*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/server/sip/SipService$ConnectivityReceiver.smali	2012-03-23 19:37:55.487786900 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/server/sip/SipService$ConnectivityReceiver.smali.original	2012-03-23 19:34:46.414339152 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1107
      iput-object p1, p0, Lcom/android/server/sip/SipService$ConnectivityReceiver;->this$0:Lcom/android/server/sip/SipService;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 1107
      invoke-direct {p0, p1}, Lcom/android/server/sip/SipService$ConnectivityReceiver;-><init>(Lcom/android/server/sip/SipService;)V
  
      return-void
***************
*** 50,63 ****
      .parameter "intent"
  
      .prologue
      invoke-virtual {p2}, Landroid/content/Intent;->getExtras()Landroid/os/Bundle;
  
      move-result-object v0
  
      .local v0, bundle:Landroid/os/Bundle;
      if-eqz v0, :cond_0
  
!     const-string v2, "networkInfo"
  
      invoke-virtual {v0, v2}, Landroid/os/Bundle;->get(Ljava/lang/String;)Ljava/lang/Object;
  
--- 52,68 ----
      .parameter "intent"
  
      .prologue
+     .line 1110
      invoke-virtual {p2}, Landroid/content/Intent;->getExtras()Landroid/os/Bundle;
  
      move-result-object v0
  
+     .line 1111
      .local v0, bundle:Landroid/os/Bundle;
      if-eqz v0, :cond_0
  
!     .line 1112
!     const-string/jumbo v2, "networkInfo"
  
      invoke-virtual {v0, v2}, Landroid/os/Bundle;->get(Ljava/lang/String;)Ljava/lang/Object;
  
***************
*** 65,70 ****
--- 70,76 ----
  
      check-cast v1, Landroid/net/NetworkInfo;
  
+     .line 1116
      .local v1, info:Landroid/net/NetworkInfo;
      iget-object v2, p0, Lcom/android/server/sip/SipService$ConnectivityReceiver;->this$0:Lcom/android/server/sip/SipService;
  
***************
*** 79,84 ****
--- 85,91 ----
  
      invoke-virtual {v2, v3}, Lcom/android/server/sip/SipService$MyExecutor;->execute(Ljava/lang/Runnable;)V
  
+     .line 1122
      .end local v1           #info:Landroid/net/NetworkInfo;
      :cond_0
      return-void
