*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/server/sip/SipSessionListenerProxy$6.smali	2012-03-23 19:37:56.060797350 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/server/sip/SipSessionListenerProxy$6.smali.original	2012-03-23 19:34:46.418339224 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .parameter
  
      .prologue
+     .line 116
      iput-object p1, p0, Lcom/android/server/sip/SipSessionListenerProxy$6;->this$0:Lcom/android/server/sip/SipSessionListenerProxy;
  
      iput-object p2, p0, Lcom/android/server/sip/SipSessionListenerProxy$6;->val$newSession:Landroid/net/sip/ISipSession;
***************
*** 50,55 ****
--- 51,57 ----
      .locals 4
  
      .prologue
+     .line 119
      :try_start_0
      iget-object v1, p0, Lcom/android/server/sip/SipSessionListenerProxy$6;->this$0:Lcom/android/server/sip/SipSessionListenerProxy;
  
***************
*** 66,81 ****
      :try_end_0
      .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
  
      :goto_0
      return-void
  
      :catch_0
      move-exception v0
  
      .local v0, t:Ljava/lang/Throwable;
      iget-object v1, p0, Lcom/android/server/sip/SipSessionListenerProxy$6;->this$0:Lcom/android/server/sip/SipSessionListenerProxy;
  
!     const-string v2, "onCallTransferring()"
  
      #calls: Lcom/android/server/sip/SipSessionListenerProxy;->handle(Ljava/lang/Throwable;Ljava/lang/String;)V
      invoke-static {v1, v0, v2}, Lcom/android/server/sip/SipSessionListenerProxy;->access$100(Lcom/android/server/sip/SipSessionListenerProxy;Ljava/lang/Throwable;Ljava/lang/String;)V
--- 68,86 ----
      :try_end_0
      .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 123
      :goto_0
      return-void
  
+     .line 120
      :catch_0
      move-exception v0
  
+     .line 121
      .local v0, t:Ljava/lang/Throwable;
      iget-object v1, p0, Lcom/android/server/sip/SipSessionListenerProxy$6;->this$0:Lcom/android/server/sip/SipSessionListenerProxy;
  
!     const-string/jumbo v2, "onCallTransferring()"
  
      #calls: Lcom/android/server/sip/SipSessionListenerProxy;->handle(Ljava/lang/Throwable;Ljava/lang/String;)V
      invoke-static {v1, v0, v2}, Lcom/android/server/sip/SipSessionListenerProxy;->access$100(Lcom/android/server/sip/SipSessionListenerProxy;Ljava/lang/Throwable;Ljava/lang/String;)V
