*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/server/sip/SipWakeupTimer.smali	2012-03-23 19:37:56.651808127 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/server/sip/SipWakeupTimer.smali.original	2012-03-23 19:34:46.423339315 -0500
***************
*** 49,56 ****
--- 49,58 ----
      .parameter "executor"
  
      .prologue
+     .line 63
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
  
+     .line 56
      new-instance v1, Ljava/util/TreeSet;
  
      new-instance v2, Lcom/android/server/sip/SipWakeupTimer$MyEventComparator;
***************
*** 63,70 ****
--- 65,74 ----
  
      iput-object v1, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
+     .line 64
      iput-object p1, p0, Lcom/android/server/sip/SipWakeupTimer;->mContext:Landroid/content/Context;
  
+     .line 65
      const-string v1, "alarm"
  
      invoke-virtual {p1, v1}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 75,80 ****
--- 79,85 ----
  
      iput-object v1, p0, Lcom/android/server/sip/SipWakeupTimer;->mAlarmManager:Landroid/app/AlarmManager;
  
+     .line 68
      new-instance v0, Landroid/content/IntentFilter;
  
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->getAction()Ljava/lang/String;
***************
*** 83,93 ****
--- 88,101 ----
  
      invoke-direct {v0, v1}, Landroid/content/IntentFilter;-><init>(Ljava/lang/String;)V
  
+     .line 69
      .local v0, filter:Landroid/content/IntentFilter;
      invoke-virtual {p1, p0, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 70
      iput-object p2, p0, Lcom/android/server/sip/SipWakeupTimer;->mExecutor:Ljava/util/concurrent/Executor;
  
+     .line 71
      return-void
  .end method
  
***************
*** 95,110 ****
--- 103,121 ----
      .locals 2
  
      .prologue
+     .line 96
      iget-object v0, p0, Lcom/android/server/sip/SipWakeupTimer;->mAlarmManager:Landroid/app/AlarmManager;
  
      iget-object v1, p0, Lcom/android/server/sip/SipWakeupTimer;->mPendingIntent:Landroid/app/PendingIntent;
  
      invoke-virtual {v0, v1}, Landroid/app/AlarmManager;->cancel(Landroid/app/PendingIntent;)V
  
+     .line 97
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/server/sip/SipWakeupTimer;->mPendingIntent:Landroid/app/PendingIntent;
  
+     .line 98
      return-void
  .end method
  
***************
*** 113,118 ****
--- 124,130 ----
      .parameter "triggerTime"
  
      .prologue
+     .line 267
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->stopped()Z
  
      move-result v2
***************
*** 127,136 ****
--- 139,150 ----
  
      if-eqz v2, :cond_1
  
+     .line 284
      :cond_0
      :goto_0
      return-void
  
+     .line 269
      :cond_1
      iget-object v2, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
***************
*** 153,158 ****
--- 167,173 ----
  
      check-cast v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
+     .line 270
      .local v0, event:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      iget-wide v2, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
***************
*** 160,167 ****
--- 175,184 ----
  
      if-nez v2, :cond_2
  
+     .line 273
      iput-wide p1, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mLastTriggerTime:J
  
+     .line 274
      iget-wide v2, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
      iget v4, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mPeriod:I
***************
*** 172,177 ****
--- 189,195 ----
  
      iput-wide v2, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
+     .line 277
      iget-object v2, p0, Lcom/android/server/sip/SipWakeupTimer;->mExecutor:Ljava/util/concurrent/Executor;
  
      iget-object v3, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mCallback:Ljava/lang/Runnable;
***************
*** 180,185 ****
--- 198,204 ----
  
      goto :goto_1
  
+     .line 283
      .end local v0           #event:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      :cond_3
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->scheduleNext()V
***************
*** 191,196 ****
--- 210,216 ----
      .locals 1
  
      .prologue
+     .line 287
      invoke-virtual {p0}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
      move-result-object v0
***************
*** 203,212 ****
--- 223,234 ----
      .parameter "event"
  
      .prologue
+     .line 127
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v3
  
+     .line 128
      .local v3, now:J
      iget-object v7, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
***************
*** 216,221 ****
--- 238,244 ----
  
      if-eqz v7, :cond_0
  
+     .line 129
      iget v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mPeriod:I
  
      int-to-long v7, v7
***************
*** 224,236 ****
--- 247,262 ----
  
      iput-wide v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
+     .line 130
      iget-object v7, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v7, p1}, Ljava/util/TreeSet;->add(Ljava/lang/Object;)Z
  
+     .line 153
      :goto_0
      return-void
  
+     .line 133
      :cond_0
      iget-object v7, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
***************
*** 240,253 ****
--- 266,282 ----
  
      check-cast v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
+     .line 134
      .local v0, firstEvent:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      iget v2, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mPeriod:I
  
+     .line 135
      .local v2, minPeriod:I
      iget v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mMaxPeriod:I
  
      if-gt v2, v7, :cond_1
  
+     .line 136
      iget v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mMaxPeriod:I
  
      div-int/2addr v7, v2
***************
*** 256,263 ****
--- 285,294 ----
  
      iput v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mPeriod:I
  
+     .line 137
      iget v1, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mMaxPeriod:I
  
+     .line 138
      .local v1, interval:I
      iget-wide v7, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
***************
*** 267,276 ****
--- 298,309 ----
  
      sub-int/2addr v1, v7
  
+     .line 139
      div-int v7, v1, v2
  
      mul-int v1, v7, v2
  
+     .line 140
      iget-wide v7, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
      int-to-long v9, v1
***************
*** 279,290 ****
--- 312,325 ----
  
      iput-wide v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
+     .line 141
      iget-object v7, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v7, p1}, Ljava/util/TreeSet;->add(Ljava/lang/Object;)Z
  
      goto :goto_0
  
+     .line 143
      .end local v1           #interval:I
      :cond_1
      iget v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mPeriod:I
***************
*** 293,298 ****
--- 328,334 ----
  
      add-long v5, v3, v7
  
+     .line 144
      .local v5, triggerTime:J
      iget-wide v7, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
***************
*** 300,309 ****
--- 336,347 ----
  
      if-gez v7, :cond_2
  
+     .line 145
      iget-wide v7, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
      iput-wide v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
+     .line 146
      iget-wide v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mLastTriggerTime:J
  
      iget v9, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mPeriod:I
***************
*** 314,328 ****
--- 352,369 ----
  
      iput-wide v7, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mLastTriggerTime:J
  
+     .line 150
      :goto_1
      iget-object v7, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v7, p1}, Ljava/util/TreeSet;->add(Ljava/lang/Object;)Z
  
+     .line 151
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->recalculatePeriods()V
  
      goto :goto_0
  
+     .line 148
      :cond_2
      iput-wide v5, p1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
***************
*** 333,340 ****
--- 374,383 ----
      .locals 7
  
      .prologue
+     .line 250
      const/4 v0, 0x0
  
+     .line 251
      .local v0, count:I
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
***************
*** 356,361 ****
--- 399,405 ----
  
      check-cast v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
+     .line 252
      .local v1, event:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      const-string v3, "_SIP.WkTimer_"
  
***************
*** 411,422 ****
--- 455,468 ----
  
      invoke-static {v3, v4}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 255
      add-int/lit8 v0, v0, 0x1
  
      const/4 v3, 0x5
  
      if-lt v0, v3, :cond_0
  
+     .line 257
      .end local v1           #event:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      :cond_1
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
***************
*** 427,445 ****
--- 473,495 ----
  
      if-le v3, v0, :cond_3
  
+     .line 258
      const-string v3, "_SIP.WkTimer_"
  
      const-string v4, "     ....."
  
      invoke-static {v3, v4}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 262
      :cond_2
      :goto_0
      return-void
  
+     .line 259
      :cond_3
      if-nez v0, :cond_2
  
+     .line 260
      const-string v3, "_SIP.WkTimer_"
  
      const-string v4, "     <empty>"
***************
*** 453,458 ****
--- 503,509 ----
      .locals 12
  
      .prologue
+     .line 101
      iget-object v8, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v8}, Ljava/util/TreeSet;->isEmpty()Z
***************
*** 461,469 ****
--- 512,522 ----
  
      if-eqz v8, :cond_0
  
+     .line 122
      :goto_0
      return-void
  
+     .line 103
      :cond_0
      iget-object v8, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
***************
*** 473,484 ****
--- 526,540 ----
  
      check-cast v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
+     .line 104
      .local v1, firstEvent:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      iget v4, v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mMaxPeriod:I
  
+     .line 105
      .local v4, minPeriod:I
      iget-wide v5, v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
+     .line 106
      .local v5, minTriggerTime:J
      iget-object v8, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
***************
*** 500,505 ****
--- 556,562 ----
  
      check-cast v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
+     .line 107
      .local v0, e:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      iget v8, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mMaxPeriod:I
  
***************
*** 509,514 ****
--- 566,572 ----
  
      iput v8, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mPeriod:I
  
+     .line 108
      iget-wide v8, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mLastTriggerTime:J
  
      iget v10, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mMaxPeriod:I
***************
*** 521,531 ****
--- 579,591 ----
  
      long-to-int v3, v8
  
+     .line 110
      .local v3, interval:I
      div-int v8, v3, v4
  
      mul-int v3, v8, v4
  
+     .line 111
      int-to-long v8, v3
  
      add-long/2addr v8, v5
***************
*** 534,539 ****
--- 594,600 ----
  
      goto :goto_1
  
+     .line 113
      .end local v0           #e:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      .end local v3           #interval:I
      :cond_1
***************
*** 547,561 ****
--- 608,625 ----
  
      invoke-direct {v7, v8}, Ljava/util/TreeSet;-><init>(Ljava/util/Comparator;)V
  
+     .line 115
      .local v7, newQueue:Ljava/util/TreeSet;,"Ljava/util/TreeSet<Lcom/android/server/sip/SipWakeupTimer$MyEvent;>;"
      iget-object v8, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v7, v8}, Ljava/util/TreeSet;->addAll(Ljava/util/Collection;)Z
  
+     .line 116
      iget-object v8, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v8}, Ljava/util/TreeSet;->clear()V
  
+     .line 117
      iput-object v7, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      goto :goto_0
***************
*** 565,570 ****
--- 629,635 ----
      .locals 7
  
      .prologue
+     .line 219
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->stopped()Z
  
      move-result v3
***************
*** 579,601 ****
  
      if-eqz v3, :cond_1
  
      :cond_0
      :goto_0
      return-void
  
      :cond_1
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mPendingIntent:Landroid/app/PendingIntent;
  
      if-eqz v3, :cond_2
  
      new-instance v3, Ljava/lang/RuntimeException;
  
!     const-string v4, "pendingIntent is not null!"
  
      invoke-direct {v3, v4}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v3
  
      :cond_2
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
--- 644,670 ----
  
      if-eqz v3, :cond_1
  
+     .line 233
      :cond_0
      :goto_0
      return-void
  
+     .line 221
      :cond_1
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mPendingIntent:Landroid/app/PendingIntent;
  
      if-eqz v3, :cond_2
  
+     .line 222
      new-instance v3, Ljava/lang/RuntimeException;
  
!     const-string/jumbo v4, "pendingIntent is not null!"
  
      invoke-direct {v3, v4}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v3
  
+     .line 225
      :cond_2
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
***************
*** 605,610 ****
--- 674,680 ----
  
      check-cast v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
+     .line 226
      .local v0, event:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      new-instance v1, Landroid/content/Intent;
  
***************
*** 614,619 ****
--- 684,690 ----
  
      invoke-direct {v1, v3}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 227
      .local v1, intent:Landroid/content/Intent;
      const-string v3, "TriggerTime"
  
***************
*** 621,626 ****
--- 692,698 ----
  
      invoke-virtual {v1, v3, v4, v5}, Landroid/content/Intent;->putExtra(Ljava/lang/String;J)Landroid/content/Intent;
  
+     .line 228
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mContext:Landroid/content/Context;
  
      const/4 v4, 0x0
***************
*** 633,638 ****
--- 705,711 ----
  
      iput-object v2, p0, Lcom/android/server/sip/SipWakeupTimer;->mPendingIntent:Landroid/app/PendingIntent;
  
+     .line 231
      .local v2, pendingIntent:Landroid/app/PendingIntent;
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mAlarmManager:Landroid/app/AlarmManager;
  
***************
*** 652,672 ****
--- 725,750 ----
      .prologue
      const-wide/16 v5, 0x3e8
  
+     .line 291
      rem-long v3, p1, v5
  
      long-to-int v1, v3
  
+     .line 292
      .local v1, ms:I
      div-long v3, p1, v5
  
      long-to-int v2, v3
  
+     .line 293
      .local v2, s:I
      div-int/lit8 v0, v2, 0x3c
  
+     .line 294
      .local v0, m:I
      rem-int/lit8 v2, v2, 0x3c
  
+     .line 295
      const-string v3, "%d.%d.%d"
  
      const/4 v4, 0x3
***************
*** 708,725 ****
--- 786,807 ----
      .locals 2
  
      .prologue
+     .line 87
      iget-object v0, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      if-nez v0, :cond_0
  
+     .line 88
      const-string v0, "_SIP.WkTimer_"
  
      const-string v1, "Timer stopped"
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 89
      const/4 v0, 0x1
  
+     .line 91
      :goto_0
      return v0
  
***************
*** 736,741 ****
--- 818,824 ----
      .parameter "callback"
  
      .prologue
+     .line 189
      monitor-enter p0
  
      :try_start_0
***************
*** 755,766 ****
--- 838,851 ----
  
      if-eqz v3, :cond_1
  
+     .line 216
      :cond_0
      :goto_0
      monitor-exit p0
  
      return-void
  
+     .line 192
      :cond_1
      :try_start_1
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
***************
*** 771,776 ****
--- 856,862 ----
  
      check-cast v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
+     .line 193
      .local v1, firstEvent:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
***************
*** 778,783 ****
--- 864,870 ----
  
      move-result-object v2
  
+     .line 194
      .local v2, iter:Ljava/util/Iterator;,"Ljava/util/Iterator<Lcom/android/server/sip/SipWakeupTimer$MyEvent;>;"
      :cond_2
      :goto_1
***************
*** 787,809 ****
--- 874,900 ----
  
      if-eqz v3, :cond_3
  
+     .line 195
      invoke-interface {v2}, Ljava/util/Iterator;->next()Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
+     .line 196
      .local v0, event:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      iget-object v3, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mCallback:Ljava/lang/Runnable;
  
      if-ne v3, p1, :cond_2
  
+     .line 197
      invoke-interface {v2}, Ljava/util/Iterator;->remove()V
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
      goto :goto_1
  
+     .line 189
      .end local v0           #event:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      .end local v1           #firstEvent:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      .end local v2           #iter:Ljava/util/Iterator;,"Ljava/util/Iterator<Lcom/android/server/sip/SipWakeupTimer$MyEvent;>;"
***************
*** 814,819 ****
--- 905,911 ----
  
      throw v3
  
+     .line 201
      .restart local v1       #firstEvent:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      .restart local v2       #iter:Ljava/util/Iterator;,"Ljava/util/Iterator<Lcom/android/server/sip/SipWakeupTimer$MyEvent;>;"
      :cond_3
***************
*** 826,835 ****
--- 918,929 ----
  
      if-eqz v3, :cond_4
  
+     .line 202
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->cancelAlarm()V
  
      goto :goto_0
  
+     .line 203
      :cond_4
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
***************
*** 839,846 ****
--- 933,942 ----
  
      if-eq v3, v1, :cond_0
  
+     .line 204
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->cancelAlarm()V
  
+     .line 205
      iget-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v3}, Ljava/util/TreeSet;->first()Ljava/lang/Object;
***************
*** 850,860 ****
--- 946,958 ----
      .end local v1           #firstEvent:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      check-cast v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
+     .line 206
      .restart local v1       #firstEvent:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      iget v3, v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mMaxPeriod:I
  
      iput v3, v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mPeriod:I
  
+     .line 207
      iget-wide v3, v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mLastTriggerTime:J
  
      iget v5, v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mPeriod:I
***************
*** 865,872 ****
--- 963,972 ----
  
      iput-wide v3, v1, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
  
+     .line 209
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->recalculatePeriods()V
  
+     .line 210
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->scheduleNext()V
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
***************
*** 880,885 ****
--- 980,986 ----
      .parameter "intent"
  
      .prologue
+     .line 238
      monitor-enter p0
  
      :try_start_0
***************
*** 887,892 ****
--- 988,994 ----
  
      move-result-object v0
  
+     .line 239
      .local v0, action:Ljava/lang/String;
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->getAction()Ljava/lang/String;
  
***************
*** 910,919 ****
--- 1012,1023 ----
  
      if-eqz v3, :cond_0
  
+     .line 241
      const/4 v3, 0x0
  
      iput-object v3, p0, Lcom/android/server/sip/SipWakeupTimer;->mPendingIntent:Landroid/app/PendingIntent;
  
+     .line 242
      const-string v3, "TriggerTime"
  
      const-wide/16 v4, -0x1
***************
*** 922,938 ****
--- 1026,1045 ----
  
      move-result-wide v1
  
+     .line 243
      .local v1, triggerTime:J
      invoke-direct {p0, v1, v2}, Lcom/android/server/sip/SipWakeupTimer;->execute(J)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 247
      .end local v1           #triggerTime:J
      :goto_0
      monitor-exit p0
  
      return-void
  
+     .line 245
      :cond_0
      :try_start_1
      const-string v3, "_SIP.WkTimer_"
***************
*** 941,947 ****
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v5, "unrecognized intent: "
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1048,1054 ----
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v5, "unrecognized intent: "
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 961,966 ****
--- 1068,1074 ----
  
      goto :goto_0
  
+     .line 238
      .end local v0           #action:Ljava/lang/String;
      :catchall_0
      move-exception v3
***************
*** 976,981 ****
--- 1084,1090 ----
      .parameter "callback"
  
      .prologue
+     .line 163
      monitor-enter p0
  
      :try_start_0
***************
*** 987,1011 ****
--- 1096,1125 ----
  
      if-eqz v5, :cond_0
  
+     .line 181
      :goto_0
      monitor-exit p0
  
      return-void
  
+     .line 165
      :cond_0
      :try_start_1
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v1
  
+     .line 166
      .local v1, now:J
      new-instance v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;
  
      invoke-direct {v0, p1, p2, v1, v2}, Lcom/android/server/sip/SipWakeupTimer$MyEvent;-><init>(ILjava/lang/Runnable;J)V
  
+     .line 167
      .local v0, event:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      invoke-direct {p0, v0}, Lcom/android/server/sip/SipWakeupTimer;->insertEvent(Lcom/android/server/sip/SipWakeupTimer$MyEvent;)V
  
+     .line 169
      iget-object v5, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v5}, Ljava/util/TreeSet;->first()Ljava/lang/Object;
***************
*** 1014,1019 ****
--- 1128,1134 ----
  
      if-ne v5, v0, :cond_2
  
+     .line 170
      iget-object v5, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v5}, Ljava/util/TreeSet;->size()I
***************
*** 1026,1042 ****
--- 1141,1161 ----
  
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->cancelAlarm()V
  
+     .line 171
      :cond_1
      invoke-direct {p0}, Lcom/android/server/sip/SipWakeupTimer;->scheduleNext()V
  
+     .line 174
      :cond_2
      iget-wide v3, v0, Lcom/android/server/sip/SipWakeupTimer$MyEvent;->mTriggerTime:J
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 181
      .local v3, triggerTime:J
      goto :goto_0
  
+     .line 163
      .end local v0           #event:Lcom/android/server/sip/SipWakeupTimer$MyEvent;
      .end local v1           #now:J
      .end local v3           #triggerTime:J
***************
*** 1052,1057 ****
--- 1171,1177 ----
      .locals 2
  
      .prologue
+     .line 77
      monitor-enter p0
  
      :try_start_0
***************
*** 1059,1093 ****
--- 1179,1220 ----
  
      invoke-virtual {v0, p0}, Landroid/content/Context;->unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  
+     .line 78
      iget-object v0, p0, Lcom/android/server/sip/SipWakeupTimer;->mPendingIntent:Landroid/app/PendingIntent;
  
      if-eqz v0, :cond_0
  
+     .line 79
      iget-object v0, p0, Lcom/android/server/sip/SipWakeupTimer;->mAlarmManager:Landroid/app/AlarmManager;
  
      iget-object v1, p0, Lcom/android/server/sip/SipWakeupTimer;->mPendingIntent:Landroid/app/PendingIntent;
  
      invoke-virtual {v0, v1}, Landroid/app/AlarmManager;->cancel(Landroid/app/PendingIntent;)V
  
+     .line 80
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/server/sip/SipWakeupTimer;->mPendingIntent:Landroid/app/PendingIntent;
  
+     .line 82
      :cond_0
      iget-object v0, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
  
      invoke-virtual {v0}, Ljava/util/TreeSet;->clear()V
  
+     .line 83
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/server/sip/SipWakeupTimer;->mEventQueue:Ljava/util/TreeSet;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 84
      monitor-exit p0
  
      return-void
  
+     .line 77
      :catchall_0
      move-exception v0
  
