*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/AsyncTask$Status.smali	2012-03-23 19:36:47.020538403 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/AsyncTask$Status.smali.original	2012-03-23 19:34:45.343319621 -0500
***************
*** 44,49 ****
--- 44,50 ----
  
      const/4 v2, 0x0
  
+     .line 234
      new-instance v0, Landroid/os/AsyncTask$Status;
  
      const-string v1, "PENDING"
***************
*** 52,57 ****
--- 53,59 ----
  
      sput-object v0, Landroid/os/AsyncTask$Status;->PENDING:Landroid/os/AsyncTask$Status;
  
+     .line 238
      new-instance v0, Landroid/os/AsyncTask$Status;
  
      const-string v1, "RUNNING"
***************
*** 60,65 ****
--- 62,68 ----
  
      sput-object v0, Landroid/os/AsyncTask$Status;->RUNNING:Landroid/os/AsyncTask$Status;
  
+     .line 242
      new-instance v0, Landroid/os/AsyncTask$Status;
  
      const-string v1, "FINISHED"
***************
*** 68,73 ****
--- 71,77 ----
  
      sput-object v0, Landroid/os/AsyncTask$Status;->FINISHED:Landroid/os/AsyncTask$Status;
  
+     .line 230
      const/4 v0, 0x3
  
      new-array v0, v0, [Landroid/os/AsyncTask$Status;
***************
*** 100,105 ****
--- 104,110 ----
      .end annotation
  
      .prologue
+     .line 230
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 110,115 ****
--- 115,121 ----
      .parameter "name"
  
      .prologue
+     .line 230
      const-class v0, Landroid/os/AsyncTask$Status;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 125,130 ****
--- 131,137 ----
      .locals 1
  
      .prologue
+     .line 230
      sget-object v0, Landroid/os/AsyncTask$Status;->$VALUES:[Landroid/os/AsyncTask$Status;
  
      invoke-virtual {v0}, [Landroid/os/AsyncTask$Status;->clone()Ljava/lang/Object;
