*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/AsyncTask.smali	2012-03-23 19:36:51.915627665 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/AsyncTask.smali.original	2012-03-23 19:34:45.371320132 -0500
***************
*** 95,106 ****
--- 95,108 ----
      .prologue
      const/4 v8, 0x0
  
+     .line 163
      new-instance v0, Landroid/os/AsyncTask$1;
  
      invoke-direct {v0}, Landroid/os/AsyncTask$1;-><init>()V
  
      sput-object v0, Landroid/os/AsyncTask;->sThreadFactory:Ljava/util/concurrent/ThreadFactory;
  
+     .line 171
      new-instance v0, Ljava/util/concurrent/LinkedBlockingQueue;
  
      const/16 v1, 0xa
***************
*** 109,114 ****
--- 111,117 ----
  
      sput-object v0, Landroid/os/AsyncTask;->sPoolWorkQueue:Ljava/util/concurrent/BlockingQueue;
  
+     .line 177
      new-instance v0, Ljava/util/concurrent/ThreadPoolExecutor;
  
      const/4 v1, 0x5
***************
*** 127,144 ****
--- 130,150 ----
  
      sput-object v0, Landroid/os/AsyncTask;->THREAD_POOL_EXECUTOR:Ljava/util/concurrent/Executor;
  
+     .line 185
      new-instance v0, Landroid/os/AsyncTask$SerialExecutor;
  
      invoke-direct {v0, v8}, Landroid/os/AsyncTask$SerialExecutor;-><init>(Landroid/os/AsyncTask$1;)V
  
      sput-object v0, Landroid/os/AsyncTask;->SERIAL_EXECUTOR:Ljava/util/concurrent/Executor;
  
+     .line 190
      new-instance v0, Landroid/os/AsyncTask$InternalHandler;
  
      invoke-direct {v0, v8}, Landroid/os/AsyncTask$InternalHandler;-><init>(Landroid/os/AsyncTask$1;)V
  
      sput-object v0, Landroid/os/AsyncTask;->sHandler:Landroid/os/AsyncTask$InternalHandler;
  
+     .line 192
      sget-object v0, Landroid/os/AsyncTask;->SERIAL_EXECUTOR:Ljava/util/concurrent/Executor;
  
      sput-object v0, Landroid/os/AsyncTask;->sDefaultExecutor:Ljava/util/concurrent/Executor;
***************
*** 150,174 ****
--- 156,185 ----
      .locals 2
  
      .prologue
+     .line 258
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 196
      sget-object v0, Landroid/os/AsyncTask$Status;->PENDING:Landroid/os/AsyncTask$Status;
  
      iput-object v0, p0, Landroid/os/AsyncTask;->mStatus:Landroid/os/AsyncTask$Status;
  
+     .line 198
      new-instance v0, Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-direct {v0}, Ljava/util/concurrent/atomic/AtomicBoolean;-><init>()V
  
      iput-object v0, p0, Landroid/os/AsyncTask;->mTaskInvoked:Ljava/util/concurrent/atomic/AtomicBoolean;
  
+     .line 259
      new-instance v0, Landroid/os/AsyncTask$2;
  
      invoke-direct {v0, p0}, Landroid/os/AsyncTask$2;-><init>(Landroid/os/AsyncTask;)V
  
      iput-object v0, p0, Landroid/os/AsyncTask;->mWorker:Landroid/os/AsyncTask$WorkerRunnable;
  
+     .line 268
      new-instance v0, Landroid/os/AsyncTask$3;
  
      iget-object v1, p0, Landroid/os/AsyncTask;->mWorker:Landroid/os/AsyncTask$WorkerRunnable;
***************
*** 177,182 ****
--- 188,194 ----
  
      iput-object v0, p0, Landroid/os/AsyncTask;->mFuture:Ljava/util/concurrent/FutureTask;
  
+     .line 288
      return-void
  .end method
  
***************
*** 185,190 ****
--- 197,203 ----
      .parameter "x0"
  
      .prologue
+     .line 156
      iget-object v0, p0, Landroid/os/AsyncTask;->mTaskInvoked:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      return-object v0
***************
*** 196,201 ****
--- 209,215 ----
      .parameter "x1"
  
      .prologue
+     .line 156
      invoke-direct {p0, p1}, Landroid/os/AsyncTask;->postResult(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
***************
*** 209,214 ****
--- 223,229 ----
      .parameter "x1"
  
      .prologue
+     .line 156
      invoke-direct {p0, p1}, Landroid/os/AsyncTask;->postResultIfNotInvoked(Ljava/lang/Object;)V
  
      return-void
***************
*** 220,225 ****
--- 235,241 ----
      .parameter "x1"
  
      .prologue
+     .line 156
      invoke-direct {p0, p1}, Landroid/os/AsyncTask;->finish(Ljava/lang/Object;)V
  
      return-void
***************
*** 230,239 ****
--- 246,257 ----
      .parameter "runnable"
  
      .prologue
+     .line 574
      sget-object v0, Landroid/os/AsyncTask;->sDefaultExecutor:Ljava/util/concurrent/Executor;
  
      invoke-interface {v0, p0}, Ljava/util/concurrent/Executor;->execute(Ljava/lang/Runnable;)V
  
+     .line 575
      return-void
  .end method
  
***************
*** 247,252 ****
--- 265,271 ----
      .end annotation
  
      .prologue
+     .line 599
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      .local p1, result:Ljava/lang/Object;,"TResult;"
      invoke-virtual {p0}, Landroid/os/AsyncTask;->isCancelled()Z
***************
*** 255,269 ****
--- 274,292 ----
  
      if-eqz v0, :cond_0
  
+     .line 600
      invoke-virtual {p0, p1}, Landroid/os/AsyncTask;->onCancelled(Ljava/lang/Object;)V
  
+     .line 604
      :goto_0
      sget-object v0, Landroid/os/AsyncTask$Status;->FINISHED:Landroid/os/AsyncTask$Status;
  
      iput-object v0, p0, Landroid/os/AsyncTask;->mStatus:Landroid/os/AsyncTask$Status;
  
+     .line 605
      return-void
  
+     .line 602
      :cond_0
      invoke-virtual {p0, p1}, Landroid/os/AsyncTask;->onPostExecute(Ljava/lang/Object;)V
  
***************
*** 274,283 ****
--- 297,308 ----
      .locals 1
  
      .prologue
+     .line 247
      sget-object v0, Landroid/os/AsyncTask;->sHandler:Landroid/os/AsyncTask$InternalHandler;
  
      invoke-virtual {v0}, Landroid/os/AsyncTask$InternalHandler;->getLooper()Landroid/os/Looper;
  
+     .line 248
      return-void
  .end method
  
***************
*** 295,300 ****
--- 320,326 ----
      .local p1, result:Ljava/lang/Object;,"TResult;"
      const/4 v5, 0x1
  
+     .line 298
      sget-object v1, Landroid/os/AsyncTask;->sHandler:Landroid/os/AsyncTask$InternalHandler;
  
      new-instance v2, Landroid/os/AsyncTask$AsyncTaskResult;
***************
*** 311,319 ****
--- 337,347 ----
  
      move-result-object v0
  
+     .line 300
      .local v0, message:Landroid/os/Message;
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 301
      return-object p1
  .end method
  
***************
*** 327,332 ****
--- 355,361 ----
      .end annotation
  
      .prologue
+     .line 291
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      .local p1, result:Ljava/lang/Object;,"TResult;"
      iget-object v1, p0, Landroid/os/AsyncTask;->mTaskInvoked:Ljava/util/concurrent/atomic/AtomicBoolean;
***************
*** 335,345 ****
--- 364,377 ----
  
      move-result v0
  
+     .line 292
      .local v0, wasTaskInvoked:Z
      if-nez v0, :cond_0
  
+     .line 293
      invoke-direct {p0, p1}, Landroid/os/AsyncTask;->postResult(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 295
      :cond_0
      return-void
  .end method
***************
*** 349,356 ****
--- 381,390 ----
      .parameter "exec"
  
      .prologue
+     .line 252
      sput-object p0, Landroid/os/AsyncTask;->sDefaultExecutor:Ljava/util/concurrent/Executor;
  
+     .line 253
      return-void
  .end method
  
***************
*** 361,366 ****
--- 395,401 ----
      .parameter "mayInterruptIfRunning"
  
      .prologue
+     .line 447
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      iget-object v0, p0, Landroid/os/AsyncTask;->mFuture:Ljava/util/concurrent/FutureTask;
  
***************
*** 391,396 ****
--- 426,432 ----
      .end annotation
  
      .prologue
+     .line 511
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      .local p1, params:[Ljava/lang/Object;,"[TParams;"
      sget-object v0, Landroid/os/AsyncTask;->sDefaultExecutor:Ljava/util/concurrent/Executor;
***************
*** 417,422 ****
--- 453,459 ----
      .end annotation
  
      .prologue
+     .line 547
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      .local p2, params:[Ljava/lang/Object;,"[TParams;"
      iget-object v0, p0, Landroid/os/AsyncTask;->mStatus:Landroid/os/AsyncTask$Status;
***************
*** 425,430 ****
--- 462,468 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 548
      sget-object v0, Landroid/os/AsyncTask$4;->$SwitchMap$android$os$AsyncTask$Status:[I
  
      iget-object v1, p0, Landroid/os/AsyncTask;->mStatus:Landroid/os/AsyncTask$Status;
***************
*** 437,459 ****
--- 475,503 ----
  
      packed-switch v0, :pswitch_data_0
  
+     .line 559
      :cond_0
      sget-object v0, Landroid/os/AsyncTask$Status;->RUNNING:Landroid/os/AsyncTask$Status;
  
      iput-object v0, p0, Landroid/os/AsyncTask;->mStatus:Landroid/os/AsyncTask$Status;
  
+     .line 561
      invoke-virtual {p0}, Landroid/os/AsyncTask;->onPreExecute()V
  
+     .line 563
      iget-object v0, p0, Landroid/os/AsyncTask;->mWorker:Landroid/os/AsyncTask$WorkerRunnable;
  
      iput-object p2, v0, Landroid/os/AsyncTask$WorkerRunnable;->mParams:[Ljava/lang/Object;
  
+     .line 564
      iget-object v0, p0, Landroid/os/AsyncTask;->mFuture:Ljava/util/concurrent/FutureTask;
  
      invoke-interface {p1, v0}, Ljava/util/concurrent/Executor;->execute(Ljava/lang/Runnable;)V
  
+     .line 566
      return-object p0
  
+     .line 550
      :pswitch_0
      new-instance v0, Ljava/lang/IllegalStateException;
  
***************
*** 463,468 ****
--- 507,513 ----
  
      throw v0
  
+     .line 553
      :pswitch_1
      new-instance v0, Ljava/lang/IllegalStateException;
  
***************
*** 472,477 ****
--- 517,523 ----
  
      throw v0
  
+     .line 548
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 495,500 ****
--- 541,547 ----
      .end annotation
  
      .prologue
+     .line 462
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      iget-object v0, p0, Landroid/os/AsyncTask;->mFuture:Ljava/util/concurrent/FutureTask;
  
***************
*** 526,531 ****
--- 573,579 ----
      .end annotation
  
      .prologue
+     .line 482
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      iget-object v0, p0, Landroid/os/AsyncTask;->mFuture:Ljava/util/concurrent/FutureTask;
  
***************
*** 540,545 ****
--- 588,594 ----
      .locals 1
  
      .prologue
+     .line 310
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      iget-object v0, p0, Landroid/os/AsyncTask;->mStatus:Landroid/os/AsyncTask$Status;
  
***************
*** 550,555 ****
--- 599,605 ----
      .locals 1
  
      .prologue
+     .line 414
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      iget-object v0, p0, Landroid/os/AsyncTask;->mFuture:Ljava/util/concurrent/FutureTask;
  
***************
*** 564,569 ****
--- 614,620 ----
      .locals 0
  
      .prologue
+     .line 401
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      return-void
  .end method
***************
*** 578,587 ****
--- 629,640 ----
      .end annotation
  
      .prologue
+     .line 385
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      .local p1, result:Ljava/lang/Object;,"TResult;"
      invoke-virtual {p0}, Landroid/os/AsyncTask;->onCancelled()V
  
+     .line 386
      return-void
  .end method
  
***************
*** 595,600 ****
--- 648,654 ----
      .end annotation
  
      .prologue
+     .line 354
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      .local p1, result:Ljava/lang/Object;,"TResult;"
      return-void
***************
*** 604,609 ****
--- 658,664 ----
      .locals 0
  
      .prologue
+     .line 338
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      return-void
  .end method
***************
*** 618,623 ****
--- 673,679 ----
      .end annotation
  
      .prologue
+     .line 367
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      .local p1, values:[Ljava/lang/Object;,"[TProgress;"
      return-void
***************
*** 633,638 ****
--- 689,695 ----
      .end annotation
  
      .prologue
+     .line 592
      .local p0, this:Landroid/os/AsyncTask;,"Landroid/os/AsyncTask<TParams;TProgress;TResult;>;"
      .local p1, values:[Ljava/lang/Object;,"[TProgress;"
      invoke-virtual {p0}, Landroid/os/AsyncTask;->isCancelled()Z
***************
*** 641,646 ****
--- 698,704 ----
  
      if-nez v0, :cond_0
  
+     .line 593
      sget-object v0, Landroid/os/AsyncTask;->sHandler:Landroid/os/AsyncTask$InternalHandler;
  
      const/4 v1, 0x2
***************
*** 655,660 ****
--- 713,719 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 596
      :cond_0
      return-void
  .end method
