*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/AsyncTask$SerialExecutor$1.smali	2012-03-23 19:36:46.742533333 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/AsyncTask$SerialExecutor$1.smali.original	2012-03-23 19:34:45.342319603 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 205
      iput-object p1, p0, Landroid/os/AsyncTask$SerialExecutor$1;->this$0:Landroid/os/AsyncTask$SerialExecutor;
  
      iput-object p2, p0, Landroid/os/AsyncTask$SerialExecutor$1;->val$r:Ljava/lang/Runnable;
***************
*** 45,50 ****
--- 46,52 ----
      .locals 2
  
      .prologue
+     .line 208
      :try_start_0
      iget-object v0, p0, Landroid/os/AsyncTask$SerialExecutor$1;->val$r:Ljava/lang/Runnable;
  
***************
*** 52,63 ****
--- 54,68 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 210
      iget-object v0, p0, Landroid/os/AsyncTask$SerialExecutor$1;->this$0:Landroid/os/AsyncTask$SerialExecutor;
  
      invoke-virtual {v0}, Landroid/os/AsyncTask$SerialExecutor;->scheduleNext()V
  
+     .line 212
      return-void
  
+     .line 210
      :catchall_0
      move-exception v0
  
