*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/BatteryStats$Uid.smali	2012-03-23 19:36:50.633604287 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/BatteryStats$Uid.smali.original	2012-03-23 19:34:45.360319931 -0500
***************
*** 35,47 ****
      .locals 3
  
      .prologue
      const/4 v0, 0x7
  
      new-array v0, v0, [Ljava/lang/String;
  
      const/4 v1, 0x0
  
!     const-string v2, "other"
  
      aput-object v2, v0, v1
  
--- 35,48 ----
      .locals 3
  
      .prologue
+     .line 282
      const/4 v0, 0x7
  
      new-array v0, v0, [Ljava/lang/String;
  
      const/4 v1, 0x0
  
!     const-string/jumbo v2, "other"
  
      aput-object v2, v0, v1
  
***************
*** 53,59 ****
  
      const/4 v1, 0x2
  
!     const-string v2, "touch"
  
      aput-object v2, v0, v1
  
--- 54,60 ----
  
      const/4 v1, 0x2
  
!     const-string/jumbo v2, "touch"
  
      aput-object v2, v0, v1
  
***************
*** 65,71 ****
  
      const/4 v1, 0x4
  
!     const-string v2, "touch_up"
  
      aput-object v2, v0, v1
  
--- 66,72 ----
  
      const/4 v1, 0x4
  
!     const-string/jumbo v2, "touch_up"
  
      aput-object v2, v0, v1
  
***************
*** 77,83 ****
  
      const/4 v1, 0x6
  
!     const-string v2, "unknown"
  
      aput-object v2, v0, v1
  
--- 78,84 ----
  
      const/4 v1, 0x6
  
!     const-string/jumbo v2, "unknown"
  
      aput-object v2, v0, v1
  
***************
*** 90,97 ****
--- 91,100 ----
      .locals 0
  
      .prologue
+     .line 202
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 365
      return-void
  .end method
  
