*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/BatteryStats$Uid$Wakelock.smali	2012-03-23 19:36:47.056539059 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/BatteryStats$Uid$Wakelock.smali.original	2012-03-23 19:34:45.343319621 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 0
  
      .prologue
+     .line 214
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
