*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Binder$1.smali	2012-03-23 19:36:50.842608098 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Binder$1.smali.original	2012-03-23 19:34:45.362319967 -0500
***************
*** 34,39 ****
--- 34,40 ----
      .parameter
  
      .prologue
+     .line 264
      iput-object p1, p0, Landroid/os/Binder$1;->this$0:Landroid/os/Binder;
  
      iput-object p3, p0, Landroid/os/Binder$1;->val$fd:Ljava/io/FileDescriptor;
***************
*** 53,58 ****
--- 54,60 ----
      .locals 4
  
      .prologue
+     .line 267
      :try_start_0
      iget-object v0, p0, Landroid/os/Binder$1;->this$0:Landroid/os/Binder;
  
***************
*** 66,77 ****
--- 68,82 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 269
      iget-object v0, p0, Landroid/os/Binder$1;->val$pw:Ljava/io/PrintWriter;
  
      invoke-virtual {v0}, Ljava/io/PrintWriter;->flush()V
  
+     .line 271
      return-void
  
+     .line 269
      :catchall_0
      move-exception v0
  
