*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Build$VERSION_CODES.smali	2012-03-23 19:36:44.649495166 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Build$VERSION_CODES.smali.original	2012-03-23 19:34:45.327319329 -0500
***************
*** 53,58 ****
--- 53,59 ----
      .locals 0
  
      .prologue
+     .line 129
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
