*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Debug$InstructionCount.smali	2012-03-23 19:36:51.793625440 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Debug$InstructionCount.smali.original	2012-03-23 19:34:45.370320113 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .locals 1
  
      .prologue
+     .line 1005
      sget v0, Ldalvik/bytecode/OpcodeInfo;->MAXIMUM_PACKED_VALUE:I
  
      add-int/lit8 v0, v0, 0x1
***************
*** 40,53 ****
--- 41,57 ----
      .locals 1
  
      .prologue
+     .line 1010
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1011
      sget v0, Landroid/os/Debug$InstructionCount;->NUM_INSTR:I
  
      new-array v0, v0, [I
  
      iput-object v0, p0, Landroid/os/Debug$InstructionCount;->mCounts:[I
  
+     .line 1012
      return-void
  .end method
  
***************
*** 57,79 ****
--- 61,88 ----
      .locals 2
  
      .prologue
+     .line 1036
      :try_start_0
      invoke-static {}, Ldalvik/system/VMDebug;->stopInstructionCounting()V
  
+     .line 1037
      iget-object v1, p0, Landroid/os/Debug$InstructionCount;->mCounts:[I
  
      invoke-static {v1}, Ldalvik/system/VMDebug;->getInstructionCount([I)V
      :try_end_0
      .catch Ljava/lang/UnsupportedOperationException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1041
      const/4 v1, 0x1
  
      :goto_0
      return v1
  
+     .line 1038
      :catch_0
      move-exception v0
  
+     .line 1039
      .local v0, uoe:Ljava/lang/UnsupportedOperationException;
      const/4 v1, 0x0
  
***************
*** 84,91 ****
--- 93,102 ----
      .locals 3
  
      .prologue
+     .line 1063
      const/4 v0, 0x0
  
+     .line 1065
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 95,117 ****
--- 106,132 ----
  
      if-ge v1, v2, :cond_1
  
+     .line 1066
      invoke-static {v1}, Ldalvik/bytecode/OpcodeInfo;->isInvoke(I)Z
  
      move-result v2
  
      if-eqz v2, :cond_0
  
+     .line 1067
      iget-object v2, p0, Landroid/os/Debug$InstructionCount;->mCounts:[I
  
      aget v2, v2, v1
  
      add-int/2addr v0, v2
  
+     .line 1065
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 1071
      :cond_1
      return v0
  .end method
***************
*** 120,127 ****
--- 135,144 ----
      .locals 3
  
      .prologue
+     .line 1049
      const/4 v0, 0x0
  
+     .line 1051
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 131,146 ****
--- 148,166 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 1052
      iget-object v2, p0, Landroid/os/Debug$InstructionCount;->mCounts:[I
  
      aget v2, v2, v1
  
      add-int/2addr v0, v2
  
+     .line 1051
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 1055
      :cond_0
      return v0
  .end method
***************
*** 149,169 ****
--- 169,194 ----
      .locals 2
  
      .prologue
+     .line 1022
      :try_start_0
      invoke-static {}, Ldalvik/system/VMDebug;->startInstructionCounting()V
  
+     .line 1023
      invoke-static {}, Ldalvik/system/VMDebug;->resetInstructionCount()V
      :try_end_0
      .catch Ljava/lang/UnsupportedOperationException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1027
      const/4 v1, 0x1
  
      :goto_0
      return v1
  
+     .line 1024
      :catch_0
      move-exception v0
  
+     .line 1025
      .local v0, uoe:Ljava/lang/UnsupportedOperationException;
      const/4 v1, 0x0
  
