*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/DropBoxManager$Entry$1.smali	2012-03-23 19:36:44.628494784 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/DropBoxManager$Entry$1.smali.original	2012-03-23 19:34:45.327319329 -0500
***************
*** 22,27 ****
--- 22,28 ----
      .locals 0
  
      .prologue
+     .line 222
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 34,58 ****
--- 35,64 ----
      .parameter "in"
  
      .prologue
+     .line 225
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 226
      .local v1, tag:Ljava/lang/String;
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v2
  
+     .line 227
      .local v2, millis:J
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
  
+     .line 228
      .local v5, flags:I
      and-int/lit8 v0, v5, 0x8
  
      if-eqz v0, :cond_0
  
+     .line 229
      new-instance v0, Landroid/os/DropBoxManager$Entry;
  
      invoke-virtual {p1}, Landroid/os/Parcel;->createByteArray()[B
***************
*** 63,68 ****
--- 69,75 ----
  
      invoke-direct/range {v0 .. v5}, Landroid/os/DropBoxManager$Entry;-><init>(Ljava/lang/String;J[BI)V
  
+     .line 231
      .end local v5           #flags:I
      :goto_0
      return-object v0
***************
*** 85,90 ****
--- 92,98 ----
      .parameter "x0"
  
      .prologue
+     .line 222
      invoke-virtual {p0, p1}, Landroid/os/DropBoxManager$Entry$1;->createFromParcel(Landroid/os/Parcel;)Landroid/os/DropBoxManager$Entry;
  
      move-result-object v0
***************
*** 97,102 ****
--- 105,111 ----
      .parameter "size"
  
      .prologue
+     .line 223
      new-array v0, p1, [Landroid/os/DropBoxManager$Entry;
  
      return-object v0
***************
*** 107,112 ****
--- 116,122 ----
      .parameter "x0"
  
      .prologue
+     .line 222
      invoke-virtual {p0, p1}, Landroid/os/DropBoxManager$Entry$1;->newArray(I)[Landroid/os/DropBoxManager$Entry;
  
      move-result-object v0
