*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Handler.smali	2012-03-23 19:36:50.774606858 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Handler.smali.original	2012-03-23 19:34:45.361319949 -0500
***************
*** 34,51 ****
--- 34,55 ----
      .locals 2
  
      .prologue
+     .line 109
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 119
      invoke-static {}, Landroid/os/Looper;->myLooper()Landroid/os/Looper;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
+     .line 120
      iget-object v0, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
      if-nez v0, :cond_0
  
+     .line 121
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "Can\'t create handler inside thread that has not called Looper.prepare()"
***************
*** 54,59 ****
--- 58,64 ----
  
      throw v0
  
+     .line 124
      :cond_0
      iget-object v0, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
***************
*** 61,70 ****
--- 66,77 ----
  
      iput-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
+     .line 125
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/os/Handler;->mCallback:Landroid/os/Handler$Callback;
  
+     .line 126
      return-void
  .end method
  
***************
*** 73,90 ****
--- 80,101 ----
      .parameter "callback"
  
      .prologue
+     .line 133
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 143
      invoke-static {}, Landroid/os/Looper;->myLooper()Landroid/os/Looper;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
+     .line 144
      iget-object v0, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
      if-nez v0, :cond_0
  
+     .line 145
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "Can\'t create handler inside thread that has not called Looper.prepare()"
***************
*** 93,98 ****
--- 104,110 ----
  
      throw v0
  
+     .line 148
      :cond_0
      iget-object v0, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
***************
*** 100,107 ****
--- 112,121 ----
  
      iput-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
+     .line 149
      iput-object p1, p0, Landroid/os/Handler;->mCallback:Landroid/os/Handler$Callback;
  
+     .line 150
      return-void
  .end method
  
***************
*** 110,127 ****
--- 124,146 ----
      .parameter "looper"
  
      .prologue
+     .line 155
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 156
      iput-object p1, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
+     .line 157
      iget-object v0, p1, Landroid/os/Looper;->mQueue:Landroid/os/MessageQueue;
  
      iput-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
+     .line 158
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/os/Handler;->mCallback:Landroid/os/Handler$Callback;
  
+     .line 159
      return-void
  .end method
  
***************
*** 131,146 ****
--- 150,170 ----
      .parameter "callback"
  
      .prologue
+     .line 165
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 166
      iput-object p1, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
+     .line 167
      iget-object v0, p1, Landroid/os/Looper;->mQueue:Landroid/os/MessageQueue;
  
      iput-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
+     .line 168
      iput-object p2, p0, Landroid/os/Handler;->mCallback:Landroid/os/Handler$Callback;
  
+     .line 169
      return-void
  .end method
  
***************
*** 149,161 ****
--- 173,188 ----
      .parameter "r"
  
      .prologue
+     .line 592
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 593
      .local v0, m:Landroid/os/Message;
      iput-object p1, v0, Landroid/os/Message;->callback:Ljava/lang/Runnable;
  
+     .line 594
      return-object v0
  .end method
  
***************
*** 165,179 ****
--- 192,210 ----
      .parameter "token"
  
      .prologue
+     .line 598
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 599
      .local v0, m:Landroid/os/Message;
      iput-object p2, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 600
      iput-object p1, v0, Landroid/os/Message;->callback:Ljava/lang/Runnable;
  
+     .line 601
      return-object v0
  .end method
  
***************
*** 182,191 ****
--- 213,224 ----
      .parameter "message"
  
      .prologue
+     .line 605
      iget-object v0, p1, Landroid/os/Message;->callback:Ljava/lang/Runnable;
  
      invoke-interface {v0}, Ljava/lang/Runnable;->run()V
  
+     .line 606
      return-void
  .end method
  
***************
*** 196,216 ****
--- 229,254 ----
      .parameter "msg"
  
      .prologue
+     .line 91
      iget-object v0, p1, Landroid/os/Message;->callback:Ljava/lang/Runnable;
  
      if-eqz v0, :cond_1
  
+     .line 92
      invoke-direct {p0, p1}, Landroid/os/Handler;->handleCallback(Landroid/os/Message;)V
  
+     .line 101
      :cond_0
      :goto_0
      return-void
  
+     .line 94
      :cond_1
      iget-object v0, p0, Landroid/os/Handler;->mCallback:Landroid/os/Handler$Callback;
  
      if-eqz v0, :cond_2
  
+     .line 95
      iget-object v0, p0, Landroid/os/Handler;->mCallback:Landroid/os/Handler$Callback;
  
      invoke-interface {v0, p1}, Landroid/os/Handler$Callback;->handleMessage(Landroid/os/Message;)Z
***************
*** 219,224 ****
--- 257,263 ----
  
      if-nez v0, :cond_0
  
+     .line 99
      :cond_2
      invoke-virtual {p0, p1}, Landroid/os/Handler;->handleMessage(Landroid/os/Message;)V
  
***************
*** 231,236 ****
--- 270,276 ----
      .parameter "prefix"
  
      .prologue
+     .line 560
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 263,272 ****
--- 303,314 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 561
      iget-object v0, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
      if-nez v0, :cond_0
  
+     .line 562
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 287,295 ****
--- 329,339 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 566
      :goto_0
      return-void
  
+     .line 564
      :cond_0
      iget-object v0, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
***************
*** 320,341 ****
--- 364,390 ----
      .locals 3
  
      .prologue
+     .line 576
      iget-object v1, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
      monitor-enter v1
  
+     .line 577
      :try_start_0
      iget-object v0, p0, Landroid/os/Handler;->mMessenger:Landroid/os/IMessenger;
  
      if-eqz v0, :cond_0
  
+     .line 578
      iget-object v0, p0, Landroid/os/Handler;->mMessenger:Landroid/os/IMessenger;
  
      monitor-exit v1
  
+     .line 581
      :goto_0
      return-object v0
  
+     .line 580
      :cond_0
      new-instance v0, Landroid/os/Handler$MessengerImpl;
  
***************
*** 345,356 ****
--- 394,407 ----
  
      iput-object v0, p0, Landroid/os/Handler;->mMessenger:Landroid/os/IMessenger;
  
+     .line 581
      iget-object v0, p0, Landroid/os/Handler;->mMessenger:Landroid/os/IMessenger;
  
      monitor-exit v1
  
      goto :goto_0
  
+     .line 582
      :catchall_0
      move-exception v0
  
***************
*** 365,370 ****
--- 416,422 ----
      .locals 1
  
      .prologue
+     .line 556
      iget-object v0, p0, Landroid/os/Handler;->mLooper:Landroid/os/Looper;
  
      return-object v0
***************
*** 375,384 ****
--- 427,438 ----
      .parameter "message"
  
      .prologue
+     .line 180
      iget-object v0, p1, Landroid/os/Message;->callback:Ljava/lang/Runnable;
  
      if-eqz v0, :cond_0
  
+     .line 181
      iget-object v0, p1, Landroid/os/Message;->callback:Ljava/lang/Runnable;
  
      invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
***************
*** 389,394 ****
--- 443,449 ----
  
      move-result-object v0
  
+     .line 183
      :goto_0
      return-object v0
  
***************
*** 425,430 ****
--- 480,486 ----
      .parameter "msg"
  
      .prologue
+     .line 85
      return-void
  .end method
  
***************
*** 433,438 ****
--- 489,495 ----
      .parameter "what"
  
      .prologue
+     .line 542
      iget-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
      const/4 v1, 0x0
***************
*** 452,457 ****
--- 509,515 ----
      .parameter "object"
  
      .prologue
+     .line 550
      iget-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
      const/4 v1, 0x0
***************
*** 467,472 ****
--- 525,531 ----
      .locals 1
  
      .prologue
+     .line 193
      invoke-static {p0}, Landroid/os/Message;->obtain(Landroid/os/Handler;)Landroid/os/Message;
  
      move-result-object v0
***************
*** 479,484 ****
--- 538,544 ----
      .parameter "what"
  
      .prologue
+     .line 204
      invoke-static {p0, p1}, Landroid/os/Message;->obtain(Landroid/os/Handler;I)Landroid/os/Message;
  
      move-result-object v0
***************
*** 493,498 ****
--- 553,559 ----
      .parameter "arg2"
  
      .prologue
+     .line 232
      invoke-static {p0, p1, p2, p3}, Landroid/os/Message;->obtain(Landroid/os/Handler;III)Landroid/os/Message;
  
      move-result-object v0
***************
*** 508,513 ****
--- 569,575 ----
      .parameter "obj"
  
      .prologue
+     .line 247
      invoke-static {p0, p1, p2, p3, p4}, Landroid/os/Message;->obtain(Landroid/os/Handler;IIILjava/lang/Object;)Landroid/os/Message;
  
      move-result-object v0
***************
*** 521,526 ****
--- 583,589 ----
      .parameter "obj"
  
      .prologue
+     .line 218
      invoke-static {p0, p1, p2}, Landroid/os/Message;->obtain(Landroid/os/Handler;ILjava/lang/Object;)Landroid/os/Message;
  
      move-result-object v0
***************
*** 533,538 ****
--- 596,602 ----
      .parameter "r"
  
      .prologue
+     .line 263
      invoke-direct {p0, p1}, Landroid/os/Handler;->getPostMessage(Ljava/lang/Runnable;)Landroid/os/Message;
  
      move-result-object v0
***************
*** 551,556 ****
--- 615,621 ----
      .parameter "r"
  
      .prologue
+     .line 351
      invoke-direct {p0, p1}, Landroid/os/Handler;->getPostMessage(Ljava/lang/Runnable;)Landroid/os/Message;
  
      move-result-object v0
***************
*** 568,573 ****
--- 633,639 ----
      .parameter "uptimeMillis"
  
      .prologue
+     .line 285
      invoke-direct {p0, p1}, Landroid/os/Handler;->getPostMessage(Ljava/lang/Runnable;)Landroid/os/Message;
  
      move-result-object v0
***************
*** 586,591 ****
--- 652,658 ----
      .parameter "uptimeMillis"
  
      .prologue
+     .line 309
      invoke-direct {p0, p1, p2}, Landroid/os/Handler;->getPostMessage(Ljava/lang/Runnable;Ljava/lang/Object;)Landroid/os/Message;
  
      move-result-object v0
***************
*** 603,608 ****
--- 670,676 ----
      .parameter "delayMillis"
  
      .prologue
+     .line 331
      invoke-direct {p0, p1}, Landroid/os/Handler;->getPostMessage(Ljava/lang/Runnable;)Landroid/os/Message;
  
      move-result-object v0
***************
*** 619,630 ****
--- 687,700 ----
      .parameter "r"
  
      .prologue
+     .line 359
      iget-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, p0, p1, v1}, Landroid/os/MessageQueue;->removeMessages(Landroid/os/Handler;Ljava/lang/Runnable;Ljava/lang/Object;)V
  
+     .line 360
      return-void
  .end method
  
***************
*** 634,643 ****
--- 704,715 ----
      .parameter "token"
  
      .prologue
+     .line 369
      iget-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
      invoke-virtual {v0, p0, p1, p2}, Landroid/os/MessageQueue;->removeMessages(Landroid/os/Handler;Ljava/lang/Runnable;Ljava/lang/Object;)V
  
+     .line 370
      return-void
  .end method
  
***************
*** 646,655 ****
--- 718,729 ----
      .parameter "token"
  
      .prologue
+     .line 534
      iget-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
      invoke-virtual {v0, p0, p1}, Landroid/os/MessageQueue;->removeCallbacksAndMessages(Landroid/os/Handler;Ljava/lang/Object;)V
  
+     .line 535
      return-void
  .end method
  
***************
*** 658,663 ****
--- 732,738 ----
      .parameter "what"
  
      .prologue
+     .line 516
      iget-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
      const/4 v1, 0x0
***************
*** 666,671 ****
--- 741,747 ----
  
      invoke-virtual {v0, p0, p1, v1, v2}, Landroid/os/MessageQueue;->removeMessages(Landroid/os/Handler;ILjava/lang/Object;Z)Z
  
+     .line 517
      return-void
  .end method
  
***************
*** 675,686 ****
--- 751,764 ----
      .parameter "object"
  
      .prologue
+     .line 525
      iget-object v0, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
      const/4 v1, 0x1
  
      invoke-virtual {v0, p0, p1, p2, v1}, Landroid/os/MessageQueue;->removeMessages(Landroid/os/Handler;ILjava/lang/Object;Z)Z
  
+     .line 526
      return-void
  .end method
  
***************
*** 689,694 ****
--- 767,773 ----
      .parameter "what"
  
      .prologue
+     .line 395
      const-wide/16 v0, 0x0
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/os/Handler;->sendEmptyMessageDelayed(IJ)Z
***************
*** 704,716 ****
--- 783,798 ----
      .parameter "uptimeMillis"
  
      .prologue
+     .line 424
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 425
      .local v0, msg:Landroid/os/Message;
      iput p1, v0, Landroid/os/Message;->what:I
  
+     .line 426
      invoke-virtual {p0, v0, p2, p3}, Landroid/os/Handler;->sendMessageAtTime(Landroid/os/Message;J)Z
  
      move-result v1
***************
*** 724,736 ****
--- 806,821 ----
      .parameter "delayMillis"
  
      .prologue
+     .line 408
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 409
      .local v0, msg:Landroid/os/Message;
      iput p1, v0, Landroid/os/Message;->what:I
  
+     .line 410
      invoke-virtual {p0, v0, p2, p3}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
      move-result v1
***************
*** 743,748 ****
--- 828,834 ----
      .parameter "msg"
  
      .prologue
+     .line 383
      const-wide/16 v0, 0x0
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
***************
*** 757,781 ****
--- 843,874 ----
      .parameter "msg"
  
      .prologue
+     .line 497
      const/4 v2, 0x0
  
+     .line 498
      .local v2, sent:Z
      iget-object v1, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
+     .line 499
      .local v1, queue:Landroid/os/MessageQueue;
      if-eqz v1, :cond_0
  
+     .line 500
      iput-object p0, p1, Landroid/os/Message;->target:Landroid/os/Handler;
  
+     .line 501
      const-wide/16 v3, 0x0
  
      invoke-virtual {v1, p1, v3, v4}, Landroid/os/MessageQueue;->enqueueMessage(Landroid/os/Message;J)Z
  
      move-result v2
  
+     .line 508
      :goto_0
      return v2
  
+     .line 504
      :cond_0
      new-instance v0, Ljava/lang/RuntimeException;
  
***************
*** 799,804 ****
--- 892,898 ----
  
      invoke-direct {v0, v3}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
+     .line 506
      .local v0, e:Ljava/lang/RuntimeException;
      const-string v3, "Looper"
  
***************
*** 817,839 ****
--- 911,940 ----
      .parameter "uptimeMillis"
  
      .prologue
+     .line 469
      const/4 v2, 0x0
  
+     .line 470
      .local v2, sent:Z
      iget-object v1, p0, Landroid/os/Handler;->mQueue:Landroid/os/MessageQueue;
  
+     .line 471
      .local v1, queue:Landroid/os/MessageQueue;
      if-eqz v1, :cond_0
  
+     .line 472
      iput-object p0, p1, Landroid/os/Message;->target:Landroid/os/Handler;
  
+     .line 473
      invoke-virtual {v1, p1, p2, p3}, Landroid/os/MessageQueue;->enqueueMessage(Landroid/os/Message;J)Z
  
      move-result v2
  
+     .line 480
      :goto_0
      return v2
  
+     .line 476
      :cond_0
      new-instance v0, Ljava/lang/RuntimeException;
  
***************
*** 857,862 ****
--- 958,964 ----
  
      invoke-direct {v0, v3}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
+     .line 478
      .local v0, e:Ljava/lang/RuntimeException;
      const-string v3, "Looper"
  
***************
*** 875,888 ****
--- 977,993 ----
      .parameter "delayMillis"
  
      .prologue
+     .line 443
      const-wide/16 v0, 0x0
  
      cmp-long v0, p2, v0
  
      if-gez v0, :cond_0
  
+     .line 444
      const-wide/16 p2, 0x0
  
+     .line 446
      :cond_0
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
***************
*** 901,906 ****
--- 1006,1012 ----
      .locals 2
  
      .prologue
+     .line 570
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 941,947 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1047,1053 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
