*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/IVibratorService$Stub$Proxy.smali	2012-03-23 19:36:51.819625913 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/IVibratorService$Stub$Proxy.smali.original	2012-03-23 19:34:45.370320113 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 94
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 95
      iput-object p1, p0, Landroid/os/IVibratorService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 96
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 99
      iget-object v0, p0, Landroid/os/IVibratorService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 55,77 ****
--- 59,86 ----
      .end annotation
  
      .prologue
+     .line 157
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 158
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 160
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.os.IVibratorService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 161
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 162
      iget-object v2, p0, Landroid/os/IVibratorService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x4
***************
*** 80,100 ****
--- 89,115 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 163
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 166
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 167
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 169
      return-void
  
+     .line 166
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 167
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 104,109 ****
--- 119,125 ----
      .locals 1
  
      .prologue
+     .line 103
      const-string v0, "android.os.IVibratorService"
  
      return-object v0
***************
*** 122,142 ****
--- 138,162 ----
  
      const/4 v3, 0x0
  
+     .line 107
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 108
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 111
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "android.os.IVibratorService"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 112
      iget-object v4, p0, Landroid/os/IVibratorService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v5, 0x1
***************
*** 145,152 ****
--- 165,174 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 113
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 114
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 155,179 ****
--- 177,207 ----
  
      if-eqz v4, :cond_0
  
+     .line 117
      .local v2, _result:Z
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 118
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 120
      return v2
  
      .end local v2           #_result:Z
      :cond_0
      move v2, v3
  
+     .line 114
      goto :goto_0
  
+     .line 117
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 118
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 190,214 ****
--- 218,248 ----
      .end annotation
  
      .prologue
+     .line 124
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 125
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 127
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.os.IVibratorService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 128
      invoke-virtual {v0, p1, p2}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 129
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 130
      iget-object v2, p0, Landroid/os/IVibratorService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x2
***************
*** 217,237 ****
--- 251,277 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 131
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 134
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 135
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 137
      return-void
  
+     .line 134
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 135
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 249,275 ****
--- 289,322 ----
      .end annotation
  
      .prologue
+     .line 140
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 141
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 143
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.os.IVibratorService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 144
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeLongArray([J)V
  
+     .line 145
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 146
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 147
      iget-object v2, p0, Landroid/os/IVibratorService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x3
***************
*** 278,298 ****
--- 325,351 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 148
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 151
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 152
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 154
      return-void
  
+     .line 151
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 152
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
