*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/MemoryFile$MemoryInputStream.smali	2012-03-23 19:36:50.481601515 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/MemoryFile$MemoryInputStream.smali.original	2012-03-23 19:34:45.359319913 -0500
***************
*** 32,43 ****
--- 32,46 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 256
      iput-object p1, p0, Landroid/os/MemoryFile$MemoryInputStream;->this$0:Landroid/os/MemoryFile;
  
      invoke-direct {p0}, Ljava/io/InputStream;-><init>()V
  
+     .line 258
      iput v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mMark:I
  
+     .line 259
      iput v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mOffset:I
  
      return-void
***************
*** 49,54 ****
--- 52,58 ----
      .parameter "x1"
  
      .prologue
+     .line 256
      invoke-direct {p0, p1}, Landroid/os/MemoryFile$MemoryInputStream;-><init>(Landroid/os/MemoryFile;)V
  
      return-void
***************
*** 65,70 ****
--- 69,75 ----
      .end annotation
  
      .prologue
+     .line 264
      iget v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mOffset:I
  
      iget-object v1, p0, Landroid/os/MemoryFile$MemoryInputStream;->this$0:Landroid/os/MemoryFile;
***************
*** 76,83 ****
--- 81,90 ----
  
      if-lt v0, v1, :cond_0
  
+     .line 265
      const/4 v0, 0x0
  
+     .line 267
      :goto_0
      return v0
  
***************
*** 101,110 ****
--- 108,119 ----
      .parameter "readlimit"
  
      .prologue
+     .line 277
      iget v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mOffset:I
  
      iput v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mMark:I
  
+     .line 278
      return-void
  .end method
  
***************
*** 112,117 ****
--- 121,127 ----
      .locals 1
  
      .prologue
+     .line 272
      const/4 v0, 0x1
  
      return v0
***************
*** 130,143 ****
--- 140,156 ----
  
      const/4 v2, 0x1
  
+     .line 287
      iget-object v1, p0, Landroid/os/MemoryFile$MemoryInputStream;->mSingleByte:[B
  
      if-nez v1, :cond_0
  
+     .line 288
      new-array v1, v2, [B
  
      iput-object v1, p0, Landroid/os/MemoryFile$MemoryInputStream;->mSingleByte:[B
  
+     .line 290
      :cond_0
      iget-object v1, p0, Landroid/os/MemoryFile$MemoryInputStream;->mSingleByte:[B
  
***************
*** 145,155 ****
--- 158,171 ----
  
      move-result v0
  
+     .line 291
      .local v0, result:I
      if-eq v0, v2, :cond_1
  
+     .line 292
      const/4 v1, -0x1
  
+     .line 294
      :goto_0
      return v1
  
***************
*** 173,178 ****
--- 189,195 ----
      .end annotation
  
      .prologue
+     .line 299
      if-ltz p2, :cond_0
  
      if-ltz p3, :cond_0
***************
*** 183,188 ****
--- 200,206 ----
  
      if-le v1, v2, :cond_1
  
+     .line 302
      :cond_0
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 190,195 ****
--- 208,214 ----
  
      throw v1
  
+     .line 304
      :cond_1
      invoke-virtual {p0}, Landroid/os/MemoryFile$MemoryInputStream;->available()I
  
***************
*** 199,214 ****
--- 218,237 ----
  
      move-result p3
  
+     .line 305
      const/4 v1, 0x1
  
      if-ge p3, v1, :cond_3
  
+     .line 306
      const/4 v0, -0x1
  
+     .line 312
      :cond_2
      :goto_0
      return v0
  
+     .line 308
      :cond_3
      iget-object v1, p0, Landroid/os/MemoryFile$MemoryInputStream;->this$0:Landroid/os/MemoryFile;
  
***************
*** 218,226 ****
--- 241,251 ----
  
      move-result v0
  
+     .line 309
      .local v0, result:I
      if-lez v0, :cond_2
  
+     .line 310
      iget v1, p0, Landroid/os/MemoryFile$MemoryInputStream;->mOffset:I
  
      add-int/2addr v1, v0
***************
*** 239,248 ****
--- 264,275 ----
      .end annotation
  
      .prologue
+     .line 282
      iget v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mMark:I
  
      iput v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mOffset:I
  
+     .line 283
      return-void
  .end method
  
***************
*** 256,261 ****
--- 283,289 ----
      .end annotation
  
      .prologue
+     .line 317
      iget v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mOffset:I
  
      int-to-long v0, v0
***************
*** 275,280 ****
--- 303,309 ----
  
      if-lez v0, :cond_0
  
+     .line 318
      iget-object v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->this$0:Landroid/os/MemoryFile;
  
      #getter for: Landroid/os/MemoryFile;->mLength:I
***************
*** 288,293 ****
--- 317,323 ----
  
      int-to-long p1, v0
  
+     .line 320
      :cond_0
      iget v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mOffset:I
  
***************
*** 299,303 ****
--- 329,334 ----
  
      iput v0, p0, Landroid/os/MemoryFile$MemoryInputStream;->mOffset:I
  
+     .line 321
      return-wide p1
  .end method
