*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Messenger$1.smali	2012-03-23 19:36:46.958537271 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/Messenger$1.smali.original	2012-03-23 19:34:45.343319621 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 92
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,53 ****
--- 45,56 ----
      .parameter "in"
  
      .prologue
+     .line 94
      invoke-virtual {p1}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 95
      .local v0, target:Landroid/os/IBinder;
      if-eqz v0, :cond_0
  
***************
*** 69,74 ****
--- 72,78 ----
      .parameter "x0"
  
      .prologue
+     .line 92
      invoke-virtual {p0, p1}, Landroid/os/Messenger$1;->createFromParcel(Landroid/os/Parcel;)Landroid/os/Messenger;
  
      move-result-object v0
***************
*** 81,86 ****
--- 85,91 ----
      .parameter "size"
  
      .prologue
+     .line 99
      new-array v0, p1, [Landroid/os/Messenger;
  
      return-object v0
***************
*** 91,96 ****
--- 96,102 ----
      .parameter "x0"
  
      .prologue
+     .line 92
      invoke-virtual {p0, p1}, Landroid/os/Messenger$1;->newArray(I)[Landroid/os/Messenger;
  
      move-result-object v0
