*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/NetworkOnMainThreadException.smali	2012-03-23 19:36:45.586512253 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/NetworkOnMainThreadException.smali.original	2012-03-23 19:34:45.335319475 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 31
      invoke-direct {p0}, Ljava/lang/RuntimeException;-><init>()V
  
      return-void
