*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/ParcelFileDescriptor.smali	2012-03-23 19:36:48.731569604 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/ParcelFileDescriptor.smali.original	2012-03-23 19:34:45.351319767 -0500
***************
*** 57,62 ****
--- 57,63 ----
      .locals 1
  
      .prologue
+     .line 421
      new-instance v0, Landroid/os/ParcelFileDescriptor$1;
  
      invoke-direct {v0}, Landroid/os/ParcelFileDescriptor$1;-><init>()V
***************
*** 71,86 ****
--- 72,91 ----
      .parameter "descriptor"
  
      .prologue
+     .line 386
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 387
      iput-object p1, p0, Landroid/os/ParcelFileDescriptor;->mParcelDescriptor:Landroid/os/ParcelFileDescriptor;
  
+     .line 388
      iget-object v0, p0, Landroid/os/ParcelFileDescriptor;->mParcelDescriptor:Landroid/os/ParcelFileDescriptor;
  
      iget-object v0, v0, Landroid/os/ParcelFileDescriptor;->mFileDescriptor:Ljava/io/FileDescriptor;
  
      iput-object v0, p0, Landroid/os/ParcelFileDescriptor;->mFileDescriptor:Ljava/io/FileDescriptor;
  
+     .line 389
      return-void
  .end method
  
***************
*** 89,98 ****
--- 94,106 ----
      .parameter "descriptor"
  
      .prologue
+     .line 392
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 393
      if-nez p1, :cond_0
  
+     .line 394
      new-instance v0, Ljava/lang/NullPointerException;
  
      const-string v1, "descriptor must not be null"
***************
*** 101,113 ****
--- 109,124 ----
  
      throw v0
  
+     .line 396
      :cond_0
      iput-object p1, p0, Landroid/os/ParcelFileDescriptor;->mFileDescriptor:Ljava/io/FileDescriptor;
  
+     .line 397
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/os/ParcelFileDescriptor;->mParcelDescriptor:Landroid/os/ParcelFileDescriptor;
  
+     .line 398
      return-void
  .end method
  
***************
*** 116,125 ****
--- 127,138 ----
      .parameter "fd"
  
      .prologue
+     .line 170
      invoke-static {p0}, Landroid/os/ParcelFileDescriptor;->getFileDescriptorFromFdNoDup(I)Ljava/io/FileDescriptor;
  
      move-result-object v0
  
+     .line 171
      .local v0, fdesc:Ljava/io/FileDescriptor;
      new-instance v1, Landroid/os/ParcelFileDescriptor;
  
***************
*** 143,155 ****
--- 156,172 ----
  
      const/4 v4, 0x0
  
+     .line 214
      new-array v0, v2, [Ljava/io/FileDescriptor;
  
+     .line 215
      .local v0, fds:[Ljava/io/FileDescriptor;
      invoke-static {v0}, Landroid/os/ParcelFileDescriptor;->createPipeNative([Ljava/io/FileDescriptor;)V
  
+     .line 216
      new-array v1, v2, [Landroid/os/ParcelFileDescriptor;
  
+     .line 217
      .local v1, pfds:[Landroid/os/ParcelFileDescriptor;
      new-instance v2, Landroid/os/ParcelFileDescriptor;
  
***************
*** 159,164 ****
--- 176,182 ----
  
      aput-object v2, v1, v4
  
+     .line 218
      new-instance v2, Landroid/os/ParcelFileDescriptor;
  
      aget-object v3, v0, v5
***************
*** 167,172 ****
--- 185,191 ----
  
      aput-object v2, v1, v5
  
+     .line 219
      return-object v1
  .end method
  
***************
*** 188,197 ****
--- 207,218 ----
      .end annotation
  
      .prologue
+     .line 127
      invoke-static {p0}, Landroid/os/Parcel;->dupFileDescriptor(Ljava/io/FileDescriptor;)Ljava/io/FileDescriptor;
  
      move-result-object v0
  
+     .line 128
      .local v0, fd:Ljava/io/FileDescriptor;
      if-eqz v0, :cond_0
  
***************
*** 226,237 ****
--- 247,261 ----
  
      const/4 v4, 0x0
  
+     .line 236
      if-nez p0, :cond_1
  
+     .line 243
      :cond_0
      :goto_0
      return-object v2
  
+     .line 237
      :cond_1
      new-instance v1, Landroid/os/MemoryFile;
  
***************
*** 239,260 ****
--- 263,289 ----
  
      invoke-direct {v1, p1, v3}, Landroid/os/MemoryFile;-><init>(Ljava/lang/String;I)V
  
+     .line 238
      .local v1, file:Landroid/os/MemoryFile;
      array-length v3, p0
  
      if-lez v3, :cond_2
  
+     .line 239
      array-length v3, p0
  
      invoke-virtual {v1, p0, v4, v4, v3}, Landroid/os/MemoryFile;->writeBytes([BIII)V
  
+     .line 241
      :cond_2
      invoke-virtual {v1}, Landroid/os/MemoryFile;->deactivate()V
  
+     .line 242
      invoke-virtual {v1}, Landroid/os/MemoryFile;->getFileDescriptor()Ljava/io/FileDescriptor;
  
      move-result-object v0
  
+     .line 243
      .local v0, fd:Ljava/io/FileDescriptor;
      if-eqz v0, :cond_0
  
***************
*** 270,279 ****
--- 299,310 ----
      .parameter "datagramSocket"
  
      .prologue
+     .line 204
      invoke-virtual {p0}, Ljava/net/DatagramSocket;->getFileDescriptor$()Ljava/io/FileDescriptor;
  
      move-result-object v0
  
+     .line 205
      .local v0, fd:Ljava/io/FileDescriptor;
      if-eqz v0, :cond_0
  
***************
*** 300,309 ****
--- 331,342 ----
      .end annotation
  
      .prologue
+     .line 152
      invoke-static {p0}, Landroid/os/ParcelFileDescriptor;->getFileDescriptorFromFd(I)Ljava/io/FileDescriptor;
  
      move-result-object v0
  
+     .line 153
      .local v0, fdesc:Ljava/io/FileDescriptor;
      new-instance v1, Landroid/os/ParcelFileDescriptor;
  
***************
*** 317,326 ****
--- 350,361 ----
      .parameter "socket"
  
      .prologue
+     .line 190
      invoke-virtual {p0}, Ljava/net/Socket;->getFileDescriptor$()Ljava/io/FileDescriptor;
  
      move-result-object v0
  
+     .line 191
      .local v0, fd:Ljava/io/FileDescriptor;
      if-eqz v0, :cond_0
  
***************
*** 362,389 ****
--- 397,431 ----
      .end annotation
  
      .prologue
+     .line 102
      invoke-virtual {p0}, Ljava/io/File;->getPath()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 103
      .local v1, path:Ljava/lang/String;
      invoke-static {}, Ljava/lang/System;->getSecurityManager()Ljava/lang/SecurityManager;
  
      move-result-object v2
  
+     .line 104
      .local v2, security:Ljava/lang/SecurityManager;
      if-eqz v2, :cond_0
  
+     .line 105
      invoke-virtual {v2, v1}, Ljava/lang/SecurityManager;->checkRead(Ljava/lang/String;)V
  
+     .line 106
      const/high16 v3, 0x2000
  
      and-int/2addr v3, p1
  
      if-eqz v3, :cond_0
  
+     .line 107
      invoke-virtual {v2, v1}, Ljava/lang/SecurityManager;->checkWrite(Ljava/lang/String;)V
  
+     .line 111
      :cond_0
      const/high16 v3, 0x3000
  
***************
*** 391,396 ****
--- 433,439 ----
  
      if-nez v3, :cond_1
  
+     .line 112
      new-instance v3, Ljava/lang/IllegalArgumentException;
  
      const-string v4, "Must specify MODE_READ_ONLY, MODE_WRITE_ONLY, or MODE_READ_WRITE"
***************
*** 399,409 ****
--- 442,454 ----
  
      throw v3
  
+     .line 116
      :cond_1
      invoke-static {v1, p1}, Landroid/os/Parcel;->openFileDescriptor(Ljava/lang/String;I)Ljava/io/FileDescriptor;
  
      move-result-object v0
  
+     .line 117
      .local v0, fd:Ljava/io/FileDescriptor;
      if-eqz v0, :cond_2
  
***************
*** 431,438 ****
--- 476,485 ----
      .end annotation
  
      .prologue
+     .line 310
      monitor-enter p0
  
+     .line 311
      :try_start_0
      iget-boolean v0, p0, Landroid/os/ParcelFileDescriptor;->mClosed:Z
  
***************
*** 440,467 ****
--- 487,520 ----
  
      monitor-exit p0
  
+     .line 321
      :goto_0
      return-void
  
+     .line 312
      :cond_0
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/os/ParcelFileDescriptor;->mClosed:Z
  
+     .line 313
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 314
      iget-object v0, p0, Landroid/os/ParcelFileDescriptor;->mParcelDescriptor:Landroid/os/ParcelFileDescriptor;
  
      if-eqz v0, :cond_1
  
+     .line 317
      iget-object v0, p0, Landroid/os/ParcelFileDescriptor;->mParcelDescriptor:Landroid/os/ParcelFileDescriptor;
  
      invoke-virtual {v0}, Landroid/os/ParcelFileDescriptor;->close()V
  
      goto :goto_0
  
+     .line 313
      :catchall_0
      move-exception v0
  
***************
*** 472,477 ****
--- 525,531 ----
  
      throw v0
  
+     .line 319
      :cond_1
      iget-object v0, p0, Landroid/os/ParcelFileDescriptor;->mFileDescriptor:Ljava/io/FileDescriptor;
  
***************
*** 484,489 ****
--- 538,544 ----
      .locals 1
  
      .prologue
+     .line 402
      const/4 v0, 0x1
  
      return v0
***************
*** 495,504 ****
--- 550,561 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 288
      iget-boolean v2, p0, Landroid/os/ParcelFileDescriptor;->mClosed:Z
  
      if-eqz v2, :cond_0
  
+     .line 289
      new-instance v2, Ljava/lang/IllegalStateException;
  
      const-string v3, "Already closed"
***************
*** 507,548 ****
--- 564,613 ----
  
      throw v2
  
+     .line 291
      :cond_0
      iget-object v2, p0, Landroid/os/ParcelFileDescriptor;->mParcelDescriptor:Landroid/os/ParcelFileDescriptor;
  
      if-eqz v2, :cond_1
  
+     .line 292
      iget-object v2, p0, Landroid/os/ParcelFileDescriptor;->mParcelDescriptor:Landroid/os/ParcelFileDescriptor;
  
      invoke-virtual {v2}, Landroid/os/ParcelFileDescriptor;->detachFd()I
  
      move-result v0
  
+     .line 293
      .local v0, fd:I
      iput-boolean v3, p0, Landroid/os/ParcelFileDescriptor;->mClosed:Z
  
      move v1, v0
  
+     .line 299
      .end local v0           #fd:I
      .local v1, fd:I
      :goto_0
      return v1
  
+     .line 296
      .end local v1           #fd:I
      :cond_1
      invoke-virtual {p0}, Landroid/os/ParcelFileDescriptor;->getFd()I
  
      move-result v0
  
+     .line 297
      .restart local v0       #fd:I
      iput-boolean v3, p0, Landroid/os/ParcelFileDescriptor;->mClosed:Z
  
+     .line 298
      iget-object v2, p0, Landroid/os/ParcelFileDescriptor;->mFileDescriptor:Ljava/io/FileDescriptor;
  
      invoke-static {v2}, Landroid/os/Parcel;->clearFileDescriptor(Ljava/io/FileDescriptor;)V
  
      move v1, v0
  
+     .line 299
      .end local v0           #fd:I
      .restart local v1       #fd:I
      goto :goto_0
***************
*** 557,562 ****
--- 622,628 ----
      .end annotation
  
      .prologue
+     .line 138
      invoke-virtual {p0}, Landroid/os/ParcelFileDescriptor;->getFileDescriptor()Ljava/io/FileDescriptor;
  
      move-result-object v0
***************
*** 577,596 ****
--- 643,667 ----
      .end annotation
  
      .prologue
+     .line 377
      :try_start_0
      iget-boolean v0, p0, Landroid/os/ParcelFileDescriptor;->mClosed:Z
  
      if-nez v0, :cond_0
  
+     .line 378
      invoke-virtual {p0}, Landroid/os/ParcelFileDescriptor;->close()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 381
      :cond_0
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 383
      return-void
  
+     .line 381
      :catchall_0
      move-exception v0
  
***************
*** 603,612 ****
--- 674,685 ----
      .locals 2
  
      .prologue
+     .line 274
      iget-boolean v0, p0, Landroid/os/ParcelFileDescriptor;->mClosed:Z
  
      if-eqz v0, :cond_0
  
+     .line 275
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "Already closed"
***************
*** 615,620 ****
--- 688,694 ----
  
      throw v0
  
+     .line 277
      :cond_0
      invoke-direct {p0}, Landroid/os/ParcelFileDescriptor;->getFdNative()I
  
***************
*** 627,632 ****
--- 701,707 ----
      .locals 1
  
      .prologue
+     .line 252
      iget-object v0, p0, Landroid/os/ParcelFileDescriptor;->mFileDescriptor:Ljava/io/FileDescriptor;
  
      return-object v0
***************
*** 642,652 ****
      .locals 2
  
      .prologue
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v1, "{ParcelFileDescriptor: "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 717,728 ----
      .locals 2
  
      .prologue
+     .line 371
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v1, "{ParcelFileDescriptor: "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 658,664 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 734,740 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 677,686 ****
--- 753,764 ----
      .parameter "flags"
  
      .prologue
+     .line 411
      iget-object v0, p0, Landroid/os/ParcelFileDescriptor;->mFileDescriptor:Ljava/io/FileDescriptor;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeFileDescriptor(Ljava/io/FileDescriptor;)V
  
+     .line 412
      and-int/lit8 v0, p2, 0x1
  
      if-eqz v0, :cond_0
***************
*** 689,703 ****
--- 767,784 ----
  
      if-nez v0, :cond_0
  
+     .line 414
      :try_start_0
      invoke-virtual {p0}, Landroid/os/ParcelFileDescriptor;->close()V
      :try_end_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 419
      :cond_0
      :goto_0
      return-void
  
+     .line 415
      :catch_0
      move-exception v0
  
