*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/ParcelUuid.smali	2012-03-23 19:36:47.920554814 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/ParcelUuid.smali.original	2012-03-23 19:34:45.348319713 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .locals 1
  
      .prologue
+     .line 110
      new-instance v0, Landroid/os/ParcelUuid$1;
  
      invoke-direct {v0}, Landroid/os/ParcelUuid$1;-><init>()V
***************
*** 42,51 ****
--- 43,55 ----
      .parameter "uuid"
  
      .prologue
+     .line 36
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 37
      iput-object p1, p0, Landroid/os/ParcelUuid;->mUuid:Ljava/util/UUID;
  
+     .line 38
      return-void
  .end method
  
***************
*** 54,59 ****
--- 58,64 ----
      .parameter "uuid"
  
      .prologue
+     .line 52
      new-instance v0, Landroid/os/ParcelUuid;
  
      invoke-static {p0}, Ljava/util/UUID;->fromString(Ljava/lang/String;)Ljava/util/UUID;
***************
*** 71,76 ****
--- 76,82 ----
      .locals 1
  
      .prologue
+     .line 125
      const/4 v0, 0x0
  
      return v0
***************
*** 83,101 ****
--- 89,112 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 93
      if-nez p1, :cond_1
  
+     .line 107
      :cond_0
      :goto_0
      return v1
  
+     .line 97
      :cond_1
      if-ne p0, p1, :cond_2
  
+     .line 98
      const/4 v1, 0x1
  
      goto :goto_0
  
+     .line 101
      :cond_2
      instance-of v2, p1, Landroid/os/ParcelUuid;
  
***************
*** 103,110 ****
--- 114,123 ----
  
      move-object v0, p1
  
+     .line 105
      check-cast v0, Landroid/os/ParcelUuid;
  
+     .line 107
      .local v0, that:Landroid/os/ParcelUuid;
      iget-object v1, p0, Landroid/os/ParcelUuid;->mUuid:Ljava/util/UUID;
  
***************
*** 121,126 ****
--- 134,140 ----
      .locals 1
  
      .prologue
+     .line 61
      iget-object v0, p0, Landroid/os/ParcelUuid;->mUuid:Ljava/util/UUID;
  
      return-object v0
***************
*** 130,135 ****
--- 144,150 ----
      .locals 1
  
      .prologue
+     .line 78
      iget-object v0, p0, Landroid/os/ParcelUuid;->mUuid:Ljava/util/UUID;
  
      invoke-virtual {v0}, Ljava/util/UUID;->hashCode()I
***************
*** 143,148 ****
--- 158,164 ----
      .locals 1
  
      .prologue
+     .line 72
      iget-object v0, p0, Landroid/os/ParcelUuid;->mUuid:Ljava/util/UUID;
  
      invoke-virtual {v0}, Ljava/util/UUID;->toString()Ljava/lang/String;
***************
*** 158,163 ****
--- 174,180 ----
      .parameter "flags"
  
      .prologue
+     .line 129
      iget-object v0, p0, Landroid/os/ParcelUuid;->mUuid:Ljava/util/UUID;
  
      invoke-virtual {v0}, Ljava/util/UUID;->getMostSignificantBits()J
***************
*** 166,171 ****
--- 183,189 ----
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 130
      iget-object v0, p0, Landroid/os/ParcelUuid;->mUuid:Ljava/util/UUID;
  
      invoke-virtual {v0}, Ljava/util/UUID;->getLeastSignificantBits()J
***************
*** 174,178 ****
--- 192,197 ----
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 131
      return-void
  .end method
