*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/PowerManager.smali	2012-03-23 19:36:45.503510739 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/PowerManager.smali.original	2012-03-23 19:34:45.335319475 -0500
***************
*** 59,66 ****
--- 59,68 ----
      .locals 0
  
      .prologue
+     .line 539
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 540
      return-void
  .end method
  
***************
*** 70,81 ****
--- 72,87 ----
      .parameter "handler"
  
      .prologue
+     .line 546
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 547
      iput-object p1, p0, Landroid/os/PowerManager;->mService:Landroid/os/IPowerManager;
  
+     .line 548
      iput-object p2, p0, Landroid/os/PowerManager;->mHandler:Landroid/os/Handler;
  
+     .line 549
      return-void
  .end method
  
***************
*** 85,90 ****
--- 91,97 ----
      .locals 2
  
      .prologue
+     .line 496
      :try_start_0
      iget-object v1, p0, Landroid/os/PowerManager;->mService:Landroid/os/IPowerManager;
  
***************
*** 94,105 ****
--- 101,115 ----
  
      move-result v1
  
+     .line 498
      :goto_0
      return v1
  
+     .line 497
      :catch_0
      move-exception v0
  
+     .line 498
      .local v0, e:Landroid/os/RemoteException;
      const/4 v1, 0x0
  
***************
*** 111,116 ****
--- 121,127 ----
      .parameter "time"
  
      .prologue
+     .line 456
      :try_start_0
      iget-object v0, p0, Landroid/os/PowerManager;->mService:Landroid/os/IPowerManager;
  
***************
*** 118,126 ****
--- 129,139 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 459
      :goto_0
      return-void
  
+     .line 457
      :catch_0
      move-exception v0
  
***************
*** 131,136 ****
--- 144,150 ----
      .locals 2
  
      .prologue
+     .line 516
      :try_start_0
      iget-object v1, p0, Landroid/os/PowerManager;->mService:Landroid/os/IPowerManager;
  
***************
*** 140,151 ****
--- 154,168 ----
  
      move-result v1
  
+     .line 518
      :goto_0
      return v1
  
+     .line 517
      :catch_0
      move-exception v0
  
+     .line 518
      .local v0, e:Landroid/os/RemoteException;
      const/4 v1, 0x0
  
***************
*** 158,173 ****
      .parameter "tag"
  
      .prologue
      if-nez p2, :cond_0
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "tag is null in PowerManager.newWakeLock"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      new-instance v0, Landroid/os/PowerManager$WakeLock;
  
--- 175,193 ----
      .parameter "tag"
  
      .prologue
+     .line 416
      if-nez p2, :cond_0
  
+     .line 417
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "tag is null in PowerManager.newWakeLock"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 419
      :cond_0
      new-instance v0, Landroid/os/PowerManager$WakeLock;
  
***************
*** 181,186 ****
--- 201,207 ----
      .parameter "reason"
  
      .prologue
+     .line 533
      :try_start_0
      iget-object v0, p0, Landroid/os/PowerManager;->mService:Landroid/os/IPowerManager;
  
***************
*** 188,196 ****
--- 209,219 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 536
      :goto_0
      return-void
  
+     .line 534
      :catch_0
      move-exception v0
  
***************
*** 202,207 ****
--- 225,231 ----
      .parameter "brightness"
  
      .prologue
+     .line 471
      :try_start_0
      iget-object v0, p0, Landroid/os/PowerManager;->mService:Landroid/os/IPowerManager;
  
***************
*** 209,217 ****
--- 233,243 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 474
      :goto_0
      return-void
  
+     .line 472
      :catch_0
      move-exception v0
  
***************
*** 224,229 ****
--- 250,256 ----
      .parameter "noChangeLights"
  
      .prologue
+     .line 440
      :try_start_0
      iget-object v0, p0, Landroid/os/PowerManager;->mService:Landroid/os/IPowerManager;
  
***************
*** 231,239 ****
--- 258,268 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 443
      :goto_0
      return-void
  
+     .line 441
      :catch_0
      move-exception v0
  
