*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/StrictMode$8.smali	2012-03-23 19:36:46.781534044 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/StrictMode$8.smali.original	2012-03-23 19:34:45.342319603 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .locals 0
  
      .prologue
+     .line 1752
      invoke-direct {p0}, Landroid/util/Singleton;-><init>()V
  
      return-void
***************
*** 39,45 ****
      .locals 1
  
      .prologue
!     const-string v0, "window"
  
      invoke-static {v0}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
--- 40,47 ----
      .locals 1
  
      .prologue
!     .line 1754
!     const-string/jumbo v0, "window"
  
      invoke-static {v0}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
***************
*** 56,61 ****
--- 58,64 ----
      .locals 1
  
      .prologue
+     .line 1752
      invoke-virtual {p0}, Landroid/os/StrictMode$8;->create()Landroid/view/IWindowManager;
  
      move-result-object v0
