*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/StrictMode$VmPolicy.smali	2012-03-23 19:36:45.157504430 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/StrictMode$VmPolicy.smali.original	2012-03-23 19:34:45.332319421 -0500
***************
*** 45,50 ****
--- 45,51 ----
      .locals 3
  
      .prologue
+     .line 524
      new-instance v0, Landroid/os/StrictMode$VmPolicy;
  
      const/4 v1, 0x0
***************
*** 76,86 ****
--- 77,90 ----
      .end annotation
  
      .prologue
+     .line 531
      .local p2, classInstanceLimit:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/Class;Ljava/lang/Integer;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 532
      if-nez p2, :cond_0
  
+     .line 533
      new-instance v0, Ljava/lang/NullPointerException;
  
      const-string v1, "classInstanceLimit == null"
***************
*** 89,99 ****
--- 93,106 ----
  
      throw v0
  
+     .line 535
      :cond_0
      iput p1, p0, Landroid/os/StrictMode$VmPolicy;->mask:I
  
+     .line 536
      iput-object p2, p0, Landroid/os/StrictMode$VmPolicy;->classInstanceLimit:Ljava/util/HashMap;
  
+     .line 537
      return-void
  .end method
  
***************
*** 104,109 ****
--- 111,117 ----
      .parameter "x2"
  
      .prologue
+     .line 520
      invoke-direct {p0, p1, p2}, Landroid/os/StrictMode$VmPolicy;-><init>(ILjava/util/HashMap;)V
  
      return-void
***************
*** 115,120 ****
--- 123,129 ----
      .locals 2
  
      .prologue
+     .line 541
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
