*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/StrictMode$StrictModeDiskReadViolation.smali	2012-03-23 19:36:45.354508021 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/StrictMode$StrictModeDiskReadViolation.smali.original	2012-03-23 19:34:45.334319457 -0500
***************
*** 20,30 ****
--- 20,32 ----
      .parameter "policyMask"
  
      .prologue
+     .line 791
      const/4 v0, 0x2
  
      const/4 v1, 0x0
  
      invoke-direct {p0, p1, v0, v1}, Landroid/os/StrictMode$StrictModeViolation;-><init>(IILjava/lang/String;)V
  
+     .line 792
      return-void
  .end method
