*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/TokenWatcher$Death.smali	2012-03-23 19:36:48.403563623 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/TokenWatcher$Death.smali.original	2012-03-23 19:34:45.350319749 -0500
***************
*** 33,46 ****
--- 33,50 ----
      .parameter "tag"
  
      .prologue
+     .line 168
      iput-object p1, p0, Landroid/os/TokenWatcher$Death;->this$0:Landroid/os/TokenWatcher;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 169
      iput-object p2, p0, Landroid/os/TokenWatcher$Death;->token:Landroid/os/IBinder;
  
+     .line 170
      iput-object p3, p0, Landroid/os/TokenWatcher$Death;->tag:Ljava/lang/String;
  
+     .line 171
      return-void
  .end method
  
***************
*** 50,55 ****
--- 54,60 ----
      .locals 3
  
      .prologue
+     .line 175
      iget-object v0, p0, Landroid/os/TokenWatcher$Death;->this$0:Landroid/os/TokenWatcher;
  
      iget-object v1, p0, Landroid/os/TokenWatcher$Death;->token:Landroid/os/IBinder;
***************
*** 58,63 ****
--- 63,69 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/os/TokenWatcher;->cleanup(Landroid/os/IBinder;Z)V
  
+     .line 176
      return-void
  .end method
  
***************
*** 70,80 ****
--- 76,88 ----
      .end annotation
  
      .prologue
+     .line 181
      :try_start_0
      iget-object v0, p0, Landroid/os/TokenWatcher$Death;->token:Landroid/os/IBinder;
  
      if-eqz v0, :cond_0
  
+     .line 182
      iget-object v0, p0, Landroid/os/TokenWatcher$Death;->this$0:Landroid/os/TokenWatcher;
  
      #getter for: Landroid/os/TokenWatcher;->mTag:Ljava/lang/String;
***************
*** 104,109 ****
--- 112,118 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 183
      iget-object v0, p0, Landroid/os/TokenWatcher$Death;->this$0:Landroid/os/TokenWatcher;
  
      iget-object v1, p0, Landroid/os/TokenWatcher$Death;->token:Landroid/os/IBinder;
***************
*** 112,122 ****
--- 121,134 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 187
      :cond_0
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 189
      return-void
  
+     .line 187
      :catchall_0
      move-exception v0
  
