*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/textservice/SpellCheckerSession$SpellCheckerSessionListenerImpl$SpellCheckerParams.smali	2012-03-23 19:37:17.728098354 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/textservice/SpellCheckerSession$SpellCheckerSessionListenerImpl$SpellCheckerParams.smali.original	2012-03-23 19:34:45.625324763 -0500
***************
*** 35,49 ****
--- 35,55 ----
      .parameter "sequentialWords"
  
      .prologue
+     .line 226
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 227
      iput p1, p0, Landroid/view/textservice/SpellCheckerSession$SpellCheckerSessionListenerImpl$SpellCheckerParams;->mWhat:I
  
+     .line 228
      iput-object p2, p0, Landroid/view/textservice/SpellCheckerSession$SpellCheckerSessionListenerImpl$SpellCheckerParams;->mTextInfos:[Landroid/view/textservice/TextInfo;
  
+     .line 229
      iput p3, p0, Landroid/view/textservice/SpellCheckerSession$SpellCheckerSessionListenerImpl$SpellCheckerParams;->mSuggestionsLimit:I
  
+     .line 230
      iput-boolean p4, p0, Landroid/view/textservice/SpellCheckerSession$SpellCheckerSessionListenerImpl$SpellCheckerParams;->mSequentialWords:Z
  
+     .line 231
      return-void
  .end method
