*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/statusbar/IStatusBarService$Stub$Proxy.smali	2012-03-23 19:37:34.731408408 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/statusbar/IStatusBarService$Stub$Proxy.smali.original	2012-03-23 19:34:46.117333735 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 259
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 260
      iput-object p1, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 261
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 264
      iget-object v0, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 54,74 ****
--- 58,82 ----
      .end annotation
  
      .prologue
+     .line 286
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 287
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 289
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 290
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x2
***************
*** 77,97 ****
--- 85,111 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 291
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 294
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 295
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 297
      return-void
  
+     .line 294
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 295
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 109,135 ****
--- 123,156 ----
      .end annotation
  
      .prologue
+     .line 300
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 301
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 303
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 304
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 305
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 306
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 307
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x3
***************
*** 138,158 ****
--- 159,185 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 308
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 311
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 312
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 314
      return-void
  
+     .line 311
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 312
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 167,187 ****
--- 194,218 ----
      .end annotation
  
      .prologue
+     .line 272
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 273
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 275
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 276
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x1
***************
*** 190,210 ****
--- 221,247 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 277
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 280
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 281
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 283
      return-void
  
+     .line 280
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 281
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 214,219 ****
--- 251,257 ----
      .locals 1
  
      .prologue
+     .line 268
      const-string v0, "com.android.internal.statusbar.IStatusBarService"
  
      return-object v0
***************
*** 228,248 ****
--- 266,290 ----
      .end annotation
  
      .prologue
+     .line 481
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 482
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 484
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 485
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xd
***************
*** 251,271 ****
--- 293,319 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 486
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 489
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 490
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 492
      return-void
  
+     .line 489
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 490
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 283,309 ****
--- 331,364 ----
      .end annotation
  
      .prologue
+     .line 495
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 496
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 498
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 499
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 500
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 501
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 502
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xe
***************
*** 312,332 ****
--- 367,393 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 503
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 506
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 507
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 509
      return-void
  
+     .line 506
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 507
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 344,370 ****
--- 405,438 ----
      .end annotation
  
      .prologue
+     .line 444
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 445
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 447
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 448
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 449
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 450
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 451
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xb
***************
*** 373,393 ****
--- 441,467 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 452
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 455
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 456
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 458
      return-void
  
+     .line 455
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 456
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 408,440 ****
--- 482,524 ----
      .end annotation
  
      .prologue
+     .line 461
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 462
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 464
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 465
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 466
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 467
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 468
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 469
      invoke-virtual {v0, p5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 470
      invoke-virtual {v0, p6}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 471
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xc
***************
*** 443,463 ****
--- 527,553 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 472
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 475
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 476
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 478
      return-void
  
+     .line 475
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 476
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 472,492 ****
--- 562,586 ----
      .end annotation
  
      .prologue
+     .line 430
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 431
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 433
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 434
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xa
***************
*** 495,515 ****
--- 589,615 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 435
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 438
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 439
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 441
      return-void
  
+     .line 438
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 439
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 550,555 ****
--- 650,656 ----
      .end annotation
  
      .prologue
+     .line 402
      .local p3, notificationKeys:Ljava/util/List;,"Ljava/util/List<Landroid/os/IBinder;>;"
      .local p4, notifications:Ljava/util/List;,"Ljava/util/List<Lcom/android/internal/statusbar/StatusBarNotification;>;"
      .local p6, binders:Ljava/util/List;,"Ljava/util/List<Landroid/os/IBinder;>;"
***************
*** 557,573 ****
--- 658,677 ----
  
      move-result-object v0
  
+     .line 403
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 405
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 406
      if-eqz p1, :cond_1
  
      invoke-interface {p1}, Lcom/android/internal/statusbar/IStatusBar;->asBinder()Landroid/os/IBinder;
***************
*** 577,588 ****
--- 681,695 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 407
      if-nez p5, :cond_2
  
+     .line 408
      const/4 v2, -0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 413
      :goto_1
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 592,631 ****
--- 699,750 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 414
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 415
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      if-eqz v2, :cond_0
  
+     .line 416
      invoke-virtual {p2, v1}, Lcom/android/internal/statusbar/StatusBarIconList;->readFromParcel(Landroid/os/Parcel;)V
  
+     .line 418
      :cond_0
      invoke-virtual {v1, p3}, Landroid/os/Parcel;->readBinderList(Ljava/util/List;)V
  
+     .line 419
      sget-object v2, Lcom/android/internal/statusbar/StatusBarNotification;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-virtual {v1, p4, v2}, Landroid/os/Parcel;->readTypedList(Ljava/util/List;Landroid/os/Parcelable$Creator;)V
  
+     .line 420
      invoke-virtual {v1, p5}, Landroid/os/Parcel;->readIntArray([I)V
  
+     .line 421
      invoke-virtual {v1, p6}, Landroid/os/Parcel;->readBinderList(Ljava/util/List;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 424
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 425
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 427
      return-void
  
+     .line 406
      :cond_1
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 411
      :cond_2
      :try_start_1
      array-length v2, p5
***************
*** 636,646 ****
--- 755,767 ----
  
      goto :goto_1
  
+     .line 424
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 425
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 656,678 ****
--- 777,804 ----
      .end annotation
  
      .prologue
+     .line 352
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 353
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 355
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 356
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 357
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x6
***************
*** 681,701 ****
--- 807,833 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 358
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 361
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 362
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 364
      return-void
  
+     .line 361
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 362
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 713,733 ****
--- 845,869 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 527
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 528
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 530
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 531
      if-eqz p1, :cond_0
  
      const/4 v2, 0x1
***************
*** 735,740 ****
--- 871,877 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 532
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x10
***************
*** 743,763 ****
--- 880,906 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 533
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 536
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 537
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 539
      return-void
  
+     .line 536
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 537
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 777,807 ****
--- 920,959 ----
      .end annotation
  
      .prologue
+     .line 317
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 318
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 320
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 321
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 322
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 323
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 324
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 325
      invoke-virtual {v0, p5}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 326
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x4
***************
*** 810,830 ****
--- 962,988 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 327
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 330
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 331
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 333
      return-void
  
+     .line 330
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 331
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 843,865 ****
--- 1001,1028 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 336
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 337
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 339
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 340
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 341
      if-eqz p2, :cond_0
  
      const/4 v2, 0x1
***************
*** 867,872 ****
--- 1030,1036 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 342
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x5
***************
*** 875,895 ****
--- 1039,1065 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 343
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 346
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 347
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 349
      return-void
  
+     .line 346
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 347
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 907,933 ****
--- 1077,1110 ----
      .end annotation
  
      .prologue
+     .line 382
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 383
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 385
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 386
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 387
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 388
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 389
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x8
***************
*** 936,956 ****
--- 1113,1139 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 390
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 393
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 394
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 396
      return-void
  
+     .line 393
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 394
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 966,988 ****
--- 1149,1176 ----
      .end annotation
  
      .prologue
+     .line 512
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 513
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 515
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 516
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 517
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xf
***************
*** 991,1011 ****
--- 1179,1205 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 518
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 521
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 522
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 524
      return-void
  
+     .line 521
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 522
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1020,1040 ****
--- 1214,1238 ----
      .end annotation
  
      .prologue
+     .line 542
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 543
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 545
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 546
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x11
***************
*** 1043,1063 ****
--- 1241,1267 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 547
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 550
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 551
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 553
      return-void
  
+     .line 550
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 551
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1075,1095 ****
--- 1279,1303 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 367
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 368
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 370
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.statusbar.IStatusBarService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 371
      if-eqz p1, :cond_0
  
      const/4 v2, 0x1
***************
*** 1097,1102 ****
--- 1305,1311 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 372
      iget-object v2, p0, Lcom/android/internal/statusbar/IStatusBarService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x7
***************
*** 1105,1125 ****
--- 1314,1340 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 373
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 376
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 377
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 379
      return-void
  
+     .line 376
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 377
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
