*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/statusbar/IStatusBar$Stub.smali	2012-03-23 19:37:34.546405036 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/statusbar/IStatusBar$Stub.smali.original	2012-03-23 19:34:46.114333681 -0500
***************
*** 58,69 ****
--- 58,72 ----
      .locals 1
  
      .prologue
+     .line 15
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 16
      const-string v0, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p0, p0, v0}, Lcom/android/internal/statusbar/IStatusBar$Stub;->attachInterface(Landroid/os/IInterface;Ljava/lang/String;)V
  
+     .line 17
      return-void
  .end method
  
***************
*** 72,84 ****
--- 75,91 ----
      .parameter "obj"
  
      .prologue
+     .line 24
      if-nez p0, :cond_0
  
+     .line 25
      const/4 v0, 0x0
  
+     .line 31
      :goto_0
      return-object v0
  
+     .line 27
      :cond_0
      const-string v1, "com.android.internal.statusbar.IStatusBar"
  
***************
*** 86,91 ****
--- 93,99 ----
  
      move-result-object v0
  
+     .line 28
      .local v0, iin:Landroid/os/IInterface;
      if-eqz v0, :cond_1
  
***************
*** 93,102 ****
--- 101,112 ----
  
      if-eqz v1, :cond_1
  
+     .line 29
      check-cast v0, Lcom/android/internal/statusbar/IStatusBar;
  
      goto :goto_0
  
+     .line 31
      :cond_1
      new-instance v0, Lcom/android/internal/statusbar/IStatusBar$Stub$Proxy;
  
***************
*** 112,117 ****
--- 122,128 ----
      .locals 0
  
      .prologue
+     .line 35
      return-object p0
  .end method
  
***************
*** 132,139 ****
--- 143,152 ----
  
      const/4 v3, 0x1
  
+     .line 39
      sparse-switch p1, :sswitch_data_0
  
+     .line 172
      invoke-super {p0, p1, p2, p3, p4}, Landroid/os/Binder;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
      move-result v3
***************
*** 141,146 ****
--- 154,160 ----
      :goto_0
      return v3
  
+     .line 43
      :sswitch_0
      const-string v4, "com.android.internal.statusbar.IStatusBar"
  
***************
*** 148,162 ****
--- 162,179 ----
  
      goto :goto_0
  
+     .line 48
      :sswitch_1
      const-string v4, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 50
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 52
      .local v0, _arg0:I
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 164,169 ****
--- 181,187 ----
  
      if-eqz v4, :cond_0
  
+     .line 53
      sget-object v4, Lcom/android/internal/statusbar/StatusBarIcon;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v4, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 172,183 ****
--- 190,203 ----
  
      check-cast v1, Lcom/android/internal/statusbar/StatusBarIcon;
  
+     .line 58
      .local v1, _arg1:Lcom/android/internal/statusbar/StatusBarIcon;
      :goto_1
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/statusbar/IStatusBar$Stub;->setIcon(ILcom/android/internal/statusbar/StatusBarIcon;)V
  
      goto :goto_0
  
+     .line 56
      .end local v1           #_arg1:Lcom/android/internal/statusbar/StatusBarIcon;
      :cond_0
      const/4 v1, 0x0
***************
*** 185,190 ****
--- 205,211 ----
      .restart local v1       #_arg1:Lcom/android/internal/statusbar/StatusBarIcon;
      goto :goto_1
  
+     .line 63
      .end local v0           #_arg0:I
      .end local v1           #_arg1:Lcom/android/internal/statusbar/StatusBarIcon;
      :sswitch_2
***************
*** 192,216 ****
--- 213,242 ----
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 65
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 66
      .restart local v0       #_arg0:I
      invoke-virtual {p0, v0}, Lcom/android/internal/statusbar/IStatusBar$Stub;->removeIcon(I)V
  
      goto :goto_0
  
+     .line 71
      .end local v0           #_arg0:I
      :sswitch_3
      const-string v4, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 73
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 75
      .local v0, _arg0:Landroid/os/IBinder;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 218,223 ****
--- 244,250 ----
  
      if-eqz v4, :cond_1
  
+     .line 76
      sget-object v4, Lcom/android/internal/statusbar/StatusBarNotification;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v4, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 226,237 ****
--- 253,266 ----
  
      check-cast v1, Lcom/android/internal/statusbar/StatusBarNotification;
  
+     .line 81
      .local v1, _arg1:Lcom/android/internal/statusbar/StatusBarNotification;
      :goto_2
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/statusbar/IStatusBar$Stub;->addNotification(Landroid/os/IBinder;Lcom/android/internal/statusbar/StatusBarNotification;)V
  
      goto :goto_0
  
+     .line 79
      .end local v1           #_arg1:Lcom/android/internal/statusbar/StatusBarNotification;
      :cond_1
      const/4 v1, 0x0
***************
*** 239,244 ****
--- 268,274 ----
      .restart local v1       #_arg1:Lcom/android/internal/statusbar/StatusBarNotification;
      goto :goto_2
  
+     .line 86
      .end local v0           #_arg0:Landroid/os/IBinder;
      .end local v1           #_arg1:Lcom/android/internal/statusbar/StatusBarNotification;
      :sswitch_4
***************
*** 246,255 ****
--- 276,287 ----
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 88
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 90
      .restart local v0       #_arg0:Landroid/os/IBinder;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 257,262 ****
--- 289,295 ----
  
      if-eqz v4, :cond_2
  
+     .line 91
      sget-object v4, Lcom/android/internal/statusbar/StatusBarNotification;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v4, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 265,276 ****
--- 298,311 ----
  
      check-cast v1, Lcom/android/internal/statusbar/StatusBarNotification;
  
+     .line 96
      .restart local v1       #_arg1:Lcom/android/internal/statusbar/StatusBarNotification;
      :goto_3
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/statusbar/IStatusBar$Stub;->updateNotification(Landroid/os/IBinder;Lcom/android/internal/statusbar/StatusBarNotification;)V
  
      goto :goto_0
  
+     .line 94
      .end local v1           #_arg1:Lcom/android/internal/statusbar/StatusBarNotification;
      :cond_2
      const/4 v1, 0x0
***************
*** 278,283 ****
--- 313,319 ----
      .restart local v1       #_arg1:Lcom/android/internal/statusbar/StatusBarNotification;
      goto :goto_3
  
+     .line 101
      .end local v0           #_arg0:Landroid/os/IBinder;
      .end local v1           #_arg1:Lcom/android/internal/statusbar/StatusBarNotification;
      :sswitch_5
***************
*** 285,353 ****
--- 321,403 ----
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 103
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 104
      .restart local v0       #_arg0:Landroid/os/IBinder;
      invoke-virtual {p0, v0}, Lcom/android/internal/statusbar/IStatusBar$Stub;->removeNotification(Landroid/os/IBinder;)V
  
      goto :goto_0
  
+     .line 109
      .end local v0           #_arg0:Landroid/os/IBinder;
      :sswitch_6
      const-string v4, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 111
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 112
      .local v0, _arg0:I
      invoke-virtual {p0, v0}, Lcom/android/internal/statusbar/IStatusBar$Stub;->disable(I)V
  
      goto/16 :goto_0
  
+     .line 117
      .end local v0           #_arg0:I
      :sswitch_7
      const-string v4, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 118
      invoke-virtual {p0}, Lcom/android/internal/statusbar/IStatusBar$Stub;->animateExpand()V
  
      goto/16 :goto_0
  
+     .line 123
      :sswitch_8
      const-string v4, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 124
      invoke-virtual {p0}, Lcom/android/internal/statusbar/IStatusBar$Stub;->animateCollapse()V
  
      goto/16 :goto_0
  
+     .line 129
      :sswitch_9
      const-string v4, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 131
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 132
      .restart local v0       #_arg0:I
      invoke-virtual {p0, v0}, Lcom/android/internal/statusbar/IStatusBar$Stub;->setSystemUiVisibility(I)V
  
      goto/16 :goto_0
  
+     .line 137
      .end local v0           #_arg0:I
      :sswitch_a
      const-string v5, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 139
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
***************
*** 356,361 ****
--- 406,412 ----
  
      move v0, v3
  
+     .line 140
      .local v0, _arg0:Z
      :goto_4
      invoke-virtual {p0, v0}, Lcom/android/internal/statusbar/IStatusBar$Stub;->topAppWindowChanged(Z)V
***************
*** 366,397 ****
--- 417,455 ----
      :cond_3
      move v0, v4
  
+     .line 139
      goto :goto_4
  
+     .line 145
      :sswitch_b
      const-string v4, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 147
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 149
      .local v0, _arg0:Landroid/os/IBinder;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
+     .line 151
      .local v1, _arg1:I
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
+     .line 152
      .local v2, _arg2:I
      invoke-virtual {p0, v0, v1, v2}, Lcom/android/internal/statusbar/IStatusBar$Stub;->setImeWindowStatus(Landroid/os/IBinder;II)V
  
      goto/16 :goto_0
  
+     .line 157
      .end local v0           #_arg0:Landroid/os/IBinder;
      .end local v1           #_arg1:I
      .end local v2           #_arg2:I
***************
*** 400,405 ****
--- 458,464 ----
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 159
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
***************
*** 408,413 ****
--- 467,473 ----
  
      move v0, v3
  
+     .line 161
      .local v0, _arg0:Z
      :goto_5
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
***************
*** 418,423 ****
--- 478,484 ----
  
      move v1, v3
  
+     .line 162
      .local v1, _arg1:Z
      :goto_6
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/statusbar/IStatusBar$Stub;->setHardKeyboardStatus(ZZ)V
***************
*** 429,452 ****
--- 490,518 ----
      :cond_4
      move v0, v4
  
+     .line 159
      goto :goto_5
  
      .restart local v0       #_arg0:Z
      :cond_5
      move v1, v4
  
+     .line 161
      goto :goto_6
  
+     .line 167
      .end local v0           #_arg0:Z
      :sswitch_d
      const-string v4, "com.android.internal.statusbar.IStatusBar"
  
      invoke-virtual {p2, v4}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 168
      invoke-virtual {p0}, Lcom/android/internal/statusbar/IStatusBar$Stub;->toggleRecentApps()V
  
      goto/16 :goto_0
  
+     .line 39
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_1
