*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/drm/mobile1/DrmRawContent$DrmInputStream.smali	2012-03-23 19:37:16.851082361 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/drm/mobile1/DrmRawContent$DrmInputStream.smali.original	2012-03-23 19:34:45.617324617 -0500
***************
*** 33,52 ****
--- 33,57 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 266
      iput-object p1, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->this$0:Landroid/drm/mobile1/DrmRawContent;
  
      invoke-direct {p0}, Ljava/io/InputStream;-><init>()V
  
+     .line 274
      iput-boolean v0, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->isClosed:Z
  
+     .line 275
      iput v0, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->offset:I
  
+     .line 276
      const/4 v0, 0x1
  
      new-array v0, v0, [B
  
      iput-object v0, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->b:[B
  
+     .line 277
      return-void
  .end method
  
***************
*** 61,66 ****
--- 66,72 ----
      .end annotation
  
      .prologue
+     .line 284
      iget-object v2, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->this$0:Landroid/drm/mobile1/DrmRawContent;
  
      #calls: Landroid/drm/mobile1/DrmRawContent;->nativeGetContentLength()I
***************
*** 68,102 ****
--- 74,116 ----
  
      move-result v1
  
+     .line 286
      .local v1, len:I
      const/4 v2, -0x1
  
      if-ne v2, v1, :cond_0
  
+     .line 287
      new-instance v2, Ljava/io/IOException;
  
      invoke-direct {v2}, Ljava/io/IOException;-><init>()V
  
      throw v2
  
+     .line 290
      :cond_0
      const/4 v2, -0x3
  
      if-ne v2, v1, :cond_2
  
+     .line 291
      const/4 v0, 0x0
  
+     .line 297
      :cond_1
      return v0
  
+     .line 293
      :cond_2
      iget v2, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->offset:I
  
      sub-int v0, v1, v2
  
+     .line 294
      .local v0, availableLen:I
      if-gez v0, :cond_1
  
+     .line 295
      new-instance v2, Ljava/io/IOException;
  
      invoke-direct {v2}, Ljava/io/IOException;-><init>()V
***************
*** 108,117 ****
--- 122,133 ----
      .locals 1
  
      .prologue
+     .line 378
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->isClosed:Z
  
+     .line 379
      return-void
  .end method
  
***************
*** 120,125 ****
--- 136,142 ----
      .parameter "readlimit"
  
      .prologue
+     .line 358
      return-void
  .end method
  
***************
*** 127,132 ****
--- 144,150 ----
      .locals 1
  
      .prologue
+     .line 351
      const/4 v0, 0x0
  
      return v0
***************
*** 145,150 ****
--- 163,169 ----
  
      const/4 v1, -0x1
  
+     .line 306
      iget-object v2, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->b:[B
  
      const/4 v3, 0x1
***************
*** 153,161 ****
--- 172,182 ----
  
      move-result v0
  
+     .line 308
      .local v0, res:I
      if-ne v1, v0, :cond_0
  
+     .line 311
      :goto_0
      return v1
  
***************
*** 179,184 ****
--- 200,206 ----
      .end annotation
  
      .prologue
+     .line 318
      const/4 v0, 0x0
  
      array-length v1, p1
***************
*** 204,217 ****
--- 226,242 ----
      .prologue
      const/4 v0, -0x1
  
+     .line 325
      if-nez p1, :cond_0
  
+     .line 326
      new-instance v0, Ljava/lang/NullPointerException;
  
      invoke-direct {v0}, Ljava/lang/NullPointerException;-><init>()V
  
      throw v0
  
+     .line 327
      :cond_0
      if-ltz p2, :cond_1
  
***************
*** 223,228 ****
--- 248,254 ----
  
      if-le v1, v2, :cond_2
  
+     .line 328
      :cond_1
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 230,235 ****
--- 256,262 ----
  
      throw v0
  
+     .line 329
      :cond_2
      const/4 v1, 0x1
  
***************
*** 237,257 ****
--- 264,289 ----
  
      if-ne v1, v2, :cond_3
  
+     .line 330
      new-instance v0, Ljava/io/IOException;
  
      invoke-direct {v0}, Ljava/io/IOException;-><init>()V
  
      throw v0
  
+     .line 332
      :cond_3
      if-nez p3, :cond_5
  
+     .line 333
      const/4 v0, 0x0
  
+     .line 344
      :cond_4
      :goto_0
      return v0
  
+     .line 335
      :cond_5
      iget-object v1, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->this$0:Landroid/drm/mobile1/DrmRawContent;
  
***************
*** 262,280 ****
--- 294,316 ----
  
      move-result p3
  
+     .line 337
      if-ne v0, p3, :cond_6
  
+     .line 338
      new-instance v0, Ljava/io/IOException;
  
      invoke-direct {v0}, Ljava/io/IOException;-><init>()V
  
      throw v0
  
+     .line 339
      :cond_6
      const/4 v1, -0x2
  
      if-eq v1, p3, :cond_4
  
+     .line 342
      iget v0, p0, Landroid/drm/mobile1/DrmRawContent$DrmInputStream;->offset:I
  
      add-int/2addr v0, p3
***************
*** 283,288 ****
--- 319,325 ----
  
      move v0, p3
  
+     .line 344
      goto :goto_0
  .end method
  
***************
*** 295,300 ****
--- 332,338 ----
      .end annotation
  
      .prologue
+     .line 364
      new-instance v0, Ljava/io/IOException;
  
      invoke-direct {v0}, Ljava/io/IOException;-><init>()V
***************
*** 312,317 ****
--- 350,356 ----
      .end annotation
  
      .prologue
+     .line 371
      const-wide/16 v0, 0x0
  
      return-wide v0
