*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/sqlite/DatabaseObjectNotClosedException.smali	2012-03-23 19:36:10.677875695 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/sqlite/DatabaseObjectNotClosedException.smali.original	2012-03-23 19:34:45.006313476 -0500
***************
*** 12,20 ****
--- 12,22 ----
      .locals 1
  
      .prologue
+     .line 29
      const-string v0, "Application did not close the cursor or database object that was opened here"
  
      invoke-direct {p0, v0}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
+     .line 30
      return-void
  .end method
