*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/sqlite/SQLiteStatement.smali	2012-03-23 19:36:11.134884028 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/sqlite/SQLiteStatement.smali.original	2012-03-23 19:34:45.009313531 -0500
***************
*** 29,38 ****
--- 29,40 ----
      .parameter "bindArgs"
  
      .prologue
+     .line 59
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, p2, p3, v0}, Landroid/database/sqlite/SQLiteProgram;-><init>(Landroid/database/sqlite/SQLiteDatabase;Ljava/lang/String;[Ljava/lang/Object;Z)V
  
+     .line 60
      return-void
  .end method
  
***************
*** 43,56 ****
--- 45,61 ----
      .prologue
      const/4 v5, 0x2
  
+     .line 216
      const/4 v3, 0x0
  
      iput v3, p0, Landroid/database/sqlite/SQLiteStatement;->mState:I
  
+     .line 218
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      invoke-virtual {v3}, Landroid/database/sqlite/SQLiteDatabase;->verifyDbIsOpen()V
  
+     .line 219
      iget v3, p0, Landroid/database/sqlite/SQLiteStatement;->mStatementType:I
  
      and-int/lit8 v3, v3, 0x40
***************
*** 65,92 ****
--- 70,103 ----
  
      move-result-object v0
  
+     .line 222
      .local v0, db:Landroid/database/sqlite/SQLiteDatabase;
      :goto_0
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      iput-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mOrigDb:Landroid/database/sqlite/SQLiteDatabase;
  
+     .line 223
      iput-object v0, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
+     .line 224
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      iget v3, v3, Landroid/database/sqlite/SQLiteDatabase;->mNativeHandle:I
  
      invoke-virtual {p0, v3}, Landroid/database/sqlite/SQLiteStatement;->setNativeHandle(I)V
  
+     .line 225
      if-nez p1, :cond_3
  
+     .line 226
      invoke-static {}, Ldalvik/system/BlockGuard;->getThreadPolicy()Ldalvik/system/BlockGuard$Policy;
  
      move-result-object v3
  
      invoke-interface {v3}, Ldalvik/system/BlockGuard$Policy;->onWriteToDisk()V
  
+     .line 237
      :goto_1
      iget v3, p0, Landroid/database/sqlite/SQLiteStatement;->mStatementType:I
  
***************
*** 96,101 ****
--- 107,113 ----
  
      if-ne v3, v4, :cond_4
  
+     .line 238
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      invoke-virtual {v3}, Landroid/database/sqlite/SQLiteDatabase;->isDbLockedByCurrentThread()Z
***************
*** 104,113 ****
--- 116,127 ----
  
      if-nez v3, :cond_0
  
+     .line 241
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      invoke-virtual {v3}, Landroid/database/sqlite/SQLiteDatabase;->setTransactionUsingExecSqlFlag()V
  
+     .line 252
      :cond_0
      :goto_2
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
***************
*** 118,147 ****
--- 132,169 ----
  
      if-nez v3, :cond_1
  
+     .line 253
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      iget-object v4, p0, Landroid/database/sqlite/SQLiteStatement;->mSql:Ljava/lang/String;
  
      invoke-virtual {v3, v4}, Landroid/database/sqlite/SQLiteDatabase;->lock(Ljava/lang/String;)V
  
+     .line 254
      iput v5, p0, Landroid/database/sqlite/SQLiteStatement;->mState:I
  
+     .line 257
      :cond_1
      invoke-virtual {p0}, Landroid/database/sqlite/SQLiteStatement;->acquireReference()V
  
+     .line 258
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v1
  
+     .line 259
      .local v1, startTime:J
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      invoke-virtual {v3}, Landroid/database/sqlite/SQLiteDatabase;->closePendingStatements()V
  
+     .line 260
      invoke-virtual {p0}, Landroid/database/sqlite/SQLiteStatement;->compileAndbindAllArgs()V
  
+     .line 261
      return-wide v1
  
+     .line 219
      .end local v0           #db:Landroid/database/sqlite/SQLiteDatabase;
      .end local v1           #startTime:J
      :cond_2
***************
*** 149,154 ****
--- 171,177 ----
  
      goto :goto_0
  
+     .line 228
      .restart local v0       #db:Landroid/database/sqlite/SQLiteDatabase;
      :cond_3
      invoke-static {}, Ldalvik/system/BlockGuard;->getThreadPolicy()Ldalvik/system/BlockGuard$Policy;
***************
*** 159,164 ****
--- 182,188 ----
  
      goto :goto_1
  
+     .line 243
      :cond_4
      iget v3, p0, Landroid/database/sqlite/SQLiteStatement;->mStatementType:I
  
***************
*** 166,171 ****
--- 190,196 ----
  
      if-ne v3, v5, :cond_0
  
+     .line 246
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      invoke-virtual {v3}, Landroid/database/sqlite/SQLiteDatabase;->inTransaction()Z
***************
*** 174,183 ****
--- 199,210 ----
  
      if-nez v3, :cond_0
  
+     .line 247
      iget-object v3, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      invoke-virtual {v3}, Landroid/database/sqlite/SQLiteDatabase;->beginTransactionNonExclusive()V
  
+     .line 248
      const/4 v3, 0x1
  
      iput v3, p0, Landroid/database/sqlite/SQLiteStatement;->mState:I
***************
*** 212,225 ****
--- 239,255 ----
      .locals 2
  
      .prologue
+     .line 268
      invoke-virtual {p0}, Landroid/database/sqlite/SQLiteStatement;->releaseReference()V
  
+     .line 269
      iget v0, p0, Landroid/database/sqlite/SQLiteStatement;->mState:I
  
      const/4 v1, 0x1
  
      if-ne v0, v1, :cond_3
  
+     .line 271
      :try_start_0
      iget-object v0, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
***************
*** 227,236 ****
--- 257,268 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 273
      iget-object v0, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      invoke-virtual {v0}, Landroid/database/sqlite/SQLiteDatabase;->endTransaction()V
  
+     .line 278
      :cond_0
      :goto_0
      iget v0, p0, Landroid/database/sqlite/SQLiteStatement;->mStatementType:I
***************
*** 249,276 ****
--- 281,315 ----
  
      if-ne v0, v1, :cond_2
  
+     .line 282
      :cond_1
      iget-object v0, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      invoke-virtual {v0}, Landroid/database/sqlite/SQLiteDatabase;->resetTransactionUsingExecSqlFlag()V
  
+     .line 284
      :cond_2
      invoke-virtual {p0}, Landroid/database/sqlite/SQLiteStatement;->clearBindings()V
  
+     .line 287
      invoke-virtual {p0}, Landroid/database/sqlite/SQLiteStatement;->release()V
  
+     .line 289
      iget-object v0, p0, Landroid/database/sqlite/SQLiteStatement;->mOrigDb:Landroid/database/sqlite/SQLiteDatabase;
  
      iput-object v0, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
+     .line 290
      iget-object v0, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      iget v0, v0, Landroid/database/sqlite/SQLiteDatabase;->mNativeHandle:I
  
      invoke-virtual {p0, v0}, Landroid/database/sqlite/SQLiteStatement;->setNativeHandle(I)V
  
+     .line 291
      return-void
  
+     .line 273
      :catchall_0
      move-exception v0
  
***************
*** 280,285 ****
--- 319,325 ----
  
      throw v0
  
+     .line 275
      :cond_3
      iget v0, p0, Landroid/database/sqlite/SQLiteStatement;->mState:I
  
***************
*** 287,292 ****
--- 327,333 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 276
      iget-object v0, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
      invoke-virtual {v0}, Landroid/database/sqlite/SQLiteDatabase;->unlock()V
***************
*** 298,303 ****
--- 339,345 ----
      .locals 2
  
      .prologue
+     .line 120
      iget v0, p0, Landroid/database/sqlite/SQLiteStatement;->mStatementType:I
  
      and-int/lit8 v0, v0, 0xf
***************
*** 314,319 ****
--- 356,362 ----
  
      if-ne v0, v1, :cond_1
  
+     .line 124
      :cond_0
      iget-object v0, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
***************
*** 321,326 ****
--- 364,370 ----
  
      invoke-virtual {v0, v1}, Landroid/database/sqlite/SQLiteDatabase;->setLastSqlStatement(Ljava/lang/String;)V
  
+     .line 126
      :cond_1
      return-void
  .end method
***************
*** 331,338 ****
--- 375,384 ----
      .locals 0
  
      .prologue
+     .line 70
      invoke-virtual {p0}, Landroid/database/sqlite/SQLiteStatement;->executeUpdateDelete()I
  
+     .line 71
      return-void
  .end method
  
***************
*** 340,358 ****
--- 386,408 ----
      .locals 2
  
      .prologue
+     .line 111
      :try_start_0
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->saveSqlAsLastSqlStatement()V
  
+     .line 112
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/database/sqlite/SQLiteStatement;->acquireAndLock(Z)J
  
+     .line 113
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->native_executeInsert()J
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result-wide v0
  
+     .line 115
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->releaseAndUnlock()V
  
      return-wide v0
***************
*** 369,383 ****
--- 419,437 ----
      .locals 2
  
      .prologue
+     .line 83
      :try_start_0
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->saveSqlAsLastSqlStatement()V
  
+     .line 84
      const/4 v1, 0x0
  
      invoke-direct {p0, v1}, Landroid/database/sqlite/SQLiteStatement;->acquireAndLock(Z)J
  
+     .line 85
      const/4 v0, 0x0
  
+     .line 86
      .local v0, numChanges:I
      iget v1, p0, Landroid/database/sqlite/SQLiteStatement;->mStatementType:I
  
***************
*** 385,401 ****
--- 439,458 ----
  
      if-lez v1, :cond_0
  
+     .line 90
      iget-object v1, p0, Landroid/database/sqlite/SQLiteStatement;->mSql:Ljava/lang/String;
  
      invoke-direct {p0, v1}, Landroid/database/sqlite/SQLiteStatement;->native_executeSql(Ljava/lang/String;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 96
      :goto_0
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->releaseAndUnlock()V
  
      return v0
  
+     .line 92
      :cond_0
      :try_start_1
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->native_execute()I
***************
*** 406,411 ****
--- 463,469 ----
  
      goto :goto_0
  
+     .line 96
      .end local v0           #numChanges:I
      :catchall_0
      move-exception v1
***************
*** 419,424 ****
--- 477,483 ----
      .locals 8
  
      .prologue
+     .line 183
      const/4 v5, 0x1
  
      :try_start_0
***************
*** 426,436 ****
--- 485,497 ----
  
      move-result-wide v3
  
+     .line 184
      .local v3, timeStart:J
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->native_1x1_blob_ashmem()Landroid/os/ParcelFileDescriptor;
  
      move-result-object v2
  
+     .line 185
      .local v2, retValue:Landroid/os/ParcelFileDescriptor;
      iget-object v5, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
***************
*** 442,447 ****
--- 503,509 ----
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Landroid/database/sqlite/SQLiteDoneException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 195
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->releaseAndUnlock()V
  
      .end local v2           #retValue:Landroid/os/ParcelFileDescriptor;
***************
*** 449,477 ****
      :goto_0
      return-object v2
  
      :catch_0
      move-exception v1
  
      .local v1, ex:Ljava/io/IOException;
      :try_start_1
      const-string v5, "SQLiteStatement"
  
!     const-string v6, "simpleQueryForBlobFileDescriptor() failed"
  
      invoke-static {v5, v6, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
      const/4 v2, 0x0
  
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->releaseAndUnlock()V
  
      goto :goto_0
  
      .end local v1           #ex:Ljava/io/IOException;
      :catch_1
      move-exception v0
  
      .local v0, e:Landroid/database/sqlite/SQLiteDoneException;
      :try_start_2
      new-instance v5, Landroid/database/sqlite/SQLiteDoneException;
--- 511,545 ----
      :goto_0
      return-object v2
  
+     .line 187
      :catch_0
      move-exception v1
  
+     .line 188
      .local v1, ex:Ljava/io/IOException;
      :try_start_1
      const-string v5, "SQLiteStatement"
  
!     const-string/jumbo v6, "simpleQueryForBlobFileDescriptor() failed"
  
      invoke-static {v5, v6, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 189
      const/4 v2, 0x0
  
+     .line 195
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->releaseAndUnlock()V
  
      goto :goto_0
  
+     .line 190
      .end local v1           #ex:Ljava/io/IOException;
      :catch_1
      move-exception v0
  
+     .line 191
      .local v0, e:Landroid/database/sqlite/SQLiteDoneException;
      :try_start_2
      new-instance v5, Landroid/database/sqlite/SQLiteDoneException;
***************
*** 502,507 ****
--- 570,576 ----
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
  
+     .line 195
      .end local v0           #e:Landroid/database/sqlite/SQLiteDoneException;
      :catchall_0
      move-exception v5
***************
*** 515,520 ****
--- 584,590 ----
      .locals 8
  
      .prologue
+     .line 137
      const/4 v5, 0x1
  
      :try_start_0
***************
*** 522,532 ****
--- 592,604 ----
  
      move-result-wide v3
  
+     .line 138
      .local v3, timeStart:J
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->native_1x1_long()J
  
      move-result-wide v1
  
+     .line 139
      .local v1, retValue:J
      iget-object v5, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
***************
*** 537,551 ****
--- 609,626 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Landroid/database/sqlite/SQLiteDoneException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 146
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->releaseAndUnlock()V
  
      return-wide v1
  
+     .line 141
      .end local v1           #retValue:J
      .end local v3           #timeStart:J
      :catch_0
      move-exception v0
  
+     .line 142
      .local v0, e:Landroid/database/sqlite/SQLiteDoneException;
      :try_start_1
      new-instance v5, Landroid/database/sqlite/SQLiteDoneException;
***************
*** 576,581 ****
--- 651,657 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 146
      .end local v0           #e:Landroid/database/sqlite/SQLiteDoneException;
      :catchall_0
      move-exception v5
***************
*** 589,594 ****
--- 665,671 ----
      .locals 7
  
      .prologue
+     .line 160
      const/4 v4, 0x1
  
      :try_start_0
***************
*** 596,606 ****
--- 673,685 ----
  
      move-result-wide v2
  
+     .line 161
      .local v2, timeStart:J
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->native_1x1_string()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 162
      .local v1, retValue:Ljava/lang/String;
      iget-object v4, p0, Landroid/database/sqlite/SQLiteStatement;->mDatabase:Landroid/database/sqlite/SQLiteDatabase;
  
***************
*** 611,625 ****
--- 690,707 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Landroid/database/sqlite/SQLiteDoneException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 169
      invoke-direct {p0}, Landroid/database/sqlite/SQLiteStatement;->releaseAndUnlock()V
  
      return-object v1
  
+     .line 164
      .end local v1           #retValue:Ljava/lang/String;
      .end local v2           #timeStart:J
      :catch_0
      move-exception v0
  
+     .line 165
      .local v0, e:Landroid/database/sqlite/SQLiteDoneException;
      :try_start_1
      new-instance v4, Landroid/database/sqlite/SQLiteDoneException;
***************
*** 650,655 ****
--- 732,738 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 169
      .end local v0           #e:Landroid/database/sqlite/SQLiteDoneException;
      :catchall_0
      move-exception v4
