*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accessibilityservice/AccessibilityService.smali	2012-03-23 19:37:16.160069761 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accessibilityservice/AccessibilityService.smali.original	2012-03-23 19:34:45.611324509 -0500
***************
*** 30,37 ****
--- 30,39 ----
      .locals 0
  
      .prologue
+     .line 194
      invoke-direct {p0}, Landroid/app/Service;-><init>()V
  
+     .line 292
      return-void
  .end method
  
***************
*** 41,46 ****
--- 43,49 ----
      .parameter "x1"
  
      .prologue
+     .line 194
      iput p1, p0, Landroid/accessibilityservice/AccessibilityService;->mConnectionId:I
  
      return p1
***************
*** 50,55 ****
--- 53,59 ----
      .locals 4
  
      .prologue
+     .line 268
      invoke-static {}, Landroid/view/accessibility/AccessibilityInteractionClient;->getInstance()Landroid/view/accessibility/AccessibilityInteractionClient;
  
      move-result-object v2
***************
*** 60,65 ****
--- 64,70 ----
  
      move-result-object v0
  
+     .line 270
      .local v0, connection:Landroid/accessibilityservice/IAccessibilityServiceConnection;
      iget-object v2, p0, Landroid/accessibilityservice/AccessibilityService;->mInfo:Landroid/accessibilityservice/AccessibilityServiceInfo;
  
***************
*** 67,72 ****
--- 72,78 ----
  
      if-eqz v0, :cond_0
  
+     .line 272
      :try_start_0
      iget-object v2, p0, Landroid/accessibilityservice/AccessibilityService;->mInfo:Landroid/accessibilityservice/AccessibilityServiceInfo;
  
***************
*** 74,86 ****
--- 80,95 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 277
      :cond_0
      :goto_0
      return-void
  
+     .line 273
      :catch_0
      move-exception v1
  
+     .line 274
      .local v1, re:Landroid/os/RemoteException;
      const-string v2, "AccessibilityService"
  
***************
*** 101,106 ****
--- 110,116 ----
      .parameter "intent"
  
      .prologue
+     .line 285
      new-instance v0, Landroid/accessibilityservice/AccessibilityService$IEventListenerWrapper;
  
      invoke-direct {v0, p0, p0}, Landroid/accessibilityservice/AccessibilityService$IEventListenerWrapper;-><init>(Landroid/accessibilityservice/AccessibilityService;Landroid/accessibilityservice/AccessibilityService;)V
***************
*** 115,120 ****
--- 125,131 ----
      .locals 0
  
      .prologue
+     .line 247
      return-void
  .end method
  
***************
*** 123,131 ****
--- 134,145 ----
      .parameter "info"
  
      .prologue
+     .line 258
      iput-object p1, p0, Landroid/accessibilityservice/AccessibilityService;->mInfo:Landroid/accessibilityservice/AccessibilityServiceInfo;
  
+     .line 259
      invoke-direct {p0}, Landroid/accessibilityservice/AccessibilityService;->sendServiceInfo()V
  
+     .line 260
      return-void
  .end method
