*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/ActionMenuItem.smali	2012-03-23 19:37:54.288765036 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/ActionMenuItem.smali.original	2012-03-23 19:34:46.402338934 -0500
***************
*** 61,88 ****
--- 61,98 ----
      .parameter "title"
  
      .prologue
+     .line 60
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 44
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIconResId:I
  
+     .line 52
      const/16 v0, 0x10
  
      iput v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
+     .line 61
      iput-object p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mContext:Landroid/content/Context;
  
+     .line 62
      iput p3, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mId:I
  
+     .line 63
      iput p2, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mGroup:I
  
+     .line 64
      iput p4, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mCategoryOrder:I
  
+     .line 65
      iput p5, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mOrdering:I
  
+     .line 66
      iput-object p6, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mTitle:Ljava/lang/CharSequence;
  
+     .line 67
      return-void
  .end method
  
***************
*** 92,97 ****
--- 102,108 ----
      .locals 1
  
      .prologue
+     .line 264
      const/4 v0, 0x0
  
      return v0
***************
*** 101,106 ****
--- 112,118 ----
      .locals 1
  
      .prologue
+     .line 259
      const/4 v0, 0x0
  
      return v0
***************
*** 110,115 ****
--- 122,128 ----
      .locals 1
  
      .prologue
+     .line 243
      const/4 v0, 0x0
  
      return-object v0
***************
*** 119,124 ****
--- 132,138 ----
      .locals 1
  
      .prologue
+     .line 233
      const/4 v0, 0x0
  
      return-object v0
***************
*** 128,133 ****
--- 142,148 ----
      .locals 1
  
      .prologue
+     .line 70
      iget-char v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mShortcutAlphabeticChar:C
  
      return v0
***************
*** 137,142 ****
--- 152,158 ----
      .locals 1
  
      .prologue
+     .line 74
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mGroup:I
  
      return v0
***************
*** 146,151 ****
--- 162,168 ----
      .locals 1
  
      .prologue
+     .line 78
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIconDrawable:Landroid/graphics/drawable/Drawable;
  
      return-object v0
***************
*** 155,160 ****
--- 172,178 ----
      .locals 1
  
      .prologue
+     .line 82
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIntent:Landroid/content/Intent;
  
      return-object v0
***************
*** 164,169 ****
--- 182,188 ----
      .locals 1
  
      .prologue
+     .line 86
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mId:I
  
      return v0
***************
*** 173,178 ****
--- 192,198 ----
      .locals 1
  
      .prologue
+     .line 90
      const/4 v0, 0x0
  
      return-object v0
***************
*** 182,187 ****
--- 202,208 ----
      .locals 1
  
      .prologue
+     .line 94
      iget-char v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mShortcutNumericChar:C
  
      return v0
***************
*** 191,196 ****
--- 212,218 ----
      .locals 1
  
      .prologue
+     .line 98
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mOrdering:I
  
      return v0
***************
*** 200,205 ****
--- 222,228 ----
      .locals 1
  
      .prologue
+     .line 102
      const/4 v0, 0x0
  
      return-object v0
***************
*** 209,214 ****
--- 232,238 ----
      .locals 1
  
      .prologue
+     .line 106
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mTitle:Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 218,223 ****
--- 242,248 ----
      .locals 1
  
      .prologue
+     .line 110
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mTitleCondensed:Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 227,232 ****
--- 252,258 ----
      .locals 1
  
      .prologue
+     .line 114
      const/4 v0, 0x0
  
      return v0
***************
*** 238,243 ****
--- 264,270 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 212
      iget-object v1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mClickListener:Landroid/view/MenuItem$OnMenuItemClickListener;
  
      if-eqz v1, :cond_0
***************
*** 250,263 ****
--- 277,293 ----
  
      if-eqz v1, :cond_0
  
+     .line 221
      :goto_0
      return v0
  
+     .line 216
      :cond_0
      iget-object v1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIntent:Landroid/content/Intent;
  
      if-eqz v1, :cond_1
  
+     .line 217
      iget-object v1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mContext:Landroid/content/Context;
  
      iget-object v2, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIntent:Landroid/content/Intent;
***************
*** 266,271 ****
--- 296,302 ----
  
      goto :goto_0
  
+     .line 221
      :cond_1
      const/4 v0, 0x0
  
***************
*** 276,281 ****
--- 307,313 ----
      .locals 1
  
      .prologue
+     .line 269
      const/4 v0, 0x0
  
      return v0
***************
*** 285,290 ****
--- 317,323 ----
      .locals 1
  
      .prologue
+     .line 118
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
      and-int/lit8 v0, v0, 0x1
***************
*** 306,311 ****
--- 339,345 ----
      .locals 1
  
      .prologue
+     .line 122
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
      and-int/lit8 v0, v0, 0x2
***************
*** 327,332 ****
--- 361,367 ----
      .locals 1
  
      .prologue
+     .line 126
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
      and-int/lit8 v0, v0, 0x10
***************
*** 348,353 ****
--- 383,389 ----
      .locals 1
  
      .prologue
+     .line 130
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
      and-int/lit8 v0, v0, 0x8
***************
*** 370,375 ****
--- 406,412 ----
      .parameter "actionProvider"
  
      .prologue
+     .line 248
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 382,387 ****
--- 419,425 ----
      .parameter "resId"
  
      .prologue
+     .line 238
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 394,399 ****
--- 432,438 ----
      .parameter "actionView"
  
      .prologue
+     .line 229
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 406,413 ****
--- 445,454 ----
      .parameter "alphaChar"
  
      .prologue
+     .line 134
      iput-char p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mShortcutAlphabeticChar:C
  
+     .line 135
      return-object p0
  .end method
  
***************
*** 416,421 ****
--- 457,463 ----
      .parameter "checkable"
  
      .prologue
+     .line 139
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
      and-int/lit8 v1, v0, -0x2
***************
*** 429,436 ****
--- 471,480 ----
  
      iput v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
+     .line 140
      return-object p0
  
+     .line 139
      :cond_0
      const/4 v0, 0x0
  
***************
*** 442,447 ****
--- 486,492 ----
      .parameter "checked"
  
      .prologue
+     .line 149
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
      and-int/lit8 v1, v0, -0x3
***************
*** 455,462 ****
--- 500,509 ----
  
      iput v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
+     .line 150
      return-object p0
  
+     .line 149
      :cond_0
      const/4 v0, 0x0
  
***************
*** 468,473 ****
--- 515,521 ----
      .parameter "enabled"
  
      .prologue
+     .line 154
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
      and-int/lit8 v1, v0, -0x11
***************
*** 481,488 ****
--- 529,538 ----
  
      iput v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
+     .line 155
      return-object p0
  
+     .line 154
      :cond_0
      const/4 v0, 0x0
  
***************
*** 494,499 ****
--- 544,550 ----
      .parameter "exclusive"
  
      .prologue
+     .line 144
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
      and-int/lit8 v1, v0, -0x5
***************
*** 507,514 ****
--- 558,567 ----
  
      iput v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
+     .line 145
      return-object p0
  
+     .line 144
      :cond_0
      const/4 v0, 0x0
  
***************
*** 520,527 ****
--- 573,582 ----
      .parameter "iconRes"
  
      .prologue
+     .line 165
      iput p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIconResId:I
  
+     .line 166
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 534,539 ****
--- 589,595 ----
  
      iput-object v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIconDrawable:Landroid/graphics/drawable/Drawable;
  
+     .line 167
      return-object p0
  .end method
  
***************
*** 542,553 ****
--- 598,612 ----
      .parameter "icon"
  
      .prologue
+     .line 159
      iput-object p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIconDrawable:Landroid/graphics/drawable/Drawable;
  
+     .line 160
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIconResId:I
  
+     .line 161
      return-object p0
  .end method
  
***************
*** 556,563 ****
--- 615,624 ----
      .parameter "intent"
  
      .prologue
+     .line 171
      iput-object p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mIntent:Landroid/content/Intent;
  
+     .line 172
      return-object p0
  .end method
  
***************
*** 566,573 ****
--- 627,636 ----
      .parameter "numericChar"
  
      .prologue
+     .line 176
      iput-char p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mShortcutNumericChar:C
  
+     .line 177
      return-object p0
  .end method
  
***************
*** 576,581 ****
--- 639,645 ----
      .parameter "listener"
  
      .prologue
+     .line 275
      return-object p0
  .end method
  
***************
*** 584,591 ****
--- 648,657 ----
      .parameter "menuItemClickListener"
  
      .prologue
+     .line 181
      iput-object p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mClickListener:Landroid/view/MenuItem$OnMenuItemClickListener;
  
+     .line 182
      return-object p0
  .end method
  
***************
*** 595,604 ****
--- 661,673 ----
      .parameter "alphaChar"
  
      .prologue
+     .line 186
      iput-char p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mShortcutNumericChar:C
  
+     .line 187
      iput-char p2, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mShortcutAlphabeticChar:C
  
+     .line 188
      return-object p0
  .end method
  
***************
*** 607,612 ****
--- 676,682 ----
      .parameter "show"
  
      .prologue
+     .line 226
      return-void
  .end method
  
***************
*** 615,622 ****
--- 685,694 ----
      .parameter "actionEnum"
  
      .prologue
+     .line 253
      invoke-virtual {p0, p1}, Lcom/android/internal/view/menu/ActionMenuItem;->setShowAsAction(I)V
  
+     .line 254
      return-object p0
  .end method
  
***************
*** 625,630 ****
--- 697,703 ----
      .parameter "title"
  
      .prologue
+     .line 197
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 637,642 ****
--- 710,716 ----
  
      iput-object v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mTitle:Ljava/lang/CharSequence;
  
+     .line 198
      return-object p0
  .end method
  
***************
*** 645,652 ****
--- 719,728 ----
      .parameter "title"
  
      .prologue
+     .line 192
      iput-object p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mTitle:Ljava/lang/CharSequence;
  
+     .line 193
      return-object p0
  .end method
  
***************
*** 655,662 ****
--- 731,740 ----
      .parameter "title"
  
      .prologue
+     .line 202
      iput-object p1, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mTitleCondensed:Ljava/lang/CharSequence;
  
+     .line 203
      return-object p0
  .end method
  
***************
*** 665,670 ****
--- 743,749 ----
      .parameter "visible"
  
      .prologue
+     .line 207
      iget v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
      and-int/lit8 v1, v0, 0x8
***************
*** 678,685 ****
--- 757,766 ----
  
      iput v0, p0, Lcom/android/internal/view/menu/ActionMenuItem;->mFlags:I
  
+     .line 208
      return-object p0
  
+     .line 207
      :cond_0
      const/16 v0, 0x8
  
