*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/ActionMenu.smali	2012-03-23 19:37:53.738755008 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/ActionMenu.smali.original	2012-03-23 19:34:46.398338860 -0500
***************
*** 29,44 ****
--- 29,48 ----
      .parameter "context"
  
      .prologue
+     .line 42
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 43
      iput-object p1, p0, Lcom/android/internal/view/menu/ActionMenu;->mContext:Landroid/content/Context;
  
+     .line 44
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
+     .line 45
      return-void
  .end method
  
***************
*** 47,59 ****
--- 51,66 ----
      .parameter "id"
  
      .prologue
+     .line 129
      iget-object v2, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
+     .line 130
      .local v2, items:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/ActionMenuItem;>;"
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
  
      move-result v1
  
+     .line 131
      .local v1, itemCount:I
      const/4 v0, 0x0
  
***************
*** 61,66 ****
--- 68,74 ----
      :goto_0
      if-ge v0, v1, :cond_1
  
+     .line 132
      invoke-virtual {v2, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v3
***************
*** 73,88 ****
--- 81,99 ----
  
      if-ne v3, p1, :cond_0
  
+     .line 137
      .end local v0           #i:I
      :goto_1
      return v0
  
+     .line 131
      .restart local v0       #i:I
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 137
      :cond_1
      const/4 v0, -0x1
  
***************
*** 95,110 ****
--- 106,125 ----
      .parameter "event"
  
      .prologue
+     .line 163
      iget-boolean v4, p0, Lcom/android/internal/view/menu/ActionMenu;->mIsQwerty:Z
  
+     .line 164
      .local v4, qwerty:Z
      iget-object v3, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
+     .line 165
      .local v3, items:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/ActionMenuItem;>;"
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v2
  
+     .line 167
      .local v2, itemCount:I
      const/4 v0, 0x0
  
***************
*** 112,123 ****
--- 127,140 ----
      :goto_0
      if-ge v0, v2, :cond_2
  
+     .line 168
      invoke-virtual {v3, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Lcom/android/internal/view/menu/ActionMenuItem;
  
+     .line 169
      .local v1, item:Lcom/android/internal/view/menu/ActionMenuItem;
      if-eqz v4, :cond_0
  
***************
*** 125,139 ****
--- 142,159 ----
  
      move-result v5
  
+     .line 171
      .local v5, shortcut:C
      :goto_1
      if-ne p1, v5, :cond_1
  
+     .line 175
      .end local v1           #item:Lcom/android/internal/view/menu/ActionMenuItem;
      .end local v5           #shortcut:C
      :goto_2
      return-object v1
  
+     .line 169
      .restart local v1       #item:Lcom/android/internal/view/menu/ActionMenuItem;
      :cond_0
      invoke-virtual {v1}, Lcom/android/internal/view/menu/ActionMenuItem;->getNumericShortcut()C
***************
*** 142,153 ****
--- 162,175 ----
  
      goto :goto_1
  
+     .line 167
      .restart local v5       #shortcut:C
      :cond_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 175
      .end local v1           #item:Lcom/android/internal/view/menu/ActionMenuItem;
      .end local v5           #shortcut:C
      :cond_2
***************
*** 165,170 ****
--- 187,193 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 56
      invoke-virtual {p0, v0, v0, v0, p1}, Lcom/android/internal/view/menu/ActionMenu;->add(IIII)Landroid/view/MenuItem;
  
      move-result-object v0
***************
*** 180,185 ****
--- 203,209 ----
      .parameter "titleRes"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenu;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 205,210 ****
--- 229,235 ----
      .parameter "title"
  
      .prologue
+     .line 64
      new-instance v0, Lcom/android/internal/view/menu/ActionMenuItem;
  
      invoke-virtual {p0}, Lcom/android/internal/view/menu/ActionMenu;->getContext()Landroid/content/Context;
***************
*** 223,233 ****
--- 248,260 ----
  
      invoke-direct/range {v0 .. v6}, Lcom/android/internal/view/menu/ActionMenuItem;-><init>(Landroid/content/Context;IIIILjava/lang/CharSequence;)V
  
+     .line 66
      .local v0, item:Lcom/android/internal/view/menu/ActionMenuItem;
      iget-object v1, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v1, p3, v0}, Ljava/util/ArrayList;->add(ILjava/lang/Object;)V
  
+     .line 67
      return-object v0
  .end method
  
***************
*** 238,243 ****
--- 265,271 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 52
      invoke-virtual {p0, v0, v0, v0, p1}, Lcom/android/internal/view/menu/ActionMenu;->add(IIILjava/lang/CharSequence;)Landroid/view/MenuItem;
  
      move-result-object v0
***************
*** 257,268 ****
--- 285,298 ----
      .parameter "outSpecificItems"
  
      .prologue
+     .line 73
      iget-object v10, p0, Lcom/android/internal/view/menu/ActionMenu;->mContext:Landroid/content/Context;
  
      invoke-virtual {v10}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
  
      move-result-object v7
  
+     .line 74
      .local v7, pm:Landroid/content/pm/PackageManager;
      const/4 v10, 0x0
  
***************
*** 276,281 ****
--- 306,312 ----
  
      move-result-object v6
  
+     .line 76
      .local v6, lri:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
      if-eqz v6, :cond_2
  
***************
*** 283,296 ****
--- 314,330 ----
  
      move-result v3
  
+     .line 78
      .local v3, N:I
      :goto_0
      and-int/lit8 v10, p7, 0x1
  
      if-nez v10, :cond_0
  
+     .line 79
      invoke-virtual {p0, p1}, Lcom/android/internal/view/menu/ActionMenu;->removeGroup(I)V
  
+     .line 82
      :cond_0
      const/4 v4, 0x0
  
***************
*** 298,309 ****
--- 332,345 ----
      :goto_1
      if-ge v4, v3, :cond_4
  
+     .line 83
      invoke-interface {v6, v4}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v8
  
      check-cast v8, Landroid/content/pm/ResolveInfo;
  
+     .line 84
      .local v8, ri:Landroid/content/pm/ResolveInfo;
      new-instance v9, Landroid/content/Intent;
  
***************
*** 316,321 ****
--- 352,358 ----
      :goto_2
      invoke-direct {v9, v10}, Landroid/content/Intent;-><init>(Landroid/content/Intent;)V
  
+     .line 86
      .local v9, rintent:Landroid/content/Intent;
      new-instance v10, Landroid/content/ComponentName;
  
***************
*** 333,338 ****
--- 370,376 ----
  
      invoke-virtual {v9, v10}, Landroid/content/Intent;->setComponent(Landroid/content/ComponentName;)Landroid/content/Intent;
  
+     .line 89
      invoke-virtual {v8, v7}, Landroid/content/pm/ResolveInfo;->loadLabel(Landroid/content/pm/PackageManager;)Ljava/lang/CharSequence;
  
      move-result-object v10
***************
*** 355,360 ****
--- 393,399 ----
  
      move-result-object v5
  
+     .line 92
      .local v5, item:Landroid/view/MenuItem;
      if-eqz p8, :cond_1
  
***************
*** 362,376 ****
--- 401,418 ----
  
      if-ltz v10, :cond_1
  
+     .line 93
      iget v10, v8, Landroid/content/pm/ResolveInfo;->specificIndex:I
  
      aput-object v5, p8, v10
  
+     .line 82
      :cond_1
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_1
  
+     .line 76
      .end local v3           #N:I
      .end local v4           #i:I
      .end local v5           #item:Landroid/view/MenuItem;
***************
*** 381,386 ****
--- 423,429 ----
  
      goto :goto_0
  
+     .line 84
      .restart local v3       #N:I
      .restart local v4       #i:I
      .restart local v8       #ri:Landroid/content/pm/ResolveInfo;
***************
*** 391,396 ****
--- 434,440 ----
  
      goto :goto_2
  
+     .line 97
      .end local v8           #ri:Landroid/content/pm/ResolveInfo;
      :cond_4
      return v3
***************
*** 401,406 ****
--- 445,451 ----
      .parameter "titleRes"
  
      .prologue
+     .line 107
      const/4 v0, 0x0
  
      return-object v0
***************
*** 414,419 ****
--- 459,465 ----
      .parameter "titleRes"
  
      .prologue
+     .line 118
      const/4 v0, 0x0
  
      return-object v0
***************
*** 427,432 ****
--- 473,479 ----
      .parameter "title"
  
      .prologue
+     .line 113
      const/4 v0, 0x0
  
      return-object v0
***************
*** 437,442 ****
--- 484,490 ----
      .parameter "title"
  
      .prologue
+     .line 102
      const/4 v0, 0x0
  
      return-object v0
***************
*** 446,455 ****
--- 494,505 ----
      .locals 1
  
      .prologue
+     .line 122
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
  
+     .line 123
      return-void
  .end method
  
***************
*** 457,462 ****
--- 507,513 ----
      .locals 0
  
      .prologue
+     .line 126
      return-void
  .end method
  
***************
*** 465,470 ****
--- 516,522 ----
      .parameter "id"
  
      .prologue
+     .line 141
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
      invoke-direct {p0, p1}, Lcom/android/internal/view/menu/ActionMenu;->findItemIndex(I)I
***************
*** 484,489 ****
--- 536,542 ----
      .locals 1
  
      .prologue
+     .line 48
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenu;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 494,499 ****
--- 547,553 ----
      .parameter "index"
  
      .prologue
+     .line 145
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 509,521 ****
--- 563,578 ----
      .locals 4
  
      .prologue
+     .line 149
      iget-object v2, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
+     .line 150
      .local v2, items:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/ActionMenuItem;>;"
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
  
      move-result v1
  
+     .line 152
      .local v1, itemCount:I
      const/4 v0, 0x0
  
***************
*** 523,528 ****
--- 580,586 ----
      :goto_0
      if-ge v0, v1, :cond_1
  
+     .line 153
      invoke-virtual {v2, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v3
***************
*** 535,550 ****
--- 593,612 ----
  
      if-eqz v3, :cond_0
  
+     .line 154
      const/4 v3, 0x1
  
+     .line 158
      :goto_1
      return v3
  
+     .line 152
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 158
      :cond_1
      const/4 v3, 0x0
  
***************
*** 557,562 ****
--- 619,625 ----
      .parameter "event"
  
      .prologue
+     .line 179
      invoke-direct {p0, p1, p2}, Lcom/android/internal/view/menu/ActionMenu;->findItemWithShortcut(ILandroid/view/KeyEvent;)Lcom/android/internal/view/menu/ActionMenuItem;
  
      move-result-object v0
***************
*** 580,594 ****
--- 643,661 ----
      .parameter "flags"
  
      .prologue
+     .line 183
      invoke-direct {p0, p1}, Lcom/android/internal/view/menu/ActionMenu;->findItemIndex(I)I
  
      move-result v0
  
+     .line 184
      .local v0, index:I
      if-gez v0, :cond_0
  
+     .line 185
      const/4 v1, 0x0
  
+     .line 188
      :goto_0
      return v1
  
***************
*** 615,629 ****
--- 682,700 ----
      .parameter "flags"
  
      .prologue
+     .line 192
      invoke-direct {p0, p1, p2}, Lcom/android/internal/view/menu/ActionMenu;->findItemWithShortcut(ILandroid/view/KeyEvent;)Lcom/android/internal/view/menu/ActionMenuItem;
  
      move-result-object v0
  
+     .line 193
      .local v0, item:Lcom/android/internal/view/menu/ActionMenuItem;
      if-nez v0, :cond_0
  
+     .line 194
      const/4 v1, 0x0
  
+     .line 197
      :goto_0
      return v1
  
***************
*** 640,659 ****
--- 711,735 ----
      .parameter "groupId"
  
      .prologue
+     .line 201
      iget-object v2, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
+     .line 202
      .local v2, items:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/ActionMenuItem;>;"
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
  
      move-result v1
  
+     .line 203
      .local v1, itemCount:I
      const/4 v0, 0x0
  
+     .line 204
      .local v0, i:I
      :goto_0
      if-ge v0, v1, :cond_1
  
+     .line 205
      invoke-virtual {v2, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v3
***************
*** 666,682 ****
--- 742,762 ----
  
      if-ne v3, p1, :cond_0
  
+     .line 206
      invoke-virtual {v2, v0}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 207
      add-int/lit8 v1, v1, -0x1
  
      goto :goto_0
  
+     .line 209
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 212
      :cond_1
      return-void
  .end method
***************
*** 686,691 ****
--- 766,772 ----
      .parameter "id"
  
      .prologue
+     .line 215
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
      invoke-direct {p0, p1}, Lcom/android/internal/view/menu/ActionMenu;->findItemIndex(I)I
***************
*** 694,699 ****
--- 775,781 ----
  
      invoke-virtual {v0, v1}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 216
      return-void
  .end method
  
***************
*** 704,716 ****
--- 786,801 ----
      .parameter "exclusive"
  
      .prologue
+     .line 220
      iget-object v3, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
+     .line 221
      .local v3, items:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/ActionMenuItem;>;"
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v2
  
+     .line 223
      .local v2, itemCount:I
      const/4 v0, 0x0
  
***************
*** 718,729 ****
--- 803,816 ----
      :goto_0
      if-ge v0, v2, :cond_1
  
+     .line 224
      invoke-virtual {v3, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Lcom/android/internal/view/menu/ActionMenuItem;
  
+     .line 225
      .local v1, item:Lcom/android/internal/view/menu/ActionMenuItem;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/ActionMenuItem;->getGroupId()I
  
***************
*** 731,745 ****
--- 818,836 ----
  
      if-ne v4, p1, :cond_0
  
+     .line 226
      invoke-virtual {v1, p2}, Lcom/android/internal/view/menu/ActionMenuItem;->setCheckable(Z)Landroid/view/MenuItem;
  
+     .line 227
      invoke-virtual {v1, p3}, Lcom/android/internal/view/menu/ActionMenuItem;->setExclusiveCheckable(Z)Lcom/android/internal/view/menu/ActionMenuItem;
  
+     .line 223
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 230
      .end local v1           #item:Lcom/android/internal/view/menu/ActionMenuItem;
      :cond_1
      return-void
***************
*** 751,763 ****
--- 842,857 ----
      .parameter "enabled"
  
      .prologue
+     .line 233
      iget-object v3, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
+     .line 234
      .local v3, items:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/ActionMenuItem;>;"
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v2
  
+     .line 236
      .local v2, itemCount:I
      const/4 v0, 0x0
  
***************
*** 765,776 ****
--- 859,872 ----
      :goto_0
      if-ge v0, v2, :cond_1
  
+     .line 237
      invoke-virtual {v3, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Lcom/android/internal/view/menu/ActionMenuItem;
  
+     .line 238
      .local v1, item:Lcom/android/internal/view/menu/ActionMenuItem;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/ActionMenuItem;->getGroupId()I
  
***************
*** 778,790 ****
--- 874,889 ----
  
      if-ne v4, p1, :cond_0
  
+     .line 239
      invoke-virtual {v1, p2}, Lcom/android/internal/view/menu/ActionMenuItem;->setEnabled(Z)Landroid/view/MenuItem;
  
+     .line 236
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 242
      .end local v1           #item:Lcom/android/internal/view/menu/ActionMenuItem;
      :cond_1
      return-void
***************
*** 796,808 ****
--- 895,910 ----
      .parameter "visible"
  
      .prologue
+     .line 245
      iget-object v3, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
+     .line 246
      .local v3, items:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/ActionMenuItem;>;"
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v2
  
+     .line 248
      .local v2, itemCount:I
      const/4 v0, 0x0
  
***************
*** 810,821 ****
--- 912,925 ----
      :goto_0
      if-ge v0, v2, :cond_1
  
+     .line 249
      invoke-virtual {v3, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Lcom/android/internal/view/menu/ActionMenuItem;
  
+     .line 250
      .local v1, item:Lcom/android/internal/view/menu/ActionMenuItem;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/ActionMenuItem;->getGroupId()I
  
***************
*** 823,835 ****
--- 927,942 ----
  
      if-ne v4, p1, :cond_0
  
+     .line 251
      invoke-virtual {v1, p2}, Lcom/android/internal/view/menu/ActionMenuItem;->setVisible(Z)Landroid/view/MenuItem;
  
+     .line 248
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 254
      .end local v1           #item:Lcom/android/internal/view/menu/ActionMenuItem;
      :cond_1
      return-void
***************
*** 840,847 ****
--- 947,956 ----
      .parameter "isQwerty"
  
      .prologue
+     .line 257
      iput-boolean p1, p0, Lcom/android/internal/view/menu/ActionMenu;->mIsQwerty:Z
  
+     .line 258
      return-void
  .end method
  
***************
*** 849,854 ****
--- 958,964 ----
      .locals 1
  
      .prologue
+     .line 261
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenu;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->size()I
