*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/ActionMenuPresenter$PopupPresenterCallback.smali	2012-03-23 19:37:53.361748133 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/ActionMenuPresenter$PopupPresenterCallback.smali.original	2012-03-23 19:34:46.394338787 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 630
      iput-object p1, p0, Lcom/android/internal/view/menu/ActionMenuPresenter$PopupPresenterCallback;->this$0:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .parameter "x1"
  
      .prologue
+     .line 630
      invoke-direct {p0, p1}, Lcom/android/internal/view/menu/ActionMenuPresenter$PopupPresenterCallback;-><init>(Lcom/android/internal/view/menu/ActionMenuPresenter;)V
  
      return-void
***************
*** 53,62 ****
--- 55,66 ----
      .parameter "allMenusAreClosing"
  
      .prologue
+     .line 642
      instance-of v0, p1, Lcom/android/internal/view/menu/SubMenuBuilder;
  
      if-eqz v0, :cond_0
  
+     .line 643
      check-cast p1, Lcom/android/internal/view/menu/SubMenuBuilder;
  
      .end local p1
***************
*** 68,73 ****
--- 72,78 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/view/menu/MenuBuilder;->close(Z)V
  
+     .line 645
      :cond_0
      return-void
  .end method
***************
*** 79,90 ****
--- 84,98 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 634
      if-nez p1, :cond_0
  
+     .line 637
      .end local p1
      :goto_0
      return v2
  
+     .line 636
      .restart local p1
      :cond_0
      iget-object v0, p0, Lcom/android/internal/view/menu/ActionMenuPresenter$PopupPresenterCallback;->this$0:Lcom/android/internal/view/menu/ActionMenuPresenter;
