*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/IconMenuView$SavedState.smali	2012-03-23 19:37:54.104761681 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/IconMenuView$SavedState.smali.original	2012-03-23 19:34:46.401338915 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .locals 1
  
      .prologue
+     .line 731
      new-instance v0, Lcom/android/internal/view/menu/IconMenuView$SavedState$1;
  
      invoke-direct {v0}, Lcom/android/internal/view/menu/IconMenuView$SavedState$1;-><init>()V
***************
*** 50,63 ****
--- 51,67 ----
      .parameter "in"
  
      .prologue
+     .line 721
      invoke-direct {p0, p1}, Landroid/view/View$BaseSavedState;-><init>(Landroid/os/Parcel;)V
  
+     .line 722
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Lcom/android/internal/view/menu/IconMenuView$SavedState;->focusedPosition:I
  
+     .line 723
      return-void
  .end method
  
***************
*** 67,72 ****
--- 71,77 ----
      .parameter "x1"
  
      .prologue
+     .line 706
      invoke-direct {p0, p1}, Lcom/android/internal/view/menu/IconMenuView$SavedState;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 78,87 ****
--- 83,95 ----
      .parameter "focusedPosition"
  
      .prologue
+     .line 713
      invoke-direct {p0, p1}, Landroid/view/View$BaseSavedState;-><init>(Landroid/os/Parcelable;)V
  
+     .line 714
      iput p2, p0, Lcom/android/internal/view/menu/IconMenuView$SavedState;->focusedPosition:I
  
+     .line 715
      return-void
  .end method
  
***************
*** 93,103 ****
--- 101,114 ----
      .parameter "flags"
  
      .prologue
+     .line 727
      invoke-super {p0, p1, p2}, Landroid/view/View$BaseSavedState;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 728
      iget v0, p0, Lcom/android/internal/view/menu/IconMenuView$SavedState;->focusedPosition:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 729
      return-void
  .end method
