*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/MenuBuilder.smali	2012-03-23 19:37:54.399767060 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/menu/MenuBuilder.smali.original	2012-03-23 19:34:46.404338971 -0500
***************
*** 147,152 ****
--- 147,153 ----
      .locals 1
  
      .prologue
+     .line 56
      const/4 v0, 0x6
  
      new-array v0, v0, [I
***************
*** 179,246 ****
--- 180,265 ----
  
      const/4 v0, 0x0
  
+     .line 186
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 116
      iput v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mDefaultShowAsAction:I
  
+     .line 142
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPreventDispatchingItemsChanged:Z
  
+     .line 143
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItemsChangedWhileDispatchPrevented:Z
  
+     .line 145
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mOptionalIconsVisible:Z
  
+     .line 147
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsClosing:Z
  
+     .line 149
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mTempShortcutItemList:Ljava/util/ArrayList;
  
+     .line 151
      new-instance v0, Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-direct {v0}, Ljava/util/concurrent/CopyOnWriteArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
+     .line 187
      iput-object p1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mContext:Landroid/content/Context;
  
+     .line 188
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mResources:Landroid/content/res/Resources;
  
+     .line 190
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
+     .line 192
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mVisibleItems:Ljava/util/ArrayList;
  
+     .line 193
      iput-boolean v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsVisibleItemsStale:Z
  
+     .line 195
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mActionItems:Ljava/util/ArrayList;
  
+     .line 196
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mNonActionItems:Ljava/util/ArrayList;
  
+     .line 197
      iput-boolean v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsActionItemsStale:Z
  
+     .line 199
      invoke-direct {p0, v1}, Lcom/android/internal/view/menu/MenuBuilder;->setShortcutsVisibleInner(Z)V
  
+     .line 200
      return-void
  .end method
  
***************
*** 252,261 ****
--- 271,282 ----
      .parameter "title"
  
      .prologue
+     .line 387
      invoke-static {p3}, Lcom/android/internal/view/menu/MenuBuilder;->getOrdering(I)I
  
      move-result v5
  
+     .line 389
      .local v5, ordering:I
      new-instance v0, Lcom/android/internal/view/menu/MenuItemImpl;
  
***************
*** 273,287 ****
--- 294,311 ----
  
      invoke-direct/range {v0 .. v7}, Lcom/android/internal/view/menu/MenuItemImpl;-><init>(Lcom/android/internal/view/menu/MenuBuilder;IIIILjava/lang/CharSequence;I)V
  
+     .line 392
      .local v0, item:Lcom/android/internal/view/menu/MenuItemImpl;
      iget-object v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mCurrentMenuInfo:Landroid/view/ContextMenu$ContextMenuInfo;
  
      if-eqz v1, :cond_0
  
+     .line 394
      iget-object v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mCurrentMenuInfo:Landroid/view/ContextMenu$ContextMenuInfo;
  
      invoke-virtual {v0, v1}, Lcom/android/internal/view/menu/MenuItemImpl;->setMenuInfo(Landroid/view/ContextMenu$ContextMenuInfo;)V
  
+     .line 397
      :cond_0
      iget-object v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
***************
*** 293,302 ****
--- 317,328 ----
  
      invoke-virtual {v1, v2, v0}, Ljava/util/ArrayList;->add(ILjava/lang/Object;)V
  
+     .line 398
      const/4 v1, 0x1
  
      invoke-virtual {p0, v1}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 400
      return-object v0
  .end method
  
***************
*** 305,310 ****
--- 331,337 ----
      .parameter "cleared"
  
      .prologue
+     .line 236
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v3}, Ljava/util/concurrent/CopyOnWriteArrayList;->isEmpty()Z
***************
*** 313,324 ****
--- 340,354 ----
  
      if-eqz v3, :cond_0
  
+     .line 248
      :goto_0
      return-void
  
+     .line 238
      :cond_0
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->stopDispatchingItemsChanged()V
  
+     .line 239
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v3}, Ljava/util/concurrent/CopyOnWriteArrayList;->iterator()Ljava/util/Iterator;
***************
*** 339,344 ****
--- 369,375 ----
  
      check-cast v2, Ljava/lang/ref/WeakReference;
  
+     .line 240
      .local v2, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      invoke-virtual {v2}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
***************
*** 346,365 ****
--- 377,400 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuPresenter;
  
+     .line 241
      .local v1, presenter:Lcom/android/internal/view/menu/MenuPresenter;
      if-nez v1, :cond_1
  
+     .line 242
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v3, v2}, Ljava/util/concurrent/CopyOnWriteArrayList;->remove(Ljava/lang/Object;)Z
  
      goto :goto_1
  
+     .line 244
      :cond_1
      invoke-interface {v1, p1}, Lcom/android/internal/view/menu/MenuPresenter;->updateMenuView(Z)V
  
      goto :goto_1
  
+     .line 247
      .end local v1           #presenter:Lcom/android/internal/view/menu/MenuPresenter;
      .end local v2           #ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      :cond_2
***************
*** 373,384 ****
--- 408,421 ----
      .parameter "state"
  
      .prologue
+     .line 290
      const-string v6, "android:menu:presenters"
  
      invoke-virtual {p1, v6}, Landroid/os/Bundle;->getSparseParcelableArray(Ljava/lang/String;)Landroid/util/SparseArray;
  
      move-result-object v4
  
+     .line 292
      .local v4, presenterStates:Landroid/util/SparseArray;,"Landroid/util/SparseArray<Landroid/os/Parcelable;>;"
      if-eqz v4, :cond_0
  
***************
*** 390,398 ****
--- 427,437 ----
  
      if-eqz v6, :cond_1
  
+     .line 308
      :cond_0
      return-void
  
+     .line 294
      :cond_1
      iget-object v6, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
***************
*** 415,420 ****
--- 454,460 ----
  
      check-cast v5, Ljava/lang/ref/WeakReference;
  
+     .line 295
      .local v5, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      invoke-virtual {v5}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
***************
*** 422,453 ****
--- 462,500 ----
  
      check-cast v3, Lcom/android/internal/view/menu/MenuPresenter;
  
+     .line 296
      .local v3, presenter:Lcom/android/internal/view/menu/MenuPresenter;
      if-nez v3, :cond_3
  
+     .line 297
      iget-object v6, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v6, v5}, Ljava/util/concurrent/CopyOnWriteArrayList;->remove(Ljava/lang/Object;)Z
  
      goto :goto_0
  
+     .line 299
      :cond_3
      invoke-interface {v3}, Lcom/android/internal/view/menu/MenuPresenter;->getId()I
  
      move-result v1
  
+     .line 300
      .local v1, id:I
      if-lez v1, :cond_2
  
+     .line 301
      invoke-virtual {v4, v1}, Landroid/util/SparseArray;->get(I)Ljava/lang/Object;
  
      move-result-object v2
  
      check-cast v2, Landroid/os/Parcelable;
  
+     .line 302
      .local v2, parcel:Landroid/os/Parcelable;
      if-eqz v2, :cond_2
  
+     .line 303
      invoke-interface {v3, v2}, Lcom/android/internal/view/menu/MenuPresenter;->onRestoreInstanceState(Landroid/os/Parcelable;)V
  
      goto :goto_0
***************
*** 458,463 ****
--- 505,511 ----
      .parameter "outState"
  
      .prologue
+     .line 267
      iget-object v6, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v6}, Ljava/util/concurrent/CopyOnWriteArrayList;->isEmpty()Z
***************
*** 466,479 ****
--- 514,530 ----
  
      if-eqz v6, :cond_0
  
+     .line 287
      :goto_0
      return-void
  
+     .line 269
      :cond_0
      new-instance v3, Landroid/util/SparseArray;
  
      invoke-direct {v3}, Landroid/util/SparseArray;-><init>()V
  
+     .line 271
      .local v3, presenterStates:Landroid/util/SparseArray;,"Landroid/util/SparseArray<Landroid/os/Parcelable;>;"
      iget-object v6, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
***************
*** 496,501 ****
--- 547,553 ----
  
      check-cast v4, Ljava/lang/ref/WeakReference;
  
+     .line 272
      .local v4, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      invoke-virtual {v4}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
***************
*** 503,536 ****
--- 555,596 ----
  
      check-cast v2, Lcom/android/internal/view/menu/MenuPresenter;
  
+     .line 273
      .local v2, presenter:Lcom/android/internal/view/menu/MenuPresenter;
      if-nez v2, :cond_2
  
+     .line 274
      iget-object v6, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v6, v4}, Ljava/util/concurrent/CopyOnWriteArrayList;->remove(Ljava/lang/Object;)Z
  
      goto :goto_1
  
+     .line 276
      :cond_2
      invoke-interface {v2}, Lcom/android/internal/view/menu/MenuPresenter;->getId()I
  
      move-result v1
  
+     .line 277
      .local v1, id:I
      if-lez v1, :cond_1
  
+     .line 278
      invoke-interface {v2}, Lcom/android/internal/view/menu/MenuPresenter;->onSaveInstanceState()Landroid/os/Parcelable;
  
      move-result-object v5
  
+     .line 279
      .local v5, state:Landroid/os/Parcelable;
      if-eqz v5, :cond_1
  
+     .line 280
      invoke-virtual {v3, v1, v5}, Landroid/util/SparseArray;->put(ILjava/lang/Object;)V
  
      goto :goto_1
  
+     .line 286
      .end local v1           #id:I
      .end local v2           #presenter:Lcom/android/internal/view/menu/MenuPresenter;
      .end local v4           #ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
***************
*** 548,553 ****
--- 608,614 ----
      .parameter "subMenu"
  
      .prologue
+     .line 251
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v4}, Ljava/util/concurrent/CopyOnWriteArrayList;->isEmpty()Z
***************
*** 558,569 ****
--- 619,633 ----
  
      const/4 v3, 0x0
  
+     .line 263
      :cond_0
      return v3
  
+     .line 253
      :cond_1
      const/4 v3, 0x0
  
+     .line 255
      .local v3, result:Z
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
***************
*** 586,591 ****
--- 650,656 ----
  
      check-cast v2, Ljava/lang/ref/WeakReference;
  
+     .line 256
      .local v2, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      invoke-virtual {v2}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
***************
*** 593,610 ****
--- 658,679 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuPresenter;
  
+     .line 257
      .local v1, presenter:Lcom/android/internal/view/menu/MenuPresenter;
      if-nez v1, :cond_3
  
+     .line 258
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v4, v2}, Ljava/util/concurrent/CopyOnWriteArrayList;->remove(Ljava/lang/Object;)Z
  
      goto :goto_0
  
+     .line 259
      :cond_3
      if-nez v3, :cond_2
  
+     .line 260
      invoke-interface {v1, p1}, Lcom/android/internal/view/menu/MenuPresenter;->onSubMenuSelected(Lcom/android/internal/view/menu/SubMenuBuilder;)Z
  
      move-result v3
***************
*** 627,632 ****
--- 696,702 ----
      .end annotation
  
      .prologue
+     .line 748
      .local p0, items:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/MenuItemImpl;>;"
      invoke-virtual {p0}, Ljava/util/ArrayList;->size()I
  
***************
*** 638,649 ****
--- 708,721 ----
      :goto_0
      if-ltz v0, :cond_1
  
+     .line 749
      invoke-virtual {p0, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 750
      .local v1, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->getOrdering()I
  
***************
*** 651,668 ****
--- 723,744 ----
  
      if-gt v2, p1, :cond_0
  
+     .line 751
      add-int/lit8 v2, v0, 0x1
  
+     .line 755
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :goto_1
      return v2
  
+     .line 748
      .restart local v1       #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_0
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_0
  
+     .line 755
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_1
      const/4 v2, 0x0
***************
*** 675,686 ****
--- 751,764 ----
      .parameter "categoryOrder"
  
      .prologue
+     .line 681
      const/high16 v1, -0x1
  
      and-int/2addr v1, p0
  
      shr-int/lit8 v0, v1, 0x10
  
+     .line 683
      .local v0, index:I
      if-ltz v0, :cond_0
  
***************
*** 690,704 ****
  
      if-lt v0, v1, :cond_1
  
      :cond_0
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
!     const-string v2, "order does not contain a valid category."
  
      invoke-direct {v1, v2}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
      :cond_1
      sget-object v1, Lcom/android/internal/view/menu/MenuBuilder;->sCategoryToOrder:[I
  
--- 768,784 ----
  
      if-lt v0, v1, :cond_1
  
+     .line 684
      :cond_0
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v2, "order does not contain a valid category."
  
      invoke-direct {v1, v2}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
+     .line 687
      :cond_1
      sget-object v1, Lcom/android/internal/view/menu/MenuBuilder;->sCategoryToOrder:[I
  
***************
*** 721,726 ****
--- 801,807 ----
      .parameter "updateChildrenOnMenuViews"
  
      .prologue
+     .line 499
      if-ltz p1, :cond_0
  
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
***************
*** 731,745 ****
--- 812,829 ----
  
      if-lt p1, v0, :cond_1
  
+     .line 504
      :cond_0
      :goto_0
      return-void
  
+     .line 501
      :cond_1
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 503
      if-eqz p2, :cond_0
  
      const/4 v0, 0x1
***************
*** 760,820 ****
--- 844,920 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1090
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->getResources()Landroid/content/res/Resources;
  
      move-result-object v0
  
+     .line 1092
      .local v0, r:Landroid/content/res/Resources;
      if-eqz p5, :cond_0
  
+     .line 1093
      iput-object p5, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderView:Landroid/view/View;
  
+     .line 1096
      iput-object v2, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderTitle:Ljava/lang/CharSequence;
  
+     .line 1097
      iput-object v2, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderIcon:Landroid/graphics/drawable/Drawable;
  
+     .line 1116
      :goto_0
      const/4 v1, 0x0
  
      invoke-virtual {p0, v1}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 1117
      return-void
  
+     .line 1099
      :cond_0
      if-lez p1, :cond_3
  
+     .line 1100
      invoke-virtual {v0, p1}, Landroid/content/res/Resources;->getText(I)Ljava/lang/CharSequence;
  
      move-result-object v1
  
      iput-object v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderTitle:Ljava/lang/CharSequence;
  
+     .line 1105
      :cond_1
      :goto_1
      if-lez p3, :cond_4
  
+     .line 1106
      invoke-virtual {v0, p3}, Landroid/content/res/Resources;->getDrawable(I)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
  
      iput-object v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderIcon:Landroid/graphics/drawable/Drawable;
  
+     .line 1112
      :cond_2
      :goto_2
      iput-object v2, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderView:Landroid/view/View;
  
      goto :goto_0
  
+     .line 1101
      :cond_3
      if-eqz p2, :cond_1
  
+     .line 1102
      iput-object p2, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderTitle:Ljava/lang/CharSequence;
  
      goto :goto_1
  
+     .line 1107
      :cond_4
      if-eqz p4, :cond_2
  
+     .line 1108
      iput-object p4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderIcon:Landroid/graphics/drawable/Drawable;
  
      goto :goto_2
***************
*** 827,832 ****
--- 927,933 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 713
      if-eqz p1, :cond_0
  
      iget-object v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mResources:Landroid/content/res/Resources;
***************
*** 852,859 ****
--- 953,962 ----
      :goto_0
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mShortcutsVisible:Z
  
+     .line 717
      return-void
  
+     .line 713
      :cond_0
      const/4 v0, 0x0
  
***************
*** 869,874 ****
--- 972,978 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 408
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mResources:Landroid/content/res/Resources;
  
      invoke-virtual {v0, p1}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
***************
*** 890,895 ****
--- 994,1000 ----
      .parameter "title"
  
      .prologue
+     .line 416
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mResources:Landroid/content/res/Resources;
  
      invoke-virtual {v0, p4}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
***************
*** 911,916 ****
--- 1016,1022 ----
      .parameter "title"
  
      .prologue
+     .line 412
      invoke-direct {p0, p1, p2, p3, p4}, Lcom/android/internal/view/menu/MenuBuilder;->addInternal(IIILjava/lang/CharSequence;)Landroid/view/MenuItem;
  
      move-result-object v0
***************
*** 925,930 ****
--- 1031,1037 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 404
      invoke-direct {p0, v0, v0, v0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->addInternal(IIILjava/lang/CharSequence;)Landroid/view/MenuItem;
  
      move-result-object v0
***************
*** 944,955 ****
--- 1051,1064 ----
      .parameter "outSpecificItems"
  
      .prologue
+     .line 441
      iget-object v10, p0, Lcom/android/internal/view/menu/MenuBuilder;->mContext:Landroid/content/Context;
  
      invoke-virtual {v10}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
  
      move-result-object v7
  
+     .line 442
      .local v7, pm:Landroid/content/pm/PackageManager;
      const/4 v10, 0x0
  
***************
*** 963,968 ****
--- 1072,1078 ----
  
      move-result-object v6
  
+     .line 444
      .local v6, lri:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
      if-eqz v6, :cond_2
  
***************
*** 970,983 ****
--- 1080,1096 ----
  
      move-result v3
  
+     .line 446
      .local v3, N:I
      :goto_0
      and-int/lit8 v10, p7, 0x1
  
      if-nez v10, :cond_0
  
+     .line 447
      invoke-virtual {p0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->removeGroup(I)V
  
+     .line 450
      :cond_0
      const/4 v4, 0x0
  
***************
*** 985,996 ****
--- 1098,1111 ----
      :goto_1
      if-ge v4, v3, :cond_4
  
+     .line 451
      invoke-interface {v6, v4}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v8
  
      check-cast v8, Landroid/content/pm/ResolveInfo;
  
+     .line 452
      .local v8, ri:Landroid/content/pm/ResolveInfo;
      new-instance v9, Landroid/content/Intent;
  
***************
*** 1003,1008 ****
--- 1118,1124 ----
      :goto_2
      invoke-direct {v9, v10}, Landroid/content/Intent;-><init>(Landroid/content/Intent;)V
  
+     .line 454
      .local v9, rintent:Landroid/content/Intent;
      new-instance v10, Landroid/content/ComponentName;
  
***************
*** 1020,1025 ****
--- 1136,1142 ----
  
      invoke-virtual {v9, v10}, Landroid/content/Intent;->setComponent(Landroid/content/ComponentName;)Landroid/content/Intent;
  
+     .line 457
      invoke-virtual {v8, v7}, Landroid/content/pm/ResolveInfo;->loadLabel(Landroid/content/pm/PackageManager;)Ljava/lang/CharSequence;
  
      move-result-object v10
***************
*** 1042,1047 ****
--- 1159,1165 ----
  
      move-result-object v5
  
+     .line 460
      .local v5, item:Landroid/view/MenuItem;
      if-eqz p8, :cond_1
  
***************
*** 1049,1063 ****
--- 1167,1184 ----
  
      if-ltz v10, :cond_1
  
+     .line 461
      iget v10, v8, Landroid/content/pm/ResolveInfo;->specificIndex:I
  
      aput-object v5, p8, v10
  
+     .line 450
      :cond_1
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_1
  
+     .line 444
      .end local v3           #N:I
      .end local v4           #i:I
      .end local v5           #item:Landroid/view/MenuItem;
***************
*** 1068,1073 ****
--- 1189,1195 ----
  
      goto :goto_0
  
+     .line 452
      .restart local v3       #N:I
      .restart local v4       #i:I
      .restart local v8       #ri:Landroid/content/pm/ResolveInfo;
***************
*** 1078,1083 ****
--- 1200,1206 ----
  
      goto :goto_2
  
+     .line 465
      .end local v8           #ri:Landroid/content/pm/ResolveInfo;
      :cond_4
      return v3
***************
*** 1088,1093 ****
--- 1211,1217 ----
      .parameter "presenter"
  
      .prologue
+     .line 215
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      new-instance v1, Ljava/lang/ref/WeakReference;
***************
*** 1096,1109 ****
--- 1220,1236 ----
  
      invoke-virtual {v0, v1}, Ljava/util/concurrent/CopyOnWriteArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 216
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mContext:Landroid/content/Context;
  
      invoke-interface {p1, v0, p0}, Lcom/android/internal/view/menu/MenuPresenter;->initForMenu(Landroid/content/Context;Lcom/android/internal/view/menu/MenuBuilder;)V
  
+     .line 217
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsActionItemsStale:Z
  
+     .line 218
      return-void
  .end method
  
***************
*** 1114,1119 ****
--- 1241,1247 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 424
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mResources:Landroid/content/res/Resources;
  
      invoke-virtual {v0, p1}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
***************
*** 1135,1140 ****
--- 1263,1269 ----
      .parameter "title"
  
      .prologue
+     .line 436
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mResources:Landroid/content/res/Resources;
  
      invoke-virtual {v0, p4}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
***************
*** 1156,1167 ****
--- 1285,1298 ----
      .parameter "title"
  
      .prologue
+     .line 428
      invoke-direct {p0, p1, p2, p3, p4}, Lcom/android/internal/view/menu/MenuBuilder;->addInternal(IIILjava/lang/CharSequence;)Landroid/view/MenuItem;
  
      move-result-object v0
  
      check-cast v0, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 429
      .local v0, item:Lcom/android/internal/view/menu/MenuItemImpl;
      new-instance v1, Lcom/android/internal/view/menu/SubMenuBuilder;
  
***************
*** 1169,1177 ****
--- 1300,1310 ----
  
      invoke-direct {v1, v2, p0, v0}, Lcom/android/internal/view/menu/SubMenuBuilder;-><init>(Landroid/content/Context;Lcom/android/internal/view/menu/MenuBuilder;Lcom/android/internal/view/menu/MenuItemImpl;)V
  
+     .line 430
      .local v1, subMenu:Lcom/android/internal/view/menu/SubMenuBuilder;
      invoke-virtual {v0, v1}, Lcom/android/internal/view/menu/MenuItemImpl;->setSubMenu(Lcom/android/internal/view/menu/SubMenuBuilder;)V
  
+     .line 432
      return-object v1
  .end method
  
***************
*** 1182,1187 ****
--- 1315,1321 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 420
      invoke-virtual {p0, v0, v0, v0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->addSubMenu(IIILjava/lang/CharSequence;)Landroid/view/SubMenu;
  
      move-result-object v0
***************
*** 1193,1206 ****
--- 1327,1343 ----
      .locals 1
  
      .prologue
+     .line 742
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mCallback:Lcom/android/internal/view/menu/MenuBuilder$Callback;
  
      if-eqz v0, :cond_0
  
+     .line 743
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mCallback:Lcom/android/internal/view/menu/MenuBuilder$Callback;
  
      invoke-interface {v0, p0}, Lcom/android/internal/view/menu/MenuBuilder$Callback;->onMenuModeChange(Lcom/android/internal/view/menu/MenuBuilder;)V
  
+     .line 745
      :cond_0
      return-void
  .end method
***************
*** 1209,1231 ****
--- 1346,1373 ----
      .locals 1
  
      .prologue
+     .line 520
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
      if-eqz v0, :cond_0
  
+     .line 521
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->collapseItemActionView(Lcom/android/internal/view/menu/MenuItemImpl;)Z
  
+     .line 523
      :cond_0
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
  
+     .line 525
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 526
      return-void
  .end method
  
***************
*** 1237,1254 ****
--- 1379,1403 ----
  
      const/4 v0, 0x0
  
+     .line 511
      iput-boolean v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPreventDispatchingItemsChanged:Z
  
+     .line 512
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->clear()V
  
+     .line 513
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->clearHeader()V
  
+     .line 514
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPreventDispatchingItemsChanged:Z
  
+     .line 515
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItemsChangedWhileDispatchPrevented:Z
  
+     .line 516
      invoke-virtual {p0, v1}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 517
      return-void
  .end method
  
***************
*** 1258,1273 ****
--- 1407,1427 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 1081
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderIcon:Landroid/graphics/drawable/Drawable;
  
+     .line 1082
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderTitle:Ljava/lang/CharSequence;
  
+     .line 1083
      iput-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderView:Landroid/view/View;
  
+     .line 1085
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 1086
      return-void
  .end method
  
***************
*** 1275,1284 ****
--- 1429,1440 ----
      .locals 1
  
      .prologue
+     .line 924
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->close(Z)V
  
+     .line 925
      return-void
  .end method
  
***************
*** 1287,1304 ****
--- 1443,1464 ----
      .parameter "allMenusAreClosing"
  
      .prologue
+     .line 908
      iget-boolean v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsClosing:Z
  
      if-eqz v3, :cond_0
  
+     .line 920
      :goto_0
      return-void
  
+     .line 910
      :cond_0
      const/4 v3, 0x1
  
      iput-boolean v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsClosing:Z
  
+     .line 911
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v3}, Ljava/util/concurrent/CopyOnWriteArrayList;->iterator()Ljava/util/Iterator;
***************
*** 1319,1324 ****
--- 1479,1485 ----
  
      check-cast v2, Ljava/lang/ref/WeakReference;
  
+     .line 912
      .local v2, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      invoke-virtual {v2}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
***************
*** 1326,1345 ****
--- 1487,1510 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuPresenter;
  
+     .line 913
      .local v1, presenter:Lcom/android/internal/view/menu/MenuPresenter;
      if-nez v1, :cond_1
  
+     .line 914
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v3, v2}, Ljava/util/concurrent/CopyOnWriteArrayList;->remove(Ljava/lang/Object;)Z
  
      goto :goto_1
  
+     .line 916
      :cond_1
      invoke-interface {v1, p0, p1}, Lcom/android/internal/view/menu/MenuPresenter;->onCloseMenu(Lcom/android/internal/view/menu/MenuBuilder;Z)V
  
      goto :goto_1
  
+     .line 919
      .end local v1           #presenter:Lcom/android/internal/view/menu/MenuPresenter;
      .end local v2           #ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      :cond_2
***************
*** 1355,1360 ****
--- 1520,1526 ----
      .parameter "item"
  
      .prologue
+     .line 1241
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v4}, Ljava/util/concurrent/CopyOnWriteArrayList;->isEmpty()Z
***************
*** 1370,1385 ****
--- 1536,1555 ----
      :cond_0
      const/4 v0, 0x0
  
+     .line 1259
      :cond_1
      :goto_0
      return v0
  
+     .line 1243
      :cond_2
      const/4 v0, 0x0
  
+     .line 1245
      .local v0, collapsed:Z
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->stopDispatchingItemsChanged()V
  
+     .line 1246
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v4}, Ljava/util/concurrent/CopyOnWriteArrayList;->iterator()Ljava/util/Iterator;
***************
*** 1401,1406 ****
--- 1571,1577 ----
  
      check-cast v3, Ljava/lang/ref/WeakReference;
  
+     .line 1247
      .local v3, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      invoke-virtual {v3}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
***************
*** 1408,1422 ****
--- 1579,1596 ----
  
      check-cast v2, Lcom/android/internal/view/menu/MenuPresenter;
  
+     .line 1248
      .local v2, presenter:Lcom/android/internal/view/menu/MenuPresenter;
      if-nez v2, :cond_4
  
+     .line 1249
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v4, v3}, Ljava/util/concurrent/CopyOnWriteArrayList;->remove(Ljava/lang/Object;)Z
  
      goto :goto_1
  
+     .line 1250
      :cond_4
      invoke-interface {v2, p0, p1}, Lcom/android/internal/view/menu/MenuPresenter;->collapseItemActionView(Lcom/android/internal/view/menu/MenuBuilder;Lcom/android/internal/view/menu/MenuItemImpl;)Z
  
***************
*** 1424,1436 ****
--- 1598,1613 ----
  
      if-eqz v0, :cond_3
  
+     .line 1254
      .end local v2           #presenter:Lcom/android/internal/view/menu/MenuPresenter;
      .end local v3           #ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      :cond_5
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->startDispatchingItemsChanged()V
  
+     .line 1256
      if-eqz v0, :cond_1
  
+     .line 1257
      const/4 v4, 0x0
  
      iput-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
***************
*** 1444,1449 ****
--- 1621,1627 ----
      .parameter "item"
  
      .prologue
+     .line 735
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mCallback:Lcom/android/internal/view/menu/MenuBuilder$Callback;
  
      if-eqz v0, :cond_0
***************
*** 1472,1477 ****
--- 1650,1656 ----
      .parameter "item"
  
      .prologue
+     .line 1219
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v4}, Ljava/util/concurrent/CopyOnWriteArrayList;->isEmpty()Z
***************
*** 1482,1497 ****
--- 1661,1680 ----
  
      const/4 v0, 0x0
  
+     .line 1237
      :cond_0
      :goto_0
      return v0
  
+     .line 1221
      :cond_1
      const/4 v0, 0x0
  
+     .line 1223
      .local v0, expanded:Z
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->stopDispatchingItemsChanged()V
  
+     .line 1224
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v4}, Ljava/util/concurrent/CopyOnWriteArrayList;->iterator()Ljava/util/Iterator;
***************
*** 1513,1518 ****
--- 1696,1702 ----
  
      check-cast v3, Ljava/lang/ref/WeakReference;
  
+     .line 1225
      .local v3, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      invoke-virtual {v3}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
***************
*** 1520,1534 ****
--- 1704,1721 ----
  
      check-cast v2, Lcom/android/internal/view/menu/MenuPresenter;
  
+     .line 1226
      .local v2, presenter:Lcom/android/internal/view/menu/MenuPresenter;
      if-nez v2, :cond_3
  
+     .line 1227
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v4, v3}, Ljava/util/concurrent/CopyOnWriteArrayList;->remove(Ljava/lang/Object;)Z
  
      goto :goto_1
  
+     .line 1228
      :cond_3
      invoke-interface {v2, p0, p1}, Lcom/android/internal/view/menu/MenuPresenter;->expandItemActionView(Lcom/android/internal/view/menu/MenuBuilder;Lcom/android/internal/view/menu/MenuItemImpl;)Z
  
***************
*** 1536,1548 ****
--- 1723,1738 ----
  
      if-eqz v0, :cond_2
  
+     .line 1232
      .end local v2           #presenter:Lcom/android/internal/view/menu/MenuPresenter;
      .end local v3           #ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      :cond_4
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->startDispatchingItemsChanged()V
  
+     .line 1234
      if-eqz v0, :cond_0
  
+     .line 1235
      iput-object p1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
      goto :goto_0
***************
*** 1553,1558 ****
--- 1743,1749 ----
      .parameter "group"
  
      .prologue
+     .line 629
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Lcom/android/internal/view/menu/MenuBuilder;->findGroupIndex(II)I
***************
*** 1568,1582 ****
--- 1759,1777 ----
      .parameter "start"
  
      .prologue
+     .line 633
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->size()I
  
      move-result v2
  
+     .line 635
      .local v2, size:I
      if-gez p2, :cond_0
  
+     .line 636
      const/4 p2, 0x0
  
+     .line 639
      :cond_0
      move v0, p2
  
***************
*** 1584,1589 ****
--- 1779,1785 ----
      :goto_0
      if-ge v0, v2, :cond_2
  
+     .line 640
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1592,1597 ****
--- 1788,1794 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 642
      .local v1, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->getGroupId()I
  
***************
*** 1599,1609 ****
--- 1796,1808 ----
  
      if-ne v3, p1, :cond_1
  
+     .line 647
      .end local v0           #i:I
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :goto_1
      return v0
  
+     .line 639
      .restart local v0       #i:I
      .restart local v1       #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_1
***************
*** 1611,1616 ****
--- 1810,1816 ----
  
      goto :goto_0
  
+     .line 647
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_2
      const/4 v0, -0x1
***************
*** 1623,1632 ****
--- 1823,1834 ----
      .parameter "id"
  
      .prologue
+     .line 598
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->size()I
  
      move-result v3
  
+     .line 599
      .local v3, size:I
      const/4 v0, 0x0
  
***************
*** 1634,1639 ****
--- 1836,1842 ----
      :goto_0
      if-ge v0, v3, :cond_2
  
+     .line 600
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1642,1647 ****
--- 1845,1851 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 601
      .local v1, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->getItemId()I
  
***************
*** 1649,1658 ****
--- 1853,1864 ----
  
      if-ne v4, p1, :cond_0
  
+     .line 612
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :goto_1
      return-object v1
  
+     .line 603
      .restart local v1       #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_0
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->hasSubMenu()Z
***************
*** 1661,1666 ****
--- 1867,1873 ----
  
      if-eqz v4, :cond_1
  
+     .line 604
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->getSubMenu()Landroid/view/SubMenu;
  
      move-result-object v4
***************
*** 1669,1687 ****
--- 1876,1898 ----
  
      move-result-object v2
  
+     .line 606
      .local v2, possibleItem:Landroid/view/MenuItem;
      if-eqz v2, :cond_1
  
      move-object v1, v2
  
+     .line 607
      goto :goto_1
  
+     .line 599
      .end local v2           #possibleItem:Landroid/view/MenuItem;
      :cond_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 612
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_2
      const/4 v1, 0x0
***************
*** 1694,1703 ****
--- 1905,1916 ----
      .parameter "id"
  
      .prologue
+     .line 616
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->size()I
  
      move-result v2
  
+     .line 618
      .local v2, size:I
      const/4 v0, 0x0
  
***************
*** 1705,1710 ****
--- 1918,1924 ----
      :goto_0
      if-ge v0, v2, :cond_1
  
+     .line 619
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1713,1718 ****
--- 1927,1933 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 620
      .local v1, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->getItemId()I
  
***************
*** 1720,1730 ****
--- 1935,1947 ----
  
      if-ne v3, p1, :cond_0
  
+     .line 625
      .end local v0           #i:I
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :goto_1
      return v0
  
+     .line 618
      .restart local v0       #i:I
      .restart local v1       #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_0
***************
*** 1732,1737 ****
--- 1949,1955 ----
  
      goto :goto_0
  
+     .line 625
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_1
      const/4 v0, -0x1
***************
*** 1749,1793 ****
--- 1967,2022 ----
  
      const/4 v11, 0x0
  
+     .line 824
      iget-object v2, p0, Lcom/android/internal/view/menu/MenuBuilder;->mTempShortcutItemList:Ljava/util/ArrayList;
  
+     .line 825
      .local v2, items:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/MenuItemImpl;>;"
      invoke-virtual {v2}, Ljava/util/ArrayList;->clear()V
  
+     .line 826
      invoke-virtual {p0, v2, p1, p2}, Lcom/android/internal/view/menu/MenuBuilder;->findItemsWithShortcutForKey(Ljava/util/List;ILandroid/view/KeyEvent;)V
  
+     .line 828
      invoke-virtual {v2}, Ljava/util/ArrayList;->isEmpty()Z
  
      move-result v9
  
      if-eqz v9, :cond_1
  
+     .line 859
      :cond_0
      :goto_0
      return-object v8
  
+     .line 832
      :cond_1
      invoke-virtual {p2}, Landroid/view/KeyEvent;->getMetaState()I
  
      move-result v3
  
+     .line 833
      .local v3, metaState:I
      new-instance v4, Landroid/view/KeyCharacterMap$KeyData;
  
      invoke-direct {v4}, Landroid/view/KeyCharacterMap$KeyData;-><init>()V
  
+     .line 835
      .local v4, possibleChars:Landroid/view/KeyCharacterMap$KeyData;
      invoke-virtual {p2, v4}, Landroid/view/KeyEvent;->getKeyData(Landroid/view/KeyCharacterMap$KeyData;)Z
  
+     .line 838
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
  
      move-result v7
  
+     .line 839
      .local v7, size:I
      const/4 v9, 0x1
  
      if-ne v7, v9, :cond_2
  
+     .line 840
      invoke-virtual {v2, v11}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v8
***************
*** 1796,1806 ****
--- 2025,2037 ----
  
      goto :goto_0
  
+     .line 843
      :cond_2
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->isQwertyMode()Z
  
      move-result v5
  
+     .line 846
      .local v5, qwerty:Z
      const/4 v0, 0x0
  
***************
*** 1808,1819 ****
--- 2039,2052 ----
      :goto_1
      if-ge v0, v7, :cond_0
  
+     .line 847
      invoke-virtual {v2, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 848
      .local v1, item:Lcom/android/internal/view/menu/MenuItemImpl;
      if-eqz v5, :cond_6
  
***************
*** 1821,1826 ****
--- 2054,2060 ----
  
      move-result v6
  
+     .line 850
      .local v6, shortcutChar:C
      :goto_2
      iget-object v9, v4, Landroid/view/KeyCharacterMap$KeyData;->meta:[C
***************
*** 1860,1867 ****
--- 2094,2103 ----
      :cond_5
      move-object v8, v1
  
+     .line 856
      goto :goto_0
  
+     .line 848
      .end local v6           #shortcutChar:C
      :cond_6
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->getNumericShortcut()C
***************
*** 1870,1875 ****
--- 2106,2112 ----
  
      goto :goto_2
  
+     .line 846
      .restart local v6       #shortcutChar:C
      :cond_7
      add-int/lit8 v0, v0, 0x1
***************
*** 1898,1930 ****
--- 2135,2174 ----
      .local p1, items:Ljava/util/List;,"Ljava/util/List<Lcom/android/internal/view/menu/MenuItemImpl;>;"
      const/16 v10, 0x43
  
+     .line 781
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->isQwertyMode()Z
  
      move-result v6
  
+     .line 782
      .local v6, qwerty:Z
      invoke-virtual {p3}, Landroid/view/KeyEvent;->getMetaState()I
  
      move-result v4
  
+     .line 783
      .local v4, metaState:I
      new-instance v5, Landroid/view/KeyCharacterMap$KeyData;
  
      invoke-direct {v5}, Landroid/view/KeyCharacterMap$KeyData;-><init>()V
  
+     .line 785
      .local v5, possibleChars:Landroid/view/KeyCharacterMap$KeyData;
      invoke-virtual {p3, v5}, Landroid/view/KeyEvent;->getKeyData(Landroid/view/KeyCharacterMap$KeyData;)Z
  
      move-result v2
  
+     .line 787
      .local v2, isKeyCodeMapped:Z
      if-nez v2, :cond_1
  
      if-eq p2, v10, :cond_1
  
+     .line 809
      :cond_0
      return-void
  
+     .line 792
      :cond_1
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
***************
*** 1932,1937 ****
--- 2176,2182 ----
  
      move-result v0
  
+     .line 793
      .local v0, N:I
      const/4 v1, 0x0
  
***************
*** 1939,1944 ****
--- 2184,2190 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 794
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v8, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1947,1952 ****
--- 2193,2199 ----
  
      check-cast v3, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 795
      .local v3, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v3}, Lcom/android/internal/view/menu/MenuItemImpl;->hasSubMenu()Z
  
***************
*** 1954,1959 ****
--- 2201,2207 ----
  
      if-eqz v8, :cond_2
  
+     .line 796
      invoke-virtual {v3}, Lcom/android/internal/view/menu/MenuItemImpl;->getSubMenu()Landroid/view/SubMenu;
  
      move-result-object v8
***************
*** 1962,1967 ****
--- 2210,2216 ----
  
      invoke-virtual {v8, p1, p2, p3}, Lcom/android/internal/view/menu/MenuBuilder;->findItemsWithShortcutForKey(Ljava/util/List;ILandroid/view/KeyEvent;)V
  
+     .line 798
      :cond_2
      if-eqz v6, :cond_5
  
***************
*** 1969,1974 ****
--- 2218,2224 ----
  
      move-result v7
  
+     .line 799
      .local v7, shortcutChar:C
      :goto_1
      and-int/lit8 v8, v4, 0x5
***************
*** 2008,2020 ****
--- 2258,2273 ----
  
      if-eqz v8, :cond_4
  
+     .line 806
      invoke-interface {p1, v3}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 793
      :cond_4
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 798
      .end local v7           #shortcutChar:C
      :cond_5
      invoke-virtual {v3}, Lcom/android/internal/view/menu/MenuItemImpl;->getNumericShortcut()C
***************
*** 2028,2043 ****
--- 2281,2300 ----
      .locals 10
  
      .prologue
+     .line 1032
      iget-boolean v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsActionItemsStale:Z
  
      if-nez v8, :cond_0
  
+     .line 1068
      :goto_0
      return-void
  
+     .line 1037
      :cond_0
      const/4 v0, 0x0
  
+     .line 1038
      .local v0, flagged:Z
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
***************
*** 2059,2064 ****
--- 2316,2322 ----
  
      check-cast v6, Ljava/lang/ref/WeakReference;
  
+     .line 1039
      .local v6, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      invoke-virtual {v6}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
***************
*** 2066,2080 ****
--- 2324,2341 ----
  
      check-cast v5, Lcom/android/internal/view/menu/MenuPresenter;
  
+     .line 1040
      .local v5, presenter:Lcom/android/internal/view/menu/MenuPresenter;
      if-nez v5, :cond_1
  
+     .line 1041
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v8, v6}, Ljava/util/concurrent/CopyOnWriteArrayList;->remove(Ljava/lang/Object;)Z
  
      goto :goto_1
  
+     .line 1043
      :cond_1
      invoke-interface {v5}, Lcom/android/internal/view/menu/MenuPresenter;->flagActionItems()Z
  
***************
*** 2084,2111 ****
--- 2345,2378 ----
  
      goto :goto_1
  
+     .line 1047
      .end local v5           #presenter:Lcom/android/internal/view/menu/MenuPresenter;
      .end local v6           #ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      :cond_2
      if-eqz v0, :cond_4
  
+     .line 1048
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mActionItems:Ljava/util/ArrayList;
  
      invoke-virtual {v8}, Ljava/util/ArrayList;->clear()V
  
+     .line 1049
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mNonActionItems:Ljava/util/ArrayList;
  
      invoke-virtual {v8}, Ljava/util/ArrayList;->clear()V
  
+     .line 1050
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->getVisibleItems()Ljava/util/ArrayList;
  
      move-result-object v7
  
+     .line 1051
      .local v7, visibleItems:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/view/menu/MenuItemImpl;>;"
      invoke-virtual {v7}, Ljava/util/ArrayList;->size()I
  
      move-result v4
  
+     .line 1052
      .local v4, itemsSize:I
      const/4 v1, 0x0
  
***************
*** 2113,2124 ****
--- 2380,2393 ----
      :goto_2
      if-ge v1, v4, :cond_5
  
+     .line 1053
      invoke-virtual {v7, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v3
  
      check-cast v3, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 1054
      .local v3, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v3}, Lcom/android/internal/view/menu/MenuItemImpl;->isActionButton()Z
  
***************
*** 2126,2140 ****
--- 2395,2412 ----
  
      if-eqz v8, :cond_3
  
+     .line 1055
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mActionItems:Ljava/util/ArrayList;
  
      invoke-virtual {v8, v3}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 1052
      :goto_3
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_2
  
+     .line 1057
      :cond_3
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mNonActionItems:Ljava/util/ArrayList;
  
***************
*** 2142,2147 ****
--- 2414,2420 ----
  
      goto :goto_3
  
+     .line 1063
      .end local v1           #i:I
      .end local v3           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      .end local v4           #itemsSize:I
***************
*** 2151,2160 ****
--- 2424,2435 ----
  
      invoke-virtual {v8}, Ljava/util/ArrayList;->clear()V
  
+     .line 1064
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mNonActionItems:Ljava/util/ArrayList;
  
      invoke-virtual {v8}, Ljava/util/ArrayList;->clear()V
  
+     .line 1065
      iget-object v8, p0, Lcom/android/internal/view/menu/MenuBuilder;->mNonActionItems:Ljava/util/ArrayList;
  
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->getVisibleItems()Ljava/util/ArrayList;
***************
*** 2163,2168 ****
--- 2438,2444 ----
  
      invoke-virtual {v8, v9}, Ljava/util/ArrayList;->addAll(Ljava/util/Collection;)Z
  
+     .line 1067
      :cond_5
      const/4 v8, 0x0
  
***************
*** 2184,2191 ****
--- 2460,2469 ----
      .end annotation
  
      .prologue
+     .line 1071
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->flagActionItems()V
  
+     .line 1072
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mActionItems:Ljava/util/ArrayList;
  
      return-object v0
***************
*** 2195,2200 ****
--- 2473,2479 ----
      .locals 1
  
      .prologue
+     .line 376
      const-string v0, "android:menu:actionviewstates"
  
      return-object v0
***************
*** 2204,2209 ****
--- 2483,2489 ----
      .locals 1
  
      .prologue
+     .line 731
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 2213,2218 ****
--- 2493,2499 ----
      .locals 1
  
      .prologue
+     .line 1263
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
      return-object v0
***************
*** 2222,2227 ****
--- 2503,2509 ----
      .locals 1
  
      .prologue
+     .line 1184
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderIcon:Landroid/graphics/drawable/Drawable;
  
      return-object v0
***************
*** 2231,2236 ****
--- 2513,2519 ----
      .locals 1
  
      .prologue
+     .line 1180
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderTitle:Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 2240,2245 ****
--- 2523,2529 ----
      .locals 1
  
      .prologue
+     .line 1188
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mHeaderView:Landroid/view/View;
  
      return-object v0
***************
*** 2250,2255 ****
--- 2534,2540 ----
      .parameter "index"
  
      .prologue
+     .line 656
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 2274,2281 ****
--- 2559,2568 ----
      .end annotation
  
      .prologue
+     .line 1076
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->flagActionItems()V
  
+     .line 1077
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mNonActionItems:Ljava/util/ArrayList;
  
      return-object v0
***************
*** 2285,2290 ****
--- 2572,2578 ----
      .locals 1
  
      .prologue
+     .line 1215
      iget-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mOptionalIconsVisible:Z
  
      return v0
***************
*** 2294,2299 ****
--- 2582,2588 ----
      .locals 1
  
      .prologue
+     .line 727
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mResources:Landroid/content/res/Resources;
  
      return-object v0
***************
*** 2303,2308 ****
--- 2592,2598 ----
      .locals 0
  
      .prologue
+     .line 1196
      return-object p0
  .end method
  
***************
*** 2319,2344 ****
--- 2609,2639 ----
      .end annotation
  
      .prologue
+     .line 989
      iget-boolean v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsVisibleItemsStale:Z
  
      if-nez v3, :cond_0
  
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mVisibleItems:Ljava/util/ArrayList;
  
+     .line 1004
      :goto_0
      return-object v3
  
+     .line 992
      :cond_0
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mVisibleItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->clear()V
  
+     .line 994
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v2
  
+     .line 996
      .local v2, itemsSize:I
      const/4 v0, 0x0
  
***************
*** 2346,2351 ****
--- 2641,2647 ----
      :goto_1
      if-ge v0, v2, :cond_2
  
+     .line 997
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 2354,2359 ****
--- 2650,2656 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 998
      .local v1, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->isVisible()Z
  
***************
*** 2365,2385 ****
--- 2662,2686 ----
  
      invoke-virtual {v3, v1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 996
      :cond_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 1001
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_2
      const/4 v3, 0x0
  
      iput-boolean v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsVisibleItemsStale:Z
  
+     .line 1002
      const/4 v3, 0x1
  
      iput-boolean v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsActionItemsStale:Z
  
+     .line 1004
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mVisibleItems:Ljava/util/ArrayList;
  
      goto :goto_0
***************
*** 2389,2398 ****
--- 2690,2701 ----
      .locals 4
  
      .prologue
+     .line 585
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->size()I
  
      move-result v2
  
+     .line 587
      .local v2, size:I
      const/4 v0, 0x0
  
***************
*** 2400,2405 ****
--- 2703,2709 ----
      :goto_0
      if-ge v0, v2, :cond_1
  
+     .line 588
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 2408,2413 ****
--- 2712,2718 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 589
      .local v1, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->isVisible()Z
  
***************
*** 2415,2432 ****
--- 2720,2741 ----
  
      if-eqz v3, :cond_0
  
+     .line 590
      const/4 v3, 0x1
  
+     .line 594
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :goto_1
      return v3
  
+     .line 587
      .restart local v1       #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 594
      .end local v1           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_1
      const/4 v3, 0x0
***************
*** 2438,2443 ****
--- 2747,2753 ----
      .locals 1
  
      .prologue
+     .line 694
      iget-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mQwertyMode:Z
  
      return v0
***************
*** 2449,2454 ****
--- 2759,2765 ----
      .parameter "event"
  
      .prologue
+     .line 660
      invoke-virtual {p0, p1, p2}, Lcom/android/internal/view/menu/MenuBuilder;->findItemWithShortcutForKey(ILandroid/view/KeyEvent;)Lcom/android/internal/view/menu/MenuItemImpl;
  
      move-result-object v0
***************
*** 2470,2475 ****
--- 2781,2787 ----
      .locals 1
  
      .prologue
+     .line 723
      iget-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mShortcutsVisible:Z
  
      return v0
***************
*** 2482,2491 ****
--- 2794,2806 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 984
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsActionItemsStale:Z
  
+     .line 985
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 986
      return-void
  .end method
  
***************
*** 2496,2505 ****
--- 2811,2823 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 974
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsVisibleItemsStale:Z
  
+     .line 975
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 976
      return-void
  .end method
  
***************
*** 2510,2531 ****
--- 2828,2856 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 935
      iget-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPreventDispatchingItemsChanged:Z
  
      if-nez v0, :cond_1
  
+     .line 936
      if-eqz p1, :cond_0
  
+     .line 937
      iput-boolean v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsVisibleItemsStale:Z
  
+     .line 938
      iput-boolean v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mIsActionItemsStale:Z
  
+     .line 941
      :cond_0
      invoke-direct {p0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->dispatchPresenterUpdate(Z)V
  
+     .line 945
      :goto_0
      return-void
  
+     .line 943
      :cond_1
      iput-boolean v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItemsChangedWhileDispatchPrevented:Z
  
***************
*** 2538,2543 ****
--- 2863,2869 ----
      .parameter "flags"
  
      .prologue
+     .line 864
      invoke-virtual {p0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->findItem(I)Landroid/view/MenuItem;
  
      move-result-object v0
***************
*** 2559,2568 ****
--- 2885,2896 ----
  
      const/4 v6, 0x1
  
+     .line 868
      move-object v1, p1
  
      check-cast v1, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 870
      .local v1, itemImpl:Lcom/android/internal/view/menu/MenuItemImpl;
      if-eqz v1, :cond_0
  
***************
*** 2575,2589 ****
--- 2903,2920 ----
      :cond_0
      move v0, v4
  
+     .line 895
      :cond_1
      :goto_0
      return v0
  
+     .line 874
      :cond_2
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->invoke()Z
  
      move-result v0
  
+     .line 876
      .local v0, invoked:Z
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->hasCollapsibleActionView()Z
  
***************
*** 2591,2608 ****
--- 2922,2942 ----
  
      if-eqz v5, :cond_3
  
+     .line 877
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->expandActionView()Z
  
      move-result v4
  
      or-int/2addr v0, v4
  
+     .line 878
      if-eqz v0, :cond_1
  
      invoke-virtual {p0, v6}, Lcom/android/internal/view/menu/MenuBuilder;->close(Z)V
  
      goto :goto_0
  
+     .line 879
      :cond_3
      invoke-interface {p1}, Landroid/view/MenuItem;->hasSubMenu()Z
  
***************
*** 2610,2628 ****
--- 2944,2966 ----
  
      if-eqz v5, :cond_5
  
+     .line 880
      invoke-virtual {p0, v4}, Lcom/android/internal/view/menu/MenuBuilder;->close(Z)V
  
+     .line 882
      invoke-interface {p1}, Landroid/view/MenuItem;->getSubMenu()Landroid/view/SubMenu;
  
      move-result-object v3
  
      check-cast v3, Lcom/android/internal/view/menu/SubMenuBuilder;
  
+     .line 883
      .local v3, subMenu:Lcom/android/internal/view/menu/SubMenuBuilder;
      invoke-interface {p1}, Landroid/view/MenuItem;->getActionProvider()Landroid/view/ActionProvider;
  
      move-result-object v2
  
+     .line 884
      .local v2, provider:Landroid/view/ActionProvider;
      if-eqz v2, :cond_4
  
***************
*** 2632,2639 ****
--- 2970,2979 ----
  
      if-eqz v4, :cond_4
  
+     .line 885
      invoke-virtual {v2, v3}, Landroid/view/ActionProvider;->onPrepareSubMenu(Landroid/view/SubMenu;)V
  
+     .line 887
      :cond_4
      invoke-direct {p0, v3}, Lcom/android/internal/view/menu/MenuBuilder;->dispatchSubMenuSelected(Lcom/android/internal/view/menu/SubMenuBuilder;)Z
  
***************
*** 2641,2652 ****
--- 2981,2994 ----
  
      or-int/2addr v0, v4
  
+     .line 888
      if-nez v0, :cond_1
  
      invoke-virtual {p0, v6}, Lcom/android/internal/view/menu/MenuBuilder;->close(Z)V
  
      goto :goto_0
  
+     .line 890
      .end local v2           #provider:Landroid/view/ActionProvider;
      .end local v3           #subMenu:Lcom/android/internal/view/menu/SubMenuBuilder;
      :cond_5
***************
*** 2654,2659 ****
--- 2996,3002 ----
  
      if-nez v4, :cond_1
  
+     .line 891
      invoke-virtual {p0, v6}, Lcom/android/internal/view/menu/MenuBuilder;->close(Z)V
  
      goto :goto_0
***************
*** 2666,2694 ****
--- 3009,3044 ----
      .parameter "flags"
  
      .prologue
+     .line 759
      invoke-virtual {p0, p1, p2}, Lcom/android/internal/view/menu/MenuBuilder;->findItemWithShortcutForKey(ILandroid/view/KeyEvent;)Lcom/android/internal/view/menu/MenuItemImpl;
  
      move-result-object v1
  
+     .line 761
      .local v1, item:Lcom/android/internal/view/menu/MenuItemImpl;
      const/4 v0, 0x0
  
+     .line 763
      .local v0, handled:Z
      if-eqz v1, :cond_0
  
+     .line 764
      invoke-virtual {p0, v1, p3}, Lcom/android/internal/view/menu/MenuBuilder;->performItemAction(Landroid/view/MenuItem;I)Z
  
      move-result v0
  
+     .line 767
      :cond_0
      and-int/lit8 v2, p3, 0x2
  
      if-eqz v2, :cond_1
  
+     .line 768
      const/4 v2, 0x1
  
      invoke-virtual {p0, v2}, Lcom/android/internal/view/menu/MenuBuilder;->close(Z)V
  
+     .line 771
      :cond_1
      return v0
  .end method
***************
*** 2698,2710 ****
--- 3048,3063 ----
      .parameter "group"
  
      .prologue
+     .line 473
      invoke-virtual {p0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->findGroupIndex(I)I
  
      move-result v0
  
+     .line 475
      .local v0, i:I
      if-ltz v0, :cond_1
  
+     .line 476
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v4}, Ljava/util/ArrayList;->size()I
***************
*** 2713,2724 ****
--- 3066,3079 ----
  
      sub-int v1, v4, v0
  
+     .line 477
      .local v1, maxRemovable:I
      const/4 v2, 0x0
  
      .local v2, numRemoved:I
      move v3, v2
  
+     .line 478
      .end local v2           #numRemoved:I
      .local v3, numRemoved:I
      :goto_0
***************
*** 2742,2747 ****
--- 3097,3103 ----
  
      if-ne v4, p1, :cond_0
  
+     .line 480
      const/4 v4, 0x0
  
      invoke-direct {p0, v0, v4}, Lcom/android/internal/view/menu/MenuBuilder;->removeItemAtInt(IZ)V
***************
*** 2752,2757 ****
--- 3108,3114 ----
      .restart local v3       #numRemoved:I
      goto :goto_0
  
+     .line 484
      .end local v3           #numRemoved:I
      .restart local v2       #numRemoved:I
      :cond_0
***************
*** 2759,2764 ****
--- 3116,3122 ----
  
      invoke-virtual {p0, v4}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 486
      .end local v1           #maxRemovable:I
      .end local v2           #numRemoved:I
      :cond_1
***************
*** 2770,2775 ****
--- 3128,3134 ----
      .parameter "id"
  
      .prologue
+     .line 469
      invoke-virtual {p0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->findItemIndex(I)I
  
      move-result v0
***************
*** 2778,2783 ****
--- 3137,3143 ----
  
      invoke-direct {p0, v0, v1}, Lcom/android/internal/view/menu/MenuBuilder;->removeItemAtInt(IZ)V
  
+     .line 470
      return-void
  .end method
  
***************
*** 2786,2795 ****
--- 3146,3157 ----
      .parameter "index"
  
      .prologue
+     .line 507
      const/4 v0, 0x1
  
      invoke-direct {p0, p1, v0}, Lcom/android/internal/view/menu/MenuBuilder;->removeItemAtInt(IZ)V
  
+     .line 508
      return-void
  .end method
  
***************
*** 2798,2803 ****
--- 3160,3166 ----
      .parameter "presenter"
  
      .prologue
+     .line 227
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
      invoke-virtual {v3}, Ljava/util/concurrent/CopyOnWriteArrayList;->iterator()Ljava/util/Iterator;
***************
*** 2819,2824 ****
--- 3182,3188 ----
  
      check-cast v2, Ljava/lang/ref/WeakReference;
  
+     .line 228
      .local v2, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      invoke-virtual {v2}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
***************
*** 2826,2836 ****
--- 3190,3202 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuPresenter;
  
+     .line 229
      .local v1, item:Lcom/android/internal/view/menu/MenuPresenter;
      if-eqz v1, :cond_1
  
      if-ne v1, p1, :cond_0
  
+     .line 230
      :cond_1
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPresenters:Ljava/util/concurrent/CopyOnWriteArrayList;
  
***************
*** 2838,2843 ****
--- 3204,3210 ----
  
      goto :goto_0
  
+     .line 233
      .end local v1           #item:Lcom/android/internal/view/menu/MenuPresenter;
      .end local v2           #ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Lcom/android/internal/view/menu/MenuPresenter;>;"
      :cond_2
***************
*** 2849,2860 ****
--- 3216,3230 ----
      .parameter "states"
  
      .prologue
+     .line 346
      if-nez p1, :cond_1
  
+     .line 373
      :cond_0
      :goto_0
      return-void
  
+     .line 350
      :cond_1
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->getActionViewStatesKey()Ljava/lang/String;
  
***************
*** 2864,2874 ****
--- 3234,3246 ----
  
      move-result-object v7
  
+     .line 353
      .local v7, viewStates:Landroid/util/SparseArray;,"Landroid/util/SparseArray<Landroid/os/Parcelable;>;"
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->size()I
  
      move-result v3
  
+     .line 354
      .local v3, itemCount:I
      const/4 v1, 0x0
  
***************
*** 2876,2890 ****
--- 3248,3265 ----
      :goto_1
      if-ge v1, v3, :cond_4
  
+     .line 355
      invoke-virtual {p0, v1}, Lcom/android/internal/view/menu/MenuBuilder;->getItem(I)Landroid/view/MenuItem;
  
      move-result-object v2
  
+     .line 356
      .local v2, item:Landroid/view/MenuItem;
      invoke-interface {v2}, Landroid/view/MenuItem;->getActionView()Landroid/view/View;
  
      move-result-object v6
  
+     .line 357
      .local v6, v:Landroid/view/View;
      if-eqz v6, :cond_2
  
***************
*** 2896,2903 ****
--- 3271,3280 ----
  
      if-eq v8, v9, :cond_2
  
+     .line 358
      invoke-virtual {v6, v7}, Landroid/view/View;->restoreHierarchyState(Landroid/util/SparseArray;)V
  
+     .line 360
      :cond_2
      invoke-interface {v2}, Landroid/view/MenuItem;->hasSubMenu()Z
  
***************
*** 2905,2925 ****
--- 3282,3306 ----
  
      if-eqz v8, :cond_3
  
+     .line 361
      invoke-interface {v2}, Landroid/view/MenuItem;->getSubMenu()Landroid/view/SubMenu;
  
      move-result-object v5
  
      check-cast v5, Lcom/android/internal/view/menu/SubMenuBuilder;
  
+     .line 362
      .local v5, subMenu:Lcom/android/internal/view/menu/SubMenuBuilder;
      invoke-virtual {v5, p1}, Lcom/android/internal/view/menu/SubMenuBuilder;->restoreActionViewStates(Landroid/os/Bundle;)V
  
+     .line 354
      .end local v5           #subMenu:Lcom/android/internal/view/menu/SubMenuBuilder;
      :cond_3
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 366
      .end local v2           #item:Landroid/view/MenuItem;
      .end local v6           #v:Landroid/view/View;
      :cond_4
***************
*** 2929,2944 ****
--- 3310,3329 ----
  
      move-result v0
  
+     .line 367
      .local v0, expandedId:I
      if-lez v0, :cond_0
  
+     .line 368
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->findItem(I)Landroid/view/MenuItem;
  
      move-result-object v4
  
+     .line 369
      .local v4, itemToExpand:Landroid/view/MenuItem;
      if-eqz v4, :cond_0
  
+     .line 370
      invoke-interface {v4}, Landroid/view/MenuItem;->expandActionView()Z
  
      goto :goto_0
***************
*** 2949,2956 ****
--- 3334,3343 ----
      .parameter "state"
  
      .prologue
+     .line 315
      invoke-direct {p0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->dispatchRestoreInstanceState(Landroid/os/Bundle;)V
  
+     .line 316
      return-void
  .end method
  
***************
*** 2959,2971 ****
--- 3346,3361 ----
      .parameter "outStates"
  
      .prologue
+     .line 319
      const/4 v5, 0x0
  
+     .line 321
      .local v5, viewStates:Landroid/util/SparseArray;,"Landroid/util/SparseArray<Landroid/os/Parcelable;>;"
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->size()I
  
      move-result v2
  
+     .line 322
      .local v2, itemCount:I
      const/4 v0, 0x0
  
***************
*** 2973,2987 ****
--- 3363,3380 ----
      :goto_0
      if-ge v0, v2, :cond_3
  
+     .line 323
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->getItem(I)Landroid/view/MenuItem;
  
      move-result-object v1
  
+     .line 324
      .local v1, item:Landroid/view/MenuItem;
      invoke-interface {v1}, Landroid/view/MenuItem;->getActionView()Landroid/view/View;
  
      move-result-object v4
  
+     .line 325
      .local v4, v:Landroid/view/View;
      if-eqz v4, :cond_1
  
***************
*** 2993,3015 ****
--- 3386,3413 ----
  
      if-eq v6, v7, :cond_1
  
+     .line 326
      if-nez v5, :cond_0
  
+     .line 327
      new-instance v5, Landroid/util/SparseArray;
  
      .end local v5           #viewStates:Landroid/util/SparseArray;,"Landroid/util/SparseArray<Landroid/os/Parcelable;>;"
      invoke-direct {v5}, Landroid/util/SparseArray;-><init>()V
  
+     .line 329
      .restart local v5       #viewStates:Landroid/util/SparseArray;,"Landroid/util/SparseArray<Landroid/os/Parcelable;>;"
      :cond_0
      invoke-virtual {v4, v5}, Landroid/view/View;->saveHierarchyState(Landroid/util/SparseArray;)V
  
+     .line 330
      invoke-interface {v1}, Landroid/view/MenuItem;->isActionViewExpanded()Z
  
      move-result v6
  
      if-eqz v6, :cond_1
  
+     .line 331
      const-string v6, "android:menu:expandedactionview"
  
      invoke-interface {v1}, Landroid/view/MenuItem;->getItemId()I
***************
*** 3018,3023 ****
--- 3416,3422 ----
  
      invoke-virtual {p1, v6, v7}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
+     .line 334
      :cond_1
      invoke-interface {v1}, Landroid/view/MenuItem;->hasSubMenu()Z
  
***************
*** 3025,3056 ****
--- 3424,3461 ----
  
      if-eqz v6, :cond_2
  
+     .line 335
      invoke-interface {v1}, Landroid/view/MenuItem;->getSubMenu()Landroid/view/SubMenu;
  
      move-result-object v3
  
      check-cast v3, Lcom/android/internal/view/menu/SubMenuBuilder;
  
+     .line 336
      .local v3, subMenu:Lcom/android/internal/view/menu/SubMenuBuilder;
      invoke-virtual {v3, p1}, Lcom/android/internal/view/menu/SubMenuBuilder;->saveActionViewStates(Landroid/os/Bundle;)V
  
+     .line 322
      .end local v3           #subMenu:Lcom/android/internal/view/menu/SubMenuBuilder;
      :cond_2
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 340
      .end local v1           #item:Landroid/view/MenuItem;
      .end local v4           #v:Landroid/view/View;
      :cond_3
      if-eqz v5, :cond_4
  
+     .line 341
      invoke-virtual {p0}, Lcom/android/internal/view/menu/MenuBuilder;->getActionViewStatesKey()Ljava/lang/String;
  
      move-result-object v6
  
      invoke-virtual {p1, v6, v5}, Landroid/os/Bundle;->putSparseParcelableArray(Ljava/lang/String;Landroid/util/SparseArray;)V
  
+     .line 343
      :cond_4
      return-void
  .end method
***************
*** 3060,3067 ****
--- 3465,3474 ----
      .parameter "outState"
  
      .prologue
+     .line 311
      invoke-direct {p0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->dispatchSaveInstanceState(Landroid/os/Bundle;)V
  
+     .line 312
      return-void
  .end method
  
***************
*** 3070,3077 ****
--- 3477,3486 ----
      .parameter "cb"
  
      .prologue
+     .line 380
      iput-object p1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mCallback:Lcom/android/internal/view/menu/MenuBuilder$Callback;
  
+     .line 381
      return-void
  .end method
  
***************
*** 3080,3087 ****
--- 3489,3498 ----
      .parameter "menuInfo"
  
      .prologue
+     .line 1207
      iput-object p1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mCurrentMenuInfo:Landroid/view/ContextMenu$ContextMenuInfo;
  
+     .line 1208
      return-void
  .end method
  
***************
*** 3090,3097 ****
--- 3501,3510 ----
      .parameter "defaultShowAsAction"
  
      .prologue
+     .line 203
      iput p1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mDefaultShowAsAction:I
  
+     .line 204
      return-object p0
  .end method
  
***************
*** 3100,3109 ****
--- 3513,3524 ----
      .parameter "item"
  
      .prologue
+     .line 529
      invoke-interface {p1}, Landroid/view/MenuItem;->getGroupId()I
  
      move-result v2
  
+     .line 531
      .local v2, group:I
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
***************
*** 3111,3116 ****
--- 3526,3532 ----
  
      move-result v0
  
+     .line 532
      .local v0, N:I
      const/4 v3, 0x0
  
***************
*** 3118,3123 ****
--- 3534,3540 ----
      :goto_0
      if-ge v3, v0, :cond_3
  
+     .line 533
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v3}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 3126,3131 ****
--- 3543,3549 ----
  
      check-cast v1, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 534
      .local v1, curItem:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->getGroupId()I
  
***************
*** 3133,3150 ****
--- 3551,3571 ----
  
      if-ne v4, v2, :cond_0
  
+     .line 535
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->isExclusiveCheckable()Z
  
      move-result v4
  
      if-nez v4, :cond_1
  
+     .line 532
      :cond_0
      :goto_1
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 536
      :cond_1
      invoke-virtual {v1}, Lcom/android/internal/view/menu/MenuItemImpl;->isCheckable()Z
  
***************
*** 3152,3157 ****
--- 3573,3579 ----
  
      if-eqz v4, :cond_0
  
+     .line 539
      if-ne v1, p1, :cond_2
  
      const/4 v4, 0x1
***************
*** 3166,3171 ****
--- 3588,3594 ----
  
      goto :goto_2
  
+     .line 542
      .end local v1           #curItem:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_3
      return-void
***************
*** 3178,3189 ****
--- 3601,3614 ----
      .parameter "exclusive"
  
      .prologue
+     .line 545
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v0
  
+     .line 547
      .local v0, N:I
      const/4 v1, 0x0
  
***************
*** 3191,3196 ****
--- 3616,3622 ----
      :goto_0
      if-ge v1, v0, :cond_1
  
+     .line 548
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 3199,3204 ****
--- 3625,3631 ----
  
      check-cast v2, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 549
      .local v2, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v2}, Lcom/android/internal/view/menu/MenuItemImpl;->getGroupId()I
  
***************
*** 3206,3220 ****
--- 3633,3651 ----
  
      if-ne v3, p1, :cond_0
  
+     .line 550
      invoke-virtual {v2, p3}, Lcom/android/internal/view/menu/MenuItemImpl;->setExclusiveCheckable(Z)V
  
+     .line 551
      invoke-virtual {v2, p2}, Lcom/android/internal/view/menu/MenuItemImpl;->setCheckable(Z)Landroid/view/MenuItem;
  
+     .line 547
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 554
      .end local v2           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_1
      return-void
***************
*** 3226,3237 ****
--- 3657,3670 ----
      .parameter "enabled"
  
      .prologue
+     .line 574
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v0
  
+     .line 576
      .local v0, N:I
      const/4 v1, 0x0
  
***************
*** 3239,3244 ****
--- 3672,3678 ----
      :goto_0
      if-ge v1, v0, :cond_1
  
+     .line 577
      iget-object v3, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 3247,3252 ****
--- 3681,3687 ----
  
      check-cast v2, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 578
      .local v2, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v2}, Lcom/android/internal/view/menu/MenuItemImpl;->getGroupId()I
  
***************
*** 3254,3266 ****
--- 3689,3704 ----
  
      if-ne v3, p1, :cond_0
  
+     .line 579
      invoke-virtual {v2, p2}, Lcom/android/internal/view/menu/MenuItemImpl;->setEnabled(Z)Landroid/view/MenuItem;
  
+     .line 576
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 582
      .end local v2           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_1
      return-void
***************
*** 3272,3286 ****
--- 3710,3727 ----
      .parameter "visible"
  
      .prologue
+     .line 557
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v4}, Ljava/util/ArrayList;->size()I
  
      move-result v0
  
+     .line 562
      .local v0, N:I
      const/4 v1, 0x0
  
+     .line 563
      .local v1, changedAtLeastOneItem:Z
      const/4 v2, 0x0
  
***************
*** 3288,3293 ****
--- 3729,3735 ----
      :goto_0
      if-ge v2, v0, :cond_1
  
+     .line 564
      iget-object v4, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v2}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 3296,3301 ****
--- 3738,3744 ----
  
      check-cast v3, Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 565
      .local v3, item:Lcom/android/internal/view/menu/MenuItemImpl;
      invoke-virtual {v3}, Lcom/android/internal/view/menu/MenuItemImpl;->getGroupId()I
  
***************
*** 3303,3308 ****
--- 3746,3752 ----
  
      if-ne v4, p1, :cond_0
  
+     .line 566
      invoke-virtual {v3, p2}, Lcom/android/internal/view/menu/MenuItemImpl;->setVisibleInt(Z)Z
  
      move-result v4
***************
*** 3311,3321 ****
--- 3755,3767 ----
  
      const/4 v1, 0x1
  
+     .line 563
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 570
      .end local v3           #item:Lcom/android/internal/view/menu/MenuItemImpl;
      :cond_1
      if-eqz v1, :cond_2
***************
*** 3324,3329 ****
--- 3770,3776 ----
  
      invoke-virtual {p0, v4}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 571
      :cond_2
      return-void
  .end method
***************
*** 3335,3340 ****
--- 3782,3788 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1163
      const/4 v1, 0x0
  
      move-object v0, p0
***************
*** 3347,3352 ****
--- 3795,3801 ----
  
      invoke-direct/range {v0 .. v5}, Lcom/android/internal/view/menu/MenuBuilder;->setHeaderInternal(ILjava/lang/CharSequence;ILandroid/graphics/drawable/Drawable;Landroid/view/View;)V
  
+     .line 1164
      return-object p0
  .end method
  
***************
*** 3359,3364 ****
--- 3808,3814 ----
  
      const/4 v1, 0x0
  
+     .line 1151
      move-object v0, p0
  
      move v3, v1
***************
*** 3369,3374 ****
--- 3819,3825 ----
  
      invoke-direct/range {v0 .. v5}, Lcom/android/internal/view/menu/MenuBuilder;->setHeaderInternal(ILjava/lang/CharSequence;ILandroid/graphics/drawable/Drawable;Landroid/view/View;)V
  
+     .line 1152
      return-object p0
  .end method
  
***************
*** 3379,3384 ****
--- 3830,3836 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1139
      const/4 v3, 0x0
  
      move-object v0, p0
***************
*** 3391,3396 ****
--- 3843,3849 ----
  
      invoke-direct/range {v0 .. v5}, Lcom/android/internal/view/menu/MenuBuilder;->setHeaderInternal(ILjava/lang/CharSequence;ILandroid/graphics/drawable/Drawable;Landroid/view/View;)V
  
+     .line 1140
      return-object p0
  .end method
  
***************
*** 3403,3408 ****
--- 3856,3862 ----
  
      const/4 v1, 0x0
  
+     .line 1127
      move-object v0, p0
  
      move-object v2, p1
***************
*** 3413,3418 ****
--- 3867,3873 ----
  
      invoke-direct/range {v0 .. v5}, Lcom/android/internal/view/menu/MenuBuilder;->setHeaderInternal(ILjava/lang/CharSequence;ILandroid/graphics/drawable/Drawable;Landroid/view/View;)V
  
+     .line 1128
      return-object p0
  .end method
  
***************
*** 3425,3430 ****
--- 3880,3886 ----
  
      const/4 v1, 0x0
  
+     .line 1175
      move-object v0, p0
  
      move v3, v1
***************
*** 3435,3440 ****
--- 3891,3897 ----
  
      invoke-direct/range {v0 .. v5}, Lcom/android/internal/view/menu/MenuBuilder;->setHeaderInternal(ILjava/lang/CharSequence;ILandroid/graphics/drawable/Drawable;Landroid/view/View;)V
  
+     .line 1176
      return-object p0
  .end method
  
***************
*** 3443,3450 ****
--- 3900,3909 ----
      .parameter "visible"
  
      .prologue
+     .line 1211
      iput-boolean p1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mOptionalIconsVisible:Z
  
+     .line 1212
      return-void
  .end method
  
***************
*** 3453,3464 ****
--- 3912,3926 ----
      .parameter "isQwerty"
  
      .prologue
+     .line 664
      iput-boolean p1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mQwertyMode:Z
  
+     .line 666
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 667
      return-void
  .end method
  
***************
*** 3467,3482 ****
--- 3929,3948 ----
      .parameter "shortcutsVisible"
  
      .prologue
+     .line 706
      iget-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mShortcutsVisible:Z
  
      if-ne v0, p1, :cond_0
  
+     .line 710
      :goto_0
      return-void
  
+     .line 708
      :cond_0
      invoke-direct {p0, p1}, Lcom/android/internal/view/menu/MenuBuilder;->setShortcutsVisibleInner(Z)V
  
+     .line 709
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
***************
*** 3488,3493 ****
--- 3954,3960 ----
      .locals 1
  
      .prologue
+     .line 651
      iget-object v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItems:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->size()I
***************
*** 3503,3520 ****
--- 3970,3992 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 960
      iput-boolean v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPreventDispatchingItemsChanged:Z
  
+     .line 962
      iget-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItemsChangedWhileDispatchPrevented:Z
  
      if-eqz v0, :cond_0
  
+     .line 963
      iput-boolean v1, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItemsChangedWhileDispatchPrevented:Z
  
+     .line 964
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Lcom/android/internal/view/menu/MenuBuilder;->onItemsChanged(Z)V
  
+     .line 966
      :cond_0
      return-void
  .end method
***************
*** 3523,3540 ****
--- 3995,4016 ----
      .locals 1
  
      .prologue
+     .line 953
      iget-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPreventDispatchingItemsChanged:Z
  
      if-nez v0, :cond_0
  
+     .line 954
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mPreventDispatchingItemsChanged:Z
  
+     .line 955
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/internal/view/menu/MenuBuilder;->mItemsChangedWhileDispatchPrevented:Z
  
+     .line 957
      :cond_0
      return-void
  .end method
