*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$4.smali	2012-03-23 19:36:20.371052450 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$4.smali.original	2012-03-23 19:34:45.061314479 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .parameter
  
      .prologue
+     .line 825
      .local p4, x2:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      iput-object p1, p0, Landroid/accounts/AccountManager$4;->this$0:Landroid/accounts/AccountManager;
  
***************
*** 61,66 ****
--- 62,68 ----
      .end annotation
  
      .prologue
+     .line 827
      iget-object v0, p0, Landroid/accounts/AccountManager$4;->this$0:Landroid/accounts/AccountManager;
  
      #getter for: Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 82,86 ****
--- 84,89 ----
  
      invoke-interface/range {v0 .. v6}, Landroid/accounts/IAccountManager;->getAuthToken(Landroid/accounts/IAccountManagerResponse;Landroid/accounts/Account;Ljava/lang/String;ZZLandroid/os/Bundle;)V
  
+     .line 830
      return-void
  .end method
