*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManagerService$6.smali	2012-03-23 19:36:20.010045867 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManagerService$6.smali.original	2012-03-23 19:34:45.058314425 -0500
***************
*** 37,42 ****
--- 37,43 ----
      .parameter
  
      .prologue
+     .line 1209
      iput-object p1, p0, Landroid/accounts/AccountManagerService$6;->this$0:Landroid/accounts/AccountManagerService;
  
      iput-object p6, p0, Landroid/accounts/AccountManagerService$6;->val$account:Landroid/accounts/Account;
***************
*** 61,66 ****
--- 62,68 ----
      .end annotation
  
      .prologue
+     .line 1211
      iget-object v0, p0, Landroid/accounts/AccountManagerService$6;->mAuthenticator:Landroid/accounts/IAccountAuthenticator;
  
      iget-object v1, p0, Landroid/accounts/AccountManagerService$6;->val$account:Landroid/accounts/Account;
***************
*** 71,76 ****
--- 73,79 ----
  
      invoke-interface {v0, p0, v1, v2, v3}, Landroid/accounts/IAccountAuthenticator;->updateCredentials(Landroid/accounts/IAccountAuthenticatorResponse;Landroid/accounts/Account;Ljava/lang/String;Landroid/os/Bundle;)V
  
+     .line 1212
      return-void
  .end method
  
***************
*** 79,84 ****
--- 82,88 ----
      .parameter "now"
  
      .prologue
+     .line 1214
      iget-object v0, p0, Landroid/accounts/AccountManagerService$6;->val$loginOptions:Landroid/os/Bundle;
  
      if-eqz v0, :cond_0
***************
*** 87,92 ****
--- 91,97 ----
  
      invoke-virtual {v0}, Landroid/os/Bundle;->keySet()Ljava/util/Set;
  
+     .line 1215
      :cond_0
      new-instance v0, Ljava/lang/StringBuilder;
  
