*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$7.smali	2012-03-23 19:36:22.195085710 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$7.smali.original	2012-03-23 19:34:45.076314753 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .parameter
  
      .prologue
+     .line 1130
      .local p4, x2:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      iput-object p1, p0, Landroid/accounts/AccountManager$7;->this$0:Landroid/accounts/AccountManager;
  
***************
*** 61,66 ****
--- 62,68 ----
      .end annotation
  
      .prologue
+     .line 1132
      iget-object v0, p0, Landroid/accounts/AccountManager$7;->this$0:Landroid/accounts/AccountManager;
  
      #getter for: Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 83,90 ****
--- 85,94 ----
      :goto_0
      invoke-interface {v1, v2, v3, v4, v0}, Landroid/accounts/IAccountManager;->confirmCredentials(Landroid/accounts/IAccountManagerResponse;Landroid/accounts/Account;Landroid/os/Bundle;Z)V
  
+     .line 1133
      return-void
  
+     .line 1132
      :cond_0
      const/4 v0, 0x0
  
