*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$6.smali	2012-03-23 19:36:20.328051665 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$6.smali.original	2012-03-23 19:34:45.061314479 -0500
***************
*** 42,47 ****
--- 42,48 ----
      .parameter
  
      .prologue
+     .line 1062
      .local p4, x2:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      iput-object p1, p0, Landroid/accounts/AccountManager$6;->this$0:Landroid/accounts/AccountManager;
  
***************
*** 71,76 ****
--- 72,78 ----
      .end annotation
  
      .prologue
+     .line 1064
      iget-object v0, p0, Landroid/accounts/AccountManager$6;->this$0:Landroid/accounts/AccountManager;
  
      #getter for: Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 97,104 ****
--- 99,108 ----
  
      invoke-interface/range {v0 .. v6}, Landroid/accounts/IAccountManager;->addAcount(Landroid/accounts/IAccountManagerResponse;Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;ZLandroid/os/Bundle;)V
  
+     .line 1066
      return-void
  
+     .line 1064
      :cond_0
      const/4 v5, 0x0
  
