*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$11.smali	2012-03-23 19:36:21.379070830 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$11.smali.original	2012-03-23 19:34:45.069314625 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .parameter
  
      .prologue
+     .line 1276
      iput-object p1, p0, Landroid/accounts/AccountManager$11;->this$0:Landroid/accounts/AccountManager;
  
      iput-object p2, p0, Landroid/accounts/AccountManager$11;->val$listener:Landroid/accounts/OnAccountsUpdateListener;
***************
*** 50,55 ****
--- 51,57 ----
      .locals 3
  
      .prologue
+     .line 1279
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager$11;->val$listener:Landroid/accounts/OnAccountsUpdateListener;
  
***************
*** 59,70 ****
--- 61,75 ----
      :try_end_0
      .catch Landroid/database/SQLException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1285
      :goto_0
      return-void
  
+     .line 1280
      :catch_0
      move-exception v0
  
+     .line 1283
      .local v0, e:Landroid/database/SQLException;
      const-string v1, "AccountManager"
  
