*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountAuthenticatorResponse.smali	2012-03-23 19:36:19.984045393 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountAuthenticatorResponse.smali.original	2012-03-23 19:34:45.058314425 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .locals 1
  
      .prologue
+     .line 88
      new-instance v0, Landroid/accounts/AccountAuthenticatorResponse$1;
  
      invoke-direct {v0}, Landroid/accounts/AccountAuthenticatorResponse$1;-><init>()V
***************
*** 44,53 ****
--- 45,57 ----
      .parameter "response"
  
      .prologue
+     .line 36
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 37
      iput-object p1, p0, Landroid/accounts/AccountAuthenticatorResponse;->mAccountAuthenticatorResponse:Landroid/accounts/IAccountAuthenticatorResponse;
  
+     .line 38
      return-void
  .end method
  
***************
*** 56,63 ****
--- 60,69 ----
      .parameter "parcel"
  
      .prologue
+     .line 40
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 41
      invoke-virtual {p1}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
***************
*** 68,73 ****
--- 74,80 ----
  
      iput-object v0, p0, Landroid/accounts/AccountAuthenticatorResponse;->mAccountAuthenticatorResponse:Landroid/accounts/IAccountAuthenticatorResponse;
  
+     .line 43
      return-void
  .end method
  
***************
*** 77,82 ****
--- 84,90 ----
      .locals 1
  
      .prologue
+     .line 81
      const/4 v0, 0x0
  
      return v0
***************
*** 88,93 ****
--- 96,102 ----
      .parameter "errorMessage"
  
      .prologue
+     .line 70
      const-string v0, "AccountAuthenticator"
  
      const/4 v1, 0x2
***************
*** 98,103 ****
--- 107,113 ----
  
      if-eqz v0, :cond_0
  
+     .line 71
      const-string v0, "AccountAuthenticator"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 130,135 ****
--- 140,146 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 74
      :cond_0
      :try_start_0
      iget-object v0, p0, Landroid/accounts/AccountAuthenticatorResponse;->mAccountAuthenticatorResponse:Landroid/accounts/IAccountAuthenticatorResponse;
***************
*** 138,146 ****
--- 149,159 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 78
      :goto_0
      return-void
  
+     .line 75
      :catch_0
      move-exception v0
  
***************
*** 151,156 ****
--- 164,170 ----
      .locals 2
  
      .prologue
+     .line 59
      const-string v0, "AccountAuthenticator"
  
      const/4 v1, 0x2
***************
*** 161,172 ****
--- 175,188 ----
  
      if-eqz v0, :cond_0
  
+     .line 60
      const-string v0, "AccountAuthenticator"
  
      const-string v1, "AccountAuthenticatorResponse.onRequestContinued"
  
      invoke-static {v0, v1}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 63
      :cond_0
      :try_start_0
      iget-object v0, p0, Landroid/accounts/AccountAuthenticatorResponse;->mAccountAuthenticatorResponse:Landroid/accounts/IAccountAuthenticatorResponse;
***************
*** 175,183 ****
--- 191,201 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 67
      :goto_0
      return-void
  
+     .line 64
      :catch_0
      move-exception v0
  
***************
*** 189,194 ****
--- 207,213 ----
      .parameter "result"
  
      .prologue
+     .line 46
      const-string v0, "AccountAuthenticator"
  
      const/4 v1, 0x2
***************
*** 199,206 ****
--- 218,227 ----
  
      if-eqz v0, :cond_0
  
+     .line 47
      invoke-virtual {p1}, Landroid/os/Bundle;->keySet()Ljava/util/Set;
  
+     .line 48
      const-string v0, "AccountAuthenticator"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 227,232 ****
--- 248,254 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 52
      :cond_0
      :try_start_0
      iget-object v0, p0, Landroid/accounts/AccountAuthenticatorResponse;->mAccountAuthenticatorResponse:Landroid/accounts/IAccountAuthenticatorResponse;
***************
*** 235,243 ****
--- 257,267 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 56
      :goto_0
      return-void
  
+     .line 53
      :catch_0
      move-exception v0
  
***************
*** 250,255 ****
--- 274,280 ----
      .parameter "flags"
  
      .prologue
+     .line 85
      iget-object v0, p0, Landroid/accounts/AccountAuthenticatorResponse;->mAccountAuthenticatorResponse:Landroid/accounts/IAccountAuthenticatorResponse;
  
      invoke-interface {v0}, Landroid/accounts/IAccountAuthenticatorResponse;->asBinder()Landroid/os/IBinder;
***************
*** 258,262 ****
--- 283,288 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 86
      return-void
  .end method
