*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask.smali	2012-03-23 19:36:21.998082118 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask.smali.original	2012-03-23 19:34:45.074314716 -0500
***************
*** 97,115 ****
--- 97,120 ----
      .end annotation
  
      .prologue
+     .line 1571
      .local p8, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      iput-object p1, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->this$0:Landroid/accounts/AccountManager;
  
+     .line 1572
      invoke-direct {p0, p1, p5, p9, p8}, Landroid/accounts/AccountManager$AmsTask;-><init>(Landroid/accounts/AccountManager;Landroid/app/Activity;Landroid/os/Handler;Landroid/accounts/AccountManagerCallback;)V
  
+     .line 1581
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mFuture:Landroid/accounts/AccountManagerFuture;
  
+     .line 1588
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mNumAccounts:I
  
+     .line 1573
      if-nez p2, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
***************
*** 120,138 ****
--- 125,150 ----
  
      throw v0
  
+     .line 1574
      :cond_0
      iput-object p2, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mAccountType:Ljava/lang/String;
  
+     .line 1575
      iput-object p3, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mAuthTokenType:Ljava/lang/String;
  
+     .line 1576
      iput-object p4, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mFeatures:[Ljava/lang/String;
  
+     .line 1577
      iput-object p6, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mAddAccountOptions:Landroid/os/Bundle;
  
+     .line 1578
      iput-object p7, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mLoginOptions:Landroid/os/Bundle;
  
+     .line 1579
      iput-object p0, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mMyCallback:Landroid/accounts/AccountManagerCallback;
  
+     .line 1580
      return-void
  .end method
  
***************
*** 142,147 ****
--- 154,160 ----
      .parameter "x1"
  
      .prologue
+     .line 1566
      invoke-virtual {p0, p1}, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->setException(Ljava/lang/Throwable;)V
  
      return-void
***************
*** 153,158 ****
--- 166,172 ----
      .parameter "x1"
  
      .prologue
+     .line 1566
      invoke-virtual {p0, p1}, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->setException(Ljava/lang/Throwable;)V
  
      return-void
***************
*** 164,169 ****
--- 178,184 ----
      .parameter "x1"
  
      .prologue
+     .line 1566
      iput p1, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mNumAccounts:I
  
      return p1
***************
*** 175,180 ****
--- 190,196 ----
      .parameter "x1"
  
      .prologue
+     .line 1566
      invoke-virtual {p0, p1}, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->setException(Ljava/lang/Throwable;)V
  
      return-void
***************
*** 191,196 ****
--- 207,213 ----
      .end annotation
  
      .prologue
+     .line 1591
      iget-object v0, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->this$0:Landroid/accounts/AccountManager;
  
      iget-object v1, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mAccountType:Ljava/lang/String;
***************
*** 205,210 ****
--- 222,228 ----
  
      invoke-virtual {v0, v1, v2, v3, v4}, Landroid/accounts/AccountManager;->getAccountsByTypeAndFeatures(Ljava/lang/String;[Ljava/lang/String;Landroid/accounts/AccountManagerCallback;Landroid/os/Handler;)Landroid/accounts/AccountManagerFuture;
  
+     .line 1678
      return-void
  .end method
  
***************
*** 225,230 ****
--- 243,249 ----
      .local p1, future:Landroid/accounts/AccountManagerFuture;,"Landroid/accounts/AccountManagerFuture<Landroid/os/Bundle;>;"
      const/4 v11, 0x1
  
+     .line 1682
      :try_start_0
      invoke-interface {p1}, Landroid/accounts/AccountManagerFuture;->getResult()Ljava/lang/Object;
  
***************
*** 232,248 ****
--- 251,270 ----
  
      check-cast v10, Landroid/os/Bundle;
  
+     .line 1683
      .local v10, result:Landroid/os/Bundle;
      iget v0, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mNumAccounts:I
  
      if-nez v0, :cond_2
  
+     .line 1684
      const-string v0, "authAccount"
  
      invoke-virtual {v10, v0}, Landroid/os/Bundle;->getString(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v7
  
+     .line 1685
      .local v7, accountName:Ljava/lang/String;
      const-string v0, "accountType"
  
***************
*** 250,255 ****
--- 272,278 ----
  
      move-result-object v8
  
+     .line 1686
      .local v8, accountType:Ljava/lang/String;
      invoke-static {v7}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
***************
*** 263,268 ****
--- 286,292 ----
  
      if-eqz v0, :cond_1
  
+     .line 1687
      :cond_0
      new-instance v0, Landroid/accounts/AuthenticatorException;
  
***************
*** 272,283 ****
--- 296,309 ----
  
      invoke-virtual {p0, v0}, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->setException(Ljava/lang/Throwable;)V
  
+     .line 1704
      .end local v7           #accountName:Ljava/lang/String;
      .end local v8           #accountType:Ljava/lang/String;
      .end local v10           #result:Landroid/os/Bundle;
      :goto_0
      return-void
  
+     .line 1690
      .restart local v7       #accountName:Ljava/lang/String;
      .restart local v8       #accountType:Ljava/lang/String;
      .restart local v10       #result:Landroid/os/Bundle;
***************
*** 286,296 ****
--- 312,324 ----
  
      invoke-direct {v1, v7, v8}, Landroid/accounts/Account;-><init>(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1691
      .local v1, account:Landroid/accounts/Account;
      const/4 v0, 0x1
  
      iput v0, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mNumAccounts:I
  
+     .line 1692
      iget-object v0, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->this$0:Landroid/accounts/AccountManager;
  
      iget-object v2, p0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->mAuthTokenType:Ljava/lang/String;
***************
*** 311,316 ****
--- 339,345 ----
  
      goto :goto_0
  
+     .line 1697
      .end local v1           #account:Landroid/accounts/Account;
      .end local v7           #accountName:Ljava/lang/String;
      .end local v8           #accountType:Ljava/lang/String;
***************
*** 318,328 ****
--- 347,359 ----
      :catch_0
      move-exception v9
  
+     .line 1698
      .local v9, e:Landroid/accounts/OperationCanceledException;
      invoke-virtual {p0, v11}, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->cancel(Z)Z
  
      goto :goto_0
  
+     .line 1696
      .end local v9           #e:Landroid/accounts/OperationCanceledException;
      .restart local v10       #result:Landroid/os/Bundle;
      :cond_2
***************
*** 335,353 ****
--- 366,388 ----
  
      goto :goto_0
  
+     .line 1699
      .end local v10           #result:Landroid/os/Bundle;
      :catch_1
      move-exception v9
  
+     .line 1700
      .local v9, e:Ljava/io/IOException;
      invoke-virtual {p0, v9}, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->setException(Ljava/lang/Throwable;)V
  
      goto :goto_0
  
+     .line 1701
      .end local v9           #e:Ljava/io/IOException;
      :catch_2
      move-exception v9
  
+     .line 1702
      .local v9, e:Landroid/accounts/AuthenticatorException;
      invoke-virtual {p0, v9}, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->setException(Ljava/lang/Throwable;)V
  
