*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManagerService$5.smali	2012-03-23 19:36:19.959044938 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManagerService$5.smali.original	2012-03-23 19:34:45.058314425 -0500
***************
*** 34,39 ****
--- 34,40 ----
      .parameter
  
      .prologue
+     .line 1177
      iput-object p1, p0, Landroid/accounts/AccountManagerService$5;->this$0:Landroid/accounts/AccountManagerService;
  
      iput-object p6, p0, Landroid/accounts/AccountManagerService$5;->val$account:Landroid/accounts/Account;
***************
*** 56,61 ****
--- 57,63 ----
      .end annotation
  
      .prologue
+     .line 1179
      iget-object v0, p0, Landroid/accounts/AccountManagerService$5;->mAuthenticator:Landroid/accounts/IAccountAuthenticator;
  
      iget-object v1, p0, Landroid/accounts/AccountManagerService$5;->val$account:Landroid/accounts/Account;
***************
*** 64,69 ****
--- 66,72 ----
  
      invoke-interface {v0, p0, v1, v2}, Landroid/accounts/IAccountAuthenticator;->confirmCredentials(Landroid/accounts/IAccountAuthenticatorResponse;Landroid/accounts/Account;Landroid/os/Bundle;)V
  
+     .line 1180
      return-void
  .end method
  
***************
*** 72,77 ****
--- 75,81 ----
      .parameter "now"
  
      .prologue
+     .line 1182
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
