*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager.smali	2012-03-23 19:36:20.459054055 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager.smali.original	2012-03-23 19:34:45.062314497 -0500
***************
*** 107,130 ****
--- 107,136 ----
      .parameter "service"
  
      .prologue
+     .line 242
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1835
      invoke-static {}, Lcom/google/android/collect/Maps;->newHashMap()Ljava/util/HashMap;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
+     .line 1843
      new-instance v0, Landroid/accounts/AccountManager$12;
  
      invoke-direct {v0, p0}, Landroid/accounts/AccountManager$12;-><init>(Landroid/accounts/AccountManager;)V
  
      iput-object v0, p0, Landroid/accounts/AccountManager;->mAccountsChangedBroadcastReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 243
      iput-object p1, p0, Landroid/accounts/AccountManager;->mContext:Landroid/content/Context;
  
+     .line 244
      iput-object p2, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
  
+     .line 245
      new-instance v0, Landroid/os/Handler;
  
      iget-object v1, p0, Landroid/accounts/AccountManager;->mContext:Landroid/content/Context;
***************
*** 137,142 ****
--- 143,149 ----
  
      iput-object v0, p0, Landroid/accounts/AccountManager;->mMainHandler:Landroid/os/Handler;
  
+     .line 246
      return-void
  .end method
  
***************
*** 147,172 ****
--- 154,186 ----
      .parameter "handler"
  
      .prologue
+     .line 251
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1835
      invoke-static {}, Lcom/google/android/collect/Maps;->newHashMap()Ljava/util/HashMap;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
+     .line 1843
      new-instance v0, Landroid/accounts/AccountManager$12;
  
      invoke-direct {v0, p0}, Landroid/accounts/AccountManager$12;-><init>(Landroid/accounts/AccountManager;)V
  
      iput-object v0, p0, Landroid/accounts/AccountManager;->mAccountsChangedBroadcastReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 252
      iput-object p1, p0, Landroid/accounts/AccountManager;->mContext:Landroid/content/Context;
  
+     .line 253
      iput-object p2, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
  
+     .line 254
      iput-object p3, p0, Landroid/accounts/AccountManager;->mMainHandler:Landroid/os/Handler;
  
+     .line 255
      return-void
  .end method
  
***************
*** 175,180 ****
--- 189,195 ----
      .parameter "x0"
  
      .prologue
+     .line 140
      iget-object v0, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
  
      return-object v0
***************
*** 185,190 ****
--- 200,206 ----
      .parameter "x0"
  
      .prologue
+     .line 140
      iget-object v0, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
      return-object v0
***************
*** 198,203 ****
--- 214,220 ----
      .parameter "x3"
  
      .prologue
+     .line 140
      invoke-direct {p0, p1, p2, p3}, Landroid/accounts/AccountManager;->postToHandler(Landroid/os/Handler;Landroid/accounts/OnAccountsUpdateListener;[Landroid/accounts/Account;)V
  
      return-void
***************
*** 208,213 ****
--- 225,231 ----
      .parameter "x0"
  
      .prologue
+     .line 140
      invoke-direct {p0}, Landroid/accounts/AccountManager;->ensureNotOnMainThread()V
  
      return-void
***************
*** 221,226 ****
--- 239,245 ----
      .parameter "x3"
  
      .prologue
+     .line 140
      invoke-direct {p0, p1, p2, p3}, Landroid/accounts/AccountManager;->postToHandler(Landroid/os/Handler;Landroid/accounts/AccountManagerCallback;Landroid/accounts/AccountManagerFuture;)V
  
      return-void
***************
*** 233,238 ****
--- 252,258 ----
      .parameter "x2"
  
      .prologue
+     .line 140
      invoke-direct {p0, p1, p2}, Landroid/accounts/AccountManager;->convertErrorToException(ILjava/lang/String;)Ljava/lang/Exception;
  
      move-result-object v0
***************
*** 245,250 ****
--- 265,271 ----
      .parameter "x0"
  
      .prologue
+     .line 140
      iget-object v0, p0, Landroid/accounts/AccountManager;->mMainHandler:Landroid/os/Handler;
  
      return-object v0
***************
*** 256,305 ****
--- 277,336 ----
      .parameter "message"
  
      .prologue
+     .line 1547
      const/4 v0, 0x3
  
      if-ne p1, v0, :cond_0
  
+     .line 1548
      new-instance v0, Ljava/io/IOException;
  
      invoke-direct {v0, p2}, Ljava/io/IOException;-><init>(Ljava/lang/String;)V
  
+     .line 1563
      :goto_0
      return-object v0
  
+     .line 1551
      :cond_0
      const/4 v0, 0x6
  
      if-ne p1, v0, :cond_1
  
+     .line 1552
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0, p2}, Ljava/lang/UnsupportedOperationException;-><init>(Ljava/lang/String;)V
  
      goto :goto_0
  
+     .line 1555
      :cond_1
      const/4 v0, 0x5
  
      if-ne p1, v0, :cond_2
  
+     .line 1556
      new-instance v0, Landroid/accounts/AuthenticatorException;
  
      invoke-direct {v0, p2}, Landroid/accounts/AuthenticatorException;-><init>(Ljava/lang/String;)V
  
      goto :goto_0
  
+     .line 1559
      :cond_2
      const/4 v0, 0x7
  
      if-ne p1, v0, :cond_3
  
+     .line 1560
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      invoke-direct {v0, p2}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      goto :goto_0
  
+     .line 1563
      :cond_3
      new-instance v0, Landroid/accounts/AuthenticatorException;
  
***************
*** 312,321 ****
--- 343,354 ----
      .locals 4
  
      .prologue
+     .line 1247
      invoke-static {}, Landroid/os/Looper;->myLooper()Landroid/os/Looper;
  
      move-result-object v1
  
+     .line 1248
      .local v1, looper:Landroid/os/Looper;
      if-eqz v1, :cond_0
  
***************
*** 327,338 ****
--- 360,373 ----
  
      if-ne v1, v2, :cond_0
  
+     .line 1249
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v2, "calling this from your main thread can lead to deadlock"
  
      invoke-direct {v0, v2}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
+     .line 1251
      .local v0, exception:Ljava/lang/IllegalStateException;
      const-string v2, "AccountManager"
  
***************
*** 340,345 ****
--- 375,381 ----
  
      invoke-static {v2, v3, v0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 1253
      iget-object v2, p0, Landroid/accounts/AccountManager;->mContext:Landroid/content/Context;
  
      invoke-virtual {v2}, Landroid/content/Context;->getApplicationInfo()Landroid/content/pm/ApplicationInfo;
***************
*** 352,359 ****
--- 388,397 ----
  
      if-lt v2, v3, :cond_0
  
+     .line 1254
      throw v0
  
+     .line 1257
      .end local v0           #exception:Ljava/lang/IllegalStateException;
      :cond_0
      return-void
***************
*** 364,369 ****
--- 402,408 ----
      .parameter "context"
  
      .prologue
+     .line 287
      if-nez p0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
***************
*** 374,379 ****
--- 413,419 ----
  
      throw v0
  
+     .line 288
      :cond_0
      const-string v0, "account"
  
***************
*** 417,427 ****
--- 457,469 ----
      .end annotation
  
      .prologue
+     .line 1814
      .local p1, allowableAccounts:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/accounts/Account;>;"
      new-instance v0, Landroid/content/Intent;
  
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
+     .line 1815
      .local v0, intent:Landroid/content/Intent;
      const-string v1, "android"
  
***************
*** 429,466 ****
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
      const-string v1, "allowableAccounts"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
      const-string v1, "allowableAccountTypes"
  
      invoke-virtual {v0, v1, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;[Ljava/lang/String;)Landroid/content/Intent;
  
      const-string v1, "addAccountOptions"
  
      invoke-virtual {v0, v1, p7}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Bundle;)Landroid/content/Intent;
  
!     const-string v1, "selectedAccount"
  
      invoke-virtual {v0, v1, p0}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
      const-string v1, "alwaysPromptForAccount"
  
      invoke-virtual {v0, v1, p3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
      const-string v1, "descriptionTextOverride"
  
      invoke-virtual {v0, v1, p4}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
      const-string v1, "authTokenType"
  
      invoke-virtual {v0, v1, p5}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
      const-string v1, "addAccountRequiredFeatures"
  
      invoke-virtual {v0, v1, p6}, Landroid/content/Intent;->putExtra(Ljava/lang/String;[Ljava/lang/String;)Landroid/content/Intent;
  
      return-object v0
  .end method
  
--- 471,517 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 1816
      const-string v1, "allowableAccounts"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
+     .line 1818
      const-string v1, "allowableAccountTypes"
  
      invoke-virtual {v0, v1, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;[Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 1820
      const-string v1, "addAccountOptions"
  
      invoke-virtual {v0, v1, p7}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Bundle;)Landroid/content/Intent;
  
!     .line 1822
!     const-string/jumbo v1, "selectedAccount"
  
      invoke-virtual {v0, v1, p0}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 1823
      const-string v1, "alwaysPromptForAccount"
  
      invoke-virtual {v0, v1, p3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     .line 1825
      const-string v1, "descriptionTextOverride"
  
      invoke-virtual {v0, v1, p4}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 1827
      const-string v1, "authTokenType"
  
      invoke-virtual {v0, v1, p5}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 1829
      const-string v1, "addAccountRequiredFeatures"
  
      invoke-virtual {v0, v1, p6}, Landroid/content/Intent;->putExtra(Ljava/lang/String;[Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 1832
      return-object v0
  .end method
  
***************
*** 485,496 ****
--- 536,549 ----
      .end annotation
  
      .prologue
+     .line 1261
      .local p2, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      .local p3, future:Landroid/accounts/AccountManagerFuture;,"Landroid/accounts/AccountManagerFuture<Landroid/os/Bundle;>;"
      if-nez p1, :cond_0
  
      iget-object p1, p0, Landroid/accounts/AccountManager;->mMainHandler:Landroid/os/Handler;
  
+     .line 1262
      :cond_0
      new-instance v0, Landroid/accounts/AccountManager$10;
  
***************
*** 498,503 ****
--- 551,557 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Handler;->post(Ljava/lang/Runnable;)Z
  
+     .line 1267
      return-void
  .end method
  
***************
*** 510,528 ****
--- 564,586 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1271
      array-length v1, p3
  
      new-array v0, v1, [Landroid/accounts/Account;
  
+     .line 1274
      .local v0, accountsCopy:[Landroid/accounts/Account;
      array-length v1, v0
  
      invoke-static {p3, v2, v0, v2, v1}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 1275
      if-nez p1, :cond_0
  
      iget-object p1, p0, Landroid/accounts/AccountManager;->mMainHandler:Landroid/os/Handler;
  
+     .line 1276
      :cond_0
      new-instance v1, Landroid/accounts/AccountManager$11;
  
***************
*** 530,535 ****
--- 588,594 ----
  
      invoke-virtual {p1, v1}, Landroid/os/Handler;->post(Ljava/lang/Runnable;)Z
  
+     .line 1287
      return-void
  .end method
  
***************
*** 538,545 ****
--- 597,606 ----
      .parameter "result"
  
      .prologue
+     .line 261
      if-eqz p0, :cond_0
  
+     .line 262
      const-string v1, "authtoken"
  
      invoke-virtual {p0, v1}, Landroid/os/Bundle;->containsKey(Ljava/lang/String;)Z
***************
*** 560,569 ****
--- 621,632 ----
  
      if-nez v1, :cond_0
  
+     .line 264
      new-instance v0, Landroid/os/Bundle;
  
      invoke-direct {v0, p0}, Landroid/os/Bundle;-><init>(Landroid/os/Bundle;)V
  
+     .line 265
      .local v0, newResult:Landroid/os/Bundle;
      const-string v1, "authtoken"
  
***************
*** 571,576 ****
--- 634,640 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 269
      .end local v0           #newResult:Landroid/os/Bundle;
      :goto_0
      return-object v0
***************
*** 615,620 ****
--- 679,685 ----
      .end annotation
  
      .prologue
+     .line 1055
      .local p6, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      if-nez p1, :cond_0
  
***************
*** 626,641 ****
--- 691,710 ----
  
      throw v0
  
+     .line 1056
      :cond_0
      new-instance v9, Landroid/os/Bundle;
  
      invoke-direct {v9}, Landroid/os/Bundle;-><init>()V
  
+     .line 1057
      .local v9, optionsIn:Landroid/os/Bundle;
      if-eqz p4, :cond_1
  
+     .line 1058
      invoke-virtual {v9, p4}, Landroid/os/Bundle;->putAll(Landroid/os/Bundle;)V
  
+     .line 1060
      :cond_1
      const-string v0, "androidPackageName"
  
***************
*** 647,652 ****
--- 716,722 ----
  
      invoke-virtual {v9, v0, v1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1062
      new-instance v0, Landroid/accounts/AccountManager$6;
  
      move-object v1, p0
***************
*** 681,686 ****
--- 751,757 ----
      .parameter "userdata"
  
      .prologue
+     .line 514
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
***************
*** 691,696 ****
--- 762,768 ----
  
      throw v1
  
+     .line 516
      :cond_0
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 703,711 ****
--- 775,785 ----
  
      return v1
  
+     .line 517
      :catch_0
      move-exception v0
  
+     .line 519
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 721,741 ****
      .parameter "updateImmediately"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "the listener is null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      iget-object v3, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
      monitor-enter v3
  
      :try_start_0
      iget-object v2, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
--- 795,819 ----
      .parameter "updateImmediately"
  
      .prologue
+     .line 1882
      if-nez p1, :cond_0
  
+     .line 1883
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "the listener is null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1885
      :cond_0
      iget-object v3, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
      monitor-enter v3
  
+     .line 1886
      :try_start_0
      iget-object v2, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
***************
*** 745,758 ****
  
      if-eqz v2, :cond_1
  
      new-instance v2, Ljava/lang/IllegalStateException;
  
!     const-string v4, "this listener is already added"
  
      invoke-direct {v2, v4}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :catchall_0
      move-exception v2
  
--- 823,838 ----
  
      if-eqz v2, :cond_1
  
+     .line 1887
      new-instance v2, Ljava/lang/IllegalStateException;
  
!     const-string/jumbo v4, "this listener is already added"
  
      invoke-direct {v2, v4}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1901
      :catchall_0
      move-exception v2
  
***************
*** 762,767 ****
--- 842,848 ----
  
      throw v2
  
+     .line 1889
      :cond_1
      :try_start_1
      iget-object v2, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
***************
*** 770,815 ****
--- 851,906 ----
  
      move-result v1
  
+     .line 1891
      .local v1, wasEmpty:Z
      iget-object v2, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
      invoke-virtual {v2, p1, p2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 1893
      if-eqz v1, :cond_2
  
+     .line 1895
      new-instance v0, Landroid/content/IntentFilter;
  
      invoke-direct {v0}, Landroid/content/IntentFilter;-><init>()V
  
+     .line 1896
      .local v0, intentFilter:Landroid/content/IntentFilter;
      const-string v2, "android.accounts.LOGIN_ACCOUNTS_CHANGED"
  
      invoke-virtual {v0, v2}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 1898
      const-string v2, "android.intent.action.DEVICE_STORAGE_OK"
  
      invoke-virtual {v0, v2}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 1899
      iget-object v2, p0, Landroid/accounts/AccountManager;->mContext:Landroid/content/Context;
  
      iget-object v4, p0, Landroid/accounts/AccountManager;->mAccountsChangedBroadcastReceiver:Landroid/content/BroadcastReceiver;
  
      invoke-virtual {v2, v4, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 1901
      .end local v0           #intentFilter:Landroid/content/IntentFilter;
      :cond_2
      monitor-exit v3
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 1903
      if-eqz p3, :cond_3
  
+     .line 1904
      invoke-virtual {p0}, Landroid/accounts/AccountManager;->getAccounts()[Landroid/accounts/Account;
  
      move-result-object v2
  
      invoke-direct {p0, p2, p1, v2}, Landroid/accounts/AccountManager;->postToHandler(Landroid/os/Handler;Landroid/accounts/OnAccountsUpdateListener;[Landroid/accounts/Account;)V
  
+     .line 1906
      :cond_3
      return-void
  .end method
***************
*** 830,835 ****
--- 921,927 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 744
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
***************
*** 840,845 ****
--- 932,938 ----
  
      throw v0
  
+     .line 745
      :cond_0
      if-nez p2, :cond_1
  
***************
*** 862,867 ****
--- 955,961 ----
  
      move-object v5, v4
  
+     .line 746
      invoke-virtual/range {v0 .. v5}, Landroid/accounts/AccountManager;->getAuthToken(Landroid/accounts/Account;Ljava/lang/String;ZLandroid/accounts/AccountManagerCallback;Landroid/os/Handler;)Landroid/accounts/AccountManagerFuture;
  
      move-result-object v0
***************
*** 872,880 ****
--- 966,976 ----
  
      check-cast v6, Landroid/os/Bundle;
  
+     .line 748
      .local v6, bundle:Landroid/os/Bundle;
      if-nez v6, :cond_2
  
+     .line 753
      const-string v0, "AccountManager"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 907,912 ****
--- 1003,1009 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 757
      :goto_0
      return-object v4
  
***************
*** 925,930 ****
--- 1022,1028 ----
      .parameter "account"
  
      .prologue
+     .line 656
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
***************
*** 935,940 ****
--- 1033,1039 ----
  
      throw v1
  
+     .line 658
      :cond_0
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 943,953 ****
--- 1042,1055 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 663
      return-void
  
+     .line 659
      :catch_0
      move-exception v0
  
+     .line 661
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 983,988 ****
--- 1085,1091 ----
      .end annotation
  
      .prologue
+     .line 1129
      .local p4, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      if-nez p1, :cond_0
  
***************
*** 994,999 ****
--- 1097,1103 ----
  
      throw v0
  
+     .line 1130
      :cond_0
      new-instance v0, Landroid/accounts/AccountManager$7;
  
***************
*** 1045,1050 ****
--- 1149,1155 ----
      .end annotation
  
      .prologue
+     .line 1238
      .local p3, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      if-nez p1, :cond_0
  
***************
*** 1056,1061 ****
--- 1161,1167 ----
  
      throw v0
  
+     .line 1239
      :cond_0
      new-instance v0, Landroid/accounts/AccountManager$9;
  
***************
*** 1084,1089 ****
--- 1190,1196 ----
      .locals 3
  
      .prologue
+     .line 375
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
  
***************
*** 1097,1105 ****
--- 1204,1214 ----
  
      return-object v1
  
+     .line 376
      :catch_0
      move-exception v0
  
+     .line 378
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 1113,1118 ****
--- 1222,1228 ----
      .parameter "type"
  
      .prologue
+     .line 400
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
  
***************
*** 1124,1132 ****
--- 1234,1244 ----
  
      return-object v1
  
+     .line 401
      :catch_0
      move-exception v0
  
+     .line 403
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 1161,1177 ****
      .end annotation
  
      .prologue
      .local p3, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<[Landroid/accounts/Account;>;"
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "type is null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      new-instance v0, Landroid/accounts/AccountManager$2;
  
--- 1273,1291 ----
      .end annotation
  
      .prologue
+     .line 478
      .local p3, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<[Landroid/accounts/Account;>;"
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "type is null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 479
      :cond_0
      new-instance v0, Landroid/accounts/AccountManager$2;
  
***************
*** 1223,1228 ****
--- 1337,1343 ----
      .end annotation
  
      .prologue
+     .line 818
      .local p5, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      if-nez p1, :cond_0
  
***************
*** 1234,1239 ****
--- 1349,1355 ----
  
      throw v0
  
+     .line 819
      :cond_0
      if-nez p2, :cond_1
  
***************
*** 1245,1260 ****
--- 1361,1380 ----
  
      throw v0
  
+     .line 820
      :cond_1
      new-instance v7, Landroid/os/Bundle;
  
      invoke-direct {v7}, Landroid/os/Bundle;-><init>()V
  
+     .line 821
      .local v7, optionsIn:Landroid/os/Bundle;
      if-eqz p3, :cond_2
  
+     .line 822
      invoke-virtual {v7, p3}, Landroid/os/Bundle;->putAll(Landroid/os/Bundle;)V
  
+     .line 824
      :cond_2
      const-string v0, "androidPackageName"
  
***************
*** 1266,1271 ****
--- 1386,1392 ----
  
      invoke-virtual {v7, v0, v1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 825
      new-instance v0, Landroid/accounts/AccountManager$4;
  
      move-object v1, p0
***************
*** 1318,1323 ****
--- 1439,1445 ----
      .end annotation
  
      .prologue
+     .line 988
      .local p5, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      if-nez p1, :cond_0
  
***************
*** 1329,1334 ****
--- 1451,1457 ----
  
      throw v0
  
+     .line 989
      :cond_0
      if-nez p2, :cond_1
  
***************
*** 1340,1355 ****
--- 1463,1482 ----
  
      throw v0
  
+     .line 990
      :cond_1
      new-instance v8, Landroid/os/Bundle;
  
      invoke-direct {v8}, Landroid/os/Bundle;-><init>()V
  
+     .line 991
      .local v8, optionsIn:Landroid/os/Bundle;
      if-eqz p3, :cond_2
  
+     .line 992
      invoke-virtual {v8, p3}, Landroid/os/Bundle;->putAll(Landroid/os/Bundle;)V
  
+     .line 994
      :cond_2
      const-string v0, "androidPackageName"
  
***************
*** 1361,1366 ****
--- 1488,1494 ----
  
      invoke-virtual {v8, v0, v1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 995
      new-instance v0, Landroid/accounts/AccountManager$5;
  
      const/4 v2, 0x0
***************
*** 1416,1421 ****
--- 1544,1550 ----
      .end annotation
  
      .prologue
+     .line 909
      .local p4, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      const/4 v3, 0x0
  
***************
*** 1472,1477 ****
--- 1601,1607 ----
      .end annotation
  
      .prologue
+     .line 1765
      .local p7, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      if-nez p1, :cond_0
  
***************
*** 1483,1488 ****
--- 1613,1619 ----
  
      throw v1
  
+     .line 1766
      :cond_0
      if-nez p2, :cond_1
  
***************
*** 1494,1499 ****
--- 1625,1631 ----
  
      throw v1
  
+     .line 1767
      :cond_1
      new-instance v0, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;
  
***************
*** 1517,1525 ****
--- 1649,1659 ----
  
      invoke-direct/range {v0 .. v9}, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;-><init>(Landroid/accounts/AccountManager;Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Landroid/app/Activity;Landroid/os/Bundle;Landroid/os/Bundle;Landroid/accounts/AccountManagerCallback;Landroid/os/Handler;)V
  
+     .line 1770
      .local v0, task:Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;
      invoke-virtual {v0}, Landroid/accounts/AccountManager$GetAuthTokenByTypeAndFeaturesTask;->start()Landroid/accounts/AccountManagerFuture;
  
+     .line 1771
      return-object v0
  .end method
  
***************
*** 1527,1532 ****
--- 1661,1667 ----
      .locals 2
  
      .prologue
+     .line 354
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
  
***************
*** 1538,1546 ****
--- 1673,1683 ----
  
      return-object v1
  
+     .line 355
      :catch_0
      move-exception v0
  
+     .line 357
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 1554,1559 ****
--- 1691,1697 ----
      .parameter "account"
  
      .prologue
+     .line 306
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
***************
*** 1564,1569 ****
--- 1702,1708 ----
  
      throw v1
  
+     .line 308
      :cond_0
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 1576,1584 ****
--- 1715,1725 ----
  
      return-object v1
  
+     .line 309
      :catch_0
      move-exception v0
  
+     .line 311
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 1593,1598 ****
--- 1734,1740 ----
      .parameter "key"
  
      .prologue
+     .line 331
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
***************
*** 1603,1608 ****
--- 1745,1751 ----
  
      throw v1
  
+     .line 332
      :cond_0
      if-nez p2, :cond_1
  
***************
*** 1614,1619 ****
--- 1757,1763 ----
  
      throw v1
  
+     .line 334
      :cond_1
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 1626,1634 ****
--- 1770,1780 ----
  
      return-object v1
  
+     .line 335
      :catch_0
      move-exception v0
  
+     .line 337
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 1663,1668 ****
--- 1809,1815 ----
      .end annotation
  
      .prologue
+     .line 433
      .local p3, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Ljava/lang/Boolean;>;"
      if-nez p1, :cond_0
  
***************
*** 1674,1679 ****
--- 1821,1827 ----
  
      throw v0
  
+     .line 434
      :cond_0
      if-nez p2, :cond_1
  
***************
*** 1685,1690 ****
--- 1833,1839 ----
  
      throw v0
  
+     .line 435
      :cond_1
      new-instance v0, Landroid/accounts/AccountManager$1;
  
***************
*** 1713,1718 ****
--- 1862,1868 ----
      .parameter "authToken"
  
      .prologue
+     .line 577
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
***************
*** 1723,1731 ****
--- 1873,1883 ----
  
      throw v1
  
+     .line 579
      :cond_0
      if-eqz p2, :cond_1
  
+     .line 580
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
  
***************
*** 1733,1744 ****
--- 1885,1899 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 586
      :cond_1
      return-void
  
+     .line 582
      :catch_0
      move-exception v0
  
+     .line 584
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 1753,1758 ****
--- 1908,1914 ----
      .parameter "authTokenType"
  
      .prologue
+     .line 606
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
***************
*** 1763,1768 ****
--- 1919,1925 ----
  
      throw v1
  
+     .line 607
      :cond_0
      if-nez p2, :cond_1
  
***************
*** 1774,1779 ****
--- 1931,1937 ----
  
      throw v1
  
+     .line 609
      :cond_1
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 1786,1794 ****
--- 1944,1954 ----
  
      return-object v1
  
+     .line 610
      :catch_0
      move-exception v0
  
+     .line 612
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 1820,1825 ****
--- 1980,1986 ----
      .end annotation
  
      .prologue
+     .line 546
      .local p2, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Ljava/lang/Boolean;>;"
      if-nez p1, :cond_0
  
***************
*** 1831,1836 ****
--- 1992,1998 ----
  
      throw v0
  
+     .line 547
      :cond_0
      new-instance v0, Landroid/accounts/AccountManager$3;
  
***************
*** 1848,1853 ****
--- 2010,2016 ----
      .parameter "listener"
  
      .prologue
+     .line 1922
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
***************
*** 1858,1868 ****
--- 2021,2033 ----
  
      throw v0
  
+     .line 1923
      :cond_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
      monitor-enter v1
  
+     .line 1924
      :try_start_0
      iget-object v0, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
***************
*** 1872,1893 ****
--- 2037,2063 ----
  
      if-nez v0, :cond_1
  
+     .line 1925
      const-string v0, "AccountManager"
  
      const-string v2, "Listener was not previously added"
  
      invoke-static {v0, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1926
      monitor-exit v1
  
+     .line 1933
      :goto_0
      return-void
  
+     .line 1928
      :cond_1
      iget-object v0, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
      invoke-virtual {v0, p1}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 1929
      iget-object v0, p0, Landroid/accounts/AccountManager;->mAccountsUpdatedListeners:Ljava/util/HashMap;
  
      invoke-virtual {v0}, Ljava/util/HashMap;->isEmpty()Z
***************
*** 1896,1907 ****
--- 2066,2079 ----
  
      if-eqz v0, :cond_2
  
+     .line 1930
      iget-object v0, p0, Landroid/accounts/AccountManager;->mContext:Landroid/content/Context;
  
      iget-object v2, p0, Landroid/accounts/AccountManager;->mAccountsChangedBroadcastReceiver:Landroid/content/BroadcastReceiver;
  
      invoke-virtual {v0, v2}, Landroid/content/Context;->unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  
+     .line 1932
      :cond_2
      monitor-exit v1
  
***************
*** 1924,1929 ****
--- 2096,2102 ----
      .parameter "authToken"
  
      .prologue
+     .line 708
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
***************
*** 1934,1939 ****
--- 2107,2113 ----
  
      throw v1
  
+     .line 709
      :cond_0
      if-nez p2, :cond_1
  
***************
*** 1945,1950 ****
--- 2119,2125 ----
  
      throw v1
  
+     .line 711
      :cond_1
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 1953,1963 ****
--- 2128,2141 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 716
      return-void
  
+     .line 712
      :catch_0
      move-exception v0
  
+     .line 714
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 1972,1977 ****
--- 2150,2156 ----
      .parameter "password"
  
      .prologue
+     .line 632
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
***************
*** 1982,1987 ****
--- 2161,2167 ----
  
      throw v1
  
+     .line 634
      :cond_0
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 1990,2000 ****
--- 2170,2183 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 639
      return-void
  
+     .line 635
      :catch_0
      move-exception v0
  
+     .line 637
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 2010,2015 ****
--- 2193,2199 ----
      .parameter "value"
  
      .prologue
+     .line 681
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
***************
*** 2020,2025 ****
--- 2204,2210 ----
  
      throw v1
  
+     .line 682
      :cond_0
      if-nez p2, :cond_1
  
***************
*** 2031,2036 ****
--- 2216,2222 ----
  
      throw v1
  
+     .line 684
      :cond_1
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 2039,2049 ****
--- 2225,2238 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 689
      return-void
  
+     .line 685
      :catch_0
      move-exception v0
  
+     .line 687
      .local v0, e:Landroid/os/RemoteException;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 2081,2086 ****
--- 2270,2276 ----
      .end annotation
  
      .prologue
+     .line 1189
      .local p5, callback:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      if-nez p1, :cond_0
  
***************
*** 2092,2097 ****
--- 2282,2288 ----
  
      throw v0
  
+     .line 1190
      :cond_0
      new-instance v0, Landroid/accounts/AccountManager$8;
  
