*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManagerService$4.smali	2012-03-23 19:36:21.496072965 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManagerService$4.smali.original	2012-03-23 19:34:45.070314643 -0500
***************
*** 40,45 ****
--- 40,46 ----
      .parameter
  
      .prologue
+     .line 1142
      iput-object p1, p0, Landroid/accounts/AccountManagerService$4;->this$0:Landroid/accounts/AccountManagerService;
  
      iput-object p6, p0, Landroid/accounts/AccountManagerService$4;->val$authTokenType:Ljava/lang/String;
***************
*** 66,71 ****
--- 67,73 ----
      .end annotation
  
      .prologue
+     .line 1144
      iget-object v0, p0, Landroid/accounts/AccountManagerService$4;->mAuthenticator:Landroid/accounts/IAccountAuthenticator;
  
      iget-object v2, p0, Landroid/accounts/AccountManagerService$4;->mAccountType:Ljava/lang/String;
***************
*** 80,85 ****
--- 82,88 ----
  
      invoke-interface/range {v0 .. v5}, Landroid/accounts/IAccountAuthenticator;->addAccount(Landroid/accounts/IAccountAuthenticatorResponse;Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Landroid/os/Bundle;)V
  
+     .line 1146
      return-void
  .end method
  
***************
*** 88,93 ****
--- 91,97 ----
      .parameter "now"
  
      .prologue
+     .line 1149
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
