*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$2.smali	2012-03-23 19:36:19.882043533 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$2.smali.original	2012-03-23 19:34:45.057314406 -0500
***************
*** 41,46 ****
--- 41,47 ----
      .parameter
  
      .prologue
+     .line 479
      .local p3, x1:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<[Landroid/accounts/Account;>;"
      iput-object p1, p0, Landroid/accounts/AccountManager$2;->this$0:Landroid/accounts/AccountManager;
  
***************
*** 65,70 ****
--- 66,72 ----
      .end annotation
  
      .prologue
+     .line 479
      invoke-virtual {p0, p1}, Landroid/accounts/AccountManager$2;->bundleToResult(Landroid/os/Bundle;)[Landroid/accounts/Account;
  
      move-result-object v0
***************
*** 82,87 ****
--- 84,90 ----
      .end annotation
  
      .prologue
+     .line 484
      const-string v3, "accounts"
  
      invoke-virtual {p1, v3}, Landroid/os/Bundle;->containsKey(Ljava/lang/String;)Z
***************
*** 90,103 ****
  
      if-nez v3, :cond_0
  
      new-instance v3, Landroid/accounts/AuthenticatorException;
  
!     const-string v4, "no result in response"
  
      invoke-direct {v3, v4}, Landroid/accounts/AuthenticatorException;-><init>(Ljava/lang/String;)V
  
      throw v3
  
      :cond_0
      const-string v3, "accounts"
  
--- 93,108 ----
  
      if-nez v3, :cond_0
  
+     .line 485
      new-instance v3, Landroid/accounts/AuthenticatorException;
  
!     const-string/jumbo v4, "no result in response"
  
      invoke-direct {v3, v4}, Landroid/accounts/AuthenticatorException;-><init>(Ljava/lang/String;)V
  
      throw v3
  
+     .line 487
      :cond_0
      const-string v3, "accounts"
  
***************
*** 105,115 ****
--- 110,122 ----
  
      move-result-object v2
  
+     .line 488
      .local v2, parcelables:[Landroid/os/Parcelable;
      array-length v3, v2
  
      new-array v0, v3, [Landroid/accounts/Account;
  
+     .line 489
      .local v0, descs:[Landroid/accounts/Account;
      const/4 v1, 0x0
  
***************
*** 119,134 ****
--- 126,144 ----
  
      if-ge v1, v3, :cond_1
  
+     .line 490
      aget-object v3, v2, v1
  
      check-cast v3, Landroid/accounts/Account;
  
      aput-object v3, v0, v1
  
+     .line 489
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 492
      :cond_1
      return-object v0
  .end method
***************
*** 142,147 ****
--- 152,158 ----
      .end annotation
  
      .prologue
+     .line 481
      iget-object v0, p0, Landroid/accounts/AccountManager$2;->this$0:Landroid/accounts/AccountManager;
  
      #getter for: Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 157,161 ****
--- 168,173 ----
  
      invoke-interface {v0, v1, v2, v3}, Landroid/accounts/IAccountManager;->getAccountsByFeatures(Landroid/accounts/IAccountManagerResponse;Ljava/lang/String;[Ljava/lang/String;)V
  
+     .line 482
      return-void
  .end method
