*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$3.smali	2012-03-23 19:36:20.826060746 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$3.smali.original	2012-03-23 19:34:45.065314552 -0500
***************
*** 38,43 ****
--- 38,44 ----
      .parameter
  
      .prologue
+     .line 547
      .local p3, x1:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Ljava/lang/Boolean;>;"
      iput-object p1, p0, Landroid/accounts/AccountManager$3;->this$0:Landroid/accounts/AccountManager;
  
***************
*** 60,65 ****
--- 61,67 ----
      .end annotation
  
      .prologue
+     .line 552
      const-string v0, "booleanResult"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->containsKey(Ljava/lang/String;)Z
***************
*** 68,81 ****
  
      if-nez v0, :cond_0
  
      new-instance v0, Landroid/accounts/AuthenticatorException;
  
!     const-string v1, "no result in response"
  
      invoke-direct {v0, v1}, Landroid/accounts/AuthenticatorException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      const-string v0, "booleanResult"
  
--- 70,85 ----
  
      if-nez v0, :cond_0
  
+     .line 553
      new-instance v0, Landroid/accounts/AuthenticatorException;
  
!     const-string/jumbo v1, "no result in response"
  
      invoke-direct {v0, v1}, Landroid/accounts/AuthenticatorException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 555
      :cond_0
      const-string v0, "booleanResult"
  
***************
*** 100,105 ****
--- 104,110 ----
      .end annotation
  
      .prologue
+     .line 547
      invoke-virtual {p0, p1}, Landroid/accounts/AccountManager$3;->bundleToResult(Landroid/os/Bundle;)Ljava/lang/Boolean;
  
      move-result-object v0
***************
*** 116,121 ****
--- 121,127 ----
      .end annotation
  
      .prologue
+     .line 549
      iget-object v0, p0, Landroid/accounts/AccountManager$3;->this$0:Landroid/accounts/AccountManager;
  
      #getter for: Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 129,133 ****
--- 135,140 ----
  
      invoke-interface {v0, v1, v2}, Landroid/accounts/IAccountManager;->removeAccount(Landroid/accounts/IAccountManagerResponse;Landroid/accounts/Account;)V
  
+     .line 550
      return-void
  .end method
