*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$BaseFutureTask$Response.smali	2012-03-23 19:36:20.718058777 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$BaseFutureTask$Response.smali.original	2012-03-23 19:34:45.064314533 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1446
      .local p0, this:Landroid/accounts/AccountManager$BaseFutureTask$Response;,"Landroid/accounts/AccountManager$BaseFutureTask<TT;>.Response;"
      iput-object p1, p0, Landroid/accounts/AccountManager$BaseFutureTask$Response;->this$1:Landroid/accounts/AccountManager$BaseFutureTask;
  
***************
*** 40,59 ****
--- 41,64 ----
      .parameter "message"
  
      .prologue
+     .line 1464
      .local p0, this:Landroid/accounts/AccountManager$BaseFutureTask$Response;,"Landroid/accounts/AccountManager$BaseFutureTask<TT;>.Response;"
      const/4 v0, 0x4
  
      if-ne p1, v0, :cond_0
  
+     .line 1465
      iget-object v0, p0, Landroid/accounts/AccountManager$BaseFutureTask$Response;->this$1:Landroid/accounts/AccountManager$BaseFutureTask;
  
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/accounts/AccountManager$BaseFutureTask;->cancel(Z)Z
  
+     .line 1469
      :goto_0
      return-void
  
+     .line 1468
      :cond_0
      iget-object v0, p0, Landroid/accounts/AccountManager$BaseFutureTask$Response;->this$1:Landroid/accounts/AccountManager$BaseFutureTask;
  
***************
*** 77,82 ****
--- 82,88 ----
      .parameter "bundle"
  
      .prologue
+     .line 1449
      .local p0, this:Landroid/accounts/AccountManager$BaseFutureTask$Response;,"Landroid/accounts/AccountManager$BaseFutureTask<TT;>.Response;"
      :try_start_0
      iget-object v1, p0, Landroid/accounts/AccountManager$BaseFutureTask$Response;->this$1:Landroid/accounts/AccountManager$BaseFutureTask;
***************
*** 85,97 ****
--- 91,106 ----
  
      move-result-object v0
  
+     .line 1450
      .local v0, result:Ljava/lang/Object;,"TT;"
      if-nez v0, :cond_0
  
+     .line 1461
      .end local v0           #result:Ljava/lang/Object;,"TT;"
      :goto_0
      return-void
  
+     .line 1453
      .restart local v0       #result:Ljava/lang/Object;,"TT;"
      :cond_0
      iget-object v1, p0, Landroid/accounts/AccountManager$BaseFutureTask$Response;->this$1:Landroid/accounts/AccountManager$BaseFutureTask;
***************
*** 104,122 ****
  
      goto :goto_0
  
      .end local v0           #result:Ljava/lang/Object;,"TT;"
      :catch_0
      move-exception v1
  
      :goto_1
      const/4 v1, 0x5
  
!     const-string v2, "no result in response"
  
      invoke-virtual {p0, v1, v2}, Landroid/accounts/AccountManager$BaseFutureTask$Response;->onError(ILjava/lang/String;)V
  
      goto :goto_0
  
      :catch_1
      move-exception v1
  
--- 113,134 ----
  
      goto :goto_0
  
+     .line 1455
      .end local v0           #result:Ljava/lang/Object;,"TT;"
      :catch_0
      move-exception v1
  
+     .line 1460
      :goto_1
      const/4 v1, 0x5
  
!     const-string/jumbo v2, "no result in response"
  
      invoke-virtual {p0, v1, v2}, Landroid/accounts/AccountManager$BaseFutureTask$Response;->onError(ILjava/lang/String;)V
  
      goto :goto_0
  
+     .line 1457
      :catch_1
      move-exception v1
  
