*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$8.smali	2012-03-23 19:36:20.693058321 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AccountManager$8.smali.original	2012-03-23 19:34:45.064314533 -0500
***************
*** 39,44 ****
--- 39,45 ----
      .parameter
  
      .prologue
+     .line 1190
      .local p4, x2:Landroid/accounts/AccountManagerCallback;,"Landroid/accounts/AccountManagerCallback<Landroid/os/Bundle;>;"
      iput-object p1, p0, Landroid/accounts/AccountManager$8;->this$0:Landroid/accounts/AccountManager;
  
***************
*** 66,71 ****
--- 67,73 ----
      .end annotation
  
      .prologue
+     .line 1192
      iget-object v0, p0, Landroid/accounts/AccountManager$8;->this$0:Landroid/accounts/AccountManager;
  
      #getter for: Landroid/accounts/AccountManager;->mService:Landroid/accounts/IAccountManager;
***************
*** 90,97 ****
--- 92,101 ----
  
      invoke-interface/range {v0 .. v5}, Landroid/accounts/IAccountManager;->updateCredentials(Landroid/accounts/IAccountManagerResponse;Landroid/accounts/Account;Ljava/lang/String;ZLandroid/os/Bundle;)V
  
+     .line 1194
      return-void
  
+     .line 1192
      :cond_0
      const/4 v4, 0x0
  
