*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AuthenticatorDescription.smali	2012-03-23 19:36:20.514055057 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/AuthenticatorDescription.smali.original	2012-03-23 19:34:45.063314515 -0500
***************
*** 40,45 ****
--- 40,46 ----
      .locals 1
  
      .prologue
+     .line 133
      new-instance v0, Landroid/accounts/AuthenticatorDescription$1;
  
      invoke-direct {v0}, Landroid/accounts/AuthenticatorDescription$1;-><init>()V
***************
*** 56,99 ****
--- 57,108 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 89
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 90
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
      iput-object v1, p0, Landroid/accounts/AuthenticatorDescription;->type:Ljava/lang/String;
  
+     .line 91
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
      iput-object v1, p0, Landroid/accounts/AuthenticatorDescription;->packageName:Ljava/lang/String;
  
+     .line 92
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
      iput v1, p0, Landroid/accounts/AuthenticatorDescription;->labelId:I
  
+     .line 93
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
      iput v1, p0, Landroid/accounts/AuthenticatorDescription;->iconId:I
  
+     .line 94
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
      iput v1, p0, Landroid/accounts/AuthenticatorDescription;->smallIconId:I
  
+     .line 95
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
      iput v1, p0, Landroid/accounts/AuthenticatorDescription;->accountPreferencesId:I
  
+     .line 96
      invoke-virtual {p1}, Landroid/os/Parcel;->readByte()B
  
      move-result v1
***************
*** 103,110 ****
--- 112,121 ----
      :goto_0
      iput-boolean v0, p0, Landroid/accounts/AuthenticatorDescription;->customTokens:Z
  
+     .line 97
      return-void
  
+     .line 96
      :cond_0
      const/4 v0, 0x0
  
***************
*** 117,122 ****
--- 128,134 ----
      .parameter "x1"
  
      .prologue
+     .line 25
      invoke-direct {p0, p1}, Landroid/accounts/AuthenticatorDescription;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 129,152 ****
--- 141,173 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 79
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 80
      iput-object p1, p0, Landroid/accounts/AuthenticatorDescription;->type:Ljava/lang/String;
  
+     .line 81
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/accounts/AuthenticatorDescription;->packageName:Ljava/lang/String;
  
+     .line 82
      iput v1, p0, Landroid/accounts/AuthenticatorDescription;->labelId:I
  
+     .line 83
      iput v1, p0, Landroid/accounts/AuthenticatorDescription;->iconId:I
  
+     .line 84
      iput v1, p0, Landroid/accounts/AuthenticatorDescription;->smallIconId:I
  
+     .line 85
      iput v1, p0, Landroid/accounts/AuthenticatorDescription;->accountPreferencesId:I
  
+     .line 86
      iput-boolean v1, p0, Landroid/accounts/AuthenticatorDescription;->customTokens:Z
  
+     .line 87
      return-void
  .end method
  
***************
*** 160,165 ****
--- 181,187 ----
      .parameter "prefId"
  
      .prologue
+     .line 66
      const/4 v7, 0x0
  
      move-object v0, p0
***************
*** 178,183 ****
--- 200,206 ----
  
      invoke-direct/range {v0 .. v7}, Landroid/accounts/AuthenticatorDescription;-><init>(Ljava/lang/String;Ljava/lang/String;IIIIZ)V
  
+     .line 67
      return-void
  .end method
  
***************
*** 192,235 ****
      .parameter "customTokens"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "type cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      if-nez p2, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "packageName cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_1
      iput-object p1, p0, Landroid/accounts/AuthenticatorDescription;->type:Ljava/lang/String;
  
      iput-object p2, p0, Landroid/accounts/AuthenticatorDescription;->packageName:Ljava/lang/String;
  
      iput p3, p0, Landroid/accounts/AuthenticatorDescription;->labelId:I
  
      iput p4, p0, Landroid/accounts/AuthenticatorDescription;->iconId:I
  
      iput p5, p0, Landroid/accounts/AuthenticatorDescription;->smallIconId:I
  
      iput p6, p0, Landroid/accounts/AuthenticatorDescription;->accountPreferencesId:I
  
      iput-boolean p7, p0, Landroid/accounts/AuthenticatorDescription;->customTokens:Z
  
      return-void
  .end method
  
--- 215,269 ----
      .parameter "customTokens"
  
      .prologue
+     .line 52
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 53
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "type cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 54
      :cond_0
      if-nez p2, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "packageName cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 55
      :cond_1
      iput-object p1, p0, Landroid/accounts/AuthenticatorDescription;->type:Ljava/lang/String;
  
+     .line 56
      iput-object p2, p0, Landroid/accounts/AuthenticatorDescription;->packageName:Ljava/lang/String;
  
+     .line 57
      iput p3, p0, Landroid/accounts/AuthenticatorDescription;->labelId:I
  
+     .line 58
      iput p4, p0, Landroid/accounts/AuthenticatorDescription;->iconId:I
  
+     .line 59
      iput p5, p0, Landroid/accounts/AuthenticatorDescription;->smallIconId:I
  
+     .line 60
      iput p6, p0, Landroid/accounts/AuthenticatorDescription;->accountPreferencesId:I
  
+     .line 61
      iput-boolean p7, p0, Landroid/accounts/AuthenticatorDescription;->customTokens:Z
  
+     .line 62
      return-void
  .end method
  
***************
*** 238,253 ****
      .parameter "type"
  
      .prologue
      if-nez p0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "type cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      new-instance v0, Landroid/accounts/AuthenticatorDescription;
  
--- 272,289 ----
      .parameter "type"
  
      .prologue
+     .line 75
      if-nez p0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "type cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 76
      :cond_0
      new-instance v0, Landroid/accounts/AuthenticatorDescription;
  
***************
*** 262,267 ****
--- 298,304 ----
      .locals 1
  
      .prologue
+     .line 101
      const/4 v0, 0x0
  
      return v0
***************
*** 272,284 ****
--- 309,324 ----
      .parameter "o"
  
      .prologue
+     .line 111
      if-ne p1, p0, :cond_0
  
      const/4 v1, 0x1
  
+     .line 114
      :goto_0
      return v1
  
+     .line 112
      :cond_0
      instance-of v1, p1, Landroid/accounts/AuthenticatorDescription;
  
***************
*** 291,298 ****
--- 331,340 ----
      :cond_1
      move-object v0, p1
  
+     .line 113
      check-cast v0, Landroid/accounts/AuthenticatorDescription;
  
+     .line 114
      .local v0, other:Landroid/accounts/AuthenticatorDescription;
      iget-object v1, p0, Landroid/accounts/AuthenticatorDescription;->type:Ljava/lang/String;
  
***************
*** 309,314 ****
--- 351,357 ----
      .locals 1
  
      .prologue
+     .line 106
      iget-object v0, p0, Landroid/accounts/AuthenticatorDescription;->type:Ljava/lang/String;
  
      invoke-virtual {v0}, Ljava/lang/String;->hashCode()I
***************
*** 322,327 ****
--- 365,371 ----
      .locals 2
  
      .prologue
+     .line 118
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 338,344 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 382,388 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 357,386 ****
--- 401,437 ----
      .parameter "flags"
  
      .prologue
+     .line 123
      iget-object v0, p0, Landroid/accounts/AuthenticatorDescription;->type:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 124
      iget-object v0, p0, Landroid/accounts/AuthenticatorDescription;->packageName:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 125
      iget v0, p0, Landroid/accounts/AuthenticatorDescription;->labelId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 126
      iget v0, p0, Landroid/accounts/AuthenticatorDescription;->iconId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 127
      iget v0, p0, Landroid/accounts/AuthenticatorDescription;->smallIconId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 128
      iget v0, p0, Landroid/accounts/AuthenticatorDescription;->accountPreferencesId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 129
      iget-boolean v0, p0, Landroid/accounts/AuthenticatorDescription;->customTokens:Z
  
      if-eqz v0, :cond_0
***************
*** 392,399 ****
--- 443,452 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeByte(B)V
  
+     .line 130
      return-void
  
+     .line 129
      :cond_0
      const/4 v0, 0x0
  
