*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/ChooseAccountTypeActivity$1.smali	2012-03-23 19:36:20.771059743 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/ChooseAccountTypeActivity$1.smali.original	2012-03-23 19:34:45.065314552 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 108
      iput-object p1, p0, Landroid/accounts/ChooseAccountTypeActivity$1;->this$0:Landroid/accounts/ChooseAccountTypeActivity;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 53,58 ****
--- 54,60 ----
      .end annotation
  
      .prologue
+     .line 110
      .local p1, parent:Landroid/widget/AdapterView;,"Landroid/widget/AdapterView<*>;"
      iget-object v1, p0, Landroid/accounts/ChooseAccountTypeActivity$1;->this$0:Landroid/accounts/ChooseAccountTypeActivity;
  
***************
*** 76,80 ****
--- 78,83 ----
      #calls: Landroid/accounts/ChooseAccountTypeActivity;->setResultAndFinish(Ljava/lang/String;)V
      invoke-static {v1, v0}, Landroid/accounts/ChooseAccountTypeActivity;->access$100(Landroid/accounts/ChooseAccountTypeActivity;Ljava/lang/String;)V
  
+     .line 111
      return-void
  .end method
