*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/ChooseAccountActivity$ViewHolder.smali	2012-03-23 19:36:20.972063409 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/ChooseAccountActivity$ViewHolder.smali.original	2012-03-23 19:34:45.066314571 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 0
  
      .prologue
+     .line 150
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 35,40 ****
--- 36,42 ----
      .parameter "x0"
  
      .prologue
+     .line 150
      invoke-direct {p0}, Landroid/accounts/ChooseAccountActivity$ViewHolder;-><init>()V
  
      return-void
