*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/ChooseAccountTypeActivity$AuthInfo.smali	2012-03-23 19:36:22.020082520 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/ChooseAccountTypeActivity$AuthInfo.smali.original	2012-03-23 19:34:45.074314716 -0500
***************
*** 30,42 ****
--- 30,47 ----
      .parameter "drawable"
  
      .prologue
+     .line 159
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 160
      iput-object p1, p0, Landroid/accounts/ChooseAccountTypeActivity$AuthInfo;->desc:Landroid/accounts/AuthenticatorDescription;
  
+     .line 161
      iput-object p2, p0, Landroid/accounts/ChooseAccountTypeActivity$AuthInfo;->name:Ljava/lang/String;
  
+     .line 162
      iput-object p3, p0, Landroid/accounts/ChooseAccountTypeActivity$AuthInfo;->drawable:Landroid/graphics/drawable/Drawable;
  
+     .line 163
      return-void
  .end method
