*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/IAccountManager$Stub.smali	2012-03-23 19:36:20.945062918 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/accounts/IAccountManager$Stub.smali.original	2012-03-23 19:34:45.066314571 -0500
***************
*** 70,81 ****
--- 70,84 ----
      .locals 1
  
      .prologue
+     .line 18
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 19
      const-string v0, "android.accounts.IAccountManager"
  
      invoke-virtual {p0, p0, v0}, Landroid/accounts/IAccountManager$Stub;->attachInterface(Landroid/os/IInterface;Ljava/lang/String;)V
  
+     .line 20
      return-void
  .end method
  
***************
*** 84,96 ****
--- 87,103 ----
      .parameter "obj"
  
      .prologue
+     .line 27
      if-nez p0, :cond_0
  
+     .line 28
      const/4 v0, 0x0
  
+     .line 34
      :goto_0
      return-object v0
  
+     .line 30
      :cond_0
      const-string v1, "android.accounts.IAccountManager"
  
***************
*** 98,103 ****
--- 105,111 ----
  
      move-result-object v0
  
+     .line 31
      .local v0, iin:Landroid/os/IInterface;
      if-eqz v0, :cond_1
  
***************
*** 105,114 ****
--- 113,124 ----
  
      if-eqz v1, :cond_1
  
+     .line 32
      check-cast v0, Landroid/accounts/IAccountManager;
  
      goto :goto_0
  
+     .line 34
      :cond_1
      new-instance v0, Landroid/accounts/IAccountManager$Stub$Proxy;
  
***************
*** 124,129 ****
--- 134,140 ----
      .locals 0
  
      .prologue
+     .line 38
      return-object p0
  .end method
  
***************
*** 144,151 ****
--- 155,164 ----
  
      const/4 v8, 0x1
  
+     .line 42
      sparse-switch p1, :sswitch_data_0
  
+     .line 383
      invoke-super {p0, p1, p2, p3, p4}, Landroid/os/Binder;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
      move-result v8
***************
*** 153,158 ****
--- 166,172 ----
      :goto_0
      return v8
  
+     .line 46
      :sswitch_0
      const-string v0, "android.accounts.IAccountManager"
  
***************
*** 160,176 ****
--- 174,193 ----
  
      goto :goto_0
  
+     .line 51
      :sswitch_1
      const-string v0, "android.accounts.IAccountManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 53
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 54
      sget-object v0, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 179,197 ****
--- 196,218 ----
  
      check-cast v1, Landroid/accounts/Account;
  
+     .line 59
      .local v1, _arg0:Landroid/accounts/Account;
      :goto_1
      invoke-virtual {p0, v1}, Landroid/accounts/IAccountManager$Stub;->getPassword(Landroid/accounts/Account;)Ljava/lang/String;
  
      move-result-object v7
  
+     .line 60
      .local v7, _result:Ljava/lang/String;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 61
      invoke-virtual {p3, v7}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
      goto :goto_0
  
+     .line 57
      .end local v1           #_arg0:Landroid/accounts/Account;
      .end local v7           #_result:Ljava/lang/String;
      :cond_0
***************
*** 200,217 ****
--- 221,241 ----
      .restart local v1       #_arg0:Landroid/accounts/Account;
      goto :goto_1
  
+     .line 66
      .end local v1           #_arg0:Landroid/accounts/Account;
      :sswitch_2
      const-string v0, "android.accounts.IAccountManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 68
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      if-eqz v0, :cond_1
  
+     .line 69
      sget-object v0, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 220,243 ****
--- 244,272 ----
  
      check-cast v1, Landroid/accounts/Account;
  
+     .line 75
      .restart local v1       #_arg0:Landroid/accounts/Account;
      :goto_2
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 76
      .local v2, _arg1:Ljava/lang/String;
      invoke-virtual {p0, v1, v2}, Landroid/accounts/IAccountManager$Stub;->getUserData(Landroid/accounts/Account;Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v7
  
+     .line 77
      .restart local v7       #_result:Ljava/lang/String;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 78
      invoke-virtual {p3, v7}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
      goto :goto_0
  
+     .line 72
      .end local v1           #_arg0:Landroid/accounts/Account;
      .end local v2           #_arg1:Ljava/lang/String;
      .end local v7           #_result:Ljava/lang/String;
***************
*** 247,291 ****
--- 276,330 ----
      .restart local v1       #_arg0:Landroid/accounts/Account;
      goto :goto_2
  
+     .line 83
      .end local v1           #_arg0:Landroid/accounts/Account;
      :sswitch_3
      const-string v0, "android.accounts.IAccountManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 84
      invoke-virtual {p0}, Landroid/accounts/IAccountManager$Stub;->getAuthenticatorTypes()[Landroid/accounts/AuthenticatorDescription;
  
      move-result-object v7
  
+     .line 85
      .local v7, _result:[Landroid/accounts/AuthenticatorDescription;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 86
      invoke-virtual {p3, v7, v8}, Landroid/os/Parcel;->writeTypedArray([Landroid/os/Parcelable;I)V
  
      goto :goto_0
  
+     .line 91
      .end local v7           #_result:[Landroid/accounts/AuthenticatorDescription;
      :sswitch_4
      const-string v0, "android.accounts.IAccountManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 93
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 94
      .local v1, _arg0:Ljava/lang/String;
      invoke-virtual {p0, v1}, Landroid/accounts/IAccountManager$Stub;->getAccounts(Ljava/lang/String;)[Landroid/accounts/Account;
  
      move-result-object v7
  
+     .line 95
      .local v7, _result:[Landroid/accounts/Account;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 96
      invoke-virtual {p3, v7, v8}, Landroid/os/Parcel;->writeTypedArray([Landroid/os/Parcelable;I)V
  
      goto :goto_0
  
+     .line 101
      .end local v1           #_arg0:Ljava/lang/String;
      .end local v7           #_result:[Landroid/accounts/Account;
      :sswitch_5
***************
*** 293,298 ****
--- 332,338 ----
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 103
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
***************
*** 301,306 ****
--- 341,347 ----
  
      move-result-object v1
  
+     .line 105
      .local v1, _arg0:Landroid/accounts/IAccountManagerResponse;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 308,313 ****
--- 349,355 ----
  
      if-eqz v0, :cond_2
  
+     .line 106
      sget-object v0, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 316,334 ****
--- 358,380 ----
  
      check-cast v2, Landroid/accounts/Account;
  
+     .line 112
      .local v2, _arg1:Landroid/accounts/Account;
      :goto_3
      invoke-virtual {p2}, Landroid/os/Parcel;->createStringArray()[Ljava/lang/String;
  
      move-result-object v3
  
+     .line 113
      .local v3, _arg2:[Ljava/lang/String;
      invoke-virtual {p0, v1, v2, v3}, Landroid/accounts/IAccountManager$Stub;->hasFeatures(Landroid/accounts/IAccountManagerResponse;Landroid/accounts/Account;[Ljava/lang/String;)V
  
+     .line 114
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 109
      .end local v2           #_arg1:Landroid/accounts/Account;
      .end local v3           #_arg2:[Ljava/lang/String;
      :cond_2
***************
*** 337,342 ****
--- 383,389 ----
      .restart local v2       #_arg1:Landroid/accounts/Account;
      goto :goto_3
  
+     .line 119
      .end local v1           #_arg0:Landroid/accounts/IAccountManagerResponse;
      .end local v2           #_arg1:Landroid/accounts/Account;
      :sswitch_6
***************
*** 344,349 ****
--- 391,397 ----
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 121
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
***************
*** 352,374 ****
--- 400,427 ----
  
      move-result-object v1
  
+     .line 123
      .restart local v1       #_arg0:Landroid/accounts/IAccountManagerResponse;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 125
      .local v2, _arg1:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->createStringArray()[Ljava/lang/String;
  
      move-result-object v3
  
+     .line 126
      .restart local v3       #_arg2:[Ljava/lang/String;
      invoke-virtual {p0, v1, v2, v3}, Landroid/accounts/IAccountManager$Stub;->getAccountsByFeatures(Landroid/accounts/IAccountManagerResponse;Ljava/lang/String;[Ljava/lang/String;)V
  
+     .line 127
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 132
      .end local v1           #_arg0:Landroid/accounts/IAccountManagerResponse;
      .end local v2           #_arg1:Ljava/lang/String;
      .end local v3           #_arg2:[Ljava/lang/String;
***************
*** 377,388 ****
--- 430,443 ----
  
      invoke-virtual {p2, v9}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 134
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v9
  
      if-eqz v9, :cond_4
  
+     .line 135
      sget-object v9, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v9, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 391,402 ****
--- 446,459 ----
  
      check-cast v1, Landroid/accounts/Account;
  
+     .line 141
      .local v1, _arg0:Landroid/accounts/Account;
      :goto_4
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 143
      .restart local v2       #_arg1:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 404,409 ****
--- 461,467 ----
  
      if-eqz v9, :cond_5
  
+     .line 144
      sget-object v9, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v9, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 412,426 ****
--- 470,487 ----
  
      check-cast v3, Landroid/os/Bundle;
  
+     .line 149
      .local v3, _arg2:Landroid/os/Bundle;
      :goto_5
      invoke-virtual {p0, v1, v2, v3}, Landroid/accounts/IAccountManager$Stub;->addAccount(Landroid/accounts/Account;Ljava/lang/String;Landroid/os/Bundle;)Z
  
      move-result v7
  
+     .line 150
      .local v7, _result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 151
      if-eqz v7, :cond_3
  
      move v0, v8
***************
*** 430,435 ****
--- 491,497 ----
  
      goto/16 :goto_0
  
+     .line 138
      .end local v1           #_arg0:Landroid/accounts/Account;
      .end local v2           #_arg1:Ljava/lang/String;
      .end local v3           #_arg2:Landroid/os/Bundle;
***************
*** 440,445 ****
--- 502,508 ----
      .restart local v1       #_arg0:Landroid/accounts/Account;
      goto :goto_4
  
+     .line 147
      .restart local v2       #_arg1:Ljava/lang/String;
      :cond_5
      const/4 v3, 0x0
***************
*** 447,452 ****
--- 510,516 ----
      .restart local v3       #_arg2:Landroid/os/Bundle;
      goto :goto_5
  
+     .line 156
      .end local v1           #_arg0:Landroid/accounts/Account;
      .end local v2           #_arg1:Ljava/lang/String;
      .end local v3           #_arg2:Landroid/os/Bundle;
***************
*** 455,460 ****
--- 519,525 ----
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 158
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
***************
*** 463,468 ****
--- 528,534 ----
  
      move-result-object v1
  
+     .line 160
      .local v1, _arg0:Landroid/accounts/IAccountManagerResponse;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 470,475 ****
--- 536,542 ----
  
      if-eqz v0, :cond_6
  
+     .line 161
      sget-object v0, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 478,491 ****
--- 545,561 ----
  
      check-cast v2, Landroid/accounts/Account;
  
+     .line 166
      .local v2, _arg1:Landroid/accounts/Account;
      :goto_6
      invoke-virtual {p0, v1, v2}, Landroid/accounts/IAccountManager$Stub;->removeAccount(Landroid/accounts/IAccountManagerResponse;Landroid/accounts/Account;)V
  
+     .line 167
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 164
      .end local v2           #_arg1:Landroid/accounts/Account;
      :cond_6
      const/4 v2, 0x0
***************
*** 493,498 ****
--- 563,569 ----
      .restart local v2       #_arg1:Landroid/accounts/Account;
      goto :goto_6
  
+     .line 172
      .end local v1           #_arg0:Landroid/accounts/IAccountManagerResponse;
      .end local v2           #_arg1:Landroid/accounts/Account;
      :sswitch_9
***************
*** 500,521 ****
--- 571,597 ----
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 174
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 176
      .local v1, _arg0:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 177
      .local v2, _arg1:Ljava/lang/String;
      invoke-virtual {p0, v1, v2}, Landroid/accounts/IAccountManager$Stub;->invalidateAuthToken(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 178
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 183
      .end local v1           #_arg0:Ljava/lang/String;
      .end local v2           #_arg1:Ljava/lang/String;
      :sswitch_a
***************
*** 523,534 ****
--- 599,612 ----
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 185
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      if-eqz v0, :cond_7
  
+     .line 186
      sget-object v0, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 537,560 ****
--- 615,643 ----
  
      check-cast v1, Landroid/accounts/Account;
  
+     .line 192
      .local v1, _arg0:Landroid/accounts/Account;
      :goto_7
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 193
      .restart local v2       #_arg1:Ljava/lang/String;
      invoke-virtual {p0, v1, v2}, Landroid/accounts/IAccountManager$Stub;->peekAuthToken(Landroid/accounts/Account;Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v7
  
+     .line 194
      .local v7, _result:Ljava/lang/String;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 195
      invoke-virtual {p3, v7}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
      goto/16 :goto_0
  
+     .line 189
      .end local v1           #_arg0:Landroid/accounts/Account;
      .end local v2           #_arg1:Ljava/lang/String;
      .end local v7           #_result:Ljava/lang/String;
***************
*** 564,581 ****
--- 647,667 ----
      .restart local v1       #_arg0:Landroid/accounts/Account;
      goto :goto_7
  
+     .line 200
      .end local v1           #_arg0:Landroid/accounts/Account;
      :sswitch_b
      const-string v0, "android.accounts.IAccountManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 202
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      if-eqz v0, :cond_8
  
+     .line 203
      sget-object v0, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 584,607 ****
--- 670,698 ----
  
      check-cast v1, Landroid/accounts/Account;
  
+     .line 209
      .restart local v1       #_arg0:Landroid/accounts/Account;
      :goto_8
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 211
      .restart local v2       #_arg1:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 212
      .local v3, _arg2:Ljava/lang/String;
      invoke-virtual {p0, v1, v2, v3}, Landroid/accounts/IAccountManager$Stub;->setAuthToken(Landroid/accounts/Account;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 213
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 206
      .end local v1           #_arg0:Landroid/accounts/Account;
      .end local v2           #_arg1:Ljava/lang/String;
      .end local v3           #_arg2:Ljava/lang/String;
***************
*** 611,628 ****
--- 702,722 ----
      .restart local v1       #_arg0:Landroid/accounts/Account;
      goto :goto_8
  
+     .line 218
      .end local v1           #_arg0:Landroid/accounts/Account;
      :sswitch_c
      const-string v0, "android.accounts.IAccountManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 220
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      if-eqz v0, :cond_9
  
+     .line 221
      sget-object v0, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 631,649 ****
--- 725,747 ----
  
      check-cast v1, Landroid/accounts/Account;
  
+     .line 227
      .restart local v1       #_arg0:Landroid/accounts/Account;
      :goto_9
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 228
      .restart local v2       #_arg1:Ljava/lang/String;
      invoke-virtual {p0, v1, v2}, Landroid/accounts/IAccountManager$Stub;->setPassword(Landroid/accounts/Account;Ljava/lang/String;)V
  
+     .line 229
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 224
      .end local v1           #_arg0:Landroid/accounts/Account;
      .end local v2           #_arg1:Ljava/lang/String;
      :cond_9
***************
*** 652,669 ****
--- 750,770 ----
      .restart local v1       #_arg0:Landroid/accounts/Account;
      goto :goto_9
  
+     .line 234
      .end local v1           #_arg0:Landroid/accounts/Account;
      :sswitch_d
      const-string v0, "android.accounts.IAccountManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 236
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      if-eqz v0, :cond_a
  
+     .line 237
      sget-object v0, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 672,685 ****
--- 773,789 ----
  
      check-cast v1, Landroid/accounts/Account;
  
+     .line 242
      .restart local v1       #_arg0:Landroid/accounts/Account;
      :goto_a
      invoke-virtual {p0, v1}, Landroid/accounts/IAccountManager$Stub;->clearPassword(Landroid/accounts/Account;)V
  
+     .line 243
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 240
      .end local v1           #_arg0:Landroid/accounts/Account;
      :cond_a
      const/4 v1, 0x0
***************
*** 687,704 ****
--- 791,811 ----
      .restart local v1       #_arg0:Landroid/accounts/Account;
      goto :goto_a
  
+     .line 248
      .end local v1           #_arg0:Landroid/accounts/Account;
      :sswitch_e
      const-string v0, "android.accounts.IAccountManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 250
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      if-eqz v0, :cond_b
  
+     .line 251
      sget-object v0, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 707,730 ****
--- 814,842 ----
  
      check-cast v1, Landroid/accounts/Account;
  
+     .line 257
      .restart local v1       #_arg0:Landroid/accounts/Account;
      :goto_b
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 259
      .restart local v2       #_arg1:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 260
      .restart local v3       #_arg2:Ljava/lang/String;
      invoke-virtual {p0, v1, v2, v3}, Landroid/accounts/IAccountManager$Stub;->setUserData(Landroid/accounts/Account;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 261
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 254
      .end local v1           #_arg0:Landroid/accounts/Account;
      .end local v2           #_arg1:Ljava/lang/String;
      .end local v3           #_arg2:Ljava/lang/String;
***************
*** 734,745 ****
--- 846,859 ----
      .restart local v1       #_arg0:Landroid/accounts/Account;
      goto :goto_b
  
+     .line 266
      .end local v1           #_arg0:Landroid/accounts/Account;
      :sswitch_f
      const-string v9, "android.accounts.IAccountManager"
  
      invoke-virtual {p2, v9}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 268
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v9
***************
*** 748,753 ****
--- 862,868 ----
  
      move-result-object v1
  
+     .line 270
      .local v1, _arg0:Landroid/accounts/IAccountManagerResponse;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 755,760 ****
--- 870,876 ----
  
      if-eqz v9, :cond_c
  
+     .line 271
      sget-object v9, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v9, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 763,774 ****
--- 879,892 ----
  
      check-cast v2, Landroid/accounts/Account;
  
+     .line 277
      .local v2, _arg1:Landroid/accounts/Account;
      :goto_c
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 279
      .restart local v3       #_arg2:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 778,783 ****
--- 896,902 ----
  
      move v4, v8
  
+     .line 281
      .local v4, _arg3:Z
      :goto_d
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
***************
*** 788,793 ****
--- 907,913 ----
  
      move v5, v8
  
+     .line 283
      .local v5, _arg4:Z
      :goto_e
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
***************
*** 796,801 ****
--- 916,922 ----
  
      if-eqz v0, :cond_f
  
+     .line 284
      sget-object v0, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 808,819 ****
--- 929,943 ----
      :goto_f
      move-object v0, p0
  
+     .line 289
      invoke-virtual/range {v0 .. v6}, Landroid/accounts/IAccountManager$Stub;->getAuthToken(Landroid/accounts/IAccountManagerResponse;Landroid/accounts/Account;Ljava/lang/String;ZZLandroid/os/Bundle;)V
  
+     .line 290
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 274
      .end local v2           #_arg1:Landroid/accounts/Account;
      .end local v3           #_arg2:Ljava/lang/String;
      .end local v4           #_arg3:Z
***************
*** 829,842 ****
--- 953,969 ----
      :cond_d
      move v4, v0
  
+     .line 279
      goto :goto_d
  
      .restart local v4       #_arg3:Z
      :cond_e
      move v5, v0
  
+     .line 281
      goto :goto_e
  
+     .line 287
      .restart local v5       #_arg4:Z
      :cond_f
      const/4 v6, 0x0
***************
*** 844,849 ****
--- 971,977 ----
      .restart local v6       #_arg5:Landroid/os/Bundle;
      goto :goto_f
  
+     .line 295
      .end local v1           #_arg0:Landroid/accounts/IAccountManagerResponse;
      .end local v2           #_arg1:Landroid/accounts/Account;
      .end local v3           #_arg2:Ljava/lang/String;
***************
*** 855,860 ****
--- 983,989 ----
  
      invoke-virtual {p2, v9}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 297
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v9
***************
*** 863,883 ****
--- 992,1016 ----
  
      move-result-object v1
  
+     .line 299
      .restart local v1       #_arg0:Landroid/accounts/IAccountManagerResponse;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 301
      .local v2, _arg1:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 303
      .restart local v3       #_arg2:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->createStringArray()[Ljava/lang/String;
  
      move-result-object v4
  
+     .line 305
      .local v4, _arg3:[Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 887,892 ****
--- 1020,1026 ----
  
      move v5, v8
  
+     .line 307
      .restart local v5       #_arg4:Z
      :goto_10
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
***************
*** 895,900 ****
--- 1029,1035 ----
  
      if-eqz v0, :cond_11
  
+     .line 308
      sget-object v0, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 907,914 ****
--- 1042,1051 ----
      :goto_11
      move-object v0, p0
  
+     .line 313
      invoke-virtual/range {v0 .. v6}, Landroid/accounts/IAccountManager$Stub;->addAcount(Landroid/accounts/IAccountManagerResponse;Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;ZLandroid/os/Bundle;)V
  
+     .line 314
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
***************
*** 918,925 ****
--- 1055,1064 ----
      :cond_10
      move v5, v0
  
+     .line 305
      goto :goto_10
  
+     .line 311
      .restart local v5       #_arg4:Z
      :cond_11
      const/4 v6, 0x0
***************
*** 927,932 ****
--- 1066,1072 ----
      .restart local v6       #_arg5:Landroid/os/Bundle;
      goto :goto_11
  
+     .line 319
      .end local v1           #_arg0:Landroid/accounts/IAccountManagerResponse;
      .end local v2           #_arg1:Ljava/lang/String;
      .end local v3           #_arg2:Ljava/lang/String;
***************
*** 938,943 ****
--- 1078,1084 ----
  
      invoke-virtual {p2, v9}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 321
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v9
***************
*** 946,951 ****
--- 1087,1093 ----
  
      move-result-object v1
  
+     .line 323
      .restart local v1       #_arg0:Landroid/accounts/IAccountManagerResponse;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 953,958 ****
--- 1095,1101 ----
  
      if-eqz v9, :cond_12
  
+     .line 324
      sget-object v9, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v9, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 961,972 ****
--- 1104,1117 ----
  
      check-cast v2, Landroid/accounts/Account;
  
+     .line 330
      .local v2, _arg1:Landroid/accounts/Account;
      :goto_12
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 332
      .restart local v3       #_arg2:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 976,981 ****
--- 1121,1127 ----
  
      move v4, v8
  
+     .line 334
      .local v4, _arg3:Z
      :goto_13
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
***************
*** 984,989 ****
--- 1130,1136 ----
  
      if-eqz v0, :cond_14
  
+     .line 335
      sget-object v0, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 996,1007 ****
--- 1143,1157 ----
      :goto_14
      move-object v0, p0
  
+     .line 340
      invoke-virtual/range {v0 .. v5}, Landroid/accounts/IAccountManager$Stub;->updateCredentials(Landroid/accounts/IAccountManagerResponse;Landroid/accounts/Account;Ljava/lang/String;ZLandroid/os/Bundle;)V
  
+     .line 341
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 327
      .end local v2           #_arg1:Landroid/accounts/Account;
      .end local v3           #_arg2:Ljava/lang/String;
      .end local v4           #_arg3:Z
***************
*** 1016,1023 ****
--- 1166,1175 ----
      :cond_13
      move v4, v0
  
+     .line 332
      goto :goto_13
  
+     .line 338
      .restart local v4       #_arg3:Z
      :cond_14
      const/4 v5, 0x0
***************
*** 1025,1030 ****
--- 1177,1183 ----
      .restart local v5       #_arg4:Landroid/os/Bundle;
      goto :goto_14
  
+     .line 346
      .end local v1           #_arg0:Landroid/accounts/IAccountManagerResponse;
      .end local v2           #_arg1:Landroid/accounts/Account;
      .end local v3           #_arg2:Ljava/lang/String;
***************
*** 1035,1040 ****
--- 1188,1194 ----
  
      invoke-virtual {p2, v9}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 348
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v9
***************
*** 1043,1053 ****
--- 1197,1209 ----
  
      move-result-object v1
  
+     .line 350
      .restart local v1       #_arg0:Landroid/accounts/IAccountManagerResponse;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 352
      .local v2, _arg1:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 1057,1066 ****
--- 1213,1224 ----
  
      move v3, v8
  
+     .line 353
      .local v3, _arg2:Z
      :goto_15
      invoke-virtual {p0, v1, v2, v3}, Landroid/accounts/IAccountManager$Stub;->editProperties(Landroid/accounts/IAccountManagerResponse;Ljava/lang/String;Z)V
  
+     .line 354
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
***************
*** 1069,1076 ****
--- 1227,1236 ----
      :cond_15
      move v3, v0
  
+     .line 352
      goto :goto_15
  
+     .line 359
      .end local v1           #_arg0:Landroid/accounts/IAccountManagerResponse;
      .end local v2           #_arg1:Ljava/lang/String;
      :sswitch_13
***************
*** 1078,1083 ****
--- 1238,1244 ----
  
      invoke-virtual {p2, v9}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 361
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v9
***************
*** 1086,1091 ****
--- 1247,1253 ----
  
      move-result-object v1
  
+     .line 363
      .restart local v1       #_arg0:Landroid/accounts/IAccountManagerResponse;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 1093,1098 ****
--- 1255,1261 ----
  
      if-eqz v9, :cond_16
  
+     .line 364
      sget-object v9, Landroid/accounts/Account;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v9, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 1101,1106 ****
--- 1264,1270 ----
  
      check-cast v2, Landroid/accounts/Account;
  
+     .line 370
      .local v2, _arg1:Landroid/accounts/Account;
      :goto_16
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
***************
*** 1109,1114 ****
--- 1273,1279 ----
  
      if-eqz v9, :cond_17
  
+     .line 371
      sget-object v9, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v9, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 1117,1122 ****
--- 1282,1288 ----
  
      check-cast v3, Landroid/os/Bundle;
  
+     .line 377
      .local v3, _arg2:Landroid/os/Bundle;
      :goto_17
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
***************
*** 1127,1140 ****
--- 1293,1309 ----
  
      move v4, v8
  
+     .line 378
      .restart local v4       #_arg3:Z
      :goto_18
      invoke-virtual {p0, v1, v2, v3, v4}, Landroid/accounts/IAccountManager$Stub;->confirmCredentials(Landroid/accounts/IAccountManagerResponse;Landroid/accounts/Account;Landroid/os/Bundle;Z)V
  
+     .line 379
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 367
      .end local v2           #_arg1:Landroid/accounts/Account;
      .end local v3           #_arg2:Landroid/os/Bundle;
      .end local v4           #_arg3:Z
***************
*** 1144,1149 ****
--- 1313,1319 ----
      .restart local v2       #_arg1:Landroid/accounts/Account;
      goto :goto_16
  
+     .line 374
      :cond_17
      const/4 v3, 0x0
  
***************
*** 1153,1160 ****
--- 1323,1332 ----
      :cond_18
      move v4, v0
  
+     .line 377
      goto :goto_18
  
+     .line 42
      nop
  
      :sswitch_data_0
