*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/nfc/ApduList.smali	2012-03-23 19:36:05.405779558 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/nfc/ApduList.smali.original	2012-03-23 19:34:44.964312710 -0500
***************
*** 35,40 ****
--- 35,41 ----
      .locals 1
  
      .prologue
+     .line 27
      new-instance v0, Landroid/nfc/ApduList$1;
  
      invoke-direct {v0}, Landroid/nfc/ApduList$1;-><init>()V
***************
*** 48,61 ****
--- 49,65 ----
      .locals 1
  
      .prologue
+     .line 16
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 14
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/nfc/ApduList;->commands:Ljava/util/ArrayList;
  
+     .line 17
      return-void
  .end method
  
***************
*** 64,81 ****
--- 68,89 ----
      .parameter "in"
  
      .prologue
+     .line 40
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 14
      new-instance v4, Ljava/util/ArrayList;
  
      invoke-direct {v4}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v4, p0, Landroid/nfc/ApduList;->commands:Ljava/util/ArrayList;
  
+     .line 41
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
+     .line 43
      .local v1, count:I
      const/4 v2, 0x0
  
***************
*** 83,106 ****
--- 91,120 ----
      :goto_0
      if-ge v2, v1, :cond_0
  
+     .line 45
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
+     .line 46
      .local v3, length:I
      new-array v0, v3, [B
  
+     .line 47
      .local v0, cmd:[B
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->readByteArray([B)V
  
+     .line 48
      iget-object v4, p0, Landroid/nfc/ApduList;->commands:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v0}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 43
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 50
      .end local v0           #cmd:[B
      .end local v3           #length:I
      :cond_0
***************
*** 113,118 ****
--- 127,133 ----
      .parameter "x1"
  
      .prologue
+     .line 12
      invoke-direct {p0, p1}, Landroid/nfc/ApduList;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 125,134 ****
--- 140,151 ----
      .parameter "command"
  
      .prologue
+     .line 20
      iget-object v0, p0, Landroid/nfc/ApduList;->commands:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 21
      return-void
  .end method
  
***************
*** 136,141 ****
--- 153,159 ----
      .locals 1
  
      .prologue
+     .line 54
      const/4 v0, 0x0
  
      return v0
***************
*** 152,157 ****
--- 170,176 ----
      .end annotation
  
      .prologue
+     .line 24
      iget-object v0, p0, Landroid/nfc/ApduList;->commands:Ljava/util/ArrayList;
  
      return-object v0
***************
*** 163,168 ****
--- 182,188 ----
      .parameter "flags"
  
      .prologue
+     .line 59
      iget-object v2, p0, Landroid/nfc/ApduList;->commands:Ljava/util/ArrayList;
  
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
***************
*** 171,176 ****
--- 191,197 ----
  
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 61
      iget-object v2, p0, Landroid/nfc/ApduList;->commands:Ljava/util/ArrayList;
  
      invoke-virtual {v2}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
***************
*** 191,205 ****
--- 212,229 ----
  
      check-cast v0, [B
  
+     .line 62
      .local v0, cmd:[B
      array-length v2, v0
  
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 63
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeByteArray([B)V
  
      goto :goto_0
  
+     .line 65
      .end local v0           #cmd:[B
      :cond_0
      return-void
