*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/nfc/NfcActivityManager$NfcActivityState.smali	2012-03-23 19:36:05.731785502 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/nfc/NfcActivityManager$NfcActivityState.smali.original	2012-03-23 19:34:44.966312746 -0500
***************
*** 32,41 ****
--- 32,43 ----
      .parameter
  
      .prologue
+     .line 47
      iput-object p1, p0, Landroid/nfc/NfcActivityManager$NfcActivityState;->this$0:Landroid/nfc/NfcActivityManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 48
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/nfc/NfcActivityManager$NfcActivityState;->resumed:Z
***************
*** 49,54 ****
--- 51,57 ----
      .locals 3
  
      .prologue
+     .line 54
      new-instance v1, Ljava/lang/StringBuilder;
  
      const-string v2, "["
***************
*** 67,72 ****
--- 70,76 ----
  
      move-result-object v0
  
+     .line 55
      .local v0, s:Ljava/lang/StringBuilder;
      iget-object v1, p0, Landroid/nfc/NfcActivityManager$NfcActivityState;->ndefMessage:Landroid/nfc/NdefMessage;
  
***************
*** 90,95 ****
--- 94,100 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 56
      iget-object v1, p0, Landroid/nfc/NfcActivityManager$NfcActivityState;->onNdefPushCompleteCallback:Landroid/nfc/NfcAdapter$OnNdefPushCompleteCallback;
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
***************
*** 100,105 ****
--- 105,111 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 57
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
