*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/nfc/TechListParcel.smali	2012-03-23 19:36:06.716803464 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/nfc/TechListParcel.smali.original	2012-03-23 19:34:44.971312838 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .locals 1
  
      .prologue
+     .line 50
      new-instance v0, Landroid/nfc/TechListParcel$1;
  
      invoke-direct {v0}, Landroid/nfc/TechListParcel$1;-><init>()V
***************
*** 42,51 ****
--- 43,55 ----
      .parameter "strings"
  
      .prologue
+     .line 27
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 28
      iput-object p1, p0, Landroid/nfc/TechListParcel;->mTechLists:[[Ljava/lang/String;
  
+     .line 29
      return-void
  .end method
  
***************
*** 55,60 ****
--- 59,65 ----
      .locals 1
  
      .prologue
+     .line 37
      const/4 v0, 0x0
  
      return v0
***************
*** 64,69 ****
--- 69,75 ----
      .locals 1
  
      .prologue
+     .line 32
      iget-object v0, p0, Landroid/nfc/TechListParcel;->mTechLists:[[Ljava/lang/String;
  
      return-object v0
***************
*** 75,104 ****
--- 81,117 ----
      .parameter "flags"
  
      .prologue
+     .line 42
      iget-object v3, p0, Landroid/nfc/TechListParcel;->mTechLists:[[Ljava/lang/String;
  
      array-length v0, v3
  
+     .line 43
      .local v0, count:I
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 44
      const/4 v1, 0x0
  
      .local v1, i:I
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 45
      iget-object v3, p0, Landroid/nfc/TechListParcel;->mTechLists:[[Ljava/lang/String;
  
      aget-object v2, v3, v1
  
+     .line 46
      .local v2, techList:[Ljava/lang/String;
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->writeStringArray([Ljava/lang/String;)V
  
+     .line 44
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 48
      .end local v2           #techList:[Ljava/lang/String;
      :cond_0
      return-void
