*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/nfc/TransceiveResult.smali	2012-03-23 19:36:05.636783771 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/nfc/TransceiveResult.smali.original	2012-03-23 19:34:44.966312746 -0500
***************
*** 38,43 ****
--- 38,44 ----
      .locals 1
  
      .prologue
+     .line 70
      new-instance v0, Landroid/nfc/TransceiveResult$1;
  
      invoke-direct {v0}, Landroid/nfc/TransceiveResult$1;-><init>()V
***************
*** 53,64 ****
--- 54,69 ----
      .parameter "data"
  
      .prologue
+     .line 38
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 39
      iput p1, p0, Landroid/nfc/TransceiveResult;->mResult:I
  
+     .line 40
      iput-object p2, p0, Landroid/nfc/TransceiveResult;->mResponseData:[B
  
+     .line 41
      return-void
  .end method
  
***************
*** 68,73 ****
--- 73,79 ----
      .locals 1
  
      .prologue
+     .line 58
      const/4 v0, 0x0
  
      return v0
***************
*** 82,91 ****
--- 88,99 ----
      .end annotation
  
      .prologue
+     .line 44
      iget v0, p0, Landroid/nfc/TransceiveResult;->mResult:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 52
      :pswitch_0
      new-instance v0, Ljava/io/IOException;
  
***************
*** 95,105 ****
--- 103,115 ----
  
      throw v0
  
+     .line 46
      :pswitch_1
      iget-object v0, p0, Landroid/nfc/TransceiveResult;->mResponseData:[B
  
      return-object v0
  
+     .line 48
      :pswitch_2
      new-instance v0, Landroid/nfc/TagLostException;
  
***************
*** 109,114 ****
--- 119,125 ----
  
      throw v0
  
+     .line 50
      :pswitch_3
      new-instance v0, Ljava/io/IOException;
  
***************
*** 118,123 ****
--- 129,135 ----
  
      throw v0
  
+     .line 44
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_1
***************
*** 133,156 ****
--- 145,173 ----
      .parameter "flags"
  
      .prologue
+     .line 63
      iget v0, p0, Landroid/nfc/TransceiveResult;->mResult:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 64
      iget v0, p0, Landroid/nfc/TransceiveResult;->mResult:I
  
      if-nez v0, :cond_0
  
+     .line 65
      iget-object v0, p0, Landroid/nfc/TransceiveResult;->mResponseData:[B
  
      array-length v0, v0
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 66
      iget-object v0, p0, Landroid/nfc/TransceiveResult;->mResponseData:[B
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeByteArray([B)V
  
+     .line 68
      :cond_0
      return-void
  .end method
