*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/DhcpStateMachine.smali	2012-03-23 19:36:54.735679089 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/DhcpStateMachine.smali.original	2012-03-23 19:34:45.403320715 -0500
***************
*** 88,135 ****
--- 88,146 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 111
      const-string v2, "DhcpStateMachine"
  
      invoke-direct {p0, v2}, Lcom/android/internal/util/StateMachine;-><init>(Ljava/lang/String;)V
  
+     .line 81
      iput-boolean v4, p0, Landroid/net/DhcpStateMachine;->mRegisteredForPreDhcpNotification:Z
  
+     .line 104
      new-instance v2, Landroid/net/DhcpStateMachine$DefaultState;
  
      invoke-direct {v2, p0}, Landroid/net/DhcpStateMachine$DefaultState;-><init>(Landroid/net/DhcpStateMachine;)V
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mDefaultState:Lcom/android/internal/util/State;
  
+     .line 105
      new-instance v2, Landroid/net/DhcpStateMachine$StoppedState;
  
      invoke-direct {v2, p0}, Landroid/net/DhcpStateMachine$StoppedState;-><init>(Landroid/net/DhcpStateMachine;)V
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mStoppedState:Lcom/android/internal/util/State;
  
+     .line 106
      new-instance v2, Landroid/net/DhcpStateMachine$WaitBeforeStartState;
  
      invoke-direct {v2, p0}, Landroid/net/DhcpStateMachine$WaitBeforeStartState;-><init>(Landroid/net/DhcpStateMachine;)V
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mWaitBeforeStartState:Lcom/android/internal/util/State;
  
+     .line 107
      new-instance v2, Landroid/net/DhcpStateMachine$RunningState;
  
      invoke-direct {v2, p0}, Landroid/net/DhcpStateMachine$RunningState;-><init>(Landroid/net/DhcpStateMachine;)V
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mRunningState:Lcom/android/internal/util/State;
  
+     .line 108
      new-instance v2, Landroid/net/DhcpStateMachine$WaitBeforeRenewalState;
  
      invoke-direct {v2, p0}, Landroid/net/DhcpStateMachine$WaitBeforeRenewalState;-><init>(Landroid/net/DhcpStateMachine;)V
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mWaitBeforeRenewalState:Lcom/android/internal/util/State;
  
+     .line 113
      iput-object p1, p0, Landroid/net/DhcpStateMachine;->mContext:Landroid/content/Context;
  
+     .line 114
      iput-object p2, p0, Landroid/net/DhcpStateMachine;->mController:Lcom/android/internal/util/StateMachine;
  
+     .line 115
      iput-object p3, p0, Landroid/net/DhcpStateMachine;->mInterfaceName:Ljava/lang/String;
  
+     .line 117
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mContext:Landroid/content/Context;
  
      const-string v3, "alarm"
***************
*** 142,147 ****
--- 153,159 ----
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mAlarmManager:Landroid/app/AlarmManager;
  
+     .line 118
      new-instance v0, Landroid/content/Intent;
  
      const-string v2, "android.net.wifi.DHCP_RENEW"
***************
*** 150,155 ****
--- 162,168 ----
  
      invoke-direct {v0, v2, v3}, Landroid/content/Intent;-><init>(Ljava/lang/String;Landroid/net/Uri;)V
  
+     .line 119
      .local v0, dhcpRenewalIntent:Landroid/content/Intent;
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mContext:Landroid/content/Context;
  
***************
*** 159,167 ****
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mDhcpRenewalIntent:Landroid/app/PendingIntent;
  
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mContext:Landroid/content/Context;
  
!     const-string v3, "power"
  
      invoke-virtual {v2, v3}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
--- 172,181 ----
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mDhcpRenewalIntent:Landroid/app/PendingIntent;
  
+     .line 121
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mContext:Landroid/content/Context;
  
!     const-string/jumbo v3, "power"
  
      invoke-virtual {v2, v3}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
***************
*** 169,174 ****
--- 183,189 ----
  
      check-cast v1, Landroid/os/PowerManager;
  
+     .line 122
      .local v1, powerManager:Landroid/os/PowerManager;
      const/4 v2, 0x1
  
***************
*** 180,195 ****
--- 195,213 ----
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mDhcpRenewWakeLock:Landroid/os/PowerManager$WakeLock;
  
+     .line 123
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mDhcpRenewWakeLock:Landroid/os/PowerManager$WakeLock;
  
      invoke-virtual {v2, v4}, Landroid/os/PowerManager$WakeLock;->setReferenceCounted(Z)V
  
+     .line 125
      new-instance v2, Landroid/net/DhcpStateMachine$1;
  
      invoke-direct {v2, p0}, Landroid/net/DhcpStateMachine$1;-><init>(Landroid/net/DhcpStateMachine;)V
  
      iput-object v2, p0, Landroid/net/DhcpStateMachine;->mBroadcastReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 135
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mContext:Landroid/content/Context;
  
      iget-object v3, p0, Landroid/net/DhcpStateMachine;->mBroadcastReceiver:Landroid/content/BroadcastReceiver;
***************
*** 202,239 ****
--- 220,264 ----
  
      invoke-virtual {v2, v3, v4}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 137
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mDefaultState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v2}, Landroid/net/DhcpStateMachine;->addState(Lcom/android/internal/util/State;)V
  
+     .line 138
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mStoppedState:Lcom/android/internal/util/State;
  
      iget-object v3, p0, Landroid/net/DhcpStateMachine;->mDefaultState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v2, v3}, Landroid/net/DhcpStateMachine;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)V
  
+     .line 139
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mWaitBeforeStartState:Lcom/android/internal/util/State;
  
      iget-object v3, p0, Landroid/net/DhcpStateMachine;->mDefaultState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v2, v3}, Landroid/net/DhcpStateMachine;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)V
  
+     .line 140
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mRunningState:Lcom/android/internal/util/State;
  
      iget-object v3, p0, Landroid/net/DhcpStateMachine;->mDefaultState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v2, v3}, Landroid/net/DhcpStateMachine;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)V
  
+     .line 141
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mWaitBeforeRenewalState:Lcom/android/internal/util/State;
  
      iget-object v3, p0, Landroid/net/DhcpStateMachine;->mDefaultState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v2, v3}, Landroid/net/DhcpStateMachine;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)V
  
+     .line 143
      iget-object v2, p0, Landroid/net/DhcpStateMachine;->mStoppedState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v2}, Landroid/net/DhcpStateMachine;->setInitialState(Lcom/android/internal/util/State;)V
  
+     .line 144
      return-void
  .end method
  
***************
*** 242,247 ****
--- 267,273 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mDhcpRenewWakeLock:Landroid/os/PowerManager$WakeLock;
  
      return-object v0
***************
*** 252,257 ****
--- 278,284 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mInterfaceName:Ljava/lang/String;
  
      return-object v0
***************
*** 263,268 ****
--- 290,296 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 274,279 ****
--- 302,308 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 284,289 ****
--- 313,319 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mStoppedState:Lcom/android/internal/util/State;
  
      return-object v0
***************
*** 295,300 ****
--- 325,331 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 306,311 ****
--- 337,343 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 316,321 ****
--- 348,354 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mDhcpRenewalIntent:Landroid/app/PendingIntent;
  
      return-object v0
***************
*** 326,331 ****
--- 359,365 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mAlarmManager:Landroid/app/AlarmManager;
  
      return-object v0
***************
*** 337,342 ****
--- 371,377 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 347,352 ****
--- 382,388 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mWaitBeforeRenewalState:Lcom/android/internal/util/State;
  
      return-object v0
***************
*** 358,363 ****
--- 394,400 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 368,373 ****
--- 405,411 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mBroadcastReceiver:Landroid/content/BroadcastReceiver;
  
      return-object v0
***************
*** 379,384 ****
--- 417,423 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 390,395 ****
--- 429,435 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 401,406 ****
--- 441,447 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 412,417 ****
--- 453,459 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 422,427 ****
--- 464,470 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 432,437 ****
--- 475,481 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-boolean v0, p0, Landroid/net/DhcpStateMachine;->mRegisteredForPreDhcpNotification:Z
  
      return v0
***************
*** 442,447 ****
--- 486,492 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mController:Lcom/android/internal/util/StateMachine;
  
      return-object v0
***************
*** 452,457 ****
--- 497,503 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mWaitBeforeStartState:Lcom/android/internal/util/State;
  
      return-object v0
***************
*** 463,468 ****
--- 509,515 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-virtual {p0, p1}, Landroid/net/DhcpStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 474,479 ****
--- 521,527 ----
      .parameter "x1"
  
      .prologue
+     .line 50
      invoke-direct {p0, p1}, Landroid/net/DhcpStateMachine;->runDhcp(Landroid/net/DhcpStateMachine$DhcpAction;)Z
  
      move-result v0
***************
*** 486,491 ****
--- 534,540 ----
      .parameter "x0"
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/net/DhcpStateMachine;->mRunningState:Lcom/android/internal/util/State;
  
      return-object v0
***************
*** 498,510 ****
--- 547,562 ----
      .parameter "intf"
  
      .prologue
+     .line 148
      new-instance v0, Landroid/net/DhcpStateMachine;
  
      invoke-direct {v0, p0, p1, p2}, Landroid/net/DhcpStateMachine;-><init>(Landroid/content/Context;Lcom/android/internal/util/StateMachine;Ljava/lang/String;)V
  
+     .line 149
      .local v0, dsm:Landroid/net/DhcpStateMachine;
      invoke-virtual {v0}, Landroid/net/DhcpStateMachine;->start()V
  
+     .line 150
      return-object v0
  .end method
  
***************
*** 519,552 ****
--- 571,612 ----
  
      const/4 v9, 0x0
  
+     .line 335
      const/4 v3, 0x0
  
+     .line 336
      .local v3, success:Z
      new-instance v0, Landroid/net/DhcpInfoInternal;
  
      invoke-direct {v0}, Landroid/net/DhcpInfoInternal;-><init>()V
  
+     .line 338
      .local v0, dhcpInfoInternal:Landroid/net/DhcpInfoInternal;
      sget-object v4, Landroid/net/DhcpStateMachine$DhcpAction;->START:Landroid/net/DhcpStateMachine$DhcpAction;
  
      if-ne p1, v4, :cond_2
  
+     .line 340
      iget-object v4, p0, Landroid/net/DhcpStateMachine;->mInterfaceName:Ljava/lang/String;
  
      invoke-static {v4, v0}, Landroid/net/NetworkUtils;->runDhcp(Ljava/lang/String;Landroid/net/DhcpInfoInternal;)Z
  
      move-result v3
  
+     .line 341
      iput-object v0, p0, Landroid/net/DhcpStateMachine;->mDhcpInfo:Landroid/net/DhcpInfoInternal;
  
+     .line 348
      :cond_0
      :goto_0
      if-eqz v3, :cond_3
  
+     .line 350
      iget v4, v0, Landroid/net/DhcpInfoInternal;->leaseDuration:I
  
      int-to-long v1, v4
  
+     .line 355
      .local v1, leaseDuration:J
      const-wide/16 v4, 0x12c
  
***************
*** 554,561 ****
--- 614,623 ----
  
      if-gez v4, :cond_1
  
+     .line 356
      const-wide/16 v1, 0x12c
  
+     .line 361
      :cond_1
      iget-object v4, p0, Landroid/net/DhcpStateMachine;->mAlarmManager:Landroid/app/AlarmManager;
  
***************
*** 573,578 ****
--- 635,641 ----
  
      invoke-virtual {v4, v10, v5, v6, v7}, Landroid/app/AlarmManager;->set(IJLandroid/app/PendingIntent;)V
  
+     .line 366
      iget-object v4, p0, Landroid/net/DhcpStateMachine;->mController:Lcom/android/internal/util/StateMachine;
  
      const/4 v5, 0x1
***************
*** 583,609 ****
--- 646,677 ----
  
      invoke-virtual {v4}, Landroid/os/Message;->sendToTarget()V
  
+     .line 375
      .end local v1           #leaseDuration:J
      :goto_1
      return v3
  
+     .line 342
      :cond_2
      sget-object v4, Landroid/net/DhcpStateMachine$DhcpAction;->RENEW:Landroid/net/DhcpStateMachine$DhcpAction;
  
      if-ne p1, v4, :cond_0
  
+     .line 344
      iget-object v4, p0, Landroid/net/DhcpStateMachine;->mInterfaceName:Ljava/lang/String;
  
      invoke-static {v4, v0}, Landroid/net/NetworkUtils;->runDhcpRenew(Ljava/lang/String;Landroid/net/DhcpInfoInternal;)Z
  
      move-result v3
  
+     .line 345
      iget-object v4, p0, Landroid/net/DhcpStateMachine;->mDhcpInfo:Landroid/net/DhcpInfoInternal;
  
      invoke-virtual {v0, v4}, Landroid/net/DhcpInfoInternal;->updateFromDhcpRequest(Landroid/net/DhcpInfoInternal;)V
  
      goto :goto_0
  
+     .line 369
      :cond_3
      const-string v4, "DhcpStateMachine"
  
***************
*** 643,652 ****
--- 711,722 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 371
      iget-object v4, p0, Landroid/net/DhcpStateMachine;->mInterfaceName:Ljava/lang/String;
  
      invoke-static {v4}, Landroid/net/NetworkUtils;->stopDhcp(Ljava/lang/String;)Z
  
+     .line 372
      iget-object v4, p0, Landroid/net/DhcpStateMachine;->mController:Lcom/android/internal/util/StateMachine;
  
      invoke-virtual {v4, v11, v10, v9}, Lcom/android/internal/util/StateMachine;->obtainMessage(III)Landroid/os/Message;
***************
*** 664,672 ****
--- 734,744 ----
      .locals 1
  
      .prologue
+     .line 163
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/net/DhcpStateMachine;->mRegisteredForPreDhcpNotification:Z
  
+     .line 164
      return-void
  .end method
