*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/DnsPinger$DnsArg.smali	2012-03-23 19:36:54.210669514 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/DnsPinger$DnsArg.smali.original	2012-03-23 19:34:45.398320625 -0500
***************
*** 30,42 ****
--- 30,46 ----
      .parameter "s"
  
      .prologue
+     .line 113
      iput-object p1, p0, Landroid/net/DnsPinger$DnsArg;->this$0:Landroid/net/DnsPinger;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 114
      iput-object p2, p0, Landroid/net/DnsPinger$DnsArg;->dns:Ljava/net/InetAddress;
  
+     .line 115
      iput p3, p0, Landroid/net/DnsPinger$DnsArg;->seq:I
  
+     .line 116
      return-void
  .end method
