*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/EthernetDataTracker.smali	2012-03-23 19:36:53.827662529 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/EthernetDataTracker.smali.original	2012-03-23 19:34:45.394320551 -0500
***************
*** 55,64 ****
--- 55,66 ----
      .locals 1
  
      .prologue
+     .line 58
      const-string v0, ""
  
      sput-object v0, Landroid/net/EthernetDataTracker;->sIfaceMatch:Ljava/lang/String;
  
+     .line 59
      const-string v0, ""
  
      sput-object v0, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
***************
*** 72,103 ****
--- 74,111 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 103
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 42
      new-instance v0, Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-direct {v0, v4}, Ljava/util/concurrent/atomic/AtomicBoolean;-><init>(Z)V
  
      iput-object v0, p0, Landroid/net/EthernetDataTracker;->mTeardownRequested:Ljava/util/concurrent/atomic/AtomicBoolean;
  
+     .line 43
      new-instance v0, Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-direct {v0, v4}, Ljava/util/concurrent/atomic/AtomicBoolean;-><init>(Z)V
  
      iput-object v0, p0, Landroid/net/EthernetDataTracker;->mPrivateDnsRouteSet:Ljava/util/concurrent/atomic/AtomicBoolean;
  
+     .line 44
      new-instance v0, Ljava/util/concurrent/atomic/AtomicInteger;
  
      invoke-direct {v0, v4}, Ljava/util/concurrent/atomic/AtomicInteger;-><init>(I)V
  
      iput-object v0, p0, Landroid/net/EthernetDataTracker;->mDefaultGatewayAddr:Ljava/util/concurrent/atomic/AtomicInteger;
  
+     .line 45
      new-instance v0, Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-direct {v0, v4}, Ljava/util/concurrent/atomic/AtomicBoolean;-><init>(Z)V
  
      iput-object v0, p0, Landroid/net/EthernetDataTracker;->mDefaultRouteSet:Ljava/util/concurrent/atomic/AtomicBoolean;
  
+     .line 104
      new-instance v0, Landroid/net/NetworkInfo;
  
      const/16 v1, 0x9
***************
*** 110,135 ****
--- 118,149 ----
  
      iput-object v0, p0, Landroid/net/EthernetDataTracker;->mNetworkInfo:Landroid/net/NetworkInfo;
  
+     .line 105
      new-instance v0, Landroid/net/LinkProperties;
  
      invoke-direct {v0}, Landroid/net/LinkProperties;-><init>()V
  
      iput-object v0, p0, Landroid/net/EthernetDataTracker;->mLinkProperties:Landroid/net/LinkProperties;
  
+     .line 106
      new-instance v0, Landroid/net/LinkCapabilities;
  
      invoke-direct {v0}, Landroid/net/LinkCapabilities;-><init>()V
  
      iput-object v0, p0, Landroid/net/EthernetDataTracker;->mLinkCapabilities:Landroid/net/LinkCapabilities;
  
+     .line 107
      sput-boolean v4, Landroid/net/EthernetDataTracker;->mLinkUp:Z
  
+     .line 109
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mNetworkInfo:Landroid/net/NetworkInfo;
  
      invoke-virtual {v0, v4}, Landroid/net/NetworkInfo;->setIsAvailable(Z)V
  
+     .line 110
      invoke-virtual {p0, v4}, Landroid/net/EthernetDataTracker;->setTeardownRequested(Z)V
  
+     .line 111
      return-void
  .end method
  
***************
*** 137,142 ****
--- 151,157 ----
      .locals 1
  
      .prologue
+     .line 38
      sget-object v0, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
  
      return-object v0
***************
*** 146,151 ****
--- 161,167 ----
      .locals 1
  
      .prologue
+     .line 38
      sget-boolean v0, Landroid/net/EthernetDataTracker;->mLinkUp:Z
  
      return v0
***************
*** 156,161 ****
--- 172,178 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      sput-boolean p0, Landroid/net/EthernetDataTracker;->mLinkUp:Z
  
      return p0
***************
*** 166,171 ****
--- 183,189 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mNetworkInfo:Landroid/net/NetworkInfo;
  
      return-object v0
***************
*** 177,182 ****
--- 195,201 ----
      .parameter "x1"
  
      .prologue
+     .line 38
      invoke-direct {p0, p1}, Landroid/net/EthernetDataTracker;->interfaceAdded(Ljava/lang/String;)V
  
      return-void
***************
*** 188,193 ****
--- 207,213 ----
      .parameter "x1"
  
      .prologue
+     .line 38
      invoke-direct {p0, p1}, Landroid/net/EthernetDataTracker;->interfaceRemoved(Ljava/lang/String;)V
  
      return-void
***************
*** 198,203 ****
--- 218,224 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mLinkProperties:Landroid/net/LinkProperties;
  
      return-object v0
***************
*** 209,214 ****
--- 230,236 ----
      .parameter "x1"
  
      .prologue
+     .line 38
      iput-object p1, p0, Landroid/net/EthernetDataTracker;->mLinkProperties:Landroid/net/LinkProperties;
  
      return-object p1
***************
*** 219,224 ****
--- 241,247 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mCsHandler:Landroid/os/Handler;
  
      return-object v0
***************
*** 228,233 ****
--- 251,257 ----
      .locals 2
  
      .prologue
+     .line 173
      const-class v1, Landroid/net/EthernetDataTracker;
  
      monitor-enter v1
***************
*** 243,248 ****
--- 267,273 ----
  
      sput-object v0, Landroid/net/EthernetDataTracker;->sInstance:Landroid/net/EthernetDataTracker;
  
+     .line 174
      :cond_0
      sget-object v0, Landroid/net/EthernetDataTracker;->sInstance:Landroid/net/EthernetDataTracker;
      :try_end_0
***************
*** 252,257 ****
--- 277,283 ----
  
      return-object v0
  
+     .line 173
      :catchall_0
      move-exception v0
  
***************
*** 265,270 ****
--- 291,297 ----
      .parameter "iface"
  
      .prologue
+     .line 114
      sget-object v1, Landroid/net/EthernetDataTracker;->sIfaceMatch:Ljava/lang/String;
  
      invoke-virtual {p1, v1}, Ljava/lang/String;->matches(Ljava/lang/String;)Z
***************
*** 273,281 ****
--- 300,310 ----
  
      if-nez v1, :cond_0
  
+     .line 130
      :goto_0
      return-void
  
+     .line 117
      :cond_0
      const-string v1, "Ethernet"
  
***************
*** 299,308 ****
--- 328,339 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 119
      sget-object v2, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
  
      monitor-enter v2
  
+     .line 120
      :try_start_0
      sget-object v1, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
  
***************
*** 312,321 ****
--- 343,354 ----
  
      if-nez v1, :cond_1
  
+     .line 121
      monitor-exit v2
  
      goto :goto_0
  
+     .line 123
      :catchall_0
      move-exception v1
  
***************
*** 325,344 ****
--- 358,381 ----
  
      throw v1
  
+     .line 122
      :cond_1
      :try_start_1
      sput-object p1, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
  
+     .line 123
      monitor-exit v2
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 125
      iget-object v1, p0, Landroid/net/EthernetDataTracker;->mNetworkInfo:Landroid/net/NetworkInfo;
  
      const/4 v2, 0x1
  
      invoke-virtual {v1, v2}, Landroid/net/NetworkInfo;->setIsAvailable(Z)V
  
+     .line 126
      iget-object v1, p0, Landroid/net/EthernetDataTracker;->mCsHandler:Landroid/os/Handler;
  
      const/4 v2, 0x3
***************
*** 349,357 ****
--- 386,396 ----
  
      move-result-object v0
  
+     .line 127
      .local v0, msg:Landroid/os/Message;
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 129
      invoke-direct {p0}, Landroid/net/EthernetDataTracker;->runDhcp()V
  
      goto :goto_0
***************
*** 364,369 ****
--- 403,409 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 133
      sget-object v1, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
  
      invoke-virtual {p1, v1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
***************
*** 372,380 ****
--- 412,422 ----
  
      if-nez v1, :cond_0
  
+     .line 151
      :goto_0
      return-void
  
+     .line 136
      :cond_0
      const-string v1, "Ethernet"
  
***************
*** 398,423 ****
--- 440,470 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 138
      sget-object v1, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
  
      invoke-static {v1}, Landroid/net/NetworkUtils;->stopDhcp(Ljava/lang/String;)Z
  
+     .line 140
      iget-object v1, p0, Landroid/net/EthernetDataTracker;->mLinkProperties:Landroid/net/LinkProperties;
  
      invoke-virtual {v1}, Landroid/net/LinkProperties;->clear()V
  
+     .line 141
      iget-object v1, p0, Landroid/net/EthernetDataTracker;->mNetworkInfo:Landroid/net/NetworkInfo;
  
      const/4 v2, 0x0
  
      invoke-virtual {v1, v2}, Landroid/net/NetworkInfo;->setIsAvailable(Z)V
  
+     .line 142
      iget-object v1, p0, Landroid/net/EthernetDataTracker;->mNetworkInfo:Landroid/net/NetworkInfo;
  
      sget-object v2, Landroid/net/NetworkInfo$DetailedState;->DISCONNECTED:Landroid/net/NetworkInfo$DetailedState;
  
      invoke-virtual {v1, v2, v4, v4}, Landroid/net/NetworkInfo;->setDetailedState(Landroid/net/NetworkInfo$DetailedState;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 144
      iget-object v1, p0, Landroid/net/EthernetDataTracker;->mCsHandler:Landroid/os/Handler;
  
      const/4 v2, 0x3
***************
*** 428,436 ****
--- 475,485 ----
  
      move-result-object v0
  
+     .line 145
      .local v0, msg:Landroid/os/Message;
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 147
      iget-object v1, p0, Landroid/net/EthernetDataTracker;->mCsHandler:Landroid/os/Handler;
  
      const/4 v2, 0x1
***************
*** 441,448 ****
--- 490,499 ----
  
      move-result-object v0
  
+     .line 148
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 150
      const-string v1, ""
  
      sput-object v1, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
***************
*** 454,459 ****
--- 505,511 ----
      .locals 2
  
      .prologue
+     .line 154
      new-instance v0, Ljava/lang/Thread;
  
      new-instance v1, Landroid/net/EthernetDataTracker$1;
***************
*** 462,470 ****
--- 514,524 ----
  
      invoke-direct {v0, v1}, Ljava/lang/Thread;-><init>(Ljava/lang/Runnable;)V
  
+     .line 169
      .local v0, dhcpThread:Ljava/lang/Thread;
      invoke-virtual {v0}, Ljava/lang/Thread;->start()V
  
+     .line 170
      return-void
  .end method
  
***************
*** 479,484 ****
--- 533,539 ----
      .end annotation
  
      .prologue
+     .line 178
      new-instance v0, Ljava/lang/CloneNotSupportedException;
  
      invoke-direct {v0}, Ljava/lang/CloneNotSupportedException;-><init>()V
***************
*** 491,500 ****
--- 546,557 ----
      .parameter "enabled"
  
      .prologue
+     .line 362
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mDefaultRouteSet:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-virtual {v0, p1}, Ljava/util/concurrent/atomic/AtomicBoolean;->set(Z)V
  
+     .line 363
      return-void
  .end method
  
***************
*** 502,507 ****
--- 559,565 ----
      .locals 1
  
      .prologue
+     .line 348
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mDefaultGatewayAddr:Ljava/util/concurrent/atomic/AtomicInteger;
  
      invoke-virtual {v0}, Ljava/util/concurrent/atomic/AtomicInteger;->get()I
***************
*** 515,520 ****
--- 573,579 ----
      .locals 2
  
      .prologue
+     .line 341
      new-instance v0, Landroid/net/LinkCapabilities;
  
      iget-object v1, p0, Landroid/net/EthernetDataTracker;->mLinkCapabilities:Landroid/net/LinkCapabilities;
***************
*** 528,533 ****
--- 587,593 ----
      .locals 2
  
      .prologue
+     .line 331
      monitor-enter p0
  
      :try_start_0
***************
*** 555,560 ****
--- 615,621 ----
      .locals 1
  
      .prologue
+     .line 324
      monitor-enter p0
  
      :try_start_0
***************
*** 578,584 ****
      .locals 1
  
      .prologue
!     const-string v0, "net.tcp.buffersize.wifi"
  
      return-object v0
  .end method
--- 639,646 ----
      .locals 1
  
      .prologue
!     .line 370
!     const-string/jumbo v0, "net.tcp.buffersize.wifi"
  
      return-object v0
  .end method
***************
*** 587,592 ****
--- 649,655 ----
      .locals 1
  
      .prologue
+     .line 259
      monitor-enter p0
  
      :try_start_0
***************
*** 614,619 ****
--- 677,683 ----
      .locals 1
  
      .prologue
+     .line 355
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mDefaultRouteSet:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-virtual {v0}, Ljava/util/concurrent/atomic/AtomicBoolean;->get()Z
***************
*** 627,632 ****
--- 691,697 ----
      .locals 1
  
      .prologue
+     .line 310
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mPrivateDnsRouteSet:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-virtual {v0}, Ljava/util/concurrent/atomic/AtomicBoolean;->get()Z
***************
*** 640,645 ****
--- 705,711 ----
      .locals 1
  
      .prologue
+     .line 186
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mTeardownRequested:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-virtual {v0}, Ljava/util/concurrent/atomic/AtomicBoolean;->get()Z
***************
*** 654,663 ****
--- 720,731 ----
      .parameter "enabled"
  
      .prologue
+     .line 317
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mPrivateDnsRouteSet:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-virtual {v0, p1}, Ljava/util/concurrent/atomic/AtomicBoolean;->set(Z)V
  
+     .line 318
      return-void
  .end method
  
***************
*** 665,678 ****
--- 733,749 ----
      .locals 2
  
      .prologue
+     .line 242
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mTeardownRequested:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, v1}, Ljava/util/concurrent/atomic/AtomicBoolean;->set(Z)V
  
+     .line 243
      invoke-direct {p0}, Landroid/net/EthernetDataTracker;->runDhcp()V
  
+     .line 244
      const/4 v0, 0x1
  
      return v0
***************
*** 683,688 ****
--- 754,760 ----
      .parameter "met"
  
      .prologue
+     .line 375
      return-void
  .end method
  
***************
*** 691,696 ****
--- 763,769 ----
      .parameter "enabled"
  
      .prologue
+     .line 303
      const-string v0, "Ethernet"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 719,724 ****
--- 792,798 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 304
      return-void
  .end method
  
***************
*** 727,732 ****
--- 801,807 ----
      .parameter "turnOn"
  
      .prologue
+     .line 252
      const/4 v0, 0x1
  
      return v0
***************
*** 737,746 ****
--- 812,823 ----
      .parameter "isRequested"
  
      .prologue
+     .line 182
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mTeardownRequested:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-virtual {v0, p1}, Ljava/util/concurrent/atomic/AtomicBoolean;->set(Z)V
  
+     .line 183
      return-void
  .end method
  
***************
*** 749,754 ****
--- 826,832 ----
      .parameter "enabled"
  
      .prologue
+     .line 298
      const-string v0, "Ethernet"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 777,782 ****
--- 855,861 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 299
      return-void
  .end method
  
***************
*** 786,806 ****
      .parameter "target"
  
      .prologue
      iput-object p1, p0, Landroid/net/EthernetDataTracker;->mContext:Landroid/content/Context;
  
      iput-object p2, p0, Landroid/net/EthernetDataTracker;->mCsHandler:Landroid/os/Handler;
  
!     const-string v9, "network_management"
  
      invoke-static {v9}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
      move-result-object v1
  
      .local v1, b:Landroid/os/IBinder;
      invoke-static {v1}, Landroid/os/INetworkManagementService$Stub;->asInterface(Landroid/os/IBinder;)Landroid/os/INetworkManagementService;
  
      move-result-object v8
  
      .local v8, service:Landroid/os/INetworkManagementService;
      new-instance v9, Landroid/net/EthernetDataTracker$InterfaceObserver;
  
--- 865,890 ----
      .parameter "target"
  
      .prologue
+     .line 193
      iput-object p1, p0, Landroid/net/EthernetDataTracker;->mContext:Landroid/content/Context;
  
+     .line 194
      iput-object p2, p0, Landroid/net/EthernetDataTracker;->mCsHandler:Landroid/os/Handler;
  
!     .line 197
!     const-string/jumbo v9, "network_management"
  
      invoke-static {v9}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
      move-result-object v1
  
+     .line 198
      .local v1, b:Landroid/os/IBinder;
      invoke-static {v1}, Landroid/os/INetworkManagementService$Stub;->asInterface(Landroid/os/IBinder;)Landroid/os/INetworkManagementService;
  
      move-result-object v8
  
+     .line 200
      .local v8, service:Landroid/os/INetworkManagementService;
      new-instance v9, Landroid/net/EthernetDataTracker$InterfaceObserver;
  
***************
*** 808,813 ****
--- 892,898 ----
  
      iput-object v9, p0, Landroid/net/EthernetDataTracker;->mInterfaceObserver:Landroid/net/EthernetDataTracker$InterfaceObserver;
  
+     .line 204
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v9
***************
*** 820,830 ****
--- 905,917 ----
  
      sput-object v9, Landroid/net/EthernetDataTracker;->sIfaceMatch:Ljava/lang/String;
  
+     .line 207
      :try_start_0
      invoke-interface {v8}, Landroid/os/INetworkManagementService;->listInterfaces()[Ljava/lang/String;
  
      move-result-object v6
  
+     .line 208
      .local v6, ifaces:[Ljava/lang/String;
      move-object v0, v6
  
***************
*** 840,845 ****
--- 927,933 ----
  
      aget-object v5, v0, v4
  
+     .line 209
      .local v5, iface:Ljava/lang/String;
      sget-object v9, Landroid/net/EthernetDataTracker;->sIfaceMatch:Ljava/lang/String;
  
***************
*** 849,860 ****
--- 937,951 ----
  
      if-eqz v9, :cond_1
  
+     .line 210
      sput-object v5, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
  
+     .line 211
      invoke-interface {v8, v5}, Landroid/os/INetworkManagementService;->getInterfaceConfig(Ljava/lang/String;)Landroid/net/InterfaceConfiguration;
  
      move-result-object v2
  
+     .line 212
      .local v2, config:Landroid/net/InterfaceConfiguration;
      invoke-virtual {v2}, Landroid/net/InterfaceConfiguration;->isActive()Z
  
***************
*** 862,871 ****
--- 953,964 ----
  
      sput-boolean v9, Landroid/net/EthernetDataTracker;->mLinkUp:Z
  
+     .line 213
      invoke-virtual {p0}, Landroid/net/EthernetDataTracker;->reconnect()Z
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 222
      .end local v0           #arr$:[Ljava/lang/String;
      .end local v2           #config:Landroid/net/InterfaceConfiguration;
      .end local v4           #i$:I
***************
*** 881,889 ****
--- 974,984 ----
      :try_end_1
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 226
      :goto_2
      return-void
  
+     .line 208
      .restart local v0       #arr$:[Ljava/lang/String;
      .restart local v4       #i$:I
      .restart local v5       #iface:Ljava/lang/String;
***************
*** 894,899 ****
--- 989,995 ----
  
      goto :goto_0
  
+     .line 217
      .end local v0           #arr$:[Ljava/lang/String;
      .end local v4           #i$:I
      .end local v5           #iface:Ljava/lang/String;
***************
*** 902,907 ****
--- 998,1004 ----
      :catch_0
      move-exception v3
  
+     .line 218
      .local v3, e:Landroid/os/RemoteException;
      const-string v9, "Ethernet"
  
***************
*** 927,936 ****
--- 1024,1035 ----
  
      goto :goto_1
  
+     .line 223
      .end local v3           #e:Landroid/os/RemoteException;
      :catch_1
      move-exception v3
  
+     .line 224
      .restart local v3       #e:Landroid/os/RemoteException;
      const-string v9, "Ethernet"
  
***************
*** 964,969 ****
--- 1063,1069 ----
      .parameter "callingUid"
  
      .prologue
+     .line 276
      const/4 v0, -0x1
  
      return v0
***************
*** 976,981 ****
--- 1076,1082 ----
      .parameter "callingUid"
  
      .prologue
+     .line 293
      const/4 v0, -0x1
  
      return v0
***************
*** 987,999 ****
--- 1088,1103 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 233
      iget-object v0, p0, Landroid/net/EthernetDataTracker;->mTeardownRequested:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-virtual {v0, v1}, Ljava/util/concurrent/atomic/AtomicBoolean;->set(Z)V
  
+     .line 234
      sget-object v0, Landroid/net/EthernetDataTracker;->mIface:Ljava/lang/String;
  
      invoke-static {v0}, Landroid/net/NetworkUtils;->stopDhcp(Ljava/lang/String;)Z
  
+     .line 235
      return v1
  .end method
