*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/INetworkManagementEventObserver$Stub.smali	2012-03-23 19:37:00.329781097 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/INetworkManagementEventObserver$Stub.smali.original	2012-03-23 19:34:45.477322065 -0500
***************
*** 42,53 ****
--- 42,56 ----
      .locals 1
  
      .prologue
+     .line 19
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 20
      const-string v0, "android.net.INetworkManagementEventObserver"
  
      invoke-virtual {p0, p0, v0}, Landroid/net/INetworkManagementEventObserver$Stub;->attachInterface(Landroid/os/IInterface;Ljava/lang/String;)V
  
+     .line 21
      return-void
  .end method
  
***************
*** 56,68 ****
--- 59,75 ----
      .parameter "obj"
  
      .prologue
+     .line 28
      if-nez p0, :cond_0
  
+     .line 29
      const/4 v0, 0x0
  
+     .line 35
      :goto_0
      return-object v0
  
+     .line 31
      :cond_0
      const-string v1, "android.net.INetworkManagementEventObserver"
  
***************
*** 70,75 ****
--- 77,83 ----
  
      move-result-object v0
  
+     .line 32
      .local v0, iin:Landroid/os/IInterface;
      if-eqz v0, :cond_1
  
***************
*** 77,86 ****
--- 85,96 ----
  
      if-eqz v1, :cond_1
  
+     .line 33
      check-cast v0, Landroid/net/INetworkManagementEventObserver;
  
      goto :goto_0
  
+     .line 35
      :cond_1
      new-instance v0, Landroid/net/INetworkManagementEventObserver$Stub$Proxy;
  
***************
*** 96,101 ****
--- 106,112 ----
      .locals 0
  
      .prologue
+     .line 39
      return-object p0
  .end method
  
***************
*** 116,123 ****
--- 127,136 ----
  
      const/4 v2, 0x1
  
+     .line 43
      sparse-switch p1, :sswitch_data_0
  
+     .line 102
      invoke-super {p0, p1, p2, p3, p4}, Landroid/os/Binder;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
      move-result v2
***************
*** 125,130 ****
--- 138,144 ----
      :goto_0
      return v2
  
+     .line 47
      :sswitch_0
      const-string v3, "android.net.INetworkManagementEventObserver"
  
***************
*** 132,146 ****
--- 146,163 ----
  
      goto :goto_0
  
+     .line 52
      :sswitch_1
      const-string v3, "android.net.INetworkManagementEventObserver"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 54
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 56
      .local v0, _arg0:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 150,163 ****
--- 167,183 ----
  
      move v1, v2
  
+     .line 57
      .local v1, _arg1:Z
      :cond_0
      invoke-virtual {p0, v0, v1}, Landroid/net/INetworkManagementEventObserver$Stub;->interfaceStatusChanged(Ljava/lang/String;Z)V
  
+     .line 58
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto :goto_0
  
+     .line 63
      .end local v0           #_arg0:Ljava/lang/String;
      .end local v1           #_arg1:Z
      :sswitch_2
***************
*** 165,174 ****
--- 185,196 ----
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 65
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 67
      .restart local v0       #_arg0:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 178,191 ****
--- 200,216 ----
  
      move v1, v2
  
+     .line 68
      .restart local v1       #_arg1:Z
      :cond_1
      invoke-virtual {p0, v0, v1}, Landroid/net/INetworkManagementEventObserver$Stub;->interfaceLinkStateChanged(Ljava/lang/String;Z)V
  
+     .line 69
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto :goto_0
  
+     .line 74
      .end local v0           #_arg0:Ljava/lang/String;
      .end local v1           #_arg1:Z
      :sswitch_3
***************
*** 193,248 ****
--- 218,286 ----
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 76
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 77
      .restart local v0       #_arg0:Ljava/lang/String;
      invoke-virtual {p0, v0}, Landroid/net/INetworkManagementEventObserver$Stub;->interfaceAdded(Ljava/lang/String;)V
  
+     .line 78
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto :goto_0
  
+     .line 83
      .end local v0           #_arg0:Ljava/lang/String;
      :sswitch_4
      const-string v3, "android.net.INetworkManagementEventObserver"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 85
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 86
      .restart local v0       #_arg0:Ljava/lang/String;
      invoke-virtual {p0, v0}, Landroid/net/INetworkManagementEventObserver$Stub;->interfaceRemoved(Ljava/lang/String;)V
  
+     .line 87
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto :goto_0
  
+     .line 92
      .end local v0           #_arg0:Ljava/lang/String;
      :sswitch_5
      const-string v3, "android.net.INetworkManagementEventObserver"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 94
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 96
      .restart local v0       #_arg0:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 97
      .local v1, _arg1:Ljava/lang/String;
      invoke-virtual {p0, v0, v1}, Landroid/net/INetworkManagementEventObserver$Stub;->limitReached(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 98
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto :goto_0
  
+     .line 43
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_1
