*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/InterfaceConfiguration$1.smali	2012-03-23 19:36:59.602767840 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/InterfaceConfiguration$1.smali.original	2012-03-23 19:34:45.469321919 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 88
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,53 ****
--- 45,56 ----
      .parameter "in"
  
      .prologue
+     .line 90
      new-instance v0, Landroid/net/InterfaceConfiguration;
  
      invoke-direct {v0}, Landroid/net/InterfaceConfiguration;-><init>()V
  
+     .line 91
      .local v0, info:Landroid/net/InterfaceConfiguration;
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
***************
*** 55,60 ****
--- 58,64 ----
  
      iput-object v1, v0, Landroid/net/InterfaceConfiguration;->hwAddr:Ljava/lang/String;
  
+     .line 92
      invoke-virtual {p1}, Landroid/os/Parcel;->readByte()B
  
      move-result v1
***************
*** 63,68 ****
--- 67,73 ----
  
      if-ne v1, v2, :cond_0
  
+     .line 93
      const/4 v1, 0x0
  
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
***************
*** 73,78 ****
--- 78,84 ----
  
      iput-object v1, v0, Landroid/net/InterfaceConfiguration;->addr:Landroid/net/LinkAddress;
  
+     .line 95
      :cond_0
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
***************
*** 80,85 ****
--- 86,92 ----
  
      iput-object v1, v0, Landroid/net/InterfaceConfiguration;->interfaceFlags:Ljava/lang/String;
  
+     .line 96
      return-object v0
  .end method
  
***************
*** 88,93 ****
--- 95,101 ----
      .parameter "x0"
  
      .prologue
+     .line 88
      invoke-virtual {p0, p1}, Landroid/net/InterfaceConfiguration$1;->createFromParcel(Landroid/os/Parcel;)Landroid/net/InterfaceConfiguration;
  
      move-result-object v0
***************
*** 100,105 ****
--- 108,114 ----
      .parameter "size"
  
      .prologue
+     .line 100
      new-array v0, p1, [Landroid/net/InterfaceConfiguration;
  
      return-object v0
***************
*** 110,115 ****
--- 119,125 ----
      .parameter "x0"
  
      .prologue
+     .line 88
      invoke-virtual {p0, p1}, Landroid/net/InterfaceConfiguration$1;->newArray(I)[Landroid/net/InterfaceConfiguration;
  
      move-result-object v0
