*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LinkCapabilities$Key.smali	2012-03-23 19:37:00.306780677 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LinkCapabilities$Key.smali.original	2012-03-23 19:34:45.477322065 -0500
***************
*** 41,46 ****
--- 41,47 ----
      .locals 0
  
      .prologue
+     .line 56
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
