*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LinkAddress$1.smali	2012-03-23 19:36:53.197651042 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LinkAddress$1.smali.original	2012-03-23 19:34:45.387320423 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 130
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,54 ****
--- 45,58 ----
      .parameter "in"
  
      .prologue
+     .line 132
      const/4 v0, 0x0
  
+     .line 133
      .local v0, address:Ljava/net/InetAddress;
      const/4 v1, 0x0
  
+     .line 134
      .local v1, prefixLength:I
      invoke-virtual {p1}, Landroid/os/Parcel;->readByte()B
  
***************
*** 58,63 ****
--- 62,68 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 136
      :try_start_0
      invoke-virtual {p1}, Landroid/os/Parcel;->createByteArray()[B
  
***************
*** 67,78 ****
--- 72,85 ----
  
      move-result-object v0
  
+     .line 137
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catch Ljava/net/UnknownHostException; {:try_start_0 .. :try_end_0} :catch_0
  
      move-result v1
  
+     .line 140
      :cond_0
      :goto_0
      new-instance v2, Landroid/net/LinkAddress;
***************
*** 81,86 ****
--- 88,94 ----
  
      return-object v2
  
+     .line 138
      :catch_0
      move-exception v2
  
***************
*** 92,97 ****
--- 100,106 ----
      .parameter "x0"
  
      .prologue
+     .line 130
      invoke-virtual {p0, p1}, Landroid/net/LinkAddress$1;->createFromParcel(Landroid/os/Parcel;)Landroid/net/LinkAddress;
  
      move-result-object v0
***************
*** 104,109 ****
--- 113,119 ----
      .parameter "size"
  
      .prologue
+     .line 144
      new-array v0, p1, [Landroid/net/LinkAddress;
  
      return-object v0
***************
*** 114,119 ****
--- 124,130 ----
      .parameter "x0"
  
      .prologue
+     .line 130
      invoke-virtual {p0, p1}, Landroid/net/LinkAddress$1;->newArray(I)[Landroid/net/LinkAddress;
  
      move-result-object v0
