*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LinkProperties$1.smali	2012-03-23 19:37:00.157777960 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LinkProperties$1.smali.original	2012-03-23 19:34:45.475322029 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 400
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 46,73 ****
--- 47,80 ----
      .prologue
      const/4 v6, 0x0
  
+     .line 402
      new-instance v4, Landroid/net/LinkProperties;
  
      invoke-direct {v4}, Landroid/net/LinkProperties;-><init>()V
  
+     .line 403
      .local v4, netProp:Landroid/net/LinkProperties;
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 404
      .local v3, iface:Ljava/lang/String;
      if-eqz v3, :cond_0
  
+     .line 406
      :try_start_0
      invoke-virtual {v4, v3}, Landroid/net/LinkProperties;->setInterfaceName(Ljava/lang/String;)V
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 411
      :cond_0
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 412
      .local v0, addressCount:I
      const/4 v2, 0x0
  
***************
*** 75,80 ****
--- 82,88 ----
      :goto_0
      if-ge v2, v0, :cond_2
  
+     .line 413
      invoke-virtual {p1, v6}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
  
      move-result-object v5
***************
*** 83,92 ****
--- 91,102 ----
  
      invoke-virtual {v4, v5}, Landroid/net/LinkProperties;->addLinkAddress(Landroid/net/LinkAddress;)V
  
+     .line 412
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 407
      .end local v0           #addressCount:I
      .end local v2           #i:I
      :catch_0
***************
*** 95,106 ****
--- 105,118 ----
      .local v1, e:Ljava/lang/Exception;
      move-object v4, v6
  
+     .line 428
      .end local v1           #e:Ljava/lang/Exception;
      .end local v4           #netProp:Landroid/net/LinkProperties;
      :cond_1
      :goto_1
      return-object v4
  
+     .line 415
      .restart local v0       #addressCount:I
      .restart local v2       #i:I
      .restart local v4       #netProp:Landroid/net/LinkProperties;
***************
*** 109,119 ****
--- 121,133 ----
  
      move-result v0
  
+     .line 416
      const/4 v2, 0x0
  
      :goto_2
      if-ge v2, v0, :cond_3
  
+     .line 418
      :try_start_1
      invoke-virtual {p1}, Landroid/os/Parcel;->createByteArray()[B
  
***************
*** 127,147 ****
--- 141,165 ----
      :try_end_1
      .catch Ljava/net/UnknownHostException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 416
      :goto_3
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_2
  
+     .line 421
      :cond_3
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 422
      const/4 v2, 0x0
  
      :goto_4
      if-ge v2, v0, :cond_4
  
+     .line 423
      invoke-virtual {p1, v6}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
  
      move-result-object v5
***************
*** 150,159 ****
--- 168,179 ----
  
      invoke-virtual {v4, v5}, Landroid/net/LinkProperties;->addRoute(Landroid/net/RouteInfo;)V
  
+     .line 422
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_4
  
+     .line 425
      :cond_4
      invoke-virtual {p1}, Landroid/os/Parcel;->readByte()B
  
***************
*** 163,168 ****
--- 183,189 ----
  
      if-ne v5, v7, :cond_1
  
+     .line 426
      invoke-virtual {p1, v6}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
  
      move-result-object v5
***************
*** 173,178 ****
--- 194,200 ----
  
      goto :goto_1
  
+     .line 419
      :catch_1
      move-exception v5
  
***************
*** 184,189 ****
--- 206,212 ----
      .parameter "x0"
  
      .prologue
+     .line 400
      invoke-virtual {p0, p1}, Landroid/net/LinkProperties$1;->createFromParcel(Landroid/os/Parcel;)Landroid/net/LinkProperties;
  
      move-result-object v0
***************
*** 196,201 ****
--- 219,225 ----
      .parameter "size"
  
      .prologue
+     .line 432
      new-array v0, p1, [Landroid/net/LinkProperties;
  
      return-object v0
***************
*** 206,211 ****
--- 230,236 ----
      .parameter "x0"
  
      .prologue
+     .line 400
      invoke-virtual {p0, p1}, Landroid/net/LinkProperties$1;->newArray(I)[Landroid/net/LinkProperties;
  
      move-result-object v0
