*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LocalSocketImpl$SocketOutputStream.smali	2012-03-23 19:36:55.050684831 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LocalSocketImpl$SocketOutputStream.smali.original	2012-03-23 19:34:45.408320807 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 103
      iput-object p1, p0, Landroid/net/LocalSocketImpl$SocketOutputStream;->this$0:Landroid/net/LocalSocketImpl;
  
      invoke-direct {p0}, Ljava/io/OutputStream;-><init>()V
***************
*** 42,51 ****
--- 43,54 ----
      .end annotation
  
      .prologue
+     .line 107
      iget-object v0, p0, Landroid/net/LocalSocketImpl$SocketOutputStream;->this$0:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->close()V
  
+     .line 108
      return-void
  .end method
  
***************
*** 59,64 ****
--- 62,68 ----
      .end annotation
  
      .prologue
+     .line 133
      iget-object v1, p0, Landroid/net/LocalSocketImpl$SocketOutputStream;->this$0:Landroid/net/LocalSocketImpl;
  
      #getter for: Landroid/net/LocalSocketImpl;->writeMonitor:Ljava/lang/Object;
***************
*** 68,73 ****
--- 72,78 ----
  
      monitor-enter v2
  
+     .line 134
      :try_start_0
      iget-object v1, p0, Landroid/net/LocalSocketImpl$SocketOutputStream;->this$0:Landroid/net/LocalSocketImpl;
  
***************
*** 76,92 ****
  
      move-result-object v0
  
      .local v0, myFd:Ljava/io/FileDescriptor;
      if-nez v0, :cond_0
  
      new-instance v1, Ljava/io/IOException;
  
!     const-string v3, "socket closed"
  
      invoke-direct {v1, v3}, Ljava/io/IOException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
      .end local v0           #myFd:Ljava/io/FileDescriptor;
      :catchall_0
      move-exception v1
--- 81,99 ----
  
      move-result-object v0
  
+     .line 135
      .local v0, myFd:Ljava/io/FileDescriptor;
      if-nez v0, :cond_0
  
      new-instance v1, Ljava/io/IOException;
  
!     const-string/jumbo v3, "socket closed"
  
      invoke-direct {v1, v3}, Ljava/io/IOException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
+     .line 137
      .end local v0           #myFd:Ljava/io/FileDescriptor;
      :catchall_0
      move-exception v1
***************
*** 97,102 ****
--- 104,110 ----
  
      throw v1
  
+     .line 136
      .restart local v0       #myFd:Ljava/io/FileDescriptor;
      :cond_0
      :try_start_1
***************
*** 105,114 ****
--- 113,124 ----
      #calls: Landroid/net/LocalSocketImpl;->write_native(ILjava/io/FileDescriptor;)V
      invoke-static {v1, p1, v0}, Landroid/net/LocalSocketImpl;->access$700(Landroid/net/LocalSocketImpl;ILjava/io/FileDescriptor;)V
  
+     .line 137
      monitor-exit v2
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 138
      return-void
  .end method
  
***************
*** 122,133 ****
--- 132,145 ----
      .end annotation
  
      .prologue
+     .line 113
      const/4 v0, 0x0
  
      array-length v1, p1
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/net/LocalSocketImpl$SocketOutputStream;->write([BII)V
  
+     .line 114
      return-void
  .end method
  
***************
*** 143,148 ****
--- 155,161 ----
      .end annotation
  
      .prologue
+     .line 119
      iget-object v1, p0, Landroid/net/LocalSocketImpl$SocketOutputStream;->this$0:Landroid/net/LocalSocketImpl;
  
      #getter for: Landroid/net/LocalSocketImpl;->writeMonitor:Ljava/lang/Object;
***************
*** 152,157 ****
--- 165,171 ----
  
      monitor-enter v2
  
+     .line 120
      :try_start_0
      iget-object v1, p0, Landroid/net/LocalSocketImpl$SocketOutputStream;->this$0:Landroid/net/LocalSocketImpl;
  
***************
*** 160,176 ****
  
      move-result-object v0
  
      .local v0, myFd:Ljava/io/FileDescriptor;
      if-nez v0, :cond_0
  
      new-instance v1, Ljava/io/IOException;
  
!     const-string v3, "socket closed"
  
      invoke-direct {v1, v3}, Ljava/io/IOException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
      .end local v0           #myFd:Ljava/io/FileDescriptor;
      :catchall_0
      move-exception v1
--- 174,192 ----
  
      move-result-object v0
  
+     .line 121
      .local v0, myFd:Ljava/io/FileDescriptor;
      if-nez v0, :cond_0
  
      new-instance v1, Ljava/io/IOException;
  
!     const-string/jumbo v3, "socket closed"
  
      invoke-direct {v1, v3}, Ljava/io/IOException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
+     .line 127
      .end local v0           #myFd:Ljava/io/FileDescriptor;
      :catchall_0
      move-exception v1
***************
*** 181,186 ****
--- 197,203 ----
  
      throw v1
  
+     .line 123
      .restart local v0       #myFd:Ljava/io/FileDescriptor;
      :cond_0
      if-ltz p2, :cond_1
***************
*** 194,199 ****
--- 211,217 ----
  
      if-le v1, v3, :cond_2
  
+     .line 124
      :cond_1
      new-instance v1, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 201,215 ****
--- 219,236 ----
  
      throw v1
  
+     .line 126
      :cond_2
      iget-object v1, p0, Landroid/net/LocalSocketImpl$SocketOutputStream;->this$0:Landroid/net/LocalSocketImpl;
  
      #calls: Landroid/net/LocalSocketImpl;->writeba_native([BIILjava/io/FileDescriptor;)V
      invoke-static {v1, p1, p2, p3, v0}, Landroid/net/LocalSocketImpl;->access$600(Landroid/net/LocalSocketImpl;[BIILjava/io/FileDescriptor;)V
  
+     .line 127
      monitor-exit v2
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 128
      return-void
  .end method
