*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LocalSocket.smali	2012-03-23 19:37:00.206778853 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LocalSocket.smali.original	2012-03-23 19:34:45.476322047 -0500
***************
*** 22,37 ****
--- 22,41 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 41
      new-instance v0, Landroid/net/LocalSocketImpl;
  
      invoke-direct {v0}, Landroid/net/LocalSocketImpl;-><init>()V
  
      invoke-direct {p0, v0}, Landroid/net/LocalSocket;-><init>(Landroid/net/LocalSocketImpl;)V
  
+     .line 42
      iput-boolean v1, p0, Landroid/net/LocalSocket;->isBound:Z
  
+     .line 43
      iput-boolean v1, p0, Landroid/net/LocalSocket;->isConnected:Z
  
+     .line 44
      return-void
  .end method
  
***************
*** 42,55 ****
--- 46,64 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 50
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 51
      iput-object p1, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
+     .line 52
      iput-boolean v0, p0, Landroid/net/LocalSocket;->isConnected:Z
  
+     .line 53
      iput-boolean v0, p0, Landroid/net/LocalSocket;->isBound:Z
  
+     .line 54
      return-void
  .end method
  
***************
*** 62,73 ****
--- 71,85 ----
      .end annotation
  
      .prologue
+     .line 70
      iget-boolean v0, p0, Landroid/net/LocalSocket;->implCreated:Z
  
      if-nez v0, :cond_1
  
+     .line 71
      monitor-enter p0
  
+     .line 72
      :try_start_0
      iget-boolean v0, p0, Landroid/net/LocalSocket;->implCreated:Z
      :try_end_0
***************
*** 75,80 ****
--- 87,93 ----
  
      if-nez v0, :cond_0
  
+     .line 74
      :try_start_1
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
***************
*** 84,100 ****
--- 97,117 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 76
      const/4 v0, 0x1
  
      :try_start_2
      iput-boolean v0, p0, Landroid/net/LocalSocket;->implCreated:Z
  
+     .line 79
      :cond_0
      monitor-exit p0
  
+     .line 81
      :cond_1
      return-void
  
+     .line 76
      :catchall_0
      move-exception v0
  
***************
*** 104,109 ****
--- 121,127 ----
  
      throw v0
  
+     .line 79
      :catchall_1
      move-exception v0
  
***************
*** 126,140 ****
--- 144,162 ----
      .end annotation
  
      .prologue
+     .line 112
      invoke-direct {p0}, Landroid/net/LocalSocket;->implCreateIfNeeded()V
  
+     .line 114
      monitor-enter p0
  
+     .line 115
      :try_start_0
      iget-boolean v0, p0, Landroid/net/LocalSocket;->isBound:Z
  
      if-eqz v0, :cond_0
  
+     .line 116
      new-instance v0, Ljava/io/IOException;
  
      const-string v1, "already bound"
***************
*** 143,148 ****
--- 165,171 ----
  
      throw v0
  
+     .line 122
      :catchall_0
      move-exception v0
  
***************
*** 152,175 ****
--- 175,203 ----
  
      throw v0
  
+     .line 119
      :cond_0
      :try_start_1
      iput-object p1, p0, Landroid/net/LocalSocket;->localAddress:Landroid/net/LocalSocketAddress;
  
+     .line 120
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      iget-object v1, p0, Landroid/net/LocalSocket;->localAddress:Landroid/net/LocalSocketAddress;
  
      invoke-virtual {v0, v1}, Landroid/net/LocalSocketImpl;->bind(Landroid/net/LocalSocketAddress;)V
  
+     .line 121
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/net/LocalSocket;->isBound:Z
  
+     .line 122
      monitor-exit p0
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 123
      return-void
  .end method
  
***************
*** 182,193 ****
--- 210,224 ----
      .end annotation
  
      .prologue
+     .line 162
      invoke-direct {p0}, Landroid/net/LocalSocket;->implCreateIfNeeded()V
  
+     .line 163
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->close()V
  
+     .line 164
      return-void
  .end method
  
***************
*** 201,213 ****
--- 232,247 ----
      .end annotation
  
      .prologue
+     .line 92
      monitor-enter p0
  
+     .line 93
      :try_start_0
      iget-boolean v0, p0, Landroid/net/LocalSocket;->isConnected:Z
  
      if-eqz v0, :cond_0
  
+     .line 94
      new-instance v0, Ljava/io/IOException;
  
      const-string v1, "already connected"
***************
*** 216,221 ****
--- 250,256 ----
  
      throw v0
  
+     .line 101
      :catchall_0
      move-exception v0
  
***************
*** 225,252 ****
--- 260,293 ----
  
      throw v0
  
+     .line 97
      :cond_0
      :try_start_1
      invoke-direct {p0}, Landroid/net/LocalSocket;->implCreateIfNeeded()V
  
+     .line 98
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, p1, v1}, Landroid/net/LocalSocketImpl;->connect(Landroid/net/LocalSocketAddress;I)V
  
+     .line 99
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/net/LocalSocket;->isConnected:Z
  
+     .line 100
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/net/LocalSocket;->isBound:Z
  
+     .line 101
      monitor-exit p0
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 102
      return-void
  .end method
  
***************
*** 261,266 ****
--- 302,308 ----
      .end annotation
  
      .prologue
+     .line 243
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 277,282 ****
--- 319,325 ----
      .end annotation
  
      .prologue
+     .line 269
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->getAncillaryFileDescriptors()[Ljava/io/FileDescriptor;
***************
*** 290,295 ****
--- 333,339 ----
      .locals 1
  
      .prologue
+     .line 289
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->getFileDescriptor()Ljava/io/FileDescriptor;
***************
*** 308,315 ****
--- 352,361 ----
      .end annotation
  
      .prologue
+     .line 141
      invoke-direct {p0}, Landroid/net/LocalSocket;->implCreateIfNeeded()V
  
+     .line 142
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->getInputStream()Ljava/io/InputStream;
***************
*** 323,328 ****
--- 369,375 ----
      .locals 1
  
      .prologue
+     .line 131
      iget-object v0, p0, Landroid/net/LocalSocket;->localAddress:Landroid/net/LocalSocketAddress;
  
      return-object v0
***************
*** 337,344 ****
--- 384,393 ----
      .end annotation
  
      .prologue
+     .line 152
      invoke-direct {p0}, Landroid/net/LocalSocket;->implCreateIfNeeded()V
  
+     .line 153
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->getOutputStream()Ljava/io/OutputStream;
***************
*** 357,362 ****
--- 406,412 ----
      .end annotation
  
      .prologue
+     .line 280
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->getPeerCredentials()Landroid/net/Credentials;
***************
*** 375,380 ****
--- 425,431 ----
      .end annotation
  
      .prologue
+     .line 191
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/16 v1, 0x1002
***************
*** 396,401 ****
--- 447,453 ----
      .locals 1
  
      .prologue
+     .line 212
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 412,417 ****
--- 464,470 ----
      .end annotation
  
      .prologue
+     .line 207
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/16 v1, 0x1001
***************
*** 438,443 ****
--- 491,497 ----
      .end annotation
  
      .prologue
+     .line 199
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/16 v1, 0x1006
***************
*** 459,464 ****
--- 513,519 ----
      .locals 1
  
      .prologue
+     .line 227
      monitor-enter p0
  
      :try_start_0
***************
*** 482,487 ****
--- 537,543 ----
      .locals 1
  
      .prologue
+     .line 222
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 493,498 ****
--- 549,555 ----
      .locals 1
  
      .prologue
+     .line 217
      monitor-enter p0
  
      :try_start_0
***************
*** 516,521 ****
--- 573,579 ----
      .locals 1
  
      .prologue
+     .line 237
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 527,532 ****
--- 585,591 ----
      .locals 1
  
      .prologue
+     .line 232
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 539,548 ****
--- 598,609 ----
      .parameter "fds"
  
      .prologue
+     .line 255
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0, p1}, Landroid/net/LocalSocketImpl;->setFileDescriptorsForSend([Ljava/io/FileDescriptor;)V
  
+     .line 256
      return-void
  .end method
  
***************
*** 556,561 ****
--- 617,623 ----
      .end annotation
  
      .prologue
+     .line 187
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/16 v1, 0x1002
***************
*** 566,571 ****
--- 628,634 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/net/LocalSocketImpl;->setOption(ILjava/lang/Object;)V
  
+     .line 188
      return-void
  .end method
  
***************
*** 579,584 ****
--- 642,648 ----
      .end annotation
  
      .prologue
+     .line 203
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/16 v1, 0x1001
***************
*** 589,594 ****
--- 653,659 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/net/LocalSocketImpl;->setOption(ILjava/lang/Object;)V
  
+     .line 204
      return-void
  .end method
  
***************
*** 602,607 ****
--- 667,673 ----
      .end annotation
  
      .prologue
+     .line 195
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/16 v1, 0x1006
***************
*** 612,617 ****
--- 678,684 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/net/LocalSocketImpl;->setOption(ILjava/lang/Object;)V
  
+     .line 196
      return-void
  .end method
  
***************
*** 624,635 ****
--- 691,705 ----
      .end annotation
  
      .prologue
+     .line 172
      invoke-direct {p0}, Landroid/net/LocalSocket;->implCreateIfNeeded()V
  
+     .line 173
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->shutdownInput()V
  
+     .line 174
      return-void
  .end method
  
***************
*** 642,653 ****
--- 712,726 ----
      .end annotation
  
      .prologue
+     .line 182
      invoke-direct {p0}, Landroid/net/LocalSocket;->implCreateIfNeeded()V
  
+     .line 183
      iget-object v0, p0, Landroid/net/LocalSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->shutdownOutput()V
  
+     .line 184
      return-void
  .end method
  
***************
*** 655,660 ****
--- 728,734 ----
      .locals 2
  
      .prologue
+     .line 59
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
