*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LocalServerSocket.smali	2012-03-23 19:36:52.485638058 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/LocalServerSocket.smali.original	2012-03-23 19:34:45.375320205 -0500
***************
*** 24,43 ****
--- 24,47 ----
      .end annotation
  
      .prologue
+     .line 67
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 68
      new-instance v0, Landroid/net/LocalSocketImpl;
  
      invoke-direct {v0, p1}, Landroid/net/LocalSocketImpl;-><init>(Ljava/io/FileDescriptor;)V
  
      iput-object v0, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
+     .line 69
      iget-object v0, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/16 v1, 0x32
  
      invoke-virtual {v0, v1}, Landroid/net/LocalSocketImpl;->listen(I)V
  
+     .line 70
      iget-object v0, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->getSockAddress()Landroid/net/LocalSocketAddress;
***************
*** 46,51 ****
--- 50,56 ----
  
      iput-object v0, p0, Landroid/net/LocalServerSocket;->localAddress:Landroid/net/LocalSocketAddress;
  
+     .line 71
      return-void
  .end method
  
***************
*** 59,96 ****
--- 64,108 ----
      .end annotation
  
      .prologue
+     .line 46
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 47
      new-instance v0, Landroid/net/LocalSocketImpl;
  
      invoke-direct {v0}, Landroid/net/LocalSocketImpl;-><init>()V
  
      iput-object v0, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
+     .line 49
      iget-object v0, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/net/LocalSocketImpl;->create(Z)V
  
+     .line 51
      new-instance v0, Landroid/net/LocalSocketAddress;
  
      invoke-direct {v0, p1}, Landroid/net/LocalSocketAddress;-><init>(Ljava/lang/String;)V
  
      iput-object v0, p0, Landroid/net/LocalServerSocket;->localAddress:Landroid/net/LocalSocketAddress;
  
+     .line 52
      iget-object v0, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
      iget-object v1, p0, Landroid/net/LocalServerSocket;->localAddress:Landroid/net/LocalSocketAddress;
  
      invoke-virtual {v0, v1}, Landroid/net/LocalSocketImpl;->bind(Landroid/net/LocalSocketAddress;)V
  
+     .line 54
      iget-object v0, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
      const/16 v1, 0x32
  
      invoke-virtual {v0, v1}, Landroid/net/LocalSocketImpl;->listen(I)V
  
+     .line 55
      return-void
  .end method
  
***************
*** 105,119 ****
--- 117,134 ----
      .end annotation
  
      .prologue
+     .line 92
      new-instance v0, Landroid/net/LocalSocketImpl;
  
      invoke-direct {v0}, Landroid/net/LocalSocketImpl;-><init>()V
  
+     .line 94
      .local v0, acceptedImpl:Landroid/net/LocalSocketImpl;
      iget-object v1, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v1, v0}, Landroid/net/LocalSocketImpl;->accept(Landroid/net/LocalSocketImpl;)V
  
+     .line 96
      new-instance v1, Landroid/net/LocalSocket;
  
      invoke-direct {v1, v0}, Landroid/net/LocalSocket;-><init>(Landroid/net/LocalSocketImpl;)V
***************
*** 130,139 ****
--- 145,156 ----
      .end annotation
  
      .prologue
+     .line 115
      iget-object v0, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->close()V
  
+     .line 116
      return-void
  .end method
  
***************
*** 141,146 ****
--- 158,164 ----
      .locals 1
  
      .prologue
+     .line 105
      iget-object v0, p0, Landroid/net/LocalServerSocket;->impl:Landroid/net/LocalSocketImpl;
  
      invoke-virtual {v0}, Landroid/net/LocalSocketImpl;->getFileDescriptor()Ljava/io/FileDescriptor;
***************
*** 154,159 ****
--- 172,178 ----
      .locals 1
  
      .prologue
+     .line 80
      iget-object v0, p0, Landroid/net/LocalServerSocket;->localAddress:Landroid/net/LocalSocketAddress;
  
      return-object v0
