*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/MailTo.smali	2012-03-23 19:36:54.318671484 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/MailTo.smali.original	2012-03-23 19:34:45.400320661 -0500
***************
*** 34,47 ****
--- 34,50 ----
      .locals 1
  
      .prologue
+     .line 169
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 170
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
+     .line 171
      return-void
  .end method
  
***************
*** 50,58 ****
      .parameter "url"
  
      .prologue
      if-eqz p0, :cond_0
  
!     const-string v0, "mailto:"
  
      invoke-virtual {p0, v0}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
  
--- 53,62 ----
      .parameter "url"
  
      .prologue
+     .line 51
      if-eqz p0, :cond_0
  
!     const-string/jumbo v0, "mailto:"
  
      invoke-virtual {p0, v0}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
  
***************
*** 60,67 ****
--- 64,73 ----
  
      if-eqz v0, :cond_0
  
+     .line 52
      const/4 v0, 0x1
  
+     .line 54
      :goto_0
      return v0
  
***************
*** 81,94 ****
--- 87,103 ----
      .end annotation
  
      .prologue
+     .line 65
      if-nez p0, :cond_0
  
+     .line 66
      new-instance v13, Ljava/lang/NullPointerException;
  
      invoke-direct {v13}, Ljava/lang/NullPointerException;-><init>()V
  
      throw v13
  
+     .line 68
      :cond_0
      invoke-static/range {p0 .. p0}, Landroid/net/MailTo;->isMailTo(Ljava/lang/String;)Z
  
***************
*** 96,101 ****
--- 105,111 ----
  
      if-nez v13, :cond_1
  
+     .line 69
      new-instance v13, Landroid/net/ParseException;
  
      const-string v14, "Not a mailto scheme"
***************
*** 104,111 ****
  
      throw v13
  
      :cond_1
!     const-string v13, "mailto:"
  
      invoke-virtual {v13}, Ljava/lang/String;->length()I
  
--- 114,122 ----
  
      throw v13
  
+     .line 72
      :cond_1
!     const-string/jumbo v13, "mailto:"
  
      invoke-virtual {v13}, Ljava/lang/String;->length()I
  
***************
*** 117,146 ****
--- 128,163 ----
  
      move-result-object v9
  
+     .line 73
      .local v9, noScheme:Ljava/lang/String;
      invoke-static {v9}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v4
  
+     .line 74
      .local v4, email:Landroid/net/Uri;
      new-instance v7, Landroid/net/MailTo;
  
      invoke-direct {v7}, Landroid/net/MailTo;-><init>()V
  
+     .line 77
      .local v7, m:Landroid/net/MailTo;
      invoke-virtual {v4}, Landroid/net/Uri;->getQuery()Ljava/lang/String;
  
      move-result-object v12
  
+     .line 78
      .local v12, query:Ljava/lang/String;
      if-eqz v12, :cond_4
  
+     .line 79
      const-string v13, "&"
  
      invoke-virtual {v12, v13}, Ljava/lang/String;->split(Ljava/lang/String;)[Ljava/lang/String;
  
      move-result-object v11
  
+     .line 80
      .local v11, queries:[Ljava/lang/String;
      move-object v3, v11
  
***************
*** 156,161 ****
--- 173,179 ----
  
      aget-object v10, v3, v5
  
+     .line 81
      .local v10, q:Ljava/lang/String;
      const-string v13, "="
  
***************
*** 163,178 ****
--- 181,199 ----
  
      move-result-object v8
  
+     .line 82
      .local v8, nameval:[Ljava/lang/String;
      array-length v13, v8
  
      if-nez v13, :cond_2
  
+     .line 80
      :goto_1
      add-int/lit8 v5, v5, 0x1
  
      goto :goto_0
  
+     .line 87
      :cond_2
      iget-object v14, v7, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
***************
*** 214,219 ****
--- 235,241 ----
  
      goto :goto_2
  
+     .line 94
      .end local v3           #arr$:[Ljava/lang/String;
      .end local v5           #i$:I
      .end local v6           #len$:I
***************
*** 225,240 ****
--- 247,266 ----
  
      move-result-object v2
  
+     .line 95
      .local v2, address:Ljava/lang/String;
      if-eqz v2, :cond_6
  
+     .line 96
      invoke-virtual {v7}, Landroid/net/MailTo;->getTo()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 97
      .local v1, addr:Ljava/lang/String;
      if-eqz v1, :cond_5
  
+     .line 98
      new-instance v13, Ljava/lang/StringBuilder;
  
      invoke-direct {v13}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 257,269 ****
  
      move-result-object v2
  
      :cond_5
      iget-object v13, v7, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
!     const-string v14, "to"
  
      invoke-virtual {v13, v14, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      .end local v1           #addr:Ljava/lang/String;
      :cond_6
      return-object v7
--- 283,297 ----
  
      move-result-object v2
  
+     .line 100
      :cond_5
      iget-object v13, v7, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
!     const-string/jumbo v14, "to"
  
      invoke-virtual {v13, v14, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 103
      .end local v1           #addr:Ljava/lang/String;
      :cond_6
      return-object v7
***************
*** 275,280 ****
--- 303,309 ----
      .locals 2
  
      .prologue
+     .line 141
      iget-object v0, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
      const-string v1, "body"
***************
*** 292,297 ****
--- 321,327 ----
      .locals 2
  
      .prologue
+     .line 123
      iget-object v0, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
      const-string v1, "cc"
***************
*** 319,324 ****
--- 349,355 ----
      .end annotation
  
      .prologue
+     .line 149
      iget-object v0, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
      return-object v0
***************
*** 328,336 ****
      .locals 2
  
      .prologue
      iget-object v0, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
!     const-string v1, "subject"
  
      invoke-virtual {v0, v1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
--- 359,368 ----
      .locals 2
  
      .prologue
+     .line 132
      iget-object v0, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
!     const-string/jumbo v1, "subject"
  
      invoke-virtual {v0, v1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
***************
*** 345,353 ****
      .locals 2
  
      .prologue
      iget-object v0, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
!     const-string v1, "to"
  
      invoke-virtual {v0, v1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
--- 377,386 ----
      .locals 2
  
      .prologue
+     .line 113
      iget-object v0, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
!     const-string/jumbo v1, "to"
  
      invoke-virtual {v0, v1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
***************
*** 362,378 ****
      .locals 4
  
      .prologue
      new-instance v2, Ljava/lang/StringBuilder;
  
!     const-string v3, "mailto:"
  
      invoke-direct {v2, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
  
      .local v2, sb:Ljava/lang/StringBuilder;
      const/16 v3, 0x3f
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
      iget-object v3, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
      invoke-virtual {v3}, Ljava/util/HashMap;->entrySet()Ljava/util/Set;
--- 395,414 ----
      .locals 4
  
      .prologue
+     .line 154
      new-instance v2, Ljava/lang/StringBuilder;
  
!     const-string/jumbo v3, "mailto:"
  
      invoke-direct {v2, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
  
+     .line 155
      .local v2, sb:Ljava/lang/StringBuilder;
      const/16 v3, 0x3f
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 156
      iget-object v3, p0, Landroid/net/MailTo;->mHeaders:Ljava/util/HashMap;
  
      invoke-virtual {v3}, Ljava/util/HashMap;->entrySet()Ljava/util/Set;
***************
*** 397,402 ****
--- 433,439 ----
  
      check-cast v0, Ljava/util/Map$Entry;
  
+     .line 157
      .local v0, header:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/lang/String;>;"
      invoke-interface {v0}, Ljava/util/Map$Entry;->getKey()Ljava/lang/Object;
  
***************
*** 410,419 ****
--- 447,458 ----
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 158
      const/16 v3, 0x3d
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 159
      invoke-interface {v0}, Ljava/util/Map$Entry;->getValue()Ljava/lang/Object;
  
      move-result-object v3
***************
*** 426,437 ****
--- 465,478 ----
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 160
      const/16 v3, 0x26
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
      goto :goto_0
  
+     .line 162
      .end local v0           #header:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/lang/String;>;"
      :cond_0
      invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
