*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/NetworkStats.smali	2012-03-23 19:36:52.595640065 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/NetworkStats.smali.original	2012-03-23 19:34:45.381320315 -0500
***************
*** 71,80 ****
--- 71,82 ----
      .locals 1
  
      .prologue
+     .line 46
      const/4 v0, 0x0
  
      sput-object v0, Landroid/net/NetworkStats;->IFACE_ALL:Ljava/lang/String;
  
+     .line 658
      new-instance v0, Landroid/net/NetworkStats$1;
  
      invoke-direct {v0}, Landroid/net/NetworkStats$1;-><init>()V
***************
*** 90,139 ****
--- 92,154 ----
      .parameter "initialSize"
  
      .prologue
+     .line 125
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 126
      iput-wide p1, p0, Landroid/net/NetworkStats;->elapsedRealtime:J
  
+     .line 127
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/net/NetworkStats;->size:I
  
+     .line 128
      new-array v0, p3, [Ljava/lang/String;
  
      iput-object v0, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
+     .line 129
      new-array v0, p3, [I
  
      iput-object v0, p0, Landroid/net/NetworkStats;->uid:[I
  
+     .line 130
      new-array v0, p3, [I
  
      iput-object v0, p0, Landroid/net/NetworkStats;->set:[I
  
+     .line 131
      new-array v0, p3, [I
  
      iput-object v0, p0, Landroid/net/NetworkStats;->tag:[I
  
+     .line 132
      new-array v0, p3, [J
  
      iput-object v0, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
+     .line 133
      new-array v0, p3, [J
  
      iput-object v0, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
+     .line 134
      new-array v0, p3, [J
  
      iput-object v0, p0, Landroid/net/NetworkStats;->txBytes:[J
  
+     .line 135
      new-array v0, p3, [J
  
      iput-object v0, p0, Landroid/net/NetworkStats;->txPackets:[J
  
+     .line 136
      new-array v0, p3, [J
  
      iput-object v0, p0, Landroid/net/NetworkStats;->operations:[J
  
+     .line 137
      return-void
  .end method
  
***************
*** 142,215 ****
--- 157,243 ----
      .parameter "parcel"
  
      .prologue
+     .line 139
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 140
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/net/NetworkStats;->elapsedRealtime:J
  
+     .line 141
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/net/NetworkStats;->size:I
  
+     .line 142
      invoke-virtual {p1}, Landroid/os/Parcel;->createStringArray()[Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
+     .line 143
      invoke-virtual {p1}, Landroid/os/Parcel;->createIntArray()[I
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/NetworkStats;->uid:[I
  
+     .line 144
      invoke-virtual {p1}, Landroid/os/Parcel;->createIntArray()[I
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/NetworkStats;->set:[I
  
+     .line 145
      invoke-virtual {p1}, Landroid/os/Parcel;->createIntArray()[I
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/NetworkStats;->tag:[I
  
+     .line 146
      invoke-virtual {p1}, Landroid/os/Parcel;->createLongArray()[J
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
+     .line 147
      invoke-virtual {p1}, Landroid/os/Parcel;->createLongArray()[J
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
+     .line 148
      invoke-virtual {p1}, Landroid/os/Parcel;->createLongArray()[J
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/NetworkStats;->txBytes:[J
  
+     .line 149
      invoke-virtual {p1}, Landroid/os/Parcel;->createLongArray()[J
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/NetworkStats;->txPackets:[J
  
+     .line 150
      invoke-virtual {p1}, Landroid/os/Parcel;->createLongArray()[J
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/NetworkStats;->operations:[J
  
+     .line 151
      return-void
  .end method
  
***************
*** 240,271 ****
--- 268,310 ----
  
      const-wide/16 v7, 0x0
  
+     .line 431
      if-eqz p1, :cond_3
  
      move-object v0, p1
  
+     .line 433
      .local v0, entry:Landroid/net/NetworkStats$Entry;
      :goto_0
      sget-object v6, Landroid/net/NetworkStats;->IFACE_ALL:Ljava/lang/String;
  
      iput-object v6, v0, Landroid/net/NetworkStats$Entry;->iface:Ljava/lang/String;
  
+     .line 434
      iput p3, v0, Landroid/net/NetworkStats$Entry;->uid:I
  
+     .line 435
      iput v10, v0, Landroid/net/NetworkStats$Entry;->set:I
  
+     .line 436
      iput v4, v0, Landroid/net/NetworkStats$Entry;->tag:I
  
+     .line 437
      iput-wide v7, v0, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
+     .line 438
      iput-wide v7, v0, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
+     .line 439
      iput-wide v7, v0, Landroid/net/NetworkStats$Entry;->txBytes:J
  
+     .line 440
      iput-wide v7, v0, Landroid/net/NetworkStats$Entry;->txPackets:J
  
+     .line 441
      iput-wide v7, v0, Landroid/net/NetworkStats$Entry;->operations:J
  
+     .line 443
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 274,279 ****
--- 313,319 ----
  
      if-ge v1, v6, :cond_7
  
+     .line 444
      if-eq p3, v10, :cond_0
  
      iget-object v6, p0, Landroid/net/NetworkStats;->uid:[I
***************
*** 285,290 ****
--- 325,331 ----
      :cond_0
      move v3, v5
  
+     .line 445
      .local v3, matchesUid:Z
      :goto_2
      if-eqz p2, :cond_1
***************
*** 302,325 ****
--- 343,370 ----
      :cond_1
      move v2, v5
  
+     .line 447
      .local v2, matchesIface:Z
      :goto_3
      if-eqz v3, :cond_2
  
      if-eqz v2, :cond_2
  
+     .line 449
      iget-object v6, p0, Landroid/net/NetworkStats;->tag:[I
  
      aget v6, v6, v1
  
      if-eqz v6, :cond_6
  
+     .line 443
      :cond_2
      :goto_4
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 431
      .end local v0           #entry:Landroid/net/NetworkStats$Entry;
      .end local v1           #i:I
      .end local v2           #matchesIface:Z
***************
*** 336,349 ****
--- 381,397 ----
      :cond_4
      move v3, v4
  
+     .line 444
      goto :goto_2
  
      .restart local v3       #matchesUid:Z
      :cond_5
      move v2, v4
  
+     .line 445
      goto :goto_3
  
+     .line 451
      .restart local v2       #matchesIface:Z
      :cond_6
      iget-wide v6, v0, Landroid/net/NetworkStats$Entry;->rxBytes:J
***************
*** 356,361 ****
--- 404,410 ----
  
      iput-wide v6, v0, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
+     .line 452
      iget-wide v6, v0, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
      iget-object v8, p0, Landroid/net/NetworkStats;->rxPackets:[J
***************
*** 366,371 ****
--- 415,421 ----
  
      iput-wide v6, v0, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
+     .line 453
      iget-wide v6, v0, Landroid/net/NetworkStats$Entry;->txBytes:J
  
      iget-object v8, p0, Landroid/net/NetworkStats;->txBytes:[J
***************
*** 376,381 ****
--- 426,432 ----
  
      iput-wide v6, v0, Landroid/net/NetworkStats$Entry;->txBytes:J
  
+     .line 454
      iget-wide v6, v0, Landroid/net/NetworkStats$Entry;->txPackets:J
  
      iget-object v8, p0, Landroid/net/NetworkStats;->txPackets:[J
***************
*** 386,391 ****
--- 437,443 ----
  
      iput-wide v6, v0, Landroid/net/NetworkStats$Entry;->txPackets:J
  
+     .line 455
      iget-wide v6, v0, Landroid/net/NetworkStats$Entry;->operations:J
  
      iget-object v8, p0, Landroid/net/NetworkStats;->operations:[J
***************
*** 398,403 ****
--- 450,456 ----
  
      goto :goto_4
  
+     .line 458
      .end local v2           #matchesIface:Z
      .end local v3           #matchesUid:Z
      :cond_7
***************
*** 409,436 ****
--- 462,495 ----
      .parameter "set"
  
      .prologue
+     .line 627
      packed-switch p0, :pswitch_data_0
  
+     .line 635
      const-string v0, "UNKNOWN"
  
      :goto_0
      return-object v0
  
+     .line 629
      :pswitch_0
      const-string v0, "ALL"
  
      goto :goto_0
  
+     .line 631
      :pswitch_1
      const-string v0, "DEFAULT"
  
      goto :goto_0
  
+     .line 633
      :pswitch_2
      const-string v0, "FOREGROUND"
  
      goto :goto_0
  
+     .line 627
      nop
  
      :pswitch_data_0
***************
*** 446,451 ****
--- 505,511 ----
      .parameter "tag"
  
      .prologue
+     .line 643
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 482,487 ****
--- 542,548 ----
      .parameter "txPackets"
  
      .prologue
+     .line 182
      const/4 v2, -0x1
  
      const/4 v3, 0x0
***************
*** 514,519 ****
--- 575,581 ----
      .parameter "entry"
  
      .prologue
+     .line 198
      iget v1, p0, Landroid/net/NetworkStats;->size:I
  
      iget-object v2, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
***************
*** 522,527 ****
--- 584,590 ----
  
      if-lt v1, v2, :cond_0
  
+     .line 199
      iget-object v1, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
      array-length v1, v1
***************
*** 536,541 ****
--- 599,605 ----
  
      div-int/lit8 v0, v1, 0x2
  
+     .line 200
      .local v0, newLength:I
      iget-object v1, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
***************
*** 547,552 ****
--- 611,617 ----
  
      iput-object v1, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
+     .line 201
      iget-object v1, p0, Landroid/net/NetworkStats;->uid:[I
  
      invoke-static {v1, v0}, Ljava/util/Arrays;->copyOf([II)[I
***************
*** 555,560 ****
--- 620,626 ----
  
      iput-object v1, p0, Landroid/net/NetworkStats;->uid:[I
  
+     .line 202
      iget-object v1, p0, Landroid/net/NetworkStats;->set:[I
  
      invoke-static {v1, v0}, Ljava/util/Arrays;->copyOf([II)[I
***************
*** 563,568 ****
--- 629,635 ----
  
      iput-object v1, p0, Landroid/net/NetworkStats;->set:[I
  
+     .line 203
      iget-object v1, p0, Landroid/net/NetworkStats;->tag:[I
  
      invoke-static {v1, v0}, Ljava/util/Arrays;->copyOf([II)[I
***************
*** 571,576 ****
--- 638,644 ----
  
      iput-object v1, p0, Landroid/net/NetworkStats;->tag:[I
  
+     .line 204
      iget-object v1, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
      invoke-static {v1, v0}, Ljava/util/Arrays;->copyOf([JI)[J
***************
*** 579,584 ****
--- 647,653 ----
  
      iput-object v1, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
+     .line 205
      iget-object v1, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
      invoke-static {v1, v0}, Ljava/util/Arrays;->copyOf([JI)[J
***************
*** 587,592 ****
--- 656,662 ----
  
      iput-object v1, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
+     .line 206
      iget-object v1, p0, Landroid/net/NetworkStats;->txBytes:[J
  
      invoke-static {v1, v0}, Ljava/util/Arrays;->copyOf([JI)[J
***************
*** 595,600 ****
--- 665,671 ----
  
      iput-object v1, p0, Landroid/net/NetworkStats;->txBytes:[J
  
+     .line 207
      iget-object v1, p0, Landroid/net/NetworkStats;->txPackets:[J
  
      invoke-static {v1, v0}, Ljava/util/Arrays;->copyOf([JI)[J
***************
*** 603,608 ****
--- 674,680 ----
  
      iput-object v1, p0, Landroid/net/NetworkStats;->txPackets:[J
  
+     .line 208
      iget-object v1, p0, Landroid/net/NetworkStats;->operations:[J
  
      invoke-static {v1, v0}, Ljava/util/Arrays;->copyOf([JI)[J
***************
*** 611,616 ****
--- 683,689 ----
  
      iput-object v1, p0, Landroid/net/NetworkStats;->operations:[J
  
+     .line 211
      .end local v0           #newLength:I
      :cond_0
      iget-object v1, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
***************
*** 621,626 ****
--- 694,700 ----
  
      aput-object v3, v1, v2
  
+     .line 212
      iget-object v1, p0, Landroid/net/NetworkStats;->uid:[I
  
      iget v2, p0, Landroid/net/NetworkStats;->size:I
***************
*** 629,634 ****
--- 703,709 ----
  
      aput v3, v1, v2
  
+     .line 213
      iget-object v1, p0, Landroid/net/NetworkStats;->set:[I
  
      iget v2, p0, Landroid/net/NetworkStats;->size:I
***************
*** 637,642 ****
--- 712,718 ----
  
      aput v3, v1, v2
  
+     .line 214
      iget-object v1, p0, Landroid/net/NetworkStats;->tag:[I
  
      iget v2, p0, Landroid/net/NetworkStats;->size:I
***************
*** 645,650 ****
--- 721,727 ----
  
      aput v3, v1, v2
  
+     .line 215
      iget-object v1, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
      iget v2, p0, Landroid/net/NetworkStats;->size:I
***************
*** 653,658 ****
--- 730,736 ----
  
      aput-wide v3, v1, v2
  
+     .line 216
      iget-object v1, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
      iget v2, p0, Landroid/net/NetworkStats;->size:I
***************
*** 661,666 ****
--- 739,745 ----
  
      aput-wide v3, v1, v2
  
+     .line 217
      iget-object v1, p0, Landroid/net/NetworkStats;->txBytes:[J
  
      iget v2, p0, Landroid/net/NetworkStats;->size:I
***************
*** 669,674 ****
--- 748,754 ----
  
      aput-wide v3, v1, v2
  
+     .line 218
      iget-object v1, p0, Landroid/net/NetworkStats;->txPackets:[J
  
      iget v2, p0, Landroid/net/NetworkStats;->size:I
***************
*** 677,682 ****
--- 757,763 ----
  
      aput-wide v3, v1, v2
  
+     .line 219
      iget-object v1, p0, Landroid/net/NetworkStats;->operations:[J
  
      iget v2, p0, Landroid/net/NetworkStats;->size:I
***************
*** 685,696 ****
--- 766,779 ----
  
      aput-wide v3, v1, v2
  
+     .line 220
      iget v1, p0, Landroid/net/NetworkStats;->size:I
  
      add-int/lit8 v1, v1, 0x1
  
      iput v1, p0, Landroid/net/NetworkStats;->size:I
  
+     .line 222
      return-object p0
  .end method
  
***************
*** 707,712 ****
--- 790,796 ----
      .parameter "operations"
  
      .prologue
+     .line 189
      new-instance v0, Landroid/net/NetworkStats$Entry;
  
      move-object/from16 v1, p1
***************
*** 740,745 ****
--- 824,830 ----
      .locals 6
  
      .prologue
+     .line 170
      new-instance v0, Landroid/net/NetworkStats;
  
      iget-wide v3, p0, Landroid/net/NetworkStats;->elapsedRealtime:J
***************
*** 748,756 ****
--- 833,843 ----
  
      invoke-direct {v0, v3, v4, v5}, Landroid/net/NetworkStats;-><init>(JI)V
  
+     .line 171
      .local v0, clone:Landroid/net/NetworkStats;
      const/4 v1, 0x0
  
+     .line 172
      .local v1, entry:Landroid/net/NetworkStats$Entry;
      const/4 v2, 0x0
  
***************
*** 760,775 ****
--- 847,866 ----
  
      if-ge v2, v3, :cond_0
  
+     .line 173
      invoke-virtual {p0, v2, v1}, Landroid/net/NetworkStats;->getValues(ILandroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats$Entry;
  
      move-result-object v1
  
+     .line 174
      invoke-virtual {v0, v1}, Landroid/net/NetworkStats;->addValues(Landroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats;
  
+     .line 172
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 176
      :cond_0
      return-object v0
  .end method
***************
*** 783,788 ****
--- 874,880 ----
      .end annotation
  
      .prologue
+     .line 42
      invoke-virtual {p0}, Landroid/net/NetworkStats;->clone()Landroid/net/NetworkStats;
  
      move-result-object v0
***************
*** 795,802 ****
--- 887,896 ----
      .parameter "another"
  
      .prologue
+     .line 300
      const/4 v0, 0x0
  
+     .line 301
      .local v0, entry:Landroid/net/NetworkStats$Entry;
      const/4 v1, 0x0
  
***************
*** 806,821 ****
--- 900,919 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 302
      invoke-virtual {p1, v1, v0}, Landroid/net/NetworkStats;->getValues(ILandroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats$Entry;
  
      move-result-object v0
  
+     .line 303
      invoke-virtual {p0, v0}, Landroid/net/NetworkStats;->combineValues(Landroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats;
  
+     .line 301
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 305
      :cond_0
      return-void
  .end method
***************
*** 825,830 ****
--- 923,929 ----
      .parameter "entry"
  
      .prologue
+     .line 282
      iget-object v1, p1, Landroid/net/NetworkStats$Entry;->iface:Ljava/lang/String;
  
      iget v2, p1, Landroid/net/NetworkStats$Entry;->uid:I
***************
*** 837,852 ****
--- 936,955 ----
  
      move-result v0
  
+     .line 283
      .local v0, i:I
      const/4 v1, -0x1
  
      if-ne v0, v1, :cond_0
  
+     .line 285
      invoke-virtual {p0, p1}, Landroid/net/NetworkStats;->addValues(Landroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats;
  
+     .line 293
      :goto_0
      return-object p0
  
+     .line 287
      :cond_0
      iget-object v1, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
***************
*** 858,863 ****
--- 961,967 ----
  
      aput-wide v2, v1, v0
  
+     .line 288
      iget-object v1, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
      aget-wide v2, v1, v0
***************
*** 868,873 ****
--- 972,978 ----
  
      aput-wide v2, v1, v0
  
+     .line 289
      iget-object v1, p0, Landroid/net/NetworkStats;->txBytes:[J
  
      aget-wide v2, v1, v0
***************
*** 878,883 ****
--- 983,989 ----
  
      aput-wide v2, v1, v0
  
+     .line 290
      iget-object v1, p0, Landroid/net/NetworkStats;->txPackets:[J
  
      aget-wide v2, v1, v0
***************
*** 888,893 ****
--- 994,1000 ----
  
      aput-wide v2, v1, v0
  
+     .line 291
      iget-object v1, p0, Landroid/net/NetworkStats;->operations:[J
  
      aget-wide v2, v1, v0
***************
*** 914,919 ****
--- 1021,1027 ----
      .parameter "operations"
  
      .prologue
+     .line 272
      new-instance v0, Landroid/net/NetworkStats$Entry;
  
      move-object/from16 v1, p1
***************
*** 957,962 ****
--- 1065,1071 ----
      .end annotation
  
      .prologue
+     .line 266
      const/4 v3, 0x0
  
      move-object v0, p0
***************
*** 988,993 ****
--- 1097,1103 ----
      .locals 1
  
      .prologue
+     .line 655
      const/4 v0, 0x0
  
      return v0
***************
*** 999,1006 ****
--- 1109,1118 ----
      .parameter "pw"
  
      .prologue
+     .line 606
      invoke-virtual {p2, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 607
      const-string v1, "NetworkStats: elapsedRealtime="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1009,1014 ****
--- 1121,1127 ----
  
      invoke-virtual {p2, v1, v2}, Ljava/io/PrintWriter;->println(J)V
  
+     .line 608
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 1017,1024 ****
--- 1130,1139 ----
  
      if-ge v0, v1, :cond_0
  
+     .line 609
      invoke-virtual {p2, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 610
      const-string v1, "  ["
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1029,1034 ****
--- 1144,1150 ----
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 611
      const-string v1, " iface="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1039,1044 ****
--- 1155,1161 ----
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 612
      const-string v1, " uid="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1049,1054 ****
--- 1166,1172 ----
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 613
      const-string v1, " set="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1063,1068 ****
--- 1181,1187 ----
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 614
      const-string v1, " tag="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1077,1082 ****
--- 1196,1202 ----
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 615
      const-string v1, " rxBytes="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1087,1092 ****
--- 1207,1213 ----
  
      invoke-virtual {p2, v1, v2}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 616
      const-string v1, " rxPackets="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1097,1102 ****
--- 1218,1224 ----
  
      invoke-virtual {p2, v1, v2}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 617
      const-string v1, " txBytes="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1107,1112 ****
--- 1229,1235 ----
  
      invoke-virtual {p2, v1, v2}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 618
      const-string v1, " txPackets="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1117,1122 ****
--- 1240,1246 ----
  
      invoke-virtual {p2, v1, v2}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 619
      const-string v1, " operations="
  
      invoke-virtual {p2, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1127,1136 ****
--- 1251,1262 ----
  
      invoke-virtual {p2, v1, v2}, Ljava/io/PrintWriter;->println(J)V
  
+     .line 608
      add-int/lit8 v0, v0, 0x1
  
      goto/16 :goto_0
  
+     .line 621
      :cond_0
      return-void
  .end method
***************
*** 1143,1148 ****
--- 1269,1275 ----
      .parameter "tag"
  
      .prologue
+     .line 311
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 1151,1156 ****
--- 1278,1284 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 312
      iget-object v1, p0, Landroid/net/NetworkStats;->uid:[I
  
      aget v1, v1, v0
***************
*** 1179,1194 ****
--- 1307,1325 ----
  
      if-eqz v1, :cond_0
  
+     .line 317
      .end local v0           #i:I
      :goto_1
      return v0
  
+     .line 311
      .restart local v0       #i:I
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 317
      :cond_1
      const/4 v0, -0x1
  
***************
*** 1204,1209 ****
--- 1335,1341 ----
      .parameter "hintIndex"
  
      .prologue
+     .line 326
      const/4 v2, 0x0
  
      .local v2, offset:I
***************
*** 1212,1230 ****
--- 1344,1366 ----
  
      if-ge v2, v3, :cond_2
  
+     .line 327
      div-int/lit8 v0, v2, 0x2
  
+     .line 331
      .local v0, halfOffset:I
      rem-int/lit8 v3, v2, 0x2
  
      if-nez v3, :cond_0
  
+     .line 332
      add-int v3, p5, v0
  
      iget v4, p0, Landroid/net/NetworkStats;->size:I
  
      rem-int v1, v3, v4
  
+     .line 337
      .local v1, i:I
      :goto_1
      iget-object v3, p0, Landroid/net/NetworkStats;->uid:[I
***************
*** 1255,1265 ****
--- 1391,1403 ----
  
      if-eqz v3, :cond_1
  
+     .line 342
      .end local v0           #halfOffset:I
      .end local v1           #i:I
      :goto_2
      return v1
  
+     .line 334
      .restart local v0       #halfOffset:I
      :cond_0
      iget v3, p0, Landroid/net/NetworkStats;->size:I
***************
*** 1277,1287 ****
--- 1415,1427 ----
      .restart local v1       #i:I
      goto :goto_1
  
+     .line 326
      :cond_1
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 342
      .end local v0           #halfOffset:I
      .end local v1           #i:I
      :cond_2
***************
*** 1294,1299 ****
--- 1434,1440 ----
      .locals 2
  
      .prologue
+     .line 243
      iget-wide v0, p0, Landroid/net/NetworkStats;->elapsedRealtime:J
  
      return-wide v0
***************
*** 1303,1308 ****
--- 1444,1450 ----
      .locals 4
  
      .prologue
+     .line 251
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v0
***************
*** 1319,1324 ****
--- 1461,1467 ----
      .parameter "recycle"
  
      .prologue
+     .line 404
      const/4 v0, 0x0
  
      const/4 v1, -0x1
***************
*** 1336,1341 ****
--- 1479,1485 ----
      .parameter "limitUid"
  
      .prologue
+     .line 412
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0, p2}, Landroid/net/NetworkStats;->getTotal(Landroid/net/NetworkStats$Entry;Ljava/util/HashSet;I)Landroid/net/NetworkStats$Entry;
***************
*** 1362,1367 ****
--- 1506,1512 ----
      .end annotation
  
      .prologue
+     .line 420
      .local p2, limitIface:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/String;>;"
      const/4 v0, -0x1
  
***************
*** 1376,1387 ****
--- 1521,1534 ----
      .locals 5
  
      .prologue
+     .line 396
      const/4 v1, 0x0
  
      invoke-virtual {p0, v1}, Landroid/net/NetworkStats;->getTotal(Landroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats$Entry;
  
      move-result-object v0
  
+     .line 397
      .local v0, entry:Landroid/net/NetworkStats$Entry;
      iget-wide v1, v0, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
***************
*** 1396,1405 ****
--- 1543,1554 ----
      .locals 6
  
      .prologue
+     .line 365
      new-instance v3, Ljava/util/HashSet;
  
      invoke-direct {v3}, Ljava/util/HashSet;-><init>()V
  
+     .line 366
      .local v3, ifaces:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/String;>;"
      iget-object v0, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
***************
*** 1415,1432 ****
--- 1564,1585 ----
  
      aget-object v2, v0, v1
  
+     .line 367
      .local v2, iface:Ljava/lang/String;
      sget-object v5, Landroid/net/NetworkStats;->IFACE_ALL:Ljava/lang/String;
  
      if-eq v2, v5, :cond_0
  
+     .line 368
      invoke-virtual {v3, v2}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 366
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 371
      .end local v2           #iface:Ljava/lang/String;
      :cond_1
      invoke-virtual {v3}, Ljava/util/HashSet;->size()I
***************
*** 1448,1457 ****
--- 1601,1612 ----
      .locals 9
  
      .prologue
+     .line 378
      new-instance v7, Landroid/util/SparseBooleanArray;
  
      invoke-direct {v7}, Landroid/util/SparseBooleanArray;-><init>()V
  
+     .line 379
      .local v7, uids:Landroid/util/SparseBooleanArray;
      iget-object v0, p0, Landroid/net/NetworkStats;->uid:[I
  
***************
*** 1467,1490 ****
--- 1622,1650 ----
  
      aget v6, v0, v2
  
+     .line 380
      .local v6, uid:I
      const/4 v8, 0x1
  
      invoke-virtual {v7, v6, v8}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 379
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 383
      .end local v6           #uid:I
      :cond_0
      invoke-virtual {v7}, Landroid/util/SparseBooleanArray;->size()I
  
      move-result v5
  
+     .line 384
      .local v5, size:I
      new-array v4, v5, [I
  
+     .line 385
      .local v4, result:[I
      const/4 v1, 0x0
  
***************
*** 1492,1507 ****
--- 1652,1670 ----
      :goto_1
      if-ge v1, v5, :cond_1
  
+     .line 386
      invoke-virtual {v7, v1}, Landroid/util/SparseBooleanArray;->keyAt(I)I
  
      move-result v8
  
      aput v8, v4, v1
  
+     .line 385
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 388
      :cond_1
      return-object v4
  .end method
***************
*** 1512,1521 ****
--- 1675,1686 ----
      .parameter "recycle"
  
      .prologue
+     .line 229
      if-eqz p2, :cond_0
  
      move-object v0, p2
  
+     .line 230
      .local v0, entry:Landroid/net/NetworkStats$Entry;
      :goto_0
      iget-object v1, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
***************
*** 1524,1579 ****
--- 1689,1754 ----
  
      iput-object v1, v0, Landroid/net/NetworkStats$Entry;->iface:Ljava/lang/String;
  
+     .line 231
      iget-object v1, p0, Landroid/net/NetworkStats;->uid:[I
  
      aget v1, v1, p1
  
      iput v1, v0, Landroid/net/NetworkStats$Entry;->uid:I
  
+     .line 232
      iget-object v1, p0, Landroid/net/NetworkStats;->set:[I
  
      aget v1, v1, p1
  
      iput v1, v0, Landroid/net/NetworkStats$Entry;->set:I
  
+     .line 233
      iget-object v1, p0, Landroid/net/NetworkStats;->tag:[I
  
      aget v1, v1, p1
  
      iput v1, v0, Landroid/net/NetworkStats$Entry;->tag:I
  
+     .line 234
      iget-object v1, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
      aget-wide v1, v1, p1
  
      iput-wide v1, v0, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
+     .line 235
      iget-object v1, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
      aget-wide v1, v1, p1
  
      iput-wide v1, v0, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
+     .line 236
      iget-object v1, p0, Landroid/net/NetworkStats;->txBytes:[J
  
      aget-wide v1, v1, p1
  
      iput-wide v1, v0, Landroid/net/NetworkStats$Entry;->txBytes:J
  
+     .line 237
      iget-object v1, p0, Landroid/net/NetworkStats;->txPackets:[J
  
      aget-wide v1, v1, p1
  
      iput-wide v1, v0, Landroid/net/NetworkStats$Entry;->txPackets:J
  
+     .line 238
      iget-object v1, p0, Landroid/net/NetworkStats;->operations:[J
  
      aget-wide v1, v1, p1
  
      iput-wide v1, v0, Landroid/net/NetworkStats$Entry;->operations:J
  
+     .line 239
      return-object v0
  
+     .line 229
      .end local v0           #entry:Landroid/net/NetworkStats$Entry;
      :cond_0
      new-instance v0, Landroid/net/NetworkStats$Entry;
***************
*** 1589,1594 ****
--- 1764,1770 ----
      .prologue
      const/4 v6, -0x1
  
+     .line 536
      new-instance v2, Landroid/net/NetworkStats;
  
      iget-wide v3, p0, Landroid/net/NetworkStats;->elapsedRealtime:J
***************
*** 1597,1620 ****
--- 1773,1802 ----
  
      invoke-direct {v2, v3, v4, v5}, Landroid/net/NetworkStats;-><init>(JI)V
  
+     .line 538
      .local v2, stats:Landroid/net/NetworkStats;
      new-instance v0, Landroid/net/NetworkStats$Entry;
  
      invoke-direct {v0}, Landroid/net/NetworkStats$Entry;-><init>()V
  
+     .line 539
      .local v0, entry:Landroid/net/NetworkStats$Entry;
      iput v6, v0, Landroid/net/NetworkStats$Entry;->uid:I
  
+     .line 540
      iput v6, v0, Landroid/net/NetworkStats$Entry;->set:I
  
+     .line 541
      const/4 v3, 0x0
  
      iput v3, v0, Landroid/net/NetworkStats$Entry;->tag:I
  
+     .line 542
      const-wide/16 v3, 0x0
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->operations:J
  
+     .line 544
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 1623,1639 ****
--- 1805,1824 ----
  
      if-ge v1, v3, :cond_1
  
+     .line 546
      iget-object v3, p0, Landroid/net/NetworkStats;->tag:[I
  
      aget v3, v3, v1
  
      if-eqz v3, :cond_0
  
+     .line 544
      :goto_1
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 548
      :cond_0
      iget-object v3, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
***************
*** 1641,1674 ****
--- 1826,1865 ----
  
      iput-object v3, v0, Landroid/net/NetworkStats$Entry;->iface:Ljava/lang/String;
  
+     .line 549
      iget-object v3, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
      aget-wide v3, v3, v1
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
+     .line 550
      iget-object v3, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
      aget-wide v3, v3, v1
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
+     .line 551
      iget-object v3, p0, Landroid/net/NetworkStats;->txBytes:[J
  
      aget-wide v3, v3, v1
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->txBytes:J
  
+     .line 552
      iget-object v3, p0, Landroid/net/NetworkStats;->txPackets:[J
  
      aget-wide v3, v3, v1
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->txPackets:J
  
+     .line 553
      invoke-virtual {v2, v0}, Landroid/net/NetworkStats;->combineValues(Landroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats;
  
      goto :goto_1
  
+     .line 556
      :cond_1
      return-object v2
  .end method
***************
*** 1677,1682 ****
--- 1868,1874 ----
      .locals 6
  
      .prologue
+     .line 564
      new-instance v2, Landroid/net/NetworkStats;
  
      iget-wide v3, p0, Landroid/net/NetworkStats;->elapsedRealtime:J
***************
*** 1685,1708 ****
--- 1877,1905 ----
  
      invoke-direct {v2, v3, v4, v5}, Landroid/net/NetworkStats;-><init>(JI)V
  
+     .line 566
      .local v2, stats:Landroid/net/NetworkStats;
      new-instance v0, Landroid/net/NetworkStats$Entry;
  
      invoke-direct {v0}, Landroid/net/NetworkStats$Entry;-><init>()V
  
+     .line 567
      .local v0, entry:Landroid/net/NetworkStats$Entry;
      sget-object v3, Landroid/net/NetworkStats;->IFACE_ALL:Ljava/lang/String;
  
      iput-object v3, v0, Landroid/net/NetworkStats$Entry;->iface:Ljava/lang/String;
  
+     .line 568
      const/4 v3, -0x1
  
      iput v3, v0, Landroid/net/NetworkStats$Entry;->set:I
  
+     .line 569
      const/4 v3, 0x0
  
      iput v3, v0, Landroid/net/NetworkStats$Entry;->tag:I
  
+     .line 571
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 1711,1727 ****
--- 1908,1927 ----
  
      if-ge v1, v3, :cond_1
  
+     .line 573
      iget-object v3, p0, Landroid/net/NetworkStats;->tag:[I
  
      aget v3, v3, v1
  
      if-eqz v3, :cond_0
  
+     .line 571
      :goto_1
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 575
      :cond_0
      iget-object v3, p0, Landroid/net/NetworkStats;->uid:[I
  
***************
*** 1729,1768 ****
--- 1929,1975 ----
  
      iput v3, v0, Landroid/net/NetworkStats$Entry;->uid:I
  
+     .line 576
      iget-object v3, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
      aget-wide v3, v3, v1
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
+     .line 577
      iget-object v3, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
      aget-wide v3, v3, v1
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
+     .line 578
      iget-object v3, p0, Landroid/net/NetworkStats;->txBytes:[J
  
      aget-wide v3, v3, v1
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->txBytes:J
  
+     .line 579
      iget-object v3, p0, Landroid/net/NetworkStats;->txPackets:[J
  
      aget-wide v3, v3, v1
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->txPackets:J
  
+     .line 580
      iget-object v3, p0, Landroid/net/NetworkStats;->operations:[J
  
      aget-wide v3, v3, v1
  
      iput-wide v3, v0, Landroid/net/NetworkStats$Entry;->operations:J
  
+     .line 581
      invoke-virtual {v2, v0}, Landroid/net/NetworkStats;->combineValues(Landroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats;
  
      goto :goto_1
  
+     .line 584
      :cond_1
      return-object v2
  .end method
***************
*** 1771,1776 ****
--- 1978,1984 ----
      .locals 1
  
      .prologue
+     .line 260
      iget-object v0, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
      array-length v0, v0
***************
*** 1782,1787 ****
--- 1990,1996 ----
      .locals 1
  
      .prologue
+     .line 255
      iget v0, p0, Landroid/net/NetworkStats;->size:I
  
      return v0
***************
*** 1792,1797 ****
--- 2001,2007 ----
      .parameter "stats"
  
      .prologue
+     .line 351
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 1800,1805 ****
--- 2010,2016 ----
  
      if-ge v0, v2, :cond_1
  
+     .line 352
      sget-object v2, Landroid/net/NetworkStats;->IFACE_ALL:Ljava/lang/String;
  
      iget-object v3, p0, Landroid/net/NetworkStats;->uid:[I
***************
*** 1818,1839 ****
--- 2029,2054 ----
  
      move-result v1
  
+     .line 353
      .local v1, j:I
      const/4 v2, -0x1
  
      if-ne v1, v2, :cond_0
  
+     .line 354
      iget-object v2, p0, Landroid/net/NetworkStats;->operations:[J
  
      const-wide/16 v3, 0x0
  
      aput-wide v3, v2, v0
  
+     .line 351
      :goto_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 356
      :cond_0
      iget-object v2, p0, Landroid/net/NetworkStats;->operations:[J
  
***************
*** 1845,1850 ****
--- 2060,2066 ----
  
      goto :goto_1
  
+     .line 359
      .end local v1           #j:I
      :cond_1
      return-void
***************
*** 1860,1865 ****
--- 2076,2082 ----
      .end annotation
  
      .prologue
+     .line 467
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Landroid/net/NetworkStats;->subtract(Landroid/net/NetworkStats;Z)Landroid/net/NetworkStats;
***************
*** 1882,1909 ****
--- 2099,2131 ----
      .prologue
      const-wide/16 v11, 0x0
  
+     .line 480
      iget-wide v0, p0, Landroid/net/NetworkStats;->elapsedRealtime:J
  
      iget-wide v2, p1, Landroid/net/NetworkStats;->elapsedRealtime:J
  
      sub-long v6, v0, v2
  
+     .line 481
      .local v6, deltaRealtime:J
      cmp-long v0, v6, v11
  
      if-gez v0, :cond_0
  
+     .line 482
      new-instance v0, Landroid/net/NetworkStats$NonMonotonicException;
  
      invoke-direct {v0, p0, p1}, Landroid/net/NetworkStats$NonMonotonicException;-><init>(Landroid/net/NetworkStats;Landroid/net/NetworkStats;)V
  
      throw v0
  
+     .line 486
      :cond_0
      new-instance v8, Landroid/net/NetworkStats$Entry;
  
      invoke-direct {v8}, Landroid/net/NetworkStats$Entry;-><init>()V
  
+     .line 487
      .local v8, entry:Landroid/net/NetworkStats$Entry;
      new-instance v10, Landroid/net/NetworkStats;
  
***************
*** 1911,1916 ****
--- 2133,2139 ----
  
      invoke-direct {v10, v6, v7, v0}, Landroid/net/NetworkStats;-><init>(JI)V
  
+     .line 488
      .local v10, result:Landroid/net/NetworkStats;
      const/4 v5, 0x0
  
***************
*** 1920,1949 ****
--- 2143,2177 ----
  
      if-ge v5, v0, :cond_5
  
+     .line 489
      iget-object v0, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
      aget-object v0, v0, v5
  
      iput-object v0, v8, Landroid/net/NetworkStats$Entry;->iface:Ljava/lang/String;
  
+     .line 490
      iget-object v0, p0, Landroid/net/NetworkStats;->uid:[I
  
      aget v0, v0, v5
  
      iput v0, v8, Landroid/net/NetworkStats$Entry;->uid:I
  
+     .line 491
      iget-object v0, p0, Landroid/net/NetworkStats;->set:[I
  
      aget v0, v0, v5
  
      iput v0, v8, Landroid/net/NetworkStats$Entry;->set:I
  
+     .line 492
      iget-object v0, p0, Landroid/net/NetworkStats;->tag:[I
  
      aget v0, v0, v5
  
      iput v0, v8, Landroid/net/NetworkStats$Entry;->tag:I
  
+     .line 495
      iget-object v1, v8, Landroid/net/NetworkStats$Entry;->iface:Ljava/lang/String;
  
      iget v2, v8, Landroid/net/NetworkStats$Entry;->uid:I
***************
*** 1958,2006 ****
--- 2186,2243 ----
  
      move-result v9
  
+     .line 496
      .local v9, j:I
      const/4 v0, -0x1
  
      if-ne v9, v0, :cond_2
  
+     .line 498
      iget-object v0, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
      aget-wide v0, v0, v5
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
+     .line 499
      iget-object v0, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
      aget-wide v0, v0, v5
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
+     .line 500
      iget-object v0, p0, Landroid/net/NetworkStats;->txBytes:[J
  
      aget-wide v0, v0, v5
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->txBytes:J
  
+     .line 501
      iget-object v0, p0, Landroid/net/NetworkStats;->txPackets:[J
  
      aget-wide v0, v0, v5
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->txPackets:J
  
+     .line 502
      iget-object v0, p0, Landroid/net/NetworkStats;->operations:[J
  
      aget-wide v0, v0, v5
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->operations:J
  
+     .line 525
      :cond_1
      :goto_1
      invoke-virtual {v10, v8}, Landroid/net/NetworkStats;->addValues(Landroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats;
  
+     .line 488
      add-int/lit8 v5, v5, 0x1
  
      goto :goto_0
  
+     .line 505
      :cond_2
      iget-object v0, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
***************
*** 2014,2019 ****
--- 2251,2257 ----
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
+     .line 506
      iget-object v0, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
      aget-wide v0, v0, v5
***************
*** 2026,2031 ****
--- 2264,2270 ----
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
+     .line 507
      iget-object v0, p0, Landroid/net/NetworkStats;->txBytes:[J
  
      aget-wide v0, v0, v5
***************
*** 2038,2043 ****
--- 2277,2283 ----
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->txBytes:J
  
+     .line 508
      iget-object v0, p0, Landroid/net/NetworkStats;->txPackets:[J
  
      aget-wide v0, v0, v5
***************
*** 2050,2055 ****
--- 2290,2296 ----
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->txPackets:J
  
+     .line 509
      iget-object v0, p0, Landroid/net/NetworkStats;->operations:[J
  
      aget-wide v0, v0, v5
***************
*** 2062,2067 ****
--- 2303,2309 ----
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->operations:J
  
+     .line 511
      iget-wide v0, v8, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
      cmp-long v0, v0, v11
***************
*** 2092,2100 ****
--- 2334,2344 ----
  
      if-gez v0, :cond_1
  
+     .line 513
      :cond_3
      if-eqz p2, :cond_4
  
+     .line 514
      iget-wide v0, v8, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
      invoke-static {v0, v1, v11, v12}, Ljava/lang/Math;->max(JJ)J
***************
*** 2103,2108 ****
--- 2347,2353 ----
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
+     .line 515
      iget-wide v0, v8, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
      invoke-static {v0, v1, v11, v12}, Ljava/lang/Math;->max(JJ)J
***************
*** 2111,2116 ****
--- 2356,2362 ----
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->rxPackets:J
  
+     .line 516
      iget-wide v0, v8, Landroid/net/NetworkStats$Entry;->txBytes:J
  
      invoke-static {v0, v1, v11, v12}, Ljava/lang/Math;->max(JJ)J
***************
*** 2119,2124 ****
--- 2365,2371 ----
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->txBytes:J
  
+     .line 517
      iget-wide v0, v8, Landroid/net/NetworkStats$Entry;->txPackets:J
  
      invoke-static {v0, v1, v11, v12}, Ljava/lang/Math;->max(JJ)J
***************
*** 2127,2132 ****
--- 2374,2380 ----
  
      iput-wide v0, v8, Landroid/net/NetworkStats$Entry;->txPackets:J
  
+     .line 518
      iget-wide v0, v8, Landroid/net/NetworkStats$Entry;->operations:J
  
      invoke-static {v0, v1, v11, v12}, Ljava/lang/Math;->max(JJ)J
***************
*** 2137,2142 ****
--- 2385,2391 ----
  
      goto/16 :goto_1
  
+     .line 520
      :cond_4
      new-instance v0, Landroid/net/NetworkStats$NonMonotonicException;
  
***************
*** 2144,2149 ****
--- 2393,2399 ----
  
      throw v0
  
+     .line 528
      .end local v9           #j:I
      :cond_5
      return-object v10
***************
*** 2153,2162 ****
--- 2403,2414 ----
      .locals 3
  
      .prologue
+     .line 648
      new-instance v0, Ljava/io/CharArrayWriter;
  
      invoke-direct {v0}, Ljava/io/CharArrayWriter;-><init>()V
  
+     .line 649
      .local v0, writer:Ljava/io/CharArrayWriter;
      const-string v1, ""
  
***************
*** 2166,2171 ****
--- 2418,2424 ----
  
      invoke-virtual {p0, v1, v2}, Landroid/net/NetworkStats;->dump(Ljava/lang/String;Ljava/io/PrintWriter;)V
  
+     .line 650
      invoke-virtual {v0}, Ljava/io/CharArrayWriter;->toString()Ljava/lang/String;
  
      move-result-object v1
***************
*** 2178,2183 ****
--- 2431,2437 ----
      .parameter "uid"
  
      .prologue
+     .line 592
      new-instance v2, Landroid/net/NetworkStats;
  
      iget-wide v3, p0, Landroid/net/NetworkStats;->elapsedRealtime:J
***************
*** 2186,2196 ****
--- 2440,2452 ----
  
      invoke-direct {v2, v3, v4, v5}, Landroid/net/NetworkStats;-><init>(JI)V
  
+     .line 594
      .local v2, stats:Landroid/net/NetworkStats;
      new-instance v0, Landroid/net/NetworkStats$Entry;
  
      invoke-direct {v0}, Landroid/net/NetworkStats$Entry;-><init>()V
  
+     .line 595
      .local v0, entry:Landroid/net/NetworkStats$Entry;
      const/4 v1, 0x0
  
***************
*** 2200,2220 ****
--- 2456,2481 ----
  
      if-ge v1, v3, :cond_1
  
+     .line 596
      invoke-virtual {p0, v1, v0}, Landroid/net/NetworkStats;->getValues(ILandroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats$Entry;
  
      move-result-object v0
  
+     .line 597
      iget v3, v0, Landroid/net/NetworkStats$Entry;->uid:I
  
      if-eq v3, p1, :cond_0
  
+     .line 598
      invoke-virtual {v2, v0}, Landroid/net/NetworkStats;->addValues(Landroid/net/NetworkStats$Entry;)Landroid/net/NetworkStats;
  
+     .line 595
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 602
      :cond_1
      return-object v2
  .end method
***************
*** 2225,2273 ****
--- 2486,2546 ----
      .parameter "flags"
  
      .prologue
+     .line 155
      iget-wide v0, p0, Landroid/net/NetworkStats;->elapsedRealtime:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 156
      iget v0, p0, Landroid/net/NetworkStats;->size:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 157
      iget-object v0, p0, Landroid/net/NetworkStats;->iface:[Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStringArray([Ljava/lang/String;)V
  
+     .line 158
      iget-object v0, p0, Landroid/net/NetworkStats;->uid:[I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeIntArray([I)V
  
+     .line 159
      iget-object v0, p0, Landroid/net/NetworkStats;->set:[I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeIntArray([I)V
  
+     .line 160
      iget-object v0, p0, Landroid/net/NetworkStats;->tag:[I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeIntArray([I)V
  
+     .line 161
      iget-object v0, p0, Landroid/net/NetworkStats;->rxBytes:[J
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeLongArray([J)V
  
+     .line 162
      iget-object v0, p0, Landroid/net/NetworkStats;->rxPackets:[J
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeLongArray([J)V
  
+     .line 163
      iget-object v0, p0, Landroid/net/NetworkStats;->txBytes:[J
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeLongArray([J)V
  
+     .line 164
      iget-object v0, p0, Landroid/net/NetworkStats;->txPackets:[J
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeLongArray([J)V
  
+     .line 165
      iget-object v0, p0, Landroid/net/NetworkStats;->operations:[J
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeLongArray([J)V
  
+     .line 166
      return-void
  .end method
