*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/NetworkConfig.smali	2012-03-23 19:36:54.658677684 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/NetworkConfig.smali.original	2012-03-23 19:34:45.402320697 -0500
***************
*** 23,36 ****
--- 23,39 ----
      .parameter "init"
  
      .prologue
+     .line 64
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 65
      const-string v1, ","
  
      invoke-virtual {p1, v1}, Ljava/lang/String;->split(Ljava/lang/String;)[Ljava/lang/String;
  
      move-result-object v0
  
+     .line 66
      .local v0, fragments:[Ljava/lang/String;
      const/4 v1, 0x0
  
***************
*** 46,51 ****
--- 49,55 ----
  
      iput-object v1, p0, Landroid/net/NetworkConfig;->name:Ljava/lang/String;
  
+     .line 67
      const/4 v1, 0x1
  
      aget-object v1, v0, v1
***************
*** 56,61 ****
--- 60,66 ----
  
      iput v1, p0, Landroid/net/NetworkConfig;->type:I
  
+     .line 68
      const/4 v1, 0x2
  
      aget-object v1, v0, v1
***************
*** 66,71 ****
--- 71,77 ----
  
      iput v1, p0, Landroid/net/NetworkConfig;->radio:I
  
+     .line 69
      const/4 v1, 0x3
  
      aget-object v1, v0, v1
***************
*** 76,81 ****
--- 82,88 ----
  
      iput v1, p0, Landroid/net/NetworkConfig;->priority:I
  
+     .line 70
      const/4 v1, 0x4
  
      aget-object v1, v0, v1
***************
*** 86,91 ****
--- 93,99 ----
  
      iput v1, p0, Landroid/net/NetworkConfig;->restoreTime:I
  
+     .line 71
      const/4 v1, 0x5
  
      aget-object v1, v0, v1
***************
*** 96,101 ****
--- 104,110 ----
  
      iput-boolean v1, p0, Landroid/net/NetworkConfig;->dependencyMet:Z
  
+     .line 72
      return-void
  .end method
  
***************
*** 105,110 ****
--- 114,120 ----
      .locals 2
  
      .prologue
+     .line 78
      iget v0, p0, Landroid/net/NetworkConfig;->type:I
  
      iget v1, p0, Landroid/net/NetworkConfig;->radio:I
