*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/NetworkInfo$1.smali	2012-03-23 19:36:53.164650440 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/NetworkInfo$1.smali.original	2012-03-23 19:34:45.387320423 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 419
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 48,77 ****
--- 49,84 ----
  
      const/4 v7, 0x0
  
+     .line 421
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
+     .line 422
      .local v1, netType:I
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
+     .line 423
      .local v2, subtype:I
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v4
  
+     .line 424
      .local v4, typeName:Ljava/lang/String;
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 425
      .local v3, subtypeName:Ljava/lang/String;
      new-instance v0, Landroid/net/NetworkInfo;
  
      invoke-direct {v0, v1, v2, v4, v3}, Landroid/net/NetworkInfo;-><init>(IILjava/lang/String;Ljava/lang/String;)V
  
+     .line 426
      .local v0, netInfo:Landroid/net/NetworkInfo;
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
***************
*** 84,89 ****
--- 91,97 ----
      #setter for: Landroid/net/NetworkInfo;->mState:Landroid/net/NetworkInfo$State;
      invoke-static {v0, v5}, Landroid/net/NetworkInfo;->access$002(Landroid/net/NetworkInfo;Landroid/net/NetworkInfo$State;)Landroid/net/NetworkInfo$State;
  
+     .line 427
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v5
***************
*** 95,100 ****
--- 103,109 ----
      #setter for: Landroid/net/NetworkInfo;->mDetailedState:Landroid/net/NetworkInfo$DetailedState;
      invoke-static {v0, v5}, Landroid/net/NetworkInfo;->access$102(Landroid/net/NetworkInfo;Landroid/net/NetworkInfo$DetailedState;)Landroid/net/NetworkInfo$DetailedState;
  
+     .line 428
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
***************
*** 107,112 ****
--- 116,122 ----
      #setter for: Landroid/net/NetworkInfo;->mIsFailover:Z
      invoke-static {v0, v5}, Landroid/net/NetworkInfo;->access$202(Landroid/net/NetworkInfo;Z)Z
  
+     .line 429
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
***************
*** 119,124 ****
--- 129,135 ----
      #setter for: Landroid/net/NetworkInfo;->mIsAvailable:Z
      invoke-static {v0, v5}, Landroid/net/NetworkInfo;->access$302(Landroid/net/NetworkInfo;Z)Z
  
+     .line 430
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
***************
*** 129,134 ****
--- 140,146 ----
      #setter for: Landroid/net/NetworkInfo;->mIsRoaming:Z
      invoke-static {v0, v6}, Landroid/net/NetworkInfo;->access$402(Landroid/net/NetworkInfo;Z)Z
  
+     .line 431
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v5
***************
*** 136,141 ****
--- 148,154 ----
      #setter for: Landroid/net/NetworkInfo;->mReason:Ljava/lang/String;
      invoke-static {v0, v5}, Landroid/net/NetworkInfo;->access$502(Landroid/net/NetworkInfo;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 432
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v5
***************
*** 143,163 ****
--- 156,180 ----
      #setter for: Landroid/net/NetworkInfo;->mExtraInfo:Ljava/lang/String;
      invoke-static {v0, v5}, Landroid/net/NetworkInfo;->access$602(Landroid/net/NetworkInfo;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 433
      return-object v0
  
      :cond_0
      move v5, v7
  
+     .line 428
      goto :goto_0
  
      :cond_1
      move v5, v7
  
+     .line 429
      goto :goto_1
  
      :cond_2
      move v6, v7
  
+     .line 430
      goto :goto_2
  .end method
  
***************
*** 166,171 ****
--- 183,189 ----
      .parameter "x0"
  
      .prologue
+     .line 419
      invoke-virtual {p0, p1}, Landroid/net/NetworkInfo$1;->createFromParcel(Landroid/os/Parcel;)Landroid/net/NetworkInfo;
  
      move-result-object v0
***************
*** 178,183 ****
--- 196,202 ----
      .parameter "size"
  
      .prologue
+     .line 437
      new-array v0, p1, [Landroid/net/NetworkInfo;
  
      return-object v0
***************
*** 188,193 ****
--- 207,213 ----
      .parameter "x0"
  
      .prologue
+     .line 419
      invoke-virtual {p0, p1}, Landroid/net/NetworkInfo$1;->newArray(I)[Landroid/net/NetworkInfo;
  
      move-result-object v0
