*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/Proxy.smali	2012-03-23 19:36:59.899773255 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/Proxy.smali.original	2012-03-23 19:34:45.473321992 -0500
***************
*** 38,47 ****
--- 38,49 ----
      .locals 1
  
      .prologue
+     .line 80
      const/4 v0, 0x0
  
      sput-object v0, Landroid/net/Proxy;->sConnectivityManager:Landroid/net/ConnectivityManager;
  
+     .line 97
      const-string v0, "^$|^[a-zA-Z0-9]+(\\-[a-zA-Z0-9]+)*(\\.[a-zA-Z0-9]+(\\-[a-zA-Z0-9]+)*)*$"
  
      invoke-static {v0}, Ljava/util/regex/Pattern;->compile(Ljava/lang/String;)Ljava/util/regex/Pattern;
***************
*** 50,55 ****
--- 52,58 ----
  
      sput-object v0, Landroid/net/Proxy;->HOSTNAME_PATTERN:Ljava/util/regex/Pattern;
  
+     .line 98
      const-string v0, "$|^(.?[a-zA-Z0-9]+(\\-[a-zA-Z0-9]+)*(\\.[a-zA-Z0-9]+(\\-[a-zA-Z0-9]+)*)*)+(,(.?[a-zA-Z0-9]+(\\-[a-zA-Z0-9]+)*(\\.[a-zA-Z0-9]+(\\-[a-zA-Z0-9]+)*)*))*$"
  
      invoke-static {v0}, Ljava/util/regex/Pattern;->compile(Ljava/lang/String;)Ljava/util/regex/Pattern;
***************
*** 58,63 ****
--- 61,67 ----
  
      sput-object v0, Landroid/net/Proxy;->EXCLLIST_PATTERN:Ljava/util/regex/Pattern;
  
+     .line 99
      return-void
  .end method
  
***************
*** 65,72 ****
--- 69,78 ----
      .locals 0
  
      .prologue
+     .line 57
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 281
      return-void
  .end method
  
***************
*** 75,80 ****
--- 81,87 ----
      .parameter "context"
  
      .prologue
+     .line 318
      new-instance v0, Landroid/net/Proxy$AndroidProxySelectorRoutePlanner;
  
      new-instance v1, Lorg/apache/http/conn/scheme/SchemeRegistry;
***************
*** 87,92 ****
--- 94,100 ----
  
      invoke-direct {v0, v1, v2, p0}, Landroid/net/Proxy$AndroidProxySelectorRoutePlanner;-><init>(Lorg/apache/http/conn/scheme/SchemeRegistry;Ljava/net/ProxySelector;Landroid/content/Context;)V
  
+     .line 320
      .local v0, ret:Landroid/net/Proxy$AndroidProxySelectorRoutePlanner;
      return-object v0
  .end method
***************
*** 95,106 ****
--- 103,116 ----
      .locals 2
  
      .prologue
+     .line 181
      const-string v1, "http.proxyHost"
  
      invoke-static {v1}, Ljava/lang/System;->getProperty(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 182
      .local v0, host:Ljava/lang/String;
      invoke-static {v0}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
***************
*** 110,115 ****
--- 120,126 ----
  
      const/4 v0, 0x0
  
+     .line 183
      .end local v0           #host:Ljava/lang/String;
      :cond_0
      return-object v0
***************
*** 121,136 ****
--- 132,150 ----
      .prologue
      const/4 v1, -0x1
  
+     .line 195
      invoke-static {}, Landroid/net/Proxy;->getDefaultHost()Ljava/lang/String;
  
      move-result-object v2
  
      if-nez v2, :cond_0
  
+     .line 199
      .local v0, e:Ljava/lang/NumberFormatException;
      :goto_0
      return v1
  
+     .line 197
      .end local v0           #e:Ljava/lang/NumberFormatException;
      :cond_0
      :try_start_0
***************
*** 148,156 ****
--- 162,172 ----
  
      goto :goto_0
  
+     .line 198
      :catch_0
      move-exception v0
  
+     .line 199
      .restart local v0       #e:Ljava/lang/NumberFormatException;
      goto :goto_0
  .end method
***************
*** 162,171 ****
--- 178,189 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 146
      invoke-static {p0, v3}, Landroid/net/Proxy;->getProxy(Landroid/content/Context;Ljava/lang/String;)Ljava/net/Proxy;
  
      move-result-object v1
  
+     .line 147
      .local v1, proxy:Ljava/net/Proxy;
      sget-object v2, Ljava/net/Proxy;->NO_PROXY:Ljava/net/Proxy;
  
***************
*** 173,181 ****
--- 191,201 ----
  
      move-object v2, v3
  
+     .line 151
      :goto_0
      return-object v2
  
+     .line 149
      :cond_0
      :try_start_0
      invoke-virtual {v1}, Ljava/net/Proxy;->address()Ljava/net/SocketAddress;
***************
*** 194,205 ****
--- 214,227 ----
  
      goto :goto_0
  
+     .line 150
      :catch_0
      move-exception v0
  
      .local v0, e:Ljava/lang/Exception;
      move-object v2, v3
  
+     .line 151
      goto :goto_0
  .end method
  
***************
*** 210,221 ****
--- 232,245 ----
      .prologue
      const/4 v3, -0x1
  
+     .line 163
      const/4 v2, 0x0
  
      invoke-static {p0, v2}, Landroid/net/Proxy;->getProxy(Landroid/content/Context;Ljava/lang/String;)Ljava/net/Proxy;
  
      move-result-object v1
  
+     .line 164
      .local v1, proxy:Ljava/net/Proxy;
      sget-object v2, Ljava/net/Proxy;->NO_PROXY:Ljava/net/Proxy;
  
***************
*** 223,231 ****
--- 247,257 ----
  
      move v2, v3
  
+     .line 168
      :goto_0
      return v2
  
+     .line 166
      :cond_0
      :try_start_0
      invoke-virtual {v1}, Ljava/net/Proxy;->address()Ljava/net/SocketAddress;
***************
*** 244,255 ****
--- 270,283 ----
  
      goto :goto_0
  
+     .line 167
      :catch_0
      move-exception v0
  
      .local v0, e:Ljava/lang/Exception;
      move v2, v3
  
+     .line 168
      goto :goto_0
  .end method
  
***************
*** 259,268 ****
--- 287,298 ----
      .parameter "url"
  
      .prologue
+     .line 218
      invoke-static {p0, p1}, Landroid/net/Proxy;->getProxy(Landroid/content/Context;Ljava/lang/String;)Ljava/net/Proxy;
  
      move-result-object v0
  
+     .line 219
      .local v0, prefProxy:Ljava/net/Proxy;
      sget-object v2, Ljava/net/Proxy;->NO_PROXY:Ljava/net/Proxy;
  
***************
*** 272,282 ****
--- 302,315 ----
  
      if-eqz v2, :cond_0
  
+     .line 220
      const/4 v2, 0x0
  
+     .line 223
      :goto_0
      return-object v2
  
+     .line 222
      :cond_0
      invoke-virtual {v0}, Ljava/net/Proxy;->address()Ljava/net/SocketAddress;
  
***************
*** 284,289 ****
--- 317,323 ----
  
      check-cast v1, Ljava/net/InetSocketAddress;
  
+     .line 223
      .local v1, sa:Ljava/net/InetSocketAddress;
      new-instance v2, Lorg/apache/http/HttpHost;
  
***************
*** 308,327 ****
--- 342,366 ----
      .parameter "url"
  
      .prologue
+     .line 111
      const-string v0, ""
  
+     .line 112
      .local v0, host:Ljava/lang/String;
      if-eqz p1, :cond_0
  
+     .line 113
      invoke-static {p1}, Ljava/net/URI;->create(Ljava/lang/String;)Ljava/net/URI;
  
      move-result-object v2
  
+     .line 114
      .local v2, uri:Ljava/net/URI;
      invoke-virtual {v2}, Ljava/net/URI;->getHost()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 117
      .end local v2           #uri:Ljava/net/URI;
      :cond_0
      invoke-static {v0}, Landroid/net/Proxy;->isLocalHost(Ljava/lang/String;)Z
***************
*** 330,339 ****
--- 369,380 ----
  
      if-nez v3, :cond_3
  
+     .line 118
      sget-object v3, Landroid/net/Proxy;->sConnectivityManager:Landroid/net/ConnectivityManager;
  
      if-nez v3, :cond_1
  
+     .line 119
      const-string v3, "connectivity"
  
      invoke-virtual {p0, v3}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 344,349 ****
--- 385,391 ----
  
      sput-object v3, Landroid/net/Proxy;->sConnectivityManager:Landroid/net/ConnectivityManager;
  
+     .line 122
      :cond_1
      sget-object v3, Landroid/net/Proxy;->sConnectivityManager:Landroid/net/ConnectivityManager;
  
***************
*** 351,359 ****
--- 393,403 ----
  
      sget-object v3, Ljava/net/Proxy;->NO_PROXY:Ljava/net/Proxy;
  
+     .line 132
      :goto_0
      return-object v3
  
+     .line 124
      :cond_2
      sget-object v3, Landroid/net/Proxy;->sConnectivityManager:Landroid/net/ConnectivityManager;
  
***************
*** 361,381 ****
--- 405,429 ----
  
      move-result-object v1
  
+     .line 126
      .local v1, proxyProperties:Landroid/net/ProxyProperties;
      if-eqz v1, :cond_3
  
+     .line 127
      invoke-virtual {v1, v0}, Landroid/net/ProxyProperties;->isExcluded(Ljava/lang/String;)Z
  
      move-result v3
  
      if-nez v3, :cond_3
  
+     .line 128
      invoke-virtual {v1}, Landroid/net/ProxyProperties;->makeProxy()Ljava/net/Proxy;
  
      move-result-object v3
  
      goto :goto_0
  
+     .line 132
      .end local v1           #proxyProperties:Landroid/net/ProxyProperties;
      :cond_3
      sget-object v3, Ljava/net/Proxy;->NO_PROXY:Ljava/net/Proxy;
***************
*** 392,406 ****
--- 440,458 ----
  
      const/4 v0, 0x0
  
+     .line 228
      if-nez p0, :cond_1
  
+     .line 242
      :cond_0
      :goto_0
      return v0
  
+     .line 232
      :cond_1
      if-eqz p0, :cond_0
  
+     .line 233
      :try_start_0
      const-string v2, "localhost"
  
***************
*** 412,419 ****
--- 464,473 ----
  
      move v0, v1
  
+     .line 234
      goto :goto_0
  
+     .line 236
      :cond_2
      invoke-static {p0}, Landroid/net/NetworkUtils;->numericToInetAddress(Ljava/lang/String;)Ljava/net/InetAddress;
  
***************
*** 429,436 ****
--- 483,492 ----
  
      move v0, v1
  
+     .line 237
      goto :goto_0
  
+     .line 240
      :catch_0
      move-exception v1
  
***************
*** 442,462 ****
--- 498,524 ----
      .parameter "p"
  
      .prologue
+     .line 325
      const/4 v1, 0x0
  
+     .line 326
      .local v1, host:Ljava/lang/String;
      const/4 v2, 0x0
  
+     .line 327
      .local v2, port:Ljava/lang/String;
      const/4 v0, 0x0
  
+     .line 328
      .local v0, exclList:Ljava/lang/String;
      if-eqz p0, :cond_0
  
+     .line 329
      invoke-virtual {p0}, Landroid/net/ProxyProperties;->getHost()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 330
      invoke-virtual {p0}, Landroid/net/ProxyProperties;->getPort()I
  
      move-result v3
***************
*** 465,477 ****
--- 527,542 ----
  
      move-result-object v2
  
+     .line 331
      invoke-virtual {p0}, Landroid/net/ProxyProperties;->getExclusionList()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 333
      :cond_0
      invoke-static {v1, v2, v0}, Landroid/net/Proxy;->setHttpProxySystemProperty(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 334
      return-void
  .end method
  
***************
*** 482,560 ****
      .parameter "exclList"
  
      .prologue
      if-eqz p2, :cond_0
  
      const-string v0, ","
  
!     const-string v1, "|"
  
      invoke-virtual {p2, v0, v1}, Ljava/lang/String;->replace(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Ljava/lang/String;
  
      move-result-object p2
  
      :cond_0
      if-eqz p0, :cond_1
  
      const-string v0, "http.proxyHost"
  
      invoke-static {v0, p0}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
      const-string v0, "https.proxyHost"
  
      invoke-static {v0, p0}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
      :goto_0
      if-eqz p1, :cond_2
  
      const-string v0, "http.proxyPort"
  
      invoke-static {v0, p1}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
      const-string v0, "https.proxyPort"
  
      invoke-static {v0, p1}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
      :goto_1
      if-eqz p2, :cond_3
  
      const-string v0, "http.nonProxyHosts"
  
      invoke-static {v0, p2}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
      const-string v0, "https.nonProxyHosts"
  
      invoke-static {v0, p2}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
      :goto_2
      return-void
  
      :cond_1
      const-string v0, "http.proxyHost"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
      const-string v0, "https.proxyHost"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
      goto :goto_0
  
      :cond_2
      const-string v0, "http.proxyPort"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
      const-string v0, "https.proxyPort"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
      goto :goto_1
  
      :cond_3
      const-string v0, "http.nonProxyHosts"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
      const-string v0, "https.nonProxyHosts"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
--- 547,642 ----
      .parameter "exclList"
  
      .prologue
+     .line 338
      if-eqz p2, :cond_0
  
      const-string v0, ","
  
!     const-string/jumbo v1, "|"
  
      invoke-virtual {p2, v0, v1}, Ljava/lang/String;->replace(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Ljava/lang/String;
  
      move-result-object p2
  
+     .line 340
      :cond_0
      if-eqz p0, :cond_1
  
+     .line 341
      const-string v0, "http.proxyHost"
  
      invoke-static {v0, p0}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 342
      const-string v0, "https.proxyHost"
  
      invoke-static {v0, p0}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 347
      :goto_0
      if-eqz p1, :cond_2
  
+     .line 348
      const-string v0, "http.proxyPort"
  
      invoke-static {v0, p1}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 349
      const-string v0, "https.proxyPort"
  
      invoke-static {v0, p1}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 354
      :goto_1
      if-eqz p2, :cond_3
  
+     .line 355
      const-string v0, "http.nonProxyHosts"
  
      invoke-static {v0, p2}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 356
      const-string v0, "https.nonProxyHosts"
  
      invoke-static {v0, p2}, Ljava/lang/System;->setProperty(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 361
      :goto_2
      return-void
  
+     .line 344
      :cond_1
      const-string v0, "http.proxyHost"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
+     .line 345
      const-string v0, "https.proxyHost"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
      goto :goto_0
  
+     .line 351
      :cond_2
      const-string v0, "http.proxyPort"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
+     .line 352
      const-string v0, "https.proxyPort"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
      goto :goto_1
  
+     .line 358
      :cond_3
      const-string v0, "http.nonProxyHosts"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
  
+     .line 359
      const-string v0, "https.nonProxyHosts"
  
      invoke-static {v0}, Ljava/lang/System;->clearProperty(Ljava/lang/String;)Ljava/lang/String;
***************
*** 569,580 ****
--- 651,664 ----
      .parameter "exclList"
  
      .prologue
+     .line 250
      sget-object v4, Landroid/net/Proxy;->HOSTNAME_PATTERN:Ljava/util/regex/Pattern;
  
      invoke-virtual {v4, p0}, Ljava/util/regex/Pattern;->matcher(Ljava/lang/CharSequence;)Ljava/util/regex/Matcher;
  
      move-result-object v2
  
+     .line 251
      .local v2, match:Ljava/util/regex/Matcher;
      sget-object v4, Landroid/net/Proxy;->EXCLLIST_PATTERN:Ljava/util/regex/Pattern;
  
***************
*** 582,587 ****
--- 666,672 ----
  
      move-result-object v1
  
+     .line 253
      .local v1, listMatch:Ljava/util/regex/Matcher;
      invoke-virtual {v2}, Ljava/util/regex/Matcher;->matches()Z
  
***************
*** 589,600 ****
--- 674,687 ----
  
      if-nez v4, :cond_0
  
+     .line 254
      new-instance v4, Ljava/lang/IllegalArgumentException;
  
      invoke-direct {v4}, Ljava/lang/IllegalArgumentException;-><init>()V
  
      throw v4
  
+     .line 257
      :cond_0
      invoke-virtual {v1}, Ljava/util/regex/Matcher;->matches()Z
  
***************
*** 602,613 ****
--- 689,702 ----
  
      if-nez v4, :cond_1
  
+     .line 258
      new-instance v4, Ljava/lang/IllegalArgumentException;
  
      invoke-direct {v4}, Ljava/lang/IllegalArgumentException;-><init>()V
  
      throw v4
  
+     .line 261
      :cond_1
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
***************
*** 621,632 ****
--- 710,723 ----
  
      if-nez v4, :cond_2
  
+     .line 262
      new-instance v4, Ljava/lang/IllegalArgumentException;
  
      invoke-direct {v4}, Ljava/lang/IllegalArgumentException;-><init>()V
  
      throw v4
  
+     .line 265
      :cond_2
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
***************
*** 634,654 ****
--- 725,749 ----
  
      if-lez v4, :cond_5
  
+     .line 266
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v4
  
      if-nez v4, :cond_3
  
+     .line 267
      new-instance v4, Ljava/lang/IllegalArgumentException;
  
      invoke-direct {v4}, Ljava/lang/IllegalArgumentException;-><init>()V
  
      throw v4
  
+     .line 269
      :cond_3
      const/4 v3, -0x1
  
+     .line 271
      .local v3, portVal:I
      :try_start_0
      invoke-static {p1}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
***************
*** 657,668 ****
--- 752,765 ----
  
      move-result v3
  
+     .line 275
      if-lez v3, :cond_4
  
      const v4, 0xffff
  
      if-le v3, v4, :cond_5
  
+     .line 276
      :cond_4
      new-instance v4, Ljava/lang/IllegalArgumentException;
  
***************
*** 670,678 ****
--- 767,777 ----
  
      throw v4
  
+     .line 272
      :catch_0
      move-exception v0
  
+     .line 273
      .local v0, ex:Ljava/lang/NumberFormatException;
      new-instance v4, Ljava/lang/IllegalArgumentException;
  
***************
*** 680,685 ****
--- 779,785 ----
  
      throw v4
  
+     .line 279
      .end local v0           #ex:Ljava/lang/NumberFormatException;
      .end local v3           #portVal:I
      :cond_5
