*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/ProxyProperties.smali	2012-03-23 19:36:53.656659411 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/ProxyProperties.smali.original	2012-03-23 19:34:45.392320515 -0500
***************
*** 34,39 ****
--- 34,40 ----
      .locals 1
  
      .prologue
+     .line 198
      new-instance v0, Landroid/net/ProxyProperties$1;
  
      invoke-direct {v0}, Landroid/net/ProxyProperties$1;-><init>()V
***************
*** 48,79 ****
--- 49,87 ----
      .parameter "source"
  
      .prologue
+     .line 54
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 55
      if-eqz p1, :cond_0
  
+     .line 56
      invoke-virtual {p1}, Landroid/net/ProxyProperties;->getHost()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
+     .line 57
      invoke-virtual {p1}, Landroid/net/ProxyProperties;->getPort()I
  
      move-result v0
  
      iput v0, p0, Landroid/net/ProxyProperties;->mPort:I
  
+     .line 58
      invoke-virtual {p1}, Landroid/net/ProxyProperties;->getExclusionList()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/ProxyProperties;->mExclusionList:Ljava/lang/String;
  
+     .line 59
      iget-object v0, p1, Landroid/net/ProxyProperties;->mParsedExclusionList:[Ljava/lang/String;
  
      iput-object v0, p0, Landroid/net/ProxyProperties;->mParsedExclusionList:[Ljava/lang/String;
  
+     .line 61
      :cond_0
      return-void
  .end method
***************
*** 85,98 ****
--- 93,111 ----
      .parameter "exclList"
  
      .prologue
+     .line 40
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 41
      iput-object p1, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
+     .line 42
      iput p2, p0, Landroid/net/ProxyProperties;->mPort:I
  
+     .line 43
      invoke-direct {p0, p3}, Landroid/net/ProxyProperties;->setExclusionList(Ljava/lang/String;)V
  
+     .line 44
      return-void
  .end method
  
***************
*** 104,119 ****
--- 117,138 ----
      .parameter "parsedExclList"
  
      .prologue
+     .line 46
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 47
      iput-object p1, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
+     .line 48
      iput p2, p0, Landroid/net/ProxyProperties;->mPort:I
  
+     .line 49
      iput-object p3, p0, Landroid/net/ProxyProperties;->mExclusionList:Ljava/lang/String;
  
+     .line 50
      iput-object p4, p0, Landroid/net/ProxyProperties;->mParsedExclusionList:[Ljava/lang/String;
  
+     .line 51
      return-void
  .end method
  
***************
*** 126,131 ****
--- 145,151 ----
      .parameter "x4"
  
      .prologue
+     .line 33
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/net/ProxyProperties;-><init>(Ljava/lang/String;ILjava/lang/String;[Ljava/lang/String;)V
  
      return-void
***************
*** 136,156 ****
--- 156,181 ----
      .parameter "exclusionList"
  
      .prologue
+     .line 86
      iput-object p1, p0, Landroid/net/ProxyProperties;->mExclusionList:Ljava/lang/String;
  
+     .line 87
      iget-object v3, p0, Landroid/net/ProxyProperties;->mExclusionList:Ljava/lang/String;
  
      if-nez v3, :cond_1
  
+     .line 88
      const/4 v3, 0x0
  
      new-array v3, v3, [Ljava/lang/String;
  
      iput-object v3, p0, Landroid/net/ProxyProperties;->mParsedExclusionList:[Ljava/lang/String;
  
+     .line 99
      :cond_0
      return-void
  
+     .line 90
      :cond_1
      invoke-virtual {p1}, Ljava/lang/String;->toLowerCase()Ljava/lang/String;
  
***************
*** 162,167 ****
--- 187,193 ----
  
      move-result-object v2
  
+     .line 91
      .local v2, splitExclusionList:[Ljava/lang/String;
      array-length v3, v2
  
***************
*** 171,176 ****
--- 197,203 ----
  
      iput-object v3, p0, Landroid/net/ProxyProperties;->mParsedExclusionList:[Ljava/lang/String;
  
+     .line 92
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 179,190 ****
--- 206,219 ----
  
      if-ge v0, v3, :cond_0
  
+     .line 93
      aget-object v3, v2, v0
  
      invoke-virtual {v3}, Ljava/lang/String;->trim()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 94
      .local v1, s:Ljava/lang/String;
      const-string v3, "."
  
***************
*** 200,205 ****
--- 229,235 ----
  
      move-result-object v1
  
+     .line 95
      :cond_2
      iget-object v3, p0, Landroid/net/ProxyProperties;->mParsedExclusionList:[Ljava/lang/String;
  
***************
*** 207,212 ****
--- 237,243 ----
  
      aput-object v1, v3, v4
  
+     .line 96
      iget-object v3, p0, Landroid/net/ProxyProperties;->mParsedExclusionList:[Ljava/lang/String;
  
      mul-int/lit8 v4, v0, 0x2
***************
*** 233,238 ****
--- 264,270 ----
  
      aput-object v5, v3, v4
  
+     .line 92
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
***************
*** 244,249 ****
--- 276,282 ----
      .locals 1
  
      .prologue
+     .line 165
      const/4 v0, 0x0
  
      return v0
***************
*** 256,265 ****
--- 289,300 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 148
      instance-of v2, p1, Landroid/net/ProxyProperties;
  
      if-nez v2, :cond_1
  
+     .line 157
      :cond_0
      :goto_0
      return v1
***************
*** 267,274 ****
--- 302,311 ----
      :cond_1
      move-object v0, p1
  
+     .line 149
      check-cast v0, Landroid/net/ProxyProperties;
  
+     .line 150
      .local v0, p:Landroid/net/ProxyProperties;
      iget-object v2, p0, Landroid/net/ProxyProperties;->mExclusionList:Ljava/lang/String;
  
***************
*** 286,291 ****
--- 323,329 ----
  
      if-eqz v2, :cond_0
  
+     .line 151
      :cond_2
      iget-object v2, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
***************
*** 309,314 ****
--- 347,353 ----
  
      if-eqz v2, :cond_0
  
+     .line 154
      :cond_3
      iget-object v2, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
***************
*** 318,323 ****
--- 357,363 ----
  
      if-eqz v2, :cond_0
  
+     .line 155
      :cond_4
      iget-object v2, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
***************
*** 327,332 ****
--- 367,373 ----
  
      if-nez v2, :cond_0
  
+     .line 156
      :cond_5
      iget v2, p0, Landroid/net/ProxyProperties;->mPort:I
  
***************
*** 334,339 ****
--- 375,381 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 157
      const/4 v1, 0x1
  
      goto :goto_0
***************
*** 343,348 ****
--- 385,391 ----
      .locals 1
  
      .prologue
+     .line 81
      iget-object v0, p0, Landroid/net/ProxyProperties;->mExclusionList:Ljava/lang/String;
  
      return-object v0
***************
*** 352,357 ****
--- 395,401 ----
      .locals 1
  
      .prologue
+     .line 72
      iget-object v0, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
      return-object v0
***************
*** 361,366 ****
--- 405,411 ----
      .locals 1
  
      .prologue
+     .line 76
      iget v0, p0, Landroid/net/ProxyProperties;->mPort:I
  
      return v0
***************
*** 370,377 ****
--- 415,424 ----
      .locals 4
  
      .prologue
+     .line 64
      const/4 v0, 0x0
  
+     .line 66
      .local v0, inetSocketAddress:Ljava/net/InetSocketAddress;
      :try_start_0
      new-instance v1, Ljava/net/InetSocketAddress;
***************
*** 388,398 ****
--- 435,447 ----
      .local v1, inetSocketAddress:Ljava/net/InetSocketAddress;
      move-object v0, v1
  
+     .line 68
      .end local v1           #inetSocketAddress:Ljava/net/InetSocketAddress;
      .restart local v0       #inetSocketAddress:Ljava/net/InetSocketAddress;
      :goto_0
      return-object v0
  
+     .line 67
      :catch_0
      move-exception v2
  
***************
*** 405,410 ****
--- 454,460 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 173
      iget-object v0, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
      if-nez v0, :cond_0
***************
*** 451,456 ****
--- 501,507 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 102
      invoke-static {p1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v4
***************
*** 467,489 ****
--- 518,545 ----
  
      if-nez v4, :cond_1
  
+     .line 114
      :cond_0
      :goto_0
      return v3
  
+     .line 105
      :cond_1
      invoke-static {p1}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v1
  
+     .line 106
      .local v1, u:Landroid/net/Uri;
      invoke-virtual {v1}, Landroid/net/Uri;->getHost()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 107
      .local v2, urlDomain:Ljava/lang/String;
      if-eqz v2, :cond_0
  
+     .line 108
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 494,499 ****
--- 550,556 ----
  
      if-ge v0, v4, :cond_0
  
+     .line 109
      iget-object v4, p0, Landroid/net/ProxyProperties;->mParsedExclusionList:[Ljava/lang/String;
  
      aget-object v4, v4, v0
***************
*** 516,526 ****
--- 573,585 ----
  
      if-eqz v4, :cond_3
  
+     .line 111
      :cond_2
      const/4 v3, 0x1
  
      goto :goto_0
  
+     .line 108
      :cond_3
      add-int/lit8 v0, v0, 0x2
  
***************
*** 531,543 ****
--- 590,605 ----
      .locals 5
  
      .prologue
+     .line 118
      sget-object v1, Ljava/net/Proxy;->NO_PROXY:Ljava/net/Proxy;
  
+     .line 119
      .local v1, proxy:Ljava/net/Proxy;
      iget-object v3, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
      if-eqz v3, :cond_0
  
+     .line 121
      :try_start_0
      new-instance v0, Ljava/net/InetSocketAddress;
  
***************
*** 547,552 ****
--- 609,615 ----
  
      invoke-direct {v0, v3, v4}, Ljava/net/InetSocketAddress;-><init>(Ljava/lang/String;I)V
  
+     .line 122
      .local v0, inetSocketAddress:Ljava/net/InetSocketAddress;
      new-instance v2, Ljava/net/Proxy;
  
***************
*** 560,565 ****
--- 623,629 ----
      .local v2, proxy:Ljava/net/Proxy;
      move-object v1, v2
  
+     .line 126
      .end local v0           #inetSocketAddress:Ljava/net/InetSocketAddress;
      .end local v2           #proxy:Ljava/net/Proxy;
      .restart local v1       #proxy:Ljava/net/Proxy;
***************
*** 567,572 ****
--- 631,637 ----
      :goto_0
      return-object v1
  
+     .line 123
      :catch_0
      move-exception v3
  
***************
*** 577,603 ****
--- 642,674 ----
      .locals 3
  
      .prologue
+     .line 131
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 132
      .local v0, sb:Ljava/lang/StringBuilder;
      iget-object v1, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
      if-eqz v1, :cond_1
  
+     .line 133
      const-string v1, "["
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 134
      iget-object v1, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 135
      const-string v1, "] "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 136
      iget v1, p0, Landroid/net/ProxyProperties;->mPort:I
  
      invoke-static {v1}, Ljava/lang/Integer;->toString(I)Ljava/lang/String;
***************
*** 606,615 ****
--- 677,688 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 137
      iget-object v1, p0, Landroid/net/ProxyProperties;->mExclusionList:Ljava/lang/String;
  
      if-eqz v1, :cond_0
  
+     .line 138
      const-string v1, " xl="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 620,625 ****
--- 693,699 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 143
      :cond_0
      :goto_0
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
***************
*** 628,633 ****
--- 702,708 ----
  
      return-object v1
  
+     .line 141
      :cond_1
      const-string v1, "[ProxyProperties.mHost == null]"
  
***************
*** 642,674 ****
--- 717,757 ----
      .parameter "flags"
  
      .prologue
+     .line 183
      iget-object v0, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
      if-eqz v0, :cond_0
  
+     .line 184
      const/4 v0, 0x1
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeByte(B)V
  
+     .line 185
      iget-object v0, p0, Landroid/net/ProxyProperties;->mHost:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 186
      iget v0, p0, Landroid/net/ProxyProperties;->mPort:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 190
      :goto_0
      iget-object v0, p0, Landroid/net/ProxyProperties;->mExclusionList:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 191
      iget-object v0, p0, Landroid/net/ProxyProperties;->mParsedExclusionList:[Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStringArray([Ljava/lang/String;)V
  
+     .line 192
      return-void
  
+     .line 188
      :cond_0
      const/4 v0, 0x0
  
