*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/RouteInfo$1.smali	2012-03-23 19:37:01.645805095 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/RouteInfo$1.smali.original	2012-03-23 19:34:45.497322429 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 182
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 46,59 ****
--- 47,64 ----
      .prologue
      const/4 v6, 0x1
  
+     .line 184
      const/4 v2, 0x0
  
+     .line 185
      .local v2, destAddr:Ljava/net/InetAddress;
      const/4 v4, 0x0
  
+     .line 186
      .local v4, prefix:I
      const/4 v3, 0x0
  
+     .line 188
      .local v3, gateway:Ljava/net/InetAddress;
      invoke-virtual {p1}, Landroid/os/Parcel;->readByte()B
  
***************
*** 61,75 ****
--- 66,83 ----
  
      if-ne v5, v6, :cond_0
  
+     .line 189
      invoke-virtual {p1}, Landroid/os/Parcel;->createByteArray()[B
  
      move-result-object v0
  
+     .line 190
      .local v0, addr:[B
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
  
+     .line 193
      :try_start_0
      invoke-static {v0}, Ljava/net/InetAddress;->getByAddress([B)Ljava/net/InetAddress;
      :try_end_0
***************
*** 77,82 ****
--- 85,91 ----
  
      move-result-object v2
  
+     .line 197
      .end local v0           #addr:[B
      :cond_0
      :goto_0
***************
*** 86,95 ****
--- 95,106 ----
  
      if-ne v5, v6, :cond_1
  
+     .line 198
      invoke-virtual {p1}, Landroid/os/Parcel;->createByteArray()[B
  
      move-result-object v0
  
+     .line 201
      .restart local v0       #addr:[B
      :try_start_1
      invoke-static {v0}, Ljava/net/InetAddress;->getByAddress([B)Ljava/net/InetAddress;
***************
*** 98,116 ****
--- 109,131 ----
  
      move-result-object v3
  
+     .line 205
      .end local v0           #addr:[B
      :cond_1
      :goto_1
      const/4 v1, 0x0
  
+     .line 207
      .local v1, dest:Landroid/net/LinkAddress;
      if-eqz v2, :cond_2
  
+     .line 208
      new-instance v1, Landroid/net/LinkAddress;
  
      .end local v1           #dest:Landroid/net/LinkAddress;
      invoke-direct {v1, v2, v4}, Landroid/net/LinkAddress;-><init>(Ljava/net/InetAddress;I)V
  
+     .line 211
      .restart local v1       #dest:Landroid/net/LinkAddress;
      :cond_2
      new-instance v5, Landroid/net/RouteInfo;
***************
*** 119,124 ****
--- 134,140 ----
  
      return-object v5
  
+     .line 194
      .end local v1           #dest:Landroid/net/LinkAddress;
      .restart local v0       #addr:[B
      :catch_0
***************
*** 126,131 ****
--- 142,148 ----
  
      goto :goto_0
  
+     .line 202
      :catch_1
      move-exception v5
  
***************
*** 137,142 ****
--- 154,160 ----
      .parameter "x0"
  
      .prologue
+     .line 182
      invoke-virtual {p0, p1}, Landroid/net/RouteInfo$1;->createFromParcel(Landroid/os/Parcel;)Landroid/net/RouteInfo;
  
      move-result-object v0
***************
*** 149,154 ****
--- 167,173 ----
      .parameter "size"
  
      .prologue
+     .line 215
      new-array v0, p1, [Landroid/net/RouteInfo;
  
      return-object v0
***************
*** 159,164 ****
--- 178,184 ----
      .parameter "x0"
  
      .prologue
+     .line 182
      invoke-virtual {p0, p1}, Landroid/net/RouteInfo$1;->newArray(I)[Landroid/net/RouteInfo;
  
      move-result-object v0
