*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/SSLSessionCache.smali	2012-03-23 19:36:53.398654707 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/SSLSessionCache.smali.original	2012-03-23 19:34:45.390320479 -0500
***************
*** 17,25 ****
      .parameter "context"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
!     const-string v3, "sslcache"
  
      const/4 v4, 0x0
  
--- 17,27 ----
      .parameter "context"
  
      .prologue
+     .line 58
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
!     .line 59
!     const-string/jumbo v3, "sslcache"
  
      const/4 v4, 0x0
  
***************
*** 27,35 ****
--- 29,39 ----
  
      move-result-object v1
  
+     .line 60
      .local v1, dir:Ljava/io/File;
      const/4 v0, 0x0
  
+     .line 62
      .local v0, cache:Lorg/apache/harmony/xnet/provider/jsse/SSLClientSessionCache;
      :try_start_0
      invoke-static {v1}, Lorg/apache/harmony/xnet/provider/jsse/FileClientSessionCache;->usingDirectory(Ljava/io/File;)Lorg/apache/harmony/xnet/provider/jsse/SSLClientSessionCache;
***************
*** 38,51 ****
--- 42,59 ----
  
      move-result-object v0
  
+     .line 66
      :goto_0
      iput-object v0, p0, Landroid/net/SSLSessionCache;->mSessionCache:Lorg/apache/harmony/xnet/provider/jsse/SSLClientSessionCache;
  
+     .line 67
      return-void
  
+     .line 63
      :catch_0
      move-exception v2
  
+     .line 64
      .local v2, e:Ljava/io/IOException;
      const-string v3, "SSLSessionCache"
  
***************
*** 82,94 ****
--- 90,105 ----
      .end annotation
  
      .prologue
+     .line 48
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 49
      invoke-static {p1}, Lorg/apache/harmony/xnet/provider/jsse/FileClientSessionCache;->usingDirectory(Ljava/io/File;)Lorg/apache/harmony/xnet/provider/jsse/SSLClientSessionCache;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/SSLSessionCache;->mSessionCache:Lorg/apache/harmony/xnet/provider/jsse/SSLClientSessionCache;
  
+     .line 50
      return-void
  .end method
