*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/Uri$StringUri.smali	2012-03-23 19:37:02.637823183 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/Uri$StringUri.smali.original	2012-03-23 19:34:45.513322721 -0500
***************
*** 46,78 ****
      .prologue
      const/4 v1, -0x2
  
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/net/Uri$AbstractHierarchicalUri;-><init>(Landroid/net/Uri$1;)V
  
      iput v1, p0, Landroid/net/Uri$StringUri;->cachedSsi:I
  
      iput v1, p0, Landroid/net/Uri$StringUri;->cachedFsi:I
  
      invoke-static {}, Landroid/net/Uri;->access$300()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/Uri$StringUri;->scheme:Ljava/lang/String;
  
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "uriString"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iput-object p1, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
      return-void
  .end method
  
--- 46,86 ----
      .prologue
      const/4 v1, -0x2
  
+     .line 462
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/net/Uri$AbstractHierarchicalUri;-><init>(Landroid/net/Uri$1;)V
  
+     .line 484
      iput v1, p0, Landroid/net/Uri$StringUri;->cachedSsi:I
  
+     .line 494
      iput v1, p0, Landroid/net/Uri$StringUri;->cachedFsi:I
  
+     .line 525
      invoke-static {}, Landroid/net/Uri;->access$300()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/net/Uri$StringUri;->scheme:Ljava/lang/String;
  
+     .line 463
      if-nez p1, :cond_0
  
+     .line 464
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "uriString"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 467
      :cond_0
      iput-object p1, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
+     .line 468
      return-void
  .end method
  
***************
*** 82,87 ****
--- 90,96 ----
      .parameter "x1"
  
      .prologue
+     .line 454
      invoke-direct {p0, p1}, Landroid/net/Uri$StringUri;-><init>(Ljava/lang/String;)V
  
      return-void
***************
*** 91,96 ****
--- 100,106 ----
      .locals 3
  
      .prologue
+     .line 498
      iget v0, p0, Landroid/net/Uri$StringUri;->cachedFsi:I
  
      const/4 v1, -0x2
***************
*** 124,129 ****
--- 134,140 ----
      .locals 2
  
      .prologue
+     .line 488
      iget v0, p0, Landroid/net/Uri$StringUri;->cachedSsi:I
  
      const/4 v1, -0x2
***************
*** 153,162 ****
--- 164,175 ----
      .locals 3
  
      .prologue
+     .line 565
      iget-object v1, p0, Landroid/net/Uri$StringUri;->authority:Landroid/net/Uri$Part;
  
      if-nez v1, :cond_0
  
+     .line 566
      iget-object v1, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->findSchemeSeparator()I
***************
*** 167,172 ****
--- 180,186 ----
  
      move-result-object v0
  
+     .line 568
      .local v0, encodedAuthority:Ljava/lang/String;
      invoke-static {v0}, Landroid/net/Uri$Part;->fromEncoded(Ljava/lang/String;)Landroid/net/Uri$Part;
  
***************
*** 174,179 ****
--- 188,194 ----
  
      iput-object v1, p0, Landroid/net/Uri$StringUri;->authority:Landroid/net/Uri$Part;
  
+     .line 571
      .end local v0           #encodedAuthority:Ljava/lang/String;
      :goto_0
      return-object v1
***************
*** 188,193 ****
--- 203,209 ----
      .locals 1
  
      .prologue
+     .line 667
      iget-object v0, p0, Landroid/net/Uri$StringUri;->fragment:Landroid/net/Uri$Part;
  
      if-nez v0, :cond_0
***************
*** 215,220 ****
--- 231,237 ----
      .locals 1
  
      .prologue
+     .line 585
      iget-object v0, p0, Landroid/net/Uri$StringUri;->path:Landroid/net/Uri$PathPart;
  
      if-nez v0, :cond_0
***************
*** 242,247 ****
--- 259,265 ----
      .locals 1
  
      .prologue
+     .line 630
      iget-object v0, p0, Landroid/net/Uri$StringUri;->query:Landroid/net/Uri$Part;
  
      if-nez v0, :cond_0
***************
*** 269,274 ****
--- 287,293 ----
      .locals 1
  
      .prologue
+     .line 541
      iget-object v0, p0, Landroid/net/Uri$StringUri;->ssp:Landroid/net/Uri$Part;
  
      if-nez v0, :cond_0
***************
*** 300,309 ****
--- 319,330 ----
      .prologue
      const/16 v3, 0x2f
  
+     .line 697
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v1
  
+     .line 700
      .local v1, length:I
      add-int/lit8 v2, p1, 0x2
  
***************
*** 325,346 ****
--- 346,372 ----
  
      if-ne v2, v3, :cond_1
  
+     .line 707
      add-int/lit8 v0, p1, 0x3
  
+     .line 708
      .local v0, end:I
      :goto_0
      if-ge v0, v1, :cond_0
  
+     .line 709
      invoke-virtual {p0, v0}, Ljava/lang/String;->charAt(I)C
  
      move-result v2
  
      sparse-switch v2, :sswitch_data_0
  
+     .line 715
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 718
      :cond_0
      :sswitch_0
      add-int/lit8 v2, p1, 0x3
***************
*** 349,354 ****
--- 375,381 ----
  
      move-result-object v2
  
+     .line 720
      .end local v0           #end:I
      :goto_1
      return-object v2
***************
*** 358,363 ****
--- 385,391 ----
  
      goto :goto_1
  
+     .line 709
      nop
  
      :sswitch_data_0
***************
*** 372,381 ****
--- 400,411 ----
      .locals 3
  
      .prologue
+     .line 676
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->findFragmentSeparator()I
  
      move-result v0
  
+     .line 677
      .local v0, fsi:I
      const/4 v1, -0x1
  
***************
*** 404,421 ****
--- 434,455 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 603
      iget-object v2, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
+     .line 604
      .local v2, uriString:Ljava/lang/String;
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->findSchemeSeparator()I
  
      move-result v1
  
+     .line 607
      .local v1, ssi:I
      const/4 v4, -0x1
  
      if-le v1, v4, :cond_3
  
+     .line 609
      add-int/lit8 v4, v1, 0x1
  
      invoke-virtual {v2}, Ljava/lang/String;->length()I
***************
*** 426,445 ****
--- 460,483 ----
  
      const/4 v0, 0x1
  
+     .line 610
      .local v0, schemeOnly:Z
      :goto_0
      if-eqz v0, :cond_2
  
+     .line 624
      .end local v0           #schemeOnly:Z
      :cond_0
      :goto_1
      return-object v3
  
+     .line 609
      :cond_1
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 616
      .restart local v0       #schemeOnly:Z
      :cond_2
      add-int/lit8 v4, v1, 0x1
***************
*** 452,457 ****
--- 490,496 ----
  
      if-ne v4, v5, :cond_0
  
+     .line 624
      .end local v0           #schemeOnly:Z
      :cond_3
      invoke-static {v2, v1}, Landroid/net/Uri$StringUri;->parsePath(Ljava/lang/String;I)Ljava/lang/String;
***************
*** 469,478 ****
--- 508,519 ----
      .prologue
      const/16 v4, 0x2f
  
+     .line 734
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v0
  
+     .line 738
      .local v0, length:I
      add-int/lit8 v3, p1, 0x2
  
***************
*** 494,544 ****
--- 535,597 ----
  
      if-ne v3, v4, :cond_0
  
+     .line 742
      add-int/lit8 v2, p1, 0x3
  
+     .line 743
      .local v2, pathStart:I
      :goto_0
      if-ge v2, v0, :cond_1
  
+     .line 744
      invoke-virtual {p0, v2}, Ljava/lang/String;->charAt(I)C
  
      move-result v3
  
      sparse-switch v3, :sswitch_data_0
  
+     .line 751
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 747
      :sswitch_0
      const-string v3, ""
  
+     .line 769
      :goto_1
      return-object v3
  
+     .line 755
      .end local v2           #pathStart:I
      :cond_0
      add-int/lit8 v2, p1, 0x1
  
+     .line 759
      .restart local v2       #pathStart:I
      :cond_1
      :sswitch_1
      move v1, v2
  
+     .line 760
      .local v1, pathEnd:I
      :goto_2
      if-ge v1, v0, :cond_2
  
+     .line 761
      invoke-virtual {p0, v1}, Ljava/lang/String;->charAt(I)C
  
      move-result v3
  
      sparse-switch v3, :sswitch_data_1
  
+     .line 766
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_2
  
+     .line 769
      :cond_2
      :sswitch_2
      invoke-virtual {p0, v2, v1}, Ljava/lang/String;->substring(II)Ljava/lang/String;
***************
*** 547,552 ****
--- 600,606 ----
  
      goto :goto_1
  
+     .line 744
      nop
  
      :sswitch_data_0
***************
*** 556,561 ****
--- 610,616 ----
          0x3f -> :sswitch_0
      .end sparse-switch
  
+     .line 761
      :sswitch_data_1
      .sparse-switch
          0x23 -> :sswitch_2
***************
*** 571,576 ****
--- 626,632 ----
  
      const/4 v6, -0x1
  
+     .line 641
      iget-object v3, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
      const/16 v4, 0x3f
***************
*** 583,603 ****
--- 639,664 ----
  
      move-result v1
  
+     .line 642
      .local v1, qsi:I
      if-ne v1, v6, :cond_1
  
+     .line 657
      :cond_0
      :goto_0
      return-object v2
  
+     .line 646
      :cond_1
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->findFragmentSeparator()I
  
      move-result v0
  
+     .line 648
      .local v0, fsi:I
      if-ne v0, v6, :cond_2
  
+     .line 649
      iget-object v2, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
      add-int/lit8 v3, v1, 0x1
***************
*** 608,616 ****
--- 669,679 ----
  
      goto :goto_0
  
+     .line 652
      :cond_2
      if-lt v0, v1, :cond_0
  
+     .line 657
      iget-object v2, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
      add-int/lit8 v3, v1, 0x1
***************
*** 626,635 ****
--- 689,700 ----
      .locals 3
  
      .prologue
+     .line 534
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->findSchemeSeparator()I
  
      move-result v0
  
+     .line 535
      .local v0, ssi:I
      const/4 v1, -0x1
  
***************
*** 656,670 ****
--- 721,738 ----
      .locals 4
  
      .prologue
+     .line 553
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->findSchemeSeparator()I
  
      move-result v1
  
+     .line 554
      .local v1, ssi:I
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->findFragmentSeparator()I
  
      move-result v0
  
+     .line 557
      .local v0, fsi:I
      const/4 v2, -0x1
  
***************
*** 698,703 ****
--- 766,772 ----
      .parameter "parcel"
  
      .prologue
+     .line 471
      new-instance v0, Landroid/net/Uri$StringUri;
  
      invoke-virtual {p0}, Landroid/os/Parcel;->readString()Ljava/lang/String;
***************
*** 715,726 ****
--- 784,797 ----
      .locals 2
  
      .prologue
+     .line 773
      invoke-virtual {p0}, Landroid/net/Uri$StringUri;->isHierarchical()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 774
      new-instance v0, Landroid/net/Uri$Builder;
  
      invoke-direct {v0}, Landroid/net/Uri$Builder;-><init>()V
***************
*** 765,770 ****
--- 836,842 ----
  
      move-result-object v0
  
+     .line 781
      :goto_0
      return-object v0
  
***************
*** 804,809 ****
--- 876,882 ----
      .locals 1
  
      .prologue
+     .line 475
      const/4 v0, 0x0
  
      return v0
***************
*** 813,818 ****
--- 886,892 ----
      .locals 1
  
      .prologue
+     .line 579
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getAuthorityPart()Landroid/net/Uri$Part;
  
      move-result-object v0
***************
*** 828,833 ****
--- 902,908 ----
      .locals 1
  
      .prologue
+     .line 575
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getAuthorityPart()Landroid/net/Uri$Part;
  
      move-result-object v0
***************
*** 843,848 ****
--- 918,924 ----
      .locals 1
  
      .prologue
+     .line 672
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getFragmentPart()Landroid/net/Uri$Part;
  
      move-result-object v0
***************
*** 858,863 ****
--- 934,940 ----
      .locals 1
  
      .prologue
+     .line 595
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getPathPart()Landroid/net/Uri$PathPart;
  
      move-result-object v0
***************
*** 873,878 ****
--- 950,956 ----
      .locals 1
  
      .prologue
+     .line 635
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getQueryPart()Landroid/net/Uri$Part;
  
      move-result-object v0
***************
*** 888,893 ****
--- 966,972 ----
      .locals 1
  
      .prologue
+     .line 545
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getSsp()Landroid/net/Uri$Part;
  
      move-result-object v0
***************
*** 903,908 ****
--- 982,988 ----
      .locals 1
  
      .prologue
+     .line 681
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getFragmentPart()Landroid/net/Uri$Part;
  
      move-result-object v0
***************
*** 918,923 ****
--- 998,1004 ----
      .locals 1
  
      .prologue
+     .line 591
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getPathPart()Landroid/net/Uri$PathPart;
  
      move-result-object v0
***************
*** 942,947 ****
--- 1023,1029 ----
      .end annotation
  
      .prologue
+     .line 599
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getPathPart()Landroid/net/Uri$PathPart;
  
      move-result-object v0
***************
*** 957,962 ****
--- 1039,1045 ----
      .locals 1
  
      .prologue
+     .line 661
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getQueryPart()Landroid/net/Uri$Part;
  
      move-result-object v0
***************
*** 972,977 ****
--- 1055,1061 ----
      .locals 3
  
      .prologue
+     .line 529
      iget-object v1, p0, Landroid/net/Uri$StringUri;->scheme:Ljava/lang/String;
  
      invoke-static {}, Landroid/net/Uri;->access$300()Ljava/lang/String;
***************
*** 982,987 ****
--- 1066,1072 ----
  
      const/4 v0, 0x1
  
+     .line 530
      .local v0, cached:Z
      :goto_0
      if-eqz v0, :cond_1
***************
*** 991,1002 ****
--- 1076,1089 ----
      :goto_1
      return-object v1
  
+     .line 529
      .end local v0           #cached:Z
      :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 530
      .restart local v0       #cached:Z
      :cond_1
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->parseScheme()Ljava/lang/String;
***************
*** 1012,1017 ****
--- 1099,1105 ----
      .locals 1
  
      .prologue
+     .line 549
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->getSsp()Landroid/net/Uri$Part;
  
      move-result-object v0
***************
*** 1031,1049 ****
--- 1119,1141 ----
  
      const/4 v2, 0x0
  
+     .line 504
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->findSchemeSeparator()I
  
      move-result v0
  
+     .line 506
      .local v0, ssi:I
      const/4 v3, -0x1
  
      if-ne v0, v3, :cond_1
  
+     .line 517
      :cond_0
      :goto_0
      return v1
  
+     .line 511
      :cond_1
      iget-object v3, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
***************
*** 1057,1064 ****
--- 1149,1158 ----
  
      move v1, v2
  
+     .line 513
      goto :goto_0
  
+     .line 517
      :cond_2
      iget-object v3, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
***************
*** 1081,1086 ****
--- 1175,1181 ----
      .locals 2
  
      .prologue
+     .line 522
      invoke-direct {p0}, Landroid/net/Uri$StringUri;->findSchemeSeparator()I
  
      move-result v0
***************
*** 1104,1109 ****
--- 1199,1205 ----
      .locals 1
  
      .prologue
+     .line 685
      iget-object v0, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
      return-object v0
***************
*** 1115,1127 ****
--- 1211,1226 ----
      .parameter "flags"
  
      .prologue
+     .line 479
      const/4 v0, 0x1
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 480
      iget-object v0, p0, Landroid/net/Uri$StringUri;->uriString:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 481
      return-void
  .end method
