*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/VpnService.smali	2012-03-23 19:37:02.663823658 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/VpnService.smali.original	2012-03-23 19:34:45.513322721 -0500
***************
*** 22,29 ****
--- 22,31 ----
      .locals 0
  
      .prologue
+     .line 105
      invoke-direct {p0}, Landroid/app/Service;-><init>()V
  
+     .line 253
      return-void
  .end method
  
***************
*** 31,36 ****
--- 33,39 ----
      .locals 1
  
      .prologue
+     .line 105
      invoke-static {}, Landroid/net/VpnService;->getService()Landroid/net/IConnectivityManager;
  
      move-result-object v0
***************
*** 42,47 ****
--- 45,51 ----
      .locals 1
  
      .prologue
+     .line 119
      const-string v0, "connectivity"
  
      invoke-static {v0}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
***************
*** 62,67 ****
--- 66,72 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 143
      :try_start_0
      invoke-static {}, Landroid/net/VpnService;->getService()Landroid/net/IConnectivityManager;
  
***************
*** 81,92 ****
--- 86,100 ----
  
      if-eqz v1, :cond_0
  
+     .line 149
      :goto_0
      return-object v0
  
+     .line 146
      :catch_0
      move-exception v0
  
+     .line 149
      :cond_0
      invoke-static {}, Lcom/android/internal/net/VpnConfig;->getIntentForConfirmation()Landroid/content/Intent;
  
***************
*** 104,109 ****
--- 112,118 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 212
      if-eqz p1, :cond_0
  
      const-string v0, "android.net.VpnService"
***************
*** 118,127 ****
--- 127,138 ----
  
      if-eqz v0, :cond_0
  
+     .line 213
      new-instance v0, Landroid/net/VpnService$Callback;
  
      invoke-direct {v0, p0, v1}, Landroid/net/VpnService$Callback;-><init>(Landroid/net/VpnService;Landroid/net/VpnService$1;)V
  
+     .line 215
      :goto_0
      return-object v0
  
***************
*** 135,142 ****
--- 146,155 ----
      .locals 0
  
      .prologue
+     .line 230
      invoke-virtual {p0}, Landroid/net/VpnService;->stopSelf()V
  
+     .line 231
      return-void
  .end method
  
***************
*** 145,158 ****
--- 158,174 ----
      .parameter "socket"
  
      .prologue
+     .line 166
      const/4 v0, 0x0
  
+     .line 168
      .local v0, dup:Landroid/os/ParcelFileDescriptor;
      :try_start_0
      invoke-static {p1}, Landroid/os/ParcelFileDescriptor;->fromFd(I)Landroid/os/ParcelFileDescriptor;
  
      move-result-object v0
  
+     .line 169
      invoke-static {}, Landroid/net/VpnService;->getService()Landroid/net/IConnectivityManager;
  
      move-result-object v2
***************
*** 164,183 ****
--- 180,204 ----
  
      move-result v2
  
+     .line 174
      :try_start_1
      invoke-virtual {v0}, Landroid/os/ParcelFileDescriptor;->close()V
      :try_end_1
      .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_2
  
+     .line 177
      :goto_0
      return v2
  
+     .line 170
      :catch_0
      move-exception v1
  
+     .line 171
      .local v1, e:Ljava/lang/Exception;
      const/4 v2, 0x0
  
+     .line 174
      :try_start_2
      invoke-virtual {v0}, Landroid/os/ParcelFileDescriptor;->close()V
      :try_end_2
***************
*** 185,207 ****
--- 206,233 ----
  
      goto :goto_0
  
+     .line 175
      :catch_1
      move-exception v3
  
      goto :goto_0
  
+     .line 173
      .end local v1           #e:Ljava/lang/Exception;
      :catchall_0
      move-exception v2
  
+     .line 174
      :try_start_3
      invoke-virtual {v0}, Landroid/os/ParcelFileDescriptor;->close()V
      :try_end_3
      .catch Ljava/lang/Exception; {:try_start_3 .. :try_end_3} :catch_3
  
+     .line 177
      :goto_1
      throw v2
  
+     .line 175
      :catch_2
      move-exception v3
  
***************
*** 218,223 ****
--- 244,250 ----
      .parameter "socket"
  
      .prologue
+     .line 199
      invoke-virtual {p1}, Ljava/net/DatagramSocket;->getFileDescriptor$()Ljava/io/FileDescriptor;
  
      move-result-object v0
***************
*** 238,243 ****
--- 265,271 ----
      .parameter "socket"
  
      .prologue
+     .line 188
      invoke-virtual {p1}, Ljava/net/Socket;->getFileDescriptor$()Ljava/io/FileDescriptor;
  
      move-result-object v0
