*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/WebAddress.smali	2012-03-23 19:36:59.580767439 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/WebAddress.smali.original	2012-03-23 19:34:45.469321919 -0500
***************
*** 34,39 ****
--- 34,40 ----
      .locals 2
  
      .prologue
+     .line 54
      const-string v0, "(?:(http|https|file)\\:\\/\\/)?(?:([-A-Za-z0-9$_.+!*\'(),;?&=]+(?:\\:[-A-Za-z0-9$_.+!*\'(),;?&=]+)?)@)?([a-zA-Z0-9\u00a0-\ud7ff\uf900-\ufdcf\ufdf0-\uffef%_-][a-zA-Z0-9\u00a0-\ud7ff\uf900-\ufdcf\ufdf0-\uffef%_\\.-]*|\\[[0-9a-fA-F:\\.]+\\])?(?:\\:([0-9]*))?(\\/?[^#]*)?.*"
  
      const/4 v1, 0x2
***************
*** 61,101 ****
--- 62,112 ----
  
      const/4 v5, -0x1
  
+     .line 63
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 64
      if-nez p1, :cond_0
  
+     .line 65
      new-instance v3, Ljava/lang/NullPointerException;
  
      invoke-direct {v3}, Ljava/lang/NullPointerException;-><init>()V
  
      throw v3
  
+     .line 70
      :cond_0
      const-string v3, ""
  
      iput-object v3, p0, Landroid/net/WebAddress;->mScheme:Ljava/lang/String;
  
+     .line 71
      const-string v3, ""
  
      iput-object v3, p0, Landroid/net/WebAddress;->mHost:Ljava/lang/String;
  
+     .line 72
      iput v5, p0, Landroid/net/WebAddress;->mPort:I
  
+     .line 73
      const-string v3, "/"
  
      iput-object v3, p0, Landroid/net/WebAddress;->mPath:Ljava/lang/String;
  
+     .line 74
      const-string v3, ""
  
      iput-object v3, p0, Landroid/net/WebAddress;->mAuthInfo:Ljava/lang/String;
  
+     .line 76
      sget-object v3, Landroid/net/WebAddress;->sAddressPattern:Ljava/util/regex/Pattern;
  
      invoke-virtual {v3, p1}, Ljava/util/regex/Pattern;->matcher(Ljava/lang/CharSequence;)Ljava/util/regex/Matcher;
  
      move-result-object v1
  
+     .line 78
      .local v1, m:Ljava/util/regex/Matcher;
      invoke-virtual {v1}, Ljava/util/regex/Matcher;->matches()Z
  
***************
*** 103,114 ****
--- 114,127 ----
  
      if-eqz v3, :cond_9
  
+     .line 79
      const/4 v3, 0x1
  
      invoke-virtual {v1, v3}, Ljava/util/regex/Matcher;->group(I)Ljava/lang/String;
  
      move-result-object v2
  
+     .line 80
      .local v2, t:Ljava/lang/String;
      if-eqz v2, :cond_1
  
***************
*** 118,123 ****
--- 131,137 ----
  
      iput-object v3, p0, Landroid/net/WebAddress;->mScheme:Ljava/lang/String;
  
+     .line 81
      :cond_1
      const/4 v3, 0x2
  
***************
*** 125,134 ****
--- 139,150 ----
  
      move-result-object v2
  
+     .line 82
      if-eqz v2, :cond_2
  
      iput-object v2, p0, Landroid/net/WebAddress;->mAuthInfo:Ljava/lang/String;
  
+     .line 83
      :cond_2
      const/4 v3, 0x3
  
***************
*** 136,145 ****
--- 152,163 ----
  
      move-result-object v2
  
+     .line 84
      if-eqz v2, :cond_3
  
      iput-object v2, p0, Landroid/net/WebAddress;->mHost:Ljava/lang/String;
  
+     .line 85
      :cond_3
      const/4 v3, 0x4
  
***************
*** 147,152 ****
--- 165,171 ----
  
      move-result-object v2
  
+     .line 86
      if-eqz v2, :cond_4
  
      invoke-virtual {v2}, Ljava/lang/String;->length()I
***************
*** 155,160 ****
--- 174,180 ----
  
      if-lez v3, :cond_4
  
+     .line 89
      :try_start_0
      invoke-static {v2}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
  
***************
*** 164,169 ****
--- 184,190 ----
      :try_end_0
      .catch Ljava/lang/NumberFormatException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 94
      :cond_4
      const/4 v3, 0x5
  
***************
*** 171,176 ****
--- 192,198 ----
  
      move-result-object v2
  
+     .line 95
      if-eqz v2, :cond_5
  
      invoke-virtual {v2}, Ljava/lang/String;->length()I
***************
*** 179,184 ****
--- 201,207 ----
  
      if-lez v3, :cond_5
  
+     .line 98
      const/4 v3, 0x0
  
      invoke-virtual {v2, v3}, Ljava/lang/String;->charAt(I)C
***************
*** 189,196 ****
--- 212,221 ----
  
      if-ne v3, v4, :cond_8
  
+     .line 99
      iput-object v2, p0, Landroid/net/WebAddress;->mPath:Ljava/lang/String;
  
+     .line 112
      :cond_5
      :goto_0
      iget v3, p0, Landroid/net/WebAddress;->mPort:I
***************
*** 207,216 ****
--- 232,243 ----
  
      if-eqz v3, :cond_a
  
+     .line 113
      const-string v3, "https"
  
      iput-object v3, p0, Landroid/net/WebAddress;->mScheme:Ljava/lang/String;
  
+     .line 120
      :cond_6
      :goto_1
      iget-object v3, p0, Landroid/net/WebAddress;->mScheme:Ljava/lang/String;
***************
*** 227,238 ****
--- 254,268 ----
  
      iput-object v3, p0, Landroid/net/WebAddress;->mScheme:Ljava/lang/String;
  
+     .line 121
      :cond_7
      return-void
  
+     .line 90
      :catch_0
      move-exception v0
  
+     .line 91
      .local v0, ex:Ljava/lang/NumberFormatException;
      new-instance v3, Landroid/net/ParseException;
  
***************
*** 242,247 ****
--- 272,278 ----
  
      throw v3
  
+     .line 101
      .end local v0           #ex:Ljava/lang/NumberFormatException;
      :cond_8
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 266,271 ****
--- 297,303 ----
  
      goto :goto_0
  
+     .line 107
      .end local v2           #t:Ljava/lang/String;
      :cond_9
      new-instance v3, Landroid/net/ParseException;
***************
*** 276,287 ****
--- 308,321 ----
  
      throw v3
  
+     .line 114
      .restart local v2       #t:Ljava/lang/String;
      :cond_a
      iget v3, p0, Landroid/net/WebAddress;->mPort:I
  
      if-ne v3, v5, :cond_6
  
+     .line 115
      iget-object v3, p0, Landroid/net/WebAddress;->mScheme:Ljava/lang/String;
  
      const-string v4, "https"
***************
*** 292,301 ****
--- 326,337 ----
  
      if-eqz v3, :cond_b
  
+     .line 116
      iput v6, p0, Landroid/net/WebAddress;->mPort:I
  
      goto :goto_1
  
+     .line 118
      :cond_b
      const/16 v3, 0x50
  
***************
*** 310,315 ****
--- 346,352 ----
      .locals 1
  
      .prologue
+     .line 175
      iget-object v0, p0, Landroid/net/WebAddress;->mAuthInfo:Ljava/lang/String;
  
      return-object v0
***************
*** 319,324 ****
--- 356,362 ----
      .locals 1
  
      .prologue
+     .line 151
      iget-object v0, p0, Landroid/net/WebAddress;->mHost:Ljava/lang/String;
  
      return-object v0
***************
*** 328,333 ****
--- 366,372 ----
      .locals 1
  
      .prologue
+     .line 167
      iget-object v0, p0, Landroid/net/WebAddress;->mPath:Ljava/lang/String;
  
      return-object v0
***************
*** 337,342 ****
--- 376,382 ----
      .locals 1
  
      .prologue
+     .line 159
      iget v0, p0, Landroid/net/WebAddress;->mPort:I
  
      return v0
***************
*** 346,351 ****
--- 386,392 ----
      .locals 1
  
      .prologue
+     .line 143
      iget-object v0, p0, Landroid/net/WebAddress;->mScheme:Ljava/lang/String;
  
      return-object v0
***************
*** 356,363 ****
--- 397,406 ----
      .parameter "authInfo"
  
      .prologue
+     .line 171
      iput-object p1, p0, Landroid/net/WebAddress;->mAuthInfo:Ljava/lang/String;
  
+     .line 172
      return-void
  .end method
  
***************
*** 366,373 ****
--- 409,418 ----
      .parameter "host"
  
      .prologue
+     .line 147
      iput-object p1, p0, Landroid/net/WebAddress;->mHost:Ljava/lang/String;
  
+     .line 148
      return-void
  .end method
  
***************
*** 376,383 ****
--- 421,430 ----
      .parameter "path"
  
      .prologue
+     .line 163
      iput-object p1, p0, Landroid/net/WebAddress;->mPath:Ljava/lang/String;
  
+     .line 164
      return-void
  .end method
  
***************
*** 386,393 ****
--- 433,442 ----
      .parameter "port"
  
      .prologue
+     .line 155
      iput p1, p0, Landroid/net/WebAddress;->mPort:I
  
+     .line 156
      return-void
  .end method
  
***************
*** 396,403 ****
--- 445,454 ----
      .parameter "scheme"
  
      .prologue
+     .line 139
      iput-object p1, p0, Landroid/net/WebAddress;->mScheme:Ljava/lang/String;
  
+     .line 140
      return-void
  .end method
  
***************
*** 405,412 ****
--- 456,465 ----
      .locals 4
  
      .prologue
+     .line 125
      const-string v1, ""
  
+     .line 126
      .local v1, port:Ljava/lang/String;
      iget v2, p0, Landroid/net/WebAddress;->mPort:I
  
***************
*** 441,446 ****
--- 494,500 ----
  
      if-eqz v2, :cond_2
  
+     .line 128
      :cond_1
      new-instance v2, Ljava/lang/StringBuilder;
  
***************
*** 466,474 ****
--- 520,530 ----
  
      move-result-object v1
  
+     .line 130
      :cond_2
      const-string v0, ""
  
+     .line 131
      .local v0, authInfo:Ljava/lang/String;
      iget-object v2, p0, Landroid/net/WebAddress;->mAuthInfo:Ljava/lang/String;
  
***************
*** 478,483 ****
--- 534,540 ----
  
      if-lez v2, :cond_3
  
+     .line 132
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 498,503 ****
--- 555,561 ----
  
      move-result-object v0
  
+     .line 135
      :cond_3
      new-instance v2, Ljava/lang/StringBuilder;
  
