*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLES10.smali	2012-03-23 19:36:13.320923892 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLES10.smali.original	2012-03-23 19:34:45.019313713 -0500
***************
*** 492,499 ****
--- 492,501 ----
      .locals 0
  
      .prologue
+     .line 265
      invoke-static {}, Landroid/opengl/GLES10;->_nativeClassInit()V
  
+     .line 266
      return-void
  .end method
  
***************
*** 501,506 ****
--- 503,509 ----
      .locals 0
  
      .prologue
+     .line 24
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 562,573 ****
--- 565,578 ----
      .parameter "pointer"
  
      .prologue
+     .line 398
      invoke-virtual {p3}, Ljava/nio/Buffer;->remaining()I
  
      move-result v0
  
      invoke-static {p0, p1, p2, p3, v0}, Landroid/opengl/GLES10;->glColorPointerBounds(IIILjava/nio/Buffer;I)V
  
+     .line 405
      const/4 v0, 0x4
  
      if-ne p0, v0, :cond_1
***************
*** 587,594 ****
--- 592,601 ----
      :cond_0
      if-ltz p2, :cond_1
  
+     .line 410
      sput-object p3, Landroid/opengl/GLES10;->_colorPointer:Ljava/nio/Buffer;
  
+     .line 412
      :cond_1
      return-void
  .end method
***************
*** 813,824 ****
--- 820,833 ----
      .parameter "pointer"
  
      .prologue
+     .line 975
      invoke-virtual {p2}, Ljava/nio/Buffer;->remaining()I
  
      move-result v0
  
      invoke-static {p0, p1, p2, v0}, Landroid/opengl/GLES10;->glNormalPointerBounds(IILjava/nio/Buffer;I)V
  
+     .line 981
      const/16 v0, 0x1406
  
      if-eq p0, v0, :cond_0
***************
*** 838,845 ****
--- 847,856 ----
      :cond_0
      if-ltz p1, :cond_1
  
+     .line 986
      sput-object p2, Landroid/opengl/GLES10;->_normalPointer:Ljava/nio/Buffer;
  
+     .line 988
      :cond_1
      return-void
  .end method
***************
*** 918,929 ****
--- 929,942 ----
      .parameter "pointer"
  
      .prologue
+     .line 1168
      invoke-virtual {p3}, Ljava/nio/Buffer;->remaining()I
  
      move-result v0
  
      invoke-static {p0, p1, p2, p3, v0}, Landroid/opengl/GLES10;->glTexCoordPointerBounds(IIILjava/nio/Buffer;I)V
  
+     .line 1175
      const/4 v0, 0x2
  
      if-eq p0, v0, :cond_0
***************
*** 956,963 ****
--- 969,978 ----
      :cond_1
      if-ltz p2, :cond_2
  
+     .line 1183
      sput-object p3, Landroid/opengl/GLES10;->_texCoordPointer:Ljava/nio/Buffer;
  
+     .line 1185
      :cond_2
      return-void
  .end method
***************
*** 1009,1020 ****
--- 1024,1037 ----
      .parameter "pointer"
  
      .prologue
+     .line 1313
      invoke-virtual {p3}, Ljava/nio/Buffer;->remaining()I
  
      move-result v0
  
      invoke-static {p0, p1, p2, p3, v0}, Landroid/opengl/GLES10;->glVertexPointerBounds(IIILjava/nio/Buffer;I)V
  
+     .line 1320
      const/4 v0, 0x2
  
      if-eq p0, v0, :cond_0
***************
*** 1047,1054 ****
--- 1064,1073 ----
      :cond_1
      if-ltz p2, :cond_2
  
+     .line 1328
      sput-object p3, Landroid/opengl/GLES10;->_vertexPointer:Ljava/nio/Buffer;
  
+     .line 1330
      :cond_2
      return-void
  .end method
