*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLES11.smali	2012-03-23 19:36:13.637929672 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLES11.smali.original	2012-03-23 19:34:45.020313731 -0500
***************
*** 256,263 ****
--- 256,265 ----
      .locals 0
  
      .prologue
+     .line 150
      invoke-static {}, Landroid/opengl/GLES11;->_nativeClassInit()V
  
+     .line 151
      return-void
  .end method
  
***************
*** 265,270 ****
--- 267,273 ----
      .locals 0
  
      .prologue
+     .line 24
      invoke-direct {p0}, Landroid/opengl/GLES10;-><init>()V
  
      return-void
***************
*** 448,459 ****
--- 451,464 ----
      .parameter "pointer"
  
      .prologue
+     .line 612
      invoke-virtual {p2}, Ljava/nio/Buffer;->remaining()I
  
      move-result v0
  
      invoke-static {p0, p1, p2, v0}, Landroid/opengl/GLES11;->glPointSizePointerOESBounds(IILjava/nio/Buffer;I)V
  
+     .line 618
      const/16 v0, 0x1406
  
      if-eq p0, v0, :cond_0
***************
*** 465,472 ****
--- 470,479 ----
      :cond_0
      if-ltz p1, :cond_1
  
+     .line 621
      sput-object p2, Landroid/opengl/GLES11;->_pointSizePointerOES:Ljava/nio/Buffer;
  
+     .line 623
      :cond_1
      return-void
  .end method
