*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLES11Ext.smali	2012-03-23 19:36:14.830951426 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLES11Ext.smali.original	2012-03-23 19:34:45.026313841 -0500
***************
*** 234,241 ****
--- 234,243 ----
      .locals 0
  
      .prologue
+     .line 135
      invoke-static {}, Landroid/opengl/GLES11Ext;->_nativeClassInit()V
  
+     .line 136
      return-void
  .end method
  
***************
*** 243,248 ****
--- 245,251 ----
      .locals 0
  
      .prologue
+     .line 24
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 520,531 ****
--- 523,536 ----
      .parameter "pointer"
  
      .prologue
+     .line 895
      invoke-virtual {p3}, Ljava/nio/Buffer;->remaining()I
  
      move-result v0
  
      invoke-static {p0, p1, p2, p3, v0}, Landroid/opengl/GLES11Ext;->glMatrixIndexPointerOESBounds(IIILjava/nio/Buffer;I)V
  
+     .line 902
      const/4 v0, 0x2
  
      if-eq p0, v0, :cond_0
***************
*** 558,565 ****
--- 563,572 ----
      :cond_1
      if-ltz p2, :cond_2
  
+     .line 910
      sput-object p3, Landroid/opengl/GLES11Ext;->_matrixIndexPointerOES:Ljava/nio/Buffer;
  
+     .line 912
      :cond_2
      return-void
  .end method
***************
*** 668,679 ****
--- 675,688 ----
      .parameter "pointer"
  
      .prologue
+     .line 930
      invoke-virtual {p3}, Ljava/nio/Buffer;->remaining()I
  
      move-result v0
  
      invoke-static {p0, p1, p2, p3, v0}, Landroid/opengl/GLES11Ext;->glWeightPointerOESBounds(IIILjava/nio/Buffer;I)V
  
+     .line 937
      return-void
  .end method
  
