*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLSurfaceView$LogWriter.smali	2012-03-23 19:36:14.168939354 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLSurfaceView$LogWriter.smali.original	2012-03-23 19:34:45.023313786 -0500
***************
*** 23,30 ****
--- 23,32 ----
      .locals 1
  
      .prologue
+     .line 1678
      invoke-direct {p0}, Ljava/io/Writer;-><init>()V
  
+     .line 1707
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 38,43 ****
--- 40,46 ----
      .locals 3
  
      .prologue
+     .line 1701
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$LogWriter;->mBuilder:Ljava/lang/StringBuilder;
  
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->length()I
***************
*** 46,51 ****
--- 49,55 ----
  
      if-lez v0, :cond_0
  
+     .line 1702
      const-string v0, "GLSurfaceView"
  
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$LogWriter;->mBuilder:Ljava/lang/StringBuilder;
***************
*** 56,61 ****
--- 60,66 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1703
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$LogWriter;->mBuilder:Ljava/lang/StringBuilder;
  
      const/4 v1, 0x0
***************
*** 68,73 ****
--- 73,79 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/lang/StringBuilder;->delete(II)Ljava/lang/StringBuilder;
  
+     .line 1705
      :cond_0
      return-void
  .end method
***************
*** 78,85 ****
--- 84,93 ----
      .locals 0
  
      .prologue
+     .line 1681
      invoke-direct {p0}, Landroid/opengl/GLSurfaceView$LogWriter;->flushBuilder()V
  
+     .line 1682
      return-void
  .end method
  
***************
*** 87,94 ****
--- 95,104 ----
      .locals 0
  
      .prologue
+     .line 1685
      invoke-direct {p0}, Landroid/opengl/GLSurfaceView$LogWriter;->flushBuilder()V
  
+     .line 1686
      return-void
  .end method
  
***************
*** 99,126 ****
--- 109,142 ----
      .parameter "count"
  
      .prologue
+     .line 1689
      const/4 v1, 0x0
  
      .local v1, i:I
      :goto_0
      if-ge v1, p3, :cond_1
  
+     .line 1690
      add-int v2, p2, v1
  
      aget-char v0, p1, v2
  
+     .line 1691
      .local v0, c:C
      const/16 v2, 0xa
  
      if-ne v0, v2, :cond_0
  
+     .line 1692
      invoke-direct {p0}, Landroid/opengl/GLSurfaceView$LogWriter;->flushBuilder()V
  
+     .line 1689
      :goto_1
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 1695
      :cond_0
      iget-object v2, p0, Landroid/opengl/GLSurfaceView$LogWriter;->mBuilder:Ljava/lang/StringBuilder;
  
***************
*** 128,133 ****
--- 144,150 ----
  
      goto :goto_1
  
+     .line 1698
      .end local v0           #c:C
      :cond_1
      return-void
