*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLSurfaceView$EglHelper.smali	2012-03-23 19:36:14.298941725 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLSurfaceView$EglHelper.smali.original	2012-03-23 19:34:45.023313786 -0500
***************
*** 34,43 ****
--- 34,45 ----
      .parameter
  
      .prologue
+     .line 976
      iput-object p1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 978
      return-void
  .end method
  
***************
*** 46,51 ****
--- 48,54 ----
      .parameter "function"
  
      .prologue
+     .line 1170
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      invoke-interface {v0}, Ljavax/microedition/khronos/egl/EGL10;->eglGetError()I
***************
*** 54,59 ****
--- 57,63 ----
  
      invoke-direct {p0, p1, v0}, Landroid/opengl/GLSurfaceView$EglHelper;->throwEglException(Ljava/lang/String;I)V
  
+     .line 1171
      return-void
  .end method
  
***************
*** 63,68 ****
--- 67,73 ----
      .parameter "error"
  
      .prologue
+     .line 1174
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 89,94 ****
--- 94,100 ----
  
      move-result-object v0
  
+     .line 1178
      .local v0, message:Ljava/lang/String;
      new-instance v1, Ljava/lang/RuntimeException;
  
***************
*** 104,113 ****
--- 110,121 ----
      .parameter "holder"
  
      .prologue
+     .line 1038
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      if-nez v4, :cond_0
  
+     .line 1039
      new-instance v4, Ljava/lang/RuntimeException;
  
      const-string v5, "egl not initialized"
***************
*** 116,126 ****
--- 124,136 ----
  
      throw v4
  
+     .line 1041
      :cond_0
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
  
      if-nez v4, :cond_1
  
+     .line 1042
      new-instance v4, Ljava/lang/RuntimeException;
  
      const-string v5, "eglDisplay not initialized"
***************
*** 129,139 ****
--- 139,151 ----
  
      throw v4
  
+     .line 1044
      :cond_1
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglConfig:Ljavax/microedition/khronos/egl/EGLConfig;
  
      if-nez v4, :cond_2
  
+     .line 1045
      new-instance v4, Ljava/lang/RuntimeException;
  
      const-string v5, "mEglConfig not initialized"
***************
*** 142,147 ****
--- 154,160 ----
  
      throw v4
  
+     .line 1051
      :cond_2
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglSurface:Ljavax/microedition/khronos/egl/EGLSurface;
  
***************
*** 153,158 ****
--- 166,172 ----
  
      if-eq v4, v5, :cond_3
  
+     .line 1057
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      iget-object v5, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
***************
*** 165,170 ****
--- 179,185 ----
  
      invoke-interface {v4, v5, v6, v7, v8}, Ljavax/microedition/khronos/egl/EGL10;->eglMakeCurrent(Ljavax/microedition/khronos/egl/EGLDisplay;Ljavax/microedition/khronos/egl/EGLSurface;Ljavax/microedition/khronos/egl/EGLSurface;Ljavax/microedition/khronos/egl/EGLContext;)Z
  
+     .line 1059
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
      #getter for: Landroid/opengl/GLSurfaceView;->mEGLWindowSurfaceFactory:Landroid/opengl/GLSurfaceView$EGLWindowSurfaceFactory;
***************
*** 180,185 ****
--- 195,201 ----
  
      invoke-interface {v4, v5, v6, v7}, Landroid/opengl/GLSurfaceView$EGLWindowSurfaceFactory;->destroySurface(Ljavax/microedition/khronos/egl/EGL10;Ljavax/microedition/khronos/egl/EGLDisplay;Ljavax/microedition/khronos/egl/EGLSurface;)V
  
+     .line 1065
      :cond_3
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
***************
*** 200,205 ****
--- 216,222 ----
  
      iput-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglSurface:Ljavax/microedition/khronos/egl/EGLSurface;
  
+     .line 1068
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglSurface:Ljavax/microedition/khronos/egl/EGLSurface;
  
      if-eqz v4, :cond_4
***************
*** 210,215 ****
--- 227,233 ----
  
      if-ne v4, v5, :cond_7
  
+     .line 1069
      :cond_4
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
***************
*** 217,241 ****
--- 235,264 ----
  
      move-result v1
  
+     .line 1070
      .local v1, error:I
      const/16 v4, 0x300b
  
      if-ne v1, v4, :cond_5
  
+     .line 1071
      const-string v4, "EglHelper"
  
      const-string v5, "createWindowSurface returned EGL_BAD_NATIVE_WINDOW."
  
      invoke-static {v4, v5}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1073
      :cond_5
      const/4 v2, 0x0
  
+     .line 1100
      .end local v1           #error:I
      :cond_6
      :goto_0
      return-object v2
  
+     .line 1080
      :cond_7
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
***************
*** 253,262 ****
--- 276,287 ----
  
      if-nez v4, :cond_8
  
+     .line 1081
      const-string v4, "eglMakeCurrent"
  
      invoke-direct {p0, v4}, Landroid/opengl/GLSurfaceView$EglHelper;->throwEglException(Ljava/lang/String;)V
  
+     .line 1084
      :cond_8
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglContext:Ljavax/microedition/khronos/egl/EGLContext;
  
***************
*** 264,269 ****
--- 289,295 ----
  
      move-result-object v2
  
+     .line 1085
      .local v2, gl:Ljavax/microedition/khronos/opengles/GL;
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
***************
*** 274,279 ****
--- 300,306 ----
  
      if-eqz v4, :cond_9
  
+     .line 1086
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
      #getter for: Landroid/opengl/GLSurfaceView;->mGLWrapper:Landroid/opengl/GLSurfaceView$GLWrapper;
***************
*** 285,290 ****
--- 312,318 ----
  
      move-result-object v2
  
+     .line 1089
      :cond_9
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
***************
*** 297,307 ****
--- 325,338 ----
  
      if-eqz v4, :cond_6
  
+     .line 1090
      const/4 v0, 0x0
  
+     .line 1091
      .local v0, configFlags:I
      const/4 v3, 0x0
  
+     .line 1092
      .local v3, log:Ljava/io/Writer;
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
***************
*** 314,321 ****
--- 345,354 ----
  
      if-eqz v4, :cond_a
  
+     .line 1093
      or-int/lit8 v0, v0, 0x1
  
+     .line 1095
      :cond_a
      iget-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
***************
*** 328,338 ****
--- 361,373 ----
  
      if-eqz v4, :cond_b
  
+     .line 1096
      new-instance v3, Landroid/opengl/GLSurfaceView$LogWriter;
  
      .end local v3           #log:Ljava/io/Writer;
      invoke-direct {v3}, Landroid/opengl/GLSurfaceView$LogWriter;-><init>()V
  
+     .line 1098
      .restart local v3       #log:Ljava/io/Writer;
      :cond_b
      invoke-static {v2, v0, v3}, Landroid/opengl/GLDebugHelper;->wrap(Ljavax/microedition/khronos/opengles/GL;ILjava/io/Writer;)Ljavax/microedition/khronos/opengles/GL;
***************
*** 346,351 ****
--- 381,387 ----
      .locals 5
  
      .prologue
+     .line 1146
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglSurface:Ljavax/microedition/khronos/egl/EGLSurface;
  
      if-eqz v0, :cond_0
***************
*** 356,361 ****
--- 392,398 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 1147
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
***************
*** 368,373 ****
--- 405,411 ----
  
      invoke-interface {v0, v1, v2, v3, v4}, Ljavax/microedition/khronos/egl/EGL10;->eglMakeCurrent(Ljavax/microedition/khronos/egl/EGLDisplay;Ljavax/microedition/khronos/egl/EGLSurface;Ljavax/microedition/khronos/egl/EGLSurface;Ljavax/microedition/khronos/egl/EGLContext;)Z
  
+     .line 1150
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
      #getter for: Landroid/opengl/GLSurfaceView;->mEGLWindowSurfaceFactory:Landroid/opengl/GLSurfaceView$EGLWindowSurfaceFactory;
***************
*** 383,392 ****
--- 421,432 ----
  
      invoke-interface {v0, v1, v2, v3}, Landroid/opengl/GLSurfaceView$EGLWindowSurfaceFactory;->destroySurface(Ljavax/microedition/khronos/egl/EGL10;Ljavax/microedition/khronos/egl/EGLDisplay;Ljavax/microedition/khronos/egl/EGLSurface;)V
  
+     .line 1151
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglSurface:Ljavax/microedition/khronos/egl/EGLSurface;
  
+     .line 1153
      :cond_0
      return-void
  .end method
***************
*** 397,406 ****
--- 437,448 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 1159
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglContext:Ljavax/microedition/khronos/egl/EGLContext;
  
      if-eqz v0, :cond_0
  
+     .line 1160
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
      #getter for: Landroid/opengl/GLSurfaceView;->mEGLContextFactory:Landroid/opengl/GLSurfaceView$EGLContextFactory;
***************
*** 416,436 ****
--- 458,483 ----
  
      invoke-interface {v0, v1, v2, v3}, Landroid/opengl/GLSurfaceView$EGLContextFactory;->destroyContext(Ljavax/microedition/khronos/egl/EGL10;Ljavax/microedition/khronos/egl/EGLDisplay;Ljavax/microedition/khronos/egl/EGLContext;)V
  
+     .line 1161
      iput-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglContext:Ljavax/microedition/khronos/egl/EGLContext;
  
+     .line 1163
      :cond_0
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
  
      if-eqz v0, :cond_1
  
+     .line 1164
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
  
      invoke-interface {v0, v1}, Ljavax/microedition/khronos/egl/EGL10;->eglTerminate(Ljavax/microedition/khronos/egl/EGLDisplay;)Z
  
+     .line 1165
      iput-object v4, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
  
+     .line 1167
      :cond_1
      return-void
  .end method
***************
*** 439,444 ****
--- 486,492 ----
      .locals 5
  
      .prologue
+     .line 1104
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
***************
*** 451,456 ****
--- 499,505 ----
  
      invoke-interface {v0, v1, v2, v3, v4}, Ljavax/microedition/khronos/egl/EGL10;->eglMakeCurrent(Ljavax/microedition/khronos/egl/EGLDisplay;Ljavax/microedition/khronos/egl/EGLSurface;Ljavax/microedition/khronos/egl/EGLSurface;Ljavax/microedition/khronos/egl/EGLContext;)Z
  
+     .line 1107
      iget-object v0, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
***************
*** 463,468 ****
--- 512,518 ----
  
      invoke-interface {v0, v1, v2, v3, v4}, Ljavax/microedition/khronos/egl/EGL10;->eglMakeCurrent(Ljavax/microedition/khronos/egl/EGLDisplay;Ljavax/microedition/khronos/egl/EGLSurface;Ljavax/microedition/khronos/egl/EGLSurface;Ljavax/microedition/khronos/egl/EGLContext;)Z
  
+     .line 1110
      return-void
  .end method
  
***************
*** 472,477 ****
--- 522,528 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 991
      invoke-static {}, Ljavax/microedition/khronos/egl/EGLContext;->getEGL()Ljavax/microedition/khronos/egl/EGL;
  
      move-result-object v1
***************
*** 480,485 ****
--- 531,537 ----
  
      iput-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
+     .line 996
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      sget-object v2, Ljavax/microedition/khronos/egl/EGL10;->EGL_DEFAULT_DISPLAY:Ljava/lang/Object;
***************
*** 490,501 ****
--- 542,555 ----
  
      iput-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
  
+     .line 998
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
  
      sget-object v2, Ljavax/microedition/khronos/egl/EGL10;->EGL_NO_DISPLAY:Ljavax/microedition/khronos/egl/EGLDisplay;
  
      if-ne v1, v2, :cond_0
  
+     .line 999
      new-instance v1, Ljava/lang/RuntimeException;
  
      const-string v2, "eglGetDisplay failed"
***************
*** 504,514 ****
--- 558,570 ----
  
      throw v1
  
+     .line 1005
      :cond_0
      const/4 v1, 0x2
  
      new-array v0, v1, [I
  
+     .line 1006
      .local v0, version:[I
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
***************
*** 520,525 ****
--- 576,582 ----
  
      if-nez v1, :cond_1
  
+     .line 1007
      new-instance v1, Ljava/lang/RuntimeException;
  
      const-string v2, "eglInitialize failed"
***************
*** 528,533 ****
--- 585,591 ----
  
      throw v1
  
+     .line 1009
      :cond_1
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
***************
*** 546,551 ****
--- 604,610 ----
  
      iput-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglConfig:Ljavax/microedition/khronos/egl/EGLConfig;
  
+     .line 1015
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->this$0:Landroid/opengl/GLSurfaceView;
  
      #getter for: Landroid/opengl/GLSurfaceView;->mEGLContextFactory:Landroid/opengl/GLSurfaceView$EGLContextFactory;
***************
*** 565,570 ****
--- 624,630 ----
  
      iput-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglContext:Ljavax/microedition/khronos/egl/EGLContext;
  
+     .line 1016
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglContext:Ljavax/microedition/khronos/egl/EGLContext;
  
      if-eqz v1, :cond_2
***************
*** 575,590 ****
--- 635,654 ----
  
      if-ne v1, v2, :cond_3
  
+     .line 1017
      :cond_2
      iput-object v5, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglContext:Ljavax/microedition/khronos/egl/EGLContext;
  
+     .line 1018
      const-string v1, "createContext"
  
      invoke-direct {p0, v1}, Landroid/opengl/GLSurfaceView$EglHelper;->throwEglException(Ljava/lang/String;)V
  
+     .line 1024
      :cond_3
      iput-object v5, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglSurface:Ljavax/microedition/khronos/egl/EGLSurface;
  
+     .line 1025
      return-void
  .end method
  
***************
*** 592,597 ****
--- 656,662 ----
      .locals 5
  
      .prologue
+     .line 1117
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      iget-object v2, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEglDisplay:Ljavax/microedition/khronos/egl/EGLDisplay;
***************
*** 604,623 ****
--- 669,692 ----
  
      if-nez v1, :cond_0
  
+     .line 1125
      iget-object v1, p0, Landroid/opengl/GLSurfaceView$EglHelper;->mEgl:Ljavax/microedition/khronos/egl/EGL10;
  
      invoke-interface {v1}, Ljavax/microedition/khronos/egl/EGL10;->eglGetError()I
  
      move-result v0
  
+     .line 1126
      .local v0, error:I
      packed-switch v0, :pswitch_data_0
  
+     .line 1136
      :pswitch_0
      const-string v1, "eglSwapBuffers"
  
      invoke-direct {p0, v1, v0}, Landroid/opengl/GLSurfaceView$EglHelper;->throwEglException(Ljava/lang/String;I)V
  
+     .line 1139
      .end local v0           #error:I
      :cond_0
      :goto_0
***************
*** 626,637 ****
--- 695,708 ----
      :goto_1
      return v1
  
+     .line 1128
      .restart local v0       #error:I
      :pswitch_1
      const/4 v1, 0x0
  
      goto :goto_1
  
+     .line 1133
      :pswitch_2
      const-string v1, "EglHelper"
  
***************
*** 665,670 ****
--- 736,742 ----
  
      goto :goto_0
  
+     .line 1126
      nop
  
      :pswitch_data_0
