*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLSurfaceView$SimpleEGLConfigChooser.smali	2012-03-23 19:36:15.304960070 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLSurfaceView$SimpleEGLConfigChooser.smali.original	2012-03-23 19:34:45.029313895 -0500
***************
*** 29,36 ****
--- 29,38 ----
  
      const/4 v5, 0x0
  
+     .line 966
      iput-object p1, p0, Landroid/opengl/GLSurfaceView$SimpleEGLConfigChooser;->this$0:Landroid/opengl/GLSurfaceView;
  
+     .line 967
      const/4 v3, 0x6
  
      if-eqz p2, :cond_0
***************
*** 48,57 ****
--- 50,61 ----
  
      invoke-direct/range {v0 .. v7}, Landroid/opengl/GLSurfaceView$ComponentSizeChooser;-><init>(Landroid/opengl/GLSurfaceView;IIIIII)V
  
+     .line 968
      return-void
  
      :cond_0
      move v6, v5
  
+     .line 967
      goto :goto_0
  .end method
