*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLU.smali	2012-03-23 19:36:13.255922706 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLU.smali.original	2012-03-23 19:34:45.019313713 -0500
***************
*** 12,17 ****
--- 12,18 ----
      .locals 1
  
      .prologue
+     .line 244
      const/16 v0, 0x20
  
      new-array v0, v0, [F
***************
*** 25,30 ****
--- 26,32 ----
      .locals 0
  
      .prologue
+     .line 26
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 35,82 ****
      .parameter "error"
  
      .prologue
      sparse-switch p0, :sswitch_data_0
  
      const/4 v0, 0x0
  
      :goto_0
      return-object v0
  
      :sswitch_0
!     const-string v0, "no error"
  
      goto :goto_0
  
      :sswitch_1
      const-string v0, "invalid enum"
  
      goto :goto_0
  
      :sswitch_2
      const-string v0, "invalid value"
  
      goto :goto_0
  
      :sswitch_3
      const-string v0, "invalid operation"
  
      goto :goto_0
  
      :sswitch_4
!     const-string v0, "stack overflow"
  
      goto :goto_0
  
      :sswitch_5
!     const-string v0, "stack underflow"
  
      goto :goto_0
  
      :sswitch_6
!     const-string v0, "out of memory"
  
      goto :goto_0
  
      :sswitch_data_0
      .sparse-switch
          0x0 -> :sswitch_0
--- 37,94 ----
      .parameter "error"
  
      .prologue
+     .line 36
      sparse-switch p0, :sswitch_data_0
  
+     .line 52
      const/4 v0, 0x0
  
      :goto_0
      return-object v0
  
+     .line 38
      :sswitch_0
!     const-string/jumbo v0, "no error"
  
      goto :goto_0
  
+     .line 40
      :sswitch_1
      const-string v0, "invalid enum"
  
      goto :goto_0
  
+     .line 42
      :sswitch_2
      const-string v0, "invalid value"
  
      goto :goto_0
  
+     .line 44
      :sswitch_3
      const-string v0, "invalid operation"
  
      goto :goto_0
  
+     .line 46
      :sswitch_4
!     const-string/jumbo v0, "stack overflow"
  
      goto :goto_0
  
+     .line 48
      :sswitch_5
!     const-string/jumbo v0, "stack underflow"
  
      goto :goto_0
  
+     .line 50
      :sswitch_6
!     const-string/jumbo v0, "out of memory"
  
      goto :goto_0
  
+     .line 36
      :sswitch_data_0
      .sparse-switch
          0x0 -> :sswitch_0
***************
*** 103,113 ****
--- 115,128 ----
      .parameter "upZ"
  
      .prologue
+     .line 75
      sget-object v0, Landroid/opengl/GLU;->sScratch:[F
  
+     .line 76
      .local v0, scratch:[F
      monitor-enter v0
  
+     .line 77
      const/4 v1, 0x0
  
      move v2, p1
***************
*** 131,144 ****
--- 146,163 ----
      :try_start_0
      invoke-static/range {v0 .. v10}, Landroid/opengl/Matrix;->setLookAtM([FIFFFFFFFFF)V
  
+     .line 79
      const/4 v1, 0x0
  
      invoke-interface {p0, v0, v1}, Ljavax/microedition/khronos/opengles/GL10;->glMultMatrixf([FI)V
  
+     .line 80
      monitor-exit v0
  
+     .line 81
      return-void
  
+     .line 80
      :catchall_0
      move-exception v1
  
***************
*** 158,163 ****
--- 177,183 ----
      .parameter "top"
  
      .prologue
+     .line 94
      const/high16 v5, -0x4080
  
      const/high16 v6, 0x3f80
***************
*** 174,179 ****
--- 194,200 ----
  
      invoke-interface/range {v0 .. v6}, Ljavax/microedition/khronos/opengles/GL10;->glOrthof(FFFFFF)V
  
+     .line 95
      return-void
  .end method
  
***************
*** 186,191 ****
--- 207,213 ----
      .parameter "zFar"
  
      .prologue
+     .line 113
      float-to-double v5, p1
  
      const-wide v7, 0x3f81df46a2529d39L
***************
*** 200,211 ****
--- 222,236 ----
  
      mul-float v4, p3, v0
  
+     .line 114
      .local v4, top:F
      neg-float v3, v4
  
+     .line 115
      .local v3, bottom:F
      mul-float v1, v3, p2
  
+     .line 116
      .local v1, left:F
      mul-float v2, v4, p2
  
***************
*** 216,223 ****
--- 241,250 ----
  
      move v6, p4
  
+     .line 117
      invoke-interface/range {v0 .. v6}, Ljavax/microedition/khronos/opengles/GL10;->glFrustumf(FFFFFF)V
  
+     .line 118
      return-void
  .end method
  
***************
*** 236,254 ****
--- 263,287 ----
      .parameter "winOffset"
  
      .prologue
+     .line 150
      sget-object v0, Landroid/opengl/GLU;->sScratch:[F
  
+     .line 151
      .local v0, scratch:[F
      monitor-enter v0
  
+     .line 152
      const/4 v6, 0x0
  
+     .line 153
      .local v6, M_OFFSET:I
      const/16 v8, 0x10
  
+     .line 154
      .local v8, V_OFFSET:I
      const/16 v7, 0x14
  
+     .line 155
      .local v7, V2_OFFSET:I
      const/4 v1, 0x0
  
***************
*** 263,286 ****
--- 296,324 ----
      :try_start_0
      invoke-static/range {v0 .. v5}, Landroid/opengl/Matrix;->multiplyMM([FI[FI[FI)V
  
+     .line 158
      const/16 v1, 0x10
  
      aput p0, v0, v1
  
+     .line 159
      const/16 v1, 0x11
  
      aput p1, v0, v1
  
+     .line 160
      const/16 v1, 0x12
  
      aput p2, v0, v1
  
+     .line 161
      const/16 v1, 0x13
  
      const/high16 v2, 0x3f80
  
      aput v2, v0, v1
  
+     .line 163
      const/16 v1, 0x14
  
      const/4 v3, 0x0
***************
*** 293,302 ****
--- 331,342 ----
  
      invoke-static/range {v0 .. v5}, Landroid/opengl/Matrix;->multiplyMV([FI[FI[FI)V
  
+     .line 166
      const/16 v1, 0x17
  
      aget v10, v0, v1
  
+     .line 167
      .local v10, w:F
      const/4 v1, 0x0
  
***************
*** 304,321 ****
--- 344,365 ----
  
      if-nez v1, :cond_0
  
+     .line 168
      const/4 v1, 0x0
  
      monitor-exit v0
  
+     .line 183
      :goto_0
      return v1
  
+     .line 171
      :cond_0
      const/high16 v1, 0x3f80
  
      div-float v9, v1, v10
  
+     .line 173
      .local v9, rw:F
      aget v1, p7, p8
  
***************
*** 347,352 ****
--- 391,397 ----
  
      aput v1, p9, p10
  
+     .line 177
      add-int/lit8 v1, p10, 0x1
  
      add-int/lit8 v2, p8, 0x1
***************
*** 381,386 ****
--- 426,432 ----
  
      aput v2, p9, v1
  
+     .line 180
      add-int/lit8 v1, p10, 0x2
  
      const/16 v2, 0x16
***************
*** 399,410 ****
--- 445,459 ----
  
      aput v2, p9, v1
  
+     .line 181
      monitor-exit v0
  
+     .line 183
      const/4 v1, 0x1
  
      goto :goto_0
  
+     .line 181
      .end local v9           #rw:F
      .end local v10           #w:F
      :catchall_0
***************
*** 432,450 ****
--- 481,505 ----
      .parameter "objOffset"
  
      .prologue
+     .line 216
      sget-object v0, Landroid/opengl/GLU;->sScratch:[F
  
+     .line 217
      .local v0, scratch:[F
      monitor-enter v0
  
+     .line 218
      const/4 v8, 0x0
  
+     .line 219
      .local v8, PM_OFFSET:I
      const/16 v7, 0x10
  
+     .line 220
      .local v7, INVPM_OFFSET:I
      const/4 v9, 0x0
  
+     .line 221
      .local v9, V_OFFSET:I
      const/4 v1, 0x0
  
***************
*** 459,464 ****
--- 514,520 ----
      :try_start_0
      invoke-static/range {v0 .. v5}, Landroid/opengl/Matrix;->multiplyMM([FI[FI[FI)V
  
+     .line 224
      const/16 v1, 0x10
  
      const/4 v2, 0x0
***************
*** 469,481 ****
--- 525,540 ----
  
      if-nez v1, :cond_0
  
+     .line 225
      const/4 v1, 0x0
  
      monitor-exit v0
  
+     .line 241
      :goto_0
      return v1
  
+     .line 228
      :cond_0
      const/4 v1, 0x0
  
***************
*** 505,510 ****
--- 564,570 ----
  
      aput v2, v0, v1
  
+     .line 231
      const/4 v1, 0x1
  
      const/high16 v2, 0x4000
***************
*** 533,538 ****
--- 593,599 ----
  
      aput v2, v0, v1
  
+     .line 234
      const/4 v1, 0x2
  
      const/high16 v2, 0x4000
***************
*** 545,556 ****
--- 606,619 ----
  
      aput v2, v0, v1
  
+     .line 235
      const/4 v1, 0x3
  
      const/high16 v2, 0x3f80
  
      aput v2, v0, v1
  
+     .line 237
      const/16 v4, 0x10
  
      const/4 v6, 0x0
***************
*** 565,576 ****
--- 628,642 ----
  
      invoke-static/range {v1 .. v6}, Landroid/opengl/Matrix;->multiplyMV([FI[FI[FI)V
  
+     .line 239
      monitor-exit v0
  
+     .line 241
      const/4 v1, 0x1
  
      goto :goto_0
  
+     .line 239
      :catchall_0
      move-exception v1
  
