*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLUtils.smali	2012-03-23 19:36:14.343942546 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLUtils.smali.original	2012-03-23 19:34:45.024313805 -0500
***************
*** 8,15 ****
--- 8,17 ----
      .locals 0
  
      .prologue
+     .line 37
      invoke-static {}, Landroid/opengl/GLUtils;->nativeClassInit()V
  
+     .line 38
      return-void
  .end method
  
***************
*** 17,24 ****
--- 19,28 ----
      .locals 0
  
      .prologue
+     .line 40
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 41
      return-void
  .end method
  
***************
*** 27,34 ****
--- 31,40 ----
      .parameter "error"
  
      .prologue
+     .line 236
      packed-switch p0, :pswitch_data_0
  
+     .line 268
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 54,134 ****
--- 60,156 ----
      :goto_0
      return-object v0
  
+     .line 238
      :pswitch_0
      const-string v0, "EGL_SUCCESS"
  
      goto :goto_0
  
+     .line 240
      :pswitch_1
      const-string v0, "EGL_NOT_INITIALIZED"
  
      goto :goto_0
  
+     .line 242
      :pswitch_2
      const-string v0, "EGL_BAD_ACCESS"
  
      goto :goto_0
  
+     .line 244
      :pswitch_3
      const-string v0, "EGL_BAD_ALLOC"
  
      goto :goto_0
  
+     .line 246
      :pswitch_4
      const-string v0, "EGL_BAD_ATTRIBUTE"
  
      goto :goto_0
  
+     .line 248
      :pswitch_5
      const-string v0, "EGL_BAD_CONFIG"
  
      goto :goto_0
  
+     .line 250
      :pswitch_6
      const-string v0, "EGL_BAD_CONTEXT"
  
      goto :goto_0
  
+     .line 252
      :pswitch_7
      const-string v0, "EGL_BAD_CURRENT_SURFACE"
  
      goto :goto_0
  
+     .line 254
      :pswitch_8
      const-string v0, "EGL_BAD_DISPLAY"
  
      goto :goto_0
  
+     .line 256
      :pswitch_9
      const-string v0, "EGL_BAD_MATCH"
  
      goto :goto_0
  
+     .line 258
      :pswitch_a
      const-string v0, "EGL_BAD_NATIVE_PIXMAP"
  
      goto :goto_0
  
+     .line 260
      :pswitch_b
      const-string v0, "EGL_BAD_NATIVE_WINDOW"
  
      goto :goto_0
  
+     .line 262
      :pswitch_c
      const-string v0, "EGL_BAD_PARAMETER"
  
      goto :goto_0
  
+     .line 264
      :pswitch_d
      const-string v0, "EGL_BAD_SURFACE"
  
      goto :goto_0
  
+     .line 266
      :pswitch_e
      const-string v0, "EGL_CONTEXT_LOST"
  
      goto :goto_0
  
+     .line 236
      :pswitch_data_0
      .packed-switch 0x3000
          :pswitch_0
***************
*** 154,161 ****
--- 176,185 ----
      .parameter "bitmap"
  
      .prologue
+     .line 49
      if-nez p0, :cond_0
  
+     .line 50
      new-instance v1, Ljava/lang/NullPointerException;
  
      const-string v2, "getInternalFormat can\'t be used with a null Bitmap"
***************
*** 164,169 ****
--- 188,194 ----
  
      throw v1
  
+     .line 52
      :cond_0
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->isRecycled()Z
  
***************
*** 171,176 ****
--- 196,202 ----
  
      if-eqz v1, :cond_1
  
+     .line 53
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      const-string v2, "bitmap is recycled"
***************
*** 179,192 ****
--- 205,221 ----
  
      throw v1
  
+     .line 55
      :cond_1
      invoke-static {p0}, Landroid/opengl/GLUtils;->native_getInternalFormat(Landroid/graphics/Bitmap;)I
  
      move-result v0
  
+     .line 56
      .local v0, result:I
      if-gez v0, :cond_2
  
+     .line 57
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      const-string v2, "Unknown internalformat"
***************
*** 195,200 ****
--- 224,230 ----
  
      throw v1
  
+     .line 59
      :cond_2
      return v0
  .end method
***************
*** 204,211 ****
--- 234,243 ----
      .parameter "bitmap"
  
      .prologue
+     .line 71
      if-nez p0, :cond_0
  
+     .line 72
      new-instance v1, Ljava/lang/NullPointerException;
  
      const-string v2, "getType can\'t be used with a null Bitmap"
***************
*** 214,219 ****
--- 246,252 ----
  
      throw v1
  
+     .line 74
      :cond_0
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->isRecycled()Z
  
***************
*** 221,226 ****
--- 254,260 ----
  
      if-eqz v1, :cond_1
  
+     .line 75
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      const-string v2, "bitmap is recycled"
***************
*** 229,242 ****
--- 263,279 ----
  
      throw v1
  
+     .line 77
      :cond_1
      invoke-static {p0}, Landroid/opengl/GLUtils;->native_getType(Landroid/graphics/Bitmap;)I
  
      move-result v0
  
+     .line 78
      .local v0, result:I
      if-gez v0, :cond_2
  
+     .line 79
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      const-string v2, "Unknown type"
***************
*** 245,250 ****
--- 282,288 ----
  
      throw v1
  
+     .line 81
      :cond_2
      return v0
  .end method
***************
*** 273,288 ****
      .parameter "border"
  
      .prologue
      if-nez p3, :cond_0
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "texImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      invoke-virtual {p3}, Landroid/graphics/Bitmap;->isRecycled()Z
  
--- 311,329 ----
      .parameter "border"
  
      .prologue
+     .line 108
      if-nez p3, :cond_0
  
+     .line 109
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "texImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 111
      :cond_0
      invoke-virtual {p3}, Landroid/graphics/Bitmap;->isRecycled()Z
  
***************
*** 290,295 ****
--- 331,337 ----
  
      if-eqz v0, :cond_1
  
+     .line 112
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "bitmap is recycled"
***************
*** 298,303 ****
--- 340,346 ----
  
      throw v0
  
+     .line 114
      :cond_1
      const/4 v4, -0x1
  
***************
*** 317,322 ****
--- 360,366 ----
  
      if-eqz v0, :cond_2
  
+     .line 115
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "invalid Bitmap format"
***************
*** 325,330 ****
--- 369,375 ----
  
      throw v0
  
+     .line 117
      :cond_2
      return-void
  .end method
***************
*** 339,354 ****
      .parameter "border"
  
      .prologue
      if-nez p3, :cond_0
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "texImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      invoke-virtual {p3}, Landroid/graphics/Bitmap;->isRecycled()Z
  
--- 384,402 ----
      .parameter "border"
  
      .prologue
+     .line 134
      if-nez p3, :cond_0
  
+     .line 135
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "texImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 137
      :cond_0
      invoke-virtual {p3}, Landroid/graphics/Bitmap;->isRecycled()Z
  
***************
*** 356,361 ****
--- 404,410 ----
  
      if-eqz v0, :cond_1
  
+     .line 138
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "bitmap is recycled"
***************
*** 364,369 ****
--- 413,419 ----
  
      throw v0
  
+     .line 140
      :cond_1
      invoke-static/range {p0 .. p5}, Landroid/opengl/GLUtils;->native_texImage2D(IIILandroid/graphics/Bitmap;II)I
  
***************
*** 371,376 ****
--- 421,427 ----
  
      if-eqz v0, :cond_2
  
+     .line 141
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "invalid Bitmap format"
***************
*** 379,384 ****
--- 430,436 ----
  
      throw v0
  
+     .line 143
      :cond_2
      return-void
  .end method
***************
*** 393,408 ****
      .prologue
      const/4 v2, -0x1
  
      if-nez p2, :cond_0
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "texImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      invoke-virtual {p2}, Landroid/graphics/Bitmap;->isRecycled()Z
  
--- 445,463 ----
      .prologue
      const/4 v2, -0x1
  
+     .line 156
      if-nez p2, :cond_0
  
+     .line 157
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "texImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 159
      :cond_0
      invoke-virtual {p2}, Landroid/graphics/Bitmap;->isRecycled()Z
  
***************
*** 410,415 ****
--- 465,471 ----
  
      if-eqz v0, :cond_1
  
+     .line 160
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "bitmap is recycled"
***************
*** 429,440 ****
--- 485,498 ----
  
      move v5, p3
  
+     .line 162
      invoke-static/range {v0 .. v5}, Landroid/opengl/GLUtils;->native_texImage2D(IIILandroid/graphics/Bitmap;II)I
  
      move-result v0
  
      if-eqz v0, :cond_2
  
+     .line 163
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "invalid Bitmap format"
***************
*** 443,448 ****
--- 501,507 ----
  
      throw v0
  
+     .line 165
      :cond_2
      return-void
  .end method
***************
*** 456,471 ****
      .parameter "bitmap"
  
      .prologue
      if-nez p4, :cond_0
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "texSubImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      invoke-virtual {p4}, Landroid/graphics/Bitmap;->isRecycled()Z
  
--- 515,533 ----
      .parameter "bitmap"
  
      .prologue
+     .line 191
      if-nez p4, :cond_0
  
+     .line 192
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "texSubImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 194
      :cond_0
      invoke-virtual {p4}, Landroid/graphics/Bitmap;->isRecycled()Z
  
***************
*** 473,478 ****
--- 535,541 ----
  
      if-eqz v0, :cond_1
  
+     .line 195
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "bitmap is recycled"
***************
*** 481,491 ****
--- 544,556 ----
  
      throw v0
  
+     .line 197
      :cond_1
      invoke-static {p4}, Landroid/opengl/GLUtils;->getType(Landroid/graphics/Bitmap;)I
  
      move-result v6
  
+     .line 198
      .local v6, type:I
      const/4 v5, -0x1
  
***************
*** 505,510 ****
--- 570,576 ----
  
      if-eqz v0, :cond_2
  
+     .line 199
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "invalid Bitmap format"
***************
*** 513,518 ****
--- 579,585 ----
  
      throw v0
  
+     .line 201
      :cond_2
      return-void
  .end method
***************
*** 528,543 ****
      .parameter "type"
  
      .prologue
      if-nez p4, :cond_0
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "texSubImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      invoke-virtual {p4}, Landroid/graphics/Bitmap;->isRecycled()Z
  
--- 595,613 ----
      .parameter "type"
  
      .prologue
+     .line 216
      if-nez p4, :cond_0
  
+     .line 217
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "texSubImage2D can\'t be used with a null Bitmap"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 219
      :cond_0
      invoke-virtual {p4}, Landroid/graphics/Bitmap;->isRecycled()Z
  
***************
*** 545,550 ****
--- 615,621 ----
  
      if-eqz v0, :cond_1
  
+     .line 220
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "bitmap is recycled"
***************
*** 553,558 ****
--- 624,630 ----
  
      throw v0
  
+     .line 222
      :cond_1
      invoke-static/range {p0 .. p6}, Landroid/opengl/GLUtils;->native_texSubImage2D(IIIILandroid/graphics/Bitmap;II)I
  
***************
*** 560,565 ****
--- 632,638 ----
  
      if-eqz v0, :cond_2
  
+     .line 223
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "invalid Bitmap format"
***************
*** 568,573 ****
--- 641,647 ----
  
      throw v0
  
+     .line 225
      :cond_2
      return-void
  .end method
