*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLWrapperBase.smali	2012-03-23 19:36:15.391961655 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/GLWrapperBase.smali.original	2012-03-23 19:34:45.030313913 -0500
***************
*** 29,52 ****
--- 29,57 ----
      .parameter "gl"
  
      .prologue
+     .line 32
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      move-object v0, p1
  
+     .line 33
      check-cast v0, Ljavax/microedition/khronos/opengles/GL10;
  
      iput-object v0, p0, Landroid/opengl/GLWrapperBase;->mgl:Ljavax/microedition/khronos/opengles/GL10;
  
+     .line 34
      instance-of v0, p1, Ljavax/microedition/khronos/opengles/GL10Ext;
  
      if-eqz v0, :cond_0
  
      move-object v0, p1
  
+     .line 35
      check-cast v0, Ljavax/microedition/khronos/opengles/GL10Ext;
  
      iput-object v0, p0, Landroid/opengl/GLWrapperBase;->mgl10Ext:Ljavax/microedition/khronos/opengles/GL10Ext;
  
+     .line 37
      :cond_0
      instance-of v0, p1, Ljavax/microedition/khronos/opengles/GL11;
  
***************
*** 54,63 ****
--- 59,70 ----
  
      move-object v0, p1
  
+     .line 38
      check-cast v0, Ljavax/microedition/khronos/opengles/GL11;
  
      iput-object v0, p0, Landroid/opengl/GLWrapperBase;->mgl11:Ljavax/microedition/khronos/opengles/GL11;
  
+     .line 40
      :cond_1
      instance-of v0, p1, Ljavax/microedition/khronos/opengles/GL11Ext;
  
***************
*** 65,84 ****
--- 72,95 ----
  
      move-object v0, p1
  
+     .line 41
      check-cast v0, Ljavax/microedition/khronos/opengles/GL11Ext;
  
      iput-object v0, p0, Landroid/opengl/GLWrapperBase;->mgl11Ext:Ljavax/microedition/khronos/opengles/GL11Ext;
  
+     .line 43
      :cond_2
      instance-of v0, p1, Ljavax/microedition/khronos/opengles/GL11ExtensionPack;
  
      if-eqz v0, :cond_3
  
+     .line 44
      check-cast p1, Ljavax/microedition/khronos/opengles/GL11ExtensionPack;
  
      .end local p1
      iput-object p1, p0, Landroid/opengl/GLWrapperBase;->mgl11ExtensionPack:Ljavax/microedition/khronos/opengles/GL11ExtensionPack;
  
+     .line 46
      :cond_3
      return-void
  .end method
