*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/MaterialIndices.smali	2012-03-23 19:36:15.232958756 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/MaterialIndices.smali.original	2012-03-23 19:34:45.028313877 -0500
***************
*** 18,33 ****
--- 18,39 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 34
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 31
      iput-object v0, p0, Landroid/opengl/MaterialIndices;->material:Landroid/opengl/Material;
  
+     .line 32
      iput-object v0, p0, Landroid/opengl/MaterialIndices;->indexBuffer:Ljava/nio/ShortBuffer;
  
+     .line 35
      iput-object p1, p0, Landroid/opengl/MaterialIndices;->material:Landroid/opengl/Material;
  
+     .line 36
      iput-object p2, p0, Landroid/opengl/MaterialIndices;->indexBuffer:Ljava/nio/ShortBuffer;
  
+     .line 37
      return-void
  .end method
  
***************
*** 37,42 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 44
      iget-object v0, p0, Landroid/opengl/MaterialIndices;->indexBuffer:Ljava/nio/ShortBuffer;
  
      return-object v0
***************
*** 46,51 ****
--- 53,59 ----
      .locals 1
  
      .prologue
+     .line 40
      iget-object v0, p0, Landroid/opengl/MaterialIndices;->material:Landroid/opengl/Material;
  
      return-object v0
