*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/Material.smali	2012-03-23 19:36:13.110920061 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/opengl/Material.smali.original	2012-03-23 19:34:45.018313695 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 1
  
      .prologue
+     .line 38
      const/4 v0, 0x4
  
      new-array v0, v0, [F
***************
*** 60,81 ****
--- 61,88 ----
      .prologue
      const/4 v1, 0x4
  
+     .line 40
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 31
      new-array v0, v1, [F
  
      iput-object v0, p0, Landroid/opengl/Material;->ka:[F
  
+     .line 32
      new-array v0, v1, [F
  
      iput-object v0, p0, Landroid/opengl/Material;->kd:[F
  
+     .line 33
      new-array v0, v1, [F
  
      iput-object v0, p0, Landroid/opengl/Material;->ks:[F
  
+     .line 41
      iput-object p1, p0, Landroid/opengl/Material;->parent:Landroid/opengl/Object3D;
  
+     .line 42
      return-void
  .end method
  
***************
*** 85,90 ****
--- 92,98 ----
      .locals 1
  
      .prologue
+     .line 49
      iget-object v0, p0, Landroid/opengl/Material;->map_kd:Ljava/lang/String;
  
      return-object v0
***************
*** 94,99 ****
--- 102,108 ----
      .locals 1
  
      .prologue
+     .line 45
      iget-object v0, p0, Landroid/opengl/Material;->name:Ljava/lang/String;
  
      return-object v0
***************
*** 117,138 ****
--- 126,152 ----
  
      const/4 v2, 0x0
  
+     .line 76
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readInt()I
  
+     .line 77
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readUTF()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/opengl/Material;->name:Ljava/lang/String;
  
+     .line 79
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readInt()I
  
+     .line 80
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readUTF()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/opengl/Material;->map_kd:Ljava/lang/String;
  
+     .line 82
      iget-object v0, p0, Landroid/opengl/Material;->parent:Landroid/opengl/Object3D;
  
      invoke-virtual {v0}, Landroid/opengl/Object3D;->hasTexcoords()Z
***************
*** 149,160 ****
--- 163,176 ----
  
      if-lez v0, :cond_0
  
+     .line 83
      iget-object v0, p0, Landroid/opengl/Material;->parent:Landroid/opengl/Object3D;
  
      iget-object v1, p0, Landroid/opengl/Material;->map_kd:Ljava/lang/String;
  
      invoke-virtual {v0, v1}, Landroid/opengl/Object3D;->loadTexture(Ljava/lang/String;)V
  
+     .line 86
      :cond_0
      iget-object v0, p0, Landroid/opengl/Material;->ka:[F
  
***************
*** 164,169 ****
--- 180,186 ----
  
      aput v1, v0, v2
  
+     .line 87
      iget-object v0, p0, Landroid/opengl/Material;->ka:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 172,177 ****
--- 189,195 ----
  
      aput v1, v0, v3
  
+     .line 88
      iget-object v0, p0, Landroid/opengl/Material;->ka:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 180,185 ****
--- 198,204 ----
  
      aput v1, v0, v4
  
+     .line 89
      iget-object v0, p0, Landroid/opengl/Material;->ka:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 188,193 ****
--- 207,213 ----
  
      aput v1, v0, v5
  
+     .line 91
      iget-object v0, p0, Landroid/opengl/Material;->kd:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 196,201 ****
--- 216,222 ----
  
      aput v1, v0, v2
  
+     .line 92
      iget-object v0, p0, Landroid/opengl/Material;->kd:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 204,209 ****
--- 225,231 ----
  
      aput v1, v0, v3
  
+     .line 93
      iget-object v0, p0, Landroid/opengl/Material;->kd:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 212,217 ****
--- 234,240 ----
  
      aput v1, v0, v4
  
+     .line 94
      iget-object v0, p0, Landroid/opengl/Material;->kd:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 220,225 ****
--- 243,249 ----
  
      aput v1, v0, v5
  
+     .line 96
      iget-object v0, p0, Landroid/opengl/Material;->ks:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 228,233 ****
--- 252,258 ----
  
      aput v1, v0, v2
  
+     .line 97
      iget-object v0, p0, Landroid/opengl/Material;->ks:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 236,241 ****
--- 261,267 ----
  
      aput v1, v0, v3
  
+     .line 98
      iget-object v0, p0, Landroid/opengl/Material;->ks:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 244,249 ****
--- 270,276 ----
  
      aput v1, v0, v4
  
+     .line 99
      iget-object v0, p0, Landroid/opengl/Material;->ks:[F
  
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
***************
*** 252,275 ****
--- 279,306 ----
  
      aput v1, v0, v5
  
+     .line 101
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
  
      move-result v0
  
      iput v0, p0, Landroid/opengl/Material;->ns:F
  
+     .line 102
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/opengl/Material;->illum:I
  
+     .line 103
      invoke-virtual {p1}, Ljava/io/DataInputStream;->readFloat()F
  
      move-result v0
  
      iput v0, p0, Landroid/opengl/Material;->d:F
  
+     .line 104
      return-void
  .end method
  
***************
*** 282,305 ****
--- 313,340 ----
  
      const/16 v3, 0x408
  
+     .line 53
      const/16 v0, 0x1200
  
      iget-object v1, p0, Landroid/opengl/Material;->kd:[F
  
      invoke-interface {p1, v3, v0, v1, v2}, Ljavax/microedition/khronos/opengles/GL10;->glMaterialfv(II[FI)V
  
+     .line 54
      const/16 v0, 0x1201
  
      iget-object v1, p0, Landroid/opengl/Material;->kd:[F
  
      invoke-interface {p1, v3, v0, v1, v2}, Ljavax/microedition/khronos/opengles/GL10;->glMaterialfv(II[FI)V
  
+     .line 55
      const/16 v0, 0x1202
  
      iget-object v1, p0, Landroid/opengl/Material;->ks:[F
  
      invoke-interface {p1, v3, v0, v1, v2}, Ljavax/microedition/khronos/opengles/GL10;->glMaterialfv(II[FI)V
  
+     .line 56
      const/16 v0, 0x1601
  
      iget v1, p0, Landroid/opengl/Material;->ns:F
***************
*** 318,323 ****
--- 353,359 ----
  
      invoke-interface {p1, v3, v0, v1}, Ljavax/microedition/khronos/opengles/GL10;->glMaterialf(IIF)V
  
+     .line 73
      return-void
  .end method
  
***************
*** 331,336 ****
--- 367,373 ----
  
      const/4 v2, 0x0
  
+     .line 107
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 395,401 ****
  
      move-result-object v0
  
!     const-string v1, "},"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 432,438 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "},"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 443,449 ****
  
      move-result-object v0
  
!     const-string v1, "},"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 480,486 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "},"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 491,503 ****
  
      move-result-object v0
  
!     const-string v1, "},"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      move-result-object v0
  
!     const-string v1, "ns="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 528,540 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "},"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      move-result-object v0
  
!     const-string/jumbo v1, "ns="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 515,521 ****
  
      move-result-object v0
  
!     const-string v1, "map_kd=\""
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 552,558 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "map_kd=\""
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
