*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/storage/MountServiceListener.smali	2012-03-23 19:36:49.648586326 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/storage/MountServiceListener.smali.original	2012-03-23 19:34:45.355319841 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 24
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 22,27 ****
--- 23,29 ----
      .parameter "newState"
  
      .prologue
+     .line 43
      return-void
  .end method
  
***************
*** 30,34 ****
--- 32,37 ----
      .parameter "connected"
  
      .prologue
+     .line 31
      return-void
  .end method
