*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/storage/OnObbStateChangeListener.smali	2012-03-23 19:36:49.438582495 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/storage/OnObbStateChangeListener.smali.original	2012-03-23 19:34:45.354319822 -0500
***************
*** 26,31 ****
--- 26,32 ----
      .locals 0
  
      .prologue
+     .line 23
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 39,43 ****
--- 40,45 ----
      .parameter "state"
  
      .prologue
+     .line 84
      return-void
  .end method
