*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/storage/StorageVolume.smali	2012-03-23 19:36:49.935591559 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/storage/StorageVolume.smali.original	2012-03-23 19:34:45.356319859 -0500
***************
*** 44,49 ****
--- 44,50 ----
      .locals 1
  
      .prologue
+     .line 181
      new-instance v0, Landroid/os/storage/StorageVolume$1;
  
      invoke-direct {v0}, Landroid/os/storage/StorageVolume$1;-><init>()V
***************
*** 65,88 ****
--- 66,99 ----
      .parameter "maxFileSize"
  
      .prologue
+     .line 59
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 60
      iput-object p1, p0, Landroid/os/storage/StorageVolume;->mPath:Ljava/lang/String;
  
+     .line 61
      iput-object p2, p0, Landroid/os/storage/StorageVolume;->mDescription:Ljava/lang/String;
  
+     .line 62
      iput-boolean p3, p0, Landroid/os/storage/StorageVolume;->mRemovable:Z
  
+     .line 63
      iput-boolean p4, p0, Landroid/os/storage/StorageVolume;->mEmulated:Z
  
+     .line 64
      iput p5, p0, Landroid/os/storage/StorageVolume;->mMtpReserveSpace:I
  
+     .line 65
      iput-boolean p7, p0, Landroid/os/storage/StorageVolume;->mAllowMassStorage:Z
  
+     .line 66
      iput p6, p0, Landroid/os/storage/StorageVolume;->mStorageId:I
  
+     .line 67
      iput-wide p8, p0, Landroid/os/storage/StorageVolume;->mMaxFileSize:J
  
+     .line 68
      return-void
  .end method
  
***************
*** 99,104 ****
--- 110,116 ----
      .parameter "x8"
  
      .prologue
+     .line 26
      invoke-direct/range {p0 .. p9}, Landroid/os/storage/StorageVolume;-><init>(Ljava/lang/String;Ljava/lang/String;ZZIIZJ)V
  
      return-void
***************
*** 115,136 ****
--- 127,157 ----
      .parameter "maxFileSize"
  
      .prologue
+     .line 46
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 47
      iput-object p1, p0, Landroid/os/storage/StorageVolume;->mPath:Ljava/lang/String;
  
+     .line 48
      iput-object p2, p0, Landroid/os/storage/StorageVolume;->mDescription:Ljava/lang/String;
  
+     .line 49
      iput-boolean p3, p0, Landroid/os/storage/StorageVolume;->mRemovable:Z
  
+     .line 50
      iput-boolean p4, p0, Landroid/os/storage/StorageVolume;->mEmulated:Z
  
+     .line 51
      iput p5, p0, Landroid/os/storage/StorageVolume;->mMtpReserveSpace:I
  
+     .line 52
      iput-boolean p6, p0, Landroid/os/storage/StorageVolume;->mAllowMassStorage:Z
  
+     .line 53
      iput-wide p7, p0, Landroid/os/storage/StorageVolume;->mMaxFileSize:J
  
+     .line 54
      return-void
  .end method
  
***************
*** 140,145 ****
--- 161,167 ----
      .locals 1
  
      .prologue
+     .line 147
      iget-boolean v0, p0, Landroid/os/storage/StorageVolume;->mAllowMassStorage:Z
  
      return v0
***************
*** 149,154 ****
--- 171,177 ----
      .locals 1
  
      .prologue
+     .line 203
      const/4 v0, 0x0
  
      return v0
***************
*** 159,164 ****
--- 182,188 ----
      .parameter "obj"
  
      .prologue
+     .line 161
      instance-of v1, p1, Landroid/os/storage/StorageVolume;
  
      if-eqz v1, :cond_0
***************
*** 169,176 ****
--- 193,202 ----
  
      move-object v0, p1
  
+     .line 162
      check-cast v0, Landroid/os/storage/StorageVolume;
  
+     .line 163
      .local v0, volume:Landroid/os/storage/StorageVolume;
      iget-object v1, p0, Landroid/os/storage/StorageVolume;->mPath:Ljava/lang/String;
  
***************
*** 180,185 ****
--- 206,212 ----
  
      move-result v1
  
+     .line 165
      .end local v0           #volume:Landroid/os/storage/StorageVolume;
      :goto_0
      return v1
***************
*** 194,199 ****
--- 221,227 ----
      .locals 1
  
      .prologue
+     .line 85
      iget-object v0, p0, Landroid/os/storage/StorageVolume;->mDescription:Ljava/lang/String;
  
      return-object v0
***************
*** 203,208 ****
--- 231,237 ----
      .locals 2
  
      .prologue
+     .line 156
      iget-wide v0, p0, Landroid/os/storage/StorageVolume;->mMaxFileSize:J
  
      return-wide v0
***************
*** 212,217 ****
--- 241,247 ----
      .locals 1
  
      .prologue
+     .line 138
      iget v0, p0, Landroid/os/storage/StorageVolume;->mMtpReserveSpace:I
  
      return v0
***************
*** 221,226 ****
--- 251,257 ----
      .locals 1
  
      .prologue
+     .line 76
      iget-object v0, p0, Landroid/os/storage/StorageVolume;->mPath:Ljava/lang/String;
  
      return-object v0
***************
*** 230,235 ****
--- 261,267 ----
      .locals 1
  
      .prologue
+     .line 113
      iget v0, p0, Landroid/os/storage/StorageVolume;->mStorageId:I
  
      return v0
***************
*** 239,244 ****
--- 271,277 ----
      .locals 1
  
      .prologue
+     .line 170
      iget-object v0, p0, Landroid/os/storage/StorageVolume;->mPath:Ljava/lang/String;
  
      invoke-virtual {v0}, Ljava/lang/String;->hashCode()I
***************
*** 252,257 ****
--- 285,291 ----
      .locals 1
  
      .prologue
+     .line 103
      iget-boolean v0, p0, Landroid/os/storage/StorageVolume;->mEmulated:Z
  
      return v0
***************
*** 261,266 ****
--- 295,301 ----
      .locals 1
  
      .prologue
+     .line 94
      iget-boolean v0, p0, Landroid/os/storage/StorageVolume;->mRemovable:Z
  
      return v0
***************
*** 271,276 ****
--- 306,312 ----
      .parameter "index"
  
      .prologue
+     .line 122
      add-int/lit8 v0, p1, 0x1
  
      shl-int/lit8 v0, v0, 0x10
***************
*** 279,284 ****
--- 315,321 ----
  
      iput v0, p0, Landroid/os/storage/StorageVolume;->mStorageId:I
  
+     .line 123
      return-void
  .end method
  
***************
*** 286,291 ****
--- 323,329 ----
      .locals 3
  
      .prologue
+     .line 175
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 409,422 ****
--- 447,463 ----
  
      const/4 v2, 0x0
  
+     .line 207
      iget-object v0, p0, Landroid/os/storage/StorageVolume;->mPath:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 208
      iget-object v0, p0, Landroid/os/storage/StorageVolume;->mDescription:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 209
      iget-boolean v0, p0, Landroid/os/storage/StorageVolume;->mRemovable:Z
  
      if-eqz v0, :cond_0
***************
*** 426,431 ****
--- 467,473 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 210
      iget-boolean v0, p0, Landroid/os/storage/StorageVolume;->mEmulated:Z
  
      if-eqz v0, :cond_1
***************
*** 435,448 ****
--- 477,493 ----
      :goto_1
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 211
      iget v0, p0, Landroid/os/storage/StorageVolume;->mStorageId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 212
      iget v0, p0, Landroid/os/storage/StorageVolume;->mMtpReserveSpace:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 213
      iget-boolean v0, p0, Landroid/os/storage/StorageVolume;->mAllowMassStorage:Z
  
      if-eqz v0, :cond_2
***************
*** 450,473 ****
--- 495,523 ----
      :goto_2
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 214
      iget-wide v0, p0, Landroid/os/storage/StorageVolume;->mMaxFileSize:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 215
      return-void
  
      :cond_0
      move v0, v2
  
+     .line 209
      goto :goto_0
  
      :cond_1
      move v0, v2
  
+     .line 210
      goto :goto_1
  
      :cond_2
      move v1, v2
  
+     .line 213
      goto :goto_2
  .end method
