*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/storage/StorageEventListener.smali	2012-03-23 19:36:50.301598232 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/os/storage/StorageEventListener.smali.original	2012-03-23 19:34:45.358319895 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 24
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 22,27 ****
--- 23,29 ----
      .parameter "newState"
  
      .prologue
+     .line 39
      return-void
  .end method
  
***************
*** 30,34 ****
--- 32,37 ----
      .parameter "connected"
  
      .prologue
+     .line 30
      return-void
  .end method
