*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/http/multipart/FilePart.smali	2012-03-23 19:37:52.908739873 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/http/multipart/FilePart.smali.original	2012-03-23 19:34:46.390338714 -0500
***************
*** 26,31 ****
--- 26,32 ----
      .locals 1
  
      .prologue
+     .line 69
      const-class v0, Lcom/android/internal/http/multipart/FilePart;
  
      invoke-static {v0}, Lorg/apache/commons/logging/LogFactory;->getLog(Ljava/lang/Class;)Lorg/apache/commons/logging/Log;
***************
*** 34,39 ****
--- 35,41 ----
  
      sput-object v0, Lcom/android/internal/http/multipart/FilePart;->LOG:Lorg/apache/commons/logging/Log;
  
+     .line 75
      const-string v0, "; filename="
  
      invoke-static {v0}, Lorg/apache/http/util/EncodingUtils;->getAsciiBytes(Ljava/lang/String;)[B
***************
*** 53,60 ****
--- 55,64 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 113
      invoke-direct {p0, p1, p2, v0, v0}, Lcom/android/internal/http/multipart/FilePart;-><init>(Ljava/lang/String;Lcom/android/internal/http/multipart/PartSource;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 114
      return-void
  .end method
  
***************
*** 66,71 ****
--- 70,76 ----
      .parameter "charset"
  
      .prologue
+     .line 93
      if-nez p3, :cond_0
  
      const-string p3, "application/octet-stream"
***************
*** 82,89 ****
--- 87,96 ----
  
      invoke-direct {p0, p1, p3, p4, v0}, Lcom/android/internal/http/multipart/PartBase;-><init>(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 100
      if-nez p2, :cond_2
  
+     .line 101
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "Source may not be null"
***************
*** 92,100 ****
--- 99,109 ----
  
      throw v0
  
+     .line 103
      :cond_2
      iput-object p2, p0, Lcom/android/internal/http/multipart/FilePart;->source:Lcom/android/internal/http/multipart/PartSource;
  
+     .line 104
      return-void
  .end method
  
***************
*** 111,122 ****
--- 120,133 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 127
      new-instance v0, Lcom/android/internal/http/multipart/FilePartSource;
  
      invoke-direct {v0, p2}, Lcom/android/internal/http/multipart/FilePartSource;-><init>(Ljava/io/File;)V
  
      invoke-direct {p0, p1, v0, v1, v1}, Lcom/android/internal/http/multipart/FilePart;-><init>(Ljava/lang/String;Lcom/android/internal/http/multipart/PartSource;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 128
      return-void
  .end method
  
***************
*** 133,144 ****
--- 144,157 ----
      .end annotation
  
      .prologue
+     .line 145
      new-instance v0, Lcom/android/internal/http/multipart/FilePartSource;
  
      invoke-direct {v0, p2}, Lcom/android/internal/http/multipart/FilePartSource;-><init>(Ljava/io/File;)V
  
      invoke-direct {p0, p1, v0, p3, p4}, Lcom/android/internal/http/multipart/FilePart;-><init>(Ljava/lang/String;Lcom/android/internal/http/multipart/PartSource;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 146
      return-void
  .end method
  
***************
*** 156,167 ****
--- 169,182 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 160
      new-instance v0, Lcom/android/internal/http/multipart/FilePartSource;
  
      invoke-direct {v0, p2, p3}, Lcom/android/internal/http/multipart/FilePartSource;-><init>(Ljava/lang/String;Ljava/io/File;)V
  
      invoke-direct {p0, p1, v0, v1, v1}, Lcom/android/internal/http/multipart/FilePart;-><init>(Ljava/lang/String;Lcom/android/internal/http/multipart/PartSource;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 161
      return-void
  .end method
  
***************
*** 179,190 ****
--- 194,207 ----
      .end annotation
  
      .prologue
+     .line 179
      new-instance v0, Lcom/android/internal/http/multipart/FilePartSource;
  
      invoke-direct {v0, p2, p3}, Lcom/android/internal/http/multipart/FilePartSource;-><init>(Ljava/lang/String;Ljava/io/File;)V
  
      invoke-direct {p0, p1, v0, p4, p5}, Lcom/android/internal/http/multipart/FilePart;-><init>(Ljava/lang/String;Lcom/android/internal/http/multipart/PartSource;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 180
      return-void
  .end method
  
***************
*** 194,205 ****
--- 211,224 ----
      .locals 2
  
      .prologue
+     .line 239
      sget-object v0, Lcom/android/internal/http/multipart/FilePart;->LOG:Lorg/apache/commons/logging/Log;
  
      const-string v1, "enter getSource()"
  
      invoke-interface {v0, v1}, Lorg/apache/commons/logging/Log;->trace(Ljava/lang/Object;)V
  
+     .line 240
      iget-object v0, p0, Lcom/android/internal/http/multipart/FilePart;->source:Lcom/android/internal/http/multipart/PartSource;
  
      return-object v0
***************
*** 209,220 ****
--- 228,241 ----
      .locals 2
  
      .prologue
+     .line 250
      sget-object v0, Lcom/android/internal/http/multipart/FilePart;->LOG:Lorg/apache/commons/logging/Log;
  
      const-string v1, "enter lengthOfData()"
  
      invoke-interface {v0, v1}, Lorg/apache/commons/logging/Log;->trace(Ljava/lang/Object;)V
  
+     .line 251
      iget-object v0, p0, Lcom/android/internal/http/multipart/FilePart;->source:Lcom/android/internal/http/multipart/PartSource;
  
      invoke-interface {v0}, Lcom/android/internal/http/multipart/PartSource;->getLength()J
***************
*** 234,245 ****
--- 255,268 ----
      .end annotation
  
      .prologue
+     .line 210
      sget-object v3, Lcom/android/internal/http/multipart/FilePart;->LOG:Lorg/apache/commons/logging/Log;
  
      const-string v4, "enter sendData(OutputStream out)"
  
      invoke-interface {v3, v4}, Lorg/apache/commons/logging/Log;->trace(Ljava/lang/Object;)V
  
+     .line 211
      invoke-virtual {p0}, Lcom/android/internal/http/multipart/FilePart;->lengthOfData()J
  
      move-result-wide v3
***************
*** 250,269 ****
--- 273,296 ----
  
      if-nez v3, :cond_0
  
+     .line 216
      sget-object v3, Lcom/android/internal/http/multipart/FilePart;->LOG:Lorg/apache/commons/logging/Log;
  
      const-string v4, "No data to send."
  
      invoke-interface {v3, v4}, Lorg/apache/commons/logging/Log;->debug(Ljava/lang/Object;)V
  
+     .line 231
      :goto_0
      return-void
  
+     .line 220
      :cond_0
      const/16 v3, 0x1000
  
      new-array v2, v3, [B
  
+     .line 221
      .local v2, tmp:[B
      iget-object v3, p0, Lcom/android/internal/http/multipart/FilePart;->source:Lcom/android/internal/http/multipart/PartSource;
  
***************
*** 271,276 ****
--- 298,304 ----
  
      move-result-object v0
  
+     .line 224
      .local v0, instream:Ljava/io/InputStream;
      :goto_1
      :try_start_0
***************
*** 281,286 ****
--- 309,315 ----
      .local v1, len:I
      if-ltz v1, :cond_1
  
+     .line 225
      const/4 v3, 0x0
  
      invoke-virtual {p1, v2, v3, v1}, Ljava/io/OutputStream;->write([BII)V
***************
*** 289,294 ****
--- 318,324 ----
  
      goto :goto_1
  
+     .line 229
      .end local v1           #len:I
      :catchall_0
      move-exception v3
***************
*** 314,354 ****
--- 344,393 ----
      .end annotation
  
      .prologue
+     .line 191
      sget-object v1, Lcom/android/internal/http/multipart/FilePart;->LOG:Lorg/apache/commons/logging/Log;
  
      const-string v2, "enter sendDispositionHeader(OutputStream out)"
  
      invoke-interface {v1, v2}, Lorg/apache/commons/logging/Log;->trace(Ljava/lang/Object;)V
  
+     .line 192
      invoke-super {p0, p1}, Lcom/android/internal/http/multipart/PartBase;->sendDispositionHeader(Ljava/io/OutputStream;)V
  
+     .line 193
      iget-object v1, p0, Lcom/android/internal/http/multipart/FilePart;->source:Lcom/android/internal/http/multipart/PartSource;
  
      invoke-interface {v1}, Lcom/android/internal/http/multipart/PartSource;->getFileName()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 194
      .local v0, filename:Ljava/lang/String;
      if-eqz v0, :cond_0
  
+     .line 195
      sget-object v1, Lcom/android/internal/http/multipart/FilePart;->FILE_NAME_BYTES:[B
  
      invoke-virtual {p1, v1}, Ljava/io/OutputStream;->write([B)V
  
+     .line 196
      sget-object v1, Lcom/android/internal/http/multipart/FilePart;->QUOTE_BYTES:[B
  
      invoke-virtual {p1, v1}, Ljava/io/OutputStream;->write([B)V
  
+     .line 197
      invoke-static {v0}, Lorg/apache/http/util/EncodingUtils;->getAsciiBytes(Ljava/lang/String;)[B
  
      move-result-object v1
  
      invoke-virtual {p1, v1}, Ljava/io/OutputStream;->write([B)V
  
+     .line 198
      sget-object v1, Lcom/android/internal/http/multipart/FilePart;->QUOTE_BYTES:[B
  
      invoke-virtual {p1, v1}, Ljava/io/OutputStream;->write([B)V
  
+     .line 200
      :cond_0
      return-void
  .end method
