*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/telephony/PhoneStateListener$1.smali	2012-03-23 19:36:18.883025316 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/telephony/PhoneStateListener$1.smali.original	2012-03-23 19:34:45.048314242 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 282
      iput-object p1, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      invoke-direct {p0}, Lcom/android/internal/telephony/IPhoneStateListener$Stub;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 297
      iget-object v0, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      iget-object v2, v0, Landroid/telephony/PhoneStateListener;->mHandler:Landroid/os/Handler;
***************
*** 59,69 ****
--- 61,73 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 299
      return-void
  
      :cond_0
      move v0, v1
  
+     .line 297
      goto :goto_0
  .end method
  
***************
*** 73,78 ****
--- 77,83 ----
      .parameter "incomingNumber"
  
      .prologue
+     .line 307
      iget-object v0, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      iget-object v0, v0, Landroid/telephony/PhoneStateListener;->mHandler:Landroid/os/Handler;
***************
*** 87,92 ****
--- 92,98 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 308
      return-void
  .end method
  
***************
*** 97,106 ****
--- 103,114 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 302
      invoke-static {p1}, Landroid/telephony/CellLocation;->newFromBundle(Landroid/os/Bundle;)Landroid/telephony/CellLocation;
  
      move-result-object v0
  
+     .line 303
      .local v0, location:Landroid/telephony/CellLocation;
      iget-object v1, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
***************
*** 114,119 ****
--- 122,128 ----
  
      invoke-virtual {v1}, Landroid/os/Message;->sendToTarget()V
  
+     .line 304
      return-void
  .end method
  
***************
*** 122,127 ****
--- 131,137 ----
      .parameter "direction"
  
      .prologue
+     .line 316
      iget-object v0, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      iget-object v0, v0, Landroid/telephony/PhoneStateListener;->mHandler:Landroid/os/Handler;
***************
*** 138,143 ****
--- 148,154 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 317
      return-void
  .end method
  
***************
*** 147,152 ****
--- 158,164 ----
      .parameter "networkType"
  
      .prologue
+     .line 311
      iget-object v0, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      iget-object v0, v0, Landroid/telephony/PhoneStateListener;->mHandler:Landroid/os/Handler;
***************
*** 159,164 ****
--- 171,177 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 313
      return-void
  .end method
  
***************
*** 169,174 ****
--- 182,188 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 292
      iget-object v0, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      iget-object v2, v0, Landroid/telephony/PhoneStateListener;->mHandler:Landroid/os/Handler;
***************
*** 188,198 ****
--- 202,214 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 294
      return-void
  
      :cond_0
      move v0, v1
  
+     .line 292
      goto :goto_0
  .end method
  
***************
*** 201,206 ****
--- 217,223 ----
      .parameter "otaspMode"
  
      .prologue
+     .line 324
      iget-object v0, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      iget-object v0, v0, Landroid/telephony/PhoneStateListener;->mHandler:Landroid/os/Handler;
***************
*** 215,220 ****
--- 232,238 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 325
      return-void
  .end method
  
***************
*** 225,230 ****
--- 243,249 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 284
      iget-object v0, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      iget-object v0, v0, Landroid/telephony/PhoneStateListener;->mHandler:Landroid/os/Handler;
***************
*** 237,242 ****
--- 256,262 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 285
      return-void
  .end method
  
***************
*** 245,250 ****
--- 265,271 ----
      .parameter "asu"
  
      .prologue
+     .line 288
      iget-object v0, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      iget-object v0, v0, Landroid/telephony/PhoneStateListener;->mHandler:Landroid/os/Handler;
***************
*** 261,266 ****
--- 282,288 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 289
      return-void
  .end method
  
***************
*** 271,276 ****
--- 293,299 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 320
      iget-object v0, p0, Landroid/telephony/PhoneStateListener$1;->this$0:Landroid/telephony/PhoneStateListener;
  
      iget-object v0, v0, Landroid/telephony/PhoneStateListener;->mHandler:Landroid/os/Handler;
***************
*** 283,287 ****
--- 306,311 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 321
      return-void
  .end method
