*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/telephony/PhoneNumberFormattingTextWatcher.smali	2012-03-23 19:36:19.064028616 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/telephony/PhoneNumberFormattingTextWatcher.smali.original	2012-03-23 19:34:45.050314279 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .locals 1
  
      .prologue
+     .line 84
      invoke-static {}, Ljava/util/Locale;->getDefault()Ljava/util/Locale;
  
      move-result-object v0
***************
*** 41,46 ****
--- 42,48 ----
  
      invoke-direct {p0, v0}, Landroid/telephony/PhoneNumberFormattingTextWatcher;-><init>(Ljava/lang/String;)V
  
+     .line 85
      return-void
  .end method
  
***************
*** 49,60 ****
--- 51,65 ----
      .parameter "countryCode"
  
      .prologue
+     .line 95
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 70
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mSelfChange:Z
  
+     .line 96
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
***************
*** 63,68 ****
--- 68,74 ----
  
      throw v0
  
+     .line 97
      :cond_0
      invoke-static {}, Lcom/android/i18n/phonenumbers/PhoneNumberUtil;->getInstance()Lcom/android/i18n/phonenumbers/PhoneNumberUtil;
  
***************
*** 74,79 ****
--- 80,86 ----
  
      iput-object v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mFormatter:Lcom/android/i18n/phonenumbers/AsYouTypeFormatter;
  
+     .line 98
      return-void
  .end method
  
***************
*** 83,88 ****
--- 90,96 ----
      .parameter "hasCursor"
  
      .prologue
+     .line 197
      if-eqz p2, :cond_0
  
      iget-object v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mFormatter:Lcom/android/i18n/phonenumbers/AsYouTypeFormatter;
***************
*** 111,116 ****
--- 119,125 ----
      .parameter "count"
  
      .prologue
+     .line 207
      move v1, p2
  
      .local v1, i:I
***************
*** 119,128 ****
--- 128,139 ----
  
      if-ge v1, v2, :cond_1
  
+     .line 208
      invoke-interface {p1, v1}, Ljava/lang/CharSequence;->charAt(I)C
  
      move-result v0
  
+     .line 209
      .local v0, c:C
      invoke-static {v0}, Landroid/telephony/PhoneNumberUtils;->isNonSeparator(C)Z
  
***************
*** 130,147 ****
--- 141,162 ----
  
      if-nez v2, :cond_0
  
+     .line 210
      const/4 v2, 0x1
  
+     .line 213
      .end local v0           #c:C
      :goto_1
      return v2
  
+     .line 207
      .restart local v0       #c:C
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 213
      .end local v0           #c:C
      :cond_1
      const/4 v2, 0x0
***************
*** 155,180 ****
--- 170,202 ----
      .parameter "cursor"
  
      .prologue
+     .line 171
      add-int/lit8 v1, p2, -0x1
  
+     .line 172
      .local v1, curIndex:I
      const/4 v2, 0x0
  
+     .line 173
      .local v2, formatted:Ljava/lang/String;
      iget-object v7, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mFormatter:Lcom/android/i18n/phonenumbers/AsYouTypeFormatter;
  
      invoke-virtual {v7}, Lcom/android/i18n/phonenumbers/AsYouTypeFormatter;->clear()V
  
+     .line 174
      const/4 v5, 0x0
  
+     .line 175
      .local v5, lastNonSeparator:C
      const/4 v3, 0x0
  
+     .line 176
      .local v3, hasCursor:Z
      invoke-interface {p1}, Ljava/lang/CharSequence;->length()I
  
      move-result v6
  
+     .line 177
      .local v6, len:I
      const/4 v4, 0x0
  
***************
*** 182,191 ****
--- 204,215 ----
      :goto_0
      if-ge v4, v6, :cond_3
  
+     .line 178
      invoke-interface {p1, v4}, Ljava/lang/CharSequence;->charAt(I)C
  
      move-result v0
  
+     .line 179
      .local v0, c:C
      invoke-static {v0}, Landroid/telephony/PhoneNumberUtils;->isNonSeparator(C)Z
  
***************
*** 193,227 ****
--- 217,261 ----
  
      if-eqz v7, :cond_1
  
+     .line 180
      if-eqz v5, :cond_0
  
+     .line 181
      invoke-direct {p0, v5, v3}, Landroid/telephony/PhoneNumberFormattingTextWatcher;->getFormattedNumber(CZ)Ljava/lang/String;
  
      move-result-object v2
  
+     .line 182
      const/4 v3, 0x0
  
+     .line 184
      :cond_0
      move v5, v0
  
+     .line 186
      :cond_1
      if-ne v4, v1, :cond_2
  
+     .line 187
      const/4 v3, 0x1
  
+     .line 177
      :cond_2
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
  
+     .line 190
      .end local v0           #c:C
      :cond_3
      if-eqz v5, :cond_4
  
+     .line 191
      invoke-direct {p0, v5, v3}, Landroid/telephony/PhoneNumberFormattingTextWatcher;->getFormattedNumber(CZ)Ljava/lang/String;
  
      move-result-object v2
  
+     .line 193
      :cond_4
      return-object v2
  .end method
***************
*** 230,243 ****
--- 264,280 ----
      .locals 1
  
      .prologue
+     .line 202
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mStopFormatting:Z
  
+     .line 203
      iget-object v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mFormatter:Lcom/android/i18n/phonenumbers/AsYouTypeFormatter;
  
      invoke-virtual {v0}, Lcom/android/i18n/phonenumbers/AsYouTypeFormatter;->clear()V
  
+     .line 204
      return-void
  .end method
  
***************
*** 252,257 ****
--- 289,295 ----
  
      const/4 v1, 0x0
  
+     .line 141
      monitor-enter p0
  
      :try_start_0
***************
*** 259,264 ****
--- 297,303 ----
  
      if-eqz v2, :cond_2
  
+     .line 143
      invoke-interface {p1}, Landroid/text/Editable;->length()I
  
      move-result v2
***************
*** 270,275 ****
--- 309,315 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 162
      :cond_0
      :goto_1
      monitor-exit p0
***************
*** 279,292 ****
--- 319,335 ----
      :cond_1
      move v0, v1
  
+     .line 143
      goto :goto_0
  
+     .line 146
      :cond_2
      :try_start_1
      iget-boolean v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mSelfChange:Z
  
      if-nez v0, :cond_0
  
+     .line 150
      invoke-static {p1}, Landroid/text/Selection;->getSelectionEnd(Ljava/lang/CharSequence;)I
  
      move-result v0
***************
*** 295,314 ****
--- 338,361 ----
  
      move-result-object v3
  
+     .line 151
      .local v3, formatted:Ljava/lang/String;
      if-eqz v3, :cond_0
  
+     .line 152
      iget-object v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mFormatter:Lcom/android/i18n/phonenumbers/AsYouTypeFormatter;
  
      invoke-virtual {v0}, Lcom/android/i18n/phonenumbers/AsYouTypeFormatter;->getRememberedPosition()I
  
      move-result v6
  
+     .line 153
      .local v6, rememberedPos:I
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mSelfChange:Z
  
+     .line 154
      const/4 v1, 0x0
  
      invoke-interface {p1}, Landroid/text/Editable;->length()I
***************
*** 325,330 ****
--- 372,378 ----
  
      invoke-interface/range {v0 .. v5}, Landroid/text/Editable;->replace(IILjava/lang/CharSequence;II)Landroid/text/Editable;
  
+     .line 157
      invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
      move-result-object v0
***************
*** 335,342 ****
--- 383,392 ----
  
      if-eqz v0, :cond_3
  
+     .line 158
      invoke-static {p1, v6}, Landroid/text/Selection;->setSelection(Landroid/text/Spannable;I)V
  
+     .line 160
      :cond_3
      const/4 v0, 0x0
  
***************
*** 346,351 ****
--- 396,402 ----
  
      goto :goto_1
  
+     .line 141
      .end local v3           #formatted:Ljava/lang/String;
      .end local v6           #rememberedPos:I
      :catchall_0
***************
*** 364,369 ****
--- 415,421 ----
      .parameter "after"
  
      .prologue
+     .line 102
      iget-boolean v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mSelfChange:Z
  
      if-nez v0, :cond_0
***************
*** 372,381 ****
--- 424,435 ----
  
      if-eqz v0, :cond_1
  
+     .line 117
      :cond_0
      :goto_0
      return-void
  
+     .line 105
      :cond_1
      if-nez p3, :cond_2
  
***************
*** 385,396 ****
--- 439,452 ----
  
      if-ne v0, p2, :cond_2
  
+     .line 107
      const/4 v0, 0x1
  
      iput v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mState:I
  
      goto :goto_0
  
+     .line 108
      :cond_2
      if-nez p4, :cond_3
  
***************
*** 404,415 ****
--- 460,473 ----
  
      if-lez p3, :cond_3
  
+     .line 110
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mState:I
  
      goto :goto_0
  
+     .line 111
      :cond_3
      if-lez p3, :cond_4
  
***************
*** 419,430 ****
--- 477,490 ----
  
      if-nez v0, :cond_4
  
+     .line 113
      const/4 v0, 0x2
  
      iput v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mState:I
  
      goto :goto_0
  
+     .line 115
      :cond_4
      const/4 v0, 0x3
  
***************
*** 443,448 ****
--- 503,509 ----
      .prologue
      const/4 v2, 0x3
  
+     .line 120
      iget-boolean v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mSelfChange:Z
  
      if-nez v0, :cond_0
***************
*** 451,465 ****
--- 512,529 ----
  
      if-eqz v0, :cond_1
  
+     .line 138
      :cond_0
      :goto_0
      return-void
  
+     .line 123
      :cond_1
      iget v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mState:I
  
      if-ne v0, v2, :cond_2
  
+     .line 124
      if-lez p4, :cond_2
  
      invoke-direct {p0, p1, p2, p4}, Landroid/telephony/PhoneNumberFormattingTextWatcher;->hasSeparator(Ljava/lang/CharSequence;II)Z
***************
*** 468,477 ****
--- 532,543 ----
  
      if-nez v0, :cond_2
  
+     .line 126
      const/4 v0, 0x2
  
      iput v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mState:I
  
+     .line 130
      :cond_2
      iget v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mState:I
  
***************
*** 487,501 ****
--- 553,570 ----
  
      if-eqz v0, :cond_3
  
+     .line 132
      invoke-direct {p0}, Landroid/telephony/PhoneNumberFormattingTextWatcher;->stopFormatting()V
  
      goto :goto_0
  
+     .line 133
      :cond_3
      iget v0, p0, Landroid/telephony/PhoneNumberFormattingTextWatcher;->mState:I
  
      if-ne v0, v2, :cond_0
  
+     .line 136
      invoke-direct {p0}, Landroid/telephony/PhoneNumberFormattingTextWatcher;->stopFormatting()V
  
      goto :goto_0
