*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/telephony/ServiceState.smali	2012-03-23 19:36:19.256032117 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/telephony/ServiceState.smali.original	2012-03-23 19:34:45.051314297 -0500
***************
*** 110,115 ****
--- 110,116 ----
      .locals 1
  
      .prologue
+     .line 229
      new-instance v0, Landroid/telephony/ServiceState$1;
  
      invoke-direct {v0}, Landroid/telephony/ServiceState$1;-><init>()V
***************
*** 123,134 ****
--- 124,138 ----
      .locals 1
  
      .prologue
+     .line 156
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 117
      const/4 v0, 0x1
  
      iput v0, p0, Landroid/telephony/ServiceState;->mState:I
  
+     .line 157
      return-void
  .end method
  
***************
*** 141,156 ****
--- 145,164 ----
  
      const/4 v1, 0x1
  
+     .line 189
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 117
      iput v1, p0, Landroid/telephony/ServiceState;->mState:I
  
+     .line 190
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/telephony/ServiceState;->mState:I
  
+     .line 191
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
***************
*** 162,185 ****
--- 170,197 ----
      :goto_0
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
+     .line 192
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
+     .line 193
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
+     .line 194
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
+     .line 195
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
***************
*** 191,202 ****
--- 203,216 ----
      :goto_1
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
+     .line 196
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
+     .line 197
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
***************
*** 208,249 ****
--- 222,270 ----
      :goto_2
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
+     .line 198
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
+     .line 199
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/telephony/ServiceState;->mSystemId:I
  
+     .line 200
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
+     .line 201
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
+     .line 202
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconIndex:I
  
+     .line 203
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconMode:I
  
+     .line 204
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
***************
*** 253,278 ****
--- 274,304 ----
      :goto_3
      iput-boolean v1, p0, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
  
+     .line 205
      return-void
  
      :cond_0
      move v0, v2
  
+     .line 191
      goto :goto_0
  
      :cond_1
      move v0, v2
  
+     .line 195
      goto :goto_1
  
      :cond_2
      move v0, v2
  
+     .line 197
      goto :goto_2
  
      :cond_3
      move v1, v2
  
+     .line 204
      goto :goto_3
  .end method
  
***************
*** 281,294 ****
--- 307,324 ----
      .parameter "s"
  
      .prologue
+     .line 164
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 117
      const/4 v0, 0x1
  
      iput v0, p0, Landroid/telephony/ServiceState;->mState:I
  
+     .line 165
      invoke-virtual {p0, p1}, Landroid/telephony/ServiceState;->copyFrom(Landroid/telephony/ServiceState;)V
  
+     .line 166
      return-void
  .end method
  
***************
*** 298,303 ****
--- 328,334 ----
      .parameter "b"
  
      .prologue
+     .line 574
      if-nez p0, :cond_1
  
      if-nez p1, :cond_0
***************
*** 325,337 ****
--- 356,371 ----
      .parameter "m"
  
      .prologue
+     .line 148
      new-instance v0, Landroid/telephony/ServiceState;
  
      invoke-direct {v0}, Landroid/telephony/ServiceState;-><init>()V
  
+     .line 149
      .local v0, ret:Landroid/telephony/ServiceState;
      invoke-direct {v0, p0}, Landroid/telephony/ServiceState;->setFromNotifierBundle(Landroid/os/Bundle;)V
  
+     .line 150
      return-object v0
  .end method
  
***************
*** 340,349 ****
--- 374,386 ----
      .parameter "rt"
  
      .prologue
+     .line 401
      packed-switch p0, :pswitch_data_0
  
+     .line 451
      const-string v0, "Unexpected"
  
+     .line 452
      .local v0, rtString:Ljava/lang/String;
      const-string v1, "PHONE"
  
***************
*** 367,372 ****
--- 404,410 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 455
      :goto_0
      new-instance v1, Ljava/lang/StringBuilder;
  
***************
*** 392,509 ****
--- 430,580 ----
  
      return-object v1
  
+     .line 403
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_0
      const-string v0, "Unknown"
  
+     .line 404
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 406
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_1
      const-string v0, "GPRS"
  
+     .line 407
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 409
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_2
      const-string v0, "EDGE"
  
+     .line 410
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 412
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_3
      const-string v0, "UMTS"
  
+     .line 413
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 415
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_4
      const-string v0, "CDMA-IS95A"
  
+     .line 416
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 418
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_5
      const-string v0, "CDMA-IS95B"
  
+     .line 419
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 421
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_6
      const-string v0, "1xRTT"
  
+     .line 422
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 424
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_7
      const-string v0, "EvDo-rev.0"
  
+     .line 425
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 427
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_8
      const-string v0, "EvDo-rev.A"
  
+     .line 428
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 430
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_9
      const-string v0, "HSDPA"
  
+     .line 431
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 433
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_a
      const-string v0, "HSUPA"
  
+     .line 434
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 436
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_b
      const-string v0, "HSPA"
  
+     .line 437
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 439
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_c
      const-string v0, "EvDo-rev.B"
  
+     .line 440
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 442
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_d
      const-string v0, "eHRPD"
  
+     .line 443
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 445
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_e
      const-string v0, "LTE"
  
+     .line 446
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 448
      .end local v0           #rtString:Ljava/lang/String;
      :pswitch_f
      const-string v0, "HSPAP"
  
+     .line 449
      .restart local v0       #rtString:Ljava/lang/String;
      goto :goto_0
  
+     .line 401
      nop
  
      :pswitch_data_0
***************
*** 532,538 ****
      .parameter "m"
  
      .prologue
!     const-string v0, "state"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
--- 603,610 ----
      .parameter "m"
  
      .prologue
!     .line 584
!     const-string/jumbo v0, "state"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
***************
*** 540,546 ****
  
      iput v0, p0, Landroid/telephony/ServiceState;->mState:I
  
!     const-string v0, "roaming"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getBoolean(Ljava/lang/String;)Z
  
--- 612,619 ----
  
      iput v0, p0, Landroid/telephony/ServiceState;->mState:I
  
!     .line 585
!     const-string/jumbo v0, "roaming"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getBoolean(Ljava/lang/String;)Z
  
***************
*** 548,554 ****
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
!     const-string v0, "operator-alpha-long"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getString(Ljava/lang/String;)Ljava/lang/String;
  
--- 621,628 ----
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
!     .line 586
!     const-string/jumbo v0, "operator-alpha-long"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getString(Ljava/lang/String;)Ljava/lang/String;
  
***************
*** 556,562 ****
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
!     const-string v0, "operator-alpha-short"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getString(Ljava/lang/String;)Ljava/lang/String;
  
--- 630,637 ----
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
!     .line 587
!     const-string/jumbo v0, "operator-alpha-short"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getString(Ljava/lang/String;)Ljava/lang/String;
  
***************
*** 564,570 ****
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
!     const-string v0, "operator-numeric"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getString(Ljava/lang/String;)Ljava/lang/String;
  
--- 639,646 ----
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
!     .line 588
!     const-string/jumbo v0, "operator-numeric"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getString(Ljava/lang/String;)Ljava/lang/String;
  
***************
*** 572,578 ****
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
!     const-string v0, "manual"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getBoolean(Ljava/lang/String;)Z
  
--- 648,655 ----
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
!     .line 589
!     const-string/jumbo v0, "manual"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getBoolean(Ljava/lang/String;)Z
  
***************
*** 580,586 ****
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
!     const-string v0, "radioTechnology"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
--- 657,664 ----
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
!     .line 590
!     const-string/jumbo v0, "radioTechnology"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
***************
*** 588,593 ****
--- 666,672 ----
  
      iput v0, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
+     .line 591
      const-string v0, "cssIndicator"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getBoolean(Ljava/lang/String;)Z
***************
*** 596,602 ****
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
!     const-string v0, "networkId"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
--- 675,682 ----
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
!     .line 592
!     const-string/jumbo v0, "networkId"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
***************
*** 604,610 ****
  
      iput v0, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
!     const-string v0, "systemId"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
--- 684,691 ----
  
      iput v0, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
!     .line 593
!     const-string/jumbo v0, "systemId"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
***************
*** 612,617 ****
--- 693,699 ----
  
      iput v0, p0, Landroid/telephony/ServiceState;->mSystemId:I
  
+     .line 594
      const-string v0, "cdmaRoamingIndicator"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
***************
*** 620,625 ****
--- 702,708 ----
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
+     .line 595
      const-string v0, "cdmaDefaultRoamingIndicator"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
***************
*** 628,633 ****
--- 711,717 ----
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
+     .line 596
      const-string v0, "emergencyOnly"
  
      invoke-virtual {p1, v0}, Landroid/os/Bundle;->getBoolean(Ljava/lang/String;)Z
***************
*** 636,641 ****
--- 720,726 ----
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
  
+     .line 597
      return-void
  .end method
  
***************
*** 650,685 ****
--- 735,786 ----
  
      const/4 v0, -0x1
  
+     .line 477
      iput p1, p0, Landroid/telephony/ServiceState;->mState:I
  
+     .line 478
      iput-boolean v1, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
+     .line 479
      iput-object v2, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
+     .line 480
      iput-object v2, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
+     .line 481
      iput-object v2, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
+     .line 482
      iput-boolean v1, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
+     .line 483
      iput v1, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
+     .line 484
      iput-boolean v1, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
+     .line 485
      iput v0, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
+     .line 486
      iput v0, p0, Landroid/telephony/ServiceState;->mSystemId:I
  
+     .line 487
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
+     .line 488
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
+     .line 489
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconIndex:I
  
+     .line 490
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconMode:I
  
+     .line 491
      iput-boolean v1, p0, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
  
+     .line 492
      return-void
  .end method
  
***************
*** 690,755 ****
--- 791,872 ----
      .parameter "s"
  
      .prologue
+     .line 169
      iget v0, p1, Landroid/telephony/ServiceState;->mState:I
  
      iput v0, p0, Landroid/telephony/ServiceState;->mState:I
  
+     .line 170
      iget-boolean v0, p1, Landroid/telephony/ServiceState;->mRoaming:Z
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
+     .line 171
      iget-object v0, p1, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
+     .line 172
      iget-object v0, p1, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
+     .line 173
      iget-object v0, p1, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
+     .line 174
      iget-boolean v0, p1, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
+     .line 175
      iget v0, p1, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
      iput v0, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
+     .line 176
      iget-boolean v0, p1, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
+     .line 177
      iget v0, p1, Landroid/telephony/ServiceState;->mNetworkId:I
  
      iput v0, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
+     .line 178
      iget v0, p1, Landroid/telephony/ServiceState;->mSystemId:I
  
      iput v0, p0, Landroid/telephony/ServiceState;->mSystemId:I
  
+     .line 179
      iget v0, p1, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
+     .line 180
      iget v0, p1, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
+     .line 181
      iget v0, p1, Landroid/telephony/ServiceState;->mCdmaEriIconIndex:I
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconIndex:I
  
+     .line 182
      iget v0, p1, Landroid/telephony/ServiceState;->mCdmaEriIconMode:I
  
      iput v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconMode:I
  
+     .line 183
      iget-boolean v0, p1, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
  
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
  
+     .line 184
      return-void
  .end method
  
***************
*** 757,762 ****
--- 874,880 ----
      .locals 1
  
      .prologue
+     .line 226
      const/4 v0, 0x0
  
      return v0
***************
*** 769,774 ****
--- 887,893 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 365
      :try_start_0
      move-object v0, p1
  
***************
*** 778,797 ****
--- 897,921 ----
      :try_end_0
      .catch Ljava/lang/ClassCastException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 370
      .local v2, s:Landroid/telephony/ServiceState;
      if-nez p1, :cond_1
  
+     .line 374
      .end local v2           #s:Landroid/telephony/ServiceState;
      :cond_0
      :goto_0
      return v3
  
+     .line 366
      :catch_0
      move-exception v1
  
+     .line 367
      .local v1, ex:Ljava/lang/ClassCastException;
      goto :goto_0
  
+     .line 374
      .end local v1           #ex:Ljava/lang/ClassCastException;
      .restart local v2       #s:Landroid/telephony/ServiceState;
      :cond_1
***************
*** 967,979 ****
      .parameter "m"
  
      .prologue
!     const-string v0, "state"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mState:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
!     const-string v0, "roaming"
  
      iget-boolean v1, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
--- 1091,1105 ----
      .parameter "m"
  
      .prologue
!     .line 606
!     const-string/jumbo v0, "state"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mState:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
!     .line 607
!     const-string/jumbo v0, "roaming"
  
      iget-boolean v1, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
***************
*** 987,1011 ****
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putBoolean(Ljava/lang/String;Z)V
  
!     const-string v0, "operator-alpha-long"
  
      iget-object v1, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v0, "operator-alpha-short"
  
      iget-object v1, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v0, "operator-numeric"
  
      iget-object v1, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v0, "manual"
  
      iget-boolean v1, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
--- 1113,1141 ----
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putBoolean(Ljava/lang/String;Z)V
  
!     .line 608
!     const-string/jumbo v0, "operator-alpha-long"
  
      iget-object v1, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 609
!     const-string/jumbo v0, "operator-alpha-short"
  
      iget-object v1, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 610
!     const-string/jumbo v0, "operator-numeric"
  
      iget-object v1, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 611
!     const-string/jumbo v0, "manual"
  
      iget-boolean v1, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
***************
*** 1019,1060 ****
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putBoolean(Ljava/lang/String;Z)V
  
!     const-string v0, "radioTechnology"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
      const-string v0, "cssIndicator"
  
      iget-boolean v1, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putBoolean(Ljava/lang/String;Z)V
  
!     const-string v0, "networkId"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
!     const-string v0, "systemId"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mSystemId:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
      const-string v0, "cdmaRoamingIndicator"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
      const-string v0, "cdmaDefaultRoamingIndicator"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
      const-string v0, "emergencyOnly"
  
      iget-boolean v1, p0, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
--- 1149,1197 ----
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putBoolean(Ljava/lang/String;Z)V
  
!     .line 612
!     const-string/jumbo v0, "radioTechnology"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
+     .line 613
      const-string v0, "cssIndicator"
  
      iget-boolean v1, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putBoolean(Ljava/lang/String;Z)V
  
!     .line 614
!     const-string/jumbo v0, "networkId"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
!     .line 615
!     const-string/jumbo v0, "systemId"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mSystemId:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
+     .line 616
      const-string v0, "cdmaRoamingIndicator"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
+     .line 617
      const-string v0, "cdmaDefaultRoamingIndicator"
  
      iget v1, p0, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
+     .line 618
      const-string v0, "emergencyOnly"
  
      iget-boolean v1, p0, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
***************
*** 1069,1074 ****
--- 1206,1212 ----
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->putBoolean(Ljava/lang/String;Z)V
  
+     .line 619
      return-void
  .end method
  
***************
*** 1076,1081 ****
--- 1214,1220 ----
      .locals 1
  
      .prologue
+     .line 282
      iget v0, p0, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
      return v0
***************
*** 1085,1090 ****
--- 1224,1230 ----
      .locals 1
  
      .prologue
+     .line 289
      iget v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconIndex:I
  
      return v0
***************
*** 1094,1099 ****
--- 1234,1240 ----
      .locals 1
  
      .prologue
+     .line 296
      iget v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconMode:I
  
      return v0
***************
*** 1103,1108 ****
--- 1244,1250 ----
      .locals 1
  
      .prologue
+     .line 275
      iget v0, p0, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
      return v0
***************
*** 1112,1117 ****
--- 1254,1260 ----
      .locals 1
  
      .prologue
+     .line 645
      iget-boolean v0, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
      if-eqz v0, :cond_0
***************
*** 1131,1136 ****
--- 1274,1280 ----
      .locals 1
  
      .prologue
+     .line 344
      iget-boolean v0, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
      return v0
***************
*** 1140,1145 ****
--- 1284,1290 ----
      .locals 1
  
      .prologue
+     .line 650
      iget v0, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
      return v0
***************
*** 1149,1154 ****
--- 1294,1300 ----
      .locals 1
  
      .prologue
+     .line 308
      iget-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
      return-object v0
***************
*** 1158,1163 ****
--- 1304,1310 ----
      .locals 1
  
      .prologue
+     .line 319
      iget-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
      return-object v0
***************
*** 1167,1172 ****
--- 1314,1320 ----
      .locals 1
  
      .prologue
+     .line 335
      iget-object v0, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
      return-object v0
***************
*** 1176,1181 ****
--- 1324,1330 ----
      .locals 1
  
      .prologue
+     .line 640
      iget v0, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
      return v0
***************
*** 1185,1190 ****
--- 1334,1340 ----
      .locals 1
  
      .prologue
+     .line 261
      iget-boolean v0, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
      return v0
***************
*** 1194,1199 ****
--- 1344,1350 ----
      .locals 1
  
      .prologue
+     .line 249
      iget v0, p0, Landroid/telephony/ServiceState;->mState:I
  
      return v0
***************
*** 1203,1208 ****
--- 1354,1360 ----
      .locals 1
  
      .prologue
+     .line 655
      iget v0, p0, Landroid/telephony/ServiceState;->mSystemId:I
  
      return v0
***************
*** 1216,1221 ****
--- 1368,1374 ----
  
      const/4 v2, 0x0
  
+     .line 349
      iget v0, p0, Landroid/telephony/ServiceState;->mState:I
  
      mul-int/lit16 v3, v0, 0x1234
***************
*** 1329,1334 ****
--- 1482,1488 ----
      .locals 1
  
      .prologue
+     .line 268
      iget-boolean v0, p0, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
  
      return v0
***************
*** 1339,1346 ****
--- 1493,1502 ----
      .parameter "roaming"
  
      .prologue
+     .line 529
      iput p1, p0, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
+     .line 530
      return-void
  .end method
  
***************
*** 1349,1356 ****
--- 1505,1514 ----
      .parameter "index"
  
      .prologue
+     .line 536
      iput p1, p0, Landroid/telephony/ServiceState;->mCdmaEriIconIndex:I
  
+     .line 537
      return-void
  .end method
  
***************
*** 1359,1366 ****
--- 1517,1526 ----
      .parameter "mode"
  
      .prologue
+     .line 543
      iput p1, p0, Landroid/telephony/ServiceState;->mCdmaEriIconMode:I
  
+     .line 544
      return-void
  .end method
  
***************
*** 1369,1376 ****
--- 1529,1538 ----
      .parameter "roaming"
  
      .prologue
+     .line 522
      iput p1, p0, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
+     .line 523
      return-void
  .end method
  
***************
*** 1379,1384 ****
--- 1541,1547 ----
      .parameter "css"
  
      .prologue
+     .line 629
      if-eqz p1, :cond_0
  
      const/4 v0, 0x1
***************
*** 1386,1393 ****
--- 1549,1558 ----
      :goto_0
      iput-boolean v0, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
+     .line 630
      return-void
  
+     .line 629
      :cond_0
      const/4 v0, 0x0
  
***************
*** 1399,1406 ****
--- 1564,1573 ----
      .parameter "emergencyOnly"
  
      .prologue
+     .line 515
      iput-boolean p1, p0, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
  
+     .line 516
      return-void
  .end method
  
***************
*** 1409,1416 ****
--- 1576,1585 ----
      .parameter "isManual"
  
      .prologue
+     .line 563
      iput-boolean p1, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
+     .line 564
      return-void
  .end method
  
***************
*** 1419,1426 ****
--- 1588,1597 ----
      .parameter "longName"
  
      .prologue
+     .line 559
      iput-object p1, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
+     .line 560
      return-void
  .end method
  
***************
*** 1431,1442 ****
--- 1602,1617 ----
      .parameter "numeric"
  
      .prologue
+     .line 547
      iput-object p1, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
+     .line 548
      iput-object p2, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
+     .line 549
      iput-object p3, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
+     .line 550
      return-void
  .end method
  
***************
*** 1445,1452 ****
--- 1620,1629 ----
      .parameter "state"
  
      .prologue
+     .line 624
      iput p1, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
+     .line 625
      return-void
  .end method
  
***************
*** 1455,1462 ****
--- 1632,1641 ----
      .parameter "roaming"
  
      .prologue
+     .line 507
      iput-boolean p1, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
+     .line 508
      return-void
  .end method
  
***************
*** 1465,1472 ****
--- 1644,1653 ----
      .parameter "state"
  
      .prologue
+     .line 503
      iput p1, p0, Landroid/telephony/ServiceState;->mState:I
  
+     .line 504
      return-void
  .end method
  
***************
*** 1474,1483 ****
--- 1655,1666 ----
      .locals 1
  
      .prologue
+     .line 499
      const/4 v0, 0x3
  
      invoke-direct {p0, v0}, Landroid/telephony/ServiceState;->setNullState(I)V
  
+     .line 500
      return-void
  .end method
  
***************
*** 1485,1494 ****
--- 1668,1679 ----
      .locals 1
  
      .prologue
+     .line 495
      const/4 v0, 0x1
  
      invoke-direct {p0, v0}, Landroid/telephony/ServiceState;->setNullState(I)V
  
+     .line 496
      return-void
  .end method
  
***************
*** 1498,1507 ****
--- 1683,1695 ----
      .parameter "networkId"
  
      .prologue
+     .line 634
      iput p1, p0, Landroid/telephony/ServiceState;->mSystemId:I
  
+     .line 635
      iput p2, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
+     .line 636
      return-void
  .end method
  
***************
*** 1509,1520 ****
--- 1697,1710 ----
      .locals 3
  
      .prologue
+     .line 460
      iget v1, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
      invoke-static {v1}, Landroid/telephony/ServiceState;->radioTechnologyToString(I)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 462
      .local v0, radioTechnology:Ljava/lang/String;
      new-instance v1, Ljava/lang/StringBuilder;
  
***************
*** 1536,1542 ****
  
      if-eqz v1, :cond_0
  
!     const-string v1, "roaming"
  
      :goto_0
      invoke-virtual {v2, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
--- 1726,1732 ----
  
      if-eqz v1, :cond_0
  
!     const-string/jumbo v1, "roaming"
  
      :goto_0
      invoke-virtual {v2, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 1715,1724 ****
--- 1905,1916 ----
  
      const/4 v2, 0x0
  
+     .line 208
      iget v0, p0, Landroid/telephony/ServiceState;->mState:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 209
      iget-boolean v0, p0, Landroid/telephony/ServiceState;->mRoaming:Z
  
      if-eqz v0, :cond_0
***************
*** 1728,1745 ****
--- 1920,1941 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 210
      iget-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaLong:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 211
      iget-object v0, p0, Landroid/telephony/ServiceState;->mOperatorAlphaShort:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 212
      iget-object v0, p0, Landroid/telephony/ServiceState;->mOperatorNumeric:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 213
      iget-boolean v0, p0, Landroid/telephony/ServiceState;->mIsManualNetworkSelection:Z
  
      if-eqz v0, :cond_1
***************
*** 1749,1758 ****
--- 1945,1956 ----
      :goto_1
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 214
      iget v0, p0, Landroid/telephony/ServiceState;->mRadioTechnology:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 215
      iget-boolean v0, p0, Landroid/telephony/ServiceState;->mCssIndicator:Z
  
      if-eqz v0, :cond_2
***************
*** 1762,1791 ****
--- 1960,1996 ----
      :goto_2
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 216
      iget v0, p0, Landroid/telephony/ServiceState;->mNetworkId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 217
      iget v0, p0, Landroid/telephony/ServiceState;->mSystemId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 218
      iget v0, p0, Landroid/telephony/ServiceState;->mCdmaRoamingIndicator:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 219
      iget v0, p0, Landroid/telephony/ServiceState;->mCdmaDefaultRoamingIndicator:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 220
      iget v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconIndex:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 221
      iget v0, p0, Landroid/telephony/ServiceState;->mCdmaEriIconMode:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 222
      iget-boolean v0, p0, Landroid/telephony/ServiceState;->mIsEmergencyOnly:Z
  
      if-eqz v0, :cond_3
***************
*** 1793,1817 ****
--- 1998,2027 ----
      :goto_3
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 223
      return-void
  
      :cond_0
      move v0, v2
  
+     .line 209
      goto :goto_0
  
      :cond_1
      move v0, v2
  
+     .line 213
      goto :goto_1
  
      :cond_2
      move v0, v2
  
+     .line 215
      goto :goto_2
  
      :cond_3
      move v1, v2
  
+     .line 222
      goto :goto_3
  .end method
