*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/textservice/ITextServicesManager$Stub.smali	2012-03-23 19:37:29.242308315 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/textservice/ITextServicesManager$Stub.smali.original	2012-03-23 19:34:46.006331711 -0500
***************
*** 50,61 ****
--- 50,64 ----
      .locals 1
  
      .prologue
+     .line 18
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 19
      const-string v0, "com.android.internal.textservice.ITextServicesManager"
  
      invoke-virtual {p0, p0, v0}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->attachInterface(Landroid/os/IInterface;Ljava/lang/String;)V
  
+     .line 20
      return-void
  .end method
  
***************
*** 64,76 ****
--- 67,83 ----
      .parameter "obj"
  
      .prologue
+     .line 27
      if-nez p0, :cond_0
  
+     .line 28
      const/4 v0, 0x0
  
+     .line 34
      :goto_0
      return-object v0
  
+     .line 30
      :cond_0
      const-string v1, "com.android.internal.textservice.ITextServicesManager"
  
***************
*** 78,83 ****
--- 85,91 ----
  
      move-result-object v0
  
+     .line 31
      .local v0, iin:Landroid/os/IInterface;
      if-eqz v0, :cond_1
  
***************
*** 85,94 ****
--- 93,104 ----
  
      if-eqz v1, :cond_1
  
+     .line 32
      check-cast v0, Lcom/android/internal/textservice/ITextServicesManager;
  
      goto :goto_0
  
+     .line 34
      :cond_1
      new-instance v0, Lcom/android/internal/textservice/ITextServicesManager$Stub$Proxy;
  
***************
*** 104,109 ****
--- 114,120 ----
      .locals 0
  
      .prologue
+     .line 38
      return-object p0
  .end method
  
***************
*** 124,131 ****
--- 135,144 ----
  
      const/4 v7, 0x1
  
+     .line 42
      sparse-switch p1, :sswitch_data_0
  
+     .line 157
      invoke-super {p0, p1, p2, p3, p4}, Landroid/os/Binder;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
      move-result v7
***************
*** 133,138 ****
--- 146,152 ----
      :goto_0
      return v7
  
+     .line 46
      :sswitch_0
      const-string v0, "com.android.internal.textservice.ITextServicesManager"
  
***************
*** 140,175 ****
--- 154,198 ----
  
      goto :goto_0
  
+     .line 51
      :sswitch_1
      const-string v8, "com.android.internal.textservice.ITextServicesManager"
  
      invoke-virtual {p2, v8}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 53
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 54
      .local v1, _arg0:Ljava/lang/String;
      invoke-virtual {p0, v1}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->getCurrentSpellChecker(Ljava/lang/String;)Landroid/view/textservice/SpellCheckerInfo;
  
      move-result-object v6
  
+     .line 55
      .local v6, _result:Landroid/view/textservice/SpellCheckerInfo;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 56
      if-eqz v6, :cond_0
  
+     .line 57
      invoke-virtual {p3, v7}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 58
      invoke-virtual {v6, p3, v7}, Landroid/view/textservice/SpellCheckerInfo;->writeToParcel(Landroid/os/Parcel;I)V
  
      goto :goto_0
  
+     .line 61
      :cond_0
      invoke-virtual {p3, v0}, Landroid/os/Parcel;->writeInt(I)V
  
      goto :goto_0
  
+     .line 67
      .end local v1           #_arg0:Ljava/lang/String;
      .end local v6           #_result:Landroid/view/textservice/SpellCheckerInfo;
      :sswitch_2
***************
*** 177,186 ****
--- 200,211 ----
  
      invoke-virtual {p2, v8}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 69
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 71
      .restart local v1       #_arg0:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 190,208 ****
--- 215,238 ----
  
      move v2, v7
  
+     .line 72
      .local v2, _arg1:Z
      :goto_1
      invoke-virtual {p0, v1, v2}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->getCurrentSpellCheckerSubtype(Ljava/lang/String;Z)Landroid/view/textservice/SpellCheckerSubtype;
  
      move-result-object v6
  
+     .line 73
      .local v6, _result:Landroid/view/textservice/SpellCheckerSubtype;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 74
      if-eqz v6, :cond_2
  
+     .line 75
      invoke-virtual {p3, v7}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 76
      invoke-virtual {v6, p3, v7}, Landroid/view/textservice/SpellCheckerSubtype;->writeToParcel(Landroid/os/Parcel;I)V
  
      goto :goto_0
***************
*** 212,219 ****
--- 242,251 ----
      :cond_1
      move v2, v0
  
+     .line 71
      goto :goto_1
  
+     .line 79
      .restart local v2       #_arg1:Z
      .restart local v6       #_result:Landroid/view/textservice/SpellCheckerSubtype;
      :cond_2
***************
*** 221,226 ****
--- 253,259 ----
  
      goto :goto_0
  
+     .line 85
      .end local v1           #_arg0:Ljava/lang/String;
      .end local v2           #_arg1:Z
      .end local v6           #_result:Landroid/view/textservice/SpellCheckerSubtype;
***************
*** 229,243 ****
--- 262,279 ----
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 87
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 89
      .restart local v1       #_arg0:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 91
      .local v2, _arg1:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
***************
*** 247,252 ****
--- 283,289 ----
  
      move-result-object v3
  
+     .line 93
      .local v3, _arg2:Lcom/android/internal/textservice/ITextServicesSessionListener;
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
***************
*** 256,261 ****
--- 293,299 ----
  
      move-result-object v4
  
+     .line 95
      .local v4, _arg3:Lcom/android/internal/textservice/ISpellCheckerSessionListener;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 263,268 ****
--- 301,307 ----
  
      if-eqz v0, :cond_3
  
+     .line 96
      sget-object v0, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 275,284 ****
--- 314,325 ----
      :goto_2
      move-object v0, p0
  
+     .line 101
      invoke-virtual/range {v0 .. v5}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->getSpellCheckerService(Ljava/lang/String;Ljava/lang/String;Lcom/android/internal/textservice/ITextServicesSessionListener;Lcom/android/internal/textservice/ISpellCheckerSessionListener;Landroid/os/Bundle;)V
  
      goto :goto_0
  
+     .line 99
      .end local v5           #_arg4:Landroid/os/Bundle;
      :cond_3
      const/4 v5, 0x0
***************
*** 286,291 ****
--- 327,333 ----
      .restart local v5       #_arg4:Landroid/os/Bundle;
      goto :goto_2
  
+     .line 106
      .end local v1           #_arg0:Ljava/lang/String;
      .end local v2           #_arg1:Ljava/lang/String;
      .end local v3           #_arg2:Lcom/android/internal/textservice/ITextServicesSessionListener;
***************
*** 296,301 ****
--- 338,344 ----
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 108
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
***************
*** 304,334 ****
--- 347,383 ----
  
      move-result-object v1
  
+     .line 109
      .local v1, _arg0:Lcom/android/internal/textservice/ISpellCheckerSessionListener;
      invoke-virtual {p0, v1}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->finishSpellCheckerService(Lcom/android/internal/textservice/ISpellCheckerSessionListener;)V
  
      goto/16 :goto_0
  
+     .line 114
      .end local v1           #_arg0:Lcom/android/internal/textservice/ISpellCheckerSessionListener;
      :sswitch_5
      const-string v0, "com.android.internal.textservice.ITextServicesManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 116
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 118
      .local v1, _arg0:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 119
      .restart local v2       #_arg1:Ljava/lang/String;
      invoke-virtual {p0, v1, v2}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->setCurrentSpellChecker(Ljava/lang/String;Ljava/lang/String;)V
  
      goto/16 :goto_0
  
+     .line 124
      .end local v1           #_arg0:Ljava/lang/String;
      .end local v2           #_arg1:Ljava/lang/String;
      :sswitch_6
***************
*** 336,355 ****
--- 385,408 ----
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 126
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 128
      .restart local v1       #_arg0:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
+     .line 129
      .local v2, _arg1:I
      invoke-virtual {p0, v1, v2}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->setCurrentSpellCheckerSubtype(Ljava/lang/String;I)V
  
      goto/16 :goto_0
  
+     .line 134
      .end local v1           #_arg0:Ljava/lang/String;
      .end local v2           #_arg1:I
      :sswitch_7
***************
*** 357,362 ****
--- 410,416 ----
  
      invoke-virtual {p2, v8}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 136
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v8
***************
*** 365,370 ****
--- 419,425 ----
  
      move v1, v7
  
+     .line 137
      .local v1, _arg0:Z
      :goto_3
      invoke-virtual {p0, v1}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->setSpellCheckerEnabled(Z)V
***************
*** 375,394 ****
--- 430,454 ----
      :cond_4
      move v1, v0
  
+     .line 136
      goto :goto_3
  
+     .line 142
      :sswitch_8
      const-string v8, "com.android.internal.textservice.ITextServicesManager"
  
      invoke-virtual {p2, v8}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 143
      invoke-virtual {p0}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->isSpellCheckerEnabled()Z
  
      move-result v6
  
+     .line 144
      .local v6, _result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 145
      if-eqz v6, :cond_5
  
      move v0, v7
***************
*** 398,420 ****
--- 458,485 ----
  
      goto/16 :goto_0
  
+     .line 150
      .end local v6           #_result:Z
      :sswitch_9
      const-string v0, "com.android.internal.textservice.ITextServicesManager"
  
      invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 151
      invoke-virtual {p0}, Lcom/android/internal/textservice/ITextServicesManager$Stub;->getEnabledSpellCheckers()[Landroid/view/textservice/SpellCheckerInfo;
  
      move-result-object v6
  
+     .line 152
      .local v6, _result:[Landroid/view/textservice/SpellCheckerInfo;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 153
      invoke-virtual {p3, v6, v7}, Landroid/os/Parcel;->writeTypedArray([Landroid/os/Parcelable;I)V
  
      goto/16 :goto_0
  
+     .line 42
      nop
  
      :sswitch_data_0
