*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState.smali	2012-03-23 19:36:55.428691724 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState.smali.original	2012-03-23 19:34:45.413320897 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 487
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .locals 0
  
      .prologue
+     .line 491
      return-void
  .end method
  
***************
*** 47,66 ****
--- 49,73 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 496
      iget v4, p1, Landroid/os/Message;->what:I
  
      sparse-switch v4, :sswitch_data_0
  
+     .line 535
      const/4 v3, 0x0
  
+     .line 537
      :goto_0
      return v3
  
+     .line 498
      :sswitch_0
      new-instance v1, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState$1;
  
      invoke-direct {v1, p0}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState$1;-><init>(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;)V
  
+     .line 510
      .local v1, listener:Landroid/content/DialogInterface$OnClickListener;
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 86,96 ****
--- 93,105 ----
  
      if-eq v4, v5, :cond_1
  
+     .line 512
      :cond_0
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v2
  
+     .line 513
      .local v2, r:Landroid/content/res/Resources;
      new-instance v4, Landroid/app/AlertDialog$Builder;
  
***************
*** 149,154 ****
--- 158,164 ----
  
      move-result-object v0
  
+     .line 519
      .local v0, dialog:Landroid/app/AlertDialog;
      invoke-virtual {v0}, Landroid/app/AlertDialog;->getWindow()Landroid/view/Window;
  
***************
*** 158,165 ****
--- 168,177 ----
  
      invoke-virtual {v4, v5}, Landroid/view/Window;->setType(I)V
  
+     .line 520
      invoke-virtual {v0}, Landroid/app/AlertDialog;->show()V
  
+     .line 521
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v5, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 172,177 ****
--- 184,190 ----
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
      invoke-static {v4, v5}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$2000(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Lcom/android/internal/util/IState;)V
  
+     .line 526
      .end local v0           #dialog:Landroid/app/AlertDialog;
      .end local v2           #r:Landroid/content/res/Resources;
      :goto_1
***************
*** 184,189 ****
--- 197,203 ----
  
      goto :goto_0
  
+     .line 523
      :cond_1
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 198,203 ****
--- 212,218 ----
  
      invoke-virtual {v4, v5}, Lcom/android/internal/util/AsyncChannel;->sendMessage(I)V
  
+     .line 524
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v5, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 212,217 ****
--- 227,233 ----
  
      goto :goto_1
  
+     .line 529
      .end local v1           #listener:Landroid/content/DialogInterface$OnClickListener;
      :sswitch_1
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 223,228 ****
--- 239,245 ----
  
      goto/16 :goto_0
  
+     .line 532
      :sswitch_2
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 233,238 ****
--- 250,256 ----
  
      goto/16 :goto_0
  
+     .line 496
      nop
  
      :sswitch_data_0
