*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pConfig.smali	2012-03-23 19:36:56.189705601 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pConfig.smali.original	2012-03-23 19:34:45.423321081 -0500
***************
*** 42,47 ****
--- 42,48 ----
      .locals 1
  
      .prologue
+     .line 141
      new-instance v0, Landroid/net/wifi/p2p/WifiP2pConfig$1;
  
      invoke-direct {v0}, Landroid/net/wifi/p2p/WifiP2pConfig$1;-><init>()V
***************
*** 55,82 ****
--- 56,89 ----
      .locals 2
  
      .prologue
+     .line 62
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 47
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->groupOwnerIntent:I
  
+     .line 60
      sget-object v0, Landroid/net/wifi/p2p/WifiP2pConfig$Persist;->SYSTEM_DEFAULT:Landroid/net/wifi/p2p/WifiP2pConfig$Persist;
  
      iput-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->persist:Landroid/net/wifi/p2p/WifiP2pConfig$Persist;
  
+     .line 64
      new-instance v0, Landroid/net/wifi/WpsInfo;
  
      invoke-direct {v0}, Landroid/net/wifi/WpsInfo;-><init>()V
  
      iput-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
  
+     .line 65
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
  
      const/4 v1, 0x0
  
      iput v1, v0, Landroid/net/wifi/WpsInfo;->setup:I
  
+     .line 66
      return-void
  .end method
  
***************
*** 85,106 ****
--- 92,119 ----
      .parameter "source"
  
      .prologue
+     .line 123
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 47
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->groupOwnerIntent:I
  
+     .line 60
      sget-object v0, Landroid/net/wifi/p2p/WifiP2pConfig$Persist;->SYSTEM_DEFAULT:Landroid/net/wifi/p2p/WifiP2pConfig$Persist;
  
      iput-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->persist:Landroid/net/wifi/p2p/WifiP2pConfig$Persist;
  
+     .line 124
      if-eqz p1, :cond_0
  
+     .line 125
      iget-object v0, p1, Landroid/net/wifi/p2p/WifiP2pConfig;->deviceAddress:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->deviceAddress:Ljava/lang/String;
  
+     .line 126
      new-instance v0, Landroid/net/wifi/WpsInfo;
  
      iget-object v1, p1, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
***************
*** 109,122 ****
--- 122,138 ----
  
      iput-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
  
+     .line 127
      iget v0, p1, Landroid/net/wifi/p2p/WifiP2pConfig;->groupOwnerIntent:I
  
      iput v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->groupOwnerIntent:I
  
+     .line 128
      iget-object v0, p1, Landroid/net/wifi/p2p/WifiP2pConfig;->persist:Landroid/net/wifi/p2p/WifiP2pConfig$Persist;
  
      iput-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->persist:Landroid/net/wifi/p2p/WifiP2pConfig$Persist;
  
+     .line 130
      :cond_0
      return-void
  .end method
***************
*** 137,158 ****
--- 153,179 ----
  
      const/4 v6, 0x2
  
+     .line 69
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 47
      const/4 v4, -0x1
  
      iput v4, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->groupOwnerIntent:I
  
+     .line 60
      sget-object v4, Landroid/net/wifi/p2p/WifiP2pConfig$Persist;->SYSTEM_DEFAULT:Landroid/net/wifi/p2p/WifiP2pConfig$Persist;
  
      iput-object v4, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->persist:Landroid/net/wifi/p2p/WifiP2pConfig$Persist;
  
+     .line 70
      const-string v4, " "
  
      invoke-virtual {p1, v4}, Ljava/lang/String;->split(Ljava/lang/String;)[Ljava/lang/String;
  
      move-result-object v3
  
+     .line 72
      .local v3, tokens:[Ljava/lang/String;
      array-length v4, v3
  
***************
*** 168,173 ****
--- 189,195 ----
  
      if-nez v4, :cond_1
  
+     .line 73
      :cond_0
      new-instance v4, Ljava/lang/IllegalArgumentException;
  
***************
*** 177,197 ****
--- 199,223 ----
  
      throw v4
  
+     .line 76
      :cond_1
      aget-object v4, v3, v8
  
      iput-object v4, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->deviceAddress:Ljava/lang/String;
  
+     .line 77
      new-instance v4, Landroid/net/wifi/WpsInfo;
  
      invoke-direct {v4}, Landroid/net/wifi/WpsInfo;-><init>()V
  
      iput-object v4, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
  
+     .line 79
      array-length v4, v3
  
      if-le v4, v6, :cond_2
  
+     .line 80
      aget-object v4, v3, v6
  
      const-string v5, "="
***************
*** 200,205 ****
--- 226,232 ----
  
      move-result-object v2
  
+     .line 83
      .local v2, nameVal:[Ljava/lang/String;
      const/4 v4, 0x1
  
***************
*** 212,242 ****
--- 239,275 ----
  
      move-result v0
  
+     .line 88
      .local v0, devPasswdId:I
      :goto_0
      packed-switch v0, :pswitch_data_0
  
+     .line 102
      :pswitch_0
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
  
      iput v7, v4, Landroid/net/wifi/WpsInfo;->setup:I
  
+     .line 106
      .end local v0           #devPasswdId:I
      .end local v2           #nameVal:[Ljava/lang/String;
      :cond_2
      :goto_1
      return-void
  
+     .line 84
      .restart local v2       #nameVal:[Ljava/lang/String;
      :catch_0
      move-exception v1
  
+     .line 85
      .local v1, e:Ljava/lang/NumberFormatException;
      const/4 v0, 0x0
  
      .restart local v0       #devPasswdId:I
      goto :goto_0
  
+     .line 90
      .end local v1           #e:Ljava/lang/NumberFormatException;
      :pswitch_1
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
***************
*** 247,252 ****
--- 280,286 ----
  
      goto :goto_1
  
+     .line 93
      :pswitch_2
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
  
***************
*** 254,259 ****
--- 288,294 ----
  
      goto :goto_1
  
+     .line 96
      :pswitch_3
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
  
***************
*** 261,266 ****
--- 296,302 ----
  
      goto :goto_1
  
+     .line 99
      :pswitch_4
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
  
***************
*** 268,273 ****
--- 304,310 ----
  
      goto :goto_1
  
+     .line 88
      nop
  
      :pswitch_data_0
***************
*** 287,292 ****
--- 324,330 ----
      .locals 1
  
      .prologue
+     .line 119
      const/4 v0, 0x0
  
      return v0
***************
*** 296,305 ****
--- 334,345 ----
      .locals 3
  
      .prologue
+     .line 109
      new-instance v0, Ljava/lang/StringBuffer;
  
      invoke-direct {v0}, Ljava/lang/StringBuffer;-><init>()V
  
+     .line 110
      .local v0, sbuf:Ljava/lang/StringBuffer;
      const-string v1, "\n address: "
  
***************
*** 311,316 ****
--- 351,357 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
  
+     .line 111
      const-string v1, "\n wps: "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
***************
*** 321,326 ****
--- 362,368 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuffer;->append(Ljava/lang/Object;)Ljava/lang/StringBuffer;
  
+     .line 112
      const-string v1, "\n groupOwnerIntent: "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
***************
*** 331,336 ****
--- 373,379 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuffer;->append(I)Ljava/lang/StringBuffer;
  
+     .line 113
      const-string v1, "\n persist: "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
***************
*** 345,350 ****
--- 388,394 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
  
+     .line 114
      invoke-virtual {v0}, Ljava/lang/StringBuffer;->toString()Ljava/lang/String;
  
      move-result-object v1
***************
*** 358,375 ****
--- 402,423 ----
      .parameter "flags"
  
      .prologue
+     .line 134
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->deviceAddress:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 135
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->wps:Landroid/net/wifi/WpsInfo;
  
      invoke-virtual {p1, v0, p2}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 136
      iget v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->groupOwnerIntent:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 137
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pConfig;->persist:Landroid/net/wifi/p2p/WifiP2pConfig$Persist;
  
      invoke-virtual {v0}, Landroid/net/wifi/p2p/WifiP2pConfig$Persist;->name()Ljava/lang/String;
***************
*** 378,382 ****
--- 426,431 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 138
      return-void
  .end method
