*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState$1.smali	2012-03-23 19:36:55.972701644 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState$1.smali.original	2012-03-23 19:34:45.421321043 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 498
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState$1;->this$2:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,51 ****
--- 43,54 ----
      .parameter "which"
  
      .prologue
+     .line 501
      const/4 v0, -0x1
  
      if-ne p2, v0, :cond_0
  
+     .line 502
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState$1;->this$2:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;
  
      iget-object v0, v0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 54,62 ****
--- 57,67 ----
  
      invoke-virtual {v0, v1}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->sendMessage(I)V
  
+     .line 506
      :goto_0
      return-void
  
+     .line 504
      :cond_0
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState$1;->this$2:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$P2pDisabledState;
  
