*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pInfo.smali	2012-03-23 19:36:55.401691233 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pInfo.smali.original	2012-03-23 19:34:45.413320897 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 1
  
      .prologue
+     .line 80
      new-instance v0, Landroid/net/wifi/p2p/WifiP2pInfo$1;
  
      invoke-direct {v0}, Landroid/net/wifi/p2p/WifiP2pInfo$1;-><init>()V
***************
*** 45,52 ****
--- 46,55 ----
      .locals 0
  
      .prologue
+     .line 41
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 42
      return-void
  .end method
  
***************
*** 55,76 ****
--- 58,85 ----
      .parameter "source"
  
      .prologue
+     .line 58
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 59
      if-eqz p1, :cond_0
  
+     .line 60
      iget-boolean v0, p1, Landroid/net/wifi/p2p/WifiP2pInfo;->groupFormed:Z
  
      iput-boolean v0, p0, Landroid/net/wifi/p2p/WifiP2pInfo;->groupFormed:Z
  
+     .line 61
      iget-boolean v0, p1, Landroid/net/wifi/p2p/WifiP2pInfo;->isGroupOwner:Z
  
      iput-boolean v0, p0, Landroid/net/wifi/p2p/WifiP2pInfo;->isGroupOwner:Z
  
+     .line 62
      iget-object v0, p1, Landroid/net/wifi/p2p/WifiP2pInfo;->groupOwnerAddress:Ljava/net/InetAddress;
  
      iput-object v0, p0, Landroid/net/wifi/p2p/WifiP2pInfo;->groupOwnerAddress:Ljava/net/InetAddress;
  
+     .line 64
      :cond_0
      return-void
  .end method
***************
*** 81,86 ****
--- 90,96 ----
      .locals 1
  
      .prologue
+     .line 54
      const/4 v0, 0x0
  
      return v0
***************
*** 90,99 ****
--- 100,111 ----
      .locals 3
  
      .prologue
+     .line 45
      new-instance v0, Ljava/lang/StringBuffer;
  
      invoke-direct {v0}, Ljava/lang/StringBuffer;-><init>()V
  
+     .line 46
      .local v0, sbuf:Ljava/lang/StringBuffer;
      const-string v1, "groupFormed: "
  
***************
*** 129,134 ****
--- 141,147 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuffer;->append(Ljava/lang/Object;)Ljava/lang/StringBuffer;
  
+     .line 49
      invoke-virtual {v0}, Ljava/lang/StringBuffer;->toString()Ljava/lang/String;
  
      move-result-object v1
***************
*** 146,151 ****
--- 159,165 ----
  
      const/4 v2, 0x0
  
+     .line 68
      iget-boolean v0, p0, Landroid/net/wifi/p2p/WifiP2pInfo;->groupFormed:Z
  
      if-eqz v0, :cond_0
***************
*** 155,160 ****
--- 169,175 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeByte(B)V
  
+     .line 69
      iget-boolean v0, p0, Landroid/net/wifi/p2p/WifiP2pInfo;->isGroupOwner:Z
  
      if-eqz v0, :cond_1
***************
*** 164,175 ****
--- 179,193 ----
      :goto_1
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeByte(B)V
  
+     .line 71
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pInfo;->groupOwnerAddress:Ljava/net/InetAddress;
  
      if-eqz v0, :cond_2
  
+     .line 72
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->writeByte(B)V
  
+     .line 73
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pInfo;->groupOwnerAddress:Ljava/net/InetAddress;
  
      invoke-virtual {v0}, Ljava/net/InetAddress;->getAddress()[B
***************
*** 178,196 ****
--- 196,218 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeByteArray([B)V
  
+     .line 77
      :goto_2
      return-void
  
      :cond_0
      move v0, v2
  
+     .line 68
      goto :goto_0
  
      :cond_1
      move v0, v2
  
+     .line 69
      goto :goto_1
  
+     .line 75
      :cond_2
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->writeByte(B)V
  
