*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService.smali	2012-03-23 19:36:55.819698856 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService.smali.original	2012-03-23 19:34:45.418320989 -0500
***************
*** 100,107 ****
--- 100,109 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 105
      sput v2, Landroid/net/wifi/p2p/WifiP2pService;->mGroupNegotiationTimeoutIndex:I
  
+     .line 164
      const/4 v0, 0x2
  
      new-array v0, v0, [Ljava/lang/String;
***************
*** 128,162 ****
      .prologue
      const/4 v5, 0x0
  
      invoke-direct {p0}, Landroid/net/wifi/p2p/IWifiP2pManager$Stub;-><init>()V
  
      const/4 v1, 0x1
  
      iput v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiState:I
  
      const/16 v1, 0xb
  
      iput v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiApState:I
  
      new-instance v1, Lcom/android/internal/util/AsyncChannel;
  
      invoke-direct {v1}, Lcom/android/internal/util/AsyncChannel;-><init>()V
  
      iput-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mReplyChannel:Lcom/android/internal/util/AsyncChannel;
  
      iput v5, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pRestartCount:I
  
      new-instance v1, Landroid/net/wifi/p2p/WifiP2pDevice;
  
      invoke-direct {v1}, Landroid/net/wifi/p2p/WifiP2pDevice;-><init>()V
  
      iput-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mThisDevice:Landroid/net/wifi/p2p/WifiP2pDevice;
  
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
  
!     const-string v1, "wifi.interface"
  
!     const-string v2, "wlan0"
  
      invoke-static {v1, v2}, Landroid/os/SystemProperties;->get(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
--- 130,172 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 167
      invoke-direct {p0}, Landroid/net/wifi/p2p/IWifiP2pManager$Stub;-><init>()V
  
+     .line 96
      const/4 v1, 0x1
  
      iput v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiState:I
  
+     .line 97
      const/16 v1, 0xb
  
      iput v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiApState:I
  
+     .line 100
      new-instance v1, Lcom/android/internal/util/AsyncChannel;
  
      invoke-direct {v1}, Lcom/android/internal/util/AsyncChannel;-><init>()V
  
      iput-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mReplyChannel:Lcom/android/internal/util/AsyncChannel;
  
+     .line 117
      iput v5, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pRestartCount:I
  
+     .line 153
      new-instance v1, Landroid/net/wifi/p2p/WifiP2pDevice;
  
      invoke-direct {v1}, Landroid/net/wifi/p2p/WifiP2pDevice;-><init>()V
  
      iput-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mThisDevice:Landroid/net/wifi/p2p/WifiP2pDevice;
  
+     .line 168
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
  
!     .line 170
!     const-string/jumbo v1, "wifi.interface"
  
!     const-string/jumbo v2, "wlan0"
  
      invoke-static {v1, v2}, Landroid/os/SystemProperties;->get(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
***************
*** 164,169 ****
--- 174,180 ----
  
      iput-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mInterface:Ljava/lang/String;
  
+     .line 171
      new-instance v1, Landroid/net/NetworkInfo;
  
      const/16 v2, 0xd
***************
*** 176,181 ****
--- 187,193 ----
  
      iput-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mNetworkInfo:Landroid/net/NetworkInfo;
  
+     .line 173
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v1}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
***************
*** 190,195 ****
--- 202,208 ----
  
      iput-boolean v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pSupported:Z
  
+     .line 176
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mThisDevice:Landroid/net/wifi/p2p/WifiP2pDevice;
  
      iget-object v2, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
***************
*** 206,211 ****
--- 219,225 ----
  
      iput-object v2, v1, Landroid/net/wifi/p2p/WifiP2pDevice;->primaryDeviceType:Ljava/lang/String;
  
+     .line 178
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mThisDevice:Landroid/net/wifi/p2p/WifiP2pDevice;
  
      invoke-direct {p0}, Landroid/net/wifi/p2p/WifiP2pService;->getDefaultDeviceName()Ljava/lang/String;
***************
*** 214,219 ****
--- 228,234 ----
  
      iput-object v2, v1, Landroid/net/wifi/p2p/WifiP2pDevice;->deviceName:Ljava/lang/String;
  
+     .line 180
      new-instance v1, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      const-string v2, "WifiP2pService"
***************
*** 224,254 ****
--- 239,276 ----
  
      iput-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pStateMachine:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
+     .line 181
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pStateMachine:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      invoke-virtual {v1}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->start()V
  
+     .line 184
      new-instance v0, Landroid/content/IntentFilter;
  
      invoke-direct {v0}, Landroid/content/IntentFilter;-><init>()V
  
+     .line 185
      .local v0, filter:Landroid/content/IntentFilter;
      const-string v1, "android.net.wifi.WIFI_STATE_CHANGED"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 186
      const-string v1, "android.intent.action.AIRPLANE_MODE"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 187
      const-string v1, "android.intent.action.EMERGENCY_CALLBACK_MODE_CHANGED"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 188
      const-string v1, "android.net.wifi.WIFI_AP_STATE_CHANGED"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 189
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
  
      new-instance v2, Landroid/net/wifi/p2p/WifiP2pService$WifiStateReceiver;
***************
*** 259,264 ****
--- 281,287 ----
  
      invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 191
      return-void
  .end method
  
***************
*** 267,272 ****
--- 290,296 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiState:I
  
      return v0
***************
*** 278,283 ****
--- 302,308 ----
      .parameter "x1"
  
      .prologue
+     .line 82
      iput p1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiState:I
  
      return p1
***************
*** 288,293 ****
--- 313,319 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiApState:I
  
      return v0
***************
*** 299,304 ****
--- 325,331 ----
      .parameter "x1"
  
      .prologue
+     .line 82
      iput p1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiApState:I
  
      return p1
***************
*** 309,314 ****
--- 336,342 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mInterface:Ljava/lang/String;
  
      return-object v0
***************
*** 319,324 ****
--- 347,353 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pStateMachine:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      return-object v0
***************
*** 329,334 ****
--- 358,364 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pRestartCount:I
  
      return v0
***************
*** 340,345 ****
--- 370,376 ----
      .parameter "x1"
  
      .prologue
+     .line 82
      iput p1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pRestartCount:I
  
      return p1
***************
*** 350,355 ****
--- 381,387 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pRestartCount:I
  
      add-int/lit8 v0, v0, 0x1
***************
*** 364,369 ****
--- 396,402 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mNetworkInfo:Landroid/net/NetworkInfo;
  
      return-object v0
***************
*** 374,379 ****
--- 407,413 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiChannel:Lcom/android/internal/util/AsyncChannel;
  
      return-object v0
***************
*** 385,390 ****
--- 419,425 ----
      .parameter "x1"
  
      .prologue
+     .line 82
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mWifiChannel:Lcom/android/internal/util/AsyncChannel;
  
      return-object p1
***************
*** 395,400 ****
--- 430,436 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mThisDevice:Landroid/net/wifi/p2p/WifiP2pDevice;
  
      return-object v0
***************
*** 405,410 ****
--- 441,447 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-boolean v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mPersistGroup:Z
  
      return v0
***************
*** 416,421 ****
--- 453,459 ----
      .parameter "x1"
  
      .prologue
+     .line 82
      iput-boolean p1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mPersistGroup:Z
  
      return p1
***************
*** 426,431 ****
--- 464,470 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mDhcpStateMachine:Landroid/net/DhcpStateMachine;
  
      return-object v0
***************
*** 437,442 ****
--- 476,482 ----
      .parameter "x1"
  
      .prologue
+     .line 82
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mDhcpStateMachine:Landroid/net/DhcpStateMachine;
  
      return-object p1
***************
*** 447,452 ****
--- 487,493 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 456,461 ****
--- 497,503 ----
      .locals 1
  
      .prologue
+     .line 82
      sget v0, Landroid/net/wifi/p2p/WifiP2pService;->mGroupNegotiationTimeoutIndex:I
  
      return v0
***************
*** 465,470 ****
--- 507,513 ----
      .locals 1
  
      .prologue
+     .line 82
      sget v0, Landroid/net/wifi/p2p/WifiP2pService;->mGroupNegotiationTimeoutIndex:I
  
      add-int/lit8 v0, v0, 0x1
***************
*** 478,483 ****
--- 521,527 ----
      .locals 1
  
      .prologue
+     .line 82
      sget-object v0, Landroid/net/wifi/p2p/WifiP2pService;->DHCP_RANGE:[Ljava/lang/String;
  
      return-object v0
***************
*** 488,493 ****
--- 532,538 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mReplyChannel:Lcom/android/internal/util/AsyncChannel;
  
      return-object v0
***************
*** 498,503 ****
--- 543,549 ----
      .parameter "x0"
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mNotification:Landroid/app/Notification;
  
      return-object v0
***************
*** 509,514 ****
--- 555,561 ----
      .parameter "x1"
  
      .prologue
+     .line 82
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mNotification:Landroid/app/Notification;
  
      return-object p1
***************
*** 518,523 ****
--- 565,571 ----
      .locals 3
  
      .prologue
+     .line 219
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
  
      const-string v1, "android.permission.ACCESS_WIFI_STATE"
***************
*** 526,531 ****
--- 574,580 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 221
      return-void
  .end method
  
***************
*** 533,538 ****
--- 582,588 ----
      .locals 3
  
      .prologue
+     .line 224
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
  
      const-string v1, "android.permission.CHANGE_WIFI_STATE"
***************
*** 541,546 ****
--- 591,597 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 226
      return-void
  .end method
  
***************
*** 548,553 ****
--- 599,605 ----
      .locals 4
  
      .prologue
+     .line 231
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
***************
*** 560,565 ****
--- 612,618 ----
  
      move-result-object v0
  
+     .line 233
      .local v0, id:Ljava/lang/String;
      new-instance v1, Ljava/lang/StringBuilder;
  
***************
*** 596,607 ****
      .locals 2
  
      .prologue
!     const-string v1, "network_management"
  
      invoke-static {v1}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
      move-result-object v0
  
      .local v0, b:Landroid/os/IBinder;
      invoke-static {v0}, Landroid/os/INetworkManagementService$Stub;->asInterface(Landroid/os/IBinder;)Landroid/os/INetworkManagementService;
  
--- 649,662 ----
      .locals 2
  
      .prologue
!     .line 194
!     const-string/jumbo v1, "network_management"
  
      invoke-static {v1}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 195
      .local v0, b:Landroid/os/IBinder;
      invoke-static {v0}, Landroid/os/INetworkManagementService$Stub;->asInterface(Landroid/os/IBinder;)Landroid/os/INetworkManagementService;
  
***************
*** 609,614 ****
--- 664,670 ----
  
      iput-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mNwService:Landroid/os/INetworkManagementService;
  
+     .line 196
      return-void
  .end method
  
***************
*** 619,624 ****
--- 675,681 ----
      .parameter "args"
  
      .prologue
+     .line 248
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService;->mContext:Landroid/content/Context;
  
      const-string v1, "android.permission.DUMP"
***************
*** 629,634 ****
--- 686,692 ----
  
      if-eqz v0, :cond_0
  
+     .line 250
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 667,672 ****
--- 725,731 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 255
      :cond_0
      return-void
  .end method
***************
*** 675,684 ****
--- 734,746 ----
      .locals 2
  
      .prologue
+     .line 241
      invoke-direct {p0}, Landroid/net/wifi/p2p/WifiP2pService;->enforceAccessPermission()V
  
+     .line 242
      invoke-direct {p0}, Landroid/net/wifi/p2p/WifiP2pService;->enforceChangePermission()V
  
+     .line 243
      new-instance v0, Landroid/os/Messenger;
  
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService;->mP2pStateMachine:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
