*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState.smali	2012-03-23 19:36:56.006702264 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState.smali.original	2012-03-23 19:34:45.421321043 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 913
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 39,44 ****
--- 40,46 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 917
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v0, v0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->this$0:Landroid/net/wifi/p2p/WifiP2pService;
***************
*** 52,57 ****
--- 54,60 ----
  
      invoke-virtual {v0, v1, v2, v2}, Landroid/net/NetworkInfo;->setDetailedState(Landroid/net/NetworkInfo$DetailedState;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 919
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      const/4 v1, 0x0
***************
*** 59,64 ****
--- 62,68 ----
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->updateThisDevice(I)V
      invoke-static {v0, v1}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$7400(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;I)V
  
+     .line 922
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #getter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
***************
*** 72,77 ****
--- 76,82 ----
  
      if-eqz v0, :cond_0
  
+     .line 923
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      const-string v1, "192.168.49.1"
***************
*** 79,89 ****
--- 84,96 ----
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->setWifiP2pInfoOnGroupFormation(Ljava/lang/String;)V
      invoke-static {v0, v1}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$7500(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Ljava/lang/String;)V
  
+     .line 924
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->sendP2pConnectionChangedBroadcast()V
      invoke-static {v0}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$7600(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;)V
  
+     .line 926
      :cond_0
      return-void
  .end method
***************
*** 94,99 ****
--- 101,107 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1066
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      const/4 v1, 0x3
***************
*** 101,111 ****
--- 109,121 ----
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->updateThisDevice(I)V
      invoke-static {v0, v1}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$7400(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;I)V
  
+     .line 1067
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->setWifiP2pInfoOnGroupTermination()V
      invoke-static {v0}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$8300(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;)V
  
+     .line 1068
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v0, v0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->this$0:Landroid/net/wifi/p2p/WifiP2pService;
***************
*** 119,129 ****
--- 129,141 ----
  
      invoke-virtual {v0, v1, v2, v2}, Landroid/net/NetworkInfo;->setDetailedState(Landroid/net/NetworkInfo$DetailedState;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1069
      iget-object v0, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->sendP2pConnectionChangedBroadcast()V
      invoke-static {v0}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$7600(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;)V
  
+     .line 1070
      return-void
  .end method
  
***************
*** 140,158 ****
--- 152,174 ----
  
      const/4 v9, 0x0
  
+     .line 931
      iget v11, p1, Landroid/os/Message;->what:I
  
      sparse-switch v11, :sswitch_data_0
  
+     .line 1062
      :cond_0
      :goto_0
      return v9
  
+     .line 934
      :sswitch_0
      iget-object v8, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v8, Ljava/lang/String;
  
+     .line 935
      .local v8, interfaceAddress:Ljava/lang/String;
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 161,169 ****
--- 177,187 ----
  
      move-result-object v4
  
+     .line 936
      .local v4, deviceAddress:Ljava/lang/String;
      if-eqz v4, :cond_1
  
+     .line 937
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #getter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
***************
*** 173,183 ****
--- 191,203 ----
  
      invoke-virtual {v11, v4}, Landroid/net/wifi/p2p/WifiP2pGroup;->addClient(Ljava/lang/String;)V
  
+     .line 938
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->updateDeviceStatus(Ljava/lang/String;I)V
      invoke-static {v11, v4, v9}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$5100(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Ljava/lang/String;I)V
  
+     .line 940
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->sendP2pPeersChangedBroadcast()V
***************
*** 188,195 ****
--- 208,217 ----
      :goto_1
      move v9, v10
  
+     .line 1062
      goto :goto_0
  
+     .line 942
      .restart local v4       #deviceAddress:Ljava/lang/String;
      .restart local v8       #interfaceAddress:Ljava/lang/String;
      :cond_1
***************
*** 218,223 ****
--- 240,246 ----
  
      goto :goto_1
  
+     .line 946
      .end local v4           #deviceAddress:Ljava/lang/String;
      .end local v8           #interfaceAddress:Ljava/lang/String;
      :sswitch_1
***************
*** 225,230 ****
--- 248,254 ----
  
      check-cast v8, Ljava/lang/String;
  
+     .line 947
      .restart local v8       #interfaceAddress:Ljava/lang/String;
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 233,246 ****
--- 257,273 ----
  
      move-result-object v4
  
+     .line 948
      .restart local v4       #deviceAddress:Ljava/lang/String;
      if-eqz v4, :cond_4
  
+     .line 949
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->updateDeviceStatus(Ljava/lang/String;I)V
      invoke-static {v9, v4, v12}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$5100(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Ljava/lang/String;I)V
  
+     .line 950
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #getter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
***************
*** 254,259 ****
--- 281,287 ----
  
      if-eqz v9, :cond_3
  
+     .line 952
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v9, v9, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->this$0:Landroid/net/wifi/p2p/WifiP2pService;
***************
*** 278,289 ****
--- 306,319 ----
  
      if-eqz v9, :cond_2
  
+     .line 953
      const-string v9, "WifiP2pService"
  
      const-string v11, "Client list empty, remove non-persistent p2p group"
  
      invoke-static {v9, v11}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 954
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #getter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
***************
*** 297,302 ****
--- 327,333 ----
  
      invoke-static {v9}, Landroid/net/wifi/WifiNative;->p2pGroupRemove(Ljava/lang/String;)Z
  
+     .line 962
      :cond_2
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 305,310 ****
--- 336,342 ----
  
      goto :goto_1
  
+     .line 958
      :cond_3
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 337,342 ****
--- 369,375 ----
  
      goto :goto_2
  
+     .line 965
      .end local v7           #i$:Ljava/util/Iterator;
      :cond_4
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 364,369 ****
--- 397,403 ----
  
      goto/16 :goto_1
  
+     .line 969
      .end local v4           #deviceAddress:Ljava/lang/String;
      .end local v8           #interfaceAddress:Ljava/lang/String;
      :sswitch_2
***************
*** 371,376 ****
--- 405,411 ----
  
      check-cast v6, Landroid/net/DhcpInfoInternal;
  
+     .line 970
      .local v6, dhcpInfo:Landroid/net/DhcpInfoInternal;
      iget v9, p1, Landroid/os/Message;->arg1:I
  
***************
*** 378,383 ****
--- 413,419 ----
  
      if-eqz v6, :cond_5
  
+     .line 973
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v11, v6, Landroid/net/DhcpInfoInternal;->serverAddress:Ljava/lang/String;
***************
*** 385,390 ****
--- 421,427 ----
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->setWifiP2pInfoOnGroupFormation(Ljava/lang/String;)V
      invoke-static {v9, v11}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$7500(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Ljava/lang/String;)V
  
+     .line 974
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->sendP2pConnectionChangedBroadcast()V
***************
*** 392,397 ****
--- 429,435 ----
  
      goto/16 :goto_1
  
+     .line 976
      :cond_5
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 408,413 ****
--- 446,452 ----
  
      goto/16 :goto_1
  
+     .line 981
      .end local v6           #dhcpInfo:Landroid/net/DhcpInfoInternal;
      :sswitch_3
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 427,432 ****
--- 466,472 ----
  
      if-eqz v11, :cond_6
  
+     .line 982
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      const v11, 0x22015
***************
*** 436,441 ****
--- 476,482 ----
  
      goto/16 :goto_1
  
+     .line 984
      :cond_6
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 446,451 ****
--- 487,493 ----
  
      goto/16 :goto_1
  
+     .line 990
      :sswitch_4
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 458,466 ****
--- 500,510 ----
  
      move-result-object v5
  
+     .line 991
      .local v5, devices:Ljava/util/Collection;,"Ljava/util/Collection<Landroid/net/wifi/p2p/WifiP2pDevice;>;"
      const/4 v0, 0x0
  
+     .line 992
      .local v0, changed:Z
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 492,497 ****
--- 536,542 ----
  
      check-cast v2, Landroid/net/wifi/p2p/WifiP2pDevice;
  
+     .line 993
      .local v2, d:Landroid/net/wifi/p2p/WifiP2pDevice;
      invoke-interface {v5, v2}, Ljava/util/Collection;->contains(Ljava/lang/Object;)Z
  
***************
*** 516,528 ****
--- 561,576 ----
  
      if-eqz v9, :cond_7
  
+     .line 994
      :cond_8
      iput v12, v2, Landroid/net/wifi/p2p/WifiP2pDevice;->status:I
  
+     .line 995
      const/4 v0, 0x1
  
      goto :goto_3
  
+     .line 999
      .end local v2           #d:Landroid/net/wifi/p2p/WifiP2pDevice;
      :cond_9
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 538,554 ****
--- 586,605 ----
  
      if-eqz v9, :cond_b
  
+     .line 1000
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->stopDhcpServer()V
      invoke-static {v9}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$7800(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;)V
  
+     .line 1008
      :goto_4
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #setter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
      invoke-static {v9, v13}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$1202(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Landroid/net/wifi/p2p/WifiP2pGroup;)Landroid/net/wifi/p2p/WifiP2pGroup;
  
+     .line 1009
      if-eqz v0, :cond_a
  
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 556,561 ****
--- 607,613 ----
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->sendP2pPeersChangedBroadcast()V
      invoke-static {v9}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$4200(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;)V
  
+     .line 1010
      :cond_a
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 571,576 ****
--- 623,629 ----
  
      goto/16 :goto_1
  
+     .line 1003
      :cond_b
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 585,590 ****
--- 638,644 ----
  
      invoke-virtual {v9, v11}, Landroid/net/DhcpStateMachine;->sendMessage(I)V
  
+     .line 1004
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v9, v9, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->this$0:Landroid/net/wifi/p2p/WifiP2pService;
***************
*** 596,601 ****
--- 650,656 ----
  
      invoke-virtual {v9}, Landroid/net/DhcpStateMachine;->quit()V
  
+     .line 1005
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v9, v9, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->this$0:Landroid/net/wifi/p2p/WifiP2pService;
***************
*** 605,610 ****
--- 660,666 ----
  
      goto :goto_4
  
+     .line 1013
      .end local v0           #changed:Z
      .end local v5           #devices:Ljava/util/Collection;,"Ljava/util/Collection<Landroid/net/wifi/p2p/WifiP2pDevice;>;"
      .end local v7           #i$:Ljava/util/Iterator;
***************
*** 613,618 ****
--- 669,675 ----
  
      check-cast v3, Landroid/net/wifi/p2p/WifiP2pDevice;
  
+     .line 1014
      .local v3, device:Landroid/net/wifi/p2p/WifiP2pDevice;
      iget-object v10, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 631,636 ****
--- 688,694 ----
  
      if-eqz v10, :cond_c
  
+     .line 1015
      iget-object v10, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      const-string v11, "Lost the group owner, killing p2p connection"
***************
*** 638,643 ****
--- 696,702 ----
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->logd(Ljava/lang/String;)V
      invoke-static {v10, v11}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$1500(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Ljava/lang/String;)V
  
+     .line 1016
      iget-object v10, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #getter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
***************
*** 653,658 ****
--- 712,718 ----
  
      goto/16 :goto_0
  
+     .line 1017
      :cond_c
      iget-object v10, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 667,672 ****
--- 727,733 ----
  
      if-eqz v10, :cond_0
  
+     .line 1018
      iget-object v10, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v10, v10, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->this$0:Landroid/net/wifi/p2p/WifiP2pService;
***************
*** 691,702 ****
--- 752,765 ----
  
      if-eqz v10, :cond_0
  
+     .line 1019
      const-string v10, "WifiP2pService"
  
      const-string v11, "Client list empty, removing a non-persistent p2p group"
  
      invoke-static {v10, v11}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1020
      iget-object v10, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #getter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
***************
*** 712,717 ****
--- 775,781 ----
  
      goto/16 :goto_0
  
+     .line 1025
      .end local v3           #device:Landroid/net/wifi/p2p/WifiP2pDevice;
      :sswitch_6
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 720,725 ****
--- 784,790 ----
  
      invoke-virtual {v9, v11}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->sendMessage(I)V
  
+     .line 1026
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->deferMessage(Landroid/os/Message;)V
***************
*** 727,737 ****
--- 792,804 ----
  
      goto/16 :goto_1
  
+     .line 1029
      :sswitch_7
      iget-object v1, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v1, Landroid/net/wifi/p2p/WifiP2pConfig;
  
+     .line 1030
      .local v1, config:Landroid/net/wifi/p2p/WifiP2pConfig;
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 758,763 ****
--- 825,831 ----
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->logd(Ljava/lang/String;)V
      invoke-static {v11, v12}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$1500(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Ljava/lang/String;)V
  
+     .line 1031
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #getter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
***************
*** 773,778 ****
--- 841,847 ----
  
      if-eqz v11, :cond_d
  
+     .line 1032
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v11, v1, Landroid/net/wifi/p2p/WifiP2pConfig;->deviceAddress:Ljava/lang/String;
***************
*** 780,790 ****
--- 849,861 ----
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->updateDeviceStatus(Ljava/lang/String;I)V
      invoke-static {v9, v11, v10}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$5100(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Ljava/lang/String;I)V
  
+     .line 1033
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #calls: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->sendP2pPeersChangedBroadcast()V
      invoke-static {v9}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$4200(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;)V
  
+     .line 1034
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      const v11, 0x2200c
***************
*** 794,799 ****
--- 865,871 ----
  
      goto/16 :goto_1
  
+     .line 1036
      :cond_d
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 804,809 ****
--- 876,882 ----
  
      goto/16 :goto_1
  
+     .line 1042
      .end local v1           #config:Landroid/net/wifi/p2p/WifiP2pConfig;
      :sswitch_8
      iget-object v9, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 833,838 ****
--- 906,912 ----
  
      goto/16 :goto_1
  
+     .line 1045
      :sswitch_9
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 845,850 ****
--- 919,925 ----
  
      goto/16 :goto_1
  
+     .line 1048
      :sswitch_a
      iget-object v11, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$GroupCreatedState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 857,862 ****
--- 932,938 ----
  
      goto/16 :goto_1
  
+     .line 1051
      :sswitch_b
      const-string v9, "WifiP2pService"
  
***************
*** 866,876 ****
--- 942,954 ----
  
      goto/16 :goto_1
  
+     .line 1054
      :sswitch_c
      invoke-static {}, Landroid/net/wifi/WifiNative;->wpsPbc()Z
  
      goto/16 :goto_1
  
+     .line 1057
      :sswitch_d
      iget-object v9, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 880,885 ****
--- 958,964 ----
  
      goto/16 :goto_1
  
+     .line 931
      nop
  
      :sswitch_data_0
