*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pManager$Channel$P2pHandler.smali	2012-03-23 19:36:55.479692656 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pManager$Channel$P2pHandler.smali.original	2012-03-23 19:34:45.414320916 -0500
***************
*** 25,34 ****
--- 25,37 ----
      .parameter "looper"
  
      .prologue
+     .line 363
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pManager$Channel$P2pHandler;->this$0:Landroid/net/wifi/p2p/WifiP2pManager$Channel;
  
+     .line 364
      invoke-direct {p0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
+     .line 365
      return-void
  .end method
  
***************
*** 39,44 ****
--- 42,48 ----
      .parameter "message"
  
      .prologue
+     .line 369
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pManager$Channel$P2pHandler;->this$0:Landroid/net/wifi/p2p/WifiP2pManager$Channel;
  
      iget v5, p1, Landroid/os/Message;->arg2:I
***************
*** 47,57 ****
--- 51,63 ----
  
      move-result-object v1
  
+     .line 370
      .local v1, listener:Ljava/lang/Object;
      iget v4, p1, Landroid/os/Message;->what:I
  
      sparse-switch v4, :sswitch_data_0
  
+     .line 416
      const-string v4, "WifiP2pManager"
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 74,84 ****
--- 80,92 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 419
      .end local v1           #listener:Ljava/lang/Object;
      :cond_0
      :goto_0
      return-void
  
+     .line 372
      .restart local v1       #listener:Ljava/lang/Object;
      :sswitch_0
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pManager$Channel$P2pHandler;->this$0:Landroid/net/wifi/p2p/WifiP2pManager$Channel;
***************
*** 90,95 ****
--- 98,104 ----
  
      if-eqz v4, :cond_0
  
+     .line 373
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pManager$Channel$P2pHandler;->this$0:Landroid/net/wifi/p2p/WifiP2pManager$Channel;
  
      #getter for: Landroid/net/wifi/p2p/WifiP2pManager$Channel;->mChannelListener:Landroid/net/wifi/p2p/WifiP2pManager$ChannelListener;
***************
*** 99,104 ****
--- 108,114 ----
  
      invoke-interface {v4}, Landroid/net/wifi/p2p/WifiP2pManager$ChannelListener;->onChannelDisconnected()V
  
+     .line 374
      iget-object v4, p0, Landroid/net/wifi/p2p/WifiP2pManager$Channel$P2pHandler;->this$0:Landroid/net/wifi/p2p/WifiP2pManager$Channel;
  
      const/4 v5, 0x0
***************
*** 108,116 ****
--- 118,128 ----
  
      goto :goto_0
  
+     .line 383
      :sswitch_1
      if-eqz v1, :cond_0
  
+     .line 384
      check-cast v1, Landroid/net/wifi/p2p/WifiP2pManager$ActionListener;
  
      .end local v1           #listener:Ljava/lang/Object;
***************
*** 120,129 ****
--- 132,143 ----
  
      goto :goto_0
  
+     .line 393
      .restart local v1       #listener:Ljava/lang/Object;
      :sswitch_2
      if-eqz v1, :cond_0
  
+     .line 394
      check-cast v1, Landroid/net/wifi/p2p/WifiP2pManager$ActionListener;
  
      .end local v1           #listener:Ljava/lang/Object;
***************
*** 131,145 ****
--- 145,162 ----
  
      goto :goto_0
  
+     .line 398
      .restart local v1       #listener:Ljava/lang/Object;
      :sswitch_3
      iget-object v2, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v2, Landroid/net/wifi/p2p/WifiP2pDeviceList;
  
+     .line 399
      .local v2, peers:Landroid/net/wifi/p2p/WifiP2pDeviceList;
      if-eqz v1, :cond_0
  
+     .line 400
      check-cast v1, Landroid/net/wifi/p2p/WifiP2pManager$PeerListListener;
  
      .end local v1           #listener:Ljava/lang/Object;
***************
*** 147,152 ****
--- 164,170 ----
  
      goto :goto_0
  
+     .line 404
      .end local v2           #peers:Landroid/net/wifi/p2p/WifiP2pDeviceList;
      .restart local v1       #listener:Ljava/lang/Object;
      :sswitch_4
***************
*** 154,162 ****
--- 172,182 ----
  
      check-cast v3, Landroid/net/wifi/p2p/WifiP2pInfo;
  
+     .line 405
      .local v3, wifiP2pInfo:Landroid/net/wifi/p2p/WifiP2pInfo;
      if-eqz v1, :cond_0
  
+     .line 406
      check-cast v1, Landroid/net/wifi/p2p/WifiP2pManager$ConnectionInfoListener;
  
      .end local v1           #listener:Ljava/lang/Object;
***************
*** 164,169 ****
--- 184,190 ----
  
      goto :goto_0
  
+     .line 410
      .end local v3           #wifiP2pInfo:Landroid/net/wifi/p2p/WifiP2pInfo;
      .restart local v1       #listener:Ljava/lang/Object;
      :sswitch_5
***************
*** 171,179 ****
--- 192,202 ----
  
      check-cast v0, Landroid/net/wifi/p2p/WifiP2pGroup;
  
+     .line 411
      .local v0, group:Landroid/net/wifi/p2p/WifiP2pGroup;
      if-eqz v1, :cond_0
  
+     .line 412
      check-cast v1, Landroid/net/wifi/p2p/WifiP2pManager$GroupInfoListener;
  
      .end local v1           #listener:Ljava/lang/Object;
***************
*** 181,186 ****
--- 204,210 ----
  
      goto :goto_0
  
+     .line 370
      :sswitch_data_0
      .sparse-switch
          0x11004 -> :sswitch_0
