*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pDevice$1.smali	2012-03-23 19:36:56.356708647 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pDevice$1.smali.original	2012-03-23 19:34:45.426321135 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 287
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,53 ****
--- 45,56 ----
      .parameter "in"
  
      .prologue
+     .line 289
      new-instance v0, Landroid/net/wifi/p2p/WifiP2pDevice;
  
      invoke-direct {v0}, Landroid/net/wifi/p2p/WifiP2pDevice;-><init>()V
  
+     .line 290
      .local v0, device:Landroid/net/wifi/p2p/WifiP2pDevice;
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
***************
*** 55,108 ****
--- 58,120 ----
  
      iput-object v1, v0, Landroid/net/wifi/p2p/WifiP2pDevice;->deviceName:Ljava/lang/String;
  
+     .line 291
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
      iput-object v1, v0, Landroid/net/wifi/p2p/WifiP2pDevice;->deviceAddress:Ljava/lang/String;
  
+     .line 292
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
      iput-object v1, v0, Landroid/net/wifi/p2p/WifiP2pDevice;->interfaceAddress:Ljava/lang/String;
  
+     .line 293
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
      iput-object v1, v0, Landroid/net/wifi/p2p/WifiP2pDevice;->primaryDeviceType:Ljava/lang/String;
  
+     .line 294
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
  
      iput-object v1, v0, Landroid/net/wifi/p2p/WifiP2pDevice;->secondaryDeviceType:Ljava/lang/String;
  
+     .line 295
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
      iput v1, v0, Landroid/net/wifi/p2p/WifiP2pDevice;->wpsConfigMethodsSupported:I
  
+     .line 296
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
      iput v1, v0, Landroid/net/wifi/p2p/WifiP2pDevice;->deviceCapability:I
  
+     .line 297
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
      iput v1, v0, Landroid/net/wifi/p2p/WifiP2pDevice;->groupCapability:I
  
+     .line 298
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
      iput v1, v0, Landroid/net/wifi/p2p/WifiP2pDevice;->status:I
  
+     .line 299
      return-object v0
  .end method
  
***************
*** 111,116 ****
--- 123,129 ----
      .parameter "x0"
  
      .prologue
+     .line 287
      invoke-virtual {p0, p1}, Landroid/net/wifi/p2p/WifiP2pDevice$1;->createFromParcel(Landroid/os/Parcel;)Landroid/net/wifi/p2p/WifiP2pDevice;
  
      move-result-object v0
***************
*** 123,128 ****
--- 136,142 ----
      .parameter "size"
  
      .prologue
+     .line 303
      new-array v0, p1, [Landroid/net/wifi/p2p/WifiP2pDevice;
  
      return-object v0
***************
*** 133,138 ****
--- 147,153 ----
      .parameter "x0"
  
      .prologue
+     .line 287
      invoke-virtual {p0, p1}, Landroid/net/wifi/p2p/WifiP2pDevice$1;->newArray(I)[Landroid/net/wifi/p2p/WifiP2pDevice;
  
      move-result-object v0
