*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService$P2pStateMachine$UserAuthorizingGroupInvitationState.smali	2012-03-23 19:36:55.455692217 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/p2p/WifiP2pService$P2pStateMachine$UserAuthorizingGroupInvitationState.smali.original	2012-03-23 19:34:45.413320897 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 824
      iput-object p1, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$UserAuthorizingGroupInvitationState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .locals 0
  
      .prologue
+     .line 828
      return-void
  .end method
  
***************
*** 47,66 ****
--- 49,73 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 833
      iget v1, p1, Landroid/os/Message;->what:I
  
      sparse-switch v1, :sswitch_data_0
  
+     .line 851
      const/4 v1, 0x0
  
+     .line 853
      :goto_0
      return v1
  
+     .line 840
      :sswitch_0
      new-instance v0, Landroid/net/wifi/p2p/WifiP2pConfig;
  
      invoke-direct {v0}, Landroid/net/wifi/p2p/WifiP2pConfig;-><init>()V
  
+     .line 841
      .local v0, config:Landroid/net/wifi/p2p/WifiP2pConfig;
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$UserAuthorizingGroupInvitationState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
***************
*** 77,93 ****
--- 84,103 ----
  
      iput-object v1, v0, Landroid/net/wifi/p2p/WifiP2pConfig;->deviceAddress:Ljava/lang/String;
  
+     .line 842
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$UserAuthorizingGroupInvitationState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      const v2, 0x2200a
  
      invoke-virtual {v1, v2, v0}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->sendMessage(ILjava/lang/Object;)V
  
+     .line 843
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$UserAuthorizingGroupInvitationState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #setter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mSavedP2pGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
      invoke-static {v1, v3}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$6902(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Landroid/net/wifi/p2p/WifiP2pGroup;)Landroid/net/wifi/p2p/WifiP2pGroup;
  
+     .line 853
      .end local v0           #config:Landroid/net/wifi/p2p/WifiP2pConfig;
      :goto_1
      :sswitch_1
***************
*** 95,106 ****
--- 105,118 ----
  
      goto :goto_0
  
+     .line 847
      :sswitch_2
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$UserAuthorizingGroupInvitationState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      #setter for: Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->mSavedP2pGroup:Landroid/net/wifi/p2p/WifiP2pGroup;
      invoke-static {v1, v3}, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;->access$6902(Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;Landroid/net/wifi/p2p/WifiP2pGroup;)Landroid/net/wifi/p2p/WifiP2pGroup;
  
+     .line 848
      iget-object v1, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$UserAuthorizingGroupInvitationState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
  
      iget-object v2, p0, Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine$UserAuthorizingGroupInvitationState;->this$1:Landroid/net/wifi/p2p/WifiP2pService$P2pStateMachine;
***************
*** 115,120 ****
--- 127,133 ----
  
      goto :goto_1
  
+     .line 833
      nop
  
      :sswitch_data_0
