*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$attr.smali	2012-03-23 19:37:30.952339497 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$attr.smali.original	2012-03-23 19:34:46.045332423 -0500
***************
*** 2165,2170 ****
--- 2165,2171 ----
      .locals 0
  
      .prologue
+     .line 552
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
