*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$drawable.smali	2012-03-23 19:37:34.440403101 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$drawable.smali.original	2012-03-23 19:34:46.112333645 -0500
***************
*** 3052,3057 ****
--- 3052,3058 ----
      .locals 0
  
      .prologue
+     .line 11900
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
