*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$id.smali	2012-03-23 19:37:53.049742443 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$id.smali.original	2012-03-23 19:34:46.391338733 -0500
***************
*** 1693,1698 ****
--- 1693,1699 ----
      .locals 0
  
      .prologue
+     .line 13455
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
