*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$interpolator.smali	2012-03-23 19:37:29.306309482 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$interpolator.smali.original	2012-03-23 19:34:46.007331729 -0500
***************
*** 47,52 ****
--- 47,53 ----
      .locals 0
  
      .prologue
+     .line 14564
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
