*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$layout.smali	2012-03-23 19:37:52.301728802 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$layout.smali.original	2012-03-23 19:34:46.383338586 -0500
***************
*** 393,398 ****
--- 393,399 ----
      .locals 0
  
      .prologue
+     .line 14608
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
