*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$plurals.smali	2012-03-23 19:37:52.276728347 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/R$plurals.smali.original	2012-03-23 19:34:46.383338586 -0500
***************
*** 61,66 ****
--- 61,67 ----
      .locals 0
  
      .prologue
+     .line 14822
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
