*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/appwidget/AppWidgetHostView$ParcelableSparseArray.smali	2012-03-23 19:37:03.009829966 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/appwidget/AppWidgetHostView$ParcelableSparseArray.smali.original	2012-03-23 19:34:45.518322813 -0500
***************
*** 45,50 ****
--- 45,51 ----
      .locals 1
  
      .prologue
+     .line 491
      new-instance v0, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray$1;
  
      invoke-direct {v0}, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray$1;-><init>()V
***************
*** 58,63 ****
--- 59,65 ----
      .locals 0
  
      .prologue
+     .line 477
      invoke-direct {p0}, Landroid/util/SparseArray;-><init>()V
  
      return-void
***************
*** 68,73 ****
--- 70,76 ----
      .parameter "x0"
  
      .prologue
+     .line 477
      invoke-direct {p0}, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;-><init>()V
  
      return-void
***************
*** 79,84 ****
--- 82,88 ----
      .locals 1
  
      .prologue
+     .line 479
      const/4 v0, 0x0
  
      return v0
***************
*** 90,114 ****
--- 94,123 ----
      .parameter "flags"
  
      .prologue
+     .line 483
      invoke-virtual {p0}, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;->size()I
  
      move-result v0
  
+     .line 484
      .local v0, count:I
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 485
      const/4 v1, 0x0
  
      .local v1, i:I
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 486
      invoke-virtual {p0, v1}, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;->keyAt(I)I
  
      move-result v2
  
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 487
      invoke-virtual {p0, v1}, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;->valueAt(I)Ljava/lang/Object;
  
      move-result-object v2
***************
*** 119,128 ****
--- 128,139 ----
  
      invoke-virtual {p1, v2, v3}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 485
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 489
      :cond_0
      return-void
  .end method
