*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/appwidget/AppWidgetHostView$ParcelableSparseArray$1.smali	2012-03-23 19:37:03.072831116 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/appwidget/AppWidgetHostView$ParcelableSparseArray$1.smali.original	2012-03-23 19:34:45.519322831 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 492
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,55 ****
--- 45,58 ----
      .parameter "source"
  
      .prologue
+     .line 494
      new-instance v0, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;
  
      const/4 v4, 0x0
  
      invoke-direct {v0, v4}, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;-><init>(Landroid/appwidget/AppWidgetHostView$1;)V
  
+     .line 495
      .local v0, array:Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;
      invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
  
***************
*** 59,69 ****
--- 62,74 ----
  
      move-result-object v3
  
+     .line 496
      .local v3, loader:Ljava/lang/ClassLoader;
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
+     .line 497
      .local v1, count:I
      const/4 v2, 0x0
  
***************
*** 71,76 ****
--- 76,82 ----
      :goto_0
      if-ge v2, v1, :cond_0
  
+     .line 498
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
***************
*** 81,90 ****
--- 87,98 ----
  
      invoke-virtual {v0, v4, v5}, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;->put(ILjava/lang/Object;)V
  
+     .line 497
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 500
      :cond_0
      return-object v0
  .end method
***************
*** 94,99 ****
--- 102,108 ----
      .parameter "x0"
  
      .prologue
+     .line 492
      invoke-virtual {p0, p1}, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray$1;->createFromParcel(Landroid/os/Parcel;)Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;
  
      move-result-object v0
***************
*** 106,111 ****
--- 115,121 ----
      .parameter "size"
  
      .prologue
+     .line 504
      new-array v0, p1, [Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;
  
      return-object v0
***************
*** 116,121 ****
--- 126,132 ----
      .parameter "x0"
  
      .prologue
+     .line 492
      invoke-virtual {p0, p1}, Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray$1;->newArray(I)[Landroid/appwidget/AppWidgetHostView$ParcelableSparseArray;
  
      move-result-object v0
