*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/appwidget/AppWidgetHostView$1.smali	2012-03-23 19:37:02.866827360 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/appwidget/AppWidgetHostView$1.smali.original	2012-03-23 19:34:45.517322795 -0500
***************
*** 22,27 ****
--- 22,28 ----
      .locals 0
  
      .prologue
+     .line 67
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 34,39 ****
--- 35,41 ----
      .parameter "clazz"
  
      .prologue
+     .line 69
      const-class v0, Landroid/widget/RemoteViews$RemoteView;
  
      invoke-virtual {p1, v0}, Ljava/lang/Class;->isAnnotationPresent(Ljava/lang/Class;)Z
