*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/appwidget/AppWidgetHost$UpdateHandler.smali	2012-03-23 19:37:02.916828270 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/appwidget/AppWidgetHost$UpdateHandler.smali.original	2012-03-23 19:34:45.518322813 -0500
***************
*** 25,34 ****
--- 25,37 ----
      .parameter "looper"
  
      .prologue
+     .line 77
      iput-object p1, p0, Landroid/appwidget/AppWidgetHost$UpdateHandler;->this$0:Landroid/appwidget/AppWidgetHost;
  
+     .line 78
      invoke-direct {p0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
+     .line 79
      return-void
  .end method
  
***************
*** 39,51 ****
--- 42,57 ----
      .parameter "msg"
  
      .prologue
+     .line 82
      iget v0, p1, Landroid/os/Message;->what:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 96
      :goto_0
      return-void
  
+     .line 84
      :pswitch_0
      iget-object v1, p0, Landroid/appwidget/AppWidgetHost$UpdateHandler;->this$0:Landroid/appwidget/AppWidgetHost;
  
***************
*** 59,64 ****
--- 65,71 ----
  
      goto :goto_0
  
+     .line 88
      :pswitch_1
      iget-object v1, p0, Landroid/appwidget/AppWidgetHost$UpdateHandler;->this$0:Landroid/appwidget/AppWidgetHost;
  
***************
*** 72,77 ****
--- 79,85 ----
  
      goto :goto_0
  
+     .line 92
      :pswitch_2
      iget-object v0, p0, Landroid/appwidget/AppWidgetHost$UpdateHandler;->this$0:Landroid/appwidget/AppWidgetHost;
  
***************
*** 83,88 ****
--- 91,97 ----
  
      goto :goto_0
  
+     .line 82
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
