*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/net/LegacyVpnInfo.smali	2012-03-23 19:37:52.500732433 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/net/LegacyVpnInfo.smali.original	2012-03-23 19:34:46.385338622 -0500
***************
*** 44,49 ****
--- 44,50 ----
      .locals 1
  
      .prologue
+     .line 53
      new-instance v0, Lcom/android/internal/net/LegacyVpnInfo$1;
  
      invoke-direct {v0}, Lcom/android/internal/net/LegacyVpnInfo$1;-><init>()V
***************
*** 57,64 ****
--- 58,67 ----
      .locals 1
  
      .prologue
+     .line 29
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 38
      const/4 v0, -0x1
  
      iput v0, p0, Lcom/android/internal/net/LegacyVpnInfo;->state:I
***************
*** 72,77 ****
--- 75,81 ----
      .locals 1
  
      .prologue
+     .line 43
      const/4 v0, 0x0
  
      return v0
***************
*** 83,99 ****
--- 87,107 ----
      .parameter "flags"
  
      .prologue
+     .line 48
      iget-object v0, p0, Lcom/android/internal/net/LegacyVpnInfo;->key:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 49
      iget v0, p0, Lcom/android/internal/net/LegacyVpnInfo;->state:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 50
      iget-object v0, p0, Lcom/android/internal/net/LegacyVpnInfo;->intent:Landroid/app/PendingIntent;
  
      invoke-virtual {p1, v0, p2}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 51
      return-void
  .end method
