*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/net/LegacyVpnInfo$1.smali	2012-03-23 19:37:52.322729186 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/net/LegacyVpnInfo$1.smali.original	2012-03-23 19:34:46.383338586 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 54
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,53 ****
--- 45,56 ----
      .parameter "in"
  
      .prologue
+     .line 57
      new-instance v0, Lcom/android/internal/net/LegacyVpnInfo;
  
      invoke-direct {v0}, Lcom/android/internal/net/LegacyVpnInfo;-><init>()V
  
+     .line 58
      .local v0, info:Lcom/android/internal/net/LegacyVpnInfo;
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
***************
*** 55,66 ****
--- 58,71 ----
  
      iput-object v1, v0, Lcom/android/internal/net/LegacyVpnInfo;->key:Ljava/lang/String;
  
+     .line 59
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
      iput v1, v0, Lcom/android/internal/net/LegacyVpnInfo;->state:I
  
+     .line 60
      const/4 v1, 0x0
  
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
***************
*** 71,76 ****
--- 76,82 ----
  
      iput-object v1, v0, Lcom/android/internal/net/LegacyVpnInfo;->intent:Landroid/app/PendingIntent;
  
+     .line 61
      return-object v0
  .end method
  
***************
*** 79,84 ****
--- 85,91 ----
      .parameter "x0"
  
      .prologue
+     .line 54
      invoke-virtual {p0, p1}, Lcom/android/internal/net/LegacyVpnInfo$1;->createFromParcel(Landroid/os/Parcel;)Lcom/android/internal/net/LegacyVpnInfo;
  
      move-result-object v0
***************
*** 91,96 ****
--- 98,104 ----
      .parameter "size"
  
      .prologue
+     .line 66
      new-array v0, p1, [Lcom/android/internal/net/LegacyVpnInfo;
  
      return-object v0
***************
*** 101,106 ****
--- 109,115 ----
      .parameter "x0"
  
      .prologue
+     .line 54
      invoke-virtual {p0, p1}, Lcom/android/internal/net/LegacyVpnInfo$1;->newArray(I)[Lcom/android/internal/net/LegacyVpnInfo;
  
      move-result-object v0
