*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/net/VpnConfig.smali	2012-03-23 19:37:52.418730937 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/net/VpnConfig.smali.original	2012-03-23 19:34:46.384338604 -0500
***************
*** 68,73 ****
--- 68,74 ----
      .locals 1
  
      .prologue
+     .line 85
      new-instance v0, Lcom/android/internal/net/VpnConfig$1;
  
      invoke-direct {v0}, Lcom/android/internal/net/VpnConfig$1;-><init>()V
***************
*** 81,92 ****
--- 82,96 ----
      .locals 2
  
      .prologue
+     .line 33
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 58
      const/4 v0, -0x1
  
      iput v0, p0, Lcom/android/internal/net/VpnConfig;->mtu:I
  
+     .line 64
      const-wide/16 v0, -0x1
  
      iput-wide v0, p0, Lcom/android/internal/net/VpnConfig;->startTime:J
***************
*** 98,107 ****
--- 102,113 ----
      .locals 3
  
      .prologue
+     .line 40
      new-instance v0, Landroid/content/Intent;
  
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
+     .line 41
      .local v0, intent:Landroid/content/Intent;
      const-string v1, "com.android.vpndialogs"
  
***************
*** 109,114 ****
--- 115,121 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 42
      return-object v0
  .end method
  
***************
*** 118,127 ****
--- 125,136 ----
      .parameter "config"
  
      .prologue
+     .line 46
      new-instance v0, Landroid/content/Intent;
  
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
+     .line 47
      .local v0, intent:Landroid/content/Intent;
      const-string v1, "com.android.vpndialogs"
  
***************
*** 129,142 ****
--- 138,154 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 48
      const-string v1, "config"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 49
      const/high16 v1, 0x5080
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 51
      const/4 v2, 0x0
  
      if-nez p1, :cond_0
***************
*** 162,167 ****
--- 174,180 ----
      .locals 1
  
      .prologue
+     .line 68
      const/4 v0, 0x0
  
      return v0
***************
*** 173,217 ****
--- 186,241 ----
      .parameter "flags"
  
      .prologue
+     .line 73
      iget-object v0, p0, Lcom/android/internal/net/VpnConfig;->user:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 74
      iget-object v0, p0, Lcom/android/internal/net/VpnConfig;->interfaze:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 75
      iget-object v0, p0, Lcom/android/internal/net/VpnConfig;->session:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 76
      iget v0, p0, Lcom/android/internal/net/VpnConfig;->mtu:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 77
      iget-object v0, p0, Lcom/android/internal/net/VpnConfig;->addresses:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 78
      iget-object v0, p0, Lcom/android/internal/net/VpnConfig;->routes:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 79
      iget-object v0, p0, Lcom/android/internal/net/VpnConfig;->dnsServers:Ljava/util/List;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStringList(Ljava/util/List;)V
  
+     .line 80
      iget-object v0, p0, Lcom/android/internal/net/VpnConfig;->searchDomains:Ljava/util/List;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStringList(Ljava/util/List;)V
  
+     .line 81
      iget-object v0, p0, Lcom/android/internal/net/VpnConfig;->configureIntent:Landroid/app/PendingIntent;
  
      invoke-virtual {p1, v0, p2}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 82
      iget-wide v0, p0, Lcom/android/internal/net/VpnConfig;->startTime:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 83
      return-void
  .end method
