*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/javax/microedition/khronos/egl/EGLContext.smali	2012-03-23 19:37:58.714845746 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/javax/microedition/khronos/egl/EGLContext.smali.original	2012-03-23 19:34:46.444339698 -0500
***************
*** 12,17 ****
--- 12,18 ----
      .locals 1
  
      .prologue
+     .line 23
      new-instance v0, Lcom/google/android/gles_jni/EGLImpl;
  
      invoke-direct {v0}, Lcom/google/android/gles_jni/EGLImpl;-><init>()V
***************
*** 25,30 ****
--- 26,32 ----
      .locals 0
  
      .prologue
+     .line 21
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 34,39 ****
--- 36,42 ----
      .locals 1
  
      .prologue
+     .line 26
      sget-object v0, Ljavax/microedition/khronos/egl/EGLContext;->EGL_INSTANCE:Ljavax/microedition/khronos/egl/EGL;
  
      return-object v0
