*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/DialogPreference.smali	2012-03-23 19:37:03.150832538 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/DialogPreference.smali.original	2012-03-23 19:34:45.520322849 -0500
***************
*** 43,52 ****
--- 43,54 ----
      .parameter "attrs"
  
      .prologue
+     .line 90
      const v0, 0x1010091
  
      invoke-direct {p0, p1, p2, v0}, Landroid/preference/DialogPreference;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 91
      return-void
  .end method
  
***************
*** 59,72 ****
--- 61,77 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 69
      invoke-direct {p0, p1, p2, p3}, Landroid/preference/Preference;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 71
      sget-object v1, Lcom/android/internal/R$styleable;->DialogPreference:[I
  
      invoke-virtual {p1, p2, v1, p3, v2}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[III)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 73
      .local v0, a:Landroid/content/res/TypedArray;
      invoke-virtual {v0, v2}, Landroid/content/res/TypedArray;->getString(I)Ljava/lang/String;
  
***************
*** 74,89 ****
--- 79,97 ----
  
      iput-object v1, p0, Landroid/preference/DialogPreference;->mDialogTitle:Ljava/lang/CharSequence;
  
+     .line 74
      iget-object v1, p0, Landroid/preference/DialogPreference;->mDialogTitle:Ljava/lang/CharSequence;
  
      if-nez v1, :cond_0
  
+     .line 77
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getTitle()Ljava/lang/CharSequence;
  
      move-result-object v1
  
      iput-object v1, p0, Landroid/preference/DialogPreference;->mDialogTitle:Ljava/lang/CharSequence;
  
+     .line 79
      :cond_0
      const/4 v1, 0x1
  
***************
*** 93,98 ****
--- 101,107 ----
  
      iput-object v1, p0, Landroid/preference/DialogPreference;->mDialogMessage:Ljava/lang/CharSequence;
  
+     .line 80
      const/4 v1, 0x2
  
      invoke-virtual {v0, v1}, Landroid/content/res/TypedArray;->getDrawable(I)Landroid/graphics/drawable/Drawable;
***************
*** 101,106 ****
--- 110,116 ----
  
      iput-object v1, p0, Landroid/preference/DialogPreference;->mDialogIcon:Landroid/graphics/drawable/Drawable;
  
+     .line 81
      const/4 v1, 0x3
  
      invoke-virtual {v0, v1}, Landroid/content/res/TypedArray;->getString(I)Ljava/lang/String;
***************
*** 109,114 ****
--- 119,125 ----
  
      iput-object v1, p0, Landroid/preference/DialogPreference;->mPositiveButtonText:Ljava/lang/CharSequence;
  
+     .line 82
      const/4 v1, 0x4
  
      invoke-virtual {v0, v1}, Landroid/content/res/TypedArray;->getString(I)Ljava/lang/String;
***************
*** 117,122 ****
--- 128,134 ----
  
      iput-object v1, p0, Landroid/preference/DialogPreference;->mNegativeButtonText:Ljava/lang/CharSequence;
  
+     .line 83
      const/4 v1, 0x5
  
      iget v2, p0, Landroid/preference/DialogPreference;->mDialogLayoutResId:I
***************
*** 127,134 ****
--- 139,148 ----
  
      iput v1, p0, Landroid/preference/DialogPreference;->mDialogLayoutResId:I
  
+     .line 85
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 87
      return-void
  .end method
  
***************
*** 137,151 ****
--- 151,168 ----
      .parameter "dialog"
  
      .prologue
+     .line 323
      invoke-virtual {p1}, Landroid/app/Dialog;->getWindow()Landroid/view/Window;
  
      move-result-object v0
  
+     .line 324
      .local v0, window:Landroid/view/Window;
      const/4 v1, 0x5
  
      invoke-virtual {v0, v1}, Landroid/view/Window;->setSoftInputMode(I)V
  
+     .line 325
      return-void
  .end method
  
***************
*** 155,160 ****
--- 172,178 ----
      .locals 1
  
      .prologue
+     .line 400
      iget-object v0, p0, Landroid/preference/DialogPreference;->mDialog:Landroid/app/Dialog;
  
      return-object v0
***************
*** 164,169 ****
--- 182,188 ----
      .locals 1
  
      .prologue
+     .line 172
      iget-object v0, p0, Landroid/preference/DialogPreference;->mDialogIcon:Landroid/graphics/drawable/Drawable;
  
      return-object v0
***************
*** 173,178 ****
--- 192,198 ----
      .locals 1
  
      .prologue
+     .line 249
      iget v0, p0, Landroid/preference/DialogPreference;->mDialogLayoutResId:I
  
      return v0
***************
*** 182,187 ****
--- 202,208 ----
      .locals 1
  
      .prologue
+     .line 145
      iget-object v0, p0, Landroid/preference/DialogPreference;->mDialogMessage:Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 191,196 ****
--- 212,218 ----
      .locals 1
  
      .prologue
+     .line 115
      iget-object v0, p0, Landroid/preference/DialogPreference;->mDialogTitle:Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 200,205 ****
--- 222,228 ----
      .locals 1
  
      .prologue
+     .line 228
      iget-object v0, p0, Landroid/preference/DialogPreference;->mNegativeButtonText:Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 209,214 ****
--- 232,238 ----
      .locals 1
  
      .prologue
+     .line 200
      iget-object v0, p0, Landroid/preference/DialogPreference;->mPositiveButtonText:Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 218,223 ****
--- 242,248 ----
      .locals 1
  
      .prologue
+     .line 316
      const/4 v0, 0x0
  
      return v0
***************
*** 227,232 ****
--- 252,258 ----
      .locals 1
  
      .prologue
+     .line 408
      iget-object v0, p0, Landroid/preference/DialogPreference;->mDialog:Landroid/app/Dialog;
  
      if-eqz v0, :cond_0
***************
*** 239,248 ****
--- 265,276 ----
  
      if-nez v0, :cond_1
  
+     .line 413
      :cond_0
      :goto_0
      return-void
  
+     .line 412
      :cond_1
      iget-object v0, p0, Landroid/preference/DialogPreference;->mDialog:Landroid/app/Dialog;
  
***************
*** 256,277 ****
--- 284,310 ----
      .parameter "view"
  
      .prologue
+     .line 352
      const v3, 0x102000b
  
      invoke-virtual {p1, v3}, Landroid/view/View;->findViewById(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 354
      .local v0, dialogMessageView:Landroid/view/View;
      if-eqz v0, :cond_2
  
+     .line 355
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getDialogMessage()Ljava/lang/CharSequence;
  
      move-result-object v1
  
+     .line 356
      .local v1, message:Ljava/lang/CharSequence;
      const/16 v2, 0x8
  
+     .line 358
      .local v2, newVisibility:I
      invoke-static {v1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
***************
*** 279,297 ****
--- 312,334 ----
  
      if-nez v3, :cond_1
  
+     .line 359
      instance-of v3, v0, Landroid/widget/TextView;
  
      if-eqz v3, :cond_0
  
      move-object v3, v0
  
+     .line 360
      check-cast v3, Landroid/widget/TextView;
  
      invoke-virtual {v3, v1}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 363
      :cond_0
      const/4 v2, 0x0
  
+     .line 366
      :cond_1
      invoke-virtual {v0}, Landroid/view/View;->getVisibility()I
  
***************
*** 299,306 ****
--- 336,345 ----
  
      if-eq v3, v2, :cond_2
  
+     .line 367
      invoke-virtual {v0, v2}, Landroid/view/View;->setVisibility(I)V
  
+     .line 370
      .end local v1           #message:Ljava/lang/CharSequence;
      .end local v2           #newVisibility:I
      :cond_2
***************
*** 311,320 ****
--- 350,361 ----
      .locals 1
  
      .prologue
+     .line 264
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Landroid/preference/DialogPreference;->showDialog(Landroid/os/Bundle;)V
  
+     .line 265
      return-void
  .end method
  
***************
*** 324,331 ****
--- 365,374 ----
      .parameter "which"
  
      .prologue
+     .line 373
      iput p2, p0, Landroid/preference/DialogPreference;->mWhichButtonClicked:I
  
+     .line 374
      return-void
  .end method
  
***************
*** 335,347 ****
--- 378,393 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 336
      iget v2, p0, Landroid/preference/DialogPreference;->mDialogLayoutResId:I
  
      if-nez v2, :cond_0
  
+     .line 341
      :goto_0
      return-object v1
  
+     .line 340
      :cond_0
      iget-object v2, p0, Landroid/preference/DialogPreference;->mBuilder:Landroid/app/AlertDialog$Builder;
  
***************
*** 353,358 ****
--- 399,405 ----
  
      move-result-object v0
  
+     .line 341
      .local v0, inflater:Landroid/view/LayoutInflater;
      iget v2, p0, Landroid/preference/DialogPreference;->mDialogLayoutResId:I
  
***************
*** 368,373 ****
--- 415,421 ----
      .parameter "positiveResult"
  
      .prologue
+     .line 392
      return-void
  .end method
  
***************
*** 376,391 ****
--- 424,442 ----
      .parameter "dialog"
  
      .prologue
+     .line 378
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getPreferenceManager()Landroid/preference/PreferenceManager;
  
      move-result-object v0
  
      invoke-virtual {v0, p0}, Landroid/preference/PreferenceManager;->unregisterOnActivityDestroyListener(Landroid/preference/PreferenceManager$OnActivityDestroyListener;)V
  
+     .line 380
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/preference/DialogPreference;->mDialog:Landroid/app/Dialog;
  
+     .line 381
      iget v0, p0, Landroid/preference/DialogPreference;->mWhichButtonClicked:I
  
      const/4 v1, -0x1
***************
*** 397,404 ****
--- 448,457 ----
      :goto_0
      invoke-virtual {p0, v0}, Landroid/preference/DialogPreference;->onDialogClosed(Z)V
  
+     .line 382
      return-void
  
+     .line 381
      :cond_0
      const/4 v0, 0x0
  
***************
*** 410,415 ****
--- 463,469 ----
      .parameter "builder"
  
      .prologue
+     .line 260
      return-void
  .end method
  
***************
*** 418,423 ****
--- 472,478 ----
      .parameter "state"
  
      .prologue
+     .line 430
      if-eqz p1, :cond_0
  
      invoke-virtual {p1}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
***************
*** 432,440 ****
--- 487,497 ----
  
      if-nez v1, :cond_2
  
+     .line 432
      :cond_0
      invoke-super {p0, p1}, Landroid/preference/Preference;->onRestoreInstanceState(Landroid/os/Parcelable;)V
  
+     .line 441
      :cond_1
      :goto_0
      return-void
***************
*** 442,449 ****
--- 499,508 ----
      :cond_2
      move-object v0, p1
  
+     .line 436
      check-cast v0, Landroid/preference/DialogPreference$SavedState;
  
+     .line 437
      .local v0, myState:Landroid/preference/DialogPreference$SavedState;
      invoke-virtual {v0}, Landroid/preference/DialogPreference$SavedState;->getSuperState()Landroid/os/Parcelable;
  
***************
*** 451,460 ****
--- 510,521 ----
  
      invoke-super {p0, v1}, Landroid/preference/Preference;->onRestoreInstanceState(Landroid/os/Parcelable;)V
  
+     .line 438
      iget-boolean v1, v0, Landroid/preference/DialogPreference$SavedState;->isDialogShowing:Z
  
      if-eqz v1, :cond_1
  
+     .line 439
      iget-object v1, v0, Landroid/preference/DialogPreference$SavedState;->dialogBundle:Landroid/os/Bundle;
  
      invoke-virtual {p0, v1}, Landroid/preference/DialogPreference;->showDialog(Landroid/os/Bundle;)V
***************
*** 466,475 ****
--- 527,538 ----
      .locals 3
  
      .prologue
+     .line 417
      invoke-super {p0}, Landroid/preference/Preference;->onSaveInstanceState()Landroid/os/Parcelable;
  
      move-result-object v1
  
+     .line 418
      .local v1, superState:Landroid/os/Parcelable;
      iget-object v2, p0, Landroid/preference/DialogPreference;->mDialog:Landroid/app/Dialog;
  
***************
*** 486,504 ****
--- 549,571 ----
      :cond_0
      move-object v0, v1
  
+     .line 425
      :goto_0
      return-object v0
  
+     .line 422
      :cond_1
      new-instance v0, Landroid/preference/DialogPreference$SavedState;
  
      invoke-direct {v0, v1}, Landroid/preference/DialogPreference$SavedState;-><init>(Landroid/os/Parcelable;)V
  
+     .line 423
      .local v0, myState:Landroid/preference/DialogPreference$SavedState;
      const/4 v2, 0x1
  
      iput-boolean v2, v0, Landroid/preference/DialogPreference$SavedState;->isDialogShowing:Z
  
+     .line 424
      iget-object v2, p0, Landroid/preference/DialogPreference;->mDialog:Landroid/app/Dialog;
  
      invoke-virtual {v2}, Landroid/app/Dialog;->onSaveInstanceState()Landroid/os/Bundle;
***************
*** 515,520 ****
--- 582,588 ----
      .parameter "dialogIconRes"
  
      .prologue
+     .line 164
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 529,534 ****
--- 597,603 ----
  
      iput-object v0, p0, Landroid/preference/DialogPreference;->mDialogIcon:Landroid/graphics/drawable/Drawable;
  
+     .line 165
      return-void
  .end method
  
***************
*** 537,544 ****
--- 606,615 ----
      .parameter "dialogIcon"
  
      .prologue
+     .line 154
      iput-object p1, p0, Landroid/preference/DialogPreference;->mDialogIcon:Landroid/graphics/drawable/Drawable;
  
+     .line 155
      return-void
  .end method
  
***************
*** 547,554 ****
--- 618,627 ----
      .parameter "dialogLayoutResId"
  
      .prologue
+     .line 239
      iput p1, p0, Landroid/preference/DialogPreference;->mDialogLayoutResId:I
  
+     .line 240
      return-void
  .end method
  
***************
*** 557,562 ****
--- 630,636 ----
      .parameter "dialogMessageResId"
  
      .prologue
+     .line 137
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 567,572 ****
--- 641,647 ----
  
      invoke-virtual {p0, v0}, Landroid/preference/DialogPreference;->setDialogMessage(Ljava/lang/CharSequence;)V
  
+     .line 138
      return-void
  .end method
  
***************
*** 575,582 ****
--- 650,659 ----
      .parameter "dialogMessage"
  
      .prologue
+     .line 129
      iput-object p1, p0, Landroid/preference/DialogPreference;->mDialogMessage:Ljava/lang/CharSequence;
  
+     .line 130
      return-void
  .end method
  
***************
*** 585,590 ****
--- 662,668 ----
      .parameter "dialogTitleResId"
  
      .prologue
+     .line 107
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 595,600 ****
--- 673,679 ----
  
      invoke-virtual {p0, v0}, Landroid/preference/DialogPreference;->setDialogTitle(Ljava/lang/CharSequence;)V
  
+     .line 108
      return-void
  .end method
  
***************
*** 603,610 ****
--- 682,691 ----
      .parameter "dialogTitle"
  
      .prologue
+     .line 99
      iput-object p1, p0, Landroid/preference/DialogPreference;->mDialogTitle:Ljava/lang/CharSequence;
  
+     .line 100
      return-void
  .end method
  
***************
*** 613,618 ****
--- 694,700 ----
      .parameter "negativeButtonTextResId"
  
      .prologue
+     .line 218
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 623,628 ****
--- 705,711 ----
  
      invoke-virtual {p0, v0}, Landroid/preference/DialogPreference;->setNegativeButtonText(Ljava/lang/CharSequence;)V
  
+     .line 219
      return-void
  .end method
  
***************
*** 631,638 ****
--- 714,723 ----
      .parameter "negativeButtonText"
  
      .prologue
+     .line 210
      iput-object p1, p0, Landroid/preference/DialogPreference;->mNegativeButtonText:Ljava/lang/CharSequence;
  
+     .line 211
      return-void
  .end method
  
***************
*** 641,646 ****
--- 726,732 ----
      .parameter "positiveButtonTextResId"
  
      .prologue
+     .line 190
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 651,656 ****
--- 737,743 ----
  
      invoke-virtual {p0, v0}, Landroid/preference/DialogPreference;->setPositiveButtonText(Ljava/lang/CharSequence;)V
  
+     .line 191
      return-void
  .end method
  
***************
*** 659,666 ****
--- 746,755 ----
      .parameter "positiveButtonText"
  
      .prologue
+     .line 182
      iput-object p1, p0, Landroid/preference/DialogPreference;->mPositiveButtonText:Ljava/lang/CharSequence;
  
+     .line 183
      return-void
  .end method
  
***************
*** 669,683 ****
--- 758,775 ----
      .parameter "state"
  
      .prologue
+     .line 275
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getContext()Landroid/content/Context;
  
      move-result-object v1
  
+     .line 277
      .local v1, context:Landroid/content/Context;
      const/4 v3, -0x2
  
      iput v3, p0, Landroid/preference/DialogPreference;->mWhichButtonClicked:I
  
+     .line 279
      new-instance v3, Landroid/app/AlertDialog$Builder;
  
      invoke-direct {v3, v1}, Landroid/app/AlertDialog$Builder;-><init>(Landroid/content/Context;)V
***************
*** 708,737 ****
--- 800,836 ----
  
      iput-object v3, p0, Landroid/preference/DialogPreference;->mBuilder:Landroid/app/AlertDialog$Builder;
  
+     .line 285
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->onCreateDialogView()Landroid/view/View;
  
      move-result-object v0
  
+     .line 286
      .local v0, contentView:Landroid/view/View;
      if-eqz v0, :cond_2
  
+     .line 287
      invoke-virtual {p0, v0}, Landroid/preference/DialogPreference;->onBindDialogView(Landroid/view/View;)V
  
+     .line 288
      iget-object v3, p0, Landroid/preference/DialogPreference;->mBuilder:Landroid/app/AlertDialog$Builder;
  
      invoke-virtual {v3, v0}, Landroid/app/AlertDialog$Builder;->setView(Landroid/view/View;)Landroid/app/AlertDialog$Builder;
  
+     .line 293
      :goto_0
      iget-object v3, p0, Landroid/preference/DialogPreference;->mBuilder:Landroid/app/AlertDialog$Builder;
  
      invoke-virtual {p0, v3}, Landroid/preference/DialogPreference;->onPrepareDialogBuilder(Landroid/app/AlertDialog$Builder;)V
  
+     .line 295
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->getPreferenceManager()Landroid/preference/PreferenceManager;
  
      move-result-object v3
  
      invoke-virtual {v3, p0}, Landroid/preference/PreferenceManager;->registerOnActivityDestroyListener(Landroid/preference/PreferenceManager$OnActivityDestroyListener;)V
  
+     .line 298
      iget-object v3, p0, Landroid/preference/DialogPreference;->mBuilder:Landroid/app/AlertDialog$Builder;
  
      invoke-virtual {v3}, Landroid/app/AlertDialog$Builder;->create()Landroid/app/AlertDialog;
***************
*** 740,750 ****
--- 839,852 ----
  
      iput-object v2, p0, Landroid/preference/DialogPreference;->mDialog:Landroid/app/Dialog;
  
+     .line 299
      .local v2, dialog:Landroid/app/Dialog;
      if-eqz p1, :cond_0
  
+     .line 300
      invoke-virtual {v2, p1}, Landroid/app/Dialog;->onRestoreInstanceState(Landroid/os/Bundle;)V
  
+     .line 302
      :cond_0
      invoke-virtual {p0}, Landroid/preference/DialogPreference;->needInputMethod()Z
  
***************
*** 752,766 ****
--- 854,873 ----
  
      if-eqz v3, :cond_1
  
+     .line 303
      invoke-direct {p0, v2}, Landroid/preference/DialogPreference;->requestInputMethod(Landroid/app/Dialog;)V
  
+     .line 305
      :cond_1
      invoke-virtual {v2, p0}, Landroid/app/Dialog;->setOnDismissListener(Landroid/content/DialogInterface$OnDismissListener;)V
  
+     .line 306
      invoke-virtual {v2}, Landroid/app/Dialog;->show()V
  
+     .line 307
      return-void
  
+     .line 290
      .end local v2           #dialog:Landroid/app/Dialog;
      :cond_2
      iget-object v3, p0, Landroid/preference/DialogPreference;->mBuilder:Landroid/app/AlertDialog$Builder;
