*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/DialogPreference$SavedState.smali	2012-03-23 19:37:04.511857356 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/DialogPreference$SavedState.smali.original	2012-03-23 19:34:45.532323067 -0500
***************
*** 38,43 ****
--- 38,44 ----
      .locals 1
  
      .prologue
+     .line 464
      new-instance v0, Landroid/preference/DialogPreference$SavedState$1;
  
      invoke-direct {v0}, Landroid/preference/DialogPreference$SavedState$1;-><init>()V
***************
*** 54,61 ****
--- 55,64 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 448
      invoke-direct {p0, p1}, Landroid/preference/Preference$BaseSavedState;-><init>(Landroid/os/Parcel;)V
  
+     .line 449
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
***************
*** 65,78 ****
--- 68,84 ----
      :goto_0
      iput-boolean v0, p0, Landroid/preference/DialogPreference$SavedState;->isDialogShowing:Z
  
+     .line 450
      invoke-virtual {p1}, Landroid/os/Parcel;->readBundle()Landroid/os/Bundle;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/preference/DialogPreference$SavedState;->dialogBundle:Landroid/os/Bundle;
  
+     .line 451
      return-void
  
+     .line 449
      :cond_0
      const/4 v0, 0x0
  
***************
*** 84,91 ****
--- 90,99 ----
      .parameter "superState"
  
      .prologue
+     .line 461
      invoke-direct {p0, p1}, Landroid/preference/Preference$BaseSavedState;-><init>(Landroid/os/Parcelable;)V
  
+     .line 462
      return-void
  .end method
  
***************
*** 97,104 ****
--- 105,114 ----
      .parameter "flags"
  
      .prologue
+     .line 455
      invoke-super {p0, p1, p2}, Landroid/preference/Preference$BaseSavedState;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 456
      iget-boolean v0, p0, Landroid/preference/DialogPreference$SavedState;->isDialogShowing:Z
  
      if-eqz v0, :cond_0
***************
*** 108,119 ****
--- 118,132 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 457
      iget-object v0, p0, Landroid/preference/DialogPreference$SavedState;->dialogBundle:Landroid/os/Bundle;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeBundle(Landroid/os/Bundle;)V
  
+     .line 458
      return-void
  
+     .line 456
      :cond_0
      const/4 v0, 0x0
  
