*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/GenericInflater.smali	2012-03-23 19:37:04.420855697 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/GenericInflater.smali.original	2012-03-23 19:34:45.531323049 -0500
***************
*** 56,61 ****
--- 56,62 ----
      .locals 3
  
      .prologue
+     .line 55
      const/4 v0, 0x2
  
      new-array v0, v0, [Ljava/lang/Class;
***************
*** 74,79 ****
--- 75,81 ----
  
      sput-object v0, Landroid/preference/GenericInflater;->mConstructorSignature:[Ljava/lang/Class;
  
+     .line 58
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
***************
*** 88,108 ****
--- 90,115 ----
      .parameter "context"
  
      .prologue
+     .line 108
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 45
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/preference/GenericInflater;->DEBUG:Z
  
+     .line 53
      const/4 v0, 0x2
  
      new-array v0, v0, [Ljava/lang/Object;
  
      iput-object v0, p0, Landroid/preference/GenericInflater;->mConstructorArgs:[Ljava/lang/Object;
  
+     .line 109
      iput-object p1, p0, Landroid/preference/GenericInflater;->mContext:Landroid/content/Context;
  
+     .line 110
      return-void
  .end method
  
***************
*** 121,146 ****
--- 128,159 ----
      .end annotation
  
      .prologue
+     .line 120
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      .local p1, original:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 45
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/preference/GenericInflater;->DEBUG:Z
  
+     .line 53
      const/4 v0, 0x2
  
      new-array v0, v0, [Ljava/lang/Object;
  
      iput-object v0, p0, Landroid/preference/GenericInflater;->mConstructorArgs:[Ljava/lang/Object;
  
+     .line 121
      iput-object p2, p0, Landroid/preference/GenericInflater;->mContext:Landroid/content/Context;
  
+     .line 122
      iget-object v0, p1, Landroid/preference/GenericInflater;->mFactory:Landroid/preference/GenericInflater$Factory;
  
      iput-object v0, p0, Landroid/preference/GenericInflater;->mFactory:Landroid/preference/GenericInflater$Factory;
  
+     .line 123
      return-void
  .end method
  
***************
*** 163,177 ****
--- 176,193 ----
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      const/4 v2, 0x0
  
+     .line 424
      :try_start_0
      iget-object v3, p0, Landroid/preference/GenericInflater;->mFactory:Landroid/preference/GenericInflater$Factory;
  
      if-nez v3, :cond_1
  
+     .line 426
      .local v2, item:Ljava/lang/Object;,"TT;"
      :goto_0
      if-nez v2, :cond_0
  
+     .line 427
      const/4 v3, -0x1
  
      const/16 v4, 0x2e
***************
*** 182,195 ****
--- 198,214 ----
  
      if-ne v3, v4, :cond_2
  
+     .line 428
      invoke-virtual {p0, p2, p3}, Landroid/preference/GenericInflater;->onCreateItem(Ljava/lang/String;Landroid/util/AttributeSet;)Ljava/lang/Object;
  
      move-result-object v2
  
+     .line 435
      :cond_0
      :goto_1
      return-object v2
  
+     .line 424
      .end local v2           #item:Ljava/lang/Object;,"TT;"
      :cond_1
      iget-object v3, p0, Landroid/preference/GenericInflater;->mFactory:Landroid/preference/GenericInflater$Factory;
***************
*** 202,207 ****
--- 221,227 ----
  
      goto :goto_0
  
+     .line 430
      .restart local v2       #item:Ljava/lang/Object;,"TT;"
      :cond_2
      const/4 v3, 0x0
***************
*** 216,232 ****
--- 236,256 ----
  
      goto :goto_1
  
+     .line 437
      .end local v2           #item:Ljava/lang/Object;,"TT;"
      :catch_0
      move-exception v0
  
+     .line 438
      .local v0, e:Landroid/view/InflateException;
      throw v0
  
+     .line 440
      .end local v0           #e:Landroid/view/InflateException;
      :catch_1
      move-exception v0
  
+     .line 441
      .local v0, e:Ljava/lang/ClassNotFoundException;
      new-instance v1, Landroid/view/InflateException;
  
***************
*** 258,273 ****
--- 282,301 ----
  
      invoke-direct {v1, v3}, Landroid/view/InflateException;-><init>(Ljava/lang/String;)V
  
+     .line 444
      .local v1, ie:Landroid/view/InflateException;
      invoke-virtual {v1, v0}, Landroid/view/InflateException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 445
      throw v1
  
+     .line 447
      .end local v0           #e:Ljava/lang/ClassNotFoundException;
      .end local v1           #ie:Landroid/view/InflateException;
      :catch_2
      move-exception v0
  
+     .line 448
      .local v0, e:Ljava/lang/Exception;
      new-instance v1, Landroid/view/InflateException;
  
***************
*** 299,307 ****
--- 327,337 ----
  
      invoke-direct {v1, v3}, Landroid/view/InflateException;-><init>(Ljava/lang/String;)V
  
+     .line 451
      .restart local v1       #ie:Landroid/view/InflateException;
      invoke-virtual {v1, v0}, Landroid/view/InflateException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 452
      throw v1
  .end method
  
***************
*** 328,339 ****
--- 358,371 ----
      .end annotation
  
      .prologue
+     .line 462
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      .local p2, parent:Ljava/lang/Object;,"TT;"
      invoke-interface {p1}, Lorg/xmlpull/v1/XmlPullParser;->getDepth()I
  
      move-result v0
  
+     .line 466
      .local v0, depth:I
      :cond_0
      :goto_0
***************
*** 357,376 ****
--- 389,412 ----
  
      if-eq v3, v4, :cond_2
  
+     .line 468
      const/4 v4, 0x2
  
      if-ne v3, v4, :cond_0
  
+     .line 472
      invoke-virtual {p0, p1, p2, p3}, Landroid/preference/GenericInflater;->onCreateCustomFromTag(Lorg/xmlpull/v1/XmlPullParser;Ljava/lang/Object;Landroid/util/AttributeSet;)Z
  
      move-result v4
  
      if-nez v4, :cond_0
  
+     .line 479
      invoke-interface {p1}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 481
      .local v2, name:Ljava/lang/String;
      invoke-direct {p0, p1, v2, p3}, Landroid/preference/GenericInflater;->createItemFromTag(Lorg/xmlpull/v1/XmlPullParser;Ljava/lang/String;Landroid/util/AttributeSet;)Ljava/lang/Object;
  
***************
*** 379,392 ****
--- 415,431 ----
      .local v1, item:Ljava/lang/Object;,"TT;"
      move-object v4, p2
  
+     .line 488
      check-cast v4, Landroid/preference/GenericInflater$Parent;
  
      invoke-interface {v4, v1}, Landroid/preference/GenericInflater$Parent;->addItemFromInflater(Ljava/lang/Object;)V
  
+     .line 493
      invoke-direct {p0, p1, v1, p3}, Landroid/preference/GenericInflater;->rInflate(Lorg/xmlpull/v1/XmlPullParser;Ljava/lang/Object;Landroid/util/AttributeSet;)V
  
      goto :goto_0
  
+     .line 499
      .end local v1           #item:Ljava/lang/Object;,"TT;"
      .end local v2           #name:Ljava/lang/String;
      :cond_2
***************
*** 421,426 ****
--- 460,466 ----
      .end annotation
  
      .prologue
+     .line 369
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      sget-object v5, Landroid/preference/GenericInflater;->sConstructorMap:Ljava/util/HashMap;
  
***************
*** 430,438 ****
--- 470,480 ----
  
      check-cast v2, Ljava/lang/reflect/Constructor;
  
+     .line 372
      .local v2, constructor:Ljava/lang/reflect/Constructor;
      if-nez v2, :cond_0
  
+     .line 375
      :try_start_0
      iget-object v5, p0, Landroid/preference/GenericInflater;->mContext:Landroid/content/Context;
  
***************
*** 463,468 ****
--- 505,511 ----
  
      move-result-object v1
  
+     .line 377
      .local v1, clazz:Ljava/lang/Class;
      sget-object v5, Landroid/preference/GenericInflater;->mConstructorSignature:[Ljava/lang/Class;
  
***************
*** 470,488 ****
--- 513,535 ----
  
      move-result-object v2
  
+     .line 378
      sget-object v5, Landroid/preference/GenericInflater;->sConstructorMap:Ljava/util/HashMap;
  
      invoke-virtual {v5, p1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 381
      .end local v1           #clazz:Ljava/lang/Class;
      :cond_0
      iget-object v0, p0, Landroid/preference/GenericInflater;->mConstructorArgs:[Ljava/lang/Object;
  
+     .line 382
      .local v0, args:[Ljava/lang/Object;
      const/4 v5, 0x1
  
      aput-object p3, v0, v5
  
+     .line 383
      invoke-virtual {v2, v0}, Ljava/lang/reflect/Constructor;->newInstance([Ljava/lang/Object;)Ljava/lang/Object;
      :try_end_0
      .catch Ljava/lang/NoSuchMethodException; {:try_start_0 .. :try_end_0} :catch_0
***************
*** 497,507 ****
--- 544,557 ----
      :cond_1
      move-object v5, p1
  
+     .line 375
      goto :goto_0
  
+     .line 385
      :catch_0
      move-exception v3
  
+     .line 386
      .local v3, e:Ljava/lang/NoSuchMethodException;
      new-instance v4, Landroid/view/InflateException;
  
***************
*** 553,576 ****
--- 603,632 ----
  
      invoke-direct {v4, v5}, Landroid/view/InflateException;-><init>(Ljava/lang/String;)V
  
+     .line 390
      .local v4, ie:Landroid/view/InflateException;
      invoke-virtual {v4, v3}, Landroid/view/InflateException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 391
      throw v4
  
+     .line 393
      .end local v3           #e:Ljava/lang/NoSuchMethodException;
      .end local v4           #ie:Landroid/view/InflateException;
      .restart local p1
      :catch_1
      move-exception v3
  
+     .line 395
      .local v3, e:Ljava/lang/ClassNotFoundException;
      throw v3
  
+     .line 396
      .end local v3           #e:Ljava/lang/ClassNotFoundException;
      :catch_2
      move-exception v3
  
+     .line 397
      .local v3, e:Ljava/lang/Exception;
      new-instance v4, Landroid/view/InflateException;
  
***************
*** 610,618 ****
--- 666,676 ----
  
      invoke-direct {v4, v5}, Landroid/view/InflateException;-><init>(Ljava/lang/String;)V
  
+     .line 401
      .restart local v4       #ie:Landroid/view/InflateException;
      invoke-virtual {v4, v3}, Landroid/view/InflateException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 402
      throw v4
  .end method
  
***************
*** 620,625 ****
--- 678,684 ----
      .locals 1
  
      .prologue
+     .line 165
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      iget-object v0, p0, Landroid/preference/GenericInflater;->mContext:Landroid/content/Context;
  
***************
*** 630,635 ****
--- 689,695 ----
      .locals 1
  
      .prologue
+     .line 157
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      iget-object v0, p0, Landroid/preference/GenericInflater;->mDefaultPackage:Ljava/lang/String;
  
***************
*** 647,652 ****
--- 707,713 ----
      .end annotation
  
      .prologue
+     .line 174
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      iget-object v0, p0, Landroid/preference/GenericInflater;->mFactory:Landroid/preference/GenericInflater$Factory;
  
***************
*** 664,669 ****
--- 725,731 ----
      .end annotation
  
      .prologue
+     .line 220
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      .local p2, root:Landroid/preference/GenericInflater$Parent;,"TP;"
      if-eqz p2, :cond_0
***************
*** 695,700 ****
--- 757,763 ----
      .end annotation
  
      .prologue
+     .line 261
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      .local p2, root:Landroid/preference/GenericInflater$Parent;,"TP;"
      invoke-virtual {p0}, Landroid/preference/GenericInflater;->getContext()Landroid/content/Context;
***************
*** 709,714 ****
--- 772,778 ----
  
      move-result-object v0
  
+     .line 263
      .local v0, parser:Landroid/content/res/XmlResourceParser;
      :try_start_0
      invoke-virtual {p0, v0, p2, p3}, Landroid/preference/GenericInflater;->inflate(Lorg/xmlpull/v1/XmlPullParser;Landroid/preference/GenericInflater$Parent;Z)Ljava/lang/Object;
***************
*** 717,722 ****
--- 781,787 ----
  
      move-result-object v1
  
+     .line 265
      invoke-interface {v0}, Landroid/content/res/XmlResourceParser;->close()V
  
      return-object v1
***************
*** 742,747 ****
--- 807,813 ----
      .end annotation
  
      .prologue
+     .line 240
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      .local p2, root:Landroid/preference/GenericInflater$Parent;,"TP;"
      if-eqz p2, :cond_0
***************
*** 779,793 ****
--- 845,862 ----
      .local p2, root:Landroid/preference/GenericInflater$Parent;,"TP;"
      const/4 v10, 0x2
  
+     .line 292
      iget-object v7, p0, Landroid/preference/GenericInflater;->mConstructorArgs:[Ljava/lang/Object;
  
      monitor-enter v7
  
+     .line 293
      :try_start_0
      invoke-static {p1}, Landroid/util/Xml;->asAttributeSet(Lorg/xmlpull/v1/XmlPullParser;)Landroid/util/AttributeSet;
  
      move-result-object v0
  
+     .line 294
      .local v0, attrs:Landroid/util/AttributeSet;
      iget-object v6, p0, Landroid/preference/GenericInflater;->mConstructorArgs:[Ljava/lang/Object;
  
***************
*** 799,806 ****
--- 868,877 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 295
      move-object v3, p2
  
+     .line 301
      .local v3, result:Landroid/preference/GenericInflater$Parent;,"TT;"
      :cond_0
      :try_start_1
***************
*** 815,823 ****
--- 886,896 ----
  
      if-ne v4, v6, :cond_0
  
+     .line 305
      :cond_1
      if-eq v4, v10, :cond_2
  
+     .line 306
      new-instance v6, Landroid/view/InflateException;
  
      new-instance v8, Ljava/lang/StringBuilder;
***************
*** 851,864 ****
--- 924,940 ----
      .catch Lorg/xmlpull/v1/XmlPullParserException; {:try_start_1 .. :try_end_1} :catch_1
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_2
  
+     .line 331
      .end local v4           #type:I
      :catch_0
      move-exception v1
  
+     .line 332
      .local v1, e:Landroid/view/InflateException;
      :try_start_2
      throw v1
  
+     .line 347
      .end local v0           #attrs:Landroid/util/AttributeSet;
      .end local v1           #e:Landroid/view/InflateException;
      .end local v3           #result:Landroid/preference/GenericInflater$Parent;,"TT;"
***************
*** 871,876 ****
--- 947,953 ----
  
      throw v6
  
+     .line 317
      .restart local v0       #attrs:Landroid/util/AttributeSet;
      .restart local v3       #result:Landroid/preference/GenericInflater$Parent;,"TT;"
      .restart local v4       #type:I
***************
*** 884,889 ****
--- 961,967 ----
  
      move-result-object v5
  
+     .line 320
      .local v5, xmlRoot:Ljava/lang/Object;,"TT;"
      check-cast v5, Landroid/preference/GenericInflater$Parent;
  
***************
*** 892,897 ****
--- 970,976 ----
  
      move-result-object v3
  
+     .line 326
      invoke-direct {p0, p1, v3, v0}, Landroid/preference/GenericInflater;->rInflate(Lorg/xmlpull/v1/XmlPullParser;Ljava/lang/Object;Landroid/util/AttributeSet;)V
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
***************
*** 899,913 ****
--- 978,995 ----
      .catch Lorg/xmlpull/v1/XmlPullParserException; {:try_start_3 .. :try_end_3} :catch_1
      .catch Ljava/io/IOException; {:try_start_3 .. :try_end_3} :catch_2
  
+     .line 346
      :try_start_4
      monitor-exit v7
  
      return-object v3
  
+     .line 334
      .end local v4           #type:I
      :catch_1
      move-exception v1
  
+     .line 335
      .local v1, e:Lorg/xmlpull/v1/XmlPullParserException;
      new-instance v2, Landroid/view/InflateException;
  
***************
*** 917,932 ****
--- 999,1018 ----
  
      invoke-direct {v2, v6}, Landroid/view/InflateException;-><init>(Ljava/lang/String;)V
  
+     .line 336
      .local v2, ex:Landroid/view/InflateException;
      invoke-virtual {v2, v1}, Landroid/view/InflateException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 337
      throw v2
  
+     .line 338
      .end local v1           #e:Lorg/xmlpull/v1/XmlPullParserException;
      .end local v2           #ex:Landroid/view/InflateException;
      :catch_2
      move-exception v1
  
+     .line 339
      .local v1, e:Ljava/io/IOException;
      new-instance v2, Landroid/view/InflateException;
  
***************
*** 962,970 ****
--- 1048,1058 ----
  
      invoke-direct {v2, v6}, Landroid/view/InflateException;-><init>(Ljava/lang/String;)V
  
+     .line 342
      .restart local v2       #ex:Landroid/view/InflateException;
      invoke-virtual {v2, v1}, Landroid/view/InflateException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 343
      throw v2
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
***************
*** 992,997 ****
--- 1080,1086 ----
      .end annotation
  
      .prologue
+     .line 514
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      .local p2, parent:Ljava/lang/Object;,"TT;"
      const/4 v0, 0x0
***************
*** 1019,1024 ****
--- 1108,1114 ----
      .end annotation
  
      .prologue
+     .line 417
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      iget-object v0, p0, Landroid/preference/GenericInflater;->mDefaultPackage:Ljava/lang/String;
  
***************
*** 1041,1046 ****
--- 1131,1137 ----
      .end annotation
  
      .prologue
+     .line 518
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      .local p1, givenRoot:Landroid/preference/GenericInflater$Parent;,"TP;"
      .local p3, xmlRoot:Landroid/preference/GenericInflater$Parent;,"TP;"
***************
*** 1052,1060 ****
--- 1143,1153 ----
      .parameter "defaultPackage"
  
      .prologue
+     .line 147
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      iput-object p1, p0, Landroid/preference/GenericInflater;->mDefaultPackage:Ljava/lang/String;
  
+     .line 148
      return-void
  .end method
  
***************
*** 1070,1081 ****
--- 1163,1176 ----
      .end annotation
  
      .prologue
+     .line 192
      .local p0, this:Landroid/preference/GenericInflater;,"Landroid/preference/GenericInflater<TT;TP;>;"
      .local p1, factory:Landroid/preference/GenericInflater$Factory;,"Landroid/preference/GenericInflater$Factory<TT;>;"
      iget-boolean v0, p0, Landroid/preference/GenericInflater;->mFactorySet:Z
  
      if-eqz v0, :cond_0
  
+     .line 193
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "A factory has already been set on this inflater"
***************
*** 1084,1092 ****
--- 1179,1189 ----
  
      throw v0
  
+     .line 196
      :cond_0
      if-nez p1, :cond_1
  
+     .line 197
      new-instance v0, Ljava/lang/NullPointerException;
  
      const-string v1, "Given factory can not be null"
***************
*** 1095,1114 ****
--- 1192,1216 ----
  
      throw v0
  
+     .line 199
      :cond_1
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/preference/GenericInflater;->mFactorySet:Z
  
+     .line 200
      iget-object v0, p0, Landroid/preference/GenericInflater;->mFactory:Landroid/preference/GenericInflater$Factory;
  
      if-nez v0, :cond_2
  
+     .line 201
      iput-object p1, p0, Landroid/preference/GenericInflater;->mFactory:Landroid/preference/GenericInflater$Factory;
  
+     .line 205
      :goto_0
      return-void
  
+     .line 203
      :cond_2
      new-instance v0, Landroid/preference/GenericInflater$FactoryMerger;
  
