*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/ListPreference.smali	2012-03-23 19:37:03.891846050 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/ListPreference.smali.original	2012-03-23 19:34:45.525322941 -0500
***************
*** 29,38 ****
--- 29,40 ----
      .parameter "context"
  
      .prologue
+     .line 64
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/preference/ListPreference;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 65
      return-void
  .end method
  
***************
*** 44,57 ****
--- 46,62 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 46
      invoke-direct {p0, p1, p2}, Landroid/preference/DialogPreference;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 48
      sget-object v1, Lcom/android/internal/R$styleable;->ListPreference:[I
  
      invoke-virtual {p1, p2, v1, v2, v2}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[III)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 50
      .local v0, a:Landroid/content/res/TypedArray;
      invoke-virtual {v0, v2}, Landroid/content/res/TypedArray;->getTextArray(I)[Ljava/lang/CharSequence;
  
***************
*** 59,64 ****
--- 64,70 ----
  
      iput-object v1, p0, Landroid/preference/ListPreference;->mEntries:[Ljava/lang/CharSequence;
  
+     .line 51
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/content/res/TypedArray;->getTextArray(I)[Ljava/lang/CharSequence;
***************
*** 67,80 ****
--- 73,89 ----
  
      iput-object v1, p0, Landroid/preference/ListPreference;->mEntryValues:[Ljava/lang/CharSequence;
  
+     .line 52
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 57
      sget-object v1, Lcom/android/internal/R$styleable;->Preference:[I
  
      invoke-virtual {p1, p2, v1, v2, v2}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[III)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 59
      const/4 v1, 0x7
  
      invoke-virtual {v0, v1}, Landroid/content/res/TypedArray;->getString(I)Ljava/lang/String;
***************
*** 83,90 ****
--- 92,101 ----
  
      iput-object v1, p0, Landroid/preference/ListPreference;->mSummary:Ljava/lang/String;
  
+     .line 60
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 61
      return-void
  .end method
  
***************
*** 94,99 ****
--- 105,111 ----
      .parameter "x1"
  
      .prologue
+     .line 38
      iput p1, p0, Landroid/preference/ListPreference;->mClickedDialogEntryIndex:I
  
      return p1
***************
*** 103,108 ****
--- 115,121 ----
      .locals 1
  
      .prologue
+     .line 224
      iget-object v0, p0, Landroid/preference/ListPreference;->mValue:Ljava/lang/String;
  
      invoke-virtual {p0, v0}, Landroid/preference/ListPreference;->findIndexOfValue(Ljava/lang/String;)I
***************
*** 119,130 ****
--- 132,145 ----
      .parameter "value"
  
      .prologue
+     .line 213
      if-eqz p1, :cond_1
  
      iget-object v1, p0, Landroid/preference/ListPreference;->mEntryValues:[Ljava/lang/CharSequence;
  
      if-eqz v1, :cond_1
  
+     .line 214
      iget-object v1, p0, Landroid/preference/ListPreference;->mEntryValues:[Ljava/lang/CharSequence;
  
      array-length v1, v1
***************
*** 135,140 ****
--- 150,156 ----
      :goto_0
      if-ltz v0, :cond_1
  
+     .line 215
      iget-object v1, p0, Landroid/preference/ListPreference;->mEntryValues:[Ljava/lang/CharSequence;
  
      aget-object v1, v1, v0
***************
*** 145,160 ****
--- 161,179 ----
  
      if-eqz v1, :cond_0
  
+     .line 220
      .end local v0           #i:I
      :goto_1
      return v0
  
+     .line 214
      .restart local v0       #i:I
      :cond_0
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_0
  
+     .line 220
      .end local v0           #i:I
      :cond_1
      const/4 v0, -0x1
***************
*** 166,171 ****
--- 185,191 ----
      .locals 1
  
      .prologue
+     .line 95
      iget-object v0, p0, Landroid/preference/ListPreference;->mEntries:[Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 175,184 ****
--- 195,206 ----
      .locals 2
  
      .prologue
+     .line 202
      invoke-direct {p0}, Landroid/preference/ListPreference;->getValueIndex()I
  
      move-result v0
  
+     .line 203
      .local v0, index:I
      if-ltz v0, :cond_0
  
***************
*** 203,208 ****
--- 225,231 ----
      .locals 1
  
      .prologue
+     .line 123
      iget-object v0, p0, Landroid/preference/ListPreference;->mEntryValues:[Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 212,221 ****
--- 235,246 ----
      .locals 4
  
      .prologue
+     .line 148
      invoke-virtual {p0}, Landroid/preference/ListPreference;->getEntry()Ljava/lang/CharSequence;
  
      move-result-object v0
  
+     .line 149
      .local v0, entry:Ljava/lang/CharSequence;
      iget-object v1, p0, Landroid/preference/ListPreference;->mSummary:Ljava/lang/String;
  
***************
*** 223,233 ****
--- 248,260 ----
  
      if-nez v0, :cond_1
  
+     .line 150
      :cond_0
      invoke-super {p0}, Landroid/preference/DialogPreference;->getSummary()Ljava/lang/CharSequence;
  
      move-result-object v1
  
+     .line 152
      :goto_0
      return-object v1
  
***************
*** 253,258 ****
--- 280,286 ----
      .locals 1
  
      .prologue
+     .line 193
      iget-object v0, p0, Landroid/preference/ListPreference;->mValue:Ljava/lang/String;
  
      return-object v0
***************
*** 263,270 ****
--- 291,300 ----
      .parameter "positiveResult"
  
      .prologue
+     .line 261
      invoke-super {p0, p1}, Landroid/preference/DialogPreference;->onDialogClosed(Z)V
  
+     .line 263
      if-eqz p1, :cond_0
  
      iget v1, p0, Landroid/preference/ListPreference;->mClickedDialogEntryIndex:I
***************
*** 275,280 ****
--- 305,311 ----
  
      if-eqz v1, :cond_0
  
+     .line 264
      iget-object v1, p0, Landroid/preference/ListPreference;->mEntryValues:[Ljava/lang/CharSequence;
  
      iget v2, p0, Landroid/preference/ListPreference;->mClickedDialogEntryIndex:I
***************
*** 285,290 ****
--- 316,322 ----
  
      move-result-object v0
  
+     .line 265
      .local v0, value:Ljava/lang/String;
      invoke-virtual {p0, v0}, Landroid/preference/ListPreference;->callChangeListener(Ljava/lang/Object;)Z
  
***************
*** 292,299 ****
--- 324,333 ----
  
      if-eqz v1, :cond_0
  
+     .line 266
      invoke-virtual {p0, v0}, Landroid/preference/ListPreference;->setValue(Ljava/lang/String;)V
  
+     .line 269
      .end local v0           #value:Ljava/lang/String;
      :cond_0
      return-void
***************
*** 305,310 ****
--- 339,345 ----
      .parameter "index"
  
      .prologue
+     .line 273
      invoke-virtual {p1, p2}, Landroid/content/res/TypedArray;->getString(I)Ljava/lang/String;
  
      move-result-object v0
***************
*** 319,326 ****
--- 354,363 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 229
      invoke-super {p0, p1}, Landroid/preference/DialogPreference;->onPrepareDialogBuilder(Landroid/app/AlertDialog$Builder;)V
  
+     .line 231
      iget-object v0, p0, Landroid/preference/ListPreference;->mEntries:[Ljava/lang/CharSequence;
  
      if-eqz v0, :cond_0
***************
*** 329,334 ****
--- 366,372 ----
  
      if-nez v0, :cond_1
  
+     .line 232
      :cond_0
      new-instance v0, Ljava/lang/IllegalStateException;
  
***************
*** 338,343 ****
--- 376,382 ----
  
      throw v0
  
+     .line 236
      :cond_1
      invoke-direct {p0}, Landroid/preference/ListPreference;->getValueIndex()I
  
***************
*** 345,350 ****
--- 384,390 ----
  
      iput v0, p0, Landroid/preference/ListPreference;->mClickedDialogEntryIndex:I
  
+     .line 237
      iget-object v0, p0, Landroid/preference/ListPreference;->mEntries:[Ljava/lang/CharSequence;
  
      iget v1, p0, Landroid/preference/ListPreference;->mClickedDialogEntryIndex:I
***************
*** 355,362 ****
--- 395,404 ----
  
      invoke-virtual {p1, v0, v1, v2}, Landroid/app/AlertDialog$Builder;->setSingleChoiceItems([Ljava/lang/CharSequence;ILandroid/content/DialogInterface$OnClickListener;)Landroid/app/AlertDialog$Builder;
  
+     .line 256
      invoke-virtual {p1, v3, v3}, Landroid/app/AlertDialog$Builder;->setPositiveButton(Ljava/lang/CharSequence;Landroid/content/DialogInterface$OnClickListener;)Landroid/app/AlertDialog$Builder;
  
+     .line 257
      return-void
  .end method
  
***************
*** 365,370 ****
--- 407,413 ----
      .parameter "state"
  
      .prologue
+     .line 296
      if-eqz p1, :cond_0
  
      invoke-virtual {p1}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
***************
*** 379,395 ****
--- 422,442 ----
  
      if-nez v1, :cond_1
  
+     .line 298
      :cond_0
      invoke-super {p0, p1}, Landroid/preference/DialogPreference;->onRestoreInstanceState(Landroid/os/Parcelable;)V
  
+     .line 305
      :goto_0
      return-void
  
      :cond_1
      move-object v0, p1
  
+     .line 302
      check-cast v0, Landroid/preference/ListPreference$SavedState;
  
+     .line 303
      .local v0, myState:Landroid/preference/ListPreference$SavedState;
      invoke-virtual {v0}, Landroid/preference/ListPreference$SavedState;->getSuperState()Landroid/os/Parcelable;
  
***************
*** 397,402 ****
--- 444,450 ----
  
      invoke-super {p0, v1}, Landroid/preference/DialogPreference;->onRestoreInstanceState(Landroid/os/Parcelable;)V
  
+     .line 304
      iget-object v1, v0, Landroid/preference/ListPreference$SavedState;->value:Ljava/lang/String;
  
      invoke-virtual {p0, v1}, Landroid/preference/ListPreference;->setValue(Ljava/lang/String;)V
***************
*** 408,417 ****
--- 456,467 ----
      .locals 3
  
      .prologue
+     .line 283
      invoke-super {p0}, Landroid/preference/DialogPreference;->onSaveInstanceState()Landroid/os/Parcelable;
  
      move-result-object v1
  
+     .line 284
      .local v1, superState:Landroid/os/Parcelable;
      invoke-virtual {p0}, Landroid/preference/ListPreference;->isPersistent()Z
  
***************
*** 419,434 ****
--- 469,487 ----
  
      if-eqz v2, :cond_0
  
+     .line 291
      .end local v1           #superState:Landroid/os/Parcelable;
      :goto_0
      return-object v1
  
+     .line 289
      .restart local v1       #superState:Landroid/os/Parcelable;
      :cond_0
      new-instance v0, Landroid/preference/ListPreference$SavedState;
  
      invoke-direct {v0, v1}, Landroid/preference/ListPreference$SavedState;-><init>(Landroid/os/Parcelable;)V
  
+     .line 290
      .local v0, myState:Landroid/preference/ListPreference$SavedState;
      invoke-virtual {p0}, Landroid/preference/ListPreference;->getValue()Ljava/lang/String;
  
***************
*** 438,443 ****
--- 491,497 ----
  
      move-object v1, v0
  
+     .line 291
      goto :goto_0
  .end method
  
***************
*** 447,452 ****
--- 501,507 ----
      .parameter "defaultValue"
  
      .prologue
+     .line 278
      if-eqz p1, :cond_0
  
      iget-object v0, p0, Landroid/preference/ListPreference;->mValue:Ljava/lang/String;
***************
*** 459,466 ****
--- 514,523 ----
      :goto_0
      invoke-virtual {p0, p2}, Landroid/preference/ListPreference;->setValue(Ljava/lang/String;)V
  
+     .line 279
      return-void
  
+     .line 278
      .restart local p2
      :cond_0
      check-cast p2, Ljava/lang/String;
***************
*** 473,478 ****
--- 530,536 ----
      .parameter "entriesResId"
  
      .prologue
+     .line 86
      invoke-virtual {p0}, Landroid/preference/ListPreference;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 487,492 ****
--- 545,551 ----
  
      invoke-virtual {p0, v0}, Landroid/preference/ListPreference;->setEntries([Ljava/lang/CharSequence;)V
  
+     .line 87
      return-void
  .end method
  
***************
*** 495,502 ****
--- 554,563 ----
      .parameter "entries"
  
      .prologue
+     .line 78
      iput-object p1, p0, Landroid/preference/ListPreference;->mEntries:[Ljava/lang/CharSequence;
  
+     .line 79
      return-void
  .end method
  
***************
*** 505,510 ****
--- 566,572 ----
      .parameter "entryValuesResId"
  
      .prologue
+     .line 114
      invoke-virtual {p0}, Landroid/preference/ListPreference;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 519,524 ****
--- 581,587 ----
  
      invoke-virtual {p0, v0}, Landroid/preference/ListPreference;->setEntryValues([Ljava/lang/CharSequence;)V
  
+     .line 115
      return-void
  .end method
  
***************
*** 527,534 ****
--- 590,599 ----
      .parameter "entryValues"
  
      .prologue
+     .line 106
      iput-object p1, p0, Landroid/preference/ListPreference;->mEntryValues:[Ljava/lang/CharSequence;
  
+     .line 107
      return-void
  .end method
  
***************
*** 537,558 ****
--- 602,628 ----
      .parameter "summary"
  
      .prologue
+     .line 167
      invoke-super {p0, p1}, Landroid/preference/DialogPreference;->setSummary(Ljava/lang/CharSequence;)V
  
+     .line 168
      if-nez p1, :cond_1
  
      iget-object v0, p0, Landroid/preference/ListPreference;->mSummary:Ljava/lang/String;
  
      if-eqz v0, :cond_1
  
+     .line 169
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/preference/ListPreference;->mSummary:Ljava/lang/String;
  
+     .line 173
      :cond_0
      :goto_0
      return-void
  
+     .line 170
      :cond_1
      if-eqz p1, :cond_0
  
***************
*** 564,569 ****
--- 634,640 ----
  
      if-nez v0, :cond_0
  
+     .line 171
      invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
      move-result-object v0
***************
*** 578,587 ****
--- 649,661 ----
      .parameter "value"
  
      .prologue
+     .line 133
      iput-object p1, p0, Landroid/preference/ListPreference;->mValue:Ljava/lang/String;
  
+     .line 135
      invoke-virtual {p0, p1}, Landroid/preference/ListPreference;->persistString(Ljava/lang/String;)Z
  
+     .line 136
      return-void
  .end method
  
***************
*** 590,599 ****
--- 664,675 ----
      .parameter "index"
  
      .prologue
+     .line 181
      iget-object v0, p0, Landroid/preference/ListPreference;->mEntryValues:[Ljava/lang/CharSequence;
  
      if-eqz v0, :cond_0
  
+     .line 182
      iget-object v0, p0, Landroid/preference/ListPreference;->mEntryValues:[Ljava/lang/CharSequence;
  
      aget-object v0, v0, p1
***************
*** 604,609 ****
--- 680,686 ----
  
      invoke-virtual {p0, v0}, Landroid/preference/ListPreference;->setValue(Ljava/lang/String;)V
  
+     .line 184
      :cond_0
      return-void
  .end method
