*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/PreferenceCategory.smali	2012-03-23 19:37:03.124832065 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/PreferenceCategory.smali.original	2012-03-23 19:34:45.520322849 -0500
***************
*** 13,22 ****
--- 13,24 ----
      .parameter "context"
  
      .prologue
+     .line 40
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/preference/PreferenceCategory;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 41
      return-void
  .end method
  
***************
*** 26,35 ****
--- 28,39 ----
      .parameter "attrs"
  
      .prologue
+     .line 36
      const v0, 0x101008c
  
      invoke-direct {p0, p1, p2, v0}, Landroid/preference/PreferenceCategory;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 37
      return-void
  .end method
  
***************
*** 40,47 ****
--- 44,53 ----
      .parameter "defStyle"
  
      .prologue
+     .line 32
      invoke-direct {p0, p1, p2, p3}, Landroid/preference/PreferenceGroup;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 33
      return-void
  .end method
  
***************
*** 51,56 ****
--- 57,63 ----
      .locals 1
  
      .prologue
+     .line 55
      const/4 v0, 0x0
  
      return v0
***************
*** 61,70 ****
--- 68,79 ----
      .parameter "preference"
  
      .prologue
+     .line 45
      instance-of v0, p1, Landroid/preference/PreferenceCategory;
  
      if-eqz v0, :cond_0
  
+     .line 46
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "Cannot add a PreferenceCategory directly to a PreferenceCategory"
***************
*** 73,78 ****
--- 82,88 ----
  
      throw v0
  
+     .line 50
      :cond_0
      invoke-super {p0, p1}, Landroid/preference/PreferenceGroup;->onPrepareAddPreference(Landroid/preference/Preference;)Z
  
