*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/PreferenceFrameLayout.smali	2012-03-23 19:37:03.445837918 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/PreferenceFrameLayout.smali.original	2012-03-23 19:34:45.522322885 -0500
***************
*** 39,48 ****
--- 39,50 ----
      .parameter "context"
  
      .prologue
+     .line 41
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/preference/PreferenceFrameLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 42
      return-void
  .end method
  
***************
*** 52,61 ****
--- 54,65 ----
      .parameter "attrs"
  
      .prologue
+     .line 45
      const v0, 0x10103d4
  
      invoke-direct {p0, p1, p2, v0}, Landroid/preference/PreferenceFrameLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 46
      return-void
  .end method
  
***************
*** 72,85 ****
--- 76,92 ----
  
      const/4 v7, 0x0
  
+     .line 49
      invoke-direct {p0, p1, p2, p3}, Landroid/widget/FrameLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 50
      sget-object v6, Lcom/android/internal/R$styleable;->PreferenceFrameLayout:[I
  
      invoke-virtual {p1, p2, v6, p3, v9}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[III)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 53
      .local v0, a:Landroid/content/res/TypedArray;
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
***************
*** 91,96 ****
--- 98,104 ----
  
      iget v5, v6, Landroid/util/DisplayMetrics;->density:F
  
+     .line 54
      .local v5, density:F
      mul-float v6, v5, v7
  
***************
*** 98,103 ****
--- 106,112 ----
  
      float-to-int v1, v6
  
+     .line 55
      .local v1, defaultBorderTop:I
      mul-float v6, v5, v7
  
***************
*** 105,110 ****
--- 114,120 ----
  
      float-to-int v2, v6
  
+     .line 56
      .local v2, defaultBottomPadding:I
      mul-float v6, v5, v7
  
***************
*** 112,117 ****
--- 122,128 ----
  
      float-to-int v3, v6
  
+     .line 57
      .local v3, defaultLeftPadding:I
      mul-float v6, v5, v7
  
***************
*** 119,124 ****
--- 130,136 ----
  
      float-to-int v4, v6
  
+     .line 59
      .local v4, defaultRightPadding:I
      invoke-virtual {v0, v9, v1}, Landroid/content/res/TypedArray;->getDimensionPixelSize(II)I
  
***************
*** 126,131 ****
--- 138,144 ----
  
      iput v6, p0, Landroid/preference/PreferenceFrameLayout;->mBorderTop:I
  
+     .line 62
      const/4 v6, 0x1
  
      invoke-virtual {v0, v6, v2}, Landroid/content/res/TypedArray;->getDimensionPixelSize(II)I
***************
*** 134,139 ****
--- 147,153 ----
  
      iput v6, p0, Landroid/preference/PreferenceFrameLayout;->mBorderBottom:I
  
+     .line 65
      const/4 v6, 0x2
  
      invoke-virtual {v0, v6, v3}, Landroid/content/res/TypedArray;->getDimensionPixelSize(II)I
***************
*** 142,147 ****
--- 156,162 ----
  
      iput v6, p0, Landroid/preference/PreferenceFrameLayout;->mBorderLeft:I
  
+     .line 68
      const/4 v6, 0x3
  
      invoke-virtual {v0, v6, v4}, Landroid/content/res/TypedArray;->getDimensionPixelSize(II)I
***************
*** 150,157 ****
--- 165,174 ----
  
      iput v6, p0, Landroid/preference/PreferenceFrameLayout;->mBorderRight:I
  
+     .line 72
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 73
      return-void
  .end method
  
***************
*** 162,191 ****
--- 179,214 ----
      .parameter "child"
  
      .prologue
+     .line 85
      invoke-virtual {p0}, Landroid/preference/PreferenceFrameLayout;->getPaddingTop()I
  
      move-result v3
  
+     .line 86
      .local v3, borderTop:I
      invoke-virtual {p0}, Landroid/preference/PreferenceFrameLayout;->getPaddingBottom()I
  
      move-result v0
  
+     .line 87
      .local v0, borderBottom:I
      invoke-virtual {p0}, Landroid/preference/PreferenceFrameLayout;->getPaddingLeft()I
  
      move-result v1
  
+     .line 88
      .local v1, borderLeft:I
      invoke-virtual {p0}, Landroid/preference/PreferenceFrameLayout;->getPaddingRight()I
  
      move-result v2
  
+     .line 90
      .local v2, borderRight:I
      invoke-virtual {p1}, Landroid/view/View;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
  
      move-result-object v5
  
+     .line 91
      .local v5, params:Landroid/view/ViewGroup$LayoutParams;
      instance-of v10, v5, Landroid/preference/PreferenceFrameLayout$LayoutParams;
  
***************
*** 199,204 ****
--- 222,228 ----
  
      move-object v4, v10
  
+     .line 94
      .local v4, layoutParams:Landroid/preference/PreferenceFrameLayout$LayoutParams;
      :goto_0
      if-eqz v4, :cond_4
***************
*** 207,257 ****
--- 231,292 ----
  
      if-eqz v10, :cond_4
  
+     .line 95
      iget-boolean v10, p0, Landroid/preference/PreferenceFrameLayout;->mPaddingApplied:Z
  
      if-eqz v10, :cond_0
  
+     .line 96
      iget v10, p0, Landroid/preference/PreferenceFrameLayout;->mBorderTop:I
  
      sub-int/2addr v3, v10
  
+     .line 97
      iget v10, p0, Landroid/preference/PreferenceFrameLayout;->mBorderBottom:I
  
      sub-int/2addr v0, v10
  
+     .line 98
      iget v10, p0, Landroid/preference/PreferenceFrameLayout;->mBorderLeft:I
  
      sub-int/2addr v1, v10
  
+     .line 99
      iget v10, p0, Landroid/preference/PreferenceFrameLayout;->mBorderRight:I
  
      sub-int/2addr v2, v10
  
+     .line 100
      const/4 v10, 0x0
  
      iput-boolean v10, p0, Landroid/preference/PreferenceFrameLayout;->mPaddingApplied:Z
  
+     .line 114
      :cond_0
      :goto_1
      invoke-virtual {p0}, Landroid/preference/PreferenceFrameLayout;->getPaddingTop()I
  
      move-result v9
  
+     .line 115
      .local v9, previousTop:I
      invoke-virtual {p0}, Landroid/preference/PreferenceFrameLayout;->getPaddingBottom()I
  
      move-result v6
  
+     .line 116
      .local v6, previousBottom:I
      invoke-virtual {p0}, Landroid/preference/PreferenceFrameLayout;->getPaddingLeft()I
  
      move-result v7
  
+     .line 117
      .local v7, previousLeft:I
      invoke-virtual {p0}, Landroid/preference/PreferenceFrameLayout;->getPaddingRight()I
  
      move-result v8
  
+     .line 118
      .local v8, previousRight:I
      if-ne v9, v3, :cond_1
  
***************
*** 261,274 ****
--- 296,313 ----
  
      if-eq v8, v2, :cond_2
  
+     .line 120
      :cond_1
      invoke-virtual {p0, v1, v3, v2, v0}, Landroid/preference/PreferenceFrameLayout;->setPadding(IIII)V
  
+     .line 123
      :cond_2
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->addView(Landroid/view/View;)V
  
+     .line 124
      return-void
  
+     .line 91
      .end local v4           #layoutParams:Landroid/preference/PreferenceFrameLayout$LayoutParams;
      .end local v6           #previousBottom:I
      .end local v7           #previousLeft:I
***************
*** 279,306 ****
--- 318,351 ----
  
      goto :goto_0
  
+     .line 105
      .restart local v4       #layoutParams:Landroid/preference/PreferenceFrameLayout$LayoutParams;
      :cond_4
      iget-boolean v10, p0, Landroid/preference/PreferenceFrameLayout;->mPaddingApplied:Z
  
      if-nez v10, :cond_0
  
+     .line 106
      iget v10, p0, Landroid/preference/PreferenceFrameLayout;->mBorderTop:I
  
      add-int/2addr v3, v10
  
+     .line 107
      iget v10, p0, Landroid/preference/PreferenceFrameLayout;->mBorderBottom:I
  
      add-int/2addr v0, v10
  
+     .line 108
      iget v10, p0, Landroid/preference/PreferenceFrameLayout;->mBorderLeft:I
  
      add-int/2addr v1, v10
  
+     .line 109
      iget v10, p0, Landroid/preference/PreferenceFrameLayout;->mBorderRight:I
  
      add-int/2addr v2, v10
  
+     .line 110
      const/4 v10, 0x1
  
      iput-boolean v10, p0, Landroid/preference/PreferenceFrameLayout;->mPaddingApplied:Z
***************
*** 313,318 ****
--- 358,364 ----
      .parameter "attrs"
  
      .prologue
+     .line 80
      new-instance v0, Landroid/preference/PreferenceFrameLayout$LayoutParams;
  
      invoke-virtual {p0}, Landroid/preference/PreferenceFrameLayout;->getContext()Landroid/content/Context;
***************
*** 329,334 ****
--- 375,381 ----
      .parameter "x0"
  
      .prologue
+     .line 29
      invoke-virtual {p0, p1}, Landroid/preference/PreferenceFrameLayout;->generateLayoutParams(Landroid/util/AttributeSet;)Landroid/preference/PreferenceFrameLayout$LayoutParams;
  
      move-result-object v0
***************
*** 341,346 ****
--- 388,394 ----
      .parameter "x0"
  
      .prologue
+     .line 29
      invoke-virtual {p0, p1}, Landroid/preference/PreferenceFrameLayout;->generateLayoutParams(Landroid/util/AttributeSet;)Landroid/preference/PreferenceFrameLayout$LayoutParams;
  
      move-result-object v0
