*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/PreferenceGroupAdapter$1.smali	2012-03-23 19:37:03.650841656 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/PreferenceGroupAdapter$1.smali.original	2012-03-23 19:34:45.523322903 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 85
      iput-object p1, p0, Landroid/preference/PreferenceGroupAdapter$1;->this$0:Landroid/preference/PreferenceGroupAdapter;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,49 ****
--- 41,52 ----
      .locals 1
  
      .prologue
+     .line 87
      iget-object v0, p0, Landroid/preference/PreferenceGroupAdapter$1;->this$0:Landroid/preference/PreferenceGroupAdapter;
  
      #calls: Landroid/preference/PreferenceGroupAdapter;->syncMyPreferences()V
      invoke-static {v0}, Landroid/preference/PreferenceGroupAdapter;->access$100(Landroid/preference/PreferenceGroupAdapter;)V
  
+     .line 88
      return-void
  .end method
