*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/TwoStatePreference.smali	2012-03-23 19:37:03.748843443 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/TwoStatePreference.smali.original	2012-03-23 19:34:45.524322922 -0500
***************
*** 29,38 ****
--- 29,40 ----
      .parameter "context"
  
      .prologue
+     .line 53
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/preference/TwoStatePreference;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 54
      return-void
  .end method
  
***************
*** 42,51 ****
--- 44,55 ----
      .parameter "attrs"
  
      .prologue
+     .line 49
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, p2, v0}, Landroid/preference/TwoStatePreference;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 50
      return-void
  .end method
  
***************
*** 56,63 ****
--- 60,69 ----
      .parameter "defStyle"
  
      .prologue
+     .line 45
      invoke-direct {p0, p1, p2, p3}, Landroid/preference/Preference;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 46
      return-void
  .end method
  
***************
*** 67,72 ****
--- 73,79 ----
      .locals 1
  
      .prologue
+     .line 164
      iget-boolean v0, p0, Landroid/preference/TwoStatePreference;->mDisableDependentsState:Z
  
      return v0
***************
*** 76,81 ****
--- 83,89 ----
      .locals 1
  
      .prologue
+     .line 153
      iget-object v0, p0, Landroid/preference/TwoStatePreference;->mSummaryOff:Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 85,90 ****
--- 93,99 ----
      .locals 1
  
      .prologue
+     .line 125
      iget-object v0, p0, Landroid/preference/TwoStatePreference;->mSummaryOn:Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 94,99 ****
--- 103,109 ----
      .locals 1
  
      .prologue
+     .line 91
      iget-boolean v0, p0, Landroid/preference/TwoStatePreference;->mChecked:Z
  
      return v0
***************
*** 105,112 ****
--- 115,124 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 58
      invoke-super {p0}, Landroid/preference/Preference;->onClick()V
  
+     .line 60
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->isChecked()Z
  
      move-result v2
***************
*** 115,124 ****
--- 127,138 ----
  
      move v0, v1
  
+     .line 62
      .local v0, newValue:Z
      :goto_0
      iput-boolean v1, p0, Landroid/preference/TwoStatePreference;->mSendClickAccessibilityEvent:Z
  
+     .line 64
      invoke-static {v0}, Ljava/lang/Boolean;->valueOf(Z)Ljava/lang/Boolean;
  
      move-result-object v1
***************
*** 129,143 ****
--- 143,160 ----
  
      if-nez v1, :cond_1
  
+     .line 69
      :goto_1
      return-void
  
+     .line 60
      .end local v0           #newValue:Z
      :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 68
      .restart local v0       #newValue:Z
      :cond_1
      invoke-virtual {p0, v0}, Landroid/preference/TwoStatePreference;->setChecked(Z)V
***************
*** 151,156 ****
--- 168,174 ----
      .parameter "index"
  
      .prologue
+     .line 179
      const/4 v0, 0x0
  
      invoke-virtual {p1, p2, v0}, Landroid/content/res/TypedArray;->getBoolean(IZ)Z
***************
*** 169,174 ****
--- 187,193 ----
      .parameter "state"
  
      .prologue
+     .line 254
      if-eqz p1, :cond_0
  
      invoke-virtual {p1}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
***************
*** 183,199 ****
--- 202,222 ----
  
      if-nez v1, :cond_1
  
+     .line 256
      :cond_0
      invoke-super {p0, p1}, Landroid/preference/Preference;->onRestoreInstanceState(Landroid/os/Parcelable;)V
  
+     .line 263
      :goto_0
      return-void
  
      :cond_1
      move-object v0, p1
  
+     .line 260
      check-cast v0, Landroid/preference/TwoStatePreference$SavedState;
  
+     .line 261
      .local v0, myState:Landroid/preference/TwoStatePreference$SavedState;
      invoke-virtual {v0}, Landroid/preference/TwoStatePreference$SavedState;->getSuperState()Landroid/os/Parcelable;
  
***************
*** 201,206 ****
--- 224,230 ----
  
      invoke-super {p0, v1}, Landroid/preference/Preference;->onRestoreInstanceState(Landroid/os/Parcelable;)V
  
+     .line 262
      iget-boolean v1, v0, Landroid/preference/TwoStatePreference$SavedState;->checked:Z
  
      invoke-virtual {p0, v1}, Landroid/preference/TwoStatePreference;->setChecked(Z)V
***************
*** 212,221 ****
--- 236,247 ----
      .locals 3
  
      .prologue
+     .line 241
      invoke-super {p0}, Landroid/preference/Preference;->onSaveInstanceState()Landroid/os/Parcelable;
  
      move-result-object v1
  
+     .line 242
      .local v1, superState:Landroid/os/Parcelable;
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->isPersistent()Z
  
***************
*** 223,238 ****
--- 249,267 ----
  
      if-eqz v2, :cond_0
  
+     .line 249
      .end local v1           #superState:Landroid/os/Parcelable;
      :goto_0
      return-object v1
  
+     .line 247
      .restart local v1       #superState:Landroid/os/Parcelable;
      :cond_0
      new-instance v0, Landroid/preference/TwoStatePreference$SavedState;
  
      invoke-direct {v0, v1}, Landroid/preference/TwoStatePreference$SavedState;-><init>(Landroid/os/Parcelable;)V
  
+     .line 248
      .local v0, myState:Landroid/preference/TwoStatePreference$SavedState;
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->isChecked()Z
  
***************
*** 242,247 ****
--- 271,277 ----
  
      move-object v1, v0
  
+     .line 249
      goto :goto_0
  .end method
  
***************
*** 251,256 ****
--- 281,287 ----
      .parameter "defaultValue"
  
      .prologue
+     .line 184
      if-eqz p1, :cond_0
  
      iget-boolean v0, p0, Landroid/preference/TwoStatePreference;->mChecked:Z
***************
*** 263,270 ****
--- 294,303 ----
      :goto_0
      invoke-virtual {p0, v0}, Landroid/preference/TwoStatePreference;->setChecked(Z)V
  
+     .line 186
      return-void
  
+     .line 184
      .restart local p2
      :cond_0
      check-cast p2, Ljava/lang/Boolean;
***************
*** 282,287 ****
--- 315,321 ----
      .parameter "view"
  
      .prologue
+     .line 192
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->getContext()Landroid/content/Context;
  
      move-result-object v2
***************
*** 290,295 ****
--- 324,330 ----
  
      move-result-object v0
  
+     .line 193
      .local v0, accessibilityManager:Landroid/view/accessibility/AccessibilityManager;
      iget-boolean v2, p0, Landroid/preference/TwoStatePreference;->mSendClickAccessibilityEvent:Z
  
***************
*** 301,327 ****
--- 336,369 ----
  
      if-eqz v2, :cond_0
  
+     .line 194
      invoke-static {}, Landroid/view/accessibility/AccessibilityEvent;->obtain()Landroid/view/accessibility/AccessibilityEvent;
  
      move-result-object v1
  
+     .line 195
      .local v1, event:Landroid/view/accessibility/AccessibilityEvent;
      const/4 v2, 0x1
  
      invoke-virtual {v1, v2}, Landroid/view/accessibility/AccessibilityEvent;->setEventType(I)V
  
+     .line 196
      invoke-virtual {p1, v1}, Landroid/view/View;->onInitializeAccessibilityEvent(Landroid/view/accessibility/AccessibilityEvent;)V
  
+     .line 197
      invoke-virtual {p1, v1}, Landroid/view/View;->dispatchPopulateAccessibilityEvent(Landroid/view/accessibility/AccessibilityEvent;)Z
  
+     .line 198
      invoke-virtual {v0, v1}, Landroid/view/accessibility/AccessibilityManager;->sendAccessibilityEvent(Landroid/view/accessibility/AccessibilityEvent;)V
  
+     .line 200
      .end local v1           #event:Landroid/view/accessibility/AccessibilityEvent;
      :cond_0
      const/4 v2, 0x0
  
      iput-boolean v2, p0, Landroid/preference/TwoStatePreference;->mSendClickAccessibilityEvent:Z
  
+     .line 201
      return-void
  .end method
  
***************
*** 330,351 ****
--- 372,399 ----
      .parameter "checked"
  
      .prologue
+     .line 77
      iget-boolean v0, p0, Landroid/preference/TwoStatePreference;->mChecked:Z
  
      if-eq v0, p1, :cond_0
  
+     .line 78
      iput-boolean p1, p0, Landroid/preference/TwoStatePreference;->mChecked:Z
  
+     .line 79
      invoke-virtual {p0, p1}, Landroid/preference/TwoStatePreference;->persistBoolean(Z)Z
  
+     .line 80
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->shouldDisableDependents()Z
  
      move-result v0
  
      invoke-virtual {p0, v0}, Landroid/preference/TwoStatePreference;->notifyDependencyChange(Z)V
  
+     .line 81
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->notifyChanged()V
  
+     .line 83
      :cond_0
      return-void
  .end method
***************
*** 355,362 ****
--- 403,412 ----
      .parameter "disableDependentsState"
  
      .prologue
+     .line 174
      iput-boolean p1, p0, Landroid/preference/TwoStatePreference;->mDisableDependentsState:Z
  
+     .line 175
      return-void
  .end method
  
***************
*** 365,370 ****
--- 415,421 ----
      .parameter "summaryResId"
  
      .prologue
+     .line 145
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 375,380 ****
--- 426,432 ----
  
      invoke-virtual {p0, v0}, Landroid/preference/TwoStatePreference;->setSummaryOff(Ljava/lang/CharSequence;)V
  
+     .line 146
      return-void
  .end method
  
***************
*** 383,398 ****
--- 435,454 ----
      .parameter "summary"
  
      .prologue
+     .line 134
      iput-object p1, p0, Landroid/preference/TwoStatePreference;->mSummaryOff:Ljava/lang/CharSequence;
  
+     .line 135
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->isChecked()Z
  
      move-result v0
  
      if-nez v0, :cond_0
  
+     .line 136
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->notifyChanged()V
  
+     .line 138
      :cond_0
      return-void
  .end method
***************
*** 402,407 ****
--- 458,464 ----
      .parameter "summaryResId"
  
      .prologue
+     .line 117
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 412,417 ****
--- 469,475 ----
  
      invoke-virtual {p0, v0}, Landroid/preference/TwoStatePreference;->setSummaryOn(Ljava/lang/CharSequence;)V
  
+     .line 118
      return-void
  .end method
  
***************
*** 420,435 ****
--- 478,497 ----
      .parameter "summary"
  
      .prologue
+     .line 106
      iput-object p1, p0, Landroid/preference/TwoStatePreference;->mSummaryOn:Ljava/lang/CharSequence;
  
+     .line 107
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->isChecked()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 108
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->notifyChanged()V
  
+     .line 110
      :cond_0
      return-void
  .end method
***************
*** 442,453 ****
--- 504,517 ----
  
      const/4 v1, 0x0
  
+     .line 96
      iget-boolean v3, p0, Landroid/preference/TwoStatePreference;->mDisableDependentsState:Z
  
      if-eqz v3, :cond_2
  
      iget-boolean v0, p0, Landroid/preference/TwoStatePreference;->mChecked:Z
  
+     .line 97
      .local v0, shouldDisable:Z
      :goto_0
      if-nez v0, :cond_0
***************
*** 464,469 ****
--- 528,534 ----
      :cond_1
      return v1
  
+     .line 96
      .end local v0           #shouldDisable:Z
      :cond_2
      iget-boolean v3, p0, Landroid/preference/TwoStatePreference;->mChecked:Z
***************
*** 485,490 ****
--- 550,556 ----
      .parameter "view"
  
      .prologue
+     .line 209
      const v4, 0x1020010
  
      invoke-virtual {p1, v4}, Landroid/view/View;->findViewById(I)Landroid/view/View;
***************
*** 493,503 ****
--- 559,572 ----
  
      check-cast v2, Landroid/widget/TextView;
  
+     .line 210
      .local v2, summaryView:Landroid/widget/TextView;
      if-eqz v2, :cond_3
  
+     .line 211
      const/4 v3, 0x1
  
+     .line 212
      .local v3, useDefaultSummary:Z
      iget-boolean v4, p0, Landroid/preference/TwoStatePreference;->mChecked:Z
  
***************
*** 507,542 ****
--- 576,622 ----
  
      if-eqz v4, :cond_4
  
+     .line 213
      iget-object v4, p0, Landroid/preference/TwoStatePreference;->mSummaryOn:Ljava/lang/CharSequence;
  
      invoke-virtual {v2, v4}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 214
      const/4 v3, 0x0
  
+     .line 220
      :cond_0
      :goto_0
      if-eqz v3, :cond_1
  
+     .line 221
      invoke-virtual {p0}, Landroid/preference/TwoStatePreference;->getSummary()Ljava/lang/CharSequence;
  
      move-result-object v1
  
+     .line 222
      .local v1, summary:Ljava/lang/CharSequence;
      if-eqz v1, :cond_1
  
+     .line 223
      invoke-virtual {v2, v1}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 224
      const/4 v3, 0x0
  
+     .line 228
      .end local v1           #summary:Ljava/lang/CharSequence;
      :cond_1
      const/16 v0, 0x8
  
+     .line 229
      .local v0, newVisibility:I
      if-nez v3, :cond_2
  
+     .line 231
      const/4 v0, 0x0
  
+     .line 233
      :cond_2
      invoke-virtual {v2}, Landroid/widget/TextView;->getVisibility()I
  
***************
*** 544,556 ****
--- 624,639 ----
  
      if-eq v0, v4, :cond_3
  
+     .line 234
      invoke-virtual {v2, v0}, Landroid/widget/TextView;->setVisibility(I)V
  
+     .line 237
      .end local v0           #newVisibility:I
      .end local v3           #useDefaultSummary:Z
      :cond_3
      return-void
  
+     .line 215
      .restart local v3       #useDefaultSummary:Z
      :cond_4
      iget-boolean v4, p0, Landroid/preference/TwoStatePreference;->mChecked:Z
***************
*** 561,570 ****
--- 644,655 ----
  
      if-eqz v4, :cond_0
  
+     .line 216
      iget-object v4, p0, Landroid/preference/TwoStatePreference;->mSummaryOff:Ljava/lang/CharSequence;
  
      invoke-virtual {v2, v4}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 217
      const/4 v3, 0x0
  
      goto :goto_0
