*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/VolumePreference$SeekBarVolumizer$1.smali	2012-03-23 19:37:03.220833815 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/preference/VolumePreference$SeekBarVolumizer$1.smali.original	2012-03-23 19:34:45.520322849 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .parameter "x0"
  
      .prologue
+     .line 237
      iput-object p1, p0, Landroid/preference/VolumePreference$SeekBarVolumizer$1;->this$1:Landroid/preference/VolumePreference$SeekBarVolumizer;
  
      invoke-direct {p0, p2}, Landroid/database/ContentObserver;-><init>(Landroid/os/Handler;)V
***************
*** 39,46 ****
--- 40,49 ----
      .parameter "selfChange"
  
      .prologue
+     .line 240
      invoke-super {p0, p1}, Landroid/database/ContentObserver;->onChange(Z)V
  
+     .line 241
      iget-object v1, p0, Landroid/preference/VolumePreference$SeekBarVolumizer$1;->this$1:Landroid/preference/VolumePreference$SeekBarVolumizer;
  
      #getter for: Landroid/preference/VolumePreference$SeekBarVolumizer;->mSeekBar:Landroid/widget/SeekBar;
***************
*** 59,64 ****
--- 62,68 ----
  
      if-eqz v1, :cond_0
  
+     .line 242
      iget-object v1, p0, Landroid/preference/VolumePreference$SeekBarVolumizer$1;->this$1:Landroid/preference/VolumePreference$SeekBarVolumizer;
  
      #getter for: Landroid/preference/VolumePreference$SeekBarVolumizer;->mAudioManager:Landroid/media/AudioManager;
***************
*** 97,102 ****
--- 101,107 ----
  
      move-result v0
  
+     .line 245
      .local v0, volume:I
      :goto_0
      iget-object v1, p0, Landroid/preference/VolumePreference$SeekBarVolumizer$1;->this$1:Landroid/preference/VolumePreference$SeekBarVolumizer;
***************
*** 108,117 ****
--- 113,124 ----
  
      invoke-virtual {v1, v0}, Landroid/widget/SeekBar;->setProgress(I)V
  
+     .line 247
      .end local v0           #volume:I
      :cond_0
      return-void
  
+     .line 242
      :cond_1
      iget-object v1, p0, Landroid/preference/VolumePreference$SeekBarVolumizer$1;->this$1:Landroid/preference/VolumePreference$SeekBarVolumizer;
  
