*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CdmaServiceStateTracker$1.smali	2012-03-23 19:37:39.083487769 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CdmaServiceStateTracker$1.smali.original	2012-03-23 19:34:46.192335103 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .parameter "x0"
  
      .prologue
+     .line 148
      iput-object p1, p0, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$1;->this$0:Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;
  
      invoke-direct {p0, p2}, Landroid/database/ContentObserver;-><init>(Landroid/os/Handler;)V
***************
*** 39,54 ****
--- 40,58 ----
      .parameter "selfChange"
  
      .prologue
+     .line 151
      iget-object v0, p0, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$1;->this$0:Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;
  
      const-string v1, "Auto time state changed"
  
      invoke-virtual {v0, v1}, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;->log(Ljava/lang/String;)V
  
+     .line 152
      iget-object v0, p0, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$1;->this$0:Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;
  
      #calls: Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;->revertToNitzTime()V
      invoke-static {v0}, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;->access$000(Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;)V
  
+     .line 153
      return-void
  .end method
