*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CdmaCallWaitingNotification.smali	2012-03-23 19:37:38.408475459 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CdmaCallWaitingNotification.smali.original	2012-03-23 19:34:46.181334903 -0500
***************
*** 38,63 ****
--- 38,74 ----
  
      const/4 v0, 0x0
  
+     .line 27
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 29
      iput-object v1, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->number:Ljava/lang/String;
  
+     .line 30
      iput v0, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->numberPresentation:I
  
+     .line 31
      iput-object v1, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->name:Ljava/lang/String;
  
+     .line 32
      iput v0, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->namePresentation:I
  
+     .line 33
      iput v0, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->numberType:I
  
+     .line 34
      iput v0, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->numberPlan:I
  
+     .line 35
      iput v0, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->isPresent:I
  
+     .line 36
      iput v0, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->signalType:I
  
+     .line 37
      iput v0, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->alertPitch:I
  
+     .line 38
      iput v0, p0, Lcom/android/internal/telephony/cdma/CdmaCallWaitingNotification;->signal:I
  
      return-void
***************
*** 68,75 ****
--- 79,88 ----
      .parameter "cli"
  
      .prologue
+     .line 58
      packed-switch p0, :pswitch_data_0
  
+     .line 64
      const-string v0, "CDMA"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 92,117 ****
--- 105,135 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 65
      sget v0, Lcom/android/internal/telephony/Connection;->PRESENTATION_UNKNOWN:I
  
      :goto_0
      return v0
  
+     .line 59
      :pswitch_0
      sget v0, Lcom/android/internal/telephony/Connection;->PRESENTATION_ALLOWED:I
  
      goto :goto_0
  
+     .line 60
      :pswitch_1
      sget v0, Lcom/android/internal/telephony/Connection;->PRESENTATION_RESTRICTED:I
  
      goto :goto_0
  
+     .line 61
      :pswitch_2
      sget v0, Lcom/android/internal/telephony/Connection;->PRESENTATION_UNKNOWN:I
  
      goto :goto_0
  
+     .line 58
      nop
  
      :pswitch_data_0
***************
*** 128,133 ****
--- 146,152 ----
      .locals 2
  
      .prologue
+     .line 42
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
