*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CDMALTEPhone$NetworkSelectMessage.smali	2012-03-23 19:37:37.650461636 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CDMALTEPhone$NetworkSelectMessage.smali.original	2012-03-23 19:34:46.171334721 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .locals 0
  
      .prologue
+     .line 54
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x0"
  
      .prologue
+     .line 54
      invoke-direct {p0}, Lcom/android/internal/telephony/cdma/CDMALTEPhone$NetworkSelectMessage;-><init>()V
  
      return-void
