*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CDMAPhone$2.smali	2012-03-23 19:37:38.304473563 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CDMAPhone$2.smali.original	2012-03-23 19:34:46.180334885 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 3
  
      .prologue
+     .line 634
      invoke-static {}, Lcom/android/internal/telephony/DataConnectionTracker$State;->values()[Lcom/android/internal/telephony/DataConnectionTracker$State;
  
      move-result-object v0
***************
*** 146,151 ****
--- 147,153 ----
      :try_end_6
      .catch Ljava/lang/NoSuchFieldError; {:try_start_6 .. :try_end_6} :catch_4
  
+     .line 342
      :goto_6
      invoke-static {}, Lcom/android/internal/telephony/DataConnectionTracker$Activity;->values()[Lcom/android/internal/telephony/DataConnectionTracker$Activity;
  
***************
*** 243,248 ****
--- 245,251 ----
  
      goto :goto_7
  
+     .line 634
      :catch_4
      move-exception v0
  
