*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CdmaInformationRecords$CdmaRedirectingNumberInfoRec.smali	2012-03-23 19:37:38.180471303 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/CdmaInformationRecords$CdmaRedirectingNumberInfoRec.smali.original	2012-03-23 19:34:46.178334847 -0500
***************
*** 45,52 ****
--- 45,54 ----
      .parameter "reason"
  
      .prologue
+     .line 191
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 192
      new-instance v0, Lcom/android/internal/telephony/cdma/CdmaInformationRecords$CdmaNumberInfoRec;
  
      const/4 v1, 0x5
***************
*** 65,72 ****
--- 67,76 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/cdma/CdmaInformationRecords$CdmaRedirectingNumberInfoRec;->numberInfoRec:Lcom/android/internal/telephony/cdma/CdmaInformationRecords$CdmaNumberInfoRec;
  
+     .line 194
      iput p6, p0, Lcom/android/internal/telephony/cdma/CdmaInformationRecords$CdmaRedirectingNumberInfoRec;->redirectingReason:I
  
+     .line 195
      return-void
  .end method
  
***************
*** 76,81 ****
--- 80,86 ----
      .locals 2
  
      .prologue
+     .line 199
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
