*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/EriManager.smali	2012-03-23 19:37:38.380474950 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/EriManager.smali.original	2012-03-23 19:34:46.181334903 -0500
***************
*** 46,69 ****
--- 46,76 ----
      .parameter "eriFileSource"
  
      .prologue
+     .line 102
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 98
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFileSource:I
  
+     .line 103
      iput-object p1, p0, Lcom/android/internal/telephony/cdma/EriManager;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
+     .line 104
      iput-object p2, p0, Lcom/android/internal/telephony/cdma/EriManager;->mContext:Landroid/content/Context;
  
+     .line 105
      iput p3, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFileSource:I
  
+     .line 106
      new-instance v0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;
  
      invoke-direct {v0, p0}, Lcom/android/internal/telephony/cdma/EriManager$EriFile;-><init>(Lcom/android/internal/telephony/cdma/EriManager;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
  
+     .line 107
      return-void
  .end method
  
***************
*** 83,99 ****
--- 90,110 ----
  
      const/4 v7, 0x0
  
+     .line 290
      iget-boolean v4, p0, Lcom/android/internal/telephony/cdma/EriManager;->isEriFileLoaded:Z
  
      if-eqz v4, :cond_0
  
+     .line 291
      invoke-direct {p0, p1}, Lcom/android/internal/telephony/cdma/EriManager;->getEriInfo(I)Lcom/android/internal/telephony/cdma/EriInfo;
  
      move-result-object v1
  
+     .line 292
      .local v1, eriInfo:Lcom/android/internal/telephony/cdma/EriInfo;
      if-eqz v1, :cond_0
  
+     .line 294
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
  
      iget v4, v1, Lcom/android/internal/telephony/cdma/EriInfo;->mIconIndex:I
***************
*** 107,134 ****
--- 118,151 ----
      .local v2, ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      move-object v3, v2
  
+     .line 478
      .end local v1           #eriInfo:Lcom/android/internal/telephony/cdma/EriInfo;
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      .local v3, ret:Ljava/lang/Object;
      :goto_0
      return-object v3
  
+     .line 302
      .end local v3           #ret:Ljava/lang/Object;
      :cond_0
      packed-switch p1, :pswitch_data_0
  
+     .line 399
      iget-boolean v4, p0, Lcom/android/internal/telephony/cdma/EriManager;->isEriFileLoaded:Z
  
      if-nez v4, :cond_2
  
+     .line 401
      const-string v4, "CDMA"
  
      const-string v5, "ERI File not loaded"
  
      invoke-static {v4, v5}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 402
      if-le p2, v8, :cond_1
  
+     .line 404
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
  
      iget-object v4, p0, Lcom/android/internal/telephony/cdma/EriManager;->mContext:Landroid/content/Context;
***************
*** 147,155 ****
--- 164,174 ----
      :goto_1
      move-object v3, v2
  
+     .line 478
      .restart local v3       #ret:Ljava/lang/Object;
      goto :goto_0
  
+     .line 305
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      .end local v3           #ret:Ljava/lang/Object;
      :pswitch_0
***************
*** 167,175 ****
--- 186,196 ----
  
      invoke-direct {v2, p0, v7, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 309
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto :goto_1
  
+     .line 312
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_1
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 188,196 ****
--- 209,219 ----
  
      invoke-direct {v2, p0, v6, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 316
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto :goto_1
  
+     .line 319
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_2
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 207,215 ****
--- 230,240 ----
  
      invoke-direct {v2, p0, v8, v6, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 323
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto :goto_1
  
+     .line 328
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_3
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 228,236 ****
--- 253,263 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 332
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto :goto_1
  
+     .line 335
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_4
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 249,257 ****
--- 276,286 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 339
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto :goto_1
  
+     .line 342
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_5
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 270,278 ****
--- 299,309 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 346
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto :goto_1
  
+     .line 349
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_6
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 291,299 ****
--- 322,332 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 353
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto :goto_1
  
+     .line 356
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_7
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 312,320 ****
--- 345,355 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 360
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 363
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_8
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 333,341 ****
--- 368,378 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 367
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 370
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_9
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 354,362 ****
--- 391,401 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 374
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 377
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_a
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 375,383 ****
--- 414,424 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 381
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 384
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_b
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 396,404 ****
--- 437,447 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 388
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 391
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_c
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 417,429 ****
--- 460,475 ----
  
      invoke-direct {v2, p0, p1, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 395
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 411
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :cond_1
      packed-switch p2, :pswitch_data_1
  
+     .line 437
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
  
      const/4 v4, -0x1
***************
*** 437,442 ****
--- 483,489 ----
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 413
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_d
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 453,461 ****
--- 500,510 ----
  
      invoke-direct {v2, p0, v7, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 418
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 421
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_e
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 474,482 ****
--- 523,533 ----
  
      invoke-direct {v2, p0, v6, v7, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 426
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 429
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :pswitch_f
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 493,517 ****
--- 544,574 ----
  
      invoke-direct {v2, p0, v8, v6, v4}, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;-><init>(Lcom/android/internal/telephony/cdma/EriManager;IILjava/lang/String;)V
  
+     .line 434
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 442
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :cond_2
      invoke-direct {p0, p1}, Lcom/android/internal/telephony/cdma/EriManager;->getEriInfo(I)Lcom/android/internal/telephony/cdma/EriInfo;
  
      move-result-object v1
  
+     .line 443
      .restart local v1       #eriInfo:Lcom/android/internal/telephony/cdma/EriInfo;
      invoke-direct {p0, p2}, Lcom/android/internal/telephony/cdma/EriManager;->getEriInfo(I)Lcom/android/internal/telephony/cdma/EriInfo;
  
      move-result-object v0
  
+     .line 444
      .local v0, defEriInfo:Lcom/android/internal/telephony/cdma/EriInfo;
      if-nez v1, :cond_4
  
+     .line 449
      if-nez v0, :cond_3
  
+     .line 450
      const-string v4, "CDMA"
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 540,545 ****
--- 597,603 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 452
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
  
      iget-object v4, p0, Lcom/android/internal/telephony/cdma/EriManager;->mContext:Landroid/content/Context;
***************
*** 557,562 ****
--- 615,621 ----
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 462
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :cond_3
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 572,577 ****
--- 631,637 ----
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 469
      .end local v2           #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      :cond_4
      new-instance v2, Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
***************
*** 587,592 ****
--- 647,653 ----
      .restart local v2       #ret:Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
      goto/16 :goto_1
  
+     .line 302
      nop
  
      :pswitch_data_0
***************
*** 606,611 ****
--- 667,673 ----
          :pswitch_c
      .end packed-switch
  
+     .line 411
      :pswitch_data_1
      .packed-switch 0x0
          :pswitch_d
***************
*** 619,624 ****
--- 681,687 ----
      .parameter "roamingIndicator"
  
      .prologue
+     .line 279
      iget-object v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
  
      iget-object v0, v0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mRoamIndTable:Ljava/util/HashMap;
***************
*** 633,638 ****
--- 696,702 ----
  
      if-eqz v0, :cond_0
  
+     .line 280
      iget-object v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
  
      iget-object v0, v0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mRoamIndTable:Ljava/util/HashMap;
***************
*** 647,652 ****
--- 711,717 ----
  
      check-cast v0, Lcom/android/internal/telephony/cdma/EriInfo;
  
+     .line 282
      :goto_0
      return-object v0
  
***************
*** 660,665 ****
--- 725,731 ----
      .locals 0
  
      .prologue
+     .line 150
      return-void
  .end method
  
***************
*** 667,672 ****
--- 733,739 ----
      .locals 0
  
      .prologue
+     .line 140
      return-void
  .end method
  
***************
*** 674,684 ****
--- 741,754 ----
      .locals 20
  
      .prologue
+     .line 157
      const/4 v13, 0x0
  
+     .line 158
      .local v13, parser:Lorg/xmlpull/v1/XmlPullParser;
      const/4 v15, 0x0
  
+     .line 159
      .local v15, stream:Ljava/io/FileInputStream;
      move-object/from16 v0, p0
  
***************
*** 688,693 ****
--- 758,764 ----
  
      move-result-object v14
  
+     .line 162
      .local v14, r:Landroid/content/res/Resources;
      :try_start_0
      const-string v2, "CDMA"
***************
*** 698,703 ****
--- 769,775 ----
  
      invoke-static {v2, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 163
      new-instance v16, Ljava/io/FileInputStream;
  
      const v2, 0x1040443
***************
*** 713,718 ****
--- 785,791 ----
      .catch Ljava/io/FileNotFoundException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Lorg/xmlpull/v1/XmlPullParserException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 165
      .end local v15           #stream:Ljava/io/FileInputStream;
      .local v16, stream:Ljava/io/FileInputStream;
      :try_start_1
***************
*** 720,731 ****
--- 793,806 ----
  
      move-result-object v13
  
+     .line 166
      const/4 v2, 0x0
  
      move-object/from16 v0, v16
  
      invoke-interface {v13, v0, v2}, Lorg/xmlpull/v1/XmlPullParser;->setInput(Ljava/io/InputStream;Ljava/lang/String;)V
  
+     .line 167
      const-string v2, "CDMA"
  
      const-string v18, "loadEriFileFromXml: opened alternate file"
***************
*** 739,749 ****
--- 814,826 ----
  
      move-object/from16 v15, v16
  
+     .line 176
      .end local v16           #stream:Ljava/io/FileInputStream;
      .restart local v15       #stream:Ljava/io/FileInputStream;
      :goto_0
      if-nez v13, :cond_0
  
+     .line 177
      const-string v2, "CDMA"
  
      const-string v18, "loadEriFileFromXml: open normal file"
***************
*** 752,769 ****
--- 829,849 ----
  
      invoke-static {v2, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 178
      const v2, 0x10f0002
  
      invoke-virtual {v14, v2}, Landroid/content/res/Resources;->getXml(I)Landroid/content/res/XmlResourceParser;
  
      move-result-object v13
  
+     .line 182
      :cond_0
      :try_start_2
      const-string v2, "EriFile"
  
      invoke-static {v13, v2}, Lcom/android/internal/util/XmlUtils;->beginDocument(Lorg/xmlpull/v1/XmlPullParser;Ljava/lang/String;)V
  
+     .line 183
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
***************
*** 788,793 ****
--- 868,874 ----
  
      iput v0, v2, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mVersionNumber:I
  
+     .line 185
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
***************
*** 812,817 ****
--- 893,899 ----
  
      iput v0, v2, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mNumberOfEriEntries:I
  
+     .line 187
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
***************
*** 836,855 ****
--- 918,942 ----
  
      iput v0, v2, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mEriFileType:I
  
+     .line 190
      const/4 v12, 0x0
  
+     .line 192
      .local v12, parsedEriEntries:I
      :cond_1
      :goto_1
      invoke-static {v13}, Lcom/android/internal/util/XmlUtils;->nextElement(Lorg/xmlpull/v1/XmlPullParser;)V
  
+     .line 193
      invoke-interface {v13}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
  
      move-result-object v11
  
+     .line 194
      .local v11, name:Ljava/lang/String;
      if-nez v11, :cond_5
  
+     .line 195
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
***************
*** 858,863 ****
--- 945,951 ----
  
      if-eq v12, v2, :cond_2
  
+     .line 196
      const-string v2, "CDMA"
  
      new-instance v18, Ljava/lang/StringBuilder;
***************
*** 912,917 ****
--- 1000,1006 ----
  
      invoke-static {v2, v0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 223
      :cond_2
      const-string v2, "CDMA"
  
***************
*** 921,926 ****
--- 1010,1016 ----
  
      invoke-static {v2, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 224
      const/4 v2, 0x1
  
      move-object/from16 v0, p0
***************
*** 930,962 ****
--- 1020,1059 ----
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
      .catch Ljava/lang/Exception; {:try_start_2 .. :try_end_2} :catch_2
  
+     .line 229
      instance-of v2, v13, Landroid/content/res/XmlResourceParser;
  
      if-eqz v2, :cond_3
  
+     .line 230
      check-cast v13, Landroid/content/res/XmlResourceParser;
  
      .end local v13           #parser:Lorg/xmlpull/v1/XmlPullParser;
      invoke-interface {v13}, Landroid/content/res/XmlResourceParser;->close()V
  
+     .line 233
      :cond_3
      if-eqz v15, :cond_4
  
+     .line 234
      :try_start_3
      invoke-virtual {v15}, Ljava/io/FileInputStream;->close()V
      :try_end_3
      .catch Ljava/io/IOException; {:try_start_3 .. :try_end_3} :catch_4
  
+     .line 240
      .end local v11           #name:Ljava/lang/String;
      .end local v12           #parsedEriEntries:I
      :cond_4
      :goto_2
      return-void
  
+     .line 168
      .restart local v13       #parser:Lorg/xmlpull/v1/XmlPullParser;
      :catch_0
      move-exception v9
  
+     .line 169
      .local v9, e:Ljava/io/FileNotFoundException;
      :goto_3
      const-string v2, "CDMA"
***************
*** 967,980 ****
--- 1064,1081 ----
  
      invoke-static {v2, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 170
      const/4 v13, 0x0
  
+     .line 174
      goto/16 :goto_0
  
+     .line 171
      .end local v9           #e:Ljava/io/FileNotFoundException;
      :catch_1
      move-exception v9
  
+     .line 172
      .local v9, e:Lorg/xmlpull/v1/XmlPullParserException;
      :goto_4
      const-string v2, "CDMA"
***************
*** 985,994 ****
--- 1086,1097 ----
  
      invoke-static {v2, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 173
      const/4 v13, 0x0
  
      goto/16 :goto_0
  
+     .line 199
      .end local v9           #e:Lorg/xmlpull/v1/XmlPullParserException;
      .restart local v11       #name:Ljava/lang/String;
      .restart local v12       #parsedEriEntries:I
***************
*** 1002,1007 ****
--- 1105,1111 ----
  
      if-eqz v2, :cond_a
  
+     .line 200
      const/4 v2, 0x0
  
      const-string v18, "Id"
***************
*** 1016,1021 ****
--- 1120,1126 ----
  
      move-result v10
  
+     .line 201
      .local v10, id:I
      const/4 v2, 0x0
  
***************
*** 1027,1032 ****
--- 1132,1138 ----
  
      move-result-object v17
  
+     .line 202
      .local v17, text:Ljava/lang/String;
      if-ltz v10, :cond_7
  
***************
*** 1034,1039 ****
--- 1140,1146 ----
  
      if-gt v10, v2, :cond_7
  
+     .line 203
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
***************
*** 1047,1052 ****
--- 1154,1160 ----
  
      goto/16 :goto_1
  
+     .line 226
      .end local v10           #id:I
      .end local v11           #name:Ljava/lang/String;
      .end local v12           #parsedEriEntries:I
***************
*** 1054,1059 ****
--- 1162,1168 ----
      :catch_2
      move-exception v9
  
+     .line 227
      .local v9, e:Ljava/lang/Exception;
      :try_start_5
      const-string v2, "CDMA"
***************
*** 1066,1083 ****
--- 1175,1196 ----
      :try_end_5
      .catchall {:try_start_5 .. :try_end_5} :catchall_0
  
+     .line 229
      instance-of v2, v13, Landroid/content/res/XmlResourceParser;
  
      if-eqz v2, :cond_6
  
+     .line 230
      check-cast v13, Landroid/content/res/XmlResourceParser;
  
      .end local v13           #parser:Lorg/xmlpull/v1/XmlPullParser;
      invoke-interface {v13}, Landroid/content/res/XmlResourceParser;->close()V
  
+     .line 233
      :cond_6
      if-eqz v15, :cond_4
  
+     .line 234
      :try_start_6
      invoke-virtual {v15}, Ljava/io/FileInputStream;->close()V
      :try_end_6
***************
*** 1085,1095 ****
--- 1198,1210 ----
  
      goto :goto_2
  
+     .line 236
      :catch_3
      move-exception v2
  
      goto :goto_2
  
+     .line 205
      .end local v9           #e:Ljava/lang/Exception;
      .restart local v10       #id:I
      .restart local v11       #name:Ljava/lang/String;
***************
*** 1135,1140 ****
--- 1250,1256 ----
  
      goto/16 :goto_1
  
+     .line 229
      .end local v10           #id:I
      .end local v11           #name:Ljava/lang/String;
      .end local v12           #parsedEriEntries:I
***************
*** 1148,1170 ****
--- 1264,1291 ----
  
      if-eqz v18, :cond_8
  
+     .line 230
      check-cast v13, Landroid/content/res/XmlResourceParser;
  
      .end local v13           #parser:Lorg/xmlpull/v1/XmlPullParser;
      invoke-interface {v13}, Landroid/content/res/XmlResourceParser;->close()V
  
+     .line 233
      :cond_8
      if-eqz v15, :cond_9
  
+     .line 234
      :try_start_8
      invoke-virtual {v15}, Ljava/io/FileInputStream;->close()V
      :try_end_8
      .catch Ljava/io/IOException; {:try_start_8 .. :try_end_8} :catch_5
  
+     .line 238
      :cond_9
      :goto_5
      throw v2
  
+     .line 208
      .restart local v11       #name:Ljava/lang/String;
      .restart local v12       #parsedEriEntries:I
      .restart local v13       #parser:Lorg/xmlpull/v1/XmlPullParser;
***************
*** 1178,1183 ****
--- 1299,1305 ----
  
      if-eqz v2, :cond_1
  
+     .line 209
      const/4 v2, 0x0
  
      const-string v18, "RoamingIndicator"
***************
*** 1192,1197 ****
--- 1314,1320 ----
  
      move-result v3
  
+     .line 211
      .local v3, roamingIndicator:I
      const/4 v2, 0x0
  
***************
*** 1207,1212 ****
--- 1330,1336 ----
  
      move-result v4
  
+     .line 212
      .local v4, iconIndex:I
      const/4 v2, 0x0
  
***************
*** 1222,1227 ****
--- 1346,1352 ----
  
      move-result v5
  
+     .line 213
      .local v5, iconMode:I
      const/4 v2, 0x0
  
***************
*** 1233,1238 ****
--- 1358,1364 ----
  
      move-result-object v6
  
+     .line 214
      .local v6, eriText:Ljava/lang/String;
      const/4 v2, 0x0
  
***************
*** 1248,1253 ****
--- 1374,1380 ----
  
      move-result v7
  
+     .line 216
      .local v7, callPromptId:I
      const/4 v2, 0x0
  
***************
*** 1263,1271 ****
--- 1390,1400 ----
  
      move-result v8
  
+     .line 217
      .local v8, alertId:I
      add-int/lit8 v12, v12, 0x1
  
+     .line 218
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
***************
*** 1293,1298 ****
--- 1422,1428 ----
  
      goto/16 :goto_1
  
+     .line 236
      .end local v3           #roamingIndicator:I
      .end local v4           #iconIndex:I
      .end local v5           #iconMode:I
***************
*** 1312,1317 ****
--- 1442,1448 ----
  
      goto :goto_5
  
+     .line 171
      .end local v15           #stream:Ljava/io/FileInputStream;
      .restart local v13       #parser:Lorg/xmlpull/v1/XmlPullParser;
      .restart local v16       #stream:Ljava/io/FileInputStream;
***************
*** 1324,1329 ****
--- 1455,1461 ----
      .restart local v15       #stream:Ljava/io/FileInputStream;
      goto/16 :goto_4
  
+     .line 168
      .end local v15           #stream:Ljava/io/FileInputStream;
      .restart local v16       #stream:Ljava/io/FileInputStream;
      :catch_7
***************
*** 1342,1357 ****
--- 1474,1492 ----
      .locals 1
  
      .prologue
+     .line 110
      new-instance v0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;
  
      invoke-direct {v0, p0}, Lcom/android/internal/telephony/cdma/EriManager$EriFile;-><init>(Lcom/android/internal/telephony/cdma/EriManager;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
  
+     .line 111
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->isEriFileLoaded:Z
  
+     .line 112
      return-void
  .end method
  
***************
*** 1361,1366 ****
--- 1496,1502 ----
      .parameter "defRoamInd"
  
      .prologue
+     .line 482
      invoke-direct {p0, p1, p2}, Lcom/android/internal/telephony/cdma/EriManager;->getEriDisplayInformation(II)Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
  
      move-result-object v0
***************
*** 1376,1381 ****
--- 1512,1518 ----
      .parameter "defRoamInd"
  
      .prologue
+     .line 486
      invoke-direct {p0, p1, p2}, Lcom/android/internal/telephony/cdma/EriManager;->getEriDisplayInformation(II)Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
  
      move-result-object v0
***************
*** 1391,1396 ****
--- 1528,1534 ----
      .parameter "defRoamInd"
  
      .prologue
+     .line 490
      invoke-direct {p0, p1, p2}, Lcom/android/internal/telephony/cdma/EriManager;->getEriDisplayInformation(II)Lcom/android/internal/telephony/cdma/EriManager$EriDisplayInformation;
  
      move-result-object v0
***************
*** 1404,1409 ****
--- 1542,1548 ----
      .locals 1
  
      .prologue
+     .line 263
      iget-object v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
  
      iget v0, v0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mEriFileType:I
***************
*** 1415,1420 ****
--- 1554,1560 ----
      .locals 1
  
      .prologue
+     .line 247
      iget-object v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
  
      iget v0, v0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mVersionNumber:I
***************
*** 1426,1431 ****
--- 1566,1572 ----
      .locals 1
  
      .prologue
+     .line 255
      iget-object v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFile:Lcom/android/internal/telephony/cdma/EriManager$EriFile;
  
      iget v0, v0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mNumberOfEriEntries:I
***************
*** 1437,1442 ****
--- 1578,1584 ----
      .locals 1
  
      .prologue
+     .line 271
      iget-boolean v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->isEriFileLoaded:Z
  
      return v0
***************
*** 1446,1470 ****
--- 1588,1618 ----
      .locals 1
  
      .prologue
+     .line 116
      iget v0, p0, Lcom/android/internal/telephony/cdma/EriManager;->mEriFileSource:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 127
      invoke-direct {p0}, Lcom/android/internal/telephony/cdma/EriManager;->loadEriFileFromXml()V
  
+     .line 130
      :goto_0
      return-void
  
+     .line 118
      :pswitch_0
      invoke-direct {p0}, Lcom/android/internal/telephony/cdma/EriManager;->loadEriFileFromModem()V
  
      goto :goto_0
  
+     .line 122
      :pswitch_1
      invoke-direct {p0}, Lcom/android/internal/telephony/cdma/EriManager;->loadEriFileFromFileSystem()V
  
      goto :goto_0
  
+     .line 116
      nop
  
      :pswitch_data_0
