*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/EriManager$EriFile.smali	2012-03-23 19:37:37.465458264 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/EriManager$EriFile.smali.original	2012-03-23 19:34:46.167334647 -0500
***************
*** 48,63 ****
--- 48,68 ----
  
      const/4 v0, -0x1
  
+     .line 55
      iput-object p1, p0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->this$0:Lcom/android/internal/telephony/cdma/EriManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 56
      iput v0, p0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mVersionNumber:I
  
+     .line 57
      iput v2, p0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mNumberOfEriEntries:I
  
+     .line 58
      iput v0, p0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mEriFileType:I
  
+     .line 59
      const/4 v0, 0x3
  
      new-array v0, v0, [Ljava/lang/String;
***************
*** 80,90 ****
--- 85,97 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mCallPromptId:[Ljava/lang/String;
  
+     .line 60
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/cdma/EriManager$EriFile;->mRoamIndTable:Ljava/util/HashMap;
  
+     .line 61
      return-void
  .end method
