*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/TtyIntent.smali	2012-03-23 19:37:38.011468220 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/TtyIntent.smali.original	2012-03-23 19:34:46.175334793 -0500
***************
*** 20,25 ****
--- 20,26 ----
      .locals 0
  
      .prologue
+     .line 19
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
