*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/audiofx/EnvironmentalReverb.smali	2012-03-23 19:35:35.432232991 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/audiofx/EnvironmentalReverb.smali.original	2012-03-23 19:34:44.727308388 -0500
***************
*** 55,60 ****
--- 55,61 ----
      .locals 1
  
      .prologue
+     .line 617
      const/16 v0, 0x1a
  
      sput v0, Landroid/media/audiofx/EnvironmentalReverb;->PROPERTY_SIZE:I
***************
*** 77,98 ****
--- 78,104 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 146
      sget-object v0, Landroid/media/audiofx/EnvironmentalReverb;->EFFECT_TYPE_ENV_REVERB:Ljava/util/UUID;
  
      sget-object v1, Landroid/media/audiofx/EnvironmentalReverb;->EFFECT_TYPE_NULL:Ljava/util/UUID;
  
      invoke-direct {p0, v0, v1, p1, p2}, Landroid/media/audiofx/AudioEffect;-><init>(Ljava/util/UUID;Ljava/util/UUID;II)V
  
+     .line 115
      iput-object v2, p0, Landroid/media/audiofx/EnvironmentalReverb;->mParamListener:Landroid/media/audiofx/EnvironmentalReverb$OnParameterChangeListener;
  
+     .line 121
      iput-object v2, p0, Landroid/media/audiofx/EnvironmentalReverb;->mBaseParamListener:Landroid/media/audiofx/EnvironmentalReverb$BaseParameterListener;
  
+     .line 126
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      iput-object v0, p0, Landroid/media/audiofx/EnvironmentalReverb;->mParamListenerLock:Ljava/lang/Object;
  
+     .line 147
      return-void
  .end method
  
***************
*** 101,106 ****
--- 107,113 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Landroid/media/audiofx/EnvironmentalReverb;->mParamListenerLock:Ljava/lang/Object;
  
      return-object v0
***************
*** 111,116 ****
--- 118,124 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Landroid/media/audiofx/EnvironmentalReverb;->mParamListener:Landroid/media/audiofx/EnvironmentalReverb$OnParameterChangeListener;
  
      return-object v0
***************
*** 129,138 ****
--- 137,148 ----
      .end annotation
  
      .prologue
+     .line 256
      const/4 v1, 0x2
  
      new-array v0, v1, [B
  
+     .line 257
      .local v0, param:[B
      const/4 v1, 0x3
  
***************
*** 142,147 ****
--- 152,158 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 258
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([B)S
  
      move-result v1
***************
*** 160,169 ****
--- 171,182 ----
      .end annotation
  
      .prologue
+     .line 227
      const/4 v1, 0x4
  
      new-array v0, v1, [B
  
+     .line 228
      .local v0, param:[B
      const/4 v1, 0x2
  
***************
*** 173,178 ****
--- 186,192 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 229
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToInt([B)I
  
      move-result v1
***************
*** 191,200 ****
--- 205,216 ----
      .end annotation
  
      .prologue
+     .line 432
      const/4 v1, 0x2
  
      new-array v0, v1, [B
  
+     .line 433
      .local v0, param:[B
      const/16 v1, 0x9
  
***************
*** 204,209 ****
--- 220,226 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 434
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([B)S
  
      move-result v1
***************
*** 222,231 ****
--- 239,250 ----
      .end annotation
  
      .prologue
+     .line 399
      const/4 v1, 0x2
  
      new-array v0, v1, [B
  
+     .line 400
      .local v0, param:[B
      const/16 v1, 0x8
  
***************
*** 235,240 ****
--- 254,260 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 401
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([B)S
  
      move-result v1
***************
*** 255,264 ****
--- 275,286 ----
      .prologue
      const/16 v3, 0xa
  
+     .line 629
      sget v2, Landroid/media/audiofx/EnvironmentalReverb;->PROPERTY_SIZE:I
  
      new-array v0, v2, [B
  
+     .line 630
      .local v0, param:[B
      invoke-virtual {p0, v3, v0}, Landroid/media/audiofx/EnvironmentalReverb;->getParameter(I[B)I
  
***************
*** 266,275 ****
--- 288,299 ----
  
      invoke-virtual {p0, v2}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 631
      new-instance v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;
  
      invoke-direct {v1}, Landroid/media/audiofx/EnvironmentalReverb$Settings;-><init>()V
  
+     .line 632
      .local v1, settings:Landroid/media/audiofx/EnvironmentalReverb$Settings;
      const/4 v2, 0x0
  
***************
*** 279,284 ****
--- 303,309 ----
  
      iput-short v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->roomLevel:S
  
+     .line 633
      const/4 v2, 0x2
  
      invoke-virtual {p0, v0, v2}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([BI)S
***************
*** 287,292 ****
--- 312,318 ----
  
      iput-short v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->roomHFLevel:S
  
+     .line 634
      const/4 v2, 0x4
  
      invoke-virtual {p0, v0, v2}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToInt([BI)I
***************
*** 295,300 ****
--- 321,327 ----
  
      iput v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->decayTime:I
  
+     .line 635
      const/16 v2, 0x8
  
      invoke-virtual {p0, v0, v2}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([BI)S
***************
*** 303,314 ****
--- 330,343 ----
  
      iput-short v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->decayHFRatio:S
  
+     .line 636
      invoke-virtual {p0, v0, v3}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([BI)S
  
      move-result v2
  
      iput-short v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->reflectionsLevel:S
  
+     .line 637
      const/16 v2, 0xc
  
      invoke-virtual {p0, v0, v2}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToInt([BI)I
***************
*** 317,322 ****
--- 346,352 ----
  
      iput v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->reflectionsDelay:I
  
+     .line 638
      const/16 v2, 0x10
  
      invoke-virtual {p0, v0, v2}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([BI)S
***************
*** 325,330 ****
--- 355,361 ----
  
      iput-short v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->reverbLevel:S
  
+     .line 639
      const/16 v2, 0x12
  
      invoke-virtual {p0, v0, v2}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToInt([BI)I
***************
*** 333,338 ****
--- 364,370 ----
  
      iput v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->reverbDelay:I
  
+     .line 640
      const/16 v2, 0x16
  
      invoke-virtual {p0, v0, v2}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([BI)S
***************
*** 341,346 ****
--- 373,379 ----
  
      iput-short v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->diffusion:S
  
+     .line 641
      const/16 v2, 0x18
  
      invoke-virtual {p0, v0, v2}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([BI)S
***************
*** 349,354 ****
--- 382,388 ----
  
      iput-short v2, v1, Landroid/media/audiofx/EnvironmentalReverb$Settings;->density:S
  
+     .line 642
      return-object v1
  .end method
  
***************
*** 363,372 ****
--- 397,408 ----
      .end annotation
  
      .prologue
+     .line 314
      const/4 v1, 0x4
  
      new-array v0, v1, [B
  
+     .line 315
      .local v0, param:[B
      const/4 v1, 0x5
  
***************
*** 376,381 ****
--- 412,418 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 316
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToInt([B)I
  
      move-result v1
***************
*** 394,403 ****
--- 431,442 ----
      .end annotation
  
      .prologue
+     .line 285
      const/4 v1, 0x2
  
      new-array v0, v1, [B
  
+     .line 286
      .local v0, param:[B
      const/4 v1, 0x4
  
***************
*** 407,412 ****
--- 446,452 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 287
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([B)S
  
      move-result v1
***************
*** 425,434 ****
--- 465,476 ----
      .end annotation
  
      .prologue
+     .line 369
      const/4 v1, 0x4
  
      new-array v0, v1, [B
  
+     .line 370
      .local v0, param:[B
      const/4 v1, 0x7
  
***************
*** 438,443 ****
--- 480,486 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 371
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToInt([B)I
  
      move-result v1
***************
*** 456,465 ****
--- 499,510 ----
      .end annotation
  
      .prologue
+     .line 342
      const/4 v1, 0x2
  
      new-array v0, v1, [B
  
+     .line 343
      .local v0, param:[B
      const/4 v1, 0x6
  
***************
*** 469,474 ****
--- 514,520 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 344
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([B)S
  
      move-result v1
***************
*** 487,496 ****
--- 533,544 ----
      .end annotation
  
      .prologue
+     .line 200
      const/4 v1, 0x2
  
      new-array v0, v1, [B
  
+     .line 201
      .local v0, param:[B
      const/4 v1, 0x1
  
***************
*** 500,505 ****
--- 548,554 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 202
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([B)S
  
      move-result v1
***************
*** 518,527 ****
--- 567,578 ----
      .end annotation
  
      .prologue
+     .line 171
      const/4 v1, 0x2
  
      new-array v0, v1, [B
  
+     .line 172
      .local v0, param:[B
      const/4 v1, 0x0
  
***************
*** 531,536 ****
--- 582,588 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 173
      invoke-virtual {p0, v0}, Landroid/media/audiofx/EnvironmentalReverb;->byteArrayToShort([B)S
  
      move-result v1
***************
*** 550,559 ****
--- 602,613 ----
      .end annotation
  
      .prologue
+     .line 243
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->shortToByteArray(S)[B
  
      move-result-object v0
  
+     .line 244
      .local v0, param:[B
      const/4 v1, 0x3
  
***************
*** 563,568 ****
--- 617,623 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 245
      return-void
  .end method
  
***************
*** 578,587 ****
--- 633,644 ----
      .end annotation
  
      .prologue
+     .line 214
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->intToByteArray(I)[B
  
      move-result-object v0
  
+     .line 215
      .local v0, param:[B
      const/4 v1, 0x2
  
***************
*** 591,596 ****
--- 648,654 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 216
      return-void
  .end method
  
***************
*** 606,615 ****
--- 664,675 ----
      .end annotation
  
      .prologue
+     .line 419
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->shortToByteArray(S)[B
  
      move-result-object v0
  
+     .line 420
      .local v0, param:[B
      const/16 v1, 0x9
  
***************
*** 619,624 ****
--- 679,685 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 421
      return-void
  .end method
  
***************
*** 634,643 ****
--- 695,706 ----
      .end annotation
  
      .prologue
+     .line 386
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->shortToByteArray(S)[B
  
      move-result-object v0
  
+     .line 387
      .local v0, param:[B
      const/16 v1, 0x8
  
***************
*** 647,652 ****
--- 710,716 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 388
      return-void
  .end method
  
***************
*** 655,671 ****
--- 719,739 ----
      .parameter "listener"
  
      .prologue
+     .line 495
      iget-object v1, p0, Landroid/media/audiofx/EnvironmentalReverb;->mParamListenerLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 496
      :try_start_0
      iget-object v0, p0, Landroid/media/audiofx/EnvironmentalReverb;->mParamListener:Landroid/media/audiofx/EnvironmentalReverb$OnParameterChangeListener;
  
      if-nez v0, :cond_0
  
+     .line 497
      iput-object p1, p0, Landroid/media/audiofx/EnvironmentalReverb;->mParamListener:Landroid/media/audiofx/EnvironmentalReverb$OnParameterChangeListener;
  
+     .line 498
      new-instance v0, Landroid/media/audiofx/EnvironmentalReverb$BaseParameterListener;
  
      const/4 v2, 0x0
***************
*** 674,688 ****
--- 742,760 ----
  
      iput-object v0, p0, Landroid/media/audiofx/EnvironmentalReverb;->mBaseParamListener:Landroid/media/audiofx/EnvironmentalReverb$BaseParameterListener;
  
+     .line 499
      iget-object v0, p0, Landroid/media/audiofx/EnvironmentalReverb;->mBaseParamListener:Landroid/media/audiofx/EnvironmentalReverb$BaseParameterListener;
  
      invoke-super {p0, v0}, Landroid/media/audiofx/AudioEffect;->setParameterListener(Landroid/media/audiofx/AudioEffect$OnParameterChangeListener;)V
  
+     .line 501
      :cond_0
      monitor-exit v1
  
+     .line 502
      return-void
  
+     .line 501
      :catchall_0
      move-exception v0
  
***************
*** 707,712 ****
--- 779,785 ----
      .prologue
      const/16 v4, 0xa
  
+     .line 656
      new-array v1, v4, [[B
  
      const/4 v2, 0x0
***************
*** 813,818 ****
--- 886,892 ----
  
      move-result-object v0
  
+     .line 667
      .local v0, param:[B
      invoke-virtual {p0, v4, v0}, Landroid/media/audiofx/EnvironmentalReverb;->setParameter(I[B)I
  
***************
*** 820,825 ****
--- 894,900 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 668
      return-void
  .end method
  
***************
*** 835,844 ****
--- 910,921 ----
      .end annotation
  
      .prologue
+     .line 301
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->intToByteArray(I)[B
  
      move-result-object v0
  
+     .line 302
      .local v0, param:[B
      const/4 v1, 0x5
  
***************
*** 848,853 ****
--- 925,931 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 303
      return-void
  .end method
  
***************
*** 863,872 ****
--- 941,952 ----
      .end annotation
  
      .prologue
+     .line 272
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->shortToByteArray(S)[B
  
      move-result-object v0
  
+     .line 273
      .local v0, param:[B
      const/4 v1, 0x4
  
***************
*** 876,881 ****
--- 956,962 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 274
      return-void
  .end method
  
***************
*** 891,900 ****
--- 972,983 ----
      .end annotation
  
      .prologue
+     .line 356
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->intToByteArray(I)[B
  
      move-result-object v0
  
+     .line 357
      .local v0, param:[B
      const/4 v1, 0x7
  
***************
*** 904,909 ****
--- 987,993 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 358
      return-void
  .end method
  
***************
*** 919,928 ****
--- 1003,1014 ----
      .end annotation
  
      .prologue
+     .line 329
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->shortToByteArray(S)[B
  
      move-result-object v0
  
+     .line 330
      .local v0, param:[B
      const/4 v1, 0x6
  
***************
*** 932,937 ****
--- 1018,1024 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 331
      return-void
  .end method
  
***************
*** 947,956 ****
--- 1034,1045 ----
      .end annotation
  
      .prologue
+     .line 187
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->shortToByteArray(S)[B
  
      move-result-object v0
  
+     .line 188
      .local v0, param:[B
      const/4 v1, 0x1
  
***************
*** 960,965 ****
--- 1049,1055 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 189
      return-void
  .end method
  
***************
*** 975,984 ****
--- 1065,1076 ----
      .end annotation
  
      .prologue
+     .line 158
      invoke-virtual {p0, p1}, Landroid/media/audiofx/EnvironmentalReverb;->shortToByteArray(S)[B
  
      move-result-object v0
  
+     .line 159
      .local v0, param:[B
      const/4 v1, 0x0
  
***************
*** 988,992 ****
--- 1080,1085 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/EnvironmentalReverb;->checkStatus(I)V
  
+     .line 160
      return-void
  .end method
