*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/audiofx/Equalizer$Settings.smali	2012-03-23 19:35:35.181228415 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/audiofx/Equalizer$Settings.smali.original	2012-03-23 19:34:44.725308352 -0500
***************
*** 27,42 ****
--- 27,46 ----
      .locals 1
  
      .prologue
+     .line 459
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 456
      const/4 v0, 0x0
  
      iput-short v0, p0, Landroid/media/audiofx/Equalizer$Settings;->numBands:S
  
+     .line 457
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/media/audiofx/Equalizer$Settings;->bandLevels:[S
  
+     .line 460
      return-void
  .end method
  
***************
*** 45,71 ****
--- 49,81 ----
      .parameter "settings"
  
      .prologue
+     .line 467
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 456
      const/4 v5, 0x0
  
      iput-short v5, p0, Landroid/media/audiofx/Equalizer$Settings;->numBands:S
  
+     .line 457
      const/4 v5, 0x0
  
      iput-object v5, p0, Landroid/media/audiofx/Equalizer$Settings;->bandLevels:[S
  
+     .line 468
      new-instance v3, Ljava/util/StringTokenizer;
  
      const-string v5, "=;"
  
      invoke-direct {v3, p1, v5}, Ljava/util/StringTokenizer;-><init>(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 469
      .local v3, st:Ljava/util/StringTokenizer;
      invoke-virtual {v3}, Ljava/util/StringTokenizer;->countTokens()I
  
      move-result v4
  
+     .line 470
      .local v4, tokens:I
      invoke-virtual {v3}, Ljava/util/StringTokenizer;->countTokens()I
  
***************
*** 75,87 ****
  
      if-ge v5, v6, :cond_0
  
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
      new-instance v6, Ljava/lang/StringBuilder;
  
      invoke-direct {v6}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v7, "settings: "
  
      invoke-virtual {v6, v7}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 85,98 ----
  
      if-ge v5, v6, :cond_0
  
+     .line 471
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
      new-instance v6, Ljava/lang/StringBuilder;
  
      invoke-direct {v6}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v7, "settings: "
  
      invoke-virtual {v6, v7}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 99,109 ****
--- 110,122 ----
  
      throw v5
  
+     .line 473
      :cond_0
      invoke-virtual {v3}, Ljava/util/StringTokenizer;->nextToken()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 474
      .local v1, key:Ljava/lang/String;
      const-string v5, "Equalizer"
  
***************
*** 113,118 ****
--- 126,132 ----
  
      if-nez v5, :cond_1
  
+     .line 475
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 137,148 ****
--- 151,164 ----
  
      throw v5
  
+     .line 479
      :cond_1
      :try_start_0
      invoke-virtual {v3}, Ljava/util/StringTokenizer;->nextToken()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 480
      const-string v5, "curPreset"
  
      invoke-virtual {v1, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
***************
*** 151,156 ****
--- 167,173 ----
  
      if-nez v5, :cond_2
  
+     .line 481
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 177,185 ****
--- 194,204 ----
      :try_end_0
      .catch Ljava/lang/NumberFormatException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 500
      :catch_0
      move-exception v2
  
+     .line 501
      .local v2, nfe:Ljava/lang/NumberFormatException;
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
***************
*** 205,210 ****
--- 224,230 ----
  
      throw v5
  
+     .line 483
      .end local v2           #nfe:Ljava/lang/NumberFormatException;
      :cond_2
      :try_start_1
***************
*** 218,228 ****
  
      iput-short v5, p0, Landroid/media/audiofx/Equalizer$Settings;->curPreset:S
  
      invoke-virtual {v3}, Ljava/util/StringTokenizer;->nextToken()Ljava/lang/String;
  
      move-result-object v1
  
!     const-string v5, "numBands"
  
      invoke-virtual {v1, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 238,250 ----
  
      iput-short v5, p0, Landroid/media/audiofx/Equalizer$Settings;->curPreset:S
  
+     .line 484
      invoke-virtual {v3}, Ljava/util/StringTokenizer;->nextToken()Ljava/lang/String;
  
      move-result-object v1
  
!     .line 485
!     const-string/jumbo v5, "numBands"
  
      invoke-virtual {v1, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 230,235 ****
--- 252,258 ----
  
      if-nez v5, :cond_3
  
+     .line 486
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 254,259 ****
--- 277,283 ----
  
      throw v5
  
+     .line 488
      :cond_3
      invoke-virtual {v3}, Ljava/util/StringTokenizer;->nextToken()Ljava/lang/String;
  
***************
*** 265,270 ****
--- 289,295 ----
  
      iput-short v5, p0, Landroid/media/audiofx/Equalizer$Settings;->numBands:S
  
+     .line 489
      invoke-virtual {v3}, Ljava/util/StringTokenizer;->countTokens()I
  
      move-result v5
***************
*** 275,287 ****
  
      if-eq v5, v6, :cond_4
  
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
      new-instance v6, Ljava/lang/StringBuilder;
  
      invoke-direct {v6}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v7, "settings: "
  
      invoke-virtual {v6, v7}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 300,313 ----
  
      if-eq v5, v6, :cond_4
  
+     .line 490
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
      new-instance v6, Ljava/lang/StringBuilder;
  
      invoke-direct {v6}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v7, "settings: "
  
      invoke-virtual {v6, v7}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 299,304 ****
--- 325,331 ----
  
      throw v5
  
+     .line 492
      :cond_4
      iget-short v5, p0, Landroid/media/audiofx/Equalizer$Settings;->numBands:S
  
***************
*** 306,311 ****
--- 333,339 ----
  
      iput-object v5, p0, Landroid/media/audiofx/Equalizer$Settings;->bandLevels:[S
  
+     .line 493
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 314,323 ****
--- 342,353 ----
  
      if-ge v0, v5, :cond_6
  
+     .line 494
      invoke-virtual {v3}, Ljava/util/StringTokenizer;->nextToken()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 495
      new-instance v5, Ljava/lang/StringBuilder;
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 350,355 ****
--- 380,386 ----
  
      if-nez v5, :cond_5
  
+     .line 496
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 374,379 ****
--- 405,411 ----
  
      throw v5
  
+     .line 498
      :cond_5
      iget-object v5, p0, Landroid/media/audiofx/Equalizer$Settings;->bandLevels:[S
  
***************
*** 389,398 ****
--- 421,432 ----
      :try_end_1
      .catch Ljava/lang/NumberFormatException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 493
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 503
      :cond_6
      return-void
  .end method
***************
*** 403,408 ****
--- 437,443 ----
      .locals 4
  
      .prologue
+     .line 508
      new-instance v1, Ljava/lang/String;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 447,452 ****
--- 482,488 ----
  
      invoke-direct {v1, v2}, Ljava/lang/String;-><init>(Ljava/lang/String;)V
  
+     .line 513
      .local v1, str:Ljava/lang/String;
      const/4 v0, 0x0
  
***************
*** 456,461 ****
--- 492,498 ----
  
      if-ge v0, v2, :cond_0
  
+     .line 514
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 498,507 ****
--- 535,546 ----
  
      move-result-object v1
  
+     .line 513
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 516
      :cond_0
      return-object v1
  .end method
