*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/audiofx/PresetReverb$BaseParameterListener.smali	2012-03-23 19:35:34.851222396 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/audiofx/PresetReverb$BaseParameterListener.smali.original	2012-03-23 19:34:44.723308316 -0500
***************
*** 27,36 ****
--- 27,38 ----
      .parameter
  
      .prologue
+     .line 191
      iput-object p1, p0, Landroid/media/audiofx/PresetReverb$BaseParameterListener;->this$0:Landroid/media/audiofx/PresetReverb;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 193
      return-void
  .end method
  
***************
*** 40,45 ****
--- 42,48 ----
      .parameter "x1"
  
      .prologue
+     .line 190
      invoke-direct {p0, p1}, Landroid/media/audiofx/PresetReverb$BaseParameterListener;-><init>(Landroid/media/audiofx/PresetReverb;)V
  
      return-void
***************
*** 59,66 ****
--- 62,71 ----
  
      const/4 v5, -0x1
  
+     .line 195
      const/4 v0, 0x0
  
+     .line 197
      .local v0, l:Landroid/media/audiofx/PresetReverb$OnParameterChangeListener;
      iget-object v3, p0, Landroid/media/audiofx/PresetReverb$BaseParameterListener;->this$0:Landroid/media/audiofx/PresetReverb;
  
***************
*** 71,76 ****
--- 76,82 ----
  
      monitor-enter v4
  
+     .line 198
      :try_start_0
      iget-object v3, p0, Landroid/media/audiofx/PresetReverb$BaseParameterListener;->this$0:Landroid/media/audiofx/PresetReverb;
  
***************
*** 81,86 ****
--- 87,93 ----
  
      if-eqz v3, :cond_0
  
+     .line 199
      iget-object v3, p0, Landroid/media/audiofx/PresetReverb$BaseParameterListener;->this$0:Landroid/media/audiofx/PresetReverb;
  
      #getter for: Landroid/media/audiofx/PresetReverb;->mParamListener:Landroid/media/audiofx/PresetReverb$OnParameterChangeListener;
***************
*** 88,105 ****
--- 95,117 ----
  
      move-result-object v0
  
+     .line 201
      :cond_0
      monitor-exit v4
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 202
      if-eqz v0, :cond_3
  
+     .line 203
      const/4 v1, -0x1
  
+     .line 204
      .local v1, p:I
      const/4 v2, -0x1
  
+     .line 206
      .local v2, v:S
      array-length v3, p3
  
***************
*** 107,118 ****
--- 119,132 ----
  
      if-ne v3, v4, :cond_1
  
+     .line 207
      iget-object v3, p0, Landroid/media/audiofx/PresetReverb$BaseParameterListener;->this$0:Landroid/media/audiofx/PresetReverb;
  
      invoke-virtual {v3, p3, v6}, Landroid/media/audiofx/PresetReverb;->byteArrayToInt([BI)I
  
      move-result v1
  
+     .line 209
      :cond_1
      array-length v3, p4
  
***************
*** 120,145 ****
--- 134,164 ----
  
      if-ne v3, v4, :cond_2
  
+     .line 210
      iget-object v3, p0, Landroid/media/audiofx/PresetReverb$BaseParameterListener;->this$0:Landroid/media/audiofx/PresetReverb;
  
      invoke-virtual {v3, p4, v6}, Landroid/media/audiofx/PresetReverb;->byteArrayToShort([BI)S
  
      move-result v2
  
+     .line 212
      :cond_2
      if-eq v1, v5, :cond_3
  
      if-eq v2, v5, :cond_3
  
+     .line 213
      iget-object v3, p0, Landroid/media/audiofx/PresetReverb$BaseParameterListener;->this$0:Landroid/media/audiofx/PresetReverb;
  
      invoke-interface {v0, v3, p2, v1, v2}, Landroid/media/audiofx/PresetReverb$OnParameterChangeListener;->onParameterChange(Landroid/media/audiofx/PresetReverb;IIS)V
  
+     .line 216
      .end local v1           #p:I
      .end local v2           #v:S
      :cond_3
      return-void
  
+     .line 201
      :catchall_0
      move-exception v3
  
