*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/audiofx/PresetReverb.smali	2012-03-23 19:35:35.498234195 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/audiofx/PresetReverb.smali.original	2012-03-23 19:34:44.728308406 -0500
***************
*** 58,79 ****
--- 58,84 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 136
      sget-object v0, Landroid/media/audiofx/PresetReverb;->EFFECT_TYPE_PRESET_REVERB:Ljava/util/UUID;
  
      sget-object v1, Landroid/media/audiofx/PresetReverb;->EFFECT_TYPE_NULL:Ljava/util/UUID;
  
      invoke-direct {p0, v0, v1, p1, p2}, Landroid/media/audiofx/AudioEffect;-><init>(Ljava/util/UUID;Ljava/util/UUID;II)V
  
+     .line 106
      iput-object v2, p0, Landroid/media/audiofx/PresetReverb;->mParamListener:Landroid/media/audiofx/PresetReverb$OnParameterChangeListener;
  
+     .line 111
      iput-object v2, p0, Landroid/media/audiofx/PresetReverb;->mBaseParamListener:Landroid/media/audiofx/PresetReverb$BaseParameterListener;
  
+     .line 116
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      iput-object v0, p0, Landroid/media/audiofx/PresetReverb;->mParamListenerLock:Ljava/lang/Object;
  
+     .line 137
      return-void
  .end method
  
***************
*** 82,87 ****
--- 87,93 ----
      .parameter "x0"
  
      .prologue
+     .line 61
      iget-object v0, p0, Landroid/media/audiofx/PresetReverb;->mParamListenerLock:Ljava/lang/Object;
  
      return-object v0
***************
*** 92,97 ****
--- 98,104 ----
      .parameter "x0"
  
      .prologue
+     .line 61
      iget-object v0, p0, Landroid/media/audiofx/PresetReverb;->mParamListener:Landroid/media/audiofx/PresetReverb$OnParameterChangeListener;
  
      return-object v0
***************
*** 112,121 ****
--- 119,130 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 164
      const/4 v1, 0x1
  
      new-array v0, v1, [S
  
+     .line 165
      .local v0, value:[S
      invoke-virtual {p0, v2, v0}, Landroid/media/audiofx/PresetReverb;->getParameter(I[S)I
  
***************
*** 123,128 ****
--- 132,138 ----
  
      invoke-virtual {p0, v1}, Landroid/media/audiofx/PresetReverb;->checkStatus(I)V
  
+     .line 166
      aget-short v1, v0, v2
  
      return v1
***************
*** 141,155 ****
--- 151,168 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 292
      new-instance v0, Landroid/media/audiofx/PresetReverb$Settings;
  
      invoke-direct {v0}, Landroid/media/audiofx/PresetReverb$Settings;-><init>()V
  
+     .line 293
      .local v0, settings:Landroid/media/audiofx/PresetReverb$Settings;
      const/4 v2, 0x1
  
      new-array v1, v2, [S
  
+     .line 294
      .local v1, value:[S
      invoke-virtual {p0, v3, v1}, Landroid/media/audiofx/PresetReverb;->getParameter(I[S)I
  
***************
*** 157,166 ****
--- 170,181 ----
  
      invoke-virtual {p0, v2}, Landroid/media/audiofx/PresetReverb;->checkStatus(I)V
  
+     .line 295
      aget-short v2, v1, v3
  
      iput-short v2, v0, Landroid/media/audiofx/PresetReverb$Settings;->preset:S
  
+     .line 296
      return-object v0
  .end method
  
***************
*** 169,185 ****
--- 184,204 ----
      .parameter "listener"
  
      .prologue
+     .line 224
      iget-object v1, p0, Landroid/media/audiofx/PresetReverb;->mParamListenerLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 225
      :try_start_0
      iget-object v0, p0, Landroid/media/audiofx/PresetReverb;->mParamListener:Landroid/media/audiofx/PresetReverb$OnParameterChangeListener;
  
      if-nez v0, :cond_0
  
+     .line 226
      iput-object p1, p0, Landroid/media/audiofx/PresetReverb;->mParamListener:Landroid/media/audiofx/PresetReverb$OnParameterChangeListener;
  
+     .line 227
      new-instance v0, Landroid/media/audiofx/PresetReverb$BaseParameterListener;
  
      const/4 v2, 0x0
***************
*** 188,202 ****
--- 207,225 ----
  
      iput-object v0, p0, Landroid/media/audiofx/PresetReverb;->mBaseParamListener:Landroid/media/audiofx/PresetReverb$BaseParameterListener;
  
+     .line 228
      iget-object v0, p0, Landroid/media/audiofx/PresetReverb;->mBaseParamListener:Landroid/media/audiofx/PresetReverb$BaseParameterListener;
  
      invoke-super {p0, v0}, Landroid/media/audiofx/AudioEffect;->setParameterListener(Landroid/media/audiofx/AudioEffect$OnParameterChangeListener;)V
  
+     .line 230
      :cond_0
      monitor-exit v1
  
+     .line 231
      return-void
  
+     .line 230
      :catchall_0
      move-exception v0
  
***************
*** 219,224 ****
--- 242,248 ----
      .end annotation
  
      .prologue
+     .line 152
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0, p1}, Landroid/media/audiofx/PresetReverb;->setParameter(IS)I
***************
*** 227,232 ****
--- 251,257 ----
  
      invoke-virtual {p0, v0}, Landroid/media/audiofx/PresetReverb;->checkStatus(I)V
  
+     .line 153
      return-void
  .end method
  
***************
*** 242,247 ****
--- 267,273 ----
      .end annotation
  
      .prologue
+     .line 309
      const/4 v0, 0x0
  
      iget-short v1, p1, Landroid/media/audiofx/PresetReverb$Settings;->preset:S
***************
*** 252,256 ****
--- 278,283 ----
  
      invoke-virtual {p0, v0}, Landroid/media/audiofx/PresetReverb;->checkStatus(I)V
  
+     .line 310
      return-void
  .end method
