*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Bitmap.smali	2012-03-23 19:35:44.989407265 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Bitmap.smali.original	2012-03-23 19:34:44.820310084 -0500
***************
*** 63,72 ****
--- 63,74 ----
      .locals 1
  
      .prologue
+     .line 69
      const/4 v0, -0x1
  
      sput v0, Landroid/graphics/Bitmap;->sDefaultDensity:I
  
+     .line 1043
      new-instance v0, Landroid/graphics/Bitmap$1;
  
      invoke-direct {v0}, Landroid/graphics/Bitmap$1;-><init>()V
***************
*** 87,98 ****
--- 89,104 ----
      .prologue
      const/4 v0, -0x1
  
+     .line 97
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 60
      iput v0, p0, Landroid/graphics/Bitmap;->mWidth:I
  
+     .line 61
      iput v0, p0, Landroid/graphics/Bitmap;->mHeight:I
  
+     .line 65
      invoke-static {}, Landroid/graphics/Bitmap;->getDefaultDensity()I
  
      move-result v0
***************
*** 101,108 ****
--- 107,116 ----
  
      iput v0, p0, Landroid/graphics/Bitmap;->mDensity:I
  
+     .line 98
      if-nez p1, :cond_0
  
+     .line 99
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "internal error: native bitmap is 0"
***************
*** 111,135 ****
--- 119,151 ----
  
      throw v0
  
+     .line 102
      :cond_0
      iput-object p2, p0, Landroid/graphics/Bitmap;->mBuffer:[B
  
+     .line 104
      iput p1, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
+     .line 105
      new-instance v0, Landroid/graphics/Bitmap$BitmapFinalizer;
  
      invoke-direct {v0, p1}, Landroid/graphics/Bitmap$BitmapFinalizer;-><init>(I)V
  
      iput-object v0, p0, Landroid/graphics/Bitmap;->mFinalizer:Landroid/graphics/Bitmap$BitmapFinalizer;
  
+     .line 107
      iput-boolean p3, p0, Landroid/graphics/Bitmap;->mIsMutable:Z
  
+     .line 108
      iput-object p4, p0, Landroid/graphics/Bitmap;->mNinePatchChunk:[B
  
+     .line 109
      if-ltz p5, :cond_1
  
+     .line 110
      iput p5, p0, Landroid/graphics/Bitmap;->mDensity:I
  
+     .line 112
      :cond_1
      return-void
  .end method
***************
*** 139,144 ****
--- 155,161 ----
      .parameter "x0"
  
      .prologue
+     .line 28
      invoke-static {p0}, Landroid/graphics/Bitmap;->nativeCreateFromParcel(Landroid/os/Parcel;)Landroid/graphics/Bitmap;
  
      move-result-object v0
***************
*** 151,156 ****
--- 168,174 ----
      .parameter "x0"
  
      .prologue
+     .line 28
      invoke-static {p0}, Landroid/graphics/Bitmap;->nativeDestructor(I)V
  
      return-void
***************
*** 162,183 ****
      .parameter "y"
  
      .prologue
      invoke-static {p1, p2}, Landroid/graphics/Bitmap;->checkXYSign(II)V
  
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getWidth()I
  
      move-result v0
  
      if-lt p1, v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "x must be < bitmap.width()"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getHeight()I
  
--- 180,205 ----
      .parameter "y"
  
      .prologue
+     .line 938
      invoke-static {p1, p2}, Landroid/graphics/Bitmap;->checkXYSign(II)V
  
+     .line 939
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getWidth()I
  
      move-result v0
  
      if-lt p1, v0, :cond_0
  
+     .line 940
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "x must be < bitmap.width()"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 942
      :cond_0
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getHeight()I
  
***************
*** 185,198 ****
  
      if-lt p2, v0, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "y must be < bitmap.height()"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_1
      return-void
  .end method
--- 207,222 ----
  
      if-lt p2, v0, :cond_1
  
+     .line 943
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "y must be < bitmap.height()"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 945
      :cond_1
      return-void
  .end method
***************
*** 208,228 ****
      .parameter "pixels"
  
      .prologue
      invoke-static {p1, p2}, Landroid/graphics/Bitmap;->checkXYSign(II)V
  
      if-gez p3, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "width must be >= 0"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      if-gez p4, :cond_1
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
      const-string v3, "height must be >= 0"
--- 232,257 ----
      .parameter "pixels"
  
      .prologue
+     .line 961
      invoke-static {p1, p2}, Landroid/graphics/Bitmap;->checkXYSign(II)V
  
+     .line 962
      if-gez p3, :cond_0
  
+     .line 963
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "width must be >= 0"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 965
      :cond_0
      if-gez p4, :cond_1
  
+     .line 966
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
      const-string v3, "height must be >= 0"
***************
*** 231,236 ****
--- 260,266 ----
  
      throw v2
  
+     .line 968
      :cond_1
      add-int v2, p1, p3
  
***************
*** 240,253 ****
  
      if-le v2, v3, :cond_2
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "x + width must be <= bitmap.width()"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_2
      add-int v2, p2, p4
  
--- 270,285 ----
  
      if-le v2, v3, :cond_2
  
+     .line 969
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "x + width must be <= bitmap.width()"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 972
      :cond_2
      add-int v2, p2, p4
  
***************
*** 257,270 ****
  
      if-le v2, v3, :cond_3
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "y + height must be <= bitmap.height()"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_3
      invoke-static {p6}, Ljava/lang/Math;->abs(I)I
  
--- 289,304 ----
  
      if-le v2, v3, :cond_3
  
+     .line 973
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "y + height must be <= bitmap.height()"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 976
      :cond_3
      invoke-static {p6}, Ljava/lang/Math;->abs(I)I
  
***************
*** 272,277 ****
--- 306,312 ----
  
      if-ge v2, p3, :cond_4
  
+     .line 977
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
      const-string v3, "abs(stride) must be >= width"
***************
*** 280,285 ****
--- 315,321 ----
  
      throw v2
  
+     .line 979
      :cond_4
      add-int/lit8 v2, p4, -0x1
  
***************
*** 287,295 ****
--- 323,333 ----
  
      add-int v0, p5, v2
  
+     .line 980
      .local v0, lastScanline:I
      array-length v1, p7
  
+     .line 981
      .local v1, length:I
      if-ltz p5, :cond_5
  
***************
*** 303,308 ****
--- 341,347 ----
  
      if-le v2, v1, :cond_6
  
+     .line 984
      :cond_5
      new-instance v2, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 310,315 ****
--- 349,355 ----
  
      throw v2
  
+     .line 986
      :cond_6
      return-void
  .end method
***************
*** 319,334 ****
--- 359,377 ----
      .parameter "errorMessage"
  
      .prologue
+     .line 212
      iget-boolean v0, p0, Landroid/graphics/Bitmap;->mRecycled:Z
  
      if-eqz v0, :cond_0
  
+     .line 213
      new-instance v0, Ljava/lang/IllegalStateException;
  
      invoke-direct {v0, p1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 215
      :cond_0
      return-void
  .end method
***************
*** 339,357 ****
      .parameter "height"
  
      .prologue
      if-gtz p0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "width must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      if-gtz p1, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "height must be > 0"
--- 382,404 ----
      .parameter "height"
  
      .prologue
+     .line 239
      if-gtz p0, :cond_0
  
+     .line 240
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "width must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 242
      :cond_0
      if-gtz p1, :cond_1
  
+     .line 243
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "height must be > 0"
***************
*** 360,365 ****
--- 407,413 ----
  
      throw v0
  
+     .line 245
      :cond_1
      return-void
  .end method
***************
*** 370,396 ****
      .parameter "y"
  
      .prologue
      if-gez p0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "x must be >= 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      if-gez p1, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "y must be >= 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_1
      return-void
  .end method
--- 418,449 ----
      .parameter "y"
  
      .prologue
+     .line 224
      if-gez p0, :cond_0
  
+     .line 225
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "x must be >= 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 227
      :cond_0
      if-gez p1, :cond_1
  
+     .line 228
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "y must be >= 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 230
      :cond_1
      return-void
  .end method
***************
*** 402,407 ****
--- 455,461 ----
      .parameter "config"
  
      .prologue
+     .line 585
      const/4 v0, 0x1
  
      invoke-static {p0, p1, p2, v0}, Landroid/graphics/Bitmap;->createBitmap(IILandroid/graphics/Bitmap$Config;Z)Landroid/graphics/Bitmap;
***************
*** 419,437 ****
      .parameter "hasAlpha"
  
      .prologue
      if-lez p0, :cond_0
  
      if-gtz p1, :cond_1
  
      :cond_0
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "width and height must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_1
      const/4 v0, 0x0
  
--- 473,494 ----
      .parameter "hasAlpha"
  
      .prologue
+     .line 602
      if-lez p0, :cond_0
  
      if-gtz p1, :cond_1
  
+     .line 603
      :cond_0
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "width and height must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 605
      :cond_1
      const/4 v0, 0x0
  
***************
*** 451,456 ****
--- 508,514 ----
  
      move-result-object v7
  
+     .line 606
      .local v7, bm:Landroid/graphics/Bitmap;
      sget-object v0, Landroid/graphics/Bitmap$Config;->ARGB_8888:Landroid/graphics/Bitmap$Config;
  
***************
*** 458,473 ****
--- 516,534 ----
  
      if-nez p3, :cond_2
  
+     .line 607
      iget v0, v7, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      const/high16 v1, -0x100
  
      invoke-static {v0, v1}, Landroid/graphics/Bitmap;->nativeErase(II)V
  
+     .line 608
      iget v0, v7, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0, p3}, Landroid/graphics/Bitmap;->nativeSetHasAlpha(IZ)V
  
+     .line 615
      :cond_2
      return-object v7
  .end method
***************
*** 479,484 ****
--- 540,546 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 455
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getWidth()I
  
      move-result v0
***************
*** 503,508 ****
--- 565,571 ----
      .parameter "height"
  
      .prologue
+     .line 471
      const/4 v5, 0x0
  
      const/4 v6, 0x0
***************
*** 535,544 ****
--- 598,610 ----
      .parameter "filter"
  
      .prologue
+     .line 495
      invoke-static/range {p1 .. p2}, Landroid/graphics/Bitmap;->checkXYSign(II)V
  
+     .line 496
      invoke-static/range {p3 .. p4}, Landroid/graphics/Bitmap;->checkWidthHeight(II)V
  
+     .line 497
      add-int v13, p1, p3
  
      invoke-virtual/range {p0 .. p0}, Landroid/graphics/Bitmap;->getWidth()I
***************
*** 547,560 ****
  
      if-le v13, v14, :cond_0
  
      new-instance v13, Ljava/lang/IllegalArgumentException;
  
!     const-string v14, "x + width must be <= bitmap.width()"
  
      invoke-direct {v13, v14}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v13
  
      :cond_0
      add-int v13, p2, p4
  
--- 613,628 ----
  
      if-le v13, v14, :cond_0
  
+     .line 498
      new-instance v13, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v14, "x + width must be <= bitmap.width()"
  
      invoke-direct {v13, v14}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v13
  
+     .line 500
      :cond_0
      add-int v13, p2, p4
  
***************
*** 564,577 ****
  
      if-le v13, v14, :cond_1
  
      new-instance v13, Ljava/lang/IllegalArgumentException;
  
!     const-string v14, "y + height must be <= bitmap.height()"
  
      invoke-direct {v13, v14}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v13
  
      :cond_1
      invoke-virtual/range {p0 .. p0}, Landroid/graphics/Bitmap;->isMutable()Z
  
--- 632,647 ----
  
      if-le v13, v14, :cond_1
  
+     .line 501
      new-instance v13, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v14, "y + height must be <= bitmap.height()"
  
      invoke-direct {v13, v14}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v13
  
+     .line 505
      :cond_1
      invoke-virtual/range {p0 .. p0}, Landroid/graphics/Bitmap;->isMutable()Z
  
***************
*** 607,629 ****
--- 677,704 ----
  
      if-eqz v13, :cond_3
  
+     .line 572
      .end local p0
      :cond_2
      :goto_0
      return-object p0
  
+     .line 510
      .restart local p0
      :cond_3
      move/from16 v9, p3
  
+     .line 511
      .local v9, neww:I
      move/from16 v8, p4
  
+     .line 512
      .local v8, newh:I
      new-instance v3, Landroid/graphics/Canvas;
  
      invoke-direct {v3}, Landroid/graphics/Canvas;-><init>()V
  
+     .line 516
      .local v3, canvas:Landroid/graphics/Canvas;
      new-instance v11, Landroid/graphics/Rect;
  
***************
*** 637,642 ****
--- 712,718 ----
  
      invoke-direct {v11, v0, v1, v13, v14}, Landroid/graphics/Rect;-><init>(IIII)V
  
+     .line 517
      .local v11, srcR:Landroid/graphics/Rect;
      new-instance v6, Landroid/graphics/RectF;
  
***************
*** 658,674 ****
--- 734,754 ----
  
      invoke-direct {v6, v13, v14, v15, v0}, Landroid/graphics/RectF;-><init>(FFFF)V
  
+     .line 519
      .local v6, dstR:Landroid/graphics/RectF;
      sget-object v7, Landroid/graphics/Bitmap$Config;->ARGB_8888:Landroid/graphics/Bitmap$Config;
  
+     .line 520
      .local v7, newConfig:Landroid/graphics/Bitmap$Config;
      invoke-virtual/range {p0 .. p0}, Landroid/graphics/Bitmap;->getConfig()Landroid/graphics/Bitmap$Config;
  
      move-result-object v4
  
+     .line 522
      .local v4, config:Landroid/graphics/Bitmap$Config;
      if-eqz v4, :cond_4
  
+     .line 523
      sget-object v13, Landroid/graphics/Bitmap$2;->$SwitchMap$android$graphics$Bitmap$Config:[I
  
      invoke-virtual {v4}, Landroid/graphics/Bitmap$Config;->ordinal()I
***************
*** 679,686 ****
--- 759,768 ----
  
      packed-switch v13, :pswitch_data_0
  
+     .line 534
      sget-object v7, Landroid/graphics/Bitmap$Config;->ARGB_8888:Landroid/graphics/Bitmap$Config;
  
+     .line 539
      :cond_4
      :goto_1
      if-eqz p5, :cond_5
***************
*** 691,696 ****
--- 773,779 ----
  
      if-eqz v13, :cond_7
  
+     .line 540
      :cond_5
      invoke-virtual/range {p0 .. p0}, Landroid/graphics/Bitmap;->hasAlpha()Z
  
***************
*** 700,708 ****
--- 783,793 ----
  
      move-result-object v2
  
+     .line 541
      .local v2, bitmap:Landroid/graphics/Bitmap;
      const/4 v10, 0x0
  
+     .line 566
      .end local v7           #newConfig:Landroid/graphics/Bitmap$Config;
      .local v10, paint:Landroid/graphics/Paint;
      :cond_6
***************
*** 713,745 ****
--- 798,839 ----
  
      iput v13, v2, Landroid/graphics/Bitmap;->mDensity:I
  
+     .line 568
      invoke-virtual {v3, v2}, Landroid/graphics/Canvas;->setBitmap(Landroid/graphics/Bitmap;)V
  
+     .line 569
      move-object/from16 v0, p0
  
      invoke-virtual {v3, v0, v11, v6, v10}, Landroid/graphics/Canvas;->drawBitmap(Landroid/graphics/Bitmap;Landroid/graphics/Rect;Landroid/graphics/RectF;Landroid/graphics/Paint;)V
  
+     .line 570
      const/4 v13, 0x0
  
      invoke-virtual {v3, v13}, Landroid/graphics/Canvas;->setBitmap(Landroid/graphics/Bitmap;)V
  
      move-object/from16 p0, v2
  
+     .line 572
      goto :goto_0
  
+     .line 525
      .end local v2           #bitmap:Landroid/graphics/Bitmap;
      .end local v10           #paint:Landroid/graphics/Paint;
      .restart local v7       #newConfig:Landroid/graphics/Bitmap$Config;
      :pswitch_0
      sget-object v7, Landroid/graphics/Bitmap$Config;->RGB_565:Landroid/graphics/Bitmap$Config;
  
+     .line 526
      goto :goto_1
  
+     .line 528
      :pswitch_1
      sget-object v7, Landroid/graphics/Bitmap$Config;->ALPHA_8:Landroid/graphics/Bitmap$Config;
  
+     .line 529
      goto :goto_1
  
+     .line 543
      :cond_7
      invoke-virtual/range {p5 .. p5}, Landroid/graphics/Matrix;->rectStaysRect()Z
  
***************
*** 749,765 ****
--- 843,862 ----
  
      const/4 v12, 0x1
  
+     .line 545
      .local v12, transformed:Z
      :goto_3
      new-instance v5, Landroid/graphics/RectF;
  
      invoke-direct {v5}, Landroid/graphics/RectF;-><init>()V
  
+     .line 546
      .local v5, deviceR:Landroid/graphics/RectF;
      move-object/from16 v0, p5
  
      invoke-virtual {v0, v5, v6}, Landroid/graphics/Matrix;->mapRect(Landroid/graphics/RectF;Landroid/graphics/RectF;)Z
  
+     .line 548
      invoke-virtual {v5}, Landroid/graphics/RectF;->width()F
  
      move-result v13
***************
*** 768,773 ****
--- 865,871 ----
  
      move-result v9
  
+     .line 549
      invoke-virtual {v5}, Landroid/graphics/RectF;->height()F
  
      move-result v13
***************
*** 776,781 ****
--- 874,880 ----
  
      move-result v8
  
+     .line 551
      if-eqz v12, :cond_8
  
      sget-object v7, Landroid/graphics/Bitmap$Config;->ARGB_8888:Landroid/graphics/Bitmap$Config;
***************
*** 798,803 ****
--- 897,903 ----
  
      move-result-object v2
  
+     .line 554
      .restart local v2       #bitmap:Landroid/graphics/Bitmap;
      iget v13, v5, Landroid/graphics/RectF;->left:F
  
***************
*** 809,835 ****
--- 909,941 ----
  
      invoke-virtual {v3, v13, v14}, Landroid/graphics/Canvas;->translate(FF)V
  
+     .line 555
      move-object/from16 v0, p5
  
      invoke-virtual {v3, v0}, Landroid/graphics/Canvas;->concat(Landroid/graphics/Matrix;)V
  
+     .line 557
      new-instance v10, Landroid/graphics/Paint;
  
      invoke-direct {v10}, Landroid/graphics/Paint;-><init>()V
  
+     .line 558
      .restart local v10       #paint:Landroid/graphics/Paint;
      move/from16 v0, p6
  
      invoke-virtual {v10, v0}, Landroid/graphics/Paint;->setFilterBitmap(Z)V
  
+     .line 559
      if-eqz v12, :cond_6
  
+     .line 560
      const/4 v13, 0x1
  
      invoke-virtual {v10, v13}, Landroid/graphics/Paint;->setAntiAlias(Z)V
  
      goto :goto_2
  
+     .line 543
      .end local v2           #bitmap:Landroid/graphics/Bitmap;
      .end local v5           #deviceR:Landroid/graphics/RectF;
      .end local v10           #paint:Landroid/graphics/Paint;
***************
*** 840,845 ****
--- 946,952 ----
  
      goto :goto_3
  
+     .line 551
      .end local v7           #newConfig:Landroid/graphics/Bitmap$Config;
      .restart local v5       #deviceR:Landroid/graphics/RectF;
      .restart local v12       #transformed:Z
***************
*** 848,853 ****
--- 955,961 ----
  
      goto :goto_4
  
+     .line 523
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 865,878 ****
--- 973,989 ----
      .parameter "config"
  
      .prologue
+     .line 639
      invoke-static {p3, p4}, Landroid/graphics/Bitmap;->checkWidthHeight(II)V
  
+     .line 640
      invoke-static {p2}, Ljava/lang/Math;->abs(I)I
  
      move-result v0
  
      if-ge v0, p3, :cond_0
  
+     .line 641
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "abs(stride) must be >= width"
***************
*** 881,886 ****
--- 992,998 ----
  
      throw v0
  
+     .line 643
      :cond_0
      add-int/lit8 v0, p4, -0x1
  
***************
*** 888,896 ****
--- 1000,1010 ----
  
      add-int v7, p1, v0
  
+     .line 644
      .local v7, lastScanline:I
      array-length v8, p0
  
+     .line 645
      .local v8, length:I
      if-ltz p1, :cond_1
  
***************
*** 904,909 ****
--- 1018,1024 ----
  
      if-le v0, v8, :cond_2
  
+     .line 647
      :cond_1
      new-instance v0, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 911,930 ****
  
      throw v0
  
      :cond_2
      if-lez p3, :cond_3
  
      if-gtz p4, :cond_4
  
      :cond_3
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "width and height must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_4
      iget v5, p5, Landroid/graphics/Bitmap$Config;->nativeInt:I
  
--- 1026,1048 ----
  
      throw v0
  
+     .line 649
      :cond_2
      if-lez p3, :cond_3
  
      if-gtz p4, :cond_4
  
+     .line 650
      :cond_3
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "width and height must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 652
      :cond_4
      iget v5, p5, Landroid/graphics/Bitmap$Config;->nativeInt:I
  
***************
*** 955,960 ****
--- 1073,1079 ----
      .parameter "config"
  
      .prologue
+     .line 672
      const/4 v1, 0x0
  
      move-object v0, p0
***************
*** 984,1023 ****
--- 1103,1151 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 422
      const-class v2, Landroid/graphics/Bitmap;
  
      monitor-enter v2
  
+     .line 424
      :try_start_0
      sget-object v5, Landroid/graphics/Bitmap;->sScaleMatrix:Landroid/graphics/Matrix;
  
+     .line 425
      .local v5, m:Landroid/graphics/Matrix;
      const/4 v0, 0x0
  
      sput-object v0, Landroid/graphics/Bitmap;->sScaleMatrix:Landroid/graphics/Matrix;
  
+     .line 426
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 428
      if-nez v5, :cond_0
  
+     .line 429
      new-instance v5, Landroid/graphics/Matrix;
  
      .end local v5           #m:Landroid/graphics/Matrix;
      invoke-direct {v5}, Landroid/graphics/Matrix;-><init>()V
  
+     .line 432
      .restart local v5       #m:Landroid/graphics/Matrix;
      :cond_0
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getWidth()I
  
      move-result v3
  
+     .line 433
      .local v3, width:I
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getHeight()I
  
      move-result v4
  
+     .line 434
      .local v4, height:I
      int-to-float v0, p1
  
***************
*** 1025,1030 ****
--- 1153,1159 ----
  
      div-float v8, v0, v2
  
+     .line 435
      .local v8, sx:F
      int-to-float v0, p2
  
***************
*** 1032,1037 ****
--- 1161,1167 ----
  
      div-float v9, v0, v2
  
+     .line 436
      .local v9, sy:F
      invoke-virtual {v5, v8, v9}, Landroid/graphics/Matrix;->setScale(FF)V
  
***************
*** 1041,1069 ****
--- 1171,1206 ----
  
      move v6, p3
  
+     .line 437
      invoke-static/range {v0 .. v6}, Landroid/graphics/Bitmap;->createBitmap(Landroid/graphics/Bitmap;IIIILandroid/graphics/Matrix;Z)Landroid/graphics/Bitmap;
  
      move-result-object v7
  
+     .line 439
      .local v7, b:Landroid/graphics/Bitmap;
      const-class v1, Landroid/graphics/Bitmap;
  
      monitor-enter v1
  
+     .line 441
      :try_start_1
      sget-object v0, Landroid/graphics/Bitmap;->sScaleMatrix:Landroid/graphics/Matrix;
  
      if-nez v0, :cond_1
  
+     .line 442
      sput-object v5, Landroid/graphics/Bitmap;->sScaleMatrix:Landroid/graphics/Matrix;
  
+     .line 444
      :cond_1
      monitor-exit v1
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_1
  
+     .line 446
      return-object v7
  
+     .line 426
      .end local v3           #width:I
      .end local v4           #height:I
      .end local v5           #m:Landroid/graphics/Matrix;
***************
*** 1080,1085 ****
--- 1217,1223 ----
  
      throw v0
  
+     .line 444
      .restart local v3       #width:I
      .restart local v4       #height:I
      .restart local v5       #m:Landroid/graphics/Matrix;
***************
*** 1101,1120 ****
--- 1239,1263 ----
      .locals 1
  
      .prologue
+     .line 81
      sget v0, Landroid/graphics/Bitmap;->sDefaultDensity:I
  
      if-ltz v0, :cond_0
  
+     .line 82
      sget v0, Landroid/graphics/Bitmap;->sDefaultDensity:I
  
+     .line 85
      :goto_0
      return v0
  
+     .line 84
      :cond_0
      sget v0, Landroid/util/DisplayMetrics;->DENSITY_DEVICE:I
  
      sput v0, Landroid/graphics/Bitmap;->sDefaultDensity:I
  
+     .line 85
      sget v0, Landroid/graphics/Bitmap;->sDefaultDensity:I
  
      goto :goto_0
***************
*** 1199,1208 ****
--- 1342,1353 ----
      .parameter "tdensity"
  
      .prologue
+     .line 808
      if-eqz p1, :cond_0
  
      if-ne p1, p2, :cond_1
  
+     .line 813
      .end local p0
      :cond_0
      :goto_0
***************
*** 1226,1233 ****
--- 1371,1380 ----
      .parameter "density"
  
      .prologue
+     .line 77
      sput p0, Landroid/graphics/Bitmap;->sDefaultDensity:I
  
+     .line 78
      return-void
  .end method
  
***************
*** 1240,1257 ****
--- 1387,1408 ----
      .parameter "stream"
  
      .prologue
+     .line 721
      const-string v0, "Can\'t compress a recycled bitmap"
  
      invoke-direct {p0, v0}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 723
      if-nez p3, :cond_0
  
+     .line 724
      new-instance v0, Ljava/lang/NullPointerException;
  
      invoke-direct {v0}, Ljava/lang/NullPointerException;-><init>()V
  
      throw v0
  
+     .line 726
      :cond_0
      if-ltz p2, :cond_1
  
***************
*** 1259,1273 ****
  
      if-le p2, v0, :cond_2
  
      :cond_1
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "quality must be 0..100"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_2
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
--- 1410,1426 ----
  
      if-le p2, v0, :cond_2
  
+     .line 727
      :cond_1
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "quality must be 0..100"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 729
      :cond_2
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
***************
*** 1290,1299 ****
--- 1443,1454 ----
      .parameter "isMutable"
  
      .prologue
+     .line 402
      const-string v1, "Can\'t copy a recycled bitmap"
  
      invoke-direct {p0, v1}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 403
      iget v1, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      iget v2, p1, Landroid/graphics/Bitmap$Config;->nativeInt:I
***************
*** 1302,1314 ****
--- 1457,1472 ----
  
      move-result-object v0
  
+     .line 404
      .local v0, b:Landroid/graphics/Bitmap;
      if-eqz v0, :cond_0
  
+     .line 405
      iget v1, p0, Landroid/graphics/Bitmap;->mDensity:I
  
      iput v1, v0, Landroid/graphics/Bitmap;->mDensity:I
  
+     .line 407
      :cond_0
      return-object v0
  .end method
***************
*** 1318,1344 ****
--- 1476,1508 ----
      .parameter "src"
  
      .prologue
+     .line 365
      const-string v6, "copyPixelsFromBuffer called on recycled bitmap"
  
      invoke-direct {p0, v6}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 367
      invoke-virtual {p1}, Ljava/nio/Buffer;->remaining()I
  
      move-result v4
  
+     .line 369
      .local v4, elements:I
      instance-of v6, p1, Ljava/nio/ByteBuffer;
  
      if-eqz v6, :cond_0
  
+     .line 370
      const/4 v5, 0x0
  
+     .line 379
      .local v5, shift:I
      :goto_0
      int-to-long v6, v4
  
      shl-long v2, v6, v5
  
+     .line 380
      .local v2, bufferBytes:J
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getByteCount()I
  
***************
*** 1346,1356 ****
--- 1510,1522 ----
  
      int-to-long v0, v6
  
+     .line 382
      .local v0, bitmapBytes:J
      cmp-long v6, v2, v0
  
      if-gez v6, :cond_3
  
+     .line 383
      new-instance v6, Ljava/lang/RuntimeException;
  
      const-string v7, "Buffer not large enough for pixels"
***************
*** 1359,1364 ****
--- 1525,1531 ----
  
      throw v6
  
+     .line 371
      .end local v0           #bitmapBytes:J
      .end local v2           #bufferBytes:J
      .end local v5           #shift:I
***************
*** 1367,1398 ****
  
      if-eqz v6, :cond_1
  
      const/4 v5, 0x1
  
      .restart local v5       #shift:I
      goto :goto_0
  
      .end local v5           #shift:I
      :cond_1
      instance-of v6, p1, Ljava/nio/IntBuffer;
  
      if-eqz v6, :cond_2
  
      const/4 v5, 0x2
  
      .restart local v5       #shift:I
      goto :goto_0
  
      .end local v5           #shift:I
      :cond_2
      new-instance v6, Ljava/lang/RuntimeException;
  
!     const-string v7, "unsupported Buffer subclass"
  
      invoke-direct {v6, v7}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v6
  
      .restart local v0       #bitmapBytes:J
      .restart local v2       #bufferBytes:J
      .restart local v5       #shift:I
--- 1534,1570 ----
  
      if-eqz v6, :cond_1
  
+     .line 372
      const/4 v5, 0x1
  
      .restart local v5       #shift:I
      goto :goto_0
  
+     .line 373
      .end local v5           #shift:I
      :cond_1
      instance-of v6, p1, Ljava/nio/IntBuffer;
  
      if-eqz v6, :cond_2
  
+     .line 374
      const/4 v5, 0x2
  
      .restart local v5       #shift:I
      goto :goto_0
  
+     .line 376
      .end local v5           #shift:I
      :cond_2
      new-instance v6, Ljava/lang/RuntimeException;
  
!     const-string/jumbo v7, "unsupported Buffer subclass"
  
      invoke-direct {v6, v7}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v6
  
+     .line 386
      .restart local v0       #bitmapBytes:J
      .restart local v2       #bufferBytes:J
      .restart local v5       #shift:I
***************
*** 1401,1406 ****
--- 1573,1579 ----
  
      invoke-static {v6, p1}, Landroid/graphics/Bitmap;->nativeCopyPixelsFromBuffer(ILjava/nio/Buffer;)V
  
+     .line 387
      return-void
  .end method
  
***************
*** 1409,1431 ****
--- 1582,1609 ----
      .parameter "dst"
  
      .prologue
+     .line 331
      invoke-virtual {p1}, Ljava/nio/Buffer;->remaining()I
  
      move-result v2
  
+     .line 333
      .local v2, elements:I
      instance-of v7, p1, Ljava/nio/ByteBuffer;
  
      if-eqz v7, :cond_0
  
+     .line 334
      const/4 v6, 0x0
  
+     .line 343
      .local v6, shift:I
      :goto_0
      int-to-long v7, v2
  
      shl-long v0, v7, v6
  
+     .line 344
      .local v0, bufferSize:J
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getByteCount()I
  
***************
*** 1433,1443 ****
--- 1611,1623 ----
  
      int-to-long v3, v7
  
+     .line 346
      .local v3, pixelSize:J
      cmp-long v7, v0, v3
  
      if-gez v7, :cond_3
  
+     .line 347
      new-instance v7, Ljava/lang/RuntimeException;
  
      const-string v8, "Buffer not large enough for pixels"
***************
*** 1446,1451 ****
--- 1626,1632 ----
  
      throw v7
  
+     .line 335
      .end local v0           #bufferSize:J
      .end local v3           #pixelSize:J
      .end local v6           #shift:I
***************
*** 1454,1485 ****
  
      if-eqz v7, :cond_1
  
      const/4 v6, 0x1
  
      .restart local v6       #shift:I
      goto :goto_0
  
      .end local v6           #shift:I
      :cond_1
      instance-of v7, p1, Ljava/nio/IntBuffer;
  
      if-eqz v7, :cond_2
  
      const/4 v6, 0x2
  
      .restart local v6       #shift:I
      goto :goto_0
  
      .end local v6           #shift:I
      :cond_2
      new-instance v7, Ljava/lang/RuntimeException;
  
!     const-string v8, "unsupported Buffer subclass"
  
      invoke-direct {v7, v8}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v7
  
      .restart local v0       #bufferSize:J
      .restart local v3       #pixelSize:J
      .restart local v6       #shift:I
--- 1635,1671 ----
  
      if-eqz v7, :cond_1
  
+     .line 336
      const/4 v6, 0x1
  
      .restart local v6       #shift:I
      goto :goto_0
  
+     .line 337
      .end local v6           #shift:I
      :cond_1
      instance-of v7, p1, Ljava/nio/IntBuffer;
  
      if-eqz v7, :cond_2
  
+     .line 338
      const/4 v6, 0x2
  
      .restart local v6       #shift:I
      goto :goto_0
  
+     .line 340
      .end local v6           #shift:I
      :cond_2
      new-instance v7, Ljava/lang/RuntimeException;
  
!     const-string/jumbo v8, "unsupported Buffer subclass"
  
      invoke-direct {v7, v8}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v7
  
+     .line 350
      .restart local v0       #bufferSize:J
      .restart local v3       #pixelSize:J
      .restart local v6       #shift:I
***************
*** 1488,1497 ****
--- 1674,1685 ----
  
      invoke-static {v7, p1}, Landroid/graphics/Bitmap;->nativeCopyPixelsToBuffer(ILjava/nio/Buffer;)V
  
+     .line 353
      invoke-virtual {p1}, Ljava/nio/Buffer;->position()I
  
      move-result v5
  
+     .line 354
      .local v5, position:I
      int-to-long v7, v5
  
***************
*** 1501,1508 ****
--- 1689,1698 ----
  
      long-to-int v5, v7
  
+     .line 355
      invoke-virtual {p1, v5}, Ljava/nio/Buffer;->position(I)Ljava/nio/Buffer;
  
+     .line 356
      return-void
  .end method
  
***************
*** 1510,1515 ****
--- 1700,1706 ----
      .locals 1
  
      .prologue
+     .line 1067
      const/4 v0, 0x0
  
      return v0
***************
*** 1520,1535 ****
--- 1711,1729 ----
      .parameter "c"
  
      .prologue
+     .line 876
      const-string v0, "Can\'t erase a recycled bitmap"
  
      invoke-direct {p0, v0}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 877
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->isMutable()Z
  
      move-result v0
  
      if-nez v0, :cond_0
  
+     .line 878
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "cannot erase immutable bitmaps"
***************
*** 1538,1548 ****
--- 1732,1744 ----
  
      throw v0
  
+     .line 880
      :cond_0
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0, p1}, Landroid/graphics/Bitmap;->nativeErase(II)V
  
+     .line 881
      return-void
  .end method
  
***************
*** 1552,1557 ****
--- 1748,1754 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 1090
      invoke-virtual {p0, v0, v0}, Landroid/graphics/Bitmap;->extractAlpha(Landroid/graphics/Paint;[I)Landroid/graphics/Bitmap;
  
      move-result-object v0
***************
*** 1565,1578 ****
--- 1762,1778 ----
      .parameter "offsetXY"
  
      .prologue
+     .line 1119
      const-string v2, "Can\'t extractAlpha on a recycled bitmap"
  
      invoke-direct {p0, v2}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 1120
      if-eqz p1, :cond_0
  
      iget v1, p1, Landroid/graphics/Paint;->mNativePaint:I
  
+     .line 1121
      .local v1, nativePaint:I
      :goto_0
      iget v2, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
***************
*** 1581,1589 ****
--- 1781,1791 ----
  
      move-result-object v0
  
+     .line 1122
      .local v0, bm:Landroid/graphics/Bitmap;
      if-nez v0, :cond_1
  
+     .line 1123
      new-instance v2, Ljava/lang/RuntimeException;
  
      const-string v3, "Failed to extractAlpha on Bitmap"
***************
*** 1592,1597 ****
--- 1794,1800 ----
  
      throw v2
  
+     .line 1120
      .end local v0           #bm:Landroid/graphics/Bitmap;
      .end local v1           #nativePaint:I
      :cond_0
***************
*** 1599,1604 ****
--- 1802,1808 ----
  
      goto :goto_0
  
+     .line 1125
      .restart local v0       #bm:Landroid/graphics/Bitmap;
      .restart local v1       #nativePaint:I
      :cond_1
***************
*** 1606,1611 ****
--- 1810,1816 ----
  
      iput v2, v0, Landroid/graphics/Bitmap;->mDensity:I
  
+     .line 1126
      return-object v0
  .end method
  
***************
*** 1613,1618 ****
--- 1818,1824 ----
      .locals 2
  
      .prologue
+     .line 833
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getRowBytes()I
  
      move-result v0
***************
*** 1630,1635 ****
--- 1836,1842 ----
      .locals 1
  
      .prologue
+     .line 841
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0}, Landroid/graphics/Bitmap;->nativeConfig(I)I
***************
*** 1647,1652 ****
--- 1854,1860 ----
      .locals 1
  
      .prologue
+     .line 135
      iget v0, p0, Landroid/graphics/Bitmap;->mDensity:I
  
      return v0
***************
*** 1656,1661 ****
--- 1864,1870 ----
      .locals 1
  
      .prologue
+     .line 204
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0}, Landroid/graphics/Bitmap;->nativeGenerationId(I)I
***************
*** 1669,1674 ****
--- 1878,1884 ----
      .locals 2
  
      .prologue
+     .line 747
      iget v0, p0, Landroid/graphics/Bitmap;->mHeight:I
  
      const/4 v1, -0x1
***************
*** 1696,1701 ****
--- 1906,1912 ----
      .locals 1
  
      .prologue
+     .line 680
      iget-object v0, p0, Landroid/graphics/Bitmap;->mNinePatchChunk:[B
  
      return-object v0
***************
*** 1707,1718 ****
--- 1918,1932 ----
      .parameter "y"
  
      .prologue
+     .line 894
      const-string v0, "Can\'t call getPixel() on a recycled bitmap"
  
      invoke-direct {p0, v0}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 895
      invoke-direct {p0, p1, p2}, Landroid/graphics/Bitmap;->checkPixelAccess(II)V
  
+     .line 896
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0, p1, p2}, Landroid/graphics/Bitmap;->nativeGetPixel(III)I
***************
*** 1733,1746 ****
--- 1947,1963 ----
      .parameter "height"
  
      .prologue
+     .line 922
      const-string v0, "Can\'t call getPixels() on a recycled bitmap"
  
      invoke-direct {p0, v0}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 923
      if-eqz p6, :cond_0
  
      if-nez p7, :cond_1
  
+     .line 929
      :cond_0
      :goto_0
      return-void
***************
*** 1762,1769 ****
--- 1979,1988 ----
  
      move-object v7, p1
  
+     .line 926
      invoke-direct/range {v0 .. v7}, Landroid/graphics/Bitmap;->checkPixelsAccess(IIIIII[I)V
  
+     .line 927
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      move-object v1, p1
***************
*** 1789,1794 ****
--- 2008,2014 ----
      .locals 1
  
      .prologue
+     .line 825
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0}, Landroid/graphics/Bitmap;->nativeRowBytes(I)I
***************
*** 1803,1808 ****
--- 2023,2029 ----
      .parameter "targetDensity"
  
      .prologue
+     .line 801
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getHeight()I
  
      move-result v0
***************
*** 1821,1826 ****
--- 2042,2048 ----
      .parameter "canvas"
  
      .prologue
+     .line 763
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getHeight()I
  
      move-result v0
***************
*** 1841,1846 ****
--- 2063,2069 ----
      .parameter "metrics"
  
      .prologue
+     .line 779
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getHeight()I
  
      move-result v0
***************
*** 1861,1866 ****
--- 2084,2090 ----
      .parameter "targetDensity"
  
      .prologue
+     .line 790
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getWidth()I
  
      move-result v0
***************
*** 1879,1884 ****
--- 2103,2109 ----
      .parameter "canvas"
  
      .prologue
+     .line 755
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getWidth()I
  
      move-result v0
***************
*** 1899,1904 ****
--- 2124,2130 ----
      .parameter "metrics"
  
      .prologue
+     .line 771
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->getWidth()I
  
      move-result v0
***************
*** 1918,1923 ****
--- 2144,2150 ----
      .locals 2
  
      .prologue
+     .line 742
      iget v0, p0, Landroid/graphics/Bitmap;->mWidth:I
  
      const/4 v1, -0x1
***************
*** 1945,1950 ****
--- 2172,2178 ----
      .locals 1
  
      .prologue
+     .line 853
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0}, Landroid/graphics/Bitmap;->nativeHasAlpha(I)Z
***************
*** 1958,1963 ****
--- 2186,2192 ----
      .locals 1
  
      .prologue
+     .line 737
      iget-boolean v0, p0, Landroid/graphics/Bitmap;->mIsMutable:Z
  
      return v0
***************
*** 1967,1972 ****
--- 2196,2202 ----
      .locals 1
  
      .prologue
+     .line 193
      iget-boolean v0, p0, Landroid/graphics/Bitmap;->mRecycled:Z
  
      return v0
***************
*** 1976,1981 ****
--- 2206,2212 ----
      .locals 1
  
      .prologue
+     .line 1223
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      return v0
***************
*** 1985,1994 ****
--- 2216,2227 ----
      .locals 1
  
      .prologue
+     .line 1150
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0}, Landroid/graphics/Bitmap;->nativePrepareToDraw(I)V
  
+     .line 1151
      return-void
  .end method
  
***************
*** 1998,2019 ****
--- 2231,2258 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 178
      iget-boolean v0, p0, Landroid/graphics/Bitmap;->mRecycled:Z
  
      if-nez v0, :cond_0
  
+     .line 179
      iput-object v1, p0, Landroid/graphics/Bitmap;->mBuffer:[B
  
+     .line 180
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0}, Landroid/graphics/Bitmap;->nativeRecycle(I)V
  
+     .line 181
      iput-object v1, p0, Landroid/graphics/Bitmap;->mNinePatchChunk:[B
  
+     .line 182
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/graphics/Bitmap;->mRecycled:Z
  
+     .line 184
      :cond_0
      return-void
  .end method
***************
*** 2023,2028 ****
--- 2262,2268 ----
      .parameter "other"
  
      .prologue
+     .line 1135
      if-eq p0, p1, :cond_0
  
      if-eqz p1, :cond_1
***************
*** 2054,2061 ****
--- 2294,2303 ----
      .parameter "density"
  
      .prologue
+     .line 152
      iput p1, p0, Landroid/graphics/Bitmap;->mDensity:I
  
+     .line 153
      return-void
  .end method
  
***************
*** 2064,2073 ****
--- 2306,2317 ----
      .parameter "hasAlpha"
  
      .prologue
+     .line 867
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0, p1}, Landroid/graphics/Bitmap;->nativeSetHasAlpha(IZ)V
  
+     .line 868
      return-void
  .end method
  
***************
*** 2076,2083 ****
--- 2320,2329 ----
      .parameter "chunk"
  
      .prologue
+     .line 163
      iput-object p1, p0, Landroid/graphics/Bitmap;->mNinePatchChunk:[B
  
+     .line 164
      return-void
  .end method
  
***************
*** 2088,2116 ****
--- 2334,2368 ----
      .parameter "color"
  
      .prologue
+     .line 1000
      const-string v0, "Can\'t call setPixel() on a recycled bitmap"
  
      invoke-direct {p0, v0}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 1001
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->isMutable()Z
  
      move-result v0
  
      if-nez v0, :cond_0
  
+     .line 1002
      new-instance v0, Ljava/lang/IllegalStateException;
  
      invoke-direct {v0}, Ljava/lang/IllegalStateException;-><init>()V
  
      throw v0
  
+     .line 1004
      :cond_0
      invoke-direct {p0, p1, p2}, Landroid/graphics/Bitmap;->checkPixelAccess(II)V
  
+     .line 1005
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      invoke-static {v0, p1, p2, p3}, Landroid/graphics/Bitmap;->nativeSetPixel(IIII)V
  
+     .line 1006
      return-void
  .end method
  
***************
*** 2125,2151 ****
--- 2377,2408 ----
      .parameter "height"
  
      .prologue
+     .line 1031
      const-string v0, "Can\'t call setPixels() on a recycled bitmap"
  
      invoke-direct {p0, v0}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 1032
      invoke-virtual {p0}, Landroid/graphics/Bitmap;->isMutable()Z
  
      move-result v0
  
      if-nez v0, :cond_0
  
+     .line 1033
      new-instance v0, Ljava/lang/IllegalStateException;
  
      invoke-direct {v0}, Ljava/lang/IllegalStateException;-><init>()V
  
      throw v0
  
+     .line 1035
      :cond_0
      if-eqz p6, :cond_1
  
      if-nez p7, :cond_2
  
+     .line 1041
      :cond_1
      :goto_0
      return-void
***************
*** 2167,2174 ****
--- 2424,2433 ----
  
      move-object v7, p1
  
+     .line 1038
      invoke-direct/range {v0 .. v7}, Landroid/graphics/Bitmap;->checkPixelsAccess(IIIIII[I)V
  
+     .line 1039
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      move-object v1, p1
***************
*** 2196,2205 ****
--- 2455,2466 ----
      .parameter "flags"
  
      .prologue
+     .line 1076
      const-string v0, "Can\'t parcel a recycled bitmap"
  
      invoke-direct {p0, v0}, Landroid/graphics/Bitmap;->checkRecycled(Ljava/lang/String;)V
  
+     .line 1077
      iget v0, p0, Landroid/graphics/Bitmap;->mNativeBitmap:I
  
      iget-boolean v1, p0, Landroid/graphics/Bitmap;->mIsMutable:Z
***************
*** 2212,2225 ****
  
      if-nez v0, :cond_0
  
      new-instance v0, Ljava/lang/RuntimeException;
  
!     const-string v1, "native writeToParcel failed"
  
      invoke-direct {v0, v1}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      return-void
  .end method
--- 2473,2488 ----
  
      if-nez v0, :cond_0
  
+     .line 1078
      new-instance v0, Ljava/lang/RuntimeException;
  
!     const-string/jumbo v1, "native writeToParcel failed"
  
      invoke-direct {v0, v1}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 1080
      :cond_0
      return-void
  .end method
