*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/BitmapRegionDecoder.smali	2012-03-23 19:35:45.497416529 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/BitmapRegionDecoder.smali.original	2012-03-23 19:34:44.824310158 -0500
***************
*** 15,28 ****
--- 15,32 ----
      .parameter "decoder"
  
      .prologue
+     .line 167
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 168
      iput p1, p0, Landroid/graphics/BitmapRegionDecoder;->mNativeBitmapRegionDecoder:I
  
+     .line 169
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/graphics/BitmapRegionDecoder;->mRecycled:Z
  
+     .line 170
      return-void
  .end method
  
***************
*** 31,46 ****
--- 35,53 ----
      .parameter "errorMessage"
  
      .prologue
+     .line 234
      iget-boolean v0, p0, Landroid/graphics/BitmapRegionDecoder;->mRecycled:Z
  
      if-eqz v0, :cond_0
  
+     .line 235
      new-instance v0, Ljava/lang/IllegalStateException;
  
      invoke-direct {v0, p1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 237
      :cond_0
      return-void
  .end method
***************
*** 80,85 ****
--- 87,93 ----
      .end annotation
  
      .prologue
+     .line 85
      invoke-static {p0, p1}, Landroid/graphics/BitmapRegionDecoder;->nativeNewInstance(Ljava/io/FileDescriptor;Z)Landroid/graphics/BitmapRegionDecoder;
  
      move-result-object v0
***************
*** 100,111 ****
--- 108,121 ----
      .prologue
      const/16 v3, 0x4000
  
+     .line 110
      invoke-virtual {p0}, Ljava/io/InputStream;->markSupported()Z
  
      move-result v2
  
      if-nez v2, :cond_0
  
+     .line 111
      new-instance v0, Ljava/io/BufferedInputStream;
  
      invoke-direct {v0, p0, v3}, Ljava/io/BufferedInputStream;-><init>(Ljava/io/InputStream;I)V
***************
*** 114,119 ****
--- 124,130 ----
      .local v0, is:Ljava/io/InputStream;
      move-object p0, v0
  
+     .line 114
      .end local v0           #is:Ljava/io/InputStream;
      .restart local p0
      :cond_0
***************
*** 121,126 ****
--- 132,138 ----
  
      if-eqz v2, :cond_1
  
+     .line 115
      check-cast p0, Landroid/content/res/AssetManager$AssetInputStream;
  
      .end local p0
***************
*** 132,144 ****
--- 144,159 ----
  
      move-result-object v2
  
+     .line 123
      :goto_0
      return-object v2
  
+     .line 122
      .restart local p0
      :cond_1
      new-array v1, v3, [B
  
+     .line 123
      .local v1, tempStorage:[B
      invoke-static {p0, v1, p1}, Landroid/graphics/BitmapRegionDecoder;->nativeNewInstance(Ljava/io/InputStream;[BZ)Landroid/graphics/BitmapRegionDecoder;
  
***************
*** 158,168 ****
--- 173,186 ----
      .end annotation
  
      .prologue
+     .line 144
      const/4 v0, 0x0
  
+     .line 145
      .local v0, decoder:Landroid/graphics/BitmapRegionDecoder;
      const/4 v1, 0x0
  
+     .line 148
      .local v1, stream:Ljava/io/InputStream;
      :try_start_0
      new-instance v2, Ljava/io/FileInputStream;
***************
*** 171,176 ****
--- 189,195 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 149
      .end local v1           #stream:Ljava/io/InputStream;
      .local v2, stream:Ljava/io/InputStream;
      :try_start_1
***************
*** 180,196 ****
--- 199,219 ----
  
      move-result-object v0
  
+     .line 151
      if-eqz v2, :cond_0
  
+     .line 153
      :try_start_2
      invoke-virtual {v2}, Ljava/io/InputStream;->close()V
      :try_end_2
      .catch Ljava/io/IOException; {:try_start_2 .. :try_end_2} :catch_0
  
+     .line 159
      :cond_0
      :goto_0
      return-object v0
  
+     .line 151
      .end local v2           #stream:Ljava/io/InputStream;
      .restart local v1       #stream:Ljava/io/InputStream;
      :catchall_0
***************
*** 199,213 ****
--- 222,239 ----
      :goto_1
      if-eqz v1, :cond_1
  
+     .line 153
      :try_start_3
      invoke-virtual {v1}, Ljava/io/InputStream;->close()V
      :try_end_3
      .catch Ljava/io/IOException; {:try_start_3 .. :try_end_3} :catch_1
  
+     .line 156
      :cond_1
      :goto_2
      throw v3
  
+     .line 154
      .end local v1           #stream:Ljava/io/InputStream;
      .restart local v2       #stream:Ljava/io/InputStream;
      :catch_0
***************
*** 222,227 ****
--- 248,254 ----
  
      goto :goto_2
  
+     .line 151
      .end local v1           #stream:Ljava/io/InputStream;
      .restart local v2       #stream:Ljava/io/InputStream;
      :catchall_1
***************
*** 247,252 ****
--- 274,280 ----
      .end annotation
  
      .prologue
+     .line 60
      or-int v0, p1, p2
  
      if-ltz v0, :cond_0
***************
*** 257,262 ****
--- 285,291 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 61
      :cond_0
      new-instance v0, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 264,269 ****
--- 293,299 ----
  
      throw v0
  
+     .line 63
      :cond_1
      invoke-static {p0, p1, p2, p3}, Landroid/graphics/BitmapRegionDecoder;->nativeNewInstance([BIIZ)Landroid/graphics/BitmapRegionDecoder;
  
***************
*** 280,289 ****
--- 310,321 ----
      .parameter "options"
  
      .prologue
+     .line 182
      const-string v0, "decodeRegion called on recycled region decoder"
  
      invoke-direct {p0, v0}, Landroid/graphics/BitmapRegionDecoder;->checkRecycled(Ljava/lang/String;)V
  
+     .line 183
      iget v0, p1, Landroid/graphics/Rect;->left:I
  
      if-ltz v0, :cond_0
***************
*** 308,322 ****
  
      if-le v0, v1, :cond_1
  
      :cond_0
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "rectangle is not inside the image"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_1
      iget v0, p0, Landroid/graphics/BitmapRegionDecoder;->mNativeBitmapRegionDecoder:I
  
--- 340,356 ----
  
      if-le v0, v1, :cond_1
  
+     .line 185
      :cond_0
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "rectangle is not inside the image"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 186
      :cond_1
      iget v0, p0, Landroid/graphics/BitmapRegionDecoder;->mNativeBitmapRegionDecoder:I
  
***************
*** 354,368 ****
--- 388,406 ----
      .end annotation
  
      .prologue
+     .line 242
      :try_start_0
      invoke-virtual {p0}, Landroid/graphics/BitmapRegionDecoder;->recycle()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 244
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 246
      return-void
  
+     .line 244
      :catchall_0
      move-exception v0
  
***************
*** 375,384 ****
--- 413,424 ----
      .locals 1
  
      .prologue
+     .line 198
      const-string v0, "getHeight called on recycled region decoder"
  
      invoke-direct {p0, v0}, Landroid/graphics/BitmapRegionDecoder;->checkRecycled(Ljava/lang/String;)V
  
+     .line 199
      iget v0, p0, Landroid/graphics/BitmapRegionDecoder;->mNativeBitmapRegionDecoder:I
  
      invoke-static {v0}, Landroid/graphics/BitmapRegionDecoder;->nativeGetHeight(I)I
***************
*** 392,401 ****
--- 432,443 ----
      .locals 1
  
      .prologue
+     .line 192
      const-string v0, "getWidth called on recycled region decoder"
  
      invoke-direct {p0, v0}, Landroid/graphics/BitmapRegionDecoder;->checkRecycled(Ljava/lang/String;)V
  
+     .line 193
      iget v0, p0, Landroid/graphics/BitmapRegionDecoder;->mNativeBitmapRegionDecoder:I
  
      invoke-static {v0}, Landroid/graphics/BitmapRegionDecoder;->nativeGetWidth(I)I
***************
*** 409,414 ****
--- 451,457 ----
      .locals 1
  
      .prologue
+     .line 226
      iget-boolean v0, p0, Landroid/graphics/BitmapRegionDecoder;->mRecycled:Z
  
      return v0
***************
*** 418,435 ****
--- 461,482 ----
      .locals 1
  
      .prologue
+     .line 213
      iget-boolean v0, p0, Landroid/graphics/BitmapRegionDecoder;->mRecycled:Z
  
      if-nez v0, :cond_0
  
+     .line 214
      iget v0, p0, Landroid/graphics/BitmapRegionDecoder;->mNativeBitmapRegionDecoder:I
  
      invoke-static {v0}, Landroid/graphics/BitmapRegionDecoder;->nativeClean(I)V
  
+     .line 215
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/graphics/BitmapRegionDecoder;->mRecycled:Z
  
+     .line 217
      :cond_0
      return-void
  .end method
