*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Color.smali	2012-03-23 19:35:44.610400355 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Color.smali.original	2012-03-23 19:34:44.817310030 -0500
***************
*** 46,57 ****
--- 46,59 ----
      .locals 3
  
      .prologue
+     .line 372
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      sput-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
+     .line 373
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
      const-string v1, "black"
***************
*** 64,69 ****
--- 66,72 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 374
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
      const-string v1, "darkgray"
***************
*** 76,81 ****
--- 79,85 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 375
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
      const-string v1, "gray"
***************
*** 88,93 ****
--- 92,98 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 376
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
      const-string v1, "lightgray"
***************
*** 100,108 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
!     const-string v1, "white"
  
      const/4 v2, -0x1
  
--- 105,114 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 377
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
!     const-string/jumbo v1, "white"
  
      const/4 v2, -0x1
  
***************
*** 112,120 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
!     const-string v1, "red"
  
      const/high16 v2, -0x1
  
--- 118,127 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 378
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
!     const-string/jumbo v1, "red"
  
      const/high16 v2, -0x1
  
***************
*** 124,129 ****
--- 131,137 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 379
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
      const-string v1, "green"
***************
*** 136,141 ****
--- 144,150 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 380
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
      const-string v1, "blue"
***************
*** 148,156 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
!     const-string v1, "yellow"
  
      const/16 v2, -0x100
  
--- 157,166 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 381
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
!     const-string/jumbo v1, "yellow"
  
      const/16 v2, -0x100
  
***************
*** 160,165 ****
--- 170,176 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 382
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
      const-string v1, "cyan"
***************
*** 172,180 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
!     const-string v1, "magenta"
  
      const v2, -0xff01
  
--- 183,192 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 383
      sget-object v0, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
  
!     const-string/jumbo v1, "magenta"
  
      const v2, -0xff01
  
***************
*** 184,189 ****
--- 196,202 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 384
      return-void
  .end method
  
***************
*** 191,196 ****
--- 204,210 ----
      .locals 0
  
      .prologue
+     .line 36
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 209,234 ****
--- 223,255 ----
  
      const/high16 v11, 0x3f80
  
+     .line 254
      invoke-static {p0, v9, v11}, Landroid/util/MathUtils;->constrain(FFF)F
  
      move-result p0
  
+     .line 255
      invoke-static {p1, v9, v11}, Landroid/util/MathUtils;->constrain(FFF)F
  
      move-result p1
  
+     .line 256
      invoke-static {p2, v9, v11}, Landroid/util/MathUtils;->constrain(FFF)F
  
      move-result p2
  
+     .line 258
      const/4 v7, 0x0
  
+     .line 259
      .local v7, red:F
      const/4 v2, 0x0
  
+     .line 260
      .local v2, green:F
      const/4 v0, 0x0
  
+     .line 262
      .local v0, blue:F
      float-to-int v9, p0
  
***************
*** 240,258 ****
--- 261,283 ----
  
      mul-float v3, v9, v10
  
+     .line 263
      .local v3, hf:F
      float-to-int v4, v3
  
+     .line 264
      .local v4, ihf:I
      int-to-float v9, v4
  
      sub-float v1, v3, v9
  
+     .line 265
      .local v1, f:F
      sub-float v9, v11, p1
  
      mul-float v5, p2, v9
  
+     .line 266
      .local v5, pv:F
      mul-float v9, p1, v1
  
***************
*** 260,265 ****
--- 285,291 ----
  
      mul-float v6, p2, v9
  
+     .line 267
      .local v6, qv:F
      sub-float v9, v11, v1
  
***************
*** 269,277 ****
--- 295,305 ----
  
      mul-float v8, p2, v9
  
+     .line 269
      .local v8, tv:F
      packed-switch v4, :pswitch_data_0
  
+     .line 302
      :goto_0
      const/high16 v9, -0x100
  
***************
*** 299,358 ****
--- 327,410 ----
  
      return v9
  
+     .line 271
      :pswitch_0
      move v7, p2
  
+     .line 272
      move v2, v8
  
+     .line 273
      move v0, v5
  
+     .line 274
      goto :goto_0
  
+     .line 276
      :pswitch_1
      move v7, v6
  
+     .line 277
      move v2, p2
  
+     .line 278
      move v0, v5
  
+     .line 279
      goto :goto_0
  
+     .line 281
      :pswitch_2
      move v7, v5
  
+     .line 282
      move v2, p2
  
+     .line 283
      move v0, v8
  
+     .line 284
      goto :goto_0
  
+     .line 286
      :pswitch_3
      move v7, v5
  
+     .line 287
      move v2, v6
  
+     .line 288
      move v0, p2
  
+     .line 289
      goto :goto_0
  
+     .line 291
      :pswitch_4
      move v7, v8
  
+     .line 292
      move v2, v5
  
+     .line 293
      move v0, p2
  
+     .line 294
      goto :goto_0
  
+     .line 296
      :pswitch_5
      move v7, p2
  
+     .line 297
      move v2, v5
  
+     .line 298
      move v0, v6
  
      goto :goto_0
  
+     .line 269
      nop
  
      :pswitch_data_0
***************
*** 371,376 ****
--- 423,429 ----
      .parameter "hsb"
  
      .prologue
+     .line 237
      const/4 v0, 0x0
  
      aget v0, p0, v0
***************
*** 396,407 ****
--- 449,462 ----
      .parameter "hsv"
  
      .prologue
+     .line 360
      array-length v0, p1
  
      const/4 v1, 0x3
  
      if-ge v0, v1, :cond_0
  
+     .line 361
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "3 components required for hsv"
***************
*** 410,415 ****
--- 465,471 ----
  
      throw v0
  
+     .line 363
      :cond_0
      invoke-static {p0, p1}, Landroid/graphics/Color;->nativeHSVToColor(I[F)I
  
***************
*** 423,428 ****
--- 479,485 ----
      .parameter "hsv"
  
      .prologue
+     .line 345
      const/16 v0, 0xff
  
      invoke-static {v0, p0}, Landroid/graphics/Color;->HSVToColor(I[F)I
***************
*** 440,451 ****
--- 497,510 ----
      .parameter "hsv"
  
      .prologue
+     .line 317
      array-length v0, p3
  
      const/4 v1, 0x3
  
      if-ge v0, v1, :cond_0
  
+     .line 318
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "3 components required for hsv"
***************
*** 454,462 ****
--- 513,523 ----
  
      throw v0
  
+     .line 320
      :cond_0
      invoke-static {p0, p1, p2, p3}, Landroid/graphics/Color;->nativeRGBToHSV(III[F)V
  
+     .line 321
      return-void
  .end method
  
***************
*** 465,470 ****
--- 526,532 ----
      .parameter "color"
  
      .prologue
+     .line 55
      ushr-int/lit8 v0, p0, 0x18
  
      return v0
***************
*** 478,483 ****
--- 540,546 ----
      .parameter "blue"
  
      .prologue
+     .line 107
      shl-int/lit8 v0, p0, 0x18
  
      shl-int/lit8 v1, p1, 0x10
***************
*** 498,503 ****
--- 561,567 ----
      .parameter "color"
  
      .prologue
+     .line 79
      and-int/lit16 v0, p0, 0xff
  
      return v0
***************
*** 508,525 ****
--- 572,593 ----
      .parameter "color"
  
      .prologue
+     .line 187
      shr-int/lit8 v4, p0, 0x10
  
      and-int/lit16 v3, v4, 0xff
  
+     .line 188
      .local v3, r:I
      shr-int/lit8 v4, p0, 0x8
  
      and-int/lit16 v2, v4, 0xff
  
+     .line 189
      .local v2, g:I
      and-int/lit16 v1, p0, 0xff
  
+     .line 191
      .local v1, b:I
      invoke-static {v3, v2}, Ljava/lang/Math;->max(II)I
  
***************
*** 529,534 ****
--- 597,603 ----
  
      move-result v0
  
+     .line 193
      .local v0, V:I
      int-to-float v4, v0
  
***************
*** 545,550 ****
--- 614,620 ----
      .parameter "hsv"
  
      .prologue
+     .line 332
      shr-int/lit8 v0, p0, 0x10
  
      and-int/lit16 v0, v0, 0xff
***************
*** 557,562 ****
--- 627,633 ----
  
      invoke-static {v0, v1, v2, p1}, Landroid/graphics/Color;->RGBToHSV(III[F)V
  
+     .line 333
      return-void
  .end method
  
***************
*** 565,570 ****
--- 636,642 ----
      .parameter "color"
  
      .prologue
+     .line 71
      shr-int/lit8 v0, p0, 0x8
  
      and-int/lit16 v0, v0, 0xff
***************
*** 577,594 ****
--- 649,670 ----
      .parameter "color"
  
      .prologue
+     .line 118
      shr-int/lit8 v10, p0, 0x10
  
      and-int/lit16 v7, v10, 0xff
  
+     .line 119
      .local v7, r:I
      shr-int/lit8 v10, p0, 0x8
  
      and-int/lit16 v6, v10, 0xff
  
+     .line 120
      .local v6, g:I
      and-int/lit16 v2, p0, 0xff
  
+     .line 122
      .local v2, b:I
      invoke-static {v7, v6}, Ljava/lang/Math;->max(II)I
  
***************
*** 598,603 ****
--- 674,680 ----
  
      move-result v1
  
+     .line 123
      .local v1, V:I
      invoke-static {v7, v6}, Ljava/lang/Math;->min(II)I
  
***************
*** 607,628 ****
--- 684,710 ----
  
      move-result v8
  
+     .line 127
      .local v8, temp:I
      if-ne v1, v8, :cond_1
  
+     .line 128
      const/4 v0, 0x0
  
+     .line 149
      .local v0, H:F
      :cond_0
      :goto_0
      return v0
  
+     .line 130
      .end local v0           #H:F
      :cond_1
      sub-int v10, v1, v8
  
      int-to-float v9, v10
  
+     .line 131
      .local v9, vtemp:F
      sub-int v10, v1, v7
  
***************
*** 630,635 ****
--- 712,718 ----
  
      div-float v5, v10, v9
  
+     .line 132
      .local v5, cr:F
      sub-int v10, v1, v6
  
***************
*** 637,642 ****
--- 720,726 ----
  
      div-float v4, v10, v9
  
+     .line 133
      .local v4, cg:F
      sub-int v10, v1, v2
  
***************
*** 644,676 ****
--- 728,767 ----
  
      div-float v3, v10, v9
  
+     .line 135
      .local v3, cb:F
      if-ne v7, v1, :cond_2
  
+     .line 136
      sub-float v0, v3, v4
  
+     .line 143
      .restart local v0       #H:F
      :goto_1
      const/high16 v10, 0x40c0
  
      div-float/2addr v0, v10
  
+     .line 144
      const/4 v10, 0x0
  
      cmpg-float v10, v0, v10
  
      if-gez v10, :cond_0
  
+     .line 145
      const/high16 v10, 0x3f80
  
      add-float/2addr v0, v10
  
      goto :goto_0
  
+     .line 137
      .end local v0           #H:F
      :cond_2
      if-ne v6, v1, :cond_3
  
+     .line 138
      const/high16 v10, 0x4000
  
      add-float/2addr v10, v5
***************
*** 680,685 ****
--- 771,777 ----
      .restart local v0       #H:F
      goto :goto_1
  
+     .line 140
      .end local v0           #H:F
      :cond_3
      const/high16 v10, 0x4080
***************
*** 703,708 ****
--- 795,801 ----
      .parameter "colorString"
  
      .prologue
+     .line 206
      const/4 v2, 0x0
  
      invoke-virtual {p0, v2}, Ljava/lang/String;->charAt(I)C
***************
*** 713,718 ****
--- 806,812 ----
  
      if-ne v2, v3, :cond_2
  
+     .line 208
      const/4 v2, 0x1
  
      invoke-virtual {p0, v2}, Ljava/lang/String;->substring(I)Ljava/lang/String;
***************
*** 725,730 ****
--- 819,825 ----
  
      move-result-wide v0
  
+     .line 209
      .local v0, color:J
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
***************
*** 734,750 ****
--- 829,849 ----
  
      if-ne v2, v3, :cond_1
  
+     .line 211
      const-wide/32 v2, -0x1000000
  
      or-long/2addr v0, v2
  
+     .line 215
      :cond_0
      long-to-int v2, v0
  
+     .line 219
      .end local v0           #color:J
      :goto_0
      return v2
  
+     .line 212
      .restart local v0       #color:J
      :cond_1
      invoke-virtual {p0}, Ljava/lang/String;->length()I
***************
*** 755,760 ****
--- 854,860 ----
  
      if-eq v2, v3, :cond_0
  
+     .line 213
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
      const-string v3, "Unknown color"
***************
*** 763,768 ****
--- 863,869 ----
  
      throw v2
  
+     .line 217
      .end local v0           #color:J
      :cond_2
      sget-object v2, Landroid/graphics/Color;->sColorNameMap:Ljava/util/HashMap;
***************
*** 779,793 ****
--- 880,897 ----
  
      check-cast v0, Ljava/lang/Integer;
  
+     .line 218
      .local v0, color:Ljava/lang/Integer;
      if-eqz v0, :cond_3
  
+     .line 219
      invoke-virtual {v0}, Ljava/lang/Integer;->intValue()I
  
      move-result v2
  
      goto :goto_0
  
+     .line 222
      :cond_3
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
***************
*** 803,808 ****
--- 907,913 ----
      .parameter "color"
  
      .prologue
+     .line 63
      shr-int/lit8 v0, p0, 0x10
  
      and-int/lit16 v0, v0, 0xff
***************
*** 817,822 ****
--- 922,928 ----
      .parameter "blue"
  
      .prologue
+     .line 93
      const/high16 v0, -0x100
  
      shl-int/lit8 v1, p0, 0x10
***************
*** 837,854 ****
--- 943,964 ----
      .parameter "color"
  
      .prologue
+     .line 160
      shr-int/lit8 v6, p0, 0x10
  
      and-int/lit16 v4, v6, 0xff
  
+     .line 161
      .local v4, r:I
      shr-int/lit8 v6, p0, 0x8
  
      and-int/lit16 v3, v6, 0xff
  
+     .line 162
      .local v3, g:I
      and-int/lit16 v2, p0, 0xff
  
+     .line 165
      .local v2, b:I
      invoke-static {v4, v3}, Ljava/lang/Math;->max(II)I
  
***************
*** 858,863 ****
--- 968,974 ----
  
      move-result v1
  
+     .line 166
      .local v1, V:I
      invoke-static {v4, v3}, Ljava/lang/Math;->min(II)I
  
***************
*** 867,881 ****
--- 978,996 ----
  
      move-result v5
  
+     .line 170
      .local v5, temp:I
      if-ne v1, v5, :cond_0
  
+     .line 171
      const/4 v0, 0x0
  
+     .line 176
      .local v0, S:F
      :goto_0
      return v0
  
+     .line 173
      .end local v0           #S:F
      :cond_0
      sub-int v6, v1, v5
