*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/DrawFilter.smali	2012-03-23 19:35:41.407341947 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/DrawFilter.smali.original	2012-03-23 19:34:44.778309318 -0500
***************
*** 12,17 ****
--- 12,18 ----
      .locals 0
  
      .prologue
+     .line 25
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 31,39 ****
--- 32,42 ----
      .end annotation
  
      .prologue
+     .line 31
      iget v0, p0, Landroid/graphics/DrawFilter;->mNativeInt:I
  
      invoke-static {v0}, Landroid/graphics/DrawFilter;->nativeDestructor(I)V
  
+     .line 32
      return-void
  .end method
