*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/LightingColorFilter.smali	2012-03-23 19:35:41.846349953 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/LightingColorFilter.smali.original	2012-03-23 19:34:44.781309374 -0500
***************
*** 10,23 ****
--- 10,26 ----
      .parameter "add"
  
      .prologue
+     .line 31
      invoke-direct {p0}, Landroid/graphics/ColorFilter;-><init>()V
  
+     .line 32
      invoke-static {p1, p2}, Landroid/graphics/LightingColorFilter;->native_CreateLightingFilter(II)I
  
      move-result v0
  
      iput v0, p0, Landroid/graphics/LightingColorFilter;->native_instance:I
  
+     .line 33
      iget v0, p0, Landroid/graphics/LightingColorFilter;->native_instance:I
  
      invoke-static {v0, p1, p2}, Landroid/graphics/LightingColorFilter;->nCreateLightingFilter(III)I
***************
*** 26,31 ****
--- 29,35 ----
  
      iput v0, p0, Landroid/graphics/LightingColorFilter;->nativeColorFilter:I
  
+     .line 34
      return-void
  .end method
  
