*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/LinearGradient.smali	2012-03-23 19:35:44.423396945 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/LinearGradient.smali.original	2012-03-23 19:34:44.816310012 -0500
***************
*** 15,22 ****
--- 15,24 ----
      .parameter "tile"
  
      .prologue
+     .line 54
      invoke-direct {p0}, Landroid/graphics/Shader;-><init>()V
  
+     .line 55
      move-object/from16 v0, p7
  
      iget v8, v0, Landroid/graphics/Shader$TileMode;->nativeInt:I
***************
*** 41,46 ****
--- 43,49 ----
  
      iput v1, p0, Landroid/graphics/LinearGradient;->native_instance:I
  
+     .line 56
      iget v2, p0, Landroid/graphics/LinearGradient;->native_instance:I
  
      move-object/from16 v0, p7
***************
*** 67,72 ****
--- 70,76 ----
  
      iput v1, p0, Landroid/graphics/LinearGradient;->native_shader:I
  
+     .line 58
      return-void
  .end method
  
***************
*** 81,102 ****
      .parameter "tile"
  
      .prologue
      invoke-direct {p0}, Landroid/graphics/Shader;-><init>()V
  
      array-length v1, p5
  
      const/4 v2, 0x2
  
      if-ge v1, v2, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
!     const-string v2, "needs >= 2 number of colors"
  
      invoke-direct {v1, v2}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
      :cond_0
      if-eqz p6, :cond_1
  
--- 85,110 ----
      .parameter "tile"
  
      .prologue
+     .line 32
      invoke-direct {p0}, Landroid/graphics/Shader;-><init>()V
  
+     .line 33
      array-length v1, p5
  
      const/4 v2, 0x2
  
      if-ge v1, v2, :cond_0
  
+     .line 34
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v2, "needs >= 2 number of colors"
  
      invoke-direct {v1, v2}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
+     .line 36
      :cond_0
      if-eqz p6, :cond_1
  
***************
*** 108,113 ****
--- 116,122 ----
  
      if-eq v1, v2, :cond_1
  
+     .line 37
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      const-string v2, "color and position arrays must be of equal length"
***************
*** 116,121 ****
--- 125,131 ----
  
      throw v1
  
+     .line 39
      :cond_1
      move-object/from16 v0, p7
  
***************
*** 141,146 ****
--- 151,157 ----
  
      iput v1, p0, Landroid/graphics/LinearGradient;->native_instance:I
  
+     .line 40
      iget v2, p0, Landroid/graphics/LinearGradient;->native_instance:I
  
      move-object/from16 v0, p7
***************
*** 167,172 ****
--- 178,184 ----
  
      iput v1, p0, Landroid/graphics/LinearGradient;->native_shader:I
  
+     .line 42
      return-void
  .end method
  
