*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Matrix$ScaleToFit.smali	2012-03-23 19:35:41.674346816 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Matrix$ScaleToFit.smali.original	2012-03-23 19:34:44.780309355 -0500
***************
*** 52,57 ****
--- 52,58 ----
  
      const/4 v2, 0x0
  
+     .line 493
      new-instance v0, Landroid/graphics/Matrix$ScaleToFit;
  
      const-string v1, "FILL"
***************
*** 60,65 ****
--- 61,67 ----
  
      sput-object v0, Landroid/graphics/Matrix$ScaleToFit;->FILL:Landroid/graphics/Matrix$ScaleToFit;
  
+     .line 500
      new-instance v0, Landroid/graphics/Matrix$ScaleToFit;
  
      const-string v1, "START"
***************
*** 68,73 ****
--- 70,76 ----
  
      sput-object v0, Landroid/graphics/Matrix$ScaleToFit;->START:Landroid/graphics/Matrix$ScaleToFit;
  
+     .line 506
      new-instance v0, Landroid/graphics/Matrix$ScaleToFit;
  
      const-string v1, "CENTER"
***************
*** 76,81 ****
--- 79,85 ----
  
      sput-object v0, Landroid/graphics/Matrix$ScaleToFit;->CENTER:Landroid/graphics/Matrix$ScaleToFit;
  
+     .line 513
      new-instance v0, Landroid/graphics/Matrix$ScaleToFit;
  
      const-string v1, "END"
***************
*** 84,89 ****
--- 88,94 ----
  
      sput-object v0, Landroid/graphics/Matrix$ScaleToFit;->END:Landroid/graphics/Matrix$ScaleToFit;
  
+     .line 488
      const/4 v0, 0x4
  
      new-array v0, v0, [Landroid/graphics/Matrix$ScaleToFit;
***************
*** 121,130 ****
--- 126,138 ----
      .end annotation
  
      .prologue
+     .line 516
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 517
      iput p3, p0, Landroid/graphics/Matrix$ScaleToFit;->nativeInt:I
  
+     .line 518
      return-void
  .end method
  
***************
*** 133,138 ****
--- 141,147 ----
      .parameter "name"
  
      .prologue
+     .line 488
      const-class v0, Landroid/graphics/Matrix$ScaleToFit;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 148,153 ****
--- 157,163 ----
      .locals 1
  
      .prologue
+     .line 488
      sget-object v0, Landroid/graphics/Matrix$ScaleToFit;->$VALUES:[Landroid/graphics/Matrix$ScaleToFit;
  
      invoke-virtual {v0}, [Landroid/graphics/Matrix$ScaleToFit;->clone()Ljava/lang/Object;
