*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Matrix$1.smali	2012-03-23 19:35:44.894405534 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Matrix$1.smali.original	2012-03-23 19:34:44.819310066 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 0
  
      .prologue
+     .line 41
      invoke-direct {p0}, Landroid/graphics/Matrix;-><init>()V
  
      return-void
***************
*** 30,35 ****
--- 31,37 ----
      .locals 2
  
      .prologue
+     .line 43
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "Matrix can not be modified"
***************
*** 44,51 ****
--- 46,55 ----
      .parameter "other"
  
      .prologue
+     .line 199
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 200
      const/4 v0, 0x0
  
      return v0
***************
*** 56,63 ****
--- 60,69 ----
      .parameter "degrees"
  
      .prologue
+     .line 181
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 182
      const/4 v0, 0x0
  
      return v0
***************
*** 70,77 ****
--- 76,85 ----
      .parameter "py"
  
      .prologue
+     .line 175
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 176
      const/4 v0, 0x0
  
      return v0
***************
*** 83,90 ****
--- 91,100 ----
      .parameter "sy"
  
      .prologue
+     .line 169
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 170
      const/4 v0, 0x0
  
      return v0
***************
*** 98,105 ****
--- 108,117 ----
      .parameter "py"
  
      .prologue
+     .line 163
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 164
      const/4 v0, 0x0
  
      return v0
***************
*** 111,118 ****
--- 123,132 ----
      .parameter "ky"
  
      .prologue
+     .line 193
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 194
      const/4 v0, 0x0
  
      return v0
***************
*** 126,133 ****
--- 140,149 ----
      .parameter "py"
  
      .prologue
+     .line 187
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 188
      const/4 v0, 0x0
  
      return v0
***************
*** 139,146 ****
--- 155,164 ----
      .parameter "dy"
  
      .prologue
+     .line 157
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 158
      const/4 v0, 0x0
  
      return v0
***************
*** 151,158 ****
--- 169,178 ----
      .parameter "other"
  
      .prologue
+     .line 151
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 152
      const/4 v0, 0x0
  
      return v0
***************
*** 163,170 ****
--- 183,192 ----
      .parameter "degrees"
  
      .prologue
+     .line 133
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 134
      const/4 v0, 0x0
  
      return v0
***************
*** 177,184 ****
--- 199,208 ----
      .parameter "py"
  
      .prologue
+     .line 127
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 128
      const/4 v0, 0x0
  
      return v0
***************
*** 190,197 ****
--- 214,223 ----
      .parameter "sy"
  
      .prologue
+     .line 121
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 122
      const/4 v0, 0x0
  
      return v0
***************
*** 205,212 ****
--- 231,240 ----
      .parameter "py"
  
      .prologue
+     .line 115
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 116
      const/4 v0, 0x0
  
      return v0
***************
*** 218,225 ****
--- 246,255 ----
      .parameter "ky"
  
      .prologue
+     .line 145
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 146
      const/4 v0, 0x0
  
      return v0
***************
*** 233,240 ****
--- 263,272 ----
      .parameter "py"
  
      .prologue
+     .line 139
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 140
      const/4 v0, 0x0
  
      return v0
***************
*** 246,253 ****
--- 278,287 ----
      .parameter "dy"
  
      .prologue
+     .line 109
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 110
      const/4 v0, 0x0
  
      return v0
***************
*** 257,264 ****
--- 291,300 ----
      .locals 0
  
      .prologue
+     .line 53
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 54
      return-void
  .end method
  
***************
*** 267,274 ****
--- 303,312 ----
      .parameter "src"
  
      .prologue
+     .line 48
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 49
      return-void
  .end method
  
***************
*** 278,285 ****
--- 316,325 ----
      .parameter "b"
  
      .prologue
+     .line 103
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 104
      const/4 v0, 0x0
  
      return v0
***************
*** 294,301 ****
--- 334,343 ----
      .parameter "pointCount"
  
      .prologue
+     .line 212
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 213
      const/4 v0, 0x0
  
      return v0
***************
*** 308,315 ****
--- 350,359 ----
      .parameter "stf"
  
      .prologue
+     .line 205
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 206
      const/4 v0, 0x0
  
      return v0
***************
*** 320,327 ****
--- 364,373 ----
      .parameter "degrees"
  
      .prologue
+     .line 78
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 79
      return-void
  .end method
  
***************
*** 332,339 ****
--- 378,387 ----
      .parameter "py"
  
      .prologue
+     .line 73
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 74
      return-void
  .end method
  
***************
*** 343,350 ****
--- 391,400 ----
      .parameter "sy"
  
      .prologue
+     .line 68
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 69
      return-void
  .end method
  
***************
*** 356,363 ****
--- 406,415 ----
      .parameter "py"
  
      .prologue
+     .line 63
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 64
      return-void
  .end method
  
***************
*** 367,374 ****
--- 419,428 ----
      .parameter "cosValue"
  
      .prologue
+     .line 88
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 89
      return-void
  .end method
  
***************
*** 380,387 ****
--- 434,443 ----
      .parameter "py"
  
      .prologue
+     .line 83
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 84
      return-void
  .end method
  
***************
*** 391,398 ****
--- 447,456 ----
      .parameter "ky"
  
      .prologue
+     .line 98
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 99
      return-void
  .end method
  
***************
*** 404,411 ****
--- 462,471 ----
      .parameter "py"
  
      .prologue
+     .line 93
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 94
      return-void
  .end method
  
***************
*** 415,422 ****
--- 475,484 ----
      .parameter "dy"
  
      .prologue
+     .line 58
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 59
      return-void
  .end method
  
***************
*** 425,431 ****
--- 487,495 ----
      .parameter "values"
  
      .prologue
+     .line 218
      invoke-virtual {p0}, Landroid/graphics/Matrix$1;->oops()V
  
+     .line 219
      return-void
  .end method
