*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/NinePatch.smali	2012-03-23 19:35:44.719402343 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/NinePatch.smali.original	2012-03-23 19:34:44.818310048 -0500
***************
*** 23,42 ****
--- 23,48 ----
      .parameter "srcName"
  
      .prologue
+     .line 52
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 42
      new-instance v0, Landroid/graphics/RectF;
  
      invoke-direct {v0}, Landroid/graphics/RectF;-><init>()V
  
      iput-object v0, p0, Landroid/graphics/NinePatch;->mRect:Landroid/graphics/RectF;
  
+     .line 53
      iput-object p1, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
+     .line 54
      iput-object p2, p0, Landroid/graphics/NinePatch;->mChunk:[B
  
+     .line 55
      iput-object p3, p0, Landroid/graphics/NinePatch;->mSrcName:Ljava/lang/String;
  
+     .line 56
      iget-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
      invoke-virtual {v0}, Landroid/graphics/Bitmap;->ni()I
***************
*** 45,50 ****
--- 51,57 ----
  
      invoke-static {v0, p2}, Landroid/graphics/NinePatch;->validateNinePatchChunk(I[B)V
  
+     .line 57
      return-void
  .end method
  
***************
*** 53,82 ****
--- 60,96 ----
      .parameter "patch"
  
      .prologue
+     .line 62
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 42
      new-instance v0, Landroid/graphics/RectF;
  
      invoke-direct {v0}, Landroid/graphics/RectF;-><init>()V
  
      iput-object v0, p0, Landroid/graphics/NinePatch;->mRect:Landroid/graphics/RectF;
  
+     .line 63
      iget-object v0, p1, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
      iput-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
+     .line 64
      iget-object v0, p1, Landroid/graphics/NinePatch;->mChunk:[B
  
      iput-object v0, p0, Landroid/graphics/NinePatch;->mChunk:[B
  
+     .line 65
      iget-object v0, p1, Landroid/graphics/NinePatch;->mSrcName:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/graphics/NinePatch;->mSrcName:Ljava/lang/String;
  
+     .line 66
      iget-object v0, p1, Landroid/graphics/NinePatch;->mPaint:Landroid/graphics/Paint;
  
      if-eqz v0, :cond_0
  
+     .line 67
      new-instance v0, Landroid/graphics/Paint;
  
      iget-object v1, p1, Landroid/graphics/NinePatch;->mPaint:Landroid/graphics/Paint;
***************
*** 85,90 ****
--- 99,105 ----
  
      iput-object v0, p0, Landroid/graphics/NinePatch;->mPaint:Landroid/graphics/Paint;
  
+     .line 69
      :cond_0
      iget-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
***************
*** 96,101 ****
--- 111,117 ----
  
      invoke-static {v0, v1}, Landroid/graphics/NinePatch;->validateNinePatchChunk(I[B)V
  
+     .line 70
      return-void
  .end method
  
***************
*** 122,133 ****
--- 138,151 ----
      .parameter "location"
  
      .prologue
+     .line 100
      invoke-virtual {p1}, Landroid/graphics/Canvas;->isHardwareAccelerated()Z
  
      move-result v0
  
      if-nez v0, :cond_1
  
+     .line 101
      iget v0, p1, Landroid/graphics/Canvas;->mNativeCanvas:I
  
      iget-object v1, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
***************
*** 157,175 ****
--- 175,197 ----
  
      invoke-static/range {v0 .. v6}, Landroid/graphics/NinePatch;->nativeDraw(ILandroid/graphics/Rect;I[BIII)V
  
+     .line 109
      :goto_1
      return-void
  
+     .line 101
      :cond_0
      const/4 v4, 0x0
  
      goto :goto_0
  
+     .line 106
      :cond_1
      iget-object v0, p0, Landroid/graphics/NinePatch;->mRect:Landroid/graphics/RectF;
  
      invoke-virtual {v0, p2}, Landroid/graphics/RectF;->set(Landroid/graphics/Rect;)V
  
+     .line 107
      iget-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
      iget-object v1, p0, Landroid/graphics/NinePatch;->mChunk:[B
***************
*** 190,201 ****
--- 212,225 ----
      .parameter "paint"
  
      .prologue
+     .line 119
      invoke-virtual {p1}, Landroid/graphics/Canvas;->isHardwareAccelerated()Z
  
      move-result v0
  
      if-nez v0, :cond_1
  
+     .line 120
      iget v0, p1, Landroid/graphics/Canvas;->mNativeCanvas:I
  
      iget-object v1, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
***************
*** 221,239 ****
--- 245,267 ----
  
      invoke-static/range {v0 .. v6}, Landroid/graphics/NinePatch;->nativeDraw(ILandroid/graphics/Rect;I[BIII)V
  
+     .line 127
      :goto_1
      return-void
  
+     .line 120
      :cond_0
      const/4 v4, 0x0
  
      goto :goto_0
  
+     .line 124
      :cond_1
      iget-object v0, p0, Landroid/graphics/NinePatch;->mRect:Landroid/graphics/RectF;
  
      invoke-virtual {v0, p2}, Landroid/graphics/RectF;->set(Landroid/graphics/Rect;)V
  
+     .line 125
      iget-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
      iget-object v1, p0, Landroid/graphics/NinePatch;->mChunk:[B
***************
*** 251,262 ****
--- 279,292 ----
      .parameter "location"
  
      .prologue
+     .line 83
      invoke-virtual {p1}, Landroid/graphics/Canvas;->isHardwareAccelerated()Z
  
      move-result v0
  
      if-nez v0, :cond_1
  
+     .line 84
      iget v0, p1, Landroid/graphics/Canvas;->mNativeCanvas:I
  
      iget-object v1, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
***************
*** 286,299 ****
--- 316,332 ----
  
      invoke-static/range {v0 .. v6}, Landroid/graphics/NinePatch;->nativeDraw(ILandroid/graphics/RectF;I[BIII)V
  
+     .line 91
      :goto_1
      return-void
  
+     .line 84
      :cond_0
      const/4 v4, 0x0
  
      goto :goto_0
  
+     .line 89
      :cond_1
      iget-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
***************
*** 310,315 ****
--- 343,349 ----
      .locals 1
  
      .prologue
+     .line 134
      iget-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
      iget v0, v0, Landroid/graphics/Bitmap;->mDensity:I
***************
*** 321,326 ****
--- 355,361 ----
      .locals 1
  
      .prologue
+     .line 142
      iget-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
      invoke-virtual {v0}, Landroid/graphics/Bitmap;->getHeight()I
***************
*** 335,340 ****
--- 370,376 ----
      .parameter "location"
  
      .prologue
+     .line 150
      iget-object v1, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
      invoke-virtual {v1}, Landroid/graphics/Bitmap;->ni()I
***************
*** 347,352 ****
--- 383,389 ----
  
      move-result v0
  
+     .line 151
      .local v0, r:I
      if-eqz v0, :cond_0
  
***************
*** 367,372 ****
--- 404,410 ----
      .locals 1
  
      .prologue
+     .line 138
      iget-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
      invoke-virtual {v0}, Landroid/graphics/Bitmap;->getWidth()I
***************
*** 380,385 ****
--- 418,424 ----
      .locals 1
  
      .prologue
+     .line 146
      iget-object v0, p0, Landroid/graphics/NinePatch;->mBitmap:Landroid/graphics/Bitmap;
  
      invoke-virtual {v0}, Landroid/graphics/Bitmap;->hasAlpha()Z
***************
*** 394,400 ****
--- 433,441 ----
      .parameter "p"
  
      .prologue
+     .line 73
      iput-object p1, p0, Landroid/graphics/NinePatch;->mPaint:Landroid/graphics/Paint;
  
+     .line 74
      return-void
  .end method
