*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Paint.smali	2012-03-23 19:35:44.946406482 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/Paint.smali.original	2012-03-23 19:34:44.820310084 -0500
***************
*** 131,136 ****
--- 131,137 ----
  
      const/4 v2, 0x0
  
+     .line 73
      new-array v0, v5, [Landroid/graphics/Paint$Style;
  
      sget-object v1, Landroid/graphics/Paint$Style;->FILL:Landroid/graphics/Paint$Style;
***************
*** 147,152 ****
--- 148,154 ----
  
      sput-object v0, Landroid/graphics/Paint;->sStyleArray:[Landroid/graphics/Paint$Style;
  
+     .line 76
      new-array v0, v5, [Landroid/graphics/Paint$Cap;
  
      sget-object v1, Landroid/graphics/Paint$Cap;->BUTT:Landroid/graphics/Paint$Cap;
***************
*** 163,168 ****
--- 165,171 ----
  
      sput-object v0, Landroid/graphics/Paint;->sCapArray:[Landroid/graphics/Paint$Cap;
  
+     .line 79
      new-array v0, v5, [Landroid/graphics/Paint$Join;
  
      sget-object v1, Landroid/graphics/Paint$Join;->MITER:Landroid/graphics/Paint$Join;
***************
*** 179,184 ****
--- 182,188 ----
  
      sput-object v0, Landroid/graphics/Paint;->sJoinArray:[Landroid/graphics/Paint$Join;
  
+     .line 82
      new-array v0, v5, [Landroid/graphics/Paint$Align;
  
      sget-object v1, Landroid/graphics/Paint$Align;->LEFT:Landroid/graphics/Paint$Align;
***************
*** 202,211 ****
--- 206,217 ----
      .locals 1
  
      .prologue
+     .line 334
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/graphics/Paint;-><init>(I)V
  
+     .line 335
      return-void
  .end method
  
***************
*** 214,241 ****
--- 220,253 ----
      .parameter "flags"
  
      .prologue
+     .line 343
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 71
      const/4 v0, 0x2
  
      iput v0, p0, Landroid/graphics/Paint;->mBidiFlags:I
  
+     .line 344
      invoke-static {}, Landroid/graphics/Paint;->native_init()I
  
      move-result v0
  
      iput v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
+     .line 345
      or-int/lit16 v0, p1, 0x100
  
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->setFlags(I)V
  
+     .line 350
      const/high16 v0, 0x3f80
  
      iput v0, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
  
      iput v0, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
+     .line 351
      return-void
  .end method
  
***************
*** 244,255 ****
--- 256,270 ----
      .parameter "paint"
  
      .prologue
+     .line 360
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 71
      const/4 v0, 0x2
  
      iput v0, p0, Landroid/graphics/Paint;->mBidiFlags:I
  
+     .line 361
      iget v0, p1, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v0}, Landroid/graphics/Paint;->native_initWithPaint(I)I
***************
*** 258,265 ****
--- 273,282 ----
  
      iput v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
+     .line 362
      invoke-direct {p0, p1}, Landroid/graphics/Paint;->setClassVariablesFrom(Landroid/graphics/Paint;)V
  
+     .line 363
      return-void
  .end method
  
***************
*** 385,454 ****
--- 402,488 ----
      .parameter "paint"
  
      .prologue
+     .line 396
      iget-object v0, p1, Landroid/graphics/Paint;->mColorFilter:Landroid/graphics/ColorFilter;
  
      iput-object v0, p0, Landroid/graphics/Paint;->mColorFilter:Landroid/graphics/ColorFilter;
  
+     .line 397
      iget-object v0, p1, Landroid/graphics/Paint;->mMaskFilter:Landroid/graphics/MaskFilter;
  
      iput-object v0, p0, Landroid/graphics/Paint;->mMaskFilter:Landroid/graphics/MaskFilter;
  
+     .line 398
      iget-object v0, p1, Landroid/graphics/Paint;->mPathEffect:Landroid/graphics/PathEffect;
  
      iput-object v0, p0, Landroid/graphics/Paint;->mPathEffect:Landroid/graphics/PathEffect;
  
+     .line 399
      iget-object v0, p1, Landroid/graphics/Paint;->mRasterizer:Landroid/graphics/Rasterizer;
  
      iput-object v0, p0, Landroid/graphics/Paint;->mRasterizer:Landroid/graphics/Rasterizer;
  
+     .line 400
      iget-object v0, p1, Landroid/graphics/Paint;->mShader:Landroid/graphics/Shader;
  
      iput-object v0, p0, Landroid/graphics/Paint;->mShader:Landroid/graphics/Shader;
  
+     .line 401
      iget-object v0, p1, Landroid/graphics/Paint;->mTypeface:Landroid/graphics/Typeface;
  
      iput-object v0, p0, Landroid/graphics/Paint;->mTypeface:Landroid/graphics/Typeface;
  
+     .line 402
      iget-object v0, p1, Landroid/graphics/Paint;->mXfermode:Landroid/graphics/Xfermode;
  
      iput-object v0, p0, Landroid/graphics/Paint;->mXfermode:Landroid/graphics/Xfermode;
  
+     .line 404
      iget-boolean v0, p1, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
      iput-boolean v0, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
+     .line 405
      iget v0, p1, Landroid/graphics/Paint;->mCompatScaling:F
  
      iput v0, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
+     .line 406
      iget v0, p1, Landroid/graphics/Paint;->mInvCompatScaling:F
  
      iput v0, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
  
+     .line 408
      iget-boolean v0, p1, Landroid/graphics/Paint;->hasShadow:Z
  
      iput-boolean v0, p0, Landroid/graphics/Paint;->hasShadow:Z
  
+     .line 409
      iget v0, p1, Landroid/graphics/Paint;->shadowDx:F
  
      iput v0, p0, Landroid/graphics/Paint;->shadowDx:F
  
+     .line 410
      iget v0, p1, Landroid/graphics/Paint;->shadowDy:F
  
      iput v0, p0, Landroid/graphics/Paint;->shadowDy:F
  
+     .line 411
      iget v0, p1, Landroid/graphics/Paint;->shadowRadius:F
  
      iput v0, p0, Landroid/graphics/Paint;->shadowRadius:F
  
+     .line 412
      iget v0, p1, Landroid/graphics/Paint;->shadowColor:I
  
      iput v0, p0, Landroid/graphics/Paint;->shadowColor:I
  
+     .line 414
      iget v0, p1, Landroid/graphics/Paint;->mBidiFlags:I
  
      iput v0, p0, Landroid/graphics/Paint;->mBidiFlags:I
  
+     .line 415
      return-void
  .end method
  
***************
*** 469,484 ****
      .prologue
      const/4 v2, 0x0
  
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v2, "text cannot be null"
  
      invoke-direct {v0, v2}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      or-int v0, p2, p3
  
--- 503,521 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1397
      if-nez p1, :cond_0
  
+     .line 1398
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v2, "text cannot be null"
  
      invoke-direct {v0, v2}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 1400
      :cond_0
      or-int v0, p2, p3
  
***************
*** 496,507 ****
--- 533,546 ----
  
      if-gez v0, :cond_1
  
+     .line 1401
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v0}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v0
  
+     .line 1404
      :cond_1
      invoke-interface {p1}, Ljava/lang/CharSequence;->length()I
  
***************
*** 514,523 ****
--- 553,564 ----
      :cond_2
      move v6, v2
  
+     .line 1424
      .end local p1
      :goto_0
      return v6
  
+     .line 1407
      .restart local p1
      :cond_3
      if-nez p2, :cond_4
***************
*** 532,537 ****
--- 573,579 ----
  
      if-ne p3, v0, :cond_4
  
+     .line 1408
      check-cast p1, Ljava/lang/String;
  
      .end local p1
***************
*** 541,546 ****
--- 583,589 ----
  
      goto :goto_0
  
+     .line 1412
      .restart local p1
      :cond_4
      sub-int v0, p3, p2
***************
*** 549,559 ****
--- 592,605 ----
  
      move-result-object v1
  
+     .line 1415
      .local v1, buf:[C
      invoke-static {p1, p2, p3, v1, v2}, Landroid/text/TextUtils;->getChars(Ljava/lang/CharSequence;II[CI)V
  
+     .line 1417
      if-eqz p4, :cond_5
  
+     .line 1418
      sub-int v3, p3, p2
  
      move-object v0, p0
***************
*** 566,577 ****
--- 612,625 ----
  
      move-result v6
  
+     .line 1423
      .local v6, result:I
      :goto_1
      invoke-static {v1}, Landroid/graphics/TemporaryBuffer;->recycle([C)V
  
      goto :goto_0
  
+     .line 1420
      .end local v6           #result:I
      :cond_5
      sub-int v0, p3, p2
***************
*** 602,617 ****
      .prologue
      const/4 v2, 0x0
  
      if-nez p1, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
--- 650,668 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1445
      if-nez p1, :cond_0
  
+     .line 1446
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1449
      :cond_0
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
***************
*** 621,646 ****
--- 672,702 ----
  
      move v1, v2
  
+     .line 1462
      :cond_1
      :goto_0
      return v1
  
+     .line 1452
      :cond_2
      iget-boolean v3, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
      if-nez v3, :cond_3
  
+     .line 1453
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/graphics/Paint;->native_breakText(Ljava/lang/String;ZF[F)I
  
      move-result v1
  
      goto :goto_0
  
+     .line 1456
      :cond_3
      invoke-virtual {p0}, Landroid/graphics/Paint;->getTextSize()F
  
      move-result v0
  
+     .line 1457
      .local v0, oldSize:F
      iget v3, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
***************
*** 648,653 ****
--- 704,710 ----
  
      invoke-virtual {p0, v3}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1458
      iget v3, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
      mul-float/2addr v3, p3
***************
*** 656,664 ****
--- 713,723 ----
  
      move-result v1
  
+     .line 1460
      .local v1, res:I
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1461
      if-eqz p4, :cond_1
  
      aget v3, p4, v2
***************
*** 683,698 ****
      .prologue
      const/4 v8, 0x0
  
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "text cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      if-ltz p2, :cond_1
  
--- 742,760 ----
      .prologue
      const/4 v8, 0x0
  
+     .line 1352
      if-nez p1, :cond_0
  
+     .line 1353
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "text cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 1355
      :cond_0
      if-ltz p2, :cond_1
  
***************
*** 706,711 ****
--- 768,774 ----
  
      if-ge v0, v1, :cond_2
  
+     .line 1356
      :cond_1
      new-instance v0, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 713,718 ****
--- 776,782 ----
  
      throw v0
  
+     .line 1359
      :cond_2
      array-length v0, p1
  
***************
*** 723,748 ****
--- 787,817 ----
      :cond_3
      move v7, v8
  
+     .line 1372
      :cond_4
      :goto_0
      return v7
  
+     .line 1362
      :cond_5
      iget-boolean v0, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
      if-nez v0, :cond_6
  
+     .line 1363
      invoke-direct/range {p0 .. p5}, Landroid/graphics/Paint;->native_breakText([CIIF[F)I
  
      move-result v7
  
      goto :goto_0
  
+     .line 1366
      :cond_6
      invoke-virtual {p0}, Landroid/graphics/Paint;->getTextSize()F
  
      move-result v6
  
+     .line 1367
      .local v6, oldSize:F
      iget v0, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
***************
*** 750,755 ****
--- 819,825 ----
  
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1368
      iget v0, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
      mul-float v4, p4, v0
***************
*** 768,776 ****
--- 838,848 ----
  
      move-result v7
  
+     .line 1370
      .local v7, res:I
      invoke-virtual {p0, v6}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1371
      if-eqz p5, :cond_4
  
      aget v0, p5, v8
***************
*** 792,801 ****
--- 864,876 ----
  
      const/4 v0, 0x0
  
+     .line 1017
      iput-boolean v1, p0, Landroid/graphics/Paint;->hasShadow:Z
  
+     .line 1018
      invoke-direct {p0, v0, v0, v0, v1}, Landroid/graphics/Paint;->nSetShadowLayer(FFFI)V
  
+     .line 1019
      return-void
  .end method
  
***************
*** 811,816 ****
--- 886,892 ----
      .end annotation
  
      .prologue
+     .line 2108
      :try_start_0
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
***************
*** 818,827 ****
--- 894,906 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 2110
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 2112
      return-void
  
+     .line 2110
      :catchall_0
      move-exception v0
  
***************
*** 837,842 ****
--- 916,922 ----
      .locals 1
  
      .prologue
+     .line 436
      iget v0, p0, Landroid/graphics/Paint;->mBidiFlags:I
  
      return v0
***************
*** 849,854 ****
--- 929,935 ----
      .locals 1
  
      .prologue
+     .line 833
      iget-object v0, p0, Landroid/graphics/Paint;->mColorFilter:Landroid/graphics/ColorFilter;
  
      return-object v0
***************
*** 860,865 ****
--- 941,947 ----
      .parameter "dst"
  
      .prologue
+     .line 797
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-virtual {p1}, Landroid/graphics/Path;->ni()I
***************
*** 887,899 ****
--- 969,984 ----
      .locals 1
  
      .prologue
+     .line 1157
      new-instance v0, Landroid/graphics/Paint$FontMetrics;
  
      invoke-direct {v0}, Landroid/graphics/Paint$FontMetrics;-><init>()V
  
+     .line 1158
      .local v0, fm:Landroid/graphics/Paint$FontMetrics;
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->getFontMetrics(Landroid/graphics/Paint$FontMetrics;)F
  
+     .line 1159
      return-object v0
  .end method
  
***************
*** 904,916 ****
--- 989,1004 ----
      .locals 1
  
      .prologue
+     .line 1193
      new-instance v0, Landroid/graphics/Paint$FontMetricsInt;
  
      invoke-direct {v0}, Landroid/graphics/Paint$FontMetricsInt;-><init>()V
  
+     .line 1194
      .local v0, fm:Landroid/graphics/Paint$FontMetricsInt;
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->getFontMetricsInt(Landroid/graphics/Paint$FontMetricsInt;)I
  
+     .line 1195
      return-object v0
  .end method
  
***************
*** 918,923 ****
--- 1006,1012 ----
      .locals 1
  
      .prologue
+     .line 1206
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->getFontMetrics(Landroid/graphics/Paint$FontMetrics;)F
***************
*** 934,939 ****
--- 1023,1029 ----
      .locals 1
  
      .prologue
+     .line 912
      iget-object v0, p0, Landroid/graphics/Paint;->mMaskFilter:Landroid/graphics/MaskFilter;
  
      return-object v0
***************
*** 943,948 ****
--- 1033,1039 ----
      .locals 1
  
      .prologue
+     .line 884
      iget-object v0, p0, Landroid/graphics/Paint;->mPathEffect:Landroid/graphics/PathEffect;
  
      return-object v0
***************
*** 952,957 ****
--- 1043,1049 ----
      .locals 1
  
      .prologue
+     .line 974
      iget-object v0, p0, Landroid/graphics/Paint;->mRasterizer:Landroid/graphics/Rasterizer;
  
      return-object v0
***************
*** 961,966 ****
--- 1053,1059 ----
      .locals 1
  
      .prologue
+     .line 806
      iget-object v0, p0, Landroid/graphics/Paint;->mShader:Landroid/graphics/Shader;
  
      return-object v0
***************
*** 970,975 ****
--- 1063,1069 ----
      .locals 2
  
      .prologue
+     .line 753
      sget-object v0, Landroid/graphics/Paint;->sCapArray:[Landroid/graphics/Paint$Cap;
  
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 987,992 ****
--- 1081,1087 ----
      .locals 2
  
      .prologue
+     .line 772
      sget-object v0, Landroid/graphics/Paint;->sJoinArray:[Landroid/graphics/Paint$Join;
  
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 1010,1015 ****
--- 1105,1111 ----
      .locals 2
  
      .prologue
+     .line 642
      sget-object v0, Landroid/graphics/Paint;->sStyleArray:[Landroid/graphics/Paint$Style;
  
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 1027,1032 ****
--- 1123,1129 ----
      .locals 2
  
      .prologue
+     .line 1030
      sget-object v0, Landroid/graphics/Paint;->sAlignArray:[Landroid/graphics/Paint$Align;
  
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 1048,1053 ****
--- 1145,1151 ----
      .parameter "bounds"
  
      .prologue
+     .line 2076
      or-int v0, p2, p3
  
      sub-int v1, p3, p2
***************
*** 1064,1091 ****
  
      if-gez v0, :cond_0
  
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v0}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v0
  
      :cond_0
      if-nez p4, :cond_1
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "need bounds Rect"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_1
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v0, p1, p2, p3, p4}, Landroid/graphics/Paint;->nativeGetStringBounds(ILjava/lang/String;IILandroid/graphics/Rect;)V
  
      return-void
  .end method
  
--- 1162,1194 ----
  
      if-gez v0, :cond_0
  
+     .line 2077
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v0}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v0
  
+     .line 2079
      :cond_0
      if-nez p4, :cond_1
  
+     .line 2080
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "need bounds Rect"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 2082
      :cond_1
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v0, p1, p2, p3, p4}, Landroid/graphics/Paint;->nativeGetStringBounds(ILjava/lang/String;IILandroid/graphics/Rect;)V
  
+     .line 2083
      return-void
  .end method
  
***************
*** 1097,1102 ****
--- 1200,1206 ----
      .parameter "bounds"
  
      .prologue
+     .line 2096
      or-int v0, p2, p3
  
      if-ltz v0, :cond_0
***************
*** 1107,1112 ****
--- 1211,1217 ----
  
      if-le v0, v1, :cond_1
  
+     .line 2097
      :cond_0
      new-instance v0, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 1114,1135 ****
  
      throw v0
  
      :cond_1
      if-nez p4, :cond_2
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "need bounds Rect"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_2
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v0, p1, p2, p3, p4}, Landroid/graphics/Paint;->nativeGetCharArrayBounds(I[CIILandroid/graphics/Rect;)V
  
      return-void
  .end method
  
--- 1219,1244 ----
  
      throw v0
  
+     .line 2099
      :cond_1
      if-nez p4, :cond_2
  
+     .line 2100
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "need bounds Rect"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 2102
      :cond_2
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v0, p1, p2, p3, p4}, Landroid/graphics/Paint;->nativeGetCharArrayBounds(I[CIILandroid/graphics/Rect;)V
  
+     .line 2103
      return-void
  .end method
  
***************
*** 1144,1159 ****
      .parameter "glyphs"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "text cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      if-eqz p6, :cond_1
  
--- 1253,1271 ----
      .parameter "glyphs"
  
      .prologue
+     .line 1621
      if-nez p1, :cond_0
  
+     .line 1622
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "text cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 1624
      :cond_0
      if-eqz p6, :cond_1
  
***************
*** 1161,1173 ****
  
      if-eq p6, v0, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "unknown flags value: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1273,1286 ----
  
      if-eq p6, v0, :cond_1
  
+     .line 1625
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "unknown flags value: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1185,1190 ****
--- 1298,1304 ----
  
      throw v0
  
+     .line 1627
      :cond_1
      or-int v0, p2, p3
  
***************
*** 1222,1233 ****
--- 1336,1349 ----
  
      if-gez v0, :cond_2
  
+     .line 1630
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v0}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v0
  
+     .line 1632
      :cond_2
      sub-int v0, p3, p2
  
***************
*** 1235,1246 ****
--- 1351,1364 ----
  
      if-le v0, v1, :cond_3
  
+     .line 1633
      new-instance v0, Ljava/lang/ArrayIndexOutOfBoundsException;
  
      invoke-direct {v0}, Ljava/lang/ArrayIndexOutOfBoundsException;-><init>()V
  
      throw v0
  
+     .line 1635
      :cond_3
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
***************
*** 1275,1280 ****
--- 1393,1399 ----
      .parameter "path"
  
      .prologue
+     .line 2058
      or-int v0, p2, p3
  
      sub-int v1, p3, p2
***************
*** 1291,1302 ****
--- 1410,1423 ----
  
      if-gez v0, :cond_0
  
+     .line 2059
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v0}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v0
  
+     .line 2061
      :cond_0
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
***************
*** 1318,1323 ****
--- 1439,1445 ----
  
      invoke-static/range {v0 .. v7}, Landroid/graphics/Paint;->native_getTextPath(IILjava/lang/String;IIFFI)V
  
+     .line 2063
      return-void
  .end method
  
***************
*** 1331,1336 ****
--- 1453,1459 ----
      .parameter "path"
  
      .prologue
+     .line 2036
      or-int v0, p2, p3
  
      if-ltz v0, :cond_0
***************
*** 1341,1346 ****
--- 1464,1470 ----
  
      if-le v0, v1, :cond_1
  
+     .line 2037
      :cond_0
      new-instance v0, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 1348,1353 ****
--- 1472,1478 ----
  
      throw v0
  
+     .line 2039
      :cond_1
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
***************
*** 1369,1374 ****
--- 1494,1500 ----
  
      invoke-static/range {v0 .. v7}, Landroid/graphics/Paint;->native_getTextPath(II[CIIFFI)V
  
+     .line 2041
      return-void
  .end method
  
***************
*** 1384,1389 ****
--- 1510,1516 ----
      .parameter "advancesIndex"
  
      .prologue
+     .line 1711
      const/4 v9, 0x0
  
      move-object v0, p0
***************
*** 1424,1439 ****
      .parameter "reserved"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v4, "text cannot be null"
  
      invoke-direct {v2, v4}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      or-int v2, p2, p3
  
--- 1551,1569 ----
      .parameter "reserved"
  
      .prologue
+     .line 1726
      if-nez p1, :cond_0
  
+     .line 1727
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v4, "text cannot be null"
  
      invoke-direct {v2, v4}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1729
      :cond_0
      or-int v2, p2, p3
  
***************
*** 1472,1483 ****
--- 1602,1615 ----
  
      if-gez v2, :cond_2
  
+     .line 1734
      new-instance v2, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v2}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v2
  
+     .line 1729
      :cond_1
      move-object/from16 v0, p7
  
***************
*** 1491,1496 ****
--- 1623,1629 ----
  
      goto :goto_0
  
+     .line 1737
      :cond_2
      instance-of v2, p1, Ljava/lang/String;
  
***************
*** 1498,1503 ****
--- 1631,1637 ----
  
      move-object v3, p1
  
+     .line 1738
      check-cast v3, Ljava/lang/String;
  
      move-object v2, p0
***************
*** 1522,1530 ****
--- 1656,1666 ----
  
      move-result v12
  
+     .line 1761
      :goto_1
      return v12
  
+     .line 1741
      :cond_3
      instance-of v2, p1, Landroid/text/SpannedString;
  
***************
*** 1534,1539 ****
--- 1670,1676 ----
  
      if-eqz v2, :cond_5
  
+     .line 1743
      :cond_4
      invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
***************
*** 1563,1568 ****
--- 1700,1706 ----
  
      goto :goto_1
  
+     .line 1746
      :cond_5
      instance-of v2, p1, Landroid/text/GraphicsOperations;
  
***************
*** 1570,1575 ****
--- 1708,1714 ----
  
      move-object v2, p1
  
+     .line 1747
      check-cast v2, Landroid/text/GraphicsOperations;
  
      move v3, p2
***************
*** 1594,1599 ****
--- 1733,1739 ----
  
      goto :goto_1
  
+     .line 1750
      :cond_6
      invoke-interface {p1}, Ljava/lang/CharSequence;->length()I
  
***************
*** 1605,1626 ****
--- 1745,1771 ----
  
      if-ne v0, p2, :cond_8
  
+     .line 1751
      :cond_7
      const/4 v12, 0x0
  
      goto :goto_1
  
+     .line 1754
      :cond_8
      sub-int v7, p5, p4
  
+     .line 1755
      .local v7, contextLen:I
      sub-int v5, p3, p2
  
+     .line 1756
      .local v5, len:I
      invoke-static {v7}, Landroid/graphics/TemporaryBuffer;->obtain(I)[C
  
      move-result-object v3
  
+     .line 1757
      .local v3, buf:[C
      const/4 v2, 0x0
  
***************
*** 1630,1635 ****
--- 1775,1781 ----
  
      invoke-static {p1, v0, v1, v3, v2}, Landroid/text/TextUtils;->getChars(Ljava/lang/CharSequence;II[CI)V
  
+     .line 1758
      sub-int v4, p2, p4
  
      const/4 v6, 0x0
***************
*** 1648,1653 ****
--- 1794,1800 ----
  
      move-result v12
  
+     .line 1760
      .local v12, result:F
      invoke-static {v3}, Landroid/graphics/TemporaryBuffer;->recycle([C)V
  
***************
*** 1666,1671 ****
--- 1813,1819 ----
      .parameter "advancesIndex"
  
      .prologue
+     .line 1808
      const/4 v9, 0x0
  
      move-object v0, p0
***************
*** 1706,1721 ****
      .parameter "reserved"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      if-eqz p6, :cond_1
  
--- 1854,1872 ----
      .parameter "reserved"
  
      .prologue
+     .line 1858
      if-nez p1, :cond_0
  
+     .line 1859
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1861
      :cond_0
      if-eqz p6, :cond_1
  
***************
*** 1725,1737 ****
  
      if-eq v0, v2, :cond_1
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "unknown flags value: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1876,1889 ----
  
      if-eq v0, v2, :cond_1
  
+     .line 1862
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "unknown flags value: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1751,1756 ****
--- 1903,1909 ----
  
      throw v2
  
+     .line 1864
      :cond_1
      or-int v2, p2, p3
  
***************
*** 1789,1800 ****
--- 1942,1955 ----
  
      if-gez v2, :cond_3
  
+     .line 1869
      new-instance v2, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v2}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v2
  
+     .line 1864
      :cond_2
      move-object/from16 v0, p7
  
***************
*** 1808,1813 ****
--- 1963,1969 ----
  
      goto :goto_0
  
+     .line 1872
      :cond_3
      invoke-virtual/range {p1 .. p1}, Ljava/lang/String;->length()I
  
***************
*** 1821,1832 ****
--- 1977,1991 ----
  
      if-ne v0, v1, :cond_5
  
+     .line 1873
      :cond_4
      const/4 v2, 0x0
  
+     .line 1892
      :goto_1
      return v2
  
+     .line 1876
      :cond_5
      move-object/from16 v0, p0
  
***************
*** 1834,1839 ****
--- 1993,1999 ----
  
      if-nez v2, :cond_6
  
+     .line 1877
      move-object/from16 v0, p0
  
      iget v2, v0, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 1862,1872 ****
--- 2022,2034 ----
  
      goto :goto_1
  
+     .line 1881
      :cond_6
      invoke-virtual/range {p0 .. p0}, Landroid/graphics/Paint;->getTextSize()F
  
      move-result v14
  
+     .line 1882
      .local v14, oldSize:F
      move-object/from16 v0, p0
  
***************
*** 1878,1883 ****
--- 2040,2046 ----
  
      invoke-virtual {v0, v2}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1883
      move-object/from16 v0, p0
  
      iget v2, v0, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 1904,1916 ****
--- 2067,2082 ----
  
      move-result v15
  
+     .line 1885
      .local v15, totalAdvance:F
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v14}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1887
      if-eqz p7, :cond_7
  
+     .line 1888
      move/from16 v13, p8
  
      .local v13, i:I
***************
*** 1922,1927 ****
--- 2088,2094 ----
      :goto_2
      if-ge v13, v12, :cond_7
  
+     .line 1889
      aget v2, p7, v13
  
      move-object/from16 v0, p0
***************
*** 1932,1941 ****
--- 2099,2110 ----
  
      aput v2, p7, v13
  
+     .line 1888
      add-int/lit8 v13, v13, 0x1
  
      goto :goto_2
  
+     .line 1892
      .end local v12           #e:I
      .end local v13           #i:I
      :cond_7
***************
*** 1960,1965 ****
--- 2129,2135 ----
      .parameter "advancesIndex"
  
      .prologue
+     .line 1649
      const/4 v9, 0x0
  
      move-object v0, p0
***************
*** 2000,2015 ****
      .parameter "reserved"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
!     const-string v2, "text cannot be null"
  
      invoke-direct {v1, v2}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
      :cond_0
      if-eqz p6, :cond_1
  
--- 2170,2188 ----
      .parameter "reserved"
  
      .prologue
+     .line 1664
      if-nez p1, :cond_0
  
+     .line 1665
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v2, "text cannot be null"
  
      invoke-direct {v1, v2}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v1
  
+     .line 1667
      :cond_0
      if-eqz p6, :cond_1
  
***************
*** 2019,2031 ****
  
      if-eq v0, v1, :cond_1
  
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "unknown flags value: "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2192,2205 ----
  
      if-eq v0, v1, :cond_1
  
+     .line 1668
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "unknown flags value: "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 2045,2050 ****
--- 2219,2225 ----
  
      throw v1
  
+     .line 1670
      :cond_1
      or-int v1, p2, p3
  
***************
*** 2089,2100 ****
--- 2264,2277 ----
  
      if-gez v1, :cond_3
  
+     .line 1676
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v1}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v1
  
+     .line 1670
      :cond_2
      move-object/from16 v0, p7
  
***************
*** 2106,2111 ****
--- 2283,2289 ----
  
      goto :goto_0
  
+     .line 1679
      :cond_3
      move-object/from16 v0, p1
  
***************
*** 2115,2131 ****
--- 2293,2313 ----
  
      if-nez p3, :cond_5
  
+     .line 1680
      :cond_4
      const/4 v1, 0x0
  
+     .line 1698
      :goto_1
      return v1
  
+     .line 1682
      :cond_5
      iget-boolean v1, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
      if-nez v1, :cond_6
  
+     .line 1683
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      move-object/from16 v2, p1
***************
*** 2152,2162 ****
--- 2334,2346 ----
  
      goto :goto_1
  
+     .line 1687
      :cond_6
      invoke-virtual {p0}, Landroid/graphics/Paint;->getTextSize()F
  
      move-result v13
  
+     .line 1688
      .local v13, oldSize:F
      iget v1, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
***************
*** 2164,2169 ****
--- 2348,2354 ----
  
      invoke-virtual {p0, v1}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1689
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      move-object/from16 v2, p1
***************
*** 2188,2198 ****
--- 2373,2386 ----
  
      move-result v14
  
+     .line 1691
      .local v14, res:F
      invoke-virtual {p0, v13}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1693
      if-eqz p7, :cond_7
  
+     .line 1694
      move/from16 v12, p8
  
      .local v12, i:I
***************
*** 2202,2207 ****
--- 2390,2396 ----
      :goto_2
      if-ge v12, v11, :cond_7
  
+     .line 1695
      aget v1, p7, v12
  
      iget v2, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
***************
*** 2210,2219 ****
--- 2399,2410 ----
  
      aput v1, p7, v12
  
+     .line 1694
      add-int/lit8 v12, v12, 0x1
  
      goto :goto_2
  
+     .line 1698
      .end local v11           #e:I
      .end local v12           #i:I
      :cond_7
***************
*** 2236,2241 ****
--- 2427,2433 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1964
      instance-of v0, p1, Ljava/lang/String;
  
      if-nez v0, :cond_0
***************
*** 2248,2253 ****
--- 2440,2446 ----
  
      if-eqz v0, :cond_1
  
+     .line 1966
      :cond_0
      invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
***************
*** 2269,2277 ****
--- 2462,2472 ----
  
      move-result v7
  
+     .line 1979
      :goto_0
      return v7
  
+     .line 1969
      :cond_1
      instance-of v0, p1, Landroid/text/GraphicsOperations;
  
***************
*** 2279,2284 ****
--- 2474,2480 ----
  
      move-object v0, p1
  
+     .line 1970
      check-cast v0, Landroid/text/GraphicsOperations;
  
      move v1, p2
***************
*** 2299,2315 ****
--- 2495,2515 ----
  
      goto :goto_0
  
+     .line 1974
      :cond_2
      sub-int v3, p3, p2
  
+     .line 1975
      .local v3, contextLen:I
      invoke-static {v3}, Landroid/graphics/TemporaryBuffer;->obtain(I)[C
  
      move-result-object v1
  
+     .line 1976
      .local v1, buf:[C
      invoke-static {p1, p2, p3, v1, v2}, Landroid/text/TextUtils;->getChars(Ljava/lang/CharSequence;II[CI)V
  
+     .line 1977
      sub-int v5, p5, p2
  
      move-object v0, p0
***************
*** 2322,2327 ****
--- 2522,2528 ----
  
      move-result v7
  
+     .line 1978
      .local v7, result:I
      invoke-static {v1}, Landroid/graphics/TemporaryBuffer;->recycle([C)V
  
***************
*** 2338,2343 ****
--- 2539,2545 ----
      .parameter "cursorOpt"
  
      .prologue
+     .line 2010
      or-int v0, p2, p3
  
      or-int/2addr v0, p5
***************
*** 2370,2375 ****
--- 2572,2578 ----
  
      if-le p6, v0, :cond_1
  
+     .line 2014
      :cond_0
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 2377,2382 ****
--- 2580,2586 ----
  
      throw v0
  
+     .line 2017
      :cond_1
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
***************
*** 2411,2418 ****
--- 2615,2624 ----
      .parameter "cursorOpt"
  
      .prologue
+     .line 1923
      add-int v8, p2, p3
  
+     .line 1924
      .local v8, contextEnd:I
      or-int v0, p2, v8
  
***************
*** 2444,2449 ****
--- 2650,2656 ----
  
      if-le p6, v0, :cond_1
  
+     .line 1928
      :cond_0
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 2451,2456 ****
--- 2658,2664 ----
  
      throw v0
  
+     .line 1931
      :cond_1
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
***************
*** 2494,2509 ****
      .prologue
      const/4 v1, 0x0
  
      if-nez p1, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      or-int v2, p2, p3
  
--- 2702,2720 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 1517
      if-nez p1, :cond_0
  
+     .line 1518
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1520
      :cond_0
      or-int v2, p2, p3
  
***************
*** 2521,2532 ****
--- 2732,2745 ----
  
      if-gez v2, :cond_1
  
+     .line 1521
      new-instance v2, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v2}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v2
  
+     .line 1523
      :cond_1
      sub-int v2, p3, p2
  
***************
*** 2534,2545 ****
--- 2747,2760 ----
  
      if-le v2, v3, :cond_2
  
+     .line 1524
      new-instance v2, Ljava/lang/ArrayIndexOutOfBoundsException;
  
      invoke-direct {v2}, Ljava/lang/ArrayIndexOutOfBoundsException;-><init>()V
  
      throw v2
  
+     .line 1527
      :cond_2
      invoke-interface {p1}, Ljava/lang/CharSequence;->length()I
  
***************
*** 2549,2565 ****
--- 2764,2783 ----
  
      if-ne p2, p3, :cond_4
  
+     .line 1546
      .end local p1
      :cond_3
      :goto_0
      return v1
  
+     .line 1530
      .restart local p1
      :cond_4
      instance-of v2, p1, Ljava/lang/String;
  
      if-eqz v2, :cond_5
  
+     .line 1531
      check-cast p1, Ljava/lang/String;
  
      .end local p1
***************
*** 2569,2574 ****
--- 2787,2793 ----
  
      goto :goto_0
  
+     .line 1533
      .restart local p1
      :cond_5
      instance-of v2, p1, Landroid/text/SpannedString;
***************
*** 2579,2584 ****
--- 2798,2804 ----
  
      if-eqz v2, :cond_7
  
+     .line 1535
      :cond_6
      invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
***************
*** 2590,2600 ****
--- 2810,2822 ----
  
      goto :goto_0
  
+     .line 1537
      :cond_7
      instance-of v2, p1, Landroid/text/GraphicsOperations;
  
      if-eqz v2, :cond_8
  
+     .line 1538
      check-cast p1, Landroid/text/GraphicsOperations;
  
      .end local p1
***************
*** 2604,2609 ****
--- 2826,2832 ----
  
      goto :goto_0
  
+     .line 1542
      .restart local p1
      :cond_8
      sub-int v2, p3, p2
***************
*** 2612,2626 ****
--- 2835,2852 ----
  
      move-result-object v0
  
+     .line 1543
      .local v0, buf:[C
      invoke-static {p1, p2, p3, v0, v1}, Landroid/text/TextUtils;->getChars(Ljava/lang/CharSequence;II[CI)V
  
+     .line 1544
      sub-int v2, p3, p2
  
      invoke-virtual {p0, v0, v1, v2, p4}, Landroid/graphics/Paint;->getTextWidths([CII[F)I
  
      move-result v1
  
+     .line 1545
      .local v1, result:I
      invoke-static {v0}, Landroid/graphics/TemporaryBuffer;->recycle([C)V
  
***************
*** 2635,2650 ****
      .parameter "widths"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v3, Ljava/lang/IllegalArgumentException;
  
!     const-string v4, "text cannot be null"
  
      invoke-direct {v3, v4}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v3
  
      :cond_0
      or-int v3, p2, p3
  
--- 2861,2879 ----
      .parameter "widths"
  
      .prologue
+     .line 1560
      if-nez p1, :cond_0
  
+     .line 1561
      new-instance v3, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v4, "text cannot be null"
  
      invoke-direct {v3, v4}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v3
  
+     .line 1563
      :cond_0
      or-int v3, p2, p3
  
***************
*** 2662,2673 ****
--- 2891,2904 ----
  
      if-gez v3, :cond_1
  
+     .line 1564
      new-instance v3, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v3}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v3
  
+     .line 1566
      :cond_1
      sub-int v3, p3, p2
  
***************
*** 2675,2686 ****
--- 2906,2919 ----
  
      if-le v3, v4, :cond_2
  
+     .line 1567
      new-instance v3, Ljava/lang/ArrayIndexOutOfBoundsException;
  
      invoke-direct {v3}, Ljava/lang/ArrayIndexOutOfBoundsException;-><init>()V
  
      throw v3
  
+     .line 1570
      :cond_2
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
***************
*** 2690,2707 ****
--- 2923,2944 ----
  
      if-ne p2, p3, :cond_5
  
+     .line 1571
      :cond_3
      const/4 v2, 0x0
  
+     .line 1584
      :cond_4
      :goto_0
      return v2
  
+     .line 1573
      :cond_5
      iget-boolean v3, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
      if-nez v3, :cond_6
  
+     .line 1574
      iget v3, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v3, p1, p2, p3, p4}, Landroid/graphics/Paint;->native_getTextWidths(ILjava/lang/String;II[F)I
***************
*** 2710,2720 ****
--- 2947,2959 ----
  
      goto :goto_0
  
+     .line 1577
      :cond_6
      invoke-virtual {p0}, Landroid/graphics/Paint;->getTextSize()F
  
      move-result v1
  
+     .line 1578
      .local v1, oldSize:F
      iget v3, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
***************
*** 2722,2742 ****
--- 2961,2985 ----
  
      invoke-virtual {p0, v3}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1579
      iget v3, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v3, p1, p2, p3, p4}, Landroid/graphics/Paint;->native_getTextWidths(ILjava/lang/String;II[F)I
  
      move-result v2
  
+     .line 1580
      .local v2, res:I
      invoke-virtual {p0, v1}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1581
      const/4 v0, 0x0
  
      .local v0, i:I
      :goto_1
      if-ge v0, v2, :cond_4
  
+     .line 1582
      aget v3, p4, v0
  
      iget v4, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
***************
*** 2745,2750 ****
--- 2988,2994 ----
  
      aput v3, p4, v0
  
+     .line 1581
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
***************
*** 2756,2761 ****
--- 3000,3006 ----
      .parameter "widths"
  
      .prologue
+     .line 1596
      const/4 v0, 0x0
  
      invoke-virtual {p1}, Ljava/lang/String;->length()I
***************
*** 2777,2792 ****
      .parameter "widths"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v3, Ljava/lang/IllegalArgumentException;
  
!     const-string v4, "text cannot be null"
  
      invoke-direct {v3, v4}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v3
  
      :cond_0
      or-int v3, p2, p3
  
--- 3022,3040 ----
      .parameter "widths"
  
      .prologue
+     .line 1480
      if-nez p1, :cond_0
  
+     .line 1481
      new-instance v3, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v4, "text cannot be null"
  
      invoke-direct {v3, v4}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v3
  
+     .line 1483
      :cond_0
      or-int v3, p2, p3
  
***************
*** 2802,2807 ****
--- 3050,3056 ----
  
      if-le p3, v3, :cond_2
  
+     .line 1485
      :cond_1
      new-instance v3, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 2809,2814 ****
--- 3058,3064 ----
  
      throw v3
  
+     .line 1488
      :cond_2
      array-length v3, p1
  
***************
*** 2816,2833 ****
--- 3066,3087 ----
  
      if-nez p3, :cond_5
  
+     .line 1489
      :cond_3
      const/4 v2, 0x0
  
+     .line 1502
      :cond_4
      :goto_0
      return v2
  
+     .line 1491
      :cond_5
      iget-boolean v3, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
      if-nez v3, :cond_6
  
+     .line 1492
      iget v3, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v3, p1, p2, p3, p4}, Landroid/graphics/Paint;->native_getTextWidths(I[CII[F)I
***************
*** 2836,2846 ****
--- 3090,3102 ----
  
      goto :goto_0
  
+     .line 1495
      :cond_6
      invoke-virtual {p0}, Landroid/graphics/Paint;->getTextSize()F
  
      move-result v1
  
+     .line 1496
      .local v1, oldSize:F
      iget v3, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
***************
*** 2848,2868 ****
--- 3104,3128 ----
  
      invoke-virtual {p0, v3}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1497
      iget v3, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v3, p1, p2, p3, p4}, Landroid/graphics/Paint;->native_getTextWidths(I[CII[F)I
  
      move-result v2
  
+     .line 1498
      .local v2, res:I
      invoke-virtual {p0, v1}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1499
      const/4 v0, 0x0
  
      .local v0, i:I
      :goto_1
      if-ge v0, v2, :cond_4
  
+     .line 1500
      aget v3, p4, v0
  
      iget v4, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
***************
*** 2871,2876 ****
--- 3131,3137 ----
  
      aput v3, p4, v0
  
+     .line 1499
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
***************
*** 2880,2885 ****
--- 3141,3147 ----
      .locals 1
  
      .prologue
+     .line 944
      iget-object v0, p0, Landroid/graphics/Paint;->mTypeface:Landroid/graphics/Typeface;
  
      return-object v0
***************
*** 2889,2894 ****
--- 3151,3157 ----
      .locals 1
  
      .prologue
+     .line 857
      iget-object v0, p0, Landroid/graphics/Paint;->mXfermode:Landroid/graphics/Xfermode;
  
      return-object v0
***************
*** 2898,2903 ****
--- 3161,3167 ----
      .locals 1
  
      .prologue
+     .line 487
      invoke-virtual {p0}, Landroid/graphics/Paint;->getFlags()I
  
      move-result v0
***************
*** 2921,2926 ****
--- 3185,3191 ----
      .locals 1
  
      .prologue
+     .line 511
      invoke-virtual {p0}, Landroid/graphics/Paint;->getFlags()I
  
      move-result v0
***************
*** 2944,2949 ****
--- 3209,3215 ----
      .locals 1
  
      .prologue
+     .line 600
      invoke-virtual {p0}, Landroid/graphics/Paint;->getFlags()I
  
      move-result v0
***************
*** 2967,2972 ****
--- 3233,3239 ----
      .locals 1
  
      .prologue
+     .line 620
      invoke-virtual {p0}, Landroid/graphics/Paint;->getFlags()I
  
      move-result v0
***************
*** 2990,2995 ****
--- 3257,3263 ----
      .locals 1
  
      .prologue
+     .line 532
      invoke-virtual {p0}, Landroid/graphics/Paint;->getFlags()I
  
      move-result v0
***************
*** 3013,3018 ****
--- 3281,3287 ----
      .locals 1
  
      .prologue
+     .line 583
      invoke-virtual {p0}, Landroid/graphics/Paint;->getFlags()I
  
      move-result v0
***************
*** 3036,3041 ****
--- 3305,3311 ----
      .locals 1
  
      .prologue
+     .line 549
      invoke-virtual {p0}, Landroid/graphics/Paint;->getFlags()I
  
      move-result v0
***************
*** 3059,3064 ****
--- 3329,3335 ----
      .locals 1
  
      .prologue
+     .line 566
      invoke-virtual {p0}, Landroid/graphics/Paint;->getFlags()I
  
      move-result v0
***************
*** 3087,3102 ****
      .prologue
      const/4 v4, 0x0
  
      if-nez p1, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      or-int v2, p2, p3
  
--- 3358,3376 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 1307
      if-nez p1, :cond_0
  
+     .line 1308
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1310
      :cond_0
      or-int v2, p2, p3
  
***************
*** 3114,3125 ****
--- 3388,3401 ----
  
      if-gez v2, :cond_1
  
+     .line 1311
      new-instance v2, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v2}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v2
  
+     .line 1314
      :cond_1
      invoke-interface {p1}, Ljava/lang/CharSequence;->length()I
  
***************
*** 3129,3147 ****
--- 3405,3427 ----
  
      if-ne p2, p3, :cond_3
  
+     .line 1315
      :cond_2
      const/4 v1, 0x0
  
+     .line 1332
      .end local p1
      :goto_0
      return v1
  
+     .line 1317
      .restart local p1
      :cond_3
      instance-of v2, p1, Ljava/lang/String;
  
      if-eqz v2, :cond_4
  
+     .line 1318
      check-cast p1, Ljava/lang/String;
  
      .end local p1
***************
*** 3151,3156 ****
--- 3431,3437 ----
  
      goto :goto_0
  
+     .line 1320
      .restart local p1
      :cond_4
      instance-of v2, p1, Landroid/text/SpannedString;
***************
*** 3161,3166 ****
--- 3442,3448 ----
  
      if-eqz v2, :cond_6
  
+     .line 1322
      :cond_5
      invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
***************
*** 3172,3182 ****
--- 3454,3466 ----
  
      goto :goto_0
  
+     .line 1324
      :cond_6
      instance-of v2, p1, Landroid/text/GraphicsOperations;
  
      if-eqz v2, :cond_7
  
+     .line 1325
      check-cast p1, Landroid/text/GraphicsOperations;
  
      .end local p1
***************
*** 3186,3191 ****
--- 3470,3476 ----
  
      goto :goto_0
  
+     .line 1328
      .restart local p1
      :cond_7
      sub-int v2, p3, p2
***************
*** 3194,3208 ****
--- 3479,3496 ----
  
      move-result-object v0
  
+     .line 1329
      .local v0, buf:[C
      invoke-static {p1, p2, p3, v0, v4}, Landroid/text/TextUtils;->getChars(Ljava/lang/CharSequence;II[CI)V
  
+     .line 1330
      sub-int v2, p3, p2
  
      invoke-virtual {p0, v0, v4, v2}, Landroid/graphics/Paint;->measureText([CII)F
  
      move-result v1
  
+     .line 1331
      .local v1, result:F
      invoke-static {v0}, Landroid/graphics/TemporaryBuffer;->recycle([C)V
  
***************
*** 3214,3229 ****
      .parameter "text"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
--- 3502,3520 ----
      .parameter "text"
  
      .prologue
+     .line 1280
      if-nez p1, :cond_0
  
+     .line 1281
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1284
      :cond_0
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
***************
*** 3231,3241 ****
--- 3522,3535 ----
  
      if-nez v2, :cond_1
  
+     .line 1285
      const/4 v2, 0x0
  
+     .line 1293
      :goto_0
      return v2
  
+     .line 1288
      :cond_1
      iget-boolean v2, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
***************
*** 3247,3257 ****
--- 3541,3553 ----
  
      goto :goto_0
  
+     .line 1289
      :cond_2
      invoke-virtual {p0}, Landroid/graphics/Paint;->getTextSize()F
  
      move-result v0
  
+     .line 1290
      .local v0, oldSize:F
      iget v2, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
***************
*** 3259,3271 ****
--- 3555,3570 ----
  
      invoke-virtual {p0, v2}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1291
      invoke-direct {p0, p1}, Landroid/graphics/Paint;->native_measureText(Ljava/lang/String;)F
  
      move-result v1
  
+     .line 1292
      .local v1, w:F
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1293
      iget v2, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
  
      mul-float/2addr v2, v1
***************
*** 3280,3295 ****
      .parameter "end"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      or-int v2, p2, p3
  
--- 3579,3597 ----
      .parameter "end"
  
      .prologue
+     .line 1250
      if-nez p1, :cond_0
  
+     .line 1251
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1253
      :cond_0
      or-int v2, p2, p3
  
***************
*** 3307,3318 ****
--- 3609,3622 ----
  
      if-gez v2, :cond_1
  
+     .line 1254
      new-instance v2, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v2}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v2
  
+     .line 1257
      :cond_1
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
***************
*** 3322,3349 ****
--- 3626,3659 ----
  
      if-ne p2, p3, :cond_3
  
+     .line 1258
      :cond_2
      const/4 v2, 0x0
  
+     .line 1268
      :goto_0
      return v2
  
+     .line 1260
      :cond_3
      iget-boolean v2, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
      if-nez v2, :cond_4
  
+     .line 1261
      invoke-direct {p0, p1, p2, p3}, Landroid/graphics/Paint;->native_measureText(Ljava/lang/String;II)F
  
      move-result v2
  
      goto :goto_0
  
+     .line 1264
      :cond_4
      invoke-virtual {p0}, Landroid/graphics/Paint;->getTextSize()F
  
      move-result v0
  
+     .line 1265
      .local v0, oldSize:F
      iget v2, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
***************
*** 3351,3363 ****
--- 3661,3676 ----
  
      invoke-virtual {p0, v2}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1266
      invoke-direct {p0, p1, p2, p3}, Landroid/graphics/Paint;->native_measureText(Ljava/lang/String;II)F
  
      move-result v1
  
+     .line 1267
      .local v1, w:F
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1268
      iget v2, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
  
      mul-float/2addr v2, v1
***************
*** 3372,3387 ****
      .parameter "count"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      :cond_0
      or-int v2, p2, p3
  
--- 3685,3703 ----
      .parameter "count"
  
      .prologue
+     .line 1218
      if-nez p1, :cond_0
  
+     .line 1219
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v3, "text cannot be null"
  
      invoke-direct {v2, v3}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1221
      :cond_0
      or-int v2, p2, p3
  
***************
*** 3393,3398 ****
--- 3709,3715 ----
  
      if-le v2, v3, :cond_2
  
+     .line 1222
      :cond_1
      new-instance v2, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 3400,3405 ****
--- 3717,3723 ----
  
      throw v2
  
+     .line 1225
      :cond_2
      array-length v2, p1
  
***************
*** 3407,3434 ****
--- 3725,3758 ----
  
      if-nez p3, :cond_4
  
+     .line 1226
      :cond_3
      const/4 v2, 0x0
  
+     .line 1236
      :goto_0
      return v2
  
+     .line 1228
      :cond_4
      iget-boolean v2, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
      if-nez v2, :cond_5
  
+     .line 1229
      invoke-direct {p0, p1, p2, p3}, Landroid/graphics/Paint;->native_measureText([CII)F
  
      move-result v2
  
      goto :goto_0
  
+     .line 1232
      :cond_5
      invoke-virtual {p0}, Landroid/graphics/Paint;->getTextSize()F
  
      move-result v0
  
+     .line 1233
      .local v0, oldSize:F
      iget v2, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
***************
*** 3436,3448 ****
--- 3760,3775 ----
  
      invoke-virtual {p0, v2}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1234
      invoke-direct {p0, p1, p2, p3}, Landroid/graphics/Paint;->native_measureText([CII)F
  
      move-result v1
  
+     .line 1235
      .local v1, w:F
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->setTextSize(F)V
  
+     .line 1236
      iget v2, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
  
      mul-float/2addr v2, v1
***************
*** 3454,3481 ****
--- 3781,3814 ----
      .locals 1
  
      .prologue
+     .line 367
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v0}, Landroid/graphics/Paint;->native_reset(I)V
  
+     .line 368
      const/16 v0, 0x100
  
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->setFlags(I)V
  
+     .line 373
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
+     .line 374
      const/high16 v0, 0x3f80
  
      iput v0, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
  
      iput v0, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
+     .line 375
      const/4 v0, 0x2
  
      iput v0, p0, Landroid/graphics/Paint;->mBidiFlags:I
  
+     .line 376
      return-void
  .end method
  
***************
*** 3484,3499 ****
--- 3817,3836 ----
      .parameter "src"
  
      .prologue
+     .line 384
      if-eq p0, p1, :cond_0
  
+     .line 386
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      iget v1, p1, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v0, v1}, Landroid/graphics/Paint;->native_set(II)V
  
+     .line 387
      invoke-direct {p0, p1}, Landroid/graphics/Paint;->setClassVariablesFrom(Landroid/graphics/Paint;)V
  
+     .line 389
      :cond_0
      return-void
  .end method
***************
*** 3506,3511 ****
--- 3843,3849 ----
      .parameter "b"
  
      .prologue
+     .line 703
      shl-int/lit8 v0, p1, 0x18
  
      shl-int/lit8 v1, p2, 0x10
***************
*** 3520,3525 ****
--- 3858,3864 ----
  
      invoke-virtual {p0, v0}, Landroid/graphics/Paint;->setColor(I)V
  
+     .line 704
      return-void
  .end method
  
***************
*** 3534,3552 ****
      .parameter "flags"
  
      .prologue
      and-int/lit8 p1, p1, 0x7
  
      const/4 v0, 0x5
  
      if-le p1, v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "unknown bidi flag: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 3873,3894 ----
      .parameter "flags"
  
      .prologue
+     .line 445
      and-int/lit8 p1, p1, 0x7
  
+     .line 446
      const/4 v0, 0x5
  
      if-le p1, v0, :cond_0
  
+     .line 447
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "unknown bidi flag: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 3564,3572 ****
--- 3906,3916 ----
  
      throw v0
  
+     .line 449
      :cond_0
      iput p1, p0, Landroid/graphics/Paint;->mBidiFlags:I
  
+     .line 450
      return-void
  .end method
  
***************
*** 3578,3597 ****
--- 3922,3947 ----
      .parameter "filter"
  
      .prologue
+     .line 843
      const/4 v0, 0x0
  
+     .line 844
      .local v0, filterNative:I
      if-eqz p1, :cond_0
  
+     .line 845
      iget v0, p1, Landroid/graphics/ColorFilter;->native_instance:I
  
+     .line 846
      :cond_0
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v1, v0}, Landroid/graphics/Paint;->native_setColorFilter(II)I
  
+     .line 847
      iput-object p1, p0, Landroid/graphics/Paint;->mColorFilter:Landroid/graphics/ColorFilter;
  
+     .line 848
      return-object p1
  .end method
  
***************
*** 3602,3607 ****
--- 3952,3958 ----
      .prologue
      const/high16 v4, 0x3f80
  
+     .line 419
      float-to-double v0, p1
  
      const-wide/high16 v2, 0x3ff0
***************
*** 3610,3633 ****
--- 3961,3990 ----
  
      if-nez v0, :cond_0
  
+     .line 420
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
+     .line 421
      iput v4, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
  
      iput v4, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
+     .line 427
      :goto_0
      return-void
  
+     .line 423
      :cond_0
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/graphics/Paint;->mHasCompatScaling:Z
  
+     .line 424
      iput p1, p0, Landroid/graphics/Paint;->mCompatScaling:F
  
+     .line 425
      div-float v0, v4, p1
  
      iput v0, p0, Landroid/graphics/Paint;->mInvCompatScaling:F
***************
*** 3658,3677 ****
--- 4015,4040 ----
      .parameter "maskfilter"
  
      .prologue
+     .line 926
      const/4 v0, 0x0
  
+     .line 927
      .local v0, maskfilterNative:I
      if-eqz p1, :cond_0
  
+     .line 928
      iget v0, p1, Landroid/graphics/MaskFilter;->native_instance:I
  
+     .line 930
      :cond_0
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v1, v0}, Landroid/graphics/Paint;->native_setMaskFilter(II)I
  
+     .line 931
      iput-object p1, p0, Landroid/graphics/Paint;->mMaskFilter:Landroid/graphics/MaskFilter;
  
+     .line 932
      return-object p1
  .end method
  
***************
*** 3680,3699 ****
--- 4043,4068 ----
      .parameter "effect"
  
      .prologue
+     .line 897
      const/4 v0, 0x0
  
+     .line 898
      .local v0, effectNative:I
      if-eqz p1, :cond_0
  
+     .line 899
      iget v0, p1, Landroid/graphics/PathEffect;->native_instance:I
  
+     .line 901
      :cond_0
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v1, v0}, Landroid/graphics/Paint;->native_setPathEffect(II)I
  
+     .line 902
      iput-object p1, p0, Landroid/graphics/Paint;->mPathEffect:Landroid/graphics/PathEffect;
  
+     .line 903
      return-object p1
  .end method
  
***************
*** 3702,3721 ****
--- 4071,4096 ----
      .parameter "rasterizer"
  
      .prologue
+     .line 988
      const/4 v0, 0x0
  
+     .line 989
      .local v0, rasterizerNative:I
      if-eqz p1, :cond_0
  
+     .line 990
      iget v0, p1, Landroid/graphics/Rasterizer;->native_instance:I
  
+     .line 992
      :cond_0
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v1, v0}, Landroid/graphics/Paint;->native_setRasterizer(II)I
  
+     .line 993
      iput-object p1, p0, Landroid/graphics/Paint;->mRasterizer:Landroid/graphics/Rasterizer;
  
+     .line 994
      return-object p1
  .end method
  
***************
*** 3724,3743 ****
--- 4099,4124 ----
      .parameter "shader"
  
      .prologue
+     .line 819
      const/4 v0, 0x0
  
+     .line 820
      .local v0, shaderNative:I
      if-eqz p1, :cond_0
  
+     .line 821
      iget v0, p1, Landroid/graphics/Shader;->native_instance:I
  
+     .line 822
      :cond_0
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v1, v0}, Landroid/graphics/Paint;->native_setShader(II)I
  
+     .line 823
      iput-object p1, p0, Landroid/graphics/Paint;->mShader:Landroid/graphics/Shader;
  
+     .line 824
      return-object p1
  .end method
  
***************
*** 3749,3754 ****
--- 4130,4136 ----
      .parameter "color"
  
      .prologue
+     .line 1003
      const/4 v0, 0x0
  
      cmpl-float v0, p1, v0
***************
*** 3760,3777 ****
--- 4142,4166 ----
      :goto_0
      iput-boolean v0, p0, Landroid/graphics/Paint;->hasShadow:Z
  
+     .line 1004
      iput p1, p0, Landroid/graphics/Paint;->shadowRadius:F
  
+     .line 1005
      iput p2, p0, Landroid/graphics/Paint;->shadowDx:F
  
+     .line 1006
      iput p3, p0, Landroid/graphics/Paint;->shadowDy:F
  
+     .line 1007
      iput p4, p0, Landroid/graphics/Paint;->shadowColor:I
  
+     .line 1008
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/graphics/Paint;->nSetShadowLayer(FFFI)V
  
+     .line 1009
      return-void
  
+     .line 1003
      :cond_0
      const/4 v0, 0x0
  
***************
*** 3786,3797 ****
--- 4175,4188 ----
      .parameter "cap"
  
      .prologue
+     .line 763
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      iget v1, p1, Landroid/graphics/Paint$Cap;->nativeInt:I
  
      invoke-static {v0, v1}, Landroid/graphics/Paint;->native_setStrokeCap(II)V
  
+     .line 764
      return-void
  .end method
  
***************
*** 3800,3811 ****
--- 4191,4204 ----
      .parameter "join"
  
      .prologue
+     .line 782
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      iget v1, p1, Landroid/graphics/Paint$Join;->nativeInt:I
  
      invoke-static {v0, v1}, Landroid/graphics/Paint;->native_setStrokeJoin(II)V
  
+     .line 783
      return-void
  .end method
  
***************
*** 3820,3831 ****
--- 4213,4226 ----
      .parameter "style"
  
      .prologue
+     .line 653
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      iget v1, p1, Landroid/graphics/Paint$Style;->nativeInt:I
  
      invoke-static {v0, v1}, Landroid/graphics/Paint;->native_setStyle(II)V
  
+     .line 654
      return-void
  .end method
  
***************
*** 3837,3848 ****
--- 4232,4245 ----
      .parameter "align"
  
      .prologue
+     .line 1042
      iget v0, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      iget v1, p1, Landroid/graphics/Paint$Align;->nativeInt:I
  
      invoke-static {v0, v1}, Landroid/graphics/Paint;->native_setTextAlign(II)V
  
+     .line 1043
      return-void
  .end method
  
***************
*** 3860,3879 ****
--- 4257,4282 ----
      .parameter "typeface"
  
      .prologue
+     .line 957
      const/4 v0, 0x0
  
+     .line 958
      .local v0, typefaceNative:I
      if-eqz p1, :cond_0
  
+     .line 959
      iget v0, p1, Landroid/graphics/Typeface;->native_instance:I
  
+     .line 961
      :cond_0
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v1, v0}, Landroid/graphics/Paint;->native_setTypeface(II)I
  
+     .line 962
      iput-object p1, p0, Landroid/graphics/Paint;->mTypeface:Landroid/graphics/Typeface;
  
+     .line 963
      return-object p1
  .end method
  
***************
*** 3885,3903 ****
--- 4288,4312 ----
      .parameter "xfermode"
  
      .prologue
+     .line 870
      const/4 v0, 0x0
  
+     .line 871
      .local v0, xfermodeNative:I
      if-eqz p1, :cond_0
  
+     .line 872
      iget v0, p1, Landroid/graphics/Xfermode;->native_instance:I
  
+     .line 873
      :cond_0
      iget v1, p0, Landroid/graphics/Paint;->mNativePaint:I
  
      invoke-static {v1, v0}, Landroid/graphics/Paint;->native_setXfermode(II)I
  
+     .line 874
      iput-object p1, p0, Landroid/graphics/Paint;->mXfermode:Landroid/graphics/Xfermode;
  
+     .line 875
      return-object p1
  .end method
