*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/PathDashPathEffect$Style.smali	2012-03-23 19:35:41.728347801 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/PathDashPathEffect$Style.smali.original	2012-03-23 19:34:44.780309355 -0500
***************
*** 48,53 ****
--- 48,54 ----
  
      const/4 v2, 0x0
  
+     .line 22
      new-instance v0, Landroid/graphics/PathDashPathEffect$Style;
  
      const-string v1, "TRANSLATE"
***************
*** 56,61 ****
--- 57,63 ----
  
      sput-object v0, Landroid/graphics/PathDashPathEffect$Style;->TRANSLATE:Landroid/graphics/PathDashPathEffect$Style;
  
+     .line 23
      new-instance v0, Landroid/graphics/PathDashPathEffect$Style;
  
      const-string v1, "ROTATE"
***************
*** 64,69 ****
--- 66,72 ----
  
      sput-object v0, Landroid/graphics/PathDashPathEffect$Style;->ROTATE:Landroid/graphics/PathDashPathEffect$Style;
  
+     .line 24
      new-instance v0, Landroid/graphics/PathDashPathEffect$Style;
  
      const-string v1, "MORPH"
***************
*** 72,77 ****
--- 75,81 ----
  
      sput-object v0, Landroid/graphics/PathDashPathEffect$Style;->MORPH:Landroid/graphics/PathDashPathEffect$Style;
  
+     .line 21
      const/4 v0, 0x3
  
      new-array v0, v0, [Landroid/graphics/PathDashPathEffect$Style;
***************
*** 105,114 ****
--- 109,121 ----
      .end annotation
  
      .prologue
+     .line 26
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 27
      iput p3, p0, Landroid/graphics/PathDashPathEffect$Style;->native_style:I
  
+     .line 28
      return-void
  .end method
  
***************
*** 117,122 ****
--- 124,130 ----
      .parameter "name"
  
      .prologue
+     .line 21
      const-class v0, Landroid/graphics/PathDashPathEffect$Style;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 132,137 ****
--- 140,146 ----
      .locals 1
  
      .prologue
+     .line 21
      sget-object v0, Landroid/graphics/PathDashPathEffect$Style;->$VALUES:[Landroid/graphics/PathDashPathEffect$Style;
  
      invoke-virtual {v0}, [Landroid/graphics/PathDashPathEffect$Style;->clone()Ljava/lang/Object;
