*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/PorterDuff.smali	2012-03-23 19:35:45.295412847 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/PorterDuff.smali.original	2012-03-23 19:34:44.822310120 -0500
***************
*** 16,22 ****
--- 16,24 ----
      .locals 0
  
      .prologue
+     .line 19
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 22
      return-void
  .end method
