*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/PorterDuff$Mode.smali	2012-03-23 19:35:44.236393536 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/PorterDuff$Mode.smali.original	2012-03-23 19:34:44.814309975 -0500
***************
*** 82,87 ****
--- 82,88 ----
  
      const/4 v4, 0x0
  
+     .line 24
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "CLEAR"
***************
*** 90,95 ****
--- 91,97 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->CLEAR:Landroid/graphics/PorterDuff$Mode;
  
+     .line 26
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "SRC"
***************
*** 98,103 ****
--- 100,106 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->SRC:Landroid/graphics/PorterDuff$Mode;
  
+     .line 28
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "DST"
***************
*** 106,111 ****
--- 109,115 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->DST:Landroid/graphics/PorterDuff$Mode;
  
+     .line 30
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "SRC_OVER"
***************
*** 114,119 ****
--- 118,124 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->SRC_OVER:Landroid/graphics/PorterDuff$Mode;
  
+     .line 32
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "DST_OVER"
***************
*** 122,127 ****
--- 127,133 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->DST_OVER:Landroid/graphics/PorterDuff$Mode;
  
+     .line 34
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "SRC_IN"
***************
*** 134,139 ****
--- 140,146 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->SRC_IN:Landroid/graphics/PorterDuff$Mode;
  
+     .line 36
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "DST_IN"
***************
*** 146,151 ****
--- 153,159 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->DST_IN:Landroid/graphics/PorterDuff$Mode;
  
+     .line 38
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "SRC_OUT"
***************
*** 158,163 ****
--- 166,172 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->SRC_OUT:Landroid/graphics/PorterDuff$Mode;
  
+     .line 40
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "DST_OUT"
***************
*** 170,175 ****
--- 179,185 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->DST_OUT:Landroid/graphics/PorterDuff$Mode;
  
+     .line 42
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "SRC_ATOP"
***************
*** 182,187 ****
--- 192,198 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->SRC_ATOP:Landroid/graphics/PorterDuff$Mode;
  
+     .line 44
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "DST_ATOP"
***************
*** 194,199 ****
--- 205,211 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->DST_ATOP:Landroid/graphics/PorterDuff$Mode;
  
+     .line 46
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "XOR"
***************
*** 206,211 ****
--- 218,224 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->XOR:Landroid/graphics/PorterDuff$Mode;
  
+     .line 49
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "DARKEN"
***************
*** 218,223 ****
--- 231,237 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->DARKEN:Landroid/graphics/PorterDuff$Mode;
  
+     .line 52
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "LIGHTEN"
***************
*** 230,235 ****
--- 244,250 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->LIGHTEN:Landroid/graphics/PorterDuff$Mode;
  
+     .line 54
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "MULTIPLY"
***************
*** 242,247 ****
--- 257,263 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->MULTIPLY:Landroid/graphics/PorterDuff$Mode;
  
+     .line 56
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "SCREEN"
***************
*** 254,259 ****
--- 270,276 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->SCREEN:Landroid/graphics/PorterDuff$Mode;
  
+     .line 58
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "ADD"
***************
*** 266,271 ****
--- 283,289 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->ADD:Landroid/graphics/PorterDuff$Mode;
  
+     .line 59
      new-instance v0, Landroid/graphics/PorterDuff$Mode;
  
      const-string v1, "OVERLAY"
***************
*** 278,283 ****
--- 296,302 ----
  
      sput-object v0, Landroid/graphics/PorterDuff$Mode;->OVERLAY:Landroid/graphics/PorterDuff$Mode;
  
+     .line 22
      const/16 v0, 0x12
  
      new-array v0, v0, [Landroid/graphics/PorterDuff$Mode;
***************
*** 397,406 ****
--- 416,428 ----
      .end annotation
  
      .prologue
+     .line 61
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 62
      iput p3, p0, Landroid/graphics/PorterDuff$Mode;->nativeInt:I
  
+     .line 63
      return-void
  .end method
  
***************
*** 409,414 ****
--- 431,437 ----
      .parameter "name"
  
      .prologue
+     .line 22
      const-class v0, Landroid/graphics/PorterDuff$Mode;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 424,429 ****
--- 447,453 ----
      .locals 1
  
      .prologue
+     .line 22
      sget-object v0, Landroid/graphics/PorterDuff$Mode;->$VALUES:[Landroid/graphics/PorterDuff$Mode;
  
      invoke-virtual {v0}, [Landroid/graphics/PorterDuff$Mode;->clone()Ljava/lang/Object;
