*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/RadialGradient.smali	2012-03-23 19:35:41.824349553 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/RadialGradient.smali.original	2012-03-23 19:34:44.781309374 -0500
***************
*** 14,35 ****
      .parameter "tile"
  
      .prologue
      invoke-direct {p0}, Landroid/graphics/Shader;-><init>()V
  
      const/4 v0, 0x0
  
      cmpg-float v0, p3, v0
  
      if-gtz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "radius must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget v5, p6, Landroid/graphics/Shader$TileMode;->nativeInt:I
  
--- 14,39 ----
      .parameter "tile"
  
      .prologue
+     .line 56
      invoke-direct {p0}, Landroid/graphics/Shader;-><init>()V
  
+     .line 57
      const/4 v0, 0x0
  
      cmpg-float v0, p3, v0
  
      if-gtz v0, :cond_0
  
+     .line 58
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "radius must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 60
      :cond_0
      iget v5, p6, Landroid/graphics/Shader$TileMode;->nativeInt:I
  
***************
*** 49,54 ****
--- 53,59 ----
  
      iput v0, p0, Landroid/graphics/RadialGradient;->native_instance:I
  
+     .line 61
      iget v0, p0, Landroid/graphics/RadialGradient;->native_instance:I
  
      iget v6, p6, Landroid/graphics/Shader$TileMode;->nativeInt:I
***************
*** 69,74 ****
--- 74,80 ----
  
      iput v0, p0, Landroid/graphics/RadialGradient;->native_shader:I
  
+     .line 63
      return-void
  .end method
  
***************
*** 82,103 ****
      .parameter "tile"
  
      .prologue
      invoke-direct {p0}, Landroid/graphics/Shader;-><init>()V
  
      const/4 v0, 0x0
  
      cmpg-float v0, p3, v0
  
      if-gtz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "radius must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      array-length v0, p4
  
--- 88,113 ----
      .parameter "tile"
  
      .prologue
+     .line 32
      invoke-direct {p0}, Landroid/graphics/Shader;-><init>()V
  
+     .line 33
      const/4 v0, 0x0
  
      cmpg-float v0, p3, v0
  
      if-gtz v0, :cond_0
  
+     .line 34
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "radius must be > 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 36
      :cond_0
      array-length v0, p4
  
***************
*** 105,118 ****
  
      if-ge v0, v1, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "needs >= 2 number of colors"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_1
      if-eqz p5, :cond_2
  
--- 115,130 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 37
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "needs >= 2 number of colors"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 39
      :cond_1
      if-eqz p5, :cond_2
  
***************
*** 122,127 ****
--- 134,140 ----
  
      if-eq v0, v1, :cond_2
  
+     .line 40
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "color and position arrays must be of equal length"
***************
*** 130,135 ****
--- 143,149 ----
  
      throw v0
  
+     .line 42
      :cond_2
      iget v5, p6, Landroid/graphics/Shader$TileMode;->nativeInt:I
  
***************
*** 149,154 ****
--- 163,169 ----
  
      iput v0, p0, Landroid/graphics/RadialGradient;->native_instance:I
  
+     .line 43
      iget v0, p0, Landroid/graphics/RadialGradient;->native_instance:I
  
      iget v6, p6, Landroid/graphics/Shader$TileMode;->nativeInt:I
***************
*** 169,174 ****
--- 184,190 ----
  
      iput v0, p0, Landroid/graphics/RadialGradient;->native_shader:I
  
+     .line 45
      return-void
  .end method
  
