*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/TableMaskFilter.smali	2012-03-23 19:35:44.539399059 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/TableMaskFilter.smali.original	2012-03-23 19:34:44.817310030 -0500
***************
*** 9,18 ****
--- 9,21 ----
      .parameter "ni"
  
      .prologue
+     .line 31
      invoke-direct {p0}, Landroid/graphics/MaskFilter;-><init>()V
  
+     .line 32
      iput p1, p0, Landroid/graphics/TableMaskFilter;->native_instance:I
  
+     .line 33
      return-void
  .end method
  
***************
*** 21,42 ****
      .parameter "table"
  
      .prologue
      invoke-direct {p0}, Landroid/graphics/MaskFilter;-><init>()V
  
      array-length v0, p1
  
      const/16 v1, 0x100
  
      if-ge v0, v1, :cond_0
  
      new-instance v0, Ljava/lang/RuntimeException;
  
!     const-string v1, "table.length must be >= 256"
  
      invoke-direct {v0, v1}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      invoke-static {p1}, Landroid/graphics/TableMaskFilter;->nativeNewTable([B)I
  
--- 24,49 ----
      .parameter "table"
  
      .prologue
+     .line 24
      invoke-direct {p0}, Landroid/graphics/MaskFilter;-><init>()V
  
+     .line 25
      array-length v0, p1
  
      const/16 v1, 0x100
  
      if-ge v0, v1, :cond_0
  
+     .line 26
      new-instance v0, Ljava/lang/RuntimeException;
  
!     const-string/jumbo v1, "table.length must be >= 256"
  
      invoke-direct {v0, v1}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 28
      :cond_0
      invoke-static {p1}, Landroid/graphics/TableMaskFilter;->nativeNewTable([B)I
  
***************
*** 44,49 ****
--- 51,57 ----
  
      iput v0, p0, Landroid/graphics/TableMaskFilter;->native_instance:I
  
+     .line 29
      return-void
  .end method
  
***************
*** 53,58 ****
--- 61,67 ----
      .parameter "max"
  
      .prologue
+     .line 36
      new-instance v0, Landroid/graphics/TableMaskFilter;
  
      invoke-static {p0, p1}, Landroid/graphics/TableMaskFilter;->nativeNewClip(II)I
***************
*** 69,74 ****
--- 78,84 ----
      .parameter "gamma"
  
      .prologue
+     .line 40
      new-instance v0, Landroid/graphics/TableMaskFilter;
  
      invoke-static {p0}, Landroid/graphics/TableMaskFilter;->nativeNewGamma(F)I
