*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/YuvImage.smali	2012-03-23 19:35:41.606345576 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/YuvImage.smali.original	2012-03-23 19:34:44.779309336 -0500
***************
*** 29,36 ****
--- 29,38 ----
      .parameter "strides"
  
      .prologue
+     .line 79
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 80
      const/16 v0, 0x11
  
      if-eq p2, v0, :cond_0
***************
*** 39,97 ****
  
      if-eq p2, v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "only support ImageFormat.NV21 and ImageFormat.YUY2 for now"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      if-lez p3, :cond_1
  
      if-gtz p4, :cond_2
  
      :cond_1
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "width and height must large than 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_2
      if-nez p1, :cond_3
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "yuv cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_3
      if-nez p5, :cond_4
  
      invoke-direct {p0, p3, p2}, Landroid/graphics/YuvImage;->calculateStrides(II)[I
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/graphics/YuvImage;->mStrides:[I
  
      :goto_0
      iput-object p1, p0, Landroid/graphics/YuvImage;->mData:[B
  
      iput p2, p0, Landroid/graphics/YuvImage;->mFormat:I
  
      iput p3, p0, Landroid/graphics/YuvImage;->mWidth:I
  
      iput p4, p0, Landroid/graphics/YuvImage;->mHeight:I
  
      return-void
  
      :cond_4
      iput-object p5, p0, Landroid/graphics/YuvImage;->mStrides:[I
  
--- 41,112 ----
  
      if-eq p2, v0, :cond_0
  
+     .line 82
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "only support ImageFormat.NV21 and ImageFormat.YUY2 for now"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 87
      :cond_0
      if-lez p3, :cond_1
  
      if-gtz p4, :cond_2
  
+     .line 88
      :cond_1
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "width and height must large than 0"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 92
      :cond_2
      if-nez p1, :cond_3
  
+     .line 93
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "yuv cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 96
      :cond_3
      if-nez p5, :cond_4
  
+     .line 97
      invoke-direct {p0, p3, p2}, Landroid/graphics/YuvImage;->calculateStrides(II)[I
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/graphics/YuvImage;->mStrides:[I
  
+     .line 102
      :goto_0
      iput-object p1, p0, Landroid/graphics/YuvImage;->mData:[B
  
+     .line 103
      iput p2, p0, Landroid/graphics/YuvImage;->mFormat:I
  
+     .line 104
      iput p3, p0, Landroid/graphics/YuvImage;->mWidth:I
  
+     .line 105
      iput p4, p0, Landroid/graphics/YuvImage;->mHeight:I
  
+     .line 106
      return-void
  
+     .line 99
      :cond_4
      iput-object p5, p0, Landroid/graphics/YuvImage;->mStrides:[I
  
***************
*** 103,117 ****
--- 118,135 ----
      .parameter "rect"
  
      .prologue
+     .line 215
      invoke-virtual {p1}, Landroid/graphics/Rect;->width()I
  
      move-result v1
  
+     .line 216
      .local v1, width:I
      invoke-virtual {p1}, Landroid/graphics/Rect;->height()I
  
      move-result v0
  
+     .line 217
      .local v0, height:I
      iget v2, p0, Landroid/graphics/YuvImage;->mFormat:I
  
***************
*** 119,152 ****
--- 137,177 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 219
      and-int/lit8 v1, v1, -0x2
  
+     .line 220
      and-int/lit8 v0, v0, -0x2
  
+     .line 221
      iget v2, p1, Landroid/graphics/Rect;->left:I
  
      and-int/lit8 v2, v2, -0x2
  
      iput v2, p1, Landroid/graphics/Rect;->left:I
  
+     .line 222
      iget v2, p1, Landroid/graphics/Rect;->top:I
  
      and-int/lit8 v2, v2, -0x2
  
      iput v2, p1, Landroid/graphics/Rect;->top:I
  
+     .line 223
      iget v2, p1, Landroid/graphics/Rect;->left:I
  
      add-int/2addr v2, v1
  
      iput v2, p1, Landroid/graphics/Rect;->right:I
  
+     .line 224
      iget v2, p1, Landroid/graphics/Rect;->top:I
  
      add-int/2addr v2, v0
  
      iput v2, p1, Landroid/graphics/Rect;->bottom:I
  
+     .line 227
      :cond_0
      iget v2, p0, Landroid/graphics/YuvImage;->mFormat:I
  
***************
*** 154,173 ****
--- 179,202 ----
  
      if-ne v2, v3, :cond_1
  
+     .line 229
      and-int/lit8 v1, v1, -0x2
  
+     .line 230
      iget v2, p1, Landroid/graphics/Rect;->left:I
  
      and-int/lit8 v2, v2, -0x2
  
      iput v2, p1, Landroid/graphics/Rect;->left:I
  
+     .line 231
      iget v2, p1, Landroid/graphics/Rect;->left:I
  
      add-int/2addr v2, v1
  
      iput v2, p1, Landroid/graphics/Rect;->right:I
  
+     .line 233
      :cond_1
      return-void
  .end method
***************
*** 182,194 ****
--- 211,226 ----
  
      const/4 v3, 0x0
  
+     .line 200
      const/4 v0, 0x0
  
+     .line 201
      .local v0, strides:[I
      const/16 v2, 0x11
  
      if-ne p2, v2, :cond_0
  
+     .line 202
      const/4 v2, 0x2
  
      new-array v0, v2, [I
***************
*** 201,211 ****
--- 233,245 ----
      .restart local v0       #strides:[I
      move-object v1, v0
  
+     .line 211
      .end local v0           #strides:[I
      .local v1, strides:[I
      :goto_0
      return-object v1
  
+     .line 206
      .end local v1           #strides:[I
      .restart local v0       #strides:[I
      :cond_0
***************
*** 213,218 ****
--- 247,253 ----
  
      if-ne p2, v2, :cond_1
  
+     .line 207
      new-array v0, v4, [I
  
      .end local v0           #strides:[I
***************
*** 223,228 ****
--- 258,264 ----
      .restart local v0       #strides:[I
      move-object v1, v0
  
+     .line 208
      .end local v0           #strides:[I
      .restart local v1       #strides:[I
      goto :goto_0
***************
*** 232,237 ****
--- 268,274 ----
      :cond_1
      move-object v1, v0
  
+     .line 211
      .end local v0           #strides:[I
      .restart local v1       #strides:[I
      goto :goto_0
***************
*** 252,259 ****
--- 289,298 ----
  
      const/4 v4, 0x0
  
+     .line 183
      const/4 v0, 0x0
  
+     .line 184
      .local v0, offsets:[I
      iget v2, p0, Landroid/graphics/YuvImage;->mFormat:I
  
***************
*** 261,266 ****
--- 300,306 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 185
      const/4 v2, 0x2
  
      new-array v0, v2, [I
***************
*** 305,315 ****
--- 345,357 ----
      .restart local v0       #offsets:[I
      move-object v1, v0
  
+     .line 196
      .end local v0           #offsets:[I
      .local v1, offsets:[I
      :goto_0
      return-object v1
  
+     .line 191
      .end local v1           #offsets:[I
      .restart local v0       #offsets:[I
      :cond_0
***************
*** 319,324 ****
--- 361,367 ----
  
      if-ne v2, v3, :cond_1
  
+     .line 192
      new-array v0, v5, [I
  
      .end local v0           #offsets:[I
***************
*** 339,344 ****
--- 382,388 ----
      .restart local v0       #offsets:[I
      move-object v1, v0
  
+     .line 193
      .end local v0           #offsets:[I
      .restart local v1       #offsets:[I
      goto :goto_0
***************
*** 348,353 ****
--- 392,398 ----
      :cond_1
      move-object v1, v0
  
+     .line 196
      .end local v0           #offsets:[I
      .restart local v1       #offsets:[I
      goto :goto_0
***************
*** 362,367 ****
--- 407,413 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 124
      new-instance v9, Landroid/graphics/Rect;
  
      iget v0, p0, Landroid/graphics/YuvImage;->mWidth:I
***************
*** 370,375 ****
--- 416,422 ----
  
      invoke-direct {v9, v2, v2, v0, v1}, Landroid/graphics/Rect;-><init>(IIII)V
  
+     .line 125
      .local v9, wholeImage:Landroid/graphics/Rect;
      invoke-virtual {v9, p1}, Landroid/graphics/Rect;->contains(Landroid/graphics/Rect;)Z
  
***************
*** 377,390 ****
  
      if-nez v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "rectangle is not inside the image"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      if-ltz p2, :cond_1
  
--- 424,439 ----
  
      if-nez v0, :cond_0
  
+     .line 126
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "rectangle is not inside the image"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 130
      :cond_0
      if-ltz p2, :cond_1
  
***************
*** 392,420 ****
  
      if-le p2, v0, :cond_2
  
      :cond_1
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "quality must be 0..100"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_2
      if-nez p3, :cond_3
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "stream cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_3
      invoke-direct {p0, p1}, Landroid/graphics/YuvImage;->adjustRectangle(Landroid/graphics/Rect;)V
  
      iget v0, p1, Landroid/graphics/Rect;->left:I
  
      iget v1, p1, Landroid/graphics/Rect;->top:I
--- 441,474 ----
  
      if-le p2, v0, :cond_2
  
+     .line 131
      :cond_1
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "quality must be 0..100"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 134
      :cond_2
      if-nez p3, :cond_3
  
+     .line 135
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "stream cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 138
      :cond_3
      invoke-direct {p0, p1}, Landroid/graphics/YuvImage;->adjustRectangle(Landroid/graphics/Rect;)V
  
+     .line 139
      iget v0, p1, Landroid/graphics/Rect;->left:I
  
      iget v1, p1, Landroid/graphics/Rect;->top:I
***************
*** 423,428 ****
--- 477,483 ----
  
      move-result-object v4
  
+     .line 141
      .local v4, offsets:[I
      iget-object v0, p0, Landroid/graphics/YuvImage;->mData:[B
  
***************
*** 457,462 ****
--- 512,518 ----
      .locals 1
  
      .prologue
+     .line 179
      iget v0, p0, Landroid/graphics/YuvImage;->mHeight:I
  
      return v0
***************
*** 466,471 ****
--- 522,528 ----
      .locals 1
  
      .prologue
+     .line 165
      iget-object v0, p0, Landroid/graphics/YuvImage;->mStrides:[I
  
      return-object v0
***************
*** 475,480 ****
--- 532,538 ----
      .locals 1
  
      .prologue
+     .line 172
      iget v0, p0, Landroid/graphics/YuvImage;->mWidth:I
  
      return v0
***************
*** 484,489 ****
--- 542,548 ----
      .locals 1
  
      .prologue
+     .line 151
      iget-object v0, p0, Landroid/graphics/YuvImage;->mData:[B
  
      return-object v0
***************
*** 493,498 ****
--- 552,558 ----
      .locals 1
  
      .prologue
+     .line 158
      iget v0, p0, Landroid/graphics/YuvImage;->mFormat:I
  
      return v0
