*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/accessibility/AccessibilityEvent$1.smali	2012-03-23 19:37:19.801136155 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/accessibility/AccessibilityEvent$1.smali.original	2012-03-23 19:34:45.641325055 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 996
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,56 ****
--- 45,60 ----
      .parameter "parcel"
  
      .prologue
+     .line 998
      invoke-static {}, Landroid/view/accessibility/AccessibilityEvent;->obtain()Landroid/view/accessibility/AccessibilityEvent;
  
      move-result-object v0
  
+     .line 999
      .local v0, event:Landroid/view/accessibility/AccessibilityEvent;
      invoke-virtual {v0, p1}, Landroid/view/accessibility/AccessibilityEvent;->initFromParcel(Landroid/os/Parcel;)V
  
+     .line 1000
      return-object v0
  .end method
  
***************
*** 59,64 ****
--- 63,69 ----
      .parameter "x0"
  
      .prologue
+     .line 996
      invoke-virtual {p0, p1}, Landroid/view/accessibility/AccessibilityEvent$1;->createFromParcel(Landroid/os/Parcel;)Landroid/view/accessibility/AccessibilityEvent;
  
      move-result-object v0
***************
*** 71,76 ****
--- 76,82 ----
      .parameter "size"
  
      .prologue
+     .line 1004
      new-array v0, p1, [Landroid/view/accessibility/AccessibilityEvent;
  
      return-object v0
***************
*** 81,86 ****
--- 87,93 ----
      .parameter "x0"
  
      .prologue
+     .line 996
      invoke-virtual {p0, p1}, Landroid/view/accessibility/AccessibilityEvent$1;->newArray(I)[Landroid/view/accessibility/AccessibilityEvent;
  
      move-result-object v0
