*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/accessibility/AccessibilityManager$1.smali	2012-03-23 19:37:19.848137013 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/accessibility/AccessibilityManager$1.smali.original	2012-03-23 19:34:45.641325055 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 107
      iput-object p1, p0, Landroid/view/accessibility/AccessibilityManager$1;->this$0:Landroid/view/accessibility/AccessibilityManager;
  
      invoke-direct {p0}, Landroid/view/accessibility/IAccessibilityManagerClient$Stub;-><init>()V
***************
*** 38,43 ****
--- 39,45 ----
      .parameter "state"
  
      .prologue
+     .line 109
      iget-object v0, p0, Landroid/view/accessibility/AccessibilityManager$1;->this$0:Landroid/view/accessibility/AccessibilityManager;
  
      iget-object v0, v0, Landroid/view/accessibility/AccessibilityManager;->mHandler:Landroid/os/Handler;
***************
*** 52,56 ****
--- 54,59 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 110
      return-void
  .end method
