*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/accessibility/IAccessibilityManagerClient$Stub$Proxy.smali	2012-03-23 19:37:19.593132363 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/accessibility/IAccessibilityManagerClient$Stub$Proxy.smali.original	2012-03-23 19:34:45.639325019 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 66
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 67
      iput-object p1, p0, Landroid/view/accessibility/IAccessibilityManagerClient$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 68
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 71
      iget-object v0, p0, Landroid/view/accessibility/IAccessibilityManagerClient$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 49,54 ****
--- 53,59 ----
      .locals 1
  
      .prologue
+     .line 75
      const-string v0, "android.view.accessibility.IAccessibilityManagerClient"
  
      return-object v0
***************
*** 64,81 ****
--- 69,90 ----
      .end annotation
  
      .prologue
+     .line 79
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 81
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.view.accessibility.IAccessibilityManagerClient"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 82
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 83
      iget-object v1, p0, Landroid/view/accessibility/IAccessibilityManagerClient$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x1
***************
*** 88,97 ****
--- 97,109 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 86
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 88
      return-void
  
+     .line 86
      :catchall_0
      move-exception v1
  
