*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/location/ComprehensiveCountryDetector$3.smali	2012-03-23 19:34:58.147553107 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/location/ComprehensiveCountryDetector$3.smali.original	2012-03-23 19:34:46.936348670 -0500
***************
*** 39,44 ****
--- 39,45 ----
      .parameter
  
      .prologue
+     .line 211
      iput-object p1, p0, Lcom/android/server/location/ComprehensiveCountryDetector$3;->this$0:Lcom/android/server/location/ComprehensiveCountryDetector;
  
      iput-object p2, p0, Lcom/android/server/location/ComprehensiveCountryDetector$3;->val$country:Landroid/location/Country;
***************
*** 60,65 ****
--- 61,67 ----
      .locals 5
  
      .prologue
+     .line 214
      iget-object v0, p0, Lcom/android/server/location/ComprehensiveCountryDetector$3;->this$0:Lcom/android/server/location/ComprehensiveCountryDetector;
  
      iget-object v1, p0, Lcom/android/server/location/ComprehensiveCountryDetector$3;->val$country:Landroid/location/Country;
***************
*** 72,76 ****
--- 74,79 ----
  
      invoke-virtual {v0, v1, v2, v3, v4}, Lcom/android/server/location/ComprehensiveCountryDetector;->runAfterDetection(Landroid/location/Country;Landroid/location/Country;ZZ)V
  
+     .line 216
      return-void
  .end method
