*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/location/ComprehensiveCountryDetector$2.smali	2012-03-23 19:34:57.619543478 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/location/ComprehensiveCountryDetector$2.smali.original	2012-03-23 19:34:46.928348525 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 93
      iput-object p1, p0, Lcom/android/server/location/ComprehensiveCountryDetector$2;->this$0:Lcom/android/server/location/ComprehensiveCountryDetector;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,51 ****
--- 42,54 ----
      .parameter "country"
  
      .prologue
+     .line 97
      iget-object v0, p0, Lcom/android/server/location/ComprehensiveCountryDetector$2;->this$0:Lcom/android/server/location/ComprehensiveCountryDetector;
  
      #setter for: Lcom/android/server/location/ComprehensiveCountryDetector;->mCountryFromLocation:Landroid/location/Country;
      invoke-static {v0, p1}, Lcom/android/server/location/ComprehensiveCountryDetector;->access$202(Lcom/android/server/location/ComprehensiveCountryDetector;Landroid/location/Country;)Landroid/location/Country;
  
+     .line 99
      iget-object v0, p0, Lcom/android/server/location/ComprehensiveCountryDetector$2;->this$0:Lcom/android/server/location/ComprehensiveCountryDetector;
  
      const/4 v1, 0x1
***************
*** 55,64 ****
--- 58,69 ----
      #calls: Lcom/android/server/location/ComprehensiveCountryDetector;->detectCountry(ZZ)Landroid/location/Country;
      invoke-static {v0, v1, v2}, Lcom/android/server/location/ComprehensiveCountryDetector;->access$100(Lcom/android/server/location/ComprehensiveCountryDetector;ZZ)Landroid/location/Country;
  
+     .line 100
      iget-object v0, p0, Lcom/android/server/location/ComprehensiveCountryDetector$2;->this$0:Lcom/android/server/location/ComprehensiveCountryDetector;
  
      #calls: Lcom/android/server/location/ComprehensiveCountryDetector;->stopLocationBasedDetector()V
      invoke-static {v0}, Lcom/android/server/location/ComprehensiveCountryDetector;->access$300(Lcom/android/server/location/ComprehensiveCountryDetector;)V
  
+     .line 101
      return-void
  .end method
