*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/location/GpsLocationProvider$GpsLocationProviderThread.smali	2012-03-23 19:34:57.979550043 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/location/GpsLocationProvider$GpsLocationProviderThread.smali.original	2012-03-23 19:34:46.934348634 -0500
***************
*** 24,35 ****
--- 24,38 ----
      .parameter
  
      .prologue
+     .line 1592
      iput-object p1, p0, Lcom/android/server/location/GpsLocationProvider$GpsLocationProviderThread;->this$0:Lcom/android/server/location/GpsLocationProvider;
  
+     .line 1593
      const-string v0, "GpsLocationProvider"
  
      invoke-direct {p0, v0}, Ljava/lang/Thread;-><init>(Ljava/lang/String;)V
  
+     .line 1594
      return-void
  .end method
  
***************
*** 39,55 ****
--- 42,62 ----
      .locals 4
  
      .prologue
+     .line 1597
      const/16 v0, 0xa
  
      invoke-static {v0}, Landroid/os/Process;->setThreadPriority(I)V
  
+     .line 1598
      iget-object v0, p0, Lcom/android/server/location/GpsLocationProvider$GpsLocationProviderThread;->this$0:Lcom/android/server/location/GpsLocationProvider;
  
      #calls: Lcom/android/server/location/GpsLocationProvider;->initialize()V
      invoke-static {v0}, Lcom/android/server/location/GpsLocationProvider;->access$2000(Lcom/android/server/location/GpsLocationProvider;)V
  
+     .line 1599
      invoke-static {}, Landroid/os/Looper;->prepare()V
  
+     .line 1600
      iget-object v0, p0, Lcom/android/server/location/GpsLocationProvider$GpsLocationProviderThread;->this$0:Lcom/android/server/location/GpsLocationProvider;
  
      new-instance v1, Lcom/android/server/location/GpsLocationProvider$ProviderHandler;
***************
*** 63,68 ****
--- 70,76 ----
      #setter for: Lcom/android/server/location/GpsLocationProvider;->mHandler:Landroid/os/Handler;
      invoke-static {v0, v1}, Lcom/android/server/location/GpsLocationProvider;->access$2102(Lcom/android/server/location/GpsLocationProvider;Landroid/os/Handler;)Landroid/os/Handler;
  
+     .line 1602
      iget-object v0, p0, Lcom/android/server/location/GpsLocationProvider$GpsLocationProviderThread;->this$0:Lcom/android/server/location/GpsLocationProvider;
  
      #getter for: Lcom/android/server/location/GpsLocationProvider;->mInitializedLatch:Ljava/util/concurrent/CountDownLatch;
***************
*** 72,78 ****
--- 80,88 ----
  
      invoke-virtual {v0}, Ljava/util/concurrent/CountDownLatch;->countDown()V
  
+     .line 1603
      invoke-static {}, Landroid/os/Looper;->loop()V
  
+     .line 1604
      return-void
  .end method
