*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/location/PassiveProvider.smali	2012-03-23 19:34:57.843547563 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/location/PassiveProvider.smali.original	2012-03-23 19:34:46.933348616 -0500
***************
*** 22,31 ****
--- 22,34 ----
      .parameter "locationManager"
  
      .prologue
+     .line 44
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 45
      iput-object p1, p0, Lcom/android/server/location/PassiveProvider;->mLocationManager:Landroid/location/ILocationManager;
  
+     .line 46
      return-void
  .end method
  
***************
*** 36,41 ****
--- 39,45 ----
      .parameter "uid"
  
      .prologue
+     .line 149
      return-void
  .end method
  
***************
*** 43,48 ****
--- 47,53 ----
      .locals 0
  
      .prologue
+     .line 101
      return-void
  .end method
  
***************
*** 50,55 ****
--- 55,61 ----
      .locals 0
  
      .prologue
+     .line 98
      return-void
  .end method
  
***************
*** 58,65 ****
--- 64,73 ----
      .parameter "enable"
  
      .prologue
+     .line 120
      iput-boolean p1, p0, Lcom/android/server/location/PassiveProvider;->mTracking:Z
  
+     .line 121
      return-void
  .end method
  
***************
*** 67,72 ****
--- 75,81 ----
      .locals 1
  
      .prologue
+     .line 90
      const/4 v0, -0x1
  
      return v0
***************
*** 76,81 ****
--- 85,91 ----
      .locals 1
  
      .prologue
+     .line 116
      const/4 v0, 0x0
  
      return-object v0
***************
*** 85,90 ****
--- 95,101 ----
      .locals 1
  
      .prologue
+     .line 49
      const-string v0, "passive"
  
      return-object v0
***************
*** 94,99 ****
--- 105,111 ----
      .locals 1
  
      .prologue
+     .line 81
      const/4 v0, -0x1
  
      return v0
***************
*** 104,115 ****
--- 116,130 ----
      .parameter "extras"
  
      .prologue
+     .line 104
      iget-boolean v0, p0, Lcom/android/server/location/PassiveProvider;->mTracking:Z
  
      if-eqz v0, :cond_0
  
+     .line 105
      const/4 v0, 0x2
  
+     .line 107
      :goto_0
      return v0
  
***************
*** 123,128 ****
--- 138,144 ----
      .locals 2
  
      .prologue
+     .line 112
      const-wide/16 v0, -0x1
  
      return-wide v0
***************
*** 132,137 ****
--- 148,154 ----
      .locals 1
  
      .prologue
+     .line 65
      const/4 v0, 0x0
  
      return v0
***************
*** 141,146 ****
--- 158,164 ----
      .locals 1
  
      .prologue
+     .line 94
      const/4 v0, 0x1
  
      return v0
***************
*** 151,156 ****
--- 169,175 ----
      .parameter "criteria"
  
      .prologue
+     .line 86
      const/4 v0, 0x0
  
      return v0
***************
*** 161,166 ****
--- 180,186 ----
      .parameter "uid"
  
      .prologue
+     .line 152
      return-void
  .end method
  
***************
*** 168,173 ****
--- 188,194 ----
      .locals 1
  
      .prologue
+     .line 124
      const/4 v0, 0x0
  
      return v0
***************
*** 177,182 ****
--- 198,204 ----
      .locals 1
  
      .prologue
+     .line 61
      const/4 v0, 0x0
  
      return v0
***************
*** 186,191 ****
--- 208,214 ----
      .locals 1
  
      .prologue
+     .line 53
      const/4 v0, 0x0
  
      return v0
***************
*** 195,200 ****
--- 218,224 ----
      .locals 1
  
      .prologue
+     .line 57
      const/4 v0, 0x0
  
      return v0
***************
*** 206,211 ****
--- 230,236 ----
      .parameter "extras"
  
      .prologue
+     .line 145
      const/4 v0, 0x0
  
      return v0
***************
*** 217,222 ****
--- 242,248 ----
      .parameter "ws"
  
      .prologue
+     .line 128
      return-void
  .end method
  
***************
*** 224,229 ****
--- 250,256 ----
      .locals 1
  
      .prologue
+     .line 69
      const/4 v0, 0x0
  
      return v0
***************
*** 233,238 ****
--- 260,266 ----
      .locals 1
  
      .prologue
+     .line 77
      const/4 v0, 0x0
  
      return v0
***************
*** 242,247 ****
--- 270,276 ----
      .locals 1
  
      .prologue
+     .line 73
      const/4 v0, 0x0
  
      return v0
***************
*** 252,261 ****
--- 281,292 ----
      .parameter "location"
  
      .prologue
+     .line 134
      iget-boolean v1, p0, Lcom/android/server/location/PassiveProvider;->mTracking:Z
  
      if-eqz v1, :cond_0
  
+     .line 137
      :try_start_0
      iget-object v1, p0, Lcom/android/server/location/PassiveProvider;->mLocationManager:Landroid/location/ILocationManager;
  
***************
*** 265,277 ****
--- 296,311 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 142
      :cond_0
      :goto_0
      return-void
  
+     .line 138
      :catch_0
      move-exception v0
  
+     .line 139
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "PassiveProvider"
  
***************
*** 288,292 ****
--- 322,327 ----
      .parameter "info"
  
      .prologue
+     .line 131
      return-void
  .end method
