*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CacheManager$CacheResult.smali	2012-03-23 19:37:07.289908013 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CacheManager$CacheResult.smali.original	2012-03-23 19:34:45.555323487 -0500
***************
*** 54,59 ****
--- 54,60 ----
      .locals 0
  
      .prologue
+     .line 97
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 65,70 ****
--- 66,72 ----
      .locals 1
  
      .prologue
+     .line 158
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->contentdisposition:Ljava/lang/String;
  
      return-object v0
***************
*** 74,79 ****
--- 76,82 ----
      .locals 2
  
      .prologue
+     .line 122
      iget-wide v0, p0, Landroid/webkit/CacheManager$CacheResult;->contentLength:J
  
      return-wide v0
***************
*** 83,88 ****
--- 86,92 ----
      .locals 1
  
      .prologue
+     .line 142
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->etag:Ljava/lang/String;
  
      return-object v0
***************
*** 92,97 ****
--- 96,102 ----
      .locals 1
  
      .prologue
+     .line 154
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->encoding:Ljava/lang/String;
  
      return-object v0
***************
*** 101,106 ****
--- 106,112 ----
      .locals 2
  
      .prologue
+     .line 130
      iget-wide v0, p0, Landroid/webkit/CacheManager$CacheResult;->expires:J
  
      return-wide v0
***************
*** 110,115 ****
--- 116,122 ----
      .locals 1
  
      .prologue
+     .line 134
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->expiresString:Ljava/lang/String;
  
      return-object v0
***************
*** 119,124 ****
--- 126,132 ----
      .locals 1
  
      .prologue
+     .line 118
      iget v0, p0, Landroid/webkit/CacheManager$CacheResult;->httpStatusCode:I
  
      return v0
***************
*** 128,133 ****
--- 136,142 ----
      .locals 1
  
      .prologue
+     .line 163
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->inStream:Ljava/io/InputStream;
  
      return-object v0
***************
*** 137,142 ****
--- 146,152 ----
      .locals 1
  
      .prologue
+     .line 138
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->lastModified:Ljava/lang/String;
  
      return-object v0
***************
*** 146,151 ****
--- 156,162 ----
      .locals 1
  
      .prologue
+     .line 126
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->localPath:Ljava/lang/String;
  
      return-object v0
***************
*** 155,160 ****
--- 166,172 ----
      .locals 1
  
      .prologue
+     .line 150
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->location:Ljava/lang/String;
  
      return-object v0
***************
*** 164,169 ****
--- 176,182 ----
      .locals 1
  
      .prologue
+     .line 146
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->mimeType:Ljava/lang/String;
  
      return-object v0
***************
*** 173,178 ****
--- 186,192 ----
      .locals 1
  
      .prologue
+     .line 167
      iget-object v0, p0, Landroid/webkit/CacheManager$CacheResult;->outStream:Ljava/io/OutputStream;
  
      return-object v0
***************
*** 183,190 ****
--- 197,206 ----
      .parameter "contentLength"
  
      .prologue
+     .line 183
      iput-wide p1, p0, Landroid/webkit/CacheManager$CacheResult;->contentLength:J
  
+     .line 184
      return-void
  .end method
  
***************
*** 193,200 ****
--- 209,218 ----
      .parameter "encoding"
  
      .prologue
+     .line 176
      iput-object p1, p0, Landroid/webkit/CacheManager$CacheResult;->encoding:Ljava/lang/String;
  
+     .line 177
      return-void
  .end method
  
***************
*** 203,209 ****
--- 221,229 ----
      .parameter "stream"
  
      .prologue
+     .line 172
      iput-object p1, p0, Landroid/webkit/CacheManager$CacheResult;->inStream:Ljava/io/InputStream;
  
+     .line 173
      return-void
  .end method
