*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CallbackProxy$4.smali	2012-03-23 19:37:13.099013943 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CallbackProxy$4.smali.original	2012-03-23 19:34:45.587324071 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 573
      iput-object p1, p0, Landroid/webkit/CallbackProxy$4;->this$0:Landroid/webkit/CallbackProxy;
  
      iput-object p2, p0, Landroid/webkit/CallbackProxy$4;->val$res:Landroid/webkit/JsResult;
***************
*** 47,55 ****
--- 48,58 ----
      .parameter "which"
  
      .prologue
+     .line 577
      iget-object v0, p0, Landroid/webkit/CallbackProxy$4;->val$res:Landroid/webkit/JsResult;
  
      invoke-virtual {v0}, Landroid/webkit/JsResult;->cancel()V
  
+     .line 578
      return-void
  .end method
