*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CallbackProxy$5.smali	2012-03-23 19:37:14.117032507 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CallbackProxy$5.smali.original	2012-03-23 19:34:45.594324199 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 566
      iput-object p1, p0, Landroid/webkit/CallbackProxy$5;->this$0:Landroid/webkit/CallbackProxy;
  
      iput-object p2, p0, Landroid/webkit/CallbackProxy$5;->val$res:Landroid/webkit/JsResult;
***************
*** 47,55 ****
--- 48,58 ----
      .parameter "which"
  
      .prologue
+     .line 570
      iget-object v0, p0, Landroid/webkit/CallbackProxy$5;->val$res:Landroid/webkit/JsResult;
  
      invoke-virtual {v0}, Landroid/webkit/JsResult;->confirm()V
  
+     .line 571
      return-void
  .end method
