*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CallbackProxy$3.smali	2012-03-23 19:37:09.046940053 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CallbackProxy$3.smali.original	2012-03-23 19:34:45.567323706 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 580
      iput-object p1, p0, Landroid/webkit/CallbackProxy$3;->this$0:Landroid/webkit/CallbackProxy;
  
      iput-object p2, p0, Landroid/webkit/CallbackProxy$3;->val$res:Landroid/webkit/JsResult;
***************
*** 46,54 ****
--- 47,57 ----
      .parameter "dialog"
  
      .prologue
+     .line 583
      iget-object v0, p0, Landroid/webkit/CallbackProxy$3;->val$res:Landroid/webkit/JsResult;
  
      invoke-virtual {v0}, Landroid/webkit/JsResult;->cancel()V
  
+     .line 584
      return-void
  .end method
