*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CallbackProxy$UploadFile.smali	2012-03-23 19:37:14.281035496 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CallbackProxy$UploadFile.smali.original	2012-03-23 19:34:45.595324217 -0500
***************
*** 39,44 ****
--- 39,45 ----
      .parameter
  
      .prologue
+     .line 1567
      iput-object p1, p0, Landroid/webkit/CallbackProxy$UploadFile;->this$0:Landroid/webkit/CallbackProxy;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 52,57 ****
--- 53,59 ----
      .parameter "x1"
  
      .prologue
+     .line 1567
      invoke-direct {p0, p1}, Landroid/webkit/CallbackProxy$UploadFile;-><init>(Landroid/webkit/CallbackProxy;)V
  
      return-void
***************
*** 63,68 ****
--- 65,71 ----
      .locals 1
  
      .prologue
+     .line 1576
      iget-object v0, p0, Landroid/webkit/CallbackProxy$UploadFile;->mValue:Landroid/net/Uri;
  
      return-object v0
***************
*** 73,93 ****
--- 76,102 ----
      .parameter "value"
  
      .prologue
+     .line 1570
      iput-object p1, p0, Landroid/webkit/CallbackProxy$UploadFile;->mValue:Landroid/net/Uri;
  
+     .line 1571
      iget-object v1, p0, Landroid/webkit/CallbackProxy$UploadFile;->this$0:Landroid/webkit/CallbackProxy;
  
      monitor-enter v1
  
+     .line 1572
      :try_start_0
      iget-object v0, p0, Landroid/webkit/CallbackProxy$UploadFile;->this$0:Landroid/webkit/CallbackProxy;
  
      invoke-virtual {v0}, Ljava/lang/Object;->notify()V
  
+     .line 1573
      monitor-exit v1
  
+     .line 1574
      return-void
  
+     .line 1573
      :catchall_0
      move-exception v0
  
***************
*** 103,108 ****
--- 112,118 ----
      .parameter "x0"
  
      .prologue
+     .line 1567
      check-cast p1, Landroid/net/Uri;
  
      .end local p1
