*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CookieSyncManager.smali	2012-03-23 19:37:13.397019376 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CookieSyncManager.smali.original	2012-03-23 19:34:45.589324107 -0500
***************
*** 17,26 ****
--- 17,28 ----
      .parameter "context"
  
      .prologue
+     .line 69
      const-string v0, "CookieSyncManager"
  
      invoke-direct {p0, p1, v0}, Landroid/webkit/WebSyncManager;-><init>(Landroid/content/Context;Ljava/lang/String;)V
  
+     .line 70
      return-void
  .end method
  
***************
*** 28,37 ****
--- 30,41 ----
      .locals 2
  
      .prologue
+     .line 221
      sget-object v0, Landroid/webkit/CookieSyncManager;->sRef:Landroid/webkit/CookieSyncManager;
  
      if-nez v0, :cond_0
  
+     .line 222
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "CookieSyncManager::createInstance() needs to be called before CookieSyncManager::getInstance()"
***************
*** 40,45 ****
--- 44,50 ----
  
      throw v0
  
+     .line 226
      :cond_0
      return-void
  .end method
***************
*** 49,60 ****
--- 54,67 ----
      .parameter "context"
  
      .prologue
+     .line 91
      const-class v2, Landroid/webkit/CookieSyncManager;
  
      monitor-enter v2
  
      if-nez p0, :cond_0
  
+     .line 92
      :try_start_0
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
***************
*** 66,71 ****
--- 73,79 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 91
      :catchall_0
      move-exception v1
  
***************
*** 73,97 ****
--- 81,110 ----
  
      throw v1
  
+     .line 95
      :cond_0
      :try_start_1
      invoke-static {p0}, Landroid/webkit/JniUtil;->setContext(Landroid/content/Context;)V
  
+     .line 96
      invoke-virtual {p0}, Landroid/content/Context;->getApplicationContext()Landroid/content/Context;
  
      move-result-object v0
  
+     .line 97
      .local v0, appContext:Landroid/content/Context;
      sget-object v1, Landroid/webkit/CookieSyncManager;->sRef:Landroid/webkit/CookieSyncManager;
  
      if-nez v1, :cond_1
  
+     .line 98
      new-instance v1, Landroid/webkit/CookieSyncManager;
  
      invoke-direct {v1, v0}, Landroid/webkit/CookieSyncManager;-><init>(Landroid/content/Context;)V
  
      sput-object v1, Landroid/webkit/CookieSyncManager;->sRef:Landroid/webkit/CookieSyncManager;
  
+     .line 100
      :cond_1
      sget-object v1, Landroid/webkit/CookieSyncManager;->sRef:Landroid/webkit/CookieSyncManager;
      :try_end_1
***************
*** 106,111 ****
--- 119,125 ----
      .locals 2
  
      .prologue
+     .line 80
      const-class v1, Landroid/webkit/CookieSyncManager;
  
      monitor-enter v1
***************
*** 113,118 ****
--- 127,133 ----
      :try_start_0
      invoke-static {}, Landroid/webkit/CookieSyncManager;->checkInstanceIsCreated()V
  
+     .line 81
      sget-object v0, Landroid/webkit/CookieSyncManager;->sRef:Landroid/webkit/CookieSyncManager;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 121,126 ****
--- 136,142 ----
  
      return-object v0
  
+     .line 80
      :catchall_0
      move-exception v0
  
***************
*** 143,153 ****
--- 159,171 ----
      .end annotation
  
      .prologue
+     .line 202
      .local p1, list:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/webkit/CookieManager$Cookie;>;"
      invoke-virtual {p1}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
  
      move-result-object v1
  
+     .line 203
      .local v1, iter:Ljava/util/Iterator;,"Ljava/util/Iterator<Landroid/webkit/CookieManager$Cookie;>;"
      :cond_0
      :goto_0
***************
*** 157,168 ****
--- 175,188 ----
  
      if-eqz v2, :cond_3
  
+     .line 204
      invoke-interface {v1}, Ljava/util/Iterator;->next()Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Landroid/webkit/CookieManager$Cookie;
  
+     .line 205
      .local v0, cookie:Landroid/webkit/CookieManager$Cookie;
      iget-byte v2, v0, Landroid/webkit/CookieManager$Cookie;->mode:B
  
***************
*** 170,179 ****
--- 190,201 ----
  
      if-eq v2, v3, :cond_0
  
+     .line 206
      iget-byte v2, v0, Landroid/webkit/CookieManager$Cookie;->mode:B
  
      if-eqz v2, :cond_1
  
+     .line 207
      iget-object v2, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
      iget-object v3, v0, Landroid/webkit/CookieManager$Cookie;->domain:Ljava/lang/String;
***************
*** 184,189 ****
--- 206,212 ----
  
      invoke-virtual {v2, v3, v4, v5}, Landroid/webkit/WebViewDatabase;->deleteCookies(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 210
      :cond_1
      iget-byte v2, v0, Landroid/webkit/CookieManager$Cookie;->mode:B
  
***************
*** 191,200 ****
--- 214,225 ----
  
      if-eq v2, v3, :cond_2
  
+     .line 211
      iget-object v2, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
      invoke-virtual {v2, v0}, Landroid/webkit/WebViewDatabase;->addCookie(Landroid/webkit/CookieManager$Cookie;)V
  
+     .line 212
      invoke-static {}, Landroid/webkit/CookieManager;->getInstance()Landroid/webkit/CookieManager;
  
      move-result-object v2
***************
*** 203,208 ****
--- 228,234 ----
  
      goto :goto_0
  
+     .line 214
      :cond_2
      invoke-static {}, Landroid/webkit/CookieManager;->getInstance()Landroid/webkit/CookieManager;
  
***************
*** 212,217 ****
--- 238,244 ----
  
      goto :goto_0
  
+     .line 218
      .end local v0           #cookie:Landroid/webkit/CookieManager$Cookie;
      :cond_3
      return-void
***************
*** 223,235 ****
--- 250,265 ----
      .locals 1
  
      .prologue
+     .line 126
      iget-object v0, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
      if-nez v0, :cond_0
  
+     .line 131
      :goto_0
      return-void
  
+     .line 130
      :cond_0
      iget-object v0, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
***************
*** 243,255 ****
--- 273,288 ----
      .parameter "now"
  
      .prologue
+     .line 167
      iget-object v0, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
      if-nez v0, :cond_0
  
+     .line 172
      :goto_0
      return-void
  
+     .line 171
      :cond_0
      iget-object v0, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
***************
*** 262,274 ****
--- 295,310 ----
      .locals 1
  
      .prologue
+     .line 153
      iget-object v0, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
      if-nez v0, :cond_0
  
+     .line 158
      :goto_0
      return-void
  
+     .line 157
      :cond_0
      iget-object v0, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
***************
*** 293,306 ****
--- 329,345 ----
      .end annotation
  
      .prologue
+     .line 112
      iget-object v0, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
      if-nez v0, :cond_0
  
+     .line 113
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
+     .line 116
      :goto_0
      return-object v0
  
***************
*** 318,329 ****
--- 357,371 ----
      .locals 1
  
      .prologue
+     .line 139
      iget-object v0, p0, Landroid/webkit/CookieSyncManager;->mDataBase:Landroid/webkit/WebViewDatabase;
  
      if-nez v0, :cond_0
  
+     .line 140
      const/4 v0, 0x0
  
+     .line 143
      :goto_0
      return v0
  
***************
*** 341,346 ****
--- 383,389 ----
      .locals 0
  
      .prologue
+     .line 61
      invoke-super {p0}, Landroid/webkit/WebSyncManager;->resetSync()V
  
      return-void
***************
*** 350,355 ****
--- 393,399 ----
      .locals 0
  
      .prologue
+     .line 61
      invoke-super {p0}, Landroid/webkit/WebSyncManager;->run()V
  
      return-void
***************
*** 359,364 ****
--- 403,409 ----
      .locals 0
  
      .prologue
+     .line 61
      invoke-super {p0}, Landroid/webkit/WebSyncManager;->startSync()V
  
      return-void
***************
*** 368,373 ****
--- 413,419 ----
      .locals 0
  
      .prologue
+     .line 61
      invoke-super {p0}, Landroid/webkit/WebSyncManager;->stopSync()V
  
      return-void
***************
*** 377,382 ****
--- 423,429 ----
      .locals 0
  
      .prologue
+     .line 61
      invoke-super {p0}, Landroid/webkit/WebSyncManager;->sync()V
  
      return-void
***************
*** 386,395 ****
--- 433,444 ----
      .locals 5
  
      .prologue
+     .line 179
      invoke-static {}, Landroid/webkit/CookieManager;->getInstance()Landroid/webkit/CookieManager;
  
      move-result-object v2
  
+     .line 181
      .local v2, manager:Landroid/webkit/CookieManager;
      invoke-virtual {v2}, Landroid/webkit/CookieManager;->acceptCookie()Z
  
***************
*** 397,405 ****
--- 446,456 ----
  
      if-nez v3, :cond_0
  
+     .line 199
      :goto_0
      return-void
  
+     .line 185
      :cond_0
      invoke-static {}, Landroid/webkit/JniUtil;->useChromiumHttpStack()Z
  
***************
*** 407,416 ****
--- 458,469 ----
  
      if-eqz v3, :cond_1
  
+     .line 186
      invoke-virtual {v2}, Landroid/webkit/CookieManager;->flushCookieStore()V
  
      goto :goto_0
  
+     .line 188
      :cond_1
      iget-wide v3, p0, Landroid/webkit/CookieSyncManager;->mLastUpdate:J
  
***************
*** 418,423 ****
--- 471,477 ----
  
      move-result-object v0
  
+     .line 189
      .local v0, cookieList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/webkit/CookieManager$Cookie;>;"
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
***************
*** 425,436 ****
--- 479,493 ----
  
      iput-wide v3, p0, Landroid/webkit/CookieSyncManager;->mLastUpdate:J
  
+     .line 190
      invoke-direct {p0, v0}, Landroid/webkit/CookieSyncManager;->syncFromRamToFlash(Ljava/util/ArrayList;)V
  
+     .line 192
      invoke-virtual {v2}, Landroid/webkit/CookieManager;->deleteLRUDomain()Ljava/util/ArrayList;
  
      move-result-object v1
  
+     .line 193
      .local v1, lruList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/webkit/CookieManager$Cookie;>;"
      invoke-direct {p0, v1}, Landroid/webkit/CookieSyncManager;->syncFromRamToFlash(Ljava/util/ArrayList;)V
  
