*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CookieManager$CookieComparator.smali	2012-03-23 19:37:10.793971910 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CookieManager$CookieComparator.smali.original	2012-03-23 19:34:45.576323871 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 208
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 42,47 ****
--- 43,49 ----
      .parameter "x0"
  
      .prologue
+     .line 208
      invoke-direct {p0}, Landroid/webkit/CookieManager$CookieComparator;-><init>()V
  
      return-void
***************
*** 55,60 ****
--- 57,63 ----
      .parameter "cookie2"
  
      .prologue
+     .line 216
      iget-object v1, p2, Landroid/webkit/CookieManager$Cookie;->path:Ljava/lang/String;
  
      invoke-virtual {v1}, Ljava/lang/String;->length()I
***************
*** 69,82 ****
--- 72,88 ----
  
      sub-int v0, v1, v2
  
+     .line 217
      .local v0, diff:I
      if-eqz v0, :cond_0
  
      move v1, v0
  
+     .line 237
      :goto_0
      return v1
  
+     .line 219
      :cond_0
      iget-object v1, p2, Landroid/webkit/CookieManager$Cookie;->domain:Ljava/lang/String;
  
***************
*** 92,125 ****
--- 98,138 ----
  
      sub-int v0, v1, v2
  
+     .line 220
      if-eqz v0, :cond_1
  
      move v1, v0
  
      goto :goto_0
  
+     .line 224
      :cond_1
      iget-object v1, p2, Landroid/webkit/CookieManager$Cookie;->value:Ljava/lang/String;
  
      if-nez v1, :cond_2
  
+     .line 227
      iget-object v1, p1, Landroid/webkit/CookieManager$Cookie;->value:Ljava/lang/String;
  
      if-eqz v1, :cond_3
  
+     .line 228
      const/4 v1, -0x1
  
      goto :goto_0
  
+     .line 230
      :cond_2
      iget-object v1, p1, Landroid/webkit/CookieManager$Cookie;->value:Ljava/lang/String;
  
      if-nez v1, :cond_3
  
+     .line 233
      const/4 v1, 0x1
  
      goto :goto_0
  
+     .line 237
      :cond_3
      iget-object v1, p1, Landroid/webkit/CookieManager$Cookie;->name:Ljava/lang/String;
  
***************
*** 138,143 ****
--- 151,157 ----
      .parameter "x1"
  
      .prologue
+     .line 208
      check-cast p1, Landroid/webkit/CookieManager$Cookie;
  
      .end local p1
