*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CookieManager$Cookie.smali	2012-03-23 19:37:14.520039854 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/CookieManager$Cookie.smali.original	2012-03-23 19:34:45.596324235 -0500
***************
*** 49,56 ****
--- 49,58 ----
      .locals 0
  
      .prologue
+     .line 146
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 147
      return-void
  .end method
  
***************
*** 60,75 ****
--- 62,82 ----
      .parameter "defaultPath"
  
      .prologue
+     .line 149
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 150
      iput-object p1, p0, Landroid/webkit/CookieManager$Cookie;->domain:Ljava/lang/String;
  
+     .line 151
      iput-object p2, p0, Landroid/webkit/CookieManager$Cookie;->path:Ljava/lang/String;
  
+     .line 152
      const-wide/16 v0, -0x1
  
      iput-wide v0, p0, Landroid/webkit/CookieManager$Cookie;->expires:J
  
+     .line 153
      return-void
  .end method
  
***************
*** 84,89 ****
--- 91,97 ----
  
      const/4 v2, 0x1
  
+     .line 166
      iget-object v4, p0, Landroid/webkit/CookieManager$Cookie;->domain:Ljava/lang/String;
  
      const-string v5, "."
***************
*** 94,99 ****
--- 102,108 ----
  
      if-eqz v4, :cond_3
  
+     .line 167
      iget-object v4, p0, Landroid/webkit/CookieManager$Cookie;->domain:Ljava/lang/String;
  
      invoke-virtual {v4, v2}, Ljava/lang/String;->substring(I)Ljava/lang/String;
***************
*** 106,127 ****
--- 115,140 ----
  
      if-eqz v4, :cond_2
  
+     .line 168
      iget-object v4, p0, Landroid/webkit/CookieManager$Cookie;->domain:Ljava/lang/String;
  
      invoke-virtual {v4}, Ljava/lang/String;->length()I
  
      move-result v0
  
+     .line 169
      .local v0, len:I
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
      move-result v1
  
+     .line 170
      .local v1, urlLen:I
      add-int/lit8 v4, v0, -0x1
  
      if-le v1, v4, :cond_0
  
+     .line 172
      sub-int v4, v1, v0
  
      invoke-virtual {p1, v4}, Ljava/lang/String;->charAt(I)C
***************
*** 132,137 ****
--- 145,151 ----
  
      if-ne v4, v5, :cond_1
  
+     .line 179
      .end local v0           #len:I
      .end local v1           #urlLen:I
      :cond_0
***************
*** 143,148 ****
--- 157,163 ----
      :cond_1
      move v2, v3
  
+     .line 172
      goto :goto_0
  
      .end local v0           #len:I
***************
*** 150,157 ****
--- 165,174 ----
      :cond_2
      move v2, v3
  
+     .line 176
      goto :goto_0
  
+     .line 179
      :cond_3
      iget-object v2, p0, Landroid/webkit/CookieManager$Cookie;->domain:Ljava/lang/String;
  
***************
*** 171,176 ****
--- 188,194 ----
  
      const/4 v3, 0x0
  
+     .line 160
      iget-object v1, p0, Landroid/webkit/CookieManager$Cookie;->value:Ljava/lang/String;
  
      if-nez v1, :cond_0
***************
*** 191,196 ****
--- 209,215 ----
  
      move v0, v2
  
+     .line 161
      .local v0, valuesMatch:Z
      :goto_2
      iget-object v1, p0, Landroid/webkit/CookieManager$Cookie;->domain:Ljava/lang/String;
***************
*** 232,237 ****
--- 251,257 ----
      :cond_0
      move v1, v3
  
+     .line 160
      goto :goto_0
  
      :cond_1
***************
*** 248,253 ****
--- 268,274 ----
      :cond_3
      move v2, v3
  
+     .line 161
      goto :goto_3
  .end method
  
***************
*** 262,267 ****
--- 283,289 ----
  
      const/4 v3, 0x0
  
+     .line 184
      iget-object v4, p0, Landroid/webkit/CookieManager$Cookie;->path:Ljava/lang/String;
  
      invoke-virtual {p1, v4}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
***************
*** 270,301 ****
  
      if-eqz v4, :cond_0
  
      iget-object v4, p0, Landroid/webkit/CookieManager$Cookie;->path:Ljava/lang/String;
  
      invoke-virtual {v4}, Ljava/lang/String;->length()I
  
      move-result v0
  
      .local v0, len:I
      if-nez v0, :cond_1
  
!     const-string v2, "webkit"
  
      const-string v4, "Empty cookie path"
  
      invoke-static {v2, v4}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
      .end local v0           #len:I
      :cond_0
      :goto_0
      return v3
  
      .restart local v0       #len:I
      :cond_1
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
      move-result v1
  
      .local v1, urlLen:I
      iget-object v4, p0, Landroid/webkit/CookieManager$Cookie;->path:Ljava/lang/String;
  
--- 292,329 ----
  
      if-eqz v4, :cond_0
  
+     .line 185
      iget-object v4, p0, Landroid/webkit/CookieManager$Cookie;->path:Ljava/lang/String;
  
      invoke-virtual {v4}, Ljava/lang/String;->length()I
  
      move-result v0
  
+     .line 186
      .local v0, len:I
      if-nez v0, :cond_1
  
!     .line 187
!     const-string/jumbo v2, "webkit"
  
      const-string v4, "Empty cookie path"
  
      invoke-static {v2, v4}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 197
      .end local v0           #len:I
      :cond_0
      :goto_0
      return v3
  
+     .line 190
      .restart local v0       #len:I
      :cond_1
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
      move-result v1
  
+     .line 191
      .local v1, urlLen:I
      iget-object v4, p0, Landroid/webkit/CookieManager$Cookie;->path:Ljava/lang/String;
  
***************
*** 309,314 ****
--- 337,343 ----
  
      if-le v1, v0, :cond_3
  
+     .line 193
      invoke-virtual {p1, v0}, Ljava/lang/String;->charAt(I)C
  
      move-result v4
***************
*** 328,333 ****
--- 357,363 ----
      :cond_3
      move v3, v2
  
+     .line 195
      goto :goto_0
  .end method
  
***************
*** 335,340 ****
--- 365,371 ----
      .locals 2
  
      .prologue
+     .line 201
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
