*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/DebugFlags.smali	2012-03-23 19:37:14.342036609 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/DebugFlags.smali.original	2012-03-23 19:34:45.595324217 -0500
***************
*** 48,53 ****
--- 48,54 ----
      .locals 0
  
      .prologue
+     .line 28
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
