*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/DeviceOrientationService.smali	2012-03-23 19:37:05.666878418 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/DeviceOrientationService.smali.original	2012-03-23 19:34:45.543323269 -0500
***************
*** 42,47 ****
--- 42,48 ----
      .locals 1
  
      .prologue
+     .line 30
      const-class v0, Landroid/webkit/DeviceOrientationService;
  
      invoke-virtual {v0}, Ljava/lang/Class;->desiredAssertionStatus()Z
***************
*** 69,78 ****
--- 70,82 ----
      .parameter "context"
  
      .prologue
+     .line 48
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 49
      iput-object p1, p0, Landroid/webkit/DeviceOrientationService;->mManager:Landroid/webkit/DeviceMotionAndOrientationManager;
  
+     .line 50
      sget-boolean v0, Landroid/webkit/DeviceOrientationService;->$assertionsDisabled:Z
  
      if-nez v0, :cond_0
***************
*** 87,95 ****
--- 91,101 ----
  
      throw v0
  
+     .line 51
      :cond_0
      iput-object p2, p0, Landroid/webkit/DeviceOrientationService;->mContext:Landroid/content/Context;
  
+     .line 52
      sget-boolean v0, Landroid/webkit/DeviceOrientationService;->$assertionsDisabled:Z
  
      if-nez v0, :cond_1
***************
*** 104,109 ****
--- 110,116 ----
  
      throw v0
  
+     .line 53
      :cond_1
      return-void
  .end method
***************
*** 113,118 ****
--- 120,126 ----
      .parameter "x0"
  
      .prologue
+     .line 30
      iget-boolean v0, p0, Landroid/webkit/DeviceOrientationService;->mIsRunning:Z
  
      return v0
***************
*** 123,128 ****
--- 131,137 ----
      .parameter "x0"
  
      .prologue
+     .line 30
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mManager:Landroid/webkit/DeviceMotionAndOrientationManager;
  
      return-object v0
***************
*** 134,139 ****
--- 143,149 ----
      .prologue
      const-wide v13, 0x4076800000000000L
  
+     .line 106
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mGravityVector:[F
  
      if-eqz v0, :cond_0
***************
*** 142,156 ****
--- 152,169 ----
  
      if-nez v0, :cond_1
  
+     .line 134
      :cond_0
      :goto_0
      return-void
  
+     .line 112
      :cond_1
      const/16 v0, 0x9
  
      new-array v7, v0, [F
  
+     .line 113
      .local v7, deviceRotationMatrix:[F
      const/4 v0, 0x0
  
***************
*** 164,176 ****
--- 177,192 ----
  
      if-eqz v0, :cond_0
  
+     .line 124
      const/4 v0, 0x3
  
      new-array v8, v0, [F
  
+     .line 125
      .local v8, rotationAngles:[F
      invoke-static {v7, v8}, Landroid/hardware/SensorManager;->getOrientation([F[F)[F
  
+     .line 126
      const/4 v0, 0x0
  
      aget v0, v8, v0
***************
*** 187,192 ****
--- 203,209 ----
  
      sub-double v1, v9, v11
  
+     .line 127
      .local v1, alpha:D
      :goto_1
      const-wide/16 v9, 0x0
***************
*** 199,204 ****
--- 216,222 ----
  
      goto :goto_1
  
+     .line 128
      :cond_2
      const/4 v0, 0x1
  
***************
*** 212,217 ****
--- 230,236 ----
  
      move-result-wide v3
  
+     .line 129
      .local v3, beta:D
      :goto_2
      const-wide v9, -0x3f99800000000000L
***************
*** 224,229 ****
--- 243,249 ----
  
      goto :goto_2
  
+     .line 130
      :cond_3
      const/4 v0, 0x2
  
***************
*** 235,240 ****
--- 255,261 ----
  
      move-result-wide v5
  
+     .line 131
      .local v5, gamma:D
      :goto_3
      const-wide v9, -0x3fa9800000000000L
***************
*** 250,255 ****
--- 271,277 ----
      :cond_4
      move-object v0, p0
  
+     .line 133
      invoke-direct/range {v0 .. v6}, Landroid/webkit/DeviceOrientationService;->maybeSendChange(DDD)V
  
      goto :goto_0
***************
*** 259,264 ****
--- 281,287 ----
      .locals 2
  
      .prologue
+     .line 137
      sget-boolean v0, Landroid/webkit/DeviceOrientationService;->$assertionsDisabled:Z
  
      if-nez v0, :cond_0
***************
*** 285,298 ****
  
      throw v0
  
      :cond_0
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mSensorManager:Landroid/hardware/SensorManager;
  
      if-nez v0, :cond_1
  
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mContext:Landroid/content/Context;
  
!     const-string v1, "sensor"
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
--- 308,323 ----
  
      throw v0
  
+     .line 138
      :cond_0
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mSensorManager:Landroid/hardware/SensorManager;
  
      if-nez v0, :cond_1
  
+     .line 139
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mContext:Landroid/content/Context;
  
!     const-string/jumbo v1, "sensor"
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
***************
*** 302,307 ****
--- 327,333 ----
  
      iput-object v0, p0, Landroid/webkit/DeviceOrientationService;->mSensorManager:Landroid/hardware/SensorManager;
  
+     .line 141
      :cond_1
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mSensorManager:Landroid/hardware/SensorManager;
  
***************
*** 317,322 ****
--- 343,349 ----
      .prologue
      const-wide/high16 v2, 0x3ff0
  
+     .line 169
      sget-boolean v0, Landroid/webkit/DeviceOrientationService;->$assertionsDisabled:Z
  
      if-nez v0, :cond_0
***************
*** 343,348 ****
--- 370,376 ----
  
      throw v0
  
+     .line 170
      :cond_0
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mAlpha:Ljava/lang/Double;
  
***************
*** 404,409 ****
--- 432,438 ----
  
      if-lez v0, :cond_2
  
+     .line 174
      :cond_1
      invoke-static {p1, p2}, Ljava/lang/Double;->valueOf(D)Ljava/lang/Double;
  
***************
*** 411,428 ****
--- 440,460 ----
  
      iput-object v0, p0, Landroid/webkit/DeviceOrientationService;->mAlpha:Ljava/lang/Double;
  
+     .line 175
      invoke-static {p3, p4}, Ljava/lang/Double;->valueOf(D)Ljava/lang/Double;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/webkit/DeviceOrientationService;->mBeta:Ljava/lang/Double;
  
+     .line 176
      invoke-static {p5, p6}, Ljava/lang/Double;->valueOf(D)Ljava/lang/Double;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/webkit/DeviceOrientationService;->mGamma:Ljava/lang/Double;
  
+     .line 177
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mManager:Landroid/webkit/DeviceMotionAndOrientationManager;
  
      iget-object v1, p0, Landroid/webkit/DeviceOrientationService;->mAlpha:Ljava/lang/Double;
***************
*** 433,442 ****
--- 465,476 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/webkit/DeviceMotionAndOrientationManager;->onOrientationChange(Ljava/lang/Double;Ljava/lang/Double;Ljava/lang/Double;)V
  
+     .line 179
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/webkit/DeviceOrientationService;->mHaveSentErrorEvent:Z
  
+     .line 181
      :cond_2
      return-void
  .end method
***************
*** 447,452 ****
--- 481,487 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 145
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->getSensorManager()Landroid/hardware/SensorManager;
  
      move-result-object v1
***************
*** 457,462 ****
--- 492,498 ----
  
      move-result-object v0
  
+     .line 146
      .local v0, sensors:Ljava/util/List;,"Ljava/util/List<Landroid/hardware/Sensor;>;"
      invoke-interface {v0}, Ljava/util/List;->isEmpty()Z
  
***************
*** 466,471 ****
--- 502,508 ----
  
      move v1, v2
  
+     .line 150
      :goto_0
      return v1
  
***************
*** 495,500 ****
--- 532,538 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 155
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->getSensorManager()Landroid/hardware/SensorManager;
  
      move-result-object v1
***************
*** 505,510 ****
--- 543,549 ----
  
      move-result-object v0
  
+     .line 156
      .local v0, sensors:Ljava/util/List;,"Ljava/util/List<Landroid/hardware/Sensor;>;"
      invoke-interface {v0}, Ljava/util/List;->isEmpty()Z
  
***************
*** 514,519 ****
--- 553,559 ----
  
      move v1, v2
  
+     .line 160
      :goto_0
      return v1
  
***************
*** 541,556 ****
--- 581,599 ----
      .locals 1
  
      .prologue
+     .line 96
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mHandler:Landroid/os/Handler;
  
      if-nez v0, :cond_0
  
+     .line 97
      new-instance v0, Landroid/os/Handler;
  
      invoke-direct {v0}, Landroid/os/Handler;-><init>()V
  
      iput-object v0, p0, Landroid/webkit/DeviceOrientationService;->mHandler:Landroid/os/Handler;
  
+     .line 99
      :cond_0
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->registerForAccelerometerSensor()Z
  
***************
*** 564,574 ****
--- 607,620 ----
  
      if-nez v0, :cond_2
  
+     .line 100
      :cond_1
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->unregisterFromSensors()V
  
+     .line 101
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->sendErrorEvent()V
  
+     .line 103
      :cond_2
      return-void
  .end method
***************
*** 577,582 ****
--- 623,629 ----
      .locals 2
  
      .prologue
+     .line 78
      sget-boolean v0, Landroid/webkit/DeviceOrientationService;->$assertionsDisabled:Z
  
      if-nez v0, :cond_0
***************
*** 603,621 ****
--- 650,672 ----
  
      throw v0
  
+     .line 80
      :cond_0
      iget-boolean v0, p0, Landroid/webkit/DeviceOrientationService;->mHaveSentErrorEvent:Z
  
      if-eqz v0, :cond_1
  
+     .line 93
      :goto_0
      return-void
  
+     .line 82
      :cond_1
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/webkit/DeviceOrientationService;->mHaveSentErrorEvent:Z
  
+     .line 83
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mHandler:Landroid/os/Handler;
  
      new-instance v1, Landroid/webkit/DeviceOrientationService$1;
***************
*** 631,642 ****
--- 682,695 ----
      .locals 1
  
      .prologue
+     .line 165
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->getSensorManager()Landroid/hardware/SensorManager;
  
      move-result-object v0
  
      invoke-virtual {v0, p0}, Landroid/hardware/SensorManager;->unregisterListener(Landroid/hardware/SensorEventListener;)V
  
+     .line 166
      return-void
  .end method
  
***************
*** 648,653 ****
--- 701,707 ----
      .parameter "accuracy"
  
      .prologue
+     .line 221
      sget-boolean v0, Landroid/webkit/DeviceOrientationService;->$assertionsDisabled:Z
  
      if-nez v0, :cond_0
***************
*** 674,679 ****
--- 728,734 ----
  
      throw v0
  
+     .line 222
      :cond_0
      return-void
  .end method
***************
*** 691,696 ****
--- 746,752 ----
  
      const/4 v2, 0x0
  
+     .line 188
      sget-boolean v0, Landroid/webkit/DeviceOrientationService;->$assertionsDisabled:Z
  
      if-nez v0, :cond_0
***************
*** 707,712 ****
--- 763,769 ----
  
      throw v0
  
+     .line 189
      :cond_0
      sget-boolean v0, Landroid/webkit/DeviceOrientationService;->$assertionsDisabled:Z
  
***************
*** 734,748 ****
--- 791,808 ----
  
      throw v0
  
+     .line 192
      :cond_1
      iget-boolean v0, p0, Landroid/webkit/DeviceOrientationService;->mIsRunning:Z
  
      if-nez v0, :cond_3
  
+     .line 218
      :cond_2
      :goto_0
      return-void
  
+     .line 196
      :cond_3
      iget-object v0, p1, Landroid/hardware/SensorEvent;->sensor:Landroid/hardware/Sensor;
  
***************
*** 752,757 ****
--- 812,818 ----
  
      packed-switch v0, :pswitch_data_0
  
+     .line 216
      sget-boolean v0, Landroid/webkit/DeviceOrientationService;->$assertionsDisabled:Z
  
      if-nez v0, :cond_2
***************
*** 762,776 ****
--- 823,840 ----
  
      throw v0
  
+     .line 198
      :pswitch_0
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mGravityVector:[F
  
      if-nez v0, :cond_4
  
+     .line 199
      new-array v0, v5, [F
  
      iput-object v0, p0, Landroid/webkit/DeviceOrientationService;->mGravityVector:[F
  
+     .line 201
      :cond_4
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mGravityVector:[F
  
***************
*** 780,785 ****
--- 844,850 ----
  
      aput v1, v0, v2
  
+     .line 202
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mGravityVector:[F
  
      iget-object v1, p1, Landroid/hardware/SensorEvent;->values:[F
***************
*** 788,793 ****
--- 853,859 ----
  
      aput v1, v0, v3
  
+     .line 203
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mGravityVector:[F
  
      iget-object v1, p1, Landroid/hardware/SensorEvent;->values:[F
***************
*** 796,814 ****
--- 862,884 ----
  
      aput v1, v0, v4
  
+     .line 204
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->getOrientationUsingGetRotationMatrix()V
  
      goto :goto_0
  
+     .line 207
      :pswitch_1
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mMagneticFieldVector:[F
  
      if-nez v0, :cond_5
  
+     .line 208
      new-array v0, v5, [F
  
      iput-object v0, p0, Landroid/webkit/DeviceOrientationService;->mMagneticFieldVector:[F
  
+     .line 210
      :cond_5
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mMagneticFieldVector:[F
  
***************
*** 818,823 ****
--- 888,894 ----
  
      aput v1, v0, v2
  
+     .line 211
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mMagneticFieldVector:[F
  
      iget-object v1, p1, Landroid/hardware/SensorEvent;->values:[F
***************
*** 826,831 ****
--- 897,903 ----
  
      aput v1, v0, v3
  
+     .line 212
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService;->mMagneticFieldVector:[F
  
      iget-object v1, p1, Landroid/hardware/SensorEvent;->values:[F
***************
*** 834,843 ****
--- 906,917 ----
  
      aput v1, v0, v4
  
+     .line 213
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->getOrientationUsingGetRotationMatrix()V
  
      goto :goto_0
  
+     .line 196
      nop
  
      :pswitch_data_0
***************
*** 851,862 ****
--- 925,939 ----
      .locals 1
  
      .prologue
+     .line 72
      iget-boolean v0, p0, Landroid/webkit/DeviceOrientationService;->mIsRunning:Z
  
      if-eqz v0, :cond_0
  
+     .line 73
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->registerForSensors()V
  
+     .line 75
      :cond_0
      return-void
  .end method
***************
*** 865,876 ****
--- 942,956 ----
      .locals 1
  
      .prologue
+     .line 56
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/webkit/DeviceOrientationService;->mIsRunning:Z
  
+     .line 57
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->registerForSensors()V
  
+     .line 58
      return-void
  .end method
  
***************
*** 878,889 ****
--- 958,972 ----
      .locals 1
  
      .prologue
+     .line 61
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/webkit/DeviceOrientationService;->mIsRunning:Z
  
+     .line 62
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->unregisterFromSensors()V
  
+     .line 63
      return-void
  .end method
  
***************
*** 891,902 ****
--- 974,988 ----
      .locals 1
  
      .prologue
+     .line 66
      iget-boolean v0, p0, Landroid/webkit/DeviceOrientationService;->mIsRunning:Z
  
      if-eqz v0, :cond_0
  
+     .line 67
      invoke-direct {p0}, Landroid/webkit/DeviceOrientationService;->unregisterFromSensors()V
  
+     .line 69
      :cond_0
      return-void
  .end method
