*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/DeviceMotionAndOrientationManager.smali	2012-03-23 19:37:06.490893445 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/DeviceMotionAndOrientationManager.smali.original	2012-03-23 19:34:45.549323377 -0500
***************
*** 16,21 ****
--- 16,22 ----
      .locals 1
  
      .prologue
+     .line 27
      const-class v0, Landroid/webkit/DeviceMotionAndOrientationManager;
  
      invoke-virtual {v0}, Ljava/lang/Class;->desiredAssertionStatus()Z
***************
*** 42,51 ****
--- 43,55 ----
      .parameter "webViewCore"
  
      .prologue
+     .line 30
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 31
      iput-object p1, p0, Landroid/webkit/DeviceMotionAndOrientationManager;->mWebViewCore:Landroid/webkit/WebViewCore;
  
+     .line 32
      return-void
  .end method
  
***************
*** 71,76 ****
--- 75,81 ----
      .parameter "interval"
  
      .prologue
+     .line 55
      iget-object v0, p0, Landroid/webkit/DeviceMotionAndOrientationManager;->mWebViewCore:Landroid/webkit/WebViewCore;
  
      if-eqz p1, :cond_0
***************
*** 113,120 ****
--- 118,127 ----
  
      invoke-static/range {v0 .. v11}, Landroid/webkit/DeviceMotionAndOrientationManager;->nativeOnMotionChange(Landroid/webkit/WebViewCore;ZDZDZDD)V
  
+     .line 60
      return-void
  
+     .line 55
      :cond_0
      const/4 v1, 0x0
  
***************
*** 159,164 ****
--- 166,172 ----
  
      const-wide/16 v8, 0x0
  
+     .line 62
      iget-object v0, p0, Landroid/webkit/DeviceMotionAndOrientationManager;->mWebViewCore:Landroid/webkit/WebViewCore;
  
      if-eqz p1, :cond_1
***************
*** 197,207 ****
--- 205,217 ----
      :cond_0
      invoke-static/range {v0 .. v9}, Landroid/webkit/DeviceMotionAndOrientationManager;->nativeOnOrientationChange(Landroid/webkit/WebViewCore;ZDZDZD)V
  
+     .line 66
      return-void
  
      :cond_1
      move v1, v10
  
+     .line 62
      goto :goto_0
  
      :cond_2
***************
*** 235,240 ****
--- 245,251 ----
      .parameter "gamma"
  
      .prologue
+     .line 48
      sget-boolean v0, Landroid/webkit/DeviceMotionAndOrientationManager;->$assertionsDisabled:Z
  
      if-nez v0, :cond_0
***************
*** 261,266 ****
--- 272,278 ----
  
      throw v0
  
+     .line 49
      :cond_0
      iget-object v0, p0, Landroid/webkit/DeviceMotionAndOrientationManager;->mWebViewCore:Landroid/webkit/WebViewCore;
  
***************
*** 278,283 ****
--- 290,296 ----
  
      invoke-static/range {v0 .. v9}, Landroid/webkit/DeviceMotionAndOrientationManager;->nativeSetMockOrientation(Landroid/webkit/WebViewCore;ZDZDZD)V
  
+     .line 51
      return-void
  .end method
  
***************
*** 285,290 ****
--- 298,304 ----
      .locals 2
  
      .prologue
+     .line 39
      sget-boolean v0, Landroid/webkit/DeviceMotionAndOrientationManager;->$assertionsDisabled:Z
  
      if-nez v0, :cond_0
***************
*** 311,320 ****
--- 325,336 ----
  
      throw v0
  
+     .line 40
      :cond_0
      iget-object v0, p0, Landroid/webkit/DeviceMotionAndOrientationManager;->mWebViewCore:Landroid/webkit/WebViewCore;
  
      invoke-static {v0}, Landroid/webkit/DeviceMotionAndOrientationManager;->nativeUseMock(Landroid/webkit/WebViewCore;)V
  
+     .line 41
      return-void
  .end method
