*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/DeviceOrientationService$1.smali	2012-03-23 19:37:08.799935548 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/DeviceOrientationService$1.smali.original	2012-03-23 19:34:45.565323669 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .locals 1
  
      .prologue
+     .line 83
      const-class v0, Landroid/webkit/DeviceOrientationService;
  
      invoke-virtual {v0}, Ljava/lang/Class;->desiredAssertionStatus()Z
***************
*** 56,61 ****
--- 57,63 ----
      .parameter
  
      .prologue
+     .line 83
      iput-object p1, p0, Landroid/webkit/DeviceOrientationService$1;->this$0:Landroid/webkit/DeviceOrientationService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 71,76 ****
--- 73,79 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 86
      sget-boolean v0, Landroid/webkit/DeviceOrientationService$1;->$assertionsDisabled:Z
  
      if-nez v0, :cond_0
***************
*** 97,102 ****
--- 100,106 ----
  
      throw v0
  
+     .line 87
      :cond_0
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService$1;->this$0:Landroid/webkit/DeviceOrientationService;
  
***************
*** 107,112 ****
--- 111,117 ----
  
      if-eqz v0, :cond_1
  
+     .line 89
      iget-object v0, p0, Landroid/webkit/DeviceOrientationService$1;->this$0:Landroid/webkit/DeviceOrientationService;
  
      #getter for: Landroid/webkit/DeviceOrientationService;->mManager:Landroid/webkit/DeviceMotionAndOrientationManager;
***************
*** 116,121 ****
--- 121,127 ----
  
      invoke-virtual {v0, v2, v2, v2}, Landroid/webkit/DeviceMotionAndOrientationManager;->onOrientationChange(Ljava/lang/Double;Ljava/lang/Double;Ljava/lang/Double;)V
  
+     .line 91
      :cond_1
      return-void
  .end method
