*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/FileLoader.smali	2012-03-23 19:37:11.787990036 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/FileLoader.smali.original	2012-03-23 19:34:45.581323961 -0500
***************
*** 30,47 ****
--- 30,52 ----
      .parameter "allowFileAccess"
  
      .prologue
+     .line 63
      invoke-direct {p0, p2}, Landroid/webkit/StreamLoader;-><init>(Landroid/webkit/LoadListener;)V
  
+     .line 64
      iput p3, p0, Landroid/webkit/FileLoader;->mType:I
  
+     .line 65
      iput-boolean p4, p0, Landroid/webkit/FileLoader;->mAllowFileAccess:Z
  
+     .line 68
      const/16 v1, 0x3f
  
      invoke-virtual {p1, v1}, Ljava/lang/String;->indexOf(I)I
  
      move-result v0
  
+     .line 69
      .local v0, index:I
      iget v1, p0, Landroid/webkit/FileLoader;->mType:I
  
***************
*** 49,54 ****
--- 54,60 ----
  
      if-ne v1, v2, :cond_1
  
+     .line 70
      if-lez v0, :cond_0
  
      const-string v1, "file:///android_asset/"
***************
*** 68,76 ****
--- 74,84 ----
      :goto_0
      iput-object v1, p0, Landroid/webkit/FileLoader;->mPath:Ljava/lang/String;
  
+     .line 85
      :goto_1
      return-void
  
+     .line 70
      :cond_0
      const-string v1, "file:///android_asset/"
  
***************
*** 88,93 ****
--- 96,102 ----
  
      goto :goto_0
  
+     .line 74
      :cond_1
      iget v1, p0, Landroid/webkit/FileLoader;->mType:I
  
***************
*** 95,100 ****
--- 104,110 ----
  
      if-ne v1, v2, :cond_3
  
+     .line 75
      if-lez v0, :cond_2
  
      const-string v1, "file:///android_res/"
***************
*** 133,138 ****
--- 143,149 ----
  
      goto :goto_2
  
+     .line 80
      :cond_3
      if-lez v0, :cond_4
  
***************
*** 178,187 ****
--- 189,200 ----
      .parameter "ex"
  
      .prologue
+     .line 88
      invoke-virtual {p1}, Ljava/lang/Exception;->getMessage()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 89
      .local v1, exMessage:Ljava/lang/String;
      iget-object v2, p0, Landroid/webkit/FileLoader;->mContext:Landroid/content/Context;
  
***************
*** 191,199 ****
--- 204,214 ----
  
      move-result-object v0
  
+     .line 90
      .local v0, errString:Ljava/lang/String;
      if-eqz v1, :cond_0
  
+     .line 91
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 216,221 ****
--- 231,237 ----
  
      move-result-object v0
  
+     .line 93
      :cond_0
      return-object v0
  .end method
***************
*** 227,232 ****
--- 243,249 ----
      .parameter "headers"
  
      .prologue
+     .line 188
      return-void
  .end method
  
***************
*** 234,239 ****
--- 251,257 ----
      .locals 17
  
      .prologue
+     .line 99
      :try_start_0
      move-object/from16 v0, p0
  
***************
*** 246,251 ****
--- 264,270 ----
  
      if-ne v12, v13, :cond_1
  
+     .line 101
      :try_start_1
      move-object/from16 v0, p0
  
***************
*** 270,275 ****
--- 289,295 ----
      .catch Ljava/io/FileNotFoundException; {:try_start_1 .. :try_end_1} :catch_0
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_7
  
+     .line 172
      :cond_0
      :goto_0
      :try_start_2
***************
*** 287,300 ****
--- 307,323 ----
  
      invoke-virtual/range {v12 .. v16}, Landroid/webkit/LoadListener;->status(IIILjava/lang/String;)V
  
+     .line 182
      const/4 v12, 0x1
  
      :goto_1
      return v12
  
+     .line 102
      :catch_0
      move-exception v5
  
+     .line 104
      .local v5, ex:Ljava/io/FileNotFoundException;
      move-object/from16 v0, p0
  
***************
*** 321,330 ****
--- 344,355 ----
  
      goto :goto_0
  
+     .line 174
      .end local v5           #ex:Ljava/io/FileNotFoundException;
      :catch_1
      move-exception v5
  
+     .line 175
      .restart local v5       #ex:Ljava/io/FileNotFoundException;
      move-object/from16 v0, p0
  
***************
*** 340,349 ****
--- 365,376 ----
  
      invoke-virtual {v12, v13, v14}, Landroid/webkit/LoadListener;->error(ILjava/lang/String;)V
  
+     .line 176
      const/4 v12, 0x0
  
      goto :goto_1
  
+     .line 106
      .end local v5           #ex:Ljava/io/FileNotFoundException;
      :cond_1
      :try_start_3
***************
*** 355,360 ****
--- 382,388 ----
  
      if-ne v12, v13, :cond_7
  
+     .line 110
      move-object/from16 v0, p0
  
      iget-object v12, v0, Landroid/webkit/FileLoader;->mPath:Ljava/lang/String;
***************
*** 371,383 ****
  
      if-nez v12, :cond_3
  
      :cond_2
!     const-string v12, "webkit"
  
      const-string v13, "Need a path to resolve the res file"
  
      invoke-static {v12, v13}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
      move-object/from16 v0, p0
  
      iget-object v12, v0, Landroid/webkit/FileLoader;->mLoadListener:Landroid/webkit/LoadListener;
--- 399,413 ----
  
      if-nez v12, :cond_3
  
+     .line 111
      :cond_2
!     const-string/jumbo v12, "webkit"
  
      const-string v13, "Need a path to resolve the res file"
  
      invoke-static {v12, v13}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 112
      move-object/from16 v0, p0
  
      iget-object v12, v0, Landroid/webkit/FileLoader;->mLoadListener:Landroid/webkit/LoadListener;
***************
*** 396,405 ****
--- 426,437 ----
  
      invoke-virtual {v12, v13, v14}, Landroid/webkit/LoadListener;->error(ILjava/lang/String;)V
  
+     .line 114
      const/4 v12, 0x0
  
      goto :goto_1
  
+     .line 117
      :cond_3
      move-object/from16 v0, p0
  
***************
*** 411,416 ****
--- 443,449 ----
  
      move-result v9
  
+     .line 118
      .local v9, slash:I
      move-object/from16 v0, p0
  
***************
*** 422,427 ****
--- 455,461 ----
  
      move-result v2
  
+     .line 119
      .local v2, dot:I
      const/4 v12, -0x1
  
***************
*** 431,438 ****
  
      if-ne v2, v12, :cond_5
  
      :cond_4
!     const-string v12, "webkit"
  
      new-instance v13, Ljava/lang/StringBuilder;
  
--- 465,473 ----
  
      if-ne v2, v12, :cond_5
  
+     .line 120
      :cond_4
!     const-string/jumbo v12, "webkit"
  
      new-instance v13, Ljava/lang/StringBuilder;
  
***************
*** 458,463 ****
--- 493,499 ----
  
      invoke-static {v12, v13}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 121
      move-object/from16 v0, p0
  
      iget-object v12, v0, Landroid/webkit/FileLoader;->mLoadListener:Landroid/webkit/LoadListener;
***************
*** 476,485 ****
--- 512,523 ----
  
      invoke-virtual {v12, v13, v14}, Landroid/webkit/LoadListener;->error(ILjava/lang/String;)V
  
+     .line 123
      const/4 v12, 0x0
  
      goto/16 :goto_1
  
+     .line 125
      :cond_5
      move-object/from16 v0, p0
  
***************
*** 491,496 ****
--- 529,535 ----
  
      move-result-object v10
  
+     .line 126
      .local v10, subClassName:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 505,513 ****
--- 544,554 ----
  
      move-result-object v7
  
+     .line 127
      .local v7, fieldName:Ljava/lang/String;
      const/4 v4, 0x0
  
+     .line 129
      .local v4, errorMsg:Ljava/lang/String;
      :try_start_4
      move-object/from16 v0, p0
***************
*** 556,566 ****
--- 597,609 ----
  
      move-result-object v1
  
+     .line 133
      .local v1, d:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
      invoke-virtual {v1, v7}, Ljava/lang/Class;->getField(Ljava/lang/String;)Ljava/lang/reflect/Field;
  
      move-result-object v6
  
+     .line 134
      .local v6, field:Ljava/lang/reflect/Field;
      const/4 v12, 0x0
  
***************
*** 568,578 ****
--- 611,623 ----
  
      move-result v8
  
+     .line 135
      .local v8, id:I
      new-instance v11, Landroid/util/TypedValue;
  
      invoke-direct {v11}, Landroid/util/TypedValue;-><init>()V
  
+     .line 136
      .local v11, value:Landroid/util/TypedValue;
      move-object/from16 v0, p0
  
***************
*** 586,597 ****
--- 631,644 ----
  
      invoke-virtual {v12, v8, v11, v13}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 137
      iget v12, v11, Landroid/util/TypedValue;->type:I
  
      const/4 v13, 0x3
  
      if-ne v12, v13, :cond_6
  
+     .line 138
      move-object/from16 v0, p0
  
      iget-object v12, v0, Landroid/webkit/FileLoader;->mContext:Landroid/content/Context;
***************
*** 626,631 ****
--- 673,679 ----
      .catch Ljava/io/FileNotFoundException; {:try_start_4 .. :try_end_4} :catch_1
      .catch Ljava/io/IOException; {:try_start_4 .. :try_end_4} :catch_7
  
+     .line 157
      .end local v1           #d:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
      .end local v6           #field:Ljava/lang/reflect/Field;
      .end local v8           #id:I
***************
*** 633,638 ****
--- 681,687 ----
      :goto_2
      if-eqz v4, :cond_0
  
+     .line 158
      :try_start_5
      move-object/from16 v0, p0
  
***************
*** 655,664 ****
--- 704,715 ----
      .catch Ljava/io/FileNotFoundException; {:try_start_5 .. :try_end_5} :catch_1
      .catch Ljava/io/IOException; {:try_start_5 .. :try_end_5} :catch_7
  
+     .line 160
      const/4 v12, 0x0
  
      goto/16 :goto_1
  
+     .line 142
      .restart local v1       #d:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
      .restart local v6       #field:Ljava/lang/reflect/Field;
      .restart local v8       #id:I
***************
*** 677,682 ****
--- 728,734 ----
  
      goto :goto_2
  
+     .line 144
      .end local v1           #d:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
      .end local v6           #field:Ljava/lang/reflect/Field;
      .end local v8           #id:I
***************
*** 684,689 ****
--- 736,742 ----
      :catch_2
      move-exception v3
  
+     .line 145
      .local v3, e:Ljava/lang/ClassNotFoundException;
      :try_start_7
      new-instance v12, Ljava/lang/StringBuilder;
***************
*** 722,733 ****
--- 775,789 ----
  
      move-result-object v4
  
+     .line 156
      goto :goto_2
  
+     .line 147
      .end local v3           #e:Ljava/lang/ClassNotFoundException;
      :catch_3
      move-exception v3
  
+     .line 148
      .local v3, e:Ljava/lang/SecurityException;
      new-instance v12, Ljava/lang/StringBuilder;
  
***************
*** 747,758 ****
--- 803,817 ----
  
      move-result-object v4
  
+     .line 156
      goto :goto_2
  
+     .line 149
      .end local v3           #e:Ljava/lang/SecurityException;
      :catch_4
      move-exception v3
  
+     .line 150
      .local v3, e:Ljava/lang/NoSuchFieldException;
      new-instance v12, Ljava/lang/StringBuilder;
  
***************
*** 800,811 ****
--- 859,873 ----
  
      move-result-object v4
  
+     .line 156
      goto/16 :goto_2
  
+     .line 152
      .end local v3           #e:Ljava/lang/NoSuchFieldException;
      :catch_5
      move-exception v3
  
+     .line 153
      .local v3, e:Ljava/lang/IllegalArgumentException;
      new-instance v12, Ljava/lang/StringBuilder;
  
***************
*** 825,836 ****
--- 887,901 ----
  
      move-result-object v4
  
+     .line 156
      goto/16 :goto_2
  
+     .line 154
      .end local v3           #e:Ljava/lang/IllegalArgumentException;
      :catch_6
      move-exception v3
  
+     .line 155
      .local v3, e:Ljava/lang/IllegalAccessException;
      new-instance v12, Ljava/lang/StringBuilder;
  
***************
*** 852,857 ****
--- 917,923 ----
  
      goto/16 :goto_2
  
+     .line 163
      .end local v2           #dot:I
      .end local v3           #e:Ljava/lang/IllegalAccessException;
      .end local v4           #errorMsg:Ljava/lang/String;
***************
*** 865,870 ****
--- 931,937 ----
  
      if-nez v12, :cond_8
  
+     .line 164
      move-object/from16 v0, p0
  
      iget-object v12, v0, Landroid/webkit/FileLoader;->mLoadListener:Landroid/webkit/LoadListener;
***************
*** 883,892 ****
--- 950,961 ----
  
      invoke-virtual {v12, v13, v14}, Landroid/webkit/LoadListener;->error(ILjava/lang/String;)V
  
+     .line 166
      const/4 v12, 0x0
  
      goto/16 :goto_1
  
+     .line 169
      :cond_8
      new-instance v12, Ljava/io/FileInputStream;
  
***************
*** 900,905 ****
--- 969,975 ----
  
      iput-object v12, v0, Landroid/webkit/FileLoader;->mDataStream:Ljava/io/InputStream;
  
+     .line 170
      new-instance v12, Ljava/io/File;
  
      move-object/from16 v0, p0
***************
*** 921,929 ****
--- 991,1001 ----
  
      goto/16 :goto_0
  
+     .line 178
      :catch_7
      move-exception v5
  
+     .line 179
      .local v5, ex:Ljava/io/IOException;
      move-object/from16 v0, p0
  
***************
*** 939,944 ****
--- 1011,1017 ----
  
      invoke-virtual {v12, v13, v14}, Landroid/webkit/LoadListener;->error(ILjava/lang/String;)V
  
+     .line 180
      const/4 v12, 0x0
  
      goto/16 :goto_1
