*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/HTML5VideoFullScreen$2.smali	2012-03-23 19:37:10.824972475 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/HTML5VideoFullScreen$2.smali.original	2012-03-23 19:34:45.576323871 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 211
      iput-object p1, p0, Landroid/webkit/HTML5VideoFullScreen$2;->this$0:Landroid/webkit/HTML5VideoFullScreen;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,53 ****
--- 43,56 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 217
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen$2;->this$0:Landroid/webkit/HTML5VideoFullScreen;
  
      iget-object v0, v0, Landroid/webkit/HTML5VideoFullScreen;->mProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      invoke-virtual {v0}, Landroid/webkit/HTML5VideoViewProxy;->dispatchOnStopFullScreen()V
  
+     .line 218
      invoke-static {}, Landroid/webkit/HTML5VideoFullScreen;->access$700()Landroid/widget/FrameLayout;
  
      move-result-object v0
***************
*** 61,72 ****
--- 64,77 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/FrameLayout;->removeView(Landroid/view/View;)V
  
+     .line 220
      invoke-static {}, Landroid/webkit/HTML5VideoFullScreen;->access$800()Landroid/view/View;
  
      move-result-object v0
  
      if-eqz v0, :cond_0
  
+     .line 221
      invoke-static {}, Landroid/webkit/HTML5VideoFullScreen;->access$700()Landroid/widget/FrameLayout;
  
      move-result-object v0
***************
*** 77,87 ****
--- 82,95 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/FrameLayout;->removeView(Landroid/view/View;)V
  
+     .line 222
      invoke-static {v2}, Landroid/webkit/HTML5VideoFullScreen;->access$802(Landroid/view/View;)Landroid/view/View;
  
+     .line 224
      :cond_0
      invoke-static {v2}, Landroid/webkit/HTML5VideoFullScreen;->access$702(Landroid/widget/FrameLayout;)Landroid/widget/FrameLayout;
  
+     .line 226
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen$2;->this$0:Landroid/webkit/HTML5VideoFullScreen;
  
      iget-object v0, v0, Landroid/webkit/HTML5VideoFullScreen;->mProxy:Landroid/webkit/HTML5VideoViewProxy;
***************
*** 96,115 ****
--- 104,127 ----
  
      invoke-virtual {v0}, Landroid/webkit/ViewManager;->showAll()V
  
+     .line 228
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen$2;->this$0:Landroid/webkit/HTML5VideoFullScreen;
  
      iput-object v2, v0, Landroid/webkit/HTML5VideoFullScreen;->mProxy:Landroid/webkit/HTML5VideoViewProxy;
  
+     .line 231
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen$2;->this$0:Landroid/webkit/HTML5VideoFullScreen;
  
      #setter for: Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
      invoke-static {v0, v2}, Landroid/webkit/HTML5VideoFullScreen;->access$202(Landroid/webkit/HTML5VideoFullScreen;Landroid/widget/MediaController;)Landroid/widget/MediaController;
  
+     .line 232
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen$2;->this$0:Landroid/webkit/HTML5VideoFullScreen;
  
      const/4 v1, 0x4
  
      iput v1, v0, Landroid/webkit/HTML5VideoFullScreen;->mCurrentState:I
  
+     .line 233
      return-void
  .end method
