*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/HTML5VideoFullScreen.smali	2012-03-23 19:37:05.253870886 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/HTML5VideoFullScreen.smali.original	2012-03-23 19:34:45.539323195 -0500
***************
*** 68,111 ****
--- 68,122 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 120
      invoke-direct {p0}, Landroid/webkit/HTML5VideoView;-><init>()V
  
+     .line 61
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mSurfaceHolder:Landroid/view/SurfaceHolder;
  
+     .line 79
      new-instance v0, Landroid/webkit/HTML5VideoFullScreen$1;
  
      invoke-direct {v0, p0}, Landroid/webkit/HTML5VideoFullScreen$1;-><init>(Landroid/webkit/HTML5VideoFullScreen;)V
  
      iput-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mSHCallback:Landroid/view/SurfaceHolder$Callback;
  
+     .line 210
      new-instance v0, Landroid/webkit/HTML5VideoFullScreen$2;
  
      invoke-direct {v0, p0}, Landroid/webkit/HTML5VideoFullScreen$2;-><init>(Landroid/webkit/HTML5VideoFullScreen;)V
  
      iput-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mCallback:Landroid/webkit/WebChromeClient$CustomViewCallback;
  
+     .line 310
      new-instance v0, Landroid/webkit/HTML5VideoFullScreen$3;
  
      invoke-direct {v0, p0}, Landroid/webkit/HTML5VideoFullScreen$3;-><init>(Landroid/webkit/HTML5VideoFullScreen;)V
  
      iput-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mBufferingUpdateListener:Landroid/media/MediaPlayer$OnBufferingUpdateListener;
  
+     .line 121
      new-instance v0, Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      invoke-direct {v0, p0, p1}, Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;-><init>(Landroid/webkit/HTML5VideoFullScreen;Landroid/content/Context;)V
  
      iput-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
+     .line 122
      iput v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mFullScreenMode:I
  
+     .line 123
      iput v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoWidth:I
  
+     .line 124
      iput v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoHeight:I
  
+     .line 125
      invoke-virtual {p0, p2, p3, p4}, Landroid/webkit/HTML5VideoFullScreen;->init(IIZ)V
  
+     .line 126
      return-void
  .end method
  
***************
*** 114,119 ****
--- 125,131 ----
      .parameter "x0"
  
      .prologue
+     .line 21
      iget v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoWidth:I
  
      return v0
***************
*** 124,129 ****
--- 136,142 ----
      .parameter "x0"
  
      .prologue
+     .line 21
      iget v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoHeight:I
  
      return v0
***************
*** 134,139 ****
--- 147,153 ----
      .parameter "x0"
  
      .prologue
+     .line 21
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      return-object v0
***************
*** 145,150 ****
--- 159,165 ----
      .parameter "x1"
  
      .prologue
+     .line 21
      iput-object p1, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      return-object p1
***************
*** 156,161 ****
--- 171,177 ----
      .parameter "x1"
  
      .prologue
+     .line 21
      iput-object p1, p0, Landroid/webkit/HTML5VideoFullScreen;->mSurfaceHolder:Landroid/view/SurfaceHolder;
  
      return-object p1
***************
*** 167,172 ****
--- 183,189 ----
      .parameter "x1"
  
      .prologue
+     .line 21
      iput p1, p0, Landroid/webkit/HTML5VideoFullScreen;->mFullScreenMode:I
  
      return p1
***************
*** 177,182 ****
--- 194,200 ----
      .parameter "x0"
  
      .prologue
+     .line 21
      invoke-direct {p0}, Landroid/webkit/HTML5VideoFullScreen;->prepareForFullScreen()V
  
      return-void
***************
*** 187,192 ****
--- 205,211 ----
      .parameter "x0"
  
      .prologue
+     .line 21
      invoke-direct {p0}, Landroid/webkit/HTML5VideoFullScreen;->getSurfaceView()Landroid/view/SurfaceView;
  
      move-result-object v0
***************
*** 198,203 ****
--- 217,223 ----
      .locals 1
  
      .prologue
+     .line 21
      sget-object v0, Landroid/webkit/HTML5VideoFullScreen;->mLayout:Landroid/widget/FrameLayout;
  
      return-object v0
***************
*** 208,213 ****
--- 228,234 ----
      .parameter "x0"
  
      .prologue
+     .line 21
      sput-object p0, Landroid/webkit/HTML5VideoFullScreen;->mLayout:Landroid/widget/FrameLayout;
  
      return-object p0
***************
*** 217,222 ****
--- 238,244 ----
      .locals 1
  
      .prologue
+     .line 21
      sget-object v0, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
      return-object v0
***************
*** 227,232 ****
--- 249,255 ----
      .parameter "x0"
  
      .prologue
+     .line 21
      sput-object p0, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
      return-object p0
***************
*** 238,243 ****
--- 261,267 ----
      .parameter "x1"
  
      .prologue
+     .line 21
      iput p1, p0, Landroid/webkit/HTML5VideoFullScreen;->mCurrentBufferPercentage:I
  
      return p1
***************
*** 247,252 ****
--- 271,277 ----
      .locals 2
  
      .prologue
+     .line 134
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mPlayer:Landroid/media/MediaPlayer;
  
      if-eqz v0, :cond_0
***************
*** 255,276 ****
--- 280,305 ----
  
      if-eqz v0, :cond_0
  
+     .line 135
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      invoke-virtual {v0, p0}, Landroid/widget/MediaController;->setMediaPlayer(Landroid/widget/MediaController$MediaPlayerControl;)V
  
+     .line 136
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      invoke-virtual {v0, v1}, Landroid/widget/MediaController;->setAnchorView(Landroid/view/View;)V
  
+     .line 138
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, v1}, Landroid/widget/MediaController;->setEnabled(Z)V
  
+     .line 140
      :cond_0
      return-void
  .end method
***************
*** 279,284 ****
--- 308,314 ----
      .locals 1
  
      .prologue
+     .line 116
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      return-object v0
***************
*** 288,297 ****
--- 318,329 ----
      .locals 3
  
      .prologue
+     .line 149
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mPlayer:Landroid/media/MediaPlayer;
  
      invoke-virtual {v1}, Landroid/media/MediaPlayer;->reset()V
  
+     .line 150
      new-instance v0, Landroid/webkit/HTML5VideoFullScreen$FullScreenMediaController;
  
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mProxy:Landroid/webkit/HTML5VideoViewProxy;
***************
*** 304,309 ****
--- 336,342 ----
  
      invoke-direct {v0, v1, v2}, Landroid/webkit/HTML5VideoFullScreen$FullScreenMediaController;-><init>(Landroid/content/Context;Landroid/view/View;)V
  
+     .line 151
      .local v0, mc:Landroid/widget/MediaController;
      sget-object v1, Landroid/webkit/HTML5VideoFullScreen;->mLayout:Landroid/widget/FrameLayout;
  
***************
*** 313,330 ****
--- 346,367 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/MediaController;->setSystemUiVisibility(I)V
  
+     .line 152
      invoke-direct {p0, v0}, Landroid/webkit/HTML5VideoFullScreen;->setMediaController(Landroid/widget/MediaController;)V
  
+     .line 153
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mPlayer:Landroid/media/MediaPlayer;
  
      const/4 v2, 0x1
  
      invoke-virtual {v1, v2}, Landroid/media/MediaPlayer;->setScreenOnWhilePlaying(Z)V
  
+     .line 154
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      invoke-virtual {p0, v1}, Landroid/webkit/HTML5VideoFullScreen;->prepareDataAndDisplayMode(Landroid/webkit/HTML5VideoViewProxy;)V
  
+     .line 155
      return-void
  .end method
  
***************
*** 333,342 ****
--- 370,382 ----
      .parameter "m"
  
      .prologue
+     .line 129
      iput-object p1, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
+     .line 130
      invoke-direct {p0}, Landroid/webkit/HTML5VideoFullScreen;->attachMediaController()V
  
+     .line 131
      return-void
  .end method
  
***************
*** 344,349 ****
--- 384,390 ----
      .locals 1
  
      .prologue
+     .line 159
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      invoke-virtual {v0}, Landroid/widget/MediaController;->isShowing()Z
***************
*** 352,364 ****
--- 393,408 ----
  
      if-eqz v0, :cond_0
  
+     .line 160
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      invoke-virtual {v0}, Landroid/widget/MediaController;->hide()V
  
+     .line 164
      :goto_0
      return-void
  
+     .line 162
      :cond_0
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
***************
*** 373,378 ****
--- 417,423 ----
      .locals 1
  
      .prologue
+     .line 288
      iget-boolean v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mCanPause:Z
  
      return v0
***************
*** 382,387 ****
--- 427,433 ----
      .locals 1
  
      .prologue
+     .line 293
      iget-boolean v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mCanSeekBack:Z
  
      return v0
***************
*** 391,396 ****
--- 437,443 ----
      .locals 1
  
      .prologue
+     .line 298
      iget-boolean v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mCanSeekForward:Z
  
      return v0
***************
*** 400,411 ****
--- 447,460 ----
      .locals 2
  
      .prologue
+     .line 144
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mPlayer:Landroid/media/MediaPlayer;
  
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mSurfaceHolder:Landroid/view/SurfaceHolder;
  
      invoke-virtual {v0, v1}, Landroid/media/MediaPlayer;->setDisplay(Landroid/view/SurfaceHolder;)V
  
+     .line 145
      return-void
  .end method
  
***************
*** 422,439 ****
--- 471,493 ----
  
      const/4 v4, 0x0
  
+     .line 239
      iput v5, p0, Landroid/webkit/HTML5VideoFullScreen;->mFullScreenMode:I
  
+     .line 240
      iput v4, p0, Landroid/webkit/HTML5VideoFullScreen;->mCurrentBufferPercentage:I
  
+     .line 241
      iget-object v2, p0, Landroid/webkit/HTML5VideoFullScreen;->mPlayer:Landroid/media/MediaPlayer;
  
      iget-object v3, p0, Landroid/webkit/HTML5VideoFullScreen;->mBufferingUpdateListener:Landroid/media/MediaPlayer$OnBufferingUpdateListener;
  
      invoke-virtual {v2, v3}, Landroid/media/MediaPlayer;->setOnBufferingUpdateListener(Landroid/media/MediaPlayer$OnBufferingUpdateListener;)V
  
+     .line 242
      iput-object p2, p0, Landroid/webkit/HTML5VideoFullScreen;->mProxy:Landroid/webkit/HTML5VideoViewProxy;
  
+     .line 244
      iget-object v2, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;->getHolder()Landroid/view/SurfaceHolder;
***************
*** 444,449 ****
--- 498,504 ----
  
      invoke-interface {v2, v3}, Landroid/view/SurfaceHolder;->addCallback(Landroid/view/SurfaceHolder$Callback;)V
  
+     .line 245
      iget-object v2, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;->getHolder()Landroid/view/SurfaceHolder;
***************
*** 454,471 ****
--- 509,530 ----
  
      invoke-interface {v2, v3}, Landroid/view/SurfaceHolder;->setType(I)V
  
+     .line 246
      iget-object v2, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      invoke-virtual {v2, v5}, Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;->setFocusable(Z)V
  
+     .line 247
      iget-object v2, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      invoke-virtual {v2, v5}, Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;->setFocusableInTouchMode(Z)V
  
+     .line 248
      iget-object v2, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;->requestFocus()Z
  
+     .line 252
      new-instance v2, Landroid/widget/FrameLayout;
  
      iget-object v3, p0, Landroid/webkit/HTML5VideoFullScreen;->mProxy:Landroid/webkit/HTML5VideoViewProxy;
***************
*** 478,489 ****
--- 537,550 ----
  
      sput-object v2, Landroid/webkit/HTML5VideoFullScreen;->mLayout:Landroid/widget/FrameLayout;
  
+     .line 253
      new-instance v1, Landroid/widget/FrameLayout$LayoutParams;
  
      const/16 v2, 0x11
  
      invoke-direct {v1, v6, v6, v2}, Landroid/widget/FrameLayout$LayoutParams;-><init>(III)V
  
+     .line 258
      .local v1, layoutParams:Landroid/widget/FrameLayout$LayoutParams;
      sget-object v2, Landroid/webkit/HTML5VideoFullScreen;->mLayout:Landroid/widget/FrameLayout;
  
***************
*** 493,527 ****
--- 554,595 ----
  
      invoke-virtual {v2, v3, v1}, Landroid/widget/FrameLayout;->addView(Landroid/view/View;Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 260
      sget-object v2, Landroid/webkit/HTML5VideoFullScreen;->mLayout:Landroid/widget/FrameLayout;
  
      invoke-virtual {v2, v4}, Landroid/widget/FrameLayout;->setVisibility(I)V
  
+     .line 261
      invoke-virtual {p3}, Landroid/webkit/WebView;->getWebChromeClient()Landroid/webkit/WebChromeClient;
  
      move-result-object v0
  
+     .line 262
      .local v0, client:Landroid/webkit/WebChromeClient;
      if-eqz v0, :cond_1
  
+     .line 263
      sget-object v2, Landroid/webkit/HTML5VideoFullScreen;->mLayout:Landroid/widget/FrameLayout;
  
      iget-object v3, p0, Landroid/webkit/HTML5VideoFullScreen;->mCallback:Landroid/webkit/WebChromeClient$CustomViewCallback;
  
      invoke-virtual {v0, v2, v3}, Landroid/webkit/WebChromeClient;->onShowCustomView(Landroid/view/View;Landroid/webkit/WebChromeClient$CustomViewCallback;)V
  
+     .line 266
      invoke-virtual {p3}, Landroid/webkit/WebView;->getViewManager()Landroid/webkit/ViewManager;
  
      move-result-object v2
  
      if-eqz v2, :cond_0
  
+     .line 267
      invoke-virtual {p3}, Landroid/webkit/WebView;->getViewManager()Landroid/webkit/ViewManager;
  
      move-result-object v2
  
      invoke-virtual {v2}, Landroid/webkit/ViewManager;->hideAll()V
  
+     .line 269
      :cond_0
      invoke-virtual {v0}, Landroid/webkit/WebChromeClient;->getVideoLoadingProgressView()Landroid/view/View;
  
***************
*** 529,548 ****
--- 597,620 ----
  
      sput-object v2, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
+     .line 270
      sget-object v2, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
      if-eqz v2, :cond_1
  
+     .line 271
      sget-object v2, Landroid/webkit/HTML5VideoFullScreen;->mLayout:Landroid/widget/FrameLayout;
  
      sget-object v3, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
      invoke-virtual {v2, v3, v1}, Landroid/widget/FrameLayout;->addView(Landroid/view/View;Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 272
      sget-object v2, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
      invoke-virtual {v2, v4}, Landroid/view/View;->setVisibility(I)V
  
+     .line 275
      :cond_1
      return-void
  .end method
***************
*** 551,556 ****
--- 623,629 ----
      .locals 1
  
      .prologue
+     .line 207
      sget-object v0, Landroid/webkit/HTML5VideoFullScreen;->mLayout:Landroid/widget/FrameLayout;
  
      if-nez v0, :cond_0
***************
*** 570,581 ****
--- 643,657 ----
      .locals 1
  
      .prologue
+     .line 303
      iget-object v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mPlayer:Landroid/media/MediaPlayer;
  
      if-eqz v0, :cond_0
  
+     .line 304
      iget v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mCurrentBufferPercentage:I
  
+     .line 306
      :goto_0
      return v0
  
***************
*** 589,594 ****
--- 665,671 ----
      .locals 1
  
      .prologue
+     .line 282
      const/4 v0, 0x1
  
      return v0
***************
*** 607,625 ****
--- 684,707 ----
  
      const/4 v3, 0x1
  
+     .line 168
      invoke-super {p0, p1}, Landroid/webkit/HTML5VideoView;->onPrepared(Landroid/media/MediaPlayer;)V
  
+     .line 170
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      invoke-virtual {v1, p0}, Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;->setOnTouchListener(Landroid/view/View$OnTouchListener;)V
  
+     .line 172
      invoke-virtual {p1, v2, v2}, Landroid/media/MediaPlayer;->getMetadata(ZZ)Landroid/media/Metadata;
  
      move-result-object v0
  
+     .line 174
      .local v0, data:Landroid/media/Metadata;
      if-eqz v0, :cond_8
  
+     .line 175
      invoke-virtual {v0, v3}, Landroid/media/Metadata;->has(I)Z
  
      move-result v1
***************
*** 638,643 ****
--- 720,726 ----
      :goto_0
      iput-boolean v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mCanPause:Z
  
+     .line 177
      invoke-virtual {v0, v4}, Landroid/media/Metadata;->has(I)Z
  
      move-result v1
***************
*** 656,661 ****
--- 739,745 ----
      :goto_1
      iput-boolean v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mCanSeekBack:Z
  
+     .line 179
      invoke-virtual {v0, v5}, Landroid/media/Metadata;->has(I)Z
  
      move-result v1
***************
*** 674,710 ****
--- 758,801 ----
      :goto_2
      iput-boolean v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mCanSeekForward:Z
  
+     .line 187
      :goto_3
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      if-eqz v1, :cond_3
  
+     .line 188
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      invoke-virtual {v1, v3}, Landroid/widget/MediaController;->setEnabled(Z)V
  
+     .line 190
      invoke-virtual {p0}, Landroid/webkit/HTML5VideoFullScreen;->getAutostart()Z
  
      move-result v1
  
      if-eqz v1, :cond_9
  
+     .line 191
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
      invoke-virtual {v1}, Landroid/widget/MediaController;->show()V
  
+     .line 196
      :cond_3
      :goto_4
      sget-object v1, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
      if-eqz v1, :cond_4
  
+     .line 197
      sget-object v1, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
      const/16 v2, 0x8
  
      invoke-virtual {v1, v2}, Landroid/view/View;->setVisibility(I)V
  
+     .line 200
      :cond_4
      invoke-virtual {p1}, Landroid/media/MediaPlayer;->getVideoWidth()I
  
***************
*** 712,723 ****
--- 803,816 ----
  
      iput v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoWidth:I
  
+     .line 201
      invoke-virtual {p1}, Landroid/media/MediaPlayer;->getVideoHeight()I
  
      move-result v1
  
      iput v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoHeight:I
  
+     .line 203
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mVideoSurfaceView:Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;
  
      invoke-virtual {v1}, Landroid/webkit/HTML5VideoFullScreen$VideoSurfaceView;->getHolder()Landroid/view/SurfaceHolder;
***************
*** 730,752 ****
--- 823,850 ----
  
      invoke-interface {v1, v2, v3}, Landroid/view/SurfaceHolder;->setFixedSize(II)V
  
+     .line 204
      return-void
  
      :cond_5
      move v1, v2
  
+     .line 175
      goto :goto_0
  
      :cond_6
      move v1, v2
  
+     .line 177
      goto :goto_1
  
      :cond_7
      move v1, v2
  
+     .line 179
      goto :goto_2
  
+     .line 182
      :cond_8
      iput-boolean v3, p0, Landroid/webkit/HTML5VideoFullScreen;->mCanSeekForward:Z
  
***************
*** 756,761 ****
--- 854,860 ----
  
      goto :goto_3
  
+     .line 193
      :cond_9
      iget-object v1, p0, Landroid/webkit/HTML5VideoFullScreen;->mMediaController:Landroid/widget/MediaController;
  
***************
*** 770,775 ****
--- 869,875 ----
      .parameter "event"
  
      .prologue
+     .line 319
      iget v0, p0, Landroid/webkit/HTML5VideoFullScreen;->mFullScreenMode:I
  
      const/4 v1, 0x2
***************
*** 780,787 ****
--- 880,889 ----
  
      if-eqz v0, :cond_0
  
+     .line 321
      invoke-direct {p0}, Landroid/webkit/HTML5VideoFullScreen;->toggleMediaControlsVisiblity()V
  
+     .line 323
      :cond_0
      const/4 v0, 0x0
  
***************
*** 793,814 ****
--- 895,921 ----
      .parameter "playerBuffering"
  
      .prologue
+     .line 328
      sget-object v0, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
      if-eqz v0, :cond_0
  
+     .line 329
      if-eqz p1, :cond_1
  
+     .line 330
      sget-object v0, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, v1}, Landroid/view/View;->setVisibility(I)V
  
+     .line 335
      :cond_0
      :goto_0
      return-void
  
+     .line 332
      :cond_1
      sget-object v0, Landroid/webkit/HTML5VideoFullScreen;->mProgressView:Landroid/view/View;
  
