*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/HTML5VideoViewProxy$VideoPlayer.smali	2012-03-23 19:37:07.382909709 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/HTML5VideoViewProxy$VideoPlayer.smali.original	2012-03-23 19:34:45.556323505 -0500
***************
*** 31,38 ****
--- 31,40 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 94
      sput-boolean v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->isVideoSelfEnded:Z
  
+     .line 97
      sput v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mBaseLayer:I
  
      return-void
***************
*** 42,47 ****
--- 44,50 ----
      .locals 0
  
      .prologue
+     .line 87
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 52,57 ****
--- 55,61 ----
      .parameter "x0"
  
      .prologue
+     .line 87
      sput-boolean p0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->isVideoSelfEnded:Z
  
      return p0
***************
*** 62,67 ****
--- 66,72 ----
      .parameter "x0"
  
      .prologue
+     .line 87
      invoke-static {p0}, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->setPlayerBuffering(Z)V
  
      return-void
***************
*** 71,96 ****
--- 76,107 ----
      .locals 1
  
      .prologue
+     .line 247
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      if-eqz v0, :cond_0
  
+     .line 248
      sget-boolean v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->isVideoSelfEnded:Z
  
      if-eqz v0, :cond_1
  
+     .line 249
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      invoke-virtual {v0}, Landroid/webkit/HTML5VideoViewProxy;->dispatchOnEnded()V
  
+     .line 253
      :cond_0
      :goto_0
      const/4 v0, 0x0
  
      sput-boolean v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->isVideoSelfEnded:Z
  
+     .line 254
      return-void
  
+     .line 251
      :cond_1
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
***************
*** 107,122 ****
--- 118,137 ----
      .parameter "webView"
  
      .prologue
+     .line 146
      const/4 v0, 0x0
  
+     .line 147
      .local v0, savePosition:I
      const/4 v1, 0x0
  
+     .line 148
      .local v1, savedIsPlaying:Z
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      if-eqz v2, :cond_1
  
+     .line 151
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoView;->getVideoLayerId()I
***************
*** 125,142 ****
--- 140,160 ----
  
      if-ne p0, v2, :cond_0
  
+     .line 152
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoView;->getCurrentPosition()I
  
      move-result v0
  
+     .line 153
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoView;->isPlaying()Z
  
      move-result v1
  
+     .line 155
      :cond_0
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
***************
*** 144,153 ****
--- 162,173 ----
  
      invoke-virtual {v2, v3}, Landroid/webkit/HTML5VideoView;->pauseAndDispatch(Landroid/webkit/HTML5VideoViewProxy;)V
  
+     .line 156
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoView;->release()V
  
+     .line 158
      :cond_1
      new-instance v2, Landroid/webkit/HTML5VideoFullScreen;
  
***************
*** 159,176 ****
--- 179,200 ----
  
      sput-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
+     .line 160
      sput-object p2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
+     .line 162
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      sget-object v3, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      invoke-virtual {v2, p1, v3}, Landroid/webkit/HTML5VideoView;->setVideoURI(Ljava/lang/String;Landroid/webkit/HTML5VideoViewProxy;)V
  
+     .line 164
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2, p0, p2, p3}, Landroid/webkit/HTML5VideoView;->enterFullScreenVideoState(ILandroid/webkit/HTML5VideoViewProxy;Landroid/webkit/WebView;)V
  
+     .line 165
      return-void
  .end method
  
***************
*** 178,196 ****
--- 202,224 ----
      .locals 2
  
      .prologue
+     .line 218
      const/4 v0, 0x0
  
+     .line 219
      .local v0, currentPosMs:I
      sget-object v1, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      if-eqz v1, :cond_0
  
+     .line 220
      sget-object v1, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v1}, Landroid/webkit/HTML5VideoView;->getCurrentPosition()I
  
      move-result v0
  
+     .line 222
      :cond_0
      return v0
  .end method
***************
*** 200,205 ****
--- 228,234 ----
      .parameter "proxy"
  
      .prologue
+     .line 213
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      if-ne v0, p0, :cond_0
***************
*** 231,236 ****
--- 260,266 ----
      .locals 1
  
      .prologue
+     .line 238
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v0}, Landroid/webkit/HTML5VideoView;->isFullScreenMode()Z
***************
*** 247,266 ****
--- 277,300 ----
  
      if-eqz v0, :cond_1
  
+     .line 239
      :cond_0
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v0}, Landroid/webkit/HTML5VideoView;->start()V
  
+     .line 241
      :cond_1
      sget v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mBaseLayer:I
  
      if-eqz v0, :cond_2
  
+     .line 242
      sget v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mBaseLayer:I
  
      invoke-static {v0}, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->setBaseLayer(I)V
  
+     .line 244
      :cond_2
      return-void
  .end method
***************
*** 270,275 ****
--- 304,310 ----
      .parameter "proxy"
  
      .prologue
+     .line 232
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      if-ne v0, p0, :cond_0
***************
*** 278,287 ****
--- 313,324 ----
  
      if-eqz v0, :cond_0
  
+     .line 233
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v0}, Landroid/webkit/HTML5VideoView;->pause()V
  
+     .line 235
      :cond_0
      return-void
  .end method
***************
*** 290,309 ****
--- 327,350 ----
      .locals 2
  
      .prologue
+     .line 135
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      if-eqz v0, :cond_0
  
+     .line 136
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      sget-object v1, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      invoke-virtual {v0, v1}, Landroid/webkit/HTML5VideoView;->pauseAndDispatch(Landroid/webkit/HTML5VideoViewProxy;)V
  
+     .line 139
      sget v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mBaseLayer:I
  
      invoke-static {v0}, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->setBaseLayer(I)V
  
+     .line 141
      :cond_0
      return-void
  .end method
***************
*** 317,344 ****
--- 358,391 ----
      .parameter "videoLayerId"
  
      .prologue
+     .line 172
      const/4 v1, -0x1
  
+     .line 173
      .local v1, currentVideoLayerId:I
      const/4 v0, 0x0
  
+     .line 174
      .local v0, backFromFullScreenMode:Z
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      if-eqz v2, :cond_0
  
+     .line 175
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoView;->getVideoLayerId()I
  
      move-result v1
  
+     .line 176
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoView;->fullScreenExited()Z
  
      move-result v0
  
+     .line 179
      :cond_0
      if-nez v0, :cond_1
  
***************
*** 352,378 ****
--- 399,431 ----
  
      if-eqz v2, :cond_5
  
+     .line 187
      :cond_1
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      if-eqz v2, :cond_3
  
+     .line 188
      if-nez v0, :cond_2
  
+     .line 189
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      sget-object v3, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      invoke-virtual {v2, v3}, Landroid/webkit/HTML5VideoView;->pauseAndDispatch(Landroid/webkit/HTML5VideoViewProxy;)V
  
+     .line 192
      :cond_2
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoView;->release()V
  
+     .line 194
      :cond_3
      sput-object p2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
+     .line 195
      new-instance v2, Landroid/webkit/HTML5VideoInline;
  
      const/4 v3, 0x0
***************
*** 381,405 ****
--- 434,463 ----
  
      sput-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
+     .line 197
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      sget-object v3, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      invoke-virtual {v2, p0, v3}, Landroid/webkit/HTML5VideoView;->setVideoURI(Ljava/lang/String;Landroid/webkit/HTML5VideoViewProxy;)V
  
+     .line 198
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2, p2}, Landroid/webkit/HTML5VideoView;->prepareDataAndDisplayMode(Landroid/webkit/HTML5VideoViewProxy;)V
  
+     .line 210
      :cond_4
      :goto_0
      return-void
  
+     .line 199
      :cond_5
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      if-ne v2, p2, :cond_6
  
+     .line 201
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoView;->isPlaying()Z
***************
*** 408,428 ****
--- 466,490 ----
  
      if-nez v2, :cond_4
  
+     .line 202
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2, p1}, Landroid/webkit/HTML5VideoView;->seekTo(I)V
  
+     .line 203
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v2}, Landroid/webkit/HTML5VideoView;->start()V
  
      goto :goto_0
  
+     .line 205
      :cond_6
      sget-object v2, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      if-eqz v2, :cond_4
  
+     .line 208
      invoke-virtual {p2}, Landroid/webkit/HTML5VideoViewProxy;->dispatchOnEnded()V
  
      goto :goto_0
***************
*** 434,439 ****
--- 496,502 ----
      .parameter "proxy"
  
      .prologue
+     .line 226
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      if-ne v0, p1, :cond_0
***************
*** 444,453 ****
--- 507,518 ----
  
      if-eqz v0, :cond_0
  
+     .line 227
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v0, p0}, Landroid/webkit/HTML5VideoView;->seekTo(I)V
  
+     .line 229
      :cond_0
      return-void
  .end method
***************
*** 457,462 ****
--- 522,528 ----
      .parameter "layer"
  
      .prologue
+     .line 108
      sget-object v5, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      if-eqz v5, :cond_1
***************
*** 477,490 ****
--- 543,559 ----
  
      if-nez v5, :cond_1
  
+     .line 111
      sput p0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mBaseLayer:I
  
+     .line 113
      sget-object v5, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v5}, Landroid/webkit/HTML5VideoView;->getVideoLayerId()I
  
      move-result v0
  
+     .line 114
      .local v0, currentVideoLayerId:I
      sget-object v5, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
***************
*** 492,497 ****
--- 561,567 ----
  
      move-result-object v3
  
+     .line 115
      .local v3, surfTexture:Landroid/graphics/SurfaceTexture;
      sget-object v5, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
***************
*** 499,504 ****
--- 569,575 ----
  
      move-result v4
  
+     .line 117
      .local v4, textureName:I
      if-eqz p0, :cond_1
  
***************
*** 508,519 ****
--- 579,592 ----
  
      if-eq v0, v5, :cond_1
  
+     .line 118
      sget-object v5, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v5}, Landroid/webkit/HTML5VideoView;->getCurrentState()I
  
      move-result v2
  
+     .line 119
      .local v2, playerState:I
      sget-object v5, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
***************
*** 523,536 ****
--- 596,612 ----
  
      if-eqz v5, :cond_0
  
+     .line 120
      const/4 v2, 0x1
  
+     .line 121
      :cond_0
      #calls: Landroid/webkit/HTML5VideoViewProxy;->nativeSendSurfaceTexture(Landroid/graphics/SurfaceTexture;IIII)Z
      invoke-static {v3, p0, v0, v4, v2}, Landroid/webkit/HTML5VideoViewProxy;->access$000(Landroid/graphics/SurfaceTexture;IIII)Z
  
      move-result v1
  
+     .line 124
      .local v1, foundInTree:Z
      const/4 v5, 0x2
  
***************
*** 538,553 ****
--- 614,632 ----
  
      if-nez v1, :cond_1
  
+     .line 126
      sget-object v5, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      sget-object v6, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mCurrentProxy:Landroid/webkit/HTML5VideoViewProxy;
  
      invoke-virtual {v5, v6}, Landroid/webkit/HTML5VideoView;->pauseAndDispatch(Landroid/webkit/HTML5VideoViewProxy;)V
  
+     .line 127
      sget-object v5, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v5}, Landroid/webkit/HTML5VideoView;->deleteSurfaceTexture()V
  
+     .line 131
      .end local v0           #currentVideoLayerId:I
      .end local v1           #foundInTree:Z
      .end local v2           #playerState:I
***************
*** 562,570 ****
--- 641,651 ----
      .parameter "playerBuffering"
  
      .prologue
+     .line 100
      sget-object v0, Landroid/webkit/HTML5VideoViewProxy$VideoPlayer;->mHTML5VideoView:Landroid/webkit/HTML5VideoView;
  
      invoke-virtual {v0, p0}, Landroid/webkit/HTML5VideoView;->setPlayerBuffering(Z)V
  
+     .line 101
      return-void
  .end method
