*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/HttpAuthHandler.smali	2012-03-23 19:37:10.354963906 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/HttpAuthHandler.smali.original	2012-03-23 19:34:45.574323833 -0500
***************
*** 8,15 ****
--- 8,17 ----
      .locals 0
  
      .prologue
+     .line 31
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
  
+     .line 32
      return-void
  .end method
  
***************
*** 19,24 ****
--- 21,27 ----
      .locals 0
  
      .prologue
+     .line 46
      return-void
  .end method
  
***************
*** 28,33 ****
--- 31,37 ----
      .parameter "password"
  
      .prologue
+     .line 52
      return-void
  .end method
  
***************
*** 35,40 ****
--- 39,45 ----
      .locals 1
  
      .prologue
+     .line 59
      const/4 v0, 0x0
  
      return v0
***************
*** 44,49 ****
--- 49,55 ----
      .locals 1
  
      .prologue
+     .line 39
      const/4 v0, 0x0
  
      return v0
