*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/JWebCoreJavaBridge.smali	2012-03-23 19:37:09.562949463 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/JWebCoreJavaBridge.smali.original	2012-03-23 19:34:45.570323761 -0500
***************
*** 51,56 ****
--- 51,57 ----
      .locals 2
  
      .prologue
+     .line 48
      new-instance v0, Ljava/lang/ref/WeakReference;
  
      const/4 v1, 0x0
***************
*** 66,75 ****
--- 67,79 ----
      .locals 0
  
      .prologue
+     .line 60
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
  
+     .line 61
      invoke-direct {p0}, Landroid/webkit/JWebCoreJavaBridge;->nativeConstructor()V
  
+     .line 63
      return-void
  .end method
  
***************
*** 78,83 ****
--- 82,88 ----
      .parameter "url"
  
      .prologue
+     .line 195
      invoke-static {}, Landroid/webkit/CookieManager;->getInstance()Landroid/webkit/CookieManager;
  
      move-result-object v0
***************
*** 93,98 ****
--- 98,104 ----
      .locals 1
  
      .prologue
+     .line 202
      invoke-static {}, Landroid/webkit/CookieManager;->getInstance()Landroid/webkit/CookieManager;
  
      move-result-object v0
***************
*** 108,119 ****
--- 114,128 ----
      .locals 1
  
      .prologue
+     .line 91
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mHasInstantTimer:Z
  
+     .line 92
      invoke-direct {p0}, Landroid/webkit/JWebCoreJavaBridge;->sharedTimerFired()V
  
+     .line 93
      return-void
  .end method
  
***************
*** 121,126 ****
--- 130,136 ----
      .locals 1
  
      .prologue
+     .line 257
      invoke-static {}, Landroid/webkit/CertTool;->getKeyStrengthList()[Ljava/lang/String;
  
      move-result-object v0
***************
*** 132,137 ****
--- 142,148 ----
      .locals 1
  
      .prologue
+     .line 209
      const/4 v0, 0x0
  
      invoke-static {v0}, Landroid/webkit/PluginManager;->getInstance(Landroid/content/Context;)Landroid/webkit/PluginManager;
***************
*** 149,154 ****
--- 160,166 ----
      .locals 1
  
      .prologue
+     .line 216
      const/4 v0, 0x0
  
      invoke-static {v0}, Landroid/webkit/PluginManager;->getInstance(Landroid/content/Context;)Landroid/webkit/PluginManager;
***************
*** 169,174 ****
--- 181,187 ----
      .parameter "url"
  
      .prologue
+     .line 262
      monitor-enter p0
  
      :try_start_0
***************
*** 180,188 ****
--- 193,203 ----
  
      check-cast v0, Landroid/webkit/WebView;
  
+     .line 263
      .local v0, current:Landroid/webkit/WebView;
      if-eqz v0, :cond_0
  
+     .line 266
      invoke-virtual {v0}, Landroid/webkit/WebView;->getContext()Landroid/content/Context;
  
      move-result-object v1
***************
*** 193,217 ****
  
      move-result-object v1
  
      :goto_0
      monitor-exit p0
  
      return-object v1
  
      :cond_0
      :try_start_1
!     const-string v1, "webkit-timers"
  
      const-string v2, "There is no active WebView for getSignedPublicKey"
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
      const-string v1, ""
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
      goto :goto_0
  
      .end local v0           #current:Landroid/webkit/WebView;
      :catchall_0
      move-exception v1
--- 208,236 ----
  
      move-result-object v1
  
+     .line 270
      :goto_0
      monitor-exit p0
  
      return-object v1
  
+     .line 269
      :cond_0
      :try_start_1
!     const-string/jumbo v1, "webkit-timers"
  
      const-string v2, "There is no active WebView for getSignedPublicKey"
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 270
      const-string v1, ""
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
      goto :goto_0
  
+     .line 262
      .end local v0           #current:Landroid/webkit/WebView;
      :catchall_0
      move-exception v1
***************
*** 238,243 ****
--- 257,263 ----
      .parameter "webview"
  
      .prologue
+     .line 79
      const-class v1, Landroid/webkit/JWebCoreJavaBridge;
  
      monitor-enter v1
***************
*** 253,263 ****
--- 273,285 ----
  
      if-eq v0, p0, :cond_0
  
+     .line 84
      :goto_0
      monitor-exit v1
  
      return-void
  
+     .line 83
      :cond_0
      :try_start_1
      sget-object v0, Landroid/webkit/JWebCoreJavaBridge;->sCurrentMainWebView:Ljava/lang/ref/WeakReference;
***************
*** 268,273 ****
--- 290,296 ----
  
      goto :goto_0
  
+     .line 79
      :catchall_0
      move-exception v0
  
***************
*** 281,290 ****
--- 304,315 ----
      .parameter "uri"
  
      .prologue
+     .line 276
      iget-object v2, p0, Landroid/webkit/JWebCoreJavaBridge;->mContentUriToFilePathMap:Ljava/util/HashMap;
  
      if-eqz v2, :cond_0
  
+     .line 277
      iget-object v2, p0, Landroid/webkit/JWebCoreJavaBridge;->mContentUriToFilePathMap:Ljava/util/HashMap;
  
      invoke-virtual {v2, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 293,310 ****
--- 318,339 ----
  
      check-cast v0, Ljava/lang/String;
  
+     .line 278
      .local v0, fileName:Ljava/lang/String;
      if-eqz v0, :cond_0
  
+     .line 286
      .end local v0           #fileName:Ljava/lang/String;
      :goto_0
      return-object v0
  
+     .line 285
      :cond_0
      invoke-static {p1}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v1
  
+     .line 286
      .local v1, jUri:Landroid/net/Uri;
      invoke-virtual {v1}, Landroid/net/Uri;->getLastPathSegment()Ljava/lang/String;
  
***************
*** 318,323 ****
--- 347,353 ----
      .parameter "webview"
  
      .prologue
+     .line 71
      const-class v1, Landroid/webkit/JWebCoreJavaBridge;
  
      monitor-enter v1
***************
*** 333,343 ****
--- 363,375 ----
  
      if-eqz v0, :cond_0
  
+     .line 76
      :goto_0
      monitor-exit v1
  
      return-void
  
+     .line 75
      :cond_0
      :try_start_1
      new-instance v0, Ljava/lang/ref/WeakReference;
***************
*** 350,355 ****
--- 382,388 ----
  
      goto :goto_0
  
+     .line 71
      :catchall_0
      move-exception v0
  
***************
*** 366,371 ****
--- 399,405 ----
      .prologue
      const/4 v7, -0x1
  
+     .line 166
      const-string v6, "\r"
  
      invoke-virtual {p2, v6}, Ljava/lang/String;->contains(Ljava/lang/CharSequence;)Z
***************
*** 382,412 ****
--- 416,452 ----
  
      if-eqz v6, :cond_7
  
+     .line 168
      :cond_0
      invoke-virtual {p2}, Ljava/lang/String;->length()I
  
      move-result v5
  
+     .line 169
      .local v5, size:I
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0, v5}, Ljava/lang/StringBuilder;-><init>(I)V
  
+     .line 170
      .local v0, buffer:Ljava/lang/StringBuilder;
      const/4 v1, 0x0
  
+     .line 171
      .local v1, i:I
      :goto_0
      if-eq v1, v7, :cond_6
  
      if-ge v1, v5, :cond_6
  
+     .line 172
      const/16 v6, 0xd
  
      invoke-virtual {p2, v6, v1}, Ljava/lang/String;->indexOf(II)I
  
      move-result v3
  
+     .line 173
      .local v3, ir:I
      const/16 v6, 0xa
  
***************
*** 414,439 ****
--- 454,485 ----
  
      move-result v2
  
+     .line 174
      .local v2, in:I
      if-ne v3, v7, :cond_2
  
      move v4, v2
  
+     .line 176
      .local v4, newi:I
      :goto_1
      if-le v4, v1, :cond_5
  
+     .line 177
      invoke-virtual {p2, v1, v4}, Ljava/lang/String;->subSequence(II)Ljava/lang/CharSequence;
  
      move-result-object v6
  
      invoke-virtual {v0, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/CharSequence;)Ljava/lang/StringBuilder;
  
+     .line 182
      :cond_1
      add-int/lit8 v1, v4, 0x1
  
+     .line 183
      goto :goto_0
  
+     .line 174
      .end local v4           #newi:I
      :cond_2
      if-ne v2, v7, :cond_3
***************
*** 454,469 ****
--- 500,518 ----
  
      goto :goto_1
  
+     .line 178
      .restart local v4       #newi:I
      :cond_5
      if-ne v4, v7, :cond_1
  
+     .line 179
      invoke-virtual {p2, v1, v5}, Ljava/lang/String;->subSequence(II)Ljava/lang/CharSequence;
  
      move-result-object v6
  
      invoke-virtual {v0, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/CharSequence;)Ljava/lang/StringBuilder;
  
+     .line 184
      .end local v2           #in:I
      .end local v3           #ir:I
      .end local v4           #newi:I
***************
*** 472,477 ****
--- 521,527 ----
  
      move-result-object p2
  
+     .line 186
      .end local v0           #buffer:Ljava/lang/StringBuilder;
      .end local v1           #i:I
      .end local v5           #size:I
***************
*** 482,487 ****
--- 532,538 ----
  
      invoke-virtual {v6, p1, p2}, Landroid/webkit/CookieManager;->setCookie(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 187
      return-void
  .end method
  
***************
*** 492,528 ****
--- 543,587 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 226
      const-wide/16 v1, 0x0
  
      cmp-long v1, p1, v1
  
      if-gtz v1, :cond_1
  
+     .line 231
      iget-boolean v1, p0, Landroid/webkit/JWebCoreJavaBridge;->mHasInstantTimer:Z
  
      if-eqz v1, :cond_0
  
+     .line 242
      :goto_0
      return-void
  
+     .line 234
      :cond_0
      iput-boolean v3, p0, Landroid/webkit/JWebCoreJavaBridge;->mHasInstantTimer:Z
  
+     .line 235
      invoke-virtual {p0, v3}, Landroid/webkit/JWebCoreJavaBridge;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 236
      .local v0, msg:Landroid/os/Message;
      invoke-virtual {p0, v0, p1, p2}, Landroid/webkit/JWebCoreJavaBridge;->sendMessageDelayed(Landroid/os/Message;J)Z
  
      goto :goto_0
  
+     .line 239
      .end local v0           #msg:Landroid/os/Message;
      :cond_1
      invoke-virtual {p0, v3}, Landroid/webkit/JWebCoreJavaBridge;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 240
      .restart local v0       #msg:Landroid/os/Message;
      invoke-virtual {p0, v0, p1, p2}, Landroid/webkit/JWebCoreJavaBridge;->sendMessageDelayed(Landroid/os/Message;J)Z
  
***************
*** 536,550 ****
--- 595,612 ----
      .locals 2
  
      .prologue
+     .line 125
      const/4 v1, 0x2
  
      invoke-virtual {p0, v1}, Landroid/webkit/JWebCoreJavaBridge;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 126
      .local v0, msg:Landroid/os/Message;
      invoke-virtual {p0, v0}, Landroid/webkit/JWebCoreJavaBridge;->sendMessage(Landroid/os/Message;)Z
  
+     .line 127
      return-void
  .end method
  
***************
*** 554,567 ****
--- 616,633 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 251
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Landroid/webkit/JWebCoreJavaBridge;->removeMessages(I)V
  
+     .line 252
      iput-boolean v1, p0, Landroid/webkit/JWebCoreJavaBridge;->mHasInstantTimer:Z
  
+     .line 253
      iput-boolean v1, p0, Landroid/webkit/JWebCoreJavaBridge;->mHasDeferredTimers:Z
  
+     .line 254
      return-void
  .end method
  
***************
*** 586,593 ****
--- 652,661 ----
      .locals 0
  
      .prologue
+     .line 67
      invoke-direct {p0}, Landroid/webkit/JWebCoreJavaBridge;->nativeFinalize()V
  
+     .line 68
      return-void
  .end method
  
***************
*** 596,629 ****
--- 664,704 ----
      .parameter "msg"
  
      .prologue
+     .line 103
      iget v0, p1, Landroid/os/Message;->what:I
  
      sparse-switch v0, :sswitch_data_0
  
+     .line 121
      :goto_0
      return-void
  
+     .line 105
      :sswitch_0
      iget-boolean v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mTimerPaused:Z
  
      if-eqz v0, :cond_0
  
+     .line 106
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mHasDeferredTimers:Z
  
      goto :goto_0
  
+     .line 108
      :cond_0
      invoke-direct {p0}, Landroid/webkit/JWebCoreJavaBridge;->fireSharedTimer()V
  
      goto :goto_0
  
+     .line 113
      :sswitch_1
      invoke-direct {p0}, Landroid/webkit/JWebCoreJavaBridge;->nativeServiceFuncPtrQueue()V
  
      goto :goto_0
  
+     .line 116
      :sswitch_2
      const/4 v0, 0x0
  
***************
*** 647,652 ****
--- 722,728 ----
  
      goto :goto_0
  
+     .line 103
      nop
  
      :sswitch_data_0
***************
*** 664,681 ****
--- 740,761 ----
      .locals 1
  
      .prologue
+     .line 135
      iget-boolean v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mTimerPaused:Z
  
      if-nez v0, :cond_0
  
+     .line 136
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mTimerPaused:Z
  
+     .line 137
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mHasDeferredTimers:Z
  
+     .line 139
      :cond_0
      return-void
  .end method
***************
*** 689,708 ****
--- 769,794 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 145
      iget-boolean v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mTimerPaused:Z
  
      if-eqz v0, :cond_0
  
+     .line 146
      iput-boolean v1, p0, Landroid/webkit/JWebCoreJavaBridge;->mTimerPaused:Z
  
+     .line 147
      iget-boolean v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mHasDeferredTimers:Z
  
      if-eqz v0, :cond_0
  
+     .line 148
      iput-boolean v1, p0, Landroid/webkit/JWebCoreJavaBridge;->mHasDeferredTimers:Z
  
+     .line 149
      invoke-direct {p0}, Landroid/webkit/JWebCoreJavaBridge;->fireSharedTimer()V
  
+     .line 152
      :cond_0
      return-void
  .end method
***************
*** 722,742 ****
--- 808,832 ----
      .parameter "contentUri"
  
      .prologue
+     .line 290
      iget-object v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mContentUriToFilePathMap:Ljava/util/HashMap;
  
      if-nez v0, :cond_0
  
+     .line 291
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mContentUriToFilePathMap:Ljava/util/HashMap;
  
+     .line 293
      :cond_0
      iget-object v0, p0, Landroid/webkit/JWebCoreJavaBridge;->mContentUriToFilePathMap:Ljava/util/HashMap;
  
      invoke-virtual {v0, p2, p1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 294
      return-void
  .end method
  
***************
*** 745,774 ****
--- 835,871 ----
      .parameter "proxyProperties"
  
      .prologue
+     .line 297
      if-nez p1, :cond_0
  
+     .line 298
      const-string v2, ""
  
      const-string v3, ""
  
      invoke-virtual {p0, v2, v3}, Landroid/webkit/JWebCoreJavaBridge;->nativeUpdateProxy(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 308
      :goto_0
      return-void
  
+     .line 302
      :cond_0
      invoke-virtual {p1}, Landroid/net/ProxyProperties;->getHost()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 303
      .local v0, host:Ljava/lang/String;
      invoke-virtual {p1}, Landroid/net/ProxyProperties;->getPort()I
  
      move-result v1
  
+     .line 304
      .local v1, port:I
      if-eqz v1, :cond_1
  
+     .line 305
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 791,796 ****
--- 888,894 ----
  
      move-result-object v0
  
+     .line 307
      :cond_1
      invoke-virtual {p1}, Landroid/net/ProxyProperties;->getExclusionList()Ljava/lang/String;
  
