*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/PluginFullScreenHolder$1.smali	2012-03-23 19:37:06.464892969 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/PluginFullScreenHolder$1.smali.original	2012-03-23 19:34:45.549323377 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 134
      iput-object p1, p0, Landroid/webkit/PluginFullScreenHolder$1;->this$0:Landroid/webkit/PluginFullScreenHolder;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .locals 4
  
      .prologue
+     .line 137
      iget-object v0, p0, Landroid/webkit/PluginFullScreenHolder$1;->this$0:Landroid/webkit/PluginFullScreenHolder;
  
      #getter for: Landroid/webkit/PluginFullScreenHolder;->mWebView:Landroid/webkit/WebView;
***************
*** 57,62 ****
--- 59,65 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 140
      iget-object v0, p0, Landroid/webkit/PluginFullScreenHolder$1;->this$0:Landroid/webkit/PluginFullScreenHolder;
  
      #getter for: Landroid/webkit/PluginFullScreenHolder;->mWebView:Landroid/webkit/WebView;
***************
*** 81,86 ****
--- 84,90 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/webkit/WebViewCore;->sendMessage(III)V
  
+     .line 143
      invoke-static {}, Landroid/webkit/PluginFullScreenHolder;->access$300()Landroid/webkit/PluginFullScreenHolder$CustomFrameLayout;
  
      move-result-object v0
***************
*** 94,103 ****
--- 98,109 ----
  
      invoke-virtual {v0, v1}, Landroid/webkit/PluginFullScreenHolder$CustomFrameLayout;->removeView(Landroid/view/View;)V
  
+     .line 144
      const/4 v0, 0x0
  
      invoke-static {v0}, Landroid/webkit/PluginFullScreenHolder;->access$302(Landroid/webkit/PluginFullScreenHolder$CustomFrameLayout;)Landroid/webkit/PluginFullScreenHolder$CustomFrameLayout;
  
+     .line 147
      iget-object v0, p0, Landroid/webkit/PluginFullScreenHolder$1;->this$0:Landroid/webkit/PluginFullScreenHolder;
  
      #getter for: Landroid/webkit/PluginFullScreenHolder;->mWebView:Landroid/webkit/WebView;
***************
*** 111,115 ****
--- 117,122 ----
  
      invoke-virtual {v0}, Landroid/webkit/ViewManager;->showAll()V
  
+     .line 148
      return-void
  .end method
