*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/StreamLoader.smali	2012-03-23 19:37:13.532021838 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/StreamLoader.smali.original	2012-03-23 19:34:45.589324107 -0500
***************
*** 36,51 ****
--- 36,55 ----
      .parameter "loadlistener"
  
      .prologue
+     .line 69
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 70
      iput-object p1, p0, Landroid/webkit/StreamLoader;->mLoadListener:Landroid/webkit/LoadListener;
  
+     .line 71
      invoke-virtual {p1}, Landroid/webkit/LoadListener;->getContext()Landroid/content/Context;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/webkit/StreamLoader;->mContext:Landroid/content/Context;
  
+     .line 72
      return-void
  .end method
  
***************
*** 53,62 ****
--- 57,68 ----
      .locals 1
  
      .prologue
+     .line 195
      iget-object v0, p0, Landroid/webkit/StreamLoader;->mDataStream:Ljava/io/InputStream;
  
      if-eqz v0, :cond_0
  
+     .line 197
      :try_start_0
      iget-object v0, p0, Landroid/webkit/StreamLoader;->mDataStream:Ljava/io/InputStream;
  
***************
*** 64,77 ****
--- 70,86 ----
      :try_end_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 202
      :cond_0
      :goto_0
      iget-object v0, p0, Landroid/webkit/StreamLoader;->mLoadListener:Landroid/webkit/LoadListener;
  
      invoke-virtual {v0}, Landroid/webkit/LoadListener;->endData()V
  
+     .line 203
      return-void
  
+     .line 198
      :catch_0
      move-exception v0
  
***************
*** 82,91 ****
--- 91,102 ----
      .locals 5
  
      .prologue
+     .line 177
      iget-object v2, p0, Landroid/webkit/StreamLoader;->mDataStream:Ljava/io/InputStream;
  
      if-eqz v2, :cond_0
  
+     .line 179
      :try_start_0
      iget-object v2, p0, Landroid/webkit/StreamLoader;->mDataStream:Ljava/io/InputStream;
  
***************
*** 95,103 ****
--- 106,116 ----
  
      move-result v0
  
+     .line 180
      .local v0, amount:I
      if-lez v0, :cond_0
  
+     .line 181
      iget-object v2, p0, Landroid/webkit/StreamLoader;->mLoadListener:Landroid/webkit/LoadListener;
  
      iget-object v3, p0, Landroid/webkit/StreamLoader;->mData:[B
***************
*** 106,120 ****
--- 119,137 ----
      :try_end_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 182
      const/4 v2, 0x0
  
+     .line 188
      .end local v0           #amount:I
      :goto_0
      return v2
  
+     .line 184
      :catch_0
      move-exception v1
  
+     .line 185
      .local v1, ex:Ljava/io/IOException;
      iget-object v2, p0, Landroid/webkit/StreamLoader;->mLoadListener:Landroid/webkit/LoadListener;
  
***************
*** 126,131 ****
--- 143,149 ----
  
      invoke-virtual {v2, v3, v4}, Landroid/webkit/LoadListener;->error(ILjava/lang/String;)V
  
+     .line 188
      .end local v1           #ex:Ljava/io/IOException;
      :cond_0
      const/4 v2, 0x1
***************
*** 137,146 ****
--- 155,166 ----
      .locals 5
  
      .prologue
+     .line 161
      new-instance v0, Landroid/net/http/Headers;
  
      invoke-direct {v0}, Landroid/net/http/Headers;-><init>()V
  
+     .line 162
      .local v0, headers:Landroid/net/http/Headers;
      iget-wide v1, p0, Landroid/webkit/StreamLoader;->mContentLength:J
  
***************
*** 150,166 ****
--- 170,190 ----
  
      if-lez v1, :cond_0
  
+     .line 163
      iget-wide v1, p0, Landroid/webkit/StreamLoader;->mContentLength:J
  
      invoke-virtual {v0, v1, v2}, Landroid/net/http/Headers;->setContentLength(J)V
  
+     .line 165
      :cond_0
      invoke-virtual {p0, v0}, Landroid/webkit/StreamLoader;->buildHeaders(Landroid/net/http/Headers;)V
  
+     .line 166
      iget-object v1, p0, Landroid/webkit/StreamLoader;->mLoadListener:Landroid/webkit/LoadListener;
  
      invoke-virtual {v1, v0}, Landroid/webkit/LoadListener;->headers(Landroid/net/http/Headers;)V
  
+     .line 167
      return-void
  .end method
  
***************
*** 178,183 ****
--- 202,208 ----
  
      const/4 v0, 0x1
  
+     .line 121
      iget-object v1, p0, Landroid/webkit/StreamLoader;->mLoadListener:Landroid/webkit/LoadListener;
  
      invoke-virtual {v1}, Landroid/webkit/LoadListener;->isSynchronous()Z
***************
*** 186,197 ****
--- 211,224 ----
  
      if-eqz v1, :cond_0
  
+     .line 122
      new-instance v0, Ljava/lang/AssertionError;
  
      invoke-direct {v0}, Ljava/lang/AssertionError;-><init>()V
  
      throw v0
  
+     .line 124
      :cond_0
      iget-object v1, p0, Landroid/webkit/StreamLoader;->mLoadListener:Landroid/webkit/LoadListener;
  
***************
*** 201,221 ****
--- 228,253 ----
  
      if-eqz v1, :cond_2
  
+     .line 125
      invoke-direct {p0}, Landroid/webkit/StreamLoader;->closeStreamAndSendEndData()V
  
+     .line 154
      :cond_1
      :goto_0
      return v0
  
+     .line 128
      :cond_2
      iget v1, p1, Landroid/os/Message;->what:I
  
      packed-switch v1, :pswitch_data_0
  
+     .line 152
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 130
      :pswitch_0
      invoke-virtual {p0}, Landroid/webkit/StreamLoader;->setupStreamAndSendStatus()Z
  
***************
*** 223,234 ****
--- 255,268 ----
  
      if-eqz v1, :cond_1
  
+     .line 133
      const/16 v1, 0x2000
  
      new-array v1, v1, [B
  
      iput-object v1, p0, Landroid/webkit/StreamLoader;->mData:[B
  
+     .line 134
      iget-object v1, p0, Landroid/webkit/StreamLoader;->mHandler:Landroid/os/Handler;
  
      const/16 v2, 0x65
***************
*** 237,251 ****
--- 271,288 ----
  
      goto :goto_0
  
+     .line 138
      :pswitch_1
      invoke-direct {p0}, Landroid/webkit/StreamLoader;->sendHeaders()V
  
+     .line 139
      iget-object v1, p0, Landroid/webkit/StreamLoader;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v2}, Landroid/os/Handler;->sendEmptyMessage(I)Z
  
      goto :goto_0
  
+     .line 142
      :pswitch_2
      invoke-direct {p0}, Landroid/webkit/StreamLoader;->sendData()Z
  
***************
*** 253,258 ****
--- 290,296 ----
  
      if-eqz v1, :cond_3
  
+     .line 143
      iget-object v1, p0, Landroid/webkit/StreamLoader;->mHandler:Landroid/os/Handler;
  
      const/16 v2, 0x67
***************
*** 261,266 ****
--- 299,305 ----
  
      goto :goto_0
  
+     .line 145
      :cond_3
      iget-object v1, p0, Landroid/webkit/StreamLoader;->mHandler:Landroid/os/Handler;
  
***************
*** 268,278 ****
--- 307,319 ----
  
      goto :goto_0
  
+     .line 149
      :pswitch_3
      invoke-direct {p0}, Landroid/webkit/StreamLoader;->closeStreamAndSendEndData()V
  
      goto :goto_0
  
+     .line 128
      nop
  
      :pswitch_data_0
***************
*** 288,311 ****
--- 329,357 ----
      .locals 2
  
      .prologue
+     .line 98
      monitor-enter p0
  
+     .line 99
      :try_start_0
      iget-object v0, p0, Landroid/webkit/StreamLoader;->mHandler:Landroid/os/Handler;
  
      if-nez v0, :cond_0
  
+     .line 100
      new-instance v0, Landroid/os/Handler;
  
      invoke-direct {v0, p0}, Landroid/os/Handler;-><init>(Landroid/os/Handler$Callback;)V
  
      iput-object v0, p0, Landroid/webkit/StreamLoader;->mHandler:Landroid/os/Handler;
  
+     .line 102
      :cond_0
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 104
      iget-object v0, p0, Landroid/webkit/StreamLoader;->mLoadListener:Landroid/webkit/LoadListener;
  
      invoke-virtual {v0}, Landroid/webkit/LoadListener;->isSynchronous()Z
***************
*** 314,329 ****
--- 360,378 ----
  
      if-nez v0, :cond_2
  
+     .line 105
      iget-object v0, p0, Landroid/webkit/StreamLoader;->mHandler:Landroid/os/Handler;
  
      const/16 v1, 0x64
  
      invoke-virtual {v0, v1}, Landroid/os/Handler;->sendEmptyMessage(I)Z
  
+     .line 118
      :cond_1
      :goto_0
      return-void
  
+     .line 102
      :catchall_0
      move-exception v0
  
***************
*** 334,339 ****
--- 383,389 ----
  
      throw v0
  
+     .line 108
      :cond_2
      invoke-virtual {p0}, Landroid/webkit/StreamLoader;->setupStreamAndSendStatus()Z
  
***************
*** 341,354 ****
--- 391,407 ----
  
      if-eqz v0, :cond_1
  
+     .line 111
      const/16 v0, 0x2000
  
      new-array v0, v0, [B
  
      iput-object v0, p0, Landroid/webkit/StreamLoader;->mData:[B
  
+     .line 112
      invoke-direct {p0}, Landroid/webkit/StreamLoader;->sendHeaders()V
  
+     .line 113
      :cond_3
      invoke-direct {p0}, Landroid/webkit/StreamLoader;->sendData()Z
  
***************
*** 364,372 ****
--- 417,427 ----
  
      if-eqz v0, :cond_3
  
+     .line 114
      :cond_4
      invoke-direct {p0}, Landroid/webkit/StreamLoader;->closeStreamAndSendEndData()V
  
+     .line 115
      iget-object v0, p0, Landroid/webkit/StreamLoader;->mLoadListener:Landroid/webkit/LoadListener;
  
      invoke-virtual {v0}, Landroid/webkit/LoadListener;->loadSynchronousMessages()V
