*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/ViewManager$1.smali	2012-03-23 19:37:09.099941020 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/ViewManager$1.smali.original	2012-03-23 19:34:45.568323725 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 211
      iput-object p1, p0, Landroid/webkit/ViewManager$1;->this$0:Landroid/webkit/ViewManager;
  
      iput-object p2, p0, Landroid/webkit/ViewManager$1;->val$sView:Landroid/view/SurfaceView;
***************
*** 45,55 ****
--- 46,58 ----
      .locals 2
  
      .prologue
+     .line 213
      iget-object v0, p0, Landroid/webkit/ViewManager$1;->val$sView:Landroid/view/SurfaceView;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, v1}, Landroid/view/SurfaceView;->setVisibility(I)V
  
+     .line 214
      return-void
  .end method
