*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebHistoryItem.smali	2012-03-23 19:37:05.201869939 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebHistoryItem.smali.original	2012-03-23 19:34:45.539323195 -0500
***************
*** 35,40 ****
--- 35,41 ----
      .locals 1
  
      .prologue
+     .line 32
      const/4 v0, 0x0
  
      sput v0, Landroid/webkit/WebHistoryItem;->sNextId:I
***************
*** 46,57 ****
--- 47,61 ----
      .locals 3
  
      .prologue
+     .line 58
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 59
      const-class v1, Landroid/webkit/WebHistoryItem;
  
      monitor-enter v1
  
+     .line 60
      :try_start_0
      sget v0, Landroid/webkit/WebHistoryItem;->sNextId:I
  
***************
*** 61,70 ****
--- 65,77 ----
  
      iput v0, p0, Landroid/webkit/WebHistoryItem;->mId:I
  
+     .line 61
      monitor-exit v1
  
+     .line 62
      return-void
  
+     .line 61
      :catchall_0
      move-exception v0
  
***************
*** 80,107 ****
--- 87,121 ----
      .parameter "item"
  
      .prologue
+     .line 80
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 81
      iget-object v0, p1, Landroid/webkit/WebHistoryItem;->mUrl:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/webkit/WebHistoryItem;->mUrl:Ljava/lang/String;
  
+     .line 82
      iget-object v0, p1, Landroid/webkit/WebHistoryItem;->mTitle:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/webkit/WebHistoryItem;->mTitle:Ljava/lang/String;
  
+     .line 83
      iget-object v0, p1, Landroid/webkit/WebHistoryItem;->mFlattenedData:[B
  
      iput-object v0, p0, Landroid/webkit/WebHistoryItem;->mFlattenedData:[B
  
+     .line 84
      iget-object v0, p1, Landroid/webkit/WebHistoryItem;->mFavicon:Landroid/graphics/Bitmap;
  
      iput-object v0, p0, Landroid/webkit/WebHistoryItem;->mFavicon:Landroid/graphics/Bitmap;
  
+     .line 85
      iget v0, p1, Landroid/webkit/WebHistoryItem;->mId:I
  
      iput v0, p0, Landroid/webkit/WebHistoryItem;->mId:I
  
+     .line 86
      return-void
  .end method
  
***************
*** 110,127 ****
--- 124,146 ----
      .parameter "data"
  
      .prologue
+     .line 68
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 69
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/webkit/WebHistoryItem;->mUrl:Ljava/lang/String;
  
+     .line 70
      iput-object p1, p0, Landroid/webkit/WebHistoryItem;->mFlattenedData:[B
  
+     .line 71
      const-class v1, Landroid/webkit/WebHistoryItem;
  
      monitor-enter v1
  
+     .line 72
      :try_start_0
      sget v0, Landroid/webkit/WebHistoryItem;->sNextId:I
  
***************
*** 131,140 ****
--- 150,162 ----
  
      iput v0, p0, Landroid/webkit/WebHistoryItem;->mId:I
  
+     .line 73
      monitor-exit v1
  
+     .line 74
      return-void
  
+     .line 73
      :catchall_0
      move-exception v0
  
***************
*** 157,172 ****
--- 179,200 ----
      .parameter "data"
  
      .prologue
+     .line 243
      iput-object p1, p0, Landroid/webkit/WebHistoryItem;->mUrl:Ljava/lang/String;
  
+     .line 244
      iput-object p2, p0, Landroid/webkit/WebHistoryItem;->mOriginalUrl:Ljava/lang/String;
  
+     .line 245
      iput-object p3, p0, Landroid/webkit/WebHistoryItem;->mTitle:Ljava/lang/String;
  
+     .line 246
      iput-object p4, p0, Landroid/webkit/WebHistoryItem;->mFavicon:Landroid/graphics/Bitmap;
  
+     .line 247
      iput-object p5, p0, Landroid/webkit/WebHistoryItem;->mFlattenedData:[B
  
+     .line 248
      return-void
  .end method
  
***************
*** 176,181 ****
--- 204,210 ----
      .locals 1
  
      .prologue
+     .line 233
      monitor-enter p0
  
      :try_start_0
***************
*** 206,211 ****
--- 235,241 ----
      .end annotation
  
      .prologue
+     .line 30
      invoke-virtual {p0}, Landroid/webkit/WebHistoryItem;->clone()Landroid/webkit/WebHistoryItem;
  
      move-result-object v0
***************
*** 217,222 ****
--- 247,253 ----
      .locals 1
  
      .prologue
+     .line 173
      iget-object v0, p0, Landroid/webkit/WebHistoryItem;->mCustomData:Ljava/lang/Object;
  
      return-object v0
***************
*** 226,231 ****
--- 257,263 ----
      .locals 1
  
      .prologue
+     .line 139
      iget-object v0, p0, Landroid/webkit/WebHistoryItem;->mFavicon:Landroid/graphics/Bitmap;
  
      return-object v0
***************
*** 235,240 ****
--- 267,273 ----
      .locals 1
  
      .prologue
+     .line 217
      iget-object v0, p0, Landroid/webkit/WebHistoryItem;->mFlattenedData:[B
  
      return-object v0
***************
*** 246,251 ****
--- 279,285 ----
      .end annotation
  
      .prologue
+     .line 97
      iget v0, p0, Landroid/webkit/WebHistoryItem;->mId:I
  
      return v0
***************
*** 255,260 ****
--- 289,295 ----
      .locals 1
  
      .prologue
+     .line 119
      iget-object v0, p0, Landroid/webkit/WebHistoryItem;->mOriginalUrl:Ljava/lang/String;
  
      return-object v0
***************
*** 264,269 ****
--- 299,305 ----
      .locals 1
  
      .prologue
+     .line 129
      iget-object v0, p0, Landroid/webkit/WebHistoryItem;->mTitle:Ljava/lang/String;
  
      return-object v0
***************
*** 273,296 ****
--- 309,338 ----
      .locals 7
  
      .prologue
+     .line 152
      iget-object v2, p0, Landroid/webkit/WebHistoryItem;->mTouchIconUrlFromLink:Ljava/lang/String;
  
      if-eqz v2, :cond_0
  
+     .line 153
      iget-object v2, p0, Landroid/webkit/WebHistoryItem;->mTouchIconUrlFromLink:Ljava/lang/String;
  
+     .line 165
      :goto_0
      return-object v2
  
+     .line 154
      :cond_0
      iget-object v2, p0, Landroid/webkit/WebHistoryItem;->mTouchIconUrlServerDefault:Ljava/lang/String;
  
      if-eqz v2, :cond_1
  
+     .line 155
      iget-object v2, p0, Landroid/webkit/WebHistoryItem;->mTouchIconUrlServerDefault:Ljava/lang/String;
  
      goto :goto_0
  
+     .line 159
      :cond_1
      :try_start_0
      new-instance v1, Ljava/net/URL;
***************
*** 299,304 ****
--- 341,347 ----
  
      invoke-direct {v1, v2}, Ljava/net/URL;-><init>(Ljava/lang/String;)V
  
+     .line 160
      .local v1, url:Ljava/net/URL;
      new-instance v2, Ljava/net/URL;
  
***************
*** 326,339 ****
--- 369,385 ----
      :try_end_0
      .catch Ljava/net/MalformedURLException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 165
      iget-object v2, p0, Landroid/webkit/WebHistoryItem;->mTouchIconUrlServerDefault:Ljava/lang/String;
  
      goto :goto_0
  
+     .line 162
      .end local v1           #url:Ljava/net/URL;
      :catch_0
      move-exception v0
  
+     .line 163
      .local v0, e:Ljava/net/MalformedURLException;
      const/4 v2, 0x0
  
***************
*** 344,349 ****
--- 390,396 ----
      .locals 1
  
      .prologue
+     .line 109
      iget-object v0, p0, Landroid/webkit/WebHistoryItem;->mUrl:Ljava/lang/String;
  
      return-object v0
***************
*** 354,363 ****
--- 401,412 ----
      .parameter "nativeFrame"
  
      .prologue
+     .line 226
      iget-object v0, p0, Landroid/webkit/WebHistoryItem;->mFlattenedData:[B
  
      invoke-direct {p0, p1, v0}, Landroid/webkit/WebHistoryItem;->inflate(I[B)V
  
+     .line 227
      return-void
  .end method
  
***************
*** 366,373 ****
--- 415,424 ----
      .parameter "data"
  
      .prologue
+     .line 187
      iput-object p1, p0, Landroid/webkit/WebHistoryItem;->mCustomData:Ljava/lang/Object;
  
+     .line 188
      return-void
  .end method
  
***************
*** 376,383 ****
--- 427,436 ----
      .parameter "icon"
  
      .prologue
+     .line 197
      iput-object p1, p0, Landroid/webkit/WebHistoryItem;->mFavicon:Landroid/graphics/Bitmap;
  
+     .line 198
      return-void
  .end method
  
***************
*** 387,401 ****
--- 440,457 ----
      .parameter "precomposed"
  
      .prologue
+     .line 206
      if-nez p2, :cond_0
  
      iget-object v0, p0, Landroid/webkit/WebHistoryItem;->mTouchIconUrlFromLink:Ljava/lang/String;
  
      if-nez v0, :cond_1
  
+     .line 207
      :cond_0
      iput-object p1, p0, Landroid/webkit/WebHistoryItem;->mTouchIconUrlFromLink:Ljava/lang/String;
  
+     .line 209
      :cond_1
      return-void
  .end method
