*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebStorage$Origin.smali	2012-03-23 19:37:11.189979131 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebStorage$Origin.smali.original	2012-03-23 19:34:45.578323907 -0500
***************
*** 30,47 ****
--- 30,53 ----
      .prologue
      const-wide/16 v1, 0x0
  
+     .line 97
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 82
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/webkit/WebStorage$Origin;->mOrigin:Ljava/lang/String;
  
+     .line 83
      iput-wide v1, p0, Landroid/webkit/WebStorage$Origin;->mQuota:J
  
+     .line 84
      iput-wide v1, p0, Landroid/webkit/WebStorage$Origin;->mUsage:J
  
+     .line 98
      iput-object p1, p0, Landroid/webkit/WebStorage$Origin;->mOrigin:Ljava/lang/String;
  
+     .line 99
      return-void
  .end method
  
***************
*** 53,72 ****
--- 59,85 ----
      .prologue
      const-wide/16 v1, 0x0
  
+     .line 92
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 82
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/webkit/WebStorage$Origin;->mOrigin:Ljava/lang/String;
  
+     .line 83
      iput-wide v1, p0, Landroid/webkit/WebStorage$Origin;->mQuota:J
  
+     .line 84
      iput-wide v1, p0, Landroid/webkit/WebStorage$Origin;->mUsage:J
  
+     .line 93
      iput-object p1, p0, Landroid/webkit/WebStorage$Origin;->mOrigin:Ljava/lang/String;
  
+     .line 94
      iput-wide p2, p0, Landroid/webkit/WebStorage$Origin;->mQuota:J
  
+     .line 95
      return-void
  .end method
  
***************
*** 79,100 ****
--- 92,121 ----
      .prologue
      const-wide/16 v1, 0x0
  
+     .line 86
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 82
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/webkit/WebStorage$Origin;->mOrigin:Ljava/lang/String;
  
+     .line 83
      iput-wide v1, p0, Landroid/webkit/WebStorage$Origin;->mQuota:J
  
+     .line 84
      iput-wide v1, p0, Landroid/webkit/WebStorage$Origin;->mUsage:J
  
+     .line 87
      iput-object p1, p0, Landroid/webkit/WebStorage$Origin;->mOrigin:Ljava/lang/String;
  
+     .line 88
      iput-wide p2, p0, Landroid/webkit/WebStorage$Origin;->mQuota:J
  
+     .line 89
      iput-wide p4, p0, Landroid/webkit/WebStorage$Origin;->mUsage:J
  
+     .line 90
      return-void
  .end method
  
***************
*** 106,111 ****
--- 127,133 ----
      .parameter "x3"
  
      .prologue
+     .line 81
      invoke-direct/range {p0 .. p5}, Landroid/webkit/WebStorage$Origin;-><init>(Ljava/lang/String;JJ)V
  
      return-void
***************
*** 118,123 ****
--- 140,146 ----
      .parameter "x2"
  
      .prologue
+     .line 81
      invoke-direct {p0, p1, p2, p3}, Landroid/webkit/WebStorage$Origin;-><init>(Ljava/lang/String;J)V
  
      return-void
***************
*** 129,134 ****
--- 152,158 ----
      .parameter "x1"
  
      .prologue
+     .line 81
      invoke-direct {p0, p1}, Landroid/webkit/WebStorage$Origin;-><init>(Ljava/lang/String;)V
  
      return-void
***************
*** 140,145 ****
--- 164,170 ----
      .locals 1
  
      .prologue
+     .line 110
      iget-object v0, p0, Landroid/webkit/WebStorage$Origin;->mOrigin:Ljava/lang/String;
  
      return-object v0
***************
*** 149,154 ****
--- 174,180 ----
      .locals 2
  
      .prologue
+     .line 118
      iget-wide v0, p0, Landroid/webkit/WebStorage$Origin;->mQuota:J
  
      return-wide v0
***************
*** 158,163 ****
--- 184,190 ----
      .locals 2
  
      .prologue
+     .line 126
      iget-wide v0, p0, Landroid/webkit/WebStorage$Origin;->mUsage:J
  
      return-wide v0
