*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebTextView$WebTextViewLayout.smali	2012-03-23 19:37:06.979902361 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebTextView$WebTextViewLayout.smali.original	2012-03-23 19:34:45.553323451 -0500
***************
*** 36,43 ****
--- 36,45 ----
      .parameter "lineHeight"
  
      .prologue
+     .line 499
      invoke-direct/range {p0 .. p10}, Landroid/text/DynamicLayout;-><init>(Ljava/lang/CharSequence;Ljava/lang/CharSequence;Landroid/text/TextPaint;ILandroid/text/Layout$Alignment;FFZLandroid/text/TextUtils$TruncateAt;I)V
  
+     .line 501
      invoke-virtual {p3}, Landroid/text/TextPaint;->descent()F
  
      move-result v1
***************
*** 48,53 ****
--- 50,56 ----
  
      sub-float v0, v1, v2
  
+     .line 502
      .local v0, paintLineHeight:F
      const/high16 v1, -0x4080
  
***************
*** 55,72 ****
--- 58,80 ----
  
      if-nez v1, :cond_0
  
+     .line 503
      iput v0, p0, Landroid/webkit/WebTextView$WebTextViewLayout;->mLineHeight:F
  
+     .line 504
      const/4 v1, 0x0
  
      iput v1, p0, Landroid/webkit/WebTextView$WebTextViewLayout;->mDifference:F
  
+     .line 511
      :goto_0
      return-void
  
+     .line 506
      :cond_0
      iput p11, p0, Landroid/webkit/WebTextView$WebTextViewLayout;->mLineHeight:F
  
+     .line 509
      sub-float v1, p11, v0
  
      const/high16 v2, 0x4000
***************
*** 85,90 ****
--- 93,99 ----
      .parameter "line"
  
      .prologue
+     .line 515
      iget v0, p0, Landroid/webkit/WebTextView$WebTextViewLayout;->mLineHeight:F
  
      int-to-float v1, p1
