*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$GetUrlData.smali	2012-03-23 19:37:05.121868480 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$GetUrlData.smali.original	2012-03-23 19:34:45.538323177 -0500
***************
*** 35,40 ****
--- 35,41 ----
      .locals 0
  
      .prologue
+     .line 773
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
