*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$EventHub$1.smali	2012-03-23 19:37:07.337908888 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$EventHub$1.smali.original	2012-03-23 19:34:45.556323505 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1076
      iput-object p1, p0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
  
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
***************
*** 38,43 ****
--- 39,45 ----
      .parameter "msg"
  
      .prologue
+     .line 1088
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 64,73 ****
--- 66,77 ----
  
      if-nez v2, :cond_1
  
+     .line 1649
      :cond_0
      :goto_0
      return-void
  
+     .line 1095
      :cond_1
      move-object/from16 v0, p0
  
***************
*** 106,111 ****
--- 110,116 ----
  
      if-ne v2, v4, :cond_0
  
+     .line 1105
      :cond_2
      move-object/from16 v0, p1
  
***************
*** 116,121 ****
--- 121,127 ----
      :pswitch_0
      goto :goto_0
  
+     .line 1127
      :pswitch_1
      move-object/from16 v0, p0
  
***************
*** 128,133 ****
--- 134,140 ----
  
      goto :goto_0
  
+     .line 1107
      :pswitch_2
      move-object/from16 v0, p0
  
***************
*** 140,145 ****
--- 147,153 ----
  
      goto :goto_0
  
+     .line 1111
      :pswitch_3
      move-object/from16 v0, p0
  
***************
*** 152,157 ****
--- 160,166 ----
  
      goto :goto_0
  
+     .line 1117
      :pswitch_4
      move-object/from16 v0, p0
  
***************
*** 161,166 ****
--- 170,176 ----
  
      monitor-enter v4
  
+     .line 1118
      :try_start_0
      move-object/from16 v0, p0
  
***************
*** 175,180 ****
--- 185,191 ----
  
      invoke-virtual {v2}, Landroid/webkit/BrowserFrame;->destroy()V
  
+     .line 1119
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 186,191 ****
--- 197,203 ----
      #setter for: Landroid/webkit/WebViewCore;->mBrowserFrame:Landroid/webkit/BrowserFrame;
      invoke-static {v2, v6}, Landroid/webkit/WebViewCore;->access$1102(Landroid/webkit/WebViewCore;Landroid/webkit/BrowserFrame;)Landroid/webkit/BrowserFrame;
  
+     .line 1120
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 199,204 ****
--- 211,217 ----
  
      invoke-virtual {v2}, Landroid/webkit/WebSettings;->onDestroyed()V
  
+     .line 1121
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 210,215 ****
--- 223,229 ----
      #setter for: Landroid/webkit/WebViewCore;->mNativeClass:I
      invoke-static {v2, v6}, Landroid/webkit/WebViewCore;->access$702(Landroid/webkit/WebViewCore;I)I
  
+     .line 1122
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 221,226 ****
--- 235,241 ----
      #setter for: Landroid/webkit/WebViewCore;->mWebView:Landroid/webkit/WebView;
      invoke-static {v2, v6}, Landroid/webkit/WebViewCore;->access$602(Landroid/webkit/WebViewCore;Landroid/webkit/WebView;)Landroid/webkit/WebView;
  
+     .line 1123
      monitor-exit v4
  
      goto/16 :goto_0
***************
*** 234,239 ****
--- 249,255 ----
  
      throw v2
  
+     .line 1131
      :pswitch_5
      move-object/from16 v0, p0
  
***************
*** 248,259 ****
--- 264,277 ----
  
      if-eqz v2, :cond_0
  
+     .line 1132
      move-object/from16 v0, p1
  
      iget v0, v0, Landroid/os/Message;->arg2:I
  
      move/from16 v39, v0
  
+     .line 1133
      .local v39, nodePointer:I
      move-object/from16 v0, p0
  
***************
*** 272,277 ****
--- 290,296 ----
  
      move-result-object v33
  
+     .line 1135
      .local v33, label:Ljava/lang/String;
      if-eqz v33, :cond_0
  
***************
*** 281,286 ****
--- 300,306 ----
  
      if-lez v2, :cond_0
  
+     .line 1136
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 310,315 ****
--- 330,336 ----
  
      goto/16 :goto_0
  
+     .line 1144
      .end local v33           #label:Ljava/lang/String;
      .end local v39           #nodePointer:I
      :pswitch_6
***************
*** 324,329 ****
--- 345,351 ----
  
      goto/16 :goto_0
  
+     .line 1149
      :pswitch_7
      move-object/from16 v0, p1
  
***************
*** 331,338 ****
--- 353,362 ----
  
      if-nez v2, :cond_3
  
+     .line 1150
      const/16 v49, 0x0
  
+     .line 1154
      .local v49, xPercent:F
      :goto_1
      move-object/from16 v0, p0
***************
*** 352,357 ****
--- 376,382 ----
  
      goto/16 :goto_0
  
+     .line 1152
      .end local v49           #xPercent:F
      :cond_3
      move-object/from16 v0, p1
***************
*** 367,372 ****
--- 392,398 ----
      .restart local v49       #xPercent:F
      goto :goto_1
  
+     .line 1158
      .end local v49           #xPercent:F
      :pswitch_8
      invoke-static {}, Landroid/webkit/CookieManager;->getInstance()Landroid/webkit/CookieManager;
***************
*** 375,380 ****
--- 401,407 ----
  
      invoke-virtual {v2}, Landroid/webkit/CookieManager;->waitForCookieOperationsToComplete()V
  
+     .line 1159
      move-object/from16 v0, p1
  
      iget-object v0, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
***************
*** 383,388 ****
--- 410,416 ----
  
      check-cast v40, Landroid/webkit/WebViewCore$GetUrlData;
  
+     .line 1160
      .local v40, param:Landroid/webkit/WebViewCore$GetUrlData;
      move-object/from16 v0, p0
  
***************
*** 403,408 ****
--- 431,437 ----
  
      goto/16 :goto_0
  
+     .line 1165
      .end local v40           #param:Landroid/webkit/WebViewCore$GetUrlData;
      :pswitch_9
      invoke-static {}, Landroid/webkit/CookieManager;->getInstance()Landroid/webkit/CookieManager;
***************
*** 411,416 ****
--- 440,446 ----
  
      invoke-virtual {v2}, Landroid/webkit/CookieManager;->waitForCookieOperationsToComplete()V
  
+     .line 1166
      move-object/from16 v0, p1
  
      iget-object v0, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
***************
*** 419,424 ****
--- 449,455 ----
  
      check-cast v40, Landroid/webkit/WebViewCore$PostUrlData;
  
+     .line 1167
      .local v40, param:Landroid/webkit/WebViewCore$PostUrlData;
      move-object/from16 v0, p0
  
***************
*** 443,448 ****
--- 474,480 ----
  
      goto/16 :goto_0
  
+     .line 1171
      .end local v40           #param:Landroid/webkit/WebViewCore$PostUrlData;
      :pswitch_a
      invoke-static {}, Landroid/webkit/CookieManager;->getInstance()Landroid/webkit/CookieManager;
***************
*** 451,456 ****
--- 483,489 ----
  
      invoke-virtual {v2}, Landroid/webkit/CookieManager;->waitForCookieOperationsToComplete()V
  
+     .line 1172
      move-object/from16 v0, p1
  
      iget-object v0, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
***************
*** 459,481 ****
--- 492,519 ----
  
      check-cast v34, Landroid/webkit/WebViewCore$BaseUrlData;
  
+     .line 1173
      .local v34, loadParams:Landroid/webkit/WebViewCore$BaseUrlData;
      move-object/from16 v0, v34
  
      iget-object v3, v0, Landroid/webkit/WebViewCore$BaseUrlData;->mBaseUrl:Ljava/lang/String;
  
+     .line 1174
      .local v3, baseUrl:Ljava/lang/String;
      if-eqz v3, :cond_4
  
+     .line 1175
      const/16 v2, 0x3a
  
      invoke-virtual {v3, v2}, Ljava/lang/String;->indexOf(I)I
  
      move-result v30
  
+     .line 1176
      .local v30, i:I
      if-lez v30, :cond_4
  
+     .line 1184
      const/4 v2, 0x0
  
      move/from16 v0, v30
***************
*** 484,489 ****
--- 522,528 ----
  
      move-result-object v46
  
+     .line 1185
      .local v46, scheme:Ljava/lang/String;
      const-string v2, "http"
  
***************
*** 525,530 ****
--- 564,570 ----
  
      if-nez v2, :cond_4
  
+     .line 1189
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 536,541 ****
--- 576,582 ----
      #calls: Landroid/webkit/WebViewCore;->nativeRegisterURLSchemeAsLocal(Ljava/lang/String;)V
      invoke-static {v2, v0}, Landroid/webkit/WebViewCore;->access$1800(Landroid/webkit/WebViewCore;Ljava/lang/String;)V
  
+     .line 1193
      .end local v30           #i:I
      .end local v46           #scheme:Ljava/lang/String;
      :cond_4
***************
*** 568,573 ****
--- 609,615 ----
  
      invoke-virtual/range {v2 .. v7}, Landroid/webkit/BrowserFrame;->loadData(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1198
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 579,584 ****
--- 621,627 ----
  
      goto/16 :goto_0
  
+     .line 1206
      .end local v3           #baseUrl:Ljava/lang/String;
      .end local v34           #loadParams:Landroid/webkit/WebViewCore$BaseUrlData;
      :pswitch_b
***************
*** 616,621 ****
--- 659,665 ----
  
      if-nez v2, :cond_5
  
+     .line 1208
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 629,634 ****
--- 673,679 ----
  
      invoke-virtual {v2}, Landroid/webkit/BrowserFrame;->didFirstLayout()V
  
+     .line 1211
      :cond_5
      move-object/from16 v0, p0
  
***************
*** 640,645 ****
--- 685,691 ----
  
      goto/16 :goto_0
  
+     .line 1215
      :pswitch_c
      move-object/from16 v0, p0
  
***************
*** 658,663 ****
--- 704,710 ----
  
      goto/16 :goto_0
  
+     .line 1219
      :pswitch_d
      move-object/from16 v0, p0
  
***************
*** 678,683 ****
--- 725,731 ----
  
      goto/16 :goto_0
  
+     .line 1223
      :pswitch_e
      move-object/from16 v0, p0
  
***************
*** 698,703 ****
--- 746,752 ----
  
      goto/16 :goto_0
  
+     .line 1227
      :pswitch_f
      move-object/from16 v0, p0
  
***************
*** 720,725 ****
--- 769,775 ----
  
      goto/16 :goto_0
  
+     .line 1231
      :pswitch_10
      move-object/from16 v0, p0
  
***************
*** 742,747 ****
--- 792,798 ----
  
      goto/16 :goto_0
  
+     .line 1235
      :pswitch_11
      move-object/from16 v0, p0
  
***************
*** 760,765 ****
--- 811,817 ----
  
      goto/16 :goto_0
  
+     .line 1241
      :pswitch_12
      move-object/from16 v0, p1
  
***************
*** 769,774 ****
--- 821,827 ----
  
      check-cast v41, Landroid/graphics/Point;
  
+     .line 1242
      .local v41, pt:Landroid/graphics/Point;
      move-object/from16 v0, p0
  
***************
*** 809,814 ****
--- 862,868 ----
  
      goto :goto_2
  
+     .line 1247
      .end local v41           #pt:Landroid/graphics/Point;
      :pswitch_13
      move-object/from16 v0, p1
***************
*** 819,824 ****
--- 873,879 ----
  
      check-cast v42, Landroid/graphics/Rect;
  
+     .line 1248
      .local v42, r:Landroid/graphics/Rect;
      move-object/from16 v0, p0
  
***************
*** 847,852 ****
--- 902,908 ----
  
      goto/16 :goto_0
  
+     .line 1255
      .end local v42           #r:Landroid/graphics/Rect;
      :pswitch_14
      move-object/from16 v0, p0
***************
*** 891,896 ****
--- 947,953 ----
  
      if-nez v2, :cond_7
  
+     .line 1258
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 908,913 ****
--- 965,971 ----
  
      goto/16 :goto_0
  
+     .line 1260
      :cond_7
      move-object/from16 v0, p0
  
***************
*** 928,933 ****
--- 986,992 ----
  
      goto/16 :goto_0
  
+     .line 1265
      :pswitch_15
      move-object/from16 v0, p0
  
***************
*** 937,942 ****
--- 996,1002 ----
  
      invoke-virtual {v2}, Landroid/webkit/WebViewCore;->stopLoading()V
  
+     .line 1266
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 952,957 ****
--- 1012,1018 ----
  
      goto/16 :goto_0
  
+     .line 1270
      :pswitch_16
      move-object/from16 v0, p0
  
***************
*** 973,978 ****
--- 1034,1040 ----
      #setter for: Landroid/webkit/WebViewCore$EventHub;->mSavedPriority:I
      invoke-static {v2, v4}, Landroid/webkit/WebViewCore$EventHub;->access$2602(Landroid/webkit/WebViewCore$EventHub;I)I
  
+     .line 1271
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 986,999 ****
--- 1048,1064 ----
  
      invoke-static {v2, v4}, Landroid/os/Process;->setThreadPriority(II)V
  
+     .line 1273
      invoke-static {}, Landroid/webkit/WebViewCore;->pauseTimers()V
  
+     .line 1274
      invoke-static {}, Landroid/webkit/JniUtil;->useChromiumHttpStack()Z
  
      move-result v2
  
      if-nez v2, :cond_8
  
+     .line 1275
      invoke-static {}, Landroid/webkit/WebViewWorker;->getHandler()Landroid/webkit/WebViewWorker;
  
      move-result-object v2
***************
*** 1004,1009 ****
--- 1069,1075 ----
  
      goto/16 :goto_0
  
+     .line 1278
      :cond_8
      move-object/from16 v0, p0
  
***************
*** 1016,1021 ****
--- 1082,1088 ----
  
      goto/16 :goto_0
  
+     .line 1283
      :pswitch_17
      move-object/from16 v0, p0
  
***************
*** 1037,1050 ****
--- 1104,1120 ----
  
      invoke-static {v2, v4}, Landroid/os/Process;->setThreadPriority(II)V
  
+     .line 1284
      invoke-static {}, Landroid/webkit/WebViewCore;->resumeTimers()V
  
+     .line 1285
      invoke-static {}, Landroid/webkit/JniUtil;->useChromiumHttpStack()Z
  
      move-result v2
  
      if-nez v2, :cond_0
  
+     .line 1286
      invoke-static {}, Landroid/webkit/WebViewWorker;->getHandler()Landroid/webkit/WebViewWorker;
  
      move-result-object v2
***************
*** 1055,1060 ****
--- 1125,1131 ----
  
      goto/16 :goto_0
  
+     .line 1292
      :pswitch_18
      move-object/from16 v0, p0
  
***************
*** 1067,1072 ****
--- 1138,1144 ----
  
      goto/16 :goto_0
  
+     .line 1296
      :pswitch_19
      move-object/from16 v0, p0
  
***************
*** 1079,1084 ****
--- 1151,1157 ----
  
      goto/16 :goto_0
  
+     .line 1300
      :pswitch_1a
      move-object/from16 v0, p0
  
***************
*** 1091,1096 ****
--- 1164,1170 ----
      #calls: Landroid/webkit/WebViewCore;->clearCache(Z)V
      invoke-static {v2, v4}, Landroid/webkit/WebViewCore;->access$3100(Landroid/webkit/WebViewCore;Z)V
  
+     .line 1301
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 1102,1112 ****
--- 1176,1188 ----
  
      goto/16 :goto_0
  
+     .line 1305
      :pswitch_1b
      sget-object v2, Landroid/webkit/BrowserFrame;->sJavaBridge:Landroid/webkit/JWebCoreJavaBridge;
  
      if-nez v2, :cond_9
  
+     .line 1306
      new-instance v2, Ljava/lang/IllegalStateException;
  
      const-string v4, "No WebView has been created in this process!"
***************
*** 1115,1120 ****
--- 1191,1197 ----
  
      throw v2
  
+     .line 1309
      :cond_9
      sget-object v4, Landroid/webkit/BrowserFrame;->sJavaBridge:Landroid/webkit/JWebCoreJavaBridge;
  
***************
*** 1138,1148 ****
--- 1215,1227 ----
  
      goto :goto_3
  
+     .line 1314
      :pswitch_1c
      sget-object v2, Landroid/webkit/BrowserFrame;->sJavaBridge:Landroid/webkit/JWebCoreJavaBridge;
  
      if-nez v2, :cond_b
  
+     .line 1315
      new-instance v2, Ljava/lang/IllegalStateException;
  
      const-string v4, "No WebView has been created in this process!"
***************
*** 1151,1156 ****
--- 1230,1236 ----
  
      throw v2
  
+     .line 1318
      :cond_b
      move-object/from16 v0, p1
  
***************
*** 1160,1169 ****
  
      check-cast v35, Ljava/util/Map;
  
      .local v35, map:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/String;>;"
      sget-object v6, Landroid/webkit/BrowserFrame;->sJavaBridge:Landroid/webkit/JWebCoreJavaBridge;
  
!     const-string v2, "type"
  
      move-object/from16 v0, v35
  
--- 1240,1250 ----
  
      check-cast v35, Ljava/util/Map;
  
+     .line 1319
      .local v35, map:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/String;>;"
      sget-object v6, Landroid/webkit/BrowserFrame;->sJavaBridge:Landroid/webkit/JWebCoreJavaBridge;
  
!     const-string/jumbo v2, "type"
  
      move-object/from16 v0, v35
  
***************
*** 1173,1179 ****
  
      check-cast v2, Ljava/lang/String;
  
!     const-string v4, "subtype"
  
      move-object/from16 v0, v35
  
--- 1254,1260 ----
  
      check-cast v2, Ljava/lang/String;
  
!     const-string/jumbo v4, "subtype"
  
      move-object/from16 v0, v35
  
***************
*** 1187,1192 ****
--- 1268,1274 ----
  
      goto/16 :goto_0
  
+     .line 1324
      .end local v35           #map:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/String;>;"
      :pswitch_1d
      move-object/from16 v0, p0
***************
*** 1216,1221 ****
--- 1298,1304 ----
  
      goto :goto_4
  
+     .line 1328
      :pswitch_1e
      move-object/from16 v0, p0
  
***************
*** 1249,1254 ****
--- 1332,1338 ----
  
      goto/16 :goto_0
  
+     .line 1333
      :pswitch_1f
      move-object/from16 v0, p1
  
***************
*** 1258,1263 ****
--- 1342,1348 ----
  
      check-cast v44, Landroid/webkit/WebViewCore$ReplaceTextData;
  
+     .line 1334
      .local v44, rep:Landroid/webkit/WebViewCore$ReplaceTextData;
      move-object/from16 v0, p0
  
***************
*** 1294,1299 ****
--- 1379,1385 ----
  
      goto/16 :goto_0
  
+     .line 1340
      .end local v44           #rep:Landroid/webkit/WebViewCore$ReplaceTextData;
      :pswitch_20
      move-object/from16 v0, p1
***************
*** 1304,1309 ****
--- 1390,1396 ----
  
      check-cast v32, Landroid/webkit/WebViewCore$JSKeyData;
  
+     .line 1341
      .local v32, jsData:Landroid/webkit/WebViewCore$JSKeyData;
      move-object/from16 v0, v32
  
***************
*** 1311,1331 ****
--- 1398,1422 ----
  
      move-object/from16 v27, v0
  
+     .line 1342
      .local v27, evt:Landroid/view/KeyEvent;
      invoke-virtual/range {v27 .. v27}, Landroid/view/KeyEvent;->getKeyCode()I
  
      move-result v7
  
+     .line 1343
      .local v7, keyCode:I
      invoke-virtual/range {v27 .. v27}, Landroid/view/KeyEvent;->getUnicodeChar()I
  
      move-result v8
  
+     .line 1344
      .local v8, keyValue:I
      move-object/from16 v0, p1
  
      iget v5, v0, Landroid/os/Message;->arg1:I
  
+     .line 1345
      .local v5, generation:I
      move-object/from16 v0, p0
  
***************
*** 1358,1363 ****
--- 1449,1455 ----
  
      goto/16 :goto_0
  
+     .line 1356
      .end local v5           #generation:I
      .end local v7           #keyCode:I
      .end local v8           #keyValue:I
***************
*** 1372,1377 ****
--- 1464,1470 ----
  
      check-cast v18, Landroid/webkit/WebViewCore$CursorData;
  
+     .line 1357
      .local v18, cDat:Landroid/webkit/WebViewCore$CursorData;
      move-object/from16 v0, p0
  
***************
*** 1388,1393 ****
--- 1481,1487 ----
  
      goto/16 :goto_0
  
+     .line 1362
      .end local v18           #cDat:Landroid/webkit/WebViewCore$CursorData;
      :pswitch_22
      invoke-static {}, Landroid/webkit/JniUtil;->useChromiumHttpStack()Z
***************
*** 1396,1407 ****
--- 1490,1503 ----
  
      if-eqz v2, :cond_d
  
+     .line 1365
      invoke-static {}, Landroid/webkit/SslCertLookupTable;->getInstance()Landroid/webkit/SslCertLookupTable;
  
      move-result-object v2
  
      invoke-virtual {v2}, Landroid/webkit/SslCertLookupTable;->clear()V
  
+     .line 1366
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 1413,1418 ****
--- 1509,1515 ----
  
      goto/16 :goto_0
  
+     .line 1368
      :cond_d
      move-object/from16 v0, p0
  
***************
*** 1433,1438 ****
--- 1530,1536 ----
  
      goto/16 :goto_0
  
+     .line 1373
      :pswitch_23
      move-object/from16 v0, p1
  
***************
*** 1442,1447 ****
--- 1540,1546 ----
  
      check-cast v48, Landroid/webkit/WebViewCore$TouchUpData;
  
+     .line 1374
      .local v48, touchUpData:Landroid/webkit/WebViewCore$TouchUpData;
      move-object/from16 v0, v48
  
***************
*** 1449,1454 ****
--- 1548,1554 ----
  
      if-eqz v2, :cond_e
  
+     .line 1375
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 1466,1471 ****
--- 1566,1572 ----
      #calls: Landroid/webkit/WebViewCore;->nativeScrollLayer(ILandroid/graphics/Rect;)V
      invoke-static {v2, v4, v6}, Landroid/webkit/WebViewCore;->access$3800(Landroid/webkit/WebViewCore;ILandroid/graphics/Rect;)V
  
+     .line 1378
      :cond_e
      move-object/from16 v0, p0
  
***************
*** 1498,1503 ****
--- 1599,1605 ----
  
      goto/16 :goto_0
  
+     .line 1384
      .end local v48           #touchUpData:Landroid/webkit/WebViewCore$TouchUpData;
      :pswitch_24
      move-object/from16 v0, p1
***************
*** 1508,1513 ****
--- 1610,1616 ----
  
      check-cast v47, Landroid/webkit/WebViewCore$TouchEventData;
  
+     .line 1385
      .local v47, ted:Landroid/webkit/WebViewCore$TouchEventData;
      move-object/from16 v0, v47
  
***************
*** 1515,1526 ****
--- 1618,1632 ----
  
      array-length v14, v2
  
+     .line 1386
      .local v14, count:I
      new-array v12, v14, [I
  
+     .line 1387
      .local v12, xArray:[I
      new-array v13, v14, [I
  
+     .line 1388
      .local v13, yArray:[I
      const/16 v17, 0x0
  
***************
*** 1530,1535 ****
--- 1636,1642 ----
  
      if-ge v0, v14, :cond_f
  
+     .line 1389
      move-object/from16 v0, v47
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$TouchEventData;->mPoints:[Landroid/graphics/Point;
***************
*** 1540,1545 ****
--- 1647,1653 ----
  
      aput v2, v12, v17
  
+     .line 1390
      move-object/from16 v0, v47
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$TouchEventData;->mPoints:[Landroid/graphics/Point;
***************
*** 1550,1559 ****
--- 1658,1669 ----
  
      aput v2, v13, v17
  
+     .line 1388
      add-int/lit8 v17, v17, 0x1
  
      goto :goto_5
  
+     .line 1392
      :cond_f
      move-object/from16 v0, v47
  
***************
*** 1561,1566 ****
--- 1671,1677 ----
  
      if-eqz v2, :cond_10
  
+     .line 1393
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 1578,1583 ****
--- 1689,1695 ----
      #calls: Landroid/webkit/WebViewCore;->nativeScrollLayer(ILandroid/graphics/Rect;)V
      invoke-static {v2, v4, v6}, Landroid/webkit/WebViewCore;->access$3800(Landroid/webkit/WebViewCore;ILandroid/graphics/Rect;)V
  
+     .line 1396
      :cond_10
      move-object/from16 v0, p0
  
***************
*** 1612,1617 ****
--- 1724,1730 ----
  
      iput-boolean v2, v0, Landroid/webkit/WebViewCore$TouchEventData;->mNativeResult:Z
  
+     .line 1398
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 1655,1660 ****
--- 1768,1774 ----
  
      goto :goto_6
  
+     .line 1408
      .end local v12           #xArray:[I
      .end local v13           #yArray:[I
      .end local v14           #count:I
***************
*** 1688,1693 ****
--- 1802,1808 ----
  
      goto :goto_7
  
+     .line 1412
      :pswitch_26
      move-object/from16 v0, p1
  
***************
*** 1697,1702 ****
--- 1812,1818 ----
  
      check-cast v32, Landroid/webkit/WebViewCore$JSInterfaceData;
  
+     .line 1413
      .local v32, jsData:Landroid/webkit/WebViewCore$JSInterfaceData;
      move-object/from16 v0, p0
  
***************
*** 1721,1726 ****
--- 1837,1843 ----
  
      goto/16 :goto_0
  
+     .line 1418
      .end local v32           #jsData:Landroid/webkit/WebViewCore$JSInterfaceData;
      :pswitch_27
      move-object/from16 v0, p1
***************
*** 1731,1736 ****
--- 1848,1854 ----
  
      check-cast v32, Landroid/webkit/WebViewCore$JSInterfaceData;
  
+     .line 1419
      .restart local v32       #jsData:Landroid/webkit/WebViewCore$JSInterfaceData;
      move-object/from16 v0, p0
  
***************
*** 1751,1756 ****
--- 1869,1875 ----
  
      goto/16 :goto_0
  
+     .line 1424
      .end local v32           #jsData:Landroid/webkit/WebViewCore$JSInterfaceData;
      :pswitch_28
      move-object/from16 v0, p0
***************
*** 1774,1779 ****
--- 1893,1899 ----
  
      goto/16 :goto_0
  
+     .line 1429
      :pswitch_29
      move-object/from16 v0, p0
  
***************
*** 1796,1801 ****
--- 1916,1922 ----
  
      goto/16 :goto_0
  
+     .line 1433
      :pswitch_2a
      move-object/from16 v0, p1
  
***************
*** 1805,1810 ****
--- 1926,1932 ----
  
      check-cast v28, Landroid/webkit/WebViewCore$CursorData;
  
+     .line 1434
      .local v28, focusData:Landroid/webkit/WebViewCore$CursorData;
      move-object/from16 v0, p0
  
***************
*** 1825,1830 ****
--- 1947,1953 ----
  
      goto/16 :goto_0
  
+     .line 1438
      .end local v28           #focusData:Landroid/webkit/WebViewCore$CursorData;
      :pswitch_2b
      move-object/from16 v0, p1
***************
*** 1835,1840 ****
--- 1958,1964 ----
  
      check-cast v23, Landroid/webkit/WebViewCore$CursorData;
  
+     .line 1439
      .local v23, cursorData:Landroid/webkit/WebViewCore$CursorData;
      move-object/from16 v0, p0
  
***************
*** 1859,1864 ****
--- 1983,1989 ----
  
      goto/16 :goto_0
  
+     .line 1444
      .end local v23           #cursorData:Landroid/webkit/WebViewCore$CursorData;
      :pswitch_2c
      move-object/from16 v0, p1
***************
*** 1869,1874 ****
--- 1994,2000 ----
  
      check-cast v19, Landroid/webkit/WebViewCore$CursorData;
  
+     .line 1445
      .local v19, cData:Landroid/webkit/WebViewCore$CursorData;
      move-object/from16 v0, p0
  
***************
*** 1895,1900 ****
--- 2021,2027 ----
      #calls: Landroid/webkit/WebViewCore;->nativeMoveMouseIfLatest(IIII)V
      invoke-static {v2, v4, v6, v9, v10}, Landroid/webkit/WebViewCore;->access$4400(Landroid/webkit/WebViewCore;IIII)V
  
+     .line 1448
      move-object/from16 v0, p1
  
      iget v2, v0, Landroid/os/Message;->arg1:I
***************
*** 1903,1908 ****
--- 2030,2036 ----
  
      if-ne v2, v4, :cond_0
  
+     .line 1449
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 1914,1919 ****
--- 2042,2048 ----
  
      goto/16 :goto_0
  
+     .line 1454
      .end local v19           #cData:Landroid/webkit/WebViewCore$CursorData;
      :pswitch_2d
      move-object/from16 v0, p1
***************
*** 1924,1935 ****
  
      check-cast v29, Landroid/os/Message;
  
      .local v29, hrefMsg:Landroid/os/Message;
      invoke-virtual/range {v29 .. v29}, Landroid/os/Message;->getData()Landroid/os/Bundle;
  
      move-result-object v2
  
!     const-string v4, "url"
  
      move-object/from16 v0, p0
  
--- 2053,2065 ----
  
      check-cast v29, Landroid/os/Message;
  
+     .line 1455
      .local v29, hrefMsg:Landroid/os/Message;
      invoke-virtual/range {v29 .. v29}, Landroid/os/Message;->getData()Landroid/os/Bundle;
  
      move-result-object v2
  
!     const-string/jumbo v4, "url"
  
      move-object/from16 v0, p0
  
***************
*** 1952,1962 ****
  
      invoke-virtual {v2, v4, v6}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
      invoke-virtual/range {v29 .. v29}, Landroid/os/Message;->getData()Landroid/os/Bundle;
  
      move-result-object v2
  
!     const-string v4, "title"
  
      move-object/from16 v0, p0
  
--- 2082,2093 ----
  
      invoke-virtual {v2, v4, v6}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1457
      invoke-virtual/range {v29 .. v29}, Landroid/os/Message;->getData()Landroid/os/Bundle;
  
      move-result-object v2
  
!     const-string/jumbo v4, "title"
  
      move-object/from16 v0, p0
  
***************
*** 1979,1989 ****
  
      invoke-virtual {v2, v4, v6}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
      invoke-virtual/range {v29 .. v29}, Landroid/os/Message;->getData()Landroid/os/Bundle;
  
      move-result-object v2
  
!     const-string v4, "src"
  
      move-object/from16 v0, p0
  
--- 2110,2121 ----
  
      invoke-virtual {v2, v4, v6}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1459
      invoke-virtual/range {v29 .. v29}, Landroid/os/Message;->getData()Landroid/os/Bundle;
  
      move-result-object v2
  
!     const-string/jumbo v4, "src"
  
      move-object/from16 v0, p0
  
***************
*** 2006,2015 ****
--- 2138,2149 ----
  
      invoke-virtual {v2, v4, v6}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1461
      invoke-virtual/range {v29 .. v29}, Landroid/os/Message;->sendToTarget()V
  
      goto/16 :goto_0
  
+     .line 1466
      .end local v29           #hrefMsg:Landroid/os/Message;
      :pswitch_2e
      move-object/from16 v0, p0
***************
*** 2021,2026 ****
--- 2155,2161 ----
      #calls: Landroid/webkit/WebViewCore;->nativeUpdateFrameCache()V
      invoke-static {v2}, Landroid/webkit/WebViewCore;->access$4900(Landroid/webkit/WebViewCore;)V
  
+     .line 1468
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 2034,2039 ****
--- 2169,2175 ----
  
      if-eqz v2, :cond_13
  
+     .line 1469
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 2047,2052 ****
--- 2183,2189 ----
  
      invoke-virtual {v2}, Landroid/webkit/WebView;->postInvalidate()V
  
+     .line 1471
      :cond_13
      move-object/from16 v0, p0
  
***************
*** 2059,2064 ****
--- 2196,2202 ----
  
      goto/16 :goto_0
  
+     .line 1475
      :pswitch_2f
      move-object/from16 v0, p1
  
***************
*** 2068,2073 ****
--- 2206,2212 ----
  
      check-cast v31, Landroid/os/Message;
  
+     .line 1476
      .local v31, imageResult:Landroid/os/Message;
      move-object/from16 v0, p0
  
***************
*** 2093,2107 ****
--- 2232,2249 ----
  
      iput v2, v0, Landroid/os/Message;->arg1:I
  
+     .line 1478
      invoke-virtual/range {v31 .. v31}, Landroid/os/Message;->sendToTarget()V
  
      goto/16 :goto_0
  
+     .line 1476
      :cond_14
      const/4 v2, 0x0
  
      goto :goto_8
  
+     .line 1482
      .end local v31           #imageResult:Landroid/os/Message;
      :pswitch_30
      move-object/from16 v0, p1
***************
*** 2112,2117 ****
--- 2254,2260 ----
  
      check-cast v26, Landroid/webkit/WebViewCore$TextSelectionData;
  
+     .line 1484
      .local v26, deleteSelectionData:Landroid/webkit/WebViewCore$TextSelectionData;
      move-object/from16 v0, p0
  
***************
*** 2136,2141 ****
--- 2279,2285 ----
  
      goto/16 :goto_0
  
+     .line 1489
      .end local v26           #deleteSelectionData:Landroid/webkit/WebViewCore$TextSelectionData;
      :pswitch_31
      move-object/from16 v0, p0
***************
*** 2157,2162 ****
--- 2301,2307 ----
  
      goto/16 :goto_0
  
+     .line 1493
      :pswitch_32
      move-object/from16 v0, p0
  
***************
*** 2177,2182 ****
--- 2322,2328 ----
  
      move-result-object v37
  
+     .line 1495
      .local v37, modifiedSelectionString:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 2203,2208 ****
--- 2349,2355 ----
  
      goto/16 :goto_0
  
+     .line 1500
      .end local v37           #modifiedSelectionString:Ljava/lang/String;
      :pswitch_33
      move-object/from16 v0, p1
***************
*** 2213,2218 ****
--- 2360,2366 ----
  
      check-cast v20, Landroid/util/SparseBooleanArray;
  
+     .line 1502
      .local v20, choices:Landroid/util/SparseBooleanArray;
      move-object/from16 v0, p1
  
***************
*** 2220,2225 ****
--- 2368,2374 ----
  
      move/from16 v22, v0
  
+     .line 1503
      .local v22, choicesSize:I
      move/from16 v0, v22
  
***************
*** 2227,2232 ****
--- 2376,2382 ----
  
      move-object/from16 v21, v0
  
+     .line 1504
      .local v21, choicesArray:[Z
      const/16 v17, 0x0
  
***************
*** 2238,2243 ****
--- 2388,2394 ----
  
      if-ge v0, v1, :cond_15
  
+     .line 1505
      move-object/from16 v0, v20
  
      move/from16 v1, v17
***************
*** 2248,2257 ****
--- 2399,2410 ----
  
      aput-boolean v2, v21, v17
  
+     .line 1504
      add-int/lit8 v17, v17, 0x1
  
      goto :goto_9
  
+     .line 1507
      :cond_15
      move-object/from16 v0, p0
  
***************
*** 2268,2273 ****
--- 2421,2427 ----
  
      goto/16 :goto_0
  
+     .line 1512
      .end local v17           #c:I
      .end local v20           #choices:Landroid/util/SparseBooleanArray;
      .end local v21           #choicesArray:[Z
***************
*** 2288,2293 ****
--- 2442,2448 ----
  
      goto/16 :goto_0
  
+     .line 1516
      :pswitch_35
      move-object/from16 v0, p0
  
***************
*** 2304,2309 ****
--- 2459,2465 ----
  
      goto/16 :goto_0
  
+     .line 1520
      :pswitch_36
      move-object/from16 v0, p0
  
***************
*** 2332,2337 ****
--- 2488,2494 ----
  
      goto :goto_a
  
+     .line 1524
      :pswitch_37
      move-object/from16 v0, p0
  
***************
*** 2360,2365 ****
--- 2517,2523 ----
  
      goto :goto_b
  
+     .line 1528
      :pswitch_38
      move-object/from16 v0, p0
  
***************
*** 2372,2377 ****
--- 2530,2536 ----
  
      goto/16 :goto_0
  
+     .line 1532
      :pswitch_39
      move-object/from16 v0, p0
  
***************
*** 2384,2389 ****
--- 2543,2549 ----
  
      goto/16 :goto_0
  
+     .line 1536
      :pswitch_3a
      move-object/from16 v0, p0
  
***************
*** 2402,2407 ****
--- 2562,2568 ----
  
      goto/16 :goto_0
  
+     .line 1540
      :pswitch_3b
      move-object/from16 v0, p0
  
***************
*** 2414,2419 ****
--- 2575,2581 ----
  
      goto/16 :goto_0
  
+     .line 1544
      :pswitch_3c
      move-object/from16 v0, p1
  
***************
*** 2423,2428 ****
--- 2585,2591 ----
  
      check-cast v45, Landroid/webkit/WebView$SaveWebArchiveMessage;
  
+     .line 1546
      .local v45, saveMessage:Landroid/webkit/WebView$SaveWebArchiveMessage;
      move-object/from16 v0, p0
  
***************
*** 2447,2452 ****
--- 2610,2616 ----
  
      iput-object v2, v0, Landroid/webkit/WebView$SaveWebArchiveMessage;->mResultFile:Ljava/lang/String;
  
+     .line 1548
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 2472,2477 ****
--- 2636,2642 ----
  
      goto/16 :goto_0
  
+     .line 1553
      .end local v45           #saveMessage:Landroid/webkit/WebView$SaveWebArchiveMessage;
      :pswitch_3d
      move-object/from16 v0, p1
***************
*** 2482,2487 ****
--- 2647,2653 ----
  
      check-cast v25, Landroid/webkit/WebViewCore$GeolocationPermissionsData;
  
+     .line 1555
      .local v25, data:Landroid/webkit/WebViewCore$GeolocationPermissionsData;
      move-object/from16 v0, p0
  
***************
*** 2506,2511 ****
--- 2672,2678 ----
  
      goto/16 :goto_0
  
+     .line 1560
      .end local v25           #data:Landroid/webkit/WebViewCore$GeolocationPermissionsData;
      :pswitch_3e
      move-object/from16 v0, p0
***************
*** 2521,2526 ****
--- 2688,2694 ----
      #calls: Landroid/webkit/WebViewCore;->nativeSplitContent(I)V
      invoke-static {v2, v4}, Landroid/webkit/WebViewCore;->access$6400(Landroid/webkit/WebViewCore;I)V
  
+     .line 1561
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 2544,2549 ****
--- 2712,2718 ----
  
      invoke-virtual {v2, v4, v6, v9}, Landroid/os/Handler;->obtainMessage(III)Landroid/os/Message;
  
+     .line 1563
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 2557,2562 ****
--- 2726,2732 ----
  
      goto/16 :goto_0
  
+     .line 1570
      :pswitch_3f
      move-object/from16 v0, p0
  
***************
*** 2569,2574 ****
--- 2739,2745 ----
  
      goto/16 :goto_0
  
+     .line 1574
      :pswitch_40
      move-object/from16 v0, p1
  
***************
*** 2580,2585 ****
--- 2751,2757 ----
  
      goto/16 :goto_0
  
+     .line 1578
      :pswitch_41
      move-object/from16 v0, p0
  
***************
*** 2600,2605 ****
--- 2772,2778 ----
  
      goto/16 :goto_0
  
+     .line 1582
      :pswitch_42
      move-object/from16 v0, p1
  
***************
*** 2609,2614 ****
--- 2782,2788 ----
  
      check-cast v38, Landroid/webkit/WebViewCore$MotionUpData;
  
+     .line 1583
      .local v38, motionUpData:Landroid/webkit/WebViewCore$MotionUpData;
      move-object/from16 v0, p0
  
***************
*** 2635,2640 ****
--- 2809,2815 ----
  
      if-nez v2, :cond_18
  
+     .line 1586
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 2644,2649 ****
--- 2819,2825 ----
      #calls: Landroid/webkit/WebViewCore;->nativeUpdateFrameCache()V
      invoke-static {v2}, Landroid/webkit/WebViewCore;->access$4900(Landroid/webkit/WebViewCore;)V
  
+     .line 1588
      :cond_18
      move-object/from16 v0, p0
  
***************
*** 2672,2677 ****
--- 2848,2854 ----
  
      move-result-object v36
  
+     .line 1591
      .local v36, message:Landroid/os/Message;
      move-object/from16 v0, p0
  
***************
*** 2692,2697 ****
--- 2869,2875 ----
  
      goto/16 :goto_0
  
+     .line 1597
      .end local v36           #message:Landroid/os/Message;
      .end local v38           #motionUpData:Landroid/webkit/WebViewCore$MotionUpData;
      :pswitch_43
***************
*** 2710,2715 ****
--- 2888,2894 ----
  
      goto/16 :goto_0
  
+     .line 1601
      :pswitch_44
      move-object/from16 v0, p0
  
***************
*** 2722,2727 ****
--- 2901,2907 ----
  
      goto/16 :goto_0
  
+     .line 1605
      :pswitch_45
      move-object/from16 v0, p0
  
***************
*** 2745,2755 ****
--- 2925,2937 ----
  
      goto/16 :goto_0
  
+     .line 1609
      :pswitch_46
      sget-object v2, Landroid/webkit/BrowserFrame;->sJavaBridge:Landroid/webkit/JWebCoreJavaBridge;
  
      if-nez v2, :cond_19
  
+     .line 1610
      new-instance v2, Ljava/lang/IllegalStateException;
  
      const-string v4, "No WebView has been created in this process!"
***************
*** 2758,2763 ****
--- 2940,2946 ----
  
      throw v2
  
+     .line 1613
      :cond_19
      sget-object v4, Landroid/webkit/BrowserFrame;->sJavaBridge:Landroid/webkit/JWebCoreJavaBridge;
  
***************
*** 2771,2776 ****
--- 2954,2960 ----
  
      goto/16 :goto_0
  
+     .line 1618
      :pswitch_47
      move-object/from16 v0, p1
  
***************
*** 2780,2785 ****
--- 2964,2970 ----
  
      check-cast v24, Landroid/webkit/WebViewCore$TouchHighlightData;
  
+     .line 1619
      .local v24, d:Landroid/webkit/WebViewCore$TouchHighlightData;
      move-object/from16 v0, v24
  
***************
*** 2787,2792 ****
--- 2972,2978 ----
  
      if-eqz v2, :cond_1a
  
+     .line 1620
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 2804,2809 ****
--- 2990,2996 ----
      #calls: Landroid/webkit/WebViewCore;->nativeScrollLayer(ILandroid/graphics/Rect;)V
      invoke-static {v2, v4, v6}, Landroid/webkit/WebViewCore;->access$3800(Landroid/webkit/WebViewCore;ILandroid/graphics/Rect;)V
  
+     .line 1623
      :cond_1a
      move-object/from16 v0, p0
  
***************
*** 2828,2833 ****
--- 3015,3021 ----
  
      move-result-object v43
  
+     .line 1625
      .local v43, rects:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/graphics/Rect;>;"
      move-object/from16 v0, p0
  
***************
*** 2854,2859 ****
--- 3042,3048 ----
  
      goto/16 :goto_0
  
+     .line 1631
      .end local v24           #d:Landroid/webkit/WebViewCore$TouchHighlightData;
      .end local v43           #rects:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/graphics/Rect;>;"
      :pswitch_48
***************
*** 2868,2873 ****
--- 3057,3063 ----
  
      goto/16 :goto_0
  
+     .line 1635
      :pswitch_49
      move-object/from16 v0, p0
  
***************
*** 2882,2887 ****
--- 3072,3078 ----
      #calls: Landroid/webkit/WebViewCore;->nativeAutoFillForm(I)V
      invoke-static {v2, v4}, Landroid/webkit/WebViewCore;->access$7400(Landroid/webkit/WebViewCore;I)V
  
+     .line 1636
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 2907,2912 ****
--- 3098,3104 ----
  
      goto/16 :goto_0
  
+     .line 1641
      :pswitch_4a
      move-object/from16 v0, p1
  
***************
*** 2916,2921 ****
--- 3108,3114 ----
  
      if-eqz v2, :cond_0
  
+     .line 1645
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/webkit/WebViewCore$EventHub$1;->this$1:Landroid/webkit/WebViewCore$EventHub;
***************
*** 2937,2942 ****
--- 3130,3136 ----
  
      goto/16 :goto_0
  
+     .line 1105
      :pswitch_data_0
      .packed-switch 0x60
          :pswitch_1
