*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$AutoFillData.smali	2012-03-23 19:37:10.616968683 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$AutoFillData.smali.original	2012-03-23 19:34:45.575323852 -0500
***************
*** 25,40 ****
--- 25,44 ----
      .locals 1
  
      .prologue
+     .line 818
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 819
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/webkit/WebViewCore$AutoFillData;->mQueryId:I
  
+     .line 820
      const-string v0, ""
  
      iput-object v0, p0, Landroid/webkit/WebViewCore$AutoFillData;->mPreview:Ljava/lang/String;
  
+     .line 821
      return-void
  .end method
  
***************
*** 44,55 ****
--- 48,63 ----
      .parameter "preview"
  
      .prologue
+     .line 823
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 824
      iput p1, p0, Landroid/webkit/WebViewCore$AutoFillData;->mQueryId:I
  
+     .line 825
      iput-object p2, p0, Landroid/webkit/WebViewCore$AutoFillData;->mPreview:Ljava/lang/String;
  
+     .line 826
      return-void
  .end method
  
***************
*** 59,64 ****
--- 67,73 ----
      .locals 1
  
      .prologue
+     .line 833
      iget-object v0, p0, Landroid/webkit/WebViewCore$AutoFillData;->mPreview:Ljava/lang/String;
  
      return-object v0
***************
*** 68,73 ****
--- 77,83 ----
      .locals 1
  
      .prologue
+     .line 829
      iget v0, p0, Landroid/webkit/WebViewCore$AutoFillData;->mQueryId:I
  
      return v0
