*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$BaseUrlData.smali	2012-03-23 19:37:10.247961953 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$BaseUrlData.smali.original	2012-03-23 19:34:45.574323833 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .locals 0
  
      .prologue
+     .line 732
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
