*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewWorker$CacheSaveData.smali	2012-03-23 19:37:09.988957230 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewWorker$CacheSaveData.smali.original	2012-03-23 19:34:45.572323797 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .locals 0
  
      .prologue
+     .line 58
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
