*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebView$InvokeListBox.smali	2012-03-23 19:37:10.499966548 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebView$InvokeListBox.smali.original	2012-03-23 19:34:45.575323852 -0500
***************
*** 46,74 ****
--- 46,81 ----
      .parameter "selection"
  
      .prologue
+     .line 9053
      iput-object p1, p0, Landroid/webkit/WebView$InvokeListBox;->this$0:Landroid/webkit/WebView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 9054
      iput p4, p0, Landroid/webkit/WebView$InvokeListBox;->mSelection:I
  
+     .line 9055
      const/4 v2, 0x0
  
      iput-boolean v2, p0, Landroid/webkit/WebView$InvokeListBox;->mMultiple:Z
  
+     .line 9057
      array-length v1, p2
  
+     .line 9058
      .local v1, length:I
      new-array v2, v1, [Landroid/webkit/WebView$InvokeListBox$Container;
  
      iput-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
+     .line 9059
      const/4 v0, 0x0
  
      .local v0, i:I
      :goto_0
      if-ge v0, v1, :cond_0
  
+     .line 9060
      iget-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
      new-instance v3, Landroid/webkit/WebView$InvokeListBox$Container;
***************
*** 79,84 ****
--- 86,92 ----
  
      aput-object v3, v2, v0
  
+     .line 9061
      iget-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
      aget-object v2, v2, v0
***************
*** 87,92 ****
--- 95,101 ----
  
      iput-object v3, v2, Landroid/webkit/WebView$InvokeListBox$Container;->mString:Ljava/lang/String;
  
+     .line 9062
      iget-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
      aget-object v2, v2, v0
***************
*** 95,110 ****
--- 104,122 ----
  
      iput v3, v2, Landroid/webkit/WebView$InvokeListBox$Container;->mEnabled:I
  
+     .line 9063
      iget-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
      aget-object v2, v2, v0
  
      iput v0, v2, Landroid/webkit/WebView$InvokeListBox$Container;->mId:I
  
+     .line 9059
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 9065
      :cond_0
      return-void
  .end method
***************
*** 118,123 ****
--- 130,136 ----
      .parameter "x4"
  
      .prologue
+     .line 8908
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/webkit/WebView$InvokeListBox;-><init>(Landroid/webkit/WebView;[Ljava/lang/String;[II)V
  
      return-void
***************
*** 131,159 ****
--- 144,179 ----
      .parameter "selected"
  
      .prologue
+     .line 9039
      iput-object p1, p0, Landroid/webkit/WebView$InvokeListBox;->this$0:Landroid/webkit/WebView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 9040
      const/4 v2, 0x1
  
      iput-boolean v2, p0, Landroid/webkit/WebView$InvokeListBox;->mMultiple:Z
  
+     .line 9041
      iput-object p4, p0, Landroid/webkit/WebView$InvokeListBox;->mSelectedArray:[I
  
+     .line 9043
      array-length v1, p2
  
+     .line 9044
      .local v1, length:I
      new-array v2, v1, [Landroid/webkit/WebView$InvokeListBox$Container;
  
      iput-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
+     .line 9045
      const/4 v0, 0x0
  
      .local v0, i:I
      :goto_0
      if-ge v0, v1, :cond_0
  
+     .line 9046
      iget-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
      new-instance v3, Landroid/webkit/WebView$InvokeListBox$Container;
***************
*** 164,169 ****
--- 184,190 ----
  
      aput-object v3, v2, v0
  
+     .line 9047
      iget-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
      aget-object v2, v2, v0
***************
*** 172,177 ****
--- 193,199 ----
  
      iput-object v3, v2, Landroid/webkit/WebView$InvokeListBox$Container;->mString:Ljava/lang/String;
  
+     .line 9048
      iget-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
      aget-object v2, v2, v0
***************
*** 180,195 ****
--- 202,220 ----
  
      iput v3, v2, Landroid/webkit/WebView$InvokeListBox$Container;->mEnabled:I
  
+     .line 9049
      iget-object v2, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
      aget-object v2, v2, v0
  
      iput v0, v2, Landroid/webkit/WebView$InvokeListBox$Container;->mId:I
  
+     .line 9045
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 9051
      :cond_0
      return-void
  .end method
***************
*** 203,208 ****
--- 228,234 ----
      .parameter "x4"
  
      .prologue
+     .line 8908
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/webkit/WebView$InvokeListBox;-><init>(Landroid/webkit/WebView;[Ljava/lang/String;[I[I)V
  
      return-void
***************
*** 213,218 ****
--- 239,245 ----
      .parameter "x0"
  
      .prologue
+     .line 8908
      iget-boolean v0, p0, Landroid/webkit/WebView$InvokeListBox;->mMultiple:Z
  
      return v0
***************
*** 223,228 ****
--- 250,256 ----
      .parameter "x0"
  
      .prologue
+     .line 8908
      iget-object v0, p0, Landroid/webkit/WebView$InvokeListBox;->mContainers:[Landroid/webkit/WebView$InvokeListBox$Container;
  
      return-object v0
***************
*** 236,241 ****
--- 264,270 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 9112
      iget-object v1, p0, Landroid/webkit/WebView$InvokeListBox;->this$0:Landroid/webkit/WebView;
  
      #getter for: Landroid/webkit/WebView;->mContext:Landroid/content/Context;
***************
*** 257,267 ****
--- 286,298 ----
  
      check-cast v4, Landroid/widget/ListView;
  
+     .line 9114
      .local v4, listView:Landroid/widget/ListView;
      new-instance v5, Landroid/webkit/WebView$InvokeListBox$MyArrayListAdapter;
  
      invoke-direct {v5, p0}, Landroid/webkit/WebView$InvokeListBox$MyArrayListAdapter;-><init>(Landroid/webkit/WebView$InvokeListBox;)V
  
+     .line 9115
      .local v5, adapter:Landroid/webkit/WebView$InvokeListBox$MyArrayListAdapter;
      new-instance v1, Landroid/app/AlertDialog$Builder;
  
***************
*** 286,296 ****
--- 317,329 ----
  
      move-result-object v6
  
+     .line 9119
      .local v6, b:Landroid/app/AlertDialog$Builder;
      iget-boolean v1, p0, Landroid/webkit/WebView$InvokeListBox;->mMultiple:Z
  
      if-eqz v1, :cond_0
  
+     .line 9120
      const v1, 0x104000a
  
      new-instance v3, Landroid/webkit/WebView$InvokeListBox$1;
***************
*** 299,304 ****
--- 332,338 ----
  
      invoke-virtual {v6, v1, v3}, Landroid/app/AlertDialog$Builder;->setPositiveButton(ILandroid/content/DialogInterface$OnClickListener;)Landroid/app/AlertDialog$Builder;
  
+     .line 9127
      const/high16 v1, 0x104
  
      new-instance v3, Landroid/webkit/WebView$InvokeListBox$2;
***************
*** 307,312 ****
--- 341,347 ----
  
      invoke-virtual {v6, v1, v3}, Landroid/app/AlertDialog$Builder;->setNegativeButton(ILandroid/content/DialogInterface$OnClickListener;)Landroid/app/AlertDialog$Builder;
  
+     .line 9134
      :cond_0
      iget-object v1, p0, Landroid/webkit/WebView$InvokeListBox;->this$0:Landroid/webkit/WebView;
  
***************
*** 317,326 ****
--- 352,364 ----
      #setter for: Landroid/webkit/WebView;->mListBoxDialog:Landroid/app/AlertDialog;
      invoke-static {v1, v3}, Landroid/webkit/WebView;->access$11102(Landroid/webkit/WebView;Landroid/app/AlertDialog;)Landroid/app/AlertDialog;
  
+     .line 9135
      invoke-virtual {v4, v5}, Landroid/widget/ListView;->setAdapter(Landroid/widget/ListAdapter;)V
  
+     .line 9136
      invoke-virtual {v4, v2}, Landroid/widget/ListView;->setFocusableInTouchMode(Z)V
  
+     .line 9143
      iget-boolean v1, p0, Landroid/webkit/WebView$InvokeListBox;->mMultiple:Z
  
      if-nez v1, :cond_1
***************
*** 330,347 ****
--- 368,389 ----
      :goto_0
      invoke-virtual {v4, v1}, Landroid/widget/ListView;->setTextFilterEnabled(Z)V
  
+     .line 9144
      iget-boolean v1, p0, Landroid/webkit/WebView$InvokeListBox;->mMultiple:Z
  
      if-eqz v1, :cond_2
  
+     .line 9145
      const/4 v1, 0x2
  
      invoke-virtual {v4, v1}, Landroid/widget/ListView;->setChoiceMode(I)V
  
+     .line 9146
      iget-object v1, p0, Landroid/webkit/WebView$InvokeListBox;->mSelectedArray:[I
  
      array-length v8, v1
  
+     .line 9147
      .local v8, length:I
      const/4 v7, 0x0
  
***************
*** 349,364 ****
--- 391,409 ----
      :goto_1
      if-ge v7, v8, :cond_3
  
+     .line 9148
      iget-object v1, p0, Landroid/webkit/WebView$InvokeListBox;->mSelectedArray:[I
  
      aget v1, v1, v7
  
      invoke-virtual {v4, v1, v2}, Landroid/widget/ListView;->setItemChecked(IZ)V
  
+     .line 9147
      add-int/lit8 v7, v7, 0x1
  
      goto :goto_1
  
+     .line 9143
      .end local v7           #i:I
      .end local v8           #length:I
      :cond_1
***************
*** 366,371 ****
--- 411,417 ----
  
      goto :goto_0
  
+     .line 9151
      :cond_2
      new-instance v1, Landroid/webkit/WebView$InvokeListBox$3;
  
***************
*** 373,394 ****
--- 419,445 ----
  
      invoke-virtual {v4, v1}, Landroid/widget/ListView;->setOnItemClickListener(Landroid/widget/AdapterView$OnItemClickListener;)V
  
+     .line 9162
      iget v1, p0, Landroid/webkit/WebView$InvokeListBox;->mSelection:I
  
      const/4 v3, -0x1
  
      if-eq v1, v3, :cond_3
  
+     .line 9163
      iget v1, p0, Landroid/webkit/WebView$InvokeListBox;->mSelection:I
  
      invoke-virtual {v4, v1}, Landroid/widget/ListView;->setSelection(I)V
  
+     .line 9164
      invoke-virtual {v4, v2}, Landroid/widget/ListView;->setChoiceMode(I)V
  
+     .line 9165
      iget v1, p0, Landroid/webkit/WebView$InvokeListBox;->mSelection:I
  
      invoke-virtual {v4, v1, v2}, Landroid/widget/ListView;->setItemChecked(IZ)V
  
+     .line 9166
      new-instance v0, Landroid/webkit/WebView$InvokeListBox$SingleDataSetObserver;
  
      iget v1, p0, Landroid/webkit/WebView$InvokeListBox;->mSelection:I
***************
*** 401,409 ****
--- 452,462 ----
  
      invoke-direct/range {v0 .. v5}, Landroid/webkit/WebView$InvokeListBox$SingleDataSetObserver;-><init>(Landroid/webkit/WebView$InvokeListBox;JLandroid/widget/ListView;Landroid/widget/Adapter;)V
  
+     .line 9168
      .local v0, observer:Landroid/database/DataSetObserver;
      invoke-virtual {v5, v0}, Landroid/webkit/WebView$InvokeListBox$MyArrayListAdapter;->registerDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 9171
      .end local v0           #observer:Landroid/database/DataSetObserver;
      :cond_3
      iget-object v1, p0, Landroid/webkit/WebView$InvokeListBox;->this$0:Landroid/webkit/WebView;
***************
*** 419,424 ****
--- 472,478 ----
  
      invoke-virtual {v1, v2}, Landroid/app/AlertDialog;->setOnCancelListener(Landroid/content/DialogInterface$OnCancelListener;)V
  
+     .line 9178
      iget-object v1, p0, Landroid/webkit/WebView$InvokeListBox;->this$0:Landroid/webkit/WebView;
  
      #getter for: Landroid/webkit/WebView;->mListBoxDialog:Landroid/app/AlertDialog;
***************
*** 428,432 ****
--- 482,487 ----
  
      invoke-virtual {v1}, Landroid/app/AlertDialog;->show()V
  
+     .line 9179
      return-void
  .end method
