*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$ShowRectData.smali	2012-03-23 19:37:09.824954241 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebViewCore$ShowRectData.smali.original	2012-03-23 19:34:45.571323779 -0500
***************
*** 41,46 ****
--- 41,47 ----
      .locals 0
  
      .prologue
+     .line 2778
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
