*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebView$1.smali	2012-03-23 19:37:12.657005883 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/WebView$1.smali.original	2012-03-23 19:34:45.584324016 -0500
***************
*** 37,42 ****
--- 37,43 ----
      .parameter
  
      .prologue
+     .line 1227
      iput-object p1, p0, Landroid/webkit/WebView$1;->this$0:Landroid/webkit/WebView;
  
      invoke-direct {p0}, Landroid/os/AsyncTask;-><init>()V
***************
*** 51,56 ****
--- 52,58 ----
      .parameter "x0"
  
      .prologue
+     .line 1227
      check-cast p1, [Ljava/lang/Void;
  
      .end local p1
***************
*** 77,86 ****
--- 79,90 ----
      .end annotation
  
      .prologue
+     .line 1231
      new-instance v1, Ljava/util/HashSet;
  
      invoke-direct {v1}, Ljava/util/HashSet;-><init>()V
  
+     .line 1232
      .local v1, installedPackages:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
      iget-object v5, p0, Landroid/webkit/WebView$1;->this$0:Landroid/webkit/WebView;
  
***************
*** 93,98 ****
--- 97,103 ----
  
      move-result-object v4
  
+     .line 1233
      .local v4, pm:Landroid/content/pm/PackageManager;
      invoke-static {}, Landroid/webkit/WebView;->access$400()Ljava/util/Set;
  
***************
*** 116,121 ****
--- 121,127 ----
  
      check-cast v2, Ljava/lang/String;
  
+     .line 1235
      .local v2, name:Ljava/lang/String;
      const/4 v5, 0x5
  
***************
*** 124,129 ****
--- 130,136 ----
  
      move-result-object v3
  
+     .line 1237
      .local v3, pInfo:Landroid/content/pm/PackageInfo;
      invoke-interface {v1, v2}, Ljava/util/Set;->add(Ljava/lang/Object;)Z
      :try_end_0
***************
*** 131,142 ****
--- 138,151 ----
  
      goto :goto_0
  
+     .line 1238
      .end local v3           #pInfo:Landroid/content/pm/PackageInfo;
      :catch_0
      move-exception v5
  
      goto :goto_0
  
+     .line 1242
      .end local v2           #name:Ljava/lang/String;
      :cond_0
      return-object v1
***************
*** 147,152 ****
--- 156,162 ----
      .parameter "x0"
  
      .prologue
+     .line 1227
      check-cast p1, Ljava/util/Set;
  
      .end local p1
***************
*** 169,174 ****
--- 179,185 ----
      .end annotation
  
      .prologue
+     .line 1248
      .local p1, installedPackages:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
      iget-object v0, p0, Landroid/webkit/WebView$1;->this$0:Landroid/webkit/WebView;
  
***************
*** 179,184 ****
--- 190,196 ----
  
      if-eqz v0, :cond_0
  
+     .line 1249
      iget-object v0, p0, Landroid/webkit/WebView$1;->this$0:Landroid/webkit/WebView;
  
      #getter for: Landroid/webkit/WebView;->mWebViewCore:Landroid/webkit/WebViewCore;
***************
*** 190,195 ****
--- 202,208 ----
  
      invoke-virtual {v0, v1, p1}, Landroid/webkit/WebViewCore;->sendMessage(ILjava/lang/Object;)V
  
+     .line 1251
      :cond_0
      return-void
  .end method
