*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/ZoomManager$FocusMovementQueue.smali	2012-03-23 19:37:05.145868918 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/webkit/ZoomManager$FocusMovementQueue.smali.original	2012-03-23 19:34:45.538323177 -0500
***************
*** 38,61 ****
--- 38,67 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 790
      iput-object p1, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->this$0:Landroid/webkit/ZoomManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 791
      const/4 v0, 0x5
  
      new-array v0, v0, [F
  
      iput-object v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mQueue:[F
  
+     .line 792
      iput v1, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSize:I
  
+     .line 793
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSum:F
  
+     .line 794
      iput v1, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mIndex:I
  
+     .line 795
      return-void
  .end method
  
***************
*** 64,69 ****
--- 70,76 ----
      .parameter "x0"
  
      .prologue
+     .line 783
      invoke-direct {p0}, Landroid/webkit/ZoomManager$FocusMovementQueue;->clear()V
  
      return-void
***************
*** 75,80 ****
--- 82,88 ----
      .parameter "x1"
  
      .prologue
+     .line 783
      invoke-direct {p0, p1}, Landroid/webkit/ZoomManager$FocusMovementQueue;->add(F)V
  
      return-void
***************
*** 85,90 ****
--- 93,99 ----
      .parameter "x0"
  
      .prologue
+     .line 783
      invoke-direct {p0}, Landroid/webkit/ZoomManager$FocusMovementQueue;->getSum()F
  
      move-result v0
***************
*** 97,120 ****
--- 106,133 ----
      .parameter "focusDelta"
  
      .prologue
+     .line 807
      iget v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSum:F
  
      add-float/2addr v0, p1
  
      iput v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSum:F
  
+     .line 808
      iget v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSize:I
  
      const/4 v1, 0x5
  
      if-ge v0, v1, :cond_0
  
+     .line 809
      iget v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSize:I
  
      add-int/lit8 v0, v0, 0x1
  
      iput v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSize:I
  
+     .line 813
      :goto_0
      iget-object v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mQueue:[F
  
***************
*** 122,127 ****
--- 135,141 ----
  
      aput p1, v0, v1
  
+     .line 814
      iget v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mIndex:I
  
      add-int/lit8 v0, v0, 0x1
***************
*** 130,137 ****
--- 144,153 ----
  
      iput v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mIndex:I
  
+     .line 815
      return-void
  
+     .line 811
      :cond_0
      iget v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSum:F
  
***************
*** 156,167 ****
--- 172,187 ----
  
      const/4 v2, 0x0
  
+     .line 798
      iput v1, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSize:I
  
+     .line 799
      iput v2, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSum:F
  
+     .line 800
      iput v1, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mIndex:I
  
+     .line 801
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 170,183 ****
--- 190,206 ----
  
      if-ge v0, v1, :cond_0
  
+     .line 802
      iget-object v1, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mQueue:[F
  
      aput v2, v1, v0
  
+     .line 801
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 804
      :cond_0
      return-void
  .end method
***************
*** 186,191 ****
--- 209,215 ----
      .locals 1
  
      .prologue
+     .line 818
      iget v0, p0, Landroid/webkit/ZoomManager$FocusMovementQueue;->mSum:F
  
      return v0
