*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothAdapter$1.smali	2012-03-23 19:35:23.332012344 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothAdapter$1.smali.original	2012-03-23 19:34:44.541304997 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .parameter "x0"
  
      .prologue
+     .line 1021
      iput-object p1, p0, Landroid/bluetooth/BluetoothAdapter$1;->this$0:Landroid/bluetooth/BluetoothAdapter;
  
      invoke-direct {p0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
***************
*** 39,46 ****
--- 40,49 ----
      .parameter "msg"
  
      .prologue
+     .line 1024
      iget v1, p1, Landroid/os/Message;->what:I
  
+     .line 1028
      .local v1, handle:I
      :try_start_0
      iget-object v2, p0, Landroid/bluetooth/BluetoothAdapter$1;->this$0:Landroid/bluetooth/BluetoothAdapter;
***************
*** 54,62 ****
--- 57,67 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1030
      :goto_0
      return-void
  
+     .line 1029
      :catch_0
      move-exception v0
  
