*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothHealth$BluetoothHealthCallbackWrapper.smali	2012-03-23 19:35:24.446032658 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothHealth$BluetoothHealthCallbackWrapper.smali.original	2012-03-23 19:34:44.557305289 -0500
***************
*** 24,33 ****
--- 24,36 ----
      .parameter "callback"
  
      .prologue
+     .line 393
      invoke-direct {p0}, Landroid/bluetooth/IBluetoothHealthCallback$Stub;-><init>()V
  
+     .line 394
      iput-object p1, p0, Landroid/bluetooth/BluetoothHealth$BluetoothHealthCallbackWrapper;->mCallback:Landroid/bluetooth/BluetoothHealthCallback;
  
+     .line 395
      return-void
  .end method
  
***************
*** 39,48 ****
--- 42,53 ----
      .parameter "status"
  
      .prologue
+     .line 400
      iget-object v0, p0, Landroid/bluetooth/BluetoothHealth$BluetoothHealthCallbackWrapper;->mCallback:Landroid/bluetooth/BluetoothHealthCallback;
  
      invoke-virtual {v0, p1, p2}, Landroid/bluetooth/BluetoothHealthCallback;->onHealthAppConfigurationStatusChange(Landroid/bluetooth/BluetoothHealthAppConfiguration;I)V
  
+     .line 401
      return-void
  .end method
  
***************
*** 56,61 ****
--- 61,67 ----
      .parameter "channelId"
  
      .prologue
+     .line 407
      iget-object v0, p0, Landroid/bluetooth/BluetoothHealth$BluetoothHealthCallbackWrapper;->mCallback:Landroid/bluetooth/BluetoothHealthCallback;
  
      move-object v1, p1
***************
*** 72,76 ****
--- 78,83 ----
  
      invoke-virtual/range {v0 .. v6}, Landroid/bluetooth/BluetoothHealthCallback;->onHealthChannelStateChange(Landroid/bluetooth/BluetoothHealthAppConfiguration;Landroid/bluetooth/BluetoothDevice;IILandroid/os/ParcelFileDescriptor;I)V
  
+     .line 409
      return-void
  .end method
