*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothAdapter$StateChangeCallbackWrapper.smali	2012-03-23 19:35:23.688018835 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothAdapter$StateChangeCallbackWrapper.smali.original	2012-03-23 19:34:44.547305106 -0500
***************
*** 27,38 ****
--- 27,41 ----
      .parameter "callback"
  
      .prologue
+     .line 1265
      iput-object p1, p0, Landroid/bluetooth/BluetoothAdapter$StateChangeCallbackWrapper;->this$0:Landroid/bluetooth/BluetoothAdapter;
  
      invoke-direct {p0}, Landroid/bluetooth/IBluetoothStateChangeCallback$Stub;-><init>()V
  
+     .line 1266
      iput-object p2, p0, Landroid/bluetooth/BluetoothAdapter$StateChangeCallbackWrapper;->mCallback:Landroid/bluetooth/BluetoothAdapter$BluetoothStateChangeCallback;
  
+     .line 1267
      return-void
  .end method
  
***************
*** 43,51 ****
--- 46,56 ----
      .parameter "on"
  
      .prologue
+     .line 1271
      iget-object v0, p0, Landroid/bluetooth/BluetoothAdapter$StateChangeCallbackWrapper;->mCallback:Landroid/bluetooth/BluetoothAdapter$BluetoothStateChangeCallback;
  
      invoke-interface {v0, p1}, Landroid/bluetooth/BluetoothAdapter$BluetoothStateChangeCallback;->onBluetoothStateChange(Z)V
  
+     .line 1272
      return-void
  .end method
