*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothDeviceProfileState$IncomingA2dp.smali	2012-03-23 19:35:23.240010667 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothDeviceProfileState$IncomingA2dp.smali.original	2012-03-23 19:34:44.540304979 -0500
***************
*** 28,37 ****
--- 28,39 ----
      .parameter
  
      .prologue
+     .line 767
      iput-object p1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
  
+     .line 768
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->mStatus:Z
***************
*** 45,50 ****
--- 47,53 ----
      .parameter "x1"
  
      .prologue
+     .line 767
      invoke-direct {p0, p1}, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;-><init>(Landroid/bluetooth/BluetoothDeviceProfileState;)V
  
      return-void
***************
*** 56,61 ****
--- 59,65 ----
      .locals 3
  
      .prologue
+     .line 773
      const-string v0, "BluetoothDeviceProfileState"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 87,92 ****
--- 91,97 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 774
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->getCurrentMessage()Landroid/os/Message;
***************
*** 98,103 ****
--- 103,109 ----
  
      iput v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->mCommand:I
  
+     .line 775
      iget v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->mCommand:I
  
      const/4 v1, 0x4
***************
*** 110,115 ****
--- 116,122 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 777
      const-string v0, "BluetoothDeviceProfileState"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 134,139 ****
--- 141,147 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 779
      :cond_0
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 145,160 ****
--- 153,171 ----
  
      iput-boolean v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->mStatus:Z
  
+     .line 780
      iget-boolean v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->mStatus:Z
  
      if-nez v0, :cond_1
  
+     .line 781
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      const/16 v1, 0x66
  
      invoke-virtual {v0, v1}, Landroid/bluetooth/BluetoothDeviceProfileState;->sendMessage(I)V
  
+     .line 782
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mService:Landroid/server/BluetoothService;
***************
*** 168,173 ****
--- 179,185 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/server/BluetoothService;->sendProfileStateMessage(II)V
  
+     .line 785
      :cond_1
      return-void
  .end method
***************
*** 183,188 ****
--- 195,201 ----
  
      const/4 v3, 0x1
  
+     .line 789
      iget-object v4, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 208,220 ****
--- 221,236 ----
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->log(Ljava/lang/String;)V
      invoke-static {v4, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2100(Landroid/bluetooth/BluetoothDeviceProfileState;Ljava/lang/String;)V
  
+     .line 790
      iget v4, p1, Landroid/os/Message;->what:I
  
      sparse-switch v4, :sswitch_data_0
  
+     .line 859
      :goto_0
      return v2
  
+     .line 792
      :sswitch_0
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 226,233 ****
--- 242,251 ----
      :sswitch_1
      move v2, v3
  
+     .line 859
      goto :goto_0
  
+     .line 796
      :sswitch_2
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 236,263 ****
--- 254,288 ----
  
      goto :goto_1
  
+     .line 802
      :sswitch_3
      iget v0, p1, Landroid/os/Message;->arg1:I
  
+     .line 803
      .local v0, val:I
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mConnectionAccessReplyReceived:Z
      invoke-static {v2, v3}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$6602(Landroid/bluetooth/BluetoothDeviceProfileState;Z)Z
  
+     .line 804
      const/4 v1, 0x0
  
+     .line 805
      .local v1, value:Z
      if-ne v0, v3, :cond_1
  
+     .line 806
      const/4 v1, 0x1
  
+     .line 808
      :cond_1
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->setTrust(I)V
      invoke-static {v2, v0}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$800(Landroid/bluetooth/BluetoothDeviceProfileState;I)V
  
+     .line 809
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->handleIncomingConnection(IZ)Z
***************
*** 265,270 ****
--- 290,296 ----
  
      goto :goto_1
  
+     .line 814
      .end local v0           #val:I
      .end local v1           #value:Z
      :sswitch_4
***************
*** 277,292 ****
--- 303,321 ----
  
      if-nez v4, :cond_0
  
+     .line 815
      iget-object v4, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->handleIncomingConnection(IZ)Z
      invoke-static {v4, v7, v2}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$6700(Landroid/bluetooth/BluetoothDeviceProfileState;IZ)Z
  
+     .line 816
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->sendConnectionAccessRemovalIntent()V
      invoke-static {v2}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$6800(Landroid/bluetooth/BluetoothDeviceProfileState;)V
  
+     .line 817
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      const/16 v4, 0x66
***************
*** 295,300 ****
--- 324,330 ----
  
      goto :goto_1
  
+     .line 822
      :sswitch_5
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 303,308 ****
--- 333,339 ----
  
      goto :goto_1
  
+     .line 825
      :sswitch_6
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 311,316 ****
--- 342,348 ----
  
      goto :goto_1
  
+     .line 835
      :sswitch_7
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 319,324 ****
--- 351,357 ----
  
      goto :goto_1
  
+     .line 842
      :sswitch_8
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 327,332 ****
--- 360,366 ----
  
      goto :goto_1
  
+     .line 851
      :sswitch_9
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 335,340 ****
--- 369,375 ----
  
      goto :goto_1
  
+     .line 854
      :sswitch_a
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 350,355 ****
--- 385,391 ----
  
      goto :goto_1
  
+     .line 790
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_0
