*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothClass$Device$Major.smali	2012-03-23 19:35:23.583016921 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothClass$Device$Major.smali.original	2012-03-23 19:34:44.545305069 -0500
***************
*** 45,50 ****
--- 45,51 ----
      .locals 0
  
      .prologue
+     .line 150
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
