*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothAssignedNumbers.smali	2012-03-23 19:35:24.111026550 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothAssignedNumbers.smali.original	2012-03-23 19:34:44.552305197 -0500
***************
*** 204,210 ****
--- 204,212 ----
      .locals 0
  
      .prologue
+     .line 518
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 519
      return-void
  .end method
