*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothClass$Device.smali	2012-03-23 19:35:24.354030980 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothClass$Device.smali.original	2012-03-23 19:34:44.556305271 -0500
***************
*** 137,143 ****
--- 137,145 ----
      .locals 0
  
      .prologue
+     .line 143
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 150
      return-void
  .end method
