*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothDeviceProfileState$OutgoingHid.smali	2012-03-23 19:35:23.842021644 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothDeviceProfileState$OutgoingHid.smali.original	2012-03-23 19:34:44.549305143 -0500
***************
*** 28,37 ****
--- 28,39 ----
      .parameter
  
      .prologue
+     .line 864
      iput-object p1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
  
+     .line 865
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->mStatus:Z
***************
*** 45,50 ****
--- 47,53 ----
      .parameter "x1"
  
      .prologue
+     .line 864
      invoke-direct {p0, p1}, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;-><init>(Landroid/bluetooth/BluetoothDeviceProfileState;)V
  
      return-void
***************
*** 56,61 ****
--- 59,65 ----
      .locals 3
  
      .prologue
+     .line 870
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 88,93 ****
--- 92,98 ----
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->log(Ljava/lang/String;)V
      invoke-static {v0, v1}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2100(Landroid/bluetooth/BluetoothDeviceProfileState;Ljava/lang/String;)V
  
+     .line 871
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->getCurrentMessage()Landroid/os/Message;
***************
*** 99,104 ****
--- 104,110 ----
  
      iput v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->mCommand:I
  
+     .line 872
      iget v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->mCommand:I
  
      const/4 v1, 0x5
***************
*** 111,116 ****
--- 117,123 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 874
      const-string v0, "BluetoothDeviceProfileState"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 135,140 ****
--- 142,148 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 876
      :cond_0
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 146,151 ****
--- 154,160 ----
  
      iput-boolean v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->mStatus:Z
  
+     .line 877
      iget-boolean v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->mStatus:Z
  
      if-nez v0, :cond_1
***************
*** 156,161 ****
--- 165,171 ----
  
      invoke-virtual {v0, v1}, Landroid/bluetooth/BluetoothDeviceProfileState;->sendMessage(I)V
  
+     .line 878
      :cond_1
      return-void
  .end method
***************
*** 165,170 ****
--- 175,181 ----
      .parameter "message"
  
      .prologue
+     .line 882
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 190,215 ****
--- 201,232 ----
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->log(Ljava/lang/String;)V
      invoke-static {v1, v2}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2100(Landroid/bluetooth/BluetoothDeviceProfileState;Ljava/lang/String;)V
  
+     .line 883
      new-instance v0, Landroid/os/Message;
  
      invoke-direct {v0}, Landroid/os/Message;-><init>()V
  
+     .line 884
      .local v0, deferMsg:Landroid/os/Message;
      iget v1, p1, Landroid/os/Message;->what:I
  
      sparse-switch v1, :sswitch_data_0
  
+     .line 939
      const/4 v1, 0x0
  
+     .line 941
      :goto_0
      return v1
  
+     .line 892
      :sswitch_0
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->deferMessage(Landroid/os/Message;)V
      invoke-static {v1, p1}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$10200(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/os/Message;)V
  
+     .line 941
      :cond_0
      :goto_1
      :sswitch_1
***************
*** 217,222 ****
--- 234,240 ----
  
      goto :goto_0
  
+     .line 896
      :sswitch_2
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 232,237 ****
--- 250,256 ----
  
      goto :goto_1
  
+     .line 899
      :sswitch_3
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 245,258 ****
--- 264,280 ----
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->transitionTo(Lcom/android/internal/util/IState;)V
      invoke-static {v1, v2}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$10400(Landroid/bluetooth/BluetoothDeviceProfileState;Lcom/android/internal/util/IState;)V
  
+     .line 906
      iget-boolean v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->mStatus:Z
  
      if-eqz v1, :cond_0
  
+     .line 907
      iget v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->mCommand:I
  
      iput v1, v0, Landroid/os/Message;->what:I
  
+     .line 908
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->deferMessage(Landroid/os/Message;)V
***************
*** 260,265 ****
--- 282,288 ----
  
      goto :goto_1
  
+     .line 913
      :sswitch_4
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 275,289 ****
--- 298,315 ----
  
      goto :goto_1
  
+     .line 922
      :sswitch_5
      iget-boolean v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->mStatus:Z
  
      if-eqz v1, :cond_0
  
+     .line 923
      iget v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->mCommand:I
  
      iput v1, v0, Landroid/os/Message;->what:I
  
+     .line 924
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->deferMessage(Landroid/os/Message;)V
***************
*** 291,296 ****
--- 317,323 ----
  
      goto :goto_1
  
+     .line 933
      :sswitch_6
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 299,304 ****
--- 326,332 ----
  
      goto :goto_1
  
+     .line 936
      :sswitch_7
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 314,319 ****
--- 342,348 ----
  
      goto :goto_1
  
+     .line 884
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_0
