*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothSocket$SocketState.smali	2012-03-23 19:35:24.311030196 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothSocket$SocketState.smali.original	2012-03-23 19:34:44.555305252 -0500
***************
*** 44,49 ****
--- 44,50 ----
  
      const/4 v2, 0x0
  
+     .line 98
      new-instance v0, Landroid/bluetooth/BluetoothSocket$SocketState;
  
      const-string v1, "INIT"
***************
*** 52,57 ****
--- 53,59 ----
  
      sput-object v0, Landroid/bluetooth/BluetoothSocket$SocketState;->INIT:Landroid/bluetooth/BluetoothSocket$SocketState;
  
+     .line 99
      new-instance v0, Landroid/bluetooth/BluetoothSocket$SocketState;
  
      const-string v1, "CONNECTED"
***************
*** 60,65 ****
--- 62,68 ----
  
      sput-object v0, Landroid/bluetooth/BluetoothSocket$SocketState;->CONNECTED:Landroid/bluetooth/BluetoothSocket$SocketState;
  
+     .line 100
      new-instance v0, Landroid/bluetooth/BluetoothSocket$SocketState;
  
      const-string v1, "CLOSED"
***************
*** 68,73 ****
--- 71,77 ----
  
      sput-object v0, Landroid/bluetooth/BluetoothSocket$SocketState;->CLOSED:Landroid/bluetooth/BluetoothSocket$SocketState;
  
+     .line 97
      const/4 v0, 0x3
  
      new-array v0, v0, [Landroid/bluetooth/BluetoothSocket$SocketState;
***************
*** 100,105 ****
--- 104,110 ----
      .end annotation
  
      .prologue
+     .line 97
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 110,115 ****
--- 115,121 ----
      .parameter "name"
  
      .prologue
+     .line 97
      const-class v0, Landroid/bluetooth/BluetoothSocket$SocketState;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 125,130 ****
--- 131,137 ----
      .locals 1
  
      .prologue
+     .line 97
      sget-object v0, Landroid/bluetooth/BluetoothSocket$SocketState;->$VALUES:[Landroid/bluetooth/BluetoothSocket$SocketState;
  
      invoke-virtual {v0}, [Landroid/bluetooth/BluetoothSocket$SocketState;->clone()Ljava/lang/Object;
