*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothHealthAppConfiguration.smali	2012-03-23 19:35:23.761020168 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothHealthAppConfiguration.smali.original	2012-03-23 19:34:44.548305125 -0500
***************
*** 34,39 ****
--- 34,40 ----
      .locals 1
  
      .prologue
+     .line 141
      new-instance v0, Landroid/bluetooth/BluetoothHealthAppConfiguration$1;
  
      invoke-direct {v0}, Landroid/bluetooth/BluetoothHealthAppConfiguration$1;-><init>()V
***************
*** 49,68 ****
--- 50,75 ----
      .parameter "dataType"
  
      .prologue
+     .line 43
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 44
      iput-object p1, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mName:Ljava/lang/String;
  
+     .line 45
      iput p2, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mDataType:I
  
+     .line 46
      const/4 v0, 0x2
  
      iput v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mRole:I
  
+     .line 47
      const/16 v0, 0xc
  
      iput v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mChannelType:I
  
+     .line 48
      return-void
  .end method
  
***************
*** 74,89 ****
--- 81,102 ----
      .parameter "channelType"
  
      .prologue
+     .line 60
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 61
      iput-object p1, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mName:Ljava/lang/String;
  
+     .line 62
      iput p2, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mDataType:I
  
+     .line 63
      iput p3, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mRole:I
  
+     .line 64
      iput p4, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mChannelType:I
  
+     .line 65
      return-void
  .end method
  
***************
*** 93,98 ****
--- 106,112 ----
      .locals 1
  
      .prologue
+     .line 98
      const/4 v0, 0x0
  
      return v0
***************
*** 105,118 ****
--- 119,135 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 69
      instance-of v2, p1, Landroid/bluetooth/BluetoothHealthAppConfiguration;
  
      if-eqz v2, :cond_0
  
      move-object v0, p1
  
+     .line 70
      check-cast v0, Landroid/bluetooth/BluetoothHealthAppConfiguration;
  
+     .line 72
      .local v0, config:Landroid/bluetooth/BluetoothHealthAppConfiguration;
      iget-object v2, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mName:Ljava/lang/String;
  
***************
*** 152,157 ****
--- 169,175 ----
  
      const/4 v1, 0x1
  
+     .line 77
      .end local v0           #config:Landroid/bluetooth/BluetoothHealthAppConfiguration;
      :cond_0
      return v1
***************
*** 161,166 ****
--- 179,185 ----
      .locals 1
  
      .prologue
+     .line 138
      iget v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mChannelType:I
  
      return v0
***************
*** 170,175 ****
--- 189,195 ----
      .locals 1
  
      .prologue
+     .line 107
      iget v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mDataType:I
  
      return v0
***************
*** 179,184 ****
--- 199,205 ----
      .locals 1
  
      .prologue
+     .line 116
      iget-object v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mName:Ljava/lang/String;
  
      return-object v0
***************
*** 188,193 ****
--- 209,215 ----
      .locals 1
  
      .prologue
+     .line 126
      iget v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mRole:I
  
      return v0
***************
*** 197,204 ****
--- 219,228 ----
      .locals 3
  
      .prologue
+     .line 82
      const/16 v0, 0x11
  
+     .line 83
      .local v0, result:I
      iget-object v1, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mName:Ljava/lang/String;
  
***************
*** 213,238 ****
--- 237,267 ----
      :goto_0
      add-int/lit16 v0, v1, 0x20f
  
+     .line 84
      mul-int/lit8 v1, v0, 0x1f
  
      iget v2, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mDataType:I
  
      add-int v0, v1, v2
  
+     .line 85
      mul-int/lit8 v1, v0, 0x1f
  
      iget v2, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mRole:I
  
      add-int v0, v1, v2
  
+     .line 86
      mul-int/lit8 v1, v0, 0x1f
  
      iget v2, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mChannelType:I
  
      add-int v0, v1, v2
  
+     .line 87
      return v0
  
+     .line 83
      :cond_0
      const/4 v1, 0x0
  
***************
*** 243,248 ****
--- 272,278 ----
      .locals 2
  
      .prologue
+     .line 92
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 314,334 ****
--- 344,369 ----
      .parameter "flags"
  
      .prologue
+     .line 160
      iget-object v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mName:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 161
      iget v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mDataType:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 162
      iget v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mRole:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 163
      iget v0, p0, Landroid/bluetooth/BluetoothHealthAppConfiguration;->mChannelType:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 164
      return-void
  .end method
