*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothDeviceProfileState$BondedDevice.smali	2012-03-23 19:35:24.805039205 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/bluetooth/BluetoothDeviceProfileState$BondedDevice.smali.original	2012-03-23 19:34:44.564305416 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 307
      iput-object p1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 307
      invoke-direct {p0, p1}, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;-><init>(Landroid/bluetooth/BluetoothDeviceProfileState;)V
  
      return-void
***************
*** 48,53 ****
--- 50,56 ----
      .locals 4
  
      .prologue
+     .line 310
      const-string v1, "BluetoothDeviceProfileState"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 79,88 ****
--- 82,93 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 311
      new-instance v0, Landroid/os/Message;
  
      invoke-direct {v0}, Landroid/os/Message;-><init>()V
  
+     .line 312
      .local v0, m:Landroid/os/Message;
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 93,103 ****
--- 98,110 ----
  
      invoke-virtual {v0, v1}, Landroid/os/Message;->copyFrom(Landroid/os/Message;)V
  
+     .line 313
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->sendMessageAtFrontOfQueue(Landroid/os/Message;)V
      invoke-static {v1, v0}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2000(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/os/Message;)V
  
+     .line 314
      return-void
  .end method
  
***************
*** 116,121 ****
--- 123,129 ----
  
      const/4 v5, 0x0
  
+     .line 317
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 141,153 ****
--- 149,164 ----
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->log(Ljava/lang/String;)V
      invoke-static {v2, v3}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2100(Landroid/bluetooth/BluetoothDeviceProfileState;Ljava/lang/String;)V
  
+     .line 318
      iget v2, p1, Landroid/os/Message;->what:I
  
      sparse-switch v2, :sswitch_data_0
  
+     .line 442
      :goto_0
      return v0
  
+     .line 321
      :sswitch_0
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 166,173 ****
--- 177,186 ----
      :sswitch_1
      move v0, v1
  
+     .line 442
      goto :goto_0
  
+     .line 324
      :sswitch_2
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 183,188 ****
--- 196,202 ----
  
      goto :goto_1
  
+     .line 327
      :sswitch_3
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 198,203 ****
--- 212,218 ----
  
      goto :goto_1
  
+     .line 331
      :sswitch_4
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 213,218 ****
--- 228,234 ----
  
      goto :goto_1
  
+     .line 335
      :sswitch_5
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 228,233 ****
--- 244,250 ----
  
      goto :goto_1
  
+     .line 339
      :sswitch_6
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 243,248 ****
--- 260,266 ----
  
      goto :goto_1
  
+     .line 343
      :sswitch_7
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 258,263 ****
--- 276,282 ----
  
      goto :goto_1
  
+     .line 346
      :sswitch_8
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 267,272 ****
--- 286,292 ----
  
      goto :goto_1
  
+     .line 349
      :sswitch_9
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 277,288 ****
--- 297,310 ----
  
      if-eqz v0, :cond_1
  
+     .line 350
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      const/16 v2, 0x32
  
      invoke-virtual {v0, v2}, Landroid/bluetooth/BluetoothDeviceProfileState;->sendMessage(I)V
  
+     .line 351
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->deferMessage(Landroid/os/Message;)V
***************
*** 290,295 ****
--- 312,318 ----
  
      goto :goto_1
  
+     .line 353
      :cond_1
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 300,311 ****
--- 323,336 ----
  
      if-eqz v0, :cond_2
  
+     .line 354
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      const/16 v2, 0x34
  
      invoke-virtual {v0, v2}, Landroid/bluetooth/BluetoothDeviceProfileState;->sendMessage(I)V
  
+     .line 355
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->deferMessage(Landroid/os/Message;)V
***************
*** 313,318 ****
--- 338,344 ----
  
      goto :goto_1
  
+     .line 357
      :cond_2
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 334,345 ****
--- 360,373 ----
  
      if-eqz v0, :cond_3
  
+     .line 359
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      const/16 v2, 0x36
  
      invoke-virtual {v0, v2}, Landroid/bluetooth/BluetoothDeviceProfileState;->sendMessage(I)V
  
+     .line 360
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->deferMessage(Landroid/os/Message;)V
***************
*** 347,352 ****
--- 375,381 ----
  
      goto/16 :goto_1
  
+     .line 363
      :cond_3
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 356,361 ****
--- 385,391 ----
  
      goto/16 :goto_1
  
+     .line 366
      :sswitch_a
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 373,378 ****
--- 403,409 ----
  
      if-nez v0, :cond_0
  
+     .line 370
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mHeadsetService:Landroid/bluetooth/BluetoothHeadset;
***************
*** 382,392 ****
--- 413,425 ----
  
      if-nez v0, :cond_6
  
+     .line 371
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mAutoConnectionPending:Z
      invoke-static {v0, v1}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$1402(Landroid/bluetooth/BluetoothDeviceProfileState;Z)Z
  
+     .line 380
      :cond_4
      :goto_2
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
***************
*** 439,444 ****
--- 472,478 ----
  
      if-nez v0, :cond_5
  
+     .line 387
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mA2dpService:Landroid/server/BluetoothA2dpService;
***************
*** 455,460 ****
--- 489,495 ----
  
      invoke-virtual {v0, v2}, Landroid/server/BluetoothA2dpService;->connect(Landroid/bluetooth/BluetoothDevice;)Z
  
+     .line 389
      :cond_5
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 476,481 ****
--- 511,517 ----
  
      if-ne v0, v7, :cond_0
  
+     .line 391
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mService:Landroid/server/BluetoothService;
***************
*** 494,499 ****
--- 530,536 ----
  
      goto/16 :goto_1
  
+     .line 372
      :cond_6
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 536,541 ****
--- 573,579 ----
  
      if-nez v0, :cond_4
  
+     .line 378
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mHeadsetService:Landroid/bluetooth/BluetoothHeadset;
***************
*** 554,559 ****
--- 592,598 ----
  
      goto/16 :goto_2
  
+     .line 396
      :sswitch_b
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 571,580 ****
--- 610,621 ----
  
      if-nez v0, :cond_0
  
+     .line 399
      iget v0, p1, Landroid/os/Message;->arg1:I
  
      if-ne v0, v6, :cond_7
  
+     .line 400
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mA2dpService:Landroid/server/BluetoothA2dpService;
***************
*** 601,612 ****
--- 642,655 ----
  
      if-nez v0, :cond_0
  
+     .line 402
      const-string v0, "BluetoothDeviceProfileState"
  
      const-string v2, "A2dp:Connect Other Profiles"
  
      invoke-static {v0, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 403
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mA2dpService:Landroid/server/BluetoothA2dpService;
***************
*** 625,635 ****
--- 668,680 ----
  
      goto/16 :goto_1
  
+     .line 405
      :cond_7
      iget v0, p1, Landroid/os/Message;->arg1:I
  
      if-ne v0, v1, :cond_0
  
+     .line 406
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mHeadsetService:Landroid/bluetooth/BluetoothHeadset;
***************
*** 639,644 ****
--- 684,690 ----
  
      if-nez v0, :cond_8
  
+     .line 407
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #calls: Landroid/bluetooth/BluetoothDeviceProfileState;->deferMessage(Landroid/os/Message;)V
***************
*** 646,651 ****
--- 692,698 ----
  
      goto/16 :goto_1
  
+     .line 409
      :cond_8
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 664,675 ****
--- 711,724 ----
  
      if-nez v0, :cond_0
  
+     .line 410
      const-string v0, "BluetoothDeviceProfileState"
  
      const-string v2, "Headset:Connect Other Profiles"
  
      invoke-static {v0, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 411
      iget-object v0, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mHeadsetService:Landroid/bluetooth/BluetoothHeadset;
***************
*** 688,693 ****
--- 737,743 ----
  
      goto/16 :goto_1
  
+     .line 420
      :sswitch_c
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
***************
*** 705,715 ****
--- 755,767 ----
  
      invoke-virtual {v2, v3}, Landroid/content/Context;->unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  
+     .line 421
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mBroadcastReceiver:Landroid/content/BroadcastReceiver;
      invoke-static {v2, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$4202(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/content/BroadcastReceiver;)Landroid/content/BroadcastReceiver;
  
+     .line 422
      iget-object v2, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mAdapter:Landroid/bluetooth/BluetoothAdapter;
***************
*** 726,746 ****
--- 778,802 ----
  
      invoke-virtual {v2, v1, v3}, Landroid/bluetooth/BluetoothAdapter;->closeProfileProxy(ILandroid/bluetooth/BluetoothProfile;)V
  
+     .line 423
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mBluetoothProfileServiceListener:Landroid/bluetooth/BluetoothProfile$ServiceListener;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$4402(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothProfile$ServiceListener;)Landroid/bluetooth/BluetoothProfile$ServiceListener;
  
+     .line 424
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mOutgoingHandsfree:Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHandsfree;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2202(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHandsfree;)Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHandsfree;
  
+     .line 425
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mPbap:Landroid/bluetooth/BluetoothDeviceProfileState$PbapServiceListener;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$4502(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothDeviceProfileState$PbapServiceListener;)Landroid/bluetooth/BluetoothDeviceProfileState$PbapServiceListener;
  
+     .line 426
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #getter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mPbapService:Landroid/bluetooth/BluetoothPbap;
***************
*** 750,790 ****
--- 806,854 ----
  
      invoke-virtual {v1}, Landroid/bluetooth/BluetoothPbap;->close()V
  
+     .line 427
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mPbapService:Landroid/bluetooth/BluetoothPbap;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$1502(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothPbap;)Landroid/bluetooth/BluetoothPbap;
  
+     .line 428
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mIncomingHid:Landroid/bluetooth/BluetoothDeviceProfileState$IncomingHid;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$3302(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothDeviceProfileState$IncomingHid;)Landroid/bluetooth/BluetoothDeviceProfileState$IncomingHid;
  
+     .line 429
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mOutgoingHid:Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$3102(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;)Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHid;
  
+     .line 430
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mIncomingHandsfree:Landroid/bluetooth/BluetoothDeviceProfileState$IncomingHandsfree;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2402(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothDeviceProfileState$IncomingHandsfree;)Landroid/bluetooth/BluetoothDeviceProfileState$IncomingHandsfree;
  
+     .line 431
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mOutgoingHandsfree:Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHandsfree;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2202(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHandsfree;)Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingHandsfree;
  
+     .line 432
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mIncomingA2dp:Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2902(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;)Landroid/bluetooth/BluetoothDeviceProfileState$IncomingA2dp;
  
+     .line 433
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mOutgoingA2dp:Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingA2dp;
      invoke-static {v1, v5}, Landroid/bluetooth/BluetoothDeviceProfileState;->access$2702(Landroid/bluetooth/BluetoothDeviceProfileState;Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingA2dp;)Landroid/bluetooth/BluetoothDeviceProfileState$OutgoingA2dp;
  
+     .line 434
      iget-object v1, p0, Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;->this$0:Landroid/bluetooth/BluetoothDeviceProfileState;
  
      #setter for: Landroid/bluetooth/BluetoothDeviceProfileState;->mBondedDevice:Landroid/bluetooth/BluetoothDeviceProfileState$BondedDevice;
***************
*** 792,797 ****
--- 856,862 ----
  
      goto/16 :goto_0
  
+     .line 318
      :sswitch_data_0
      .sparse-switch
          -0x1 -> :sswitch_c
***************
*** 814,819 ****
--- 879,885 ----
          0x67 -> :sswitch_b
      .end sparse-switch
  
+     .line 380
      :array_0
      .array-data 0x4
          0x2t 0x0t 0x0t 0x0t
***************
*** 821,826 ****
--- 887,893 ----
          0x3t 0x0t 0x0t 0x0t
      .end array-data
  
+     .line 372
      :array_1
      .array-data 0x4
          0x2t 0x0t 0x0t 0x0t
