*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/AlteredCharSequence.smali	2012-03-23 19:35:18.355921605 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/AlteredCharSequence.smali.original	2012-03-23 19:34:44.471303721 -0500
***************
*** 35,50 ****
--- 35,56 ----
      .parameter "subend"
  
      .prologue
+     .line 43
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 44
      iput-object p1, p0, Landroid/text/AlteredCharSequence;->mSource:Ljava/lang/CharSequence;
  
+     .line 45
      iput-object p2, p0, Landroid/text/AlteredCharSequence;->mChars:[C
  
+     .line 46
      iput p3, p0, Landroid/text/AlteredCharSequence;->mStart:I
  
+     .line 47
      iput p4, p0, Landroid/text/AlteredCharSequence;->mEnd:I
  
+     .line 48
      return-void
  .end method
  
***************
*** 57,62 ****
--- 63,69 ----
      .parameter "x4"
  
      .prologue
+     .line 25
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/text/AlteredCharSequence;-><init>(Ljava/lang/CharSequence;[CII)V
  
      return-void
***************
*** 70,79 ****
--- 77,88 ----
      .parameter "subend"
  
      .prologue
+     .line 36
      instance-of v0, p0, Landroid/text/Spanned;
  
      if-eqz v0, :cond_0
  
+     .line 37
      new-instance v0, Landroid/text/AlteredCharSequence$AlteredSpanned;
  
      const/4 v5, 0x0
***************
*** 88,93 ****
--- 97,103 ----
  
      invoke-direct/range {v0 .. v5}, Landroid/text/AlteredCharSequence$AlteredSpanned;-><init>(Ljava/lang/CharSequence;[CIILandroid/text/AlteredCharSequence$1;)V
  
+     .line 39
      :goto_0
      return-object v0
  
***************
*** 106,111 ****
--- 116,122 ----
      .parameter "off"
  
      .prologue
+     .line 90
      iget v0, p0, Landroid/text/AlteredCharSequence;->mStart:I
  
      if-lt p1, v0, :cond_0
***************
*** 114,119 ****
--- 125,131 ----
  
      if-ge p1, v0, :cond_0
  
+     .line 91
      iget-object v0, p0, Landroid/text/AlteredCharSequence;->mChars:[C
  
      iget v1, p0, Landroid/text/AlteredCharSequence;->mStart:I
***************
*** 122,127 ****
--- 134,140 ----
  
      aget-char v0, v0, v1
  
+     .line 93
      :goto_0
      return v0
  
***************
*** 143,166 ****
--- 156,184 ----
      .parameter "off"
  
      .prologue
+     .line 106
      iget-object v0, p0, Landroid/text/AlteredCharSequence;->mSource:Ljava/lang/CharSequence;
  
      invoke-static {v0, p1, p2, p3, p4}, Landroid/text/TextUtils;->getChars(Ljava/lang/CharSequence;II[CI)V
  
+     .line 108
      iget v0, p0, Landroid/text/AlteredCharSequence;->mStart:I
  
      invoke-static {v0, p1}, Ljava/lang/Math;->max(II)I
  
      move-result p1
  
+     .line 109
      iget v0, p0, Landroid/text/AlteredCharSequence;->mEnd:I
  
      invoke-static {v0, p2}, Ljava/lang/Math;->min(II)I
  
      move-result p2
  
+     .line 111
      if-le p1, p2, :cond_0
  
+     .line 112
      iget-object v0, p0, Landroid/text/AlteredCharSequence;->mChars:[C
  
      iget v1, p0, Landroid/text/AlteredCharSequence;->mStart:I
***************
*** 171,176 ****
--- 189,195 ----
  
      invoke-static {v0, v1, p3, p4, v2}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 113
      :cond_0
      return-void
  .end method
***************
*** 179,184 ****
--- 198,204 ----
      .locals 1
  
      .prologue
+     .line 97
      iget-object v0, p0, Landroid/text/AlteredCharSequence;->mSource:Ljava/lang/CharSequence;
  
      invoke-interface {v0}, Ljava/lang/CharSequence;->length()I
***************
*** 194,199 ****
--- 214,220 ----
      .parameter "end"
  
      .prologue
+     .line 101
      iget-object v0, p0, Landroid/text/AlteredCharSequence;->mSource:Ljava/lang/CharSequence;
  
      invoke-interface {v0, p1, p2}, Ljava/lang/CharSequence;->subSequence(II)Ljava/lang/CharSequence;
***************
*** 223,238 ****
--- 244,263 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 116
      invoke-virtual {p0}, Landroid/text/AlteredCharSequence;->length()I
  
      move-result v0
  
+     .line 118
      .local v0, len:I
      new-array v1, v0, [C
  
+     .line 119
      .local v1, ret:[C
      invoke-virtual {p0, v2, v0, v1, v2}, Landroid/text/AlteredCharSequence;->getChars(II[CI)V
  
+     .line 120
      invoke-static {v1}, Ljava/lang/String;->valueOf([C)Ljava/lang/String;
  
      move-result-object v2
***************
*** 247,257 ****
--- 272,286 ----
      .parameter "subend"
  
      .prologue
+     .line 51
      iput-object p1, p0, Landroid/text/AlteredCharSequence;->mChars:[C
  
+     .line 52
      iput p2, p0, Landroid/text/AlteredCharSequence;->mStart:I
  
+     .line 53
      iput p3, p0, Landroid/text/AlteredCharSequence;->mEnd:I
  
+     .line 54
      return-void
  .end method
