*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/DynamicLayout.smali	2012-03-23 19:35:19.794947845 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/DynamicLayout.smali.original	2012-03-23 19:34:44.489304049 -0500
***************
*** 83,88 ****
--- 83,89 ----
      .locals 2
  
      .prologue
+     .line 491
      new-instance v0, Landroid/text/StaticLayout;
  
      const/4 v1, 0x0
***************
*** 91,96 ****
--- 92,98 ----
  
      sput-object v0, Landroid/text/DynamicLayout;->sStaticLayout:Landroid/text/StaticLayout;
  
+     .line 493
      const/4 v0, 0x0
  
      new-array v0, v0, [Ljava/lang/Object;
***************
*** 111,116 ****
--- 113,119 ----
      .parameter "includepad"
  
      .prologue
+     .line 47
      move-object v0, p0
  
      move-object v1, p1
***************
*** 131,136 ****
--- 134,140 ----
  
      invoke-direct/range {v0 .. v8}, Landroid/text/DynamicLayout;-><init>(Ljava/lang/CharSequence;Ljava/lang/CharSequence;Landroid/text/TextPaint;ILandroid/text/Layout$Alignment;FFZ)V
  
+     .line 49
      return-void
  .end method
  
***************
*** 146,151 ****
--- 150,156 ----
      .parameter "includepad"
  
      .prologue
+     .line 61
      const/4 v9, 0x0
  
      const/4 v10, 0x0
***************
*** 170,175 ****
--- 175,181 ----
  
      invoke-direct/range {v0 .. v10}, Landroid/text/DynamicLayout;-><init>(Ljava/lang/CharSequence;Ljava/lang/CharSequence;Landroid/text/TextPaint;ILandroid/text/Layout$Alignment;FFZLandroid/text/TextUtils$TruncateAt;I)V
  
+     .line 63
      return-void
  .end method
  
***************
*** 187,192 ****
--- 193,199 ----
      .parameter "ellipsizedWidth"
  
      .prologue
+     .line 78
      sget-object v6, Landroid/text/TextDirectionHeuristics;->FIRSTSTRONG_LTR:Landroid/text/TextDirectionHeuristic;
  
      move-object v0, p0
***************
*** 213,218 ****
--- 220,226 ----
  
      invoke-direct/range {v0 .. v11}, Landroid/text/DynamicLayout;-><init>(Ljava/lang/CharSequence;Ljava/lang/CharSequence;Landroid/text/TextPaint;ILandroid/text/Layout$Alignment;Landroid/text/TextDirectionHeuristic;FFZLandroid/text/TextUtils$TruncateAt;I)V
  
+     .line 80
      return-void
  .end method
  
***************
*** 231,236 ****
--- 239,245 ----
      .parameter "ellipsizedWidth"
  
      .prologue
+     .line 97
      if-nez p10, :cond_2
  
      move-object/from16 v3, p2
***************
*** 252,271 ****
--- 261,284 ----
  
      invoke-direct/range {v2 .. v9}, Landroid/text/Layout;-><init>(Ljava/lang/CharSequence;Landroid/text/TextPaint;ILandroid/text/Layout$Alignment;Landroid/text/TextDirectionHeuristic;FF)V
  
+     .line 104
      move-object/from16 v0, p1
  
      move-object/from16 v1, p0
  
      iput-object v0, v1, Landroid/text/DynamicLayout;->mBase:Ljava/lang/CharSequence;
  
+     .line 105
      move-object/from16 v0, p2
  
      move-object/from16 v1, p0
  
      iput-object v0, v1, Landroid/text/DynamicLayout;->mDisplay:Ljava/lang/CharSequence;
  
+     .line 107
      if-eqz p10, :cond_4
  
+     .line 108
      new-instance v2, Landroid/text/PackedIntVector;
  
      const/4 v3, 0x5
***************
*** 276,293 ****
--- 289,309 ----
  
      iput-object v2, v0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
  
+     .line 109
      move/from16 v0, p11
  
      move-object/from16 v1, p0
  
      iput v0, v1, Landroid/text/DynamicLayout;->mEllipsizedWidth:I
  
+     .line 110
      move-object/from16 v0, p10
  
      move-object/from16 v1, p0
  
      iput-object v0, v1, Landroid/text/DynamicLayout;->mEllipsizeAt:Landroid/text/TextUtils$TruncateAt;
  
+     .line 117
      :goto_1
      new-instance v2, Landroid/text/PackedObjectVector;
  
***************
*** 299,347 ****
--- 315,373 ----
  
      iput-object v2, v0, Landroid/text/DynamicLayout;->mObjects:Landroid/text/PackedObjectVector;
  
+     .line 119
      move/from16 v0, p9
  
      move-object/from16 v1, p0
  
      iput-boolean v0, v1, Landroid/text/DynamicLayout;->mIncludePad:Z
  
+     .line 129
      if-eqz p10, :cond_0
  
+     .line 130
      invoke-virtual/range {p0 .. p0}, Landroid/text/DynamicLayout;->getText()Ljava/lang/CharSequence;
  
      move-result-object v13
  
      check-cast v13, Landroid/text/Layout$Ellipsizer;
  
+     .line 132
      .local v13, e:Landroid/text/Layout$Ellipsizer;
      move-object/from16 v0, p0
  
      iput-object v0, v13, Landroid/text/Layout$Ellipsizer;->mLayout:Landroid/text/Layout;
  
+     .line 133
      move/from16 v0, p11
  
      iput v0, v13, Landroid/text/Layout$Ellipsizer;->mWidth:I
  
+     .line 134
      move-object/from16 v0, p10
  
      iput-object v0, v13, Landroid/text/Layout$Ellipsizer;->mMethod:Landroid/text/TextUtils$TruncateAt;
  
+     .line 135
      const/4 v2, 0x1
  
      move-object/from16 v0, p0
  
      iput-boolean v2, v0, Landroid/text/DynamicLayout;->mEllipsize:Z
  
+     .line 144
      .end local v13           #e:Landroid/text/Layout$Ellipsizer;
      :cond_0
      if-eqz p10, :cond_5
  
+     .line 145
      const/4 v2, 0x5
  
      new-array v0, v2, [I
  
      move-object/from16 v18, v0
  
+     .line 146
      .local v18, start:[I
      const/4 v2, 0x3
  
***************
*** 349,354 ****
--- 375,381 ----
  
      aput v3, v18, v2
  
+     .line 151
      :goto_2
      const/4 v2, 0x1
  
***************
*** 360,376 ****
--- 387,407 ----
  
      aput-object v3, v12, v2
  
+     .line 153
      .local v12, dirs:[Landroid/text/Layout$Directions;
      invoke-virtual/range {p3 .. p3}, Landroid/text/TextPaint;->getFontMetricsInt()Landroid/graphics/Paint$FontMetricsInt;
  
      move-result-object v14
  
+     .line 154
      .local v14, fm:Landroid/graphics/Paint$FontMetricsInt;
      iget v10, v14, Landroid/graphics/Paint$FontMetricsInt;->ascent:I
  
+     .line 155
      .local v10, asc:I
      iget v11, v14, Landroid/graphics/Paint$FontMetricsInt;->descent:I
  
+     .line 157
      .local v11, desc:I
      const/4 v2, 0x0
  
***************
*** 378,393 ****
--- 409,427 ----
  
      aput v3, v18, v2
  
+     .line 158
      const/4 v2, 0x1
  
      const/4 v3, 0x0
  
      aput v3, v18, v2
  
+     .line 159
      const/4 v2, 0x2
  
      aput v11, v18, v2
  
+     .line 160
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
***************
*** 398,409 ****
--- 432,445 ----
  
      invoke-virtual {v2, v3, v0}, Landroid/text/PackedIntVector;->insertAt(I[I)V
  
+     .line 162
      const/4 v2, 0x1
  
      sub-int v3, v11, v10
  
      aput v3, v18, v2
  
+     .line 163
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
***************
*** 414,419 ****
--- 450,456 ----
  
      invoke-virtual {v2, v3, v0}, Landroid/text/PackedIntVector;->insertAt(I[I)V
  
+     .line 165
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/text/DynamicLayout;->mObjects:Landroid/text/PackedObjectVector;
***************
*** 422,427 ****
--- 459,465 ----
  
      invoke-virtual {v2, v3, v12}, Landroid/text/PackedObjectVector;->insertAt(I[Ljava/lang/Object;)V
  
+     .line 169
      const/4 v2, 0x0
  
      const/4 v3, 0x0
***************
*** 436,453 ****
--- 474,494 ----
  
      invoke-direct {v0, v1, v2, v3, v4}, Landroid/text/DynamicLayout;->reflow(Ljava/lang/CharSequence;III)V
  
+     .line 171
      move-object/from16 v0, p1
  
      instance-of v2, v0, Landroid/text/Spannable;
  
      if-eqz v2, :cond_7
  
+     .line 172
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/text/DynamicLayout;->mWatcher:Landroid/text/DynamicLayout$ChangeWatcher;
  
      if-nez v2, :cond_1
  
+     .line 173
      new-instance v2, Landroid/text/DynamicLayout$ChangeWatcher;
  
      move-object/from16 v0, p0
***************
*** 461,468 ****
--- 502,511 ----
      :cond_1
      move-object/from16 v16, p1
  
+     .line 176
      check-cast v16, Landroid/text/Spannable;
  
+     .line 177
      .local v16, sp:Landroid/text/Spannable;
      const/4 v2, 0x0
  
***************
*** 480,485 ****
--- 523,529 ----
  
      check-cast v17, [Landroid/text/DynamicLayout$ChangeWatcher;
  
+     .line 178
      .local v17, spans:[Landroid/text/DynamicLayout$ChangeWatcher;
      const/4 v15, 0x0
  
***************
*** 491,506 ****
--- 535,553 ----
  
      if-ge v15, v2, :cond_6
  
+     .line 179
      aget-object v2, v17, v15
  
      move-object/from16 v0, v16
  
      invoke-interface {v0, v2}, Landroid/text/Spannable;->removeSpan(Ljava/lang/Object;)V
  
+     .line 178
      add-int/lit8 v15, v15, 0x1
  
      goto :goto_3
  
+     .line 97
      .end local v10           #asc:I
      .end local v11           #desc:I
      .end local v12           #dirs:[Landroid/text/Layout$Directions;
***************
*** 533,538 ****
--- 580,586 ----
  
      goto/16 :goto_0
  
+     .line 112
      :cond_4
      new-instance v2, Landroid/text/PackedIntVector;
  
***************
*** 544,555 ****
--- 592,605 ----
  
      iput-object v2, v0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
  
+     .line 113
      move/from16 v0, p4
  
      move-object/from16 v1, p0
  
      iput v0, v1, Landroid/text/DynamicLayout;->mEllipsizedWidth:I
  
+     .line 114
      const/4 v2, 0x0
  
      move-object/from16 v0, p0
***************
*** 558,563 ****
--- 608,614 ----
  
      goto/16 :goto_1
  
+     .line 148
      :cond_5
      const/4 v2, 0x3
  
***************
*** 568,573 ****
--- 619,625 ----
      .restart local v18       #start:[I
      goto/16 :goto_2
  
+     .line 181
      .restart local v10       #asc:I
      .restart local v11       #desc:I
      .restart local v12       #dirs:[Landroid/text/Layout$Directions;
***************
*** 592,597 ****
--- 644,650 ----
  
      invoke-interface {v0, v2, v3, v4, v5}, Landroid/text/Spannable;->setSpan(Ljava/lang/Object;III)V
  
+     .line 185
      .end local v15           #i:I
      .end local v16           #sp:Landroid/text/Spannable;
      .end local v17           #spans:[Landroid/text/DynamicLayout$ChangeWatcher;
***************
*** 608,613 ****
--- 661,667 ----
      .parameter "x4"
  
      .prologue
+     .line 33
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/text/DynamicLayout;->reflow(Ljava/lang/CharSequence;III)V
  
      return-void
***************
*** 621,626 ****
--- 675,681 ----
      .parameter "after"
  
      .prologue
+     .line 188
      move-object/from16 v0, p0
  
      iget-object v4, v0, Landroid/text/DynamicLayout;->mBase:Ljava/lang/CharSequence;
***************
*** 629,647 ****
--- 684,706 ----
  
      if-eq v0, v4, :cond_0
  
+     .line 364
      :goto_0
      return-void
  
+     .line 191
      :cond_0
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/text/DynamicLayout;->mDisplay:Ljava/lang/CharSequence;
  
+     .line 192
      .local v3, text:Ljava/lang/CharSequence;
      invoke-interface {v3}, Ljava/lang/CharSequence;->length()I
  
      move-result v29
  
+     .line 196
      .local v29, len:I
      const/16 v4, 0xa
  
***************
*** 651,671 ****
--- 710,737 ----
  
      move-result v23
  
+     .line 197
      .local v23, find:I
      if-gez v23, :cond_4
  
+     .line 198
      const/16 v23, 0x0
  
+     .line 203
      :goto_1
      sub-int v19, p2, v23
  
+     .line 204
      .local v19, diff:I
      add-int p3, p3, v19
  
+     .line 205
      add-int p4, p4, v19
  
+     .line 206
      sub-int p2, p2, v19
  
+     .line 211
      const/16 v4, 0xa
  
      add-int v5, p2, p4
***************
*** 674,706 ****
--- 740,781 ----
  
      move-result v30
  
+     .line 212
      .local v30, look:I
      if-gez v30, :cond_5
  
+     .line 213
      move/from16 v30, v29
  
+     .line 217
      :goto_2
      add-int v4, p2, p4
  
      sub-int v17, v30, v4
  
+     .line 218
      .local v17, change:I
      add-int p3, p3, v17
  
+     .line 219
      add-int p4, p4, v17
  
+     .line 223
      instance-of v4, v3, Landroid/text/Spanned;
  
      if-eqz v4, :cond_7
  
      move-object/from16 v33, v3
  
+     .line 224
      check-cast v33, Landroid/text/Spanned;
  
+     .line 228
      .local v33, sp:Landroid/text/Spanned;
      :cond_1
      const/4 v15, 0x0
  
+     .line 230
      .local v15, again:Z
      add-int v4, p2, p4
  
***************
*** 714,719 ****
--- 789,795 ----
  
      move-result-object v24
  
+     .line 233
      .local v24, force:[Ljava/lang/Object;
      const/16 v26, 0x0
  
***************
*** 727,732 ****
--- 803,809 ----
  
      if-ge v0, v4, :cond_6
  
+     .line 234
      aget-object v4, v24, v26
  
      move-object/from16 v0, v33
***************
*** 735,740 ****
--- 812,818 ----
  
      move-result v34
  
+     .line 235
      .local v34, st:I
      aget-object v4, v24, v26
  
***************
*** 744,749 ****
--- 822,828 ----
  
      move-result v20
  
+     .line 237
      .local v20, en:I
      move/from16 v0, v34
  
***************
*** 751,766 ****
--- 830,851 ----
  
      if-ge v0, v1, :cond_2
  
+     .line 238
      const/4 v15, 0x1
  
+     .line 240
      sub-int v19, p2, v34
  
+     .line 241
      add-int p3, p3, v19
  
+     .line 242
      add-int p4, p4, v19
  
+     .line 243
      sub-int p2, p2, v19
  
+     .line 246
      :cond_2
      add-int v4, p2, p4
  
***************
*** 768,788 ****
--- 853,879 ----
  
      if-le v0, v4, :cond_3
  
+     .line 247
      const/4 v15, 0x1
  
+     .line 249
      add-int v4, p2, p4
  
      sub-int v19, v20, v4
  
+     .line 250
      add-int p3, p3, v19
  
+     .line 251
      add-int p4, p4, v19
  
+     .line 233
      :cond_3
      add-int/lit8 v26, v26, 0x1
  
      goto :goto_3
  
+     .line 200
      .end local v15           #again:Z
      .end local v17           #change:I
      .end local v19           #diff:I
***************
*** 797,802 ****
--- 888,894 ----
  
      goto :goto_1
  
+     .line 215
      .restart local v19       #diff:I
      .restart local v30       #look:I
      :cond_5
***************
*** 804,809 ****
--- 896,902 ----
  
      goto :goto_2
  
+     .line 254
      .restart local v15       #again:Z
      .restart local v17       #change:I
      .restart local v24       #force:[Ljava/lang/Object;
***************
*** 812,817 ****
--- 905,911 ----
      :cond_6
      if-nez v15, :cond_1
  
+     .line 259
      .end local v15           #again:Z
      .end local v24           #force:[Ljava/lang/Object;
      .end local v26           #i:I
***************
*** 825,830 ****
--- 919,925 ----
  
      move-result v35
  
+     .line 260
      .local v35, startline:I
      move-object/from16 v0, p0
  
***************
*** 834,839 ****
--- 929,935 ----
  
      move-result v36
  
+     .line 262
      .local v36, startv:I
      add-int v4, p2, p3
  
***************
*** 843,848 ****
--- 939,945 ----
  
      move-result v21
  
+     .line 263
      .local v21, endline:I
      add-int v4, p2, p4
  
***************
*** 850,859 ****
--- 947,958 ----
  
      if-ne v4, v0, :cond_8
  
+     .line 264
      invoke-virtual/range {p0 .. p0}, Landroid/text/DynamicLayout;->getLineCount()I
  
      move-result v21
  
+     .line 265
      :cond_8
      move-object/from16 v0, p0
  
***************
*** 863,868 ****
--- 962,968 ----
  
      move-result v22
  
+     .line 266
      .local v22, endv:I
      invoke-virtual/range {p0 .. p0}, Landroid/text/DynamicLayout;->getLineCount()I
  
***************
*** 874,899 ****
--- 974,1005 ----
  
      const/16 v28, 0x1
  
+     .line 272
      .local v28, islast:Z
      :goto_4
      sget-object v5, Landroid/text/DynamicLayout;->sLock:[Ljava/lang/Object;
  
      monitor-enter v5
  
+     .line 273
      :try_start_0
      sget-object v2, Landroid/text/DynamicLayout;->sStaticLayout:Landroid/text/StaticLayout;
  
+     .line 274
      .local v2, reflowed:Landroid/text/StaticLayout;
      const/4 v4, 0x0
  
      sput-object v4, Landroid/text/DynamicLayout;->sStaticLayout:Landroid/text/StaticLayout;
  
+     .line 275
      monitor-exit v5
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 277
      if-nez v2, :cond_10
  
+     .line 278
      new-instance v2, Landroid/text/StaticLayout;
  
      .end local v2           #reflowed:Landroid/text/StaticLayout;
***************
*** 901,906 ****
--- 1007,1013 ----
  
      invoke-direct {v2, v4}, Landroid/text/StaticLayout;-><init>(Ljava/lang/CharSequence;)V
  
+     .line 283
      .restart local v2       #reflowed:Landroid/text/StaticLayout;
      :goto_5
      add-int v5, p2, p4
***************
*** 943,952 ****
--- 1050,1061 ----
  
      invoke-virtual/range {v2 .. v14}, Landroid/text/StaticLayout;->generate(Ljava/lang/CharSequence;IILandroid/text/TextPaint;ILandroid/text/TextDirectionHeuristic;FFZZFLandroid/text/TextUtils$TruncateAt;)V
  
+     .line 287
      invoke-virtual {v2}, Landroid/text/StaticLayout;->getLineCount()I
  
      move-result v31
  
+     .line 293
      .local v31, n:I
      add-int v4, p2, p4
  
***************
*** 964,971 ****
--- 1073,1082 ----
  
      if-ne v4, v5, :cond_9
  
+     .line 295
      add-int/lit8 v31, v31, -0x1
  
+     .line 299
      :cond_9
      move-object/from16 v0, p0
  
***************
*** 977,982 ****
--- 1088,1094 ----
  
      invoke-virtual {v4, v0, v5}, Landroid/text/PackedIntVector;->deleteAt(II)V
  
+     .line 300
      move-object/from16 v0, p0
  
      iget-object v4, v0, Landroid/text/DynamicLayout;->mObjects:Landroid/text/PackedObjectVector;
***************
*** 987,1004 ****
--- 1099,1119 ----
  
      invoke-virtual {v4, v0, v5}, Landroid/text/PackedObjectVector;->deleteAt(II)V
  
+     .line 304
      move/from16 v0, v31
  
      invoke-virtual {v2, v0}, Landroid/text/StaticLayout;->getLineTop(I)I
  
      move-result v25
  
+     .line 305
      .local v25, ht:I
      const/16 v38, 0x0
  
      .local v38, toppad:I
      const/16 v16, 0x0
  
+     .line 307
      .local v16, botpad:I
      move-object/from16 v0, p0
  
***************
*** 1008,1025 ****
--- 1123,1144 ----
  
      if-nez v35, :cond_a
  
+     .line 308
      invoke-virtual {v2}, Landroid/text/StaticLayout;->getTopPadding()I
  
      move-result v38
  
+     .line 309
      move/from16 v0, v38
  
      move-object/from16 v1, p0
  
      iput v0, v1, Landroid/text/DynamicLayout;->mTopPadding:I
  
+     .line 310
      sub-int v25, v25, v38
  
+     .line 312
      :cond_a
      move-object/from16 v0, p0
  
***************
*** 1029,1046 ****
--- 1148,1169 ----
  
      if-eqz v28, :cond_b
  
+     .line 313
      invoke-virtual {v2}, Landroid/text/StaticLayout;->getBottomPadding()I
  
      move-result v16
  
+     .line 314
      move/from16 v0, v16
  
      move-object/from16 v1, p0
  
      iput v0, v1, Landroid/text/DynamicLayout;->mBottomPadding:I
  
+     .line 315
      add-int v25, v25, v16
  
+     .line 318
      :cond_b
      move-object/from16 v0, p0
  
***************
*** 1054,1059 ****
--- 1177,1183 ----
  
      invoke-virtual {v4, v0, v5, v6}, Landroid/text/PackedIntVector;->adjustValuesBelow(III)V
  
+     .line 319
      move-object/from16 v0, p0
  
      iget-object v4, v0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
***************
*** 1068,1085 ****
--- 1192,1212 ----
  
      invoke-virtual {v4, v0, v5, v6}, Landroid/text/PackedIntVector;->adjustValuesBelow(III)V
  
+     .line 325
      move-object/from16 v0, p0
  
      iget-boolean v4, v0, Landroid/text/DynamicLayout;->mEllipsize:Z
  
      if-eqz v4, :cond_11
  
+     .line 326
      const/4 v4, 0x5
  
      new-array v0, v4, [I
  
      move-object/from16 v27, v0
  
+     .line 327
      .local v27, ints:[I
      const/4 v4, 0x3
  
***************
*** 1087,1092 ****
--- 1214,1220 ----
  
      aput v5, v27, v4
  
+     .line 332
      :goto_6
      const/4 v4, 0x1
  
***************
*** 1094,1099 ****
--- 1222,1228 ----
  
      move-object/from16 v32, v0
  
+     .line 334
      .local v32, objects:[Landroid/text/Layout$Directions;
      const/16 v26, 0x0
  
***************
*** 1105,1110 ****
--- 1234,1240 ----
  
      if-ge v0, v1, :cond_13
  
+     .line 335
      const/4 v5, 0x0
  
      move/from16 v0, v26
***************
*** 1138,1143 ****
--- 1268,1274 ----
  
      aput v4, v27, v5
  
+     .line 339
      move/from16 v0, v26
  
      invoke-virtual {v2, v0}, Landroid/text/StaticLayout;->getLineTop(I)I
***************
*** 1146,1167 ****
--- 1277,1303 ----
  
      add-int v37, v4, v36
  
+     .line 340
      .local v37, top:I
      if-lez v26, :cond_c
  
+     .line 341
      sub-int v37, v37, v38
  
+     .line 342
      :cond_c
      const/4 v4, 0x1
  
      aput v37, v27, v4
  
+     .line 344
      move/from16 v0, v26
  
      invoke-virtual {v2, v0}, Landroid/text/StaticLayout;->getLineDescent(I)I
  
      move-result v18
  
+     .line 345
      .local v18, desc:I
      add-int/lit8 v4, v31, -0x1
  
***************
*** 1169,1181 ****
--- 1305,1320 ----
  
      if-ne v0, v4, :cond_d
  
+     .line 346
      add-int v18, v18, v16
  
+     .line 348
      :cond_d
      const/4 v4, 0x2
  
      aput v18, v27, v4
  
+     .line 349
      const/4 v4, 0x0
  
      move/from16 v0, v26
***************
*** 1186,1197 ****
--- 1325,1338 ----
  
      aput-object v5, v32, v4
  
+     .line 351
      move-object/from16 v0, p0
  
      iget-boolean v4, v0, Landroid/text/DynamicLayout;->mEllipsize:Z
  
      if-eqz v4, :cond_e
  
+     .line 352
      const/4 v4, 0x3
  
      move/from16 v0, v26
***************
*** 1202,1207 ****
--- 1343,1349 ----
  
      aput v5, v27, v4
  
+     .line 353
      const/4 v4, 0x4
  
      move/from16 v0, v26
***************
*** 1212,1217 ****
--- 1354,1360 ----
  
      aput v5, v27, v4
  
+     .line 356
      :cond_e
      move-object/from16 v0, p0
  
***************
*** 1223,1228 ****
--- 1366,1372 ----
  
      invoke-virtual {v4, v5, v0}, Landroid/text/PackedIntVector;->insertAt(I[I)V
  
+     .line 357
      move-object/from16 v0, p0
  
      iget-object v4, v0, Landroid/text/DynamicLayout;->mObjects:Landroid/text/PackedObjectVector;
***************
*** 1233,1242 ****
--- 1377,1388 ----
  
      invoke-virtual {v4, v5, v0}, Landroid/text/PackedObjectVector;->insertAt(I[Ljava/lang/Object;)V
  
+     .line 334
      add-int/lit8 v26, v26, 0x1
  
      goto :goto_7
  
+     .line 266
      .end local v2           #reflowed:Landroid/text/StaticLayout;
      .end local v16           #botpad:I
      .end local v18           #desc:I
***************
*** 1253,1258 ****
--- 1399,1405 ----
  
      goto/16 :goto_4
  
+     .line 275
      .restart local v28       #islast:Z
      :catchall_0
      move-exception v4
***************
*** 1264,1275 ****
--- 1411,1424 ----
  
      throw v4
  
+     .line 280
      .restart local v2       #reflowed:Landroid/text/StaticLayout;
      :cond_10
      invoke-virtual {v2}, Landroid/text/StaticLayout;->prepare()V
  
      goto/16 :goto_5
  
+     .line 329
      .restart local v16       #botpad:I
      .restart local v25       #ht:I
      .restart local v31       #n:I
***************
*** 1284,1289 ****
--- 1433,1439 ----
      .restart local v27       #ints:[I
      goto/16 :goto_6
  
+     .line 335
      .restart local v26       #i:I
      .restart local v32       #objects:[Landroid/text/Layout$Directions;
      :cond_12
***************
*** 1291,1306 ****
--- 1441,1460 ----
  
      goto :goto_8
  
+     .line 360
      :cond_13
      sget-object v5, Landroid/text/DynamicLayout;->sLock:[Ljava/lang/Object;
  
      monitor-enter v5
  
+     .line 361
      :try_start_2
      sput-object v2, Landroid/text/DynamicLayout;->sStaticLayout:Landroid/text/StaticLayout;
  
+     .line 362
      invoke-virtual {v2}, Landroid/text/StaticLayout;->finish()V
  
+     .line 363
      monitor-exit v5
  
      goto/16 :goto_0
***************
*** 1321,1326 ****
--- 1475,1481 ----
      .locals 1
  
      .prologue
+     .line 408
      iget v0, p0, Landroid/text/DynamicLayout;->mBottomPadding:I
  
      return v0
***************
*** 1331,1342 ****
--- 1486,1500 ----
      .parameter "line"
  
      .prologue
+     .line 471
      iget-object v0, p0, Landroid/text/DynamicLayout;->mEllipsizeAt:Landroid/text/TextUtils$TruncateAt;
  
      if-nez v0, :cond_0
  
+     .line 472
      const/4 v0, 0x0
  
+     .line 475
      :goto_0
      return v0
  
***************
*** 1357,1368 ****
--- 1515,1529 ----
      .parameter "line"
  
      .prologue
+     .line 462
      iget-object v0, p0, Landroid/text/DynamicLayout;->mEllipsizeAt:Landroid/text/TextUtils$TruncateAt;
  
      if-nez v0, :cond_0
  
+     .line 463
      const/4 v0, 0x0
  
+     .line 466
      :goto_0
      return v0
  
***************
*** 1382,1387 ****
--- 1543,1549 ----
      .locals 1
  
      .prologue
+     .line 413
      iget v0, p0, Landroid/text/DynamicLayout;->mEllipsizedWidth:I
  
      return v0
***************
*** 1394,1399 ****
--- 1556,1562 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 388
      iget-object v1, p0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
  
      invoke-virtual {v1, p1, v0}, Landroid/text/PackedIntVector;->getValue(II)I
***************
*** 1416,1421 ****
--- 1579,1585 ----
      .locals 1
  
      .prologue
+     .line 368
      iget-object v0, p0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
  
      invoke-virtual {v0}, Landroid/text/PackedIntVector;->size()I
***************
*** 1432,1437 ****
--- 1596,1602 ----
      .parameter "line"
  
      .prologue
+     .line 378
      iget-object v0, p0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
  
      const/4 v1, 0x2
***************
*** 1448,1453 ****
--- 1613,1619 ----
      .parameter "line"
  
      .prologue
+     .line 398
      iget-object v0, p0, Landroid/text/DynamicLayout;->mObjects:Landroid/text/PackedObjectVector;
  
      const/4 v1, 0x0
***************
*** 1466,1471 ****
--- 1632,1638 ----
      .parameter "line"
  
      .prologue
+     .line 383
      iget-object v0, p0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
  
      const/4 v1, 0x0
***************
*** 1486,1491 ****
--- 1653,1659 ----
      .parameter "line"
  
      .prologue
+     .line 373
      iget-object v0, p0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
  
      const/4 v1, 0x1
***************
*** 1502,1507 ****
--- 1670,1676 ----
      .parameter "line"
  
      .prologue
+     .line 393
      iget-object v0, p0, Landroid/text/DynamicLayout;->mInts:Landroid/text/PackedIntVector;
  
      const/4 v1, 0x0
***************
*** 1519,1524 ****
--- 1688,1694 ----
      .locals 1
  
      .prologue
+     .line 403
      iget v0, p0, Landroid/text/DynamicLayout;->mTopPadding:I
  
      return v0
