*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/DynamicLayout$ChangeWatcher.smali	2012-03-23 19:35:19.395940570 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/DynamicLayout$ChangeWatcher.smali.original	2012-03-23 19:34:44.483303939 -0500
***************
*** 37,50 ****
--- 37,53 ----
      .parameter "layout"
  
      .prologue
+     .line 417
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 418
      new-instance v0, Ljava/lang/ref/WeakReference;
  
      invoke-direct {v0, p1}, Ljava/lang/ref/WeakReference;-><init>(Ljava/lang/Object;)V
  
      iput-object v0, p0, Landroid/text/DynamicLayout$ChangeWatcher;->mLayout:Ljava/lang/ref/WeakReference;
  
+     .line 419
      return-void
  .end method
  
***************
*** 56,61 ****
--- 59,65 ----
      .parameter "after"
  
      .prologue
+     .line 422
      iget-object v1, p0, Landroid/text/DynamicLayout$ChangeWatcher;->mLayout:Ljava/lang/ref/WeakReference;
  
      invoke-virtual {v1}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
***************
*** 64,86 ****
--- 68,95 ----
  
      check-cast v0, Landroid/text/DynamicLayout;
  
+     .line 424
      .local v0, ml:Landroid/text/DynamicLayout;
      if-eqz v0, :cond_1
  
+     .line 425
      #calls: Landroid/text/DynamicLayout;->reflow(Ljava/lang/CharSequence;III)V
      invoke-static {v0, p1, p2, p3, p4}, Landroid/text/DynamicLayout;->access$000(Landroid/text/DynamicLayout;Ljava/lang/CharSequence;III)V
  
+     .line 428
      .end local p1
      :cond_0
      :goto_0
      return-void
  
+     .line 426
      .restart local p1
      :cond_1
      instance-of v1, p1, Landroid/text/Spannable;
  
      if-eqz v1, :cond_0
  
+     .line 427
      check-cast p1, Landroid/text/Spannable;
  
      .end local p1
***************
*** 96,101 ****
--- 105,111 ----
      .parameter "s"
  
      .prologue
+     .line 438
      return-void
  .end method
  
***************
*** 107,112 ****
--- 117,123 ----
      .parameter "after"
  
      .prologue
+     .line 431
      return-void
  .end method
  
***************
*** 118,133 ****
--- 129,147 ----
      .parameter "end"
  
      .prologue
+     .line 441
      instance-of v0, p2, Landroid/text/style/UpdateLayout;
  
      if-eqz v0, :cond_0
  
+     .line 442
      sub-int v0, p4, p3
  
      sub-int v1, p4, p3
  
      invoke-direct {p0, p1, p3, v0, v1}, Landroid/text/DynamicLayout$ChangeWatcher;->reflow(Ljava/lang/CharSequence;III)V
  
+     .line 443
      :cond_0
      return-void
  .end method
***************
*** 142,163 ****
--- 156,181 ----
      .parameter "nend"
  
      .prologue
+     .line 451
      instance-of v0, p2, Landroid/text/style/UpdateLayout;
  
      if-eqz v0, :cond_0
  
+     .line 452
      sub-int v0, p4, p3
  
      sub-int v1, p4, p3
  
      invoke-direct {p0, p1, p3, v0, v1}, Landroid/text/DynamicLayout$ChangeWatcher;->reflow(Ljava/lang/CharSequence;III)V
  
+     .line 453
      sub-int v0, p6, p5
  
      sub-int v1, p6, p5
  
      invoke-direct {p0, p1, p5, v0, v1}, Landroid/text/DynamicLayout$ChangeWatcher;->reflow(Ljava/lang/CharSequence;III)V
  
+     .line 455
      :cond_0
      return-void
  .end method
***************
*** 170,185 ****
--- 188,206 ----
      .parameter "end"
  
      .prologue
+     .line 446
      instance-of v0, p2, Landroid/text/style/UpdateLayout;
  
      if-eqz v0, :cond_0
  
+     .line 447
      sub-int v0, p4, p3
  
      sub-int v1, p4, p3
  
      invoke-direct {p0, p1, p3, v0, v1}, Landroid/text/DynamicLayout$ChangeWatcher;->reflow(Ljava/lang/CharSequence;III)V
  
+     .line 448
      :cond_0
      return-void
  .end method
***************
*** 192,198 ****
--- 213,221 ----
      .parameter "after"
  
      .prologue
+     .line 434
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/text/DynamicLayout$ChangeWatcher;->reflow(Ljava/lang/CharSequence;III)V
  
+     .line 435
      return-void
  .end method
