*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/HtmlToSpannedConverter$Underline.smali	2012-03-23 19:35:18.877931124 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/HtmlToSpannedConverter$Underline.smali.original	2012-03-23 19:34:44.477303829 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 0
  
      .prologue
+     .line 788
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 29,34 ****
--- 30,36 ----
      .parameter "x0"
  
      .prologue
+     .line 788
      invoke-direct {p0}, Landroid/text/HtmlToSpannedConverter$Underline;-><init>()V
  
      return-void
