*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/HtmlToSpannedConverter.smali	2012-03-23 19:35:21.631981344 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/HtmlToSpannedConverter.smali.original	2012-03-23 19:34:44.514304505 -0500
***************
*** 59,64 ****
--- 59,65 ----
      .locals 1
  
      .prologue
+     .line 387
      const/4 v0, 0x6
  
      new-array v0, v0, [F
***************
*** 67,72 ****
--- 68,74 ----
  
      sput-object v0, Landroid/text/HtmlToSpannedConverter;->HEADER_SIZES:[F
  
+     .line 822
      invoke-static {}, Landroid/text/HtmlToSpannedConverter;->buildColorMap()Ljava/util/HashMap;
  
      move-result-object v0
***************
*** 75,80 ****
--- 77,83 ----
  
      return-void
  
+     .line 387
      nop
  
      :array_0
***************
*** 96,117 ****
--- 99,127 ----
      .parameter "parser"
  
      .prologue
+     .line 399
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 400
      iput-object p1, p0, Landroid/text/HtmlToSpannedConverter;->mSource:Ljava/lang/String;
  
+     .line 401
      new-instance v0, Landroid/text/SpannableStringBuilder;
  
      invoke-direct {v0}, Landroid/text/SpannableStringBuilder;-><init>()V
  
      iput-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
+     .line 402
      iput-object p2, p0, Landroid/text/HtmlToSpannedConverter;->mImageGetter:Landroid/text/Html$ImageGetter;
  
+     .line 403
      iput-object p3, p0, Landroid/text/HtmlToSpannedConverter;->mTagHandler:Landroid/text/Html$TagHandler;
  
+     .line 404
      iput-object p4, p0, Landroid/text/HtmlToSpannedConverter;->mReader:Lorg/xml/sax/XMLReader;
  
+     .line 405
      return-void
  .end method
  
***************
*** 129,138 ****
--- 139,150 ----
      .end annotation
  
      .prologue
+     .line 825
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
+     .line 826
      .local v0, map:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/String;Ljava/lang/Integer;>;"
      const-string v1, "aqua"
  
***************
*** 144,149 ****
--- 156,162 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 827
      const-string v1, "black"
  
      const/4 v2, 0x0
***************
*** 154,159 ****
--- 167,173 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 828
      const-string v1, "blue"
  
      const/16 v2, 0xff
***************
*** 164,169 ****
--- 178,184 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 829
      const-string v1, "fuchsia"
  
      const v2, 0xff00ff
***************
*** 174,179 ****
--- 189,195 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 830
      const-string v1, "green"
  
      const v2, 0x8000
***************
*** 184,189 ****
--- 200,206 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 831
      const-string v1, "grey"
  
      const v2, 0x808080
***************
*** 194,199 ****
--- 211,217 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 832
      const-string v1, "lime"
  
      const v2, 0xff00
***************
*** 204,210 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v1, "maroon"
  
      const/high16 v2, 0x80
  
--- 222,229 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 833
!     const-string/jumbo v1, "maroon"
  
      const/high16 v2, 0x80
  
***************
*** 214,220 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v1, "navy"
  
      const/16 v2, 0x80
  
--- 233,240 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 834
!     const-string/jumbo v1, "navy"
  
      const/16 v2, 0x80
  
***************
*** 224,230 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v1, "olive"
  
      const v2, 0x808000
  
--- 244,251 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 835
!     const-string/jumbo v1, "olive"
  
      const v2, 0x808000
  
***************
*** 234,240 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v1, "purple"
  
      const v2, 0x800080
  
--- 255,262 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 836
!     const-string/jumbo v1, "purple"
  
      const v2, 0x800080
  
***************
*** 244,250 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v1, "red"
  
      const/high16 v2, 0xff
  
--- 266,273 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 837
!     const-string/jumbo v1, "red"
  
      const/high16 v2, 0xff
  
***************
*** 254,260 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v1, "silver"
  
      const v2, 0xc0c0c0
  
--- 277,284 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 838
!     const-string/jumbo v1, "silver"
  
      const v2, 0xc0c0c0
  
***************
*** 264,270 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v1, "teal"
  
      const v2, 0x8080
  
--- 288,295 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 839
!     const-string/jumbo v1, "teal"
  
      const v2, 0x8080
  
***************
*** 274,280 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v1, "white"
  
      const v2, 0xffffff
  
--- 299,306 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 840
!     const-string/jumbo v1, "white"
  
      const v2, 0xffffff
  
***************
*** 284,290 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v1, "yellow"
  
      const v2, 0xffff00
  
--- 310,317 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 841
!     const-string/jumbo v1, "yellow"
  
      const v2, 0xffff00
  
***************
*** 294,299 ****
--- 321,327 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 842
      return-object v0
  .end method
  
***************
*** 304,332 ****
--- 332,367 ----
      .parameter "repl"
  
      .prologue
+     .line 586
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v0
  
+     .line 587
      .local v0, len:I
      invoke-static {p0, p1}, Landroid/text/HtmlToSpannedConverter;->getLast(Landroid/text/Spanned;Ljava/lang/Class;)Ljava/lang/Object;
  
      move-result-object v1
  
+     .line 588
      .local v1, obj:Ljava/lang/Object;
      invoke-virtual {p0, v1}, Landroid/text/SpannableStringBuilder;->getSpanStart(Ljava/lang/Object;)I
  
      move-result v2
  
+     .line 590
      .local v2, where:I
      invoke-virtual {p0, v1}, Landroid/text/SpannableStringBuilder;->removeSpan(Ljava/lang/Object;)V
  
+     .line 592
      if-eq v2, v0, :cond_0
  
+     .line 593
      const/16 v3, 0x21
  
      invoke-virtual {p0, p2, v2, v0, v3}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 596
      :cond_0
      return-void
  .end method
***************
*** 336,345 ****
--- 371,382 ----
      .parameter "text"
  
      .prologue
+     .line 676
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v1
  
+     .line 677
      .local v1, len:I
      const-class v4, Landroid/text/HtmlToSpannedConverter$Href;
  
***************
*** 347,371 ****
--- 384,414 ----
  
      move-result-object v2
  
+     .line 678
      .local v2, obj:Ljava/lang/Object;
      invoke-virtual {p0, v2}, Landroid/text/SpannableStringBuilder;->getSpanStart(Ljava/lang/Object;)I
  
      move-result v3
  
+     .line 680
      .local v3, where:I
      invoke-virtual {p0, v2}, Landroid/text/SpannableStringBuilder;->removeSpan(Ljava/lang/Object;)V
  
+     .line 682
      if-eq v3, v1, :cond_0
  
      move-object v0, v2
  
+     .line 683
      check-cast v0, Landroid/text/HtmlToSpannedConverter$Href;
  
+     .line 685
      .local v0, h:Landroid/text/HtmlToSpannedConverter$Href;
      iget-object v4, v0, Landroid/text/HtmlToSpannedConverter$Href;->mHref:Ljava/lang/String;
  
      if-eqz v4, :cond_0
  
+     .line 686
      new-instance v4, Landroid/text/style/URLSpan;
  
      iget-object v5, v0, Landroid/text/HtmlToSpannedConverter$Href;->mHref:Ljava/lang/String;
***************
*** 376,381 ****
--- 419,425 ----
  
      invoke-virtual {p0, v4, v3, v1, v5}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 690
      .end local v0           #h:Landroid/text/HtmlToSpannedConverter$Href;
      :cond_0
      return-void
***************
*** 392,401 ****
--- 436,447 ----
  
      const/16 v14, 0x21
  
+     .line 631
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v9
  
+     .line 632
      .local v9, len:I
      const-class v0, Landroid/text/HtmlToSpannedConverter$Font;
  
***************
*** 403,422 ****
--- 449,473 ----
  
      move-result-object v11
  
+     .line 633
      .local v11, obj:Ljava/lang/Object;
      invoke-virtual {p0, v11}, Landroid/text/SpannableStringBuilder;->getSpanStart(Ljava/lang/Object;)I
  
      move-result v13
  
+     .line 635
      .local v13, where:I
      invoke-virtual {p0, v11}, Landroid/text/SpannableStringBuilder;->removeSpan(Ljava/lang/Object;)V
  
+     .line 637
      if-eq v13, v9, :cond_1
  
      move-object v8, v11
  
+     .line 638
      check-cast v8, Landroid/text/HtmlToSpannedConverter$Font;
  
+     .line 640
      .local v8, f:Landroid/text/HtmlToSpannedConverter$Font;
      iget-object v0, v8, Landroid/text/HtmlToSpannedConverter$Font;->mColor:Ljava/lang/String;
  
***************
*** 426,431 ****
--- 477,483 ----
  
      if-nez v0, :cond_0
  
+     .line 641
      iget-object v0, v8, Landroid/text/HtmlToSpannedConverter$Font;->mColor:Ljava/lang/String;
  
      const-string v3, "@"
***************
*** 436,445 ****
--- 488,499 ----
  
      if-eqz v0, :cond_2
  
+     .line 642
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v12
  
+     .line 643
      .local v12, res:Landroid/content/res/Resources;
      iget-object v0, v8, Landroid/text/HtmlToSpannedConverter$Font;->mColor:Ljava/lang/String;
  
***************
*** 449,454 ****
--- 503,509 ----
  
      move-result-object v10
  
+     .line 644
      .local v10, name:Ljava/lang/String;
      const-string v0, "color"
  
***************
*** 458,470 ****
--- 513,528 ----
  
      move-result v7
  
+     .line 645
      .local v7, colorRes:I
      if-eqz v7, :cond_0
  
+     .line 646
      invoke-virtual {v12, v7}, Landroid/content/res/Resources;->getColorStateList(I)Landroid/content/res/ColorStateList;
  
      move-result-object v4
  
+     .line 647
      .local v4, colors:Landroid/content/res/ColorStateList;
      new-instance v0, Landroid/text/style/TextAppearanceSpan;
  
***************
*** 476,481 ****
--- 534,540 ----
  
      invoke-virtual {p0, v0, v13, v9, v14}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 661
      .end local v4           #colors:Landroid/content/res/ColorStateList;
      .end local v7           #colorRes:I
      .end local v10           #name:Ljava/lang/String;
***************
*** 486,491 ****
--- 545,551 ----
  
      if-eqz v0, :cond_1
  
+     .line 662
      new-instance v0, Landroid/text/style/TypefaceSpan;
  
      iget-object v1, v8, Landroid/text/HtmlToSpannedConverter$Font;->mFace:Ljava/lang/String;
***************
*** 494,503 ****
--- 554,565 ----
  
      invoke-virtual {p0, v0, v13, v9, v14}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 666
      .end local v8           #f:Landroid/text/HtmlToSpannedConverter$Font;
      :cond_1
      return-void
  
+     .line 652
      .restart local v8       #f:Landroid/text/HtmlToSpannedConverter$Font;
      :cond_2
      iget-object v0, v8, Landroid/text/HtmlToSpannedConverter$Font;->mColor:Ljava/lang/String;
***************
*** 506,516 ****
--- 568,580 ----
  
      move-result v6
  
+     .line 653
      .local v6, c:I
      const/4 v0, -0x1
  
      if-eq v6, v0, :cond_0
  
+     .line 654
      new-instance v0, Landroid/text/style/ForegroundColorSpan;
  
      const/high16 v1, -0x100
***************
*** 531,540 ****
--- 595,606 ----
      .prologue
      const/16 v7, 0x21
  
+     .line 693
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v1
  
+     .line 694
      .local v1, len:I
      const-class v4, Landroid/text/HtmlToSpannedConverter$Header;
  
***************
*** 542,555 ****
--- 608,624 ----
  
      move-result-object v2
  
+     .line 696
      .local v2, obj:Ljava/lang/Object;
      invoke-virtual {p0, v2}, Landroid/text/SpannableStringBuilder;->getSpanStart(Ljava/lang/Object;)I
  
      move-result v3
  
+     .line 698
      .local v3, where:I
      invoke-virtual {p0, v2}, Landroid/text/SpannableStringBuilder;->removeSpan(Ljava/lang/Object;)V
  
+     .line 701
      :goto_0
      if-le v1, v3, :cond_0
  
***************
*** 563,579 ****
--- 632,652 ----
  
      if-ne v4, v5, :cond_0
  
+     .line 702
      add-int/lit8 v1, v1, -0x1
  
      goto :goto_0
  
+     .line 705
      :cond_0
      if-eq v3, v1, :cond_1
  
      move-object v0, v2
  
+     .line 706
      check-cast v0, Landroid/text/HtmlToSpannedConverter$Header;
  
+     .line 708
      .local v0, h:Landroid/text/HtmlToSpannedConverter$Header;
      new-instance v4, Landroid/text/style/RelativeSizeSpan;
  
***************
*** 590,595 ****
--- 663,669 ----
  
      invoke-virtual {p0, v4, v3, v1, v7}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 710
      new-instance v4, Landroid/text/style/StyleSpan;
  
      const/4 v5, 0x1
***************
*** 598,603 ****
--- 672,678 ----
  
      invoke-virtual {p0, v4, v3, v1, v7}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 713
      .end local v0           #h:Landroid/text/HtmlToSpannedConverter$Header;
      :cond_1
      return-void
***************
*** 610,615 ****
--- 685,691 ----
      .prologue
      const/4 v2, -0x1
  
+     .line 852
      sget-object v3, Landroid/text/HtmlToSpannedConverter;->COLORS:Ljava/util/HashMap;
  
      invoke-virtual {p0}, Ljava/lang/String;->toLowerCase()Ljava/lang/String;
***************
*** 622,637 ****
--- 698,717 ----
  
      check-cast v0, Ljava/lang/Integer;
  
+     .line 853
      .local v0, i:Ljava/lang/Integer;
      if-eqz v0, :cond_0
  
+     .line 854
      invoke-virtual {v0}, Ljava/lang/Integer;->intValue()I
  
      move-result v2
  
+     .line 859
      :goto_0
      return v2
  
+     .line 857
      :cond_0
      const/4 v3, -0x1
  
***************
*** 644,652 ****
--- 724,734 ----
  
      goto :goto_0
  
+     .line 858
      :catch_0
      move-exception v1
  
+     .line 859
      .local v1, nfe:Ljava/lang/NumberFormatException;
      goto :goto_0
  .end method
***************
*** 657,662 ****
--- 739,745 ----
      .parameter "kind"
  
      .prologue
+     .line 570
      const/4 v1, 0x0
  
      invoke-interface {p0}, Landroid/text/Spanned;->length()I
***************
*** 667,679 ****
--- 750,765 ----
  
      move-result-object v0
  
+     .line 572
      .local v0, objs:[Ljava/lang/Object;
      array-length v1, v0
  
      if-nez v1, :cond_0
  
+     .line 573
      const/4 v1, 0x0
  
+     .line 575
      :goto_0
      return-object v1
  
***************
*** 692,701 ****
--- 778,789 ----
      .parameter "text"
  
      .prologue
+     .line 562
      const-string v0, "\n"
  
      invoke-virtual {p0, v0}, Landroid/text/SpannableStringBuilder;->append(Ljava/lang/CharSequence;)Landroid/text/SpannableStringBuilder;
  
+     .line 563
      return-void
  .end method
  
***************
*** 710,715 ****
--- 798,804 ----
  
      const/4 v3, 0x2
  
+     .line 496
      const-string v0, "br"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
***************
*** 718,733 ****
  
      if-eqz v0, :cond_1
  
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleBr(Landroid/text/SpannableStringBuilder;)V
  
      :cond_0
      :goto_0
      return-void
  
      :cond_1
!     const-string v0, "p"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 807,825 ----
  
      if-eqz v0, :cond_1
  
+     .line 497
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleBr(Landroid/text/SpannableStringBuilder;)V
  
+     .line 542
      :cond_0
      :goto_0
      return-void
  
+     .line 498
      :cond_1
!     const-string/jumbo v0, "p"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 735,746 ****
--- 827,840 ----
  
      if-eqz v0, :cond_2
  
+     .line 499
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
      goto :goto_0
  
+     .line 500
      :cond_2
      const-string v0, "div"
  
***************
*** 750,763 ****
  
      if-eqz v0, :cond_3
  
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
      goto :goto_0
  
      :cond_3
!     const-string v0, "strong"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 844,859 ----
  
      if-eqz v0, :cond_3
  
+     .line 501
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
      goto :goto_0
  
+     .line 502
      :cond_3
!     const-string/jumbo v0, "strong"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 765,770 ****
--- 861,867 ----
  
      if-eqz v0, :cond_4
  
+     .line 503
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Bold;
***************
*** 777,782 ****
--- 874,880 ----
  
      goto :goto_0
  
+     .line 504
      :cond_4
      const-string v0, "b"
  
***************
*** 786,791 ****
--- 884,890 ----
  
      if-eqz v0, :cond_5
  
+     .line 505
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Bold;
***************
*** 798,803 ****
--- 897,903 ----
  
      goto :goto_0
  
+     .line 506
      :cond_5
      const-string v0, "em"
  
***************
*** 807,812 ****
--- 907,913 ----
  
      if-eqz v0, :cond_6
  
+     .line 507
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Italic;
***************
*** 819,824 ****
--- 920,926 ----
  
      goto :goto_0
  
+     .line 508
      :cond_6
      const-string v0, "cite"
  
***************
*** 828,833 ****
--- 930,936 ----
  
      if-eqz v0, :cond_7
  
+     .line 509
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Italic;
***************
*** 840,845 ****
--- 943,949 ----
  
      goto :goto_0
  
+     .line 510
      :cond_7
      const-string v0, "dfn"
  
***************
*** 849,854 ****
--- 953,959 ----
  
      if-eqz v0, :cond_8
  
+     .line 511
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Italic;
***************
*** 861,866 ****
--- 966,972 ----
  
      goto/16 :goto_0
  
+     .line 512
      :cond_8
      const-string v0, "i"
  
***************
*** 870,875 ****
--- 976,982 ----
  
      if-eqz v0, :cond_9
  
+     .line 513
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Italic;
***************
*** 882,887 ****
--- 989,995 ----
  
      goto/16 :goto_0
  
+     .line 514
      :cond_9
      const-string v0, "big"
  
***************
*** 891,896 ****
--- 999,1005 ----
  
      if-eqz v0, :cond_a
  
+     .line 515
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Big;
***************
*** 905,912 ****
  
      goto/16 :goto_0
  
      :cond_a
!     const-string v0, "small"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1014,1022 ----
  
      goto/16 :goto_0
  
+     .line 516
      :cond_a
!     const-string/jumbo v0, "small"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 914,919 ****
--- 1024,1030 ----
  
      if-eqz v0, :cond_b
  
+     .line 517
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Small;
***************
*** 928,933 ****
--- 1039,1045 ----
  
      goto/16 :goto_0
  
+     .line 518
      :cond_b
      const-string v0, "font"
  
***************
*** 937,948 ****
--- 1049,1062 ----
  
      if-eqz v0, :cond_c
  
+     .line 519
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->endFont(Landroid/text/SpannableStringBuilder;)V
  
      goto/16 :goto_0
  
+     .line 520
      :cond_c
      const-string v0, "blockquote"
  
***************
*** 952,961 ****
--- 1066,1077 ----
  
      if-eqz v0, :cond_d
  
+     .line 521
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
+     .line 522
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Blockquote;
***************
*** 968,975 ****
  
      goto/16 :goto_0
  
      :cond_d
!     const-string v0, "tt"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1084,1092 ----
  
      goto/16 :goto_0
  
+     .line 523
      :cond_d
!     const-string/jumbo v0, "tt"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 977,989 ****
  
      if-eqz v0, :cond_e
  
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Monospace;
  
      new-instance v2, Landroid/text/style/TypefaceSpan;
  
!     const-string v3, "monospace"
  
      invoke-direct {v2, v3}, Landroid/text/style/TypefaceSpan;-><init>(Ljava/lang/String;)V
  
--- 1094,1107 ----
  
      if-eqz v0, :cond_e
  
+     .line 524
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Monospace;
  
      new-instance v2, Landroid/text/style/TypefaceSpan;
  
!     const-string/jumbo v3, "monospace"
  
      invoke-direct {v2, v3}, Landroid/text/style/TypefaceSpan;-><init>(Ljava/lang/String;)V
  
***************
*** 991,996 ****
--- 1109,1115 ----
  
      goto/16 :goto_0
  
+     .line 526
      :cond_e
      const-string v0, "a"
  
***************
*** 1000,1013 ****
  
      if-eqz v0, :cond_f
  
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->endA(Landroid/text/SpannableStringBuilder;)V
  
      goto/16 :goto_0
  
      :cond_f
!     const-string v0, "u"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1119,1134 ----
  
      if-eqz v0, :cond_f
  
+     .line 527
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->endA(Landroid/text/SpannableStringBuilder;)V
  
      goto/16 :goto_0
  
+     .line 528
      :cond_f
!     const-string/jumbo v0, "u"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1015,1020 ****
--- 1136,1142 ----
  
      if-eqz v0, :cond_10
  
+     .line 529
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Underline;
***************
*** 1027,1034 ****
  
      goto/16 :goto_0
  
      :cond_10
!     const-string v0, "sup"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1149,1157 ----
  
      goto/16 :goto_0
  
+     .line 530
      :cond_10
!     const-string/jumbo v0, "sup"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1036,1041 ****
--- 1159,1165 ----
  
      if-eqz v0, :cond_11
  
+     .line 531
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Super;
***************
*** 1048,1055 ****
  
      goto/16 :goto_0
  
      :cond_11
!     const-string v0, "sub"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1172,1180 ----
  
      goto/16 :goto_0
  
+     .line 532
      :cond_11
!     const-string/jumbo v0, "sub"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1057,1062 ****
--- 1182,1188 ----
  
      if-eqz v0, :cond_12
  
+     .line 533
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const-class v1, Landroid/text/HtmlToSpannedConverter$Sub;
***************
*** 1069,1074 ****
--- 1195,1201 ----
  
      goto/16 :goto_0
  
+     .line 534
      :cond_12
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
***************
*** 1104,1124 ****
--- 1231,1255 ----
  
      if-gt v0, v1, :cond_13
  
+     .line 537
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
+     .line 538
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->endHeader(Landroid/text/SpannableStringBuilder;)V
  
      goto/16 :goto_0
  
+     .line 539
      :cond_13
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mTagHandler:Landroid/text/Html$TagHandler;
  
      if-eqz v0, :cond_0
  
+     .line 540
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mTagHandler:Landroid/text/Html$TagHandler;
  
      iget-object v1, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
***************
*** 1137,1146 ****
--- 1268,1279 ----
      .prologue
      const/16 v2, 0xa
  
+     .line 545
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v0
  
+     .line 547
      .local v0, len:I
      const/4 v1, 0x1
  
***************
*** 1154,1159 ****
--- 1287,1293 ----
  
      if-ne v1, v2, :cond_2
  
+     .line 548
      const/4 v1, 0x2
  
      if-lt v0, v1, :cond_1
***************
*** 1166,1175 ****
--- 1300,1311 ----
  
      if-ne v1, v2, :cond_1
  
+     .line 559
      :cond_0
      :goto_0
      return-void
  
+     .line 552
      :cond_1
      const-string v1, "\n"
  
***************
*** 1177,1185 ****
--- 1313,1323 ----
  
      goto :goto_0
  
+     .line 556
      :cond_2
      if-eqz v0, :cond_0
  
+     .line 557
      const-string v1, "\n\n"
  
      invoke-virtual {p0, v1}, Landroid/text/SpannableStringBuilder;->append(Ljava/lang/CharSequence;)Landroid/text/SpannableStringBuilder;
***************
*** 1197,1202 ****
--- 1335,1341 ----
  
      const/4 v2, 0x0
  
+     .line 445
      const-string v0, "br"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
***************
*** 1205,1216 ****
  
      if-eqz v0, :cond_1
  
      :cond_0
      :goto_0
      return-void
  
      :cond_1
!     const-string v0, "p"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1344,1357 ----
  
      if-eqz v0, :cond_1
  
+     .line 493
      :cond_0
      :goto_0
      return-void
  
+     .line 448
      :cond_1
!     const-string/jumbo v0, "p"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1218,1229 ****
--- 1359,1372 ----
  
      if-eqz v0, :cond_2
  
+     .line 449
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
      goto :goto_0
  
+     .line 450
      :cond_2
      const-string v0, "div"
  
***************
*** 1233,1246 ****
  
      if-eqz v0, :cond_3
  
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
      goto :goto_0
  
      :cond_3
!     const-string v0, "strong"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1376,1391 ----
  
      if-eqz v0, :cond_3
  
+     .line 451
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
      goto :goto_0
  
+     .line 452
      :cond_3
!     const-string/jumbo v0, "strong"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1248,1253 ****
--- 1393,1399 ----
  
      if-eqz v0, :cond_4
  
+     .line 453
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Bold;
***************
*** 1258,1263 ****
--- 1404,1410 ----
  
      goto :goto_0
  
+     .line 454
      :cond_4
      const-string v0, "b"
  
***************
*** 1267,1272 ****
--- 1414,1420 ----
  
      if-eqz v0, :cond_5
  
+     .line 455
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Bold;
***************
*** 1277,1282 ****
--- 1425,1431 ----
  
      goto :goto_0
  
+     .line 456
      :cond_5
      const-string v0, "em"
  
***************
*** 1286,1291 ****
--- 1435,1441 ----
  
      if-eqz v0, :cond_6
  
+     .line 457
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Italic;
***************
*** 1296,1301 ****
--- 1446,1452 ----
  
      goto :goto_0
  
+     .line 458
      :cond_6
      const-string v0, "cite"
  
***************
*** 1305,1310 ****
--- 1456,1462 ----
  
      if-eqz v0, :cond_7
  
+     .line 459
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Italic;
***************
*** 1315,1320 ****
--- 1467,1473 ----
  
      goto :goto_0
  
+     .line 460
      :cond_7
      const-string v0, "dfn"
  
***************
*** 1324,1329 ****
--- 1477,1483 ----
  
      if-eqz v0, :cond_8
  
+     .line 461
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Italic;
***************
*** 1334,1339 ****
--- 1488,1494 ----
  
      goto :goto_0
  
+     .line 462
      :cond_8
      const-string v0, "i"
  
***************
*** 1343,1348 ****
--- 1498,1504 ----
  
      if-eqz v0, :cond_9
  
+     .line 463
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Italic;
***************
*** 1353,1358 ****
--- 1509,1515 ----
  
      goto/16 :goto_0
  
+     .line 464
      :cond_9
      const-string v0, "big"
  
***************
*** 1362,1367 ****
--- 1519,1525 ----
  
      if-eqz v0, :cond_a
  
+     .line 465
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Big;
***************
*** 1372,1379 ****
  
      goto/16 :goto_0
  
      :cond_a
!     const-string v0, "small"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1530,1538 ----
  
      goto/16 :goto_0
  
+     .line 466
      :cond_a
!     const-string/jumbo v0, "small"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1381,1386 ****
--- 1540,1546 ----
  
      if-eqz v0, :cond_b
  
+     .line 467
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Small;
***************
*** 1391,1396 ****
--- 1551,1557 ----
  
      goto/16 :goto_0
  
+     .line 468
      :cond_b
      const-string v0, "font"
  
***************
*** 1400,1411 ****
--- 1561,1574 ----
  
      if-eqz v0, :cond_c
  
+     .line 469
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0, p2}, Landroid/text/HtmlToSpannedConverter;->startFont(Landroid/text/SpannableStringBuilder;Lorg/xml/sax/Attributes;)V
  
      goto/16 :goto_0
  
+     .line 470
      :cond_c
      const-string v0, "blockquote"
  
***************
*** 1415,1424 ****
--- 1578,1589 ----
  
      if-eqz v0, :cond_d
  
+     .line 471
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
+     .line 472
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Blockquote;
***************
*** 1429,1436 ****
  
      goto/16 :goto_0
  
      :cond_d
!     const-string v0, "tt"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1594,1602 ----
  
      goto/16 :goto_0
  
+     .line 473
      :cond_d
!     const-string/jumbo v0, "tt"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1438,1443 ****
--- 1604,1610 ----
  
      if-eqz v0, :cond_e
  
+     .line 474
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Monospace;
***************
*** 1448,1453 ****
--- 1615,1621 ----
  
      goto/16 :goto_0
  
+     .line 475
      :cond_e
      const-string v0, "a"
  
***************
*** 1457,1470 ****
  
      if-eqz v0, :cond_f
  
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0, p2}, Landroid/text/HtmlToSpannedConverter;->startA(Landroid/text/SpannableStringBuilder;Lorg/xml/sax/Attributes;)V
  
      goto/16 :goto_0
  
      :cond_f
!     const-string v0, "u"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1625,1640 ----
  
      if-eqz v0, :cond_f
  
+     .line 476
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0, p2}, Landroid/text/HtmlToSpannedConverter;->startA(Landroid/text/SpannableStringBuilder;Lorg/xml/sax/Attributes;)V
  
      goto/16 :goto_0
  
+     .line 477
      :cond_f
!     const-string/jumbo v0, "u"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1472,1477 ****
--- 1642,1648 ----
  
      if-eqz v0, :cond_10
  
+     .line 478
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Underline;
***************
*** 1482,1489 ****
  
      goto/16 :goto_0
  
      :cond_10
!     const-string v0, "sup"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1653,1661 ----
  
      goto/16 :goto_0
  
+     .line 479
      :cond_10
!     const-string/jumbo v0, "sup"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1491,1496 ****
--- 1663,1669 ----
  
      if-eqz v0, :cond_11
  
+     .line 480
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Super;
***************
*** 1501,1508 ****
  
      goto/16 :goto_0
  
      :cond_11
!     const-string v0, "sub"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 1674,1682 ----
  
      goto/16 :goto_0
  
+     .line 481
      :cond_11
!     const-string/jumbo v0, "sub"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 1510,1515 ****
--- 1684,1690 ----
  
      if-eqz v0, :cond_12
  
+     .line 482
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Sub;
***************
*** 1520,1525 ****
--- 1695,1701 ----
  
      goto/16 :goto_0
  
+     .line 483
      :cond_12
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
***************
*** 1559,1568 ****
--- 1735,1746 ----
  
      if-gt v0, v1, :cond_13
  
+     .line 486
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-static {v0}, Landroid/text/HtmlToSpannedConverter;->handleP(Landroid/text/SpannableStringBuilder;)V
  
+     .line 487
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      new-instance v1, Landroid/text/HtmlToSpannedConverter$Header;
***************
*** 1579,1584 ****
--- 1757,1763 ----
  
      goto/16 :goto_0
  
+     .line 488
      :cond_13
      const-string v0, "img"
  
***************
*** 1588,1593 ****
--- 1767,1773 ----
  
      if-eqz v0, :cond_14
  
+     .line 489
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      iget-object v1, p0, Landroid/text/HtmlToSpannedConverter;->mImageGetter:Landroid/text/Html$ImageGetter;
***************
*** 1596,1606 ****
--- 1776,1788 ----
  
      goto/16 :goto_0
  
+     .line 490
      :cond_14
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mTagHandler:Landroid/text/Html$TagHandler;
  
      if-eqz v0, :cond_0
  
+     .line 491
      iget-object v0, p0, Landroid/text/HtmlToSpannedConverter;->mTagHandler:Landroid/text/Html$TagHandler;
  
      iget-object v1, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
***************
*** 1618,1632 ****
--- 1800,1817 ----
      .parameter "mark"
  
      .prologue
+     .line 580
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v0
  
+     .line 581
      .local v0, len:I
      const/16 v1, 0x11
  
      invoke-virtual {p0, p1, v0, v0, v1}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 582
      return-void
  .end method
  
***************
*** 1636,1641 ****
--- 1821,1827 ----
      .parameter "attributes"
  
      .prologue
+     .line 669
      const-string v2, ""
  
      const-string v3, "href"
***************
*** 1644,1654 ****
--- 1830,1842 ----
  
      move-result-object v0
  
+     .line 671
      .local v0, href:Ljava/lang/String;
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v1
  
+     .line 672
      .local v1, len:I
      new-instance v2, Landroid/text/HtmlToSpannedConverter$Href;
  
***************
*** 1658,1663 ****
--- 1846,1852 ----
  
      invoke-virtual {p0, v2, v1, v1, v3}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 673
      return-void
  .end method
  
***************
*** 1667,1672 ****
--- 1856,1862 ----
      .parameter "attributes"
  
      .prologue
+     .line 623
      const-string v3, ""
  
      const-string v4, "color"
***************
*** 1675,1680 ****
--- 1865,1871 ----
  
      move-result-object v0
  
+     .line 624
      .local v0, color:Ljava/lang/String;
      const-string v3, ""
  
***************
*** 1684,1694 ****
--- 1875,1887 ----
  
      move-result-object v1
  
+     .line 626
      .local v1, face:Ljava/lang/String;
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v2
  
+     .line 627
      .local v2, len:I
      new-instance v3, Landroid/text/HtmlToSpannedConverter$Font;
  
***************
*** 1698,1703 ****
--- 1891,1897 ----
  
      invoke-virtual {p0, v3, v2, v2, v4}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 628
      return-void
  .end method
  
***************
*** 1710,1736 ****
      .prologue
      const/4 v5, 0x0
  
      const-string v3, ""
  
!     const-string v4, "src"
  
      invoke-interface {p1, v3, v4}, Lorg/xml/sax/Attributes;->getValue(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v2
  
      .local v2, src:Ljava/lang/String;
      const/4 v0, 0x0
  
      .local v0, d:Landroid/graphics/drawable/Drawable;
      if-eqz p2, :cond_0
  
      invoke-interface {p2, v2}, Landroid/text/Html$ImageGetter;->getDrawable(Ljava/lang/String;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v0
  
      :cond_0
      if-nez v0, :cond_1
  
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v3
--- 1904,1936 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 601
      const-string v3, ""
  
!     const-string/jumbo v4, "src"
  
      invoke-interface {p1, v3, v4}, Lorg/xml/sax/Attributes;->getValue(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v2
  
+     .line 602
      .local v2, src:Ljava/lang/String;
      const/4 v0, 0x0
  
+     .line 604
      .local v0, d:Landroid/graphics/drawable/Drawable;
      if-eqz p2, :cond_0
  
+     .line 605
      invoke-interface {p2, v2}, Landroid/text/Html$ImageGetter;->getDrawable(Ljava/lang/String;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v0
  
+     .line 608
      :cond_0
      if-nez v0, :cond_1
  
+     .line 609
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v3
***************
*** 1741,1746 ****
--- 1941,1947 ----
  
      move-result-object v0
  
+     .line 611
      invoke-virtual {v0}, Landroid/graphics/drawable/Drawable;->getIntrinsicWidth()I
  
      move-result v3
***************
*** 1751,1766 ****
  
      invoke-virtual {v0, v5, v5, v3, v4}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
  
      :cond_1
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v1
  
      .local v1, len:I
!     const-string v3, "\ufffc"
  
      invoke-virtual {p0, v3}, Landroid/text/SpannableStringBuilder;->append(Ljava/lang/CharSequence;)Landroid/text/SpannableStringBuilder;
  
      new-instance v3, Landroid/text/style/ImageSpan;
  
      invoke-direct {v3, v0, v2}, Landroid/text/style/ImageSpan;-><init>(Landroid/graphics/drawable/Drawable;Ljava/lang/String;)V
--- 1952,1970 ----
  
      invoke-virtual {v0, v5, v5, v3, v4}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
  
+     .line 614
      :cond_1
      invoke-virtual {p0}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v1
  
+     .line 615
      .local v1, len:I
!     const-string/jumbo v3, "\ufffc"
  
      invoke-virtual {p0, v3}, Landroid/text/SpannableStringBuilder;->append(Ljava/lang/CharSequence;)Landroid/text/SpannableStringBuilder;
  
+     .line 617
      new-instance v3, Landroid/text/style/ImageSpan;
  
      invoke-direct {v3, v0, v2}, Landroid/text/style/ImageSpan;-><init>(Landroid/graphics/drawable/Drawable;Ljava/lang/String;)V
***************
*** 1773,1778 ****
--- 1977,1983 ----
  
      invoke-virtual {p0, v3, v1, v4, v5}, Landroid/text/SpannableStringBuilder;->setSpan(Ljava/lang/Object;III)V
  
+     .line 619
      return-void
  .end method
  
***************
*** 1794,1803 ****
--- 1999,2010 ----
  
      const/16 v7, 0x20
  
+     .line 740
      new-instance v4, Ljava/lang/StringBuilder;
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 747
      .local v4, sb:Ljava/lang/StringBuilder;
      const/4 v1, 0x0
  
***************
*** 1805,1845 ****
--- 2012,2062 ----
      :goto_0
      if-ge v1, p3, :cond_5
  
+     .line 748
      add-int v5, v1, p2
  
      aget-char v0, p1, v5
  
+     .line 750
      .local v0, c:C
      if-eq v0, v7, :cond_0
  
      if-ne v0, v8, :cond_4
  
+     .line 752
      :cond_0
      invoke-virtual {v4}, Ljava/lang/StringBuilder;->length()I
  
      move-result v2
  
+     .line 754
      .local v2, len:I
      if-nez v2, :cond_3
  
+     .line 755
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-virtual {v5}, Landroid/text/SpannableStringBuilder;->length()I
  
      move-result v2
  
+     .line 757
      if-nez v2, :cond_2
  
+     .line 758
      const/16 v3, 0xa
  
+     .line 766
      .local v3, pred:C
      :goto_1
      if-eq v3, v7, :cond_1
  
      if-eq v3, v8, :cond_1
  
+     .line 767
      invoke-virtual {v4, v7}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 747
      .end local v2           #len:I
      .end local v3           #pred:C
      :cond_1
***************
*** 1848,1853 ****
--- 2065,2071 ----
  
      goto :goto_0
  
+     .line 760
      .restart local v2       #len:I
      :cond_2
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
***************
*** 1861,1866 ****
--- 2079,2085 ----
      .restart local v3       #pred:C
      goto :goto_1
  
+     .line 763
      .end local v3           #pred:C
      :cond_3
      add-int/lit8 v5, v2, -0x1
***************
*** 1872,1877 ****
--- 2091,2097 ----
      .restart local v3       #pred:C
      goto :goto_1
  
+     .line 770
      .end local v2           #len:I
      .end local v3           #pred:C
      :cond_4
***************
*** 1879,1890 ****
--- 2099,2112 ----
  
      goto :goto_2
  
+     .line 774
      .end local v0           #c:C
      :cond_5
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      invoke-virtual {v5, v4}, Landroid/text/SpannableStringBuilder;->append(Ljava/lang/CharSequence;)Landroid/text/SpannableStringBuilder;
  
+     .line 775
      return-void
  .end method
  
***************
*** 1894,1903 ****
--- 2116,2127 ----
      .prologue
      const/16 v9, 0xa
  
+     .line 409
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mReader:Lorg/xml/sax/XMLReader;
  
      invoke-interface {v5, p0}, Lorg/xml/sax/XMLReader;->setContentHandler(Lorg/xml/sax/ContentHandler;)V
  
+     .line 411
      :try_start_0
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mReader:Lorg/xml/sax/XMLReader;
  
***************
*** 1916,1921 ****
--- 2140,2146 ----
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Lorg/xml/sax/SAXException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 421
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      const/4 v6, 0x0
***************
*** 1932,1937 ****
--- 2157,2163 ----
  
      move-result-object v3
  
+     .line 422
      .local v3, obj:[Ljava/lang/Object;
      const/4 v2, 0x0
  
***************
*** 1941,1946 ****
--- 2167,2173 ----
  
      if-ge v2, v5, :cond_2
  
+     .line 423
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      aget-object v6, v3, v2
***************
*** 1949,1954 ****
--- 2176,2182 ----
  
      move-result v4
  
+     .line 424
      .local v4, start:I
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
***************
*** 1958,1968 ****
--- 2186,2198 ----
  
      move-result v1
  
+     .line 427
      .local v1, end:I
      add-int/lit8 v5, v1, -0x2
  
      if-ltz v5, :cond_0
  
+     .line 428
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      add-int/lit8 v6, v1, -0x1
***************
*** 1983,2004 ****
--- 2213,2239 ----
  
      if-ne v5, v9, :cond_0
  
+     .line 430
      add-int/lit8 v1, v1, -0x1
  
+     .line 434
      :cond_0
      if-ne v1, v4, :cond_1
  
+     .line 435
      iget-object v5, p0, Landroid/text/HtmlToSpannedConverter;->mSpannableStringBuilder:Landroid/text/SpannableStringBuilder;
  
      aget-object v6, v3, v2
  
      invoke-virtual {v5, v6}, Landroid/text/SpannableStringBuilder;->removeSpan(Ljava/lang/Object;)V
  
+     .line 422
      :goto_1
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 412
      .end local v1           #end:I
      .end local v2           #i:I
      .end local v3           #obj:[Ljava/lang/Object;
***************
*** 2006,2011 ****
--- 2241,2247 ----
      :catch_0
      move-exception v0
  
+     .line 414
      .local v0, e:Ljava/io/IOException;
      new-instance v5, Ljava/lang/RuntimeException;
  
***************
*** 2013,2022 ****
--- 2249,2260 ----
  
      throw v5
  
+     .line 415
      .end local v0           #e:Ljava/io/IOException;
      :catch_1
      move-exception v0
  
+     .line 417
      .local v0, e:Lorg/xml/sax/SAXException;
      new-instance v5, Ljava/lang/RuntimeException;
  
***************
*** 2024,2029 ****
--- 2262,2268 ----
  
      throw v5
  
+     .line 437
      .end local v0           #e:Lorg/xml/sax/SAXException;
      .restart local v1       #end:I
      .restart local v2       #i:I
***************
*** 2040,2045 ****
--- 2279,2285 ----
  
      goto :goto_1
  
+     .line 441
      .end local v1           #end:I
      .end local v4           #start:I
      :cond_2
***************
*** 2057,2062 ****
--- 2297,2303 ----
      .end annotation
  
      .prologue
+     .line 722
      return-void
  .end method
  
***************
*** 2072,2079 ****
--- 2313,2322 ----
      .end annotation
  
      .prologue
+     .line 736
      invoke-direct {p0, p2}, Landroid/text/HtmlToSpannedConverter;->handleEndTag(Ljava/lang/String;)V
  
+     .line 737
      return-void
  .end method
  
***************
*** 2087,2092 ****
--- 2330,2336 ----
      .end annotation
  
      .prologue
+     .line 728
      return-void
  .end method
  
***************
*** 2102,2107 ****
--- 2346,2352 ----
      .end annotation
  
      .prologue
+     .line 778
      return-void
  .end method
  
***************
*** 2116,2121 ****
--- 2361,2367 ----
      .end annotation
  
      .prologue
+     .line 781
      return-void
  .end method
  
***************
*** 2124,2129 ****
--- 2370,2376 ----
      .parameter "locator"
  
      .prologue
+     .line 716
      return-void
  .end method
  
***************
*** 2137,2142 ****
--- 2384,2390 ----
      .end annotation
  
      .prologue
+     .line 784
      return-void
  .end method
  
***************
*** 2149,2154 ****
--- 2397,2403 ----
      .end annotation
  
      .prologue
+     .line 719
      return-void
  .end method
  
***************
*** 2165,2172 ****
--- 2414,2423 ----
      .end annotation
  
      .prologue
+     .line 732
      invoke-direct {p0, p2, p4}, Landroid/text/HtmlToSpannedConverter;->handleStartTag(Ljava/lang/String;Lorg/xml/sax/Attributes;)V
  
+     .line 733
      return-void
  .end method
  
***************
*** 2181,2185 ****
--- 2432,2437 ----
      .end annotation
  
      .prologue
+     .line 725
      return-void
  .end method
