*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/InputFilter$AllCaps.smali	2012-03-23 19:35:19.271938309 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/InputFilter$AllCaps.smali.original	2012-03-23 19:34:44.481303903 -0500
***************
*** 22,27 ****
--- 22,28 ----
      .locals 0
  
      .prologue
+     .line 49
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 43,54 ****
--- 44,57 ----
  
      const/4 v5, 0x0
  
+     .line 52
      move v6, p2
  
      .local v6, i:I
      :goto_0
      if-ge v6, p3, :cond_2
  
+     .line 53
      invoke-interface {p1, v6}, Ljava/lang/CharSequence;->charAt(I)C
  
      move-result v0
***************
*** 59,71 ****
--- 62,77 ----
  
      if-eqz v0, :cond_1
  
+     .line 54
      sub-int v0, p3, p2
  
      new-array v8, v0, [C
  
+     .line 55
      .local v8, v:[C
      invoke-static {p1, p2, p3, v8, v5}, Landroid/text/TextUtils;->getChars(Ljava/lang/CharSequence;II[CI)V
  
+     .line 56
      new-instance v0, Ljava/lang/String;
  
      invoke-direct {v0, v8}, Ljava/lang/String;-><init>([C)V
***************
*** 74,84 ****
--- 80,92 ----
  
      move-result-object v7
  
+     .line 58
      .local v7, s:Ljava/lang/String;
      instance-of v0, p1, Landroid/text/Spanned;
  
      if-eqz v0, :cond_0
  
+     .line 59
      new-instance v4, Landroid/text/SpannableString;
  
      invoke-direct {v4, v7}, Landroid/text/SpannableString;-><init>(Ljava/lang/CharSequence;)V
***************
*** 86,91 ****
--- 94,100 ----
      .local v4, sp:Landroid/text/SpannableString;
      move-object v0, p1
  
+     .line 60
      check-cast v0, Landroid/text/Spanned;
  
      move v1, p2
***************
*** 94,99 ****
--- 103,109 ----
  
      invoke-static/range {v0 .. v5}, Landroid/text/TextUtils;->copySpansFrom(Landroid/text/Spanned;IILjava/lang/Class;Landroid/text/Spannable;I)V
  
+     .line 69
      .end local v4           #sp:Landroid/text/SpannableString;
      .end local v7           #s:Ljava/lang/String;
      .end local v8           #v:[C
***************
*** 105,112 ****
--- 115,124 ----
      :cond_0
      move-object v4, v7
  
+     .line 64
      goto :goto_1
  
+     .line 52
      .end local v7           #s:Ljava/lang/String;
      .end local v8           #v:[C
      :cond_1
***************
*** 117,121 ****
--- 129,134 ----
      :cond_2
      move-object v4, v3
  
+     .line 69
      goto :goto_1
  .end method
