*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/Layout$Directions.smali	2012-03-23 19:35:21.571980251 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/Layout$Directions.smali.original	2012-03-23 19:34:44.513304486 -0500
***************
*** 24,32 ****
--- 24,35 ----
      .parameter "dirs"
  
      .prologue
+     .line 1713
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1714
      iput-object p1, p0, Landroid/text/Layout$Directions;->mDirections:[I
  
+     .line 1715
      return-void
  .end method
