*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/PackedObjectVector.smali	2012-03-23 19:35:18.401922444 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/PackedObjectVector.smali.original	2012-03-23 19:34:44.472303739 -0500
***************
*** 35,44 ****
--- 35,47 ----
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      const/4 v2, 0x0
  
+     .line 33
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 34
      iput p1, p0, Landroid/text/PackedObjectVector;->mColumns:I
  
+     .line 35
      invoke-static {v2}, Lcom/android/internal/util/ArrayUtils;->idealIntArraySize(I)I
  
      move-result v0
***************
*** 49,60 ****
--- 52,66 ----
  
      iput v0, p0, Landroid/text/PackedObjectVector;->mRows:I
  
+     .line 37
      iput v2, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
+     .line 38
      iget v0, p0, Landroid/text/PackedObjectVector;->mRows:I
  
      iput v0, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
+     .line 40
      iget v0, p0, Landroid/text/PackedObjectVector;->mRows:I
  
      iget v1, p0, Landroid/text/PackedObjectVector;->mColumns:I
***************
*** 65,70 ****
--- 71,77 ----
  
      iput-object v0, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
+     .line 41
      return-void
  .end method
  
***************
*** 75,86 ****
--- 82,95 ----
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      const/4 v6, 0x0
  
+     .line 112
      invoke-virtual {p0}, Landroid/text/PackedObjectVector;->size()I
  
      move-result v3
  
      add-int/lit8 v1, v3, 0x1
  
+     .line 113
      .local v1, newsize:I
      iget v3, p0, Landroid/text/PackedObjectVector;->mColumns:I
  
***************
*** 94,105 ****
--- 103,116 ----
  
      div-int v1, v3, v4
  
+     .line 114
      iget v3, p0, Landroid/text/PackedObjectVector;->mColumns:I
  
      mul-int/2addr v3, v1
  
      new-array v2, v3, [Ljava/lang/Object;
  
+     .line 116
      .local v2, newvalues:[Ljava/lang/Object;
      iget v3, p0, Landroid/text/PackedObjectVector;->mRows:I
  
***************
*** 111,116 ****
--- 122,128 ----
  
      sub-int v0, v3, v4
  
+     .line 118
      .local v0, after:I
      iget-object v3, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
***************
*** 122,127 ****
--- 134,140 ----
  
      invoke-static {v3, v6, v2, v6, v4}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 119
      iget-object v3, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
      iget v4, p0, Landroid/text/PackedObjectVector;->mRows:I
***************
*** 144,149 ****
--- 157,163 ----
  
      invoke-static {v3, v4, v2, v5, v6}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 121
      iget v3, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
      iget v4, p0, Landroid/text/PackedObjectVector;->mRows:I
***************
*** 154,163 ****
--- 168,180 ----
  
      iput v3, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
+     .line 122
      iput v1, p0, Landroid/text/PackedObjectVector;->mRows:I
  
+     .line 123
      iput-object v2, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
+     .line 124
      return-void
  .end method
  
***************
*** 166,184 ****
--- 183,205 ----
      .parameter "where"
  
      .prologue
+     .line 129
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      iget v5, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
      if-ne p1, v5, :cond_0
  
+     .line 166
      :goto_0
      return-void
  
+     .line 132
      :cond_0
      iget v5, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
      if-le p1, v5, :cond_2
  
+     .line 134
      iget v5, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
      add-int/2addr v5, p1
***************
*** 191,196 ****
--- 212,218 ----
  
      sub-int v3, v5, v6
  
+     .line 136
      .local v3, moving:I
      iget v5, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
***************
*** 210,215 ****
--- 232,238 ----
  
      if-ge v1, v5, :cond_4
  
+     .line 138
      iget v5, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
      iget v6, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
***************
*** 222,227 ****
--- 245,251 ----
  
      add-int v0, v5, v6
  
+     .line 140
      .local v0, destrow:I
      const/4 v2, 0x0
  
***************
*** 231,236 ****
--- 255,261 ----
  
      if-ge v2, v5, :cond_1
  
+     .line 142
      iget-object v5, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
      iget v6, p0, Landroid/text/PackedObjectVector;->mColumns:I
***************
*** 241,246 ****
--- 266,272 ----
  
      aget-object v4, v5, v6
  
+     .line 144
      .local v4, val:Ljava/lang/Object;
      iget-object v5, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
***************
*** 252,267 ****
--- 278,296 ----
  
      aput-object v4, v5, v6
  
+     .line 140
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_2
  
+     .line 136
      .end local v4           #val:Ljava/lang/Object;
      :cond_1
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 150
      .end local v0           #destrow:I
      .end local v1           #i:I
      .end local v2           #j:I
***************
*** 271,276 ****
--- 300,306 ----
  
      sub-int v3, v5, p1
  
+     .line 152
      .restart local v3       #moving:I
      add-int v5, p1, v3
  
***************
*** 280,285 ****
--- 310,316 ----
      :goto_3
      if-lt v1, p1, :cond_4
  
+     .line 154
      sub-int v5, v1, p1
  
      iget v6, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
***************
*** 292,297 ****
--- 323,329 ----
  
      sub-int v0, v5, v3
  
+     .line 156
      .restart local v0       #destrow:I
      const/4 v2, 0x0
  
***************
*** 301,306 ****
--- 333,339 ----
  
      if-ge v2, v5, :cond_3
  
+     .line 158
      iget-object v5, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
      iget v6, p0, Landroid/text/PackedObjectVector;->mColumns:I
***************
*** 311,316 ****
--- 344,350 ----
  
      aget-object v4, v5, v6
  
+     .line 160
      .restart local v4       #val:Ljava/lang/Object;
      iget-object v5, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
***************
*** 322,337 ****
--- 356,374 ----
  
      aput-object v4, v5, v6
  
+     .line 156
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_4
  
+     .line 152
      .end local v4           #val:Ljava/lang/Object;
      :cond_3
      add-int/lit8 v1, v1, -0x1
  
      goto :goto_3
  
+     .line 165
      .end local v0           #destrow:I
      .end local v2           #j:I
      :cond_4
***************
*** 348,370 ****
--- 385,411 ----
      .parameter "count"
  
      .prologue
+     .line 85
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      add-int v0, p1, p2
  
      invoke-direct {p0, v0}, Landroid/text/PackedObjectVector;->moveRowGapTo(I)V
  
+     .line 87
      iget v0, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
      sub-int/2addr v0, p2
  
      iput v0, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
+     .line 88
      iget v0, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
      add-int/2addr v0, p2
  
      iput v0, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
+     .line 90
      iget v0, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
      invoke-virtual {p0}, Landroid/text/PackedObjectVector;->size()I
***************
*** 375,380 ****
--- 416,422 ----
  
      if-le v0, v1, :cond_0
  
+     .line 95
      :cond_0
      return-void
  .end method
***************
*** 383,388 ****
--- 425,431 ----
      .locals 6
  
      .prologue
+     .line 171
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      const/4 v0, 0x0
  
***************
*** 392,397 ****
--- 435,441 ----
  
      if-ge v0, v3, :cond_3
  
+     .line 173
      const/4 v1, 0x0
  
      .local v1, j:I
***************
*** 400,405 ****
--- 444,450 ----
  
      if-ge v1, v3, :cond_2
  
+     .line 175
      iget-object v3, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
      iget v4, p0, Landroid/text/PackedObjectVector;->mColumns:I
***************
*** 410,415 ****
--- 455,461 ----
  
      aget-object v2, v3, v4
  
+     .line 177
      .local v2, val:Ljava/lang/Object;
      iget v3, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
***************
*** 423,428 ****
--- 469,475 ----
  
      if-lt v0, v3, :cond_1
  
+     .line 178
      :cond_0
      sget-object v3, Ljava/lang/System;->out:Ljava/io/PrintStream;
  
***************
*** 446,456 ****
--- 493,505 ----
  
      invoke-virtual {v3, v4}, Ljava/io/PrintStream;->print(Ljava/lang/String;)V
  
+     .line 173
      :goto_2
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 180
      :cond_1
      sget-object v3, Ljava/lang/System;->out:Ljava/io/PrintStream;
  
***************
*** 482,487 ****
--- 531,537 ----
  
      goto :goto_2
  
+     .line 183
      .end local v2           #val:Ljava/lang/Object;
      :cond_2
      sget-object v3, Ljava/lang/System;->out:Ljava/io/PrintStream;
***************
*** 490,499 ****
--- 540,551 ----
  
      invoke-virtual {v3, v4}, Ljava/io/PrintStream;->print(Ljava/lang/String;)V
  
+     .line 171
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 186
      .end local v1           #j:I
      :cond_3
      sget-object v3, Ljava/lang/System;->out:Ljava/io/PrintStream;
***************
*** 502,507 ****
--- 554,560 ----
  
      invoke-virtual {v3, v4}, Ljava/io/PrintStream;->print(Ljava/lang/String;)V
  
+     .line 187
      return-void
  .end method
  
***************
*** 516,530 ****
--- 569,586 ----
      .end annotation
  
      .prologue
+     .line 46
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      iget v1, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
      if-lt p1, v1, :cond_0
  
+     .line 47
      iget v1, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
      add-int/2addr p1, v1
  
+     .line 49
      :cond_0
      iget-object v1, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
***************
*** 536,541 ****
--- 592,598 ----
  
      aget-object v0, v1, v2
  
+     .line 51
      .local v0, value:Ljava/lang/Object;
      return-object v0
  .end method
***************
*** 551,566 ****
--- 608,627 ----
      .end annotation
  
      .prologue
+     .line 66
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      .local p2, values:[Ljava/lang/Object;,"[TE;"
      invoke-direct {p0, p1}, Landroid/text/PackedObjectVector;->moveRowGapTo(I)V
  
+     .line 68
      iget v1, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
      if-nez v1, :cond_0
  
+     .line 69
      invoke-direct {p0}, Landroid/text/PackedObjectVector;->growBuffer()V
  
+     .line 71
      :cond_0
      iget v1, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
***************
*** 568,581 ****
--- 629,645 ----
  
      iput v1, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
+     .line 72
      iget v1, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
      add-int/lit8 v1, v1, -0x1
  
      iput v1, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
+     .line 74
      if-nez p2, :cond_1
  
+     .line 75
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 584,597 ****
--- 648,664 ----
  
      if-ge v0, v1, :cond_2
  
+     .line 76
      const/4 v1, 0x0
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/text/PackedObjectVector;->setValue(IILjava/lang/Object;)V
  
+     .line 75
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 78
      .end local v0           #i:I
      :cond_1
      const/4 v0, 0x0
***************
*** 602,615 ****
--- 669,685 ----
  
      if-ge v0, v1, :cond_2
  
+     .line 79
      aget-object v1, p2, v0
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/text/PackedObjectVector;->setValue(IILjava/lang/Object;)V
  
+     .line 78
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 80
      :cond_2
      return-void
  .end method
***************
*** 626,641 ****
--- 696,714 ----
      .end annotation
  
      .prologue
+     .line 57
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      .local p3, value:Ljava/lang/Object;,"TE;"
      iget v0, p0, Landroid/text/PackedObjectVector;->mRowGapStart:I
  
      if-lt p1, v0, :cond_0
  
+     .line 58
      iget v0, p0, Landroid/text/PackedObjectVector;->mRowGapLength:I
  
      add-int/2addr p1, v0
  
+     .line 60
      :cond_0
      iget-object v0, p0, Landroid/text/PackedObjectVector;->mValues:[Ljava/lang/Object;
  
***************
*** 647,652 ****
--- 720,726 ----
  
      aput-object p3, v0, v1
  
+     .line 61
      return-void
  .end method
  
***************
*** 654,659 ****
--- 728,734 ----
      .locals 2
  
      .prologue
+     .line 100
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      iget v0, p0, Landroid/text/PackedObjectVector;->mRows:I
  
***************
*** 668,673 ****
--- 743,749 ----
      .locals 1
  
      .prologue
+     .line 106
      .local p0, this:Landroid/text/PackedObjectVector;,"Landroid/text/PackedObjectVector<TE;>;"
      iget v0, p0, Landroid/text/PackedObjectVector;->mColumns:I
  
