*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/PackedIntVector.smali	2012-03-23 19:35:19.446941500 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/PackedIntVector.smali.original	2012-03-23 19:34:44.483303939 -0500
***************
*** 25,52 ****
--- 25,60 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 43
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 44
      iput p1, p0, Landroid/text/PackedIntVector;->mColumns:I
  
+     .line 45
      iput v0, p0, Landroid/text/PackedIntVector;->mRows:I
  
+     .line 47
      iput v0, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
+     .line 48
      iget v0, p0, Landroid/text/PackedIntVector;->mRows:I
  
      iput v0, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
+     .line 50
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/text/PackedIntVector;->mValues:[I
  
+     .line 51
      mul-int/lit8 v0, p1, 0x2
  
      new-array v0, v0, [I
  
      iput-object v0, p0, Landroid/text/PackedIntVector;->mValueGap:[I
  
+     .line 52
      return-void
  .end method
  
***************
*** 56,63 ****
--- 64,73 ----
      .prologue
      const/4 v9, 0x0
  
+     .line 254
      iget v1, p0, Landroid/text/PackedIntVector;->mColumns:I
  
+     .line 255
      .local v1, columns:I
      invoke-virtual {p0}, Landroid/text/PackedIntVector;->size()I
  
***************
*** 65,70 ****
--- 75,81 ----
  
      add-int/lit8 v3, v7, 0x1
  
+     .line 256
      .local v3, newsize:I
      mul-int v7, v3, v1
  
***************
*** 74,89 ****
--- 85,104 ----
  
      div-int v3, v7, v1
  
+     .line 257
      mul-int v7, v3, v1
  
      new-array v4, v7, [I
  
+     .line 259
      .local v4, newvalues:[I
      iget-object v6, p0, Landroid/text/PackedIntVector;->mValueGap:[I
  
+     .line 260
      .local v6, valuegap:[I
      iget v5, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
+     .line 262
      .local v5, rowgapstart:I
      iget v7, p0, Landroid/text/PackedIntVector;->mRows:I
  
***************
*** 93,109 ****
--- 108,127 ----
  
      sub-int v0, v7, v8
  
+     .line 264
      .local v0, after:I
      iget-object v7, p0, Landroid/text/PackedIntVector;->mValues:[I
  
      if-eqz v7, :cond_0
  
+     .line 265
      iget-object v7, p0, Landroid/text/PackedIntVector;->mValues:[I
  
      mul-int v8, v1, v5
  
      invoke-static {v7, v9, v4, v9, v8}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 266
      iget-object v7, p0, Landroid/text/PackedIntVector;->mValues:[I
  
      iget v8, p0, Landroid/text/PackedIntVector;->mRows:I
***************
*** 120,125 ****
--- 138,144 ----
  
      invoke-static {v7, v8, v4, v9, v10}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 271
      :cond_0
      const/4 v2, 0x0
  
***************
*** 127,136 ****
--- 146,157 ----
      :goto_0
      if-ge v2, v1, :cond_2
  
+     .line 272
      aget v7, v6, v2
  
      if-lt v7, v5, :cond_1
  
+     .line 273
      aget v7, v6, v2
  
      iget v8, p0, Landroid/text/PackedIntVector;->mRows:I
***************
*** 141,157 ****
--- 162,182 ----
  
      aput v7, v6, v2
  
+     .line 275
      aget v7, v6, v2
  
      if-ge v7, v5, :cond_1
  
+     .line 276
      aput v5, v6, v2
  
+     .line 271
      :cond_1
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 281
      :cond_2
      iget v7, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
***************
*** 163,172 ****
--- 188,200 ----
  
      iput v7, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
+     .line 282
      iput v3, p0, Landroid/text/PackedIntVector;->mRows:I
  
+     .line 283
      iput-object v4, p0, Landroid/text/PackedIntVector;->mValues:[I
  
+     .line 284
      return-void
  .end method
  
***************
*** 175,192 ****
--- 203,224 ----
      .parameter "where"
  
      .prologue
+     .line 314
      iget v9, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
      if-ne p1, v9, :cond_0
  
+     .line 367
      :goto_0
      return-void
  
+     .line 316
      :cond_0
      iget v9, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
      if-le p1, v9, :cond_4
  
+     .line 317
      iget v9, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
      add-int/2addr v9, p1
***************
*** 199,213 ****
--- 231,249 ----
  
      sub-int v5, v9, v10
  
+     .line 318
      .local v5, moving:I
      iget v0, p0, Landroid/text/PackedIntVector;->mColumns:I
  
+     .line 319
      .local v0, columns:I
      iget-object v7, p0, Landroid/text/PackedIntVector;->mValueGap:[I
  
+     .line 320
      .local v7, valuegap:[I
      iget-object v8, p0, Landroid/text/PackedIntVector;->mValues:[I
  
+     .line 321
      .local v8, values:[I
      iget v9, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
***************
*** 215,220 ****
--- 251,257 ----
  
      add-int v2, v9, v10
  
+     .line 323
      .local v2, gapend:I
      move v3, v2
  
***************
*** 224,235 ****
--- 261,274 ----
  
      if-ge v3, v9, :cond_8
  
+     .line 324
      sub-int v9, v3, v2
  
      iget v10, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
      add-int v1, v9, v10
  
+     .line 326
      .local v1, destrow:I
      const/4 v4, 0x0
  
***************
*** 237,270 ****
--- 276,315 ----
      :goto_2
      if-ge v4, v0, :cond_3
  
+     .line 327
      mul-int v9, v3, v0
  
      add-int/2addr v9, v4
  
      aget v6, v8, v9
  
+     .line 329
      .local v6, val:I
      aget v9, v7, v4
  
      if-lt v3, v9, :cond_1
  
+     .line 330
      add-int v9, v4, v0
  
      aget v9, v7, v9
  
      add-int/2addr v6, v9
  
+     .line 333
      :cond_1
      aget v9, v7, v4
  
      if-lt v1, v9, :cond_2
  
+     .line 334
      add-int v9, v4, v0
  
      aget v9, v7, v9
  
      sub-int/2addr v6, v9
  
+     .line 337
      :cond_2
      mul-int v9, v1, v0
  
***************
*** 272,287 ****
--- 317,335 ----
  
      aput v6, v8, v9
  
+     .line 326
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_2
  
+     .line 323
      .end local v6           #val:I
      :cond_3
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_1
  
+     .line 341
      .end local v0           #columns:I
      .end local v1           #destrow:I
      .end local v2           #gapend:I
***************
*** 295,309 ****
--- 343,361 ----
  
      sub-int v5, v9, p1
  
+     .line 342
      .restart local v5       #moving:I
      iget v0, p0, Landroid/text/PackedIntVector;->mColumns:I
  
+     .line 343
      .restart local v0       #columns:I
      iget-object v7, p0, Landroid/text/PackedIntVector;->mValueGap:[I
  
+     .line 344
      .restart local v7       #valuegap:[I
      iget-object v8, p0, Landroid/text/PackedIntVector;->mValues:[I
  
+     .line 345
      .restart local v8       #values:[I
      iget v9, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
***************
*** 311,316 ****
--- 363,369 ----
  
      add-int v2, v9, v10
  
+     .line 347
      .restart local v2       #gapend:I
      add-int v9, p1, v5
  
***************
*** 320,331 ****
--- 373,386 ----
      :goto_3
      if-lt v3, p1, :cond_8
  
+     .line 348
      sub-int v9, v3, p1
  
      add-int/2addr v9, v2
  
      sub-int v1, v9, v5
  
+     .line 350
      .restart local v1       #destrow:I
      const/4 v4, 0x0
  
***************
*** 333,366 ****
--- 388,427 ----
      :goto_4
      if-ge v4, v0, :cond_7
  
+     .line 351
      mul-int v9, v3, v0
  
      add-int/2addr v9, v4
  
      aget v6, v8, v9
  
+     .line 353
      .restart local v6       #val:I
      aget v9, v7, v4
  
      if-lt v3, v9, :cond_5
  
+     .line 354
      add-int v9, v4, v0
  
      aget v9, v7, v9
  
      add-int/2addr v6, v9
  
+     .line 357
      :cond_5
      aget v9, v7, v4
  
      if-lt v1, v9, :cond_6
  
+     .line 358
      add-int v9, v4, v0
  
      aget v9, v7, v9
  
      sub-int/2addr v6, v9
  
+     .line 361
      :cond_6
      mul-int v9, v1, v0
  
***************
*** 368,383 ****
--- 429,447 ----
  
      aput v6, v8, v9
  
+     .line 350
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_4
  
+     .line 347
      .end local v6           #val:I
      :cond_7
      add-int/lit8 v3, v3, -0x1
  
      goto :goto_3
  
+     .line 366
      .end local v1           #destrow:I
      .end local v4           #j:I
      :cond_8
***************
*** 392,424 ****
--- 456,496 ----
      .parameter "where"
  
      .prologue
+     .line 291
      iget-object v2, p0, Landroid/text/PackedIntVector;->mValueGap:[I
  
+     .line 292
      .local v2, valuegap:[I
      iget-object v3, p0, Landroid/text/PackedIntVector;->mValues:[I
  
+     .line 293
      .local v3, values:[I
      iget v0, p0, Landroid/text/PackedIntVector;->mColumns:I
  
+     .line 295
      .local v0, columns:I
      aget v4, v2, p1
  
      if-ne p2, v4, :cond_0
  
+     .line 308
      :goto_0
      return-void
  
+     .line 297
      :cond_0
      aget v4, v2, p1
  
      if-le p2, v4, :cond_1
  
+     .line 298
      aget v1, v2, p1
  
      .local v1, i:I
      :goto_1
      if-ge v1, p2, :cond_2
  
+     .line 299
      mul-int v4, v1, v0
  
      add-int/2addr v4, p1
***************
*** 433,442 ****
--- 505,516 ----
  
      aput v5, v3, v4
  
+     .line 298
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 302
      .end local v1           #i:I
      :cond_1
      move v1, p2
***************
*** 447,452 ****
--- 521,527 ----
  
      if-ge v1, v4, :cond_2
  
+     .line 303
      mul-int v4, v1, v0
  
      add-int/2addr v4, p1
***************
*** 461,470 ****
--- 536,547 ----
  
      aput v5, v3, v4
  
+     .line 302
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_2
  
+     .line 307
      :cond_2
      aput p2, v2, p1
  
***************
*** 478,499 ****
--- 555,581 ----
      .parameter "value"
  
      .prologue
+     .line 123
      iget v1, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
      if-lt p1, v1, :cond_0
  
+     .line 124
      iget v1, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
      add-int/2addr p1, v1
  
+     .line 127
      :cond_0
      iget-object v0, p0, Landroid/text/PackedIntVector;->mValueGap:[I
  
+     .line 128
      .local v0, valuegap:[I
      aget v1, v0, p2
  
      if-lt p1, v1, :cond_1
  
+     .line 129
      iget v1, p0, Landroid/text/PackedIntVector;->mColumns:I
  
      add-int/2addr v1, p2
***************
*** 502,507 ****
--- 584,590 ----
  
      sub-int/2addr p3, v1
  
+     .line 132
      :cond_1
      iget-object v1, p0, Landroid/text/PackedIntVector;->mValues:[I
  
***************
*** 513,518 ****
--- 596,602 ----
  
      aput p3, v1, v2
  
+     .line 133
      return-void
  .end method
  
***************
*** 525,530 ****
--- 609,615 ----
      .parameter "delta"
  
      .prologue
+     .line 149
      or-int v0, p1, p2
  
      if-ltz v0, :cond_0
***************
*** 541,546 ****
--- 626,632 ----
  
      if-lt p2, v0, :cond_1
  
+     .line 151
      :cond_0
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 570,587 ****
--- 656,677 ----
  
      throw v0
  
+     .line 154
      :cond_1
      iget v0, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
      if-lt p1, v0, :cond_2
  
+     .line 155
      iget v0, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
      add-int/2addr p1, v0
  
+     .line 158
      :cond_2
      invoke-direct {p0, p2, p1}, Landroid/text/PackedIntVector;->moveValueGapTo(II)V
  
+     .line 159
      iget-object v0, p0, Landroid/text/PackedIntVector;->mValueGap:[I
  
      iget v1, p0, Landroid/text/PackedIntVector;->mColumns:I
***************
*** 594,599 ****
--- 684,690 ----
  
      aput v2, v0, v1
  
+     .line 160
      return-void
  .end method
  
***************
*** 603,608 ****
--- 694,700 ----
      .parameter "count"
  
      .prologue
+     .line 216
      or-int v0, p1, p2
  
      if-ltz v0, :cond_0
***************
*** 615,620 ****
--- 707,713 ----
  
      if-le v0, v1, :cond_1
  
+     .line 217
      :cond_0
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 644,666 ****
--- 737,763 ----
  
      throw v0
  
+     .line 220
      :cond_1
      add-int v0, p1, p2
  
      invoke-direct {p0, v0}, Landroid/text/PackedIntVector;->moveRowGapTo(I)V
  
+     .line 222
      iget v0, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
      sub-int/2addr v0, p2
  
      iput v0, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
+     .line 223
      iget v0, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
      add-int/2addr v0, p2
  
      iput v0, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
+     .line 227
      return-void
  .end method
  
***************
*** 670,677 ****
--- 767,776 ----
      .parameter "column"
  
      .prologue
+     .line 67
      iget v0, p0, Landroid/text/PackedIntVector;->mColumns:I
  
+     .line 69
      .local v0, columns:I
      or-int v3, p1, p2
  
***************
*** 685,690 ****
--- 784,790 ----
  
      if-lt p2, v0, :cond_1
  
+     .line 70
      :cond_0
      new-instance v3, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 714,728 ****
--- 814,831 ----
  
      throw v3
  
+     .line 73
      :cond_1
      iget v3, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
      if-lt p1, v3, :cond_2
  
+     .line 74
      iget v3, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
      add-int/2addr p1, v3
  
+     .line 77
      :cond_2
      iget-object v3, p0, Landroid/text/PackedIntVector;->mValues:[I
  
***************
*** 732,751 ****
--- 835,858 ----
  
      aget v1, v3, v4
  
+     .line 79
      .local v1, value:I
      iget-object v2, p0, Landroid/text/PackedIntVector;->mValueGap:[I
  
+     .line 80
      .local v2, valuegap:[I
      aget v3, v2, p2
  
      if-lt p1, v3, :cond_3
  
+     .line 81
      add-int v3, p2, v0
  
      aget v3, v2, v3
  
      add-int/2addr v1, v3
  
+     .line 84
      :cond_3
      return v1
  .end method
***************
*** 756,761 ****
--- 863,869 ----
      .parameter "values"
  
      .prologue
+     .line 176
      if-ltz p1, :cond_0
  
      invoke-virtual {p0}, Landroid/text/PackedIntVector;->size()I
***************
*** 764,769 ****
--- 872,878 ----
  
      if-le p1, v1, :cond_1
  
+     .line 177
      :cond_0
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 771,777 ****
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "row "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 880,886 ----
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "row "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 789,794 ****
--- 898,904 ----
  
      throw v1
  
+     .line 180
      :cond_1
      if-eqz p2, :cond_2
  
***************
*** 800,812 ****
  
      if-ge v1, v2, :cond_2
  
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "value count "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 910,923 ----
  
      if-ge v1, v2, :cond_2
  
+     .line 181
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "value count "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 826,840 ****
--- 937,955 ----
  
      throw v1
  
+     .line 184
      :cond_2
      invoke-direct {p0, p1}, Landroid/text/PackedIntVector;->moveRowGapTo(I)V
  
+     .line 186
      iget v1, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
      if-nez v1, :cond_3
  
+     .line 187
      invoke-direct {p0}, Landroid/text/PackedIntVector;->growBuffer()V
  
+     .line 190
      :cond_3
      iget v1, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
***************
*** 842,855 ****
--- 957,973 ----
  
      iput v1, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
+     .line 191
      iget v1, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
      add-int/lit8 v1, v1, -0x1
  
      iput v1, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
+     .line 193
      if-nez p2, :cond_4
  
+     .line 194
      iget v1, p0, Landroid/text/PackedIntVector;->mColumns:I
  
      add-int/lit8 v0, v1, -0x1
***************
*** 858,871 ****
--- 976,992 ----
      :goto_0
      if-ltz v0, :cond_5
  
+     .line 195
      const/4 v1, 0x0
  
      invoke-direct {p0, p1, v0, v1}, Landroid/text/PackedIntVector;->setValueInternal(III)V
  
+     .line 194
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_0
  
+     .line 198
      .end local v0           #i:I
      :cond_4
      iget v1, p0, Landroid/text/PackedIntVector;->mColumns:I
***************
*** 876,889 ****
--- 997,1013 ----
      :goto_1
      if-ltz v0, :cond_5
  
+     .line 199
      aget v1, p2, v0
  
      invoke-direct {p0, p1, v0, v1}, Landroid/text/PackedIntVector;->setValueInternal(III)V
  
+     .line 198
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_1
  
+     .line 202
      :cond_5
      return-void
  .end method
***************
*** 895,900 ****
--- 1019,1025 ----
      .parameter "value"
  
      .prologue
+     .line 98
      or-int v1, p1, p2
  
      if-ltz v1, :cond_0
***************
*** 909,914 ****
--- 1034,1040 ----
  
      if-lt p2, v1, :cond_1
  
+     .line 99
      :cond_0
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 938,960 ****
--- 1064,1091 ----
  
      throw v1
  
+     .line 102
      :cond_1
      iget v1, p0, Landroid/text/PackedIntVector;->mRowGapStart:I
  
      if-lt p1, v1, :cond_2
  
+     .line 103
      iget v1, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
  
      add-int/2addr p1, v1
  
+     .line 106
      :cond_2
      iget-object v0, p0, Landroid/text/PackedIntVector;->mValueGap:[I
  
+     .line 107
      .local v0, valuegap:[I
      aget v1, v0, p2
  
      if-lt p1, v1, :cond_3
  
+     .line 108
      iget v1, p0, Landroid/text/PackedIntVector;->mColumns:I
  
      add-int/2addr v1, p2
***************
*** 963,968 ****
--- 1094,1100 ----
  
      sub-int/2addr p3, v1
  
+     .line 111
      :cond_3
      iget-object v1, p0, Landroid/text/PackedIntVector;->mValues:[I
  
***************
*** 974,979 ****
--- 1106,1112 ----
  
      aput p3, v1, v2
  
+     .line 112
      return-void
  .end method
  
***************
*** 981,986 ****
--- 1114,1120 ----
      .locals 2
  
      .prologue
+     .line 236
      iget v0, p0, Landroid/text/PackedIntVector;->mRows:I
  
      iget v1, p0, Landroid/text/PackedIntVector;->mRowGapLength:I
***************
*** 994,999 ****
--- 1128,1134 ----
      .locals 1
  
      .prologue
+     .line 246
      iget v0, p0, Landroid/text/PackedIntVector;->mColumns:I
  
      return v0
