*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/SpannableStringInternal.smali	2012-03-23 19:35:19.480942120 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/SpannableStringInternal.smali.original	2012-03-23 19:34:44.484303957 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .locals 1
  
      .prologue
+     .line 366
      const/4 v0, 0x0
  
      new-array v0, v0, [Ljava/lang/Object;
***************
*** 46,53 ****
--- 47,56 ----
      .parameter "end"
  
      .prologue
+     .line 26
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 27
      if-nez p2, :cond_2
  
      invoke-interface {p1}, Ljava/lang/CharSequence;->length()I
***************
*** 56,67 ****
--- 59,72 ----
  
      if-ne p3, v7, :cond_2
  
+     .line 28
      invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
      move-result-object v7
  
      iput-object v7, p0, Landroid/text/SpannableStringInternal;->mText:Ljava/lang/String;
  
+     .line 32
      :goto_0
      const/4 v7, 0x0
  
***************
*** 69,93 ****
--- 74,103 ----
  
      move-result v3
  
+     .line 33
      .local v3, initial:I
      new-array v7, v3, [Ljava/lang/Object;
  
      iput-object v7, p0, Landroid/text/SpannableStringInternal;->mSpans:[Ljava/lang/Object;
  
+     .line 34
      mul-int/lit8 v7, v3, 0x3
  
      new-array v7, v7, [I
  
      iput-object v7, p0, Landroid/text/SpannableStringInternal;->mSpanData:[I
  
+     .line 36
      instance-of v7, p1, Landroid/text/Spanned;
  
      if-eqz v7, :cond_3
  
      move-object v4, p1
  
+     .line 37
      check-cast v4, Landroid/text/Spanned;
  
+     .line 38
      .local v4, sp:Landroid/text/Spanned;
      const-class v7, Ljava/lang/Object;
  
***************
*** 95,100 ****
--- 105,111 ----
  
      move-result-object v5
  
+     .line 40
      .local v5, spans:[Ljava/lang/Object;
      const/4 v2, 0x0
  
***************
*** 104,115 ****
--- 115,128 ----
  
      if-ge v2, v7, :cond_3
  
+     .line 41
      aget-object v7, v5, v2
  
      invoke-interface {v4, v7}, Landroid/text/Spanned;->getSpanStart(Ljava/lang/Object;)I
  
      move-result v6
  
+     .line 42
      .local v6, st:I
      aget-object v7, v5, v2
  
***************
*** 117,122 ****
--- 130,136 ----
  
      move-result v0
  
+     .line 43
      .local v0, en:I
      aget-object v7, v5, v2
  
***************
*** 124,139 ****
--- 138,158 ----
  
      move-result v1
  
+     .line 45
      .local v1, fl:I
      if-ge v6, p2, :cond_0
  
+     .line 46
      move v6, p2
  
+     .line 47
      :cond_0
      if-le v0, p3, :cond_1
  
+     .line 48
      move v0, p3
  
+     .line 50
      :cond_1
      aget-object v7, v5, v2
  
***************
*** 143,152 ****
--- 162,173 ----
  
      invoke-virtual {p0, v7, v8, v9, v1}, Landroid/text/SpannableStringInternal;->setSpan(Ljava/lang/Object;III)V
  
+     .line 40
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_1
  
+     .line 30
      .end local v0           #en:I
      .end local v1           #fl:I
      .end local v2           #i:I
***************
*** 167,172 ****
--- 188,194 ----
  
      goto :goto_0
  
+     .line 53
      .restart local v3       #initial:I
      :cond_3
      return-void
***************
*** 179,186 ****
--- 201,210 ----
      .parameter "end"
  
      .prologue
+     .line 340
      if-ge p3, p2, :cond_0
  
+     .line 341
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 219,234 ****
--- 243,261 ----
  
      throw v1
  
+     .line 346
      :cond_0
      invoke-virtual {p0}, Landroid/text/SpannableStringInternal;->length()I
  
      move-result v0
  
+     .line 348
      .local v0, len:I
      if-gt p2, v0, :cond_1
  
      if-le p3, v0, :cond_2
  
+     .line 349
      :cond_1
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 272,282 ****
--- 299,311 ----
  
      throw v1
  
+     .line 354
      :cond_2
      if-ltz p2, :cond_3
  
      if-gez p3, :cond_4
  
+     .line 355
      :cond_3
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
***************
*** 316,321 ****
--- 345,351 ----
  
      throw v1
  
+     .line 359
      :cond_4
      return-void
  .end method
***************
*** 326,331 ****
--- 356,362 ----
      .parameter "end"
  
      .prologue
+     .line 336
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 370,375 ****
--- 401,407 ----
      .parameter "end"
  
      .prologue
+     .line 308
      const-class v3, Landroid/text/SpanWatcher;
  
      invoke-virtual {p0, p2, p3, v3}, Landroid/text/SpannableStringInternal;->getSpans(IILjava/lang/Class;)[Ljava/lang/Object;
***************
*** 378,386 ****
--- 410,420 ----
  
      check-cast v2, [Landroid/text/SpanWatcher;
  
+     .line 309
      .local v2, recip:[Landroid/text/SpanWatcher;
      array-length v1, v2
  
+     .line 311
      .local v1, n:I
      const/4 v0, 0x0
  
***************
*** 388,393 ****
--- 422,428 ----
      :goto_0
      if-ge v0, v1, :cond_0
  
+     .line 312
      aget-object v4, v2, v0
  
      move-object v3, p0
***************
*** 396,405 ****
--- 431,442 ----
  
      invoke-interface {v4, v3, p1, p2, p3}, Landroid/text/SpanWatcher;->onSpanAdded(Landroid/text/Spannable;Ljava/lang/Object;II)V
  
+     .line 311
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 314
      :cond_0
      return-void
  .end method
***************
*** 413,418 ****
--- 450,456 ----
      .parameter "en"
  
      .prologue
+     .line 326
      invoke-static {p2, p4}, Ljava/lang/Math;->min(II)I
  
      move-result v0
***************
*** 429,437 ****
--- 467,477 ----
  
      check-cast v9, [Landroid/text/SpanWatcher;
  
+     .line 328
      .local v9, recip:[Landroid/text/SpanWatcher;
      array-length v8, v9
  
+     .line 330
      .local v8, n:I
      const/4 v7, 0x0
  
***************
*** 439,444 ****
--- 479,485 ----
      :goto_0
      if-ge v7, v8, :cond_0
  
+     .line 331
      aget-object v0, v9, v7
  
      move-object v1, p0
***************
*** 457,466 ****
--- 498,509 ----
  
      invoke-interface/range {v0 .. v6}, Landroid/text/SpanWatcher;->onSpanChanged(Landroid/text/Spannable;Ljava/lang/Object;IIII)V
  
+     .line 330
      add-int/lit8 v7, v7, 0x1
  
      goto :goto_0
  
+     .line 333
      :cond_0
      return-void
  .end method
***************
*** 472,477 ****
--- 515,521 ----
      .parameter "end"
  
      .prologue
+     .line 317
      const-class v3, Landroid/text/SpanWatcher;
  
      invoke-virtual {p0, p2, p3, v3}, Landroid/text/SpannableStringInternal;->getSpans(IILjava/lang/Class;)[Ljava/lang/Object;
***************
*** 480,488 ****
--- 524,534 ----
  
      check-cast v2, [Landroid/text/SpanWatcher;
  
+     .line 318
      .local v2, recip:[Landroid/text/SpanWatcher;
      array-length v1, v2
  
+     .line 320
      .local v1, n:I
      const/4 v0, 0x0
  
***************
*** 490,495 ****
--- 536,542 ----
      :goto_0
      if-ge v0, v1, :cond_0
  
+     .line 321
      aget-object v4, v2, v0
  
      move-object v3, p0
***************
*** 498,507 ****
--- 545,556 ----
  
      invoke-interface {v4, v3, p1, p2, p3}, Landroid/text/SpanWatcher;->onSpanRemoved(Landroid/text/Spannable;Ljava/lang/Object;II)V
  
+     .line 320
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 323
      :cond_0
      return-void
  .end method
***************
*** 513,518 ****
--- 562,568 ----
      .parameter "i"
  
      .prologue
+     .line 60
      iget-object v0, p0, Landroid/text/SpannableStringInternal;->mText:Ljava/lang/String;
  
      invoke-virtual {v0, p1}, Ljava/lang/String;->charAt(I)C
***************
*** 530,539 ****
--- 580,591 ----
      .parameter "off"
  
      .prologue
+     .line 70
      iget-object v0, p0, Landroid/text/SpannableStringInternal;->mText:Ljava/lang/String;
  
      invoke-virtual {v0, p1, p2, p3, p4}, Ljava/lang/String;->getChars(II[CI)V
  
+     .line 71
      return-void
  .end method
  
***************
*** 542,555 ****
--- 594,611 ----
      .parameter "what"
  
      .prologue
+     .line 178
      iget v0, p0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
+     .line 179
      .local v0, count:I
      iget-object v3, p0, Landroid/text/SpannableStringInternal;->mSpans:[Ljava/lang/Object;
  
+     .line 180
      .local v3, spans:[Ljava/lang/Object;
      iget-object v1, p0, Landroid/text/SpannableStringInternal;->mSpanData:[I
  
+     .line 182
      .local v1, data:[I
      add-int/lit8 v2, v0, -0x1
  
***************
*** 557,580 ****
--- 613,641 ----
      :goto_0
      if-ltz v2, :cond_1
  
+     .line 183
      aget-object v4, v3, v2
  
      if-ne v4, p1, :cond_0
  
+     .line 184
      mul-int/lit8 v4, v2, 0x3
  
      add-int/lit8 v4, v4, 0x1
  
      aget v4, v1, v4
  
+     .line 188
      :goto_1
      return v4
  
+     .line 182
      :cond_0
      add-int/lit8 v2, v2, -0x1
  
      goto :goto_0
  
+     .line 188
      :cond_1
      const/4 v4, -0x1
  
***************
*** 586,599 ****
--- 647,664 ----
      .parameter "what"
  
      .prologue
+     .line 192
      iget v0, p0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
+     .line 193
      .local v0, count:I
      iget-object v3, p0, Landroid/text/SpannableStringInternal;->mSpans:[Ljava/lang/Object;
  
+     .line 194
      .local v3, spans:[Ljava/lang/Object;
      iget-object v1, p0, Landroid/text/SpannableStringInternal;->mSpanData:[I
  
+     .line 196
      .local v1, data:[I
      add-int/lit8 v2, v0, -0x1
  
***************
*** 601,624 ****
--- 666,694 ----
      :goto_0
      if-ltz v2, :cond_1
  
+     .line 197
      aget-object v4, v3, v2
  
      if-ne v4, p1, :cond_0
  
+     .line 198
      mul-int/lit8 v4, v2, 0x3
  
      add-int/lit8 v4, v4, 0x2
  
      aget v4, v1, v4
  
+     .line 202
      :goto_1
      return v4
  
+     .line 196
      :cond_0
      add-int/lit8 v2, v2, -0x1
  
      goto :goto_0
  
+     .line 202
      :cond_1
      const/4 v4, 0x0
  
***************
*** 630,643 ****
--- 700,717 ----
      .parameter "what"
  
      .prologue
+     .line 164
      iget v0, p0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
+     .line 165
      .local v0, count:I
      iget-object v3, p0, Landroid/text/SpannableStringInternal;->mSpans:[Ljava/lang/Object;
  
+     .line 166
      .local v3, spans:[Ljava/lang/Object;
      iget-object v1, p0, Landroid/text/SpannableStringInternal;->mSpanData:[I
  
+     .line 168
      .local v1, data:[I
      add-int/lit8 v2, v0, -0x1
  
***************
*** 645,668 ****
--- 719,747 ----
      :goto_0
      if-ltz v2, :cond_1
  
+     .line 169
      aget-object v4, v3, v2
  
      if-ne v4, p1, :cond_0
  
+     .line 170
      mul-int/lit8 v4, v2, 0x3
  
      add-int/lit8 v4, v4, 0x0
  
      aget v4, v1, v4
  
+     .line 174
      :goto_1
      return v4
  
+     .line 168
      :cond_0
      add-int/lit8 v2, v2, -0x1
  
      goto :goto_0
  
+     .line 174
      :cond_1
      const/4 v4, -0x1
  
***************
*** 685,714 ****
--- 764,800 ----
      .end annotation
  
      .prologue
+     .line 206
      .local p3, kind:Ljava/lang/Class;,"Ljava/lang/Class<TT;>;"
      const/4 v2, 0x0
  
+     .line 208
      .local v2, count:I
      move-object/from16 v0, p0
  
      iget v12, v0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
+     .line 209
      .local v12, spanCount:I
      move-object/from16 v0, p0
  
      iget-object v15, v0, Landroid/text/SpannableStringInternal;->mSpans:[Ljava/lang/Object;
  
+     .line 210
      .local v15, spans:[Ljava/lang/Object;
      move-object/from16 v0, p0
  
      iget-object v4, v0, Landroid/text/SpannableStringInternal;->mSpanData:[I
  
+     .line 211
      .local v4, data:[I
      const/4 v10, 0x0
  
+     .line 212
      .local v10, ret:[Ljava/lang/Object;
      const/4 v11, 0x0
  
+     .line 214
      .local v11, ret1:Ljava/lang/Object;
      const/4 v5, 0x0
  
***************
*** 721,726 ****
--- 807,813 ----
      :goto_0
      if-ge v5, v12, :cond_a
  
+     .line 215
      if-eqz p3, :cond_0
  
      aget-object v16, v15, v5
***************
*** 737,742 ****
--- 824,830 ----
  
      move v2, v3
  
+     .line 214
      .end local v3           #count:I
      .restart local v2       #count:I
      :goto_1
***************
*** 748,753 ****
--- 836,842 ----
      .restart local v3       #count:I
      goto :goto_0
  
+     .line 219
      :cond_0
      mul-int/lit8 v16, v5, 0x3
  
***************
*** 755,760 ****
--- 844,850 ----
  
      aget v14, v4, v16
  
+     .line 220
      .local v14, spanStart:I
      mul-int/lit8 v16, v5, 0x3
  
***************
*** 762,767 ****
--- 852,858 ----
  
      aget v13, v4, v16
  
+     .line 222
      .local v13, spanEnd:I
      move/from16 v0, p2
  
***************
*** 769,778 ****
--- 860,871 ----
  
      move v2, v3
  
+     .line 223
      .end local v3           #count:I
      .restart local v2       #count:I
      goto :goto_1
  
+     .line 225
      .end local v2           #count:I
      .restart local v3       #count:I
      :cond_1
***************
*** 782,791 ****
--- 875,886 ----
  
      move v2, v3
  
+     .line 226
      .end local v3           #count:I
      .restart local v2       #count:I
      goto :goto_1
  
+     .line 229
      .end local v2           #count:I
      .restart local v3       #count:I
      :cond_2
***************
*** 797,812 ****
--- 892,910 ----
  
      if-eq v0, v1, :cond_4
  
+     .line 230
      move/from16 v0, p2
  
      if-ne v14, v0, :cond_3
  
      move v2, v3
  
+     .line 231
      .end local v3           #count:I
      .restart local v2       #count:I
      goto :goto_1
  
+     .line 233
      .end local v2           #count:I
      .restart local v3       #count:I
      :cond_3
***************
*** 816,832 ****
--- 914,934 ----
  
      move v2, v3
  
+     .line 234
      .end local v3           #count:I
      .restart local v2       #count:I
      goto :goto_1
  
+     .line 238
      .end local v2           #count:I
      .restart local v3       #count:I
      :cond_4
      if-nez v3, :cond_5
  
+     .line 239
      aget-object v11, v15, v5
  
+     .line 240
      .restart local v11       #ret1:Ljava/lang/Object;
      add-int/lit8 v2, v3, 0x1
  
***************
*** 834,839 ****
--- 936,942 ----
      .restart local v2       #count:I
      goto :goto_1
  
+     .line 242
      .end local v2           #count:I
      .end local v11           #ret1:Ljava/lang/Object;
      .restart local v3       #count:I
***************
*** 844,849 ****
--- 947,953 ----
  
      if-ne v3, v0, :cond_6
  
+     .line 243
      sub-int v16, v12, v5
  
      add-int/lit8 v16, v16, 0x1
***************
*** 862,871 ****
--- 966,977 ----
  
      check-cast v10, [Ljava/lang/Object;
  
+     .line 244
      const/16 v16, 0x0
  
      aput-object v11, v10, v16
  
+     .line 247
      :cond_6
      mul-int/lit8 v16, v5, 0x3
  
***************
*** 877,891 ****
--- 983,1000 ----
  
      and-int v9, v16, v17
  
+     .line 248
      .local v9, prio:I
      if-eqz v9, :cond_9
  
+     .line 251
      const/4 v6, 0x0
  
      .local v6, j:I
      :goto_2
      if-ge v6, v3, :cond_7
  
+     .line 252
      aget-object v16, v10, v6
  
      move-object/from16 v0, p0
***************
*** 900,908 ****
--- 1009,1019 ----
  
      and-int v8, v16, v17
  
+     .line 254
      .local v8, p:I
      if-le v9, v8, :cond_8
  
+     .line 259
      .end local v8           #p:I
      :cond_7
      add-int/lit8 v16, v6, 0x1
***************
*** 915,930 ****
--- 1026,1045 ----
  
      invoke-static {v10, v6, v10, v0, v1}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 260
      aget-object v16, v15, v5
  
      aput-object v16, v10, v6
  
+     .line 261
      add-int/lit8 v2, v3, 0x1
  
+     .line 262
      .end local v3           #count:I
      .restart local v2       #count:I
      goto/16 :goto_1
  
+     .line 251
      .end local v2           #count:I
      .restart local v3       #count:I
      .restart local v8       #p:I
***************
*** 933,938 ****
--- 1048,1054 ----
  
      goto :goto_2
  
+     .line 263
      .end local v6           #j:I
      .end local v8           #p:I
      :cond_9
***************
*** 946,951 ****
--- 1062,1068 ----
  
      goto/16 :goto_1
  
+     .line 268
      .end local v2           #count:I
      .end local v9           #prio:I
      .end local v13           #spanEnd:I
***************
*** 954,968 ****
--- 1071,1088 ----
      :cond_a
      if-nez v3, :cond_b
  
+     .line 269
      invoke-static/range {p3 .. p3}, Lcom/android/internal/util/ArrayUtils;->emptyArray(Ljava/lang/Class;)[Ljava/lang/Object;
  
      move-result-object v16
  
      check-cast v16, [Ljava/lang/Object;
  
+     .line 282
      :goto_3
      return-object v16
  
+     .line 271
      :cond_b
      const/16 v16, 0x1
  
***************
*** 970,975 ****
--- 1090,1096 ----
  
      if-ne v3, v0, :cond_c
  
+     .line 272
      const/16 v16, 0x1
  
      move-object/from16 v0, p3
***************
*** 986,1001 ****
--- 1107,1125 ----
  
      check-cast v10, [Ljava/lang/Object;
  
+     .line 273
      const/16 v16, 0x0
  
      aput-object v11, v10, v16
  
      move-object/from16 v16, v10
  
+     .line 274
      check-cast v16, [Ljava/lang/Object;
  
      goto :goto_3
  
+     .line 276
      :cond_c
      array-length v0, v10
  
***************
*** 1007,1016 ****
--- 1131,1142 ----
  
      move-object/from16 v16, v10
  
+     .line 277
      check-cast v16, [Ljava/lang/Object;
  
      goto :goto_3
  
+     .line 280
      :cond_d
      move-object/from16 v0, p3
  
***************
*** 1024,1029 ****
--- 1150,1156 ----
  
      check-cast v7, [Ljava/lang/Object;
  
+     .line 281
      .local v7, nret:[Ljava/lang/Object;
      const/16 v16, 0x0
  
***************
*** 1035,1040 ****
--- 1162,1168 ----
  
      invoke-static {v10, v0, v7, v1, v3}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 282
      check-cast v7, [Ljava/lang/Object;
  
      .end local v7           #nret:[Ljava/lang/Object;
***************
*** 1047,1052 ****
--- 1175,1181 ----
      .locals 1
  
      .prologue
+     .line 56
      iget-object v0, p0, Landroid/text/SpannableStringInternal;->mText:Ljava/lang/String;
  
      invoke-virtual {v0}, Ljava/lang/String;->length()I
***************
*** 1063,1081 ****
--- 1192,1216 ----
      .parameter "kind"
  
      .prologue
+     .line 286
      iget v0, p0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
+     .line 287
      .local v0, count:I
      iget-object v4, p0, Landroid/text/SpannableStringInternal;->mSpans:[Ljava/lang/Object;
  
+     .line 288
      .local v4, spans:[Ljava/lang/Object;
      iget-object v1, p0, Landroid/text/SpannableStringInternal;->mSpanData:[I
  
+     .line 290
      .local v1, data:[I
      if-nez p3, :cond_0
  
+     .line 291
      const-class p3, Ljava/lang/Object;
  
+     .line 294
      :cond_0
      const/4 v3, 0x0
  
***************
*** 1083,1094 ****
--- 1218,1231 ----
      :goto_0
      if-ge v3, v0, :cond_3
  
+     .line 295
      mul-int/lit8 v6, v3, 0x3
  
      add-int/lit8 v6, v6, 0x0
  
      aget v5, v1, v6
  
+     .line 296
      .local v5, st:I
      mul-int/lit8 v6, v3, 0x3
  
***************
*** 1096,1101 ****
--- 1233,1239 ----
  
      aget v2, v1, v6
  
+     .line 298
      .local v2, en:I
      if-le v5, p1, :cond_1
  
***************
*** 1109,1116 ****
--- 1247,1256 ----
  
      if-eqz v6, :cond_1
  
+     .line 299
      move p2, v5
  
+     .line 300
      :cond_1
      if-le v2, p1, :cond_2
  
***************
*** 1124,1136 ****
--- 1264,1279 ----
  
      if-eqz v6, :cond_2
  
+     .line 301
      move p2, v2
  
+     .line 294
      :cond_2
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 304
      .end local v2           #en:I
      .end local v5           #st:I
      :cond_3
***************
*** 1142,1155 ****
--- 1285,1302 ----
      .parameter "what"
  
      .prologue
+     .line 140
      iget v1, p0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
+     .line 141
      .local v1, count:I
      iget-object v6, p0, Landroid/text/SpannableStringInternal;->mSpans:[Ljava/lang/Object;
  
+     .line 142
      .local v6, spans:[Ljava/lang/Object;
      iget-object v2, p0, Landroid/text/SpannableStringInternal;->mSpanData:[I
  
+     .line 144
      .local v2, data:[I
      add-int/lit8 v3, v1, -0x1
  
***************
*** 1157,1172 ****
--- 1304,1322 ----
      :goto_0
      if-ltz v3, :cond_0
  
+     .line 145
      aget-object v7, v6, v3
  
      if-ne v7, p1, :cond_1
  
+     .line 146
      mul-int/lit8 v7, v3, 0x3
  
      add-int/lit8 v7, v7, 0x0
  
      aget v5, v2, v7
  
+     .line 147
      .local v5, ostart:I
      mul-int/lit8 v7, v3, 0x3
  
***************
*** 1174,1189 ****
--- 1324,1342 ----
  
      aget v4, v2, v7
  
+     .line 149
      .local v4, oend:I
      add-int/lit8 v7, v3, 0x1
  
      sub-int v0, v1, v7
  
+     .line 151
      .local v0, c:I
      add-int/lit8 v7, v3, 0x1
  
      invoke-static {v6, v7, v6, v3, v0}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 152
      add-int/lit8 v7, v3, 0x1
  
      mul-int/lit8 v7, v7, 0x3
***************
*** 1194,1213 ****
--- 1347,1370 ----
  
      invoke-static {v2, v7, v2, v8, v9}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 155
      iget v7, p0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
      add-int/lit8 v7, v7, -0x1
  
      iput v7, p0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
+     .line 157
      invoke-direct {p0, p1, v5, v4}, Landroid/text/SpannableStringInternal;->sendSpanRemoved(Ljava/lang/Object;II)V
  
+     .line 161
      .end local v0           #c:I
      .end local v4           #oend:I
      .end local v5           #ostart:I
      :cond_0
      return-void
  
+     .line 144
      :cond_1
      add-int/lit8 v3, v3, -0x1
  
***************
*** 1222,1234 ****
      .parameter "flags"
  
      .prologue
      move/from16 v7, p2
  
      .local v7, nstart:I
      move/from16 v8, p3
  
      .local v8, nend:I
!     const-string v3, "setSpan"
  
      move-object/from16 v0, p0
  
--- 1379,1394 ----
      .parameter "flags"
  
      .prologue
+     .line 74
      move/from16 v7, p2
  
+     .line 75
      .local v7, nstart:I
      move/from16 v8, p3
  
+     .line 77
      .local v8, nend:I
!     const-string/jumbo v3, "setSpan"
  
      move-object/from16 v0, p0
  
***************
*** 1238,1249 ****
--- 1398,1411 ----
  
      invoke-direct {v0, v3, v1, v2}, Landroid/text/SpannableStringInternal;->checkRange(Ljava/lang/String;II)V
  
+     .line 79
      and-int/lit8 v3, p4, 0x33
  
      const/16 v4, 0x33
  
      if-ne v3, v4, :cond_1
  
+     .line 80
      if-eqz p2, :cond_0
  
      invoke-virtual/range {p0 .. p0}, Landroid/text/SpannableStringInternal;->length()I
***************
*** 1254,1259 ****
--- 1416,1422 ----
  
      if-eq v0, v3, :cond_0
  
+     .line 81
      add-int/lit8 v3, p2, -0x1
  
      move-object/from16 v0, p0
***************
*** 1262,1272 ****
--- 1425,1437 ----
  
      move-result v9
  
+     .line 83
      .local v9, c:C
      const/16 v3, 0xa
  
      if-eq v9, v3, :cond_0
  
+     .line 84
      new-instance v3, Ljava/lang/RuntimeException;
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 1315,1320 ****
--- 1480,1486 ----
  
      throw v3
  
+     .line 89
      .end local v9           #c:C
      :cond_0
      if-eqz p3, :cond_1
***************
*** 1327,1332 ****
--- 1493,1499 ----
  
      if-eq v0, v3, :cond_1
  
+     .line 90
      add-int/lit8 v3, p3, -0x1
  
      move-object/from16 v0, p0
***************
*** 1335,1345 ****
--- 1502,1514 ----
  
      move-result v9
  
+     .line 92
      .restart local v9       #c:C
      const/16 v3, 0xa
  
      if-eq v9, v3, :cond_1
  
+     .line 93
      new-instance v3, Ljava/lang/RuntimeException;
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 1388,1399 ****
--- 1557,1570 ----
  
      throw v3
  
+     .line 99
      .end local v9           #c:C
      :cond_1
      move-object/from16 v0, p0
  
      iget v10, v0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
+     .line 100
      .local v10, count:I
      move-object/from16 v0, p0
  
***************
*** 1401,1411 ****
--- 1572,1584 ----
  
      move-object/from16 v16, v0
  
+     .line 101
      .local v16, spans:[Ljava/lang/Object;
      move-object/from16 v0, p0
  
      iget-object v11, v0, Landroid/text/SpannableStringInternal;->mSpanData:[I
  
+     .line 103
      .local v11, data:[I
      const/4 v12, 0x0
  
***************
*** 1413,1430 ****
--- 1586,1606 ----
      :goto_0
      if-ge v12, v10, :cond_4
  
+     .line 104
      aget-object v3, v16, v12
  
      move-object/from16 v0, p1
  
      if-ne v3, v0, :cond_3
  
+     .line 105
      mul-int/lit8 v3, v12, 0x3
  
      add-int/lit8 v3, v3, 0x0
  
      aget v5, v11, v3
  
+     .line 106
      .local v5, ostart:I
      mul-int/lit8 v3, v12, 0x3
  
***************
*** 1432,1437 ****
--- 1608,1614 ----
  
      aget v6, v11, v3
  
+     .line 108
      .local v6, oend:I
      mul-int/lit8 v3, v12, 0x3
  
***************
*** 1439,1450 ****
--- 1616,1629 ----
  
      aput p2, v11, v3
  
+     .line 109
      mul-int/lit8 v3, v12, 0x3
  
      add-int/lit8 v3, v3, 0x1
  
      aput p3, v11, v3
  
+     .line 110
      mul-int/lit8 v3, v12, 0x3
  
      add-int/lit8 v3, v3, 0x2
***************
*** 1455,1473 ****
--- 1634,1656 ----
  
      move-object/from16 v4, p1
  
+     .line 112
      invoke-direct/range {v3 .. v8}, Landroid/text/SpannableStringInternal;->sendSpanChanged(Ljava/lang/Object;IIII)V
  
+     .line 137
      .end local v5           #ostart:I
      .end local v6           #oend:I
      :cond_2
      :goto_1
      return-void
  
+     .line 103
      :cond_3
      add-int/lit8 v12, v12, 0x1
  
      goto :goto_0
  
+     .line 117
      :cond_4
      move-object/from16 v0, p0
  
***************
*** 1483,1488 ****
--- 1666,1672 ----
  
      if-lt v3, v4, :cond_5
  
+     .line 118
      move-object/from16 v0, p0
  
      iget v3, v0, Landroid/text/SpannableStringInternal;->mSpanCount:I
***************
*** 1493,1506 ****
--- 1677,1693 ----
  
      move-result v14
  
+     .line 119
      .local v14, newsize:I
      new-array v15, v14, [Ljava/lang/Object;
  
+     .line 120
      .local v15, newtags:[Ljava/lang/Object;
      mul-int/lit8 v3, v14, 0x3
  
      new-array v13, v3, [I
  
+     .line 122
      .local v13, newdata:[I
      move-object/from16 v0, p0
  
***************
*** 1522,1527 ****
--- 1709,1715 ----
  
      invoke-static {v3, v4, v15, v0, v1}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 123
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/text/SpannableStringInternal;->mSpanData:[I
***************
*** 1544,1557 ****
--- 1732,1748 ----
  
      invoke-static {v3, v4, v13, v0, v1}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 125
      move-object/from16 v0, p0
  
      iput-object v15, v0, Landroid/text/SpannableStringInternal;->mSpans:[Ljava/lang/Object;
  
+     .line 126
      move-object/from16 v0, p0
  
      iput-object v13, v0, Landroid/text/SpannableStringInternal;->mSpanData:[I
  
+     .line 129
      .end local v13           #newdata:[I
      .end local v14           #newsize:I
      .end local v15           #newtags:[Ljava/lang/Object;
***************
*** 1566,1571 ****
--- 1757,1763 ----
  
      aput-object p1, v3, v4
  
+     .line 130
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/text/SpannableStringInternal;->mSpanData:[I
***************
*** 1580,1585 ****
--- 1772,1778 ----
  
      aput p2, v3, v4
  
+     .line 131
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/text/SpannableStringInternal;->mSpanData:[I
***************
*** 1594,1599 ****
--- 1787,1793 ----
  
      aput p3, v3, v4
  
+     .line 132
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/text/SpannableStringInternal;->mSpanData:[I
***************
*** 1608,1613 ****
--- 1802,1808 ----
  
      aput p4, v3, v4
  
+     .line 133
      move-object/from16 v0, p0
  
      iget v3, v0, Landroid/text/SpannableStringInternal;->mSpanCount:I
***************
*** 1618,1629 ****
--- 1813,1826 ----
  
      iput v3, v0, Landroid/text/SpannableStringInternal;->mSpanCount:I
  
+     .line 135
      move-object/from16 v0, p0
  
      instance-of v3, v0, Landroid/text/Spannable;
  
      if-eqz v3, :cond_2
  
+     .line 136
      move-object/from16 v0, p0
  
      move-object/from16 v1, p1
***************
*** 1637,1642 ****
--- 1834,1840 ----
      .locals 1
  
      .prologue
+     .line 64
      iget-object v0, p0, Landroid/text/SpannableStringInternal;->mText:Ljava/lang/String;
  
      return-object v0
