*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/TextUtils$SimpleStringSplitter.smali	2012-03-23 19:35:19.970951056 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/TextUtils$SimpleStringSplitter.smali.original	2012-03-23 19:34:44.491304085 -0500
***************
*** 45,54 ****
--- 45,57 ----
      .parameter "delimiter"
  
      .prologue
+     .line 386
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 387
      iput-char p1, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mDelimiter:C
  
+     .line 388
      return-void
  .end method
  
***************
*** 58,63 ****
--- 61,67 ----
      .locals 2
  
      .prologue
+     .line 405
      iget v0, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mPosition:I
  
      iget v1, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mLength:I
***************
*** 88,93 ****
--- 92,98 ----
      .end annotation
  
      .prologue
+     .line 401
      return-object p0
  .end method
  
***************
*** 95,100 ****
--- 100,106 ----
      .locals 1
  
      .prologue
+     .line 376
      invoke-virtual {p0}, Landroid/text/TextUtils$SimpleStringSplitter;->next()Ljava/lang/String;
  
      move-result-object v0
***************
*** 106,111 ****
--- 112,118 ----
      .locals 5
  
      .prologue
+     .line 409
      iget-object v2, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mString:Ljava/lang/String;
  
      iget-char v3, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mDelimiter:C
***************
*** 116,128 ****
--- 123,138 ----
  
      move-result v0
  
+     .line 410
      .local v0, end:I
      const/4 v2, -0x1
  
      if-ne v0, v2, :cond_0
  
+     .line 411
      iget v0, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mLength:I
  
+     .line 413
      :cond_0
      iget-object v2, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mString:Ljava/lang/String;
  
***************
*** 132,142 ****
--- 142,154 ----
  
      move-result-object v1
  
+     .line 414
      .local v1, nextString:Ljava/lang/String;
      add-int/lit8 v2, v0, 0x1
  
      iput v2, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mPosition:I
  
+     .line 415
      return-object v1
  .end method
  
***************
*** 144,149 ****
--- 156,162 ----
      .locals 1
  
      .prologue
+     .line 419
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 156,167 ****
--- 169,183 ----
      .parameter "string"
  
      .prologue
+     .line 395
      iput-object p1, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mString:Ljava/lang/String;
  
+     .line 396
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mPosition:I
  
+     .line 397
      iget-object v0, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mString:Ljava/lang/String;
  
      invoke-virtual {v0}, Ljava/lang/String;->length()I
***************
*** 170,174 ****
--- 186,191 ----
  
      iput v0, p0, Landroid/text/TextUtils$SimpleStringSplitter;->mLength:I
  
+     .line 398
      return-void
  .end method
