*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/TextDirectionHeuristics$TriState.smali	2012-03-23 19:35:19.002933403 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/TextDirectionHeuristics$TriState.smali.original	2012-03-23 19:34:44.479303866 -0500
***************
*** 44,49 ****
--- 44,50 ----
  
      const/4 v2, 0x0
  
+     .line 70
      new-instance v0, Landroid/text/TextDirectionHeuristics$TriState;
  
      const-string v1, "TRUE"
***************
*** 68,73 ****
--- 69,75 ----
  
      sput-object v0, Landroid/text/TextDirectionHeuristics$TriState;->UNKNOWN:Landroid/text/TextDirectionHeuristics$TriState;
  
+     .line 69
      const/4 v0, 0x3
  
      new-array v0, v0, [Landroid/text/TextDirectionHeuristics$TriState;
***************
*** 100,105 ****
--- 102,108 ----
      .end annotation
  
      .prologue
+     .line 69
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 110,115 ****
--- 113,119 ----
      .parameter "name"
  
      .prologue
+     .line 69
      const-class v0, Landroid/text/TextDirectionHeuristics$TriState;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 125,130 ****
--- 129,135 ----
      .locals 1
  
      .prologue
+     .line 69
      sget-object v0, Landroid/text/TextDirectionHeuristics$TriState;->$VALUES:[Landroid/text/TextDirectionHeuristics$TriState;
  
      invoke-virtual {v0}, [Landroid/text/TextDirectionHeuristics$TriState;->clone()Ljava/lang/Object;
