*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/TextDirectionHeuristics$TextDirectionHeuristicImpl.smali	2012-03-23 19:35:20.292956927 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/TextDirectionHeuristics$TextDirectionHeuristicImpl.smali.original	2012-03-23 19:34:44.496304176 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "algorithm"
  
      .prologue
+     .line 82
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 83
      iput-object p1, p0, Landroid/text/TextDirectionHeuristics$TextDirectionHeuristicImpl;->mAlgorithm:Landroid/text/TextDirectionHeuristics$TextDirectionAlgorithm;
  
+     .line 84
      return-void
  .end method
  
***************
*** 41,46 ****
--- 44,50 ----
      .parameter "count"
  
      .prologue
+     .line 103
      sget-object v0, Landroid/text/TextDirectionHeuristics$1;->$SwitchMap$android$text$TextDirectionHeuristics$TriState:[I
  
      iget-object v1, p0, Landroid/text/TextDirectionHeuristics$TextDirectionHeuristicImpl;->mAlgorithm:Landroid/text/TextDirectionHeuristics$TextDirectionAlgorithm;
***************
*** 57,62 ****
--- 61,67 ----
  
      packed-switch v0, :pswitch_data_0
  
+     .line 109
      invoke-virtual {p0}, Landroid/text/TextDirectionHeuristics$TextDirectionHeuristicImpl;->defaultIsRtl()Z
  
      move-result v0
***************
*** 64,79 ****
--- 69,87 ----
      :goto_0
      return v0
  
+     .line 105
      :pswitch_0
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 107
      :pswitch_1
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 103
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 93,98 ****
--- 101,107 ----
      .parameter "count"
  
      .prologue
+     .line 93
      if-eqz p1, :cond_0
  
      if-ltz p2, :cond_0
***************
*** 105,110 ****
--- 114,120 ----
  
      if-ge v0, p2, :cond_1
  
+     .line 94
      :cond_0
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
***************
*** 112,126 ****
--- 122,139 ----
  
      throw v0
  
+     .line 96
      :cond_1
      iget-object v0, p0, Landroid/text/TextDirectionHeuristics$TextDirectionHeuristicImpl;->mAlgorithm:Landroid/text/TextDirectionHeuristics$TextDirectionAlgorithm;
  
      if-nez v0, :cond_2
  
+     .line 97
      invoke-virtual {p0}, Landroid/text/TextDirectionHeuristics$TextDirectionHeuristicImpl;->defaultIsRtl()Z
  
      move-result v0
  
+     .line 99
      :goto_0
      return v0
  
