*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/TextUtils$1.smali	2012-03-23 19:35:19.905949870 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/TextUtils$1.smali.original	2012-03-23 19:34:44.491304085 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 643
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,93 ****
--- 45,106 ----
      .parameter "p"
  
      .prologue
+     .line 649
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 651
      .local v0, kind:I
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 652
      .local v2, string:Ljava/lang/String;
      if-nez v2, :cond_1
  
+     .line 653
      const/4 v2, 0x0
  
+     .line 762
      .end local v2           #string:Ljava/lang/String;
      :cond_0
      :goto_0
      return-object v2
  
+     .line 656
      .restart local v2       #string:Ljava/lang/String;
      :cond_1
      const/4 v3, 0x1
  
      if-eq v0, v3, :cond_0
  
+     .line 660
      new-instance v1, Landroid/text/SpannableString;
  
      invoke-direct {v1, v2}, Landroid/text/SpannableString;-><init>(Ljava/lang/CharSequence;)V
  
+     .line 663
      .local v1, sp:Landroid/text/SpannableString;
      :goto_1
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 665
      if-nez v0, :cond_2
  
      move-object v2, v1
  
+     .line 762
      goto :goto_0
  
+     .line 668
      :cond_2
      packed-switch v0, :pswitch_data_0
  
+     .line 758
      new-instance v3, Ljava/lang/RuntimeException;
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 112,117 ****
--- 125,131 ----
  
      throw v3
  
+     .line 670
      :pswitch_0
      new-instance v3, Landroid/text/style/AlignmentSpan$Standard;
  
***************
*** 122,127 ****
--- 136,142 ----
  
      goto :goto_1
  
+     .line 674
      :pswitch_1
      new-instance v3, Landroid/text/style/ForegroundColorSpan;
  
***************
*** 132,137 ****
--- 147,153 ----
  
      goto :goto_1
  
+     .line 678
      :pswitch_2
      new-instance v3, Landroid/text/style/RelativeSizeSpan;
  
***************
*** 142,147 ****
--- 158,164 ----
  
      goto :goto_1
  
+     .line 682
      :pswitch_3
      new-instance v3, Landroid/text/style/ScaleXSpan;
  
***************
*** 152,157 ****
--- 169,175 ----
  
      goto :goto_1
  
+     .line 686
      :pswitch_4
      new-instance v3, Landroid/text/style/StrikethroughSpan;
  
***************
*** 162,167 ****
--- 180,186 ----
  
      goto :goto_1
  
+     .line 690
      :pswitch_5
      new-instance v3, Landroid/text/style/UnderlineSpan;
  
***************
*** 172,177 ****
--- 191,197 ----
  
      goto :goto_1
  
+     .line 694
      :pswitch_6
      new-instance v3, Landroid/text/style/StyleSpan;
  
***************
*** 182,187 ****
--- 202,208 ----
  
      goto :goto_1
  
+     .line 698
      :pswitch_7
      new-instance v3, Landroid/text/style/BulletSpan;
  
***************
*** 192,197 ****
--- 213,219 ----
  
      goto :goto_1
  
+     .line 702
      :pswitch_8
      new-instance v3, Landroid/text/style/QuoteSpan;
  
***************
*** 202,207 ****
--- 224,230 ----
  
      goto :goto_1
  
+     .line 706
      :pswitch_9
      new-instance v3, Landroid/text/style/LeadingMarginSpan$Standard;
  
***************
*** 212,217 ****
--- 235,241 ----
  
      goto :goto_1
  
+     .line 710
      :pswitch_a
      new-instance v3, Landroid/text/style/URLSpan;
  
***************
*** 222,227 ****
--- 246,252 ----
  
      goto/16 :goto_1
  
+     .line 714
      :pswitch_b
      new-instance v3, Landroid/text/style/BackgroundColorSpan;
  
***************
*** 232,237 ****
--- 257,263 ----
  
      goto/16 :goto_1
  
+     .line 718
      :pswitch_c
      new-instance v3, Landroid/text/style/TypefaceSpan;
  
***************
*** 242,247 ****
--- 268,274 ----
  
      goto/16 :goto_1
  
+     .line 722
      :pswitch_d
      new-instance v3, Landroid/text/style/SuperscriptSpan;
  
***************
*** 252,257 ****
--- 279,285 ----
  
      goto/16 :goto_1
  
+     .line 726
      :pswitch_e
      new-instance v3, Landroid/text/style/SubscriptSpan;
  
***************
*** 262,267 ****
--- 290,296 ----
  
      goto/16 :goto_1
  
+     .line 730
      :pswitch_f
      new-instance v3, Landroid/text/style/AbsoluteSizeSpan;
  
***************
*** 272,277 ****
--- 301,307 ----
  
      goto/16 :goto_1
  
+     .line 734
      :pswitch_10
      new-instance v3, Landroid/text/style/TextAppearanceSpan;
  
***************
*** 282,287 ****
--- 312,318 ----
  
      goto/16 :goto_1
  
+     .line 738
      :pswitch_11
      new-instance v3, Landroid/text/Annotation;
  
***************
*** 292,297 ****
--- 323,329 ----
  
      goto/16 :goto_1
  
+     .line 742
      :pswitch_12
      new-instance v3, Landroid/text/style/SuggestionSpan;
  
***************
*** 302,307 ****
--- 334,340 ----
  
      goto/16 :goto_1
  
+     .line 746
      :pswitch_13
      new-instance v3, Landroid/text/style/SpellCheckSpan;
  
***************
*** 312,317 ****
--- 345,351 ----
  
      goto/16 :goto_1
  
+     .line 750
      :pswitch_14
      new-instance v3, Landroid/text/style/SuggestionRangeSpan;
  
***************
*** 322,327 ****
--- 356,362 ----
  
      goto/16 :goto_1
  
+     .line 754
      :pswitch_15
      new-instance v3, Landroid/text/style/EasyEditSpan;
  
***************
*** 332,337 ****
--- 367,373 ----
  
      goto/16 :goto_1
  
+     .line 668
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 364,369 ****
--- 400,406 ----
      .parameter "x0"
  
      .prologue
+     .line 643
      invoke-virtual {p0, p1}, Landroid/text/TextUtils$1;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/CharSequence;
  
      move-result-object v0
***************
*** 376,381 ****
--- 413,419 ----
      .parameter "size"
  
      .prologue
+     .line 767
      new-array v0, p1, [Ljava/lang/CharSequence;
  
      return-object v0
***************
*** 386,391 ****
--- 424,430 ----
      .parameter "x0"
  
      .prologue
+     .line 643
      invoke-virtual {p0, p1}, Landroid/text/TextUtils$1;->newArray(I)[Ljava/lang/CharSequence;
  
      move-result-object v0
