*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/server/NetworkManagementSocketTagger.smali	2012-03-23 19:37:55.323783911 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/server/NetworkManagementSocketTagger.smali.original	2012-03-23 19:34:46.412339117 -0500
***************
*** 35,40 ****
--- 35,41 ----
      .locals 1
  
      .prologue
+     .line 48
      new-instance v0, Lcom/android/server/NetworkManagementSocketTagger$1;
  
      invoke-direct {v0}, Lcom/android/server/NetworkManagementSocketTagger$1;-><init>()V
***************
*** 48,55 ****
--- 49,58 ----
      .locals 0
  
      .prologue
+     .line 37
      invoke-direct {p0}, Ldalvik/system/SocketTagger;-><init>()V
  
+     .line 113
      return-void
  .end method
  
***************
*** 57,62 ****
--- 60,66 ----
      .locals 1
  
      .prologue
+     .line 64
      sget-object v0, Lcom/android/server/NetworkManagementSocketTagger;->threadSocketTags:Ljava/lang/ThreadLocal;
  
      invoke-virtual {v0}, Ljava/lang/ThreadLocal;->get()Ljava/lang/Object;
***************
*** 74,85 ****
--- 78,91 ----
      .locals 1
  
      .prologue
+     .line 56
      new-instance v0, Lcom/android/server/NetworkManagementSocketTagger;
  
      invoke-direct {v0}, Lcom/android/server/NetworkManagementSocketTagger;-><init>()V
  
      invoke-static {v0}, Ldalvik/system/SocketTagger;->set(Ldalvik/system/SocketTagger;)V
  
+     .line 57
      return-void
  .end method
  
***************
*** 88,93 ****
--- 94,100 ----
      .parameter "string"
  
      .prologue
+     .line 138
      invoke-static {p0}, Ljava/lang/Long;->decode(Ljava/lang/String;)Ljava/lang/Long;
  
      move-result-object v0
***************
*** 122,143 ****
      .parameter "uid"
  
      .prologue
      const/4 v1, 0x0
  
      invoke-static {v1, p0}, Lcom/android/server/NetworkManagementSocketTagger;->native_deleteTagData(II)I
  
      move-result v0
  
      .local v0, errno:I
      if-gez v0, :cond_0
  
      const-string v1, "NetworkManagementSocketTagger"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "problem clearing counters for uid "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 129,153 ----
      .parameter "uid"
  
      .prologue
+     .line 126
      const/4 v1, 0x0
  
      invoke-static {v1, p0}, Lcom/android/server/NetworkManagementSocketTagger;->native_deleteTagData(II)I
  
      move-result v0
  
+     .line 127
      .local v0, errno:I
      if-gez v0, :cond_0
  
+     .line 128
      const-string v1, "NetworkManagementSocketTagger"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "problem clearing counters for uid "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 163,168 ****
--- 173,179 ----
  
      invoke-static {v1, v2}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 130
      :cond_0
      return-void
  .end method
***************
*** 173,192 ****
      .parameter "counterSet"
  
      .prologue
      invoke-static {p1, p0}, Lcom/android/server/NetworkManagementSocketTagger;->native_setCounterSet(II)I
  
      move-result v0
  
      .local v0, errno:I
      if-gez v0, :cond_0
  
      const-string v1, "NetworkManagementSocketTagger"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "setKernelCountSet("
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 184,206 ----
      .parameter "counterSet"
  
      .prologue
+     .line 119
      invoke-static {p1, p0}, Lcom/android/server/NetworkManagementSocketTagger;->native_setCounterSet(II)I
  
      move-result v0
  
+     .line 120
      .local v0, errno:I
      if-gez v0, :cond_0
  
+     .line 121
      const-string v1, "NetworkManagementSocketTagger"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "setKernelCountSet("
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 222,227 ****
--- 236,242 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 123
      :cond_0
      return-void
  .end method
***************
*** 231,236 ****
--- 246,252 ----
      .parameter "tag"
  
      .prologue
+     .line 60
      sget-object v0, Lcom/android/server/NetworkManagementSocketTagger;->threadSocketTags:Ljava/lang/ThreadLocal;
  
      invoke-virtual {v0}, Ljava/lang/ThreadLocal;->get()Ljava/lang/Object;
***************
*** 241,246 ****
--- 257,263 ----
  
      iput p0, v0, Lcom/android/server/NetworkManagementSocketTagger$SocketTags;->statsTag:I
  
+     .line 61
      return-void
  .end method
  
***************
*** 249,254 ****
--- 266,272 ----
      .parameter "uid"
  
      .prologue
+     .line 68
      sget-object v0, Lcom/android/server/NetworkManagementSocketTagger;->threadSocketTags:Ljava/lang/ThreadLocal;
  
      invoke-virtual {v0}, Ljava/lang/ThreadLocal;->get()Ljava/lang/Object;
***************
*** 259,264 ****
--- 277,283 ----
  
      iput p0, v0, Lcom/android/server/NetworkManagementSocketTagger$SocketTags;->statsUid:I
  
+     .line 69
      return-void
  .end method
  
***************
*** 271,299 ****
      .prologue
      const/4 v1, -0x1
  
      if-ne p2, v1, :cond_1
  
      if-ne p3, v1, :cond_1
  
      :cond_0
      :goto_0
      return-void
  
      :cond_1
      invoke-static {p1, p2, p3}, Lcom/android/server/NetworkManagementSocketTagger;->native_tagSocketFd(Ljava/io/FileDescriptor;II)I
  
      move-result v0
  
      .local v0, errno:I
      if-gez v0, :cond_0
  
      const-string v1, "NetworkManagementSocketTagger"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "tagSocketFd("
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 290,323 ----
      .prologue
      const/4 v1, -0x1
  
+     .line 84
      if-ne p2, v1, :cond_1
  
      if-ne p3, v1, :cond_1
  
+     .line 92
      :cond_0
      :goto_0
      return-void
  
+     .line 86
      :cond_1
      invoke-static {p1, p2, p3}, Lcom/android/server/NetworkManagementSocketTagger;->native_tagSocketFd(Ljava/io/FileDescriptor;II)I
  
      move-result v0
  
+     .line 87
      .local v0, errno:I
      if-gez v0, :cond_0
  
+     .line 88
      const-string v1, "NetworkManagementSocketTagger"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "tagSocketFd("
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 353,358 ****
--- 377,383 ----
      .prologue
      const/4 v3, -0x1
  
+     .line 103
      sget-object v2, Lcom/android/server/NetworkManagementSocketTagger;->threadSocketTags:Ljava/lang/ThreadLocal;
  
      invoke-virtual {v2}, Ljava/lang/ThreadLocal;->get()Ljava/lang/Object;
***************
*** 361,366 ****
--- 386,392 ----
  
      check-cast v1, Lcom/android/server/NetworkManagementSocketTagger$SocketTags;
  
+     .line 105
      .local v1, options:Lcom/android/server/NetworkManagementSocketTagger$SocketTags;
      iget v2, v1, Lcom/android/server/NetworkManagementSocketTagger$SocketTags;->statsTag:I
  
***************
*** 370,394 ****
  
      if-ne v2, v3, :cond_1
  
      :cond_0
      :goto_0
      return-void
  
      :cond_1
      invoke-static {p1}, Lcom/android/server/NetworkManagementSocketTagger;->native_untagSocketFd(Ljava/io/FileDescriptor;)I
  
      move-result v0
  
      .local v0, errno:I
      if-gez v0, :cond_0
  
      const-string v2, "NetworkManagementSocketTagger"
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "untagSocket("
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 396,424 ----
  
      if-ne v2, v3, :cond_1
  
+     .line 111
      :cond_0
      :goto_0
      return-void
  
+     .line 107
      :cond_1
      invoke-static {p1}, Lcom/android/server/NetworkManagementSocketTagger;->native_untagSocketFd(Ljava/io/FileDescriptor;)I
  
      move-result v0
  
+     .line 108
      .local v0, errno:I
      if-gez v0, :cond_0
  
+     .line 109
      const-string v2, "NetworkManagementSocketTagger"
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "untagSocket("
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 433,438 ****
--- 463,469 ----
      .end annotation
  
      .prologue
+     .line 73
      sget-object v1, Lcom/android/server/NetworkManagementSocketTagger;->threadSocketTags:Ljava/lang/ThreadLocal;
  
      invoke-virtual {v1}, Ljava/lang/ThreadLocal;->get()Ljava/lang/Object;
***************
*** 441,446 ****
--- 472,478 ----
  
      check-cast v0, Lcom/android/server/NetworkManagementSocketTagger$SocketTags;
  
+     .line 79
      .local v0, options:Lcom/android/server/NetworkManagementSocketTagger$SocketTags;
      iget v1, v0, Lcom/android/server/NetworkManagementSocketTagger$SocketTags;->statsTag:I
  
***************
*** 448,453 ****
--- 480,486 ----
  
      invoke-direct {p0, p1, v1, v2}, Lcom/android/server/NetworkManagementSocketTagger;->tagSocketFd(Ljava/io/FileDescriptor;II)V
  
+     .line 80
      return-void
  .end method
  
***************
*** 461,467 ****
--- 494,502 ----
      .end annotation
  
      .prologue
+     .line 99
      invoke-direct {p0, p1}, Lcom/android/server/NetworkManagementSocketTagger;->unTagSocketFd(Ljava/io/FileDescriptor;)V
  
+     .line 100
      return-void
  .end method
