*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState.smali	2012-03-23 19:35:43.815385858 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState.smali.original	2012-03-23 19:34:44.811309920 -0500
***************
*** 44,55 ****
--- 44,59 ----
      .parameter "res"
  
      .prologue
+     .line 314
      invoke-direct {p0}, Landroid/graphics/drawable/Drawable$ConstantState;-><init>()V
  
+     .line 315
      if-eqz p1, :cond_0
  
+     .line 316
      if-eqz p3, :cond_1
  
+     .line 317
      iget-object v0, p1, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mDrawable:Landroid/graphics/drawable/Drawable;
  
      invoke-virtual {v0}, Landroid/graphics/drawable/Drawable;->getConstantState()Landroid/graphics/drawable/Drawable$ConstantState;
***************
*** 62,105 ****
--- 66,119 ----
  
      iput-object v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mDrawable:Landroid/graphics/drawable/Drawable;
  
+     .line 321
      :goto_0
      iget-object v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mDrawable:Landroid/graphics/drawable/Drawable;
  
      invoke-virtual {v0, p2}, Landroid/graphics/drawable/Drawable;->setCallback(Landroid/graphics/drawable/Drawable$Callback;)V
  
+     .line 322
      iget-boolean v0, p1, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mPivotXRel:Z
  
      iput-boolean v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mPivotXRel:Z
  
+     .line 323
      iget v0, p1, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mPivotX:F
  
      iput v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mPivotX:F
  
+     .line 324
      iget-boolean v0, p1, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mPivotYRel:Z
  
      iput-boolean v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mPivotYRel:Z
  
+     .line 325
      iget v0, p1, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mPivotY:F
  
      iput v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mPivotY:F
  
+     .line 326
      iget v0, p1, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mFramesCount:I
  
      iput v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mFramesCount:I
  
+     .line 327
      iget v0, p1, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mFrameDuration:I
  
      iput v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mFrameDuration:I
  
+     .line 328
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mCheckedConstantState:Z
  
      iput-boolean v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mCanConstantState:Z
  
+     .line 330
      :cond_0
      return-void
  
+     .line 319
      :cond_1
      iget-object v0, p1, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mDrawable:Landroid/graphics/drawable/Drawable;
  
***************
*** 124,133 ****
--- 138,149 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 348
      iget-boolean v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mCheckedConstantState:Z
  
      if-nez v0, :cond_0
  
+     .line 349
      iget-object v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mDrawable:Landroid/graphics/drawable/Drawable;
  
      invoke-virtual {v0}, Landroid/graphics/drawable/Drawable;->getConstantState()Landroid/graphics/drawable/Drawable$ConstantState;
***************
*** 141,153 ****
--- 157,172 ----
      :goto_0
      iput-boolean v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mCanConstantState:Z
  
+     .line 350
      iput-boolean v1, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mCheckedConstantState:Z
  
+     .line 353
      :cond_0
      iget-boolean v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mCanConstantState:Z
  
      return v0
  
+     .line 349
      :cond_1
      const/4 v0, 0x0
  
***************
*** 158,163 ****
--- 177,183 ----
      .locals 1
  
      .prologue
+     .line 344
      iget v0, p0, Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;->mChangingConfigurations:I
  
      return v0
***************
*** 169,174 ****
--- 189,195 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 334
      new-instance v0, Landroid/graphics/drawable/AnimatedRotateDrawable;
  
      invoke-direct {v0, p0, v1, v1}, Landroid/graphics/drawable/AnimatedRotateDrawable;-><init>(Landroid/graphics/drawable/AnimatedRotateDrawable$AnimatedRotateState;Landroid/content/res/Resources;Landroid/graphics/drawable/AnimatedRotateDrawable$1;)V
***************
*** 181,186 ****
--- 202,208 ----
      .parameter "res"
  
      .prologue
+     .line 339
      new-instance v0, Landroid/graphics/drawable/AnimatedRotateDrawable;
  
      const/4 v1, 0x0
