*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/drawable/BitmapDrawable.smali	2012-03-23 19:35:42.255357411 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/drawable/BitmapDrawable.smali.original	2012-03-23 19:34:44.798309684 -0500
***************
*** 41,54 ****
--- 41,57 ----
      .end annotation
  
      .prologue
+     .line 81
      invoke-direct {p0}, Landroid/graphics/drawable/Drawable;-><init>()V
  
+     .line 66
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mDstRect:Landroid/graphics/Rect;
  
+     .line 82
      new-instance v1, Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      const/4 v0, 0x0
***************
*** 59,64 ****
--- 62,68 ----
  
      iput-object v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
+     .line 83
      return-void
  .end method
  
***************
*** 67,80 ****
--- 71,87 ----
      .parameter "res"
  
      .prologue
+     .line 90
      invoke-direct {p0}, Landroid/graphics/drawable/Drawable;-><init>()V
  
+     .line 66
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mDstRect:Landroid/graphics/Rect;
  
+     .line 91
      new-instance v1, Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      const/4 v0, 0x0
***************
*** 85,96 ****
--- 92,105 ----
  
      iput-object v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
+     .line 92
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mTargetDensity:I
  
      iput v1, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTargetDensity:I
  
+     .line 93
      return-void
  .end method
  
***************
*** 100,117 ****
--- 109,129 ----
      .parameter "bitmap"
  
      .prologue
+     .line 110
      new-instance v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      invoke-direct {v0, p2}, Landroid/graphics/drawable/BitmapDrawable$BitmapState;-><init>(Landroid/graphics/Bitmap;)V
  
      invoke-direct {p0, v0, p1}, Landroid/graphics/drawable/BitmapDrawable;-><init>(Landroid/graphics/drawable/BitmapDrawable$BitmapState;Landroid/content/res/Resources;)V
  
+     .line 111
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mTargetDensity:I
  
      iput v1, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTargetDensity:I
  
+     .line 112
      return-void
  .end method
  
***************
*** 121,126 ****
--- 133,139 ----
      .parameter "is"
  
      .prologue
+     .line 157
      new-instance v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      invoke-static {p2}, Landroid/graphics/BitmapFactory;->decodeStream(Ljava/io/InputStream;)Landroid/graphics/Bitmap;
***************
*** 133,148 ****
--- 146,164 ----
  
      invoke-direct {p0, v0, v1}, Landroid/graphics/drawable/BitmapDrawable;-><init>(Landroid/graphics/drawable/BitmapDrawable$BitmapState;Landroid/content/res/Resources;)V
  
+     .line 158
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mTargetDensity:I
  
      iput v1, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTargetDensity:I
  
+     .line 159
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
      if-nez v0, :cond_0
  
+     .line 160
      const-string v0, "BitmapDrawable"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 165,170 ****
--- 181,187 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 162
      :cond_0
      return-void
  .end method
***************
*** 175,180 ****
--- 192,198 ----
      .parameter "filepath"
  
      .prologue
+     .line 132
      new-instance v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      invoke-static {p2}, Landroid/graphics/BitmapFactory;->decodeFile(Ljava/lang/String;)Landroid/graphics/Bitmap;
***************
*** 187,202 ****
--- 205,223 ----
  
      invoke-direct {p0, v0, v1}, Landroid/graphics/drawable/BitmapDrawable;-><init>(Landroid/graphics/drawable/BitmapDrawable$BitmapState;Landroid/content/res/Resources;)V
  
+     .line 133
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mTargetDensity:I
  
      iput v1, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTargetDensity:I
  
+     .line 134
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
      if-nez v0, :cond_0
  
+     .line 135
      const-string v0, "BitmapDrawable"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 219,224 ****
--- 240,246 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 137
      :cond_0
      return-void
  .end method
***************
*** 230,235 ****
--- 252,258 ----
      .end annotation
  
      .prologue
+     .line 102
      new-instance v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      invoke-direct {v0, p1}, Landroid/graphics/drawable/BitmapDrawable$BitmapState;-><init>(Landroid/graphics/Bitmap;)V
***************
*** 238,243 ****
--- 261,267 ----
  
      invoke-direct {p0, v0, v1}, Landroid/graphics/drawable/BitmapDrawable;-><init>(Landroid/graphics/drawable/BitmapDrawable$BitmapState;Landroid/content/res/Resources;)V
  
+     .line 103
      return-void
  .end method
  
***************
*** 247,264 ****
--- 271,293 ----
      .parameter "res"
  
      .prologue
+     .line 548
      invoke-direct {p0}, Landroid/graphics/drawable/Drawable;-><init>()V
  
+     .line 66
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mDstRect:Landroid/graphics/Rect;
  
+     .line 549
      iput-object p1, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
+     .line 550
      if-eqz p2, :cond_0
  
+     .line 551
      invoke-virtual {p2}, Landroid/content/res/Resources;->getDisplayMetrics()Landroid/util/DisplayMetrics;
  
      move-result-object v0
***************
*** 267,272 ****
--- 296,302 ----
  
      iput v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mTargetDensity:I
  
+     .line 555
      :goto_0
      if-eqz p1, :cond_1
  
***************
*** 275,282 ****
--- 305,314 ----
      :goto_1
      invoke-direct {p0, v0}, Landroid/graphics/drawable/BitmapDrawable;->setBitmap(Landroid/graphics/Bitmap;)V
  
+     .line 556
      return-void
  
+     .line 553
      :cond_0
      iget v0, p1, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTargetDensity:I
  
***************
*** 284,289 ****
--- 316,322 ----
  
      goto :goto_0
  
+     .line 555
      :cond_1
      const/4 v0, 0x0
  
***************
*** 297,302 ****
--- 330,336 ----
      .parameter "x2"
  
      .prologue
+     .line 58
      invoke-direct {p0, p1, p2}, Landroid/graphics/drawable/BitmapDrawable;-><init>(Landroid/graphics/drawable/BitmapDrawable$BitmapState;Landroid/content/res/Resources;)V
  
      return-void
***************
*** 309,314 ****
--- 343,349 ----
      .end annotation
  
      .prologue
+     .line 146
      new-instance v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      invoke-static {p1}, Landroid/graphics/BitmapFactory;->decodeStream(Ljava/io/InputStream;)Landroid/graphics/Bitmap;
***************
*** 321,330 ****
--- 356,367 ----
  
      invoke-direct {p0, v0, v1}, Landroid/graphics/drawable/BitmapDrawable;-><init>(Landroid/graphics/drawable/BitmapDrawable$BitmapState;Landroid/content/res/Resources;)V
  
+     .line 147
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
      if-nez v0, :cond_0
  
+     .line 148
      const-string v0, "BitmapDrawable"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 347,352 ****
--- 384,390 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 150
      :cond_0
      return-void
  .end method
***************
*** 358,363 ****
--- 396,402 ----
      .end annotation
  
      .prologue
+     .line 121
      new-instance v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      invoke-static {p1}, Landroid/graphics/BitmapFactory;->decodeFile(Ljava/lang/String;)Landroid/graphics/Bitmap;
***************
*** 370,379 ****
--- 409,420 ----
  
      invoke-direct {p0, v0, v1}, Landroid/graphics/drawable/BitmapDrawable;-><init>(Landroid/graphics/drawable/BitmapDrawable$BitmapState;Landroid/content/res/Resources;)V
  
+     .line 122
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
      if-nez v0, :cond_0
  
+     .line 123
      const-string v0, "BitmapDrawable"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 396,401 ****
--- 437,443 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 125
      :cond_0
      return-void
  .end method
***************
*** 404,409 ****
--- 446,452 ----
      .locals 2
  
      .prologue
+     .line 179
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
      iget v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mTargetDensity:I
***************
*** 414,419 ****
--- 457,463 ----
  
      iput v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapWidth:I
  
+     .line 180
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
      iget v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mTargetDensity:I
***************
*** 424,429 ****
--- 468,474 ----
  
      iput v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapHeight:I
  
+     .line 181
      return-void
  .end method
  
***************
*** 432,453 ****
--- 477,505 ----
      .parameter "bitmap"
  
      .prologue
+     .line 184
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
      if-eq p1, v0, :cond_0
  
+     .line 185
      iput-object p1, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
+     .line 186
      if-eqz p1, :cond_1
  
+     .line 187
      invoke-direct {p0}, Landroid/graphics/drawable/BitmapDrawable;->computeBitmapSize()V
  
+     .line 191
      :goto_0
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->invalidateSelf()V
  
+     .line 193
      :cond_0
      return-void
  
+     .line 189
      :cond_1
      const/4 v0, -0x1
  
***************
*** 469,509 ****
--- 521,572 ----
  
      const/4 v11, 0x0
  
+     .line 367
      iget-object v6, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
+     .line 368
      .local v6, bitmap:Landroid/graphics/Bitmap;
      if-eqz v6, :cond_2
  
+     .line 369
      iget-object v8, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
+     .line 370
      .local v8, state:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
      iget-boolean v0, v8, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mRebuildShader:Z
  
      if-eqz v0, :cond_0
  
+     .line 371
      iget-object v9, v8, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTileModeX:Landroid/graphics/Shader$TileMode;
  
+     .line 372
      .local v9, tmx:Landroid/graphics/Shader$TileMode;
      iget-object v10, v8, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTileModeY:Landroid/graphics/Shader$TileMode;
  
+     .line 374
      .local v10, tmy:Landroid/graphics/Shader$TileMode;
      if-nez v9, :cond_3
  
      if-nez v10, :cond_3
  
+     .line 375
      iget-object v0, v8, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
  
      invoke-virtual {v0, v12}, Landroid/graphics/Paint;->setShader(Landroid/graphics/Shader;)Landroid/graphics/Shader;
  
+     .line 381
      .end local v9           #tmx:Landroid/graphics/Shader$TileMode;
      .end local v10           #tmy:Landroid/graphics/Shader$TileMode;
      :goto_0
      iput-boolean v11, v8, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mRebuildShader:Z
  
+     .line 382
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mDstRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0, v0}, Landroid/graphics/drawable/BitmapDrawable;->copyBounds(Landroid/graphics/Rect;)V
  
+     .line 385
      :cond_0
      iget-object v0, v8, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
  
***************
*** 511,527 ****
--- 574,594 ----
  
      move-result-object v7
  
+     .line 386
      .local v7, shader:Landroid/graphics/Shader;
      if-nez v7, :cond_6
  
+     .line 387
      iget-boolean v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mApplyGravity:Z
  
      if-eqz v0, :cond_1
  
+     .line 388
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->getResolvedLayoutDirectionSelf()I
  
      move-result v5
  
+     .line 389
      .local v5, layoutDirection:I
      iget v0, v8, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mGravity:I
  
***************
*** 537,544 ****
--- 604,613 ----
  
      invoke-static/range {v0 .. v5}, Landroid/view/Gravity;->apply(IIILandroid/graphics/Rect;Landroid/graphics/Rect;I)V
  
+     .line 391
      iput-boolean v11, p0, Landroid/graphics/drawable/BitmapDrawable;->mApplyGravity:Z
  
+     .line 393
      .end local v5           #layoutDirection:I
      :cond_1
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mDstRect:Landroid/graphics/Rect;
***************
*** 547,558 ****
--- 616,629 ----
  
      invoke-virtual {p1, v6, v12, v0, v1}, Landroid/graphics/Canvas;->drawBitmap(Landroid/graphics/Bitmap;Landroid/graphics/Rect;Landroid/graphics/Rect;Landroid/graphics/Paint;)V
  
+     .line 402
      .end local v7           #shader:Landroid/graphics/Shader;
      .end local v8           #state:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
      :cond_2
      :goto_1
      return-void
  
+     .line 377
      .restart local v8       #state:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
      .restart local v9       #tmx:Landroid/graphics/Shader$TileMode;
      .restart local v10       #tmy:Landroid/graphics/Shader$TileMode;
***************
*** 579,596 ****
--- 650,671 ----
  
      goto :goto_0
  
+     .line 395
      .restart local v7       #shader:Landroid/graphics/Shader;
      :cond_6
      iget-boolean v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mApplyGravity:Z
  
      if-eqz v0, :cond_7
  
+     .line 396
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mDstRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0, v0}, Landroid/graphics/drawable/BitmapDrawable;->copyBounds(Landroid/graphics/Rect;)V
  
+     .line 397
      iput-boolean v11, p0, Landroid/graphics/drawable/BitmapDrawable;->mApplyGravity:Z
  
+     .line 399
      :cond_7
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mDstRect:Landroid/graphics/Rect;
  
***************
*** 605,610 ****
--- 680,686 ----
      .locals 1
  
      .prologue
+     .line 175
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
      return-object v0
***************
*** 614,619 ****
--- 690,696 ----
      .locals 2
  
      .prologue
+     .line 356
      invoke-super {p0}, Landroid/graphics/drawable/Drawable;->getChangingConfigurations()I
  
      move-result v0
***************
*** 631,636 ****
--- 708,714 ----
      .locals 2
  
      .prologue
+     .line 503
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->getChangingConfigurations()I
***************
*** 639,644 ****
--- 717,723 ----
  
      iput v1, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mChangingConfigurations:I
  
+     .line 504
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      return-object v0
***************
*** 648,653 ****
--- 727,733 ----
      .locals 1
  
      .prologue
+     .line 244
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mGravity:I
***************
*** 659,664 ****
--- 739,745 ----
      .locals 1
  
      .prologue
+     .line 488
      iget v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapHeight:I
  
      return v0
***************
*** 668,673 ****
--- 749,755 ----
      .locals 1
  
      .prologue
+     .line 483
      iget v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapWidth:I
  
      return v0
***************
*** 679,684 ****
--- 761,767 ----
      .prologue
      const/4 v1, -0x3
  
+     .line 493
      iget-object v2, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget v2, v2, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mGravity:I
***************
*** 687,699 ****
--- 770,785 ----
  
      if-eq v2, v3, :cond_1
  
+     .line 497
      :cond_0
      :goto_0
      return v1
  
+     .line 496
      :cond_1
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
+     .line 497
      .local v0, bm:Landroid/graphics/Bitmap;
      if-eqz v0, :cond_0
  
***************
*** 724,729 ****
--- 810,816 ----
      .locals 1
  
      .prologue
+     .line 168
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
***************
*** 735,740 ****
--- 822,828 ----
      .locals 1
  
      .prologue
+     .line 289
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTileModeX:Landroid/graphics/Shader$TileMode;
***************
*** 746,751 ****
--- 834,840 ----
      .locals 1
  
      .prologue
+     .line 299
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTileModeY:Landroid/graphics/Shader$TileMode;
***************
*** 770,783 ****
--- 859,875 ----
  
      const/4 v7, -0x1
  
+     .line 437
      invoke-super {p0, p1, p2, p3}, Landroid/graphics/drawable/Drawable;->inflate(Landroid/content/res/Resources;Lorg/xmlpull/v1/XmlPullParser;Landroid/util/AttributeSet;)V
  
+     .line 439
      sget-object v5, Lcom/android/internal/R$styleable;->BitmapDrawable:[I
  
      invoke-virtual {p1, p3, v5}, Landroid/content/res/Resources;->obtainAttributes(Landroid/util/AttributeSet;[I)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 441
      .local v0, a:Landroid/content/res/TypedArray;
      const/4 v5, 0x1
  
***************
*** 785,793 ****
--- 877,887 ----
  
      move-result v2
  
+     .line 442
      .local v2, id:I
      if-nez v2, :cond_0
  
+     .line 443
      new-instance v5, Lorg/xmlpull/v1/XmlPullParserException;
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 816,829 ****
--- 910,926 ----
  
      throw v5
  
+     .line 446
      :cond_0
      invoke-static {p1, v2}, Landroid/graphics/BitmapFactory;->decodeResource(Landroid/content/res/Resources;I)Landroid/graphics/Bitmap;
  
      move-result-object v1
  
+     .line 447
      .local v1, bitmap:Landroid/graphics/Bitmap;
      if-nez v1, :cond_1
  
+     .line 448
      new-instance v5, Lorg/xmlpull/v1/XmlPullParserException;
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 852,874 ****
--- 949,976 ----
  
      throw v5
  
+     .line 451
      :cond_1
      iget-object v5, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iput-object v1, v5, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mBitmap:Landroid/graphics/Bitmap;
  
+     .line 452
      invoke-direct {p0, v1}, Landroid/graphics/drawable/BitmapDrawable;->setBitmap(Landroid/graphics/Bitmap;)V
  
+     .line 453
      invoke-virtual {p1}, Landroid/content/res/Resources;->getDisplayMetrics()Landroid/util/DisplayMetrics;
  
      move-result-object v5
  
      invoke-virtual {p0, v5}, Landroid/graphics/drawable/BitmapDrawable;->setTargetDensity(Landroid/util/DisplayMetrics;)V
  
+     .line 455
      iget-object v5, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v3, v5, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
  
+     .line 456
      .local v3, paint:Landroid/graphics/Paint;
      const/4 v5, 0x2
  
***************
*** 882,887 ****
--- 984,990 ----
  
      invoke-virtual {v3, v5}, Landroid/graphics/Paint;->setAntiAlias(Z)V
  
+     .line 458
      const/4 v5, 0x3
  
      invoke-virtual {v3}, Landroid/graphics/Paint;->isFilterBitmap()Z
***************
*** 894,899 ****
--- 997,1003 ----
  
      invoke-virtual {v3, v5}, Landroid/graphics/Paint;->setFilterBitmap(Z)V
  
+     .line 460
      const/4 v5, 0x4
  
      invoke-virtual {v3}, Landroid/graphics/Paint;->isDither()Z
***************
*** 906,911 ****
--- 1010,1016 ----
  
      invoke-virtual {v3, v5}, Landroid/graphics/Paint;->setDither(Z)V
  
+     .line 462
      const/16 v5, 0x77
  
      invoke-virtual {v0, v8, v5}, Landroid/content/res/TypedArray;->getInt(II)I
***************
*** 914,936 ****
--- 1019,1047 ----
  
      invoke-virtual {p0, v5}, Landroid/graphics/drawable/BitmapDrawable;->setGravity(I)V
  
+     .line 463
      const/4 v5, 0x5
  
      invoke-virtual {v0, v5, v7}, Landroid/content/res/TypedArray;->getInt(II)I
  
      move-result v4
  
+     .line 464
      .local v4, tileMode:I
      if-eq v4, v7, :cond_2
  
+     .line 465
      packed-switch v4, :pswitch_data_0
  
+     .line 478
      :cond_2
      :goto_0
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 479
      return-void
  
+     .line 467
      :pswitch_0
      sget-object v5, Landroid/graphics/Shader$TileMode;->CLAMP:Landroid/graphics/Shader$TileMode;
  
***************
*** 940,945 ****
--- 1051,1057 ----
  
      goto :goto_0
  
+     .line 470
      :pswitch_1
      sget-object v5, Landroid/graphics/Shader$TileMode;->REPEAT:Landroid/graphics/Shader$TileMode;
  
***************
*** 949,954 ****
--- 1061,1067 ----
  
      goto :goto_0
  
+     .line 473
      :pswitch_2
      sget-object v5, Landroid/graphics/Shader$TileMode;->MIRROR:Landroid/graphics/Shader$TileMode;
  
***************
*** 958,963 ****
--- 1071,1077 ----
  
      goto :goto_0
  
+     .line 465
      nop
  
      :pswitch_data_0
***************
*** 972,977 ****
--- 1086,1092 ----
      .locals 2
  
      .prologue
+     .line 427
      iget-boolean v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mMutated:Z
  
      if-nez v0, :cond_0
***************
*** 982,987 ****
--- 1097,1103 ----
  
      if-ne v0, p0, :cond_0
  
+     .line 428
      new-instance v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
***************
*** 990,999 ****
--- 1106,1117 ----
  
      iput-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
+     .line 429
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mMutated:Z
  
+     .line 431
      :cond_0
      return-object p0
  .end method
***************
*** 1003,1014 ****
--- 1121,1135 ----
      .parameter "bounds"
  
      .prologue
+     .line 361
      invoke-super {p0, p1}, Landroid/graphics/drawable/Drawable;->onBoundsChange(Landroid/graphics/Rect;)V
  
+     .line 362
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mApplyGravity:Z
  
+     .line 363
      return-void
  .end method
  
***************
*** 1017,1022 ****
--- 1138,1144 ----
      .parameter "alpha"
  
      .prologue
+     .line 406
      iget-object v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v1, v1, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
***************
*** 1025,1041 ****
--- 1147,1167 ----
  
      move-result v0
  
+     .line 407
      .local v0, oldAlpha:I
      if-eq p1, v0, :cond_0
  
+     .line 408
      iget-object v1, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v1, v1, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
  
      invoke-virtual {v1, p1}, Landroid/graphics/Paint;->setAlpha(I)V
  
+     .line 409
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->invalidateSelf()V
  
+     .line 411
      :cond_0
      return-void
  .end method
***************
*** 1045,1058 ****
--- 1171,1187 ----
      .parameter "aa"
  
      .prologue
+     .line 266
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
  
      invoke-virtual {v0, p1}, Landroid/graphics/Paint;->setAntiAlias(Z)V
  
+     .line 267
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->invalidateSelf()V
  
+     .line 268
      return-void
  .end method
  
***************
*** 1061,1074 ****
--- 1190,1206 ----
      .parameter "cf"
  
      .prologue
+     .line 415
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
  
      invoke-virtual {v0, p1}, Landroid/graphics/Paint;->setColorFilter(Landroid/graphics/ColorFilter;)Landroid/graphics/ColorFilter;
  
+     .line 416
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->invalidateSelf()V
  
+     .line 417
      return-void
  .end method
  
***************
*** 1077,1090 ****
--- 1209,1225 ----
      .parameter "dither"
  
      .prologue
+     .line 278
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
  
      invoke-virtual {v0, p1}, Landroid/graphics/Paint;->setDither(Z)V
  
+     .line 279
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->invalidateSelf()V
  
+     .line 280
      return-void
  .end method
  
***************
*** 1093,1106 ****
--- 1228,1244 ----
      .parameter "filter"
  
      .prologue
+     .line 272
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mPaint:Landroid/graphics/Paint;
  
      invoke-virtual {v0, p1}, Landroid/graphics/Paint;->setFilterBitmap(Z)V
  
+     .line 273
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->invalidateSelf()V
  
+     .line 274
      return-void
  .end method
  
***************
*** 1109,1130 ****
--- 1247,1273 ----
      .parameter "gravity"
  
      .prologue
+     .line 252
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mGravity:I
  
      if-eq v0, p1, :cond_0
  
+     .line 253
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iput p1, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mGravity:I
  
+     .line 254
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mApplyGravity:Z
  
+     .line 255
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->invalidateSelf()V
  
+     .line 257
      :cond_0
      return-void
  .end method
***************
*** 1134,1143 ****
--- 1277,1288 ----
      .parameter "density"
  
      .prologue
+     .line 230
      iget v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mTargetDensity:I
  
      if-eq v0, p1, :cond_2
  
+     .line 231
      if-nez p1, :cond_0
  
      const/16 p1, 0xa0
***************
*** 1146,1160 ****
--- 1291,1309 ----
      :cond_0
      iput p1, p0, Landroid/graphics/drawable/BitmapDrawable;->mTargetDensity:I
  
+     .line 232
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmap:Landroid/graphics/Bitmap;
  
      if-eqz v0, :cond_1
  
+     .line 233
      invoke-direct {p0}, Landroid/graphics/drawable/BitmapDrawable;->computeBitmapSize()V
  
+     .line 235
      :cond_1
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->invalidateSelf()V
  
+     .line 237
      :cond_2
      return-void
  .end method
***************
*** 1164,1175 ****
--- 1313,1326 ----
      .parameter "canvas"
  
      .prologue
+     .line 206
      invoke-virtual {p1}, Landroid/graphics/Canvas;->getDensity()I
  
      move-result v0
  
      invoke-virtual {p0, v0}, Landroid/graphics/drawable/BitmapDrawable;->setTargetDensity(I)V
  
+     .line 207
      return-void
  .end method
  
***************
*** 1178,1187 ****
--- 1329,1340 ----
      .parameter "metrics"
  
      .prologue
+     .line 218
      iget v0, p1, Landroid/util/DisplayMetrics;->densityDpi:I
  
      invoke-virtual {p0, v0}, Landroid/graphics/drawable/BitmapDrawable;->setTargetDensity(I)V
  
+     .line 219
      return-void
  .end method
  
***************
*** 1190,1201 ****
--- 1343,1356 ----
      .parameter "mode"
  
      .prologue
+     .line 314
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTileModeY:Landroid/graphics/Shader$TileMode;
  
      invoke-virtual {p0, p1, v0}, Landroid/graphics/drawable/BitmapDrawable;->setTileModeXY(Landroid/graphics/Shader$TileMode;Landroid/graphics/Shader$TileMode;)V
  
+     .line 315
      return-void
  .end method
  
***************
*** 1205,1212 ****
--- 1360,1369 ----
      .parameter "ymode"
  
      .prologue
+     .line 345
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
+     .line 346
      .local v0, state:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
      iget-object v1, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTileModeX:Landroid/graphics/Shader$TileMode;
  
***************
*** 1216,1232 ****
--- 1373,1394 ----
  
      if-eq v1, p2, :cond_1
  
+     .line 347
      :cond_0
      iput-object p1, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTileModeX:Landroid/graphics/Shader$TileMode;
  
+     .line 348
      iput-object p2, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTileModeY:Landroid/graphics/Shader$TileMode;
  
+     .line 349
      const/4 v1, 0x1
  
      iput-boolean v1, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mRebuildShader:Z
  
+     .line 350
      invoke-virtual {p0}, Landroid/graphics/drawable/BitmapDrawable;->invalidateSelf()V
  
+     .line 352
      :cond_1
      return-void
  .end method
***************
*** 1236,1246 ****
--- 1398,1410 ----
      .parameter "mode"
  
      .prologue
+     .line 329
      iget-object v0, p0, Landroid/graphics/drawable/BitmapDrawable;->mBitmapState:Landroid/graphics/drawable/BitmapDrawable$BitmapState;
  
      iget-object v0, v0, Landroid/graphics/drawable/BitmapDrawable$BitmapState;->mTileModeX:Landroid/graphics/Shader$TileMode;
  
      invoke-virtual {p0, v0, p1}, Landroid/graphics/drawable/BitmapDrawable;->setTileModeXY(Landroid/graphics/Shader$TileMode;Landroid/graphics/Shader$TileMode;)V
  
+     .line 330
      return-void
  .end method
