*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/drawable/GradientDrawable$GradientState.smali	2012-03-23 19:35:42.487361641 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/drawable/GradientDrawable$GradientState.smali.original	2012-03-23 19:34:44.800309720 -0500
***************
*** 83,110 ****
--- 83,121 ----
  
      const/high16 v0, 0x3f00
  
+     .line 916
      invoke-direct {p0}, Landroid/graphics/drawable/Drawable$ConstantState;-><init>()V
  
+     .line 887
      iput v2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mShape:I
  
+     .line 888
      iput v2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradient:I
  
+     .line 896
      iput v1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeWidth:I
  
+     .line 903
      iput v1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mWidth:I
  
+     .line 904
      iput v1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mHeight:I
  
+     .line 909
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterX:F
  
+     .line 910
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterY:F
  
+     .line 911
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradientRadius:F
  
+     .line 917
      sget-object v0, Landroid/graphics/drawable/GradientDrawable$Orientation;->TOP_BOTTOM:Landroid/graphics/drawable/GradientDrawable$Orientation;
  
      iput-object v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mOrientation:Landroid/graphics/drawable/GradientDrawable$Orientation;
  
+     .line 918
      return-void
  .end method
  
***************
*** 119,162 ****
--- 130,188 ----
  
      const/high16 v0, 0x3f00
  
+     .line 925
      invoke-direct {p0}, Landroid/graphics/drawable/Drawable$ConstantState;-><init>()V
  
+     .line 887
      iput v2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mShape:I
  
+     .line 888
      iput v2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradient:I
  
+     .line 896
      iput v1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeWidth:I
  
+     .line 903
      iput v1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mWidth:I
  
+     .line 904
      iput v1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mHeight:I
  
+     .line 909
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterX:F
  
+     .line 910
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterY:F
  
+     .line 911
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradientRadius:F
  
+     .line 926
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mChangingConfigurations:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mChangingConfigurations:I
  
+     .line 927
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mShape:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mShape:I
  
+     .line 928
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradient:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradient:I
  
+     .line 929
      iget-object v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mOrientation:Landroid/graphics/drawable/GradientDrawable$Orientation;
  
      iput-object v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mOrientation:Landroid/graphics/drawable/GradientDrawable$Orientation;
  
+     .line 930
      iget-object v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mColors:[I
  
      if-eqz v0, :cond_0
  
+     .line 931
      iget-object v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mColors:[I
  
      invoke-virtual {v0}, [I->clone()Ljava/lang/Object;
***************
*** 167,177 ****
--- 193,205 ----
  
      iput-object v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mColors:[I
  
+     .line 933
      :cond_0
      iget-object v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mPositions:[F
  
      if-eqz v0, :cond_1
  
+     .line 934
      iget-object v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mPositions:[F
  
      invoke-virtual {v0}, [F->clone()Ljava/lang/Object;
***************
*** 182,220 ****
--- 210,257 ----
  
      iput-object v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mPositions:[F
  
+     .line 936
      :cond_1
      iget-boolean v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mHasSolidColor:Z
  
      iput-boolean v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mHasSolidColor:Z
  
+     .line 937
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mSolidColor:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mSolidColor:I
  
+     .line 938
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeWidth:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeWidth:I
  
+     .line 939
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeColor:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeColor:I
  
+     .line 940
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeDashWidth:F
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeDashWidth:F
  
+     .line 941
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeDashGap:F
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeDashGap:F
  
+     .line 942
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mRadius:F
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mRadius:F
  
+     .line 943
      iget-object v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mRadiusArray:[F
  
      if-eqz v0, :cond_2
  
+     .line 944
      iget-object v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mRadiusArray:[F
  
      invoke-virtual {v0}, [F->clone()Ljava/lang/Object;
***************
*** 225,235 ****
--- 262,274 ----
  
      iput-object v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mRadiusArray:[F
  
+     .line 946
      :cond_2
      iget-object v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mPadding:Landroid/graphics/Rect;
  
      if-eqz v0, :cond_3
  
+     .line 947
      new-instance v0, Landroid/graphics/Rect;
  
      iget-object v1, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mPadding:Landroid/graphics/Rect;
***************
*** 238,288 ****
--- 277,339 ----
  
      iput-object v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mPadding:Landroid/graphics/Rect;
  
+     .line 949
      :cond_3
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mWidth:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mWidth:I
  
+     .line 950
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mHeight:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mHeight:I
  
+     .line 951
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mInnerRadiusRatio:F
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mInnerRadiusRatio:F
  
+     .line 952
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mThicknessRatio:F
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mThicknessRatio:F
  
+     .line 953
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mInnerRadius:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mInnerRadius:I
  
+     .line 954
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mThickness:I
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mThickness:I
  
+     .line 955
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterX:F
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterX:F
  
+     .line 956
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterY:F
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterY:F
  
+     .line 957
      iget v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradientRadius:F
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradientRadius:F
  
+     .line 958
      iget-boolean v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mUseLevel:Z
  
      iput-boolean v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mUseLevel:Z
  
+     .line 959
      iget-boolean v0, p1, Landroid/graphics/drawable/GradientDrawable$GradientState;->mUseLevelForShape:Z
  
      iput-boolean v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mUseLevelForShape:Z
  
+     .line 960
      return-void
  .end method
  
***************
*** 298,325 ****
--- 349,388 ----
  
      const/high16 v0, 0x3f00
  
+     .line 920
      invoke-direct {p0}, Landroid/graphics/drawable/Drawable$ConstantState;-><init>()V
  
+     .line 887
      iput v2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mShape:I
  
+     .line 888
      iput v2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradient:I
  
+     .line 896
      iput v1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeWidth:I
  
+     .line 903
      iput v1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mWidth:I
  
+     .line 904
      iput v1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mHeight:I
  
+     .line 909
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterX:F
  
+     .line 910
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterY:F
  
+     .line 911
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradientRadius:F
  
+     .line 921
      iput-object p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mOrientation:Landroid/graphics/drawable/GradientDrawable$Orientation;
  
+     .line 922
      iput-object p2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mColors:[I
  
+     .line 923
      return-void
  .end method
  
***************
*** 328,333 ****
--- 391,397 ----
      .parameter "x0"
  
      .prologue
+     .line 885
      iget-boolean v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mUseLevel:Z
  
      return v0
***************
*** 339,344 ****
--- 403,409 ----
      .parameter "x1"
  
      .prologue
+     .line 885
      iput-boolean p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mUseLevel:Z
  
      return p1
***************
*** 349,354 ****
--- 414,420 ----
      .parameter "x0"
  
      .prologue
+     .line 885
      iget-boolean v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mUseLevelForShape:Z
  
      return v0
***************
*** 360,365 ****
--- 426,432 ----
      .parameter "x1"
  
      .prologue
+     .line 885
      iput-boolean p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mUseLevelForShape:Z
  
      return p1
***************
*** 370,375 ****
--- 437,443 ----
      .parameter "x0"
  
      .prologue
+     .line 885
      iget v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterX:F
  
      return v0
***************
*** 381,386 ****
--- 449,455 ----
      .parameter "x1"
  
      .prologue
+     .line 885
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterX:F
  
      return p1
***************
*** 391,396 ****
--- 460,466 ----
      .parameter "x0"
  
      .prologue
+     .line 885
      iget v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterY:F
  
      return v0
***************
*** 402,407 ****
--- 472,478 ----
      .parameter "x1"
  
      .prologue
+     .line 885
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterY:F
  
      return p1
***************
*** 412,417 ****
--- 483,489 ----
      .parameter "x0"
  
      .prologue
+     .line 885
      iget v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradientRadius:F
  
      return v0
***************
*** 423,428 ****
--- 495,501 ----
      .parameter "x1"
  
      .prologue
+     .line 885
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradientRadius:F
  
      return p1
***************
*** 434,439 ****
--- 507,513 ----
      .locals 1
  
      .prologue
+     .line 974
      iget v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mChangingConfigurations:I
  
      return v0
***************
*** 443,448 ****
--- 517,523 ----
      .locals 2
  
      .prologue
+     .line 964
      new-instance v0, Landroid/graphics/drawable/GradientDrawable;
  
      const/4 v1, 0x0
***************
*** 457,462 ****
--- 532,538 ----
      .parameter "res"
  
      .prologue
+     .line 969
      new-instance v0, Landroid/graphics/drawable/GradientDrawable;
  
      const/4 v1, 0x0
***************
*** 471,484 ****
--- 547,564 ----
      .parameter "radii"
  
      .prologue
+     .line 1017
      iput-object p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mRadiusArray:[F
  
+     .line 1018
      if-nez p1, :cond_0
  
+     .line 1019
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mRadius:F
  
+     .line 1021
      :cond_0
      return-void
  .end method
***************
*** 488,508 ****
--- 568,593 ----
      .parameter "radius"
  
      .prologue
+     .line 1009
      const/4 v0, 0x0
  
      cmpg-float v0, p1, v0
  
      if-gez v0, :cond_0
  
+     .line 1010
      const/4 p1, 0x0
  
+     .line 1012
      :cond_0
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mRadius:F
  
+     .line 1013
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mRadiusArray:[F
  
+     .line 1014
      return-void
  .end method
  
***************
*** 512,521 ****
--- 597,609 ----
      .parameter "y"
  
      .prologue
+     .line 986
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterX:F
  
+     .line 987
      iput p2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mCenterY:F
  
+     .line 988
      return-void
  .end method
  
***************
*** 524,531 ****
--- 612,621 ----
      .parameter "gradientRadius"
  
      .prologue
+     .line 1029
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradientRadius:F
  
+     .line 1030
      return-void
  .end method
  
***************
*** 534,541 ****
--- 624,633 ----
      .parameter "gradient"
  
      .prologue
+     .line 982
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mGradient:I
  
+     .line 983
      return-void
  .end method
  
***************
*** 544,551 ****
--- 636,645 ----
      .parameter "shape"
  
      .prologue
+     .line 978
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mShape:I
  
+     .line 979
      return-void
  .end method
  
***************
*** 555,564 ****
--- 649,661 ----
      .parameter "height"
  
      .prologue
+     .line 1024
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mWidth:I
  
+     .line 1025
      iput p2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mHeight:I
  
+     .line 1026
      return-void
  .end method
  
***************
*** 567,582 ****
--- 664,683 ----
      .parameter "argb"
  
      .prologue
+     .line 991
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mHasSolidColor:Z
  
+     .line 992
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mSolidColor:I
  
+     .line 993
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mColors:[I
  
+     .line 994
      return-void
  .end method
  
***************
*** 586,595 ****
--- 687,699 ----
      .parameter "color"
  
      .prologue
+     .line 997
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeWidth:I
  
+     .line 998
      iput p2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeColor:I
  
+     .line 999
      return-void
  .end method
  
***************
*** 601,613 ****
--- 705,722 ----
      .parameter "dashGap"
  
      .prologue
+     .line 1002
      iput p1, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeWidth:I
  
+     .line 1003
      iput p2, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeColor:I
  
+     .line 1004
      iput p3, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeDashWidth:F
  
+     .line 1005
      iput p4, p0, Landroid/graphics/drawable/GradientDrawable$GradientState;->mStrokeDashGap:F
  
+     .line 1006
      return-void
  .end method
