*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/drawable/LayerDrawable$ChildDrawable.smali	2012-03-23 19:35:43.622382338 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/graphics/drawable/LayerDrawable$ChildDrawable.smali.original	2012-03-23 19:34:44.809309884 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .locals 0
  
      .prologue
+     .line 588
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
