*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/AbstractSynthesisCallback.smali	2012-03-23 19:36:33.403290092 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/AbstractSynthesisCallback.smali.original	2012-03-23 19:34:45.190316831 -0500
***************
*** 11,16 ****
--- 11,17 ----
      .locals 0
  
      .prologue
+     .line 22
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
