*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/AudioPlaybackHandler$MessageLoop.smali	2012-03-23 19:36:32.491273462 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/AudioPlaybackHandler$MessageLoop.smali.original	2012-03-23 19:34:45.181316667 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 197
      iput-object p1, p0, Landroid/speech/tts/AudioPlaybackHandler$MessageLoop;->this$0:Landroid/speech/tts/AudioPlaybackHandler;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .parameter "x1"
  
      .prologue
+     .line 197
      invoke-direct {p0, p1}, Landroid/speech/tts/AudioPlaybackHandler$MessageLoop;-><init>(Landroid/speech/tts/AudioPlaybackHandler;)V
  
      return-void
***************
*** 51,59 ****
--- 53,63 ----
      .locals 5
  
      .prologue
+     .line 201
      :goto_0
      const/4 v1, 0x0
  
+     .line 203
      .local v1, entry:Landroid/speech/tts/AudioPlaybackHandler$ListEntry;
      :try_start_0
      iget-object v3, p0, Landroid/speech/tts/AudioPlaybackHandler$MessageLoop;->this$0:Landroid/speech/tts/AudioPlaybackHandler;
***************
*** 75,95 ****
--- 79,104 ----
      :try_end_0
      .catch Ljava/lang/InterruptedException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 208
      iget v3, v1, Landroid/speech/tts/AudioPlaybackHandler$ListEntry;->mWhat:I
  
      const/4 v4, -0x1
  
      if-ne v3, v4, :cond_0
  
+     .line 210
      :goto_1
      return-void
  
+     .line 204
      :catch_0
      move-exception v2
  
+     .line 205
      .local v2, ie:Ljava/lang/InterruptedException;
      goto :goto_1
  
+     .line 218
      .end local v2           #ie:Ljava/lang/InterruptedException;
      :cond_0
      iget-object v3, p0, Landroid/speech/tts/AudioPlaybackHandler$MessageLoop;->this$0:Landroid/speech/tts/AudioPlaybackHandler;
***************
*** 99,109 ****
--- 108,120 ----
      #calls: Landroid/speech/tts/AudioPlaybackHandler;->setCurrentParams(Landroid/speech/tts/MessageParams;)V
      invoke-static {v3, v4}, Landroid/speech/tts/AudioPlaybackHandler;->access$400(Landroid/speech/tts/AudioPlaybackHandler;Landroid/speech/tts/MessageParams;)V
  
+     .line 219
      iget-object v3, p0, Landroid/speech/tts/AudioPlaybackHandler$MessageLoop;->this$0:Landroid/speech/tts/AudioPlaybackHandler;
  
      #calls: Landroid/speech/tts/AudioPlaybackHandler;->handleMessage(Landroid/speech/tts/AudioPlaybackHandler$ListEntry;)V
      invoke-static {v3, v1}, Landroid/speech/tts/AudioPlaybackHandler;->access$500(Landroid/speech/tts/AudioPlaybackHandler;Landroid/speech/tts/AudioPlaybackHandler$ListEntry;)V
  
+     .line 220
      iget-object v3, p0, Landroid/speech/tts/AudioPlaybackHandler$MessageLoop;->this$0:Landroid/speech/tts/AudioPlaybackHandler;
  
      const/4 v4, 0x0
