*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/BlockingMediaPlayer$4.smali	2012-03-23 19:36:32.670276726 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/BlockingMediaPlayer$4.smali.original	2012-03-23 19:34:45.183316703 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 118
      iput-object p1, p0, Landroid/speech/tts/BlockingMediaPlayer$4;->this$0:Landroid/speech/tts/BlockingMediaPlayer;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "mp"
  
      .prologue
+     .line 121
      iget-object v0, p0, Landroid/speech/tts/BlockingMediaPlayer$4;->this$0:Landroid/speech/tts/BlockingMediaPlayer;
  
      const/4 v1, 0x1
***************
*** 48,53 ****
--- 50,56 ----
      #setter for: Landroid/speech/tts/BlockingMediaPlayer;->mFinished:Z
      invoke-static {v0, v1}, Landroid/speech/tts/BlockingMediaPlayer;->access$302(Landroid/speech/tts/BlockingMediaPlayer;Z)Z
  
+     .line 122
      iget-object v0, p0, Landroid/speech/tts/BlockingMediaPlayer$4;->this$0:Landroid/speech/tts/BlockingMediaPlayer;
  
      #getter for: Landroid/speech/tts/BlockingMediaPlayer;->mDone:Landroid/os/ConditionVariable;
***************
*** 57,61 ****
--- 60,65 ----
  
      invoke-virtual {v0}, Landroid/os/ConditionVariable;->open()V
  
+     .line 123
      return-void
  .end method
