*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/EventLogTags.smali	2012-03-23 19:36:32.648276324 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/EventLogTags.smali.original	2012-03-23 19:34:45.182316685 -0500
***************
*** 14,19 ****
--- 14,20 ----
      .locals 0
  
      .prologue
+     .line 11
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 29,34 ****
--- 30,36 ----
      .parameter "pitch"
  
      .prologue
+     .line 24
      const v0, 0x128e2
  
      const/4 v1, 0x6
***************
*** 73,78 ****
--- 75,81 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 25
      return-void
  .end method
  
***************
*** 89,94 ****
--- 92,98 ----
      .parameter "audioLatency"
  
      .prologue
+     .line 20
      const v0, 0x128e1
  
      const/16 v1, 0x9
***************
*** 157,161 ****
--- 161,166 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 21
      return-void
  .end method
