*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/MessageParams.smali	2012-03-23 19:36:33.526292335 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/MessageParams.smali.original	2012-03-23 19:34:45.191316849 -0500
***************
*** 24,35 ****
--- 24,39 ----
      .parameter "callingApp"
  
      .prologue
+     .line 28
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 29
      iput-object p1, p0, Landroid/speech/tts/MessageParams;->mDispatcher:Landroid/speech/tts/TextToSpeechService$UtteranceProgressDispatcher;
  
+     .line 30
      iput-object p2, p0, Landroid/speech/tts/MessageParams;->mCallingApp:Ljava/lang/String;
  
+     .line 31
      return-void
  .end method
  
***************
*** 39,44 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 38
      iget-object v0, p0, Landroid/speech/tts/MessageParams;->mCallingApp:Ljava/lang/String;
  
      return-object v0
***************
*** 48,53 ****
--- 53,59 ----
      .locals 1
  
      .prologue
+     .line 34
      iget-object v0, p0, Landroid/speech/tts/MessageParams;->mDispatcher:Landroid/speech/tts/TextToSpeechService$UtteranceProgressDispatcher;
  
      return-object v0
***************
*** 60,65 ****
--- 66,72 ----
      .locals 2
  
      .prologue
+     .line 43
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
