*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeech$10.smali	2012-03-23 19:36:33.592293538 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeech$10.smali.original	2012-03-23 19:34:45.192316867 -0500
***************
*** 40,45 ****
--- 40,46 ----
      .parameter
  
      .prologue
+     .line 1064
      iput-object p1, p0, Landroid/speech/tts/TextToSpeech$10;->this$0:Landroid/speech/tts/TextToSpeech;
  
      iput-object p2, p0, Landroid/speech/tts/TextToSpeech$10;->val$loc:Ljava/util/Locale;
***************
*** 61,66 ****
--- 62,68 ----
      .end annotation
  
      .prologue
+     .line 1067
      iget-object v0, p0, Landroid/speech/tts/TextToSpeech$10;->val$loc:Ljava/util/Locale;
  
      invoke-virtual {v0}, Ljava/util/Locale;->getISO3Language()Ljava/lang/String;
***************
*** 100,105 ****
--- 102,108 ----
      .end annotation
  
      .prologue
+     .line 1064
      invoke-virtual {p0, p1}, Landroid/speech/tts/TextToSpeech$10;->run(Landroid/speech/tts/ITextToSpeechService;)Ljava/lang/Integer;
  
      move-result-object v0
