*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeech$7.smali	2012-03-23 19:36:33.548292736 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeech$7.smali.original	2012-03-23 19:34:45.191316849 -0500
***************
*** 37,42 ****
--- 37,43 ----
      .parameter
  
      .prologue
+     .line 926
      iput-object p1, p0, Landroid/speech/tts/TextToSpeech$7;->this$0:Landroid/speech/tts/TextToSpeech;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 56,61 ****
--- 57,63 ----
      .end annotation
  
      .prologue
+     .line 929
      iget-object v0, p0, Landroid/speech/tts/TextToSpeech$7;->this$0:Landroid/speech/tts/TextToSpeech;
  
      #calls: Landroid/speech/tts/TextToSpeech;->getPackageName()Ljava/lang/String;
***************
*** 84,89 ****
--- 86,92 ----
      .end annotation
  
      .prologue
+     .line 926
      invoke-virtual {p0, p1}, Landroid/speech/tts/TextToSpeech$7;->run(Landroid/speech/tts/ITextToSpeechService;)Ljava/lang/Integer;
  
      move-result-object v0
