*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeechService$SynthesisSpeechItem.smali	2012-03-23 19:36:33.005282834 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeechService$SynthesisSpeechItem.smali.original	2012-03-23 19:34:45.185316741 -0500
***************
*** 37,48 ****
--- 37,52 ----
      .parameter "text"
  
      .prologue
+     .line 521
      iput-object p1, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->this$0:Landroid/speech/tts/TextToSpeechService;
  
+     .line 522
      invoke-direct {p0, p1, p2, p3}, Landroid/speech/tts/TextToSpeechService$SpeechItem;-><init>(Landroid/speech/tts/TextToSpeechService;Ljava/lang/String;Landroid/os/Bundle;)V
  
+     .line 523
      iput-object p4, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mText:Ljava/lang/String;
  
+     .line 524
      new-instance v0, Landroid/speech/tts/SynthesisRequest;
  
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mText:Ljava/lang/String;
***************
*** 53,58 ****
--- 57,63 ----
  
      iput-object v0, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mSynthesisRequest:Landroid/speech/tts/SynthesisRequest;
  
+     .line 525
      #calls: Landroid/speech/tts/TextToSpeechService;->getSettingsLocale()[Ljava/lang/String;
      invoke-static {p1}, Landroid/speech/tts/TextToSpeechService;->access$400(Landroid/speech/tts/TextToSpeechService;)[Ljava/lang/String;
  
***************
*** 60,69 ****
--- 65,76 ----
  
      iput-object v0, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mDefaultLocale:[Ljava/lang/String;
  
+     .line 526
      iget-object v0, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mSynthesisRequest:Landroid/speech/tts/SynthesisRequest;
  
      invoke-direct {p0, v0}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->setRequestParams(Landroid/speech/tts/SynthesisRequest;)V
  
+     .line 527
      new-instance v0, Landroid/speech/tts/EventLogger;
  
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mSynthesisRequest:Landroid/speech/tts/SynthesisRequest;
***************
*** 81,86 ****
--- 88,94 ----
  
      iput-object v0, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mEventLogger:Landroid/speech/tts/EventLogger;
  
+     .line 528
      return-void
  .end method
  
***************
*** 88,93 ****
--- 96,102 ----
      .locals 2
  
      .prologue
+     .line 600
      invoke-direct {p0}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->hasLanguage()Z
  
      move-result v0
***************
*** 100,105 ****
--- 109,115 ----
  
      aget-object v0, v0, v1
  
+     .line 601
      :goto_0
      return-object v0
  
***************
*** 119,125 ****
      .locals 2
  
      .prologue
!     const-string v0, "pitch"
  
      const/16 v1, 0x64
  
--- 129,136 ----
      .locals 2
  
      .prologue
!     .line 614
!     const-string/jumbo v0, "pitch"
  
      const/16 v1, 0x64
  
***************
*** 134,140 ****
      .locals 2
  
      .prologue
!     const-string v0, "rate"
  
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->this$0:Landroid/speech/tts/TextToSpeechService;
  
--- 145,152 ----
      .locals 2
  
      .prologue
!     .line 610
!     const-string/jumbo v0, "rate"
  
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->this$0:Landroid/speech/tts/TextToSpeechService;
  
***************
*** 154,159 ****
--- 166,172 ----
      .locals 2
  
      .prologue
+     .line 605
      invoke-direct {p0}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->hasLanguage()Z
  
      move-result v0
***************
*** 166,176 ****
  
      aget-object v0, v0, v1
  
      :goto_0
      return-object v0
  
      :cond_0
!     const-string v0, "variant"
  
      const-string v1, ""
  
--- 179,190 ----
  
      aget-object v0, v0, v1
  
+     .line 606
      :goto_0
      return-object v0
  
      :cond_0
!     const-string/jumbo v0, "variant"
  
      const-string v1, ""
  
***************
*** 185,190 ****
--- 199,205 ----
      .locals 2
  
      .prologue
+     .line 596
      const-string v0, "language"
  
      const/4 v1, 0x0
***************
*** 215,220 ****
--- 230,236 ----
      .parameter "request"
  
      .prologue
+     .line 570
      invoke-virtual {p0}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->getLanguage()Ljava/lang/String;
  
      move-result-object v0
***************
*** 229,246 ****
--- 245,265 ----
  
      invoke-virtual {p1, v0, v1, v2}, Landroid/speech/tts/SynthesisRequest;->setLanguage(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 571
      invoke-direct {p0}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->getSpeechRate()I
  
      move-result v0
  
      invoke-virtual {p1, v0}, Landroid/speech/tts/SynthesisRequest;->setSpeechRate(I)V
  
+     .line 573
      invoke-direct {p0}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->getPitch()I
  
      move-result v0
  
      invoke-virtual {p1, v0}, Landroid/speech/tts/SynthesisRequest;->setPitch(I)V
  
+     .line 574
      return-void
  .end method
  
***************
*** 250,255 ****
--- 269,275 ----
      .locals 8
  
      .prologue
+     .line 565
      new-instance v0, Landroid/speech/tts/PlaybackSynthesisCallback;
  
      invoke-virtual {p0}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->getStreamType()I
***************
*** 288,293 ****
--- 308,314 ----
      .locals 3
  
      .prologue
+     .line 592
      const-string v0, "language"
  
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mDefaultLocale:[Ljava/lang/String;
***************
*** 307,312 ****
--- 328,334 ----
      .locals 1
  
      .prologue
+     .line 531
      iget-object v0, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mText:Ljava/lang/String;
  
      return-object v0
***************
*** 318,336 ****
--- 340,362 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 536
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mText:Ljava/lang/String;
  
      if-nez v1, :cond_0
  
+     .line 537
      const-string v1, "TextToSpeechService"
  
      const-string v2, "Got null text"
  
      invoke-static {v1, v2}, Landroid/util/Log;->wtf(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 544
      :goto_0
      return v0
  
+     .line 540
      :cond_0
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mText:Ljava/lang/String;
  
***************
*** 342,347 ****
--- 368,374 ----
  
      if-lt v1, v2, :cond_1
  
+     .line 541
      const-string v1, "TextToSpeechService"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 378,383 ****
--- 405,411 ----
  
      goto :goto_0
  
+     .line 544
      :cond_1
      const/4 v0, 0x1
  
***************
*** 390,401 ****
--- 418,432 ----
      .prologue
      const/4 v1, -0x1
  
+     .line 550
      iget-object v2, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mEventLogger:Landroid/speech/tts/EventLogger;
  
      invoke-virtual {v2}, Landroid/speech/tts/EventLogger;->onRequestProcessingStart()V
  
+     .line 551
      monitor-enter p0
  
+     .line 554
      :try_start_0
      invoke-virtual {p0}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->isStopped()Z
  
***************
*** 403,414 ****
--- 434,448 ----
  
      if-eqz v2, :cond_1
  
+     .line 555
      monitor-exit p0
  
+     .line 561
      :cond_0
      :goto_0
      return v1
  
+     .line 557
      :cond_1
      invoke-virtual {p0}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->createSynthesisCallback()Landroid/speech/tts/AbstractSynthesisCallback;
  
***************
*** 416,434 ****
--- 450,472 ----
  
      iput-object v2, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mSynthesisCallback:Landroid/speech/tts/AbstractSynthesisCallback;
  
+     .line 558
      iget-object v0, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mSynthesisCallback:Landroid/speech/tts/AbstractSynthesisCallback;
  
+     .line 559
      .local v0, synthesisCallback:Landroid/speech/tts/AbstractSynthesisCallback;
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 560
      iget-object v2, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->this$0:Landroid/speech/tts/TextToSpeechService;
  
      iget-object v3, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mSynthesisRequest:Landroid/speech/tts/SynthesisRequest;
  
      invoke-virtual {v2, v3, v0}, Landroid/speech/tts/TextToSpeechService;->onSynthesizeText(Landroid/speech/tts/SynthesisRequest;Landroid/speech/tts/SynthesisCallback;)V
  
+     .line 561
      invoke-virtual {v0}, Landroid/speech/tts/AbstractSynthesisCallback;->isDone()Z
  
      move-result v2
***************
*** 439,444 ****
--- 477,483 ----
  
      goto :goto_0
  
+     .line 559
      .end local v0           #synthesisCallback:Landroid/speech/tts/AbstractSynthesisCallback;
      :catchall_0
      move-exception v1
***************
*** 455,481 ****
--- 494,528 ----
      .locals 2
  
      .prologue
+     .line 579
      monitor-enter p0
  
+     .line 580
      :try_start_0
      iget-object v0, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->mSynthesisCallback:Landroid/speech/tts/AbstractSynthesisCallback;
  
+     .line 581
      .local v0, synthesisCallback:Landroid/speech/tts/AbstractSynthesisCallback;
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 582
      if-eqz v0, :cond_0
  
+     .line 586
      invoke-virtual {v0}, Landroid/speech/tts/AbstractSynthesisCallback;->stop()V
  
+     .line 587
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;->this$0:Landroid/speech/tts/TextToSpeechService;
  
      invoke-virtual {v1}, Landroid/speech/tts/TextToSpeechService;->onStop()V
  
+     .line 589
      :cond_0
      return-void
  
+     .line 581
      .end local v0           #synthesisCallback:Landroid/speech/tts/AbstractSynthesisCallback;
      :catchall_0
      move-exception v1
