*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeechService$1.smali	2012-03-23 19:36:32.932281504 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeechService$1.smali.original	2012-03-23 19:34:45.185316741 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 749
      iput-object p1, p0, Landroid/speech/tts/TextToSpeechService$1;->this$0:Landroid/speech/tts/TextToSpeechService;
  
      invoke-direct {p0}, Landroid/speech/tts/ITextToSpeechService$Stub;-><init>()V
***************
*** 36,41 ****
--- 37,43 ----
      .parameter "args"
  
      .prologue
+     .line 857
      move-object v0, p1
  
      .local v0, arr$:[Ljava/lang/Object;
***************
*** 50,70 ****
--- 52,76 ----
  
      aget-object v3, v0, v1
  
+     .line 858
      .local v3, o:Ljava/lang/Object;
      if-nez v3, :cond_0
  
      const/4 v4, 0x0
  
+     .line 860
      .end local v3           #o:Ljava/lang/Object;
      :goto_1
      return v4
  
+     .line 857
      .restart local v3       #o:Ljava/lang/Object;
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 860
      .end local v3           #o:Ljava/lang/Object;
      :cond_1
      const/4 v4, 0x1
***************
*** 77,82 ****
--- 83,89 ----
      .parameter "in"
  
      .prologue
+     .line 853
      invoke-virtual {p1}, Ljava/lang/String;->intern()Ljava/lang/String;
  
      move-result-object v0
***************
*** 93,121 ****
--- 100,135 ----
      .parameter "variant"
  
      .prologue
+     .line 819
      iget-object v2, p0, Landroid/speech/tts/TextToSpeechService$1;->this$0:Landroid/speech/tts/TextToSpeechService;
  
      invoke-virtual {v2, p1, p2, p3}, Landroid/speech/tts/TextToSpeechService;->onGetFeaturesForLanguage(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Ljava/util/Set;
  
      move-result-object v0
  
+     .line 820
      .local v0, features:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
      const/4 v1, 0x0
  
+     .line 821
      .local v1, featuresArray:[Ljava/lang/String;
      if-eqz v0, :cond_0
  
+     .line 822
      invoke-interface {v0}, Ljava/util/Set;->size()I
  
      move-result v2
  
      new-array v1, v2, [Ljava/lang/String;
  
+     .line 823
      invoke-interface {v0, v1}, Ljava/util/Set;->toArray([Ljava/lang/Object;)[Ljava/lang/Object;
  
+     .line 827
      :goto_0
      return-object v1
  
+     .line 825
      :cond_0
      const/4 v2, 0x0
  
***************
*** 128,133 ****
--- 142,148 ----
      .locals 1
  
      .prologue
+     .line 803
      iget-object v0, p0, Landroid/speech/tts/TextToSpeechService$1;->this$0:Landroid/speech/tts/TextToSpeechService;
  
      invoke-virtual {v0}, Landroid/speech/tts/TextToSpeechService;->onGetLanguage()[Ljava/lang/String;
***************
*** 144,149 ****
--- 159,165 ----
      .parameter "variant"
  
      .prologue
+     .line 811
      const/4 v0, 0x1
  
      new-array v0, v0, [Ljava/lang/Object;
***************
*** 158,165 ****
--- 174,183 ----
  
      if-nez v0, :cond_0
  
+     .line 812
      const/4 v0, -0x1
  
+     .line 815
      :goto_0
      return v0
  
***************
*** 177,182 ****
--- 195,201 ----
      .locals 1
  
      .prologue
+     .line 791
      iget-object v0, p0, Landroid/speech/tts/TextToSpeechService$1;->this$0:Landroid/speech/tts/TextToSpeechService;
  
      #getter for: Landroid/speech/tts/TextToSpeechService;->mSynthHandler:Landroid/speech/tts/TextToSpeechService$SynthHandler;
***************
*** 222,227 ****
--- 241,247 ----
      .parameter "variant"
  
      .prologue
+     .line 835
      const/4 v0, 0x1
  
      new-array v0, v0, [Ljava/lang/Object;
***************
*** 236,243 ****
--- 256,265 ----
  
      if-nez v0, :cond_0
  
+     .line 836
      const/4 v0, -0x1
  
+     .line 839
      :goto_0
      return v0
  
***************
*** 259,264 ****
--- 281,287 ----
      .parameter "params"
  
      .prologue
+     .line 773
      const/4 v1, 0x3
  
      new-array v1, v1, [Ljava/lang/Object;
***************
*** 281,291 ****
--- 304,317 ----
  
      if-nez v1, :cond_0
  
+     .line 774
      const/4 v1, -0x1
  
+     .line 778
      :goto_0
      return v1
  
+     .line 777
      :cond_0
      new-instance v0, Landroid/speech/tts/TextToSpeechService$AudioSpeechItem;
  
***************
*** 297,302 ****
--- 323,329 ----
  
      invoke-direct {v0, v1, v2, p4, p2}, Landroid/speech/tts/TextToSpeechService$AudioSpeechItem;-><init>(Landroid/speech/tts/TextToSpeechService;Ljava/lang/String;Landroid/os/Bundle;Landroid/net/Uri;)V
  
+     .line 778
      .local v0, item:Landroid/speech/tts/TextToSpeechService$SpeechItem;
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$1;->this$0:Landroid/speech/tts/TextToSpeechService;
  
***************
*** 320,325 ****
--- 347,353 ----
      .parameter "params"
  
      .prologue
+     .line 782
      const/4 v1, 0x2
  
      new-array v1, v1, [Ljava/lang/Object;
***************
*** 338,348 ****
--- 366,379 ----
  
      if-nez v1, :cond_0
  
+     .line 783
      const/4 v1, -0x1
  
+     .line 787
      :goto_0
      return v1
  
+     .line 786
      :cond_0
      new-instance v0, Landroid/speech/tts/TextToSpeechService$SilenceSpeechItem;
  
***************
*** 358,363 ****
--- 389,395 ----
  
      invoke-direct/range {v0 .. v5}, Landroid/speech/tts/TextToSpeechService$SilenceSpeechItem;-><init>(Landroid/speech/tts/TextToSpeechService;Ljava/lang/String;Landroid/os/Bundle;J)V
  
+     .line 787
      .local v0, item:Landroid/speech/tts/TextToSpeechService$SpeechItem;
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$1;->this$0:Landroid/speech/tts/TextToSpeechService;
  
***************
*** 379,384 ****
--- 411,417 ----
      .parameter "cb"
  
      .prologue
+     .line 844
      const/4 v0, 0x1
  
      new-array v0, v0, [Ljava/lang/Object;
***************
*** 393,401 ****
--- 426,436 ----
  
      if-nez v0, :cond_0
  
+     .line 849
      :goto_0
      return-void
  
+     .line 848
      :cond_0
      iget-object v0, p0, Landroid/speech/tts/TextToSpeechService$1;->this$0:Landroid/speech/tts/TextToSpeechService;
  
***************
*** 417,422 ****
--- 452,458 ----
      .parameter "params"
  
      .prologue
+     .line 752
      const/4 v1, 0x3
  
      new-array v1, v1, [Ljava/lang/Object;
***************
*** 439,449 ****
--- 475,488 ----
  
      if-nez v1, :cond_0
  
+     .line 753
      const/4 v1, -0x1
  
+     .line 757
      :goto_0
      return v1
  
+     .line 756
      :cond_0
      new-instance v0, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;
  
***************
*** 455,460 ****
--- 494,500 ----
  
      invoke-direct {v0, v1, v2, p4, p2}, Landroid/speech/tts/TextToSpeechService$SynthesisSpeechItem;-><init>(Landroid/speech/tts/TextToSpeechService;Ljava/lang/String;Landroid/os/Bundle;Ljava/lang/String;)V
  
+     .line 757
      .local v0, item:Landroid/speech/tts/TextToSpeechService$SpeechItem;
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$1;->this$0:Landroid/speech/tts/TextToSpeechService;
  
***************
*** 475,480 ****
--- 515,521 ----
      .parameter "callingApp"
  
      .prologue
+     .line 795
      const/4 v0, 0x1
  
      new-array v0, v0, [Ljava/lang/Object;
***************
*** 489,496 ****
--- 530,539 ----
  
      if-nez v0, :cond_0
  
+     .line 796
      const/4 v0, -0x1
  
+     .line 799
      :goto_0
      return v0
  
***************
*** 523,528 ****
--- 566,572 ----
      .prologue
      const/4 v6, 0x1
  
+     .line 762
      const/4 v1, 0x4
  
      new-array v1, v1, [Ljava/lang/Object;
***************
*** 547,562 ****
--- 591,610 ----
  
      if-nez v1, :cond_0
  
+     .line 763
      const/4 v1, -0x1
  
+     .line 769
      :goto_0
      return v1
  
+     .line 766
      :cond_0
      new-instance v5, Ljava/io/File;
  
      invoke-direct {v5, p3}, Ljava/io/File;-><init>(Ljava/lang/String;)V
  
+     .line 767
      .local v5, file:Ljava/io/File;
      new-instance v0, Landroid/speech/tts/TextToSpeechService$SynthesisToFileSpeechItem;
  
***************
*** 572,577 ****
--- 620,626 ----
  
      invoke-direct/range {v0 .. v5}, Landroid/speech/tts/TextToSpeechService$SynthesisToFileSpeechItem;-><init>(Landroid/speech/tts/TextToSpeechService;Ljava/lang/String;Landroid/os/Bundle;Ljava/lang/String;Ljava/io/File;)V
  
+     .line 769
      .local v0, item:Landroid/speech/tts/TextToSpeechService$SpeechItem;
      iget-object v1, p0, Landroid/speech/tts/TextToSpeechService$1;->this$0:Landroid/speech/tts/TextToSpeechService;
  
