*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeechService$AudioSpeechItem.smali	2012-03-23 19:36:32.839279808 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeechService$AudioSpeechItem.smali.original	2012-03-23 19:34:45.184316722 -0500
***************
*** 31,40 ****
--- 31,43 ----
      .parameter "uri"
  
      .prologue
+     .line 685
      iput-object p1, p0, Landroid/speech/tts/TextToSpeechService$AudioSpeechItem;->this$0:Landroid/speech/tts/TextToSpeechService;
  
+     .line 686
      invoke-direct {p0, p1, p2, p3}, Landroid/speech/tts/TextToSpeechService$SpeechItem;-><init>(Landroid/speech/tts/TextToSpeechService;Ljava/lang/String;Landroid/os/Bundle;)V
  
+     .line 687
      new-instance v0, Landroid/speech/tts/BlockingMediaPlayer;
  
      invoke-virtual {p0}, Landroid/speech/tts/TextToSpeechService$AudioSpeechItem;->getStreamType()I
***************
*** 45,50 ****
--- 48,54 ----
  
      iput-object v0, p0, Landroid/speech/tts/TextToSpeechService$AudioSpeechItem;->mPlayer:Landroid/speech/tts/BlockingMediaPlayer;
  
+     .line 688
      return-void
  .end method
  
***************
*** 54,59 ****
--- 58,64 ----
      .locals 1
  
      .prologue
+     .line 692
      const/4 v0, 0x1
  
      return v0
***************
*** 63,68 ****
--- 68,74 ----
      .locals 3
  
      .prologue
+     .line 697
      new-instance v0, Landroid/speech/tts/AudioMessageParams;
  
      invoke-virtual {p0}, Landroid/speech/tts/TextToSpeechService$AudioSpeechItem;->getCallingApp()Ljava/lang/String;
***************
*** 75,80 ****
--- 81,87 ----
  
      iput-object v0, p0, Landroid/speech/tts/TextToSpeechService$AudioSpeechItem;->mToken:Landroid/speech/tts/AudioMessageParams;
  
+     .line 698
      iget-object v0, p0, Landroid/speech/tts/TextToSpeechService$AudioSpeechItem;->this$0:Landroid/speech/tts/TextToSpeechService;
  
      #getter for: Landroid/speech/tts/TextToSpeechService;->mAudioPlaybackHandler:Landroid/speech/tts/AudioPlaybackHandler;
***************
*** 86,91 ****
--- 93,99 ----
  
      invoke-virtual {v0, v1}, Landroid/speech/tts/AudioPlaybackHandler;->enqueueAudio(Landroid/speech/tts/AudioMessageParams;)V
  
+     .line 699
      const/4 v0, 0x0
  
      return v0
***************
*** 95,99 ****
--- 103,108 ----
      .locals 0
  
      .prologue
+     .line 705
      return-void
  .end method
