*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeech$5.smali	2012-03-23 19:36:33.635294322 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeech$5.smali.original	2012-03-23 19:34:45.193316886 -0500
***************
*** 42,47 ****
--- 42,48 ----
      .parameter
  
      .prologue
+     .line 887
      iput-object p1, p0, Landroid/speech/tts/TextToSpeech$5;->this$0:Landroid/speech/tts/TextToSpeech;
  
      iput-object p2, p0, Landroid/speech/tts/TextToSpeech$5;->val$locale:Ljava/util/Locale;
***************
*** 63,68 ****
--- 64,70 ----
      .end annotation
  
      .prologue
+     .line 887
      invoke-virtual {p0, p1}, Landroid/speech/tts/TextToSpeech$5;->run(Landroid/speech/tts/ITextToSpeechService;)Ljava/util/Set;
  
      move-result-object v0
***************
*** 92,97 ****
--- 94,100 ----
      .end annotation
  
      .prologue
+     .line 890
      iget-object v2, p0, Landroid/speech/tts/TextToSpeech$5;->val$locale:Ljava/util/Locale;
  
      invoke-virtual {v2}, Ljava/util/Locale;->getISO3Language()Ljava/lang/String;
***************
*** 114,129 ****
--- 117,136 ----
  
      move-result-object v1
  
+     .line 892
      .local v1, features:[Ljava/lang/String;
      if-eqz v1, :cond_0
  
+     .line 893
      new-instance v0, Ljava/util/HashSet;
  
      invoke-direct {v0}, Ljava/util/HashSet;-><init>()V
  
+     .line 894
      .local v0, featureSet:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
      invoke-static {v0, v1}, Ljava/util/Collections;->addAll(Ljava/util/Collection;[Ljava/lang/Object;)Z
  
+     .line 897
      .end local v0           #featureSet:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
      :goto_0
      return-object v0
