*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeech$11.smali	2012-03-23 19:36:32.884280628 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/tts/TextToSpeech$11.smali.original	2012-03-23 19:34:45.184316722 -0500
***************
*** 46,51 ****
--- 46,52 ----
      .parameter
  
      .prologue
+     .line 1091
      iput-object p1, p0, Landroid/speech/tts/TextToSpeech$11;->this$0:Landroid/speech/tts/TextToSpeech;
  
      iput-object p2, p0, Landroid/speech/tts/TextToSpeech$11;->val$text:Ljava/lang/String;
***************
*** 71,76 ****
--- 72,78 ----
      .end annotation
  
      .prologue
+     .line 1094
      iget-object v0, p0, Landroid/speech/tts/TextToSpeech$11;->this$0:Landroid/speech/tts/TextToSpeech;
  
      #calls: Landroid/speech/tts/TextToSpeech;->getPackageName()Ljava/lang/String;
***************
*** 112,117 ****
--- 114,120 ----
      .end annotation
  
      .prologue
+     .line 1091
      invoke-virtual {p0, p1}, Landroid/speech/tts/TextToSpeech$11;->run(Landroid/speech/tts/ITextToSpeechService;)Ljava/lang/Integer;
  
      move-result-object v0
