*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/CountryDetector$ListenerTransport.smali	2012-03-23 19:36:17.862006697 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/CountryDetector$ListenerTransport.smali.original	2012-03-23 19:34:45.041314115 -0500
***************
*** 27,47 ****
--- 27,53 ----
      .parameter "looper"
  
      .prologue
+     .line 65
      invoke-direct {p0}, Landroid/location/ICountryListener$Stub;-><init>()V
  
+     .line 66
      iput-object p1, p0, Landroid/location/CountryDetector$ListenerTransport;->mListener:Landroid/location/CountryListener;
  
+     .line 67
      if-eqz p2, :cond_0
  
+     .line 68
      new-instance v0, Landroid/os/Handler;
  
      invoke-direct {v0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
      iput-object v0, p0, Landroid/location/CountryDetector$ListenerTransport;->mHandler:Landroid/os/Handler;
  
+     .line 72
      :goto_0
      return-void
  
+     .line 70
      :cond_0
      new-instance v0, Landroid/os/Handler;
  
***************
*** 57,62 ****
--- 63,69 ----
      .parameter "x0"
  
      .prologue
+     .line 59
      iget-object v0, p0, Landroid/location/CountryDetector$ListenerTransport;->mListener:Landroid/location/CountryListener;
  
      return-object v0
***************
*** 69,74 ****
--- 76,82 ----
      .parameter "country"
  
      .prologue
+     .line 75
      iget-object v0, p0, Landroid/location/CountryDetector$ListenerTransport;->mHandler:Landroid/os/Handler;
  
      new-instance v1, Landroid/location/CountryDetector$ListenerTransport$1;
***************
*** 77,81 ****
--- 85,90 ----
  
      invoke-virtual {v0, v1}, Landroid/os/Handler;->post(Ljava/lang/Runnable;)Z
  
+     .line 80
      return-void
  .end method
