*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/CountryDetector$ListenerTransport$1.smali	2012-03-23 19:36:17.270995920 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/CountryDetector$ListenerTransport$1.smali.original	2012-03-23 19:34:45.038314059 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 75
      iput-object p1, p0, Landroid/location/CountryDetector$ListenerTransport$1;->this$0:Landroid/location/CountryDetector$ListenerTransport;
  
      iput-object p2, p0, Landroid/location/CountryDetector$ListenerTransport$1;->val$country:Landroid/location/Country;
***************
*** 45,50 ****
--- 46,52 ----
      .locals 2
  
      .prologue
+     .line 77
      iget-object v0, p0, Landroid/location/CountryDetector$ListenerTransport$1;->this$0:Landroid/location/CountryDetector$ListenerTransport;
  
      #getter for: Landroid/location/CountryDetector$ListenerTransport;->mListener:Landroid/location/CountryListener;
***************
*** 56,60 ****
--- 58,63 ----
  
      invoke-interface {v0, v1}, Landroid/location/CountryListener;->onCountryDetected(Landroid/location/Country;)V
  
+     .line 78
      return-void
  .end method
