*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/Country$1.smali	2012-03-23 19:36:17.966008595 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/Country$1.smali.original	2012-03-23 19:34:45.042314133 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 109
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "in"
  
      .prologue
+     .line 111
      new-instance v0, Landroid/location/Country;
  
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
***************
*** 64,69 ****
--- 66,72 ----
      .parameter "x0"
  
      .prologue
+     .line 109
      invoke-virtual {p0, p1}, Landroid/location/Country$1;->createFromParcel(Landroid/os/Parcel;)Landroid/location/Country;
  
      move-result-object v0
***************
*** 76,81 ****
--- 79,85 ----
      .parameter "size"
  
      .prologue
+     .line 115
      new-array v0, p1, [Landroid/location/Country;
  
      return-object v0
***************
*** 86,91 ****
--- 90,96 ----
      .parameter "x0"
  
      .prologue
+     .line 109
      invoke-virtual {p0, p1}, Landroid/location/Country$1;->newArray(I)[Landroid/location/Country;
  
      move-result-object v0
