*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/ICountryDetector$Stub.smali	2012-03-23 19:36:17.889007190 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/ICountryDetector$Stub.smali.original	2012-03-23 19:34:45.042314133 -0500
***************
*** 38,49 ****
--- 38,52 ----
      .locals 1
  
      .prologue
+     .line 19
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 20
      const-string v0, "android.location.ICountryDetector"
  
      invoke-virtual {p0, p0, v0}, Landroid/location/ICountryDetector$Stub;->attachInterface(Landroid/os/IInterface;Ljava/lang/String;)V
  
+     .line 21
      return-void
  .end method
  
***************
*** 52,64 ****
--- 55,71 ----
      .parameter "obj"
  
      .prologue
+     .line 28
      if-nez p0, :cond_0
  
+     .line 29
      const/4 v0, 0x0
  
+     .line 35
      :goto_0
      return-object v0
  
+     .line 31
      :cond_0
      const-string v1, "android.location.ICountryDetector"
  
***************
*** 66,71 ****
--- 73,79 ----
  
      move-result-object v0
  
+     .line 32
      .local v0, iin:Landroid/os/IInterface;
      if-eqz v0, :cond_1
  
***************
*** 73,82 ****
--- 81,92 ----
  
      if-eqz v1, :cond_1
  
+     .line 33
      check-cast v0, Landroid/location/ICountryDetector;
  
      goto :goto_0
  
+     .line 35
      :cond_1
      new-instance v0, Landroid/location/ICountryDetector$Stub$Proxy;
  
***************
*** 92,97 ****
--- 102,108 ----
      .locals 0
  
      .prologue
+     .line 39
      return-object p0
  .end method
  
***************
*** 110,117 ****
--- 121,130 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 43
      sparse-switch p1, :sswitch_data_0
  
+     .line 83
      invoke-super {p0, p1, p2, p3, p4}, Landroid/os/Binder;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
      move-result v2
***************
*** 119,124 ****
--- 132,138 ----
      :goto_0
      return v2
  
+     .line 47
      :sswitch_0
      const-string v3, "android.location.ICountryDetector"
  
***************
*** 126,151 ****
--- 140,172 ----
  
      goto :goto_0
  
+     .line 52
      :sswitch_1
      const-string v3, "android.location.ICountryDetector"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 53
      invoke-virtual {p0}, Landroid/location/ICountryDetector$Stub;->detectCountry()Landroid/location/Country;
  
      move-result-object v1
  
+     .line 54
      .local v1, _result:Landroid/location/Country;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 55
      if-eqz v1, :cond_0
  
+     .line 56
      invoke-virtual {p3, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 57
      invoke-virtual {v1, p3, v2}, Landroid/location/Country;->writeToParcel(Landroid/os/Parcel;I)V
  
      goto :goto_0
  
+     .line 60
      :cond_0
      const/4 v3, 0x0
  
***************
*** 153,164 ****
--- 174,187 ----
  
      goto :goto_0
  
+     .line 66
      .end local v1           #_result:Landroid/location/Country;
      :sswitch_2
      const-string v3, "android.location.ICountryDetector"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 68
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v3
***************
*** 167,185 ****
--- 190,212 ----
  
      move-result-object v0
  
+     .line 69
      .local v0, _arg0:Landroid/location/ICountryListener;
      invoke-virtual {p0, v0}, Landroid/location/ICountryDetector$Stub;->addCountryListener(Landroid/location/ICountryListener;)V
  
+     .line 70
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto :goto_0
  
+     .line 75
      .end local v0           #_arg0:Landroid/location/ICountryListener;
      :sswitch_3
      const-string v3, "android.location.ICountryDetector"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 77
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v3
***************
*** 188,200 ****
--- 215,230 ----
  
      move-result-object v0
  
+     .line 78
      .restart local v0       #_arg0:Landroid/location/ICountryListener;
      invoke-virtual {p0, v0}, Landroid/location/ICountryDetector$Stub;->removeCountryListener(Landroid/location/ICountryListener;)V
  
+     .line 79
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto :goto_0
  
+     .line 43
      nop
  
      :sswitch_data_0
