*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/ILocationManager$Stub.smali	2012-03-23 19:36:18.081010691 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/ILocationManager$Stub.smali.original	2012-03-23 19:34:45.043314151 -0500
***************
*** 92,103 ****
--- 92,106 ----
      .locals 1
  
      .prologue
+     .line 19
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 20
      const-string v0, "android.location.ILocationManager"
  
      invoke-virtual {p0, p0, v0}, Landroid/location/ILocationManager$Stub;->attachInterface(Landroid/os/IInterface;Ljava/lang/String;)V
  
+     .line 21
      return-void
  .end method
  
***************
*** 106,118 ****
--- 109,125 ----
      .parameter "obj"
  
      .prologue
+     .line 28
      if-nez p0, :cond_0
  
+     .line 29
      const/4 v0, 0x0
  
+     .line 35
      :goto_0
      return-object v0
  
+     .line 31
      :cond_0
      const-string v1, "android.location.ILocationManager"
  
***************
*** 120,125 ****
--- 127,133 ----
  
      move-result-object v0
  
+     .line 32
      .local v0, iin:Landroid/os/IInterface;
      if-eqz v0, :cond_1
  
***************
*** 127,136 ****
--- 135,146 ----
  
      if-eqz v1, :cond_1
  
+     .line 33
      check-cast v0, Landroid/location/ILocationManager;
  
      goto :goto_0
  
+     .line 35
      :cond_1
      new-instance v0, Landroid/location/ILocationManager$Stub$Proxy;
  
***************
*** 146,151 ****
--- 156,162 ----
      .locals 0
  
      .prologue
+     .line 39
      return-object p0
  .end method
  
***************
*** 162,169 ****
--- 173,182 ----
      .end annotation
  
      .prologue
+     .line 43
      sparse-switch p1, :sswitch_data_0
  
+     .line 518
      invoke-super/range {p0 .. p4}, Landroid/os/Binder;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
      move-result v2
***************
*** 171,176 ****
--- 184,190 ----
      :goto_0
      return v2
  
+     .line 47
      :sswitch_0
      const-string v2, "android.location.ILocationManager"
  
***************
*** 178,187 ****
--- 192,203 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 48
      const/4 v2, 0x1
  
      goto :goto_0
  
+     .line 52
      :sswitch_1
      const-string v2, "android.location.ILocationManager"
  
***************
*** 189,211 ****
--- 205,232 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 53
      invoke-virtual/range {p0 .. p0}, Landroid/location/ILocationManager$Stub;->getAllProviders()Ljava/util/List;
  
      move-result-object v51
  
+     .line 54
      .local v51, _result:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 55
      move-object/from16 v0, p3
  
      move-object/from16 v1, v51
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeStringList(Ljava/util/List;)V
  
+     .line 56
      const/4 v2, 0x1
  
      goto :goto_0
  
+     .line 60
      .end local v51           #_result:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
      :sswitch_2
      const-string v2, "android.location.ILocationManager"
***************
*** 214,225 ****
--- 235,248 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 62
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      if-eqz v2, :cond_0
  
+     .line 63
      sget-object v2, Landroid/location/Criteria;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 230,235 ****
--- 253,259 ----
  
      check-cast v3, Landroid/location/Criteria;
  
+     .line 69
      .local v3, _arg0:Landroid/location/Criteria;
      :goto_1
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 240,245 ****
--- 264,270 ----
  
      const/4 v4, 0x1
  
+     .line 70
      .local v4, _arg1:Z
      :goto_2
      move-object/from16 v0, p0
***************
*** 248,266 ****
--- 273,295 ----
  
      move-result-object v51
  
+     .line 71
      .restart local v51       #_result:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 72
      move-object/from16 v0, p3
  
      move-object/from16 v1, v51
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeStringList(Ljava/util/List;)V
  
+     .line 73
      const/4 v2, 0x1
  
      goto :goto_0
  
+     .line 66
      .end local v3           #_arg0:Landroid/location/Criteria;
      .end local v4           #_arg1:Z
      .end local v51           #_result:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
***************
*** 270,280 ****
--- 299,311 ----
      .restart local v3       #_arg0:Landroid/location/Criteria;
      goto :goto_1
  
+     .line 69
      :cond_1
      const/4 v4, 0x0
  
      goto :goto_2
  
+     .line 77
      .end local v3           #_arg0:Landroid/location/Criteria;
      :sswitch_3
      const-string v2, "android.location.ILocationManager"
***************
*** 283,294 ****
--- 314,327 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 79
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      if-eqz v2, :cond_2
  
+     .line 80
      sget-object v2, Landroid/location/Criteria;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 299,304 ****
--- 332,338 ----
  
      check-cast v3, Landroid/location/Criteria;
  
+     .line 86
      .restart local v3       #_arg0:Landroid/location/Criteria;
      :goto_3
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 309,314 ****
--- 343,349 ----
  
      const/4 v4, 0x1
  
+     .line 87
      .restart local v4       #_arg1:Z
      :goto_4
      move-object/from16 v0, p0
***************
*** 317,335 ****
--- 352,374 ----
  
      move-result-object v50
  
+     .line 88
      .local v50, _result:Ljava/lang/String;
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 89
      move-object/from16 v0, p3
  
      move-object/from16 v1, v50
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 90
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 83
      .end local v3           #_arg0:Landroid/location/Criteria;
      .end local v4           #_arg1:Z
      .end local v50           #_result:Ljava/lang/String;
***************
*** 339,349 ****
--- 378,390 ----
      .restart local v3       #_arg0:Landroid/location/Criteria;
      goto :goto_3
  
+     .line 86
      :cond_3
      const/4 v4, 0x0
  
      goto :goto_4
  
+     .line 94
      .end local v3           #_arg0:Landroid/location/Criteria;
      :sswitch_4
      const-string v2, "android.location.ILocationManager"
***************
*** 352,361 ****
--- 393,404 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 96
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 98
      .local v3, _arg0:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 363,368 ****
--- 406,412 ----
  
      if-eqz v2, :cond_4
  
+     .line 99
      sget-object v2, Landroid/location/Criteria;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 373,378 ****
--- 417,423 ----
  
      check-cast v4, Landroid/location/Criteria;
  
+     .line 104
      .local v4, _arg1:Landroid/location/Criteria;
      :goto_5
      move-object/from16 v0, p0
***************
*** 381,389 ****
--- 426,436 ----
  
      move-result v50
  
+     .line 105
      .local v50, _result:Z
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 106
      if-eqz v50, :cond_5
  
      const/4 v2, 0x1
***************
*** 393,402 ****
--- 440,451 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 107
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 102
      .end local v4           #_arg1:Landroid/location/Criteria;
      .end local v50           #_result:Z
      :cond_4
***************
*** 405,416 ****
--- 454,467 ----
      .restart local v4       #_arg1:Landroid/location/Criteria;
      goto :goto_5
  
+     .line 106
      .restart local v50       #_result:Z
      :cond_5
      const/4 v2, 0x0
  
      goto :goto_6
  
+     .line 111
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v4           #_arg1:Landroid/location/Criteria;
      .end local v50           #_result:Z
***************
*** 421,430 ****
--- 472,483 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 113
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 115
      .restart local v3       #_arg0:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 432,437 ****
--- 485,491 ----
  
      if-eqz v2, :cond_6
  
+     .line 116
      sget-object v2, Landroid/location/Criteria;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 442,458 ****
--- 496,515 ----
  
      check-cast v4, Landroid/location/Criteria;
  
+     .line 122
      .restart local v4       #_arg1:Landroid/location/Criteria;
      :goto_7
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v5
  
+     .line 124
      .local v5, _arg2:J
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readFloat()F
  
      move-result v7
  
+     .line 126
      .local v7, _arg3:F
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 462,467 ****
--- 519,525 ----
  
      const/4 v8, 0x1
  
+     .line 128
      .local v8, _arg4:Z
      :goto_8
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
***************
*** 475,488 ****
--- 533,550 ----
      .local v9, _arg5:Landroid/location/ILocationListener;
      move-object/from16 v2, p0
  
+     .line 129
      invoke-virtual/range {v2 .. v9}, Landroid/location/ILocationManager$Stub;->requestLocationUpdates(Ljava/lang/String;Landroid/location/Criteria;JFZLandroid/location/ILocationListener;)V
  
+     .line 130
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 131
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 119
      .end local v4           #_arg1:Landroid/location/Criteria;
      .end local v5           #_arg2:J
      .end local v7           #_arg3:F
***************
*** 494,499 ****
--- 556,562 ----
      .restart local v4       #_arg1:Landroid/location/Criteria;
      goto :goto_7
  
+     .line 126
      .restart local v5       #_arg2:J
      .restart local v7       #_arg3:F
      :cond_7
***************
*** 501,506 ****
--- 564,570 ----
  
      goto :goto_8
  
+     .line 135
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v4           #_arg1:Landroid/location/Criteria;
      .end local v5           #_arg2:J
***************
*** 512,521 ****
--- 576,587 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 137
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 139
      .restart local v3       #_arg0:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 523,528 ****
--- 589,595 ----
  
      if-eqz v2, :cond_8
  
+     .line 140
      sget-object v2, Landroid/location/Criteria;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 533,549 ****
--- 600,619 ----
  
      check-cast v4, Landroid/location/Criteria;
  
+     .line 146
      .restart local v4       #_arg1:Landroid/location/Criteria;
      :goto_9
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v5
  
+     .line 148
      .restart local v5       #_arg2:J
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readFloat()F
  
      move-result v7
  
+     .line 150
      .restart local v7       #_arg3:F
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 553,558 ****
--- 623,629 ----
  
      const/4 v8, 0x1
  
+     .line 152
      .restart local v8       #_arg4:Z
      :goto_a
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 561,566 ****
--- 632,638 ----
  
      if-eqz v2, :cond_a
  
+     .line 153
      sget-object v2, Landroid/app/PendingIntent;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 575,588 ****
--- 647,664 ----
      :goto_b
      move-object/from16 v2, p0
  
+     .line 158
      invoke-virtual/range {v2 .. v9}, Landroid/location/ILocationManager$Stub;->requestLocationUpdatesPI(Ljava/lang/String;Landroid/location/Criteria;JFZLandroid/app/PendingIntent;)V
  
+     .line 159
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 160
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 143
      .end local v4           #_arg1:Landroid/location/Criteria;
      .end local v5           #_arg2:J
      .end local v7           #_arg3:F
***************
*** 594,599 ****
--- 670,676 ----
      .restart local v4       #_arg1:Landroid/location/Criteria;
      goto :goto_9
  
+     .line 150
      .restart local v5       #_arg2:J
      .restart local v7       #_arg3:F
      :cond_9
***************
*** 601,606 ****
--- 678,684 ----
  
      goto :goto_a
  
+     .line 156
      .restart local v8       #_arg4:Z
      :cond_a
      const/4 v9, 0x0
***************
*** 608,613 ****
--- 686,692 ----
      .restart local v9       #_arg5:Landroid/app/PendingIntent;
      goto :goto_b
  
+     .line 164
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v4           #_arg1:Landroid/location/Criteria;
      .end local v5           #_arg2:J
***************
*** 621,626 ****
--- 700,706 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 166
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v2
***************
*** 629,645 ****
--- 709,729 ----
  
      move-result-object v3
  
+     .line 167
      .local v3, _arg0:Landroid/location/ILocationListener;
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3}, Landroid/location/ILocationManager$Stub;->removeUpdates(Landroid/location/ILocationListener;)V
  
+     .line 168
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 169
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 173
      .end local v3           #_arg0:Landroid/location/ILocationListener;
      :sswitch_8
      const-string v2, "android.location.ILocationManager"
***************
*** 648,659 ****
--- 732,745 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 175
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      if-eqz v2, :cond_b
  
+     .line 176
      sget-object v2, Landroid/app/PendingIntent;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 664,681 ****
--- 750,771 ----
  
      check-cast v3, Landroid/app/PendingIntent;
  
+     .line 181
      .local v3, _arg0:Landroid/app/PendingIntent;
      :goto_c
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3}, Landroid/location/ILocationManager$Stub;->removeUpdatesPI(Landroid/app/PendingIntent;)V
  
+     .line 182
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 183
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 179
      .end local v3           #_arg0:Landroid/app/PendingIntent;
      :cond_b
      const/4 v3, 0x0
***************
*** 683,688 ****
--- 773,779 ----
      .restart local v3       #_arg0:Landroid/app/PendingIntent;
      goto :goto_c
  
+     .line 187
      .end local v3           #_arg0:Landroid/app/PendingIntent;
      :sswitch_9
      const-string v2, "android.location.ILocationManager"
***************
*** 691,696 ****
--- 782,788 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 189
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v2
***************
*** 699,704 ****
--- 791,797 ----
  
      move-result-object v3
  
+     .line 190
      .local v3, _arg0:Landroid/location/IGpsStatusListener;
      move-object/from16 v0, p0
  
***************
*** 706,714 ****
--- 799,809 ----
  
      move-result v50
  
+     .line 191
      .restart local v50       #_result:Z
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 192
      if-eqz v50, :cond_c
  
      const/4 v2, 0x1
***************
*** 718,732 ****
--- 813,830 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 193
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 192
      :cond_c
      const/4 v2, 0x0
  
      goto :goto_d
  
+     .line 197
      .end local v3           #_arg0:Landroid/location/IGpsStatusListener;
      .end local v50           #_result:Z
      :sswitch_a
***************
*** 736,741 ****
--- 834,840 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 199
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v2
***************
*** 744,760 ****
--- 843,863 ----
  
      move-result-object v3
  
+     .line 200
      .restart local v3       #_arg0:Landroid/location/IGpsStatusListener;
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3}, Landroid/location/ILocationManager$Stub;->removeGpsStatusListener(Landroid/location/IGpsStatusListener;)V
  
+     .line 201
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 202
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 206
      .end local v3           #_arg0:Landroid/location/IGpsStatusListener;
      :sswitch_b
      const-string v2, "android.location.ILocationManager"
***************
*** 763,768 ****
--- 866,872 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 208
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v2
***************
*** 771,787 ****
--- 875,895 ----
  
      move-result-object v3
  
+     .line 209
      .local v3, _arg0:Landroid/location/ILocationListener;
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3}, Landroid/location/ILocationManager$Stub;->locationCallbackFinished(Landroid/location/ILocationListener;)V
  
+     .line 210
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 211
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 215
      .end local v3           #_arg0:Landroid/location/ILocationListener;
      :sswitch_c
      const-string v2, "android.location.ILocationManager"
***************
*** 790,804 ****
--- 898,915 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 217
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 219
      .local v3, _arg0:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v4
  
+     .line 221
      .local v4, _arg1:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 806,811 ****
--- 917,923 ----
  
      if-eqz v2, :cond_d
  
+     .line 222
      sget-object v2, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 816,821 ****
--- 928,934 ----
  
      check-cast v5, Landroid/os/Bundle;
  
+     .line 227
      .local v5, _arg2:Landroid/os/Bundle;
      :goto_e
      move-object/from16 v0, p0
***************
*** 824,832 ****
--- 937,947 ----
  
      move-result v50
  
+     .line 228
      .restart local v50       #_result:Z
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 229
      if-eqz v50, :cond_e
  
      const/4 v2, 0x1
***************
*** 836,860 ****
--- 951,980 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 230
      if-eqz v5, :cond_f
  
+     .line 231
      const/4 v2, 0x1
  
      move-object/from16 v0, p3
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 232
      const/4 v2, 0x1
  
      move-object/from16 v0, p3
  
      invoke-virtual {v5, v0, v2}, Landroid/os/Bundle;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 237
      :goto_10
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 225
      .end local v5           #_arg2:Landroid/os/Bundle;
      .end local v50           #_result:Z
      :cond_d
***************
*** 863,874 ****
--- 983,996 ----
      .restart local v5       #_arg2:Landroid/os/Bundle;
      goto :goto_e
  
+     .line 229
      .restart local v50       #_result:Z
      :cond_e
      const/4 v2, 0x0
  
      goto :goto_f
  
+     .line 235
      :cond_f
      const/4 v2, 0x0
  
***************
*** 878,883 ****
--- 1000,1006 ----
  
      goto :goto_10
  
+     .line 241
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v4           #_arg1:Ljava/lang/String;
      .end local v5           #_arg2:Landroid/os/Bundle;
***************
*** 889,913 ****
--- 1012,1041 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 243
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readDouble()D
  
      move-result-wide v11
  
+     .line 245
      .local v11, _arg0:D
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readDouble()D
  
      move-result-wide v13
  
+     .line 247
      .local v13, _arg1:D
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readFloat()F
  
      move-result v5
  
+     .line 249
      .local v5, _arg2:F
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v16
  
+     .line 251
      .local v16, _arg3:J
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 915,920 ****
--- 1043,1049 ----
  
      if-eqz v2, :cond_10
  
+     .line 252
      sget-object v2, Landroid/app/PendingIntent;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 933,946 ****
--- 1062,1079 ----
  
      move-object/from16 v18, v8
  
+     .line 257
      invoke-virtual/range {v10 .. v18}, Landroid/location/ILocationManager$Stub;->addProximityAlert(DDFJLandroid/app/PendingIntent;)V
  
+     .line 258
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 259
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 255
      .end local v8           #_arg4:Landroid/app/PendingIntent;
      :cond_10
      const/4 v8, 0x0
***************
*** 948,953 ****
--- 1081,1087 ----
      .restart local v8       #_arg4:Landroid/app/PendingIntent;
      goto :goto_11
  
+     .line 263
      .end local v5           #_arg2:F
      .end local v8           #_arg4:Landroid/app/PendingIntent;
      .end local v11           #_arg0:D
***************
*** 960,971 ****
--- 1094,1107 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 265
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      if-eqz v2, :cond_11
  
+     .line 266
      sget-object v2, Landroid/app/PendingIntent;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 976,993 ****
--- 1112,1133 ----
  
      check-cast v3, Landroid/app/PendingIntent;
  
+     .line 271
      .local v3, _arg0:Landroid/app/PendingIntent;
      :goto_12
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3}, Landroid/location/ILocationManager$Stub;->removeProximityAlert(Landroid/app/PendingIntent;)V
  
+     .line 272
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 273
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 269
      .end local v3           #_arg0:Landroid/app/PendingIntent;
      :cond_11
      const/4 v3, 0x0
***************
*** 995,1000 ****
--- 1135,1141 ----
      .restart local v3       #_arg0:Landroid/app/PendingIntent;
      goto :goto_12
  
+     .line 277
      .end local v3           #_arg0:Landroid/app/PendingIntent;
      :sswitch_f
      const-string v2, "android.location.ILocationManager"
***************
*** 1003,1012 ****
--- 1144,1155 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 279
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 280
      .local v3, _arg0:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 1014,1030 ****
--- 1157,1177 ----
  
      move-result-object v50
  
+     .line 281
      .local v50, _result:Landroid/os/Bundle;
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 282
      if-eqz v50, :cond_12
  
+     .line 283
      const/4 v2, 0x1
  
      move-object/from16 v0, p3
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 284
      const/4 v2, 0x1
  
      move-object/from16 v0, v50
***************
*** 1033,1043 ****
--- 1180,1192 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/os/Bundle;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 289
      :goto_13
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 287
      :cond_12
      const/4 v2, 0x0
  
***************
*** 1047,1052 ****
--- 1196,1202 ----
  
      goto :goto_13
  
+     .line 293
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v50           #_result:Landroid/os/Bundle;
      :sswitch_10
***************
*** 1056,1065 ****
--- 1206,1217 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 295
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 296
      .restart local v3       #_arg0:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 1067,1075 ****
--- 1219,1229 ----
  
      move-result v50
  
+     .line 297
      .local v50, _result:Z
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 298
      if-eqz v50, :cond_13
  
      const/4 v2, 0x1
***************
*** 1079,1093 ****
--- 1233,1250 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 299
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 298
      :cond_13
      const/4 v2, 0x0
  
      goto :goto_14
  
+     .line 303
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v50           #_result:Z
      :sswitch_11
***************
*** 1097,1106 ****
--- 1254,1265 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 305
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 306
      .restart local v3       #_arg0:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 1108,1124 ****
--- 1267,1287 ----
  
      move-result-object v50
  
+     .line 307
      .local v50, _result:Landroid/location/Location;
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 308
      if-eqz v50, :cond_14
  
+     .line 309
      const/4 v2, 0x1
  
      move-object/from16 v0, p3
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 310
      const/4 v2, 0x1
  
      move-object/from16 v0, v50
***************
*** 1127,1137 ****
--- 1290,1302 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/location/Location;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 315
      :goto_15
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 313
      :cond_14
      const/4 v2, 0x0
  
***************
*** 1141,1146 ****
--- 1306,1312 ----
  
      goto :goto_15
  
+     .line 319
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v50           #_result:Landroid/location/Location;
      :sswitch_12
***************
*** 1150,1161 ****
--- 1316,1329 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 321
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      if-eqz v2, :cond_15
  
+     .line 322
      sget-object v2, Landroid/location/Location;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 1166,1171 ****
--- 1334,1340 ----
  
      check-cast v3, Landroid/location/Location;
  
+     .line 328
      .local v3, _arg0:Landroid/location/Location;
      :goto_16
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 1176,1193 ****
--- 1345,1366 ----
  
      const/4 v4, 0x1
  
+     .line 329
      .local v4, _arg1:Z
      :goto_17
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3, v4}, Landroid/location/ILocationManager$Stub;->reportLocation(Landroid/location/Location;Z)V
  
+     .line 330
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 331
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 325
      .end local v3           #_arg0:Landroid/location/Location;
      .end local v4           #_arg1:Z
      :cond_15
***************
*** 1196,1206 ****
--- 1369,1381 ----
      .restart local v3       #_arg0:Landroid/location/Location;
      goto :goto_16
  
+     .line 328
      :cond_16
      const/4 v4, 0x0
  
      goto :goto_17
  
+     .line 335
      .end local v3           #_arg0:Landroid/location/Location;
      :sswitch_13
      const-string v2, "android.location.ILocationManager"
***************
*** 1209,1221 ****
--- 1384,1399 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 336
      invoke-virtual/range {p0 .. p0}, Landroid/location/ILocationManager$Stub;->geocoderIsPresent()Z
  
      move-result v50
  
+     .line 337
      .local v50, _result:Z
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 338
      if-eqz v50, :cond_17
  
      const/4 v2, 0x1
***************
*** 1225,1239 ****
--- 1403,1420 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 339
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 338
      :cond_17
      const/4 v2, 0x0
  
      goto :goto_18
  
+     .line 343
      .end local v50           #_result:Z
      :sswitch_14
      const-string v2, "android.location.ILocationManager"
***************
*** 1242,1261 ****
--- 1423,1446 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 345
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readDouble()D
  
      move-result-wide v11
  
+     .line 347
      .restart local v11       #_arg0:D
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readDouble()D
  
      move-result-wide v13
  
+     .line 349
      .restart local v13       #_arg1:D
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
  
+     .line 351
      .local v5, _arg2:I
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 1263,1268 ****
--- 1448,1454 ----
  
      if-eqz v2, :cond_18
  
+     .line 352
      sget-object v2, Landroid/location/GeocoderParams;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 1273,1278 ****
--- 1459,1465 ----
  
      check-cast v7, Landroid/location/GeocoderParams;
  
+     .line 358
      .local v7, _arg3:Landroid/location/GeocoderParams;
      :goto_19
      new-instance v25, Ljava/util/ArrayList;
***************
*** 1290,1318 ****
--- 1477,1511 ----
  
      move-object/from16 v24, v7
  
+     .line 359
      invoke-virtual/range {v18 .. v25}, Landroid/location/ILocationManager$Stub;->getFromLocation(DDILandroid/location/GeocoderParams;Ljava/util/List;)Ljava/lang/String;
  
      move-result-object v50
  
+     .line 360
      .local v50, _result:Ljava/lang/String;
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 361
      move-object/from16 v0, p3
  
      move-object/from16 v1, v50
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 362
      move-object/from16 v0, p3
  
      move-object/from16 v1, v25
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeTypedList(Ljava/util/List;)V
  
+     .line 363
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 355
      .end local v7           #_arg3:Landroid/location/GeocoderParams;
      .end local v25           #_arg4:Ljava/util/List;,"Ljava/util/List<Landroid/location/Address;>;"
      .end local v50           #_result:Ljava/lang/String;
***************
*** 1322,1327 ****
--- 1515,1521 ----
      .restart local v7       #_arg3:Landroid/location/GeocoderParams;
      goto :goto_19
  
+     .line 367
      .end local v5           #_arg2:I
      .end local v7           #_arg3:Landroid/location/GeocoderParams;
      .end local v11           #_arg0:D
***************
*** 1333,1367 ****
--- 1527,1568 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 369
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 371
      .local v3, _arg0:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readDouble()D
  
      move-result-wide v13
  
+     .line 373
      .restart local v13       #_arg1:D
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readDouble()D
  
      move-result-wide v5
  
+     .line 375
      .local v5, _arg2:D
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readDouble()D
  
      move-result-wide v16
  
+     .line 377
      .local v16, _arg3:D
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readDouble()D
  
      move-result-wide v34
  
+     .line 379
      .local v34, _arg4:D
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v9
  
+     .line 381
      .local v9, _arg5:I
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 1369,1374 ****
--- 1570,1576 ----
  
      if-eqz v2, :cond_19
  
+     .line 382
      sget-object v2, Landroid/location/GeocoderParams;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 1379,1384 ****
--- 1581,1587 ----
  
      check-cast v37, Landroid/location/GeocoderParams;
  
+     .line 388
      .local v37, _arg6:Landroid/location/GeocoderParams;
      :goto_1a
      new-instance v38, Ljava/util/ArrayList;
***************
*** 1398,1426 ****
--- 1601,1635 ----
  
      move/from16 v36, v9
  
+     .line 389
      invoke-virtual/range {v26 .. v38}, Landroid/location/ILocationManager$Stub;->getFromLocationName(Ljava/lang/String;DDDDILandroid/location/GeocoderParams;Ljava/util/List;)Ljava/lang/String;
  
      move-result-object v50
  
+     .line 390
      .restart local v50       #_result:Ljava/lang/String;
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 391
      move-object/from16 v0, p3
  
      move-object/from16 v1, v50
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 392
      move-object/from16 v0, p3
  
      move-object/from16 v1, v38
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeTypedList(Ljava/util/List;)V
  
+     .line 393
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 385
      .end local v37           #_arg6:Landroid/location/GeocoderParams;
      .end local v38           #_arg7:Ljava/util/List;,"Ljava/util/List<Landroid/location/Address;>;"
      .end local v50           #_result:Ljava/lang/String;
***************
*** 1430,1435 ****
--- 1639,1645 ----
      .restart local v37       #_arg6:Landroid/location/GeocoderParams;
      goto :goto_1a
  
+     .line 397
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v5           #_arg2:D
      .end local v9           #_arg5:I
***************
*** 1444,1453 ****
--- 1654,1665 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 399
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 401
      .restart local v3       #_arg0:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 1457,1462 ****
--- 1669,1675 ----
  
      const/4 v4, 0x1
  
+     .line 403
      .restart local v4       #_arg1:Z
      :goto_1b
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 1467,1472 ****
--- 1680,1686 ----
  
      const/4 v5, 0x1
  
+     .line 405
      .local v5, _arg2:Z
      :goto_1c
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 1477,1482 ****
--- 1691,1697 ----
  
      const/4 v7, 0x1
  
+     .line 407
      .local v7, _arg3:Z
      :goto_1d
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 1487,1492 ****
--- 1702,1708 ----
  
      const/4 v8, 0x1
  
+     .line 409
      .local v8, _arg4:Z
      :goto_1e
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 1497,1502 ****
--- 1713,1719 ----
  
      const/4 v9, 0x1
  
+     .line 411
      .local v9, _arg5:Z
      :goto_1f
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 1507,1512 ****
--- 1724,1730 ----
  
      const/16 v37, 0x1
  
+     .line 413
      .local v37, _arg6:Z
      :goto_20
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
***************
*** 1517,1528 ****
--- 1735,1748 ----
  
      const/16 v47, 0x1
  
+     .line 415
      .local v47, _arg7:Z
      :goto_21
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v48
  
+     .line 417
      .local v48, _arg8:I
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 1545,1558 ****
--- 1765,1782 ----
  
      move/from16 v46, v37
  
+     .line 418
      invoke-virtual/range {v39 .. v49}, Landroid/location/ILocationManager$Stub;->addTestProvider(Ljava/lang/String;ZZZZZZZII)V
  
+     .line 419
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 420
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 401
      .end local v4           #_arg1:Z
      .end local v5           #_arg2:Z
      .end local v7           #_arg3:Z
***************
*** 1567,1608 ****
--- 1791,1839 ----
  
      goto :goto_1b
  
+     .line 403
      .restart local v4       #_arg1:Z
      :cond_1b
      const/4 v5, 0x0
  
      goto :goto_1c
  
+     .line 405
      .restart local v5       #_arg2:Z
      :cond_1c
      const/4 v7, 0x0
  
      goto :goto_1d
  
+     .line 407
      .restart local v7       #_arg3:Z
      :cond_1d
      const/4 v8, 0x0
  
      goto :goto_1e
  
+     .line 409
      .restart local v8       #_arg4:Z
      :cond_1e
      const/4 v9, 0x0
  
      goto :goto_1f
  
+     .line 411
      .restart local v9       #_arg5:Z
      :cond_1f
      const/16 v37, 0x0
  
      goto :goto_20
  
+     .line 413
      .restart local v37       #_arg6:Z
      :cond_20
      const/16 v47, 0x0
  
      goto :goto_21
  
+     .line 424
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v4           #_arg1:Z
      .end local v5           #_arg2:Z
***************
*** 1617,1637 ****
--- 1848,1873 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 426
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 427
      .restart local v3       #_arg0:Ljava/lang/String;
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3}, Landroid/location/ILocationManager$Stub;->removeTestProvider(Ljava/lang/String;)V
  
+     .line 428
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 429
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 433
      .end local v3           #_arg0:Ljava/lang/String;
      :sswitch_18
      const-string v2, "android.location.ILocationManager"
***************
*** 1640,1649 ****
--- 1876,1887 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 435
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 437
      .restart local v3       #_arg0:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 1651,1656 ****
--- 1889,1895 ----
  
      if-eqz v2, :cond_21
  
+     .line 438
      sget-object v2, Landroid/location/Location;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 1661,1678 ****
--- 1900,1921 ----
  
      check-cast v4, Landroid/location/Location;
  
+     .line 443
      .local v4, _arg1:Landroid/location/Location;
      :goto_22
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3, v4}, Landroid/location/ILocationManager$Stub;->setTestProviderLocation(Ljava/lang/String;Landroid/location/Location;)V
  
+     .line 444
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 445
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 441
      .end local v4           #_arg1:Landroid/location/Location;
      :cond_21
      const/4 v4, 0x0
***************
*** 1680,1685 ****
--- 1923,1929 ----
      .restart local v4       #_arg1:Landroid/location/Location;
      goto :goto_22
  
+     .line 449
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v4           #_arg1:Landroid/location/Location;
      :sswitch_19
***************
*** 1689,1709 ****
--- 1933,1958 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 451
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 452
      .restart local v3       #_arg0:Ljava/lang/String;
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3}, Landroid/location/ILocationManager$Stub;->clearTestProviderLocation(Ljava/lang/String;)V
  
+     .line 453
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 454
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 458
      .end local v3           #_arg0:Ljava/lang/String;
      :sswitch_1a
      const-string v2, "android.location.ILocationManager"
***************
*** 1712,1721 ****
--- 1961,1972 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 460
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 462
      .restart local v3       #_arg0:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 1725,1748 ****
--- 1976,2004 ----
  
      const/4 v4, 0x1
  
+     .line 463
      .local v4, _arg1:Z
      :goto_23
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3, v4}, Landroid/location/ILocationManager$Stub;->setTestProviderEnabled(Ljava/lang/String;Z)V
  
+     .line 464
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 465
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 462
      .end local v4           #_arg1:Z
      :cond_22
      const/4 v4, 0x0
  
      goto :goto_23
  
+     .line 469
      .end local v3           #_arg0:Ljava/lang/String;
      :sswitch_1b
      const-string v2, "android.location.ILocationManager"
***************
*** 1751,1771 ****
--- 2007,2032 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 471
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 472
      .restart local v3       #_arg0:Ljava/lang/String;
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3}, Landroid/location/ILocationManager$Stub;->clearTestProviderEnabled(Ljava/lang/String;)V
  
+     .line 473
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 474
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 478
      .end local v3           #_arg0:Ljava/lang/String;
      :sswitch_1c
      const-string v2, "android.location.ILocationManager"
***************
*** 1774,1788 ****
--- 2035,2052 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 480
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 482
      .restart local v3       #_arg0:Ljava/lang/String;
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
  
+     .line 484
      .local v4, _arg1:I
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 1790,1795 ****
--- 2054,2060 ----
  
      if-eqz v2, :cond_23
  
+     .line 485
      sget-object v2, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      move-object/from16 v0, p2
***************
*** 1800,1805 ****
--- 2065,2071 ----
  
      check-cast v5, Landroid/os/Bundle;
  
+     .line 491
      .local v5, _arg2:Landroid/os/Bundle;
      :goto_24
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readLong()J
***************
*** 1817,1830 ****
--- 2083,2100 ----
  
      move-wide/from16 v22, v16
  
+     .line 492
      invoke-virtual/range {v18 .. v23}, Landroid/location/ILocationManager$Stub;->setTestProviderStatus(Ljava/lang/String;ILandroid/os/Bundle;J)V
  
+     .line 493
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 494
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 488
      .end local v5           #_arg2:Landroid/os/Bundle;
      .end local v16           #_arg3:J
      :cond_23
***************
*** 1833,1838 ****
--- 2103,2109 ----
      .restart local v5       #_arg2:Landroid/os/Bundle;
      goto :goto_24
  
+     .line 498
      .end local v3           #_arg0:Ljava/lang/String;
      .end local v4           #_arg1:I
      .end local v5           #_arg2:Landroid/os/Bundle;
***************
*** 1843,1863 ****
--- 2114,2139 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 500
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 501
      .restart local v3       #_arg0:Ljava/lang/String;
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v3}, Landroid/location/ILocationManager$Stub;->clearTestProviderStatus(Ljava/lang/String;)V
  
+     .line 502
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 503
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 507
      .end local v3           #_arg0:Ljava/lang/String;
      :sswitch_1e
      const-string v2, "android.location.ILocationManager"
***************
*** 1866,1880 ****
--- 2142,2159 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 509
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
+     .line 511
      .local v3, _arg0:I
      invoke-virtual/range {p2 .. p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
  
+     .line 512
      .restart local v4       #_arg1:I
      move-object/from16 v0, p0
  
***************
*** 1882,1890 ****
--- 2161,2171 ----
  
      move-result v50
  
+     .line 513
      .local v50, _result:Z
      invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 514
      if-eqz v50, :cond_24
  
      const/4 v2, 0x1
***************
*** 1894,1908 ****
--- 2175,2192 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 515
      const/4 v2, 0x1
  
      goto/16 :goto_0
  
+     .line 514
      :cond_24
      const/4 v2, 0x0
  
      goto :goto_25
  
+     .line 43
      nop
  
      :sswitch_data_0
