*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/ILocationProvider$Stub.smali	2012-03-23 19:36:16.765986712 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/ILocationProvider$Stub.smali.original	2012-03-23 19:34:45.036314023 -0500
***************
*** 76,87 ****
--- 76,90 ----
      .locals 1
  
      .prologue
+     .line 19
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 20
      const-string v0, "android.location.ILocationProvider"
  
      invoke-virtual {p0, p0, v0}, Landroid/location/ILocationProvider$Stub;->attachInterface(Landroid/os/IInterface;Ljava/lang/String;)V
  
+     .line 21
      return-void
  .end method
  
***************
*** 90,102 ****
--- 93,109 ----
      .parameter "obj"
  
      .prologue
+     .line 28
      if-nez p0, :cond_0
  
+     .line 29
      const/4 v0, 0x0
  
+     .line 35
      :goto_0
      return-object v0
  
+     .line 31
      :cond_0
      const-string v1, "android.location.ILocationProvider"
  
***************
*** 104,109 ****
--- 111,117 ----
  
      move-result-object v0
  
+     .line 32
      .local v0, iin:Landroid/os/IInterface;
      if-eqz v0, :cond_1
  
***************
*** 111,120 ****
--- 119,130 ----
  
      if-eqz v1, :cond_1
  
+     .line 33
      check-cast v0, Landroid/location/ILocationProvider;
  
      goto :goto_0
  
+     .line 35
      :cond_1
      new-instance v0, Landroid/location/ILocationProvider$Stub$Proxy;
  
***************
*** 130,135 ****
--- 140,146 ----
      .locals 0
  
      .prologue
+     .line 39
      return-object p0
  .end method
  
***************
*** 150,157 ****
--- 161,170 ----
  
      const/4 v6, 0x1
  
+     .line 43
      sparse-switch p1, :sswitch_data_0
  
+     .line 282
      invoke-super {p0, p1, p2, p3, p4}, Landroid/os/Binder;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
      move-result v6
***************
*** 159,164 ****
--- 172,178 ----
      :goto_0
      return v6
  
+     .line 47
      :sswitch_0
      const-string v5, "android.location.ILocationProvider"
  
***************
*** 166,183 ****
--- 180,201 ----
  
      goto :goto_0
  
+     .line 52
      :sswitch_1
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 53
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->requiresNetwork()Z
  
      move-result v3
  
+     .line 54
      .local v3, _result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 55
      if-eqz v3, :cond_0
  
      move v5, v6
***************
*** 187,205 ****
--- 205,227 ----
  
      goto :goto_0
  
+     .line 60
      .end local v3           #_result:Z
      :sswitch_2
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 61
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->requiresSatellite()Z
  
      move-result v3
  
+     .line 62
      .restart local v3       #_result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 63
      if-eqz v3, :cond_1
  
      move v5, v6
***************
*** 209,227 ****
--- 231,253 ----
  
      goto :goto_0
  
+     .line 68
      .end local v3           #_result:Z
      :sswitch_3
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 69
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->requiresCell()Z
  
      move-result v3
  
+     .line 70
      .restart local v3       #_result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 71
      if-eqz v3, :cond_2
  
      move v5, v6
***************
*** 231,249 ****
--- 257,279 ----
  
      goto :goto_0
  
+     .line 76
      .end local v3           #_result:Z
      :sswitch_4
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 77
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->hasMonetaryCost()Z
  
      move-result v3
  
+     .line 78
      .restart local v3       #_result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 79
      if-eqz v3, :cond_3
  
      move v5, v6
***************
*** 253,271 ****
--- 283,305 ----
  
      goto :goto_0
  
+     .line 84
      .end local v3           #_result:Z
      :sswitch_5
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 85
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->supportsAltitude()Z
  
      move-result v3
  
+     .line 86
      .restart local v3       #_result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 87
      if-eqz v3, :cond_4
  
      move v5, v6
***************
*** 275,293 ****
--- 309,331 ----
  
      goto :goto_0
  
+     .line 92
      .end local v3           #_result:Z
      :sswitch_6
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 93
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->supportsSpeed()Z
  
      move-result v3
  
+     .line 94
      .restart local v3       #_result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 95
      if-eqz v3, :cond_5
  
      move v5, v6
***************
*** 297,315 ****
--- 335,357 ----
  
      goto :goto_0
  
+     .line 100
      .end local v3           #_result:Z
      :sswitch_7
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 101
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->supportsBearing()Z
  
      move-result v3
  
+     .line 102
      .restart local v3       #_result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 103
      if-eqz v3, :cond_6
  
      move v5, v6
***************
*** 319,353 ****
--- 361,402 ----
  
      goto/16 :goto_0
  
+     .line 108
      .end local v3           #_result:Z
      :sswitch_8
      const-string v5, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 109
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->getPowerRequirement()I
  
      move-result v3
  
+     .line 110
      .local v3, _result:I
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 111
      invoke-virtual {p3, v3}, Landroid/os/Parcel;->writeInt(I)V
  
      goto/16 :goto_0
  
+     .line 116
      .end local v3           #_result:I
      :sswitch_9
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 118
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v7
  
      if-eqz v7, :cond_8
  
+     .line 119
      sget-object v7, Landroid/location/Criteria;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v7, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 356,370 ****
--- 405,422 ----
  
      check-cast v0, Landroid/location/Criteria;
  
+     .line 124
      .local v0, _arg0:Landroid/location/Criteria;
      :goto_1
      invoke-virtual {p0, v0}, Landroid/location/ILocationProvider$Stub;->meetsCriteria(Landroid/location/Criteria;)Z
  
      move-result v3
  
+     .line 125
      .local v3, _result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 126
      if-eqz v3, :cond_7
  
      move v5, v6
***************
*** 374,379 ****
--- 426,432 ----
  
      goto/16 :goto_0
  
+     .line 122
      .end local v0           #_arg0:Landroid/location/Criteria;
      .end local v3           #_result:Z
      :cond_8
***************
*** 382,459 ****
--- 435,532 ----
      .restart local v0       #_arg0:Landroid/location/Criteria;
      goto :goto_1
  
+     .line 131
      .end local v0           #_arg0:Landroid/location/Criteria;
      :sswitch_a
      const-string v5, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 132
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->getAccuracy()I
  
      move-result v3
  
+     .line 133
      .local v3, _result:I
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 134
      invoke-virtual {p3, v3}, Landroid/os/Parcel;->writeInt(I)V
  
      goto/16 :goto_0
  
+     .line 139
      .end local v3           #_result:I
      :sswitch_b
      const-string v5, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 140
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->enable()V
  
+     .line 141
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 146
      :sswitch_c
      const-string v5, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 147
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->disable()V
  
+     .line 148
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 153
      :sswitch_d
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 155
      new-instance v0, Landroid/os/Bundle;
  
      invoke-direct {v0}, Landroid/os/Bundle;-><init>()V
  
+     .line 156
      .local v0, _arg0:Landroid/os/Bundle;
      invoke-virtual {p0, v0}, Landroid/location/ILocationProvider$Stub;->getStatus(Landroid/os/Bundle;)I
  
      move-result v3
  
+     .line 157
      .restart local v3       #_result:I
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 158
      invoke-virtual {p3, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 159
      if-eqz v0, :cond_9
  
+     .line 160
      invoke-virtual {p3, v6}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 161
      invoke-virtual {v0, p3, v6}, Landroid/os/Bundle;->writeToParcel(Landroid/os/Parcel;I)V
  
      goto/16 :goto_0
  
+     .line 164
      :cond_9
      invoke-virtual {p3, v5}, Landroid/os/Parcel;->writeInt(I)V
  
      goto/16 :goto_0
  
+     .line 170
      .end local v0           #_arg0:Landroid/os/Bundle;
      .end local v3           #_result:I
      :sswitch_e
***************
*** 461,500 ****
--- 534,582 ----
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 171
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->getStatusUpdateTime()J
  
      move-result-wide v3
  
+     .line 172
      .local v3, _result:J
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 173
      invoke-virtual {p3, v3, v4}, Landroid/os/Parcel;->writeLong(J)V
  
      goto/16 :goto_0
  
+     .line 178
      .end local v3           #_result:J
      :sswitch_f
      const-string v5, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 179
      invoke-virtual {p0}, Landroid/location/ILocationProvider$Stub;->getInternalState()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 180
      .local v3, _result:Ljava/lang/String;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 181
      invoke-virtual {p3, v3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
      goto/16 :goto_0
  
+     .line 186
      .end local v3           #_result:Ljava/lang/String;
      :sswitch_10
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 188
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v7
***************
*** 503,512 ****
--- 585,596 ----
  
      move v0, v6
  
+     .line 189
      .local v0, _arg0:Z
      :goto_2
      invoke-virtual {p0, v0}, Landroid/location/ILocationProvider$Stub;->enableLocationTracking(Z)V
  
+     .line 190
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
***************
*** 515,531 ****
--- 599,619 ----
      :cond_a
      move v0, v5
  
+     .line 188
      goto :goto_2
  
+     .line 195
      :sswitch_11
      const-string v5, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 197
      invoke-virtual {p2}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
+     .line 199
      .local v0, _arg0:J
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 533,538 ****
--- 621,627 ----
  
      if-eqz v5, :cond_b
  
+     .line 200
      sget-object v5, Landroid/os/WorkSource;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v5, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 541,554 ****
--- 630,646 ----
  
      check-cast v2, Landroid/os/WorkSource;
  
+     .line 205
      .local v2, _arg1:Landroid/os/WorkSource;
      :goto_3
      invoke-virtual {p0, v0, v1, v2}, Landroid/location/ILocationProvider$Stub;->setMinTime(JLandroid/os/WorkSource;)V
  
+     .line 206
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 203
      .end local v2           #_arg1:Landroid/os/WorkSource;
      :cond_b
      const/4 v2, 0x0
***************
*** 556,561 ****
--- 648,654 ----
      .restart local v2       #_arg1:Landroid/os/WorkSource;
      goto :goto_3
  
+     .line 211
      .end local v0           #_arg0:J
      .end local v2           #_arg1:Landroid/os/WorkSource;
      :sswitch_12
***************
*** 563,572 ****
--- 656,667 ----
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 213
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 215
      .local v0, _arg0:I
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 574,579 ****
--- 669,675 ----
  
      if-eqz v5, :cond_c
  
+     .line 216
      sget-object v5, Landroid/net/NetworkInfo;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v5, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 582,595 ****
--- 678,694 ----
  
      check-cast v2, Landroid/net/NetworkInfo;
  
+     .line 221
      .local v2, _arg1:Landroid/net/NetworkInfo;
      :goto_4
      invoke-virtual {p0, v0, v2}, Landroid/location/ILocationProvider$Stub;->updateNetworkState(ILandroid/net/NetworkInfo;)V
  
+     .line 222
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 219
      .end local v2           #_arg1:Landroid/net/NetworkInfo;
      :cond_c
      const/4 v2, 0x0
***************
*** 597,602 ****
--- 696,702 ----
      .restart local v2       #_arg1:Landroid/net/NetworkInfo;
      goto :goto_4
  
+     .line 227
      .end local v0           #_arg0:I
      .end local v2           #_arg1:Landroid/net/NetworkInfo;
      :sswitch_13
***************
*** 604,615 ****
--- 704,717 ----
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 229
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
  
      if-eqz v5, :cond_d
  
+     .line 230
      sget-object v5, Landroid/location/Location;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v5, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 618,631 ****
--- 720,736 ----
  
      check-cast v0, Landroid/location/Location;
  
+     .line 235
      .local v0, _arg0:Landroid/location/Location;
      :goto_5
      invoke-virtual {p0, v0}, Landroid/location/ILocationProvider$Stub;->updateLocation(Landroid/location/Location;)V
  
+     .line 236
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 233
      .end local v0           #_arg0:Landroid/location/Location;
      :cond_d
      const/4 v0, 0x0
***************
*** 633,648 ****
--- 738,756 ----
      .restart local v0       #_arg0:Landroid/location/Location;
      goto :goto_5
  
+     .line 241
      .end local v0           #_arg0:Landroid/location/Location;
      :sswitch_14
      const-string v7, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 243
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 245
      .local v0, _arg0:Ljava/lang/String;
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
***************
*** 650,655 ****
--- 758,764 ----
  
      if-eqz v7, :cond_e
  
+     .line 246
      sget-object v7, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v7, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 658,672 ****
--- 767,784 ----
  
      check-cast v2, Landroid/os/Bundle;
  
+     .line 251
      .local v2, _arg1:Landroid/os/Bundle;
      :goto_6
      invoke-virtual {p0, v0, v2}, Landroid/location/ILocationProvider$Stub;->sendExtraCommand(Ljava/lang/String;Landroid/os/Bundle;)Z
  
      move-result v3
  
+     .line 252
      .local v3, _result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 253
      if-eqz v3, :cond_f
  
      move v7, v6
***************
*** 674,687 ****
--- 786,803 ----
      :goto_7
      invoke-virtual {p3, v7}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 254
      if-eqz v2, :cond_10
  
+     .line 255
      invoke-virtual {p3, v6}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 256
      invoke-virtual {v2, p3, v6}, Landroid/os/Bundle;->writeToParcel(Landroid/os/Parcel;I)V
  
      goto/16 :goto_0
  
+     .line 249
      .end local v2           #_arg1:Landroid/os/Bundle;
      .end local v3           #_result:Z
      :cond_e
***************
*** 694,706 ****
--- 810,825 ----
      :cond_f
      move v7, v5
  
+     .line 253
      goto :goto_7
  
+     .line 259
      :cond_10
      invoke-virtual {p3, v5}, Landroid/os/Parcel;->writeInt(I)V
  
      goto/16 :goto_0
  
+     .line 265
      .end local v0           #_arg0:Ljava/lang/String;
      .end local v2           #_arg1:Landroid/os/Bundle;
      .end local v3           #_result:Z
***************
*** 709,742 ****
--- 828,869 ----
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 267
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 268
      .local v0, _arg0:I
      invoke-virtual {p0, v0}, Landroid/location/ILocationProvider$Stub;->addListener(I)V
  
+     .line 269
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 274
      .end local v0           #_arg0:I
      :sswitch_16
      const-string v5, "android.location.ILocationProvider"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 276
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 277
      .restart local v0       #_arg0:I
      invoke-virtual {p0, v0}, Landroid/location/ILocationProvider$Stub;->removeListener(I)V
  
+     .line 278
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto/16 :goto_0
  
+     .line 43
      nop
  
      :sswitch_data_0
