*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/LocationManager$ListenerTransport.smali	2012-03-23 19:36:17.836006223 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/LocationManager$ListenerTransport.smali.original	2012-03-23 19:34:45.041314115 -0500
***************
*** 40,62 ****
--- 40,68 ----
      .parameter "looper"
  
      .prologue
+     .line 169
      iput-object p1, p0, Landroid/location/LocationManager$ListenerTransport;->this$0:Landroid/location/LocationManager;
  
      invoke-direct {p0}, Landroid/location/ILocationListener$Stub;-><init>()V
  
+     .line 170
      iput-object p2, p0, Landroid/location/LocationManager$ListenerTransport;->mListener:Landroid/location/LocationListener;
  
+     .line 172
      if-nez p3, :cond_0
  
+     .line 173
      new-instance v0, Landroid/location/LocationManager$ListenerTransport$1;
  
      invoke-direct {v0, p0, p1}, Landroid/location/LocationManager$ListenerTransport$1;-><init>(Landroid/location/LocationManager$ListenerTransport;Landroid/location/LocationManager;)V
  
      iput-object v0, p0, Landroid/location/LocationManager$ListenerTransport;->mListenerHandler:Landroid/os/Handler;
  
+     .line 187
      :goto_0
      return-void
  
+     .line 180
      :cond_0
      new-instance v0, Landroid/location/LocationManager$ListenerTransport$2;
  
***************
*** 72,81 ****
--- 78,89 ----
      .parameter "msg"
  
      .prologue
+     .line 224
      iget v6, p1, Landroid/os/Message;->what:I
  
      packed-switch v6, :pswitch_data_0
  
+     .line 244
      :goto_0
      :try_start_0
      iget-object v6, p0, Landroid/location/LocationManager$ListenerTransport;->this$0:Landroid/location/LocationManager;
***************
*** 89,97 ****
--- 97,107 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 248
      :goto_1
      return-void
  
+     .line 226
      :pswitch_0
      new-instance v3, Landroid/location/Location;
  
***************
*** 101,106 ****
--- 111,117 ----
  
      invoke-direct {v3, v6}, Landroid/location/Location;-><init>(Landroid/location/Location;)V
  
+     .line 227
      .local v3, location:Landroid/location/Location;
      iget-object v6, p0, Landroid/location/LocationManager$ListenerTransport;->mListener:Landroid/location/LocationListener;
  
***************
*** 108,133 ****
  
      goto :goto_0
  
      .end local v3           #location:Landroid/location/Location;
      :pswitch_1
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/Bundle;
  
      .local v0, b:Landroid/os/Bundle;
!     const-string v6, "provider"
  
      invoke-virtual {v0, v6}, Landroid/os/Bundle;->getString(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v4
  
      .local v4, provider:Ljava/lang/String;
!     const-string v6, "status"
  
      invoke-virtual {v0, v6}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
      move-result v5
  
      .local v5, status:I
      const-string v6, "extras"
  
--- 119,148 ----
  
      goto :goto_0
  
+     .line 230
      .end local v3           #location:Landroid/location/Location;
      :pswitch_1
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/Bundle;
  
+     .line 231
      .local v0, b:Landroid/os/Bundle;
!     const-string/jumbo v6, "provider"
  
      invoke-virtual {v0, v6}, Landroid/os/Bundle;->getString(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v4
  
+     .line 232
      .local v4, provider:Ljava/lang/String;
!     const-string/jumbo v6, "status"
  
      invoke-virtual {v0, v6}, Landroid/os/Bundle;->getInt(Ljava/lang/String;)I
  
      move-result v5
  
+     .line 233
      .local v5, status:I
      const-string v6, "extras"
  
***************
*** 135,140 ****
--- 150,156 ----
  
      move-result-object v2
  
+     .line 234
      .local v2, extras:Landroid/os/Bundle;
      iget-object v6, p0, Landroid/location/LocationManager$ListenerTransport;->mListener:Landroid/location/LocationListener;
  
***************
*** 142,147 ****
--- 158,164 ----
  
      goto :goto_0
  
+     .line 237
      .end local v0           #b:Landroid/os/Bundle;
      .end local v2           #extras:Landroid/os/Bundle;
      .end local v4           #provider:Ljava/lang/String;
***************
*** 157,162 ****
--- 174,180 ----
  
      goto :goto_0
  
+     .line 240
      :pswitch_3
      iget-object v7, p0, Landroid/location/LocationManager$ListenerTransport;->mListener:Landroid/location/LocationListener;
  
***************
*** 168,176 ****
--- 186,196 ----
  
      goto :goto_0
  
+     .line 245
      :catch_0
      move-exception v1
  
+     .line 246
      .local v1, e:Landroid/os/RemoteException;
      const-string v6, "LocationManager"
  
***************
*** 180,185 ****
--- 200,206 ----
  
      goto :goto_1
  
+     .line 224
      nop
  
      :pswitch_data_0
***************
*** 197,202 ****
--- 218,224 ----
      .parameter "x1"
  
      .prologue
+     .line 160
      invoke-direct {p0, p1}, Landroid/location/LocationManager$ListenerTransport;->_handleMessage(Landroid/os/Message;)V
  
      return-void
***************
*** 209,229 ****
--- 231,256 ----
      .parameter "location"
  
      .prologue
+     .line 190
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 191
      .local v0, msg:Landroid/os/Message;
      const/4 v1, 0x1
  
      iput v1, v0, Landroid/os/Message;->what:I
  
+     .line 192
      iput-object p1, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 193
      iget-object v1, p0, Landroid/location/LocationManager$ListenerTransport;->mListenerHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 194
      return-void
  .end method
  
***************
*** 232,252 ****
--- 259,284 ----
      .parameter "provider"
  
      .prologue
+     .line 217
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 218
      .local v0, msg:Landroid/os/Message;
      const/4 v1, 0x4
  
      iput v1, v0, Landroid/os/Message;->what:I
  
+     .line 219
      iput-object p1, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 220
      iget-object v1, p0, Landroid/location/LocationManager$ListenerTransport;->mListenerHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 221
      return-void
  .end method
  
***************
*** 255,275 ****
--- 287,312 ----
      .parameter "provider"
  
      .prologue
+     .line 210
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 211
      .local v0, msg:Landroid/os/Message;
      const/4 v1, 0x3
  
      iput v1, v0, Landroid/os/Message;->what:I
  
+     .line 212
      iput-object p1, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 213
      iget-object v1, p0, Landroid/location/LocationManager$ListenerTransport;->mListenerHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 214
      return-void
  .end method
  
***************
*** 280,319 ****
      .parameter "extras"
  
      .prologue
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v1
  
      .local v1, msg:Landroid/os/Message;
      const/4 v2, 0x2
  
      iput v2, v1, Landroid/os/Message;->what:I
  
      new-instance v0, Landroid/os/Bundle;
  
      invoke-direct {v0}, Landroid/os/Bundle;-><init>()V
  
      .local v0, b:Landroid/os/Bundle;
!     const-string v2, "provider"
  
      invoke-virtual {v0, v2, p1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v2, "status"
  
      invoke-virtual {v0, v2, p2}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
      if-eqz p3, :cond_0
  
      const-string v2, "extras"
  
      invoke-virtual {v0, v2, p3}, Landroid/os/Bundle;->putBundle(Ljava/lang/String;Landroid/os/Bundle;)V
  
      :cond_0
      iput-object v0, v1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      iget-object v2, p0, Landroid/location/LocationManager$ListenerTransport;->mListenerHandler:Landroid/os/Handler;
  
      invoke-virtual {v2, v1}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
      return-void
  .end method
--- 317,366 ----
      .parameter "extras"
  
      .prologue
+     .line 197
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v1
  
+     .line 198
      .local v1, msg:Landroid/os/Message;
      const/4 v2, 0x2
  
      iput v2, v1, Landroid/os/Message;->what:I
  
+     .line 199
      new-instance v0, Landroid/os/Bundle;
  
      invoke-direct {v0}, Landroid/os/Bundle;-><init>()V
  
+     .line 200
      .local v0, b:Landroid/os/Bundle;
!     const-string/jumbo v2, "provider"
  
      invoke-virtual {v0, v2, p1}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 201
!     const-string/jumbo v2, "status"
  
      invoke-virtual {v0, v2, p2}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
+     .line 202
      if-eqz p3, :cond_0
  
+     .line 203
      const-string v2, "extras"
  
      invoke-virtual {v0, v2, p3}, Landroid/os/Bundle;->putBundle(Ljava/lang/String;Landroid/os/Bundle;)V
  
+     .line 205
      :cond_0
      iput-object v0, v1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 206
      iget-object v2, p0, Landroid/location/LocationManager$ListenerTransport;->mListenerHandler:Landroid/os/Handler;
  
      invoke-virtual {v2, v1}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 207
      return-void
  .end method
