*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/LocationManager$GpsStatusListenerTransport$Nmea.smali	2012-03-23 19:36:17.120993185 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/location/LocationManager$GpsStatusListenerTransport$Nmea.smali.original	2012-03-23 19:34:45.038314059 -0500
***************
*** 30,42 ****
--- 30,46 ----
      .parameter "nmea"
  
      .prologue
+     .line 1214
      iput-object p1, p0, Landroid/location/LocationManager$GpsStatusListenerTransport$Nmea;->this$1:Landroid/location/LocationManager$GpsStatusListenerTransport;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1215
      iput-wide p2, p0, Landroid/location/LocationManager$GpsStatusListenerTransport$Nmea;->mTimestamp:J
  
+     .line 1216
      iput-object p4, p0, Landroid/location/LocationManager$GpsStatusListenerTransport$Nmea;->mNmea:Ljava/lang/String;
  
+     .line 1217
      return-void
  .end method
