*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/mms/pdu/Base64.smali	2012-03-23 19:37:57.845829899 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/mms/pdu/Base64.smali.original	2012-03-23 19:34:46.435339535 -0500
***************
*** 20,45 ****
--- 20,50 ----
      .prologue
      const/16 v3, 0xff
  
+     .line 37
      new-array v1, v3, [B
  
      sput-object v1, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
+     .line 41
      const/4 v0, 0x0
  
      .local v0, i:I
      :goto_0
      if-ge v0, v3, :cond_0
  
+     .line 42
      sget-object v1, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      const/4 v2, -0x1
  
      aput-byte v2, v1, v0
  
+     .line 41
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 44
      :cond_0
      const/16 v0, 0x5a
  
***************
*** 48,53 ****
--- 53,59 ----
  
      if-lt v0, v1, :cond_1
  
+     .line 45
      sget-object v1, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      add-int/lit8 v2, v0, -0x41
***************
*** 56,65 ****
--- 62,73 ----
  
      aput-byte v2, v1, v0
  
+     .line 44
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_1
  
+     .line 47
      :cond_1
      const/16 v0, 0x7a
  
***************
*** 68,73 ****
--- 76,82 ----
  
      if-lt v0, v1, :cond_2
  
+     .line 48
      sget-object v1, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      add-int/lit8 v2, v0, -0x61
***************
*** 78,87 ****
--- 87,98 ----
  
      aput-byte v2, v1, v0
  
+     .line 47
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_2
  
+     .line 50
      :cond_2
      const/16 v0, 0x39
  
***************
*** 90,95 ****
--- 101,107 ----
  
      if-lt v0, v1, :cond_3
  
+     .line 51
      sget-object v1, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      add-int/lit8 v2, v0, -0x30
***************
*** 100,109 ****
--- 112,123 ----
  
      aput-byte v2, v1, v0
  
+     .line 50
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_3
  
+     .line 54
      :cond_3
      sget-object v1, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
***************
*** 113,118 ****
--- 127,133 ----
  
      aput-byte v3, v1, v2
  
+     .line 55
      sget-object v1, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      const/16 v2, 0x2f
***************
*** 121,126 ****
--- 136,142 ----
  
      aput-byte v3, v1, v2
  
+     .line 56
      return-void
  .end method
  
***************
*** 128,133 ****
--- 144,150 ----
      .locals 0
  
      .prologue
+     .line 20
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 138,167 ****
--- 155,191 ----
      .parameter "base64Data"
  
      .prologue
+     .line 66
      invoke-static {p0}, Lcom/google/android/mms/pdu/Base64;->discardNonBase64([B)[B
  
      move-result-object p0
  
+     .line 69
      array-length v12, p0
  
      if-nez v12, :cond_1
  
+     .line 70
      const/4 v12, 0x0
  
      new-array v5, v12, [B
  
+     .line 123
      :cond_0
      :goto_0
      return-object v5
  
+     .line 73
      :cond_1
      array-length v12, p0
  
      div-int/lit8 v11, v12, 0x4
  
+     .line 74
      .local v11, numberQuadruple:I
      const/4 v5, 0x0
  
+     .line 75
      .local v5, decodedData:[B
      const/4 v0, 0x0
  
***************
*** 180,194 ****
--- 204,222 ----
      .local v9, marker0:B
      const/4 v10, 0x0
  
+     .line 79
      .local v10, marker1:B
      const/4 v6, 0x0
  
+     .line 80
      .local v6, encodedIndex:I
      const/4 v4, 0x0
  
+     .line 83
      .local v4, dataIndex:I
      array-length v8, p0
  
+     .line 85
      .local v8, lastData:I
      :cond_2
      add-int/lit8 v12, v8, -0x1
***************
*** 199,241 ****
--- 227,278 ----
  
      if-ne v12, v13, :cond_3
  
+     .line 86
      add-int/lit8 v8, v8, -0x1
  
      if-nez v8, :cond_2
  
+     .line 87
      const/4 v12, 0x0
  
      new-array v5, v12, [B
  
      goto :goto_0
  
+     .line 90
      :cond_3
      sub-int v12, v8, v11
  
      new-array v5, v12, [B
  
+     .line 93
      const/4 v7, 0x0
  
      .local v7, i:I
      :goto_1
      if-ge v7, v11, :cond_0
  
+     .line 94
      mul-int/lit8 v4, v7, 0x4
  
+     .line 95
      add-int/lit8 v12, v4, 0x2
  
      aget-byte v9, p0, v12
  
+     .line 96
      add-int/lit8 v12, v4, 0x3
  
      aget-byte v10, p0, v12
  
+     .line 98
      sget-object v12, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      aget-byte v13, p0, v4
  
      aget-byte v0, v12, v13
  
+     .line 99
      sget-object v12, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      add-int/lit8 v13, v4, 0x1
***************
*** 244,249 ****
--- 281,287 ----
  
      aget-byte v1, v12, v13
  
+     .line 101
      const/16 v12, 0x3d
  
      if-eq v9, v12, :cond_5
***************
*** 252,265 ****
--- 290,306 ----
  
      if-eq v10, v12, :cond_5
  
+     .line 103
      sget-object v12, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      aget-byte v2, v12, v9
  
+     .line 104
      sget-object v12, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      aget-byte v3, v12, v10
  
+     .line 106
      shl-int/lit8 v12, v0, 0x2
  
      shr-int/lit8 v13, v1, 0x4
***************
*** 270,275 ****
--- 311,317 ----
  
      aput-byte v12, v5, v6
  
+     .line 107
      add-int/lit8 v12, v6, 0x1
  
      and-int/lit8 v13, v1, 0xf
***************
*** 286,291 ****
--- 328,334 ----
  
      aput-byte v13, v5, v12
  
+     .line 109
      add-int/lit8 v12, v6, 0x2
  
      shl-int/lit8 v13, v2, 0x6
***************
*** 296,314 ****
--- 339,361 ----
  
      aput-byte v13, v5, v12
  
+     .line 121
      :cond_4
      :goto_2
      add-int/lit8 v6, v6, 0x3
  
+     .line 93
      add-int/lit8 v7, v7, 0x1
  
      goto :goto_1
  
+     .line 110
      :cond_5
      const/16 v12, 0x3d
  
      if-ne v9, v12, :cond_6
  
+     .line 112
      shl-int/lit8 v12, v0, 0x2
  
      shr-int/lit8 v13, v1, 0x4
***************
*** 321,335 ****
--- 368,385 ----
  
      goto :goto_2
  
+     .line 113
      :cond_6
      const/16 v12, 0x3d
  
      if-ne v10, v12, :cond_4
  
+     .line 115
      sget-object v12, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
      aget-byte v2, v12, v9
  
+     .line 117
      shl-int/lit8 v12, v0, 0x2
  
      shr-int/lit8 v13, v1, 0x4
***************
*** 340,345 ****
--- 390,396 ----
  
      aput-byte v12, v5, v6
  
+     .line 118
      add-int/lit8 v12, v6, 0x1
  
      and-int/lit8 v13, v1, 0xf
***************
*** 366,378 ****
--- 417,432 ----
      .prologue
      const/4 v6, 0x0
  
+     .line 152
      array-length v5, p0
  
      new-array v2, v5, [B
  
+     .line 153
      .local v2, groomedData:[B
      const/4 v0, 0x0
  
+     .line 155
      .local v0, bytesCopied:I
      const/4 v3, 0x0
  
***************
*** 382,387 ****
--- 436,442 ----
  
      if-ge v3, v5, :cond_1
  
+     .line 156
      aget-byte v5, p0, v3
  
      invoke-static {v5}, Lcom/google/android/mms/pdu/Base64;->isBase64(B)Z
***************
*** 390,395 ****
--- 445,451 ----
  
      if-eqz v5, :cond_0
  
+     .line 157
      add-int/lit8 v1, v0, 0x1
  
      .end local v0           #bytesCopied:I
***************
*** 400,405 ****
--- 456,462 ----
  
      move v0, v1
  
+     .line 155
      .end local v1           #bytesCopied:I
      .restart local v0       #bytesCopied:I
      :cond_0
***************
*** 407,418 ****
--- 464,478 ----
  
      goto :goto_0
  
+     .line 161
      :cond_1
      new-array v4, v0, [B
  
+     .line 163
      .local v4, packedData:[B
      invoke-static {v2, v6, v4, v6, v0}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 165
      return-object v4
  .end method
  
***************
*** 423,436 ****
--- 483,499 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 133
      const/16 v1, 0x3d
  
      if-ne p0, v1, :cond_1
  
+     .line 138
      :cond_0
      :goto_0
      return v0
  
+     .line 135
      :cond_1
      sget-object v1, Lcom/google/android/mms/pdu/Base64;->base64Alphabet:[B
  
***************
*** 440,445 ****
--- 503,509 ----
  
      if-ne v1, v2, :cond_0
  
+     .line 136
      const/4 v0, 0x0
  
      goto :goto_0
