*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/mms/pdu/GenericPdu.smali	2012-03-23 19:37:57.658826490 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/mms/pdu/GenericPdu.smali.original	2012-03-23 19:34:46.433339497 -0500
***************
*** 12,29 ****
--- 12,33 ----
      .locals 1
  
      .prologue
+     .line 31
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 26
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
+     .line 32
      new-instance v0, Lcom/google/android/mms/pdu/PduHeaders;
  
      invoke-direct {v0}, Lcom/google/android/mms/pdu/PduHeaders;-><init>()V
  
      iput-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
+     .line 33
      return-void
  .end method
  
***************
*** 32,45 ****
--- 36,53 ----
      .parameter "headers"
  
      .prologue
+     .line 40
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 26
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
+     .line 41
      iput-object p1, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
+     .line 42
      return-void
  .end method
  
***************
*** 49,54 ****
--- 57,63 ----
      .locals 2
  
      .prologue
+     .line 101
      iget-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x89
***************
*** 64,69 ****
--- 73,79 ----
      .locals 2
  
      .prologue
+     .line 59
      iget-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x8c
***************
*** 79,84 ****
--- 89,95 ----
      .locals 2
  
      .prologue
+     .line 79
      iget-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x8d
***************
*** 94,99 ****
--- 105,111 ----
      .locals 1
  
      .prologue
+     .line 50
      iget-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      return-object v0
***************
*** 104,115 ****
--- 116,129 ----
      .parameter "value"
  
      .prologue
+     .line 111
      iget-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x89
  
      invoke-virtual {v0, p1, v1}, Lcom/google/android/mms/pdu/PduHeaders;->setEncodedStringValue(Lcom/google/android/mms/pdu/EncodedStringValue;I)V
  
+     .line 112
      return-void
  .end method
  
***************
*** 123,134 ****
--- 137,150 ----
      .end annotation
  
      .prologue
+     .line 70
      iget-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x8c
  
      invoke-virtual {v0, p1, v1}, Lcom/google/android/mms/pdu/PduHeaders;->setOctet(II)V
  
+     .line 71
      return-void
  .end method
  
***************
*** 142,152 ****
--- 158,170 ----
      .end annotation
  
      .prologue
+     .line 90
      iget-object v0, p0, Lcom/google/android/mms/pdu/GenericPdu;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x8d
  
      invoke-virtual {v0, p1, v1}, Lcom/google/android/mms/pdu/PduHeaders;->setOctet(II)V
  
+     .line 91
      return-void
  .end method
