*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/mms/pdu/SendConf.smali	2012-03-23 19:37:57.611825633 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/mms/pdu/SendConf.smali.original	2012-03-23 19:34:46.433339497 -0500
***************
*** 13,24 ****
--- 13,27 ----
      .end annotation
  
      .prologue
+     .line 32
      invoke-direct {p0}, Lcom/google/android/mms/pdu/GenericPdu;-><init>()V
  
+     .line 33
      const/16 v0, 0x81
  
      invoke-virtual {p0, v0}, Lcom/google/android/mms/pdu/SendConf;->setMessageType(I)V
  
+     .line 34
      return-void
  .end method
  
***************
*** 27,34 ****
--- 30,39 ----
      .parameter "headers"
  
      .prologue
+     .line 42
      invoke-direct {p0, p1}, Lcom/google/android/mms/pdu/GenericPdu;-><init>(Lcom/google/android/mms/pdu/PduHeaders;)V
  
+     .line 43
      return-void
  .end method
  
***************
*** 38,43 ****
--- 43,49 ----
      .locals 2
  
      .prologue
+     .line 51
      iget-object v0, p0, Lcom/google/android/mms/pdu/SendConf;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x8b
***************
*** 53,58 ****
--- 59,65 ----
      .locals 2
  
      .prologue
+     .line 70
      iget-object v0, p0, Lcom/google/android/mms/pdu/SendConf;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x92
***************
*** 68,73 ****
--- 75,81 ----
      .locals 2
  
      .prologue
+     .line 89
      iget-object v0, p0, Lcom/google/android/mms/pdu/SendConf;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x98
***************
*** 84,95 ****
--- 92,105 ----
      .parameter "value"
  
      .prologue
+     .line 61
      iget-object v0, p0, Lcom/google/android/mms/pdu/SendConf;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x8b
  
      invoke-virtual {v0, p1, v1}, Lcom/google/android/mms/pdu/PduHeaders;->setTextString([BI)V
  
+     .line 62
      return-void
  .end method
  
***************
*** 103,114 ****
--- 113,126 ----
      .end annotation
  
      .prologue
+     .line 80
      iget-object v0, p0, Lcom/google/android/mms/pdu/SendConf;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x92
  
      invoke-virtual {v0, p1, v1}, Lcom/google/android/mms/pdu/PduHeaders;->setOctet(II)V
  
+     .line 81
      return-void
  .end method
  
***************
*** 117,127 ****
--- 129,141 ----
      .parameter "value"
  
      .prologue
+     .line 99
      iget-object v0, p0, Lcom/google/android/mms/pdu/SendConf;->mPduHeaders:Lcom/google/android/mms/pdu/PduHeaders;
  
      const/16 v1, 0x98
  
      invoke-virtual {v0, p1, v1}, Lcom/google/android/mms/pdu/PduHeaders;->setTextString([BI)V
  
+     .line 100
      return-void
  .end method
