*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/collect/Maps.smali	2012-03-23 19:37:58.323838616 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/collect/Maps.smali.original	2012-03-23 19:34:46.440339626 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 24
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 28,33 ****
--- 29,35 ----
      .end annotation
  
      .prologue
+     .line 31
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
