*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityStack$1.smali	2012-03-23 19:35:03.750655278 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityStack$1.smali.original	2012-03-23 19:34:47.090351479 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 285
      iput-object p1, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
***************
*** 46,59 ****
--- 47,63 ----
  
      const/4 v2, 0x0
  
+     .line 291
      iget v3, p1, Landroid/os/Message;->what:I
  
      packed-switch v3, :pswitch_data_0
  
+     .line 353
      :goto_0
      :pswitch_0
      return-void
  
+     .line 293
      :pswitch_1
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
***************
*** 61,66 ****
--- 65,71 ----
  
      monitor-enter v3
  
+     .line 294
      :try_start_0
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
***************
*** 72,93 ****
--- 77,102 ----
  
      if-eqz v2, :cond_0
  
+     .line 295
      const-string v2, "ActivityManager"
  
      const-string v4, "Sleep timeout!  Sleeping now."
  
      invoke-static {v2, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 296
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      const/4 v4, 0x1
  
      iput-boolean v4, v2, Lcom/android/server/am/ActivityStack;->mSleepTimeout:Z
  
+     .line 297
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      invoke-virtual {v2}, Lcom/android/server/am/ActivityStack;->checkReadyForSleepLocked()V
  
+     .line 299
      :cond_0
      monitor-exit v3
  
***************
*** 102,112 ****
--- 111,123 ----
  
      throw v2
  
+     .line 302
      :pswitch_2
      iget-object v1, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v1, Lcom/android/server/am/ActivityRecord;
  
+     .line 305
      .local v1, r:Lcom/android/server/am/ActivityRecord;
      const-string v3, "ActivityManager"
  
***************
*** 130,135 ****
--- 141,147 ----
  
      invoke-static {v3, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 306
      iget-object v3, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      if-eqz v1, :cond_1
***************
*** 141,146 ****
--- 153,159 ----
  
      goto :goto_0
  
+     .line 309
      .end local v1           #r:Lcom/android/server/am/ActivityRecord;
      :pswitch_3
      iget-object v3, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
***************
*** 151,162 ****
--- 164,177 ----
  
      if-eqz v3, :cond_2
  
+     .line 310
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      iget-object v2, v2, Lcom/android/server/am/ActivityStack;->mService:Lcom/android/server/am/ActivityManagerService;
  
      iput-boolean v5, v2, Lcom/android/server/am/ActivityManagerService;->mDidDexOpt:Z
  
+     .line 311
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      iget-object v2, v2, Lcom/android/server/am/ActivityStack;->mHandler:Landroid/os/Handler;
***************
*** 167,177 ****
--- 182,194 ----
  
      move-result-object v0
  
+     .line 312
      .local v0, nmsg:Landroid/os/Message;
      iget-object v2, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      iput-object v2, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 313
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      iget-object v2, v2, Lcom/android/server/am/ActivityStack;->mHandler:Landroid/os/Handler;
***************
*** 180,191 ****
--- 197,210 ----
  
      goto :goto_0
  
+     .line 318
      .end local v0           #nmsg:Landroid/os/Message;
      :cond_2
      iget-object v1, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v1, Lcom/android/server/am/ActivityRecord;
  
+     .line 319
      .restart local v1       #r:Lcom/android/server/am/ActivityRecord;
      const-string v3, "ActivityManager"
  
***************
*** 209,214 ****
--- 228,234 ----
  
      invoke-static {v3, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 320
      iget-object v4, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      if-eqz v1, :cond_3
***************
*** 225,236 ****
--- 245,258 ----
  
      goto :goto_1
  
+     .line 323
      .end local v1           #r:Lcom/android/server/am/ActivityRecord;
      :pswitch_4
      iget-object v1, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v1, Lcom/android/server/am/ActivityRecord;
  
+     .line 326
      .restart local v1       #r:Lcom/android/server/am/ActivityRecord;
      const-string v3, "ActivityManager"
  
***************
*** 254,259 ****
--- 276,282 ----
  
      invoke-static {v3, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 327
      iget-object v3, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      if-eqz v1, :cond_4
***************
*** 265,276 ****
--- 288,301 ----
  
      goto/16 :goto_0
  
+     .line 330
      .end local v1           #r:Lcom/android/server/am/ActivityRecord;
      :pswitch_5
      iget-object v1, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v1, Lcom/android/server/am/ActivityRecord;
  
+     .line 331
      .restart local v1       #r:Lcom/android/server/am/ActivityRecord;
      iget-object v4, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
***************
*** 288,293 ****
--- 313,319 ----
  
      goto :goto_2
  
+     .line 334
      .end local v1           #r:Lcom/android/server/am/ActivityRecord;
      :pswitch_6
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
***************
*** 298,309 ****
--- 324,337 ----
  
      if-eqz v2, :cond_6
  
+     .line 335
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      iget-object v2, v2, Lcom/android/server/am/ActivityStack;->mService:Lcom/android/server/am/ActivityManagerService;
  
      iput-boolean v5, v2, Lcom/android/server/am/ActivityManagerService;->mDidDexOpt:Z
  
+     .line 336
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      iget-object v2, v2, Lcom/android/server/am/ActivityStack;->mHandler:Landroid/os/Handler;
***************
*** 314,319 ****
--- 342,348 ----
  
      move-result-object v0
  
+     .line 337
      .restart local v0       #nmsg:Landroid/os/Message;
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
***************
*** 323,328 ****
--- 352,358 ----
  
      goto/16 :goto_0
  
+     .line 340
      .end local v0           #nmsg:Landroid/os/Message;
      :cond_6
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
***************
*** 331,336 ****
--- 361,367 ----
  
      monitor-enter v3
  
+     .line 341
      :try_start_1
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
***************
*** 342,359 ****
--- 373,393 ----
  
      if-eqz v2, :cond_7
  
+     .line 342
      const-string v2, "ActivityManager"
  
      const-string v4, "Launch timeout has expired, giving up wake lock!"
  
      invoke-static {v2, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 343
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
      iget-object v2, v2, Lcom/android/server/am/ActivityStack;->mLaunchingActivity:Landroid/os/PowerManager$WakeLock;
  
      invoke-virtual {v2}, Landroid/os/PowerManager$WakeLock;->release()V
  
+     .line 345
      :cond_7
      monitor-exit v3
  
***************
*** 368,373 ****
--- 402,408 ----
  
      throw v2
  
+     .line 348
      :pswitch_7
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
***************
*** 375,380 ****
--- 410,416 ----
  
      monitor-enter v3
  
+     .line 349
      :try_start_2
      iget-object v2, p0, Lcom/android/server/am/ActivityStack$1;->this$0:Lcom/android/server/am/ActivityStack;
  
***************
*** 382,387 ****
--- 418,424 ----
  
      invoke-virtual {v2, v4}, Lcom/android/server/am/ActivityStack;->resumeTopActivityLocked(Lcom/android/server/am/ActivityRecord;)Z
  
+     .line 350
      monitor-exit v3
  
      goto/16 :goto_0
***************
*** 395,400 ****
--- 432,438 ----
  
      throw v2
  
+     .line 291
      :pswitch_data_0
      .packed-switch 0x8
          :pswitch_1
