*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityManagerService$5.smali	2012-03-23 19:35:01.044605933 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityManagerService$5.smali.original	2012-03-23 19:34:47.062350970 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .parameter
  
      .prologue
+     .line 3145
      iput-object p1, p0, Lcom/android/server/am/ActivityManagerService$5;->this$0:Lcom/android/server/am/ActivityManagerService;
  
      iput-object p2, p0, Lcom/android/server/am/ActivityManagerService$5;->val$cur:Lcom/android/server/am/ActivityRecord;
***************
*** 50,59 ****
--- 51,62 ----
      .locals 6
  
      .prologue
+     .line 3148
      iget-object v2, p0, Lcom/android/server/am/ActivityManagerService$5;->this$0:Lcom/android/server/am/ActivityManagerService;
  
      monitor-enter v2
  
+     .line 3149
      :try_start_0
      new-instance v0, Lcom/android/server/am/LaunchWarningWindow;
  
***************
*** 67,75 ****
--- 70,80 ----
  
      invoke-direct {v0, v1, v3, v4}, Lcom/android/server/am/LaunchWarningWindow;-><init>(Landroid/content/Context;Lcom/android/server/am/ActivityRecord;Lcom/android/server/am/ActivityRecord;)V
  
+     .line 3150
      .local v0, d:Landroid/app/Dialog;
      invoke-virtual {v0}, Landroid/app/Dialog;->show()V
  
+     .line 3151
      iget-object v1, p0, Lcom/android/server/am/ActivityManagerService$5;->this$0:Lcom/android/server/am/ActivityManagerService;
  
      iget-object v1, v1, Lcom/android/server/am/ActivityManagerService;->mHandler:Landroid/os/Handler;
***************
*** 82,91 ****
--- 87,99 ----
  
      invoke-virtual {v1, v3, v4, v5}, Landroid/os/Handler;->postDelayed(Ljava/lang/Runnable;J)Z
  
+     .line 3160
      monitor-exit v2
  
+     .line 3161
      return-void
  
+     .line 3160
      .end local v0           #d:Landroid/app/Dialog;
      :catchall_0
      move-exception v1
