*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityManagerService$11.smali	2012-03-23 19:35:01.182608451 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityManagerService$11.smali.original	2012-03-23 19:34:47.064351005 -0500
***************
*** 43,48 ****
--- 43,49 ----
      .parameter
  
      .prologue
+     .line 7587
      iput-object p1, p0, Lcom/android/server/am/ActivityManagerService$11;->this$0:Lcom/android/server/am/ActivityManagerService;
  
      iput-object p3, p0, Lcom/android/server/am/ActivityManagerService$11;->val$report:Ljava/lang/String;
***************
*** 70,90 ****
--- 71,95 ----
      .prologue
      const/4 v12, 0x0
  
+     .line 7590
      iget-object v8, p0, Lcom/android/server/am/ActivityManagerService$11;->val$report:Ljava/lang/String;
  
      if-eqz v8, :cond_0
  
+     .line 7591
      iget-object v8, p0, Lcom/android/server/am/ActivityManagerService$11;->val$sb:Ljava/lang/StringBuilder;
  
      iget-object v9, p0, Lcom/android/server/am/ActivityManagerService$11;->val$report:Ljava/lang/String;
  
      invoke-virtual {v8, v9}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 7593
      :cond_0
      iget-object v8, p0, Lcom/android/server/am/ActivityManagerService$11;->val$logFile:Ljava/io/File;
  
      if-eqz v8, :cond_1
  
+     .line 7595
      :try_start_0
      iget-object v8, p0, Lcom/android/server/am/ActivityManagerService$11;->val$sb:Ljava/lang/StringBuilder;
  
***************
*** 102,107 ****
--- 107,113 ----
      :try_end_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 7600
      :cond_1
      :goto_0
      iget-object v8, p0, Lcom/android/server/am/ActivityManagerService$11;->val$crashInfo:Landroid/app/ApplicationErrorReport$CrashInfo;
***************
*** 114,119 ****
--- 120,126 ----
  
      if-eqz v8, :cond_2
  
+     .line 7601
      iget-object v8, p0, Lcom/android/server/am/ActivityManagerService$11;->val$sb:Ljava/lang/StringBuilder;
  
      iget-object v9, p0, Lcom/android/server/am/ActivityManagerService$11;->val$crashInfo:Landroid/app/ApplicationErrorReport$CrashInfo;
***************
*** 122,127 ****
--- 129,135 ----
  
      invoke-virtual {v8, v9}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 7604
      :cond_2
      new-instance v8, Ljava/lang/StringBuilder;
  
***************
*** 143,148 ****
--- 151,157 ----
  
      move-result-object v7
  
+     .line 7605
      .local v7, setting:Ljava/lang/String;
      iget-object v8, p0, Lcom/android/server/am/ActivityManagerService$11;->this$0:Lcom/android/server/am/ActivityManagerService;
  
***************
*** 156,172 ****
--- 165,185 ----
  
      move-result v4
  
+     .line 7606
      .local v4, lines:I
      if-lez v4, :cond_3
  
+     .line 7607
      iget-object v8, p0, Lcom/android/server/am/ActivityManagerService$11;->val$sb:Ljava/lang/StringBuilder;
  
      const-string v9, "\n"
  
      invoke-virtual {v8, v9}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 7610
      const/4 v2, 0x0
  
+     .line 7612
      .local v2, input:Ljava/io/InputStreamReader;
      :try_start_1
      new-instance v8, Ljava/lang/ProcessBuilder;
***************
*** 258,263 ****
--- 271,277 ----
  
      move-result-object v5
  
+     .line 7616
      .local v5, logcat:Ljava/lang/Process;
      :try_start_2
      invoke-virtual {v5}, Ljava/lang/Process;->getOutputStream()Ljava/io/OutputStream;
***************
*** 269,274 ****
--- 283,289 ----
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
      .catch Ljava/io/IOException; {:try_start_2 .. :try_end_2} :catch_6
  
+     .line 7617
      :goto_1
      :try_start_3
      invoke-virtual {v5}, Ljava/lang/Process;->getErrorStream()Ljava/io/InputStream;
***************
*** 280,285 ****
--- 295,301 ----
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
      .catch Ljava/io/IOException; {:try_start_3 .. :try_end_3} :catch_5
  
+     .line 7618
      :goto_2
      :try_start_4
      new-instance v3, Ljava/io/InputStreamReader;
***************
*** 293,298 ****
--- 309,315 ----
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
      .catch Ljava/io/IOException; {:try_start_4 .. :try_end_4} :catch_4
  
+     .line 7621
      .end local v2           #input:Ljava/io/InputStreamReader;
      .local v3, input:Ljava/io/InputStreamReader;
      const/16 v8, 0x2000
***************
*** 300,305 ****
--- 317,323 ----
      :try_start_5
      new-array v0, v8, [C
  
+     .line 7622
      .local v0, buf:[C
      :goto_3
      invoke-virtual {v3, v0}, Ljava/io/InputStreamReader;->read([C)I
***************
*** 320,325 ****
--- 338,344 ----
  
      goto :goto_3
  
+     .line 7623
      .end local v0           #buf:[C
      .end local v6           #num:I
      :catch_0
***************
*** 327,332 ****
--- 346,352 ----
  
      move-object v2, v3
  
+     .line 7624
      .end local v3           #input:Ljava/io/InputStreamReader;
      .end local v5           #logcat:Ljava/lang/Process;
      .local v1, e:Ljava/io/IOException;
***************
*** 341,346 ****
--- 361,367 ----
      :try_end_6
      .catchall {:try_start_6 .. :try_end_6} :catchall_0
  
+     .line 7626
      if-eqz v2, :cond_3
  
      :try_start_7
***************
*** 348,353 ****
--- 369,375 ----
      :try_end_7
      .catch Ljava/io/IOException; {:try_start_7 .. :try_end_7} :catch_3
  
+     .line 7630
      .end local v1           #e:Ljava/io/IOException;
      .end local v2           #input:Ljava/io/InputStreamReader;
      :cond_3
***************
*** 364,376 ****
--- 386,401 ----
  
      invoke-virtual {v8, v9, v10}, Landroid/os/DropBoxManager;->addText(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 7631
      return-void
  
+     .line 7596
      .end local v4           #lines:I
      .end local v7           #setting:Ljava/lang/String;
      :catch_1
      move-exception v1
  
+     .line 7597
      .restart local v1       #e:Ljava/io/IOException;
      const-string v8, "ActivityManager"
  
***************
*** 398,403 ****
--- 423,429 ----
  
      goto/16 :goto_0
  
+     .line 7626
      .end local v1           #e:Ljava/io/IOException;
      .restart local v2       #input:Ljava/io/InputStreamReader;
      .restart local v4       #lines:I
***************
*** 453,470 ****
--- 479,499 ----
      .restart local v2       #input:Ljava/io/InputStreamReader;
      goto :goto_6
  
+     .line 7623
      .end local v5           #logcat:Ljava/lang/Process;
      :catch_4
      move-exception v1
  
      goto :goto_4
  
+     .line 7617
      .restart local v5       #logcat:Ljava/lang/Process;
      :catch_5
      move-exception v8
  
      goto :goto_2
  
+     .line 7616
      :catch_6
      move-exception v8
  
