*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityManagerService$9.smali	2012-03-23 19:35:01.700617896 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityManagerService$9.smali.original	2012-03-23 19:34:47.070351113 -0500
***************
*** 34,39 ****
--- 34,40 ----
      .parameter
  
      .prologue
+     .line 7381
      iput-object p1, p0, Lcom/android/server/am/ActivityManagerService$9;->this$0:Lcom/android/server/am/ActivityManagerService;
  
      iput-object p3, p0, Lcom/android/server/am/ActivityManagerService$9;->val$sb:Ljava/lang/StringBuilder;
***************
*** 53,62 ****
--- 54,65 ----
      .locals 5
  
      .prologue
+     .line 7385
      iget-object v2, p0, Lcom/android/server/am/ActivityManagerService$9;->val$sb:Ljava/lang/StringBuilder;
  
      monitor-enter v2
  
+     .line 7386
      :try_start_0
      iget-object v1, p0, Lcom/android/server/am/ActivityManagerService$9;->val$sb:Ljava/lang/StringBuilder;
  
***************
*** 64,69 ****
--- 67,73 ----
  
      move-result-object v0
  
+     .line 7387
      .local v0, report:Ljava/lang/String;
      iget-object v1, p0, Lcom/android/server/am/ActivityManagerService$9;->val$sb:Ljava/lang/StringBuilder;
  
***************
*** 77,105 ****
--- 81,115 ----
  
      invoke-virtual {v1, v3, v4}, Ljava/lang/StringBuilder;->delete(II)Ljava/lang/StringBuilder;
  
+     .line 7388
      iget-object v1, p0, Lcom/android/server/am/ActivityManagerService$9;->val$sb:Ljava/lang/StringBuilder;
  
      invoke-virtual {v1}, Ljava/lang/StringBuilder;->trimToSize()V
  
+     .line 7389
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 7390
      invoke-virtual {v0}, Ljava/lang/String;->length()I
  
      move-result v1
  
      if-eqz v1, :cond_0
  
+     .line 7391
      iget-object v1, p0, Lcom/android/server/am/ActivityManagerService$9;->val$dbox:Landroid/os/DropBoxManager;
  
      iget-object v2, p0, Lcom/android/server/am/ActivityManagerService$9;->val$dropboxTag:Ljava/lang/String;
  
      invoke-virtual {v1, v2, v0}, Landroid/os/DropBoxManager;->addText(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 7393
      :cond_0
      return-void
  
+     .line 7389
      .end local v0           #report:Ljava/lang/String;
      :catchall_0
      move-exception v1
