*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityStack$ActivityState.smali	2012-03-23 19:35:00.736600317 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ActivityStack$ActivityState.smali.original	2012-03-23 19:34:47.060350933 -0500
***************
*** 60,65 ****
--- 60,66 ----
  
      const/4 v3, 0x0
  
+     .line 127
      new-instance v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      const-string v1, "INITIALIZING"
***************
*** 68,73 ****
--- 69,75 ----
  
      sput-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->INITIALIZING:Lcom/android/server/am/ActivityStack$ActivityState;
  
+     .line 128
      new-instance v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      const-string v1, "RESUMED"
***************
*** 76,81 ****
--- 78,84 ----
  
      sput-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->RESUMED:Lcom/android/server/am/ActivityStack$ActivityState;
  
+     .line 129
      new-instance v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      const-string v1, "PAUSING"
***************
*** 84,89 ****
--- 87,93 ----
  
      sput-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->PAUSING:Lcom/android/server/am/ActivityStack$ActivityState;
  
+     .line 130
      new-instance v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      const-string v1, "PAUSED"
***************
*** 92,97 ****
--- 96,102 ----
  
      sput-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->PAUSED:Lcom/android/server/am/ActivityStack$ActivityState;
  
+     .line 131
      new-instance v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      const-string v1, "STOPPING"
***************
*** 100,105 ****
--- 105,111 ----
  
      sput-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->STOPPING:Lcom/android/server/am/ActivityStack$ActivityState;
  
+     .line 132
      new-instance v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      const-string v1, "STOPPED"
***************
*** 110,115 ****
--- 116,122 ----
  
      sput-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->STOPPED:Lcom/android/server/am/ActivityStack$ActivityState;
  
+     .line 133
      new-instance v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      const-string v1, "FINISHING"
***************
*** 120,125 ****
--- 127,133 ----
  
      sput-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->FINISHING:Lcom/android/server/am/ActivityStack$ActivityState;
  
+     .line 134
      new-instance v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      const-string v1, "DESTROYING"
***************
*** 130,135 ****
--- 138,144 ----
  
      sput-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->DESTROYING:Lcom/android/server/am/ActivityStack$ActivityState;
  
+     .line 135
      new-instance v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      const-string v1, "DESTROYED"
***************
*** 140,145 ****
--- 149,155 ----
  
      sput-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->DESTROYED:Lcom/android/server/am/ActivityStack$ActivityState;
  
+     .line 126
      const/16 v0, 0x9
  
      new-array v0, v0, [Lcom/android/server/am/ActivityStack$ActivityState;
***************
*** 204,209 ****
--- 214,220 ----
      .end annotation
  
      .prologue
+     .line 126
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 214,219 ****
--- 225,231 ----
      .parameter "name"
  
      .prologue
+     .line 126
      const-class v0, Lcom/android/server/am/ActivityStack$ActivityState;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 229,234 ****
--- 241,247 ----
      .locals 1
  
      .prologue
+     .line 126
      sget-object v0, Lcom/android/server/am/ActivityStack$ActivityState;->$VALUES:[Lcom/android/server/am/ActivityStack$ActivityState;
  
      invoke-virtual {v0}, [Lcom/android/server/am/ActivityStack$ActivityState;->clone()Ljava/lang/Object;
