*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/AppErrorDialog.smali	2012-03-23 19:35:03.821656574 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/AppErrorDialog.smali.original	2012-03-23 19:34:47.091351497 -0500
***************
*** 33,55 ****
--- 33,61 ----
  
      const/4 v5, 0x0
  
+     .line 43
      invoke-direct {p0, p1}, Lcom/android/server/am/BaseErrorDialog;-><init>(Landroid/content/Context;)V
  
+     .line 87
      new-instance v2, Lcom/android/server/am/AppErrorDialog$1;
  
      invoke-direct {v2, p0}, Lcom/android/server/am/AppErrorDialog$1;-><init>(Lcom/android/server/am/AppErrorDialog;)V
  
      iput-object v2, p0, Lcom/android/server/am/AppErrorDialog;->mHandler:Landroid/os/Handler;
  
+     .line 45
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v1
  
+     .line 47
      .local v1, res:Landroid/content/res/Resources;
      iput-object p3, p0, Lcom/android/server/am/AppErrorDialog;->mProc:Lcom/android/server/am/ProcessRecord;
  
+     .line 48
      iput-object p2, p0, Lcom/android/server/am/AppErrorDialog;->mResult:Lcom/android/server/am/AppErrorResult;
  
+     .line 50
      iget-object v2, p3, Lcom/android/server/am/ProcessRecord;->pkgList:Ljava/util/HashSet;
  
      invoke-virtual {v2}, Ljava/util/HashSet;->size()I
***************
*** 71,76 ****
--- 77,83 ----
      .local v0, name:Ljava/lang/CharSequence;
      if-eqz v0, :cond_2
  
+     .line 52
      const v2, 0x10403a7
  
      const/4 v3, 0x2
***************
*** 95,103 ****
--- 102,112 ----
  
      invoke-virtual {p0, v2}, Lcom/android/server/am/AppErrorDialog;->setMessage(Ljava/lang/CharSequence;)V
  
+     .line 62
      :goto_0
      invoke-virtual {p0, v5}, Lcom/android/server/am/AppErrorDialog;->setCancelable(Z)V
  
+     .line 64
      const/4 v2, -0x1
  
      const v3, 0x10403ae
***************
*** 114,123 ****
--- 123,134 ----
  
      invoke-virtual {p0, v2, v3, v4}, Lcom/android/server/am/AppErrorDialog;->setButton(ILjava/lang/CharSequence;Landroid/os/Message;)V
  
+     .line 68
      iget-object v2, p3, Lcom/android/server/am/ProcessRecord;->errorReportReceiver:Landroid/content/ComponentName;
  
      if-eqz v2, :cond_0
  
+     .line 69
      const/4 v2, -0x2
  
      const v3, 0x10403af
***************
*** 134,139 ****
--- 145,151 ----
  
      invoke-virtual {p0, v2, v3, v4}, Lcom/android/server/am/AppErrorDialog;->setButton(ILjava/lang/CharSequence;Landroid/os/Message;)V
  
+     .line 74
      :cond_0
      const v2, 0x10403a6
  
***************
*** 143,148 ****
--- 155,161 ----
  
      invoke-virtual {p0, v2}, Lcom/android/server/am/AppErrorDialog;->setTitle(Ljava/lang/CharSequence;)V
  
+     .line 75
      invoke-virtual {p0}, Lcom/android/server/am/AppErrorDialog;->getWindow()Landroid/view/Window;
  
      move-result-object v2
***************
*** 151,156 ****
--- 164,170 ----
  
      invoke-virtual {v2, v3}, Landroid/view/Window;->addFlags(I)V
  
+     .line 76
      invoke-virtual {p0}, Lcom/android/server/am/AppErrorDialog;->getWindow()Landroid/view/Window;
  
      move-result-object v2
***************
*** 179,188 ****
--- 193,204 ----
  
      invoke-virtual {v2, v3}, Landroid/view/Window;->setTitle(Ljava/lang/CharSequence;)V
  
+     .line 77
      iget-boolean v2, p3, Lcom/android/server/am/ProcessRecord;->persistent:Z
  
      if-eqz v2, :cond_1
  
+     .line 78
      invoke-virtual {p0}, Lcom/android/server/am/AppErrorDialog;->getWindow()Landroid/view/Window;
  
      move-result-object v2
***************
*** 191,196 ****
--- 207,213 ----
  
      invoke-virtual {v2, v3}, Landroid/view/Window;->setType(I)V
  
+     .line 82
      :cond_1
      iget-object v2, p0, Lcom/android/server/am/AppErrorDialog;->mHandler:Landroid/os/Handler;
  
***************
*** 204,215 ****
--- 221,235 ----
  
      invoke-virtual {v2, v3, v4, v5}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 85
      return-void
  
+     .line 56
      .end local v0           #name:Ljava/lang/CharSequence;
      :cond_2
      iget-object v0, p3, Lcom/android/server/am/ProcessRecord;->processName:Ljava/lang/String;
  
+     .line 57
      .restart local v0       #name:Ljava/lang/CharSequence;
      const v2, 0x10403a8
  
***************
*** 235,240 ****
--- 255,261 ----
      .parameter "x0"
  
      .prologue
+     .line 29
      iget-object v0, p0, Lcom/android/server/am/AppErrorDialog;->mProc:Lcom/android/server/am/ProcessRecord;
  
      return-object v0
***************
*** 245,250 ****
--- 266,272 ----
      .parameter "x0"
  
      .prologue
+     .line 29
      iget-object v0, p0, Lcom/android/server/am/AppErrorDialog;->mResult:Lcom/android/server/am/AppErrorResult;
  
      return-object v0
