*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/AppWaitingForDebuggerDialog.smali	2012-03-23 19:35:01.741618644 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/AppWaitingForDebuggerDialog.smali.original	2012-03-23 19:34:47.071351131 -0500
***************
*** 21,38 ****
--- 21,43 ----
      .parameter "app"
  
      .prologue
+     .line 31
      invoke-direct {p0, p2}, Lcom/android/server/am/BaseErrorDialog;-><init>(Landroid/content/Context;)V
  
+     .line 61
      new-instance v1, Lcom/android/server/am/AppWaitingForDebuggerDialog$1;
  
      invoke-direct {v1, p0}, Lcom/android/server/am/AppWaitingForDebuggerDialog$1;-><init>(Lcom/android/server/am/AppWaitingForDebuggerDialog;)V
  
      iput-object v1, p0, Lcom/android/server/am/AppWaitingForDebuggerDialog;->mHandler:Landroid/os/Handler;
  
+     .line 32
      iput-object p1, p0, Lcom/android/server/am/AppWaitingForDebuggerDialog;->mService:Lcom/android/server/am/ActivityManagerService;
  
+     .line 33
      iput-object p3, p0, Lcom/android/server/am/AppWaitingForDebuggerDialog;->mProc:Lcom/android/server/am/ProcessRecord;
  
+     .line 34
      invoke-virtual {p2}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
  
      move-result-object v1
***************
*** 45,58 ****
--- 50,66 ----
  
      iput-object v1, p0, Lcom/android/server/am/AppWaitingForDebuggerDialog;->mAppName:Ljava/lang/CharSequence;
  
+     .line 36
      const/4 v1, 0x0
  
      invoke-virtual {p0, v1}, Lcom/android/server/am/AppWaitingForDebuggerDialog;->setCancelable(Z)V
  
+     .line 38
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 39
      .local v0, text:Ljava/lang/StringBuilder;
      iget-object v1, p0, Lcom/android/server/am/AppWaitingForDebuggerDialog;->mAppName:Ljava/lang/CharSequence;
  
***************
*** 66,102 ****
--- 74,118 ----
  
      if-lez v1, :cond_0
  
+     .line 40
      const-string v1, "Application "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 41
      iget-object v1, p0, Lcom/android/server/am/AppWaitingForDebuggerDialog;->mAppName:Ljava/lang/CharSequence;
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/CharSequence;)Ljava/lang/StringBuilder;
  
+     .line 42
      const-string v1, " (process "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 43
      iget-object v1, p3, Lcom/android/server/am/ProcessRecord;->processName:Ljava/lang/String;
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 44
      const-string v1, ")"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 50
      :goto_0
      const-string v1, " is waiting for the debugger to attach."
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 52
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
  
      invoke-virtual {p0, v1}, Lcom/android/server/am/AppWaitingForDebuggerDialog;->setMessage(Ljava/lang/CharSequence;)V
  
+     .line 53
      const/4 v1, -0x1
  
      const-string v2, "Force Close"
***************
*** 111,120 ****
--- 127,138 ----
  
      invoke-virtual {p0, v1, v2, v3}, Lcom/android/server/am/AppWaitingForDebuggerDialog;->setButton(ILjava/lang/CharSequence;Landroid/os/Message;)V
  
+     .line 54
      const-string v1, "Waiting For Debugger"
  
      invoke-virtual {p0, v1}, Lcom/android/server/am/AppWaitingForDebuggerDialog;->setTitle(Ljava/lang/CharSequence;)V
  
+     .line 55
      invoke-virtual {p0}, Lcom/android/server/am/AppWaitingForDebuggerDialog;->getWindow()Landroid/view/Window;
  
      move-result-object v1
***************
*** 143,155 ****
--- 161,176 ----
  
      invoke-virtual {v1, v2}, Landroid/view/Window;->setTitle(Ljava/lang/CharSequence;)V
  
+     .line 56
      return-void
  
+     .line 46
      :cond_0
      const-string v1, "Process "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 47
      iget-object v1, p3, Lcom/android/server/am/ProcessRecord;->processName:Ljava/lang/String;
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 163,167 ****
--- 184,189 ----
      .locals 0
  
      .prologue
+     .line 59
      return-void
  .end method
