*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/BaseErrorDialog$1.smali	2012-03-23 19:35:01.066606335 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/BaseErrorDialog$1.smali.original	2012-03-23 19:34:47.062350970 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 70
      iput-object p1, p0, Lcom/android/server/am/BaseErrorDialog$1;->this$0:Lcom/android/server/am/BaseErrorDialog;
  
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
***************
*** 38,47 ****
--- 39,50 ----
      .parameter "msg"
  
      .prologue
+     .line 72
      iget v0, p1, Landroid/os/Message;->what:I
  
      if-nez v0, :cond_0
  
+     .line 73
      iget-object v0, p0, Lcom/android/server/am/BaseErrorDialog$1;->this$0:Lcom/android/server/am/BaseErrorDialog;
  
      const/4 v1, 0x0
***************
*** 49,54 ****
--- 52,58 ----
      #setter for: Lcom/android/server/am/BaseErrorDialog;->mConsuming:Z
      invoke-static {v0, v1}, Lcom/android/server/am/BaseErrorDialog;->access$002(Lcom/android/server/am/BaseErrorDialog;Z)Z
  
+     .line 74
      iget-object v0, p0, Lcom/android/server/am/BaseErrorDialog$1;->this$0:Lcom/android/server/am/BaseErrorDialog;
  
      const/4 v1, 0x1
***************
*** 56,61 ****
--- 60,66 ----
      #calls: Lcom/android/server/am/BaseErrorDialog;->setEnabled(Z)V
      invoke-static {v0, v1}, Lcom/android/server/am/BaseErrorDialog;->access$100(Lcom/android/server/am/BaseErrorDialog;Z)V
  
+     .line 76
      :cond_0
      return-void
  .end method
