*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/CoreSettingsObserver.smali	2012-03-23 19:35:01.540614978 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/CoreSettingsObserver.smali.original	2012-03-23 19:34:47.068351077 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .locals 3
  
      .prologue
+     .line 37
      const-class v0, Lcom/android/server/am/CoreSettingsObserver;
  
      invoke-virtual {v0}, Ljava/lang/Class;->getSimpleName()Ljava/lang/String;
***************
*** 38,49 ****
--- 39,52 ----
  
      sput-object v0, Lcom/android/server/am/CoreSettingsObserver;->LOG_TAG:Ljava/lang/String;
  
+     .line 40
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      sput-object v0, Lcom/android/server/am/CoreSettingsObserver;->sCoreSettingToTypeMap:Ljava/util/Map;
  
+     .line 43
      sget-object v0, Lcom/android/server/am/CoreSettingsObserver;->sCoreSettingToTypeMap:Ljava/util/Map;
  
      const-string v1, "long_press_timeout"
***************
*** 52,57 ****
--- 55,61 ----
  
      invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 45
      return-void
  .end method
  
***************
*** 60,81 ****
--- 64,91 ----
      .parameter "activityManagerService"
  
      .prologue
+     .line 52
      iget-object v0, p1, Lcom/android/server/am/ActivityManagerService;->mHandler:Landroid/os/Handler;
  
      invoke-direct {p0, v0}, Landroid/database/ContentObserver;-><init>(Landroid/os/Handler;)V
  
+     .line 47
      new-instance v0, Landroid/os/Bundle;
  
      invoke-direct {v0}, Landroid/os/Bundle;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/am/CoreSettingsObserver;->mCoreSettings:Landroid/os/Bundle;
  
+     .line 53
      iput-object p1, p0, Lcom/android/server/am/CoreSettingsObserver;->mActivityManagerService:Lcom/android/server/am/ActivityManagerService;
  
+     .line 54
      invoke-direct {p0}, Lcom/android/server/am/CoreSettingsObserver;->beginObserveCoreSettings()V
  
+     .line 55
      invoke-direct {p0}, Lcom/android/server/am/CoreSettingsObserver;->sendCoreSettings()V
  
+     .line 56
      return-void
  .end method
  
***************
*** 83,88 ****
--- 93,99 ----
      .locals 5
  
      .prologue
+     .line 75
      sget-object v3, Lcom/android/server/am/CoreSettingsObserver;->sCoreSettingToTypeMap:Ljava/util/Map;
  
      invoke-interface {v3}, Ljava/util/Map;->keySet()Ljava/util/Set;
***************
*** 107,117 ****
--- 118,130 ----
  
      check-cast v1, Ljava/lang/String;
  
+     .line 76
      .local v1, setting:Ljava/lang/String;
      invoke-static {v1}, Landroid/provider/Settings$Secure;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v2
  
+     .line 77
      .local v2, uri:Landroid/net/Uri;
      iget-object v3, p0, Lcom/android/server/am/CoreSettingsObserver;->mActivityManagerService:Lcom/android/server/am/ActivityManagerService;
  
***************
*** 127,132 ****
--- 140,146 ----
  
      goto :goto_0
  
+     .line 80
      .end local v1           #setting:Ljava/lang/String;
      .end local v2           #uri:Landroid/net/Uri;
      :cond_0
***************
*** 138,147 ****
--- 152,163 ----
      .parameter "snapshot"
  
      .prologue
+     .line 83
      iget-object v8, p0, Lcom/android/server/am/CoreSettingsObserver;->mActivityManagerService:Lcom/android/server/am/ActivityManagerService;
  
      iget-object v0, v8, Lcom/android/server/am/ActivityManagerService;->mContext:Landroid/content/Context;
  
+     .line 84
      .local v0, context:Landroid/content/Context;
      sget-object v8, Lcom/android/server/am/CoreSettingsObserver;->sCoreSettingToTypeMap:Ljava/util/Map;
  
***************
*** 168,173 ****
--- 184,190 ----
  
      check-cast v1, Ljava/util/Map$Entry;
  
+     .line 85
      .local v1, entry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/lang/Class<*>;>;"
      invoke-interface {v1}, Ljava/util/Map$Entry;->getKey()Ljava/lang/Object;
  
***************
*** 175,180 ****
--- 192,198 ----
  
      check-cast v3, Ljava/lang/String;
  
+     .line 86
      .local v3, setting:Ljava/lang/String;
      invoke-interface {v1}, Ljava/util/Map$Entry;->getValue()Ljava/lang/Object;
  
***************
*** 182,193 ****
--- 200,213 ----
  
      check-cast v5, Ljava/lang/Class;
  
+     .line 88
      .local v5, type:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
      :try_start_0
      const-class v8, Ljava/lang/String;
  
      if-ne v5, v8, :cond_1
  
+     .line 89
      invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v8
***************
*** 196,201 ****
--- 216,222 ----
  
      move-result-object v6
  
+     .line 91
      .local v6, value:Ljava/lang/String;
      invoke-virtual {p1, v3, v6}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
      :try_end_0
***************
*** 203,212 ****
--- 224,235 ----
  
      goto :goto_0
  
+     .line 105
      .end local v6           #value:Ljava/lang/String;
      :catch_0
      move-exception v4
  
+     .line 106
      .local v4, snfe:Landroid/provider/Settings$SettingNotFoundException;
      sget-object v8, Lcom/android/server/am/CoreSettingsObserver;->LOG_TAG:Ljava/lang/String;
  
***************
*** 238,243 ****
--- 261,267 ----
  
      goto :goto_0
  
+     .line 92
      .end local v4           #snfe:Landroid/provider/Settings$SettingNotFoundException;
      :cond_1
      :try_start_1
***************
*** 245,250 ****
--- 269,275 ----
  
      if-ne v5, v8, :cond_2
  
+     .line 93
      invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v8
***************
*** 253,269 ****
--- 278,297 ----
  
      move-result v6
  
+     .line 95
      .local v6, value:I
      invoke-virtual {p1, v3, v6}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
      goto :goto_0
  
+     .line 96
      .end local v6           #value:I
      :cond_2
      sget-object v8, Ljava/lang/Float;->TYPE:Ljava/lang/Class;
  
      if-ne v5, v8, :cond_3
  
+     .line 97
      invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v8
***************
*** 272,288 ****
--- 300,319 ----
  
      move-result v6
  
+     .line 99
      .local v6, value:F
      invoke-virtual {p1, v3, v6}, Landroid/os/Bundle;->putFloat(Ljava/lang/String;F)V
  
      goto :goto_0
  
+     .line 100
      .end local v6           #value:F
      :cond_3
      sget-object v8, Ljava/lang/Long;->TYPE:Ljava/lang/Class;
  
      if-ne v5, v8, :cond_0
  
+     .line 101
      invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v8
***************
*** 291,296 ****
--- 322,328 ----
  
      move-result-wide v6
  
+     .line 103
      .local v6, value:J
      invoke-virtual {p1, v3, v6, v7}, Landroid/os/Bundle;->putLong(Ljava/lang/String;J)V
      :try_end_1
***************
*** 298,303 ****
--- 330,336 ----
  
      goto :goto_0
  
+     .line 109
      .end local v1           #entry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/lang/Class<*>;>;"
      .end local v3           #setting:Ljava/lang/String;
      .end local v5           #type:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
***************
*** 310,325 ****
--- 343,361 ----
      .locals 2
  
      .prologue
+     .line 70
      iget-object v0, p0, Lcom/android/server/am/CoreSettingsObserver;->mCoreSettings:Landroid/os/Bundle;
  
      invoke-direct {p0, v0}, Lcom/android/server/am/CoreSettingsObserver;->populateCoreSettings(Landroid/os/Bundle;)V
  
+     .line 71
      iget-object v0, p0, Lcom/android/server/am/CoreSettingsObserver;->mActivityManagerService:Lcom/android/server/am/ActivityManagerService;
  
      iget-object v1, p0, Lcom/android/server/am/CoreSettingsObserver;->mCoreSettings:Landroid/os/Bundle;
  
      invoke-virtual {v0, v1}, Lcom/android/server/am/ActivityManagerService;->onCoreSettingsChange(Landroid/os/Bundle;)V
  
+     .line 72
      return-void
  .end method
  
***************
*** 329,334 ****
--- 365,371 ----
      .locals 1
  
      .prologue
+     .line 59
      iget-object v0, p0, Lcom/android/server/am/CoreSettingsObserver;->mCoreSettings:Landroid/os/Bundle;
  
      invoke-virtual {v0}, Landroid/os/Bundle;->clone()Ljava/lang/Object;
***************
*** 345,361 ****
--- 382,403 ----
      .parameter "selfChange"
  
      .prologue
+     .line 64
      iget-object v1, p0, Lcom/android/server/am/CoreSettingsObserver;->mActivityManagerService:Lcom/android/server/am/ActivityManagerService;
  
      monitor-enter v1
  
+     .line 65
      :try_start_0
      invoke-direct {p0}, Lcom/android/server/am/CoreSettingsObserver;->sendCoreSettings()V
  
+     .line 66
      monitor-exit v1
  
+     .line 67
      return-void
  
+     .line 66
      :catchall_0
      move-exception v0
  
