*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/EventLogTags.smali	2012-03-23 19:35:00.978604731 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/EventLogTags.smali.original	2012-03-23 19:34:47.062350970 -0500
***************
*** 88,93 ****
--- 88,94 ----
      .locals 0
  
      .prologue
+     .line 11
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 100,105 ****
--- 101,107 ----
      .parameter "time"
  
      .prologue
+     .line 179
      const/16 v0, 0x7539
  
      const/4 v1, 0x3
***************
*** 128,133 ****
--- 130,136 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 180
      return-void
  .end method
  
***************
*** 139,144 ****
--- 142,148 ----
      .parameter "reason"
  
      .prologue
+     .line 175
      const/16 v0, 0x7538
  
      const/4 v1, 0x4
***************
*** 171,176 ****
--- 175,181 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 176
      return-void
  .end method
  
***************
*** 182,187 ****
--- 187,193 ----
      .parameter "app"
  
      .prologue
+     .line 243
      const/16 v0, 0x7549
  
      const/4 v1, 0x4
***************
*** 214,219 ****
--- 220,226 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 244
      return-void
  .end method
  
***************
*** 225,230 ****
--- 232,238 ----
      .parameter "broadcastfilter"
  
      .prologue
+     .line 239
      const/16 v0, 0x7548
  
      const/4 v1, 0x4
***************
*** 261,266 ****
--- 269,275 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 240
      return-void
  .end method
  
***************
*** 275,280 ****
--- 284,290 ----
      .parameter "line"
  
      .prologue
+     .line 279
      const/16 v0, 0x7557
  
      const/4 v1, 0x7
***************
*** 323,328 ****
--- 333,339 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 280
      return-void
  .end method
  
***************
*** 337,342 ****
--- 348,354 ----
      .parameter "flags"
  
      .prologue
+     .line 163
      const/16 v0, 0x7535
  
      const/4 v1, 0x7
***************
*** 385,390 ****
--- 397,403 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 164
      return-void
  .end method
  
***************
*** 396,401 ****
--- 409,415 ----
      .parameter "pid"
  
      .prologue
+     .line 247
      const/16 v0, 0x754e
  
      const/4 v1, 0x4
***************
*** 428,433 ****
--- 442,448 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 248
      return-void
  .end method
  
***************
*** 436,445 ****
--- 451,462 ----
      .parameter "taskId"
  
      .prologue
+     .line 159
      const/16 v0, 0x7534
  
      invoke-static {v0, p0}, Landroid/util/EventLog;->writeEvent(II)I
  
+     .line 160
      return-void
  .end method
  
***************
*** 451,456 ****
--- 468,474 ----
      .parameter "reason"
  
      .prologue
+     .line 215
      const/16 v0, 0x7542
  
      const/4 v1, 0x4
***************
*** 483,488 ****
--- 501,507 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 216
      return-void
  .end method
  
***************
*** 493,498 ****
--- 512,518 ----
      .parameter "pid"
  
      .prologue
+     .line 251
      const/16 v0, 0x754f
  
      const/4 v1, 0x3
***************
*** 521,526 ****
--- 541,547 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 252
      return-void
  .end method
  
***************
*** 529,538 ****
--- 550,561 ----
      .parameter "pid"
  
      .prologue
+     .line 259
      const/16 v0, 0x7551
  
      invoke-static {v0, p0}, Landroid/util/EventLog;->writeEvent(II)I
  
+     .line 260
      return-void
  .end method
  
***************
*** 543,548 ****
--- 566,572 ----
      .parameter "currentlyPausing"
  
      .prologue
+     .line 191
      const/16 v0, 0x753c
  
      const/4 v1, 0x3
***************
*** 567,572 ****
--- 591,597 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 192
      return-void
  .end method
  
***************
*** 578,583 ****
--- 603,609 ----
      .parameter "reason"
  
      .prologue
+     .line 147
      const/16 v0, 0x7531
  
      const/4 v1, 0x4
***************
*** 610,615 ****
--- 636,642 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 148
      return-void
  .end method
  
***************
*** 621,626 ****
--- 648,654 ----
      .parameter "reason"
  
      .prologue
+     .line 235
      const/16 v0, 0x7547
  
      const/4 v1, 0x4
***************
*** 653,658 ****
--- 681,687 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 236
      return-void
  .end method
  
***************
*** 661,670 ****
--- 690,701 ----
      .parameter "numProcesses"
  
      .prologue
+     .line 211
      const/16 v0, 0x7541
  
      invoke-static {v0, p0}, Landroid/util/EventLog;->writeEvent(II)I
  
+     .line 212
      return-void
  .end method
  
***************
*** 679,684 ****
--- 710,716 ----
      .parameter "flags"
  
      .prologue
+     .line 155
      const/16 v0, 0x7533
  
      const/4 v1, 0x7
***************
*** 727,732 ****
--- 759,765 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 156
      return-void
  .end method
  
***************
*** 735,744 ****
--- 768,779 ----
      .parameter "componentName"
  
      .prologue
+     .line 227
      const/16 v0, 0x7545
  
      invoke-static {v0, p0}, Landroid/util/EventLog;->writeEvent(ILjava/lang/String;)I
  
+     .line 228
      return-void
  .end method
  
***************
*** 747,756 ****
--- 782,793 ----
      .parameter "componentName"
  
      .prologue
+     .line 231
      const/16 v0, 0x7546
  
      invoke-static {v0, p0}, Landroid/util/EventLog;->writeEvent(ILjava/lang/String;)I
  
+     .line 232
      return-void
  .end method
  
***************
*** 760,765 ****
--- 797,803 ----
      .parameter "componentName"
  
      .prologue
+     .line 195
      const/16 v0, 0x753d
  
      const/4 v1, 0x2
***************
*** 780,785 ****
--- 818,824 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 196
      return-void
  .end method
  
***************
*** 789,794 ****
--- 828,834 ----
      .parameter "processName"
  
      .prologue
+     .line 203
      const/16 v0, 0x753f
  
      const/4 v1, 0x2
***************
*** 809,814 ****
--- 849,855 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 204
      return-void
  .end method
  
***************
*** 818,823 ****
--- 859,865 ----
      .parameter "processName"
  
      .prologue
+     .line 183
      const/16 v0, 0x753a
  
      const/4 v1, 0x2
***************
*** 838,843 ****
--- 880,886 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 184
      return-void
  .end method
  
***************
*** 847,852 ****
--- 890,896 ----
      .parameter "processName"
  
      .prologue
+     .line 187
      const/16 v0, 0x753b
  
      const/4 v1, 0x2
***************
*** 867,872 ****
--- 911,917 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 188
      return-void
  .end method
  
***************
*** 876,881 ****
--- 921,927 ----
      .parameter "processName"
  
      .prologue
+     .line 207
      const/16 v0, 0x7540
  
      const/4 v1, 0x2
***************
*** 896,901 ****
--- 942,948 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 208
      return-void
  .end method
  
***************
*** 908,913 ****
--- 955,961 ----
      .parameter "component"
  
      .prologue
+     .line 199
      const/16 v0, 0x753e
  
      const/4 v1, 0x5
***************
*** 944,949 ****
--- 992,998 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 200
      return-void
  .end method
  
***************
*** 953,958 ****
--- 1002,1008 ----
      .parameter "pid"
  
      .prologue
+     .line 255
      const/16 v0, 0x7550
  
      const/4 v1, 0x2
***************
*** 973,978 ****
--- 1023,1029 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 256
      return-void
  .end method
  
***************
*** 983,988 ****
--- 1034,1040 ----
      .parameter "processName"
  
      .prologue
+     .line 275
      const/16 v0, 0x7555
  
      const/4 v1, 0x3
***************
*** 1011,1016 ****
--- 1063,1069 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 276
      return-void
  .end method
  
***************
*** 1021,1026 ****
--- 1074,1080 ----
      .parameter "name"
  
      .prologue
+     .line 271
      const/16 v0, 0x7554
  
      const/4 v1, 0x3
***************
*** 1045,1050 ****
--- 1099,1105 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 272
      return-void
  .end method
  
***************
*** 1055,1060 ****
--- 1110,1116 ----
      .parameter "componentName"
  
      .prologue
+     .line 223
      const/16 v0, 0x7544
  
      const/4 v1, 0x3
***************
*** 1083,1088 ****
--- 1139,1145 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 224
      return-void
  .end method
  
***************
*** 1093,1098 ****
--- 1150,1156 ----
      .parameter "componentName"
  
      .prologue
+     .line 219
      const/16 v0, 0x7543
  
      const/4 v1, 0x3
***************
*** 1121,1126 ****
--- 1179,1185 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 220
      return-void
  .end method
  
***************
*** 1131,1136 ****
--- 1190,1196 ----
      .parameter "componentName"
  
      .prologue
+     .line 167
      const/16 v0, 0x7536
  
      const/4 v1, 0x3
***************
*** 1159,1164 ****
--- 1219,1225 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 168
      return-void
  .end method
  
***************
*** 1169,1174 ****
--- 1230,1236 ----
      .parameter "componentName"
  
      .prologue
+     .line 171
      const/16 v0, 0x7537
  
      const/4 v1, 0x3
***************
*** 1197,1202 ****
--- 1259,1265 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 172
      return-void
  .end method
  
***************
*** 1206,1211 ****
--- 1269,1275 ----
      .parameter "time"
  
      .prologue
+     .line 267
      const/16 v0, 0x7553
  
      const/4 v1, 0x2
***************
*** 1226,1231 ****
--- 1290,1296 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 268
      return-void
  .end method
  
***************
*** 1236,1241 ****
--- 1301,1307 ----
      .parameter "pid"
  
      .prologue
+     .line 263
      const/16 v0, 0x7552
  
      const/4 v1, 0x3
***************
*** 1264,1269 ****
--- 1330,1336 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 264
      return-void
  .end method
  
***************
*** 1272,1281 ****
--- 1339,1350 ----
      .parameter "task"
  
      .prologue
+     .line 151
      const/16 v0, 0x7532
  
      invoke-static {v0, p0}, Landroid/util/EventLog;->writeEvent(II)I
  
+     .line 152
      return-void
  .end method
  
***************
*** 1288,1293 ****
--- 1357,1363 ----
      .parameter "message"
  
      .prologue
+     .line 283
      const/16 v0, 0x7558
  
      const/4 v1, 0x5
***************
*** 1324,1329 ****
--- 1394,1400 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 284
      return-void
  .end method
  
***************
*** 1332,1341 ****
--- 1403,1414 ----
      .parameter "time"
  
      .prologue
+     .line 139
      const/16 v0, 0xbe0
  
      invoke-static {v0, p0, p1}, Landroid/util/EventLog;->writeEvent(IJ)I
  
+     .line 140
      return-void
  .end method
  
***************
*** 1344,1353 ****
--- 1417,1428 ----
      .parameter "time"
  
      .prologue
+     .line 143
      const/16 v0, 0xbea
  
      invoke-static {v0, p0, p1}, Landroid/util/EventLog;->writeEvent(IJ)I
  
+     .line 144
      return-void
  .end method
  
***************
*** 1356,1365 ****
--- 1431,1442 ----
      .parameter "configMask"
  
      .prologue
+     .line 131
      const/16 v0, 0xa9f
  
      invoke-static {v0, p0}, Landroid/util/EventLog;->writeEvent(II)I
  
+     .line 132
      return-void
  .end method
  
***************
*** 1373,1378 ****
--- 1450,1456 ----
      .parameter "softirq"
  
      .prologue
+     .line 135
      const/16 v0, 0xaa1
  
      const/4 v1, 0x6
***************
*** 1429,1433 ****
--- 1507,1512 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 136
      return-void
  .end method
