*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/LaunchWarningWindow.smali	2012-03-23 19:35:01.319610950 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/LaunchWarningWindow.smali.original	2012-03-23 19:34:47.065351023 -0500
***************
*** 17,28 ****
--- 17,31 ----
  
      const/4 v8, 0x1
  
+     .line 31
      const v3, 0x10302f9
  
      invoke-direct {p0, p1, v3}, Landroid/app/Dialog;-><init>(Landroid/content/Context;I)V
  
+     .line 33
      invoke-virtual {p0, v5}, Lcom/android/server/am/LaunchWarningWindow;->requestWindowFeature(I)Z
  
+     .line 34
      invoke-virtual {p0}, Lcom/android/server/am/LaunchWarningWindow;->getWindow()Landroid/view/Window;
  
      move-result-object v3
***************
*** 31,36 ****
--- 34,40 ----
  
      invoke-virtual {v3, v4}, Landroid/view/Window;->setType(I)V
  
+     .line 35
      invoke-virtual {p0}, Lcom/android/server/am/LaunchWarningWindow;->getWindow()Landroid/view/Window;
  
      move-result-object v3
***************
*** 39,48 ****
--- 43,54 ----
  
      invoke-virtual {v3, v4}, Landroid/view/Window;->addFlags(I)V
  
+     .line 38
      const v3, 0x109005b
  
      invoke-virtual {p0, v3}, Lcom/android/server/am/LaunchWarningWindow;->setContentView(I)V
  
+     .line 39
      const v3, 0x10403b1
  
      invoke-virtual {p1, v3}, Landroid/content/Context;->getText(I)Ljava/lang/CharSequence;
***************
*** 51,60 ****
--- 57,68 ----
  
      invoke-virtual {p0, v3}, Lcom/android/server/am/LaunchWarningWindow;->setTitle(Ljava/lang/CharSequence;)V
  
+     .line 41
      new-instance v1, Landroid/util/TypedValue;
  
      invoke-direct {v1}, Landroid/util/TypedValue;-><init>()V
  
+     .line 42
      .local v1, out:Landroid/util/TypedValue;
      invoke-virtual {p0}, Lcom/android/server/am/LaunchWarningWindow;->getContext()Landroid/content/Context;
  
***************
*** 68,73 ****
--- 76,82 ----
  
      invoke-virtual {v3, v4, v1, v8}, Landroid/content/res/Resources$Theme;->resolveAttribute(ILandroid/util/TypedValue;Z)Z
  
+     .line 43
      invoke-virtual {p0}, Lcom/android/server/am/LaunchWarningWindow;->getWindow()Landroid/view/Window;
  
      move-result-object v3
***************
*** 76,81 ****
--- 85,91 ----
  
      invoke-virtual {v3, v5, v4}, Landroid/view/Window;->setFeatureDrawableResource(II)V
  
+     .line 45
      const v3, 0x10202b5
  
      invoke-virtual {p0, v3}, Lcom/android/server/am/LaunchWarningWindow;->findViewById(I)Landroid/view/View;
***************
*** 84,89 ****
--- 94,100 ----
  
      check-cast v0, Landroid/widget/ImageView;
  
+     .line 46
      .local v0, icon:Landroid/widget/ImageView;
      iget-object v3, p3, Lcom/android/server/am/ActivityRecord;->info:Landroid/content/pm/ActivityInfo;
  
***************
*** 99,104 ****
--- 110,116 ----
  
      invoke-virtual {v0, v3}, Landroid/widget/ImageView;->setImageDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 47
      const v3, 0x10202b6
  
      invoke-virtual {p0, v3}, Lcom/android/server/am/LaunchWarningWindow;->findViewById(I)Landroid/view/View;
***************
*** 107,112 ****
--- 119,125 ----
  
      check-cast v2, Landroid/widget/TextView;
  
+     .line 48
      .local v2, text:Landroid/widget/TextView;
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
***************
*** 140,145 ****
--- 153,159 ----
  
      invoke-virtual {v2, v3}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 50
      const v3, 0x10202b7
  
      invoke-virtual {p0, v3}, Lcom/android/server/am/LaunchWarningWindow;->findViewById(I)Landroid/view/View;
***************
*** 149,154 ****
--- 163,169 ----
      .end local v0           #icon:Landroid/widget/ImageView;
      check-cast v0, Landroid/widget/ImageView;
  
+     .line 51
      .restart local v0       #icon:Landroid/widget/ImageView;
      iget-object v3, p2, Lcom/android/server/am/ActivityRecord;->info:Landroid/content/pm/ActivityInfo;
  
***************
*** 164,169 ****
--- 179,185 ----
  
      invoke-virtual {v0, v3}, Landroid/widget/ImageView;->setImageDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 52
      const v3, 0x10202b8
  
      invoke-virtual {p0, v3}, Lcom/android/server/am/LaunchWarningWindow;->findViewById(I)Landroid/view/View;
***************
*** 173,178 ****
--- 189,195 ----
      .end local v2           #text:Landroid/widget/TextView;
      check-cast v2, Landroid/widget/TextView;
  
+     .line 53
      .restart local v2       #text:Landroid/widget/TextView;
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
***************
*** 206,210 ****
--- 223,228 ----
  
      invoke-virtual {v2, v3}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 55
      return-void
  .end method
