*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ServiceRecord$1.smali	2012-03-23 19:35:00.876602871 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ServiceRecord$1.smali.original	2012-03-23 19:34:47.061350952 -0500
***************
*** 42,47 ****
--- 42,48 ----
      .parameter
  
      .prologue
+     .line 351
      iput-object p1, p0, Lcom/android/server/am/ServiceRecord$1;->this$0:Lcom/android/server/am/ServiceRecord;
  
      iput-object p2, p0, Lcom/android/server/am/ServiceRecord$1;->val$localPackageName:Ljava/lang/String;
***************
*** 69,92 ****
--- 70,98 ----
  
      const/4 v9, 0x1
  
+     .line 353
      invoke-static {}, Landroid/app/NotificationManager;->getService()Landroid/app/INotificationManager;
  
      move-result-object v0
  
      check-cast v0, Lcom/android/server/NotificationManagerService;
  
+     .line 355
      .local v0, nm:Lcom/android/server/NotificationManagerService;
      if-nez v0, :cond_0
  
+     .line 372
      :goto_0
      return-void
  
+     .line 359
      :cond_0
      const/4 v1, 0x1
  
      :try_start_0
      new-array v7, v1, [I
  
+     .line 360
      .local v7, outId:[I
      iget-object v1, p0, Lcom/android/server/am/ServiceRecord$1;->val$localPackageName:Ljava/lang/String;
  
***************
*** 106,115 ****
--- 112,123 ----
  
      goto :goto_0
  
+     .line 362
      .end local v7           #outId:[I
      :catch_0
      move-exception v8
  
+     .line 363
      .local v8, e:Ljava/lang/RuntimeException;
      const-string v1, "ActivityManager"
  
***************
*** 117,122 ****
--- 125,131 ----
  
      invoke-static {v1, v2, v8}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 367
      iget-object v1, p0, Lcom/android/server/am/ServiceRecord$1;->this$0:Lcom/android/server/am/ServiceRecord;
  
      iget-object v1, v1, Lcom/android/server/am/ServiceRecord;->ams:Lcom/android/server/am/ActivityManagerService;
***************
*** 135,140 ****
--- 144,150 ----
  
      invoke-virtual/range {v1 .. v6}, Lcom/android/server/am/ActivityManagerService;->setServiceForeground(Landroid/content/ComponentName;Landroid/os/IBinder;ILandroid/app/Notification;Z)V
  
+     .line 369
      iget-object v1, p0, Lcom/android/server/am/ServiceRecord$1;->this$0:Lcom/android/server/am/ServiceRecord;
  
      iget-object v1, v1, Lcom/android/server/am/ServiceRecord;->ams:Lcom/android/server/am/ActivityManagerService;
