*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ServiceRecord.smali	2012-03-23 19:35:02.913640015 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/ServiceRecord.smali.original	2012-03-23 19:34:47.082351332 -0500
***************
*** 146,247 ****
--- 146,270 ----
      .parameter "restarter"
  
      .prologue
+     .line 274
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 72
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->bindings:Ljava/util/HashMap;
  
+     .line 75
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->connections:Ljava/util/HashMap;
  
+     .line 151
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->deliveredStarts:Ljava/util/ArrayList;
  
+     .line 153
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->pendingStarts:Ljava/util/ArrayList;
  
+     .line 275
      iput-object p1, p0, Lcom/android/server/am/ServiceRecord;->ams:Lcom/android/server/am/ActivityManagerService;
  
+     .line 276
      iput-object p2, p0, Lcom/android/server/am/ServiceRecord;->stats:Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;
  
+     .line 277
      iput-object p3, p0, Lcom/android/server/am/ServiceRecord;->name:Landroid/content/ComponentName;
  
+     .line 278
      invoke-virtual {p3}, Landroid/content/ComponentName;->flattenToShortString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->shortName:Ljava/lang/String;
  
+     .line 279
      iput-object p4, p0, Lcom/android/server/am/ServiceRecord;->intent:Landroid/content/Intent$FilterComparison;
  
+     .line 280
      iput-object p5, p0, Lcom/android/server/am/ServiceRecord;->serviceInfo:Landroid/content/pm/ServiceInfo;
  
+     .line 281
      iget-object v0, p5, Landroid/content/pm/ServiceInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->appInfo:Landroid/content/pm/ApplicationInfo;
  
+     .line 282
      iget-object v0, p5, Landroid/content/pm/ServiceInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iget-object v0, v0, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->packageName:Ljava/lang/String;
  
+     .line 283
      iget-object v0, p5, Landroid/content/pm/ServiceInfo;->processName:Ljava/lang/String;
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->processName:Ljava/lang/String;
  
+     .line 284
      iget-object v0, p5, Landroid/content/pm/ServiceInfo;->permission:Ljava/lang/String;
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->permission:Ljava/lang/String;
  
+     .line 285
      iget-object v0, p5, Landroid/content/pm/ServiceInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iget-object v0, v0, Landroid/content/pm/ApplicationInfo;->sourceDir:Ljava/lang/String;
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->baseDir:Ljava/lang/String;
  
+     .line 286
      iget-object v0, p5, Landroid/content/pm/ServiceInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iget-object v0, v0, Landroid/content/pm/ApplicationInfo;->publicSourceDir:Ljava/lang/String;
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->resDir:Ljava/lang/String;
  
+     .line 287
      iget-object v0, p5, Landroid/content/pm/ServiceInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iget-object v0, v0, Landroid/content/pm/ApplicationInfo;->dataDir:Ljava/lang/String;
  
      iput-object v0, p0, Lcom/android/server/am/ServiceRecord;->dataDir:Ljava/lang/String;
  
+     .line 288
      iget-boolean v0, p5, Landroid/content/pm/ServiceInfo;->exported:Z
  
      iput-boolean v0, p0, Lcom/android/server/am/ServiceRecord;->exported:Z
  
+     .line 289
      iput-object p6, p0, Lcom/android/server/am/ServiceRecord;->restarter:Ljava/lang/Runnable;
  
+     .line 290
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Lcom/android/server/am/ServiceRecord;->createTime:J
  
+     .line 291
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Lcom/android/server/am/ServiceRecord;->lastActivity:J
  
+     .line 292
      return-void
  .end method
  
***************
*** 251,265 ****
--- 274,292 ----
      .locals 4
  
      .prologue
+     .line 378
      iget v2, p0, Lcom/android/server/am/ServiceRecord;->foregroundId:I
  
      if-eqz v2, :cond_0
  
+     .line 381
      iget-object v1, p0, Lcom/android/server/am/ServiceRecord;->packageName:Ljava/lang/String;
  
+     .line 382
      .local v1, localPackageName:Ljava/lang/String;
      iget v0, p0, Lcom/android/server/am/ServiceRecord;->foregroundId:I
  
+     .line 383
      .local v0, localForegroundId:I
      iget-object v2, p0, Lcom/android/server/am/ServiceRecord;->ams:Lcom/android/server/am/ActivityManagerService;
  
***************
*** 271,276 ****
--- 298,304 ----
  
      invoke-virtual {v2, v3}, Landroid/os/Handler;->post(Ljava/lang/Runnable;)Z
  
+     .line 399
      .end local v0           #localForegroundId:I
      .end local v1           #localPackageName:Ljava/lang/String;
      :cond_0
***************
*** 281,286 ****
--- 309,315 ----
      .locals 2
  
      .prologue
+     .line 402
      iget-object v1, p0, Lcom/android/server/am/ServiceRecord;->deliveredStarts:Ljava/util/ArrayList;
  
      invoke-virtual {v1}, Ljava/util/ArrayList;->size()I
***************
*** 293,298 ****
--- 322,328 ----
      :goto_0
      if-ltz v0, :cond_0
  
+     .line 403
      iget-object v1, p0, Lcom/android/server/am/ServiceRecord;->deliveredStarts:Ljava/util/ArrayList;
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 303,317 ****
--- 333,350 ----
  
      invoke-virtual {v1}, Lcom/android/server/am/ServiceRecord$StartItem;->removeUriPermissionsLocked()V
  
+     .line 402
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_0
  
+     .line 405
      :cond_0
      iget-object v1, p0, Lcom/android/server/am/ServiceRecord;->deliveredStarts:Ljava/util/ArrayList;
  
      invoke-virtual {v1}, Ljava/util/ArrayList;->clear()V
  
+     .line 406
      return-void
  .end method
  
***************
*** 321,326 ****
--- 354,360 ----
      .parameter "prefix"
  
      .prologue
+     .line 194
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "intent={"
***************
*** 329,334 ****
--- 363,369 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 195
      move-object/from16 v0, p0
  
      iget-object v3, v0, Lcom/android/server/am/ServiceRecord;->intent:Landroid/content/Intent$FilterComparison;
***************
*** 351,362 ****
--- 386,399 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 196
      const/16 v3, 0x7d
  
      move-object/from16 v0, p1
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(C)V
  
+     .line 197
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "packageName="
***************
*** 373,378 ****
--- 410,416 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 198
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "processName="
***************
*** 389,400 ****
--- 427,440 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 199
      move-object/from16 v0, p0
  
      iget-object v3, v0, Lcom/android/server/am/ServiceRecord;->permission:Ljava/lang/String;
  
      if-eqz v3, :cond_0
  
+     .line 200
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "permission="
***************
*** 411,426 ****
--- 451,469 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 202
      :cond_0
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v7
  
+     .line 203
      .local v7, now:J
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v20
  
+     .line 204
      .local v20, nowReal:J
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 438,443 ****
--- 481,487 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 205
      move-object/from16 v0, p0
  
      iget-object v3, v0, Lcom/android/server/am/ServiceRecord;->resDir:Ljava/lang/String;
***************
*** 452,457 ****
--- 496,502 ----
  
      if-nez v3, :cond_1
  
+     .line 206
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "resDir="
***************
*** 468,473 ****
--- 513,519 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 208
      :cond_1
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 485,490 ****
--- 531,537 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 209
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "app="
***************
*** 501,506 ****
--- 548,554 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 210
      move-object/from16 v0, p0
  
      iget-boolean v3, v0, Lcom/android/server/am/ServiceRecord;->isForeground:Z
***************
*** 513,518 ****
--- 561,567 ----
  
      if-eqz v3, :cond_3
  
+     .line 211
      :cond_2
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 530,535 ****
--- 579,585 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 212
      const-string v3, " foregroundId="
  
      move-object/from16 v0, p1
***************
*** 544,549 ****
--- 594,600 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 213
      const-string v3, " foregroundNoti="
  
      move-object/from16 v0, p1
***************
*** 558,563 ****
--- 609,615 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 215
      :cond_3
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 567,572 ****
--- 619,625 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 216
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/server/am/ServiceRecord;->createTime:J
***************
*** 577,588 ****
--- 630,643 ----
  
      invoke-static {v3, v4, v0, v1, v2}, Landroid/util/TimeUtils;->formatDuration(JJLjava/io/PrintWriter;)V
  
+     .line 217
      const-string v3, " lastActivity="
  
      move-object/from16 v0, p1
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 218
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/server/am/ServiceRecord;->lastActivity:J
***************
*** 591,602 ****
--- 646,659 ----
  
      invoke-static {v3, v4, v7, v8, v0}, Landroid/util/TimeUtils;->formatDuration(JJLjava/io/PrintWriter;)V
  
+     .line 219
      const-string v3, ""
  
      move-object/from16 v0, p1
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 220
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "executingStart="
***************
*** 605,610 ****
--- 662,668 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 221
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/server/am/ServiceRecord;->executingStart:J
***************
*** 613,624 ****
--- 671,684 ----
  
      invoke-static {v3, v4, v7, v8, v0}, Landroid/util/TimeUtils;->formatDuration(JJLjava/io/PrintWriter;)V
  
+     .line 222
      const-string v3, " restartTime="
  
      move-object/from16 v0, p1
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 223
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/server/am/ServiceRecord;->restartTime:J
***************
*** 627,638 ****
--- 687,700 ----
  
      invoke-static {v3, v4, v7, v8, v0}, Landroid/util/TimeUtils;->formatDuration(JJLjava/io/PrintWriter;)V
  
+     .line 224
      const-string v3, ""
  
      move-object/from16 v0, p1
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 225
      move-object/from16 v0, p0
  
      iget-boolean v3, v0, Lcom/android/server/am/ServiceRecord;->startRequested:Z
***************
*** 645,650 ****
--- 707,713 ----
  
      if-eqz v3, :cond_5
  
+     .line 226
      :cond_4
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 662,667 ****
--- 725,731 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 227
      const-string v3, " stopIfKilled="
  
      move-object/from16 v0, p1
***************
*** 676,681 ****
--- 740,746 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 228
      const-string v3, " callStart="
  
      move-object/from16 v0, p1
***************
*** 690,695 ****
--- 755,761 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 229
      const-string v3, " lastStartId="
  
      move-object/from16 v0, p1
***************
*** 704,709 ****
--- 770,776 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(I)V
  
+     .line 231
      :cond_5
      move-object/from16 v0, p0
  
***************
*** 743,748 ****
--- 810,816 ----
  
      if-eqz v3, :cond_7
  
+     .line 233
      :cond_6
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 760,765 ****
--- 828,834 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 234
      const-string v3, " restartCount="
  
      move-object/from16 v0, p1
***************
*** 774,785 ****
--- 843,856 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 235
      const-string v3, " restartDelay="
  
      move-object/from16 v0, p1
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 236
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/server/am/ServiceRecord;->restartDelay:J
***************
*** 788,799 ****
--- 859,872 ----
  
      invoke-static {v3, v4, v7, v8, v0}, Landroid/util/TimeUtils;->formatDuration(JJLjava/io/PrintWriter;)V
  
+     .line 237
      const-string v3, " nextRestartTime="
  
      move-object/from16 v0, p1
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 238
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/server/am/ServiceRecord;->nextRestartTime:J
***************
*** 802,807 ****
--- 875,881 ----
  
      invoke-static {v3, v4, v7, v8, v0}, Landroid/util/TimeUtils;->formatDuration(JJLjava/io/PrintWriter;)V
  
+     .line 239
      const-string v3, " crashCount="
  
      move-object/from16 v0, p1
***************
*** 816,821 ****
--- 890,896 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(I)V
  
+     .line 241
      :cond_7
      move-object/from16 v0, p0
  
***************
*** 827,832 ****
--- 902,908 ----
  
      if-lez v3, :cond_8
  
+     .line 242
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "Delivered Starts:"
***************
*** 835,840 ****
--- 911,917 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 243
      move-object/from16 v0, p0
  
      iget-object v6, v0, Lcom/android/server/am/ServiceRecord;->deliveredStarts:Ljava/util/ArrayList;
***************
*** 847,852 ****
--- 924,930 ----
  
      invoke-virtual/range {v3 .. v8}, Lcom/android/server/am/ServiceRecord;->dumpStartList(Ljava/io/PrintWriter;Ljava/lang/String;Ljava/util/List;J)V
  
+     .line 245
      :cond_8
      move-object/from16 v0, p0
  
***************
*** 858,863 ****
--- 936,942 ----
  
      if-lez v3, :cond_9
  
+     .line 246
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "Pending Starts:"
***************
*** 866,871 ****
--- 945,951 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 247
      move-object/from16 v0, p0
  
      iget-object v12, v0, Lcom/android/server/am/ServiceRecord;->pendingStarts:Ljava/util/ArrayList;
***************
*** 880,885 ****
--- 960,966 ----
  
      invoke-virtual/range {v9 .. v14}, Lcom/android/server/am/ServiceRecord;->dumpStartList(Ljava/io/PrintWriter;Ljava/lang/String;Ljava/util/List;J)V
  
+     .line 249
      :cond_9
      move-object/from16 v0, p0
  
***************
*** 891,896 ****
--- 972,978 ----
  
      if-lez v3, :cond_a
  
+     .line 250
      move-object/from16 v0, p0
  
      iget-object v3, v0, Lcom/android/server/am/ServiceRecord;->bindings:Ljava/util/HashMap;
***************
*** 903,908 ****
--- 985,991 ----
  
      move-result-object v18
  
+     .line 251
      .local v18, it:Ljava/util/Iterator;,"Ljava/util/Iterator<Lcom/android/server/am/IntentBindRecord;>;"
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 912,917 ****
--- 995,1001 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 252
      :goto_0
      invoke-interface/range {v18 .. v18}, Ljava/util/Iterator;->hasNext()Z
  
***************
*** 919,930 ****
--- 1003,1016 ----
  
      if-eqz v3, :cond_a
  
+     .line 253
      invoke-interface/range {v18 .. v18}, Ljava/util/Iterator;->next()Ljava/lang/Object;
  
      move-result-object v15
  
      check-cast v15, Lcom/android/server/am/IntentBindRecord;
  
+     .line 254
      .local v15, b:Lcom/android/server/am/IntentBindRecord;
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 934,939 ****
--- 1020,1026 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 255
      invoke-static {v15}, Ljava/lang/System;->identityHashCode(Ljava/lang/Object;)I
  
      move-result v3
***************
*** 946,957 ****
--- 1033,1046 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 256
      const-string v3, "}:"
  
      move-object/from16 v0, p1
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 257
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 978,983 ****
--- 1067,1073 ----
  
      goto :goto_0
  
+     .line 260
      .end local v15           #b:Lcom/android/server/am/IntentBindRecord;
      .end local v18           #it:Ljava/util/Iterator;,"Ljava/util/Iterator<Lcom/android/server/am/IntentBindRecord;>;"
      :cond_a
***************
*** 991,996 ****
--- 1081,1087 ----
  
      if-lez v3, :cond_c
  
+     .line 261
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "All Connections:"
***************
*** 999,1004 ****
--- 1090,1096 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 262
      move-object/from16 v0, p0
  
      iget-object v3, v0, Lcom/android/server/am/ServiceRecord;->connections:Ljava/util/HashMap;
***************
*** 1011,1016 ****
--- 1103,1109 ----
  
      move-result-object v19
  
+     .line 263
      .local v19, it:Ljava/util/Iterator;,"Ljava/util/Iterator<Ljava/util/ArrayList<Lcom/android/server/am/ConnectionRecord;>;>;"
      :cond_b
      invoke-interface/range {v19 .. v19}, Ljava/util/Iterator;->hasNext()Z
***************
*** 1019,1030 ****
--- 1112,1125 ----
  
      if-eqz v3, :cond_c
  
+     .line 264
      invoke-interface/range {v19 .. v19}, Ljava/util/Iterator;->next()Ljava/lang/Object;
  
      move-result-object v16
  
      check-cast v16, Ljava/util/ArrayList;
  
+     .line 265
      .local v16, c:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/server/am/ConnectionRecord;>;"
      const/16 v17, 0x0
  
***************
*** 1038,1043 ****
--- 1133,1139 ----
  
      if-ge v0, v3, :cond_b
  
+     .line 266
      invoke-virtual/range {p1 .. p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "  "
***************
*** 1054,1063 ****
--- 1150,1161 ----
  
      invoke-virtual {v0, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 265
      add-int/lit8 v17, v17, 0x1
  
      goto :goto_1
  
+     .line 270
      .end local v16           #c:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/server/am/ConnectionRecord;>;"
      .end local v17           #i:I
      .end local v19           #it:Ljava/util/Iterator;,"Ljava/util/Iterator<Ljava/util/ArrayList<Lcom/android/server/am/ConnectionRecord;>;>;"
***************
*** 1084,1094 ****
--- 1182,1194 ----
      .end annotation
  
      .prologue
+     .line 157
      .local p3, list:Ljava/util/List;,"Ljava/util/List<Lcom/android/server/am/ServiceRecord$StartItem;>;"
      invoke-interface {p3}, Ljava/util/List;->size()I
  
      move-result v0
  
+     .line 158
      .local v0, N:I
      const/4 v1, 0x0
  
***************
*** 1096,1107 ****
--- 1196,1209 ----
      :goto_0
      if-ge v1, v0, :cond_7
  
+     .line 159
      invoke-interface {p3, v1}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v2
  
      check-cast v2, Lcom/android/server/am/ServiceRecord$StartItem;
  
+     .line 160
      .local v2, si:Lcom/android/server/am/ServiceRecord$StartItem;
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 1111,1116 ****
--- 1213,1219 ----
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 161
      const-string v3, " id="
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1119,1143 ****
--- 1222,1251 ----
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 162
      const-wide/16 v3, 0x0
  
      cmp-long v3, p4, v3
  
      if-eqz v3, :cond_0
  
+     .line 163
      const-string v3, " dur="
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 164
      iget-wide v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->deliveredTime:J
  
      invoke-static {v3, v4, p4, p5, p1}, Landroid/util/TimeUtils;->formatDuration(JJLjava/io/PrintWriter;)V
  
+     .line 166
      :cond_0
      iget v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->deliveryCount:I
  
      if-eqz v3, :cond_1
  
+     .line 167
      const-string v3, " dc="
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1146,1156 ****
--- 1254,1266 ----
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 169
      :cond_1
      iget v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->doneExecutingCount:I
  
      if-eqz v3, :cond_2
  
+     .line 170
      const-string v3, " dxc="
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1159,1175 ****
--- 1269,1288 ----
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 172
      :cond_2
      const-string v3, ""
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 173
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "  intent="
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 174
      iget-object v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->intent:Landroid/content/Intent;
  
      if-eqz v3, :cond_6
***************
*** 1182,1225 ****
--- 1295,1346 ----
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 176
      :goto_1
      iget v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->targetPermissionUid:I
  
      if-ltz v3, :cond_3
  
+     .line 177
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "  targetPermissionUid="
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 178
      iget v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->targetPermissionUid:I
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->println(I)V
  
+     .line 180
      :cond_3
      iget-object v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->uriPermissions:Lcom/android/server/am/UriPermissionOwner;
  
      if-eqz v3, :cond_5
  
+     .line 181
      iget-object v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->uriPermissions:Lcom/android/server/am/UriPermissionOwner;
  
      iget-object v3, v3, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
      if-eqz v3, :cond_4
  
+     .line 182
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "  readUriPermissions="
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 183
      iget-object v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->uriPermissions:Lcom/android/server/am/UriPermissionOwner;
  
      iget-object v3, v3, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 185
      :cond_4
      iget-object v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->uriPermissions:Lcom/android/server/am/UriPermissionOwner;
  
***************
*** 1227,1249 ****
--- 1348,1374 ----
  
      if-eqz v3, :cond_5
  
+     .line 186
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v3, "  writeUriPermissions="
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 187
      iget-object v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->uriPermissions:Lcom/android/server/am/UriPermissionOwner;
  
      iget-object v3, v3, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {p1, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 158
      :cond_5
      add-int/lit8 v1, v1, 0x1
  
      goto/16 :goto_0
  
+     .line 175
      :cond_6
      const-string v3, "null"
  
***************
*** 1251,1256 ****
--- 1376,1382 ----
  
      goto :goto_1
  
+     .line 191
      .end local v2           #si:Lcom/android/server/am/ServiceRecord$StartItem;
      :cond_7
      return-void
***************
*** 1262,1273 ****
--- 1388,1401 ----
      .parameter "remove"
  
      .prologue
+     .line 318
      iget-object v3, p0, Lcom/android/server/am/ServiceRecord;->deliveredStarts:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v0
  
+     .line 319
      .local v0, N:I
      const/4 v1, 0x0
  
***************
*** 1275,1280 ****
--- 1403,1409 ----
      :goto_0
      if-ge v1, v0, :cond_2
  
+     .line 320
      iget-object v3, p0, Lcom/android/server/am/ServiceRecord;->deliveredStarts:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1283,1310 ****
--- 1412,1444 ----
  
      check-cast v2, Lcom/android/server/am/ServiceRecord$StartItem;
  
+     .line 321
      .local v2, si:Lcom/android/server/am/ServiceRecord$StartItem;
      iget v3, v2, Lcom/android/server/am/ServiceRecord$StartItem;->id:I
  
      if-ne v3, p1, :cond_1
  
+     .line 322
      if-eqz p2, :cond_0
  
      iget-object v3, p0, Lcom/android/server/am/ServiceRecord;->deliveredStarts:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v1}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 327
      .end local v2           #si:Lcom/android/server/am/ServiceRecord$StartItem;
      :cond_0
      :goto_1
      return-object v2
  
+     .line 319
      .restart local v2       #si:Lcom/android/server/am/ServiceRecord$StartItem;
      :cond_1
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 327
      .end local v2           #si:Lcom/android/server/am/ServiceRecord$StartItem;
      :cond_2
      const/4 v2, 0x0
***************
*** 1316,1321 ****
--- 1450,1456 ----
      .locals 1
  
      .prologue
+     .line 331
      iget v0, p0, Lcom/android/server/am/ServiceRecord;->lastStartId:I
  
      return v0
***************
*** 1327,1344 ****
--- 1462,1483 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 335
      iget v0, p0, Lcom/android/server/am/ServiceRecord;->lastStartId:I
  
      add-int/lit8 v0, v0, 0x1
  
      iput v0, p0, Lcom/android/server/am/ServiceRecord;->lastStartId:I
  
+     .line 336
      iget v0, p0, Lcom/android/server/am/ServiceRecord;->lastStartId:I
  
      if-ge v0, v1, :cond_0
  
+     .line 337
      iput v1, p0, Lcom/android/server/am/ServiceRecord;->lastStartId:I
  
+     .line 339
      :cond_0
      iget v0, p0, Lcom/android/server/am/ServiceRecord;->lastStartId:I
  
***************
*** 1349,1363 ****
--- 1488,1505 ----
      .locals 8
  
      .prologue
+     .line 343
      iget-object v0, p0, Lcom/android/server/am/ServiceRecord;->appInfo:Landroid/content/pm/ApplicationInfo;
  
      iget v3, v0, Landroid/content/pm/ApplicationInfo;->uid:I
  
+     .line 344
      .local v3, appUid:I
      iget-object v0, p0, Lcom/android/server/am/ServiceRecord;->app:Lcom/android/server/am/ProcessRecord;
  
      iget v4, v0, Lcom/android/server/am/ProcessRecord;->pid:I
  
+     .line 345
      .local v4, appPid:I
      iget v0, p0, Lcom/android/server/am/ServiceRecord;->foregroundId:I
  
***************
*** 1367,1380 ****
--- 1509,1526 ----
  
      if-eqz v0, :cond_0
  
+     .line 348
      iget-object v2, p0, Lcom/android/server/am/ServiceRecord;->packageName:Ljava/lang/String;
  
+     .line 349
      .local v2, localPackageName:Ljava/lang/String;
      iget v5, p0, Lcom/android/server/am/ServiceRecord;->foregroundId:I
  
+     .line 350
      .local v5, localForegroundId:I
      iget-object v6, p0, Lcom/android/server/am/ServiceRecord;->foregroundNoti:Landroid/app/Notification;
  
+     .line 351
      .local v6, localForegroundNoti:Landroid/app/Notification;
      iget-object v0, p0, Lcom/android/server/am/ServiceRecord;->ams:Lcom/android/server/am/ActivityManagerService;
  
***************
*** 1388,1393 ****
--- 1534,1540 ----
  
      invoke-virtual {v7, v0}, Landroid/os/Handler;->post(Ljava/lang/Runnable;)Z
  
+     .line 375
      .end local v2           #localPackageName:Ljava/lang/String;
      .end local v5           #localForegroundId:I
      .end local v6           #localForegroundNoti:Landroid/app/Notification;
***************
*** 1401,1414 ****
--- 1548,1565 ----
      .prologue
      const-wide/16 v1, 0x0
  
+     .line 312
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/server/am/ServiceRecord;->restartCount:I
  
+     .line 313
      iput-wide v1, p0, Lcom/android/server/am/ServiceRecord;->restartDelay:J
  
+     .line 314
      iput-wide v1, p0, Lcom/android/server/am/ServiceRecord;->restartTime:J
  
+     .line 315
      return-void
  .end method
  
***************
*** 1418,1427 ****
--- 1569,1580 ----
      .parameter "app"
  
      .prologue
+     .line 296
      new-instance v2, Landroid/content/Intent$FilterComparison;
  
      invoke-direct {v2, p1}, Landroid/content/Intent$FilterComparison;-><init>(Landroid/content/Intent;)V
  
+     .line 297
      .local v2, filter:Landroid/content/Intent$FilterComparison;
      iget-object v4, p0, Lcom/android/server/am/ServiceRecord;->bindings:Ljava/util/HashMap;
  
***************
*** 1431,1449 ****
--- 1584,1606 ----
  
      check-cast v3, Lcom/android/server/am/IntentBindRecord;
  
+     .line 298
      .local v3, i:Lcom/android/server/am/IntentBindRecord;
      if-nez v3, :cond_0
  
+     .line 299
      new-instance v3, Lcom/android/server/am/IntentBindRecord;
  
      .end local v3           #i:Lcom/android/server/am/IntentBindRecord;
      invoke-direct {v3, p0, v2}, Lcom/android/server/am/IntentBindRecord;-><init>(Lcom/android/server/am/ServiceRecord;Landroid/content/Intent$FilterComparison;)V
  
+     .line 300
      .restart local v3       #i:Lcom/android/server/am/IntentBindRecord;
      iget-object v4, p0, Lcom/android/server/am/ServiceRecord;->bindings:Ljava/util/HashMap;
  
      invoke-virtual {v4, v2, v3}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 302
      :cond_0
      iget-object v4, v3, Lcom/android/server/am/IntentBindRecord;->apps:Ljava/util/HashMap;
  
***************
*** 1453,1468 ****
--- 1610,1628 ----
  
      check-cast v0, Lcom/android/server/am/AppBindRecord;
  
+     .line 303
      .local v0, a:Lcom/android/server/am/AppBindRecord;
      if-eqz v0, :cond_1
  
      move-object v1, v0
  
+     .line 308
      .end local v0           #a:Lcom/android/server/am/AppBindRecord;
      .local v1, a:Ljava/lang/Object;
      :goto_0
      return-object v1
  
+     .line 306
      .end local v1           #a:Ljava/lang/Object;
      .restart local v0       #a:Lcom/android/server/am/AppBindRecord;
      :cond_1
***************
*** 1471,1476 ****
--- 1631,1637 ----
      .end local v0           #a:Lcom/android/server/am/AppBindRecord;
      invoke-direct {v0, p0, v3, p2}, Lcom/android/server/am/AppBindRecord;-><init>(Lcom/android/server/am/ServiceRecord;Lcom/android/server/am/IntentBindRecord;Lcom/android/server/am/ProcessRecord;)V
  
+     .line 307
      .restart local v0       #a:Lcom/android/server/am/AppBindRecord;
      iget-object v4, v3, Lcom/android/server/am/IntentBindRecord;->apps:Ljava/util/HashMap;
  
***************
*** 1478,1483 ****
--- 1639,1645 ----
  
      move-object v1, v0
  
+     .line 308
      .restart local v1       #a:Ljava/lang/Object;
      goto :goto_0
  .end method
***************
*** 1486,1500 ****
--- 1648,1666 ----
      .locals 3
  
      .prologue
+     .line 409
      iget-object v1, p0, Lcom/android/server/am/ServiceRecord;->stringName:Ljava/lang/String;
  
      if-eqz v1, :cond_0
  
+     .line 410
      iget-object v1, p0, Lcom/android/server/am/ServiceRecord;->stringName:Ljava/lang/String;
  
+     .line 416
      :goto_0
      return-object v1
  
+     .line 412
      :cond_0
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 1502,1507 ****
--- 1668,1674 ----
  
      invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(I)V
  
+     .line 413
      .local v0, sb:Ljava/lang/StringBuilder;
      const-string v1, "ServiceRecord{"
  
***************
*** 1537,1542 ****
--- 1704,1710 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 416
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
