*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/UriPermissionOwner.smali	2012-03-23 19:35:01.870620996 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/UriPermissionOwner.smali.original	2012-03-23 19:34:47.072351149 -0500
***************
*** 48,59 ****
--- 48,63 ----
      .parameter "_owner"
  
      .prologue
+     .line 42
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 43
      iput-object p1, p0, Lcom/android/server/am/UriPermissionOwner;->service:Lcom/android/server/am/ActivityManagerService;
  
+     .line 44
      iput-object p2, p0, Lcom/android/server/am/UriPermissionOwner;->owner:Ljava/lang/Object;
  
+     .line 45
      return-void
  .end method
  
***************
*** 62,71 ****
--- 66,77 ----
      .parameter "token"
  
      .prologue
+     .line 55
      instance-of v0, p0, Lcom/android/server/am/UriPermissionOwner$ExternalToken;
  
      if-eqz v0, :cond_0
  
+     .line 56
      check-cast p0, Lcom/android/server/am/UriPermissionOwner$ExternalToken;
  
      .end local p0
***************
*** 73,78 ****
--- 79,85 ----
  
      move-result-object v0
  
+     .line 58
      :goto_0
      return-object v0
  
***************
*** 90,110 ****
--- 97,121 ----
      .parameter "perm"
  
      .prologue
+     .line 135
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
      if-nez v0, :cond_0
  
+     .line 136
      new-instance v0, Ljava/util/HashSet;
  
      invoke-direct {v0}, Ljava/util/HashSet;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
+     .line 138
      :cond_0
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v0, p1}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 139
      return-void
  .end method
  
***************
*** 113,133 ****
--- 124,148 ----
      .parameter "perm"
  
      .prologue
+     .line 142
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
      if-nez v0, :cond_0
  
+     .line 143
      new-instance v0, Ljava/util/HashSet;
  
      invoke-direct {v0}, Ljava/util/HashSet;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
+     .line 145
      :cond_0
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v0, p1}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 146
      return-void
  .end method
  
***************
*** 135,150 ****
--- 150,168 ----
      .locals 1
  
      .prologue
+     .line 48
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->externalToken:Landroid/os/Binder;
  
      if-nez v0, :cond_0
  
+     .line 49
      new-instance v0, Lcom/android/server/am/UriPermissionOwner$ExternalToken;
  
      invoke-direct {v0, p0}, Lcom/android/server/am/UriPermissionOwner$ExternalToken;-><init>(Lcom/android/server/am/UriPermissionOwner;)V
  
      iput-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->externalToken:Landroid/os/Binder;
  
+     .line 51
      :cond_0
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->externalToken:Landroid/os/Binder;
  
***************
*** 156,165 ****
--- 174,185 ----
      .parameter "perm"
  
      .prologue
+     .line 149
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v0, p1}, Ljava/util/HashSet;->remove(Ljava/lang/Object;)Z
  
+     .line 150
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v0}, Ljava/util/HashSet;->size()I
***************
*** 168,177 ****
--- 188,199 ----
  
      if-nez v0, :cond_0
  
+     .line 151
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
+     .line 153
      :cond_0
      return-void
  .end method
***************
*** 184,189 ****
--- 206,212 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 94
      and-int/lit8 v2, p2, 0x1
  
      if-eqz v2, :cond_3
***************
*** 192,203 ****
--- 215,228 ----
  
      if-eqz v2, :cond_3
  
+     .line 96
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v2}, Ljava/util/HashSet;->iterator()Ljava/util/Iterator;
  
      move-result-object v0
  
+     .line 97
      .local v0, it:Ljava/util/Iterator;,"Ljava/util/Iterator<Lcom/android/server/am/UriPermission;>;"
      :cond_0
      :goto_0
***************
*** 207,218 ****
--- 232,245 ----
  
      if-eqz v2, :cond_2
  
+     .line 98
      invoke-interface {v0}, Ljava/util/Iterator;->next()Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Lcom/android/server/am/UriPermission;
  
+     .line 99
      .local v1, perm:Lcom/android/server/am/UriPermission;
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->uri:Landroid/net/Uri;
  
***************
*** 222,231 ****
--- 249,260 ----
  
      if-eqz v2, :cond_0
  
+     .line 100
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->readOwners:Ljava/util/HashSet;
  
      invoke-virtual {v2, p0}, Ljava/util/HashSet;->remove(Ljava/lang/Object;)Z
  
+     .line 101
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->readOwners:Ljava/util/HashSet;
  
      invoke-virtual {v2}, Ljava/util/HashSet;->size()I
***************
*** 240,260 ****
--- 269,293 ----
  
      if-nez v2, :cond_1
  
+     .line 103
      iget v2, v1, Lcom/android/server/am/UriPermission;->modeFlags:I
  
      and-int/lit8 v2, v2, -0x2
  
      iput v2, v1, Lcom/android/server/am/UriPermission;->modeFlags:I
  
+     .line 104
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->service:Lcom/android/server/am/ActivityManagerService;
  
      invoke-virtual {v2, v1}, Lcom/android/server/am/ActivityManagerService;->removeUriPermissionIfNeededLocked(Lcom/android/server/am/UriPermission;)V
  
+     .line 106
      :cond_1
      invoke-interface {v0}, Ljava/util/Iterator;->remove()V
  
      goto :goto_0
  
+     .line 109
      .end local v1           #perm:Lcom/android/server/am/UriPermission;
      :cond_2
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
***************
*** 265,272 ****
--- 298,307 ----
  
      if-nez v2, :cond_3
  
+     .line 110
      iput-object v3, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
+     .line 113
      .end local v0           #it:Ljava/util/Iterator;,"Ljava/util/Iterator<Lcom/android/server/am/UriPermission;>;"
      :cond_3
      and-int/lit8 v2, p2, 0x2
***************
*** 277,288 ****
--- 312,325 ----
  
      if-eqz v2, :cond_7
  
+     .line 115
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v2}, Ljava/util/HashSet;->iterator()Ljava/util/Iterator;
  
      move-result-object v0
  
+     .line 116
      .restart local v0       #it:Ljava/util/Iterator;,"Ljava/util/Iterator<Lcom/android/server/am/UriPermission;>;"
      :cond_4
      :goto_1
***************
*** 292,303 ****
--- 329,342 ----
  
      if-eqz v2, :cond_6
  
+     .line 117
      invoke-interface {v0}, Ljava/util/Iterator;->next()Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Lcom/android/server/am/UriPermission;
  
+     .line 118
      .restart local v1       #perm:Lcom/android/server/am/UriPermission;
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->uri:Landroid/net/Uri;
  
***************
*** 307,316 ****
--- 346,357 ----
  
      if-eqz v2, :cond_4
  
+     .line 119
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->writeOwners:Ljava/util/HashSet;
  
      invoke-virtual {v2, p0}, Ljava/util/HashSet;->remove(Ljava/lang/Object;)Z
  
+     .line 120
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->writeOwners:Ljava/util/HashSet;
  
      invoke-virtual {v2}, Ljava/util/HashSet;->size()I
***************
*** 325,345 ****
--- 366,390 ----
  
      if-nez v2, :cond_5
  
+     .line 122
      iget v2, v1, Lcom/android/server/am/UriPermission;->modeFlags:I
  
      and-int/lit8 v2, v2, -0x3
  
      iput v2, v1, Lcom/android/server/am/UriPermission;->modeFlags:I
  
+     .line 123
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->service:Lcom/android/server/am/ActivityManagerService;
  
      invoke-virtual {v2, v1}, Lcom/android/server/am/ActivityManagerService;->removeUriPermissionIfNeededLocked(Lcom/android/server/am/UriPermission;)V
  
+     .line 125
      :cond_5
      invoke-interface {v0}, Ljava/util/Iterator;->remove()V
  
      goto :goto_1
  
+     .line 128
      .end local v1           #perm:Lcom/android/server/am/UriPermission;
      :cond_6
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
***************
*** 350,357 ****
--- 395,404 ----
  
      if-nez v2, :cond_7
  
+     .line 129
      iput-object v3, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
+     .line 132
      .end local v0           #it:Ljava/util/Iterator;,"Ljava/util/Iterator<Lcom/android/server/am/UriPermission;>;"
      :cond_7
      return-void
***************
*** 361,370 ****
--- 408,419 ----
      .locals 1
  
      .prologue
+     .line 62
      const/4 v0, 0x3
  
      invoke-virtual {p0, v0}, Lcom/android/server/am/UriPermissionOwner;->removeUriPermissionsLocked(I)V
  
+     .line 64
      return-void
  .end method
  
***************
*** 375,380 ****
--- 424,430 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 67
      and-int/lit8 v2, p1, 0x1
  
      if-eqz v2, :cond_2
***************
*** 383,388 ****
--- 433,439 ----
  
      if-eqz v2, :cond_2
  
+     .line 69
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v2}, Ljava/util/HashSet;->iterator()Ljava/util/Iterator;
***************
*** 404,414 ****
--- 455,467 ----
  
      check-cast v1, Lcom/android/server/am/UriPermission;
  
+     .line 70
      .local v1, perm:Lcom/android/server/am/UriPermission;
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->readOwners:Ljava/util/HashSet;
  
      invoke-virtual {v2, p0}, Ljava/util/HashSet;->remove(Ljava/lang/Object;)Z
  
+     .line 71
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->readOwners:Ljava/util/HashSet;
  
      invoke-virtual {v2}, Ljava/util/HashSet;->size()I
***************
*** 423,444 ****
--- 476,501 ----
  
      if-nez v2, :cond_0
  
+     .line 73
      iget v2, v1, Lcom/android/server/am/UriPermission;->modeFlags:I
  
      and-int/lit8 v2, v2, -0x2
  
      iput v2, v1, Lcom/android/server/am/UriPermission;->modeFlags:I
  
+     .line 74
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->service:Lcom/android/server/am/ActivityManagerService;
  
      invoke-virtual {v2, v1}, Lcom/android/server/am/ActivityManagerService;->removeUriPermissionIfNeededLocked(Lcom/android/server/am/UriPermission;)V
  
      goto :goto_0
  
+     .line 77
      .end local v1           #perm:Lcom/android/server/am/UriPermission;
      :cond_1
      iput-object v3, p0, Lcom/android/server/am/UriPermissionOwner;->readUriPermissions:Ljava/util/HashSet;
  
+     .line 79
      .end local v0           #i$:Ljava/util/Iterator;
      :cond_2
      and-int/lit8 v2, p1, 0x2
***************
*** 449,454 ****
--- 506,512 ----
  
      if-eqz v2, :cond_5
  
+     .line 81
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v2}, Ljava/util/HashSet;->iterator()Ljava/util/Iterator;
***************
*** 470,480 ****
--- 528,540 ----
  
      check-cast v1, Lcom/android/server/am/UriPermission;
  
+     .line 82
      .restart local v1       #perm:Lcom/android/server/am/UriPermission;
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->writeOwners:Ljava/util/HashSet;
  
      invoke-virtual {v2, p0}, Ljava/util/HashSet;->remove(Ljava/lang/Object;)Z
  
+     .line 83
      iget-object v2, v1, Lcom/android/server/am/UriPermission;->writeOwners:Ljava/util/HashSet;
  
      invoke-virtual {v2}, Ljava/util/HashSet;->size()I
***************
*** 489,510 ****
--- 549,574 ----
  
      if-nez v2, :cond_3
  
+     .line 85
      iget v2, v1, Lcom/android/server/am/UriPermission;->modeFlags:I
  
      and-int/lit8 v2, v2, -0x3
  
      iput v2, v1, Lcom/android/server/am/UriPermission;->modeFlags:I
  
+     .line 86
      iget-object v2, p0, Lcom/android/server/am/UriPermissionOwner;->service:Lcom/android/server/am/ActivityManagerService;
  
      invoke-virtual {v2, v1}, Lcom/android/server/am/ActivityManagerService;->removeUriPermissionIfNeededLocked(Lcom/android/server/am/UriPermission;)V
  
      goto :goto_1
  
+     .line 89
      .end local v1           #perm:Lcom/android/server/am/UriPermission;
      :cond_4
      iput-object v3, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
+     .line 91
      .end local v0           #i$:Ljava/util/Iterator;
      :cond_5
      return-void
***************
*** 515,524 ****
--- 579,590 ----
      .parameter "perm"
  
      .prologue
+     .line 156
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v0, p1}, Ljava/util/HashSet;->remove(Ljava/lang/Object;)Z
  
+     .line 157
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
      invoke-virtual {v0}, Ljava/util/HashSet;->size()I
***************
*** 527,536 ****
--- 593,604 ----
  
      if-nez v0, :cond_0
  
+     .line 158
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->writeUriPermissions:Ljava/util/HashSet;
  
+     .line 160
      :cond_0
      return-void
  .end method
***************
*** 539,544 ****
--- 607,613 ----
      .locals 1
  
      .prologue
+     .line 164
      iget-object v0, p0, Lcom/android/server/am/UriPermissionOwner;->owner:Ljava/lang/Object;
  
      invoke-virtual {v0}, Ljava/lang/Object;->toString()Ljava/lang/String;
