*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/UsageStatsService.smali	2012-03-23 19:35:03.231645814 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/am/UsageStatsService.smali.original	2012-03-23 19:34:47.085351386 -0500
***************
*** 104,109 ****
--- 104,110 ----
      .locals 1
  
      .prologue
+     .line 89
      const/16 v0, 0x9
  
      new-array v0, v0, [I
***************
*** 133,140 ****
--- 134,143 ----
      .parameter "dir"
  
      .prologue
+     .line 244
      invoke-direct {p0}, Lcom/android/internal/app/IUsageStats$Stub;-><init>()V
  
+     .line 121
      new-instance v4, Ljava/util/concurrent/atomic/AtomicInteger;
  
      const/4 v5, -0x1
***************
*** 143,148 ****
--- 146,152 ----
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mLastWriteDay:Ljava/util/concurrent/atomic/AtomicInteger;
  
+     .line 122
      new-instance v4, Ljava/util/concurrent/atomic/AtomicLong;
  
      const-wide/16 v5, 0x0
***************
*** 151,156 ****
--- 155,161 ----
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mLastWriteElapsedTime:Ljava/util/concurrent/atomic/AtomicLong;
  
+     .line 123
      new-instance v4, Ljava/util/concurrent/atomic/AtomicBoolean;
  
      const/4 v5, 0x0
***************
*** 159,194 ****
--- 164,205 ----
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mUnforcedDiskWriteRunning:Ljava/util/concurrent/atomic/AtomicBoolean;
  
+     .line 245
      new-instance v4, Ljava/util/HashMap;
  
      invoke-direct {v4}, Ljava/util/HashMap;-><init>()V
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
+     .line 246
      new-instance v4, Ljava/util/HashMap;
  
      invoke-direct {v4}, Ljava/util/HashMap;-><init>()V
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
+     .line 247
      new-instance v4, Ljava/lang/Object;
  
      invoke-direct {v4}, Ljava/lang/Object;-><init>()V
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
+     .line 248
      new-instance v4, Ljava/lang/Object;
  
      invoke-direct {v4}, Ljava/lang/Object;-><init>()V
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mFileLock:Ljava/lang/Object;
  
+     .line 249
      new-instance v4, Ljava/io/File;
  
      invoke-direct {v4, p1}, Ljava/io/File;-><init>(Ljava/lang/String;)V
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mDir:Ljava/io/File;
  
+     .line 250
      const-string v4, "GMT+0"
  
      invoke-static {v4}, Ljava/util/TimeZone;->getTimeZone(Ljava/lang/String;)Ljava/util/TimeZone;
***************
*** 201,224 ****
--- 212,240 ----
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
  
+     .line 252
      iget-object v4, p0, Lcom/android/server/am/UsageStatsService;->mDir:Ljava/io/File;
  
      invoke-virtual {v4}, Ljava/io/File;->mkdir()Z
  
+     .line 255
      iget-object v4, p0, Lcom/android/server/am/UsageStatsService;->mDir:Ljava/io/File;
  
      invoke-virtual {v4}, Ljava/io/File;->getParentFile()Ljava/io/File;
  
      move-result-object v2
  
+     .line 256
      .local v2, parentDir:Ljava/io/File;
      invoke-virtual {v2}, Ljava/io/File;->list()[Ljava/lang/String;
  
      move-result-object v0
  
+     .line 257
      .local v0, fList:[Ljava/lang/String;
      if-eqz v0, :cond_1
  
+     .line 258
      new-instance v4, Ljava/lang/StringBuilder;
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 243,258 ****
--- 259,278 ----
  
      move-result-object v3
  
+     .line 259
      .local v3, prefix:Ljava/lang/String;
      array-length v1, v0
  
+     .line 260
      .local v1, i:I
      :cond_0
      :goto_0
      if-lez v1, :cond_1
  
+     .line 261
      add-int/lit8 v1, v1, -0x1
  
+     .line 262
      aget-object v4, v0, v1
  
      invoke-virtual {v4, v3}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
***************
*** 261,266 ****
--- 281,287 ----
  
      if-eqz v4, :cond_0
  
+     .line 263
      const-string v4, "UsageStats"
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 285,290 ****
--- 306,312 ----
  
      invoke-static {v4, v5}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 264
      new-instance v4, Ljava/io/File;
  
      aget-object v5, v0, v1
***************
*** 295,300 ****
--- 317,323 ----
  
      goto :goto_0
  
+     .line 270
      .end local v1           #i:I
      .end local v3           #prefix:Ljava/lang/String;
      :cond_1
***************
*** 306,311 ****
--- 329,335 ----
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mFileLeaf:Ljava/lang/String;
  
+     .line 271
      new-instance v4, Ljava/io/File;
  
      iget-object v5, p0, Lcom/android/server/am/UsageStatsService;->mDir:Ljava/io/File;
***************
*** 316,321 ****
--- 340,346 ----
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mFile:Ljava/io/File;
  
+     .line 272
      new-instance v4, Lcom/android/internal/os/AtomicFile;
  
      new-instance v5, Ljava/io/File;
***************
*** 330,339 ****
--- 355,367 ----
  
      iput-object v4, p0, Lcom/android/server/am/UsageStatsService;->mHistoryFile:Lcom/android/internal/os/AtomicFile;
  
+     .line 273
      invoke-direct {p0}, Lcom/android/server/am/UsageStatsService;->readStatsFromFile()V
  
+     .line 274
      invoke-direct {p0}, Lcom/android/server/am/UsageStatsService;->readHistoryStatsFromFile()V
  
+     .line 275
      iget-object v4, p0, Lcom/android/server/am/UsageStatsService;->mLastWriteElapsedTime:Ljava/util/concurrent/atomic/AtomicLong;
  
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
***************
*** 342,347 ****
--- 370,376 ----
  
      invoke-virtual {v4, v5, v6}, Ljava/util/concurrent/atomic/AtomicLong;->set(J)V
  
+     .line 277
      iget-object v4, p0, Lcom/android/server/am/UsageStatsService;->mLastWriteDay:Ljava/util/concurrent/atomic/AtomicInteger;
  
      iget-object v5, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
***************
*** 354,359 ****
--- 383,389 ----
  
      invoke-virtual {v4, v5}, Ljava/util/concurrent/atomic/AtomicInteger;->set(I)V
  
+     .line 278
      return-void
  .end method
  
***************
*** 361,366 ****
--- 391,397 ----
      .locals 1
  
      .prologue
+     .line 69
      sget-object v0, Lcom/android/server/am/UsageStatsService;->LAUNCH_TIME_BINS:[I
  
      return-object v0
***************
*** 373,378 ****
--- 404,410 ----
      .parameter "x2"
  
      .prologue
+     .line 69
      invoke-direct {p0, p1, p2}, Lcom/android/server/am/UsageStatsService;->writeStatsToFile(ZZ)V
  
      return-void
***************
*** 383,388 ****
--- 415,421 ----
      .parameter "x0"
  
      .prologue
+     .line 69
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mUnforcedDiskWriteRunning:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      return-object v0
***************
*** 393,398 ****
--- 426,432 ----
      .parameter "x0"
  
      .prologue
+     .line 69
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
      return-object v0
***************
*** 402,443 ****
--- 436,487 ----
      .locals 8
  
      .prologue
+     .line 443
      invoke-direct {p0}, Lcom/android/server/am/UsageStatsService;->getUsageStatsFileListFLOCK()Ljava/util/ArrayList;
  
      move-result-object v2
  
+     .line 444
      .local v2, fileList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
      if-nez v2, :cond_1
  
+     .line 462
      :cond_0
      return-void
  
+     .line 448
      :cond_1
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
  
      move-result v0
  
+     .line 449
      .local v0, count:I
      const/4 v5, 0x5
  
      if-le v0, v5, :cond_0
  
+     .line 453
      invoke-static {v2}, Ljava/util/Collections;->sort(Ljava/util/List;)V
  
+     .line 454
      add-int/lit8 v0, v0, -0x5
  
+     .line 456
      const/4 v4, 0x0
  
      .local v4, i:I
      :goto_0
      if-ge v4, v0, :cond_0
  
+     .line 457
      invoke-virtual {v2, v4}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v3
  
      check-cast v3, Ljava/lang/String;
  
+     .line 458
      .local v3, fileName:Ljava/lang/String;
      new-instance v1, Ljava/io/File;
  
***************
*** 445,450 ****
--- 489,495 ----
  
      invoke-direct {v1, v5, v3}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 459
      .local v1, file:Ljava/io/File;
      const-string v5, "UsageStats"
  
***************
*** 468,475 ****
--- 513,522 ----
  
      invoke-static {v5, v6}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 460
      invoke-virtual {v1}, Ljava/io/File;->delete()Z
  
+     .line 456
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
***************
*** 494,514 ****
--- 541,566 ----
      .end annotation
  
      .prologue
+     .line 856
      .local p4, packages:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/String;>;"
      invoke-direct {p0}, Lcom/android/server/am/UsageStatsService;->getUsageStatsFileListFLOCK()Ljava/util/ArrayList;
  
      move-result-object v9
  
+     .line 857
      .local v9, fileList:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
      if-nez v9, :cond_1
  
+     .line 884
      :cond_0
      :goto_0
      return-void
  
+     .line 860
      :cond_1
      invoke-static {v9}, Ljava/util/Collections;->sort(Ljava/util/List;)V
  
+     .line 861
      invoke-interface {v9}, Ljava/util/List;->iterator()Ljava/util/Iterator;
  
      move-result-object v10
***************
*** 528,533 ****
--- 580,586 ----
  
      check-cast v8, Ljava/lang/String;
  
+     .line 862
      .local v8, file:Ljava/lang/String;
      if-eqz p3, :cond_3
  
***************
*** 539,544 ****
--- 592,598 ----
  
      if-nez v0, :cond_2
  
+     .line 867
      :cond_3
      new-instance v6, Ljava/io/File;
  
***************
*** 546,551 ****
--- 600,606 ----
  
      invoke-direct {v6, v0, v8}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 868
      .local v6, dFile:Ljava/io/File;
      const-string v0, "usage-"
  
***************
*** 557,562 ****
--- 612,618 ----
  
      move-result-object v3
  
+     .line 870
      .local v3, dateStr:Ljava/lang/String;
      :try_start_0
      invoke-direct {p0, v6}, Lcom/android/server/am/UsageStatsService;->getParcelForFile(Ljava/io/File;)Landroid/os/Parcel;
***************
*** 572,581 ****
--- 628,640 ----
  
      move-object v5, p4
  
+     .line 871
      invoke-direct/range {v0 .. v5}, Lcom/android/server/am/UsageStatsService;->collectDumpInfoFromParcelFLOCK(Landroid/os/Parcel;Ljava/io/PrintWriter;Ljava/lang/String;ZLjava/util/HashSet;)V
  
+     .line 873
      if-eqz p3, :cond_2
  
+     .line 875
      invoke-virtual {v6}, Ljava/io/File;->delete()Z
      :try_end_0
      .catch Ljava/io/FileNotFoundException; {:try_start_0 .. :try_end_0} :catch_0
***************
*** 583,592 ****
--- 642,653 ----
  
      goto :goto_1
  
+     .line 877
      .end local v1           #in:Landroid/os/Parcel;
      :catch_0
      move-exception v7
  
+     .line 878
      .local v7, e:Ljava/io/FileNotFoundException;
      const-string v0, "UsageStats"
  
***************
*** 622,631 ****
--- 683,694 ----
  
      goto :goto_0
  
+     .line 880
      .end local v7           #e:Ljava/io/FileNotFoundException;
      :catch_1
      move-exception v7
  
+     .line 881
      .local v7, e:Ljava/io/IOException;
      const-string v0, "UsageStats"
  
***************
*** 684,689 ****
--- 747,753 ----
      .end annotation
  
      .prologue
+     .line 888
      .local p5, packages:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/String;>;"
      new-instance v10, Ljava/lang/StringBuilder;
  
***************
*** 691,729 ****
--- 755,802 ----
  
      invoke-direct {v10, v13}, Ljava/lang/StringBuilder;-><init>(I)V
  
+     .line 889
      .local v10, sb:Ljava/lang/StringBuilder;
      if-eqz p4, :cond_1
  
+     .line 890
      const-string v13, "D:"
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 891
      const/4 v13, 0x4
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 892
      const/16 v13, 0x2c
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 897
      :goto_0
      move-object/from16 v0, p3
  
      invoke-virtual {v10, v0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 899
      invoke-virtual/range {p1 .. p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v12
  
+     .line 900
      .local v12, vers:I
      const/16 v13, 0x3ef
  
      if-eq v12, v13, :cond_2
  
+     .line 901
      const-string v13, " (old data version)"
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 902
      invoke-virtual {v10}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v13
***************
*** 732,740 ****
--- 805,815 ----
  
      invoke-virtual {v0, v13}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 995
      :cond_0
      return-void
  
+     .line 894
      .end local v12           #vers:I
      :cond_1
      const-string v13, "Date: "
***************
*** 743,748 ****
--- 818,824 ----
  
      goto :goto_0
  
+     .line 906
      .restart local v12       #vers:I
      :cond_2
      invoke-virtual {v10}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
***************
*** 753,785 ****
--- 829,869 ----
  
      invoke-virtual {v0, v13}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 907
      invoke-virtual/range {p1 .. p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
+     .line 909
      .local v1, N:I
      :goto_1
      if-lez v1, :cond_0
  
+     .line 910
      add-int/lit8 v1, v1, -0x1
  
+     .line 911
      invoke-virtual/range {p1 .. p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v8
  
+     .line 912
      .local v8, pkgName:Ljava/lang/String;
      if-eqz v8, :cond_0
  
+     .line 915
      const/4 v13, 0x0
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->setLength(I)V
  
+     .line 916
      new-instance v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
      move-object/from16 v0, p1
  
      invoke-direct {v9, p0, v0}, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;-><init>(Lcom/android/server/am/UsageStatsService;Landroid/os/Parcel;)V
  
+     .line 917
      .local v9, pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      if-eqz p5, :cond_4
  
***************
*** 791,796 ****
--- 875,881 ----
  
      if-nez v13, :cond_4
  
+     .line 993
      :cond_3
      invoke-virtual {v10}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
***************
*** 802,845 ****
--- 887,941 ----
  
      goto :goto_1
  
+     .line 920
      :cond_4
      if-eqz p4, :cond_7
  
+     .line 921
      const-string v13, "P:"
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 922
      invoke-virtual {v10, v8}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 923
      const/16 v13, 0x2c
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 924
      iget v13, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mLaunchCount:I
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 925
      const/16 v13, 0x2c
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 926
      iget-wide v13, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mUsageTime:J
  
      invoke-virtual {v10, v13, v14}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 927
      const/16 v13, 0xa
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 928
      iget-object v13, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mLaunchTimes:Ljava/util/HashMap;
  
      invoke-virtual {v13}, Ljava/util/HashMap;->size()I
  
      move-result v2
  
+     .line 929
      .local v2, NC:I
      if-lez v2, :cond_3
  
+     .line 930
      iget-object v13, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mLaunchTimes:Ljava/util/HashMap;
  
      invoke-virtual {v13}, Ljava/util/HashMap;->entrySet()Ljava/util/Set;
***************
*** 864,880 ****
--- 960,979 ----
  
      check-cast v4, Ljava/util/Map$Entry;
  
+     .line 931
      .local v4, ent:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Lcom/android/server/am/UsageStatsService$TimeStats;>;"
      const-string v13, "A:"
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 932
      invoke-interface {v4}, Ljava/util/Map$Entry;->getKey()Ljava/lang/Object;
  
      move-result-object v3
  
      check-cast v3, Ljava/lang/String;
  
+     .line 933
      .local v3, activity:Ljava/lang/String;
      invoke-virtual {v3, v8}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
  
***************
*** 882,891 ****
--- 981,992 ----
  
      if-eqz v13, :cond_5
  
+     .line 934
      const/16 v13, 0x2a
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 935
      invoke-virtual {v8}, Ljava/lang/String;->length()I
  
      move-result v13
***************
*** 900,905 ****
--- 1001,1007 ----
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 940
      :goto_3
      invoke-interface {v4}, Ljava/util/Map$Entry;->getValue()Ljava/lang/Object;
  
***************
*** 907,921 ****
--- 1009,1026 ----
  
      check-cast v11, Lcom/android/server/am/UsageStatsService$TimeStats;
  
+     .line 941
      .local v11, times:Lcom/android/server/am/UsageStatsService$TimeStats;
      const/16 v13, 0x2c
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 942
      iget v13, v11, Lcom/android/server/am/UsageStatsService$TimeStats;->count:I
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 943
      const/4 v5, 0x0
  
      .local v5, i:I
***************
*** 924,943 ****
--- 1029,1052 ----
  
      if-ge v5, v13, :cond_6
  
+     .line 944
      const-string v13, ","
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 945
      iget-object v13, v11, Lcom/android/server/am/UsageStatsService$TimeStats;->times:[I
  
      aget v13, v13, v5
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 943
      add-int/lit8 v5, v5, 0x1
  
      goto :goto_4
  
+     .line 938
      .end local v5           #i:I
      .end local v11           #times:Lcom/android/server/am/UsageStatsService$TimeStats;
      :cond_5
***************
*** 945,950 ****
--- 1054,1060 ----
  
      goto :goto_3
  
+     .line 947
      .restart local v5       #i:I
      .restart local v11       #times:Lcom/android/server/am/UsageStatsService$TimeStats;
      :cond_6
***************
*** 954,959 ****
--- 1064,1070 ----
  
      goto :goto_2
  
+     .line 952
      .end local v2           #NC:I
      .end local v3           #activity:Ljava/lang/String;
      .end local v4           #ent:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Lcom/android/server/am/UsageStatsService$TimeStats;>;"
***************
*** 965,1005 ****
--- 1076,1126 ----
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 953
      invoke-virtual {v10, v8}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 954
      const-string v13, ": "
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 955
      iget v13, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mLaunchCount:I
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 956
      const-string v13, " times, "
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 957
      iget-wide v13, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mUsageTime:J
  
      invoke-virtual {v10, v13, v14}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 958
      const-string v13, " ms"
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 959
      const/16 v13, 0xa
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 960
      iget-object v13, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mLaunchTimes:Ljava/util/HashMap;
  
      invoke-virtual {v13}, Ljava/util/HashMap;->size()I
  
      move-result v2
  
+     .line 961
      .restart local v2       #NC:I
      if-lez v2, :cond_3
  
+     .line 962
      iget-object v13, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mLaunchTimes:Ljava/util/HashMap;
  
      invoke-virtual {v13}, Ljava/util/HashMap;->entrySet()Ljava/util/Set;
***************
*** 1024,1034 ****
--- 1145,1157 ----
  
      check-cast v4, Ljava/util/Map$Entry;
  
+     .line 963
      .restart local v4       #ent:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Lcom/android/server/am/UsageStatsService$TimeStats;>;"
      const-string v13, "    "
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 964
      invoke-interface {v4}, Ljava/util/Map$Entry;->getKey()Ljava/lang/Object;
  
      move-result-object v13
***************
*** 1037,1063 ****
--- 1160,1192 ----
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 965
      invoke-interface {v4}, Ljava/util/Map$Entry;->getValue()Ljava/lang/Object;
  
      move-result-object v11
  
      check-cast v11, Lcom/android/server/am/UsageStatsService$TimeStats;
  
+     .line 966
      .restart local v11       #times:Lcom/android/server/am/UsageStatsService$TimeStats;
      const-string v13, ": "
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 967
      iget v13, v11, Lcom/android/server/am/UsageStatsService$TimeStats;->count:I
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 968
      const-string v13, " starts"
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 969
      const/4 v7, 0x0
  
+     .line 970
      .local v7, lastBin:I
      const/4 v5, 0x0
  
***************
*** 1067,1113 ****
--- 1196,1252 ----
  
      if-ge v5, v13, :cond_9
  
+     .line 971
      iget-object v13, v11, Lcom/android/server/am/UsageStatsService$TimeStats;->times:[I
  
      aget v13, v13, v5
  
      if-eqz v13, :cond_8
  
+     .line 972
      const-string v13, ", "
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 973
      invoke-virtual {v10, v7}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 974
      const/16 v13, 0x2d
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 975
      sget-object v13, Lcom/android/server/am/UsageStatsService;->LAUNCH_TIME_BINS:[I
  
      aget v13, v13, v5
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 976
      const-string v13, "ms="
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 977
      iget-object v13, v11, Lcom/android/server/am/UsageStatsService$TimeStats;->times:[I
  
      aget v13, v13, v5
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 979
      :cond_8
      sget-object v13, Lcom/android/server/am/UsageStatsService;->LAUNCH_TIME_BINS:[I
  
      aget v7, v13, v5
  
+     .line 970
      add-int/lit8 v5, v5, 0x1
  
      goto :goto_6
  
+     .line 981
      :cond_9
      iget-object v13, v11, Lcom/android/server/am/UsageStatsService$TimeStats;->times:[I
  
***************
*** 1117,1136 ****
--- 1256,1280 ----
  
      if-eqz v13, :cond_a
  
+     .line 982
      const-string v13, ", "
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 983
      const-string v13, ">="
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 984
      invoke-virtual {v10, v7}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 985
      const-string v13, "ms="
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 986
      iget-object v13, v11, Lcom/android/server/am/UsageStatsService$TimeStats;->times:[I
  
      const/16 v14, 0x9
***************
*** 1139,1144 ****
--- 1283,1289 ----
  
      invoke-virtual {v10, v13}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 988
      :cond_a
      const/16 v13, 0xa
  
***************
*** 1151,1160 ****
--- 1296,1307 ----
      .locals 7
  
      .prologue
+     .line 593
      iget-object v4, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
      monitor-enter v4
  
+     .line 596
      :try_start_0
      new-instance v2, Ljava/util/HashMap;
  
***************
*** 1162,1172 ****
--- 1309,1321 ----
  
      invoke-direct {v2, v3}, Ljava/util/HashMap;-><init>(Ljava/util/Map;)V
  
+     .line 598
      .local v2, tmpLastResumeTimes:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;>;"
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
      invoke-interface {v3}, Ljava/util/Map;->clear()V
  
+     .line 599
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v3}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
***************
*** 1198,1203 ****
--- 1347,1353 ----
  
      check-cast v1, Landroid/content/pm/PackageInfo;
  
+     .line 600
      .local v1, info:Landroid/content/pm/PackageInfo;
      iget-object v3, v1, Landroid/content/pm/PackageInfo;->packageName:Ljava/lang/String;
  
***************
*** 1207,1212 ****
--- 1357,1363 ----
  
      if-eqz v3, :cond_0
  
+     .line 601
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
      iget-object v5, v1, Landroid/content/pm/PackageInfo;->packageName:Ljava/lang/String;
***************
*** 1221,1226 ****
--- 1372,1378 ----
  
      goto :goto_0
  
+     .line 604
      .end local v0           #i$:Ljava/util/Iterator;
      .end local v1           #info:Landroid/content/pm/PackageInfo;
      .end local v2           #tmpLastResumeTimes:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;>;"
***************
*** 1241,1246 ****
--- 1393,1399 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 605
      return-void
  .end method
  
***************
*** 1251,1265 ****
--- 1404,1421 ----
      .prologue
      const/16 v7, 0xa
  
+     .line 284
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 285
      .local v2, sb:Ljava/lang/StringBuilder;
      iget-object v4, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
  
      monitor-enter v4
  
+     .line 286
      :try_start_0
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
  
***************
*** 1269,1278 ****
--- 1425,1437 ----
  
      invoke-virtual {v3, v5, v6}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 287
      if-eqz p1, :cond_0
  
+     .line 288
      invoke-virtual {v2, p1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 290
      :cond_0
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
  
***************
*** 1284,1289 ****
--- 1443,1449 ----
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 291
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
  
      const/4 v5, 0x2
***************
*** 1296,1311 ****
--- 1456,1475 ----
  
      add-int/lit8 v1, v3, 0x1
  
+     .line 292
      .local v1, mm:I
      if-ge v1, v7, :cond_1
  
+     .line 293
      const-string v3, "0"
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 295
      :cond_1
      invoke-virtual {v2, v1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 296
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
  
      const/4 v5, 0x5
***************
*** 1314,1339 ****
--- 1478,1509 ----
  
      move-result v0
  
+     .line 297
      .local v0, dd:I
      if-ge v0, v7, :cond_2
  
+     .line 298
      const-string v3, "0"
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 300
      :cond_2
      invoke-virtual {v2, v0}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 301
      monitor-exit v4
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 302
      invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v3
  
      return-object v3
  
+     .line 301
      .end local v0           #dd:I
      .end local v1           #mm:I
      :catchall_0
***************
*** 1359,1387 ****
--- 1529,1564 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 306
      new-instance v2, Ljava/io/FileInputStream;
  
      invoke-direct {v2, p1}, Ljava/io/FileInputStream;-><init>(Ljava/io/File;)V
  
+     .line 307
      .local v2, stream:Ljava/io/FileInputStream;
      invoke-static {v2}, Lcom/android/server/am/UsageStatsService;->readFully(Ljava/io/FileInputStream;)[B
  
      move-result-object v1
  
+     .line 308
      .local v1, raw:[B
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 309
      .local v0, in:Landroid/os/Parcel;
      array-length v3, v1
  
      invoke-virtual {v0, v1, v4, v3}, Landroid/os/Parcel;->unmarshall([BII)V
  
+     .line 310
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->setDataPosition(I)V
  
+     .line 311
      invoke-virtual {v2}, Ljava/io/FileInputStream;->close()V
  
+     .line 312
      return-object v0
  .end method
  
***************
*** 1389,1404 ****
--- 1566,1585 ----
      .locals 2
  
      .prologue
+     .line 672
      sget-object v1, Lcom/android/server/am/UsageStatsService;->sService:Lcom/android/internal/app/IUsageStats;
  
      if-eqz v1, :cond_0
  
+     .line 673
      sget-object v1, Lcom/android/server/am/UsageStatsService;->sService:Lcom/android/internal/app/IUsageStats;
  
+     .line 677
      .local v0, b:Landroid/os/IBinder;
      :goto_0
      return-object v1
  
+     .line 675
      .end local v0           #b:Landroid/os/IBinder;
      :cond_0
      const-string v1, "usagestats"
***************
*** 1407,1412 ****
--- 1588,1594 ----
  
      move-result-object v0
  
+     .line 676
      .restart local v0       #b:Landroid/os/IBinder;
      invoke-static {v0}, Lcom/android/server/am/UsageStatsService;->asInterface(Landroid/os/IBinder;)Lcom/android/internal/app/IUsageStats;
  
***************
*** 1414,1419 ****
--- 1596,1602 ----
  
      sput-object v1, Lcom/android/server/am/UsageStatsService;->sService:Lcom/android/internal/app/IUsageStats;
  
+     .line 677
      sget-object v1, Lcom/android/server/am/UsageStatsService;->sService:Lcom/android/internal/app/IUsageStats;
  
      goto :goto_0
***************
*** 1432,1456 ****
--- 1615,1645 ----
      .end annotation
  
      .prologue
+     .line 423
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mDir:Ljava/io/File;
  
      invoke-virtual {v6}, Ljava/io/File;->list()[Ljava/lang/String;
  
      move-result-object v1
  
+     .line 424
      .local v1, fList:[Ljava/lang/String;
      if-nez v1, :cond_1
  
+     .line 425
      const/4 v3, 0x0
  
+     .line 438
      :cond_0
      return-object v3
  
+     .line 427
      :cond_1
      new-instance v3, Ljava/util/ArrayList;
  
      invoke-direct {v3}, Ljava/util/ArrayList;-><init>()V
  
+     .line 428
      .local v3, fileList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
      move-object v0, v1
  
***************
*** 1466,1471 ****
--- 1655,1661 ----
  
      aget-object v2, v0, v4
  
+     .line 429
      .local v2, file:Ljava/lang/String;
      const-string v6, "usage-"
  
***************
*** 1475,1485 ****
--- 1665,1677 ----
  
      if-nez v6, :cond_2
  
+     .line 428
      :goto_1
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
  
+     .line 432
      :cond_2
      const-string v6, ".bak"
  
***************
*** 1489,1494 ****
--- 1681,1687 ----
  
      if-eqz v6, :cond_3
  
+     .line 433
      new-instance v6, Ljava/io/File;
  
      iget-object v7, p0, Lcom/android/server/am/UsageStatsService;->mDir:Ljava/io/File;
***************
*** 1499,1504 ****
--- 1692,1698 ----
  
      goto :goto_1
  
+     .line 436
      :cond_3
      invoke-virtual {v3, v2}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
***************
*** 1517,1532 ****
--- 1711,1730 ----
      .prologue
      const/4 v6, 0x0
  
+     .line 836
      const/4 v4, 0x0
  
+     .line 837
      .local v4, pos:I
      invoke-virtual {p0}, Ljava/io/FileInputStream;->available()I
  
      move-result v1
  
+     .line 838
      .local v1, avail:I
      new-array v2, v1, [B
  
+     .line 840
      .local v2, data:[B
      :cond_0
      :goto_0
***************
*** 1538,1568 ****
--- 1736,1774 ----
  
      move-result v0
  
+     .line 841
      .local v0, amt:I
      if-gtz v0, :cond_1
  
+     .line 842
      return-object v2
  
+     .line 844
      :cond_1
      add-int/2addr v4, v0
  
+     .line 845
      invoke-virtual {p0}, Ljava/io/FileInputStream;->available()I
  
      move-result v1
  
+     .line 846
      array-length v5, v2
  
      sub-int/2addr v5, v4
  
      if-le v1, v5, :cond_0
  
+     .line 847
      add-int v5, v4, v1
  
      new-array v3, v5, [B
  
+     .line 848
      .local v3, newData:[B
      invoke-static {v2, v6, v3, v6, v4}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 849
      move-object v2, v3
  
      goto :goto_0
***************
*** 1573,1580 ****
--- 1779,1788 ----
      .parameter "file"
  
      .prologue
+     .line 363
      const/4 v4, 0x0
  
+     .line 365
      .local v4, fis:Ljava/io/FileInputStream;
      :try_start_0
      iget-object v12, p0, Lcom/android/server/am/UsageStatsService;->mHistoryFile:Lcom/android/internal/os/AtomicFile;
***************
*** 1583,1618 ****
--- 1791,1833 ----
  
      move-result-object v4
  
+     .line 366
      invoke-static {}, Landroid/util/Xml;->newPullParser()Lorg/xmlpull/v1/XmlPullParser;
  
      move-result-object v9
  
+     .line 367
      .local v9, parser:Lorg/xmlpull/v1/XmlPullParser;
      const/4 v12, 0x0
  
      invoke-interface {v9, v4, v12}, Lorg/xmlpull/v1/XmlPullParser;->setInput(Ljava/io/InputStream;Ljava/lang/String;)V
  
+     .line 368
      invoke-interface {v9}, Lorg/xmlpull/v1/XmlPullParser;->getEventType()I
  
      move-result v3
  
+     .line 369
      .local v3, eventType:I
      :goto_0
      const/4 v12, 0x2
  
      if-eq v3, v12, :cond_0
  
+     .line 370
      invoke-interface {v9}, Lorg/xmlpull/v1/XmlPullParser;->next()I
  
      move-result v3
  
      goto :goto_0
  
+     .line 372
      :cond_0
      invoke-interface {v9}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
  
      move-result-object v11
  
+     .line 373
      .local v11, tagName:Ljava/lang/String;
      const-string v12, "usage-history"
  
***************
*** 1622,1647 ****
--- 1837,1868 ----
  
      if-eqz v12, :cond_3
  
+     .line 374
      const/4 v10, 0x0
  
+     .line 376
      .local v10, pkg:Ljava/lang/String;
      :cond_1
      invoke-interface {v9}, Lorg/xmlpull/v1/XmlPullParser;->next()I
  
      move-result v3
  
+     .line 377
      const/4 v12, 0x2
  
      if-ne v3, v12, :cond_7
  
+     .line 378
      invoke-interface {v9}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
  
      move-result-object v11
  
+     .line 379
      invoke-interface {v9}, Lorg/xmlpull/v1/XmlPullParser;->getDepth()I
  
      move-result v1
  
+     .line 380
      .local v1, depth:I
      const-string v12, "pkg"
  
***************
*** 1655,1660 ****
--- 1876,1882 ----
  
      if-ne v1, v12, :cond_5
  
+     .line 381
      const/4 v12, 0x0
  
      const-string v13, "name"
***************
*** 1667,1672 ****
--- 1889,1895 ----
  
      move-result-object v10
  
+     .line 405
      .end local v1           #depth:I
      :cond_2
      :goto_1
***************
*** 1674,1688 ****
--- 1897,1914 ----
  
      if-ne v3, v12, :cond_1
  
+     .line 412
      .end local v10           #pkg:Ljava/lang/String;
      :cond_3
      if-eqz v4, :cond_4
  
+     .line 414
      :try_start_1
      invoke-virtual {v4}, Ljava/io/FileInputStream;->close()V
      :try_end_1
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_5
  
+     .line 419
      .end local v3           #eventType:I
      .end local v9           #parser:Lorg/xmlpull/v1/XmlPullParser;
      .end local v11           #tagName:Ljava/lang/String;
***************
*** 1690,1695 ****
--- 1916,1922 ----
      :goto_2
      return-void
  
+     .line 382
      .restart local v1       #depth:I
      .restart local v3       #eventType:I
      .restart local v9       #parser:Lorg/xmlpull/v1/XmlPullParser;
***************
*** 1711,1716 ****
--- 1938,1944 ----
  
      if-eqz v10, :cond_2
  
+     .line 383
      const/4 v12, 0x0
  
      const-string v13, "name"
***************
*** 1719,1724 ****
--- 1947,1953 ----
  
      move-result-object v0
  
+     .line 384
      .local v0, comp:Ljava/lang/String;
      const/4 v12, 0x0
  
***************
*** 1732,1747 ****
--- 1961,1979 ----
  
      move-result-object v7
  
+     .line 385
      .local v7, lastResumeTimeStr:Ljava/lang/String;
      if-eqz v0, :cond_2
  
      if-eqz v7, :cond_2
  
+     .line 387
      :try_start_3
      invoke-static {v7}, Ljava/lang/Long;->parseLong(Ljava/lang/String;)J
  
      move-result-wide v5
  
+     .line 388
      .local v5, lastResumeTime:J
      iget-object v13, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
***************
*** 1752,1757 ****
--- 1984,1990 ----
      .catch Lorg/xmlpull/v1/XmlPullParserException; {:try_start_3 .. :try_end_3} :catch_1
      .catch Ljava/io/IOException; {:try_start_3 .. :try_end_3} :catch_3
  
+     .line 389
      :try_start_4
      iget-object v12, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
***************
*** 1761,1779 ****
--- 1994,2016 ----
  
      check-cast v8, Ljava/util/Map;
  
+     .line 390
      .local v8, lrt:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;"
      if-nez v8, :cond_6
  
+     .line 391
      new-instance v8, Ljava/util/HashMap;
  
      .end local v8           #lrt:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;"
      invoke-direct {v8}, Ljava/util/HashMap;-><init>()V
  
+     .line 392
      .restart local v8       #lrt:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;"
      iget-object v12, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
      invoke-interface {v12, v10, v8}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 394
      :cond_6
      invoke-static {v5, v6}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
  
***************
*** 1781,1786 ****
--- 2018,2024 ----
  
      invoke-interface {v8, v0, v12}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 395
      monitor-exit v13
  
      goto :goto_1
***************
*** 1801,1812 ****
--- 2039,2052 ----
      .catch Lorg/xmlpull/v1/XmlPullParserException; {:try_start_5 .. :try_end_5} :catch_1
      .catch Ljava/io/IOException; {:try_start_5 .. :try_end_5} :catch_3
  
+     .line 396
      .end local v5           #lastResumeTime:J
      :catch_0
      move-exception v12
  
      goto :goto_1
  
+     .line 400
      .end local v0           #comp:Ljava/lang/String;
      .end local v1           #depth:I
      .end local v7           #lastResumeTimeStr:Ljava/lang/String;
***************
*** 1815,1820 ****
--- 2055,2061 ----
  
      if-ne v3, v12, :cond_2
  
+     .line 401
      :try_start_6
      const-string v12, "pkg"
  
***************
*** 1832,1841 ****
--- 2073,2084 ----
  
      if-eqz v12, :cond_2
  
+     .line 402
      const/4 v10, 0x0
  
      goto :goto_1
  
+     .line 407
      .end local v3           #eventType:I
      .end local v9           #parser:Lorg/xmlpull/v1/XmlPullParser;
      .end local v10           #pkg:Ljava/lang/String;
***************
*** 1843,1848 ****
--- 2086,2092 ----
      :catch_1
      move-exception v2
  
+     .line 408
      .local v2, e:Lorg/xmlpull/v1/XmlPullParserException;
      :try_start_7
      const-string v12, "UsageStats"
***************
*** 1869,1876 ****
--- 2113,2122 ----
      :try_end_7
      .catchall {:try_start_7 .. :try_end_7} :catchall_1
  
+     .line 412
      if-eqz v4, :cond_4
  
+     .line 414
      :try_start_8
      invoke-virtual {v4}, Ljava/io/FileInputStream;->close()V
      :try_end_8
***************
*** 1878,1892 ****
--- 2124,2141 ----
  
      goto :goto_2
  
+     .line 415
      :catch_2
      move-exception v12
  
      goto :goto_2
  
+     .line 409
      .end local v2           #e:Lorg/xmlpull/v1/XmlPullParserException;
      :catch_3
      move-exception v2
  
+     .line 410
      .local v2, e:Ljava/io/IOException;
      :try_start_9
      const-string v12, "UsageStats"
***************
*** 1913,1920 ****
--- 2162,2171 ----
      :try_end_9
      .catchall {:try_start_9 .. :try_end_9} :catchall_1
  
+     .line 412
      if-eqz v4, :cond_4
  
+     .line 414
      :try_start_a
      invoke-virtual {v4}, Ljava/io/FileInputStream;->close()V
      :try_end_a
***************
*** 1922,1947 ****
--- 2173,2203 ----
  
      goto/16 :goto_2
  
+     .line 415
      :catch_4
      move-exception v12
  
      goto/16 :goto_2
  
+     .line 412
      .end local v2           #e:Ljava/io/IOException;
      :catchall_1
      move-exception v12
  
      if-eqz v4, :cond_8
  
+     .line 414
      :try_start_b
      invoke-virtual {v4}, Ljava/io/FileInputStream;->close()V
      :try_end_b
      .catch Ljava/io/IOException; {:try_start_b .. :try_end_b} :catch_6
  
+     .line 416
      :cond_8
      :goto_3
      throw v12
  
+     .line 415
      .restart local v3       #eventType:I
      .restart local v9       #parser:Lorg/xmlpull/v1/XmlPullParser;
      .restart local v11       #tagName:Ljava/lang/String;
***************
*** 1963,1972 ****
--- 2219,2230 ----
      .locals 2
  
      .prologue
+     .line 355
      iget-object v1, p0, Lcom/android/server/am/UsageStatsService;->mFileLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 356
      :try_start_0
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mHistoryFile:Lcom/android/internal/os/AtomicFile;
  
***************
*** 1980,1994 ****
--- 2238,2256 ----
  
      if-eqz v0, :cond_0
  
+     .line 357
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mHistoryFile:Lcom/android/internal/os/AtomicFile;
  
      invoke-direct {p0, v0}, Lcom/android/server/am/UsageStatsService;->readHistoryStatsFLOCK(Lcom/android/internal/os/AtomicFile;)V
  
+     .line 359
      :cond_0
      monitor-exit v1
  
+     .line 360
      return-void
  
+     .line 359
      :catchall_0
      move-exception v0
  
***************
*** 2009,2069 ****
--- 2271,2345 ----
      .end annotation
  
      .prologue
+     .line 333
      invoke-direct {p0, p1}, Lcom/android/server/am/UsageStatsService;->getParcelForFile(Ljava/io/File;)Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 334
      .local v1, in:Landroid/os/Parcel;
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
  
+     .line 335
      .local v4, vers:I
      const/16 v5, 0x3ef
  
      if-eq v4, v5, :cond_1
  
+     .line 336
      const-string v5, "UsageStats"
  
      const-string v6, "Usage stats version changed; dropping"
  
      invoke-static {v5, v6}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 352
      :cond_0
      return-void
  
+     .line 339
      :cond_1
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 340
      .local v0, N:I
      :goto_0
      if-lez v0, :cond_0
  
+     .line 341
      add-int/lit8 v0, v0, -0x1
  
+     .line 342
      invoke-virtual {v1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
+     .line 343
      .local v2, pkgName:Ljava/lang/String;
      if-eqz v2, :cond_0
  
+     .line 347
      new-instance v3, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
      invoke-direct {v3, p0, v1}, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;-><init>(Lcom/android/server/am/UsageStatsService;Landroid/os/Parcel;)V
  
+     .line 348
      .local v3, pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
      monitor-enter v6
  
+     .line 349
      :try_start_0
      iget-object v5, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
      invoke-interface {v5, v2, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 350
      monitor-exit v6
  
      goto :goto_0
***************
*** 2082,2094 ****
--- 2358,2373 ----
      .locals 6
  
      .prologue
+     .line 316
      iget-object v1, p0, Lcom/android/server/am/UsageStatsService;->mFile:Ljava/io/File;
  
+     .line 317
      .local v1, newFile:Ljava/io/File;
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mFileLock:Ljava/lang/Object;
  
      monitor-enter v3
  
+     .line 319
      :try_start_0
      invoke-virtual {v1}, Ljava/io/File;->exists()Z
  
***************
*** 2096,2118 ****
--- 2375,2402 ----
  
      if-eqz v2, :cond_0
  
+     .line 320
      invoke-direct {p0, v1}, Lcom/android/server/am/UsageStatsService;->readStatsFLOCK(Ljava/io/File;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 329
      :goto_0
      :try_start_1
      monitor-exit v3
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 330
      return-void
  
+     .line 323
      :cond_0
      :try_start_2
      invoke-direct {p0}, Lcom/android/server/am/UsageStatsService;->checkFileLimitFLOCK()V
  
+     .line 324
      invoke-virtual {v1}, Ljava/io/File;->createNewFile()Z
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
***************
*** 2120,2128 ****
--- 2404,2414 ----
  
      goto :goto_0
  
+     .line 326
      :catch_0
      move-exception v0
  
+     .line 327
      .local v0, e:Ljava/io/IOException;
      :try_start_3
      const-string v2, "UsageStats"
***************
*** 2159,2164 ****
--- 2445,2451 ----
  
      goto :goto_0
  
+     .line 329
      .end local v0           #e:Ljava/io/IOException;
      :catchall_0
      move-exception v2
***************
*** 2176,2183 ****
--- 2463,2472 ----
      .parameter "value"
  
      .prologue
+     .line 1004
      if-eqz p0, :cond_1
  
+     .line 1005
      move-object v1, p0
  
      .local v1, arr$:[Ljava/lang/String;
***************
*** 2192,2197 ****
--- 2481,2487 ----
  
      aget-object v0, v1, v2
  
+     .line 1006
      .local v0, arg:Ljava/lang/String;
      invoke-virtual {p1, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 2199,2206 ****
--- 2489,2498 ----
  
      if-eqz v4, :cond_0
  
+     .line 1007
      const/4 v4, 0x1
  
+     .line 1011
      .end local v0           #arg:Ljava/lang/String;
      .end local v1           #arr$:[Ljava/lang/String;
      .end local v2           #i$:I
***************
*** 2208,2213 ****
--- 2500,2506 ----
      :goto_1
      return v4
  
+     .line 1005
      .restart local v0       #arg:Ljava/lang/String;
      .restart local v1       #arr$:[Ljava/lang/String;
      .restart local v2       #i$:I
***************
*** 2217,2222 ****
--- 2510,2516 ----
  
      goto :goto_0
  
+     .line 1011
      .end local v0           #arg:Ljava/lang/String;
      .end local v1           #arr$:[Ljava/lang/String;
      .end local v2           #i$:I
***************
*** 2235,2244 ****
--- 2529,2541 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1021
      if-eqz p0, :cond_0
  
+     .line 1022
      array-length v0, p0
  
+     .line 1023
      .local v0, N:I
      const/4 v1, 0x0
  
***************
*** 2246,2251 ****
--- 2543,2549 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 1024
      aget-object v3, p0, v1
  
      invoke-virtual {p1, v3}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
***************
*** 2254,2270 ****
--- 2552,2572 ----
  
      if-eqz v3, :cond_1
  
+     .line 1025
      add-int/lit8 v1, v1, 0x1
  
+     .line 1026
      if-ge v1, v0, :cond_0
  
      aget-object v2, p0, v1
  
+     .line 1030
      .end local v0           #N:I
      .end local v1           #i:I
      :cond_0
      return-object v2
  
+     .line 1023
      .restart local v0       #N:I
      .restart local v1       #i:I
      :cond_1
***************
*** 2278,2300 ****
--- 2580,2607 ----
      .parameter "historyFile"
  
      .prologue
+     .line 608
      const/4 v2, 0x0
  
+     .line 610
      .local v2, fos:Ljava/io/FileOutputStream;
      :try_start_0
      invoke-virtual {p1}, Lcom/android/internal/os/AtomicFile;->startWrite()Ljava/io/FileOutputStream;
  
      move-result-object v2
  
+     .line 611
      new-instance v5, Lcom/android/internal/util/FastXmlSerializer;
  
      invoke-direct {v5}, Lcom/android/internal/util/FastXmlSerializer;-><init>()V
  
+     .line 612
      .local v5, out:Lorg/xmlpull/v1/XmlSerializer;
      const-string v7, "utf-8"
  
      invoke-interface {v5, v2, v7}, Lorg/xmlpull/v1/XmlSerializer;->setOutput(Ljava/io/OutputStream;Ljava/lang/String;)V
  
+     .line 613
      const/4 v7, 0x0
  
      const/4 v8, 0x1
***************
*** 2305,2328 ****
--- 2612,2639 ----
  
      invoke-interface {v5, v7, v8}, Lorg/xmlpull/v1/XmlSerializer;->startDocument(Ljava/lang/String;Ljava/lang/Boolean;)V
  
+     .line 614
      const-string v7, "http://xmlpull.org/v1/doc/features.html#indent-output"
  
      const/4 v8, 0x1
  
      invoke-interface {v5, v7, v8}, Lorg/xmlpull/v1/XmlSerializer;->setFeature(Ljava/lang/String;Z)V
  
+     .line 615
      const/4 v7, 0x0
  
      const-string v8, "usage-history"
  
      invoke-interface {v5, v7, v8}, Lorg/xmlpull/v1/XmlSerializer;->startTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 616
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
      monitor-enter v8
      :try_end_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 617
      :try_start_1
      iget-object v7, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
***************
*** 2347,2352 ****
--- 2658,2664 ----
  
      check-cast v6, Ljava/util/Map$Entry;
  
+     .line 618
      .local v6, pkgEntry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;>;"
      const/4 v7, 0x0
  
***************
*** 2354,2359 ****
--- 2666,2672 ----
  
      invoke-interface {v5, v7, v9}, Lorg/xmlpull/v1/XmlSerializer;->startTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 619
      const/4 v9, 0x0
  
      const-string v10, "name"
***************
*** 2366,2371 ****
--- 2679,2685 ----
  
      invoke-interface {v5, v9, v10, v7}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 620
      invoke-interface {v6}, Ljava/util/Map$Entry;->getValue()Ljava/lang/Object;
  
      move-result-object v7
***************
*** 2394,2399 ****
--- 2708,2714 ----
  
      check-cast v0, Ljava/util/Map$Entry;
  
+     .line 621
      .local v0, compEntry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/lang/Long;>;"
      const/4 v7, 0x0
  
***************
*** 2401,2406 ****
--- 2716,2722 ----
  
      invoke-interface {v5, v7, v9}, Lorg/xmlpull/v1/XmlSerializer;->startTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 622
      const/4 v9, 0x0
  
      const-string v10, "name"
***************
*** 2413,2418 ****
--- 2729,2735 ----
  
      invoke-interface {v5, v9, v10, v7}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 623
      const/4 v9, 0x0
  
      const-string v10, "lrt"
***************
*** 2429,2434 ****
--- 2746,2752 ----
  
      invoke-interface {v5, v9, v10, v7}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 624
      const/4 v7, 0x0
  
      const-string v9, "comp"
***************
*** 2437,2442 ****
--- 2755,2761 ----
  
      goto :goto_1
  
+     .line 628
      .end local v0           #compEntry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/lang/Long;>;"
      .end local v4           #i$:Ljava/util/Iterator;
      .end local v6           #pkgEntry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;>;"
***************
*** 2452,2461 ****
--- 2771,2782 ----
      :try_end_2
      .catch Ljava/io/IOException; {:try_start_2 .. :try_end_2} :catch_0
  
+     .line 633
      .end local v5           #out:Lorg/xmlpull/v1/XmlSerializer;
      :catch_0
      move-exception v1
  
+     .line 634
      .local v1, e:Ljava/io/IOException;
      const-string v7, "UsageStats"
  
***************
*** 2479,2493 ****
--- 2800,2818 ----
  
      invoke-static {v7, v8}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 635
      if-eqz v2, :cond_0
  
+     .line 636
      invoke-virtual {p1, v2}, Lcom/android/internal/os/AtomicFile;->failWrite(Ljava/io/FileOutputStream;)V
  
+     .line 639
      .end local v1           #e:Ljava/io/IOException;
      :cond_0
      :goto_2
      return-void
  
+     .line 626
      .restart local v4       #i$:Ljava/util/Iterator;
      .restart local v5       #out:Lorg/xmlpull/v1/XmlSerializer;
      .restart local v6       #pkgEntry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;>;"
***************
*** 2501,2506 ****
--- 2826,2832 ----
  
      goto/16 :goto_0
  
+     .line 628
      .end local v4           #i$:Ljava/util/Iterator;
      .end local v6           #pkgEntry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;>;"
      :cond_2
***************
*** 2508,2513 ****
--- 2834,2840 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
+     .line 629
      const/4 v7, 0x0
  
      :try_start_4
***************
*** 2515,2522 ****
--- 2842,2851 ----
  
      invoke-interface {v5, v7, v8}, Lorg/xmlpull/v1/XmlSerializer;->endTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 630
      invoke-interface {v5}, Lorg/xmlpull/v1/XmlSerializer;->endDocument()V
  
+     .line 632
      invoke-virtual {p1, v2}, Lcom/android/internal/os/AtomicFile;->finishWrite(Ljava/io/FileOutputStream;)V
      :try_end_4
      .catch Ljava/io/IOException; {:try_start_4 .. :try_end_4} :catch_0
***************
*** 2534,2576 ****
--- 2863,2916 ----
      .end annotation
  
      .prologue
+     .line 565
      new-instance v1, Ljava/io/FileOutputStream;
  
      invoke-direct {v1, p1}, Ljava/io/FileOutputStream;-><init>(Ljava/io/File;)V
  
+     .line 567
      .local v1, stream:Ljava/io/FileOutputStream;
      :try_start_0
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 568
      .local v0, out:Landroid/os/Parcel;
      invoke-direct {p0, v0}, Lcom/android/server/am/UsageStatsService;->writeStatsToParcelFLOCK(Landroid/os/Parcel;)V
  
+     .line 569
      invoke-virtual {v0}, Landroid/os/Parcel;->marshall()[B
  
      move-result-object v2
  
      invoke-virtual {v1, v2}, Ljava/io/FileOutputStream;->write([B)V
  
+     .line 570
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 571
      invoke-virtual {v1}, Ljava/io/FileOutputStream;->flush()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 573
      invoke-static {v1}, Landroid/os/FileUtils;->sync(Ljava/io/FileOutputStream;)Z
  
+     .line 574
      invoke-virtual {v1}, Ljava/io/FileOutputStream;->close()V
  
+     .line 576
      return-void
  
+     .line 573
      .end local v0           #out:Landroid/os/Parcel;
      :catchall_0
      move-exception v2
  
      invoke-static {v1}, Landroid/os/FileUtils;->sync(Ljava/io/FileOutputStream;)Z
  
+     .line 574
      invoke-virtual {v1}, Ljava/io/FileOutputStream;->close()V
  
      throw v2
***************
*** 2586,2595 ****
--- 2926,2937 ----
  
      const/4 v7, 0x0
  
+     .line 478
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
  
      monitor-enter v8
  
+     .line 479
      :try_start_0
      iget-object v9, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
  
***************
*** 2599,2604 ****
--- 2941,2947 ----
  
      invoke-virtual {v9, v10, v11}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 480
      iget-object v9, p0, Lcom/android/server/am/UsageStatsService;->mCal:Ljava/util/Calendar;
  
      const/4 v10, 0x6
***************
*** 2607,2617 ****
--- 2950,2962 ----
  
      move-result v1
  
+     .line 481
      .local v1, curDay:I
      monitor-exit v8
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 482
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mLastWriteDay:Ljava/util/concurrent/atomic/AtomicInteger;
  
      invoke-virtual {v8}, Ljava/util/concurrent/atomic/AtomicInteger;->get()I
***************
*** 2622,2636 ****
--- 2967,2984 ----
  
      move v4, v6
  
+     .line 487
      .local v4, dayChanged:Z
      :goto_0
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v2
  
+     .line 491
      .local v2, currElapsedTime:J
      if-nez p1, :cond_3
  
+     .line 492
      if-nez v4, :cond_2
  
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mLastWriteElapsedTime:Ljava/util/concurrent/atomic/AtomicLong;
***************
*** 2647,2656 ****
--- 2995,3006 ----
  
      if-gez v8, :cond_2
  
+     .line 562
      :cond_0
      :goto_1
      return-void
  
+     .line 481
      .end local v1           #curDay:I
      .end local v2           #currElapsedTime:J
      .end local v4           #dayChanged:Z
***************
*** 2668,2675 ****
--- 3018,3027 ----
      :cond_1
      move v4, v7
  
+     .line 482
      goto :goto_0
  
+     .line 497
      .restart local v2       #currElapsedTime:J
      .restart local v4       #dayChanged:Z
      :cond_2
***************
*** 2681,2686 ****
--- 3033,3039 ----
  
      if-eqz v6, :cond_0
  
+     .line 498
      new-instance v6, Lcom/android/server/am/UsageStatsService$1;
  
      const-string v7, "UsageStatsService_DiskWriter"
***************
*** 2691,2701 ****
--- 3044,3056 ----
  
      goto :goto_1
  
+     .line 513
      :cond_3
      iget-object v7, p0, Lcom/android/server/am/UsageStatsService;->mFileLock:Ljava/lang/Object;
  
      monitor-enter v7
  
+     .line 515
      :try_start_2
      const-string v6, "usage-"
  
***************
*** 2705,2712 ****
--- 3060,3069 ----
  
      iput-object v6, p0, Lcom/android/server/am/UsageStatsService;->mFileLeaf:Ljava/lang/String;
  
+     .line 517
      const/4 v0, 0x0
  
+     .line 518
      .local v0, backupFile:Ljava/io/File;
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mFile:Ljava/io/File;
  
***************
*** 2720,2725 ****
--- 3077,3083 ----
  
      if-eqz v6, :cond_5
  
+     .line 519
      new-instance v0, Ljava/io/File;
  
      .end local v0           #backupFile:Ljava/io/File;
***************
*** 2749,2754 ****
--- 3107,3113 ----
  
      invoke-direct {v0, v6}, Ljava/io/File;-><init>(Ljava/lang/String;)V
  
+     .line 520
      .restart local v0       #backupFile:Ljava/io/File;
      invoke-virtual {v0}, Ljava/io/File;->exists()Z
  
***************
*** 2756,2761 ****
--- 3115,3121 ----
  
      if-nez v6, :cond_4
  
+     .line 521
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mFile:Ljava/io/File;
  
      invoke-virtual {v6, v0}, Ljava/io/File;->renameTo(Ljava/io/File;)Z
***************
*** 2764,2779 ****
--- 3124,3142 ----
  
      if-nez v6, :cond_5
  
+     .line 522
      const-string v6, "UsageStats"
  
      const-string v8, "Failed to persist new stats"
  
      invoke-static {v6, v8}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 523
      monitor-exit v7
  
      goto :goto_1
  
+     .line 560
      .end local v0           #backupFile:Ljava/io/File;
      :catchall_1
      move-exception v6
***************
*** 2784,2789 ****
--- 3147,3153 ----
  
      throw v6
  
+     .line 526
      .restart local v0       #backupFile:Ljava/io/File;
      :cond_4
      :try_start_3
***************
*** 2793,2814 ****
--- 3157,3183 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_1
  
+     .line 532
      :cond_5
      :try_start_4
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mFile:Ljava/io/File;
  
      invoke-direct {p0, v6}, Lcom/android/server/am/UsageStatsService;->writeStatsFLOCK(Ljava/io/File;)V
  
+     .line 533
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mLastWriteElapsedTime:Ljava/util/concurrent/atomic/AtomicLong;
  
      invoke-virtual {v6, v2, v3}, Ljava/util/concurrent/atomic/AtomicLong;->set(J)V
  
+     .line 534
      if-eqz v4, :cond_6
  
+     .line 535
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mLastWriteDay:Ljava/util/concurrent/atomic/AtomicInteger;
  
      invoke-virtual {v6, v1}, Ljava/util/concurrent/atomic/AtomicInteger;->set(I)V
  
+     .line 537
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
      monitor-enter v8
***************
*** 2816,2830 ****
--- 3185,3202 ----
      .catchall {:try_start_4 .. :try_end_4} :catchall_1
      .catch Ljava/io/IOException; {:try_start_4 .. :try_end_4} :catch_0
  
+     .line 538
      :try_start_5
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
      invoke-interface {v6}, Ljava/util/Map;->clear()V
  
+     .line 539
      monitor-exit v8
      :try_end_5
      .catchall {:try_start_5 .. :try_end_5} :catchall_2
  
+     .line 540
      :try_start_6
      new-instance v6, Ljava/io/File;
  
***************
*** 2836,2861 ****
--- 3208,3239 ----
  
      iput-object v6, p0, Lcom/android/server/am/UsageStatsService;->mFile:Ljava/io/File;
  
+     .line 541
      invoke-direct {p0}, Lcom/android/server/am/UsageStatsService;->checkFileLimitFLOCK()V
  
+     .line 544
      :cond_6
      if-nez v4, :cond_7
  
      if-eqz p2, :cond_8
  
+     .line 546
      :cond_7
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mHistoryFile:Lcom/android/internal/os/AtomicFile;
  
      invoke-direct {p0, v6}, Lcom/android/server/am/UsageStatsService;->writeHistoryStatsFLOCK(Lcom/android/internal/os/AtomicFile;)V
  
+     .line 550
      :cond_8
      if-eqz v0, :cond_9
  
+     .line 551
      invoke-virtual {v0}, Ljava/io/File;->delete()Z
      :try_end_6
      .catchall {:try_start_6 .. :try_end_6} :catchall_1
      .catch Ljava/io/IOException; {:try_start_6 .. :try_end_6} :catch_0
  
+     .line 560
      :cond_9
      :goto_2
      :try_start_7
***************
*** 2865,2870 ****
--- 3243,3249 ----
  
      goto/16 :goto_1
  
+     .line 539
      :catchall_2
      move-exception v6
  
***************
*** 2879,2887 ****
--- 3258,3268 ----
      .catchall {:try_start_9 .. :try_end_9} :catchall_1
      .catch Ljava/io/IOException; {:try_start_9 .. :try_end_9} :catch_0
  
+     .line 553
      :catch_0
      move-exception v5
  
+     .line 554
      .local v5, e:Ljava/io/IOException;
      :try_start_a
      const-string v6, "UsageStats"
***************
*** 2908,2919 ****
--- 3289,3303 ----
  
      invoke-static {v6, v8}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 555
      if-eqz v0, :cond_9
  
+     .line 556
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mFile:Ljava/io/File;
  
      invoke-virtual {v6}, Ljava/io/File;->delete()Z
  
+     .line 557
      iget-object v6, p0, Lcom/android/server/am/UsageStatsService;->mFile:Ljava/io/File;
  
      invoke-virtual {v0, v6}, Ljava/io/File;->renameTo(Ljava/io/File;)Z
***************
*** 2928,2948 ****
--- 3312,3336 ----
      .parameter "out"
  
      .prologue
+     .line 579
      iget-object v5, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
      monitor-enter v5
  
+     .line 580
      const/16 v4, 0x3ef
  
      :try_start_0
      invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 581
      iget-object v4, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
      invoke-interface {v4}, Ljava/util/Map;->keySet()Ljava/util/Set;
  
      move-result-object v2
  
+     .line 582
      .local v2, keys:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
      invoke-interface {v2}, Ljava/util/Set;->size()I
  
***************
*** 2950,2955 ****
--- 3338,3344 ----
  
      invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 583
      invoke-interface {v2}, Ljava/util/Set;->iterator()Ljava/util/Iterator;
  
      move-result-object v0
***************
*** 2968,2973 ****
--- 3357,3363 ----
  
      check-cast v1, Ljava/lang/String;
  
+     .line 584
      .local v1, key:Ljava/lang/String;
      iget-object v4, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
***************
*** 2977,2989 ****
--- 3367,3382 ----
  
      check-cast v3, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
+     .line 585
      .local v3, pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 586
      invoke-virtual {v3, p1}, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->writeToParcel(Landroid/os/Parcel;)V
  
      goto :goto_0
  
+     .line 588
      .end local v0           #i$:Ljava/util/Iterator;
      .end local v1           #key:Ljava/lang/String;
      .end local v2           #keys:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
***************
*** 3005,3010 ****
--- 3398,3404 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 589
      return-void
  .end method
  
***************
*** 3017,3022 ****
--- 3411,3417 ----
      .parameter "args"
  
      .prologue
+     .line 1038
      iget-object v10, p0, Lcom/android/server/am/UsageStatsService;->mContext:Landroid/content/Context;
  
      const-string v11, "android.permission.DUMP"
***************
*** 3027,3032 ****
--- 3422,3428 ----
  
      if-eqz v10, :cond_0
  
+     .line 1040
      new-instance v10, Ljava/lang/StringBuilder;
  
      invoke-direct {v10}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 3077,3085 ****
--- 3473,3483 ----
  
      invoke-virtual {p2, v10}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 1080
      :goto_0
      return-void
  
+     .line 1046
      :cond_0
      const-string v10, "--checkin"
  
***************
*** 3087,3092 ****
--- 3485,3491 ----
  
      move-result v3
  
+     .line 1047
      .local v3, isCheckinRequest:Z
      if-nez v3, :cond_1
  
***************
*** 3101,3106 ****
--- 3500,3506 ----
      :cond_1
      const/4 v4, 0x1
  
+     .line 1048
      .local v4, isCompactOutput:Z
      :goto_1
      if-nez v3, :cond_2
***************
*** 3116,3121 ****
--- 3516,3522 ----
      :cond_2
      const/4 v1, 0x1
  
+     .line 1049
      .local v1, deleteAfterPrint:Z
      :goto_2
      const-string v10, "--packages"
***************
*** 3124,3144 ****
--- 3525,3550 ----
  
      move-result-object v9
  
+     .line 1054
      .local v9, rawPackages:Ljava/lang/String;
      if-nez v1, :cond_3
  
+     .line 1055
      const/4 v10, 0x1
  
      const/4 v11, 0x0
  
      invoke-direct {p0, v10, v11}, Lcom/android/server/am/UsageStatsService;->writeStatsToFile(ZZ)V
  
+     .line 1058
      :cond_3
      const/4 v8, 0x0
  
+     .line 1059
      .local v8, packages:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/String;>;"
      if-eqz v9, :cond_7
  
+     .line 1060
      const-string v10, "*"
  
      invoke-virtual {v10, v9}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
***************
*** 3147,3158 ****
--- 3553,3566 ----
  
      if-nez v10, :cond_8
  
+     .line 1062
      const-string v10, ","
  
      invoke-virtual {v9, v10}, Ljava/lang/String;->split(Ljava/lang/String;)[Ljava/lang/String;
  
      move-result-object v7
  
+     .line 1063
      .local v7, names:[Ljava/lang/String;
      move-object v0, v7
  
***************
*** 3168,3189 ****
--- 3576,3602 ----
  
      aget-object v6, v0, v2
  
+     .line 1064
      .local v6, n:Ljava/lang/String;
      if-nez v8, :cond_4
  
+     .line 1065
      new-instance v8, Ljava/util/HashSet;
  
      .end local v8           #packages:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/String;>;"
      invoke-direct {v8}, Ljava/util/HashSet;-><init>()V
  
+     .line 1067
      .restart local v8       #packages:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/String;>;"
      :cond_4
      invoke-virtual {v8, v6}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 1063
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_3
  
+     .line 1047
      .end local v0           #arr$:[Ljava/lang/String;
      .end local v1           #deleteAfterPrint:Z
      .end local v2           #i$:I
***************
*** 3198,3215 ****
--- 3611,3631 ----
  
      goto :goto_1
  
+     .line 1048
      .restart local v4       #isCompactOutput:Z
      :cond_6
      const/4 v1, 0x0
  
      goto :goto_2
  
+     .line 1070
      .restart local v1       #deleteAfterPrint:Z
      .restart local v8       #packages:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/String;>;"
      .restart local v9       #rawPackages:Ljava/lang/String;
      :cond_7
      if-eqz v3, :cond_8
  
+     .line 1073
      const-string v10, "UsageStats"
  
      const-string v11, "Checkin without packages"
***************
*** 3218,3231 ****
--- 3634,3650 ----
  
      goto :goto_0
  
+     .line 1077
      :cond_8
      iget-object v11, p0, Lcom/android/server/am/UsageStatsService;->mFileLock:Ljava/lang/Object;
  
      monitor-enter v11
  
+     .line 1078
      :try_start_0
      invoke-direct {p0, p2, v4, v1, v8}, Lcom/android/server/am/UsageStatsService;->collectDumpInfoFLOCK(Ljava/io/PrintWriter;ZZLjava/util/HashSet;)V
  
+     .line 1079
      monitor-exit v11
  
      goto :goto_0
***************
*** 3244,3249 ****
--- 3663,3669 ----
      .locals 5
  
      .prologue
+     .line 781
      invoke-static {}, Landroid/os/Binder;->getCallingPid()I
  
      move-result v0
***************
*** 3254,3262 ****
--- 3674,3684 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 786
      :goto_0
      return-void
  
+     .line 784
      :cond_0
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mContext:Landroid/content/Context;
  
***************
*** 3283,3298 ****
--- 3705,3723 ----
      .prologue
      const/4 v10, 0x0
  
+     .line 809
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mContext:Landroid/content/Context;
  
      const-string v5, "android.permission.PACKAGE_USAGE_STATS"
  
      invoke-virtual {v0, v5, v10}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 811
      iget-object v12, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
      monitor-enter v12
  
+     .line 812
      :try_start_0
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
***************
*** 3300,3319 ****
--- 3725,3750 ----
  
      move-result v11
  
+     .line 813
      .local v11, size:I
      if-gtz v11, :cond_0
  
+     .line 814
      monitor-exit v12
  
+     .line 831
      :goto_0
      return-object v10
  
+     .line 816
      :cond_0
      new-array v10, v11, [Lcom/android/internal/os/PkgUsageStats;
  
+     .line 817
      .local v10, retArr:[Lcom/android/internal/os/PkgUsageStats;
      const/4 v7, 0x0
  
+     .line 818
      .local v7, i:I
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
***************
*** 3339,3344 ****
--- 3770,3776 ----
  
      check-cast v6, Ljava/util/Map$Entry;
  
+     .line 819
      .local v6, entry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;>;"
      invoke-interface {v6}, Ljava/util/Map$Entry;->getKey()Ljava/lang/Object;
  
***************
*** 3346,3357 ****
--- 3778,3792 ----
  
      check-cast v1, Ljava/lang/String;
  
+     .line 820
      .local v1, pkg:Ljava/lang/String;
      const-wide/16 v3, 0x0
  
+     .line 821
      .local v3, usageTime:J
      const/4 v2, 0x0
  
+     .line 823
      .local v2, launchCount:I
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
***************
*** 3361,3373 ****
--- 3796,3812 ----
  
      check-cast v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
+     .line 824
      .local v9, pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      if-eqz v9, :cond_1
  
+     .line 825
      iget-wide v3, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mUsageTime:J
  
+     .line 826
      iget v2, v9, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mLaunchCount:I
  
+     .line 828
      :cond_1
      new-instance v0, Lcom/android/internal/os/PkgUsageStats;
  
***************
*** 3381,3390 ****
--- 3820,3832 ----
  
      aput-object v0, v10, v7
  
+     .line 829
      add-int/lit8 v7, v7, 0x1
  
+     .line 830
      goto :goto_1
  
+     .line 831
      .end local v1           #pkg:Ljava/lang/String;
      .end local v2           #launchCount:I
      .end local v3           #usageTime:J
***************
*** 3395,3400 ****
--- 3837,3843 ----
  
      goto :goto_0
  
+     .line 832
      .end local v7           #i:I
      .end local v8           #i$:Ljava/util/Iterator;
      .end local v10           #retArr:[Lcom/android/internal/os/PkgUsageStats;
***************
*** 3416,3427 ****
--- 3859,3872 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 789
      iget-object v7, p0, Lcom/android/server/am/UsageStatsService;->mContext:Landroid/content/Context;
  
      const-string v8, "android.permission.PACKAGE_USAGE_STATS"
  
      invoke-virtual {v7, v8, v0}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 792
      if-eqz p1, :cond_0
  
      invoke-virtual {p1}, Landroid/content/ComponentName;->getPackageName()Ljava/lang/String;
***************
*** 3431,3447 ****
--- 3876,3895 ----
      .local v1, pkgName:Ljava/lang/String;
      if-nez v1, :cond_1
  
+     .line 804
      .end local v1           #pkgName:Ljava/lang/String;
      :cond_0
      :goto_0
      return-object v0
  
+     .line 796
      .restart local v1       #pkgName:Ljava/lang/String;
      :cond_1
      iget-object v7, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
      monitor-enter v7
  
+     .line 797
      :try_start_0
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
***************
*** 3451,3456 ****
--- 3899,3905 ----
  
      check-cast v6, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
+     .line 798
      .local v6, pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
***************
*** 3460,3474 ****
--- 3909,3926 ----
  
      check-cast v5, Ljava/util/Map;
  
+     .line 799
      .local v5, lastResumeTimes:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;"
      if-nez v6, :cond_2
  
      if-nez v5, :cond_2
  
+     .line 800
      monitor-exit v7
  
      goto :goto_0
  
+     .line 805
      .end local v5           #lastResumeTimes:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;"
      .end local v6           #pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      :catchall_0
***************
*** 3480,3485 ****
--- 3932,3938 ----
  
      throw v0
  
+     .line 802
      .restart local v5       #lastResumeTimes:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;"
      .restart local v6       #pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      :cond_2
***************
*** 3488,3499 ****
--- 3941,3954 ----
      :try_start_1
      iget v2, v6, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mLaunchCount:I
  
+     .line 803
      .local v2, launchCount:I
      :goto_1
      if-eqz v6, :cond_4
  
      iget-wide v3, v6, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->mUsageTime:J
  
+     .line 804
      .local v3, usageTime:J
      :goto_2
      new-instance v0, Lcom/android/internal/os/PkgUsageStats;
***************
*** 3506,3511 ****
--- 3961,3967 ----
  
      goto :goto_0
  
+     .line 802
      .end local v2           #launchCount:I
      .end local v3           #usageTime:J
      :cond_3
***************
*** 3513,3518 ****
--- 3969,3975 ----
  
      goto :goto_1
  
+     .line 803
      .restart local v2       #launchCount:I
      :cond_4
      const-wide/16 v3, 0x0
***************
*** 3524,3535 ****
--- 3981,3994 ----
      .locals 3
  
      .prologue
+     .line 651
      new-instance v0, Lcom/android/server/am/UsageStatsService$2;
  
      invoke-direct {v0, p0}, Lcom/android/server/am/UsageStatsService$2;-><init>(Lcom/android/server/am/UsageStatsService;)V
  
      iput-object v0, p0, Lcom/android/server/am/UsageStatsService;->mPackageMonitor:Lcom/android/internal/content/PackageMonitor;
  
+     .line 659
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mPackageMonitor:Lcom/android/internal/content/PackageMonitor;
  
      iget-object v1, p0, Lcom/android/server/am/UsageStatsService;->mContext:Landroid/content/Context;
***************
*** 3538,3545 ****
--- 3997,4006 ----
  
      invoke-virtual {v0, v1, v2}, Lcom/android/internal/content/PackageMonitor;->register(Landroid/content/Context;Z)V
  
+     .line 660
      invoke-direct {p0}, Lcom/android/server/am/UsageStatsService;->filterHistoryStats()V
  
+     .line 661
      return-void
  .end method
  
***************
*** 3551,3558 ****
--- 4012,4021 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 762
      invoke-virtual {p0}, Lcom/android/server/am/UsageStatsService;->enforceCallingPermission()V
  
+     .line 764
      if-eqz p1, :cond_0
  
      invoke-virtual {p1}, Landroid/content/ComponentName;->getPackageName()Ljava/lang/String;
***************
*** 3562,3580 ****
--- 4025,4047 ----
      .local v0, pkgName:Ljava/lang/String;
      if-nez v0, :cond_1
  
+     .line 778
      .end local v0           #pkgName:Ljava/lang/String;
      :cond_0
      :goto_0
      return-void
  
+     .line 770
      .restart local v0       #pkgName:Ljava/lang/String;
      :cond_1
      invoke-direct {p0, v2, v2}, Lcom/android/server/am/UsageStatsService;->writeStatsToFile(ZZ)V
  
+     .line 772
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
      monitor-enter v3
  
+     .line 773
      :try_start_0
      iget-object v2, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
***************
*** 3584,3598 ****
--- 4051,4068 ----
  
      check-cast v1, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
+     .line 774
      .local v1, pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      if-eqz v1, :cond_2
  
+     .line 775
      invoke-virtual {p1}, Landroid/content/ComponentName;->getClassName()Ljava/lang/String;
  
      move-result-object v2
  
      invoke-virtual {v1, v2, p2}, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->addLaunchTime(Ljava/lang/String;I)V
  
+     .line 777
      :cond_2
      monitor-exit v3
  
***************
*** 3616,3627 ****
--- 4086,4100 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 731
      invoke-virtual {p0}, Lcom/android/server/am/UsageStatsService;->enforceCallingPermission()V
  
+     .line 733
      iget-object v3, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
      monitor-enter v3
  
+     .line 735
      if-eqz p1, :cond_0
  
      :try_start_0
***************
*** 3632,3654 ****
--- 4105,4132 ----
      .local v0, pkgName:Ljava/lang/String;
      if-nez v0, :cond_1
  
+     .line 737
      .end local v0           #pkgName:Ljava/lang/String;
      :cond_0
      monitor-exit v3
  
+     .line 759
      :goto_0
      return-void
  
+     .line 739
      .restart local v0       #pkgName:Ljava/lang/String;
      :cond_1
      iget-boolean v2, p0, Lcom/android/server/am/UsageStatsService;->mIsResumed:Z
  
      if-nez v2, :cond_2
  
+     .line 742
      monitor-exit v3
  
      goto :goto_0
  
+     .line 755
      .end local v0           #pkgName:Ljava/lang/String;
      :catchall_0
      move-exception v2
***************
*** 3659,3664 ****
--- 4137,4143 ----
  
      throw v2
  
+     .line 744
      .restart local v0       #pkgName:Ljava/lang/String;
      :cond_2
      const/4 v2, 0x0
***************
*** 3666,3671 ****
--- 4145,4151 ----
      :try_start_1
      iput-boolean v2, p0, Lcom/android/server/am/UsageStatsService;->mIsResumed:Z
  
+     .line 748
      iget-object v2, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
      invoke-interface {v2, v0}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 3674,3682 ****
--- 4154,4164 ----
  
      check-cast v1, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
+     .line 749
      .local v1, pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      if-nez v1, :cond_3
  
+     .line 751
      const-string v2, "UsageStats"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 3699,3715 ****
--- 4181,4201 ----
  
      invoke-static {v2, v4}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 752
      monitor-exit v3
  
      goto :goto_0
  
+     .line 754
      :cond_3
      invoke-virtual {v1}, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->updatePause()V
  
+     .line 755
      monitor-exit v3
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 758
      invoke-direct {p0, v4, v4}, Lcom/android/server/am/UsageStatsService;->writeStatsToFile(ZZ)V
  
      goto :goto_0
***************
*** 3724,3735 ****
--- 4210,4224 ----
  
      const/4 v5, 0x1
  
+     .line 681
      invoke-virtual {p0}, Lcom/android/server/am/UsageStatsService;->enforceCallingPermission()V
  
+     .line 683
      iget-object v7, p0, Lcom/android/server/am/UsageStatsService;->mStatsLock:Ljava/lang/Object;
  
      monitor-enter v7
  
+     .line 684
      if-eqz p1, :cond_0
  
      :try_start_0
***************
*** 3740,3752 ****
--- 4229,4244 ----
      .local v1, pkgName:Ljava/lang/String;
      if-nez v1, :cond_1
  
+     .line 686
      .end local v1           #pkgName:Ljava/lang/String;
      :cond_0
      monitor-exit v7
  
+     .line 728
      :goto_0
      return-void
  
+     .line 689
      .restart local v1       #pkgName:Ljava/lang/String;
      :cond_1
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mLastResumedPkg:Ljava/lang/String;
***************
*** 3755,3769 ****
--- 4247,4264 ----
  
      move-result v4
  
+     .line 690
      .local v4, samePackage:Z
      iget-boolean v8, p0, Lcom/android/server/am/UsageStatsService;->mIsResumed:Z
  
      if-eqz v8, :cond_2
  
+     .line 691
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mLastResumedPkg:Ljava/lang/String;
  
      if-eqz v8, :cond_2
  
+     .line 696
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
      iget-object v9, p0, Lcom/android/server/am/UsageStatsService;->mLastResumedPkg:Ljava/lang/String;
***************
*** 3774,3784 ****
--- 4269,4282 ----
  
      check-cast v2, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
+     .line 697
      .local v2, pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      if-eqz v2, :cond_2
  
+     .line 698
      invoke-virtual {v2}, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->updatePause()V
  
+     .line 703
      .end local v2           #pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      :cond_2
      if-eqz v4, :cond_6
***************
*** 3797,3816 ****
--- 4295,4318 ----
  
      move v3, v5
  
+     .line 706
      .local v3, sameComp:Z
      :goto_1
      const/4 v8, 0x1
  
      iput-boolean v8, p0, Lcom/android/server/am/UsageStatsService;->mIsResumed:Z
  
+     .line 707
      iput-object v1, p0, Lcom/android/server/am/UsageStatsService;->mLastResumedPkg:Ljava/lang/String;
  
+     .line 708
      invoke-virtual {p1}, Landroid/content/ComponentName;->getClassName()Ljava/lang/String;
  
      move-result-object v8
  
      iput-object v8, p0, Lcom/android/server/am/UsageStatsService;->mLastResumedComp:Ljava/lang/String;
  
+     .line 711
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
      invoke-interface {v8, v1}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 3819,3837 ****
--- 4321,4343 ----
  
      check-cast v2, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
+     .line 712
      .restart local v2       #pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      if-nez v2, :cond_3
  
+     .line 713
      new-instance v2, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
  
      .end local v2           #pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      invoke-direct {v2, p0}, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;-><init>(Lcom/android/server/am/UsageStatsService;)V
  
+     .line 714
      .restart local v2       #pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mStats:Ljava/util/Map;
  
      invoke-interface {v8, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 716
      :cond_3
      iget-object v8, p0, Lcom/android/server/am/UsageStatsService;->mLastResumedComp:Ljava/lang/String;
  
***************
*** 3840,3851 ****
--- 4346,4360 ----
      :goto_2
      invoke-virtual {v2, v8, v5}, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->updateResume(Ljava/lang/String;Z)V
  
+     .line 717
      if-nez v3, :cond_4
  
+     .line 718
      iget-object v5, p0, Lcom/android/server/am/UsageStatsService;->mLastResumedComp:Ljava/lang/String;
  
      invoke-virtual {v2, v5}, Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;->addLaunchCount(Ljava/lang/String;)V
  
+     .line 721
      :cond_4
      iget-object v5, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
***************
*** 3855,3873 ****
--- 4364,4386 ----
  
      check-cast v0, Ljava/util/Map;
  
+     .line 722
      .local v0, componentResumeTimes:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;"
      if-nez v0, :cond_5
  
+     .line 723
      new-instance v0, Ljava/util/HashMap;
  
      .end local v0           #componentResumeTimes:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;"
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
+     .line 724
      .restart local v0       #componentResumeTimes:Ljava/util/Map;,"Ljava/util/Map<Ljava/lang/String;Ljava/lang/Long;>;"
      iget-object v5, p0, Lcom/android/server/am/UsageStatsService;->mLastResumeTimes:Ljava/util/Map;
  
      invoke-interface {v5, v1, v0}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 726
      :cond_5
      iget-object v5, p0, Lcom/android/server/am/UsageStatsService;->mLastResumedComp:Ljava/lang/String;
  
***************
*** 3881,3886 ****
--- 4394,4400 ----
  
      invoke-interface {v0, v5, v6}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 727
      monitor-exit v7
  
      goto :goto_0
***************
*** 3904,3909 ****
--- 4418,4424 ----
      :cond_6
      move v3, v6
  
+     .line 703
      goto :goto_1
  
      .restart local v2       #pus:Lcom/android/server/am/UsageStatsService$PkgUsageStatsExtended;
***************
*** 3911,3916 ****
--- 4426,4432 ----
      :cond_7
      move v5, v6
  
+     .line 716
      goto :goto_2
  .end method
  
***************
*** 3919,3926 ****
--- 4435,4444 ----
      .parameter "context"
  
      .prologue
+     .line 642
      iput-object p1, p0, Lcom/android/server/am/UsageStatsService;->mContext:Landroid/content/Context;
  
+     .line 643
      const-string v0, "usagestats"
  
      invoke-virtual {p0}, Lcom/android/server/am/UsageStatsService;->asBinder()Landroid/os/IBinder;
***************
*** 3929,3934 ****
--- 4447,4453 ----
  
      invoke-static {v0, v1}, Landroid/os/ServiceManager;->addService(Ljava/lang/String;Landroid/os/IBinder;)V
  
+     .line 644
      return-void
  .end method
  
***************
*** 3938,3951 ****
--- 4457,4473 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 664
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mPackageMonitor:Lcom/android/internal/content/PackageMonitor;
  
      if-eqz v0, :cond_0
  
+     .line 665
      iget-object v0, p0, Lcom/android/server/am/UsageStatsService;->mPackageMonitor:Lcom/android/internal/content/PackageMonitor;
  
      invoke-virtual {v0}, Lcom/android/internal/content/PackageMonitor;->unregister()V
  
+     .line 667
      :cond_0
      const-string v0, "UsageStats"
  
***************
*** 3953,3959 ****
--- 4475,4483 ----
  
      invoke-static {v0, v1}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 668
      invoke-direct {p0, v2, v2}, Lcom/android/server/am/UsageStatsService;->writeStatsToFile(ZZ)V
  
+     .line 669
      return-void
  .end method
