*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/database/SortCursor.smali	2012-03-23 19:37:34.182398397 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/database/SortCursor.smali.original	2012-03-23 19:34:46.108333571 -0500
***************
*** 36,91 ****
--- 36,104 ----
      .prologue
      const/16 v7, 0x40
  
+     .line 56
      invoke-direct {p0}, Landroid/database/AbstractCursor;-><init>()V
  
+     .line 34
      iput v7, p0, Lcom/android/internal/database/SortCursor;->ROWCACHESIZE:I
  
+     .line 35
      new-array v5, v7, [I
  
      iput-object v5, p0, Lcom/android/internal/database/SortCursor;->mRowNumCache:[I
  
+     .line 36
      new-array v5, v7, [I
  
      iput-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursorCache:[I
  
+     .line 38
      const/4 v5, -0x1
  
      iput v5, p0, Lcom/android/internal/database/SortCursor;->mLastCacheHit:I
  
+     .line 40
      new-instance v5, Lcom/android/internal/database/SortCursor$1;
  
      invoke-direct {v5, p0}, Lcom/android/internal/database/SortCursor$1;-><init>(Lcom/android/internal/database/SortCursor;)V
  
      iput-object v5, p0, Lcom/android/internal/database/SortCursor;->mObserver:Landroid/database/DataSetObserver;
  
+     .line 57
      iput-object p1, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
+     .line 59
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      array-length v3, v5
  
+     .line 60
      .local v3, length:I
      new-array v5, v3, [I
  
      iput-object v5, p0, Lcom/android/internal/database/SortCursor;->mSortColumns:[I
  
+     .line 61
      const/4 v1, 0x0
  
      .local v1, i:I
      :goto_0
      if-ge v1, v3, :cond_1
  
+     .line 62
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v5, v5, v1
  
      if-nez v5, :cond_0
  
+     .line 61
      :goto_1
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 65
      :cond_0
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
***************
*** 95,106 ****
--- 108,121 ----
  
      invoke-interface {v5, v6}, Landroid/database/Cursor;->registerDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 67
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v5, v5, v1
  
      invoke-interface {v5}, Landroid/database/Cursor;->moveToFirst()Z
  
+     .line 70
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mSortColumns:[I
  
      iget-object v6, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
***************
*** 115,127 ****
--- 130,145 ----
  
      goto :goto_1
  
+     .line 72
      :cond_1
      const/4 v5, 0x0
  
      iput-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
+     .line 73
      const-string v4, ""
  
+     .line 74
      .local v4, smallest:Ljava/lang/String;
      const/4 v2, 0x0
  
***************
*** 129,134 ****
--- 147,153 ----
      :goto_2
      if-ge v2, v3, :cond_5
  
+     .line 75
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v5, v5, v2
***************
*** 145,156 ****
--- 164,177 ----
  
      if-eqz v5, :cond_3
  
+     .line 74
      :cond_2
      :goto_3
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_2
  
+     .line 77
      :cond_3
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
***************
*** 164,169 ****
--- 185,191 ----
  
      move-result-object v0
  
+     .line 78
      .local v0, current:Ljava/lang/String;
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
***************
*** 175,183 ****
--- 197,207 ----
  
      if-gez v5, :cond_2
  
+     .line 79
      :cond_4
      move-object v4, v0
  
+     .line 80
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v5, v5, v2
***************
*** 186,191 ****
--- 210,216 ----
  
      goto :goto_3
  
+     .line 84
      .end local v0           #current:Ljava/lang/String;
      :cond_5
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mRowNumCache:[I
***************
*** 197,212 ****
--- 222,240 ----
      :goto_4
      if-ltz v1, :cond_6
  
+     .line 85
      iget-object v5, p0, Lcom/android/internal/database/SortCursor;->mRowNumCache:[I
  
      const/4 v6, -0x2
  
      aput v6, v5, v1
  
+     .line 84
      add-int/lit8 v1, v1, -0x1
  
      goto :goto_4
  
+     .line 87
      :cond_6
      filled-new-array {v7, v3}, [I
  
***************
*** 222,227 ****
--- 250,256 ----
  
      iput-object v5, p0, Lcom/android/internal/database/SortCursor;->mCurRowNumCache:[[I
  
+     .line 88
      return-void
  .end method
  
***************
*** 231,236 ****
--- 260,266 ----
      .parameter "x1"
  
      .prologue
+     .line 28
      iput p1, p0, Lcom/android/internal/database/SortCursor;->mPos:I
  
      return p1
***************
*** 242,247 ****
--- 272,278 ----
      .parameter "x1"
  
      .prologue
+     .line 28
      iput p1, p0, Lcom/android/internal/database/SortCursor;->mPos:I
  
      return p1
***************
*** 253,262 ****
--- 284,295 ----
      .locals 3
  
      .prologue
+     .line 267
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      array-length v1, v2
  
+     .line 268
      .local v1, length:I
      const/4 v0, 0x0
  
***************
*** 264,280 ****
--- 297,316 ----
      :goto_0
      if-ge v0, v1, :cond_1
  
+     .line 269
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v2, v2, v0
  
      if-nez v2, :cond_0
  
+     .line 268
      :goto_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 270
      :cond_0
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
***************
*** 284,289 ****
--- 320,326 ----
  
      goto :goto_1
  
+     .line 272
      :cond_1
      return-void
  .end method
***************
*** 292,301 ****
--- 329,340 ----
      .locals 3
  
      .prologue
+     .line 258
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      array-length v1, v2
  
+     .line 259
      .local v1, length:I
      const/4 v0, 0x0
  
***************
*** 303,319 ****
--- 342,361 ----
      :goto_0
      if-ge v0, v1, :cond_1
  
+     .line 260
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v2, v2, v0
  
      if-nez v2, :cond_0
  
+     .line 259
      :goto_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 261
      :cond_0
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
***************
*** 323,328 ****
--- 365,371 ----
  
      goto :goto_1
  
+     .line 263
      :cond_1
      return-void
  .end method
***************
*** 332,337 ****
--- 375,381 ----
      .parameter "column"
  
      .prologue
+     .line 234
      iget-object v0, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0, p1}, Landroid/database/Cursor;->getBlob(I)[B
***************
*** 345,368 ****
--- 389,417 ----
      .locals 4
  
      .prologue
+     .line 240
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      if-eqz v2, :cond_0
  
+     .line 241
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v2}, Landroid/database/Cursor;->getColumnNames()[Ljava/lang/String;
  
      move-result-object v2
  
+     .line 248
      :goto_0
      return-object v2
  
+     .line 245
      :cond_0
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      array-length v1, v2
  
+     .line 246
      .local v1, length:I
      const/4 v0, 0x0
  
***************
*** 370,381 ****
--- 419,432 ----
      :goto_1
      if-ge v0, v1, :cond_2
  
+     .line 247
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v2, v2, v0
  
      if-eqz v2, :cond_1
  
+     .line 248
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v2, v2, v0
***************
*** 386,396 ****
--- 437,449 ----
  
      goto :goto_0
  
+     .line 246
      :cond_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 251
      :cond_2
      new-instance v2, Ljava/lang/IllegalStateException;
  
***************
*** 405,417 ****
--- 458,473 ----
      .locals 4
  
      .prologue
+     .line 93
      const/4 v0, 0x0
  
+     .line 94
      .local v0, count:I
      iget-object v3, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      array-length v2, v3
  
+     .line 95
      .local v2, length:I
      const/4 v1, 0x0
  
***************
*** 419,430 ****
--- 475,488 ----
      :goto_0
      if-ge v1, v2, :cond_1
  
+     .line 96
      iget-object v3, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v3, v3, v1
  
      if-eqz v3, :cond_0
  
+     .line 97
      iget-object v3, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v3, v3, v1
***************
*** 435,445 ****
--- 493,505 ----
  
      add-int/2addr v0, v3
  
+     .line 95
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 100
      :cond_1
      return v0
  .end method
***************
*** 449,454 ****
--- 509,515 ----
      .parameter "column"
  
      .prologue
+     .line 217
      iget-object v0, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0, p1}, Landroid/database/Cursor;->getDouble(I)D
***************
*** 463,468 ****
--- 524,530 ----
      .parameter "column"
  
      .prologue
+     .line 211
      iget-object v0, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0, p1}, Landroid/database/Cursor;->getFloat(I)F
***************
*** 477,482 ****
--- 539,545 ----
      .parameter "column"
  
      .prologue
+     .line 199
      iget-object v0, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0, p1}, Landroid/database/Cursor;->getInt(I)I
***************
*** 491,496 ****
--- 554,560 ----
      .parameter "column"
  
      .prologue
+     .line 205
      iget-object v0, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0, p1}, Landroid/database/Cursor;->getLong(I)J
***************
*** 505,510 ****
--- 569,575 ----
      .parameter "column"
  
      .prologue
+     .line 193
      iget-object v0, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0, p1}, Landroid/database/Cursor;->getShort(I)S
***************
*** 519,524 ****
--- 584,590 ----
      .parameter "column"
  
      .prologue
+     .line 187
      iget-object v0, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0, p1}, Landroid/database/Cursor;->getString(I)Ljava/lang/String;
***************
*** 533,538 ****
--- 599,605 ----
      .parameter "column"
  
      .prologue
+     .line 222
      iget-object v0, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0, p1}, Landroid/database/Cursor;->getType(I)I
***************
*** 547,552 ****
--- 614,620 ----
      .parameter "column"
  
      .prologue
+     .line 228
      iget-object v0, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0, p1}, Landroid/database/Cursor;->isNull(I)Z
***************
*** 566,579 ****
--- 634,651 ----
  
      const/4 v8, 0x1
  
+     .line 106
      if-ne p1, p2, :cond_0
  
+     .line 181
      :goto_0
      return v8
  
+     .line 117
      :cond_0
      rem-int/lit8 v0, p2, 0x40
  
+     .line 119
      .local v0, cache_entry:I
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mRowNumCache:[I
  
***************
*** 581,590 ****
--- 653,664 ----
  
      if-ne v9, p2, :cond_2
  
+     .line 120
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursorCache:[I
  
      aget v7, v9, v0
  
+     .line 121
      .local v7, which:I
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
***************
*** 592,611 ****
  
      iput-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      if-nez v9, :cond_1
  
      const-string v8, "SortCursor"
  
!     const-string v9, "onMove: cache results in a null cursor."
  
      invoke-static {v8, v9}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
      const/4 v8, 0x0
  
      goto :goto_0
  
      :cond_1
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
--- 666,689 ----
  
      iput-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
+     .line 122
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
      if-nez v9, :cond_1
  
+     .line 123
      const-string v8, "SortCursor"
  
!     const-string/jumbo v9, "onMove: cache results in a null cursor."
  
      invoke-static {v8, v9}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 124
      const/4 v8, 0x0
  
      goto :goto_0
  
+     .line 126
      :cond_1
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
***************
*** 617,658 ****
--- 695,744 ----
  
      invoke-interface {v9, v10}, Landroid/database/Cursor;->moveToPosition(I)Z
  
+     .line 127
      iput v0, p0, Lcom/android/internal/database/SortCursor;->mLastCacheHit:I
  
      goto :goto_0
  
+     .line 131
      .end local v7           #which:I
      :cond_2
      const/4 v9, 0x0
  
      iput-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
+     .line 132
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      array-length v4, v9
  
+     .line 134
      .local v4, length:I
      iget v9, p0, Lcom/android/internal/database/SortCursor;->mLastCacheHit:I
  
      if-ltz v9, :cond_4
  
+     .line 135
      const/4 v2, 0x0
  
      .local v2, i:I
      :goto_1
      if-ge v2, v4, :cond_4
  
+     .line 136
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v9, v9, v2
  
      if-nez v9, :cond_3
  
+     .line 135
      :goto_2
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_1
  
+     .line 137
      :cond_3
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
***************
*** 670,681 ****
--- 756,769 ----
  
      goto :goto_2
  
+     .line 141
      .end local v2           #i:I
      :cond_4
      if-lt p2, p1, :cond_5
  
      if-ne p1, v12, :cond_8
  
+     .line 142
      :cond_5
      const/4 v2, 0x0
  
***************
*** 683,699 ****
--- 771,790 ----
      :goto_3
      if-ge v2, v4, :cond_7
  
+     .line 143
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v9, v9, v2
  
      if-nez v9, :cond_6
  
+     .line 142
      :goto_4
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_3
  
+     .line 144
      :cond_6
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
***************
*** 703,720 ****
--- 794,816 ----
  
      goto :goto_4
  
+     .line 146
      :cond_7
      const/4 p1, 0x0
  
+     .line 148
      .end local v2           #i:I
      :cond_8
      if-gez p1, :cond_9
  
+     .line 149
      const/4 p1, 0x0
  
+     .line 153
      :cond_9
      const/4 v6, -0x1
  
+     .line 154
      .local v6, smallestIdx:I
      move v2, p1
  
***************
*** 722,738 ****
--- 818,838 ----
      :goto_5
      if-gt v2, p2, :cond_e
  
+     .line 155
      const-string v5, ""
  
+     .line 156
      .local v5, smallest:Ljava/lang/String;
      const/4 v6, -0x1
  
+     .line 157
      const/4 v3, 0x0
  
      .local v3, j:I
      :goto_6
      if-ge v3, v4, :cond_d
  
+     .line 158
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v9, v9, v3
***************
*** 749,760 ****
--- 849,862 ----
  
      if-eqz v9, :cond_b
  
+     .line 157
      :cond_a
      :goto_7
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_6
  
+     .line 161
      :cond_b
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
***************
*** 768,773 ****
--- 870,876 ----
  
      move-result-object v1
  
+     .line 162
      .local v1, current:Ljava/lang/String;
      if-ltz v6, :cond_c
  
***************
*** 777,793 ****
--- 880,900 ----
  
      if-gez v9, :cond_a
  
+     .line 163
      :cond_c
      move-object v5, v1
  
+     .line 164
      move v6, v3
  
      goto :goto_7
  
+     .line 167
      .end local v1           #current:Ljava/lang/String;
      :cond_d
      if-ne v2, p2, :cond_10
  
+     .line 172
      .end local v3           #j:I
      .end local v5           #smallest:Ljava/lang/String;
      :cond_e
***************
*** 797,821 ****
--- 904,933 ----
  
      iput-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursor:Landroid/database/Cursor;
  
+     .line 173
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mRowNumCache:[I
  
      aput p2, v9, v0
  
+     .line 174
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursorCache:[I
  
      aput v6, v9, v0
  
+     .line 175
      const/4 v2, 0x0
  
      :goto_8
      if-ge v2, v4, :cond_12
  
+     .line 176
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v9, v9, v2
  
      if-eqz v9, :cond_f
  
+     .line 177
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCurRowNumCache:[[I
  
      aget-object v9, v9, v0
***************
*** 830,840 ****
--- 942,954 ----
  
      aput v10, v9, v2
  
+     .line 175
      :cond_f
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_8
  
+     .line 168
      .restart local v3       #j:I
      .restart local v5       #smallest:Ljava/lang/String;
      :cond_10
***************
*** 844,860 ****
--- 958,977 ----
  
      if-eqz v9, :cond_11
  
+     .line 169
      iget-object v9, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v9, v9, v6
  
      invoke-interface {v9}, Landroid/database/Cursor;->moveToNext()Z
  
+     .line 154
      :cond_11
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_5
  
+     .line 180
      .end local v3           #j:I
      .end local v5           #smallest:Ljava/lang/String;
      :cond_12
***************
*** 868,877 ****
--- 985,996 ----
      .parameter "observer"
  
      .prologue
+     .line 276
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      array-length v1, v2
  
+     .line 277
      .local v1, length:I
      const/4 v0, 0x0
  
***************
*** 879,901 ****
--- 998,1024 ----
      :goto_0
      if-ge v0, v1, :cond_1
  
+     .line 278
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v2, v2, v0
  
      if-eqz v2, :cond_0
  
+     .line 279
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v2, v2, v0
  
      invoke-interface {v2, p1}, Landroid/database/Cursor;->registerDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 277
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 282
      :cond_1
      return-void
  .end method
***************
*** 904,913 ****
--- 1027,1038 ----
      .locals 3
  
      .prologue
+     .line 297
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      array-length v1, v2
  
+     .line 298
      .local v1, length:I
      const/4 v0, 0x0
  
***************
*** 915,931 ****
--- 1040,1059 ----
      :goto_0
      if-ge v0, v1, :cond_2
  
+     .line 299
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v2, v2, v0
  
      if-nez v2, :cond_1
  
+     .line 298
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 301
      :cond_1
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
***************
*** 937,944 ****
--- 1065,1074 ----
  
      if-nez v2, :cond_0
  
+     .line 302
      const/4 v2, 0x0
  
+     .line 306
      :goto_1
      return v2
  
***************
*** 953,962 ****
--- 1083,1094 ----
      .parameter "observer"
  
      .prologue
+     .line 286
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      array-length v1, v2
  
+     .line 287
      .local v1, length:I
      const/4 v0, 0x0
  
***************
*** 964,986 ****
--- 1096,1122 ----
      :goto_0
      if-ge v0, v1, :cond_1
  
+     .line 288
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v2, v2, v0
  
      if-eqz v2, :cond_0
  
+     .line 289
      iget-object v2, p0, Lcom/android/internal/database/SortCursor;->mCursors:[Landroid/database/Cursor;
  
      aget-object v2, v2, v0
  
      invoke-interface {v2, p1}, Landroid/database/Cursor;->unregisterDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 287
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 292
      :cond_1
      return-void
  .end method
