*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/AppWidgetService$2.smali	2012-03-23 19:35:00.445595012 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/AppWidgetService$2.smali.original	2012-03-23 19:34:46.955349017 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 820
      iput-object p1, p0, Lcom/android/server/AppWidgetService$2;->this$0:Lcom/android/server/AppWidgetService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,51 ****
--- 43,54 ----
      .parameter "service"
  
      .prologue
+     .line 823
      invoke-static {p2}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->asInterface(Landroid/os/IBinder;)Lcom/android/internal/widget/IRemoteViewsFactory;
  
      move-result-object v0
  
+     .line 826
      .local v0, cb:Lcom/android/internal/widget/IRemoteViewsFactory;
      :try_start_0
      invoke-interface {v0}, Lcom/android/internal/widget/IRemoteViewsFactory;->onDataSetChangedAsync()V
***************
*** 53,58 ****
--- 56,62 ----
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/lang/RuntimeException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 832
      :goto_0
      iget-object v2, p0, Lcom/android/server/AppWidgetService$2;->this$0:Lcom/android/server/AppWidgetService;
  
***************
*** 60,79 ****
--- 64,88 ----
  
      invoke-virtual {v2, p0}, Landroid/content/Context;->unbindService(Landroid/content/ServiceConnection;)V
  
+     .line 833
      return-void
  
+     .line 827
      :catch_0
      move-exception v1
  
+     .line 828
      .local v1, e:Landroid/os/RemoteException;
      invoke-virtual {v1}, Landroid/os/RemoteException;->printStackTrace()V
  
      goto :goto_0
  
+     .line 829
      .end local v1           #e:Landroid/os/RemoteException;
      :catch_1
      move-exception v1
  
+     .line 830
      .local v1, e:Ljava/lang/RuntimeException;
      invoke-virtual {v1}, Ljava/lang/RuntimeException;->printStackTrace()V
  
***************
*** 85,89 ****
--- 94,99 ----
      .parameter "name"
  
      .prologue
+     .line 837
      return-void
  .end method
