*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/AppWidgetService$Host.smali	2012-03-23 19:35:04.278664908 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/AppWidgetService$Host.smali.original	2012-03-23 19:34:47.096351587 -0500
***************
*** 44,51 ****
--- 44,53 ----
      .locals 1
  
      .prologue
+     .line 104
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 108
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
