*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/AttributeCache.smali	2012-03-23 19:34:56.394521139 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/AttributeCache.smali.original	2012-03-23 19:34:46.895347923 -0500
***************
*** 39,44 ****
--- 39,45 ----
      .locals 1
  
      .prologue
+     .line 36
      const/4 v0, 0x0
  
      sput-object v0, Lcom/android/server/AttributeCache;->sInstance:Lcom/android/server/AttributeCache;
***************
*** 51,72 ****
--- 52,78 ----
      .parameter "context"
  
      .prologue
+     .line 73
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 39
      new-instance v0, Ljava/util/WeakHashMap;
  
      invoke-direct {v0}, Ljava/util/WeakHashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/AttributeCache;->mPackages:Ljava/util/WeakHashMap;
  
+     .line 41
      new-instance v0, Landroid/content/res/Configuration;
  
      invoke-direct {v0}, Landroid/content/res/Configuration;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/AttributeCache;->mConfiguration:Landroid/content/res/Configuration;
  
+     .line 74
      iput-object p1, p0, Lcom/android/server/AttributeCache;->mContext:Landroid/content/Context;
  
+     .line 75
      return-void
  .end method
  
***************
*** 75,90 ****
--- 81,99 ----
      .parameter "context"
  
      .prologue
+     .line 64
      sget-object v0, Lcom/android/server/AttributeCache;->sInstance:Lcom/android/server/AttributeCache;
  
      if-nez v0, :cond_0
  
+     .line 65
      new-instance v0, Lcom/android/server/AttributeCache;
  
      invoke-direct {v0, p0}, Lcom/android/server/AttributeCache;-><init>(Landroid/content/Context;)V
  
      sput-object v0, Lcom/android/server/AttributeCache;->sInstance:Lcom/android/server/AttributeCache;
  
+     .line 67
      :cond_0
      return-void
  .end method
***************
*** 93,98 ****
--- 102,108 ----
      .locals 1
  
      .prologue
+     .line 70
      sget-object v0, Lcom/android/server/AttributeCache;->sInstance:Lcom/android/server/AttributeCache;
  
      return-object v0
***************
*** 109,116 ****
--- 119,128 ----
      .prologue
      const/4 v6, 0x0
  
+     .line 98
      monitor-enter p0
  
+     .line 99
      :try_start_0
      iget-object v7, p0, Lcom/android/server/AttributeCache;->mPackages:Ljava/util/WeakHashMap;
  
***************
*** 120,134 ****
--- 132,150 ----
  
      check-cast v5, Lcom/android/server/AttributeCache$Package;
  
+     .line 100
      .local v5, pkg:Lcom/android/server/AttributeCache$Package;
      const/4 v4, 0x0
  
+     .line 101
      .local v4, map:Ljava/util/HashMap;,"Ljava/util/HashMap<[ILcom/android/server/AttributeCache$Entry;>;"
      const/4 v2, 0x0
  
+     .line 102
      .local v2, ent:Lcom/android/server/AttributeCache$Entry;
      if-eqz v5, :cond_0
  
+     .line 103
      #getter for: Lcom/android/server/AttributeCache$Package;->mMap:Landroid/util/SparseArray;
      invoke-static {v5}, Lcom/android/server/AttributeCache$Package;->access$000(Lcom/android/server/AttributeCache$Package;)Landroid/util/SparseArray;
  
***************
*** 141,149 ****
--- 157,167 ----
      .end local v4           #map:Ljava/util/HashMap;,"Ljava/util/HashMap<[ILcom/android/server/AttributeCache$Entry;>;"
      check-cast v4, Ljava/util/HashMap;
  
+     .line 104
      .restart local v4       #map:Ljava/util/HashMap;,"Ljava/util/HashMap<[ILcom/android/server/AttributeCache$Entry;>;"
      if-eqz v4, :cond_2
  
+     .line 105
      invoke-virtual {v4, p3}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v2
***************
*** 151,168 ****
--- 169,190 ----
      .end local v2           #ent:Lcom/android/server/AttributeCache$Entry;
      check-cast v2, Lcom/android/server/AttributeCache$Entry;
  
+     .line 106
      .restart local v2       #ent:Lcom/android/server/AttributeCache$Entry;
      if-eqz v2, :cond_2
  
+     .line 107
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-object v6, v2
  
+     .line 137
      :goto_0
      return-object v6
  
+     .line 113
      :cond_0
      :try_start_1
      iget-object v7, p0, Lcom/android/server/AttributeCache;->mContext:Landroid/content/Context;
***************
*** 176,189 ****
--- 198,214 ----
  
      move-result-object v0
  
+     .line 114
      .local v0, context:Landroid/content/Context;
      if-nez v0, :cond_1
  
+     .line 115
      :try_start_2
      monitor-exit p0
  
      goto :goto_0
  
+     .line 138
      .end local v0           #context:Landroid/content/Context;
      .end local v2           #ent:Lcom/android/server/AttributeCache$Entry;
      .end local v4           #map:Ljava/util/HashMap;,"Ljava/util/HashMap<[ILcom/android/server/AttributeCache$Entry;>;"
***************
*** 197,214 ****
--- 222,242 ----
  
      throw v6
  
+     .line 117
      .restart local v2       #ent:Lcom/android/server/AttributeCache$Entry;
      .restart local v4       #map:Ljava/util/HashMap;,"Ljava/util/HashMap<[ILcom/android/server/AttributeCache$Entry;>;"
      .restart local v5       #pkg:Lcom/android/server/AttributeCache$Package;
      :catch_0
      move-exception v1
  
+     .line 118
      .local v1, e:Landroid/content/pm/PackageManager$NameNotFoundException;
      :try_start_3
      monitor-exit p0
  
      goto :goto_0
  
+     .line 120
      .end local v1           #e:Landroid/content/pm/PackageManager$NameNotFoundException;
      .restart local v0       #context:Landroid/content/Context;
      :cond_1
***************
*** 217,222 ****
--- 245,251 ----
      .end local v5           #pkg:Lcom/android/server/AttributeCache$Package;
      invoke-direct {v5, v0}, Lcom/android/server/AttributeCache$Package;-><init>(Landroid/content/Context;)V
  
+     .line 121
      .restart local v5       #pkg:Lcom/android/server/AttributeCache$Package;
      iget-object v7, p0, Lcom/android/server/AttributeCache;->mPackages:Ljava/util/WeakHashMap;
  
***************
*** 226,240 ****
--- 255,272 ----
      :cond_2
      move-object v3, v2
  
+     .line 124
      .end local v2           #ent:Lcom/android/server/AttributeCache$Entry;
      .local v3, ent:Lcom/android/server/AttributeCache$Entry;
      if-nez v4, :cond_3
  
+     .line 125
      new-instance v4, Ljava/util/HashMap;
  
      .end local v4           #map:Ljava/util/HashMap;,"Ljava/util/HashMap<[ILcom/android/server/AttributeCache$Entry;>;"
      invoke-direct {v4}, Ljava/util/HashMap;-><init>()V
  
+     .line 126
      .restart local v4       #map:Ljava/util/HashMap;,"Ljava/util/HashMap<[ILcom/android/server/AttributeCache$Entry;>;"
      #getter for: Lcom/android/server/AttributeCache$Package;->mMap:Landroid/util/SparseArray;
      invoke-static {v5}, Lcom/android/server/AttributeCache$Package;->access$000(Lcom/android/server/AttributeCache$Package;)Landroid/util/SparseArray;
***************
*** 245,250 ****
--- 277,283 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
+     .line 130
      :cond_3
      :try_start_4
      new-instance v2, Lcom/android/server/AttributeCache$Entry;
***************
*** 262,267 ****
--- 295,301 ----
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
      .catch Landroid/content/res/Resources$NotFoundException; {:try_start_4 .. :try_end_4} :catch_1
  
+     .line 132
      .end local v3           #ent:Lcom/android/server/AttributeCache$Entry;
      .restart local v2       #ent:Lcom/android/server/AttributeCache$Entry;
      :try_start_5
***************
*** 270,275 ****
--- 304,310 ----
      .catchall {:try_start_5 .. :try_end_5} :catchall_0
      .catch Landroid/content/res/Resources$NotFoundException; {:try_start_5 .. :try_end_5} :catch_2
  
+     .line 137
      :try_start_6
      monitor-exit p0
  
***************
*** 277,282 ****
--- 312,318 ----
  
      goto :goto_0
  
+     .line 133
      .end local v2           #ent:Lcom/android/server/AttributeCache$Entry;
      .restart local v3       #ent:Lcom/android/server/AttributeCache$Entry;
      :catch_1
***************
*** 284,289 ****
--- 320,326 ----
  
      move-object v2, v3
  
+     .line 134
      .end local v3           #ent:Lcom/android/server/AttributeCache$Entry;
      .local v1, e:Landroid/content/res/Resources$NotFoundException;
      .restart local v2       #ent:Lcom/android/server/AttributeCache$Entry;
***************
*** 294,299 ****
--- 331,337 ----
  
      goto :goto_0
  
+     .line 133
      .end local v1           #e:Landroid/content/res/Resources$NotFoundException;
      :catch_2
      move-exception v1
***************
*** 306,322 ****
--- 344,365 ----
      .parameter "packageName"
  
      .prologue
+     .line 78
      monitor-enter p0
  
+     .line 79
      :try_start_0
      iget-object v0, p0, Lcom/android/server/AttributeCache;->mPackages:Ljava/util/WeakHashMap;
  
      invoke-virtual {v0, p1}, Ljava/util/WeakHashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 80
      monitor-exit p0
  
+     .line 81
      return-void
  
+     .line 80
      :catchall_0
      move-exception v0
  
***************
*** 332,339 ****
--- 375,384 ----
      .parameter "config"
  
      .prologue
+     .line 84
      monitor-enter p0
  
+     .line 85
      :try_start_0
      iget-object v1, p0, Lcom/android/server/AttributeCache;->mConfiguration:Landroid/content/res/Configuration;
  
***************
*** 341,346 ****
--- 386,392 ----
  
      move-result v0
  
+     .line 86
      .local v0, changes:I
      const v1, -0x400000a1
  
***************
*** 348,362 ****
--- 394,412 ----
  
      if-eqz v1, :cond_0
  
+     .line 92
      iget-object v1, p0, Lcom/android/server/AttributeCache;->mPackages:Ljava/util/WeakHashMap;
  
      invoke-virtual {v1}, Ljava/util/WeakHashMap;->clear()V
  
+     .line 94
      :cond_0
      monitor-exit p0
  
+     .line 95
      return-void
  
+     .line 94
      .end local v0           #changes:I
      :catchall_0
      move-exception v1
