*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$PerformRestoreTask.smali	2012-03-23 19:35:08.941749939 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$PerformRestoreTask.smali.original	2012-03-23 19:34:47.316355600 -0500
***************
*** 111,152 ****
--- 111,165 ----
      .parameter "filterSet"
  
      .prologue
+     .line 4074
      iput-object p1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 4075
      sget-object v4, Lcom/android/server/BackupManagerService$RestoreState;->INITIAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      iput-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mCurrentState:Lcom/android/server/BackupManagerService$RestoreState;
  
+     .line 4076
      const/4 v4, 0x0
  
      iput-boolean v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mFinished:Z
  
+     .line 4077
      const/4 v4, 0x0
  
      iput-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mPmAgent:Lcom/android/server/PackageManagerBackupAgent;
  
+     .line 4079
      iput-object p2, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
+     .line 4080
      iput-object p3, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mObserver:Landroid/app/backup/IRestoreObserver;
  
+     .line 4081
      iput-wide p4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mToken:J
  
+     .line 4082
      iput-object p6, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTargetPackage:Landroid/content/pm/PackageInfo;
  
+     .line 4083
      iput p7, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mPmToken:I
  
+     .line 4084
      iput-boolean p8, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mNeedFullBackup:Z
  
+     .line 4086
      if-eqz p9, :cond_0
  
+     .line 4087
      new-instance v4, Ljava/util/HashSet;
  
      invoke-direct {v4}, Ljava/util/HashSet;-><init>()V
  
      iput-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mFilterSet:Ljava/util/HashSet;
  
+     .line 4088
      move-object/from16 v0, p9
  
      .local v0, arr$:[Ljava/lang/String;
***************
*** 161,175 ****
--- 174,191 ----
  
      aget-object v3, v0, v1
  
+     .line 4089
      .local v3, pkg:Ljava/lang/String;
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mFilterSet:Ljava/util/HashSet;
  
      invoke-virtual {v4, v3}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 4088
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 4092
      .end local v0           #arr$:[Ljava/lang/String;
      .end local v1           #i$:I
      .end local v2           #len$:I
***************
*** 179,184 ****
--- 195,201 ----
  
      iput-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mFilterSet:Ljava/util/HashSet;
  
+     .line 4096
      :cond_1
      :try_start_0
      new-instance v4, Ljava/io/File;
***************
*** 195,203 ****
--- 212,222 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 4100
      :goto_1
      return-void
  
+     .line 4097
      :catch_0
      move-exception v4
  
***************
*** 210,219 ****
--- 229,240 ----
      .locals 3
  
      .prologue
+     .line 4511
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupDataName:Ljava/io/File;
  
      invoke-virtual {v0}, Ljava/io/File;->delete()Z
  
+     .line 4512
      :try_start_0
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupData:Landroid/os/ParcelFileDescriptor;
  
***************
*** 225,230 ****
--- 246,252 ----
      :try_end_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_2
  
+     .line 4513
      :cond_0
      :goto_0
      :try_start_1
***************
*** 238,243 ****
--- 260,266 ----
      :try_end_1
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 4514
      :cond_1
      :goto_1
      const/4 v0, 0x0
***************
*** 246,261 ****
--- 269,287 ----
  
      iput-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupData:Landroid/os/ParcelFileDescriptor;
  
+     .line 4529
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mNewStateName:Ljava/io/File;
  
      invoke-virtual {v0}, Ljava/io/File;->delete()Z
  
+     .line 4533
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mCurrentPackage:Landroid/content/pm/PackageInfo;
  
      iget-object v0, v0, Landroid/content/pm/PackageInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      if-eqz v0, :cond_2
  
+     .line 4536
      :try_start_2
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
***************
*** 270,275 ****
--- 296,302 ----
  
      invoke-interface {v0, v1}, Landroid/app/IActivityManager;->unbindBackupAgent(Landroid/content/pm/ApplicationInfo;)V
  
+     .line 4544
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTargetPackage:Landroid/content/pm/PackageInfo;
  
      if-nez v0, :cond_2
***************
*** 286,291 ****
--- 313,319 ----
  
      if-eqz v0, :cond_2
  
+     .line 4546
      const-string v0, "BackupManagerService"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 314,319 ****
--- 342,348 ----
  
      invoke-static {v0, v1}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4548
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      #getter for: Lcom/android/server/BackupManagerService;->mActivityManager:Landroid/app/IActivityManager;
***************
*** 337,342 ****
--- 366,372 ----
      :try_end_2
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_0
  
+     .line 4559
      :cond_2
      :goto_2
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
***************
*** 347,358 ****
--- 377,390 ----
  
      invoke-virtual {v0, v1, p0}, Lcom/android/server/BackupManagerService$BackupHandler;->removeMessages(ILjava/lang/Object;)V
  
+     .line 4560
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v1, v0, Lcom/android/server/BackupManagerService;->mCurrentOpLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 4561
      :try_start_3
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
***************
*** 360,369 ****
--- 392,404 ----
  
      invoke-virtual {v0}, Landroid/util/SparseArray;->clear()V
  
+     .line 4562
      monitor-exit v1
  
+     .line 4563
      return-void
  
+     .line 4562
      :catchall_0
      move-exception v0
  
***************
*** 373,388 ****
--- 408,426 ----
  
      throw v0
  
+     .line 4552
      :catch_0
      move-exception v0
  
      goto :goto_2
  
+     .line 4513
      :catch_1
      move-exception v0
  
      goto :goto_1
  
+     .line 4512
      :catch_2
      move-exception v0
  
***************
*** 393,398 ****
--- 431,437 ----
      .locals 2
  
      .prologue
+     .line 4506
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mCurrentPackage:Landroid/content/pm/PackageInfo;
***************
*** 401,408 ****
--- 440,449 ----
  
      invoke-virtual {v0, v1}, Lcom/android/server/BackupManagerService;->clearApplicationDataSynchronous(Ljava/lang/String;)V
  
+     .line 4507
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->agentCleanup()V
  
+     .line 4508
      return-void
  .end method
  
***************
*** 414,419 ****
--- 455,461 ----
  
      const/4 v9, 0x0
  
+     .line 4137
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v4, v4, Lcom/android/server/BackupManagerService;->mBackupHandler:Lcom/android/server/BackupManagerService$BackupHandler;
***************
*** 422,429 ****
--- 464,473 ----
  
      invoke-virtual {v4, v5}, Lcom/android/server/BackupManagerService$BackupHandler;->removeMessages(I)V
  
+     .line 4140
      iput v6, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
+     .line 4144
      const/16 v4, 0xb0e
  
      const/4 v5, 0x2
***************
*** 453,477 ****
--- 497,526 ----
  
      invoke-static {v4, v5}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4148
      new-instance v4, Ljava/util/ArrayList;
  
      invoke-direct {v4}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mRestorePackages:Ljava/util/ArrayList;
  
+     .line 4149
      new-instance v2, Landroid/content/pm/PackageInfo;
  
      invoke-direct {v2}, Landroid/content/pm/PackageInfo;-><init>()V
  
+     .line 4150
      .local v2, omPackage:Landroid/content/pm/PackageInfo;
      const-string v4, "@pm@"
  
      iput-object v4, v2, Landroid/content/pm/PackageInfo;->packageName:Ljava/lang/String;
  
+     .line 4151
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mRestorePackages:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v2}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 4153
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-virtual {v4}, Lcom/android/server/BackupManagerService;->allAgentPackages()Ljava/util/List;
***************
*** 480,493 ****
--- 529,545 ----
  
      iput-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mAgentPackages:Ljava/util/List;
  
+     .line 4154
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTargetPackage:Landroid/content/pm/PackageInfo;
  
      if-nez v4, :cond_3
  
+     .line 4157
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mFilterSet:Ljava/util/HashSet;
  
      if-eqz v4, :cond_1
  
+     .line 4158
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mAgentPackages:Ljava/util/List;
  
      invoke-interface {v4}, Ljava/util/List;->size()I
***************
*** 500,505 ****
--- 552,558 ----
      :goto_0
      if-ltz v1, :cond_1
  
+     .line 4159
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mAgentPackages:Ljava/util/List;
  
      invoke-interface {v4, v1}, Ljava/util/List;->get(I)Ljava/lang/Object;
***************
*** 508,513 ****
--- 561,567 ----
  
      check-cast v3, Landroid/content/pm/PackageInfo;
  
+     .line 4160
      .local v3, pkg:Landroid/content/pm/PackageInfo;
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mFilterSet:Ljava/util/HashSet;
  
***************
*** 519,533 ****
--- 573,590 ----
  
      if-nez v4, :cond_0
  
+     .line 4161
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mAgentPackages:Ljava/util/List;
  
      invoke-interface {v4, v1}, Ljava/util/List;->remove(I)Ljava/lang/Object;
  
+     .line 4158
      :cond_0
      add-int/lit8 v1, v1, -0x1
  
      goto :goto_0
  
+     .line 4171
      .end local v1           #i:I
      .end local v3           #pkg:Landroid/content/pm/PackageInfo;
      :cond_1
***************
*** 537,542 ****
--- 594,600 ----
  
      invoke-virtual {v4, v5}, Ljava/util/ArrayList;->addAll(Ljava/util/Collection;)Z
  
+     .line 4178
      :goto_1
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mObserver:Landroid/app/backup/IRestoreObserver;
      :try_end_0
***************
*** 544,549 ****
--- 602,608 ----
  
      if-eqz v4, :cond_2
  
+     .line 4182
      :try_start_1
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mObserver:Landroid/app/backup/IRestoreObserver;
  
***************
*** 557,574 ****
--- 616,637 ----
      :try_end_1
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 4195
      :cond_2
      :goto_2
      iput v9, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
+     .line 4196
      sget-object v4, Lcom/android/server/BackupManagerService$RestoreState;->DOWNLOAD_DATA:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v4}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
+     .line 4197
      .end local v2           #omPackage:Landroid/content/pm/PackageInfo;
      :goto_3
      return-void
  
+     .line 4174
      .restart local v2       #omPackage:Landroid/content/pm/PackageInfo;
      :cond_3
      :try_start_2
***************
*** 582,591 ****
--- 645,656 ----
  
      goto :goto_1
  
+     .line 4188
      .end local v2           #omPackage:Landroid/content/pm/PackageInfo;
      :catch_0
      move-exception v0
  
+     .line 4190
      .local v0, e:Landroid/os/RemoteException;
      const-string v4, "BackupManagerService"
  
***************
*** 593,609 ****
--- 658,677 ----
  
      invoke-static {v4, v5}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4191
      sget-object v4, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v4}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
      goto :goto_3
  
+     .line 4183
      .end local v0           #e:Landroid/os/RemoteException;
      .restart local v2       #omPackage:Landroid/content/pm/PackageInfo;
      :catch_1
      move-exception v0
  
+     .line 4184
      .restart local v0       #e:Landroid/os/RemoteException;
      :try_start_3
      const-string v4, "BackupManagerService"
***************
*** 612,617 ****
--- 680,686 ----
  
      invoke-static {v4, v5}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4185
      const/4 v4, 0x0
  
      iput-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mObserver:Landroid/app/backup/IRestoreObserver;
***************
*** 629,634 ****
--- 698,704 ----
  
      const/4 v6, 0x0
  
+     .line 4208
      :try_start_0
      iget-object v2, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
***************
*** 652,667 ****
--- 722,740 ----
  
      iput v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
+     .line 4210
      iget v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
      if-eqz v1, :cond_0
  
+     .line 4211
      const-string v1, "BackupManagerService"
  
      const-string v2, "Error starting restore operation"
  
      invoke-static {v1, v2}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4212
      const/16 v1, 0xb0f
  
      const/4 v2, 0x0
***************
*** 670,687 ****
--- 743,764 ----
  
      invoke-static {v1, v2}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4213
      sget-object v1, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v1}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 4226
      :goto_0
      return-void
  
+     .line 4216
      :catch_0
      move-exception v0
  
+     .line 4217
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "BackupManagerService"
  
***************
*** 689,708 ****
--- 766,789 ----
  
      invoke-static {v1, v2}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4218
      new-array v1, v6, [Ljava/lang/Object;
  
      invoke-static {v7, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4219
      const/4 v1, 0x1
  
      iput v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
+     .line 4220
      sget-object v1, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v1}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
      goto :goto_0
  
+     .line 4225
      .end local v0           #e:Landroid/os/RemoteException;
      :cond_0
      sget-object v1, Lcom/android/server/BackupManagerService$RestoreState;->PM_METADATA:Lcom/android/server/BackupManagerService$RestoreState;
***************
*** 716,721 ****
--- 797,803 ----
      .locals 2
  
      .prologue
+     .line 4106
      sget-object v0, Lcom/android/server/BackupManagerService$4;->$SwitchMap$com$android$server$BackupManagerService$RestoreState:[I
  
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mCurrentState:Lcom/android/server/BackupManagerService$RestoreState;
***************
*** 728,756 ****
--- 810,844 ----
  
      packed-switch v0, :pswitch_data_0
  
+     .line 4131
      :goto_0
      return-void
  
+     .line 4108
      :pswitch_0
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->beginRestore()V
  
      goto :goto_0
  
+     .line 4112
      :pswitch_1
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->downloadRestoreData()V
  
      goto :goto_0
  
+     .line 4116
      :pswitch_2
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->restorePmMetadata()V
  
      goto :goto_0
  
+     .line 4120
      :pswitch_3
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->restoreNextAgent()V
  
      goto :goto_0
  
+     .line 4124
      :pswitch_4
      iget-boolean v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mFinished:Z
  
***************
*** 758,763 ****
--- 846,852 ----
  
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->finalizeRestore()V
  
+     .line 4128
      :goto_1
      const/4 v0, 0x1
  
***************
*** 765,770 ****
--- 854,860 ----
  
      goto :goto_0
  
+     .line 4126
      :cond_0
      const-string v0, "BackupManagerService"
  
***************
*** 774,779 ****
--- 864,870 ----
  
      goto :goto_1
  
+     .line 4106
      nop
  
      :pswitch_data_0
***************
*** 791,798 ****
--- 882,891 ----
      .parameter "nextState"
  
      .prologue
+     .line 4590
      iput-object p1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mCurrentState:Lcom/android/server/BackupManagerService$RestoreState;
  
+     .line 4591
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v1, v1, Lcom/android/server/BackupManagerService;->mBackupHandler:Lcom/android/server/BackupManagerService$BackupHandler;
***************
*** 803,808 ****
--- 896,902 ----
  
      move-result-object v0
  
+     .line 4592
      .local v0, msg:Landroid/os/Message;
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
***************
*** 810,815 ****
--- 904,910 ----
  
      invoke-virtual {v1, v0}, Lcom/android/server/BackupManagerService$BackupHandler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 4593
      return-void
  .end method
  
***************
*** 819,824 ****
--- 914,920 ----
      .prologue
      const/16 v4, 0x8
  
+     .line 4403
      :try_start_0
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
***************
*** 826,836 ****
--- 922,934 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 4408
      :goto_0
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mObserver:Landroid/app/backup/IRestoreObserver;
  
      if-eqz v1, :cond_0
  
+     .line 4410
      :try_start_1
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mObserver:Landroid/app/backup/IRestoreObserver;
  
***************
*** 840,845 ****
--- 938,944 ----
      :try_end_1
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 4419
      :cond_0
      :goto_1
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTargetPackage:Landroid/content/pm/PackageInfo;
***************
*** 850,855 ****
--- 949,955 ----
  
      if-eqz v1, :cond_1
  
+     .line 4420
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v2, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mPmAgent:Lcom/android/server/PackageManagerBackupAgent;
***************
*** 860,880 ****
--- 960,984 ----
  
      iput-object v2, v1, Lcom/android/server/BackupManagerService;->mAncestralPackages:Ljava/util/Set;
  
+     .line 4421
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-wide v2, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mToken:J
  
      iput-wide v2, v1, Lcom/android/server/BackupManagerService;->mAncestralToken:J
  
+     .line 4422
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-virtual {v1}, Lcom/android/server/BackupManagerService;->writeRestoreTokens()V
  
+     .line 4427
      :cond_1
      iget v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mPmToken:I
  
      if-lez v1, :cond_2
  
+     .line 4430
      :try_start_2
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
***************
*** 886,891 ****
--- 990,996 ----
      :try_end_2
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_2
  
+     .line 4435
      :cond_2
      :goto_2
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
***************
*** 894,899 ****
--- 999,1005 ----
  
      invoke-virtual {v1, v4}, Lcom/android/server/BackupManagerService$BackupHandler;->removeMessages(I)V
  
+     .line 4436
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v1, v1, Lcom/android/server/BackupManagerService;->mBackupHandler:Lcom/android/server/BackupManagerService$BackupHandler;
***************
*** 902,924 ****
--- 1008,1035 ----
  
      invoke-virtual {v1, v4, v2, v3}, Lcom/android/server/BackupManagerService$BackupHandler;->sendEmptyMessageDelayed(IJ)Z
  
+     .line 4440
      const-string v1, "BackupManagerService"
  
      const-string v2, "Restore complete."
  
      invoke-static {v1, v2}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4441
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v1, v1, Lcom/android/server/BackupManagerService;->mWakelock:Landroid/os/PowerManager$WakeLock;
  
      invoke-virtual {v1}, Landroid/os/PowerManager$WakeLock;->release()V
  
+     .line 4442
      return-void
  
+     .line 4404
      :catch_0
      move-exception v0
  
+     .line 4405
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "BackupManagerService"
  
***************
*** 928,937 ****
--- 1039,1050 ----
  
      goto :goto_0
  
+     .line 4411
      .end local v0           #e:Landroid/os/RemoteException;
      :catch_1
      move-exception v0
  
+     .line 4412
      .restart local v0       #e:Landroid/os/RemoteException;
      const-string v1, "BackupManagerService"
  
***************
*** 941,946 ****
--- 1054,1060 ----
  
      goto :goto_1
  
+     .line 4431
      .end local v0           #e:Landroid/os/RemoteException;
      :catch_2
      move-exception v1
***************
*** 952,957 ****
--- 1066,1072 ----
      .locals 4
  
      .prologue
+     .line 4579
      const-string v0, "BackupManagerService"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 978,983 ****
--- 1093,1099 ----
  
      invoke-static {v0, v1}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4580
      const/16 v0, 0xb10
  
      const/4 v1, 0x2
***************
*** 1000,1011 ****
--- 1116,1130 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4583
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->agentErrorCleanup()V
  
+     .line 4584
      sget-object v0, Lcom/android/server/BackupManagerService$RestoreState;->RUNNING_QUEUE:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
+     .line 4585
      return-void
  .end method
  
***************
*** 1019,1028 ****
--- 1138,1150 ----
      .prologue
      const/4 v8, 0x0
  
+     .line 4448
      iput-object p1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mCurrentPackage:Landroid/content/pm/PackageInfo;
  
+     .line 4449
      iget-object v7, p1, Landroid/content/pm/PackageInfo;->packageName:Ljava/lang/String;
  
+     .line 4451
      .local v7, packageName:Ljava/lang/String;
      const-string v0, "BackupManagerService"
  
***************
*** 1046,1051 ****
--- 1168,1174 ----
  
      invoke-static {v0, v1}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4454
      new-instance v0, Ljava/io/File;
  
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
***************
*** 1074,1079 ****
--- 1197,1203 ----
  
      iput-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupDataName:Ljava/io/File;
  
+     .line 4455
      new-instance v0, Ljava/io/File;
  
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStateDir:Ljava/io/File;
***************
*** 1100,1105 ****
--- 1224,1230 ----
  
      iput-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mNewStateName:Ljava/io/File;
  
+     .line 4456
      new-instance v0, Ljava/io/File;
  
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStateDir:Ljava/io/File;
***************
*** 1108,1119 ****
--- 1233,1246 ----
  
      iput-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mSavedStateName:Ljava/io/File;
  
+     .line 4458
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-virtual {v0}, Lcom/android/server/BackupManagerService;->generateToken()I
  
      move-result v4
  
+     .line 4461
      .local v4, token:I
      :try_start_0
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupDataName:Ljava/io/File;
***************
*** 1126,1131 ****
--- 1253,1259 ----
  
      iput-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupData:Landroid/os/ParcelFileDescriptor;
  
+     .line 4466
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupData:Landroid/os/ParcelFileDescriptor;
***************
*** 1136,1141 ****
--- 1264,1270 ----
  
      if-eqz v0, :cond_0
  
+     .line 4469
      const-string v0, "BackupManagerService"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 1158,1163 ****
--- 1287,1293 ----
  
      invoke-static {v0, v1}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4470
      const/16 v0, 0xb0f
  
      const/4 v1, 0x0
***************
*** 1166,1191 ****
--- 1296,1327 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4471
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupData:Landroid/os/ParcelFileDescriptor;
  
      invoke-virtual {v0}, Landroid/os/ParcelFileDescriptor;->close()V
  
+     .line 4472
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupDataName:Ljava/io/File;
  
      invoke-virtual {v0}, Ljava/io/File;->delete()Z
  
+     .line 4473
      sget-object v0, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
+     .line 4500
      :goto_0
      return-void
  
+     .line 4478
      :cond_0
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupData:Landroid/os/ParcelFileDescriptor;
  
      invoke-virtual {v0}, Landroid/os/ParcelFileDescriptor;->close()V
  
+     .line 4479
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupDataName:Ljava/io/File;
  
      const/high16 v1, 0x1000
***************
*** 1196,1201 ****
--- 1332,1338 ----
  
      iput-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupData:Landroid/os/ParcelFileDescriptor;
  
+     .line 4482
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mNewStateName:Ljava/io/File;
  
      const/high16 v1, 0x3c00
***************
*** 1206,1217 ****
--- 1343,1356 ----
  
      iput-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mNewState:Landroid/os/ParcelFileDescriptor;
  
+     .line 4488
      iget-object v0, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      const-wide/32 v1, 0xea60
  
      invoke-virtual {v0, v4, v1, v2, p0}, Lcom/android/server/BackupManagerService;->prepareOperationTimeout(IJLcom/android/server/BackupManagerService$BackupRestoreTask;)V
  
+     .line 4489
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupData:Landroid/os/ParcelFileDescriptor;
  
      iget-object v3, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mNewState:Landroid/os/ParcelFileDescriptor;
***************
*** 1230,1238 ****
--- 1369,1379 ----
  
      goto :goto_0
  
+     .line 4490
      :catch_0
      move-exception v6
  
+     .line 4491
      .local v6, e:Ljava/lang/Exception;
      const-string v0, "BackupManagerService"
  
***************
*** 1256,1261 ****
--- 1397,1403 ----
  
      invoke-static {v0, v1, v6}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 4492
      const/16 v0, 0xb10
  
      const/4 v1, 0x2
***************
*** 1274,1281 ****
--- 1416,1425 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4493
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->agentErrorCleanup()V
  
+     .line 4498
      sget-object v0, Lcom/android/server/BackupManagerService$RestoreState;->RUNNING_QUEUE:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
***************
*** 1287,1292 ****
--- 1431,1437 ----
      .locals 5
  
      .prologue
+     .line 4568
      iget-object v1, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mBackupDataName:Ljava/io/File;
  
      invoke-virtual {v1}, Ljava/io/File;->length()J
***************
*** 1295,1300 ****
--- 1440,1446 ----
  
      long-to-int v0, v1
  
+     .line 4569
      .local v0, size:I
      const/16 v1, 0xb11
  
***************
*** 1320,1331 ****
--- 1466,1480 ----
  
      invoke-static {v1, v2}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4571
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->agentCleanup()V
  
+     .line 4573
      sget-object v1, Lcom/android/server/BackupManagerService$RestoreState;->RUNNING_QUEUE:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v1}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
+     .line 4574
      return-void
  .end method
  
***************
*** 1335,1340 ****
--- 1484,1490 ----
      .prologue
      const/4 v12, 0x1
  
+     .line 4293
      :try_start_0
      iget-object v8, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
***************
*** 1342,1356 ****
--- 1492,1509 ----
  
      move-result-object v7
  
+     .line 4295
      .local v7, packageName:Ljava/lang/String;
      if-nez v7, :cond_0
  
+     .line 4296
      const-string v8, "BackupManagerService"
  
      const-string v9, "Error getting next restore package"
  
      invoke-static {v8, v9}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4297
      const/16 v8, 0xb0f
  
      const/4 v9, 0x0
***************
*** 1359,1372 ****
--- 1512,1528 ----
  
      invoke-static {v8, v9}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4298
      sget-object v8, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v8}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
+     .line 4397
      .end local v7           #packageName:Ljava/lang/String;
      :goto_0
      return-void
  
+     .line 4300
      .restart local v7       #packageName:Ljava/lang/String;
      :cond_0
      const-string v8, ""
***************
*** 1377,1388 ****
--- 1533,1546 ----
  
      if-eqz v8, :cond_1
  
+     .line 4301
      const-string v8, "BackupManagerService"
  
      const-string v9, "No next package, finishing restore"
  
      invoke-static {v8, v9}, Landroid/util/Slog;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4302
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v8
***************
*** 1393,1398 ****
--- 1551,1557 ----
  
      long-to-int v5, v8
  
+     .line 4303
      .local v5, millis:I
      const/16 v8, 0xb12
  
***************
*** 1420,1425 ****
--- 1579,1585 ----
  
      invoke-static {v8, v9}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4304
      sget-object v8, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v8}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
***************
*** 1428,1438 ****
--- 1588,1600 ----
  
      goto :goto_0
  
+     .line 4392
      .end local v5           #millis:I
      .end local v7           #packageName:Ljava/lang/String;
      :catch_0
      move-exception v1
  
+     .line 4393
      .local v1, e:Landroid/os/RemoteException;
      const-string v8, "BackupManagerService"
  
***************
*** 1440,1453 ****
--- 1602,1618 ----
  
      invoke-static {v8, v9}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4394
      iput v12, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
+     .line 4395
      sget-object v8, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v8}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
      goto :goto_0
  
+     .line 4308
      .end local v1           #e:Landroid/os/RemoteException;
      .restart local v7       #packageName:Ljava/lang/String;
      :cond_1
***************
*** 1458,1463 ****
--- 1623,1629 ----
  
      if-eqz v8, :cond_2
  
+     .line 4310
      :try_start_2
      iget-object v8, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mObserver:Landroid/app/backup/IRestoreObserver;
  
***************
*** 1467,1472 ****
--- 1633,1639 ----
      :try_end_2
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_1
  
+     .line 4317
      :cond_2
      :goto_1
      :try_start_3
***************
*** 1476,1484 ****
--- 1643,1653 ----
  
      move-result-object v4
  
+     .line 4318
      .local v4, metaInfo:Lcom/android/server/PackageManagerBackupAgent$Metadata;
      if-nez v4, :cond_3
  
+     .line 4319
      const-string v8, "BackupManagerService"
  
      new-instance v9, Ljava/lang/StringBuilder;
***************
*** 1501,1506 ****
--- 1670,1676 ----
  
      invoke-static {v8, v9}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4320
      const/16 v8, 0xb10
  
      const/4 v9, 0x2
***************
*** 1519,1534 ****
--- 1689,1707 ----
  
      invoke-static {v8, v9}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4322
      sget-object v8, Lcom/android/server/BackupManagerService$RestoreState;->RUNNING_QUEUE:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v8}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
      goto/16 :goto_0
  
+     .line 4311
      .end local v4           #metaInfo:Lcom/android/server/PackageManagerBackupAgent$Metadata;
      :catch_1
      move-exception v1
  
+     .line 4312
      .restart local v1       #e:Landroid/os/RemoteException;
      const-string v8, "BackupManagerService"
  
***************
*** 1536,1541 ****
--- 1709,1715 ----
  
      invoke-static {v8, v9}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4313
      const/4 v8, 0x0
  
      iput-object v8, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mObserver:Landroid/app/backup/IRestoreObserver;
***************
*** 1544,1554 ****
--- 1718,1730 ----
  
      goto :goto_1
  
+     .line 4328
      .end local v1           #e:Landroid/os/RemoteException;
      .restart local v4       #metaInfo:Lcom/android/server/PackageManagerBackupAgent$Metadata;
      :cond_3
      const/16 v2, 0x40
  
+     .line 4329
      .local v2, flags:I
      :try_start_4
      iget-object v8, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
***************
*** 1565,1570 ****
--- 1741,1747 ----
  
      move-result-object v6
  
+     .line 4338
      .local v6, packageInfo:Landroid/content/pm/PackageInfo;
      :try_start_5
      iget v8, v4, Lcom/android/server/PackageManagerBackupAgent$Metadata;->versionCode:I
***************
*** 1573,1578 ****
--- 1750,1756 ----
  
      if-le v8, v9, :cond_5
  
+     .line 4342
      iget-object v8, v6, Landroid/content/pm/PackageInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iget v8, v8, Landroid/content/pm/ApplicationInfo;->flags:I
***************
*** 1583,1588 ****
--- 1761,1767 ----
  
      if-nez v8, :cond_4
  
+     .line 4344
      new-instance v8, Ljava/lang/StringBuilder;
  
      invoke-direct {v8}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1615,1620 ****
--- 1794,1800 ----
  
      move-result-object v3
  
+     .line 4346
      .local v3, message:Ljava/lang/String;
      const-string v8, "BackupManagerService"
  
***************
*** 1648,1653 ****
--- 1828,1834 ----
  
      invoke-static {v8, v9}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4347
      const/16 v8, 0xb10
  
      const/4 v9, 0x2
***************
*** 1664,1680 ****
--- 1845,1864 ----
  
      invoke-static {v8, v9}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4349
      sget-object v8, Lcom/android/server/BackupManagerService$RestoreState;->RUNNING_QUEUE:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v8}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
      goto/16 :goto_0
  
+     .line 4330
      .end local v3           #message:Ljava/lang/String;
      .end local v6           #packageInfo:Landroid/content/pm/PackageInfo;
      :catch_2
      move-exception v1
  
+     .line 4331
      .local v1, e:Landroid/content/pm/PackageManager$NameNotFoundException;
      const-string v8, "BackupManagerService"
  
***************
*** 1682,1687 ****
--- 1866,1872 ----
  
      invoke-static {v8, v9, v1}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 4332
      const/16 v8, 0xb10
  
      const/4 v9, 0x2
***************
*** 1700,1711 ****
--- 1885,1898 ----
  
      invoke-static {v8, v9}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4334
      sget-object v8, Lcom/android/server/BackupManagerService$RestoreState;->RUNNING_QUEUE:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v8}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
      goto/16 :goto_0
  
+     .line 4352
      .end local v1           #e:Landroid/content/pm/PackageManager$NameNotFoundException;
      .restart local v6       #packageInfo:Landroid/content/pm/PackageInfo;
      :cond_4
***************
*** 1751,1756 ****
--- 1938,1944 ----
  
      invoke-static {v8, v9}, Landroid/util/Slog;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4358
      :cond_5
      iget-object v8, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
***************
*** 1763,1768 ****
--- 1951,1957 ----
  
      if-nez v8, :cond_6
  
+     .line 4359
      const-string v8, "BackupManagerService"
  
      new-instance v9, Ljava/lang/StringBuilder;
***************
*** 1785,1790 ****
--- 1974,1980 ----
  
      invoke-static {v8, v9}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4360
      const/16 v8, 0xb10
  
      const/4 v9, 0x2
***************
*** 1803,1814 ****
--- 1993,2006 ----
  
      invoke-static {v8, v9}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4362
      sget-object v8, Lcom/android/server/BackupManagerService$RestoreState;->RUNNING_QUEUE:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v8}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
      goto/16 :goto_0
  
+     .line 4366
      :cond_6
      const-string v8, "BackupManagerService"
  
***************
*** 1862,1867 ****
--- 2054,2060 ----
  
      invoke-static {v8, v9}, Landroid/util/Slog;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4372
      iget-object v8, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v9, v6, Landroid/content/pm/PackageInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
***************
*** 1872,1880 ****
--- 2065,2075 ----
  
      move-result-object v0
  
+     .line 4375
      .local v0, agent:Landroid/app/IBackupAgent;
      if-nez v0, :cond_7
  
+     .line 4376
      const-string v8, "BackupManagerService"
  
      new-instance v9, Ljava/lang/StringBuilder;
***************
*** 1897,1902 ****
--- 2092,2098 ----
  
      invoke-static {v8, v9}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4377
      const/16 v8, 0xb10
  
      const/4 v9, 0x2
***************
*** 1915,1920 ****
--- 2111,2117 ----
  
      invoke-static {v8, v9}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4379
      sget-object v8, Lcom/android/server/BackupManagerService$RestoreState;->RUNNING_QUEUE:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v8}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
***************
*** 1923,1928 ****
--- 2120,2126 ----
  
      goto/16 :goto_0
  
+     .line 4385
      :cond_7
      :try_start_6
      iget v8, v4, Lcom/android/server/PackageManagerBackupAgent$Metadata;->versionCode:I
***************
*** 1931,1936 ****
--- 2129,2135 ----
  
      invoke-virtual {p0, v6, v8, v0, v9}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->initiateOneRestore(Landroid/content/pm/PackageInfo;ILandroid/app/IBackupAgent;Z)V
  
+     .line 4386
      iget v8, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mCount:I
  
      add-int/lit8 v8, v8, 0x1
***************
*** 1942,1950 ****
--- 2141,2151 ----
  
      goto/16 :goto_0
  
+     .line 4387
      :catch_3
      move-exception v1
  
+     .line 4388
      .local v1, e:Ljava/lang/Exception;
      :try_start_7
      const-string v8, "BackupManagerService"
***************
*** 1973,1980 ****
--- 2174,2183 ----
  
      invoke-static {v8, v9}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4389
      invoke-virtual {p0}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->agentErrorCleanup()V
  
+     .line 4390
      sget-object v8, Lcom/android/server/BackupManagerService$RestoreState;->RUNNING_QUEUE:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v8}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
***************
*** 1996,2001 ****
--- 2199,2205 ----
  
      const/4 v8, 0x0
  
+     .line 4230
      :try_start_0
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
***************
*** 2003,2017 ****
--- 2207,2224 ----
  
      move-result-object v3
  
+     .line 4231
      .local v3, packageName:Ljava/lang/String;
      if-nez v3, :cond_1
  
+     .line 4232
      const-string v4, "BackupManagerService"
  
      const-string v5, "Error getting first restore package"
  
      invoke-static {v4, v5}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4233
      const/16 v4, 0xb0f
  
      const/4 v5, 0x0
***************
*** 2020,2038 ****
--- 2227,2249 ----
  
      invoke-static {v4, v5}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4234
      const/4 v4, 0x1
  
      iput v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
+     .line 4235
      sget-object v4, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v4}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
+     .line 4289
      .end local v3           #packageName:Ljava/lang/String;
      :cond_0
      :goto_0
      return-void
  
+     .line 4237
      .restart local v3       #packageName:Ljava/lang/String;
      :cond_1
      const-string v4, ""
***************
*** 2043,2054 ****
--- 2254,2267 ----
  
      if-eqz v4, :cond_2
  
+     .line 4238
      const-string v4, "BackupManagerService"
  
      const-string v5, "No restore data available"
  
      invoke-static {v4, v5}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4239
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v4
***************
*** 2059,2064 ****
--- 2272,2278 ----
  
      long-to-int v1, v4
  
+     .line 4240
      .local v1, millis:I
      const/16 v4, 0xb12
  
***************
*** 2086,2095 ****
--- 2300,2311 ----
  
      invoke-static {v4, v5}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4241
      const/4 v4, 0x0
  
      iput v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
+     .line 4242
      sget-object v4, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v4}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
***************
*** 2098,2108 ****
--- 2314,2326 ----
  
      goto :goto_0
  
+     .line 4278
      .end local v1           #millis:I
      .end local v3           #packageName:Ljava/lang/String;
      :catch_0
      move-exception v0
  
+     .line 4279
      .local v0, e:Landroid/os/RemoteException;
      const-string v4, "BackupManagerService"
  
***************
*** 2110,2133 ****
--- 2328,2356 ----
  
      invoke-static {v4, v5}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4280
      new-array v4, v8, [Ljava/lang/Object;
  
      invoke-static {v11, v4}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4281
      iput v9, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
+     .line 4282
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v4, v4, Lcom/android/server/BackupManagerService;->mBackupHandler:Lcom/android/server/BackupManagerService$BackupHandler;
  
      invoke-virtual {v4, v10, p0}, Lcom/android/server/BackupManagerService$BackupHandler;->removeMessages(ILjava/lang/Object;)V
  
+     .line 4283
      sget-object v4, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v4}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
      goto :goto_0
  
+     .line 4244
      .end local v0           #e:Landroid/os/RemoteException;
      .restart local v3       #packageName:Ljava/lang/String;
      :cond_2
***************
*** 2140,2145 ****
--- 2363,2369 ----
  
      if-nez v4, :cond_3
  
+     .line 4245
      const-string v4, "BackupManagerService"
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 2168,2173 ****
--- 2392,2398 ----
  
      invoke-static {v4, v5}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4247
      const/16 v4, 0xb10
  
      const/4 v5, 0x2
***************
*** 2188,2209 ****
--- 2413,2438 ----
  
      invoke-static {v4, v5}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4249
      sget-object v4, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v4}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
  
      goto/16 :goto_0
  
+     .line 4254
      :cond_3
      new-instance v2, Landroid/content/pm/PackageInfo;
  
      invoke-direct {v2}, Landroid/content/pm/PackageInfo;-><init>()V
  
+     .line 4255
      .local v2, omPackage:Landroid/content/pm/PackageInfo;
      const-string v4, "@pm@"
  
      iput-object v4, v2, Landroid/content/pm/PackageInfo;->packageName:Ljava/lang/String;
  
+     .line 4256
      new-instance v4, Lcom/android/server/PackageManagerBackupAgent;
  
      iget-object v5, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
***************
*** 2219,2224 ****
--- 2448,2454 ----
  
      iput-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mPmAgent:Lcom/android/server/PackageManagerBackupAgent;
  
+     .line 4258
      const/4 v4, 0x0
  
      iget-object v5, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mPmAgent:Lcom/android/server/PackageManagerBackupAgent;
***************
*** 2235,2240 ****
--- 2465,2471 ----
  
      invoke-virtual {p0, v2, v4, v5, v6}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->initiateOneRestore(Landroid/content/pm/PackageInfo;ILandroid/app/IBackupAgent;Z)V
  
+     .line 4269
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mPmAgent:Lcom/android/server/PackageManagerBackupAgent;
  
      invoke-virtual {v4}, Lcom/android/server/PackageManagerBackupAgent;->hasMetadata()Z
***************
*** 2243,2254 ****
--- 2474,2487 ----
  
      if-nez v4, :cond_0
  
+     .line 4270
      const-string v4, "BackupManagerService"
  
      const-string v5, "No restore metadata available, so not restoring settings"
  
      invoke-static {v4, v5}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 4271
      const/16 v4, 0xb10
  
      const/4 v5, 0x2
***************
*** 2269,2278 ****
--- 2502,2513 ----
  
      invoke-static {v4, v5}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 4273
      const/4 v4, 0x1
  
      iput v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->mStatus:I
  
+     .line 4274
      iget-object v4, p0, Lcom/android/server/BackupManagerService$PerformRestoreTask;->this$0:Lcom/android/server/BackupManagerService;
  
      iget-object v4, v4, Lcom/android/server/BackupManagerService;->mBackupHandler:Lcom/android/server/BackupManagerService$BackupHandler;
***************
*** 2281,2286 ****
--- 2516,2522 ----
  
      invoke-virtual {v4, v5, p0}, Lcom/android/server/BackupManagerService$BackupHandler;->removeMessages(ILjava/lang/Object;)V
  
+     .line 4275
      sget-object v4, Lcom/android/server/BackupManagerService$RestoreState;->FINAL:Lcom/android/server/BackupManagerService$RestoreState;
  
      invoke-virtual {p0, v4}, Lcom/android/server/BackupManagerService$PerformRestoreTask;->executeNextState(Lcom/android/server/BackupManagerService$RestoreState;)V
