*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$RestoreParams.smali	2012-03-23 19:34:50.386411582 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$RestoreParams.smali.original	2012-03-23 19:34:46.810346375 -0500
***************
*** 44,69 ****
--- 44,78 ----
      .parameter "_needFullBackup"
  
      .prologue
+     .line 272
      iput-object p1, p0, Lcom/android/server/BackupManagerService$RestoreParams;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 273
      iput-object p2, p0, Lcom/android/server/BackupManagerService$RestoreParams;->transport:Lcom/android/internal/backup/IBackupTransport;
  
+     .line 274
      iput-object p3, p0, Lcom/android/server/BackupManagerService$RestoreParams;->observer:Landroid/app/backup/IRestoreObserver;
  
+     .line 275
      iput-wide p4, p0, Lcom/android/server/BackupManagerService$RestoreParams;->token:J
  
+     .line 276
      iput-object p6, p0, Lcom/android/server/BackupManagerService$RestoreParams;->pkgInfo:Landroid/content/pm/PackageInfo;
  
+     .line 277
      iput p7, p0, Lcom/android/server/BackupManagerService$RestoreParams;->pmToken:I
  
+     .line 278
      iput-boolean p8, p0, Lcom/android/server/BackupManagerService$RestoreParams;->needFullBackup:Z
  
+     .line 279
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/server/BackupManagerService$RestoreParams;->filterSet:[Ljava/lang/String;
  
+     .line 280
      return-void
  .end method
  
***************
*** 78,103 ****
--- 87,121 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 283
      iput-object p1, p0, Lcom/android/server/BackupManagerService$RestoreParams;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 284
      iput-object p2, p0, Lcom/android/server/BackupManagerService$RestoreParams;->transport:Lcom/android/internal/backup/IBackupTransport;
  
+     .line 285
      iput-object p3, p0, Lcom/android/server/BackupManagerService$RestoreParams;->observer:Landroid/app/backup/IRestoreObserver;
  
+     .line 286
      iput-wide p4, p0, Lcom/android/server/BackupManagerService$RestoreParams;->token:J
  
+     .line 287
      iput-object v1, p0, Lcom/android/server/BackupManagerService$RestoreParams;->pkgInfo:Landroid/content/pm/PackageInfo;
  
+     .line 288
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/server/BackupManagerService$RestoreParams;->pmToken:I
  
+     .line 289
      iput-boolean p6, p0, Lcom/android/server/BackupManagerService$RestoreParams;->needFullBackup:Z
  
+     .line 290
      iput-object v1, p0, Lcom/android/server/BackupManagerService$RestoreParams;->filterSet:[Ljava/lang/String;
  
+     .line 291
      return-void
  .end method
  
***************
*** 111,137 ****
--- 129,164 ----
      .parameter "_needFullBackup"
  
      .prologue
+     .line 294
      iput-object p1, p0, Lcom/android/server/BackupManagerService$RestoreParams;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 295
      iput-object p2, p0, Lcom/android/server/BackupManagerService$RestoreParams;->transport:Lcom/android/internal/backup/IBackupTransport;
  
+     .line 296
      iput-object p3, p0, Lcom/android/server/BackupManagerService$RestoreParams;->observer:Landroid/app/backup/IRestoreObserver;
  
+     .line 297
      iput-wide p4, p0, Lcom/android/server/BackupManagerService$RestoreParams;->token:J
  
+     .line 298
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/server/BackupManagerService$RestoreParams;->pkgInfo:Landroid/content/pm/PackageInfo;
  
+     .line 299
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/server/BackupManagerService$RestoreParams;->pmToken:I
  
+     .line 300
      iput-boolean p7, p0, Lcom/android/server/BackupManagerService$RestoreParams;->needFullBackup:Z
  
+     .line 301
      iput-object p6, p0, Lcom/android/server/BackupManagerService$RestoreParams;->filterSet:[Ljava/lang/String;
  
+     .line 302
      return-void
  .end method
