*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$ClearParams.smali	2012-03-23 19:35:08.988750795 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$ClearParams.smali.original	2012-03-23 19:34:47.317355618 -0500
***************
*** 30,42 ****
--- 30,46 ----
      .parameter "_info"
  
      .prologue
+     .line 309
      iput-object p1, p0, Lcom/android/server/BackupManagerService$ClearParams;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 310
      iput-object p2, p0, Lcom/android/server/BackupManagerService$ClearParams;->transport:Lcom/android/internal/backup/IBackupTransport;
  
+     .line 311
      iput-object p3, p0, Lcom/android/server/BackupManagerService$ClearParams;->packageInfo:Landroid/content/pm/PackageInfo;
  
+     .line 312
      return-void
  .end method
