*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$PerformClearTask.smali	2012-03-23 19:34:50.761418419 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$PerformClearTask.smali.original	2012-03-23 19:34:46.814346447 -0500
***************
*** 33,46 ****
--- 33,50 ----
      .parameter "packageInfo"
  
      .prologue
+     .line 4600
      iput-object p1, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 4601
      iput-object p2, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
+     .line 4602
      iput-object p3, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->mPackage:Landroid/content/pm/PackageInfo;
  
+     .line 4603
      return-void
  .end method
  
***************
*** 50,55 ****
--- 54,60 ----
      .locals 4
  
      .prologue
+     .line 4608
      :try_start_0
      new-instance v0, Ljava/io/File;
  
***************
*** 65,70 ****
--- 70,76 ----
  
      invoke-direct {v0, v2, v3}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 4609
      .local v0, stateDir:Ljava/io/File;
      new-instance v1, Ljava/io/File;
  
***************
*** 74,82 ****
--- 80,90 ----
  
      invoke-direct {v1, v0, v2}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 4610
      .local v1, stateFile:Ljava/io/File;
      invoke-virtual {v1}, Ljava/io/File;->delete()Z
  
+     .line 4614
      iget-object v2, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
      iget-object v3, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->mPackage:Landroid/content/pm/PackageInfo;
***************
*** 86,91 ****
--- 94,100 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 4620
      :try_start_1
      iget-object v2, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
***************
*** 93,98 ****
--- 102,108 ----
      :try_end_1
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 4626
      :goto_0
      iget-object v2, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->this$0:Lcom/android/server/BackupManagerService;
  
***************
*** 103,113 ****
--- 113,126 ----
      :goto_1
      invoke-virtual {v2}, Landroid/os/PowerManager$WakeLock;->release()V
  
+     .line 4628
      return-void
  
+     .line 4618
      :catchall_0
      move-exception v2
  
+     .line 4620
      :try_start_2
      iget-object v3, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
***************
*** 115,120 ****
--- 128,134 ----
      :try_end_2
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_2
  
+     .line 4626
      :goto_2
      iget-object v3, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->this$0:Lcom/android/server/BackupManagerService;
  
***************
*** 122,132 ****
--- 136,149 ----
  
      invoke-virtual {v3}, Landroid/os/PowerManager$WakeLock;->release()V
  
+     .line 4618
      throw v2
  
+     .line 4615
      :catch_0
      move-exception v2
  
+     .line 4620
      :try_start_3
      iget-object v2, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->mTransport:Lcom/android/internal/backup/IBackupTransport;
  
***************
*** 134,139 ****
--- 151,157 ----
      :try_end_3
      .catch Landroid/os/RemoteException; {:try_start_3 .. :try_end_3} :catch_3
  
+     .line 4626
      :goto_3
      iget-object v2, p0, Lcom/android/server/BackupManagerService$PerformClearTask;->this$0:Lcom/android/server/BackupManagerService;
  
***************
*** 141,146 ****
--- 159,165 ----
  
      goto :goto_1
  
+     .line 4621
      .restart local v0       #stateDir:Ljava/io/File;
      .restart local v1       #stateFile:Ljava/io/File;
      :catch_1
