*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$FullRestoreParams.smali	2012-03-23 19:35:04.914676505 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BackupManagerService$FullRestoreParams.smali.original	2012-03-23 19:34:47.108351806 -0500
***************
*** 25,35 ****
--- 25,38 ----
      .parameter "input"
  
      .prologue
+     .line 346
      iput-object p1, p0, Lcom/android/server/BackupManagerService$FullRestoreParams;->this$0:Lcom/android/server/BackupManagerService;
  
      invoke-direct {p0, p1}, Lcom/android/server/BackupManagerService$FullParams;-><init>(Lcom/android/server/BackupManagerService;)V
  
+     .line 347
      iput-object p2, p0, Lcom/android/server/BackupManagerService$FullRestoreParams;->fd:Landroid/os/ParcelFileDescriptor;
  
+     .line 348
      return-void
  .end method
