*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BatteryService.smali	2012-03-23 19:34:56.368520665 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/BatteryService.smali.original	2012-03-23 19:34:46.895347923 -0500
***************
*** 100,105 ****
--- 100,106 ----
      .locals 3
  
      .prologue
+     .line 72
      const-class v0, Lcom/android/server/BatteryService;
  
      invoke-virtual {v0}, Ljava/lang/Class;->getSimpleName()Ljava/lang/String;
***************
*** 108,113 ****
--- 109,115 ----
  
      sput-object v0, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
+     .line 83
      const/4 v0, 0x2
  
      new-array v0, v0, [Ljava/lang/String;
***************
*** 135,176 ****
--- 137,187 ----
      .parameter "lights"
  
      .prologue
+     .line 128
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 119
      const/4 v0, -0x1
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLastPlugType:I
  
+     .line 126
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/server/BatteryService;->mSentLowBatteryBroadcast:Z
  
+     .line 179
      new-instance v0, Lcom/android/server/BatteryService$1;
  
      invoke-direct {v0, p0}, Lcom/android/server/BatteryService$1;-><init>(Lcom/android/server/BatteryService;)V
  
      iput-object v0, p0, Lcom/android/server/BatteryService;->mPowerSupplyObserver:Landroid/os/UEventObserver;
  
+     .line 186
      new-instance v0, Lcom/android/server/BatteryService$2;
  
      invoke-direct {v0, p0}, Lcom/android/server/BatteryService$2;-><init>(Lcom/android/server/BatteryService;)V
  
      iput-object v0, p0, Lcom/android/server/BatteryService;->mInvalidChargerObserver:Landroid/os/UEventObserver;
  
+     .line 129
      iput-object p1, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
+     .line 130
      new-instance v0, Lcom/android/server/BatteryService$Led;
  
      invoke-direct {v0, p0, p1, p2}, Lcom/android/server/BatteryService$Led;-><init>(Lcom/android/server/BatteryService;Landroid/content/Context;Lcom/android/server/LightsService;)V
  
      iput-object v0, p0, Lcom/android/server/BatteryService;->mLed:Lcom/android/server/BatteryService$Led;
  
+     .line 131
      invoke-static {}, Lcom/android/server/am/BatteryStatsService;->getService()Lcom/android/internal/app/IBatteryStats;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/server/BatteryService;->mBatteryStats:Lcom/android/internal/app/IBatteryStats;
  
+     .line 133
      iget-object v0, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 185,190 ****
--- 196,202 ----
  
      iput v0, p0, Lcom/android/server/BatteryService;->mCriticalBatteryLevel:I
  
+     .line 135
      iget-object v0, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 199,204 ****
--- 211,217 ----
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLowBatteryWarningLevel:I
  
+     .line 137
      iget-object v0, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 213,224 ****
--- 226,239 ----
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLowBatteryCloseWarningLevel:I
  
+     .line 140
      iget-object v0, p0, Lcom/android/server/BatteryService;->mPowerSupplyObserver:Landroid/os/UEventObserver;
  
      const-string v1, "SUBSYSTEM=power_supply"
  
      invoke-virtual {v0, v1}, Landroid/os/UEventObserver;->startObserving(Ljava/lang/String;)V
  
+     .line 143
      new-instance v0, Ljava/io/File;
  
      const-string v1, "/sys/devices/virtual/switch/invalid_charger/state"
***************
*** 231,245 ****
--- 246,263 ----
  
      if-eqz v0, :cond_0
  
+     .line 144
      iget-object v0, p0, Lcom/android/server/BatteryService;->mInvalidChargerObserver:Landroid/os/UEventObserver;
  
      const-string v1, "DEVPATH=/devices/virtual/switch/invalid_charger"
  
      invoke-virtual {v0, v1}, Landroid/os/UEventObserver;->startObserving(Ljava/lang/String;)V
  
+     .line 148
      :cond_0
      invoke-direct {p0}, Lcom/android/server/BatteryService;->update()V
  
+     .line 149
      return-void
  .end method
  
***************
*** 248,253 ****
--- 266,272 ----
      .parameter "x0"
  
      .prologue
+     .line 71
      invoke-direct {p0}, Lcom/android/server/BatteryService;->update()V
  
      return-void
***************
*** 258,263 ****
--- 277,283 ----
      .parameter "x0"
  
      .prologue
+     .line 71
      iget v0, p0, Lcom/android/server/BatteryService;->mInvalidCharger:I
  
      return v0
***************
*** 269,274 ****
--- 289,295 ----
      .parameter "x1"
  
      .prologue
+     .line 71
      iput p1, p0, Lcom/android/server/BatteryService;->mInvalidCharger:I
  
      return p1
***************
*** 279,284 ****
--- 300,306 ----
      .parameter "x0"
  
      .prologue
+     .line 71
      iget-object v0, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 289,294 ****
--- 311,317 ----
      .parameter "x0"
  
      .prologue
+     .line 71
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryLevel:I
  
      return v0
***************
*** 299,304 ****
--- 322,328 ----
      .parameter "x0"
  
      .prologue
+     .line 71
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryStatus:I
  
      return v0
***************
*** 309,314 ****
--- 333,339 ----
      .parameter "x0"
  
      .prologue
+     .line 71
      iget v0, p0, Lcom/android/server/BatteryService;->mLowBatteryWarningLevel:I
  
      return v0
***************
*** 323,338 ****
--- 348,366 ----
  
      const v1, 0x10804f6
  
+     .line 474
      iget v2, p0, Lcom/android/server/BatteryService;->mBatteryStatus:I
  
      const/4 v3, 0x2
  
      if-ne v2, v3, :cond_1
  
+     .line 486
      :cond_0
      :goto_0
      return v0
  
+     .line 476
      :cond_1
      iget v2, p0, Lcom/android/server/BatteryService;->mBatteryStatus:I
  
***************
*** 342,349 ****
--- 370,379 ----
  
      move v0, v1
  
+     .line 477
      goto :goto_0
  
+     .line 478
      :cond_2
      iget v2, p0, Lcom/android/server/BatteryService;->mBatteryStatus:I
  
***************
*** 357,362 ****
--- 387,393 ----
  
      if-ne v2, v3, :cond_5
  
+     .line 480
      :cond_3
      invoke-virtual {p0}, Lcom/android/server/BatteryService;->isPowered()Z
  
***************
*** 373,380 ****
--- 404,413 ----
      :cond_4
      move v0, v1
  
+     .line 483
      goto :goto_0
  
+     .line 486
      :cond_5
      const v0, 0x1080512
  
***************
*** 385,403 ****
--- 418,440 ----
      .locals 11
  
      .prologue
+     .line 408
      const-string v7, "batteryinfo"
  
      invoke-static {v7}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 409
      .local v0, batteryInfoService:Landroid/os/IBinder;
      if-nez v0, :cond_1
  
+     .line 443
      :cond_0
      :goto_0
      return-void
  
+     .line 411
      :cond_1
      iget-object v7, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
***************
*** 409,414 ****
--- 446,452 ----
  
      check-cast v1, Landroid/os/DropBoxManager;
  
+     .line 412
      .local v1, db:Landroid/os/DropBoxManager;
      if-eqz v1, :cond_0
  
***************
*** 420,430 ****
--- 458,471 ----
  
      if-eqz v7, :cond_0
  
+     .line 414
      const/4 v2, 0x0
  
+     .line 415
      .local v2, dumpFile:Ljava/io/File;
      const/4 v4, 0x0
  
+     .line 418
      .local v4, dumpStream:Ljava/io/FileOutputStream;
      :try_start_0
      new-instance v3, Ljava/io/File;
***************
*** 437,442 ****
--- 478,484 ----
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_1
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_3
  
+     .line 419
      .end local v2           #dumpFile:Ljava/io/File;
      .local v3, dumpFile:Ljava/io/File;
      :try_start_1
***************
*** 448,453 ****
--- 490,496 ----
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_8
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_6
  
+     .line 420
      .end local v4           #dumpStream:Ljava/io/FileOutputStream;
      .local v5, dumpStream:Ljava/io/FileOutputStream;
      :try_start_2
***************
*** 459,466 ****
--- 502,511 ----
  
      invoke-interface {v0, v7, v8}, Landroid/os/IBinder;->dump(Ljava/io/FileDescriptor;[Ljava/lang/String;)V
  
+     .line 421
      invoke-static {v5}, Landroid/os/FileUtils;->sync(Ljava/io/FileOutputStream;)Z
  
+     .line 424
      const-string v7, "BATTERY_DISCHARGE_INFO"
  
      const/4 v8, 0x2
***************
*** 471,483 ****
--- 516,531 ----
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_9
      .catch Ljava/io/IOException; {:try_start_2 .. :try_end_2} :catch_7
  
+     .line 431
      if-eqz v5, :cond_2
  
+     .line 433
      :try_start_3
      invoke-virtual {v5}, Ljava/io/FileOutputStream;->close()V
      :try_end_3
      .catch Ljava/io/IOException; {:try_start_3 .. :try_end_3} :catch_0
  
+     .line 438
      :cond_2
      :goto_1
      if-eqz v3, :cond_7
***************
*** 488,493 ****
--- 536,542 ----
  
      if-nez v7, :cond_7
  
+     .line 439
      sget-object v7, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
      new-instance v8, Ljava/lang/StringBuilder;
***************
*** 524,529 ****
--- 573,579 ----
      .restart local v2       #dumpFile:Ljava/io/File;
      goto :goto_0
  
+     .line 434
      .end local v2           #dumpFile:Ljava/io/File;
      .end local v4           #dumpStream:Ljava/io/FileOutputStream;
      .restart local v3       #dumpFile:Ljava/io/File;
***************
*** 531,536 ****
--- 581,587 ----
      :catch_0
      move-exception v6
  
+     .line 435
      .local v6, e:Ljava/io/IOException;
      sget-object v7, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
***************
*** 540,545 ****
--- 591,597 ----
  
      goto :goto_1
  
+     .line 425
      .end local v3           #dumpFile:Ljava/io/File;
      .end local v5           #dumpStream:Ljava/io/FileOutputStream;
      .end local v6           #e:Ljava/io/IOException;
***************
*** 548,553 ****
--- 600,606 ----
      :catch_1
      move-exception v6
  
+     .line 426
      .local v6, e:Landroid/os/RemoteException;
      :goto_2
      :try_start_4
***************
*** 559,571 ****
--- 612,627 ----
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
  
+     .line 431
      if-eqz v4, :cond_3
  
+     .line 433
      :try_start_5
      invoke-virtual {v4}, Ljava/io/FileOutputStream;->close()V
      :try_end_5
      .catch Ljava/io/IOException; {:try_start_5 .. :try_end_5} :catch_2
  
+     .line 438
      .end local v6           #e:Landroid/os/RemoteException;
      :cond_3
      :goto_3
***************
*** 577,582 ****
--- 633,639 ----
  
      if-nez v7, :cond_0
  
+     .line 439
      sget-object v7, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
      new-instance v8, Ljava/lang/StringBuilder;
***************
*** 605,614 ****
--- 662,673 ----
  
      goto/16 :goto_0
  
+     .line 434
      .restart local v6       #e:Landroid/os/RemoteException;
      :catch_2
      move-exception v6
  
+     .line 435
      .local v6, e:Ljava/io/IOException;
      sget-object v7, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
***************
*** 618,627 ****
--- 677,688 ----
  
      goto :goto_3
  
+     .line 427
      .end local v6           #e:Ljava/io/IOException;
      :catch_3
      move-exception v6
  
+     .line 428
      .restart local v6       #e:Ljava/io/IOException;
      :goto_4
      :try_start_6
***************
*** 633,645 ****
--- 694,709 ----
      :try_end_6
      .catchall {:try_start_6 .. :try_end_6} :catchall_0
  
+     .line 431
      if-eqz v4, :cond_4
  
+     .line 433
      :try_start_7
      invoke-virtual {v4}, Ljava/io/FileOutputStream;->close()V
      :try_end_7
      .catch Ljava/io/IOException; {:try_start_7 .. :try_end_7} :catch_4
  
+     .line 438
      :cond_4
      :goto_5
      if-eqz v2, :cond_0
***************
*** 650,655 ****
--- 714,720 ----
  
      if-nez v7, :cond_0
  
+     .line 439
      sget-object v7, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
      new-instance v8, Ljava/lang/StringBuilder;
***************
*** 678,686 ****
--- 743,753 ----
  
      goto/16 :goto_0
  
+     .line 434
      :catch_4
      move-exception v6
  
+     .line 435
      sget-object v7, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
      const-string v8, "failed to close dumpsys output stream"
***************
*** 689,694 ****
--- 756,762 ----
  
      goto :goto_5
  
+     .line 431
      .end local v6           #e:Ljava/io/IOException;
      :catchall_0
      move-exception v7
***************
*** 696,706 ****
--- 764,776 ----
      :goto_6
      if-eqz v4, :cond_5
  
+     .line 433
      :try_start_8
      invoke-virtual {v4}, Ljava/io/FileOutputStream;->close()V
      :try_end_8
      .catch Ljava/io/IOException; {:try_start_8 .. :try_end_8} :catch_5
  
+     .line 438
      :cond_5
      :goto_7
      if-eqz v2, :cond_6
***************
*** 711,716 ****
--- 781,787 ----
  
      if-nez v8, :cond_6
  
+     .line 439
      sget-object v8, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
      new-instance v9, Ljava/lang/StringBuilder;
***************
*** 740,748 ****
--- 811,821 ----
      :cond_6
      throw v7
  
+     .line 434
      :catch_5
      move-exception v6
  
+     .line 435
      .restart local v6       #e:Ljava/io/IOException;
      sget-object v8, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
***************
*** 752,757 ****
--- 825,831 ----
  
      goto :goto_7
  
+     .line 431
      .end local v2           #dumpFile:Ljava/io/File;
      .end local v6           #e:Ljava/io/IOException;
      .restart local v3       #dumpFile:Ljava/io/File;
***************
*** 781,786 ****
--- 855,861 ----
      .restart local v2       #dumpFile:Ljava/io/File;
      goto :goto_6
  
+     .line 427
      .end local v2           #dumpFile:Ljava/io/File;
      .restart local v3       #dumpFile:Ljava/io/File;
      :catch_6
***************
*** 809,814 ****
--- 884,890 ----
      .restart local v2       #dumpFile:Ljava/io/File;
      goto :goto_4
  
+     .line 425
      .end local v2           #dumpFile:Ljava/io/File;
      .restart local v3       #dumpFile:Ljava/io/File;
      :catch_8
***************
*** 858,869 ****
--- 934,947 ----
      .parameter "duration"
  
      .prologue
+     .line 446
      iget-object v7, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v7}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v0
  
+     .line 447
      .local v0, cr:Landroid/content/ContentResolver;
      const-string v7, "battery_discharge_threshold"
  
***************
*** 871,876 ****
--- 949,955 ----
  
      move-result-object v2
  
+     .line 449
      .local v2, dischargeThresholdString:Ljava/lang/String;
      const-string v7, "battery_discharge_duration_threshold"
  
***************
*** 878,898 ****
--- 957,981 ----
  
      move-result-object v5
  
+     .line 452
      .local v5, durationThresholdString:Ljava/lang/String;
      if-eqz v2, :cond_0
  
      if-eqz v5, :cond_0
  
+     .line 454
      :try_start_0
      invoke-static {v5}, Ljava/lang/Long;->parseLong(Ljava/lang/String;)J
  
      move-result-wide v3
  
+     .line 455
      .local v3, durationThreshold:J
      invoke-static {v2}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
  
      move-result v1
  
+     .line 456
      .local v1, dischargeThreshold:I
      cmp-long v7, p1, v3
  
***************
*** 906,924 ****
--- 989,1011 ----
  
      if-lt v7, v1, :cond_0
  
+     .line 459
      invoke-direct {p0}, Lcom/android/server/BatteryService;->logBatteryStats()V
      :try_end_0
      .catch Ljava/lang/NumberFormatException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 471
      .end local v1           #dischargeThreshold:I
      .end local v3           #durationThreshold:J
      :cond_0
      :goto_0
      return-void
  
+     .line 465
      :catch_0
      move-exception v6
  
+     .line 466
      .local v6, e:Ljava/lang/NumberFormatException;
      sget-object v7, Lcom/android/server/BatteryService;->TAG:Ljava/lang/String;
  
***************
*** 962,972 ****
--- 1049,1062 ----
      .locals 14
  
      .prologue
+     .line 238
      const/4 v9, 0x0
  
+     .line 239
      .local v9, logOutlier:Z
      const-wide/16 v7, 0x0
  
+     .line 241
      .local v7, dischargeDuration:J
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryLevel:I
  
***************
*** 979,992 ****
--- 1069,1085 ----
      :goto_0
      iput-boolean v0, p0, Lcom/android/server/BatteryService;->mBatteryLevelCritical:Z
  
+     .line 242
      iget-boolean v0, p0, Lcom/android/server/BatteryService;->mAcOnline:Z
  
      if-eqz v0, :cond_d
  
+     .line 243
      const/4 v0, 0x1
  
      iput v0, p0, Lcom/android/server/BatteryService;->mPlugType:I
  
+     .line 252
      :goto_1
      :try_start_0
      iget-object v0, p0, Lcom/android/server/BatteryService;->mBatteryStats:Lcom/android/internal/app/IBatteryStats;
***************
*** 1007,1017 ****
--- 1100,1113 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 259
      :goto_2
      invoke-direct {p0}, Lcom/android/server/BatteryService;->shutdownIfNoPower()V
  
+     .line 260
      invoke-direct {p0}, Lcom/android/server/BatteryService;->shutdownIfOverTemp()V
  
+     .line 262
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryStatus:I
  
      iget v1, p0, Lcom/android/server/BatteryService;->mLastBatteryStatus:I
***************
*** 1060,1065 ****
--- 1156,1162 ----
  
      if-eq v0, v1, :cond_b
  
+     .line 271
      :cond_0
      iget v0, p0, Lcom/android/server/BatteryService;->mPlugType:I
  
***************
*** 1067,1076 ****
--- 1164,1175 ----
  
      if-eq v0, v1, :cond_1
  
+     .line 272
      iget v0, p0, Lcom/android/server/BatteryService;->mLastPlugType:I
  
      if-nez v0, :cond_f
  
+     .line 277
      iget-wide v0, p0, Lcom/android/server/BatteryService;->mDischargeStartTime:J
  
      const-wide/16 v2, 0x0
***************
*** 1085,1090 ****
--- 1184,1190 ----
  
      if-eq v0, v1, :cond_1
  
+     .line 278
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v0
***************
*** 1093,1100 ****
--- 1193,1202 ----
  
      sub-long v7, v0, v2
  
+     .line 279
      const/4 v9, 0x1
  
+     .line 280
      const/16 v0, 0xaaa
  
      const/4 v1, 0x3
***************
*** 1131,1140 ****
--- 1233,1244 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 283
      const-wide/16 v0, 0x0
  
      iput-wide v0, p0, Lcom/android/server/BatteryService;->mDischargeStartTime:J
  
+     .line 291
      :cond_1
      :goto_3
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryStatus:I
***************
*** 1161,1166 ****
--- 1265,1271 ----
  
      if-eq v0, v1, :cond_3
  
+     .line 295
      :cond_2
      const/16 v1, 0xaa3
  
***************
*** 1221,1226 ****
--- 1326,1332 ----
  
      invoke-static {v1, v2}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 299
      :cond_3
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryLevel:I
  
***************
*** 1240,1245 ****
--- 1346,1352 ----
  
      if-eq v0, v1, :cond_5
  
+     .line 302
      :cond_4
      const/16 v0, 0xaa2
  
***************
*** 1279,1284 ****
--- 1386,1392 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 305
      :cond_5
      iget-boolean v0, p0, Lcom/android/server/BatteryService;->mBatteryLevelCritical:Z
  
***************
*** 1292,1297 ****
--- 1400,1406 ----
  
      if-nez v0, :cond_6
  
+     .line 309
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v0
***************
*** 1300,1307 ****
--- 1409,1418 ----
  
      sub-long v7, v0, v2
  
+     .line 310
      const/4 v9, 0x1
  
+     .line 313
      :cond_6
      iget v0, p0, Lcom/android/server/BatteryService;->mPlugType:I
  
***************
*** 1309,1314 ****
--- 1420,1426 ----
  
      const/4 v11, 0x1
  
+     .line 314
      .local v11, plugged:Z
      :goto_5
      iget v0, p0, Lcom/android/server/BatteryService;->mLastPlugType:I
***************
*** 1317,1322 ****
--- 1429,1435 ----
  
      const/4 v10, 0x1
  
+     .line 322
      .local v10, oldPlugged:Z
      :goto_6
      if-nez v11, :cond_13
***************
*** 1344,1362 ****
--- 1457,1479 ----
      :cond_7
      const/4 v12, 0x1
  
+     .line 327
      .local v12, sendBatteryLow:Z
      :goto_7
      invoke-direct {p0}, Lcom/android/server/BatteryService;->sendIntent()V
  
+     .line 332
      new-instance v13, Landroid/content/Intent;
  
      invoke-direct {v13}, Landroid/content/Intent;-><init>()V
  
+     .line 333
      .local v13, statusIntent:Landroid/content/Intent;
      const/high16 v0, 0x1000
  
      invoke-virtual {v13, v0}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 334
      iget v0, p0, Lcom/android/server/BatteryService;->mPlugType:I
  
      if-eqz v0, :cond_14
***************
*** 1365,1400 ****
--- 1482,1525 ----
  
      if-nez v0, :cond_14
  
+     .line 335
      const-string v0, "android.intent.action.ACTION_POWER_CONNECTED"
  
      invoke-virtual {v13, v0}, Landroid/content/Intent;->setAction(Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 336
      iget-object v0, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0, v13}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 343
      :cond_8
      :goto_8
      if-eqz v12, :cond_15
  
+     .line 344
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Lcom/android/server/BatteryService;->mSentLowBatteryBroadcast:Z
  
+     .line 345
      const-string v0, "android.intent.action.BATTERY_LOW"
  
      invoke-virtual {v13, v0}, Landroid/content/Intent;->setAction(Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 346
      iget-object v0, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0, v13}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 354
      :cond_9
      :goto_9
      iget-object v0, p0, Lcom/android/server/BatteryService;->mLed:Lcom/android/server/BatteryService$Led;
  
      invoke-virtual {v0}, Lcom/android/server/BatteryService$Led;->updateLightsLocked()V
  
+     .line 357
      if-eqz v9, :cond_a
  
      const-wide/16 v0, 0x0
***************
*** 1403,1447 ****
--- 1528,1583 ----
  
      if-eqz v0, :cond_a
  
+     .line 358
      invoke-direct {p0, v7, v8}, Lcom/android/server/BatteryService;->logOutlier(J)V
  
+     .line 361
      :cond_a
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryStatus:I
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLastBatteryStatus:I
  
+     .line 362
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryHealth:I
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLastBatteryHealth:I
  
+     .line 363
      iget-boolean v0, p0, Lcom/android/server/BatteryService;->mBatteryPresent:Z
  
      iput-boolean v0, p0, Lcom/android/server/BatteryService;->mLastBatteryPresent:Z
  
+     .line 364
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryLevel:I
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLastBatteryLevel:I
  
+     .line 365
      iget v0, p0, Lcom/android/server/BatteryService;->mPlugType:I
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLastPlugType:I
  
+     .line 366
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryVoltage:I
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLastBatteryVoltage:I
  
+     .line 367
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryTemperature:I
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLastBatteryTemperature:I
  
+     .line 368
      iget-boolean v0, p0, Lcom/android/server/BatteryService;->mBatteryLevelCritical:Z
  
      iput-boolean v0, p0, Lcom/android/server/BatteryService;->mLastBatteryLevelCritical:Z
  
+     .line 369
      iget v0, p0, Lcom/android/server/BatteryService;->mInvalidCharger:I
  
      iput v0, p0, Lcom/android/server/BatteryService;->mLastInvalidCharger:I
  
+     .line 371
      .end local v10           #oldPlugged:Z
      .end local v11           #plugged:Z
      .end local v12           #sendBatteryLow:Z
***************
*** 1449,1470 ****
--- 1585,1610 ----
      :cond_b
      return-void
  
+     .line 241
      :cond_c
      const/4 v0, 0x0
  
      goto/16 :goto_0
  
+     .line 244
      :cond_d
      iget-boolean v0, p0, Lcom/android/server/BatteryService;->mUsbOnline:Z
  
      if-eqz v0, :cond_e
  
+     .line 245
      const/4 v0, 0x2
  
      iput v0, p0, Lcom/android/server/BatteryService;->mPlugType:I
  
      goto/16 :goto_1
  
+     .line 247
      :cond_e
      const/4 v0, 0x0
  
***************
*** 1472,1516 ****
--- 1612,1664 ----
  
      goto/16 :goto_1
  
+     .line 285
      :cond_f
      iget v0, p0, Lcom/android/server/BatteryService;->mPlugType:I
  
      if-nez v0, :cond_1
  
+     .line 287
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Lcom/android/server/BatteryService;->mDischargeStartTime:J
  
+     .line 288
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryLevel:I
  
      iput v0, p0, Lcom/android/server/BatteryService;->mDischargeStartLevel:I
  
      goto/16 :goto_3
  
+     .line 295
      :cond_10
      const/4 v0, 0x0
  
      goto/16 :goto_4
  
+     .line 313
      :cond_11
      const/4 v11, 0x0
  
      goto/16 :goto_5
  
+     .line 314
      .restart local v11       #plugged:Z
      :cond_12
      const/4 v10, 0x0
  
      goto/16 :goto_6
  
+     .line 322
      .restart local v10       #oldPlugged:Z
      :cond_13
      const/4 v12, 0x0
  
      goto/16 :goto_7
  
+     .line 338
      .restart local v12       #sendBatteryLow:Z
      .restart local v13       #statusIntent:Landroid/content/Intent;
      :cond_14
***************
*** 1522,1537 ****
--- 1670,1688 ----
  
      if-eqz v0, :cond_8
  
+     .line 339
      const-string v0, "android.intent.action.ACTION_POWER_DISCONNECTED"
  
      invoke-virtual {v13, v0}, Landroid/content/Intent;->setAction(Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 340
      iget-object v0, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0, v13}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
      goto/16 :goto_8
  
+     .line 347
      :cond_15
      iget-boolean v0, p0, Lcom/android/server/BatteryService;->mSentLowBatteryBroadcast:Z
  
***************
*** 1543,1562 ****
--- 1694,1717 ----
  
      if-lt v0, v1, :cond_9
  
+     .line 348
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/server/BatteryService;->mSentLowBatteryBroadcast:Z
  
+     .line 349
      const-string v0, "android.intent.action.BATTERY_OKAY"
  
      invoke-virtual {v13, v0}, Landroid/content/Intent;->setAction(Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 350
      iget-object v0, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0, v13}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
      goto/16 :goto_9
  
+     .line 255
      .end local v10           #oldPlugged:Z
      .end local v11           #plugged:Z
      .end local v12           #sendBatteryLow:Z
***************
*** 1571,1593 ****
--- 1726,1752 ----
      .locals 4
  
      .prologue
+     .line 375
      new-instance v1, Landroid/content/Intent;
  
      const-string v2, "android.intent.action.BATTERY_CHANGED"
  
      invoke-direct {v1, v2}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 376
      .local v1, intent:Landroid/content/Intent;
      const/high16 v2, 0x6000
  
      invoke-virtual {v1, v2}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 379
      iget v2, p0, Lcom/android/server/BatteryService;->mBatteryLevel:I
  
      invoke-direct {p0, v2}, Lcom/android/server/BatteryService;->getIcon(I)I
  
      move-result v0
  
+     .line 381
      .local v0, icon:I
      const-string v2, "status"
  
***************
*** 1595,1662 ****
--- 1754,1833 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 382
      const-string v2, "health"
  
      iget v3, p0, Lcom/android/server/BatteryService;->mBatteryHealth:I
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 383
      const-string v2, "present"
  
      iget-boolean v3, p0, Lcom/android/server/BatteryService;->mBatteryPresent:Z
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     .line 384
      const-string v2, "level"
  
      iget v3, p0, Lcom/android/server/BatteryService;->mBatteryLevel:I
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 385
      const-string v2, "scale"
  
      const/16 v3, 0x64
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 386
      const-string v2, "icon-small"
  
      invoke-virtual {v1, v2, v0}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 387
      const-string v2, "plugged"
  
      iget v3, p0, Lcom/android/server/BatteryService;->mPlugType:I
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 388
      const-string v2, "voltage"
  
      iget v3, p0, Lcom/android/server/BatteryService;->mBatteryVoltage:I
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 389
      const-string v2, "temperature"
  
      iget v3, p0, Lcom/android/server/BatteryService;->mBatteryTemperature:I
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 390
      const-string v2, "technology"
  
      iget-object v3, p0, Lcom/android/server/BatteryService;->mBatteryTechnology:Ljava/lang/String;
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 391
      const-string v2, "invalid_charger"
  
      iget v3, p0, Lcom/android/server/BatteryService;->mInvalidCharger:I
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 404
      const/4 v2, 0x0
  
      invoke-static {v1, v2}, Landroid/app/ActivityManagerNative;->broadcastStickyIntent(Landroid/content/Intent;Ljava/lang/String;)V
  
+     .line 405
      return-void
  .end method
  
***************
*** 1664,1669 ****
--- 1835,1841 ----
      .locals 3
  
      .prologue
+     .line 211
      iget v1, p0, Lcom/android/server/BatteryService;->mBatteryLevel:I
  
      if-nez v1, :cond_0
***************
*** 1680,1691 ****
--- 1852,1865 ----
  
      if-eqz v1, :cond_0
  
+     .line 212
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.intent.action.ACTION_REQUEST_SHUTDOWN"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 213
      .local v0, intent:Landroid/content/Intent;
      const-string v1, "android.intent.extra.KEY_CONFIRM"
  
***************
*** 1693,1706 ****
--- 1867,1883 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     .line 214
      const/high16 v1, 0x1000
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 215
      iget-object v1, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v1, v0}, Landroid/content/Context;->startActivity(Landroid/content/Intent;)V
  
+     .line 217
      .end local v0           #intent:Landroid/content/Intent;
      :cond_0
      return-void
***************
*** 1710,1715 ****
--- 1887,1893 ----
      .locals 3
  
      .prologue
+     .line 222
      iget v1, p0, Lcom/android/server/BatteryService;->mBatteryTemperature:I
  
      const/16 v2, 0x2a8
***************
*** 1722,1733 ****
--- 1900,1913 ----
  
      if-eqz v1, :cond_0
  
+     .line 223
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.intent.action.ACTION_REQUEST_SHUTDOWN"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 224
      .local v0, intent:Landroid/content/Intent;
      const-string v1, "android.intent.extra.KEY_CONFIRM"
  
***************
*** 1735,1748 ****
--- 1915,1931 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     .line 225
      const/high16 v1, 0x1000
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 226
      iget-object v1, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v1, v0}, Landroid/content/Context;->startActivity(Landroid/content/Intent;)V
  
+     .line 228
      .end local v0           #intent:Landroid/content/Intent;
      :cond_0
      return-void
***************
*** 1752,1770 ****
--- 1935,1957 ----
      .locals 1
  
      .prologue
+     .line 233
      monitor-enter p0
  
      :try_start_0
      invoke-direct {p0}, Lcom/android/server/BatteryService;->native_update()V
  
+     .line 234
      invoke-direct {p0}, Lcom/android/server/BatteryService;->processValues()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 235
      monitor-exit p0
  
      return-void
  
+     .line 233
      :catchall_0
      move-exception v0
  
***************
*** 1782,1787 ****
--- 1969,1975 ----
      .parameter "args"
  
      .prologue
+     .line 492
      iget-object v0, p0, Lcom/android/server/BatteryService;->mContext:Landroid/content/Context;
  
      const-string v1, "android.permission.DUMP"
***************
*** 1792,1797 ****
--- 1980,1986 ----
  
      if-eqz v0, :cond_1
  
+     .line 495
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1830,1839 ****
--- 2019,2030 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 543
      :cond_0
      :goto_0
      return-void
  
+     .line 501
      :cond_1
      if-eqz p3, :cond_2
  
***************
*** 1853,1866 ****
--- 2044,2060 ----
  
      if-eqz v0, :cond_0
  
+     .line 502
      :cond_2
      monitor-enter p0
  
+     .line 503
      :try_start_0
      const-string v0, "Current Battery Service state:"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 504
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1883,1888 ****
--- 2077,2083 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 505
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1905,1910 ****
--- 2100,2106 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 506
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1927,1932 ****
--- 2123,2129 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 507
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1949,1954 ****
--- 2146,2152 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 508
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1971,1976 ****
--- 2169,2175 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 509
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1993,2002 ****
--- 2192,2203 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 510
      const-string v0, "  scale: 100"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 511
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2019,2024 ****
--- 2220,2226 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 512
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2041,2046 ****
--- 2243,2249 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 513
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2063,2068 ****
--- 2266,2272 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 514
      monitor-exit p0
  
      goto/16 :goto_0
***************
*** 2081,2086 ****
--- 2285,2291 ----
      .locals 1
  
      .prologue
+     .line 199
      iget v0, p0, Lcom/android/server/BatteryService;->mBatteryLevel:I
  
      return v0
***************
*** 2090,2095 ****
--- 2295,2301 ----
      .locals 1
  
      .prologue
+     .line 176
      iget v0, p0, Lcom/android/server/BatteryService;->mPlugType:I
  
      return v0
***************
*** 2101,2106 ****
--- 2307,2313 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 153
      iget-boolean v1, p0, Lcom/android/server/BatteryService;->mAcOnline:Z
  
      if-nez v1, :cond_0
***************
*** 2132,2169 ****
--- 2339,2386 ----
  
      const/4 v1, 0x1
  
+     .line 159
      iget v3, p0, Lcom/android/server/BatteryService;->mBatteryStatus:I
  
      if-ne v3, v1, :cond_1
  
+     .line 172
      :cond_0
      :goto_0
      return v1
  
+     .line 162
      :cond_1
      if-nez p1, :cond_2
  
      move v1, v2
  
+     .line 163
      goto :goto_0
  
+     .line 165
      :cond_2
      const/4 v0, 0x0
  
+     .line 166
      .local v0, plugTypeBit:I
      iget-boolean v3, p0, Lcom/android/server/BatteryService;->mAcOnline:Z
  
      if-eqz v3, :cond_3
  
+     .line 167
      or-int/lit8 v0, v0, 0x1
  
+     .line 169
      :cond_3
      iget-boolean v3, p0, Lcom/android/server/BatteryService;->mUsbOnline:Z
  
      if-eqz v3, :cond_4
  
+     .line 170
      or-int/lit8 v0, v0, 0x2
  
+     .line 172
      :cond_4
      and-int v3, p1, v0
  
***************
*** 2178,2186 ****
--- 2395,2406 ----
      .locals 0
  
      .prologue
+     .line 204
      invoke-direct {p0}, Lcom/android/server/BatteryService;->shutdownIfNoPower()V
  
+     .line 205
      invoke-direct {p0}, Lcom/android/server/BatteryService;->shutdownIfOverTemp()V
  
+     .line 206
      return-void
  .end method
