*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/ConnectivityService$1.smali	2012-03-23 19:35:08.387739836 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/ConnectivityService$1.smali.original	2012-03-23 19:34:47.310355490 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1376
      iput-object p1, p0, Lcom/android/server/ConnectivityService$1;->this$0:Lcom/android/server/ConnectivityService;
  
      invoke-direct {p0}, Landroid/net/INetworkPolicyListener$Stub;-><init>()V
***************
*** 38,43 ****
--- 39,45 ----
      .parameter "meteredIfaces"
  
      .prologue
+     .line 1401
      iget-object v4, p0, Lcom/android/server/ConnectivityService$1;->this$0:Lcom/android/server/ConnectivityService;
  
      #getter for: Lcom/android/server/ConnectivityService;->mContext:Landroid/content/Context;
***************
*** 51,56 ****
--- 53,59 ----
  
      invoke-virtual {v4, v5, v6}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1407
      iget-object v4, p0, Lcom/android/server/ConnectivityService$1;->this$0:Lcom/android/server/ConnectivityService;
  
      #getter for: Lcom/android/server/ConnectivityService;->mRulesLock:Ljava/lang/Object;
***************
*** 60,65 ****
--- 63,69 ----
  
      monitor-enter v5
  
+     .line 1408
      :try_start_0
      iget-object v4, p0, Lcom/android/server/ConnectivityService$1;->this$0:Lcom/android/server/ConnectivityService;
  
***************
*** 70,75 ****
--- 74,80 ----
  
      invoke-virtual {v4}, Ljava/util/HashSet;->clear()V
  
+     .line 1409
      move-object v0, p1
  
      .local v0, arr$:[Ljava/lang/String;
***************
*** 84,89 ****
--- 89,95 ----
  
      aget-object v2, v0, v1
  
+     .line 1410
      .local v2, iface:Ljava/lang/String;
      iget-object v4, p0, Lcom/android/server/ConnectivityService$1;->this$0:Lcom/android/server/ConnectivityService;
  
***************
*** 94,109 ****
--- 100,119 ----
  
      invoke-virtual {v4, v2}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 1409
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 1412
      .end local v2           #iface:Ljava/lang/String;
      :cond_0
      monitor-exit v5
  
+     .line 1413
      return-void
  
+     .line 1412
      .end local v0           #arr$:[Ljava/lang/String;
      .end local v1           #i$:I
      .end local v3           #len$:I
***************
*** 123,128 ****
--- 133,139 ----
      .parameter "uidRules"
  
      .prologue
+     .line 1380
      iget-object v1, p0, Lcom/android/server/ConnectivityService$1;->this$0:Lcom/android/server/ConnectivityService;
  
      #getter for: Lcom/android/server/ConnectivityService;->mContext:Landroid/content/Context;
***************
*** 136,141 ****
--- 147,153 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1386
      iget-object v1, p0, Lcom/android/server/ConnectivityService$1;->this$0:Lcom/android/server/ConnectivityService;
  
      #getter for: Lcom/android/server/ConnectivityService;->mRulesLock:Ljava/lang/Object;
***************
*** 145,150 ****
--- 157,163 ----
  
      monitor-enter v2
  
+     .line 1388
      :try_start_0
      iget-object v1, p0, Lcom/android/server/ConnectivityService$1;->this$0:Lcom/android/server/ConnectivityService;
  
***************
*** 159,172 ****
--- 172,188 ----
  
      move-result v0
  
+     .line 1389
      .local v0, oldRules:I
      if-ne v0, p2, :cond_0
  
      monitor-exit v2
  
+     .line 1396
      :goto_0
      return-void
  
+     .line 1391
      :cond_0
      iget-object v1, p0, Lcom/android/server/ConnectivityService$1;->this$0:Lcom/android/server/ConnectivityService;
  
***************
*** 177,182 ****
--- 193,199 ----
  
      invoke-virtual {v1, p1, p2}, Landroid/util/SparseIntArray;->put(II)V
  
+     .line 1392
      monitor-exit v2
  
      goto :goto_0
