*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/ConnectivityService$MyHandler.smali	2012-03-23 19:34:53.286464465 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/ConnectivityService$MyHandler.smali.original	2012-03-23 19:34:46.857347228 -0500
***************
*** 25,34 ****
--- 25,37 ----
      .parameter "looper"
  
      .prologue
+     .line 2337
      iput-object p1, p0, Lcom/android/server/ConnectivityService$MyHandler;->this$0:Lcom/android/server/ConnectivityService;
  
+     .line 2338
      invoke-direct {p0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
+     .line 2339
      return-void
  .end method
  
***************
*** 39,54 ****
--- 42,60 ----
      .parameter "msg"
  
      .prologue
+     .line 2344
      move-object/from16 v0, p1
  
      iget v15, v0, Landroid/os/Message;->what:I
  
      sparse-switch v15, :sswitch_data_0
  
+     .line 2468
      :cond_0
      :goto_0
      return-void
  
+     .line 2346
      :sswitch_0
      move-object/from16 v0, p1
  
***************
*** 56,71 ****
--- 62,80 ----
  
      check-cast v5, Landroid/net/NetworkInfo;
  
+     .line 2347
      .local v5, info:Landroid/net/NetworkInfo;
      invoke-virtual {v5}, Landroid/net/NetworkInfo;->getType()I
  
      move-result v13
  
+     .line 2348
      .local v13, type:I
      invoke-virtual {v5}, Landroid/net/NetworkInfo;->getState()Landroid/net/NetworkInfo$State;
  
      move-result-object v12
  
+     .line 2350
      .local v12, state:Landroid/net/NetworkInfo$State;
      sget-object v15, Landroid/net/NetworkInfo$State;->CONNECTED:Landroid/net/NetworkInfo$State;
  
***************
*** 75,80 ****
--- 84,90 ----
  
      if-ne v12, v15, :cond_2
  
+     .line 2352
      :cond_1
      move-object/from16 v0, p0
  
***************
*** 131,136 ****
--- 141,147 ----
      #calls: Lcom/android/server/ConnectivityService;->log(Ljava/lang/String;)V
      invoke-static/range {v15 .. v16}, Lcom/android/server/ConnectivityService;->access$100(Lcom/android/server/ConnectivityService;Ljava/lang/String;)V
  
+     .line 2364
      :cond_2
      invoke-virtual {v5}, Landroid/net/NetworkInfo;->getType()I
  
***************
*** 160,170 ****
--- 171,183 ----
  
      or-int v4, v15, v16
  
+     .line 2367
      .local v4, eventLogParam:I
      const v15, 0xc364
  
      invoke-static {v15, v4}, Landroid/util/EventLog;->writeEvent(II)I
  
+     .line 2370
      invoke-virtual {v5}, Landroid/net/NetworkInfo;->getDetailedState()Landroid/net/NetworkInfo$DetailedState;
  
      move-result-object v15
***************
*** 175,180 ****
--- 188,194 ----
  
      if-ne v15, v0, :cond_3
  
+     .line 2372
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/ConnectivityService$MyHandler;->this$0:Lcom/android/server/ConnectivityService;
***************
*** 184,194 ****
--- 198,210 ----
  
      goto/16 :goto_0
  
+     .line 2373
      :cond_3
      sget-object v15, Landroid/net/NetworkInfo$State;->DISCONNECTED:Landroid/net/NetworkInfo$State;
  
      if-ne v12, v15, :cond_4
  
+     .line 2374
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/ConnectivityService$MyHandler;->this$0:Lcom/android/server/ConnectivityService;
***************
*** 198,208 ****
--- 214,226 ----
  
      goto/16 :goto_0
  
+     .line 2375
      :cond_4
      sget-object v15, Landroid/net/NetworkInfo$State;->SUSPENDED:Landroid/net/NetworkInfo$State;
  
      if-ne v12, v15, :cond_5
  
+     .line 2383
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/ConnectivityService$MyHandler;->this$0:Lcom/android/server/ConnectivityService;
***************
*** 212,222 ****
--- 230,242 ----
  
      goto/16 :goto_0
  
+     .line 2384
      :cond_5
      sget-object v15, Landroid/net/NetworkInfo$State;->CONNECTED:Landroid/net/NetworkInfo$State;
  
      if-ne v12, v15, :cond_0
  
+     .line 2385
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/ConnectivityService$MyHandler;->this$0:Lcom/android/server/ConnectivityService;
***************
*** 226,231 ****
--- 246,252 ----
  
      goto/16 :goto_0
  
+     .line 2389
      .end local v4           #eventLogParam:I
      .end local v5           #info:Landroid/net/NetworkInfo;
      .end local v12           #state:Landroid/net/NetworkInfo$State;
***************
*** 237,242 ****
--- 258,264 ----
  
      check-cast v5, Landroid/net/NetworkInfo;
  
+     .line 2393
      .restart local v5       #info:Landroid/net/NetworkInfo;
      move-object/from16 v0, p0
  
***************
*** 253,262 ****
--- 275,286 ----
  
      goto/16 :goto_0
  
+     .line 2396
      .end local v5           #info:Landroid/net/NetworkInfo;
      :sswitch_2
      const/4 v1, 0x0
  
+     .line 2397
      .local v1, causedBy:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 266,271 ****
--- 290,296 ----
  
      monitor-enter v16
  
+     .line 2398
      :try_start_0
      move-object/from16 v0, p1
  
***************
*** 301,306 ****
--- 326,332 ----
  
      if-eqz v15, :cond_6
  
+     .line 2400
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/ConnectivityService$MyHandler;->this$0:Lcom/android/server/ConnectivityService;
***************
*** 312,317 ****
--- 338,344 ----
  
      invoke-virtual {v15}, Landroid/os/PowerManager$WakeLock;->release()V
  
+     .line 2401
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/ConnectivityService$MyHandler;->this$0:Lcom/android/server/ConnectivityService;
***************
*** 321,333 ****
--- 348,363 ----
  
      move-result-object v1
  
+     .line 2403
      :cond_6
      monitor-exit v16
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 2404
      if-eqz v1, :cond_0
  
+     .line 2405
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/ConnectivityService$MyHandler;->this$0:Lcom/android/server/ConnectivityService;
***************
*** 363,368 ****
--- 393,399 ----
  
      goto/16 :goto_0
  
+     .line 2403
      :catchall_0
      move-exception v15
  
***************
*** 373,378 ****
--- 404,410 ----
  
      throw v15
  
+     .line 2409
      .end local v1           #causedBy:Ljava/lang/String;
      :sswitch_3
      move-object/from16 v0, p1
***************
*** 381,402 ****
--- 413,438 ----
  
      check-cast v14, Lcom/android/server/ConnectivityService$FeatureUser;
  
+     .line 2410
      .local v14, u:Lcom/android/server/ConnectivityService$FeatureUser;
      invoke-virtual {v14}, Lcom/android/server/ConnectivityService$FeatureUser;->expire()V
  
      goto/16 :goto_0
  
+     .line 2414
      .end local v14           #u:Lcom/android/server/ConnectivityService$FeatureUser;
      :sswitch_4
      move-object/from16 v0, p1
  
      iget v8, v0, Landroid/os/Message;->arg1:I
  
+     .line 2415
      .local v8, netType:I
      move-object/from16 v0, p1
  
      iget v2, v0, Landroid/os/Message;->arg2:I
  
+     .line 2416
      .local v2, condition:I
      move-object/from16 v0, p0
  
***************
*** 407,412 ****
--- 443,449 ----
  
      goto/16 :goto_0
  
+     .line 2421
      .end local v2           #condition:I
      .end local v8           #netType:I
      :sswitch_5
***************
*** 414,424 ****
--- 451,463 ----
  
      iget v8, v0, Landroid/os/Message;->arg1:I
  
+     .line 2422
      .restart local v8       #netType:I
      move-object/from16 v0, p1
  
      iget v11, v0, Landroid/os/Message;->arg2:I
  
+     .line 2423
      .local v11, sequence:I
      move-object/from16 v0, p0
  
***************
*** 429,434 ****
--- 468,474 ----
  
      goto/16 :goto_0
  
+     .line 2428
      .end local v8           #netType:I
      .end local v11           #sequence:I
      :sswitch_6
***************
*** 436,441 ****
--- 476,482 ----
  
      iget v10, v0, Landroid/os/Message;->arg1:I
  
+     .line 2429
      .local v10, preference:I
      move-object/from16 v0, p0
  
***************
*** 446,451 ****
--- 487,493 ----
  
      goto/16 :goto_0
  
+     .line 2434
      .end local v10           #preference:I
      :sswitch_7
      move-object/from16 v0, p1
***************
*** 460,465 ****
--- 502,508 ----
  
      const/4 v3, 0x1
  
+     .line 2435
      .local v3, enabled:Z
      :goto_1
      move-object/from16 v0, p0
***************
*** 471,482 ****
--- 514,527 ----
  
      goto/16 :goto_0
  
+     .line 2434
      .end local v3           #enabled:Z
      :cond_7
      const/4 v3, 0x0
  
      goto :goto_1
  
+     .line 2440
      :sswitch_8
      move-object/from16 v0, p0
  
***************
*** 487,492 ****
--- 532,538 ----
  
      goto/16 :goto_0
  
+     .line 2445
      :sswitch_9
      move-object/from16 v0, p1
  
***************
*** 500,505 ****
--- 546,552 ----
  
      const/4 v7, 0x1
  
+     .line 2446
      .local v7, met:Z
      :goto_2
      move-object/from16 v0, p0
***************
*** 519,530 ****
--- 566,579 ----
  
      goto/16 :goto_0
  
+     .line 2445
      .end local v7           #met:Z
      :cond_8
      const/4 v7, 0x0
  
      goto :goto_2
  
+     .line 2451
      :sswitch_a
      move-object/from16 v0, p0
  
***************
*** 541,546 ****
--- 590,596 ----
  
      if-eq v15, v0, :cond_0
  
+     .line 2452
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/ConnectivityService$MyHandler;->this$0:Lcom/android/server/ConnectivityService;
***************
*** 561,566 ****
--- 611,617 ----
  
      goto/16 :goto_0
  
+     .line 2458
      :sswitch_b
      move-object/from16 v0, p1
  
***************
*** 568,573 ****
--- 619,625 ----
  
      check-cast v6, Landroid/content/Intent;
  
+     .line 2459
      .local v6, intent:Landroid/content/Intent;
      move-object/from16 v0, p0
  
***************
*** 578,589 ****
--- 630,643 ----
  
      goto/16 :goto_0
  
+     .line 2463
      .end local v6           #intent:Landroid/content/Intent;
      :sswitch_c
      move-object/from16 v0, p1
  
      iget v9, v0, Landroid/os/Message;->arg1:I
  
+     .line 2464
      .local v9, networkType:I
      move-object/from16 v0, p1
  
***************
*** 597,602 ****
--- 651,657 ----
  
      const/4 v3, 0x1
  
+     .line 2465
      .restart local v3       #enabled:Z
      :goto_3
      move-object/from16 v0, p0
***************
*** 608,619 ****
--- 663,676 ----
  
      goto/16 :goto_0
  
+     .line 2464
      .end local v3           #enabled:Z
      :cond_9
      const/4 v3, 0x0
  
      goto :goto_3
  
+     .line 2344
      nop
  
      :sswitch_data_0
