*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/CountryDetectorService$1$1.smali	2012-03-23 19:34:58.516559835 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/CountryDetectorService$1$1.smali.original	2012-03-23 19:34:46.939348724 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 173
      iput-object p1, p0, Lcom/android/server/CountryDetectorService$1$1;->this$1:Lcom/android/server/CountryDetectorService$1;
  
      iput-object p2, p0, Lcom/android/server/CountryDetectorService$1$1;->val$country:Landroid/location/Country;
***************
*** 45,50 ****
--- 46,52 ----
      .locals 2
  
      .prologue
+     .line 175
      iget-object v0, p0, Lcom/android/server/CountryDetectorService$1$1;->this$1:Lcom/android/server/CountryDetectorService$1;
  
      iget-object v0, v0, Lcom/android/server/CountryDetectorService$1;->this$0:Lcom/android/server/CountryDetectorService;
***************
*** 53,57 ****
--- 55,60 ----
  
      invoke-virtual {v0, v1}, Lcom/android/server/CountryDetectorService;->notifyReceivers(Landroid/location/Country;)V
  
+     .line 176
      return-void
  .end method
