*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DeviceStorageMonitorService.smali	2012-03-23 19:34:53.900475661 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DeviceStorageMonitorService.smali.original	2012-03-23 19:34:46.863347341 -0500
***************
*** 105,134 ****
--- 105,143 ----
  
      const/high16 v4, 0x1000
  
+     .line 303
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 74
      iput-boolean v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mLowMemFlag:Z
  
+     .line 75
      iput-boolean v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mMemFullFlag:Z
  
+     .line 85
      const-wide/16 v0, -0x1
  
      iput-wide v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mThreadStartTime:J
  
+     .line 86
      iput-boolean v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearSucceeded:Z
  
+     .line 107
      new-instance v0, Lcom/android/server/DeviceStorageMonitorService$1;
  
      invoke-direct {v0, p0}, Lcom/android/server/DeviceStorageMonitorService$1;-><init>(Lcom/android/server/DeviceStorageMonitorService;)V
  
      iput-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mHandler:Landroid/os/Handler;
  
+     .line 304
      const-wide/16 v0, 0x0
  
      iput-wide v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mLastReportedFreeMemTime:J
  
+     .line 305
      iput-object p1, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
+     .line 306
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
***************
*** 137,142 ****
--- 146,152 ----
  
      iput-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mContentResolver:Landroid/content/ContentResolver;
  
+     .line 308
      new-instance v0, Landroid/os/StatFs;
  
      const-string v1, "/data"
***************
*** 145,150 ****
--- 155,161 ----
  
      iput-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mDataFileStats:Landroid/os/StatFs;
  
+     .line 309
      new-instance v0, Landroid/os/StatFs;
  
      const-string v1, "/system"
***************
*** 153,158 ****
--- 164,170 ----
  
      iput-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mSystemFileStats:Landroid/os/StatFs;
  
+     .line 310
      new-instance v0, Landroid/os/StatFs;
  
      const-string v1, "/cache"
***************
*** 161,166 ****
--- 173,179 ----
  
      iput-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mCacheFileStats:Landroid/os/StatFs;
  
+     .line 312
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mDataFileStats:Landroid/os/StatFs;
  
      invoke-virtual {v0}, Landroid/os/StatFs;->getBlockCount()I
***************
*** 185,190 ****
--- 198,204 ----
  
      iput-wide v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mTotalMemory:J
  
+     .line 314
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.intent.action.DEVICE_STORAGE_LOW"
***************
*** 193,202 ****
--- 207,218 ----
  
      iput-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageLowIntent:Landroid/content/Intent;
  
+     .line 315
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageLowIntent:Landroid/content/Intent;
  
      invoke-virtual {v0, v4}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 316
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.intent.action.DEVICE_STORAGE_OK"
***************
*** 205,214 ****
--- 221,232 ----
  
      iput-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageOkIntent:Landroid/content/Intent;
  
+     .line 317
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageOkIntent:Landroid/content/Intent;
  
      invoke-virtual {v0, v4}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 318
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.intent.action.DEVICE_STORAGE_FULL"
***************
*** 217,226 ****
--- 235,246 ----
  
      iput-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageFullIntent:Landroid/content/Intent;
  
+     .line 319
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageFullIntent:Landroid/content/Intent;
  
      invoke-virtual {v0, v4}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 320
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.intent.action.DEVICE_STORAGE_NOT_FULL"
***************
*** 229,254 ****
--- 249,279 ----
  
      iput-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageNotFullIntent:Landroid/content/Intent;
  
+     .line 321
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageNotFullIntent:Landroid/content/Intent;
  
      invoke-virtual {v0, v4}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 323
      invoke-direct {p0}, Lcom/android/server/DeviceStorageMonitorService;->getMemThreshold()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mMemLowThreshold:J
  
+     .line 324
      invoke-direct {p0}, Lcom/android/server/DeviceStorageMonitorService;->getMemFullThreshold()I
  
      move-result v0
  
      iput v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mMemFullThreshold:I
  
+     .line 325
      const/4 v0, 0x1
  
      invoke-direct {p0, v0}, Lcom/android/server/DeviceStorageMonitorService;->checkMemory(Z)V
  
+     .line 326
      return-void
  .end method
  
***************
*** 258,263 ****
--- 283,289 ----
      .parameter "x1"
  
      .prologue
+     .line 58
      invoke-direct {p0, p1}, Lcom/android/server/DeviceStorageMonitorService;->checkMemory(Z)V
  
      return-void
***************
*** 269,274 ****
--- 295,301 ----
      .parameter "x1"
  
      .prologue
+     .line 58
      iput-boolean p1, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearSucceeded:Z
  
      return p1
***************
*** 280,285 ****
--- 307,313 ----
      .parameter "x1"
  
      .prologue
+     .line 58
      iput-boolean p1, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearingCache:Z
  
      return p1
***************
*** 292,297 ****
--- 320,326 ----
      .parameter "x2"
  
      .prologue
+     .line 58
      invoke-direct {p0, p1, p2, p3}, Lcom/android/server/DeviceStorageMonitorService;->postCheckMemoryMsg(ZJ)V
  
      return-void
***************
*** 301,318 ****
--- 330,350 ----
      .locals 2
  
      .prologue
+     .line 387
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      iget-object v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageFullIntent:Landroid/content/Intent;
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->removeStickyBroadcast(Landroid/content/Intent;)V
  
+     .line 388
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      iget-object v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageNotFullIntent:Landroid/content/Intent;
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 389
      return-void
  .end method
  
***************
*** 320,325 ****
--- 352,358 ----
      .locals 3
  
      .prologue
+     .line 364
      iget-object v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      const-string v2, "notification"
***************
*** 330,352 ****
--- 363,389 ----
  
      check-cast v0, Landroid/app/NotificationManager;
  
+     .line 368
      .local v0, mNotificationMgr:Landroid/app/NotificationManager;
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/app/NotificationManager;->cancel(I)V
  
+     .line 370
      iget-object v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      iget-object v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageLowIntent:Landroid/content/Intent;
  
      invoke-virtual {v1, v2}, Landroid/content/Context;->removeStickyBroadcast(Landroid/content/Intent;)V
  
+     .line 371
      iget-object v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      iget-object v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageOkIntent:Landroid/content/Intent;
  
      invoke-virtual {v1, v2}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 372
      return-void
  .end method
  
***************
*** 359,368 ****
--- 396,407 ----
  
      const/4 v6, 0x0
  
+     .line 203
      iget-boolean v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearingCache:Z
  
      if-eqz v2, :cond_1
  
+     .line 206
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v2
***************
*** 371,376 ****
--- 410,416 ----
  
      sub-long v0, v2, v4
  
+     .line 207
      .local v0, diffTime:J
      const-wide/32 v2, 0x927c0
  
***************
*** 378,389 ****
--- 418,431 ----
  
      if-lez v2, :cond_0
  
+     .line 208
      const-string v2, "DeviceStorageMonitorService"
  
      const-string v3, "Thread that clears cache file seems to run for ever"
  
      invoke-static {v2, v3}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 254
      .end local v0           #diffTime:J
      :cond_0
      :goto_0
***************
*** 391,401 ****
--- 433,446 ----
  
      invoke-direct {p0, v7, v2, v3}, Lcom/android/server/DeviceStorageMonitorService;->postCheckMemoryMsg(ZJ)V
  
+     .line 255
      return-void
  
+     .line 211
      :cond_1
      invoke-direct {p0}, Lcom/android/server/DeviceStorageMonitorService;->restatDataDir()V
  
+     .line 215
      iget-wide v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mFreeMem:J
  
      iget-wide v4, p0, Lcom/android/server/DeviceStorageMonitorService;->mMemLowThreshold:J
***************
*** 404,425 ****
--- 449,476 ----
  
      if-gez v2, :cond_4
  
+     .line 216
      iget-boolean v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mLowMemFlag:Z
  
      if-nez v2, :cond_2
  
+     .line 217
      if-eqz p1, :cond_3
  
+     .line 221
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v2
  
      iput-wide v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mThreadStartTime:J
  
+     .line 222
      iput-boolean v6, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearSucceeded:Z
  
+     .line 223
      invoke-direct {p0}, Lcom/android/server/DeviceStorageMonitorService;->clearCache()V
  
+     .line 240
      :cond_2
      :goto_1
      iget-wide v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mFreeMem:J
***************
*** 432,447 ****
--- 483,502 ----
  
      if-gez v2, :cond_5
  
+     .line 241
      iget-boolean v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mMemFullFlag:Z
  
      if-nez v2, :cond_0
  
+     .line 242
      invoke-direct {p0}, Lcom/android/server/DeviceStorageMonitorService;->sendFullNotification()V
  
+     .line 243
      iput-boolean v7, p0, Lcom/android/server/DeviceStorageMonitorService;->mMemFullFlag:Z
  
      goto :goto_0
  
+     .line 225
      :cond_3
      const-string v2, "DeviceStorageMonitorService"
  
***************
*** 449,484 ****
--- 504,548 ----
  
      invoke-static {v2, v3}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 226
      invoke-direct {p0}, Lcom/android/server/DeviceStorageMonitorService;->sendNotification()V
  
+     .line 227
      iput-boolean v7, p0, Lcom/android/server/DeviceStorageMonitorService;->mLowMemFlag:Z
  
      goto :goto_1
  
+     .line 234
      :cond_4
      iget-boolean v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mLowMemFlag:Z
  
      if-eqz v2, :cond_2
  
+     .line 235
      const-string v2, "DeviceStorageMonitorService"
  
      const-string v3, "Memory available. Cancelling notification"
  
      invoke-static {v2, v3}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 236
      invoke-direct {p0}, Lcom/android/server/DeviceStorageMonitorService;->cancelNotification()V
  
+     .line 237
      iput-boolean v6, p0, Lcom/android/server/DeviceStorageMonitorService;->mLowMemFlag:Z
  
      goto :goto_1
  
+     .line 246
      :cond_5
      iget-boolean v2, p0, Lcom/android/server/DeviceStorageMonitorService;->mMemFullFlag:Z
  
      if-eqz v2, :cond_0
  
+     .line 247
      invoke-direct {p0}, Lcom/android/server/DeviceStorageMonitorService;->cancelFullNotification()V
  
+     .line 248
      iput-boolean v6, p0, Lcom/android/server/DeviceStorageMonitorService;->mMemFullFlag:Z
  
      goto :goto_0
***************
*** 490,510 ****
--- 554,578 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 182
      iget-object v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearCacheObserver:Lcom/android/server/DeviceStorageMonitorService$CachePackageDataObserver;
  
      if-nez v1, :cond_0
  
+     .line 184
      new-instance v1, Lcom/android/server/DeviceStorageMonitorService$CachePackageDataObserver;
  
      invoke-direct {v1, p0}, Lcom/android/server/DeviceStorageMonitorService$CachePackageDataObserver;-><init>(Lcom/android/server/DeviceStorageMonitorService;)V
  
      iput-object v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearCacheObserver:Lcom/android/server/DeviceStorageMonitorService$CachePackageDataObserver;
  
+     .line 186
      :cond_0
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearingCache:Z
  
+     .line 189
      :try_start_0
      const-string v1, "package"
  
***************
*** 524,535 ****
--- 592,606 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 196
      :goto_0
      return-void
  
+     .line 191
      :catch_0
      move-exception v0
  
+     .line 192
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "DeviceStorageMonitorService"
  
***************
*** 553,560 ****
--- 624,633 ----
  
      invoke-static {v1, v2}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 193
      iput-boolean v5, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearingCache:Z
  
+     .line 194
      iput-boolean v5, p0, Lcom/android/server/DeviceStorageMonitorService;->mClearSucceeded:Z
  
      goto :goto_0
***************
*** 564,569 ****
--- 637,643 ----
      .locals 4
  
      .prologue
+     .line 291
      iget-object v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mContentResolver:Landroid/content/ContentResolver;
  
      const-string v2, "sys_storage_full_threshold_bytes"
***************
*** 574,579 ****
--- 648,654 ----
  
      move-result v0
  
+     .line 296
      .local v0, value:I
      return v0
  .end method
***************
*** 582,587 ****
--- 657,663 ----
      .locals 7
  
      .prologue
+     .line 271
      iget-object v4, p0, Lcom/android/server/DeviceStorageMonitorService;->mContentResolver:Landroid/content/ContentResolver;
  
      const-string v5, "sys_storage_threshold_percentage"
***************
*** 594,604 ****
--- 670,682 ----
  
      int-to-long v2, v4
  
+     .line 276
      .local v2, value:J
      iget-wide v4, p0, Lcom/android/server/DeviceStorageMonitorService;->mTotalMemory:J
  
      mul-long/2addr v2, v4
  
+     .line 277
      iget-object v4, p0, Lcom/android/server/DeviceStorageMonitorService;->mContentResolver:Landroid/content/ContentResolver;
  
      const-string v5, "sys_storage_threshold_max_bytes"
***************
*** 611,616 ****
--- 689,695 ----
  
      int-to-long v0, v4
  
+     .line 282
      .local v0, maxValue:J
      cmp-long v4, v2, v0
  
***************
*** 637,646 ****
--- 716,727 ----
  
      const/4 v1, 0x1
  
+     .line 259
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v0, v1}, Landroid/os/Handler;->removeMessages(I)V
  
+     .line 260
      iget-object v3, p0, Lcom/android/server/DeviceStorageMonitorService;->mHandler:Landroid/os/Handler;
  
      iget-object v4, p0, Lcom/android/server/DeviceStorageMonitorService;->mHandler:Landroid/os/Handler;
***************
*** 656,666 ****
--- 737,749 ----
  
      invoke-virtual {v3, v0, p2, p3}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 263
      return-void
  
      :cond_0
      move v0, v2
  
+     .line 260
      goto :goto_0
  .end method
  
***************
*** 668,673 ****
--- 751,757 ----
      .locals 20
  
      .prologue
+     .line 131
      :try_start_0
      move-object/from16 v0, p0
  
***************
*** 677,682 ****
--- 761,767 ----
  
      invoke-virtual/range {v15 .. v16}, Landroid/os/StatFs;->restat(Ljava/lang/String;)V
  
+     .line 132
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/DeviceStorageMonitorService;->mDataFileStats:Landroid/os/StatFs;
***************
*** 711,716 ****
--- 796,802 ----
      :try_end_0
      .catch Ljava/lang/IllegalArgumentException; {:try_start_0 .. :try_end_0} :catch_2
  
+     .line 138
      :goto_0
      const-string v15, "debug.freemem"
  
***************
*** 718,723 ****
--- 804,810 ----
  
      move-result-object v4
  
+     .line 139
      .local v4, debugFreeMem:Ljava/lang/String;
      const-string v15, ""
  
***************
*** 727,732 ****
--- 814,820 ----
  
      if-nez v15, :cond_0
  
+     .line 140
      invoke-static {v4}, Ljava/lang/Long;->parseLong(Ljava/lang/String;)J
  
      move-result-wide v15
***************
*** 735,740 ****
--- 823,829 ----
  
      iput-wide v15, v0, Lcom/android/server/DeviceStorageMonitorService;->mFreeMem:J
  
+     .line 143
      :cond_0
      move-object/from16 v0, p0
  
***************
*** 756,766 ****
--- 845,857 ----
  
      mul-long v7, v15, v17
  
+     .line 147
      .local v7, freeMemLogInterval:J
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v2
  
+     .line 148
      .local v2, currTime:J
      move-object/from16 v0, p0
  
***************
*** 782,797 ****
--- 873,891 ----
  
      if-ltz v15, :cond_2
  
+     .line 150
      :cond_1
      move-object/from16 v0, p0
  
      iput-wide v2, v0, Lcom/android/server/DeviceStorageMonitorService;->mLastReportedFreeMemTime:J
  
+     .line 151
      const-wide/16 v11, -0x1
  
      .local v11, mFreeSystem:J
      const-wide/16 v9, -0x1
  
+     .line 153
      .local v9, mFreeCache:J
      :try_start_1
      move-object/from16 v0, p0
***************
*** 802,807 ****
--- 896,902 ----
  
      invoke-virtual/range {v15 .. v16}, Landroid/os/StatFs;->restat(Ljava/lang/String;)V
  
+     .line 154
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/DeviceStorageMonitorService;->mSystemFileStats:Landroid/os/StatFs;
***************
*** 832,837 ****
--- 927,933 ----
  
      mul-long v11, v15, v17
  
+     .line 160
      :goto_1
      :try_start_2
      move-object/from16 v0, p0
***************
*** 842,847 ****
--- 938,944 ----
  
      invoke-virtual/range {v15 .. v16}, Landroid/os/StatFs;->restat(Ljava/lang/String;)V
  
+     .line 161
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/server/DeviceStorageMonitorService;->mCacheFileStats:Landroid/os/StatFs;
***************
*** 872,877 ****
--- 969,975 ----
  
      mul-long v9, v15, v17
  
+     .line 166
      :goto_2
      const/16 v15, 0xaba
  
***************
*** 915,920 ****
--- 1013,1019 ----
  
      invoke-static/range {v15 .. v16}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 170
      .end local v9           #mFreeCache:J
      .end local v11           #mFreeSystem:J
      :cond_2
***************
*** 930,935 ****
--- 1029,1035 ----
  
      move-result-wide v13
  
+     .line 174
      .local v13, threshold:J
      move-object/from16 v0, p0
  
***************
*** 943,948 ****
--- 1043,1049 ----
  
      sub-long v5, v15, v17
  
+     .line 175
      .local v5, delta:J
      cmp-long v15, v5, v13
  
***************
*** 954,959 ****
--- 1055,1061 ----
  
      if-gez v15, :cond_4
  
+     .line 176
      :cond_3
      move-object/from16 v0, p0
  
***************
*** 963,968 ****
--- 1065,1071 ----
  
      iput-wide v15, v0, Lcom/android/server/DeviceStorageMonitorService;->mLastReportedFreeMem:J
  
+     .line 177
      const/16 v15, 0xab8
  
      move-object/from16 v0, p0
***************
*** 973,981 ****
--- 1076,1086 ----
  
      invoke-static/range {v15 .. v17}, Landroid/util/EventLog;->writeEvent(IJ)I
  
+     .line 179
      :cond_4
      return-void
  
+     .line 163
      .end local v5           #delta:J
      .end local v13           #threshold:J
      .restart local v9       #mFreeCache:J
***************
*** 985,995 ****
--- 1090,1102 ----
  
      goto :goto_2
  
+     .line 156
      :catch_1
      move-exception v15
  
      goto/16 :goto_1
  
+     .line 134
      .end local v2           #currTime:J
      .end local v4           #debugFreeMem:Ljava/lang/String;
      .end local v7           #freeMemLogInterval:J
***************
*** 1005,1016 ****
--- 1112,1125 ----
      .locals 2
  
      .prologue
+     .line 379
      iget-object v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      iget-object v1, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageFullIntent:Landroid/content/Intent;
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->sendStickyBroadcast(Landroid/content/Intent;)V
  
+     .line 380
      return-void
  .end method
  
***************
*** 1020,1037 ****
--- 1129,1149 ----
      .prologue
      const/4 v9, 0x0
  
+     .line 337
      const/16 v6, 0xab9
  
      iget-wide v7, p0, Lcom/android/server/DeviceStorageMonitorService;->mFreeMem:J
  
      invoke-static {v6, v7, v8}, Landroid/util/EventLog;->writeEvent(IJ)I
  
+     .line 339
      new-instance v2, Landroid/content/Intent;
  
      const-string v6, "android.intent.action.MANAGE_PACKAGE_STORAGE"
  
      invoke-direct {v2, v6}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 340
      .local v2, lowMemIntent:Landroid/content/Intent;
      const-string v6, "memory"
  
***************
*** 1039,1048 ****
--- 1151,1162 ----
  
      invoke-virtual {v2, v6, v7, v8}, Landroid/content/Intent;->putExtra(Ljava/lang/String;J)Landroid/content/Intent;
  
+     .line 341
      const/high16 v6, 0x1000
  
      invoke-virtual {v2, v6}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 342
      iget-object v6, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      const-string v7, "notification"
***************
*** 1053,1058 ****
--- 1167,1173 ----
  
      check-cast v3, Landroid/app/NotificationManager;
  
+     .line 345
      .local v3, mNotificationMgr:Landroid/app/NotificationManager;
      iget-object v6, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
***************
*** 1062,1067 ****
--- 1177,1183 ----
  
      move-result-object v5
  
+     .line 347
      .local v5, title:Ljava/lang/CharSequence;
      iget-object v6, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
***************
*** 1071,1076 ****
--- 1187,1193 ----
  
      move-result-object v0
  
+     .line 349
      .local v0, details:Ljava/lang/CharSequence;
      iget-object v6, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
***************
*** 1078,1115 ****
--- 1195,1240 ----
  
      move-result-object v1
  
+     .line 350
      .local v1, intent:Landroid/app/PendingIntent;
      new-instance v4, Landroid/app/Notification;
  
      invoke-direct {v4}, Landroid/app/Notification;-><init>()V
  
+     .line 351
      .local v4, notification:Landroid/app/Notification;
      const v6, 0x10804ee
  
      iput v6, v4, Landroid/app/Notification;->icon:I
  
+     .line 352
      iput-object v5, v4, Landroid/app/Notification;->tickerText:Ljava/lang/CharSequence;
  
+     .line 353
      iget v6, v4, Landroid/app/Notification;->flags:I
  
      or-int/lit8 v6, v6, 0x20
  
      iput v6, v4, Landroid/app/Notification;->flags:I
  
+     .line 354
      iget-object v6, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v4, v6, v5, v0, v1}, Landroid/app/Notification;->setLatestEventInfo(Landroid/content/Context;Ljava/lang/CharSequence;Ljava/lang/CharSequence;Landroid/app/PendingIntent;)V
  
+     .line 355
      const/4 v6, 0x1
  
      invoke-virtual {v3, v6, v4}, Landroid/app/NotificationManager;->notify(ILandroid/app/Notification;)V
  
+     .line 356
      iget-object v6, p0, Lcom/android/server/DeviceStorageMonitorService;->mContext:Landroid/content/Context;
  
      iget-object v7, p0, Lcom/android/server/DeviceStorageMonitorService;->mStorageLowIntent:Landroid/content/Intent;
  
      invoke-virtual {v6, v7}, Landroid/content/Context;->sendStickyBroadcast(Landroid/content/Intent;)V
  
+     .line 357
      return-void
  .end method
  
***************
*** 1119,1124 ****
--- 1244,1250 ----
      .locals 2
  
      .prologue
+     .line 407
      iget-wide v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mMemLowThreshold:J
  
      return-wide v0
***************
*** 1128,1133 ****
--- 1254,1260 ----
      .locals 1
  
      .prologue
+     .line 417
      iget-boolean v0, p0, Lcom/android/server/DeviceStorageMonitorService;->mLowMemFlag:Z
  
      return v0
***************
*** 1137,1154 ****
--- 1264,1285 ----
      .locals 4
  
      .prologue
+     .line 392
      invoke-static {}, Lcom/android/server/DeviceStorageMonitorService;->getCallingUid()I
  
      move-result v0
  
+     .line 393
      .local v0, callingUid:I
      const/16 v1, 0x3e8
  
      if-eq v0, v1, :cond_0
  
+     .line 398
      :goto_0
      return-void
  
+     .line 397
      :cond_0
      const/4 v1, 0x1
  
