*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DevicePolicyManagerService$3.smali	2012-03-23 19:35:00.516596305 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DevicePolicyManagerService$3.smali.original	2012-03-23 19:34:46.955349017 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1715
      iput-object p1, p0, Lcom/android/server/DevicePolicyManagerService$3;->this$0:Lcom/android/server/DevicePolicyManagerService;
  
      iput-object p2, p0, Lcom/android/server/DevicePolicyManagerService$3;->val$result:Landroid/os/RemoteCallback;
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "intent"
  
      .prologue
+     .line 1719
      :try_start_0
      iget-object v0, p0, Lcom/android/server/DevicePolicyManagerService$3;->val$result:Landroid/os/RemoteCallback;
  
***************
*** 57,65 ****
--- 59,69 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1722
      :goto_0
      return-void
  
+     .line 1720
      :catch_0
      move-exception v0
  
