*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DiskStatsService.smali	2012-03-23 19:35:00.346593207 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DiskStatsService.smali.original	2012-03-23 19:34:46.954348998 -0500
***************
*** 13,22 ****
--- 13,25 ----
      .parameter "context"
  
      .prologue
+     .line 39
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 40
      iput-object p1, p0, Lcom/android/server/DiskStatsService;->mContext:Landroid/content/Context;
  
+     .line 41
      return-void
  .end method
  
***************
*** 29,34 ****
--- 32,38 ----
      .prologue
      const-wide/16 v9, 0x0
  
+     .line 87
      :try_start_0
      new-instance v5, Landroid/os/StatFs;
  
***************
*** 38,43 ****
--- 42,48 ----
  
      invoke-direct {v5, v8}, Landroid/os/StatFs;-><init>(Ljava/lang/String;)V
  
+     .line 88
      .local v5, statfs:Landroid/os/StatFs;
      invoke-virtual {v5}, Landroid/os/StatFs;->getBlockSize()I
  
***************
*** 45,50 ****
--- 50,56 ----
  
      int-to-long v2, v8
  
+     .line 89
      .local v2, bsize:J
      invoke-virtual {v5}, Landroid/os/StatFs;->getAvailableBlocks()I
  
***************
*** 52,57 ****
--- 58,64 ----
  
      int-to-long v0, v8
  
+     .line 90
      .local v0, avail:J
      invoke-virtual {v5}, Landroid/os/StatFs;->getBlockCount()I
  
***************
*** 59,64 ****
--- 66,72 ----
  
      int-to-long v6, v8
  
+     .line 91
      .local v6, total:J
      cmp-long v8, v2, v9
  
***************
*** 68,73 ****
--- 76,82 ----
  
      if-gtz v8, :cond_1
  
+     .line 92
      :cond_0
      new-instance v8, Ljava/lang/IllegalArgumentException;
  
***************
*** 115,120 ****
--- 124,130 ----
      :try_end_0
      .catch Ljava/lang/IllegalArgumentException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 104
      .end local v0           #avail:J
      .end local v2           #bsize:J
      .end local v5           #statfs:Landroid/os/StatFs;
***************
*** 122,144 ****
--- 132,159 ----
      :catch_0
      move-exception v4
  
+     .line 105
      .local v4, e:Ljava/lang/IllegalArgumentException;
      invoke-virtual {p3, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 106
      const-string v8, "-Error: "
  
      invoke-virtual {p3, v8}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 107
      invoke-virtual {v4}, Ljava/lang/IllegalArgumentException;->toString()Ljava/lang/String;
  
      move-result-object v8
  
      invoke-virtual {p3, v8}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 110
      .end local v4           #e:Ljava/lang/IllegalArgumentException;
      :goto_0
      return-void
  
+     .line 96
      .restart local v0       #avail:J
      .restart local v2       #bsize:J
      .restart local v5       #statfs:Landroid/os/StatFs;
***************
*** 147,156 ****
--- 162,173 ----
      :try_start_1
      invoke-virtual {p3, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 97
      const-string v8, "-Free: "
  
      invoke-virtual {p3, v8}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 98
      mul-long v8, v0, v2
  
      const-wide/16 v10, 0x400
***************
*** 159,168 ****
--- 176,187 ----
  
      invoke-virtual {p3, v8, v9}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 99
      const-string v8, "K / "
  
      invoke-virtual {p3, v8}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 100
      mul-long v8, v6, v2
  
      const-wide/16 v10, 0x400
***************
*** 171,180 ****
--- 190,201 ----
  
      invoke-virtual {p3, v8, v9}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 101
      const-string v8, "K total = "
  
      invoke-virtual {p3, v8}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 102
      const-wide/16 v8, 0x64
  
      mul-long/2addr v8, v0
***************
*** 183,188 ****
--- 204,210 ----
  
      invoke-virtual {p3, v8, v9}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 103
      const-string v8, "% free"
  
      invoke-virtual {p3, v8}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
***************
*** 201,210 ****
--- 223,234 ----
      .parameter "args"
  
      .prologue
+     .line 48
      const/16 v11, 0x200
  
      new-array v9, v11, [B
  
+     .line 49
      .local v9, junk:[B
      const/4 v8, 0x0
  
***************
*** 222,227 ****
--- 246,252 ----
  
      goto :goto_0
  
+     .line 51
      :cond_0
      new-instance v10, Ljava/io/File;
  
***************
*** 233,249 ****
--- 258,278 ----
  
      invoke-direct {v10, v11, v12}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 52
      .local v10, tmp:Ljava/io/File;
      const/4 v6, 0x0
  
+     .line 53
      .local v6, fos:Ljava/io/FileOutputStream;
      const/4 v5, 0x0
  
+     .line 55
      .local v5, error:Ljava/io/IOException;
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v2
  
+     .line 57
      .local v2, before:J
      :try_start_0
      new-instance v7, Ljava/io/FileOutputStream;
***************
*** 253,258 ****
--- 282,288 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 58
      .end local v6           #fos:Ljava/io/FileOutputStream;
      .local v7, fos:Ljava/io/FileOutputStream;
      :try_start_1
***************
*** 261,266 ****
--- 291,297 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_1
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_4
  
+     .line 62
      if-eqz v7, :cond_1
  
      :try_start_2
***************
*** 272,277 ****
--- 303,309 ----
      :goto_1
      move-object v6, v7
  
+     .line 65
      .end local v7           #fos:Ljava/io/FileOutputStream;
      .restart local v6       #fos:Ljava/io/FileOutputStream;
      :cond_2
***************
*** 280,285 ****
--- 312,318 ----
  
      move-result-wide v0
  
+     .line 66
      .local v0, after:J
      invoke-virtual {v10}, Ljava/io/File;->exists()Z
  
***************
*** 289,307 ****
--- 322,344 ----
  
      invoke-virtual {v10}, Ljava/io/File;->delete()Z
  
+     .line 68
      :cond_3
      if-eqz v5, :cond_5
  
+     .line 69
      const-string v11, "Test-Error: "
  
      invoke-virtual {p2, v11}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 70
      invoke-virtual {v5}, Ljava/io/IOException;->toString()Ljava/lang/String;
  
      move-result-object v11
  
      invoke-virtual {p2, v11}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 77
      :goto_3
      invoke-static {}, Landroid/os/Environment;->getDataDirectory()Ljava/io/File;
  
***************
*** 311,316 ****
--- 348,354 ----
  
      invoke-direct {p0, v11, v12, p2}, Lcom/android/server/DiskStatsService;->reportFreeSpace(Ljava/io/File;Ljava/lang/String;Ljava/io/PrintWriter;)V
  
+     .line 78
      invoke-static {}, Landroid/os/Environment;->getDownloadCacheDirectory()Ljava/io/File;
  
      move-result-object v11
***************
*** 319,324 ****
--- 357,363 ----
  
      invoke-direct {p0, v11, v12, p2}, Lcom/android/server/DiskStatsService;->reportFreeSpace(Ljava/io/File;Ljava/lang/String;Ljava/io/PrintWriter;)V
  
+     .line 79
      new-instance v11, Ljava/io/File;
  
      const-string v12, "/system"
***************
*** 329,344 ****
--- 368,387 ----
  
      invoke-direct {p0, v11, v12, p2}, Lcom/android/server/DiskStatsService;->reportFreeSpace(Ljava/io/File;Ljava/lang/String;Ljava/io/PrintWriter;)V
  
+     .line 83
      return-void
  
+     .line 59
      .end local v0           #after:J
      :catch_0
      move-exception v4
  
+     .line 60
      .local v4, e:Ljava/io/IOException;
      :goto_4
      move-object v5, v4
  
+     .line 62
      if-eqz v6, :cond_2
  
      :try_start_3
***************
*** 369,390 ****
--- 412,437 ----
      :goto_6
      throw v11
  
+     .line 72
      .restart local v0       #after:J
      :cond_5
      const-string v11, "Latency: "
  
      invoke-virtual {p2, v11}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 73
      sub-long v11, v0, v2
  
      invoke-virtual {p2, v11, v12}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 74
      const-string v11, "ms [512B Data Write]"
  
      invoke-virtual {p2, v11}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
      goto :goto_3
  
+     .line 62
      .end local v0           #after:J
      :catch_2
      move-exception v12
***************
*** 407,412 ****
--- 454,460 ----
      .restart local v6       #fos:Ljava/io/FileOutputStream;
      goto :goto_5
  
+     .line 59
      .end local v6           #fos:Ljava/io/FileOutputStream;
      .restart local v7       #fos:Ljava/io/FileOutputStream;
      :catch_4
