*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DockObserver$1.smali	2012-03-23 19:34:57.480540943 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DockObserver$1.smali.original	2012-03-23 19:34:46.926348487 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 130
      iput-object p1, p0, Lcom/android/server/DockObserver$1;->this$0:Lcom/android/server/DockObserver;
  
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
***************
*** 46,61 ****
--- 47,66 ----
  
      const/4 v10, 0x1
  
+     .line 133
      iget v7, p1, Landroid/os/Message;->what:I
  
      packed-switch v7, :pswitch_data_0
  
+     .line 199
      :goto_0
      return-void
  
+     .line 135
      :pswitch_0
      monitor-enter p0
  
+     .line 136
      :try_start_0
      invoke-static {}, Lcom/android/server/DockObserver;->access$000()Ljava/lang/String;
  
***************
*** 88,93 ****
--- 93,99 ----
  
      invoke-static {v7, v8}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 138
      iget-object v7, p0, Lcom/android/server/DockObserver$1;->this$0:Lcom/android/server/DockObserver;
  
      #getter for: Lcom/android/server/DockObserver;->mContext:Landroid/content/Context;
***************
*** 99,104 ****
--- 105,111 ----
  
      move-result-object v1
  
+     .line 140
      .local v1, cr:Landroid/content/ContentResolver;
      const-string v7, "device_provisioned"
  
***************
*** 110,115 ****
--- 117,123 ----
  
      if-nez v7, :cond_0
  
+     .line 142
      invoke-static {}, Lcom/android/server/DockObserver;->access$000()Ljava/lang/String;
  
      move-result-object v7
***************
*** 118,127 ****
--- 126,137 ----
  
      invoke-static {v7, v8}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 143
      monitor-exit p0
  
      goto :goto_0
  
+     .line 196
      .end local v1           #cr:Landroid/content/ContentResolver;
      :catchall_0
      move-exception v7
***************
*** 132,137 ****
--- 142,148 ----
  
      throw v7
  
+     .line 146
      .restart local v1       #cr:Landroid/content/ContentResolver;
      :cond_0
      :try_start_1
***************
*** 141,151 ****
--- 152,164 ----
  
      invoke-direct {v2, v7}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 147
      .local v2, intent:Landroid/content/Intent;
      const/high16 v7, 0x2000
  
      invoke-virtual {v2, v7}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 148
      const-string v7, "android.intent.extra.DOCK_STATE"
  
      iget-object v8, p0, Lcom/android/server/DockObserver$1;->this$0:Lcom/android/server/DockObserver;
***************
*** 157,169 ****
--- 170,185 ----
  
      invoke-virtual {v2, v7, v8}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 151
      invoke-static {}, Landroid/server/BluetoothService;->readDockBluetoothAddress()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 152
      .local v0, address:Ljava/lang/String;
      if-eqz v0, :cond_1
  
+     .line 153
      const-string v7, "android.bluetooth.device.extra.DEVICE"
  
      invoke-static {}, Landroid/bluetooth/BluetoothAdapter;->getDefaultAdapter()Landroid/bluetooth/BluetoothAdapter;
***************
*** 176,181 ****
--- 192,198 ----
  
      invoke-virtual {v2, v7, v8}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 158
      :cond_1
      const-string v7, "dock_sounds_enabled"
  
***************
*** 187,194 ****
--- 204,213 ----
  
      if-ne v7, v10, :cond_4
  
+     .line 161
      const/4 v6, 0x0
  
+     .line 162
      .local v6, whichSound:Ljava/lang/String;
      iget-object v7, p0, Lcom/android/server/DockObserver$1;->this$0:Lcom/android/server/DockObserver;
  
***************
*** 199,204 ****
--- 218,224 ----
  
      if-nez v7, :cond_6
  
+     .line 163
      iget-object v7, p0, Lcom/android/server/DockObserver$1;->this$0:Lcom/android/server/DockObserver;
  
      #getter for: Lcom/android/server/DockObserver;->mPreviousDockState:I
***************
*** 226,245 ****
--- 246,270 ----
  
      if-ne v7, v13, :cond_5
  
+     .line 166
      :cond_2
      const-string v6, "desk_undock_sound"
  
+     .line 180
      :cond_3
      :goto_1
      if-eqz v6, :cond_4
  
+     .line 181
      invoke-static {v1, v6}, Landroid/provider/Settings$System;->getString(Landroid/content/ContentResolver;Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v4
  
+     .line 182
      .local v4, soundPath:Ljava/lang/String;
      if-eqz v4, :cond_4
  
+     .line 183
      new-instance v7, Ljava/lang/StringBuilder;
  
      invoke-direct {v7}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 262,270 ****
--- 287,297 ----
  
      move-result-object v5
  
+     .line 184
      .local v5, soundUri:Landroid/net/Uri;
      if-eqz v5, :cond_4
  
+     .line 185
      iget-object v7, p0, Lcom/android/server/DockObserver$1;->this$0:Lcom/android/server/DockObserver;
  
      #getter for: Lcom/android/server/DockObserver;->mContext:Landroid/content/Context;
***************
*** 276,290 ****
--- 303,321 ----
  
      move-result-object v3
  
+     .line 186
      .local v3, sfx:Landroid/media/Ringtone;
      if-eqz v3, :cond_4
  
+     .line 187
      const/4 v7, 0x1
  
      invoke-virtual {v3, v7}, Landroid/media/Ringtone;->setStreamType(I)V
  
+     .line 188
      invoke-virtual {v3}, Landroid/media/Ringtone;->play()V
  
+     .line 195
      .end local v3           #sfx:Landroid/media/Ringtone;
      .end local v4           #soundPath:Ljava/lang/String;
      .end local v5           #soundUri:Landroid/net/Uri;
***************
*** 299,308 ****
--- 330,341 ----
  
      invoke-virtual {v7, v2}, Landroid/content/Context;->sendStickyBroadcast(Landroid/content/Intent;)V
  
+     .line 196
      monitor-exit p0
  
      goto/16 :goto_0
  
+     .line 167
      .restart local v6       #whichSound:Ljava/lang/String;
      :cond_5
      iget-object v7, p0, Lcom/android/server/DockObserver$1;->this$0:Lcom/android/server/DockObserver;
***************
*** 314,323 ****
--- 347,358 ----
  
      if-ne v7, v11, :cond_3
  
+     .line 168
      const-string v6, "car_undock_sound"
  
      goto :goto_1
  
+     .line 171
      :cond_6
      iget-object v7, p0, Lcom/android/server/DockObserver$1;->this$0:Lcom/android/server/DockObserver;
  
***************
*** 346,356 ****
--- 381,393 ----
  
      if-ne v7, v13, :cond_8
  
+     .line 174
      :cond_7
      const-string v6, "desk_dock_sound"
  
      goto :goto_1
  
+     .line 175
      :cond_8
      iget-object v7, p0, Lcom/android/server/DockObserver$1;->this$0:Lcom/android/server/DockObserver;
  
***************
*** 361,372 ****
--- 398,411 ----
  
      if-ne v7, v11, :cond_3
  
+     .line 176
      const-string v6, "car_dock_sound"
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
      goto :goto_1
  
+     .line 133
      nop
  
      :pswitch_data_0
