*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DropBoxManagerService$EntryFile.smali	2012-03-23 19:35:04.353666274 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DropBoxManagerService$EntryFile.smali.original	2012-03-23 19:34:47.097351606 -0500
***************
*** 47,68 ****
--- 47,75 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 547
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 548
      iput-object v1, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->tag:Ljava/lang/String;
  
+     .line 549
      iput-wide p1, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->timestampMillis:J
  
+     .line 550
      const/4 v0, 0x1
  
      iput v0, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->flags:I
  
+     .line 551
      iput-object v1, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
+     .line 552
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->blocks:I
  
+     .line 553
      return-void
  .end method
  
***************
*** 72,81 ****
--- 79,91 ----
      .parameter "blockSize"
  
      .prologue
+     .line 504
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 505
      iput-object p1, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
+     .line 506
      iget-object v6, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
      invoke-virtual {v6}, Ljava/io/File;->length()J
***************
*** 98,107 ****
--- 108,119 ----
  
      iput v6, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->blocks:I
  
+     .line 508
      invoke-virtual {p1}, Ljava/io/File;->getName()Ljava/lang/String;
  
      move-result-object v5
  
+     .line 509
      .local v5, name:Ljava/lang/String;
      const/16 v6, 0x40
  
***************
*** 109,135 ****
--- 121,154 ----
  
      move-result v0
  
+     .line 510
      .local v0, at:I
      if-gez v0, :cond_0
  
+     .line 511
      const/4 v6, 0x0
  
      iput-object v6, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->tag:Ljava/lang/String;
  
+     .line 512
      const-wide/16 v6, 0x0
  
      iput-wide v6, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->timestampMillis:J
  
+     .line 513
      const/4 v6, 0x1
  
      iput v6, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->flags:I
  
+     .line 541
      :goto_0
      return-void
  
+     .line 517
      :cond_0
      const/4 v2, 0x0
  
+     .line 518
      .local v2, flags:I
      const/4 v6, 0x0
  
***************
*** 143,148 ****
--- 162,168 ----
  
      iput-object v6, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->tag:Ljava/lang/String;
  
+     .line 519
      const-string v6, ".gz"
  
      invoke-virtual {v5, v6}, Ljava/lang/String;->endsWith(Ljava/lang/String;)Z
***************
*** 151,158 ****
--- 171,180 ----
  
      if-eqz v6, :cond_1
  
+     .line 520
      or-int/lit8 v2, v2, 0x4
  
+     .line 521
      const/4 v6, 0x0
  
      invoke-virtual {v5}, Ljava/lang/String;->length()I
***************
*** 165,170 ****
--- 187,193 ----
  
      move-result-object v5
  
+     .line 523
      :cond_1
      const-string v6, ".lost"
  
***************
*** 174,181 ****
--- 197,206 ----
  
      if-eqz v6, :cond_2
  
+     .line 524
      or-int/lit8 v2, v2, 0x1
  
+     .line 525
      add-int/lit8 v6, v0, 0x1
  
      invoke-virtual {v5}, Ljava/lang/String;->length()I
***************
*** 188,196 ****
--- 213,223 ----
  
      move-result-object v5
  
+     .line 536
      :goto_1
      iput v2, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->flags:I
  
+     .line 539
      :try_start_0
      invoke-static {v5}, Ljava/lang/Long;->valueOf(Ljava/lang/String;)Ljava/lang/Long;
  
***************
*** 202,213 ****
--- 229,242 ----
  
      move-result-wide v3
  
+     .line 540
      .local v3, millis:J
      :goto_2
      iput-wide v3, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->timestampMillis:J
  
      goto :goto_0
  
+     .line 526
      .end local v3           #millis:J
      :cond_2
      const-string v6, ".txt"
***************
*** 218,225 ****
--- 247,256 ----
  
      if-eqz v6, :cond_3
  
+     .line 527
      or-int/lit8 v2, v2, 0x2
  
+     .line 528
      add-int/lit8 v6, v0, 0x1
  
      invoke-virtual {v5}, Ljava/lang/String;->length()I
***************
*** 234,239 ****
--- 265,271 ----
  
      goto :goto_1
  
+     .line 529
      :cond_3
      const-string v6, ".dat"
  
***************
*** 243,248 ****
--- 275,281 ----
  
      if-eqz v6, :cond_4
  
+     .line 530
      add-int/lit8 v6, v0, 0x1
  
      invoke-virtual {v5}, Ljava/lang/String;->length()I
***************
*** 257,273 ****
--- 290,309 ----
  
      goto :goto_1
  
+     .line 532
      :cond_4
      const/4 v6, 0x1
  
      iput v6, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->flags:I
  
+     .line 533
      const-wide/16 v6, 0x0
  
      iput-wide v6, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->timestampMillis:J
  
      goto :goto_0
  
+     .line 539
      :catch_0
      move-exception v1
  
***************
*** 293,300 ****
--- 329,338 ----
      .end annotation
  
      .prologue
+     .line 467
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 468
      and-int/lit8 v0, p6, 0x1
  
      if-eqz v0, :cond_0
***************
*** 305,317 ****
--- 343,359 ----
  
      throw v0
  
+     .line 470
      :cond_0
      iput-object p3, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->tag:Ljava/lang/String;
  
+     .line 471
      iput-wide p4, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->timestampMillis:J
  
+     .line 472
      iput p6, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->flags:I
  
+     .line 473
      new-instance v1, Ljava/io/File;
  
      new-instance v0, Ljava/lang/StringBuilder;
***************
*** 366,371 ****
--- 408,414 ----
  
      iput-object v1, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
+     .line 477
      iget-object v0, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
      invoke-virtual {p1, v0}, Ljava/io/File;->renameTo(Ljava/io/File;)Z
***************
*** 374,379 ****
--- 417,423 ----
  
      if-nez v0, :cond_3
  
+     .line 478
      new-instance v0, Ljava/io/IOException;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 410,415 ****
--- 454,460 ----
  
      throw v0
  
+     .line 473
      :cond_1
      const-string v0, ".dat"
  
***************
*** 420,425 ****
--- 465,471 ----
  
      goto :goto_1
  
+     .line 480
      :cond_3
      iget-object v0, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
***************
*** 443,448 ****
--- 489,495 ----
  
      iput v0, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->blocks:I
  
+     .line 481
      return-void
  .end method
  
***************
*** 458,473 ****
--- 505,525 ----
      .end annotation
  
      .prologue
+     .line 490
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 491
      iput-object p2, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->tag:Ljava/lang/String;
  
+     .line 492
      iput-wide p3, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->timestampMillis:J
  
+     .line 493
      const/4 v0, 0x1
  
      iput v0, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->flags:I
  
+     .line 494
      new-instance v0, Ljava/io/File;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 506,515 ****
--- 558,569 ----
  
      iput-object v0, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
+     .line 495
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->blocks:I
  
+     .line 496
      new-instance v0, Ljava/io/FileOutputStream;
  
      iget-object v1, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
***************
*** 518,523 ****
--- 572,578 ----
  
      invoke-virtual {v0}, Ljava/io/FileOutputStream;->close()V
  
+     .line 497
      return-void
  .end method
  
***************
*** 534,539 ****
--- 589,595 ----
  
      const/4 v0, -0x1
  
+     .line 445
      iget-wide v3, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->timestampMillis:J
  
      iget-wide v5, p1, Lcom/android/server/DropBoxManagerService$EntryFile;->timestampMillis:J
***************
*** 542,551 ****
--- 598,609 ----
  
      if-gez v3, :cond_1
  
+     .line 453
      :cond_0
      :goto_0
      return v0
  
+     .line 446
      :cond_1
      iget-wide v3, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->timestampMillis:J
  
***************
*** 559,564 ****
--- 617,623 ----
  
      goto :goto_0
  
+     .line 447
      :cond_2
      iget-object v3, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
***************
*** 578,588 ****
--- 637,649 ----
  
      goto :goto_0
  
+     .line 448
      :cond_3
      iget-object v3, p1, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
      if-nez v3, :cond_0
  
+     .line 449
      iget-object v3, p0, Lcom/android/server/DropBoxManagerService$EntryFile;->file:Ljava/io/File;
  
      if-eqz v3, :cond_4
***************
*** 591,596 ****
--- 652,658 ----
  
      goto :goto_0
  
+     .line 450
      :cond_4
      if-ne p0, p1, :cond_5
  
***************
*** 598,603 ****
--- 660,666 ----
  
      goto :goto_0
  
+     .line 451
      :cond_5
      invoke-virtual {p0}, Ljava/lang/Object;->hashCode()I
  
***************
*** 609,614 ****
--- 672,678 ----
  
      if-lt v3, v4, :cond_0
  
+     .line 452
      invoke-virtual {p0}, Ljava/lang/Object;->hashCode()I
  
      move-result v0
***************
*** 626,631 ****
--- 690,696 ----
      :cond_6
      move v0, v2
  
+     .line 453
      goto :goto_0
  .end method
  
***************
*** 634,639 ****
--- 699,705 ----
      .parameter "x0"
  
      .prologue
+     .line 436
      check-cast p1, Lcom/android/server/DropBoxManagerService$EntryFile;
  
      .end local p1
