*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DropBoxManagerService$1$1.smali	2012-03-23 19:34:51.396430001 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/DropBoxManagerService$1$1.smali.original	2012-03-23 19:34:46.820346555 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 111
      iput-object p1, p0, Lcom/android/server/DropBoxManagerService$1$1;->this$1:Lcom/android/server/DropBoxManagerService$1;
  
      invoke-direct {p0}, Ljava/lang/Thread;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .locals 3
  
      .prologue
+     .line 114
      :try_start_0
      iget-object v1, p0, Lcom/android/server/DropBoxManagerService$1$1;->this$1:Lcom/android/server/DropBoxManagerService$1;
  
***************
*** 45,50 ****
--- 47,53 ----
      #calls: Lcom/android/server/DropBoxManagerService;->init()V
      invoke-static {v1}, Lcom/android/server/DropBoxManagerService;->access$200(Lcom/android/server/DropBoxManagerService;)V
  
+     .line 115
      iget-object v1, p0, Lcom/android/server/DropBoxManagerService$1$1;->this$1:Lcom/android/server/DropBoxManagerService$1;
  
      iget-object v1, v1, Lcom/android/server/DropBoxManagerService$1;->this$0:Lcom/android/server/DropBoxManagerService;
***************
*** 54,65 ****
--- 57,71 ----
      :try_end_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 119
      :goto_0
      return-void
  
+     .line 116
      :catch_0
      move-exception v0
  
+     .line 117
      .local v0, e:Ljava/io/IOException;
      const-string v1, "DropBoxManagerService"
  
