*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/InputMethodManagerService$InputMethodFileManager.smali	2012-03-23 19:34:50.455412840 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/InputMethodManagerService$InputMethodFileManager.smali.original	2012-03-23 19:34:46.812346410 -0500
***************
*** 88,104 ****
--- 88,108 ----
      .end annotation
  
      .prologue
+     .line 3169
      .local p1, methodMap:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/String;Landroid/view/inputmethod/InputMethodInfo;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 3167
      new-instance v3, Ljava/util/HashMap;
  
      invoke-direct {v3}, Ljava/util/HashMap;-><init>()V
  
      iput-object v3, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mSubtypesMap:Ljava/util/HashMap;
  
+     .line 3170
      if-nez p1, :cond_0
  
+     .line 3171
      new-instance v3, Ljava/lang/NullPointerException;
  
      const-string v4, "methodMap is null"
***************
*** 107,115 ****
--- 111,121 ----
  
      throw v3
  
+     .line 3173
      :cond_0
      iput-object p1, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mMethodMap:Ljava/util/HashMap;
  
+     .line 3174
      new-instance v2, Ljava/io/File;
  
      invoke-static {}, Landroid/os/Environment;->getDataDirectory()Ljava/io/File;
***************
*** 120,125 ****
--- 126,132 ----
  
      invoke-direct {v2, v3, v4}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 3175
      .local v2, systemDir:Ljava/io/File;
      new-instance v0, Ljava/io/File;
  
***************
*** 127,132 ****
--- 134,140 ----
  
      invoke-direct {v0, v2, v3}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 3176
      .local v0, inputMethodDir:Ljava/io/File;
      invoke-virtual {v0}, Ljava/io/File;->mkdirs()Z
  
***************
*** 134,139 ****
--- 142,148 ----
  
      if-nez v3, :cond_1
  
+     .line 3177
      const-string v3, "InputManagerService"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 160,165 ****
--- 169,175 ----
  
      invoke-static {v3, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 3179
      :cond_1
      new-instance v1, Ljava/io/File;
  
***************
*** 167,172 ****
--- 177,183 ----
  
      invoke-direct {v1, v0, v3}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 3180
      .local v1, subtypeFile:Ljava/io/File;
      new-instance v3, Lcom/android/internal/os/AtomicFile;
  
***************
*** 174,194 ****
--- 185,209 ----
  
      iput-object v3, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mAdditionalInputMethodSubtypeFile:Lcom/android/internal/os/AtomicFile;
  
+     .line 3181
      invoke-virtual {v1}, Ljava/io/File;->exists()Z
  
      move-result v3
  
      if-nez v3, :cond_2
  
+     .line 3183
      iget-object v3, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mSubtypesMap:Ljava/util/HashMap;
  
      iget-object v4, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mAdditionalInputMethodSubtypeFile:Lcom/android/internal/os/AtomicFile;
  
      invoke-static {v3, v4, p1}, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->writeAdditionalInputMethodSubtypes(Ljava/util/HashMap;Lcom/android/internal/os/AtomicFile;Ljava/util/HashMap;)V
  
+     .line 3188
      :goto_0
      return-void
  
+     .line 3186
      :cond_2
      iget-object v3, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mSubtypesMap:Ljava/util/HashMap;
  
***************
*** 205,210 ****
--- 220,226 ----
      .parameter "x1"
  
      .prologue
+     .line 3151
      invoke-direct {p0, p1}, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->deleteAllInputMethodSubtypes(Ljava/lang/String;)V
  
      return-void
***************
*** 215,229 ****
--- 231,248 ----
      .parameter "imiId"
  
      .prologue
+     .line 3191
      iget-object v1, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mMethodMap:Ljava/util/HashMap;
  
      monitor-enter v1
  
+     .line 3192
      :try_start_0
      iget-object v0, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mSubtypesMap:Ljava/util/HashMap;
  
      invoke-virtual {v0, p1}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 3193
      iget-object v0, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mSubtypesMap:Ljava/util/HashMap;
  
      iget-object v2, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mAdditionalInputMethodSubtypeFile:Lcom/android/internal/os/AtomicFile;
***************
*** 232,241 ****
--- 251,263 ----
  
      invoke-static {v0, v2, v3}, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->writeAdditionalInputMethodSubtypes(Ljava/util/HashMap;Lcom/android/internal/os/AtomicFile;Ljava/util/HashMap;)V
  
+     .line 3195
      monitor-exit v1
  
+     .line 3196
      return-void
  
+     .line 3195
      :catchall_0
      move-exception v0
  
***************
*** 266,295 ****
--- 288,324 ----
      .end annotation
  
      .prologue
+     .line 3270
      .local p0, allSubtypes:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/String;Ljava/util/List<Landroid/view/inputmethod/InputMethodSubtype;>;>;"
      if-eqz p0, :cond_0
  
      if-nez p1, :cond_1
  
+     .line 3342
      :cond_0
      :goto_0
      return-void
  
+     .line 3271
      :cond_1
      invoke-virtual/range {p0 .. p0}, Ljava/util/HashMap;->clear()V
  
+     .line 3272
      const/4 v15, 0x0
  
+     .line 3274
      .local v15, fis:Ljava/io/FileInputStream;
      :try_start_0
      invoke-virtual/range {p1 .. p1}, Lcom/android/internal/os/AtomicFile;->openRead()Ljava/io/FileInputStream;
  
      move-result-object v15
  
+     .line 3275
      invoke-static {}, Landroid/util/Xml;->newPullParser()Lorg/xmlpull/v1/XmlPullParser;
  
      move-result-object v17
  
+     .line 3276
      .local v17, parser:Lorg/xmlpull/v1/XmlPullParser;
      const/16 v20, 0x0
  
***************
*** 299,308 ****
--- 328,339 ----
  
      invoke-interface {v0, v15, v1}, Lorg/xmlpull/v1/XmlPullParser;->setInput(Ljava/io/InputStream;Ljava/lang/String;)V
  
+     .line 3277
      invoke-interface/range {v17 .. v17}, Lorg/xmlpull/v1/XmlPullParser;->getEventType()I
  
      move-result v19
  
+     .line 3280
      .local v19, type:I
      :cond_2
      invoke-interface/range {v17 .. v17}, Lorg/xmlpull/v1/XmlPullParser;->next()I
***************
*** 325,335 ****
--- 356,368 ----
  
      if-ne v0, v1, :cond_2
  
+     .line 3281
      :cond_3
      invoke-interface/range {v17 .. v17}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
  
      move-result-object v14
  
+     .line 3282
      .local v14, firstNodeName:Ljava/lang/String;
      const-string v20, "subtypes"
  
***************
*** 341,346 ****
--- 374,380 ----
  
      if-nez v20, :cond_4
  
+     .line 3283
      new-instance v20, Lorg/xmlpull/v1/XmlPullParserException;
  
      const-string v21, "Xml doesn\'t start with subtypes"
***************
*** 354,365 ****
--- 388,401 ----
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_2
      .catch Ljava/lang/NumberFormatException; {:try_start_0 .. :try_end_0} :catch_4
  
+     .line 3324
      .end local v14           #firstNodeName:Ljava/lang/String;
      .end local v17           #parser:Lorg/xmlpull/v1/XmlPullParser;
      .end local v19           #type:I
      :catch_0
      move-exception v12
  
+     .line 3325
      .local v12, e:Lorg/xmlpull/v1/XmlPullParserException;
      :try_start_1
      const-string v20, "InputManagerService"
***************
*** 388,395 ****
--- 424,433 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 3334
      if-eqz v15, :cond_0
  
+     .line 3336
      :try_start_2
      invoke-virtual {v15}, Ljava/io/FileInputStream;->close()V
      :try_end_2
***************
*** 397,405 ****
--- 435,445 ----
  
      goto :goto_0
  
+     .line 3337
      :catch_1
      move-exception v13
  
+     .line 3338
      .local v13, e1:Ljava/io/IOException;
      const-string v20, "InputManagerService"
  
***************
*** 411,416 ****
--- 451,457 ----
  
      goto :goto_0
  
+     .line 3285
      .end local v13           #e1:Ljava/io/IOException;
      .restart local v14       #firstNodeName:Ljava/lang/String;
      .restart local v17       #parser:Lorg/xmlpull/v1/XmlPullParser;
***************
*** 421,432 ****
--- 462,476 ----
  
      move-result v11
  
+     .line 3286
      .local v11, depth:I
      const/4 v10, 0x0
  
+     .line 3287
      .local v10, currentImiId:Ljava/lang/String;
      const/16 v18, 0x0
  
+     .line 3289
      .local v18, tempSubtypesArray:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/view/inputmethod/InputMethodSubtype;>;"
      :cond_5
      :goto_2
***************
*** 459,464 ****
--- 503,509 ----
  
      if-eq v0, v1, :cond_c
  
+     .line 3290
      const/16 v20, 0x2
  
      move/from16 v0, v19
***************
*** 467,476 ****
--- 512,523 ----
  
      if-ne v0, v1, :cond_5
  
+     .line 3292
      invoke-interface/range {v17 .. v17}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
  
      move-result-object v16
  
+     .line 3293
      .local v16, nodeName:Ljava/lang/String;
      const-string v20, "imi"
  
***************
*** 484,489 ****
--- 531,537 ----
  
      if-eqz v20, :cond_8
  
+     .line 3294
      const/16 v20, 0x0
  
      const-string v21, "id"
***************
*** 498,509 ****
--- 546,559 ----
  
      move-result-object v10
  
+     .line 3295
      invoke-static {v10}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v20
  
      if-eqz v20, :cond_7
  
+     .line 3296
      const-string v20, "InputManagerService"
  
      const-string v21, "Invalid imi id found in subtypes.xml"
***************
*** 517,522 ****
--- 567,573 ----
  
      goto :goto_2
  
+     .line 3327
      .end local v10           #currentImiId:Ljava/lang/String;
      .end local v11           #depth:I
      .end local v14           #firstNodeName:Ljava/lang/String;
***************
*** 527,532 ****
--- 578,584 ----
      :catch_2
      move-exception v12
  
+     .line 3328
      .local v12, e:Ljava/io/IOException;
      :try_start_4
      const-string v20, "InputManagerService"
***************
*** 555,562 ****
--- 607,616 ----
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
  
+     .line 3334
      if-eqz v15, :cond_0
  
+     .line 3336
      :try_start_5
      invoke-virtual {v15}, Ljava/io/FileInputStream;->close()V
      :try_end_5
***************
*** 564,572 ****
--- 618,628 ----
  
      goto/16 :goto_0
  
+     .line 3337
      :catch_3
      move-exception v13
  
+     .line 3338
      .restart local v13       #e1:Ljava/io/IOException;
      const-string v20, "InputManagerService"
  
***************
*** 574,579 ****
--- 630,636 ----
  
      goto/16 :goto_1
  
+     .line 3299
      .end local v12           #e:Ljava/io/IOException;
      .end local v13           #e1:Ljava/io/IOException;
      .restart local v10       #currentImiId:Ljava/lang/String;
***************
*** 590,595 ****
--- 647,653 ----
      .end local v18           #tempSubtypesArray:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/view/inputmethod/InputMethodSubtype;>;"
      invoke-direct/range {v18 .. v18}, Ljava/util/ArrayList;-><init>()V
  
+     .line 3300
      .restart local v18       #tempSubtypesArray:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/view/inputmethod/InputMethodSubtype;>;"
      move-object/from16 v0, p0
  
***************
*** 604,609 ****
--- 662,668 ----
  
      goto/16 :goto_2
  
+     .line 3330
      .end local v10           #currentImiId:Ljava/lang/String;
      .end local v11           #depth:I
      .end local v14           #firstNodeName:Ljava/lang/String;
***************
*** 614,619 ****
--- 673,679 ----
      :catch_4
      move-exception v12
  
+     .line 3331
      .local v12, e:Ljava/lang/NumberFormatException;
      :try_start_7
      const-string v20, "InputManagerService"
***************
*** 642,649 ****
--- 702,711 ----
      :try_end_7
      .catchall {:try_start_7 .. :try_end_7} :catchall_0
  
+     .line 3334
      if-eqz v15, :cond_0
  
+     .line 3336
      :try_start_8
      invoke-virtual {v15}, Ljava/io/FileInputStream;->close()V
      :try_end_8
***************
*** 651,659 ****
--- 713,723 ----
  
      goto/16 :goto_0
  
+     .line 3337
      :catch_5
      move-exception v13
  
+     .line 3338
      .restart local v13       #e1:Ljava/io/IOException;
      const-string v20, "InputManagerService"
  
***************
*** 661,666 ****
--- 725,731 ----
  
      goto/16 :goto_1
  
+     .line 3301
      .end local v12           #e:Ljava/lang/NumberFormatException;
      .end local v13           #e1:Ljava/io/IOException;
      .restart local v10       #currentImiId:Ljava/lang/String;
***************
*** 684,689 ****
--- 749,755 ----
  
      if-eqz v20, :cond_5
  
+     .line 3302
      invoke-static {v10}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v20
***************
*** 692,697 ****
--- 758,764 ----
  
      if-nez v18, :cond_b
  
+     .line 3303
      :cond_9
      const-string v20, "InputManagerService"
  
***************
*** 724,729 ****
--- 791,797 ----
  
      goto/16 :goto_2
  
+     .line 3334
      .end local v10           #currentImiId:Ljava/lang/String;
      .end local v11           #depth:I
      .end local v14           #firstNodeName:Ljava/lang/String;
***************
*** 736,750 ****
--- 804,821 ----
  
      if-eqz v15, :cond_a
  
+     .line 3336
      :try_start_a
      invoke-virtual {v15}, Ljava/io/FileInputStream;->close()V
      :try_end_a
      .catch Ljava/io/IOException; {:try_start_a .. :try_end_a} :catch_6
  
+     .line 3334
      :cond_a
      :goto_3
      throw v20
  
+     .line 3306
      .restart local v10       #currentImiId:Ljava/lang/String;
      .restart local v11       #depth:I
      .restart local v14       #firstNodeName:Ljava/lang/String;
***************
*** 776,781 ****
--- 847,853 ----
  
      move-result v5
  
+     .line 3308
      .local v5, icon:I
      const/16 v20, 0x0
  
***************
*** 799,804 ****
--- 871,877 ----
  
      move-result v4
  
+     .line 3310
      .local v4, label:I
      const/16 v20, 0x0
  
***************
*** 814,819 ****
--- 887,893 ----
  
      move-result-object v6
  
+     .line 3312
      .local v6, imeSubtypeLocale:Ljava/lang/String;
      const/16 v20, 0x0
  
***************
*** 829,834 ****
--- 903,909 ----
  
      move-result-object v7
  
+     .line 3314
      .local v7, imeSubtypeMode:Ljava/lang/String;
      const/16 v20, 0x0
  
***************
*** 844,849 ****
--- 919,925 ----
  
      move-result-object v8
  
+     .line 3316
      .local v8, imeSubtypeExtraValue:Ljava/lang/String;
      const-string v20, "1"
  
***************
*** 869,879 ****
--- 945,957 ----
  
      move-result v9
  
+     .line 3318
      .local v9, isAuxiliary:Z
      new-instance v3, Landroid/view/inputmethod/InputMethodSubtype;
  
      invoke-direct/range {v3 .. v9}, Landroid/view/inputmethod/InputMethodSubtype;-><init>(IILjava/lang/String;Ljava/lang/String;Ljava/lang/String;Z)V
  
+     .line 3321
      .local v3, subtype:Landroid/view/inputmethod/InputMethodSubtype;
      move-object/from16 v0, v18
  
***************
*** 886,891 ****
--- 964,970 ----
  
      goto/16 :goto_2
  
+     .line 3337
      .end local v3           #subtype:Landroid/view/inputmethod/InputMethodSubtype;
      .end local v4           #label:I
      .end local v5           #icon:I
***************
*** 903,908 ****
--- 982,988 ----
      :catch_6
      move-exception v13
  
+     .line 3338
      .restart local v13       #e1:Ljava/io/IOException;
      const-string v21, "InputManagerService"
  
***************
*** 912,917 ****
--- 992,998 ----
  
      goto/16 :goto_3
  
+     .line 3334
      .end local v13           #e1:Ljava/io/IOException;
      .restart local v10       #currentImiId:Ljava/lang/String;
      .restart local v11       #depth:I
***************
*** 922,927 ****
--- 1003,1009 ----
      :cond_c
      if-eqz v15, :cond_0
  
+     .line 3336
      :try_start_c
      invoke-virtual {v15}, Ljava/io/FileInputStream;->close()V
      :try_end_c
***************
*** 929,937 ****
--- 1011,1021 ----
  
      goto/16 :goto_0
  
+     .line 3337
      :catch_7
      move-exception v13
  
+     .line 3338
      .restart local v13       #e1:Ljava/io/IOException;
      const-string v20, "InputManagerService"
  
***************
*** 965,970 ****
--- 1049,1055 ----
      .end annotation
  
      .prologue
+     .line 3225
      .local p0, allSubtypes:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/String;Ljava/util/List<Landroid/view/inputmethod/InputMethodSubtype;>;>;"
      .local p2, methodMap:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/String;Landroid/view/inputmethod/InputMethodInfo;>;"
      if-eqz p2, :cond_1
***************
*** 977,1001 ****
--- 1062,1091 ----
  
      const/4 v6, 0x1
  
+     .line 3226
      .local v6, isSetMethodMap:Z
      :goto_0
      const/4 v2, 0x0
  
+     .line 3228
      .local v2, fos:Ljava/io/FileOutputStream;
      :try_start_0
      invoke-virtual {p1}, Lcom/android/internal/os/AtomicFile;->startWrite()Ljava/io/FileOutputStream;
  
      move-result-object v2
  
+     .line 3229
      new-instance v7, Lcom/android/internal/util/FastXmlSerializer;
  
      invoke-direct {v7}, Lcom/android/internal/util/FastXmlSerializer;-><init>()V
  
+     .line 3230
      .local v7, out:Lorg/xmlpull/v1/XmlSerializer;
      const-string v10, "utf-8"
  
      invoke-interface {v7, v2, v10}, Lorg/xmlpull/v1/XmlSerializer;->setOutput(Ljava/io/OutputStream;Ljava/lang/String;)V
  
+     .line 3231
      const/4 v10, 0x0
  
      const/4 v11, 0x1
***************
*** 1006,1023 ****
--- 1096,1116 ----
  
      invoke-interface {v7, v10, v11}, Lorg/xmlpull/v1/XmlSerializer;->startDocument(Ljava/lang/String;Ljava/lang/Boolean;)V
  
+     .line 3232
      const-string v10, "http://xmlpull.org/v1/doc/features.html#indent-output"
  
      const/4 v11, 0x1
  
      invoke-interface {v7, v10, v11}, Lorg/xmlpull/v1/XmlSerializer;->setFeature(Ljava/lang/String;Z)V
  
+     .line 3233
      const/4 v10, 0x0
  
      const-string v11, "subtypes"
  
      invoke-interface {v7, v10, v11}, Lorg/xmlpull/v1/XmlSerializer;->startTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3234
      invoke-virtual {p0}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
  
      move-result-object v10
***************
*** 1040,1045 ****
--- 1133,1139 ----
  
      check-cast v5, Ljava/lang/String;
  
+     .line 3235
      .local v5, imiId:Ljava/lang/String;
      if-eqz v6, :cond_2
  
***************
*** 1049,1054 ****
--- 1143,1149 ----
  
      if-nez v10, :cond_2
  
+     .line 3236
      const-string v10, "InputManagerService"
  
      new-instance v11, Ljava/lang/StringBuilder;
***************
*** 1075,1086 ****
--- 1170,1183 ----
  
      goto :goto_1
  
+     .line 3260
      .end local v4           #i$:Ljava/util/Iterator;
      .end local v5           #imiId:Ljava/lang/String;
      .end local v7           #out:Lorg/xmlpull/v1/XmlSerializer;
      :catch_0
      move-exception v1
  
+     .line 3261
      .local v1, e:Ljava/io/IOException;
      const-string v10, "InputManagerService"
  
***************
*** 1088,1102 ****
--- 1185,1203 ----
  
      invoke-static {v10, v11, v1}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 3262
      if-eqz v2, :cond_0
  
+     .line 3263
      invoke-virtual {p1, v2}, Lcom/android/internal/os/AtomicFile;->failWrite(Ljava/io/FileOutputStream;)V
  
+     .line 3266
      .end local v1           #e:Ljava/io/IOException;
      :cond_0
      :goto_2
      return-void
  
+     .line 3225
      .end local v2           #fos:Ljava/io/FileOutputStream;
      .end local v6           #isSetMethodMap:Z
      :cond_1
***************
*** 1104,1109 ****
--- 1205,1211 ----
  
      goto :goto_0
  
+     .line 3239
      .restart local v2       #fos:Ljava/io/FileOutputStream;
      .restart local v4       #i$:Ljava/util/Iterator;
      .restart local v5       #imiId:Ljava/lang/String;
***************
*** 1117,1139 ****
--- 1219,1245 ----
  
      invoke-interface {v7, v10, v11}, Lorg/xmlpull/v1/XmlSerializer;->startTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3240
      const/4 v10, 0x0
  
      const-string v11, "id"
  
      invoke-interface {v7, v10, v11, v5}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3241
      invoke-virtual {p0, v5}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v9
  
      check-cast v9, Ljava/util/List;
  
+     .line 3242
      .local v9, subtypesList:Ljava/util/List;,"Ljava/util/List<Landroid/view/inputmethod/InputMethodSubtype;>;"
      invoke-interface {v9}, Ljava/util/List;->size()I
  
      move-result v0
  
+     .line 3243
      .local v0, N:I
      const/4 v3, 0x0
  
***************
*** 1141,1152 ****
--- 1247,1260 ----
      :goto_3
      if-ge v3, v0, :cond_4
  
+     .line 3244
      invoke-interface {v9, v3}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v8
  
      check-cast v8, Landroid/view/inputmethod/InputMethodSubtype;
  
+     .line 3245
      .local v8, subtype:Landroid/view/inputmethod/InputMethodSubtype;
      const/4 v10, 0x0
  
***************
*** 1154,1159 ****
--- 1262,1268 ----
  
      invoke-interface {v7, v10, v11}, Lorg/xmlpull/v1/XmlSerializer;->startTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3246
      const/4 v10, 0x0
  
      const-string v11, "icon"
***************
*** 1168,1173 ****
--- 1277,1283 ----
  
      invoke-interface {v7, v10, v11, v12}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3247
      const/4 v10, 0x0
  
      const-string v11, "label"
***************
*** 1182,1187 ****
--- 1292,1298 ----
  
      invoke-interface {v7, v10, v11, v12}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3248
      const/4 v10, 0x0
  
      const-string v11, "imeSubtypeLocale"
***************
*** 1192,1197 ****
--- 1303,1309 ----
  
      invoke-interface {v7, v10, v11, v12}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3249
      const/4 v10, 0x0
  
      const-string v11, "imeSubtypeMode"
***************
*** 1202,1207 ****
--- 1314,1320 ----
  
      invoke-interface {v7, v10, v11, v12}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3250
      const/4 v10, 0x0
  
      const-string v11, "imeSubtypeExtraValue"
***************
*** 1212,1217 ****
--- 1325,1331 ----
  
      invoke-interface {v7, v10, v11, v12}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3251
      const/4 v11, 0x0
  
      const-string v12, "isAuxiliary"
***************
*** 1231,1251 ****
--- 1345,1369 ----
  
      invoke-interface {v7, v11, v12, v10}, Lorg/xmlpull/v1/XmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3253
      const/4 v10, 0x0
  
      const-string v11, "subtype"
  
      invoke-interface {v7, v10, v11}, Lorg/xmlpull/v1/XmlSerializer;->endTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3243
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_3
  
+     .line 3251
      :cond_3
      const/4 v10, 0x0
  
      goto :goto_4
  
+     .line 3255
      .end local v8           #subtype:Landroid/view/inputmethod/InputMethodSubtype;
      :cond_4
      const/4 v10, 0x0
***************
*** 1256,1261 ****
--- 1374,1380 ----
  
      goto/16 :goto_1
  
+     .line 3257
      .end local v0           #N:I
      .end local v3           #i:I
      .end local v5           #imiId:Ljava/lang/String;
***************
*** 1267,1274 ****
--- 1386,1395 ----
  
      invoke-interface {v7, v10, v11}, Lorg/xmlpull/v1/XmlSerializer;->endTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 3258
      invoke-interface {v7}, Lorg/xmlpull/v1/XmlSerializer;->endDocument()V
  
+     .line 3259
      invoke-virtual {p1, v2}, Lcom/android/internal/os/AtomicFile;->finishWrite(Ljava/io/FileOutputStream;)V
      :try_end_1
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_0
***************
*** 1284,1301 ****
--- 1405,1426 ----
      .parameter "additionalSubtypes"
  
      .prologue
+     .line 3200
      iget-object v5, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mMethodMap:Ljava/util/HashMap;
  
      monitor-enter v5
  
+     .line 3201
      :try_start_0
      new-instance v3, Ljava/util/ArrayList;
  
      invoke-direct {v3}, Ljava/util/ArrayList;-><init>()V
  
+     .line 3202
      .local v3, subtypes:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/view/inputmethod/InputMethodSubtype;>;"
      array-length v0, p2
  
+     .line 3203
      .local v0, N:I
      const/4 v1, 0x0
  
***************
*** 1303,1310 ****
--- 1428,1437 ----
      :goto_0
      if-ge v1, v0, :cond_1
  
+     .line 3204
      aget-object v2, p2, v1
  
+     .line 3205
      .local v2, subtype:Landroid/view/inputmethod/InputMethodSubtype;
      invoke-virtual {v3, v2}, Ljava/util/ArrayList;->contains(Ljava/lang/Object;)Z
  
***************
*** 1312,1324 ****
--- 1439,1454 ----
  
      if-nez v4, :cond_0
  
+     .line 3206
      invoke-virtual {v3, v2}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 3203
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 3209
      .end local v2           #subtype:Landroid/view/inputmethod/InputMethodSubtype;
      :cond_1
      iget-object v4, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mSubtypesMap:Ljava/util/HashMap;
***************
*** 1329,1334 ****
--- 1459,1465 ----
  
      invoke-virtual {v4, v6, v3}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 3210
      iget-object v4, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mSubtypesMap:Ljava/util/HashMap;
  
      iget-object v6, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mAdditionalInputMethodSubtypeFile:Lcom/android/internal/os/AtomicFile;
***************
*** 1337,1346 ****
--- 1468,1480 ----
  
      invoke-static {v4, v6, v7}, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->writeAdditionalInputMethodSubtypes(Ljava/util/HashMap;Lcom/android/internal/os/AtomicFile;Ljava/util/HashMap;)V
  
+     .line 3212
      monitor-exit v5
  
+     .line 3213
      return-void
  
+     .line 3212
      .end local v0           #N:I
      .end local v1           #i:I
      .end local v3           #subtypes:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/view/inputmethod/InputMethodSubtype;>;"
***************
*** 1370,1379 ****
--- 1504,1515 ----
      .end annotation
  
      .prologue
+     .line 3216
      iget-object v1, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mMethodMap:Ljava/util/HashMap;
  
      monitor-enter v1
  
+     .line 3217
      :try_start_0
      iget-object v0, p0, Lcom/android/server/InputMethodManagerService$InputMethodFileManager;->mSubtypesMap:Ljava/util/HashMap;
  
***************
*** 1381,1386 ****
--- 1517,1523 ----
  
      return-object v0
  
+     .line 3218
      :catchall_0
      move-exception v0
  
