*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/InputMethodManagerService$3.smali	2012-03-23 19:34:55.036496376 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/InputMethodManagerService$3.smali.original	2012-03-23 19:34:46.875347557 -0500
***************
*** 40,45 ****
--- 40,46 ----
      .parameter
  
      .prologue
+     .line 2111
      iput-object p1, p0, Lcom/android/server/InputMethodManagerService$3;->this$0:Lcom/android/server/InputMethodManagerService;
  
      iput-object p2, p0, Lcom/android/server/InputMethodManagerService$3;->val$pm:Landroid/content/pm/PackageManager;
***************
*** 57,69 ****
--- 58,73 ----
      .parameter "imi2"
  
      .prologue
+     .line 2114
      if-nez p2, :cond_0
  
      const/4 v2, 0x0
  
+     .line 2121
      :goto_0
      return v2
  
+     .line 2115
      :cond_0
      if-nez p1, :cond_1
  
***************
*** 71,81 ****
--- 75,87 ----
  
      goto :goto_0
  
+     .line 2116
      :cond_1
      iget-object v2, p0, Lcom/android/server/InputMethodManagerService$3;->val$pm:Landroid/content/pm/PackageManager;
  
      if-nez v2, :cond_2
  
+     .line 2117
      invoke-virtual {p1}, Landroid/view/inputmethod/InputMethodInfo;->getId()Ljava/lang/String;
  
      move-result-object v2
***************
*** 90,95 ****
--- 96,102 ----
  
      goto :goto_0
  
+     .line 2119
      :cond_2
      new-instance v2, Ljava/lang/StringBuilder;
  
***************
*** 123,128 ****
--- 130,136 ----
  
      move-result-object v0
  
+     .line 2120
      .local v0, imiId1:Ljava/lang/CharSequence;
      new-instance v2, Ljava/lang/StringBuilder;
  
***************
*** 156,161 ****
--- 164,170 ----
  
      move-result-object v1
  
+     .line 2121
      .local v1, imiId2:Ljava/lang/CharSequence;
      invoke-virtual {v0}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
***************
*** 178,183 ****
--- 187,193 ----
      .parameter "x1"
  
      .prologue
+     .line 2111
      check-cast p1, Landroid/view/inputmethod/InputMethodInfo;
  
      .end local p1
