*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/LightsService$Light.smali	2012-03-23 19:34:58.754564175 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/LightsService$Light.smali.original	2012-03-23 19:34:46.942348780 -0500
***************
*** 37,48 ****
--- 37,51 ----
      .parameter "id"
  
      .prologue
+     .line 62
      iput-object p1, p0, Lcom/android/server/LightsService$Light;->this$0:Lcom/android/server/LightsService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 63
      iput p2, p0, Lcom/android/server/LightsService$Light;->mId:I
  
+     .line 64
      return-void
  .end method
  
***************
*** 53,58 ****
--- 56,62 ----
      .parameter "x2"
  
      .prologue
+     .line 60
      invoke-direct {p0, p1, p2}, Lcom/android/server/LightsService$Light;-><init>(Lcom/android/server/LightsService;I)V
  
      return-void
***************
*** 63,68 ****
--- 67,73 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      invoke-direct {p0}, Lcom/android/server/LightsService$Light;->stopFlashing()V
  
      return-void
***************
*** 77,82 ****
--- 82,88 ----
      .parameter "brightnessMode"
  
      .prologue
+     .line 117
      iget v0, p0, Lcom/android/server/LightsService$Light;->mColor:I
  
      if-ne p1, v0, :cond_0
***************
*** 93,107 ****
--- 99,118 ----
  
      if-eq p4, v0, :cond_1
  
+     .line 120
      :cond_0
      iput p1, p0, Lcom/android/server/LightsService$Light;->mColor:I
  
+     .line 121
      iput p2, p0, Lcom/android/server/LightsService$Light;->mMode:I
  
+     .line 122
      iput p3, p0, Lcom/android/server/LightsService$Light;->mOnMS:I
  
+     .line 123
      iput p4, p0, Lcom/android/server/LightsService$Light;->mOffMS:I
  
+     .line 124
      iget-object v0, p0, Lcom/android/server/LightsService$Light;->this$0:Lcom/android/server/LightsService;
  
      #getter for: Lcom/android/server/LightsService;->mNativePointer:I
***************
*** 124,129 ****
--- 135,141 ----
      #calls: Lcom/android/server/LightsService;->setLight_native(IIIIIII)V
      invoke-static/range {v0 .. v6}, Lcom/android/server/LightsService;->access$200(IIIIIII)V
  
+     .line 126
      :cond_1
      return-void
  .end method
***************
*** 132,139 ****
--- 144,153 ----
      .locals 6
  
      .prologue
+     .line 111
      monitor-enter p0
  
+     .line 112
      :try_start_0
      iget v1, p0, Lcom/android/server/LightsService$Light;->mColor:I
  
***************
*** 149,158 ****
--- 163,175 ----
  
      invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
+     .line 113
      monitor-exit p0
  
+     .line 114
      return-void
  
+     .line 113
      :catchall_0
      move-exception v0
  
***************
*** 169,180 ****
--- 186,199 ----
      .locals 2
  
      .prologue
+     .line 92
      const v0, 0xffffff
  
      const/4 v1, 0x7
  
      invoke-virtual {p0, v0, v1}, Lcom/android/server/LightsService$Light;->pulse(II)V
  
+     .line 93
      return-void
  .end method
  
***************
*** 184,191 ****
--- 203,212 ----
      .parameter "onMS"
  
      .prologue
+     .line 96
      monitor-enter p0
  
+     .line 97
      :try_start_0
      iget v0, p0, Lcom/android/server/LightsService$Light;->mColor:I
  
***************
*** 195,200 ****
--- 216,222 ----
  
      if-nez v0, :cond_0
  
+     .line 98
      const/4 v2, 0x2
  
      const/16 v4, 0x3e8
***************
*** 209,214 ****
--- 231,237 ----
  
      invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
+     .line 99
      iget-object v0, p0, Lcom/android/server/LightsService$Light;->this$0:Lcom/android/server/LightsService;
  
      #getter for: Lcom/android/server/LightsService;->mH:Landroid/os/Handler;
***************
*** 233,243 ****
--- 256,269 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 101
      :cond_0
      monitor-exit p0
  
+     .line 102
      return-void
  
+     .line 101
      :catchall_0
      move-exception v0
  
***************
*** 253,262 ****
--- 279,290 ----
      .parameter "brightness"
  
      .prologue
+     .line 67
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Lcom/android/server/LightsService$Light;->setBrightness(II)V
  
+     .line 68
      return-void
  .end method
  
***************
*** 266,275 ****
--- 294,306 ----
      .parameter "brightnessMode"
  
      .prologue
+     .line 71
      monitor-enter p0
  
+     .line 72
      and-int/lit16 v1, p1, 0xff
  
+     .line 73
      .local v1, color:I
      const/high16 v0, -0x100
  
***************
*** 283,288 ****
--- 314,320 ----
  
      or-int/2addr v1, v0
  
+     .line 74
      const/4 v2, 0x0
  
      const/4 v3, 0x0
***************
*** 296,305 ****
--- 328,340 ----
      :try_start_0
      invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
+     .line 75
      monitor-exit p0
  
+     .line 76
      return-void
  
+     .line 75
      :catchall_0
      move-exception v0
  
***************
*** 315,322 ****
--- 350,359 ----
      .parameter "color"
  
      .prologue
+     .line 79
      monitor-enter p0
  
+     .line 80
      const/4 v2, 0x0
  
      const/4 v3, 0x0
***************
*** 332,341 ****
--- 369,381 ----
      :try_start_0
      invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
+     .line 81
      monitor-exit p0
  
+     .line 82
      return-void
  
+     .line 81
      :catchall_0
      move-exception v0
  
***************
*** 354,361 ****
--- 394,403 ----
      .parameter "offMS"
  
      .prologue
+     .line 85
      monitor-enter p0
  
+     .line 86
      const/4 v5, 0x0
  
      move-object v0, p0
***************
*** 371,380 ****
--- 413,425 ----
      :try_start_0
      invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
+     .line 87
      monitor-exit p0
  
+     .line 88
      return-void
  
+     .line 87
      :catchall_0
      move-exception v0
  
***************
*** 389,396 ****
--- 434,443 ----
      .locals 6
  
      .prologue
+     .line 105
      monitor-enter p0
  
+     .line 106
      const/4 v1, 0x0
  
      const/4 v2, 0x0
***************
*** 406,415 ****
--- 453,465 ----
      :try_start_0
      invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
+     .line 107
      monitor-exit p0
  
+     .line 108
      return-void
  
+     .line 107
      :catchall_0
      move-exception v0
  
