*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/LocationManagerService$ProximityAlert.smali	2012-03-23 19:34:58.419558067 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/LocationManagerService$ProximityAlert.smali.original	2012-03-23 19:34:46.938348706 -0500
***************
*** 44,65 ****
--- 44,73 ----
      .parameter "intent"
  
      .prologue
+     .line 1372
      iput-object p1, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->this$0:Lcom/android/server/LocationManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1373
      iput p2, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mUid:I
  
+     .line 1374
      iput-wide p3, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mLatitude:D
  
+     .line 1375
      iput-wide p5, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mLongitude:D
  
+     .line 1376
      iput p7, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mRadius:F
  
+     .line 1377
      iput-wide p8, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mExpiration:J
  
+     .line 1378
      iput-object p10, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mIntent:Landroid/app/PendingIntent;
  
+     .line 1380
      new-instance v0, Landroid/location/Location;
  
      const-string v1, ""
***************
*** 68,81 ****
--- 76,92 ----
  
      iput-object v0, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mLocation:Landroid/location/Location;
  
+     .line 1381
      iget-object v0, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mLocation:Landroid/location/Location;
  
      invoke-virtual {v0, p3, p4}, Landroid/location/Location;->setLatitude(D)V
  
+     .line 1382
      iget-object v0, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mLocation:Landroid/location/Location;
  
      invoke-virtual {v0, p5, p6}, Landroid/location/Location;->setLongitude(D)V
  
+     .line 1383
      return-void
  .end method
  
***************
*** 87,92 ****
--- 98,104 ----
      .parameter "prefix"
  
      .prologue
+     .line 1410
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 105,110 ****
--- 117,123 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 1411
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 143,148 ****
--- 156,162 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 1412
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 181,186 ****
--- 195,201 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 1413
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 207,212 ****
--- 222,228 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 1414
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 227,232 ****
--- 243,249 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 1415
      iget-object v0, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mLocation:Landroid/location/Location;
  
      new-instance v1, Landroid/util/PrintWriterPrinter;
***************
*** 253,258 ****
--- 270,276 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/location/Location;->dump(Landroid/util/Printer;Ljava/lang/String;)V
  
+     .line 1416
      return-void
  .end method
  
***************
*** 260,265 ****
--- 278,284 ----
      .locals 2
  
      .prologue
+     .line 1386
      iget-wide v0, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mExpiration:J
  
      return-wide v0
***************
*** 269,274 ****
--- 288,294 ----
      .locals 1
  
      .prologue
+     .line 1390
      iget-object v0, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mIntent:Landroid/app/PendingIntent;
  
      return-object v0
***************
*** 281,297 ****
--- 301,321 ----
      .parameter "accuracy"
  
      .prologue
+     .line 1394
      new-instance v0, Landroid/location/Location;
  
      const-string v3, ""
  
      invoke-direct {v0, v3}, Landroid/location/Location;-><init>(Ljava/lang/String;)V
  
+     .line 1395
      .local v0, loc:Landroid/location/Location;
      invoke-virtual {v0, p1, p2}, Landroid/location/Location;->setLatitude(D)V
  
+     .line 1396
      invoke-virtual {v0, p3, p4}, Landroid/location/Location;->setLongitude(D)V
  
+     .line 1398
      iget-object v3, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mLocation:Landroid/location/Location;
  
      invoke-virtual {v0, v3}, Landroid/location/Location;->distanceTo(Landroid/location/Location;)F
***************
*** 300,305 ****
--- 324,330 ----
  
      float-to-double v1, v3
  
+     .line 1399
      .local v1, radius:D
      iget v3, p0, Lcom/android/server/LocationManagerService$ProximityAlert;->mRadius:F
  
***************
*** 328,333 ****
--- 353,359 ----
      .locals 2
  
      .prologue
+     .line 1404
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
