*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$ObbState.smali	2012-03-23 19:34:53.142461838 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$ObbState.smali.original	2012-03-23 19:34:46.856347211 -0500
***************
*** 44,61 ****
--- 44,67 ----
      .end annotation
  
      .prologue
+     .line 211
      iput-object p1, p0, Lcom/android/server/MountService$ObbState;->this$0:Lcom/android/server/MountService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 212
      iput-object p2, p0, Lcom/android/server/MountService$ObbState;->filename:Ljava/lang/String;
  
+     .line 213
      iput p3, p0, Lcom/android/server/MountService$ObbState;->callerUid:I
  
+     .line 214
      iput-object p4, p0, Lcom/android/server/MountService$ObbState;->token:Landroid/os/storage/IObbActionListener;
  
+     .line 215
      iput p5, p0, Lcom/android/server/MountService$ObbState;->nonce:I
  
+     .line 216
      return-void
  .end method
  
***************
*** 67,78 ****
--- 73,86 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 236
      new-instance v0, Lcom/android/server/MountService$UnmountObbAction;
  
      iget-object v1, p0, Lcom/android/server/MountService$ObbState;->this$0:Lcom/android/server/MountService;
  
      invoke-direct {v0, v1, p0, v3}, Lcom/android/server/MountService$UnmountObbAction;-><init>(Lcom/android/server/MountService;Lcom/android/server/MountService$ObbState;Z)V
  
+     .line 237
      .local v0, action:Lcom/android/server/MountService$ObbAction;
      iget-object v1, p0, Lcom/android/server/MountService$ObbState;->this$0:Lcom/android/server/MountService;
  
***************
*** 94,99 ****
--- 102,108 ----
  
      invoke-virtual {v1, v2}, Lcom/android/server/MountService$ObbActionHandler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 238
      return-void
  .end method
  
***************
*** 101,106 ****
--- 110,116 ----
      .locals 1
  
      .prologue
+     .line 231
      iget-object v0, p0, Lcom/android/server/MountService$ObbState;->token:Landroid/os/storage/IObbActionListener;
  
      invoke-interface {v0}, Landroid/os/storage/IObbActionListener;->asBinder()Landroid/os/IBinder;
***************
*** 119,124 ****
--- 129,135 ----
      .end annotation
  
      .prologue
+     .line 241
      invoke-virtual {p0}, Lcom/android/server/MountService$ObbState;->getBinder()Landroid/os/IBinder;
  
      move-result-object v0
***************
*** 127,132 ****
--- 138,144 ----
  
      invoke-interface {v0, p0, v1}, Landroid/os/IBinder;->linkToDeath(Landroid/os/IBinder$DeathRecipient;I)V
  
+     .line 242
      return-void
  .end method
  
***************
*** 134,158 ****
--- 146,175 ----
      .locals 2
  
      .prologue
+     .line 250
      new-instance v0, Ljava/lang/StringBuilder;
  
      const-string v1, "ObbState{"
  
      invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
  
+     .line 251
      .local v0, sb:Ljava/lang/StringBuilder;
      const-string v1, "filename="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 252
      iget-object v1, p0, Lcom/android/server/MountService$ObbState;->filename:Ljava/lang/String;
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 253
      const-string v1, ",token="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 254
      iget-object v1, p0, Lcom/android/server/MountService$ObbState;->token:Landroid/os/storage/IObbActionListener;
  
      invoke-virtual {v1}, Ljava/lang/Object;->toString()Ljava/lang/String;
***************
*** 161,178 ****
--- 178,199 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 255
      const-string v1, ",callerUid="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 256
      iget v1, p0, Lcom/android/server/MountService$ObbState;->callerUid:I
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 257
      const/16 v1, 0x7d
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 258
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
***************
*** 184,189 ****
--- 205,211 ----
      .locals 2
  
      .prologue
+     .line 245
      invoke-virtual {p0}, Lcom/android/server/MountService$ObbState;->getBinder()Landroid/os/IBinder;
  
      move-result-object v0
***************
*** 192,196 ****
--- 214,219 ----
  
      invoke-interface {v0, p0, v1}, Landroid/os/IBinder;->unlinkToDeath(Landroid/os/IBinder$DeathRecipient;I)Z
  
+     .line 246
      return-void
  .end method
