*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$DefaultContainerConnection.smali	2012-03-23 19:35:11.366794159 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$DefaultContainerConnection.smali.original	2012-03-23 19:34:47.373356640 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 280
      iput-object p1, p0, Lcom/android/server/MountService$DefaultContainerConnection;->this$0:Lcom/android/server/MountService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,51 ****
--- 43,54 ----
      .parameter "service"
  
      .prologue
+     .line 284
      invoke-static {p2}, Lcom/android/internal/app/IMediaContainerService$Stub;->asInterface(Landroid/os/IBinder;)Lcom/android/internal/app/IMediaContainerService;
  
      move-result-object v0
  
+     .line 285
      .local v0, imcs:Lcom/android/internal/app/IMediaContainerService;
      iget-object v1, p0, Lcom/android/server/MountService$DefaultContainerConnection;->this$0:Lcom/android/server/MountService;
  
***************
*** 69,74 ****
--- 72,78 ----
  
      invoke-virtual {v1, v2}, Lcom/android/server/MountService$ObbActionHandler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 286
      return-void
  .end method
  
***************
*** 77,81 ****
--- 81,86 ----
      .parameter "name"
  
      .prologue
+     .line 291
      return-void
  .end method
