*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$5.smali	2012-03-23 19:34:55.663507810 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$5.smali.original	2012-03-23 19:34:46.885347740 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1876
      iput-object p1, p0, Lcom/android/server/MountService$5;->this$0:Lcom/android/server/MountService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .locals 3
  
      .prologue
+     .line 1878
      iget-object v0, p0, Lcom/android/server/MountService$5;->this$0:Lcom/android/server/MountService;
  
      #getter for: Lcom/android/server/MountService;->mConnector:Lcom/android/server/NativeDaemonConnector;
***************
*** 59,63 ****
--- 61,66 ----
  
      invoke-virtual {v0, v1}, Lcom/android/server/NativeDaemonConnector;->doCommand(Ljava/lang/String;)Ljava/util/ArrayList;
  
+     .line 1879
      return-void
  .end method
