*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$UmsEnableCallBack.smali	2012-03-23 19:35:00.084588428 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$UmsEnableCallBack.smali.original	2012-03-23 19:34:46.952348961 -0500
***************
*** 29,42 ****
--- 29,46 ----
      .parameter "force"
  
      .prologue
+     .line 326
      iput-object p1, p0, Lcom/android/server/MountService$UmsEnableCallBack;->this$0:Lcom/android/server/MountService;
  
+     .line 327
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, p2, p4, v0}, Lcom/android/server/MountService$UnmountCallBack;-><init>(Lcom/android/server/MountService;Ljava/lang/String;ZZ)V
  
+     .line 328
      iput-object p3, p0, Lcom/android/server/MountService$UmsEnableCallBack;->method:Ljava/lang/String;
  
+     .line 329
      return-void
  .end method
  
***************
*** 46,53 ****
--- 50,59 ----
      .locals 4
  
      .prologue
+     .line 333
      invoke-super {p0}, Lcom/android/server/MountService$UnmountCallBack;->handleFinished()V
  
+     .line 334
      iget-object v0, p0, Lcom/android/server/MountService$UmsEnableCallBack;->this$0:Lcom/android/server/MountService;
  
      iget-object v1, p0, Lcom/android/server/MountService$UmsEnableCallBack;->path:Ljava/lang/String;
***************
*** 59,63 ****
--- 65,70 ----
      #calls: Lcom/android/server/MountService;->doShareUnshareVolume(Ljava/lang/String;Ljava/lang/String;Z)V
      invoke-static {v0, v1, v2, v3}, Lcom/android/server/MountService;->access$200(Lcom/android/server/MountService;Ljava/lang/String;Ljava/lang/String;Z)V
  
+     .line 335
      return-void
  .end method
