*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$MountObbAction.smali	2012-03-23 19:34:59.405576047 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$MountObbAction.smali.original	2012-03-23 19:34:46.946348853 -0500
***************
*** 28,39 ****
--- 28,43 ----
      .parameter "key"
  
      .prologue
+     .line 2284
      iput-object p1, p0, Lcom/android/server/MountService$MountObbAction;->this$0:Lcom/android/server/MountService;
  
+     .line 2285
      invoke-direct {p0, p1, p2}, Lcom/android/server/MountService$ObbAction;-><init>(Lcom/android/server/MountService;Lcom/android/server/MountService$ObbState;)V
  
+     .line 2286
      iput-object p3, p0, Lcom/android/server/MountService$MountObbAction;->mKey:Ljava/lang/String;
  
+     .line 2287
      return-void
  .end method
  
***************
*** 43,52 ****
--- 47,58 ----
      .locals 1
  
      .prologue
+     .line 2372
      const/16 v0, 0x14
  
      invoke-virtual {p0, v0}, Lcom/android/server/MountService$MountObbAction;->sendNewStatusOrIgnore(I)V
  
+     .line 2373
      return-void
  .end method
  
***************
*** 60,79 ****
--- 66,89 ----
      .end annotation
  
      .prologue
+     .line 2291
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->this$0:Lcom/android/server/MountService;
  
      #calls: Lcom/android/server/MountService;->waitForReady()V
      invoke-static {v11}, Lcom/android/server/MountService;->access$2500(Lcom/android/server/MountService;)V
  
+     .line 2292
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->this$0:Lcom/android/server/MountService;
  
      #calls: Lcom/android/server/MountService;->warnOnNotMounted()V
      invoke-static {v11}, Lcom/android/server/MountService;->access$2600(Lcom/android/server/MountService;)V
  
+     .line 2294
      invoke-virtual {p0}, Lcom/android/server/MountService$MountObbAction;->getObbInfo()Landroid/content/res/ObbInfo;
  
      move-result-object v9
  
+     .line 2296
      .local v9, obbInfo:Landroid/content/res/ObbInfo;
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->this$0:Lcom/android/server/MountService;
  
***************
*** 90,95 ****
--- 100,106 ----
  
      if-nez v11, :cond_0
  
+     .line 2297
      const-string v11, "MountService"
  
      new-instance v12, Ljava/lang/StringBuilder;
***************
*** 126,138 ****
--- 137,152 ----
  
      invoke-static {v11, v12}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 2299
      const/16 v11, 0x19
  
      invoke-virtual {p0, v11}, Lcom/android/server/MountService$MountObbAction;->sendNewStatusOrIgnore(I)V
  
+     .line 2368
      :goto_0
      return-void
  
+     .line 2304
      :cond_0
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->this$0:Lcom/android/server/MountService;
  
***************
*** 143,148 ****
--- 157,163 ----
  
      monitor-enter v12
  
+     .line 2305
      :try_start_0
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->this$0:Lcom/android/server/MountService;
  
***************
*** 157,169 ****
--- 172,187 ----
  
      move-result v6
  
+     .line 2306
      .local v6, isMounted:Z
      monitor-exit v12
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 2307
      if-eqz v6, :cond_1
  
+     .line 2308
      const-string v11, "MountService"
  
      new-instance v12, Ljava/lang/StringBuilder;
***************
*** 188,199 ****
--- 206,219 ----
  
      invoke-static {v11, v12}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 2309
      const/16 v11, 0x18
  
      invoke-virtual {p0, v11}, Lcom/android/server/MountService$MountObbAction;->sendNewStatusOrIgnore(I)V
  
      goto :goto_0
  
+     .line 2306
      .end local v6           #isMounted:Z
      :catchall_0
      move-exception v11
***************
*** 205,210 ****
--- 225,231 ----
  
      throw v11
  
+     .line 2317
      .restart local v6       #isMounted:Z
      :cond_1
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->mObbState:Lcom/android/server/MountService$ObbState;
***************
*** 213,228 ****
--- 234,253 ----
  
      iput-object v12, v11, Lcom/android/server/MountService$ObbState;->filename:Ljava/lang/String;
  
+     .line 2320
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->mKey:Ljava/lang/String;
  
      if-nez v11, :cond_3
  
+     .line 2321
      const-string v5, "none"
  
+     .line 2342
      .local v5, hashedKey:Ljava/lang/String;
      :goto_1
      const/4 v10, 0x0
  
+     .line 2343
      .local v10, rc:I
      const-string v11, "obb mount %s %s %d"
  
***************
*** 258,263 ****
--- 283,289 ----
  
      move-result-object v1
  
+     .line 2346
      .local v1, cmd:Ljava/lang/String;
      :try_start_2
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->this$0:Lcom/android/server/MountService;
***************
*** 271,280 ****
--- 297,308 ----
      :try_end_2
      .catch Lcom/android/server/NativeDaemonConnectorException; {:try_start_2 .. :try_end_2} :catch_2
  
+     .line 2354
      :cond_2
      :goto_2
      if-nez v10, :cond_4
  
+     .line 2358
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->this$0:Lcom/android/server/MountService;
  
      #getter for: Lcom/android/server/MountService;->mObbMounts:Ljava/util/Map;
***************
*** 284,289 ****
--- 312,318 ----
  
      monitor-enter v12
  
+     .line 2359
      :try_start_3
      iget-object v11, p0, Lcom/android/server/MountService$MountObbAction;->this$0:Lcom/android/server/MountService;
  
***************
*** 292,307 ****
--- 321,339 ----
      #calls: Lcom/android/server/MountService;->addObbStateLocked(Lcom/android/server/MountService$ObbState;)V
      invoke-static {v11, v13}, Lcom/android/server/MountService;->access$2800(Lcom/android/server/MountService;Lcom/android/server/MountService$ObbState;)V
  
+     .line 2360
      monitor-exit v12
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_1
  
+     .line 2362
      const/4 v11, 0x1
  
      invoke-virtual {p0, v11}, Lcom/android/server/MountService$MountObbAction;->sendNewStatusOrIgnore(I)V
  
      goto/16 :goto_0
  
+     .line 2324
      .end local v1           #cmd:Ljava/lang/String;
      .end local v5           #hashedKey:Ljava/lang/String;
      .end local v10           #rc:I
***************
*** 313,318 ****
--- 345,351 ----
  
      move-result-object v4
  
+     .line 2326
      .local v4, factory:Ljavax/crypto/SecretKeyFactory;
      new-instance v8, Ljavax/crypto/spec/PBEKeySpec;
  
***************
*** 330,340 ****
--- 363,375 ----
  
      invoke-direct {v8, v11, v12, v13, v14}, Ljavax/crypto/spec/PBEKeySpec;-><init>([C[BII)V
  
+     .line 2328
      .local v8, ks:Ljava/security/spec/KeySpec;
      invoke-virtual {v4, v8}, Ljavax/crypto/SecretKeyFactory;->generateSecret(Ljava/security/spec/KeySpec;)Ljavax/crypto/SecretKey;
  
      move-result-object v7
  
+     .line 2329
      .local v7, key:Ljavax/crypto/SecretKey;
      new-instance v0, Ljava/math/BigInteger;
  
***************
*** 344,349 ****
--- 379,385 ----
  
      invoke-direct {v0, v11}, Ljava/math/BigInteger;-><init>([B)V
  
+     .line 2330
      .local v0, bi:Ljava/math/BigInteger;
      const/16 v11, 0x10
  
***************
*** 357,362 ****
--- 393,399 ----
      .restart local v5       #hashedKey:Ljava/lang/String;
      goto :goto_1
  
+     .line 2331
      .end local v0           #bi:Ljava/math/BigInteger;
      .end local v4           #factory:Ljavax/crypto/SecretKeyFactory;
      .end local v5           #hashedKey:Ljava/lang/String;
***************
*** 365,370 ****
--- 402,408 ----
      :catch_0
      move-exception v3
  
+     .line 2332
      .local v3, e:Ljava/security/NoSuchAlgorithmException;
      const-string v11, "MountService"
  
***************
*** 372,387 ****
--- 410,428 ----
  
      invoke-static {v11, v12, v3}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 2333
      const/16 v11, 0x14
  
      invoke-virtual {p0, v11}, Lcom/android/server/MountService$MountObbAction;->sendNewStatusOrIgnore(I)V
  
      goto/16 :goto_0
  
+     .line 2335
      .end local v3           #e:Ljava/security/NoSuchAlgorithmException;
      :catch_1
      move-exception v3
  
+     .line 2336
      .local v3, e:Ljava/security/spec/InvalidKeySpecException;
      const-string v11, "MountService"
  
***************
*** 389,400 ****
--- 430,443 ----
  
      invoke-static {v11, v12, v3}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 2337
      const/16 v11, 0x14
  
      invoke-virtual {p0, v11}, Lcom/android/server/MountService$MountObbAction;->sendNewStatusOrIgnore(I)V
  
      goto/16 :goto_0
  
+     .line 2347
      .end local v3           #e:Ljava/security/spec/InvalidKeySpecException;
      .restart local v1       #cmd:Ljava/lang/String;
      .restart local v5       #hashedKey:Ljava/lang/String;
***************
*** 402,421 ****
--- 445,468 ----
      :catch_2
      move-exception v3
  
+     .line 2348
      .local v3, e:Lcom/android/server/NativeDaemonConnectorException;
      invoke-virtual {v3}, Lcom/android/server/NativeDaemonConnectorException;->getCode()I
  
      move-result v2
  
+     .line 2349
      .local v2, code:I
      const/16 v11, 0x195
  
      if-eq v2, v11, :cond_2
  
+     .line 2350
      const/4 v10, -0x1
  
      goto :goto_2
  
+     .line 2360
      .end local v2           #code:I
      .end local v3           #e:Lcom/android/server/NativeDaemonConnectorException;
      :catchall_1
***************
*** 428,433 ****
--- 475,481 ----
  
      throw v11
  
+     .line 2364
      :cond_4
      const-string v11, "MountService"
  
***************
*** 451,456 ****
--- 499,505 ----
  
      invoke-static {v11, v12}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 2366
      const/16 v11, 0x15
  
      invoke-virtual {p0, v11}, Lcom/android/server/MountService$MountObbAction;->sendNewStatusOrIgnore(I)V
***************
*** 462,500 ****
--- 511,557 ----
      .locals 2
  
      .prologue
+     .line 2377
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 2378
      .local v0, sb:Ljava/lang/StringBuilder;
      const-string v1, "MountObbAction{"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 2379
      const-string v1, "filename="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 2380
      iget-object v1, p0, Lcom/android/server/MountService$MountObbAction;->mObbState:Lcom/android/server/MountService$ObbState;
  
      iget-object v1, v1, Lcom/android/server/MountService$ObbState;->filename:Ljava/lang/String;
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 2381
      const-string v1, ",callerUid="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 2382
      iget-object v1, p0, Lcom/android/server/MountService$MountObbAction;->mObbState:Lcom/android/server/MountService$ObbState;
  
      iget v1, v1, Lcom/android/server/MountService$ObbState;->callerUid:I
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 2383
      const-string v1, ",token="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 2384
      iget-object v1, p0, Lcom/android/server/MountService$MountObbAction;->mObbState:Lcom/android/server/MountService$ObbState;
  
      iget-object v1, v1, Lcom/android/server/MountService$ObbState;->token:Landroid/os/storage/IObbActionListener;
***************
*** 512,521 ****
--- 569,580 ----
      :goto_0
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 2385
      const-string v1, ",binder="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 2386
      iget-object v1, p0, Lcom/android/server/MountService$MountObbAction;->mObbState:Lcom/android/server/MountService$ObbState;
  
      iget-object v1, v1, Lcom/android/server/MountService$ObbState;->token:Landroid/os/storage/IObbActionListener;
***************
*** 535,555 ****
--- 594,618 ----
      :goto_1
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 2387
      const/16 v1, 0x7d
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 2388
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
  
      return-object v1
  
+     .line 2384
      :cond_0
      const-string v1, "NULL"
  
      goto :goto_0
  
+     .line 2386
      :cond_1
      const-string v1, "null"
  
