*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$4.smali	2012-03-23 19:34:59.973586405 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/MountService$4.smali.original	2012-03-23 19:34:46.951348943 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1033
      iput-object p1, p0, Lcom/android/server/MountService$4;->this$0:Lcom/android/server/MountService;
  
      iput-object p2, p0, Lcom/android/server/MountService$4;->val$volumes:Ljava/util/ArrayList;
***************
*** 42,47 ****
--- 43,49 ----
      .locals 9
  
      .prologue
+     .line 1038
      :try_start_0
      const-string v4, "MountService"
  
***************
*** 49,54 ****
--- 51,57 ----
  
      invoke-static {v4, v5}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1039
      iget-object v4, p0, Lcom/android/server/MountService$4;->val$volumes:Ljava/util/ArrayList;
  
      invoke-virtual {v4}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
***************
*** 70,75 ****
--- 73,79 ----
  
      check-cast v2, Ljava/lang/String;
  
+     .line 1040
      .local v2, path:Ljava/lang/String;
      iget-object v4, p0, Lcom/android/server/MountService$4;->this$0:Lcom/android/server/MountService;
  
***************
*** 85,90 ****
--- 89,95 ----
  
      if-eqz v4, :cond_0
  
+     .line 1041
      iget-object v4, p0, Lcom/android/server/MountService$4;->this$0:Lcom/android/server/MountService;
  
      const-string v5, "ums"
***************
*** 94,99 ****
--- 99,105 ----
      #calls: Lcom/android/server/MountService;->doShareUnshareVolume(Ljava/lang/String;Ljava/lang/String;Z)V
      invoke-static {v4, v2, v5, v6}, Lcom/android/server/MountService;->access$200(Lcom/android/server/MountService;Ljava/lang/String;Ljava/lang/String;Z)V
  
+     .line 1042
      iget-object v4, p0, Lcom/android/server/MountService$4;->this$0:Lcom/android/server/MountService;
  
      #calls: Lcom/android/server/MountService;->doMountVolume(Ljava/lang/String;)I
***************
*** 104,109 ****
--- 110,116 ----
      .local v3, rc:I
      if-eqz v3, :cond_0
  
+     .line 1043
      const-string v4, "MountService"
  
      const-string v5, "Failed to remount {%s} on UMS enabled-disconnect (%d)"
***************
*** 134,145 ****
--- 141,154 ----
  
      goto :goto_0
  
+     .line 1049
      .end local v1           #i$:Ljava/util/Iterator;
      .end local v2           #path:Ljava/lang/String;
      .end local v3           #rc:I
      :catch_0
      move-exception v0
  
+     .line 1050
      .local v0, ex:Ljava/lang/Exception;
      const-string v4, "MountService"
  
***************
*** 147,152 ****
--- 156,162 ----
  
      invoke-static {v4, v5, v0}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 1052
      .end local v0           #ex:Ljava/lang/Exception;
      :cond_1
      return-void
