*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/NotificationPlayer$Command.smali	2012-03-23 19:35:08.409740237 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/NotificationPlayer$Command.smali.original	2012-03-23 19:34:47.310355490 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .locals 0
  
      .prologue
+     .line 48
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 43,48 ****
--- 44,50 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-direct {p0}, Lcom/android/server/NotificationPlayer$Command;-><init>()V
  
      return-void
***************
*** 54,59 ****
--- 56,62 ----
      .locals 2
  
      .prologue
+     .line 57
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
