*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/PowerManagerService$WakeLock.smali	2012-03-23 19:35:09.286756229 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/PowerManagerService$WakeLock.smali.original	2012-03-23 19:34:47.320355672 -0500
***************
*** 54,71 ****
--- 54,78 ----
  
      const/4 v3, 0x1
  
+     .line 667
      iput-object p1, p0, Lcom/android/server/PowerManagerService$WakeLock;->this$0:Lcom/android/server/PowerManagerService;
  
+     .line 668
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 701
      iput-boolean v3, p0, Lcom/android/server/PowerManagerService$WakeLock;->activated:Z
  
+     .line 669
      iput p2, p0, Lcom/android/server/PowerManagerService$WakeLock;->flags:I
  
+     .line 670
      iput-object p3, p0, Lcom/android/server/PowerManagerService$WakeLock;->binder:Landroid/os/IBinder;
  
+     .line 671
      iput-object p4, p0, Lcom/android/server/PowerManagerService$WakeLock;->tag:Ljava/lang/String;
  
+     .line 672
      #getter for: Lcom/android/server/PowerManagerService;->MY_UID:I
      invoke-static {p1}, Lcom/android/server/PowerManagerService;->access$200(Lcom/android/server/PowerManagerService;)I
  
***************
*** 78,85 ****
--- 85,94 ----
      :goto_0
      iput v1, p0, Lcom/android/server/PowerManagerService$WakeLock;->uid:I
  
+     .line 673
      iput p6, p0, Lcom/android/server/PowerManagerService$WakeLock;->pid:I
  
+     .line 674
      #getter for: Lcom/android/server/PowerManagerService;->MY_UID:I
      invoke-static {p1}, Lcom/android/server/PowerManagerService;->access$200(Lcom/android/server/PowerManagerService;)I
  
***************
*** 107,112 ****
--- 116,122 ----
  
      if-nez v1, :cond_3
  
+     .line 677
      :cond_0
      and-int/lit8 v1, p2, 0x3f
  
***************
*** 117,122 ****
--- 127,133 ----
      :goto_1
      iput v1, p0, Lcom/android/server/PowerManagerService$WakeLock;->monitorType:I
  
+     .line 684
      :goto_2
      const/4 v1, 0x0
  
***************
*** 125,143 ****
--- 136,158 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 688
      :goto_3
      return-void
  
      :cond_1
      move v1, p5
  
+     .line 672
      goto :goto_0
  
      :cond_2
      move v1, v3
  
+     .line 677
      goto :goto_1
  
+     .line 681
      :cond_3
      const/4 v1, -0x1
  
***************
*** 145,153 ****
--- 160,170 ----
  
      goto :goto_2
  
+     .line 685
      :catch_0
      move-exception v0
  
+     .line 686
      .local v0, e:Landroid/os/RemoteException;
      invoke-virtual {p0}, Lcom/android/server/PowerManagerService$WakeLock;->binderDied()V
  
***************
*** 160,165 ****
--- 177,183 ----
      .locals 5
  
      .prologue
+     .line 690
      iget-object v0, p0, Lcom/android/server/PowerManagerService$WakeLock;->this$0:Lcom/android/server/PowerManagerService;
  
      #getter for: Lcom/android/server/PowerManagerService;->mLocks:Lcom/android/server/PowerManagerService$LockList;
***************
*** 169,174 ****
--- 187,193 ----
  
      monitor-enter v1
  
+     .line 691
      :try_start_0
      iget-object v0, p0, Lcom/android/server/PowerManagerService$WakeLock;->this$0:Lcom/android/server/PowerManagerService;
  
***************
*** 181,190 ****
--- 200,212 ----
      #calls: Lcom/android/server/PowerManagerService;->releaseWakeLockLocked(Landroid/os/IBinder;IZ)V
      invoke-static {v0, v2, v3, v4}, Lcom/android/server/PowerManagerService;->access$400(Lcom/android/server/PowerManagerService;Landroid/os/IBinder;IZ)V
  
+     .line 692
      monitor-exit v1
  
+     .line 693
      return-void
  
+     .line 692
      :catchall_0
      move-exception v0
  
