*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/PowerManagerService$12.smali	2012-03-23 19:35:08.728746054 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/PowerManagerService$12.smali.original	2012-03-23 19:34:47.314355564 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .parameter
  
      .prologue
+     .line 2620
      iput-object p1, p0, Lcom/android/server/PowerManagerService$12;->this$0:Lcom/android/server/PowerManagerService;
  
      iput-object p3, p0, Lcom/android/server/PowerManagerService$12;->val$message:Ljava/lang/String;
***************
*** 43,48 ****
--- 44,50 ----
      .locals 2
  
      .prologue
+     .line 2621
      new-instance v0, Ljava/lang/RuntimeException;
  
      iget-object v1, p0, Lcom/android/server/PowerManagerService$12;->val$message:Ljava/lang/String;
