*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/PowerManagerService$5.smali	2012-03-23 19:34:55.338501884 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/PowerManagerService$5.smali.original	2012-03-23 19:34:46.882347686 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1422
      iput-object p1, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 48,53 ****
--- 49,55 ----
  
      const/4 v5, 0x0
  
+     .line 1429
      :goto_0
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
***************
*** 58,63 ****
--- 60,66 ----
  
      monitor-enter v1
  
+     .line 1430
      :try_start_0
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
***************
*** 70,75 ****
--- 73,79 ----
  
      aget v10, v0, v3
  
+     .line 1431
      .local v10, value:I
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
***************
*** 82,87 ****
--- 86,92 ----
  
      aget v11, v0, v3
  
+     .line 1432
      .local v11, why:I
      const/4 v8, 0x0
  
***************
*** 89,94 ****
--- 94,100 ----
      :goto_1
      if-ge v8, v13, :cond_0
  
+     .line 1433
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      #getter for: Lcom/android/server/PowerManagerService;->mBroadcastQueue:[I
***************
*** 109,114 ****
--- 115,121 ----
  
      aput v3, v0, v8
  
+     .line 1434
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      #getter for: Lcom/android/server/PowerManagerService;->mBroadcastWhy:[I
***************
*** 129,138 ****
--- 136,147 ----
  
      aput v3, v0, v8
  
+     .line 1432
      add-int/lit8 v8, v8, 0x1
  
      goto :goto_1
  
+     .line 1436
      :cond_0
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
***************
*** 140,145 ****
--- 149,155 ----
  
      move-result-object v9
  
+     .line 1437
      .local v9, policy:Landroid/view/WindowManagerPolicy;
      if-ne v10, v12, :cond_1
  
***************
*** 152,157 ****
--- 162,168 ----
  
      if-nez v0, :cond_1
  
+     .line 1438
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      const/4 v3, 0x1
***************
*** 159,164 ****
--- 170,176 ----
      #setter for: Lcom/android/server/PowerManagerService;->mPreparingForScreenOn:Z
      invoke-static {v0, v3}, Lcom/android/server/PowerManagerService;->access$3102(Lcom/android/server/PowerManagerService;Z)Z
  
+     .line 1439
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      #getter for: Lcom/android/server/PowerManagerService;->mBroadcastWakeLock:Lcom/android/server/PowerManagerService$UnsynchronizedWakeLock;
***************
*** 168,173 ****
--- 180,186 ----
  
      invoke-virtual {v0}, Lcom/android/server/PowerManagerService$UnsynchronizedWakeLock;->acquire()V
  
+     .line 1440
      const/16 v0, 0xaa5
  
      iget-object v3, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
***************
*** 181,193 ****
--- 194,209 ----
  
      invoke-static {v0, v3}, Landroid/util/EventLog;->writeEvent(II)I
  
+     .line 1443
      :cond_1
      monitor-exit v1
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1444
      if-ne v10, v12, :cond_3
  
+     .line 1445
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
***************
*** 196,201 ****
--- 212,218 ----
  
      iput-wide v3, v0, Lcom/android/server/PowerManagerService;->mScreenOnStart:J
  
+     .line 1447
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      #getter for: Lcom/android/server/PowerManagerService;->mScreenOnListener:Landroid/view/WindowManagerPolicy$ScreenOnListener;
***************
*** 205,210 ****
--- 222,228 ----
  
      invoke-interface {v9, v0}, Landroid/view/WindowManagerPolicy;->screenTurningOn(Landroid/view/WindowManagerPolicy$ScreenOnListener;)V
  
+     .line 1449
      :try_start_1
      invoke-static {}, Landroid/app/ActivityManagerNative;->getDefault()Landroid/app/IActivityManager;
  
***************
*** 214,219 ****
--- 232,238 ----
      :try_end_1
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 1457
      :goto_2
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
***************
*** 230,235 ****
--- 249,255 ----
  
      if-eqz v0, :cond_2
  
+     .line 1458
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      #getter for: Lcom/android/server/PowerManagerService;->mContext:Landroid/content/Context;
***************
*** 266,271 ****
--- 286,292 ----
  
      goto/16 :goto_0
  
+     .line 1443
      .end local v8           #i:I
      .end local v9           #policy:Landroid/view/WindowManagerPolicy;
      .end local v10           #value:I
***************
*** 280,285 ****
--- 301,307 ----
  
      throw v0
  
+     .line 1461
      .restart local v8       #i:I
      .restart local v9       #policy:Landroid/view/WindowManagerPolicy;
      .restart local v10       #value:I
***************
*** 294,299 ****
--- 316,322 ----
  
      monitor-enter v1
  
+     .line 1462
      const/16 v0, 0xaa7
  
      const/4 v3, 0x2
***************
*** 330,335 ****
--- 353,359 ----
  
      invoke-static {v0, v3}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 1464
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      #getter for: Lcom/android/server/PowerManagerService;->mBroadcastWakeLock:Lcom/android/server/PowerManagerService$UnsynchronizedWakeLock;
***************
*** 339,344 ****
--- 363,369 ----
  
      invoke-virtual {v0}, Lcom/android/server/PowerManagerService$UnsynchronizedWakeLock;->release()V
  
+     .line 1465
      monitor-exit v1
  
      goto/16 :goto_0
***************
*** 352,360 ****
--- 377,387 ----
  
      throw v0
  
+     .line 1468
      :cond_3
      if-nez v10, :cond_5
  
+     .line 1469
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
***************
*** 363,370 ****
--- 390,399 ----
  
      iput-wide v3, v0, Lcom/android/server/PowerManagerService;->mScreenOffStart:J
  
+     .line 1471
      invoke-interface {v9, v11}, Landroid/view/WindowManagerPolicy;->screenTurnedOff(I)V
  
+     .line 1473
      :try_start_4
      invoke-static {}, Landroid/app/ActivityManagerNative;->getDefault()Landroid/app/IActivityManager;
  
***************
*** 374,379 ****
--- 403,409 ----
      :try_end_4
      .catch Landroid/os/RemoteException; {:try_start_4 .. :try_end_4} :catch_0
  
+     .line 1478
      :goto_3
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
***************
*** 390,395 ****
--- 420,426 ----
  
      if-eqz v0, :cond_4
  
+     .line 1479
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      #getter for: Lcom/android/server/PowerManagerService;->mContext:Landroid/content/Context;
***************
*** 426,431 ****
--- 457,463 ----
  
      goto/16 :goto_0
  
+     .line 1482
      :cond_4
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
***************
*** 436,441 ****
--- 468,474 ----
  
      monitor-enter v1
  
+     .line 1483
      const/16 v0, 0xaa7
  
      const/4 v3, 0x2
***************
*** 472,477 ****
--- 505,511 ----
  
      invoke-static {v0, v3}, Landroid/util/EventLog;->writeEvent(I[Ljava/lang/Object;)I
  
+     .line 1485
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      iget-object v3, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
***************
*** 486,491 ****
--- 520,526 ----
      #calls: Lcom/android/server/PowerManagerService;->updateLightsLocked(II)V
      invoke-static {v0, v3, v4}, Lcom/android/server/PowerManagerService;->access$3200(Lcom/android/server/PowerManagerService;II)V
  
+     .line 1486
      iget-object v0, p0, Lcom/android/server/PowerManagerService$5;->this$0:Lcom/android/server/PowerManagerService;
  
      #getter for: Lcom/android/server/PowerManagerService;->mBroadcastWakeLock:Lcom/android/server/PowerManagerService$UnsynchronizedWakeLock;
***************
*** 495,500 ****
--- 530,536 ----
  
      invoke-virtual {v0}, Lcom/android/server/PowerManagerService$UnsynchronizedWakeLock;->release()V
  
+     .line 1487
      monitor-exit v1
  
      goto/16 :goto_0
***************
*** 508,521 ****
--- 544,560 ----
  
      throw v0
  
+     .line 1496
      :cond_5
      return-void
  
+     .line 1474
      :catch_0
      move-exception v0
  
      goto :goto_3
  
+     .line 1450
      :catch_1
      move-exception v0
  
