*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/PowerManagerService$BootCompletedReceiver.smali	2012-03-23 19:35:11.297792901 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/PowerManagerService$BootCompletedReceiver.smali.original	2012-03-23 19:34:47.373356640 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 395
      iput-object p1, p0, Lcom/android/server/PowerManagerService$BootCompletedReceiver;->this$0:Lcom/android/server/PowerManagerService;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 395
      invoke-direct {p0, p1}, Lcom/android/server/PowerManagerService$BootCompletedReceiver;-><init>(Lcom/android/server/PowerManagerService;)V
  
      return-void
***************
*** 50,58 ****
--- 52,62 ----
      .parameter "intent"
  
      .prologue
+     .line 398
      iget-object v0, p0, Lcom/android/server/PowerManagerService$BootCompletedReceiver;->this$0:Lcom/android/server/PowerManagerService;
  
      invoke-virtual {v0}, Lcom/android/server/PowerManagerService;->bootCompleted()V
  
+     .line 399
      return-void
  .end method
