*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/StatusBarManagerService$7.smali	2012-03-23 19:34:54.771491543 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/StatusBarManagerService$7.smali.original	2012-03-23 19:34:46.872347503 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 576
      iput-object p1, p0, Lcom/android/server/StatusBarManagerService$7;->this$0:Lcom/android/server/StatusBarManagerService;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 39,48 ****
--- 40,51 ----
      .parameter "intent"
  
      .prologue
+     .line 578
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 579
      .local v0, action:Ljava/lang/String;
      const-string v1, "android.intent.action.CLOSE_SYSTEM_DIALOGS"
  
***************
*** 60,70 ****
--- 63,75 ----
  
      if-eqz v1, :cond_1
  
+     .line 581
      :cond_0
      iget-object v1, p0, Lcom/android/server/StatusBarManagerService$7;->this$0:Lcom/android/server/StatusBarManagerService;
  
      invoke-virtual {v1}, Lcom/android/server/StatusBarManagerService;->collapse()V
  
+     .line 594
      :cond_1
      return-void
  .end method
