*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/StatusBarManagerService$3.smali	2012-03-23 19:34:59.601579621 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/StatusBarManagerService$3.smali.original	2012-03-23 19:34:46.948348889 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .parameter
  
      .prologue
+     .line 282
      iput-object p1, p0, Lcom/android/server/StatusBarManagerService$3;->this$0:Lcom/android/server/StatusBarManagerService;
  
      iput-object p2, p0, Lcom/android/server/StatusBarManagerService$3;->val$token:Landroid/os/IBinder;
***************
*** 55,66 ****
--- 56,69 ----
      .locals 4
  
      .prologue
+     .line 284
      iget-object v0, p0, Lcom/android/server/StatusBarManagerService$3;->this$0:Lcom/android/server/StatusBarManagerService;
  
      iget-object v0, v0, Lcom/android/server/StatusBarManagerService;->mBar:Lcom/android/internal/statusbar/IStatusBar;
  
      if-eqz v0, :cond_0
  
+     .line 286
      :try_start_0
      iget-object v0, p0, Lcom/android/server/StatusBarManagerService$3;->this$0:Lcom/android/server/StatusBarManagerService;
  
***************
*** 76,85 ****
--- 79,90 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 290
      :cond_0
      :goto_0
      return-void
  
+     .line 287
      :catch_0
      move-exception v0
  
