*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/StatusBarManagerService$5.smali	2012-03-23 19:34:51.494431786 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/StatusBarManagerService$5.smali.original	2012-03-23 19:34:46.822346591 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 325
      iput-object p1, p0, Lcom/android/server/StatusBarManagerService$5;->this$0:Lcom/android/server/StatusBarManagerService;
  
      iput-boolean p2, p0, Lcom/android/server/StatusBarManagerService$5;->val$enabled:Z
***************
*** 45,50 ****
--- 46,52 ----
      .locals 2
  
      .prologue
+     .line 327
      iget-object v0, p0, Lcom/android/server/StatusBarManagerService$5;->this$0:Lcom/android/server/StatusBarManagerService;
  
      iget-object v0, v0, Lcom/android/server/StatusBarManagerService;->mWindowManager:Lcom/android/server/wm/WindowManagerService;
***************
*** 53,57 ****
--- 55,60 ----
  
      invoke-virtual {v0, v1}, Lcom/android/server/wm/WindowManagerService;->setHardKeyboardEnabled(Z)V
  
+     .line 328
      return-void
  .end method
