*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/StatusBarManagerService$DisableRecord.smali	2012-03-23 19:34:50.350410926 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/StatusBarManagerService$DisableRecord.smali.original	2012-03-23 19:34:46.809346356 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .parameter
  
      .prologue
+     .line 79
      iput-object p1, p0, Lcom/android/server/StatusBarManagerService$DisableRecord;->this$0:Lcom/android/server/StatusBarManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 46,51 ****
--- 47,53 ----
      .parameter "x1"
  
      .prologue
+     .line 79
      invoke-direct {p0, p1}, Lcom/android/server/StatusBarManagerService$DisableRecord;-><init>(Lcom/android/server/StatusBarManagerService;)V
  
      return-void
***************
*** 59,64 ****
--- 61,67 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 85
      const-string v0, "StatusBarManagerService"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 83,88 ****
--- 86,92 ----
  
      invoke-static {v0, v1}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 86
      iget-object v0, p0, Lcom/android/server/StatusBarManagerService$DisableRecord;->this$0:Lcom/android/server/StatusBarManagerService;
  
      iget-object v1, p0, Lcom/android/server/StatusBarManagerService$DisableRecord;->token:Landroid/os/IBinder;
***************
*** 91,99 ****
--- 95,105 ----
  
      invoke-virtual {v0, v3, v1, v2}, Lcom/android/server/StatusBarManagerService;->disable(ILandroid/os/IBinder;Ljava/lang/String;)V
  
+     .line 87
      iget-object v0, p0, Lcom/android/server/StatusBarManagerService$DisableRecord;->token:Landroid/os/IBinder;
  
      invoke-interface {v0, p0, v3}, Landroid/os/IBinder;->unlinkToDeath(Landroid/os/IBinder$DeathRecipient;I)Z
  
+     .line 88
      return-void
  .end method
