*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/ThrottleService$MyHandler.smali	2012-03-23 19:35:04.323665728 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/ThrottleService$MyHandler.smali.original	2012-03-23 19:34:47.097351606 -0500
***************
*** 25,34 ****
--- 25,37 ----
      .parameter "l"
  
      .prologue
+     .line 388
      iput-object p1, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
+     .line 389
      invoke-direct {p0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
+     .line 390
      return-void
  .end method
  
***************
*** 47,63 ****
--- 50,70 ----
  
      const/4 v4, 0x0
  
+     .line 708
      invoke-static {}, Ljava/util/GregorianCalendar;->getInstance()Ljava/util/Calendar;
  
      move-result-object v1
  
+     .line 709
      .local v1, end:Ljava/util/Calendar;
      invoke-virtual {v1, p1, p2}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 710
      invoke-virtual {v1, v7}, Ljava/util/Calendar;->get(I)I
  
      move-result v0
  
+     .line 711
      .local v0, day:I
      iget-object v3, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 68,87 ****
--- 75,99 ----
  
      invoke-virtual {v1, v7, v3}, Ljava/util/Calendar;->set(II)V
  
+     .line 712
      invoke-virtual {v1, v8, v4}, Ljava/util/Calendar;->set(II)V
  
+     .line 713
      const/16 v3, 0xc
  
      invoke-virtual {v1, v3, v4}, Ljava/util/Calendar;->set(II)V
  
+     .line 714
      const/16 v3, 0xd
  
      invoke-virtual {v1, v3, v4}, Ljava/util/Calendar;->set(II)V
  
+     .line 715
      const/16 v3, 0xe
  
      invoke-virtual {v1, v3, v4}, Ljava/util/Calendar;->set(II)V
  
+     .line 716
      iget-object v3, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mPolicyResetDay:I
***************
*** 91,103 ****
--- 103,118 ----
  
      if-lt v0, v3, :cond_1
  
+     .line 717
      invoke-virtual {v1, v6}, Ljava/util/Calendar;->get(I)I
  
      move-result v2
  
+     .line 718
      .local v2, month:I
      if-ne v2, v8, :cond_0
  
+     .line 719
      invoke-virtual {v1, v5}, Ljava/util/Calendar;->get(I)I
  
      move-result v3
***************
*** 106,118 ****
--- 121,136 ----
  
      invoke-virtual {v1, v5, v3}, Ljava/util/Calendar;->set(II)V
  
+     .line 720
      const/4 v2, -0x1
  
+     .line 722
      :cond_0
      add-int/lit8 v3, v2, 0x1
  
      invoke-virtual {v1, v6, v3}, Ljava/util/Calendar;->set(II)V
  
+     .line 726
      .end local v2           #month:I
      :cond_1
      const-string v3, "persist.throttle.testing"
***************
*** 129,146 ****
--- 147,168 ----
  
      if-eqz v3, :cond_2
  
+     .line 727
      invoke-static {}, Ljava/util/GregorianCalendar;->getInstance()Ljava/util/Calendar;
  
      move-result-object v1
  
+     .line 728
      invoke-virtual {v1, p1, p2}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 729
      const/16 v3, 0xd
  
      const/16 v4, 0x258
  
      invoke-virtual {v1, v3, v4}, Ljava/util/Calendar;->add(II)V
  
+     .line 731
      :cond_2
      return-object v1
  .end method
***************
*** 154,170 ****
--- 176,195 ----
  
      const/4 v3, 0x2
  
+     .line 734
      invoke-virtual {p1}, Ljava/util/Calendar;->clone()Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Ljava/util/Calendar;
  
+     .line 735
      .local v1, start:Ljava/util/Calendar;
      invoke-virtual {p1, v3}, Ljava/util/Calendar;->get(I)I
  
      move-result v0
  
+     .line 736
      .local v0, month:I
      invoke-virtual {p1, v3}, Ljava/util/Calendar;->get(I)I
  
***************
*** 172,179 ****
--- 197,206 ----
  
      if-nez v2, :cond_0
  
+     .line 737
      const/16 v0, 0xc
  
+     .line 738
      invoke-virtual {v1, v4}, Ljava/util/Calendar;->get(I)I
  
      move-result v2
***************
*** 182,192 ****
--- 209,221 ----
  
      invoke-virtual {v1, v4, v2}, Ljava/util/Calendar;->set(II)V
  
+     .line 740
      :cond_0
      add-int/lit8 v2, v0, -0x1
  
      invoke-virtual {v1, v3, v2}, Ljava/util/Calendar;->set(II)V
  
+     .line 743
      const-string v2, "persist.throttle.testing"
  
      invoke-static {v2}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
***************
*** 201,206 ****
--- 230,236 ----
  
      if-eqz v2, :cond_1
  
+     .line 744
      invoke-virtual {p1}, Ljava/util/Calendar;->clone()Ljava/lang/Object;
  
      move-result-object v1
***************
*** 208,213 ****
--- 238,244 ----
      .end local v1           #start:Ljava/util/Calendar;
      check-cast v1, Ljava/util/Calendar;
  
+     .line 745
      .restart local v1       #start:Ljava/util/Calendar;
      const/16 v2, 0xd
  
***************
*** 215,220 ****
--- 246,252 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/util/Calendar;->add(II)V
  
+     .line 747
      :cond_1
      return-object v1
  .end method
***************
*** 224,229 ****
--- 256,262 ----
      .parameter "currentTotal"
  
      .prologue
+     .line 588
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 239,244 ****
--- 272,278 ----
  
      move-result-wide v15
  
+     .line 589
      .local v15, threshold:J
      const-wide/16 v21, 0x0
  
***************
*** 246,257 ****
--- 280,294 ----
  
      if-nez v21, :cond_1
  
+     .line 590
      invoke-direct/range {p0 .. p0}, Lcom/android/server/ThrottleService$MyHandler;->clearThrottleAndNotification()V
  
+     .line 663
      :cond_0
      :goto_0
      return-void
  
+     .line 596
      :cond_1
      move-object/from16 v0, p0
  
***************
*** 270,275 ****
--- 307,313 ----
  
      if-nez v21, :cond_2
  
+     .line 597
      const-string v21, "ThrottleService"
  
      const-string v22, "missing trusted time, skipping throttle check"
***************
*** 278,288 ****
--- 316,328 ----
  
      goto :goto_0
  
+     .line 602
      :cond_2
      cmp-long v21, p1, v15
  
      if-lez v21, :cond_3
  
+     .line 603
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 306,311 ****
--- 346,352 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 604
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 321,326 ****
--- 362,368 ----
  
      invoke-virtual/range {v21 .. v22}, Ljava/util/concurrent/atomic/AtomicInteger;->set(I)V
  
+     .line 605
      const-string v21, "ThrottleService"
  
      new-instance v22, Ljava/lang/StringBuilder;
***************
*** 353,358 ****
--- 395,401 ----
  
      invoke-static/range {v21 .. v22}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 607
      :try_start_0
      move-object/from16 v0, p0
  
***************
*** 410,415 ****
--- 453,459 ----
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 613
      :goto_1
      move-object/from16 v0, p0
  
***************
*** 426,431 ****
--- 470,476 ----
  
      invoke-virtual/range {v21 .. v22}, Landroid/app/NotificationManager;->cancel(I)V
  
+     .line 615
      const v21, 0x1040455
  
      const v22, 0x1040456
***************
*** 446,451 ****
--- 491,497 ----
  
      invoke-direct {v0, v1, v2, v3, v4}, Lcom/android/server/ThrottleService$MyHandler;->postNotification(IIII)V
  
+     .line 620
      new-instance v5, Landroid/content/Intent;
  
      const-string v21, "android.net.thrott.THROTTLE_ACTION"
***************
*** 454,459 ****
--- 500,506 ----
  
      invoke-direct {v5, v0}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 621
      .local v5, broadcast:Landroid/content/Intent;
      const-string v21, "level"
  
***************
*** 478,483 ****
--- 525,531 ----
  
      invoke-virtual {v5, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 623
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 495,504 ****
--- 543,554 ----
  
      goto/16 :goto_0
  
+     .line 609
      .end local v5           #broadcast:Landroid/content/Intent;
      :catch_0
      move-exception v6
  
+     .line 610
      .local v6, e:Ljava/lang/Exception;
      const-string v21, "ThrottleService"
  
***************
*** 526,535 ****
--- 576,587 ----
  
      goto :goto_1
  
+     .line 627
      .end local v6           #e:Ljava/lang/Exception;
      :cond_3
      invoke-direct/range {p0 .. p0}, Lcom/android/server/ThrottleService$MyHandler;->clearThrottleAndNotification()V
  
+     .line 628
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 545,550 ****
--- 597,603 ----
  
      if-eqz v21, :cond_0
  
+     .line 640
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 560,565 ****
--- 613,619 ----
  
      move-result-wide v13
  
+     .line 641
      .local v13, start:J
      move-object/from16 v0, p0
  
***************
*** 576,592 ****
--- 630,650 ----
  
      move-result-wide v7
  
+     .line 642
      .local v7, end:J
      sub-long v11, v7, v13
  
+     .line 643
      .local v11, periodLength:J
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v9
  
+     .line 644
      .local v9, now:J
      sub-long v17, v9, v13
  
+     .line 645
      .local v17, timeUsed:J
      const-wide/16 v21, 0x2
  
***************
*** 598,603 ****
--- 656,662 ----
  
      div-long v19, v21, v23
  
+     .line 646
      .local v19, warningThreshold:J
      cmp-long v21, p1, v19
  
***************
*** 611,616 ****
--- 670,676 ----
  
      if-lez v21, :cond_4
  
+     .line 647
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 624,629 ****
--- 684,690 ----
  
      if-nez v21, :cond_0
  
+     .line 648
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 635,640 ****
--- 696,702 ----
      #setter for: Lcom/android/server/ThrottleService;->mWarningNotificationSent:Z
      invoke-static/range {v21 .. v22}, Lcom/android/server/ThrottleService;->access$1902(Lcom/android/server/ThrottleService;Z)Z
  
+     .line 649
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 650,655 ****
--- 712,718 ----
  
      invoke-virtual/range {v21 .. v22}, Landroid/app/NotificationManager;->cancel(I)V
  
+     .line 650
      const v21, 0x1040453
  
      const v22, 0x1040454
***************
*** 672,677 ****
--- 735,741 ----
  
      goto/16 :goto_0
  
+     .line 656
      :cond_4
      move-object/from16 v0, p0
  
***************
*** 692,697 ****
--- 756,762 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 657
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 707,712 ****
--- 772,778 ----
  
      invoke-virtual/range {v21 .. v22}, Landroid/app/NotificationManager;->cancel(I)V
  
+     .line 658
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 729,734 ****
--- 795,801 ----
  
      const/4 v6, -0x1
  
+     .line 692
      iget-object v2, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mThrottleIndex:Ljava/util/concurrent/atomic/AtomicInteger;
***************
*** 742,747 ****
--- 809,815 ----
  
      if-eqz v2, :cond_0
  
+     .line 693
      iget-object v2, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mThrottleIndex:Ljava/util/concurrent/atomic/AtomicInteger;
***************
*** 751,756 ****
--- 819,825 ----
  
      invoke-virtual {v2, v7}, Ljava/util/concurrent/atomic/AtomicInteger;->set(I)V
  
+     .line 695
      :try_start_0
      iget-object v2, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 774,779 ****
--- 843,849 ----
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 699
      :goto_0
      new-instance v0, Landroid/content/Intent;
  
***************
*** 781,791 ****
--- 851,863 ----
  
      invoke-direct {v0, v2}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 700
      .local v0, broadcast:Landroid/content/Intent;
      const-string v2, "level"
  
      invoke-virtual {v0, v2, v6}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 701
      iget-object v2, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mContext:Landroid/content/Context;
***************
*** 795,800 ****
--- 867,873 ----
  
      invoke-virtual {v2, v0}, Landroid/content/Context;->sendStickyBroadcast(Landroid/content/Intent;)V
  
+     .line 702
      iget-object v2, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mNotificationManager:Landroid/app/NotificationManager;
***************
*** 806,823 ****
--- 879,900 ----
  
      invoke-virtual {v2, v3}, Landroid/app/NotificationManager;->cancel(I)V
  
+     .line 703
      iget-object v2, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #setter for: Lcom/android/server/ThrottleService;->mWarningNotificationSent:Z
      invoke-static {v2, v7}, Lcom/android/server/ThrottleService;->access$1902(Lcom/android/server/ThrottleService;Z)Z
  
+     .line 705
      .end local v0           #broadcast:Landroid/content/Intent;
      :cond_0
      return-void
  
+     .line 696
      :catch_0
      move-exception v1
  
+     .line 697
      .local v1, e:Ljava/lang/Exception;
      const-string v2, "ThrottleService"
  
***************
*** 848,853 ****
--- 925,931 ----
      .locals 5
  
      .prologue
+     .line 575
      iget-object v1, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mThrottleIndex:Ljava/util/concurrent/atomic/AtomicInteger;
***************
*** 863,868 ****
--- 941,947 ----
  
      if-ne v1, v2, :cond_0
  
+     .line 577
      :try_start_0
      iget-object v1, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 884,889 ****
--- 963,969 ----
  
      invoke-interface {v1, v2, v3, v4}, Landroid/os/INetworkManagementService;->setInterfaceThrottle(Ljava/lang/String;II)V
  
+     .line 578
      iget-object v1, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mNMService:Landroid/os/INetworkManagementService;
***************
*** 924,936 ****
--- 1004,1019 ----
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 584
      :cond_0
      :goto_0
      return-void
  
+     .line 580
      :catch_0
      move-exception v0
  
+     .line 581
      .local v0, e:Ljava/lang/Exception;
      const-string v1, "ThrottleService"
  
***************
*** 961,966 ****
--- 1044,1050 ----
      .locals 17
  
      .prologue
+     .line 433
      const-string v13, "persist.throttle.testing"
  
      invoke-static {v13}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
***************
*** 973,978 ****
--- 1057,1063 ----
  
      move-result v9
  
+     .line 435
      .local v9, testing:Z
      move-object/from16 v0, p0
  
***************
*** 993,998 ****
--- 1078,1084 ----
  
      move-result v8
  
+     .line 437
      .local v8, pollingPeriod:I
      move-object/from16 v0, p0
  
***************
*** 1020,1025 ****
--- 1106,1112 ----
      #setter for: Lcom/android/server/ThrottleService;->mPolicyPollPeriodSec:I
      invoke-static {v13, v14}, Lcom/android/server/ThrottleService;->access$402(Lcom/android/server/ThrottleService;I)I
  
+     .line 441
      move-object/from16 v0, p0
  
      iget-object v13, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1041,1046 ****
--- 1128,1134 ----
  
      int-to-long v3, v13
  
+     .line 443
      .local v3, defaultThreshold:J
      move-object/from16 v0, p0
  
***************
*** 1061,1066 ****
--- 1149,1155 ----
  
      move-result v5
  
+     .line 445
      .local v5, defaultValue:I
      move-object/from16 v0, p0
  
***************
*** 1081,1086 ****
--- 1170,1176 ----
  
      move-result-wide v10
  
+     .line 447
      .local v10, threshold:J
      move-object/from16 v0, p0
  
***************
*** 1101,1106 ****
--- 1191,1197 ----
  
      move-result v12
  
+     .line 450
      .local v12, value:I
      move-object/from16 v0, p0
  
***************
*** 1113,1118 ****
--- 1204,1210 ----
  
      invoke-virtual {v13, v10, v11}, Ljava/util/concurrent/atomic/AtomicLong;->set(J)V
  
+     .line 451
      move-object/from16 v0, p0
  
      iget-object v13, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1124,1131 ****
--- 1216,1225 ----
  
      invoke-virtual {v13, v12}, Ljava/util/concurrent/atomic/AtomicInteger;->set(I)V
  
+     .line 452
      if-eqz v9, :cond_0
  
+     .line 453
      move-object/from16 v0, p0
  
      iget-object v13, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1135,1140 ****
--- 1229,1235 ----
      #setter for: Lcom/android/server/ThrottleService;->mPolicyPollPeriodSec:I
      invoke-static {v13, v14}, Lcom/android/server/ThrottleService;->access$402(Lcom/android/server/ThrottleService;I)I
  
+     .line 454
      move-object/from16 v0, p0
  
      iget-object v13, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1148,1153 ****
--- 1243,1249 ----
  
      invoke-virtual {v13, v14, v15}, Ljava/util/concurrent/atomic/AtomicLong;->set(J)V
  
+     .line 457
      :cond_0
      move-object/from16 v0, p0
  
***************
*** 1177,1182 ****
--- 1273,1279 ----
      #setter for: Lcom/android/server/ThrottleService;->mPolicyResetDay:I
      invoke-static {v13, v14}, Lcom/android/server/ThrottleService;->access$702(Lcom/android/server/ThrottleService;I)I
  
+     .line 459
      move-object/from16 v0, p0
  
      iget-object v13, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1216,1226 ****
--- 1313,1325 ----
  
      if-le v13, v14, :cond_2
  
+     .line 461
      :cond_1
      new-instance v6, Ljava/util/Random;
  
      invoke-direct {v6}, Ljava/util/Random;-><init>()V
  
+     .line 462
      .local v6, g:Ljava/util/Random;
      move-object/from16 v0, p0
  
***************
*** 1237,1242 ****
--- 1336,1342 ----
      #setter for: Lcom/android/server/ThrottleService;->mPolicyResetDay:I
      invoke-static {v13, v14}, Lcom/android/server/ThrottleService;->access$702(Lcom/android/server/ThrottleService;I)I
  
+     .line 463
      move-object/from16 v0, p0
  
      iget-object v13, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1263,1268 ****
--- 1363,1369 ----
  
      invoke-static {v13, v14, v15}, Landroid/provider/Settings$Secure;->putInt(Landroid/content/ContentResolver;Ljava/lang/String;I)Z
  
+     .line 466
      .end local v6           #g:Ljava/util/Random;
      :cond_2
      move-object/from16 v0, p0
***************
*** 1276,1281 ****
--- 1377,1383 ----
  
      if-nez v13, :cond_3
  
+     .line 467
      move-object/from16 v0, p0
  
      iget-object v13, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1289,1294 ****
--- 1391,1397 ----
  
      invoke-virtual {v13, v14, v15}, Ljava/util/concurrent/atomic/AtomicLong;->set(J)V
  
+     .line 470
      :cond_3
      move-object/from16 v0, p0
  
***************
*** 1309,1314 ****
--- 1412,1418 ----
  
      move-result v2
  
+     .line 472
      .local v2, defaultNotificationType:I
      move-object/from16 v0, p0
  
***************
*** 1336,1341 ****
--- 1440,1446 ----
      #setter for: Lcom/android/server/ThrottleService;->mPolicyNotificationsAllowedMask:I
      invoke-static {v13, v14}, Lcom/android/server/ThrottleService;->access$902(Lcom/android/server/ThrottleService;I)I
  
+     .line 475
      move-object/from16 v0, p0
  
      iget-object v13, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1357,1362 ****
--- 1462,1468 ----
  
      move-result v7
  
+     .line 478
      .local v7, maxNtpCacheAgeSec:I
      move-object/from16 v0, p0
  
***************
*** 1369,1374 ****
--- 1475,1481 ----
      #setter for: Lcom/android/server/ThrottleService;->mMaxNtpCacheAge:J
      invoke-static {v13, v14, v15}, Lcom/android/server/ThrottleService;->access$1002(Lcom/android/server/ThrottleService;J)J
  
+     .line 480
      move-object/from16 v0, p0
  
      iget-object v13, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1388,1393 ****
--- 1495,1501 ----
  
      if-eqz v13, :cond_4
  
+     .line 481
      const-string v13, "ThrottleService"
  
      new-instance v14, Ljava/lang/StringBuilder;
***************
*** 1532,1537 ****
--- 1640,1646 ----
  
      invoke-static {v13, v14}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 489
      :cond_4
      move-object/from16 v0, p0
  
***************
*** 1546,1561 ****
--- 1655,1674 ----
  
      invoke-virtual {v13, v14}, Ljava/util/concurrent/atomic/AtomicInteger;->set(I)V
  
+     .line 491
      invoke-direct/range {p0 .. p0}, Lcom/android/server/ThrottleService$MyHandler;->onResetAlarm()V
  
+     .line 493
      invoke-direct/range {p0 .. p0}, Lcom/android/server/ThrottleService$MyHandler;->onPollAlarm()V
  
+     .line 495
      new-instance v1, Landroid/content/Intent;
  
      const-string v13, "android.net.thrott.POLICY_CHANGED_ACTION"
  
      invoke-direct {v1, v13}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 496
      .local v1, broadcast:Landroid/content/Intent;
      move-object/from16 v0, p0
  
***************
*** 1568,1573 ****
--- 1681,1687 ----
  
      invoke-virtual {v13, v1}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 497
      return-void
  .end method
  
***************
*** 1575,1584 ****
--- 1689,1700 ----
      .locals 29
  
      .prologue
+     .line 500
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v15
  
+     .line 501
      .local v15, now:J
      move-object/from16 v0, p0
  
***************
*** 1605,1610 ****
--- 1721,1727 ----
  
      add-long v13, v15, v25
  
+     .line 504
      .local v13, next:J
      move-object/from16 v0, p0
  
***************
*** 1636,1641 ****
--- 1753,1759 ----
  
      if-lez v25, :cond_0
  
+     .line 505
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1653,1658 ****
--- 1771,1777 ----
  
      if-eqz v25, :cond_0
  
+     .line 507
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1661,1672 ****
--- 1780,1794 ----
  
      invoke-virtual/range {v25 .. v25}, Lcom/android/server/ThrottleService;->dispatchReset()V
  
+     .line 511
      :cond_0
      const-wide/16 v8, 0x0
  
+     .line 512
      .local v8, incRead:J
      const-wide/16 v10, 0x0
  
+     .line 514
      .local v10, incWrite:J
      :try_start_0
      move-object/from16 v0, p0
***************
*** 1684,1689 ****
--- 1806,1812 ----
  
      move-result-object v22
  
+     .line 515
      .local v22, stats:Landroid/net/NetworkStats;
      move-object/from16 v0, p0
  
***************
*** 1716,1721 ****
--- 1839,1845 ----
  
      move-result v12
  
+     .line 518
      .local v12, index:I
      const/16 v25, -0x1
  
***************
*** 1723,1728 ****
--- 1847,1853 ----
  
      if-eq v12, v0, :cond_5
  
+     .line 519
      const/16 v25, 0x0
  
      move-object/from16 v0, v22
***************
*** 1733,1738 ****
--- 1858,1864 ----
  
      move-result-object v7
  
+     .line 520
      .local v7, entry:Landroid/net/NetworkStats$Entry;
      iget-wide v0, v7, Landroid/net/NetworkStats$Entry;->rxBytes:J
  
***************
*** 1751,1756 ****
--- 1877,1883 ----
  
      sub-long v8, v25, v27
  
+     .line 521
      iget-wide v0, v7, Landroid/net/NetworkStats$Entry;->txBytes:J
  
      move-wide/from16 v25, v0
***************
*** 1768,1773 ****
--- 1895,1901 ----
  
      sub-long v10, v25, v27
  
+     .line 529
      .end local v7           #entry:Landroid/net/NetworkStats$Entry;
      :goto_0
      const-wide/16 v25, 0x0
***************
*** 1782,1787 ****
--- 1910,1916 ----
  
      if-gez v25, :cond_2
  
+     .line 530
      :cond_1
      move-object/from16 v0, p0
  
***************
*** 1796,1801 ****
--- 1925,1931 ----
  
      add-long v8, v8, v25
  
+     .line 531
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1809,1814 ****
--- 1939,1945 ----
  
      add-long v10, v10, v25
  
+     .line 532
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1820,1825 ****
--- 1951,1957 ----
      #setter for: Lcom/android/server/ThrottleService;->mLastRead:J
      invoke-static/range {v25 .. v27}, Lcom/android/server/ThrottleService;->access$1302(Lcom/android/server/ThrottleService;J)J
  
+     .line 533
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1834,1839 ****
--- 1966,1972 ----
      .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 542
      .end local v12           #index:I
      .end local v22           #stats:Landroid/net/NetworkStats;
      :cond_2
***************
*** 1850,1858 ****
--- 1983,1993 ----
  
      move-result v21
  
+     .line 544
      .local v21, roaming:Z
      if-nez v21, :cond_3
  
+     .line 545
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 1868,1873 ****
--- 2003,2009 ----
  
      invoke-virtual {v0, v8, v9, v10, v11}, Lcom/android/server/ThrottleService$DataRecorder;->addData(JJ)V
  
+     .line 548
      :cond_3
      move-object/from16 v0, p0
  
***************
*** 1886,1891 ****
--- 2022,2028 ----
  
      move-result-wide v17
  
+     .line 549
      .local v17, periodRx:J
      move-object/from16 v0, p0
  
***************
*** 1904,1912 ****
--- 2041,2051 ----
  
      move-result-wide v19
  
+     .line 550
      .local v19, periodTx:J
      add-long v23, v17, v19
  
+     .line 551
      .local v23, total:J
      move-object/from16 v0, p0
  
***************
*** 1929,1934 ****
--- 2068,2074 ----
  
      if-eqz v25, :cond_4
  
+     .line 552
      const-string v25, "ThrottleService"
  
      new-instance v26, Ljava/lang/StringBuilder;
***************
*** 1993,1998 ****
--- 2133,2139 ----
  
      invoke-static/range {v25 .. v26}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 555
      :cond_4
      move-object/from16 v0, p0
  
***************
*** 2004,2009 ****
--- 2145,2151 ----
  
      invoke-static {v0, v8, v9}, Lcom/android/server/ThrottleService;->access$1314(Lcom/android/server/ThrottleService;J)J
  
+     .line 556
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 2014,2025 ****
--- 2156,2169 ----
  
      invoke-static {v0, v10, v11}, Lcom/android/server/ThrottleService;->access$1414(Lcom/android/server/ThrottleService;J)J
  
+     .line 558
      move-object/from16 v0, p0
  
      move-wide/from16 v1, v23
  
      invoke-direct {v0, v1, v2}, Lcom/android/server/ThrottleService$MyHandler;->checkThrottleAndPostNotification(J)V
  
+     .line 560
      new-instance v5, Landroid/content/Intent;
  
      const-string v25, "android.net.thrott.POLL_ACTION"
***************
*** 2028,2033 ****
--- 2172,2178 ----
  
      invoke-direct {v5, v0}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 561
      .local v5, broadcast:Landroid/content/Intent;
      const-string v25, "cycleRead"
  
***************
*** 2037,2042 ****
--- 2182,2188 ----
  
      invoke-virtual {v5, v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;J)Landroid/content/Intent;
  
+     .line 562
      const-string v25, "cycleWrite"
  
      move-object/from16 v0, v25
***************
*** 2045,2050 ****
--- 2191,2197 ----
  
      invoke-virtual {v5, v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;J)Landroid/content/Intent;
  
+     .line 563
      const-string v25, "cycleStart"
  
      move-object/from16 v0, p0
***************
*** 2074,2079 ****
--- 2221,2227 ----
  
      invoke-virtual {v5, v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;J)Landroid/content/Intent;
  
+     .line 564
      const-string v25, "cycleEnd"
  
      move-object/from16 v0, p0
***************
*** 2103,2108 ****
--- 2251,2257 ----
  
      invoke-virtual {v5, v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;J)Landroid/content/Intent;
  
+     .line 565
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 2118,2123 ****
--- 2267,2273 ----
  
      invoke-virtual {v0, v5}, Landroid/content/Context;->sendStickyBroadcast(Landroid/content/Intent;)V
  
+     .line 566
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 2129,2134 ****
--- 2279,2285 ----
      #setter for: Lcom/android/server/ThrottleService;->mPollStickyBroadcast:Landroid/content/Intent;
      invoke-static {v0, v5}, Lcom/android/server/ThrottleService;->access$1502(Lcom/android/server/ThrottleService;Landroid/content/Intent;)Landroid/content/Intent;
  
+     .line 568
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 2153,2158 ****
--- 2304,2310 ----
  
      invoke-virtual/range {v25 .. v26}, Landroid/app/AlarmManager;->cancel(Landroid/app/PendingIntent;)V
  
+     .line 569
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
***************
*** 2185,2192 ****
--- 2337,2346 ----
  
      invoke-virtual {v0, v1, v13, v14, v2}, Landroid/app/AlarmManager;->set(IJLandroid/app/PendingIntent;)V
  
+     .line 570
      return-void
  
+     .line 524
      .end local v5           #broadcast:Landroid/content/Intent;
      .end local v17           #periodRx:J
      .end local v19           #periodTx:J
***************
*** 2234,2244 ****
--- 2388,2400 ----
  
      goto/16 :goto_0
  
+     .line 535
      .end local v12           #index:I
      .end local v22           #stats:Landroid/net/NetworkStats;
      :catch_0
      move-exception v6
  
+     .line 536
      .local v6, e:Ljava/lang/IllegalStateException;
      const-string v25, "ThrottleService"
  
***************
*** 2266,2275 ****
--- 2422,2433 ----
  
      goto/16 :goto_1
  
+     .line 537
      .end local v6           #e:Ljava/lang/IllegalStateException;
      :catch_1
      move-exception v6
  
+     .line 538
      .local v6, e:Landroid/os/RemoteException;
      const-string v25, "ThrottleService"
  
***************
*** 2302,2307 ****
--- 2460,2466 ----
      .locals 4
  
      .prologue
+     .line 417
      iget-object v0, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mThrottleIndex:Ljava/util/concurrent/atomic/AtomicInteger;
***************
*** 2313,2318 ****
--- 2472,2478 ----
  
      invoke-virtual {v0, v1}, Ljava/util/concurrent/atomic/AtomicInteger;->set(I)V
  
+     .line 419
      iget-object v0, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      new-instance v1, Lcom/android/server/ThrottleService$DataRecorder;
***************
*** 2331,2336 ****
--- 2491,2497 ----
      #setter for: Lcom/android/server/ThrottleService;->mRecorder:Lcom/android/server/ThrottleService$DataRecorder;
      invoke-static {v0, v1}, Lcom/android/server/ThrottleService;->access$102(Lcom/android/server/ThrottleService;Lcom/android/server/ThrottleService$DataRecorder;)Lcom/android/server/ThrottleService$DataRecorder;
  
+     .line 422
      iget-object v0, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mHandler:Landroid/os/Handler;
***************
*** 2346,2351 ****
--- 2507,2513 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 427
      iget-object v0, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mHandler:Landroid/os/Handler;
***************
*** 2370,2375 ****
--- 2532,2538 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 429
      return-void
  .end method
  
***************
*** 2379,2384 ****
--- 2542,2548 ----
      .prologue
      const/4 v10, 0x0
  
+     .line 751
      iget-object v6, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mPolicyThreshold:Ljava/util/concurrent/atomic/AtomicLong;
***************
*** 2396,2401 ****
--- 2560,2566 ----
  
      if-eqz v6, :cond_0
  
+     .line 752
      const-string v6, "ThrottleService"
  
      new-instance v7, Ljava/lang/StringBuilder;
***************
*** 2456,2461 ****
--- 2621,2627 ----
  
      invoke-static {v6, v7}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 757
      :cond_0
      iget-object v6, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 2479,2484 ****
--- 2645,2651 ----
  
      if-lez v6, :cond_1
  
+     .line 758
      iget-object v6, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mTime:Landroid/util/TrustedTime;
***************
*** 2488,2493 ****
--- 2655,2661 ----
  
      invoke-interface {v6}, Landroid/util/TrustedTime;->forceRefresh()Z
  
+     .line 763
      :cond_1
      iget-object v6, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 2502,2507 ****
--- 2670,2676 ----
  
      if-eqz v6, :cond_3
  
+     .line 764
      iget-object v6, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mTime:Landroid/util/TrustedTime;
***************
*** 2513,2528 ****
--- 2682,2700 ----
  
      move-result-wide v1
  
+     .line 765
      .local v1, now:J
      invoke-direct {p0, v1, v2}, Lcom/android/server/ThrottleService$MyHandler;->calculatePeriodEnd(J)Ljava/util/Calendar;
  
      move-result-object v0
  
+     .line 766
      .local v0, end:Ljava/util/Calendar;
      invoke-direct {p0, v0}, Lcom/android/server/ThrottleService$MyHandler;->calculatePeriodStart(Ljava/util/Calendar;)Ljava/util/Calendar;
  
      move-result-object v5
  
+     .line 768
      .local v5, start:Ljava/util/Calendar;
      iget-object v6, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 2537,2544 ****
--- 2709,2718 ----
  
      if-eqz v6, :cond_2
  
+     .line 769
      invoke-direct {p0}, Lcom/android/server/ThrottleService$MyHandler;->onPollAlarm()V
  
+     .line 772
      :cond_2
      iget-object v6, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 2556,2567 ****
--- 2730,2743 ----
  
      invoke-virtual {v6, v7}, Landroid/app/AlarmManager;->cancel(Landroid/app/PendingIntent;)V
  
+     .line 773
      invoke-virtual {v0}, Ljava/util/Calendar;->getTimeInMillis()J
  
      move-result-wide v6
  
      sub-long v3, v6, v1
  
+     .line 775
      .local v3, offset:J
      iget-object v6, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 2587,2592 ****
--- 2763,2769 ----
  
      invoke-virtual {v6, v7, v8, v9, v10}, Landroid/app/AlarmManager;->set(IJLandroid/app/PendingIntent;)V
  
+     .line 781
      .end local v0           #end:Ljava/util/Calendar;
      .end local v1           #now:J
      .end local v3           #offset:J
***************
*** 2605,2614 ****
--- 2782,2793 ----
      .prologue
      const/4 v7, 0x0
  
+     .line 666
      new-instance v0, Landroid/content/Intent;
  
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
+     .line 668
      .local v0, intent:Landroid/content/Intent;
      const-string v5, "com.android.phone"
  
***************
*** 2616,2625 ****
--- 2795,2806 ----
  
      invoke-virtual {v0, v5, v6}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 669
      const/high16 v5, 0x4000
  
      invoke-virtual {v0, v5}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 671
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mContext:Landroid/content/Context;
***************
*** 2631,2651 ****
--- 2812,2836 ----
  
      move-result-object v2
  
+     .line 673
      .local v2, pi:Landroid/app/PendingIntent;
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v3
  
+     .line 674
      .local v3, r:Landroid/content/res/Resources;
      invoke-virtual {v3, p1}, Landroid/content/res/Resources;->getText(I)Ljava/lang/CharSequence;
  
      move-result-object v4
  
+     .line 675
      .local v4, title:Ljava/lang/CharSequence;
      invoke-virtual {v3, p2}, Landroid/content/res/Resources;->getText(I)Ljava/lang/CharSequence;
  
      move-result-object v1
  
+     .line 676
      .local v1, message:Ljava/lang/CharSequence;
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 2656,2661 ****
--- 2841,2847 ----
  
      if-nez v5, :cond_0
  
+     .line 677
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      new-instance v6, Landroid/app/Notification;
***************
*** 2665,2670 ****
--- 2851,2857 ----
      #setter for: Lcom/android/server/ThrottleService;->mThrottlingNotification:Landroid/app/Notification;
      invoke-static {v5, v6}, Lcom/android/server/ThrottleService;->access$2002(Lcom/android/server/ThrottleService;Landroid/app/Notification;)Landroid/app/Notification;
  
+     .line 678
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mThrottlingNotification:Landroid/app/Notification;
***************
*** 2676,2681 ****
--- 2863,2869 ----
  
      iput-wide v6, v5, Landroid/app/Notification;->when:J
  
+     .line 680
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mThrottlingNotification:Landroid/app/Notification;
***************
*** 2685,2690 ****
--- 2873,2879 ----
  
      iput p3, v5, Landroid/app/Notification;->icon:I
  
+     .line 681
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mThrottlingNotification:Landroid/app/Notification;
***************
*** 2698,2703 ****
--- 2887,2893 ----
  
      iput v6, v5, Landroid/app/Notification;->defaults:I
  
+     .line 683
      :cond_0
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
***************
*** 2708,2713 ****
--- 2898,2904 ----
  
      iput p4, v5, Landroid/app/Notification;->flags:I
  
+     .line 684
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mThrottlingNotification:Landroid/app/Notification;
***************
*** 2717,2722 ****
--- 2908,2914 ----
  
      iput-object v4, v5, Landroid/app/Notification;->tickerText:Ljava/lang/CharSequence;
  
+     .line 685
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mThrottlingNotification:Landroid/app/Notification;
***************
*** 2733,2738 ****
--- 2925,2931 ----
  
      invoke-virtual {v5, v6, v4, v1, v2}, Landroid/app/Notification;->setLatestEventInfo(Landroid/content/Context;Ljava/lang/CharSequence;Ljava/lang/CharSequence;Landroid/app/PendingIntent;)V
  
+     .line 687
      iget-object v5, p0, Lcom/android/server/ThrottleService$MyHandler;->this$0:Lcom/android/server/ThrottleService;
  
      #getter for: Lcom/android/server/ThrottleService;->mNotificationManager:Landroid/app/NotificationManager;
***************
*** 2758,2763 ****
--- 2951,2957 ----
  
      invoke-virtual {v5, v6, v7}, Landroid/app/NotificationManager;->notify(ILandroid/app/Notification;)V
  
+     .line 688
      return-void
  .end method
  
***************
*** 2768,2805 ****
--- 2962,3007 ----
      .parameter "msg"
  
      .prologue
+     .line 394
      iget v0, p1, Landroid/os/Message;->what:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 410
      :goto_0
      return-void
  
+     .line 396
      :pswitch_0
      invoke-direct {p0}, Lcom/android/server/ThrottleService$MyHandler;->onRebootRecovery()V
  
      goto :goto_0
  
+     .line 399
      :pswitch_1
      invoke-direct {p0}, Lcom/android/server/ThrottleService$MyHandler;->onPolicyChanged()V
  
      goto :goto_0
  
+     .line 402
      :pswitch_2
      invoke-direct {p0}, Lcom/android/server/ThrottleService$MyHandler;->onPollAlarm()V
  
      goto :goto_0
  
+     .line 405
      :pswitch_3
      invoke-direct {p0}, Lcom/android/server/ThrottleService$MyHandler;->onResetAlarm()V
  
      goto :goto_0
  
+     .line 408
      :pswitch_4
      invoke-direct {p0}, Lcom/android/server/ThrottleService$MyHandler;->onIfaceUp()V
  
      goto :goto_0
  
+     .line 394
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
