*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/ThrottleService$DataRecorder.smali	2012-03-23 19:35:11.341793704 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/ThrottleService$DataRecorder.smali.original	2012-03-23 19:34:47.373356640 -0500
***************
*** 49,64 ****
--- 49,69 ----
      .parameter "parent"
  
      .prologue
+     .line 804
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 800
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mImsi:Ljava/lang/String;
  
+     .line 805
      iput-object p1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mContext:Landroid/content/Context;
  
+     .line 806
      iput-object p2, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
+     .line 808
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mContext:Landroid/content/Context;
  
      const-string v1, "phone"
***************
*** 71,115 ****
--- 76,130 ----
  
      iput-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mTelephonyManager:Landroid/telephony/TelephonyManager;
  
+     .line 811
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v1
  
+     .line 812
      const/4 v0, 0x6
  
      :try_start_0
      iput v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
+     .line 813
      iget v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
      new-array v0, v0, [J
  
      iput-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodRxData:[J
  
+     .line 814
      iget v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
      new-array v0, v0, [J
  
      iput-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodTxData:[J
  
+     .line 816
      invoke-static {}, Ljava/util/Calendar;->getInstance()Ljava/util/Calendar;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodStart:Ljava/util/Calendar;
  
+     .line 817
      invoke-static {}, Ljava/util/Calendar;->getInstance()Ljava/util/Calendar;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodEnd:Ljava/util/Calendar;
  
+     .line 819
      invoke-direct {p0}, Lcom/android/server/ThrottleService$DataRecorder;->retrieve()V
  
+     .line 820
      monitor-exit v1
  
+     .line 821
      return-void
  
+     .line 820
      :catchall_0
      move-exception v0
  
***************
*** 127,136 ****
--- 142,153 ----
      .prologue
      const/4 v10, 0x3
  
+     .line 947
      invoke-virtual {p1}, Ljava/io/File;->listFiles()[Ljava/io/File;
  
      move-result-object v2
  
+     .line 949
      .local v2, files:[Ljava/io/File;
      if-eqz v2, :cond_0
  
***************
*** 138,147 ****
--- 155,166 ----
  
      if-gt v6, v10, :cond_1
  
+     .line 963
      :cond_0
      :goto_0
      return-void
  
+     .line 950
      :cond_1
      const-string v6, "ThrottleService"
  
***************
*** 149,157 ****
--- 168,178 ----
  
      invoke-static {v6, v7}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 952
      :cond_2
      const/4 v5, 0x0
  
+     .line 953
      .local v5, oldest:Ljava/io/File;
      move-object v0, v2
  
***************
*** 167,172 ****
--- 188,194 ----
  
      aget-object v1, v0, v3
  
+     .line 954
      .local v1, f:Ljava/io/File;
      if-eqz v5, :cond_3
  
***************
*** 182,199 ****
--- 204,225 ----
  
      if-lez v6, :cond_4
  
+     .line 955
      :cond_3
      move-object v5, v1
  
+     .line 953
      :cond_4
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_1
  
+     .line 958
      .end local v1           #f:Ljava/io/File;
      :cond_5
      if-eqz v5, :cond_0
  
+     .line 959
      const-string v6, "ThrottleService"
  
      new-instance v7, Ljava/lang/StringBuilder;
***************
*** 216,227 ****
--- 242,256 ----
  
      invoke-static {v6, v7}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 960
      invoke-virtual {v5}, Ljava/io/File;->delete()Z
  
+     .line 961
      invoke-virtual {p1}, Ljava/io/File;->listFiles()[Ljava/io/File;
  
      move-result-object v2
  
+     .line 962
      array-length v6, v2
  
      if-gt v6, v10, :cond_2
***************
*** 233,246 ****
--- 262,278 ----
      .locals 1
  
      .prologue
+     .line 935
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mImsi:Ljava/lang/String;
  
      if-eqz v0, :cond_1
  
+     .line 942
      :cond_0
      :goto_0
      return-void
  
+     .line 937
      :cond_1
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mTelephonyManager:Landroid/telephony/TelephonyManager;
  
***************
*** 250,259 ****
--- 282,293 ----
  
      iput-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mImsi:Ljava/lang/String;
  
+     .line 938
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mImsi:Ljava/lang/String;
  
      if-eqz v0, :cond_0
  
+     .line 941
      invoke-direct {p0}, Lcom/android/server/ThrottleService$DataRecorder;->retrieve()V
  
      goto :goto_0
***************
*** 263,272 ****
--- 297,308 ----
      .locals 7
  
      .prologue
+     .line 915
      invoke-static {}, Landroid/os/Environment;->getDataDirectory()Ljava/io/File;
  
      move-result-object v0
  
+     .line 916
      .local v0, dataDir:Ljava/io/File;
      new-instance v4, Ljava/io/File;
  
***************
*** 274,295 ****
--- 310,336 ----
  
      invoke-direct {v4, v0, v5}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 917
      .local v4, throttleDir:Ljava/io/File;
      invoke-virtual {v4}, Ljava/io/File;->mkdirs()Z
  
+     .line 918
      iget-object v5, p0, Lcom/android/server/ThrottleService$DataRecorder;->mTelephonyManager:Landroid/telephony/TelephonyManager;
  
      invoke-virtual {v5}, Landroid/telephony/TelephonyManager;->getSubscriberId()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 920
      .local v3, mImsi:Ljava/lang/String;
      if-nez v3, :cond_0
  
+     .line 921
      invoke-direct {p0, v4}, Lcom/android/server/ThrottleService$DataRecorder;->useMRUFile(Ljava/io/File;)Ljava/io/File;
  
      move-result-object v1
  
+     .line 928
      .local v1, dataFile:Ljava/io/File;
      :goto_0
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
***************
*** 298,307 ****
--- 339,351 ----
  
      invoke-virtual {v1, v5, v6}, Ljava/io/File;->setLastModified(J)Z
  
+     .line 929
      invoke-direct {p0, v4}, Lcom/android/server/ThrottleService$DataRecorder;->checkAndDeleteLRUDataFile(Ljava/io/File;)V
  
+     .line 930
      return-object v1
  
+     .line 924
      .end local v1           #dataFile:Ljava/io/File;
      :cond_0
      invoke-virtual {v3}, Ljava/lang/String;->hashCode()I
***************
*** 312,317 ****
--- 356,362 ----
  
      move-result-object v2
  
+     .line 925
      .local v2, imsiHash:Ljava/lang/String;
      new-instance v1, Ljava/io/File;
  
***************
*** 325,351 ****
--- 370,402 ----
      .locals 7
  
      .prologue
+     .line 994
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 995
      .local v0, builder:Ljava/lang/StringBuilder;
      const/4 v5, 0x1
  
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 996
      const-string v5, ":"
  
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 997
      iget v5, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 998
      const-string v5, ":"
  
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 999
      const/4 v2, 0x0
  
      .local v2, i:I
***************
*** 354,373 ****
--- 405,428 ----
  
      if-ge v2, v5, :cond_0
  
+     .line 1000
      iget-object v5, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodRxData:[J
  
      aget-wide v5, v5, v2
  
      invoke-virtual {v0, v5, v6}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 1001
      const-string v5, ":"
  
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 999
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 1003
      :cond_0
      const/4 v2, 0x0
  
***************
*** 376,404 ****
--- 431,465 ----
  
      if-ge v2, v5, :cond_1
  
+     .line 1004
      iget-object v5, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodTxData:[J
  
      aget-wide v5, v5, v2
  
      invoke-virtual {v0, v5, v6}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 1005
      const-string v5, ":"
  
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 1003
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_1
  
+     .line 1007
      :cond_1
      iget v5, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
  
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 1008
      const-string v5, ":"
  
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 1009
      iget-object v5, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodStart:Ljava/util/Calendar;
  
      invoke-virtual {v5}, Ljava/util/Calendar;->getTimeInMillis()J
***************
*** 407,416 ****
--- 468,479 ----
  
      invoke-virtual {v0, v5, v6}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 1010
      const-string v5, ":"
  
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 1011
      iget-object v5, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodEnd:Ljava/util/Calendar;
  
      invoke-virtual {v5}, Ljava/util/Calendar;->getTimeInMillis()J
***************
*** 419,426 ****
--- 482,491 ----
  
      invoke-virtual {v0, v5, v6}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 1013
      const/4 v3, 0x0
  
+     .line 1015
      .local v3, out:Ljava/io/BufferedWriter;
      :try_start_0
      new-instance v4, Ljava/io/BufferedWriter;
***************
*** 440,445 ****
--- 505,511 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1016
      .end local v3           #out:Ljava/io/BufferedWriter;
      .local v4, out:Ljava/io/BufferedWriter;
      :try_start_1
***************
*** 452,459 ****
--- 518,527 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_1
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_4
  
+     .line 1021
      if-eqz v4, :cond_2
  
+     .line 1023
      :try_start_2
      invoke-virtual {v4}, Ljava/io/BufferedWriter;->close()V
      :try_end_2
***************
*** 463,477 ****
--- 531,548 ----
      :goto_2
      move-object v3, v4
  
+     .line 1027
      .end local v4           #out:Ljava/io/BufferedWriter;
      .restart local v3       #out:Ljava/io/BufferedWriter;
      :cond_3
      :goto_3
      return-void
  
+     .line 1017
      :catch_0
      move-exception v1
  
+     .line 1018
      .local v1, e:Ljava/io/IOException;
      :goto_4
      :try_start_3
***************
*** 483,490 ****
--- 554,563 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
+     .line 1021
      if-eqz v3, :cond_3
  
+     .line 1023
      :try_start_4
      invoke-virtual {v3}, Ljava/io/BufferedWriter;->close()V
      :try_end_4
***************
*** 492,502 ****
--- 565,577 ----
  
      goto :goto_3
  
+     .line 1024
      :catch_1
      move-exception v5
  
      goto :goto_3
  
+     .line 1021
      .end local v1           #e:Ljava/io/IOException;
      :catchall_0
      move-exception v5
***************
*** 504,518 ****
--- 579,596 ----
      :goto_5
      if-eqz v3, :cond_4
  
+     .line 1023
      :try_start_5
      invoke-virtual {v3}, Ljava/io/BufferedWriter;->close()V
      :try_end_5
      .catch Ljava/lang/Exception; {:try_start_5 .. :try_end_5} :catch_2
  
+     .line 1021
      :cond_4
      :goto_6
      throw v5
  
+     .line 1024
      :catch_2
      move-exception v6
  
***************
*** 525,530 ****
--- 603,609 ----
  
      goto :goto_2
  
+     .line 1021
      :catchall_1
      move-exception v5
  
***************
*** 534,539 ****
--- 613,619 ----
      .restart local v3       #out:Ljava/io/BufferedWriter;
      goto :goto_5
  
+     .line 1017
      .end local v3           #out:Ljava/io/BufferedWriter;
      .restart local v4       #out:Ljava/io/BufferedWriter;
      :catch_4
***************
*** 550,555 ****
--- 630,636 ----
      .locals 21
  
      .prologue
+     .line 1031
      const/16 v18, 0x0
  
      move-object/from16 v0, p0
***************
*** 558,570 ****
--- 639,654 ----
  
      invoke-direct {v0, v1}, Lcom/android/server/ThrottleService$DataRecorder;->zeroData(I)V
  
+     .line 1033
      invoke-direct/range {p0 .. p0}, Lcom/android/server/ThrottleService$DataRecorder;->getDataFile()Ljava/io/File;
  
      move-result-object v6
  
+     .line 1035
      .local v6, f:Ljava/io/File;
      const/16 v16, 0x0
  
+     .line 1037
      .local v16, s:Ljava/io/FileInputStream;
      :try_start_0
      invoke-virtual {v6}, Ljava/io/File;->length()J
***************
*** 581,586 ****
--- 665,671 ----
  
      new-array v2, v0, [B
  
+     .line 1038
      .local v2, buffer:[B
      new-instance v17, Ljava/io/FileInputStream;
  
***************
*** 591,596 ****
--- 676,682 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1039
      .end local v16           #s:Ljava/io/FileInputStream;
      .local v17, s:Ljava/io/FileInputStream;
      :try_start_1
***************
*** 601,619 ****
--- 687,709 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_2
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_6
  
+     .line 1044
      if-eqz v17, :cond_0
  
+     .line 1046
      :try_start_2
      invoke-virtual/range {v17 .. v17}, Ljava/io/FileInputStream;->close()V
      :try_end_2
      .catch Ljava/lang/Exception; {:try_start_2 .. :try_end_2} :catch_4
  
+     .line 1050
      :cond_0
      :goto_0
      new-instance v4, Ljava/lang/String;
  
      invoke-direct {v4, v2}, Ljava/lang/String;-><init>([B)V
  
+     .line 1051
      .local v4, data:Ljava/lang/String;
      if-eqz v4, :cond_1
  
***************
*** 623,628 ****
--- 713,719 ----
  
      if-nez v18, :cond_4
  
+     .line 1052
      :cond_1
      const-string v18, "ThrottleService"
  
***************
*** 632,637 ****
--- 723,729 ----
  
      move-object/from16 v16, v17
  
+     .line 1107
      .end local v2           #buffer:[B
      .end local v4           #data:Ljava/lang/String;
      .end local v17           #s:Ljava/io/FileInputStream;
***************
*** 640,648 ****
--- 732,742 ----
      :goto_1
      return-void
  
+     .line 1040
      :catch_0
      move-exception v5
  
+     .line 1041
      .local v5, e:Ljava/io/IOException;
      :goto_2
      :try_start_3
***************
*** 654,661 ****
--- 748,757 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
+     .line 1044
      if-eqz v16, :cond_2
  
+     .line 1046
      :try_start_4
      invoke-virtual/range {v16 .. v16}, Ljava/io/FileInputStream;->close()V
      :try_end_4
***************
*** 663,673 ****
--- 759,771 ----
  
      goto :goto_1
  
+     .line 1047
      :catch_1
      move-exception v18
  
      goto :goto_1
  
+     .line 1044
      .end local v5           #e:Ljava/io/IOException;
      :catchall_0
      move-exception v18
***************
*** 675,689 ****
--- 773,790 ----
      :goto_3
      if-eqz v16, :cond_3
  
+     .line 1046
      :try_start_5
      invoke-virtual/range {v16 .. v16}, Ljava/io/FileInputStream;->close()V
      :try_end_5
      .catch Ljava/lang/Exception; {:try_start_5 .. :try_end_5} :catch_3
  
+     .line 1044
      :cond_3
      :goto_4
      throw v18
  
+     .line 1055
      .end local v16           #s:Ljava/io/FileInputStream;
      .restart local v2       #buffer:[B
      .restart local v4       #data:Ljava/lang/String;
***************
*** 697,705 ****
--- 798,808 ----
  
      move-result-object v8
  
+     .line 1056
      .local v8, parsed:[Ljava/lang/String;
      const/4 v9, 0x0
  
+     .line 1057
      .local v9, parsedUsed:I
      array-length v0, v8
  
***************
*** 713,718 ****
--- 816,822 ----
  
      if-ge v0, v1, :cond_5
  
+     .line 1058
      const-string v18, "ThrottleService"
  
      const-string v19, "reading data file with insufficient length - ignoring"
***************
*** 721,730 ****
--- 825,836 ----
  
      move-object/from16 v16, v17
  
+     .line 1059
      .end local v17           #s:Ljava/io/FileInputStream;
      .restart local v16       #s:Ljava/io/FileInputStream;
      goto :goto_1
  
+     .line 1069
      .end local v16           #s:Ljava/io/FileInputStream;
      .restart local v17       #s:Ljava/io/FileInputStream;
      :cond_5
***************
*** 747,752 ****
--- 853,859 ----
  
      if-eq v0, v1, :cond_6
  
+     .line 1070
      const-string v18, "ThrottleService"
  
      const-string v19, "reading data file with bad version - ignoring"
***************
*** 757,766 ****
--- 864,875 ----
  
      move-object/from16 v16, v17
  
+     .line 1071
      .end local v17           #s:Ljava/io/FileInputStream;
      .restart local v16       #s:Ljava/io/FileInputStream;
      goto :goto_1
  
+     .line 1074
      .end local v16           #s:Ljava/io/FileInputStream;
      .restart local v17       #s:Ljava/io/FileInputStream;
      :cond_6
***************
*** 775,780 ****
--- 884,890 ----
  
      move-result v11
  
+     .line 1075
      .local v11, periodCount:I
      array-length v0, v8
  
***************
*** 790,795 ****
--- 900,906 ----
  
      if-eq v0, v1, :cond_7
  
+     .line 1076
      const-string v18, "ThrottleService"
  
      new-instance v19, Ljava/lang/StringBuilder;
***************
*** 838,852 ****
--- 949,966 ----
  
      move-object/from16 v16, v17
  
+     .line 1078
      .end local v17           #s:Ljava/io/FileInputStream;
      .restart local v16       #s:Ljava/io/FileInputStream;
      goto/16 :goto_1
  
+     .line 1080
      .end local v16           #s:Ljava/io/FileInputStream;
      .restart local v17       #s:Ljava/io/FileInputStream;
      :cond_7
      new-array v13, v11, [J
  
+     .line 1081
      .local v13, periodRxData:[J
      const/4 v7, 0x0
  
***************
*** 858,863 ****
--- 972,978 ----
      :goto_5
      if-ge v7, v11, :cond_8
  
+     .line 1082
      add-int/lit8 v9, v10, 0x1
  
      .end local v10           #parsedUsed:I
***************
*** 872,877 ****
--- 987,993 ----
      :try_end_7
      .catch Ljava/lang/Exception; {:try_start_7 .. :try_end_7} :catch_5
  
+     .line 1081
      add-int/lit8 v7, v7, 0x1
  
      move v10, v9
***************
*** 880,897 ****
--- 996,1016 ----
      .restart local v10       #parsedUsed:I
      goto :goto_5
  
+     .line 1084
      :cond_8
      :try_start_8
      new-array v15, v11, [J
      :try_end_8
      .catch Ljava/lang/Exception; {:try_start_8 .. :try_end_8} :catch_2
  
+     .line 1085
      .local v15, periodTxData:[J
      const/4 v7, 0x0
  
      :goto_6
      if-ge v7, v11, :cond_9
  
+     .line 1086
      add-int/lit8 v9, v10, 0x1
  
      .end local v10           #parsedUsed:I
***************
*** 905,910 ****
--- 1024,1030 ----
  
      aput-wide v18, v15, v7
  
+     .line 1085
      add-int/lit8 v7, v7, 0x1
  
      move v10, v9
***************
*** 913,918 ****
--- 1033,1039 ----
      .restart local v10       #parsedUsed:I
      goto :goto_6
  
+     .line 1089
      :cond_9
      add-int/lit8 v9, v10, 0x1
  
***************
*** 924,929 ****
--- 1045,1051 ----
  
      move-result v3
  
+     .line 1091
      .local v3, currentPeriod:I
      new-instance v14, Ljava/util/GregorianCalendar;
  
***************
*** 931,936 ****
--- 1053,1059 ----
      :try_end_9
      .catch Ljava/lang/Exception; {:try_start_9 .. :try_end_9} :catch_5
  
+     .line 1092
      .local v14, periodStart:Ljava/util/Calendar;
      add-int/lit8 v10, v9, 0x1
  
***************
*** 947,958 ****
--- 1070,1083 ----
  
      invoke-virtual {v14, v0, v1}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 1093
      new-instance v12, Ljava/util/GregorianCalendar;
  
      invoke-direct {v12}, Ljava/util/GregorianCalendar;-><init>()V
      :try_end_a
      .catch Ljava/lang/Exception; {:try_start_a .. :try_end_a} :catch_2
  
+     .line 1094
      .local v12, periodEnd:Ljava/util/Calendar;
      add-int/lit8 v9, v10, 0x1
  
***************
*** 971,976 ****
--- 1096,1102 ----
      :try_end_b
      .catch Ljava/lang/Exception; {:try_start_b .. :try_end_b} :catch_5
  
+     .line 1099
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
***************
*** 979,1019 ****
--- 1105,1154 ----
  
      monitor-enter v19
  
+     .line 1100
      :try_start_c
      move-object/from16 v0, p0
  
      iput v11, v0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
+     .line 1101
      move-object/from16 v0, p0
  
      iput-object v13, v0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodRxData:[J
  
+     .line 1102
      move-object/from16 v0, p0
  
      iput-object v15, v0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodTxData:[J
  
+     .line 1103
      move-object/from16 v0, p0
  
      iput v3, v0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
  
+     .line 1104
      move-object/from16 v0, p0
  
      iput-object v14, v0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodStart:Ljava/util/Calendar;
  
+     .line 1105
      move-object/from16 v0, p0
  
      iput-object v12, v0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodEnd:Ljava/util/Calendar;
  
+     .line 1106
      monitor-exit v19
      :try_end_c
      .catchall {:try_start_c .. :try_end_c} :catchall_1
  
      move-object/from16 v16, v17
  
+     .line 1107
      .end local v17           #s:Ljava/io/FileInputStream;
      .restart local v16       #s:Ljava/io/FileInputStream;
      goto/16 :goto_1
  
+     .line 1095
      .end local v3           #currentPeriod:I
      .end local v7           #i:I
      .end local v9           #parsedUsed:I
***************
*** 1030,1035 ****
--- 1165,1171 ----
  
      move v9, v10
  
+     .line 1096
      .end local v10           #parsedUsed:I
      .local v5, e:Ljava/lang/Exception;
      .restart local v9       #parsedUsed:I
***************
*** 1042,1051 ****
--- 1178,1189 ----
  
      move-object/from16 v16, v17
  
+     .line 1097
      .end local v17           #s:Ljava/io/FileInputStream;
      .restart local v16       #s:Ljava/io/FileInputStream;
      goto/16 :goto_1
  
+     .line 1106
      .end local v5           #e:Ljava/lang/Exception;
      .end local v16           #s:Ljava/io/FileInputStream;
      .restart local v3       #currentPeriod:I
***************
*** 1066,1071 ****
--- 1204,1210 ----
  
      throw v18
  
+     .line 1047
      .end local v2           #buffer:[B
      .end local v3           #currentPeriod:I
      .end local v4           #data:Ljava/lang/String;
***************
*** 1092,1097 ****
--- 1231,1237 ----
  
      goto/16 :goto_0
  
+     .line 1095
      .restart local v4       #data:Ljava/lang/String;
      .restart local v8       #parsed:[Ljava/lang/String;
      .restart local v9       #parsedUsed:I
***************
*** 1100,1105 ****
--- 1240,1246 ----
  
      goto :goto_7
  
+     .line 1044
      .end local v4           #data:Ljava/lang/String;
      .end local v8           #parsed:[Ljava/lang/String;
      .end local v9           #parsedUsed:I
***************
*** 1112,1117 ****
--- 1253,1259 ----
      .restart local v16       #s:Ljava/io/FileInputStream;
      goto/16 :goto_3
  
+     .line 1040
      .end local v16           #s:Ljava/io/FileInputStream;
      .restart local v17       #s:Ljava/io/FileInputStream;
      :catch_6
***************
*** 1129,1145 ****
--- 1271,1292 ----
      .parameter "end"
  
      .prologue
+     .line 868
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v1
  
+     .line 869
      :try_start_0
      iput-object p1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodEnd:Ljava/util/Calendar;
  
+     .line 870
      monitor-exit v1
  
+     .line 871
      return-void
  
+     .line 870
      :catchall_0
      move-exception v0
  
***************
*** 1155,1171 ****
--- 1302,1323 ----
      .parameter "start"
  
      .prologue
+     .line 880
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v1
  
+     .line 881
      :try_start_0
      iput-object p1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodStart:Ljava/util/Calendar;
  
+     .line 882
      monitor-exit v1
  
+     .line 883
      return-void
  
+     .line 882
      :catchall_0
      move-exception v0
  
***************
*** 1181,1196 ****
--- 1333,1352 ----
      .parameter "dir"
  
      .prologue
+     .line 966
      const/4 v5, 0x0
  
+     .line 967
      .local v5, newest:Ljava/io/File;
      invoke-virtual {p1}, Ljava/io/File;->listFiles()[Ljava/io/File;
  
      move-result-object v2
  
+     .line 969
      .local v2, files:[Ljava/io/File;
      if-eqz v2, :cond_2
  
+     .line 970
      move-object v0, v2
  
      .local v0, arr$:[Ljava/io/File;
***************
*** 1205,1210 ****
--- 1361,1367 ----
  
      aget-object v1, v0, v3
  
+     .line 971
      .local v1, f:Ljava/io/File;
      if-eqz v5, :cond_0
  
***************
*** 1220,1233 ****
--- 1377,1393 ----
  
      if-gez v6, :cond_1
  
+     .line 972
      :cond_0
      move-object v5, v1
  
+     .line 970
      :cond_1
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 976
      .end local v0           #arr$:[Ljava/io/File;
      .end local v1           #f:Ljava/io/File;
      .end local v3           #i$:I
***************
*** 1235,1240 ****
--- 1395,1401 ----
      :cond_2
      if-nez v5, :cond_3
  
+     .line 977
      new-instance v5, Ljava/io/File;
  
      .end local v5           #newest:Ljava/io/File;
***************
*** 1242,1247 ****
--- 1403,1409 ----
  
      invoke-direct {v5, p1, v6}, Ljava/io/File;-><init>(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 979
      .restart local v5       #newest:Ljava/io/File;
      :cond_3
      return-object v5
***************
*** 1252,1261 ****
--- 1414,1425 ----
      .parameter "field"
  
      .prologue
+     .line 892
      iget-object v2, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v2
  
+     .line 893
      const/4 v0, 0x0
  
      .local v0, period:I
***************
*** 1265,1295 ****
--- 1429,1466 ----
  
      if-ge v0, v1, :cond_0
  
+     .line 894
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodRxData:[J
  
      const-wide/16 v3, 0x0
  
      aput-wide v3, v1, v0
  
+     .line 895
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodTxData:[J
  
      const-wide/16 v3, 0x0
  
      aput-wide v3, v1, v0
  
+     .line 893
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 897
      :cond_0
      const/4 v1, 0x0
  
      iput v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
  
+     .line 898
      monitor-exit v2
  
+     .line 900
      return-void
  
+     .line 898
      :catchall_0
      move-exception v1
  
***************
*** 1308,1319 ****
--- 1479,1493 ----
      .parameter "bytesWritten"
  
      .prologue
+     .line 905
      invoke-direct {p0}, Lcom/android/server/ThrottleService$DataRecorder;->checkForSubscriberId()V
  
+     .line 907
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v1
  
+     .line 908
      :try_start_0
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodRxData:[J
  
***************
*** 1325,1330 ****
--- 1499,1505 ----
  
      aput-wide v3, v0, v2
  
+     .line 909
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodTxData:[J
  
      iget v2, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
***************
*** 1335,1348 ****
--- 1510,1527 ----
  
      aput-wide v3, v0, v2
  
+     .line 910
      monitor-exit v1
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 911
      invoke-direct {p0}, Lcom/android/server/ThrottleService$DataRecorder;->record()V
  
+     .line 912
      return-void
  
+     .line 910
      :catchall_0
      move-exception v0
  
***************
*** 1358,1367 ****
--- 1537,1548 ----
      .locals 2
  
      .prologue
+     .line 886
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v1
  
+     .line 887
      :try_start_0
      iget v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
***************
*** 1369,1374 ****
--- 1550,1556 ----
  
      return v0
  
+     .line 888
      :catchall_0
      move-exception v0
  
***************
*** 1383,1392 ****
--- 1565,1576 ----
      .locals 4
  
      .prologue
+     .line 862
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v1
  
+     .line 863
      :try_start_0
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodEnd:Ljava/util/Calendar;
  
***************
*** 1398,1403 ****
--- 1582,1588 ----
  
      return-wide v2
  
+     .line 864
      :catchall_0
      move-exception v0
  
***************
*** 1413,1422 ****
--- 1598,1609 ----
      .parameter "which"
  
      .prologue
+     .line 1110
      iget-object v2, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v2
  
+     .line 1111
      :try_start_0
      iget v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
***************
*** 1426,1445 ****
--- 1613,1636 ----
  
      monitor-exit v2
  
+     .line 1114
      :goto_0
      return-wide v0
  
+     .line 1112
      :cond_0
      iget v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
  
      sub-int p1, v0, p1
  
+     .line 1113
      if-gez p1, :cond_1
  
      iget v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
      add-int/2addr p1, v0
  
+     .line 1114
      :cond_1
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodRxData:[J
  
***************
*** 1449,1454 ****
--- 1640,1646 ----
  
      goto :goto_0
  
+     .line 1115
      :catchall_0
      move-exception v0
  
***************
*** 1463,1472 ****
--- 1655,1666 ----
      .locals 4
  
      .prologue
+     .line 874
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v1
  
+     .line 875
      :try_start_0
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodStart:Ljava/util/Calendar;
  
***************
*** 1478,1483 ****
--- 1672,1678 ----
  
      return-wide v2
  
+     .line 876
      :catchall_0
      move-exception v0
  
***************
*** 1493,1502 ****
--- 1688,1699 ----
      .parameter "which"
  
      .prologue
+     .line 1118
      iget-object v2, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v2
  
+     .line 1119
      :try_start_0
      iget v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
***************
*** 1506,1525 ****
--- 1703,1726 ----
  
      monitor-exit v2
  
+     .line 1122
      :goto_0
      return-wide v0
  
+     .line 1120
      :cond_0
      iget v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
  
      sub-int p1, v0, p1
  
+     .line 1121
      if-gez p1, :cond_1
  
      iget v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
  
      add-int/2addr p1, v0
  
+     .line 1122
      :cond_1
      iget-object v0, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodTxData:[J
  
***************
*** 1529,1534 ****
--- 1730,1736 ----
  
      goto :goto_0
  
+     .line 1123
      :catchall_0
      move-exception v0
  
***************
*** 1545,1554 ****
--- 1747,1759 ----
      .parameter "end"
  
      .prologue
+     .line 825
      invoke-direct {p0}, Lcom/android/server/ThrottleService$DataRecorder;->checkForSubscriberId()V
  
+     .line 826
      const/4 v0, 0x1
  
+     .line 828
      .local v0, startNewPeriod:Z
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodStart:Ljava/util/Calendar;
  
***************
*** 1566,1587 ****
--- 1771,1799 ----
  
      if-eqz v1, :cond_0
  
+     .line 834
      const/4 v0, 0x0
  
+     .line 855
      :goto_0
      invoke-direct {p0, p1}, Lcom/android/server/ThrottleService$DataRecorder;->setPeriodStart(Ljava/util/Calendar;)V
  
+     .line 856
      invoke-direct {p0, p2}, Lcom/android/server/ThrottleService$DataRecorder;->setPeriodEnd(Ljava/util/Calendar;)V
  
+     .line 857
      invoke-direct {p0}, Lcom/android/server/ThrottleService$DataRecorder;->record()V
  
+     .line 858
      return v0
  
+     .line 848
      :cond_0
      iget-object v2, p0, Lcom/android/server/ThrottleService$DataRecorder;->mParent:Lcom/android/server/ThrottleService;
  
      monitor-enter v2
  
+     .line 849
      :try_start_0
      iget v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
  
***************
*** 1589,1594 ****
--- 1801,1807 ----
  
      iput v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
  
+     .line 850
      iget v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
  
      iget v3, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodCount:I
***************
*** 1599,1604 ****
--- 1812,1818 ----
  
      iput v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
  
+     .line 851
      :cond_1
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodRxData:[J
  
***************
*** 1608,1613 ****
--- 1822,1828 ----
  
      aput-wide v4, v1, v3
  
+     .line 852
      iget-object v1, p0, Lcom/android/server/ThrottleService$DataRecorder;->mPeriodTxData:[J
  
      iget v3, p0, Lcom/android/server/ThrottleService$DataRecorder;->mCurrentPeriod:I
***************
*** 1616,1621 ****
--- 1831,1837 ----
  
      aput-wide v4, v1, v3
  
+     .line 853
      monitor-exit v2
  
      goto :goto_0
