*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/UiModeManagerService$4.smali	2012-03-23 19:35:09.033751616 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/UiModeManagerService$4.smali.original	2012-03-23 19:34:47.318355636 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 225
      iput-object p1, p0, Lcom/android/server/UiModeManagerService$4;->this$0:Lcom/android/server/UiModeManagerService;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 41,46 ****
--- 42,48 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 228
      iget-object v1, p0, Lcom/android/server/UiModeManagerService$4;->this$0:Lcom/android/server/UiModeManagerService;
  
      const-string v2, "plugged"
***************
*** 57,68 ****
--- 59,72 ----
      #setter for: Lcom/android/server/UiModeManagerService;->mCharging:Z
      invoke-static {v1, v0}, Lcom/android/server/UiModeManagerService;->access$502(Lcom/android/server/UiModeManagerService;Z)Z
  
+     .line 229
      iget-object v0, p0, Lcom/android/server/UiModeManagerService$4;->this$0:Lcom/android/server/UiModeManagerService;
  
      iget-object v1, v0, Lcom/android/server/UiModeManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 230
      :try_start_0
      iget-object v0, p0, Lcom/android/server/UiModeManagerService$4;->this$0:Lcom/android/server/UiModeManagerService;
  
***************
*** 73,78 ****
--- 77,83 ----
  
      if-eqz v0, :cond_1
  
+     .line 231
      iget-object v0, p0, Lcom/android/server/UiModeManagerService$4;->this$0:Lcom/android/server/UiModeManagerService;
  
      const/4 v2, 0x0
***************
*** 81,91 ****
--- 86,99 ----
  
      invoke-virtual {v0, v2, v3}, Lcom/android/server/UiModeManagerService;->updateLocked(II)V
  
+     .line 233
      :cond_1
      monitor-exit v1
  
+     .line 234
      return-void
  
+     .line 233
      :catchall_0
      move-exception v0
  
