*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/UiModeManagerService$5.smali	2012-03-23 19:35:07.298719977 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/UiModeManagerService$5.smali.original	2012-03-23 19:34:47.129352190 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 237
      iput-object p1, p0, Lcom/android/server/UiModeManagerService$5;->this$0:Lcom/android/server/UiModeManagerService;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 41,46 ****
--- 42,48 ----
      .prologue
      const/4 v2, 0x2
  
+     .line 240
      const-string v0, "android.intent.action.AIRPLANE_MODE"
  
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
***************
*** 53,58 ****
--- 55,61 ----
  
      if-eqz v0, :cond_1
  
+     .line 241
      const-string v0, "state"
  
      const/4 v1, 0x0
***************
*** 63,68 ****
--- 66,72 ----
  
      if-nez v0, :cond_0
  
+     .line 243
      iget-object v0, p0, Lcom/android/server/UiModeManagerService$5;->this$0:Lcom/android/server/UiModeManagerService;
  
      #getter for: Lcom/android/server/UiModeManagerService;->mHandler:Landroid/os/Handler;
***************
*** 72,81 ****
--- 76,87 ----
  
      invoke-virtual {v0, v2}, Landroid/os/Handler;->sendEmptyMessage(I)Z
  
+     .line 249
      :cond_0
      :goto_0
      return-void
  
+     .line 247
      :cond_1
      iget-object v0, p0, Lcom/android/server/UiModeManagerService$5;->this$0:Lcom/android/server/UiModeManagerService;
  
