*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/UiModeManagerService.smali	2012-03-23 19:34:55.261500480 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/UiModeManagerService.smali.original	2012-03-23 19:34:46.879347632 -0500
***************
*** 100,105 ****
--- 100,106 ----
      .locals 1
  
      .prologue
+     .line 61
      const-class v0, Landroid/app/UiModeManager;
  
      invoke-virtual {v0}, Ljava/lang/Class;->getSimpleName()Ljava/lang/String;
***************
*** 120,209 ****
--- 121,232 ----
  
      const/4 v4, 0x0
  
+     .line 326
      invoke-direct {p0}, Landroid/app/IUiModeManager$Stub;-><init>()V
  
+     .line 85
      new-instance v2, Ljava/lang/Object;
  
      invoke-direct {v2}, Ljava/lang/Object;-><init>()V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mLock:Ljava/lang/Object;
  
+     .line 87
      iput v4, p0, Lcom/android/server/UiModeManagerService;->mDockState:I
  
+     .line 88
      iput v4, p0, Lcom/android/server/UiModeManagerService;->mLastBroadcastState:I
  
+     .line 90
      iput v3, p0, Lcom/android/server/UiModeManagerService;->mNightMode:I
  
+     .line 91
      iput-boolean v4, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
+     .line 92
      iput-boolean v4, p0, Lcom/android/server/UiModeManagerService;->mCharging:Z
  
+     .line 97
      iput v4, p0, Lcom/android/server/UiModeManagerService;->mCurUiMode:I
  
+     .line 98
      iput v4, p0, Lcom/android/server/UiModeManagerService;->mSetUiMode:I
  
+     .line 100
      iput-boolean v4, p0, Lcom/android/server/UiModeManagerService;->mHoldingConfiguration:Z
  
+     .line 101
      new-instance v2, Landroid/content/res/Configuration;
  
      invoke-direct {v2}, Landroid/content/res/Configuration;-><init>()V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mConfiguration:Landroid/content/res/Configuration;
  
+     .line 126
      new-instance v2, Lcom/android/server/UiModeManagerService$1;
  
      invoke-direct {v2, p0}, Lcom/android/server/UiModeManagerService$1;-><init>(Lcom/android/server/UiModeManagerService;)V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mResultReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 207
      new-instance v2, Lcom/android/server/UiModeManagerService$2;
  
      invoke-direct {v2, p0}, Lcom/android/server/UiModeManagerService$2;-><init>(Lcom/android/server/UiModeManagerService;)V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mTwilightUpdateReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 216
      new-instance v2, Lcom/android/server/UiModeManagerService$3;
  
      invoke-direct {v2, p0}, Lcom/android/server/UiModeManagerService$3;-><init>(Lcom/android/server/UiModeManagerService;)V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mDockModeReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 225
      new-instance v2, Lcom/android/server/UiModeManagerService$4;
  
      invoke-direct {v2, p0}, Lcom/android/server/UiModeManagerService$4;-><init>(Lcom/android/server/UiModeManagerService;)V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mBatteryReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 237
      new-instance v2, Lcom/android/server/UiModeManagerService$5;
  
      invoke-direct {v2, p0}, Lcom/android/server/UiModeManagerService$5;-><init>(Lcom/android/server/UiModeManagerService;)V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mUpdateLocationReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 254
      new-instance v2, Lcom/android/server/UiModeManagerService$6;
  
      invoke-direct {v2, p0}, Lcom/android/server/UiModeManagerService$6;-><init>(Lcom/android/server/UiModeManagerService;)V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mEmptyLocationListener:Landroid/location/LocationListener;
  
+     .line 268
      new-instance v2, Lcom/android/server/UiModeManagerService$7;
  
      invoke-direct {v2, p0}, Lcom/android/server/UiModeManagerService$7;-><init>(Lcom/android/server/UiModeManagerService;)V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mLocationListener:Landroid/location/LocationListener;
  
+     .line 647
      new-instance v2, Lcom/android/server/UiModeManagerService$8;
  
      invoke-direct {v2, p0}, Lcom/android/server/UiModeManagerService$8;-><init>(Lcom/android/server/UiModeManagerService;)V
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mHandler:Landroid/os/Handler;
  
+     .line 327
      iput-object p1, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
+     .line 329
      const-string v2, "uimode"
  
      invoke-static {v2, p0}, Landroid/os/ServiceManager;->addService(Ljava/lang/String;Landroid/os/IBinder;)V
  
+     .line 331
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      const-string v5, "alarm"
***************
*** 216,221 ****
--- 239,245 ----
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mAlarmManager:Landroid/app/AlarmManager;
  
+     .line 333
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      const-string v5, "location"
***************
*** 228,233 ****
--- 252,258 ----
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mLocationManager:Landroid/location/LocationManager;
  
+     .line 335
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      iget-object v5, p0, Lcom/android/server/UiModeManagerService;->mTwilightUpdateReceiver:Landroid/content/BroadcastReceiver;
***************
*** 240,245 ****
--- 265,271 ----
  
      invoke-virtual {v2, v5, v6}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 337
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      iget-object v5, p0, Lcom/android/server/UiModeManagerService;->mDockModeReceiver:Landroid/content/BroadcastReceiver;
***************
*** 252,257 ****
--- 278,284 ----
  
      invoke-virtual {v2, v5, v6}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 339
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      iget-object v5, p0, Lcom/android/server/UiModeManagerService;->mBatteryReceiver:Landroid/content/BroadcastReceiver;
***************
*** 264,286 ****
--- 291,317 ----
  
      invoke-virtual {v2, v5, v6}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 341
      new-instance v0, Landroid/content/IntentFilter;
  
      const-string v2, "android.intent.action.AIRPLANE_MODE"
  
      invoke-direct {v0, v2}, Landroid/content/IntentFilter;-><init>(Ljava/lang/String;)V
  
+     .line 342
      .local v0, filter:Landroid/content/IntentFilter;
      const-string v2, "android.intent.action.TIMEZONE_CHANGED"
  
      invoke-virtual {v0, v2}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 343
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      iget-object v5, p0, Lcom/android/server/UiModeManagerService;->mUpdateLocationReceiver:Landroid/content/BroadcastReceiver;
  
      invoke-virtual {v2, v5, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 345
      const-string v2, "power"
  
      invoke-virtual {p1, v2}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 289,294 ****
--- 320,326 ----
  
      check-cast v1, Landroid/os/PowerManager;
  
+     .line 346
      .local v1, powerManager:Landroid/os/PowerManager;
      const/16 v2, 0x1a
  
***************
*** 300,309 ****
--- 332,343 ----
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mWakeLock:Landroid/os/PowerManager$WakeLock;
  
+     .line 348
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mConfiguration:Landroid/content/res/Configuration;
  
      invoke-virtual {v2}, Landroid/content/res/Configuration;->setToDefaults()V
  
+     .line 350
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v2
***************
*** 321,326 ****
--- 355,361 ----
      :goto_0
      iput-boolean v2, p0, Lcom/android/server/UiModeManagerService;->mCarModeKeepsScreenOn:Z
  
+     .line 352
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v2
***************
*** 336,341 ****
--- 371,377 ----
      :goto_1
      iput-boolean v3, p0, Lcom/android/server/UiModeManagerService;->mDeskModeKeepsScreenOn:Z
  
+     .line 355
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v2}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
***************
*** 350,365 ****
--- 386,404 ----
  
      iput v2, p0, Lcom/android/server/UiModeManagerService;->mNightMode:I
  
+     .line 357
      return-void
  
      :cond_0
      move v2, v4
  
+     .line 350
      goto :goto_0
  
      :cond_1
      move v3, v4
  
+     .line 352
      goto :goto_1
  .end method
  
***************
*** 368,373 ****
--- 407,413 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-boolean v0, p0, Lcom/android/server/UiModeManagerService;->mHoldingConfiguration:Z
  
      return v0
***************
*** 379,384 ****
--- 419,425 ----
      .parameter "x1"
  
      .prologue
+     .line 60
      iput-boolean p1, p0, Lcom/android/server/UiModeManagerService;->mHoldingConfiguration:Z
  
      return p1
***************
*** 389,394 ****
--- 430,436 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mConfiguration:Landroid/content/res/Configuration;
  
      return-object v0
***************
*** 399,404 ****
--- 441,447 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mLocationListener:Landroid/location/LocationListener;
  
      return-object v0
***************
*** 408,413 ****
--- 451,457 ----
      .locals 1
  
      .prologue
+     .line 60
      sget-object v0, Lcom/android/server/UiModeManagerService;->TAG:Ljava/lang/String;
  
      return-object v0
***************
*** 418,423 ****
--- 462,468 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget v0, p0, Lcom/android/server/UiModeManagerService;->mNightMode:I
  
      return v0
***************
*** 428,433 ****
--- 473,479 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mHandler:Landroid/os/Handler;
  
      return-object v0
***************
*** 439,444 ****
--- 485,491 ----
      .parameter "x1"
  
      .prologue
+     .line 60
      iput-boolean p1, p0, Lcom/android/server/UiModeManagerService;->mCharging:Z
  
      return p1
***************
*** 449,454 ****
--- 496,502 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-boolean v0, p0, Lcom/android/server/UiModeManagerService;->mSystemReady:Z
  
      return v0
***************
*** 459,464 ****
--- 507,513 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mLocation:Landroid/location/Location;
  
      return-object v0
***************
*** 470,475 ****
--- 519,525 ----
      .parameter "x1"
  
      .prologue
+     .line 60
      iput-object p1, p0, Lcom/android/server/UiModeManagerService;->mLocation:Landroid/location/Location;
  
      return-object p1
***************
*** 480,485 ****
--- 530,536 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mEmptyLocationListener:Landroid/location/LocationListener;
  
      return-object v0
***************
*** 490,495 ****
--- 541,547 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mLocationManager:Landroid/location/LocationManager;
  
      return-object v0
***************
*** 501,510 ****
--- 553,564 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 606
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mStatusBarManager:Landroid/app/StatusBarManager;
  
      if-nez v2, :cond_0
  
+     .line 607
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      const-string v4, "statusbar"
***************
*** 517,527 ****
--- 571,583 ----
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mStatusBarManager:Landroid/app/StatusBarManager;
  
+     .line 615
      :cond_0
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mStatusBarManager:Landroid/app/StatusBarManager;
  
      if-eqz v2, :cond_1
  
+     .line 616
      iget-object v4, p0, Lcom/android/server/UiModeManagerService;->mStatusBarManager:Landroid/app/StatusBarManager;
  
      iget-boolean v2, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
***************
*** 533,543 ****
--- 589,601 ----
      :goto_0
      invoke-virtual {v4, v2}, Landroid/app/StatusBarManager;->disable(I)V
  
+     .line 621
      :cond_1
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mNotificationManager:Landroid/app/NotificationManager;
  
      if-nez v2, :cond_2
  
+     .line 622
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      const-string v4, "notification"
***************
*** 550,564 ****
--- 608,625 ----
  
      iput-object v2, p0, Lcom/android/server/UiModeManagerService;->mNotificationManager:Landroid/app/NotificationManager;
  
+     .line 626
      :cond_2
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mNotificationManager:Landroid/app/NotificationManager;
  
      if-eqz v2, :cond_3
  
+     .line 627
      iget-boolean v2, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
      if-eqz v2, :cond_5
  
+     .line 628
      new-instance v0, Landroid/content/Intent;
  
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
***************
*** 567,594 ****
--- 628,661 ----
  
      invoke-direct {v0, v2, v4}, Landroid/content/Intent;-><init>(Landroid/content/Context;Ljava/lang/Class;)V
  
+     .line 630
      .local v0, carModeOffIntent:Landroid/content/Intent;
      new-instance v1, Landroid/app/Notification;
  
      invoke-direct {v1}, Landroid/app/Notification;-><init>()V
  
+     .line 631
      .local v1, n:Landroid/app/Notification;
      const v2, 0x10804ec
  
      iput v2, v1, Landroid/app/Notification;->icon:I
  
+     .line 632
      const/4 v2, 0x4
  
      iput v2, v1, Landroid/app/Notification;->defaults:I
  
+     .line 633
      const/4 v2, 0x2
  
      iput v2, v1, Landroid/app/Notification;->flags:I
  
+     .line 634
      const-wide/16 v4, 0x0
  
      iput-wide v4, v1, Landroid/app/Notification;->when:J
  
+     .line 635
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      iget-object v4, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
***************
*** 615,624 ****
--- 682,693 ----
  
      invoke-virtual {v1, v2, v4, v5, v6}, Landroid/app/Notification;->setLatestEventInfo(Landroid/content/Context;Ljava/lang/CharSequence;Ljava/lang/CharSequence;Landroid/app/PendingIntent;)V
  
+     .line 640
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mNotificationManager:Landroid/app/NotificationManager;
  
      invoke-virtual {v2, v3, v1}, Landroid/app/NotificationManager;->notify(ILandroid/app/Notification;)V
  
+     .line 645
      .end local v0           #carModeOffIntent:Landroid/content/Intent;
      .end local v1           #n:Landroid/app/Notification;
      :cond_3
***************
*** 628,635 ****
--- 697,706 ----
      :cond_4
      move v2, v3
  
+     .line 616
      goto :goto_0
  
+     .line 642
      :cond_5
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mNotificationManager:Landroid/app/NotificationManager;
  
***************
*** 643,661 ****
--- 714,736 ----
      .parameter "category"
  
      .prologue
+     .line 115
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.intent.action.MAIN"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 116
      .local v0, intent:Landroid/content/Intent;
      invoke-virtual {v0, p0}, Landroid/content/Intent;->addCategory(Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 117
      const/high16 v1, 0x1020
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 119
      return-object v0
  .end method
  
***************
*** 664,682 ****
--- 739,761 ----
      .parameter "state"
  
      .prologue
+     .line 444
      packed-switch p0, :pswitch_data_0
  
+     .line 450
      :pswitch_0
      const/4 v0, 0x0
  
      :goto_0
      return v0
  
+     .line 448
      :pswitch_1
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 444
      nop
  
      :pswitch_data_0
***************
*** 695,722 ****
--- 774,808 ----
      .parameter "flags"
  
      .prologue
+     .line 360
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 361
      const/4 v0, 0x0
  
      :try_start_0
      invoke-virtual {p0, v0}, Lcom/android/server/UiModeManagerService;->setCarModeLocked(Z)V
  
+     .line 362
      iget-boolean v0, p0, Lcom/android/server/UiModeManagerService;->mSystemReady:Z
  
      if-eqz v0, :cond_0
  
+     .line 363
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0, p1}, Lcom/android/server/UiModeManagerService;->updateLocked(II)V
  
+     .line 365
      :cond_0
      monitor-exit v1
  
+     .line 366
      return-void
  
+     .line 365
      :catchall_0
      move-exception v0
  
***************
*** 734,739 ****
--- 820,826 ----
      .parameter "args"
  
      .prologue
+     .line 834
      iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      const-string v1, "android.permission.DUMP"
***************
*** 744,749 ****
--- 831,837 ----
  
      if-eqz v0, :cond_0
  
+     .line 837
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 782,800 ****
--- 870,892 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 858
      :goto_0
      return-void
  
+     .line 843
      :cond_0
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 844
      :try_start_0
      const-string v0, "Current UI Mode Service state:"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 845
      const-string v0, "  mDockState="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 803,808 ****
--- 895,901 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 846
      const-string v0, " mLastBroadcastState="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 811,816 ****
--- 904,910 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(I)V
  
+     .line 847
      const-string v0, "  mNightMode="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 819,824 ****
--- 913,919 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 848
      const-string v0, " mCarModeEnabled="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 827,832 ****
--- 922,928 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 849
      const-string v0, " mComputedNightMode="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 835,840 ****
--- 931,937 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 850
      const-string v0, "  mCurUiMode=0x"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 847,852 ****
--- 944,950 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 851
      const-string v0, " mSetUiMode=0x"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 859,864 ****
--- 957,963 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 852
      const-string v0, "  mHoldingConfiguration="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 867,872 ****
--- 966,972 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 853
      const-string v0, " mSystemReady="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 875,884 ****
--- 975,986 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 854
      iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mLocation:Landroid/location/Location;
  
      if-eqz v0, :cond_1
  
+     .line 855
      const-string v0, "  mLocation="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 887,892 ****
--- 989,995 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 857
      :cond_1
      monitor-exit v1
  
***************
*** 907,934 ****
--- 1010,1044 ----
      .parameter "flags"
  
      .prologue
+     .line 369
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 370
      const/4 v0, 0x1
  
      :try_start_0
      invoke-virtual {p0, v0}, Lcom/android/server/UiModeManagerService;->setCarModeLocked(Z)V
  
+     .line 371
      iget-boolean v0, p0, Lcom/android/server/UiModeManagerService;->mSystemReady:Z
  
      if-eqz v0, :cond_0
  
+     .line 372
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Lcom/android/server/UiModeManagerService;->updateLocked(II)V
  
+     .line 374
      :cond_0
      monitor-exit v1
  
+     .line 375
      return-void
  
+     .line 374
      :catchall_0
      move-exception v0
  
***************
*** 943,952 ****
--- 1053,1064 ----
      .locals 2
  
      .prologue
+     .line 378
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 379
      :try_start_0
      iget v0, p0, Lcom/android/server/UiModeManagerService;->mCurUiMode:I
  
***************
*** 956,961 ****
--- 1068,1074 ----
  
      return v0
  
+     .line 380
      :catchall_0
      move-exception v0
  
***************
*** 975,980 ****
--- 1088,1094 ----
      .end annotation
  
      .prologue
+     .line 409
      iget v0, p0, Lcom/android/server/UiModeManagerService;->mNightMode:I
  
      return v0
***************
*** 984,989 ****
--- 1098,1104 ----
      .locals 1
  
      .prologue
+     .line 422
      iget-boolean v0, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
      if-nez v0, :cond_0
***************
*** 1009,1020 ****
--- 1124,1138 ----
      .parameter "enabled"
  
      .prologue
+     .line 426
      iget-boolean v0, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
      if-eq v0, p1, :cond_0
  
+     .line 427
      iput-boolean p1, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
+     .line 429
      :cond_0
      return-void
  .end method
***************
*** 1029,1040 ****
--- 1147,1161 ----
      .end annotation
  
      .prologue
+     .line 384
      iget-object v3, p0, Lcom/android/server/UiModeManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v3
  
+     .line 385
      packed-switch p1, :pswitch_data_0
  
+     .line 391
      :try_start_0
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
***************
*** 1060,1065 ****
--- 1181,1187 ----
  
      throw v2
  
+     .line 405
      :catchall_0
      move-exception v2
  
***************
*** 1069,1074 ****
--- 1191,1197 ----
  
      throw v2
  
+     .line 393
      :pswitch_0
      :try_start_1
      invoke-virtual {p0}, Lcom/android/server/UiModeManagerService;->isDoingNightMode()Z
***************
*** 1077,1096 ****
--- 1200,1224 ----
  
      if-nez v2, :cond_0
  
+     .line 394
      monitor-exit v3
  
+     .line 406
      :goto_0
      return-void
  
+     .line 397
      :cond_0
      iget v2, p0, Lcom/android/server/UiModeManagerService;->mNightMode:I
  
      if-eq v2, p1, :cond_1
  
+     .line 398
      invoke-static {}, Landroid/os/Binder;->clearCallingIdentity()J
  
      move-result-wide v0
  
+     .line 399
      .local v0, ident:J
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
***************
*** 1102,1117 ****
--- 1230,1249 ----
  
      invoke-static {v2, v4, p1}, Landroid/provider/Settings$Secure;->putInt(Landroid/content/ContentResolver;Ljava/lang/String;I)Z
  
+     .line 401
      invoke-static {v0, v1}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
+     .line 402
      iput p1, p0, Lcom/android/server/UiModeManagerService;->mNightMode:I
  
+     .line 403
      const/4 v2, 0x0
  
      const/4 v4, 0x0
  
      invoke-virtual {p0, v2, v4}, Lcom/android/server/UiModeManagerService;->updateLocked(II)V
  
+     .line 405
      .end local v0           #ident:J
      :cond_1
      monitor-exit v3
***************
*** 1120,1125 ****
--- 1252,1258 ----
  
      goto :goto_0
  
+     .line 385
      nop
  
      :pswitch_data_0
***************
*** 1138,1152 ****
--- 1271,1288 ----
  
      const/4 v1, 0x0
  
+     .line 413
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 414
      const/4 v3, 0x1
  
      :try_start_0
      iput-boolean v3, p0, Lcom/android/server/UiModeManagerService;->mSystemReady:Z
  
+     .line 415
      iget v3, p0, Lcom/android/server/UiModeManagerService;->mDockState:I
  
      const/4 v4, 0x2
***************
*** 1156,1182 ****
--- 1292,1324 ----
      :goto_0
      iput-boolean v0, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
+     .line 416
      const/4 v0, 0x0
  
      const/4 v1, 0x0
  
      invoke-virtual {p0, v0, v1}, Lcom/android/server/UiModeManagerService;->updateLocked(II)V
  
+     .line 417
      iget-object v0, p0, Lcom/android/server/UiModeManagerService;->mHandler:Landroid/os/Handler;
  
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/os/Handler;->sendEmptyMessage(I)Z
  
+     .line 418
      monitor-exit v2
  
+     .line 419
      return-void
  
      :cond_0
      move v0, v1
  
+     .line 415
      goto :goto_0
  
+     .line 418
      :catchall_0
      move-exception v0
  
***************
*** 1192,1218 ****
--- 1334,1367 ----
      .parameter "sendIt"
  
      .prologue
+     .line 455
      const/4 v1, 0x1
  
+     .line 456
      .local v1, uiMode:I
      iget-boolean v2, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
      if-eqz v2, :cond_2
  
+     .line 457
      const/4 v1, 0x3
  
+     .line 461
      :cond_0
      :goto_0
      iget-boolean v2, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
      if-eqz v2, :cond_5
  
+     .line 462
      iget v2, p0, Lcom/android/server/UiModeManagerService;->mNightMode:I
  
      if-nez v2, :cond_4
  
+     .line 463
      invoke-virtual {p0}, Lcom/android/server/UiModeManagerService;->updateTwilightLocked()V
  
+     .line 464
      iget-boolean v2, p0, Lcom/android/server/UiModeManagerService;->mComputedNightMode:Z
  
      if-eqz v2, :cond_3
***************
*** 1222,1230 ****
--- 1371,1381 ----
      :goto_1
      or-int/2addr v1, v2
  
+     .line 482
      :goto_2
      iput v1, p0, Lcom/android/server/UiModeManagerService;->mCurUiMode:I
  
+     .line 484
      iget-boolean v2, p0, Lcom/android/server/UiModeManagerService;->mHoldingConfiguration:Z
  
      if-nez v2, :cond_1
***************
*** 1233,1246 ****
--- 1384,1401 ----
  
      if-eq v1, v2, :cond_1
  
+     .line 485
      iput v1, p0, Lcom/android/server/UiModeManagerService;->mSetUiMode:I
  
+     .line 486
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mConfiguration:Landroid/content/res/Configuration;
  
      iput v1, v2, Landroid/content/res/Configuration;->uiMode:I
  
+     .line 488
      if-eqz p1, :cond_1
  
+     .line 490
      :try_start_0
      invoke-static {}, Landroid/app/ActivityManagerNative;->getDefault()Landroid/app/IActivityManager;
  
***************
*** 1252,1261 ****
--- 1407,1418 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 496
      :cond_1
      :goto_3
      return-void
  
+     .line 458
      :cond_2
      iget v2, p0, Lcom/android/server/UiModeManagerService;->mDockState:I
  
***************
*** 1265,1279 ****
--- 1422,1439 ----
  
      if-eqz v2, :cond_0
  
+     .line 459
      const/4 v1, 0x2
  
      goto :goto_0
  
+     .line 464
      :cond_3
      const/16 v2, 0x10
  
      goto :goto_1
  
+     .line 467
      :cond_4
      iget v2, p0, Lcom/android/server/UiModeManagerService;->mNightMode:I
  
***************
*** 1283,1288 ****
--- 1443,1449 ----
  
      goto :goto_2
  
+     .line 471
      :cond_5
      and-int/lit8 v2, v1, -0x31
  
***************
*** 1290,1298 ****
--- 1451,1461 ----
  
      goto :goto_2
  
+     .line 491
      :catch_0
      move-exception v0
  
+     .line 492
      .local v0, e:Landroid/os/RemoteException;
      sget-object v2, Lcom/android/server/UiModeManagerService;->TAG:Ljava/lang/String;
  
***************
*** 1312,1328 ****
--- 1475,1495 ----
  
      const/4 v1, 0x0
  
+     .line 432
      iget-object v2, p0, Lcom/android/server/UiModeManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 433
      :try_start_0
      iget v3, p0, Lcom/android/server/UiModeManagerService;->mDockState:I
  
      if-eq p1, v3, :cond_0
  
+     .line 434
      iput p1, p0, Lcom/android/server/UiModeManagerService;->mDockState:I
  
+     .line 435
      iget v3, p0, Lcom/android/server/UiModeManagerService;->mDockState:I
  
      const/4 v4, 0x2
***************
*** 1332,1357 ****
--- 1499,1530 ----
      :goto_0
      invoke-virtual {p0, v0}, Lcom/android/server/UiModeManagerService;->setCarModeLocked(Z)V
  
+     .line 436
      iget-boolean v0, p0, Lcom/android/server/UiModeManagerService;->mSystemReady:Z
  
      if-eqz v0, :cond_0
  
+     .line 437
      const/4 v0, 0x1
  
      const/4 v1, 0x0
  
      invoke-virtual {p0, v0, v1}, Lcom/android/server/UiModeManagerService;->updateLocked(II)V
  
+     .line 440
      :cond_0
      monitor-exit v2
  
+     .line 441
      return-void
  
      :cond_1
      move v0, v1
  
+     .line 435
      goto :goto_0
  
+     .line 440
      :catchall_0
      move-exception v0
  
***************
*** 1368,1383 ****
--- 1541,1560 ----
      .parameter "disableFlags"
  
      .prologue
+     .line 499
      invoke-static {}, Landroid/os/Binder;->clearCallingIdentity()J
  
      move-result-wide v11
  
+     .line 502
      .local v11, ident:J
      const/4 v9, 0x0
  
+     .line 503
      .local v9, action:Ljava/lang/String;
      const/4 v14, 0x0
  
+     .line 504
      .local v14, oldAction:Ljava/lang/String;
      :try_start_0
      iget v1, p0, Lcom/android/server/UiModeManagerService;->mLastBroadcastState:I
***************
*** 1386,1411 ****
--- 1563,1595 ----
  
      if-ne v1, v3, :cond_7
  
+     .line 505
      invoke-direct {p0}, Lcom/android/server/UiModeManagerService;->adjustStatusBarCarModeLocked()V
  
+     .line 506
      sget-object v14, Landroid/app/UiModeManager;->ACTION_EXIT_CAR_MODE:Ljava/lang/String;
  
+     .line 511
      :cond_0
      :goto_0
      iget-boolean v1, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
      if-eqz v1, :cond_8
  
+     .line 512
      iget v1, p0, Lcom/android/server/UiModeManagerService;->mLastBroadcastState:I
  
      const/4 v3, 0x2
  
      if-eq v1, v3, :cond_2
  
+     .line 513
      invoke-direct {p0}, Lcom/android/server/UiModeManagerService;->adjustStatusBarCarModeLocked()V
  
+     .line 515
      if-eqz v14, :cond_1
  
+     .line 516
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      new-instance v3, Landroid/content/Intent;
***************
*** 1414,1434 ****
--- 1598,1623 ----
  
      invoke-virtual {v1, v3}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 518
      :cond_1
      const/4 v1, 0x2
  
      iput v1, p0, Lcom/android/server/UiModeManagerService;->mLastBroadcastState:I
  
+     .line 519
      sget-object v9, Landroid/app/UiModeManager;->ACTION_ENTER_CAR_MODE:Ljava/lang/String;
  
+     .line 534
      :cond_2
      :goto_1
      if-eqz v9, :cond_b
  
+     .line 547
      new-instance v2, Landroid/content/Intent;
  
      invoke-direct {v2, v9}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 548
      .local v2, intent:Landroid/content/Intent;
      const-string v1, "enableFlags"
  
***************
*** 1436,1447 ****
--- 1625,1638 ----
  
      invoke-virtual {v2, v1, v0}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 549
      const-string v1, "disableFlags"
  
      move/from16 v0, p2
  
      invoke-virtual {v2, v1, v0}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 550
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      const/4 v3, 0x0
***************
*** 1458,1467 ****
--- 1649,1660 ----
  
      invoke-virtual/range {v1 .. v8}, Landroid/content/Context;->sendOrderedBroadcast(Landroid/content/Intent;Ljava/lang/String;Landroid/content/BroadcastReceiver;Landroid/os/Handler;ILjava/lang/String;Landroid/os/Bundle;)V
  
+     .line 555
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Lcom/android/server/UiModeManagerService;->mHoldingConfiguration:Z
  
+     .line 587
      .end local v2           #intent:Landroid/content/Intent;
      :cond_3
      :goto_2
***************
*** 1469,1474 ****
--- 1662,1668 ----
  
      invoke-virtual {p0, v1}, Lcom/android/server/UiModeManagerService;->updateConfigurationLocked(Z)V
  
+     .line 590
      iget-boolean v1, p0, Lcom/android/server/UiModeManagerService;->mCharging:Z
  
      if-eqz v1, :cond_f
***************
*** 1495,1500 ****
--- 1689,1695 ----
      :cond_5
      const/4 v13, 0x1
  
+     .line 593
      .local v13, keepScreenOn:Z
      :goto_3
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mWakeLock:Landroid/os/PowerManager$WakeLock;
***************
*** 1505,1524 ****
--- 1700,1724 ----
  
      if-eq v13, v1, :cond_6
  
+     .line 594
      if-eqz v13, :cond_10
  
+     .line 595
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mWakeLock:Landroid/os/PowerManager$WakeLock;
  
      invoke-virtual {v1}, Landroid/os/PowerManager$WakeLock;->acquire()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 601
      :cond_6
      :goto_4
      invoke-static {v11, v12}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
+     .line 603
      return-void
  
+     .line 507
      .end local v13           #keepScreenOn:Z
      :cond_7
      :try_start_1
***************
*** 1530,1539 ****
--- 1730,1741 ----
  
      if-eqz v1, :cond_0
  
+     .line 508
      sget-object v14, Landroid/app/UiModeManager;->ACTION_EXIT_DESK_MODE:Ljava/lang/String;
  
      goto :goto_0
  
+     .line 521
      :cond_8
      iget v1, p0, Lcom/android/server/UiModeManagerService;->mDockState:I
  
***************
*** 1543,1548 ****
--- 1745,1751 ----
  
      if-eqz v1, :cond_a
  
+     .line 522
      iget v1, p0, Lcom/android/server/UiModeManagerService;->mLastBroadcastState:I
  
      invoke-static {v1}, Lcom/android/server/UiModeManagerService;->isDeskDockState(I)Z
***************
*** 1551,1558 ****
--- 1754,1763 ----
  
      if-nez v1, :cond_2
  
+     .line 523
      if-eqz v14, :cond_9
  
+     .line 524
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
      new-instance v3, Landroid/content/Intent;
***************
*** 1561,1596 ****
--- 1766,1809 ----
  
      invoke-virtual {v1, v3}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 526
      :cond_9
      iget v1, p0, Lcom/android/server/UiModeManagerService;->mDockState:I
  
      iput v1, p0, Lcom/android/server/UiModeManagerService;->mLastBroadcastState:I
  
+     .line 527
      sget-object v9, Landroid/app/UiModeManager;->ACTION_ENTER_DESK_MODE:Ljava/lang/String;
  
      goto :goto_1
  
+     .line 530
      :cond_a
      const/4 v1, 0x0
  
      iput v1, p0, Lcom/android/server/UiModeManagerService;->mLastBroadcastState:I
  
+     .line 531
      move-object v9, v14
  
      goto/16 :goto_1
  
+     .line 557
      :cond_b
      const/4 v10, 0x0
  
+     .line 558
      .local v10, homeIntent:Landroid/content/Intent;
      iget-boolean v1, p0, Lcom/android/server/UiModeManagerService;->mCarModeEnabled:Z
  
      if-eqz v1, :cond_d
  
+     .line 559
      and-int/lit8 v1, p1, 0x1
  
      if-eqz v1, :cond_c
  
+     .line 561
      const-string v1, "android.intent.category.CAR_DOCK"
  
      invoke-static {v1}, Lcom/android/server/UiModeManagerService;->buildHomeIntent(Ljava/lang/String;)Landroid/content/Intent;
***************
*** 1599,1608 ****
--- 1812,1823 ----
  
      move-result-object v10
  
+     .line 579
      :cond_c
      :goto_5
      if-eqz v10, :cond_3
  
+     .line 581
      :try_start_2
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mContext:Landroid/content/Context;
  
***************
*** 1613,1623 ****
--- 1828,1840 ----
  
      goto :goto_2
  
+     .line 582
      :catch_0
      move-exception v1
  
      goto :goto_2
  
+     .line 563
      :cond_d
      :try_start_3
      iget v1, p0, Lcom/android/server/UiModeManagerService;->mDockState:I
***************
*** 1628,1637 ****
--- 1845,1856 ----
  
      if-eqz v1, :cond_e
  
+     .line 564
      and-int/lit8 v1, p1, 0x1
  
      if-eqz v1, :cond_c
  
+     .line 566
      const-string v1, "android.intent.category.DESK_DOCK"
  
      invoke-static {v1}, Lcom/android/server/UiModeManagerService;->buildHomeIntent(Ljava/lang/String;)Landroid/content/Intent;
***************
*** 1640,1650 ****
--- 1859,1871 ----
  
      goto :goto_5
  
+     .line 569
      :cond_e
      and-int/lit8 v1, p2, 0x1
  
      if-eqz v1, :cond_c
  
+     .line 570
      const-string v1, "android.intent.category.HOME"
  
      invoke-static {v1}, Lcom/android/server/UiModeManagerService;->buildHomeIntent(Ljava/lang/String;)Landroid/content/Intent;
***************
*** 1653,1664 ****
--- 1874,1887 ----
  
      goto :goto_5
  
+     .line 590
      .end local v10           #homeIntent:Landroid/content/Intent;
      :cond_f
      const/4 v13, 0x0
  
      goto :goto_3
  
+     .line 597
      .restart local v13       #keepScreenOn:Z
      :cond_10
      iget-object v1, p0, Lcom/android/server/UiModeManagerService;->mWakeLock:Landroid/os/PowerManager$WakeLock;
***************
*** 1669,1674 ****
--- 1892,1898 ----
  
      goto :goto_4
  
+     .line 601
      .end local v13           #keepScreenOn:Z
      :catchall_0
      move-exception v1
***************
*** 1682,1706 ****
--- 1906,1935 ----
      .locals 17
  
      .prologue
+     .line 785
      move-object/from16 v0, p0
  
      iget-object v4, v0, Lcom/android/server/UiModeManagerService;->mLocation:Landroid/location/Location;
  
      if-nez v4, :cond_0
  
+     .line 830
      :goto_0
      return-void
  
+     .line 788
      :cond_0
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v2
  
+     .line 791
      .local v2, currentTime:J
      new-instance v1, Lcom/android/server/TwilightCalculator;
  
      invoke-direct {v1}, Lcom/android/server/TwilightCalculator;-><init>()V
  
+     .line 792
      .local v1, tw:Lcom/android/server/TwilightCalculator;
      move-object/from16 v0, p0
  
***************
*** 1720,1735 ****
--- 1949,1968 ----
  
      invoke-virtual/range {v1 .. v7}, Lcom/android/server/TwilightCalculator;->calculateTwilight(JDD)V
  
+     .line 794
      iget v4, v1, Lcom/android/server/TwilightCalculator;->mState:I
  
      if-nez v4, :cond_2
  
+     .line 795
      const/4 v14, 0x0
  
+     .line 801
      .local v14, nightMode:Z
      :goto_1
      const-wide/16 v12, 0x0
  
+     .line 802
      .local v12, nextUpdate:J
      iget-wide v4, v1, Lcom/android/server/TwilightCalculator;->mSunrise:J
  
***************
*** 1747,1757 ****
--- 1980,1992 ----
  
      if-nez v4, :cond_3
  
+     .line 804
      :cond_1
      const-wide/32 v4, 0x2932e00
  
      add-long v12, v2, v4
  
+     .line 823
      :goto_2
      new-instance v16, Landroid/content/Intent;
  
***************
*** 1761,1766 ****
--- 1996,2002 ----
  
      invoke-direct {v0, v4}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 824
      .local v16, updateIntent:Landroid/content/Intent;
      move-object/from16 v0, p0
  
***************
*** 1776,1781 ****
--- 2012,2018 ----
  
      move-result-object v15
  
+     .line 826
      .local v15, pendingIntent:Landroid/app/PendingIntent;
      move-object/from16 v0, p0
  
***************
*** 1783,1788 ****
--- 2020,2026 ----
  
      invoke-virtual {v4, v15}, Landroid/app/AlarmManager;->cancel(Landroid/app/PendingIntent;)V
  
+     .line 827
      move-object/from16 v0, p0
  
      iget-object v4, v0, Lcom/android/server/UiModeManagerService;->mAlarmManager:Landroid/app/AlarmManager;
***************
*** 1791,1802 ****
--- 2029,2042 ----
  
      invoke-virtual {v4, v5, v12, v13, v15}, Landroid/app/AlarmManager;->set(IJLandroid/app/PendingIntent;)V
  
+     .line 829
      move-object/from16 v0, p0
  
      iput-boolean v14, v0, Lcom/android/server/UiModeManagerService;->mComputedNightMode:Z
  
      goto :goto_0
  
+     .line 797
      .end local v12           #nextUpdate:J
      .end local v14           #nightMode:Z
      .end local v15           #pendingIntent:Landroid/app/PendingIntent;
***************
*** 1807,1827 ****
--- 2047,2071 ----
      .restart local v14       #nightMode:Z
      goto :goto_1
  
+     .line 806
      .restart local v12       #nextUpdate:J
      :cond_3
      iget v11, v1, Lcom/android/server/TwilightCalculator;->mState:I
  
+     .line 808
      .local v11, mLastTwilightState:I
      const-wide/32 v4, 0xea60
  
      add-long/2addr v12, v4
  
+     .line 809
      iget-wide v4, v1, Lcom/android/server/TwilightCalculator;->mSunset:J
  
      cmp-long v4, v2, v4
  
      if-lez v4, :cond_4
  
+     .line 811
      const-wide/32 v4, 0x5265c00
  
      add-long v5, v2, v4
***************
*** 1846,1862 ****
--- 2090,2109 ----
  
      invoke-virtual/range {v4 .. v10}, Lcom/android/server/TwilightCalculator;->calculateTwilight(JDD)V
  
+     .line 816
      :cond_4
      const/4 v4, 0x1
  
      if-ne v11, v4, :cond_5
  
+     .line 817
      iget-wide v4, v1, Lcom/android/server/TwilightCalculator;->mSunrise:J
  
      add-long/2addr v12, v4
  
      goto :goto_2
  
+     .line 819
      :cond_5
      iget-wide v4, v1, Lcom/android/server/TwilightCalculator;->mSunset:J
  
