*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/WallpaperManagerService$MyPackageMonitor.smali	2012-03-23 19:34:50.622415885 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/WallpaperManagerService$MyPackageMonitor.smali.original	2012-03-23 19:34:46.813346429 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 228
      iput-object p1, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      invoke-direct {p0}, Lcom/android/internal/content/PackageMonitor;-><init>()V
***************
*** 42,49 ****
--- 43,52 ----
  
      const/4 v7, 0x2
  
+     .line 277
      const/4 v1, 0x0
  
+     .line 278
      .local v1, changed:Z
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 51,56 ****
--- 54,60 ----
  
      monitor-enter v4
  
+     .line 279
      :try_start_0
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 58,63 ****
--- 62,68 ----
  
      if-eqz v3, :cond_1
  
+     .line 280
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v3, v3, Lcom/android/server/WallpaperManagerService;->mWallpaperComponent:Landroid/content/ComponentName;
***************
*** 70,85 ****
--- 75,94 ----
  
      move-result v0
  
+     .line 281
      .local v0, change:I
      if-eq v0, v8, :cond_0
  
      if-ne v0, v7, :cond_1
  
+     .line 283
      :cond_0
      const/4 v1, 0x1
  
+     .line 284
      if-eqz p1, :cond_1
  
+     .line 285
      const-string v3, "WallpaperService"
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 106,117 ****
--- 115,128 ----
  
      invoke-static {v3, v5}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 286
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      const/4 v5, 0x0
  
      invoke-virtual {v3, v5}, Lcom/android/server/WallpaperManagerService;->clearWallpaperLocked(Z)V
  
+     .line 290
      .end local v0           #change:I
      :cond_1
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
***************
*** 120,125 ****
--- 131,137 ----
  
      if-eqz v3, :cond_3
  
+     .line 291
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v3, v3, Lcom/android/server/WallpaperManagerService;->mNextWallpaperComponent:Landroid/content/ComponentName;
***************
*** 132,142 ****
--- 144,156 ----
  
      move-result v0
  
+     .line 292
      .restart local v0       #change:I
      if-eq v0, v8, :cond_2
  
      if-ne v0, v7, :cond_3
  
+     .line 294
      :cond_2
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 144,149 ****
--- 158,164 ----
  
      iput-object v5, v3, Lcom/android/server/WallpaperManagerService;->mNextWallpaperComponent:Landroid/content/ComponentName;
  
+     .line 297
      .end local v0           #change:I
      :cond_3
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
***************
*** 168,173 ****
--- 183,189 ----
  
      if-eqz v3, :cond_4
  
+     .line 300
      :try_start_1
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 188,193 ****
--- 204,210 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/content/pm/PackageManager$NameNotFoundException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 307
      :cond_4
      :goto_0
      :try_start_2
***************
*** 213,218 ****
--- 230,236 ----
  
      if-eqz v3, :cond_5
  
+     .line 310
      :try_start_3
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 233,248 ****
--- 251,270 ----
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
      .catch Landroid/content/pm/PackageManager$NameNotFoundException; {:try_start_3 .. :try_end_3} :catch_1
  
+     .line 316
      :cond_5
      :goto_1
      :try_start_4
      monitor-exit v4
  
+     .line 317
      return v1
  
+     .line 302
      :catch_0
      move-exception v2
  
+     .line 303
      .local v2, e:Landroid/content/pm/PackageManager$NameNotFoundException;
      const-string v3, "WallpaperService"
  
***************
*** 270,275 ****
--- 292,298 ----
  
      invoke-static {v3, v5}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 304
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      const/4 v5, 0x0
***************
*** 278,283 ****
--- 301,307 ----
  
      goto :goto_0
  
+     .line 316
      .end local v2           #e:Landroid/content/pm/PackageManager$NameNotFoundException;
      :catchall_0
      move-exception v3
***************
*** 288,296 ****
--- 312,322 ----
  
      throw v3
  
+     .line 312
      :catch_1
      move-exception v2
  
+     .line 313
      .restart local v2       #e:Landroid/content/pm/PackageManager$NameNotFoundException;
      :try_start_5
      iget-object v3, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
***************
*** 312,317 ****
--- 338,344 ----
      .parameter "doit"
  
      .prologue
+     .line 268
      invoke-virtual {p0, p4}, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->doPackagesChanged(Z)Z
  
      move-result v0
***************
*** 324,335 ****
--- 351,364 ----
      .parameter "packageName"
  
      .prologue
+     .line 247
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v1, v0, Lcom/android/server/WallpaperManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 248
      :try_start_0
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 351,373 ****
--- 380,407 ----
  
      if-nez v0, :cond_1
  
+     .line 250
      :cond_0
      monitor-exit v1
  
+     .line 254
      :goto_0
      return-void
  
+     .line 252
      :cond_1
      monitor-exit v1
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 253
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->doPackagesChanged(Z)Z
  
      goto :goto_0
  
+     .line 252
      :catchall_0
      move-exception v0
  
***************
*** 385,396 ****
--- 419,432 ----
      .parameter "uid"
  
      .prologue
+     .line 231
      iget-object v1, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v2, v1, Lcom/android/server/WallpaperManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 232
      :try_start_0
      iget-object v1, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 412,432 ****
--- 448,472 ----
  
      if-eqz v1, :cond_0
  
+     .line 234
      iget-object v1, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      const/4 v3, 0x0
  
      iput-boolean v3, v1, Lcom/android/server/WallpaperManagerService;->mWallpaperUpdating:Z
  
+     .line 235
      iget-object v1, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v0, v1, Lcom/android/server/WallpaperManagerService;->mWallpaperComponent:Landroid/content/ComponentName;
  
+     .line 236
      .local v0, comp:Landroid/content/ComponentName;
      iget-object v1, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      invoke-virtual {v1}, Lcom/android/server/WallpaperManagerService;->clearWallpaperComponentLocked()V
  
+     .line 237
      iget-object v1, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      const/4 v3, 0x0
***************
*** 439,462 ****
--- 479,507 ----
  
      if-nez v1, :cond_0
  
+     .line 238
      const-string v1, "WallpaperService"
  
      const-string v3, "Wallpaper no longer available; reverting to default"
  
      invoke-static {v1, v3}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 239
      iget-object v1, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      const/4 v3, 0x0
  
      invoke-virtual {v1, v3}, Lcom/android/server/WallpaperManagerService;->clearWallpaperLocked(Z)V
  
+     .line 242
      .end local v0           #comp:Landroid/content/ComponentName;
      :cond_0
      monitor-exit v2
  
+     .line 243
      return-void
  
+     .line 242
      :catchall_0
      move-exception v1
  
***************
*** 473,484 ****
--- 518,531 ----
      .parameter "uid"
  
      .prologue
+     .line 258
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v1, v0, Lcom/android/server/WallpaperManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 259
      :try_start_0
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 500,516 ****
--- 547,567 ----
  
      if-eqz v0, :cond_0
  
+     .line 261
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->this$0:Lcom/android/server/WallpaperManagerService;
  
      const/4 v2, 0x1
  
      iput-boolean v2, v0, Lcom/android/server/WallpaperManagerService;->mWallpaperUpdating:Z
  
+     .line 263
      :cond_0
      monitor-exit v1
  
+     .line 264
      return-void
  
+     .line 263
      :catchall_0
      move-exception v0
  
***************
*** 525,533 ****
--- 576,586 ----
      .locals 1
  
      .prologue
+     .line 273
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Lcom/android/server/WallpaperManagerService$MyPackageMonitor;->doPackagesChanged(Z)Z
  
+     .line 274
      return-void
  .end method
