*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/WallpaperManagerService$WallpaperConnection.smali	2012-03-23 19:35:11.119789654 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/WallpaperManagerService$WallpaperConnection.smali.original	2012-03-23 19:34:47.371356604 -0500
***************
*** 36,53 ****
--- 36,57 ----
      .parameter "info"
  
      .prologue
+     .line 180
      iput-object p1, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      invoke-direct {p0}, Landroid/service/wallpaper/IWallpaperConnection$Stub;-><init>()V
  
+     .line 176
      new-instance v0, Landroid/os/Binder;
  
      invoke-direct {v0}, Landroid/os/Binder;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->mToken:Landroid/os/Binder;
  
+     .line 181
      iput-object p2, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->mInfo:Landroid/app/WallpaperInfo;
  
+     .line 182
      return-void
  .end method
  
***************
*** 58,65 ****
--- 62,71 ----
      .parameter "engine"
  
      .prologue
+     .line 215
      iput-object p1, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->mEngine:Landroid/service/wallpaper/IWallpaperEngine;
  
+     .line 216
      return-void
  .end method
  
***************
*** 69,80 ****
--- 75,88 ----
      .parameter "service"
  
      .prologue
+     .line 185
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v1, v0, Lcom/android/server/WallpaperManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 186
      :try_start_0
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 82,87 ****
--- 90,96 ----
  
      if-ne v0, p0, :cond_0
  
+     .line 187
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
***************
*** 90,115 ****
--- 99,130 ----
  
      iput-wide v2, v0, Lcom/android/server/WallpaperManagerService;->mLastDiedTime:J
  
+     .line 188
      invoke-static {p2}, Landroid/service/wallpaper/IWallpaperService$Stub;->asInterface(Landroid/os/IBinder;)Landroid/service/wallpaper/IWallpaperService;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->mService:Landroid/service/wallpaper/IWallpaperService;
  
+     .line 189
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      invoke-virtual {v0, p0}, Lcom/android/server/WallpaperManagerService;->attachServiceLocked(Lcom/android/server/WallpaperManagerService$WallpaperConnection;)V
  
+     .line 194
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      #calls: Lcom/android/server/WallpaperManagerService;->saveSettingsLocked()V
      invoke-static {v0}, Lcom/android/server/WallpaperManagerService;->access$100(Lcom/android/server/WallpaperManagerService;)V
  
+     .line 196
      :cond_0
      monitor-exit v1
  
+     .line 197
      return-void
  
+     .line 196
      :catchall_0
      move-exception v0
  
***************
*** 125,151 ****
--- 140,171 ----
      .parameter "name"
  
      .prologue
+     .line 200
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v1, v0, Lcom/android/server/WallpaperManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 201
      const/4 v0, 0x0
  
      :try_start_0
      iput-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->mService:Landroid/service/wallpaper/IWallpaperService;
  
+     .line 202
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->mEngine:Landroid/service/wallpaper/IWallpaperEngine;
  
+     .line 203
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v0, v0, Lcom/android/server/WallpaperManagerService;->mWallpaperConnection:Lcom/android/server/WallpaperManagerService$WallpaperConnection;
  
      if-ne v0, p0, :cond_0
  
+     .line 204
      const-string v0, "WallpaperService"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 172,177 ****
--- 192,198 ----
  
      invoke-static {v0, v2}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 205
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-boolean v0, v0, Lcom/android/server/WallpaperManagerService;->mWallpaperUpdating:Z
***************
*** 194,216 ****
--- 215,242 ----
  
      if-lez v0, :cond_0
  
+     .line 207
      const-string v0, "WallpaperService"
  
      const-string v2, "Reverting to built-in wallpaper!"
  
      invoke-static {v0, v2}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 208
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Lcom/android/server/WallpaperManagerService;->clearWallpaperLocked(Z)V
  
+     .line 211
      :cond_0
      monitor-exit v1
  
+     .line 212
      return-void
  
+     .line 211
      :catchall_0
      move-exception v0
  
***************
*** 226,237 ****
--- 252,265 ----
      .parameter "name"
  
      .prologue
+     .line 219
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      iget-object v1, v0, Lcom/android/server/WallpaperManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 220
      :try_start_0
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
***************
*** 239,244 ****
--- 267,273 ----
  
      if-ne v0, p0, :cond_0
  
+     .line 221
      iget-object v0, p0, Lcom/android/server/WallpaperManagerService$WallpaperConnection;->this$0:Lcom/android/server/WallpaperManagerService;
  
      invoke-virtual {v0, p1}, Lcom/android/server/WallpaperManagerService;->updateWallpaperBitmapLocked(Ljava/lang/String;)Landroid/os/ParcelFileDescriptor;
***************
*** 247,252 ****
--- 276,282 ----
  
      monitor-exit v1
  
+     .line 223
      :goto_0
      return-object v0
  
***************
*** 257,262 ****
--- 287,293 ----
  
      goto :goto_0
  
+     .line 224
      :catchall_0
      move-exception v0
  
