*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/WifiService$2.smali	2012-03-23 19:34:53.325465176 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/WifiService$2.smali.original	2012-03-23 19:34:46.857347228 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 394
      iput-object p1, p0, Lcom/android/server/WifiService$2;->this$0:Lcom/android/server/WifiService;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 41,46 ****
--- 42,48 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 397
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
      move-result-object v2
***************
*** 53,64 ****
--- 55,68 ----
  
      if-eqz v2, :cond_2
  
+     .line 398
      const-string v2, "wifi_state"
  
      invoke-virtual {p2, v2, v1}, Landroid/content/Intent;->getIntExtra(Ljava/lang/String;I)I
  
      move-result v0
  
+     .line 401
      .local v0, wifiState:I
      iget-object v2, p0, Lcom/android/server/WifiService$2;->this$0:Lcom/android/server/WifiService;
  
***************
*** 70,91 ****
--- 74,99 ----
      #setter for: Lcom/android/server/WifiService;->mWifiEnabled:Z
      invoke-static {v2, v1}, Lcom/android/server/WifiService;->access$1402(Lcom/android/server/WifiService;Z)Z
  
+     .line 404
      iget-object v1, p0, Lcom/android/server/WifiService$2;->this$0:Lcom/android/server/WifiService;
  
      #calls: Lcom/android/server/WifiService;->resetNotification()V
      invoke-static {v1}, Lcom/android/server/WifiService;->access$1500(Lcom/android/server/WifiService;)V
  
+     .line 421
      .end local v0           #wifiState:I
      :cond_0
      :goto_1
      return-void
  
+     .line 401
      .restart local v0       #wifiState:I
      :cond_1
      const/4 v1, 0x0
  
      goto :goto_0
  
+     .line 405
      .end local v0           #wifiState:I
      :cond_2
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
***************
*** 100,105 ****
--- 108,114 ----
  
      if-eqz v1, :cond_3
  
+     .line 407
      iget-object v2, p0, Lcom/android/server/WifiService$2;->this$0:Lcom/android/server/WifiService;
  
      const-string v1, "networkInfo"
***************
*** 112,117 ****
--- 121,127 ----
  
      iput-object v1, v2, Lcom/android/server/WifiService;->mNetworkInfo:Landroid/net/NetworkInfo;
  
+     .line 410
      sget-object v1, Lcom/android/server/WifiService$4;->$SwitchMap$android$net$NetworkInfo$DetailedState:[I
  
      iget-object v2, p0, Lcom/android/server/WifiService$2;->this$0:Lcom/android/server/WifiService;
***************
*** 132,143 ****
--- 142,155 ----
  
      goto :goto_1
  
+     .line 413
      :pswitch_0
      iget-object v1, p0, Lcom/android/server/WifiService$2;->this$0:Lcom/android/server/WifiService;
  
      #calls: Lcom/android/server/WifiService;->evaluateTrafficStatsPolling()V
      invoke-static {v1}, Lcom/android/server/WifiService;->access$1600(Lcom/android/server/WifiService;)V
  
+     .line 414
      iget-object v1, p0, Lcom/android/server/WifiService$2;->this$0:Lcom/android/server/WifiService;
  
      #calls: Lcom/android/server/WifiService;->resetNotification()V
***************
*** 145,150 ****
--- 157,163 ----
  
      goto :goto_1
  
+     .line 417
      :cond_3
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
***************
*** 158,163 ****
--- 171,177 ----
  
      if-eqz v1, :cond_0
  
+     .line 419
      iget-object v1, p0, Lcom/android/server/WifiService$2;->this$0:Lcom/android/server/WifiService;
  
      #calls: Lcom/android/server/WifiService;->checkAndSetNotification()V
***************
*** 165,170 ****
--- 179,185 ----
  
      goto :goto_1
  
+     .line 410
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
