*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/WifiService$3.smali	2012-03-23 19:34:53.667471413 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/WifiService$3.smali.original	2012-03-23 19:34:46.860347286 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 917
      iput-object p1, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "pluggedType"
  
      .prologue
+     .line 1041
      and-int v0, p1, p2
  
      if-eqz v0, :cond_0
***************
*** 62,67 ****
--- 64,70 ----
  
      const/4 v1, 0x1
  
+     .line 1009
      iget-object v2, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #getter for: Lcom/android/server/WifiService;->mContext:Landroid/content/Context;
***************
*** 79,96 ****
--- 82,103 ----
  
      move-result v0
  
+     .line 1013
      .local v0, wifiSleepPolicy:I
      if-ne v0, v4, :cond_1
  
+     .line 1022
      :cond_0
      :goto_0
      return v1
  
+     .line 1016
      :cond_1
      if-ne v0, v1, :cond_2
  
      if-nez p2, :cond_0
  
+     .line 1022
      :cond_2
      invoke-direct {p0, p1, p2}, Lcom/android/server/WifiService$3;->shouldDeviceStayAwake(II)Z
  
***************
*** 107,116 ****
--- 114,125 ----
      .parameter "intent"
  
      .prologue
+     .line 920
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 922
      .local v0, action:Ljava/lang/String;
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
***************
*** 131,136 ****
--- 140,146 ----
  
      move-result-wide v1
  
+     .line 925
      .local v1, idleMillis:J
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
***************
*** 151,156 ****
--- 161,167 ----
  
      move-result v5
  
+     .line 928
      .local v5, stayAwakeConditions:I
      const-string v8, "android.intent.action.SCREEN_ON"
  
***************
*** 160,165 ****
--- 171,177 ----
  
      if-eqz v8, :cond_2
  
+     .line 932
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #getter for: Lcom/android/server/WifiService;->mAlarmManager:Landroid/app/AlarmManager;
***************
*** 176,181 ****
--- 188,194 ----
  
      invoke-virtual {v8, v9}, Landroid/app/AlarmManager;->cancel(Landroid/app/PendingIntent;)V
  
+     .line 933
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      const/4 v9, 0x0
***************
*** 183,193 ****
--- 196,208 ----
      #setter for: Lcom/android/server/WifiService;->mScreenOff:Z
      invoke-static {v8, v9}, Lcom/android/server/WifiService;->access$2002(Lcom/android/server/WifiService;Z)Z
  
+     .line 934
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #calls: Lcom/android/server/WifiService;->evaluateTrafficStatsPolling()V
      invoke-static {v8}, Lcom/android/server/WifiService;->access$1600(Lcom/android/server/WifiService;)V
  
+     .line 935
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #getter for: Lcom/android/server/WifiService;->mWifiStateMachine:Landroid/net/wifi/WifiStateMachine;
***************
*** 199,204 ****
--- 214,220 ----
  
      invoke-virtual {v8, v9}, Landroid/net/wifi/WifiStateMachine;->enableRssiPolling(Z)V
  
+     .line 936
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #getter for: Lcom/android/server/WifiService;->mBackgroundScanSupported:Z
***************
*** 208,213 ****
--- 224,230 ----
  
      if-eqz v8, :cond_0
  
+     .line 937
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #getter for: Lcom/android/server/WifiService;->mWifiStateMachine:Landroid/net/wifi/WifiStateMachine;
***************
*** 219,224 ****
--- 236,242 ----
  
      invoke-virtual {v8, v9}, Landroid/net/wifi/WifiStateMachine;->enableBackgroundScanCommand(Z)V
  
+     .line 939
      :cond_0
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
***************
*** 229,234 ****
--- 247,253 ----
  
      invoke-virtual {v8}, Landroid/net/wifi/WifiStateMachine;->enableAllNetworks()V
  
+     .line 940
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      const/4 v9, 0x0
***************
*** 236,245 ****
--- 255,266 ----
      #calls: Lcom/android/server/WifiService;->setDeviceIdleAndUpdateWifi(Z)V
      invoke-static {v8, v9}, Lcom/android/server/WifiService;->access$2200(Lcom/android/server/WifiService;Z)V
  
+     .line 998
      :cond_1
      :goto_0
      return-void
  
+     .line 941
      :cond_2
      const-string v8, "android.intent.action.SCREEN_OFF"
  
***************
*** 249,254 ****
--- 270,276 ----
  
      if-eqz v8, :cond_5
  
+     .line 945
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      const/4 v9, 0x1
***************
*** 256,266 ****
--- 278,290 ----
      #setter for: Lcom/android/server/WifiService;->mScreenOff:Z
      invoke-static {v8, v9}, Lcom/android/server/WifiService;->access$2002(Lcom/android/server/WifiService;Z)Z
  
+     .line 946
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #calls: Lcom/android/server/WifiService;->evaluateTrafficStatsPolling()V
      invoke-static {v8}, Lcom/android/server/WifiService;->access$1600(Lcom/android/server/WifiService;)V
  
+     .line 947
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #getter for: Lcom/android/server/WifiService;->mWifiStateMachine:Landroid/net/wifi/WifiStateMachine;
***************
*** 272,277 ****
--- 296,302 ----
  
      invoke-virtual {v8, v9}, Landroid/net/wifi/WifiStateMachine;->enableRssiPolling(Z)V
  
+     .line 948
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #getter for: Lcom/android/server/WifiService;->mBackgroundScanSupported:Z
***************
*** 281,286 ****
--- 306,312 ----
  
      if-eqz v8, :cond_3
  
+     .line 949
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #getter for: Lcom/android/server/WifiService;->mWifiStateMachine:Landroid/net/wifi/WifiStateMachine;
***************
*** 292,297 ****
--- 318,324 ----
  
      invoke-virtual {v8, v9}, Landroid/net/wifi/WifiStateMachine;->enableBackgroundScanCommand(Z)V
  
+     .line 957
      :cond_3
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
***************
*** 306,311 ****
--- 333,339 ----
  
      if-nez v8, :cond_1
  
+     .line 959
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      iget-object v8, v8, Lcom/android/server/WifiService;->mNetworkInfo:Landroid/net/NetworkInfo;
***************
*** 318,323 ****
--- 346,352 ----
  
      if-ne v8, v9, :cond_4
  
+     .line 961
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #getter for: Lcom/android/server/WifiService;->mAlarmManager:Landroid/app/AlarmManager;
***************
*** 344,349 ****
--- 373,379 ----
  
      goto :goto_0
  
+     .line 964
      :cond_4
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
***************
*** 354,359 ****
--- 384,390 ----
  
      goto :goto_0
  
+     .line 967
      :cond_5
      const-string v8, "com.android.server.WifiManager.action.DEVICE_IDLE"
  
***************
*** 363,368 ****
--- 394,400 ----
  
      if-eqz v8, :cond_6
  
+     .line 968
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      const/4 v9, 0x1
***************
*** 372,377 ****
--- 404,410 ----
  
      goto :goto_0
  
+     .line 969
      :cond_6
      const-string v8, "android.intent.action.BATTERY_CHANGED"
  
***************
*** 381,386 ****
--- 414,420 ----
  
      if-eqz v8, :cond_8
  
+     .line 977
      const-string v8, "plugged"
  
      const/4 v9, 0x0
***************
*** 389,394 ****
--- 423,429 ----
  
      move-result v3
  
+     .line 981
      .local v3, pluggedType:I
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
***************
*** 418,429 ****
--- 453,466 ----
  
      if-nez v8, :cond_7
  
+     .line 983
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v8
  
      add-long v6, v8, v1
  
+     .line 987
      .local v6, triggerTime:J
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
***************
*** 443,448 ****
--- 480,486 ----
  
      invoke-virtual {v8, v9, v6, v7, v10}, Landroid/app/AlarmManager;->set(IJLandroid/app/PendingIntent;)V
  
+     .line 989
      .end local v6           #triggerTime:J
      :cond_7
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
***************
*** 452,457 ****
--- 490,496 ----
  
      goto/16 :goto_0
  
+     .line 990
      .end local v3           #pluggedType:I
      :cond_8
      const-string v8, "android.bluetooth.adapter.action.CONNECTION_STATE_CHANGED"
***************
*** 462,467 ****
--- 501,507 ----
  
      if-eqz v8, :cond_9
  
+     .line 991
      const-string v8, "android.bluetooth.adapter.extra.CONNECTION_STATE"
  
      const/4 v9, 0x0
***************
*** 470,475 ****
--- 510,516 ----
  
      move-result v4
  
+     .line 993
      .local v4, state:I
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
***************
*** 482,487 ****
--- 523,529 ----
  
      goto/16 :goto_0
  
+     .line 994
      .end local v4           #state:I
      :cond_9
      const-string v8, "android.intent.action.EMERGENCY_CALLBACK_MODE_CHANGED"
***************
*** 492,497 ****
--- 534,540 ----
  
      if-eqz v8, :cond_1
  
+     .line 995
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      const-string v9, "phoneinECMState"
***************
*** 505,510 ****
--- 548,554 ----
      #setter for: Lcom/android/server/WifiService;->mEmergencyCallbackMode:Z
      invoke-static {v8, v9}, Lcom/android/server/WifiService;->access$2402(Lcom/android/server/WifiService;Z)Z
  
+     .line 996
      iget-object v8, p0, Lcom/android/server/WifiService$3;->this$0:Lcom/android/server/WifiService;
  
      #calls: Lcom/android/server/WifiService;->updateWifiState()V
