*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/search/SearchManagerService$GlobalSearchProviderObserver.smali	2012-03-23 19:37:15.219052602 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/search/SearchManagerService$GlobalSearchProviderObserver.smali.original	2012-03-23 19:34:45.603324363 -0500
***************
*** 27,43 ****
      .parameter "resolver"
  
      .prologue
      iput-object p1, p0, Landroid/server/search/SearchManagerService$GlobalSearchProviderObserver;->this$0:Landroid/server/search/SearchManagerService;
  
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/database/ContentObserver;-><init>(Landroid/os/Handler;)V
  
      iput-object p2, p0, Landroid/server/search/SearchManagerService$GlobalSearchProviderObserver;->mResolver:Landroid/content/ContentResolver;
  
      iget-object v0, p0, Landroid/server/search/SearchManagerService$GlobalSearchProviderObserver;->mResolver:Landroid/content/ContentResolver;
  
!     const-string v1, "search_global_search_activity"
  
      invoke-static {v1}, Landroid/provider/Settings$Secure;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
  
--- 27,47 ----
      .parameter "resolver"
  
      .prologue
+     .line 124
      iput-object p1, p0, Landroid/server/search/SearchManagerService$GlobalSearchProviderObserver;->this$0:Landroid/server/search/SearchManagerService;
  
+     .line 125
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/database/ContentObserver;-><init>(Landroid/os/Handler;)V
  
+     .line 126
      iput-object p2, p0, Landroid/server/search/SearchManagerService$GlobalSearchProviderObserver;->mResolver:Landroid/content/ContentResolver;
  
+     .line 127
      iget-object v0, p0, Landroid/server/search/SearchManagerService$GlobalSearchProviderObserver;->mResolver:Landroid/content/ContentResolver;
  
!     const-string/jumbo v1, "search_global_search_activity"
  
      invoke-static {v1}, Landroid/provider/Settings$Secure;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
  
***************
*** 47,52 ****
--- 51,57 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 131
      return-void
  .end method
  
***************
*** 57,62 ****
--- 62,68 ----
      .parameter "selfChange"
  
      .prologue
+     .line 135
      iget-object v1, p0, Landroid/server/search/SearchManagerService$GlobalSearchProviderObserver;->this$0:Landroid/server/search/SearchManagerService;
  
      #calls: Landroid/server/search/SearchManagerService;->getSearchables()Landroid/server/search/Searchables;
***************
*** 66,82 ****
--- 72,91 ----
  
      invoke-virtual {v1}, Landroid/server/search/Searchables;->buildSearchableList()V
  
+     .line 136
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.search.action.GLOBAL_SEARCH_ACTIVITY_CHANGED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 137
      .local v0, intent:Landroid/content/Intent;
      const/high16 v1, 0x2000
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 138
      iget-object v1, p0, Landroid/server/search/SearchManagerService$GlobalSearchProviderObserver;->this$0:Landroid/server/search/SearchManagerService;
  
      #getter for: Landroid/server/search/SearchManagerService;->mContext:Landroid/content/Context;
***************
*** 86,90 ****
--- 95,100 ----
  
      invoke-virtual {v1, v0}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 139
      return-void
  .end method
