*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/service/wallpaper/WallpaperService$Engine.smali	2012-03-23 19:36:08.372833663 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/service/wallpaper/WallpaperService$Engine.smali.original	2012-03-23 19:34:44.997313313 -0500
***************
*** 112,195 ****
--- 112,212 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 124
      iput-object p1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->this$0:Landroid/service/wallpaper/WallpaperService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 132
      iput-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mInitializing:Z
  
+     .line 134
      iput-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mScreenOn:Z
  
+     .line 150
      const/16 v0, 0x10
  
      iput v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowFlags:I
  
+     .line 151
      const/4 v0, 0x4
  
      iput v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowPrivateFlags:I
  
+     .line 153
      iget v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowFlags:I
  
      iput v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mCurWindowFlags:I
  
+     .line 154
      iget v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowPrivateFlags:I
  
      iput v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mCurWindowPrivateFlags:I
  
+     .line 155
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mVisibleInsets:Landroid/graphics/Rect;
  
+     .line 156
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWinFrame:Landroid/graphics/Rect;
  
+     .line 157
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mContentInsets:Landroid/graphics/Rect;
  
+     .line 158
      new-instance v0, Landroid/content/res/Configuration;
  
      invoke-direct {v0}, Landroid/content/res/Configuration;-><init>()V
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mConfiguration:Landroid/content/res/Configuration;
  
+     .line 160
      new-instance v0, Landroid/view/WindowManager$LayoutParams;
  
      invoke-direct {v0}, Landroid/view/WindowManager$LayoutParams;-><init>()V
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
  
+     .line 165
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mLock:Ljava/lang/Object;
  
+     .line 174
      new-instance v0, Landroid/service/wallpaper/WallpaperService$Engine$1;
  
      invoke-direct {v0, p0}, Landroid/service/wallpaper/WallpaperService$Engine$1;-><init>(Landroid/service/wallpaper/WallpaperService$Engine;)V
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 187
      new-instance v0, Landroid/service/wallpaper/WallpaperService$Engine$2;
  
      invoke-direct {v0, p0}, Landroid/service/wallpaper/WallpaperService$Engine$2;-><init>(Landroid/service/wallpaper/WallpaperService$Engine;)V
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
  
+     .line 232
      new-instance v0, Landroid/service/wallpaper/WallpaperService$Engine$3;
  
      invoke-direct {v0, p0}, Landroid/service/wallpaper/WallpaperService$Engine$3;-><init>(Landroid/service/wallpaper/WallpaperService$Engine;)V
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mInputHandler:Landroid/view/InputHandler;
  
+     .line 249
      new-instance v0, Landroid/service/wallpaper/WallpaperService$Engine$4;
  
      invoke-direct {v0, p0}, Landroid/service/wallpaper/WallpaperService$Engine$4;-><init>(Landroid/service/wallpaper/WallpaperService$Engine;)V
***************
*** 205,210 ****
--- 222,228 ----
      .parameter "x1"
  
      .prologue
+     .line 124
      invoke-direct {p0, p1}, Landroid/service/wallpaper/WallpaperService$Engine;->dispatchPointer(Landroid/view/MotionEvent;)V
  
      return-void
***************
*** 215,230 ****
--- 233,251 ----
      .parameter "event"
  
      .prologue
+     .line 527
      invoke-virtual {p1}, Landroid/view/MotionEvent;->isTouchEvent()Z
  
      move-result v1
  
      if-eqz v1, :cond_0
  
+     .line 528
      iget-object v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 529
      :try_start_0
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getAction()I
  
***************
*** 234,246 ****
--- 255,270 ----
  
      if-ne v1, v3, :cond_1
  
+     .line 530
      iput-object p1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mPendingMove:Landroid/view/MotionEvent;
  
+     .line 534
      :goto_0
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 535
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      const/16 v2, 0x2738
***************
*** 249,263 ****
--- 273,290 ----
  
      move-result-object v0
  
+     .line 536
      .local v0, msg:Landroid/os/Message;
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      invoke-virtual {v1, v0}, Lcom/android/internal/os/HandlerCaller;->sendMessage(Landroid/os/Message;)V
  
+     .line 538
      .end local v0           #msg:Landroid/os/Message;
      :cond_0
      return-void
  
+     .line 532
      :cond_1
      const/4 v1, 0x0
  
***************
*** 266,271 ****
--- 293,299 ----
  
      goto :goto_0
  
+     .line 534
      :catchall_0
      move-exception v1
  
***************
*** 285,300 ****
--- 313,332 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 734
      iget-boolean v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mDestroyed:Z
  
      if-eqz v1, :cond_0
  
+     .line 761
      :goto_0
      return-void
  
+     .line 738
      :cond_0
      iput-object p1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mIWallpaperEngine:Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;
  
+     .line 739
      #getter for: Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
      invoke-static {p1}, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->access$100(Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;)Lcom/android/internal/os/HandlerCaller;
  
***************
*** 302,323 ****
--- 334,360 ----
  
      iput-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
+     .line 740
      iget-object v1, p1, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->mConnection:Landroid/service/wallpaper/IWallpaperConnection;
  
      iput-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mConnection:Landroid/service/wallpaper/IWallpaperConnection;
  
+     .line 741
      iget-object v1, p1, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->mWindowToken:Landroid/os/IBinder;
  
      iput-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowToken:Landroid/os/IBinder;
  
+     .line 742
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
  
      invoke-virtual {v1}, Lcom/android/internal/view/BaseSurfaceHolder;->setSizeFromLayout()V
  
+     .line 743
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mInitializing:Z
  
+     .line 744
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->this$0:Landroid/service/wallpaper/WallpaperService;
  
      invoke-virtual {v1}, Landroid/service/wallpaper/WallpaperService;->getMainLooper()Landroid/os/Looper;
***************
*** 330,344 ****
  
      iput-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
  
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindow:Lcom/android/internal/view/BaseIWindow;
  
      iget-object v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
  
      invoke-virtual {v1, v2}, Lcom/android/internal/view/BaseIWindow;->setSession(Landroid/view/IWindowSession;)V
  
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->this$0:Landroid/service/wallpaper/WallpaperService;
  
!     const-string v2, "power"
  
      invoke-virtual {v1, v2}, Landroid/service/wallpaper/WallpaperService;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
--- 367,383 ----
  
      iput-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
  
+     .line 746
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindow:Lcom/android/internal/view/BaseIWindow;
  
      iget-object v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
  
      invoke-virtual {v1, v2}, Lcom/android/internal/view/BaseIWindow;->setSession(Landroid/view/IWindowSession;)V
  
+     .line 748
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->this$0:Landroid/service/wallpaper/WallpaperService;
  
!     const-string/jumbo v2, "power"
  
      invoke-virtual {v1, v2}, Landroid/service/wallpaper/WallpaperService;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
***************
*** 352,384 ****
--- 391,431 ----
  
      iput-boolean v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mScreenOn:Z
  
+     .line 750
      new-instance v0, Landroid/content/IntentFilter;
  
      invoke-direct {v0}, Landroid/content/IntentFilter;-><init>()V
  
+     .line 751
      .local v0, filter:Landroid/content/IntentFilter;
      const-string v1, "android.intent.action.SCREEN_ON"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 752
      const-string v1, "android.intent.action.SCREEN_OFF"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 753
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->this$0:Landroid/service/wallpaper/WallpaperService;
  
      iget-object v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mReceiver:Landroid/content/BroadcastReceiver;
  
      invoke-virtual {v1, v2, v0}, Landroid/service/wallpaper/WallpaperService;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 756
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
  
      invoke-virtual {p0, v1}, Landroid/service/wallpaper/WallpaperService$Engine;->onCreate(Landroid/view/SurfaceHolder;)V
  
+     .line 758
      iput-boolean v3, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mInitializing:Z
  
+     .line 759
      iput-boolean v3, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mReportedVisible:Z
  
+     .line 760
      invoke-virtual {p0, v3, v3, v3}, Landroid/service/wallpaper/WallpaperService$Engine;->updateSurface(ZZZ)V
  
      goto :goto_0
***************
*** 390,440 ****
--- 437,500 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 881
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mDestroyed:Z
  
      if-eqz v0, :cond_1
  
+     .line 922
      :cond_0
      :goto_0
      return-void
  
+     .line 885
      :cond_1
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mDestroyed:Z
  
+     .line 887
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mVisible:Z
  
      if-eqz v0, :cond_2
  
+     .line 888
      iput-boolean v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mVisible:Z
  
+     .line 890
      invoke-virtual {p0, v2}, Landroid/service/wallpaper/WallpaperService$Engine;->onVisibilityChanged(Z)V
  
+     .line 893
      :cond_2
      invoke-virtual {p0}, Landroid/service/wallpaper/WallpaperService$Engine;->reportSurfaceDestroyed()V
  
+     .line 896
      invoke-virtual {p0}, Landroid/service/wallpaper/WallpaperService$Engine;->onDestroy()V
  
+     .line 898
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->this$0:Landroid/service/wallpaper/WallpaperService;
  
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mReceiver:Landroid/content/BroadcastReceiver;
  
      invoke-virtual {v0, v1}, Landroid/service/wallpaper/WallpaperService;->unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  
+     .line 900
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mCreated:Z
  
      if-eqz v0, :cond_0
  
+     .line 905
      :try_start_0
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mInputChannel:Landroid/view/InputChannel;
  
      if-eqz v0, :cond_3
  
+     .line 906
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mInputChannel:Landroid/view/InputChannel;
  
      invoke-static {v0}, Landroid/view/InputQueue;->unregisterInputChannel(Landroid/view/InputChannel;)V
  
+     .line 909
      :cond_3
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
  
***************
*** 444,449 ****
--- 504,510 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 912
      :goto_1
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
  
***************
*** 451,472 ****
--- 512,538 ----
  
      invoke-virtual {v0}, Landroid/view/Surface;->release()V
  
+     .line 913
      iput-boolean v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mCreated:Z
  
+     .line 917
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mInputChannel:Landroid/view/InputChannel;
  
      if-eqz v0, :cond_0
  
+     .line 918
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mInputChannel:Landroid/view/InputChannel;
  
      invoke-virtual {v0}, Landroid/view/InputChannel;->dispose()V
  
+     .line 919
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mInputChannel:Landroid/view/InputChannel;
  
      goto :goto_0
  
+     .line 910
      :catch_0
      move-exception v0
  
***************
*** 478,487 ****
--- 544,555 ----
      .parameter "cmd"
  
      .prologue
+     .line 849
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mDestroyed:Z
  
      if-nez v0, :cond_1
  
+     .line 850
      iget-object v1, p1, Landroid/service/wallpaper/WallpaperService$WallpaperCommand;->action:Ljava/lang/String;
  
      iget v2, p1, Landroid/service/wallpaper/WallpaperService$WallpaperCommand;->x:I
***************
*** 500,511 ****
--- 568,581 ----
  
      move-result-object v7
  
+     .line 855
      .local v7, result:Landroid/os/Bundle;
      :goto_0
      iget-boolean v0, p1, Landroid/service/wallpaper/WallpaperService$WallpaperCommand;->sync:Z
  
      if-eqz v0, :cond_0
  
+     .line 858
      :try_start_0
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
  
***************
*** 519,528 ****
--- 589,600 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 862
      :cond_0
      :goto_1
      return-void
  
+     .line 853
      .end local v7           #result:Landroid/os/Bundle;
      :cond_1
      const/4 v7, 0x0
***************
*** 530,535 ****
--- 602,608 ----
      .restart local v7       #result:Landroid/os/Bundle;
      goto :goto_0
  
+     .line 859
      :catch_0
      move-exception v0
  
***************
*** 542,565 ****
--- 615,644 ----
      .parameter "desiredHeight"
  
      .prologue
+     .line 764
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mDestroyed:Z
  
      if-nez v0, :cond_0
  
+     .line 767
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mIWallpaperEngine:Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;
  
      iput p1, v0, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->mReqWidth:I
  
+     .line 768
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mIWallpaperEngine:Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;
  
      iput p2, v0, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->mReqHeight:I
  
+     .line 769
      invoke-virtual {p0, p1, p2}, Landroid/service/wallpaper/WallpaperService$Engine;->onDesiredSizeChanged(II)V
  
+     .line 770
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Landroid/service/wallpaper/WallpaperService$Engine;->doOffsetsChanged(Z)V
  
+     .line 772
      :cond_0
      return-void
  .end method
***************
*** 573,586 ****
--- 652,668 ----
  
      const/4 v0, 0x0
  
+     .line 801
      iget-boolean v10, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mDestroyed:Z
  
      if-eqz v10, :cond_1
  
+     .line 845
      :cond_0
      :goto_0
      return-void
  
+     .line 805
      :cond_1
      if-nez p1, :cond_2
  
***************
*** 588,634 ****
--- 670,728 ----
  
      if-eqz v10, :cond_0
  
+     .line 814
      :cond_2
      iget-object v10, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mLock:Ljava/lang/Object;
  
      monitor-enter v10
  
+     .line 815
      :try_start_0
      iget v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mPendingXOffset:F
  
+     .line 816
      .local v1, xOffset:F
      iget v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mPendingYOffset:F
  
+     .line 817
      .local v2, yOffset:F
      iget v3, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mPendingXOffsetStep:F
  
+     .line 818
      .local v3, xOffsetStep:F
      iget v4, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mPendingYOffsetStep:F
  
+     .line 819
      .local v4, yOffsetStep:F
      iget-boolean v9, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mPendingSync:Z
  
+     .line 820
      .local v9, sync:Z
      const/4 v11, 0x0
  
      iput-boolean v11, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mPendingSync:Z
  
+     .line 821
      const/4 v11, 0x0
  
      iput-boolean v11, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mOffsetMessageEnqueued:Z
  
+     .line 822
      monitor-exit v10
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 824
      iget-boolean v10, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceCreated:Z
  
      if-eqz v10, :cond_3
  
+     .line 825
      iget-boolean v10, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mReportedVisible:Z
  
      if-eqz v10, :cond_6
  
+     .line 828
      iget-object v10, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mIWallpaperEngine:Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;
  
      iget v10, v10, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->mReqWidth:I
***************
*** 637,642 ****
--- 731,737 ----
  
      sub-int v8, v10, v11
  
+     .line 829
      .local v8, availw:I
      if-lez v8, :cond_4
  
***************
*** 650,655 ****
--- 745,751 ----
  
      neg-int v5, v10
  
+     .line 830
      .local v5, xPixels:I
      :goto_1
      iget-object v10, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mIWallpaperEngine:Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;
***************
*** 660,665 ****
--- 756,762 ----
  
      sub-int v7, v10, v11
  
+     .line 831
      .local v7, availh:I
      if-lez v7, :cond_5
  
***************
*** 677,684 ****
--- 774,783 ----
      :goto_2
      move-object v0, p0
  
+     .line 832
      invoke-virtual/range {v0 .. v6}, Landroid/service/wallpaper/WallpaperService$Engine;->onOffsetsChanged(FFFFII)V
  
+     .line 838
      .end local v5           #xPixels:I
      .end local v6           #yPixels:I
      .end local v7           #availh:I
***************
*** 687,692 ****
--- 786,792 ----
      :goto_3
      if-eqz v9, :cond_0
  
+     .line 841
      :try_start_1
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
  
***************
*** 702,712 ****
--- 802,814 ----
  
      goto :goto_0
  
+     .line 842
      :catch_0
      move-exception v0
  
      goto :goto_0
  
+     .line 822
      .end local v1           #xOffset:F
      .end local v2           #yOffset:F
      .end local v3           #xOffsetStep:F
***************
*** 731,736 ****
--- 833,839 ----
      :cond_4
      move v5, v0
  
+     .line 829
      goto :goto_1
  
      .restart local v5       #xPixels:I
***************
*** 738,745 ****
--- 841,850 ----
      :cond_5
      move v6, v0
  
+     .line 831
      goto :goto_2
  
+     .line 834
      .end local v5           #xPixels:I
      .end local v7           #availh:I
      .end local v8           #availw:I
***************
*** 756,769 ****
--- 861,878 ----
      .parameter "visible"
  
      .prologue
+     .line 775
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mDestroyed:Z
  
      if-nez v0, :cond_0
  
+     .line 776
      iput-boolean p1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mVisible:Z
  
+     .line 777
      invoke-virtual {p0}, Landroid/service/wallpaper/WallpaperService$Engine;->reportVisibility()V
  
+     .line 779
      :cond_0
      return-void
  .end method
***************
*** 776,781 ****
--- 885,891 ----
      .parameter "args"
  
      .prologue
+     .line 487
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mInitializing="
***************
*** 786,791 ****
--- 896,902 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 488
      const-string v0, " mDestroyed="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 794,802 ****
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string v0, "mVisible="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
--- 905,914 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 489
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string/jumbo v0, "mVisible="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 804,809 ****
--- 916,922 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 490
      const-string v0, " mScreenOn="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 812,817 ****
--- 925,931 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 491
      const-string v0, " mReportedVisible="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 820,825 ****
--- 934,940 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 492
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mCreated="
***************
*** 830,835 ****
--- 945,951 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 493
      const-string v0, " mSurfaceCreated="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 838,843 ****
--- 954,960 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 494
      const-string v0, " mIsCreating="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 846,851 ****
--- 963,969 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 495
      const-string v0, " mDrawingAllowed="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 854,862 ****
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string v0, "mWidth="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
--- 972,981 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 496
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string/jumbo v0, "mWidth="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 864,869 ****
--- 983,989 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 497
      const-string v0, " mCurWidth="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 872,877 ****
--- 992,998 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 498
      const-string v0, " mHeight="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 880,885 ****
--- 1001,1007 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 499
      const-string v0, " mCurHeight="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 888,896 ****
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(I)V
  
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string v0, "mType="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
--- 1010,1019 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(I)V
  
+     .line 500
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string/jumbo v0, "mType="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 898,903 ****
--- 1021,1027 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 501
      const-string v0, " mWindowFlags="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 906,911 ****
--- 1030,1036 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 502
      const-string v0, " mCurWindowFlags="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 914,919 ****
--- 1039,1045 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(I)V
  
+     .line 503
      const-string v0, " mWindowPrivateFlags="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 922,927 ****
--- 1048,1054 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 504
      const-string v0, " mCurWindowPrivateFlags="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 930,941 ****
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(I)V
  
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string v0, "mVisibleInsets="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mVisibleInsets:Landroid/graphics/Rect;
  
      invoke-virtual {v0}, Landroid/graphics/Rect;->toShortString()Ljava/lang/String;
--- 1057,1070 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(I)V
  
+     .line 505
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string/jumbo v0, "mVisibleInsets="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 506
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mVisibleInsets:Landroid/graphics/Rect;
  
      invoke-virtual {v0}, Landroid/graphics/Rect;->toShortString()Ljava/lang/String;
***************
*** 944,949 ****
--- 1073,1079 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 507
      const-string v0, " mWinFrame="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 956,961 ****
--- 1086,1092 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 508
      const-string v0, " mContentInsets="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 968,973 ****
--- 1099,1105 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 509
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mConfiguration="
***************
*** 978,983 ****
--- 1110,1116 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 510
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mLayout="
***************
*** 988,997 ****
--- 1121,1132 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 511
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 512
      :try_start_0
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 1003,1008 ****
--- 1138,1144 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(F)V
  
+     .line 513
      const-string v0, " mPendingXOffset="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1011,1026 ****
--- 1147,1165 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(F)V
  
+     .line 514
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mPendingXOffsetStep="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 515
      iget v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mPendingXOffsetStep:F
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(F)V
  
+     .line 516
      const-string v0, " mPendingXOffsetStep="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1029,1044 ****
--- 1168,1186 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(F)V
  
+     .line 517
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mOffsetMessageEnqueued="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 518
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mOffsetMessageEnqueued:Z
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 519
      const-string v0, " mPendingSync="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 1047,1056 ****
--- 1189,1200 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 520
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mPendingMove:Landroid/view/MotionEvent;
  
      if-eqz v0, :cond_0
  
+     .line 521
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mPendingMove="
***************
*** 1061,1071 ****
--- 1205,1218 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 523
      :cond_0
      monitor-exit v1
  
+     .line 524
      return-void
  
+     .line 523
      :catchall_0
      move-exception v0
  
***************
*** 1080,1085 ****
--- 1227,1233 ----
      .locals 1
  
      .prologue
+     .line 329
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mIWallpaperEngine:Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;
  
      iget v0, v0, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->mReqHeight:I
***************
*** 1091,1096 ****
--- 1239,1245 ----
      .locals 1
  
      .prologue
+     .line 320
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mIWallpaperEngine:Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;
  
      iget v0, v0, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->mReqWidth:I
***************
*** 1102,1107 ****
--- 1251,1257 ----
      .locals 1
  
      .prologue
+     .line 311
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
  
      return-object v0
***************
*** 1111,1116 ****
--- 1261,1267 ----
      .locals 1
  
      .prologue
+     .line 347
      iget-object v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mIWallpaperEngine:Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;
  
      iget-boolean v0, v0, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;->mIsPreview:Z
***************
*** 1122,1127 ****
--- 1273,1279 ----
      .locals 1
  
      .prologue
+     .line 338
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mReportedVisible:Z
  
      return v0
***************
*** 1137,1142 ****
--- 1289,1295 ----
      .parameter "resultRequested"
  
      .prologue
+     .line 448
      const/4 v0, 0x0
  
      return-object v0
***************
*** 1147,1152 ****
--- 1300,1306 ----
      .parameter "surfaceHolder"
  
      .prologue
+     .line 390
      return-void
  .end method
  
***************
*** 1156,1161 ****
--- 1310,1316 ----
      .parameter "desiredHeight"
  
      .prologue
+     .line 456
      return-void
  .end method
  
***************
*** 1163,1168 ****
--- 1318,1324 ----
      .locals 0
  
      .prologue
+     .line 398
      return-void
  .end method
  
***************
*** 1176,1181 ****
--- 1332,1338 ----
      .parameter "yPixelOffset"
  
      .prologue
+     .line 427
      return-void
  .end method
  
***************
*** 1187,1192 ****
--- 1344,1350 ----
      .parameter "height"
  
      .prologue
+     .line 463
      return-void
  .end method
  
***************
*** 1195,1200 ****
--- 1353,1359 ----
      .parameter "holder"
  
      .prologue
+     .line 477
      return-void
  .end method
  
***************
*** 1203,1208 ****
--- 1362,1368 ----
      .parameter "holder"
  
      .prologue
+     .line 484
      return-void
  .end method
  
***************
*** 1211,1216 ****
--- 1371,1377 ----
      .parameter "holder"
  
      .prologue
+     .line 470
      return-void
  .end method
  
***************
*** 1219,1224 ****
--- 1380,1386 ----
      .parameter "event"
  
      .prologue
+     .line 416
      return-void
  .end method
  
***************
*** 1227,1232 ****
--- 1389,1395 ----
      .parameter "visible"
  
      .prologue
+     .line 406
      return-void
  .end method
  
***************
*** 1234,1260 ****
--- 1397,1429 ----
      .locals 6
  
      .prologue
+     .line 865
      iget-boolean v5, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceCreated:Z
  
      if-eqz v5, :cond_1
  
+     .line 866
      const/4 v5, 0x0
  
      iput-boolean v5, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceCreated:Z
  
+     .line 867
      iget-object v5, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
  
      invoke-virtual {v5}, Lcom/android/internal/view/BaseSurfaceHolder;->ungetCallbacks()V
  
+     .line 868
      iget-object v5, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
  
      invoke-virtual {v5}, Lcom/android/internal/view/BaseSurfaceHolder;->getCallbacks()[Landroid/view/SurfaceHolder$Callback;
  
      move-result-object v2
  
+     .line 869
      .local v2, callbacks:[Landroid/view/SurfaceHolder$Callback;
      if-eqz v2, :cond_0
  
+     .line 870
      move-object v0, v2
  
      .local v0, arr$:[Landroid/view/SurfaceHolder$Callback;
***************
*** 1269,1283 ****
--- 1438,1455 ----
  
      aget-object v1, v0, v3
  
+     .line 871
      .local v1, c:Landroid/view/SurfaceHolder$Callback;
      iget-object v5, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
  
      invoke-interface {v1, v5}, Landroid/view/SurfaceHolder$Callback;->surfaceDestroyed(Landroid/view/SurfaceHolder;)V
  
+     .line 870
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 876
      .end local v0           #arr$:[Landroid/view/SurfaceHolder$Callback;
      .end local v1           #c:Landroid/view/SurfaceHolder$Callback;
      .end local v3           #i$:I
***************
*** 1287,1292 ****
--- 1459,1465 ----
  
      invoke-virtual {p0, v5}, Landroid/service/wallpaper/WallpaperService$Engine;->onSurfaceDestroyed(Landroid/view/SurfaceHolder;)V
  
+     .line 878
      .end local v2           #callbacks:[Landroid/view/SurfaceHolder$Callback;
      :cond_1
      return-void
***************
*** 1298,1307 ****
--- 1471,1482 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 782
      iget-boolean v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mDestroyed:Z
  
      if-nez v2, :cond_1
  
+     .line 783
      iget-boolean v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mVisible:Z
  
      if-eqz v2, :cond_2
***************
*** 1312,1334 ****
--- 1487,1516 ----
  
      const/4 v0, 0x1
  
+     .line 784
      .local v0, visible:Z
      :goto_0
      iget-boolean v2, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mReportedVisible:Z
  
      if-eq v2, v0, :cond_1
  
+     .line 785
      iput-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mReportedVisible:Z
  
+     .line 788
      if-eqz v0, :cond_0
  
+     .line 792
      invoke-virtual {p0, v1}, Landroid/service/wallpaper/WallpaperService$Engine;->doOffsetsChanged(Z)V
  
+     .line 793
      invoke-virtual {p0, v1, v1, v1}, Landroid/service/wallpaper/WallpaperService$Engine;->updateSurface(ZZZ)V
  
+     .line 795
      :cond_0
      invoke-virtual {p0, v0}, Landroid/service/wallpaper/WallpaperService$Engine;->onVisibilityChanged(Z)V
  
+     .line 798
      .end local v0           #visible:Z
      :cond_1
      return-void
***************
*** 1336,1341 ****
--- 1518,1524 ----
      :cond_2
      move v0, v1
  
+     .line 783
      goto :goto_0
  .end method
  
***************
*** 1346,1351 ****
--- 1529,1535 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 375
      if-eqz p1, :cond_1
  
      iget v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowPrivateFlags:I
***************
*** 1355,1369 ****
--- 1539,1557 ----
      :goto_0
      iput v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowPrivateFlags:I
  
+     .line 380
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mCreated:Z
  
      if-eqz v0, :cond_0
  
+     .line 381
      invoke-virtual {p0, v1, v1, v1}, Landroid/service/wallpaper/WallpaperService$Engine;->updateSurface(ZZZ)V
  
+     .line 383
      :cond_0
      return-void
  
+     .line 375
      :cond_1
      iget v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowPrivateFlags:I
  
***************
*** 1379,1384 ****
--- 1567,1573 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 358
      if-eqz p1, :cond_1
  
      iget v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowFlags:I
***************
*** 1388,1402 ****
--- 1577,1595 ----
      :goto_0
      iput v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowFlags:I
  
+     .line 361
      iget-boolean v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mCreated:Z
  
      if-eqz v0, :cond_0
  
+     .line 362
      invoke-virtual {p0, v1, v1, v1}, Landroid/service/wallpaper/WallpaperService$Engine;->updateSurface(ZZZ)V
  
+     .line 364
      :cond_0
      return-void
  
+     .line 358
      :cond_1
      iget v0, p0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowFlags:I
  
***************
*** 1412,1429 ****
--- 1605,1625 ----
      .parameter "redrawNeeded"
  
      .prologue
+     .line 541
      move-object/from16 v0, p0
  
      iget-boolean v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mDestroyed:Z
  
      if-eqz v2, :cond_0
  
+     .line 542
      const-string v2, "WallpaperService"
  
      const-string v3, "Ignoring updateSurface: destroyed"
  
      invoke-static {v2, v3}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 545
      :cond_0
      move-object/from16 v0, p0
  
***************
*** 1433,1443 ****
--- 1629,1641 ----
  
      move-result v26
  
+     .line 546
      .local v26, myWidth:I
      if-gtz v26, :cond_1
  
      const/16 v26, -0x1
  
+     .line 547
      :cond_1
      move-object/from16 v0, p0
  
***************
*** 1447,1457 ****
--- 1645,1657 ----
  
      move-result v25
  
+     .line 548
      .local v25, myHeight:I
      if-gtz v25, :cond_2
  
      const/16 v25, -0x1
  
+     .line 550
      :cond_2
      move-object/from16 v0, p0
  
***************
*** 1461,1466 ****
--- 1661,1667 ----
  
      const/16 v18, 0x1
  
+     .line 551
      .local v18, creating:Z
      :goto_0
      move-object/from16 v0, p0
***************
*** 1471,1476 ****
--- 1672,1678 ----
  
      const/16 v29, 0x1
  
+     .line 552
      .local v29, surfaceCreating:Z
      :goto_1
      move-object/from16 v0, p0
***************
*** 1489,1494 ****
--- 1691,1697 ----
  
      const/16 v21, 0x1
  
+     .line 553
      .local v21, formatChanged:Z
      :goto_2
      move-object/from16 v0, p0
***************
*** 1510,1515 ****
--- 1713,1719 ----
      :cond_3
      const/16 v28, 0x1
  
+     .line 554
      .local v28, sizeChanged:Z
      :goto_3
      move-object/from16 v0, p0
***************
*** 1528,1533 ****
--- 1732,1738 ----
  
      const/16 v30, 0x1
  
+     .line 555
      .local v30, typeChanged:Z
      :goto_4
      move-object/from16 v0, p0
***************
*** 1553,1558 ****
--- 1758,1764 ----
      :cond_4
      const/16 v20, 0x1
  
+     .line 557
      .local v20, flagsChanged:Z
      :goto_5
      if-nez p1, :cond_5
***************
*** 1571,1576 ****
--- 1777,1783 ----
  
      if-eqz p3, :cond_6
  
+     .line 564
      :cond_5
      :try_start_0
      move/from16 v0, v26
***************
*** 1579,1590 ****
--- 1786,1799 ----
  
      iput v0, v1, Landroid/service/wallpaper/WallpaperService$Engine;->mWidth:I
  
+     .line 565
      move/from16 v0, v25
  
      move-object/from16 v1, p0
  
      iput v0, v1, Landroid/service/wallpaper/WallpaperService$Engine;->mHeight:I
  
+     .line 566
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 1597,1602 ****
--- 1806,1812 ----
  
      iput v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mFormat:I
  
+     .line 567
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 1609,1614 ****
--- 1819,1825 ----
  
      iput v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mType:I
  
+     .line 569
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1617,1622 ****
--- 1828,1834 ----
  
      iput v3, v2, Landroid/view/WindowManager$LayoutParams;->x:I
  
+     .line 570
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1625,1630 ****
--- 1837,1843 ----
  
      iput v3, v2, Landroid/view/WindowManager$LayoutParams;->y:I
  
+     .line 571
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1633,1638 ****
--- 1846,1852 ----
  
      iput v0, v2, Landroid/view/WindowManager$LayoutParams;->width:I
  
+     .line 572
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1641,1646 ****
--- 1855,1861 ----
  
      iput v0, v2, Landroid/view/WindowManager$LayoutParams;->height:I
  
+     .line 574
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1651,1656 ****
--- 1866,1872 ----
  
      iput v3, v2, Landroid/view/WindowManager$LayoutParams;->format:I
  
+     .line 576
      move-object/from16 v0, p0
  
      iget v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowFlags:I
***************
*** 1659,1664 ****
--- 1875,1881 ----
  
      iput v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mCurWindowFlags:I
  
+     .line 577
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1675,1680 ****
--- 1892,1898 ----
  
      iput v3, v2, Landroid/view/WindowManager$LayoutParams;->flags:I
  
+     .line 582
      move-object/from16 v0, p0
  
      iget v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mWindowPrivateFlags:I
***************
*** 1683,1688 ****
--- 1901,1907 ----
  
      iput v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mCurWindowPrivateFlags:I
  
+     .line 583
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1693,1698 ****
--- 1912,1918 ----
  
      iput v3, v2, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
+     .line 585
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1703,1708 ****
--- 1923,1929 ----
  
      iput v3, v2, Landroid/view/WindowManager$LayoutParams;->memoryType:I
  
+     .line 586
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1713,1724 ****
--- 1934,1947 ----
  
      iput-object v3, v2, Landroid/view/WindowManager$LayoutParams;->token:Landroid/os/IBinder;
  
+     .line 588
      move-object/from16 v0, p0
  
      iget-boolean v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mCreated:Z
  
      if-nez v2, :cond_e
  
+     .line 589
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1731,1736 ****
--- 1954,1960 ----
  
      iput v3, v2, Landroid/view/WindowManager$LayoutParams;->type:I
  
+     .line 590
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1739,1744 ****
--- 1963,1969 ----
  
      iput v3, v2, Landroid/view/WindowManager$LayoutParams;->gravity:I
  
+     .line 591
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1757,1762 ****
--- 1982,1988 ----
  
      invoke-virtual {v2, v3}, Landroid/view/WindowManager$LayoutParams;->setTitle(Ljava/lang/CharSequence;)V
  
+     .line 592
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mLayout:Landroid/view/WindowManager$LayoutParams;
***************
*** 1765,1770 ****
--- 1991,1997 ----
  
      iput v3, v2, Landroid/view/WindowManager$LayoutParams;->windowAnimations:I
  
+     .line 594
      new-instance v2, Landroid/view/InputChannel;
  
      invoke-direct {v2}, Landroid/view/InputChannel;-><init>()V
***************
*** 1773,1778 ****
--- 2000,2006 ----
  
      iput-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mInputChannel:Landroid/view/InputChannel;
  
+     .line 595
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
***************
*** 1807,1822 ****
--- 2035,2053 ----
  
      if-gez v2, :cond_d
  
+     .line 597
      const-string v2, "WallpaperService"
  
      const-string v3, "Failed to add window while updating wallpaper surface."
  
      invoke-static {v2, v3}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 730
      :cond_6
      :goto_6
      return-void
  
+     .line 550
      .end local v18           #creating:Z
      .end local v20           #flagsChanged:Z
      .end local v21           #formatChanged:Z
***************
*** 1828,1863 ****
--- 2059,2100 ----
  
      goto/16 :goto_0
  
+     .line 551
      .restart local v18       #creating:Z
      :cond_8
      const/16 v29, 0x0
  
      goto/16 :goto_1
  
+     .line 552
      .restart local v29       #surfaceCreating:Z
      :cond_9
      const/16 v21, 0x0
  
      goto/16 :goto_2
  
+     .line 553
      .restart local v21       #formatChanged:Z
      :cond_a
      const/16 v28, 0x0
  
      goto/16 :goto_3
  
+     .line 554
      .restart local v28       #sizeChanged:Z
      :cond_b
      const/16 v30, 0x0
  
      goto/16 :goto_4
  
+     .line 555
      .restart local v30       #typeChanged:Z
      :cond_c
      const/16 v20, 0x0
  
      goto/16 :goto_5
  
+     .line 600
      .restart local v20       #flagsChanged:Z
      :cond_d
      const/4 v2, 0x1
***************
*** 1866,1871 ****
--- 2103,2109 ----
  
      iput-boolean v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mCreated:Z
  
+     .line 602
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mInputChannel:Landroid/view/InputChannel;
***************
*** 1880,1885 ****
--- 2118,2124 ----
  
      invoke-static {v2, v3, v4}, Landroid/view/InputQueue;->registerInputChannel(Landroid/view/InputChannel;Landroid/view/InputHandler;Landroid/os/MessageQueue;)V
  
+     .line 606
      :cond_e
      move-object/from16 v0, p0
  
***************
*** 1889,1900 ****
--- 2128,2141 ----
  
      invoke-virtual {v2}, Ljava/util/concurrent/locks/ReentrantLock;->lock()V
  
+     .line 607
      const/4 v2, 0x1
  
      move-object/from16 v0, p0
  
      iput-boolean v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mDrawingAllowed:Z
  
+     .line 609
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
***************
*** 1951,1956 ****
--- 2192,2198 ----
  
      move-result v27
  
+     .line 617
      .local v27, relayoutResult:I
      move-object/from16 v0, p0
  
***************
*** 1960,1965 ****
--- 2202,2208 ----
  
      move-result v31
  
+     .line 618
      .local v31, w:I
      move-object/from16 v0, p0
  
***************
*** 1969,1982 ****
--- 2212,2228 ----
  
      if-eq v2, v0, :cond_f
  
+     .line 619
      const/16 v28, 0x1
  
+     .line 620
      move/from16 v0, v31
  
      move-object/from16 v1, p0
  
      iput v0, v1, Landroid/service/wallpaper/WallpaperService$Engine;->mCurWidth:I
  
+     .line 622
      :cond_f
      move-object/from16 v0, p0
  
***************
*** 1986,1991 ****
--- 2232,2238 ----
  
      move-result v22
  
+     .line 623
      .local v22, h:I
      move-object/from16 v0, p0
  
***************
*** 1995,2008 ****
--- 2242,2258 ----
  
      if-eq v2, v0, :cond_10
  
+     .line 624
      const/16 v28, 0x1
  
+     .line 625
      move/from16 v0, v22
  
      move-object/from16 v1, p0
  
      iput v0, v1, Landroid/service/wallpaper/WallpaperService$Engine;->mCurHeight:I
  
+     .line 628
      :cond_10
      move-object/from16 v0, p0
  
***************
*** 2014,2019 ****
--- 2264,2270 ----
  
      invoke-virtual {v2, v0, v1}, Lcom/android/internal/view/BaseSurfaceHolder;->setSurfaceFrameSize(II)V
  
+     .line 629
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 2022,2027 ****
--- 2273,2279 ----
  
      invoke-virtual {v2}, Ljava/util/concurrent/locks/ReentrantLock;->unlock()V
  
+     .line 631
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 2034,2045 ****
--- 2286,2299 ----
  
      if-nez v2, :cond_11
  
+     .line 632
      invoke-virtual/range {p0 .. p0}, Landroid/service/wallpaper/WallpaperService$Engine;->reportSurfaceDestroyed()V
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
      goto/16 :goto_6
  
+     .line 724
      .end local v22           #h:I
      .end local v27           #relayoutResult:I
      .end local v31           #w:I
***************
*** 2048,2059 ****
--- 2302,2315 ----
  
      goto/16 :goto_6
  
+     .line 637
      .restart local v22       #h:I
      .restart local v27       #relayoutResult:I
      .restart local v31       #w:I
      :cond_11
      const/16 v19, 0x0
  
+     .line 640
      .local v19, didSurface:Z
      :try_start_1
      move-object/from16 v0, p0
***************
*** 2062,2077 ****
--- 2318,2337 ----
  
      invoke-virtual {v2}, Lcom/android/internal/view/BaseSurfaceHolder;->ungetCallbacks()V
  
+     .line 642
      if-eqz v29, :cond_12
  
+     .line 643
      const/4 v2, 0x1
  
      move-object/from16 v0, p0
  
      iput-boolean v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mIsCreating:Z
  
+     .line 644
      const/16 v19, 0x1
  
+     .line 647
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 2080,2085 ****
--- 2340,2346 ----
  
      invoke-virtual {v0, v2}, Landroid/service/wallpaper/WallpaperService$Engine;->onSurfaceCreated(Landroid/view/SurfaceHolder;)V
  
+     .line 648
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 2088,2096 ****
--- 2349,2359 ----
  
      move-result-object v17
  
+     .line 649
      .local v17, callbacks:[Landroid/view/SurfaceHolder$Callback;
      if-eqz v17, :cond_12
  
+     .line 650
      move-object/from16 v15, v17
  
      .local v15, arr$:[Landroid/view/SurfaceHolder$Callback;
***************
*** 2111,2116 ****
--- 2374,2380 ----
  
      aget-object v16, v15, v23
  
+     .line 651
      .local v16, c:Landroid/view/SurfaceHolder$Callback;
      move-object/from16 v0, p0
  
***************
*** 2120,2129 ****
--- 2384,2395 ----
  
      invoke-interface {v0, v2}, Landroid/view/SurfaceHolder$Callback;->surfaceCreated(Landroid/view/SurfaceHolder;)V
  
+     .line 650
      add-int/lit8 v23, v23, 0x1
  
      goto :goto_7
  
+     .line 656
      .end local v15           #arr$:[Landroid/view/SurfaceHolder$Callback;
      .end local v16           #c:Landroid/view/SurfaceHolder$Callback;
      .end local v17           #callbacks:[Landroid/view/SurfaceHolder$Callback;
***************
*** 2142,2147 ****
--- 2408,2414 ----
      :goto_8
      or-int p3, p3, v2
  
+     .line 659
      if-nez p2, :cond_14
  
      if-nez v18, :cond_14
***************
*** 2152,2160 ****
--- 2419,2429 ----
  
      if-eqz v28, :cond_16
  
+     .line 672
      :cond_14
      const/16 v19, 0x1
  
+     .line 673
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 2175,2180 ****
--- 2444,2450 ----
  
      invoke-virtual {v0, v2, v3, v4, v5}, Landroid/service/wallpaper/WallpaperService$Engine;->onSurfaceChanged(Landroid/view/SurfaceHolder;III)V
  
+     .line 675
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 2183,2191 ****
--- 2453,2463 ----
  
      move-result-object v17
  
+     .line 676
      .restart local v17       #callbacks:[Landroid/view/SurfaceHolder$Callback;
      if-eqz v17, :cond_16
  
+     .line 677
      move-object/from16 v15, v17
  
      .restart local v15       #arr$:[Landroid/view/SurfaceHolder$Callback;
***************
*** 2206,2211 ****
--- 2478,2484 ----
  
      aget-object v16, v15, v23
  
+     .line 678
      .restart local v16       #c:Landroid/view/SurfaceHolder$Callback;
      move-object/from16 v0, p0
  
***************
*** 2227,2236 ****
--- 2500,2511 ----
  
      invoke-interface {v0, v2, v3, v4, v5}, Landroid/view/SurfaceHolder$Callback;->surfaceChanged(Landroid/view/SurfaceHolder;III)V
  
+     .line 677
      add-int/lit8 v23, v23, 0x1
  
      goto :goto_9
  
+     .line 656
      .end local v15           #arr$:[Landroid/view/SurfaceHolder$Callback;
      .end local v16           #c:Landroid/view/SurfaceHolder$Callback;
      .end local v17           #callbacks:[Landroid/view/SurfaceHolder$Callback;
***************
*** 2241,2249 ****
--- 2516,2526 ----
  
      goto :goto_8
  
+     .line 684
      :cond_16
      if-eqz p3, :cond_18
  
+     .line 685
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 2252,2257 ****
--- 2529,2535 ----
  
      invoke-virtual {v0, v2}, Landroid/service/wallpaper/WallpaperService$Engine;->onSurfaceRedrawNeeded(Landroid/view/SurfaceHolder;)V
  
+     .line 686
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceHolder:Lcom/android/internal/view/BaseSurfaceHolder;
***************
*** 2260,2268 ****
--- 2538,2548 ----
  
      move-result-object v17
  
+     .line 687
      .restart local v17       #callbacks:[Landroid/view/SurfaceHolder$Callback;
      if-eqz v17, :cond_18
  
+     .line 688
      move-object/from16 v15, v17
  
      .restart local v15       #arr$:[Landroid/view/SurfaceHolder$Callback;
***************
*** 2283,2288 ****
--- 2563,2569 ----
  
      aget-object v16, v15, v23
  
+     .line 689
      .restart local v16       #c:Landroid/view/SurfaceHolder$Callback;
      move-object/from16 v0, v16
  
***************
*** 2290,2295 ****
--- 2571,2577 ----
  
      if-eqz v2, :cond_17
  
+     .line 690
      check-cast v16, Landroid/view/SurfaceHolder$Callback2;
  
      .end local v16           #c:Landroid/view/SurfaceHolder$Callback;
***************
*** 2301,2311 ****
--- 2583,2595 ----
  
      invoke-interface {v0, v2}, Landroid/view/SurfaceHolder$Callback2;->surfaceRedrawNeeded(Landroid/view/SurfaceHolder;)V
  
+     .line 688
      :cond_17
      add-int/lit8 v23, v23, 0x1
  
      goto :goto_a
  
+     .line 697
      .end local v15           #arr$:[Landroid/view/SurfaceHolder$Callback;
      .end local v17           #callbacks:[Landroid/view/SurfaceHolder$Callback;
      .end local v23           #i$:I
***************
*** 2319,2336 ****
--- 2603,2623 ----
  
      if-nez v2, :cond_1a
  
+     .line 703
      move-object/from16 v0, p0
  
      iget-boolean v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mIsCreating:Z
  
      if-eqz v2, :cond_19
  
+     .line 710
      const/4 v2, 0x1
  
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v2}, Landroid/service/wallpaper/WallpaperService$Engine;->onVisibilityChanged(Z)V
  
+     .line 714
      :cond_19
      const/4 v2, 0x0
  
***************
*** 2340,2345 ****
--- 2627,2633 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 718
      :cond_1a
      const/4 v2, 0x0
  
***************
*** 2348,2361 ****
--- 2636,2652 ----
  
      iput-boolean v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mIsCreating:Z
  
+     .line 719
      const/4 v2, 0x1
  
      move-object/from16 v0, p0
  
      iput-boolean v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceCreated:Z
  
+     .line 720
      if-eqz p3, :cond_6
  
+     .line 721
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
***************
*** 2368,2373 ****
--- 2659,2665 ----
  
      goto/16 :goto_6
  
+     .line 718
      :catchall_0
      move-exception v2
  
***************
*** 2377,2390 ****
--- 2669,2685 ----
  
      iput-boolean v3, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mIsCreating:Z
  
+     .line 719
      const/4 v3, 0x1
  
      move-object/from16 v0, p0
  
      iput-boolean v3, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSurfaceCreated:Z
  
+     .line 720
      if-eqz p3, :cond_1b
  
+     .line 721
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/service/wallpaper/WallpaperService$Engine;->mSession:Landroid/view/IWindowSession;
