*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/service/wallpaper/WallpaperService$Engine$3.smali	2012-03-23 19:36:08.419834520 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/service/wallpaper/WallpaperService$Engine$3.smali.original	2012-03-23 19:34:44.997313313 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 232
      iput-object p1, p0, Landroid/service/wallpaper/WallpaperService$Engine$3;->this$1:Landroid/service/wallpaper/WallpaperService$Engine;
  
      invoke-direct {p0}, Lcom/android/internal/view/BaseInputHandler;-><init>()V
***************
*** 39,57 ****
--- 40,62 ----
      .parameter "finishedCallback"
  
      .prologue
+     .line 236
      const/4 v0, 0x0
  
+     .line 238
      .local v0, handled:Z
      :try_start_0
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getSource()I
  
      move-result v1
  
+     .line 239
      .local v1, source:I
      and-int/lit8 v2, v1, 0x2
  
      if-eqz v2, :cond_0
  
+     .line 240
      iget-object v2, p0, Landroid/service/wallpaper/WallpaperService$Engine$3;->this$1:Landroid/service/wallpaper/WallpaperService$Engine;
  
      #calls: Landroid/service/wallpaper/WallpaperService$Engine;->dispatchPointer(Landroid/view/MotionEvent;)V
***************
*** 59,71 ****
--- 64,80 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 241
      const/4 v0, 0x1
  
+     .line 244
      :cond_0
      invoke-virtual {p2, v0}, Landroid/view/InputQueue$FinishedCallback;->finished(Z)V
  
+     .line 246
      return-void
  
+     .line 244
      .end local v1           #source:I
      :catchall_0
      move-exception v2
