*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/service/wallpaper/WallpaperService$IWallpaperServiceWrapper.smali	2012-03-23 19:36:07.714821663 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/service/wallpaper/WallpaperService$IWallpaperServiceWrapper.smali.original	2012-03-23 19:34:44.994313264 -0500
***************
*** 27,38 ****
--- 27,41 ----
      .parameter "context"
  
      .prologue
+     .line 1064
      iput-object p1, p0, Landroid/service/wallpaper/WallpaperService$IWallpaperServiceWrapper;->this$0:Landroid/service/wallpaper/WallpaperService;
  
      invoke-direct {p0}, Landroid/service/wallpaper/IWallpaperService$Stub;-><init>()V
  
+     .line 1065
      iput-object p2, p0, Landroid/service/wallpaper/WallpaperService$IWallpaperServiceWrapper;->mTarget:Landroid/service/wallpaper/WallpaperService;
  
+     .line 1066
      return-void
  .end method
  
***************
*** 48,53 ****
--- 51,57 ----
      .parameter "reqHeight"
  
      .prologue
+     .line 1070
      new-instance v0, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;
  
      iget-object v1, p0, Landroid/service/wallpaper/WallpaperService$IWallpaperServiceWrapper;->this$0:Landroid/service/wallpaper/WallpaperService;
***************
*** 68,72 ****
--- 72,77 ----
  
      invoke-direct/range {v0 .. v8}, Landroid/service/wallpaper/WallpaperService$IWallpaperEngineWrapper;-><init>(Landroid/service/wallpaper/WallpaperService;Landroid/service/wallpaper/WallpaperService;Landroid/service/wallpaper/IWallpaperConnection;Landroid/os/IBinder;IZII)V
  
+     .line 1072
      return-void
  .end method
