*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/mtp/MtpDatabase.smali	2012-03-23 19:37:17.234089346 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/mtp/MtpDatabase.smali.original	2012-03-23 19:34:45.620324673 -0500
***************
*** 122,127 ****
--- 122,128 ----
  
      const/4 v2, 0x0
  
+     .line 80
      new-array v0, v3, [Ljava/lang/String;
  
      const-string v1, "_id"
***************
*** 130,135 ****
--- 131,137 ----
  
      sput-object v0, Landroid/mtp/MtpDatabase;->ID_PROJECTION:[Ljava/lang/String;
  
+     .line 83
      new-array v0, v4, [Ljava/lang/String;
  
      const-string v1, "_id"
***************
*** 142,147 ****
--- 144,150 ----
  
      sput-object v0, Landroid/mtp/MtpDatabase;->PATH_PROJECTION:[Ljava/lang/String;
  
+     .line 87
      new-array v0, v6, [Ljava/lang/String;
  
      const-string v1, "_id"
***************
*** 162,167 ****
--- 165,171 ----
  
      sput-object v0, Landroid/mtp/MtpDatabase;->PATH_SIZE_FORMAT_PROJECTION:[Ljava/lang/String;
  
+     .line 93
      const/4 v0, 0x7
  
      new-array v0, v0, [Ljava/lang/String;
***************
*** 170,176 ****
  
      aput-object v1, v0, v2
  
!     const-string v1, "storage_id"
  
      aput-object v1, v0, v3
  
--- 174,180 ----
  
      aput-object v1, v0, v2
  
!     const-string/jumbo v1, "storage_id"
  
      aput-object v1, v0, v3
  
***************
*** 178,184 ****
  
      aput-object v1, v0, v4
  
!     const-string v1, "parent"
  
      aput-object v1, v0, v5
  
--- 182,188 ----
  
      aput-object v1, v0, v4
  
!     const-string/jumbo v1, "parent"
  
      aput-object v1, v0, v5
  
***************
*** 200,209 ****
  
      sput-object v0, Landroid/mtp/MtpDatabase;->OBJECT_INFO_PROJECTION:[Ljava/lang/String;
  
!     const-string v0, "media_jni"
  
      invoke-static {v0}, Ljava/lang/System;->loadLibrary(Ljava/lang/String;)V
  
      const/16 v0, 0xa
  
      new-array v0, v0, [I
--- 204,215 ----
  
      sput-object v0, Landroid/mtp/MtpDatabase;->OBJECT_INFO_PROJECTION:[Ljava/lang/String;
  
!     .line 120
!     const-string/jumbo v0, "media_jni"
  
      invoke-static {v0}, Ljava/lang/System;->loadLibrary(Ljava/lang/String;)V
  
+     .line 519
      const/16 v0, 0xa
  
      new-array v0, v0, [I
***************
*** 212,217 ****
--- 218,224 ----
  
      sput-object v0, Landroid/mtp/MtpDatabase;->FILE_PROPERTIES:[I
  
+     .line 534
      const/16 v0, 0x13
  
      new-array v0, v0, [I
***************
*** 220,225 ****
--- 227,233 ----
  
      sput-object v0, Landroid/mtp/MtpDatabase;->AUDIO_PROPERTIES:[I
  
+     .line 559
      const/16 v0, 0xf
  
      new-array v0, v0, [I
***************
*** 228,233 ****
--- 236,242 ----
  
      sput-object v0, Landroid/mtp/MtpDatabase;->VIDEO_PROPERTIES:[I
  
+     .line 580
      const/16 v0, 0xc
  
      new-array v0, v0, [I
***************
*** 236,241 ****
--- 245,251 ----
  
      sput-object v0, Landroid/mtp/MtpDatabase;->IMAGE_PROPERTIES:[I
  
+     .line 598
      const/16 v0, 0x19
  
      new-array v0, v0, [I
***************
*** 246,251 ****
--- 256,262 ----
  
      return-void
  
+     .line 519
      :array_0
      .array-data 0x4
          0x1t 0xdct 0x0t 0x0t
***************
*** 260,265 ****
--- 271,277 ----
          0x4et 0xdct 0x0t 0x0t
      .end array-data
  
+     .line 534
      :array_1
      .array-data 0x4
          0x1t 0xdct 0x0t 0x0t
***************
*** 283,288 ****
--- 295,301 ----
          0x96t 0xdct 0x0t 0x0t
      .end array-data
  
+     .line 559
      :array_2
      .array-data 0x4
          0x1t 0xdct 0x0t 0x0t
***************
*** 302,307 ****
--- 315,321 ----
          0x48t 0xdct 0x0t 0x0t
      .end array-data
  
+     .line 580
      :array_3
      .array-data 0x4
          0x1t 0xdct 0x0t 0x0t
***************
*** 318,323 ****
--- 332,338 ----
          0x48t 0xdct 0x0t 0x0t
      .end array-data
  
+     .line 598
      :array_4
      .array-data 0x4
          0x1t 0xdct 0x0t 0x0t
***************
*** 356,390 ****
      .parameter "subDirectories"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      new-instance v10, Ljava/util/HashMap;
  
      invoke-direct {v10}, Ljava/util/HashMap;-><init>()V
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mStorageMap:Ljava/util/HashMap;
  
      new-instance v10, Ljava/util/HashMap;
  
      invoke-direct {v10}, Ljava/util/HashMap;-><init>()V
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mPropertyGroupsByProperty:Ljava/util/HashMap;
  
      new-instance v10, Ljava/util/HashMap;
  
      invoke-direct {v10}, Ljava/util/HashMap;-><init>()V
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mPropertyGroupsByFormat:Ljava/util/HashMap;
  
      invoke-direct {p0}, Landroid/mtp/MtpDatabase;->native_setup()V
  
      iput-object p1, p0, Landroid/mtp/MtpDatabase;->mContext:Landroid/content/Context;
  
      invoke-virtual {p1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v10
  
!     const-string v11, "media"
  
      invoke-virtual {v10, v11}, Landroid/content/ContentResolver;->acquireProvider(Ljava/lang/String;)Landroid/content/IContentProvider;
  
--- 371,412 ----
      .parameter "subDirectories"
  
      .prologue
+     .line 124
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 63
      new-instance v10, Ljava/util/HashMap;
  
      invoke-direct {v10}, Ljava/util/HashMap;-><init>()V
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mStorageMap:Ljava/util/HashMap;
  
+     .line 66
      new-instance v10, Ljava/util/HashMap;
  
      invoke-direct {v10}, Ljava/util/HashMap;-><init>()V
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mPropertyGroupsByProperty:Ljava/util/HashMap;
  
+     .line 70
      new-instance v10, Ljava/util/HashMap;
  
      invoke-direct {v10}, Ljava/util/HashMap;-><init>()V
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mPropertyGroupsByFormat:Ljava/util/HashMap;
  
+     .line 125
      invoke-direct {p0}, Landroid/mtp/MtpDatabase;->native_setup()V
  
+     .line 127
      iput-object p1, p0, Landroid/mtp/MtpDatabase;->mContext:Landroid/content/Context;
  
+     .line 128
      invoke-virtual {p1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v10
  
!     const-string/jumbo v11, "media"
  
      invoke-virtual {v10, v11}, Landroid/content/ContentResolver;->acquireProvider(Ljava/lang/String;)Landroid/content/IContentProvider;
  
***************
*** 392,434 ****
--- 414,466 ----
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
  
+     .line 129
      iput-object p2, p0, Landroid/mtp/MtpDatabase;->mVolumeName:Ljava/lang/String;
  
+     .line 130
      move-object/from16 v0, p3
  
      iput-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaStoragePath:Ljava/lang/String;
  
+     .line 131
      invoke-static {p2}, Landroid/provider/MediaStore$Files;->getMtpObjectsUri(Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v10
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mObjectsUri:Landroid/net/Uri;
  
+     .line 132
      new-instance v10, Landroid/media/MediaScanner;
  
      invoke-direct {v10, p1}, Landroid/media/MediaScanner;-><init>(Landroid/content/Context;)V
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mMediaScanner:Landroid/media/MediaScanner;
  
+     .line 134
      move-object/from16 v0, p4
  
      iput-object v0, p0, Landroid/mtp/MtpDatabase;->mSubDirectories:[Ljava/lang/String;
  
+     .line 135
      if-eqz p4, :cond_2
  
+     .line 137
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 138
      .local v1, builder:Ljava/lang/StringBuilder;
      const-string v10, "("
  
      invoke-virtual {v1, v10}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 139
      move-object/from16 v0, p4
  
      array-length v2, v0
  
+     .line 140
      .local v2, count:I
      const/4 v4, 0x0
  
***************
*** 436,475 ****
--- 468,515 ----
      :goto_0
      if-ge v4, v2, :cond_1
  
+     .line 141
      const-string v10, "_data=? OR _data LIKE ?"
  
      invoke-virtual {v1, v10}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 143
      add-int/lit8 v10, v2, -0x1
  
      if-eq v4, v10, :cond_0
  
+     .line 144
      const-string v10, " OR "
  
      invoke-virtual {v1, v10}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 140
      :cond_0
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
  
+     .line 147
      :cond_1
      const-string v10, ")"
  
      invoke-virtual {v1, v10}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 148
      invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v10
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mSubDirectoriesWhere:Ljava/lang/String;
  
+     .line 151
      mul-int/lit8 v10, v2, 0x2
  
      new-array v10, v10, [Ljava/lang/String;
  
      iput-object v10, p0, Landroid/mtp/MtpDatabase;->mSubDirectoriesWhereArgs:[Ljava/lang/String;
  
+     .line 152
      const/4 v4, 0x0
  
      const/4 v5, 0x0
***************
*** 482,489 ****
--- 522,531 ----
      :goto_1
      if-ge v4, v2, :cond_2
  
+     .line 153
      aget-object v9, p4, v4
  
+     .line 154
      .local v9, path:Ljava/lang/String;
      iget-object v10, p0, Landroid/mtp/MtpDatabase;->mSubDirectoriesWhereArgs:[Ljava/lang/String;
  
***************
*** 493,498 ****
--- 535,541 ----
      .restart local v5       #j:I
      aput-object v9, v10, v6
  
+     .line 155
      iget-object v10, p0, Landroid/mtp/MtpDatabase;->mSubDirectoriesWhereArgs:[Ljava/lang/String;
  
      add-int/lit8 v6, v5, 0x1
***************
*** 519,528 ****
--- 562,573 ----
  
      aput-object v11, v10, v5
  
+     .line 152
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_1
  
+     .line 160
      .end local v1           #builder:Ljava/lang/StringBuilder;
      .end local v2           #count:I
      .end local v4           #i:I
***************
*** 539,561 ****
--- 584,612 ----
  
      iget-object v8, v10, Landroid/content/res/Configuration;->locale:Ljava/util/Locale;
  
+     .line 161
      .local v8, locale:Ljava/util/Locale;
      if-eqz v8, :cond_3
  
+     .line 162
      invoke-virtual {v8}, Ljava/util/Locale;->getLanguage()Ljava/lang/String;
  
      move-result-object v7
  
+     .line 163
      .local v7, language:Ljava/lang/String;
      invoke-virtual {v8}, Ljava/util/Locale;->getCountry()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 164
      .local v3, country:Ljava/lang/String;
      if-eqz v7, :cond_3
  
+     .line 165
      if-eqz v3, :cond_4
  
+     .line 166
      iget-object v10, p0, Landroid/mtp/MtpDatabase;->mMediaScanner:Landroid/media/MediaScanner;
  
      new-instance v11, Ljava/lang/StringBuilder;
***************
*** 582,595 ****
--- 633,649 ----
  
      invoke-virtual {v10, v11}, Landroid/media/MediaScanner;->setLocale(Ljava/lang/String;)V
  
+     .line 172
      .end local v3           #country:Ljava/lang/String;
      .end local v7           #language:Ljava/lang/String;
      :cond_3
      :goto_2
      invoke-direct {p0, p1}, Landroid/mtp/MtpDatabase;->initDeviceProperties(Landroid/content/Context;)V
  
+     .line 173
      return-void
  
+     .line 168
      .restart local v3       #country:Ljava/lang/String;
      .restart local v7       #language:Ljava/lang/String;
      :cond_4
***************
*** 610,615 ****
--- 664,670 ----
      .parameter "modified"
  
      .prologue
+     .line 262
      invoke-direct {p0, p1}, Landroid/mtp/MtpDatabase;->inStorageSubDirectory(Ljava/lang/String;)Z
  
      move-result v0
***************
*** 618,632 ****
--- 673,691 ----
  
      const/4 v0, -0x1
  
+     .line 301
      :cond_0
      :goto_0
      return v0
  
+     .line 265
      :cond_1
      if-eqz p1, :cond_3
  
+     .line 266
      const/4 v6, 0x0
  
+     .line 268
      .local v6, c:Landroid/database/Cursor;
      :try_start_0
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
***************
*** 651,656 ****
--- 710,716 ----
  
      move-result-object v6
  
+     .line 270
      if-eqz v6, :cond_2
  
      invoke-interface {v6}, Landroid/database/Cursor;->getCount()I
***************
*** 659,664 ****
--- 719,725 ----
  
      if-lez v0, :cond_2
  
+     .line 271
      const-string v0, "MtpDatabase"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 684,702 ****
--- 745,769 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 272
      const/4 v0, -0x1
  
+     .line 277
      if-eqz v6, :cond_0
  
+     .line 278
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 277
      :cond_2
      if-eqz v6, :cond_3
  
+     .line 278
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
+     .line 283
      .end local v6           #c:Landroid/database/Cursor;
      :cond_3
      :goto_1
***************
*** 704,718 ****
--- 771,788 ----
  
      iput-boolean v0, p0, Landroid/mtp/MtpDatabase;->mDatabaseModified:Z
  
+     .line 284
      new-instance v9, Landroid/content/ContentValues;
  
      invoke-direct {v9}, Landroid/content/ContentValues;-><init>()V
  
+     .line 285
      .local v9, values:Landroid/content/ContentValues;
      const-string v0, "_data"
  
      invoke-virtual {v9, v0, p1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 286
      const-string v0, "format"
  
      invoke-static {p2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 721,727 ****
  
      invoke-virtual {v9, v0, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     const-string v0, "parent"
  
      invoke-static {p3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 791,798 ----
  
      invoke-virtual {v9, v0, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     .line 287
!     const-string/jumbo v0, "parent"
  
      invoke-static {p3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 729,735 ****
  
      invoke-virtual {v9, v0, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     const-string v0, "storage_id"
  
      invoke-static {p4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 800,807 ----
  
      invoke-virtual {v9, v0, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     .line 288
!     const-string/jumbo v0, "storage_id"
  
      invoke-static {p4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 737,742 ****
--- 809,815 ----
  
      invoke-virtual {v9, v0, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 289
      const-string v0, "_size"
  
      invoke-static/range {p5 .. p6}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
***************
*** 745,750 ****
--- 818,824 ----
  
      invoke-virtual {v9, v0, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
+     .line 290
      const-string v0, "date_modified"
  
      invoke-static/range {p7 .. p8}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
***************
*** 753,758 ****
--- 827,833 ----
  
      invoke-virtual {v9, v0, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
+     .line 293
      :try_start_1
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
  
***************
*** 762,770 ****
--- 837,847 ----
  
      move-result-object v8
  
+     .line 294
      .local v8, uri:Landroid/net/Uri;
      if-eqz v8, :cond_5
  
+     .line 295
      invoke-virtual {v8}, Landroid/net/Uri;->getPathSegments()Ljava/util/List;
  
      move-result-object v0
***************
*** 785,796 ****
--- 862,875 ----
  
      goto/16 :goto_0
  
+     .line 274
      .end local v8           #uri:Landroid/net/Uri;
      .end local v9           #values:Landroid/content/ContentValues;
      .restart local v6       #c:Landroid/database/Cursor;
      :catch_0
      move-exception v7
  
+     .line 275
      .local v7, e:Landroid/os/RemoteException;
      :try_start_2
      const-string v0, "MtpDatabase"
***************
*** 801,823 ****
--- 880,907 ----
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
  
+     .line 277
      if-eqz v6, :cond_3
  
+     .line 278
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_1
  
+     .line 277
      .end local v7           #e:Landroid/os/RemoteException;
      :catchall_0
      move-exception v0
  
      if-eqz v6, :cond_4
  
+     .line 278
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_4
      throw v0
  
+     .line 297
      .end local v6           #c:Landroid/database/Cursor;
      .restart local v8       #uri:Landroid/net/Uri;
      .restart local v9       #values:Landroid/content/ContentValues;
***************
*** 826,835 ****
--- 910,921 ----
  
      goto/16 :goto_0
  
+     .line 299
      .end local v8           #uri:Landroid/net/Uri;
      :catch_1
      move-exception v7
  
+     .line 300
      .restart local v7       #e:Landroid/os/RemoteException;
      const-string v0, "MtpDatabase"
  
***************
*** 837,842 ****
--- 923,929 ----
  
      invoke-static {v0, v1, v7}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 301
      const/4 v0, -0x1
  
      goto/16 :goto_0
***************
*** 862,890 ****
--- 949,987 ----
  
      const/4 v1, 0x0
  
+     .line 344
      if-ne p1, v0, :cond_6
  
+     .line 346
      if-nez p2, :cond_3
  
+     .line 348
      if-nez p3, :cond_1
  
+     .line 350
      const/4 v3, 0x0
  
+     .line 351
      .local v3, where:Ljava/lang/String;
      const/4 v4, 0x0
  
+     .line 415
      .local v4, whereArgs:[Ljava/lang/String;
      :goto_0
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mSubDirectoriesWhere:Ljava/lang/String;
  
      if-eqz v0, :cond_0
  
+     .line 416
      if-nez v3, :cond_c
  
+     .line 417
      iget-object v3, p0, Landroid/mtp/MtpDatabase;->mSubDirectoriesWhere:Ljava/lang/String;
  
+     .line 418
      iget-object v4, p0, Landroid/mtp/MtpDatabase;->mSubDirectoriesWhereArgs:[Ljava/lang/String;
  
+     .line 436
      :cond_0
      :goto_1
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
***************
*** 901,916 ****
--- 998,1017 ----
  
      return-object v0
  
+     .line 353
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_1
      if-ne p3, v0, :cond_2
  
+     .line 355
      const/4 p3, 0x0
  
+     .line 357
      :cond_2
      const-string v3, "format=?"
  
+     .line 358
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v2, [Ljava/lang/String;
  
***************
*** 923,935 ****
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_3
      if-nez p3, :cond_4
  
!     const-string v3, "parent=?"
  
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v2, [Ljava/lang/String;
  
--- 1024,1039 ----
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
+     .line 362
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_3
      if-nez p3, :cond_4
  
!     .line 364
!     const-string/jumbo v3, "parent=?"
  
+     .line 365
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v2, [Ljava/lang/String;
  
***************
*** 942,957 ****
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_4
      if-ne p3, v0, :cond_5
  
      const/4 p3, 0x0
  
      :cond_5
!     const-string v3, "parent=? AND parent=?"
  
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v5, [Ljava/lang/String;
  
--- 1046,1065 ----
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
+     .line 367
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_4
      if-ne p3, v0, :cond_5
  
+     .line 369
      const/4 p3, 0x0
  
+     .line 371
      :cond_5
!     const-string/jumbo v3, "parent=? AND parent=?"
  
+     .line 372
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v5, [Ljava/lang/String;
  
***************
*** 970,984 ****
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_6
      if-nez p2, :cond_9
  
      if-nez p3, :cond_7
  
!     const-string v3, "storage_id=?"
  
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v2, [Ljava/lang/String;
  
--- 1078,1096 ----
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
+     .line 378
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_6
      if-nez p2, :cond_9
  
+     .line 380
      if-nez p3, :cond_7
  
!     .line 382
!     const-string/jumbo v3, "storage_id=?"
  
+     .line 383
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v2, [Ljava/lang/String;
  
***************
*** 991,1006 ****
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_7
      if-ne p3, v0, :cond_8
  
      const/4 p3, 0x0
  
      :cond_8
!     const-string v3, "storage_id=? AND parent=?"
  
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v5, [Ljava/lang/String;
  
--- 1103,1122 ----
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
+     .line 385
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_7
      if-ne p3, v0, :cond_8
  
+     .line 387
      const/4 p3, 0x0
  
+     .line 389
      :cond_8
!     const-string/jumbo v3, "storage_id=? AND parent=?"
  
+     .line 390
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v5, [Ljava/lang/String;
  
***************
*** 1019,1031 ****
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_9
      if-nez p3, :cond_a
  
!     const-string v3, "storage_id=? AND format=?"
  
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v5, [Ljava/lang/String;
  
--- 1135,1150 ----
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
+     .line 395
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_9
      if-nez p3, :cond_a
  
!     .line 397
!     const-string/jumbo v3, "storage_id=? AND format=?"
  
+     .line 398
      .restart local v3       #where:Ljava/lang/String;
      new-array v4, v5, [Ljava/lang/String;
  
***************
*** 1044,1059 ****
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_a
      if-ne p3, v0, :cond_b
  
      const/4 p3, 0x0
  
      :cond_b
!     const-string v3, "storage_id=? AND format=? AND parent=?"
  
      .restart local v3       #where:Ljava/lang/String;
      const/4 v0, 0x3
  
--- 1163,1182 ----
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto :goto_0
  
+     .line 401
      .end local v3           #where:Ljava/lang/String;
      .end local v4           #whereArgs:[Ljava/lang/String;
      :cond_a
      if-ne p3, v0, :cond_b
  
+     .line 403
      const/4 p3, 0x0
  
+     .line 405
      :cond_b
!     const-string/jumbo v3, "storage_id=? AND format=? AND parent=?"
  
+     .line 406
      .restart local v3       #where:Ljava/lang/String;
      const/4 v0, 0x3
  
***************
*** 1080,1085 ****
--- 1203,1209 ----
      .restart local v4       #whereArgs:[Ljava/lang/String;
      goto/16 :goto_0
  
+     .line 420
      :cond_c
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 1105,1110 ****
--- 1229,1235 ----
  
      move-result-object v3
  
+     .line 423
      array-length v0, v4
  
      iget-object v1, p0, Landroid/mtp/MtpDatabase;->mSubDirectoriesWhereArgs:[Ljava/lang/String;
***************
*** 1115,1120 ****
--- 1240,1246 ----
  
      new-array v8, v0, [Ljava/lang/String;
  
+     .line 426
      .local v8, newWhereArgs:[Ljava/lang/String;
      const/4 v6, 0x0
  
***************
*** 1124,1137 ****
--- 1250,1266 ----
  
      if-ge v6, v0, :cond_d
  
+     .line 427
      aget-object v0, v4, v6
  
      aput-object v0, v8, v6
  
+     .line 426
      add-int/lit8 v6, v6, 0x1
  
      goto :goto_2
  
+     .line 429
      :cond_d
      const/4 v7, 0x0
  
***************
*** 1143,1160 ****
--- 1272,1292 ----
  
      if-ge v7, v0, :cond_e
  
+     .line 430
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mSubDirectoriesWhereArgs:[Ljava/lang/String;
  
      aget-object v0, v0, v7
  
      aput-object v0, v8, v6
  
+     .line 429
      add-int/lit8 v6, v6, 0x1
  
      add-int/lit8 v7, v7, 0x1
  
      goto :goto_3
  
+     .line 432
      :cond_e
      move-object v4, v8
  
***************
*** 1172,1187 ****
--- 1304,1324 ----
  
      const/4 v0, 0x1
  
+     .line 883
      iput-boolean v0, p0, Landroid/mtp/MtpDatabase;->mDatabaseModified:Z
  
+     .line 884
      const/4 v9, 0x0
  
+     .line 885
      .local v9, path:Ljava/lang/String;
      const/4 v8, 0x0
  
+     .line 887
      .local v8, format:I
      const/4 v6, 0x0
  
+     .line 889
      .local v6, c:Landroid/database/Cursor;
      :try_start_0
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
***************
*** 1210,1215 ****
--- 1347,1353 ----
  
      move-result-object v6
  
+     .line 891
      if-eqz v6, :cond_3
  
      invoke-interface {v6}, Landroid/database/Cursor;->moveToNext()Z
***************
*** 1218,1229 ****
--- 1356,1369 ----
  
      if-eqz v0, :cond_3
  
+     .line 894
      const/4 v0, 0x1
  
      invoke-interface {v6, v0}, Landroid/database/Cursor;->getString(I)Ljava/lang/String;
  
      move-result-object v9
  
+     .line 895
      const/4 v0, 0x3
  
      invoke-interface {v6, v0}, Landroid/database/Cursor;->getInt(I)I
***************
*** 1233,1245 ****
--- 1373,1388 ----
  
      move-result v8
  
+     .line 900
      if-eqz v9, :cond_0
  
      if-nez v8, :cond_5
  
+     .line 926
      :cond_0
      if-eqz v6, :cond_1
  
+     .line 927
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_1
***************
*** 1249,1257 ****
--- 1392,1402 ----
      :goto_0
      return v0
  
+     .line 926
      :cond_3
      if-eqz v6, :cond_4
  
+     .line 927
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_4
***************
*** 1259,1264 ****
--- 1404,1410 ----
  
      goto :goto_0
  
+     .line 905
      :cond_5
      :try_start_1
      invoke-direct {p0, v9}, Landroid/mtp/MtpDatabase;->isStorageSubDirectory(Ljava/lang/String;)Z
***************
*** 1270,1288 ****
--- 1416,1439 ----
  
      if-eqz v0, :cond_6
  
+     .line 906
      const/16 v0, 0x200d
  
+     .line 926
      if-eqz v6, :cond_2
  
+     .line 927
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 909
      :cond_6
      const/16 v0, 0x3001
  
      if-ne v8, v0, :cond_7
  
+     .line 911
      :try_start_2
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mVolumeName:Ljava/lang/String;
  
***************
*** 1290,1295 ****
--- 1441,1447 ----
  
      move-result-object v10
  
+     .line 912
      .local v10, uri:Landroid/net/Uri;
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
  
***************
*** 1323,1328 ****
--- 1475,1481 ----
  
      invoke-interface {v0, v10, v1, v2}, Landroid/content/IContentProvider;->delete(Landroid/net/Uri;Ljava/lang/String;[Ljava/lang/String;)I
  
+     .line 916
      .end local v10           #uri:Landroid/net/Uri;
      :cond_7
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mVolumeName:Ljava/lang/String;
***************
*** 1333,1338 ****
--- 1486,1492 ----
  
      move-result-object v10
  
+     .line 917
      .restart local v10       #uri:Landroid/net/Uri;
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
  
***************
*** 1349,1365 ****
--- 1503,1524 ----
  
      if-lez v0, :cond_8
  
+     .line 918
      const/16 v0, 0x2001
  
+     .line 926
      if-eqz v6, :cond_2
  
+     .line 927
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 926
      :cond_8
      if-eqz v6, :cond_9
  
+     .line 927
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_9
***************
*** 1367,1376 ****
--- 1526,1537 ----
  
      goto :goto_0
  
+     .line 922
      .end local v10           #uri:Landroid/net/Uri;
      :catch_0
      move-exception v7
  
+     .line 923
      .local v7, e:Landroid/os/RemoteException;
      :try_start_3
      const-string v0, "MtpDatabase"
***************
*** 1381,1388 ****
--- 1542,1551 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
+     .line 926
      if-eqz v6, :cond_a
  
+     .line 927
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_a
***************
*** 1390,1401 ****
--- 1553,1566 ----
  
      goto :goto_0
  
+     .line 926
      .end local v7           #e:Landroid/os/RemoteException;
      :catchall_0
      move-exception v0
  
      if-eqz v6, :cond_b
  
+     .line 927
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_b
***************
*** 1410,1423 ****
--- 1575,1592 ----
      .parameter "succeeded"
  
      .prologue
+     .line 306
      if-eqz p4, :cond_3
  
+     .line 309
      const v4, 0xba05
  
      if-ne p3, v4, :cond_2
  
+     .line 311
      move-object v2, p1
  
+     .line 312
      .local v2, name:Ljava/lang/String;
      const/16 v4, 0x2f
  
***************
*** 1425,1439 ****
--- 1594,1611 ----
  
      move-result v1
  
+     .line 313
      .local v1, lastSlash:I
      if-ltz v1, :cond_0
  
+     .line 314
      add-int/lit8 v4, v1, 0x1
  
      invoke-virtual {v2, v4}, Ljava/lang/String;->substring(I)Ljava/lang/String;
  
      move-result-object v2
  
+     .line 317
      :cond_0
      const-string v4, ".pla"
  
***************
*** 1443,1448 ****
--- 1615,1621 ----
  
      if-eqz v4, :cond_1
  
+     .line 318
      const/4 v4, 0x0
  
      invoke-virtual {v2}, Ljava/lang/String;->length()I
***************
*** 1455,1460 ****
--- 1628,1634 ----
  
      move-result-object v2
  
+     .line 321
      :cond_1
      new-instance v3, Landroid/content/ContentValues;
  
***************
*** 1462,1476 ****
  
      invoke-direct {v3, v4}, Landroid/content/ContentValues;-><init>(I)V
  
      .local v3, values:Landroid/content/ContentValues;
      const-string v4, "_data"
  
      invoke-virtual {v3, v4, p1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v4, "name"
  
      invoke-virtual {v3, v4, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
      const-string v4, "format"
  
      invoke-static {p3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
--- 1636,1653 ----
  
      invoke-direct {v3, v4}, Landroid/content/ContentValues;-><init>(I)V
  
+     .line 322
      .local v3, values:Landroid/content/ContentValues;
      const-string v4, "_data"
  
      invoke-virtual {v3, v4, p1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 323
!     const-string/jumbo v4, "name"
  
      invoke-virtual {v3, v4, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 324
      const-string v4, "format"
  
      invoke-static {p3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 1479,1484 ****
--- 1656,1662 ----
  
      invoke-virtual {v3, v4, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 325
      const-string v4, "date_modified"
  
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
***************
*** 1495,1501 ****
  
      invoke-virtual {v3, v4, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
!     const-string v4, "media_scanner_new_object_id"
  
      invoke-static {p2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 1673,1680 ----
  
      invoke-virtual {v3, v4, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
!     .line 326
!     const-string/jumbo v4, "media_scanner_new_object_id"
  
      invoke-static {p2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 1503,1508 ****
--- 1682,1688 ----
  
      invoke-virtual {v3, v4, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 328
      :try_start_0
      iget-object v4, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
  
***************
*** 1512,1529 ****
--- 1692,1712 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 338
      .end local v1           #lastSlash:I
      .end local v2           #name:Ljava/lang/String;
      .end local v3           #values:Landroid/content/ContentValues;
      :goto_0
      return-void
  
+     .line 329
      .restart local v1       #lastSlash:I
      .restart local v2       #name:Ljava/lang/String;
      .restart local v3       #values:Landroid/content/ContentValues;
      :catch_0
      move-exception v0
  
+     .line 330
      .local v0, e:Landroid/os/RemoteException;
      const-string v4, "MtpDatabase"
  
***************
*** 1533,1538 ****
--- 1716,1722 ----
  
      goto :goto_0
  
+     .line 333
      .end local v0           #e:Landroid/os/RemoteException;
      .end local v1           #lastSlash:I
      .end local v2           #name:Ljava/lang/String;
***************
*** 1546,1551 ****
--- 1730,1736 ----
  
      goto :goto_0
  
+     .line 336
      :cond_3
      invoke-direct {p0, p2}, Landroid/mtp/MtpDatabase;->deleteFile(I)I
  
***************
*** 1563,1575 ****
--- 1748,1763 ----
  
      const/4 v10, 0x0
  
+     .line 770
      sparse-switch p1, :sswitch_data_0
  
+     .line 795
      const/16 v6, 0x200a
  
      :goto_0
      return v6
  
+     .line 774
      :sswitch_0
      iget-object v6, p0, Landroid/mtp/MtpDatabase;->mDeviceProperties:Landroid/content/SharedPreferences;
  
***************
*** 1583,1615 ****
  
      move-result-object v4
  
      .local v4, value:Ljava/lang/String;
      invoke-virtual {v4}, Ljava/lang/String;->length()I
  
      move-result v3
  
      .local v3, length:I
      const/16 v6, 0xff
  
      if-le v3, v6, :cond_0
  
      const/16 v3, 0xff
  
      :cond_0
      invoke-virtual {v4, v10, v3, p3, v10}, Ljava/lang/String;->getChars(II[CI)V
  
      aput-char v10, p3, v3
  
      move v6, v7
  
      goto :goto_0
  
      .end local v3           #length:I
      .end local v4           #value:Ljava/lang/String;
      :sswitch_1
      iget-object v6, p0, Landroid/mtp/MtpDatabase;->mContext:Landroid/content/Context;
  
!     const-string v8, "window"
  
      invoke-virtual {v6, v8}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
--- 1771,1810 ----
  
      move-result-object v4
  
+     .line 775
      .local v4, value:Ljava/lang/String;
      invoke-virtual {v4}, Ljava/lang/String;->length()I
  
      move-result v3
  
+     .line 776
      .local v3, length:I
      const/16 v6, 0xff
  
      if-le v3, v6, :cond_0
  
+     .line 777
      const/16 v3, 0xff
  
+     .line 779
      :cond_0
      invoke-virtual {v4, v10, v3, p3, v10}, Ljava/lang/String;->getChars(II[CI)V
  
+     .line 780
      aput-char v10, p3, v3
  
      move v6, v7
  
+     .line 781
      goto :goto_0
  
+     .line 785
      .end local v3           #length:I
      .end local v4           #value:Ljava/lang/String;
      :sswitch_1
      iget-object v6, p0, Landroid/mtp/MtpDatabase;->mContext:Landroid/content/Context;
  
!     const-string/jumbo v8, "window"
  
      invoke-virtual {v6, v8}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
***************
*** 1621,1636 ****
--- 1816,1834 ----
  
      move-result-object v0
  
+     .line 787
      .local v0, display:Landroid/view/Display;
      invoke-virtual {v0}, Landroid/view/Display;->getMaximumSizeDimension()I
  
      move-result v5
  
+     .line 788
      .local v5, width:I
      invoke-virtual {v0}, Landroid/view/Display;->getMaximumSizeDimension()I
  
      move-result v1
  
+     .line 789
      .local v1, height:I
      new-instance v6, Ljava/lang/StringBuilder;
  
***************
*** 1644,1650 ****
  
      move-result-object v6
  
!     const-string v8, "x"
  
      invoke-virtual {v6, v8}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1842,1848 ----
  
      move-result-object v6
  
!     const-string/jumbo v8, "x"
  
      invoke-virtual {v6, v8}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1662,1667 ****
--- 1860,1866 ----
  
      move-result-object v2
  
+     .line 790
      .local v2, imageSize:Ljava/lang/String;
      invoke-virtual {v2}, Ljava/lang/String;->length()I
  
***************
*** 1669,1674 ****
--- 1868,1874 ----
  
      invoke-virtual {v2, v10, v6, p3, v10}, Ljava/lang/String;->getChars(II[CI)V
  
+     .line 791
      invoke-virtual {v2}, Ljava/lang/String;->length()I
  
      move-result v6
***************
*** 1677,1684 ****
--- 1877,1886 ----
  
      move v6, v7
  
+     .line 792
      goto :goto_0
  
+     .line 770
      :sswitch_data_0
      .sparse-switch
          0x5003 -> :sswitch_1
***************
*** 1694,1709 ****
--- 1896,1915 ----
      .parameter "parent"
  
      .prologue
+     .line 466
      const/4 v0, 0x0
  
+     .line 468
      .local v0, c:Landroid/database/Cursor;
      :try_start_0
      invoke-direct {p0, p1, p2, p3}, Landroid/mtp/MtpDatabase;->createObjectQuery(III)Landroid/database/Cursor;
  
      move-result-object v0
  
+     .line 469
      if-eqz v0, :cond_1
  
+     .line 470
      invoke-interface {v0}, Landroid/database/Cursor;->getCount()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1711,1738 ****
--- 1917,1952 ----
  
      move-result v2
  
+     .line 475
      if-eqz v0, :cond_0
  
+     .line 476
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
+     .line 479
      :cond_0
      :goto_0
      return v2
  
+     .line 475
      :cond_1
      if-eqz v0, :cond_2
  
+     .line 476
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
+     .line 479
      :cond_2
      :goto_1
      const/4 v2, -0x1
  
      goto :goto_0
  
+     .line 472
      :catch_0
      move-exception v1
  
+     .line 473
      .local v1, e:Landroid/os/RemoteException;
      :try_start_1
      const-string v2, "MtpDatabase"
***************
*** 1743,1760 ****
--- 1957,1978 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 475
      if-eqz v0, :cond_2
  
+     .line 476
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      goto :goto_1
  
+     .line 475
      .end local v1           #e:Landroid/os/RemoteException;
      :catchall_0
      move-exception v2
  
      if-eqz v0, :cond_3
  
+     .line 476
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      :cond_3
***************
*** 1774,1781 ****
--- 1992,2001 ----
  
      const/4 v2, 0x0
  
+     .line 850
      if-nez p1, :cond_1
  
+     .line 852
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaStoragePath:Ljava/lang/String;
  
      iget-object v1, p0, Landroid/mtp/MtpDatabase;->mMediaStoragePath:Ljava/lang/String;
***************
*** 1786,1791 ****
--- 2006,2012 ----
  
      invoke-virtual {v0, v2, v1, p2, v2}, Ljava/lang/String;->getChars(II[CI)V
  
+     .line 853
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaStoragePath:Ljava/lang/String;
  
      invoke-virtual {v0}, Ljava/lang/String;->length()I
***************
*** 1794,1816 ****
--- 2015,2042 ----
  
      aput-char v2, p2, v0
  
+     .line 854
      const-wide/16 v0, 0x0
  
      aput-wide v0, p3, v2
  
+     .line 855
      const-wide/16 v0, 0x3001
  
      aput-wide v0, p3, v3
  
      move v0, v9
  
+     .line 877
      :cond_0
      :goto_0
      return v0
  
+     .line 858
      :cond_1
      const/4 v6, 0x0
  
+     .line 860
      .local v6, c:Landroid/database/Cursor;
      :try_start_0
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
***************
*** 1839,1844 ****
--- 2065,2071 ----
  
      move-result-object v6
  
+     .line 862
      if-eqz v6, :cond_3
  
      invoke-interface {v6}, Landroid/database/Cursor;->moveToNext()Z
***************
*** 1847,1858 ****
--- 2074,2087 ----
  
      if-eqz v0, :cond_3
  
+     .line 863
      const/4 v0, 0x1
  
      invoke-interface {v6, v0}, Landroid/database/Cursor;->getString(I)Ljava/lang/String;
  
      move-result-object v8
  
+     .line 864
      .local v8, path:Ljava/lang/String;
      const/4 v0, 0x0
  
***************
*** 1864,1869 ****
--- 2093,2099 ----
  
      invoke-virtual {v8, v0, v1, p2, v2}, Ljava/lang/String;->getChars(II[CI)V
  
+     .line 865
      invoke-virtual {v8}, Ljava/lang/String;->length()I
  
      move-result v0
***************
*** 1872,1877 ****
--- 2102,2108 ----
  
      aput-char v1, p2, v0
  
+     .line 866
      const/4 v0, 0x0
  
      const/4 v1, 0x2
***************
*** 1882,1887 ****
--- 2113,2119 ----
  
      aput-wide v1, p3, v0
  
+     .line 867
      const/4 v0, 0x1
  
      const/4 v1, 0x3
***************
*** 1895,1902 ****
--- 2127,2136 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 876
      if-eqz v6, :cond_2
  
+     .line 877
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_2
***************
*** 1904,1922 ****
--- 2138,2161 ----
  
      goto :goto_0
  
+     .line 870
      .end local v8           #path:Ljava/lang/String;
      :cond_3
      const/16 v0, 0x2009
  
+     .line 876
      if-eqz v6, :cond_0
  
+     .line 877
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 872
      :catch_0
      move-exception v7
  
+     .line 873
      .local v7, e:Landroid/os/RemoteException;
      :try_start_1
      const-string v0, "MtpDatabase"
***************
*** 1927,1946 ****
--- 2166,2190 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 874
      const/16 v0, 0x2002
  
+     .line 876
      if-eqz v6, :cond_0
  
+     .line 877
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 876
      .end local v7           #e:Landroid/os/RemoteException;
      :catchall_0
      move-exception v0
  
      if-eqz v6, :cond_4
  
+     .line 877
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_4
***************
*** 1955,1962 ****
--- 2199,2208 ----
      .parameter "outSizeModified"
  
      .prologue
+     .line 815
      const/4 v7, 0x0
  
+     .line 817
      .local v7, c:Landroid/database/Cursor;
      :try_start_0
      iget-object v1, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
***************
*** 1985,1990 ****
--- 2231,2237 ----
  
      move-result-object v7
  
+     .line 819
      if-eqz v7, :cond_3
  
      invoke-interface {v7}, Landroid/database/Cursor;->moveToNext()Z
***************
*** 1993,1998 ****
--- 2240,2246 ----
  
      if-eqz v1, :cond_3
  
+     .line 820
      const/4 v1, 0x0
  
      const/4 v2, 0x1
***************
*** 2003,2008 ****
--- 2251,2257 ----
  
      aput v2, p2, v1
  
+     .line 821
      const/4 v1, 0x1
  
      const/4 v2, 0x2
***************
*** 2013,2018 ****
--- 2262,2268 ----
  
      aput v2, p2, v1
  
+     .line 822
      const/4 v1, 0x2
  
      const/4 v2, 0x3
***************
*** 2023,2034 ****
--- 2273,2286 ----
  
      aput v2, p2, v1
  
+     .line 825
      const/4 v1, 0x4
  
      invoke-interface {v7, v1}, Landroid/database/Cursor;->getString(I)Ljava/lang/String;
  
      move-result-object v11
  
+     .line 826
      .local v11, path:Ljava/lang/String;
      const/16 v1, 0x2f
  
***************
*** 2036,2052 ****
--- 2288,2307 ----
  
      move-result v10
  
+     .line 827
      .local v10, lastSlash:I
      if-ltz v10, :cond_2
  
      add-int/lit8 v12, v10, 0x1
  
+     .line 828
      .local v12, start:I
      :goto_0
      invoke-virtual {v11}, Ljava/lang/String;->length()I
  
      move-result v9
  
+     .line 829
      .local v9, end:I
      sub-int v1, v9, v12
  
***************
*** 2054,2061 ****
--- 2309,2318 ----
  
      if-le v1, v2, :cond_0
  
+     .line 830
      add-int/lit16 v9, v12, 0xff
  
+     .line 832
      :cond_0
      const/4 v1, 0x0
  
***************
*** 2063,2074 ****
--- 2320,2333 ----
  
      invoke-virtual {v11, v12, v9, v0, v1}, Ljava/lang/String;->getChars(II[CI)V
  
+     .line 833
      sub-int v1, v9, v12
  
      const/4 v2, 0x0
  
      aput-char v2, p3, v1
  
+     .line 835
      const/4 v1, 0x0
  
      const/4 v2, 0x5
***************
*** 2079,2084 ****
--- 2338,2344 ----
  
      aput-wide v2, p4, v1
  
+     .line 836
      const/4 v1, 0x1
  
      const/4 v2, 0x6
***************
*** 2092,2103 ****
--- 2352,2367 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 837
      const/4 v1, 0x1
  
+     .line 842
      if-eqz v7, :cond_1
  
+     .line 843
      invoke-interface {v7}, Landroid/database/Cursor;->close()V
  
+     .line 846
      .end local v9           #end:I
      .end local v10           #lastSlash:I
      .end local v11           #path:Ljava/lang/String;
***************
*** 2106,2111 ****
--- 2370,2376 ----
      :goto_1
      return v1
  
+     .line 827
      .restart local v10       #lastSlash:I
      .restart local v11       #path:Ljava/lang/String;
      :cond_2
***************
*** 2113,2134 ****
--- 2378,2404 ----
  
      goto :goto_0
  
+     .line 842
      .end local v10           #lastSlash:I
      .end local v11           #path:Ljava/lang/String;
      :cond_3
      if-eqz v7, :cond_4
  
+     .line 843
      invoke-interface {v7}, Landroid/database/Cursor;->close()V
  
+     .line 846
      :cond_4
      :goto_2
      const/4 v1, 0x0
  
      goto :goto_1
  
+     .line 839
      :catch_0
      move-exception v8
  
+     .line 840
      .local v8, e:Landroid/os/RemoteException;
      :try_start_1
      const-string v1, "MtpDatabase"
***************
*** 2139,2156 ****
--- 2409,2430 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 842
      if-eqz v7, :cond_4
  
+     .line 843
      invoke-interface {v7}, Landroid/database/Cursor;->close()V
  
      goto :goto_2
  
+     .line 842
      .end local v8           #e:Landroid/os/RemoteException;
      :catchall_0
      move-exception v1
  
      if-eqz v7, :cond_5
  
+     .line 843
      invoke-interface {v7}, Landroid/database/Cursor;->close()V
  
      :cond_5
***************
*** 2166,2173 ****
--- 2440,2449 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 440
      const/4 v0, 0x0
  
+     .line 442
      .local v0, c:Landroid/database/Cursor;
      :try_start_0
      invoke-direct {p0, p1, p2, p3}, Landroid/mtp/MtpDatabase;->createObjectQuery(III)Landroid/database/Cursor;
***************
*** 2177,2206 ****
--- 2453,2490 ----
  
      move-result-object v0
  
+     .line 443
      if-nez v0, :cond_2
  
+     .line 458
      if-eqz v0, :cond_0
  
+     .line 459
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      :cond_0
      move-object v4, v5
  
+     .line 462
      :cond_1
      :goto_0
      return-object v4
  
+     .line 446
      :cond_2
      :try_start_1
      invoke-interface {v0}, Landroid/database/Cursor;->getCount()I
  
      move-result v1
  
+     .line 447
      .local v1, count:I
      if-lez v1, :cond_4
  
+     .line 448
      new-array v4, v1, [I
  
+     .line 449
      .local v4, result:[I
      const/4 v3, 0x0
  
***************
*** 2208,2215 ****
--- 2492,2501 ----
      :goto_1
      if-ge v3, v1, :cond_3
  
+     .line 450
      invoke-interface {v0}, Landroid/database/Cursor;->moveToNext()Z
  
+     .line 451
      const/4 v6, 0x0
  
      invoke-interface {v0, v6}, Landroid/database/Cursor;->getInt(I)I
***************
*** 2221,2242 ****
--- 2507,2533 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 449
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_1
  
+     .line 458
      :cond_3
      if-eqz v0, :cond_1
  
+     .line 459
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 458
      .end local v3           #i:I
      .end local v4           #result:[I
      :cond_4
      if-eqz v0, :cond_5
  
+     .line 459
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      .end local v1           #count:I
***************
*** 2244,2254 ****
--- 2535,2548 ----
      :goto_2
      move-object v4, v5
  
+     .line 462
      goto :goto_0
  
+     .line 455
      :catch_0
      move-exception v2
  
+     .line 456
      .local v2, e:Landroid/os/RemoteException;
      :try_start_2
      const-string v6, "MtpDatabase"
***************
*** 2259,2276 ****
--- 2553,2574 ----
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
  
+     .line 458
      if-eqz v0, :cond_5
  
+     .line 459
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      goto :goto_2
  
+     .line 458
      .end local v2           #e:Landroid/os/RemoteException;
      :catchall_0
      move-exception v5
  
      if-eqz v0, :cond_6
  
+     .line 459
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      :cond_6
***************
*** 2288,2304 ****
--- 2586,2606 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 671
      if-eqz p6, :cond_0
  
+     .line 672
      new-instance v2, Landroid/mtp/MtpPropertyList;
  
      const v3, 0xa807
  
      invoke-direct {v2, v4, v3}, Landroid/mtp/MtpPropertyList;-><init>(II)V
  
+     .line 692
      :goto_0
      return-object v2
  
+     .line 676
      :cond_0
      const-wide v2, 0xffffffffL
  
***************
*** 2306,2311 ****
--- 2608,2614 ----
  
      if-nez v2, :cond_2
  
+     .line 677
      iget-object v2, p0, Landroid/mtp/MtpDatabase;->mPropertyGroupsByFormat:Ljava/util/HashMap;
  
      invoke-static {p3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 2318,2330 ****
--- 2621,2636 ----
  
      check-cast v0, Landroid/mtp/MtpPropertyGroup;
  
+     .line 678
      .local v0, propertyGroup:Landroid/mtp/MtpPropertyGroup;
      if-nez v0, :cond_1
  
+     .line 679
      invoke-direct {p0, p3}, Landroid/mtp/MtpDatabase;->getSupportedObjectProperties(I)[I
  
      move-result-object v1
  
+     .line 680
      .local v1, propertyList:[I
      new-instance v0, Landroid/mtp/MtpPropertyGroup;
  
***************
*** 2335,2340 ****
--- 2641,2647 ----
  
      invoke-direct {v0, p0, v2, v3, v1}, Landroid/mtp/MtpPropertyGroup;-><init>(Landroid/mtp/MtpDatabase;Landroid/content/IContentProvider;Ljava/lang/String;[I)V
  
+     .line 681
      .restart local v0       #propertyGroup:Landroid/mtp/MtpPropertyGroup;
      iget-object v2, p0, Landroid/mtp/MtpDatabase;->mPropertyGroupsByFormat:Ljava/util/HashMap;
  
***************
*** 2344,2349 ****
--- 2651,2657 ----
  
      invoke-virtual {v2, v3, v0}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 692
      .end local v1           #propertyList:[I
      :cond_1
      :goto_1
***************
*** 2355,2360 ****
--- 2663,2669 ----
  
      goto :goto_0
  
+     .line 684
      .end local v0           #propertyGroup:Landroid/mtp/MtpPropertyGroup;
      :cond_2
      iget-object v2, p0, Landroid/mtp/MtpDatabase;->mPropertyGroupsByProperty:Ljava/util/HashMap;
***************
*** 2369,2377 ****
--- 2678,2688 ----
  
      check-cast v0, Landroid/mtp/MtpPropertyGroup;
  
+     .line 685
      .restart local v0       #propertyGroup:Landroid/mtp/MtpPropertyGroup;
      if-nez v0, :cond_1
  
+     .line 686
      const/4 v2, 0x1
  
      new-array v1, v2, [I
***************
*** 2380,2385 ****
--- 2691,2697 ----
  
      aput v2, v1, v4
  
+     .line 687
      .restart local v1       #propertyList:[I
      new-instance v0, Landroid/mtp/MtpPropertyGroup;
  
***************
*** 2390,2395 ****
--- 2702,2708 ----
  
      invoke-direct {v0, p0, v2, v3, v1}, Landroid/mtp/MtpPropertyGroup;-><init>(Landroid/mtp/MtpDatabase;Landroid/content/IContentProvider;Ljava/lang/String;[I)V
  
+     .line 688
      .restart local v0       #propertyGroup:Landroid/mtp/MtpPropertyGroup;
      iget-object v2, p0, Landroid/mtp/MtpDatabase;->mPropertyGroupsByProperty:Ljava/util/HashMap;
  
***************
*** 2411,2416 ****
--- 2724,2730 ----
      .prologue
      const/4 v11, 0x0
  
+     .line 933
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mVolumeName:Ljava/lang/String;
  
      int-to-long v2, p1
***************
*** 2419,2427 ****
--- 2733,2743 ----
  
      move-result-object v1
  
+     .line 934
      .local v1, uri:Landroid/net/Uri;
      const/4 v6, 0x0
  
+     .line 936
      .local v6, c:Landroid/database/Cursor;
      :try_start_0
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mMediaProvider:Landroid/content/IContentProvider;
***************
*** 2441,2470 ****
--- 2757,2794 ----
  
      move-result-object v6
  
+     .line 937
      if-nez v6, :cond_2
  
+     .line 952
      if-eqz v6, :cond_0
  
+     .line 953
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_0
      move-object v10, v11
  
+     .line 956
      :cond_1
      :goto_0
      return-object v10
  
+     .line 940
      :cond_2
      :try_start_1
      invoke-interface {v6}, Landroid/database/Cursor;->getCount()I
  
      move-result v7
  
+     .line 941
      .local v7, count:I
      if-lez v7, :cond_4
  
+     .line 942
      new-array v10, v7, [I
  
+     .line 943
      .local v10, result:[I
      const/4 v9, 0x0
  
***************
*** 2472,2479 ****
--- 2796,2805 ----
      :goto_1
      if-ge v9, v7, :cond_3
  
+     .line 944
      invoke-interface {v6}, Landroid/database/Cursor;->moveToNext()Z
  
+     .line 945
      const/4 v0, 0x0
  
      invoke-interface {v6, v0}, Landroid/database/Cursor;->getInt(I)I
***************
*** 2485,2506 ****
--- 2811,2837 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 943
      add-int/lit8 v9, v9, 0x1
  
      goto :goto_1
  
+     .line 952
      :cond_3
      if-eqz v6, :cond_1
  
+     .line 953
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 952
      .end local v9           #i:I
      .end local v10           #result:[I
      :cond_4
      if-eqz v6, :cond_5
  
+     .line 953
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      .end local v7           #count:I
***************
*** 2508,2518 ****
--- 2839,2852 ----
      :goto_2
      move-object v10, v11
  
+     .line 956
      goto :goto_0
  
+     .line 949
      :catch_0
      move-exception v8
  
+     .line 950
      .local v8, e:Landroid/os/RemoteException;
      :try_start_2
      const-string v0, "MtpDatabase"
***************
*** 2523,2540 ****
--- 2857,2878 ----
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
  
+     .line 952
      if-eqz v6, :cond_5
  
+     .line 953
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_2
  
+     .line 952
      .end local v8           #e:Landroid/os/RemoteException;
      :catchall_0
      move-exception v0
  
      if-eqz v6, :cond_6
  
+     .line 953
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_6
***************
*** 2545,2550 ****
--- 2883,2889 ----
      .locals 1
  
      .prologue
+     .line 516
      const/4 v0, 0x0
  
      return-object v0
***************
*** 2554,2559 ****
--- 2893,2899 ----
      .locals 1
  
      .prologue
+     .line 660
      const/4 v0, 0x3
  
      new-array v0, v0, [I
***************
*** 2577,2609 ****
--- 2917,2956 ----
      .parameter "format"
  
      .prologue
+     .line 636
      sparse-switch p1, :sswitch_data_0
  
+     .line 655
      sget-object v0, Landroid/mtp/MtpDatabase;->FILE_PROPERTIES:[I
  
      :goto_0
      return-object v0
  
+     .line 642
      :sswitch_0
      sget-object v0, Landroid/mtp/MtpDatabase;->AUDIO_PROPERTIES:[I
  
      goto :goto_0
  
+     .line 646
      :sswitch_1
      sget-object v0, Landroid/mtp/MtpDatabase;->VIDEO_PROPERTIES:[I
  
      goto :goto_0
  
+     .line 651
      :sswitch_2
      sget-object v0, Landroid/mtp/MtpDatabase;->IMAGE_PROPERTIES:[I
  
      goto :goto_0
  
+     .line 653
      :sswitch_3
      sget-object v0, Landroid/mtp/MtpDatabase;->ALL_PROPERTIES:[I
  
      goto :goto_0
  
+     .line 636
      :sswitch_data_0
      .sparse-switch
          0x0 -> :sswitch_3
***************
*** 2626,2631 ****
--- 2973,2979 ----
      .locals 1
  
      .prologue
+     .line 483
      const/16 v0, 0x19
  
      new-array v0, v0, [I
***************
*** 2669,2684 ****
--- 3017,3035 ----
      .parameter "path"
  
      .prologue
+     .line 230
      iget-object v5, p0, Landroid/mtp/MtpDatabase;->mSubDirectories:[Ljava/lang/String;
  
      if-nez v5, :cond_1
  
      const/4 v0, 0x1
  
+     .line 244
      :cond_0
      :goto_0
      return v0
  
+     .line 231
      :cond_1
      if-nez p1, :cond_2
  
***************
*** 2686,2699 ****
--- 3037,3053 ----
  
      goto :goto_0
  
+     .line 233
      :cond_2
      const/4 v0, 0x0
  
+     .line 234
      .local v0, allowed:Z
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
      move-result v2
  
+     .line 235
      .local v2, pathLength:I
      const/4 v1, 0x0
  
***************
*** 2707,2721 ****
--- 3061,3078 ----
  
      if-nez v0, :cond_0
  
+     .line 236
      iget-object v5, p0, Landroid/mtp/MtpDatabase;->mSubDirectories:[Ljava/lang/String;
  
      aget-object v3, v5, v1
  
+     .line 237
      .local v3, subdir:Ljava/lang/String;
      invoke-virtual {v3}, Ljava/lang/String;->length()I
  
      move-result v4
  
+     .line 238
      .local v4, subdirLength:I
      if-ge v4, v2, :cond_3
  
***************
*** 2733,2740 ****
--- 3090,3099 ----
  
      if-eqz v5, :cond_3
  
+     .line 241
      const/4 v0, 0x1
  
+     .line 235
      :cond_3
      add-int/lit8 v1, v1, 0x1
  
***************
*** 2748,2755 ****
--- 3107,3116 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 193
      const-string v10, "device-properties"
  
+     .line 194
      .local v10, devicePropertiesName:Ljava/lang/String;
      const-string v1, "device-properties"
  
***************
*** 2759,2770 ****
--- 3120,3133 ----
  
      iput-object v1, p0, Landroid/mtp/MtpDatabase;->mDeviceProperties:Landroid/content/SharedPreferences;
  
+     .line 195
      const-string v1, "device-properties"
  
      invoke-virtual {p1, v1}, Landroid/content/Context;->getDatabasePath(Ljava/lang/String;)Ljava/io/File;
  
      move-result-object v9
  
+     .line 197
      .local v9, databaseFile:Ljava/io/File;
      invoke-virtual {v9}, Ljava/io/File;->exists()Z
  
***************
*** 2772,2782 ****
--- 3135,3148 ----
  
      if-eqz v1, :cond_2
  
+     .line 200
      const/4 v0, 0x0
  
+     .line 201
      .local v0, db:Landroid/database/sqlite/SQLiteDatabase;
      const/4 v8, 0x0
  
+     .line 203
      .local v8, c:Landroid/database/Cursor;
      :try_start_0
      const-string v1, "device-properties"
***************
*** 2789,2797 ****
  
      move-result-object v0
  
      if-eqz v0, :cond_4
  
!     const-string v1, "properties"
  
      const/4 v2, 0x3
  
--- 3155,3165 ----
  
      move-result-object v0
  
+     .line 204
      if-eqz v0, :cond_4
  
!     .line 205
!     const-string/jumbo v1, "properties"
  
      const/4 v2, 0x3
  
***************
*** 2811,2817 ****
  
      const/4 v3, 0x2
  
!     const-string v4, "value"
  
      aput-object v4, v2, v3
  
--- 3179,3185 ----
  
      const/4 v3, 0x2
  
!     const-string/jumbo v4, "value"
  
      aput-object v4, v2, v3
  
***************
*** 2829,2842 ****
--- 3197,3213 ----
  
      move-result-object v8
  
+     .line 207
      if-eqz v8, :cond_4
  
+     .line 208
      iget-object v1, p0, Landroid/mtp/MtpDatabase;->mDeviceProperties:Landroid/content/SharedPreferences;
  
      invoke-interface {v1}, Landroid/content/SharedPreferences;->edit()Landroid/content/SharedPreferences$Editor;
  
      move-result-object v11
  
+     .line 209
      .local v11, e:Landroid/content/SharedPreferences$Editor;
      :goto_0
      invoke-interface {v8}, Landroid/database/Cursor;->moveToNext()Z
***************
*** 2845,2856 ****
--- 3216,3229 ----
  
      if-eqz v1, :cond_3
  
+     .line 210
      const/4 v1, 0x1
  
      invoke-interface {v8, v1}, Landroid/database/Cursor;->getString(I)Ljava/lang/String;
  
      move-result-object v12
  
+     .line 211
      .local v12, name:Ljava/lang/String;
      const/4 v1, 0x2
  
***************
*** 2858,2863 ****
--- 3231,3237 ----
  
      move-result-object v13
  
+     .line 212
      .local v13, value:Ljava/lang/String;
      invoke-interface {v11, v12, v13}, Landroid/content/SharedPreferences$Editor;->putString(Ljava/lang/String;Ljava/lang/String;)Landroid/content/SharedPreferences$Editor;
      :try_end_0
***************
*** 2866,2877 ****
--- 3240,3253 ----
  
      goto :goto_0
  
+     .line 217
      .end local v11           #e:Landroid/content/SharedPreferences$Editor;
      .end local v12           #name:Ljava/lang/String;
      .end local v13           #value:Ljava/lang/String;
      :catch_0
      move-exception v11
  
+     .line 218
      .local v11, e:Ljava/lang/Exception;
      :try_start_1
      const-string v1, "MtpDatabase"
***************
*** 2882,2906 ****
--- 3258,3287 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 220
      if-eqz v8, :cond_0
  
      invoke-interface {v8}, Landroid/database/Cursor;->close()V
  
+     .line 221
      :cond_0
      if-eqz v0, :cond_1
  
      invoke-virtual {v0}, Landroid/database/sqlite/SQLiteDatabase;->close()V
  
+     .line 223
      .end local v11           #e:Ljava/lang/Exception;
      :cond_1
      :goto_1
      invoke-virtual {v9}, Ljava/io/File;->delete()Z
  
+     .line 225
      .end local v0           #db:Landroid/database/sqlite/SQLiteDatabase;
      .end local v8           #c:Landroid/database/Cursor;
      :cond_2
      return-void
  
+     .line 214
      .restart local v0       #db:Landroid/database/sqlite/SQLiteDatabase;
      .restart local v8       #c:Landroid/database/Cursor;
      .local v11, e:Landroid/content/SharedPreferences$Editor;
***************
*** 2911,2922 ****
--- 3292,3305 ----
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
      .catch Ljava/lang/Exception; {:try_start_2 .. :try_end_2} :catch_0
  
+     .line 220
      .end local v11           #e:Landroid/content/SharedPreferences$Editor;
      :cond_4
      if-eqz v8, :cond_5
  
      invoke-interface {v8}, Landroid/database/Cursor;->close()V
  
+     .line 221
      :cond_5
      if-eqz v0, :cond_1
  
***************
*** 2924,2929 ****
--- 3307,3313 ----
  
      goto :goto_1
  
+     .line 220
      :catchall_0
      move-exception v1
  
***************
*** 2931,2936 ****
--- 3315,3321 ----
  
      invoke-interface {v8}, Landroid/database/Cursor;->close()V
  
+     .line 221
      :cond_6
      if-eqz v0, :cond_7
  
***************
*** 2947,2960 ****
--- 3332,3348 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 250
      iget-object v2, p0, Landroid/mtp/MtpDatabase;->mSubDirectories:[Ljava/lang/String;
  
      if-nez v2, :cond_1
  
+     .line 256
      :cond_0
      :goto_0
      return v1
  
+     .line 251
      :cond_1
      const/4 v0, 0x0
  
***************
*** 2966,2971 ****
--- 3354,3360 ----
  
      if-ge v0, v2, :cond_0
  
+     .line 252
      iget-object v2, p0, Landroid/mtp/MtpDatabase;->mSubDirectories:[Ljava/lang/String;
  
      aget-object v2, v2, v0
***************
*** 2976,2985 ****
--- 3365,3376 ----
  
      if-eqz v2, :cond_2
  
+     .line 253
      const/4 v1, 0x1
  
      goto :goto_0
  
+     .line 251
      :cond_2
      add-int/lit8 v0, v0, 0x1
  
***************
*** 2998,3008 ****
--- 3389,3402 ----
      .parameter "newName"
  
      .prologue
+     .line 696
      const/4 v7, 0x0
  
+     .line 699
      .local v7, c:Landroid/database/Cursor;
      const/4 v13, 0x0
  
+     .line 700
      .local v13, path:Ljava/lang/String;
      const/4 v1, 0x1
  
***************
*** 3016,3021 ****
--- 3410,3416 ----
  
      aput-object v2, v5, v1
  
+     .line 702
      .local v5, whereArgs:[Ljava/lang/String;
      :try_start_0
      move-object/from16 v0, p0
***************
*** 3036,3041 ****
--- 3431,3437 ----
  
      move-result-object v7
  
+     .line 703
      if-eqz v7, :cond_0
  
      invoke-interface {v7}, Landroid/database/Cursor;->moveToNext()Z
***************
*** 3044,3049 ****
--- 3440,3446 ----
  
      if-eqz v1, :cond_0
  
+     .line 704
      const/4 v1, 0x1
  
      invoke-interface {v7, v1}, Landroid/database/Cursor;->getString(I)Ljava/lang/String;
***************
*** 3053,3075 ****
--- 3450,3479 ----
  
      move-result-object v13
  
+     .line 710
      :cond_0
      if-eqz v7, :cond_1
  
+     .line 711
      invoke-interface {v7}, Landroid/database/Cursor;->close()V
  
+     .line 714
      :cond_1
      if-nez v13, :cond_4
  
+     .line 715
      const/16 v1, 0x2009
  
+     .line 755
      :cond_2
      :goto_0
      return v1
  
+     .line 706
      :catch_0
      move-exception v8
  
+     .line 707
      .local v8, e:Landroid/os/RemoteException;
      :try_start_1
      const-string v1, "MtpDatabase"
***************
*** 3080,3104 ****
--- 3484,3514 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 708
      const/16 v1, 0x2002
  
+     .line 710
      if-eqz v7, :cond_2
  
+     .line 711
      invoke-interface {v7}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 710
      .end local v8           #e:Landroid/os/RemoteException;
      :catchall_0
      move-exception v1
  
      if-eqz v7, :cond_3
  
+     .line 711
      invoke-interface {v7}, Landroid/database/Cursor;->close()V
  
      :cond_3
      throw v1
  
+     .line 719
      :cond_4
      move-object/from16 v0, p0
  
***************
*** 3108,3122 ****
--- 3518,3535 ----
  
      if-eqz v1, :cond_5
  
+     .line 720
      const/16 v1, 0x200d
  
      goto :goto_0
  
+     .line 724
      :cond_5
      new-instance v12, Ljava/io/File;
  
      invoke-direct {v12, v13}, Ljava/io/File;-><init>(Ljava/lang/String;)V
  
+     .line 725
      .local v12, oldFile:Ljava/io/File;
      const/16 v1, 0x2f
  
***************
*** 3124,3138 ****
--- 3537,3554 ----
  
      move-result v9
  
+     .line 726
      .local v9, lastSlash:I
      const/4 v1, 0x1
  
      if-gt v9, v1, :cond_6
  
+     .line 727
      const/16 v1, 0x2002
  
      goto :goto_0
  
+     .line 729
      :cond_6
      new-instance v1, Ljava/lang/StringBuilder;
  
***************
*** 3160,3185 ****
  
      move-result-object v11
  
      .local v11, newPath:Ljava/lang/String;
      new-instance v10, Ljava/io/File;
  
      invoke-direct {v10, v11}, Ljava/io/File;-><init>(Ljava/lang/String;)V
  
      .local v10, newFile:Ljava/io/File;
      invoke-virtual {v12, v10}, Ljava/io/File;->renameTo(Ljava/io/File;)Z
  
      move-result v14
  
      .local v14, success:Z
      if-nez v14, :cond_7
  
      const-string v1, "MtpDatabase"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "renaming "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 3576,3605 ----
  
      move-result-object v11
  
+     .line 730
      .local v11, newPath:Ljava/lang/String;
      new-instance v10, Ljava/io/File;
  
      invoke-direct {v10, v11}, Ljava/io/File;-><init>(Ljava/lang/String;)V
  
+     .line 731
      .local v10, newFile:Ljava/io/File;
      invoke-virtual {v12, v10}, Ljava/io/File;->renameTo(Ljava/io/File;)Z
  
      move-result v14
  
+     .line 732
      .local v14, success:Z
      if-nez v14, :cond_7
  
+     .line 733
      const-string v1, "MtpDatabase"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "renaming "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 3211,3225 ****
--- 3631,3648 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 734
      const/16 v1, 0x2002
  
      goto/16 :goto_0
  
+     .line 738
      :cond_7
      new-instance v16, Landroid/content/ContentValues;
  
      invoke-direct/range {v16 .. v16}, Landroid/content/ContentValues;-><init>()V
  
+     .line 739
      .local v16, values:Landroid/content/ContentValues;
      const-string v1, "_data"
  
***************
*** 3227,3234 ****
--- 3650,3659 ----
  
      invoke-virtual {v0, v1, v11}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 740
      const/4 v15, 0x0
  
+     .line 744
      .local v15, updated:I
      :try_start_2
      move-object/from16 v0, p0
***************
*** 3249,3257 ****
--- 3674,3684 ----
  
      move-result v15
  
+     .line 748
      :goto_1
      if-nez v15, :cond_8
  
+     .line 749
      const-string v1, "MtpDatabase"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 3284,3298 ****
--- 3711,3729 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 751
      invoke-virtual {v10, v12}, Ljava/io/File;->renameTo(Ljava/io/File;)Z
  
+     .line 752
      const/16 v1, 0x2002
  
      goto/16 :goto_0
  
+     .line 745
      :catch_1
      move-exception v8
  
+     .line 746
      .restart local v8       #e:Landroid/os/RemoteException;
      const-string v1, "MtpDatabase"
  
***************
*** 3302,3307 ****
--- 3733,3739 ----
  
      goto :goto_1
  
+     .line 755
      .end local v8           #e:Landroid/os/RemoteException;
      :cond_8
      const/16 v1, 0x2001
***************
*** 3313,3322 ****
--- 3745,3756 ----
      .locals 3
  
      .prologue
+     .line 984
      iget-boolean v0, p0, Landroid/mtp/MtpDatabase;->mDatabaseModified:Z
  
      if-eqz v0, :cond_0
  
+     .line 985
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mContext:Landroid/content/Context;
  
      new-instance v1, Landroid/content/Intent;
***************
*** 3327,3336 ****
--- 3761,3772 ----
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 986
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/mtp/MtpDatabase;->mDatabaseModified:Z
  
+     .line 988
      :cond_0
      return-void
  .end method
***************
*** 3339,3348 ****
--- 3775,3786 ----
      .locals 1
  
      .prologue
+     .line 980
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/mtp/MtpDatabase;->mDatabaseModified:Z
  
+     .line 981
      return-void
  .end method
  
***************
*** 3353,3365 ****
--- 3791,3806 ----
      .parameter "stringValue"
  
      .prologue
+     .line 800
      packed-switch p1, :pswitch_data_0
  
+     .line 810
      const/16 v1, 0x200a
  
      :goto_0
      return v1
  
+     .line 804
      :pswitch_0
      iget-object v1, p0, Landroid/mtp/MtpDatabase;->mDeviceProperties:Landroid/content/SharedPreferences;
  
***************
*** 3367,3372 ****
--- 3808,3814 ----
  
      move-result-object v0
  
+     .line 805
      .local v0, e:Landroid/content/SharedPreferences$Editor;
      invoke-static {p1}, Ljava/lang/Integer;->toString(I)Ljava/lang/String;
  
***************
*** 3374,3379 ****
--- 3816,3822 ----
  
      invoke-interface {v0, v1, p4}, Landroid/content/SharedPreferences$Editor;->putString(Ljava/lang/String;Ljava/lang/String;)Landroid/content/SharedPreferences$Editor;
  
+     .line 806
      invoke-interface {v0}, Landroid/content/SharedPreferences$Editor;->commit()Z
  
      move-result v1
***************
*** 3389,3394 ****
--- 3832,3838 ----
  
      goto :goto_0
  
+     .line 800
      nop
  
      :pswitch_data_0
***************
*** 3406,3418 ****
--- 3850,3865 ----
      .parameter "stringValue"
  
      .prologue
+     .line 760
      packed-switch p2, :pswitch_data_0
  
+     .line 765
      const v0, 0xa80a
  
      :goto_0
      return v0
  
+     .line 762
      :pswitch_0
      invoke-direct {p0, p1, p5}, Landroid/mtp/MtpDatabase;->renameFile(ILjava/lang/String;)I
  
***************
*** 3420,3425 ****
--- 3867,3873 ----
  
      goto :goto_0
  
+     .line 760
      :pswitch_data_0
      .packed-switch 0xdc07
          :pswitch_0
***************
*** 3432,3441 ****
--- 3880,3891 ----
      .parameter "references"
  
      .prologue
+     .line 960
      const/4 v6, 0x1
  
      iput-boolean v6, p0, Landroid/mtp/MtpDatabase;->mDatabaseModified:Z
  
+     .line 961
      iget-object v6, p0, Landroid/mtp/MtpDatabase;->mVolumeName:Ljava/lang/String;
  
      int-to-long v7, p1
***************
*** 3444,3455 ****
--- 3894,3908 ----
  
      move-result-object v3
  
+     .line 962
      .local v3, uri:Landroid/net/Uri;
      array-length v0, p2
  
+     .line 963
      .local v0, count:I
      new-array v5, v0, [Landroid/content/ContentValues;
  
+     .line 964
      .local v5, valuesList:[Landroid/content/ContentValues;
      const/4 v2, 0x0
  
***************
*** 3457,3466 ****
--- 3910,3921 ----
      :goto_0
      if-ge v2, v0, :cond_0
  
+     .line 965
      new-instance v4, Landroid/content/ContentValues;
  
      invoke-direct {v4}, Landroid/content/ContentValues;-><init>()V
  
+     .line 966
      .local v4, values:Landroid/content/ContentValues;
      const-string v6, "_id"
  
***************
*** 3472,3483 ****
--- 3927,3941 ----
  
      invoke-virtual {v4, v6, v7}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 967
      aput-object v4, v5, v2
  
+     .line 964
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 970
      .end local v4           #values:Landroid/content/ContentValues;
      :cond_0
      :try_start_0
***************
*** 3491,3504 ****
--- 3949,3966 ----
  
      if-lez v6, :cond_1
  
+     .line 971
      const/16 v6, 0x2001
  
+     .line 976
      :goto_1
      return v6
  
+     .line 973
      :catch_0
      move-exception v1
  
+     .line 974
      .local v1, e:Landroid/os/RemoteException;
      const-string v6, "MtpDatabase"
  
***************
*** 3506,3511 ****
--- 3968,3974 ----
  
      invoke-static {v6, v7, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 976
      .end local v1           #e:Landroid/os/RemoteException;
      :cond_1
      const/16 v6, 0x2002
***************
*** 3520,3525 ****
--- 3983,3989 ----
      .parameter "storage"
  
      .prologue
+     .line 185
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mStorageMap:Ljava/util/HashMap;
  
      invoke-virtual {p1}, Landroid/mtp/MtpStorage;->getPath()Ljava/lang/String;
***************
*** 3528,3533 ****
--- 3992,3998 ----
  
      invoke-virtual {v0, v1, p1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 186
      return-void
  .end method
  
***************
*** 3540,3554 ****
--- 4005,4023 ----
      .end annotation
  
      .prologue
+     .line 178
      :try_start_0
      invoke-direct {p0}, Landroid/mtp/MtpDatabase;->native_finalize()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 180
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 182
      return-void
  
+     .line 180
      :catchall_0
      move-exception v0
  
***************
*** 3562,3567 ****
--- 4031,4037 ----
      .parameter "storage"
  
      .prologue
+     .line 189
      iget-object v0, p0, Landroid/mtp/MtpDatabase;->mStorageMap:Ljava/util/HashMap;
  
      invoke-virtual {p1}, Landroid/mtp/MtpStorage;->getPath()Ljava/lang/String;
***************
*** 3570,3574 ****
--- 4040,4045 ----
  
      invoke-virtual {v0, v1}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 190
      return-void
  .end method
