*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/AnimatorSet$1.smali	2012-03-23 19:35:41.129336877 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/AnimatorSet$1.smali.original	2012-03-23 19:34:44.775309264 -0500
***************
*** 29,40 ****
--- 29,42 ----
      .parameter
  
      .prologue
+     .line 506
      iput-object p1, p0, Landroid/animation/AnimatorSet$1;->this$0:Landroid/animation/AnimatorSet;
  
      iput-object p2, p0, Landroid/animation/AnimatorSet$1;->val$nodesToStart:Ljava/util/ArrayList;
  
      invoke-direct {p0}, Landroid/animation/AnimatorListenerAdapter;-><init>()V
  
+     .line 507
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/animation/AnimatorSet$1;->canceled:Z
***************
*** 49,58 ****
--- 51,62 ----
      .parameter "anim"
  
      .prologue
+     .line 509
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/animation/AnimatorSet$1;->canceled:Z
  
+     .line 510
      return-void
  .end method
  
***************
*** 61,76 ****
--- 65,83 ----
      .parameter "anim"
  
      .prologue
+     .line 512
      iget-boolean v3, p0, Landroid/animation/AnimatorSet$1;->canceled:Z
  
      if-nez v3, :cond_0
  
+     .line 513
      iget-object v3, p0, Landroid/animation/AnimatorSet$1;->val$nodesToStart:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v2
  
+     .line 514
      .local v2, numNodes:I
      const/4 v0, 0x0
  
***************
*** 78,83 ****
--- 85,91 ----
      :goto_0
      if-ge v0, v2, :cond_0
  
+     .line 515
      iget-object v3, p0, Landroid/animation/AnimatorSet$1;->val$nodesToStart:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 86,96 ****
--- 94,106 ----
  
      check-cast v1, Landroid/animation/AnimatorSet$Node;
  
+     .line 516
      .local v1, node:Landroid/animation/AnimatorSet$Node;
      iget-object v3, v1, Landroid/animation/AnimatorSet$Node;->animation:Landroid/animation/Animator;
  
      invoke-virtual {v3}, Landroid/animation/Animator;->start()V
  
+     .line 517
      iget-object v3, p0, Landroid/animation/AnimatorSet$1;->this$0:Landroid/animation/AnimatorSet;
  
      #getter for: Landroid/animation/AnimatorSet;->mPlayingSet:Ljava/util/ArrayList;
***************
*** 102,111 ****
--- 112,123 ----
  
      invoke-virtual {v3, v4}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 514
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 520
      .end local v0           #i:I
      .end local v1           #node:Landroid/animation/AnimatorSet$Node;
      .end local v2           #numNodes:I
