*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/AnimatorSet$Builder.smali	2012-03-23 19:35:40.932333285 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/AnimatorSet$Builder.smali.original	2012-03-23 19:34:44.774309246 -0500
***************
*** 27,36 ****
--- 27,38 ----
      .parameter "anim"
  
      .prologue
+     .line 1025
      iput-object p1, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1026
      #getter for: Landroid/animation/AnimatorSet;->mNodeMap:Ljava/util/HashMap;
      invoke-static {p1}, Landroid/animation/AnimatorSet;->access$100(Landroid/animation/AnimatorSet;)Ljava/util/HashMap;
  
***************
*** 44,59 ****
--- 46,64 ----
  
      iput-object v0, p0, Landroid/animation/AnimatorSet$Builder;->mCurrentNode:Landroid/animation/AnimatorSet$Node;
  
+     .line 1027
      iget-object v0, p0, Landroid/animation/AnimatorSet$Builder;->mCurrentNode:Landroid/animation/AnimatorSet$Node;
  
      if-nez v0, :cond_0
  
+     .line 1028
      new-instance v0, Landroid/animation/AnimatorSet$Node;
  
      invoke-direct {v0, p2}, Landroid/animation/AnimatorSet$Node;-><init>(Landroid/animation/Animator;)V
  
      iput-object v0, p0, Landroid/animation/AnimatorSet$Builder;->mCurrentNode:Landroid/animation/AnimatorSet$Node;
  
+     .line 1029
      #getter for: Landroid/animation/AnimatorSet;->mNodeMap:Ljava/util/HashMap;
      invoke-static {p1}, Landroid/animation/AnimatorSet;->access$100(Landroid/animation/AnimatorSet;)Ljava/util/HashMap;
  
***************
*** 63,68 ****
--- 68,74 ----
  
      invoke-virtual {v0, p2, v1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 1030
      #getter for: Landroid/animation/AnimatorSet;->mNodes:Ljava/util/ArrayList;
      invoke-static {p1}, Landroid/animation/AnimatorSet;->access$400(Landroid/animation/AnimatorSet;)Ljava/util/ArrayList;
  
***************
*** 72,77 ****
--- 78,84 ----
  
      invoke-virtual {v0, v1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 1032
      :cond_0
      return-void
  .end method
***************
*** 83,88 ****
--- 90,96 ----
      .parameter "delay"
  
      .prologue
+     .line 1103
      const/4 v1, 0x2
  
      new-array v1, v1, [F
***************
*** 93,105 ****
--- 101,117 ----
  
      move-result-object v0
  
+     .line 1104
      .local v0, anim:Landroid/animation/ValueAnimator;
      invoke-virtual {v0, p1, p2}, Landroid/animation/ValueAnimator;->setDuration(J)Landroid/animation/ValueAnimator;
  
+     .line 1105
      invoke-virtual {p0, v0}, Landroid/animation/AnimatorSet$Builder;->after(Landroid/animation/Animator;)Landroid/animation/AnimatorSet$Builder;
  
+     .line 1106
      return-object p0
  
+     .line 1103
      nop
  
      :array_0
***************
*** 114,119 ****
--- 126,132 ----
      .parameter "anim"
  
      .prologue
+     .line 1082
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
      #getter for: Landroid/animation/AnimatorSet;->mNodeMap:Ljava/util/HashMap;
***************
*** 127,140 ****
--- 140,156 ----
  
      check-cast v1, Landroid/animation/AnimatorSet$Node;
  
+     .line 1083
      .local v1, node:Landroid/animation/AnimatorSet$Node;
      if-nez v1, :cond_0
  
+     .line 1084
      new-instance v1, Landroid/animation/AnimatorSet$Node;
  
      .end local v1           #node:Landroid/animation/AnimatorSet$Node;
      invoke-direct {v1, p1}, Landroid/animation/AnimatorSet$Node;-><init>(Landroid/animation/Animator;)V
  
+     .line 1085
      .restart local v1       #node:Landroid/animation/AnimatorSet$Node;
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
***************
*** 145,150 ****
--- 161,167 ----
  
      invoke-virtual {v2, p1, v1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 1086
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
      #getter for: Landroid/animation/AnimatorSet;->mNodes:Ljava/util/ArrayList;
***************
*** 154,159 ****
--- 171,177 ----
  
      invoke-virtual {v2, v1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 1088
      :cond_0
      new-instance v0, Landroid/animation/AnimatorSet$Dependency;
  
***************
*** 161,171 ****
--- 179,191 ----
  
      invoke-direct {v0, v1, v2}, Landroid/animation/AnimatorSet$Dependency;-><init>(Landroid/animation/AnimatorSet$Node;I)V
  
+     .line 1089
      .local v0, dependency:Landroid/animation/AnimatorSet$Dependency;
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->mCurrentNode:Landroid/animation/AnimatorSet$Node;
  
      invoke-virtual {v2, v0}, Landroid/animation/AnimatorSet$Node;->addDependency(Landroid/animation/AnimatorSet$Dependency;)V
  
+     .line 1090
      return-object p0
  .end method
  
***************
*** 174,179 ****
--- 194,200 ----
      .parameter "anim"
  
      .prologue
+     .line 1062
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
      #getter for: Landroid/animation/AnimatorSet;->mNodeMap:Ljava/util/HashMap;
***************
*** 187,200 ****
--- 208,224 ----
  
      check-cast v1, Landroid/animation/AnimatorSet$Node;
  
+     .line 1063
      .local v1, node:Landroid/animation/AnimatorSet$Node;
      if-nez v1, :cond_0
  
+     .line 1064
      new-instance v1, Landroid/animation/AnimatorSet$Node;
  
      .end local v1           #node:Landroid/animation/AnimatorSet$Node;
      invoke-direct {v1, p1}, Landroid/animation/AnimatorSet$Node;-><init>(Landroid/animation/Animator;)V
  
+     .line 1065
      .restart local v1       #node:Landroid/animation/AnimatorSet$Node;
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
***************
*** 205,210 ****
--- 229,235 ----
  
      invoke-virtual {v2, p1, v1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 1066
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
      #getter for: Landroid/animation/AnimatorSet;->mNodes:Ljava/util/ArrayList;
***************
*** 214,219 ****
--- 239,245 ----
  
      invoke-virtual {v2, v1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 1068
      :cond_0
      new-instance v0, Landroid/animation/AnimatorSet$Dependency;
  
***************
*** 223,231 ****
--- 249,259 ----
  
      invoke-direct {v0, v2, v3}, Landroid/animation/AnimatorSet$Dependency;-><init>(Landroid/animation/AnimatorSet$Node;I)V
  
+     .line 1069
      .local v0, dependency:Landroid/animation/AnimatorSet$Dependency;
      invoke-virtual {v1, v0}, Landroid/animation/AnimatorSet$Node;->addDependency(Landroid/animation/AnimatorSet$Dependency;)V
  
+     .line 1070
      return-object p0
  .end method
  
***************
*** 234,239 ****
--- 262,268 ----
      .parameter "anim"
  
      .prologue
+     .line 1042
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
      #getter for: Landroid/animation/AnimatorSet;->mNodeMap:Ljava/util/HashMap;
***************
*** 247,260 ****
--- 276,292 ----
  
      check-cast v1, Landroid/animation/AnimatorSet$Node;
  
+     .line 1043
      .local v1, node:Landroid/animation/AnimatorSet$Node;
      if-nez v1, :cond_0
  
+     .line 1044
      new-instance v1, Landroid/animation/AnimatorSet$Node;
  
      .end local v1           #node:Landroid/animation/AnimatorSet$Node;
      invoke-direct {v1, p1}, Landroid/animation/AnimatorSet$Node;-><init>(Landroid/animation/Animator;)V
  
+     .line 1045
      .restart local v1       #node:Landroid/animation/AnimatorSet$Node;
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
***************
*** 265,270 ****
--- 297,303 ----
  
      invoke-virtual {v2, p1, v1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 1046
      iget-object v2, p0, Landroid/animation/AnimatorSet$Builder;->this$0:Landroid/animation/AnimatorSet;
  
      #getter for: Landroid/animation/AnimatorSet;->mNodes:Ljava/util/ArrayList;
***************
*** 274,279 ****
--- 307,313 ----
  
      invoke-virtual {v2, v1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 1048
      :cond_0
      new-instance v0, Landroid/animation/AnimatorSet$Dependency;
  
***************
*** 283,290 ****
--- 317,326 ----
  
      invoke-direct {v0, v2, v3}, Landroid/animation/AnimatorSet$Dependency;-><init>(Landroid/animation/AnimatorSet$Node;I)V
  
+     .line 1049
      .local v0, dependency:Landroid/animation/AnimatorSet$Dependency;
      invoke-virtual {v1, v0}, Landroid/animation/AnimatorSet$Node;->addDependency(Landroid/animation/AnimatorSet$Dependency;)V
  
+     .line 1050
      return-object p0
  .end method
