*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/Keyframe$FloatKeyframe.smali	2012-03-23 19:35:41.218338502 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/Keyframe$FloatKeyframe.smali.original	2012-03-23 19:34:44.776309282 -0500
***************
*** 24,37 ****
--- 24,41 ----
      .parameter "fraction"
  
      .prologue
+     .line 331
      invoke-direct {p0}, Landroid/animation/Keyframe;-><init>()V
  
+     .line 332
      iput p1, p0, Landroid/animation/Keyframe$FloatKeyframe;->mFraction:F
  
+     .line 333
      sget-object v0, Ljava/lang/Float;->TYPE:Ljava/lang/Class;
  
      iput-object v0, p0, Landroid/animation/Keyframe$FloatKeyframe;->mValueType:Ljava/lang/Class;
  
+     .line 334
      return-void
  .end method
  
***************
*** 41,60 ****
--- 45,70 ----
      .parameter "value"
  
      .prologue
+     .line 324
      invoke-direct {p0}, Landroid/animation/Keyframe;-><init>()V
  
+     .line 325
      iput p1, p0, Landroid/animation/Keyframe$FloatKeyframe;->mFraction:F
  
+     .line 326
      iput p2, p0, Landroid/animation/Keyframe$FloatKeyframe;->mValue:F
  
+     .line 327
      sget-object v0, Ljava/lang/Float;->TYPE:Ljava/lang/Class;
  
      iput-object v0, p0, Landroid/animation/Keyframe$FloatKeyframe;->mValueType:Ljava/lang/Class;
  
+     .line 328
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/animation/Keyframe$FloatKeyframe;->mHasValue:Z
  
+     .line 329
      return-void
  .end method
  
***************
*** 64,69 ****
--- 74,80 ----
      .locals 3
  
      .prologue
+     .line 353
      new-instance v0, Landroid/animation/Keyframe$FloatKeyframe;
  
      invoke-virtual {p0}, Landroid/animation/Keyframe$FloatKeyframe;->getFraction()F
***************
*** 74,79 ****
--- 85,91 ----
  
      invoke-direct {v0, v1, v2}, Landroid/animation/Keyframe$FloatKeyframe;-><init>(FF)V
  
+     .line 354
      .local v0, kfClone:Landroid/animation/Keyframe$FloatKeyframe;
      invoke-virtual {p0}, Landroid/animation/Keyframe$FloatKeyframe;->getInterpolator()Landroid/animation/TimeInterpolator;
  
***************
*** 81,86 ****
--- 93,99 ----
  
      invoke-virtual {v0, v1}, Landroid/animation/Keyframe$FloatKeyframe;->setInterpolator(Landroid/animation/TimeInterpolator;)V
  
+     .line 355
      return-object v0
  .end method
  
***************
*** 88,93 ****
--- 101,107 ----
      .locals 1
  
      .prologue
+     .line 318
      invoke-virtual {p0}, Landroid/animation/Keyframe$FloatKeyframe;->clone()Landroid/animation/Keyframe$FloatKeyframe;
  
      move-result-object v0
***************
*** 104,109 ****
--- 118,124 ----
      .end annotation
  
      .prologue
+     .line 318
      invoke-virtual {p0}, Landroid/animation/Keyframe$FloatKeyframe;->clone()Landroid/animation/Keyframe$FloatKeyframe;
  
      move-result-object v0
***************
*** 115,120 ****
--- 130,136 ----
      .locals 1
  
      .prologue
+     .line 337
      iget v0, p0, Landroid/animation/Keyframe$FloatKeyframe;->mValue:F
  
      return v0
***************
*** 124,129 ****
--- 140,146 ----
      .locals 1
  
      .prologue
+     .line 341
      iget v0, p0, Landroid/animation/Keyframe$FloatKeyframe;->mValue:F
  
      invoke-static {v0}, Ljava/lang/Float;->valueOf(F)Ljava/lang/Float;
***************
*** 138,143 ****
--- 155,161 ----
      .parameter "value"
  
      .prologue
+     .line 345
      if-eqz p1, :cond_0
  
      invoke-virtual {p1}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
***************
*** 148,153 ****
--- 166,172 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 346
      check-cast p1, Ljava/lang/Float;
  
      .end local p1
***************
*** 157,166 ****
--- 176,187 ----
  
      iput v0, p0, Landroid/animation/Keyframe$FloatKeyframe;->mValue:F
  
+     .line 347
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/animation/Keyframe$FloatKeyframe;->mHasValue:Z
  
+     .line 349
      :cond_0
      return-void
  .end method
