*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/Keyframe$ObjectKeyframe.smali	2012-03-23 19:35:40.574326757 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/Keyframe$ObjectKeyframe.smali.original	2012-03-23 19:34:44.771309190 -0500
***************
*** 25,36 ****
--- 25,40 ----
      .parameter "value"
  
      .prologue
+     .line 246
      invoke-direct {p0}, Landroid/animation/Keyframe;-><init>()V
  
+     .line 247
      iput p1, p0, Landroid/animation/Keyframe$ObjectKeyframe;->mFraction:F
  
+     .line 248
      iput-object p2, p0, Landroid/animation/Keyframe$ObjectKeyframe;->mValue:Ljava/lang/Object;
  
+     .line 249
      if-eqz p2, :cond_0
  
      const/4 v0, 0x1
***************
*** 38,43 ****
--- 42,48 ----
      :goto_0
      iput-boolean v0, p0, Landroid/animation/Keyframe$ObjectKeyframe;->mHasValue:Z
  
+     .line 250
      iget-boolean v0, p0, Landroid/animation/Keyframe$ObjectKeyframe;->mHasValue:Z
  
      if-eqz v0, :cond_1
***************
*** 49,61 ****
--- 54,69 ----
      :goto_1
      iput-object v0, p0, Landroid/animation/Keyframe$ObjectKeyframe;->mValueType:Ljava/lang/Class;
  
+     .line 251
      return-void
  
+     .line 249
      :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 250
      :cond_1
      const-class v0, Ljava/lang/Object;
  
***************
*** 68,73 ****
--- 76,82 ----
      .locals 3
  
      .prologue
+     .line 264
      new-instance v0, Landroid/animation/Keyframe$ObjectKeyframe;
  
      invoke-virtual {p0}, Landroid/animation/Keyframe$ObjectKeyframe;->getFraction()F
***************
*** 78,83 ****
--- 87,93 ----
  
      invoke-direct {v0, v1, v2}, Landroid/animation/Keyframe$ObjectKeyframe;-><init>(FLjava/lang/Object;)V
  
+     .line 265
      .local v0, kfClone:Landroid/animation/Keyframe$ObjectKeyframe;
      invoke-virtual {p0}, Landroid/animation/Keyframe$ObjectKeyframe;->getInterpolator()Landroid/animation/TimeInterpolator;
  
***************
*** 85,90 ****
--- 95,101 ----
  
      invoke-virtual {v0, v1}, Landroid/animation/Keyframe$ObjectKeyframe;->setInterpolator(Landroid/animation/TimeInterpolator;)V
  
+     .line 266
      return-object v0
  .end method
  
***************
*** 92,97 ****
--- 103,109 ----
      .locals 1
  
      .prologue
+     .line 239
      invoke-virtual {p0}, Landroid/animation/Keyframe$ObjectKeyframe;->clone()Landroid/animation/Keyframe$ObjectKeyframe;
  
      move-result-object v0
***************
*** 108,113 ****
--- 120,126 ----
      .end annotation
  
      .prologue
+     .line 239
      invoke-virtual {p0}, Landroid/animation/Keyframe$ObjectKeyframe;->clone()Landroid/animation/Keyframe$ObjectKeyframe;
  
      move-result-object v0
***************
*** 119,124 ****
--- 132,138 ----
      .locals 1
  
      .prologue
+     .line 254
      iget-object v0, p0, Landroid/animation/Keyframe$ObjectKeyframe;->mValue:Ljava/lang/Object;
  
      return-object v0
***************
*** 129,136 ****
--- 143,152 ----
      .parameter "value"
  
      .prologue
+     .line 258
      iput-object p1, p0, Landroid/animation/Keyframe$ObjectKeyframe;->mValue:Ljava/lang/Object;
  
+     .line 259
      if-eqz p1, :cond_0
  
      const/4 v0, 0x1
***************
*** 138,145 ****
--- 154,163 ----
      :goto_0
      iput-boolean v0, p0, Landroid/animation/Keyframe$ObjectKeyframe;->mHasValue:Z
  
+     .line 260
      return-void
  
+     .line 259
      :cond_0
      const/4 v0, 0x0
  
