*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/LayoutTransition$2.smali	2012-03-23 19:35:41.280339632 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/LayoutTransition$2.smali.original	2012-03-23 19:34:44.777309300 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 695
      iput-object p1, p0, Landroid/animation/LayoutTransition$2;->this$0:Landroid/animation/LayoutTransition;
  
      iput-object p2, p0, Landroid/animation/LayoutTransition$2;->val$child:Landroid/view/View;
***************
*** 43,48 ****
--- 44,50 ----
      .parameter "animation"
  
      .prologue
+     .line 698
      iget-object v0, p0, Landroid/animation/LayoutTransition$2;->this$0:Landroid/animation/LayoutTransition;
  
      #getter for: Landroid/animation/LayoutTransition;->pendingAnimations:Ljava/util/HashMap;
***************
*** 54,58 ****
--- 56,61 ----
  
      invoke-virtual {v0, v1}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 699
      return-void
  .end method
