*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/LayoutTransition$5.smali	2012-03-23 19:35:40.233320539 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/LayoutTransition$5.smali.original	2012-03-23 19:34:44.768309136 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 964
      iput-object p1, p0, Landroid/animation/LayoutTransition$5;->this$0:Landroid/animation/LayoutTransition;
  
      iput-object p2, p0, Landroid/animation/LayoutTransition$5;->val$child:Landroid/view/View;
***************
*** 48,53 ****
--- 49,55 ----
      .parameter "anim"
  
      .prologue
+     .line 967
      iget-object v2, p0, Landroid/animation/LayoutTransition$5;->this$0:Landroid/animation/LayoutTransition;
  
      #getter for: Landroid/animation/LayoutTransition;->currentAppearingAnimations:Ljava/util/LinkedHashMap;
***************
*** 59,64 ****
--- 61,67 ----
  
      invoke-virtual {v2, v3}, Ljava/util/LinkedHashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 968
      iget-object v2, p0, Landroid/animation/LayoutTransition$5;->this$0:Landroid/animation/LayoutTransition;
  
      #getter for: Landroid/animation/LayoutTransition;->mListeners:Ljava/util/ArrayList;
***************
*** 84,89 ****
--- 87,93 ----
  
      check-cast v1, Landroid/animation/LayoutTransition$TransitionListener;
  
+     .line 969
      .local v1, listener:Landroid/animation/LayoutTransition$TransitionListener;
      iget-object v2, p0, Landroid/animation/LayoutTransition$5;->this$0:Landroid/animation/LayoutTransition;
  
***************
*** 97,102 ****
--- 101,107 ----
  
      goto :goto_0
  
+     .line 971
      .end local v1           #listener:Landroid/animation/LayoutTransition$TransitionListener;
      :cond_0
      return-void
