*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/LayoutTransition$4.smali	2012-03-23 19:35:40.315322035 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/animation/LayoutTransition$4.smali.original	2012-03-23 19:34:44.769309154 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .parameter
  
      .prologue
+     .line 761
      iput-object p1, p0, Landroid/animation/LayoutTransition$4;->this$0:Landroid/animation/LayoutTransition;
  
      iput-object p2, p0, Landroid/animation/LayoutTransition$4;->val$parent:Landroid/view/ViewGroup;
***************
*** 58,69 ****
--- 59,72 ----
      .parameter "animator"
  
      .prologue
+     .line 776
      iget-object v0, p0, Landroid/animation/LayoutTransition$4;->val$child:Landroid/view/View;
  
      iget-object v1, p0, Landroid/animation/LayoutTransition$4;->val$listener:Landroid/view/View$OnLayoutChangeListener;
  
      invoke-virtual {v0, v1}, Landroid/view/View;->removeOnLayoutChangeListener(Landroid/view/View$OnLayoutChangeListener;)V
  
+     .line 777
      iget-object v0, p0, Landroid/animation/LayoutTransition$4;->this$0:Landroid/animation/LayoutTransition;
  
      #getter for: Landroid/animation/LayoutTransition;->layoutChangeListenerMap:Ljava/util/HashMap;
***************
*** 75,80 ****
--- 78,84 ----
  
      invoke-virtual {v0, v1}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 778
      return-void
  .end method
  
***************
*** 83,88 ****
--- 87,93 ----
      .parameter "animator"
  
      .prologue
+     .line 782
      iget-object v2, p0, Landroid/animation/LayoutTransition$4;->this$0:Landroid/animation/LayoutTransition;
  
      #getter for: Landroid/animation/LayoutTransition;->currentChangingAnimations:Ljava/util/LinkedHashMap;
***************
*** 94,99 ****
--- 99,105 ----
  
      invoke-virtual {v2, v3}, Ljava/util/LinkedHashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 783
      iget-object v2, p0, Landroid/animation/LayoutTransition$4;->this$0:Landroid/animation/LayoutTransition;
  
      #getter for: Landroid/animation/LayoutTransition;->mListeners:Ljava/util/ArrayList;
***************
*** 103,108 ****
--- 109,115 ----
  
      if-eqz v2, :cond_1
  
+     .line 784
      iget-object v2, p0, Landroid/animation/LayoutTransition$4;->this$0:Landroid/animation/LayoutTransition;
  
      #getter for: Landroid/animation/LayoutTransition;->mListeners:Ljava/util/ArrayList;
***************
*** 128,133 ****
--- 135,141 ----
  
      check-cast v1, Landroid/animation/LayoutTransition$TransitionListener;
  
+     .line 785
      .local v1, listener:Landroid/animation/LayoutTransition$TransitionListener;
      iget-object v3, p0, Landroid/animation/LayoutTransition$4;->this$0:Landroid/animation/LayoutTransition;
  
***************
*** 153,158 ****
--- 161,167 ----
  
      goto :goto_1
  
+     .line 790
      .end local v0           #i$:Ljava/util/Iterator;
      .end local v1           #listener:Landroid/animation/LayoutTransition$TransitionListener;
      :cond_1
***************
*** 164,169 ****
--- 173,179 ----
      .parameter "animator"
  
      .prologue
+     .line 765
      iget-object v2, p0, Landroid/animation/LayoutTransition$4;->this$0:Landroid/animation/LayoutTransition;
  
      #getter for: Landroid/animation/LayoutTransition;->mListeners:Ljava/util/ArrayList;
***************
*** 173,178 ****
--- 183,189 ----
  
      if-eqz v2, :cond_1
  
+     .line 766
      iget-object v2, p0, Landroid/animation/LayoutTransition$4;->this$0:Landroid/animation/LayoutTransition;
  
      #getter for: Landroid/animation/LayoutTransition;->mListeners:Ljava/util/ArrayList;
***************
*** 198,203 ****
--- 209,215 ----
  
      check-cast v1, Landroid/animation/LayoutTransition$TransitionListener;
  
+     .line 767
      .local v1, listener:Landroid/animation/LayoutTransition$TransitionListener;
      iget-object v3, p0, Landroid/animation/LayoutTransition$4;->this$0:Landroid/animation/LayoutTransition;
  
***************
*** 223,228 ****
--- 235,241 ----
  
      goto :goto_1
  
+     .line 772
      .end local v0           #i$:Ljava/util/Iterator;
      .end local v1           #listener:Landroid/animation/LayoutTransition$TransitionListener;
      :cond_1
