*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/AssetFileDescriptor$AutoCloseInputStream.smali	2012-03-23 19:36:25.684149334 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/AssetFileDescriptor$AutoCloseInputStream.smali.original	2012-03-23 19:34:45.105315281 -0500
***************
*** 29,46 ****
--- 29,49 ----
      .end annotation
  
      .prologue
+     .line 172
      invoke-virtual {p1}, Landroid/content/res/AssetFileDescriptor;->getParcelFileDescriptor()Landroid/os/ParcelFileDescriptor;
  
      move-result-object v0
  
      invoke-direct {p0, v0}, Landroid/os/ParcelFileDescriptor$AutoCloseInputStream;-><init>(Landroid/os/ParcelFileDescriptor;)V
  
+     .line 173
      invoke-virtual {p1}, Landroid/content/res/AssetFileDescriptor;->getStartOffset()J
  
      move-result-wide v0
  
      invoke-super {p0, v0, v1}, Landroid/os/ParcelFileDescriptor$AutoCloseInputStream;->skip(J)J
  
+     .line 174
      invoke-virtual {p1}, Landroid/content/res/AssetFileDescriptor;->getLength()J
  
      move-result-wide v0
***************
*** 51,56 ****
--- 54,60 ----
  
      iput-wide v0, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
+     .line 175
      return-void
  .end method
  
***************
*** 65,70 ****
--- 69,75 ----
      .end annotation
  
      .prologue
+     .line 179
      iget-wide v0, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
      const-wide/16 v2, 0x0
***************
*** 106,111 ****
--- 111,117 ----
      .parameter "readlimit"
  
      .prologue
+     .line 224
      iget-wide v0, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
      const-wide/16 v2, 0x0
***************
*** 114,122 ****
--- 120,130 ----
  
      if-ltz v0, :cond_0
  
+     .line 229
      :goto_0
      return-void
  
+     .line 228
      :cond_0
      invoke-super {p0, p1}, Landroid/os/ParcelFileDescriptor$AutoCloseInputStream;->mark(I)V
  
***************
*** 127,132 ****
--- 135,141 ----
      .locals 4
  
      .prologue
+     .line 233
      iget-wide v0, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
      const-wide/16 v2, 0x0
***************
*** 135,142 ****
--- 144,153 ----
  
      if-ltz v0, :cond_0
  
+     .line 234
      const/4 v0, 0x0
  
+     .line 236
      :goto_0
      return v0
  
***************
*** 163,175 ****
--- 174,189 ----
  
      const/4 v2, -0x1
  
+     .line 186
      new-array v0, v4, [B
  
+     .line 187
      .local v0, buffer:[B
      invoke-virtual {p0, v0, v3, v4}, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->read([BII)I
  
      move-result v1
  
+     .line 188
      .local v1, result:I
      if-ne v1, v2, :cond_0
  
***************
*** 194,199 ****
--- 208,214 ----
      .end annotation
  
      .prologue
+     .line 206
      const/4 v0, 0x0
  
      array-length v1, p1
***************
*** 219,230 ****
--- 234,247 ----
      .prologue
      const-wide/16 v3, 0x0
  
+     .line 193
      iget-wide v1, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
      cmp-long v1, v1, v3
  
      if-ltz v1, :cond_3
  
+     .line 194
      iget-wide v1, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
      cmp-long v1, v1, v3
***************
*** 233,242 ****
--- 250,261 ----
  
      const/4 v0, -0x1
  
+     .line 201
      :cond_0
      :goto_0
      return v0
  
+     .line 195
      :cond_1
      int-to-long v1, p3
  
***************
*** 250,260 ****
--- 269,281 ----
  
      long-to-int p3, v1
  
+     .line 196
      :cond_2
      invoke-super {p0, p1, p2, p3}, Landroid/os/ParcelFileDescriptor$AutoCloseInputStream;->read([BII)I
  
      move-result v0
  
+     .line 197
      .local v0, res:I
      if-ltz v0, :cond_0
  
***************
*** 268,273 ****
--- 289,295 ----
  
      goto :goto_0
  
+     .line 201
      .end local v0           #res:I
      :cond_3
      invoke-super {p0, p1, p2, p3}, Landroid/os/ParcelFileDescriptor$AutoCloseInputStream;->read([BII)I
***************
*** 286,291 ****
--- 308,314 ----
      .end annotation
  
      .prologue
+     .line 241
      monitor-enter p0
  
      :try_start_0
***************
*** 299,309 ****
--- 322,334 ----
  
      if-ltz v0, :cond_0
  
+     .line 246
      :goto_0
      monitor-exit p0
  
      return-void
  
+     .line 245
      :cond_0
      :try_start_1
      invoke-super {p0}, Landroid/os/ParcelFileDescriptor$AutoCloseInputStream;->reset()V
***************
*** 312,317 ****
--- 337,343 ----
  
      goto :goto_0
  
+     .line 241
      :catchall_0
      move-exception v0
  
***************
*** 332,343 ****
--- 358,371 ----
      .prologue
      const-wide/16 v4, 0x0
  
+     .line 211
      iget-wide v2, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
      cmp-long v2, v2, v4
  
      if-ltz v2, :cond_3
  
+     .line 212
      iget-wide v2, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
      cmp-long v2, v2, v4
***************
*** 346,355 ****
--- 374,385 ----
  
      const-wide/16 v0, -0x1
  
+     .line 219
      :cond_0
      :goto_0
      return-wide v0
  
+     .line 213
      :cond_1
      iget-wide v2, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
***************
*** 359,369 ****
--- 389,401 ----
  
      iget-wide p1, p0, Landroid/content/res/AssetFileDescriptor$AutoCloseInputStream;->mRemaining:J
  
+     .line 214
      :cond_2
      invoke-super {p0, p1, p2}, Landroid/os/ParcelFileDescriptor$AutoCloseInputStream;->skip(J)J
  
      move-result-wide v0
  
+     .line 215
      .local v0, res:J
      cmp-long v2, v0, v4
  
***************
*** 377,382 ****
--- 409,415 ----
  
      goto :goto_0
  
+     .line 219
      .end local v0           #res:J
      :cond_3
      invoke-super {p0, p1, p2}, Landroid/os/ParcelFileDescriptor$AutoCloseInputStream;->skip(J)J
