*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/AssetManager.smali	2012-03-23 19:36:25.732150209 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/AssetManager.smali.original	2012-03-23 19:34:45.106315299 -0500
***************
*** 79,90 ****
--- 79,92 ----
      .locals 1
  
      .prologue
+     .line 64
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      sput-object v0, Landroid/content/res/AssetManager;->sSync:Ljava/lang/Object;
  
+     .line 65
      const/4 v0, 0x0
  
      sput-object v0, Landroid/content/res/AssetManager;->sSystem:Landroid/content/res/AssetManager;
***************
*** 98,136 ****
--- 100,150 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 87
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 67
      new-instance v0, Landroid/util/TypedValue;
  
      invoke-direct {v0}, Landroid/util/TypedValue;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/AssetManager;->mValue:Landroid/util/TypedValue;
  
+     .line 68
      const/4 v0, 0x2
  
      new-array v0, v0, [J
  
      iput-object v0, p0, Landroid/content/res/AssetManager;->mOffsets:[J
  
+     .line 74
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
+     .line 76
      iput v1, p0, Landroid/content/res/AssetManager;->mNumRefs:I
  
+     .line 77
      iput-boolean v1, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
+     .line 88
      monitor-enter p0
  
+     .line 93
      :try_start_0
      invoke-direct {p0}, Landroid/content/res/AssetManager;->init()V
  
+     .line 95
      invoke-static {}, Landroid/content/res/AssetManager;->ensureSystemAssets()V
  
+     .line 96
      monitor-exit p0
  
+     .line 97
      return-void
  
+     .line 96
      :catchall_0
      move-exception v0
  
***************
*** 148,177 ****
--- 162,199 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 109
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 67
      new-instance v0, Landroid/util/TypedValue;
  
      invoke-direct {v0}, Landroid/util/TypedValue;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/AssetManager;->mValue:Landroid/util/TypedValue;
  
+     .line 68
      const/4 v0, 0x2
  
      new-array v0, v0, [J
  
      iput-object v0, p0, Landroid/content/res/AssetManager;->mOffsets:[J
  
+     .line 74
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
+     .line 76
      iput v1, p0, Landroid/content/res/AssetManager;->mNumRefs:I
  
+     .line 77
      iput-boolean v1, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
+     .line 116
      invoke-direct {p0}, Landroid/content/res/AssetManager;->init()V
  
+     .line 118
      return-void
  .end method
  
***************
*** 181,186 ****
--- 203,209 ----
      .parameter "x1"
  
      .prologue
+     .line 35
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->getAssetLength(I)J
  
      move-result-wide v0
***************
*** 194,199 ****
--- 217,223 ----
      .parameter "x1"
  
      .prologue
+     .line 35
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->readAssetChar(I)I
  
      move-result v0
***************
*** 207,212 ****
--- 231,237 ----
      .parameter "x1"
  
      .prologue
+     .line 35
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->getAssetRemainingLength(I)J
  
      move-result-wide v0
***************
*** 220,225 ****
--- 245,251 ----
      .parameter "x1"
  
      .prologue
+     .line 35
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->destroyAsset(I)V
  
      return-void
***************
*** 231,236 ****
--- 257,263 ----
      .parameter "x1"
  
      .prologue
+     .line 35
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->decRefsLocked(I)V
  
      return-void
***************
*** 244,249 ****
--- 271,277 ----
      .parameter "x3"
  
      .prologue
+     .line 35
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/content/res/AssetManager;->seekAsset(IJI)J
  
      move-result-wide v0
***************
*** 260,265 ****
--- 288,294 ----
      .parameter "x4"
  
      .prologue
+     .line 35
      invoke-direct {p0, p1, p2, p3, p4}, Landroid/content/res/AssetManager;->readAsset(I[BII)I
  
      move-result v0
***************
*** 281,298 ****
--- 310,331 ----
      .parameter "id"
  
      .prologue
+     .line 763
      iget v0, p0, Landroid/content/res/AssetManager;->mNumRefs:I
  
      add-int/lit8 v0, v0, -0x1
  
      iput v0, p0, Landroid/content/res/AssetManager;->mNumRefs:I
  
+     .line 766
      iget v0, p0, Landroid/content/res/AssetManager;->mNumRefs:I
  
      if-nez v0, :cond_0
  
+     .line 767
      invoke-direct {p0}, Landroid/content/res/AssetManager;->destroy()V
  
+     .line 769
      :cond_0
      return-void
  .end method
***************
*** 313,345 ****
--- 346,386 ----
      .locals 3
  
      .prologue
+     .line 100
      sget-object v2, Landroid/content/res/AssetManager;->sSync:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 101
      :try_start_0
      sget-object v1, Landroid/content/res/AssetManager;->sSystem:Landroid/content/res/AssetManager;
  
      if-nez v1, :cond_0
  
+     .line 102
      new-instance v0, Landroid/content/res/AssetManager;
  
      const/4 v1, 0x1
  
      invoke-direct {v0, v1}, Landroid/content/res/AssetManager;-><init>(Z)V
  
+     .line 103
      .local v0, system:Landroid/content/res/AssetManager;
      const/4 v1, 0x0
  
      invoke-virtual {v0, v1}, Landroid/content/res/AssetManager;->makeStringBlocks(Z)V
  
+     .line 104
      sput-object v0, Landroid/content/res/AssetManager;->sSystem:Landroid/content/res/AssetManager;
  
+     .line 106
      :cond_0
      monitor-exit v2
  
+     .line 107
      return-void
  
+     .line 106
      :catchall_0
      move-exception v1
  
***************
*** 381,388 ****
--- 422,431 ----
      .locals 1
  
      .prologue
+     .line 126
      invoke-static {}, Landroid/content/res/AssetManager;->ensureSystemAssets()V
  
+     .line 127
      sget-object v0, Landroid/content/res/AssetManager;->sSystem:Landroid/content/res/AssetManager;
  
      return-object v0
***************
*** 393,404 ****
--- 436,449 ----
      .parameter "id"
  
      .prologue
+     .line 756
      iget v0, p0, Landroid/content/res/AssetManager;->mNumRefs:I
  
      add-int/lit8 v0, v0, 0x1
  
      iput v0, p0, Landroid/content/res/AssetManager;->mNumRefs:I
  
+     .line 757
      return-void
  .end method
  
***************
*** 461,478 ****
--- 506,528 ----
      .parameter "paths"
  
      .prologue
+     .line 615
      if-nez p1, :cond_1
  
+     .line 616
      const/4 v0, 0x0
  
+     .line 624
      :cond_0
      return-object v0
  
+     .line 619
      :cond_1
      array-length v2, p1
  
      new-array v0, v2, [I
  
+     .line 620
      .local v0, cookies:[I
      const/4 v1, 0x0
  
***************
*** 482,487 ****
--- 532,538 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 621
      aget-object v2, p1, v1
  
      invoke-virtual {p0, v2}, Landroid/content/res/AssetManager;->addAssetPath(Ljava/lang/String;)I
***************
*** 490,495 ****
--- 541,547 ----
  
      aput v2, v0, v1
  
+     .line 620
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
***************
*** 499,526 ****
--- 551,585 ----
      .locals 1
  
      .prologue
+     .line 134
      monitor-enter p0
  
+     .line 137
      :try_start_0
      iget-boolean v0, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
      if-eqz v0, :cond_0
  
+     .line 138
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
+     .line 139
      invoke-virtual {p0}, Ljava/lang/Object;->hashCode()I
  
      move-result v0
  
      invoke-direct {p0, v0}, Landroid/content/res/AssetManager;->decRefsLocked(I)V
  
+     .line 141
      :cond_0
      monitor-exit p0
  
+     .line 142
      return-void
  
+     .line 141
      :catchall_0
      move-exception v0
  
***************
*** 535,547 ****
--- 594,609 ----
      .locals 3
  
      .prologue
+     .line 504
      monitor-enter p0
  
+     .line 505
      :try_start_0
      iget-boolean v1, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
      if-nez v1, :cond_0
  
+     .line 506
      new-instance v1, Ljava/lang/RuntimeException;
  
      const-string v2, "Assetmanager has been closed"
***************
*** 550,555 ****
--- 612,618 ----
  
      throw v1
  
+     .line 511
      :catchall_0
      move-exception v1
  
***************
*** 559,573 ****
--- 622,639 ----
  
      throw v1
  
+     .line 508
      :cond_0
      :try_start_1
      invoke-direct {p0}, Landroid/content/res/AssetManager;->newTheme()I
  
      move-result v0
  
+     .line 509
      .local v0, res:I
      invoke-direct {p0, v0}, Landroid/content/res/AssetManager;->incRefsLocked(I)V
  
+     .line 510
      monitor-exit p0
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
***************
*** 579,605 ****
--- 645,678 ----
      .locals 1
  
      .prologue
+     .line 246
      iget-object v0, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
      if-nez v0, :cond_1
  
+     .line 247
      monitor-enter p0
  
+     .line 248
      :try_start_0
      iget-object v0, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
      if-nez v0, :cond_0
  
+     .line 249
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Landroid/content/res/AssetManager;->makeStringBlocks(Z)V
  
+     .line 251
      :cond_0
      monitor-exit p0
  
+     .line 253
      :cond_1
      return-void
  
+     .line 251
      :catchall_0
      move-exception v0
  
***************
*** 619,633 ****
--- 692,710 ----
      .end annotation
  
      .prologue
+     .line 532
      :try_start_0
      invoke-direct {p0}, Landroid/content/res/AssetManager;->destroy()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 534
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 536
      return-void
  
+     .line 534
      :catchall_0
      move-exception v0
  
***************
*** 654,659 ****
--- 731,737 ----
      .parameter "id"
  
      .prologue
+     .line 274
      iget-object v0, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
      add-int/lit8 v1, p1, -0x1
***************
*** 673,683 ****
--- 751,764 ----
      .parameter "bagEntryId"
  
      .prologue
+     .line 167
      monitor-enter p0
  
+     .line 168
      :try_start_0
      iget-object v1, p0, Landroid/content/res/AssetManager;->mValue:Landroid/util/TypedValue;
  
+     .line 169
      .local v1, tmpValue:Landroid/util/TypedValue;
      const/4 v2, 0x1
  
***************
*** 685,699 ****
--- 766,783 ----
  
      move-result v0
  
+     .line 170
      .local v0, block:I
      if-ltz v0, :cond_1
  
+     .line 171
      iget v2, v1, Landroid/util/TypedValue;->type:I
  
      const/4 v3, 0x3
  
      if-ne v2, v3, :cond_0
  
+     .line 172
      iget-object v2, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
      aget-object v2, v2, v0
***************
*** 706,714 ****
--- 790,800 ----
  
      monitor-exit p0
  
+     .line 177
      :goto_0
      return-object v2
  
+     .line 174
      :cond_0
      invoke-virtual {v1}, Landroid/util/TypedValue;->coerceToString()Ljava/lang/CharSequence;
  
***************
*** 718,723 ****
--- 804,810 ----
  
      goto :goto_0
  
+     .line 176
      .end local v0           #block:I
      .end local v1           #tmpValue:Landroid/util/TypedValue;
      :catchall_0
***************
*** 737,742 ****
--- 824,830 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 177
      const/4 v2, 0x0
  
      goto :goto_0
***************
*** 759,768 ****
--- 847,858 ----
      .parameter "id"
  
      .prologue
+     .line 186
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->getArrayStringResource(I)[Ljava/lang/String;
  
      move-result-object v0
  
+     .line 187
      .local v0, retArray:[Ljava/lang/String;
      return-object v0
  .end method
***************
*** 772,782 ****
--- 862,875 ----
      .parameter "ident"
  
      .prologue
+     .line 149
      monitor-enter p0
  
+     .line 150
      :try_start_0
      iget-object v1, p0, Landroid/content/res/AssetManager;->mValue:Landroid/util/TypedValue;
  
+     .line 151
      .local v1, tmpValue:Landroid/util/TypedValue;
      const/4 v2, 0x0
  
***************
*** 786,800 ****
--- 879,896 ----
  
      move-result v0
  
+     .line 152
      .local v0, block:I
      if-ltz v0, :cond_1
  
+     .line 153
      iget v2, v1, Landroid/util/TypedValue;->type:I
  
      const/4 v3, 0x3
  
      if-ne v2, v3, :cond_0
  
+     .line 154
      iget-object v2, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
      aget-object v2, v2, v0
***************
*** 807,815 ****
--- 903,913 ----
  
      monitor-exit p0
  
+     .line 159
      :goto_0
      return-object v2
  
+     .line 156
      :cond_0
      invoke-virtual {v1}, Landroid/util/TypedValue;->coerceToString()Ljava/lang/CharSequence;
  
***************
*** 819,824 ****
--- 917,923 ----
  
      goto :goto_0
  
+     .line 158
      .end local v0           #block:I
      .end local v1           #tmpValue:Landroid/util/TypedValue;
      :catchall_0
***************
*** 838,843 ****
--- 937,943 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 159
      const/4 v2, 0x0
  
      goto :goto_0
***************
*** 848,866 ****
--- 948,971 ----
      .parameter "id"
  
      .prologue
+     .line 213
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->getArrayStringInfo(I)[I
  
      move-result-object v5
  
+     .line 214
      .local v5, rawInfoArray:[I
      array-length v6, v5
  
+     .line 215
      .local v6, rawInfoArrayLen:I
      div-int/lit8 v3, v6, 0x2
  
+     .line 218
      .local v3, infoArrayLen:I
      new-array v7, v3, [Ljava/lang/CharSequence;
  
+     .line 219
      .local v7, retArray:[Ljava/lang/CharSequence;
      const/4 v1, 0x0
  
***************
*** 871,883 ****
--- 976,991 ----
      :goto_0
      if-ge v1, v6, :cond_1
  
+     .line 220
      aget v0, v5, v1
  
+     .line 221
      .local v0, block:I
      add-int/lit8 v8, v1, 0x1
  
      aget v2, v5, v8
  
+     .line 222
      .local v2, index:I
      if-ltz v2, :cond_0
  
***************
*** 892,908 ****
--- 1000,1019 ----
      :goto_1
      aput-object v8, v7, v4
  
+     .line 219
      add-int/lit8 v1, v1, 0x2
  
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
  
+     .line 222
      :cond_0
      const/4 v8, 0x0
  
      goto :goto_1
  
+     .line 224
      .end local v0           #block:I
      .end local v2           #index:I
      :cond_1
***************
*** 922,945 ****
--- 1033,1061 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 196
      int-to-short v2, p2
  
      invoke-direct {p0, p1, v2, p3, p4}, Landroid/content/res/AssetManager;->loadResourceValue(ISLandroid/util/TypedValue;Z)I
  
      move-result v0
  
+     .line 197
      .local v0, block:I
      if-ltz v0, :cond_1
  
+     .line 198
      iget v2, p3, Landroid/util/TypedValue;->type:I
  
      const/4 v3, 0x3
  
      if-eq v2, v3, :cond_0
  
+     .line 204
      :goto_0
      return v1
  
+     .line 201
      :cond_0
      iget-object v2, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
***************
*** 955,960 ****
--- 1071,1077 ----
  
      goto :goto_0
  
+     .line 204
      :cond_1
      const/4 v1, 0x0
  
***************
*** 971,1002 ****
--- 1088,1128 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 229
      invoke-static {p1, p2, p3, p4}, Landroid/content/res/AssetManager;->loadThemeAttributeValue(IILandroid/util/TypedValue;Z)I
  
      move-result v0
  
+     .line 230
      .local v0, block:I
      if-ltz v0, :cond_2
  
+     .line 231
      iget v3, p3, Landroid/util/TypedValue;->type:I
  
      const/4 v4, 0x3
  
      if-eq v3, v4, :cond_0
  
+     .line 242
      :goto_0
      return v2
  
+     .line 234
      :cond_0
      iget-object v1, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
+     .line 235
      .local v1, blocks:[Landroid/content/res/StringBlock;
      if-nez v1, :cond_1
  
+     .line 236
      invoke-virtual {p0}, Landroid/content/res/AssetManager;->ensureStringBlocks()V
  
+     .line 237
      iget-object v1, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
+     .line 239
      :cond_1
      aget-object v3, v1, v0
  
***************
*** 1010,1015 ****
--- 1136,1142 ----
  
      goto :goto_0
  
+     .line 242
      .end local v1           #blocks:[Landroid/content/res/StringBlock;
      :cond_2
      const/4 v2, 0x0
***************
*** 1033,1038 ****
--- 1160,1166 ----
      .parameter "copyFromSystem"
  
      .prologue
+     .line 256
      if-eqz p1, :cond_0
  
      sget-object v3, Landroid/content/res/AssetManager;->sSystem:Landroid/content/res/AssetManager;
***************
*** 1041,1065 ****
--- 1169,1198 ----
  
      array-length v2, v3
  
+     .line 257
      .local v2, sysNum:I
      :goto_0
      invoke-direct {p0}, Landroid/content/res/AssetManager;->getStringBlockCount()I
  
      move-result v1
  
+     .line 258
      .local v1, num:I
      new-array v3, v1, [Landroid/content/res/StringBlock;
  
      iput-object v3, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
+     .line 261
      const/4 v0, 0x0
  
      .local v0, i:I
      :goto_1
      if-ge v0, v1, :cond_2
  
+     .line 262
      if-ge v0, v2, :cond_1
  
+     .line 263
      iget-object v3, p0, Landroid/content/res/AssetManager;->mStringBlocks:[Landroid/content/res/StringBlock;
  
      sget-object v4, Landroid/content/res/AssetManager;->sSystem:Landroid/content/res/AssetManager;
***************
*** 1070,1080 ****
--- 1203,1215 ----
  
      aput-object v4, v3, v0
  
+     .line 261
      :goto_2
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 256
      .end local v0           #i:I
      .end local v1           #num:I
      .end local v2           #sysNum:I
***************
*** 1083,1088 ****
--- 1218,1224 ----
  
      goto :goto_0
  
+     .line 265
      .restart local v0       #i:I
      .restart local v1       #num:I
      .restart local v2       #sysNum:I
***************
*** 1103,1108 ****
--- 1239,1245 ----
  
      goto :goto_2
  
+     .line 268
      :cond_2
      return-void
  .end method
***************
*** 1117,1122 ****
--- 1254,1260 ----
      .end annotation
  
      .prologue
+     .line 289
      const/4 v0, 0x2
  
      invoke-virtual {p0, p1, v0}, Landroid/content/res/AssetManager;->open(Ljava/lang/String;I)Ljava/io/InputStream;
***************
*** 1137,1149 ****
--- 1275,1290 ----
      .end annotation
  
      .prologue
+     .line 311
      monitor-enter p0
  
+     .line 312
      :try_start_0
      iget-boolean v2, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
      if-nez v2, :cond_0
  
+     .line 313
      new-instance v2, Ljava/lang/RuntimeException;
  
      const-string v3, "Assetmanager has been closed"
***************
*** 1152,1157 ****
--- 1293,1299 ----
  
      throw v2
  
+     .line 321
      :catchall_0
      move-exception v2
  
***************
*** 1161,1181 ****
--- 1303,1327 ----
  
      throw v2
  
+     .line 315
      :cond_0
      :try_start_1
      invoke-direct {p0, p1, p2}, Landroid/content/res/AssetManager;->openAsset(Ljava/lang/String;I)I
  
      move-result v0
  
+     .line 316
      .local v0, asset:I
      if-eqz v0, :cond_1
  
+     .line 317
      new-instance v1, Landroid/content/res/AssetManager$AssetInputStream;
  
      const/4 v2, 0x0
  
      invoke-direct {v1, p0, v0, v2}, Landroid/content/res/AssetManager$AssetInputStream;-><init>(Landroid/content/res/AssetManager;ILandroid/content/res/AssetManager$1;)V
  
+     .line 318
      .local v1, res:Landroid/content/res/AssetManager$AssetInputStream;
      invoke-virtual {v1}, Ljava/lang/Object;->hashCode()I
  
***************
*** 1183,1198 ****
--- 1329,1347 ----
  
      invoke-direct {p0, v2}, Landroid/content/res/AssetManager;->incRefsLocked(I)V
  
+     .line 319
      monitor-exit p0
  
      return-object v1
  
+     .line 321
      .end local v1           #res:Landroid/content/res/AssetManager$AssetInputStream;
      :cond_1
      monitor-exit p0
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 322
      new-instance v2, Ljava/io/FileNotFoundException;
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 1228,1240 ****
--- 1377,1392 ----
      .end annotation
  
      .prologue
+     .line 327
      monitor-enter p0
  
+     .line 328
      :try_start_0
      iget-boolean v0, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
      if-nez v0, :cond_0
  
+     .line 329
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v2, "Assetmanager has been closed"
***************
*** 1243,1248 ****
--- 1395,1401 ----
  
      throw v0
  
+     .line 335
      :catchall_0
      move-exception v0
  
***************
*** 1252,1257 ****
--- 1405,1411 ----
  
      throw v0
  
+     .line 331
      :cond_0
      :try_start_1
      iget-object v0, p0, Landroid/content/res/AssetManager;->mOffsets:[J
***************
*** 1260,1268 ****
--- 1414,1424 ----
  
      move-result-object v1
  
+     .line 332
      .local v1, pfd:Landroid/os/ParcelFileDescriptor;
      if-eqz v1, :cond_1
  
+     .line 333
      new-instance v0, Landroid/content/res/AssetFileDescriptor;
  
      iget-object v2, p0, Landroid/content/res/AssetManager;->mOffsets:[J
***************
*** 1283,1293 ****
--- 1439,1451 ----
  
      return-object v0
  
+     .line 335
      :cond_1
      monitor-exit p0
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 336
      new-instance v0, Ljava/io/FileNotFoundException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 1324,1329 ****
--- 1482,1488 ----
      .end annotation
  
      .prologue
+     .line 390
      const/4 v0, 0x2
  
      invoke-virtual {p0, p1, p2, v0}, Landroid/content/res/AssetManager;->openNonAsset(ILjava/lang/String;I)Ljava/io/InputStream;
***************
*** 1345,1357 ****
--- 1504,1519 ----
      .end annotation
  
      .prologue
+     .line 403
      monitor-enter p0
  
+     .line 404
      :try_start_0
      iget-boolean v2, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
      if-nez v2, :cond_0
  
+     .line 405
      new-instance v2, Ljava/lang/RuntimeException;
  
      const-string v3, "Assetmanager has been closed"
***************
*** 1360,1365 ****
--- 1522,1528 ----
  
      throw v2
  
+     .line 413
      :catchall_0
      move-exception v2
  
***************
*** 1369,1389 ****
--- 1532,1556 ----
  
      throw v2
  
+     .line 407
      :cond_0
      :try_start_1
      invoke-direct {p0, p1, p2, p3}, Landroid/content/res/AssetManager;->openNonAssetNative(ILjava/lang/String;I)I
  
      move-result v0
  
+     .line 408
      .local v0, asset:I
      if-eqz v0, :cond_1
  
+     .line 409
      new-instance v1, Landroid/content/res/AssetManager$AssetInputStream;
  
      const/4 v2, 0x0
  
      invoke-direct {v1, p0, v0, v2}, Landroid/content/res/AssetManager$AssetInputStream;-><init>(Landroid/content/res/AssetManager;ILandroid/content/res/AssetManager$1;)V
  
+     .line 410
      .local v1, res:Landroid/content/res/AssetManager$AssetInputStream;
      invoke-virtual {v1}, Ljava/lang/Object;->hashCode()I
  
***************
*** 1391,1406 ****
--- 1558,1576 ----
  
      invoke-direct {p0, v2}, Landroid/content/res/AssetManager;->incRefsLocked(I)V
  
+     .line 411
      monitor-exit p0
  
      return-object v1
  
+     .line 413
      .end local v1           #res:Landroid/content/res/AssetManager$AssetInputStream;
      :cond_1
      monitor-exit p0
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 414
      new-instance v2, Ljava/io/FileNotFoundException;
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 1436,1441 ****
--- 1606,1612 ----
      .end annotation
  
      .prologue
+     .line 364
      const/4 v0, 0x0
  
      const/4 v1, 0x2
***************
*** 1458,1463 ****
--- 1629,1635 ----
      .end annotation
  
      .prologue
+     .line 378
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0, p1, p2}, Landroid/content/res/AssetManager;->openNonAsset(ILjava/lang/String;I)Ljava/io/InputStream;
***************
*** 1478,1490 ****
--- 1650,1665 ----
      .end annotation
  
      .prologue
+     .line 424
      monitor-enter p0
  
+     .line 425
      :try_start_0
      iget-boolean v0, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
      if-nez v0, :cond_0
  
+     .line 426
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v2, "Assetmanager has been closed"
***************
*** 1493,1498 ****
--- 1668,1674 ----
  
      throw v0
  
+     .line 433
      :catchall_0
      move-exception v0
  
***************
*** 1502,1507 ****
--- 1678,1684 ----
  
      throw v0
  
+     .line 428
      :cond_0
      :try_start_1
      iget-object v0, p0, Landroid/content/res/AssetManager;->mOffsets:[J
***************
*** 1510,1518 ****
--- 1687,1697 ----
  
      move-result-object v1
  
+     .line 430
      .local v1, pfd:Landroid/os/ParcelFileDescriptor;
      if-eqz v1, :cond_1
  
+     .line 431
      new-instance v0, Landroid/content/res/AssetFileDescriptor;
  
      iget-object v2, p0, Landroid/content/res/AssetManager;->mOffsets:[J
***************
*** 1533,1543 ****
--- 1712,1724 ----
  
      return-object v0
  
+     .line 433
      :cond_1
      monitor-exit p0
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 434
      new-instance v0, Ljava/io/FileNotFoundException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 1573,1578 ****
--- 1754,1760 ----
      .end annotation
  
      .prologue
+     .line 419
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0, p1}, Landroid/content/res/AssetManager;->openNonAssetFd(ILjava/lang/String;)Landroid/content/res/AssetFileDescriptor;
***************
*** 1593,1605 ****
--- 1775,1790 ----
      .end annotation
  
      .prologue
+     .line 483
      monitor-enter p0
  
+     .line 484
      :try_start_0
      iget-boolean v2, p0, Landroid/content/res/AssetManager;->mOpen:Z
  
      if-nez v2, :cond_0
  
+     .line 485
      new-instance v2, Ljava/lang/RuntimeException;
  
      const-string v3, "Assetmanager has been closed"
***************
*** 1608,1613 ****
--- 1793,1799 ----
  
      throw v2
  
+     .line 493
      :catchall_0
      move-exception v2
  
***************
*** 1617,1635 ****
--- 1803,1825 ----
  
      throw v2
  
+     .line 487
      :cond_0
      :try_start_1
      invoke-direct {p0, p1, p2}, Landroid/content/res/AssetManager;->openXmlAssetNative(ILjava/lang/String;)I
  
      move-result v1
  
+     .line 488
      .local v1, xmlBlock:I
      if-eqz v1, :cond_1
  
+     .line 489
      new-instance v0, Landroid/content/res/XmlBlock;
  
      invoke-direct {v0, p0, v1}, Landroid/content/res/XmlBlock;-><init>(Landroid/content/res/AssetManager;I)V
  
+     .line 490
      .local v0, res:Landroid/content/res/XmlBlock;
      invoke-virtual {v0}, Ljava/lang/Object;->hashCode()I
  
***************
*** 1637,1652 ****
--- 1827,1845 ----
  
      invoke-direct {p0, v2}, Landroid/content/res/AssetManager;->incRefsLocked(I)V
  
+     .line 491
      monitor-exit p0
  
      return-object v0
  
+     .line 493
      .end local v0           #res:Landroid/content/res/XmlBlock;
      :cond_1
      monitor-exit p0
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 494
      new-instance v2, Ljava/io/FileNotFoundException;
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 1682,1687 ****
--- 1875,1881 ----
      .end annotation
  
      .prologue
+     .line 470
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0, p1}, Landroid/content/res/AssetManager;->openXmlBlockAsset(ILjava/lang/String;)Landroid/content/res/XmlBlock;
***************
*** 1702,1719 ****
--- 1896,1917 ----
      .end annotation
  
      .prologue
+     .line 455
      invoke-virtual {p0, p1, p2}, Landroid/content/res/AssetManager;->openXmlBlockAsset(ILjava/lang/String;)Landroid/content/res/XmlBlock;
  
      move-result-object v0
  
+     .line 456
      .local v0, block:Landroid/content/res/XmlBlock;
      invoke-virtual {v0}, Landroid/content/res/XmlBlock;->newParser()Landroid/content/res/XmlResourceParser;
  
      move-result-object v1
  
+     .line 457
      .local v1, rp:Landroid/content/res/XmlResourceParser;
      invoke-virtual {v0}, Landroid/content/res/XmlBlock;->close()V
  
+     .line 458
      return-object v1
  .end method
  
***************
*** 1727,1732 ****
--- 1925,1931 ----
      .end annotation
  
      .prologue
+     .line 444
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0, p1}, Landroid/content/res/AssetManager;->openXmlResourceParser(ILjava/lang/String;)Landroid/content/res/XmlResourceParser;
***************
*** 1741,1757 ****
--- 1940,1962 ----
      .parameter "theme"
  
      .prologue
+     .line 515
      monitor-enter p0
  
+     .line 516
      :try_start_0
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->deleteTheme(I)V
  
+     .line 517
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->decRefsLocked(I)V
  
+     .line 518
      monitor-exit p0
  
+     .line 519
      return-void
  
+     .line 518
      :catchall_0
      move-exception v0
  
***************
*** 1779,1793 ****
--- 1984,2003 ----
      .parameter "id"
  
      .prologue
+     .line 498
      monitor-enter p0
  
+     .line 499
      :try_start_0
      invoke-direct {p0, p1}, Landroid/content/res/AssetManager;->decRefsLocked(I)V
  
+     .line 500
      monitor-exit p0
  
+     .line 501
      return-void
  
+     .line 500
      :catchall_0
      move-exception v0
  
