*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/ColorStateList.smali	2012-03-23 19:36:26.441163138 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/ColorStateList.smali.original	2012-03-23 19:34:45.111315391 -0500
***************
*** 49,54 ****
--- 49,55 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 68
      const/4 v0, 0x1
  
      new-array v0, v0, [[I
***************
*** 59,70 ****
--- 60,73 ----
  
      sput-object v0, Landroid/content/res/ColorStateList;->EMPTY:[[I
  
+     .line 69
      new-instance v0, Landroid/util/SparseArray;
  
      invoke-direct {v0}, Landroid/util/SparseArray;-><init>()V
  
      sput-object v0, Landroid/content/res/ColorStateList;->sCache:Landroid/util/SparseArray;
  
+     .line 312
      new-instance v0, Landroid/content/res/ColorStateList$1;
  
      invoke-direct {v0}, Landroid/content/res/ColorStateList$1;-><init>()V
***************
*** 78,89 ****
--- 81,95 ----
      .locals 1
  
      .prologue
+     .line 72
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 66
      const/high16 v0, -0x1
  
      iput v0, p0, Landroid/content/res/ColorStateList;->mDefaultColor:I
  
+     .line 72
      return-void
  .end method
  
***************
*** 93,118 ****
--- 99,131 ----
      .parameter "colors"
  
      .prologue
+     .line 78
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 66
      const/high16 v1, -0x1
  
      iput v1, p0, Landroid/content/res/ColorStateList;->mDefaultColor:I
  
+     .line 79
      iput-object p1, p0, Landroid/content/res/ColorStateList;->mStateSpecs:[[I
  
+     .line 80
      iput-object p2, p0, Landroid/content/res/ColorStateList;->mColors:[I
  
+     .line 82
      array-length v1, p1
  
      if-lez v1, :cond_1
  
+     .line 83
      const/4 v1, 0x0
  
      aget v1, p2, v1
  
      iput v1, p0, Landroid/content/res/ColorStateList;->mDefaultColor:I
  
+     .line 85
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 121,141 ****
--- 134,158 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 86
      aget-object v1, p1, v0
  
      array-length v1, v1
  
      if-nez v1, :cond_0
  
+     .line 87
      aget v1, p2, v0
  
      iput v1, p0, Landroid/content/res/ColorStateList;->mDefaultColor:I
  
+     .line 85
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 91
      .end local v0           #i:I
      :cond_1
      return-void
***************
*** 155,164 ****
--- 172,183 ----
      .prologue
      const/4 v3, 0x2
  
+     .line 118
      invoke-static {p1}, Landroid/util/Xml;->asAttributeSet(Lorg/xmlpull/v1/XmlPullParser;)Landroid/util/AttributeSet;
  
      move-result-object v0
  
+     .line 122
      .local v0, attrs:Landroid/util/AttributeSet;
      :cond_0
      invoke-interface {p1}, Lorg/xmlpull/v1/XmlPullParser;->next()I
***************
*** 172,180 ****
--- 191,201 ----
  
      if-ne v1, v2, :cond_0
  
+     .line 125
      :cond_1
      if-eq v1, v3, :cond_2
  
+     .line 126
      new-instance v2, Lorg/xmlpull/v1/XmlPullParserException;
  
      const-string v3, "No start tag found"
***************
*** 183,188 ****
--- 204,210 ----
  
      throw v2
  
+     .line 129
      :cond_2
      invoke-static {p0, p1, v0}, Landroid/content/res/ColorStateList;->createFromXmlInner(Landroid/content/res/Resources;Lorg/xmlpull/v1/XmlPullParser;Landroid/util/AttributeSet;)Landroid/content/res/ColorStateList;
  
***************
*** 204,215 ****
      .end annotation
  
      .prologue
      invoke-interface {p1}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
  
      move-result-object v1
  
      .local v1, name:Ljava/lang/String;
!     const-string v2, "selector"
  
      invoke-virtual {v1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 226,239 ----
      .end annotation
  
      .prologue
+     .line 141
      invoke-interface {p1}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 143
      .local v1, name:Ljava/lang/String;
!     const-string/jumbo v2, "selector"
  
      invoke-virtual {v1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 217,231 ****
--- 241,259 ----
  
      if-eqz v2, :cond_0
  
+     .line 144
      new-instance v0, Landroid/content/res/ColorStateList;
  
      invoke-direct {v0}, Landroid/content/res/ColorStateList;-><init>()V
  
+     .line 150
      .local v0, colorStateList:Landroid/content/res/ColorStateList;
      invoke-direct {v0, p0, p1, p2}, Landroid/content/res/ColorStateList;->inflate(Landroid/content/res/Resources;Lorg/xmlpull/v1/XmlPullParser;Landroid/util/AttributeSet;)V
  
+     .line 151
      return-object v0
  
+     .line 146
      .end local v0           #colorStateList:Landroid/content/res/ColorStateList;
      :cond_0
      new-instance v2, Lorg/xmlpull/v1/XmlPullParserException;
***************
*** 274,299 ****
--- 302,333 ----
      .end annotation
  
      .prologue
+     .line 178
      invoke-interface/range {p2 .. p2}, Lorg/xmlpull/v1/XmlPullParser;->getDepth()I
  
      move-result v22
  
      add-int/lit8 v10, v22, 0x1
  
+     .line 181
      .local v10, innerDepth:I
      const/16 v13, 0x14
  
+     .line 182
      .local v13, listAllocated:I
      const/4 v14, 0x0
  
+     .line 183
      .local v14, listSize:I
      new-array v5, v13, [I
  
+     .line 184
      .local v5, colorList:[I
      new-array v0, v13, [[I
  
      move-object/from16 v20, v0
  
+     .line 187
      .local v20, stateSpecList:[[I
      :cond_0
      :goto_0
***************
*** 325,330 ****
--- 359,365 ----
  
      if-eq v0, v1, :cond_b
  
+     .line 189
      :cond_1
      const/16 v22, 0x2
  
***************
*** 334,339 ****
--- 369,375 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 193
      if-gt v7, v10, :cond_0
  
      invoke-interface/range {p2 .. p2}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
***************
*** 348,369 ****
--- 384,411 ----
  
      if-eqz v22, :cond_0
  
+     .line 197
      const/4 v6, 0x0
  
+     .line 198
      .local v6, colorRes:I
      const/high16 v4, -0x1
  
+     .line 199
      .local v4, color:I
      const/4 v8, 0x0
  
+     .line 202
      .local v8, haveColor:Z
      const/4 v11, 0x0
  
+     .line 203
      .local v11, j:I
      invoke-interface/range {p3 .. p3}, Landroid/util/AttributeSet;->getAttributeCount()I
  
      move-result v17
  
+     .line 204
      .local v17, numAttrs:I
      move/from16 v0, v17
  
***************
*** 371,376 ****
--- 413,419 ----
  
      move-object/from16 v19, v0
  
+     .line 205
      .local v19, stateSpec:[I
      const/4 v9, 0x0
  
***************
*** 384,398 ****
--- 427,444 ----
  
      if-ge v9, v0, :cond_2
  
+     .line 206
      move-object/from16 v0, p3
  
      invoke-interface {v0, v9}, Landroid/util/AttributeSet;->getAttributeNameResource(I)I
  
      move-result v18
  
+     .line 207
      .local v18, stateResId:I
      if-nez v18, :cond_7
  
+     .line 221
      .end local v18           #stateResId:I
      :cond_2
      move-object/from16 v0, v19
***************
*** 401,414 ****
--- 447,463 ----
  
      move-result-object v19
  
+     .line 223
      if-eqz v6, :cond_a
  
+     .line 224
      move-object/from16 v0, p1
  
      invoke-virtual {v0, v6}, Landroid/content/res/Resources;->getColor(I)I
  
      move-result v4
  
+     .line 231
      :cond_3
      if-eqz v14, :cond_4
  
***************
*** 420,430 ****
--- 469,481 ----
  
      if-nez v22, :cond_5
  
+     .line 232
      :cond_4
      move-object/from16 v0, p0
  
      iput v4, v0, Landroid/content/res/ColorStateList;->mDefaultColor:I
  
+     .line 235
      :cond_5
      add-int/lit8 v22, v14, 0x1
  
***************
*** 432,445 ****
--- 483,499 ----
  
      if-lt v0, v13, :cond_6
  
+     .line 236
      add-int/lit8 v22, v14, 0x1
  
      invoke-static/range {v22 .. v22}, Lcom/android/internal/util/ArrayUtils;->idealIntArraySize(I)I
  
      move-result v13
  
+     .line 238
      new-array v15, v13, [I
  
+     .line 239
      .local v15, ncolor:[I
      const/16 v22, 0x0
  
***************
*** 451,460 ****
--- 505,516 ----
  
      invoke-static {v5, v0, v15, v1, v14}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 241
      new-array v0, v13, [[I
  
      move-object/from16 v16, v0
  
+     .line 242
      .local v16, nstate:[[I
      const/16 v22, 0x0
  
***************
*** 470,490 ****
--- 526,553 ----
  
      invoke-static {v0, v1, v2, v3, v14}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 244
      move-object v5, v15
  
+     .line 245
      move-object/from16 v20, v16
  
+     .line 248
      .end local v15           #ncolor:[I
      .end local v16           #nstate:[[I
      :cond_6
      aput v4, v5, v14
  
+     .line 249
      aput-object v19, v20, v14
  
+     .line 250
      add-int/lit8 v14, v14, 0x1
  
+     .line 251
      goto/16 :goto_0
  
+     .line 208
      .restart local v18       #stateResId:I
      :cond_7
      const v22, 0x10101a5
***************
*** 495,500 ****
--- 558,564 ----
  
      if-ne v0, v1, :cond_8
  
+     .line 209
      const/16 v22, 0x0
  
      move-object/from16 v0, p3
***************
*** 505,522 ****
--- 569,590 ----
  
      move-result v6
  
+     .line 211
      if-nez v6, :cond_c
  
+     .line 212
      move-object/from16 v0, p3
  
      invoke-interface {v0, v9, v4}, Landroid/util/AttributeSet;->getAttributeIntValue(II)I
  
      move-result v4
  
+     .line 213
      const/4 v8, 0x1
  
      move v11, v12
  
+     .line 205
      .end local v12           #j:I
      .end local v18           #stateResId:I
      .restart local v11       #j:I
***************
*** 529,534 ****
--- 597,603 ----
      .restart local v12       #j:I
      goto :goto_1
  
+     .line 216
      .restart local v18       #stateResId:I
      :cond_8
      add-int/lit8 v11, v12, 0x1
***************
*** 563,574 ****
--- 632,645 ----
  
      goto :goto_3
  
+     .line 225
      .end local v11           #j:I
      .end local v18           #stateResId:I
      .restart local v12       #j:I
      :cond_a
      if-nez v8, :cond_3
  
+     .line 226
      new-instance v22, Lorg/xmlpull/v1/XmlPullParserException;
  
      new-instance v23, Ljava/lang/StringBuilder;
***************
*** 597,602 ****
--- 668,674 ----
  
      throw v22
  
+     .line 253
      .end local v4           #color:I
      .end local v6           #colorRes:I
      .end local v7           #depth:I
***************
*** 616,621 ****
--- 688,694 ----
  
      iput-object v0, v1, Landroid/content/res/ColorStateList;->mColors:[I
  
+     .line 254
      new-array v0, v14, [[I
  
      move-object/from16 v22, v0
***************
*** 626,631 ****
--- 699,705 ----
  
      iput-object v0, v1, Landroid/content/res/ColorStateList;->mStateSpecs:[[I
  
+     .line 255
      const/16 v22, 0x0
  
      move-object/from16 v0, p0
***************
*** 644,649 ****
--- 718,724 ----
  
      invoke-static {v5, v0, v1, v2, v14}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 256
      const/16 v22, 0x0
  
      move-object/from16 v0, p0
***************
*** 664,669 ****
--- 739,745 ----
  
      invoke-static {v0, v1, v2, v3, v14}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 257
      return-void
  
      .restart local v4       #color:I
***************
*** 688,697 ****
--- 764,775 ----
      .parameter "color"
  
      .prologue
+     .line 98
      sget-object v4, Landroid/content/res/ColorStateList;->sCache:Landroid/util/SparseArray;
  
      monitor-enter v4
  
+     .line 99
      :try_start_0
      sget-object v3, Landroid/content/res/ColorStateList;->sCache:Landroid/util/SparseArray;
  
***************
*** 701,706 ****
--- 779,785 ----
  
      check-cast v2, Ljava/lang/ref/WeakReference;
  
+     .line 100
      .local v2, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Landroid/content/res/ColorStateList;>;"
      if-eqz v2, :cond_0
  
***************
*** 712,736 ****
--- 791,820 ----
  
      move-object v0, v3
  
+     .line 102
      .local v0, csl:Landroid/content/res/ColorStateList;
      :goto_0
      if-eqz v0, :cond_1
  
+     .line 103
      monitor-exit v4
  
      move-object v1, v0
  
+     .line 108
      .end local v0           #csl:Landroid/content/res/ColorStateList;
      .local v1, csl:Ljava/lang/Object;
      :goto_1
      return-object v1
  
+     .line 100
      .end local v1           #csl:Ljava/lang/Object;
      :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 106
      .restart local v0       #csl:Landroid/content/res/ColorStateList;
      :cond_1
      new-instance v0, Landroid/content/res/ColorStateList;
***************
*** 748,753 ****
--- 832,838 ----
  
      invoke-direct {v0, v3, v5}, Landroid/content/res/ColorStateList;-><init>([[I[I)V
  
+     .line 107
      .restart local v0       #csl:Landroid/content/res/ColorStateList;
      sget-object v3, Landroid/content/res/ColorStateList;->sCache:Landroid/util/SparseArray;
  
***************
*** 757,762 ****
--- 842,848 ----
  
      invoke-virtual {v3, p0, v5}, Landroid/util/SparseArray;->put(ILjava/lang/Object;)V
  
+     .line 108
      monitor-exit v4
  
      move-object v1, v0
***************
*** 764,769 ****
--- 850,856 ----
      .restart local v1       #csl:Ljava/lang/Object;
      goto :goto_1
  
+     .line 109
      .end local v0           #csl:Landroid/content/res/ColorStateList;
      .end local v1           #csl:Ljava/lang/Object;
      .end local v2           #ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Landroid/content/res/ColorStateList;>;"
***************
*** 783,788 ****
--- 870,876 ----
      .locals 1
  
      .prologue
+     .line 300
      const/4 v0, 0x0
  
      return v0
***************
*** 794,803 ****
--- 882,893 ----
      .parameter "defaultColor"
  
      .prologue
+     .line 273
      iget-object v3, p0, Landroid/content/res/ColorStateList;->mStateSpecs:[[I
  
      array-length v1, v3
  
+     .line 274
      .local v1, setLength:I
      const/4 v0, 0x0
  
***************
*** 805,814 ****
--- 895,906 ----
      :goto_0
      if-ge v0, v1, :cond_0
  
+     .line 275
      iget-object v3, p0, Landroid/content/res/ColorStateList;->mStateSpecs:[[I
  
      aget-object v2, v3, v0
  
+     .line 276
      .local v2, stateSpec:[I
      invoke-static {v2, p1}, Landroid/util/StateSet;->stateSetMatches([I[I)Z
  
***************
*** 816,830 ****
--- 908,925 ----
  
      if-eqz v3, :cond_1
  
+     .line 277
      iget-object v3, p0, Landroid/content/res/ColorStateList;->mColors:[I
  
      aget p2, v3, v0
  
+     .line 280
      .end local v2           #stateSpec:[I
      .end local p2
      :cond_0
      return p2
  
+     .line 274
      .restart local v2       #stateSpec:[I
      .restart local p2
      :cond_1
***************
*** 837,842 ****
--- 932,938 ----
      .locals 1
  
      .prologue
+     .line 289
      iget v0, p0, Landroid/content/res/ColorStateList;->mDefaultColor:I
  
      return v0
***************
*** 848,853 ****
--- 944,950 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 260
      iget-object v1, p0, Landroid/content/res/ColorStateList;->mStateSpecs:[[I
  
      array-length v1, v1
***************
*** 867,872 ****
--- 964,970 ----
      .locals 2
  
      .prologue
+     .line 293
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 933,947 ****
--- 1031,1048 ----
      .parameter "alpha"
  
      .prologue
+     .line 160
      iget-object v3, p0, Landroid/content/res/ColorStateList;->mColors:[I
  
      array-length v3, v3
  
      new-array v0, v3, [I
  
+     .line 162
      .local v0, colors:[I
      array-length v2, v0
  
+     .line 163
      .local v2, len:I
      const/4 v1, 0x0
  
***************
*** 949,954 ****
--- 1050,1056 ----
      :goto_0
      if-ge v1, v2, :cond_0
  
+     .line 164
      iget-object v3, p0, Landroid/content/res/ColorStateList;->mColors:[I
  
      aget v3, v3, v1
***************
*** 963,972 ****
--- 1065,1076 ----
  
      aput v3, v0, v1
  
+     .line 163
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 167
      :cond_0
      new-instance v3, Landroid/content/res/ColorStateList;
  
***************
*** 983,1015 ****
--- 1087,1126 ----
      .parameter "flags"
  
      .prologue
+     .line 304
      iget-object v2, p0, Landroid/content/res/ColorStateList;->mStateSpecs:[[I
  
      array-length v0, v2
  
+     .line 305
      .local v0, N:I
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 306
      const/4 v1, 0x0
  
      .local v1, i:I
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 307
      iget-object v2, p0, Landroid/content/res/ColorStateList;->mStateSpecs:[[I
  
      aget-object v2, v2, v1
  
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->writeIntArray([I)V
  
+     .line 306
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 309
      :cond_0
      iget-object v2, p0, Landroid/content/res/ColorStateList;->mColors:[I
  
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->writeIntArray([I)V
  
+     .line 310
      return-void
  .end method
