*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/CompatibilityInfo$1.smali	2012-03-23 19:36:25.949154166 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/CompatibilityInfo$1.smali.original	2012-03-23 19:34:45.107315317 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 1
  
      .prologue
+     .line 39
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/content/res/CompatibilityInfo;-><init>(Landroid/content/res/CompatibilityInfo$1;)V
