*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/Resources.smali	2012-03-23 19:36:26.117157230 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/Resources.smali.original	2012-03-23 19:34:45.109315355 -0500
***************
*** 150,183 ****
--- 150,189 ----
      .locals 2
  
      .prologue
+     .line 81
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      sput-object v0, Landroid/content/res/Resources;->mSync:Ljava/lang/Object;
  
+     .line 82
      const/4 v0, 0x0
  
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
+     .line 87
      new-instance v0, Landroid/util/LongSparseArray;
  
      invoke-direct {v0}, Landroid/util/LongSparseArray;-><init>()V
  
      sput-object v0, Landroid/content/res/Resources;->sPreloadedDrawables:Landroid/util/LongSparseArray;
  
+     .line 89
      new-instance v0, Landroid/util/SparseArray;
  
      invoke-direct {v0}, Landroid/util/SparseArray;-><init>()V
  
      sput-object v0, Landroid/content/res/Resources;->mPreloadedColorStateLists:Landroid/util/SparseArray;
  
+     .line 91
      new-instance v0, Landroid/util/LongSparseArray;
  
      invoke-direct {v0}, Landroid/util/LongSparseArray;-><init>()V
  
      sput-object v0, Landroid/content/res/Resources;->sPreloadedColorDrawables:Landroid/util/LongSparseArray;
  
+     .line 121
      new-instance v0, Landroid/content/res/Resources$1;
  
      const/4 v1, 0x0
***************
*** 197,290 ****
--- 203,317 ----
  
      const/4 v1, 0x0
  
+     .line 2201
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 95
      new-instance v0, Landroid/util/TypedValue;
  
      invoke-direct {v0}, Landroid/util/TypedValue;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 96
      new-instance v0, Landroid/content/res/Configuration;
  
      invoke-direct {v0}, Landroid/content/res/Configuration;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mTmpConfig:Landroid/content/res/Configuration;
  
+     .line 99
      new-instance v0, Landroid/util/LongSparseArray;
  
      invoke-direct {v0}, Landroid/util/LongSparseArray;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mDrawableCache:Landroid/util/LongSparseArray;
  
+     .line 101
      new-instance v0, Landroid/util/SparseArray;
  
      invoke-direct {v0}, Landroid/util/SparseArray;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mColorStateListCache:Landroid/util/SparseArray;
  
+     .line 103
      new-instance v0, Landroid/util/LongSparseArray;
  
      invoke-direct {v0}, Landroid/util/LongSparseArray;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mColorDrawableCache:Landroid/util/LongSparseArray;
  
+     .line 107
      iput-object v1, p0, Landroid/content/res/Resources;->mCachedStyledAttributes:Landroid/content/res/TypedArray;
  
+     .line 108
      iput-object v1, p0, Landroid/content/res/Resources;->mLastRetrievedAttrs:Ljava/lang/RuntimeException;
  
+     .line 110
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/content/res/Resources;->mLastCachedXmlBlockIndex:I
  
+     .line 111
      new-array v0, v2, [I
  
      fill-array-data v0, :array_0
  
      iput-object v0, p0, Landroid/content/res/Resources;->mCachedXmlBlockIds:[I
  
+     .line 112
      new-array v0, v2, [Landroid/content/res/XmlBlock;
  
      iput-object v0, p0, Landroid/content/res/Resources;->mCachedXmlBlocks:[Landroid/content/res/XmlBlock;
  
+     .line 115
      new-instance v0, Landroid/content/res/Configuration;
  
      invoke-direct {v0}, Landroid/content/res/Configuration;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
  
+     .line 116
      new-instance v0, Landroid/util/DisplayMetrics;
  
      invoke-direct {v0}, Landroid/util/DisplayMetrics;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
  
+     .line 2202
      invoke-static {}, Landroid/content/res/AssetManager;->getSystem()Landroid/content/res/AssetManager;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
+     .line 2206
      iget-object v0, p0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
  
      invoke-virtual {v0}, Landroid/content/res/Configuration;->setToDefaults()V
  
+     .line 2207
      iget-object v0, p0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
  
      invoke-virtual {v0}, Landroid/util/DisplayMetrics;->setToDefaults()V
  
+     .line 2208
      invoke-virtual {p0, v1, v1}, Landroid/content/res/Resources;->updateConfiguration(Landroid/content/res/Configuration;Landroid/util/DisplayMetrics;)V
  
+     .line 2209
      iget-object v0, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v0}, Landroid/content/res/AssetManager;->ensureStringBlocks()V
  
+     .line 2210
      sget-object v0, Landroid/content/res/CompatibilityInfo;->DEFAULT_COMPATIBILITY_INFO:Landroid/content/res/CompatibilityInfo;
  
      iput-object v0, p0, Landroid/content/res/Resources;->mCompatibilityInfo:Landroid/content/res/CompatibilityInfo;
  
+     .line 2211
      return-void
  
+     .line 111
      nop
  
      :array_0
***************
*** 303,314 ****
--- 330,343 ----
      .parameter "config"
  
      .prologue
+     .line 185
      const/4 v0, 0x0
  
      check-cast v0, Landroid/content/res/CompatibilityInfo;
  
      invoke-direct {p0, p1, p2, p3, v0}, Landroid/content/res/Resources;-><init>(Landroid/content/res/AssetManager;Landroid/util/DisplayMetrics;Landroid/content/res/Configuration;Landroid/content/res/CompatibilityInfo;)V
  
+     .line 186
      return-void
  .end method
  
***************
*** 324,405 ****
--- 353,454 ----
  
      const/4 v1, 0x4
  
+     .line 201
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 95
      new-instance v0, Landroid/util/TypedValue;
  
      invoke-direct {v0}, Landroid/util/TypedValue;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 96
      new-instance v0, Landroid/content/res/Configuration;
  
      invoke-direct {v0}, Landroid/content/res/Configuration;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mTmpConfig:Landroid/content/res/Configuration;
  
+     .line 99
      new-instance v0, Landroid/util/LongSparseArray;
  
      invoke-direct {v0}, Landroid/util/LongSparseArray;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mDrawableCache:Landroid/util/LongSparseArray;
  
+     .line 101
      new-instance v0, Landroid/util/SparseArray;
  
      invoke-direct {v0}, Landroid/util/SparseArray;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mColorStateListCache:Landroid/util/SparseArray;
  
+     .line 103
      new-instance v0, Landroid/util/LongSparseArray;
  
      invoke-direct {v0}, Landroid/util/LongSparseArray;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mColorDrawableCache:Landroid/util/LongSparseArray;
  
+     .line 107
      iput-object v2, p0, Landroid/content/res/Resources;->mCachedStyledAttributes:Landroid/content/res/TypedArray;
  
+     .line 108
      iput-object v2, p0, Landroid/content/res/Resources;->mLastRetrievedAttrs:Ljava/lang/RuntimeException;
  
+     .line 110
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/content/res/Resources;->mLastCachedXmlBlockIndex:I
  
+     .line 111
      new-array v0, v1, [I
  
      fill-array-data v0, :array_0
  
      iput-object v0, p0, Landroid/content/res/Resources;->mCachedXmlBlockIds:[I
  
+     .line 112
      new-array v0, v1, [Landroid/content/res/XmlBlock;
  
      iput-object v0, p0, Landroid/content/res/Resources;->mCachedXmlBlocks:[Landroid/content/res/XmlBlock;
  
+     .line 115
      new-instance v0, Landroid/content/res/Configuration;
  
      invoke-direct {v0}, Landroid/content/res/Configuration;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
  
+     .line 116
      new-instance v0, Landroid/util/DisplayMetrics;
  
      invoke-direct {v0}, Landroid/util/DisplayMetrics;-><init>()V
  
      iput-object v0, p0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
  
+     .line 202
      iput-object p1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
+     .line 203
      iget-object v0, p0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
  
      invoke-virtual {v0}, Landroid/util/DisplayMetrics;->setToDefaults()V
  
+     .line 204
      iput-object p4, p0, Landroid/content/res/Resources;->mCompatibilityInfo:Landroid/content/res/CompatibilityInfo;
  
+     .line 205
      invoke-virtual {p0, p3, p2}, Landroid/content/res/Resources;->updateConfiguration(Landroid/content/res/Configuration;Landroid/util/DisplayMetrics;)V
  
+     .line 206
      invoke-virtual {p1}, Landroid/content/res/AssetManager;->ensureStringBlocks()V
  
+     .line 207
      return-void
  
+     .line 111
      :array_0
      .array-data 0x4
          0x0t 0x0t 0x0t 0x0t
***************
*** 415,420 ****
--- 464,470 ----
      .parameter "x1"
  
      .prologue
+     .line 71
      invoke-direct {p0, p1}, Landroid/content/res/Resources;->getCachedStyledAttributes(I)Landroid/content/res/TypedArray;
  
      move-result-object v0
***************
*** 427,464 ****
--- 477,522 ----
      .parameter "quantityCode"
  
      .prologue
+     .line 295
      packed-switch p0, :pswitch_data_0
  
+     .line 301
      const v0, 0x1000004
  
      :goto_0
      return v0
  
+     .line 296
      :pswitch_0
      const v0, 0x1000005
  
      goto :goto_0
  
+     .line 297
      :pswitch_1
      const v0, 0x1000006
  
      goto :goto_0
  
+     .line 298
      :pswitch_2
      const v0, 0x1000007
  
      goto :goto_0
  
+     .line 299
      :pswitch_3
      const v0, 0x1000008
  
      goto :goto_0
  
+     .line 300
      :pswitch_4
      const v0, 0x1000009
  
      goto :goto_0
  
+     .line 295
      nop
  
      :pswitch_data_0
***************
*** 488,498 ****
--- 546,558 ----
      .end annotation
  
      .prologue
+     .line 1513
      .local p1, cache:Landroid/util/LongSparseArray;,"Landroid/util/LongSparseArray<Ljava/lang/ref/WeakReference<Landroid/graphics/drawable/Drawable$ConstantState;>;>;"
      invoke-virtual {p1}, Landroid/util/LongSparseArray;->size()I
  
      move-result v0
  
+     .line 1518
      .local v0, N:I
      const/4 v2, 0x0
  
***************
*** 500,523 ****
--- 560,588 ----
      :goto_0
      if-ge v2, v0, :cond_1
  
+     .line 1519
      invoke-virtual {p1, v2}, Landroid/util/LongSparseArray;->valueAt(I)Ljava/lang/Object;
  
      move-result-object v3
  
      check-cast v3, Ljava/lang/ref/WeakReference;
  
+     .line 1520
      .local v3, ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Landroid/graphics/drawable/Drawable$ConstantState;>;"
      if-eqz v3, :cond_0
  
+     .line 1521
      invoke-virtual {v3}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Landroid/graphics/drawable/Drawable$ConstantState;
  
+     .line 1522
      .local v1, cs:Landroid/graphics/drawable/Drawable$ConstantState;
      if-eqz v1, :cond_0
  
+     .line 1523
      invoke-virtual {v1}, Landroid/graphics/drawable/Drawable$ConstantState;->getChangingConfigurations()I
  
      move-result v4
***************
*** 528,543 ****
--- 593,611 ----
  
      if-eqz v4, :cond_0
  
+     .line 1531
      const/4 v4, 0x0
  
      invoke-virtual {p1, v2, v4}, Landroid/util/LongSparseArray;->setValueAt(ILjava/lang/Object;)V
  
+     .line 1518
      .end local v1           #cs:Landroid/graphics/drawable/Drawable$ConstantState;
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 1542
      .end local v3           #ref:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Landroid/graphics/drawable/Drawable$ConstantState;>;"
      :cond_1
      return-void
***************
*** 556,561 ****
--- 624,630 ----
      .end annotation
  
      .prologue
+     .line 134
      sget-object v0, Landroid/content/res/Resources;->EMPTY_ARRAY:Landroid/util/LongSparseArray;
  
      return-object v0
***************
*** 566,575 ****
--- 635,646 ----
      .parameter "key"
  
      .prologue
+     .line 2085
      iget-object v3, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v3
  
+     .line 2086
      :try_start_0
      iget-object v2, p0, Landroid/content/res/Resources;->mColorStateListCache:Landroid/util/SparseArray;
  
***************
*** 579,616 ****
--- 650,696 ----
  
      check-cast v1, Ljava/lang/ref/WeakReference;
  
+     .line 2087
      .local v1, wr:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Landroid/content/res/ColorStateList;>;"
      if-eqz v1, :cond_1
  
+     .line 2088
      invoke-virtual {v1}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Landroid/content/res/ColorStateList;
  
+     .line 2089
      .local v0, entry:Landroid/content/res/ColorStateList;
      if-eqz v0, :cond_0
  
+     .line 2093
      monitor-exit v3
  
+     .line 2100
      .end local v0           #entry:Landroid/content/res/ColorStateList;
      :goto_0
      return-object v0
  
+     .line 2096
      .restart local v0       #entry:Landroid/content/res/ColorStateList;
      :cond_0
      iget-object v2, p0, Landroid/content/res/Resources;->mColorStateListCache:Landroid/util/SparseArray;
  
      invoke-virtual {v2, p1}, Landroid/util/SparseArray;->delete(I)V
  
+     .line 2099
      .end local v0           #entry:Landroid/content/res/ColorStateList;
      :cond_1
      monitor-exit v3
  
+     .line 2100
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 2099
      .end local v1           #wr:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Landroid/content/res/ColorStateList;>;"
      :catchall_0
      move-exception v2
***************
*** 640,650 ****
--- 720,732 ----
      .end annotation
  
      .prologue
+     .line 1983
      .local p1, drawableCache:Landroid/util/LongSparseArray;,"Landroid/util/LongSparseArray<Ljava/lang/ref/WeakReference<Landroid/graphics/drawable/Drawable$ConstantState;>;>;"
      iget-object v3, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v3
  
+     .line 1984
      :try_start_0
      invoke-virtual {p1, p2, p3}, Landroid/util/LongSparseArray;->get(J)Ljava/lang/Object;
  
***************
*** 652,691 ****
--- 734,782 ----
  
      check-cast v1, Ljava/lang/ref/WeakReference;
  
+     .line 1985
      .local v1, wr:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Landroid/graphics/drawable/Drawable$ConstantState;>;"
      if-eqz v1, :cond_1
  
+     .line 1986
      invoke-virtual {v1}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Landroid/graphics/drawable/Drawable$ConstantState;
  
+     .line 1987
      .local v0, entry:Landroid/graphics/drawable/Drawable$ConstantState;
      if-eqz v0, :cond_0
  
+     .line 1991
      invoke-virtual {v0, p0}, Landroid/graphics/drawable/Drawable$ConstantState;->newDrawable(Landroid/content/res/Resources;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
      monitor-exit v3
  
+     .line 1998
      .end local v0           #entry:Landroid/graphics/drawable/Drawable$ConstantState;
      :goto_0
      return-object v2
  
+     .line 1994
      .restart local v0       #entry:Landroid/graphics/drawable/Drawable$ConstantState;
      :cond_0
      invoke-virtual {p1, p2, p3}, Landroid/util/LongSparseArray;->delete(J)V
  
+     .line 1997
      .end local v0           #entry:Landroid/graphics/drawable/Drawable$ConstantState;
      :cond_1
      monitor-exit v3
  
+     .line 1998
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 1997
      .end local v1           #wr:Ljava/lang/ref/WeakReference;,"Ljava/lang/ref/WeakReference<Landroid/graphics/drawable/Drawable$ConstantState;>;"
      :catchall_0
      move-exception v2
***************
*** 702,725 ****
--- 793,823 ----
      .parameter "len"
  
      .prologue
+     .line 2168
      iget-object v3, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v3
  
+     .line 2169
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mCachedStyledAttributes:Landroid/content/res/TypedArray;
  
+     .line 2170
      .local v0, attrs:Landroid/content/res/TypedArray;
      if-eqz v0, :cond_1
  
+     .line 2171
      const/4 v2, 0x0
  
      iput-object v2, p0, Landroid/content/res/Resources;->mCachedStyledAttributes:Landroid/content/res/TypedArray;
  
+     .line 2177
      iput p1, v0, Landroid/content/res/TypedArray;->mLength:I
  
+     .line 2178
      mul-int/lit8 v1, p1, 0x6
  
+     .line 2179
      .local v1, fullLen:I
      iget-object v2, v0, Landroid/content/res/TypedArray;->mData:[I
  
***************
*** 727,739 ****
--- 825,840 ----
  
      if-lt v2, v1, :cond_0
  
+     .line 2180
      monitor-exit v3
  
+     .line 2195
      .end local v0           #attrs:Landroid/content/res/TypedArray;
      .end local v1           #fullLen:I
      :goto_0
      return-object v0
  
+     .line 2182
      .restart local v0       #attrs:Landroid/content/res/TypedArray;
      .restart local v1       #fullLen:I
      :cond_0
***************
*** 741,756 ****
--- 842,860 ----
  
      iput-object v2, v0, Landroid/content/res/TypedArray;->mData:[I
  
+     .line 2183
      add-int/lit8 v2, p1, 0x1
  
      new-array v2, v2, [I
  
      iput-object v2, v0, Landroid/content/res/TypedArray;->mIndices:[I
  
+     .line 2184
      monitor-exit v3
  
      goto :goto_0
  
+     .line 2198
      .end local v0           #attrs:Landroid/content/res/TypedArray;
      .end local v1           #fullLen:I
      :catchall_0
***************
*** 762,767 ****
--- 866,872 ----
  
      throw v2
  
+     .line 2195
      .restart local v0       #attrs:Landroid/content/res/TypedArray;
      :cond_1
      :try_start_1
***************
*** 789,803 ****
--- 894,911 ----
      .locals 2
  
      .prologue
+     .line 286
      sget-object v1, Landroid/content/res/Resources;->mSync:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 287
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mPluralRule:Llibcore/icu/NativePluralRules;
  
      if-nez v0, :cond_0
  
+     .line 288
      iget-object v0, p0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
  
      iget-object v0, v0, Landroid/content/res/Configuration;->locale:Ljava/util/Locale;
***************
*** 808,813 ****
--- 916,922 ----
  
      iput-object v0, p0, Landroid/content/res/Resources;->mPluralRule:Llibcore/icu/NativePluralRules;
  
+     .line 290
      :cond_0
      iget-object v0, p0, Landroid/content/res/Resources;->mPluralRule:Llibcore/icu/NativePluralRules;
  
***************
*** 815,820 ****
--- 924,930 ----
  
      return-object v0
  
+     .line 291
      :catchall_0
      move-exception v0
  
***************
*** 829,857 ****
--- 939,974 ----
      .locals 3
  
      .prologue
+     .line 216
      sget-object v2, Landroid/content/res/Resources;->mSync:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 217
      :try_start_0
      sget-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
+     .line 218
      .local v0, ret:Landroid/content/res/Resources;
      if-nez v0, :cond_0
  
+     .line 219
      new-instance v0, Landroid/content/res/Resources;
  
      .end local v0           #ret:Landroid/content/res/Resources;
      invoke-direct {v0}, Landroid/content/res/Resources;-><init>()V
  
+     .line 220
      .restart local v0       #ret:Landroid/content/res/Resources;
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
+     .line 223
      :cond_0
      monitor-exit v2
  
      return-object v0
  
+     .line 224
      :catchall_0
      move-exception v1
  
***************
*** 868,873 ****
--- 985,991 ----
      .parameter "targetSdkVersion"
  
      .prologue
+     .line 139
      const v0, 0x1030005
  
      const v1, 0x103006b
***************
*** 890,901 ****
--- 1008,1022 ----
      .parameter "deviceDefault"
  
      .prologue
+     .line 148
      if-eqz p0, :cond_0
  
+     .line 157
      .end local p0
      :goto_0
      return p0
  
+     .line 151
      .restart local p0
      :cond_0
      const/16 v0, 0xb
***************
*** 904,911 ****
--- 1025,1034 ----
  
      move p0, p2
  
+     .line 152
      goto :goto_0
  
+     .line 154
      :cond_1
      const/16 v0, 0xe
  
***************
*** 913,923 ****
--- 1036,1048 ----
  
      move p0, p3
  
+     .line 155
      goto :goto_0
  
      :cond_2
      move p0, p4
  
+     .line 157
      goto :goto_0
  .end method
  
***************
*** 926,963 ****
      .parameter "quantityCode"
  
      .prologue
      packed-switch p0, :pswitch_data_0
  
!     const-string v0, "other"
  
      :goto_0
      return-object v0
  
      :pswitch_0
!     const-string v0, "zero"
  
      goto :goto_0
  
      :pswitch_1
!     const-string v0, "one"
  
      goto :goto_0
  
      :pswitch_2
!     const-string v0, "two"
  
      goto :goto_0
  
      :pswitch_3
      const-string v0, "few"
  
      goto :goto_0
  
      :pswitch_4
!     const-string v0, "many"
  
      goto :goto_0
  
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
--- 1051,1096 ----
      .parameter "quantityCode"
  
      .prologue
+     .line 306
      packed-switch p0, :pswitch_data_0
  
!     .line 312
!     const-string/jumbo v0, "other"
  
      :goto_0
      return-object v0
  
+     .line 307
      :pswitch_0
!     const-string/jumbo v0, "zero"
  
      goto :goto_0
  
+     .line 308
      :pswitch_1
!     const-string/jumbo v0, "one"
  
      goto :goto_0
  
+     .line 309
      :pswitch_2
!     const-string/jumbo v0, "two"
  
      goto :goto_0
  
+     .line 310
      :pswitch_3
      const-string v0, "few"
  
      goto :goto_0
  
+     .line 311
      :pswitch_4
!     const-string/jumbo v0, "many"
  
      goto :goto_0
  
+     .line 306
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
***************
*** 974,983 ****
--- 1107,1118 ----
      .parameter "metrics"
  
      .prologue
+     .line 1563
      const/4 v0, 0x0
  
      invoke-static {p0, p1, v0}, Landroid/content/res/Resources;->updateSystemConfiguration(Landroid/content/res/Configuration;Landroid/util/DisplayMetrics;Landroid/content/res/CompatibilityInfo;)V
  
+     .line 1564
      return-void
  .end method
  
***************
*** 988,1001 ****
--- 1123,1139 ----
      .parameter "compat"
  
      .prologue
+     .line 1552
      sget-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
      if-eqz v0, :cond_0
  
+     .line 1553
      sget-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
      invoke-virtual {v0, p0, p1, p2}, Landroid/content/res/Resources;->updateConfiguration(Landroid/content/res/Configuration;Landroid/util/DisplayMetrics;Landroid/content/res/CompatibilityInfo;)V
  
+     .line 1557
      :cond_0
      return-void
  .end method
***************
*** 1006,1021 ****
--- 1144,1163 ----
      .locals 1
  
      .prologue
+     .line 1859
      iget-boolean v0, p0, Landroid/content/res/Resources;->mPreloading:Z
  
      if-eqz v0, :cond_0
  
+     .line 1860
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/content/res/Resources;->mPreloading:Z
  
+     .line 1861
      invoke-virtual {p0}, Landroid/content/res/Resources;->flushLayoutCache()V
  
+     .line 1863
      :cond_0
      return-void
  .end method
***************
*** 1024,1038 ****
--- 1166,1183 ----
      .locals 6
  
      .prologue
+     .line 1825
      iget-object v4, p0, Landroid/content/res/Resources;->mCachedXmlBlockIds:[I
  
      monitor-enter v4
  
+     .line 1827
      :try_start_0
      iget-object v3, p0, Landroid/content/res/Resources;->mCachedXmlBlockIds:[I
  
      array-length v1, v3
  
+     .line 1828
      .local v1, num:I
      const/4 v0, 0x0
  
***************
*** 1040,1060 ****
--- 1185,1210 ----
      :goto_0
      if-ge v0, v1, :cond_1
  
+     .line 1829
      iget-object v3, p0, Landroid/content/res/Resources;->mCachedXmlBlockIds:[I
  
      const/4 v5, 0x0
  
      aput v5, v3, v0
  
+     .line 1830
      iget-object v3, p0, Landroid/content/res/Resources;->mCachedXmlBlocks:[Landroid/content/res/XmlBlock;
  
      aget-object v2, v3, v0
  
+     .line 1831
      .local v2, oldBlock:Landroid/content/res/XmlBlock;
      if-eqz v2, :cond_0
  
+     .line 1832
      invoke-virtual {v2}, Landroid/content/res/XmlBlock;->close()V
  
+     .line 1834
      :cond_0
      iget-object v3, p0, Landroid/content/res/Resources;->mCachedXmlBlocks:[Landroid/content/res/XmlBlock;
  
***************
*** 1062,1077 ****
--- 1212,1231 ----
  
      aput-object v5, v3, v0
  
+     .line 1828
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 1836
      .end local v2           #oldBlock:Landroid/content/res/XmlBlock;
      :cond_1
      monitor-exit v4
  
+     .line 1837
      return-void
  
+     .line 1836
      .end local v0           #i:I
      .end local v1           #num:I
      :catchall_0
***************
*** 1094,1099 ****
--- 1248,1254 ----
      .end annotation
  
      .prologue
+     .line 882
      const-string v0, "anim"
  
      invoke-virtual {p0, p1, v0}, Landroid/content/res/Resources;->loadXmlResourceParser(ILjava/lang/String;)Landroid/content/res/XmlResourceParser;
***************
*** 1107,1112 ****
--- 1262,1268 ----
      .locals 1
  
      .prologue
+     .line 1816
      iget-object v0, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      return-object v0
***************
*** 1124,1141 ****
--- 1280,1301 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 799
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 800
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 801
      .local v0, value:Landroid/util/TypedValue;
      const/4 v3, 0x1
  
      invoke-virtual {p0, p1, v0, v3}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 802
      iget v3, v0, Landroid/util/TypedValue;->type:I
  
      const/16 v4, 0x10
***************
*** 1148,1153 ****
--- 1308,1314 ----
  
      if-gt v3, v4, :cond_1
  
+     .line 804
      iget v3, v0, Landroid/util/TypedValue;->data:I
  
      if-eqz v3, :cond_0
***************
*** 1162,1167 ****
--- 1323,1329 ----
  
      goto :goto_0
  
+     .line 806
      :cond_1
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 1213,1218 ****
--- 1375,1381 ----
  
      throw v1
  
+     .line 809
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 1234,1251 ****
--- 1397,1418 ----
      .end annotation
  
      .prologue
+     .line 747
      iget-object v3, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v3
  
+     .line 748
      :try_start_0
      iget-object v1, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 749
      .local v1, value:Landroid/util/TypedValue;
      const/4 v2, 0x1
  
      invoke-virtual {p0, p1, v1, v2}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 750
      iget v2, v1, Landroid/util/TypedValue;->type:I
  
      const/16 v4, 0x10
***************
*** 1258,1270 ****
--- 1425,1440 ----
  
      if-gt v2, v4, :cond_0
  
+     .line 752
      iget v2, v1, Landroid/util/TypedValue;->data:I
  
      monitor-exit v3
  
+     .line 755
      :goto_0
      return v2
  
+     .line 753
      :cond_0
      iget v2, v1, Landroid/util/TypedValue;->type:I
  
***************
*** 1272,1283 ****
--- 1442,1455 ----
  
      if-ne v2, v4, :cond_1
  
+     .line 754
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      invoke-virtual {p0, v2, p1}, Landroid/content/res/Resources;->loadColorStateList(Landroid/util/TypedValue;I)Landroid/content/res/ColorStateList;
  
      move-result-object v0
  
+     .line 755
      .local v0, csl:Landroid/content/res/ColorStateList;
      invoke-virtual {v0}, Landroid/content/res/ColorStateList;->getDefaultColor()I
  
***************
*** 1287,1292 ****
--- 1459,1465 ----
  
      goto :goto_0
  
+     .line 760
      .end local v0           #csl:Landroid/content/res/ColorStateList;
      .end local v1           #value:Landroid/util/TypedValue;
      :catchall_0
***************
*** 1298,1303 ****
--- 1471,1477 ----
  
      throw v2
  
+     .line 757
      .restart local v1       #value:Landroid/util/TypedValue;
      :cond_1
      :try_start_1
***************
*** 1364,1381 ****
--- 1538,1559 ----
      .end annotation
  
      .prologue
+     .line 778
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 779
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 780
      .local v0, value:Landroid/util/TypedValue;
      const/4 v1, 0x1
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 781
      invoke-virtual {p0, v0, p1}, Landroid/content/res/Resources;->loadColorStateList(Landroid/util/TypedValue;I)Landroid/content/res/ColorStateList;
  
      move-result-object v1
***************
*** 1384,1389 ****
--- 1562,1568 ----
  
      return-object v1
  
+     .line 782
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 1399,1404 ****
--- 1578,1584 ----
      .locals 1
  
      .prologue
+     .line 1596
      iget-object v0, p0, Landroid/content/res/Resources;->mCompatibilityInfo:Landroid/content/res/CompatibilityInfo;
  
      if-eqz v0, :cond_0
***************
*** 1418,1423 ****
--- 1598,1604 ----
      .locals 1
  
      .prologue
+     .line 1585
      iget-object v0, p0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
  
      return-object v0
***************
*** 1433,1456 ****
--- 1614,1642 ----
      .end annotation
  
      .prologue
+     .line 537
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 538
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 539
      .local v0, value:Landroid/util/TypedValue;
      const/4 v1, 0x1
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 540
      iget v1, v0, Landroid/util/TypedValue;->type:I
  
      const/4 v3, 0x5
  
      if-ne v1, v3, :cond_0
  
+     .line 541
      iget v1, v0, Landroid/util/TypedValue;->data:I
  
      iget-object v3, p0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
***************
*** 1463,1468 ****
--- 1649,1655 ----
  
      return v1
  
+     .line 543
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 1514,1519 ****
--- 1701,1707 ----
  
      throw v1
  
+     .line 546
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 1535,1558 ****
--- 1723,1751 ----
      .end annotation
  
      .prologue
+     .line 569
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 570
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 571
      .local v0, value:Landroid/util/TypedValue;
      const/4 v1, 0x1
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 572
      iget v1, v0, Landroid/util/TypedValue;->type:I
  
      const/4 v3, 0x5
  
      if-ne v1, v3, :cond_0
  
+     .line 573
      iget v1, v0, Landroid/util/TypedValue;->data:I
  
      iget-object v3, p0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
***************
*** 1565,1570 ****
--- 1758,1764 ----
  
      return v1
  
+     .line 576
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 1616,1621 ****
--- 1810,1816 ----
  
      throw v1
  
+     .line 579
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 1637,1660 ****
--- 1832,1860 ----
      .end annotation
  
      .prologue
+     .line 603
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 604
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 605
      .local v0, value:Landroid/util/TypedValue;
      const/4 v1, 0x1
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 606
      iget v1, v0, Landroid/util/TypedValue;->type:I
  
      const/4 v3, 0x5
  
      if-ne v1, v3, :cond_0
  
+     .line 607
      iget v1, v0, Landroid/util/TypedValue;->data:I
  
      iget-object v3, p0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
***************
*** 1667,1672 ****
--- 1867,1873 ----
  
      return v1
  
+     .line 610
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 1718,1723 ****
--- 1919,1925 ----
  
      throw v1
  
+     .line 613
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 1733,1738 ****
--- 1935,1941 ----
      .locals 1
  
      .prologue
+     .line 1575
      iget-object v0, p0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
  
      return-object v0
***************
*** 1748,1765 ****
--- 1951,1972 ----
      .end annotation
  
      .prologue
+     .line 661
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 662
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 663
      .local v0, value:Landroid/util/TypedValue;
      const/4 v1, 0x1
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 664
      invoke-virtual {p0, v0, p1}, Landroid/content/res/Resources;->loadDrawable(Landroid/util/TypedValue;I)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
***************
*** 1768,1773 ****
--- 1975,1981 ----
  
      return-object v1
  
+     .line 665
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 1790,1807 ****
--- 1998,2019 ----
      .end annotation
  
      .prologue
+     .line 689
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 690
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 691
      .local v0, value:Landroid/util/TypedValue;
      const/4 v1, 0x1
  
      invoke-virtual {p0, p1, p2, v0, v1}, Landroid/content/res/Resources;->getValueForDensity(IILandroid/util/TypedValue;Z)V
  
+     .line 700
      iget v1, v0, Landroid/util/TypedValue;->density:I
  
      if-lez v1, :cond_0
***************
*** 1812,1825 ****
--- 2024,2040 ----
  
      if-eq v1, v3, :cond_0
  
+     .line 701
      iget v1, v0, Landroid/util/TypedValue;->density:I
  
      if-ne v1, p2, :cond_1
  
+     .line 702
      sget v1, Landroid/util/DisplayMetrics;->DENSITY_DEVICE:I
  
      iput v1, v0, Landroid/util/TypedValue;->density:I
  
+     .line 708
      :cond_0
      :goto_0
      invoke-virtual {p0, v0, p1}, Landroid/content/res/Resources;->loadDrawable(Landroid/util/TypedValue;I)Landroid/graphics/drawable/Drawable;
***************
*** 1830,1835 ****
--- 2045,2051 ----
  
      return-object v1
  
+     .line 704
      :cond_1
      iget v1, v0, Landroid/util/TypedValue;->density:I
  
***************
*** 1843,1848 ****
--- 2059,2065 ----
  
      goto :goto_0
  
+     .line 709
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 1861,1884 ****
--- 2078,2106 ----
      .parameter "pbase"
  
      .prologue
+     .line 634
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 635
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 636
      .local v0, value:Landroid/util/TypedValue;
      const/4 v1, 0x1
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 637
      iget v1, v0, Landroid/util/TypedValue;->type:I
  
      const/4 v3, 0x6
  
      if-ne v1, v3, :cond_0
  
+     .line 638
      iget v1, v0, Landroid/util/TypedValue;->data:I
  
      int-to-float v3, p2
***************
*** 1893,1898 ****
--- 2115,2121 ----
  
      return v1
  
+     .line 640
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 1944,1949 ****
--- 2167,2173 ----
  
      throw v1
  
+     .line 643
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 1962,1967 ****
--- 2186,2192 ----
      .parameter "defPackage"
  
      .prologue
+     .line 1631
      :try_start_0
      invoke-static {p1}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
      :try_end_0
***************
*** 1969,1980 ****
--- 2194,2208 ----
  
      move-result v0
  
+     .line 1635
      :goto_0
      return v0
  
+     .line 1632
      :catch_0
      move-exception v0
  
+     .line 1635
      iget-object v0, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v0, p1, p2, p3}, Landroid/content/res/AssetManager;->getResourceIdentifier(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)I
***************
*** 1994,2010 ****
--- 2222,2242 ----
      .end annotation
  
      .prologue
+     .line 484
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v1, p1}, Landroid/content/res/AssetManager;->getArrayIntResource(I)[I
  
      move-result-object v0
  
+     .line 485
      .local v0, res:[I
      if-eqz v0, :cond_0
  
+     .line 486
      return-object v0
  
+     .line 488
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2045,2062 ****
--- 2277,2298 ----
      .end annotation
  
      .prologue
+     .line 824
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 825
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 826
      .local v0, value:Landroid/util/TypedValue;
      const/4 v1, 0x1
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 827
      iget v1, v0, Landroid/util/TypedValue;->type:I
  
      const/16 v3, 0x10
***************
*** 2069,2080 ****
--- 2305,2318 ----
  
      if-gt v1, v3, :cond_0
  
+     .line 829
      iget v1, v0, Landroid/util/TypedValue;->data:I
  
      monitor-exit v2
  
      return v1
  
+     .line 831
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2126,2131 ****
--- 2364,2370 ----
  
      throw v1
  
+     .line 834
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 2147,2152 ****
--- 2386,2392 ----
      .end annotation
  
      .prologue
+     .line 858
      const-string v0, "layout"
  
      invoke-virtual {p0, p1, v0}, Landroid/content/res/Resources;->loadXmlResourceParser(ILjava/lang/String;)Landroid/content/res/XmlResourceParser;
***************
*** 2166,2189 ****
--- 2406,2434 ----
      .end annotation
  
      .prologue
+     .line 721
      invoke-virtual {p0, p1}, Landroid/content/res/Resources;->openRawResource(I)Ljava/io/InputStream;
  
      move-result-object v0
  
+     .line 722
      .local v0, is:Ljava/io/InputStream;
      invoke-static {v0}, Landroid/graphics/Movie;->decodeStream(Ljava/io/InputStream;)Landroid/graphics/Movie;
  
      move-result-object v1
  
+     .line 724
      .local v1, movie:Landroid/graphics/Movie;
      :try_start_0
      invoke-virtual {v0}, Ljava/io/InputStream;->close()V
      :try_end_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 729
      :goto_0
      return-object v1
  
+     .line 726
      :catch_0
      move-exception v2
  
***************
*** 2201,2206 ****
--- 2446,2452 ----
      .end annotation
  
      .prologue
+     .line 410
      invoke-virtual {p0, p1, p2}, Landroid/content/res/Resources;->getQuantityText(II)Ljava/lang/CharSequence;
  
      move-result-object v0
***************
*** 2224,2229 ****
--- 2470,2476 ----
      .end annotation
  
      .prologue
+     .line 387
      invoke-virtual {p0, p1, p2}, Landroid/content/res/Resources;->getQuantityText(II)Ljava/lang/CharSequence;
  
      move-result-object v1
***************
*** 2232,2237 ****
--- 2479,2485 ----
  
      move-result-object v0
  
+     .line 388
      .local v0, raw:Ljava/lang/String;
      iget-object v1, p0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
  
***************
*** 2255,2264 ****
--- 2503,2514 ----
      .end annotation
  
      .prologue
+     .line 270
      invoke-direct {p0}, Landroid/content/res/Resources;->getPluralRule()Llibcore/icu/NativePluralRules;
  
      move-result-object v2
  
+     .line 271
      .local v2, rule:Llibcore/icu/NativePluralRules;
      iget-object v3, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
***************
*** 2274,2289 ****
--- 2524,2542 ----
  
      move-result-object v0
  
+     .line 273
      .local v0, res:Ljava/lang/CharSequence;
      if-eqz v0, :cond_0
  
      move-object v1, v0
  
+     .line 278
      .end local v0           #res:Ljava/lang/CharSequence;
      .local v1, res:Ljava/lang/CharSequence;
      :goto_0
      return-object v1
  
+     .line 276
      .end local v1           #res:Ljava/lang/CharSequence;
      .restart local v0       #res:Ljava/lang/CharSequence;
      :cond_0
***************
*** 2295,2308 ****
--- 2548,2564 ----
  
      move-result-object v0
  
+     .line 277
      if-eqz v0, :cond_1
  
      move-object v1, v0
  
+     .line 278
      .end local v0           #res:Ljava/lang/CharSequence;
      .restart local v1       #res:Ljava/lang/CharSequence;
      goto :goto_0
  
+     .line 280
      .end local v1           #res:Ljava/lang/CharSequence;
      .restart local v0       #res:Ljava/lang/CharSequence;
      :cond_1
***************
*** 2373,2389 ****
--- 2629,2648 ----
      .end annotation
  
      .prologue
+     .line 1710
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v1, p1}, Landroid/content/res/AssetManager;->getResourceEntryName(I)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 1711
      .local v0, str:Ljava/lang/String;
      if-eqz v0, :cond_0
  
      return-object v0
  
+     .line 1712
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2424,2440 ****
--- 2683,2702 ----
      .end annotation
  
      .prologue
+     .line 1653
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v1, p1}, Landroid/content/res/AssetManager;->getResourceName(I)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 1654
      .local v0, str:Ljava/lang/String;
      if-eqz v0, :cond_0
  
      return-object v0
  
+     .line 1655
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2475,2491 ****
--- 2737,2756 ----
      .end annotation
  
      .prologue
+     .line 1672
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v1, p1}, Landroid/content/res/AssetManager;->getResourcePackageName(I)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 1673
      .local v0, str:Ljava/lang/String;
      if-eqz v0, :cond_0
  
      return-object v0
  
+     .line 1674
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2526,2542 ****
--- 2791,2810 ----
      .end annotation
  
      .prologue
+     .line 1691
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v1, p1}, Landroid/content/res/AssetManager;->getResourceTypeName(I)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 1692
      .local v0, str:Ljava/lang/String;
      if-eqz v0, :cond_0
  
      return-object v0
  
+     .line 1693
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2577,2595 ****
--- 2845,2867 ----
      .end annotation
  
      .prologue
+     .line 331
      invoke-virtual {p0, p1}, Landroid/content/res/Resources;->getText(I)Ljava/lang/CharSequence;
  
      move-result-object v0
  
+     .line 332
      .local v0, res:Ljava/lang/CharSequence;
      if-eqz v0, :cond_0
  
+     .line 333
      invoke-virtual {v0}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
      move-result-object v1
  
      return-object v1
  
+     .line 335
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2631,2640 ****
--- 2903,2914 ----
      .end annotation
  
      .prologue
+     .line 359
      invoke-virtual {p0, p1}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 360
      .local v0, raw:Ljava/lang/String;
      iget-object v1, p0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
  
***************
*** 2657,2673 ****
--- 2931,2951 ----
      .end annotation
  
      .prologue
+     .line 464
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v1, p1}, Landroid/content/res/AssetManager;->getResourceStringArray(I)[Ljava/lang/String;
  
      move-result-object v0
  
+     .line 465
      .local v0, res:[Ljava/lang/String;
      if-eqz v0, :cond_0
  
+     .line 466
      return-object v0
  
+     .line 468
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2708,2724 ****
--- 2986,3006 ----
      .end annotation
  
      .prologue
+     .line 243
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v1, p1}, Landroid/content/res/AssetManager;->getResourceText(I)Ljava/lang/CharSequence;
  
      move-result-object v0
  
+     .line 244
      .local v0, res:Ljava/lang/CharSequence;
      if-eqz v0, :cond_0
  
+     .line 245
      return-object v0
  
+     .line 247
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2755,2760 ****
--- 3037,3043 ----
      .parameter "def"
  
      .prologue
+     .line 428
      if-eqz p1, :cond_0
  
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
***************
*** 2763,2768 ****
--- 3046,3052 ----
  
      move-result-object v0
  
+     .line 429
      .local v0, res:Ljava/lang/CharSequence;
      :goto_0
      if-eqz v0, :cond_1
***************
*** 2771,2776 ****
--- 3055,3061 ----
      :goto_1
      return-object v0
  
+     .line 428
      :cond_0
      const/4 v0, 0x0
  
***************
*** 2780,2785 ****
--- 3065,3071 ----
      :cond_1
      move-object v0, p2
  
+     .line 429
      goto :goto_1
  .end method
  
***************
*** 2793,2809 ****
--- 3079,3099 ----
      .end annotation
  
      .prologue
+     .line 444
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v1, p1}, Landroid/content/res/AssetManager;->getResourceTextArray(I)[Ljava/lang/CharSequence;
  
      move-result-object v0
  
+     .line 445
      .local v0, res:[Ljava/lang/CharSequence;
      if-eqz v0, :cond_0
  
+     .line 446
      return-object v0
  
+     .line 448
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2846,2851 ****
--- 3136,3142 ----
      .end annotation
  
      .prologue
+     .line 1015
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      const/4 v2, 0x0
***************
*** 2854,2864 ****
--- 3145,3158 ----
  
      move-result v0
  
+     .line 1016
      .local v0, found:Z
      if-eqz v0, :cond_0
  
+     .line 1017
      return-void
  
+     .line 1019
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2901,2907 ****
      .end annotation
  
      .prologue
!     const-string v1, "string"
  
      const/4 v2, 0x0
  
--- 3195,3202 ----
      .end annotation
  
      .prologue
!     .line 1068
!     const-string/jumbo v1, "string"
  
      const/4 v2, 0x0
  
***************
*** 2909,2921 ****
--- 3204,3220 ----
  
      move-result v0
  
+     .line 1069
      .local v0, id:I
      if-eqz v0, :cond_0
  
+     .line 1070
      invoke-virtual {p0, v0, p2, p3}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 1071
      return-void
  
+     .line 1073
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 2955,2971 ****
--- 3254,3274 ----
      .end annotation
  
      .prologue
+     .line 1039
      iget-object v1, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v1, p1, p2, p3, p4}, Landroid/content/res/AssetManager;->getResourceValue(IILandroid/util/TypedValue;Z)Z
  
      move-result v0
  
+     .line 1040
      .local v0, found:Z
      if-eqz v0, :cond_0
  
+     .line 1041
      return-void
  
+     .line 1043
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 3006,3012 ****
      .end annotation
  
      .prologue
!     const-string v0, "xml"
  
      invoke-virtual {p0, p1, v0}, Landroid/content/res/Resources;->loadXmlResourceParser(ILjava/lang/String;)Landroid/content/res/XmlResourceParser;
  
--- 3309,3316 ----
      .end annotation
  
      .prologue
!     .line 907
!     const-string/jumbo v0, "xml"
  
      invoke-virtual {p0, p1, v0}, Landroid/content/res/Resources;->loadXmlResourceParser(ILjava/lang/String;)Landroid/content/res/XmlResourceParser;
  
***************
*** 3026,3031 ****
--- 3330,3336 ----
      .end annotation
  
      .prologue
+     .line 2011
      iget v7, p1, Landroid/util/TypedValue;->assetCookie:I
  
      shl-int/lit8 v7, v7, 0x18
***************
*** 3034,3039 ****
--- 3339,3345 ----
  
      or-int v4, v7, v8
  
+     .line 2015
      .local v4, key:I
      iget v7, p1, Landroid/util/TypedValue;->type:I
  
***************
*** 3047,3052 ****
--- 3353,3359 ----
  
      if-gt v7, v8, :cond_2
  
+     .line 2018
      sget-object v7, Landroid/content/res/Resources;->mPreloadedColorStateLists:Landroid/util/SparseArray;
  
      invoke-virtual {v7, v4}, Landroid/util/SparseArray;->get(I)Ljava/lang/Object;
***************
*** 3055,3070 ****
--- 3362,3380 ----
  
      check-cast v0, Landroid/content/res/ColorStateList;
  
+     .line 2019
      .local v0, csl:Landroid/content/res/ColorStateList;
      if-eqz v0, :cond_0
  
      move-object v1, v0
  
+     .line 2081
      .end local v0           #csl:Landroid/content/res/ColorStateList;
      .local v1, csl:Landroid/content/res/ColorStateList;
      :goto_0
      return-object v1
  
+     .line 2023
      .end local v1           #csl:Landroid/content/res/ColorStateList;
      .restart local v0       #csl:Landroid/content/res/ColorStateList;
      :cond_0
***************
*** 3074,3083 ****
--- 3384,3395 ----
  
      move-result-object v0
  
+     .line 2024
      iget-boolean v7, p0, Landroid/content/res/Resources;->mPreloading:Z
  
      if-eqz v7, :cond_1
  
+     .line 2025
      sget-object v7, Landroid/content/res/Resources;->mPreloadedColorStateLists:Landroid/util/SparseArray;
  
      invoke-virtual {v7, v4, v0}, Landroid/util/SparseArray;->put(ILjava/lang/Object;)V
***************
*** 3085,3109 ****
--- 3397,3426 ----
      :cond_1
      move-object v1, v0
  
+     .line 2028
      .end local v0           #csl:Landroid/content/res/ColorStateList;
      .restart local v1       #csl:Landroid/content/res/ColorStateList;
      goto :goto_0
  
+     .line 2031
      .end local v1           #csl:Landroid/content/res/ColorStateList;
      :cond_2
      invoke-direct {p0, v4}, Landroid/content/res/Resources;->getCachedColorStateList(I)Landroid/content/res/ColorStateList;
  
      move-result-object v0
  
+     .line 2032
      .restart local v0       #csl:Landroid/content/res/ColorStateList;
      if-eqz v0, :cond_3
  
      move-object v1, v0
  
+     .line 2033
      .end local v0           #csl:Landroid/content/res/ColorStateList;
      .restart local v1       #csl:Landroid/content/res/ColorStateList;
      goto :goto_0
  
+     .line 2036
      .end local v1           #csl:Landroid/content/res/ColorStateList;
      .restart local v0       #csl:Landroid/content/res/ColorStateList;
      :cond_3
***************
*** 3116,3130 ****
--- 3433,3450 ----
      .end local v0           #csl:Landroid/content/res/ColorStateList;
      check-cast v0, Landroid/content/res/ColorStateList;
  
+     .line 2037
      .restart local v0       #csl:Landroid/content/res/ColorStateList;
      if-eqz v0, :cond_4
  
      move-object v1, v0
  
+     .line 2038
      .end local v0           #csl:Landroid/content/res/ColorStateList;
      .restart local v1       #csl:Landroid/content/res/ColorStateList;
      goto :goto_0
  
+     .line 2041
      .end local v1           #csl:Landroid/content/res/ColorStateList;
      .restart local v0       #csl:Landroid/content/res/ColorStateList;
      :cond_4
***************
*** 3132,3137 ****
--- 3452,3458 ----
  
      if-nez v7, :cond_5
  
+     .line 2042
      new-instance v7, Landroid/content/res/Resources$NotFoundException;
  
      new-instance v8, Ljava/lang/StringBuilder;
***************
*** 3156,3161 ****
--- 3477,3483 ----
  
      throw v7
  
+     .line 2046
      :cond_5
      iget-object v7, p1, Landroid/util/TypedValue;->string:Ljava/lang/CharSequence;
  
***************
*** 3163,3168 ****
--- 3485,3491 ----
  
      move-result-object v3
  
+     .line 2048
      .local v3, file:Ljava/lang/String;
      const-string v7, ".xml"
  
***************
*** 3172,3177 ****
--- 3495,3501 ----
  
      if-eqz v7, :cond_7
  
+     .line 2050
      :try_start_0
      iget v7, p1, Landroid/util/TypedValue;->assetCookie:I
  
***************
*** 3181,3201 ****
--- 3505,3530 ----
  
      move-result-object v6
  
+     .line 2052
      .local v6, rp:Landroid/content/res/XmlResourceParser;
      invoke-static {p0, v6}, Landroid/content/res/ColorStateList;->createFromXml(Landroid/content/res/Resources;Lorg/xmlpull/v1/XmlPullParser;)Landroid/content/res/ColorStateList;
  
      move-result-object v0
  
+     .line 2053
      invoke-interface {v6}, Landroid/content/res/XmlResourceParser;->close()V
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 2067
      if-eqz v0, :cond_6
  
+     .line 2068
      iget-boolean v7, p0, Landroid/content/res/Resources;->mPreloading:Z
  
      if-eqz v7, :cond_8
  
+     .line 2069
      sget-object v7, Landroid/content/res/Resources;->mPreloadedColorStateLists:Landroid/util/SparseArray;
  
      invoke-virtual {v7, v4, v0}, Landroid/util/SparseArray;->put(ILjava/lang/Object;)V
***************
*** 3204,3219 ****
--- 3533,3551 ----
      :goto_1
      move-object v1, v0
  
+     .line 2081
      .end local v0           #csl:Landroid/content/res/ColorStateList;
      .restart local v1       #csl:Landroid/content/res/ColorStateList;
      goto :goto_0
  
+     .line 2054
      .end local v1           #csl:Landroid/content/res/ColorStateList;
      .end local v6           #rp:Landroid/content/res/XmlResourceParser;
      .restart local v0       #csl:Landroid/content/res/ColorStateList;
      :catch_0
      move-exception v2
  
+     .line 2055
      .local v2, e:Ljava/lang/Exception;
      new-instance v5, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 3251,3261 ****
--- 3583,3596 ----
  
      invoke-direct {v5, v7}, Landroid/content/res/Resources$NotFoundException;-><init>(Ljava/lang/String;)V
  
+     .line 2058
      .local v5, rnf:Landroid/content/res/Resources$NotFoundException;
      invoke-virtual {v5, v2}, Landroid/content/res/Resources$NotFoundException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 2059
      throw v5
  
+     .line 2062
      .end local v2           #e:Ljava/lang/Exception;
      .end local v5           #rnf:Landroid/content/res/Resources$NotFoundException;
      :cond_7
***************
*** 3303,3314 ****
--- 3638,3651 ----
  
      throw v7
  
+     .line 2071
      .restart local v6       #rp:Landroid/content/res/XmlResourceParser;
      :cond_8
      iget-object v8, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v8
  
+     .line 2075
      :try_start_1
      iget-object v7, p0, Landroid/content/res/Resources;->mColorStateListCache:Landroid/util/SparseArray;
  
***************
*** 3318,3323 ****
--- 3655,3661 ----
  
      invoke-virtual {v7, v4, v9}, Landroid/util/SparseArray;->put(ILjava/lang/Object;)V
  
+     .line 2077
      monitor-exit v8
  
      goto :goto_1
***************
*** 3343,3348 ****
--- 3681,3687 ----
      .end annotation
  
      .prologue
+     .line 1876
      move-object/from16 v0, p1
  
      iget v13, v0, Landroid/util/TypedValue;->assetCookie:I
***************
*** 3361,3369 ****
--- 3700,3710 ----
  
      or-long v9, v13, v15
  
+     .line 1877
      .local v9, key:J
      const/4 v8, 0x0
  
+     .line 1878
      .local v8, isColorDrawable:Z
      move-object/from16 v0, p1
  
***************
*** 3381,3388 ****
--- 3722,3731 ----
  
      if-gt v13, v14, :cond_0
  
+     .line 1880
      const/4 v8, 0x1
  
+     .line 1882
      :cond_0
      if-eqz v8, :cond_1
  
***************
*** 3397,3412 ****
--- 3740,3758 ----
  
      move-result-object v3
  
+     .line 1884
      .local v3, dr:Landroid/graphics/drawable/Drawable;
      if-eqz v3, :cond_2
  
      move-object v4, v3
  
+     .line 1977
      .end local v3           #dr:Landroid/graphics/drawable/Drawable;
      .local v4, dr:Landroid/graphics/drawable/Drawable;
      :goto_1
      return-object v4
  
+     .line 1882
      .end local v4           #dr:Landroid/graphics/drawable/Drawable;
      :cond_1
      move-object/from16 v0, p0
***************
*** 3415,3420 ****
--- 3761,3767 ----
  
      goto :goto_0
  
+     .line 1888
      .restart local v3       #dr:Landroid/graphics/drawable/Drawable;
      :cond_2
      if-eqz v8, :cond_5
***************
*** 3429,3468 ****
--- 3776,3824 ----
  
      move-object v2, v13
  
+     .line 1889
      .local v2, cs:Landroid/graphics/drawable/Drawable$ConstantState;
      :goto_2
      if-eqz v2, :cond_6
  
+     .line 1890
      move-object/from16 v0, p0
  
      invoke-virtual {v2, v0}, Landroid/graphics/drawable/Drawable$ConstantState;->newDrawable(Landroid/content/res/Resources;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v3
  
+     .line 1952
      :cond_3
      :goto_3
      if-eqz v3, :cond_4
  
+     .line 1953
      move-object/from16 v0, p1
  
      iget v13, v0, Landroid/util/TypedValue;->changingConfigurations:I
  
      invoke-virtual {v3, v13}, Landroid/graphics/drawable/Drawable;->setChangingConfigurations(I)V
  
+     .line 1954
      invoke-virtual {v3}, Landroid/graphics/drawable/Drawable;->getConstantState()Landroid/graphics/drawable/Drawable$ConstantState;
  
      move-result-object v2
  
+     .line 1955
      if-eqz v2, :cond_4
  
+     .line 1956
      move-object/from16 v0, p0
  
      iget-boolean v13, v0, Landroid/content/res/Resources;->mPreloading:Z
  
      if-eqz v13, :cond_b
  
+     .line 1957
      if-eqz v8, :cond_a
  
+     .line 1958
      sget-object v13, Landroid/content/res/Resources;->sPreloadedColorDrawables:Landroid/util/LongSparseArray;
  
      invoke-virtual {v13, v9, v10, v2}, Landroid/util/LongSparseArray;->put(JLjava/lang/Object;)V
***************
*** 3471,3480 ****
--- 3827,3838 ----
      :goto_4
      move-object v4, v3
  
+     .line 1977
      .end local v3           #dr:Landroid/graphics/drawable/Drawable;
      .restart local v4       #dr:Landroid/graphics/drawable/Drawable;
      goto :goto_1
  
+     .line 1888
      .end local v2           #cs:Landroid/graphics/drawable/Drawable$ConstantState;
      .end local v4           #dr:Landroid/graphics/drawable/Drawable;
      .restart local v3       #dr:Landroid/graphics/drawable/Drawable;
***************
*** 3491,3496 ****
--- 3849,3855 ----
  
      goto :goto_2
  
+     .line 1892
      .restart local v2       #cs:Landroid/graphics/drawable/Drawable$ConstantState;
      :cond_6
      move-object/from16 v0, p1
***************
*** 3509,3514 ****
--- 3868,3874 ----
  
      if-gt v13, v14, :cond_7
  
+     .line 1894
      new-instance v3, Landroid/graphics/drawable/ColorDrawable;
  
      .end local v3           #dr:Landroid/graphics/drawable/Drawable;
***************
*** 3518,3533 ****
--- 3878,3896 ----
  
      invoke-direct {v3, v13}, Landroid/graphics/drawable/ColorDrawable;-><init>(I)V
  
+     .line 1897
      .restart local v3       #dr:Landroid/graphics/drawable/Drawable;
      :cond_7
      if-nez v3, :cond_3
  
+     .line 1898
      move-object/from16 v0, p1
  
      iget-object v13, v0, Landroid/util/TypedValue;->string:Ljava/lang/CharSequence;
  
      if-nez v13, :cond_8
  
+     .line 1899
      new-instance v13, Landroid/content/res/Resources$NotFoundException;
  
      new-instance v14, Ljava/lang/StringBuilder;
***************
*** 3554,3559 ****
--- 3917,3923 ----
  
      throw v13
  
+     .line 1903
      :cond_8
      move-object/from16 v0, p1
  
***************
*** 3563,3568 ****
--- 3927,3933 ----
  
      move-result-object v6
  
+     .line 1918
      .local v6, file:Ljava/lang/String;
      const-string v13, ".xml"
  
***************
*** 3572,3577 ****
--- 3937,3943 ----
  
      if-eqz v13, :cond_9
  
+     .line 1920
      :try_start_0
      move-object/from16 v0, p1
  
***************
*** 3587,3592 ****
--- 3953,3959 ----
  
      move-result-object v12
  
+     .line 1922
      .local v12, rp:Landroid/content/res/XmlResourceParser;
      move-object/from16 v0, p0
  
***************
*** 3594,3609 ****
--- 3961,3979 ----
  
      move-result-object v3
  
+     .line 1923
      invoke-interface {v12}, Landroid/content/res/XmlResourceParser;->close()V
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
      goto/16 :goto_3
  
+     .line 1924
      .end local v12           #rp:Landroid/content/res/XmlResourceParser;
      :catch_0
      move-exception v5
  
+     .line 1925
      .local v5, e:Ljava/lang/Exception;
      new-instance v11, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 3641,3651 ****
--- 4011,4024 ----
  
      invoke-direct {v11, v13}, Landroid/content/res/Resources$NotFoundException;-><init>(Ljava/lang/String;)V
  
+     .line 1928
      .local v11, rnf:Landroid/content/res/Resources$NotFoundException;
      invoke-virtual {v11, v5}, Landroid/content/res/Resources$NotFoundException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 1929
      throw v11
  
+     .line 1934
      .end local v5           #e:Ljava/lang/Exception;
      .end local v11           #rnf:Landroid/content/res/Resources$NotFoundException;
      :cond_9
***************
*** 3664,3669 ****
--- 4037,4043 ----
  
      move-result-object v7
  
+     .line 1937
      .local v7, is:Ljava/io/InputStream;
      const/4 v13, 0x0
  
***************
*** 3675,3690 ****
--- 4049,4067 ----
  
      move-result-object v3
  
+     .line 1939
      invoke-virtual {v7}, Ljava/io/InputStream;->close()V
      :try_end_1
      .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_1
  
      goto/16 :goto_3
  
+     .line 1941
      .end local v7           #is:Ljava/io/InputStream;
      :catch_1
      move-exception v5
  
+     .line 1942
      .restart local v5       #e:Ljava/lang/Exception;
      new-instance v11, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 3722,3732 ****
--- 4099,4112 ----
  
      invoke-direct {v11, v13}, Landroid/content/res/Resources$NotFoundException;-><init>(Ljava/lang/String;)V
  
+     .line 1945
      .restart local v11       #rnf:Landroid/content/res/Resources$NotFoundException;
      invoke-virtual {v11, v5}, Landroid/content/res/Resources$NotFoundException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 1946
      throw v11
  
+     .line 1960
      .end local v5           #e:Ljava/lang/Exception;
      .end local v6           #file:Ljava/lang/String;
      .end local v11           #rnf:Landroid/content/res/Resources$NotFoundException;
***************
*** 3737,3742 ****
--- 4117,4123 ----
  
      goto/16 :goto_4
  
+     .line 1963
      :cond_b
      move-object/from16 v0, p0
  
***************
*** 3744,3751 ****
--- 4125,4134 ----
  
      monitor-enter v14
  
+     .line 1967
      if-eqz v8, :cond_c
  
+     .line 1968
      :try_start_2
      move-object/from16 v0, p0
  
***************
*** 3757,3762 ****
--- 4140,4146 ----
  
      invoke-virtual {v13, v9, v10, v15}, Landroid/util/LongSparseArray;->put(JLjava/lang/Object;)V
  
+     .line 1972
      :goto_5
      monitor-exit v14
  
***************
*** 3771,3776 ****
--- 4155,4161 ----
  
      throw v13
  
+     .line 1970
      :cond_c
      :try_start_3
      move-object/from16 v0, p0
***************
*** 3799,3822 ****
--- 4184,4212 ----
      .end annotation
  
      .prologue
+     .line 2105
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
  
+     .line 2106
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 2107
      .local v0, value:Landroid/util/TypedValue;
      const/4 v1, 0x1
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 2108
      iget v1, v0, Landroid/util/TypedValue;->type:I
  
      const/4 v3, 0x3
  
      if-ne v1, v3, :cond_0
  
+     .line 2109
      iget-object v1, v0, Landroid/util/TypedValue;->string:Ljava/lang/CharSequence;
  
      invoke-virtual {v1}, Ljava/lang/Object;->toString()Ljava/lang/String;
***************
*** 3833,3838 ****
--- 4223,4229 ----
  
      return-object v1
  
+     .line 2112
      :cond_0
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 3884,3889 ****
--- 4275,4281 ----
  
      throw v1
  
+     .line 2115
      .end local v0           #value:Landroid/util/TypedValue;
      :catchall_0
      move-exception v1
***************
*** 3908,3915 ****
--- 4300,4309 ----
      .end annotation
  
      .prologue
+     .line 2120
      if-eqz p2, :cond_5
  
+     .line 2123
      :try_start_0
      iget-object v8, p0, Landroid/content/res/Resources;->mCachedXmlBlockIds:[I
  
***************
*** 3917,3927 ****
--- 4311,4323 ----
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 2125
      :try_start_1
      iget-object v7, p0, Landroid/content/res/Resources;->mCachedXmlBlockIds:[I
  
      array-length v3, v7
  
+     .line 2126
      .local v3, num:I
      const/4 v2, 0x0
  
***************
*** 3929,3940 ****
--- 4325,4338 ----
      :goto_0
      if-ge v2, v3, :cond_1
  
+     .line 2127
      iget-object v7, p0, Landroid/content/res/Resources;->mCachedXmlBlockIds:[I
  
      aget v7, v7, v2
  
      if-ne v7, p2, :cond_0
  
+     .line 2130
      iget-object v7, p0, Landroid/content/res/Resources;->mCachedXmlBlocks:[Landroid/content/res/XmlBlock;
  
      aget-object v7, v7, v2
***************
*** 3945,3958 ****
--- 4343,4359 ----
  
      monitor-exit v8
  
+     .line 2150
      :goto_1
      return-object v7
  
+     .line 2126
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 2136
      :cond_1
      iget-object v7, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
***************
*** 3960,3998 ****
--- 4361,4409 ----
  
      move-result-object v0
  
+     .line 2138
      .local v0, block:Landroid/content/res/XmlBlock;
      if-eqz v0, :cond_4
  
+     .line 2139
      iget v7, p0, Landroid/content/res/Resources;->mLastCachedXmlBlockIndex:I
  
      add-int/lit8 v5, v7, 0x1
  
+     .line 2140
      .local v5, pos:I
      if-lt v5, v3, :cond_2
  
      const/4 v5, 0x0
  
+     .line 2141
      :cond_2
      iput v5, p0, Landroid/content/res/Resources;->mLastCachedXmlBlockIndex:I
  
+     .line 2142
      iget-object v7, p0, Landroid/content/res/Resources;->mCachedXmlBlocks:[Landroid/content/res/XmlBlock;
  
      aget-object v4, v7, v5
  
+     .line 2143
      .local v4, oldBlock:Landroid/content/res/XmlBlock;
      if-eqz v4, :cond_3
  
+     .line 2144
      invoke-virtual {v4}, Landroid/content/res/XmlBlock;->close()V
  
+     .line 2146
      :cond_3
      iget-object v7, p0, Landroid/content/res/Resources;->mCachedXmlBlockIds:[I
  
      aput p2, v7, v5
  
+     .line 2147
      iget-object v7, p0, Landroid/content/res/Resources;->mCachedXmlBlocks:[Landroid/content/res/XmlBlock;
  
      aput-object v0, v7, v5
  
+     .line 2150
      invoke-virtual {v0}, Landroid/content/res/XmlBlock;->newParser()Landroid/content/res/XmlResourceParser;
  
      move-result-object v7
***************
*** 4001,4006 ****
--- 4412,4418 ----
  
      goto :goto_1
  
+     .line 2152
      .end local v0           #block:Landroid/content/res/XmlBlock;
      .end local v2           #i:I
      .end local v3           #num:I
***************
*** 4018,4026 ****
--- 4430,4440 ----
      :try_end_2
      .catch Ljava/lang/Exception; {:try_start_2 .. :try_end_2} :catch_0
  
+     .line 2153
      :catch_0
      move-exception v1
  
+     .line 2154
      .local v1, e:Ljava/lang/Exception;
      new-instance v6, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 4068,4078 ****
--- 4482,4495 ----
  
      invoke-direct {v6, v7}, Landroid/content/res/Resources$NotFoundException;-><init>(Ljava/lang/String;)V
  
+     .line 2157
      .local v6, rnf:Landroid/content/res/Resources$NotFoundException;
      invoke-virtual {v6, v1}, Landroid/content/res/Resources$NotFoundException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 2158
      throw v6
  
+     .line 2152
      .end local v1           #e:Ljava/lang/Exception;
      .end local v6           #rnf:Landroid/content/res/Resources$NotFoundException;
      .restart local v0       #block:Landroid/content/res/XmlBlock;
***************
*** 4084,4089 ****
--- 4501,4507 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
+     .line 2162
      .end local v0           #block:Landroid/content/res/XmlBlock;
      .end local v2           #i:I
      .end local v3           #num:I
***************
*** 4141,4146 ****
--- 4559,4565 ----
      .locals 1
  
      .prologue
+     .line 1370
      new-instance v0, Landroid/content/res/Resources$Theme;
  
      invoke-direct {v0, p0}, Landroid/content/res/Resources$Theme;-><init>(Landroid/content/res/Resources;)V
***************
*** 4154,4161 ****
--- 4573,4582 ----
      .parameter "attrs"
  
      .prologue
+     .line 1386
      array-length v1, p2
  
+     .line 1387
      .local v1, len:I
      invoke-direct {p0, v1}, Landroid/content/res/Resources;->getCachedStyledAttributes(I)Landroid/content/res/TypedArray;
  
***************
*** 4164,4171 ****
--- 4585,4594 ----
      .local v0, array:Landroid/content/res/TypedArray;
      move-object v2, p1
  
+     .line 1393
      check-cast v2, Landroid/content/res/XmlBlock$Parser;
  
+     .line 1394
      .local v2, parser:Landroid/content/res/XmlBlock$Parser;
      iget-object v3, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
***************
*** 4177,4186 ****
--- 4600,4612 ----
  
      invoke-virtual {v3, v4, p2, v5, v6}, Landroid/content/res/AssetManager;->retrieveAttributes(I[I[I[I)Z
  
+     .line 1397
      iput-object p2, v0, Landroid/content/res/TypedArray;->mRsrcs:[I
  
+     .line 1398
      iput-object v2, v0, Landroid/content/res/TypedArray;->mXml:Landroid/content/res/XmlBlock$Parser;
  
+     .line 1400
      return-object v0
  .end method
  
***************
*** 4196,4210 ****
--- 4622,4639 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 506
      iget-object v2, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
      invoke-virtual {v2, p1}, Landroid/content/res/AssetManager;->getArraySize(I)I
  
      move-result v1
  
+     .line 507
      .local v1, len:I
      if-gez v1, :cond_0
  
+     .line 508
      new-instance v2, Landroid/content/res/Resources$NotFoundException;
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 4233,4243 ****
--- 4662,4674 ----
  
      throw v2
  
+     .line 512
      :cond_0
      invoke-direct {p0, v1}, Landroid/content/res/Resources;->getCachedStyledAttributes(I)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 513
      .local v0, array:Landroid/content/res/TypedArray;
      iget-object v2, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
***************
*** 4249,4258 ****
--- 4680,4691 ----
  
      iput v2, v0, Landroid/content/res/TypedArray;->mLength:I
  
+     .line 514
      iget-object v2, v0, Landroid/content/res/TypedArray;->mIndices:[I
  
      aput v4, v2, v4
  
+     .line 516
      return-object v0
  .end method
  
***************
*** 4266,4275 ****
--- 4699,4710 ----
      .end annotation
  
      .prologue
+     .line 925
      iget-object v1, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v1
  
+     .line 926
      :try_start_0
      iget-object v0, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
***************
*** 4281,4286 ****
--- 4716,4722 ----
  
      return-object v0
  
+     .line 927
      :catchall_0
      move-exception v0
  
***************
*** 4302,4311 ****
--- 4738,4749 ----
      .end annotation
  
      .prologue
+     .line 944
      const/4 v2, 0x1
  
      invoke-virtual {p0, p1, p2, v2}, Landroid/content/res/Resources;->getValue(ILandroid/util/TypedValue;Z)V
  
+     .line 947
      :try_start_0
      iget-object v2, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
***************
*** 4327,4335 ****
--- 4765,4775 ----
  
      return-object v2
  
+     .line 949
      :catch_0
      move-exception v0
  
+     .line 950
      .local v0, e:Ljava/lang/Exception;
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 4373,4381 ****
--- 4813,4823 ----
  
      invoke-direct {v1, v2}, Landroid/content/res/Resources$NotFoundException;-><init>(Ljava/lang/String;)V
  
+     .line 952
      .local v1, rnf:Landroid/content/res/Resources$NotFoundException;
      invoke-virtual {v1, v0}, Landroid/content/res/Resources$NotFoundException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 953
      throw v1
  .end method
  
***************
*** 4389,4401 ****
--- 4831,4846 ----
      .end annotation
  
      .prologue
+     .line 979
      iget-object v4, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v4
  
+     .line 980
      :try_start_0
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
+     .line 981
      .local v2, value:Landroid/util/TypedValue;
      const/4 v3, 0x1
  
***************
*** 4403,4408 ****
--- 4848,4854 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 984
      :try_start_1
      iget-object v3, p0, Landroid/content/res/Resources;->mAssets:Landroid/content/res/AssetManager;
  
***************
*** 4426,4434 ****
--- 4872,4882 ----
  
      return-object v3
  
+     .line 986
      :catch_0
      move-exception v0
  
+     .line 987
      .local v0, e:Ljava/lang/Exception;
      new-instance v1, Landroid/content/res/Resources$NotFoundException;
  
***************
*** 4472,4482 ****
--- 4920,4933 ----
  
      invoke-direct {v1, v3}, Landroid/content/res/Resources$NotFoundException;-><init>(Ljava/lang/String;)V
  
+     .line 991
      .local v1, rnf:Landroid/content/res/Resources$NotFoundException;
      invoke-virtual {v1, v0}, Landroid/content/res/Resources$NotFoundException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;
  
+     .line 992
      throw v1
  
+     .line 995
      .end local v0           #e:Ljava/lang/Exception;
      .end local v1           #rnf:Landroid/content/res/Resources$NotFoundException;
      .end local v2           #value:Landroid/util/TypedValue;
***************
*** 4506,4527 ****
--- 4957,4983 ----
  
      const/4 v5, 0x0
  
+     .line 1771
      sget-object v6, Lcom/android/internal/R$styleable;->Extra:[I
  
      invoke-virtual {p0, p2, v6}, Landroid/content/res/Resources;->obtainAttributes(Landroid/util/AttributeSet;[I)Landroid/content/res/TypedArray;
  
      move-result-object v2
  
+     .line 1774
      .local v2, sa:Landroid/content/res/TypedArray;
      invoke-virtual {v2, v5}, Landroid/content/res/TypedArray;->getString(I)Ljava/lang/String;
  
      move-result-object v1
  
+     .line 1776
      .local v1, name:Ljava/lang/String;
      if-nez v1, :cond_0
  
+     .line 1777
      invoke-virtual {v2}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 1778
      new-instance v4, Lorg/xmlpull/v1/XmlPullParserException;
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 4560,4592 ****
--- 5016,5056 ----
  
      throw v4
  
+     .line 1783
      :cond_0
      invoke-virtual {v2, v4}, Landroid/content/res/TypedArray;->peekValue(I)Landroid/util/TypedValue;
  
      move-result-object v3
  
+     .line 1785
      .local v3, v:Landroid/util/TypedValue;
      if-eqz v3, :cond_6
  
+     .line 1786
      iget v6, v3, Landroid/util/TypedValue;->type:I
  
      const/4 v7, 0x3
  
      if-ne v6, v7, :cond_1
  
+     .line 1787
      invoke-virtual {v3}, Landroid/util/TypedValue;->coerceToString()Ljava/lang/CharSequence;
  
      move-result-object v0
  
+     .line 1788
      .local v0, cs:Ljava/lang/CharSequence;
      invoke-virtual {p3, v1, v0}, Landroid/os/Bundle;->putCharSequence(Ljava/lang/String;Ljava/lang/CharSequence;)V
  
+     .line 1809
      .end local v0           #cs:Ljava/lang/CharSequence;
      :goto_0
      invoke-virtual {v2}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 1810
      return-void
  
+     .line 1789
      :cond_1
      iget v6, v3, Landroid/util/TypedValue;->type:I
  
***************
*** 4594,4599 ****
--- 5058,5064 ----
  
      if-ne v6, v7, :cond_3
  
+     .line 1790
      iget v6, v3, Landroid/util/TypedValue;->data:I
  
      if-eqz v6, :cond_2
***************
*** 4608,4613 ****
--- 5073,5079 ----
  
      goto :goto_1
  
+     .line 1791
      :cond_3
      iget v4, v3, Landroid/util/TypedValue;->type:I
  
***************
*** 4621,4632 ****
--- 5087,5100 ----
  
      if-gt v4, v5, :cond_4
  
+     .line 1793
      iget v4, v3, Landroid/util/TypedValue;->data:I
  
      invoke-virtual {p3, v1, v4}, Landroid/os/Bundle;->putInt(Ljava/lang/String;I)V
  
      goto :goto_0
  
+     .line 1794
      :cond_4
      iget v4, v3, Landroid/util/TypedValue;->type:I
  
***************
*** 4634,4639 ****
--- 5102,5108 ----
  
      if-ne v4, v5, :cond_5
  
+     .line 1795
      invoke-virtual {v3}, Landroid/util/TypedValue;->getFloat()F
  
      move-result v4
***************
*** 4642,4650 ****
--- 5111,5121 ----
  
      goto :goto_0
  
+     .line 1797
      :cond_5
      invoke-virtual {v2}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 1798
      new-instance v4, Lorg/xmlpull/v1/XmlPullParserException;
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 4683,4691 ****
--- 5154,5164 ----
  
      throw v4
  
+     .line 1803
      :cond_6
      invoke-virtual {v2}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 1804
      new-instance v4, Lorg/xmlpull/v1/XmlPullParserException;
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 4739,4748 ****
--- 5212,5223 ----
      .prologue
      const/4 v4, 0x3
  
+     .line 1729
      invoke-interface {p1}, Landroid/content/res/XmlResourceParser;->getDepth()I
  
      move-result v1
  
+     .line 1732
      .local v1, outerDepth:I
      :cond_0
      :goto_0
***************
*** 4763,4768 ****
--- 5238,5244 ----
  
      if-le v3, v1, :cond_3
  
+     .line 1733
      :cond_1
      if-eq v2, v4, :cond_0
  
***************
*** 4770,4779 ****
--- 5246,5257 ----
  
      if-eq v2, v3, :cond_0
  
+     .line 1737
      invoke-interface {p1}, Landroid/content/res/XmlResourceParser;->getName()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 1738
      .local v0, nodeName:Ljava/lang/String;
      const-string v3, "extra"
  
***************
*** 4783,4801 ****
--- 5261,5283 ----
  
      if-eqz v3, :cond_2
  
+     .line 1739
      const-string v3, "extra"
  
      invoke-virtual {p0, v3, p1, p2}, Landroid/content/res/Resources;->parseBundleExtra(Ljava/lang/String;Landroid/util/AttributeSet;Landroid/os/Bundle;)V
  
+     .line 1740
      invoke-static {p1}, Lcom/android/internal/util/XmlUtils;->skipCurrentTag(Lorg/xmlpull/v1/XmlPullParser;)V
  
      goto :goto_0
  
+     .line 1743
      :cond_2
      invoke-static {p1}, Lcom/android/internal/util/XmlUtils;->skipCurrentTag(Lorg/xmlpull/v1/XmlPullParser;)V
  
      goto :goto_0
  
+     .line 1746
      .end local v0           #nodeName:Ljava/lang/String;
      :cond_3
      return-void
***************
*** 4806,4819 ****
--- 5288,5304 ----
      .parameter "ci"
  
      .prologue
+     .line 1605
      iput-object p1, p0, Landroid/content/res/Resources;->mCompatibilityInfo:Landroid/content/res/CompatibilityInfo;
  
+     .line 1606
      iget-object v0, p0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
  
      iget-object v1, p0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
  
      invoke-virtual {p0, v0, v1}, Landroid/content/res/Resources;->updateConfiguration(Landroid/content/res/Configuration;Landroid/util/DisplayMetrics;)V
  
+     .line 1607
      return-void
  .end method
  
***************
*** 4821,4835 ****
--- 5306,5323 ----
      .locals 3
  
      .prologue
+     .line 1845
      sget-object v1, Landroid/content/res/Resources;->mSync:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 1846
      :try_start_0
      sget-boolean v0, Landroid/content/res/Resources;->mPreloaded:Z
  
      if-eqz v0, :cond_0
  
+     .line 1847
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v2, "Resources already preloaded"
***************
*** 4838,4843 ****
--- 5326,5332 ----
  
      throw v0
  
+     .line 1851
      :catchall_0
      move-exception v0
  
***************
*** 4847,4866 ****
--- 5336,5359 ----
  
      throw v0
  
+     .line 1849
      :cond_0
      const/4 v0, 0x1
  
      :try_start_1
      sput-boolean v0, Landroid/content/res/Resources;->mPreloaded:Z
  
+     .line 1850
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/content/res/Resources;->mPreloading:Z
  
+     .line 1851
      monitor-exit v1
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 1852
      return-void
  .end method
  
***************
*** 4870,4879 ****
--- 5363,5374 ----
      .parameter "metrics"
  
      .prologue
+     .line 1408
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, p2, v0}, Landroid/content/res/Resources;->updateConfiguration(Landroid/content/res/Configuration;Landroid/util/DisplayMetrics;Landroid/content/res/CompatibilityInfo;)V
  
+     .line 1409
      return-void
  .end method
  
***************
*** 4884,4889 ****
--- 5379,5385 ----
      .parameter "compat"
  
      .prologue
+     .line 1416
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
***************
*** 4892,4899 ****
--- 5388,5397 ----
  
      monitor-enter v21
  
+     .line 1423
      if-eqz p3, :cond_0
  
+     .line 1424
      :try_start_0
      move-object/from16 v0, p3
  
***************
*** 4901,4909 ****
--- 5399,5409 ----
  
      iput-object v0, v1, Landroid/content/res/Resources;->mCompatibilityInfo:Landroid/content/res/CompatibilityInfo;
  
+     .line 1426
      :cond_0
      if-eqz p2, :cond_1
  
+     .line 1427
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
***************
*** 4912,4917 ****
--- 5412,5418 ----
  
      invoke-virtual {v2, v0}, Landroid/util/DisplayMetrics;->setTo(Landroid/util/DisplayMetrics;)V
  
+     .line 1438
      :cond_1
      move-object/from16 v0, p0
  
***************
*** 4919,4924 ****
--- 5420,5426 ----
  
      if-eqz v2, :cond_2
  
+     .line 1439
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mCompatibilityInfo:Landroid/content/res/CompatibilityInfo;
***************
*** 4929,4940 ****
--- 5431,5445 ----
  
      invoke-virtual {v2, v3}, Landroid/content/res/CompatibilityInfo;->applyToDisplayMetrics(Landroid/util/DisplayMetrics;)V
  
+     .line 1441
      :cond_2
      const v20, 0xfffffff
  
+     .line 1442
      .local v20, configChanges:I
      if-eqz p1, :cond_5
  
+     .line 1443
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mTmpConfig:Landroid/content/res/Configuration;
***************
*** 4943,4954 ****
--- 5448,5461 ----
  
      invoke-virtual {v2, v0}, Landroid/content/res/Configuration;->setTo(Landroid/content/res/Configuration;)V
  
+     .line 1444
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mCompatibilityInfo:Landroid/content/res/CompatibilityInfo;
  
      if-eqz v2, :cond_3
  
+     .line 1445
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mCompatibilityInfo:Landroid/content/res/CompatibilityInfo;
***************
*** 4959,4964 ****
--- 5466,5472 ----
  
      invoke-virtual {v2, v3}, Landroid/content/res/CompatibilityInfo;->applyToConfiguration(Landroid/content/res/Configuration;)V
  
+     .line 1447
      :cond_3
      move-object/from16 v0, p0
  
***************
*** 4968,4973 ****
--- 5476,5482 ----
  
      if-nez v2, :cond_4
  
+     .line 1448
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mTmpConfig:Landroid/content/res/Configuration;
***************
*** 4978,4983 ****
--- 5487,5493 ----
  
      iput-object v3, v2, Landroid/content/res/Configuration;->locale:Ljava/util/Locale;
  
+     .line 1450
      :cond_4
      move-object/from16 v0, p0
  
***************
*** 4991,5000 ****
--- 5501,5512 ----
  
      move-result v20
  
+     .line 1451
      invoke-static/range {v20 .. v20}, Landroid/content/pm/ActivityInfo;->activityInfoConfigToNative(I)I
  
      move-result v20
  
+     .line 1453
      :cond_5
      move-object/from16 v0, p0
  
***************
*** 5004,5009 ****
--- 5516,5522 ----
  
      if-nez v2, :cond_6
  
+     .line 1454
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
***************
*** 5014,5019 ****
--- 5527,5533 ----
  
      iput-object v3, v2, Landroid/content/res/Configuration;->locale:Ljava/util/Locale;
  
+     .line 1456
      :cond_6
      move-object/from16 v0, p0
  
***************
*** 5035,5042 ****
--- 5549,5558 ----
  
      iput v3, v2, Landroid/util/DisplayMetrics;->scaledDensity:F
  
+     .line 1458
      const/4 v5, 0x0
  
+     .line 1459
      .local v5, locale:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 5046,5051 ****
--- 5562,5568 ----
  
      if-eqz v2, :cond_7
  
+     .line 1460
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
***************
*** 5056,5061 ****
--- 5573,5579 ----
  
      move-result-object v5
  
+     .line 1461
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mConfiguration:Landroid/content/res/Configuration;
***************
*** 5068,5073 ****
--- 5586,5592 ----
  
      if-eqz v2, :cond_7
  
+     .line 1462
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 5100,5105 ****
--- 5619,5625 ----
  
      move-result-object v5
  
+     .line 1466
      :cond_7
      move-object/from16 v0, p0
  
***************
*** 5115,5126 ****
--- 5635,5648 ----
  
      if-lt v2, v3, :cond_a
  
+     .line 1467
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mMetrics:Landroid/util/DisplayMetrics;
  
      iget v12, v2, Landroid/util/DisplayMetrics;->widthPixels:I
  
+     .line 1468
      .local v12, width:I
      move-object/from16 v0, p0
  
***************
*** 5128,5133 ****
--- 5650,5656 ----
  
      iget v13, v2, Landroid/util/DisplayMetrics;->heightPixels:I
  
+     .line 1475
      .local v13, height:I
      :goto_0
      move-object/from16 v0, p0
***************
*** 5136,5141 ****
--- 5659,5665 ----
  
      iget v10, v2, Landroid/content/res/Configuration;->keyboardHidden:I
  
+     .line 1476
      .local v10, keyboardHidden:I
      const/4 v2, 0x1
  
***************
*** 5151,5158 ****
--- 5675,5684 ----
  
      if-ne v2, v3, :cond_8
  
+     .line 1479
      const/4 v10, 0x3
  
+     .line 1481
      :cond_8
      move-object/from16 v0, p0
  
***************
*** 5258,5263 ****
--- 5784,5790 ----
  
      invoke-virtual/range {v2 .. v19}, Landroid/content/res/AssetManager;->setConfiguration(IILjava/lang/String;IIIIIIIIIIIIII)V
  
+     .line 1496
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mDrawableCache:Landroid/util/LongSparseArray;
***************
*** 5268,5273 ****
--- 5795,5801 ----
  
      invoke-direct {v0, v2, v1}, Landroid/content/res/Resources;->clearDrawableCache(Landroid/util/LongSparseArray;I)V
  
+     .line 1497
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mColorDrawableCache:Landroid/util/LongSparseArray;
***************
*** 5278,5299 ****
--- 5806,5832 ----
  
      invoke-direct {v0, v2, v1}, Landroid/content/res/Resources;->clearDrawableCache(Landroid/util/LongSparseArray;I)V
  
+     .line 1499
      move-object/from16 v0, p0
  
      iget-object v2, v0, Landroid/content/res/Resources;->mColorStateListCache:Landroid/util/SparseArray;
  
      invoke-virtual {v2}, Landroid/util/SparseArray;->clear()V
  
+     .line 1501
      invoke-virtual/range {p0 .. p0}, Landroid/content/res/Resources;->flushLayoutCache()V
  
+     .line 1502
      monitor-exit v21
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1503
      sget-object v3, Landroid/content/res/Resources;->mSync:Ljava/lang/Object;
  
      monitor-enter v3
  
+     .line 1504
      :try_start_1
      move-object/from16 v0, p0
  
***************
*** 5301,5306 ****
--- 5834,5840 ----
  
      if-eqz v2, :cond_9
  
+     .line 1505
      move-object/from16 v0, p1
  
      iget-object v2, v0, Landroid/content/res/Configuration;->locale:Ljava/util/Locale;
***************
*** 5313,5325 ****
--- 5847,5862 ----
  
      iput-object v2, v0, Landroid/content/res/Resources;->mPluralRule:Llibcore/icu/NativePluralRules;
  
+     .line 1507
      :cond_9
      monitor-exit v3
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_1
  
+     .line 1508
      return-void
  
+     .line 1471
      .end local v10           #keyboardHidden:I
      .end local v12           #width:I
      .end local v13           #height:I
***************
*** 5331,5336 ****
--- 5868,5874 ----
  
      iget v12, v2, Landroid/util/DisplayMetrics;->heightPixels:I
  
+     .line 1473
      .restart local v12       #width:I
      move-object/from16 v0, p0
  
***************
*** 5341,5346 ****
--- 5879,5885 ----
      .restart local v13       #height:I
      goto/16 :goto_0
  
+     .line 1502
      .end local v5           #locale:Ljava/lang/String;
      .end local v12           #width:I
      .end local v13           #height:I
***************
*** 5354,5359 ****
--- 5893,5899 ----
  
      throw v2
  
+     .line 1507
      .restart local v5       #locale:Ljava/lang/String;
      .restart local v10       #keyboardHidden:I
      .restart local v12       #width:I
