*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/Resources$NotFoundException.smali	2012-03-23 19:36:26.377161970 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/res/Resources$NotFoundException.smali.original	2012-03-23 19:34:45.111315391 -0500
***************
*** 19,26 ****
--- 19,28 ----
      .locals 0
  
      .prologue
+     .line 165
      invoke-direct {p0}, Ljava/lang/RuntimeException;-><init>()V
  
+     .line 166
      return-void
  .end method
  
***************
*** 29,35 ****
--- 31,39 ----
      .parameter "name"
  
      .prologue
+     .line 169
      invoke-direct {p0, p1}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
+     .line 170
      return-void
  .end method
