*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/usb/UsbSettingsManager.smali	2012-03-23 19:34:54.578488024 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/usb/UsbSettingsManager.smali.original	2012-03-23 19:34:46.870347467 -0500
***************
*** 85,90 ****
--- 85,91 ----
      .locals 2
  
      .prologue
+     .line 66
      new-instance v0, Ljava/io/File;
  
      const-string v1, "/data/system/usb_device_manager.xml"
***************
*** 101,138 ****
--- 102,146 ----
      .parameter "context"
  
      .prologue
+     .line 367
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 72
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePermissionMap:Ljava/util/HashMap;
  
+     .line 75
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPermissionMap:Ljava/util/HashMap;
  
+     .line 78
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
+     .line 81
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
+     .line 84
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
+     .line 365
      new-instance v0, Lcom/android/server/usb/UsbSettingsManager$MyPackageMonitor;
  
      const/4 v1, 0x0
***************
*** 141,173 ****
--- 149,189 ----
  
      iput-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mPackageMonitor:Lcom/android/server/usb/UsbSettingsManager$MyPackageMonitor;
  
+     .line 368
      iput-object p1, p0, Lcom/android/server/usb/UsbSettingsManager;->mContext:Landroid/content/Context;
  
+     .line 369
      invoke-virtual {p1}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mPackageManager:Landroid/content/pm/PackageManager;
  
+     .line 370
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 371
      :try_start_0
      invoke-direct {p0}, Lcom/android/server/usb/UsbSettingsManager;->readSettingsLocked()V
  
+     .line 372
      monitor-exit v1
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 373
      iget-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mPackageMonitor:Lcom/android/server/usb/UsbSettingsManager$MyPackageMonitor;
  
      const/4 v1, 0x1
  
      invoke-virtual {v0, p1, v1}, Lcom/android/server/usb/UsbSettingsManager$MyPackageMonitor;->register(Landroid/content/Context;Z)V
  
+     .line 374
      return-void
  
+     .line 372
      :catchall_0
      move-exception v0
  
***************
*** 185,190 ****
--- 201,207 ----
      .parameter "x1"
  
      .prologue
+     .line 62
      invoke-direct {p0, p1}, Lcom/android/server/usb/UsbSettingsManager;->handlePackageUpdate(Ljava/lang/String;)V
  
      return-void
***************
*** 196,203 ****
--- 213,222 ----
      .parameter "filter"
  
      .prologue
+     .line 700
      const/4 v0, 0x0
  
+     .line 701
      .local v0, changed:Z
      iget-object v3, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
***************
*** 224,229 ****
--- 243,249 ----
  
      check-cast v2, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
  
+     .line 702
      .local v2, test:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      invoke-virtual {p2, v2}, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;->matches(Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;)Z
  
***************
*** 231,244 ****
--- 251,267 ----
  
      if-eqz v3, :cond_0
  
+     .line 703
      iget-object v3, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v3, v2}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 704
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 707
      .end local v2           #test:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      :cond_1
      return v0
***************
*** 250,257 ****
--- 273,282 ----
      .parameter "filter"
  
      .prologue
+     .line 689
      const/4 v0, 0x0
  
+     .line 690
      .local v0, changed:Z
      iget-object v3, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
***************
*** 278,283 ****
--- 303,309 ----
  
      check-cast v2, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
  
+     .line 691
      .local v2, test:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      invoke-virtual {p2, v2}, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;->matches(Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;)Z
  
***************
*** 285,298 ****
--- 311,327 ----
  
      if-eqz v3, :cond_0
  
+     .line 692
      iget-object v3, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v3, v2}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 693
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 696
      .end local v2           #test:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      :cond_1
      return v0
***************
*** 303,315 ****
--- 332,347 ----
      .parameter "packageName"
  
      .prologue
+     .line 956
      const/4 v0, 0x0
  
+     .line 957
      .local v0, cleared:Z
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v5
  
+     .line 958
      :try_start_0
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
***************
*** 319,324 ****
--- 351,357 ----
  
      if-eqz v4, :cond_1
  
+     .line 960
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v4}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 329,334 ****
--- 362,368 ----
  
      move-result-object v3
  
+     .line 961
      .local v3, keys:[Ljava/lang/Object;
      const/4 v1, 0x0
  
***************
*** 338,345 ****
--- 372,381 ----
  
      if-ge v1, v4, :cond_1
  
+     .line 962
      aget-object v2, v3, v1
  
+     .line 963
      .local v2, key:Ljava/lang/Object;
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
***************
*** 353,369 ****
--- 389,409 ----
  
      if-eqz v4, :cond_0
  
+     .line 964
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v4, v2}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 965
      const/4 v0, 0x1
  
+     .line 961
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 969
      .end local v1           #i:I
      .end local v2           #key:Ljava/lang/Object;
      .end local v3           #keys:[Ljava/lang/Object;
***************
*** 376,381 ****
--- 416,422 ----
  
      if-eqz v4, :cond_3
  
+     .line 971
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v4}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 386,391 ****
--- 427,433 ----
  
      move-result-object v3
  
+     .line 972
      .restart local v3       #keys:[Ljava/lang/Object;
      const/4 v1, 0x0
  
***************
*** 395,402 ****
--- 437,446 ----
  
      if-ge v1, v4, :cond_3
  
+     .line 973
      aget-object v2, v3, v1
  
+     .line 974
      .restart local v2       #key:Ljava/lang/Object;
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
***************
*** 410,426 ****
--- 454,474 ----
  
      if-eqz v4, :cond_2
  
+     .line 975
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v4, v2}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 976
      const/4 v0, 0x1
  
+     .line 972
      :cond_2
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 980
      .end local v1           #i:I
      .end local v2           #key:Ljava/lang/Object;
      .end local v3           #keys:[Ljava/lang/Object;
***************
*** 429,434 ****
--- 477,483 ----
  
      return v0
  
+     .line 981
      :catchall_0
      move-exception v4
  
***************
*** 457,466 ****
--- 506,517 ----
      .end annotation
  
      .prologue
+     .line 521
      new-instance v2, Ljava/util/ArrayList;
  
      invoke-direct {v2}, Ljava/util/ArrayList;-><init>()V
  
+     .line 522
      .local v2, matches:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/content/pm/ResolveInfo;>;"
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mPackageManager:Landroid/content/pm/PackageManager;
  
***************
*** 470,480 ****
--- 521,533 ----
  
      move-result-object v4
  
+     .line 524
      .local v4, resolveInfos:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
      invoke-interface {v4}, Ljava/util/List;->size()I
  
      move-result v0
  
+     .line 525
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 482,493 ****
--- 535,548 ----
      :goto_0
      if-ge v1, v0, :cond_1
  
+     .line 526
      invoke-interface {v4, v1}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v3
  
      check-cast v3, Landroid/content/pm/ResolveInfo;
  
+     .line 527
      .local v3, resolveInfo:Landroid/content/pm/ResolveInfo;
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
***************
*** 501,513 ****
--- 556,571 ----
  
      if-eqz v5, :cond_0
  
+     .line 528
      invoke-virtual {v2, v3}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 525
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 531
      .end local v3           #resolveInfo:Landroid/content/pm/ResolveInfo;
      :cond_1
      return-object v2
***************
*** 531,540 ****
--- 589,600 ----
      .end annotation
  
      .prologue
+     .line 506
      new-instance v2, Ljava/util/ArrayList;
  
      invoke-direct {v2}, Ljava/util/ArrayList;-><init>()V
  
+     .line 507
      .local v2, matches:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/content/pm/ResolveInfo;>;"
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mPackageManager:Landroid/content/pm/PackageManager;
  
***************
*** 544,554 ****
--- 604,616 ----
  
      move-result-object v4
  
+     .line 509
      .local v4, resolveInfos:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
      invoke-interface {v4}, Ljava/util/List;->size()I
  
      move-result v0
  
+     .line 510
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 556,567 ****
--- 618,631 ----
      :goto_0
      if-ge v1, v0, :cond_1
  
+     .line 511
      invoke-interface {v4, v1}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v3
  
      check-cast v3, Landroid/content/pm/ResolveInfo;
  
+     .line 512
      .local v3, resolveInfo:Landroid/content/pm/ResolveInfo;
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
***************
*** 575,587 ****
--- 639,654 ----
  
      if-eqz v5, :cond_0
  
+     .line 513
      invoke-virtual {v2, v3}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 510
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 516
      .end local v3           #resolveInfo:Landroid/content/pm/ResolveInfo;
      :cond_1
      return-object v2
***************
*** 592,603 ****
--- 659,673 ----
      .parameter "packageName"
  
      .prologue
+     .line 747
      iget-object v6, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v6
  
+     .line 749
      const/4 v1, 0x0
  
+     .line 752
      .local v1, changed:Z
      :try_start_0
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mPackageManager:Landroid/content/pm/PackageManager;
***************
*** 611,633 ****
--- 681,708 ----
  
      move-result-object v4
  
+     .line 759
      .local v4, info:Landroid/content/pm/PackageInfo;
      :try_start_1
      iget-object v0, v4, Landroid/content/pm/PackageInfo;->activities:[Landroid/content/pm/ActivityInfo;
  
+     .line 760
      .local v0, activities:[Landroid/content/pm/ActivityInfo;
      if-nez v0, :cond_0
  
      monitor-exit v6
  
+     .line 777
      .end local v0           #activities:[Landroid/content/pm/ActivityInfo;
      .end local v4           #info:Landroid/content/pm/PackageInfo;
      :goto_0
      return-void
  
+     .line 754
      :catch_0
      move-exception v2
  
+     .line 755
      .local v2, e:Landroid/content/pm/PackageManager$NameNotFoundException;
      const-string v5, "UsbSettingsManager"
  
***************
*** 651,660 ****
--- 726,737 ----
  
      invoke-static {v5, v7, v2}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 756
      monitor-exit v6
  
      goto :goto_0
  
+     .line 776
      .end local v2           #e:Landroid/content/pm/PackageManager$NameNotFoundException;
      :catchall_0
      move-exception v5
***************
*** 665,670 ****
--- 742,748 ----
  
      throw v5
  
+     .line 761
      .restart local v0       #activities:[Landroid/content/pm/ActivityInfo;
      .restart local v4       #info:Landroid/content/pm/PackageInfo;
      :cond_0
***************
*** 677,682 ****
--- 755,761 ----
  
      if-ge v3, v5, :cond_3
  
+     .line 763
      aget-object v5, v0, v3
  
      const-string v7, "android.hardware.usb.action.USB_DEVICE_ATTACHED"
***************
*** 687,694 ****
--- 766,775 ----
  
      if-eqz v5, :cond_1
  
+     .line 765
      const/4 v1, 0x1
  
+     .line 767
      :cond_1
      aget-object v5, v0, v3
  
***************
*** 700,717 ****
--- 781,803 ----
  
      if-eqz v5, :cond_2
  
+     .line 769
      const/4 v1, 0x1
  
+     .line 761
      :cond_2
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_1
  
+     .line 773
      :cond_3
      if-eqz v1, :cond_4
  
+     .line 774
      invoke-direct {p0}, Lcom/android/server/usb/UsbSettingsManager;->writeSettingsLocked()V
  
+     .line 776
      :cond_4
      monitor-exit v6
      :try_end_2
***************
*** 727,737 ****
--- 813,826 ----
      .parameter "metaDataName"
  
      .prologue
+     .line 712
      const/4 v3, 0x0
  
+     .line 713
      .local v3, parser:Landroid/content/res/XmlResourceParser;
      const/4 v0, 0x0
  
+     .line 716
      .local v0, changed:Z
      :try_start_0
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mPackageManager:Landroid/content/pm/PackageManager;
***************
*** 743,764 ****
--- 832,858 ----
  
      move-result-object v3
  
+     .line 717
      if-nez v3, :cond_1
  
      const/4 v5, 0x0
  
+     .line 739
      if-eqz v3, :cond_0
  
      invoke-interface {v3}, Landroid/content/res/XmlResourceParser;->close()V
  
+     .line 741
      :cond_0
      :goto_0
      return v5
  
+     .line 719
      :cond_1
      :try_start_1
      invoke-static {v3}, Lcom/android/internal/util/XmlUtils;->nextElement(Lorg/xmlpull/v1/XmlPullParser;)V
  
+     .line 720
      :goto_1
      invoke-interface {v3}, Landroid/content/res/XmlResourceParser;->getEventType()I
  
***************
*** 768,777 ****
--- 862,873 ----
  
      if-eq v5, v6, :cond_6
  
+     .line 721
      invoke-interface {v3}, Landroid/content/res/XmlResourceParser;->getName()Ljava/lang/String;
  
      move-result-object v4
  
+     .line 722
      .local v4, tagName:Ljava/lang/String;
      const-string v5, "usb-device"
  
***************
*** 781,790 ****
--- 877,888 ----
  
      if-eqz v5, :cond_4
  
+     .line 723
      invoke-static {v3}, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;->read(Lorg/xmlpull/v1/XmlPullParser;)Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
  
      move-result-object v2
  
+     .line 724
      .local v2, filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      invoke-direct {p0, p1, v2}, Lcom/android/server/usb/UsbSettingsManager;->clearCompatibleMatchesLocked(Ljava/lang/String;Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;)Z
  
***************
*** 792,799 ****
--- 890,899 ----
  
      if-eqz v5, :cond_2
  
+     .line 725
      const/4 v0, 0x1
  
+     .line 734
      .end local v2           #filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      :cond_2
      :goto_2
***************
*** 804,813 ****
--- 904,915 ----
  
      goto :goto_1
  
+     .line 736
      .end local v4           #tagName:Ljava/lang/String;
      :catch_0
      move-exception v1
  
+     .line 737
      .local v1, e:Ljava/lang/Exception;
      :try_start_2
      const-string v5, "UsbSettingsManager"
***************
*** 838,843 ****
--- 940,946 ----
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
  
+     .line 739
      if-eqz v3, :cond_3
  
      .end local v1           #e:Ljava/lang/Exception;
***************
*** 847,854 ****
--- 950,959 ----
      :cond_3
      move v5, v0
  
+     .line 741
      goto :goto_0
  
+     .line 728
      .restart local v4       #tagName:Ljava/lang/String;
      :cond_4
      :try_start_3
***************
*** 860,869 ****
--- 965,976 ----
  
      if-eqz v5, :cond_2
  
+     .line 729
      invoke-static {v3}, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;->read(Lorg/xmlpull/v1/XmlPullParser;)Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
  
      move-result-object v2
  
+     .line 730
      .local v2, filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      invoke-direct {p0, p1, v2}, Lcom/android/server/usb/UsbSettingsManager;->clearCompatibleMatchesLocked(Ljava/lang/String;Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;)Z
      :try_end_3
***************
*** 874,883 ****
--- 981,992 ----
  
      if-eqz v5, :cond_2
  
+     .line 731
      const/4 v0, 0x1
  
      goto :goto_2
  
+     .line 739
      .end local v2           #filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      .end local v4           #tagName:Ljava/lang/String;
      :catchall_0
***************
*** 908,918 ****
--- 1017,1030 ----
  
      const/4 v6, 0x1
  
+     .line 470
      iget-object v0, p1, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
+     .line 472
      .local v0, ai:Landroid/content/pm/ActivityInfo;
      const/4 v3, 0x0
  
+     .line 474
      .local v3, parser:Landroid/content/res/XmlResourceParser;
      :try_start_0
      iget-object v7, p0, Lcom/android/server/usb/UsbSettingsManager;->mPackageManager:Landroid/content/pm/PackageManager;
***************
*** 921,928 ****
--- 1033,1042 ----
  
      move-result-object v3
  
+     .line 475
      if-nez v3, :cond_1
  
+     .line 476
      const-string v6, "UsbSettingsManager"
  
      new-instance v7, Ljava/lang/StringBuilder;
***************
*** 948,966 ****
--- 1062,1084 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 500
      if-eqz v3, :cond_0
  
      :goto_0
      invoke-interface {v3}, Landroid/content/res/XmlResourceParser;->close()V
  
+     .line 502
      :cond_0
      :goto_1
      return v5
  
+     .line 480
      :cond_1
      :try_start_1
      invoke-static {v3}, Lcom/android/internal/util/XmlUtils;->nextElement(Lorg/xmlpull/v1/XmlPullParser;)V
  
+     .line 481
      :goto_2
      invoke-interface {v3}, Landroid/content/res/XmlResourceParser;->getEventType()I
  
***************
*** 968,977 ****
--- 1086,1097 ----
  
      if-eq v7, v6, :cond_7
  
+     .line 482
      invoke-interface {v3}, Landroid/content/res/XmlResourceParser;->getName()Ljava/lang/String;
  
      move-result-object v4
  
+     .line 483
      .local v4, tagName:Ljava/lang/String;
      if-eqz p3, :cond_3
  
***************
*** 983,992 ****
--- 1103,1114 ----
  
      if-eqz v7, :cond_3
  
+     .line 484
      invoke-static {v3}, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;->read(Lorg/xmlpull/v1/XmlPullParser;)Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
  
      move-result-object v2
  
+     .line 485
      .local v2, filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      invoke-virtual {v2, p3}, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;->matches(Landroid/hardware/usb/UsbDevice;)Z
      :try_end_1
***************
*** 997,1002 ****
--- 1119,1125 ----
  
      if-eqz v7, :cond_5
  
+     .line 500
      if-eqz v3, :cond_2
  
      invoke-interface {v3}, Landroid/content/res/XmlResourceParser;->close()V
***************
*** 1004,1011 ****
--- 1127,1136 ----
      :cond_2
      move v5, v6
  
+     .line 486
      goto :goto_1
  
+     .line 489
      .end local v2           #filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      :cond_3
      if-eqz p4, :cond_5
***************
*** 1019,1028 ****
--- 1144,1155 ----
  
      if-eqz v7, :cond_5
  
+     .line 490
      invoke-static {v3}, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;->read(Lorg/xmlpull/v1/XmlPullParser;)Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
  
      move-result-object v2
  
+     .line 491
      .local v2, filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      invoke-virtual {v2, p4}, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;->matches(Landroid/hardware/usb/UsbAccessory;)Z
      :try_end_2
***************
*** 1033,1038 ****
--- 1160,1166 ----
  
      if-eqz v7, :cond_5
  
+     .line 500
      if-eqz v3, :cond_4
  
      invoke-interface {v3}, Landroid/content/res/XmlResourceParser;->close()V
***************
*** 1040,1047 ****
--- 1168,1177 ----
      :cond_4
      move v5, v6
  
+     .line 492
      goto :goto_1
  
+     .line 495
      .end local v2           #filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      :cond_5
      :try_start_3
***************
*** 1052,1061 ****
--- 1182,1193 ----
  
      goto :goto_2
  
+     .line 497
      .end local v4           #tagName:Ljava/lang/String;
      :catch_0
      move-exception v1
  
+     .line 498
      .local v1, e:Ljava/lang/Exception;
      :try_start_4
      const-string v6, "UsbSettingsManager"
***************
*** 1086,1091 ****
--- 1218,1224 ----
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
  
+     .line 500
      if-eqz v3, :cond_0
  
      goto :goto_0
***************
*** 1118,1130 ****
--- 1251,1266 ----
      .end annotation
  
      .prologue
+     .line 378
      const/4 v3, 0x0
  
+     .line 379
      .local v3, packageName:Ljava/lang/String;
      invoke-interface {p1}, Lorg/xmlpull/v1/XmlPullParser;->getAttributeCount()I
  
      move-result v0
  
+     .line 380
      .local v0, count:I
      const/4 v2, 0x0
  
***************
*** 1132,1137 ****
--- 1268,1274 ----
      :goto_0
      if-ge v2, v0, :cond_0
  
+     .line 381
      const-string v4, "package"
  
      invoke-interface {p1, v2}, Lorg/xmlpull/v1/XmlPullParser;->getAttributeName(I)Ljava/lang/String;
***************
*** 1144,1156 ****
--- 1281,1296 ----
  
      if-eqz v4, :cond_2
  
+     .line 382
      invoke-interface {p1, v2}, Lorg/xmlpull/v1/XmlPullParser;->getAttributeValue(I)Ljava/lang/String;
  
      move-result-object v3
  
+     .line 386
      :cond_0
      invoke-static {p1}, Lcom/android/internal/util/XmlUtils;->nextElement(Lorg/xmlpull/v1/XmlPullParser;)V
  
+     .line 387
      const-string v4, "usb-device"
  
      invoke-interface {p1}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
***************
*** 1163,1189 ****
--- 1303,1335 ----
  
      if-eqz v4, :cond_3
  
+     .line 388
      invoke-static {p1}, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;->read(Lorg/xmlpull/v1/XmlPullParser;)Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
  
      move-result-object v1
  
+     .line 389
      .local v1, filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v4, v1, v3}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 394
      .end local v1           #filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      :cond_1
      :goto_1
      invoke-static {p1}, Lcom/android/internal/util/XmlUtils;->nextElement(Lorg/xmlpull/v1/XmlPullParser;)V
  
+     .line 395
      return-void
  
+     .line 380
      :cond_2
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 390
      :cond_3
      const-string v4, "usb-accessory"
  
***************
*** 1197,1206 ****
--- 1343,1354 ----
  
      if-eqz v4, :cond_1
  
+     .line 391
      invoke-static {p1}, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;->read(Lorg/xmlpull/v1/XmlPullParser;)Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
  
      move-result-object v1
  
+     .line 392
      .local v1, filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
***************
*** 1213,1220 ****
--- 1361,1370 ----
      .locals 7
  
      .prologue
+     .line 398
      const/4 v2, 0x0
  
+     .line 400
      .local v2, stream:Ljava/io/FileInputStream;
      :try_start_0
      new-instance v3, Ljava/io/FileInputStream;
***************
*** 1227,1232 ****
--- 1377,1383 ----
      .catch Ljava/io/FileNotFoundException; {:try_start_0 .. :try_end_0} :catch_6
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_5
  
+     .line 401
      .end local v2           #stream:Ljava/io/FileInputStream;
      .local v3, stream:Ljava/io/FileInputStream;
      :try_start_1
***************
*** 1234,1246 ****
--- 1385,1400 ----
  
      move-result-object v1
  
+     .line 402
      .local v1, parser:Lorg/xmlpull/v1/XmlPullParser;
      const/4 v5, 0x0
  
      invoke-interface {v1, v3, v5}, Lorg/xmlpull/v1/XmlPullParser;->setInput(Ljava/io/InputStream;Ljava/lang/String;)V
  
+     .line 404
      invoke-static {v1}, Lcom/android/internal/util/XmlUtils;->nextElement(Lorg/xmlpull/v1/XmlPullParser;)V
  
+     .line 405
      :goto_0
      invoke-interface {v1}, Lorg/xmlpull/v1/XmlPullParser;->getEventType()I
  
***************
*** 1250,1259 ****
--- 1404,1415 ----
  
      if-eq v5, v6, :cond_2
  
+     .line 406
      invoke-interface {v1}, Lorg/xmlpull/v1/XmlPullParser;->getName()Ljava/lang/String;
  
      move-result-object v4
  
+     .line 407
      .local v4, tagName:Ljava/lang/String;
      const-string v5, "preference"
  
***************
*** 1263,1268 ****
--- 1419,1425 ----
  
      if-eqz v5, :cond_1
  
+     .line 408
      invoke-direct {p0, v1}, Lcom/android/server/usb/UsbSettingsManager;->readPreference(Lorg/xmlpull/v1/XmlPullParser;)V
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_1
***************
*** 1271,1276 ****
--- 1428,1434 ----
  
      goto :goto_0
  
+     .line 413
      .end local v1           #parser:Lorg/xmlpull/v1/XmlPullParser;
      .end local v4           #tagName:Ljava/lang/String;
      :catch_0
***************
*** 1278,1297 ****
--- 1436,1459 ----
  
      move-object v2, v3
  
+     .line 419
      .end local v3           #stream:Ljava/io/FileInputStream;
      .restart local v2       #stream:Ljava/io/FileInputStream;
      :goto_1
      if-eqz v2, :cond_0
  
+     .line 421
      :try_start_2
      invoke-virtual {v2}, Ljava/io/FileInputStream;->close()V
      :try_end_2
      .catch Ljava/io/IOException; {:try_start_2 .. :try_end_2} :catch_2
  
+     .line 426
      :cond_0
      :goto_2
      return-void
  
+     .line 410
      .end local v2           #stream:Ljava/io/FileInputStream;
      .restart local v1       #parser:Lorg/xmlpull/v1/XmlPullParser;
      .restart local v3       #stream:Ljava/io/FileInputStream;
***************
*** 1306,1311 ****
--- 1468,1474 ----
  
      goto :goto_0
  
+     .line 415
      .end local v1           #parser:Lorg/xmlpull/v1/XmlPullParser;
      .end local v4           #tagName:Ljava/lang/String;
      :catch_1
***************
*** 1313,1318 ****
--- 1476,1482 ----
  
      move-object v2, v3
  
+     .line 416
      .end local v3           #stream:Ljava/io/FileInputStream;
      .local v0, e:Ljava/lang/Exception;
      .restart local v2       #stream:Ljava/io/FileInputStream;
***************
*** 1324,1337 ****
--- 1488,1504 ----
  
      invoke-static {v5, v6, v0}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 417
      sget-object v5, Lcom/android/server/usb/UsbSettingsManager;->sSettingsFile:Ljava/io/File;
  
      invoke-virtual {v5}, Ljava/io/File;->delete()Z
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
  
+     .line 419
      if-eqz v2, :cond_0
  
+     .line 421
      :try_start_5
      invoke-virtual {v2}, Ljava/io/FileInputStream;->close()V
      :try_end_5
***************
*** 1339,1356 ****
--- 1506,1526 ----
  
      goto :goto_2
  
+     .line 422
      .end local v0           #e:Ljava/lang/Exception;
      :catch_2
      move-exception v5
  
      goto :goto_2
  
+     .line 419
      .end local v2           #stream:Ljava/io/FileInputStream;
      .restart local v1       #parser:Lorg/xmlpull/v1/XmlPullParser;
      .restart local v3       #stream:Ljava/io/FileInputStream;
      :cond_2
      if-eqz v3, :cond_3
  
+     .line 421
      :try_start_6
      invoke-virtual {v3}, Ljava/io/FileInputStream;->close()V
      :try_end_6
***************
*** 1360,1369 ****
--- 1530,1541 ----
      :goto_4
      move-object v2, v3
  
+     .line 425
      .end local v3           #stream:Ljava/io/FileInputStream;
      .restart local v2       #stream:Ljava/io/FileInputStream;
      goto :goto_2
  
+     .line 419
      .end local v1           #parser:Lorg/xmlpull/v1/XmlPullParser;
      :catchall_0
      move-exception v5
***************
*** 1371,1385 ****
--- 1543,1560 ----
      :goto_5
      if-eqz v2, :cond_4
  
+     .line 421
      :try_start_7
      invoke-virtual {v2}, Ljava/io/FileInputStream;->close()V
      :try_end_7
      .catch Ljava/io/IOException; {:try_start_7 .. :try_end_7} :catch_3
  
+     .line 419
      :cond_4
      :goto_6
      throw v5
  
+     .line 422
      :catch_3
      move-exception v6
  
***************
*** 1393,1398 ****
--- 1568,1574 ----
  
      goto :goto_4
  
+     .line 419
      .end local v1           #parser:Lorg/xmlpull/v1/XmlPullParser;
      :catchall_1
      move-exception v5
***************
*** 1403,1413 ****
--- 1579,1591 ----
      .restart local v2       #stream:Ljava/io/FileInputStream;
      goto :goto_5
  
+     .line 415
      :catch_5
      move-exception v0
  
      goto :goto_3
  
+     .line 413
      :catch_6
      move-exception v5
  
***************
*** 1421,1430 ****
--- 1599,1610 ----
      .parameter "pi"
  
      .prologue
+     .line 812
      invoke-static {}, Landroid/os/Binder;->getCallingUid()I
  
      move-result v4
  
+     .line 816
      .local v4, uid:I
      :try_start_0
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mPackageManager:Landroid/content/pm/PackageManager;
***************
*** 1435,1445 ****
--- 1615,1627 ----
  
      move-result-object v0
  
+     .line 817
      .local v0, aInfo:Landroid/content/pm/ApplicationInfo;
      iget v5, v0, Landroid/content/pm/ApplicationInfo;->uid:I
  
      if-eq v5, v4, :cond_0
  
+     .line 818
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 1476,1485 ****
--- 1658,1669 ----
      :try_end_0
      .catch Landroid/content/pm/PackageManager$NameNotFoundException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 821
      .end local v0           #aInfo:Landroid/content/pm/ApplicationInfo;
      :catch_0
      move-exception v1
  
+     .line 822
      .local v1, e:Landroid/content/pm/PackageManager$NameNotFoundException;
      new-instance v5, Ljava/lang/IllegalArgumentException;
  
***************
*** 1511,1516 ****
--- 1695,1701 ----
  
      throw v5
  
+     .line 825
      .end local v1           #e:Landroid/content/pm/PackageManager$NameNotFoundException;
      .restart local v0       #aInfo:Landroid/content/pm/ApplicationInfo;
      :cond_0
***************
*** 1518,1523 ****
--- 1703,1709 ----
  
      move-result-wide v2
  
+     .line 826
      .local v2, identity:J
      const-string v5, "com.android.systemui"
  
***************
*** 1525,1546 ****
--- 1711,1737 ----
  
      invoke-virtual {p1, v5, v6}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 828
      const/high16 v5, 0x1000
  
      invoke-virtual {p1, v5}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 829
      const-string v5, "android.intent.extra.INTENT"
  
      invoke-virtual {p1, v5, p3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 830
      const-string v5, "package"
  
      invoke-virtual {p1, v5, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 831
      const-string v5, "uid"
  
      invoke-virtual {p1, v5, v4}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 833
      :try_start_1
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mContext:Landroid/content/Context;
  
***************
*** 1549,1562 ****
--- 1740,1757 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/content/ActivityNotFoundException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 837
      :goto_0
      invoke-static {v2, v3}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
+     .line 839
      return-void
  
+     .line 834
      :catch_1
      move-exception v1
  
+     .line 835
      .local v1, e:Landroid/content/ActivityNotFoundException;
      :try_start_2
      const-string v5, "UsbSettingsManager"
***************
*** 1569,1574 ****
--- 1764,1770 ----
  
      goto :goto_0
  
+     .line 837
      .end local v1           #e:Landroid/content/ActivityNotFoundException;
      :catchall_0
      move-exception v5
***************
*** 1601,1620 ****
--- 1797,1821 ----
      .end annotation
  
      .prologue
+     .line 590
      .local p2, matches:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/content/pm/ResolveInfo;>;"
      invoke-virtual {p2}, Ljava/util/ArrayList;->size()I
  
      move-result v1
  
+     .line 593
      .local v1, count:I
      if-nez v1, :cond_1
  
+     .line 594
      if-eqz p5, :cond_0
  
+     .line 595
      invoke-virtual/range {p5 .. p5}, Landroid/hardware/usb/UsbAccessory;->getUri()Ljava/lang/String;
  
      move-result-object v8
  
+     .line 596
      .local v8, uri:Ljava/lang/String;
      if-eqz v8, :cond_0
  
***************
*** 1624,1633 ****
--- 1825,1836 ----
  
      if-lez v9, :cond_0
  
+     .line 599
      new-instance v3, Landroid/content/Intent;
  
      invoke-direct {v3}, Landroid/content/Intent;-><init>()V
  
+     .line 600
      .local v3, dialogIntent:Landroid/content/Intent;
      const-string v9, "com.android.systemui"
  
***************
*** 1635,1654 ****
--- 1838,1861 ----
  
      invoke-virtual {v3, v9, v10}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 602
      const/high16 v9, 0x1000
  
      invoke-virtual {v3, v9}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 603
      const-string v9, "accessory"
  
      move-object/from16 v0, p5
  
      invoke-virtual {v3, v9, v0}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 604
      const-string v9, "uri"
  
      invoke-virtual {v3, v9, v8}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 606
      :try_start_0
      iget-object v9, p0, Lcom/android/server/usb/UsbSettingsManager;->mContext:Landroid/content/Context;
  
***************
*** 1656,1672 ****
--- 1863,1882 ----
      :try_end_0
      .catch Landroid/content/ActivityNotFoundException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 686
      .end local v3           #dialogIntent:Landroid/content/Intent;
      .end local v8           #uri:Ljava/lang/String;
      :cond_0
      :goto_0
      return-void
  
+     .line 607
      .restart local v3       #dialogIntent:Landroid/content/Intent;
      .restart local v8       #uri:Ljava/lang/String;
      :catch_0
      move-exception v4
  
+     .line 608
      .local v4, e:Landroid/content/ActivityNotFoundException;
      const-string v9, "UsbSettingsManager"
  
***************
*** 1676,1687 ****
--- 1886,1899 ----
  
      goto :goto_0
  
+     .line 617
      .end local v3           #dialogIntent:Landroid/content/Intent;
      .end local v4           #e:Landroid/content/ActivityNotFoundException;
      .end local v8           #uri:Ljava/lang/String;
      :cond_1
      const/4 v2, 0x0
  
+     .line 618
      .local v2, defaultRI:Landroid/content/pm/ResolveInfo;
      const/4 v9, 0x1
  
***************
*** 1689,1694 ****
--- 1901,1907 ----
  
      if-nez p3, :cond_2
  
+     .line 621
      const/4 v9, 0x0
  
      invoke-virtual {p2, v9}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1697,1702 ****
--- 1910,1916 ----
  
      check-cast v6, Landroid/content/pm/ResolveInfo;
  
+     .line 622
      .local v6, rInfo:Landroid/content/pm/ResolveInfo;
      iget-object v9, v6, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
***************
*** 1718,1743 ****
--- 1932,1962 ----
  
      if-eqz v9, :cond_2
  
+     .line 625
      move-object v2, v6
  
+     .line 629
      .end local v6           #rInfo:Landroid/content/pm/ResolveInfo;
      :cond_2
      if-nez v2, :cond_3
  
      if-eqz p3, :cond_3
  
+     .line 631
      const/4 v5, 0x0
  
      .local v5, i:I
      :goto_1
      if-ge v5, v1, :cond_3
  
+     .line 632
      invoke-virtual {p2, v5}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v6
  
      check-cast v6, Landroid/content/pm/ResolveInfo;
  
+     .line 633
      .restart local v6       #rInfo:Landroid/content/pm/ResolveInfo;
      iget-object v9, v6, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
***************
*** 1753,1767 ****
--- 1972,1990 ----
  
      if-eqz v9, :cond_5
  
+     .line 635
      move-object v2, v6
  
+     .line 641
      .end local v5           #i:I
      .end local v6           #rInfo:Landroid/content/pm/ResolveInfo;
      :cond_3
      if-eqz v2, :cond_7
  
+     .line 643
      if-eqz p4, :cond_6
  
+     .line 644
      iget-object v9, v2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
      iget-object v9, v9, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
***************
*** 1772,1777 ****
--- 1995,2001 ----
  
      invoke-virtual {p0, v0, v9}, Lcom/android/server/usb/UsbSettingsManager;->grantDevicePermission(Landroid/hardware/usb/UsbDevice;I)V
  
+     .line 651
      :cond_4
      :goto_2
      :try_start_1
***************
*** 1789,1794 ****
--- 2013,2019 ----
  
      invoke-virtual {p1, v9}, Landroid/content/Intent;->setComponent(Landroid/content/ComponentName;)Landroid/content/Intent;
  
+     .line 654
      iget-object v9, p0, Lcom/android/server/usb/UsbSettingsManager;->mContext:Landroid/content/Context;
  
      invoke-virtual {v9, p1}, Landroid/content/Context;->startActivity(Landroid/content/Intent;)V
***************
*** 1797,1805 ****
--- 2022,2032 ----
  
      goto :goto_0
  
+     .line 655
      :catch_1
      move-exception v4
  
+     .line 656
      .restart local v4       #e:Landroid/content/ActivityNotFoundException;
      const-string v9, "UsbSettingsManager"
  
***************
*** 1809,1814 ****
--- 2036,2042 ----
  
      goto :goto_0
  
+     .line 631
      .end local v4           #e:Landroid/content/ActivityNotFoundException;
      .restart local v5       #i:I
      .restart local v6       #rInfo:Landroid/content/pm/ResolveInfo;
***************
*** 1817,1827 ****
--- 2045,2057 ----
  
      goto :goto_1
  
+     .line 645
      .end local v5           #i:I
      .end local v6           #rInfo:Landroid/content/pm/ResolveInfo;
      :cond_6
      if-eqz p5, :cond_4
  
+     .line 646
      iget-object v9, v2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
      iget-object v9, v9, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
***************
*** 1834,1859 ****
--- 2064,2094 ----
  
      goto :goto_2
  
+     .line 659
      :cond_7
      new-instance v7, Landroid/content/Intent;
  
      invoke-direct {v7}, Landroid/content/Intent;-><init>()V
  
+     .line 660
      .local v7, resolverIntent:Landroid/content/Intent;
      const/high16 v9, 0x1000
  
      invoke-virtual {v7, v9}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 662
      const/4 v9, 0x1
  
      if-ne v1, v9, :cond_9
  
+     .line 664
      const-string v9, "com.android.systemui"
  
      const-string v10, "com.android.systemui.usb.UsbConfirmActivity"
  
      invoke-virtual {v7, v9, v10}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 666
      const-string v10, "rinfo"
  
      const/4 v9, 0x0
***************
*** 1866,1879 ****
--- 2101,2117 ----
  
      invoke-virtual {v7, v10, v9}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 668
      if-eqz p4, :cond_8
  
+     .line 669
      const-string v9, "device"
  
      move-object/from16 v0, p4
  
      invoke-virtual {v7, v9, v0}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 681
      :goto_3
      :try_start_2
      iget-object v9, p0, Lcom/android/server/usb/UsbSettingsManager;->mContext:Landroid/content/Context;
***************
*** 1884,1892 ****
--- 2122,2132 ----
  
      goto/16 :goto_0
  
+     .line 682
      :catch_2
      move-exception v4
  
+     .line 683
      .restart local v4       #e:Landroid/content/ActivityNotFoundException;
      const-string v9, "UsbSettingsManager"
  
***************
*** 1912,1917 ****
--- 2152,2158 ----
  
      goto/16 :goto_0
  
+     .line 671
      .end local v4           #e:Landroid/content/ActivityNotFoundException;
      :cond_8
      const-string v9, "accessory"
***************
*** 1922,1927 ****
--- 2163,2169 ----
  
      goto :goto_3
  
+     .line 675
      :cond_9
      const-string v9, "com.android.systemui"
  
***************
*** 1929,1938 ****
--- 2171,2182 ----
  
      invoke-virtual {v7, v9, v10}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 677
      const-string v9, "rlist"
  
      invoke-virtual {v7, v9, p2}, Landroid/content/Intent;->putParcelableArrayListExtra(Ljava/lang/String;Ljava/util/ArrayList;)Landroid/content/Intent;
  
+     .line 678
      const-string v9, "android.intent.extra.INTENT"
  
      invoke-virtual {v7, v9, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
***************
*** 1944,1951 ****
--- 2188,2197 ----
      .locals 10
  
      .prologue
+     .line 429
      const/4 v2, 0x0
  
+     .line 431
      .local v2, fos:Ljava/io/FileOutputStream;
      :try_start_0
      new-instance v3, Ljava/io/FileOutputStream;
***************
*** 1954,1974 ****
--- 2200,2224 ----
  
      invoke-direct {v3, v7}, Ljava/io/FileOutputStream;-><init>(Ljava/io/File;)V
  
+     .line 433
      .local v3, fstr:Ljava/io/FileOutputStream;
      new-instance v6, Ljava/io/BufferedOutputStream;
  
      invoke-direct {v6, v3}, Ljava/io/BufferedOutputStream;-><init>(Ljava/io/OutputStream;)V
  
+     .line 434
      .local v6, str:Ljava/io/BufferedOutputStream;
      new-instance v5, Lcom/android/internal/util/FastXmlSerializer;
  
      invoke-direct {v5}, Lcom/android/internal/util/FastXmlSerializer;-><init>()V
  
+     .line 435
      .local v5, serializer:Lcom/android/internal/util/FastXmlSerializer;
      const-string v7, "utf-8"
  
      invoke-virtual {v5, v6, v7}, Lcom/android/internal/util/FastXmlSerializer;->setOutput(Ljava/io/OutputStream;Ljava/lang/String;)V
  
+     .line 436
      const/4 v7, 0x0
  
      const/4 v8, 0x1
***************
*** 1979,1996 ****
--- 2229,2249 ----
  
      invoke-virtual {v5, v7, v8}, Lcom/android/internal/util/FastXmlSerializer;->startDocument(Ljava/lang/String;Ljava/lang/Boolean;)V
  
+     .line 437
      const-string v7, "http://xmlpull.org/v1/doc/features.html#indent-output"
  
      const/4 v8, 0x1
  
      invoke-virtual {v5, v7, v8}, Lcom/android/internal/util/FastXmlSerializer;->setFeature(Ljava/lang/String;Z)V
  
+     .line 438
      const/4 v7, 0x0
  
      const-string v8, "settings"
  
      invoke-virtual {v5, v7, v8}, Lcom/android/internal/util/FastXmlSerializer;->startTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 440
      iget-object v7, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v7}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 2015,2020 ****
--- 2268,2274 ----
  
      check-cast v1, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
  
+     .line 441
      .local v1, filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      const/4 v7, 0x0
  
***************
*** 2022,2027 ****
--- 2276,2282 ----
  
      invoke-virtual {v5, v7, v8}, Lcom/android/internal/util/FastXmlSerializer;->startTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 442
      const/4 v8, 0x0
  
      const-string v9, "package"
***************
*** 2036,2043 ****
--- 2291,2300 ----
  
      invoke-virtual {v5, v8, v9, v7}, Lcom/android/internal/util/FastXmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 443
      invoke-virtual {v1, v5}, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;->write(Lorg/xmlpull/v1/XmlSerializer;)V
  
+     .line 444
      const/4 v7, 0x0
  
      const-string v8, "preference"
***************
*** 2048,2053 ****
--- 2305,2311 ----
  
      goto :goto_0
  
+     .line 460
      .end local v1           #filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      .end local v3           #fstr:Ljava/io/FileOutputStream;
      .end local v4           #i$:Ljava/util/Iterator;
***************
*** 2056,2061 ****
--- 2314,2320 ----
      :catch_0
      move-exception v0
  
+     .line 461
      .local v0, e:Ljava/lang/Exception;
      const-string v7, "UsbSettingsManager"
  
***************
*** 2063,2076 ****
--- 2322,2338 ----
  
      invoke-static {v7, v8, v0}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 462
      sget-object v7, Lcom/android/server/usb/UsbSettingsManager;->sSettingsFile:Ljava/io/File;
  
      invoke-virtual {v7}, Ljava/io/File;->delete()Z
  
+     .line 464
      .end local v0           #e:Ljava/lang/Exception;
      :goto_1
      return-void
  
+     .line 447
      .restart local v3       #fstr:Ljava/io/FileOutputStream;
      .restart local v4       #i$:Ljava/util/Iterator;
      .restart local v5       #serializer:Lcom/android/internal/util/FastXmlSerializer;
***************
*** 2100,2105 ****
--- 2362,2368 ----
  
      check-cast v1, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
  
+     .line 448
      .local v1, filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      const/4 v7, 0x0
  
***************
*** 2107,2112 ****
--- 2370,2376 ----
  
      invoke-virtual {v5, v7, v8}, Lcom/android/internal/util/FastXmlSerializer;->startTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 449
      const/4 v8, 0x0
  
      const-string v9, "package"
***************
*** 2121,2128 ****
--- 2385,2394 ----
  
      invoke-virtual {v5, v8, v9, v7}, Lcom/android/internal/util/FastXmlSerializer;->attribute(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 450
      invoke-virtual {v1, v5}, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;->write(Lorg/xmlpull/v1/XmlSerializer;)V
  
+     .line 451
      const/4 v7, 0x0
  
      const-string v8, "preference"
***************
*** 2131,2136 ****
--- 2397,2403 ----
  
      goto :goto_2
  
+     .line 454
      .end local v1           #filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      :cond_1
      const/4 v7, 0x0
***************
*** 2139,2150 ****
--- 2406,2421 ----
  
      invoke-virtual {v5, v7, v8}, Lcom/android/internal/util/FastXmlSerializer;->endTag(Ljava/lang/String;Ljava/lang/String;)Lorg/xmlpull/v1/XmlSerializer;
  
+     .line 455
      invoke-virtual {v5}, Lcom/android/internal/util/FastXmlSerializer;->endDocument()V
  
+     .line 457
      invoke-virtual {v6}, Ljava/io/BufferedOutputStream;->flush()V
  
+     .line 458
      invoke-static {v3}, Landroid/os/FileUtils;->sync(Ljava/io/FileOutputStream;)Z
  
+     .line 459
      invoke-virtual {v6}, Ljava/io/BufferedOutputStream;->close()V
      :try_end_1
      .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_0
***************
*** 2159,2188 ****
--- 2430,2465 ----
      .parameter "accessory"
  
      .prologue
+     .line 562
      new-instance v1, Landroid/content/Intent;
  
      const-string v0, "android.hardware.usb.action.USB_ACCESSORY_ATTACHED"
  
      invoke-direct {v1, v0}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 563
      .local v1, intent:Landroid/content/Intent;
      const-string v0, "accessory"
  
      invoke-virtual {v1, v0, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 564
      const/high16 v0, 0x1000
  
      invoke-virtual {v1, v0}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 568
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v4
  
+     .line 569
      :try_start_0
      invoke-direct {p0, p1, v1}, Lcom/android/server/usb/UsbSettingsManager;->getAccessoryMatchesLocked(Landroid/hardware/usb/UsbAccessory;Landroid/content/Intent;)Ljava/util/ArrayList;
  
      move-result-object v2
  
+     .line 572
      .local v2, matches:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/content/pm/ResolveInfo;>;"
      iget-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
***************
*** 2196,2206 ****
--- 2473,2485 ----
  
      check-cast v3, Ljava/lang/String;
  
+     .line 573
      .local v3, defaultPackage:Ljava/lang/String;
      monitor-exit v4
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 575
      const/4 v4, 0x0
  
      move-object v0, p0
***************
*** 2209,2216 ****
--- 2488,2497 ----
  
      invoke-direct/range {v0 .. v5}, Lcom/android/server/usb/UsbSettingsManager;->resolveActivity(Landroid/content/Intent;Ljava/util/ArrayList;Ljava/lang/String;Landroid/hardware/usb/UsbDevice;Landroid/hardware/usb/UsbAccessory;)V
  
+     .line 576
      return-void
  
+     .line 573
      .end local v2           #matches:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/content/pm/ResolveInfo;>;"
      .end local v3           #defaultPackage:Ljava/lang/String;
      :catchall_0
***************
*** 2229,2253 ****
--- 2510,2539 ----
      .parameter "accessory"
  
      .prologue
+     .line 580
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPermissionMap:Ljava/util/HashMap;
  
      invoke-virtual {v1, p1}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 582
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.hardware.usb.action.USB_ACCESSORY_DETACHED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 584
      .local v0, intent:Landroid/content/Intent;
      const-string v1, "accessory"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 585
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mContext:Landroid/content/Context;
  
      invoke-virtual {v1, v0}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 586
      return-void
  .end method
  
***************
*** 2256,2267 ****
--- 2542,2555 ----
      .parameter "accessory"
  
      .prologue
+     .line 806
      invoke-virtual {p0, p1}, Lcom/android/server/usb/UsbSettingsManager;->hasPermission(Landroid/hardware/usb/UsbAccessory;)Z
  
      move-result v0
  
      if-nez v0, :cond_0
  
+     .line 807
      new-instance v0, Ljava/lang/SecurityException;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 2286,2291 ****
--- 2574,2580 ----
  
      throw v0
  
+     .line 809
      :cond_0
      return-void
  .end method
***************
*** 2295,2306 ****
--- 2584,2597 ----
      .parameter "device"
  
      .prologue
+     .line 800
      invoke-virtual {p0, p1}, Lcom/android/server/usb/UsbSettingsManager;->hasPermission(Landroid/hardware/usb/UsbDevice;)Z
  
      move-result v0
  
      if-nez v0, :cond_0
  
+     .line 801
      new-instance v0, Ljava/lang/SecurityException;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 2325,2330 ****
--- 2616,2622 ----
  
      throw v0
  
+     .line 803
      :cond_0
      return-void
  .end method
***************
*** 2334,2343 ****
--- 2626,2637 ----
      .parameter "packageName"
  
      .prologue
+     .line 948
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 949
      :try_start_0
      invoke-direct {p0, p1}, Lcom/android/server/usb/UsbSettingsManager;->clearPackageDefaultsLocked(Ljava/lang/String;)Z
  
***************
*** 2345,2357 ****
--- 2639,2655 ----
  
      if-eqz v0, :cond_0
  
+     .line 950
      invoke-direct {p0}, Lcom/android/server/usb/UsbSettingsManager;->writeSettingsLocked()V
  
+     .line 952
      :cond_0
      monitor-exit v1
  
+     .line 953
      return-void
  
+     .line 952
      :catchall_0
      move-exception v0
  
***************
*** 2367,2396 ****
--- 2665,2700 ----
      .parameter "device"
  
      .prologue
+     .line 535
      new-instance v1, Landroid/content/Intent;
  
      const-string v0, "android.hardware.usb.action.USB_DEVICE_ATTACHED"
  
      invoke-direct {v1, v0}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 536
      .local v1, intent:Landroid/content/Intent;
      const-string v0, "device"
  
      invoke-virtual {v1, v0, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 537
      const/high16 v0, 0x1000
  
      invoke-virtual {v1, v0}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 541
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v4
  
+     .line 542
      :try_start_0
      invoke-direct {p0, p1, v1}, Lcom/android/server/usb/UsbSettingsManager;->getDeviceMatchesLocked(Landroid/hardware/usb/UsbDevice;Landroid/content/Intent;)Ljava/util/ArrayList;
  
      move-result-object v2
  
+     .line 545
      .local v2, matches:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/content/pm/ResolveInfo;>;"
      iget-object v0, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
***************
*** 2404,2414 ****
--- 2708,2720 ----
  
      check-cast v3, Ljava/lang/String;
  
+     .line 546
      .local v3, defaultPackage:Ljava/lang/String;
      monitor-exit v4
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 548
      const/4 v5, 0x0
  
      move-object v0, p0
***************
*** 2417,2424 ****
--- 2723,2732 ----
  
      invoke-direct/range {v0 .. v5}, Lcom/android/server/usb/UsbSettingsManager;->resolveActivity(Landroid/content/Intent;Ljava/util/ArrayList;Ljava/lang/String;Landroid/hardware/usb/UsbDevice;Landroid/hardware/usb/UsbAccessory;)V
  
+     .line 549
      return-void
  
+     .line 546
      .end local v2           #matches:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/content/pm/ResolveInfo;>;"
      .end local v3           #defaultPackage:Ljava/lang/String;
      :catchall_0
***************
*** 2437,2442 ****
--- 2745,2751 ----
      .parameter "device"
  
      .prologue
+     .line 553
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePermissionMap:Ljava/util/HashMap;
  
      invoke-virtual {p1}, Landroid/hardware/usb/UsbDevice;->getDeviceName()Ljava/lang/String;
***************
*** 2445,2465 ****
--- 2754,2778 ----
  
      invoke-virtual {v1, v2}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 555
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.hardware.usb.action.USB_DEVICE_DETACHED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 556
      .local v0, intent:Landroid/content/Intent;
      const-string v1, "device"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 558
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mContext:Landroid/content/Context;
  
      invoke-virtual {v1, v0}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 559
      return-void
  .end method
  
***************
*** 2469,2483 ****
--- 2782,2799 ----
      .parameter "pw"
  
      .prologue
+     .line 985
      iget-object v8, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v8
  
+     .line 986
      :try_start_0
      const-string v7, "  Device permissions:"
  
      invoke-virtual {p2, v7}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 987
      iget-object v7, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePermissionMap:Ljava/util/HashMap;
  
      invoke-virtual {v7}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 2502,2507 ****
--- 2818,2824 ----
  
      check-cast v2, Ljava/lang/String;
  
+     .line 988
      .local v2, deviceName:Ljava/lang/String;
      new-instance v7, Ljava/lang/StringBuilder;
  
***************
*** 2529,2534 ****
--- 2846,2852 ----
  
      invoke-virtual {p2, v7}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 989
      iget-object v7, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePermissionMap:Ljava/util/HashMap;
  
      invoke-virtual {v7, v2}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 2537,2547 ****
--- 2855,2867 ----
  
      check-cast v6, Landroid/util/SparseBooleanArray;
  
+     .line 990
      .local v6, uidList:Landroid/util/SparseBooleanArray;
      invoke-virtual {v6}, Landroid/util/SparseBooleanArray;->size()I
  
      move-result v1
  
+     .line 991
      .local v1, count:I
      const/4 v4, 0x0
  
***************
*** 2549,2554 ****
--- 2869,2875 ----
      :goto_1
      if-ge v4, v1, :cond_0
  
+     .line 992
      new-instance v7, Ljava/lang/StringBuilder;
  
      invoke-direct {v7}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2577,2586 ****
--- 2898,2909 ----
  
      invoke-virtual {p2, v7}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 991
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_1
  
+     .line 994
      :cond_0
      const-string v7, ""
  
***************
*** 2588,2593 ****
--- 2911,2917 ----
  
      goto :goto_0
  
+     .line 1014
      .end local v1           #count:I
      .end local v2           #deviceName:Ljava/lang/String;
      .end local v4           #i:I
***************
*** 2602,2607 ****
--- 2926,2932 ----
  
      throw v7
  
+     .line 996
      .restart local v5       #i$:Ljava/util/Iterator;
      :cond_1
      :try_start_1
***************
*** 2609,2614 ****
--- 2934,2940 ----
  
      invoke-virtual {p2, v7}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 997
      iget-object v7, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPermissionMap:Ljava/util/HashMap;
  
      invoke-virtual {v7}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 2632,2637 ****
--- 2958,2964 ----
  
      check-cast v0, Landroid/hardware/usb/UsbAccessory;
  
+     .line 998
      .local v0, accessory:Landroid/hardware/usb/UsbAccessory;
      new-instance v7, Ljava/lang/StringBuilder;
  
***************
*** 2659,2664 ****
--- 2986,2992 ----
  
      invoke-virtual {p2, v7}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 999
      iget-object v7, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPermissionMap:Ljava/util/HashMap;
  
      invoke-virtual {v7, v0}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 2667,2677 ****
--- 2995,3007 ----
  
      check-cast v6, Landroid/util/SparseBooleanArray;
  
+     .line 1000
      .restart local v6       #uidList:Landroid/util/SparseBooleanArray;
      invoke-virtual {v6}, Landroid/util/SparseBooleanArray;->size()I
  
      move-result v1
  
+     .line 1001
      .restart local v1       #count:I
      const/4 v4, 0x0
  
***************
*** 2679,2684 ****
--- 3009,3015 ----
      :goto_3
      if-ge v4, v1, :cond_2
  
+     .line 1002
      new-instance v7, Ljava/lang/StringBuilder;
  
      invoke-direct {v7}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2707,2716 ****
--- 3038,3049 ----
  
      invoke-virtual {p2, v7}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 1001
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_3
  
+     .line 1004
      :cond_2
      const-string v7, ""
  
***************
*** 2718,2723 ****
--- 3051,3057 ----
  
      goto :goto_2
  
+     .line 1006
      .end local v0           #accessory:Landroid/hardware/usb/UsbAccessory;
      .end local v1           #count:I
      .end local v4           #i:I
***************
*** 2727,2732 ****
--- 3061,3067 ----
  
      invoke-virtual {p2, v7}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 1007
      iget-object v7, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v7}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 2750,2755 ****
--- 3085,3091 ----
  
      check-cast v3, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
  
+     .line 1008
      .local v3, filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      new-instance v7, Ljava/lang/StringBuilder;
  
***************
*** 2791,2802 ****
--- 3127,3140 ----
  
      goto :goto_4
  
+     .line 1010
      .end local v3           #filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      :cond_4
      const-string v7, "  Accessory preferences:"
  
      invoke-virtual {p2, v7}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 1011
      iget-object v7, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v7}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 2820,2825 ****
--- 3158,3164 ----
  
      check-cast v3, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
  
+     .line 1012
      .local v3, filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      new-instance v7, Ljava/lang/StringBuilder;
  
***************
*** 2861,2872 ****
--- 3200,3213 ----
  
      goto :goto_5
  
+     .line 1014
      .end local v3           #filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      :cond_5
      monitor-exit v8
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 1015
      return-void
  .end method
  
***************
*** 2876,2885 ****
--- 3217,3228 ----
      .parameter "uid"
  
      .prologue
+     .line 929
      iget-object v2, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 930
      :try_start_0
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPermissionMap:Ljava/util/HashMap;
  
***************
*** 2889,2897 ****
--- 3232,3242 ----
  
      check-cast v0, Landroid/util/SparseBooleanArray;
  
+     .line 931
      .local v0, uidList:Landroid/util/SparseBooleanArray;
      if-nez v0, :cond_0
  
+     .line 932
      new-instance v0, Landroid/util/SparseBooleanArray;
  
      .end local v0           #uidList:Landroid/util/SparseBooleanArray;
***************
*** 2899,2918 ****
--- 3244,3268 ----
  
      invoke-direct {v0, v1}, Landroid/util/SparseBooleanArray;-><init>(I)V
  
+     .line 933
      .restart local v0       #uidList:Landroid/util/SparseBooleanArray;
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPermissionMap:Ljava/util/HashMap;
  
      invoke-virtual {v1, p1, v0}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 935
      :cond_0
      const/4 v1, 0x1
  
      invoke-virtual {v0, p2, v1}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 936
      monitor-exit v2
  
+     .line 937
      return-void
  
+     .line 936
      .end local v0           #uidList:Landroid/util/SparseBooleanArray;
      :catchall_0
      move-exception v1
***************
*** 2930,2944 ****
--- 3280,3297 ----
      .parameter "uid"
  
      .prologue
+     .line 917
      iget-object v3, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v3
  
+     .line 918
      :try_start_0
      invoke-virtual {p1}, Landroid/hardware/usb/UsbDevice;->getDeviceName()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 919
      .local v0, deviceName:Ljava/lang/String;
      iget-object v2, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePermissionMap:Ljava/util/HashMap;
  
***************
*** 2948,2956 ****
--- 3301,3311 ----
  
      check-cast v1, Landroid/util/SparseBooleanArray;
  
+     .line 920
      .local v1, uidList:Landroid/util/SparseBooleanArray;
      if-nez v1, :cond_0
  
+     .line 921
      new-instance v1, Landroid/util/SparseBooleanArray;
  
      .end local v1           #uidList:Landroid/util/SparseBooleanArray;
***************
*** 2958,2977 ****
--- 3313,3337 ----
  
      invoke-direct {v1, v2}, Landroid/util/SparseBooleanArray;-><init>(I)V
  
+     .line 922
      .restart local v1       #uidList:Landroid/util/SparseBooleanArray;
      iget-object v2, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePermissionMap:Ljava/util/HashMap;
  
      invoke-virtual {v2, v0, v1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 924
      :cond_0
      const/4 v2, 0x1
  
      invoke-virtual {v1, p2, v2}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 925
      monitor-exit v3
  
+     .line 926
      return-void
  
+     .line 925
      .end local v0           #deviceName:Ljava/lang/String;
      .end local v1           #uidList:Landroid/util/SparseBooleanArray;
      :catchall_0
***************
*** 2991,3000 ****
--- 3351,3362 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 940
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 941
      :try_start_0
      iget-object v2, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
***************
*** 3010,3018 ****
--- 3372,3382 ----
  
      monitor-exit v1
  
+     .line 943
      :goto_0
      return v0
  
+     .line 942
      :cond_0
      iget-object v2, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
***************
*** 3030,3035 ****
--- 3394,3400 ----
  
      goto :goto_0
  
+     .line 944
      :catchall_0
      move-exception v0
  
***************
*** 3039,3044 ****
--- 3404,3410 ----
  
      throw v0
  
+     .line 943
      :cond_1
      const/4 v0, 0x0
  
***************
*** 3055,3064 ****
--- 3421,3432 ----
      .parameter "accessory"
  
      .prologue
+     .line 790
      iget-object v2, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 791
      :try_start_0
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPermissionMap:Ljava/util/HashMap;
  
***************
*** 3068,3080 ****
--- 3436,3451 ----
  
      check-cast v0, Landroid/util/SparseBooleanArray;
  
+     .line 792
      .local v0, uidList:Landroid/util/SparseBooleanArray;
      if-nez v0, :cond_0
  
+     .line 793
      const/4 v1, 0x0
  
      monitor-exit v2
  
+     .line 795
      :goto_0
      return v1
  
***************
*** 3091,3096 ****
--- 3462,3468 ----
  
      goto :goto_0
  
+     .line 796
      .end local v0           #uidList:Landroid/util/SparseBooleanArray;
      :catchall_0
      move-exception v1
***************
*** 3107,3116 ****
--- 3479,3490 ----
      .parameter "device"
  
      .prologue
+     .line 780
      iget-object v2, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 781
      :try_start_0
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePermissionMap:Ljava/util/HashMap;
  
***************
*** 3124,3136 ****
--- 3498,3513 ----
  
      check-cast v0, Landroid/util/SparseBooleanArray;
  
+     .line 782
      .local v0, uidList:Landroid/util/SparseBooleanArray;
      if-nez v0, :cond_0
  
+     .line 783
      const/4 v1, 0x0
  
      monitor-exit v2
  
+     .line 785
      :goto_0
      return v1
  
***************
*** 3147,3152 ****
--- 3524,3530 ----
  
      goto :goto_0
  
+     .line 786
      .end local v0           #uidList:Landroid/util/SparseBooleanArray;
      :catchall_0
      move-exception v1
***************
*** 3165,3174 ****
--- 3543,3554 ----
      .parameter "pi"
  
      .prologue
+     .line 862
      new-instance v0, Landroid/content/Intent;
  
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
+     .line 865
      .local v0, intent:Landroid/content/Intent;
      invoke-virtual {p0, p1}, Lcom/android/server/usb/UsbSettingsManager;->hasPermission(Landroid/hardware/usb/UsbAccessory;)Z
  
***************
*** 3176,3191 ****
--- 3556,3574 ----
  
      if-eqz v1, :cond_0
  
+     .line 866
      const-string v1, "accessory"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 867
      const-string v1, "permission"
  
      const/4 v2, 0x1
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     .line 869
      :try_start_0
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mContext:Landroid/content/Context;
  
***************
*** 3195,3212 ****
--- 3578,3599 ----
      :try_end_0
      .catch Landroid/app/PendingIntent$CanceledException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 878
      :goto_0
      return-void
  
+     .line 876
      :cond_0
      const-string v1, "accessory"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 877
      invoke-direct {p0, v0, p2, p3}, Lcom/android/server/usb/UsbSettingsManager;->requestPermissionDialog(Landroid/content/Intent;Ljava/lang/String;Landroid/app/PendingIntent;)V
  
      goto :goto_0
  
+     .line 870
      :catch_0
      move-exception v1
  
***************
*** 3220,3229 ****
--- 3607,3618 ----
      .parameter "pi"
  
      .prologue
+     .line 842
      new-instance v0, Landroid/content/Intent;
  
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
+     .line 845
      .local v0, intent:Landroid/content/Intent;
      invoke-virtual {p0, p1}, Lcom/android/server/usb/UsbSettingsManager;->hasPermission(Landroid/hardware/usb/UsbDevice;)Z
  
***************
*** 3231,3246 ****
--- 3620,3638 ----
  
      if-eqz v1, :cond_0
  
+     .line 846
      const-string v1, "device"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 847
      const-string v1, "permission"
  
      const/4 v2, 0x1
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     .line 849
      :try_start_0
      iget-object v1, p0, Lcom/android/server/usb/UsbSettingsManager;->mContext:Landroid/content/Context;
  
***************
*** 3250,3267 ****
--- 3642,3663 ----
      :try_end_0
      .catch Landroid/app/PendingIntent$CanceledException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 859
      :goto_0
      return-void
  
+     .line 857
      :cond_0
      const-string v1, "device"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 858
      invoke-direct {p0, v0, p2, p3}, Lcom/android/server/usb/UsbSettingsManager;->requestPermissionDialog(Landroid/content/Intent;Ljava/lang/String;Landroid/app/PendingIntent;)V
  
      goto :goto_0
  
+     .line 850
      :catch_0
      move-exception v1
  
***************
*** 3278,3297 ****
--- 3674,3698 ----
  
      const/4 v3, 0x0
  
+     .line 899
      new-instance v1, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
  
      invoke-direct {v1, p1}, Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;-><init>(Landroid/hardware/usb/UsbAccessory;)V
  
+     .line 900
      .local v1, filter:Lcom/android/server/usb/UsbSettingsManager$AccessoryFilter;
      const/4 v0, 0x0
  
+     .line 901
      .local v0, changed:Z
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v4
  
+     .line 902
      if-nez p2, :cond_3
  
+     .line 903
      :try_start_0
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
***************
*** 3303,3324 ****
--- 3704,3731 ----
  
      move v0, v2
  
+     .line 910
      :cond_0
      :goto_0
      if-eqz v0, :cond_1
  
+     .line 911
      invoke-direct {p0}, Lcom/android/server/usb/UsbSettingsManager;->writeSettingsLocked()V
  
+     .line 913
      :cond_1
      monitor-exit v4
  
+     .line 914
      return-void
  
      :cond_2
      move v0, v3
  
+     .line 903
      goto :goto_0
  
+     .line 905
      :cond_3
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
***************
*** 3334,3348 ****
--- 3741,3758 ----
  
      move v0, v2
  
+     .line 906
      :goto_1
      if-eqz v0, :cond_0
  
+     .line 907
      iget-object v2, p0, Lcom/android/server/usb/UsbSettingsManager;->mAccessoryPreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v2, v1, p2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      goto :goto_0
  
+     .line 913
      :catchall_0
      move-exception v2
  
***************
*** 3355,3360 ****
--- 3765,3771 ----
      :cond_4
      move v0, v3
  
+     .line 905
      goto :goto_1
  .end method
  
***************
*** 3368,3387 ****
--- 3779,3803 ----
  
      const/4 v3, 0x0
  
+     .line 881
      new-instance v1, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
  
      invoke-direct {v1, p1}, Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;-><init>(Landroid/hardware/usb/UsbDevice;)V
  
+     .line 882
      .local v1, filter:Lcom/android/server/usb/UsbSettingsManager$DeviceFilter;
      const/4 v0, 0x0
  
+     .line 883
      .local v0, changed:Z
      iget-object v4, p0, Lcom/android/server/usb/UsbSettingsManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v4
  
+     .line 884
      if-nez p2, :cond_3
  
+     .line 885
      :try_start_0
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
***************
*** 3393,3414 ****
--- 3809,3836 ----
  
      move v0, v2
  
+     .line 892
      :cond_0
      :goto_0
      if-eqz v0, :cond_1
  
+     .line 893
      invoke-direct {p0}, Lcom/android/server/usb/UsbSettingsManager;->writeSettingsLocked()V
  
+     .line 895
      :cond_1
      monitor-exit v4
  
+     .line 896
      return-void
  
      :cond_2
      move v0, v3
  
+     .line 885
      goto :goto_0
  
+     .line 887
      :cond_3
      iget-object v5, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
***************
*** 3424,3438 ****
--- 3846,3863 ----
  
      move v0, v2
  
+     .line 888
      :goto_1
      if-eqz v0, :cond_0
  
+     .line 889
      iget-object v2, p0, Lcom/android/server/usb/UsbSettingsManager;->mDevicePreferenceMap:Ljava/util/HashMap;
  
      invoke-virtual {v2, v1, p2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      goto :goto_0
  
+     .line 895
      :catchall_0
      move-exception v2
  
***************
*** 3445,3449 ****
--- 3870,3875 ----
      :cond_4
      move v0, v3
  
+     .line 887
      goto :goto_1
  .end method
