*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ActivityNotFoundException.smali	2012-03-23 19:36:26.482163886 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ActivityNotFoundException.smali.original	2012-03-23 19:34:45.112315409 -0500
***************
*** 8,15 ****
--- 8,17 ----
      .locals 0
  
      .prologue
+     .line 27
      invoke-direct {p0}, Ljava/lang/RuntimeException;-><init>()V
  
+     .line 28
      return-void
  .end method
  
***************
*** 18,24 ****
--- 20,28 ----
      .parameter "name"
  
      .prologue
+     .line 32
      invoke-direct {p0, p1}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
+     .line 33
      return-void
  .end method
