*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/AsyncTaskLoader.smali	2012-03-23 19:36:25.257141548 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/AsyncTaskLoader.smali.original	2012-03-23 19:34:45.102315227 -0500
***************
*** 61,73 ****
--- 61,76 ----
      .parameter "context"
  
      .prologue
+     .line 107
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      invoke-direct {p0, p1}, Landroid/content/Loader;-><init>(Landroid/content/Context;)V
  
+     .line 103
      const-wide/16 v0, -0x2710
  
      iput-wide v0, p0, Landroid/content/AsyncTaskLoader;->mLastLoadCompleteTime:J
  
+     .line 108
      return-void
  .end method
  
***************
*** 82,118 ****
--- 85,129 ----
  
      const/4 v0, 0x0
  
+     .line 151
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      if-eqz v1, :cond_1
  
+     .line 152
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mCancellingTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      if-eqz v1, :cond_2
  
+     .line 157
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      iget-boolean v1, v1, Landroid/content/AsyncTaskLoader$LoadTask;->waiting:Z
  
      if-eqz v1, :cond_0
  
+     .line 158
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      iput-boolean v0, v1, Landroid/content/AsyncTaskLoader$LoadTask;->waiting:Z
  
+     .line 159
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mHandler:Landroid/os/Handler;
  
      iget-object v2, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      invoke-virtual {v1, v2}, Landroid/os/Handler;->removeCallbacks(Ljava/lang/Runnable;)V
  
+     .line 161
      :cond_0
      iput-object v3, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
+     .line 181
      :cond_1
      :goto_0
      return v0
  
+     .line 163
      :cond_2
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
***************
*** 120,139 ****
--- 131,154 ----
  
      if-eqz v1, :cond_3
  
+     .line 167
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      iput-boolean v0, v1, Landroid/content/AsyncTaskLoader$LoadTask;->waiting:Z
  
+     .line 168
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mHandler:Landroid/os/Handler;
  
      iget-object v2, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      invoke-virtual {v1, v2}, Landroid/os/Handler;->removeCallbacks(Ljava/lang/Runnable;)V
  
+     .line 169
      iput-object v3, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      goto :goto_0
  
+     .line 172
      :cond_3
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
***************
*** 141,153 ****
--- 156,171 ----
  
      move-result v0
  
+     .line 174
      .local v0, cancelled:Z
      if-eqz v0, :cond_4
  
+     .line 175
      iget-object v1, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      iput-object v1, p0, Landroid/content/AsyncTaskLoader;->mCancellingTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
+     .line 177
      :cond_4
      iput-object v3, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
***************
*** 169,195 ****
--- 187,219 ----
      .end annotation
  
      .prologue
+     .line 215
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      .local p1, task:Landroid/content/AsyncTaskLoader$LoadTask;,"Landroid/content/AsyncTaskLoader<TD;>.LoadTask;"
      .local p2, data:Ljava/lang/Object;,"TD;"
      invoke-virtual {p0, p2}, Landroid/content/AsyncTaskLoader;->onCanceled(Ljava/lang/Object;)V
  
+     .line 216
      iget-object v0, p0, Landroid/content/AsyncTaskLoader;->mCancellingTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      if-ne v0, p1, :cond_0
  
+     .line 218
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/AsyncTaskLoader;->mLastLoadCompleteTime:J
  
+     .line 219
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/AsyncTaskLoader;->mCancellingTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
+     .line 220
      invoke-virtual {p0}, Landroid/content/AsyncTaskLoader;->executePendingTask()V
  
+     .line 222
      :cond_0
      return-void
  .end method
***************
*** 209,214 ****
--- 233,239 ----
      .end annotation
  
      .prologue
+     .line 225
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      .local p1, task:Landroid/content/AsyncTaskLoader$LoadTask;,"Landroid/content/AsyncTaskLoader<TD;>.LoadTask;"
      .local p2, data:Ljava/lang/Object;,"TD;"
***************
*** 216,226 ****
--- 241,254 ----
  
      if-eq v0, p1, :cond_0
  
+     .line 227
      invoke-virtual {p0, p1, p2}, Landroid/content/AsyncTaskLoader;->dispatchOnCancelled(Landroid/content/AsyncTaskLoader$LoadTask;Ljava/lang/Object;)V
  
+     .line 239
      :goto_0
      return-void
  
+     .line 229
      :cond_0
      invoke-virtual {p0}, Landroid/content/AsyncTaskLoader;->isAbandoned()Z
  
***************
*** 228,237 ****
--- 256,267 ----
  
      if-eqz v0, :cond_1
  
+     .line 231
      invoke-virtual {p0, p2}, Landroid/content/AsyncTaskLoader;->onCanceled(Ljava/lang/Object;)V
  
      goto :goto_0
  
+     .line 233
      :cond_1
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
***************
*** 239,248 ****
--- 269,280 ----
  
      iput-wide v0, p0, Landroid/content/AsyncTaskLoader;->mLastLoadCompleteTime:J
  
+     .line 234
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
+     .line 236
      invoke-virtual {p0, p2}, Landroid/content/AsyncTaskLoader;->deliverResult(Ljava/lang/Object;)V
  
      goto :goto_0
***************
*** 256,271 ****
      .parameter "args"
  
      .prologue
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      invoke-super {p0, p1, p2, p3, p4}, Landroid/content/Loader;->dump(Ljava/lang/String;Ljava/io/FileDescriptor;Ljava/io/PrintWriter;[Ljava/lang/String;)V
  
      iget-object v0, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      if-eqz v0, :cond_0
  
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string v0, "mTask="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
--- 288,306 ----
      .parameter "args"
  
      .prologue
+     .line 281
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      invoke-super {p0, p1, p2, p3, p4}, Landroid/content/Loader;->dump(Ljava/lang/String;Ljava/io/FileDescriptor;Ljava/io/PrintWriter;[Ljava/lang/String;)V
  
+     .line 282
      iget-object v0, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      if-eqz v0, :cond_0
  
+     .line 283
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string/jumbo v0, "mTask="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 273,278 ****
--- 308,314 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/Object;)V
  
+     .line 284
      const-string v0, " waiting="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 283,293 ****
--- 319,331 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 286
      :cond_0
      iget-object v0, p0, Landroid/content/AsyncTaskLoader;->mCancellingTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      if-eqz v0, :cond_1
  
+     .line 287
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mCancellingTask="
***************
*** 298,303 ****
--- 336,342 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/Object;)V
  
+     .line 288
      const-string v0, " waiting="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 308,313 ****
--- 347,353 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 290
      :cond_1
      iget-wide v0, p0, Landroid/content/AsyncTaskLoader;->mUpdateThrottle:J
  
***************
*** 317,336 ****
  
      if-eqz v0, :cond_2
  
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string v0, "mUpdateThrottle="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      iget-wide v0, p0, Landroid/content/AsyncTaskLoader;->mUpdateThrottle:J
  
      invoke-static {v0, v1, p3}, Landroid/util/TimeUtils;->formatDuration(JLjava/io/PrintWriter;)V
  
      const-string v0, " mLastLoadCompleteTime="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      iget-wide v0, p0, Landroid/content/AsyncTaskLoader;->mLastLoadCompleteTime:J
  
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
--- 357,380 ----
  
      if-eqz v0, :cond_2
  
+     .line 291
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
!     const-string/jumbo v0, "mUpdateThrottle="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 292
      iget-wide v0, p0, Landroid/content/AsyncTaskLoader;->mUpdateThrottle:J
  
      invoke-static {v0, v1, p3}, Landroid/util/TimeUtils;->formatDuration(JLjava/io/PrintWriter;)V
  
+     .line 293
      const-string v0, " mLastLoadCompleteTime="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 294
      iget-wide v0, p0, Landroid/content/AsyncTaskLoader;->mLastLoadCompleteTime:J
  
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
***************
*** 339,346 ****
--- 383,392 ----
  
      invoke-static {v0, v1, v2, v3, p3}, Landroid/util/TimeUtils;->formatDuration(JJLjava/io/PrintWriter;)V
  
+     .line 296
      invoke-virtual {p3}, Ljava/io/PrintWriter;->println()V
  
+     .line 298
      :cond_2
      return-void
  .end method
***************
*** 349,354 ****
--- 395,401 ----
      .locals 8
  
      .prologue
+     .line 192
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      iget-object v2, p0, Landroid/content/AsyncTaskLoader;->mCancellingTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
***************
*** 358,381 ****
--- 405,432 ----
  
      if-eqz v2, :cond_1
  
+     .line 193
      iget-object v2, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      iget-boolean v2, v2, Landroid/content/AsyncTaskLoader$LoadTask;->waiting:Z
  
      if-eqz v2, :cond_0
  
+     .line 194
      iget-object v2, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      const/4 v3, 0x0
  
      iput-boolean v3, v2, Landroid/content/AsyncTaskLoader$LoadTask;->waiting:Z
  
+     .line 195
      iget-object v2, p0, Landroid/content/AsyncTaskLoader;->mHandler:Landroid/os/Handler;
  
      iget-object v3, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      invoke-virtual {v2, v3}, Landroid/os/Handler;->removeCallbacks(Ljava/lang/Runnable;)V
  
+     .line 197
      :cond_0
      iget-wide v2, p0, Landroid/content/AsyncTaskLoader;->mUpdateThrottle:J
  
***************
*** 385,394 ****
--- 436,447 ----
  
      if-lez v2, :cond_2
  
+     .line 198
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v0
  
+     .line 199
      .local v0, now:J
      iget-wide v2, p0, Landroid/content/AsyncTaskLoader;->mLastLoadCompleteTime:J
  
***************
*** 400,411 ****
--- 453,466 ----
  
      if-gez v2, :cond_2
  
+     .line 204
      iget-object v2, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
      const/4 v3, 0x1
  
      iput-boolean v3, v2, Landroid/content/AsyncTaskLoader$LoadTask;->waiting:Z
  
+     .line 205
      iget-object v2, p0, Landroid/content/AsyncTaskLoader;->mHandler:Landroid/os/Handler;
  
      iget-object v3, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
***************
*** 418,428 ****
--- 473,485 ----
  
      invoke-virtual {v2, v3, v4, v5}, Landroid/os/Handler;->postAtTime(Ljava/lang/Runnable;J)Z
  
+     .line 212
      .end local v0           #now:J
      :cond_1
      :goto_0
      return-void
  
+     .line 210
      :cond_2
      iget-object v3, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
***************
*** 455,460 ****
--- 512,518 ----
      .end annotation
  
      .prologue
+     .line 189
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      .local p1, data:Ljava/lang/Object;,"TD;"
      return-void
***************
*** 464,482 ****
--- 522,545 ----
      .locals 1
  
      .prologue
+     .line 126
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      invoke-super {p0}, Landroid/content/Loader;->onForceLoad()V
  
+     .line 127
      invoke-virtual {p0}, Landroid/content/AsyncTaskLoader;->cancelLoad()Z
  
+     .line 128
      new-instance v0, Landroid/content/AsyncTaskLoader$LoadTask;
  
      invoke-direct {v0, p0}, Landroid/content/AsyncTaskLoader$LoadTask;-><init>(Landroid/content/AsyncTaskLoader;)V
  
      iput-object v0, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
+     .line 130
      invoke-virtual {p0}, Landroid/content/AsyncTaskLoader;->executePendingTask()V
  
+     .line 131
      return-void
  .end method
  
***************
*** 489,494 ****
--- 552,558 ----
      .end annotation
  
      .prologue
+     .line 255
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      invoke-virtual {p0}, Landroid/content/AsyncTaskLoader;->loadInBackground()Ljava/lang/Object;
  
***************
*** 502,522 ****
--- 566,590 ----
      .parameter "delayMS"
  
      .prologue
+     .line 118
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      iput-wide p1, p0, Landroid/content/AsyncTaskLoader;->mUpdateThrottle:J
  
+     .line 119
      const-wide/16 v0, 0x0
  
      cmp-long v0, p1, v0
  
      if-eqz v0, :cond_0
  
+     .line 120
      new-instance v0, Landroid/os/Handler;
  
      invoke-direct {v0}, Landroid/os/Handler;-><init>()V
  
      iput-object v0, p0, Landroid/content/AsyncTaskLoader;->mHandler:Landroid/os/Handler;
  
+     .line 122
      :cond_0
      return-void
  .end method
***************
*** 525,536 ****
--- 593,607 ----
      .locals 2
  
      .prologue
+     .line 269
      .local p0, this:Landroid/content/AsyncTaskLoader;,"Landroid/content/AsyncTaskLoader<TD;>;"
      iget-object v0, p0, Landroid/content/AsyncTaskLoader;->mTask:Landroid/content/AsyncTaskLoader$LoadTask;
  
+     .line 270
      .local v0, task:Landroid/content/AsyncTaskLoader$LoadTask;,"Landroid/content/AsyncTaskLoader<TD;>.LoadTask;"
      if-eqz v0, :cond_0
  
+     .line 272
      :try_start_0
      #getter for: Landroid/content/AsyncTaskLoader$LoadTask;->done:Ljava/util/concurrent/CountDownLatch;
      invoke-static {v0}, Landroid/content/AsyncTaskLoader$LoadTask;->access$000(Landroid/content/AsyncTaskLoader$LoadTask;)Ljava/util/concurrent/CountDownLatch;
***************
*** 541,550 ****
--- 612,623 ----
      :try_end_0
      .catch Ljava/lang/InterruptedException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 277
      :cond_0
      :goto_0
      return-void
  
+     .line 273
      :catch_0
      move-exception v1
  
