*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ClipboardManager$1.smali	2012-03-23 19:36:31.718259366 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ClipboardManager$1.smali.original	2012-03-23 19:34:45.168316431 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 55
      iput-object p1, p0, Landroid/content/ClipboardManager$1;->this$0:Landroid/content/ClipboardManager;
  
      invoke-direct {p0}, Landroid/content/IOnPrimaryClipChangedListener$Stub;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .locals 2
  
      .prologue
+     .line 57
      iget-object v0, p0, Landroid/content/ClipboardManager$1;->this$0:Landroid/content/ClipboardManager;
  
      #getter for: Landroid/content/ClipboardManager;->mHandler:Landroid/os/Handler;
***************
*** 48,52 ****
--- 50,55 ----
  
      invoke-virtual {v0, v1}, Landroid/os/Handler;->sendEmptyMessage(I)Z
  
+     .line 58
      return-void
  .end method
