*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ContentService$1.smali	2012-03-23 19:36:28.051192497 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ContentService$1.smali.original	2012-03-23 19:34:45.128315701 -0500
***************
*** 40,45 ****
--- 40,46 ----
      .parameter
  
      .prologue
+     .line 88
      iput-object p1, p0, Landroid/content/ContentService$1;->this$0:Landroid/content/ContentService;
  
      iput-object p2, p0, Landroid/content/ContentService$1;->val$pidCounts:Landroid/util/SparseIntArray;
***************
*** 57,62 ****
--- 58,64 ----
      .parameter "rhs"
  
      .prologue
+     .line 91
      iget-object v2, p0, Landroid/content/ContentService$1;->val$pidCounts:Landroid/util/SparseIntArray;
  
      invoke-virtual {p1}, Ljava/lang/Integer;->intValue()I
***************
*** 67,72 ****
--- 69,75 ----
  
      move-result v0
  
+     .line 92
      .local v0, lc:I
      iget-object v2, p0, Landroid/content/ContentService$1;->val$pidCounts:Landroid/util/SparseIntArray;
  
***************
*** 78,98 ****
--- 81,107 ----
  
      move-result v1
  
+     .line 93
      .local v1, rc:I
      if-ge v0, v1, :cond_0
  
+     .line 94
      const/4 v2, 0x1
  
+     .line 98
      :goto_0
      return v2
  
+     .line 95
      :cond_0
      if-le v0, v1, :cond_1
  
+     .line 96
      const/4 v2, -0x1
  
      goto :goto_0
  
+     .line 98
      :cond_1
      const/4 v2, 0x0
  
***************
*** 105,110 ****
--- 114,120 ----
      .parameter "x1"
  
      .prologue
+     .line 88
      check-cast p1, Ljava/lang/Integer;
  
      .end local p1
