*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ContentResolver$CursorWrapperInner.smali	2012-03-23 19:36:25.329142860 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ContentResolver$CursorWrapperInner.smali.original	2012-03-23 19:34:45.103315245 -0500
***************
*** 36,59 ****
--- 36,65 ----
      .parameter "icp"
  
      .prologue
+     .line 1584
      iput-object p1, p0, Landroid/content/ContentResolver$CursorWrapperInner;->this$0:Landroid/content/ContentResolver;
  
+     .line 1585
      invoke-direct {p0, p2}, Landroid/database/CrossProcessCursorWrapper;-><init>(Landroid/database/Cursor;)V
  
+     .line 1581
      invoke-static {}, Ldalvik/system/CloseGuard;->get()Ldalvik/system/CloseGuard;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mCloseGuard:Ldalvik/system/CloseGuard;
  
+     .line 1586
      iput-object p3, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mContentProvider:Landroid/content/IContentProvider;
  
+     .line 1587
      iget-object v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mCloseGuard:Ldalvik/system/CloseGuard;
  
      const-string v1, "close"
  
      invoke-virtual {v0, v1}, Ldalvik/system/CloseGuard;->open(Ljava/lang/String;)V
  
+     .line 1588
      return-void
  .end method
  
***************
*** 63,88 ****
--- 69,100 ----
      .locals 2
  
      .prologue
+     .line 1592
      invoke-super {p0}, Landroid/database/CrossProcessCursorWrapper;->close()V
  
+     .line 1593
      iget-object v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->this$0:Landroid/content/ContentResolver;
  
      iget-object v1, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mContentProvider:Landroid/content/IContentProvider;
  
      invoke-virtual {v0, v1}, Landroid/content/ContentResolver;->releaseProvider(Landroid/content/IContentProvider;)Z
  
+     .line 1594
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mProviderReleased:Z
  
+     .line 1596
      iget-object v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mCloseGuard:Ldalvik/system/CloseGuard;
  
      if-eqz v0, :cond_0
  
+     .line 1597
      iget-object v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mCloseGuard:Ldalvik/system/CloseGuard;
  
      invoke-virtual {v0}, Ldalvik/system/CloseGuard;->close()V
  
+     .line 1599
      :cond_0
      return-void
  .end method
***************
*** 96,110 ****
--- 108,125 ----
      .end annotation
  
      .prologue
+     .line 1604
      :try_start_0
      iget-object v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mCloseGuard:Ldalvik/system/CloseGuard;
  
      if-eqz v0, :cond_0
  
+     .line 1605
      iget-object v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mCloseGuard:Ldalvik/system/CloseGuard;
  
      invoke-virtual {v0}, Ldalvik/system/CloseGuard;->warnIfOpen()V
  
+     .line 1608
      :cond_0
      iget-boolean v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mProviderReleased:Z
  
***************
*** 114,125 ****
--- 129,142 ----
  
      if-eqz v0, :cond_1
  
+     .line 1611
      const-string v0, "CursorWrapperInner"
  
      const-string v1, "Cursor finalized without prior close()"
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1612
      iget-object v0, p0, Landroid/content/ContentResolver$CursorWrapperInner;->this$0:Landroid/content/ContentResolver;
  
      iget-object v1, p0, Landroid/content/ContentResolver$CursorWrapperInner;->mContentProvider:Landroid/content/IContentProvider;
***************
*** 128,138 ****
--- 145,158 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1615
      :cond_1
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 1617
      return-void
  
+     .line 1615
      :catchall_0
      move-exception v0
  
