*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ContentUris.smali	2012-03-23 19:36:31.432254150 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ContentUris.smali.original	2012-03-23 19:34:45.162316321 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 70
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 19,24 ****
--- 20,26 ----
      .parameter "id"
  
      .prologue
+     .line 98
      invoke-static {p1, p2}, Ljava/lang/String;->valueOf(J)Ljava/lang/String;
  
      move-result-object v0
***************
*** 35,44 ****
--- 37,48 ----
      .parameter "contentUri"
  
      .prologue
+     .line 85
      invoke-virtual {p0}, Landroid/net/Uri;->getLastPathSegment()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 86
      .local v0, last:Ljava/lang/String;
      if-nez v0, :cond_0
  
***************
*** 61,66 ****
--- 65,71 ----
      .parameter "id"
  
      .prologue
+     .line 110
      invoke-virtual {p0}, Landroid/net/Uri;->buildUpon()Landroid/net/Uri$Builder;
  
      move-result-object v0
