*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ContentProviderOperation$Builder.smali	2012-03-23 19:36:27.770187373 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/ContentProviderOperation$Builder.smali.original	2012-03-23 19:34:45.125315646 -0500
***************
*** 51,73 ****
      .parameter "uri"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      if-nez p2, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "uri must not be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iput p1, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      iput-object p2, p0, Landroid/content/ContentProviderOperation$Builder;->mUri:Landroid/net/Uri;
  
      return-void
  .end method
  
--- 51,79 ----
      .parameter "uri"
  
      .prologue
+     .line 413
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 414
      if-nez p2, :cond_0
  
+     .line 415
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "uri must not be null"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 417
      :cond_0
      iput p1, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
+     .line 418
      iput-object p2, p0, Landroid/content/ContentProviderOperation$Builder;->mUri:Landroid/net/Uri;
  
+     .line 419
      return-void
  .end method
  
***************
*** 78,83 ****
--- 84,90 ----
      .parameter "x2"
  
      .prologue
+     .line 401
      invoke-direct {p0, p1, p2}, Landroid/content/ContentProviderOperation$Builder;-><init>(ILandroid/net/Uri;)V
  
      return-void
***************
*** 88,93 ****
--- 95,101 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      return v0
***************
*** 98,103 ****
--- 106,112 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mUri:Landroid/net/Uri;
  
      return-object v0
***************
*** 108,113 ****
--- 117,123 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      return-object v0
***************
*** 118,123 ****
--- 128,134 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelection:Ljava/lang/String;
  
      return-object v0
***************
*** 128,133 ****
--- 139,145 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgs:[Ljava/lang/String;
  
      return-object v0
***************
*** 138,143 ****
--- 150,156 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mExpectedCount:Ljava/lang/Integer;
  
      return-object v0
***************
*** 148,153 ****
--- 161,167 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgsBackReferences:Ljava/util/Map;
  
      return-object v0
***************
*** 158,163 ****
--- 172,178 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValuesBackReferences:Landroid/content/ContentValues;
  
      return-object v0
***************
*** 168,173 ****
--- 183,189 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget-boolean v0, p0, Landroid/content/ContentProviderOperation$Builder;->mYieldAllowed:Z
  
      return v0
***************
*** 179,190 ****
--- 195,208 ----
      .locals 2
  
      .prologue
+     .line 423
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      const/4 v1, 0x2
  
      if-ne v0, v1, :cond_2
  
+     .line 424
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      if-eqz v0, :cond_0
***************
*** 210,215 ****
--- 228,234 ----
  
      if-nez v0, :cond_2
  
+     .line 426
      :cond_1
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
***************
*** 219,224 ****
--- 238,244 ----
  
      throw v0
  
+     .line 429
      :cond_2
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
***************
*** 226,231 ****
--- 246,252 ----
  
      if-ne v0, v1, :cond_5
  
+     .line 430
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      if-eqz v0, :cond_3
***************
*** 256,261 ****
--- 277,283 ----
  
      if-nez v0, :cond_5
  
+     .line 433
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "Empty values"
***************
*** 264,269 ****
--- 286,292 ----
  
      throw v0
  
+     .line 436
      :cond_5
      new-instance v0, Landroid/content/ContentProviderOperation;
  
***************
*** 279,284 ****
--- 302,308 ----
      .parameter "count"
  
      .prologue
+     .line 585
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      const/4 v1, 0x2
***************
*** 297,310 ****
  
      if-eq v0, v1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "only updates, deletes, and asserts can have expected counts"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      invoke-static {p1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 321,336 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 586
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "only updates, deletes, and asserts can have expected counts"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 589
      :cond_0
      invoke-static {p1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 312,317 ****
--- 338,344 ----
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mExpectedCount:Ljava/lang/Integer;
  
+     .line 590
      return-object p0
  .end method
  
***************
*** 323,328 ****
--- 350,356 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 564
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      const/4 v1, 0x2
***************
*** 341,366 ****
  
      if-eq v0, v1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "only updates, deletes, and asserts can have selections"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iput-object p1, p0, Landroid/content/ContentProviderOperation$Builder;->mSelection:Ljava/lang/String;
  
      if-nez p2, :cond_1
  
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgs:[Ljava/lang/String;
  
      :goto_0
      return-object p0
  
      :cond_1
      array-length v0, p2
  
--- 369,400 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 565
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "only updates, deletes, and asserts can have selections"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 568
      :cond_0
      iput-object p1, p0, Landroid/content/ContentProviderOperation$Builder;->mSelection:Ljava/lang/String;
  
+     .line 569
      if-nez p2, :cond_1
  
+     .line 570
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgs:[Ljava/lang/String;
  
+     .line 575
      :goto_0
      return-object p0
  
+     .line 572
      :cond_1
      array-length v0, p2
  
***************
*** 368,373 ****
--- 402,408 ----
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgs:[Ljava/lang/String;
  
+     .line 573
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgs:[Ljava/lang/String;
  
      array-length v1, p2
***************
*** 383,388 ****
--- 418,424 ----
      .parameter "previousResult"
  
      .prologue
+     .line 483
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      const/4 v1, 0x2
***************
*** 401,425 ****
  
      if-eq v0, v1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "only updates, deletes, and asserts can have selection back-references"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgsBackReferences:Ljava/util/Map;
  
      if-nez v0, :cond_1
  
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgsBackReferences:Ljava/util/Map;
  
      :cond_1
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgsBackReferences:Ljava/util/Map;
  
--- 437,465 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 484
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "only updates, deletes, and asserts can have selection back-references"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 487
      :cond_0
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgsBackReferences:Ljava/util/Map;
  
      if-nez v0, :cond_1
  
+     .line 488
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgsBackReferences:Ljava/util/Map;
  
+     .line 490
      :cond_1
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mSelectionArgsBackReferences:Ljava/util/Map;
  
***************
*** 433,438 ****
--- 473,479 ----
  
      invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 491
      return-object p0
  .end method
  
***************
*** 442,447 ****
--- 483,489 ----
      .parameter "value"
  
      .prologue
+     .line 523
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      const/4 v1, 0x1
***************
*** 460,501 ****
  
      if-eq v0, v1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "only inserts and updates can have values"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      if-nez v0, :cond_1
  
      new-instance v0, Landroid/content/ContentValues;
  
      invoke-direct {v0}, Landroid/content/ContentValues;-><init>()V
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      :cond_1
      if-nez p2, :cond_2
  
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      invoke-virtual {v0, p1}, Landroid/content/ContentValues;->putNull(Ljava/lang/String;)V
  
      .end local p2
      :goto_0
      return-object p0
  
      .restart local p2
      :cond_2
      instance-of v0, p2, Ljava/lang/String;
  
      if-eqz v0, :cond_3
  
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, Ljava/lang/String;
--- 502,551 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 524
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "only inserts and updates can have values"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 526
      :cond_0
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      if-nez v0, :cond_1
  
+     .line 527
      new-instance v0, Landroid/content/ContentValues;
  
      invoke-direct {v0}, Landroid/content/ContentValues;-><init>()V
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
+     .line 529
      :cond_1
      if-nez p2, :cond_2
  
+     .line 530
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      invoke-virtual {v0, p1}, Landroid/content/ContentValues;->putNull(Ljava/lang/String;)V
  
+     .line 552
      .end local p2
      :goto_0
      return-object p0
  
+     .line 531
      .restart local p2
      :cond_2
      instance-of v0, p2, Ljava/lang/String;
  
      if-eqz v0, :cond_3
  
+     .line 532
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, Ljava/lang/String;
***************
*** 505,516 ****
--- 555,568 ----
  
      goto :goto_0
  
+     .line 533
      .restart local p2
      :cond_3
      instance-of v0, p2, Ljava/lang/Byte;
  
      if-eqz v0, :cond_4
  
+     .line 534
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, Ljava/lang/Byte;
***************
*** 520,531 ****
--- 572,585 ----
  
      goto :goto_0
  
+     .line 535
      .restart local p2
      :cond_4
      instance-of v0, p2, Ljava/lang/Short;
  
      if-eqz v0, :cond_5
  
+     .line 536
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, Ljava/lang/Short;
***************
*** 535,546 ****
--- 589,602 ----
  
      goto :goto_0
  
+     .line 537
      .restart local p2
      :cond_5
      instance-of v0, p2, Ljava/lang/Integer;
  
      if-eqz v0, :cond_6
  
+     .line 538
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, Ljava/lang/Integer;
***************
*** 550,561 ****
--- 606,619 ----
  
      goto :goto_0
  
+     .line 539
      .restart local p2
      :cond_6
      instance-of v0, p2, Ljava/lang/Long;
  
      if-eqz v0, :cond_7
  
+     .line 540
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, Ljava/lang/Long;
***************
*** 565,576 ****
--- 623,636 ----
  
      goto :goto_0
  
+     .line 541
      .restart local p2
      :cond_7
      instance-of v0, p2, Ljava/lang/Float;
  
      if-eqz v0, :cond_8
  
+     .line 542
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, Ljava/lang/Float;
***************
*** 580,591 ****
--- 640,653 ----
  
      goto :goto_0
  
+     .line 543
      .restart local p2
      :cond_8
      instance-of v0, p2, Ljava/lang/Double;
  
      if-eqz v0, :cond_9
  
+     .line 544
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, Ljava/lang/Double;
***************
*** 595,606 ****
--- 657,670 ----
  
      goto :goto_0
  
+     .line 545
      .restart local p2
      :cond_9
      instance-of v0, p2, Ljava/lang/Boolean;
  
      if-eqz v0, :cond_a
  
+     .line 546
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, Ljava/lang/Boolean;
***************
*** 610,621 ****
--- 674,687 ----
  
      goto :goto_0
  
+     .line 547
      .restart local p2
      :cond_a
      instance-of v0, p2, [B
  
      if-eqz v0, :cond_b
  
+     .line 548
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      check-cast p2, [B
***************
*** 627,632 ****
--- 693,699 ----
  
      goto :goto_0
  
+     .line 550
      .restart local p2
      :cond_b
      new-instance v0, Ljava/lang/IllegalArgumentException;
***************
*** 668,673 ****
--- 735,741 ----
      .parameter "previousResult"
  
      .prologue
+     .line 465
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      const/4 v1, 0x1
***************
*** 686,710 ****
  
      if-eq v0, v1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "only inserts, updates, and asserts can have value back-references"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValuesBackReferences:Landroid/content/ContentValues;
  
      if-nez v0, :cond_1
  
      new-instance v0, Landroid/content/ContentValues;
  
      invoke-direct {v0}, Landroid/content/ContentValues;-><init>()V
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValuesBackReferences:Landroid/content/ContentValues;
  
      :cond_1
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValuesBackReferences:Landroid/content/ContentValues;
  
--- 754,782 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 466
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "only inserts, updates, and asserts can have value back-references"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 469
      :cond_0
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValuesBackReferences:Landroid/content/ContentValues;
  
      if-nez v0, :cond_1
  
+     .line 470
      new-instance v0, Landroid/content/ContentValues;
  
      invoke-direct {v0}, Landroid/content/ContentValues;-><init>()V
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValuesBackReferences:Landroid/content/ContentValues;
  
+     .line 472
      :cond_1
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValuesBackReferences:Landroid/content/ContentValues;
  
***************
*** 714,719 ****
--- 786,792 ----
  
      invoke-virtual {v0, p1, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 473
      return-object p0
  .end method
  
***************
*** 722,727 ****
--- 795,801 ----
      .parameter "backReferences"
  
      .prologue
+     .line 449
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      const/4 v1, 0x1
***************
*** 740,756 ****
  
      if-eq v0, v1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "only inserts, updates, and asserts can have value back-references"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iput-object p1, p0, Landroid/content/ContentProviderOperation$Builder;->mValuesBackReferences:Landroid/content/ContentValues;
  
      return-object p0
  .end method
  
--- 814,833 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 450
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "only inserts, updates, and asserts can have value back-references"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 453
      :cond_0
      iput-object p1, p0, Landroid/content/ContentProviderOperation$Builder;->mValuesBackReferences:Landroid/content/ContentValues;
  
+     .line 454
      return-object p0
  .end method
  
***************
*** 759,764 ****
--- 836,842 ----
      .parameter "values"
  
      .prologue
+     .line 502
      iget v0, p0, Landroid/content/ContentProviderOperation$Builder;->mType:I
  
      const/4 v1, 0x1
***************
*** 777,806 ****
  
      if-eq v0, v1, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "only inserts, updates, and asserts can have values"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      if-nez v0, :cond_1
  
      new-instance v0, Landroid/content/ContentValues;
  
      invoke-direct {v0}, Landroid/content/ContentValues;-><init>()V
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      :cond_1
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      invoke-virtual {v0, p1}, Landroid/content/ContentValues;->putAll(Landroid/content/ContentValues;)V
  
      return-object p0
  .end method
  
--- 855,889 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 503
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "only inserts, updates, and asserts can have values"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 506
      :cond_0
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      if-nez v0, :cond_1
  
+     .line 507
      new-instance v0, Landroid/content/ContentValues;
  
      invoke-direct {v0}, Landroid/content/ContentValues;-><init>()V
  
      iput-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
+     .line 509
      :cond_1
      iget-object v0, p0, Landroid/content/ContentProviderOperation$Builder;->mValues:Landroid/content/ContentValues;
  
      invoke-virtual {v0, p1}, Landroid/content/ContentValues;->putAll(Landroid/content/ContentValues;)V
  
+     .line 510
      return-object p0
  .end method
  
***************
*** 809,815 ****
--- 892,900 ----
      .parameter "yieldAllowed"
  
      .prologue
+     .line 594
      iput-boolean p1, p0, Landroid/content/ContentProviderOperation$Builder;->mYieldAllowed:Z
  
+     .line 595
      return-object p0
  .end method
