*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/IntentSender.smali	2012-03-23 19:36:26.710168043 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/IntentSender.smali.original	2012-03-23 19:34:45.114315445 -0500
***************
*** 38,43 ****
--- 38,44 ----
      .locals 1
  
      .prologue
+     .line 263
      new-instance v0, Landroid/content/IntentSender$1;
  
      invoke-direct {v0}, Landroid/content/IntentSender$1;-><init>()V
***************
*** 52,61 ****
--- 53,65 ----
      .parameter "target"
  
      .prologue
+     .line 310
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 311
      iput-object p1, p0, Landroid/content/IntentSender;->mTarget:Landroid/content/IIntentSender;
  
+     .line 312
      return-void
  .end method
  
***************
*** 64,77 ****
--- 68,84 ----
      .parameter "target"
  
      .prologue
+     .line 315
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 316
      invoke-static {p1}, Landroid/content/IIntentSender$Stub;->asInterface(Landroid/os/IBinder;)Landroid/content/IIntentSender;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/IntentSender;->mTarget:Landroid/content/IIntentSender;
  
+     .line 317
      return-void
  .end method
  
***************
*** 80,89 ****
--- 87,98 ----
      .parameter "in"
  
      .prologue
+     .line 300
      invoke-virtual {p0}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 301
      .local v0, b:Landroid/os/IBinder;
      if-eqz v0, :cond_0
  
***************
*** 106,111 ****
--- 115,121 ----
      .parameter "out"
  
      .prologue
+     .line 285
      if-eqz p0, :cond_0
  
      iget-object v0, p0, Landroid/content/IntentSender;->mTarget:Landroid/content/IIntentSender;
***************
*** 117,124 ****
--- 127,136 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 287
      return-void
  
+     .line 285
      :cond_0
      const/4 v0, 0x0
  
***************
*** 131,136 ****
--- 143,149 ----
      .locals 1
  
      .prologue
+     .line 256
      const/4 v0, 0x0
  
      return v0
***************
*** 141,150 ****
--- 154,165 ----
      .parameter "otherObj"
  
      .prologue
+     .line 232
      instance-of v0, p1, Landroid/content/IntentSender;
  
      if-eqz v0, :cond_0
  
+     .line 233
      iget-object v0, p0, Landroid/content/IntentSender;->mTarget:Landroid/content/IIntentSender;
  
      invoke-interface {v0}, Landroid/content/IIntentSender;->asBinder()Landroid/os/IBinder;
***************
*** 164,169 ****
--- 179,185 ----
  
      move-result v0
  
+     .line 236
      :goto_0
      return v0
  
***************
*** 178,183 ****
--- 194,200 ----
      .locals 1
  
      .prologue
+     .line 306
      iget-object v0, p0, Landroid/content/IntentSender;->mTarget:Landroid/content/IIntentSender;
  
      return-object v0
***************
*** 187,192 ****
--- 204,210 ----
      .locals 3
  
      .prologue
+     .line 217
      :try_start_0
      invoke-static {}, Landroid/app/ActivityManagerNative;->getDefault()Landroid/app/IActivityManager;
  
***************
*** 200,211 ****
--- 218,232 ----
  
      move-result-object v1
  
+     .line 221
      :goto_0
      return-object v1
  
+     .line 219
      :catch_0
      move-exception v0
  
+     .line 221
      .local v0, e:Landroid/os/RemoteException;
      const/4 v1, 0x0
  
***************
*** 216,221 ****
--- 237,243 ----
      .locals 1
  
      .prologue
+     .line 241
      iget-object v0, p0, Landroid/content/IntentSender;->mTarget:Landroid/content/IIntentSender;
  
      invoke-interface {v0}, Landroid/content/IIntentSender;->asBinder()Landroid/os/IBinder;
***************
*** 243,248 ****
--- 265,271 ----
      .end annotation
  
      .prologue
+     .line 157
      const/4 v6, 0x0
  
      move-object v0, p0
***************
*** 259,264 ****
--- 282,288 ----
  
      invoke-virtual/range {v0 .. v6}, Landroid/content/IntentSender;->sendIntent(Landroid/content/Context;ILandroid/content/Intent;Landroid/content/IntentSender$OnFinished;Landroid/os/Handler;Ljava/lang/String;)V
  
+     .line 158
      return-void
  .end method
  
***************
*** 279,284 ****
--- 303,309 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 190
      if-eqz p3, :cond_1
  
      :try_start_0
***************
*** 290,295 ****
--- 315,321 ----
  
      move-result-object v3
  
+     .line 193
      .local v3, resolvedType:Ljava/lang/String;
      :goto_0
      iget-object v0, p0, Landroid/content/IntentSender;->mTarget:Landroid/content/IIntentSender;
***************
*** 311,319 ****
--- 337,347 ----
  
      move-result v7
  
+     .line 198
      .local v7, res:I
      if-gez v7, :cond_2
  
+     .line 199
      new-instance v0, Landroid/content/IntentSender$SendIntentException;
  
      invoke-direct {v0}, Landroid/content/IntentSender$SendIntentException;-><init>()V
***************
*** 322,332 ****
--- 350,362 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 201
      .end local v3           #resolvedType:Ljava/lang/String;
      .end local v7           #res:I
      :catch_0
      move-exception v6
  
+     .line 202
      .local v6, e:Landroid/os/RemoteException;
      new-instance v0, Landroid/content/IntentSender$SendIntentException;
  
***************
*** 338,345 ****
--- 368,377 ----
      :cond_1
      move-object v3, v4
  
+     .line 190
      goto :goto_0
  
+     .line 204
      .restart local v3       #resolvedType:Ljava/lang/String;
      .restart local v7       #res:I
      :cond_2
***************
*** 350,366 ****
--- 382,401 ----
      .locals 2
  
      .prologue
+     .line 246
      new-instance v0, Ljava/lang/StringBuilder;
  
      const/16 v1, 0x80
  
      invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(I)V
  
+     .line 247
      .local v0, sb:Ljava/lang/StringBuilder;
      const-string v1, "IntentSender{"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 248
      invoke-static {p0}, Ljava/lang/System;->identityHashCode(Ljava/lang/Object;)I
  
      move-result v1
***************
*** 371,380 ****
--- 406,417 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 249
      const-string v1, ": "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 250
      iget-object v1, p0, Landroid/content/IntentSender;->mTarget:Landroid/content/IIntentSender;
  
      if-eqz v1, :cond_0
***************
*** 388,403 ****
--- 425,443 ----
      :goto_0
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
  
+     .line 251
      const/16 v1, 0x7d
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 252
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
  
      return-object v1
  
+     .line 250
      :cond_0
      const/4 v1, 0x0
  
***************
*** 410,415 ****
--- 450,456 ----
      .parameter "flags"
  
      .prologue
+     .line 260
      iget-object v0, p0, Landroid/content/IntentSender;->mTarget:Landroid/content/IIntentSender;
  
      invoke-interface {v0}, Landroid/content/IIntentSender;->asBinder()Landroid/os/IBinder;
***************
*** 418,422 ****
--- 459,464 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 261
      return-void
  .end method
