*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/MutableContextWrapper.smali	2012-03-23 19:36:24.506127852 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/MutableContextWrapper.smali.original	2012-03-23 19:34:45.095315099 -0500
***************
*** 9,16 ****
--- 9,18 ----
      .parameter "base"
  
      .prologue
+     .line 25
      invoke-direct {p0, p1}, Landroid/content/ContextWrapper;-><init>(Landroid/content/Context;)V
  
+     .line 26
      return-void
  .end method
  
***************
*** 21,27 ****
--- 23,31 ----
      .parameter "base"
  
      .prologue
+     .line 36
      iput-object p1, p0, Landroid/content/MutableContextWrapper;->mBase:Landroid/content/Context;
  
+     .line 37
      return-void
  .end method
