*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/PeriodicSync.smali	2012-03-23 19:36:27.689185895 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/PeriodicSync.smali.original	2012-03-23 19:34:45.125315646 -0500
***************
*** 34,39 ****
--- 34,40 ----
      .locals 1
  
      .prologue
+     .line 57
      new-instance v0, Landroid/content/PeriodicSync$1;
  
      invoke-direct {v0}, Landroid/content/PeriodicSync$1;-><init>()V
***************
*** 51,70 ****
--- 52,77 ----
      .parameter "period"
  
      .prologue
+     .line 39
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 40
      iput-object p1, p0, Landroid/content/PeriodicSync;->account:Landroid/accounts/Account;
  
+     .line 41
      iput-object p2, p0, Landroid/content/PeriodicSync;->authority:Ljava/lang/String;
  
+     .line 42
      new-instance v0, Landroid/os/Bundle;
  
      invoke-direct {v0, p3}, Landroid/os/Bundle;-><init>(Landroid/os/Bundle;)V
  
      iput-object v0, p0, Landroid/content/PeriodicSync;->extras:Landroid/os/Bundle;
  
+     .line 43
      iput-wide p4, p0, Landroid/content/PeriodicSync;->period:J
  
+     .line 44
      return-void
  .end method
  
***************
*** 74,79 ****
--- 81,87 ----
      .locals 1
  
      .prologue
+     .line 47
      const/4 v0, 0x0
  
      return v0
***************
*** 88,99 ****
--- 96,110 ----
  
      const/4 v2, 0x0
  
+     .line 69
      if-ne p1, p0, :cond_1
  
+     .line 79
      :cond_0
      :goto_0
      return v1
  
+     .line 73
      :cond_1
      instance-of v3, p1, Landroid/content/PeriodicSync;
  
***************
*** 101,113 ****
--- 112,127 ----
  
      move v1, v2
  
+     .line 74
      goto :goto_0
  
      :cond_2
      move-object v0, p1
  
+     .line 77
      check-cast v0, Landroid/content/PeriodicSync;
  
+     .line 79
      .local v0, other:Landroid/content/PeriodicSync;
      iget-object v3, p0, Landroid/content/PeriodicSync;->account:Landroid/accounts/Account;
  
***************
*** 159,179 ****
--- 173,198 ----
      .parameter "flags"
  
      .prologue
+     .line 51
      iget-object v0, p0, Landroid/content/PeriodicSync;->account:Landroid/accounts/Account;
  
      invoke-virtual {v0, p1, p2}, Landroid/accounts/Account;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 52
      iget-object v0, p0, Landroid/content/PeriodicSync;->authority:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 53
      iget-object v0, p0, Landroid/content/PeriodicSync;->extras:Landroid/os/Bundle;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeBundle(Landroid/os/Bundle;)V
  
+     .line 54
      iget-wide v0, p0, Landroid/content/PeriodicSync;->period:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 55
      return-void
  .end method
