*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncAdapterType.smali	2012-03-23 19:36:27.035173970 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncAdapterType.smali.original	2012-03-23 19:34:45.118315519 -0500
***************
*** 42,47 ****
--- 42,48 ----
      .locals 1
  
      .prologue
+     .line 217
      new-instance v0, Landroid/content/SyncAdapterType$1;
  
      invoke-direct {v0}, Landroid/content/SyncAdapterType$1;-><init>()V
***************
*** 60,65 ****
--- 61,67 ----
  
      const/4 v6, 0x0
  
+     .line 207
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v1
***************
*** 112,122 ****
--- 114,126 ----
  
      invoke-direct/range {v0 .. v7}, Landroid/content/SyncAdapterType;-><init>(Ljava/lang/String;Ljava/lang/String;ZZZZLjava/lang/String;)V
  
+     .line 215
      return-void
  
      :cond_1
      move v3, v6
  
+     .line 207
      goto :goto_0
  
      :cond_2
***************
*** 140,160 ****
  
      const/4 v1, 0x1
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-static {p1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "the authority must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 144,167 ----
  
      const/4 v1, 0x1
  
+     .line 77
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 78
      invoke-static {p1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 79
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "the authority must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 172,177 ****
--- 179,185 ----
  
      throw v0
  
+     .line 81
      :cond_0
      invoke-static {p2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
***************
*** 179,191 ****
  
      if-eqz v0, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "the accountType must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 187,200 ----
  
      if-eqz v0, :cond_1
  
+     .line 82
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "the accountType must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 203,227 ****
--- 212,245 ----
  
      throw v0
  
+     .line 84
      :cond_1
      iput-object p1, p0, Landroid/content/SyncAdapterType;->authority:Ljava/lang/String;
  
+     .line 85
      iput-object p2, p0, Landroid/content/SyncAdapterType;->accountType:Ljava/lang/String;
  
+     .line 86
      iput-boolean v1, p0, Landroid/content/SyncAdapterType;->userVisible:Z
  
+     .line 87
      iput-boolean v1, p0, Landroid/content/SyncAdapterType;->supportsUploading:Z
  
+     .line 88
      iput-boolean v2, p0, Landroid/content/SyncAdapterType;->isAlwaysSyncable:Z
  
+     .line 89
      iput-boolean v2, p0, Landroid/content/SyncAdapterType;->allowParallelSyncs:Z
  
+     .line 90
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/SyncAdapterType;->settingsActivity:Ljava/lang/String;
  
+     .line 91
      iput-boolean v1, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
+     .line 92
      return-void
  .end method
  
***************
*** 235,255 ****
      .prologue
      const/4 v1, 0x0
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-static {p1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "the authority must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 253,276 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 38
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 39
      invoke-static {p1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 40
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "the authority must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 267,272 ****
--- 288,294 ----
  
      throw v0
  
+     .line 42
      :cond_0
      invoke-static {p2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
***************
*** 274,286 ****
  
      if-eqz v0, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "the accountType must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 296,309 ----
  
      if-eqz v0, :cond_1
  
+     .line 43
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "the accountType must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 298,322 ****
--- 321,354 ----
  
      throw v0
  
+     .line 45
      :cond_1
      iput-object p1, p0, Landroid/content/SyncAdapterType;->authority:Ljava/lang/String;
  
+     .line 46
      iput-object p2, p0, Landroid/content/SyncAdapterType;->accountType:Ljava/lang/String;
  
+     .line 47
      iput-boolean p3, p0, Landroid/content/SyncAdapterType;->userVisible:Z
  
+     .line 48
      iput-boolean p4, p0, Landroid/content/SyncAdapterType;->supportsUploading:Z
  
+     .line 49
      iput-boolean v1, p0, Landroid/content/SyncAdapterType;->isAlwaysSyncable:Z
  
+     .line 50
      iput-boolean v1, p0, Landroid/content/SyncAdapterType;->allowParallelSyncs:Z
  
+     .line 51
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/SyncAdapterType;->settingsActivity:Ljava/lang/String;
  
+     .line 52
      iput-boolean v1, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
+     .line 53
      return-void
  .end method
  
***************
*** 331,351 ****
      .parameter "settingsActivity"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-static {p1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "the authority must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 363,386 ----
      .parameter "settingsActivity"
  
      .prologue
+     .line 60
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 61
      invoke-static {p1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 62
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "the authority must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 363,368 ****
--- 398,404 ----
  
      throw v0
  
+     .line 64
      :cond_0
      invoke-static {p2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
***************
*** 370,382 ****
  
      if-eqz v0, :cond_1
  
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "the accountType must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 406,419 ----
  
      if-eqz v0, :cond_1
  
+     .line 65
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "the accountType must not be empty: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 394,418 ****
--- 431,464 ----
  
      throw v0
  
+     .line 67
      :cond_1
      iput-object p1, p0, Landroid/content/SyncAdapterType;->authority:Ljava/lang/String;
  
+     .line 68
      iput-object p2, p0, Landroid/content/SyncAdapterType;->accountType:Ljava/lang/String;
  
+     .line 69
      iput-boolean p3, p0, Landroid/content/SyncAdapterType;->userVisible:Z
  
+     .line 70
      iput-boolean p4, p0, Landroid/content/SyncAdapterType;->supportsUploading:Z
  
+     .line 71
      iput-boolean p5, p0, Landroid/content/SyncAdapterType;->isAlwaysSyncable:Z
  
+     .line 72
      iput-boolean p6, p0, Landroid/content/SyncAdapterType;->allowParallelSyncs:Z
  
+     .line 73
      iput-object p7, p0, Landroid/content/SyncAdapterType;->settingsActivity:Ljava/lang/String;
  
+     .line 74
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
+     .line 75
      return-void
  .end method
  
***************
*** 422,427 ****
--- 468,474 ----
      .parameter "accountType"
  
      .prologue
+     .line 152
      new-instance v0, Landroid/content/SyncAdapterType;
  
      invoke-direct {v0, p0, p1}, Landroid/content/SyncAdapterType;-><init>(Ljava/lang/String;Ljava/lang/String;)V
***************
*** 435,452 ****
      .locals 2
  
      .prologue
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->allowParallelSyncs:Z
  
--- 482,502 ----
      .locals 2
  
      .prologue
+     .line 116
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
+     .line 117
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string/jumbo v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 120
      :cond_0
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->allowParallelSyncs:Z
  
***************
*** 457,462 ****
--- 507,513 ----
      .locals 1
  
      .prologue
+     .line 189
      const/4 v0, 0x0
  
      return v0
***************
*** 471,482 ****
--- 522,536 ----
  
      const/4 v2, 0x0
  
+     .line 156
      if-ne p1, p0, :cond_1
  
+     .line 160
      :cond_0
      :goto_0
      return v1
  
+     .line 157
      :cond_1
      instance-of v3, p1, Landroid/content/SyncAdapterType;
  
***************
*** 489,496 ****
--- 543,552 ----
      :cond_2
      move-object v0, p1
  
+     .line 158
      check-cast v0, Landroid/content/SyncAdapterType;
  
+     .line 160
      .local v0, other:Landroid/content/SyncAdapterType;
      iget-object v3, p0, Landroid/content/SyncAdapterType;->authority:Ljava/lang/String;
  
***************
*** 522,539 ****
      .locals 2
  
      .prologue
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget-object v0, p0, Landroid/content/SyncAdapterType;->settingsActivity:Ljava/lang/String;
  
--- 578,598 ----
      .locals 2
  
      .prologue
+     .line 144
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
+     .line 145
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string/jumbo v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 148
      :cond_0
      iget-object v0, p0, Landroid/content/SyncAdapterType;->settingsActivity:Ljava/lang/String;
  
***************
*** 544,551 ****
--- 603,612 ----
      .locals 3
  
      .prologue
+     .line 164
      const/16 v0, 0x11
  
+     .line 165
      .local v0, result:I
      iget-object v1, p0, Landroid/content/SyncAdapterType;->authority:Ljava/lang/String;
  
***************
*** 555,560 ****
--- 616,622 ----
  
      add-int/lit16 v0, v1, 0x20f
  
+     .line 166
      mul-int/lit8 v1, v0, 0x1f
  
      iget-object v2, p0, Landroid/content/SyncAdapterType;->accountType:Ljava/lang/String;
***************
*** 565,570 ****
--- 627,633 ----
  
      add-int v0, v1, v2
  
+     .line 168
      return v0
  .end method
  
***************
*** 572,589 ****
      .locals 2
  
      .prologue
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isAlwaysSyncable:Z
  
--- 635,655 ----
      .locals 2
  
      .prologue
+     .line 132
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
+     .line 133
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string/jumbo v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 136
      :cond_0
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isAlwaysSyncable:Z
  
***************
*** 594,611 ****
      .locals 2
  
      .prologue
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->userVisible:Z
  
--- 660,680 ----
      .locals 2
  
      .prologue
+     .line 103
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
+     .line 104
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string/jumbo v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 107
      :cond_0
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->userVisible:Z
  
***************
*** 616,633 ****
      .locals 2
  
      .prologue
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->supportsUploading:Z
  
--- 685,705 ----
      .locals 2
  
      .prologue
+     .line 95
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
+     .line 96
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string/jumbo v1, "this method is not allowed to be called when this is a key"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 99
      :cond_0
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->supportsUploading:Z
  
***************
*** 638,647 ****
--- 710,721 ----
      .locals 2
  
      .prologue
+     .line 172
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
+     .line 173
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 670,676 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 744,750 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 680,685 ****
--- 754,760 ----
  
      move-result-object v0
  
+     .line 177
      :goto_0
      return-object v0
  
***************
*** 772,778 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 847,853 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 795,804 ****
--- 870,881 ----
  
      const/4 v2, 0x0
  
+     .line 193
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isKey:Z
  
      if-eqz v0, :cond_0
  
+     .line 194
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "keys aren\'t parcelable"
***************
*** 807,821 ****
--- 884,901 ----
  
      throw v0
  
+     .line 197
      :cond_0
      iget-object v0, p0, Landroid/content/SyncAdapterType;->authority:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 198
      iget-object v0, p0, Landroid/content/SyncAdapterType;->accountType:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 199
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->userVisible:Z
  
      if-eqz v0, :cond_1
***************
*** 825,830 ****
--- 905,911 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 200
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->supportsUploading:Z
  
      if-eqz v0, :cond_2
***************
*** 834,839 ****
--- 915,921 ----
      :goto_1
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 201
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->isAlwaysSyncable:Z
  
      if-eqz v0, :cond_3
***************
*** 843,848 ****
--- 925,931 ----
      :goto_2
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 202
      iget-boolean v0, p0, Landroid/content/SyncAdapterType;->allowParallelSyncs:Z
  
      if-eqz v0, :cond_4
***************
*** 850,878 ****
--- 933,967 ----
      :goto_3
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 203
      iget-object v0, p0, Landroid/content/SyncAdapterType;->settingsActivity:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 204
      return-void
  
      :cond_1
      move v0, v2
  
+     .line 199
      goto :goto_0
  
      :cond_2
      move v0, v2
  
+     .line 200
      goto :goto_1
  
      :cond_3
      move v0, v2
  
+     .line 201
      goto :goto_2
  
      :cond_4
      move v1, v2
  
+     .line 202
      goto :goto_3
  .end method
