*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncStats.smali	2012-03-23 19:36:27.112175374 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncStats.smali.original	2012-03-23 19:34:45.119315537 -0500
***************
*** 44,49 ****
--- 44,50 ----
      .locals 1
  
      .prologue
+     .line 169
      new-instance v0, Landroid/content/SyncStats$1;
  
      invoke-direct {v0}, Landroid/content/SyncStats$1;-><init>()V
***************
*** 59,84 ****
--- 60,96 ----
      .prologue
      const-wide/16 v0, 0x0
  
+     .line 96
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 97
      iput-wide v0, p0, Landroid/content/SyncStats;->numAuthExceptions:J
  
+     .line 98
      iput-wide v0, p0, Landroid/content/SyncStats;->numIoExceptions:J
  
+     .line 99
      iput-wide v0, p0, Landroid/content/SyncStats;->numParseExceptions:J
  
+     .line 100
      iput-wide v0, p0, Landroid/content/SyncStats;->numConflictDetectedExceptions:J
  
+     .line 101
      iput-wide v0, p0, Landroid/content/SyncStats;->numInserts:J
  
+     .line 102
      iput-wide v0, p0, Landroid/content/SyncStats;->numUpdates:J
  
+     .line 103
      iput-wide v0, p0, Landroid/content/SyncStats;->numDeletes:J
  
+     .line 104
      iput-wide v0, p0, Landroid/content/SyncStats;->numEntries:J
  
+     .line 105
      iput-wide v0, p0, Landroid/content/SyncStats;->numSkippedEntries:J
  
+     .line 106
      return-void
  .end method
  
***************
*** 87,148 ****
--- 99,171 ----
      .parameter "in"
  
      .prologue
+     .line 108
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 109
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncStats;->numAuthExceptions:J
  
+     .line 110
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncStats;->numIoExceptions:J
  
+     .line 111
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncStats;->numParseExceptions:J
  
+     .line 112
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncStats;->numConflictDetectedExceptions:J
  
+     .line 113
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncStats;->numInserts:J
  
+     .line 114
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncStats;->numUpdates:J
  
+     .line 115
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncStats;->numDeletes:J
  
+     .line 116
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncStats;->numEntries:J
  
+     .line 117
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncStats;->numSkippedEntries:J
  
+     .line 118
      return-void
  .end method
  
***************
*** 154,177 ****
--- 177,210 ----
      .prologue
      const-wide/16 v0, 0x0
  
+     .line 142
      iput-wide v0, p0, Landroid/content/SyncStats;->numAuthExceptions:J
  
+     .line 143
      iput-wide v0, p0, Landroid/content/SyncStats;->numIoExceptions:J
  
+     .line 144
      iput-wide v0, p0, Landroid/content/SyncStats;->numParseExceptions:J
  
+     .line 145
      iput-wide v0, p0, Landroid/content/SyncStats;->numConflictDetectedExceptions:J
  
+     .line 146
      iput-wide v0, p0, Landroid/content/SyncStats;->numInserts:J
  
+     .line 147
      iput-wide v0, p0, Landroid/content/SyncStats;->numUpdates:J
  
+     .line 148
      iput-wide v0, p0, Landroid/content/SyncStats;->numDeletes:J
  
+     .line 149
      iput-wide v0, p0, Landroid/content/SyncStats;->numEntries:J
  
+     .line 150
      iput-wide v0, p0, Landroid/content/SyncStats;->numSkippedEntries:J
  
+     .line 151
      return-void
  .end method
  
***************
*** 179,184 ****
--- 212,218 ----
      .locals 1
  
      .prologue
+     .line 154
      const/4 v0, 0x0
  
      return v0
***************
*** 190,204 ****
--- 224,241 ----
      .prologue
      const-wide/16 v4, 0x0
  
+     .line 122
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 123
      .local v0, sb:Ljava/lang/StringBuilder;
      const-string v1, " stats ["
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 124
      iget-wide v1, p0, Landroid/content/SyncStats;->numAuthExceptions:J
  
      cmp-long v1, v1, v4
***************
*** 215,220 ****
--- 252,258 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 125
      :cond_0
      iget-wide v1, p0, Landroid/content/SyncStats;->numIoExceptions:J
  
***************
*** 232,237 ****
--- 270,276 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 126
      :cond_1
      iget-wide v1, p0, Landroid/content/SyncStats;->numParseExceptions:J
  
***************
*** 249,254 ****
--- 288,294 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 127
      :cond_2
      iget-wide v1, p0, Landroid/content/SyncStats;->numConflictDetectedExceptions:J
  
***************
*** 256,261 ****
--- 296,302 ----
  
      if-lez v1, :cond_3
  
+     .line 128
      const-string v1, " numConflictDetectedExceptions: "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 266,271 ****
--- 307,313 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 129
      :cond_3
      iget-wide v1, p0, Landroid/content/SyncStats;->numInserts:J
  
***************
*** 283,288 ****
--- 325,331 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 130
      :cond_4
      iget-wide v1, p0, Landroid/content/SyncStats;->numUpdates:J
  
***************
*** 300,305 ****
--- 343,349 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 131
      :cond_5
      iget-wide v1, p0, Landroid/content/SyncStats;->numDeletes:J
  
***************
*** 317,322 ****
--- 361,367 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 132
      :cond_6
      iget-wide v1, p0, Landroid/content/SyncStats;->numEntries:J
  
***************
*** 334,339 ****
--- 379,385 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 133
      :cond_7
      iget-wide v1, p0, Landroid/content/SyncStats;->numSkippedEntries:J
  
***************
*** 351,361 ****
--- 397,409 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 134
      :cond_8
      const-string v1, "]"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 135
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
***************
*** 369,409 ****
--- 417,467 ----
      .parameter "flags"
  
      .prologue
+     .line 158
      iget-wide v0, p0, Landroid/content/SyncStats;->numAuthExceptions:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 159
      iget-wide v0, p0, Landroid/content/SyncStats;->numIoExceptions:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 160
      iget-wide v0, p0, Landroid/content/SyncStats;->numParseExceptions:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 161
      iget-wide v0, p0, Landroid/content/SyncStats;->numConflictDetectedExceptions:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 162
      iget-wide v0, p0, Landroid/content/SyncStats;->numInserts:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 163
      iget-wide v0, p0, Landroid/content/SyncStats;->numUpdates:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 164
      iget-wide v0, p0, Landroid/content/SyncStats;->numDeletes:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 165
      iget-wide v0, p0, Landroid/content/SyncStats;->numEntries:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 166
      iget-wide v0, p0, Landroid/content/SyncStats;->numSkippedEntries:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 167
      return-void
  .end method
