*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$SyncHandlerMessagePayload.smali	2012-03-23 19:36:27.739186807 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$SyncHandlerMessagePayload.smali.original	2012-03-23 19:34:45.125315646 -0500
***************
*** 30,42 ****
--- 30,46 ----
      .parameter "syncResult"
  
      .prologue
+     .line 706
      iput-object p1, p0, Landroid/content/SyncManager$SyncHandlerMessagePayload;->this$0:Landroid/content/SyncManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 707
      iput-object p2, p0, Landroid/content/SyncManager$SyncHandlerMessagePayload;->activeSyncContext:Landroid/content/SyncManager$ActiveSyncContext;
  
+     .line 708
      iput-object p3, p0, Landroid/content/SyncManager$SyncHandlerMessagePayload;->syncResult:Landroid/content/SyncResult;
  
+     .line 709
      return-void
  .end method
