*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$InitializerServiceConnection$1.smali	2012-03-23 19:36:24.070119902 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$InitializerServiceConnection$1.smali.original	2012-03-23 19:34:45.091315026 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 463
      iput-object p1, p0, Landroid/content/SyncManager$InitializerServiceConnection$1;->this$0:Landroid/content/SyncManager$InitializerServiceConnection;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .locals 2
  
      .prologue
+     .line 465
      iget-object v0, p0, Landroid/content/SyncManager$InitializerServiceConnection$1;->this$0:Landroid/content/SyncManager$InitializerServiceConnection;
  
      #getter for: Landroid/content/SyncManager$InitializerServiceConnection;->mContext:Landroid/content/Context;
***************
*** 49,54 ****
--- 51,57 ----
  
      if-eqz v0, :cond_0
  
+     .line 466
      iget-object v0, p0, Landroid/content/SyncManager$InitializerServiceConnection$1;->this$0:Landroid/content/SyncManager$InitializerServiceConnection;
  
      #getter for: Landroid/content/SyncManager$InitializerServiceConnection;->mContext:Landroid/content/Context;
***************
*** 60,65 ****
--- 63,69 ----
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->unbindService(Landroid/content/ServiceConnection;)V
  
+     .line 467
      iget-object v0, p0, Landroid/content/SyncManager$InitializerServiceConnection$1;->this$0:Landroid/content/SyncManager$InitializerServiceConnection;
  
      const/4 v1, 0x0
***************
*** 67,72 ****
--- 71,77 ----
      #setter for: Landroid/content/SyncManager$InitializerServiceConnection;->mContext:Landroid/content/Context;
      invoke-static {v0, v1}, Landroid/content/SyncManager$InitializerServiceConnection;->access$702(Landroid/content/SyncManager$InitializerServiceConnection;Landroid/content/Context;)Landroid/content/Context;
  
+     .line 469
      :cond_0
      return-void
  .end method
