*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$SyncHandler$SyncNotificationInfo.smali	2012-03-23 19:36:26.970172784 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$SyncHandler$SyncNotificationInfo.smali.original	2012-03-23 19:34:45.117315501 -0500
***************
*** 28,41 ****
--- 28,44 ----
      .parameter
  
      .prologue
+     .line 1586
      iput-object p1, p0, Landroid/content/SyncManager$SyncHandler$SyncNotificationInfo;->this$1:Landroid/content/SyncManager$SyncHandler;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1588
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/content/SyncManager$SyncHandler$SyncNotificationInfo;->isActive:Z
  
+     .line 1592
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/content/SyncManager$SyncHandler$SyncNotificationInfo;->startTime:Ljava/lang/Long;
***************
*** 49,61 ****
--- 52,67 ----
      .locals 2
  
      .prologue
+     .line 1600
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 1601
      .local v0, sb:Ljava/lang/StringBuilder;
      invoke-virtual {p0, v0}, Landroid/content/SyncManager$SyncHandler$SyncNotificationInfo;->toString(Ljava/lang/StringBuilder;)V
  
+     .line 1602
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
***************
*** 68,73 ****
--- 74,80 ----
      .parameter "sb"
  
      .prologue
+     .line 1595
      const-string v0, "isActive "
  
      invoke-virtual {p1, v0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 90,94 ****
--- 97,102 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
  
+     .line 1596
      return-void
  .end method
