*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$9.smali	2012-03-23 19:36:31.254250904 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$9.smali.original	2012-03-23 19:34:45.159316266 -0500
***************
*** 37,42 ****
--- 37,43 ----
      .parameter
  
      .prologue
+     .line 1323
      iput-object p1, p0, Landroid/content/SyncManager$9;->this$0:Landroid/content/SyncManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 52,57 ****
--- 53,59 ----
      .parameter "rhs"
  
      .prologue
+     .line 1327
      iget v1, p2, Landroid/content/SyncManager$AccountSyncStats;->times:I
  
      iget v2, p1, Landroid/content/SyncManager$AccountSyncStats;->times:I
***************
*** 60,68 ****
--- 62,72 ----
  
      move-result v0
  
+     .line 1328
      .local v0, compare:I
      if-nez v0, :cond_0
  
+     .line 1329
      iget-wide v1, p2, Landroid/content/SyncManager$AccountSyncStats;->elapsedTime:J
  
      iget-wide v3, p1, Landroid/content/SyncManager$AccountSyncStats;->elapsedTime:J
***************
*** 71,76 ****
--- 75,81 ----
  
      move-result v0
  
+     .line 1331
      :cond_0
      return v0
  .end method
***************
*** 81,86 ****
--- 86,92 ----
      .parameter "x1"
  
      .prologue
+     .line 1323
      check-cast p1, Landroid/content/SyncManager$AccountSyncStats;
  
      .end local p1
