*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncStatusInfo.smali	2012-03-23 19:36:24.417126230 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncStatusInfo.smali.original	2012-03-23 19:34:45.095315099 -0500
***************
*** 73,78 ****
--- 73,79 ----
      .locals 1
  
      .prologue
+     .line 160
      new-instance v0, Landroid/content/SyncStatusInfo$1;
  
      invoke-direct {v0}, Landroid/content/SyncStatusInfo$1;-><init>()V
***************
*** 87,96 ****
--- 88,100 ----
      .parameter "authorityId"
  
      .prologue
+     .line 49
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 50
      iput p1, p0, Landroid/content/SyncStatusInfo;->authorityId:I
  
+     .line 51
      return-void
  .end method
  
***************
*** 105,116 ****
--- 109,123 ----
  
      const/4 v4, 0x1
  
+     .line 95
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 96
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
+     .line 97
      .local v2, version:I
      const/4 v3, 0x2
  
***************
*** 118,123 ****
--- 125,131 ----
  
      if-eq v2, v4, :cond_0
  
+     .line 98
      const-string v3, "SyncStatusInfo"
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 140,145 ****
--- 148,154 ----
  
      invoke-static {v3, v6}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 100
      :cond_0
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
***************
*** 147,224 ****
--- 156,246 ----
  
      iput v3, p0, Landroid/content/SyncStatusInfo;->authorityId:I
  
+     .line 101
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v6
  
      iput-wide v6, p0, Landroid/content/SyncStatusInfo;->totalElapsedTime:J
  
+     .line 102
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      iput v3, p0, Landroid/content/SyncStatusInfo;->numSyncs:I
  
+     .line 103
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      iput v3, p0, Landroid/content/SyncStatusInfo;->numSourcePoll:I
  
+     .line 104
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      iput v3, p0, Landroid/content/SyncStatusInfo;->numSourceServer:I
  
+     .line 105
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      iput v3, p0, Landroid/content/SyncStatusInfo;->numSourceLocal:I
  
+     .line 106
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      iput v3, p0, Landroid/content/SyncStatusInfo;->numSourceUser:I
  
+     .line 107
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v6
  
      iput-wide v6, p0, Landroid/content/SyncStatusInfo;->lastSuccessTime:J
  
+     .line 108
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      iput v3, p0, Landroid/content/SyncStatusInfo;->lastSuccessSource:I
  
+     .line 109
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v6
  
      iput-wide v6, p0, Landroid/content/SyncStatusInfo;->lastFailureTime:J
  
+     .line 110
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      iput v3, p0, Landroid/content/SyncStatusInfo;->lastFailureSource:I
  
+     .line 111
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
      iput-object v3, p0, Landroid/content/SyncStatusInfo;->lastFailureMesg:Ljava/lang/String;
  
+     .line 112
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v6
  
      iput-wide v6, p0, Landroid/content/SyncStatusInfo;->initialFailureTime:J
  
+     .line 113
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
***************
*** 230,235 ****
--- 252,258 ----
      :goto_0
      iput-boolean v3, p0, Landroid/content/SyncStatusInfo;->pending:Z
  
+     .line 114
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
***************
*** 241,250 ****
--- 264,276 ----
      :cond_1
      iput-boolean v5, p0, Landroid/content/SyncStatusInfo;->initialize:Z
  
+     .line 115
      if-ne v2, v4, :cond_4
  
+     .line 116
      iput-object v8, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
+     .line 128
      :cond_2
      :goto_1
      return-void
***************
*** 252,271 ****
--- 278,302 ----
      :cond_3
      move v3, v5
  
+     .line 113
      goto :goto_0
  
+     .line 118
      :cond_4
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 119
      .local v0, N:I
      if-gez v0, :cond_5
  
+     .line 120
      iput-object v8, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      goto :goto_1
  
+     .line 122
      :cond_5
      new-instance v3, Ljava/util/ArrayList;
  
***************
*** 273,284 ****
--- 304,317 ----
  
      iput-object v3, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
+     .line 123
      const/4 v1, 0x0
  
      .local v1, i:I
      :goto_2
      if-ge v1, v0, :cond_2
  
+     .line 124
      iget-object v3, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
***************
*** 291,296 ****
--- 324,330 ----
  
      invoke-virtual {v3, v4}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 123
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_2
***************
*** 301,310 ****
--- 335,346 ----
      .parameter "index"
  
      .prologue
+     .line 136
      iget-object v2, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      if-nez v2, :cond_0
  
+     .line 137
      new-instance v2, Ljava/util/ArrayList;
  
      const/4 v3, 0x0
***************
*** 313,321 ****
--- 349,359 ----
  
      iput-object v2, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
+     .line 140
      :cond_0
      add-int/lit8 v1, p1, 0x1
  
+     .line 141
      .local v1, requiredSize:I
      iget-object v2, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
***************
*** 325,330 ****
--- 363,369 ----
  
      if-ge v2, v1, :cond_1
  
+     .line 142
      iget-object v2, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
***************
*** 335,340 ****
--- 374,380 ----
      :goto_0
      if-ge v0, v1, :cond_1
  
+     .line 143
      iget-object v2, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      const-wide/16 v3, 0x0
***************
*** 345,354 ****
--- 385,396 ----
  
      invoke-virtual {v2, v3}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 142
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 146
      .end local v0           #i:I
      :cond_1
      return-void
***************
*** 360,365 ****
--- 402,408 ----
      .locals 1
  
      .prologue
+     .line 65
      const/4 v0, 0x0
  
      return v0
***************
*** 370,380 ****
--- 413,425 ----
      .parameter "def"
  
      .prologue
+     .line 55
      :try_start_0
      iget-object v1, p0, Landroid/content/SyncStatusInfo;->lastFailureMesg:Ljava/lang/String;
  
      if-eqz v1, :cond_0
  
+     .line 56
      iget-object v1, p0, Landroid/content/SyncStatusInfo;->lastFailureMesg:Ljava/lang/String;
  
      invoke-static {v1}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
***************
*** 383,397 ****
--- 428,445 ----
  
      move-result p1
  
+     .line 61
      .end local p1
      :cond_0
      :goto_0
      return p1
  
+     .line 58
      .restart local p1
      :catch_0
      move-exception v0
  
+     .line 59
      .local v0, e:Ljava/lang/NumberFormatException;
      const-string v1, "Sync"
  
***************
*** 425,430 ****
--- 473,479 ----
      .parameter "index"
  
      .prologue
+     .line 149
      iget-object v0, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      if-eqz v0, :cond_0
***************
*** 439,447 ****
--- 488,498 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 150
      :cond_0
      const-wide/16 v0, 0x0
  
+     .line 152
      :goto_0
      return-wide v0
  
***************
*** 466,477 ****
--- 517,531 ----
      .parameter "index"
  
      .prologue
+     .line 156
      invoke-direct {p0, p1}, Landroid/content/SyncStatusInfo;->ensurePeriodicSyncTimeSize(I)V
  
+     .line 157
      iget-object v0, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 158
      return-void
  .end method
  
***************
*** 481,488 ****
--- 535,544 ----
      .parameter "when"
  
      .prologue
+     .line 131
      invoke-direct {p0, p1}, Landroid/content/SyncStatusInfo;->ensurePeriodicSyncTimeSize(I)V
  
+     .line 132
      iget-object v0, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      invoke-static {p2, p3}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
***************
*** 491,496 ****
--- 547,553 ----
  
      invoke-virtual {v0, p1, v1}, Ljava/util/ArrayList;->set(ILjava/lang/Object;)Ljava/lang/Object;
  
+     .line 133
      return-void
  .end method
  
***************
*** 504,565 ****
--- 561,637 ----
  
      const/4 v5, 0x0
  
+     .line 69
      const/4 v3, 0x2
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 70
      iget v3, p0, Landroid/content/SyncStatusInfo;->authorityId:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 71
      iget-wide v6, p0, Landroid/content/SyncStatusInfo;->totalElapsedTime:J
  
      invoke-virtual {p1, v6, v7}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 72
      iget v3, p0, Landroid/content/SyncStatusInfo;->numSyncs:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 73
      iget v3, p0, Landroid/content/SyncStatusInfo;->numSourcePoll:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 74
      iget v3, p0, Landroid/content/SyncStatusInfo;->numSourceServer:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 75
      iget v3, p0, Landroid/content/SyncStatusInfo;->numSourceLocal:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 76
      iget v3, p0, Landroid/content/SyncStatusInfo;->numSourceUser:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 77
      iget-wide v6, p0, Landroid/content/SyncStatusInfo;->lastSuccessTime:J
  
      invoke-virtual {p1, v6, v7}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 78
      iget v3, p0, Landroid/content/SyncStatusInfo;->lastSuccessSource:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 79
      iget-wide v6, p0, Landroid/content/SyncStatusInfo;->lastFailureTime:J
  
      invoke-virtual {p1, v6, v7}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 80
      iget v3, p0, Landroid/content/SyncStatusInfo;->lastFailureSource:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 81
      iget-object v3, p0, Landroid/content/SyncStatusInfo;->lastFailureMesg:Ljava/lang/String;
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 82
      iget-wide v6, p0, Landroid/content/SyncStatusInfo;->initialFailureTime:J
  
      invoke-virtual {p1, v6, v7}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 83
      iget-boolean v3, p0, Landroid/content/SyncStatusInfo;->pending:Z
  
      if-eqz v3, :cond_0
***************
*** 569,574 ****
--- 641,647 ----
      :goto_0
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 84
      iget-boolean v3, p0, Landroid/content/SyncStatusInfo;->initialize:Z
  
      if-eqz v3, :cond_1
***************
*** 576,585 ****
--- 649,660 ----
      :goto_1
      invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 85
      iget-object v3, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      if-eqz v3, :cond_2
  
+     .line 86
      iget-object v3, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
***************
*** 588,593 ****
--- 663,669 ----
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 87
      iget-object v3, p0, Landroid/content/SyncStatusInfo;->periodicSyncTimes:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
***************
*** 612,617 ****
--- 688,694 ----
  
      move-result-wide v1
  
+     .line 88
      .local v1, periodicSyncTime:J
      invoke-virtual {p1, v1, v2}, Landroid/os/Parcel;->writeLong(J)V
  
***************
*** 622,639 ****
--- 699,720 ----
      :cond_0
      move v3, v5
  
+     .line 83
      goto :goto_0
  
      :cond_1
      move v4, v5
  
+     .line 84
      goto :goto_1
  
+     .line 91
      :cond_2
      const/4 v3, -0x1
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 93
      :cond_3
      return-void
  .end method
