*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncInfo.smali	2012-03-23 19:36:24.192122126 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncInfo.smali.original	2012-03-23 19:34:45.093315063 -0500
***************
*** 34,39 ****
--- 34,40 ----
      .locals 1
  
      .prologue
+     .line 79
      new-instance v0, Landroid/content/SyncInfo$1;
  
      invoke-direct {v0}, Landroid/content/SyncInfo$1;-><init>()V
***************
*** 51,66 ****
--- 52,73 ----
      .parameter "startTime"
  
      .prologue
+     .line 50
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 51
      iput p1, p0, Landroid/content/SyncInfo;->authorityId:I
  
+     .line 52
      iput-object p2, p0, Landroid/content/SyncInfo;->account:Landroid/accounts/Account;
  
+     .line 53
      iput-object p3, p0, Landroid/content/SyncInfo;->authority:Ljava/lang/String;
  
+     .line 54
      iput-wide p4, p0, Landroid/content/SyncInfo;->startTime:J
  
+     .line 55
      return-void
  .end method
  
***************
*** 69,100 ****
--- 76,113 ----
      .parameter "parcel"
  
      .prologue
+     .line 71
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 72
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/SyncInfo;->authorityId:I
  
+     .line 73
      new-instance v0, Landroid/accounts/Account;
  
      invoke-direct {v0, p1}, Landroid/accounts/Account;-><init>(Landroid/os/Parcel;)V
  
      iput-object v0, p0, Landroid/content/SyncInfo;->account:Landroid/accounts/Account;
  
+     .line 74
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/SyncInfo;->authority:Ljava/lang/String;
  
+     .line 75
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/content/SyncInfo;->startTime:J
  
+     .line 76
      return-void
  .end method
  
***************
*** 104,109 ****
--- 117,123 ----
      .locals 1
  
      .prologue
+     .line 59
      const/4 v0, 0x0
  
      return v0
***************
*** 115,137 ****
--- 129,156 ----
      .parameter "flags"
  
      .prologue
+     .line 64
      iget v0, p0, Landroid/content/SyncInfo;->authorityId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 65
      iget-object v0, p0, Landroid/content/SyncInfo;->account:Landroid/accounts/Account;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, p1, v1}, Landroid/accounts/Account;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 66
      iget-object v0, p0, Landroid/content/SyncInfo;->authority:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 67
      iget-wide v0, p0, Landroid/content/SyncInfo;->startTime:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 68
      return-void
  .end method
