*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$SyncTimeTracker.smali	2012-03-23 19:36:23.916117094 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$SyncTimeTracker.smali.original	2012-03-23 19:34:45.090315007 -0500
***************
*** 30,43 ****
--- 30,46 ----
      .parameter
  
      .prologue
+     .line 1490
      iput-object p1, p0, Landroid/content/SyncManager$SyncTimeTracker;->this$0:Landroid/content/SyncManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1492
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/content/SyncManager$SyncTimeTracker;->mLastWasSyncing:Z
  
+     .line 1494
      const-wide/16 v0, 0x0
  
      iput-wide v0, p0, Landroid/content/SyncManager$SyncTimeTracker;->mWhenSyncStarted:J
***************
*** 51,56 ****
--- 54,60 ----
      .parameter "x1"
  
      .prologue
+     .line 1490
      invoke-direct {p0, p1}, Landroid/content/SyncManager$SyncTimeTracker;-><init>(Landroid/content/SyncManager;)V
  
      return-void
***************
*** 62,67 ****
--- 66,72 ----
      .locals 6
  
      .prologue
+     .line 1513
      monitor-enter p0
  
      :try_start_0
***************
*** 73,89 ****
--- 78,97 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1516
      :goto_0
      monitor-exit p0
  
      return-wide v2
  
+     .line 1515
      :cond_0
      :try_start_1
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v0
  
+     .line 1516
      .local v0, now:J
      iget-wide v2, p0, Landroid/content/SyncManager$SyncTimeTracker;->mTimeSpentSyncing:J
  
***************
*** 97,102 ****
--- 105,111 ----
  
      goto :goto_0
  
+     .line 1513
      .end local v0           #now:J
      :catchall_0
      move-exception v2
***************
*** 110,115 ****
--- 119,125 ----
      .locals 7
  
      .prologue
+     .line 1500
      monitor-enter p0
  
      :try_start_0
***************
*** 125,130 ****
--- 135,141 ----
  
      const/4 v0, 0x1
  
+     .line 1501
      .local v0, isSyncInProgress:Z
      :goto_0
      iget-boolean v3, p0, Landroid/content/SyncManager$SyncTimeTracker;->mLastWasSyncing:Z
***************
*** 133,149 ****
--- 144,163 ----
  
      if-ne v0, v3, :cond_1
  
+     .line 1509
      :goto_1
      monitor-exit p0
  
      return-void
  
+     .line 1500
      .end local v0           #isSyncInProgress:Z
      :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 1502
      .restart local v0       #isSyncInProgress:Z
      :cond_1
      :try_start_1
***************
*** 151,161 ****
--- 165,178 ----
  
      move-result-wide v1
  
+     .line 1503
      .local v1, now:J
      if-eqz v0, :cond_2
  
+     .line 1504
      iput-wide v1, p0, Landroid/content/SyncManager$SyncTimeTracker;->mWhenSyncStarted:J
  
+     .line 1508
      :goto_2
      iput-boolean v0, p0, Landroid/content/SyncManager$SyncTimeTracker;->mLastWasSyncing:Z
      :try_end_1
***************
*** 163,168 ****
--- 180,186 ----
  
      goto :goto_1
  
+     .line 1500
      .end local v0           #isSyncInProgress:Z
      .end local v1           #now:J
      :catchall_0
***************
*** 172,177 ****
--- 190,196 ----
  
      throw v3
  
+     .line 1506
      .restart local v0       #isSyncInProgress:Z
      .restart local v1       #now:J
      :cond_2
