*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$4.smali	2012-03-23 19:36:25.073138192 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncManager$4.smali.original	2012-03-23 19:34:45.100315191 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 246
      iput-object p1, p0, Landroid/content/SyncManager$4;->this$0:Landroid/content/SyncManager;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 39,44 ****
--- 40,46 ----
      .parameter "intent"
  
      .prologue
+     .line 248
      iget-object v1, p0, Landroid/content/SyncManager$4;->this$0:Landroid/content/SyncManager;
  
      #getter for: Landroid/content/SyncManager;->mDataConnectionIsConnected:Z
***************
*** 46,51 ****
--- 48,54 ----
  
      move-result v0
  
+     .line 252
      .local v0, wasConnected:Z
      iget-object v1, p0, Landroid/content/SyncManager$4;->this$0:Landroid/content/SyncManager;
  
***************
*** 59,64 ****
--- 62,68 ----
      #setter for: Landroid/content/SyncManager;->mDataConnectionIsConnected:Z
      invoke-static {v1, v2}, Landroid/content/SyncManager;->access$402(Landroid/content/SyncManager;Z)Z
  
+     .line 253
      iget-object v1, p0, Landroid/content/SyncManager$4;->this$0:Landroid/content/SyncManager;
  
      #getter for: Landroid/content/SyncManager;->mDataConnectionIsConnected:Z
***************
*** 68,75 ****
--- 72,81 ----
  
      if-eqz v1, :cond_2
  
+     .line 254
      if-nez v0, :cond_1
  
+     .line 255
      const-string v1, "SyncManager"
  
      const/4 v2, 0x2
***************
*** 80,91 ****
--- 86,99 ----
  
      if-eqz v1, :cond_0
  
+     .line 256
      const-string v1, "SyncManager"
  
      const-string v2, "Reconnection detected: clearing all backoffs"
  
      invoke-static {v1, v2}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 258
      :cond_0
      iget-object v1, p0, Landroid/content/SyncManager$4;->this$0:Landroid/content/SyncManager;
  
***************
*** 100,111 ****
--- 108,121 ----
  
      invoke-virtual {v1, v2}, Landroid/content/SyncStorageEngine;->clearAllBackoffs(Landroid/content/SyncQueue;)V
  
+     .line 260
      :cond_1
      iget-object v1, p0, Landroid/content/SyncManager$4;->this$0:Landroid/content/SyncManager;
  
      #calls: Landroid/content/SyncManager;->sendCheckAlarmsMessage()V
      invoke-static {v1}, Landroid/content/SyncManager;->access$100(Landroid/content/SyncManager;)V
  
+     .line 262
      :cond_2
      return-void
  .end method
