*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncStorageEngine$PendingOperation.smali	2012-03-23 19:36:28.096193317 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncStorageEngine$PendingOperation.smali.original	2012-03-23 19:34:45.128315701 -0500
***************
*** 40,53 ****
--- 40,58 ----
      .parameter "expedited"
  
      .prologue
+     .line 145
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 146
      iput-object p1, p0, Landroid/content/SyncStorageEngine$PendingOperation;->account:Landroid/accounts/Account;
  
+     .line 147
      iput p2, p0, Landroid/content/SyncStorageEngine$PendingOperation;->syncSource:I
  
+     .line 148
      iput-object p3, p0, Landroid/content/SyncStorageEngine$PendingOperation;->authority:Ljava/lang/String;
  
+     .line 149
      if-eqz p4, :cond_0
  
      new-instance v0, Landroid/os/Bundle;
***************
*** 60,71 ****
--- 65,79 ----
      :cond_0
      iput-object p4, p0, Landroid/content/SyncStorageEngine$PendingOperation;->extras:Landroid/os/Bundle;
  
+     .line 150
      iput-boolean p5, p0, Landroid/content/SyncStorageEngine$PendingOperation;->expedited:Z
  
+     .line 151
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/content/SyncStorageEngine$PendingOperation;->authorityId:I
  
+     .line 152
      return-void
  .end method
  
***************
*** 74,104 ****
--- 82,120 ----
      .parameter "other"
  
      .prologue
+     .line 154
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 155
      iget-object v0, p1, Landroid/content/SyncStorageEngine$PendingOperation;->account:Landroid/accounts/Account;
  
      iput-object v0, p0, Landroid/content/SyncStorageEngine$PendingOperation;->account:Landroid/accounts/Account;
  
+     .line 156
      iget v0, p1, Landroid/content/SyncStorageEngine$PendingOperation;->syncSource:I
  
      iput v0, p0, Landroid/content/SyncStorageEngine$PendingOperation;->syncSource:I
  
+     .line 157
      iget-object v0, p1, Landroid/content/SyncStorageEngine$PendingOperation;->authority:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/SyncStorageEngine$PendingOperation;->authority:Ljava/lang/String;
  
+     .line 158
      iget-object v0, p1, Landroid/content/SyncStorageEngine$PendingOperation;->extras:Landroid/os/Bundle;
  
      iput-object v0, p0, Landroid/content/SyncStorageEngine$PendingOperation;->extras:Landroid/os/Bundle;
  
+     .line 159
      iget v0, p1, Landroid/content/SyncStorageEngine$PendingOperation;->authorityId:I
  
      iput v0, p0, Landroid/content/SyncStorageEngine$PendingOperation;->authorityId:I
  
+     .line 160
      iget-boolean v0, p1, Landroid/content/SyncStorageEngine$PendingOperation;->expedited:Z
  
      iput-boolean v0, p0, Landroid/content/SyncStorageEngine$PendingOperation;->expedited:Z
  
+     .line 161
      return-void
  .end method
