*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncStorageEngine$DayStats.smali	2012-03-23 19:36:23.798114942 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/SyncStorageEngine$DayStats.smali.original	2012-03-23 19:34:45.089314989 -0500
***************
*** 32,40 ****
--- 32,43 ----
      .parameter "day"
  
      .prologue
+     .line 217
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 218
      iput p1, p0, Landroid/content/SyncStorageEngine$DayStats;->day:I
  
+     .line 219
      return-void
  .end method
