*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioManager$FocusEventHandlerDelegate$1.smali	2012-03-23 19:35:33.634200204 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioManager$FocusEventHandlerDelegate$1.smali.original	2012-03-23 19:34:44.712308115 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .parameter
  
      .prologue
+     .line 1550
      iput-object p1, p0, Landroid/media/AudioManager$FocusEventHandlerDelegate$1;->this$1:Landroid/media/AudioManager$FocusEventHandlerDelegate;
  
      iput-object p3, p0, Landroid/media/AudioManager$FocusEventHandlerDelegate$1;->val$this$0:Landroid/media/AudioManager;
***************
*** 44,51 ****
--- 45,54 ----
      .parameter "msg"
  
      .prologue
+     .line 1553
      const/4 v0, 0x0
  
+     .line 1554
      .local v0, listener:Landroid/media/AudioManager$OnAudioFocusChangeListener;
      iget-object v1, p0, Landroid/media/AudioManager$FocusEventHandlerDelegate$1;->this$1:Landroid/media/AudioManager$FocusEventHandlerDelegate;
  
***************
*** 58,63 ****
--- 61,67 ----
  
      monitor-enter v2
  
+     .line 1555
      :try_start_0
      iget-object v1, p0, Landroid/media/AudioManager$FocusEventHandlerDelegate$1;->this$1:Landroid/media/AudioManager$FocusEventHandlerDelegate;
  
***************
*** 72,90 ****
--- 76,99 ----
  
      move-result-object v0
  
+     .line 1556
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1557
      if-eqz v0, :cond_0
  
+     .line 1558
      iget v1, p1, Landroid/os/Message;->what:I
  
      invoke-interface {v0, v1}, Landroid/media/AudioManager$OnAudioFocusChangeListener;->onAudioFocusChange(I)V
  
+     .line 1560
      :cond_0
      return-void
  
+     .line 1556
      :catchall_0
      move-exception v1
  
