*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioService$RemoteControlStackEntry.smali	2012-03-23 19:35:38.555289941 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioService$RemoteControlStackEntry.smali.original	2012-03-23 19:34:44.754308880 -0500
***************
*** 35,54 ****
--- 35,60 ----
      .parameter "eventReceiver"
  
      .prologue
+     .line 3138
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 3139
      iput-object p1, p0, Landroid/media/AudioService$RemoteControlStackEntry;->mMediaIntent:Landroid/app/PendingIntent;
  
+     .line 3140
      iput-object p2, p0, Landroid/media/AudioService$RemoteControlStackEntry;->mReceiverComponent:Landroid/content/ComponentName;
  
+     .line 3141
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/media/AudioService$RemoteControlStackEntry;->mCallingUid:I
  
+     .line 3142
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/media/AudioService$RemoteControlStackEntry;->mRcClient:Landroid/media/IRemoteControlClient;
  
+     .line 3143
      return-void
  .end method
  
***************
*** 63,72 ****
--- 69,81 ----
      .end annotation
  
      .prologue
+     .line 3160
      invoke-virtual {p0}, Landroid/media/AudioService$RemoteControlStackEntry;->unlinkToRcClientDeath()V
  
+     .line 3161
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 3162
      return-void
  .end method
  
***************
*** 74,79 ****
--- 83,89 ----
      .locals 4
  
      .prologue
+     .line 3146
      iget-object v1, p0, Landroid/media/AudioService$RemoteControlStackEntry;->mRcClientDeathHandler:Landroid/media/AudioService$RcClientDeathHandler;
  
      if-eqz v1, :cond_0
***************
*** 87,92 ****
--- 97,103 ----
  
      if-eqz v1, :cond_0
  
+     .line 3148
      :try_start_0
      iget-object v1, p0, Landroid/media/AudioService$RemoteControlStackEntry;->mRcClientDeathHandler:Landroid/media/AudioService$RcClientDeathHandler;
  
***************
*** 101,119 ****
--- 112,134 ----
  
      invoke-interface {v1, v2, v3}, Landroid/os/IBinder;->unlinkToDeath(Landroid/os/IBinder$DeathRecipient;I)Z
  
+     .line 3149
      const/4 v1, 0x0
  
      iput-object v1, p0, Landroid/media/AudioService$RemoteControlStackEntry;->mRcClientDeathHandler:Landroid/media/AudioService$RcClientDeathHandler;
      :try_end_0
      .catch Ljava/util/NoSuchElementException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 3156
      :cond_0
      :goto_0
      return-void
  
+     .line 3150
      :catch_0
      move-exception v0
  
+     .line 3152
      .local v0, e:Ljava/util/NoSuchElementException;
      const-string v1, "AudioService"
  
***************
*** 143,148 ****
--- 158,164 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 3153
      invoke-virtual {v0}, Ljava/util/NoSuchElementException;->printStackTrace()V
  
      goto :goto_0
