*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioTrack$NativeEventHandlerDelegate$1.smali	2012-03-23 19:35:37.931278561 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioTrack$NativeEventHandlerDelegate$1.smali.original	2012-03-23 19:34:44.749308790 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .parameter
  
      .prologue
+     .line 1084
      iput-object p1, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate$1;->this$1:Landroid/media/AudioTrack$NativeEventHandlerDelegate;
  
      iput-object p3, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate$1;->val$this$0:Landroid/media/AudioTrack;
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "msg"
  
      .prologue
+     .line 1087
      iget-object v1, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate$1;->this$1:Landroid/media/AudioTrack$NativeEventHandlerDelegate;
  
      #getter for: Landroid/media/AudioTrack$NativeEventHandlerDelegate;->mAudioTrack:Landroid/media/AudioTrack;
***************
*** 53,65 ****
--- 55,70 ----
  
      if-nez v1, :cond_1
  
+     .line 1110
      :cond_0
      :goto_0
      return-void
  
+     .line 1090
      :cond_1
      const/4 v0, 0x0
  
+     .line 1091
      .local v0, listener:Landroid/media/AudioTrack$OnPlaybackPositionUpdateListener;
      iget-object v1, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate$1;->this$1:Landroid/media/AudioTrack$NativeEventHandlerDelegate;
  
***************
*** 72,77 ****
--- 77,83 ----
  
      monitor-enter v2
  
+     .line 1092
      :try_start_0
      iget-object v1, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate$1;->this$1:Landroid/media/AudioTrack$NativeEventHandlerDelegate;
  
***************
*** 85,98 ****
--- 91,107 ----
  
      move-result-object v0
  
+     .line 1093
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1094
      iget v1, p1, Landroid/os/Message;->what:I
  
      packed-switch v1, :pswitch_data_0
  
+     .line 1106
      const-string v1, "AudioTrack-Java"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 119,124 ****
--- 128,134 ----
  
      goto :goto_0
  
+     .line 1093
      :catchall_0
      move-exception v1
  
***************
*** 129,137 ****
--- 139,149 ----
  
      throw v1
  
+     .line 1096
      :pswitch_0
      if-eqz v0, :cond_0
  
+     .line 1097
      iget-object v1, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate$1;->this$1:Landroid/media/AudioTrack$NativeEventHandlerDelegate;
  
      #getter for: Landroid/media/AudioTrack$NativeEventHandlerDelegate;->mAudioTrack:Landroid/media/AudioTrack;
***************
*** 143,151 ****
--- 155,165 ----
  
      goto :goto_0
  
+     .line 1101
      :pswitch_1
      if-eqz v0, :cond_0
  
+     .line 1102
      iget-object v1, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate$1;->this$1:Landroid/media/AudioTrack$NativeEventHandlerDelegate;
  
      #getter for: Landroid/media/AudioTrack$NativeEventHandlerDelegate;->mAudioTrack:Landroid/media/AudioTrack;
***************
*** 157,162 ****
--- 171,177 ----
  
      goto :goto_0
  
+     .line 1094
      nop
  
      :pswitch_data_0
