*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioRecord$NativeEventHandler.smali	2012-03-23 19:35:39.931315032 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioRecord$NativeEventHandler.smali.original	2012-03-23 19:34:44.766309100 -0500
***************
*** 28,39 ****
--- 28,43 ----
      .parameter "looper"
  
      .prologue
+     .line 725
      iput-object p1, p0, Landroid/media/AudioRecord$NativeEventHandler;->this$0:Landroid/media/AudioRecord;
  
+     .line 726
      invoke-direct {p0, p3}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
+     .line 727
      iput-object p2, p0, Landroid/media/AudioRecord$NativeEventHandler;->mAudioRecord:Landroid/media/AudioRecord;
  
+     .line 728
      return-void
  .end method
  
***************
*** 44,51 ****
--- 48,57 ----
      .parameter "msg"
  
      .prologue
+     .line 732
      const/4 v0, 0x0
  
+     .line 733
      .local v0, listener:Landroid/media/AudioRecord$OnRecordPositionUpdateListener;
      iget-object v1, p0, Landroid/media/AudioRecord$NativeEventHandler;->this$0:Landroid/media/AudioRecord;
  
***************
*** 56,61 ****
--- 62,68 ----
  
      monitor-enter v2
  
+     .line 734
      :try_start_0
      iget-object v1, p0, Landroid/media/AudioRecord$NativeEventHandler;->mAudioRecord:Landroid/media/AudioRecord;
  
***************
*** 64,77 ****
--- 71,87 ----
  
      move-result-object v0
  
+     .line 735
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 737
      iget v1, p1, Landroid/os/Message;->what:I
  
      packed-switch v1, :pswitch_data_0
  
+     .line 749
      const-string v1, "AudioRecord-Java"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 96,105 ****
--- 106,117 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 753
      :cond_0
      :goto_0
      return-void
  
+     .line 735
      :catchall_0
      move-exception v1
  
***************
*** 110,133 ****
--- 122,150 ----
  
      throw v1
  
+     .line 739
      :pswitch_0
      if-eqz v0, :cond_0
  
+     .line 740
      iget-object v1, p0, Landroid/media/AudioRecord$NativeEventHandler;->mAudioRecord:Landroid/media/AudioRecord;
  
      invoke-interface {v0, v1}, Landroid/media/AudioRecord$OnRecordPositionUpdateListener;->onMarkerReached(Landroid/media/AudioRecord;)V
  
      goto :goto_0
  
+     .line 744
      :pswitch_1
      if-eqz v0, :cond_0
  
+     .line 745
      iget-object v1, p0, Landroid/media/AudioRecord$NativeEventHandler;->mAudioRecord:Landroid/media/AudioRecord;
  
      invoke-interface {v0, v1}, Landroid/media/AudioRecord$OnRecordPositionUpdateListener;->onPeriodicNotification(Landroid/media/AudioRecord;)V
  
      goto :goto_0
  
+     .line 737
      :pswitch_data_0
      .packed-switch 0x2
          :pswitch_0
