*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioService$AudioHandler$1.smali	2012-03-23 19:35:33.981206532 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioService$AudioHandler$1.smali.original	2012-03-23 19:34:44.715308170 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 2148
      iput-object p1, p0, Landroid/media/AudioService$AudioHandler$1;->this$1:Landroid/media/AudioService$AudioHandler;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,50 ****
--- 42,53 ----
      .parameter "mp"
  
      .prologue
+     .line 2150
      iget-object v0, p0, Landroid/media/AudioService$AudioHandler$1;->this$1:Landroid/media/AudioService$AudioHandler;
  
      #calls: Landroid/media/AudioService$AudioHandler;->cleanupPlayer(Landroid/media/MediaPlayer;)V
      invoke-static {v0, p1}, Landroid/media/AudioService$AudioHandler;->access$4500(Landroid/media/AudioService$AudioHandler;Landroid/media/MediaPlayer;)V
  
+     .line 2151
      return-void
  .end method
