*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioService$AudioHandler$2.smali	2012-03-23 19:35:38.511289137 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioService$AudioHandler$2.smali.original	2012-03-23 19:34:44.753308862 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 2153
      iput-object p1, p0, Landroid/media/AudioService$AudioHandler$2;->this$1:Landroid/media/AudioService$AudioHandler;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 43,53 ****
--- 44,56 ----
      .parameter "extra"
  
      .prologue
+     .line 2155
      iget-object v0, p0, Landroid/media/AudioService$AudioHandler$2;->this$1:Landroid/media/AudioService$AudioHandler;
  
      #calls: Landroid/media/AudioService$AudioHandler;->cleanupPlayer(Landroid/media/MediaPlayer;)V
      invoke-static {v0, p1}, Landroid/media/AudioService$AudioHandler;->access$4500(Landroid/media/AudioService$AudioHandler;Landroid/media/MediaPlayer;)V
  
+     .line 2156
      const/4 v0, 0x1
  
      return v0
