*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioTrack$NativeEventHandlerDelegate.smali	2012-03-23 19:35:38.897296177 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioTrack$NativeEventHandlerDelegate.smali.original	2012-03-23 19:34:44.757308936 -0500
***************
*** 30,60 ****
--- 30,68 ----
      .parameter "handler"
  
      .prologue
+     .line 1070
      iput-object p1, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate;->this$0:Landroid/media/AudioTrack;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1071
      iput-object p2, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate;->mAudioTrack:Landroid/media/AudioTrack;
  
+     .line 1074
      if-eqz p3, :cond_0
  
+     .line 1075
      invoke-virtual {p3}, Landroid/os/Handler;->getLooper()Landroid/os/Looper;
  
      move-result-object v0
  
+     .line 1082
      .local v0, looper:Landroid/os/Looper;
      :goto_0
      if-eqz v0, :cond_1
  
+     .line 1084
      new-instance v1, Landroid/media/AudioTrack$NativeEventHandlerDelegate$1;
  
      invoke-direct {v1, p0, v0, p1}, Landroid/media/AudioTrack$NativeEventHandlerDelegate$1;-><init>(Landroid/media/AudioTrack$NativeEventHandlerDelegate;Landroid/os/Looper;Landroid/media/AudioTrack;)V
  
      iput-object v1, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate;->mHandler:Landroid/os/Handler;
  
+     .line 1115
      :goto_1
      return-void
  
+     .line 1078
      .end local v0           #looper:Landroid/os/Looper;
      :cond_0
      #getter for: Landroid/media/AudioTrack;->mInitializationLooper:Landroid/os/Looper;
***************
*** 65,70 ****
--- 73,79 ----
      .restart local v0       #looper:Landroid/os/Looper;
      goto :goto_0
  
+     .line 1113
      :cond_1
      const/4 v1, 0x0
  
***************
*** 78,83 ****
--- 87,93 ----
      .parameter "x0"
  
      .prologue
+     .line 1066
      iget-object v0, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate;->mAudioTrack:Landroid/media/AudioTrack;
  
      return-object v0
***************
*** 89,94 ****
--- 99,105 ----
      .locals 1
  
      .prologue
+     .line 1118
      iget-object v0, p0, Landroid/media/AudioTrack$NativeEventHandlerDelegate;->mHandler:Landroid/os/Handler;
  
      return-object v0
