*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioFormat.smali	2012-03-23 19:35:34.486215740 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/AudioFormat.smali.original	2012-03-23 19:34:44.719308242 -0500
***************
*** 126,131 ****
--- 126,132 ----
      .locals 0
  
      .prologue
+     .line 25
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
