*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/EncoderCapabilities.smali	2012-03-23 19:35:38.653291727 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/EncoderCapabilities.smali.original	2012-03-23 19:34:44.755308899 -0500
***************
*** 21,32 ****
      .locals 1
  
      .prologue
!     const-string v0, "media_jni"
  
      invoke-static {v0}, Ljava/lang/System;->loadLibrary(Ljava/lang/String;)V
  
      invoke-static {}, Landroid/media/EncoderCapabilities;->native_init()V
  
      return-void
  .end method
  
--- 21,35 ----
      .locals 1
  
      .prologue
!     .line 104
!     const-string/jumbo v0, "media_jni"
  
      invoke-static {v0}, Ljava/lang/System;->loadLibrary(Ljava/lang/String;)V
  
+     .line 105
      invoke-static {}, Landroid/media/EncoderCapabilities;->native_init()V
  
+     .line 106
      return-void
  .end method
  
***************
*** 34,39 ****
--- 37,43 ----
      .locals 0
  
      .prologue
+     .line 154
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 52,74 ****
--- 56,83 ----
      .end annotation
  
      .prologue
+     .line 143
      invoke-static {}, Landroid/media/EncoderCapabilities;->native_get_num_audio_encoders()I
  
      move-result v2
  
+     .line 144
      .local v2, nEncoders:I
      if-nez v2, :cond_1
  
      const/4 v0, 0x0
  
+     .line 150
      :cond_0
      return-object v0
  
+     .line 146
      :cond_1
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
+     .line 147
      .local v0, encoderList:Ljava/util/List;,"Ljava/util/List<Landroid/media/EncoderCapabilities$AudioEncoderCap;>;"
      const/4 v1, 0x0
  
***************
*** 76,87 ****
--- 85,98 ----
      :goto_0
      if-ge v1, v2, :cond_0
  
+     .line 148
      invoke-static {v1}, Landroid/media/EncoderCapabilities;->native_get_audio_encoder_cap(I)Landroid/media/EncoderCapabilities$AudioEncoderCap;
  
      move-result-object v3
  
      invoke-interface {v0, v3}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 147
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
***************
*** 91,111 ****
--- 102,127 ----
      .locals 4
  
      .prologue
+     .line 113
      invoke-static {}, Landroid/media/EncoderCapabilities;->native_get_num_file_formats()I
  
      move-result v2
  
+     .line 114
      .local v2, nFormats:I
      if-nez v2, :cond_1
  
      const/4 v0, 0x0
  
+     .line 120
      :cond_0
      return-object v0
  
+     .line 116
      :cond_1
      new-array v0, v2, [I
  
+     .line 117
      .local v0, formats:[I
      const/4 v1, 0x0
  
***************
*** 113,124 ****
--- 129,142 ----
      :goto_0
      if-ge v1, v2, :cond_0
  
+     .line 118
      invoke-static {v1}, Landroid/media/EncoderCapabilities;->native_get_file_format(I)I
  
      move-result v3
  
      aput v3, v0, v1
  
+     .line 117
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
***************
*** 137,159 ****
--- 155,182 ----
      .end annotation
  
      .prologue
+     .line 128
      invoke-static {}, Landroid/media/EncoderCapabilities;->native_get_num_video_encoders()I
  
      move-result v2
  
+     .line 129
      .local v2, nEncoders:I
      if-nez v2, :cond_1
  
      const/4 v0, 0x0
  
+     .line 135
      :cond_0
      return-object v0
  
+     .line 131
      :cond_1
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
+     .line 132
      .local v0, encoderList:Ljava/util/List;,"Ljava/util/List<Landroid/media/EncoderCapabilities$VideoEncoderCap;>;"
      const/4 v1, 0x0
  
***************
*** 161,172 ****
--- 184,197 ----
      :goto_0
      if-ge v1, v2, :cond_0
  
+     .line 133
      invoke-static {v1}, Landroid/media/EncoderCapabilities;->native_get_video_encoder_cap(I)Landroid/media/EncoderCapabilities$VideoEncoderCap;
  
      move-result-object v3
  
      invoke-interface {v0, v3}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 132
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
