*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/IRemoteControlClient$Stub.smali	2012-03-23 19:35:39.392305203 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/IRemoteControlClient$Stub.smali.original	2012-03-23 19:34:44.761309008 -0500
***************
*** 40,51 ****
--- 40,54 ----
      .locals 1
  
      .prologue
+     .line 21
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 22
      const-string v0, "android.media.IRemoteControlClient"
  
      invoke-virtual {p0, p0, v0}, Landroid/media/IRemoteControlClient$Stub;->attachInterface(Landroid/os/IInterface;Ljava/lang/String;)V
  
+     .line 23
      return-void
  .end method
  
***************
*** 54,66 ****
--- 57,73 ----
      .parameter "obj"
  
      .prologue
+     .line 30
      if-nez p0, :cond_0
  
+     .line 31
      const/4 v0, 0x0
  
+     .line 37
      :goto_0
      return-object v0
  
+     .line 33
      :cond_0
      const-string v1, "android.media.IRemoteControlClient"
  
***************
*** 68,73 ****
--- 75,81 ----
  
      move-result-object v0
  
+     .line 34
      .local v0, iin:Landroid/os/IInterface;
      if-eqz v0, :cond_1
  
***************
*** 75,84 ****
--- 83,94 ----
  
      if-eqz v1, :cond_1
  
+     .line 35
      check-cast v0, Landroid/media/IRemoteControlClient;
  
      goto :goto_0
  
+     .line 37
      :cond_1
      new-instance v0, Landroid/media/IRemoteControlClient$Stub$Proxy;
  
***************
*** 94,99 ****
--- 104,110 ----
      .locals 0
  
      .prologue
+     .line 41
      return-object p0
  .end method
  
***************
*** 112,119 ****
--- 123,132 ----
      .prologue
      const/4 v4, 0x1
  
+     .line 45
      sparse-switch p1, :sswitch_data_0
  
+     .line 91
      invoke-super {p0, p1, p2, p3, p4}, Landroid/os/Binder;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
      move-result v4
***************
*** 121,126 ****
--- 134,140 ----
      :goto_0
      return v4
  
+     .line 49
      :sswitch_0
      const-string v5, "android.media.IRemoteControlClient"
  
***************
*** 128,162 ****
--- 142,183 ----
  
      goto :goto_0
  
+     .line 54
      :sswitch_1
      const-string v5, "android.media.IRemoteControlClient"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 56
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 58
      .local v0, _arg0:I
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
+     .line 60
      .local v1, _arg1:I
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
+     .line 62
      .local v2, _arg2:I
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
+     .line 63
      .local v3, _arg3:I
      invoke-virtual {p0, v0, v1, v2, v3}, Landroid/media/IRemoteControlClient$Stub;->onInformationRequested(IIII)V
  
      goto :goto_0
  
+     .line 68
      .end local v0           #_arg0:I
      .end local v1           #_arg1:I
      .end local v2           #_arg2:I
***************
*** 166,186 ****
--- 187,211 ----
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 70
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 71
      .restart local v0       #_arg0:I
      invoke-virtual {p0, v0}, Landroid/media/IRemoteControlClient$Stub;->setCurrentClientGenerationId(I)V
  
      goto :goto_0
  
+     .line 76
      .end local v0           #_arg0:I
      :sswitch_3
      const-string v5, "android.media.IRemoteControlClient"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 78
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v5
***************
*** 189,205 ****
--- 214,233 ----
  
      move-result-object v0
  
+     .line 79
      .local v0, _arg0:Landroid/media/IRemoteControlDisplay;
      invoke-virtual {p0, v0}, Landroid/media/IRemoteControlClient$Stub;->plugRemoteControlDisplay(Landroid/media/IRemoteControlDisplay;)V
  
      goto :goto_0
  
+     .line 84
      .end local v0           #_arg0:Landroid/media/IRemoteControlDisplay;
      :sswitch_4
      const-string v5, "android.media.IRemoteControlClient"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 86
      invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v5
***************
*** 208,218 ****
--- 236,248 ----
  
      move-result-object v0
  
+     .line 87
      .restart local v0       #_arg0:Landroid/media/IRemoteControlDisplay;
      invoke-virtual {p0, v0}, Landroid/media/IRemoteControlClient$Stub;->unplugRemoteControlDisplay(Landroid/media/IRemoteControlDisplay;)V
  
      goto :goto_0
  
+     .line 45
      nop
  
      :sswitch_data_0
