*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/Metadata.smali	2012-03-23 19:35:38.630291307 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/Metadata.smali.original	2012-03-23 19:34:44.754308880 -0500
***************
*** 142,151 ****
--- 142,153 ----
      .locals 1
  
      .prologue
+     .line 207
      sget-object v0, Ljava/util/Collections;->EMPTY_SET:Ljava/util/Set;
  
      sput-object v0, Landroid/media/Metadata;->MATCH_NONE:Ljava/util/Set;
  
+     .line 211
      const/4 v0, 0x0
  
      invoke-static {v0}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 165,178 ****
--- 167,183 ----
      .locals 1
  
      .prologue
+     .line 266
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 260
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
+     .line 266
      return-void
  .end method
  
***************
*** 181,186 ****
--- 186,192 ----
      .parameter "val"
  
      .prologue
+     .line 525
      if-lez p1, :cond_0
  
      const/16 v0, 0x1f
***************
*** 191,198 ****
  
      if-ge p1, v0, :cond_1
  
      :cond_0
!     const-string v0, "media.Metadata"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
--- 197,205 ----
  
      if-ge p1, v0, :cond_1
  
+     .line 526
      :cond_0
!     const-string/jumbo v0, "media.Metadata"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
***************
*** 214,221 ****
--- 221,230 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 527
      const/4 v0, 0x0
  
+     .line 529
      :goto_0
      return v0
  
***************
*** 231,236 ****
--- 240,246 ----
      .parameter "expectedType"
  
      .prologue
+     .line 536
      iget-object v2, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
      invoke-static {p1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 247,266 ****
--- 257,280 ----
  
      move-result v0
  
+     .line 538
      .local v0, pos:I
      iget-object v2, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
      invoke-virtual {v2, v0}, Landroid/os/Parcel;->setDataPosition(I)V
  
+     .line 540
      iget-object v2, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
      invoke-virtual {v2}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
+     .line 541
      .local v1, type:I
      if-eq v1, p2, :cond_0
  
+     .line 542
      new-instance v2, Ljava/lang/IllegalStateException;
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 295,300 ****
--- 309,315 ----
  
      throw v2
  
+     .line 544
      :cond_0
      return-void
  .end method
***************
*** 303,308 ****
--- 318,324 ----
      .locals 1
  
      .prologue
+     .line 511
      const/16 v0, 0x2000
  
      return v0
***************
*** 312,317 ****
--- 328,334 ----
      .locals 1
  
      .prologue
+     .line 505
      const/16 v0, 0x1f
  
      return v0
***************
*** 321,326 ****
--- 338,344 ----
      .locals 1
  
      .prologue
+     .line 517
      const/4 v0, 0x7
  
      return v0
***************
*** 334,372 ****
      .prologue
      const/16 v9, 0xc
  
      const/4 v3, 0x0
  
      .local v3, recCount:I
      const/4 v0, 0x0
  
      .local v0, error:Z
      iget-object v6, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
      invoke-virtual {v6}, Ljava/util/HashMap;->clear()V
  
      :goto_0
      if-le p2, v9, :cond_0
  
      invoke-virtual {p1}, Landroid/os/Parcel;->dataPosition()I
  
      move-result v5
  
      .local v5, start:I
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
  
      .local v4, size:I
      if-gt v4, v9, :cond_2
  
!     const-string v6, "media.Metadata"
  
      const-string v7, "Record is too short"
  
      invoke-static {v6, v7}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
      const/4 v0, 0x1
  
      .end local v4           #size:I
      .end local v5           #start:I
      :cond_0
--- 352,400 ----
      .prologue
      const/16 v9, 0xc
  
+     .line 293
      const/4 v3, 0x0
  
+     .line 294
      .local v3, recCount:I
      const/4 v0, 0x0
  
+     .line 296
      .local v0, error:Z
      iget-object v6, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
      invoke-virtual {v6}, Ljava/util/HashMap;->clear()V
  
+     .line 297
      :goto_0
      if-le p2, v9, :cond_0
  
+     .line 298
      invoke-virtual {p1}, Landroid/os/Parcel;->dataPosition()I
  
      move-result v5
  
+     .line 300
      .local v5, start:I
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
  
+     .line 302
      .local v4, size:I
      if-gt v4, v9, :cond_2
  
!     .line 303
!     const-string/jumbo v6, "media.Metadata"
  
      const-string v7, "Record is too short"
  
      invoke-static {v6, v7}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 304
      const/4 v0, 0x1
  
+     .line 340
      .end local v4           #size:I
      .end local v5           #start:I
      :cond_0
***************
*** 375,382 ****
  
      if-eqz v0, :cond_7
  
      :cond_1
!     const-string v6, "media.Metadata"
  
      new-instance v7, Ljava/lang/StringBuilder;
  
--- 403,411 ----
  
      if-eqz v0, :cond_7
  
+     .line 341
      :cond_1
!     const-string/jumbo v6, "media.Metadata"
  
      new-instance v7, Ljava/lang/StringBuilder;
  
***************
*** 398,412 ****
--- 427,445 ----
  
      invoke-static {v6, v7}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 342
      iget-object v6, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
      invoke-virtual {v6}, Ljava/util/HashMap;->clear()V
  
+     .line 343
      const/4 v6, 0x0
  
+     .line 345
      :goto_2
      return v6
  
+     .line 309
      .restart local v4       #size:I
      .restart local v5       #start:I
      :cond_2
***************
*** 414,419 ****
--- 447,453 ----
  
      move-result v1
  
+     .line 310
      .local v1, metadataId:I
      invoke-direct {p0, v1}, Landroid/media/Metadata;->checkMetadataId(I)Z
  
***************
*** 421,430 ****
--- 455,467 ----
  
      if-nez v6, :cond_3
  
+     .line 311
      const/4 v0, 0x1
  
+     .line 312
      goto :goto_1
  
+     .line 318
      :cond_3
      iget-object v6, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
***************
*** 438,453 ****
  
      if-eqz v6, :cond_4
  
!     const-string v6, "media.Metadata"
  
      const-string v7, "Duplicate metadata ID found"
  
      invoke-static {v6, v7}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
      const/4 v0, 0x1
  
      goto :goto_1
  
      :cond_4
      iget-object v6, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
--- 475,494 ----
  
      if-eqz v6, :cond_4
  
!     .line 319
!     const-string/jumbo v6, "media.Metadata"
  
      const-string v7, "Duplicate metadata ID found"
  
      invoke-static {v6, v7}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 320
      const/4 v0, 0x1
  
+     .line 321
      goto :goto_1
  
+     .line 324
      :cond_4
      iget-object v6, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
***************
*** 465,474 ****
--- 506,517 ----
  
      invoke-virtual {v6, v7, v8}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 327
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
+     .line 328
      .local v2, metadataType:I
      if-lez v2, :cond_5
  
***************
*** 476,483 ****
  
      if-le v2, v6, :cond_6
  
      :cond_5
!     const-string v6, "media.Metadata"
  
      new-instance v7, Ljava/lang/StringBuilder;
  
--- 519,527 ----
  
      if-le v2, v6, :cond_6
  
+     .line 329
      :cond_5
!     const-string/jumbo v6, "media.Metadata"
  
      new-instance v7, Ljava/lang/StringBuilder;
  
***************
*** 499,519 ****
--- 543,570 ----
  
      invoke-static {v6, v7}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 330
      const/4 v0, 0x1
  
+     .line 331
      goto :goto_1
  
+     .line 335
      :cond_6
      add-int v6, v5, v4
  
      invoke-virtual {p1, v6}, Landroid/os/Parcel;->setDataPosition(I)V
  
+     .line 336
      sub-int/2addr p2, v4
  
+     .line 337
      add-int/lit8 v3, v3, 0x1
  
+     .line 338
      goto/16 :goto_0
  
+     .line 345
      .end local v1           #metadataId:I
      .end local v2           #metadataType:I
      .end local v4           #size:I
***************
*** 533,542 ****
--- 584,595 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 451
      const/4 v1, 0x3
  
      invoke-direct {p0, p1, v1}, Landroid/media/Metadata;->checkType(II)V
  
+     .line 452
      iget-object v1, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
***************
*** 559,568 ****
--- 612,623 ----
      .parameter "key"
  
      .prologue
+     .line 477
      const/4 v0, 0x7
  
      invoke-direct {p0, p1, v0}, Landroid/media/Metadata;->checkType(II)V
  
+     .line 478
      iget-object v0, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
      invoke-virtual {v0}, Landroid/os/Parcel;->createByteArray()[B
***************
*** 577,592 ****
--- 632,650 ----
      .parameter "key"
  
      .prologue
+     .line 485
      const/4 v5, 0x6
  
      invoke-direct {p0, p1, v5}, Landroid/media/Metadata;->checkType(II)V
  
+     .line 486
      iget-object v5, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
      invoke-virtual {v5}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v1
  
+     .line 487
      .local v1, timeSinceEpoch:J
      iget-object v5, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
***************
*** 594,599 ****
--- 652,658 ----
  
      move-result-object v3
  
+     .line 489
      .local v3, timeZone:Ljava/lang/String;
      invoke-virtual {v3}, Ljava/lang/String;->length()I
  
***************
*** 601,626 ****
--- 660,691 ----
  
      if-nez v5, :cond_0
  
+     .line 490
      new-instance v5, Ljava/util/Date;
  
      invoke-direct {v5, v1, v2}, Ljava/util/Date;-><init>(J)V
  
+     .line 496
      :goto_0
      return-object v5
  
+     .line 492
      :cond_0
      invoke-static {v3}, Ljava/util/TimeZone;->getTimeZone(Ljava/lang/String;)Ljava/util/TimeZone;
  
      move-result-object v4
  
+     .line 493
      .local v4, tz:Ljava/util/TimeZone;
      invoke-static {v4}, Ljava/util/Calendar;->getInstance(Ljava/util/TimeZone;)Ljava/util/Calendar;
  
      move-result-object v0
  
+     .line 495
      .local v0, cal:Ljava/util/Calendar;
      invoke-virtual {v0, v1, v2}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 496
      invoke-virtual {v0}, Ljava/util/Calendar;->getTime()Ljava/util/Date;
  
      move-result-object v5
***************
*** 633,642 ****
--- 698,709 ----
      .parameter "key"
  
      .prologue
+     .line 469
      const/4 v0, 0x5
  
      invoke-direct {p0, p1, v0}, Landroid/media/Metadata;->checkType(II)V
  
+     .line 470
      iget-object v0, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
      invoke-virtual {v0}, Landroid/os/Parcel;->readDouble()D
***************
*** 651,660 ****
--- 718,729 ----
      .parameter "key"
  
      .prologue
+     .line 443
      const/4 v0, 0x2
  
      invoke-direct {p0, p1, v0}, Landroid/media/Metadata;->checkType(II)V
  
+     .line 444
      iget-object v0, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
      invoke-virtual {v0}, Landroid/os/Parcel;->readInt()I
***************
*** 669,678 ****
--- 738,749 ----
      .parameter "key"
  
      .prologue
+     .line 459
      const/4 v0, 0x4
  
      invoke-direct {p0, p1, v0}, Landroid/media/Metadata;->checkType(II)V
  
+     .line 462
      iget-object v0, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
      invoke-virtual {v0}, Landroid/os/Parcel;->readLong()J
***************
*** 687,696 ****
--- 758,769 ----
      .parameter "key"
  
      .prologue
+     .line 435
      const/4 v0, 0x1
  
      invoke-direct {p0, p1, v0}, Landroid/media/Metadata;->checkType(II)V
  
+     .line 436
      iget-object v0, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
      invoke-virtual {v0}, Landroid/os/Parcel;->readString()Ljava/lang/String;
***************
*** 705,716 ****
--- 778,791 ----
      .parameter "metadataId"
  
      .prologue
+     .line 421
      invoke-direct {p0, p1}, Landroid/media/Metadata;->checkMetadataId(I)Z
  
      move-result v0
  
      if-nez v0, :cond_0
  
+     .line 422
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 735,740 ****
--- 810,816 ----
  
      throw v0
  
+     .line 424
      :cond_0
      iget-object v0, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
***************
*** 762,767 ****
--- 838,844 ----
      .end annotation
  
      .prologue
+     .line 414
      iget-object v0, p0, Landroid/media/Metadata;->mKeyToPosMap:Ljava/util/HashMap;
  
      invoke-virtual {v0}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 780,792 ****
  
      const/4 v3, 0x0
  
      invoke-virtual {p1}, Landroid/os/Parcel;->dataAvail()I
  
      move-result v4
  
      if-ge v4, v5, :cond_0
  
!     const-string v4, "media.Metadata"
  
      new-instance v5, Ljava/lang/StringBuilder;
  
--- 857,871 ----
  
      const/4 v3, 0x0
  
+     .line 378
      invoke-virtual {p1}, Landroid/os/Parcel;->dataAvail()I
  
      move-result v4
  
      if-ge v4, v5, :cond_0
  
!     .line 379
!     const-string/jumbo v4, "media.Metadata"
  
      new-instance v5, Ljava/lang/StringBuilder;
  
***************
*** 812,830 ****
--- 891,913 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 407
      :goto_0
      return v3
  
+     .line 383
      :cond_0
      invoke-virtual {p1}, Landroid/os/Parcel;->dataPosition()I
  
      move-result v1
  
+     .line 384
      .local v1, pin:I
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
+     .line 387
      .local v2, size:I
      invoke-virtual {p1}, Landroid/os/Parcel;->dataAvail()I
  
***************
*** 836,843 ****
  
      if-ge v2, v5, :cond_2
  
      :cond_1
!     const-string v4, "media.Metadata"
  
      new-instance v5, Ljava/lang/StringBuilder;
  
--- 919,927 ----
  
      if-ge v2, v5, :cond_2
  
+     .line 388
      :cond_1
!     const-string/jumbo v4, "media.Metadata"
  
      new-instance v5, Ljava/lang/StringBuilder;
  
***************
*** 883,903 ****
  
      invoke-static {v4, v5}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->setDataPosition(I)V
  
      goto :goto_0
  
      :cond_2
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      .local v0, kShouldBeMetaMarker:I
      const v4, 0x4d455441
  
      if-eq v0, v4, :cond_3
  
!     const-string v4, "media.Metadata"
  
      new-instance v5, Ljava/lang/StringBuilder;
  
--- 967,991 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 389
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->setDataPosition(I)V
  
      goto :goto_0
  
+     .line 394
      :cond_2
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 395
      .local v0, kShouldBeMetaMarker:I
      const v4, 0x4d455441
  
      if-eq v0, v4, :cond_3
  
!     .line 396
!     const-string/jumbo v4, "media.Metadata"
  
      new-instance v5, Ljava/lang/StringBuilder;
  
***************
*** 923,932 ****
--- 1011,1022 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 397
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->setDataPosition(I)V
  
      goto :goto_0
  
+     .line 402
      :cond_3
      add-int/lit8 v4, v2, -0x8
  
***************
*** 936,948 ****
--- 1026,1041 ----
  
      if-nez v4, :cond_4
  
+     .line 403
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->setDataPosition(I)V
  
      goto :goto_0
  
+     .line 406
      :cond_4
      iput-object p1, p0, Landroid/media/Metadata;->mParcel:Landroid/os/Parcel;
  
+     .line 407
      const/4 v3, 0x1
  
      goto/16 :goto_0
