*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/RemoteControlClient$MetadataEditor.smali	2012-03-23 19:35:38.739293295 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/RemoteControlClient$MetadataEditor.smali.original	2012-03-23 19:34:44.755308899 -0500
***************
*** 40,53 ****
--- 40,56 ----
      .parameter
  
      .prologue
+     .line 315
      iput-object p1, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 312
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mApplied:Z
  
+     .line 315
      return-void
  .end method
  
***************
*** 57,62 ****
--- 60,66 ----
      .parameter "x1"
  
      .prologue
+     .line 295
      invoke-direct {p0, p1}, Landroid/media/RemoteControlClient$MetadataEditor;-><init>(Landroid/media/RemoteControlClient;)V
  
      return-void
***************
*** 68,73 ****
--- 72,78 ----
      .locals 4
  
      .prologue
+     .line 447
      monitor-enter p0
  
      :try_start_0
***************
*** 75,80 ****
--- 80,86 ----
  
      if-eqz v0, :cond_0
  
+     .line 448
      const-string v0, "RemoteControlClient"
  
      const-string v1, "Can\'t apply a previously applied MetadataEditor"
***************
*** 83,93 ****
--- 89,101 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_1
  
+     .line 471
      :goto_0
      monitor-exit p0
  
      return-void
  
+     .line 451
      :cond_0
      :try_start_1
      iget-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
***************
*** 101,106 ****
--- 109,115 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_1
  
+     .line 453
      :try_start_2
      iget-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
  
***************
*** 113,118 ****
--- 122,128 ----
      #setter for: Landroid/media/RemoteControlClient;->mMetadata:Landroid/os/Bundle;
      invoke-static {v0, v2}, Landroid/media/RemoteControlClient;->access$702(Landroid/media/RemoteControlClient;Landroid/os/Bundle;)Landroid/os/Bundle;
  
+     .line 454
      iget-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
  
      #getter for: Landroid/media/RemoteControlClient;->mArtwork:Landroid/graphics/Bitmap;
***************
*** 137,142 ****
--- 147,153 ----
  
      if-nez v0, :cond_1
  
+     .line 455
      iget-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
  
      #getter for: Landroid/media/RemoteControlClient;->mArtwork:Landroid/graphics/Bitmap;
***************
*** 146,151 ****
--- 157,163 ----
  
      invoke-virtual {v0}, Landroid/graphics/Bitmap;->recycle()V
  
+     .line 457
      :cond_1
      iget-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
  
***************
*** 154,163 ****
--- 166,177 ----
      #setter for: Landroid/media/RemoteControlClient;->mArtwork:Landroid/graphics/Bitmap;
      invoke-static {v0, v2}, Landroid/media/RemoteControlClient;->access$802(Landroid/media/RemoteControlClient;Landroid/graphics/Bitmap;)Landroid/graphics/Bitmap;
  
+     .line 458
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mEditorArtwork:Landroid/graphics/Bitmap;
  
+     .line 459
      iget-boolean v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mMetadataChanged:Z
  
      iget-boolean v2, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mArtworkChanged:Z
***************
*** 166,182 ****
--- 180,199 ----
  
      if-eqz v0, :cond_3
  
+     .line 461
      iget-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
  
      #calls: Landroid/media/RemoteControlClient;->sendMetadataWithArtwork_syncCacheLock()V
      invoke-static {v0}, Landroid/media/RemoteControlClient;->access$900(Landroid/media/RemoteControlClient;)V
  
+     .line 469
      :cond_2
      :goto_1
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mApplied:Z
  
+     .line 470
      monitor-exit v1
  
      goto :goto_0
***************
*** 193,198 ****
--- 210,216 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_1
  
+     .line 447
      :catchall_1
      move-exception v0
  
***************
*** 200,211 ****
--- 218,231 ----
  
      throw v0
  
+     .line 462
      :cond_3
      :try_start_4
      iget-boolean v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mMetadataChanged:Z
  
      if-eqz v0, :cond_4
  
+     .line 464
      iget-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
  
      #calls: Landroid/media/RemoteControlClient;->sendMetadata_syncCacheLock()V
***************
*** 213,223 ****
--- 233,245 ----
  
      goto :goto_1
  
+     .line 465
      :cond_4
      iget-boolean v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mArtworkChanged:Z
  
      if-eqz v0, :cond_2
  
+     .line 467
      iget-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
  
      #calls: Landroid/media/RemoteControlClient;->sendArtwork_syncCacheLock()V
***************
*** 232,237 ****
--- 254,260 ----
      .locals 2
  
      .prologue
+     .line 432
      monitor-enter p0
  
      :try_start_0
***************
*** 239,244 ****
--- 262,268 ----
  
      if-eqz v0, :cond_0
  
+     .line 433
      const-string v0, "RemoteControlClient"
  
      const-string v1, "Can\'t clear a previously applied MetadataEditor"
***************
*** 247,263 ****
--- 271,290 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 438
      :goto_0
      monitor-exit p0
  
      return-void
  
+     .line 436
      :cond_0
      :try_start_1
      iget-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mEditorMetadata:Landroid/os/Bundle;
  
      invoke-virtual {v0}, Landroid/os/Bundle;->clear()V
  
+     .line 437
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mEditorArtwork:Landroid/graphics/Bitmap;
***************
*** 266,271 ****
--- 293,299 ----
  
      goto :goto_0
  
+     .line 432
      :catchall_0
      move-exception v0
  
***************
*** 283,288 ****
--- 311,317 ----
      .end annotation
  
      .prologue
+     .line 320
      new-instance v0, Ljava/lang/CloneNotSupportedException;
  
      invoke-direct {v0}, Ljava/lang/CloneNotSupportedException;-><init>()V
***************
*** 301,306 ****
--- 330,336 ----
      .end annotation
  
      .prologue
+     .line 409
      monitor-enter p0
  
      :try_start_0
***************
*** 308,313 ****
--- 338,344 ----
  
      if-eqz v1, :cond_0
  
+     .line 410
      const-string v1, "RemoteControlClient"
  
      const-string v2, "Can\'t edit a previously applied MetadataEditor"
***************
*** 318,323 ****
--- 349,355 ----
  
      move-object v0, p0
  
+     .line 424
      .end local p0
      .local v0, this:Landroid/media/RemoteControlClient$MetadataEditor;
      :goto_0
***************
*** 325,330 ****
--- 357,363 ----
  
      return-object v0
  
+     .line 413
      .end local v0           #this:Landroid/media/RemoteControlClient$MetadataEditor;
      .restart local p0
      :cond_0
***************
*** 332,337 ****
--- 365,371 ----
  
      if-eq p1, v1, :cond_1
  
+     .line 414
      :try_start_1
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
***************
*** 359,364 ****
--- 393,399 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 409
      :catchall_0
      move-exception v1
  
***************
*** 366,371 ****
--- 401,407 ----
  
      throw v1
  
+     .line 416
      :cond_1
      :try_start_2
      iget-object v1, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
***************
*** 386,391 ****
--- 422,428 ----
  
      if-lez v1, :cond_2
  
+     .line 417
      iget-object v1, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
  
      iget-object v2, p0, Landroid/media/RemoteControlClient$MetadataEditor;->this$0:Landroid/media/RemoteControlClient;
***************
*** 409,414 ****
--- 446,452 ----
  
      iput-object v1, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mEditorArtwork:Landroid/graphics/Bitmap;
  
+     .line 423
      :goto_1
      const/4 v1, 0x1
  
***************
*** 416,425 ****
--- 454,465 ----
  
      move-object v0, p0
  
+     .line 424
      .end local p0
      .restart local v0       #this:Landroid/media/RemoteControlClient$MetadataEditor;
      goto :goto_0
  
+     .line 421
      .end local v0           #this:Landroid/media/RemoteControlClient$MetadataEditor;
      .restart local p0
      :cond_2
***************
*** 441,446 ****
--- 481,487 ----
      .end annotation
  
      .prologue
+     .line 385
      monitor-enter p0
  
      :try_start_0
***************
*** 448,453 ****
--- 489,495 ----
  
      if-eqz v1, :cond_0
  
+     .line 386
      const-string v1, "RemoteControlClient"
  
      const-string v2, "Can\'t edit a previously applied MetadataEditor"
***************
*** 458,463 ****
--- 500,506 ----
  
      move-object v0, p0
  
+     .line 394
      .end local p0
      .local v0, this:Landroid/media/RemoteControlClient$MetadataEditor;
      :goto_0
***************
*** 465,470 ****
--- 508,514 ----
  
      return-object v0
  
+     .line 389
      .end local v0           #this:Landroid/media/RemoteControlClient$MetadataEditor;
      .restart local p0
      :cond_0
***************
*** 480,485 ****
--- 524,530 ----
  
      if-nez v1, :cond_1
  
+     .line 390
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 506,511 ****
--- 551,557 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 385
      :catchall_0
      move-exception v1
  
***************
*** 513,518 ****
--- 559,565 ----
  
      throw v1
  
+     .line 392
      :cond_1
      :try_start_2
      iget-object v1, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mEditorMetadata:Landroid/os/Bundle;
***************
*** 523,528 ****
--- 570,576 ----
  
      invoke-virtual {v1, v2, p2, p3}, Landroid/os/Bundle;->putLong(Ljava/lang/String;J)V
  
+     .line 393
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mMetadataChanged:Z
***************
*** 531,536 ****
--- 579,585 ----
  
      move-object v0, p0
  
+     .line 394
      .end local p0
      .restart local v0       #this:Landroid/media/RemoteControlClient$MetadataEditor;
      goto :goto_0
***************
*** 547,552 ****
--- 596,602 ----
      .end annotation
  
      .prologue
+     .line 356
      monitor-enter p0
  
      :try_start_0
***************
*** 554,559 ****
--- 604,610 ----
  
      if-eqz v1, :cond_0
  
+     .line 357
      const-string v1, "RemoteControlClient"
  
      const-string v2, "Can\'t edit a previously applied MetadataEditor"
***************
*** 564,569 ****
--- 615,621 ----
  
      move-object v0, p0
  
+     .line 365
      .end local p0
      .local v0, this:Landroid/media/RemoteControlClient$MetadataEditor;
      :goto_0
***************
*** 571,576 ****
--- 623,629 ----
  
      return-object v0
  
+     .line 360
      .end local v0           #this:Landroid/media/RemoteControlClient$MetadataEditor;
      .restart local p0
      :cond_0
***************
*** 586,591 ****
--- 639,645 ----
  
      if-nez v1, :cond_1
  
+     .line 361
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 612,617 ****
--- 666,672 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 356
      :catchall_0
      move-exception v1
  
***************
*** 619,624 ****
--- 674,680 ----
  
      throw v1
  
+     .line 363
      :cond_1
      :try_start_2
      iget-object v1, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mEditorMetadata:Landroid/os/Bundle;
***************
*** 629,634 ****
--- 685,691 ----
  
      invoke-virtual {v1, v2, p2}, Landroid/os/Bundle;->putString(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 364
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/media/RemoteControlClient$MetadataEditor;->mMetadataChanged:Z
***************
*** 637,642 ****
--- 694,700 ----
  
      move-object v0, p0
  
+     .line 365
      .end local p0
      .restart local v0       #this:Landroid/media/RemoteControlClient$MetadataEditor;
      goto :goto_0
