*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/TimedText$Justification.smali	2012-03-23 19:35:33.899205036 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/TimedText$Justification.smali.original	2012-03-23 19:34:44.714308152 -0500
***************
*** 30,42 ****
--- 30,46 ----
      .prologue
      const/4 v0, -0x1
  
+     .line 174
      iput-object p1, p0, Landroid/media/TimedText$Justification;->this$0:Landroid/media/TimedText;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 167
      iput v0, p0, Landroid/media/TimedText$Justification;->horizontalJustification:I
  
+     .line 172
      iput v0, p0, Landroid/media/TimedText$Justification;->verticalJustification:I
  
+     .line 174
      return-void
  .end method
