*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/ToneGenerator.smali	2012-03-23 19:35:38.059280895 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/ToneGenerator.smali.original	2012-03-23 19:34:44.749308790 -0500
***************
*** 218,227 ****
--- 218,230 ----
      .parameter "volume"
  
      .prologue
+     .line 739
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 740
      invoke-direct {p0, p1, p2}, Landroid/media/ToneGenerator;->native_setup(II)V
  
+     .line 741
      return-void
  .end method
  
***************
*** 237,242 ****
--- 240,246 ----
      .locals 0
  
      .prologue
+     .line 879
      invoke-direct {p0}, Landroid/media/ToneGenerator;->native_finalize()V
  
      return-void
***************
*** 250,255 ****
--- 254,260 ----
      .parameter "toneType"
  
      .prologue
+     .line 850
      const/4 v0, -0x1
  
      invoke-virtual {p0, p1, v0}, Landroid/media/ToneGenerator;->startTone(II)Z
