*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/sip/ISipService$Stub$Proxy.smali	2012-03-23 19:37:02.313817275 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/sip/ISipService$Stub$Proxy.smali.original	2012-03-23 19:34:45.508322631 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 167
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 168
      iput-object p1, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 169
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 172
      iget-object v0, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 55,77 ****
--- 59,86 ----
      .end annotation
  
      .prologue
+     .line 230
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 231
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 233
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.sip.ISipService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 234
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 235
      iget-object v2, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x3
***************
*** 80,100 ****
--- 89,115 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 236
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 239
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 240
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 242
      return-void
  
+     .line 239
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 240
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 111,141 ****
--- 126,163 ----
      .end annotation
  
      .prologue
+     .line 297
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 298
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 301
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.sip.ISipService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 302
      if-eqz p1, :cond_0
  
+     .line 303
      const/4 v3, 0x1
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 304
      const/4 v3, 0x0
  
      invoke-virtual {p1, v0, v3}, Landroid/net/sip/SipProfile;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 309
      :goto_0
      if-eqz p2, :cond_1
  
***************
*** 146,151 ****
--- 168,174 ----
      :goto_1
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 310
      iget-object v3, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x7
***************
*** 154,161 ****
--- 177,186 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 311
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 312
      invoke-virtual {v1}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v3
***************
*** 166,178 ****
--- 191,207 ----
  
      move-result-object v2
  
+     .line 315
      .local v2, _result:Landroid/net/sip/ISipSession;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 316
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 318
      return-object v2
  
+     .line 307
      .end local v2           #_result:Landroid/net/sip/ISipSession;
      :cond_0
      const/4 v3, 0x0
***************
*** 184,198 ****
--- 213,230 ----
  
      goto :goto_0
  
+     .line 315
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 316
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
  
+     .line 309
      :cond_1
      const/4 v3, 0x0
  
***************
*** 203,208 ****
--- 235,241 ----
      .locals 1
  
      .prologue
+     .line 176
      const-string v0, "android.net.sip.ISipService"
  
      return-object v0
***************
*** 217,237 ****
--- 250,274 ----
      .end annotation
  
      .prologue
+     .line 340
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 341
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 344
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.sip.ISipService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 345
      iget-object v3, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x9
***************
*** 240,247 ****
--- 277,286 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 346
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 347
      sget-object v3, Landroid/net/sip/SipProfile;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-virtual {v1, v3}, Landroid/os/Parcel;->createTypedArray(Landroid/os/Parcelable$Creator;)[Ljava/lang/Object;
***************
*** 252,270 ****
--- 291,314 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 350
      .local v2, _result:[Landroid/net/sip/SipProfile;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 351
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 353
      return-object v2
  
+     .line 350
      .end local v2           #_result:[Landroid/net/sip/SipProfile;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 351
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 280,302 ****
--- 324,351 ----
      .end annotation
  
      .prologue
+     .line 322
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 323
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 326
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.sip.ISipService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 327
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 328
      iget-object v3, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x8
***************
*** 305,312 ****
--- 354,363 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 329
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 330
      invoke-virtual {v1}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v3
***************
*** 317,335 ****
--- 368,391 ----
  
      move-result-object v2
  
+     .line 333
      .local v2, _result:Landroid/net/sip/ISipSession;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 334
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 336
      return-object v2
  
+     .line 333
      .end local v2           #_result:Landroid/net/sip/ISipSession;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 334
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 347,369 ****
--- 403,430 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 245
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 246
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 249
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.sip.ISipService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 250
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 251
      iget-object v3, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x4
***************
*** 372,379 ****
--- 433,442 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 252
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 253
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 384,403 ****
--- 447,471 ----
  
      const/4 v2, 0x1
  
+     .line 256
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 257
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 259
      return v2
  
+     .line 256
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 257
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 415,437 ****
--- 483,510 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 263
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 264
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 267
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.sip.ISipService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 268
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 269
      iget-object v3, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x5
***************
*** 440,447 ****
--- 513,522 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 270
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 271
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 452,471 ****
--- 527,551 ----
  
      const/4 v2, 0x1
  
+     .line 274
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 275
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 277
      return v2
  
+     .line 274
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 275
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 481,511 ****
--- 561,598 ----
      .end annotation
  
      .prologue
+     .line 180
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 181
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 183
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.sip.ISipService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 184
      if-eqz p1, :cond_0
  
+     .line 185
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 186
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/net/sip/SipProfile;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 191
      :goto_0
      iget-object v2, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 515,530 ****
--- 602,622 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 192
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 195
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 196
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 198
      return-void
  
+     .line 189
      :cond_0
      const/4 v2, 0x0
  
***************
*** 535,545 ****
--- 627,639 ----
  
      goto :goto_0
  
+     .line 195
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 196
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 557,598 ****
--- 651,702 ----
      .end annotation
  
      .prologue
+     .line 201
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 202
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 204
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.sip.ISipService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 205
      if-eqz p1, :cond_0
  
+     .line 206
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 207
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/net/sip/SipProfile;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 212
      :goto_0
      if-eqz p2, :cond_1
  
+     .line 213
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 214
      const/4 v2, 0x0
  
      invoke-virtual {p2, v0, v2}, Landroid/app/PendingIntent;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 219
      :goto_1
      if-eqz p3, :cond_2
  
***************
*** 603,608 ****
--- 707,713 ----
      :goto_2
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 220
      iget-object v2, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x2
***************
*** 611,626 ****
--- 716,736 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 221
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 224
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 225
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 227
      return-void
  
+     .line 210
      :cond_0
      const/4 v2, 0x0
  
***************
*** 631,645 ****
--- 741,758 ----
  
      goto :goto_0
  
+     .line 224
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 225
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
  
+     .line 217
      :cond_1
      const/4 v2, 0x0
  
***************
*** 650,655 ****
--- 763,769 ----
  
      goto :goto_1
  
+     .line 219
      :cond_2
      const/4 v2, 0x0
  
***************
*** 667,689 ****
--- 781,808 ----
      .end annotation
  
      .prologue
+     .line 281
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 282
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 284
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.sip.ISipService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 285
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 286
      if-eqz p2, :cond_0
  
      invoke-interface {p2}, Landroid/net/sip/ISipSessionListener;->asBinder()Landroid/os/IBinder;
***************
*** 693,698 ****
--- 812,818 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 287
      iget-object v2, p0, Landroid/net/sip/ISipService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x6
***************
*** 701,726 ****
--- 821,853 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 288
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 291
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 292
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 294
      return-void
  
+     .line 286
      :cond_0
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 291
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 292
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
