*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/sip/SipAudioCall$1.smali	2012-03-23 19:37:01.926810218 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/sip/SipAudioCall$1.smali.original	2012-03-23 19:34:45.502322521 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 375
      iput-object p1, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      invoke-direct {p0}, Landroid/net/sip/SipSession$Listener;-><init>()V
***************
*** 38,43 ****
--- 39,45 ----
      .parameter "session"
  
      .prologue
+     .line 479
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
***************
*** 46,52 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "sip call busy: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 48,54 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "sip call busy: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 62,67 ****
--- 64,70 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 480
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #getter for: Landroid/net/sip/SipAudioCall;->mListener:Landroid/net/sip/SipAudioCall$Listener;
***************
*** 69,77 ****
--- 72,82 ----
  
      move-result-object v0
  
+     .line 481
      .local v0, listener:Landroid/net/sip/SipAudioCall$Listener;
      if-eqz v0, :cond_0
  
+     .line 483
      :try_start_0
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 79,84 ****
--- 84,90 ----
      :try_end_0
      .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 488
      :cond_0
      :goto_0
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
***************
*** 88,98 ****
--- 94,107 ----
      #calls: Landroid/net/sip/SipAudioCall;->close(Z)V
      invoke-static {v2, v3}, Landroid/net/sip/SipAudioCall;->access$900(Landroid/net/sip/SipAudioCall;Z)V
  
+     .line 489
      return-void
  
+     .line 484
      :catch_0
      move-exception v1
  
+     .line 485
      .local v1, t:Ljava/lang/Throwable;
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
***************
*** 102,108 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "onCallBusy(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 111,117 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "onCallBusy(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 128,133 ****
--- 137,143 ----
      .parameter "message"
  
      .prologue
+     .line 494
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
***************
*** 136,142 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "sip call change failed: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 146,152 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "sip call change failed: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 152,167 ****
--- 162,180 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 495
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #setter for: Landroid/net/sip/SipAudioCall;->mErrorCode:I
      invoke-static {v2, p2}, Landroid/net/sip/SipAudioCall;->access$1002(Landroid/net/sip/SipAudioCall;I)I
  
+     .line 496
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #setter for: Landroid/net/sip/SipAudioCall;->mErrorMessage:Ljava/lang/String;
      invoke-static {v2, p3}, Landroid/net/sip/SipAudioCall;->access$1102(Landroid/net/sip/SipAudioCall;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 497
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #getter for: Landroid/net/sip/SipAudioCall;->mListener:Landroid/net/sip/SipAudioCall$Listener;
***************
*** 169,177 ****
--- 182,192 ----
  
      move-result-object v0
  
+     .line 498
      .local v0, listener:Landroid/net/sip/SipAudioCall$Listener;
      if-eqz v0, :cond_0
  
+     .line 500
      :try_start_0
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 186,198 ****
--- 201,216 ----
      :try_end_0
      .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 506
      :cond_0
      :goto_0
      return-void
  
+     .line 502
      :catch_0
      move-exception v1
  
+     .line 503
      .local v1, t:Ljava/lang/Throwable;
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
***************
*** 202,208 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "onCallBusy(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 220,226 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "onCallBusy(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 226,231 ****
--- 244,250 ----
      .parameter "session"
  
      .prologue
+     .line 455
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
***************
*** 234,240 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "sip call ended: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 253,259 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "sip call ended: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 267,272 ****
--- 286,292 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 457
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #getter for: Landroid/net/sip/SipAudioCall;->mTransferringSession:Landroid/net/sip/SipSession;
***************
*** 276,281 ****
--- 296,302 ----
  
      if-ne p1, v2, :cond_1
  
+     .line 458
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      const/4 v3, 0x0
***************
*** 283,292 ****
--- 304,315 ----
      #setter for: Landroid/net/sip/SipAudioCall;->mTransferringSession:Landroid/net/sip/SipSession;
      invoke-static {v2, v3}, Landroid/net/sip/SipAudioCall;->access$602(Landroid/net/sip/SipAudioCall;Landroid/net/sip/SipSession;)Landroid/net/sip/SipSession;
  
+     .line 475
      :cond_0
      :goto_0
      return-void
  
+     .line 463
      :cond_1
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 306,311 ****
--- 329,335 ----
  
      if-ne p1, v2, :cond_0
  
+     .line 466
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #getter for: Landroid/net/sip/SipAudioCall;->mListener:Landroid/net/sip/SipAudioCall$Listener;
***************
*** 313,321 ****
--- 337,347 ----
  
      move-result-object v0
  
+     .line 467
      .local v0, listener:Landroid/net/sip/SipAudioCall$Listener;
      if-eqz v0, :cond_2
  
+     .line 469
      :try_start_0
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 323,328 ****
--- 349,355 ----
      :try_end_0
      .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 474
      :cond_2
      :goto_1
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
***************
*** 331,339 ****
--- 358,368 ----
  
      goto :goto_0
  
+     .line 470
      :catch_0
      move-exception v1
  
+     .line 471
      .local v1, t:Ljava/lang/Throwable;
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
***************
*** 343,349 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "onCallEnded(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 372,378 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "onCallEnded(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 368,378 ****
--- 397,409 ----
      .parameter "sessionDescription"
  
      .prologue
+     .line 429
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #setter for: Landroid/net/sip/SipAudioCall;->mPeerSd:Ljava/lang/String;
      invoke-static {v2, p2}, Landroid/net/sip/SipAudioCall;->access$502(Landroid/net/sip/SipAudioCall;Ljava/lang/String;)Ljava/lang/String;
  
+     .line 430
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
***************
*** 381,387 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "onCallEstablished()"
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 412,418 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "onCallEstablished()"
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 404,409 ****
--- 435,441 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 433
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #getter for: Landroid/net/sip/SipAudioCall;->mTransferringSession:Landroid/net/sip/SipSession;
***************
*** 422,436 ****
--- 454,471 ----
  
      if-ne p1, v2, :cond_1
  
+     .line 435
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #calls: Landroid/net/sip/SipAudioCall;->transferToNewSession()V
      invoke-static {v2}, Landroid/net/sip/SipAudioCall;->access$700(Landroid/net/sip/SipAudioCall;)V
  
+     .line 451
      :cond_0
      :goto_0
      return-void
  
+     .line 439
      :cond_1
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 439,447 ****
--- 474,484 ----
  
      move-result-object v0
  
+     .line 440
      .local v0, listener:Landroid/net/sip/SipAudioCall$Listener;
      if-eqz v0, :cond_0
  
+     .line 442
      :try_start_0
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 452,457 ****
--- 489,495 ----
  
      if-eqz v2, :cond_2
  
+     .line 443
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      invoke-virtual {v0, v2}, Landroid/net/sip/SipAudioCall$Listener;->onCallHeld(Landroid/net/sip/SipAudioCall;)V
***************
*** 460,468 ****
--- 498,508 ----
  
      goto :goto_0
  
+     .line 447
      :catch_0
      move-exception v1
  
+     .line 448
      .local v1, t:Ljava/lang/Throwable;
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
***************
*** 472,478 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "onCallEstablished(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 512,518 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "onCallEstablished(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 490,495 ****
--- 530,536 ----
  
      goto :goto_0
  
+     .line 445
      .end local v1           #t:Ljava/lang/Throwable;
      :cond_2
      :try_start_1
***************
*** 508,513 ****
--- 549,555 ----
      .parameter "sessionDescription"
  
      .prologue
+     .line 538
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
***************
*** 516,522 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "onCallTransferring mSipSession:"
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 558,564 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "onCallTransferring mSipSession:"
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 549,561 ****
--- 591,606 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 540
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #setter for: Landroid/net/sip/SipAudioCall;->mTransferringSession:Landroid/net/sip/SipSession;
      invoke-static {v2, p1}, Landroid/net/sip/SipAudioCall;->access$602(Landroid/net/sip/SipAudioCall;Landroid/net/sip/SipSession;)Landroid/net/sip/SipSession;
  
+     .line 542
      if-nez p2, :cond_0
  
+     .line 543
      :try_start_0
      invoke-virtual {p1}, Landroid/net/sip/SipSession;->getPeerProfile()Landroid/net/sip/SipProfile;
  
***************
*** 576,584 ****
--- 621,631 ----
  
      invoke-virtual {p1, v2, v3, v4}, Landroid/net/sip/SipSession;->makeCall(Landroid/net/sip/SipProfile;Ljava/lang/String;I)V
  
+     .line 553
      :goto_0
      return-void
  
+     .line 546
      :cond_0
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 591,596 ****
--- 638,644 ----
  
      move-result-object v0
  
+     .line 547
      .local v0, answer:Ljava/lang/String;
      const/4 v2, 0x5
  
***************
*** 600,618 ****
  
      goto :goto_0
  
      .end local v0           #answer:Ljava/lang/String;
      :catch_0
      move-exception v1
  
      .local v1, e:Ljava/lang/Throwable;
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
  
!     const-string v3, "onCallTransferring()"
  
      invoke-static {v2, v3, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
      invoke-virtual {p1}, Landroid/net/sip/SipSession;->endCall()V
  
      goto :goto_0
--- 648,669 ----
  
      goto :goto_0
  
+     .line 549
      .end local v0           #answer:Ljava/lang/String;
      :catch_0
      move-exception v1
  
+     .line 550
      .local v1, e:Ljava/lang/Throwable;
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
  
!     const-string/jumbo v3, "onCallTransferring()"
  
      invoke-static {v2, v3, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 551
      invoke-virtual {p1}, Landroid/net/sip/SipSession;->endCall()V
  
      goto :goto_0
***************
*** 623,628 ****
--- 674,680 ----
      .parameter "session"
  
      .prologue
+     .line 378
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
***************
*** 647,652 ****
--- 699,705 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 379
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #getter for: Landroid/net/sip/SipAudioCall;->mListener:Landroid/net/sip/SipAudioCall$Listener;
***************
*** 654,662 ****
--- 707,717 ----
  
      move-result-object v0
  
+     .line 380
      .local v0, listener:Landroid/net/sip/SipAudioCall$Listener;
      if-eqz v0, :cond_0
  
+     .line 382
      :try_start_0
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 664,676 ****
--- 719,734 ----
      :try_end_0
      .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 387
      :cond_0
      :goto_0
      return-void
  
+     .line 383
      :catch_0
      move-exception v1
  
+     .line 384
      .local v1, t:Ljava/lang/Throwable;
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
***************
*** 680,686 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "onCalling(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 738,744 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "onCalling(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 706,716 ****
--- 764,776 ----
      .parameter "message"
  
      .prologue
+     .line 511
      iget-object v0, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #calls: Landroid/net/sip/SipAudioCall;->onError(ILjava/lang/String;)V
      invoke-static {v0, p2, p3}, Landroid/net/sip/SipAudioCall;->access$1200(Landroid/net/sip/SipAudioCall;ILjava/lang/String;)V
  
+     .line 512
      return-void
  .end method
  
***************
*** 719,724 ****
--- 779,785 ----
      .parameter "session"
  
      .prologue
+     .line 517
      return-void
  .end method
  
***************
*** 728,733 ****
--- 789,795 ----
      .parameter "duration"
  
      .prologue
+     .line 533
      return-void
  .end method
  
***************
*** 738,743 ****
--- 800,806 ----
      .parameter "message"
  
      .prologue
+     .line 528
      return-void
  .end method
  
***************
*** 746,751 ****
--- 809,815 ----
      .parameter "session"
  
      .prologue
+     .line 522
      return-void
  .end method
  
***************
*** 756,765 ****
--- 820,831 ----
      .parameter "sessionDescription"
  
      .prologue
+     .line 406
      iget-object v3, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      monitor-enter v3
  
+     .line 407
      :try_start_0
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 800,815 ****
--- 866,885 ----
  
      if-nez v2, :cond_1
  
+     .line 411
      :cond_0
      invoke-virtual {p1}, Landroid/net/sip/SipSession;->endCall()V
  
+     .line 412
      monitor-exit v3
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 424
      :goto_0
      return-void
  
+     .line 417
      :cond_1
      :try_start_1
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
***************
*** 823,828 ****
--- 893,899 ----
  
      move-result-object v0
  
+     .line 418
      .local v0, answer:Ljava/lang/String;
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 838,843 ****
--- 909,915 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Ljava/lang/Throwable; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 423
      .end local v0           #answer:Ljava/lang/String;
      :goto_1
      :try_start_2
***************
*** 854,872 ****
  
      throw v2
  
      :catch_0
      move-exception v1
  
      .local v1, e:Ljava/lang/Throwable;
      :try_start_3
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
  
!     const-string v4, "onRinging()"
  
      invoke-static {v2, v4, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
      invoke-virtual {p1}, Landroid/net/sip/SipSession;->endCall()V
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
--- 926,947 ----
  
      throw v2
  
+     .line 419
      :catch_0
      move-exception v1
  
+     .line 420
      .local v1, e:Ljava/lang/Throwable;
      :try_start_3
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
  
!     const-string/jumbo v4, "onRinging()"
  
      invoke-static {v2, v4, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 421
      invoke-virtual {p1}, Landroid/net/sip/SipSession;->endCall()V
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
***************
*** 879,884 ****
--- 954,960 ----
      .parameter "session"
  
      .prologue
+     .line 391
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
      move-result-object v2
***************
*** 887,893 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "sip call ringing back: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 963,969 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "sip call ringing back: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 903,908 ****
--- 979,985 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 392
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
      #getter for: Landroid/net/sip/SipAudioCall;->mListener:Landroid/net/sip/SipAudioCall$Listener;
***************
*** 910,918 ****
--- 987,997 ----
  
      move-result-object v0
  
+     .line 393
      .local v0, listener:Landroid/net/sip/SipAudioCall$Listener;
      if-eqz v0, :cond_0
  
+     .line 395
      :try_start_0
      iget-object v2, p0, Landroid/net/sip/SipAudioCall$1;->this$0:Landroid/net/sip/SipAudioCall;
  
***************
*** 920,932 ****
--- 999,1014 ----
      :try_end_0
      .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 400
      :cond_0
      :goto_0
      return-void
  
+     .line 396
      :catch_0
      move-exception v1
  
+     .line 397
      .local v1, t:Ljava/lang/Throwable;
      invoke-static {}, Landroid/net/sip/SipAudioCall;->access$000()Ljava/lang/String;
  
***************
*** 936,942 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "onRingingBack(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1018,1024 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "onRingingBack(): "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
