*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/sip/SipSession$Listener.smali	2012-03-23 19:37:02.256816236 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/sip/SipSession$Listener.smali.original	2012-03-23 19:34:45.507322612 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 0
  
      .prologue
+     .line 111
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 31,36 ****
--- 32,38 ----
      .parameter "session"
  
      .prologue
+     .line 163
      return-void
  .end method
  
***************
*** 41,46 ****
--- 43,49 ----
      .parameter "errorMessage"
  
      .prologue
+     .line 197
      return-void
  .end method
  
***************
*** 49,54 ****
--- 52,58 ----
      .parameter "session"
  
      .prologue
+     .line 155
      return-void
  .end method
  
***************
*** 58,63 ****
--- 62,68 ----
      .parameter "sessionDescription"
  
      .prologue
+     .line 147
      return-void
  .end method
  
***************
*** 67,72 ****
--- 72,78 ----
      .parameter "sessionDescription"
  
      .prologue
+     .line 174
      return-void
  .end method
  
***************
*** 75,80 ****
--- 81,87 ----
      .parameter "session"
  
      .prologue
+     .line 118
      return-void
  .end method
  
***************
*** 85,90 ****
--- 92,98 ----
      .parameter "errorMessage"
  
      .prologue
+     .line 186
      return-void
  .end method
  
***************
*** 93,98 ****
--- 101,107 ----
      .parameter "session"
  
      .prologue
+     .line 205
      return-void
  .end method
  
***************
*** 102,107 ****
--- 111,117 ----
      .parameter "duration"
  
      .prologue
+     .line 214
      return-void
  .end method
  
***************
*** 112,117 ****
--- 122,128 ----
      .parameter "errorMessage"
  
      .prologue
+     .line 225
      return-void
  .end method
  
***************
*** 120,125 ****
--- 131,137 ----
      .parameter "session"
  
      .prologue
+     .line 233
      return-void
  .end method
  
***************
*** 130,135 ****
--- 142,148 ----
      .parameter "sessionDescription"
  
      .prologue
+     .line 129
      return-void
  .end method
  
***************
*** 138,142 ****
--- 151,156 ----
      .parameter "session"
  
      .prologue
+     .line 137
      return-void
  .end method
