*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/location/GpsNetInitiatedHandler$GpsNiResponse.smali	2012-03-23 19:37:34.341401298 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/location/GpsNetInitiatedHandler$GpsNiResponse.smali.original	2012-03-23 19:34:46.110333607 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 0
  
      .prologue
+     .line 114
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
