*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/AbsSavedState$1.smali	2012-03-23 19:37:20.106141717 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/AbsSavedState$1.smali.original	2012-03-23 19:34:45.643325092 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 1
  
      .prologue
+     .line 27
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/view/AbsSavedState;-><init>(Landroid/view/AbsSavedState$1;)V
