*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ActionMode.smali	2012-03-23 19:37:20.409147243 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ActionMode.smali.original	2012-03-23 19:34:45.646325147 -0500
***************
*** 20,27 ****
--- 20,29 ----
      .locals 0
  
      .prologue
+     .line 25
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 184
      return-void
  .end method
  
***************
*** 46,51 ****
--- 48,54 ----
      .locals 1
  
      .prologue
+     .line 53
      iget-object v0, p0, Landroid/view/ActionMode;->mTag:Ljava/lang/Object;
  
      return-object v0
***************
*** 61,66 ****
--- 64,70 ----
      .locals 1
  
      .prologue
+     .line 164
      const/4 v0, 0x1
  
      return v0
***************
*** 80,87 ****
--- 84,93 ----
      .parameter "tag"
  
      .prologue
+     .line 39
      iput-object p1, p0, Landroid/view/ActionMode;->mTag:Ljava/lang/Object;
  
+     .line 40
      return-void
  .end method
  
