*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/DisplayList.smali	2012-03-23 19:37:25.698243689 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/DisplayList.smali.original	2012-03-23 19:34:45.949330672 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 28
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
