*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/GLES20TextureLayer.smali	2012-03-23 19:37:25.772245038 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/GLES20TextureLayer.smali.original	2012-03-23 19:34:45.952330727 -0500
***************
*** 15,26 ****
--- 15,29 ----
      .parameter "isOpaque"
  
      .prologue
+     .line 32
      invoke-direct {p0}, Landroid/view/GLES20Layer;-><init>()V
  
+     .line 33
      const/4 v1, 0x2
  
      new-array v0, v1, [I
  
+     .line 34
      .local v0, layerInfo:[I
      invoke-static {p1, v0}, Landroid/view/GLES20Canvas;->nCreateTextureLayer(Z[I)I
  
***************
*** 28,43 ****
--- 31,49 ----
  
      iput v1, p0, Landroid/view/GLES20TextureLayer;->mLayer:I
  
+     .line 36
      iget v1, p0, Landroid/view/GLES20TextureLayer;->mLayer:I
  
      if-eqz v1, :cond_0
  
+     .line 37
      const/4 v1, 0x0
  
      aget v1, v0, v1
  
      iput v1, p0, Landroid/view/GLES20TextureLayer;->mTexture:I
  
+     .line 38
      new-instance v1, Landroid/view/GLES20Layer$Finalizer;
  
      iget v2, p0, Landroid/view/GLES20TextureLayer;->mLayer:I
***************
*** 46,54 ****
--- 52,62 ----
  
      iput-object v1, p0, Landroid/view/GLES20TextureLayer;->mFinalizer:Landroid/view/GLES20Layer$Finalizer;
  
+     .line 42
      :goto_0
      return-void
  
+     .line 40
      :cond_0
      const/4 v1, 0x0
  
***************
*** 64,69 ****
--- 72,78 ----
      .parameter "currentCanvas"
  
      .prologue
+     .line 65
      return-void
  .end method
  
***************
*** 71,76 ****
--- 80,86 ----
      .locals 1
  
      .prologue
+     .line 55
      const/4 v0, 0x0
  
      return-object v0
***************
*** 80,89 ****
--- 90,101 ----
      .locals 3
  
      .prologue
+     .line 68
      iget-object v0, p0, Landroid/view/GLES20TextureLayer;->mSurface:Landroid/graphics/SurfaceTexture;
  
      if-nez v0, :cond_0
  
+     .line 69
      new-instance v0, Landroid/graphics/SurfaceTexture;
  
      iget v1, p0, Landroid/view/GLES20TextureLayer;->mTexture:I
***************
*** 94,99 ****
--- 106,112 ----
  
      iput-object v0, p0, Landroid/view/GLES20TextureLayer;->mSurface:Landroid/graphics/SurfaceTexture;
  
+     .line 71
      :cond_0
      iget-object v0, p0, Landroid/view/GLES20TextureLayer;->mSurface:Landroid/graphics/SurfaceTexture;
  
***************
*** 104,109 ****
--- 117,123 ----
      .locals 1
  
      .prologue
+     .line 46
      iget v0, p0, Landroid/view/GLES20TextureLayer;->mLayer:I
  
      if-eqz v0, :cond_0
***************
*** 129,134 ****
--- 143,149 ----
      .parameter "height"
  
      .prologue
+     .line 51
      return-void
  .end method
  
***************
*** 137,148 ****
--- 152,165 ----
      .parameter "matrix"
  
      .prologue
+     .line 82
      iget v0, p0, Landroid/view/GLES20TextureLayer;->mLayer:I
  
      iget v1, p1, Landroid/graphics/Matrix;->native_instance:I
  
      invoke-static {v0, v1}, Landroid/view/GLES20Canvas;->nSetTextureLayerTransform(II)V
  
+     .line 83
      return-void
  .end method
  
***************
*** 151,156 ****
--- 168,174 ----
      .parameter "currentCanvas"
  
      .prologue
+     .line 60
      const/4 v0, 0x0
  
      return-object v0
***************
*** 163,175 ****
--- 181,196 ----
      .parameter "isOpaque"
  
      .prologue
+     .line 76
      invoke-super {p0, p1, p2, p3}, Landroid/view/GLES20Layer;->update(IIZ)V
  
+     .line 77
      iget v0, p0, Landroid/view/GLES20TextureLayer;->mLayer:I
  
      iget-object v1, p0, Landroid/view/GLES20TextureLayer;->mSurface:Landroid/graphics/SurfaceTexture;
  
      invoke-static {v0, p1, p2, p3, v1}, Landroid/view/GLES20Canvas;->nUpdateTextureLayer(IIIZLandroid/graphics/SurfaceTexture;)V
  
+     .line 78
      return-void
  .end method
