*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/GLES20Canvas.smali	2012-03-23 19:37:20.356146276 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/GLES20Canvas.smali.original	2012-03-23 19:34:45.645325129 -0500
***************
*** 54,59 ****
--- 54,60 ----
      .locals 1
  
      .prologue
+     .line 72
      invoke-static {}, Landroid/view/GLES20Canvas;->nIsAvailable()Z
  
      move-result v0
***************
*** 69,94 ****
--- 70,100 ----
      .parameter "translucent"
  
      .prologue
+     .line 92
      invoke-direct {p0}, Landroid/view/HardwareCanvas;-><init>()V
  
+     .line 60
      const/4 v0, 0x2
  
      new-array v0, v0, [F
  
      iput-object v0, p0, Landroid/view/GLES20Canvas;->mPoint:[F
  
+     .line 61
      const/4 v0, 0x4
  
      new-array v0, v0, [F
  
      iput-object v0, p0, Landroid/view/GLES20Canvas;->mLine:[F
  
+     .line 63
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/view/GLES20Canvas;->mClipBounds:Landroid/graphics/Rect;
  
+     .line 93
      if-nez p2, :cond_0
  
      const/4 v0, 0x1
***************
*** 96,111 ****
--- 102,121 ----
      :goto_0
      iput-boolean v0, p0, Landroid/view/GLES20Canvas;->mOpaque:Z
  
+     .line 94
      invoke-static {p1}, Landroid/view/GLES20Canvas;->nCreateLayerRenderer(I)I
  
      move-result v0
  
      iput v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
+     .line 95
      invoke-direct {p0}, Landroid/view/GLES20Canvas;->setupFinalizer()V
  
+     .line 96
      return-void
  
+     .line 93
      :cond_0
      const/4 v0, 0x0
  
***************
*** 117,126 ****
--- 127,138 ----
      .parameter "translucent"
  
      .prologue
+     .line 86
      const/4 v0, 0x0
  
      invoke-direct {p0, v0, p1}, Landroid/view/GLES20Canvas;-><init>(ZZ)V
  
+     .line 87
      return-void
  .end method
  
***************
*** 130,155 ****
--- 142,172 ----
      .parameter "translucent"
  
      .prologue
+     .line 98
      invoke-direct {p0}, Landroid/view/HardwareCanvas;-><init>()V
  
+     .line 60
      const/4 v0, 0x2
  
      new-array v0, v0, [F
  
      iput-object v0, p0, Landroid/view/GLES20Canvas;->mPoint:[F
  
+     .line 61
      const/4 v0, 0x4
  
      new-array v0, v0, [F
  
      iput-object v0, p0, Landroid/view/GLES20Canvas;->mLine:[F
  
+     .line 63
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/view/GLES20Canvas;->mClipBounds:Landroid/graphics/Rect;
  
+     .line 99
      if-nez p2, :cond_0
  
      const/4 v0, 0x1
***************
*** 157,180 ****
--- 174,203 ----
      :goto_0
      iput-boolean v0, p0, Landroid/view/GLES20Canvas;->mOpaque:Z
  
+     .line 101
      if-eqz p1, :cond_1
  
+     .line 102
      invoke-static {}, Landroid/view/GLES20Canvas;->nCreateDisplayListRenderer()I
  
      move-result v0
  
      iput v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
+     .line 107
      :goto_1
      invoke-direct {p0}, Landroid/view/GLES20Canvas;->setupFinalizer()V
  
+     .line 108
      return-void
  
+     .line 99
      :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 104
      :cond_1
      invoke-static {}, Landroid/view/GLES20Canvas;->nCreateRenderer()I
  
***************
*** 190,195 ****
--- 213,219 ----
      .parameter "x0"
  
      .prologue
+     .line 42
      invoke-static {p0}, Landroid/view/GLES20Canvas;->nDestroyRenderer(I)V
  
      return-void
***************
*** 200,207 ****
--- 224,233 ----
      .parameter "displayList"
  
      .prologue
+     .line 350
      invoke-static {p0}, Landroid/view/GLES20Canvas;->nDestroyDisplayList(I)V
  
+     .line 351
      return-void
  .end method
  
***************
*** 209,216 ****
--- 235,244 ----
      .locals 0
  
      .prologue
+     .line 243
      invoke-static {}, Landroid/view/GLES20Canvas;->nDisableVsync()V
  
+     .line 244
      return-void
  .end method
  
***************
*** 219,226 ****
--- 247,256 ----
      .parameter "level"
  
      .prologue
+     .line 313
      invoke-static {p0}, Landroid/view/GLES20Canvas;->nFlushCaches(I)V
  
+     .line 314
      return-void
  .end method
  
***************
*** 229,234 ****
--- 259,265 ----
      .parameter "displayList"
  
      .prologue
+     .line 356
      invoke-static {p0}, Landroid/view/GLES20Canvas;->nGetDisplayListSize(I)I
  
      move-result v0
***************
*** 240,247 ****
--- 271,280 ----
      .locals 0
  
      .prologue
+     .line 334
      invoke-static {}, Landroid/view/GLES20Canvas;->nInitCaches()V
  
+     .line 335
      return-void
  .end method
  
***************
*** 249,254 ****
--- 282,288 ----
      .locals 1
  
      .prologue
+     .line 75
      sget-boolean v0, Landroid/view/GLES20Canvas;->sIsAvailable:Z
  
      return v0
***************
*** 258,263 ****
--- 292,298 ----
      .locals 1
  
      .prologue
+     .line 232
      invoke-static {}, Landroid/view/GLES20Canvas;->nIsBackBufferPreserved()Z
  
      move-result v0
***************
*** 500,505 ****
--- 535,541 ----
      .locals 1
  
      .prologue
+     .line 217
      invoke-static {}, Landroid/view/GLES20Canvas;->nPreserveBackBuffer()Z
  
      move-result v0
***************
*** 512,532 ****
--- 548,573 ----
      .parameter "paint"
  
      .prologue
+     .line 1157
      invoke-virtual {p1}, Landroid/graphics/Paint;->getColorFilter()Landroid/graphics/ColorFilter;
  
      move-result-object v0
  
+     .line 1158
      .local v0, filter:Landroid/graphics/ColorFilter;
      if-eqz v0, :cond_0
  
+     .line 1159
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v2, v0, Landroid/graphics/ColorFilter;->nativeColorFilter:I
  
      invoke-static {v1, v2}, Landroid/view/GLES20Canvas;->nSetupColorFilter(II)V
  
+     .line 1160
      const/4 v1, 0x4
  
+     .line 1162
      :goto_0
      return v1
  
***************
*** 540,549 ****
--- 581,592 ----
      .locals 2
  
      .prologue
+     .line 111
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      if-nez v0, :cond_0
  
+     .line 112
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "Could not create GLES20Canvas renderer"
***************
*** 552,557 ****
--- 595,601 ----
  
      throw v0
  
+     .line 114
      :cond_0
      new-instance v0, Landroid/view/GLES20Canvas$CanvasFinalizer;
  
***************
*** 561,566 ****
--- 605,611 ----
  
      iput-object v0, p0, Landroid/view/GLES20Canvas;->mFinalizer:Landroid/view/GLES20Canvas$CanvasFinalizer;
  
+     .line 116
      return-void
  .end method
  
***************
*** 570,575 ****
--- 615,621 ----
      .parameter "paint"
  
      .prologue
+     .line 1119
      invoke-virtual {p1}, Landroid/graphics/Bitmap;->getConfig()Landroid/graphics/Bitmap$Config;
  
      move-result-object v1
***************
*** 578,608 ****
--- 624,661 ----
  
      if-ne v1, v2, :cond_0
  
+     .line 1120
      invoke-direct {p0, p2}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v1
  
+     .line 1129
      :goto_0
      return v1
  
+     .line 1123
      :cond_0
      invoke-virtual {p2}, Landroid/graphics/Paint;->getColorFilter()Landroid/graphics/ColorFilter;
  
      move-result-object v0
  
+     .line 1124
      .local v0, filter:Landroid/graphics/ColorFilter;
      if-eqz v0, :cond_1
  
+     .line 1125
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v2, v0, Landroid/graphics/ColorFilter;->nativeColorFilter:I
  
      invoke-static {v1, v2}, Landroid/view/GLES20Canvas;->nSetupColorFilter(II)V
  
+     .line 1126
      const/4 v1, 0x4
  
      goto :goto_0
  
+     .line 1129
      :cond_1
      const/4 v1, 0x0
  
***************
*** 614,626 ****
--- 667,682 ----
      .parameter "paint"
  
      .prologue
+     .line 1133
      const/4 v1, 0x0
  
+     .line 1135
      .local v1, modifiers:I
      iget-boolean v3, p1, Landroid/graphics/Paint;->hasShadow:Z
  
      if-eqz v3, :cond_0
  
+     .line 1136
      iget v3, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v4, p1, Landroid/graphics/Paint;->shadowRadius:F
***************
*** 633,672 ****
--- 689,738 ----
  
      invoke-static {v3, v4, v5, v6, v7}, Landroid/view/GLES20Canvas;->nSetupShadow(IFFFI)V
  
+     .line 1138
      or-int/lit8 v1, v1, 0x1
  
+     .line 1141
      :cond_0
      invoke-virtual {p1}, Landroid/graphics/Paint;->getShader()Landroid/graphics/Shader;
  
      move-result-object v2
  
+     .line 1142
      .local v2, shader:Landroid/graphics/Shader;
      if-eqz v2, :cond_1
  
+     .line 1143
      iget v3, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v4, v2, Landroid/graphics/Shader;->native_shader:I
  
      invoke-static {v3, v4}, Landroid/view/GLES20Canvas;->nSetupShader(II)V
  
+     .line 1144
      or-int/lit8 v1, v1, 0x2
  
+     .line 1147
      :cond_1
      invoke-virtual {p1}, Landroid/graphics/Paint;->getColorFilter()Landroid/graphics/ColorFilter;
  
      move-result-object v0
  
+     .line 1148
      .local v0, filter:Landroid/graphics/ColorFilter;
      if-eqz v0, :cond_2
  
+     .line 1149
      iget v3, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v4, v0, Landroid/graphics/ColorFilter;->nativeColorFilter:I
  
      invoke-static {v3, v4}, Landroid/view/GLES20Canvas;->nSetupColorFilter(II)V
  
+     .line 1150
      or-int/lit8 v1, v1, 0x4
  
+     .line 1153
      :cond_2
      return v1
  .end method
***************
*** 675,682 ****
--- 741,750 ----
      .locals 0
  
      .prologue
+     .line 325
      invoke-static {}, Landroid/view/GLES20Canvas;->nTerminateCaches()V
  
+     .line 326
      return-void
  .end method
  
***************
*** 687,692 ****
--- 755,761 ----
      .parameter "drawGLFunction"
  
      .prologue
+     .line 270
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, p1}, Landroid/view/GLES20Canvas;->nCallDrawGLFunction(II)Z
***************
*** 701,706 ****
--- 770,776 ----
      .parameter "path"
  
      .prologue
+     .line 408
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 714,719 ****
--- 784,790 ----
      .parameter "op"
  
      .prologue
+     .line 413
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 729,734 ****
--- 800,806 ----
      .parameter "bottom"
  
      .prologue
+     .line 418
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      sget-object v1, Landroid/graphics/Region$Op;->INTERSECT:Landroid/graphics/Region$Op;
***************
*** 759,764 ****
--- 831,837 ----
      .parameter "op"
  
      .prologue
+     .line 426
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v5, p5, Landroid/graphics/Region$Op;->nativeInt:I
***************
*** 786,791 ****
--- 859,865 ----
      .parameter "bottom"
  
      .prologue
+     .line 431
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      sget-object v1, Landroid/graphics/Region$Op;->INTERSECT:Landroid/graphics/Region$Op;
***************
*** 812,817 ****
--- 886,892 ----
      .parameter "rect"
  
      .prologue
+     .line 439
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v1, p1, Landroid/graphics/Rect;->left:I
***************
*** 839,844 ****
--- 914,920 ----
      .parameter "op"
  
      .prologue
+     .line 445
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v1, p1, Landroid/graphics/Rect;->left:I
***************
*** 863,868 ****
--- 939,945 ----
      .parameter "rect"
  
      .prologue
+     .line 450
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v1, p1, Landroid/graphics/RectF;->left:F
***************
*** 890,895 ****
--- 967,973 ----
      .parameter "op"
  
      .prologue
+     .line 456
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v1, p1, Landroid/graphics/RectF;->left:F
***************
*** 914,919 ****
--- 992,998 ----
      .parameter "region"
  
      .prologue
+     .line 461
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 927,932 ****
--- 1006,1012 ----
      .parameter "op"
  
      .prologue
+     .line 466
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 939,950 ****
--- 1019,1032 ----
      .parameter "matrix"
  
      .prologue
+     .line 542
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v1, p1, Landroid/graphics/Matrix;->native_instance:I
  
      invoke-static {v0, v1}, Landroid/view/GLES20Canvas;->nConcatMatrix(II)V
  
+     .line 543
      return-void
  .end method
  
***************
*** 956,961 ****
--- 1038,1044 ----
      .parameter "b"
  
      .prologue
+     .line 671
      and-int/lit16 v0, p1, 0xff
  
      shl-int/lit8 v0, v0, 0x18
***************
*** 978,983 ****
--- 1061,1067 ----
  
      invoke-virtual {p0, v0}, Landroid/view/GLES20Canvas;->drawColor(I)V
  
+     .line 672
      return-void
  .end method
  
***************
*** 990,999 ****
--- 1074,1085 ----
      .parameter "paint"
  
      .prologue
+     .line 659
      invoke-direct {p0, p5}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v9
  
+     .line 660
      .local v9, modifiers:I
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 1015,1026 ****
--- 1101,1114 ----
  
      invoke-static/range {v0 .. v8}, Landroid/view/GLES20Canvas;->nDrawArc(IFFFFFFZI)V
  
+     .line 662
      if-eqz v9, :cond_0
  
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v9}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 663
      :cond_0
      return-void
  .end method
***************
*** 1035,1050 ****
--- 1123,1141 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 690
      if-eqz p4, :cond_1
  
      invoke-direct {p0, p1, p4}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Bitmap;Landroid/graphics/Paint;)I
  
      move-result v6
  
+     .line 691
      .local v6, modifiers:I
      :goto_0
      if-nez p4, :cond_2
  
+     .line 692
      .local v5, nativePaint:I
      :goto_1
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 1059,1070 ****
--- 1150,1163 ----
  
      invoke-static/range {v0 .. v5}, Landroid/view/GLES20Canvas;->nDrawBitmap(II[BFFI)V
  
+     .line 693
      if-eqz v6, :cond_0
  
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v6}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 694
      :cond_0
      return-void
  
***************
*** 1073,1080 ****
--- 1166,1175 ----
      :cond_1
      move v6, v5
  
+     .line 690
      goto :goto_0
  
+     .line 691
      .restart local v6       #modifiers:I
      :cond_2
      iget v5, p4, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 1091,1106 ****
--- 1186,1204 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 702
      if-eqz p3, :cond_1
  
      invoke-direct {p0, p1, p3}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Bitmap;Landroid/graphics/Paint;)I
  
      move-result v0
  
+     .line 703
      .local v0, modifiers:I
      :goto_0
      if-nez p3, :cond_2
  
+     .line 704
      .local v1, nativePaint:I
      :goto_1
      iget v2, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 1113,1124 ****
--- 1211,1224 ----
  
      invoke-static {v2, v3, v4, v5, v1}, Landroid/view/GLES20Canvas;->nDrawBitmap(II[BII)V
  
+     .line 706
      if-eqz v0, :cond_0
  
      iget v2, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v2, v0}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 707
      :cond_0
      return-void
  
***************
*** 1127,1134 ****
--- 1227,1236 ----
      :cond_1
      move v0, v1
  
+     .line 702
      goto :goto_0
  
+     .line 703
      .restart local v0       #modifiers:I
      :cond_2
      iget v1, p3, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 1144,1149 ****
--- 1246,1252 ----
      .parameter "paint"
  
      .prologue
+     .line 715
      if-eqz p4, :cond_1
  
      move-object/from16 v0, p0
***************
*** 1156,1186 ****
--- 1259,1295 ----
  
      move-result v17
  
+     .line 716
      .local v17, modifiers:I
      :goto_0
      if-nez p4, :cond_2
  
      const/4 v14, 0x0
  
+     .line 719
      .local v14, nativePaint:I
      :goto_1
      if-nez p2, :cond_3
  
+     .line 720
      const/16 v19, 0x0
  
      .local v19, top:I
      move/from16 v16, v19
  
+     .line 721
      .local v16, left:I
      invoke-virtual/range {p1 .. p1}, Landroid/graphics/Bitmap;->getWidth()I
  
      move-result v18
  
+     .line 722
      .local v18, right:I
      invoke-virtual/range {p1 .. p1}, Landroid/graphics/Bitmap;->getHeight()I
  
      move-result v15
  
+     .line 730
      .local v15, bottom:I
      :goto_2
      move-object/from16 v0, p0
***************
*** 1235,1240 ****
--- 1344,1350 ----
  
      invoke-static/range {v3 .. v14}, Landroid/view/GLES20Canvas;->nDrawBitmap(II[BFFFFFFFFI)V
  
+     .line 732
      if-eqz v17, :cond_0
  
      move-object/from16 v0, p0
***************
*** 1245,1253 ****
--- 1355,1365 ----
  
      invoke-static {v3, v0}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 733
      :cond_0
      return-void
  
+     .line 715
      .end local v14           #nativePaint:I
      .end local v15           #bottom:I
      .end local v16           #left:I
***************
*** 1259,1264 ****
--- 1371,1377 ----
  
      goto :goto_0
  
+     .line 716
      .restart local v17       #modifiers:I
      :cond_2
      move-object/from16 v0, p4
***************
*** 1267,1272 ****
--- 1380,1386 ----
  
      goto :goto_1
  
+     .line 724
      .restart local v14       #nativePaint:I
      :cond_3
      move-object/from16 v0, p2
***************
*** 1275,1280 ****
--- 1389,1395 ----
  
      move/from16 v16, v0
  
+     .line 725
      .restart local v16       #left:I
      move-object/from16 v0, p2
  
***************
*** 1282,1287 ****
--- 1397,1403 ----
  
      move/from16 v18, v0
  
+     .line 726
      .restart local v18       #right:I
      move-object/from16 v0, p2
  
***************
*** 1289,1294 ****
--- 1405,1411 ----
  
      move/from16 v19, v0
  
+     .line 727
      .restart local v19       #top:I
      move-object/from16 v0, p2
  
***************
*** 1306,1311 ****
--- 1423,1429 ----
      .parameter "paint"
  
      .prologue
+     .line 738
      if-eqz p4, :cond_1
  
      move-object/from16 v0, p4
***************
*** 1314,1334 ****
--- 1432,1456 ----
  
      move-result v13
  
+     .line 739
      .local v13, modifiers:I
      :goto_0
      if-nez p4, :cond_2
  
      const/4 v12, 0x0
  
+     .line 742
      .local v12, nativePaint:I
      :goto_1
      if-nez p2, :cond_3
  
+     .line 743
      const/4 v5, 0x0
  
      .local v5, top:F
      move v4, v5
  
+     .line 744
      .local v4, left:F
      invoke-virtual {p1}, Landroid/graphics/Bitmap;->getWidth()I
  
***************
*** 1336,1341 ****
--- 1458,1464 ----
  
      int-to-float v6, v1
  
+     .line 745
      .local v6, right:F
      invoke-virtual {p1}, Landroid/graphics/Bitmap;->getHeight()I
  
***************
*** 1343,1348 ****
--- 1466,1472 ----
  
      int-to-float v7, v1
  
+     .line 753
      .local v7, bottom:F
      :goto_2
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 1369,1383 ****
--- 1493,1510 ----
  
      invoke-static/range {v1 .. v12}, Landroid/view/GLES20Canvas;->nDrawBitmap(II[BFFFFFFFFI)V
  
+     .line 755
      if-eqz v13, :cond_0
  
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v1, v13}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 756
      :cond_0
      return-void
  
+     .line 738
      .end local v4           #left:F
      .end local v5           #top:F
      .end local v6           #right:F
***************
*** 1389,1394 ****
--- 1516,1522 ----
  
      goto :goto_0
  
+     .line 739
      .restart local v13       #modifiers:I
      :cond_2
      move-object/from16 v0, p4
***************
*** 1397,1402 ****
--- 1525,1531 ----
  
      goto :goto_1
  
+     .line 747
      .restart local v12       #nativePaint:I
      :cond_3
      move-object/from16 v0, p2
***************
*** 1405,1410 ****
--- 1534,1540 ----
  
      int-to-float v4, v1
  
+     .line 748
      .restart local v4       #left:F
      move-object/from16 v0, p2
  
***************
*** 1412,1417 ****
--- 1542,1548 ----
  
      int-to-float v6, v1
  
+     .line 749
      .restart local v6       #right:F
      move-object/from16 v0, p2
  
***************
*** 1419,1424 ****
--- 1550,1556 ----
  
      int-to-float v5, v1
  
+     .line 750
      .restart local v5       #top:F
      move-object/from16 v0, p2
  
***************
*** 1443,1448 ****
--- 1575,1581 ----
      .parameter "paint"
  
      .prologue
+     .line 766
      if-eqz p9, :cond_1
  
      move-object/from16 v0, p9
***************
*** 1451,1456 ****
--- 1584,1590 ----
  
      move-result v14
  
+     .line 767
      .local v14, modifier:I
      :goto_0
      if-eqz p8, :cond_2
***************
*** 1469,1483 ****
--- 1603,1620 ----
  
      move/from16 v5, p7
  
+     .line 768
      invoke-static/range {v1 .. v6}, Landroid/graphics/Bitmap;->createBitmap([IIIIILandroid/graphics/Bitmap$Config;)Landroid/graphics/Bitmap;
  
      move-result-object v13
  
+     .line 769
      .local v13, b:Landroid/graphics/Bitmap;
      if-nez p9, :cond_3
  
      const/4 v12, 0x0
  
+     .line 770
      .local v12, nativePaint:I
      :goto_2
      iget v7, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 1492,1508 ****
--- 1629,1649 ----
  
      invoke-static/range {v7 .. v12}, Landroid/view/GLES20Canvas;->nDrawBitmap(II[BFFI)V
  
+     .line 771
      invoke-virtual {v13}, Landroid/graphics/Bitmap;->recycle()V
  
+     .line 772
      if-eqz v14, :cond_0
  
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v1, v14}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 773
      :cond_0
      return-void
  
+     .line 766
      .end local v6           #config:Landroid/graphics/Bitmap$Config;
      .end local v12           #nativePaint:I
      .end local v13           #b:Landroid/graphics/Bitmap;
***************
*** 1512,1523 ****
--- 1653,1666 ----
  
      goto :goto_0
  
+     .line 767
      .restart local v14       #modifier:I
      :cond_2
      sget-object v6, Landroid/graphics/Bitmap$Config;->RGB_565:Landroid/graphics/Bitmap$Config;
  
      goto :goto_1
  
+     .line 769
      .restart local v6       #config:Landroid/graphics/Bitmap$Config;
      .restart local v13       #b:Landroid/graphics/Bitmap;
      :cond_3
***************
*** 1541,1546 ****
--- 1684,1690 ----
      .parameter "paint"
  
      .prologue
+     .line 779
      int-to-float v4, p4
  
      int-to-float v5, p5
***************
*** 1563,1568 ****
--- 1707,1713 ----
  
      invoke-virtual/range {v0 .. v9}, Landroid/view/GLES20Canvas;->drawBitmap([IIIFFIIZLandroid/graphics/Paint;)V
  
+     .line 780
      return-void
  .end method
  
***************
*** 1578,1583 ****
--- 1723,1729 ----
      .parameter "paint"
  
      .prologue
+     .line 785
      if-ltz p2, :cond_0
  
      if-ltz p3, :cond_0
***************
*** 1586,1591 ****
--- 1732,1738 ----
  
      if-gez p7, :cond_1
  
+     .line 786
      :cond_0
      new-instance v1, Ljava/lang/ArrayIndexOutOfBoundsException;
  
***************
*** 1593,1607 ****
--- 1740,1757 ----
  
      throw v1
  
+     .line 789
      :cond_1
      if-eqz p2, :cond_2
  
      if-nez p3, :cond_3
  
+     .line 805
      :cond_2
      :goto_0
      return-void
  
+     .line 793
      :cond_3
      add-int/lit8 v1, p2, 0x1
  
***************
*** 1609,1614 ****
--- 1759,1765 ----
  
      mul-int v11, v1, v2
  
+     .line 794
      .local v11, count:I
      move-object/from16 v0, p4
  
***************
*** 1620,1629 ****
--- 1771,1783 ----
  
      invoke-static {v1, v0, v2}, Landroid/view/GLES20Canvas;->checkRange(III)V
  
+     .line 797
      const/16 p6, 0x0
  
+     .line 798
      const/16 p7, 0x0
  
+     .line 800
      if-eqz p8, :cond_4
  
      move-object/from16 v0, p8
***************
*** 1632,1643 ****
--- 1786,1799 ----
  
      move-result v12
  
+     .line 801
      .local v12, modifiers:I
      :goto_1
      if-nez p8, :cond_5
  
      const/4 v10, 0x0
  
+     .line 802
      .local v10, nativePaint:I
      :goto_2
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 1660,1665 ****
--- 1816,1822 ----
  
      invoke-static/range {v1 .. v10}, Landroid/view/GLES20Canvas;->nDrawBitmapMesh(II[BII[FI[III)V
  
+     .line 804
      if-eqz v12, :cond_2
  
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 1668,1673 ****
--- 1825,1831 ----
  
      goto :goto_0
  
+     .line 800
      .end local v10           #nativePaint:I
      .end local v12           #modifiers:I
      :cond_4
***************
*** 1675,1680 ****
--- 1833,1839 ----
  
      goto :goto_1
  
+     .line 801
      .restart local v12       #modifiers:I
      :cond_5
      move-object/from16 v0, p8
***************
*** 1692,1701 ****
--- 1851,1862 ----
      .parameter "paint"
  
      .prologue
+     .line 813
      invoke-direct {p0, p4}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v0
  
+     .line 814
      .local v0, modifiers:I
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 1703,1714 ****
--- 1864,1877 ----
  
      invoke-static {v1, p1, p2, p3, v2}, Landroid/view/GLES20Canvas;->nDrawCircle(IFFFI)V
  
+     .line 815
      if-eqz v0, :cond_0
  
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v1, v0}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 816
      :cond_0
      return-void
  .end method
***************
*** 1718,1727 ****
--- 1881,1892 ----
      .parameter "color"
  
      .prologue
+     .line 823
      sget-object v0, Landroid/graphics/PorterDuff$Mode;->SRC_OVER:Landroid/graphics/PorterDuff$Mode;
  
      invoke-virtual {p0, p1, v0}, Landroid/view/GLES20Canvas;->drawColor(ILandroid/graphics/PorterDuff$Mode;)V
  
+     .line 824
      return-void
  .end method
  
***************
*** 1731,1742 ****
--- 1896,1909 ----
      .parameter "mode"
  
      .prologue
+     .line 828
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v1, p2, Landroid/graphics/PorterDuff$Mode;->nativeInt:I
  
      invoke-static {v0, p1, v1}, Landroid/view/GLES20Canvas;->nDrawColor(III)V
  
+     .line 829
      return-void
  .end method
  
***************
*** 1748,1753 ****
--- 1915,1921 ----
      .parameter "dirty"
  
      .prologue
+     .line 363
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      check-cast p1, Landroid/view/GLES20DisplayList;
***************
*** 1774,1783 ****
--- 1942,1953 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 382
      move-object v0, p1
  
      check-cast v0, Landroid/view/GLES20Layer;
  
+     .line 383
      .local v0, glLayer:Landroid/view/GLES20Layer;
      if-eqz p4, :cond_1
  
***************
*** 1785,1794 ****
--- 1955,1966 ----
  
      move-result v1
  
+     .line 384
      .local v1, modifier:I
      :goto_0
      if-nez p4, :cond_2
  
+     .line 385
      .local v2, nativePaint:I
      :goto_1
      iget v3, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 1799,1810 ****
--- 1971,1984 ----
  
      invoke-static {v3, v4, p2, p3, v2}, Landroid/view/GLES20Canvas;->nDrawLayer(IIFFI)V
  
+     .line 386
      if-eqz v1, :cond_0
  
      iget v3, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v3, v1}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 387
      :cond_0
      return-void
  
***************
*** 1813,1820 ****
--- 1987,1996 ----
      :cond_1
      move v1, v2
  
+     .line 383
      goto :goto_0
  
+     .line 384
      .restart local v1       #modifier:I
      :cond_2
      iget v2, p4, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 1833,1866 ****
--- 2009,2048 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 835
      iget-object v0, p0, Landroid/view/GLES20Canvas;->mLine:[F
  
      aput p1, v0, v2
  
+     .line 836
      iget-object v0, p0, Landroid/view/GLES20Canvas;->mLine:[F
  
      const/4 v1, 0x1
  
      aput p2, v0, v1
  
+     .line 837
      iget-object v0, p0, Landroid/view/GLES20Canvas;->mLine:[F
  
      const/4 v1, 0x2
  
      aput p3, v0, v1
  
+     .line 838
      iget-object v0, p0, Landroid/view/GLES20Canvas;->mLine:[F
  
      const/4 v1, 0x3
  
      aput p4, v0, v1
  
+     .line 839
      iget-object v0, p0, Landroid/view/GLES20Canvas;->mLine:[F
  
      const/4 v1, 0x4
  
      invoke-virtual {p0, v0, v2, v1, p5}, Landroid/view/GLES20Canvas;->drawLines([FIILandroid/graphics/Paint;)V
  
+     .line 840
      return-void
  .end method
  
***************
*** 1872,1877 ****
--- 2054,2060 ----
      .parameter "paint"
  
      .prologue
+     .line 844
      or-int v1, p2, p3
  
      if-ltz v1, :cond_0
***************
*** 1882,1887 ****
--- 2065,2071 ----
  
      if-le v1, v2, :cond_1
  
+     .line 845
      :cond_0
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
***************
*** 1891,1901 ****
--- 2075,2087 ----
  
      throw v1
  
+     .line 847
      :cond_1
      invoke-direct {p0, p4}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v0
  
+     .line 848
      .local v0, modifiers:I
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 1903,1914 ****
--- 2089,2102 ----
  
      invoke-static {v1, p1, p2, p3, v2}, Landroid/view/GLES20Canvas;->nDrawLines(I[FIII)V
  
+     .line 849
      if-eqz v0, :cond_2
  
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v1, v0}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 850
      :cond_2
      return-void
  .end method
***************
*** 1919,1930 ****
--- 2107,2120 ----
      .parameter "paint"
  
      .prologue
+     .line 857
      const/4 v0, 0x0
  
      array-length v1, p1
  
      invoke-virtual {p0, p1, v0, v1, p2}, Landroid/view/GLES20Canvas;->drawLines([FIILandroid/graphics/Paint;)V
  
+     .line 858
      return-void
  .end method
  
***************
*** 1934,1943 ****
--- 2124,2135 ----
      .parameter "paint"
  
      .prologue
+     .line 862
      invoke-direct {p0, p2}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v6
  
+     .line 863
      .local v6, modifiers:I
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 1953,1964 ****
--- 2145,2158 ----
  
      invoke-static/range {v0 .. v5}, Landroid/view/GLES20Canvas;->nDrawOval(IFFFFI)V
  
+     .line 864
      if-eqz v6, :cond_0
  
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v6}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 865
      :cond_0
      return-void
  .end method
***************
*** 1968,1980 ****
--- 2162,2177 ----
      .parameter "paint"
  
      .prologue
+     .line 872
      iget-object v6, p0, Landroid/view/GLES20Canvas;->mClipBounds:Landroid/graphics/Rect;
  
+     .line 873
      .local v6, r:Landroid/graphics/Rect;
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v6}, Landroid/view/GLES20Canvas;->nGetClipBounds(ILandroid/graphics/Rect;)Z
  
+     .line 874
      iget v0, v6, Landroid/graphics/Rect;->left:I
  
      int-to-float v1, v0
***************
*** 1997,2002 ****
--- 2194,2200 ----
  
      invoke-virtual/range {v0 .. v5}, Landroid/view/GLES20Canvas;->drawRect(FFFFLandroid/graphics/Paint;)V
  
+     .line 875
      return-void
  .end method
  
***************
*** 2010,2025 ****
--- 2208,2226 ----
      .prologue
      const/4 v8, 0x0
  
+     .line 677
      if-eqz p4, :cond_1
  
      invoke-direct {p0, p4}, Landroid/view/GLES20Canvas;->setupColorFilter(Landroid/graphics/Paint;)I
  
      move-result v9
  
+     .line 678
      .local v9, modifier:I
      :goto_0
      if-nez p4, :cond_2
  
+     .line 679
      .local v8, nativePaint:I
      :goto_1
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 2040,2051 ****
--- 2241,2254 ----
  
      invoke-static/range {v0 .. v8}, Landroid/view/GLES20Canvas;->nDrawPatch(II[B[BFFFFI)V
  
+     .line 681
      if-eqz v9, :cond_0
  
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v9}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 682
      :cond_0
      return-void
  
***************
*** 2054,2061 ****
--- 2257,2266 ----
      :cond_1
      move v9, v8
  
+     .line 677
      goto :goto_0
  
+     .line 678
      .restart local v9       #modifier:I
      :cond_2
      iget v8, p4, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 2069,2087 ****
--- 2274,2296 ----
      .parameter "paint"
  
      .prologue
+     .line 879
      invoke-direct {p0, p2}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v0
  
+     .line 880
      .local v0, modifiers:I
      iget-boolean v1, p1, Landroid/graphics/Path;->isSimplePath:Z
  
      if-eqz v1, :cond_2
  
+     .line 881
      iget-object v1, p1, Landroid/graphics/Path;->rects:Landroid/graphics/Region;
  
      if-eqz v1, :cond_0
  
+     .line 882
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget-object v2, p1, Landroid/graphics/Path;->rects:Landroid/graphics/Region;
***************
*** 2092,2097 ****
--- 2301,2307 ----
  
      invoke-static {v1, v2, v3}, Landroid/view/GLES20Canvas;->nDrawRects(III)V
  
+     .line 887
      :cond_0
      :goto_0
      if-eqz v0, :cond_1
***************
*** 2100,2108 ****
--- 2310,2320 ----
  
      invoke-static {v1, v0}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 888
      :cond_1
      return-void
  
+     .line 885
      :cond_2
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 2120,2125 ****
--- 2332,2338 ----
      .parameter "picture"
  
      .prologue
+     .line 895
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 2133,2138 ****
--- 2346,2352 ----
      .parameter "dst"
  
      .prologue
+     .line 900
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 2146,2151 ****
--- 2360,2366 ----
      .parameter "dst"
  
      .prologue
+     .line 905
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 2162,2183 ****
--- 2377,2402 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 910
      iget-object v0, p0, Landroid/view/GLES20Canvas;->mPoint:[F
  
      aput p1, v0, v2
  
+     .line 911
      iget-object v0, p0, Landroid/view/GLES20Canvas;->mPoint:[F
  
      const/4 v1, 0x1
  
      aput p2, v0, v1
  
+     .line 912
      iget-object v0, p0, Landroid/view/GLES20Canvas;->mPoint:[F
  
      const/4 v1, 0x2
  
      invoke-virtual {p0, v0, v2, v1, p3}, Landroid/view/GLES20Canvas;->drawPoints([FIILandroid/graphics/Paint;)V
  
+     .line 913
      return-void
  .end method
  
***************
*** 2189,2198 ****
--- 2408,2419 ----
      .parameter "paint"
  
      .prologue
+     .line 922
      invoke-direct {p0, p4}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v0
  
+     .line 923
      .local v0, modifiers:I
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 2200,2211 ****
--- 2421,2434 ----
  
      invoke-static {v1, p1, p2, p3, v2}, Landroid/view/GLES20Canvas;->nDrawPoints(I[FIII)V
  
+     .line 924
      if-eqz v0, :cond_0
  
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v1, v0}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 925
      :cond_0
      return-void
  .end method
***************
*** 2216,2227 ****
--- 2439,2452 ----
      .parameter "paint"
  
      .prologue
+     .line 917
      const/4 v0, 0x0
  
      array-length v1, p1
  
      invoke-virtual {p0, p1, v0, v1, p2}, Landroid/view/GLES20Canvas;->drawPoints([FIILandroid/graphics/Paint;)V
  
+     .line 918
      return-void
  .end method
  
***************
*** 2232,2237 ****
--- 2457,2463 ----
      .parameter "paint"
  
      .prologue
+     .line 938
      return-void
  .end method
  
***************
*** 2244,2249 ****
--- 2470,2476 ----
      .parameter "paint"
  
      .prologue
+     .line 933
      return-void
  .end method
  
***************
*** 2254,2259 ****
--- 2481,2487 ----
      .parameter "b"
  
      .prologue
+     .line 962
      const/high16 v0, -0x100
  
      and-int/lit16 v1, p1, 0xff
***************
*** 2274,2279 ****
--- 2502,2508 ----
  
      invoke-virtual {p0, v0}, Landroid/view/GLES20Canvas;->drawColor(I)V
  
+     .line 963
      return-void
  .end method
  
***************
*** 2286,2295 ****
--- 2515,2526 ----
      .parameter "paint"
  
      .prologue
+     .line 942
      invoke-direct {p0, p5}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v6
  
+     .line 943
      .local v6, modifiers:I
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 2305,2316 ****
--- 2536,2549 ----
  
      invoke-static/range {v0 .. v5}, Landroid/view/GLES20Canvas;->nDrawRect(IFFFFI)V
  
+     .line 944
      if-eqz v6, :cond_0
  
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v6}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 945
      :cond_0
      return-void
  .end method
***************
*** 2321,2326 ****
--- 2554,2560 ----
      .parameter "paint"
  
      .prologue
+     .line 952
      iget v0, p1, Landroid/graphics/Rect;->left:I
  
      int-to-float v1, v0
***************
*** 2343,2348 ****
--- 2577,2583 ----
  
      invoke-virtual/range {v0 .. v5}, Landroid/view/GLES20Canvas;->drawRect(FFFFLandroid/graphics/Paint;)V
  
+     .line 953
      return-void
  .end method
  
***************
*** 2352,2357 ****
--- 2587,2593 ----
      .parameter "paint"
  
      .prologue
+     .line 957
      iget v1, p1, Landroid/graphics/RectF;->left:F
  
      iget v2, p1, Landroid/graphics/RectF;->top:F
***************
*** 2366,2371 ****
--- 2602,2608 ----
  
      invoke-virtual/range {v0 .. v5}, Landroid/view/GLES20Canvas;->drawRect(FFFFLandroid/graphics/Paint;)V
  
+     .line 958
      return-void
  .end method
  
***************
*** 2377,2386 ****
--- 2614,2625 ----
      .parameter "paint"
  
      .prologue
+     .line 967
      invoke-direct {p0, p4}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v8
  
+     .line 968
      .local v8, modifiers:I
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 2400,2411 ****
--- 2639,2652 ----
  
      invoke-static/range {v0 .. v7}, Landroid/view/GLES20Canvas;->nDrawRoundRect(IFFFFFFI)V
  
+     .line 970
      if-eqz v8, :cond_0
  
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v8}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 971
      :cond_0
      return-void
  .end method
***************
*** 2420,2431 ****
--- 2661,2674 ----
      .parameter "paint"
  
      .prologue
+     .line 995
      move-object/from16 v0, p6
  
      invoke-direct {p0, v0}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v9
  
+     .line 997
      .local v9, modifiers:I
      :try_start_0
      instance-of v1, p1, Ljava/lang/String;
***************
*** 2440,2445 ****
--- 2683,2689 ----
  
      if-eqz v1, :cond_2
  
+     .line 999
      :cond_0
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 2467,2472 ****
--- 2711,2717 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1012
      :goto_0
      if-eqz v9, :cond_1
  
***************
*** 2474,2488 ****
--- 2719,2736 ----
  
      invoke-static {v1, v9}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 1014
      :cond_1
      return-void
  
+     .line 1001
      :cond_2
      :try_start_1
      instance-of v1, p1, Landroid/text/GraphicsOperations;
  
      if-eqz v1, :cond_4
  
+     .line 1002
      move-object v0, p1
  
      check-cast v0, Landroid/text/GraphicsOperations;
***************
*** 2507,2512 ****
--- 2755,2761 ----
  
      goto :goto_0
  
+     .line 1012
      :catchall_0
      move-exception v1
  
***************
*** 2519,2524 ****
--- 2768,2774 ----
      :cond_3
      throw v1
  
+     .line 1005
      :cond_4
      sub-int v1, p3, p2
  
***************
*** 2527,2537 ****
--- 2777,2789 ----
  
      move-result-object v2
  
+     .line 1006
      .local v2, buf:[C
      const/4 v1, 0x0
  
      invoke-static {p1, p2, p3, v2, v1}, Landroid/text/TextUtils;->getChars(Ljava/lang/CharSequence;II[CI)V
  
+     .line 1007
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      const/4 v3, 0x0
***************
*** 2552,2557 ****
--- 2804,2810 ----
  
      invoke-static/range {v1 .. v8}, Landroid/view/GLES20Canvas;->nDrawText(I[CIIFFII)V
  
+     .line 1009
      invoke-static {v2}, Landroid/graphics/TemporaryBuffer;->recycle([C)V
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
***************
*** 2567,2576 ****
--- 2820,2831 ----
      .parameter "paint"
  
      .prologue
+     .line 1035
      invoke-direct {p0, p4}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v8
  
+     .line 1037
      .local v8, modifiers:I
      :try_start_0
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 2595,2609 ****
--- 2850,2867 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1040
      if-eqz v8, :cond_0
  
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v8}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 1042
      :cond_0
      return-void
  
+     .line 1040
      :catchall_0
      move-exception v0
  
***************
*** 2627,2632 ****
--- 2885,2891 ----
      .parameter "paint"
  
      .prologue
+     .line 1018
      or-int v0, p2, p3
  
      sub-int v1, p3, p2
***************
*** 2643,2659 ****
--- 2902,2921 ----
  
      if-gez v0, :cond_0
  
+     .line 1019
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v0}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v0
  
+     .line 1022
      :cond_0
      invoke-direct {p0, p6}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v8
  
+     .line 1024
      .local v8, modifiers:I
      :try_start_0
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 2676,2690 ****
--- 2938,2955 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1026
      if-eqz v8, :cond_1
  
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v8}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 1028
      :cond_1
      return-void
  
+     .line 1026
      :catchall_0
      move-exception v0
  
***************
*** 2708,2713 ****
--- 2973,2979 ----
      .parameter "paint"
  
      .prologue
+     .line 978
      or-int v0, p2, p3
  
      add-int v1, p2, p3
***************
*** 2724,2740 ****
--- 2990,3009 ----
  
      if-gez v0, :cond_0
  
+     .line 979
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v0}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v0
  
+     .line 982
      :cond_0
      invoke-direct {p0, p6}, Landroid/view/GLES20Canvas;->setupModifiers(Landroid/graphics/Paint;)I
  
      move-result v8
  
+     .line 984
      .local v8, modifiers:I
      :try_start_0
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 2757,2771 ****
--- 3026,3043 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 986
      if-eqz v8, :cond_1
  
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, v8}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 988
      :cond_1
      return-void
  
+     .line 986
      :catchall_0
      move-exception v0
  
***************
*** 2788,2793 ****
--- 3060,3066 ----
      .parameter "paint"
  
      .prologue
+     .line 1053
      return-void
  .end method
  
***************
*** 2802,2807 ****
--- 3075,3081 ----
      .parameter "paint"
  
      .prologue
+     .line 1048
      return-void
  .end method
  
***************
*** 2818,2823 ****
--- 3092,3098 ----
      .parameter "paint"
  
      .prologue
+     .line 1080
      or-int v2, p2, p3
  
      sub-int v4, p3, p2
***************
*** 2834,2845 ****
--- 3109,3122 ----
  
      if-gez v2, :cond_0
  
+     .line 1081
      new-instance v2, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v2}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v2
  
+     .line 1084
      :cond_0
      move-object/from16 v0, p9
  
***************
*** 2847,2857 ****
--- 3124,3136 ----
  
      move-result v12
  
+     .line 1086
      .local v12, modifiers:I
      if-nez p8, :cond_3
  
      const/4 v10, 0x0
  
+     .line 1087
      .local v10, flags:I
      :goto_0
      :try_start_0
***************
*** 2867,2872 ****
--- 3146,3152 ----
  
      if-eqz v2, :cond_4
  
+     .line 1089
      :cond_1
      iget v2, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 2894,2899 ****
--- 3174,3180 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1104
      :goto_1
      if-eqz v12, :cond_2
  
***************
*** 2901,2915 ****
--- 3182,3199 ----
  
      invoke-static {v2, v12}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 1106
      :cond_2
      return-void
  
+     .line 1086
      .end local v10           #flags:I
      :cond_3
      const/4 v10, 0x1
  
      goto :goto_0
  
+     .line 1091
      .restart local v10       #flags:I
      :cond_4
      :try_start_1
***************
*** 2917,2922 ****
--- 3201,3207 ----
  
      if-eqz v2, :cond_6
  
+     .line 1092
      move-object v0, p1
  
      check-cast v0, Landroid/text/GraphicsOperations;
***************
*** 2945,2950 ****
--- 3230,3236 ----
  
      goto :goto_1
  
+     .line 1104
      :catchall_0
      move-exception v2
  
***************
*** 2957,2974 ****
--- 3243,3264 ----
      :cond_5
      throw v2
  
+     .line 1095
      :cond_6
      sub-int v7, p5, p4
  
+     .line 1096
      .local v7, contextLen:I
      sub-int v5, p3, p2
  
+     .line 1097
      .local v5, len:I
      :try_start_2
      invoke-static {v7}, Landroid/graphics/TemporaryBuffer;->obtain(I)[C
  
      move-result-object v3
  
+     .line 1098
      .local v3, buf:[C
      const/4 v2, 0x0
  
***************
*** 2978,2983 ****
--- 3268,3274 ----
  
      invoke-static {p1, v0, v1, v3, v2}, Landroid/text/TextUtils;->getChars(Ljava/lang/CharSequence;II[CI)V
  
+     .line 1099
      iget v2, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      sub-int v4, p2, p4
***************
*** 2994,2999 ****
--- 3285,3291 ----
  
      invoke-static/range {v2 .. v11}, Landroid/view/GLES20Canvas;->nDrawTextRun(I[CIIIIFFII)V
  
+     .line 1101
      invoke-static {v3}, Landroid/graphics/TemporaryBuffer;->recycle([C)V
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
***************
*** 3014,3019 ****
--- 3306,3312 ----
      .parameter "paint"
  
      .prologue
+     .line 1058
      or-int v1, p2, p3
  
      array-length v2, p1
***************
*** 3026,3037 ****
--- 3319,3332 ----
  
      if-gez v1, :cond_0
  
+     .line 1059
      new-instance v1, Ljava/lang/IndexOutOfBoundsException;
  
      invoke-direct {v1}, Ljava/lang/IndexOutOfBoundsException;-><init>()V
  
      throw v1
  
+     .line 1061
      :cond_0
      if-eqz p8, :cond_1
  
***************
*** 3041,3046 ****
--- 3336,3342 ----
  
      if-eq v0, v1, :cond_1
  
+     .line 1062
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 3067,3072 ****
--- 3363,3369 ----
  
      throw v1
  
+     .line 1065
      :cond_1
      move-object/from16 v0, p9
  
***************
*** 3074,3079 ****
--- 3371,3377 ----
  
      move-result v11
  
+     .line 1067
      .local v11, modifiers:I
      :try_start_0
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 3102,3116 ****
--- 3400,3417 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1070
      if-eqz v11, :cond_2
  
      iget v1, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v1, v11}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 1072
      :cond_2
      return-void
  
+     .line 1070
      :catchall_0
      move-exception v1
  
***************
*** 3140,3145 ****
--- 3441,3447 ----
      .parameter "paint"
  
      .prologue
+     .line 1116
      return-void
  .end method
  
***************
*** 3148,3153 ****
--- 3450,3456 ----
      .parameter "bounds"
  
      .prologue
+     .line 471
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, p1}, Landroid/view/GLES20Canvas;->nGetClipBounds(ILandroid/graphics/Rect;)Z
***************
*** 3162,3167 ****
--- 3465,3471 ----
      .parameter "displayList"
  
      .prologue
+     .line 344
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, p1}, Landroid/view/GLES20Canvas;->nGetDisplayList(II)I
***************
*** 3175,3180 ****
--- 3479,3485 ----
      .locals 1
  
      .prologue
+     .line 649
      iget-object v0, p0, Landroid/view/GLES20Canvas;->mFilter:Landroid/graphics/DrawFilter;
  
      return-object v0
***************
*** 3184,3189 ****
--- 3489,3495 ----
      .locals 1
  
      .prologue
+     .line 175
      iget v0, p0, Landroid/view/GLES20Canvas;->mHeight:I
  
      return v0
***************
*** 3194,3205 ****
--- 3500,3513 ----
      .parameter "matrix"
  
      .prologue
+     .line 535
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v1, p1, Landroid/graphics/Matrix;->native_instance:I
  
      invoke-static {v0, v1}, Landroid/view/GLES20Canvas;->nGetMatrix(II)V
  
+     .line 536
      return-void
  .end method
  
***************
*** 3207,3212 ****
--- 3515,3521 ----
      .locals 1
  
      .prologue
+     .line 185
      invoke-static {}, Landroid/view/GLES20Canvas;->nGetMaximumTextureHeight()I
  
      move-result v0
***************
*** 3218,3223 ****
--- 3527,3533 ----
      .locals 1
  
      .prologue
+     .line 180
      invoke-static {}, Landroid/view/GLES20Canvas;->nGetMaximumTextureWidth()I
  
      move-result v0
***************
*** 3229,3234 ****
--- 3539,3545 ----
      .locals 1
  
      .prologue
+     .line 633
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0}, Landroid/view/GLES20Canvas;->nGetSaveCount(I)I
***************
*** 3242,3247 ****
--- 3553,3559 ----
      .locals 1
  
      .prologue
+     .line 170
      iget v0, p0, Landroid/view/GLES20Canvas;->mWidth:I
  
      return v0
***************
*** 3251,3260 ****
--- 3563,3574 ----
      .locals 1
  
      .prologue
+     .line 392
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0}, Landroid/view/GLES20Canvas;->nInterrupt(I)V
  
+     .line 393
      return-void
  .end method
  
***************
*** 3262,3267 ****
--- 3576,3582 ----
      .locals 1
  
      .prologue
+     .line 165
      iget-boolean v0, p0, Landroid/view/GLES20Canvas;->mOpaque:Z
  
      return v0
***************
*** 3271,3280 ****
--- 3586,3597 ----
      .locals 1
  
      .prologue
+     .line 263
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0}, Landroid/view/GLES20Canvas;->nFinish(I)V
  
+     .line 264
      return-void
  .end method
  
***************
*** 3283,3290 ****
--- 3600,3609 ----
      .parameter "dirty"
  
      .prologue
+     .line 250
      if-eqz p1, :cond_0
  
+     .line 251
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v1, p1, Landroid/graphics/Rect;->left:I
***************
*** 3299,3307 ****
--- 3618,3628 ----
  
      invoke-static/range {v0 .. v5}, Landroid/view/GLES20Canvas;->nPrepareDirty(IIIIIZ)V
  
+     .line 255
      :goto_0
      return-void
  
+     .line 253
      :cond_0
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
***************
*** 3317,3322 ****
--- 3638,3644 ----
      .parameter "displayList"
  
      .prologue
+     .line 372
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      check-cast p1, Landroid/view/GLES20DisplayList;
***************
*** 3328,3333 ****
--- 3650,3656 ----
  
      invoke-static {v0, v1}, Landroid/view/GLES20Canvas;->nOutputDisplayList(II)V
  
+     .line 373
      return-void
  .end method
  
***************
*** 3340,3345 ****
--- 3663,3669 ----
      .parameter "type"
  
      .prologue
+     .line 478
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v5, p5, Landroid/graphics/Canvas$EdgeType;->nativeInt:I
***************
*** 3365,3370 ****
--- 3689,3695 ----
      .parameter "type"
  
      .prologue
+     .line 486
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 3378,3383 ****
--- 3703,3709 ----
      .parameter "type"
  
      .prologue
+     .line 491
      iget v1, p1, Landroid/graphics/RectF;->left:F
  
      iget v2, p1, Landroid/graphics/RectF;->top:F
***************
*** 3401,3410 ****
--- 3727,3738 ----
      .locals 1
  
      .prologue
+     .line 119
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0}, Landroid/view/GLES20Canvas;->nResetDisplayListRenderer(I)V
  
+     .line 120
      return-void
  .end method
  
***************
*** 3412,3421 ****
--- 3740,3751 ----
      .locals 1
  
      .prologue
+     .line 619
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0}, Landroid/view/GLES20Canvas;->nRestore(I)V
  
+     .line 620
      return-void
  .end method
  
***************
*** 3424,3433 ****
--- 3754,3765 ----
      .parameter "saveCount"
  
      .prologue
+     .line 626
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, p1}, Landroid/view/GLES20Canvas;->nRestoreToCount(II)V
  
+     .line 627
      return-void
  .end method
  
***************
*** 3435,3444 ****
--- 3767,3778 ----
      .locals 1
  
      .prologue
+     .line 396
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0}, Landroid/view/GLES20Canvas;->nResume(I)V
  
+     .line 397
      return-void
  .end method
  
***************
*** 3447,3456 ****
--- 3781,3792 ----
      .parameter "degrees"
  
      .prologue
+     .line 514
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, p1}, Landroid/view/GLES20Canvas;->nRotate(IF)V
  
+     .line 515
      return-void
  .end method
  
***************
*** 3458,3463 ****
--- 3794,3800 ----
      .locals 2
  
      .prologue
+     .line 553
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      const/4 v1, 0x3
***************
*** 3474,3479 ****
--- 3811,3817 ----
      .parameter "saveFlags"
  
      .prologue
+     .line 558
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, p1}, Landroid/view/GLES20Canvas;->nSave(II)I
***************
*** 3495,3500 ****
--- 3833,3839 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 581
      cmpg-float v0, p1, p3
  
      if-gez v0, :cond_3
***************
*** 3503,3518 ****
--- 3842,3860 ----
  
      if-gez v0, :cond_3
  
+     .line 582
      if-eqz p5, :cond_1
  
      invoke-direct {p0, p5}, Landroid/view/GLES20Canvas;->setupColorFilter(Landroid/graphics/Paint;)I
  
      move-result v8
  
+     .line 583
      .local v8, modifier:I
      :goto_0
      if-nez p5, :cond_2
  
+     .line 584
      .local v5, nativePaint:I
      :goto_1
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 3531,3536 ****
--- 3873,3879 ----
  
      move-result v7
  
+     .line 585
      .local v7, count:I
      if-eqz v8, :cond_0
  
***************
*** 3538,3543 ****
--- 3881,3887 ----
  
      invoke-static {v0, v8}, Landroid/view/GLES20Canvas;->nResetModifiers(II)V
  
+     .line 588
      .end local v5           #nativePaint:I
      .end local v7           #count:I
      .end local v8           #modifier:I
***************
*** 3548,3561 ****
--- 3892,3908 ----
      :cond_1
      move v8, v5
  
+     .line 582
      goto :goto_0
  
+     .line 583
      .restart local v8       #modifier:I
      :cond_2
      iget v5, p5, Landroid/graphics/Paint;->mNativePaint:I
  
      goto :goto_1
  
+     .line 588
      .end local v8           #modifier:I
      :cond_3
      invoke-virtual {p0, p6}, Landroid/view/GLES20Canvas;->save(I)I
***************
*** 3574,3581 ****
--- 3921,3930 ----
      .prologue
      const/4 v9, 0x0
  
+     .line 565
      if-eqz p1, :cond_1
  
+     .line 566
      iget v1, p1, Landroid/graphics/RectF;->left:F
  
      iget v2, p1, Landroid/graphics/RectF;->top:F
***************
*** 3594,3603 ****
--- 3943,3954 ----
  
      move-result v7
  
+     .line 573
      :cond_0
      :goto_0
      return v7
  
+     .line 569
      :cond_1
      if-eqz p2, :cond_2
  
***************
*** 3605,3614 ****
--- 3956,3967 ----
  
      move-result v8
  
+     .line 570
      .local v8, modifier:I
      :goto_1
      if-nez p2, :cond_3
  
+     .line 571
      .local v9, nativePaint:I
      :goto_2
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
***************
*** 3617,3622 ****
--- 3970,3976 ----
  
      move-result v7
  
+     .line 572
      .local v7, count:I
      if-eqz v8, :cond_0
  
***************
*** 3632,3639 ****
--- 3986,3995 ----
      :cond_2
      move v8, v9
  
+     .line 569
      goto :goto_1
  
+     .line 570
      .restart local v8       #modifier:I
      :cond_3
      iget v9, p2, Landroid/graphics/Paint;->mNativePaint:I
***************
*** 3651,3656 ****
--- 4007,4013 ----
      .parameter "saveFlags"
  
      .prologue
+     .line 608
      cmpg-float v0, p1, p3
  
      if-gez v0, :cond_0
***************
*** 3659,3664 ****
--- 4016,4022 ----
  
      if-gez v0, :cond_0
  
+     .line 609
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      move v1, p1
***************
*** 3677,3682 ****
--- 4035,4041 ----
  
      move-result v0
  
+     .line 611
      :goto_0
      return v0
  
***************
*** 3695,3702 ****
--- 4054,4063 ----
      .parameter "saveFlags"
  
      .prologue
+     .line 596
      if-eqz p1, :cond_0
  
+     .line 597
      iget v1, p1, Landroid/graphics/RectF;->left:F
  
      iget v2, p1, Landroid/graphics/RectF;->top:F
***************
*** 3715,3720 ****
--- 4076,4082 ----
  
      move-result v0
  
+     .line 600
      :goto_0
      return v0
  
***************
*** 3734,3743 ****
--- 4096,4107 ----
      .parameter "sy"
  
      .prologue
+     .line 521
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, p1, p2}, Landroid/view/GLES20Canvas;->nScale(IFF)V
  
+     .line 522
      return-void
  .end method
  
***************
*** 3746,3753 ****
--- 4110,4119 ----
      .parameter "filter"
  
      .prologue
+     .line 644
      iput-object p1, p0, Landroid/view/GLES20Canvas;->mFilter:Landroid/graphics/DrawFilter;
  
+     .line 645
      return-void
  .end method
  
***************
*** 3756,3767 ****
--- 4122,4135 ----
      .parameter "matrix"
  
      .prologue
+     .line 528
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      iget v1, p1, Landroid/graphics/Matrix;->native_instance:I
  
      invoke-static {v0, v1}, Landroid/view/GLES20Canvas;->nSetMatrix(II)V
  
+     .line 529
      return-void
  .end method
  
***************
*** 3771,3784 ****
--- 4139,4156 ----
      .parameter "height"
  
      .prologue
+     .line 197
      iput p1, p0, Landroid/view/GLES20Canvas;->mWidth:I
  
+     .line 198
      iput p2, p0, Landroid/view/GLES20Canvas;->mHeight:I
  
+     .line 200
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, p1, p2}, Landroid/view/GLES20Canvas;->nSetViewport(III)V
  
+     .line 201
      return-void
  .end method
  
***************
*** 3788,3797 ****
--- 4160,4171 ----
      .parameter "sy"
  
      .prologue
+     .line 507
      iget v0, p0, Landroid/view/GLES20Canvas;->mRenderer:I
  
      invoke-static {v0, p1, p2}, Landroid/view/GLES20Canvas;->nSkew(IFF)V
  
+     .line 508
      return-void
  .end method
  
***************
*** 3803,3808 ****
--- 4177,4183 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 500
      cmpl-float v0, p1, v1
  
      if-nez v0, :cond_0
***************
*** 3816,3821 ****
--- 4191,4197 ----
  
      invoke-static {v0, p1, p2}, Landroid/view/GLES20Canvas;->nTranslate(IFF)V
  
+     .line 501
      :cond_1
      return-void
  .end method
