*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/HardwareLayer.smali	2012-03-23 19:37:22.768190259 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/HardwareLayer.smali.original	2012-03-23 19:34:45.900329779 -0500
***************
*** 22,31 ****
--- 22,33 ----
      .prologue
      const/4 v1, -0x1
  
+     .line 45
      const/4 v0, 0x0
  
      invoke-direct {p0, v1, v1, v0}, Landroid/view/HardwareLayer;-><init>(IIZ)V
  
+     .line 46
      return-void
  .end method
  
***************
*** 36,49 ****
--- 38,56 ----
      .parameter "isOpaque"
  
      .prologue
+     .line 56
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 57
      iput p1, p0, Landroid/view/HardwareLayer;->mWidth:I
  
+     .line 58
      iput p2, p0, Landroid/view/HardwareLayer;->mHeight:I
  
+     .line 59
      iput-boolean p3, p0, Landroid/view/HardwareLayer;->mOpaque:Z
  
+     .line 60
      return-void
  .end method
  
***************
*** 65,70 ****
--- 72,78 ----
      .locals 1
  
      .prologue
+     .line 77
      iget v0, p0, Landroid/view/HardwareLayer;->mHeight:I
  
      return v0
***************
*** 74,79 ****
--- 82,88 ----
      .locals 1
  
      .prologue
+     .line 68
      iget v0, p0, Landroid/view/HardwareLayer;->mWidth:I
  
      return v0
***************
*** 83,88 ****
--- 92,98 ----
      .locals 1
  
      .prologue
+     .line 86
      iget-boolean v0, p0, Landroid/view/HardwareLayer;->mOpaque:Z
  
      return v0
***************
*** 107,117 ****
--- 117,131 ----
      .parameter "isOpaque"
  
      .prologue
+     .line 150
      iput p1, p0, Landroid/view/HardwareLayer;->mWidth:I
  
+     .line 151
      iput p2, p0, Landroid/view/HardwareLayer;->mHeight:I
  
+     .line 152
      iput-boolean p3, p0, Landroid/view/HardwareLayer;->mOpaque:Z
  
+     .line 153
      return-void
  .end method
