*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/KeyEvent$DispatcherState.smali	2012-03-23 19:37:26.171252314 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/KeyEvent$DispatcherState.smali.original	2012-03-23 19:34:45.957330817 -0500
***************
*** 27,34 ****
--- 27,36 ----
      .locals 1
  
      .prologue
+     .line 2603
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 2606
      new-instance v0, Landroid/util/SparseIntArray;
  
      invoke-direct {v0}, Landroid/util/SparseIntArray;-><init>()V
***************
*** 45,54 ****
--- 47,58 ----
      .parameter "event"
  
      .prologue
+     .line 2674
      invoke-virtual {p1}, Landroid/view/KeyEvent;->getKeyCode()I
  
      move-result v1
  
+     .line 2676
      .local v1, keyCode:I
      iget-object v2, p0, Landroid/view/KeyEvent$DispatcherState;->mActiveLongPresses:Landroid/util/SparseIntArray;
  
***************
*** 56,89 ****
--- 60,101 ----
  
      move-result v0
  
+     .line 2677
      .local v0, index:I
      if-ltz v0, :cond_0
  
+     .line 2679
      const/16 v2, 0x120
  
      invoke-static {p1, v2}, Landroid/view/KeyEvent;->access$076(Landroid/view/KeyEvent;I)I
  
+     .line 2680
      iget-object v2, p0, Landroid/view/KeyEvent$DispatcherState;->mActiveLongPresses:Landroid/util/SparseIntArray;
  
      invoke-virtual {v2, v0}, Landroid/util/SparseIntArray;->removeAt(I)V
  
+     .line 2682
      :cond_0
      iget v2, p0, Landroid/view/KeyEvent$DispatcherState;->mDownKeyCode:I
  
      if-ne v2, v1, :cond_1
  
+     .line 2684
      const/16 v2, 0x200
  
      invoke-static {p1, v2}, Landroid/view/KeyEvent;->access$076(Landroid/view/KeyEvent;I)I
  
+     .line 2685
      const/4 v2, 0x0
  
      iput v2, p0, Landroid/view/KeyEvent$DispatcherState;->mDownKeyCode:I
  
+     .line 2686
      const/4 v2, 0x0
  
      iput-object v2, p0, Landroid/view/KeyEvent$DispatcherState;->mDownTarget:Ljava/lang/Object;
  
+     .line 2688
      :cond_1
      return-void
  .end method
***************
*** 93,98 ****
--- 105,111 ----
      .parameter "event"
  
      .prologue
+     .line 2654
      iget v0, p0, Landroid/view/KeyEvent$DispatcherState;->mDownKeyCode:I
  
      invoke-virtual {p1}, Landroid/view/KeyEvent;->getKeyCode()I
***************
*** 117,122 ****
--- 130,136 ----
      .parameter "event"
  
      .prologue
+     .line 2664
      iget-object v0, p0, Landroid/view/KeyEvent$DispatcherState;->mActiveLongPresses:Landroid/util/SparseIntArray;
  
      invoke-virtual {p1}, Landroid/view/KeyEvent;->getKeyCode()I
***************
*** 127,132 ****
--- 141,147 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/util/SparseIntArray;->put(II)V
  
+     .line 2665
      return-void
  .end method
  
***************
*** 134,151 ****
--- 149,170 ----
      .locals 1
  
      .prologue
+     .line 2613
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/view/KeyEvent$DispatcherState;->mDownKeyCode:I
  
+     .line 2614
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/view/KeyEvent$DispatcherState;->mDownTarget:Ljava/lang/Object;
  
+     .line 2615
      iget-object v0, p0, Landroid/view/KeyEvent$DispatcherState;->mActiveLongPresses:Landroid/util/SparseIntArray;
  
      invoke-virtual {v0}, Landroid/util/SparseIntArray;->clear()V
  
+     .line 2616
      return-void
  .end method
  
***************
*** 154,171 ****
--- 173,194 ----
      .parameter "target"
  
      .prologue
+     .line 2622
      iget-object v0, p0, Landroid/view/KeyEvent$DispatcherState;->mDownTarget:Ljava/lang/Object;
  
      if-ne v0, p1, :cond_0
  
+     .line 2624
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/view/KeyEvent$DispatcherState;->mDownKeyCode:I
  
+     .line 2625
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/view/KeyEvent$DispatcherState;->mDownTarget:Ljava/lang/Object;
  
+     .line 2627
      :cond_0
      return-void
  .end method
***************
*** 176,187 ****
--- 199,212 ----
      .parameter "target"
  
      .prologue
+     .line 2640
      invoke-virtual {p1}, Landroid/view/KeyEvent;->getAction()I
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 2641
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "Can only start tracking on a down event"
***************
*** 190,195 ****
--- 215,221 ----
  
      throw v0
  
+     .line 2645
      :cond_0
      invoke-virtual {p1}, Landroid/view/KeyEvent;->getKeyCode()I
  
***************
*** 197,203 ****
--- 223,231 ----
  
      iput v0, p0, Landroid/view/KeyEvent$DispatcherState;->mDownKeyCode:I
  
+     .line 2646
      iput-object p2, p0, Landroid/view/KeyEvent$DispatcherState;->mDownTarget:Ljava/lang/Object;
  
+     .line 2647
      return-void
  .end method
