*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ScaleGestureDetector$SimpleOnScaleGestureListener.smali	2012-03-23 19:37:20.431147643 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ScaleGestureDetector$SimpleOnScaleGestureListener.smali.original	2012-03-23 19:34:45.646325147 -0500
***************
*** 22,27 ****
--- 22,28 ----
      .locals 0
  
      .prologue
+     .line 109
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 34,39 ****
--- 35,41 ----
      .parameter "detector"
  
      .prologue
+     .line 112
      const/4 v0, 0x0
  
      return v0
***************
*** 44,49 ****
--- 46,52 ----
      .parameter "detector"
  
      .prologue
+     .line 116
      const/4 v0, 0x1
  
      return v0
***************
*** 54,58 ****
--- 57,62 ----
      .parameter "detector"
  
      .prologue
+     .line 121
      return-void
  .end method
