*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/Surface$CompatibleCanvas.smali	2012-03-23 19:37:26.130251566 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/Surface$CompatibleCanvas.smali.original	2012-03-23 19:34:45.956330799 -0500
***************
*** 26,35 ****
--- 26,37 ----
      .parameter
  
      .prologue
+     .line 323
      iput-object p1, p0, Landroid/view/Surface$CompatibleCanvas;->this$0:Landroid/view/Surface;
  
      invoke-direct {p0}, Landroid/graphics/Canvas;-><init>()V
  
+     .line 325
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/view/Surface$CompatibleCanvas;->mOrigMatrix:Landroid/graphics/Matrix;
***************
*** 43,48 ****
--- 45,51 ----
      .parameter "x1"
  
      .prologue
+     .line 323
      invoke-direct {p0, p1}, Landroid/view/Surface$CompatibleCanvas;-><init>(Landroid/view/Surface;)V
  
      return-void
***************
*** 54,63 ****
--- 57,68 ----
      .locals 3
  
      .prologue
+     .line 338
      invoke-super {p0}, Landroid/graphics/Canvas;->getHeight()I
  
      move-result v0
  
+     .line 339
      .local v0, h:I
      iget-object v1, p0, Landroid/view/Surface$CompatibleCanvas;->this$0:Landroid/view/Surface;
  
***************
*** 68,73 ****
--- 73,79 ----
  
      if-eqz v1, :cond_0
  
+     .line 340
      int-to-float v1, v0
  
      iget-object v2, p0, Landroid/view/Surface$CompatibleCanvas;->this$0:Landroid/view/Surface;
***************
*** 87,92 ****
--- 93,99 ----
  
      float-to-int v0, v1
  
+     .line 342
      :cond_0
      return v0
  .end method
***************
*** 96,118 ****
--- 103,130 ----
      .parameter "m"
  
      .prologue
+     .line 360
      invoke-super {p0, p1}, Landroid/graphics/Canvas;->getMatrix(Landroid/graphics/Matrix;)V
  
+     .line 361
      iget-object v0, p0, Landroid/view/Surface$CompatibleCanvas;->mOrigMatrix:Landroid/graphics/Matrix;
  
      if-nez v0, :cond_0
  
+     .line 362
      new-instance v0, Landroid/graphics/Matrix;
  
      invoke-direct {v0}, Landroid/graphics/Matrix;-><init>()V
  
      iput-object v0, p0, Landroid/view/Surface$CompatibleCanvas;->mOrigMatrix:Landroid/graphics/Matrix;
  
+     .line 364
      :cond_0
      iget-object v0, p0, Landroid/view/Surface$CompatibleCanvas;->mOrigMatrix:Landroid/graphics/Matrix;
  
      invoke-virtual {v0, p1}, Landroid/graphics/Matrix;->set(Landroid/graphics/Matrix;)V
  
+     .line 365
      return-void
  .end method
  
***************
*** 120,129 ****
--- 132,143 ----
      .locals 3
  
      .prologue
+     .line 329
      invoke-super {p0}, Landroid/graphics/Canvas;->getWidth()I
  
      move-result v0
  
+     .line 330
      .local v0, w:I
      iget-object v1, p0, Landroid/view/Surface$CompatibleCanvas;->this$0:Landroid/view/Surface;
  
***************
*** 134,139 ****
--- 148,154 ----
  
      if-eqz v1, :cond_0
  
+     .line 331
      int-to-float v1, v0
  
      iget-object v2, p0, Landroid/view/Surface$CompatibleCanvas;->this$0:Landroid/view/Surface;
***************
*** 153,158 ****
--- 168,174 ----
  
      float-to-int v0, v1
  
+     .line 333
      :cond_0
      return v0
  .end method
***************
*** 162,167 ****
--- 178,184 ----
      .parameter "matrix"
  
      .prologue
+     .line 347
      iget-object v1, p0, Landroid/view/Surface$CompatibleCanvas;->this$0:Landroid/view/Surface;
  
      #getter for: Landroid/view/Surface;->mCompatibleMatrix:Landroid/graphics/Matrix;
***************
*** 183,194 ****
--- 200,214 ----
  
      if-eqz v1, :cond_1
  
+     .line 350
      :cond_0
      invoke-super {p0, p1}, Landroid/graphics/Canvas;->setMatrix(Landroid/graphics/Matrix;)V
  
+     .line 356
      :goto_0
      return-void
  
+     .line 352
      :cond_1
      new-instance v0, Landroid/graphics/Matrix;
  
***************
*** 201,209 ****
--- 221,231 ----
  
      invoke-direct {v0, v1}, Landroid/graphics/Matrix;-><init>(Landroid/graphics/Matrix;)V
  
+     .line 353
      .local v0, m:Landroid/graphics/Matrix;
      invoke-virtual {v0, p1}, Landroid/graphics/Matrix;->preConcat(Landroid/graphics/Matrix;)Z
  
+     .line 354
      invoke-super {p0, v0}, Landroid/graphics/Canvas;->setMatrix(Landroid/graphics/Matrix;)V
  
      goto :goto_0
