*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/SurfaceSession.smali	2012-03-23 19:37:26.484258022 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/SurfaceSession.smali.original	2012-03-23 19:34:45.961330891 -0500
***************
*** 12,21 ****
--- 12,24 ----
      .locals 0
  
      .prologue
+     .line 28
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 29
      invoke-direct {p0}, Landroid/view/SurfaceSession;->init()V
  
+     .line 30
      return-void
  .end method
  
***************
*** 36,43 ****
--- 39,48 ----
      .end annotation
  
      .prologue
+     .line 41
      invoke-direct {p0}, Landroid/view/SurfaceSession;->destroy()V
  
+     .line 42
      return-void
  .end method
  
