*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/SurfaceView$3.smali	2012-03-23 19:37:22.855191846 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/SurfaceView$3.smali.original	2012-03-23 19:34:45.901329797 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 164
      iput-object p1, p0, Landroid/view/SurfaceView$3;->this$0:Landroid/view/SurfaceView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 44,49 ****
--- 45,51 ----
  
      const/4 v2, 0x0
  
+     .line 168
      iget-object v3, p0, Landroid/view/SurfaceView$3;->this$0:Landroid/view/SurfaceView;
  
      iget-object v0, p0, Landroid/view/SurfaceView$3;->this$0:Landroid/view/SurfaceView;
***************
*** 67,81 ****
--- 69,86 ----
      :goto_0
      iput-boolean v0, v3, Landroid/view/SurfaceView;->mHaveFrame:Z
  
+     .line 169
      iget-object v0, p0, Landroid/view/SurfaceView$3;->this$0:Landroid/view/SurfaceView;
  
      #calls: Landroid/view/SurfaceView;->updateWindow(ZZ)V
      invoke-static {v0, v2, v2}, Landroid/view/SurfaceView;->access$000(Landroid/view/SurfaceView;ZZ)V
  
+     .line 170
      return v1
  
      :cond_0
      move v0, v2
  
+     .line 168
      goto :goto_0
  .end method
