*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/TouchDelegate.smali	2012-03-23 19:37:20.540149631 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/TouchDelegate.smali.original	2012-03-23 19:34:45.647325165 -0500
***************
*** 32,41 ****
--- 32,44 ----
      .parameter "delegateView"
  
      .prologue
+     .line 90
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 91
      iput-object p1, p0, Landroid/view/TouchDelegate;->mBounds:Landroid/graphics/Rect;
  
+     .line 93
      invoke-virtual {p2}, Landroid/view/View;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 50,61 ****
--- 53,66 ----
  
      iput v0, p0, Landroid/view/TouchDelegate;->mSlop:I
  
+     .line 94
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0, p1}, Landroid/graphics/Rect;-><init>(Landroid/graphics/Rect;)V
  
      iput-object v0, p0, Landroid/view/TouchDelegate;->mSlopBounds:Landroid/graphics/Rect;
  
+     .line 95
      iget-object v0, p0, Landroid/view/TouchDelegate;->mSlopBounds:Landroid/graphics/Rect;
  
      iget v1, p0, Landroid/view/TouchDelegate;->mSlop:I
***************
*** 68,75 ****
--- 73,82 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/graphics/Rect;->inset(II)V
  
+     .line 96
      iput-object p2, p0, Landroid/view/TouchDelegate;->mDelegateView:Landroid/view/View;
  
+     .line 97
      return-void
  .end method
  
***************
*** 80,91 ****
--- 87,100 ----
      .parameter "event"
  
      .prologue
+     .line 107
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getX()F
  
      move-result v9
  
      float-to-int v7, v9
  
+     .line 108
      .local v7, x:I
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getY()F
  
***************
*** 93,107 ****
--- 102,120 ----
  
      float-to-int v8, v9
  
+     .line 109
      .local v8, y:I
      const/4 v4, 0x0
  
+     .line 110
      .local v4, sendToDelegate:Z
      const/4 v3, 0x1
  
+     .line 111
      .local v3, hit:Z
      const/4 v2, 0x0
  
+     .line 113
      .local v2, handled:Z
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getAction()I
  
***************
*** 109,123 ****
--- 122,140 ----
  
      packed-switch v9, :pswitch_data_0
  
+     .line 137
      :cond_0
      :goto_0
      if-eqz v4, :cond_1
  
+     .line 138
      iget-object v1, p0, Landroid/view/TouchDelegate;->mDelegateView:Landroid/view/View;
  
+     .line 140
      .local v1, delegateView:Landroid/view/View;
      if-eqz v3, :cond_2
  
+     .line 142
      invoke-virtual {v1}, Landroid/view/View;->getWidth()I
  
      move-result v9
***************
*** 136,153 ****
--- 153,174 ----
  
      invoke-virtual {p1, v9, v10}, Landroid/view/MotionEvent;->setLocation(FF)V
  
+     .line 149
      :goto_1
      invoke-virtual {v1, p1}, Landroid/view/View;->dispatchTouchEvent(Landroid/view/MotionEvent;)Z
  
      move-result v2
  
+     .line 151
      .end local v1           #delegateView:Landroid/view/View;
      :cond_1
      return v2
  
+     .line 115
      :pswitch_0
      iget-object v0, p0, Landroid/view/TouchDelegate;->mBounds:Landroid/graphics/Rect;
  
+     .line 117
      .local v0, bounds:Landroid/graphics/Rect;
      invoke-virtual {v0, v7, v8}, Landroid/graphics/Rect;->contains(II)Z
  
***************
*** 155,176 ****
--- 176,203 ----
  
      if-eqz v9, :cond_0
  
+     .line 118
      const/4 v9, 0x1
  
      iput-boolean v9, p0, Landroid/view/TouchDelegate;->mDelegateTargeted:Z
  
+     .line 119
      const/4 v4, 0x1
  
      goto :goto_0
  
+     .line 124
      .end local v0           #bounds:Landroid/graphics/Rect;
      :pswitch_1
      iget-boolean v4, p0, Landroid/view/TouchDelegate;->mDelegateTargeted:Z
  
+     .line 125
      if-eqz v4, :cond_0
  
+     .line 126
      iget-object v6, p0, Landroid/view/TouchDelegate;->mSlopBounds:Landroid/graphics/Rect;
  
+     .line 127
      .local v6, slopBounds:Landroid/graphics/Rect;
      invoke-virtual {v6, v7, v8}, Landroid/graphics/Rect;->contains(II)Z
  
***************
*** 178,201 ****
--- 205,233 ----
  
      if-nez v9, :cond_0
  
+     .line 128
      const/4 v3, 0x0
  
      goto :goto_0
  
+     .line 133
      .end local v6           #slopBounds:Landroid/graphics/Rect;
      :pswitch_2
      iget-boolean v4, p0, Landroid/view/TouchDelegate;->mDelegateTargeted:Z
  
+     .line 134
      const/4 v9, 0x0
  
      iput-boolean v9, p0, Landroid/view/TouchDelegate;->mDelegateTargeted:Z
  
      goto :goto_0
  
+     .line 146
      .restart local v1       #delegateView:Landroid/view/View;
      :cond_2
      iget v5, p0, Landroid/view/TouchDelegate;->mSlop:I
  
+     .line 147
      .local v5, slop:I
      mul-int/lit8 v9, v5, 0x2
  
***************
*** 213,218 ****
--- 245,251 ----
  
      goto :goto_1
  
+     .line 113
      nop
  
      :pswitch_data_0
