*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/View$1.smali	2012-03-23 19:37:25.251235538 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/View$1.smali.original	2012-03-23 19:34:45.943330563 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .parameter
  
      .prologue
+     .line 3018
      iput-object p1, p0, Landroid/view/View$1;->this$0:Landroid/view/View;
  
      iput-object p2, p0, Landroid/view/View$1;->val$handlerName:Ljava/lang/String;
***************
*** 48,57 ****
--- 49,60 ----
      .parameter "v"
  
      .prologue
+     .line 3022
      iget-object v3, p0, Landroid/view/View$1;->mHandler:Ljava/lang/reflect/Method;
  
      if-nez v3, :cond_0
  
+     .line 3024
      :try_start_0
      iget-object v3, p0, Landroid/view/View$1;->this$0:Landroid/view/View;
  
***************
*** 83,88 ****
--- 86,92 ----
      :try_end_0
      .catch Ljava/lang/NoSuchMethodException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 3039
      :cond_0
      :try_start_1
      iget-object v3, p0, Landroid/view/View$1;->mHandler:Ljava/lang/reflect/Method;
***************
*** 108,118 ****
--- 112,125 ----
      .catch Ljava/lang/IllegalAccessException; {:try_start_1 .. :try_end_1} :catch_1
      .catch Ljava/lang/reflect/InvocationTargetException; {:try_start_1 .. :try_end_1} :catch_2
  
+     .line 3047
      return-void
  
+     .line 3026
      :catch_0
      move-exception v0
  
+     .line 3027
      .local v0, e:Ljava/lang/NoSuchMethodException;
      iget-object v3, p0, Landroid/view/View$1;->this$0:Landroid/view/View;
  
***************
*** 120,125 ****
--- 127,133 ----
  
      move-result v1
  
+     .line 3028
      .local v1, id:I
      const/4 v3, -0x1
  
***************
*** 127,132 ****
--- 135,141 ----
  
      const-string v2, ""
  
+     .line 3031
      .local v2, idText:Ljava/lang/String;
      :goto_0
      new-instance v3, Ljava/lang/IllegalStateException;
***************
*** 201,206 ****
--- 210,216 ----
  
      throw v3
  
+     .line 3028
      .end local v2           #idText:Ljava/lang/String;
      :cond_1
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 243,253 ****
--- 253,265 ----
  
      goto :goto_0
  
+     .line 3040
      .end local v0           #e:Ljava/lang/NoSuchMethodException;
      .end local v1           #id:I
      :catch_1
      move-exception v0
  
+     .line 3041
      .local v0, e:Ljava/lang/IllegalAccessException;
      new-instance v3, Ljava/lang/IllegalStateException;
  
***************
*** 257,266 ****
--- 269,280 ----
  
      throw v3
  
+     .line 3043
      .end local v0           #e:Ljava/lang/IllegalAccessException;
      :catch_2
      move-exception v0
  
+     .line 3044
      .local v0, e:Ljava/lang/reflect/InvocationTargetException;
      new-instance v3, Ljava/lang/IllegalStateException;
  
