*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/View$MeasureSpec.smali	2012-03-23 19:37:20.484148611 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/View$MeasureSpec.smali.original	2012-03-23 19:34:45.646325147 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .locals 0
  
      .prologue
+     .line 13986
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "measureSpec"
  
      .prologue
+     .line 14036
      const/high16 v0, -0x4000
  
      and-int/2addr v0, p0
***************
*** 53,58 ****
--- 55,61 ----
      .parameter "measureSpec"
  
      .prologue
+     .line 14046
      const v0, 0x3fffffff
  
      and-int/2addr v0, p0
***************
*** 66,71 ****
--- 69,75 ----
      .parameter "mode"
  
      .prologue
+     .line 14024
      add-int v0, p0, p1
  
      return v0
***************
*** 76,90 ****
--- 80,97 ----
      .parameter "measureSpec"
  
      .prologue
+     .line 14057
      invoke-static {p0}, Landroid/view/View$MeasureSpec;->getMode(I)I
  
      move-result v0
  
+     .line 14058
      .local v0, mode:I
      invoke-static {p0}, Landroid/view/View$MeasureSpec;->getSize(I)I
  
      move-result v2
  
+     .line 14060
      .local v2, size:I
      new-instance v1, Ljava/lang/StringBuilder;
  
***************
*** 92,135 ****
--- 99,151 ----
  
      invoke-direct {v1, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
  
+     .line 14062
      .local v1, sb:Ljava/lang/StringBuilder;
      if-nez v0, :cond_0
  
+     .line 14063
      const-string v3, "UNSPECIFIED "
  
      invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 14071
      :goto_0
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 14072
      invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v3
  
      return-object v3
  
+     .line 14064
      :cond_0
      const/high16 v3, 0x4000
  
      if-ne v0, v3, :cond_1
  
+     .line 14065
      const-string v3, "EXACTLY "
  
      invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      goto :goto_0
  
+     .line 14066
      :cond_1
      const/high16 v3, -0x8000
  
      if-ne v0, v3, :cond_2
  
+     .line 14067
      const-string v3, "AT_MOST "
  
      invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      goto :goto_0
  
+     .line 14069
      :cond_2
      invoke-virtual {v1, v0}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
