*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewRootImpl$TrackballAxis.smali	2012-03-23 19:37:24.712225709 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewRootImpl$TrackballAxis.smali.original	2012-03-23 19:34:45.934330399 -0500
***************
*** 43,54 ****
--- 43,57 ----
      .locals 2
  
      .prologue
+     .line 4238
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 4261
      const/high16 v0, 0x3f80
  
      iput v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->acceleration:F
  
+     .line 4262
      const-wide/16 v0, 0x0
  
      iput-wide v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->lastMoveTime:J
***************
*** 65,108 ****
--- 68,120 ----
      .parameter "axis"
  
      .prologue
+     .line 4287
      const/4 v6, 0x0
  
      cmpl-float v6, p1, v6
  
      if-lez v6, :cond_3
  
+     .line 4288
      const/high16 v6, 0x4316
  
      mul-float/2addr v6, p1
  
      float-to-long v3, v6
  
+     .line 4289
      .local v3, normTime:J
      iget v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->dir:I
  
      if-gez v6, :cond_0
  
+     .line 4291
      const/4 v6, 0x0
  
      iput v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
+     .line 4292
      const/4 v6, 0x0
  
      iput v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->step:I
  
+     .line 4293
      const/high16 v6, 0x3f80
  
      iput v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->acceleration:F
  
+     .line 4294
      const-wide/16 v6, 0x0
  
      iput-wide v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->lastMoveTime:J
  
+     .line 4296
      :cond_0
      const/4 v6, 0x1
  
      iput v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->dir:I
  
+     .line 4314
      :goto_0
      const-wide/16 v6, 0x0
  
***************
*** 110,129 ****
--- 122,146 ----
  
      if-lez v6, :cond_2
  
+     .line 4315
      iget-wide v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->lastMoveTime:J
  
      sub-long v1, p2, v6
  
+     .line 4316
      .local v1, delta:J
      iput-wide p2, p0, Landroid/view/ViewRootImpl$TrackballAxis;->lastMoveTime:J
  
+     .line 4317
      iget v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->acceleration:F
  
+     .line 4318
      .local v0, acc:F
      cmp-long v6, v1, v3
  
      if-gez v6, :cond_7
  
+     .line 4320
      sub-long v6, v3, v1
  
      long-to-float v6, v6
***************
*** 132,137 ****
--- 149,155 ----
  
      mul-float v5, v6, v7
  
+     .line 4321
      .local v5, scale:F
      const/high16 v6, 0x3f80
  
***************
*** 141,146 ****
--- 159,165 ----
  
      mul-float/2addr v0, v5
  
+     .line 4325
      :cond_1
      const/high16 v6, 0x41a0
  
***************
*** 152,157 ****
--- 171,177 ----
      :goto_1
      iput v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->acceleration:F
  
+     .line 4336
      .end local v1           #delta:J
      .end local v5           #scale:F
      :cond_2
***************
*** 162,167 ****
--- 182,188 ----
  
      iput v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
+     .line 4337
      iget v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
      invoke-static {v6}, Ljava/lang/Math;->abs(F)F
***************
*** 172,177 ****
--- 193,199 ----
  
      return v6
  
+     .line 4297
      .end local v3           #normTime:J
      :cond_3
      const/4 v6, 0x0
***************
*** 180,185 ****
--- 202,208 ----
  
      if-gez v6, :cond_5
  
+     .line 4298
      neg-float v6, p1
  
      const/high16 v7, 0x4316
***************
*** 188,214 ****
--- 211,243 ----
  
      float-to-long v3, v6
  
+     .line 4299
      .restart local v3       #normTime:J
      iget v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->dir:I
  
      if-lez v6, :cond_4
  
+     .line 4301
      const/4 v6, 0x0
  
      iput v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
+     .line 4302
      const/4 v6, 0x0
  
      iput v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->step:I
  
+     .line 4303
      const/high16 v6, 0x3f80
  
      iput v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->acceleration:F
  
+     .line 4304
      const-wide/16 v6, 0x0
  
      iput-wide v6, p0, Landroid/view/ViewRootImpl$TrackballAxis;->lastMoveTime:J
  
+     .line 4306
      :cond_4
      const/4 v6, -0x1
  
***************
*** 216,221 ****
--- 245,251 ----
  
      goto :goto_0
  
+     .line 4308
      .end local v3           #normTime:J
      :cond_5
      const-wide/16 v3, 0x0
***************
*** 223,228 ****
--- 253,259 ----
      .restart local v3       #normTime:J
      goto :goto_0
  
+     .line 4325
      .restart local v0       #acc:F
      .restart local v1       #delta:J
      .restart local v5       #scale:F
***************
*** 231,236 ****
--- 262,268 ----
  
      goto :goto_1
  
+     .line 4328
      .end local v5           #scale:F
      :cond_7
      sub-long v6, v1, v3
***************
*** 241,246 ****
--- 273,279 ----
  
      mul-float v5, v6, v7
  
+     .line 4329
      .restart local v5       #scale:F
      const/high16 v6, 0x3f80
  
***************
*** 250,255 ****
--- 283,289 ----
  
      div-float/2addr v0, v5
  
+     .line 4333
      :cond_8
      const/high16 v6, 0x3f80
  
***************
*** 281,293 ****
--- 315,330 ----
  
      const/high16 v6, 0x3f80
  
+     .line 4352
      const/4 v2, 0x0
  
+     .line 4353
      .local v2, movement:I
      const/4 v4, 0x0
  
      iput v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->nonAccelMovement:I
  
+     .line 4355
      :goto_0
      iget v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
***************
*** 299,325 ****
--- 336,367 ----
  
      move v1, v3
  
+     .line 4356
      .local v1, dir:I
      :goto_1
      iget v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->step:I
  
      packed-switch v4, :pswitch_data_0
  
+     .line 4389
      iget v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->absPosition:F
  
      cmpg-float v4, v4, v6
  
      if-gez v4, :cond_3
  
+     .line 4390
      :cond_0
      return v2
  
+     .line 4355
      .end local v1           #dir:I
      :cond_1
      const/4 v1, -0x1
  
      goto :goto_1
  
+     .line 4361
      .restart local v1       #dir:I
      :pswitch_0
      iget v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->absPosition:F
***************
*** 328,345 ****
--- 370,391 ----
  
      if-ltz v4, :cond_0
  
+     .line 4364
      add-int/2addr v2, v1
  
+     .line 4365
      iget v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->nonAccelMovement:I
  
      add-int/2addr v4, v1
  
      iput v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->nonAccelMovement:I
  
+     .line 4366
      iput v3, p0, Landroid/view/ViewRootImpl$TrackballAxis;->step:I
  
      goto :goto_0
  
+     .line 4372
      :pswitch_1
      iget v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->absPosition:F
  
***************
*** 347,360 ****
--- 393,409 ----
  
      if-ltz v4, :cond_0
  
+     .line 4375
      add-int/2addr v2, v1
  
+     .line 4376
      iget v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->nonAccelMovement:I
  
      add-int/2addr v4, v1
  
      iput v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->nonAccelMovement:I
  
+     .line 4377
      iget v7, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
      if-lez v1, :cond_2
***************
*** 366,371 ****
--- 415,421 ----
  
      iput v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
+     .line 4378
      iget v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
      invoke-static {v4}, Ljava/lang/Math;->abs(F)F
***************
*** 374,379 ****
--- 424,430 ----
  
      iput v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->absPosition:F
  
+     .line 4379
      const/4 v4, 0x2
  
      iput v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->step:I
***************
*** 383,393 ****
--- 434,447 ----
      :cond_2
      move v4, v5
  
+     .line 4377
      goto :goto_2
  
+     .line 4392
      :cond_3
      add-int/2addr v2, v1
  
+     .line 4393
      iget v7, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
      if-ltz v1, :cond_4
***************
*** 399,404 ****
--- 453,459 ----
  
      iput v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
+     .line 4394
      iget v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
      invoke-static {v4}, Ljava/lang/Math;->abs(F)F
***************
*** 407,419 ****
--- 462,477 ----
  
      iput v4, p0, Landroid/view/ViewRootImpl$TrackballAxis;->absPosition:F
  
+     .line 4395
      iget v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->acceleration:F
  
+     .line 4396
      .local v0, acc:F
      const v4, 0x3f8ccccd
  
      mul-float/2addr v0, v4
  
+     .line 4397
      const/high16 v4, 0x41a0
  
      cmpg-float v4, v0, v4
***************
*** 429,442 ****
--- 487,503 ----
      :cond_4
      move v4, v6
  
+     .line 4393
      goto :goto_3
  
+     .line 4397
      .restart local v0       #acc:F
      :cond_5
      iget v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->acceleration:F
  
      goto :goto_4
  
+     .line 4356
      nop
  
      :pswitch_data_0
***************
*** 451,473 ****
--- 512,540 ----
      .parameter "_step"
  
      .prologue
+     .line 4268
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->position:F
  
+     .line 4269
      const/high16 v0, 0x3f80
  
      iput v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->acceleration:F
  
+     .line 4270
      const-wide/16 v0, 0x0
  
      iput-wide v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->lastMoveTime:J
  
+     .line 4271
      iput p1, p0, Landroid/view/ViewRootImpl$TrackballAxis;->step:I
  
+     .line 4272
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/view/ViewRootImpl$TrackballAxis;->dir:I
  
+     .line 4273
      return-void
  .end method
