*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewGroup$1.smali	2012-03-23 19:37:26.219253190 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewGroup$1.smali.original	2012-03-23 19:34:45.957330817 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 2529
      iput-object p1, p0, Landroid/view/ViewGroup$1;->this$0:Landroid/view/ViewGroup;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,49 ****
--- 41,52 ----
      .locals 1
  
      .prologue
+     .line 2531
      iget-object v0, p0, Landroid/view/ViewGroup$1;->this$0:Landroid/view/ViewGroup;
  
      #calls: Landroid/view/ViewGroup;->notifyAnimationListener()V
      invoke-static {v0}, Landroid/view/ViewGroup;->access$000(Landroid/view/ViewGroup;)V
  
+     .line 2532
      return-void
  .end method
