*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewTreeObserver$InternalInsetsInfo.smali	2012-03-23 19:37:23.458202842 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewTreeObserver$InternalInsetsInfo.smali.original	2012-03-23 19:34:45.913330015 -0500
***************
*** 39,58 ****
--- 39,62 ----
      .locals 1
  
      .prologue
+     .line 122
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 127
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->contentInsets:Landroid/graphics/Rect;
  
+     .line 133
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->visibleInsets:Landroid/graphics/Rect;
  
+     .line 140
      new-instance v0, Landroid/graphics/Region;
  
      invoke-direct {v0}, Landroid/graphics/Region;-><init>()V
***************
*** 71,82 ****
--- 75,89 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 190
      if-nez p1, :cond_1
  
+     .line 205
      :cond_0
      :goto_0
      return v3
  
+     .line 193
      :cond_1
      :try_start_0
      move-object v0, p1
***************
*** 85,90 ****
--- 92,98 ----
  
      move-object v2, v0
  
+     .line 194
      .local v2, other:Landroid/view/ViewTreeObserver$InternalInsetsInfo;
      iget v4, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->mTouchableInsets:I
  
***************
*** 92,97 ****
--- 100,106 ----
  
      if-ne v4, v5, :cond_0
  
+     .line 197
      iget-object v4, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->contentInsets:Landroid/graphics/Rect;
  
      iget-object v5, v2, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->contentInsets:Landroid/graphics/Rect;
***************
*** 102,107 ****
--- 111,117 ----
  
      if-eqz v4, :cond_0
  
+     .line 200
      iget-object v4, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->visibleInsets:Landroid/graphics/Rect;
  
      iget-object v5, v2, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->visibleInsets:Landroid/graphics/Rect;
***************
*** 112,117 ****
--- 122,128 ----
  
      if-eqz v4, :cond_0
  
+     .line 203
      iget-object v4, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->touchableRegion:Landroid/graphics/Region;
  
      iget-object v5, v2, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->touchableRegion:Landroid/graphics/Region;
***************
*** 124,133 ****
--- 135,146 ----
  
      goto :goto_0
  
+     .line 204
      .end local v2           #other:Landroid/view/ViewTreeObserver$InternalInsetsInfo;
      :catch_0
      move-exception v1
  
+     .line 205
      .local v1, e:Ljava/lang/ClassCastException;
      goto :goto_0
  .end method
***************
*** 136,141 ****
--- 149,155 ----
      .locals 1
  
      .prologue
+     .line 176
      iget v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->mTouchableInsets:I
  
      return v0
***************
*** 145,166 ****
--- 159,185 ----
      .locals 1
  
      .prologue
+     .line 182
      iget-object v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->contentInsets:Landroid/graphics/Rect;
  
      invoke-virtual {v0}, Landroid/graphics/Rect;->setEmpty()V
  
+     .line 183
      iget-object v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->visibleInsets:Landroid/graphics/Rect;
  
      invoke-virtual {v0}, Landroid/graphics/Rect;->setEmpty()V
  
+     .line 184
      iget-object v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->touchableRegion:Landroid/graphics/Region;
  
      invoke-virtual {v0}, Landroid/graphics/Region;->setEmpty()V
  
+     .line 185
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->mTouchableInsets:I
  
+     .line 186
      return-void
  .end method
  
***************
*** 169,196 ****
--- 188,220 ----
      .parameter "other"
  
      .prologue
+     .line 210
      iget-object v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->contentInsets:Landroid/graphics/Rect;
  
      iget-object v1, p1, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->contentInsets:Landroid/graphics/Rect;
  
      invoke-virtual {v0, v1}, Landroid/graphics/Rect;->set(Landroid/graphics/Rect;)V
  
+     .line 211
      iget-object v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->visibleInsets:Landroid/graphics/Rect;
  
      iget-object v1, p1, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->visibleInsets:Landroid/graphics/Rect;
  
      invoke-virtual {v0, v1}, Landroid/graphics/Rect;->set(Landroid/graphics/Rect;)V
  
+     .line 212
      iget-object v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->touchableRegion:Landroid/graphics/Region;
  
      iget-object v1, p1, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->touchableRegion:Landroid/graphics/Region;
  
      invoke-virtual {v0, v1}, Landroid/graphics/Region;->set(Landroid/graphics/Region;)Z
  
+     .line 213
      iget v0, p1, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->mTouchableInsets:I
  
      iput v0, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->mTouchableInsets:I
  
+     .line 214
      return-void
  .end method
  
***************
*** 199,205 ****
--- 223,231 ----
      .parameter "val"
  
      .prologue
+     .line 172
      iput p1, p0, Landroid/view/ViewTreeObserver$InternalInsetsInfo;->mTouchableInsets:I
  
+     .line 173
      return-void
  .end method
