*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewDebug$LooperProfiler$1.smali	2012-03-23 19:37:26.000249196 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewDebug$LooperProfiler$1.smali.original	2012-03-23 19:34:45.954330763 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 521
      iput-object p1, p0, Landroid/view/ViewDebug$LooperProfiler$1;->this$0:Landroid/view/ViewDebug$LooperProfiler;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,49 ****
--- 41,52 ----
      .locals 1
  
      .prologue
+     .line 524
      iget-object v0, p0, Landroid/view/ViewDebug$LooperProfiler$1;->this$0:Landroid/view/ViewDebug$LooperProfiler;
  
      #calls: Landroid/view/ViewDebug$LooperProfiler;->saveTraces()V
      invoke-static {v0}, Landroid/view/ViewDebug$LooperProfiler;->access$000(Landroid/view/ViewDebug$LooperProfiler;)V
  
+     .line 525
      return-void
  .end method
