*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewRootImpl$3.smali	2012-03-23 19:37:23.978212324 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewRootImpl$3.smali.original	2012-03-23 19:34:45.919330125 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 4410
      iput-object p1, p0, Landroid/view/ViewRootImpl$3;->this$0:Landroid/view/ViewRootImpl;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "callback"
  
      .prologue
+     .line 4424
      return-void
  .end method
  
***************
*** 48,53 ****
--- 50,56 ----
      .locals 1
  
      .prologue
+     .line 4416
      iget-object v0, p0, Landroid/view/ViewRootImpl$3;->this$0:Landroid/view/ViewRootImpl;
  
      #getter for: Landroid/view/ViewRootImpl;->mSurface:Landroid/view/Surface;
***************
*** 62,67 ****
--- 65,71 ----
      .locals 1
  
      .prologue
+     .line 4455
      const/4 v0, 0x0
  
      return-object v0
***************
*** 71,76 ****
--- 75,81 ----
      .locals 1
  
      .prologue
+     .line 4420
      const/4 v0, 0x0
  
      return v0
***************
*** 80,85 ****
--- 85,91 ----
      .locals 1
  
      .prologue
+     .line 4445
      const/4 v0, 0x0
  
      return-object v0
***************
*** 90,95 ****
--- 96,102 ----
      .parameter "dirty"
  
      .prologue
+     .line 4449
      const/4 v0, 0x0
  
      return-object v0
***************
*** 100,105 ****
--- 107,113 ----
      .parameter "callback"
  
      .prologue
+     .line 4427
      return-void
  .end method
  
***************
*** 109,114 ****
--- 117,123 ----
      .parameter "height"
  
      .prologue
+     .line 4430
      return-void
  .end method
  
***************
*** 117,122 ****
--- 126,132 ----
      .parameter "format"
  
      .prologue
+     .line 4436
      return-void
  .end method
  
***************
*** 125,130 ****
--- 135,141 ----
      .parameter "screenOn"
  
      .prologue
+     .line 4442
      return-void
  .end method
  
***************
*** 132,137 ****
--- 143,149 ----
      .locals 0
  
      .prologue
+     .line 4433
      return-void
  .end method
  
***************
*** 140,145 ****
--- 152,158 ----
      .parameter "type"
  
      .prologue
+     .line 4439
      return-void
  .end method
  
***************
*** 148,152 ****
--- 161,166 ----
      .parameter "canvas"
  
      .prologue
+     .line 4453
      return-void
  .end method
