*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/View$3.smali	2012-03-23 19:37:26.618260466 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/View$3.smali.original	2012-03-23 19:34:45.962330909 -0500
***************
*** 40,45 ****
--- 40,46 ----
      .parameter
  
      .prologue
+     .line 5162
      iput-object p1, p0, Landroid/view/View$3;->this$0:Landroid/view/View;
  
      iput p2, p0, Landroid/view/View$3;->val$childViewId:I
***************
*** 56,61 ****
--- 57,63 ----
      .parameter "t"
  
      .prologue
+     .line 5165
      iget v0, p1, Landroid/view/View;->mID:I
  
      iget v1, p0, Landroid/view/View$3;->val$childViewId:I
***************
*** 78,83 ****
--- 80,86 ----
      .parameter "x0"
  
      .prologue
+     .line 5162
      check-cast p1, Landroid/view/View;
  
      .end local p1
