*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/View$11.smali	2012-03-23 19:37:23.508203754 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/View$11.smali.original	2012-03-23 19:34:45.914330033 -0500
***************
*** 29,34 ****
--- 29,35 ----
      .parameter "x0"
  
      .prologue
+     .line 13915
      invoke-direct {p0, p1}, Landroid/util/FloatProperty;-><init>(Ljava/lang/String;)V
  
      return-void
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "object"
  
      .prologue
+     .line 13923
      invoke-virtual {p1}, Landroid/view/View;->getRotationY()F
  
      move-result v0
***************
*** 57,62 ****
--- 59,65 ----
      .parameter "x0"
  
      .prologue
+     .line 13915
      check-cast p1, Landroid/view/View;
  
      .end local p1
***************
*** 73,80 ****
--- 76,85 ----
      .parameter "value"
  
      .prologue
+     .line 13918
      invoke-virtual {p1, p2}, Landroid/view/View;->setRotationY(F)V
  
+     .line 13919
      return-void
  .end method
  
***************
*** 84,89 ****
--- 89,95 ----
      .parameter "x1"
  
      .prologue
+     .line 13915
      check-cast p1, Landroid/view/View;
  
      .end local p1
