*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewRootImpl$ResizedInfo.smali	2012-03-23 19:37:27.078268854 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewRootImpl$ResizedInfo.smali.original	2012-03-23 19:34:45.972331091 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 248
      iput-object p1, p0, Landroid/view/ViewRootImpl$ResizedInfo;->this$0:Landroid/view/ViewRootImpl;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
