*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewDebug$5.smali	2012-03-23 19:37:27.271272374 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/ViewDebug$5.smali.original	2012-03-23 19:34:45.974331127 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .parameter
  
      .prologue
+     .line 1130
      iput-object p1, p0, Landroid/view/ViewDebug$5;->val$operation:Landroid/view/ViewDebug$ViewOperation;
  
      iput-object p2, p0, Landroid/view/ViewDebug$5;->val$duration:[J
***************
*** 50,55 ****
--- 51,57 ----
      .locals 7
  
      .prologue
+     .line 1133
      :try_start_0
      iget-object v3, p0, Landroid/view/ViewDebug$5;->val$operation:Landroid/view/ViewDebug$ViewOperation;
  
***************
*** 57,72 ****
--- 59,77 ----
  
      move-result-object v0
  
+     .line 1134
      .local v0, data:[Ljava/lang/Object;,"[TT;"
      invoke-static {}, Landroid/os/Debug;->threadCpuTimeNanos()J
  
      move-result-wide v1
  
+     .line 1136
      .local v1, start:J
      iget-object v3, p0, Landroid/view/ViewDebug$5;->val$operation:Landroid/view/ViewDebug$ViewOperation;
  
      invoke-interface {v3, v0}, Landroid/view/ViewDebug$ViewOperation;->run([Ljava/lang/Object;)V
  
+     .line 1137
      iget-object v3, p0, Landroid/view/ViewDebug$5;->val$duration:[J
  
      const/4 v4, 0x0
***************
*** 79,96 ****
--- 84,105 ----
  
      aput-wide v5, v3, v4
  
+     .line 1139
      iget-object v3, p0, Landroid/view/ViewDebug$5;->val$operation:Landroid/view/ViewDebug$ViewOperation;
  
      invoke-interface {v3, v0}, Landroid/view/ViewDebug$ViewOperation;->post([Ljava/lang/Object;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1141
      iget-object v3, p0, Landroid/view/ViewDebug$5;->val$latch:Ljava/util/concurrent/CountDownLatch;
  
      invoke-virtual {v3}, Ljava/util/concurrent/CountDownLatch;->countDown()V
  
+     .line 1143
      return-void
  
+     .line 1141
      .end local v0           #data:[Ljava/lang/Object;,"[TT;"
      .end local v1           #start:J
      :catchall_0
