*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/VolumePanel$4.smali	2012-03-23 19:37:23.316200252 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/VolumePanel$4.smali.original	2012-03-23 19:34:45.906329887 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 263
      iput-object p1, p0, Landroid/view/VolumePanel$4;->this$0:Landroid/view/VolumePanel;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 41,50 ****
--- 42,53 ----
      .prologue
      const/4 v3, 0x6
  
+     .line 266
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 268
      .local v0, action:Ljava/lang/String;
      const-string v1, "android.media.RINGER_MODE_CHANGED"
  
***************
*** 54,63 ****
--- 57,68 ----
  
      if-eqz v1, :cond_0
  
+     .line 269
      iget-object v1, p0, Landroid/view/VolumePanel$4;->this$0:Landroid/view/VolumePanel;
  
      invoke-virtual {v1, v3}, Landroid/view/VolumePanel;->removeMessages(I)V
  
+     .line 270
      iget-object v1, p0, Landroid/view/VolumePanel$4;->this$0:Landroid/view/VolumePanel;
  
      iget-object v2, p0, Landroid/view/VolumePanel$4;->this$0:Landroid/view/VolumePanel;
***************
*** 68,73 ****
--- 73,79 ----
  
      invoke-virtual {v1, v2}, Landroid/view/VolumePanel;->sendMessage(Landroid/os/Message;)Z
  
+     .line 272
      :cond_0
      return-void
  .end method
