*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/VolumePanel$3.smali	2012-03-23 19:37:19.011121749 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/VolumePanel$3.smali.original	2012-03-23 19:34:45.634324927 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 223
      iput-object p1, p0, Landroid/view/VolumePanel$3;->this$0:Landroid/view/VolumePanel;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "dialog"
  
      .prologue
+     .line 225
      iget-object v0, p0, Landroid/view/VolumePanel$3;->this$0:Landroid/view/VolumePanel;
  
      const/4 v1, -0x1
***************
*** 48,53 ****
--- 50,56 ----
      #setter for: Landroid/view/VolumePanel;->mActiveStreamType:I
      invoke-static {v0, v1}, Landroid/view/VolumePanel;->access$202(Landroid/view/VolumePanel;I)I
  
+     .line 226
      iget-object v0, p0, Landroid/view/VolumePanel$3;->this$0:Landroid/view/VolumePanel;
  
      #getter for: Landroid/view/VolumePanel;->mAudioManager:Landroid/media/AudioManager;
***************
*** 64,68 ****
--- 67,72 ----
  
      invoke-virtual {v0, v1}, Landroid/media/AudioManager;->forceVolumeControlStream(I)V
  
+     .line 227
      return-void
  .end method
