*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/VolumePanel.smali	2012-03-23 19:37:19.993139657 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/VolumePanel.smali.original	2012-03-23 19:34:45.642325073 -0500
***************
*** 103,110 ****
--- 103,112 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 63
      sput-boolean v2, Landroid/view/VolumePanel;->LOGD:Z
  
+     .line 170
      const/4 v0, 0x6
  
      new-array v0, v0, [Landroid/view/VolumePanel$StreamResources;
***************
*** 160,173 ****
--- 162,179 ----
  
      const/4 v6, -0x2
  
+     .line 193
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
  
+     .line 117
      const/4 v4, -0x1
  
      iput v4, p0, Landroid/view/VolumePanel;->mActiveStreamType:I
  
+     .line 194
      iput-object p1, p0, Landroid/view/VolumePanel;->mContext:Landroid/content/Context;
  
+     .line 195
      const-string v4, "audio"
  
      invoke-virtual {p1, v4}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 178,185 ****
--- 184,193 ----
  
      iput-object v4, p0, Landroid/view/VolumePanel;->mAudioManager:Landroid/media/AudioManager;
  
+     .line 196
      iput-object p2, p0, Landroid/view/VolumePanel;->mAudioService:Landroid/media/AudioService;
  
+     .line 198
      const-string v4, "layout_inflater"
  
      invoke-virtual {p1, v4}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 188,193 ****
--- 196,202 ----
  
      check-cast v0, Landroid/view/LayoutInflater;
  
+     .line 200
      .local v0, inflater:Landroid/view/LayoutInflater;
      const v4, 0x10900af
  
***************
*** 197,202 ****
--- 206,212 ----
  
      iput-object v2, p0, Landroid/view/VolumePanel;->mView:Landroid/view/View;
  
+     .line 201
      .local v2, view:Landroid/view/View;
      iget-object v4, p0, Landroid/view/VolumePanel;->mView:Landroid/view/View;
  
***************
*** 206,211 ****
--- 216,222 ----
  
      invoke-virtual {v4, v5}, Landroid/view/View;->setOnTouchListener(Landroid/view/View$OnTouchListener;)V
  
+     .line 207
      iget-object v4, p0, Landroid/view/VolumePanel;->mView:Landroid/view/View;
  
      const v5, 0x102032e
***************
*** 218,223 ****
--- 229,235 ----
  
      iput-object v4, p0, Landroid/view/VolumePanel;->mPanel:Landroid/view/ViewGroup;
  
+     .line 208
      iget-object v4, p0, Landroid/view/VolumePanel;->mView:Landroid/view/View;
  
      const v5, 0x102032f
***************
*** 230,235 ****
--- 242,248 ----
  
      iput-object v4, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
+     .line 209
      iget-object v4, p0, Landroid/view/VolumePanel;->mView:Landroid/view/View;
  
      const v5, 0x1020331
***************
*** 242,247 ****
--- 255,261 ----
  
      iput-object v4, p0, Landroid/view/VolumePanel;->mMoreButton:Landroid/view/View;
  
+     .line 210
      iget-object v4, p0, Landroid/view/VolumePanel;->mView:Landroid/view/View;
  
      const v5, 0x1020330
***************
*** 254,259 ****
--- 268,274 ----
  
      iput-object v4, p0, Landroid/view/VolumePanel;->mDivider:Landroid/view/View;
  
+     .line 212
      new-instance v4, Landroid/view/VolumePanel$2;
  
      const v5, 0x10302fa
***************
*** 262,279 ****
--- 277,297 ----
  
      iput-object v4, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
+     .line 221
      iget-object v4, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
      const-string v5, "Volume control"
  
      invoke-virtual {v4, v5}, Landroid/app/Dialog;->setTitle(Ljava/lang/CharSequence;)V
  
+     .line 222
      iget-object v4, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
      iget-object v5, p0, Landroid/view/VolumePanel;->mView:Landroid/view/View;
  
      invoke-virtual {v4, v5}, Landroid/app/Dialog;->setContentView(Landroid/view/View;)V
  
+     .line 223
      iget-object v4, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
      new-instance v5, Landroid/view/VolumePanel$3;
***************
*** 282,305 ****
--- 300,328 ----
  
      invoke-virtual {v4, v5}, Landroid/app/Dialog;->setOnDismissListener(Landroid/content/DialogInterface$OnDismissListener;)V
  
+     .line 230
      iget-object v4, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
      invoke-virtual {v4}, Landroid/app/Dialog;->getWindow()Landroid/view/Window;
  
      move-result-object v3
  
+     .line 231
      .local v3, window:Landroid/view/Window;
      const/16 v4, 0x30
  
      invoke-virtual {v3, v4}, Landroid/view/Window;->setGravity(I)V
  
+     .line 232
      invoke-virtual {v3}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v1
  
+     .line 233
      .local v1, lp:Landroid/view/WindowManager$LayoutParams;
      iput-object v8, v1, Landroid/view/WindowManager$LayoutParams;->token:Landroid/os/IBinder;
  
+     .line 235
      iget-object v4, p0, Landroid/view/VolumePanel;->mContext:Landroid/content/Context;
  
      invoke-virtual {v4}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 314,333 ****
--- 337,362 ----
  
      iput v4, v1, Landroid/view/WindowManager$LayoutParams;->y:I
  
+     .line 237
      const/16 v4, 0x7e4
  
      iput v4, v1, Landroid/view/WindowManager$LayoutParams;->type:I
  
+     .line 238
      iput v6, v1, Landroid/view/WindowManager$LayoutParams;->width:I
  
+     .line 239
      iput v6, v1, Landroid/view/WindowManager$LayoutParams;->height:I
  
+     .line 240
      invoke-virtual {v3, v1}, Landroid/view/Window;->setAttributes(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 241
      const v4, 0x40028
  
      invoke-virtual {v3, v4}, Landroid/view/Window;->addFlags(I)V
  
+     .line 244
      invoke-static {}, Landroid/media/AudioSystem;->getNumStreamTypes()I
  
      move-result v4
***************
*** 336,347 ****
--- 365,378 ----
  
      iput-object v4, p0, Landroid/view/VolumePanel;->mToneGenerators:[Landroid/media/ToneGenerator;
  
+     .line 245
      new-instance v4, Landroid/os/Vibrator;
  
      invoke-direct {v4}, Landroid/os/Vibrator;-><init>()V
  
      iput-object v4, p0, Landroid/view/VolumePanel;->mVibrator:Landroid/os/Vibrator;
  
+     .line 247
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v4
***************
*** 354,359 ****
--- 385,391 ----
  
      iput-boolean v4, p0, Landroid/view/VolumePanel;->mVoiceCapable:Z
  
+     .line 248
      iget-boolean v4, p0, Landroid/view/VolumePanel;->mVoiceCapable:Z
  
      if-nez v4, :cond_0
***************
*** 363,390 ****
--- 395,429 ----
      :goto_0
      iput-boolean v4, p0, Landroid/view/VolumePanel;->mShowCombinedVolumes:Z
  
+     .line 250
      iget-boolean v4, p0, Landroid/view/VolumePanel;->mShowCombinedVolumes:Z
  
      if-nez v4, :cond_1
  
+     .line 251
      iget-object v4, p0, Landroid/view/VolumePanel;->mMoreButton:Landroid/view/View;
  
      invoke-virtual {v4, v7}, Landroid/view/View;->setVisibility(I)V
  
+     .line 252
      iget-object v4, p0, Landroid/view/VolumePanel;->mDivider:Landroid/view/View;
  
      invoke-virtual {v4, v7}, Landroid/view/View;->setVisibility(I)V
  
+     .line 257
      :goto_1
      invoke-direct {p0}, Landroid/view/VolumePanel;->listenToRingerMode()V
  
+     .line 258
      return-void
  
+     .line 248
      :cond_0
      const/4 v4, 0x0
  
      goto :goto_0
  
+     .line 254
      :cond_1
      iget-object v4, p0, Landroid/view/VolumePanel;->mMoreButton:Landroid/view/View;
  
***************
*** 398,403 ****
--- 437,443 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      invoke-direct {p0}, Landroid/view/VolumePanel;->resetTimeout()V
  
      return-void
***************
*** 408,413 ****
--- 448,454 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      invoke-direct {p0}, Landroid/view/VolumePanel;->forceTimeout()V
  
      return-void
***************
*** 418,423 ****
--- 459,465 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget v0, p0, Landroid/view/VolumePanel;->mActiveStreamType:I
  
      return v0
***************
*** 429,434 ****
--- 471,477 ----
      .parameter "x1"
  
      .prologue
+     .line 60
      iput p1, p0, Landroid/view/VolumePanel;->mActiveStreamType:I
  
      return p1
***************
*** 439,444 ****
--- 482,488 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Landroid/view/VolumePanel;->mAudioManager:Landroid/media/AudioManager;
  
      return-object v0
***************
*** 448,460 ****
--- 492,507 ----
      .locals 5
  
      .prologue
+     .line 329
      iget-boolean v3, p0, Landroid/view/VolumePanel;->mShowCombinedVolumes:Z
  
      if-nez v3, :cond_1
  
+     .line 341
      :cond_0
      return-void
  
+     .line 331
      :cond_1
      const/4 v0, 0x0
  
***************
*** 466,477 ****
--- 513,526 ----
  
      if-ge v0, v3, :cond_0
  
+     .line 333
      sget-object v3, Landroid/view/VolumePanel;->STREAMS:[Landroid/view/VolumePanel$StreamResources;
  
      aget-object v3, v3, v0
  
      iget v2, v3, Landroid/view/VolumePanel$StreamResources;->streamType:I
  
+     .line 334
      .local v2, streamType:I
      sget-object v3, Landroid/view/VolumePanel;->STREAMS:[Landroid/view/VolumePanel$StreamResources;
  
***************
*** 485,496 ****
--- 534,547 ----
  
      if-ne v2, v3, :cond_3
  
+     .line 331
      :cond_2
      :goto_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 337
      :cond_3
      iget-object v3, p0, Landroid/view/VolumePanel;->mStreamControls:Ljava/util/HashMap;
  
***************
*** 504,509 ****
--- 555,561 ----
  
      check-cast v1, Landroid/view/VolumePanel$StreamControl;
  
+     .line 338
      .local v1, sc:Landroid/view/VolumePanel$StreamControl;
      iget-object v3, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
***************
*** 511,516 ****
--- 563,569 ----
  
      invoke-virtual {v3, v4}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
  
+     .line 339
      invoke-direct {p0, v1}, Landroid/view/VolumePanel;->updateSlider(Landroid/view/VolumePanel$StreamControl;)V
  
      goto :goto_1
***************
*** 522,541 ****
--- 575,598 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 368
      iget-object v2, p0, Landroid/view/VolumePanel;->mMoreButton:Landroid/view/View;
  
      invoke-virtual {v2, v3}, Landroid/view/View;->setVisibility(I)V
  
+     .line 369
      iget-object v2, p0, Landroid/view/VolumePanel;->mDivider:Landroid/view/View;
  
      invoke-virtual {v2, v3}, Landroid/view/View;->setVisibility(I)V
  
+     .line 370
      iget-object v2, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
      invoke-virtual {v2}, Landroid/view/ViewGroup;->getChildCount()I
  
      move-result v0
  
+     .line 371
      .local v0, count:I
      const/4 v1, 0x1
  
***************
*** 543,548 ****
--- 600,606 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 372
      iget-object v2, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
      invoke-virtual {v2, v1}, Landroid/view/ViewGroup;->getChildAt(I)Landroid/view/View;
***************
*** 553,562 ****
--- 611,622 ----
  
      invoke-virtual {v2, v3}, Landroid/view/View;->setVisibility(I)V
  
+     .line 371
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 374
      :cond_0
      return-void
  .end method
***************
*** 567,572 ****
--- 627,633 ----
      .prologue
      const/4 v9, 0x0
  
+     .line 281
      iget-object v7, p0, Landroid/view/VolumePanel;->mContext:Landroid/content/Context;
  
      const-string v8, "layout_inflater"
***************
*** 577,582 ****
--- 638,644 ----
  
      check-cast v1, Landroid/view/LayoutInflater;
  
+     .line 283
      .local v1, inflater:Landroid/view/LayoutInflater;
      new-instance v7, Ljava/util/HashMap;
  
***************
*** 588,599 ****
--- 650,663 ----
  
      iput-object v7, p0, Landroid/view/VolumePanel;->mStreamControls:Ljava/util/HashMap;
  
+     .line 284
      iget-object v7, p0, Landroid/view/VolumePanel;->mContext:Landroid/content/Context;
  
      invoke-virtual {v7}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v3
  
+     .line 285
      .local v3, res:Landroid/content/res/Resources;
      const/4 v0, 0x0
  
***************
*** 605,617 ****
--- 669,684 ----
  
      if-ge v0, v7, :cond_3
  
+     .line 286
      sget-object v7, Landroid/view/VolumePanel;->STREAMS:[Landroid/view/VolumePanel$StreamResources;
  
      aget-object v5, v7, v0
  
+     .line 287
      .local v5, streamRes:Landroid/view/VolumePanel$StreamResources;
      iget v6, v5, Landroid/view/VolumePanel$StreamResources;->streamType:I
  
+     .line 288
      .local v6, streamType:I
      iget-boolean v7, p0, Landroid/view/VolumePanel;->mVoiceCapable:Z
  
***************
*** 621,636 ****
--- 688,707 ----
  
      if-ne v5, v7, :cond_0
  
+     .line 289
      sget-object v5, Landroid/view/VolumePanel$StreamResources;->RingerStream:Landroid/view/VolumePanel$StreamResources;
  
+     .line 291
      :cond_0
      new-instance v4, Landroid/view/VolumePanel$StreamControl;
  
      invoke-direct {v4, p0, v9}, Landroid/view/VolumePanel$StreamControl;-><init>(Landroid/view/VolumePanel;Landroid/view/VolumePanel$1;)V
  
+     .line 292
      .local v4, sc:Landroid/view/VolumePanel$StreamControl;
      iput v6, v4, Landroid/view/VolumePanel$StreamControl;->streamType:I
  
+     .line 293
      const v7, 0x10900b0
  
      invoke-virtual {v1, v7, v9}, Landroid/view/LayoutInflater;->inflate(ILandroid/view/ViewGroup;)Landroid/view/View;
***************
*** 641,650 ****
--- 712,723 ----
  
      iput-object v7, v4, Landroid/view/VolumePanel$StreamControl;->group:Landroid/view/ViewGroup;
  
+     .line 294
      iget-object v7, v4, Landroid/view/VolumePanel$StreamControl;->group:Landroid/view/ViewGroup;
  
      invoke-virtual {v7, v4}, Landroid/view/ViewGroup;->setTag(Ljava/lang/Object;)V
  
+     .line 295
      iget-object v7, v4, Landroid/view/VolumePanel$StreamControl;->group:Landroid/view/ViewGroup;
  
      const v8, 0x1020332
***************
*** 657,666 ****
--- 730,741 ----
  
      iput-object v7, v4, Landroid/view/VolumePanel$StreamControl;->icon:Landroid/widget/ImageView;
  
+     .line 296
      iget-object v7, v4, Landroid/view/VolumePanel$StreamControl;->icon:Landroid/widget/ImageView;
  
      invoke-virtual {v7, v4}, Landroid/widget/ImageView;->setTag(Ljava/lang/Object;)V
  
+     .line 297
      iget-object v7, v4, Landroid/view/VolumePanel$StreamControl;->icon:Landroid/widget/ImageView;
  
      iget v8, v5, Landroid/view/VolumePanel$StreamResources;->descRes:I
***************
*** 671,690 ****
--- 746,769 ----
  
      invoke-virtual {v7, v8}, Landroid/widget/ImageView;->setContentDescription(Ljava/lang/CharSequence;)V
  
+     .line 298
      iget v7, v5, Landroid/view/VolumePanel$StreamResources;->iconRes:I
  
      iput v7, v4, Landroid/view/VolumePanel$StreamControl;->iconRes:I
  
+     .line 299
      iget v7, v5, Landroid/view/VolumePanel$StreamResources;->iconMuteRes:I
  
      iput v7, v4, Landroid/view/VolumePanel$StreamControl;->iconMuteRes:I
  
+     .line 300
      iget-object v7, v4, Landroid/view/VolumePanel$StreamControl;->icon:Landroid/widget/ImageView;
  
      iget v8, v4, Landroid/view/VolumePanel$StreamControl;->iconRes:I
  
      invoke-virtual {v7, v8}, Landroid/widget/ImageView;->setImageResource(I)V
  
+     .line 301
      iget-object v7, v4, Landroid/view/VolumePanel$StreamControl;->group:Landroid/view/ViewGroup;
  
      const v8, 0x10202d9
***************
*** 697,702 ****
--- 776,782 ----
  
      iput-object v7, v4, Landroid/view/VolumePanel$StreamControl;->seekbarView:Landroid/widget/SeekBar;
  
+     .line 302
      const/4 v7, 0x6
  
      if-eq v6, v7, :cond_1
***************
*** 706,711 ****
--- 786,792 ----
      :cond_1
      const/4 v2, 0x1
  
+     .line 304
      .local v2, plusOne:I
      :goto_1
      iget-object v7, v4, Landroid/view/VolumePanel$StreamControl;->seekbarView:Landroid/widget/SeekBar;
***************
*** 720,733 ****
--- 801,817 ----
  
      invoke-virtual {v7, v8}, Landroid/widget/SeekBar;->setMax(I)V
  
+     .line 305
      iget-object v7, v4, Landroid/view/VolumePanel$StreamControl;->seekbarView:Landroid/widget/SeekBar;
  
      invoke-virtual {v7, p0}, Landroid/widget/SeekBar;->setOnSeekBarChangeListener(Landroid/widget/SeekBar$OnSeekBarChangeListener;)V
  
+     .line 306
      iget-object v7, v4, Landroid/view/VolumePanel$StreamControl;->seekbarView:Landroid/widget/SeekBar;
  
      invoke-virtual {v7, v4}, Landroid/widget/SeekBar;->setTag(Ljava/lang/Object;)V
  
+     .line 307
      iget-object v7, p0, Landroid/view/VolumePanel;->mStreamControls:Ljava/util/HashMap;
  
      invoke-static {v6}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 736,751 ****
--- 820,838 ----
  
      invoke-virtual {v7, v8, v4}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 285
      add-int/lit8 v0, v0, 0x1
  
      goto/16 :goto_0
  
+     .line 302
      .end local v2           #plusOne:I
      :cond_2
      const/4 v2, 0x0
  
      goto :goto_1
  
+     .line 309
      .end local v4           #sc:Landroid/view/VolumePanel$StreamControl;
      .end local v5           #streamRes:Landroid/view/VolumePanel$StreamResources;
      .end local v6           #streamType:I
***************
*** 759,770 ****
--- 846,859 ----
      .prologue
      const/4 v4, 0x4
  
+     .line 359
      iget-object v2, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
      invoke-virtual {v2}, Landroid/view/ViewGroup;->getChildCount()I
  
      move-result v0
  
+     .line 360
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 772,777 ****
--- 861,867 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 361
      iget-object v2, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
      invoke-virtual {v2, v1}, Landroid/view/ViewGroup;->getChildAt(I)Landroid/view/View;
***************
*** 782,800 ****
--- 872,894 ----
  
      invoke-virtual {v2, v3}, Landroid/view/View;->setVisibility(I)V
  
+     .line 360
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 363
      :cond_0
      iget-object v2, p0, Landroid/view/VolumePanel;->mMoreButton:Landroid/view/View;
  
      invoke-virtual {v2, v4}, Landroid/view/View;->setVisibility(I)V
  
+     .line 364
      iget-object v2, p0, Landroid/view/VolumePanel;->mDivider:Landroid/view/View;
  
      invoke-virtual {v2, v4}, Landroid/view/View;->setVisibility(I)V
  
+     .line 365
      return-void
  .end method
  
***************
*** 804,817 ****
--- 898,914 ----
      .prologue
      const/4 v0, 0x5
  
+     .line 665
      invoke-virtual {p0, v0}, Landroid/view/VolumePanel;->removeMessages(I)V
  
+     .line 666
      invoke-virtual {p0, v0}, Landroid/view/VolumePanel;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
      invoke-virtual {p0, v0}, Landroid/view/VolumePanel;->sendMessage(Landroid/os/Message;)Z
  
+     .line 667
      return-void
  .end method
  
***************
*** 820,827 ****
--- 917,926 ----
      .parameter "streamType"
  
      .prologue
+     .line 574
      monitor-enter p0
  
+     .line 575
      :try_start_0
      iget-object v1, p0, Landroid/view/VolumePanel;->mToneGenerators:[Landroid/media/ToneGenerator;
  
***************
*** 831,836 ****
--- 930,936 ----
  
      if-nez v1, :cond_0
  
+     .line 577
      :try_start_1
      iget-object v1, p0, Landroid/view/VolumePanel;->mToneGenerators:[Landroid/media/ToneGenerator;
  
***************
*** 845,850 ****
--- 945,951 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Ljava/lang/RuntimeException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 585
      :cond_0
      :goto_0
      :try_start_2
***************
*** 856,869 ****
--- 957,973 ----
  
      return-object v1
  
+     .line 578
      :catch_0
      move-exception v0
  
+     .line 579
      .local v0, e:Ljava/lang/RuntimeException;
      sget-boolean v1, Landroid/view/VolumePanel;->LOGD:Z
  
      if-eqz v1, :cond_0
  
+     .line 580
      const-string v1, "VolumePanel"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 888,893 ****
--- 992,998 ----
  
      goto :goto_0
  
+     .line 586
      .end local v0           #e:Ljava/lang/RuntimeException;
      :catchall_0
      move-exception v1
***************
*** 903,908 ****
--- 1008,1014 ----
      .locals 1
  
      .prologue
+     .line 355
      iget-object v0, p0, Landroid/view/VolumePanel;->mMoreButton:Landroid/view/View;
  
      invoke-virtual {v0}, Landroid/view/View;->getVisibility()I
***************
*** 927,932 ****
--- 1033,1039 ----
      .parameter "streamType"
  
      .prologue
+     .line 277
      iget-object v0, p0, Landroid/view/VolumePanel;->mAudioManager:Landroid/media/AudioManager;
  
      invoke-virtual {v0, p1}, Landroid/media/AudioManager;->isStreamMute(I)Z
***************
*** 940,954 ****
--- 1047,1064 ----
      .locals 3
  
      .prologue
+     .line 261
      new-instance v0, Landroid/content/IntentFilter;
  
      invoke-direct {v0}, Landroid/content/IntentFilter;-><init>()V
  
+     .line 262
      .local v0, filter:Landroid/content/IntentFilter;
      const-string v1, "android.media.RINGER_MODE_CHANGED"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 263
      iget-object v1, p0, Landroid/view/VolumePanel;->mContext:Landroid/content/Context;
  
      new-instance v2, Landroid/view/VolumePanel$4;
***************
*** 957,962 ****
--- 1067,1073 ----
  
      invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 274
      return-void
  .end method
  
***************
*** 965,974 ****
--- 1076,1087 ----
      .parameter "activeStreamType"
  
      .prologue
+     .line 312
      iget-object v1, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
      invoke-virtual {v1}, Landroid/view/ViewGroup;->removeAllViews()V
  
+     .line 314
      iget-object v1, p0, Landroid/view/VolumePanel;->mStreamControls:Ljava/util/HashMap;
  
      invoke-static {p1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 981,989 ****
--- 1094,1104 ----
  
      check-cast v0, Landroid/view/VolumePanel$StreamControl;
  
+     .line 315
      .local v0, active:Landroid/view/VolumePanel$StreamControl;
      if-nez v0, :cond_0
  
+     .line 316
      const-string v1, "VolumePanel"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 1006,1020 ****
--- 1121,1139 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 317
      const/4 v1, -0x1
  
      iput v1, p0, Landroid/view/VolumePanel;->mActiveStreamType:I
  
+     .line 325
      :goto_0
      invoke-direct {p0}, Landroid/view/VolumePanel;->addOtherVolumes()V
  
+     .line 326
      return-void
  
+     .line 319
      :cond_0
      iget-object v1, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
***************
*** 1022,1035 ****
--- 1141,1157 ----
  
      invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
  
+     .line 320
      iput p1, p0, Landroid/view/VolumePanel;->mActiveStreamType:I
  
+     .line 321
      iget-object v1, v0, Landroid/view/VolumePanel$StreamControl;->group:Landroid/view/ViewGroup;
  
      const/4 v2, 0x0
  
      invoke-virtual {v1, v2}, Landroid/view/ViewGroup;->setVisibility(I)V
  
+     .line 322
      invoke-direct {p0, v0}, Landroid/view/VolumePanel;->updateSlider(Landroid/view/VolumePanel$StreamControl;)V
  
      goto :goto_0
***************
*** 1041,1048 ****
--- 1163,1172 ----
      .prologue
      const/4 v0, 0x5
  
+     .line 660
      invoke-virtual {p0, v0}, Landroid/view/VolumePanel;->removeMessages(I)V
  
+     .line 661
      invoke-virtual {p0, v0}, Landroid/view/VolumePanel;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
***************
*** 1051,1056 ****
--- 1175,1181 ----
  
      invoke-virtual {p0, v0, v1, v2}, Landroid/view/VolumePanel;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 662
      return-void
  .end method
  
***************
*** 1060,1065 ****
--- 1185,1191 ----
      .parameter "resMuteId"
  
      .prologue
+     .line 595
      iget-object v1, p0, Landroid/view/VolumePanel;->mStreamControls:Ljava/util/HashMap;
  
      const/4 v2, 0x3
***************
*** 1074,1086 ****
--- 1200,1216 ----
  
      check-cast v0, Landroid/view/VolumePanel$StreamControl;
  
+     .line 596
      .local v0, sc:Landroid/view/VolumePanel$StreamControl;
      if-eqz v0, :cond_0
  
+     .line 597
      iput p1, v0, Landroid/view/VolumePanel$StreamControl;->iconRes:I
  
+     .line 598
      iput p2, v0, Landroid/view/VolumePanel$StreamControl;->iconMuteRes:I
  
+     .line 599
      iget-object v2, v0, Landroid/view/VolumePanel$StreamControl;->icon:Landroid/widget/ImageView;
  
      iget v1, v0, Landroid/view/VolumePanel$StreamControl;->streamType:I
***************
*** 1096,1104 ****
--- 1226,1236 ----
      :goto_0
      invoke-virtual {v2, v1}, Landroid/widget/ImageView;->setImageResource(I)V
  
+     .line 601
      :cond_0
      return-void
  
+     .line 599
      :cond_1
      iget v1, v0, Landroid/view/VolumePanel$StreamControl;->iconRes:I
  
***************
*** 1110,1115 ****
--- 1242,1248 ----
      .parameter "sc"
  
      .prologue
+     .line 345
      iget-object v1, p1, Landroid/view/VolumePanel$StreamControl;->seekbarView:Landroid/widget/SeekBar;
  
      iget-object v2, p0, Landroid/view/VolumePanel;->mAudioManager:Landroid/media/AudioManager;
***************
*** 1122,1133 ****
--- 1255,1268 ----
  
      invoke-virtual {v1, v2}, Landroid/widget/SeekBar;->setProgress(I)V
  
+     .line 346
      iget v1, p1, Landroid/view/VolumePanel$StreamControl;->streamType:I
  
      invoke-direct {p0, v1}, Landroid/view/VolumePanel;->isMuted(I)Z
  
      move-result v0
  
+     .line 347
      .local v0, muted:Z
      iget-object v2, p1, Landroid/view/VolumePanel$StreamControl;->icon:Landroid/widget/ImageView;
  
***************
*** 1138,1143 ****
--- 1273,1279 ----
      :goto_0
      invoke-virtual {v2, v1}, Landroid/widget/ImageView;->setImageResource(I)V
  
+     .line 348
      iget v1, p1, Landroid/view/VolumePanel$StreamControl;->streamType:I
  
      const/4 v2, 0x2
***************
*** 1156,1170 ****
--- 1292,1309 ----
  
      if-eqz v1, :cond_0
  
+     .line 350
      iget-object v1, p1, Landroid/view/VolumePanel$StreamControl;->icon:Landroid/widget/ImageView;
  
      const v2, 0x108029a
  
      invoke-virtual {v1, v2}, Landroid/widget/ImageView;->setImageResource(I)V
  
+     .line 352
      :cond_0
      return-void
  
+     .line 347
      :cond_1
      iget v1, p1, Landroid/view/VolumePanel$StreamControl;->iconRes:I
  
***************
*** 1175,1186 ****
--- 1314,1327 ----
      .locals 4
  
      .prologue
+     .line 377
      iget-object v3, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
      invoke-virtual {v3}, Landroid/view/ViewGroup;->getChildCount()I
  
      move-result v0
  
+     .line 378
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 1188,1193 ****
--- 1329,1335 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 379
      iget-object v3, p0, Landroid/view/VolumePanel;->mSliderGroup:Landroid/view/ViewGroup;
  
      invoke-virtual {v3, v1}, Landroid/view/ViewGroup;->getChildAt(I)Landroid/view/View;
***************
*** 1200,1212 ****
--- 1342,1357 ----
  
      check-cast v2, Landroid/view/VolumePanel$StreamControl;
  
+     .line 380
      .local v2, sc:Landroid/view/VolumePanel$StreamControl;
      invoke-direct {p0, v2}, Landroid/view/VolumePanel;->updateSlider(Landroid/view/VolumePanel$StreamControl;)V
  
+     .line 378
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 382
      .end local v2           #sc:Landroid/view/VolumePanel$StreamControl;
      :cond_0
      return-void
***************
*** 1219,1232 ****
--- 1364,1380 ----
      .parameter "msg"
  
      .prologue
+     .line 616
      iget v0, p1, Landroid/os/Message;->what:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 657
      :cond_0
      :goto_0
      return-void
  
+     .line 619
      :pswitch_0
      iget v0, p1, Landroid/os/Message;->arg1:I
  
***************
*** 1236,1251 ****
--- 1384,1402 ----
  
      goto :goto_0
  
+     .line 624
      :pswitch_1
      invoke-virtual {p0}, Landroid/view/VolumePanel;->onFreeResources()V
  
      goto :goto_0
  
+     .line 629
      :pswitch_2
      invoke-virtual {p0}, Landroid/view/VolumePanel;->onStopSounds()V
  
      goto :goto_0
  
+     .line 634
      :pswitch_3
      iget v0, p1, Landroid/os/Message;->arg1:I
  
***************
*** 1255,1265 ****
--- 1406,1418 ----
  
      goto :goto_0
  
+     .line 639
      :pswitch_4
      invoke-virtual {p0}, Landroid/view/VolumePanel;->onVibrate()V
  
      goto :goto_0
  
+     .line 644
      :pswitch_5
      iget-object v0, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
***************
*** 1269,1284 ****
--- 1422,1440 ----
  
      if-eqz v0, :cond_0
  
+     .line 645
      iget-object v0, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
      invoke-virtual {v0}, Landroid/app/Dialog;->dismiss()V
  
+     .line 646
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/view/VolumePanel;->mActiveStreamType:I
  
      goto :goto_0
  
+     .line 651
      :pswitch_6
      iget-object v0, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
***************
*** 1288,1297 ****
--- 1444,1455 ----
  
      if-eqz v0, :cond_0
  
+     .line 652
      invoke-direct {p0}, Landroid/view/VolumePanel;->updateStates()V
  
      goto :goto_0
  
+     .line 616
      nop
  
      :pswitch_data_0
***************
*** 1311,1325 ****
--- 1469,1487 ----
      .parameter "v"
  
      .prologue
+     .line 688
      iget-object v0, p0, Landroid/view/VolumePanel;->mMoreButton:Landroid/view/View;
  
      if-ne p1, v0, :cond_0
  
+     .line 689
      invoke-direct {p0}, Landroid/view/VolumePanel;->expand()V
  
+     .line 691
      :cond_0
      invoke-direct {p0}, Landroid/view/VolumePanel;->resetTimeout()V
  
+     .line 692
      return-void
  .end method
  
***************
*** 1327,1334 ****
--- 1489,1498 ----
      .locals 3
  
      .prologue
+     .line 604
      monitor-enter p0
  
+     .line 605
      :try_start_0
      iget-object v1, p0, Landroid/view/VolumePanel;->mToneGenerators:[Landroid/media/ToneGenerator;
  
***************
*** 1340,1357 ****
--- 1504,1524 ----
      :goto_0
      if-ltz v0, :cond_1
  
+     .line 606
      iget-object v1, p0, Landroid/view/VolumePanel;->mToneGenerators:[Landroid/media/ToneGenerator;
  
      aget-object v1, v1, v0
  
      if-eqz v1, :cond_0
  
+     .line 607
      iget-object v1, p0, Landroid/view/VolumePanel;->mToneGenerators:[Landroid/media/ToneGenerator;
  
      aget-object v1, v1, v0
  
      invoke-virtual {v1}, Landroid/media/ToneGenerator;->release()V
  
+     .line 609
      :cond_0
      iget-object v1, p0, Landroid/view/VolumePanel;->mToneGenerators:[Landroid/media/ToneGenerator;
  
***************
*** 1359,1373 ****
--- 1526,1544 ----
  
      aput-object v2, v1, v0
  
+     .line 605
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_0
  
+     .line 611
      :cond_1
      monitor-exit p0
  
+     .line 612
      return-void
  
+     .line 611
      .end local v0           #i:I
      :catchall_0
      move-exception v1
***************
*** 1387,1417 ****
--- 1558,1596 ----
      .prologue
      const/4 v2, 0x3
  
+     .line 532
      invoke-virtual {p0, v2}, Landroid/view/VolumePanel;->hasMessages(I)Z
  
      move-result v1
  
      if-eqz v1, :cond_0
  
+     .line 533
      invoke-virtual {p0, v2}, Landroid/view/VolumePanel;->removeMessages(I)V
  
+     .line 535
      invoke-virtual {p0}, Landroid/view/VolumePanel;->onStopSounds()V
  
+     .line 538
      :cond_0
      monitor-enter p0
  
+     .line 539
      :try_start_0
      invoke-direct {p0, p1}, Landroid/view/VolumePanel;->getOrCreateToneGenerator(I)Landroid/media/ToneGenerator;
  
      move-result-object v0
  
+     .line 540
      .local v0, toneGen:Landroid/media/ToneGenerator;
      if-eqz v0, :cond_1
  
+     .line 541
      const/16 v1, 0x18
  
      invoke-virtual {v0, v1}, Landroid/media/ToneGenerator;->startTone(I)Z
  
+     .line 542
      const/4 v1, 0x3
  
      invoke-virtual {p0, v1}, Landroid/view/VolumePanel;->obtainMessage(I)Landroid/os/Message;
***************
*** 1422,1432 ****
--- 1601,1614 ----
  
      invoke-virtual {p0, v1, v2, v3}, Landroid/view/VolumePanel;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 544
      :cond_1
      monitor-exit p0
  
+     .line 545
      return-void
  
+     .line 544
      .end local v0           #toneGen:Landroid/media/ToneGenerator;
      :catchall_0
      move-exception v1
***************
*** 1445,1454 ****
--- 1627,1638 ----
      .parameter "fromUser"
  
      .prologue
+     .line 671
      invoke-virtual {p1}, Landroid/widget/SeekBar;->getTag()Ljava/lang/Object;
  
      move-result-object v1
  
+     .line 672
      .local v1, tag:Ljava/lang/Object;
      if-eqz p3, :cond_0
  
***************
*** 1458,1465 ****
--- 1642,1651 ----
  
      move-object v0, v1
  
+     .line 673
      check-cast v0, Landroid/view/VolumePanel$StreamControl;
  
+     .line 674
      .local v0, sc:Landroid/view/VolumePanel$StreamControl;
      iget-object v2, p0, Landroid/view/VolumePanel;->mAudioManager:Landroid/media/AudioManager;
  
***************
*** 1471,1476 ****
--- 1657,1663 ----
  
      if-eq v2, p2, :cond_0
  
+     .line 675
      iget-object v2, p0, Landroid/view/VolumePanel;->mAudioManager:Landroid/media/AudioManager;
  
      iget v3, v0, Landroid/view/VolumePanel$StreamControl;->streamType:I
***************
*** 1479,1488 ****
--- 1666,1677 ----
  
      invoke-virtual {v2, v3, p2, v4}, Landroid/media/AudioManager;->setStreamVolume(III)V
  
+     .line 678
      .end local v0           #sc:Landroid/view/VolumePanel$StreamControl;
      :cond_0
      invoke-direct {p0}, Landroid/view/VolumePanel;->resetTimeout()V
  
+     .line 679
      return-void
  .end method
  
***************
*** 1496,1501 ****
--- 1685,1691 ----
  
      const/4 v7, 0x1
  
+     .line 427
      iget-object v4, p0, Landroid/view/VolumePanel;->mAudioService:Landroid/media/AudioService;
  
      invoke-virtual {v4, p1}, Landroid/media/AudioService;->isStreamMute(I)Z
***************
*** 1510,1530 ****
  
      move-result v0
  
      .local v0, index:I
      :goto_0
      iput-boolean v8, p0, Landroid/view/VolumePanel;->mRingIsSilent:Z
  
      sget-boolean v4, Landroid/view/VolumePanel;->LOGD:Z
  
      if-eqz v4, :cond_0
  
      const-string v4, "VolumePanel"
  
      new-instance v5, Ljava/lang/StringBuilder;
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v6, "onShowVolumeChanged(streamType: "
  
      invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1700,1723 ----
  
      move-result v0
  
+     .line 431
      .local v0, index:I
      :goto_0
      iput-boolean v8, p0, Landroid/view/VolumePanel;->mRingIsSilent:Z
  
+     .line 433
      sget-boolean v4, Landroid/view/VolumePanel;->LOGD:Z
  
      if-eqz v4, :cond_0
  
+     .line 434
      const-string v4, "VolumePanel"
  
      new-instance v5, Ljava/lang/StringBuilder;
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v6, "onShowVolumeChanged(streamType: "
  
      invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1560,1565 ****
--- 1753,1759 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 440
      :cond_0
      iget-object v4, p0, Landroid/view/VolumePanel;->mAudioService:Landroid/media/AudioService;
  
***************
*** 1567,1575 ****
--- 1761,1771 ----
  
      move-result v1
  
+     .line 442
      .local v1, max:I
      packed-switch p1, :pswitch_data_0
  
+     .line 503
      :cond_1
      :goto_1
      :pswitch_0
***************
*** 1585,1593 ****
--- 1781,1791 ----
  
      check-cast v3, Landroid/view/VolumePanel$StreamControl;
  
+     .line 504
      .local v3, sc:Landroid/view/VolumePanel$StreamControl;
      if-eqz v3, :cond_3
  
+     .line 505
      iget-object v4, v3, Landroid/view/VolumePanel$StreamControl;->seekbarView:Landroid/widget/SeekBar;
  
      invoke-virtual {v4}, Landroid/widget/SeekBar;->getMax()I
***************
*** 1596,1610 ****
--- 1794,1811 ----
  
      if-eq v4, v1, :cond_2
  
+     .line 506
      iget-object v4, v3, Landroid/view/VolumePanel$StreamControl;->seekbarView:Landroid/widget/SeekBar;
  
      invoke-virtual {v4, v1}, Landroid/widget/SeekBar;->setMax(I)V
  
+     .line 508
      :cond_2
      iget-object v4, v3, Landroid/view/VolumePanel$StreamControl;->seekbarView:Landroid/widget/SeekBar;
  
      invoke-virtual {v4, v0}, Landroid/widget/SeekBar;->setProgress(I)V
  
+     .line 511
      :cond_3
      iget-object v4, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
***************
*** 1614,1640 ****
--- 1815,1847 ----
  
      if-nez v4, :cond_5
  
+     .line 512
      iget-object v4, p0, Landroid/view/VolumePanel;->mAudioManager:Landroid/media/AudioManager;
  
      invoke-virtual {v4, p1}, Landroid/media/AudioManager;->forceVolumeControlStream(I)V
  
+     .line 513
      iget-object v4, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
      iget-object v5, p0, Landroid/view/VolumePanel;->mView:Landroid/view/View;
  
      invoke-virtual {v4, v5}, Landroid/app/Dialog;->setContentView(Landroid/view/View;)V
  
+     .line 515
      iget-boolean v4, p0, Landroid/view/VolumePanel;->mShowCombinedVolumes:Z
  
      if-eqz v4, :cond_4
  
+     .line 516
      invoke-direct {p0}, Landroid/view/VolumePanel;->collapse()V
  
+     .line 518
      :cond_4
      iget-object v4, p0, Landroid/view/VolumePanel;->mDialog:Landroid/app/Dialog;
  
      invoke-virtual {v4}, Landroid/app/Dialog;->show()V
  
+     .line 522
      :cond_5
      and-int/lit8 v4, p2, 0x10
  
***************
*** 1664,1669 ****
--- 1871,1877 ----
  
      if-eqz v4, :cond_6
  
+     .line 526
      const/4 v4, 0x4
  
      invoke-virtual {p0, v4}, Landroid/view/VolumePanel;->obtainMessage(I)Landroid/os/Message;
***************
*** 1674,1682 ****
--- 1882,1892 ----
  
      invoke-virtual {p0, v4, v5, v6}, Landroid/view/VolumePanel;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 528
      :cond_6
      return-void
  
+     .line 427
      .end local v0           #index:I
      .end local v1           #max:I
      .end local v3           #sc:Landroid/view/VolumePanel$StreamControl;
***************
*** 1689,1694 ****
--- 1899,1905 ----
  
      goto/16 :goto_0
  
+     .line 446
      .restart local v0       #index:I
      .restart local v1       #max:I
      :pswitch_1
***************
*** 1698,1710 ****
--- 1909,1924 ----
  
      move-result-object v2
  
+     .line 448
      .local v2, ringuri:Landroid/net/Uri;
      if-nez v2, :cond_1
  
+     .line 449
      iput-boolean v7, p0, Landroid/view/VolumePanel;->mRingIsSilent:Z
  
      goto :goto_1
  
+     .line 456
      .end local v2           #ringuri:Landroid/net/Uri;
      :pswitch_2
      iget-object v4, p0, Landroid/view/VolumePanel;->mAudioManager:Landroid/media/AudioManager;
***************
*** 1719,1724 ****
--- 1933,1939 ----
  
      if-eqz v4, :cond_8
  
+     .line 460
      const v4, 0x1080293
  
      const v5, 0x1080294
***************
*** 1727,1732 ****
--- 1942,1948 ----
  
      goto/16 :goto_1
  
+     .line 462
      :cond_8
      const v4, 0x108029b
  
***************
*** 1736,1748 ****
--- 1952,1968 ----
  
      goto/16 :goto_1
  
+     .line 473
      :pswitch_3
      add-int/lit8 v0, v0, 0x1
  
+     .line 474
      add-int/lit8 v1, v1, 0x1
  
+     .line 475
      goto/16 :goto_1
  
+     .line 483
      :pswitch_4
      iget-object v4, p0, Landroid/view/VolumePanel;->mContext:Landroid/content/Context;
  
***************
*** 1752,1772 ****
--- 1972,1997 ----
  
      move-result-object v2
  
+     .line 485
      .restart local v2       #ringuri:Landroid/net/Uri;
      if-nez v2, :cond_1
  
+     .line 486
      iput-boolean v7, p0, Landroid/view/VolumePanel;->mRingIsSilent:Z
  
      goto/16 :goto_1
  
+     .line 497
      .end local v2           #ringuri:Landroid/net/Uri;
      :pswitch_5
      add-int/lit8 v0, v0, 0x1
  
+     .line 498
      add-int/lit8 v1, v1, 0x1
  
      goto/16 :goto_1
  
+     .line 442
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_3
***************
*** 1784,1789 ****
--- 2009,2015 ----
      .parameter "seekBar"
  
      .prologue
+     .line 682
      return-void
  .end method
  
***************
*** 1791,1803 ****
--- 2017,2032 ----
      .locals 4
  
      .prologue
+     .line 549
      monitor-enter p0
  
+     .line 550
      :try_start_0
      invoke-static {}, Landroid/media/AudioSystem;->getNumStreamTypes()I
  
      move-result v1
  
+     .line 551
      .local v1, numStreamTypes:I
      add-int/lit8 v0, v1, -0x1
  
***************
*** 1805,1830 ****
--- 2034,2066 ----
      :goto_0
      if-ltz v0, :cond_1
  
+     .line 552
      iget-object v3, p0, Landroid/view/VolumePanel;->mToneGenerators:[Landroid/media/ToneGenerator;
  
      aget-object v2, v3, v0
  
+     .line 553
      .local v2, toneGen:Landroid/media/ToneGenerator;
      if-eqz v2, :cond_0
  
+     .line 554
      invoke-virtual {v2}, Landroid/media/ToneGenerator;->stopTone()V
  
+     .line 551
      :cond_0
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_0
  
+     .line 557
      .end local v2           #toneGen:Landroid/media/ToneGenerator;
      :cond_1
      monitor-exit p0
  
+     .line 558
      return-void
  
+     .line 557
      .end local v0           #i:I
      .end local v1           #numStreamTypes:I
      :catchall_0
***************
*** 1842,1847 ****
--- 2078,2084 ----
      .parameter "seekBar"
  
      .prologue
+     .line 685
      return-void
  .end method
  
***************
*** 1849,1854 ****
--- 2086,2092 ----
      .locals 3
  
      .prologue
+     .line 563
      iget-object v0, p0, Landroid/view/VolumePanel;->mAudioService:Landroid/media/AudioService;
  
      invoke-virtual {v0}, Landroid/media/AudioService;->getRingerMode()I
***************
*** 1859,1867 ****
--- 2097,2107 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 568
      :goto_0
      return-void
  
+     .line 567
      :cond_0
      iget-object v0, p0, Landroid/view/VolumePanel;->mVibrator:Landroid/os/Vibrator;
  
***************
*** 1882,1887 ****
--- 2122,2128 ----
  
      const/4 v3, 0x2
  
+     .line 400
      sget-boolean v0, Landroid/view/VolumePanel;->LOGD:Z
  
      if-eqz v0, :cond_0
***************
*** 1892,1898 ****
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "onVolumeChanged(streamType: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2133,2139 ----
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "onVolumeChanged(streamType: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1924,1945 ****
--- 2165,2191 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 402
      :cond_0
      and-int/lit8 v0, p2, 0x1
  
      if-eqz v0, :cond_2
  
+     .line 403
      iget v0, p0, Landroid/view/VolumePanel;->mActiveStreamType:I
  
      const/4 v1, -0x1
  
      if-ne v0, v1, :cond_1
  
+     .line 404
      invoke-direct {p0, p1}, Landroid/view/VolumePanel;->reorderSliders(I)V
  
+     .line 406
      :cond_1
      invoke-virtual {p0, p1, p2}, Landroid/view/VolumePanel;->onShowVolumeChanged(II)V
  
+     .line 409
      :cond_2
      and-int/lit8 v0, p2, 0x4
  
***************
*** 1949,1956 ****
--- 2195,2204 ----
  
      if-nez v0, :cond_3
  
+     .line 410
      invoke-virtual {p0, v3}, Landroid/view/VolumePanel;->removeMessages(I)V
  
+     .line 411
      invoke-virtual {p0, v3, p1, p2}, Landroid/view/VolumePanel;->obtainMessage(III)Landroid/os/Message;
  
      move-result-object v0
***************
*** 1959,1980 ****
--- 2207,2234 ----
  
      invoke-virtual {p0, v0, v1, v2}, Landroid/view/VolumePanel;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 414
      :cond_3
      and-int/lit8 v0, p2, 0x8
  
      if-eqz v0, :cond_4
  
+     .line 415
      invoke-virtual {p0, v3}, Landroid/view/VolumePanel;->removeMessages(I)V
  
+     .line 416
      const/4 v0, 0x4
  
      invoke-virtual {p0, v0}, Landroid/view/VolumePanel;->removeMessages(I)V
  
+     .line 417
      invoke-virtual {p0}, Landroid/view/VolumePanel;->onStopSounds()V
  
+     .line 420
      :cond_4
      invoke-virtual {p0, v4}, Landroid/view/VolumePanel;->removeMessages(I)V
  
+     .line 421
      invoke-virtual {p0, v4}, Landroid/view/VolumePanel;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
***************
*** 1983,1990 ****
--- 2237,2246 ----
  
      invoke-virtual {p0, v0, v1, v2}, Landroid/view/VolumePanel;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 423
      invoke-direct {p0}, Landroid/view/VolumePanel;->resetTimeout()V
  
+     .line 424
      return-void
  .end method
  
***************
*** 1996,2022 ****
--- 2252,2284 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 385
      invoke-virtual {p0, v1}, Landroid/view/VolumePanel;->hasMessages(I)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 391
      :goto_0
      return-void
  
+     .line 386
      :cond_0
      iget-object v0, p0, Landroid/view/VolumePanel;->mStreamControls:Ljava/util/HashMap;
  
      if-nez v0, :cond_1
  
+     .line 387
      invoke-direct {p0}, Landroid/view/VolumePanel;->createSliders()V
  
+     .line 389
      :cond_1
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Landroid/view/VolumePanel;->removeMessages(I)V
  
+     .line 390
      invoke-virtual {p0, v1, p1, p2}, Landroid/view/VolumePanel;->obtainMessage(III)Landroid/os/Message;
  
      move-result-object v0
