*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/WindowManagerImpl$CompatModeWrapper.smali	2012-03-23 19:37:26.417256800 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/WindowManagerImpl$CompatModeWrapper.smali.original	2012-03-23 19:34:45.960330873 -0500
***************
*** 32,39 ****
--- 32,41 ----
      .parameter "ci"
  
      .prologue
+     .line 119
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 120
      instance-of v0, p1, Landroid/view/WindowManagerImpl$CompatModeWrapper;
  
      if-eqz v0, :cond_0
***************
*** 46,53 ****
--- 48,57 ----
      :goto_0
      iput-object p1, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mWindowManager:Landroid/view/WindowManagerImpl;
  
+     .line 127
      if-nez p2, :cond_1
  
+     .line 128
      iget-object v0, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mWindowManager:Landroid/view/WindowManagerImpl;
  
      invoke-virtual {v0}, Landroid/view/WindowManagerImpl;->getDefaultDisplay()Landroid/view/Display;
***************
*** 56,72 ****
--- 60,80 ----
  
      iput-object v0, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mDefaultDisplay:Landroid/view/Display;
  
+     .line 135
      :goto_1
      iput-object p2, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mCompatibilityInfo:Landroid/view/CompatibilityInfoHolder;
  
+     .line 136
      return-void
  
+     .line 120
      .restart local p1
      :cond_0
      check-cast p1, Landroid/view/WindowManagerImpl;
  
      goto :goto_0
  
+     .line 131
      .end local p1
      :cond_1
      iget-object v0, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mWindowManager:Landroid/view/WindowManagerImpl;
***************
*** 96,107 ****
--- 104,117 ----
      .parameter "params"
  
      .prologue
+     .line 140
      iget-object v0, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mWindowManager:Landroid/view/WindowManagerImpl;
  
      iget-object v1, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mCompatibilityInfo:Landroid/view/CompatibilityInfoHolder;
  
      invoke-virtual {v0, p1, p2, v1}, Landroid/view/WindowManagerImpl;->addView(Landroid/view/View;Landroid/view/ViewGroup$LayoutParams;Landroid/view/CompatibilityInfoHolder;)V
  
+     .line 141
      return-void
  .end method
  
***************
*** 109,114 ****
--- 119,125 ----
      .locals 1
  
      .prologue
+     .line 156
      iget-object v0, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mDefaultDisplay:Landroid/view/Display;
  
      return-object v0
***************
*** 118,123 ****
--- 129,135 ----
      .locals 1
  
      .prologue
+     .line 166
      iget-object v0, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mWindowManager:Landroid/view/WindowManagerImpl;
  
      invoke-virtual {v0}, Landroid/view/WindowManagerImpl;->isHardwareAccelerated()Z
***************
*** 132,141 ****
--- 144,155 ----
      .parameter "view"
  
      .prologue
+     .line 151
      iget-object v0, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mWindowManager:Landroid/view/WindowManagerImpl;
  
      invoke-virtual {v0, p1}, Landroid/view/WindowManagerImpl;->removeView(Landroid/view/View;)V
  
+     .line 152
      return-void
  .end method
  
***************
*** 144,153 ****
--- 158,169 ----
      .parameter "view"
  
      .prologue
+     .line 161
      iget-object v0, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mWindowManager:Landroid/view/WindowManagerImpl;
  
      invoke-virtual {v0, p1}, Landroid/view/WindowManagerImpl;->removeViewImmediate(Landroid/view/View;)V
  
+     .line 162
      return-void
  .end method
  
***************
*** 157,165 ****
--- 173,183 ----
      .parameter "params"
  
      .prologue
+     .line 145
      iget-object v0, p0, Landroid/view/WindowManagerImpl$CompatModeWrapper;->mWindowManager:Landroid/view/WindowManagerImpl;
  
      invoke-virtual {v0, p1, p2}, Landroid/view/WindowManagerImpl;->updateViewLayout(Landroid/view/View;Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 147
      return-void
  .end method
