*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/Window.smali	2012-03-23 19:37:19.100123373 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/Window.smali.original	2012-03-23 19:34:45.635324946 -0500
***************
*** 110,149 ****
--- 110,164 ----
  
      const/4 v1, 0x0
  
+     .line 373
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 121
      iput-boolean v1, p0, Landroid/view/Window;->mIsActive:Z
  
+     .line 122
      iput-boolean v1, p0, Landroid/view/Window;->mHasChildren:Z
  
+     .line 123
      iput-boolean v1, p0, Landroid/view/Window;->mCloseOnTouchOutside:Z
  
+     .line 124
      iput-boolean v1, p0, Landroid/view/Window;->mSetCloseOnTouchOutside:Z
  
+     .line 125
      iput v1, p0, Landroid/view/Window;->mForcedWindowFlags:I
  
+     .line 127
      iput v0, p0, Landroid/view/Window;->mFeatures:I
  
+     .line 128
      iput v0, p0, Landroid/view/Window;->mLocalFeatures:I
  
+     .line 130
      iput-boolean v1, p0, Landroid/view/Window;->mHaveWindowFormat:Z
  
+     .line 131
      iput-boolean v1, p0, Landroid/view/Window;->mHaveDimAmount:Z
  
+     .line 132
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/view/Window;->mDefaultWindowFormat:I
  
+     .line 134
      iput-boolean v1, p0, Landroid/view/Window;->mHasSoftInputMode:Z
  
+     .line 139
      new-instance v0, Landroid/view/WindowManager$LayoutParams;
  
      invoke-direct {v0}, Landroid/view/WindowManager$LayoutParams;-><init>()V
  
      iput-object v0, p0, Landroid/view/Window;->mWindowAttributes:Landroid/view/WindowManager$LayoutParams;
  
+     .line 374
      iput-object p1, p0, Landroid/view/Window;->mContext:Landroid/content/Context;
  
+     .line 375
      return-void
  .end method
  
***************
*** 152,157 ****
--- 167,173 ----
      .parameter "x0"
  
      .prologue
+     .line 45
      iget-object v0, p0, Landroid/view/Window;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 162,167 ****
--- 178,184 ----
      .parameter "x0"
  
      .prologue
+     .line 45
      iget-object v0, p0, Landroid/view/Window;->mAppName:Ljava/lang/String;
  
      return-object v0
***************
*** 172,177 ****
--- 189,195 ----
      .parameter "x0"
  
      .prologue
+     .line 45
      iget-object v0, p0, Landroid/view/Window;->mContainer:Landroid/view/Window;
  
      return-object v0
***************
*** 182,187 ****
--- 200,206 ----
      .parameter "x0"
  
      .prologue
+     .line 45
      iget-object v0, p0, Landroid/view/Window;->mAppToken:Landroid/os/IBinder;
  
      return-object v0
***************
*** 192,203 ****
--- 211,224 ----
      .parameter "context"
  
      .prologue
+     .line 471
      invoke-virtual {p0}, Landroid/content/Context;->getApplicationContext()Landroid/content/Context;
  
      move-result-object v0
  
      check-cast v0, Landroid/app/Application;
  
+     .line 472
      .local v0, app:Landroid/app/Application;
      if-eqz v0, :cond_0
  
***************
*** 222,233 ****
--- 243,256 ----
      .parameter "event"
  
      .prologue
+     .line 841
      invoke-virtual {p2}, Landroid/view/MotionEvent;->getX()F
  
      move-result v4
  
      float-to-int v2, v4
  
+     .line 842
      .local v2, x:I
      invoke-virtual {p2}, Landroid/view/MotionEvent;->getY()F
  
***************
*** 235,240 ****
--- 258,264 ----
  
      float-to-int v3, v4
  
+     .line 843
      .local v3, y:I
      invoke-static {p1}, Landroid/view/ViewConfiguration;->get(Landroid/content/Context;)Landroid/view/ViewConfiguration;
  
***************
*** 244,254 ****
--- 268,280 ----
  
      move-result v1
  
+     .line 844
      .local v1, slop:I
      invoke-virtual {p0}, Landroid/view/Window;->getDecorView()Landroid/view/View;
  
      move-result-object v0
  
+     .line 845
      .local v0, decorView:Landroid/view/View;
      neg-int v4, v1
  
***************
*** 296,303 ****
--- 322,331 ----
      .parameter "flags"
  
      .prologue
+     .line 713
      invoke-virtual {p0, p1, p1}, Landroid/view/Window;->setFlags(II)V
  
+     .line 714
      return-void
  .end method
  
***************
*** 309,318 ****
--- 337,348 ----
      .parameter "flags"
  
      .prologue
+     .line 723
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0, p1}, Landroid/view/Window;->setFlags(II)V
  
+     .line 724
      return-void
  .end method
  
***************
*** 326,335 ****
--- 356,367 ----
      .locals 1
  
      .prologue
+     .line 434
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/view/Window;->mDestroyed:Z
  
+     .line 435
      return-void
  .end method
  
***************
*** 338,343 ****
--- 370,376 ----
      .parameter "id"
  
      .prologue
+     .line 901
      invoke-virtual {p0}, Landroid/view/Window;->getDecorView()Landroid/view/View;
  
      move-result-object v0
***************
*** 353,358 ****
--- 386,392 ----
      .locals 1
  
      .prologue
+     .line 796
      iget-object v0, p0, Landroid/view/Window;->mWindowAttributes:Landroid/view/WindowManager$LayoutParams;
  
      return-object v0
***************
*** 362,367 ****
--- 396,402 ----
      .locals 1
  
      .prologue
+     .line 565
      iget-object v0, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      return-object v0
***************
*** 371,376 ****
--- 406,412 ----
      .locals 1
  
      .prologue
+     .line 425
      iget-object v0, p0, Landroid/view/Window;->mContainer:Landroid/view/Window;
  
      return-object v0
***************
*** 380,385 ****
--- 416,422 ----
      .locals 1
  
      .prologue
+     .line 384
      iget-object v0, p0, Landroid/view/Window;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 395,400 ****
--- 432,438 ----
      .locals 1
  
      .prologue
+     .line 1175
      iget v0, p0, Landroid/view/Window;->mFeatures:I
  
      return v0
***************
*** 404,409 ****
--- 442,448 ----
      .locals 1
  
      .prologue
+     .line 804
      iget v0, p0, Landroid/view/Window;->mForcedWindowFlags:I
  
      return v0
***************
*** 416,421 ****
--- 455,461 ----
      .locals 1
  
      .prologue
+     .line 1197
      iget v0, p0, Landroid/view/Window;->mLocalFeatures:I
  
      return v0
***************
*** 428,433 ****
--- 468,474 ----
      .locals 1
  
      .prologue
+     .line 548
      iget-object v0, p0, Landroid/view/Window;->mWindowManager:Landroid/view/WindowManager;
  
      return-object v0
***************
*** 437,449 ****
--- 478,493 ----
      .locals 2
  
      .prologue
+     .line 392
      monitor-enter p0
  
+     .line 393
      :try_start_0
      iget-object v0, p0, Landroid/view/Window;->mWindowStyle:Landroid/content/res/TypedArray;
  
      if-nez v0, :cond_0
  
+     .line 394
      iget-object v0, p0, Landroid/view/Window;->mContext:Landroid/content/Context;
  
      sget-object v1, Lcom/android/internal/R$styleable;->Window:[I
***************
*** 454,459 ****
--- 498,504 ----
  
      iput-object v0, p0, Landroid/view/Window;->mWindowStyle:Landroid/content/res/TypedArray;
  
+     .line 397
      :cond_0
      iget-object v0, p0, Landroid/view/Window;->mWindowStyle:Landroid/content/res/TypedArray;
  
***************
*** 461,466 ****
--- 506,512 ----
  
      return-object v0
  
+     .line 398
      :catchall_0
      move-exception v0
  
***************
*** 475,480 ****
--- 521,527 ----
      .locals 1
  
      .prologue
+     .line 429
      iget-boolean v0, p0, Landroid/view/Window;->mHasChildren:Z
  
      return v0
***************
*** 487,492 ****
--- 534,540 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 1185
      invoke-virtual {p0}, Landroid/view/Window;->getFeatures()I
  
      move-result v1
***************
*** 510,515 ****
--- 558,564 ----
      .locals 1
  
      .prologue
+     .line 811
      iget-boolean v0, p0, Landroid/view/Window;->mHasSoftInputMode:Z
  
      return v0
***************
*** 519,524 ****
--- 568,574 ----
      .locals 1
  
      .prologue
+     .line 1223
      iget-boolean v0, p0, Landroid/view/Window;->mHaveDimAmount:Z
  
      return v0
***************
*** 531,536 ****
--- 581,587 ----
      .locals 1
  
      .prologue
+     .line 889
      iget-boolean v0, p0, Landroid/view/Window;->mIsActive:Z
  
      return v0
***************
*** 540,545 ****
--- 591,597 ----
      .locals 1
  
      .prologue
+     .line 439
      iget-boolean v0, p0, Landroid/view/Window;->mDestroyed:Z
  
      return v0
***************
*** 555,570 ****
--- 607,625 ----
      .locals 2
  
      .prologue
+     .line 877
      iget-object v0, p0, Landroid/view/Window;->mContainer:Landroid/view/Window;
  
      if-eqz v0, :cond_1
  
+     .line 878
      iget-object v0, p0, Landroid/view/Window;->mContainer:Landroid/view/Window;
  
      iget-object v0, v0, Landroid/view/Window;->mActiveChild:Landroid/view/Window;
  
      if-eqz v0, :cond_0
  
+     .line 879
      iget-object v0, p0, Landroid/view/Window;->mContainer:Landroid/view/Window;
  
      iget-object v0, v0, Landroid/view/Window;->mActiveChild:Landroid/view/Window;
***************
*** 573,590 ****
--- 628,649 ----
  
      iput-boolean v1, v0, Landroid/view/Window;->mIsActive:Z
  
+     .line 881
      :cond_0
      iget-object v0, p0, Landroid/view/Window;->mContainer:Landroid/view/Window;
  
      iput-object p0, v0, Landroid/view/Window;->mActiveChild:Landroid/view/Window;
  
+     .line 883
      :cond_1
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/view/Window;->mIsActive:Z
  
+     .line 884
      invoke-virtual {p0}, Landroid/view/Window;->onActive()V
  
+     .line 885
      return-void
  .end method
  
***************
*** 614,623 ****
--- 673,684 ----
      .parameter "featureId"
  
      .prologue
+     .line 871
      const/4 v1, 0x1
  
      shl-int v0, v1, p1
  
+     .line 872
      .local v0, flag:I
      iget v1, p0, Landroid/view/Window;->mFeatures:I
  
***************
*** 627,632 ****
--- 688,694 ----
  
      iput v1, p0, Landroid/view/Window;->mFeatures:I
  
+     .line 873
      iget v1, p0, Landroid/view/Window;->mLocalFeatures:I
  
      iget-object v2, p0, Landroid/view/Window;->mContainer:Landroid/view/Window;
***************
*** 649,654 ****
--- 711,717 ----
  
      iput v1, p0, Landroid/view/Window;->mLocalFeatures:I
  
+     .line 874
      return-void
  .end method
  
***************
*** 659,666 ****
--- 722,731 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 861
      shl-int v0, v2, p1
  
+     .line 862
      .local v0, flag:I
      iget v1, p0, Landroid/view/Window;->mFeatures:I
  
***************
*** 668,673 ****
--- 733,739 ----
  
      iput v1, p0, Landroid/view/Window;->mFeatures:I
  
+     .line 863
      iget v3, p0, Landroid/view/Window;->mLocalFeatures:I
  
      iget-object v1, p0, Landroid/view/Window;->mContainer:Landroid/view/Window;
***************
*** 687,692 ****
--- 753,759 ----
  
      iput v1, p0, Landroid/view/Window;->mLocalFeatures:I
  
+     .line 864
      iget v1, p0, Landroid/view/Window;->mFeatures:I
  
      and-int/2addr v1, v0
***************
*** 701,708 ****
--- 768,777 ----
      :cond_0
      move v1, v0
  
+     .line 863
      goto :goto_0
  
+     .line 864
      :cond_1
      const/4 v1, 0x0
  
***************
*** 720,739 ****
--- 789,812 ----
      .parameter "a"
  
      .prologue
+     .line 783
      iget-object v0, p0, Landroid/view/Window;->mWindowAttributes:Landroid/view/WindowManager$LayoutParams;
  
      invoke-virtual {v0, p1}, Landroid/view/WindowManager$LayoutParams;->copyFrom(Landroid/view/WindowManager$LayoutParams;)I
  
+     .line 784
      iget-object v0, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v0, :cond_0
  
+     .line 785
      iget-object v0, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      iget-object v1, p0, Landroid/view/Window;->mWindowAttributes:Landroid/view/WindowManager$LayoutParams;
  
      invoke-interface {v0, v1}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 787
      :cond_0
      return-void
  .end method
***************
*** 746,751 ****
--- 819,825 ----
      .parameter "resid"
  
      .prologue
+     .line 1012
      iget-object v0, p0, Landroid/view/Window;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 758,763 ****
--- 832,838 ----
  
      invoke-virtual {p0, v0}, Landroid/view/Window;->setBackgroundDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 1013
      return-void
  .end method
  
***************
*** 766,773 ****
--- 841,850 ----
      .parameter "callback"
  
      .prologue
+     .line 558
      iput-object p1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
+     .line 559
      return-void
  .end method
  
***************
*** 782,793 ****
--- 859,873 ----
      .parameter "close"
  
      .prologue
+     .line 816
      iput-boolean p1, p0, Landroid/view/Window;->mCloseOnTouchOutside:Z
  
+     .line 817
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/view/Window;->mSetCloseOnTouchOutside:Z
  
+     .line 818
      return-void
  .end method
  
***************
*** 796,811 ****
--- 876,895 ----
      .parameter "close"
  
      .prologue
+     .line 822
      iget-boolean v0, p0, Landroid/view/Window;->mSetCloseOnTouchOutside:Z
  
      if-nez v0, :cond_0
  
+     .line 823
      iput-boolean p1, p0, Landroid/view/Window;->mCloseOnTouchOutside:Z
  
+     .line 824
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/view/Window;->mSetCloseOnTouchOutside:Z
  
+     .line 826
      :cond_0
      return-void
  .end method
***************
*** 815,840 ****
--- 899,930 ----
      .parameter "container"
  
      .prologue
+     .line 409
      iput-object p1, p0, Landroid/view/Window;->mContainer:Landroid/view/Window;
  
+     .line 410
      if-eqz p1, :cond_0
  
+     .line 412
      iget v0, p0, Landroid/view/Window;->mFeatures:I
  
      or-int/lit8 v0, v0, 0x2
  
      iput v0, p0, Landroid/view/Window;->mFeatures:I
  
+     .line 413
      iget v0, p0, Landroid/view/Window;->mLocalFeatures:I
  
      or-int/lit8 v0, v0, 0x2
  
      iput v0, p0, Landroid/view/Window;->mLocalFeatures:I
  
+     .line 414
      const/4 v0, 0x1
  
      iput-boolean v0, p1, Landroid/view/Window;->mHasChildren:Z
  
+     .line 416
      :cond_0
      return-void
  .end method
***************
*** 853,879 ****
--- 943,976 ----
      .parameter "format"
  
      .prologue
+     .line 1211
      iput p1, p0, Landroid/view/Window;->mDefaultWindowFormat:I
  
+     .line 1212
      iget-boolean v1, p0, Landroid/view/Window;->mHaveWindowFormat:Z
  
      if-nez v1, :cond_0
  
+     .line 1213
      invoke-virtual {p0}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v0
  
+     .line 1214
      .local v0, attrs:Landroid/view/WindowManager$LayoutParams;
      iput p1, v0, Landroid/view/WindowManager$LayoutParams;->format:I
  
+     .line 1215
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v1, :cond_0
  
+     .line 1216
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      invoke-interface {v1, v0}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 1219
      .end local v0           #attrs:Landroid/view/WindowManager$LayoutParams;
      :cond_0
      return-void
***************
*** 884,908 ****
--- 981,1011 ----
      .parameter "amount"
  
      .prologue
+     .line 764
      invoke-virtual {p0}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v0
  
+     .line 765
      .local v0, attrs:Landroid/view/WindowManager$LayoutParams;
      iput p1, v0, Landroid/view/WindowManager$LayoutParams;->dimAmount:F
  
+     .line 766
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/view/Window;->mHaveDimAmount:Z
  
+     .line 767
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v1, :cond_0
  
+     .line 768
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      invoke-interface {v1, v0}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 770
      :cond_0
      return-void
  .end method
***************
*** 928,937 ****
--- 1031,1042 ----
      .parameter "mask"
  
      .prologue
+     .line 744
      invoke-virtual {p0}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v0
  
+     .line 745
      .local v0, attrs:Landroid/view/WindowManager$LayoutParams;
      iget v1, v0, Landroid/view/WindowManager$LayoutParams;->flags:I
  
***************
*** 945,962 ****
--- 1050,1070 ----
  
      iput v1, v0, Landroid/view/WindowManager$LayoutParams;->flags:I
  
+     .line 746
      const/high16 v1, 0x800
  
      and-int/2addr v1, p2
  
      if-eqz v1, :cond_0
  
+     .line 747
      iget v1, v0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
      or-int/lit8 v1, v1, 0x8
  
      iput v1, v0, Landroid/view/WindowManager$LayoutParams;->privateFlags:I
  
+     .line 749
      :cond_0
      iget v1, p0, Landroid/view/Window;->mForcedWindowFlags:I
  
***************
*** 964,977 ****
--- 1072,1088 ----
  
      iput v1, p0, Landroid/view/Window;->mForcedWindowFlags:I
  
+     .line 750
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v1, :cond_1
  
+     .line 751
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      invoke-interface {v1, v0}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 753
      :cond_1
      return-void
  .end method
***************
*** 981,1016 ****
--- 1092,1136 ----
      .parameter "format"
  
      .prologue
+     .line 658
      invoke-virtual {p0}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v0
  
+     .line 659
      .local v0, attrs:Landroid/view/WindowManager$LayoutParams;
      if-eqz p1, :cond_1
  
+     .line 660
      iput p1, v0, Landroid/view/WindowManager$LayoutParams;->format:I
  
+     .line 661
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/view/Window;->mHaveWindowFormat:Z
  
+     .line 666
      :goto_0
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v1, :cond_0
  
+     .line 667
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      invoke-interface {v1, v0}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 669
      :cond_0
      return-void
  
+     .line 663
      :cond_1
      iget v1, p0, Landroid/view/Window;->mDefaultWindowFormat:I
  
      iput v1, v0, Landroid/view/WindowManager$LayoutParams;->format:I
  
+     .line 664
      const/4 v1, 0x0
  
      iput-boolean v1, p0, Landroid/view/Window;->mHaveWindowFormat:Z
***************
*** 1023,1043 ****
--- 1143,1168 ----
      .parameter "gravity"
  
      .prologue
+     .line 625
      invoke-virtual {p0}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v0
  
+     .line 626
      .local v0, attrs:Landroid/view/WindowManager$LayoutParams;
      iput p1, v0, Landroid/view/WindowManager$LayoutParams;->gravity:I
  
+     .line 627
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v1, :cond_0
  
+     .line 628
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      invoke-interface {v1, v0}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 630
      :cond_0
      return-void
  .end method
***************
*** 1048,1070 ****
--- 1173,1201 ----
      .parameter "height"
  
      .prologue
+     .line 605
      invoke-virtual {p0}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v0
  
+     .line 606
      .local v0, attrs:Landroid/view/WindowManager$LayoutParams;
      iput p1, v0, Landroid/view/WindowManager$LayoutParams;->width:I
  
+     .line 607
      iput p2, v0, Landroid/view/WindowManager$LayoutParams;->height:I
  
+     .line 608
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v1, :cond_0
  
+     .line 609
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      invoke-interface {v1, v0}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 611
      :cond_0
      return-void
  .end method
***************
*** 1074,1104 ****
--- 1205,1243 ----
      .parameter "mode"
  
      .prologue
+     .line 694
      invoke-virtual {p0}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v0
  
+     .line 695
      .local v0, attrs:Landroid/view/WindowManager$LayoutParams;
      if-eqz p1, :cond_1
  
+     .line 696
      iput p1, v0, Landroid/view/WindowManager$LayoutParams;->softInputMode:I
  
+     .line 697
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/view/Window;->mHasSoftInputMode:Z
  
+     .line 701
      :goto_0
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v1, :cond_0
  
+     .line 702
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      invoke-interface {v1, v0}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 704
      :cond_0
      return-void
  
+     .line 699
      :cond_1
      const/4 v1, 0x0
  
***************
*** 1118,1138 ****
--- 1257,1282 ----
      .parameter "type"
  
      .prologue
+     .line 639
      invoke-virtual {p0}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v0
  
+     .line 640
      .local v0, attrs:Landroid/view/WindowManager$LayoutParams;
      iput p1, v0, Landroid/view/WindowManager$LayoutParams;->type:I
  
+     .line 641
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v1, :cond_0
  
+     .line 642
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      invoke-interface {v1, v0}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 644
      :cond_0
      return-void
  .end method
***************
*** 1142,1147 ****
--- 1286,1292 ----
      .parameter "uiOptions"
  
      .prologue
+     .line 1251
      return-void
  .end method
  
***************
*** 1151,1156 ****
--- 1296,1302 ----
      .parameter "mask"
  
      .prologue
+     .line 1259
      return-void
  .end method
  
***************
*** 1162,1182 ****
--- 1308,1333 ----
      .parameter "resId"
  
      .prologue
+     .line 679
      invoke-virtual {p0}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v0
  
+     .line 680
      .local v0, attrs:Landroid/view/WindowManager$LayoutParams;
      iput p1, v0, Landroid/view/WindowManager$LayoutParams;->windowAnimations:I
  
+     .line 681
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      if-eqz v1, :cond_0
  
+     .line 682
      iget-object v1, p0, Landroid/view/Window;->mCallback:Landroid/view/Window$Callback;
  
      invoke-interface {v1, v0}, Landroid/view/Window$Callback;->onWindowAttributesChanged(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 684
      :cond_0
      return-void
  .end method
***************
*** 1188,1197 ****
--- 1339,1350 ----
      .parameter "appName"
  
      .prologue
+     .line 450
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, p2, p3, v0}, Landroid/view/Window;->setWindowManager(Landroid/view/WindowManager;Landroid/os/IBinder;Ljava/lang/String;Z)V
  
+     .line 451
      return-void
  .end method
  
***************
*** 1203,1218 ****
--- 1356,1376 ----
      .parameter "hardwareAccelerated"
  
      .prologue
+     .line 462
      iput-object p2, p0, Landroid/view/Window;->mAppToken:Landroid/os/IBinder;
  
+     .line 463
      iput-object p3, p0, Landroid/view/Window;->mAppName:Ljava/lang/String;
  
+     .line 464
      if-nez p1, :cond_0
  
+     .line 465
      invoke-static {}, Landroid/view/WindowManagerImpl;->getDefault()Landroid/view/WindowManagerImpl;
  
      move-result-object p1
  
+     .line 467
      :cond_0
      new-instance v0, Landroid/view/Window$LocalWindowManager;
  
***************
*** 1220,1225 ****
--- 1378,1384 ----
  
      iput-object v0, p0, Landroid/view/Window;->mWindowManager:Landroid/view/WindowManager;
  
+     .line 468
      return-void
  .end method
  
***************
*** 1229,1234 ****
--- 1388,1394 ----
      .parameter "event"
  
      .prologue
+     .line 833
      iget-boolean v0, p0, Landroid/view/Window;->mCloseOnTouchOutside:Z
  
      if-eqz v0, :cond_0
***************
*** 1251,1258 ****
--- 1411,1420 ----
  
      if-eqz v0, :cond_0
  
+     .line 835
      const/4 v0, 0x1
  
+     .line 837
      :goto_0
      return v0
  
