*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/app/IMediaContainerService$Stub$Proxy.smali	2012-03-23 19:37:48.424658104 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/app/IMediaContainerService$Stub$Proxy.smali.original	2012-03-23 19:34:46.343337856 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 178
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 179
      iput-object p1, p0, Lcom/android/internal/app/IMediaContainerService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 180
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 183
      iget-object v0, p0, Lcom/android/internal/app/IMediaContainerService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 55,77 ****
--- 59,86 ----
      .end annotation
  
      .prologue
+     .line 352
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 353
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 356
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.app.IMediaContainerService"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 357
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 358
      iget-object v4, p0, Lcom/android/internal/app/IMediaContainerService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v5, 0x7
***************
*** 80,106 ****
--- 89,122 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 359
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 360
      invoke-virtual {v1}, Landroid/os/Parcel;->readLong()J
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result-wide v2
  
+     .line 363
      .local v2, _result:J
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 364
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 366
      return-wide v2
  
+     .line 363
      .end local v2           #_result:J
      :catchall_0
      move-exception v4
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 364
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v4
***************
*** 120,150 ****
--- 136,173 ----
  
      const/4 v3, 0x0
  
+     .line 305
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 306
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 309
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.app.IMediaContainerService"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 310
      if-eqz p1, :cond_0
  
+     .line 311
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 312
      const/4 v4, 0x0
  
      invoke-virtual {p1, v0, v4}, Landroid/net/Uri;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 317
      :goto_0
      iget-object v4, p0, Lcom/android/internal/app/IMediaContainerService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 154,161 ****
--- 177,186 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 318
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 319
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 164,177 ****
--- 189,206 ----
  
      if-eqz v4, :cond_1
  
+     .line 322
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 323
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 325
      return v2
  
+     .line 315
      .end local v2           #_result:Z
      :cond_0
      const/4 v4, 0x0
***************
*** 183,193 ****
--- 212,224 ----
  
      goto :goto_0
  
+     .line 322
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 323
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 195,200 ****
--- 226,232 ----
      :cond_1
      move v2, v3
  
+     .line 319
      goto :goto_1
  .end method
  
***************
*** 213,246 ****
--- 245,286 ----
  
      const/4 v3, 0x0
  
+     .line 280
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 281
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 284
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.app.IMediaContainerService"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 285
      if-eqz p1, :cond_0
  
+     .line 286
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 287
      const/4 v4, 0x0
  
      invoke-virtual {p1, v0, v4}, Landroid/net/Uri;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 292
      :goto_0
      invoke-virtual {v0, p2, p3}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 293
      iget-object v4, p0, Lcom/android/internal/app/IMediaContainerService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v5, 0x4
***************
*** 249,256 ****
--- 289,298 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 294
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 295
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 259,272 ****
--- 301,318 ----
  
      if-eqz v4, :cond_1
  
+     .line 298
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 299
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 301
      return v2
  
+     .line 290
      .end local v2           #_result:Z
      :cond_0
      const/4 v4, 0x0
***************
*** 278,288 ****
--- 324,336 ----
  
      goto :goto_0
  
+     .line 298
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 299
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 290,295 ****
--- 338,344 ----
      :cond_1
      move v2, v3
  
+     .line 295
      goto :goto_1
  .end method
  
***************
*** 304,345 ****
--- 353,404 ----
      .end annotation
  
      .prologue
+     .line 218
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 219
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 222
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.app.IMediaContainerService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 223
      if-eqz p1, :cond_0
  
+     .line 224
      const/4 v3, 0x1
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 225
      const/4 v3, 0x0
  
      invoke-virtual {p1, v0, v3}, Landroid/net/Uri;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 230
      :goto_0
      if-eqz p2, :cond_1
  
+     .line 231
      const/4 v3, 0x1
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 232
      const/4 v3, 0x0
  
      invoke-virtual {p2, v0, v3}, Landroid/os/ParcelFileDescriptor;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 237
      :goto_1
      iget-object v3, p0, Lcom/android/internal/app/IMediaContainerService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 349,369 ****
--- 408,434 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 238
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 239
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 242
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 243
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 245
      return v2
  
+     .line 228
      .end local v2           #_result:I
      :cond_0
      const/4 v3, 0x0
***************
*** 375,389 ****
--- 440,457 ----
  
      goto :goto_0
  
+     .line 242
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 243
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
  
+     .line 235
      :cond_1
      const/4 v3, 0x0
  
***************
*** 408,445 ****
--- 476,523 ----
      .end annotation
  
      .prologue
+     .line 191
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 192
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 195
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.app.IMediaContainerService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 196
      if-eqz p1, :cond_0
  
+     .line 197
      const/4 v3, 0x1
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 198
      const/4 v3, 0x0
  
      invoke-virtual {p1, v0, v3}, Landroid/net/Uri;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 203
      :goto_0
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 204
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 205
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 206
      iget-object v3, p0, Lcom/android/internal/app/IMediaContainerService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x1
***************
*** 448,468 ****
--- 526,552 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 207
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 208
      invoke-virtual {v1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result-object v2
  
+     .line 211
      .local v2, _result:Ljava/lang/String;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 212
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 214
      return-object v2
  
+     .line 201
      .end local v2           #_result:Ljava/lang/String;
      :cond_0
      const/4 v3, 0x0
***************
*** 474,484 ****
--- 558,570 ----
  
      goto :goto_0
  
+     .line 211
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 212
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 488,493 ****
--- 574,580 ----
      .locals 1
  
      .prologue
+     .line 187
      const-string v0, "com.android.internal.app.IMediaContainerService"
  
      return-object v0
***************
*** 505,540 ****
--- 592,636 ----
      .end annotation
  
      .prologue
+     .line 249
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 250
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 253
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.app.IMediaContainerService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 254
      if-eqz p1, :cond_0
  
+     .line 255
      const/4 v3, 0x1
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 256
      const/4 v3, 0x0
  
      invoke-virtual {p1, v0, v3}, Landroid/net/Uri;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 261
      :goto_0
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 262
      invoke-virtual {v0, p3, p4}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 263
      iget-object v3, p0, Lcom/android/internal/app/IMediaContainerService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x3
***************
*** 543,556 ****
--- 639,655 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 264
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 265
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_1
  
+     .line 266
      sget-object v3, Landroid/content/pm/PackageInfoLite;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 561,574 ****
--- 660,677 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 273
      .local v2, _result:Landroid/content/pm/PackageInfoLite;
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 274
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 276
      return-object v2
  
+     .line 259
      .end local v2           #_result:Landroid/content/pm/PackageInfoLite;
      :cond_0
      const/4 v3, 0x0
***************
*** 580,594 ****
--- 683,700 ----
  
      goto :goto_0
  
+     .line 273
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 274
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
  
+     .line 269
      :cond_1
      const/4 v2, 0x0
  
***************
*** 606,628 ****
--- 712,739 ----
      .end annotation
  
      .prologue
+     .line 329
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 330
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 333
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.app.IMediaContainerService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 334
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 335
      iget-object v3, p0, Lcom/android/internal/app/IMediaContainerService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x6
***************
*** 631,644 ****
--- 742,758 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 336
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 337
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 338
      sget-object v3, Landroid/content/res/ObbInfo;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 649,662 ****
--- 763,780 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 345
      .local v2, _result:Landroid/content/res/ObbInfo;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 346
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 348
      return-object v2
  
+     .line 341
      .end local v2           #_result:Landroid/content/res/ObbInfo;
      :cond_0
      const/4 v2, 0x0
***************
*** 664,675 ****
--- 782,795 ----
      .restart local v2       #_result:Landroid/content/res/ObbInfo;
      goto :goto_0
  
+     .line 345
      .end local v2           #_result:Landroid/content/res/ObbInfo;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 346
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
