*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/app/NetInitiatedActivity$1.smali	2012-03-23 19:37:48.763664286 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/app/NetInitiatedActivity$1.smali.original	2012-03-23 19:34:46.346337912 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 57
      iput-object p1, p0, Lcom/android/internal/app/NetInitiatedActivity$1;->this$0:Lcom/android/internal/app/NetInitiatedActivity;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 39,44 ****
--- 40,46 ----
      .parameter "intent"
  
      .prologue
+     .line 60
      const-string v0, "NetInitiatedActivity"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 65,70 ****
--- 67,73 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 61
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
      move-result-object v0
***************
*** 73,83 ****
--- 76,88 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 62
      iget-object v0, p0, Lcom/android/internal/app/NetInitiatedActivity$1;->this$0:Lcom/android/internal/app/NetInitiatedActivity;
  
      #calls: Lcom/android/internal/app/NetInitiatedActivity;->handleNIVerify(Landroid/content/Intent;)V
      invoke-static {v0, p2}, Lcom/android/internal/app/NetInitiatedActivity;->access$000(Lcom/android/internal/app/NetInitiatedActivity;Landroid/content/Intent;)V
  
+     .line 64
      :cond_0
      return-void
  .end method
