*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/app/ResolverActivity$ResolveListAdapter.smali	2012-03-23 19:37:49.761682485 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/app/ResolverActivity$ResolveListAdapter.smali.original	2012-03-23 19:34:46.356338093 -0500
***************
*** 56,61 ****
--- 56,62 ----
      .end annotation
  
      .prologue
+     .line 234
      .local p5, rList:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
      move-object/from16 v0, p1
  
***************
*** 65,70 ****
--- 66,72 ----
  
      invoke-direct/range {p0 .. p0}, Landroid/widget/BaseAdapter;-><init>()V
  
+     .line 235
      new-instance v2, Landroid/content/Intent;
  
      move-object/from16 v0, p3
***************
*** 75,80 ****
--- 77,83 ----
  
      iput-object v2, v0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mIntent:Landroid/content/Intent;
  
+     .line 236
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mIntent:Landroid/content/Intent;
***************
*** 83,88 ****
--- 86,92 ----
  
      invoke-virtual {v2, v3}, Landroid/content/Intent;->setComponent(Landroid/content/ComponentName;)Landroid/content/Intent;
  
+     .line 237
      const-string v2, "layout_inflater"
  
      move-object/from16 v0, p2
***************
*** 97,104 ****
--- 101,110 ----
  
      iput-object v2, v0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mInflater:Landroid/view/LayoutInflater;
  
+     .line 239
      if-nez p5, :cond_0
  
+     .line 240
      #getter for: Lcom/android/internal/app/ResolverActivity;->mPm:Landroid/content/pm/PackageManager;
      invoke-static/range {p1 .. p1}, Lcom/android/internal/app/ResolverActivity;->access$200(Lcom/android/internal/app/ResolverActivity;)Landroid/content/pm/PackageManager;
  
***************
*** 124,129 ****
--- 130,136 ----
  
      move-result-object p5
  
+     .line 245
      :cond_0
      if-eqz p5, :cond_e
  
***************
*** 134,139 ****
--- 141,147 ----
      .local v14, N:I
      if-lez v14, :cond_e
  
+     .line 248
      const/4 v2, 0x0
  
      move-object/from16 v0, p5
***************
*** 144,149 ****
--- 152,158 ----
  
      check-cast v12, Landroid/content/pm/ResolveInfo;
  
+     .line 249
      .local v12, r0:Landroid/content/pm/ResolveInfo;
      const/16 v16, 0x1
  
***************
*** 153,158 ****
--- 162,168 ----
  
      if-ge v0, v14, :cond_4
  
+     .line 250
      move-object/from16 v0, p5
  
      move/from16 v1, v16
***************
*** 163,168 ****
--- 173,179 ----
  
      check-cast v4, Landroid/content/pm/ResolveInfo;
  
+     .line 257
      .local v4, ri:Landroid/content/pm/ResolveInfo;
      iget v2, v12, Landroid/content/pm/ResolveInfo;->priority:I
  
***************
*** 176,197 ****
--- 187,212 ----
  
      if-eq v2, v3, :cond_3
  
+     .line 259
      :cond_1
      :goto_2
      move/from16 v0, v16
  
      if-ge v0, v14, :cond_3
  
+     .line 260
      move-object/from16 v0, p5
  
      move/from16 v1, v16
  
      invoke-interface {v0, v1}, Ljava/util/List;->remove(I)Ljava/lang/Object;
  
+     .line 261
      add-int/lit8 v14, v14, -0x1
  
      goto :goto_2
  
+     .line 240
      .end local v4           #ri:Landroid/content/pm/ResolveInfo;
      .end local v12           #r0:Landroid/content/pm/ResolveInfo;
      .end local v14           #N:I
***************
*** 201,206 ****
--- 216,222 ----
  
      goto :goto_0
  
+     .line 249
      .restart local v4       #ri:Landroid/content/pm/ResolveInfo;
      .restart local v12       #r0:Landroid/content/pm/ResolveInfo;
      .restart local v14       #N:I
***************
*** 210,221 ****
--- 226,239 ----
  
      goto :goto_1
  
+     .line 265
      .end local v4           #ri:Landroid/content/pm/ResolveInfo;
      :cond_4
      const/4 v2, 0x1
  
      if-le v14, v2, :cond_5
  
+     .line 266
      new-instance v18, Landroid/content/pm/ResolveInfo$DisplayNameComparator;
  
      #getter for: Lcom/android/internal/app/ResolverActivity;->mPm:Landroid/content/pm/PackageManager;
***************
*** 227,232 ****
--- 245,251 ----
  
      invoke-direct {v0, v2}, Landroid/content/pm/ResolveInfo$DisplayNameComparator;-><init>(Landroid/content/pm/PackageManager;)V
  
+     .line 268
      .local v18, rComparator:Landroid/content/pm/ResolveInfo$DisplayNameComparator;
      move-object/from16 v0, p5
  
***************
*** 234,239 ****
--- 253,259 ----
  
      invoke-static {v0, v1}, Ljava/util/Collections;->sort(Ljava/util/List;Ljava/util/Comparator;)V
  
+     .line 271
      .end local v18           #rComparator:Landroid/content/pm/ResolveInfo$DisplayNameComparator;
      :cond_5
      new-instance v2, Ljava/util/ArrayList;
***************
*** 244,251 ****
--- 264,273 ----
  
      iput-object v2, v0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
  
+     .line 274
      if-eqz p4, :cond_9
  
+     .line 275
      const/16 v16, 0x0
  
      :goto_3
***************
*** 257,272 ****
--- 279,298 ----
  
      if-ge v0, v2, :cond_9
  
+     .line 276
      aget-object v7, p4, v16
  
+     .line 277
      .local v7, ii:Landroid/content/Intent;
      if-nez v7, :cond_6
  
+     .line 275
      :goto_4
      add-int/lit8 v16, v16, 0x1
  
      goto :goto_3
  
+     .line 280
      :cond_6
      invoke-virtual/range {p1 .. p1}, Lcom/android/internal/app/ResolverActivity;->getPackageManager()Landroid/content/pm/PackageManager;
  
***************
*** 278,286 ****
--- 304,314 ----
  
      move-result-object v15
  
+     .line 282
      .local v15, ai:Landroid/content/pm/ActivityInfo;
      if-nez v15, :cond_7
  
+     .line 283
      const-string v2, "ResolverActivity"
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 305,326 ****
--- 333,359 ----
  
      goto :goto_4
  
+     .line 287
      :cond_7
      new-instance v4, Landroid/content/pm/ResolveInfo;
  
      invoke-direct {v4}, Landroid/content/pm/ResolveInfo;-><init>()V
  
+     .line 288
      .restart local v4       #ri:Landroid/content/pm/ResolveInfo;
      iput-object v15, v4, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
+     .line 289
      instance-of v2, v7, Landroid/content/pm/LabeledIntent;
  
      if-eqz v2, :cond_8
  
      move-object/from16 v17, v7
  
+     .line 290
      check-cast v17, Landroid/content/pm/LabeledIntent;
  
+     .line 291
      .local v17, li:Landroid/content/pm/LabeledIntent;
      invoke-virtual/range {v17 .. v17}, Landroid/content/pm/LabeledIntent;->getSourcePackage()Ljava/lang/String;
  
***************
*** 328,351 ****
--- 361,388 ----
  
      iput-object v2, v4, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
  
+     .line 292
      invoke-virtual/range {v17 .. v17}, Landroid/content/pm/LabeledIntent;->getLabelResource()I
  
      move-result v2
  
      iput v2, v4, Landroid/content/pm/ResolveInfo;->labelRes:I
  
+     .line 293
      invoke-virtual/range {v17 .. v17}, Landroid/content/pm/LabeledIntent;->getNonLocalizedLabel()Ljava/lang/CharSequence;
  
      move-result-object v2
  
      iput-object v2, v4, Landroid/content/pm/ResolveInfo;->nonLocalizedLabel:Ljava/lang/CharSequence;
  
+     .line 294
      invoke-virtual/range {v17 .. v17}, Landroid/content/pm/LabeledIntent;->getIconResource()I
  
      move-result v2
  
      iput v2, v4, Landroid/content/pm/ResolveInfo;->icon:I
  
+     .line 296
      .end local v17           #li:Landroid/content/pm/LabeledIntent;
      :cond_8
      move-object/from16 v0, p0
***************
*** 372,377 ****
--- 409,415 ----
  
      goto :goto_4
  
+     .line 303
      .end local v4           #ri:Landroid/content/pm/ResolveInfo;
      .end local v7           #ii:Landroid/content/Intent;
      .end local v15           #ai:Landroid/content/pm/ActivityInfo;
***************
*** 387,395 ****
--- 425,435 ----
      .end local v12           #r0:Landroid/content/pm/ResolveInfo;
      check-cast v12, Landroid/content/pm/ResolveInfo;
  
+     .line 304
      .restart local v12       #r0:Landroid/content/pm/ResolveInfo;
      const/4 v10, 0x0
  
+     .line 305
      .local v10, start:I
      #getter for: Lcom/android/internal/app/ResolverActivity;->mPm:Landroid/content/pm/PackageManager;
      invoke-static/range {p1 .. p1}, Lcom/android/internal/app/ResolverActivity;->access$200(Lcom/android/internal/app/ResolverActivity;)Landroid/content/pm/PackageManager;
***************
*** 400,405 ****
--- 440,446 ----
  
      move-result-object v13
  
+     .line 306
      .local v13, r0Label:Ljava/lang/CharSequence;
      const/16 v16, 0x1
  
***************
*** 408,419 ****
--- 449,463 ----
  
      if-ge v0, v14, :cond_d
  
+     .line 307
      if-nez v13, :cond_a
  
+     .line 308
      iget-object v2, v12, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
      iget-object v13, v2, Landroid/content/pm/ActivityInfo;->packageName:Ljava/lang/String;
  
+     .line 310
      :cond_a
      move-object/from16 v0, p5
  
***************
*** 425,430 ****
--- 469,475 ----
  
      check-cast v4, Landroid/content/pm/ResolveInfo;
  
+     .line 311
      .restart local v4       #ri:Landroid/content/pm/ResolveInfo;
      #getter for: Lcom/android/internal/app/ResolverActivity;->mPm:Landroid/content/pm/PackageManager;
      invoke-static/range {p1 .. p1}, Lcom/android/internal/app/ResolverActivity;->access$200(Lcom/android/internal/app/ResolverActivity;)Landroid/content/pm/PackageManager;
***************
*** 435,449 ****
--- 480,497 ----
  
      move-result-object v19
  
+     .line 312
      .local v19, riLabel:Ljava/lang/CharSequence;
      if-nez v19, :cond_b
  
+     .line 313
      iget-object v2, v4, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
      iget-object v0, v2, Landroid/content/pm/ActivityInfo;->packageName:Ljava/lang/String;
  
      move-object/from16 v19, v0
  
+     .line 315
      :cond_b
      move-object/from16 v0, v19
  
***************
*** 453,463 ****
--- 501,513 ----
  
      if-eqz v2, :cond_c
  
+     .line 306
      :goto_6
      add-int/lit8 v16, v16, 0x1
  
      goto :goto_5
  
+     .line 318
      :cond_c
      add-int/lit8 v11, v16, -0x1
  
***************
*** 467,480 ****
--- 517,534 ----
  
      invoke-direct/range {v8 .. v13}, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->processGroup(Ljava/util/List;IILandroid/content/pm/ResolveInfo;Ljava/lang/CharSequence;)V
  
+     .line 319
      move-object v12, v4
  
+     .line 320
      move-object/from16 v13, v19
  
+     .line 321
      move/from16 v10, v16
  
      goto :goto_6
  
+     .line 324
      .end local v4           #ri:Landroid/content/pm/ResolveInfo;
      .end local v19           #riLabel:Ljava/lang/CharSequence;
      :cond_d
***************
*** 486,491 ****
--- 540,546 ----
  
      invoke-direct/range {v8 .. v13}, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->processGroup(Ljava/util/List;IILandroid/content/pm/ResolveInfo;Ljava/lang/CharSequence;)V
  
+     .line 326
      .end local v10           #start:I
      .end local v12           #r0:Landroid/content/pm/ResolveInfo;
      .end local v13           #r0Label:Ljava/lang/CharSequence;
***************
*** 500,505 ****
--- 555,561 ----
      .parameter "x0"
  
      .prologue
+     .line 227
      iget-object v0, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
  
      return-object v0
***************
*** 511,516 ****
--- 567,573 ----
      .parameter "info"
  
      .prologue
+     .line 424
      const v3, 0x1020014
  
      invoke-virtual {p1, v3}, Landroid/view/View;->findViewById(I)Landroid/view/View;
***************
*** 519,524 ****
--- 576,582 ----
  
      check-cast v1, Landroid/widget/TextView;
  
+     .line 425
      .local v1, text:Landroid/widget/TextView;
      const v3, 0x1020015
  
***************
*** 528,533 ****
--- 586,592 ----
  
      check-cast v2, Landroid/widget/TextView;
  
+     .line 426
      .local v2, text2:Landroid/widget/TextView;
      const v3, 0x1020006
  
***************
*** 537,564 ****
--- 596,629 ----
  
      check-cast v0, Landroid/widget/ImageView;
  
+     .line 427
      .local v0, icon:Landroid/widget/ImageView;
      iget-object v3, p2, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->displayLabel:Ljava/lang/CharSequence;
  
      invoke-virtual {v1, v3}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 428
      iget-object v3, p2, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->extendedInfo:Ljava/lang/CharSequence;
  
      if-eqz v3, :cond_1
  
+     .line 429
      const/4 v3, 0x0
  
      invoke-virtual {v2, v3}, Landroid/widget/TextView;->setVisibility(I)V
  
+     .line 430
      iget-object v3, p2, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->extendedInfo:Ljava/lang/CharSequence;
  
      invoke-virtual {v2, v3}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 434
      :goto_0
      iget-object v3, p2, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->displayIcon:Landroid/graphics/drawable/Drawable;
  
      if-nez v3, :cond_0
  
+     .line 435
      iget-object v3, p2, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->ri:Landroid/content/pm/ResolveInfo;
  
      iget-object v4, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->this$0:Lcom/android/internal/app/ResolverActivity;
***************
*** 574,586 ****
--- 639,654 ----
  
      iput-object v3, p2, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->displayIcon:Landroid/graphics/drawable/Drawable;
  
+     .line 437
      :cond_0
      iget-object v3, p2, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->displayIcon:Landroid/graphics/drawable/Drawable;
  
      invoke-virtual {v0, v3}, Landroid/widget/ImageView;->setImageDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 438
      return-void
  
+     .line 432
      :cond_1
      const/16 v3, 0x8
  
***************
*** 610,625 ****
--- 678,696 ----
      .end annotation
  
      .prologue
+     .line 331
      .local p1, rList:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
      sub-int v1, p3, p2
  
      add-int/lit8 v12, v1, 0x1
  
+     .line 332
      .local v12, num:I
      const/4 v1, 0x1
  
      if-ne v12, v1, :cond_1
  
+     .line 334
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
***************
*** 642,653 ****
--- 713,727 ----
  
      invoke-interface {v15, v1}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 372
      :cond_0
      return-void
  
+     .line 336
      :cond_1
      const/4 v14, 0x0
  
+     .line 337
      .local v14, usePkg:Z
      move-object/from16 v0, p4
  
***************
*** 668,688 ****
--- 742,768 ----
  
      move-result-object v13
  
+     .line 338
      .local v13, startApp:Ljava/lang/CharSequence;
      if-nez v13, :cond_2
  
+     .line 339
      const/4 v14, 0x1
  
+     .line 341
      :cond_2
      if-nez v14, :cond_5
  
+     .line 343
      new-instance v7, Ljava/util/HashSet;
  
      invoke-direct {v7}, Ljava/util/HashSet;-><init>()V
  
+     .line 345
      .local v7, duplicates:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/CharSequence;>;"
      invoke-virtual {v7, v13}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 346
      add-int/lit8 v8, p2, 0x1
  
      .local v8, j:I
***************
*** 691,696 ****
--- 771,777 ----
  
      if-gt v8, v0, :cond_4
  
+     .line 347
      move-object/from16 v0, p1
  
      invoke-interface {v0, v8}, Ljava/util/List;->get(I)Ljava/lang/Object;
***************
*** 699,704 ****
--- 780,786 ----
  
      check-cast v10, Landroid/content/pm/ResolveInfo;
  
+     .line 348
      .local v10, jRi:Landroid/content/pm/ResolveInfo;
      iget-object v1, v10, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
***************
*** 717,722 ****
--- 799,805 ----
  
      move-result-object v9
  
+     .line 349
      .local v9, jApp:Ljava/lang/CharSequence;
      if-eqz v9, :cond_3
  
***************
*** 726,739 ****
--- 809,825 ----
  
      if-eqz v1, :cond_6
  
+     .line 350
      :cond_3
      const/4 v14, 0x1
  
+     .line 357
      .end local v9           #jApp:Ljava/lang/CharSequence;
      .end local v10           #jRi:Landroid/content/pm/ResolveInfo;
      :cond_4
      invoke-virtual {v7}, Ljava/util/HashSet;->clear()V
  
+     .line 359
      .end local v7           #duplicates:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/CharSequence;>;"
      .end local v8           #j:I
      :cond_5
***************
*** 745,750 ****
--- 831,837 ----
  
      if-gt v11, v0, :cond_0
  
+     .line 360
      move-object/from16 v0, p1
  
      invoke-interface {v0, v11}, Ljava/util/List;->get(I)Ljava/lang/Object;
***************
*** 753,761 ****
--- 840,850 ----
  
      check-cast v3, Landroid/content/pm/ResolveInfo;
  
+     .line 361
      .local v3, add:Landroid/content/pm/ResolveInfo;
      if-eqz v14, :cond_7
  
+     .line 363
      move-object/from16 v0, p0
  
      iget-object v15, v0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
***************
*** 778,788 ****
--- 867,879 ----
  
      invoke-interface {v15, v1}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 359
      :goto_2
      add-int/lit8 v11, v11, 0x1
  
      goto :goto_1
  
+     .line 353
      .end local v3           #add:Landroid/content/pm/ResolveInfo;
      .end local v11           #k:I
      .restart local v7       #duplicates:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/CharSequence;>;"
***************
*** 792,801 ****
--- 883,894 ----
      :cond_6
      invoke-virtual {v7, v9}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 346
      add-int/lit8 v8, v8, 0x1
  
      goto :goto_0
  
+     .line 367
      .end local v7           #duplicates:Ljava/util/HashSet;,"Ljava/util/HashSet<Ljava/lang/CharSequence;>;"
      .end local v8           #j:I
      .end local v9           #jApp:Ljava/lang/CharSequence;
***************
*** 847,852 ****
--- 940,946 ----
      .locals 1
  
      .prologue
+     .line 400
      iget-object v0, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
  
      if-eqz v0, :cond_0
***************
*** 871,876 ****
--- 965,971 ----
      .parameter "position"
  
      .prologue
+     .line 404
      invoke-static {p1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
      move-result-object v0
***************
*** 883,888 ****
--- 978,984 ----
      .parameter "position"
  
      .prologue
+     .line 408
      int-to-long v0, p1
  
      return-wide v0
***************
*** 895,902 ****
--- 991,1000 ----
      .parameter "parent"
  
      .prologue
+     .line 413
      if-nez p2, :cond_0
  
+     .line 414
      iget-object v1, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mInflater:Landroid/view/LayoutInflater;
  
      const v2, 0x109007f
***************
*** 907,912 ****
--- 1005,1011 ----
  
      move-result-object v0
  
+     .line 419
      .local v0, view:Landroid/view/View;
      :goto_0
      iget-object v1, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
***************
*** 919,926 ****
--- 1018,1027 ----
  
      invoke-direct {p0, v0, v1}, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->bindView(Landroid/view/View;Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;)V
  
+     .line 420
      return-object v0
  
+     .line 417
      .end local v0           #view:Landroid/view/View;
      :cond_0
      move-object v0, p2
***************
*** 934,948 ****
--- 1035,1053 ----
      .parameter "position"
  
      .prologue
+     .line 383
      iget-object v3, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
  
      if-nez v3, :cond_0
  
+     .line 384
      const/4 v2, 0x0
  
+     .line 396
      :goto_0
      return-object v2
  
+     .line 387
      :cond_0
      iget-object v3, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
  
***************
*** 952,957 ****
--- 1057,1063 ----
  
      check-cast v1, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;
  
+     .line 389
      .local v1, dri:Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;
      new-instance v2, Landroid/content/Intent;
  
***************
*** 964,978 ****
--- 1070,1087 ----
      :goto_1
      invoke-direct {v2, v3}, Landroid/content/Intent;-><init>(Landroid/content/Intent;)V
  
+     .line 391
      .local v2, intent:Landroid/content/Intent;
      const/high16 v3, 0x300
  
      invoke-virtual {v2, v3}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 393
      iget-object v3, v1, Lcom/android/internal/app/ResolverActivity$DisplayResolveInfo;->ri:Landroid/content/pm/ResolveInfo;
  
      iget-object v0, v3, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
+     .line 394
      .local v0, ai:Landroid/content/pm/ActivityInfo;
      new-instance v3, Landroid/content/ComponentName;
  
***************
*** 988,993 ****
--- 1097,1103 ----
  
      goto :goto_0
  
+     .line 389
      .end local v0           #ai:Landroid/content/pm/ActivityInfo;
      .end local v2           #intent:Landroid/content/Intent;
      :cond_1
***************
*** 1001,1012 ****
--- 1111,1125 ----
      .parameter "position"
  
      .prologue
+     .line 375
      iget-object v0, p0, Lcom/android/internal/app/ResolverActivity$ResolveListAdapter;->mList:Ljava/util/List;
  
      if-nez v0, :cond_0
  
+     .line 376
      const/4 v0, 0x0
  
+     .line 379
      :goto_0
      return-object v0
  
