*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/Manifest$permission_group.smali	2012-03-23 19:36:02.214721370 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/Manifest$permission_group.smali.original	2012-03-23 19:34:44.949312436 -0500
***************
*** 43,48 ****
--- 43,49 ----
      .locals 0
  
      .prologue
+     .line 668
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
