*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$animator.smali	2012-03-23 19:36:19.854043021 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$animator.smali.original	2012-03-23 19:34:45.056314387 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 0
  
      .prologue
+     .line 30
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
