*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$attr.smali	2012-03-23 19:35:23.056007312 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$attr.smali.original	2012-03-23 19:34:44.536304905 -0500
***************
*** 1925,1930 ****
--- 1925,1931 ----
      .locals 0
  
      .prologue
+     .line 55
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
