*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$drawable.smali	2012-03-23 19:36:18.348015559 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$drawable.smali.original	2012-03-23 19:34:45.044314169 -0500
***************
*** 384,389 ****
--- 384,390 ----
      .locals 0
  
      .prologue
+     .line 9896
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
