*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$id.smali	2012-03-23 19:37:17.526094670 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$id.smali.original	2012-03-23 19:34:45.622324709 -0500
***************
*** 113,118 ****
--- 113,119 ----
      .locals 0
  
      .prologue
+     .line 10107
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
