*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$layout.smali	2012-03-23 19:36:52.385636234 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$layout.smali.original	2012-03-23 19:34:45.374320187 -0500
***************
*** 69,74 ****
--- 69,75 ----
      .locals 0
  
      .prologue
+     .line 10263
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
