*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$menu.smali	2012-03-23 19:35:16.185882034 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$menu.smali.original	2012-03-23 19:34:44.442303191 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 0
  
      .prologue
+     .line 10305
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
