*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$plurals.smali	2012-03-23 19:36:52.357635725 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$plurals.smali.original	2012-03-23 19:34:45.374320187 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 0
  
      .prologue
+     .line 10312
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
