*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$raw.smali	2012-03-23 19:35:23.021006673 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$raw.smali.original	2012-03-23 19:34:44.536304905 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 0
  
      .prologue
+     .line 10314
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
