*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$xml.smali	2012-03-23 19:36:18.426016981 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/R$xml.smali.original	2012-03-23 19:34:45.045314187 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 0
  
      .prologue
+     .line 11104
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
