*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/service/textservice/SpellCheckerService$SpellCheckerServiceBinder.smali	2012-03-23 19:36:07.474817287 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/service/textservice/SpellCheckerService$SpellCheckerServiceBinder.smali.original	2012-03-23 19:34:44.992313231 -0500
***************
*** 33,46 ****
--- 33,49 ----
      .parameter "service"
  
      .prologue
+     .line 238
      invoke-direct {p0}, Lcom/android/internal/textservice/ISpellCheckerService$Stub;-><init>()V
  
+     .line 239
      new-instance v0, Ljava/lang/ref/WeakReference;
  
      invoke-direct {v0, p1}, Ljava/lang/ref/WeakReference;-><init>(Ljava/lang/Object;)V
  
      iput-object v0, p0, Landroid/service/textservice/SpellCheckerService$SpellCheckerServiceBinder;->mInternalServiceRef:Ljava/lang/ref/WeakReference;
  
+     .line 240
      return-void
  .end method
  
***************
*** 53,58 ****
--- 56,62 ----
      .parameter "bundle"
  
      .prologue
+     .line 245
      iget-object v3, p0, Landroid/service/textservice/SpellCheckerService$SpellCheckerServiceBinder;->mInternalServiceRef:Ljava/lang/ref/WeakReference;
  
      invoke-virtual {v3}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
***************
*** 61,84 ****
--- 65,93 ----
  
      check-cast v1, Landroid/service/textservice/SpellCheckerService;
  
+     .line 246
      .local v1, service:Landroid/service/textservice/SpellCheckerService;
      if-nez v1, :cond_0
  
      const/4 v0, 0x0
  
+     .line 251
      :goto_0
      return-object v0
  
+     .line 247
      :cond_0
      invoke-virtual {v1}, Landroid/service/textservice/SpellCheckerService;->createSession()Landroid/service/textservice/SpellCheckerService$Session;
  
      move-result-object v2
  
+     .line 248
      .local v2, session:Landroid/service/textservice/SpellCheckerService$Session;
      new-instance v0, Landroid/service/textservice/SpellCheckerService$InternalISpellCheckerSession;
  
      invoke-direct {v0, p1, p2, p3, v2}, Landroid/service/textservice/SpellCheckerService$InternalISpellCheckerSession;-><init>(Ljava/lang/String;Lcom/android/internal/textservice/ISpellCheckerSessionListener;Landroid/os/Bundle;Landroid/service/textservice/SpellCheckerService$Session;)V
  
+     .line 250
      .local v0, internalSession:Landroid/service/textservice/SpellCheckerService$InternalISpellCheckerSession;
      invoke-virtual {v2}, Landroid/service/textservice/SpellCheckerService$Session;->onCreate()V
  
