*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/storage/ExternalStorageFormatter$2.smali	2012-03-23 19:37:51.475713741 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/storage/ExternalStorageFormatter$2.smali.original	2012-03-23 19:34:46.376338459 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 169
      iput-object p1, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->this$0:Lcom/android/internal/os/storage/ExternalStorageFormatter;
  
      iput-object p2, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->val$mountService:Landroid/os/storage/IMountService;
***************
*** 47,54 ****
--- 48,57 ----
      .locals 5
  
      .prologue
+     .line 172
      const/4 v1, 0x0
  
+     .line 174
      .local v1, success:Z
      :try_start_0
      iget-object v2, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->val$mountService:Landroid/os/storage/IMountService;
***************
*** 59,69 ****
--- 62,75 ----
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 175
      const/4 v1, 0x1
  
+     .line 180
      :goto_0
      if-eqz v1, :cond_0
  
+     .line 181
      iget-object v2, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->this$0:Lcom/android/internal/os/storage/ExternalStorageFormatter;
  
      #getter for: Lcom/android/internal/os/storage/ExternalStorageFormatter;->mFactoryReset:Z
***************
*** 73,78 ****
--- 79,85 ----
  
      if-eqz v2, :cond_0
  
+     .line 182
      iget-object v2, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->this$0:Lcom/android/internal/os/storage/ExternalStorageFormatter;
  
      new-instance v3, Landroid/content/Intent;
***************
*** 83,98 ****
--- 90,109 ----
  
      invoke-virtual {v2, v3}, Lcom/android/internal/os/storage/ExternalStorageFormatter;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 184
      iget-object v2, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->this$0:Lcom/android/internal/os/storage/ExternalStorageFormatter;
  
      invoke-virtual {v2}, Lcom/android/internal/os/storage/ExternalStorageFormatter;->stopSelf()V
  
+     .line 200
      :goto_1
      return-void
  
+     .line 176
      :catch_0
      move-exception v0
  
+     .line 177
      .local v0, e:Ljava/lang/Exception;
      iget-object v2, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->this$0:Lcom/android/internal/os/storage/ExternalStorageFormatter;
  
***************
*** 108,113 ****
--- 119,125 ----
  
      goto :goto_0
  
+     .line 190
      .end local v0           #e:Ljava/lang/Exception;
      :cond_0
      if-nez v1, :cond_1
***************
*** 121,126 ****
--- 133,139 ----
  
      if-eqz v2, :cond_1
  
+     .line 191
      iget-object v2, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->this$0:Lcom/android/internal/os/storage/ExternalStorageFormatter;
  
      new-instance v3, Landroid/content/Intent;
***************
*** 131,136 ****
--- 144,150 ----
  
      invoke-virtual {v2, v3}, Lcom/android/internal/os/storage/ExternalStorageFormatter;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 199
      :goto_2
      iget-object v2, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->this$0:Lcom/android/internal/os/storage/ExternalStorageFormatter;
  
***************
*** 138,143 ****
--- 152,158 ----
  
      goto :goto_1
  
+     .line 194
      :cond_1
      :try_start_1
      iget-object v2, p0, Lcom/android/internal/os/storage/ExternalStorageFormatter$2;->val$mountService:Landroid/os/storage/IMountService;
***************
*** 150,158 ****
--- 165,175 ----
  
      goto :goto_2
  
+     .line 195
      :catch_1
      move-exception v0
  
+     .line 196
      .local v0, e:Landroid/os/RemoteException;
      const-string v2, "ExternalStorageFormatter"
  
