*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState.smali	2012-03-23 19:34:56.568524313 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState.smali.original	2012-03-23 19:34:46.900348013 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1080
      iput-object p1, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState;->this$1:Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 39,59 ****
--- 40,64 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 1083
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState;->this$1:Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;
  
      #calls: Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->setAvailable(Z)V
      invoke-static {v0, v1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->access$700(Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;Z)V
  
+     .line 1084
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState;->this$1:Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;
  
      #calls: Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->setLastError(I)V
      invoke-static {v0, v1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->access$1000(Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;I)V
  
+     .line 1085
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState;->this$1:Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;
  
      #calls: Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->setTethered(Z)V
      invoke-static {v0, v1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->access$800(Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;Z)V
  
+     .line 1086
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState;->this$1:Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;
  
      iget-object v0, v0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
***************
*** 61,66 ****
--- 66,72 ----
      #calls: Lcom/android/server/connectivity/Tethering;->sendTetherStateChangedBroadcast()V
      invoke-static {v0}, Lcom/android/server/connectivity/Tethering;->access$900(Lcom/android/server/connectivity/Tethering;)V
  
+     .line 1087
      return-void
  .end method
  
***************
*** 69,86 ****
--- 75,97 ----
      .parameter "message"
  
      .prologue
+     .line 1090
      const/4 v0, 0x1
  
+     .line 1091
      .local v0, retValue:Z
      iget v1, p1, Landroid/os/Message;->what:I
  
      packed-switch v1, :pswitch_data_0
  
+     .line 1096
      const/4 v0, 0x0
  
+     .line 1099
      :goto_0
      return v0
  
+     .line 1093
      :pswitch_0
      iget-object v1, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState;->this$1:Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;
  
***************
*** 96,101 ****
--- 107,113 ----
  
      goto :goto_0
  
+     .line 1091
      :pswitch_data_0
      .packed-switch 0x5
          :pswitch_0
