*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/connectivity/Tethering$TetherMasterSM$StartTetheringErrorState.smali	2012-03-23 19:34:57.128534524 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/connectivity/Tethering$TetherMasterSM$StartTetheringErrorState.smali.original	2012-03-23 19:34:46.919348361 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1476
      iput-object p1, p0, Lcom/android/server/connectivity/Tethering$TetherMasterSM$StartTetheringErrorState;->this$1:Lcom/android/server/connectivity/Tethering$TetherMasterSM;
  
      invoke-direct {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherMasterSM$ErrorState;-><init>(Lcom/android/server/connectivity/Tethering$TetherMasterSM;)V
***************
*** 37,52 ****
--- 38,56 ----
      .locals 2
  
      .prologue
+     .line 1479
      const-string v0, "Tethering"
  
      const-string v1, "Error in startTethering"
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1480
      const/16 v0, 0x9
  
      invoke-virtual {p0, v0}, Lcom/android/server/connectivity/Tethering$TetherMasterSM$StartTetheringErrorState;->notify(I)V
  
+     .line 1482
      :try_start_0
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherMasterSM$StartTetheringErrorState;->this$1:Lcom/android/server/connectivity/Tethering$TetherMasterSM;
  
***************
*** 63,71 ****
--- 67,77 ----
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1484
      :goto_0
      return-void
  
+     .line 1483
      :catch_0
      move-exception v0
  
