*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/connectivity/Tethering$TetherInterfaceSM.smali	2012-03-23 19:34:56.923530786 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/connectivity/Tethering$TetherInterfaceSM.smali.original	2012-03-23 19:34:46.909348178 -0500
***************
*** 84,145 ****
--- 84,160 ----
      .parameter "usb"
  
      .prologue
+     .line 775
      iput-object p1, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
  
+     .line 776
      invoke-direct {p0, p2, p3}, Lcom/android/internal/util/StateMachine;-><init>(Ljava/lang/String;Landroid/os/Looper;)V
  
+     .line 777
      iput-object p2, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mIfaceName:Ljava/lang/String;
  
+     .line 778
      iput-boolean p4, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mUsb:Z
  
+     .line 779
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->setLastError(I)V
  
+     .line 781
      new-instance v0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$InitialState;
  
      invoke-direct {v0, p0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$InitialState;-><init>(Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;)V
  
      iput-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mInitialState:Lcom/android/internal/util/State;
  
+     .line 782
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mInitialState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->addState(Lcom/android/internal/util/State;)V
  
+     .line 783
      new-instance v0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$StartingState;
  
      invoke-direct {v0, p0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$StartingState;-><init>(Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;)V
  
      iput-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mStartingState:Lcom/android/internal/util/State;
  
+     .line 784
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mStartingState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->addState(Lcom/android/internal/util/State;)V
  
+     .line 785
      new-instance v0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$TetheredState;
  
      invoke-direct {v0, p0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$TetheredState;-><init>(Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;)V
  
      iput-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mTetheredState:Lcom/android/internal/util/State;
  
+     .line 786
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mTetheredState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->addState(Lcom/android/internal/util/State;)V
  
+     .line 787
      new-instance v0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState;
  
      invoke-direct {v0, p0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM$UnavailableState;-><init>(Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;)V
  
      iput-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mUnavailableState:Lcom/android/internal/util/State;
  
+     .line 788
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mUnavailableState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->addState(Lcom/android/internal/util/State;)V
  
+     .line 790
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mInitialState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->setInitialState(Lcom/android/internal/util/State;)V
  
+     .line 791
      return-void
  .end method
  
***************
*** 149,154 ****
--- 164,170 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-direct {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->setLastError(I)V
  
      return-void
***************
*** 159,164 ****
--- 175,181 ----
      .parameter "x0"
  
      .prologue
+     .line 732
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mStartingState:Lcom/android/internal/util/State;
  
      return-object v0
***************
*** 170,175 ****
--- 187,193 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 180,185 ****
--- 198,204 ----
      .parameter "x0"
  
      .prologue
+     .line 732
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mUnavailableState:Lcom/android/internal/util/State;
  
      return-object v0
***************
*** 191,196 ****
--- 210,216 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 201,206 ****
--- 221,227 ----
      .parameter "x0"
  
      .prologue
+     .line 732
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mInitialState:Lcom/android/internal/util/State;
  
      return-object v0
***************
*** 212,217 ****
--- 233,239 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 222,227 ****
--- 244,250 ----
      .parameter "x0"
  
      .prologue
+     .line 732
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mTetheredState:Lcom/android/internal/util/State;
  
      return-object v0
***************
*** 233,238 ****
--- 256,262 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 244,249 ****
--- 268,274 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 255,260 ****
--- 280,286 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 266,271 ****
--- 292,298 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 277,282 ****
--- 304,310 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 288,293 ****
--- 316,322 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 299,304 ****
--- 328,334 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 310,315 ****
--- 340,346 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 321,326 ****
--- 352,358 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-virtual {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 332,337 ****
--- 364,370 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-direct {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->setAvailable(Z)V
  
      return-void
***************
*** 343,348 ****
--- 376,382 ----
      .parameter "x1"
  
      .prologue
+     .line 732
      invoke-direct {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->setTethered(Z)V
  
      return-void
***************
*** 353,358 ****
--- 387,393 ----
      .parameter "available"
  
      .prologue
+     .line 834
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
  
      #getter for: Lcom/android/server/connectivity/Tethering;->mPublicSync:Ljava/lang/Object;
***************
*** 362,374 ****
--- 397,413 ----
  
      monitor-enter v1
  
+     .line 835
      :try_start_0
      iput-boolean p1, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mAvailable:Z
  
+     .line 836
      monitor-exit v1
  
+     .line 837
      return-void
  
+     .line 836
      :catchall_0
      move-exception v0
  
***************
*** 384,389 ****
--- 423,429 ----
      .parameter "error"
  
      .prologue
+     .line 814
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
  
      #getter for: Lcom/android/server/connectivity/Tethering;->mPublicSync:Ljava/lang/Object;
***************
*** 393,411 ****
--- 433,455 ----
  
      monitor-enter v1
  
+     .line 815
      :try_start_0
      iput p1, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mLastError:I
  
+     .line 817
      invoke-virtual {p0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->isErrored()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 818
      iget-boolean v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mUsb:Z
  
      if-eqz v0, :cond_0
  
+     .line 821
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
  
      const/4 v2, 0x0
***************
*** 413,423 ****
--- 457,470 ----
      #calls: Lcom/android/server/connectivity/Tethering;->configureUsbIface(Z)Z
      invoke-static {v0, v2}, Lcom/android/server/connectivity/Tethering;->access$600(Lcom/android/server/connectivity/Tethering;Z)Z
  
+     .line 824
      :cond_0
      monitor-exit v1
  
+     .line 825
      return-void
  
+     .line 824
      :catchall_0
      move-exception v0
  
***************
*** 433,438 ****
--- 480,486 ----
      .parameter "tethered"
  
      .prologue
+     .line 846
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
  
      #getter for: Lcom/android/server/connectivity/Tethering;->mPublicSync:Ljava/lang/Object;
***************
*** 442,454 ****
--- 490,506 ----
  
      monitor-enter v1
  
+     .line 847
      :try_start_0
      iput-boolean p1, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mTethered:Z
  
+     .line 848
      monitor-exit v1
  
+     .line 849
      return-void
  
+     .line 848
      :catchall_0
      move-exception v0
  
***************
*** 465,470 ****
--- 517,523 ----
      .locals 2
  
      .prologue
+     .line 808
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
  
      #getter for: Lcom/android/server/connectivity/Tethering;->mPublicSync:Ljava/lang/Object;
***************
*** 474,479 ****
--- 527,533 ----
  
      monitor-enter v1
  
+     .line 809
      :try_start_0
      iget v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mLastError:I
  
***************
*** 481,486 ****
--- 535,541 ----
  
      return v0
  
+     .line 810
      :catchall_0
      move-exception v0
  
***************
*** 495,500 ****
--- 550,556 ----
      .locals 2
  
      .prologue
+     .line 828
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
  
      #getter for: Lcom/android/server/connectivity/Tethering;->mPublicSync:Ljava/lang/Object;
***************
*** 504,509 ****
--- 560,566 ----
  
      monitor-enter v1
  
+     .line 829
      :try_start_0
      iget-boolean v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mAvailable:Z
  
***************
*** 511,516 ****
--- 568,574 ----
  
      return v0
  
+     .line 830
      :catchall_0
      move-exception v0
  
***************
*** 525,530 ****
--- 583,589 ----
      .locals 2
  
      .prologue
+     .line 852
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
  
      #getter for: Lcom/android/server/connectivity/Tethering;->mPublicSync:Ljava/lang/Object;
***************
*** 534,539 ****
--- 593,599 ----
  
      monitor-enter v1
  
+     .line 853
      :try_start_0
      iget v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mLastError:I
  
***************
*** 551,556 ****
--- 611,617 ----
  
      goto :goto_0
  
+     .line 854
      :catchall_0
      move-exception v0
  
***************
*** 565,570 ****
--- 626,632 ----
      .locals 2
  
      .prologue
+     .line 840
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->this$0:Lcom/android/server/connectivity/Tethering;
  
      #getter for: Lcom/android/server/connectivity/Tethering;->mPublicSync:Ljava/lang/Object;
***************
*** 574,579 ****
--- 636,642 ----
  
      monitor-enter v1
  
+     .line 841
      :try_start_0
      iget-boolean v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mTethered:Z
  
***************
*** 581,586 ****
--- 644,650 ----
  
      return v0
  
+     .line 842
      :catchall_0
      move-exception v0
  
***************
*** 596,607 ****
--- 660,674 ----
      .parameter "error"
  
      .prologue
+     .line 1104
      invoke-direct {p0, p1}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->setLastError(I)V
  
+     .line 1105
      iget-object v0, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mInitialState:Lcom/android/internal/util/State;
  
      invoke-virtual {p0, v0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->transitionTo(Lcom/android/internal/util/IState;)V
  
+     .line 1106
      return-void
  .end method
  
***************
*** 609,618 ****
--- 676,687 ----
      .locals 4
  
      .prologue
+     .line 794
      new-instance v1, Ljava/lang/String;
  
      invoke-direct {v1}, Ljava/lang/String;-><init>()V
  
+     .line 795
      .local v1, res:Ljava/lang/String;
      new-instance v2, Ljava/lang/StringBuilder;
  
***************
*** 638,647 ****
--- 707,718 ----
  
      move-result-object v1
  
+     .line 796
      invoke-virtual {p0}, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->getCurrentState()Lcom/android/internal/util/IState;
  
      move-result-object v0
  
+     .line 797
      .local v0, current:Lcom/android/internal/util/IState;
      iget-object v2, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mInitialState:Lcom/android/internal/util/State;
  
***************
*** 665,670 ****
--- 736,742 ----
  
      move-result-object v1
  
+     .line 798
      :cond_0
      iget-object v2, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mStartingState:Lcom/android/internal/util/State;
  
***************
*** 688,693 ****
--- 760,766 ----
  
      move-result-object v1
  
+     .line 799
      :cond_1
      iget-object v2, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mTetheredState:Lcom/android/internal/util/State;
  
***************
*** 711,716 ****
--- 784,790 ----
  
      move-result-object v1
  
+     .line 800
      :cond_2
      iget-object v2, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mUnavailableState:Lcom/android/internal/util/State;
  
***************
*** 734,739 ****
--- 808,814 ----
  
      move-result-object v1
  
+     .line 801
      :cond_3
      iget-boolean v2, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mAvailable:Z
  
***************
*** 757,762 ****
--- 832,838 ----
  
      move-result-object v1
  
+     .line 802
      :cond_4
      iget-boolean v2, p0, Lcom/android/server/connectivity/Tethering$TetherInterfaceSM;->mTethered:Z
  
***************
*** 780,785 ****
--- 856,862 ----
  
      move-result-object v1
  
+     .line 803
      :cond_5
      new-instance v2, Ljava/lang/StringBuilder;
  
***************
*** 805,809 ****
--- 882,887 ----
  
      move-result-object v1
  
+     .line 804
      return-object v1
  .end method
