*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/backup/IBackupManager$Stub$Proxy.smali	2012-03-23 19:36:40.089412013 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/backup/IBackupManager$Stub$Proxy.smali.original	2012-03-23 19:34:45.293318709 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 301
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 302
      iput-object p1, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 303
      return-void
  .end method
  
***************
*** 52,74 ****
--- 55,82 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 678
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 679
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 681
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 682
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 683
      if-eqz p2, :cond_0
  
      const/4 v2, 0x1
***************
*** 76,85 ****
--- 84,96 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 684
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 685
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 686
      if-eqz p5, :cond_1
  
      invoke-interface {p5}, Landroid/app/backup/IFullBackupRestoreObserver;->asBinder()Landroid/os/IBinder;
***************
*** 89,94 ****
--- 100,106 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 687
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xf
***************
*** 97,122 ****
--- 109,141 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 688
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 691
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 692
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 694
      return-void
  
+     .line 686
      :cond_1
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 691
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 692
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 133,157 ****
--- 152,182 ----
      .end annotation
  
      .prologue
+     .line 364
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 365
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 367
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 368
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 369
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 370
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x3
***************
*** 160,180 ****
--- 185,211 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 371
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 374
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 375
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 377
      return-void
  
+     .line 374
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 375
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 190,212 ****
--- 221,248 ----
      .end annotation
  
      .prologue
+     .line 384
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 385
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 387
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 388
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 389
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x4
***************
*** 215,235 ****
--- 251,277 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 390
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 393
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 394
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 396
      return-void
  
+     .line 393
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 394
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 239,244 ****
--- 281,287 ----
      .locals 1
  
      .prologue
+     .line 306
      iget-object v0, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 253,273 ****
--- 296,320 ----
      .end annotation
  
      .prologue
+     .line 572
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 573
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 575
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 576
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xc
***************
*** 276,296 ****
--- 323,349 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 577
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 580
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 581
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 583
      return-void
  
+     .line 580
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 581
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 307,331 ****
--- 360,390 ----
      .end annotation
  
      .prologue
+     .line 844
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 845
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 848
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 849
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 850
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 851
      iget-object v3, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x15
***************
*** 334,341 ****
--- 393,402 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 852
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 853
      invoke-virtual {v1}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
  
      move-result-object v3
***************
*** 346,364 ****
--- 407,430 ----
  
      move-result-object v2
  
+     .line 856
      .local v2, _result:Landroid/app/backup/IRestoreSession;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 857
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 859
      return-object v2
  
+     .line 856
      .end local v2           #_result:Landroid/app/backup/IRestoreSession;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 857
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 374,396 ****
--- 440,467 ----
      .end annotation
  
      .prologue
+     .line 345
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 346
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 348
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 349
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 350
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x2
***************
*** 399,419 ****
--- 470,496 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 351
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 354
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 355
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 357
      return-void
  
+     .line 354
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 355
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 429,451 ****
--- 506,533 ----
      .end annotation
  
      .prologue
+     .line 322
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 323
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 325
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 326
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 327
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x1
***************
*** 454,474 ****
--- 536,562 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 328
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 331
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 332
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 334
      return-void
  
+     .line 331
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 332
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 493,523 ****
--- 581,618 ----
  
      const/4 v3, 0x0
  
+     .line 611
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 612
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 614
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 615
      if-eqz p1, :cond_0
  
+     .line 616
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 617
      const/4 v4, 0x0
  
      invoke-virtual {p1, v0, v4}, Landroid/os/ParcelFileDescriptor;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 622
      :goto_0
      if-eqz p2, :cond_1
  
***************
*** 526,531 ****
--- 621,627 ----
      :goto_1
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 623
      if-eqz p3, :cond_2
  
      move v4, v2
***************
*** 533,538 ****
--- 629,635 ----
      :goto_2
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 624
      if-eqz p4, :cond_3
  
      move v4, v2
***************
*** 540,552 ****
--- 637,652 ----
      :goto_3
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 625
      if-eqz p5, :cond_4
  
      :goto_4
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 626
      invoke-virtual {v0, p6}, Landroid/os/Parcel;->writeStringArray([Ljava/lang/String;)V
  
+     .line 627
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xd
***************
*** 555,570 ****
--- 655,675 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 628
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 631
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 632
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 634
      return-void
  
+     .line 620
      :cond_0
      const/4 v4, 0x0
  
***************
*** 575,585 ****
--- 680,692 ----
  
      goto :goto_0
  
+     .line 631
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 632
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 587,607 ****
--- 694,718 ----
      :cond_1
      move v4, v3
  
+     .line 622
      goto :goto_1
  
      :cond_2
      move v4, v3
  
+     .line 623
      goto :goto_2
  
      :cond_3
      move v4, v3
  
+     .line 624
      goto :goto_3
  
      :cond_4
      move v2, v3
  
+     .line 625
      goto :goto_4
  .end method
  
***************
*** 615,645 ****
--- 726,763 ----
      .end annotation
  
      .prologue
+     .line 643
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 644
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 646
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 647
      if-eqz p1, :cond_0
  
+     .line 648
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 649
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/os/ParcelFileDescriptor;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 654
      :goto_0
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 649,664 ****
--- 767,787 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 655
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 658
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 659
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 661
      return-void
  
+     .line 652
      :cond_0
      const/4 v2, 0x0
  
***************
*** 669,679 ****
--- 792,804 ----
  
      goto :goto_0
  
+     .line 658
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 659
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 689,711 ****
--- 814,841 ----
      .end annotation
  
      .prologue
+     .line 776
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 777
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 780
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 781
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 782
      iget-object v3, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x13
***************
*** 714,727 ****
--- 844,860 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 783
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 784
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 785
      sget-object v3, Landroid/content/Intent;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 732,745 ****
--- 865,882 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 792
      .local v2, _result:Landroid/content/Intent;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 793
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 795
      return-object v2
  
+     .line 788
      .end local v2           #_result:Landroid/content/Intent;
      :cond_0
      const/4 v2, 0x0
***************
*** 747,758 ****
--- 884,897 ----
      .restart local v2       #_result:Landroid/content/Intent;
      goto :goto_0
  
+     .line 792
      .end local v2           #_result:Landroid/content/Intent;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 793
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 767,787 ****
--- 906,930 ----
      .end annotation
  
      .prologue
+     .line 701
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 702
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 705
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 706
      iget-object v3, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x10
***************
*** 790,816 ****
--- 933,966 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 707
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 708
      invoke-virtual {v1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result-object v2
  
+     .line 711
      .local v2, _result:Ljava/lang/String;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 712
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 714
      return-object v2
  
+     .line 711
      .end local v2           #_result:Ljava/lang/String;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 712
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 826,848 ****
--- 976,1003 ----
      .end annotation
  
      .prologue
+     .line 807
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 808
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 811
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 812
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 813
      iget-object v3, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x14
***************
*** 851,877 ****
--- 1006,1039 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 814
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 815
      invoke-virtual {v1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result-object v2
  
+     .line 818
      .local v2, _result:Ljava/lang/String;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 819
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 821
      return-object v2
  
+     .line 818
      .end local v2           #_result:Ljava/lang/String;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 819
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 881,886 ****
--- 1043,1049 ----
      .locals 1
  
      .prologue
+     .line 310
      const-string v0, "android.app.backup.IBackupManager"
  
      return-object v0
***************
*** 897,917 ****
--- 1060,1084 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 545
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 546
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 549
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 550
      iget-object v3, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xb
***************
*** 920,927 ****
--- 1087,1096 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 551
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 552
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 932,951 ****
--- 1101,1125 ----
  
      const/4 v2, 0x1
  
+     .line 555
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 556
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 558
      return v2
  
+     .line 555
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 556
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 962,982 ****
--- 1136,1160 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 496
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 497
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 500
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 501
      iget-object v3, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x9
***************
*** 985,992 ****
--- 1163,1172 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 502
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 503
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 997,1016 ****
--- 1177,1201 ----
  
      const/4 v2, 0x1
  
+     .line 506
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 507
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 509
      return v2
  
+     .line 506
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 507
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1025,1045 ****
--- 1210,1234 ----
      .end annotation
  
      .prologue
+     .line 722
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 723
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 726
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 727
      iget-object v3, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x11
***************
*** 1048,1074 ****
--- 1237,1270 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 728
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 729
      invoke-virtual {v1}, Landroid/os/Parcel;->createStringArray()[Ljava/lang/String;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result-object v2
  
+     .line 732
      .local v2, _result:[Ljava/lang/String;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 733
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 735
      return-object v2
  
+     .line 732
      .end local v2           #_result:[Ljava/lang/String;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 733
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1084,1106 ****
--- 1280,1307 ----
      .end annotation
  
      .prologue
+     .line 871
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 872
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 874
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 875
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 876
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x16
***************
*** 1109,1129 ****
--- 1310,1336 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 877
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 880
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 881
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 883
      return-void
  
+     .line 880
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 881
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1140,1164 ****
--- 1347,1377 ----
      .end annotation
  
      .prologue
+     .line 403
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 404
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 406
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 407
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 408
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 409
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x5
***************
*** 1167,1187 ****
--- 1380,1406 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 410
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 413
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 414
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 416
      return-void
  
+     .line 413
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 414
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1197,1219 ****
--- 1416,1443 ----
      .end annotation
  
      .prologue
+     .line 749
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 750
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 753
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 754
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 755
      iget-object v3, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x12
***************
*** 1222,1248 ****
--- 1446,1479 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 756
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 757
      invoke-virtual {v1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result-object v2
  
+     .line 760
      .local v2, _result:Ljava/lang/String;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 761
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 763
      return-object v2
  
+     .line 760
      .end local v2           #_result:Ljava/lang/String;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 761
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1260,1280 ****
--- 1491,1515 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 456
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 457
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 459
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 460
      if-eqz p1, :cond_0
  
      const/4 v2, 0x1
***************
*** 1282,1287 ****
--- 1517,1523 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 461
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x7
***************
*** 1290,1310 ****
--- 1526,1552 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 462
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 465
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 466
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 468
      return-void
  
+     .line 465
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 466
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1322,1342 ****
--- 1564,1588 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 428
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 429
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 431
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 432
      if-eqz p1, :cond_0
  
      const/4 v2, 0x1
***************
*** 1344,1349 ****
--- 1590,1596 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 433
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x6
***************
*** 1352,1372 ****
--- 1599,1625 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 434
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 437
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 438
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 440
      return-void
  
+     .line 437
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 438
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1385,1409 ****
--- 1638,1668 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 520
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 521
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 524
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 525
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 526
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 527
      iget-object v3, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xa
***************
*** 1412,1419 ****
--- 1671,1680 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 528
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 529
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1424,1443 ****
--- 1685,1709 ----
  
      const/4 v2, 0x1
  
+     .line 532
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 533
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 535
      return v2
  
+     .line 532
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 533
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1455,1475 ****
--- 1721,1745 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 476
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 477
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 479
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.app.backup.IBackupManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 480
      if-eqz p1, :cond_0
  
      const/4 v2, 0x1
***************
*** 1477,1482 ****
--- 1747,1753 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 481
      iget-object v2, p0, Landroid/app/backup/IBackupManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x8
***************
*** 1485,1505 ****
--- 1756,1782 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 482
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 485
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 486
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 488
      return-void
  
+     .line 485
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 486
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
