*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/backup/IFullBackupRestoreObserver$Stub$Proxy.smali	2012-03-23 19:36:41.067429848 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/backup/IFullBackupRestoreObserver$Stub$Proxy.smali.original	2012-03-23 19:34:45.301318855 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 104
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 105
      iput-object p1, p0, Landroid/app/backup/IFullBackupRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 106
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 109
      iget-object v0, p0, Landroid/app/backup/IFullBackupRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 49,54 ****
--- 53,59 ----
      .locals 1
  
      .prologue
+     .line 113
      const-string v0, "android.app.backup.IFullBackupRestoreObserver"
  
      return-object v0
***************
*** 64,81 ****
--- 69,90 ----
      .end annotation
  
      .prologue
+     .line 138
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 140
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IFullBackupRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 141
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 142
      iget-object v1, p0, Landroid/app/backup/IFullBackupRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x2
***************
*** 88,97 ****
--- 97,109 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 145
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 147
      return-void
  
+     .line 145
      :catchall_0
      move-exception v1
  
***************
*** 109,124 ****
--- 121,139 ----
      .end annotation
  
      .prologue
+     .line 153
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 155
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IFullBackupRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 156
      iget-object v1, p0, Landroid/app/backup/IFullBackupRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x3
***************
*** 131,140 ****
--- 146,158 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 159
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 161
      return-void
  
+     .line 159
      :catchall_0
      move-exception v1
  
***************
*** 152,167 ****
--- 170,188 ----
      .end annotation
  
      .prologue
+     .line 200
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 202
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IFullBackupRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 203
      iget-object v1, p0, Landroid/app/backup/IFullBackupRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x6
***************
*** 174,183 ****
--- 195,207 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 206
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 208
      return-void
  
+     .line 206
      :catchall_0
      move-exception v1
  
***************
*** 196,213 ****
--- 220,241 ----
      .end annotation
  
      .prologue
+     .line 185
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 187
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IFullBackupRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 188
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 189
      iget-object v1, p0, Landroid/app/backup/IFullBackupRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x5
***************
*** 220,229 ****
--- 248,260 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 192
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 194
      return-void
  
+     .line 192
      :catchall_0
      move-exception v1
  
***************
*** 241,256 ****
--- 272,290 ----
      .end annotation
  
      .prologue
+     .line 120
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 122
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IFullBackupRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 123
      iget-object v1, p0, Landroid/app/backup/IFullBackupRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x1
***************
*** 263,272 ****
--- 297,309 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 126
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 128
      return-void
  
+     .line 126
      :catchall_0
      move-exception v1
  
***************
*** 284,299 ****
--- 321,339 ----
      .end annotation
  
      .prologue
+     .line 167
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 169
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IFullBackupRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 170
      iget-object v1, p0, Landroid/app/backup/IFullBackupRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x4
***************
*** 306,315 ****
--- 346,358 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 173
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 175
      return-void
  
+     .line 173
      :catchall_0
      move-exception v1
  
***************
*** 327,342 ****
--- 370,388 ----
      .end annotation
  
      .prologue
+     .line 214
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 216
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IFullBackupRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 217
      iget-object v1, p0, Landroid/app/backup/IFullBackupRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x7
***************
*** 349,358 ****
--- 395,407 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 220
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 222
      return-void
  
+     .line 220
      :catchall_0
      move-exception v1
  
