*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/backup/IRestoreObserver$Stub$Proxy.smali	2012-03-23 19:36:40.439418396 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/backup/IRestoreObserver$Stub$Proxy.smali.original	2012-03-23 19:34:45.296318765 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 91
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 92
      iput-object p1, p0, Landroid/app/backup/IRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 93
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 96
      iget-object v0, p0, Landroid/app/backup/IRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 49,54 ****
--- 53,59 ----
      .locals 1
  
      .prologue
+     .line 100
      const-string v0, "android.app.backup.IRestoreObserver"
  
      return-object v0
***************
*** 65,84 ****
--- 70,94 ----
      .end annotation
  
      .prologue
+     .line 153
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 155
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 156
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 157
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 158
      iget-object v1, p0, Landroid/app/backup/IRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x3
***************
*** 91,100 ****
--- 101,113 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 161
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 163
      return-void
  
+     .line 161
      :catchall_0
      move-exception v1
  
***************
*** 113,130 ****
--- 126,147 ----
      .end annotation
  
      .prologue
+     .line 172
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 174
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 175
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 176
      iget-object v1, p0, Landroid/app/backup/IRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x4
***************
*** 137,146 ****
--- 154,166 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 179
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 181
      return-void
  
+     .line 179
      :catchall_0
      move-exception v1
  
***************
*** 159,178 ****
--- 179,202 ----
      .end annotation
  
      .prologue
+     .line 114
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 116
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 117
      const/4 v1, 0x0
  
      invoke-virtual {v0, p1, v1}, Landroid/os/Parcel;->writeTypedArray([Landroid/os/Parcelable;I)V
  
+     .line 118
      iget-object v1, p0, Landroid/app/backup/IRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x1
***************
*** 185,194 ****
--- 209,221 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 121
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 123
      return-void
  
+     .line 121
      :catchall_0
      move-exception v1
  
***************
*** 207,224 ****
--- 234,255 ----
      .end annotation
  
      .prologue
+     .line 132
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 134
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.backup.IRestoreObserver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 135
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 136
      iget-object v1, p0, Landroid/app/backup/IRestoreObserver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x2
***************
*** 231,240 ****
--- 262,274 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 139
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 141
      return-void
  
+     .line 139
      :catchall_0
      move-exception v1
  
