*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/backup/RestoreSession$RestoreObserverWrapper$1.smali	2012-03-23 19:36:40.993428498 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/backup/RestoreSession$RestoreObserverWrapper$1.smali.original	2012-03-23 19:34:45.300318837 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .parameter
  
      .prologue
+     .line 194
      iput-object p1, p0, Landroid/app/backup/RestoreSession$RestoreObserverWrapper$1;->this$1:Landroid/app/backup/RestoreSession$RestoreObserverWrapper;
  
      iput-object p3, p0, Landroid/app/backup/RestoreSession$RestoreObserverWrapper$1;->val$this$0:Landroid/app/backup/RestoreSession;
***************
*** 44,56 ****
--- 45,60 ----
      .parameter "msg"
  
      .prologue
+     .line 197
      iget v0, p1, Landroid/os/Message;->what:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 211
      :goto_0
      return-void
  
+     .line 199
      :pswitch_0
      iget-object v0, p0, Landroid/app/backup/RestoreSession$RestoreObserverWrapper$1;->this$1:Landroid/app/backup/RestoreSession$RestoreObserverWrapper;
  
***************
*** 62,67 ****
--- 66,72 ----
  
      goto :goto_0
  
+     .line 202
      :pswitch_1
      iget-object v0, p0, Landroid/app/backup/RestoreSession$RestoreObserverWrapper$1;->this$1:Landroid/app/backup/RestoreSession$RestoreObserverWrapper;
  
***************
*** 77,82 ****
--- 82,88 ----
  
      goto :goto_0
  
+     .line 205
      :pswitch_2
      iget-object v0, p0, Landroid/app/backup/RestoreSession$RestoreObserverWrapper$1;->this$1:Landroid/app/backup/RestoreSession$RestoreObserverWrapper;
  
***************
*** 88,93 ****
--- 94,100 ----
  
      goto :goto_0
  
+     .line 208
      :pswitch_3
      iget-object v0, p0, Landroid/app/backup/RestoreSession$RestoreObserverWrapper$1;->this$1:Landroid/app/backup/RestoreSession$RestoreObserverWrapper;
  
***************
*** 103,108 ****
--- 110,116 ----
  
      goto :goto_0
  
+     .line 197
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
