*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothA2dpService.smali	2012-03-23 19:37:14.860046054 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothA2dpService.smali.original	2012-03-23 19:34:45.599324289 -0500
***************
*** 56,71 ****
--- 56,75 ----
      .parameter "bluetoothService"
  
      .prologue
+     .line 131
      invoke-direct {p0}, Landroid/bluetooth/IBluetoothA2dp$Stub;-><init>()V
  
+     .line 69
      new-instance v0, Landroid/server/BluetoothA2dpService$1;
  
      invoke-direct {v0, p0}, Landroid/server/BluetoothA2dpService$1;-><init>(Landroid/server/BluetoothA2dpService;)V
  
      iput-object v0, p0, Landroid/server/BluetoothA2dpService;->mReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 132
      iput-object p1, p0, Landroid/server/BluetoothA2dpService;->mContext:Landroid/content/Context;
  
+     .line 134
      const-string v0, "audio"
  
      invoke-virtual {p1, v0}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 76,87 ****
--- 80,94 ----
  
      iput-object v0, p0, Landroid/server/BluetoothA2dpService;->mAudioManager:Landroid/media/AudioManager;
  
+     .line 136
      iput-object p2, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
+     .line 137
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      if-nez v0, :cond_0
  
+     .line 138
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "Platform does not support Bluetooth"
***************
*** 90,95 ****
--- 97,103 ----
  
      throw v0
  
+     .line 141
      :cond_0
      invoke-direct {p0}, Landroid/server/BluetoothA2dpService;->initNative()Z
  
***************
*** 97,102 ****
--- 105,111 ----
  
      if-nez v0, :cond_1
  
+     .line 142
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "Could not init BluetoothA2dpService"
***************
*** 105,110 ****
--- 114,120 ----
  
      throw v0
  
+     .line 145
      :cond_1
      invoke-static {}, Landroid/bluetooth/BluetoothAdapter;->getDefaultAdapter()Landroid/bluetooth/BluetoothAdapter;
  
***************
*** 112,117 ****
--- 122,128 ----
  
      iput-object v0, p0, Landroid/server/BluetoothA2dpService;->mAdapter:Landroid/bluetooth/BluetoothAdapter;
  
+     .line 147
      new-instance v0, Landroid/content/IntentFilter;
  
      const-string v1, "android.bluetooth.adapter.action.STATE_CHANGED"
***************
*** 120,143 ****
--- 131,158 ----
  
      iput-object v0, p0, Landroid/server/BluetoothA2dpService;->mIntentFilter:Landroid/content/IntentFilter;
  
+     .line 148
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mIntentFilter:Landroid/content/IntentFilter;
  
      const-string v1, "android.bluetooth.device.action.ACL_CONNECTED"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 149
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mIntentFilter:Landroid/content/IntentFilter;
  
      const-string v1, "android.bluetooth.device.action.ACL_DISCONNECTED"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 150
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mIntentFilter:Landroid/content/IntentFilter;
  
      const-string v1, "android.media.VOLUME_CHANGED_ACTION"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 151
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mContext:Landroid/content/Context;
  
      iget-object v1, p0, Landroid/server/BluetoothA2dpService;->mReceiver:Landroid/content/BroadcastReceiver;
***************
*** 146,157 ****
--- 161,174 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 153
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
+     .line 155
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      invoke-virtual {v0}, Landroid/server/BluetoothService;->isEnabled()Z
***************
*** 160,176 ****
--- 177,197 ----
  
      if-eqz v0, :cond_2
  
+     .line 156
      invoke-direct {p0}, Landroid/server/BluetoothA2dpService;->onBluetoothEnable()V
  
+     .line 157
      :cond_2
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/server/BluetoothA2dpService;->mTargetA2dpState:I
  
+     .line 158
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      invoke-virtual {v0, p0}, Landroid/server/BluetoothService;->setA2dpService(Landroid/server/BluetoothA2dpService;)V
  
+     .line 159
      return-void
  .end method
  
***************
*** 179,184 ****
--- 200,206 ----
      .parameter "x0"
  
      .prologue
+     .line 47
      invoke-direct {p0}, Landroid/server/BluetoothA2dpService;->onBluetoothEnable()V
  
      return-void
***************
*** 189,194 ****
--- 211,217 ----
      .parameter "x0"
  
      .prologue
+     .line 47
      invoke-direct {p0}, Landroid/server/BluetoothA2dpService;->onBluetoothDisable()V
  
      return-void
***************
*** 199,204 ****
--- 222,228 ----
      .parameter "x0"
  
      .prologue
+     .line 47
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      return-object v0
***************
*** 212,217 ****
--- 236,242 ----
      .parameter "x3"
  
      .prologue
+     .line 47
      invoke-direct {p0, p1, p2, p3}, Landroid/server/BluetoothA2dpService;->handleSinkStateChange(Landroid/bluetooth/BluetoothDevice;II)V
  
      return-void
***************
*** 223,228 ****
--- 248,254 ----
      .parameter "x1"
  
      .prologue
+     .line 47
      invoke-direct {p0, p1}, Landroid/server/BluetoothA2dpService;->isPhoneDocked(Landroid/bluetooth/BluetoothDevice;)Z
  
      move-result v0
***************
*** 235,240 ****
--- 261,267 ----
      .parameter "x0"
  
      .prologue
+     .line 47
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      return-object v0
***************
*** 246,251 ****
--- 273,279 ----
      .parameter "x1"
  
      .prologue
+     .line 47
      invoke-direct {p0, p1}, Landroid/server/BluetoothA2dpService;->avrcpVolumeUpNative(Ljava/lang/String;)Z
  
      move-result v0
***************
*** 259,264 ****
--- 287,293 ----
      .parameter "x1"
  
      .prologue
+     .line 47
      invoke-direct {p0, p1}, Landroid/server/BluetoothA2dpService;->avrcpVolumeDownNative(Ljava/lang/String;)Z
  
      move-result v0
***************
*** 271,276 ****
--- 300,306 ----
      .parameter "device"
  
      .prologue
+     .line 191
      monitor-enter p0
  
      :try_start_0
***************
*** 282,287 ****
--- 312,318 ----
  
      if-nez v0, :cond_0
  
+     .line 192
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      const/4 v1, 0x0
***************
*** 294,304 ****
--- 325,337 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 194
      :cond_0
      monitor-exit p0
  
      return-void
  
+     .line 191
      :catchall_0
      move-exception v0
  
***************
*** 312,317 ****
--- 345,351 ----
      .parameter "connectedDevice"
  
      .prologue
+     .line 543
      iget-object v2, p0, Landroid/server/BluetoothA2dpService;->mAdapter:Landroid/bluetooth/BluetoothAdapter;
  
      invoke-virtual {v2}, Landroid/bluetooth/BluetoothAdapter;->getBondedDevices()Ljava/util/Set;
***************
*** 337,342 ****
--- 371,377 ----
  
      check-cast v0, Landroid/bluetooth/BluetoothDevice;
  
+     .line 544
      .local v0, device:Landroid/bluetooth/BluetoothDevice;
      invoke-virtual {p0, v0}, Landroid/server/BluetoothA2dpService;->getPriority(Landroid/bluetooth/BluetoothDevice;)I
  
***************
*** 352,363 ****
--- 387,400 ----
  
      if-nez v2, :cond_0
  
+     .line 546
      const/16 v2, 0x64
  
      invoke-virtual {p0, v0, v2}, Landroid/server/BluetoothA2dpService;->setPriority(Landroid/bluetooth/BluetoothDevice;I)Z
  
      goto :goto_0
  
+     .line 549
      .end local v0           #device:Landroid/bluetooth/BluetoothDevice;
      :cond_1
      return-void
***************
*** 378,406 ****
--- 415,449 ----
  
      const/4 v2, 0x2
  
+     .line 552
      const/4 v0, 0x1
  
+     .line 554
      .local v0, result:Z
      iget v1, p0, Landroid/server/BluetoothA2dpService;->mTargetA2dpState:I
  
      if-eq p1, v1, :cond_0
  
+     .line 555
      if-ne p1, v3, :cond_1
  
      iget v1, p0, Landroid/server/BluetoothA2dpService;->mTargetA2dpState:I
  
      if-ne v1, v2, :cond_1
  
+     .line 557
      iget-object v1, p0, Landroid/server/BluetoothA2dpService;->mAudioManager:Landroid/media/AudioManager;
  
      const-string v2, "A2dpSuspended=true"
  
      invoke-virtual {v1, v2}, Landroid/media/AudioManager;->setParameters(Ljava/lang/String;)V
  
+     .line 565
      :cond_0
      :goto_0
      return v0
  
+     .line 558
      :cond_1
      if-ne p1, v2, :cond_2
  
***************
*** 408,413 ****
--- 451,457 ----
  
      if-ne v1, v3, :cond_2
  
+     .line 560
      iget-object v1, p0, Landroid/server/BluetoothA2dpService;->mAudioManager:Landroid/media/AudioManager;
  
      const-string v2, "A2dpSuspended=false"
***************
*** 416,421 ****
--- 460,466 ----
  
      goto :goto_0
  
+     .line 562
      :cond_2
      const/4 v0, 0x0
  
***************
*** 433,438 ****
--- 478,484 ----
      .parameter "value"
  
      .prologue
+     .line 171
      const-string v0, "disconnected"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
***************
*** 441,451 ****
--- 487,500 ----
  
      if-eqz v0, :cond_0
  
+     .line 172
      const/4 v0, 0x0
  
+     .line 179
      :goto_0
      return v0
  
+     .line 173
      :cond_0
      const-string v0, "connecting"
  
***************
*** 455,464 ****
--- 504,515 ----
  
      if-eqz v0, :cond_1
  
+     .line 174
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 175
      :cond_1
      const-string v0, "connected"
  
***************
*** 468,479 ****
  
      if-eqz v0, :cond_2
  
      const/4 v0, 0x2
  
      goto :goto_0
  
      :cond_2
!     const-string v0, "playing"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 519,532 ----
  
      if-eqz v0, :cond_2
  
+     .line 176
      const/4 v0, 0x2
  
      goto :goto_0
  
+     .line 177
      :cond_2
!     const-string/jumbo v0, "playing"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 481,490 ****
--- 534,545 ----
  
      if-eqz v0, :cond_3
  
+     .line 178
      const/16 v0, 0xa
  
      goto :goto_0
  
+     .line 179
      :cond_3
      const/4 v0, -0x1
  
***************
*** 504,538 ****
--- 559,600 ----
      .parameter "prevState"
  
      .prologue
+     .line 532
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.bluetooth.a2dp.profile.action.PLAYING_STATE_CHANGED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 533
      .local v0, intent:Landroid/content/Intent;
      const-string v1, "android.bluetooth.device.extra.DEVICE"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 534
      const-string v1, "android.bluetooth.profile.extra.PREVIOUS_STATE"
  
      invoke-virtual {v0, v1, p3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 535
      const-string v1, "android.bluetooth.profile.extra.STATE"
  
      invoke-virtual {v0, v1, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 536
      const/high16 v1, 0x1000
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 537
      iget-object v1, p0, Landroid/server/BluetoothA2dpService;->mContext:Landroid/content/Context;
  
      const-string v2, "android.permission.BLUETOOTH"
  
      invoke-virtual {v1, v0, v2}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;Ljava/lang/String;)V
  
+     .line 539
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 573,578 ****
--- 635,641 ----
  
      invoke-static {v1}, Landroid/server/BluetoothA2dpService;->log(Ljava/lang/String;)V
  
+     .line 540
      return-void
  .end method
  
***************
*** 585,592 ****
--- 648,657 ----
      .prologue
      const/4 v3, 0x2
  
+     .line 501
      if-eq p3, p2, :cond_1
  
+     .line 502
      iget-object v1, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      invoke-static {p3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 595,606 ****
--- 660,674 ----
  
      invoke-virtual {v1, p1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 504
      invoke-direct {p0, p3}, Landroid/server/BluetoothA2dpService;->checkSinkSuspendState(I)Z
  
+     .line 505
      const/4 v1, -0x1
  
      iput v1, p0, Landroid/server/BluetoothA2dpService;->mTargetA2dpState:I
  
+     .line 507
      invoke-virtual {p0, p1}, Landroid/server/BluetoothA2dpService;->getPriority(Landroid/bluetooth/BluetoothDevice;)I
  
      move-result v1
***************
*** 609,620 ****
--- 677,691 ----
  
      if-ne p3, v3, :cond_0
  
+     .line 511
      const/16 v1, 0x3e8
  
      invoke-virtual {p0, p1, v1}, Landroid/server/BluetoothA2dpService;->setPriority(Landroid/bluetooth/BluetoothDevice;I)Z
  
+     .line 514
      invoke-direct {p0, p1}, Landroid/server/BluetoothA2dpService;->adjustOtherSinkPriorities(Landroid/bluetooth/BluetoothDevice;)V
  
+     .line 517
      :cond_0
      new-instance v0, Landroid/content/Intent;
  
***************
*** 622,650 ****
--- 693,727 ----
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 518
      .local v0, intent:Landroid/content/Intent;
      const-string v1, "android.bluetooth.device.extra.DEVICE"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 519
      const-string v1, "android.bluetooth.profile.extra.PREVIOUS_STATE"
  
      invoke-virtual {v0, v1, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 520
      const-string v1, "android.bluetooth.profile.extra.STATE"
  
      invoke-virtual {v0, v1, p3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 521
      const/high16 v1, 0x1000
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 522
      iget-object v1, p0, Landroid/server/BluetoothA2dpService;->mContext:Landroid/content/Context;
  
      const-string v2, "android.permission.BLUETOOTH"
  
      invoke-virtual {v1, v0, v2}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;Ljava/lang/String;)V
  
+     .line 524
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 685,694 ****
--- 762,773 ----
  
      invoke-static {v1}, Landroid/server/BluetoothA2dpService;->log(Ljava/lang/String;)V
  
+     .line 526
      iget-object v1, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      invoke-virtual {v1, p1, v3, p3, p2}, Landroid/server/BluetoothService;->sendConnectionStateChange(Landroid/bluetooth/BluetoothDevice;III)V
  
+     .line 529
      .end local v0           #intent:Landroid/content/Intent;
      :cond_1
      return-void
***************
*** 704,709 ****
--- 783,789 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 243
      monitor-enter p0
  
      :try_start_0
***************
*** 729,744 ****
--- 809,827 ----
  
      if-nez v2, :cond_1
  
+     .line 254
      :cond_0
      :goto_0
      monitor-exit p0
  
      return v1
  
+     .line 248
      :cond_1
      :try_start_1
      invoke-direct {p0, p1}, Landroid/server/BluetoothA2dpService;->addAudioSink(Landroid/bluetooth/BluetoothDevice;)V
  
+     .line 250
      iget-object v2, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      invoke-virtual {p1}, Landroid/bluetooth/BluetoothDevice;->getAddress()Ljava/lang/String;
***************
*** 751,763 ****
--- 834,849 ----
  
      move-result-object v0
  
+     .line 251
      .local v0, path:Ljava/lang/String;
      if-eqz v0, :cond_0
  
+     .line 254
      const/4 v1, 0x1
  
      goto :goto_0
  
+     .line 243
      .end local v0           #path:Ljava/lang/String;
      :catchall_0
      move-exception v1
***************
*** 774,779 ****
--- 860,866 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 318
      monitor-enter p0
  
      :try_start_0
***************
*** 789,803 ****
--- 876,893 ----
  
      move-result-object v0
  
+     .line 319
      .local v0, path:Ljava/lang/String;
      if-nez v0, :cond_0
  
+     .line 329
      :goto_0
      :pswitch_0
      monitor-exit p0
  
      return v2
  
+     .line 323
      :cond_0
      :try_start_1
      invoke-virtual {p0, p1}, Landroid/server/BluetoothA2dpService;->getConnectionState(Landroid/bluetooth/BluetoothDevice;)I
***************
*** 806,819 ****
--- 896,912 ----
  
      move-result v1
  
+     .line 324
      .local v1, state:I
      packed-switch v1, :pswitch_data_0
  
+     .line 329
      :pswitch_1
      const/4 v2, 0x1
  
      goto :goto_0
  
+     .line 318
      .end local v0           #path:Ljava/lang/String;
      .end local v1           #state:I
      :catchall_0
***************
*** 823,828 ****
--- 916,922 ----
  
      throw v2
  
+     .line 324
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
***************
*** 839,844 ****
--- 933,939 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 118
      iget-object v4, p0, Landroid/server/BluetoothA2dpService;->mContext:Landroid/content/Context;
  
      const/4 v5, 0x0
***************
*** 853,870 ****
--- 948,969 ----
  
      move-result-object v1
  
+     .line 119
      .local v1, i:Landroid/content/Intent;
      if-eqz v1, :cond_0
  
+     .line 120
      const-string v4, "android.intent.extra.DOCK_STATE"
  
      invoke-virtual {v1, v4, v3}, Landroid/content/Intent;->getIntExtra(Ljava/lang/String;I)I
  
      move-result v2
  
+     .line 121
      .local v2, state:I
      if-eqz v2, :cond_0
  
+     .line 122
      const-string v4, "android.bluetooth.device.extra.DEVICE"
  
      invoke-virtual {v1, v4}, Landroid/content/Intent;->getParcelableExtra(Ljava/lang/String;)Landroid/os/Parcelable;
***************
*** 873,878 ****
--- 972,978 ----
  
      check-cast v0, Landroid/bluetooth/BluetoothDevice;
  
+     .line 123
      .local v0, dockDevice:Landroid/bluetooth/BluetoothDevice;
      if-eqz v0, :cond_0
  
***************
*** 882,889 ****
--- 982,991 ----
  
      if-eqz v4, :cond_0
  
+     .line 124
      const/4 v3, 0x1
  
+     .line 128
      .end local v0           #dockDevice:Landroid/bluetooth/BluetoothDevice;
      .end local v2           #state:I
      :cond_0
***************
*** 895,900 ****
--- 997,1003 ----
      .parameter "device"
  
      .prologue
+     .line 183
      iget-object v1, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      invoke-virtual {p1}, Landroid/bluetooth/BluetoothDevice;->getAddress()Ljava/lang/String;
***************
*** 905,910 ****
--- 1008,1014 ----
  
      move-result-object v0
  
+     .line 184
      .local v0, uuids:[Landroid/os/ParcelUuid;
      if-eqz v0, :cond_0
  
***************
*** 916,923 ****
--- 1020,1029 ----
  
      if-eqz v1, :cond_0
  
+     .line 185
      const/4 v1, 0x1
  
+     .line 187
      :goto_0
      return v1
  
***************
*** 932,941 ****
--- 1038,1049 ----
      .parameter "msg"
  
      .prologue
+     .line 600
      const-string v0, "BluetoothA2dpService"
  
      invoke-static {v0, p0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 601
      return-void
  .end method
  
***************
*** 943,948 ****
--- 1051,1057 ----
      .locals 8
  
      .prologue
+     .line 217
      monitor-enter p0
  
      :try_start_0
***************
*** 954,959 ****
--- 1063,1069 ----
  
      if-nez v6, :cond_1
  
+     .line 218
      iget-object v6, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      invoke-virtual {v6}, Ljava/util/HashMap;->size()I
***************
*** 962,967 ****
--- 1072,1078 ----
  
      new-array v2, v6, [Landroid/bluetooth/BluetoothDevice;
  
+     .line 219
      .local v2, devices:[Landroid/bluetooth/BluetoothDevice;
      iget-object v6, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
***************
*** 976,981 ****
--- 1087,1093 ----
      .end local v2           #devices:[Landroid/bluetooth/BluetoothDevice;
      check-cast v2, [Landroid/bluetooth/BluetoothDevice;
  
+     .line 220
      .restart local v2       #devices:[Landroid/bluetooth/BluetoothDevice;
      move-object v0, v2
  
***************
*** 991,1009 ****
--- 1103,1125 ----
  
      aget-object v1, v0, v3
  
+     .line 221
      .local v1, device:Landroid/bluetooth/BluetoothDevice;
      invoke-virtual {p0, v1}, Landroid/server/BluetoothA2dpService;->getConnectionState(Landroid/bluetooth/BluetoothDevice;)I
  
      move-result v5
  
+     .line 222
      .local v5, state:I
      sparse-switch v5, :sswitch_data_0
  
+     .line 220
      :goto_1
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 226
      :sswitch_0
      iget-object v6, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
***************
*** 1017,1022 ****
--- 1133,1139 ----
  
      invoke-direct {p0, v6}, Landroid/server/BluetoothA2dpService;->disconnectSinkNative(Ljava/lang/String;)Z
  
+     .line 228
      const/4 v6, 0x0
  
      invoke-direct {p0, v1, v5, v6}, Landroid/server/BluetoothA2dpService;->handleSinkStateChange(Landroid/bluetooth/BluetoothDevice;II)V
***************
*** 1025,1030 ****
--- 1142,1148 ----
  
      goto :goto_1
  
+     .line 217
      .end local v0           #arr$:[Landroid/bluetooth/BluetoothDevice;
      .end local v1           #device:Landroid/bluetooth/BluetoothDevice;
      .end local v2           #devices:[Landroid/bluetooth/BluetoothDevice;
***************
*** 1038,1043 ****
--- 1156,1162 ----
  
      throw v6
  
+     .line 231
      .restart local v0       #arr$:[Landroid/bluetooth/BluetoothDevice;
      .restart local v1       #device:Landroid/bluetooth/BluetoothDevice;
      .restart local v2       #devices:[Landroid/bluetooth/BluetoothDevice;
***************
*** 1054,1059 ****
--- 1173,1179 ----
  
      goto :goto_1
  
+     .line 236
      .end local v1           #device:Landroid/bluetooth/BluetoothDevice;
      .end local v5           #state:I
      :cond_0
***************
*** 1061,1066 ****
--- 1181,1187 ----
  
      invoke-virtual {v6}, Ljava/util/HashMap;->clear()V
  
+     .line 239
      .end local v0           #arr$:[Landroid/bluetooth/BluetoothDevice;
      .end local v2           #devices:[Landroid/bluetooth/BluetoothDevice;
      .end local v3           #i$:I
***************
*** 1074,1083 ****
--- 1195,1206 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 240
      monitor-exit p0
  
      return-void
  
+     .line 222
      nop
  
      :sswitch_data_0
***************
*** 1093,1098 ****
--- 1216,1222 ----
      .locals 12
  
      .prologue
+     .line 197
      monitor-enter p0
  
      :try_start_0
***************
*** 1106,1120 ****
--- 1230,1247 ----
  
      move-result-object v3
  
+     .line 198
      .local v3, devices:Ljava/lang/String;
      if-eqz v3, :cond_1
  
+     .line 199
      const-string v9, ","
  
      invoke-virtual {v3, v9}, Ljava/lang/String;->split(Ljava/lang/String;)[Ljava/lang/String;
  
      move-result-object v7
  
+     .line 200
      .local v7, paths:[Ljava/lang/String;
      move-object v1, v7
  
***************
*** 1130,1135 ****
--- 1257,1263 ----
  
      aget-object v6, v1, v4
  
+     .line 201
      .local v6, path:Ljava/lang/String;
      iget-object v9, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
***************
*** 1137,1142 ****
--- 1265,1271 ----
  
      move-result-object v0
  
+     .line 202
      .local v0, address:Ljava/lang/String;
      iget-object v9, p0, Landroid/server/BluetoothA2dpService;->mAdapter:Landroid/bluetooth/BluetoothAdapter;
  
***************
*** 1144,1149 ****
--- 1273,1279 ----
  
      move-result-object v2
  
+     .line 203
      .local v2, device:Landroid/bluetooth/BluetoothDevice;
      iget-object v9, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
***************
*** 1151,1159 ****
--- 1281,1291 ----
  
      move-result-object v8
  
+     .line 204
      .local v8, remoteUuids:[Landroid/os/ParcelUuid;
      if-eqz v8, :cond_0
  
+     .line 205
      const/4 v9, 0x2
  
      new-array v9, v9, [Landroid/os/ParcelUuid;
***************
*** 1176,1188 ****
--- 1308,1323 ----
  
      if-eqz v9, :cond_0
  
+     .line 208
      invoke-direct {p0, v2}, Landroid/server/BluetoothA2dpService;->addAudioSink(Landroid/bluetooth/BluetoothDevice;)V
  
+     .line 200
      :cond_0
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
  
+     .line 212
      .end local v0           #address:Ljava/lang/String;
      .end local v1           #arr$:[Ljava/lang/String;
      .end local v2           #device:Landroid/bluetooth/BluetoothDevice;
***************
*** 1198,1203 ****
--- 1333,1339 ----
  
      invoke-virtual {v9, v10}, Landroid/media/AudioManager;->setParameters(Ljava/lang/String;)V
  
+     .line 213
      iget-object v9, p0, Landroid/server/BluetoothA2dpService;->mAudioManager:Landroid/media/AudioManager;
  
      const-string v10, "A2dpSuspended=false"
***************
*** 1206,1215 ****
--- 1342,1353 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 214
      monitor-exit p0
  
      return-void
  
+     .line 197
      .end local v3           #devices:Ljava/lang/String;
      :catchall_0
      move-exception v9
***************
*** 1225,1248 ****
--- 1363,1392 ----
      .parameter "result"
  
      .prologue
+     .line 578
      if-nez p2, :cond_0
  
+     .line 579
      if-eqz p1, :cond_0
  
+     .line 580
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      invoke-virtual {v3, p1}, Landroid/server/BluetoothService;->getAddressFromObjectPath(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 581
      .local v0, address:Ljava/lang/String;
      if-nez v0, :cond_1
  
+     .line 587
      .end local v0           #address:Ljava/lang/String;
      :cond_0
      :goto_0
      return-void
  
+     .line 582
      .restart local v0       #address:Ljava/lang/String;
      :cond_1
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mAdapter:Landroid/bluetooth/BluetoothAdapter;
***************
*** 1251,1261 ****
--- 1395,1407 ----
  
      move-result-object v1
  
+     .line 583
      .local v1, device:Landroid/bluetooth/BluetoothDevice;
      invoke-virtual {p0, v1}, Landroid/server/BluetoothA2dpService;->getConnectionState(Landroid/bluetooth/BluetoothDevice;)I
  
      move-result v2
  
+     .line 584
      .local v2, state:I
      const/4 v3, 0x0
  
***************
*** 1272,1277 ****
--- 1418,1424 ----
      .prologue
      const/16 v7, 0xa
  
+     .line 461
      monitor-enter p0
  
      :try_start_0
***************
*** 1285,1302 ****
--- 1432,1452 ----
  
      if-nez v5, :cond_1
  
+     .line 498
      :cond_0
      :goto_0
      monitor-exit p0
  
      return-void
  
+     .line 465
      :cond_1
      const/4 v5, 0x0
  
      :try_start_1
      aget-object v2, p2, v5
  
+     .line 466
      .local v2, name:Ljava/lang/String;
      iget-object v5, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
***************
*** 1304,1315 ****
  
      move-result-object v0
  
      .local v0, address:Ljava/lang/String;
      if-nez v0, :cond_2
  
      const-string v5, "BluetoothA2dpService"
  
!     const-string v6, "onSinkPropertyChanged: Address of the remote device in null"
  
      invoke-static {v5, v6}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
      :try_end_1
--- 1454,1467 ----
  
      move-result-object v0
  
+     .line 467
      .local v0, address:Ljava/lang/String;
      if-nez v0, :cond_2
  
+     .line 468
      const-string v5, "BluetoothA2dpService"
  
!     const-string/jumbo v6, "onSinkPropertyChanged: Address of the remote device in null"
  
      invoke-static {v5, v6}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
      :try_end_1
***************
*** 1317,1322 ****
--- 1469,1475 ----
  
      goto :goto_0
  
+     .line 461
      .end local v0           #address:Ljava/lang/String;
      .end local v2           #name:Ljava/lang/String;
      :catchall_0
***************
*** 1326,1331 ****
--- 1479,1485 ----
  
      throw v5
  
+     .line 472
      .restart local v0       #address:Ljava/lang/String;
      .restart local v2       #name:Ljava/lang/String;
      :cond_2
***************
*** 1336,1341 ****
--- 1490,1496 ----
  
      move-result-object v1
  
+     .line 474
      .local v1, device:Landroid/bluetooth/BluetoothDevice;
      const-string v5, "State"
  
***************
*** 1345,1350 ****
--- 1500,1506 ----
  
      if-eqz v5, :cond_0
  
+     .line 475
      const/4 v5, 0x1
  
      aget-object v5, p2, v5
***************
*** 1353,1358 ****
--- 1509,1515 ----
  
      move-result v4
  
+     .line 476
      .local v4, state:I
      new-instance v5, Ljava/lang/StringBuilder;
  
***************
*** 1386,1391 ****
--- 1543,1549 ----
  
      invoke-static {v5}, Landroid/server/BluetoothA2dpService;->log(Ljava/lang/String;)V
  
+     .line 478
      iget-object v5, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      invoke-virtual {v5, v1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1394,1407 ****
--- 1552,1568 ----
  
      if-nez v5, :cond_3
  
+     .line 481
      invoke-direct {p0, v1}, Landroid/server/BluetoothA2dpService;->addAudioSink(Landroid/bluetooth/BluetoothDevice;)V
  
+     .line 482
      const/4 v5, 0x0
  
      invoke-direct {p0, v1, v5, v4}, Landroid/server/BluetoothA2dpService;->handleSinkStateChange(Landroid/bluetooth/BluetoothDevice;II)V
  
      goto :goto_0
  
+     .line 484
      :cond_3
      if-ne v4, v7, :cond_4
  
***************
*** 1409,1422 ****
--- 1570,1586 ----
  
      if-nez v5, :cond_4
  
+     .line 485
      iput-object v1, p0, Landroid/server/BluetoothA2dpService;->mPlayingA2dpDevice:Landroid/bluetooth/BluetoothDevice;
  
+     .line 486
      const/16 v5, 0xb
  
      invoke-direct {p0, v1, v4, v5}, Landroid/server/BluetoothA2dpService;->handleSinkPlayingStateChange(Landroid/bluetooth/BluetoothDevice;II)V
  
      goto :goto_0
  
+     .line 487
      :cond_4
      const/4 v5, 0x2
  
***************
*** 1426,1435 ****
--- 1590,1601 ----
  
      if-eqz v5, :cond_5
  
+     .line 488
      const/4 v5, 0x0
  
      iput-object v5, p0, Landroid/server/BluetoothA2dpService;->mPlayingA2dpDevice:Landroid/bluetooth/BluetoothDevice;
  
+     .line 489
      const/16 v5, 0xb
  
      const/16 v6, 0xa
***************
*** 1438,1448 ****
--- 1604,1616 ----
  
      goto :goto_0
  
+     .line 492
      :cond_5
      const/4 v5, 0x0
  
      iput-object v5, p0, Landroid/server/BluetoothA2dpService;->mPlayingA2dpDevice:Landroid/bluetooth/BluetoothDevice;
  
+     .line 493
      iget-object v5, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      invoke-virtual {v5, v1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1455,1460 ****
--- 1623,1629 ----
  
      move-result v3
  
+     .line 494
      .local v3, prevState:I
      invoke-direct {p0, v1, v3, v4}, Landroid/server/BluetoothA2dpService;->handleSinkStateChange(Landroid/bluetooth/BluetoothDevice;II)V
      :try_end_2
***************
*** 1479,1484 ****
--- 1648,1654 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 437
      monitor-enter p0
  
      :try_start_0
***************
*** 1490,1499 ****
--- 1660,1671 ----
  
      invoke-virtual {v3, v4, v5}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 439
      invoke-virtual {p1}, Landroid/bluetooth/BluetoothDevice;->getAddress()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 440
      .local v0, address:Ljava/lang/String;
      invoke-static {v0}, Landroid/bluetooth/BluetoothAdapter;->checkBluetoothAddress(Ljava/lang/String;)Z
      :try_end_0
***************
*** 1503,1513 ****
--- 1675,1687 ----
  
      if-nez v3, :cond_0
  
+     .line 449
      :goto_0
      monitor-exit p0
  
      return v2
  
+     .line 443
      :cond_0
      :try_start_1
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
***************
*** 1516,1524 ****
--- 1690,1700 ----
  
      move-result-object v1
  
+     .line 444
      .local v1, data:Ljava/lang/Integer;
      if-nez v1, :cond_1
  
+     .line 445
      const-string v3, "BluetoothA2dpService"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 1551,1556 ****
--- 1727,1733 ----
  
      goto :goto_0
  
+     .line 437
      .end local v0           #address:Ljava/lang/String;
      .end local v1           #data:Ljava/lang/Integer;
      :catchall_0
***************
*** 1560,1565 ****
--- 1737,1743 ----
  
      throw v2
  
+     .line 448
      .restart local v0       #address:Ljava/lang/String;
      .restart local v1       #data:Ljava/lang/Integer;
      :cond_1
***************
*** 1594,1599 ****
--- 1772,1778 ----
  
      invoke-static {v2}, Landroid/server/BluetoothA2dpService;->log(Ljava/lang/String;)V
  
+     .line 449
      iget-object v2, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      invoke-virtual {v1}, Ljava/lang/Integer;->intValue()I
***************
*** 1614,1619 ****
--- 1793,1799 ----
      .parameter "device"
  
      .prologue
+     .line 266
      monitor-enter p0
  
      :try_start_0
***************
*** 1625,1630 ****
--- 1805,1811 ----
  
      invoke-virtual {v2, v3, v4}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 268
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1651,1656 ****
--- 1832,1838 ----
  
      invoke-static {v2}, Landroid/server/BluetoothA2dpService;->log(Ljava/lang/String;)V
  
+     .line 269
      invoke-direct {p0, p1}, Landroid/server/BluetoothA2dpService;->isConnectSinkFeasible(Landroid/bluetooth/BluetoothDevice;)Z
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1661,1671 ****
--- 1843,1855 ----
  
      const/4 v2, 0x0
  
+     .line 277
      :goto_0
      monitor-exit p0
  
      return v2
  
+     .line 271
      :cond_0
      :try_start_1
      iget-object v2, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
***************
*** 1693,1698 ****
--- 1877,1883 ----
  
      check-cast v1, Landroid/bluetooth/BluetoothDevice;
  
+     .line 272
      .local v1, sinkDevice:Landroid/bluetooth/BluetoothDevice;
      invoke-virtual {p0, v1}, Landroid/server/BluetoothA2dpService;->getConnectionState(Landroid/bluetooth/BluetoothDevice;)I
  
***************
*** 1700,1711 ****
--- 1885,1898 ----
  
      if-eqz v2, :cond_1
  
+     .line 273
      invoke-virtual {p0, v1}, Landroid/server/BluetoothA2dpService;->disconnect(Landroid/bluetooth/BluetoothDevice;)Z
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
      goto :goto_1
  
+     .line 266
      .end local v0           #i$:Ljava/util/Iterator;
      .end local v1           #sinkDevice:Landroid/bluetooth/BluetoothDevice;
      :catchall_0
***************
*** 1715,1720 ****
--- 1902,1908 ----
  
      throw v2
  
+     .line 277
      .restart local v0       #i$:Ljava/util/Iterator;
      :cond_2
      :try_start_2
***************
*** 1744,1749 ****
--- 1932,1938 ----
  
      const/4 v3, 0x0
  
+     .line 281
      monitor-enter p0
  
      :try_start_0
***************
*** 1759,1769 ****
--- 1948,1960 ----
  
      move v2, v3
  
+     .line 314
      :goto_0
      monitor-exit p0
  
      return v2
  
+     .line 283
      :cond_0
      :try_start_1
      iget-object v2, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
***************
*** 1778,1783 ****
--- 1969,1975 ----
  
      move-result v1
  
+     .line 286
      .local v1, state:I
      const/4 v2, 0x3
  
***************
*** 1797,1807 ****
--- 1989,2002 ----
  
      move v2, v3
  
+     .line 290
      goto :goto_0
  
+     .line 293
      :cond_1
      packed-switch v1, :pswitch_data_0
  
+     .line 301
      iget-object v2, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
      invoke-virtual {p1}, Landroid/bluetooth/BluetoothDevice;->getAddress()Ljava/lang/String;
***************
*** 1812,1817 ****
--- 2007,2013 ----
  
      move-result-object v0
  
+     .line 304
      .local v0, path:Ljava/lang/String;
      invoke-virtual {p0, p1}, Landroid/server/BluetoothA2dpService;->getPriority(Landroid/bluetooth/BluetoothDevice;)I
  
***************
*** 1819,1839 ****
--- 2015,2039 ----
  
      if-ge v2, v6, :cond_2
  
+     .line 305
      const/16 v2, 0x3e8
  
      invoke-virtual {p0, p1, v2}, Landroid/server/BluetoothA2dpService;->setPriority(Landroid/bluetooth/BluetoothDevice;I)Z
  
+     .line 307
      :cond_2
      const/4 v2, 0x1
  
      invoke-direct {p0, p1, v1, v2}, Landroid/server/BluetoothA2dpService;->handleSinkStateChange(Landroid/bluetooth/BluetoothDevice;II)V
  
+     .line 309
      invoke-direct {p0, v0}, Landroid/server/BluetoothA2dpService;->connectSinkNative(Ljava/lang/String;)Z
  
      move-result v2
  
      if-nez v2, :cond_3
  
+     .line 311
      iget-object v2, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      invoke-virtual {v2, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1852,1876 ****
--- 2052,2081 ----
  
      move v2, v3
  
+     .line 312
      goto :goto_0
  
      .end local v0           #path:Ljava/lang/String;
      :pswitch_0
      move v2, v3
  
+     .line 296
      goto :goto_0
  
      :pswitch_1
      move v2, v4
  
+     .line 298
      goto :goto_0
  
      .restart local v0       #path:Ljava/lang/String;
      :cond_3
      move v2, v4
  
+     .line 314
      goto :goto_0
  
+     .line 281
      .end local v0           #path:Ljava/lang/String;
      .end local v1           #state:I
      :catchall_0
***************
*** 1880,1885 ****
--- 2085,2091 ----
  
      throw v2
  
+     .line 286
      :array_0
      .array-data 0x4
          0x1t 0x0t 0x0t 0x0t
***************
*** 1887,1892 ****
--- 2093,2099 ----
          0x3t 0x0t 0x0t 0x0t
      .end array-data
  
+     .line 293
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_1
***************
*** 1900,1905 ****
--- 2107,2113 ----
      .parameter "device"
  
      .prologue
+     .line 333
      monitor-enter p0
  
      :try_start_0
***************
*** 1911,1916 ****
--- 2119,2125 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 335
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1937,1942 ****
--- 2146,2152 ----
  
      invoke-static {v0}, Landroid/server/BluetoothA2dpService;->log(Ljava/lang/String;)V
  
+     .line 336
      invoke-direct {p0, p1}, Landroid/server/BluetoothA2dpService;->isDisconnectSinkFeasible(Landroid/bluetooth/BluetoothDevice;)Z
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1947,1952 ****
--- 2157,2163 ----
  
      const/4 v0, 0x0
  
+     .line 337
      :goto_0
      monitor-exit p0
  
***************
*** 1968,1973 ****
--- 2179,2185 ----
  
      goto :goto_0
  
+     .line 333
      :catchall_0
      move-exception v0
  
***************
*** 1983,1988 ****
--- 2195,2201 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 341
      monitor-enter p0
  
      :try_start_0
***************
*** 1990,1995 ****
--- 2203,2209 ----
  
      move-result v1
  
+     .line 342
      .local v1, state:I
      iget-object v2, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
  
***************
*** 2001,2020 ****
--- 2215,2238 ----
  
      move-result-object v0
  
+     .line 344
      .local v0, path:Ljava/lang/String;
      packed-switch v1, :pswitch_data_0
  
+     .line 350
      :pswitch_0
      const/4 v2, 0x3
  
      invoke-direct {p0, p1, v1, v2}, Landroid/server/BluetoothA2dpService;->handleSinkStateChange(Landroid/bluetooth/BluetoothDevice;II)V
  
+     .line 351
      invoke-direct {p0, v0}, Landroid/server/BluetoothA2dpService;->disconnectSinkNative(Ljava/lang/String;)Z
  
      move-result v2
  
      if-nez v2, :cond_0
  
+     .line 353
      iget-object v2, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      invoke-virtual {v2, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 2033,2038 ****
--- 2251,2257 ----
  
      move v2, v3
  
+     .line 356
      :goto_0
      monitor-exit p0
  
***************
*** 2041,2053 ****
--- 2260,2275 ----
      :pswitch_1
      move v2, v3
  
+     .line 347
      goto :goto_0
  
+     .line 356
      :cond_0
      const/4 v2, 0x1
  
      goto :goto_0
  
+     .line 341
      .end local v0           #path:Ljava/lang/String;
      .end local v1           #state:I
      :catchall_0
***************
*** 2057,2062 ****
--- 2279,2285 ----
  
      throw v2
  
+     .line 344
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_1
***************
*** 2073,2078 ****
--- 2296,2302 ----
      .parameter "args"
  
      .prologue
+     .line 591
      monitor-enter p0
  
      :try_start_0
***************
*** 2086,2102 ****
--- 2310,2329 ----
  
      if-eqz v3, :cond_1
  
+     .line 597
      :cond_0
      monitor-exit p0
  
      return-void
  
+     .line 592
      :cond_1
      :try_start_1
      const-string v3, "Cached audio devices:"
  
      invoke-virtual {p2, v3}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 593
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      invoke-virtual {v3}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 2121,2126 ****
--- 2348,2354 ----
  
      check-cast v0, Landroid/bluetooth/BluetoothDevice;
  
+     .line 594
      .local v0, device:Landroid/bluetooth/BluetoothDevice;
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
***************
*** 2134,2139 ****
--- 2362,2368 ----
  
      move-result v2
  
+     .line 595
      .local v2, state:I
      new-instance v3, Ljava/lang/StringBuilder;
  
***************
*** 2167,2172 ****
--- 2396,2402 ----
  
      goto :goto_0
  
+     .line 591
      .end local v0           #device:Landroid/bluetooth/BluetoothDevice;
      .end local v1           #i$:Ljava/util/Iterator;
      .end local v2           #state:I
***************
*** 2187,2201 ****
--- 2417,2435 ----
      .end annotation
  
      .prologue
+     .line 164
      :try_start_0
      invoke-direct {p0}, Landroid/server/BluetoothA2dpService;->cleanupNative()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 166
      invoke-super {p0}, Landroid/bluetooth/IBluetoothA2dp$Stub;->finalize()V
  
+     .line 168
      return-void
  
+     .line 166
      :catchall_0
      move-exception v0
  
***************
*** 2217,2222 ****
--- 2451,2457 ----
      .end annotation
  
      .prologue
+     .line 401
      monitor-enter p0
  
      :try_start_0
***************
*** 2228,2233 ****
--- 2463,2469 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 402
      const/4 v1, 0x1
  
      new-array v1, v1, [I
***************
*** 2244,2254 ****
--- 2480,2492 ----
  
      move-result-object v0
  
+     .line 404
      .local v0, sinks:Ljava/util/List;,"Ljava/util/List<Landroid/bluetooth/BluetoothDevice;>;"
      monitor-exit p0
  
      return-object v0
  
+     .line 401
      .end local v0           #sinks:Ljava/util/List;,"Ljava/util/List<Landroid/bluetooth/BluetoothDevice;>;"
      :catchall_0
      move-exception v1
***************
*** 2263,2268 ****
--- 2501,2507 ----
      .parameter "device"
  
      .prologue
+     .line 393
      monitor-enter p0
  
      :try_start_0
***************
*** 2274,2279 ****
--- 2513,2519 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 394
      iget-object v1, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
      invoke-virtual {v1, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 2284,2294 ****
--- 2524,2537 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 395
      .local v0, state:Ljava/lang/Integer;
      if-nez v0, :cond_0
  
+     .line 396
      const/4 v1, 0x0
  
+     .line 397
      :goto_0
      monitor-exit p0
  
***************
*** 2304,2309 ****
--- 2547,2553 ----
  
      goto :goto_0
  
+     .line 393
      .end local v0           #state:Ljava/lang/Integer;
      :catchall_0
      move-exception v1
***************
*** 2327,2332 ****
--- 2571,2577 ----
      .end annotation
  
      .prologue
+     .line 408
      monitor-enter p0
  
      :try_start_0
***************
*** 2338,2347 ****
--- 2583,2594 ----
  
      invoke-virtual {v8, v9, v10}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 409
      new-instance v6, Ljava/util/ArrayList;
  
      invoke-direct {v6}, Ljava/util/ArrayList;-><init>()V
  
+     .line 410
      .local v6, sinks:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/bluetooth/BluetoothDevice;>;"
      iget-object v8, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
***************
*** 2367,2377 ****
--- 2614,2626 ----
  
      check-cast v1, Landroid/bluetooth/BluetoothDevice;
  
+     .line 411
      .local v1, device:Landroid/bluetooth/BluetoothDevice;
      invoke-virtual {p0, v1}, Landroid/server/BluetoothA2dpService;->getConnectionState(Landroid/bluetooth/BluetoothDevice;)I
  
      move-result v5
  
+     .line 412
      .local v5, sinkState:I
      move-object v0, p1
  
***************
*** 2387,2401 ****
--- 2636,2653 ----
  
      aget v7, v0, v3
  
+     .line 413
      .local v7, state:I
      if-ne v7, v5, :cond_1
  
+     .line 414
      invoke-virtual {v6, v1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      goto :goto_0
  
+     .line 408
      .end local v0           #arr$:[I
      .end local v1           #device:Landroid/bluetooth/BluetoothDevice;
      .end local v3           #i$:I
***************
*** 2410,2415 ****
--- 2662,2668 ----
  
      throw v8
  
+     .line 412
      .restart local v0       #arr$:[I
      .restart local v1       #device:Landroid/bluetooth/BluetoothDevice;
      .restart local v3       #i$:I
***************
*** 2422,2427 ****
--- 2675,2681 ----
  
      goto :goto_1
  
+     .line 419
      .end local v0           #arr$:[I
      .end local v1           #device:Landroid/bluetooth/BluetoothDevice;
      .end local v3           #i$:I
***************
*** 2439,2444 ****
--- 2693,2699 ----
      .parameter "device"
  
      .prologue
+     .line 423
      monitor-enter p0
  
      :try_start_0
***************
*** 2450,2455 ****
--- 2705,2711 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 424
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
***************
*** 2476,2481 ****
--- 2732,2738 ----
  
      return v0
  
+     .line 423
      :catchall_0
      move-exception v0
  
***************
*** 2489,2494 ****
--- 2746,2752 ----
      .parameter "device"
  
      .prologue
+     .line 258
      monitor-enter p0
  
      :try_start_0
***************
*** 2500,2505 ****
--- 2758,2764 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 260
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2526,2531 ****
--- 2785,2791 ----
  
      invoke-static {v0}, Landroid/server/BluetoothA2dpService;->log(Ljava/lang/String;)V
  
+     .line 261
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mPlayingA2dpDevice:Landroid/bluetooth/BluetoothDevice;
  
      invoke-virtual {p1, v0}, Landroid/bluetooth/BluetoothDevice;->equals(Ljava/lang/Object;)Z
***************
*** 2538,2543 ****
--- 2798,2804 ----
  
      const/4 v0, 0x1
  
+     .line 262
      :goto_0
      monitor-exit p0
  
***************
*** 2548,2553 ****
--- 2809,2815 ----
  
      goto :goto_0
  
+     .line 258
      :catchall_0
      move-exception v0
  
***************
*** 2563,2568 ****
--- 2825,2831 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 377
      monitor-enter p0
  
      :try_start_0
***************
*** 2574,2584 ****
  
      invoke-virtual {v3, v4, v5}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "resumeSink("
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2837,2848 ----
  
      invoke-virtual {v3, v4, v5}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 379
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "resumeSink("
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 2606,2611 ****
--- 2870,2876 ----
  
      invoke-static {v3}, Landroid/server/BluetoothA2dpService;->log(Ljava/lang/String;)V
  
+     .line 380
      if-eqz p1, :cond_0
  
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
***************
*** 2614,2625 ****
--- 2879,2892 ----
  
      if-nez v3, :cond_1
  
+     .line 389
      :cond_0
      :goto_0
      monitor-exit p0
  
      return v2
  
+     .line 383
      :cond_1
      :try_start_1
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
***************
*** 2632,2637 ****
--- 2899,2905 ----
  
      move-result-object v0
  
+     .line 384
      .local v0, path:Ljava/lang/String;
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
***************
*** 2641,2655 ****
--- 2909,2926 ----
  
      check-cast v1, Ljava/lang/Integer;
  
+     .line 385
      .local v1, state:Ljava/lang/Integer;
      if-eqz v0, :cond_0
  
      if-eqz v1, :cond_0
  
+     .line 388
      const/16 v2, 0xa
  
      iput v2, p0, Landroid/server/BluetoothA2dpService;->mTargetA2dpState:I
  
+     .line 389
      invoke-virtual {v1}, Ljava/lang/Integer;->intValue()I
  
      move-result v2
***************
*** 2662,2667 ****
--- 2933,2939 ----
  
      goto :goto_0
  
+     .line 377
      .end local v0           #path:Ljava/lang/String;
      .end local v1           #state:Ljava/lang/Integer;
      :catchall_0
***************
*** 2678,2683 ****
--- 2950,2956 ----
      .parameter "priority"
  
      .prologue
+     .line 430
      monitor-enter p0
  
      :try_start_0
***************
*** 2689,2694 ****
--- 2962,2968 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 432
      iget-object v0, p0, Landroid/server/BluetoothA2dpService;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
***************
*** 2713,2718 ****
--- 2987,2993 ----
  
      return v0
  
+     .line 430
      :catchall_0
      move-exception v0
  
***************
*** 2728,2733 ****
--- 3003,3009 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 360
      monitor-enter p0
  
      :try_start_0
***************
*** 2739,2749 ****
  
      invoke-virtual {v3, v4, v5}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "suspendSink("
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 3015,3026 ----
  
      invoke-virtual {v3, v4, v5}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 362
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "suspendSink("
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 2771,2776 ****
--- 3048,3054 ----
  
      invoke-static {v3}, Landroid/server/BluetoothA2dpService;->log(Ljava/lang/String;)V
  
+     .line 363
      if-eqz p1, :cond_0
  
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
***************
*** 2779,2790 ****
--- 3057,3070 ----
  
      if-nez v3, :cond_1
  
+     .line 373
      :cond_0
      :goto_0
      monitor-exit p0
  
      return v2
  
+     .line 366
      :cond_1
      :try_start_1
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mBluetoothService:Landroid/server/BluetoothService;
***************
*** 2797,2802 ****
--- 3077,3083 ----
  
      move-result-object v0
  
+     .line 367
      .local v0, path:Ljava/lang/String;
      iget-object v3, p0, Landroid/server/BluetoothA2dpService;->mAudioDevices:Ljava/util/HashMap;
  
***************
*** 2806,2820 ****
--- 3087,3104 ----
  
      check-cast v1, Ljava/lang/Integer;
  
+     .line 368
      .local v1, state:Ljava/lang/Integer;
      if-eqz v0, :cond_0
  
      if-eqz v1, :cond_0
  
+     .line 372
      const/4 v2, 0x2
  
      iput v2, p0, Landroid/server/BluetoothA2dpService;->mTargetA2dpState:I
  
+     .line 373
      invoke-virtual {v1}, Ljava/lang/Integer;->intValue()I
  
      move-result v2
***************
*** 2827,2832 ****
--- 3111,3117 ----
  
      goto :goto_0
  
+     .line 360
      .end local v0           #path:Ljava/lang/String;
      .end local v1           #state:Ljava/lang/Integer;
      :catchall_0
