*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothService$3.smali	2012-03-23 19:37:15.690061190 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothService$3.smali.original	2012-03-23 19:34:45.606324417 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1950
      iput-object p1, p0, Landroid/server/BluetoothService$3;->this$0:Landroid/server/BluetoothService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,51 ****
--- 43,54 ----
      .parameter "proxy"
  
      .prologue
+     .line 1952
      const/4 v0, 0x1
  
      if-ne p1, v0, :cond_1
  
+     .line 1953
      iget-object v0, p0, Landroid/server/BluetoothService$3;->this$0:Landroid/server/BluetoothService;
  
      check-cast p2, Landroid/bluetooth/BluetoothHeadset;
***************
*** 54,69 ****
--- 57,75 ----
      #setter for: Landroid/server/BluetoothService;->mHeadsetProxy:Landroid/bluetooth/BluetoothHeadset;
      invoke-static {v0, p2}, Landroid/server/BluetoothService;->access$902(Landroid/server/BluetoothService;Landroid/bluetooth/BluetoothHeadset;)Landroid/bluetooth/BluetoothHeadset;
  
+     .line 1959
      :cond_0
      :goto_0
      return-void
  
+     .line 1954
      .restart local p2
      :cond_1
      const/4 v0, 0x4
  
      if-ne p1, v0, :cond_2
  
+     .line 1955
      iget-object v0, p0, Landroid/server/BluetoothService$3;->this$0:Landroid/server/BluetoothService;
  
      check-cast p2, Landroid/bluetooth/BluetoothInputDevice;
***************
*** 74,85 ****
--- 80,93 ----
  
      goto :goto_0
  
+     .line 1956
      .restart local p2
      :cond_2
      const/4 v0, 0x5
  
      if-ne p1, v0, :cond_0
  
+     .line 1957
      iget-object v0, p0, Landroid/server/BluetoothService$3;->this$0:Landroid/server/BluetoothService;
  
      check-cast p2, Landroid/bluetooth/BluetoothPan;
***************
*** 98,121 ****
--- 106,134 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 1961
      const/4 v0, 0x1
  
      if-ne p1, v0, :cond_1
  
+     .line 1962
      iget-object v0, p0, Landroid/server/BluetoothService$3;->this$0:Landroid/server/BluetoothService;
  
      #setter for: Landroid/server/BluetoothService;->mHeadsetProxy:Landroid/bluetooth/BluetoothHeadset;
      invoke-static {v0, v1}, Landroid/server/BluetoothService;->access$902(Landroid/server/BluetoothService;Landroid/bluetooth/BluetoothHeadset;)Landroid/bluetooth/BluetoothHeadset;
  
+     .line 1968
      :cond_0
      :goto_0
      return-void
  
+     .line 1963
      :cond_1
      const/4 v0, 0x4
  
      if-ne p1, v0, :cond_2
  
+     .line 1964
      iget-object v0, p0, Landroid/server/BluetoothService$3;->this$0:Landroid/server/BluetoothService;
  
      #setter for: Landroid/server/BluetoothService;->mInputDevice:Landroid/bluetooth/BluetoothInputDevice;
***************
*** 123,133 ****
--- 136,148 ----
  
      goto :goto_0
  
+     .line 1965
      :cond_2
      const/4 v0, 0x5
  
      if-ne p1, v0, :cond_0
  
+     .line 1966
      iget-object v0, p0, Landroid/server/BluetoothService$3;->this$0:Landroid/server/BluetoothService;
  
      #setter for: Landroid/server/BluetoothService;->mPan:Landroid/bluetooth/BluetoothPan;
