*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothPanProfileHandler$1.smali	2012-03-23 19:37:14.757044176 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothPanProfileHandler$1.smali.original	2012-03-23 19:34:45.598324271 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 106
      iput-object p1, p0, Landroid/server/BluetoothPanProfileHandler$1;->this$0:Landroid/server/BluetoothPanProfileHandler;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 39,44 ****
--- 40,46 ----
      .parameter "intent"
  
      .prologue
+     .line 109
      const-string v0, "android.bluetooth.adapter.extra.STATE"
  
      const/16 v1, 0xa
***************
*** 51,56 ****
--- 53,59 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 111
      iget-object v0, p0, Landroid/server/BluetoothPanProfileHandler$1;->this$0:Landroid/server/BluetoothPanProfileHandler;
  
      const/4 v1, 0x1
***************
*** 58,63 ****
--- 61,67 ----
      #setter for: Landroid/server/BluetoothPanProfileHandler;->mTetheringOn:Z
      invoke-static {v0, v1}, Landroid/server/BluetoothPanProfileHandler;->access$002(Landroid/server/BluetoothPanProfileHandler;Z)Z
  
+     .line 112
      iget-object v0, p0, Landroid/server/BluetoothPanProfileHandler$1;->this$0:Landroid/server/BluetoothPanProfileHandler;
  
      #getter for: Landroid/server/BluetoothPanProfileHandler;->mContext:Landroid/content/Context;
***************
*** 74,79 ****
--- 78,84 ----
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  
+     .line 114
      :cond_0
      return-void
  .end method
