*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothAdapterProperties.smali	2012-03-23 19:37:14.808045108 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothAdapterProperties.smali.original	2012-03-23 19:34:45.599324289 -0500
***************
*** 32,49 ****
--- 32,54 ----
      .parameter "service"
  
      .prologue
+     .line 33
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 34
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Landroid/server/BluetoothAdapterProperties;->mPropertiesMap:Ljava/util/Map;
  
+     .line 35
      iput-object p1, p0, Landroid/server/BluetoothAdapterProperties;->mContext:Landroid/content/Context;
  
+     .line 36
      iput-object p2, p0, Landroid/server/BluetoothAdapterProperties;->mService:Landroid/server/BluetoothService;
  
+     .line 37
      return-void
  .end method
  
***************
*** 53,58 ****
--- 58,64 ----
      .locals 1
  
      .prologue
+     .line 51
      monitor-enter p0
  
      :try_start_0
***************
*** 62,71 ****
--- 68,79 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 52
      monitor-exit p0
  
      return-void
  
+     .line 51
      :catchall_0
      move-exception v0
  
***************
*** 78,83 ****
--- 86,92 ----
      .locals 13
  
      .prologue
+     .line 63
      monitor-enter p0
  
      :try_start_0
***************
*** 89,98 ****
--- 98,109 ----
  
      invoke-virtual {v9, v10, v11}, Landroid/content/Context;->enforceCallingOrSelfPermission(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 66
      iget-object v9, p0, Landroid/server/BluetoothAdapterProperties;->mPropertiesMap:Ljava/util/Map;
  
      invoke-interface {v9}, Ljava/util/Map;->clear()V
  
+     .line 68
      iget-object v9, p0, Landroid/server/BluetoothAdapterProperties;->mService:Landroid/server/BluetoothService;
  
      invoke-virtual {v9}, Landroid/server/BluetoothService;->getAdapterPropertiesNative()[Ljava/lang/Object;
***************
*** 107,115 ****
--- 118,128 ----
  
      move-object v7, v0
  
+     .line 71
      .local v7, properties:[Ljava/lang/String;
      if-nez v7, :cond_1
  
+     .line 72
      const-string v9, "BluetoothAdapterProperties"
  
      const-string v10, "*Error*: GetAdapterProperties returned NULL"
***************
*** 118,129 ****
--- 131,144 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 104
      :cond_0
      :goto_0
      monitor-exit p0
  
      return-void
  
+     .line 76
      :cond_1
      const/4 v2, 0x0
  
***************
*** 134,147 ****
--- 149,166 ----
  
      if-ge v2, v9, :cond_7
  
+     .line 77
      aget-object v5, v7, v2
  
+     .line 78
      .local v5, name:Ljava/lang/String;
      const/4 v6, 0x0
  
+     .line 79
      .local v6, newValue:Ljava/lang/String;
      if-nez v5, :cond_2
  
+     .line 80
      const-string v9, "BluetoothAdapterProperties"
  
      new-instance v10, Ljava/lang/StringBuilder;
***************
*** 170,180 ****
--- 189,201 ----
  
      invoke-static {v9, v10}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 76
      :goto_2
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_1
  
+     .line 83
      :cond_2
      const-string v9, "Devices"
  
***************
*** 192,202 ****
--- 213,225 ----
  
      if-eqz v9, :cond_6
  
+     .line 84
      :cond_3
      new-instance v8, Ljava/lang/StringBuilder;
  
      invoke-direct {v8}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 85
      .local v8, str:Ljava/lang/StringBuilder;
      add-int/lit8 v2, v2, 0x1
  
***************
*** 210,215 ****
--- 233,239 ----
  
      move-result v4
  
+     .line 86
      .local v4, len:I
      const/4 v3, 0x0
  
***************
*** 217,243 ****
--- 241,273 ----
      :goto_3
      if-ge v3, v4, :cond_4
  
+     .line 87
      add-int/lit8 v2, v2, 0x1
  
      aget-object v9, v7, v2
  
      invoke-virtual {v8, v9}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 88
      const-string v9, ","
  
      invoke-virtual {v8, v9}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 86
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_3
  
+     .line 90
      :cond_4
      if-lez v4, :cond_5
  
+     .line 91
      invoke-virtual {v8}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v6
  
+     .line 96
      .end local v3           #j:I
      .end local v4           #len:I
      .end local v8           #str:Ljava/lang/StringBuilder;
***************
*** 251,256 ****
--- 281,287 ----
  
      goto :goto_2
  
+     .line 63
      .end local v2           #i:I
      .end local v5           #name:Ljava/lang/String;
      .end local v6           #newValue:Ljava/lang/String;
***************
*** 262,267 ****
--- 293,299 ----
  
      throw v9
  
+     .line 94
      .restart local v2       #i:I
      .restart local v5       #name:Ljava/lang/String;
      .restart local v6       #newValue:Ljava/lang/String;
***************
*** 274,279 ****
--- 306,312 ----
  
      goto :goto_4
  
+     .line 100
      .end local v5           #name:Ljava/lang/String;
      .end local v6           #newValue:Ljava/lang/String;
      :cond_7
***************
*** 283,291 ****
--- 316,326 ----
  
      move-result-object v1
  
+     .line 101
      .local v1, adapterPath:Ljava/lang/String;
      if-eqz v1, :cond_0
  
+     .line 102
      iget-object v9, p0, Landroid/server/BluetoothAdapterProperties;->mPropertiesMap:Ljava/util/Map;
  
      const-string v10, "ObjectPath"
***************
*** 319,324 ****
--- 354,360 ----
      .locals 1
  
      .prologue
+     .line 47
      const-string v0, "ObjectPath"
  
      invoke-virtual {p0, v0}, Landroid/server/BluetoothAdapterProperties;->getProperty(Ljava/lang/String;)Ljava/lang/String;
***************
*** 333,338 ****
--- 369,375 ----
      .parameter "name"
  
      .prologue
+     .line 40
      monitor-enter p0
  
      :try_start_0
***************
*** 344,351 ****
--- 381,390 ----
  
      if-eqz v0, :cond_0
  
+     .line 41
      invoke-virtual {p0}, Landroid/server/BluetoothAdapterProperties;->getAllProperties()V
  
+     .line 43
      :cond_0
      iget-object v0, p0, Landroid/server/BluetoothAdapterProperties;->mPropertiesMap:Ljava/util/Map;
  
***************
*** 361,366 ****
--- 400,406 ----
  
      return-object v0
  
+     .line 40
      :catchall_0
      move-exception v0
  
***************
*** 373,378 ****
--- 413,419 ----
      .locals 1
  
      .prologue
+     .line 55
      monitor-enter p0
  
      :try_start_0
***************
*** 402,407 ****
--- 443,449 ----
      .parameter "value"
  
      .prologue
+     .line 59
      monitor-enter p0
  
      :try_start_0
***************
*** 411,420 ****
--- 453,464 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 60
      monitor-exit p0
  
      return-void
  
+     .line 59
      :catchall_0
      move-exception v0
  
