*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothService$RemoteService.smali	2012-03-23 19:37:15.411056103 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothService$RemoteService.smali.original	2012-03-23 19:34:45.604324381 -0500
***************
*** 27,38 ****
--- 27,42 ----
      .parameter "uuid"
  
      .prologue
+     .line 179
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 180
      iput-object p1, p0, Landroid/server/BluetoothService$RemoteService;->address:Ljava/lang/String;
  
+     .line 181
      iput-object p2, p0, Landroid/server/BluetoothService$RemoteService;->uuid:Landroid/os/ParcelUuid;
  
+     .line 182
      return-void
  .end method
  
***************
*** 45,58 ****
--- 49,65 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 185
      instance-of v2, p1, Landroid/server/BluetoothService$RemoteService;
  
      if-eqz v2, :cond_0
  
      move-object v0, p1
  
+     .line 186
      check-cast v0, Landroid/server/BluetoothService$RemoteService;
  
+     .line 187
      .local v0, service:Landroid/server/BluetoothService$RemoteService;
      iget-object v2, p0, Landroid/server/BluetoothService$RemoteService;->address:Ljava/lang/String;
  
***************
*** 76,81 ****
--- 83,89 ----
  
      const/4 v1, 0x1
  
+     .line 189
      .end local v0           #service:Landroid/server/BluetoothService$RemoteService;
      :cond_0
      return v1
***************
*** 87,94 ****
--- 95,104 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 194
      const/4 v0, 0x1
  
+     .line 195
      .local v0, hash:I
      iget-object v1, p0, Landroid/server/BluetoothService$RemoteService;->address:Ljava/lang/String;
  
***************
*** 99,104 ****
--- 109,115 ----
      :goto_0
      add-int/lit8 v0, v1, 0x1f
  
+     .line 196
      mul-int/lit8 v1, v0, 0x1f
  
      iget-object v3, p0, Landroid/server/BluetoothService$RemoteService;->uuid:Landroid/os/ParcelUuid;
***************
*** 108,115 ****
--- 119,128 ----
      :goto_1
      add-int v0, v1, v2
  
+     .line 197
      return v0
  
+     .line 195
      :cond_0
      iget-object v1, p0, Landroid/server/BluetoothService$RemoteService;->address:Ljava/lang/String;
  
***************
*** 119,124 ****
--- 132,138 ----
  
      goto :goto_0
  
+     .line 196
      :cond_1
      iget-object v2, p0, Landroid/server/BluetoothService$RemoteService;->uuid:Landroid/os/ParcelUuid;
  
