*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothHealthProfileHandler$1.smali	2012-03-23 19:37:15.279053695 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothHealthProfileHandler$1.smali.original	2012-03-23 19:34:45.603324363 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 84
      iput-object p1, p0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
  
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
***************
*** 38,53 ****
--- 39,57 ----
      .parameter "msg"
  
      .prologue
+     .line 87
      move-object/from16 v0, p1
  
      iget v1, v0, Landroid/os/Message;->what:I
  
      packed-switch v1, :pswitch_data_0
  
+     .line 154
      :cond_0
      :goto_0
      return-void
  
+     .line 89
      :pswitch_0
      move-object/from16 v0, p1
  
***************
*** 55,68 ****
--- 59,75 ----
  
      check-cast v14, Landroid/bluetooth/BluetoothHealthAppConfiguration;
  
+     .line 91
      .local v14, registerApp:Landroid/bluetooth/BluetoothHealthAppConfiguration;
      invoke-virtual {v14}, Landroid/bluetooth/BluetoothHealthAppConfiguration;->getRole()I
  
      move-result v16
  
+     .line 92
      .local v16, role:I
      const/4 v13, 0x0
  
+     .line 94
      .local v13, path:Ljava/lang/String;
      const/4 v1, 0x2
  
***************
*** 70,75 ****
--- 77,83 ----
  
      if-ne v0, v1, :cond_1
  
+     .line 95
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
***************
*** 102,110 ****
--- 110,120 ----
  
      move-result-object v13
  
+     .line 103
      :goto_1
      if-nez v13, :cond_2
  
+     .line 104
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
***************
*** 114,119 ****
--- 124,130 ----
      #calls: Landroid/server/BluetoothHealthProfileHandler;->callHealthApplicationStatusCallback(Landroid/bluetooth/BluetoothHealthAppConfiguration;I)V
      invoke-static {v1, v14, v2}, Landroid/server/BluetoothHealthProfileHandler;->access$400(Landroid/server/BluetoothHealthProfileHandler;Landroid/bluetooth/BluetoothHealthAppConfiguration;I)V
  
+     .line 106
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
***************
*** 127,132 ****
--- 138,144 ----
  
      goto :goto_0
  
+     .line 98
      :cond_1
      move-object/from16 v0, p0
  
***************
*** 177,182 ****
--- 189,195 ----
  
      goto :goto_1
  
+     .line 108
      :cond_2
      move-object/from16 v0, p0
  
***************
*** 189,194 ****
--- 202,208 ----
  
      invoke-virtual {v1, v14, v13}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 109
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
***************
*** 200,205 ****
--- 214,220 ----
  
      goto/16 :goto_0
  
+     .line 115
      .end local v13           #path:Ljava/lang/String;
      .end local v14           #registerApp:Landroid/bluetooth/BluetoothHealthAppConfiguration;
      .end local v16           #role:I
***************
*** 212,217 ****
--- 227,233 ----
  
      check-cast v17, Landroid/bluetooth/BluetoothHealthAppConfiguration;
  
+     .line 119
      .local v17, unregisterApp:Landroid/bluetooth/BluetoothHealthAppConfiguration;
      move-object/from16 v0, p0
  
***************
*** 241,246 ****
--- 257,263 ----
  
      check-cast v8, Landroid/server/BluetoothHealthProfileHandler$HealthChannel;
  
+     .line 120
      .local v8, chan:Landroid/server/BluetoothHealthProfileHandler$HealthChannel;
      #getter for: Landroid/server/BluetoothHealthProfileHandler$HealthChannel;->mConfig:Landroid/bluetooth/BluetoothHealthAppConfiguration;
      invoke-static {v8}, Landroid/server/BluetoothHealthProfileHandler$HealthChannel;->access$800(Landroid/server/BluetoothHealthProfileHandler$HealthChannel;)Landroid/bluetooth/BluetoothHealthAppConfiguration;
***************
*** 262,267 ****
--- 279,285 ----
  
      if-eqz v1, :cond_3
  
+     .line 122
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
***************
*** 282,287 ****
--- 300,306 ----
  
      goto :goto_2
  
+     .line 126
      .end local v8           #chan:Landroid/server/BluetoothHealthProfileHandler$HealthChannel;
      :cond_4
      move-object/from16 v0, p0
***************
*** 314,322 ****
--- 333,343 ----
  
      move-result v15
  
+     .line 128
      .local v15, result:Z
      if-eqz v15, :cond_5
  
+     .line 129
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
***************
*** 328,333 ****
--- 349,355 ----
      #calls: Landroid/server/BluetoothHealthProfileHandler;->callHealthApplicationStatusCallback(Landroid/bluetooth/BluetoothHealthAppConfiguration;I)V
      invoke-static {v1, v0, v2}, Landroid/server/BluetoothHealthProfileHandler;->access$400(Landroid/server/BluetoothHealthProfileHandler;Landroid/bluetooth/BluetoothHealthAppConfiguration;I)V
  
+     .line 131
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
***************
*** 341,346 ****
--- 363,369 ----
  
      invoke-virtual {v1, v0}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 132
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
***************
*** 356,361 ****
--- 379,385 ----
  
      goto/16 :goto_0
  
+     .line 134
      :cond_5
      move-object/from16 v0, p0
  
***************
*** 370,375 ****
--- 394,400 ----
  
      goto/16 :goto_0
  
+     .line 139
      .end local v12           #i$:Ljava/util/Iterator;
      .end local v15           #result:Z
      .end local v17           #unregisterApp:Landroid/bluetooth/BluetoothHealthAppConfiguration;
***************
*** 380,385 ****
--- 405,411 ----
  
      check-cast v8, Landroid/server/BluetoothHealthProfileHandler$HealthChannel;
  
+     .line 140
      .restart local v8       #chan:Landroid/server/BluetoothHealthProfileHandler$HealthChannel;
      move-object/from16 v0, p0
  
***************
*** 403,408 ****
--- 429,435 ----
  
      move-result-object v11
  
+     .line 142
      .local v11, deviceObjectPath:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 424,429 ****
--- 451,457 ----
  
      check-cast v10, Ljava/lang/String;
  
+     .line 143
      .local v10, configPath:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 439,444 ****
--- 467,473 ----
  
      move-result-object v9
  
+     .line 145
      .local v9, channelType:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 460,473 ****
--- 489,505 ----
  
      if-nez v1, :cond_0
  
+     .line 147
      #getter for: Landroid/server/BluetoothHealthProfileHandler$HealthChannel;->mState:I
      invoke-static {v8}, Landroid/server/BluetoothHealthProfileHandler$HealthChannel;->access$900(Landroid/server/BluetoothHealthProfileHandler$HealthChannel;)I
  
      move-result v4
  
+     .line 148
      .local v4, prevState:I
      const/4 v5, 0x0
  
+     .line 149
      .local v5, state:I
      move-object/from16 v0, p0
  
***************
*** 493,498 ****
--- 525,531 ----
      #calls: Landroid/server/BluetoothHealthProfileHandler;->callHealthChannelCallback(Landroid/bluetooth/BluetoothHealthAppConfiguration;Landroid/bluetooth/BluetoothDevice;IILandroid/os/ParcelFileDescriptor;I)V
      invoke-static/range {v1 .. v7}, Landroid/server/BluetoothHealthProfileHandler;->access$1300(Landroid/server/BluetoothHealthProfileHandler;Landroid/bluetooth/BluetoothHealthAppConfiguration;Landroid/bluetooth/BluetoothDevice;IILandroid/os/ParcelFileDescriptor;I)V
  
+     .line 151
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/server/BluetoothHealthProfileHandler$1;->this$0:Landroid/server/BluetoothHealthProfileHandler;
***************
*** 506,511 ****
--- 539,545 ----
  
      goto/16 :goto_0
  
+     .line 87
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
