*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothAdapterStateMachine$WarmUp.smali	2012-03-23 19:37:16.046067682 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/server/BluetoothAdapterStateMachine$WarmUp.smali.original	2012-03-23 19:34:45.610324491 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 291
      iput-object p1, p0, Landroid/server/BluetoothAdapterStateMachine$WarmUp;->this$0:Landroid/server/BluetoothAdapterStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 291
      invoke-direct {p0, p1}, Landroid/server/BluetoothAdapterStateMachine$WarmUp;-><init>(Landroid/server/BluetoothAdapterStateMachine;)V
  
      return-void
***************
*** 48,53 ****
--- 50,56 ----
      .locals 0
  
      .prologue
+     .line 296
      return-void
  .end method
  
***************
*** 56,61 ****
--- 59,65 ----
      .parameter "message"
  
      .prologue
+     .line 300
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 79,97 ****
--- 83,106 ----
      #calls: Landroid/server/BluetoothAdapterStateMachine;->log(Ljava/lang/String;)V
      invoke-static {v1}, Landroid/server/BluetoothAdapterStateMachine;->access$600(Ljava/lang/String;)V
  
+     .line 302
      const/4 v0, 0x1
  
+     .line 303
      .local v0, retValue:Z
      iget v1, p1, Landroid/os/Message;->what:I
  
      sparse-switch v1, :sswitch_data_0
  
+     .line 327
      const/4 v0, 0x0
  
+     .line 329
      .end local v0           #retValue:Z
      :goto_0
      return v0
  
+     .line 305
      .restart local v0       #retValue:Z
      :sswitch_0
      iget-object v1, p0, Landroid/server/BluetoothAdapterStateMachine$WarmUp;->this$0:Landroid/server/BluetoothAdapterStateMachine;
***************
*** 101,106 ****
--- 110,116 ----
      #calls: Landroid/server/BluetoothAdapterStateMachine;->removeMessages(I)V
      invoke-static {v1, v2}, Landroid/server/BluetoothAdapterStateMachine;->access$2600(Landroid/server/BluetoothAdapterStateMachine;I)V
  
+     .line 306
      iget-object v1, p0, Landroid/server/BluetoothAdapterStateMachine$WarmUp;->this$0:Landroid/server/BluetoothAdapterStateMachine;
  
      iget-object v2, p0, Landroid/server/BluetoothAdapterStateMachine$WarmUp;->this$0:Landroid/server/BluetoothAdapterStateMachine;
***************
*** 115,120 ****
--- 125,131 ----
  
      goto :goto_0
  
+     .line 309
      :sswitch_1
      const-string v1, "BluetoothAdapterStateMachine"
  
***************
*** 122,132 ****
--- 133,145 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 310
      iget-object v1, p0, Landroid/server/BluetoothAdapterStateMachine$WarmUp;->this$0:Landroid/server/BluetoothAdapterStateMachine;
  
      #calls: Landroid/server/BluetoothAdapterStateMachine;->shutoffBluetooth()V
      invoke-static {v1}, Landroid/server/BluetoothAdapterStateMachine;->access$2900(Landroid/server/BluetoothAdapterStateMachine;)V
  
+     .line 311
      iget-object v1, p0, Landroid/server/BluetoothAdapterStateMachine$WarmUp;->this$0:Landroid/server/BluetoothAdapterStateMachine;
  
      iget-object v2, p0, Landroid/server/BluetoothAdapterStateMachine$WarmUp;->this$0:Landroid/server/BluetoothAdapterStateMachine;
***************
*** 139,144 ****
--- 152,158 ----
      #calls: Landroid/server/BluetoothAdapterStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
      invoke-static {v1, v2}, Landroid/server/BluetoothAdapterStateMachine;->access$3000(Landroid/server/BluetoothAdapterStateMachine;Lcom/android/internal/util/IState;)V
  
+     .line 312
      iget-object v1, p0, Landroid/server/BluetoothAdapterStateMachine$WarmUp;->this$0:Landroid/server/BluetoothAdapterStateMachine;
  
      const/16 v2, 0xa
***************
*** 148,153 ****
--- 162,168 ----
  
      goto :goto_0
  
+     .line 321
      :sswitch_2
      iget-object v1, p0, Landroid/server/BluetoothAdapterStateMachine$WarmUp;->this$0:Landroid/server/BluetoothAdapterStateMachine;
  
***************
*** 156,161 ****
--- 171,177 ----
  
      goto :goto_0
  
+     .line 324
      :sswitch_3
      const-string v1, "BluetoothAdapterStateMachine"
  
***************
*** 183,188 ****
--- 199,205 ----
  
      goto :goto_0
  
+     .line 303
      nop
  
      :sswitch_data_0
