*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/AppInterface$CommandType.smali	2012-03-23 19:37:43.727572453 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/AppInterface$CommandType.smali.original	2012-03-23 19:34:46.288336853 -0500
***************
*** 78,83 ****
--- 78,84 ----
  
      const/4 v4, 0x1
  
+     .line 47
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "DISPLAY_TEXT"
***************
*** 88,93 ****
--- 89,95 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->DISPLAY_TEXT:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 48
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "GET_INKEY"
***************
*** 98,103 ****
--- 100,106 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->GET_INKEY:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 49
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "GET_INPUT"
***************
*** 108,113 ****
--- 111,117 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->GET_INPUT:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 50
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "LAUNCH_BROWSER"
***************
*** 118,123 ****
--- 122,128 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->LAUNCH_BROWSER:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 51
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "PLAY_TONE"
***************
*** 130,135 ****
--- 135,141 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->PLAY_TONE:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 52
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "REFRESH"
***************
*** 138,143 ****
--- 144,150 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->REFRESH:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 53
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "SELECT_ITEM"
***************
*** 150,155 ****
--- 157,163 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->SELECT_ITEM:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 54
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "SEND_SS"
***************
*** 162,167 ****
--- 170,176 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->SEND_SS:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 55
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "SEND_USSD"
***************
*** 174,179 ****
--- 183,189 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->SEND_USSD:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 56
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "SEND_SMS"
***************
*** 186,191 ****
--- 196,202 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->SEND_SMS:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 57
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "SEND_DTMF"
***************
*** 198,203 ****
--- 209,215 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->SEND_DTMF:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 58
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "SET_UP_EVENT_LIST"
***************
*** 208,213 ****
--- 220,226 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->SET_UP_EVENT_LIST:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 59
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "SET_UP_IDLE_MODE_TEXT"
***************
*** 220,225 ****
--- 233,239 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->SET_UP_IDLE_MODE_TEXT:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 60
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "SET_UP_MENU"
***************
*** 232,237 ****
--- 246,252 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->SET_UP_MENU:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 61
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "SET_UP_CALL"
***************
*** 244,249 ****
--- 259,265 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->SET_UP_CALL:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 62
      new-instance v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      const-string v1, "PROVIDE_LOCAL_INFORMATION"
***************
*** 256,261 ****
--- 272,278 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->PROVIDE_LOCAL_INFORMATION:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
+     .line 46
      const/16 v0, 0x10
  
      new-array v0, v0, [Lcom/android/internal/telephony/cat/AppInterface$CommandType;
***************
*** 363,372 ****
--- 380,392 ----
      .end annotation
  
      .prologue
+     .line 66
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 67
      iput p3, p0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->mValue:I
  
+     .line 68
      return-void
  .end method
  
***************
*** 375,380 ****
--- 395,401 ----
      .parameter "value"
  
      .prologue
+     .line 83
      invoke-static {}, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->values()[Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      move-result-object v0
***************
*** 391,411 ****
--- 412,436 ----
  
      aget-object v1, v0, v2
  
+     .line 84
      .local v1, e:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
      iget v4, v1, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->mValue:I
  
      if-ne v4, p0, :cond_0
  
+     .line 88
      .end local v1           #e:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
      :goto_1
      return-object v1
  
+     .line 83
      .restart local v1       #e:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 88
      .end local v1           #e:Lcom/android/internal/telephony/cat/AppInterface$CommandType;
      :cond_1
      const/4 v1, 0x0
***************
*** 418,423 ****
--- 443,449 ----
      .parameter "name"
  
      .prologue
+     .line 46
      const-class v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 433,438 ****
--- 459,465 ----
      .locals 1
  
      .prologue
+     .line 46
      sget-object v0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->$VALUES:[Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/cat/AppInterface$CommandType;->clone()Ljava/lang/Object;
***************
*** 450,455 ****
--- 477,483 ----
      .locals 1
  
      .prologue
+     .line 71
      iget v0, p0, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->mValue:I
  
      return v0
