*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/CatCmdMessage.smali	2012-03-23 19:37:43.852574732 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/CatCmdMessage.smali.original	2012-03-23 19:34:46.289336872 -0500
***************
*** 50,55 ****
--- 50,56 ----
      .locals 1
  
      .prologue
+     .line 133
      new-instance v0, Lcom/android/internal/telephony/cat/CatCmdMessage$1;
  
      invoke-direct {v0}, Lcom/android/internal/telephony/cat/CatCmdMessage$1;-><init>()V
***************
*** 66,79 ****
--- 67,85 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 91
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 33
      iput-object v2, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
+     .line 34
      iput-object v2, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mToneSettings:Lcom/android/internal/telephony/cat/ToneSettings;
  
+     .line 35
      iput-object v2, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
+     .line 92
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
  
      move-result-object v0
***************
*** 82,87 ****
--- 88,94 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCmdDet:Lcom/android/internal/telephony/cat/CommandDetails;
  
+     .line 93
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
  
      move-result-object v0
***************
*** 90,95 ****
--- 97,103 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mTextMsg:Lcom/android/internal/telephony/cat/TextMessage;
  
+     .line 94
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
  
      move-result-object v0
***************
*** 98,103 ****
--- 106,112 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mMenu:Lcom/android/internal/telephony/cat/Menu;
  
+     .line 95
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
  
      move-result-object v0
***************
*** 106,111 ****
--- 115,121 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mInput:Lcom/android/internal/telephony/cat/Input;
  
+     .line 96
      sget-object v0, Lcom/android/internal/telephony/cat/CatCmdMessage$2;->$SwitchMap$com$android$internal$telephony$cat$AppInterface$CommandType:[I
  
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/CatCmdMessage;->getCmdType()Lcom/android/internal/telephony/cat/AppInterface$CommandType;
***************
*** 120,128 ****
--- 130,140 ----
  
      packed-switch v0, :pswitch_data_0
  
+     .line 111
      :goto_0
      return-void
  
+     .line 98
      :pswitch_0
      new-instance v0, Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
***************
*** 130,135 ****
--- 142,148 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
+     .line 99
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
***************
*** 138,143 ****
--- 151,157 ----
  
      iput-object v1, v0, Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;->url:Ljava/lang/String;
  
+     .line 100
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
      invoke-static {}, Lcom/android/internal/telephony/cat/LaunchBrowserMode;->values()[Lcom/android/internal/telephony/cat/LaunchBrowserMode;
***************
*** 154,159 ****
--- 168,174 ----
  
      goto :goto_0
  
+     .line 103
      :pswitch_1
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
  
***************
*** 165,170 ****
--- 180,186 ----
  
      goto :goto_0
  
+     .line 106
      :pswitch_2
      new-instance v0, Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
***************
*** 172,177 ****
--- 188,194 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
+     .line 107
      iget-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
***************
*** 182,187 ****
--- 199,205 ----
  
      iput-object v0, v1, Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;->confirmMsg:Lcom/android/internal/telephony/cat/TextMessage;
  
+     .line 108
      iget-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
***************
*** 194,199 ****
--- 212,218 ----
  
      goto :goto_0
  
+     .line 96
      nop
  
      :pswitch_data_0
***************
*** 211,228 ****
--- 230,253 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 53
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 33
      iput-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
+     .line 34
      iput-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mToneSettings:Lcom/android/internal/telephony/cat/ToneSettings;
  
+     .line 35
      iput-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
+     .line 54
      iget-object v1, p1, Lcom/android/internal/telephony/cat/CommandParams;->cmdDet:Lcom/android/internal/telephony/cat/CommandDetails;
  
      iput-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCmdDet:Lcom/android/internal/telephony/cat/CommandDetails;
  
+     .line 55
      sget-object v1, Lcom/android/internal/telephony/cat/CatCmdMessage$2;->$SwitchMap$com$android$internal$telephony$cat$AppInterface$CommandType:[I
  
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/CatCmdMessage;->getCmdType()Lcom/android/internal/telephony/cat/AppInterface$CommandType;
***************
*** 237,246 ****
--- 262,273 ----
  
      packed-switch v1, :pswitch_data_0
  
+     .line 89
      .end local p1
      :goto_0
      return-void
  
+     .line 58
      .restart local p1
      :pswitch_0
      check-cast p1, Lcom/android/internal/telephony/cat/SelectItemParams;
***************
*** 252,257 ****
--- 279,285 ----
  
      goto :goto_0
  
+     .line 66
      .restart local p1
      :pswitch_1
      check-cast p1, Lcom/android/internal/telephony/cat/DisplayTextParams;
***************
*** 263,268 ****
--- 291,297 ----
  
      goto :goto_0
  
+     .line 70
      .restart local p1
      :pswitch_2
      check-cast p1, Lcom/android/internal/telephony/cat/GetInputParams;
***************
*** 278,295 ****
--- 307,327 ----
      :pswitch_3
      move-object v1, p1
  
+     .line 73
      check-cast v1, Lcom/android/internal/telephony/cat/LaunchBrowserParams;
  
      iget-object v1, v1, Lcom/android/internal/telephony/cat/LaunchBrowserParams;->confirmMsg:Lcom/android/internal/telephony/cat/TextMessage;
  
      iput-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mTextMsg:Lcom/android/internal/telephony/cat/TextMessage;
  
+     .line 74
      new-instance v1, Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
      invoke-direct {v1, p0}, Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;-><init>(Lcom/android/internal/telephony/cat/CatCmdMessage;)V
  
      iput-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
+     .line 75
      iget-object v2, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
      move-object v1, p1
***************
*** 300,305 ****
--- 332,338 ----
  
      iput-object v1, v2, Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;->url:Ljava/lang/String;
  
+     .line 76
      iget-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
      check-cast p1, Lcom/android/internal/telephony/cat/LaunchBrowserParams;
***************
*** 315,333 ****
--- 348,370 ----
      :pswitch_4
      move-object v0, p1
  
+     .line 79
      check-cast v0, Lcom/android/internal/telephony/cat/PlayToneParams;
  
+     .line 80
      .local v0, params:Lcom/android/internal/telephony/cat/PlayToneParams;
      iget-object v1, v0, Lcom/android/internal/telephony/cat/PlayToneParams;->settings:Lcom/android/internal/telephony/cat/ToneSettings;
  
      iput-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mToneSettings:Lcom/android/internal/telephony/cat/ToneSettings;
  
+     .line 81
      iget-object v1, v0, Lcom/android/internal/telephony/cat/PlayToneParams;->textMsg:Lcom/android/internal/telephony/cat/TextMessage;
  
      iput-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mTextMsg:Lcom/android/internal/telephony/cat/TextMessage;
  
      goto :goto_0
  
+     .line 84
      .end local v0           #params:Lcom/android/internal/telephony/cat/PlayToneParams;
      :pswitch_5
      new-instance v1, Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
***************
*** 336,341 ****
--- 373,379 ----
  
      iput-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
+     .line 85
      iget-object v2, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
      move-object v1, p1
***************
*** 346,351 ****
--- 384,390 ----
  
      iput-object v1, v2, Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;->confirmMsg:Lcom/android/internal/telephony/cat/TextMessage;
  
+     .line 86
      iget-object v1, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
      check-cast p1, Lcom/android/internal/telephony/cat/CallSetupParams;
***************
*** 357,362 ****
--- 396,402 ----
  
      goto :goto_0
  
+     .line 55
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 381,386 ****
--- 421,427 ----
      .locals 1
  
      .prologue
+     .line 144
      const/4 v0, 0x0
  
      return v0
***************
*** 390,395 ****
--- 431,437 ----
      .locals 1
  
      .prologue
+     .line 157
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mInput:Lcom/android/internal/telephony/cat/Input;
  
      return-object v0
***************
*** 399,404 ****
--- 441,447 ----
      .locals 1
  
      .prologue
+     .line 161
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mTextMsg:Lcom/android/internal/telephony/cat/TextMessage;
  
      return-object v0
***************
*** 408,413 ****
--- 451,457 ----
      .locals 1
  
      .prologue
+     .line 165
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
      return-object v0
***************
*** 417,422 ****
--- 461,467 ----
      .locals 1
  
      .prologue
+     .line 173
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
      return-object v0
***************
*** 426,431 ****
--- 471,477 ----
      .locals 1
  
      .prologue
+     .line 149
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCmdDet:Lcom/android/internal/telephony/cat/CommandDetails;
  
      iget v0, v0, Lcom/android/internal/telephony/cat/CommandDetails;->typeOfCommand:I
***************
*** 441,446 ****
--- 487,493 ----
      .locals 1
  
      .prologue
+     .line 153
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mMenu:Lcom/android/internal/telephony/cat/Menu;
  
      return-object v0
***************
*** 450,455 ****
--- 497,503 ----
      .locals 1
  
      .prologue
+     .line 169
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mToneSettings:Lcom/android/internal/telephony/cat/ToneSettings;
  
      return-object v0
***************
*** 463,484 ****
--- 511,537 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 114
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCmdDet:Lcom/android/internal/telephony/cat/CommandDetails;
  
      invoke-virtual {p1, v0, v2}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 115
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mTextMsg:Lcom/android/internal/telephony/cat/TextMessage;
  
      invoke-virtual {p1, v0, v2}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 116
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mMenu:Lcom/android/internal/telephony/cat/Menu;
  
      invoke-virtual {p1, v0, v2}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 117
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mInput:Lcom/android/internal/telephony/cat/Input;
  
      invoke-virtual {p1, v0, v2}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 118
      sget-object v0, Lcom/android/internal/telephony/cat/CatCmdMessage$2;->$SwitchMap$com$android$internal$telephony$cat$AppInterface$CommandType:[I
  
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/CatCmdMessage;->getCmdType()Lcom/android/internal/telephony/cat/AppInterface$CommandType;
***************
*** 493,501 ****
--- 546,556 ----
  
      packed-switch v0, :pswitch_data_0
  
+     .line 131
      :goto_0
      return-void
  
+     .line 120
      :pswitch_0
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
***************
*** 503,508 ****
--- 558,564 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 121
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mBrowserSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;
  
      iget-object v0, v0, Lcom/android/internal/telephony/cat/CatCmdMessage$BrowserSettings;->mode:Lcom/android/internal/telephony/cat/LaunchBrowserMode;
***************
*** 515,520 ****
--- 571,577 ----
  
      goto :goto_0
  
+     .line 124
      :pswitch_1
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mToneSettings:Lcom/android/internal/telephony/cat/ToneSettings;
  
***************
*** 522,527 ****
--- 579,585 ----
  
      goto :goto_0
  
+     .line 127
      :pswitch_2
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
***************
*** 529,534 ****
--- 587,593 ----
  
      invoke-virtual {p1, v0, v2}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 128
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CatCmdMessage;->mCallSettings:Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;
  
      iget-object v0, v0, Lcom/android/internal/telephony/cat/CatCmdMessage$CallSettings;->callMsg:Lcom/android/internal/telephony/cat/TextMessage;
***************
*** 537,542 ****
--- 596,602 ----
  
      goto :goto_0
  
+     .line 118
      nop
  
      :pswitch_data_0
