*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/CatService$1.smali	2012-03-23 19:37:43.626570611 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/CatService$1.smali.original	2012-03-23 19:34:46.286336817 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 3
  
      .prologue
+     .line 635
      invoke-static {}, Lcom/android/internal/telephony/cat/ResultCode;->values()[Lcom/android/internal/telephony/cat/ResultCode;
  
      move-result-object v0
***************
*** 258,263 ****
--- 259,265 ----
      :try_end_d
      .catch Ljava/lang/NoSuchFieldError; {:try_start_d .. :try_end_d} :catch_f
  
+     .line 209
      :goto_d
      invoke-static {}, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->values()[Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
***************
*** 586,591 ****
--- 588,594 ----
  
      goto/16 :goto_e
  
+     .line 635
      :catch_f
      move-exception v0
  
