*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/CommandParamsFactory$1.smali	2012-03-23 19:37:45.473604292 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/CommandParamsFactory$1.smali.original	2012-03-23 19:34:46.317337383 -0500
***************
*** 23,28 ****
--- 23,29 ----
      .locals 3
  
      .prologue
+     .line 127
      invoke-static {}, Lcom/android/internal/telephony/cat/AppInterface$CommandType;->values()[Lcom/android/internal/telephony/cat/AppInterface$CommandType;
  
      move-result-object v0
