*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/CommandParams.smali	2012-03-23 19:37:43.542569080 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/CommandParams.smali.original	2012-03-23 19:34:46.285336799 -0500
***************
*** 13,22 ****
--- 13,25 ----
      .parameter "cmdDet"
  
      .prologue
+     .line 28
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 29
      iput-object p1, p0, Lcom/android/internal/telephony/cat/CommandParams;->cmdDet:Lcom/android/internal/telephony/cat/CommandDetails;
  
+     .line 30
      return-void
  .end method
  
***************
*** 26,31 ****
--- 29,35 ----
      .locals 1
  
      .prologue
+     .line 33
      iget-object v0, p0, Lcom/android/internal/telephony/cat/CommandParams;->cmdDet:Lcom/android/internal/telephony/cat/CommandDetails;
  
      iget v0, v0, Lcom/android/internal/telephony/cat/CommandDetails;->typeOfCommand:I
***************
*** 42,47 ****
--- 46,52 ----
      .parameter "icon"
  
      .prologue
+     .line 36
      const/4 v0, 0x1
  
      return v0
