*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/FontSize.smali	2012-03-23 19:37:43.597570084 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/FontSize.smali.original	2012-03-23 19:34:46.286336817 -0500
***************
*** 39,44 ****
--- 39,45 ----
  
      const/4 v2, 0x0
  
+     .line 26
      new-instance v0, Lcom/android/internal/telephony/cat/FontSize;
  
      const-string v1, "NORMAL"
***************
*** 47,52 ****
--- 48,54 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/FontSize;->NORMAL:Lcom/android/internal/telephony/cat/FontSize;
  
+     .line 27
      new-instance v0, Lcom/android/internal/telephony/cat/FontSize;
  
      const-string v1, "LARGE"
***************
*** 55,60 ****
--- 57,63 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/FontSize;->LARGE:Lcom/android/internal/telephony/cat/FontSize;
  
+     .line 28
      new-instance v0, Lcom/android/internal/telephony/cat/FontSize;
  
      const-string v1, "SMALL"
***************
*** 63,68 ****
--- 66,72 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/FontSize;->SMALL:Lcom/android/internal/telephony/cat/FontSize;
  
+     .line 25
      const/4 v0, 0x3
  
      new-array v0, v0, [Lcom/android/internal/telephony/cat/FontSize;
***************
*** 96,105 ****
--- 100,112 ----
      .end annotation
  
      .prologue
+     .line 32
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 33
      iput p3, p0, Lcom/android/internal/telephony/cat/FontSize;->mValue:I
  
+     .line 34
      return-void
  .end method
  
***************
*** 108,113 ****
--- 115,121 ----
      .parameter "value"
  
      .prologue
+     .line 43
      invoke-static {}, Lcom/android/internal/telephony/cat/FontSize;->values()[Lcom/android/internal/telephony/cat/FontSize;
  
      move-result-object v0
***************
*** 124,144 ****
--- 132,156 ----
  
      aget-object v1, v0, v2
  
+     .line 44
      .local v1, e:Lcom/android/internal/telephony/cat/FontSize;
      iget v4, v1, Lcom/android/internal/telephony/cat/FontSize;->mValue:I
  
      if-ne v4, p0, :cond_0
  
+     .line 48
      .end local v1           #e:Lcom/android/internal/telephony/cat/FontSize;
      :goto_1
      return-object v1
  
+     .line 43
      .restart local v1       #e:Lcom/android/internal/telephony/cat/FontSize;
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 48
      .end local v1           #e:Lcom/android/internal/telephony/cat/FontSize;
      :cond_1
      const/4 v1, 0x0
***************
*** 151,156 ****
--- 163,169 ----
      .parameter "name"
  
      .prologue
+     .line 25
      const-class v0, Lcom/android/internal/telephony/cat/FontSize;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 166,171 ****
--- 179,185 ----
      .locals 1
  
      .prologue
+     .line 25
      sget-object v0, Lcom/android/internal/telephony/cat/FontSize;->$VALUES:[Lcom/android/internal/telephony/cat/FontSize;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/cat/FontSize;->clone()Ljava/lang/Object;
