*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/LaunchBrowserMode.smali	2012-03-23 19:37:45.429603489 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/LaunchBrowserMode.smali.original	2012-03-23 19:34:46.316337365 -0500
***************
*** 35,40 ****
--- 35,41 ----
  
      const/4 v2, 0x0
  
+     .line 27
      new-instance v0, Lcom/android/internal/telephony/cat/LaunchBrowserMode;
  
      const-string v1, "LAUNCH_IF_NOT_ALREADY_LAUNCHED"
***************
*** 43,48 ****
--- 44,50 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/LaunchBrowserMode;->LAUNCH_IF_NOT_ALREADY_LAUNCHED:Lcom/android/internal/telephony/cat/LaunchBrowserMode;
  
+     .line 32
      new-instance v0, Lcom/android/internal/telephony/cat/LaunchBrowserMode;
  
      const-string v1, "USE_EXISTING_BROWSER"
***************
*** 51,56 ****
--- 53,59 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/LaunchBrowserMode;->USE_EXISTING_BROWSER:Lcom/android/internal/telephony/cat/LaunchBrowserMode;
  
+     .line 34
      new-instance v0, Lcom/android/internal/telephony/cat/LaunchBrowserMode;
  
      const-string v1, "LAUNCH_NEW_BROWSER"
***************
*** 59,64 ****
--- 62,68 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/LaunchBrowserMode;->LAUNCH_NEW_BROWSER:Lcom/android/internal/telephony/cat/LaunchBrowserMode;
  
+     .line 25
      const/4 v0, 0x3
  
      new-array v0, v0, [Lcom/android/internal/telephony/cat/LaunchBrowserMode;
***************
*** 91,96 ****
--- 95,101 ----
      .end annotation
  
      .prologue
+     .line 25
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 101,106 ****
--- 106,112 ----
      .parameter "name"
  
      .prologue
+     .line 25
      const-class v0, Lcom/android/internal/telephony/cat/LaunchBrowserMode;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 116,121 ****
--- 122,128 ----
      .locals 1
  
      .prologue
+     .line 25
      sget-object v0, Lcom/android/internal/telephony/cat/LaunchBrowserMode;->$VALUES:[Lcom/android/internal/telephony/cat/LaunchBrowserMode;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/cat/LaunchBrowserMode;->clone()Ljava/lang/Object;
