*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/ResultException$1.smali	2012-03-23 19:37:44.464585892 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/ResultException$1.smali.original	2012-03-23 19:34:46.302337109 -0500
***************
*** 23,28 ****
--- 23,29 ----
      .locals 3
  
      .prologue
+     .line 35
      invoke-static {}, Lcom/android/internal/telephony/cat/ResultCode;->values()[Lcom/android/internal/telephony/cat/ResultCode;
  
      move-result-object v0
