*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/RilMessageDecoder$StateStart.smali	2012-03-23 19:37:44.836592676 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/RilMessageDecoder$StateStart.smali.original	2012-03-23 19:34:46.310337255 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 104
      iput-object p1, p0, Lcom/android/internal/telephony/cat/RilMessageDecoder$StateStart;->this$0:Lcom/android/internal/telephony/cat/RilMessageDecoder;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 104
      invoke-direct {p0, p1}, Lcom/android/internal/telephony/cat/RilMessageDecoder$StateStart;-><init>(Lcom/android/internal/telephony/cat/RilMessageDecoder;)V
  
      return-void
***************
*** 51,60 ****
--- 53,64 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 107
      iget v0, p1, Landroid/os/Message;->what:I
  
      if-ne v0, v2, :cond_1
  
+     .line 108
      iget-object v1, p0, Lcom/android/internal/telephony/cat/RilMessageDecoder$StateStart;->this$0:Lcom/android/internal/telephony/cat/RilMessageDecoder;
  
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
***************
*** 68,73 ****
--- 72,78 ----
  
      if-eqz v0, :cond_0
  
+     .line 109
      iget-object v0, p0, Lcom/android/internal/telephony/cat/RilMessageDecoder$StateStart;->this$0:Lcom/android/internal/telephony/cat/RilMessageDecoder;
  
      iget-object v1, p0, Lcom/android/internal/telephony/cat/RilMessageDecoder$StateStart;->this$0:Lcom/android/internal/telephony/cat/RilMessageDecoder;
***************
*** 80,89 ****
--- 85,96 ----
      #calls: Lcom/android/internal/telephony/cat/RilMessageDecoder;->transitionTo(Lcom/android/internal/util/IState;)V
      invoke-static {v0, v1}, Lcom/android/internal/telephony/cat/RilMessageDecoder;->access$400(Lcom/android/internal/telephony/cat/RilMessageDecoder;Lcom/android/internal/util/IState;)V
  
+     .line 115
      :cond_0
      :goto_0
      return v2
  
+     .line 112
      :cond_1
      new-instance v0, Ljava/lang/StringBuilder;
  
