*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/Tone$1.smali	2012-03-23 19:37:45.083597181 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/Tone$1.smali.original	2012-03-23 19:34:46.313337309 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 181
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "in"
  
      .prologue
+     .line 183
      invoke-static {}, Lcom/android/internal/telephony/cat/Tone;->values()[Lcom/android/internal/telephony/cat/Tone;
  
      move-result-object v0
***************
*** 62,67 ****
--- 64,70 ----
      .parameter "x0"
  
      .prologue
+     .line 181
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/cat/Tone$1;->createFromParcel(Landroid/os/Parcel;)Lcom/android/internal/telephony/cat/Tone;
  
      move-result-object v0
***************
*** 74,79 ****
--- 77,83 ----
      .parameter "size"
  
      .prologue
+     .line 187
      new-array v0, p1, [Lcom/android/internal/telephony/cat/Tone;
  
      return-object v0
***************
*** 84,89 ****
--- 88,94 ----
      .parameter "x0"
  
      .prologue
+     .line 181
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/cat/Tone$1;->newArray(I)[Lcom/android/internal/telephony/cat/Tone;
  
      move-result-object v0
