*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/Tone.smali	2012-03-23 19:37:44.658589430 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/Tone.smali.original	2012-03-23 19:34:46.305337163 -0500
***************
*** 108,113 ****
--- 108,114 ----
  
      const/4 v4, 0x1
  
+     .line 34
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "DIAL"
***************
*** 118,123 ****
--- 119,125 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->DIAL:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 39
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "BUSY"
***************
*** 126,131 ****
--- 128,134 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->BUSY:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 44
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "CONGESTION"
***************
*** 134,139 ****
--- 137,143 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->CONGESTION:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 49
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "RADIO_PATH_ACK"
***************
*** 142,147 ****
--- 146,152 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->RADIO_PATH_ACK:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 54
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "RADIO_PATH_NOT_AVAILABLE"
***************
*** 150,155 ****
--- 155,161 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->RADIO_PATH_NOT_AVAILABLE:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 59
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "ERROR_SPECIAL_INFO"
***************
*** 160,165 ****
--- 166,172 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->ERROR_SPECIAL_INFO:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 64
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "CALL_WAITING"
***************
*** 172,177 ****
--- 179,185 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->CALL_WAITING:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 69
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "RINGING"
***************
*** 184,189 ****
--- 192,198 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->RINGING:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 76
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "GENERAL_BEEP"
***************
*** 196,201 ****
--- 205,211 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->GENERAL_BEEP:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 81
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "POSITIVE_ACK"
***************
*** 208,213 ****
--- 218,224 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->POSITIVE_ACK:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 86
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "NEGATIVE_ACK"
***************
*** 220,225 ****
--- 231,237 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->NEGATIVE_ACK:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 91
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "INCOMING_SPEECH_CALL"
***************
*** 232,237 ****
--- 244,250 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->INCOMING_SPEECH_CALL:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 96
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "INCOMING_SMS"
***************
*** 244,249 ****
--- 257,263 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->INCOMING_SMS:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 104
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "CRITICAL_ALERT"
***************
*** 256,261 ****
--- 270,276 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->CRITICAL_ALERT:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 109
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "VIBRATE_ONLY"
***************
*** 268,273 ****
--- 283,289 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->VIBRATE_ONLY:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 116
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "HAPPY"
***************
*** 280,285 ****
--- 296,302 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->HAPPY:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 121
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "SAD"
***************
*** 292,297 ****
--- 309,315 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->SAD:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 126
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "URGENT"
***************
*** 304,309 ****
--- 322,328 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->URGENT:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 131
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "QUESTION"
***************
*** 316,321 ****
--- 335,341 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->QUESTION:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 136
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "MESSAGE_RECEIVED"
***************
*** 328,333 ****
--- 348,354 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->MESSAGE_RECEIVED:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 139
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "MELODY_1"
***************
*** 340,345 ****
--- 361,367 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->MELODY_1:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 140
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "MELODY_2"
***************
*** 352,357 ****
--- 374,380 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->MELODY_2:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 141
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "MELODY_3"
***************
*** 364,369 ****
--- 387,393 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->MELODY_3:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 142
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "MELODY_4"
***************
*** 376,381 ****
--- 400,406 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->MELODY_4:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 143
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "MELODY_5"
***************
*** 388,393 ****
--- 413,419 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->MELODY_5:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 144
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "MELODY_6"
***************
*** 400,405 ****
--- 426,432 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->MELODY_6:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 145
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "MELODY_7"
***************
*** 412,417 ****
--- 439,445 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->MELODY_7:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 146
      new-instance v0, Lcom/android/internal/telephony/cat/Tone;
  
      const-string v1, "MELODY_8"
***************
*** 424,429 ****
--- 452,458 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->MELODY_8:Lcom/android/internal/telephony/cat/Tone;
  
+     .line 28
      const/16 v0, 0x1c
  
      new-array v0, v0, [Lcom/android/internal/telephony/cat/Tone;
***************
*** 588,593 ****
--- 617,623 ----
  
      sput-object v0, Lcom/android/internal/telephony/cat/Tone;->$VALUES:[Lcom/android/internal/telephony/cat/Tone;
  
+     .line 181
      new-instance v0, Lcom/android/internal/telephony/cat/Tone$1;
  
      invoke-direct {v0}, Lcom/android/internal/telephony/cat/Tone$1;-><init>()V
***************
*** 609,618 ****
--- 639,651 ----
      .end annotation
  
      .prologue
+     .line 150
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 151
      iput p3, p0, Lcom/android/internal/telephony/cat/Tone;->mValue:I
  
+     .line 152
      return-void
  .end method
  
***************
*** 630,643 ****
--- 663,679 ----
      .end annotation
  
      .prologue
+     .line 169
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 170
      invoke-virtual {p3}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Lcom/android/internal/telephony/cat/Tone;->mValue:I
  
+     .line 171
      return-void
  .end method
  
***************
*** 646,651 ****
--- 682,688 ----
      .parameter "value"
  
      .prologue
+     .line 161
      invoke-static {}, Lcom/android/internal/telephony/cat/Tone;->values()[Lcom/android/internal/telephony/cat/Tone;
  
      move-result-object v0
***************
*** 662,682 ****
--- 699,723 ----
  
      aget-object v1, v0, v2
  
+     .line 162
      .local v1, e:Lcom/android/internal/telephony/cat/Tone;
      iget v4, v1, Lcom/android/internal/telephony/cat/Tone;->mValue:I
  
      if-ne v4, p0, :cond_0
  
+     .line 166
      .end local v1           #e:Lcom/android/internal/telephony/cat/Tone;
      :goto_1
      return-object v1
  
+     .line 161
      .restart local v1       #e:Lcom/android/internal/telephony/cat/Tone;
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 166
      .end local v1           #e:Lcom/android/internal/telephony/cat/Tone;
      :cond_1
      const/4 v1, 0x0
***************
*** 689,694 ****
--- 730,736 ----
      .parameter "name"
  
      .prologue
+     .line 28
      const-class v0, Lcom/android/internal/telephony/cat/Tone;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 704,709 ****
--- 746,752 ----
      .locals 1
  
      .prologue
+     .line 28
      sget-object v0, Lcom/android/internal/telephony/cat/Tone;->$VALUES:[Lcom/android/internal/telephony/cat/Tone;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/cat/Tone;->clone()Ljava/lang/Object;
***************
*** 721,726 ****
--- 764,770 ----
      .locals 1
  
      .prologue
+     .line 178
      const/4 v0, 0x0
  
      return v0
***************
*** 732,742 ****
--- 776,788 ----
      .parameter "flags"
  
      .prologue
+     .line 174
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/Tone;->ordinal()I
  
      move-result v0
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 175
      return-void
  .end method
