*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/ValueParser.smali	2012-03-23 19:37:44.807592149 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cat/ValueParser.smali.original	2012-03-23 19:34:46.310337255 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 27
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 23,47 ****
--- 24,54 ----
      .end annotation
  
      .prologue
+     .line 276
      if-eqz p0, :cond_1
  
+     .line 277
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v2
  
+     .line 278
      .local v2, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v3
  
+     .line 279
      .local v3, valueIndex:I
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getLength()I
  
      move-result v1
  
+     .line 280
      .local v1, length:I
      if-eqz v1, :cond_0
  
+     .line 282
      :try_start_0
      invoke-static {v2, v3, v1}, Lcom/android/internal/telephony/IccUtils;->adnStringFieldToString([BII)Ljava/lang/String;
      :try_end_0
***************
*** 49,66 ****
--- 56,76 ----
  
      move-result-object v4
  
+     .line 291
      .end local v1           #length:I
      .end local v2           #rawValue:[B
      .end local v3           #valueIndex:I
      :goto_0
      return-object v4
  
+     .line 284
      .restart local v1       #length:I
      .restart local v2       #rawValue:[B
      .restart local v3       #valueIndex:I
      :catch_0
      move-exception v0
  
+     .line 285
      .local v0, e:Ljava/lang/IndexOutOfBoundsException;
      new-instance v4, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 70,81 ****
--- 80,93 ----
  
      throw v4
  
+     .line 288
      .end local v0           #e:Ljava/lang/IndexOutOfBoundsException;
      :cond_0
      const-string v4, "Defualt Message"
  
      goto :goto_0
  
+     .line 291
      .end local v1           #length:I
      .end local v2           #rawValue:[B
      .end local v3           #valueIndex:I
***************
*** 95,114 ****
--- 107,130 ----
      .end annotation
  
      .prologue
+     .line 40
      new-instance v0, Lcom/android/internal/telephony/cat/CommandDetails;
  
      invoke-direct {v0}, Lcom/android/internal/telephony/cat/CommandDetails;-><init>()V
  
+     .line 41
      .local v0, cmdDet:Lcom/android/internal/telephony/cat/CommandDetails;
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v2
  
+     .line 42
      .local v2, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v3
  
+     .line 44
      .local v3, valueIndex:I
      :try_start_0
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->isComprehensionRequired()Z
***************
*** 117,128 ****
--- 133,146 ----
  
      iput-boolean v4, v0, Lcom/android/internal/telephony/cat/CommandDetails;->compRequired:Z
  
+     .line 45
      aget-byte v4, v2, v3
  
      and-int/lit16 v4, v4, 0xff
  
      iput v4, v0, Lcom/android/internal/telephony/cat/CommandDetails;->commandNumber:I
  
+     .line 46
      add-int/lit8 v4, v3, 0x1
  
      aget-byte v4, v2, v4
***************
*** 131,136 ****
--- 149,155 ----
  
      iput v4, v0, Lcom/android/internal/telephony/cat/CommandDetails;->typeOfCommand:I
  
+     .line 47
      add-int/lit8 v4, v3, 0x2
  
      aget-byte v4, v2, v4
***************
*** 141,151 ****
--- 160,173 ----
      :try_end_0
      .catch Ljava/lang/IndexOutOfBoundsException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 48
      return-object v0
  
+     .line 49
      :catch_0
      move-exception v1
  
+     .line 50
      .local v1, e:Ljava/lang/IndexOutOfBoundsException;
      new-instance v4, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 166,185 ****
--- 188,211 ----
      .end annotation
  
      .prologue
+     .line 65
      new-instance v0, Lcom/android/internal/telephony/cat/DeviceIdentities;
  
      invoke-direct {v0}, Lcom/android/internal/telephony/cat/DeviceIdentities;-><init>()V
  
+     .line 66
      .local v0, devIds:Lcom/android/internal/telephony/cat/DeviceIdentities;
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v2
  
+     .line 67
      .local v2, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v3
  
+     .line 69
      .local v3, valueIndex:I
      :try_start_0
      aget-byte v4, v2, v3
***************
*** 188,193 ****
--- 214,220 ----
  
      iput v4, v0, Lcom/android/internal/telephony/cat/DeviceIdentities;->sourceId:I
  
+     .line 70
      add-int/lit8 v4, v3, 0x1
  
      aget-byte v4, v2, v4
***************
*** 198,208 ****
--- 225,238 ----
      :try_end_0
      .catch Ljava/lang/IndexOutOfBoundsException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 71
      return-object v0
  
+     .line 72
      :catch_0
      move-exception v1
  
+     .line 73
      .local v1, e:Ljava/lang/IndexOutOfBoundsException;
      new-instance v4, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 223,243 ****
--- 253,278 ----
      .end annotation
  
      .prologue
+     .line 86
      const/4 v2, 0x0
  
+     .line 87
      .local v2, timeInterval:I
      sget-object v3, Lcom/android/internal/telephony/cat/Duration$TimeUnit;->SECOND:Lcom/android/internal/telephony/cat/Duration$TimeUnit;
  
+     .line 89
      .local v3, timeUnit:Lcom/android/internal/telephony/cat/Duration$TimeUnit;
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v1
  
+     .line 90
      .local v1, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v4
  
+     .line 93
      .local v4, valueIndex:I
      :try_start_0
      invoke-static {}, Lcom/android/internal/telephony/cat/Duration$TimeUnit;->values()[Lcom/android/internal/telephony/cat/Duration$TimeUnit;
***************
*** 250,255 ****
--- 285,291 ----
  
      aget-object v3, v5, v6
  
+     .line 94
      add-int/lit8 v5, v4, 0x1
  
      aget-byte v5, v1, v5
***************
*** 258,272 ****
--- 294,311 ----
  
      and-int/lit16 v2, v5, 0xff
  
+     .line 98
      new-instance v5, Lcom/android/internal/telephony/cat/Duration;
  
      invoke-direct {v5, v2, v3}, Lcom/android/internal/telephony/cat/Duration;-><init>(ILcom/android/internal/telephony/cat/Duration$TimeUnit;)V
  
      return-object v5
  
+     .line 95
      :catch_0
      move-exception v0
  
+     .line 96
      .local v0, e:Ljava/lang/IndexOutOfBoundsException;
      new-instance v5, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 287,306 ****
--- 326,349 ----
      .end annotation
  
      .prologue
+     .line 161
      new-instance v1, Lcom/android/internal/telephony/cat/IconId;
  
      invoke-direct {v1}, Lcom/android/internal/telephony/cat/IconId;-><init>()V
  
+     .line 163
      .local v1, id:Lcom/android/internal/telephony/cat/IconId;
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v2
  
+     .line 164
      .local v2, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v3
  
+     .line 166
      .local v3, valueIndex:I
      add-int/lit8 v4, v3, 0x1
  
***************
*** 318,323 ****
--- 361,367 ----
      :goto_0
      iput-boolean v5, v1, Lcom/android/internal/telephony/cat/IconId;->selfExplanatory:Z
  
+     .line 167
      aget-byte v5, v2, v4
  
      and-int/lit16 v5, v5, 0xff
***************
*** 326,341 ****
--- 370,389 ----
      :try_end_0
      .catch Ljava/lang/IndexOutOfBoundsException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 172
      return-object v1
  
+     .line 166
      :cond_0
      const/4 v5, 0x0
  
      goto :goto_0
  
+     .line 168
      :catch_0
      move-exception v0
  
+     .line 169
      .local v0, e:Ljava/lang/IndexOutOfBoundsException;
      new-instance v5, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 356,389 ****
--- 404,445 ----
      .end annotation
  
      .prologue
+     .line 109
      const/4 v2, 0x0
  
+     .line 111
      .local v2, item:Lcom/android/internal/telephony/cat/Item;
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v4
  
+     .line 112
      .local v4, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v7
  
+     .line 113
      .local v7, valueIndex:I
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getLength()I
  
      move-result v3
  
+     .line 115
      .local v3, length:I
      if-eqz v3, :cond_0
  
+     .line 116
      add-int/lit8 v6, v3, -0x1
  
+     .line 119
      .local v6, textLen:I
      :try_start_0
      aget-byte v8, v4, v7
  
      and-int/lit16 v1, v8, 0xff
  
+     .line 120
      .local v1, id:I
      add-int/lit8 v8, v7, 0x1
  
***************
*** 391,396 ****
--- 447,453 ----
  
      move-result-object v5
  
+     .line 122
      .local v5, text:Ljava/lang/String;
      new-instance v2, Lcom/android/internal/telephony/cat/Item;
  
***************
*** 399,404 ****
--- 456,462 ----
      :try_end_0
      .catch Ljava/lang/IndexOutOfBoundsException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 128
      .end local v1           #id:I
      .end local v5           #text:Ljava/lang/String;
      .end local v6           #textLen:I
***************
*** 406,416 ****
--- 464,476 ----
      :cond_0
      return-object v2
  
+     .line 123
      .end local v2           #item:Lcom/android/internal/telephony/cat/Item;
      .restart local v6       #textLen:I
      :catch_0
      move-exception v0
  
+     .line 124
      .local v0, e:Ljava/lang/IndexOutOfBoundsException;
      new-instance v8, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 431,448 ****
--- 491,512 ----
      .end annotation
  
      .prologue
+     .line 139
      const/4 v1, 0x0
  
+     .line 141
      .local v1, id:I
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v2
  
+     .line 142
      .local v2, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v3
  
+     .line 145
      .local v3, valueIndex:I
      :try_start_0
      aget-byte v4, v2, v3
***************
*** 451,461 ****
--- 515,528 ----
  
      and-int/lit16 v1, v4, 0xff
  
+     .line 150
      return v1
  
+     .line 146
      :catch_0
      move-exception v0
  
+     .line 147
      .local v0, e:Ljava/lang/IndexOutOfBoundsException;
      new-instance v4, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 476,501 ****
      .end annotation
  
      .prologue
      const-string v8, "ValueParser"
  
!     const-string v9, "retrieveItemsIconId:"
  
      invoke-static {v8, v9}, Lcom/android/internal/telephony/cat/CatLog;->d(Ljava/lang/String;Ljava/lang/String;)V
  
      new-instance v1, Lcom/android/internal/telephony/cat/ItemsIconId;
  
      invoke-direct {v1}, Lcom/android/internal/telephony/cat/ItemsIconId;-><init>()V
  
      .local v1, id:Lcom/android/internal/telephony/cat/ItemsIconId;
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v5
  
      .local v5, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v6
  
      .local v6, valueIndex:I
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getLength()I
  
--- 543,573 ----
      .end annotation
  
      .prologue
+     .line 185
      const-string v8, "ValueParser"
  
!     const-string/jumbo v9, "retrieveItemsIconId:"
  
      invoke-static {v8, v9}, Lcom/android/internal/telephony/cat/CatLog;->d(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 186
      new-instance v1, Lcom/android/internal/telephony/cat/ItemsIconId;
  
      invoke-direct {v1}, Lcom/android/internal/telephony/cat/ItemsIconId;-><init>()V
  
+     .line 188
      .local v1, id:Lcom/android/internal/telephony/cat/ItemsIconId;
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v5
  
+     .line 189
      .local v5, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v6
  
+     .line 190
      .local v6, valueIndex:I
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getLength()I
  
***************
*** 503,513 ****
--- 575,587 ----
  
      add-int/lit8 v4, v8, -0x1
  
+     .line 191
      .local v4, numOfItems:I
      new-array v8, v4, [I
  
      iput-object v8, v1, Lcom/android/internal/telephony/cat/ItemsIconId;->recordNumbers:[I
  
+     .line 195
      add-int/lit8 v7, v6, 0x1
  
      .end local v6           #valueIndex:I
***************
*** 524,529 ****
--- 598,604 ----
      :goto_0
      iput-boolean v8, v1, Lcom/android/internal/telephony/cat/ItemsIconId;->selfExplanatory:Z
  
+     .line 197
      const/4 v2, 0x0
  
      .local v2, index:I
***************
*** 534,539 ****
--- 609,615 ----
      :goto_1
      if-ge v3, v4, :cond_1
  
+     .line 198
      iget-object v8, v1, Lcom/android/internal/telephony/cat/ItemsIconId;->recordNumbers:[I
      :try_end_0
      .catch Ljava/lang/IndexOutOfBoundsException; {:try_start_0 .. :try_end_0} :catch_0
***************
*** 563,579 ****
--- 639,658 ----
      .restart local v7       #valueIndex:I
      goto :goto_1
  
+     .line 195
      .end local v3           #index:I
      :cond_0
      const/4 v8, 0x0
  
      goto :goto_0
  
+     .line 200
      :catch_0
      move-exception v0
  
      move v6, v7
  
+     .line 201
      .end local v7           #valueIndex:I
      .local v0, e:Ljava/lang/IndexOutOfBoundsException;
      .restart local v6       #valueIndex:I
***************
*** 586,591 ****
--- 665,671 ----
  
      throw v8
  
+     .line 203
      .end local v0           #e:Ljava/lang/IndexOutOfBoundsException;
      .end local v6           #valueIndex:I
      .restart local v3       #index:I
***************
*** 593,598 ****
--- 673,679 ----
      :cond_1
      return-object v1
  
+     .line 200
      .end local v3           #index:I
      .end local v7           #valueIndex:I
      .restart local v2       #index:I
***************
*** 625,654 ****
--- 706,742 ----
      .end annotation
  
      .prologue
+     .line 216
      new-instance v18, Ljava/util/ArrayList;
  
      invoke-direct/range {v18 .. v18}, Ljava/util/ArrayList;-><init>()V
  
+     .line 218
      .local v18, lst:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/telephony/cat/TextAttribute;>;"
      invoke-virtual/range {p0 .. p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v19
  
+     .line 219
      .local v19, rawValue:[B
      invoke-virtual/range {p0 .. p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v21
  
+     .line 220
      .local v21, valueIndex:I
      invoke-virtual/range {p0 .. p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getLength()I
  
      move-result v17
  
+     .line 222
      .local v17, length:I
      if-eqz v17, :cond_5
  
+     .line 224
      div-int/lit8 v16, v17, 0x4
  
+     .line 227
      .local v16, itemCount:I
      const/4 v15, 0x0
  
***************
*** 658,663 ****
--- 746,752 ----
  
      if-ge v15, v0, :cond_6
  
+     .line 228
      :try_start_0
      aget-byte v22, v19, v21
  
***************
*** 665,670 ****
--- 754,760 ----
  
      and-int/lit16 v2, v0, 0xff
  
+     .line 229
      .local v2, start:I
      add-int/lit8 v22, v21, 0x1
  
***************
*** 674,679 ****
--- 764,770 ----
  
      and-int/lit16 v3, v0, 0xff
  
+     .line 230
      .local v3, textLength:I
      add-int/lit8 v22, v21, 0x2
  
***************
*** 683,688 ****
--- 774,780 ----
  
      and-int/lit16 v14, v0, 0xff
  
+     .line 231
      .local v14, format:I
      add-int/lit8 v22, v21, 0x3
  
***************
*** 692,720 ****
--- 784,819 ----
  
      and-int/lit16 v12, v0, 0xff
  
+     .line 233
      .local v12, colorValue:I
      and-int/lit8 v11, v14, 0x3
  
+     .line 234
      .local v11, alignValue:I
      invoke-static {v11}, Lcom/android/internal/telephony/cat/TextAlignment;->fromInt(I)Lcom/android/internal/telephony/cat/TextAlignment;
  
      move-result-object v4
  
+     .line 236
      .local v4, align:Lcom/android/internal/telephony/cat/TextAlignment;
      shr-int/lit8 v22, v14, 0x2
  
      and-int/lit8 v20, v22, 0x3
  
+     .line 237
      .local v20, sizeValue:I
      invoke-static/range {v20 .. v20}, Lcom/android/internal/telephony/cat/FontSize;->fromInt(I)Lcom/android/internal/telephony/cat/FontSize;
  
      move-result-object v5
  
+     .line 238
      .local v5, size:Lcom/android/internal/telephony/cat/FontSize;
      if-nez v5, :cond_0
  
+     .line 240
      sget-object v5, Lcom/android/internal/telephony/cat/FontSize;->NORMAL:Lcom/android/internal/telephony/cat/FontSize;
  
+     .line 243
      :cond_0
      and-int/lit8 v22, v14, 0x10
  
***************
*** 722,727 ****
--- 821,827 ----
  
      const/4 v6, 0x1
  
+     .line 244
      .local v6, bold:Z
      :goto_1
      and-int/lit8 v22, v14, 0x20
***************
*** 730,735 ****
--- 830,836 ----
  
      const/4 v7, 0x1
  
+     .line 245
      .local v7, italic:Z
      :goto_2
      and-int/lit8 v22, v14, 0x40
***************
*** 738,743 ****
--- 839,845 ----
  
      const/4 v8, 0x1
  
+     .line 246
      .local v8, underlined:Z
      :goto_3
      and-int/lit16 v0, v14, 0x80
***************
*** 748,764 ****
--- 850,869 ----
  
      const/4 v9, 0x1
  
+     .line 248
      .local v9, strikeThrough:Z
      :goto_4
      invoke-static {v12}, Lcom/android/internal/telephony/cat/TextColor;->fromInt(I)Lcom/android/internal/telephony/cat/TextColor;
  
      move-result-object v10
  
+     .line 250
      .local v10, color:Lcom/android/internal/telephony/cat/TextColor;
      new-instance v1, Lcom/android/internal/telephony/cat/TextAttribute;
  
      invoke-direct/range {v1 .. v10}, Lcom/android/internal/telephony/cat/TextAttribute;-><init>(IILcom/android/internal/telephony/cat/TextAlignment;Lcom/android/internal/telephony/cat/FontSize;ZZZZLcom/android/internal/telephony/cat/TextColor;)V
  
+     .line 253
      .local v1, attr:Lcom/android/internal/telephony/cat/TextAttribute;
      move-object/from16 v0, v18
  
***************
*** 766,777 ****
--- 871,884 ----
      :try_end_0
      .catch Ljava/lang/IndexOutOfBoundsException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 227
      add-int/lit8 v15, v15, 0x1
  
      add-int/lit8 v21, v21, 0x4
  
      goto :goto_0
  
+     .line 243
      .end local v1           #attr:Lcom/android/internal/telephony/cat/TextAttribute;
      .end local v6           #bold:Z
      .end local v7           #italic:Z
***************
*** 783,806 ****
--- 890,917 ----
  
      goto :goto_1
  
+     .line 244
      .restart local v6       #bold:Z
      :cond_2
      const/4 v7, 0x0
  
      goto :goto_2
  
+     .line 245
      .restart local v7       #italic:Z
      :cond_3
      const/4 v8, 0x0
  
      goto :goto_3
  
+     .line 246
      .restart local v8       #underlined:Z
      :cond_4
      const/4 v9, 0x0
  
      goto :goto_4
  
+     .line 258
      .end local v2           #start:I
      .end local v3           #textLength:I
      .end local v4           #align:Lcom/android/internal/telephony/cat/TextAlignment;
***************
*** 815,820 ****
--- 926,932 ----
      :catch_0
      move-exception v13
  
+     .line 259
      .local v13, e:Ljava/lang/IndexOutOfBoundsException;
      new-instance v22, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 824,829 ****
--- 936,942 ----
  
      throw v22
  
+     .line 262
      .end local v13           #e:Ljava/lang/IndexOutOfBoundsException;
      .end local v15           #i:I
      .end local v16           #itemCount:I
***************
*** 845,885 ****
--- 958,1007 ----
      .end annotation
  
      .prologue
+     .line 304
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getRawValue()[B
  
      move-result-object v2
  
+     .line 305
      .local v2, rawValue:[B
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getValueIndex()I
  
      move-result v6
  
+     .line 306
      .local v6, valueIndex:I
      const/4 v0, 0x0
  
+     .line 307
      .local v0, codingScheme:B
      const/4 v3, 0x0
  
+     .line 308
      .local v3, text:Ljava/lang/String;
      invoke-virtual {p0}, Lcom/android/internal/telephony/cat/ComprehensionTlv;->getLength()I
  
      move-result v5
  
+     .line 311
      .local v5, textLen:I
      if-nez v5, :cond_0
  
      move-object v4, v3
  
+     .line 333
      .end local v3           #text:Ljava/lang/String;
      .local v4, text:Ljava/lang/String;
      :goto_0
      return-object v4
  
+     .line 315
      .end local v4           #text:Ljava/lang/String;
      .restart local v3       #text:Ljava/lang/String;
      :cond_0
      add-int/lit8 v5, v5, -0x1
  
+     .line 319
      :try_start_0
      aget-byte v7, v2, v6
  
***************
*** 887,894 ****
--- 1009,1018 ----
  
      int-to-byte v0, v7
  
+     .line 321
      if-nez v0, :cond_1
  
+     .line 322
      add-int/lit8 v7, v6, 0x1
  
      mul-int/lit8 v8, v5, 0x8
***************
*** 902,911 ****
--- 1026,1037 ----
      :goto_1
      move-object v4, v3
  
+     .line 333
      .end local v3           #text:Ljava/lang/String;
      .restart local v4       #text:Ljava/lang/String;
      goto :goto_0
  
+     .line 324
      .end local v4           #text:Ljava/lang/String;
      .restart local v3       #text:Ljava/lang/String;
      :cond_1
***************
*** 913,918 ****
--- 1039,1045 ----
  
      if-ne v0, v7, :cond_2
  
+     .line 325
      add-int/lit8 v7, v6, 0x1
  
      invoke-static {v2, v7, v5}, Lcom/android/internal/telephony/GsmAlphabet;->gsm8BitUnpackedToString([BII)Ljava/lang/String;
***************
*** 921,931 ****
--- 1048,1060 ----
  
      goto :goto_1
  
+     .line 327
      :cond_2
      const/16 v7, 0x8
  
      if-ne v0, v7, :cond_3
  
+     .line 328
      new-instance v3, Ljava/lang/String;
  
      .end local v3           #text:Ljava/lang/String;
***************
*** 938,943 ****
--- 1067,1073 ----
      .restart local v3       #text:Ljava/lang/String;
      goto :goto_1
  
+     .line 330
      :cond_3
      new-instance v7, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 950,959 ****
--- 1080,1091 ----
      .catch Ljava/lang/IndexOutOfBoundsException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/io/UnsupportedEncodingException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 334
      .end local v3           #text:Ljava/lang/String;
      :catch_0
      move-exception v1
  
+     .line 335
      .local v1, e:Ljava/lang/IndexOutOfBoundsException;
      new-instance v7, Lcom/android/internal/telephony/cat/ResultException;
  
***************
*** 963,972 ****
--- 1095,1106 ----
  
      throw v7
  
+     .line 336
      .end local v1           #e:Ljava/lang/IndexOutOfBoundsException;
      :catch_1
      move-exception v1
  
+     .line 338
      .local v1, e:Ljava/io/UnsupportedEncodingException;
      new-instance v7, Lcom/android/internal/telephony/cat/ResultException;
  
