*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/IWifiManager$Stub$Proxy.smali	2012-03-23 19:36:57.918737131 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/IWifiManager$Stub$Proxy.smali.original	2012-03-23 19:34:45.445321481 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 434
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 435
      iput-object p1, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 436
      return-void
  .end method
  
***************
*** 47,71 ****
--- 50,80 ----
      .end annotation
  
      .prologue
+     .line 894
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 895
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 897
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 898
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 899
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 900
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x1a
***************
*** 74,94 ****
--- 83,109 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 901
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 904
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 905
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 907
      return-void
  
+     .line 904
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 905
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 111,147 ****
--- 126,172 ----
  
      const/4 v3, 0x0
  
+     .line 796
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 797
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 800
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 801
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 802
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 803
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 804
      if-eqz p4, :cond_0
  
+     .line 805
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 806
      const/4 v4, 0x0
  
      invoke-virtual {p4, v0, v4}, Landroid/os/WorkSource;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 811
      :goto_0
      iget-object v4, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 151,158 ****
--- 176,185 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 812
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 813
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 161,174 ****
--- 188,205 ----
  
      if-eqz v4, :cond_1
  
+     .line 816
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 817
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 819
      return v2
  
+     .line 809
      .end local v2           #_result:Z
      :cond_0
      const/4 v4, 0x0
***************
*** 180,190 ****
--- 211,223 ----
  
      goto :goto_0
  
+     .line 816
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 817
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 192,197 ****
--- 225,231 ----
      :cond_1
      move v2, v3
  
+     .line 813
      goto :goto_1
  .end method
  
***************
*** 205,235 ****
--- 239,276 ----
      .end annotation
  
      .prologue
+     .line 464
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 465
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 468
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 469
      if-eqz p1, :cond_0
  
+     .line 470
      const/4 v3, 0x1
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 471
      const/4 v3, 0x0
  
      invoke-virtual {p1, v0, v3}, Landroid/net/wifi/WifiConfiguration;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 476
      :goto_0
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 239,259 ****
--- 280,306 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 477
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 478
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 481
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 482
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 484
      return v2
  
+     .line 474
      .end local v2           #_result:I
      :cond_0
      const/4 v3, 0x0
***************
*** 265,275 ****
--- 312,324 ----
  
      goto :goto_0
  
+     .line 481
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 482
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 285,307 ****
--- 334,361 ----
      .end annotation
  
      .prologue
+     .line 1034
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1035
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1037
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1038
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 1039
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x22
***************
*** 310,330 ****
--- 364,390 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1040
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1043
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1044
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1046
      return-void
  
+     .line 1043
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1044
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 334,339 ****
--- 394,400 ----
      .locals 1
  
      .prologue
+     .line 439
      iget-object v0, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 348,368 ****
--- 409,433 ----
      .end annotation
  
      .prologue
+     .line 1049
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1050
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1052
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1053
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x23
***************
*** 371,391 ****
--- 436,462 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1054
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1057
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1058
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1060
      return-void
  
+     .line 1057
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1058
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 403,425 ****
--- 474,501 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 525
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 526
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 529
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 530
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 531
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x5
***************
*** 428,435 ****
--- 504,513 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 532
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 533
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 440,459 ****
--- 518,542 ----
  
      const/4 v2, 0x1
  
+     .line 536
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 537
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 539
      return v2
  
+     .line 536
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 537
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 468,488 ****
--- 551,575 ----
      .end annotation
  
      .prologue
+     .line 592
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 593
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 595
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 596
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x9
***************
*** 491,511 ****
--- 578,604 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 597
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 600
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 601
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 603
      return-void
  
+     .line 600
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 601
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 526,548 ****
--- 619,646 ----
  
      const/4 v3, 0x0
  
+     .line 506
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 507
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 510
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 511
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 512
      if-eqz p2, :cond_0
  
      move v4, v2
***************
*** 550,555 ****
--- 648,654 ----
      :goto_0
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 513
      iget-object v4, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v5, 0x4
***************
*** 558,565 ****
--- 657,666 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 514
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 515
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 568,597 ****
--- 669,705 ----
  
      if-eqz v4, :cond_1
  
+     .line 518
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 519
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 521
      return v2
  
      .end local v2           #_result:Z
      :cond_0
      move v4, v3
  
+     .line 512
      goto :goto_0
  
      :cond_1
      move v2, v3
  
+     .line 515
      goto :goto_1
  
+     .line 518
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 519
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 606,626 ****
--- 714,738 ----
      .end annotation
  
      .prologue
+     .line 1085
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1086
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1089
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1090
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x25
***************
*** 629,655 ****
--- 741,774 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1091
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1092
      invoke-virtual {v1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result-object v2
  
+     .line 1095
      .local v2, _result:Ljava/lang/String;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1096
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1098
      return-object v2
  
+     .line 1095
      .end local v2           #_result:Ljava/lang/String;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1096
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 674,694 ****
--- 793,817 ----
      .end annotation
  
      .prologue
+     .line 447
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 448
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 451
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 452
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x1
***************
*** 697,704 ****
--- 820,829 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 453
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 454
      sget-object v3, Landroid/net/wifi/WifiConfiguration;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-virtual {v1, v3}, Landroid/os/Parcel;->createTypedArrayList(Landroid/os/Parcelable$Creator;)Ljava/util/ArrayList;
***************
*** 707,725 ****
--- 832,855 ----
  
      move-result-object v2
  
+     .line 457
      .local v2, _result:Ljava/util/List;,"Ljava/util/List<Landroid/net/wifi/WifiConfiguration;>;"
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 458
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 460
      return-object v2
  
+     .line 457
      .end local v2           #_result:Ljava/util/List;,"Ljava/util/List<Landroid/net/wifi/WifiConfiguration;>;"
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 458
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 734,754 ****
--- 864,888 ----
      .end annotation
  
      .prologue
+     .line 634
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 635
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 638
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 639
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xc
***************
*** 757,770 ****
--- 891,907 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 640
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 641
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 642
      sget-object v3, Landroid/net/wifi/WifiInfo;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 775,788 ****
--- 912,929 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 649
      .local v2, _result:Landroid/net/wifi/WifiInfo;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 650
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 652
      return-object v2
  
+     .line 645
      .end local v2           #_result:Landroid/net/wifi/WifiInfo;
      :cond_0
      const/4 v2, 0x0
***************
*** 790,801 ****
--- 931,944 ----
      .restart local v2       #_result:Landroid/net/wifi/WifiInfo;
      goto :goto_0
  
+     .line 649
      .end local v2           #_result:Landroid/net/wifi/WifiInfo;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 650
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 810,830 ****
--- 953,977 ----
      .end annotation
  
      .prologue
+     .line 774
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 775
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 778
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 779
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x14
***************
*** 833,846 ****
--- 980,996 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 780
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 781
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 782
      sget-object v3, Landroid/net/DhcpInfo;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 851,864 ****
--- 1001,1018 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 789
      .local v2, _result:Landroid/net/DhcpInfo;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 790
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 792
      return-object v2
  
+     .line 785
      .end local v2           #_result:Landroid/net/DhcpInfo;
      :cond_0
      const/4 v2, 0x0
***************
*** 866,877 ****
--- 1020,1033 ----
      .restart local v2       #_result:Landroid/net/DhcpInfo;
      goto :goto_0
  
+     .line 789
      .end local v2           #_result:Landroid/net/DhcpInfo;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 790
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 886,906 ****
--- 1042,1066 ----
      .end annotation
  
      .prologue
+     .line 723
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 724
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 727
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 728
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x11
***************
*** 909,935 ****
--- 1069,1102 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 729
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 730
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 733
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 734
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 736
      return v2
  
+     .line 733
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 734
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 939,944 ****
--- 1106,1112 ----
      .locals 1
  
      .prologue
+     .line 443
      const-string v0, "android.net.wifi.IWifiManager"
  
      return-object v0
***************
*** 953,973 ****
--- 1121,1145 ----
      .end annotation
  
      .prologue
+     .line 1063
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1064
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1067
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1068
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x24
***************
*** 976,989 ****
--- 1148,1164 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1069
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1070
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 1071
      sget-object v3, Landroid/os/Messenger;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 994,1007 ****
--- 1169,1186 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1078
      .local v2, _result:Landroid/os/Messenger;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1079
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1081
      return-object v2
  
+     .line 1074
      .end local v2           #_result:Landroid/os/Messenger;
      :cond_0
      const/4 v2, 0x0
***************
*** 1009,1020 ****
--- 1188,1201 ----
      .restart local v2       #_result:Landroid/os/Messenger;
      goto :goto_0
  
+     .line 1078
      .end local v2           #_result:Landroid/os/Messenger;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1079
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1039,1059 ****
--- 1220,1244 ----
      .end annotation
  
      .prologue
+     .line 575
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 576
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 579
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 580
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x8
***************
*** 1062,1069 ****
--- 1247,1256 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 581
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 582
      sget-object v3, Landroid/net/wifi/ScanResult;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-virtual {v1, v3}, Landroid/os/Parcel;->createTypedArrayList(Landroid/os/Parcelable$Creator;)Ljava/util/ArrayList;
***************
*** 1072,1090 ****
--- 1259,1282 ----
  
      move-result-object v2
  
+     .line 585
      .local v2, _result:Ljava/util/List;,"Ljava/util/List<Landroid/net/wifi/ScanResult;>;"
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 586
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 588
      return-object v2
  
+     .line 585
      .end local v2           #_result:Ljava/util/List;,"Ljava/util/List<Landroid/net/wifi/ScanResult;>;"
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 586
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1099,1119 ****
--- 1291,1315 ----
      .end annotation
  
      .prologue
+     .line 963
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 964
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 967
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 968
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x1e
***************
*** 1122,1135 ****
--- 1318,1334 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 969
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 970
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 971
      sget-object v3, Landroid/net/wifi/WifiConfiguration;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 1140,1153 ****
--- 1339,1356 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 978
      .local v2, _result:Landroid/net/wifi/WifiConfiguration;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 979
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 981
      return-object v2
  
+     .line 974
      .end local v2           #_result:Landroid/net/wifi/WifiConfiguration;
      :cond_0
      const/4 v2, 0x0
***************
*** 1155,1166 ****
--- 1358,1371 ----
      .restart local v2       #_result:Landroid/net/wifi/WifiConfiguration;
      goto :goto_0
  
+     .line 978
      .end local v2           #_result:Landroid/net/wifi/WifiConfiguration;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 979
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1175,1195 ****
--- 1380,1404 ----
      .end annotation
  
      .prologue
+     .line 946
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 947
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 950
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 951
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x1d
***************
*** 1198,1224 ****
--- 1407,1440 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 952
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 953
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 956
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 957
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 959
      return v2
  
+     .line 956
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 957
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1233,1253 ****
--- 1449,1473 ----
      .end annotation
  
      .prologue
+     .line 674
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 675
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 678
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 679
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xe
***************
*** 1256,1282 ****
--- 1476,1509 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 680
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 681
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 684
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 685
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 687
      return v2
  
+     .line 684
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 685
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1291,1311 ****
--- 1518,1542 ----
      .end annotation
  
      .prologue
+     .line 863
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 864
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 866
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 867
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x18
***************
*** 1314,1334 ****
--- 1545,1571 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 868
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 871
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 872
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 874
      return-void
  
+     .line 871
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 872
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1345,1365 ****
--- 1582,1606 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 740
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 741
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 744
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 745
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x12
***************
*** 1368,1375 ****
--- 1609,1618 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 746
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 747
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1380,1399 ****
--- 1623,1647 ----
  
      const/4 v2, 0x1
  
+     .line 750
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 751
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 753
      return v2
  
+     .line 750
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 751
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1410,1430 ****
--- 1658,1682 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 877
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 878
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 881
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 882
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x19
***************
*** 1433,1440 ****
--- 1685,1694 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 883
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 884
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1445,1464 ****
--- 1699,1723 ----
  
      const/4 v2, 0x1
  
+     .line 887
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 888
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 890
      return v2
  
+     .line 887
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 888
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1475,1495 ****
--- 1734,1758 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 543
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 544
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 547
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 548
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x6
***************
*** 1498,1505 ****
--- 1761,1770 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 549
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 550
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1510,1529 ****
--- 1775,1799 ----
  
      const/4 v2, 0x1
  
+     .line 553
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 554
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 556
      return v2
  
+     .line 553
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 554
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1538,1558 ****
--- 1808,1832 ----
      .end annotation
  
      .prologue
+     .line 620
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 621
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 623
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 624
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xb
***************
*** 1561,1581 ****
--- 1835,1861 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 625
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 628
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 629
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 631
      return-void
  
+     .line 628
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 629
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1590,1610 ****
--- 1870,1894 ----
      .end annotation
  
      .prologue
+     .line 606
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 607
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 609
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 610
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xa
***************
*** 1613,1633 ****
--- 1897,1923 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 611
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 614
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 615
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 617
      return-void
  
+     .line 614
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 615
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1642,1662 ****
--- 1932,1956 ----
      .end annotation
  
      .prologue
+     .line 910
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 911
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 913
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 914
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x1b
***************
*** 1665,1685 ****
--- 1959,1985 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 915
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 918
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 919
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 921
      return-void
  
+     .line 918
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 919
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1697,1719 ****
--- 1997,2024 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 845
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 846
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 849
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 850
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 851
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x17
***************
*** 1722,1729 ****
--- 2027,2036 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 852
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 853
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1734,1753 ****
--- 2041,2065 ----
  
      const/4 v2, 0x1
  
+     .line 856
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 857
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 859
      return v2
  
+     .line 856
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 857
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1765,1787 ****
--- 2077,2104 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 488
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 489
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 492
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 493
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 494
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x3
***************
*** 1790,1797 ****
--- 2107,2116 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 495
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 496
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1802,1821 ****
--- 2121,2145 ----
  
      const/4 v2, 0x1
  
+     .line 499
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 500
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 502
      return v2
  
+     .line 499
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 500
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1832,1852 ****
--- 2156,2180 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 757
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 758
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 761
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 762
      iget-object v3, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x13
***************
*** 1855,1862 ****
--- 2183,2192 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 763
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 764
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1867,1886 ****
--- 2197,2221 ----
  
      const/4 v2, 0x1
  
+     .line 767
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 768
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 770
      return v2
  
+     .line 767
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 768
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1899,1921 ****
--- 2234,2261 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 691
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 692
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 694
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 695
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 696
      if-eqz p2, :cond_0
  
      const/4 v2, 0x1
***************
*** 1923,1928 ****
--- 2263,2269 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 697
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xf
***************
*** 1931,1951 ****
--- 2272,2298 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 698
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 701
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 702
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 704
      return-void
  
+     .line 701
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 702
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1964,1986 ****
--- 2311,2338 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 707
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 708
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 710
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 711
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 712
      if-eqz p2, :cond_0
  
      const/4 v2, 0x1
***************
*** 1988,1993 ****
--- 2340,2346 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 713
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x10
***************
*** 1996,2016 ****
--- 2349,2375 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 714
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 717
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 718
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 720
      return-void
  
+     .line 717
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 718
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 2026,2056 ****
--- 2385,2422 ----
      .end annotation
  
      .prologue
+     .line 985
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 986
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 988
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 989
      if-eqz p1, :cond_0
  
+     .line 990
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 991
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/net/wifi/WifiConfiguration;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 996
      :goto_0
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 2060,2075 ****
--- 2426,2446 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 997
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1000
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1001
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1003
      return-void
  
+     .line 994
      :cond_0
      const/4 v2, 0x0
  
***************
*** 2080,2090 ****
--- 2451,2463 ----
  
      goto :goto_0
  
+     .line 1000
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1001
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 2105,2141 ****
--- 2478,2522 ----
  
      const/4 v3, 0x0
  
+     .line 924
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 925
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 927
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 928
      if-eqz p1, :cond_0
  
+     .line 929
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 930
      const/4 v4, 0x0
  
      invoke-virtual {p1, v0, v4}, Landroid/net/wifi/WifiConfiguration;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 935
      :goto_0
      if-eqz p2, :cond_1
  
      :goto_1
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 936
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x1c
***************
*** 2144,2159 ****
--- 2525,2545 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 937
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 940
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 941
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 943
      return-void
  
+     .line 933
      :cond_0
      const/4 v4, 0x0
  
***************
*** 2164,2174 ****
--- 2550,2562 ----
  
      goto :goto_0
  
+     .line 940
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 941
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 2176,2181 ****
--- 2564,2570 ----
      :cond_1
      move v2, v3
  
+     .line 935
      goto :goto_1
  .end method
  
***************
*** 2193,2213 ****
--- 2582,2606 ----
  
      const/4 v3, 0x0
  
+     .line 656
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 657
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 660
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 661
      if-eqz p1, :cond_0
  
      move v4, v2
***************
*** 2215,2220 ****
--- 2608,2614 ----
      :goto_0
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 662
      iget-object v4, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v5, 0xd
***************
*** 2223,2230 ****
--- 2617,2626 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 663
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 664
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 2233,2262 ****
--- 2629,2665 ----
  
      if-eqz v4, :cond_1
  
+     .line 667
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 668
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 670
      return v2
  
      .end local v2           #_result:Z
      :cond_0
      move v4, v3
  
+     .line 661
      goto :goto_0
  
      :cond_1
      move v2, v3
  
+     .line 664
      goto :goto_1
  
+     .line 667
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 668
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 2274,2294 ****
--- 2677,2701 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 560
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 561
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 563
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 564
      if-eqz p1, :cond_0
  
      const/4 v2, 0x1
***************
*** 2296,2301 ****
--- 2703,2709 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 565
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x7
***************
*** 2304,2324 ****
--- 2712,2738 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 566
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 569
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 570
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 572
      return-void
  
+     .line 569
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 570
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 2333,2353 ****
--- 2747,2771 ----
      .end annotation
  
      .prologue
+     .line 1006
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1007
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1009
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1010
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x20
***************
*** 2356,2376 ****
--- 2774,2800 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1011
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1014
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1015
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1017
      return-void
  
+     .line 1014
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1015
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 2385,2405 ****
--- 2809,2833 ----
      .end annotation
  
      .prologue
+     .line 1020
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1021
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1023
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1024
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x21
***************
*** 2408,2428 ****
--- 2836,2862 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1025
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1028
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1029
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1031
      return-void
  
+     .line 1028
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1029
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 2439,2471 ****
--- 2873,2913 ----
      .end annotation
  
      .prologue
+     .line 823
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 824
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 826
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.net.wifi.IWifiManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 827
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 828
      if-eqz p2, :cond_0
  
+     .line 829
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 830
      const/4 v2, 0x0
  
      invoke-virtual {p2, v0, v2}, Landroid/os/WorkSource;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 835
      :goto_0
      iget-object v2, p0, Landroid/net/wifi/IWifiManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 2475,2490 ****
--- 2917,2937 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 836
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 839
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 840
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 842
      return-void
  
+     .line 833
      :cond_0
      const/4 v2, 0x0
  
***************
*** 2495,2505 ****
--- 2942,2954 ----
  
      goto :goto_0
  
+     .line 839
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 840
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
