*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/SupplicantState$1.smali	2012-03-23 19:36:59.194760400 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/SupplicantState$1.smali.original	2012-03-23 19:34:45.463321809 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 253
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "in"
  
      .prologue
+     .line 255
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
***************
*** 60,65 ****
--- 62,68 ----
      .parameter "x0"
  
      .prologue
+     .line 253
      invoke-virtual {p0, p1}, Landroid/net/wifi/SupplicantState$1;->createFromParcel(Landroid/os/Parcel;)Landroid/net/wifi/SupplicantState;
  
      move-result-object v0
***************
*** 72,77 ****
--- 75,81 ----
      .parameter "size"
  
      .prologue
+     .line 259
      new-array v0, p1, [Landroid/net/wifi/SupplicantState;
  
      return-object v0
***************
*** 82,87 ****
--- 86,92 ----
      .parameter "x0"
  
      .prologue
+     .line 253
      invoke-virtual {p0, p1}, Landroid/net/wifi/SupplicantState$1;->newArray(I)[Landroid/net/wifi/SupplicantState;
  
      move-result-object v0
