*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$3.smali	2012-03-23 19:36:57.182723709 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$3.smali.original	2012-03-23 19:34:45.437321335 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .parameter "x0"
  
      .prologue
+     .line 305
      iput-object p1, p0, Landroid/net/wifi/WifiWatchdogStateMachine$3;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      invoke-direct {p0, p2}, Landroid/database/ContentObserver;-><init>(Landroid/os/Handler;)V
***************
*** 39,49 ****
--- 40,52 ----
      .parameter "selfChange"
  
      .prologue
+     .line 308
      iget-object v0, p0, Landroid/net/wifi/WifiWatchdogStateMachine$3;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      const v1, 0x21006
  
      invoke-virtual {v0, v1}, Landroid/net/wifi/WifiWatchdogStateMachine;->sendMessage(I)V
  
+     .line 309
      return-void
  .end method
