*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DriverUnloadingState$1.smali	2012-03-23 19:36:58.729751920 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DriverUnloadingState$1.smali.original	2012-03-23 19:34:45.456321682 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 2077
      iput-object p1, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
      iput-object p2, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->val$message:Landroid/os/Message;
***************
*** 45,50 ****
--- 46,52 ----
      .locals 2
  
      .prologue
+     .line 2080
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
      iget-object v0, v0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 56,67 ****
--- 58,71 ----
  
      invoke-virtual {v0}, Landroid/os/PowerManager$WakeLock;->acquire()V
  
+     .line 2081
      invoke-static {}, Landroid/net/wifi/WifiNative;->unloadDriver()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 2083
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
      iget-object v0, v0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 70,81 ****
--- 74,87 ----
  
      invoke-virtual {v0, v1}, Landroid/net/wifi/WifiStateMachine;->sendMessage(I)V
  
+     .line 2085
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->val$message:Landroid/os/Message;
  
      iget v0, v0, Landroid/os/Message;->arg1:I
  
      sparse-switch v0, :sswitch_data_0
  
+     .line 2110
      :goto_0
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
***************
*** 88,95 ****
--- 94,103 ----
  
      invoke-virtual {v0}, Landroid/os/PowerManager$WakeLock;->release()V
  
+     .line 2111
      return-void
  
+     .line 2088
      :sswitch_0
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
***************
*** 104,109 ****
--- 112,118 ----
  
      goto :goto_0
  
+     .line 2092
      :sswitch_1
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
***************
*** 118,123 ****
--- 127,133 ----
  
      goto :goto_0
  
+     .line 2096
      :cond_0
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
***************
*** 128,133 ****
--- 138,144 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->loge(Ljava/lang/String;)V
      invoke-static {v0, v1}, Landroid/net/wifi/WifiStateMachine;->access$200(Landroid/net/wifi/WifiStateMachine;Ljava/lang/String;)V
  
+     .line 2097
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
      iget-object v0, v0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 136,141 ****
--- 147,153 ----
  
      invoke-virtual {v0, v1}, Landroid/net/wifi/WifiStateMachine;->sendMessage(I)V
  
+     .line 2099
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->val$message:Landroid/os/Message;
  
      iget v0, v0, Landroid/os/Message;->arg1:I
***************
*** 144,149 ****
--- 156,162 ----
  
      goto :goto_0
  
+     .line 2102
      :sswitch_2
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
***************
*** 156,161 ****
--- 169,175 ----
  
      goto :goto_0
  
+     .line 2106
      :sswitch_3
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverUnloadingState;
  
***************
*** 168,173 ****
--- 182,188 ----
  
      goto :goto_0
  
+     .line 2085
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_0
***************
*** 176,181 ****
--- 191,197 ----
          0xe -> :sswitch_1
      .end sparse-switch
  
+     .line 2099
      :sswitch_data_1
      .sparse-switch
          0x1 -> :sswitch_2
