*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiConfiguration.smali	2012-03-23 19:36:58.851754144 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiConfiguration.smali.original	2012-03-23 19:34:45.458321719 -0500
***************
*** 120,155 ****
      .locals 3
  
      .prologue
      const/4 v0, 0x4
  
      new-array v0, v0, [Ljava/lang/String;
  
      const/4 v1, 0x0
  
!     const-string v2, "wep_key0"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x1
  
!     const-string v2, "wep_key1"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x2
  
!     const-string v2, "wep_key2"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x3
  
!     const-string v2, "wep_key3"
  
      aput-object v2, v0, v1
  
      sput-object v0, Landroid/net/wifi/WifiConfiguration;->wepKeyVarNames:[Ljava/lang/String;
  
      new-instance v0, Landroid/net/wifi/WifiConfiguration$1;
  
      invoke-direct {v0}, Landroid/net/wifi/WifiConfiguration$1;-><init>()V
--- 120,157 ----
      .locals 3
  
      .prologue
+     .line 39
      const/4 v0, 0x4
  
      new-array v0, v0, [Ljava/lang/String;
  
      const/4 v1, 0x0
  
!     const-string/jumbo v2, "wep_key0"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x1
  
!     const-string/jumbo v2, "wep_key1"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x2
  
!     const-string/jumbo v2, "wep_key2"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x3
  
!     const-string/jumbo v2, "wep_key3"
  
      aput-object v2, v0, v1
  
      sput-object v0, Landroid/net/wifi/WifiConfiguration;->wepKeyVarNames:[Ljava/lang/String;
  
+     .line 589
      new-instance v0, Landroid/net/wifi/WifiConfiguration$1;
  
      invoke-direct {v0}, Landroid/net/wifi/WifiConfiguration$1;-><init>()V
***************
*** 169,176 ****
--- 171,180 ----
  
      const/4 v8, 0x0
  
+     .line 365
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 73
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v6, "eap"
***************
*** 179,192 ****
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->eap:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string v6, "phase2"
  
      invoke-direct {v5, p0, v6, v8}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->phase2:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v6, "identity"
--- 183,198 ----
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->eap:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 75
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string/jumbo v6, "phase2"
  
      invoke-direct {v5, p0, v6, v8}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->phase2:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 77
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v6, "identity"
***************
*** 195,200 ****
--- 201,207 ----
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->identity:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 79
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v6, "anonymous_identity"
***************
*** 203,216 ****
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->anonymous_identity:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string v6, "password"
  
      invoke-direct {v5, p0, v6, v8}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->password:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v6, "client_cert"
--- 210,225 ----
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->anonymous_identity:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 81
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string/jumbo v6, "password"
  
      invoke-direct {v5, p0, v6, v8}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->password:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 83
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v6, "client_cert"
***************
*** 219,232 ****
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->client_cert:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string v6, "private_key"
  
      invoke-direct {v5, p0, v6, v8}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->private_key:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v6, "ca_cert"
--- 228,243 ----
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->client_cert:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 85
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string/jumbo v6, "private_key"
  
      invoke-direct {v5, p0, v6, v8}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->private_key:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 87
      new-instance v5, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v6, "ca_cert"
***************
*** 235,240 ****
--- 246,252 ----
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->ca_cert:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 90
      const/16 v5, 0x8
  
      new-array v5, v5, [Landroid/net/wifi/WifiConfiguration$EnterpriseField;
***************
*** 285,338 ****
--- 297,363 ----
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->enterpriseFields:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 366
      const/4 v5, -0x1
  
      iput v5, p0, Landroid/net/wifi/WifiConfiguration;->networkId:I
  
+     .line 367
      iput-object v8, p0, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
  
+     .line 368
      iput-object v8, p0, Landroid/net/wifi/WifiConfiguration;->BSSID:Ljava/lang/String;
  
+     .line 369
      iput v9, p0, Landroid/net/wifi/WifiConfiguration;->priority:I
  
+     .line 370
      iput-boolean v9, p0, Landroid/net/wifi/WifiConfiguration;->hiddenSSID:Z
  
+     .line 371
      iput v9, p0, Landroid/net/wifi/WifiConfiguration;->disableReason:I
  
+     .line 372
      new-instance v5, Ljava/util/BitSet;
  
      invoke-direct {v5}, Ljava/util/BitSet;-><init>()V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
+     .line 373
      new-instance v5, Ljava/util/BitSet;
  
      invoke-direct {v5}, Ljava/util/BitSet;-><init>()V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedProtocols:Ljava/util/BitSet;
  
+     .line 374
      new-instance v5, Ljava/util/BitSet;
  
      invoke-direct {v5}, Ljava/util/BitSet;-><init>()V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedAuthAlgorithms:Ljava/util/BitSet;
  
+     .line 375
      new-instance v5, Ljava/util/BitSet;
  
      invoke-direct {v5}, Ljava/util/BitSet;-><init>()V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedPairwiseCiphers:Ljava/util/BitSet;
  
+     .line 376
      new-instance v5, Ljava/util/BitSet;
  
      invoke-direct {v5}, Ljava/util/BitSet;-><init>()V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedGroupCiphers:Ljava/util/BitSet;
  
+     .line 377
      new-array v5, v10, [Ljava/lang/String;
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->wepKeys:[Ljava/lang/String;
  
+     .line 378
      const/4 v2, 0x0
  
      .local v2, i:I
***************
*** 343,356 ****
--- 368,384 ----
  
      if-ge v2, v5, :cond_0
  
+     .line 379
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->wepKeys:[Ljava/lang/String;
  
      aput-object v8, v5, v2
  
+     .line 378
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 380
      :cond_0
      iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->enterpriseFields:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
***************
*** 366,394 ****
--- 394,428 ----
  
      aget-object v1, v0, v3
  
+     .line 381
      .local v1, field:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      invoke-virtual {v1, v8}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;->setValue(Ljava/lang/String;)V
  
+     .line 380
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_1
  
+     .line 383
      .end local v1           #field:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      :cond_1
      sget-object v5, Landroid/net/wifi/WifiConfiguration$IpAssignment;->UNASSIGNED:Landroid/net/wifi/WifiConfiguration$IpAssignment;
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->ipAssignment:Landroid/net/wifi/WifiConfiguration$IpAssignment;
  
+     .line 384
      sget-object v5, Landroid/net/wifi/WifiConfiguration$ProxySettings;->UNASSIGNED:Landroid/net/wifi/WifiConfiguration$ProxySettings;
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->proxySettings:Landroid/net/wifi/WifiConfiguration$ProxySettings;
  
+     .line 385
      new-instance v5, Landroid/net/LinkProperties;
  
      invoke-direct {v5}, Landroid/net/LinkProperties;-><init>()V
  
      iput-object v5, p0, Landroid/net/wifi/WifiConfiguration;->linkProperties:Landroid/net/LinkProperties;
  
+     .line 386
      return-void
  .end method
  
***************
*** 401,408 ****
--- 435,444 ----
  
      const/4 v3, 0x0
  
+     .line 529
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 73
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v2, "eap"
***************
*** 411,424 ****
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->eap:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string v2, "phase2"
  
      invoke-direct {v1, p0, v2, v3}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->phase2:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v2, "identity"
--- 447,462 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->eap:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 75
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string/jumbo v2, "phase2"
  
      invoke-direct {v1, p0, v2, v3}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->phase2:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 77
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v2, "identity"
***************
*** 427,432 ****
--- 465,471 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->identity:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 79
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v2, "anonymous_identity"
***************
*** 435,448 ****
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->anonymous_identity:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string v2, "password"
  
      invoke-direct {v1, p0, v2, v3}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->password:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v2, "client_cert"
--- 474,489 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->anonymous_identity:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 81
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string/jumbo v2, "password"
  
      invoke-direct {v1, p0, v2, v3}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->password:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 83
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v2, "client_cert"
***************
*** 451,464 ****
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->client_cert:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string v2, "private_key"
  
      invoke-direct {v1, p0, v2, v3}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->private_key:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v2, "ca_cert"
--- 492,507 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->client_cert:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 85
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
!     const-string/jumbo v2, "private_key"
  
      invoke-direct {v1, p0, v2, v3}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;-><init>(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Landroid/net/wifi/WifiConfiguration$1;)V
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->private_key:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 87
      new-instance v1, Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      const-string v2, "ca_cert"
***************
*** 467,472 ****
--- 510,516 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->ca_cert:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 90
      const/16 v1, 0x8
  
      new-array v1, v1, [Landroid/net/wifi/WifiConfiguration$EnterpriseField;
***************
*** 519,554 ****
--- 563,607 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->enterpriseFields:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
+     .line 530
      if-eqz p1, :cond_2
  
+     .line 531
      iget v1, p1, Landroid/net/wifi/WifiConfiguration;->networkId:I
  
      iput v1, p0, Landroid/net/wifi/WifiConfiguration;->networkId:I
  
+     .line 532
      iget v1, p1, Landroid/net/wifi/WifiConfiguration;->status:I
  
      iput v1, p0, Landroid/net/wifi/WifiConfiguration;->status:I
  
+     .line 533
      iget v1, p1, Landroid/net/wifi/WifiConfiguration;->disableReason:I
  
      iput v1, p0, Landroid/net/wifi/WifiConfiguration;->disableReason:I
  
+     .line 534
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
  
+     .line 535
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->BSSID:Ljava/lang/String;
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->BSSID:Ljava/lang/String;
  
+     .line 536
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
  
+     .line 538
      new-array v1, v4, [Ljava/lang/String;
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->wepKeys:[Ljava/lang/String;
  
+     .line 539
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 559,564 ****
--- 612,618 ----
  
      if-ge v0, v1, :cond_0
  
+     .line 540
      iget-object v1, p0, Landroid/net/wifi/WifiConfiguration;->wepKeys:[Ljava/lang/String;
  
      iget-object v2, p1, Landroid/net/wifi/WifiConfiguration;->wepKeys:[Ljava/lang/String;
***************
*** 567,589 ****
--- 621,648 ----
  
      aput-object v2, v1, v0
  
+     .line 539
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 542
      :cond_0
      iget v1, p1, Landroid/net/wifi/WifiConfiguration;->wepTxKeyIndex:I
  
      iput v1, p0, Landroid/net/wifi/WifiConfiguration;->wepTxKeyIndex:I
  
+     .line 543
      iget v1, p1, Landroid/net/wifi/WifiConfiguration;->priority:I
  
      iput v1, p0, Landroid/net/wifi/WifiConfiguration;->priority:I
  
+     .line 544
      iget-boolean v1, p1, Landroid/net/wifi/WifiConfiguration;->hiddenSSID:Z
  
      iput-boolean v1, p0, Landroid/net/wifi/WifiConfiguration;->hiddenSSID:Z
  
+     .line 545
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
      invoke-virtual {v1}, Ljava/util/BitSet;->clone()Ljava/lang/Object;
***************
*** 594,599 ****
--- 653,659 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
+     .line 546
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->allowedProtocols:Ljava/util/BitSet;
  
      invoke-virtual {v1}, Ljava/util/BitSet;->clone()Ljava/lang/Object;
***************
*** 604,609 ****
--- 664,670 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->allowedProtocols:Ljava/util/BitSet;
  
+     .line 547
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->allowedAuthAlgorithms:Ljava/util/BitSet;
  
      invoke-virtual {v1}, Ljava/util/BitSet;->clone()Ljava/lang/Object;
***************
*** 614,619 ****
--- 675,681 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->allowedAuthAlgorithms:Ljava/util/BitSet;
  
+     .line 548
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->allowedPairwiseCiphers:Ljava/util/BitSet;
  
      invoke-virtual {v1}, Ljava/util/BitSet;->clone()Ljava/lang/Object;
***************
*** 624,629 ****
--- 686,692 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->allowedPairwiseCiphers:Ljava/util/BitSet;
  
+     .line 549
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->allowedGroupCiphers:Ljava/util/BitSet;
  
      invoke-virtual {v1}, Ljava/util/BitSet;->clone()Ljava/lang/Object;
***************
*** 634,639 ****
--- 697,703 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->allowedGroupCiphers:Ljava/util/BitSet;
  
+     .line 551
      const/4 v0, 0x0
  
      :goto_1
***************
*** 643,648 ****
--- 707,713 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 552
      iget-object v1, p0, Landroid/net/wifi/WifiConfiguration;->enterpriseFields:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      aget-object v1, v1, v0
***************
*** 657,675 ****
--- 722,744 ----
  
      invoke-virtual {v1, v2}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;->setValue(Ljava/lang/String;)V
  
+     .line 551
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 554
      :cond_1
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->ipAssignment:Landroid/net/wifi/WifiConfiguration$IpAssignment;
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->ipAssignment:Landroid/net/wifi/WifiConfiguration$IpAssignment;
  
+     .line 555
      iget-object v1, p1, Landroid/net/wifi/WifiConfiguration;->proxySettings:Landroid/net/wifi/WifiConfiguration$ProxySettings;
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->proxySettings:Landroid/net/wifi/WifiConfiguration$ProxySettings;
  
+     .line 556
      new-instance v1, Landroid/net/LinkProperties;
  
      iget-object v2, p1, Landroid/net/wifi/WifiConfiguration;->linkProperties:Landroid/net/LinkProperties;
***************
*** 678,683 ****
--- 747,753 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->linkProperties:Landroid/net/LinkProperties;
  
+     .line 558
      .end local v0           #i:I
      :cond_2
      return-void
***************
*** 688,693 ****
--- 758,764 ----
      .parameter "x0"
  
      .prologue
+     .line 30
      invoke-static {p0}, Landroid/net/wifi/WifiConfiguration;->readBitSet(Landroid/os/Parcel;)Ljava/util/BitSet;
  
      move-result-object v0
***************
*** 700,714 ****
--- 771,788 ----
      .parameter "src"
  
      .prologue
+     .line 491
      invoke-virtual {p0}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 493
      .local v0, cardinality:I
      new-instance v2, Ljava/util/BitSet;
  
      invoke-direct {v2}, Ljava/util/BitSet;-><init>()V
  
+     .line 494
      .local v2, set:Ljava/util/BitSet;
      const/4 v1, 0x0
  
***************
*** 716,731 ****
--- 790,808 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 495
      invoke-virtual {p0}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      invoke-virtual {v2, v3}, Ljava/util/BitSet;->set(I)V
  
+     .line 494
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 497
      :cond_0
      return-object v2
  .end method
***************
*** 736,743 ****
--- 813,822 ----
      .parameter "set"
  
      .prologue
+     .line 501
      const/4 v0, -0x1
  
+     .line 503
      .local v0, nextSetBit:I
      invoke-virtual {p1}, Ljava/util/BitSet;->cardinality()I
  
***************
*** 745,750 ****
--- 824,830 ----
  
      invoke-virtual {p0, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 505
      :goto_0
      add-int/lit8 v1, v0, 0x1
  
***************
*** 756,765 ****
--- 836,847 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 506
      invoke-virtual {p0, v0}, Landroid/os/Parcel;->writeInt(I)V
  
      goto :goto_0
  
+     .line 507
      :cond_0
      return-void
  .end method
***************
*** 770,775 ****
--- 852,858 ----
      .locals 1
  
      .prologue
+     .line 525
      const/4 v0, 0x0
  
      return v0
***************
*** 787,792 ****
--- 870,876 ----
  
      const/4 v0, 0x1
  
+     .line 511
      iget-object v4, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
      invoke-virtual {v4, v0}, Ljava/util/BitSet;->get(I)Z
***************
*** 795,803 ****
--- 879,889 ----
  
      if-eqz v4, :cond_0
  
+     .line 520
      :goto_0
      return v0
  
+     .line 513
      :cond_0
      iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
***************
*** 809,816 ****
--- 895,904 ----
  
      move v0, v1
  
+     .line 514
      goto :goto_0
  
+     .line 515
      :cond_1
      iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
***************
*** 822,829 ****
--- 910,919 ----
  
      move v0, v2
  
+     .line 516
      goto :goto_0
  
+     .line 517
      :cond_2
      iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
***************
*** 835,842 ****
--- 925,934 ----
  
      move v0, v3
  
+     .line 518
      goto :goto_0
  
+     .line 520
      :cond_3
      const/4 v0, 0x0
  
***************
*** 849,867 ****
--- 941,963 ----
      .prologue
      const/16 v14, 0xa
  
+     .line 390
      new-instance v9, Ljava/lang/StringBuilder;
  
      invoke-direct {v9}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 391
      .local v9, sbuf:Ljava/lang/StringBuilder;
      iget v11, p0, Landroid/net/wifi/WifiConfiguration;->status:I
  
      if-nez v11, :cond_2
  
+     .line 392
      const-string v11, "* "
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 396
      :cond_0
      :goto_0
      const-string v11, "ID: "
***************
*** 914,923 ****
--- 1010,1021 ----
  
      invoke-virtual {v11, v14}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 399
      const-string v11, " KeyMgmt:"
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 400
      const/4 v5, 0x0
  
      .local v5, k:I
***************
*** 930,935 ****
--- 1028,1034 ----
  
      if-ge v5, v11, :cond_4
  
+     .line 401
      iget-object v11, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
      invoke-virtual {v11, v5}, Ljava/util/BitSet;->get(I)Z
***************
*** 938,965 ****
--- 1037,1069 ----
  
      if-eqz v11, :cond_1
  
+     .line 402
      const-string v11, " "
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 403
      sget-object v11, Landroid/net/wifi/WifiConfiguration$KeyMgmt;->strings:[Ljava/lang/String;
  
      array-length v11, v11
  
      if-ge v5, v11, :cond_3
  
+     .line 404
      sget-object v11, Landroid/net/wifi/WifiConfiguration$KeyMgmt;->strings:[Ljava/lang/String;
  
      aget-object v11, v11, v5
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 400
      :cond_1
      :goto_2
      add-int/lit8 v5, v5, 0x1
  
      goto :goto_1
  
+     .line 393
      .end local v5           #k:I
      :cond_2
      iget v11, p0, Landroid/net/wifi/WifiConfiguration;->status:I
***************
*** 968,973 ****
--- 1072,1078 ----
  
      if-ne v11, v12, :cond_0
  
+     .line 394
      const-string v11, "- DSBLE: "
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 986,991 ****
--- 1091,1097 ----
  
      goto :goto_0
  
+     .line 406
      .restart local v5       #k:I
      :cond_3
      const-string v11, "??"
***************
*** 994,1004 ****
--- 1100,1112 ----
  
      goto :goto_2
  
+     .line 410
      :cond_4
      const-string v11, " Protocols:"
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 411
      const/4 v7, 0x0
  
      .local v7, p:I
***************
*** 1011,1016 ****
--- 1119,1125 ----
  
      if-ge v7, v11, :cond_7
  
+     .line 412
      iget-object v11, p0, Landroid/net/wifi/WifiConfiguration;->allowedProtocols:Ljava/util/BitSet;
  
      invoke-virtual {v11, v7}, Ljava/util/BitSet;->get(I)Z
***************
*** 1019,1046 ****
--- 1128,1160 ----
  
      if-eqz v11, :cond_5
  
+     .line 413
      const-string v11, " "
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 414
      sget-object v11, Landroid/net/wifi/WifiConfiguration$Protocol;->strings:[Ljava/lang/String;
  
      array-length v11, v11
  
      if-ge v7, v11, :cond_6
  
+     .line 415
      sget-object v11, Landroid/net/wifi/WifiConfiguration$Protocol;->strings:[Ljava/lang/String;
  
      aget-object v11, v11, v7
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 411
      :cond_5
      :goto_4
      add-int/lit8 v7, v7, 0x1
  
      goto :goto_3
  
+     .line 417
      :cond_6
      const-string v11, "??"
  
***************
*** 1048,1060 ****
--- 1162,1177 ----
  
      goto :goto_4
  
+     .line 421
      :cond_7
      invoke-virtual {v9, v14}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 422
      const-string v11, " AuthAlgorithms:"
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 423
      const/4 v0, 0x0
  
      .local v0, a:I
***************
*** 1067,1072 ****
--- 1184,1190 ----
  
      if-ge v0, v11, :cond_a
  
+     .line 424
      iget-object v11, p0, Landroid/net/wifi/WifiConfiguration;->allowedAuthAlgorithms:Ljava/util/BitSet;
  
      invoke-virtual {v11, v0}, Ljava/util/BitSet;->get(I)Z
***************
*** 1075,1102 ****
--- 1193,1225 ----
  
      if-eqz v11, :cond_8
  
+     .line 425
      const-string v11, " "
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 426
      sget-object v11, Landroid/net/wifi/WifiConfiguration$AuthAlgorithm;->strings:[Ljava/lang/String;
  
      array-length v11, v11
  
      if-ge v0, v11, :cond_9
  
+     .line 427
      sget-object v11, Landroid/net/wifi/WifiConfiguration$AuthAlgorithm;->strings:[Ljava/lang/String;
  
      aget-object v11, v11, v0
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 423
      :cond_8
      :goto_6
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_5
  
+     .line 429
      :cond_9
      const-string v11, "??"
  
***************
*** 1104,1116 ****
--- 1227,1242 ----
  
      goto :goto_6
  
+     .line 433
      :cond_a
      invoke-virtual {v9, v14}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 434
      const-string v11, " PairwiseCiphers:"
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 435
      const/4 v8, 0x0
  
      .local v8, pc:I
***************
*** 1123,1128 ****
--- 1249,1255 ----
  
      if-ge v8, v11, :cond_d
  
+     .line 436
      iget-object v11, p0, Landroid/net/wifi/WifiConfiguration;->allowedPairwiseCiphers:Ljava/util/BitSet;
  
      invoke-virtual {v11, v8}, Ljava/util/BitSet;->get(I)Z
***************
*** 1131,1158 ****
--- 1258,1290 ----
  
      if-eqz v11, :cond_b
  
+     .line 437
      const-string v11, " "
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 438
      sget-object v11, Landroid/net/wifi/WifiConfiguration$PairwiseCipher;->strings:[Ljava/lang/String;
  
      array-length v11, v11
  
      if-ge v8, v11, :cond_c
  
+     .line 439
      sget-object v11, Landroid/net/wifi/WifiConfiguration$PairwiseCipher;->strings:[Ljava/lang/String;
  
      aget-object v11, v11, v8
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 435
      :cond_b
      :goto_8
      add-int/lit8 v8, v8, 0x1
  
      goto :goto_7
  
+     .line 441
      :cond_c
      const-string v11, "??"
  
***************
*** 1160,1172 ****
--- 1292,1307 ----
  
      goto :goto_8
  
+     .line 445
      :cond_d
      invoke-virtual {v9, v14}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 446
      const-string v11, " GroupCiphers:"
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 447
      const/4 v3, 0x0
  
      .local v3, gc:I
***************
*** 1179,1184 ****
--- 1314,1320 ----
  
      if-ge v3, v11, :cond_10
  
+     .line 448
      iget-object v11, p0, Landroid/net/wifi/WifiConfiguration;->allowedGroupCiphers:Ljava/util/BitSet;
  
      invoke-virtual {v11, v3}, Ljava/util/BitSet;->get(I)Z
***************
*** 1187,1214 ****
--- 1323,1355 ----
  
      if-eqz v11, :cond_e
  
+     .line 449
      const-string v11, " "
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 450
      sget-object v11, Landroid/net/wifi/WifiConfiguration$GroupCipher;->strings:[Ljava/lang/String;
  
      array-length v11, v11
  
      if-ge v3, v11, :cond_f
  
+     .line 451
      sget-object v11, Landroid/net/wifi/WifiConfiguration$GroupCipher;->strings:[Ljava/lang/String;
  
      aget-object v11, v11, v3
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 447
      :cond_e
      :goto_a
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_9
  
+     .line 453
      :cond_f
      const-string v11, "??"
  
***************
*** 1216,1221 ****
--- 1357,1363 ----
  
      goto :goto_a
  
+     .line 457
      :cond_10
      invoke-virtual {v9, v14}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
***************
*** 1225,1238 ****
--- 1367,1383 ----
  
      invoke-virtual {v11, v12}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 458
      iget-object v11, p0, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
  
      if-eqz v11, :cond_11
  
+     .line 459
      const/16 v11, 0x2a
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 462
      :cond_11
      iget-object v1, p0, Landroid/net/wifi/WifiConfiguration;->enterpriseFields:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
***************
*** 1248,1253 ****
--- 1393,1399 ----
  
      aget-object v2, v1, v4
  
+     .line 463
      .local v2, field:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      invoke-virtual {v9, v14}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
***************
*** 1283,1307 ****
--- 1429,1458 ----
  
      invoke-virtual {v11, v12}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 464
      invoke-virtual {v2}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;->value()Ljava/lang/String;
  
      move-result-object v10
  
+     .line 465
      .local v10, value:Ljava/lang/String;
      if-eqz v10, :cond_12
  
      invoke-virtual {v9, v10}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 462
      :cond_12
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_b
  
+     .line 467
      .end local v2           #field:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      .end local v10           #value:Ljava/lang/String;
      :cond_13
      invoke-virtual {v9, v14}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 468
      new-instance v11, Ljava/lang/StringBuilder;
  
      invoke-direct {v11}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1328,1337 ****
--- 1479,1490 ----
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 469
      const-string v11, "\n"
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 470
      new-instance v11, Ljava/lang/StringBuilder;
  
      invoke-direct {v11}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1358,1367 ****
--- 1511,1522 ----
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 471
      const-string v11, "\n"
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 472
      iget-object v11, p0, Landroid/net/wifi/WifiConfiguration;->linkProperties:Landroid/net/LinkProperties;
  
      invoke-virtual {v11}, Landroid/net/LinkProperties;->toString()Ljava/lang/String;
***************
*** 1370,1379 ****
--- 1525,1536 ----
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 473
      const-string v11, "\n"
  
      invoke-virtual {v9, v11}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 475
      invoke-virtual {v9}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v11
***************
*** 1387,1416 ****
--- 1544,1580 ----
      .parameter "flags"
  
      .prologue
+     .line 562
      iget v5, p0, Landroid/net/wifi/WifiConfiguration;->networkId:I
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 563
      iget v5, p0, Landroid/net/wifi/WifiConfiguration;->status:I
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 564
      iget v5, p0, Landroid/net/wifi/WifiConfiguration;->disableReason:I
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 565
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 566
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->BSSID:Ljava/lang/String;
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 567
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 568
      iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->wepKeys:[Ljava/lang/String;
  
      .local v0, arr$:[Ljava/lang/String;
***************
*** 1425,1447 ****
--- 1589,1616 ----
  
      aget-object v4, v0, v2
  
+     .line 569
      .local v4, wepKey:Ljava/lang/String;
      invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 568
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 570
      .end local v4           #wepKey:Ljava/lang/String;
      :cond_0
      iget v5, p0, Landroid/net/wifi/WifiConfiguration;->wepTxKeyIndex:I
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 571
      iget v5, p0, Landroid/net/wifi/WifiConfiguration;->priority:I
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 572
      iget-boolean v5, p0, Landroid/net/wifi/WifiConfiguration;->hiddenSSID:Z
  
      if-eqz v5, :cond_1
***************
*** 1451,1476 ****
--- 1620,1651 ----
      :goto_1
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 574
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
      invoke-static {p1, v5}, Landroid/net/wifi/WifiConfiguration;->writeBitSet(Landroid/os/Parcel;Ljava/util/BitSet;)V
  
+     .line 575
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedProtocols:Ljava/util/BitSet;
  
      invoke-static {p1, v5}, Landroid/net/wifi/WifiConfiguration;->writeBitSet(Landroid/os/Parcel;Ljava/util/BitSet;)V
  
+     .line 576
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedAuthAlgorithms:Ljava/util/BitSet;
  
      invoke-static {p1, v5}, Landroid/net/wifi/WifiConfiguration;->writeBitSet(Landroid/os/Parcel;Ljava/util/BitSet;)V
  
+     .line 577
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedPairwiseCiphers:Ljava/util/BitSet;
  
      invoke-static {p1, v5}, Landroid/net/wifi/WifiConfiguration;->writeBitSet(Landroid/os/Parcel;Ljava/util/BitSet;)V
  
+     .line 578
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->allowedGroupCiphers:Ljava/util/BitSet;
  
      invoke-static {p1, v5}, Landroid/net/wifi/WifiConfiguration;->writeBitSet(Landroid/os/Parcel;Ljava/util/BitSet;)V
  
+     .line 580
      iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->enterpriseFields:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      .local v0, arr$:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
***************
*** 1483,1488 ****
--- 1658,1664 ----
  
      aget-object v1, v0, v2
  
+     .line 581
      .local v1, field:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      invoke-virtual {v1}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;->value()Ljava/lang/String;
  
***************
*** 1490,1499 ****
--- 1666,1677 ----
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 580
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_2
  
+     .line 572
      .end local v1           #field:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      .local v0, arr$:[Ljava/lang/String;
      :cond_1
***************
*** 1501,1506 ****
--- 1679,1685 ----
  
      goto :goto_1
  
+     .line 583
      .local v0, arr$:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      :cond_2
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->ipAssignment:Landroid/net/wifi/WifiConfiguration$IpAssignment;
***************
*** 1511,1516 ****
--- 1690,1696 ----
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 584
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->proxySettings:Landroid/net/wifi/WifiConfiguration$ProxySettings;
  
      invoke-virtual {v5}, Landroid/net/wifi/WifiConfiguration$ProxySettings;->name()Ljava/lang/String;
***************
*** 1519,1527 ****
--- 1699,1709 ----
  
      invoke-virtual {p1, v5}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 585
      iget-object v5, p0, Landroid/net/wifi/WifiConfiguration;->linkProperties:Landroid/net/LinkProperties;
  
      invoke-virtual {p1, v5, p2}, Landroid/os/Parcel;->writeParcelable(Landroid/os/Parcelable;I)V
  
+     .line 586
      return-void
  .end method
