*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiApConfigStore$ActiveState.smali	2012-03-23 19:36:58.356745119 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiApConfigStore$ActiveState.smali.original	2012-03-23 19:34:45.450321573 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 113
      iput-object p1, p0, Landroid/net/wifi/WifiApConfigStore$ActiveState;->this$0:Landroid/net/wifi/WifiApConfigStore;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .locals 2
  
      .prologue
+     .line 115
      new-instance v0, Ljava/lang/Thread;
  
      new-instance v1, Landroid/net/wifi/WifiApConfigStore$ActiveState$1;
***************
*** 47,52 ****
--- 49,55 ----
  
      invoke-virtual {v0}, Ljava/lang/Thread;->start()V
  
+     .line 121
      return-void
  .end method
  
***************
*** 55,80 ****
--- 58,89 ----
      .parameter "message"
  
      .prologue
+     .line 124
      iget v0, p1, Landroid/os/Message;->what:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 134
      const/4 v0, 0x0
  
+     .line 136
      :goto_0
      return v0
  
+     .line 128
      :pswitch_0
      iget-object v0, p0, Landroid/net/wifi/WifiApConfigStore$ActiveState;->this$0:Landroid/net/wifi/WifiApConfigStore;
  
      #calls: Landroid/net/wifi/WifiApConfigStore;->deferMessage(Landroid/os/Message;)V
      invoke-static {v0, p1}, Landroid/net/wifi/WifiApConfigStore;->access$500(Landroid/net/wifi/WifiApConfigStore;Landroid/os/Message;)V
  
+     .line 136
      :goto_1
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 131
      :pswitch_1
      iget-object v0, p0, Landroid/net/wifi/WifiApConfigStore$ActiveState;->this$0:Landroid/net/wifi/WifiApConfigStore;
  
***************
*** 90,95 ****
--- 99,105 ----
  
      goto :goto_1
  
+     .line 124
      :pswitch_data_0
      .packed-switch 0x20019
          :pswitch_0
