*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DriverStartingState.smali	2012-03-23 19:36:57.527730002 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DriverStartingState.smali.original	2012-03-23 19:34:45.440321391 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 2449
      iput-object p1, p0, Landroid/net/wifi/WifiStateMachine$DriverStartingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .locals 2
  
      .prologue
+     .line 2453
      const v0, 0xc365
  
      invoke-virtual {p0}, Landroid/net/wifi/WifiStateMachine$DriverStartingState;->getName()Ljava/lang/String;
***************
*** 45,50 ****
--- 47,53 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(ILjava/lang/String;)I
  
+     .line 2454
      return-void
  .end method
  
***************
*** 53,67 ****
--- 56,74 ----
      .parameter "message"
  
      .prologue
+     .line 2458
      iget v1, p1, Landroid/os/Message;->what:I
  
      sparse-switch v1, :sswitch_data_0
  
+     .line 2489
      const/4 v1, 0x0
  
+     .line 2492
      :goto_0
      return v1
  
+     .line 2460
      :sswitch_0
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$DriverStartingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 70,75 ****
--- 77,83 ----
  
      move-result-object v0
  
+     .line 2465
      .local v0, state:Landroid/net/wifi/SupplicantState;
      invoke-static {v0}, Landroid/net/wifi/SupplicantState;->isDriverActive(Landroid/net/wifi/SupplicantState;)Z
  
***************
*** 77,82 ****
--- 85,91 ----
  
      if-eqz v1, :cond_0
  
+     .line 2466
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$DriverStartingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$DriverStartingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 89,94 ****
--- 98,104 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
      invoke-static {v1, v2}, Landroid/net/wifi/WifiStateMachine;->access$6500(Landroid/net/wifi/WifiStateMachine;Lcom/android/internal/util/IState;)V
  
+     .line 2491
      .end local v0           #state:Landroid/net/wifi/SupplicantState;
      :cond_0
      :goto_1
***************
*** 98,107 ****
--- 108,119 ----
  
      invoke-static {v1, v2}, Landroid/util/EventLog;->writeEvent(II)I
  
+     .line 2492
      const/4 v1, 0x1
  
      goto :goto_0
  
+     .line 2486
      :sswitch_1
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$DriverStartingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 110,115 ****
--- 122,128 ----
  
      goto :goto_1
  
+     .line 2458
      :sswitch_data_0
      .sparse-switch
          0x2000d -> :sswitch_1
