*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$ConnectingState.smali	2012-03-23 19:36:58.176741836 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$ConnectingState.smali.original	2012-03-23 19:34:45.448321536 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 2890
      iput-object p1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 39,44 ****
--- 40,46 ----
      .prologue
      const v7, 0x20010
  
+     .line 2895
      const v4, 0xc365
  
      invoke-virtual {p0}, Landroid/net/wifi/WifiStateMachine$ConnectingState;->getName()Ljava/lang/String;
***************
*** 47,52 ****
--- 49,55 ----
  
      invoke-static {v4, v5}, Landroid/util/EventLog;->writeEvent(ILjava/lang/String;)I
  
+     .line 2898
      :try_start_0
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 67,72 ****
--- 70,76 ----
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 2905
      :goto_0
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 81,86 ****
--- 85,91 ----
  
      if-nez v4, :cond_0
  
+     .line 2907
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iget-object v5, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 106,111 ****
--- 111,117 ----
      #setter for: Landroid/net/wifi/WifiStateMachine;->mDhcpStateMachine:Landroid/net/DhcpStateMachine;
      invoke-static {v4, v5}, Landroid/net/wifi/WifiStateMachine;->access$10602(Landroid/net/wifi/WifiStateMachine;Landroid/net/DhcpStateMachine;)Landroid/net/DhcpStateMachine;
  
+     .line 2909
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #getter for: Landroid/net/wifi/WifiStateMachine;->mDhcpStateMachine:Landroid/net/DhcpStateMachine;
***************
*** 115,120 ****
--- 121,127 ----
  
      invoke-virtual {v4}, Landroid/net/DhcpStateMachine;->registerForPreDhcpNotification()V
  
+     .line 2910
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #getter for: Landroid/net/wifi/WifiStateMachine;->mDhcpStateMachine:Landroid/net/DhcpStateMachine;
***************
*** 126,137 ****
--- 133,147 ----
  
      invoke-virtual {v4, v5}, Landroid/net/DhcpStateMachine;->sendMessage(I)V
  
+     .line 2929
      :goto_1
      return-void
  
+     .line 2899
      :catch_0
      move-exception v3
  
+     .line 2900
      .local v3, re:Landroid/os/RemoteException;
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 158,167 ****
--- 168,179 ----
  
      goto :goto_0
  
+     .line 2901
      .end local v3           #re:Landroid/os/RemoteException;
      :catch_1
      move-exception v1
  
+     .line 2902
      .local v1, e:Ljava/lang/IllegalStateException;
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 188,193 ****
--- 200,206 ----
  
      goto :goto_0
  
+     .line 2912
      .end local v1           #e:Ljava/lang/IllegalStateException;
      :cond_0
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 201,211 ****
--- 214,226 ----
  
      move-result-object v0
  
+     .line 2914
      .local v0, dhcpInfoInternal:Landroid/net/DhcpInfoInternal;
      new-instance v2, Landroid/net/InterfaceConfiguration;
  
      invoke-direct {v2}, Landroid/net/InterfaceConfiguration;-><init>()V
  
+     .line 2915
      .local v2, ifcg:Landroid/net/InterfaceConfiguration;
      invoke-virtual {v0}, Landroid/net/DhcpInfoInternal;->makeLinkAddress()Landroid/net/LinkAddress;
  
***************
*** 213,222 ****
--- 228,239 ----
  
      iput-object v4, v2, Landroid/net/InterfaceConfiguration;->addr:Landroid/net/LinkAddress;
  
+     .line 2916
      const-string v4, "[up]"
  
      iput-object v4, v2, Landroid/net/InterfaceConfiguration;->interfaceFlags:Ljava/lang/String;
  
+     .line 2918
      :try_start_1
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 234,239 ****
--- 251,257 ----
  
      invoke-interface {v4, v5, v2}, Landroid/os/INetworkManagementService;->setInterfaceConfig(Ljava/lang/String;Landroid/net/InterfaceConfiguration;)V
  
+     .line 2920
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      const v5, 0x2000f
***************
*** 245,253 ****
--- 263,273 ----
  
      goto :goto_1
  
+     .line 2921
      :catch_2
      move-exception v3
  
+     .line 2922
      .restart local v3       #re:Landroid/os/RemoteException;
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 272,287 ****
--- 292,310 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->loge(Ljava/lang/String;)V
      invoke-static {v4, v5}, Landroid/net/wifi/WifiStateMachine;->access$200(Landroid/net/wifi/WifiStateMachine;Ljava/lang/String;)V
  
+     .line 2923
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-virtual {v4, v7}, Landroid/net/wifi/WifiStateMachine;->sendMessage(I)V
  
      goto/16 :goto_1
  
+     .line 2924
      .end local v3           #re:Landroid/os/RemoteException;
      :catch_3
      move-exception v1
  
+     .line 2925
      .restart local v1       #e:Ljava/lang/IllegalStateException;
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 306,311 ****
--- 329,335 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->loge(Ljava/lang/String;)V
      invoke-static {v4, v5}, Landroid/net/wifi/WifiStateMachine;->access$200(Landroid/net/wifi/WifiStateMachine;Ljava/lang/String;)V
  
+     .line 2926
      iget-object v4, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-virtual {v4, v7}, Landroid/net/wifi/WifiStateMachine;->sendMessage(I)V
***************
*** 324,342 ****
--- 348,370 ----
  
      const/4 v1, 0x0
  
+     .line 2934
      iget v3, p1, Landroid/os/Message;->what:I
  
      sparse-switch v3, :sswitch_data_0
  
+     .line 2992
      :cond_0
      :goto_0
      return v1
  
+     .line 2936
      :sswitch_0
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-virtual {v1}, Landroid/net/wifi/WifiStateMachine;->handlePreDhcpSetup()V
  
+     .line 2937
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #getter for: Landroid/net/wifi/WifiStateMachine;->mDhcpStateMachine:Landroid/net/DhcpStateMachine;
***************
*** 348,353 ****
--- 376,382 ----
  
      invoke-virtual {v1, v3}, Landroid/net/DhcpStateMachine;->sendMessage(I)V
  
+     .line 2991
      :cond_1
      :goto_1
      :sswitch_1
***************
*** 359,375 ****
--- 388,408 ----
  
      move v1, v2
  
+     .line 2992
      goto :goto_0
  
+     .line 2940
      :sswitch_2
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-virtual {v1}, Landroid/net/wifi/WifiStateMachine;->handlePostDhcpSetup()V
  
+     .line 2941
      iget v1, p1, Landroid/os/Message;->arg1:I
  
      if-ne v1, v2, :cond_2
  
+     .line 2942
      iget-object v3, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iget-object v1, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
***************
*** 379,384 ****
--- 412,418 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->handleSuccessfulIpConfiguration(Landroid/net/DhcpInfoInternal;)V
      invoke-static {v3, v1}, Landroid/net/wifi/WifiStateMachine;->access$10700(Landroid/net/wifi/WifiStateMachine;Landroid/net/DhcpInfoInternal;)V
  
+     .line 2943
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iget-object v3, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 393,408 ****
--- 427,445 ----
  
      goto :goto_1
  
+     .line 2944
      :cond_2
      iget v1, p1, Landroid/os/Message;->arg1:I
  
      if-ne v1, v4, :cond_1
  
+     .line 2945
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #calls: Landroid/net/wifi/WifiStateMachine;->handleFailedIpConfiguration()V
      invoke-static {v1}, Landroid/net/wifi/WifiStateMachine;->access$11000(Landroid/net/wifi/WifiStateMachine;)V
  
+     .line 2946
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iget-object v3, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 417,422 ****
--- 454,460 ----
  
      goto :goto_1
  
+     .line 2950
      :sswitch_3
      iget-object v3, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 427,432 ****
--- 465,471 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->handleSuccessfulIpConfiguration(Landroid/net/DhcpInfoInternal;)V
      invoke-static {v3, v1}, Landroid/net/wifi/WifiStateMachine;->access$10700(Landroid/net/wifi/WifiStateMachine;Landroid/net/DhcpInfoInternal;)V
  
+     .line 2951
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iget-object v3, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 441,452 ****
--- 480,493 ----
  
      goto :goto_1
  
+     .line 2954
      :sswitch_4
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #calls: Landroid/net/wifi/WifiStateMachine;->handleFailedIpConfiguration()V
      invoke-static {v1}, Landroid/net/wifi/WifiStateMachine;->access$11000(Landroid/net/wifi/WifiStateMachine;)V
  
+     .line 2955
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iget-object v3, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 461,469 ****
--- 502,512 ----
  
      goto :goto_1
  
+     .line 2958
      :sswitch_5
      invoke-static {}, Landroid/net/wifi/WifiNative;->disconnectCommand()Z
  
+     .line 2959
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iget-object v3, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 478,486 ****
--- 521,531 ----
  
      goto :goto_1
  
+     .line 2963
      :sswitch_6
      iget v0, p1, Landroid/os/Message;->arg1:I
  
+     .line 2964
      .local v0, netId:I
      iget-object v3, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 497,502 ****
--- 542,548 ----
  
      goto :goto_1
  
+     .line 2969
      .end local v0           #netId:I
      :sswitch_7
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 506,522 ****
--- 552,571 ----
  
      goto/16 :goto_1
  
+     .line 2975
      :sswitch_8
      iget v1, p1, Landroid/os/Message;->arg1:I
  
      if-ne v1, v4, :cond_1
  
+     .line 2976
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      const v3, 0x2004a
  
      invoke-virtual {v1, v3}, Landroid/net/wifi/WifiStateMachine;->sendMessage(I)V
  
+     .line 2977
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #calls: Landroid/net/wifi/WifiStateMachine;->deferMessage(Landroid/os/Message;)V
***************
*** 524,529 ****
--- 573,579 ----
  
      goto/16 :goto_1
  
+     .line 2982
      :sswitch_9
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 532,537 ****
--- 582,588 ----
  
      goto/16 :goto_1
  
+     .line 2986
      :sswitch_a
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$ConnectingState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 540,545 ****
--- 591,597 ----
  
      goto/16 :goto_1
  
+     .line 2934
      nop
  
      :sswitch_data_0
