*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiConfigStore$1.smali	2012-03-23 19:36:56.790716561 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiConfigStore$1.smali.original	2012-03-23 19:34:45.431321226 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 3
  
      .prologue
+     .line 688
      invoke-static {}, Landroid/net/wifi/WifiConfiguration$ProxySettings;->values()[Landroid/net/wifi/WifiConfiguration$ProxySettings;
  
      move-result-object v0
***************
*** 82,87 ****
--- 83,89 ----
      :try_end_2
      .catch Ljava/lang/NoSuchFieldError; {:try_start_2 .. :try_end_2} :catch_3
  
+     .line 643
      :goto_2
      invoke-static {}, Landroid/net/wifi/WifiConfiguration$IpAssignment;->values()[Landroid/net/wifi/WifiConfiguration$IpAssignment;
  
***************
*** 158,163 ****
--- 160,166 ----
  
      goto :goto_3
  
+     .line 688
      :catch_3
      move-exception v0
  
