*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateTracker$WifiStateReceiver.smali	2012-03-23 19:36:58.384745628 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateTracker$WifiStateReceiver.smali.original	2012-03-23 19:34:45.450321573 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 207
      iput-object p1, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 207
      invoke-direct {p0, p1}, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;-><init>(Landroid/net/wifi/WifiStateTracker;)V
  
      return-void
***************
*** 50,55 ****
--- 52,58 ----
      .parameter "intent"
  
      .prologue
+     .line 211
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
      move-result-object v2
***************
*** 62,70 ****
  
      if-eqz v2, :cond_2
  
      iget-object v3, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
!     const-string v2, "networkInfo"
  
      invoke-virtual {p2, v2}, Landroid/content/Intent;->getParcelableExtra(Ljava/lang/String;)Landroid/os/Parcelable;
  
--- 65,74 ----
  
      if-eqz v2, :cond_2
  
+     .line 212
      iget-object v3, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
!     const-string/jumbo v2, "networkInfo"
  
      invoke-virtual {p2, v2}, Landroid/content/Intent;->getParcelableExtra(Ljava/lang/String;)Landroid/os/Parcelable;
  
***************
*** 75,80 ****
--- 79,85 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mNetworkInfo:Landroid/net/NetworkInfo;
      invoke-static {v3, v2}, Landroid/net/wifi/WifiStateTracker;->access$102(Landroid/net/wifi/WifiStateTracker;Landroid/net/NetworkInfo;)Landroid/net/NetworkInfo;
  
+     .line 214
      iget-object v3, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      const-string v2, "linkProperties"
***************
*** 88,93 ****
--- 93,99 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLinkProperties:Landroid/net/LinkProperties;
      invoke-static {v3, v2}, Landroid/net/wifi/WifiStateTracker;->access$202(Landroid/net/wifi/WifiStateTracker;Landroid/net/LinkProperties;)Landroid/net/LinkProperties;
  
+     .line 216
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      #getter for: Landroid/net/wifi/WifiStateTracker;->mLinkProperties:Landroid/net/LinkProperties;
***************
*** 97,102 ****
--- 103,109 ----
  
      if-nez v2, :cond_0
  
+     .line 217
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      new-instance v3, Landroid/net/LinkProperties;
***************
*** 106,111 ****
--- 113,119 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLinkProperties:Landroid/net/LinkProperties;
      invoke-static {v2, v3}, Landroid/net/wifi/WifiStateTracker;->access$202(Landroid/net/wifi/WifiStateTracker;Landroid/net/LinkProperties;)Landroid/net/LinkProperties;
  
+     .line 219
      :cond_0
      iget-object v3, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
***************
*** 120,125 ****
--- 128,134 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLinkCapabilities:Landroid/net/LinkCapabilities;
      invoke-static {v3, v2}, Landroid/net/wifi/WifiStateTracker;->access$302(Landroid/net/wifi/WifiStateTracker;Landroid/net/LinkCapabilities;)Landroid/net/LinkCapabilities;
  
+     .line 221
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      #getter for: Landroid/net/wifi/WifiStateTracker;->mLinkCapabilities:Landroid/net/LinkCapabilities;
***************
*** 129,134 ****
--- 138,144 ----
  
      if-nez v2, :cond_1
  
+     .line 222
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      new-instance v3, Landroid/net/LinkCapabilities;
***************
*** 138,147 ****
--- 148,159 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLinkCapabilities:Landroid/net/LinkCapabilities;
      invoke-static {v2, v3}, Landroid/net/wifi/WifiStateTracker;->access$302(Landroid/net/wifi/WifiStateTracker;Landroid/net/LinkCapabilities;)Landroid/net/LinkCapabilities;
  
+     .line 254
      :cond_1
      :goto_0
      return-void
  
+     .line 224
      :cond_2
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
***************
*** 155,163 ****
  
      if-eqz v2, :cond_5
  
      iget-object v3, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
!     const-string v2, "networkInfo"
  
      invoke-virtual {p2, v2}, Landroid/content/Intent;->getParcelableExtra(Ljava/lang/String;)Landroid/os/Parcelable;
  
--- 167,176 ----
  
      if-eqz v2, :cond_5
  
+     .line 225
      iget-object v3, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
!     const-string/jumbo v2, "networkInfo"
  
      invoke-virtual {p2, v2}, Landroid/content/Intent;->getParcelableExtra(Ljava/lang/String;)Landroid/os/Parcelable;
  
***************
*** 168,173 ****
--- 181,187 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mNetworkInfo:Landroid/net/NetworkInfo;
      invoke-static {v3, v2}, Landroid/net/wifi/WifiStateTracker;->access$102(Landroid/net/wifi/WifiStateTracker;Landroid/net/NetworkInfo;)Landroid/net/NetworkInfo;
  
+     .line 227
      iget-object v3, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      const-string v2, "linkProperties"
***************
*** 181,186 ****
--- 195,201 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLinkProperties:Landroid/net/LinkProperties;
      invoke-static {v3, v2}, Landroid/net/wifi/WifiStateTracker;->access$202(Landroid/net/wifi/WifiStateTracker;Landroid/net/LinkProperties;)Landroid/net/LinkProperties;
  
+     .line 229
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      #getter for: Landroid/net/wifi/WifiStateTracker;->mLinkProperties:Landroid/net/LinkProperties;
***************
*** 190,195 ****
--- 205,211 ----
  
      if-nez v2, :cond_3
  
+     .line 230
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      new-instance v3, Landroid/net/LinkProperties;
***************
*** 199,204 ****
--- 215,221 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLinkProperties:Landroid/net/LinkProperties;
      invoke-static {v2, v3}, Landroid/net/wifi/WifiStateTracker;->access$202(Landroid/net/wifi/WifiStateTracker;Landroid/net/LinkProperties;)Landroid/net/LinkProperties;
  
+     .line 232
      :cond_3
      iget-object v3, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
***************
*** 213,218 ****
--- 230,236 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLinkCapabilities:Landroid/net/LinkCapabilities;
      invoke-static {v3, v2}, Landroid/net/wifi/WifiStateTracker;->access$302(Landroid/net/wifi/WifiStateTracker;Landroid/net/LinkCapabilities;)Landroid/net/LinkCapabilities;
  
+     .line 234
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      #getter for: Landroid/net/wifi/WifiStateTracker;->mLinkCapabilities:Landroid/net/LinkCapabilities;
***************
*** 222,227 ****
--- 240,246 ----
  
      if-nez v2, :cond_4
  
+     .line 235
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      new-instance v3, Landroid/net/LinkCapabilities;
***************
*** 231,236 ****
--- 250,256 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLinkCapabilities:Landroid/net/LinkCapabilities;
      invoke-static {v2, v3}, Landroid/net/wifi/WifiStateTracker;->access$302(Landroid/net/wifi/WifiStateTracker;Landroid/net/LinkCapabilities;)Landroid/net/LinkCapabilities;
  
+     .line 239
      :cond_4
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
***************
*** 243,248 ****
--- 263,269 ----
  
      move-result-object v1
  
+     .line 240
      .local v1, state:Landroid/net/NetworkInfo$State;
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
***************
*** 253,263 ****
--- 274,286 ----
  
      if-eq v2, v1, :cond_1
  
+     .line 243
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLastState:Landroid/net/NetworkInfo$State;
      invoke-static {v2, v1}, Landroid/net/wifi/WifiStateTracker;->access$402(Landroid/net/wifi/WifiStateTracker;Landroid/net/NetworkInfo$State;)Landroid/net/NetworkInfo$State;
  
+     .line 245
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      #getter for: Landroid/net/wifi/WifiStateTracker;->mCsHandler:Landroid/os/Handler;
***************
*** 282,292 ****
--- 305,317 ----
  
      move-result-object v0
  
+     .line 247
      .local v0, msg:Landroid/os/Message;
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
      goto/16 :goto_0
  
+     .line 248
      .end local v0           #msg:Landroid/os/Message;
      .end local v1           #state:Landroid/net/NetworkInfo$State;
      :cond_5
***************
*** 302,307 ****
--- 327,333 ----
  
      if-eqz v2, :cond_1
  
+     .line 249
      iget-object v3, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      const-string v2, "linkProperties"
***************
*** 315,320 ****
--- 341,347 ----
      #setter for: Landroid/net/wifi/WifiStateTracker;->mLinkProperties:Landroid/net/LinkProperties;
      invoke-static {v3, v2}, Landroid/net/wifi/WifiStateTracker;->access$202(Landroid/net/wifi/WifiStateTracker;Landroid/net/LinkProperties;)Landroid/net/LinkProperties;
  
+     .line 251
      iget-object v2, p0, Landroid/net/wifi/WifiStateTracker$WifiStateReceiver;->this$0:Landroid/net/wifi/WifiStateTracker;
  
      #getter for: Landroid/net/wifi/WifiStateTracker;->mCsHandler:Landroid/os/Handler;
***************
*** 335,340 ****
--- 362,368 ----
  
      move-result-object v0
  
+     .line 252
      .restart local v0       #msg:Landroid/os/Message;
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
