*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DriverUnloadedState.smali	2012-03-23 19:36:58.120740815 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DriverUnloadedState.smali.original	2012-03-23 19:34:45.447321517 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 2150
      iput-object p1, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadedState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .locals 2
  
      .prologue
+     .line 2154
      const v0, 0xc365
  
      invoke-virtual {p0}, Landroid/net/wifi/WifiStateMachine$DriverUnloadedState;->getName()Ljava/lang/String;
***************
*** 45,50 ****
--- 47,53 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(ILjava/lang/String;)I
  
+     .line 2155
      return-void
  .end method
  
***************
*** 53,67 ****
--- 56,74 ----
      .parameter "message"
  
      .prologue
+     .line 2159
      iget v0, p1, Landroid/os/Message;->what:I
  
      sparse-switch v0, :sswitch_data_0
  
+     .line 2168
      const/4 v0, 0x0
  
+     .line 2171
      :goto_0
      return v0
  
+     .line 2161
      :sswitch_0
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadedState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 74,79 ****
--- 81,87 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/util/AsyncChannel;->sendMessage(I)V
  
+     .line 2162
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadedState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadedState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 86,91 ****
--- 94,100 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
      invoke-static {v0, v1}, Landroid/net/wifi/WifiStateMachine;->access$3500(Landroid/net/wifi/WifiStateMachine;Lcom/android/internal/util/IState;)V
  
+     .line 2170
      :goto_1
      const v0, 0xc366
  
***************
*** 93,102 ****
--- 102,113 ----
  
      invoke-static {v0, v1}, Landroid/util/EventLog;->writeEvent(II)I
  
+     .line 2171
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 2165
      :sswitch_1
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverUnloadedState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 111,116 ****
--- 122,128 ----
  
      goto :goto_1
  
+     .line 2159
      nop
  
      :sswitch_data_0
