*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$WalledGardenState.smali	2012-03-23 19:36:56.813716981 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$WalledGardenState.smali.original	2012-03-23 19:34:45.432321245 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1049
      iput-object p1, p0, Landroid/net/wifi/WifiWatchdogStateMachine$WalledGardenState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .locals 4
  
      .prologue
+     .line 1052
      iget-object v0, p0, Landroid/net/wifi/WifiWatchdogStateMachine$WalledGardenState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      const v1, 0x21064
***************
*** 56,61 ****
--- 58,64 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 1053
      return-void
  .end method
  
***************
*** 66,83 ****
--- 69,90 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 1057
      iget v1, p1, Landroid/os/Message;->what:I
  
      const v2, 0x21064
  
      if-eq v1, v2, :cond_1
  
+     .line 1058
      const/4 v0, 0x0
  
+     .line 1073
      :cond_0
      :goto_0
      return v0
  
+     .line 1061
      :cond_1
      iget v1, p1, Landroid/os/Message;->arg1:I
  
***************
*** 90,100 ****
--- 97,109 ----
  
      if-ne v1, v2, :cond_0
  
+     .line 1067
      iget-object v1, p0, Landroid/net/wifi/WifiWatchdogStateMachine$WalledGardenState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      #calls: Landroid/net/wifi/WifiWatchdogStateMachine;->setWalledGardenNotificationVisible(Z)V
      invoke-static {v1, v0}, Landroid/net/wifi/WifiWatchdogStateMachine;->access$1200(Landroid/net/wifi/WifiWatchdogStateMachine;Z)V
  
+     .line 1068
      iget-object v1, p0, Landroid/net/wifi/WifiWatchdogStateMachine$WalledGardenState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      #getter for: Landroid/net/wifi/WifiWatchdogStateMachine;->mPoorNetworkDetectionEnabled:Z
***************
*** 104,109 ****
--- 113,119 ----
  
      if-eqz v1, :cond_2
  
+     .line 1069
      iget-object v1, p0, Landroid/net/wifi/WifiWatchdogStateMachine$WalledGardenState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$WalledGardenState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
***************
*** 118,123 ****
--- 128,134 ----
  
      goto :goto_0
  
+     .line 1071
      :cond_2
      iget-object v1, p0, Landroid/net/wifi/WifiWatchdogStateMachine$WalledGardenState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
