*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DriverLoadingState$1.smali	2012-03-23 19:36:59.262761640 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DriverLoadingState$1.smali.original	2012-03-23 19:34:45.464321827 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 1948
      iput-object p1, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
      iput-object p2, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->val$message:Landroid/os/Message;
***************
*** 45,50 ****
--- 46,52 ----
      .locals 2
  
      .prologue
+     .line 1950
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
      iget-object v0, v0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 56,67 ****
--- 58,71 ----
  
      invoke-virtual {v0}, Landroid/os/PowerManager$WakeLock;->acquire()V
  
+     .line 1952
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->val$message:Landroid/os/Message;
  
      iget v0, v0, Landroid/os/Message;->arg1:I
  
      sparse-switch v0, :sswitch_data_0
  
+     .line 1961
      :goto_0
      invoke-static {}, Landroid/net/wifi/WifiNative;->loadDriver()Z
  
***************
*** 69,74 ****
--- 73,79 ----
  
      if-eqz v0, :cond_0
  
+     .line 1963
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
      iget-object v0, v0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState;->this$0:Landroid/net/wifi/WifiStateMachine;
***************
*** 77,82 ****
--- 82,88 ----
  
      invoke-virtual {v0, v1}, Landroid/net/wifi/WifiStateMachine;->sendMessage(I)V
  
+     .line 1976
      :goto_1
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
***************
*** 89,96 ****
--- 95,104 ----
  
      invoke-virtual {v0}, Landroid/os/PowerManager$WakeLock;->release()V
  
+     .line 1977
      return-void
  
+     .line 1954
      :sswitch_0
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
***************
*** 103,108 ****
--- 111,117 ----
  
      goto :goto_0
  
+     .line 1957
      :sswitch_1
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
***************
*** 115,120 ****
--- 124,130 ----
  
      goto :goto_0
  
+     .line 1965
      :cond_0
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
***************
*** 125,136 ****
--- 135,148 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->loge(Ljava/lang/String;)V
      invoke-static {v0, v1}, Landroid/net/wifi/WifiStateMachine;->access$200(Landroid/net/wifi/WifiStateMachine;Ljava/lang/String;)V
  
+     .line 1966
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->val$message:Landroid/os/Message;
  
      iget v0, v0, Landroid/os/Message;->arg1:I
  
      sparse-switch v0, :sswitch_data_1
  
+     .line 1974
      :goto_2
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
***************
*** 142,147 ****
--- 154,160 ----
  
      goto :goto_1
  
+     .line 1968
      :sswitch_2
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
***************
*** 154,159 ****
--- 167,173 ----
  
      goto :goto_2
  
+     .line 1971
      :sswitch_3
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DriverLoadingState$1;->this$1:Landroid/net/wifi/WifiStateMachine$DriverLoadingState;
  
***************
*** 166,171 ****
--- 180,186 ----
  
      goto :goto_2
  
+     .line 1952
      nop
  
      :sswitch_data_0
***************
*** 174,179 ****
--- 189,195 ----
          0xc -> :sswitch_1
      .end sparse-switch
  
+     .line 1966
      :sswitch_data_1
      .sparse-switch
          0x2 -> :sswitch_2
