*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$3.smali	2012-03-23 19:36:57.777734560 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$3.smali.original	2012-03-23 19:34:45.443321445 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 1768
      iput-object p1, p0, Landroid/net/wifi/WifiStateMachine$3;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      iput-object p2, p0, Landroid/net/wifi/WifiStateMachine$3;->val$config:Landroid/net/wifi/WifiConfiguration;
***************
*** 45,50 ****
--- 46,52 ----
      .locals 6
  
      .prologue
+     .line 1771
      :try_start_0
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$3;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 62,73 ****
  
      move-result-object v4
  
!     const-string v5, "wl0.1"
  
      invoke-interface {v2, v3, v4, v5}, Landroid/os/INetworkManagementService;->startAccessPoint(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Ljava/lang/String;)V
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
      :goto_0
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$3;->this$0:Landroid/net/wifi/WifiStateMachine;
  
--- 64,76 ----
  
      move-result-object v4
  
!     const-string/jumbo v5, "wl0.1"
  
      invoke-interface {v2, v3, v4, v5}, Landroid/os/INetworkManagementService;->startAccessPoint(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Ljava/lang/String;)V
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1784
      :goto_0
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$3;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 75,86 ****
--- 78,92 ----
  
      invoke-virtual {v2, v3}, Landroid/net/wifi/WifiStateMachine;->sendMessage(I)V
  
+     .line 1785
      :goto_1
      return-void
  
+     .line 1772
      :catch_0
      move-exception v0
  
+     .line 1773
      .local v0, e:Ljava/lang/Exception;
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$3;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 105,110 ****
--- 111,117 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->loge(Ljava/lang/String;)V
      invoke-static {v2, v3}, Landroid/net/wifi/WifiStateMachine;->access$200(Landroid/net/wifi/WifiStateMachine;Ljava/lang/String;)V
  
+     .line 1775
      :try_start_1
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$3;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 122,127 ****
--- 129,135 ----
  
      invoke-interface {v2, v3}, Landroid/os/INetworkManagementService;->stopAccessPoint(Ljava/lang/String;)V
  
+     .line 1776
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$3;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #getter for: Landroid/net/wifi/WifiStateMachine;->mNwService:Landroid/os/INetworkManagementService;
***************
*** 138,144 ****
  
      move-result-object v4
  
!     const-string v5, "wl0.1"
  
      invoke-interface {v2, v3, v4, v5}, Landroid/os/INetworkManagementService;->startAccessPoint(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Ljava/lang/String;)V
      :try_end_1
--- 146,152 ----
  
      move-result-object v4
  
!     const-string/jumbo v5, "wl0.1"
  
      invoke-interface {v2, v3, v4, v5}, Landroid/os/INetworkManagementService;->startAccessPoint(Landroid/net/wifi/WifiConfiguration;Ljava/lang/String;Ljava/lang/String;)V
      :try_end_1
***************
*** 146,154 ****
--- 154,164 ----
  
      goto :goto_0
  
+     .line 1777
      :catch_1
      move-exception v1
  
+     .line 1778
      .local v1, e1:Ljava/lang/Exception;
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$3;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 173,178 ****
--- 183,189 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->loge(Ljava/lang/String;)V
      invoke-static {v2, v3}, Landroid/net/wifi/WifiStateMachine;->access$200(Landroid/net/wifi/WifiStateMachine;Ljava/lang/String;)V
  
+     .line 1779
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$3;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      const v3, 0x20017
