*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$DefaultState.smali	2012-03-23 19:36:57.311726061 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$DefaultState.smali.original	2012-03-23 19:34:45.438321353 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 562
      iput-object p1, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 40,52 ****
--- 41,56 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 565
      iget v0, p1, Landroid/os/Message;->what:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 577
      :goto_0
      return v1
  
+     .line 567
      :pswitch_0
      iget-object v0, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
***************
*** 55,60 ****
--- 59,65 ----
  
      goto :goto_0
  
+     .line 565
      nop
  
      :pswitch_data_0
