*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState.smali	2012-03-23 19:36:55.228688079 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState.smali.original	2012-03-23 19:34:45.410320843 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 992
      iput-object p1, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,46 ****
--- 38,49 ----
      .locals 4
  
      .prologue
+     .line 996
      iget-object v0, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      invoke-static {v0}, Landroid/net/wifi/WifiWatchdogStateMachine;->access$4908(Landroid/net/wifi/WifiWatchdogStateMachine;)I
  
+     .line 997
      iget-object v0, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      const v1, 0x21065
***************
*** 60,65 ****
--- 63,69 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 998
      return-void
  .end method
  
***************
*** 70,87 ****
--- 74,95 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 1002
      iget v2, p1, Landroid/os/Message;->what:I
  
      const v3, 0x21065
  
      if-eq v2, v3, :cond_1
  
+     .line 1003
      const/4 v1, 0x0
  
+     .line 1045
      :cond_0
      :goto_0
      return v1
  
+     .line 1006
      :cond_1
      iget v2, p1, Landroid/os/Message;->arg1:I
  
***************
*** 94,99 ****
--- 102,108 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 1013
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      iget-boolean v2, v2, Landroid/net/wifi/WifiWatchdogStateMachine;->mDisableAPNextFailure:Z
***************
*** 136,141 ****
--- 145,151 ----
  
      if-lt v2, v3, :cond_6
  
+     .line 1015
      :cond_2
      invoke-static {}, Landroid/net/wifi/WifiWatchdogStateMachine;->access$5200()Z
  
***************
*** 143,148 ****
--- 153,159 ----
  
      if-eqz v2, :cond_3
  
+     .line 1016
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      const-string v3, "Would disable bad network, but device has no mobile data!  Going idle..."
***************
*** 150,155 ****
--- 161,167 ----
      #calls: Landroid/net/wifi/WifiWatchdogStateMachine;->log(Ljava/lang/String;)V
      invoke-static {v2, v3}, Landroid/net/wifi/WifiWatchdogStateMachine;->access$5300(Landroid/net/wifi/WifiWatchdogStateMachine;Ljava/lang/String;)V
  
+     .line 1019
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      iget-object v3, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
***************
*** 164,169 ****
--- 176,182 ----
  
      goto :goto_0
  
+     .line 1024
      :cond_3
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
***************
*** 199,205 ****
  
      move-result-object v3
  
!     const-string v4, "numCheckFailures "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 212,218 ----
  
      move-result-object v3
  
!     const-string/jumbo v4, "numCheckFailures "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 244,249 ****
--- 257,263 ----
      #calls: Landroid/net/wifi/WifiWatchdogStateMachine;->log(Ljava/lang/String;)V
      invoke-static {v2, v3}, Landroid/net/wifi/WifiWatchdogStateMachine;->access$5300(Landroid/net/wifi/WifiWatchdogStateMachine;Ljava/lang/String;)V
  
+     .line 1027
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      #getter for: Landroid/net/wifi/WifiWatchdogStateMachine;->mConnectionInfo:Landroid/net/wifi/WifiInfo;
***************
*** 255,260 ****
--- 269,275 ----
  
      move-result v0
  
+     .line 1028
      .local v0, networkId:I
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
***************
*** 262,267 ****
--- 277,283 ----
  
      if-nez v2, :cond_4
  
+     .line 1029
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      #getter for: Landroid/net/wifi/WifiWatchdogStateMachine;->mWifiManager:Landroid/net/wifi/WifiManager;
***************
*** 284,293 ****
--- 300,311 ----
  
      invoke-virtual {v2, v3, v4}, Landroid/net/wifi/WifiManager;->asyncConnect(Landroid/content/Context;Landroid/os/Handler;)V
  
+     .line 1030
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      iput-boolean v1, v2, Landroid/net/wifi/WifiWatchdogStateMachine;->mHasConnectedWifiManager:Z
  
+     .line 1032
      :cond_4
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
***************
*** 298,303 ****
--- 316,322 ----
  
      invoke-virtual {v2, v0, v1}, Landroid/net/wifi/WifiManager;->disableNetwork(II)V
  
+     .line 1033
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      #getter for: Landroid/net/wifi/WifiWatchdogStateMachine;->mShowDisabledNotification:Z
***************
*** 320,330 ****
--- 339,351 ----
  
      if-eqz v2, :cond_5
  
+     .line 1034
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      #calls: Landroid/net/wifi/WifiWatchdogStateMachine;->setDisabledNetworkNotificationVisible(Z)V
      invoke-static {v2, v1}, Landroid/net/wifi/WifiWatchdogStateMachine;->access$1100(Landroid/net/wifi/WifiWatchdogStateMachine;Z)V
  
+     .line 1036
      :cond_5
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
***************
*** 340,345 ****
--- 361,367 ----
  
      goto/16 :goto_0
  
+     .line 1038
      .end local v0           #networkId:I
      :cond_6
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
***************
*** 370,376 ****
  
      move-result-object v3
  
!     const-string v4, "numCheckFailures "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 392,398 ----
  
      move-result-object v3
  
!     const-string/jumbo v4, "numCheckFailures "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 415,420 ****
--- 437,443 ----
      #calls: Landroid/net/wifi/WifiWatchdogStateMachine;->log(Ljava/lang/String;)V
      invoke-static {v2, v3}, Landroid/net/wifi/WifiWatchdogStateMachine;->access$5300(Landroid/net/wifi/WifiWatchdogStateMachine;Ljava/lang/String;)V
  
+     .line 1041
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      #getter for: Landroid/net/wifi/WifiWatchdogStateMachine;->mWifiManager:Landroid/net/wifi/WifiManager;
***************
*** 435,440 ****
--- 458,464 ----
  
      invoke-virtual {v2, v3}, Landroid/net/wifi/WifiManager;->addToBlacklist(Ljava/lang/String;)Z
  
+     .line 1042
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      #getter for: Landroid/net/wifi/WifiWatchdogStateMachine;->mWifiManager:Landroid/net/wifi/WifiManager;
***************
*** 444,449 ****
--- 468,474 ----
  
      invoke-virtual {v2}, Landroid/net/wifi/WifiManager;->reassociate()Z
  
+     .line 1043
      iget-object v2, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      iget-object v3, p0, Landroid/net/wifi/WifiWatchdogStateMachine$DnsCheckFailureState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
