*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiConfiguration$1.smali	2012-03-23 19:36:56.913718804 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiConfiguration$1.smali.original	2012-03-23 19:34:45.433321263 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 590
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,53 ****
--- 45,56 ----
      .parameter "in"
  
      .prologue
+     .line 592
      new-instance v1, Landroid/net/wifi/WifiConfiguration;
  
      invoke-direct {v1}, Landroid/net/wifi/WifiConfiguration;-><init>()V
  
+     .line 593
      .local v1, config:Landroid/net/wifi/WifiConfiguration;
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
***************
*** 55,90 ****
--- 58,99 ----
  
      iput v6, v1, Landroid/net/wifi/WifiConfiguration;->networkId:I
  
+     .line 594
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v6
  
      iput v6, v1, Landroid/net/wifi/WifiConfiguration;->status:I
  
+     .line 595
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v6
  
      iput v6, v1, Landroid/net/wifi/WifiConfiguration;->disableReason:I
  
+     .line 596
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v6
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->SSID:Ljava/lang/String;
  
+     .line 597
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v6
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->BSSID:Ljava/lang/String;
  
+     .line 598
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v6
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
  
+     .line 599
      const/4 v3, 0x0
  
      .local v3, i:I
***************
*** 95,100 ****
--- 104,110 ----
  
      if-ge v3, v6, :cond_0
  
+     .line 600
      iget-object v6, v1, Landroid/net/wifi/WifiConfiguration;->wepKeys:[Ljava/lang/String;
  
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
***************
*** 103,112 ****
--- 113,124 ----
  
      aput-object v7, v6, v3
  
+     .line 599
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 601
      :cond_0
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
***************
*** 114,125 ****
--- 126,139 ----
  
      iput v6, v1, Landroid/net/wifi/WifiConfiguration;->wepTxKeyIndex:I
  
+     .line 602
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v6
  
      iput v6, v1, Landroid/net/wifi/WifiConfiguration;->priority:I
  
+     .line 603
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v6
***************
*** 131,136 ****
--- 145,151 ----
      :goto_1
      iput-boolean v6, v1, Landroid/net/wifi/WifiConfiguration;->hiddenSSID:Z
  
+     .line 604
      #calls: Landroid/net/wifi/WifiConfiguration;->readBitSet(Landroid/os/Parcel;)Ljava/util/BitSet;
      invoke-static {p1}, Landroid/net/wifi/WifiConfiguration;->access$100(Landroid/os/Parcel;)Ljava/util/BitSet;
  
***************
*** 138,143 ****
--- 153,159 ----
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
  
+     .line 605
      #calls: Landroid/net/wifi/WifiConfiguration;->readBitSet(Landroid/os/Parcel;)Ljava/util/BitSet;
      invoke-static {p1}, Landroid/net/wifi/WifiConfiguration;->access$100(Landroid/os/Parcel;)Ljava/util/BitSet;
  
***************
*** 145,150 ****
--- 161,167 ----
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->allowedProtocols:Ljava/util/BitSet;
  
+     .line 606
      #calls: Landroid/net/wifi/WifiConfiguration;->readBitSet(Landroid/os/Parcel;)Ljava/util/BitSet;
      invoke-static {p1}, Landroid/net/wifi/WifiConfiguration;->access$100(Landroid/os/Parcel;)Ljava/util/BitSet;
  
***************
*** 152,157 ****
--- 169,175 ----
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->allowedAuthAlgorithms:Ljava/util/BitSet;
  
+     .line 607
      #calls: Landroid/net/wifi/WifiConfiguration;->readBitSet(Landroid/os/Parcel;)Ljava/util/BitSet;
      invoke-static {p1}, Landroid/net/wifi/WifiConfiguration;->access$100(Landroid/os/Parcel;)Ljava/util/BitSet;
  
***************
*** 159,164 ****
--- 177,183 ----
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->allowedPairwiseCiphers:Ljava/util/BitSet;
  
+     .line 608
      #calls: Landroid/net/wifi/WifiConfiguration;->readBitSet(Landroid/os/Parcel;)Ljava/util/BitSet;
      invoke-static {p1}, Landroid/net/wifi/WifiConfiguration;->access$100(Landroid/os/Parcel;)Ljava/util/BitSet;
  
***************
*** 166,171 ****
--- 185,191 ----
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->allowedGroupCiphers:Ljava/util/BitSet;
  
+     .line 610
      iget-object v0, v1, Landroid/net/wifi/WifiConfiguration;->enterpriseFields:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
  
      .local v0, arr$:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
***************
*** 180,185 ****
--- 200,206 ----
  
      aget-object v2, v0, v4
  
+     .line 611
      .local v2, field:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
***************
*** 187,196 ****
--- 208,219 ----
  
      invoke-virtual {v2, v6}, Landroid/net/wifi/WifiConfiguration$EnterpriseField;->setValue(Ljava/lang/String;)V
  
+     .line 610
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_2
  
+     .line 603
      .end local v0           #arr$:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      .end local v2           #field:Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      .end local v4           #i$:I
***************
*** 200,205 ****
--- 223,229 ----
  
      goto :goto_1
  
+     .line 614
      .restart local v0       #arr$:[Landroid/net/wifi/WifiConfiguration$EnterpriseField;
      .restart local v4       #i$:I
      .restart local v5       #len$:I
***************
*** 214,219 ****
--- 238,244 ----
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->ipAssignment:Landroid/net/wifi/WifiConfiguration$IpAssignment;
  
+     .line 615
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v6
***************
*** 224,229 ****
--- 249,255 ----
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->proxySettings:Landroid/net/wifi/WifiConfiguration$ProxySettings;
  
+     .line 616
      const/4 v6, 0x0
  
      invoke-virtual {p1, v6}, Landroid/os/Parcel;->readParcelable(Ljava/lang/ClassLoader;)Landroid/os/Parcelable;
***************
*** 234,239 ****
--- 260,266 ----
  
      iput-object v6, v1, Landroid/net/wifi/WifiConfiguration;->linkProperties:Landroid/net/LinkProperties;
  
+     .line 617
      return-object v1
  .end method
  
***************
*** 242,247 ****
--- 269,275 ----
      .parameter "x0"
  
      .prologue
+     .line 590
      invoke-virtual {p0, p1}, Landroid/net/wifi/WifiConfiguration$1;->createFromParcel(Landroid/os/Parcel;)Landroid/net/wifi/WifiConfiguration;
  
      move-result-object v0
***************
*** 254,259 ****
--- 282,288 ----
      .parameter "size"
  
      .prologue
+     .line 621
      new-array v0, p1, [Landroid/net/wifi/WifiConfiguration;
  
      return-object v0
***************
*** 264,269 ****
--- 293,299 ----
      .parameter "x0"
  
      .prologue
+     .line 590
      invoke-virtual {p0, p1}, Landroid/net/wifi/WifiConfiguration$1;->newArray(I)[Landroid/net/wifi/WifiConfiguration;
  
      move-result-object v0
