*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DefaultState.smali	2012-03-23 19:36:57.992738480 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiStateMachine$DefaultState.smali.original	2012-03-23 19:34:45.446321499 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1793
      iput-object p1, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 42,51 ****
--- 43,54 ----
  
      const/4 v1, 0x1
  
+     .line 1797
      iget v2, p1, Landroid/os/Message;->what:I
  
      sparse-switch v2, :sswitch_data_0
  
+     .line 1891
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 69,83 ****
--- 72,89 ----
      #calls: Landroid/net/wifi/WifiStateMachine;->loge(Ljava/lang/String;)V
      invoke-static {v0, v2}, Landroid/net/wifi/WifiStateMachine;->access$200(Landroid/net/wifi/WifiStateMachine;Ljava/lang/String;)V
  
+     .line 1894
      :goto_0
      :sswitch_0
      return v1
  
+     .line 1799
      :sswitch_1
      iget v0, p1, Landroid/os/Message;->arg1:I
  
      if-nez v0, :cond_0
  
+     .line 1800
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #getter for: Landroid/net/wifi/WifiStateMachine;->mWifiP2pChannel:Lcom/android/internal/util/AsyncChannel;
***************
*** 91,96 ****
--- 97,103 ----
  
      goto :goto_0
  
+     .line 1802
      :cond_0
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 119,124 ****
--- 126,132 ----
  
      goto :goto_0
  
+     .line 1806
      :sswitch_2
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 147,152 ****
--- 155,161 ----
  
      goto :goto_0
  
+     .line 1811
      :sswitch_3
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 162,167 ****
--- 171,177 ----
  
      goto :goto_0
  
+     .line 1821
      :sswitch_4
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 178,183 ****
--- 188,194 ----
  
      goto :goto_0
  
+     .line 1824
      :sswitch_5
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 193,198 ****
--- 204,210 ----
  
      goto :goto_0
  
+     .line 1827
      :sswitch_6
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 208,224 ****
--- 220,239 ----
  
      goto :goto_0
  
+     .line 1877
      :sswitch_7
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-virtual {v2, v0}, Landroid/net/wifi/WifiStateMachine;->setWifiEnabled(Z)V
  
+     .line 1878
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-virtual {v0, v1}, Landroid/net/wifi/WifiStateMachine;->setWifiEnabled(Z)V
  
      goto :goto_0
  
+     .line 1882
      :sswitch_8
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
***************
*** 239,249 ****
--- 254,266 ----
  
      goto/16 :goto_0
  
+     .line 1887
      :sswitch_9
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      invoke-virtual {v2, v0}, Landroid/net/wifi/WifiStateMachine;->setWifiEnabled(Z)V
  
+     .line 1888
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$DefaultState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #calls: Landroid/net/wifi/WifiStateMachine;->deferMessage(Landroid/os/Message;)V
***************
*** 251,256 ****
--- 268,274 ----
  
      goto/16 :goto_0
  
+     .line 1797
      :sswitch_data_0
      .sparse-switch
          0x11000 -> :sswitch_1
