*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$OnlineState.smali	2012-03-23 19:36:55.200687567 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/wifi/WifiWatchdogStateMachine$OnlineState.smali.original	2012-03-23 19:34:45.410320843 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 989
      iput-object p1, p0, Landroid/net/wifi/WifiWatchdogStateMachine$OnlineState;->this$0:Landroid/net/wifi/WifiWatchdogStateMachine;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
