*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/sms/BearerData$TimeStamp.smali	2012-03-23 19:37:38.681480439 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/sms/BearerData$TimeStamp.smali.original	2012-03-23 19:34:46.185334975 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 1
  
      .prologue
+     .line 235
      invoke-static {}, Ljava/util/TimeZone;->getDefault()Ljava/util/TimeZone;
  
      move-result-object v0
***************
*** 29,34 ****
--- 30,36 ----
  
      invoke-direct {p0, v0}, Landroid/text/format/Time;-><init>(Ljava/lang/String;)V
  
+     .line 236
      return-void
  .end method
  
***************
*** 43,52 ****
--- 45,56 ----
  
      const/4 v8, 0x0
  
+     .line 239
      new-instance v5, Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;
  
      invoke-direct {v5}, Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;-><init>()V
  
+     .line 241
      .local v5, ts:Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;
      const/4 v7, 0x0
  
***************
*** 56,61 ****
--- 60,66 ----
  
      move-result v6
  
+     .line 242
      .local v6, year:I
      const/16 v7, 0x63
  
***************
*** 66,75 ****
--- 71,82 ----
      :cond_0
      move-object v5, v8
  
+     .line 259
      .end local v5           #ts:Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;
      :goto_0
      return-object v5
  
+     .line 243
      .restart local v5       #ts:Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;
      :cond_1
      const/16 v7, 0x60
***************
*** 81,92 ****
--- 88,101 ----
      :goto_1
      iput v7, v5, Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;->year:I
  
+     .line 244
      aget-byte v7, p0, v9
  
      invoke-static {v7}, Lcom/android/internal/telephony/IccUtils;->cdmaBcdByteToInt(B)I
  
      move-result v3
  
+     .line 245
      .local v3, month:I
      if-lt v3, v9, :cond_2
  
***************
*** 99,116 ****
--- 108,128 ----
  
      goto :goto_0
  
+     .line 243
      .end local v3           #month:I
      :cond_3
      add-int/lit16 v7, v6, 0x7d0
  
      goto :goto_1
  
+     .line 246
      .restart local v3       #month:I
      :cond_4
      add-int/lit8 v7, v3, -0x1
  
      iput v7, v5, Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;->month:I
  
+     .line 247
      const/4 v7, 0x2
  
      aget-byte v7, p0, v7
***************
*** 119,124 ****
--- 131,137 ----
  
      move-result v0
  
+     .line 248
      .local v0, day:I
      if-lt v0, v9, :cond_5
  
***************
*** 131,139 ****
--- 144,154 ----
  
      goto :goto_0
  
+     .line 249
      :cond_6
      iput v0, v5, Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;->monthDay:I
  
+     .line 250
      const/4 v7, 0x3
  
      aget-byte v7, p0, v7
***************
*** 142,147 ****
--- 157,163 ----
  
      move-result v1
  
+     .line 251
      .local v1, hour:I
      if-ltz v1, :cond_7
  
***************
*** 154,162 ****
--- 170,180 ----
  
      goto :goto_0
  
+     .line 252
      :cond_8
      iput v1, v5, Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;->hour:I
  
+     .line 253
      const/4 v7, 0x4
  
      aget-byte v7, p0, v7
***************
*** 165,170 ****
--- 183,189 ----
  
      move-result v2
  
+     .line 254
      .local v2, minute:I
      if-ltz v2, :cond_9
  
***************
*** 175,183 ****
--- 194,204 ----
  
      goto :goto_0
  
+     .line 255
      :cond_a
      iput v2, v5, Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;->minute:I
  
+     .line 256
      const/4 v7, 0x5
  
      aget-byte v7, p0, v7
***************
*** 186,191 ****
--- 207,213 ----
  
      move-result v4
  
+     .line 257
      .local v4, second:I
      if-ltz v4, :cond_b
  
***************
*** 196,201 ****
--- 218,224 ----
  
      goto :goto_0
  
+     .line 258
      :cond_c
      iput v4, v5, Lcom/android/internal/telephony/cdma/sms/BearerData$TimeStamp;->second:I
  
***************
*** 208,227 ****
      .locals 3
  
      .prologue
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
      .local v0, builder:Ljava/lang/StringBuilder;
      const-string v1, "TimeStamp "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "{ year="
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 231,253 ----
      .locals 3
  
      .prologue
+     .line 264
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 265
      .local v0, builder:Ljava/lang/StringBuilder;
      const-string v1, "TimeStamp "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 266
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "{ year="
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 239,244 ****
--- 265,271 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 267
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 261,266 ****
--- 288,294 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 268
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 283,288 ****
--- 311,317 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 269
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 305,310 ****
--- 334,340 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 270
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 327,332 ****
--- 357,363 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 271
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 349,358 ****
--- 380,391 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 272
      const-string v1, " }"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 273
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
