*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/sms/CdmaSmsAddress.smali	2012-03-23 19:37:38.608479106 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/sms/CdmaSmsAddress.smali.original	2012-03-23 19:34:46.184334957 -0500
***************
*** 58,63 ****
--- 58,64 ----
      .locals 4
  
      .prologue
+     .line 133
      const/16 v1, 0xc
  
      new-array v1, v1, [C
***************
*** 66,71 ****
--- 67,73 ----
  
      sput-object v1, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numericCharsDialable:[C
  
+     .line 137
      const/16 v1, 0x8
  
      new-array v1, v1, [C
***************
*** 74,79 ****
--- 76,82 ----
  
      sput-object v1, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numericCharsSugar:[C
  
+     .line 141
      new-instance v1, Landroid/util/SparseBooleanArray;
  
      sget-object v2, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numericCharsDialable:[C
***************
*** 90,95 ****
--- 93,99 ----
  
      sput-object v1, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numericCharDialableMap:Landroid/util/SparseBooleanArray;
  
+     .line 144
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 100,105 ****
--- 104,110 ----
  
      if-ge v0, v1, :cond_0
  
+     .line 145
      sget-object v1, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numericCharDialableMap:Landroid/util/SparseBooleanArray;
  
      sget-object v2, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numericCharsDialable:[C
***************
*** 110,119 ****
--- 115,126 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 144
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 147
      :cond_0
      const/4 v0, 0x0
  
***************
*** 124,129 ****
--- 131,137 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 148
      sget-object v1, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numericCharDialableMap:Landroid/util/SparseBooleanArray;
  
      sget-object v2, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numericCharsSugar:[C
***************
*** 134,146 ****
--- 142,157 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 147
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 150
      :cond_1
      return-void
  
+     .line 133
      nop
  
      :array_0
***************
*** 159,164 ****
--- 170,176 ----
          0x23t 0x0t
      .end array-data
  
+     .line 137
      :array_1
      .array-data 0x2
          0x28t 0x0t
***************
*** 176,183 ****
--- 188,197 ----
      .locals 0
  
      .prologue
+     .line 93
      invoke-direct {p0}, Lcom/android/internal/telephony/SmsAddress;-><init>()V
  
+     .line 94
      return-void
  .end method
  
***************
*** 186,200 ****
--- 200,217 ----
      .parameter "address"
  
      .prologue
+     .line 159
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 160
      .local v0, builder:Ljava/lang/StringBuilder;
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v3
  
+     .line 161
      .local v3, len:I
      const/4 v2, 0x0
  
***************
*** 202,211 ****
--- 219,230 ----
      :goto_0
      if-ge v2, v3, :cond_2
  
+     .line 162
      invoke-virtual {p0, v2}, Ljava/lang/String;->charAt(I)C
  
      move-result v1
  
+     .line 163
      .local v1, c:C
      sget-object v5, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numericCharDialableMap:Landroid/util/SparseBooleanArray;
  
***************
*** 213,228 ****
--- 232,250 ----
  
      move-result v4
  
+     .line 164
      .local v4, mapIndex:I
      if-gez v4, :cond_0
  
      const/4 v5, 0x0
  
+     .line 168
      .end local v1           #c:C
      .end local v4           #mapIndex:I
      :goto_1
      return-object v5
  
+     .line 165
      .restart local v1       #c:C
      .restart local v4       #mapIndex:I
      :cond_0
***************
*** 234,249 ****
--- 256,274 ----
  
      if-nez v5, :cond_1
  
+     .line 161
      :goto_2
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 166
      :cond_1
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
      goto :goto_2
  
+     .line 168
      .end local v1           #c:C
      .end local v4           #mapIndex:I
      :cond_2
***************
*** 259,273 ****
--- 284,301 ----
      .parameter "address"
  
      .prologue
+     .line 176
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 177
      .local v0, builder:Ljava/lang/StringBuilder;
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v3
  
+     .line 178
      .local v3, len:I
      const/4 v2, 0x0
  
***************
*** 275,284 ****
--- 303,314 ----
      :goto_0
      if-ge v2, v3, :cond_2
  
+     .line 179
      invoke-virtual {p0, v2}, Ljava/lang/String;->charAt(I)C
  
      move-result v1
  
+     .line 180
      .local v1, c:C
      const/16 v4, 0x20
  
***************
*** 296,312 ****
--- 326,345 ----
  
      if-ne v1, v4, :cond_1
  
+     .line 178
      :cond_0
      :goto_1
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 181
      :cond_1
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
      goto :goto_1
  
+     .line 183
      .end local v1           #c:C
      :cond_2
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
***************
*** 327,362 ****
--- 360,406 ----
  
      const/4 v3, 0x0
  
+     .line 197
      new-instance v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;
  
      invoke-direct {v0}, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;-><init>()V
  
+     .line 198
      .local v0, addr:Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;
      iput-object p0, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->address:Ljava/lang/String;
  
+     .line 199
      iput v3, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->ton:I
  
+     .line 200
      const/4 v2, 0x0
  
+     .line 201
      .local v2, origBytes:[B
      invoke-static {p0}, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->filterNumericSugar(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
+     .line 202
      .local v1, filteredAddr:Ljava/lang/String;
      if-eqz v1, :cond_0
  
+     .line 203
      invoke-static {v1}, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->parseToDtmf(Ljava/lang/String;)[B
  
      move-result-object v2
  
+     .line 205
      :cond_0
      if-eqz v2, :cond_2
  
+     .line 206
      iput v3, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->digitMode:I
  
+     .line 207
      iput v3, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numberMode:I
  
+     .line 208
      const/16 v3, 0x2b
  
      invoke-virtual {p0, v3}, Ljava/lang/String;->indexOf(I)I
***************
*** 365,405 ****
--- 409,460 ----
  
      if-eq v3, v5, :cond_1
  
+     .line 209
      iput v4, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->ton:I
  
+     .line 223
      :cond_1
      :goto_0
      iput-object v2, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->origBytes:[B
  
+     .line 224
      array-length v3, v2
  
      iput v3, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numberOfDigits:I
  
+     .line 225
      .end local v0           #addr:Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;
      :goto_1
      return-object v0
  
+     .line 212
      .restart local v0       #addr:Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;
      :cond_2
      invoke-static {p0}, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->filterWhitespace(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
+     .line 213
      invoke-static {v1}, Lcom/android/internal/telephony/cdma/sms/UserData;->stringToAscii(Ljava/lang/String;)[B
  
      move-result-object v2
  
+     .line 214
      if-nez v2, :cond_3
  
+     .line 215
      const/4 v0, 0x0
  
      goto :goto_1
  
+     .line 217
      :cond_3
      iput v4, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->digitMode:I
  
+     .line 218
      iput v4, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->numberMode:I
  
+     .line 219
      const/16 v3, 0x40
  
      invoke-virtual {p0, v3}, Ljava/lang/String;->indexOf(I)I
***************
*** 408,413 ****
--- 463,469 ----
  
      if-eq v3, v5, :cond_1
  
+     .line 220
      const/4 v3, 0x2
  
      iput v3, v0, Lcom/android/internal/telephony/cdma/sms/CdmaSmsAddress;->ton:I
***************
*** 420,432 ****
--- 476,491 ----
      .parameter "address"
  
      .prologue
+     .line 118
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v1
  
+     .line 119
      .local v1, digits:I
      new-array v3, v1, [B
  
+     .line 120
      .local v3, result:[B
      const/4 v2, 0x0
  
***************
*** 434,446 ****
--- 493,508 ----
      :goto_0
      if-ge v2, v1, :cond_4
  
+     .line 121
      invoke-virtual {p0, v2}, Ljava/lang/String;->charAt(I)C
  
      move-result v0
  
+     .line 122
      .local v0, c:C
      const/4 v4, 0x0
  
+     .line 123
      .local v4, val:I
      const/16 v5, 0x31
  
***************
*** 452,466 ****
--- 514,531 ----
  
      add-int/lit8 v4, v0, -0x30
  
+     .line 128
      :goto_1
      int-to-byte v5, v4
  
      aput-byte v5, v3, v2
  
+     .line 120
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 124
      :cond_0
      const/16 v5, 0x30
  
***************
*** 470,475 ****
--- 535,541 ----
  
      goto :goto_1
  
+     .line 125
      :cond_1
      const/16 v5, 0x2a
  
***************
*** 479,484 ****
--- 545,551 ----
  
      goto :goto_1
  
+     .line 126
      :cond_2
      const/16 v5, 0x23
  
***************
*** 488,496 ****
--- 555,565 ----
  
      goto :goto_1
  
+     .line 127
      :cond_3
      const/4 v3, 0x0
  
+     .line 130
      .end local v0           #c:C
      .end local v3           #result:[B
      .end local v4           #val:I
***************
*** 504,523 ****
      .locals 3
  
      .prologue
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
      .local v0, builder:Ljava/lang/StringBuilder;
      const-string v1, "CdmaSmsAddress "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "{ digitMode="
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 573,595 ----
      .locals 3
  
      .prologue
+     .line 98
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 99
      .local v0, builder:Ljava/lang/StringBuilder;
      const-string v1, "CdmaSmsAddress "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 100
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "{ digitMode="
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 535,540 ****
--- 607,613 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 101
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 557,562 ****
--- 630,636 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 102
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 579,584 ****
--- 653,659 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 103
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 601,606 ****
--- 676,682 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 104
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 623,628 ****
--- 699,705 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 105
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 651,656 ****
--- 728,734 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 106
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 677,686 ****
--- 755,766 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 107
      const-string v1, " }"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 108
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
