*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/sms/CdmaSmsSubaddress.smali	2012-03-23 19:37:38.654479945 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/sms/CdmaSmsSubaddress.smali.original	2012-03-23 19:34:46.185334975 -0500
***************
*** 16,21 ****
--- 16,22 ----
      .locals 0
  
      .prologue
+     .line 20
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
