*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/sms/SmsEnvelope.smali	2012-03-23 19:37:38.576478524 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/cdma/sms/SmsEnvelope.smali.original	2012-03-23 19:34:46.184334957 -0500
***************
*** 54,64 ****
--- 54,67 ----
      .locals 1
  
      .prologue
+     .line 121
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 68
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/telephony/cdma/sms/SmsEnvelope;->teleService:I
  
+     .line 123
      return-void
  .end method
