*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/AbstractWindowedCursor.smali	2012-03-23 19:36:10.590874108 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/AbstractWindowedCursor.smali.original	2012-03-23 19:34:45.005313457 -0500
***************
*** 12,17 ****
--- 12,18 ----
      .locals 0
  
      .prologue
+     .line 36
      invoke-direct {p0}, Landroid/database/AbstractCursor;-><init>()V
  
      return-void
***************
*** 23,34 ****
--- 24,38 ----
      .locals 2
  
      .prologue
+     .line 136
      invoke-super {p0}, Landroid/database/AbstractCursor;->checkPosition()V
  
+     .line 138
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      if-nez v0, :cond_0
  
+     .line 139
      new-instance v0, Landroid/database/StaleDataException;
  
      const-string v1, "Attempting to access a closed CursorWindow.Most probable cause: cursor is deactivated prior to calling this method."
***************
*** 37,42 ****
--- 41,47 ----
  
      throw v0
  
+     .line 142
      :cond_0
      return-void
  .end method
***************
*** 46,64 ****
--- 51,73 ----
      .parameter "name"
  
      .prologue
+     .line 197
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      if-nez v0, :cond_0
  
+     .line 198
      new-instance v0, Landroid/database/CursorWindow;
  
      invoke-direct {v0, p1}, Landroid/database/CursorWindow;-><init>(Ljava/lang/String;)V
  
      iput-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
+     .line 202
      :goto_0
      return-void
  
+     .line 200
      :cond_0
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
***************
*** 71,88 ****
--- 80,101 ----
      .locals 1
  
      .prologue
+     .line 183
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      if-eqz v0, :cond_0
  
+     .line 184
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      invoke-virtual {v0}, Landroid/database/CursorWindow;->close()V
  
+     .line 185
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
+     .line 187
      :cond_0
      return-void
  .end method
***************
*** 93,106 ****
--- 106,122 ----
      .parameter "buffer"
  
      .prologue
+     .line 56
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 57
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
  
      invoke-virtual {v0, v1, p1, p2}, Landroid/database/CursorWindow;->copyStringToBuffer(IILandroid/database/CharArrayBuffer;)V
  
+     .line 58
      return-void
  .end method
  
***************
*** 109,116 ****
--- 125,134 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 44
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 45
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
***************
*** 127,134 ****
--- 145,154 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 86
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 87
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
***************
*** 145,152 ****
--- 165,174 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 80
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 81
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
***************
*** 163,170 ****
--- 185,194 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 68
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 69
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
***************
*** 181,188 ****
--- 205,214 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 74
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 75
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
***************
*** 199,206 ****
--- 225,234 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 62
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 63
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
***************
*** 217,224 ****
--- 245,254 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 50
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 51
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
***************
*** 235,242 ****
--- 265,274 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 130
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 131
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
***************
*** 252,257 ****
--- 284,290 ----
      .locals 1
  
      .prologue
+     .line 146
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      return-object v0
***************
*** 261,266 ****
--- 294,300 ----
      .locals 1
  
      .prologue
+     .line 175
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      if-eqz v0, :cond_0
***************
*** 283,288 ****
--- 317,323 ----
      .end annotation
  
      .prologue
+     .line 101
      invoke-virtual {p0, p1}, Landroid/database/AbstractWindowedCursor;->getType(I)I
  
      move-result v0
***************
*** 309,314 ****
--- 344,350 ----
      .end annotation
  
      .prologue
+     .line 125
      invoke-virtual {p0, p1}, Landroid/database/AbstractWindowedCursor;->getType(I)I
  
      move-result v0
***************
*** 337,342 ****
--- 373,379 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 117
      invoke-virtual {p0, p1}, Landroid/database/AbstractWindowedCursor;->getType(I)I
  
      move-result v1
***************
*** 357,364 ****
--- 394,403 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 92
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->checkPosition()V
  
+     .line 93
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      iget v1, p0, Landroid/database/AbstractWindowedCursor;->mPos:I
***************
*** 387,392 ****
--- 426,432 ----
      .end annotation
  
      .prologue
+     .line 109
      invoke-virtual {p0, p1}, Landroid/database/AbstractWindowedCursor;->getType(I)I
  
      move-result v0
***************
*** 410,419 ****
--- 450,462 ----
      .locals 0
  
      .prologue
+     .line 207
      invoke-super {p0}, Landroid/database/AbstractCursor;->onDeactivateOrClose()V
  
+     .line 208
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->closeWindow()V
  
+     .line 209
      return-void
  .end method
  
***************
*** 422,435 ****
--- 465,482 ----
      .parameter "window"
  
      .prologue
+     .line 163
      iget-object v0, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
      if-eq p1, v0, :cond_0
  
+     .line 164
      invoke-virtual {p0}, Landroid/database/AbstractWindowedCursor;->closeWindow()V
  
+     .line 165
      iput-object p1, p0, Landroid/database/AbstractWindowedCursor;->mWindow:Landroid/database/CursorWindow;
  
+     .line 167
      :cond_0
      return-void
  .end method
