*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/CharArrayBuffer.smali	2012-03-23 19:36:08.618838148 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/CharArrayBuffer.smali.original	2012-03-23 19:34:44.999313349 -0500
***************
*** 15,26 ****
--- 15,29 ----
      .parameter "size"
  
      .prologue
+     .line 23
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 24
      new-array v0, p1, [C
  
      iput-object v0, p0, Landroid/database/CharArrayBuffer;->data:[C
  
+     .line 25
      return-void
  .end method
  
***************
*** 29,37 ****
--- 32,43 ----
      .parameter "buf"
  
      .prologue
+     .line 27
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 28
      iput-object p1, p0, Landroid/database/CharArrayBuffer;->data:[C
  
+     .line 29
      return-void
  .end method
