*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/CursorJoiner$1.smali	2012-03-23 19:36:10.070864625 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/CursorJoiner$1.smali.original	2012-03-23 19:34:45.003313421 -0500
***************
*** 23,28 ****
--- 23,29 ----
      .locals 3
  
      .prologue
+     .line 122
      invoke-static {}, Landroid/database/CursorJoiner$Result;->values()[Landroid/database/CursorJoiner$Result;
  
      move-result-object v0
