*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/DataSetObserver.smali	2012-03-23 19:36:10.205867087 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/DataSetObserver.smali.original	2012-03-23 19:34:45.004313439 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 24
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 19,24 ****
--- 20,26 ----
      .locals 0
  
      .prologue
+     .line 31
      return-void
  .end method
  
***************
*** 26,30 ****
--- 28,33 ----
      .locals 0
  
      .prologue
+     .line 40
      return-void
  .end method
