*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/Observable.smali	2012-03-23 19:36:08.971844585 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/database/Observable.smali.original	2012-03-23 19:34:45.000313367 -0500
***************
*** 30,38 ****
--- 30,40 ----
      .locals 1
  
      .prologue
+     .line 24
      .local p0, this:Landroid/database/Observable;,"Landroid/database/Observable<TT;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 29
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
***************
*** 54,63 ****
--- 56,67 ----
      .end annotation
  
      .prologue
+     .line 39
      .local p0, this:Landroid/database/Observable;,"Landroid/database/Observable<TT;>;"
      .local p1, observer:Ljava/lang/Object;,"TT;"
      if-nez p1, :cond_0
  
+     .line 40
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "The observer is null."
***************
*** 66,76 ****
--- 70,82 ----
  
      throw v0
  
+     .line 42
      :cond_0
      iget-object v1, p0, Landroid/database/Observable;->mObservers:Ljava/util/ArrayList;
  
      monitor-enter v1
  
+     .line 43
      :try_start_0
      iget-object v0, p0, Landroid/database/Observable;->mObservers:Ljava/util/ArrayList;
  
***************
*** 80,85 ****
--- 86,92 ----
  
      if-eqz v0, :cond_1
  
+     .line 44
      new-instance v0, Ljava/lang/IllegalStateException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 110,115 ****
--- 117,123 ----
  
      throw v0
  
+     .line 47
      :catchall_0
      move-exception v0
  
***************
*** 119,134 ****
--- 127,145 ----
  
      throw v0
  
+     .line 46
      :cond_1
      :try_start_1
      iget-object v0, p0, Landroid/database/Observable;->mObservers:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 47
      monitor-exit v1
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 48
      return-void
  .end method
  
***************
*** 136,155 ****
--- 147,171 ----
      .locals 2
  
      .prologue
+     .line 74
      .local p0, this:Landroid/database/Observable;,"Landroid/database/Observable<TT;>;"
      iget-object v1, p0, Landroid/database/Observable;->mObservers:Ljava/util/ArrayList;
  
      monitor-enter v1
  
+     .line 75
      :try_start_0
      iget-object v0, p0, Landroid/database/Observable;->mObservers:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
  
+     .line 76
      monitor-exit v1
  
+     .line 77
      return-void
  
+     .line 76
      :catchall_0
      move-exception v0
  
***************
*** 170,179 ****
--- 186,197 ----
      .end annotation
  
      .prologue
+     .line 58
      .local p0, this:Landroid/database/Observable;,"Landroid/database/Observable<TT;>;"
      .local p1, observer:Ljava/lang/Object;,"TT;"
      if-nez p1, :cond_0
  
+     .line 59
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
      const-string v2, "The observer is null."
***************
*** 182,192 ****
--- 200,212 ----
  
      throw v1
  
+     .line 61
      :cond_0
      iget-object v2, p0, Landroid/database/Observable;->mObservers:Ljava/util/ArrayList;
  
      monitor-enter v2
  
+     .line 62
      :try_start_0
      iget-object v1, p0, Landroid/database/Observable;->mObservers:Ljava/util/ArrayList;
  
***************
*** 194,204 ****
--- 214,226 ----
  
      move-result v0
  
+     .line 63
      .local v0, index:I
      const/4 v1, -0x1
  
      if-ne v0, v1, :cond_1
  
+     .line 64
      new-instance v1, Ljava/lang/IllegalStateException;
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 229,234 ****
--- 251,257 ----
  
      throw v1
  
+     .line 67
      .end local v0           #index:I
      :catchall_0
      move-exception v1
***************
*** 239,244 ****
--- 262,268 ----
  
      throw v1
  
+     .line 66
      .restart local v0       #index:I
      :cond_1
      :try_start_1
***************
*** 246,254 ****
--- 270,280 ----
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 67
      monitor-exit v2
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 68
      return-void
  .end method
