*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/style/AbsoluteSizeSpan.smali	2012-03-23 19:35:21.160972756 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/style/AbsoluteSizeSpan.smali.original	2012-03-23 19:34:44.507304377 -0500
***************
*** 18,27 ****
--- 18,30 ----
      .parameter "size"
  
      .prologue
+     .line 32
      invoke-direct {p0}, Landroid/text/style/MetricAffectingSpan;-><init>()V
  
+     .line 33
      iput p1, p0, Landroid/text/style/AbsoluteSizeSpan;->mSize:I
  
+     .line 34
      return-void
  .end method
  
***************
*** 31,42 ****
--- 34,49 ----
      .parameter "dip"
  
      .prologue
+     .line 41
      invoke-direct {p0}, Landroid/text/style/MetricAffectingSpan;-><init>()V
  
+     .line 42
      iput p1, p0, Landroid/text/style/AbsoluteSizeSpan;->mSize:I
  
+     .line 43
      iput-boolean p2, p0, Landroid/text/style/AbsoluteSizeSpan;->mDip:Z
  
+     .line 44
      return-void
  .end method
  
***************
*** 45,58 ****
--- 52,68 ----
      .parameter "src"
  
      .prologue
+     .line 46
      invoke-direct {p0}, Landroid/text/style/MetricAffectingSpan;-><init>()V
  
+     .line 47
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mSize:I
  
+     .line 48
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
***************
*** 64,71 ****
--- 74,83 ----
      :goto_0
      iput-boolean v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mDip:Z
  
+     .line 49
      return-void
  
+     .line 48
      :cond_0
      const/4 v0, 0x0
  
***************
*** 78,83 ****
--- 90,96 ----
      .locals 1
  
      .prologue
+     .line 56
      const/4 v0, 0x0
  
      return v0
***************
*** 87,92 ****
--- 100,106 ----
      .locals 1
  
      .prologue
+     .line 69
      iget-boolean v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mDip:Z
  
      return v0
***************
*** 96,101 ****
--- 110,116 ----
      .locals 1
  
      .prologue
+     .line 65
      iget v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mSize:I
  
      return v0
***************
*** 105,110 ****
--- 120,126 ----
      .locals 1
  
      .prologue
+     .line 52
      const/16 v0, 0x10
  
      return v0
***************
*** 115,124 ****
--- 131,142 ----
      .parameter "ds"
  
      .prologue
+     .line 74
      iget-boolean v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mDip:Z
  
      if-eqz v0, :cond_0
  
+     .line 75
      iget v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mSize:I
  
      int-to-float v0, v0
***************
*** 129,137 ****
--- 147,157 ----
  
      invoke-virtual {p1, v0}, Landroid/text/TextPaint;->setTextSize(F)V
  
+     .line 79
      :goto_0
      return-void
  
+     .line 77
      :cond_0
      iget v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mSize:I
  
***************
*** 147,156 ****
--- 167,178 ----
      .parameter "ds"
  
      .prologue
+     .line 83
      iget-boolean v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mDip:Z
  
      if-eqz v0, :cond_0
  
+     .line 84
      iget v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mSize:I
  
      int-to-float v0, v0
***************
*** 161,169 ****
--- 183,193 ----
  
      invoke-virtual {p1, v0}, Landroid/text/TextPaint;->setTextSize(F)V
  
+     .line 88
      :goto_0
      return-void
  
+     .line 86
      :cond_0
      iget v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mSize:I
  
***************
*** 180,189 ****
--- 204,215 ----
      .parameter "flags"
  
      .prologue
+     .line 60
      iget v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mSize:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 61
      iget-boolean v0, p0, Landroid/text/style/AbsoluteSizeSpan;->mDip:Z
  
      if-eqz v0, :cond_0
***************
*** 193,200 ****
--- 219,228 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 62
      return-void
  
+     .line 61
      :cond_0
      const/4 v0, 0x0
  
