*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/style/ForegroundColorSpan.smali	2012-03-23 19:35:20.425959352 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/style/ForegroundColorSpan.smali.original	2012-03-23 19:34:44.498304213 -0500
***************
*** 17,26 ****
--- 17,29 ----
      .parameter "color"
  
      .prologue
+     .line 29
      invoke-direct {p0}, Landroid/text/style/CharacterStyle;-><init>()V
  
+     .line 30
      iput p1, p0, Landroid/text/style/ForegroundColorSpan;->mColor:I
  
+     .line 31
      return-void
  .end method
  
***************
*** 29,42 ****
--- 32,48 ----
      .parameter "src"
  
      .prologue
+     .line 33
      invoke-direct {p0}, Landroid/text/style/CharacterStyle;-><init>()V
  
+     .line 34
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/ForegroundColorSpan;->mColor:I
  
+     .line 35
      return-void
  .end method
  
***************
*** 46,51 ****
--- 52,58 ----
      .locals 1
  
      .prologue
+     .line 42
      const/4 v0, 0x0
  
      return v0
***************
*** 55,60 ****
--- 62,68 ----
      .locals 1
  
      .prologue
+     .line 50
      iget v0, p0, Landroid/text/style/ForegroundColorSpan;->mColor:I
  
      return v0
***************
*** 64,69 ****
--- 72,78 ----
      .locals 1
  
      .prologue
+     .line 38
      const/4 v0, 0x2
  
      return v0
***************
*** 74,83 ****
--- 83,94 ----
      .parameter "ds"
  
      .prologue
+     .line 55
      iget v0, p0, Landroid/text/style/ForegroundColorSpan;->mColor:I
  
      invoke-virtual {p1, v0}, Landroid/text/TextPaint;->setColor(I)V
  
+     .line 56
      return-void
  .end method
  
***************
*** 87,95 ****
--- 98,108 ----
      .parameter "flags"
  
      .prologue
+     .line 46
      iget v0, p0, Landroid/text/style/ForegroundColorSpan;->mColor:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 47
      return-void
  .end method
