*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/style/SuggestionSpan.smali	2012-03-23 19:35:20.579962162 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/style/SuggestionSpan.smali.original	2012-03-23 19:34:44.500304249 -0500
***************
*** 64,69 ****
--- 64,70 ----
      .locals 1
  
      .prologue
+     .line 272
      new-instance v0, Landroid/text/style/SuggestionSpan$1;
  
      invoke-direct {v0}, Landroid/text/style/SuggestionSpan$1;-><init>()V
***************
*** 94,102 ****
--- 95,105 ----
      .end annotation
  
      .prologue
+     .line 123
      .local p5, notificationTargetClass:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
      invoke-direct {p0}, Landroid/text/style/CharacterStyle;-><init>()V
  
+     .line 124
      const/4 v1, 0x5
  
      array-length v2, p3
***************
*** 105,110 ****
--- 108,114 ----
  
      move-result v0
  
+     .line 125
      .local v0, N:I
      invoke-static {p3, v0}, Ljava/util/Arrays;->copyOf([Ljava/lang/Object;I)[Ljava/lang/Object;
  
***************
*** 114,125 ****
--- 118,132 ----
  
      iput-object v1, p0, Landroid/text/style/SuggestionSpan;->mSuggestions:[Ljava/lang/String;
  
+     .line 126
      iput p4, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
  
+     .line 127
      if-eqz p1, :cond_0
  
      if-nez p2, :cond_0
  
+     .line 128
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v1
***************
*** 136,150 ****
--- 143,160 ----
  
      iput-object v1, p0, Landroid/text/style/SuggestionSpan;->mLocaleString:Ljava/lang/String;
  
+     .line 133
      :goto_0
      if-eqz p5, :cond_1
  
+     .line 134
      invoke-virtual {p5}, Ljava/lang/Class;->getCanonicalName()Ljava/lang/String;
  
      move-result-object v1
  
      iput-object v1, p0, Landroid/text/style/SuggestionSpan;->mNotificationTargetClassName:Ljava/lang/String;
  
+     .line 138
      :goto_1
      iget-object v1, p0, Landroid/text/style/SuggestionSpan;->mSuggestions:[Ljava/lang/String;
  
***************
*** 158,167 ****
--- 168,180 ----
  
      iput v1, p0, Landroid/text/style/SuggestionSpan;->mHashCode:I
  
+     .line 140
      invoke-direct {p0, p1}, Landroid/text/style/SuggestionSpan;->initStyle(Landroid/content/Context;)V
  
+     .line 141
      return-void
  
+     .line 130
      :cond_0
      invoke-virtual {p2}, Ljava/util/Locale;->toString()Ljava/lang/String;
  
***************
*** 171,176 ****
--- 184,190 ----
  
      goto :goto_0
  
+     .line 136
      :cond_1
      const-string v1, ""
  
***************
*** 188,193 ****
--- 202,208 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 101
      move-object v0, p0
  
      move-object v1, p1
***************
*** 200,205 ****
--- 215,221 ----
  
      invoke-direct/range {v0 .. v5}, Landroid/text/style/SuggestionSpan;-><init>(Landroid/content/Context;Ljava/util/Locale;[Ljava/lang/String;ILjava/lang/Class;)V
  
+     .line 102
      return-void
  .end method
  
***************
*** 208,281 ****
--- 224,310 ----
      .parameter "src"
  
      .prologue
+     .line 179
      invoke-direct {p0}, Landroid/text/style/CharacterStyle;-><init>()V
  
+     .line 180
      invoke-virtual {p1}, Landroid/os/Parcel;->readStringArray()[Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/text/style/SuggestionSpan;->mSuggestions:[Ljava/lang/String;
  
+     .line 181
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
  
+     .line 182
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/text/style/SuggestionSpan;->mLocaleString:Ljava/lang/String;
  
+     .line 183
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/text/style/SuggestionSpan;->mNotificationTargetClassName:Ljava/lang/String;
  
+     .line 184
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/SuggestionSpan;->mHashCode:I
  
+     .line 185
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineColor:I
  
+     .line 186
      invoke-virtual {p1}, Landroid/os/Parcel;->readFloat()F
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineThickness:F
  
+     .line 187
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineColor:I
  
+     .line 188
      invoke-virtual {p1}, Landroid/os/Parcel;->readFloat()F
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineThickness:F
  
+     .line 189
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineColor:I
  
+     .line 190
      invoke-virtual {p1}, Landroid/os/Parcel;->readFloat()F
  
      move-result v0
  
      iput v0, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineThickness:F
  
+     .line 191
      return-void
  .end method
  
***************
*** 288,293 ****
--- 317,323 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 110
      move-object v0, p0
  
      move-object v2, p1
***************
*** 300,305 ****
--- 330,336 ----
  
      invoke-direct/range {v0 .. v5}, Landroid/text/style/SuggestionSpan;-><init>(Landroid/content/Context;Ljava/util/Locale;[Ljava/lang/String;ILjava/lang/Class;)V
  
+     .line 111
      return-void
  .end method
  
***************
*** 310,315 ****
--- 341,347 ----
      .parameter "notificationTargetClassName"
  
      .prologue
+     .line 268
      const/4 v0, 0x4
  
      new-array v0, v0, [Ljava/lang/Object;
***************
*** 360,385 ****
--- 392,427 ----
  
      const/4 v3, 0x0
  
+     .line 144
      if-nez p1, :cond_0
  
+     .line 145
      iput v3, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineThickness:F
  
+     .line 146
      iput v3, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineThickness:F
  
+     .line 147
      iput v3, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineThickness:F
  
+     .line 148
      iput v4, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineColor:I
  
+     .line 149
      iput v4, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineColor:I
  
+     .line 150
      iput v4, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineColor:I
  
+     .line 177
      :goto_0
      return-void
  
+     .line 154
      :cond_0
      const v0, 0x10103a9
  
+     .line 155
      .local v0, defStyle:I
      sget-object v2, Lcom/android/internal/R$styleable;->SuggestionSpan:[I
  
***************
*** 387,392 ****
--- 429,435 ----
  
      move-result-object v1
  
+     .line 157
      .local v1, typedArray:Landroid/content/res/TypedArray;
      invoke-virtual {v1, v6, v3}, Landroid/content/res/TypedArray;->getDimension(IF)F
  
***************
*** 394,439 ****
--- 437,491 ----
  
      iput v2, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineThickness:F
  
+     .line 159
      invoke-virtual {v1, v5, v4}, Landroid/content/res/TypedArray;->getColor(II)I
  
      move-result v2
  
      iput v2, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineColor:I
  
+     .line 162
      const v0, 0x10103a8
  
+     .line 163
      sget-object v2, Lcom/android/internal/R$styleable;->SuggestionSpan:[I
  
      invoke-virtual {p1, v7, v2, v0, v5}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[III)Landroid/content/res/TypedArray;
  
      move-result-object v1
  
+     .line 165
      invoke-virtual {v1, v6, v3}, Landroid/content/res/TypedArray;->getDimension(IF)F
  
      move-result v2
  
      iput v2, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineThickness:F
  
+     .line 167
      invoke-virtual {v1, v5, v4}, Landroid/content/res/TypedArray;->getColor(II)I
  
      move-result v2
  
      iput v2, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineColor:I
  
+     .line 170
      const v0, 0x10103aa
  
+     .line 171
      sget-object v2, Lcom/android/internal/R$styleable;->SuggestionSpan:[I
  
      invoke-virtual {p1, v7, v2, v0, v5}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[III)Landroid/content/res/TypedArray;
  
      move-result-object v1
  
+     .line 173
      invoke-virtual {v1, v6, v3}, Landroid/content/res/TypedArray;->getDimension(IF)F
  
      move-result v2
  
      iput v2, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineThickness:F
  
+     .line 175
      invoke-virtual {v1, v5, v4}, Landroid/content/res/TypedArray;->getColor(II)I
  
      move-result v2
***************
*** 449,454 ****
--- 501,507 ----
      .locals 1
  
      .prologue
+     .line 230
      const/4 v0, 0x0
  
      return v0
***************
*** 461,470 ****
--- 514,525 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 255
      instance-of v1, p1, Landroid/text/style/SuggestionSpan;
  
      if-eqz v1, :cond_0
  
+     .line 256
      check-cast p1, Landroid/text/style/SuggestionSpan;
  
      .end local p1
***************
*** 478,483 ****
--- 533,539 ----
  
      const/4 v0, 0x1
  
+     .line 258
      :cond_0
      return v0
  .end method
***************
*** 486,491 ****
--- 542,548 ----
      .locals 1
  
      .prologue
+     .line 221
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
  
      return v0
***************
*** 495,500 ****
--- 552,558 ----
      .locals 1
  
      .prologue
+     .line 204
      iget-object v0, p0, Landroid/text/style/SuggestionSpan;->mLocaleString:Ljava/lang/String;
  
      return-object v0
***************
*** 504,509 ****
--- 562,568 ----
      .locals 1
  
      .prologue
+     .line 217
      iget-object v0, p0, Landroid/text/style/SuggestionSpan;->mNotificationTargetClassName:Ljava/lang/String;
  
      return-object v0
***************
*** 513,518 ****
--- 572,578 ----
      .locals 1
  
      .prologue
+     .line 250
      const/16 v0, 0x13
  
      return v0
***************
*** 522,527 ****
--- 582,588 ----
      .locals 1
  
      .prologue
+     .line 197
      iget-object v0, p0, Landroid/text/style/SuggestionSpan;->mSuggestions:[Ljava/lang/String;
  
      return-object v0
***************
*** 535,540 ****
--- 596,602 ----
  
      const/4 v4, 0x0
  
+     .line 310
      iget v5, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
  
      and-int/lit8 v5, v5, 0x2
***************
*** 543,548 ****
--- 605,611 ----
  
      move v2, v3
  
+     .line 311
      .local v2, misspelled:Z
      :goto_0
      iget v5, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
***************
*** 553,558 ****
--- 616,622 ----
  
      move v1, v3
  
+     .line 312
      .local v1, easy:Z
      :goto_1
      iget v5, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
***************
*** 563,576 ****
--- 627,644 ----
  
      move v0, v3
  
+     .line 313
      .local v0, autoCorrection:Z
      :goto_2
      if-eqz v1, :cond_5
  
+     .line 314
      if-nez v2, :cond_4
  
+     .line 315
      iget v4, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineColor:I
  
+     .line 322
      :cond_0
      :goto_3
      return v4
***************
*** 581,609 ****
--- 649,683 ----
      :cond_1
      move v2, v4
  
+     .line 310
      goto :goto_0
  
      .restart local v2       #misspelled:Z
      :cond_2
      move v1, v4
  
+     .line 311
      goto :goto_1
  
      .restart local v1       #easy:Z
      :cond_3
      move v0, v4
  
+     .line 312
      goto :goto_2
  
+     .line 317
      .restart local v0       #autoCorrection:Z
      :cond_4
      iget v4, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineColor:I
  
      goto :goto_3
  
+     .line 319
      :cond_5
      if-eqz v0, :cond_0
  
+     .line 320
      iget v4, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineColor:I
  
      goto :goto_3
***************
*** 613,618 ****
--- 687,693 ----
      .locals 1
  
      .prologue
+     .line 263
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mHashCode:I
  
      return v0
***************
*** 623,630 ****
--- 698,707 ----
      .parameter "flags"
  
      .prologue
+     .line 225
      iput p1, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
  
+     .line 226
      return-void
  .end method
  
***************
*** 637,642 ****
--- 714,720 ----
  
      const/4 v4, 0x0
  
+     .line 287
      iget v5, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
  
      and-int/lit8 v5, v5, 0x2
***************
*** 645,650 ****
--- 723,729 ----
  
      move v2, v3
  
+     .line 288
      .local v2, misspelled:Z
      :goto_0
      iget v5, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
***************
*** 655,660 ****
--- 734,740 ----
  
      move v1, v3
  
+     .line 289
      .local v1, easy:Z
      :goto_1
      iget v5, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
***************
*** 665,682 ****
--- 745,766 ----
  
      move v0, v3
  
+     .line 290
      .local v0, autoCorrection:Z
      :goto_2
      if-eqz v1, :cond_5
  
+     .line 291
      if-nez v2, :cond_4
  
+     .line 292
      iget v3, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineColor:I
  
      iget v4, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineThickness:F
  
      invoke-virtual {p1, v3, v4}, Landroid/text/TextPaint;->setUnderlineText(IF)V
  
+     .line 301
      :cond_0
      :goto_3
      return-void
***************
*** 687,712 ****
--- 771,801 ----
      :cond_1
      move v2, v4
  
+     .line 287
      goto :goto_0
  
      .restart local v2       #misspelled:Z
      :cond_2
      move v1, v4
  
+     .line 288
      goto :goto_1
  
      .restart local v1       #easy:Z
      :cond_3
      move v0, v4
  
+     .line 289
      goto :goto_2
  
+     .line 293
      .restart local v0       #autoCorrection:Z
      :cond_4
      iget v3, p1, Landroid/text/TextPaint;->underlineColor:I
  
      if-nez v3, :cond_0
  
+     .line 296
      iget v3, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineColor:I
  
      iget v4, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineThickness:F
***************
*** 715,723 ****
--- 804,814 ----
  
      goto :goto_3
  
+     .line 298
      :cond_5
      if-eqz v0, :cond_0
  
+     .line 299
      iget v3, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineColor:I
  
      iget v4, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineThickness:F
***************
*** 733,781 ****
--- 824,884 ----
      .parameter "flags"
  
      .prologue
+     .line 235
      iget-object v0, p0, Landroid/text/style/SuggestionSpan;->mSuggestions:[Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStringArray([Ljava/lang/String;)V
  
+     .line 236
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mFlags:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 237
      iget-object v0, p0, Landroid/text/style/SuggestionSpan;->mLocaleString:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 238
      iget-object v0, p0, Landroid/text/style/SuggestionSpan;->mNotificationTargetClassName:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 239
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mHashCode:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 240
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineColor:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 241
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mEasyCorrectUnderlineThickness:F
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeFloat(F)V
  
+     .line 242
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineColor:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 243
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mMisspelledUnderlineThickness:F
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeFloat(F)V
  
+     .line 244
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineColor:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 245
      iget v0, p0, Landroid/text/style/SuggestionSpan;->mAutoCorrectionUnderlineThickness:F
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeFloat(F)V
  
+     .line 246
      return-void
  .end method
