*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/style/TypefaceSpan.smali	2012-03-23 19:35:20.511960922 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/style/TypefaceSpan.smali.original	2012-03-23 19:34:44.499304231 -0500
***************
*** 16,29 ****
--- 16,32 ----
      .parameter "src"
  
      .prologue
+     .line 40
      invoke-direct {p0}, Landroid/text/style/MetricAffectingSpan;-><init>()V
  
+     .line 41
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/text/style/TypefaceSpan;->mFamily:Ljava/lang/String;
  
+     .line 42
      return-void
  .end method
  
***************
*** 32,41 ****
--- 35,47 ----
      .parameter "family"
  
      .prologue
+     .line 36
      invoke-direct {p0}, Landroid/text/style/MetricAffectingSpan;-><init>()V
  
+     .line 37
      iput-object p1, p0, Landroid/text/style/TypefaceSpan;->mFamily:Ljava/lang/String;
  
+     .line 38
      return-void
  .end method
  
***************
*** 45,65 ****
--- 51,76 ----
      .parameter "family"
  
      .prologue
+     .line 76
      invoke-virtual {p0}, Landroid/graphics/Paint;->getTypeface()Landroid/graphics/Typeface;
  
      move-result-object v1
  
+     .line 77
      .local v1, old:Landroid/graphics/Typeface;
      if-nez v1, :cond_2
  
+     .line 78
      const/4 v2, 0x0
  
+     .line 83
      .local v2, oldStyle:I
      :goto_0
      invoke-static {p1, v2}, Landroid/graphics/Typeface;->create(Ljava/lang/String;I)Landroid/graphics/Typeface;
  
      move-result-object v3
  
+     .line 84
      .local v3, tf:Landroid/graphics/Typeface;
      invoke-virtual {v3}, Landroid/graphics/Typeface;->getStyle()I
  
***************
*** 69,97 ****
--- 80,115 ----
  
      and-int v0, v2, v4
  
+     .line 86
      .local v0, fake:I
      and-int/lit8 v4, v0, 0x1
  
      if-eqz v4, :cond_0
  
+     .line 87
      const/4 v4, 0x1
  
      invoke-virtual {p0, v4}, Landroid/graphics/Paint;->setFakeBoldText(Z)V
  
+     .line 90
      :cond_0
      and-int/lit8 v4, v0, 0x2
  
      if-eqz v4, :cond_1
  
+     .line 91
      const/high16 v4, -0x4180
  
      invoke-virtual {p0, v4}, Landroid/graphics/Paint;->setTextSkewX(F)V
  
+     .line 94
      :cond_1
      invoke-virtual {p0, v3}, Landroid/graphics/Paint;->setTypeface(Landroid/graphics/Typeface;)Landroid/graphics/Typeface;
  
+     .line 95
      return-void
  
+     .line 80
      .end local v0           #fake:I
      .end local v2           #oldStyle:I
      .end local v3           #tf:Landroid/graphics/Typeface;
***************
*** 110,115 ****
--- 128,134 ----
      .locals 1
  
      .prologue
+     .line 49
      const/4 v0, 0x0
  
      return v0
***************
*** 119,124 ****
--- 138,144 ----
      .locals 1
  
      .prologue
+     .line 60
      iget-object v0, p0, Landroid/text/style/TypefaceSpan;->mFamily:Ljava/lang/String;
  
      return-object v0
***************
*** 128,133 ****
--- 148,154 ----
      .locals 1
  
      .prologue
+     .line 45
      const/16 v0, 0xd
  
      return v0
***************
*** 138,147 ****
--- 159,170 ----
      .parameter "ds"
  
      .prologue
+     .line 65
      iget-object v0, p0, Landroid/text/style/TypefaceSpan;->mFamily:Ljava/lang/String;
  
      invoke-static {p1, v0}, Landroid/text/style/TypefaceSpan;->apply(Landroid/graphics/Paint;Ljava/lang/String;)V
  
+     .line 66
      return-void
  .end method
  
***************
*** 150,159 ****
--- 173,184 ----
      .parameter "paint"
  
      .prologue
+     .line 70
      iget-object v0, p0, Landroid/text/style/TypefaceSpan;->mFamily:Ljava/lang/String;
  
      invoke-static {p1, v0}, Landroid/text/style/TypefaceSpan;->apply(Landroid/graphics/Paint;Ljava/lang/String;)V
  
+     .line 71
      return-void
  .end method
  
***************
*** 163,171 ****
--- 188,198 ----
      .parameter "flags"
  
      .prologue
+     .line 53
      iget-object v0, p0, Landroid/text/style/TypefaceSpan;->mFamily:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 54
      return-void
  .end method
