*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/test/InterpreterEx.smali	2012-03-23 19:37:40.550514521 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/test/InterpreterEx.smali.original	2012-03-23 19:34:46.230335796 -0500
***************
*** 13,21 ****
--- 13,24 ----
      .parameter "result"
  
      .prologue
+     .line 127
      invoke-direct {p0}, Ljava/lang/Exception;-><init>()V
  
+     .line 128
      iput-object p1, p0, Lcom/android/internal/telephony/test/InterpreterEx;->result:Ljava/lang/String;
  
+     .line 129
      return-void
  .end method
