*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/test/InvalidStateEx.smali	2012-03-23 19:37:40.659516507 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/test/InvalidStateEx.smali.original	2012-03-23 19:34:46.231335815 -0500
***************
*** 8,14 ****
--- 8,16 ----
      .locals 0
  
      .prologue
+     .line 126
      invoke-direct {p0}, Ljava/lang/Exception;-><init>()V
  
+     .line 128
      return-void
  .end method
