*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/test/LineReader.smali	2012-03-23 19:37:40.391511621 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/test/LineReader.smali.original	2012-03-23 19:34:46.227335741 -0500
***************
*** 19,34 ****
--- 19,38 ----
      .parameter "s"
  
      .prologue
+     .line 55
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 48
      const/16 v0, 0x1000
  
      new-array v0, v0, [B
  
      iput-object v0, p0, Lcom/android/internal/telephony/test/LineReader;->buffer:[B
  
+     .line 56
      iput-object p1, p0, Lcom/android/internal/telephony/test/LineReader;->inStream:Ljava/io/InputStream;
  
+     .line 57
      return-void
  .end method
  
***************
*** 38,43 ****
--- 42,48 ----
      .locals 1
  
      .prologue
+     .line 62
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/test/LineReader;->getNextLine(Z)Ljava/lang/String;
***************
*** 54,64 ****
--- 59,71 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 81
      const/4 v1, 0x0
  
      .local v1, i:I
      move v2, v1
  
+     .line 87
      .end local v1           #i:I
      .local v2, i:I
      :goto_0
***************
*** 72,88 ****
--- 79,98 ----
  
      move-result v3
  
+     .line 89
      .local v3, result:I
      if-gez v3, :cond_0
  
      move v1, v2
  
+     .line 116
      .end local v2           #i:I
      .end local v3           #result:I
      .restart local v1       #i:I
      :goto_1
      return-object v4
  
+     .line 93
      .end local v1           #i:I
      .restart local v2       #i:I
      .restart local v3       #result:I
***************
*** 96,101 ****
--- 106,112 ----
      :cond_1
      move v1, v2
  
+     .line 113
      .end local v2           #i:I
      .end local v3           #result:I
      .restart local v1       #i:I
***************
*** 117,122 ****
--- 128,134 ----
  
      goto :goto_1
  
+     .line 95
      .end local v1           #i:I
      .restart local v2       #i:I
      .restart local v3       #result:I
***************
*** 129,139 ****
--- 141,153 ----
  
      if-ne v3, v5, :cond_4
  
+     .line 96
      :cond_3
      if-nez v2, :cond_1
  
      goto :goto_0
  
+     .line 104
      :cond_4
      :try_start_2
      iget-object v5, p0, Lcom/android/internal/telephony/test/LineReader;->buffer:[B
***************
*** 155,176 ****
--- 169,194 ----
  
      move v2, v1
  
+     .line 105
      .end local v1           #i:I
      .restart local v2       #i:I
      goto :goto_0
  
+     .line 106
      .end local v3           #result:I
      :catch_0
      move-exception v0
  
      move v1, v2
  
+     .line 107
      .end local v2           #i:I
      .local v0, ex:Ljava/io/IOException;
      .restart local v1       #i:I
      :goto_3
      goto :goto_1
  
+     .line 108
      .end local v0           #ex:Ljava/io/IOException;
      .end local v1           #i:I
      .restart local v2       #i:I
***************
*** 179,184 ****
--- 197,203 ----
  
      move v1, v2
  
+     .line 109
      .end local v2           #i:I
      .local v0, ex:Ljava/lang/IndexOutOfBoundsException;
      .restart local v1       #i:I
***************
*** 191,200 ****
--- 210,221 ----
  
      goto :goto_2
  
+     .line 114
      .end local v0           #ex:Ljava/lang/IndexOutOfBoundsException;
      :catch_2
      move-exception v0
  
+     .line 115
      .local v0, ex:Ljava/io/UnsupportedEncodingException;
      sget-object v5, Ljava/lang/System;->err:Ljava/io/PrintStream;
  
***************
*** 204,209 ****
--- 225,231 ----
  
      goto :goto_1
  
+     .line 108
      .end local v0           #ex:Ljava/io/UnsupportedEncodingException;
      .restart local v3       #result:I
      :catch_3
***************
*** 211,216 ****
--- 233,239 ----
  
      goto :goto_4
  
+     .line 106
      :catch_4
      move-exception v0
  
***************
*** 221,226 ****
--- 244,250 ----
      .locals 1
  
      .prologue
+     .line 68
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/test/LineReader;->getNextLine(Z)Ljava/lang/String;
