*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/usb/IUsbManager$Stub$Proxy.smali	2012-03-23 19:36:02.833732657 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/usb/IUsbManager$Stub$Proxy.smali.original	2012-03-23 19:34:44.952312492 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 299
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 300
      iput-object p1, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 301
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 304
      iget-object v0, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 55,77 ****
--- 59,86 ----
      .end annotation
  
      .prologue
+     .line 643
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 644
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 646
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 647
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 648
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xe
***************
*** 80,100 ****
--- 89,115 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 649
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 652
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 653
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 655
      return-void
  
+     .line 652
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 653
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 109,129 ****
--- 124,148 ----
      .end annotation
  
      .prologue
+     .line 357
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 358
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 361
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 362
      iget-object v3, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x3
***************
*** 132,145 ****
--- 151,167 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 363
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 364
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 365
      sget-object v3, Landroid/hardware/usb/UsbAccessory;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 150,163 ****
--- 172,189 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 372
      .local v2, _result:Landroid/hardware/usb/UsbAccessory;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 373
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 375
      return-object v2
  
+     .line 368
      .end local v2           #_result:Landroid/hardware/usb/UsbAccessory;
      :cond_0
      const/4 v2, 0x0
***************
*** 165,176 ****
--- 191,204 ----
      .restart local v2       #_result:Landroid/hardware/usb/UsbAccessory;
      goto :goto_0
  
+     .line 372
      .end local v2           #_result:Landroid/hardware/usb/UsbAccessory;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 373
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 186,206 ****
--- 214,238 ----
      .end annotation
  
      .prologue
+     .line 313
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 314
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 316
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 317
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x1
***************
*** 209,238 ****
--- 241,278 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 318
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 319
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      if-eqz v2, :cond_0
  
+     .line 320
      invoke-virtual {p1, v1}, Landroid/os/Bundle;->readFromParcel(Landroid/os/Parcel;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 324
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 325
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 327
      return-void
  
+     .line 324
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 325
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 242,247 ****
--- 282,288 ----
      .locals 1
  
      .prologue
+     .line 308
      const-string v0, "android.hardware.usb.IUsbManager"
  
      return-object v0
***************
*** 258,291 ****
--- 299,340 ----
      .end annotation
  
      .prologue
+     .line 601
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 602
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 604
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 605
      if-eqz p1, :cond_0
  
+     .line 606
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 607
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/hardware/usb/UsbAccessory;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 612
      :goto_0
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 613
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xc
***************
*** 294,309 ****
--- 343,363 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 614
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 617
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 618
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 620
      return-void
  
+     .line 610
      :cond_0
      const/4 v2, 0x0
  
***************
*** 314,324 ****
--- 368,380 ----
  
      goto :goto_0
  
+     .line 617
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 618
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 335,368 ****
--- 391,432 ----
      .end annotation
  
      .prologue
+     .line 578
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 579
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 581
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 582
      if-eqz p1, :cond_0
  
+     .line 583
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 584
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/hardware/usb/UsbDevice;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 589
      :goto_0
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 590
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xb
***************
*** 371,386 ****
--- 435,455 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 591
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 594
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 595
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 597
      return-void
  
+     .line 587
      :cond_0
      const/4 v2, 0x0
  
***************
*** 391,401 ****
--- 460,472 ----
  
      goto :goto_0
  
+     .line 594
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 595
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 415,445 ****
--- 486,523 ----
  
      const/4 v3, 0x0
  
+     .line 487
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 488
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 491
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 492
      if-eqz p1, :cond_0
  
+     .line 493
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 494
      const/4 v4, 0x0
  
      invoke-virtual {p1, v0, v4}, Landroid/hardware/usb/UsbAccessory;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 499
      :goto_0
      iget-object v4, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 449,456 ****
--- 527,536 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 500
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 501
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 459,472 ****
--- 539,556 ----
  
      if-eqz v4, :cond_1
  
+     .line 504
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 505
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 507
      return v2
  
+     .line 497
      .end local v2           #_result:Z
      :cond_0
      const/4 v4, 0x0
***************
*** 478,488 ****
--- 562,574 ----
  
      goto :goto_0
  
+     .line 504
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 505
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 490,495 ****
--- 576,582 ----
      :cond_1
      move v2, v3
  
+     .line 501
      goto :goto_1
  .end method
  
***************
*** 505,527 ****
--- 592,619 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 624
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 625
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 628
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 629
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 630
      iget-object v3, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xd
***************
*** 530,537 ****
--- 622,631 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 631
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 632
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 542,561 ****
--- 636,660 ----
  
      const/4 v2, 0x1
  
+     .line 635
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 636
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 638
      return v2
  
+     .line 635
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 636
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 575,605 ****
--- 674,711 ----
  
      const/4 v3, 0x0
  
+     .line 462
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 463
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 466
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 467
      if-eqz p1, :cond_0
  
+     .line 468
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 469
      const/4 v4, 0x0
  
      invoke-virtual {p1, v0, v4}, Landroid/hardware/usb/UsbDevice;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 474
      :goto_0
      iget-object v4, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 609,616 ****
--- 715,724 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 475
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 476
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 619,632 ****
--- 727,744 ----
  
      if-eqz v4, :cond_1
  
+     .line 479
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 480
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 482
      return v2
  
+     .line 472
      .end local v2           #_result:Z
      :cond_0
      const/4 v4, 0x0
***************
*** 638,648 ****
--- 750,762 ----
  
      goto :goto_0
  
+     .line 479
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 480
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 650,655 ****
--- 764,770 ----
      :cond_1
      move v2, v3
  
+     .line 476
      goto :goto_1
  .end method
  
***************
*** 663,693 ****
--- 778,815 ----
      .end annotation
  
      .prologue
+     .line 382
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 383
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 386
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 387
      if-eqz p1, :cond_0
  
+     .line 388
      const/4 v3, 0x1
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 389
      const/4 v3, 0x0
  
      invoke-virtual {p1, v0, v3}, Landroid/hardware/usb/UsbAccessory;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 394
      :goto_0
      iget-object v3, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 697,710 ****
--- 819,835 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 395
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 396
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_1
  
+     .line 397
      sget-object v3, Landroid/os/ParcelFileDescriptor;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 715,728 ****
--- 840,857 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 404
      .local v2, _result:Landroid/os/ParcelFileDescriptor;
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 405
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 407
      return-object v2
  
+     .line 392
      .end local v2           #_result:Landroid/os/ParcelFileDescriptor;
      :cond_0
      const/4 v3, 0x0
***************
*** 734,748 ****
--- 863,880 ----
  
      goto :goto_0
  
+     .line 404
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 405
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
  
+     .line 400
      :cond_1
      const/4 v2, 0x0
  
***************
*** 760,782 ****
--- 892,919 ----
      .end annotation
  
      .prologue
+     .line 333
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 334
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 337
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 338
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 339
      iget-object v3, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x2
***************
*** 785,798 ****
--- 922,938 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 340
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 341
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 342
      sget-object v3, Landroid/os/ParcelFileDescriptor;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 803,816 ****
--- 943,960 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 349
      .local v2, _result:Landroid/os/ParcelFileDescriptor;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 350
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 352
      return-object v2
  
+     .line 345
      .end local v2           #_result:Landroid/os/ParcelFileDescriptor;
      :cond_0
      const/4 v2, 0x0
***************
*** 818,829 ****
--- 962,975 ----
      .restart local v2       #_result:Landroid/os/ParcelFileDescriptor;
      goto :goto_0
  
+     .line 349
      .end local v2           #_result:Landroid/os/ParcelFileDescriptor;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 350
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 841,884 ****
--- 987,1041 ----
      .end annotation
  
      .prologue
+     .line 548
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 549
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 551
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 552
      if-eqz p1, :cond_0
  
+     .line 553
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 554
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/hardware/usb/UsbAccessory;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 559
      :goto_0
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 560
      if-eqz p3, :cond_1
  
+     .line 561
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 562
      const/4 v2, 0x0
  
      invoke-virtual {p3, v0, v2}, Landroid/app/PendingIntent;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 567
      :goto_1
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 888,903 ****
--- 1045,1065 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 568
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 571
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 572
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 574
      return-void
  
+     .line 557
      :cond_0
      const/4 v2, 0x0
  
***************
*** 908,922 ****
--- 1070,1087 ----
  
      goto :goto_0
  
+     .line 571
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 572
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
  
+     .line 565
      :cond_1
      const/4 v2, 0x0
  
***************
*** 940,983 ****
--- 1105,1159 ----
      .end annotation
  
      .prologue
+     .line 515
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 516
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 518
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 519
      if-eqz p1, :cond_0
  
+     .line 520
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 521
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/hardware/usb/UsbDevice;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 526
      :goto_0
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 527
      if-eqz p3, :cond_1
  
+     .line 528
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 529
      const/4 v2, 0x0
  
      invoke-virtual {p3, v0, v2}, Landroid/app/PendingIntent;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 534
      :goto_1
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 987,1002 ****
--- 1163,1183 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 535
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 538
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 539
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 541
      return-void
  
+     .line 524
      :cond_0
      const/4 v2, 0x0
  
***************
*** 1007,1021 ****
--- 1188,1205 ----
  
      goto :goto_0
  
+     .line 538
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 539
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
  
+     .line 532
      :cond_1
      const/4 v2, 0x0
  
***************
*** 1038,1071 ****
--- 1222,1263 ----
      .end annotation
  
      .prologue
+     .line 439
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 440
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 442
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 443
      if-eqz p1, :cond_0
  
+     .line 444
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 445
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/hardware/usb/UsbAccessory;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 450
      :goto_0
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 451
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x6
***************
*** 1074,1089 ****
--- 1266,1286 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 452
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 455
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 456
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 458
      return-void
  
+     .line 448
      :cond_0
      const/4 v2, 0x0
  
***************
*** 1094,1104 ****
--- 1291,1303 ----
  
      goto :goto_0
  
+     .line 455
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 456
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1117,1139 ****
--- 1316,1343 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 659
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 660
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 662
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 663
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 664
      if-eqz p2, :cond_0
  
      const/4 v2, 0x1
***************
*** 1141,1146 ****
--- 1345,1351 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 665
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xf
***************
*** 1149,1169 ****
--- 1354,1380 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 666
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 669
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 670
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 672
      return-void
  
+     .line 669
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 670
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1180,1213 ****
--- 1391,1432 ----
      .end annotation
  
      .prologue
+     .line 414
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 415
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 417
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 418
      if-eqz p1, :cond_0
  
+     .line 419
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 420
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/hardware/usb/UsbDevice;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 425
      :goto_0
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 426
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x5
***************
*** 1216,1231 ****
--- 1435,1455 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 427
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 430
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 431
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 433
      return-void
  
+     .line 423
      :cond_0
      const/4 v2, 0x0
  
***************
*** 1236,1246 ****
--- 1460,1472 ----
  
      goto :goto_0
  
+     .line 430
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 431
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1256,1278 ****
--- 1482,1509 ----
      .end annotation
  
      .prologue
+     .line 676
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 677
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 679
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.hardware.usb.IUsbManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 680
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 681
      iget-object v2, p0, Landroid/hardware/usb/IUsbManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x10
***************
*** 1281,1301 ****
--- 1512,1538 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 682
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 685
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 686
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 688
      return-void
  
+     .line 685
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 686
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
