*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/EffectColor.smali	2012-03-23 19:35:37.493270575 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/EffectColor.smali.original	2012-03-23 19:34:44.745308716 -0500
***************
*** 38,43 ****
--- 38,44 ----
  
      const/4 v7, 0x0
  
+     .line 76
      move-object v0, p0
  
      move-object v2, v1
***************
*** 48,53 ****
--- 49,55 ----
  
      invoke-direct/range {v0 .. v8}, Landroid/media/videoeditor/EffectColor;-><init>(Landroid/media/videoeditor/MediaItem;Ljava/lang/String;JJII)V
  
+     .line 77
      return-void
  .end method
  
***************
*** 61,70 ****
--- 63,75 ----
      .parameter "color"
  
      .prologue
+     .line 95
      invoke-direct/range {p0 .. p6}, Landroid/media/videoeditor/Effect;-><init>(Landroid/media/videoeditor/MediaItem;Ljava/lang/String;JJ)V
  
+     .line 96
      packed-switch p7, :pswitch_data_0
  
+     .line 119
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 89,97 ****
--- 94,104 ----
  
      throw v0
  
+     .line 99
      :pswitch_0
      sparse-switch p8, :sswitch_data_0
  
+     .line 107
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 116,129 ****
--- 123,140 ----
  
      throw v0
  
+     .line 103
      :sswitch_0
      iput p8, p0, Landroid/media/videoeditor/EffectColor;->mColor:I
  
+     .line 122
      :goto_0
      iput p7, p0, Landroid/media/videoeditor/EffectColor;->mType:I
  
+     .line 123
      return-void
  
+     .line 114
      :pswitch_1
      const/4 v0, -0x1
  
***************
*** 131,136 ****
--- 142,148 ----
  
      goto :goto_0
  
+     .line 96
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 140,145 ****
--- 152,158 ----
          :pswitch_1
      .end packed-switch
  
+     .line 99
      :sswitch_data_0
      .sparse-switch
          0xff00 -> :sswitch_0
***************
*** 154,159 ****
--- 167,173 ----
      .locals 1
  
      .prologue
+     .line 140
      iget v0, p0, Landroid/media/videoeditor/EffectColor;->mColor:I
  
      return v0
***************
*** 163,168 ****
--- 177,183 ----
      .locals 1
  
      .prologue
+     .line 131
      iget v0, p0, Landroid/media/videoeditor/EffectColor;->mType:I
  
      return v0
