*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$AudioFormat.smali	2012-03-23 19:35:36.875259305 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$AudioFormat.smali.original	2012-03-23 19:34:44.740308626 -0500
***************
*** 46,51 ****
--- 46,52 ----
      .parameter
  
      .prologue
+     .line 176
      iput-object p1, p0, Landroid/media/videoeditor/MediaArtistNativeHelper$AudioFormat;->this$0:Landroid/media/videoeditor/MediaArtistNativeHelper;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
