*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaItem$1.smali	2012-03-23 19:35:37.368268295 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaItem$1.smali.original	2012-03-23 19:34:44.744308698 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 594
      iput-object p1, p0, Landroid/media/videoeditor/MediaItem$1;->this$0:Landroid/media/videoeditor/MediaItem;
  
      iput-object p2, p0, Landroid/media/videoeditor/MediaItem$1;->val$bitmaps:[Landroid/graphics/Bitmap;
***************
*** 47,55 ****
--- 48,58 ----
      .parameter "index"
  
      .prologue
+     .line 596
      iget-object v0, p0, Landroid/media/videoeditor/MediaItem$1;->val$bitmaps:[Landroid/graphics/Bitmap;
  
      aput-object p1, v0, p2
  
+     .line 597
      return-void
  .end method
