*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$Version.smali	2012-03-23 19:35:36.018243677 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$Version.smali.original	2012-03-23 19:34:44.733308498 -0500
***************
*** 38,43 ****
--- 38,44 ----
      .parameter
  
      .prologue
+     .line 141
      iput-object p1, p0, Landroid/media/videoeditor/MediaArtistNativeHelper$Version;->this$0:Landroid/media/videoeditor/MediaArtistNativeHelper;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 53,72 ****
--- 54,78 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 163
      new-instance v0, Landroid/media/videoeditor/MediaArtistNativeHelper$Version;
  
      iget-object v1, p0, Landroid/media/videoeditor/MediaArtistNativeHelper$Version;->this$0:Landroid/media/videoeditor/MediaArtistNativeHelper;
  
      invoke-direct {v0, v1}, Landroid/media/videoeditor/MediaArtistNativeHelper$Version;-><init>(Landroid/media/videoeditor/MediaArtistNativeHelper;)V
  
+     .line 165
      .local v0, version:Landroid/media/videoeditor/MediaArtistNativeHelper$Version;
      iput v2, v0, Landroid/media/videoeditor/MediaArtistNativeHelper$Version;->major:I
  
+     .line 166
      iput v2, v0, Landroid/media/videoeditor/MediaArtistNativeHelper$Version;->minor:I
  
+     .line 167
      const/4 v1, 0x1
  
      iput v1, v0, Landroid/media/videoeditor/MediaArtistNativeHelper$Version;->revision:I
  
+     .line 169
      return-object v0
  .end method
