*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$BackgroundMusicSettings.smali	2012-03-23 19:35:37.347267912 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$BackgroundMusicSettings.smali.original	2012-03-23 19:34:44.744308698 -0500
***************
*** 41,46 ****
--- 41,47 ----
      .locals 0
  
      .prologue
+     .line 1139
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
