*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$PreviewClipProperties.smali	2012-03-23 19:35:36.600254291 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$PreviewClipProperties.smali.original	2012-03-23 19:34:44.737308570 -0500
***************
*** 23,28 ****
--- 23,29 ----
      .locals 0
  
      .prologue
+     .line 1488
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
