*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$EffectSettings.smali	2012-03-23 19:35:36.462251773 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/MediaArtistNativeHelper$EffectSettings.smali.original	2012-03-23 19:34:44.737308570 -0500
***************
*** 73,78 ****
--- 73,79 ----
      .locals 0
  
      .prologue
+     .line 1196
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
