*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/TransitionCrossfade.smali	2012-03-23 19:35:36.187246759 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/media/videoeditor/TransitionCrossfade.smali.original	2012-03-23 19:34:44.735308534 -0500
***************
*** 10,15 ****
--- 10,16 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 33
      const-wide/16 v4, 0x0
  
      const/4 v6, 0x0
***************
*** 22,27 ****
--- 23,29 ----
  
      invoke-direct/range {v0 .. v6}, Landroid/media/videoeditor/TransitionCrossfade;-><init>(Ljava/lang/String;Landroid/media/videoeditor/MediaItem;Landroid/media/videoeditor/MediaItem;JI)V
  
+     .line 34
      return-void
  .end method
  
***************
*** 34,41 ****
--- 36,45 ----
      .parameter "behavior"
  
      .prologue
+     .line 52
      invoke-direct/range {p0 .. p6}, Landroid/media/videoeditor/Transition;-><init>(Ljava/lang/String;Landroid/media/videoeditor/MediaItem;Landroid/media/videoeditor/MediaItem;JI)V
  
+     .line 53
      return-void
  .end method
  
***************
*** 45,51 ****
--- 49,57 ----
      .locals 0
  
      .prologue
+     .line 60
      invoke-super {p0}, Landroid/media/videoeditor/Transition;->generate()V
  
+     .line 61
      return-void
  .end method
