*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/method/AllCapsTransformationMethod.smali	2012-03-23 19:35:22.706000930 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/method/AllCapsTransformationMethod.smali.original	2012-03-23 19:34:44.530304796 -0500
***************
*** 22,29 ****
--- 22,31 ----
      .parameter "context"
  
      .prologue
+     .line 36
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 37
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v0
***************
*** 36,41 ****
--- 38,44 ----
  
      iput-object v0, p0, Landroid/text/method/AllCapsTransformationMethod;->mLocale:Ljava/util/Locale;
  
+     .line 38
      return-void
  .end method
  
***************
*** 47,56 ****
--- 50,61 ----
      .parameter "view"
  
      .prologue
+     .line 42
      iget-boolean v0, p0, Landroid/text/method/AllCapsTransformationMethod;->mEnabled:Z
  
      if-eqz v0, :cond_1
  
+     .line 43
      if-eqz p1, :cond_0
  
      invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
***************
*** 63,76 ****
--- 68,84 ----
  
      move-result-object v0
  
+     .line 46
      :goto_0
      return-object v0
  
+     .line 43
      :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 45
      :cond_1
      const-string v0, "AllCapsTransformationMethod"
  
***************
*** 80,85 ****
--- 88,94 ----
  
      move-object v0, p1
  
+     .line 46
      goto :goto_0
  .end method
  
***************
*** 92,97 ****
--- 101,107 ----
      .parameter "previouslyFocusedRect"
  
      .prologue
+     .line 52
      return-void
  .end method
  
***************
*** 100,106 ****
--- 110,118 ----
      .parameter "allowLengthChanges"
  
      .prologue
+     .line 56
      iput-boolean p1, p0, Landroid/text/method/AllCapsTransformationMethod;->mEnabled:Z
  
+     .line 57
      return-void
  .end method
