*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/method/ArrowKeyMovementMethod.smali	2012-03-23 19:35:22.064989240 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/method/ArrowKeyMovementMethod.smali.original	2012-03-23 19:34:44.520304613 -0500
***************
*** 17,22 ****
--- 17,23 ----
      .locals 1
  
      .prologue
+     .line 327
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
***************
*** 30,35 ****
--- 31,37 ----
      .locals 0
  
      .prologue
+     .line 32
      invoke-direct {p0}, Landroid/text/method/BaseMovementMethod;-><init>()V
  
      return-void
***************
*** 41,46 ****
--- 43,49 ----
      .parameter "layout"
  
      .prologue
+     .line 39
      invoke-static {p0}, Landroid/text/Selection;->getSelectionEnd(Ljava/lang/CharSequence;)I
  
      move-result v0
***************
*** 60,75 ****
--- 63,81 ----
      .locals 1
  
      .prologue
+     .line 320
      sget-object v0, Landroid/text/method/ArrowKeyMovementMethod;->sInstance:Landroid/text/method/ArrowKeyMovementMethod;
  
      if-nez v0, :cond_0
  
+     .line 321
      new-instance v0, Landroid/text/method/ArrowKeyMovementMethod;
  
      invoke-direct {v0}, Landroid/text/method/ArrowKeyMovementMethod;-><init>()V
  
      sput-object v0, Landroid/text/method/ArrowKeyMovementMethod;->sInstance:Landroid/text/method/ArrowKeyMovementMethod;
  
+     .line 324
      :cond_0
      sget-object v0, Landroid/text/method/ArrowKeyMovementMethod;->sInstance:Landroid/text/method/ArrowKeyMovementMethod;
  
***************
*** 81,90 ****
--- 87,98 ----
      .parameter "widget"
  
      .prologue
+     .line 46
      new-instance v0, Landroid/graphics/Rect;
  
      invoke-direct {v0}, Landroid/graphics/Rect;-><init>()V
  
+     .line 47
      .local v0, rect:Landroid/graphics/Rect;
      invoke-virtual {p0, v0}, Landroid/widget/TextView;->getGlobalVisibleRect(Landroid/graphics/Rect;)Z
  
***************
*** 112,117 ****
--- 120,126 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 34
      invoke-static {p0, v0}, Landroid/text/method/MetaKeyKeyListener;->getMetaState(Ljava/lang/CharSequence;I)I
  
      move-result v1
***************
*** 144,166 ****
--- 153,179 ----
      .parameter "buffer"
  
      .prologue
+     .line 168
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 169
      invoke-interface {p2}, Landroid/text/Spannable;->length()I
  
      move-result v0
  
      invoke-static {p2, v0}, Landroid/text/Selection;->extendSelection(Landroid/text/Spannable;I)V
  
+     .line 173
      :goto_0
      const/4 v0, 0x1
  
      return v0
  
+     .line 171
      :cond_0
      invoke-interface {p2}, Landroid/text/Spannable;->length()I
  
***************
*** 175,180 ****
--- 188,194 ----
      .locals 1
  
      .prologue
+     .line 299
      const/4 v0, 0x1
  
      return v0
***************
*** 186,195 ****
--- 200,211 ----
      .parameter "buffer"
  
      .prologue
+     .line 100
      invoke-virtual {p1}, Landroid/widget/TextView;->getLayout()Landroid/text/Layout;
  
      move-result-object v0
  
+     .line 101
      .local v0, layout:Landroid/text/Layout;
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
***************
*** 197,206 ****
--- 213,224 ----
  
      if-eqz v1, :cond_0
  
+     .line 102
      invoke-static {p2, v0}, Landroid/text/Selection;->extendDown(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
      move-result v1
  
+     .line 104
      :goto_0
      return v1
  
***************
*** 218,223 ****
--- 236,242 ----
      .parameter "buffer"
  
      .prologue
+     .line 221
      invoke-virtual {p0, p1, p2}, Landroid/text/method/ArrowKeyMovementMethod;->lineEnd(Landroid/widget/TextView;Landroid/text/Spannable;)Z
  
      move-result v0
***************
*** 234,241 ****
--- 253,262 ----
      .parameter "event"
  
      .prologue
+     .line 53
      packed-switch p3, :pswitch_data_0
  
+     .line 65
      :cond_0
      invoke-super/range {p0 .. p5}, Landroid/text/method/BaseMovementMethod;->handleMovementKey(Landroid/widget/TextView;Landroid/text/Spannable;IILandroid/view/KeyEvent;)Z
  
***************
*** 244,249 ****
--- 265,271 ----
      :goto_0
      return v0
  
+     .line 55
      :pswitch_0
      invoke-static {p4}, Landroid/view/KeyEvent;->metaStateHasNoModifiers(I)Z
  
***************
*** 251,256 ****
--- 273,279 ----
  
      if-eqz v0, :cond_0
  
+     .line 56
      invoke-virtual {p5}, Landroid/view/KeyEvent;->getAction()I
  
      move-result v0
***************
*** 271,282 ****
--- 294,307 ----
  
      if-eqz v0, :cond_0
  
+     .line 60
      invoke-virtual {p1}, Landroid/widget/TextView;->showContextMenu()Z
  
      move-result v0
  
      goto :goto_0
  
+     .line 53
      nop
  
      :pswitch_data_0
***************
*** 291,296 ****
--- 316,322 ----
      .parameter "buffer"
  
      .prologue
+     .line 216
      invoke-virtual {p0, p1, p2}, Landroid/text/method/ArrowKeyMovementMethod;->lineStart(Landroid/widget/TextView;Landroid/text/Spannable;)Z
  
      move-result v0
***************
*** 304,313 ****
--- 330,341 ----
      .parameter "text"
  
      .prologue
+     .line 304
      const/4 v0, 0x0
  
      invoke-static {p2, v0}, Landroid/text/Selection;->setSelection(Landroid/text/Spannable;I)V
  
+     .line 305
      return-void
  .end method
  
***************
*** 317,326 ****
--- 345,356 ----
      .parameter "buffer"
  
      .prologue
+     .line 70
      invoke-virtual {p1}, Landroid/widget/TextView;->getLayout()Landroid/text/Layout;
  
      move-result-object v0
  
+     .line 71
      .local v0, layout:Landroid/text/Layout;
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
***************
*** 328,337 ****
--- 358,369 ----
  
      if-eqz v1, :cond_0
  
+     .line 72
      invoke-static {p2, v0}, Landroid/text/Selection;->extendLeft(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
      move-result v1
  
+     .line 74
      :goto_0
      return v1
  
***************
*** 349,366 ****
--- 381,402 ----
      .parameter "buffer"
  
      .prologue
+     .line 199
      invoke-virtual {p1}, Landroid/widget/TextView;->getSelectionEnd()I
  
      move-result v0
  
+     .line 200
      .local v0, selectionEnd:I
      invoke-virtual {p1}, Landroid/widget/TextView;->getWordIterator()Landroid/text/method/WordIterator;
  
      move-result-object v1
  
+     .line 201
      .local v1, wordIterator:Landroid/text/method/WordIterator;
      invoke-virtual {v1, p2, v0, v0}, Landroid/text/method/WordIterator;->setCharSequence(Ljava/lang/CharSequence;II)V
  
+     .line 202
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
      move-result v2
***************
*** 378,387 ****
--- 414,425 ----
      .parameter "buffer"
  
      .prologue
+     .line 188
      invoke-virtual {p1}, Landroid/widget/TextView;->getLayout()Landroid/text/Layout;
  
      move-result-object v0
  
+     .line 189
      .local v0, layout:Landroid/text/Layout;
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
***************
*** 389,398 ****
--- 427,438 ----
  
      if-eqz v1, :cond_0
  
+     .line 190
      invoke-static {p2, v0}, Landroid/text/Selection;->extendToRightEdge(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
      move-result v1
  
+     .line 192
      :goto_0
      return v1
  
***************
*** 410,419 ****
--- 450,461 ----
      .parameter "buffer"
  
      .prologue
+     .line 178
      invoke-virtual {p1}, Landroid/widget/TextView;->getLayout()Landroid/text/Layout;
  
      move-result-object v0
  
+     .line 179
      .local v0, layout:Landroid/text/Layout;
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
***************
*** 421,430 ****
--- 463,474 ----
  
      if-eqz v1, :cond_0
  
+     .line 180
      invoke-static {p2, v0}, Landroid/text/Selection;->extendToLeftEdge(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
      move-result v1
  
+     .line 182
      :goto_0
      return v1
  
***************
*** 443,468 ****
--- 487,517 ----
      .parameter "dir"
  
      .prologue
+     .line 309
      and-int/lit16 v0, p3, 0x82
  
      if-eqz v0, :cond_1
  
+     .line 310
      invoke-virtual {p1}, Landroid/widget/TextView;->getLayout()Landroid/text/Layout;
  
      move-result-object v0
  
      if-nez v0, :cond_0
  
+     .line 312
      invoke-interface {p2}, Landroid/text/Spannable;->length()I
  
      move-result v0
  
      invoke-static {p2, v0}, Landroid/text/Selection;->setSelection(Landroid/text/Spannable;I)V
  
+     .line 317
      :cond_0
      :goto_0
      return-void
  
+     .line 315
      :cond_1
      invoke-interface {p2}, Landroid/text/Spannable;->length()I
  
***************
*** 482,513 ****
--- 531,570 ----
      .prologue
      const/4 v5, 0x1
  
+     .line 226
      const/4 v2, -0x1
  
+     .line 227
      .local v2, initialScrollX:I
      const/4 v3, -0x1
  
+     .line 228
      .local v3, initialScrollY:I
      invoke-virtual {p3}, Landroid/view/MotionEvent;->getAction()I
  
      move-result v0
  
+     .line 230
      .local v0, action:I
      if-ne v0, v5, :cond_0
  
+     .line 231
      invoke-static {p1, p2}, Landroid/text/method/Touch;->getInitialScrollX(Landroid/widget/TextView;Landroid/text/Spannable;)I
  
      move-result v2
  
+     .line 232
      invoke-static {p1, p2}, Landroid/text/method/Touch;->getInitialScrollY(Landroid/widget/TextView;Landroid/text/Spannable;)I
  
      move-result v3
  
+     .line 235
      :cond_0
      invoke-static {p1, p2, p3}, Landroid/text/method/Touch;->onTouchEvent(Landroid/widget/TextView;Landroid/text/Spannable;Landroid/view/MotionEvent;)Z
  
      move-result v1
  
+     .line 237
      .local v1, handled:Z
      invoke-virtual {p1}, Landroid/widget/TextView;->isFocused()Z
  
***************
*** 521,534 ****
--- 578,594 ----
  
      if-nez v6, :cond_1
  
+     .line 238
      if-nez v0, :cond_2
  
+     .line 239
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
      move-result v6
  
      if-eqz v6, :cond_1
  
+     .line 240
      invoke-virtual {p3}, Landroid/view/MotionEvent;->getX()F
  
      move-result v6
***************
*** 541,546 ****
--- 601,607 ----
  
      move-result v4
  
+     .line 242
      .local v4, offset:I
      sget-object v6, Landroid/text/method/ArrowKeyMovementMethod;->LAST_TAP_DOWN:Ljava/lang/Object;
  
***************
*** 548,571 ****
--- 609,636 ----
  
      invoke-interface {p2, v6, v4, v4, v7}, Landroid/text/Spannable;->setSpan(Ljava/lang/Object;III)V
  
+     .line 248
      invoke-virtual {p1}, Landroid/widget/TextView;->getParent()Landroid/view/ViewParent;
  
      move-result-object v6
  
      invoke-interface {v6, v5}, Landroid/view/ViewParent;->requestDisallowInterceptTouchEvent(Z)V
  
+     .line 294
      .end local v1           #handled:Z
      .end local v4           #offset:I
      :cond_1
      :goto_0
      return v1
  
+     .line 250
      .restart local v1       #handled:Z
      :cond_2
      const/4 v6, 0x2
  
      if-ne v0, v6, :cond_3
  
+     .line 251
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
      move-result v6
***************
*** 574,581 ****
--- 639,648 ----
  
      if-eqz v1, :cond_1
  
+     .line 258
      invoke-virtual {p1}, Landroid/widget/TextView;->cancelLongPress()V
  
+     .line 263
      invoke-virtual {p3}, Landroid/view/MotionEvent;->getX()F
  
      move-result v6
***************
*** 588,604 ****
--- 655,675 ----
  
      move-result v4
  
+     .line 265
      .restart local v4       #offset:I
      invoke-static {p2, v4}, Landroid/text/Selection;->extendSelection(Landroid/text/Spannable;I)V
  
      move v1, v5
  
+     .line 266
      goto :goto_0
  
+     .line 268
      .end local v4           #offset:I
      :cond_3
      if-ne v0, v5, :cond_1
  
+     .line 273
      if-ltz v3, :cond_4
  
      invoke-virtual {p1}, Landroid/widget/TextView;->getScrollY()I
***************
*** 616,628 ****
--- 687,702 ----
  
      if-eq v2, v6, :cond_6
  
+     .line 275
      :cond_5
      invoke-virtual {p1}, Landroid/widget/TextView;->moveCursorToVisibleOffset()Z
  
      move v1, v5
  
+     .line 276
      goto :goto_0
  
+     .line 279
      :cond_6
      invoke-virtual {p3}, Landroid/view/MotionEvent;->getX()F
  
***************
*** 636,641 ****
--- 710,716 ----
  
      move-result v4
  
+     .line 280
      .restart local v4       #offset:I
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
***************
*** 643,664 ****
--- 718,745 ----
  
      if-eqz v6, :cond_8
  
+     .line 281
      sget-object v6, Landroid/text/method/ArrowKeyMovementMethod;->LAST_TAP_DOWN:Ljava/lang/Object;
  
      invoke-interface {p2, v6}, Landroid/text/Spannable;->removeSpan(Ljava/lang/Object;)V
  
+     .line 282
      invoke-static {p2, v4}, Landroid/text/Selection;->extendSelection(Landroid/text/Spannable;I)V
  
+     .line 287
      :cond_7
      :goto_1
      invoke-static {p2}, Landroid/text/method/MetaKeyKeyListener;->adjustMetaAfterKeypress(Landroid/text/Spannable;)V
  
+     .line 288
      invoke-static {p2}, Landroid/text/method/MetaKeyKeyListener;->resetLockedMeta(Landroid/text/Spannable;)V
  
      move v1, v5
  
+     .line 290
      goto :goto_0
  
+     .line 283
      :cond_8
      invoke-virtual {p1}, Landroid/widget/TextView;->shouldIgnoreActionUpEvent()Z
  
***************
*** 666,671 ****
--- 747,753 ----
  
      if-nez v6, :cond_7
  
+     .line 284
      invoke-static {p2, v4}, Landroid/text/Selection;->setSelection(Landroid/text/Spannable;I)V
  
      goto :goto_1
***************
*** 677,691 ****
--- 759,776 ----
      .parameter "buffer"
  
      .prologue
+     .line 134
      invoke-virtual {p1}, Landroid/widget/TextView;->getLayout()Landroid/text/Layout;
  
      move-result-object v1
  
+     .line 135
      .local v1, layout:Landroid/text/Layout;
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
      move-result v3
  
+     .line 136
      .local v3, selecting:Z
      invoke-static {p2, v1}, Landroid/text/method/ArrowKeyMovementMethod;->getCurrentLineTop(Landroid/text/Spannable;Landroid/text/Layout;)I
  
***************
*** 697,716 ****
--- 782,806 ----
  
      add-int v4, v5, v6
  
+     .line 137
      .local v4, targetY:I
      const/4 v0, 0x0
  
+     .line 139
      .local v0, handled:Z
      :cond_0
      invoke-static {p2}, Landroid/text/Selection;->getSelectionEnd(Ljava/lang/CharSequence;)I
  
      move-result v2
  
+     .line 140
      .local v2, previousSelectionEnd:I
      if-eqz v3, :cond_1
  
+     .line 141
      invoke-static {p2, v1}, Landroid/text/Selection;->extendDown(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
+     .line 145
      :goto_0
      invoke-static {p2}, Landroid/text/Selection;->getSelectionEnd(Ljava/lang/CharSequence;)I
  
***************
*** 718,734 ****
--- 808,828 ----
  
      if-ne v5, v2, :cond_2
  
+     .line 153
      :goto_1
      return v0
  
+     .line 143
      :cond_1
      invoke-static {p2, v1}, Landroid/text/Selection;->moveDown(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
      goto :goto_0
  
+     .line 148
      :cond_2
      const/4 v0, 0x1
  
+     .line 149
      invoke-static {p2, v1}, Landroid/text/method/ArrowKeyMovementMethod;->getCurrentLineTop(Landroid/text/Spannable;Landroid/text/Layout;)I
  
      move-result v5
***************
*** 744,758 ****
--- 838,855 ----
      .parameter "buffer"
  
      .prologue
+     .line 110
      invoke-virtual {p1}, Landroid/widget/TextView;->getLayout()Landroid/text/Layout;
  
      move-result-object v1
  
+     .line 111
      .local v1, layout:Landroid/text/Layout;
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
      move-result v3
  
+     .line 112
      .local v3, selecting:Z
      invoke-static {p2, v1}, Landroid/text/method/ArrowKeyMovementMethod;->getCurrentLineTop(Landroid/text/Spannable;Landroid/text/Layout;)I
  
***************
*** 764,783 ****
--- 861,885 ----
  
      sub-int v4, v5, v6
  
+     .line 113
      .local v4, targetY:I
      const/4 v0, 0x0
  
+     .line 115
      .local v0, handled:Z
      :cond_0
      invoke-static {p2}, Landroid/text/Selection;->getSelectionEnd(Ljava/lang/CharSequence;)I
  
      move-result v2
  
+     .line 116
      .local v2, previousSelectionEnd:I
      if-eqz v3, :cond_1
  
+     .line 117
      invoke-static {p2, v1}, Landroid/text/Selection;->extendUp(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
+     .line 121
      :goto_0
      invoke-static {p2}, Landroid/text/Selection;->getSelectionEnd(Ljava/lang/CharSequence;)I
  
***************
*** 785,801 ****
--- 887,907 ----
  
      if-ne v5, v2, :cond_2
  
+     .line 129
      :goto_1
      return v0
  
+     .line 119
      :cond_1
      invoke-static {p2, v1}, Landroid/text/Selection;->moveUp(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
      goto :goto_0
  
+     .line 124
      :cond_2
      const/4 v0, 0x1
  
+     .line 125
      invoke-static {p2, v1}, Landroid/text/method/ArrowKeyMovementMethod;->getCurrentLineTop(Landroid/text/Spannable;Landroid/text/Layout;)I
  
      move-result v5
***************
*** 811,820 ****
--- 917,928 ----
      .parameter "buffer"
  
      .prologue
+     .line 80
      invoke-virtual {p1}, Landroid/widget/TextView;->getLayout()Landroid/text/Layout;
  
      move-result-object v0
  
+     .line 81
      .local v0, layout:Landroid/text/Layout;
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
***************
*** 822,831 ****
--- 930,941 ----
  
      if-eqz v1, :cond_0
  
+     .line 82
      invoke-static {p2, v0}, Landroid/text/Selection;->extendRight(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
      move-result v1
  
+     .line 84
      :goto_0
      return v1
  
***************
*** 843,860 ****
--- 953,974 ----
      .parameter "buffer"
  
      .prologue
+     .line 208
      invoke-virtual {p1}, Landroid/widget/TextView;->getSelectionEnd()I
  
      move-result v0
  
+     .line 209
      .local v0, selectionEnd:I
      invoke-virtual {p1}, Landroid/widget/TextView;->getWordIterator()Landroid/text/method/WordIterator;
  
      move-result-object v1
  
+     .line 210
      .local v1, wordIterator:Landroid/text/method/WordIterator;
      invoke-virtual {v1, p2, v0, v0}, Landroid/text/method/WordIterator;->setCharSequence(Ljava/lang/CharSequence;II)V
  
+     .line 211
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
      move-result v2
***************
*** 874,892 ****
--- 988,1010 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 158
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 159
      invoke-static {p2, v1}, Landroid/text/Selection;->extendSelection(Landroid/text/Spannable;I)V
  
+     .line 163
      :goto_0
      const/4 v0, 0x1
  
      return v0
  
+     .line 161
      :cond_0
      invoke-static {p2, v1}, Landroid/text/Selection;->setSelection(Landroid/text/Spannable;I)V
  
***************
*** 899,908 ****
--- 1017,1028 ----
      .parameter "buffer"
  
      .prologue
+     .line 90
      invoke-virtual {p1}, Landroid/widget/TextView;->getLayout()Landroid/text/Layout;
  
      move-result-object v0
  
+     .line 91
      .local v0, layout:Landroid/text/Layout;
      invoke-static {p2}, Landroid/text/method/ArrowKeyMovementMethod;->isSelecting(Landroid/text/Spannable;)Z
  
***************
*** 910,919 ****
--- 1030,1041 ----
  
      if-eqz v1, :cond_0
  
+     .line 92
      invoke-static {p2, v0}, Landroid/text/Selection;->extendUp(Landroid/text/Spannable;Landroid/text/Layout;)Z
  
      move-result v1
  
+     .line 94
      :goto_0
      return v1
  
