*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/method/PasswordTransformationMethod.smali	2012-03-23 19:35:22.184991429 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/method/PasswordTransformationMethod.smali.original	2012-03-23 19:34:44.522304651 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .locals 1
  
      .prologue
+     .line 266
      const/16 v0, 0x2022
  
      sput-char v0, Landroid/text/method/PasswordTransformationMethod;->DOT:C
***************
*** 39,46 ****
--- 40,49 ----
      .locals 0
  
      .prologue
+     .line 35
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 258
      return-void
  .end method
  
***************
*** 48,53 ****
--- 51,57 ----
      .locals 1
  
      .prologue
+     .line 35
      sget-char v0, Landroid/text/method/PasswordTransformationMethod;->DOT:C
  
      return v0
***************
*** 57,71 ****
--- 61,79 ----
      .locals 1
  
      .prologue
+     .line 64
      sget-object v0, Landroid/text/method/PasswordTransformationMethod;->sInstance:Landroid/text/method/PasswordTransformationMethod;
  
      if-eqz v0, :cond_0
  
+     .line 65
      sget-object v0, Landroid/text/method/PasswordTransformationMethod;->sInstance:Landroid/text/method/PasswordTransformationMethod;
  
+     .line 68
      :goto_0
      return-object v0
  
+     .line 67
      :cond_0
      new-instance v0, Landroid/text/method/PasswordTransformationMethod;
  
***************
*** 73,78 ****
--- 81,87 ----
  
      sput-object v0, Landroid/text/method/PasswordTransformationMethod;->sInstance:Landroid/text/method/PasswordTransformationMethod;
  
+     .line 68
      sget-object v0, Landroid/text/method/PasswordTransformationMethod;->sInstance:Landroid/text/method/PasswordTransformationMethod;
  
      goto :goto_0
***************
*** 83,88 ****
--- 92,98 ----
      .parameter "sp"
  
      .prologue
+     .line 133
      const/4 v2, 0x0
  
      invoke-interface {p0}, Landroid/text/Spannable;->length()I
***************
*** 97,102 ****
--- 107,113 ----
  
      check-cast v1, [Landroid/text/method/PasswordTransformationMethod$Visible;
  
+     .line 134
      .local v1, old:[Landroid/text/method/PasswordTransformationMethod$Visible;
      const/4 v0, 0x0
  
***************
*** 106,119 ****
--- 117,133 ----
  
      if-ge v0, v2, :cond_0
  
+     .line 135
      aget-object v2, v1, v0
  
      invoke-interface {p0, v2}, Landroid/text/Spannable;->removeSpan(Ljava/lang/Object;)V
  
+     .line 134
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 137
      :cond_0
      return-void
  .end method
***************
*** 125,130 ****
--- 139,145 ----
      .parameter "s"
  
      .prologue
+     .line 118
      return-void
  .end method
  
***************
*** 136,141 ****
--- 151,157 ----
      .parameter "after"
  
      .prologue
+     .line 74
      return-void
  .end method
  
***************
*** 147,160 ****
--- 163,179 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 39
      instance-of v3, p1, Landroid/text/Spannable;
  
      if-eqz v3, :cond_1
  
      move-object v1, p1
  
+     .line 40
      check-cast v1, Landroid/text/Spannable;
  
+     .line 48
      .local v1, sp:Landroid/text/Spannable;
      invoke-interface {v1}, Landroid/text/Spannable;->length()I
  
***************
*** 168,173 ****
--- 187,193 ----
  
      check-cast v2, [Landroid/text/method/PasswordTransformationMethod$ViewReference;
  
+     .line 50
      .local v2, vr:[Landroid/text/method/PasswordTransformationMethod$ViewReference;
      const/4 v0, 0x0
  
***************
*** 177,193 ****
--- 197,217 ----
  
      if-ge v0, v3, :cond_0
  
+     .line 51
      aget-object v3, v2, v0
  
      invoke-interface {v1, v3}, Landroid/text/Spannable;->removeSpan(Ljava/lang/Object;)V
  
+     .line 50
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 54
      :cond_0
      invoke-static {v1}, Landroid/text/method/PasswordTransformationMethod;->removeVisibleSpans(Landroid/text/Spannable;)V
  
+     .line 56
      new-instance v3, Landroid/text/method/PasswordTransformationMethod$ViewReference;
  
      invoke-direct {v3, p2}, Landroid/text/method/PasswordTransformationMethod$ViewReference;-><init>(Landroid/view/View;)V
***************
*** 196,201 ****
--- 220,226 ----
  
      invoke-interface {v1, v3, v5, v5, v4}, Landroid/text/Spannable;->setSpan(Ljava/lang/Object;III)V
  
+     .line 60
      .end local v0           #i:I
      .end local v1           #sp:Landroid/text/Spannable;
      .end local v2           #vr:[Landroid/text/method/PasswordTransformationMethod$ViewReference;
***************
*** 216,234 ****
--- 241,264 ----
      .parameter "previouslyFocusedRect"
  
      .prologue
+     .line 123
      if-nez p3, :cond_0
  
+     .line 124
      instance-of v1, p2, Landroid/text/Spannable;
  
      if-eqz v1, :cond_0
  
      move-object v0, p2
  
+     .line 125
      check-cast v0, Landroid/text/Spannable;
  
+     .line 127
      .local v0, sp:Landroid/text/Spannable;
      invoke-static {v0}, Landroid/text/method/PasswordTransformationMethod;->removeVisibleSpans(Landroid/text/Spannable;)V
  
+     .line 130
      .end local v0           #sp:Landroid/text/Spannable;
      :cond_0
      return-void
***************
*** 242,255 ****
--- 272,288 ----
      .parameter "count"
  
      .prologue
+     .line 78
      instance-of v5, p1, Landroid/text/Spannable;
  
      if-eqz v5, :cond_0
  
      move-object v2, p1
  
+     .line 79
      check-cast v2, Landroid/text/Spannable;
  
+     .line 80
      .local v2, sp:Landroid/text/Spannable;
      const/4 v5, 0x0
  
***************
*** 265,286 ****
--- 298,323 ----
  
      check-cast v4, [Landroid/text/method/PasswordTransformationMethod$ViewReference;
  
+     .line 82
      .local v4, vr:[Landroid/text/method/PasswordTransformationMethod$ViewReference;
      array-length v5, v4
  
      if-nez v5, :cond_1
  
+     .line 114
      .end local v2           #sp:Landroid/text/Spannable;
      .end local v4           #vr:[Landroid/text/method/PasswordTransformationMethod$ViewReference;
      :cond_0
      :goto_0
      return-void
  
+     .line 93
      .restart local v2       #sp:Landroid/text/Spannable;
      .restart local v4       #vr:[Landroid/text/method/PasswordTransformationMethod$ViewReference;
      :cond_1
      const/4 v3, 0x0
  
+     .line 94
      .local v3, v:Landroid/view/View;
      const/4 v0, 0x0
  
***************
*** 292,297 ****
--- 329,335 ----
  
      if-ge v0, v5, :cond_2
  
+     .line 95
      aget-object v5, v4, v0
  
      invoke-virtual {v5}, Landroid/text/method/PasswordTransformationMethod$ViewReference;->get()Ljava/lang/Object;
***************
*** 301,314 ****
--- 339,355 ----
      .end local v3           #v:Landroid/view/View;
      check-cast v3, Landroid/view/View;
  
+     .line 94
      .restart local v3       #v:Landroid/view/View;
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 98
      :cond_2
      if-eqz v3, :cond_0
  
+     .line 102
      invoke-static {}, Landroid/text/method/TextKeyListener;->getInstance()Landroid/text/method/TextKeyListener;
  
      move-result-object v5
***************
*** 321,339 ****
--- 362,385 ----
  
      move-result v1
  
+     .line 103
      .local v1, pref:I
      and-int/lit8 v5, v1, 0x8
  
      if-eqz v5, :cond_0
  
+     .line 104
      if-lez p4, :cond_0
  
+     .line 105
      invoke-static {v2}, Landroid/text/method/PasswordTransformationMethod;->removeVisibleSpans(Landroid/text/Spannable;)V
  
+     .line 107
      const/4 v5, 0x1
  
      if-ne p4, v5, :cond_0
  
+     .line 108
      new-instance v5, Landroid/text/method/PasswordTransformationMethod$Visible;
  
      invoke-direct {v5, v2, p0}, Landroid/text/method/PasswordTransformationMethod$Visible;-><init>(Landroid/text/Spannable;Landroid/text/method/PasswordTransformationMethod;)V
