*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/util/AbstractMessageParser$YouTubeVideo.smali	2012-03-23 19:37:56.868812083 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/util/AbstractMessageParser$YouTubeVideo.smali.original	2012-03-23 19:34:46.425339353 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .locals 1
  
      .prologue
+     .line 871
      const-string v0, "(?i)http://(?:[a-z0-9]+\\.)?youtube\\.[a-z0-9]+(?:\\.[a-z0-9]+)?/watch\\?.*\\bv=([-_a-zA-Z0-9=]+).*"
  
      invoke-static {v0}, Ljava/util/regex/Pattern;->compile(Ljava/lang/String;)Ljava/util/regex/Pattern;
***************
*** 44,55 ****
--- 45,59 ----
      .parameter "text"
  
      .prologue
+     .line 878
      sget-object v0, Lcom/google/android/util/AbstractMessageParser$Token$Type;->YOUTUBE_VIDEO:Lcom/google/android/util/AbstractMessageParser$Token$Type;
  
      invoke-direct {p0, v0, p2}, Lcom/google/android/util/AbstractMessageParser$Token;-><init>(Lcom/google/android/util/AbstractMessageParser$Token$Type;Ljava/lang/String;)V
  
+     .line 879
      iput-object p1, p0, Lcom/google/android/util/AbstractMessageParser$YouTubeVideo;->docid:Ljava/lang/String;
  
+     .line 880
      return-void
  .end method
  
***************
*** 61,83 ****
--- 65,95 ----
      .parameter "extraParams"
  
      .prologue
+     .line 931
      const-string v0, ""
  
+     .line 933
      .local v0, protocol:Ljava/lang/String;
      if-eqz p0, :cond_0
  
+     .line 934
      const-string v0, "http://"
  
+     .line 937
      :cond_0
      if-nez p1, :cond_1
  
+     .line 938
      const-string p1, ""
  
+     .line 941
      :cond_1
      if-nez p3, :cond_3
  
+     .line 942
      const-string p3, ""
  
+     .line 947
      :cond_2
      :goto_0
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 92,98 ****
  
      move-result-object v1
  
!     const-string v2, "youtube.com/watch?"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 104,110 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "youtube.com/watch?"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 102,108 ****
  
      move-result-object v1
  
!     const-string v2, "v="
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 114,120 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "v="
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 118,123 ****
--- 130,136 ----
  
      return-object v1
  
+     .line 943
      :cond_3
      invoke-virtual {p3}, Ljava/lang/String;->length()I
  
***************
*** 125,130 ****
--- 138,144 ----
  
      if-lez v1, :cond_2
  
+     .line 944
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 151,156 ****
--- 165,171 ----
      .parameter "docid"
  
      .prologue
+     .line 906
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 177,182 ****
--- 192,198 ----
      .parameter "docid"
  
      .prologue
+     .line 911
      const/4 v0, 0x0
  
      invoke-static {p0, v0}, Lcom/google/android/util/AbstractMessageParser$YouTubeVideo;->getURL(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
***************
*** 192,201 ****
--- 208,220 ----
      .parameter "extraParams"
  
      .prologue
+     .line 916
      if-nez p1, :cond_1
  
+     .line 917
      const-string p1, ""
  
+     .line 921
      :cond_0
      :goto_0
      new-instance v0, Ljava/lang/StringBuilder;
***************
*** 212,218 ****
  
      move-result-object v0
  
!     const-string v1, "v="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 231,237 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "v="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 228,233 ****
--- 247,253 ----
  
      return-object v0
  
+     .line 918
      :cond_1
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
***************
*** 235,240 ****
--- 255,261 ----
  
      if-lez v0, :cond_0
  
+     .line 919
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 262,273 ****
--- 283,296 ----
      .parameter "text"
  
      .prologue
+     .line 889
      sget-object v1, Lcom/google/android/util/AbstractMessageParser$YouTubeVideo;->URL_PATTERN:Ljava/util/regex/Pattern;
  
      invoke-virtual {v1, p0}, Ljava/util/regex/Pattern;->matcher(Ljava/lang/CharSequence;)Ljava/util/regex/Matcher;
  
      move-result-object v0
  
+     .line 890
      .local v0, m:Ljava/util/regex/Matcher;
      invoke-virtual {v0}, Ljava/util/regex/Matcher;->matches()Z
  
***************
*** 275,280 ****
--- 298,304 ----
  
      if-eqz v1, :cond_0
  
+     .line 891
      new-instance v1, Lcom/google/android/util/AbstractMessageParser$YouTubeVideo;
  
      const/4 v2, 0x1
***************
*** 285,290 ****
--- 309,315 ----
  
      invoke-direct {v1, v2, p1}, Lcom/google/android/util/AbstractMessageParser$YouTubeVideo;-><init>(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 893
      :goto_0
      return-object v1
  
***************
*** 300,305 ****
--- 325,331 ----
      .locals 1
  
      .prologue
+     .line 882
      iget-object v0, p0, Lcom/google/android/util/AbstractMessageParser$YouTubeVideo;->docid:Ljava/lang/String;
  
      return-object v0
***************
*** 318,327 ****
--- 344,355 ----
      .end annotation
  
      .prologue
+     .line 898
      invoke-super {p0}, Lcom/google/android/util/AbstractMessageParser$Token;->getInfo()Ljava/util/List;
  
      move-result-object v0
  
+     .line 899
      .local v0, info:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
      iget-object v1, p0, Lcom/google/android/util/AbstractMessageParser$YouTubeVideo;->docid:Ljava/lang/String;
  
***************
*** 331,336 ****
--- 359,365 ----
  
      invoke-interface {v0, v1}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 900
      iget-object v1, p0, Lcom/google/android/util/AbstractMessageParser$YouTubeVideo;->docid:Ljava/lang/String;
  
      invoke-static {v1}, Lcom/google/android/util/AbstractMessageParser$YouTubeVideo;->getURL(Ljava/lang/String;)Ljava/lang/String;
***************
*** 339,344 ****
--- 368,374 ----
  
      invoke-interface {v0, v1}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 901
      return-object v0
  .end method
  
***************
*** 346,351 ****
--- 376,382 ----
      .locals 1
  
      .prologue
+     .line 884
      const/4 v0, 0x0
  
      return v0
***************
*** 355,360 ****
--- 386,392 ----
      .locals 1
  
      .prologue
+     .line 885
      const/4 v0, 0x1
  
      return v0
