*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/util/AbstractMessageParser$Format.smali	2012-03-23 19:37:57.049815384 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/util/AbstractMessageParser$Format.smali.original	2012-03-23 19:34:46.427339389 -0500
***************
*** 29,34 ****
--- 29,35 ----
      .parameter "start"
  
      .prologue
+     .line 1183
      sget-object v0, Lcom/google/android/util/AbstractMessageParser$Token$Type;->FORMAT:Lcom/google/android/util/AbstractMessageParser$Token$Type;
  
      invoke-static {p1}, Ljava/lang/String;->valueOf(C)Ljava/lang/String;
***************
*** 37,46 ****
--- 38,50 ----
  
      invoke-direct {p0, v0, v1}, Lcom/google/android/util/AbstractMessageParser$Token;-><init>(Lcom/google/android/util/AbstractMessageParser$Token$Type;Ljava/lang/String;)V
  
+     .line 1184
      iput-char p1, p0, Lcom/google/android/util/AbstractMessageParser$Format;->ch:C
  
+     .line 1185
      iput-boolean p2, p0, Lcom/google/android/util/AbstractMessageParser$Format;->start:Z
  
+     .line 1186
      return-void
  .end method
  
***************
*** 49,63 ****
      .parameter "ch"
  
      .prologue
      sparse-switch p1, :sswitch_data_0
  
      new-instance v0, Ljava/lang/AssertionError;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "unknown format \'"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 53,69 ----
      .parameter "ch"
  
      .prologue
+     .line 1224
      sparse-switch p1, :sswitch_data_0
  
+     .line 1229
      new-instance v0, Ljava/lang/AssertionError;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "unknown format \'"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 81,107 ****
  
      throw v0
  
      :sswitch_0
      const-string v0, "</b>"
  
      :goto_0
      return-object v0
  
      :sswitch_1
      const-string v0, "</i>"
  
      goto :goto_0
  
      :sswitch_2
      const-string v0, "</font></b>"
  
      goto :goto_0
  
      :sswitch_3
!     const-string v0, "\u201d</font>"
  
      goto :goto_0
  
      :sswitch_data_0
      .sparse-switch
          0x22 -> :sswitch_3
--- 87,119 ----
  
      throw v0
  
+     .line 1225
      :sswitch_0
      const-string v0, "</b>"
  
+     .line 1228
      :goto_0
      return-object v0
  
+     .line 1226
      :sswitch_1
      const-string v0, "</i>"
  
      goto :goto_0
  
+     .line 1227
      :sswitch_2
      const-string v0, "</font></b>"
  
      goto :goto_0
  
+     .line 1228
      :sswitch_3
!     const-string/jumbo v0, "\u201d</font>"
  
      goto :goto_0
  
+     .line 1224
      :sswitch_data_0
      .sparse-switch
          0x22 -> :sswitch_3
***************
*** 116,130 ****
      .parameter "ch"
  
      .prologue
      sparse-switch p1, :sswitch_data_0
  
      new-instance v0, Ljava/lang/AssertionError;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "unknown format \'"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 128,144 ----
      .parameter "ch"
  
      .prologue
+     .line 1214
      sparse-switch p1, :sswitch_data_0
  
+     .line 1219
      new-instance v0, Ljava/lang/AssertionError;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "unknown format \'"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 148,174 ****
--- 162,194 ----
  
      throw v0
  
+     .line 1215
      :sswitch_0
      const-string v0, "<b>"
  
+     .line 1218
      :goto_0
      return-object v0
  
+     .line 1216
      :sswitch_1
      const-string v0, "<i>"
  
      goto :goto_0
  
+     .line 1217
      :sswitch_2
      const-string v0, "<b><font color=\"#005FFF\">"
  
      goto :goto_0
  
+     .line 1218
      :sswitch_3
      const-string v0, "<font color=\"#999999\">\u201c"
  
      goto :goto_0
  
+     .line 1214
      nop
  
      :sswitch_data_0
***************
*** 186,191 ****
--- 206,212 ----
      .locals 2
  
      .prologue
+     .line 1210
      iget-char v0, p0, Lcom/google/android/util/AbstractMessageParser$Format;->ch:C
  
      const/16 v1, 0x5e
***************
*** 216,221 ****
--- 237,243 ----
      .end annotation
  
      .prologue
+     .line 1207
      new-instance v0, Ljava/lang/UnsupportedOperationException;
  
      invoke-direct {v0}, Ljava/lang/UnsupportedOperationException;-><init>()V
***************
*** 227,232 ****
--- 249,255 ----
      .locals 1
  
      .prologue
+     .line 1190
      const/4 v0, 0x1
  
      return v0
***************
*** 236,241 ****
--- 259,265 ----
      .locals 1
  
      .prologue
+     .line 1211
      iget-boolean v0, p0, Lcom/google/android/util/AbstractMessageParser$Format;->start:Z
  
      return v0
***************
*** 246,251 ****
--- 270,276 ----
      .parameter "matched"
  
      .prologue
+     .line 1188
      iput-boolean p1, p0, Lcom/google/android/util/AbstractMessageParser$Format;->matched:Z
  
      return-void
***************
*** 256,265 ****
--- 281,292 ----
      .parameter "caps"
  
      .prologue
+     .line 1195
      iget-boolean v0, p0, Lcom/google/android/util/AbstractMessageParser$Format;->matched:Z
  
      if-eqz v0, :cond_1
  
+     .line 1196
      iget-boolean v0, p0, Lcom/google/android/util/AbstractMessageParser$Format;->start:Z
  
      if-eqz v0, :cond_0
***************
*** 270,278 ****
--- 297,307 ----
  
      move-result-object v0
  
+     .line 1199
      :goto_0
      return-object v0
  
+     .line 1196
      :cond_0
      iget-char v0, p0, Lcom/google/android/util/AbstractMessageParser$Format;->ch:C
  
***************
*** 282,287 ****
--- 311,317 ----
  
      goto :goto_0
  
+     .line 1199
      :cond_1
      iget-char v0, p0, Lcom/google/android/util/AbstractMessageParser$Format;->ch:C
  
