*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/util/AbstractMessageParser$Token.smali	2012-03-23 19:37:57.234818757 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/util/AbstractMessageParser$Token.smali.original	2012-03-23 19:34:46.429339425 -0500
***************
*** 33,44 ****
--- 33,48 ----
      .parameter "text"
  
      .prologue
+     .line 686
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 687
      iput-object p1, p0, Lcom/google/android/util/AbstractMessageParser$Token;->type:Lcom/google/android/util/AbstractMessageParser$Token$Type;
  
+     .line 688
      iput-object p2, p0, Lcom/google/android/util/AbstractMessageParser$Token;->text:Ljava/lang/String;
  
+     .line 689
      return-void
  .end method
  
***************
*** 48,53 ****
--- 52,58 ----
      .locals 1
  
      .prologue
+     .line 716
      const/4 v0, 0x0
  
      return v0
***************
*** 66,75 ****
--- 71,82 ----
      .end annotation
  
      .prologue
+     .line 701
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
+     .line 702
      .local v0, info:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
      invoke-virtual {p0}, Lcom/google/android/util/AbstractMessageParser$Token;->getType()Lcom/google/android/util/AbstractMessageParser$Token$Type;
  
***************
*** 81,86 ****
--- 88,94 ----
  
      invoke-interface {v0, v1}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 703
      return-object v0
  .end method
  
***************
*** 88,93 ****
--- 96,102 ----
      .locals 1
  
      .prologue
+     .line 707
      iget-object v0, p0, Lcom/google/android/util/AbstractMessageParser$Token;->text:Ljava/lang/String;
  
      return-object v0
***************
*** 97,102 ****
--- 106,112 ----
      .locals 1
  
      .prologue
+     .line 692
      iget-object v0, p0, Lcom/google/android/util/AbstractMessageParser$Token;->type:Lcom/google/android/util/AbstractMessageParser$Token$Type;
  
      return-object v0
***************
*** 106,111 ****
--- 116,122 ----
      .locals 1
  
      .prologue
+     .line 711
      invoke-virtual {p0}, Lcom/google/android/util/AbstractMessageParser$Token;->isHtml()Z
  
      move-result v0
***************
*** 130,135 ****
--- 141,147 ----
      .locals 1
  
      .prologue
+     .line 709
      const/4 v0, 0x0
  
      return v0
***************
*** 139,144 ****
--- 151,157 ----
      .locals 1
  
      .prologue
+     .line 717
      const/4 v0, 0x0
  
      return v0
***************
*** 149,157 ****
      .parameter "caps"
  
      .prologue
      new-instance v0, Ljava/lang/AssertionError;
  
!     const-string v1, "not html"
  
      invoke-direct {v0, v1}, Ljava/lang/AssertionError;-><init>(Ljava/lang/Object;)V
  
--- 162,171 ----
      .parameter "caps"
  
      .prologue
+     .line 713
      new-instance v0, Ljava/lang/AssertionError;
  
!     const-string/jumbo v1, "not html"
  
      invoke-direct {v0, v1}, Ljava/lang/AssertionError;-><init>(Ljava/lang/Object;)V
  
