*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/util/SmileyResources.smali	2012-03-23 19:37:56.803810899 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/util/SmileyResources.smali.original	2012-03-23 19:34:46.425339353 -0500
***************
*** 29,48 ****
--- 29,52 ----
      .parameter "smileyResIds"
  
      .prologue
+     .line 35
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 28
      new-instance v1, Ljava/util/HashMap;
  
      invoke-direct {v1}, Ljava/util/HashMap;-><init>()V
  
      iput-object v1, p0, Lcom/google/android/util/SmileyResources;->mSmileyToRes:Ljava/util/HashMap;
  
+     .line 56
      new-instance v1, Lcom/google/android/util/AbstractMessageParser$TrieNode;
  
      invoke-direct {v1}, Lcom/google/android/util/AbstractMessageParser$TrieNode;-><init>()V
  
      iput-object v1, p0, Lcom/google/android/util/SmileyResources;->smileys:Lcom/google/android/util/AbstractMessageParser$TrieNode;
  
+     .line 36
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 51,56 ****
--- 55,61 ----
  
      if-ge v0, v1, :cond_0
  
+     .line 37
      iget-object v1, p0, Lcom/google/android/util/SmileyResources;->smileys:Lcom/google/android/util/AbstractMessageParser$TrieNode;
  
      aget-object v2, p1, v0
***************
*** 59,64 ****
--- 64,70 ----
  
      invoke-static {v1, v2, v3}, Lcom/google/android/util/AbstractMessageParser$TrieNode;->addToTrie(Lcom/google/android/util/AbstractMessageParser$TrieNode;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 38
      iget-object v1, p0, Lcom/google/android/util/SmileyResources;->mSmileyToRes:Ljava/util/HashMap;
  
      aget-object v2, p1, v0
***************
*** 71,80 ****
--- 77,88 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 36
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 40
      :cond_0
      return-void
  .end method
***************
*** 85,90 ****
--- 93,99 ----
      .locals 1
  
      .prologue
+     .line 71
      const/4 v0, 0x0
  
      return-object v0
***************
*** 94,99 ****
--- 103,109 ----
      .locals 1
  
      .prologue
+     .line 63
      const/4 v0, 0x0
  
      return-object v0
***************
*** 112,117 ****
--- 122,128 ----
      .end annotation
  
      .prologue
+     .line 59
      const/4 v0, 0x0
  
      return-object v0
***************
*** 122,127 ****
--- 133,139 ----
      .parameter "smiley"
  
      .prologue
+     .line 49
      iget-object v1, p0, Lcom/google/android/util/SmileyResources;->mSmileyToRes:Ljava/util/HashMap;
  
      invoke-virtual {v1, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 130,140 ****
--- 142,155 ----
  
      check-cast v0, Ljava/lang/Integer;
  
+     .line 50
      .local v0, i:Ljava/lang/Integer;
      if-nez v0, :cond_0
  
+     .line 51
      const/4 v1, -0x1
  
+     .line 53
      :goto_0
      return v1
  
***************
*** 150,155 ****
--- 165,171 ----
      .locals 1
  
      .prologue
+     .line 67
      iget-object v0, p0, Lcom/google/android/util/SmileyResources;->smileys:Lcom/google/android/util/AbstractMessageParser$TrieNode;
  
      return-object v0
