*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/BaseIWindow.smali	2012-03-23 19:37:53.071742844 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/BaseIWindow.smali.original	2012-03-23 19:34:46.392338751 -0500
***************
*** 14,19 ****
--- 14,20 ----
      .locals 0
  
      .prologue
+     .line 28
      invoke-direct {p0}, Landroid/view/IWindow$Stub;-><init>()V
  
      return-void
***************
*** 26,31 ****
--- 27,33 ----
      .parameter "reason"
  
      .prologue
+     .line 59
      return-void
  .end method
  
***************
*** 34,39 ****
--- 36,42 ----
      .parameter "visible"
  
      .prologue
+     .line 47
      return-void
  .end method
  
***************
*** 42,47 ****
--- 45,51 ----
      .parameter "event"
  
      .prologue
+     .line 71
      return-void
  .end method
  
***************
*** 49,54 ****
--- 53,59 ----
      .locals 0
  
      .prologue
+     .line 50
      return-void
  .end method
  
***************
*** 60,67 ****
--- 65,74 ----
      .parameter "localChanges"
  
      .prologue
+     .line 75
      iput p1, p0, Lcom/android/internal/view/BaseIWindow;->mSeq:I
  
+     .line 76
      return-void
  .end method
  
***************
*** 75,82 ****
--- 82,91 ----
      .parameter "sync"
  
      .prologue
+     .line 80
      if-eqz p6, :cond_0
  
+     .line 82
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/view/BaseIWindow;->mSession:Landroid/view/IWindowSession;
  
***************
*** 90,99 ****
--- 99,110 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 86
      :cond_0
      :goto_0
      return-void
  
+     .line 83
      :catch_0
      move-exception v0
  
***************
*** 109,116 ****
--- 120,129 ----
      .parameter "sync"
  
      .prologue
+     .line 62
      if-eqz p5, :cond_0
  
+     .line 64
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/view/BaseIWindow;->mSession:Landroid/view/IWindowSession;
  
***************
*** 122,131 ****
--- 135,146 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 68
      :cond_0
      :goto_0
      return-void
  
+     .line 65
      :catch_0
      move-exception v0
  
***************
*** 139,144 ****
--- 154,160 ----
      .parameter "out"
  
      .prologue
+     .line 56
      return-void
  .end method
  
***************
*** 152,159 ****
--- 168,177 ----
      .parameter "newConfig"
  
      .prologue
+     .line 38
      if-eqz p5, :cond_0
  
+     .line 40
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/view/BaseIWindow;->mSession:Landroid/view/IWindowSession;
  
***************
*** 161,170 ****
--- 179,190 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 44
      :cond_0
      :goto_0
      return-void
  
+     .line 41
      :catch_0
      move-exception v0
  
***************
*** 176,183 ****
--- 196,205 ----
      .parameter "session"
  
      .prologue
+     .line 33
      iput-object p1, p0, Lcom/android/internal/view/BaseIWindow;->mSession:Landroid/view/IWindowSession;
  
+     .line 34
      return-void
  .end method
  
***************
*** 187,191 ****
--- 209,214 ----
      .parameter "touchEnabled"
  
      .prologue
+     .line 53
      return-void
  .end method
