*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/IInputMethodManager$Stub$Proxy.smali	2012-03-23 19:37:53.257746235 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/IInputMethodManager$Stub$Proxy.smali.original	2012-03-23 19:34:46.393338769 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 418
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 419
      iput-object p1, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 420
      return-void
  .end method
  
***************
*** 51,71 ****
--- 54,78 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 533
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 534
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 536
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 537
      if-eqz p1, :cond_1
  
      invoke-interface {p1}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
***************
*** 75,80 ****
--- 82,88 ----
      :goto_0
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 538
      if-eqz p2, :cond_0
  
      invoke-interface {p2}, Lcom/android/internal/view/IInputContext;->asBinder()Landroid/os/IBinder;
***************
*** 84,93 ****
--- 92,104 ----
      :cond_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 539
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 540
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 541
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x6
***************
*** 96,121 ****
--- 107,139 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 542
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 545
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 546
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 548
      return-void
  
      :cond_1
      move-object v3, v2
  
+     .line 537
      goto :goto_0
  
+     .line 545
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 546
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 125,130 ****
--- 143,149 ----
      .locals 1
  
      .prologue
+     .line 423
      iget-object v0, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 140,160 ****
--- 159,183 ----
      .end annotation
  
      .prologue
+     .line 599
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 600
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 602
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 603
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
***************
*** 164,169 ****
--- 187,193 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 604
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x9
***************
*** 172,197 ****
--- 196,228 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 605
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 608
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 609
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 611
      return-void
  
+     .line 603
      :cond_0
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 608
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 609
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 206,226 ****
--- 237,261 ----
      .end annotation
  
      .prologue
+     .line 864
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 865
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 868
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 869
      iget-object v3, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x17
***************
*** 229,242 ****
--- 264,280 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 870
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 871
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 872
      sget-object v3, Landroid/view/inputmethod/InputMethodSubtype;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 247,260 ****
--- 285,302 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 879
      .local v2, _result:Landroid/view/inputmethod/InputMethodSubtype;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 880
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 882
      return-object v2
  
+     .line 875
      .end local v2           #_result:Landroid/view/inputmethod/InputMethodSubtype;
      :cond_0
      const/4 v2, 0x0
***************
*** 262,273 ****
--- 304,317 ----
      .restart local v2       #_result:Landroid/view/inputmethod/InputMethodSubtype;
      goto :goto_0
  
+     .line 879
      .end local v2           #_result:Landroid/view/inputmethod/InputMethodSubtype;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 880
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 292,312 ****
--- 336,360 ----
      .end annotation
  
      .prologue
+     .line 448
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 449
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 452
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 453
      iget-object v3, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x2
***************
*** 315,322 ****
--- 363,372 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 454
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 455
      sget-object v3, Landroid/view/inputmethod/InputMethodInfo;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-virtual {v1, v3}, Landroid/os/Parcel;->createTypedArrayList(Landroid/os/Parcelable$Creator;)Ljava/util/ArrayList;
***************
*** 325,343 ****
--- 375,398 ----
  
      move-result-object v2
  
+     .line 458
      .local v2, _result:Ljava/util/List;,"Ljava/util/List<Landroid/view/inputmethod/InputMethodInfo;>;"
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 459
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 461
      return-object v2
  
+     .line 458
      .end local v2           #_result:Ljava/util/List;,"Ljava/util/List<Landroid/view/inputmethod/InputMethodInfo;>;"
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 459
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 370,406 ****
--- 425,469 ----
  
      const/4 v4, 0x0
  
+     .line 465
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 466
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 469
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v5, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v5}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 470
      if-eqz p1, :cond_0
  
+     .line 471
      const/4 v5, 0x1
  
      invoke-virtual {v0, v5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 472
      const/4 v5, 0x0
  
      invoke-virtual {p1, v0, v5}, Landroid/view/inputmethod/InputMethodInfo;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 477
      :goto_0
      if-eqz p2, :cond_1
  
      :goto_1
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 478
      iget-object v3, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x3
***************
*** 409,416 ****
--- 472,481 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 479
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 480
      sget-object v3, Landroid/view/inputmethod/InputMethodSubtype;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-virtual {v1, v3}, Landroid/os/Parcel;->createTypedArrayList(Landroid/os/Parcelable$Creator;)Ljava/util/ArrayList;
***************
*** 419,431 ****
--- 484,500 ----
  
      move-result-object v2
  
+     .line 483
      .local v2, _result:Ljava/util/List;,"Ljava/util/List<Landroid/view/inputmethod/InputMethodSubtype;>;"
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 484
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 486
      return-object v2
  
+     .line 475
      .end local v2           #_result:Ljava/util/List;,"Ljava/util/List<Landroid/view/inputmethod/InputMethodSubtype;>;"
      :cond_0
      const/4 v5, 0x0
***************
*** 437,447 ****
--- 506,518 ----
  
      goto :goto_0
  
+     .line 483
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 484
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 449,454 ****
--- 520,526 ----
      :cond_1
      move v3, v4
  
+     .line 477
      goto :goto_1
  .end method
  
***************
*** 471,491 ****
--- 543,567 ----
      .end annotation
  
      .prologue
+     .line 431
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 432
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 435
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 436
      iget-object v3, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x1
***************
*** 494,501 ****
--- 570,579 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 437
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 438
      sget-object v3, Landroid/view/inputmethod/InputMethodInfo;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-virtual {v1, v3}, Landroid/os/Parcel;->createTypedArrayList(Landroid/os/Parcelable$Creator;)Ljava/util/ArrayList;
***************
*** 504,522 ****
--- 582,605 ----
  
      move-result-object v2
  
+     .line 441
      .local v2, _result:Ljava/util/List;,"Ljava/util/List<Landroid/view/inputmethod/InputMethodInfo;>;"
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 442
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 444
      return-object v2
  
+     .line 441
      .end local v2           #_result:Ljava/util/List;,"Ljava/util/List<Landroid/view/inputmethod/InputMethodInfo;>;"
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 442
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 526,531 ****
--- 609,615 ----
      .locals 1
  
      .prologue
+     .line 427
      const-string v0, "com.android.internal.view.IInputMethodManager"
  
      return-object v0
***************
*** 540,560 ****
--- 624,648 ----
      .end annotation
  
      .prologue
+     .line 490
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 491
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 494
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 495
      iget-object v3, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x4
***************
*** 563,576 ****
--- 651,667 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 496
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 497
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 498
      sget-object v3, Landroid/view/inputmethod/InputMethodSubtype;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 581,594 ****
--- 672,689 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 505
      .local v2, _result:Landroid/view/inputmethod/InputMethodSubtype;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 506
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 508
      return-object v2
  
+     .line 501
      .end local v2           #_result:Landroid/view/inputmethod/InputMethodSubtype;
      :cond_0
      const/4 v2, 0x0
***************
*** 596,607 ****
--- 691,704 ----
      .restart local v2       #_result:Landroid/view/inputmethod/InputMethodSubtype;
      goto :goto_0
  
+     .line 505
      .end local v2           #_result:Landroid/view/inputmethod/InputMethodSubtype;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 506
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 616,636 ****
--- 713,737 ----
      .end annotation
  
      .prologue
+     .line 515
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 516
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 519
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 520
      iget-object v4, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v5, 0x5
***************
*** 639,646 ****
--- 740,749 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 521
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 522
      invoke-virtual {p0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
  
      move-result-object v4
***************
*** 649,654 ****
--- 752,758 ----
  
      move-result-object v3
  
+     .line 523
      .local v3, cl:Ljava/lang/ClassLoader;
      invoke-virtual {v1, v3}, Landroid/os/Parcel;->readArrayList(Ljava/lang/ClassLoader;)Ljava/util/ArrayList;
      :try_end_0
***************
*** 656,668 ****
--- 760,776 ----
  
      move-result-object v2
  
+     .line 526
      .local v2, _result:Ljava/util/List;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 527
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 529
      return-object v2
  
+     .line 526
      .end local v2           #_result:Ljava/util/List;
      .end local v3           #cl:Ljava/lang/ClassLoader;
      :catchall_0
***************
*** 670,675 ****
--- 778,784 ----
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 527
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v4
***************
*** 686,710 ****
--- 795,825 ----
      .end annotation
  
      .prologue
+     .line 757
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 758
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 760
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 761
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 762
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 763
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x11
***************
*** 713,733 ****
--- 828,854 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 764
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 767
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 768
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 770
      return-void
  
+     .line 767
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 768
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 749,769 ****
--- 870,894 ----
  
      const/4 v3, 0x0
  
+     .line 640
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 641
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 644
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 645
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
***************
*** 773,790 ****
--- 898,920 ----
      :goto_0
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 646
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 647
      if-eqz p3, :cond_1
  
+     .line 648
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 649
      const/4 v4, 0x0
  
      invoke-virtual {p3, v0, v4}, Landroid/os/ResultReceiver;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 654
      :goto_1
      iget-object v4, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 794,801 ****
--- 924,933 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 655
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 656
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 804,823 ****
--- 936,960 ----
  
      if-eqz v4, :cond_2
  
+     .line 659
      .local v2, _result:Z
      :goto_2
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 660
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 662
      return v2
  
+     .line 645
      .end local v2           #_result:Z
      :cond_0
      const/4 v4, 0x0
  
      goto :goto_0
  
+     .line 652
      :cond_1
      const/4 v4, 0x0
  
***************
*** 828,838 ****
--- 965,977 ----
  
      goto :goto_1
  
+     .line 659
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 660
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 840,845 ****
--- 979,985 ----
      :cond_2
      move v2, v3
  
+     .line 656
      goto :goto_2
  .end method
  
***************
*** 859,894 ****
--- 999,1043 ----
  
      const/4 v3, 0x0
  
+     .line 838
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 839
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 842
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 843
      if-eqz p1, :cond_0
  
+     .line 844
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 845
      const/4 v4, 0x0
  
      invoke-virtual {p1, v0, v4}, Landroid/text/style/SuggestionSpan;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 850
      :goto_0
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 851
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 852
      iget-object v4, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v5, 0x16
***************
*** 897,904 ****
--- 1046,1055 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 853
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 854
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 907,920 ****
--- 1058,1075 ----
  
      if-eqz v4, :cond_1
  
+     .line 857
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 858
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 860
      return v2
  
+     .line 848
      .end local v2           #_result:Z
      :cond_0
      const/4 v4, 0x0
***************
*** 926,936 ****
--- 1081,1093 ----
  
      goto :goto_0
  
+     .line 857
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 858
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 938,943 ****
--- 1095,1101 ----
      :cond_1
      move v2, v3
  
+     .line 854
      goto :goto_1
  .end method
  
***************
*** 951,975 ****
--- 1109,1138 ----
      .end annotation
  
      .prologue
+     .line 823
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 824
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 826
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 827
      const/4 v2, 0x0
  
      invoke-virtual {v0, p1, v2}, Landroid/os/Parcel;->writeTypedArray([Landroid/os/Parcelable;I)V
  
+     .line 828
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x15
***************
*** 978,998 ****
--- 1141,1167 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 829
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 832
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 833
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 835
      return-void
  
+     .line 832
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 833
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1008,1028 ****
--- 1177,1201 ----
      .end annotation
  
      .prologue
+     .line 551
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 552
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 554
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 555
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
***************
*** 1032,1037 ****
--- 1205,1211 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 556
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x7
***************
*** 1040,1065 ****
--- 1214,1246 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 557
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 560
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 561
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 563
      return-void
  
+     .line 555
      :cond_0
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 560
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 561
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1076,1097 ****
--- 1257,1283 ----
      .end annotation
  
      .prologue
+     .line 947
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 949
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 950
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 951
      const/4 v1, 0x0
  
      invoke-virtual {v0, p2, v1}, Landroid/os/Parcel;->writeTypedArray([Landroid/os/Parcelable;I)V
  
+     .line 952
      iget-object v1, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v2, 0x1b
***************
*** 1104,1113 ****
--- 1290,1302 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 955
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 957
      return-void
  
+     .line 955
      :catchall_0
      move-exception v1
  
***************
*** 1130,1160 ****
--- 1319,1356 ----
  
      const/4 v3, 0x0
  
+     .line 886
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 887
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 890
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 891
      if-eqz p1, :cond_0
  
+     .line 892
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 893
      const/4 v4, 0x0
  
      invoke-virtual {p1, v0, v4}, Landroid/view/inputmethod/InputMethodSubtype;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 898
      :goto_0
      iget-object v4, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 1164,1171 ****
--- 1360,1369 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 899
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 900
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1174,1187 ****
--- 1372,1389 ----
  
      if-eqz v4, :cond_1
  
+     .line 903
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 904
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 906
      return v2
  
+     .line 896
      .end local v2           #_result:Z
      :cond_0
      const/4 v4, 0x0
***************
*** 1193,1203 ****
--- 1395,1407 ----
  
      goto :goto_0
  
+     .line 903
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 904
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1205,1210 ****
--- 1409,1415 ----
      :cond_1
      move v2, v3
  
+     .line 900
      goto :goto_1
  .end method
  
***************
*** 1220,1246 ****
--- 1425,1458 ----
      .end annotation
  
      .prologue
+     .line 806
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 807
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 809
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 810
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 811
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 812
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 813
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x14
***************
*** 1249,1269 ****
--- 1461,1487 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 814
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 817
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 818
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 820
      return-void
  
+     .line 817
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 818
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1280,1304 ****
--- 1498,1528 ----
      .end annotation
  
      .prologue
+     .line 718
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 719
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 721
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 722
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 723
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 724
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xf
***************
*** 1307,1327 ****
--- 1531,1557 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 725
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 728
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 729
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 731
      return-void
  
+     .line 728
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 729
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1339,1373 ****
--- 1569,1612 ----
      .end annotation
  
      .prologue
+     .line 734
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 735
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 737
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 738
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 739
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 740
      if-eqz p3, :cond_0
  
+     .line 741
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 742
      const/4 v2, 0x0
  
      invoke-virtual {p3, v0, v2}, Landroid/view/inputmethod/InputMethodSubtype;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 747
      :goto_0
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 1377,1392 ****
--- 1616,1636 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 748
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 751
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 752
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 754
      return-void
  
+     .line 745
      :cond_0
      const/4 v2, 0x0
  
***************
*** 1397,1407 ****
--- 1641,1653 ----
  
      goto :goto_0
  
+     .line 751
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 752
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1422,1444 ****
--- 1668,1695 ----
  
      const/4 v3, 0x0
  
+     .line 928
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 929
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 932
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 933
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 934
      if-eqz p2, :cond_0
  
      move v4, v2
***************
*** 1446,1451 ****
--- 1697,1703 ----
      :goto_0
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 935
      iget-object v4, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v5, 0x1a
***************
*** 1454,1461 ****
--- 1706,1715 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 936
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 937
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1464,1493 ****
--- 1718,1754 ----
  
      if-eqz v4, :cond_1
  
+     .line 940
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 941
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 943
      return v2
  
      .end local v2           #_result:Z
      :cond_0
      move v4, v3
  
+     .line 934
      goto :goto_0
  
      :cond_1
      move v2, v3
  
+     .line 937
      goto :goto_1
  
+     .line 940
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 941
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1504,1524 ****
--- 1765,1789 ----
      .end annotation
  
      .prologue
+     .line 702
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 703
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 705
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 706
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
***************
*** 1528,1535 ****
--- 1793,1802 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 707
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 708
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xe
***************
*** 1538,1563 ****
--- 1805,1837 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 709
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 712
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 713
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 715
      return-void
  
+     .line 706
      :cond_0
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 712
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 713
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1573,1593 ****
--- 1847,1871 ----
      .end annotation
  
      .prologue
+     .line 687
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 688
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 690
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 691
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
***************
*** 1597,1602 ****
--- 1875,1881 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 692
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xd
***************
*** 1605,1630 ****
--- 1884,1916 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 693
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 696
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 697
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 699
      return-void
  
+     .line 691
      :cond_0
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 696
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 697
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1641,1665 ****
--- 1927,1957 ----
      .end annotation
  
      .prologue
+     .line 773
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 774
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 776
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 777
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 778
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 779
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x12
***************
*** 1668,1688 ****
--- 1960,1986 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 780
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 783
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 784
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 786
      return-void
  
+     .line 783
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 784
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 1704,1724 ****
--- 2002,2026 ----
  
      const/4 v3, 0x0
  
+     .line 614
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 615
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 618
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 619
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
***************
*** 1728,1745 ****
--- 2030,2052 ----
      :goto_0
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 620
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 621
      if-eqz p3, :cond_1
  
+     .line 622
      const/4 v4, 0x1
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 623
      const/4 v4, 0x0
  
      invoke-virtual {p3, v0, v4}, Landroid/os/ResultReceiver;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 628
      :goto_1
      iget-object v4, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 1749,1756 ****
--- 2056,2065 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 629
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 630
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1759,1778 ****
--- 2068,2092 ----
  
      if-eqz v4, :cond_2
  
+     .line 633
      .local v2, _result:Z
      :goto_2
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 634
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 636
      return v2
  
+     .line 619
      .end local v2           #_result:Z
      :cond_0
      const/4 v4, 0x0
  
      goto :goto_0
  
+     .line 626
      :cond_1
      const/4 v4, 0x0
  
***************
*** 1783,1793 ****
--- 2097,2109 ----
  
      goto :goto_1
  
+     .line 633
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 634
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1795,1800 ****
--- 2111,2117 ----
      :cond_2
      move v2, v3
  
+     .line 630
      goto :goto_2
  .end method
  
***************
*** 1818,1838 ****
--- 2135,2159 ----
  
      const/4 v4, 0x0
  
+     .line 566
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 567
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 570
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v6, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v6}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 571
      if-eqz p1, :cond_1
  
      invoke-interface {p1}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
***************
*** 1842,1847 ****
--- 2163,2169 ----
      :goto_0
      invoke-virtual {v0, v6}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 572
      if-eqz p2, :cond_0
  
      invoke-interface {p2}, Lcom/android/internal/view/IInputContext;->asBinder()Landroid/os/IBinder;
***************
*** 1851,1866 ****
--- 2173,2192 ----
      :cond_0
      invoke-virtual {v0, v5}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 573
      if-eqz p3, :cond_2
  
+     .line 574
      const/4 v5, 0x1
  
      invoke-virtual {v0, v5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 575
      const/4 v5, 0x0
  
      invoke-virtual {p3, v0, v5}, Landroid/view/inputmethod/EditorInfo;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 580
      :goto_1
      if-eqz p4, :cond_3
  
***************
*** 1869,1879 ****
--- 2195,2207 ----
      :goto_2
      invoke-virtual {v0, v5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 581
      if-eqz p5, :cond_4
  
      :goto_3
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 582
      iget-object v3, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x8
***************
*** 1882,1895 ****
--- 2210,2226 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 583
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 584
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_5
  
+     .line 585
      sget-object v3, Lcom/android/internal/view/InputBindResult;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 1900,1919 ****
--- 2231,2255 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 592
      .local v2, _result:Lcom/android/internal/view/InputBindResult;
      :goto_4
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 593
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 595
      return-object v2
  
      .end local v2           #_result:Lcom/android/internal/view/InputBindResult;
      :cond_1
      move-object v6, v5
  
+     .line 571
      goto :goto_0
  
+     .line 578
      :cond_2
      const/4 v5, 0x0
  
***************
*** 1924,1934 ****
--- 2260,2272 ----
  
      goto :goto_1
  
+     .line 592
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 593
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1936,1948 ****
--- 2274,2289 ----
      :cond_3
      move v5, v4
  
+     .line 580
      goto :goto_2
  
      :cond_4
      move v3, v4
  
+     .line 581
      goto :goto_3
  
+     .line 588
      :cond_5
      const/4 v2, 0x0
  
***************
*** 1962,1984 ****
--- 2303,2330 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 910
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 911
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 914
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 915
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 916
      iget-object v3, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x19
***************
*** 1987,1994 ****
--- 2333,2342 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 917
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 918
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1999,2018 ****
--- 2347,2371 ----
  
      const/4 v2, 0x1
  
+     .line 921
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 922
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 924
      return v2
  
+     .line 921
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 922
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 2030,2056 ****
--- 2383,2416 ----
      .end annotation
  
      .prologue
+     .line 789
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 790
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 792
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 793
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 794
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 795
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 796
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x13
***************
*** 2059,2079 ****
--- 2419,2445 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 797
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 800
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 801
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 803
      return-void
  
+     .line 800
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 801
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 2099,2119 ****
--- 2465,2489 ----
  
      const/4 v3, 0x0
  
+     .line 666
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 667
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 669
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.view.IInputMethodManager"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 670
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
***************
*** 2123,2130 ****
--- 2493,2502 ----
      :goto_0
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 671
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 672
      if-eqz p3, :cond_1
  
      move v4, v2
***************
*** 2132,2137 ****
--- 2504,2510 ----
      :goto_1
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 673
      if-eqz p4, :cond_2
  
      move v4, v2
***************
*** 2139,2153 ****
--- 2512,2530 ----
      :goto_2
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 674
      invoke-virtual {v0, p5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 675
      if-eqz p6, :cond_3
  
      :goto_3
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 676
      invoke-virtual {v0, p7}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 677
      iget-object v2, p0, Lcom/android/internal/view/IInputMethodManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xc
***************
*** 2156,2171 ****
--- 2533,2553 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 678
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 681
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 682
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 684
      return-void
  
+     .line 670
      :cond_0
      const/4 v4, 0x0
  
***************
*** 2174,2196 ****
--- 2556,2583 ----
      :cond_1
      move v4, v3
  
+     .line 672
      goto :goto_1
  
      :cond_2
      move v4, v3
  
+     .line 673
      goto :goto_2
  
      :cond_3
      move v2, v3
  
+     .line 675
      goto :goto_3
  
+     .line 681
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 682
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
