*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/IInputConnectionWrapper$MyHandler.smali	2012-03-23 19:37:53.094743264 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/view/IInputConnectionWrapper$MyHandler.smali.original	2012-03-23 19:34:46.392338751 -0500
***************
*** 25,34 ****
--- 25,37 ----
      .parameter "looper"
  
      .prologue
+     .line 71
      iput-object p1, p0, Lcom/android/internal/view/IInputConnectionWrapper$MyHandler;->this$0:Lcom/android/internal/view/IInputConnectionWrapper;
  
+     .line 72
      invoke-direct {p0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
+     .line 73
      return-void
  .end method
  
***************
*** 39,47 ****
--- 42,52 ----
      .parameter "msg"
  
      .prologue
+     .line 77
      iget-object v0, p0, Lcom/android/internal/view/IInputConnectionWrapper$MyHandler;->this$0:Lcom/android/internal/view/IInputConnectionWrapper;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/view/IInputConnectionWrapper;->executeMessage(Landroid/os/Message;)V
  
+     .line 78
      return-void
  .end method
