*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/CallerInfoAsyncQuery.smali	2012-03-23 19:37:41.143525334 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/CallerInfoAsyncQuery.smali.original	2012-03-23 19:34:46.240335979 -0500
***************
*** 42,49 ****
--- 42,51 ----
      .locals 0
  
      .prologue
+     .line 302
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 303
      return-void
  .end method
  
***************
*** 52,57 ****
--- 54,60 ----
      .parameter "x0"
  
      .prologue
+     .line 42
      invoke-direct {p0}, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->release()V
  
      return-void
***************
*** 63,72 ****
--- 66,77 ----
      .parameter "contactRef"
  
      .prologue
+     .line 446
      if-eqz p1, :cond_0
  
      if-nez p2, :cond_1
  
+     .line 447
      :cond_0
      new-instance v0, Lcom/android/internal/telephony/CallerInfoAsyncQuery$QueryPoolException;
  
***************
*** 76,81 ****
--- 81,87 ----
  
      throw v0
  
+     .line 449
      :cond_1
      new-instance v0, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
***************
*** 85,100 ****
--- 91,109 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
+     .line 450
      iget-object v0, p0, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
      #setter for: Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->mQueryContext:Landroid/content/Context;
      invoke-static {v0, p1}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->access$302(Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;Landroid/content/Context;)Landroid/content/Context;
  
+     .line 451
      iget-object v0, p0, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
      #setter for: Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->mQueryUri:Landroid/net/Uri;
      invoke-static {v0, p2}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->access$402(Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;Landroid/net/Uri;)Landroid/net/Uri;
  
+     .line 452
      return-void
  .end method
  
***************
*** 104,126 ****
--- 113,140 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 458
      iget-object v0, p0, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
      #setter for: Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->mQueryContext:Landroid/content/Context;
      invoke-static {v0, v1}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->access$302(Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;Landroid/content/Context;)Landroid/content/Context;
  
+     .line 459
      iget-object v0, p0, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
      #setter for: Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->mQueryUri:Landroid/net/Uri;
      invoke-static {v0, v1}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->access$402(Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;Landroid/net/Uri;)Landroid/net/Uri;
  
+     .line 460
      iget-object v0, p0, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
      #setter for: Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->mCallerInfo:Lcom/android/internal/telephony/CallerInfo;
      invoke-static {v0, v1}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->access$502(Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;Lcom/android/internal/telephony/CallerInfo;)Lcom/android/internal/telephony/CallerInfo;
  
+     .line 461
      iput-object v1, p0, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
+     .line 462
      return-void
  .end method
  
***************
*** 129,140 ****
--- 143,157 ----
      .parameter "uri"
  
      .prologue
+     .line 465
      if-eqz p0, :cond_1
  
+     .line 466
      invoke-virtual {p0}, Landroid/net/Uri;->toString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 467
      .local v1, uriString:Ljava/lang/String;
      const/16 v2, 0x2f
  
***************
*** 142,150 ****
--- 159,169 ----
  
      move-result v0
  
+     .line 468
      .local v0, indexOfLastSlash:I
      if-lez v0, :cond_0
  
+     .line 469
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 169,174 ****
--- 188,194 ----
  
      move-result-object v1
  
+     .line 474
      .end local v0           #indexOfLastSlash:I
      .end local v1           #uriString:Ljava/lang/String;
      :cond_0
***************
*** 192,217 ****
--- 212,244 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 312
      new-instance v8, Lcom/android/internal/telephony/CallerInfoAsyncQuery;
  
      invoke-direct {v8}, Lcom/android/internal/telephony/CallerInfoAsyncQuery;-><init>()V
  
+     .line 313
      .local v8, c:Lcom/android/internal/telephony/CallerInfoAsyncQuery;
      invoke-direct {v8, p1, p2}, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->allocate(Landroid/content/Context;Landroid/net/Uri;)V
  
+     .line 318
      new-instance v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;
  
      invoke-direct {v2, v4}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;-><init>(Lcom/android/internal/telephony/CallerInfoAsyncQuery$1;)V
  
+     .line 319
      .local v2, cw:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;
      iput-object p3, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->listener:Lcom/android/internal/telephony/CallerInfoAsyncQuery$OnQueryCompleteListener;
  
+     .line 320
      iput-object p4, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->cookie:Ljava/lang/Object;
  
+     .line 321
      const/4 v0, 0x1
  
      iput v0, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->event:I
  
+     .line 323
      iget-object v0, v8, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
      move v1, p0
***************
*** 226,231 ****
--- 253,259 ----
  
      invoke-virtual/range {v0 .. v7}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->startQuery(ILjava/lang/Object;Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 325
      return-object v8
  .end method
  
***************
*** 242,258 ****
  
      const/4 v4, 0x0
  
      invoke-static {p2}, Landroid/telephony/PhoneNumberUtils;->isUriNumber(Ljava/lang/String;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
      sget-object v3, Landroid/provider/ContactsContract$Data;->CONTENT_URI:Landroid/net/Uri;
  
      .local v3, contactRef:Landroid/net/Uri;
!     const-string v5, "upper(data1)=? AND mimetype=\'vnd.android.cursor.item/sip_address\'"
  
      .local v5, selection:Ljava/lang/String;
      new-array v6, v7, [Ljava/lang/String;
  
--- 270,290 ----
  
      const/4 v4, 0x0
  
+     .line 353
      invoke-static {p2}, Landroid/telephony/PhoneNumberUtils;->isUriNumber(Ljava/lang/String;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 358
      sget-object v3, Landroid/provider/ContactsContract$Data;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 374
      .local v3, contactRef:Landroid/net/Uri;
!     const-string/jumbo v5, "upper(data1)=? AND mimetype=\'vnd.android.cursor.item/sip_address\'"
  
+     .line 377
      .local v5, selection:Ljava/lang/String;
      new-array v6, v7, [Ljava/lang/String;
  
***************
*** 264,299 ****
--- 296,340 ----
  
      aput-object v1, v6, v0
  
+     .line 396
      .local v6, selectionArgs:[Ljava/lang/String;
      :goto_0
      new-instance v8, Lcom/android/internal/telephony/CallerInfoAsyncQuery;
  
      invoke-direct {v8}, Lcom/android/internal/telephony/CallerInfoAsyncQuery;-><init>()V
  
+     .line 397
      .local v8, c:Lcom/android/internal/telephony/CallerInfoAsyncQuery;
      invoke-direct {v8, p1, v3}, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->allocate(Landroid/content/Context;Landroid/net/Uri;)V
  
+     .line 400
      new-instance v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;
  
      invoke-direct {v2, v4}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;-><init>(Lcom/android/internal/telephony/CallerInfoAsyncQuery$1;)V
  
+     .line 401
      .local v2, cw:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;
      iput-object p3, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->listener:Lcom/android/internal/telephony/CallerInfoAsyncQuery$OnQueryCompleteListener;
  
+     .line 402
      iput-object p4, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->cookie:Ljava/lang/Object;
  
+     .line 403
      iput-object p2, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->number:Ljava/lang/String;
  
+     .line 406
      invoke-static {p2, p1}, Landroid/telephony/PhoneNumberUtils;->isLocalEmergencyNumber(Ljava/lang/String;Landroid/content/Context;)Z
  
      move-result v0
  
      if-eqz v0, :cond_1
  
+     .line 407
      const/4 v0, 0x4
  
      iput v0, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->event:I
  
+     .line 414
      :goto_1
      iget-object v0, v8, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
***************
*** 303,310 ****
--- 344,353 ----
  
      invoke-virtual/range {v0 .. v7}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->startQuery(ILjava/lang/Object;Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 421
      return-object v8
  
+     .line 381
      .end local v2           #cw:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;
      .end local v3           #contactRef:Landroid/net/Uri;
      .end local v5           #selection:Ljava/lang/String;
***************
*** 321,335 ****
--- 364,381 ----
  
      move-result-object v3
  
+     .line 382
      .restart local v3       #contactRef:Landroid/net/Uri;
      const/4 v5, 0x0
  
+     .line 383
      .restart local v5       #selection:Ljava/lang/String;
      const/4 v6, 0x0
  
      .restart local v6       #selectionArgs:[Ljava/lang/String;
      goto :goto_0
  
+     .line 408
      .restart local v2       #cw:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;
      .restart local v8       #c:Lcom/android/internal/telephony/CallerInfoAsyncQuery;
      :cond_1
***************
*** 339,350 ****
--- 385,398 ----
  
      if-eqz v0, :cond_2
  
+     .line 409
      const/4 v0, 0x5
  
      iput v0, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->event:I
  
      goto :goto_1
  
+     .line 411
      :cond_2
      iput v7, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->event:I
  
***************
*** 362,380 ****
--- 410,433 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 433
      new-instance v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;
  
      invoke-direct {v2, v3}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;-><init>(Lcom/android/internal/telephony/CallerInfoAsyncQuery$1;)V
  
+     .line 434
      .local v2, cw:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;
      iput-object p2, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->listener:Lcom/android/internal/telephony/CallerInfoAsyncQuery$OnQueryCompleteListener;
  
+     .line 435
      iput-object p3, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->cookie:Ljava/lang/Object;
  
+     .line 436
      const/4 v0, 0x2
  
      iput v0, v2, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CookieWrapper;->event:I
  
+     .line 438
      iget-object v0, p0, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->mHandler:Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;
  
      move v1, p1
***************
*** 389,393 ****
--- 442,447 ----
  
      invoke-virtual/range {v0 .. v7}, Lcom/android/internal/telephony/CallerInfoAsyncQuery$CallerInfoAsyncQueryHandler;->startQuery(ILjava/lang/Object;Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 439
      return-void
  .end method
