*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/CallTracker.smali	2012-03-23 19:37:39.721499403 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/CallTracker.smali.original	2012-03-23 19:34:46.202335285 -0500
***************
*** 52,57 ****
--- 52,58 ----
      .locals 0
  
      .prologue
+     .line 30
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
  
      return-void
***************
*** 61,66 ****
--- 62,68 ----
      .locals 1
  
      .prologue
+     .line 116
      iget v0, p0, Lcom/android/internal/telephony/CallTracker;->pendingOperations:I
  
      if-nez v0, :cond_0
***************
*** 88,95 ****
--- 90,99 ----
      .locals 0
  
      .prologue
+     .line 90
      invoke-virtual {p0}, Lcom/android/internal/telephony/CallTracker;->pollCallsWhenSafe()V
  
+     .line 91
      return-void
  .end method
  
***************
*** 98,103 ****
--- 102,108 ----
      .parameter "e"
  
      .prologue
+     .line 82
      if-eqz p1, :cond_0
  
      instance-of v0, p1, Lcom/android/internal/telephony/CommandException;
***************
*** 134,149 ****
--- 139,157 ----
      .parameter "what"
  
      .prologue
+     .line 103
      iget v0, p0, Lcom/android/internal/telephony/CallTracker;->pendingOperations:I
  
      add-int/lit8 v0, v0, 0x1
  
      iput v0, p0, Lcom/android/internal/telephony/CallTracker;->pendingOperations:I
  
+     .line 104
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/internal/telephony/CallTracker;->lastRelevantPoll:Landroid/os/Message;
  
+     .line 105
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/CallTracker;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
***************
*** 155,173 ****
--- 163,185 ----
      .locals 3
  
      .prologue
+     .line 74
      invoke-virtual {p0}, Lcom/android/internal/telephony/CallTracker;->obtainMessage()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 76
      .local v0, msg:Landroid/os/Message;
      const/4 v1, 0x3
  
      iput v1, v0, Landroid/os/Message;->what:I
  
+     .line 77
      const-wide/16 v1, 0xfa
  
      invoke-virtual {p0, v0, v1, v2}, Lcom/android/internal/telephony/CallTracker;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 78
      return-void
  .end method
  
***************
*** 177,202 ****
--- 189,219 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 64
      iput-boolean v1, p0, Lcom/android/internal/telephony/CallTracker;->needsPoll:Z
  
+     .line 66
      invoke-direct {p0}, Lcom/android/internal/telephony/CallTracker;->checkNoOperationsPending()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 67
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/CallTracker;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/internal/telephony/CallTracker;->lastRelevantPoll:Landroid/os/Message;
  
+     .line 68
      iget-object v0, p0, Lcom/android/internal/telephony/CallTracker;->cm:Lcom/android/internal/telephony/CommandsInterface;
  
      iget-object v1, p0, Lcom/android/internal/telephony/CallTracker;->lastRelevantPoll:Landroid/os/Message;
  
      invoke-interface {v0, v1}, Lcom/android/internal/telephony/CommandsInterface;->getCurrentCalls(Landroid/os/Message;)V
  
+     .line 70
      :cond_0
      return-void
  .end method
