*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/Call.smali	2012-03-23 19:37:41.168525790 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/Call.smali.original	2012-03-23 19:34:46.240335979 -0500
***************
*** 24,39 ****
--- 24,43 ----
      .locals 1
  
      .prologue
+     .line 26
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 48
      sget-object v0, Lcom/android/internal/telephony/Call$State;->IDLE:Lcom/android/internal/telephony/Call$State;
  
      iput-object v0, p0, Lcom/android/internal/telephony/Call;->state:Lcom/android/internal/telephony/Call$State;
  
+     .line 57
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/internal/telephony/Call;->isGeneric:Z
  
+     .line 59
      const-string v0, "Call"
  
      iput-object v0, p0, Lcom/android/internal/telephony/Call;->LOG_TAG:Ljava/lang/String;
***************
*** 59,71 ****
--- 63,78 ----
      .locals 10
  
      .prologue
+     .line 172
      const-wide v6, 0x7fffffffffffffffL
  
+     .line 173
      .local v6, time:J
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->getConnections()Ljava/util/List;
  
      move-result-object v2
  
+     .line 175
      .local v2, l:Ljava/util/List;
      invoke-interface {v2}, Ljava/util/List;->size()I
  
***************
*** 73,83 ****
--- 80,93 ----
  
      if-nez v8, :cond_0
  
+     .line 176
      const-wide/16 v8, 0x0
  
+     .line 188
      :goto_0
      return-wide v8
  
+     .line 179
      :cond_0
      const/4 v1, 0x0
  
***************
*** 90,106 ****
--- 100,119 ----
      :goto_1
      if-ge v1, v3, :cond_2
  
+     .line 180
      invoke-interface {v2, v1}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Lcom/android/internal/telephony/Connection;
  
+     .line 183
      .local v0, c:Lcom/android/internal/telephony/Connection;
      invoke-virtual {v0}, Lcom/android/internal/telephony/Connection;->getConnectTime()J
  
      move-result-wide v4
  
+     .line 185
      .local v4, t:J
      cmp-long v8, v4, v6
  
***************
*** 108,113 ****
--- 121,127 ----
  
      move-wide v6, v4
  
+     .line 179
      :cond_1
      add-int/lit8 v1, v1, 0x1
  
***************
*** 118,123 ****
--- 132,138 ----
      :cond_2
      move-wide v8, v6
  
+     .line 188
      goto :goto_0
  .end method
  
***************
*** 125,140 ****
--- 140,159 ----
      .locals 10
  
      .prologue
+     .line 122
      const-wide v7, 0x7fffffffffffffffL
  
+     .line 124
      .local v7, time:J
      const/4 v1, 0x0
  
+     .line 126
      .local v1, earliest:Lcom/android/internal/telephony/Connection;
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->getConnections()Ljava/util/List;
  
      move-result-object v3
  
+     .line 128
      .local v3, l:Ljava/util/List;
      invoke-interface {v3}, Ljava/util/List;->size()I
  
***************
*** 142,152 ****
--- 161,174 ----
  
      if-nez v9, :cond_0
  
+     .line 129
      const/4 v9, 0x0
  
+     .line 144
      :goto_0
      return-object v9
  
+     .line 132
      :cond_0
      const/4 v2, 0x0
  
***************
*** 159,184 ****
--- 181,212 ----
      :goto_1
      if-ge v2, v4, :cond_2
  
+     .line 133
      invoke-interface {v3, v2}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Lcom/android/internal/telephony/Connection;
  
+     .line 136
      .local v0, c:Lcom/android/internal/telephony/Connection;
      invoke-virtual {v0}, Lcom/android/internal/telephony/Connection;->getCreateTime()J
  
      move-result-wide v5
  
+     .line 138
      .local v5, t:J
      cmp-long v9, v5, v7
  
      if-gez v9, :cond_1
  
+     .line 139
      move-object v1, v0
  
+     .line 140
      move-wide v7, v5
  
+     .line 132
      :cond_1
      add-int/lit8 v2, v2, 0x1
  
***************
*** 189,194 ****
--- 217,223 ----
      :cond_2
      move-object v9, v1
  
+     .line 144
      goto :goto_0
  .end method
  
***************
*** 196,208 ****
--- 225,240 ----
      .locals 10
  
      .prologue
+     .line 150
      const-wide v6, 0x7fffffffffffffffL
  
+     .line 152
      .local v6, time:J
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->getConnections()Ljava/util/List;
  
      move-result-object v2
  
+     .line 154
      .local v2, l:Ljava/util/List;
      invoke-interface {v2}, Ljava/util/List;->size()I
  
***************
*** 210,220 ****
--- 242,255 ----
  
      if-nez v8, :cond_0
  
+     .line 155
      const-wide/16 v8, 0x0
  
+     .line 167
      :goto_0
      return-wide v8
  
+     .line 158
      :cond_0
      const/4 v1, 0x0
  
***************
*** 227,243 ****
--- 262,281 ----
      :goto_1
      if-ge v1, v3, :cond_2
  
+     .line 159
      invoke-interface {v2, v1}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Lcom/android/internal/telephony/Connection;
  
+     .line 162
      .local v0, c:Lcom/android/internal/telephony/Connection;
      invoke-virtual {v0}, Lcom/android/internal/telephony/Connection;->getCreateTime()J
  
      move-result-wide v4
  
+     .line 164
      .local v4, t:J
      cmp-long v8, v4, v6
  
***************
*** 245,250 ****
--- 283,289 ----
  
      move-wide v6, v4
  
+     .line 158
      :cond_1
      add-int/lit8 v1, v1, 0x1
  
***************
*** 255,260 ****
--- 294,300 ----
      :cond_2
      move-wide v8, v6
  
+     .line 167
      goto :goto_0
  .end method
  
***************
*** 262,271 ****
--- 302,313 ----
      .locals 10
  
      .prologue
+     .line 208
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->getConnections()Ljava/util/List;
  
      move-result-object v2
  
+     .line 209
      .local v2, l:Ljava/util/List;
      invoke-interface {v2}, Ljava/util/List;->size()I
  
***************
*** 273,289 ****
--- 315,336 ----
  
      if-nez v9, :cond_1
  
+     .line 210
      const/4 v3, 0x0
  
+     .line 225
      :cond_0
      return-object v3
  
+     .line 213
      :cond_1
      const-wide/16 v7, 0x0
  
+     .line 214
      .local v7, time:J
      const/4 v3, 0x0
  
+     .line 215
      .local v3, latest:Lcom/android/internal/telephony/Connection;
      const/4 v1, 0x0
  
***************
*** 296,321 ****
--- 343,374 ----
      :goto_0
      if-ge v1, v4, :cond_0
  
+     .line 216
      invoke-interface {v2, v1}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Lcom/android/internal/telephony/Connection;
  
+     .line 217
      .local v0, c:Lcom/android/internal/telephony/Connection;
      invoke-virtual {v0}, Lcom/android/internal/telephony/Connection;->getCreateTime()J
  
      move-result-wide v5
  
+     .line 219
      .local v5, t:J
      cmp-long v9, v5, v7
  
      if-lez v9, :cond_2
  
+     .line 220
      move-object v3, v0
  
+     .line 221
      move-wide v7, v5
  
+     .line 215
      :cond_2
      add-int/lit8 v1, v1, 0x1
  
***************
*** 329,334 ****
--- 382,388 ----
      .locals 1
  
      .prologue
+     .line 102
      iget-object v0, p0, Lcom/android/internal/telephony/Call;->state:Lcom/android/internal/telephony/Call$State;
  
      return-object v0
***************
*** 346,351 ****
--- 400,406 ----
      .locals 4
  
      .prologue
+     .line 247
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->getState()Lcom/android/internal/telephony/Call$State;
  
      move-result-object v1
***************
*** 356,373 ****
--- 411,432 ----
  
      if-eqz v1, :cond_0
  
+     .line 249
      :try_start_0
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->hangup()V
      :try_end_0
      .catch Lcom/android/internal/telephony/CallStateException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 254
      :cond_0
      :goto_0
      return-void
  
+     .line 250
      :catch_0
      move-exception v0
  
+     .line 251
      .local v0, ex:Lcom/android/internal/telephony/CallStateException;
      const-string v1, "Call"
  
***************
*** 399,404 ****
--- 458,464 ----
      .parameter "c"
  
      .prologue
+     .line 80
      invoke-virtual {p1}, Lcom/android/internal/telephony/Connection;->getCall()Lcom/android/internal/telephony/Call;
  
      move-result-object v0
***************
*** 422,434 ****
--- 482,497 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 88
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->getConnections()Ljava/util/List;
  
      move-result-object v0
  
+     .line 90
      .local v0, connections:Ljava/util/List;
      if-nez v0, :cond_1
  
+     .line 94
      :cond_0
      :goto_0
      return v1
***************
*** 449,454 ****
--- 512,518 ----
      .locals 1
  
      .prologue
+     .line 194
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->getState()Lcom/android/internal/telephony/Call$State;
  
      move-result-object v0
***************
*** 464,469 ****
--- 528,534 ----
      .locals 1
  
      .prologue
+     .line 233
      iget-boolean v0, p0, Lcom/android/internal/telephony/Call;->isGeneric:Z
  
      return v0
***************
*** 473,478 ****
--- 538,544 ----
      .locals 1
  
      .prologue
+     .line 112
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->getState()Lcom/android/internal/telephony/Call$State;
  
      move-result-object v0
***************
*** 501,506 ****
--- 567,573 ----
      .locals 1
  
      .prologue
+     .line 199
      invoke-virtual {p0}, Lcom/android/internal/telephony/Call;->getState()Lcom/android/internal/telephony/Call$State;
  
      move-result-object v0
***************
*** 517,523 ****
--- 584,592 ----
      .parameter "generic"
  
      .prologue
+     .line 240
      iput-boolean p1, p0, Lcom/android/internal/telephony/Call;->isGeneric:Z
  
+     .line 241
      return-void
  .end method
