*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/CommandsInterface$RadioState.smali	2012-03-23 19:37:36.624442928 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/CommandsInterface$RadioState.smali.original	2012-03-23 19:34:46.154334411 -0500
***************
*** 66,71 ****
--- 66,72 ----
  
      const/4 v3, 0x1
  
+     .line 29
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "RADIO_OFF"
***************
*** 74,79 ****
--- 75,81 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->RADIO_OFF:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 30
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "RADIO_UNAVAILABLE"
***************
*** 82,87 ****
--- 84,90 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->RADIO_UNAVAILABLE:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 31
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "SIM_NOT_READY"
***************
*** 90,95 ****
--- 93,99 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->SIM_NOT_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 32
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "SIM_LOCKED_OR_ABSENT"
***************
*** 98,103 ****
--- 102,108 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->SIM_LOCKED_OR_ABSENT:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 34
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "SIM_READY"
***************
*** 106,111 ****
--- 111,117 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->SIM_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 35
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "RUIM_NOT_READY"
***************
*** 116,121 ****
--- 122,128 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->RUIM_NOT_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 36
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "RUIM_READY"
***************
*** 126,131 ****
--- 133,139 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->RUIM_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 37
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "RUIM_LOCKED_OR_ABSENT"
***************
*** 136,141 ****
--- 144,150 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->RUIM_LOCKED_OR_ABSENT:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 39
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "NV_NOT_READY"
***************
*** 146,151 ****
--- 155,161 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->NV_NOT_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 40
      new-instance v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      const-string v1, "NV_READY"
***************
*** 156,161 ****
--- 166,172 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->NV_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
+     .line 28
      const/16 v0, 0xa
  
      new-array v0, v0, [Lcom/android/internal/telephony/CommandsInterface$RadioState;
***************
*** 227,236 ****
--- 238,250 ----
      .end annotation
  
      .prologue
+     .line 53
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 54
      iput p3, p0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->stateType:I
  
+     .line 55
      return-void
  .end method
  
***************
*** 239,244 ****
--- 253,259 ----
      .parameter "name"
  
      .prologue
+     .line 28
      const-class v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 254,259 ****
--- 269,275 ----
      .locals 1
  
      .prologue
+     .line 28
      sget-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->$VALUES:[Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/CommandsInterface$RadioState;->clone()Ljava/lang/Object;
***************
*** 271,276 ****
--- 287,293 ----
      .locals 1
  
      .prologue
+     .line 58
      iget v0, p0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->stateType:I
  
      return v0
***************
*** 280,285 ****
--- 297,303 ----
      .locals 1
  
      .prologue
+     .line 62
      sget-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->RADIO_UNAVAILABLE:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      if-eq p0, v0, :cond_0
***************
*** 301,312 ****
--- 319,332 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 88
      invoke-static {}, Lcom/android/internal/telephony/BaseCommands;->getLteOnCdmaModeStatic()I
  
      move-result v1
  
      if-ne v1, v0, :cond_1
  
+     .line 91
      :cond_0
      :goto_0
      return v0
***************
*** 345,356 ****
--- 365,378 ----
  
      const/4 v0, 0x0
  
+     .line 78
      invoke-static {}, Lcom/android/internal/telephony/BaseCommands;->getLteOnCdmaModeStatic()I
  
      move-result v2
  
      if-ne v2, v1, :cond_1
  
+     .line 81
      :cond_0
      :goto_0
      return v0
***************
*** 378,383 ****
--- 400,406 ----
      .locals 1
  
      .prologue
+     .line 74
      sget-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->NV_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      if-ne p0, v0, :cond_0
***************
*** 397,402 ****
--- 420,426 ----
      .locals 1
  
      .prologue
+     .line 43
      sget-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->SIM_NOT_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      if-eq p0, v0, :cond_0
***************
*** 445,450 ****
--- 469,475 ----
      .locals 1
  
      .prologue
+     .line 70
      sget-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->RUIM_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      if-ne p0, v0, :cond_0
***************
*** 464,469 ****
--- 489,495 ----
      .locals 1
  
      .prologue
+     .line 66
      sget-object v0, Lcom/android/internal/telephony/CommandsInterface$RadioState;->SIM_READY:Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      if-ne p0, v0, :cond_0
