*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/CommandException$Error.smali	2012-03-23 19:37:37.172452921 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/CommandException$Error.smali.original	2012-03-23 19:34:46.163334575 -0500
***************
*** 72,77 ****
--- 72,78 ----
  
      const/4 v3, 0x0
  
+     .line 30
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "INVALID_RESPONSE"
***************
*** 80,85 ****
--- 81,87 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->INVALID_RESPONSE:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 31
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "RADIO_NOT_AVAILABLE"
***************
*** 88,93 ****
--- 90,96 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->RADIO_NOT_AVAILABLE:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 32
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "GENERIC_FAILURE"
***************
*** 96,101 ****
--- 99,105 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->GENERIC_FAILURE:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 33
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "PASSWORD_INCORRECT"
***************
*** 104,109 ****
--- 108,114 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->PASSWORD_INCORRECT:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 34
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "SIM_PIN2"
***************
*** 112,117 ****
--- 117,123 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->SIM_PIN2:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 35
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "SIM_PUK2"
***************
*** 122,127 ****
--- 128,134 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->SIM_PUK2:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 36
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "REQUEST_NOT_SUPPORTED"
***************
*** 132,137 ****
--- 139,145 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->REQUEST_NOT_SUPPORTED:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 37
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "OP_NOT_ALLOWED_DURING_VOICE_CALL"
***************
*** 142,147 ****
--- 150,156 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->OP_NOT_ALLOWED_DURING_VOICE_CALL:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 38
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "OP_NOT_ALLOWED_BEFORE_REG_NW"
***************
*** 152,157 ****
--- 161,167 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->OP_NOT_ALLOWED_BEFORE_REG_NW:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 39
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "SMS_FAIL_RETRY"
***************
*** 162,167 ****
--- 172,178 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->SMS_FAIL_RETRY:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 40
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "SIM_ABSENT"
***************
*** 172,177 ****
--- 183,189 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->SIM_ABSENT:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 41
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "SUBSCRIPTION_NOT_AVAILABLE"
***************
*** 182,187 ****
--- 194,200 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->SUBSCRIPTION_NOT_AVAILABLE:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 42
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "MODE_NOT_SUPPORTED"
***************
*** 192,197 ****
--- 205,211 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->MODE_NOT_SUPPORTED:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 43
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "FDN_CHECK_FAILURE"
***************
*** 202,207 ****
--- 216,222 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->FDN_CHECK_FAILURE:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 44
      new-instance v0, Lcom/android/internal/telephony/CommandException$Error;
  
      const-string v1, "ILLEGAL_SIM_OR_ME"
***************
*** 212,217 ****
--- 227,233 ----
  
      sput-object v0, Lcom/android/internal/telephony/CommandException$Error;->ILLEGAL_SIM_OR_ME:Lcom/android/internal/telephony/CommandException$Error;
  
+     .line 29
      const/16 v0, 0xf
  
      new-array v0, v0, [Lcom/android/internal/telephony/CommandException$Error;
***************
*** 312,317 ****
--- 328,334 ----
      .end annotation
  
      .prologue
+     .line 29
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 322,327 ****
--- 339,345 ----
      .parameter "name"
  
      .prologue
+     .line 29
      const-class v0, Lcom/android/internal/telephony/CommandException$Error;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 337,342 ****
--- 355,361 ----
      .locals 1
  
      .prologue
+     .line 29
      sget-object v0, Lcom/android/internal/telephony/CommandException$Error;->$VALUES:[Lcom/android/internal/telephony/CommandException$Error;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/CommandException$Error;->clone()Ljava/lang/Object;
