*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/Connection$DisconnectCause.smali	2012-03-23 19:37:36.651443420 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/Connection$DisconnectCause.smali.original	2012-03-23 19:34:46.154334411 -0500
***************
*** 116,121 ****
--- 116,122 ----
  
      const/4 v3, 0x0
  
+     .line 34
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "NOT_DISCONNECTED"
***************
*** 124,129 ****
--- 125,131 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->NOT_DISCONNECTED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 35
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "INCOMING_MISSED"
***************
*** 132,137 ****
--- 134,140 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->INCOMING_MISSED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 36
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "NORMAL"
***************
*** 140,145 ****
--- 143,149 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->NORMAL:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 37
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "LOCAL"
***************
*** 148,153 ****
--- 152,158 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->LOCAL:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 38
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "BUSY"
***************
*** 156,161 ****
--- 161,167 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->BUSY:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 39
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CONGESTION"
***************
*** 166,171 ****
--- 172,178 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CONGESTION:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 40
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "MMI"
***************
*** 176,181 ****
--- 183,189 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->MMI:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 41
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "INVALID_NUMBER"
***************
*** 186,191 ****
--- 194,200 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->INVALID_NUMBER:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 42
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "NUMBER_UNREACHABLE"
***************
*** 196,201 ****
--- 205,211 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->NUMBER_UNREACHABLE:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 43
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "SERVER_UNREACHABLE"
***************
*** 206,211 ****
--- 216,222 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->SERVER_UNREACHABLE:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 44
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "INVALID_CREDENTIALS"
***************
*** 216,221 ****
--- 227,233 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->INVALID_CREDENTIALS:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 45
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "OUT_OF_NETWORK"
***************
*** 226,231 ****
--- 238,244 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->OUT_OF_NETWORK:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 46
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "SERVER_ERROR"
***************
*** 236,241 ****
--- 249,255 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->SERVER_ERROR:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 47
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "TIMED_OUT"
***************
*** 246,251 ****
--- 260,266 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->TIMED_OUT:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 48
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "LOST_SIGNAL"
***************
*** 256,261 ****
--- 271,277 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->LOST_SIGNAL:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 49
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "LIMIT_EXCEEDED"
***************
*** 266,271 ****
--- 282,288 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->LIMIT_EXCEEDED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 50
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "INCOMING_REJECTED"
***************
*** 276,281 ****
--- 293,299 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->INCOMING_REJECTED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 51
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "POWER_OFF"
***************
*** 286,291 ****
--- 304,310 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->POWER_OFF:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 52
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "OUT_OF_SERVICE"
***************
*** 296,301 ****
--- 315,321 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->OUT_OF_SERVICE:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 53
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "ICC_ERROR"
***************
*** 306,311 ****
--- 326,332 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->ICC_ERROR:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 54
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CALL_BARRED"
***************
*** 316,321 ****
--- 337,343 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CALL_BARRED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 55
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "FDN_BLOCKED"
***************
*** 326,331 ****
--- 348,354 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->FDN_BLOCKED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 56
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CS_RESTRICTED"
***************
*** 336,341 ****
--- 359,365 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CS_RESTRICTED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 57
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CS_RESTRICTED_NORMAL"
***************
*** 346,351 ****
--- 370,376 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CS_RESTRICTED_NORMAL:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 58
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CS_RESTRICTED_EMERGENCY"
***************
*** 356,361 ****
--- 381,387 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CS_RESTRICTED_EMERGENCY:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 59
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "UNOBTAINABLE_NUMBER"
***************
*** 366,371 ****
--- 392,398 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->UNOBTAINABLE_NUMBER:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 60
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_LOCKED_UNTIL_POWER_CYCLE"
***************
*** 376,381 ****
--- 403,409 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_LOCKED_UNTIL_POWER_CYCLE:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 61
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_DROP"
***************
*** 386,391 ****
--- 414,420 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_DROP:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 62
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_INTERCEPT"
***************
*** 396,401 ****
--- 425,431 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_INTERCEPT:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 63
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_REORDER"
***************
*** 406,411 ****
--- 436,442 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_REORDER:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 64
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_SO_REJECT"
***************
*** 416,421 ****
--- 447,453 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_SO_REJECT:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 65
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_RETRY_ORDER"
***************
*** 426,431 ****
--- 458,464 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_RETRY_ORDER:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 66
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_ACCESS_FAILURE"
***************
*** 436,441 ****
--- 469,475 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_ACCESS_FAILURE:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 67
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_PREEMPTED"
***************
*** 446,451 ****
--- 480,486 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_PREEMPTED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 68
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_NOT_EMERGENCY"
***************
*** 456,461 ****
--- 491,497 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_NOT_EMERGENCY:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 69
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "CDMA_ACCESS_BLOCKED"
***************
*** 466,471 ****
--- 502,508 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->CDMA_ACCESS_BLOCKED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 70
      new-instance v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      const-string v1, "ERROR_UNSPECIFIED"
***************
*** 476,481 ****
--- 513,519 ----
  
      sput-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->ERROR_UNSPECIFIED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
+     .line 33
      const/16 v0, 0x25
  
      new-array v0, v0, [Lcom/android/internal/telephony/Connection$DisconnectCause;
***************
*** 708,713 ****
--- 746,752 ----
      .end annotation
  
      .prologue
+     .line 33
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 718,723 ****
--- 757,763 ----
      .parameter "name"
  
      .prologue
+     .line 33
      const-class v0, Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 733,738 ****
--- 773,779 ----
      .locals 1
  
      .prologue
+     .line 33
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->$VALUES:[Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/Connection$DisconnectCause;->clone()Ljava/lang/Object;
