*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataCallState$SetupResult.smali	2012-03-23 19:37:39.874502194 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataCallState$SetupResult.smali.original	2012-03-23 19:34:46.205335341 -0500
***************
*** 58,63 ****
--- 58,64 ----
  
      const/4 v3, 0x0
  
+     .line 56
      new-instance v0, Lcom/android/internal/telephony/DataCallState$SetupResult;
  
      const-string v1, "SUCCESS"
***************
*** 66,71 ****
--- 67,73 ----
  
      sput-object v0, Lcom/android/internal/telephony/DataCallState$SetupResult;->SUCCESS:Lcom/android/internal/telephony/DataCallState$SetupResult;
  
+     .line 57
      new-instance v0, Lcom/android/internal/telephony/DataCallState$SetupResult;
  
      const-string v1, "ERR_BadCommand"
***************
*** 74,79 ****
--- 76,82 ----
  
      sput-object v0, Lcom/android/internal/telephony/DataCallState$SetupResult;->ERR_BadCommand:Lcom/android/internal/telephony/DataCallState$SetupResult;
  
+     .line 58
      new-instance v0, Lcom/android/internal/telephony/DataCallState$SetupResult;
  
      const-string v1, "ERR_UnacceptableParameter"
***************
*** 82,87 ****
--- 85,91 ----
  
      sput-object v0, Lcom/android/internal/telephony/DataCallState$SetupResult;->ERR_UnacceptableParameter:Lcom/android/internal/telephony/DataCallState$SetupResult;
  
+     .line 59
      new-instance v0, Lcom/android/internal/telephony/DataCallState$SetupResult;
  
      const-string v1, "ERR_GetLastErrorFromRil"
***************
*** 90,95 ****
--- 94,100 ----
  
      sput-object v0, Lcom/android/internal/telephony/DataCallState$SetupResult;->ERR_GetLastErrorFromRil:Lcom/android/internal/telephony/DataCallState$SetupResult;
  
+     .line 60
      new-instance v0, Lcom/android/internal/telephony/DataCallState$SetupResult;
  
      const-string v1, "ERR_Stale"
***************
*** 98,103 ****
--- 103,109 ----
  
      sput-object v0, Lcom/android/internal/telephony/DataCallState$SetupResult;->ERR_Stale:Lcom/android/internal/telephony/DataCallState$SetupResult;
  
+     .line 61
      new-instance v0, Lcom/android/internal/telephony/DataCallState$SetupResult;
  
      const-string v1, "ERR_RilError"
***************
*** 108,113 ****
--- 114,120 ----
  
      sput-object v0, Lcom/android/internal/telephony/DataCallState$SetupResult;->ERR_RilError:Lcom/android/internal/telephony/DataCallState$SetupResult;
  
+     .line 55
      const/4 v0, 0x6
  
      new-array v0, v0, [Lcom/android/internal/telephony/DataCallState$SetupResult;
***************
*** 154,161 ****
--- 161,170 ----
      .end annotation
  
      .prologue
+     .line 65
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 66
      const/4 v0, 0x0
  
      invoke-static {v0}, Lcom/android/internal/telephony/DataConnection$FailCause;->fromInt(I)Lcom/android/internal/telephony/DataConnection$FailCause;
***************
*** 164,169 ****
--- 173,179 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataCallState$SetupResult;->mFailCause:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 67
      return-void
  .end method
  
***************
*** 172,177 ****
--- 182,188 ----
      .parameter "name"
  
      .prologue
+     .line 55
      const-class v0, Lcom/android/internal/telephony/DataCallState$SetupResult;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 187,192 ****
--- 198,204 ----
      .locals 1
  
      .prologue
+     .line 55
      sget-object v0, Lcom/android/internal/telephony/DataCallState$SetupResult;->$VALUES:[Lcom/android/internal/telephony/DataCallState$SetupResult;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/DataCallState$SetupResult;->clone()Ljava/lang/Object;
***************
*** 204,209 ****
--- 216,222 ----
      .locals 2
  
      .prologue
+     .line 71
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
