*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataConnection$DcInactiveState.smali	2012-03-23 19:37:36.705444405 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataConnection$DcInactiveState.smali.original	2012-03-23 19:34:46.155334429 -0500
***************
*** 32,45 ****
--- 32,49 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 738
      iput-object p1, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
  
+     .line 739
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mConnectionParams:Lcom/android/internal/telephony/DataConnection$ConnectionParams;
  
+     .line 740
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mFailCause:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 741
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mDisconnectParams:Lcom/android/internal/telephony/DataConnection$DisconnectParams;
  
      return-void
***************
*** 51,56 ****
--- 55,61 ----
      .parameter "x1"
  
      .prologue
+     .line 738
      invoke-direct {p0, p1}, Lcom/android/internal/telephony/DataConnection$DcInactiveState;-><init>(Lcom/android/internal/telephony/DataConnection;)V
  
      return-void
***************
*** 62,67 ****
--- 67,73 ----
      .locals 3
  
      .prologue
+     .line 758
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      iget v1, v0, Lcom/android/internal/telephony/DataConnection;->mTag:I
***************
*** 70,75 ****
--- 76,82 ----
  
      iput v1, v0, Lcom/android/internal/telephony/DataConnection;->mTag:I
  
+     .line 767
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mConnectionParams:Lcom/android/internal/telephony/DataConnection$ConnectionParams;
  
      if-eqz v0, :cond_0
***************
*** 78,83 ****
--- 85,91 ----
  
      if-eqz v0, :cond_0
  
+     .line 769
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mConnectionParams:Lcom/android/internal/telephony/DataConnection$ConnectionParams;
***************
*** 87,97 ****
--- 95,107 ----
      #calls: Lcom/android/internal/telephony/DataConnection;->notifyConnectCompleted(Lcom/android/internal/telephony/DataConnection$ConnectionParams;Lcom/android/internal/telephony/DataConnection$FailCause;)V
      invoke-static {v0, v1, v2}, Lcom/android/internal/telephony/DataConnection;->access$700(Lcom/android/internal/telephony/DataConnection;Lcom/android/internal/telephony/DataConnection$ConnectionParams;Lcom/android/internal/telephony/DataConnection$FailCause;)V
  
+     .line 771
      :cond_0
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mDisconnectParams:Lcom/android/internal/telephony/DataConnection$DisconnectParams;
  
      if-eqz v0, :cond_1
  
+     .line 773
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mDisconnectParams:Lcom/android/internal/telephony/DataConnection$DisconnectParams;
***************
*** 99,109 ****
--- 109,121 ----
      #calls: Lcom/android/internal/telephony/DataConnection;->notifyDisconnectCompleted(Lcom/android/internal/telephony/DataConnection$DisconnectParams;)V
      invoke-static {v0, v1}, Lcom/android/internal/telephony/DataConnection;->access$1000(Lcom/android/internal/telephony/DataConnection;Lcom/android/internal/telephony/DataConnection$DisconnectParams;)V
  
+     .line 775
      :cond_1
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/DataConnection;->clearSettings()V
  
+     .line 776
      return-void
  .end method
  
***************
*** 113,124 ****
--- 125,140 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 781
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mConnectionParams:Lcom/android/internal/telephony/DataConnection$ConnectionParams;
  
+     .line 782
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mFailCause:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 783
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mDisconnectParams:Lcom/android/internal/telephony/DataConnection$DisconnectParams;
  
+     .line 784
      return-void
  .end method
  
***************
*** 127,142 ****
--- 143,162 ----
      .parameter "msg"
  
      .prologue
+     .line 790
      iget v2, p1, Landroid/os/Message;->what:I
  
      sparse-switch v2, :sswitch_data_0
  
+     .line 823
      const/4 v1, 0x0
  
+     .line 826
      .local v1, retVal:Z
      :goto_0
      return v1
  
+     .line 793
      .end local v1           #retVal:Z
      :sswitch_0
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
***************
*** 145,150 ****
--- 165,171 ----
  
      invoke-virtual {v2, v3}, Lcom/android/internal/telephony/DataConnection;->log(Ljava/lang/String;)V
  
+     .line 795
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      iget-object v2, v2, Lcom/android/internal/telephony/DataConnection;->mAc:Lcom/android/internal/util/AsyncChannel;
***************
*** 153,169 ****
--- 174,194 ----
  
      invoke-virtual {v2, p1, v3}, Lcom/android/internal/util/AsyncChannel;->replyToMessage(Landroid/os/Message;I)V
  
+     .line 796
      const/4 v1, 0x1
  
+     .line 797
      .restart local v1       #retVal:Z
      goto :goto_0
  
+     .line 800
      .end local v1           #retVal:Z
      :sswitch_1
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Lcom/android/internal/telephony/DataConnection$ConnectionParams;
  
+     .line 801
      .local v0, cp:Lcom/android/internal/telephony/DataConnection$ConnectionParams;
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
***************
*** 171,176 ****
--- 196,202 ----
  
      iput v2, v0, Lcom/android/internal/telephony/DataConnection$ConnectionParams;->tag:I
  
+     .line 803
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 200,205 ****
--- 226,232 ----
  
      invoke-virtual {v2, v3}, Lcom/android/internal/telephony/DataConnection;->log(Ljava/lang/String;)V
  
+     .line 806
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      const/4 v3, 0x1
***************
*** 207,216 ****
--- 234,245 ----
      #setter for: Lcom/android/internal/telephony/DataConnection;->mRefCount:I
      invoke-static {v2, v3}, Lcom/android/internal/telephony/DataConnection;->access$502(Lcom/android/internal/telephony/DataConnection;I)I
  
+     .line 807
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      invoke-virtual {v2, v0}, Lcom/android/internal/telephony/DataConnection;->onConnect(Lcom/android/internal/telephony/DataConnection$ConnectionParams;)V
  
+     .line 808
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      iget-object v3, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
***************
*** 223,233 ****
--- 252,265 ----
      #calls: Lcom/android/internal/telephony/DataConnection;->transitionTo(Lcom/android/internal/util/IState;)V
      invoke-static {v2, v3}, Lcom/android/internal/telephony/DataConnection;->access$1200(Lcom/android/internal/telephony/DataConnection;Lcom/android/internal/util/IState;)V
  
+     .line 809
      const/4 v1, 0x1
  
+     .line 810
      .restart local v1       #retVal:Z
      goto :goto_0
  
+     .line 813
      .end local v0           #cp:Lcom/android/internal/telephony/DataConnection$ConnectionParams;
      .end local v1           #retVal:Z
      :sswitch_2
***************
*** 237,242 ****
--- 269,275 ----
  
      invoke-virtual {v2, v3}, Lcom/android/internal/telephony/DataConnection;->log(Ljava/lang/String;)V
  
+     .line 814
      iget-object v3, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      iget-object v2, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
***************
*** 246,256 ****
--- 279,292 ----
      #calls: Lcom/android/internal/telephony/DataConnection;->notifyDisconnectCompleted(Lcom/android/internal/telephony/DataConnection$DisconnectParams;)V
      invoke-static {v3, v2}, Lcom/android/internal/telephony/DataConnection;->access$1000(Lcom/android/internal/telephony/DataConnection;Lcom/android/internal/telephony/DataConnection$DisconnectParams;)V
  
+     .line 815
      const/4 v1, 0x1
  
+     .line 816
      .restart local v1       #retVal:Z
      goto :goto_0
  
+     .line 790
      :sswitch_data_0
      .sparse-switch
          0x40000 -> :sswitch_1
***************
*** 266,279 ****
--- 302,319 ----
      .parameter "retryOverride"
  
      .prologue
+     .line 746
      iput-object p1, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mConnectionParams:Lcom/android/internal/telephony/DataConnection$ConnectionParams;
  
+     .line 747
      iput-object p2, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mFailCause:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 748
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->this$0:Lcom/android/internal/telephony/DataConnection;
  
      iput p3, v0, Lcom/android/internal/telephony/DataConnection;->mRetryOverride:I
  
+     .line 749
      return-void
  .end method
  
***************
*** 282,288 ****
--- 322,330 ----
      .parameter "dp"
  
      .prologue
+     .line 753
      iput-object p1, p0, Lcom/android/internal/telephony/DataConnection$DcInactiveState;->mDisconnectParams:Lcom/android/internal/telephony/DataConnection$DisconnectParams;
  
+     .line 754
      return-void
  .end method
