*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataConnection$FailCause.smali	2012-03-23 19:37:39.902502703 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataConnection$FailCause.smali.original	2012-03-23 19:34:46.205335341 -0500
***************
*** 110,115 ****
--- 110,116 ----
  
      const/4 v8, 0x0
  
+     .line 103
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "NONE"
***************
*** 118,123 ****
--- 119,125 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->NONE:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 107
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "OPERATOR_BARRED"
***************
*** 126,131 ****
--- 128,134 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->OPERATOR_BARRED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 108
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "INSUFFICIENT_RESOURCES"
***************
*** 136,141 ****
--- 139,145 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->INSUFFICIENT_RESOURCES:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 109
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "MISSING_UNKNOWN_APN"
***************
*** 146,151 ****
--- 150,156 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->MISSING_UNKNOWN_APN:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 110
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "UNKNOWN_PDP_ADDRESS_TYPE"
***************
*** 158,163 ****
--- 163,169 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->UNKNOWN_PDP_ADDRESS_TYPE:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 111
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "USER_AUTHENTICATION"
***************
*** 170,175 ****
--- 176,182 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->USER_AUTHENTICATION:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 112
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "ACTIVATION_REJECT_GGSN"
***************
*** 182,187 ****
--- 189,195 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->ACTIVATION_REJECT_GGSN:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 113
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "ACTIVATION_REJECT_UNSPECIFIED"
***************
*** 194,199 ****
--- 202,208 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->ACTIVATION_REJECT_UNSPECIFIED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 114
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "SERVICE_OPTION_NOT_SUPPORTED"
***************
*** 204,209 ****
--- 213,219 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->SERVICE_OPTION_NOT_SUPPORTED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 115
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "SERVICE_OPTION_NOT_SUBSCRIBED"
***************
*** 216,221 ****
--- 226,232 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->SERVICE_OPTION_NOT_SUBSCRIBED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 116
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "SERVICE_OPTION_OUT_OF_ORDER"
***************
*** 228,233 ****
--- 239,245 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->SERVICE_OPTION_OUT_OF_ORDER:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 117
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "NSAPI_IN_USE"
***************
*** 240,245 ****
--- 252,258 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->NSAPI_IN_USE:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 118
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "ONLY_IPV4_ALLOWED"
***************
*** 252,257 ****
--- 265,271 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->ONLY_IPV4_ALLOWED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 119
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "ONLY_IPV6_ALLOWED"
***************
*** 264,269 ****
--- 278,284 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->ONLY_IPV6_ALLOWED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 120
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "ONLY_SINGLE_BEARER_ALLOWED"
***************
*** 276,281 ****
--- 291,297 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->ONLY_SINGLE_BEARER_ALLOWED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 121
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "PROTOCOL_ERRORS"
***************
*** 288,293 ****
--- 304,310 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->PROTOCOL_ERRORS:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 125
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "REGISTRATION_FAIL"
***************
*** 300,305 ****
--- 317,323 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->REGISTRATION_FAIL:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 126
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "GPRS_REGISTRATION_FAIL"
***************
*** 312,317 ****
--- 330,336 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->GPRS_REGISTRATION_FAIL:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 127
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "SIGNAL_LOST"
***************
*** 324,329 ****
--- 343,349 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->SIGNAL_LOST:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 128
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "PREF_RADIO_TECH_CHANGED"
***************
*** 336,341 ****
--- 356,362 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->PREF_RADIO_TECH_CHANGED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 129
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "RADIO_POWER_OFF"
***************
*** 348,353 ****
--- 369,375 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->RADIO_POWER_OFF:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 130
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "TETHERED_CALL_ACTIVE"
***************
*** 360,365 ****
--- 382,388 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->TETHERED_CALL_ACTIVE:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 131
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "ERROR_UNSPECIFIED"
***************
*** 372,377 ****
--- 395,401 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->ERROR_UNSPECIFIED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 135
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "UNKNOWN"
***************
*** 384,389 ****
--- 408,414 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->UNKNOWN:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 136
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "RADIO_NOT_AVAILABLE"
***************
*** 396,401 ****
--- 421,427 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->RADIO_NOT_AVAILABLE:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 137
      new-instance v4, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      const-string v5, "UNACCEPTABLE_NETWORK_PARAMETER"
***************
*** 408,413 ****
--- 434,440 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->UNACCEPTABLE_NETWORK_PARAMETER:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 102
      const/16 v4, 0x1a
  
      new-array v4, v4, [Lcom/android/internal/telephony/DataConnection$FailCause;
***************
*** 560,571 ****
--- 587,600 ----
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->$VALUES:[Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 142
      new-instance v4, Ljava/util/HashMap;
  
      invoke-direct {v4}, Ljava/util/HashMap;-><init>()V
  
      sput-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->sErrorCodeToFailCauseMap:Ljava/util/HashMap;
  
+     .line 143
      invoke-static {}, Lcom/android/internal/telephony/DataConnection$FailCause;->values()[Lcom/android/internal/telephony/DataConnection$FailCause;
  
      move-result-object v0
***************
*** 582,587 ****
--- 611,617 ----
  
      aget-object v1, v0, v2
  
+     .line 144
      .local v1, fc:Lcom/android/internal/telephony/DataConnection$FailCause;
      sget-object v4, Lcom/android/internal/telephony/DataConnection$FailCause;->sErrorCodeToFailCauseMap:Ljava/util/HashMap;
  
***************
*** 595,604 ****
--- 625,636 ----
  
      invoke-virtual {v4, v5, v1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 143
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 146
      .end local v1           #fc:Lcom/android/internal/telephony/DataConnection$FailCause;
      :cond_0
      return-void
***************
*** 616,625 ****
--- 648,660 ----
      .end annotation
  
      .prologue
+     .line 148
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 149
      iput p3, p0, Lcom/android/internal/telephony/DataConnection$FailCause;->mErrorCode:I
  
+     .line 150
      return-void
  .end method
  
***************
*** 628,633 ****
--- 663,669 ----
      .parameter "errorCode"
  
      .prologue
+     .line 176
      sget-object v1, Lcom/android/internal/telephony/DataConnection$FailCause;->sErrorCodeToFailCauseMap:Ljava/util/HashMap;
  
      invoke-static {p0}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 640,650 ****
--- 676,689 ----
  
      check-cast v0, Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 177
      .local v0, fc:Lcom/android/internal/telephony/DataConnection$FailCause;
      if-nez v0, :cond_0
  
+     .line 178
      sget-object v0, Lcom/android/internal/telephony/DataConnection$FailCause;->UNKNOWN:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 180
      :cond_0
      return-object v0
  .end method
***************
*** 654,659 ****
--- 693,699 ----
      .parameter "name"
  
      .prologue
+     .line 102
      const-class v0, Lcom/android/internal/telephony/DataConnection$FailCause;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 669,674 ****
--- 709,715 ----
      .locals 1
  
      .prologue
+     .line 102
      sget-object v0, Lcom/android/internal/telephony/DataConnection$FailCause;->$VALUES:[Lcom/android/internal/telephony/DataConnection$FailCause;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/DataConnection$FailCause;->clone()Ljava/lang/Object;
***************
*** 686,691 ****
--- 727,733 ----
      .locals 1
  
      .prologue
+     .line 153
      iget v0, p0, Lcom/android/internal/telephony/DataConnection$FailCause;->mErrorCode:I
  
      return v0
***************
*** 695,700 ****
--- 737,743 ----
      .locals 1
  
      .prologue
+     .line 165
      sget-object v0, Lcom/android/internal/telephony/DataConnection$FailCause;->OPERATOR_BARRED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
      if-eq p0, v0, :cond_0
***************
*** 759,764 ****
--- 802,808 ----
      .locals 1
  
      .prologue
+     .line 157
      sget-object v0, Lcom/android/internal/telephony/DataConnection$FailCause;->OPERATOR_BARRED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
      if-eq p0, v0, :cond_0
