*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataConnection$DisconnectParams.smali	2012-03-23 19:37:41.119524896 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataConnection$DisconnectParams.smali.original	2012-03-23 19:34:46.240335979 -0500
***************
*** 29,39 ****
--- 29,43 ----
      .parameter "onCompletedMsg"
  
      .prologue
+     .line 89
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 90
      iput-object p1, p0, Lcom/android/internal/telephony/DataConnection$DisconnectParams;->reason:Ljava/lang/String;
  
+     .line 91
      iput-object p2, p0, Lcom/android/internal/telephony/DataConnection$DisconnectParams;->onCompletedMsg:Landroid/os/Message;
  
+     .line 92
      return-void
  .end method
