*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataConnectionTracker.smali	2012-03-23 19:37:40.886520646 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/DataConnectionTracker.smali.original	2012-03-23 19:34:46.236335905 -0500
***************
*** 296,310 ****
      .locals 2
  
      .prologue
      const-string v0, "com.android.internal.telephony"
  
      sput-object v0, Lcom/android/internal/telephony/DataConnectionTracker;->ACTION_DATA_CONNECTION_TRACKER_MESSENGER:Ljava/lang/String;
  
      const-string v0, "EXTRA_MESSENGER"
  
      sput-object v0, Lcom/android/internal/telephony/DataConnectionTracker;->EXTRA_MESSENGER:Ljava/lang/String;
  
!     const-string v0, "persist.radio.apn_delay"
  
      const/16 v1, 0x1388
  
--- 296,313 ----
      .locals 2
  
      .prologue
+     .line 97
      const-string v0, "com.android.internal.telephony"
  
      sput-object v0, Lcom/android/internal/telephony/DataConnectionTracker;->ACTION_DATA_CONNECTION_TRACKER_MESSENGER:Ljava/lang/String;
  
+     .line 99
      const-string v0, "EXTRA_MESSENGER"
  
      sput-object v0, Lcom/android/internal/telephony/DataConnectionTracker;->EXTRA_MESSENGER:Ljava/lang/String;
  
!     .line 157
!     const-string/jumbo v0, "persist.radio.apn_delay"
  
      const/16 v1, 0x1388
  
***************
*** 314,319 ****
--- 317,323 ----
  
      sput v0, Lcom/android/internal/telephony/DataConnectionTracker;->APN_DELAY_MILLIS:I
  
+     .line 172
      const/4 v0, 0x1
  
      sput-boolean v0, Lcom/android/internal/telephony/DataConnectionTracker;->sPolicyDataEnabled:Z
***************
*** 334,381 ****
--- 338,399 ----
  
      const/4 v9, 0x0
  
+     .line 491
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
  
+     .line 160
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataEnabledLock:Ljava/lang/Object;
  
+     .line 164
      iput-boolean v8, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mInternalDataEnabled:Z
  
+     .line 169
      iput-boolean v8, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mUserDataEnabled:Z
  
+     .line 174
      const/16 v0, 0x8
  
      new-array v0, v0, [Z
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
+     .line 176
      iput v9, p0, Lcom/android/internal/telephony/DataConnectionTracker;->enabledCount:I
  
+     .line 179
      const-string v0, "default"
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mRequestedApnType:Ljava/lang/String;
  
+     .line 237
      iput v9, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mFailDataSetupCounter:I
  
+     .line 239
      sget-object v0, Lcom/android/internal/telephony/DataConnection$FailCause;->ERROR_UNSPECIFIED:Lcom/android/internal/telephony/DataConnection$FailCause;
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mFailDataSetupFailCause:Lcom/android/internal/telephony/DataConnection$FailCause;
  
+     .line 245
      sget-object v0, Lcom/android/internal/telephony/DataConnectionTracker$Activity;->NONE:Lcom/android/internal/telephony/DataConnectionTracker$Activity;
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mActivity:Lcom/android/internal/telephony/DataConnectionTracker$Activity;
  
+     .line 246
      sget-object v0, Lcom/android/internal/telephony/DataConnectionTracker$State;->IDLE:Lcom/android/internal/telephony/DataConnectionTracker$State;
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mState:Lcom/android/internal/telephony/DataConnectionTracker$State;
  
+     .line 247
      iput-object v10, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataConnectionTracker:Landroid/os/Handler;
  
+     .line 253
      iput-boolean v9, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mNetStatPollEnabled:Z
  
+     .line 255
      new-instance v0, Lcom/android/internal/telephony/DataConnectionTracker$TxRxSum;
  
      move-object v1, p0
***************
*** 386,391 ****
--- 404,410 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataStallTxRxSum:Lcom/android/internal/telephony/DataConnectionTracker$TxRxSum;
  
+     .line 257
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v0
***************
*** 394,455 ****
--- 413,492 ----
  
      iput v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataStallAlarmTag:I
  
+     .line 259
      iput-object v10, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataStallAlarmIntent:Landroid/app/PendingIntent;
  
+     .line 263
      iput v9, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mNoRecvPollCount:I
  
+     .line 266
      iput-boolean v9, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mIsWifiConnected:Z
  
+     .line 269
      iput-object v10, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mReconnectIntent:Landroid/app/PendingIntent;
  
+     .line 275
      iput-boolean v9, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mAutoAttachOnCreation:Z
  
+     .line 280
      iput-boolean v8, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mIsScreenOn:Z
  
+     .line 283
      new-instance v0, Ljava/util/concurrent/atomic/AtomicInteger;
  
      invoke-direct {v0, v9}, Ljava/util/concurrent/atomic/AtomicInteger;-><init>(I)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mUniqueIdGenerator:Ljava/util/concurrent/atomic/AtomicInteger;
  
+     .line 286
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataConnections:Ljava/util/HashMap;
  
+     .line 290
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataConnectionAsyncChannels:Ljava/util/HashMap;
  
+     .line 294
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mApnToDataConnectionId:Ljava/util/HashMap;
  
+     .line 304
      iput-object v10, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mAllApns:Ljava/util/ArrayList;
  
+     .line 307
      iput-object v10, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPreferredApn:Lcom/android/internal/telephony/ApnSetting;
  
+     .line 310
      iput-boolean v9, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mIsPsRestricted:Z
  
+     .line 313
      iput-boolean v9, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mIsDisposed:Z
  
+     .line 315
      new-instance v0, Lcom/android/internal/telephony/DataConnectionTracker$1;
  
      invoke-direct {v0, p0}, Lcom/android/internal/telephony/DataConnectionTracker$1;-><init>(Lcom/android/internal/telephony/DataConnectionTracker;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mIntentReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 492
      iput-object p1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
+     .line 494
      new-instance v6, Landroid/content/IntentFilter;
  
      invoke-direct {v6}, Landroid/content/IntentFilter;-><init>()V
  
+     .line 495
      .local v6, filter:Landroid/content/IntentFilter;
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->getActionIntentReconnectAlarm()Ljava/lang/String;
  
***************
*** 457,482 ****
--- 494,525 ----
  
      invoke-virtual {v6, v0}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 496
      const-string v0, "android.intent.action.SCREEN_ON"
  
      invoke-virtual {v6, v0}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 497
      const-string v0, "android.intent.action.SCREEN_OFF"
  
      invoke-virtual {v6, v0}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 498
      const-string v0, "android.net.wifi.STATE_CHANGE"
  
      invoke-virtual {v6, v0}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 499
      const-string v0, "android.net.wifi.WIFI_STATE_CHANGED"
  
      invoke-virtual {v6, v0}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 500
      const-string v0, "com.android.internal.telephony.dataconnectiontracker.intent_set_fail_data_setup_counter"
  
      invoke-virtual {v6, v0}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 502
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
***************
*** 487,493 ****
  
      move-result-object v0
  
!     const-string v1, "mobile_data"
  
      invoke-static {v0, v1, v8}, Landroid/provider/Settings$Secure;->getInt(Landroid/content/ContentResolver;Ljava/lang/String;I)I
  
--- 530,536 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "mobile_data"
  
      invoke-static {v0, v1, v8}, Landroid/provider/Settings$Secure;->getInt(Landroid/content/ContentResolver;Ljava/lang/String;I)I
  
***************
*** 500,505 ****
--- 543,549 ----
      :goto_0
      iput-boolean v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mUserDataEnabled:Z
  
+     .line 507
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
***************
*** 512,520 ****
  
      invoke-virtual {v0, v1, v6, v10, v2}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;Ljava/lang/String;Landroid/os/Handler;)Landroid/content/Intent;
  
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
!     const-string v1, "net.def_data_on_boot"
  
      invoke-static {v1, v8}, Landroid/os/SystemProperties;->getBoolean(Ljava/lang/String;Z)Z
  
--- 556,565 ----
  
      invoke-virtual {v0, v1, v6, v10, v2}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;Ljava/lang/String;Landroid/os/Handler;)Landroid/content/Intent;
  
+     .line 512
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
!     const-string/jumbo v1, "net.def_data_on_boot"
  
      invoke-static {v1, v8}, Landroid/os/SystemProperties;->getBoolean(Ljava/lang/String;Z)Z
  
***************
*** 522,539 ****
--- 567,587 ----
  
      aput-boolean v1, v0, v9
  
+     .line 514
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
      aget-boolean v0, v0, v9
  
      if-eqz v0, :cond_0
  
+     .line 515
      iget v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->enabledCount:I
  
      add-int/lit8 v0, v0, 0x1
  
      iput v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->enabledCount:I
  
+     .line 518
      :cond_0
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
***************
*** 545,550 ****
--- 593,599 ----
  
      move-result-object v7
  
+     .line 519
      .local v7, sp:Landroid/content/SharedPreferences;
      const-string v0, "disabled_on_boot_key"
  
***************
*** 554,559 ****
--- 603,609 ----
  
      iput-boolean v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mAutoAttachOnCreation:Z
  
+     .line 522
      new-instance v0, Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;
  
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 562,567 ****
--- 612,618 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataRoamingSettingObserver:Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;
  
+     .line 523
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataRoamingSettingObserver:Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;
  
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 572,583 ****
--- 623,636 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;->register(Landroid/content/Context;)V
  
+     .line 524
      return-void
  
      .end local v7           #sp:Landroid/content/SharedPreferences;
      :cond_1
      move v0, v9
  
+     .line 502
      goto :goto_0
  .end method
  
***************
*** 586,591 ****
--- 639,645 ----
      .parameter "x0"
  
      .prologue
+     .line 59
      invoke-direct {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->handleDataOnRoamingChange()V
  
      return-void
***************
*** 595,600 ****
--- 649,655 ----
      .locals 1
  
      .prologue
+     .line 612
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/PhoneBase;->getServiceState()Landroid/telephony/ServiceState;
***************
*** 607,620 ****
--- 662,678 ----
  
      if-eqz v0, :cond_1
  
+     .line 613
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->getDataOnRoamingEnabled()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 614
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->resetAllRetryCounts()V
  
+     .line 616
      :cond_0
      const v0, 0x4200b
  
***************
*** 624,629 ****
--- 682,688 ----
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->sendMessage(Landroid/os/Message;)Z
  
+     .line 618
      :cond_1
      return-void
  .end method
***************
*** 634,639 ****
--- 693,699 ----
      .parameter "apnId"
  
      .prologue
+     .line 887
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {p0, p2}, Lcom/android/internal/telephony/DataConnectionTracker;->apnIdToType(I)Ljava/lang/String;
***************
*** 644,649 ****
--- 704,710 ----
  
      invoke-virtual {v0, p1, v1, v2}, Lcom/android/internal/telephony/PhoneBase;->notifyDataConnection(Ljava/lang/String;Ljava/lang/String;Lcom/android/internal/telephony/Phone$DataState;)V
  
+     .line 888
      return-void
  .end method
  
***************
*** 653,658 ****
--- 714,720 ----
      .parameter "apnId"
  
      .prologue
+     .line 869
      sget-object v0, Lcom/android/internal/telephony/DataConnectionTracker$2;->$SwitchMap$com$android$internal$telephony$DataConnectionTracker$State:[I
  
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mState:Lcom/android/internal/telephony/DataConnectionTracker$State;
***************
*** 665,674 ****
--- 727,738 ----
  
      packed-switch v0, :pswitch_data_0
  
+     .line 883
      :goto_0
      :pswitch_0
      return-void
  
+     .line 875
      :pswitch_1
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
***************
*** 682,687 ****
--- 746,752 ----
  
      goto :goto_0
  
+     .line 879
      :pswitch_2
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
***************
*** 693,698 ****
--- 758,764 ----
  
      invoke-virtual {v0, p1, v1, v2}, Lcom/android/internal/telephony/PhoneBase;->notifyDataConnection(Ljava/lang/String;Ljava/lang/String;Lcom/android/internal/telephony/Phone$DataState;)V
  
+     .line 880
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {p0, p2}, Lcom/android/internal/telephony/DataConnectionTracker;->apnIdToType(I)Ljava/lang/String;
***************
*** 705,710 ****
--- 771,777 ----
  
      goto :goto_0
  
+     .line 869
      nop
  
      :pswitch_data_0
***************
*** 725,732 ****
--- 792,801 ----
      .parameter "id"
  
      .prologue
+     .line 810
      packed-switch p1, :pswitch_data_0
  
+     .line 828
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 753,803 ****
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
      const-string v0, "default"
  
      :goto_0
      return-object v0
  
      :pswitch_0
      const-string v0, "default"
  
      goto :goto_0
  
      :pswitch_1
!     const-string v0, "mms"
  
      goto :goto_0
  
      :pswitch_2
!     const-string v0, "supl"
  
      goto :goto_0
  
      :pswitch_3
      const-string v0, "dun"
  
      goto :goto_0
  
      :pswitch_4
      const-string v0, "hipri"
  
      goto :goto_0
  
      :pswitch_5
      const-string v0, "ims"
  
      goto :goto_0
  
      :pswitch_6
      const-string v0, "fota"
  
      goto :goto_0
  
      :pswitch_7
      const-string v0, "cbs"
  
      goto :goto_0
  
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
--- 822,882 ----
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 829
      const-string v0, "default"
  
      :goto_0
      return-object v0
  
+     .line 812
      :pswitch_0
      const-string v0, "default"
  
      goto :goto_0
  
+     .line 814
      :pswitch_1
!     const-string/jumbo v0, "mms"
  
      goto :goto_0
  
+     .line 816
      :pswitch_2
!     const-string/jumbo v0, "supl"
  
      goto :goto_0
  
+     .line 818
      :pswitch_3
      const-string v0, "dun"
  
      goto :goto_0
  
+     .line 820
      :pswitch_4
      const-string v0, "hipri"
  
      goto :goto_0
  
+     .line 822
      :pswitch_5
      const-string v0, "ims"
  
      goto :goto_0
  
+     .line 824
      :pswitch_6
      const-string v0, "fota"
  
      goto :goto_0
  
+     .line 826
      :pswitch_7
      const-string v0, "cbs"
  
      goto :goto_0
  
+     .line 810
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
***************
*** 816,821 ****
--- 895,901 ----
      .parameter "type"
  
      .prologue
+     .line 788
      const-string v0, "default"
  
      invoke-static {p1, v0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
***************
*** 824,836 ****
  
      if-eqz v0, :cond_0
  
      const/4 v0, 0x0
  
      :goto_0
      return v0
  
      :cond_0
!     const-string v0, "mms"
  
      invoke-static {p1, v0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
  
--- 904,919 ----
  
      if-eqz v0, :cond_0
  
+     .line 789
      const/4 v0, 0x0
  
+     .line 805
      :goto_0
      return v0
  
+     .line 790
      :cond_0
!     const-string/jumbo v0, "mms"
  
      invoke-static {p1, v0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
  
***************
*** 838,849 ****
  
      if-eqz v0, :cond_1
  
      const/4 v0, 0x1
  
      goto :goto_0
  
      :cond_1
!     const-string v0, "supl"
  
      invoke-static {p1, v0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
  
--- 921,934 ----
  
      if-eqz v0, :cond_1
  
+     .line 791
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 792
      :cond_1
!     const-string/jumbo v0, "supl"
  
      invoke-static {p1, v0}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
  
***************
*** 851,860 ****
--- 936,947 ----
  
      if-eqz v0, :cond_2
  
+     .line 793
      const/4 v0, 0x2
  
      goto :goto_0
  
+     .line 794
      :cond_2
      const-string v0, "dun"
  
***************
*** 864,873 ****
--- 951,962 ----
  
      if-eqz v0, :cond_3
  
+     .line 795
      const/4 v0, 0x3
  
      goto :goto_0
  
+     .line 796
      :cond_3
      const-string v0, "hipri"
  
***************
*** 877,886 ****
--- 966,977 ----
  
      if-eqz v0, :cond_4
  
+     .line 797
      const/4 v0, 0x4
  
      goto :goto_0
  
+     .line 798
      :cond_4
      const-string v0, "ims"
  
***************
*** 890,899 ****
--- 981,992 ----
  
      if-eqz v0, :cond_5
  
+     .line 799
      const/4 v0, 0x5
  
      goto :goto_0
  
+     .line 800
      :cond_5
      const-string v0, "fota"
  
***************
*** 903,912 ****
--- 996,1007 ----
  
      if-eqz v0, :cond_6
  
+     .line 801
      const/4 v0, 0x6
  
      goto :goto_0
  
+     .line 802
      :cond_6
      const-string v0, "cbs"
  
***************
*** 916,925 ****
--- 1011,1022 ----
  
      if-eqz v0, :cond_7
  
+     .line 803
      const/4 v0, 0x7
  
      goto :goto_0
  
+     .line 805
      :cond_7
      const/4 v0, -0x1
  
***************
*** 930,941 ****
--- 1027,1040 ----
      .locals 3
  
      .prologue
+     .line 537
      new-instance v0, Landroid/content/Intent;
  
      sget-object v1, Lcom/android/internal/telephony/DataConnectionTracker;->ACTION_DATA_CONNECTION_TRACKER_MESSENGER:Ljava/lang/String;
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 538
      .local v0, intent:Landroid/content/Intent;
      sget-object v1, Lcom/android/internal/telephony/DataConnectionTracker;->EXTRA_MESSENGER:Ljava/lang/String;
  
***************
*** 945,950 ****
--- 1044,1050 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 539
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v1}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
***************
*** 953,958 ****
--- 1053,1059 ----
  
      invoke-virtual {v1, v0}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
  
+     .line 540
      return-void
  .end method
  
***************
*** 961,977 ****
--- 1062,1082 ----
      .parameter "cause"
  
      .prologue
+     .line 1105
      const v1, 0x4201e
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 1106
      .local v0, msg:Landroid/os/Message;
      iput-object p1, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 1107
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->sendMessage(Landroid/os/Message;)Z
  
+     .line 1108
      return-void
  .end method
  
***************
*** 986,991 ****
--- 1091,1097 ----
  
      const/4 v2, 0x0
  
+     .line 964
      monitor-enter p0
  
      :try_start_0
***************
*** 1015,1037 ****
--- 1121,1147 ----
  
      invoke-virtual {p0, v4}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 965
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/DataConnectionTracker;->apnTypeToId(Ljava/lang/String;)I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v0
  
+     .line 966
      .local v0, id:I
      const/4 v4, -0x1
  
      if-ne v0, v4, :cond_1
  
+     .line 981
      :cond_0
      :goto_0
      monitor-exit p0
  
      return v1
  
+     .line 969
      :cond_1
      :try_start_1
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnIdEnabled(I)Z
***************
*** 1040,1049 ****
--- 1150,1161 ----
  
      if-eqz v4, :cond_0
  
+     .line 970
      const/4 v1, 0x0
  
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->setEnabled(IZ)V
  
+     .line 971
      const-string v1, "default"
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnTypeActive(Ljava/lang/String;)Z
***************
*** 1052,1057 ****
--- 1164,1170 ----
  
      if-eqz v1, :cond_3
  
+     .line 972
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
      const/4 v4, 0x0
***************
*** 1064,1081 ****
--- 1177,1198 ----
  
      move v1, v2
  
+     .line 973
      goto :goto_0
  
      :cond_2
      move v1, v3
  
+     .line 975
      goto :goto_0
  
      :cond_3
      move v1, v3
  
+     .line 978
      goto :goto_0
  
+     .line 964
      .end local v0           #id:I
      :catchall_0
      move-exception v1
***************
*** 1089,1094 ****
--- 1206,1212 ----
      .locals 4
  
      .prologue
+     .line 527
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataConnectionAsyncChannels:Ljava/util/HashMap;
  
      invoke-virtual {v2}, Ljava/util/HashMap;->values()Ljava/util/Collection;
***************
*** 1113,1133 ****
--- 1231,1255 ----
  
      check-cast v0, Lcom/android/internal/telephony/DataConnectionAc;
  
+     .line 528
      .local v0, dcac:Lcom/android/internal/telephony/DataConnectionAc;
      invoke-virtual {v0}, Lcom/android/internal/telephony/DataConnectionAc;->disconnect()V
  
      goto :goto_0
  
+     .line 530
      .end local v0           #dcac:Lcom/android/internal/telephony/DataConnectionAc;
      :cond_0
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataConnectionAsyncChannels:Ljava/util/HashMap;
  
      invoke-virtual {v2}, Ljava/util/HashMap;->clear()V
  
+     .line 531
      const/4 v2, 0x1
  
      iput-boolean v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mIsDisposed:Z
  
+     .line 532
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v2}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
***************
*** 1138,1143 ****
--- 1260,1266 ----
  
      invoke-virtual {v2, v3}, Landroid/content/Context;->unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  
+     .line 533
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataRoamingSettingObserver:Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;
  
      iget-object v3, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 1148,1153 ****
--- 1271,1277 ----
  
      invoke-virtual {v2, v3}, Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;->unregister(Landroid/content/Context;)V
  
+     .line 534
      return-void
  .end method
  
***************
*** 1158,1163 ****
--- 1282,1288 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 926
      monitor-enter p0
  
      :try_start_0
***************
*** 1167,1184 ****
--- 1292,1313 ----
  
      move-result v0
  
+     .line 927
      .local v0, id:I
      const/4 v2, -0x1
  
      if-ne v0, v2, :cond_0
  
+     .line 928
      const/4 v1, 0x3
  
+     .line 946
      :goto_0
      monitor-exit p0
  
      return v1
  
+     .line 932
      :cond_0
      :try_start_1
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 1241,1260 ****
  
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnTypeAvailable(Ljava/lang/String;)Z
  
      move-result v2
  
      if-nez v2, :cond_1
  
!     const-string v1, "type not available"
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
      const/4 v1, 0x2
  
      goto :goto_0
  
      :cond_1
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnIdEnabled(I)Z
  
--- 1370,1393 ----
  
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 936
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnTypeAvailable(Ljava/lang/String;)Z
  
      move-result v2
  
      if-nez v2, :cond_1
  
!     .line 937
!     const-string/jumbo v1, "type not available"
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 938
      const/4 v1, 0x2
  
      goto :goto_0
  
+     .line 941
      :cond_1
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnIdEnabled(I)Z
  
***************
*** 1262,1271 ****
--- 1395,1406 ----
  
      if-eqz v2, :cond_2
  
+     .line 942
      const/4 v1, 0x0
  
      goto :goto_0
  
+     .line 944
      :cond_2
      const/4 v2, 0x1
  
***************
*** 1275,1280 ****
--- 1410,1416 ----
  
      goto :goto_0
  
+     .line 926
      .end local v0           #id:I
      :catchall_0
      move-exception v1
***************
*** 1288,1322 ****
      .locals 5
  
      .prologue
      iget-object v3, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v3}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
  
      move-result-object v1
  
      .local v1, c:Landroid/content/Context;
      invoke-virtual {v1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v3
  
!     const-string v4, "tether_dun_apn"
  
      invoke-static {v3, v4}, Landroid/provider/Settings$Secure;->getString(Landroid/content/ContentResolver;Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v0
  
      .local v0, apnData:Ljava/lang/String;
      invoke-static {v0}, Lcom/android/internal/telephony/ApnSetting;->fromString(Ljava/lang/String;)Lcom/android/internal/telephony/ApnSetting;
  
      move-result-object v2
  
      .local v2, dunSetting:Lcom/android/internal/telephony/ApnSetting;
      if-eqz v2, :cond_0
  
      .end local v2           #dunSetting:Lcom/android/internal/telephony/ApnSetting;
      :goto_0
      return-object v2
  
      .restart local v2       #dunSetting:Lcom/android/internal/telephony/ApnSetting;
      :cond_0
      invoke-virtual {v1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
--- 1424,1464 ----
      .locals 5
  
      .prologue
+     .line 558
      iget-object v3, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v3}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
  
      move-result-object v1
  
+     .line 559
      .local v1, c:Landroid/content/Context;
      invoke-virtual {v1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v3
  
!     const-string/jumbo v4, "tether_dun_apn"
  
      invoke-static {v3, v4}, Landroid/provider/Settings$Secure;->getString(Landroid/content/ContentResolver;Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 561
      .local v0, apnData:Ljava/lang/String;
      invoke-static {v0}, Lcom/android/internal/telephony/ApnSetting;->fromString(Ljava/lang/String;)Lcom/android/internal/telephony/ApnSetting;
  
      move-result-object v2
  
+     .line 562
      .local v2, dunSetting:Lcom/android/internal/telephony/ApnSetting;
      if-eqz v2, :cond_0
  
+     .line 565
      .end local v2           #dunSetting:Lcom/android/internal/telephony/ApnSetting;
      :goto_0
      return-object v2
  
+     .line 564
      .restart local v2       #dunSetting:Lcom/android/internal/telephony/ApnSetting;
      :cond_0
      invoke-virtual {v1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 1329,1334 ****
--- 1471,1477 ----
  
      move-result-object v0
  
+     .line 565
      invoke-static {v0}, Lcom/android/internal/telephony/ApnSetting;->fromString(Ljava/lang/String;)Lcom/android/internal/telephony/ApnSetting;
  
      move-result-object v2
***************
*** 1347,1363 ****
--- 1490,1510 ----
      .parameter "apnType"
  
      .prologue
+     .line 581
      const/4 v0, 0x0
  
+     .line 582
      .local v0, result:Ljava/lang/String;
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mActiveApn:Lcom/android/internal/telephony/ApnSetting;
  
      if-eqz v1, :cond_0
  
+     .line 583
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mActiveApn:Lcom/android/internal/telephony/ApnSetting;
  
      iget-object v0, v1, Lcom/android/internal/telephony/ApnSetting;->apn:Ljava/lang/String;
  
+     .line 585
      :cond_0
      return-object v0
  .end method
***************
*** 1366,1389 ****
--- 1513,1541 ----
      .locals 3
  
      .prologue
+     .line 570
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mActiveApn:Lcom/android/internal/telephony/ApnSetting;
  
      if-eqz v1, :cond_0
  
+     .line 571
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mActiveApn:Lcom/android/internal/telephony/ApnSetting;
  
      iget-object v0, v1, Lcom/android/internal/telephony/ApnSetting;->types:[Ljava/lang/String;
  
+     .line 576
      .local v0, result:[Ljava/lang/String;
      :goto_0
      return-object v0
  
+     .line 573
      .end local v0           #result:[Ljava/lang/String;
      :cond_0
      const/4 v1, 0x1
  
      new-array v0, v1, [Ljava/lang/String;
  
+     .line 574
      .restart local v0       #result:[Ljava/lang/String;
      const/4 v1, 0x0
  
***************
*** 1398,1403 ****
--- 1550,1556 ----
      .locals 1
  
      .prologue
+     .line 543
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mActivity:Lcom/android/internal/telephony/DataConnectionTracker$Activity;
  
      return-object v0
***************
*** 1407,1416 ****
--- 1560,1571 ----
      .locals 3
  
      .prologue
+     .line 770
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataEnabledLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 771
      :try_start_0
      iget-boolean v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mInternalDataEnabled:Z
  
***************
*** 1430,1441 ****
--- 1585,1598 ----
  
      const/4 v0, 0x1
  
+     .line 773
      .local v0, result:Z
      :goto_0
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 774
      if-nez v0, :cond_0
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 1458,1472 ****
--- 1615,1632 ----
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 775
      :cond_0
      return v0
  
+     .line 771
      .end local v0           #result:Z
      :cond_1
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 773
      :catchall_0
      move-exception v1
  
***************
*** 1484,1489 ****
--- 1644,1650 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 604
      :try_start_0
      iget-object v3, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
***************
*** 1495,1500 ****
--- 1656,1662 ----
  
      move-result-object v0
  
+     .line 605
      .local v0, resolver:Landroid/content/ContentResolver;
      const-string v3, "data_roaming"
  
***************
*** 1508,1521 ****
--- 1670,1686 ----
  
      const/4 v2, 0x1
  
+     .line 607
      .end local v0           #resolver:Landroid/content/ContentResolver;
      :cond_0
      :goto_0
      return v2
  
+     .line 606
      :catch_0
      move-exception v1
  
+     .line 607
      .local v1, snfe:Landroid/provider/Settings$SettingNotFoundException;
      goto :goto_0
  .end method
***************
*** 1525,1534 ****
--- 1690,1701 ----
      .parameter "apnType"
  
      .prologue
+     .line 846
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/DataConnectionTracker;->apnTypeToId(Ljava/lang/String;)I
  
      move-result v1
  
+     .line 847
      .local v1, id:I
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnIdEnabled(I)Z
  
***************
*** 1536,1541 ****
--- 1703,1709 ----
  
      if-eqz v2, :cond_0
  
+     .line 849
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataConnectionAsyncChannels:Ljava/util/HashMap;
  
      const/4 v3, 0x0
***************
*** 1550,1560 ****
--- 1718,1730 ----
  
      check-cast v0, Lcom/android/internal/telephony/DataConnectionAc;
  
+     .line 850
      .local v0, dcac:Lcom/android/internal/telephony/DataConnectionAc;
      invoke-virtual {v0}, Lcom/android/internal/telephony/DataConnectionAc;->getLinkCapabilitiesSync()Landroid/net/LinkCapabilities;
  
      move-result-object v2
  
+     .line 852
      .end local v0           #dcac:Lcom/android/internal/telephony/DataConnectionAc;
      :goto_0
      return-object v2
***************
*** 1572,1581 ****
--- 1742,1753 ----
      .parameter "apnType"
  
      .prologue
+     .line 834
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/DataConnectionTracker;->apnTypeToId(Ljava/lang/String;)I
  
      move-result v1
  
+     .line 836
      .local v1, id:I
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnIdEnabled(I)Z
  
***************
*** 1583,1588 ****
--- 1755,1761 ----
  
      if-eqz v2, :cond_0
  
+     .line 838
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataConnectionAsyncChannels:Ljava/util/HashMap;
  
      const/4 v3, 0x0
***************
*** 1597,1607 ****
--- 1770,1782 ----
  
      check-cast v0, Lcom/android/internal/telephony/DataConnectionAc;
  
+     .line 839
      .local v0, dcac:Lcom/android/internal/telephony/DataConnectionAc;
      invoke-virtual {v0}, Lcom/android/internal/telephony/DataConnectionAc;->getLinkPropertiesSync()Landroid/net/LinkProperties;
  
      move-result-object v2
  
+     .line 841
      .end local v0           #dcac:Lcom/android/internal/telephony/DataConnectionAc;
      :goto_0
      return-object v2
***************
*** 1619,1624 ****
--- 1794,1800 ----
      .parameter "forDefault"
  
      .prologue
+     .line 1152
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v1}, Lcom/android/internal/telephony/PhoneBase;->getServiceState()Landroid/telephony/ServiceState;
***************
*** 1629,1634 ****
--- 1805,1811 ----
  
      move-result v0
  
+     .line 1154
      .local v0, rt:I
      const/4 v1, 0x4
  
***************
*** 1658,1677 ****
  
      if-ne v0, v1, :cond_1
  
      :cond_0
!     const-string v1, "ro.cdma.data_retry_config"
  
      invoke-static {v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
      :goto_0
      return-object v1
  
      :cond_1
      if-eqz p1, :cond_2
  
!     const-string v1, "ro.gsm.data_retry_config"
  
      invoke-static {v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
  
--- 1835,1858 ----
  
      if-ne v0, v1, :cond_1
  
+     .line 1162
      :cond_0
!     const-string/jumbo v1, "ro.cdma.data_retry_config"
  
      invoke-static {v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
+     .line 1168
      :goto_0
      return-object v1
  
+     .line 1165
      :cond_1
      if-eqz p1, :cond_2
  
!     .line 1166
!     const-string/jumbo v1, "ro.gsm.data_retry_config"
  
      invoke-static {v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
  
***************
*** 1679,1686 ****
  
      goto :goto_0
  
      :cond_2
!     const-string v1, "ro.gsm.2nd_data_retry_config"
  
      invoke-static {v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
  
--- 1860,1868 ----
  
      goto :goto_0
  
+     .line 1168
      :cond_2
!     const-string/jumbo v1, "ro.gsm.2nd_data_retry_config"
  
      invoke-static {v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
  
***************
*** 1704,1713 ****
--- 1886,1897 ----
  
      const/4 v7, 0x1
  
+     .line 653
      iget v8, p1, Landroid/os/Message;->what:I
  
      sparse-switch v8, :sswitch_data_0
  
+     .line 757
      const-string v7, "DATA"
  
      new-instance v8, Ljava/lang/StringBuilder;
***************
*** 1730,1739 ****
--- 1914,1925 ----
  
      invoke-static {v7, v8}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 760
      :cond_0
      :goto_0
      return-void
  
+     .line 655
      :sswitch_0
      new-instance v7, Ljava/lang/StringBuilder;
  
***************
*** 1755,1764 ****
--- 1941,1952 ----
  
      invoke-virtual {p0, v7}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 656
      iget-object v2, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v2, Lcom/android/internal/telephony/DataConnectionAc;
  
+     .line 657
      .local v2, dcac:Lcom/android/internal/telephony/DataConnectionAc;
      iget-object v7, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataConnectionAsyncChannels:Ljava/util/HashMap;
  
***************
*** 1774,1783 ****
--- 1962,1973 ----
  
      invoke-virtual {v7, v8}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 658
      invoke-virtual {v2}, Lcom/android/internal/telephony/DataConnectionAc;->disconnected()V
  
      goto :goto_0
  
+     .line 662
      .end local v2           #dcac:Lcom/android/internal/telephony/DataConnectionAc;
      :sswitch_1
      iget v7, p1, Landroid/os/Message;->arg1:I
***************
*** 1788,1796 ****
--- 1978,1988 ----
  
      goto :goto_0
  
+     .line 666
      :sswitch_2
      const/4 v5, 0x0
  
+     .line 667
      .local v5, reason:Ljava/lang/String;
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 1798,1814 ****
--- 1990,2009 ----
  
      if-eqz v7, :cond_1
  
+     .line 668
      iget-object v5, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      .end local v5           #reason:Ljava/lang/String;
      check-cast v5, Ljava/lang/String;
  
+     .line 670
      .restart local v5       #reason:Ljava/lang/String;
      :cond_1
      invoke-virtual {p0, v5}, Lcom/android/internal/telephony/DataConnectionTracker;->onTrySetupData(Ljava/lang/String;)Z
  
      goto :goto_0
  
+     .line 674
      .end local v5           #reason:Ljava/lang/String;
      :sswitch_3
      iget v7, p1, Landroid/os/Message;->arg1:I
***************
*** 1817,1822 ****
--- 2012,2018 ----
  
      goto :goto_0
  
+     .line 678
      :sswitch_4
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->getDataOnRoamingEnabled()Z
  
***************
*** 1824,1856 ****
--- 2020,2059 ----
  
      if-nez v7, :cond_2
  
+     .line 679
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->resetAllRetryCounts()V
  
+     .line 681
      :cond_2
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->onRoamingOff()V
  
      goto :goto_0
  
+     .line 685
      :sswitch_5
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->onRoamingOn()V
  
      goto :goto_0
  
+     .line 689
      :sswitch_6
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->onRadioAvailable()V
  
      goto :goto_0
  
+     .line 693
      :sswitch_7
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->onRadioOffOrNotAvailable()V
  
      goto :goto_0
  
+     .line 697
      :sswitch_8
      iget v7, p1, Landroid/os/Message;->arg1:I
  
      iput v7, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mCidActive:I
  
+     .line 698
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v7, Landroid/os/AsyncResult;
***************
*** 1859,1864 ****
--- 2062,2068 ----
  
      goto :goto_0
  
+     .line 702
      :sswitch_9
      new-instance v7, Ljava/lang/StringBuilder;
  
***************
*** 1880,1885 ****
--- 2084,2090 ----
  
      invoke-virtual {p0, v7}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 703
      iget v8, p1, Landroid/os/Message;->arg1:I
  
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
***************
*** 1890,1905 ****
--- 2095,2113 ----
  
      goto/16 :goto_0
  
+     .line 707
      :sswitch_a
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->onVoiceCallStarted()V
  
      goto/16 :goto_0
  
+     .line 711
      :sswitch_b
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->onVoiceCallEnded()V
  
      goto/16 :goto_0
  
+     .line 715
      :sswitch_c
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 1909,1919 ****
--- 2117,2129 ----
  
      goto/16 :goto_0
  
+     .line 719
      :sswitch_d
      iget v8, p1, Landroid/os/Message;->arg1:I
  
      if-nez v8, :cond_3
  
+     .line 720
      .local v6, tearDown:Z
      :goto_1
      iget v8, p1, Landroid/os/Message;->arg2:I
***************
*** 1930,1937 ****
--- 2140,2149 ----
      :cond_3
      move v6, v7
  
+     .line 719
      goto :goto_1
  
+     .line 724
      :sswitch_e
      iget v8, p1, Landroid/os/Message;->arg1:I
  
***************
*** 1939,1944 ****
--- 2151,2157 ----
  
      move v3, v7
  
+     .line 725
      .local v3, enabled:Z
      :goto_2
      invoke-virtual {p0, v3}, Lcom/android/internal/telephony/DataConnectionTracker;->onSetInternalDataEnabled(Z)V
***************
*** 1949,1961 ****
--- 2162,2177 ----
      :cond_4
      move v3, v6
  
+     .line 724
      goto :goto_2
  
+     .line 729
      :sswitch_f
      const-string v7, "EVENT_RESET_DONE"
  
      invoke-virtual {p0, v7}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 730
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v7, Landroid/os/AsyncResult;
***************
*** 1964,1969 ****
--- 2180,2186 ----
  
      goto/16 :goto_0
  
+     .line 734
      :sswitch_10
      iget v8, p1, Landroid/os/Message;->arg1:I
  
***************
*** 1971,1976 ****
--- 2188,2194 ----
  
      move v3, v7
  
+     .line 735
      .restart local v3       #enabled:Z
      :goto_3
      new-instance v7, Ljava/lang/StringBuilder;
***************
*** 1993,1998 ****
--- 2211,2217 ----
  
      invoke-virtual {p0, v7}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 736
      invoke-virtual {p0, v3}, Lcom/android/internal/telephony/DataConnectionTracker;->onSetUserDataEnabled(Z)V
  
      goto/16 :goto_0
***************
*** 2001,2008 ****
--- 2220,2229 ----
      :cond_5
      move v3, v6
  
+     .line 734
      goto :goto_3
  
+     .line 740
      :sswitch_11
      iget v8, p1, Landroid/os/Message;->arg1:I
  
***************
*** 2010,2015 ****
--- 2231,2237 ----
  
      move v4, v7
  
+     .line 741
      .local v4, met:Z
      :goto_4
      new-instance v7, Ljava/lang/StringBuilder;
***************
*** 2032,2044 ****
--- 2254,2269 ----
  
      invoke-virtual {p0, v7}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 742
      invoke-virtual {p1}, Landroid/os/Message;->getData()Landroid/os/Bundle;
  
      move-result-object v1
  
+     .line 743
      .local v1, bundle:Landroid/os/Bundle;
      if-eqz v1, :cond_0
  
+     .line 744
      const-string v7, "apnType"
  
      invoke-virtual {v1, v7}, Landroid/os/Bundle;->get(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 2047,2055 ****
--- 2272,2282 ----
  
      check-cast v0, Ljava/lang/String;
  
+     .line 745
      .local v0, apnType:Ljava/lang/String;
      if-eqz v0, :cond_0
  
+     .line 746
      invoke-virtual {p0, v0, v4}, Lcom/android/internal/telephony/DataConnectionTracker;->onSetDependencyMet(Ljava/lang/String;Z)V
  
      goto/16 :goto_0
***************
*** 2060,2067 ****
--- 2287,2296 ----
      :cond_6
      move v4, v6
  
+     .line 740
      goto :goto_4
  
+     .line 752
      :sswitch_12
      iget v8, p1, Landroid/os/Message;->arg1:I
  
***************
*** 2069,2074 ****
--- 2298,2304 ----
  
      move v3, v7
  
+     .line 753
      .restart local v3       #enabled:Z
      :goto_5
      invoke-virtual {p0, v3}, Lcom/android/internal/telephony/DataConnectionTracker;->onSetPolicyDataEnabled(Z)V
***************
*** 2079,2086 ****
--- 2309,2318 ----
      :cond_7
      move v3, v6
  
+     .line 752
      goto :goto_5
  
+     .line 653
      :sswitch_data_0
      .sparse-switch
          0x11004 -> :sswitch_0
***************
*** 2110,2121 ****
--- 2342,2355 ----
      .parameter "id"
  
      .prologue
+     .line 909
      monitor-enter p0
  
      const/4 v0, -0x1
  
      if-eq p1, v0, :cond_0
  
+     .line 910
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
***************
*** 2123,2128 ****
--- 2357,2363 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 912
      :goto_0
      monitor-exit p0
  
***************
*** 2133,2138 ****
--- 2368,2374 ----
  
      goto :goto_0
  
+     .line 909
      :catchall_0
      move-exception v0
  
***************
*** 2150,2155 ****
--- 2386,2392 ----
  
      const/4 v2, 0x0
  
+     .line 548
      const-string v3, "dun"
  
      invoke-virtual {v3, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
***************
*** 2158,2170 ****
--- 2395,2410 ----
  
      if-eqz v3, :cond_2
  
+     .line 549
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->fetchDunApn()Lcom/android/internal/telephony/ApnSetting;
  
      move-result-object v0
  
+     .line 550
      .local v0, dunApn:Lcom/android/internal/telephony/ApnSetting;
      if-eqz v0, :cond_2
  
+     .line 551
      iget-object v3, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mActiveApn:Lcom/android/internal/telephony/ApnSetting;
  
      if-eqz v3, :cond_1
***************
*** 2185,2190 ****
--- 2425,2431 ----
  
      if-eqz v3, :cond_1
  
+     .line 554
      .end local v0           #dunApn:Lcom/android/internal/telephony/ApnSetting;
      :cond_0
      :goto_0
***************
*** 2194,2201 ****
--- 2435,2444 ----
      :cond_1
      move v1, v2
  
+     .line 551
      goto :goto_0
  
+     .line 554
      .end local v0           #dunApn:Lcom/android/internal/telephony/ApnSetting;
      :cond_2
      iget-object v3, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mActiveApn:Lcom/android/internal/telephony/ApnSetting;
***************
*** 2224,2233 ****
--- 2467,2479 ----
      .parameter "apnType"
  
      .prologue
+     .line 901
      if-nez p1, :cond_0
  
+     .line 902
      const/4 v0, 0x0
  
+     .line 904
      :goto_0
      return v0
  
***************
*** 2256,2265 ****
--- 2502,2513 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 450
      iget-object v1, p1, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
      if-eqz v1, :cond_0
  
+     .line 451
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2282,2308 ****
--- 2530,2562 ----
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 465
      :goto_0
      return v0
  
+     .line 454
      :cond_0
      iget v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mFailDataSetupCounter:I
  
      if-gtz v1, :cond_1
  
+     .line 455
      const-string v0, "isDataSetupCompleteOk return true"
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 456
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 458
      :cond_1
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mFailDataSetupFailCause:Lcom/android/internal/telephony/DataConnection$FailCause;
  
      iput-object v1, p1, Landroid/os/AsyncResult;->result:Ljava/lang/Object;
  
+     .line 460
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2337,2342 ****
--- 2591,2597 ----
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 464
      iget v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mFailDataSetupCounter:I
  
      add-int/lit8 v1, v1, -0x1
***************
*** 2353,2362 ****
--- 2608,2619 ----
      .locals 3
  
      .prologue
+     .line 780
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataEnabledLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 781
      :try_start_0
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
***************
*** 2377,2388 ****
--- 2634,2647 ----
      :cond_0
      const/4 v0, 0x1
  
+     .line 782
      .local v0, result:Z
      :goto_0
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 783
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2403,2416 ****
--- 2662,2678 ----
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 784
      return v0
  
+     .line 781
      .end local v0           #result:Z
      :cond_1
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 782
      :catchall_0
      move-exception v1
  
***************
*** 2433,2438 ****
--- 2695,2701 ----
      .parameter "reason"
  
      .prologue
+     .line 858
      const/4 v0, 0x0
  
      .local v0, id:I
***************
*** 2441,2452 ****
--- 2704,2717 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 859
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
      aget-boolean v1, v1, v0
  
      if-eqz v1, :cond_0
  
+     .line 860
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->apnIdToType(I)Ljava/lang/String;
***************
*** 2455,2468 ****
--- 2720,2736 ----
  
      invoke-virtual {v1, p1, v2}, Lcom/android/internal/telephony/PhoneBase;->notifyDataConnection(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 858
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 863
      :cond_1
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/DataConnectionTracker;->notifyOffApnsOfAvailability(Ljava/lang/String;)V
  
+     .line 864
      return-void
  .end method
  
***************
*** 2471,2481 ****
      .parameter "reason"
  
      .prologue
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "notifyOffApnsOfAvailability - reason= "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2739,2750 ----
      .parameter "reason"
  
      .prologue
+     .line 892
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "notifyOffApnsOfAvailability - reason= "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 2491,2496 ****
--- 2760,2766 ----
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 893
      const/4 v0, 0x0
  
      .local v0, id:I
***************
*** 2499,2517 ****
--- 2769,2791 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 894
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnIdEnabled(I)Z
  
      move-result v1
  
      if-nez v1, :cond_0
  
+     .line 895
      invoke-direct {p0, p1, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->notifyApnIdDisconnected(Ljava/lang/String;I)V
  
+     .line 893
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 898
      :cond_1
      return-void
  .end method
***************
*** 2521,2526 ****
--- 2795,2801 ----
      .parameter "intent"
  
      .prologue
+     .line 482
      const v1, 0x42011
  
      invoke-virtual {p1}, Landroid/content/Intent;->getAction()Ljava/lang/String;
***************
*** 2531,2536 ****
--- 2806,2812 ----
  
      move-result-object v0
  
+     .line 483
      .local v0, msg:Landroid/os/Message;
      const-string v1, "data.stall.alram.tag"
  
***************
*** 2542,2549 ****
--- 2818,2827 ----
  
      iput v1, v0, Landroid/os/Message;->arg1:I
  
+     .line 484
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->sendMessage(Landroid/os/Message;)Z
  
+     .line 485
      return-void
  .end method
  
***************
*** 2554,2565 ****
      .prologue
      const/4 v4, 0x0
  
!     const-string v2, "reason"
  
      invoke-virtual {p1, v2}, Landroid/content/Intent;->getStringExtra(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
      .local v1, reason:Ljava/lang/String;
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mState:Lcom/android/internal/telephony/DataConnectionTracker$State;
  
--- 2832,2845 ----
      .prologue
      const/4 v4, 0x0
  
!     .line 469
!     const-string/jumbo v2, "reason"
  
      invoke-virtual {p1, v2}, Landroid/content/Intent;->getStringExtra(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
+     .line 470
      .local v1, reason:Ljava/lang/String;
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mState:Lcom/android/internal/telephony/DataConnectionTracker$State;
  
***************
*** 2567,2587 ****
--- 2847,2873 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 471
      const v2, 0x42018
  
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/DataConnectionTracker;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 472
      .local v0, msg:Landroid/os/Message;
      iput v4, v0, Landroid/os/Message;->arg1:I
  
+     .line 473
      iput v4, v0, Landroid/os/Message;->arg2:I
  
+     .line 474
      iput-object v1, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 475
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->sendMessage(Landroid/os/Message;)Z
  
+     .line 477
      .end local v0           #msg:Landroid/os/Message;
      :cond_0
      const v2, 0x42003
***************
*** 2592,2597 ****
--- 2878,2884 ----
  
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/DataConnectionTracker;->sendMessage(Landroid/os/Message;)Z
  
+     .line 478
      return-void
  .end method
  
***************
*** 2609,2619 ****
      .parameter "tag"
  
      .prologue
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v1, "onDataStallAlarm: not impleted tag="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2896,2907 ----
      .parameter "tag"
  
      .prologue
+     .line 648
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v1, "onDataStallAlarm: not impleted tag="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 2629,2634 ****
--- 2917,2923 ----
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->loge(Ljava/lang/String;)V
  
+     .line 649
      return-void
  .end method
  
***************
*** 2645,2650 ****
--- 2934,2940 ----
  
      const/4 v4, 0x1
  
+     .line 998
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2733,2742 ****
--- 3023,3035 ----
  
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 1003
      if-ne p2, v4, :cond_3
  
+     .line 1004
      monitor-enter p0
  
+     .line 1005
      :try_start_0
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
***************
*** 2744,2770 ****
--- 3037,3068 ----
  
      if-nez v2, :cond_0
  
+     .line 1006
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
      const/4 v3, 0x1
  
      aput-boolean v3, v2, p1
  
+     .line 1007
      iget v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->enabledCount:I
  
      add-int/lit8 v2, v2, 0x1
  
      iput v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->enabledCount:I
  
+     .line 1009
      :cond_0
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1010
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/DataConnectionTracker;->apnIdToType(I)Ljava/lang/String;
  
      move-result-object v1
  
+     .line 1011
      .local v1, type:Ljava/lang/String;
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->isApnTypeActive(Ljava/lang/String;)Z
  
***************
*** 2772,2786 ****
--- 3070,3088 ----
  
      if-nez v2, :cond_2
  
+     .line 1012
      iput-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mRequestedApnType:Ljava/lang/String;
  
+     .line 1013
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->onEnableNewApn()V
  
+     .line 1044
      .end local v1           #type:Ljava/lang/String;
      :cond_1
      :goto_0
      return-void
  
+     .line 1009
      :catchall_0
      move-exception v2
  
***************
*** 2791,2796 ****
--- 3093,3099 ----
  
      throw v2
  
+     .line 1015
      .restart local v1       #type:Ljava/lang/String;
      :cond_2
      const-string v2, "apnSwitched"
***************
*** 2799,2811 ****
--- 3102,3117 ----
  
      goto :goto_0
  
+     .line 1019
      .end local v1           #type:Ljava/lang/String;
      :cond_3
      const/4 v0, 0x0
  
+     .line 1020
      .local v0, didDisable:Z
      monitor-enter p0
  
+     .line 1021
      :try_start_2
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
***************
*** 2813,2851 ****
--- 3119,3165 ----
  
      if-eqz v2, :cond_4
  
+     .line 1022
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
      const/4 v3, 0x0
  
      aput-boolean v3, v2, p1
  
+     .line 1023
      iget v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->enabledCount:I
  
      add-int/lit8 v2, v2, -0x1
  
      iput v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->enabledCount:I
  
+     .line 1024
      const/4 v0, 0x1
  
+     .line 1026
      :cond_4
      monitor-exit p0
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_1
  
+     .line 1027
      if-eqz v0, :cond_1
  
      iget v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->enabledCount:I
  
      if-nez v2, :cond_1
  
+     .line 1028
      const-string v2, "dataDisabled"
  
      invoke-virtual {p0, v4, p1, v2}, Lcom/android/internal/telephony/DataConnectionTracker;->onCleanUpConnection(ZILjava/lang/String;)V
  
+     .line 1032
      const-string v2, "dataDisabled"
  
      invoke-direct {p0, v2, p1}, Lcom/android/internal/telephony/DataConnectionTracker;->notifyApnIdDisconnected(Ljava/lang/String;I)V
  
+     .line 1033
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->dataEnabled:[Z
  
      aget-boolean v2, v2, v5
***************
*** 2860,2873 ****
--- 3174,3190 ----
  
      if-nez v2, :cond_1
  
+     .line 1039
      const-string v2, "default"
  
      iput-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mRequestedApnType:Ljava/lang/String;
  
+     .line 1040
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->onEnableNewApn()V
  
      goto :goto_0
  
+     .line 1026
      :catchall_1
      move-exception v2
  
***************
*** 2883,2888 ****
--- 3200,3206 ----
      .locals 0
  
      .prologue
+     .line 1053
      return-void
  .end method
  
***************
*** 2897,2908 ****
--- 3215,3229 ----
      .parameter "ar"
  
      .prologue
+     .line 1063
      const-string v1, "EVENT_RESET_DONE"
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 1064
      const/4 v0, 0x0
  
+     .line 1065
      .local v0, reason:Ljava/lang/String;
      iget-object v1, p1, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
***************
*** 2910,2924 ****
--- 3231,3248 ----
  
      if-eqz v1, :cond_0
  
+     .line 1066
      iget-object v0, p1, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
      .end local v0           #reason:Ljava/lang/String;
      check-cast v0, Ljava/lang/String;
  
+     .line 1068
      .restart local v0       #reason:Ljava/lang/String;
      :cond_0
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->gotoIdleAndNotifyDataConnection(Ljava/lang/String;)V
  
+     .line 1069
      return-void
  .end method
  
***************
*** 2934,2939 ****
--- 3258,3264 ----
      .parameter "met"
  
      .prologue
+     .line 1132
      return-void
  .end method
  
***************
*** 2942,2982 ****
      .parameter "enabled"
  
      .prologue
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataEnabledLock:Ljava/lang/Object;
  
      monitor-enter v1
  
      :try_start_0
      iput-boolean p1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mInternalDataEnabled:Z
  
      if-eqz p1, :cond_0
  
!     const-string v0, "onSetInternalDataEnabled: changed to enabled, try to setup data call"
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->resetAllRetryCounts()V
  
      const-string v0, "dataEnabled"
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->onTrySetupData(Ljava/lang/String;)Z
  
      :goto_0
      monitor-exit v1
  
      return-void
  
      :cond_0
!     const-string v0, "onSetInternalDataEnabled: changed to disabled, cleanUpAllConnections"
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->cleanUpAllConnections(Ljava/lang/String;)V
  
      goto :goto_0
  
      :catchall_0
      move-exception v0
  
--- 3267,3318 ----
      .parameter "enabled"
  
      .prologue
+     .line 1091
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataEnabledLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 1092
      :try_start_0
      iput-boolean p1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mInternalDataEnabled:Z
  
+     .line 1093
      if-eqz p1, :cond_0
  
!     .line 1094
!     const-string/jumbo v0, "onSetInternalDataEnabled: changed to enabled, try to setup data call"
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 1095
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->resetAllRetryCounts()V
  
+     .line 1096
      const-string v0, "dataEnabled"
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->onTrySetupData(Ljava/lang/String;)Z
  
+     .line 1101
      :goto_0
      monitor-exit v1
  
+     .line 1102
      return-void
  
+     .line 1098
      :cond_0
!     const-string/jumbo v0, "onSetInternalDataEnabled: changed to disabled, cleanUpAllConnections"
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 1099
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->cleanUpAllConnections(Ljava/lang/String;)V
  
      goto :goto_0
  
+     .line 1101
      :catchall_0
      move-exception v0
  
***************
*** 2992,3033 ****
--- 3328,3380 ----
      .parameter "enabled"
  
      .prologue
+     .line 1135
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataEnabledLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 1136
      :try_start_0
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->getAnyDataEnabled()Z
  
      move-result v0
  
+     .line 1137
      .local v0, prevEnabled:Z
      sget-boolean v1, Lcom/android/internal/telephony/DataConnectionTracker;->sPolicyDataEnabled:Z
  
      if-eq v1, p1, :cond_0
  
+     .line 1138
      sput-boolean p1, Lcom/android/internal/telephony/DataConnectionTracker;->sPolicyDataEnabled:Z
  
+     .line 1139
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->getAnyDataEnabled()Z
  
      move-result v1
  
      if-eq v0, v1, :cond_0
  
+     .line 1140
      if-nez v0, :cond_1
  
+     .line 1141
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->resetAllRetryCounts()V
  
+     .line 1142
      const-string v1, "dataEnabled"
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->onTrySetupData(Ljava/lang/String;)Z
  
+     .line 1148
      :cond_0
      :goto_0
      monitor-exit v2
  
+     .line 1149
      return-void
  
+     .line 1144
      :cond_1
      const-string v1, "dataDisabled"
  
***************
*** 3035,3040 ****
--- 3382,3388 ----
  
      goto :goto_0
  
+     .line 1148
      .end local v0           #prevEnabled:Z
      :catchall_0
      move-exception v1
***************
*** 3051,3072 ****
--- 3399,3425 ----
      .parameter "enabled"
  
      .prologue
+     .line 1113
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataEnabledLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 1114
      :try_start_0
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->getAnyDataEnabled()Z
  
      move-result v0
  
+     .line 1115
      .local v0, prevEnabled:Z
      iget-boolean v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mUserDataEnabled:Z
  
      if-eq v1, p1, :cond_0
  
+     .line 1116
      iput-boolean p1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mUserDataEnabled:Z
  
+     .line 1117
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v1}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
***************
*** 3077,3083 ****
  
      move-result-object v3
  
!     const-string v4, "mobile_data"
  
      if-eqz p1, :cond_1
  
--- 3430,3436 ----
  
      move-result-object v3
  
!     const-string/jumbo v4, "mobile_data"
  
      if-eqz p1, :cond_1
  
***************
*** 3086,3116 ****
--- 3439,3477 ----
      :goto_0
      invoke-static {v3, v4, v1}, Landroid/provider/Settings$Secure;->putInt(Landroid/content/ContentResolver;Ljava/lang/String;I)Z
  
+     .line 1119
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->getAnyDataEnabled()Z
  
      move-result v1
  
      if-eq v0, v1, :cond_0
  
+     .line 1120
      if-nez v0, :cond_2
  
+     .line 1121
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->resetAllRetryCounts()V
  
+     .line 1122
      const-string v1, "dataEnabled"
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->onTrySetupData(Ljava/lang/String;)Z
  
+     .line 1128
      :cond_0
      :goto_1
      monitor-exit v2
  
+     .line 1129
      return-void
  
+     .line 1117
      :cond_1
      const/4 v1, 0x0
  
      goto :goto_0
  
+     .line 1124
      :cond_2
      const-string v1, "dataDisabled"
  
***************
*** 3118,3123 ****
--- 3479,3485 ----
  
      goto :goto_1
  
+     .line 1128
      .end local v0           #prevEnabled:Z
      :catchall_0
      move-exception v1
***************
*** 3142,3147 ****
--- 3504,3510 ----
      .locals 3
  
      .prologue
+     .line 1174
      iget-object v2, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mDataConnections:Ljava/util/HashMap;
  
      invoke-virtual {v2}, Ljava/util/HashMap;->values()Ljava/util/Collection;
***************
*** 3166,3176 ****
--- 3529,3541 ----
  
      check-cast v0, Lcom/android/internal/telephony/DataConnection;
  
+     .line 1175
      .local v0, dc:Lcom/android/internal/telephony/DataConnection;
      invoke-virtual {v0}, Lcom/android/internal/telephony/DataConnection;->resetRetryCount()V
  
      goto :goto_0
  
+     .line 1177
      .end local v0           #dc:Lcom/android/internal/telephony/DataConnection;
      :cond_0
      return-void
***************
*** 3184,3195 ****
--- 3549,3562 ----
      .parameter "enabled"
  
      .prologue
+     .line 592
      invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->getDataOnRoamingEnabled()Z
  
      move-result v1
  
      if-eq v1, p1, :cond_0
  
+     .line 593
      iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v1}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
***************
*** 3200,3205 ****
--- 3567,3573 ----
  
      move-result-object v0
  
+     .line 594
      .local v0, resolver:Landroid/content/ContentResolver;
      const-string v2, "data_roaming"
  
***************
*** 3210,3219 ****
--- 3578,3589 ----
      :goto_0
      invoke-static {v0, v2, v1}, Landroid/provider/Settings$Secure;->putInt(Landroid/content/ContentResolver;Ljava/lang/String;I)Z
  
+     .line 597
      .end local v0           #resolver:Landroid/content/ContentResolver;
      :cond_0
      return-void
  
+     .line 594
      .restart local v0       #resolver:Landroid/content/ContentResolver;
      :cond_1
      const/4 v1, 0x0
***************
*** 3227,3237 ****
      .parameter "enable"
  
      .prologue
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "setEnabled("
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 3597,3608 ----
      .parameter "enable"
  
      .prologue
+     .line 987
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "setEnabled("
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 3283,3297 ****
--- 3654,3671 ----
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 990
      const v1, 0x4200d
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 991
      .local v0, msg:Landroid/os/Message;
      iput p1, v0, Landroid/os/Message;->arg1:I
  
+     .line 992
      if-eqz p2, :cond_0
  
      const/4 v1, 0x1
***************
*** 3299,3308 ****
--- 3673,3685 ----
      :goto_0
      iput v1, v0, Landroid/os/Message;->arg2:I
  
+     .line 993
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->sendMessage(Landroid/os/Message;)Z
  
+     .line 994
      return-void
  
+     .line 992
      :cond_0
      const/4 v1, 0x0
  
***************
*** 3316,3326 ****
      .prologue
      const/4 v2, 0x1
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "setInternalDataEnabled("
  
      invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 3693,3704 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 1082
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "setInternalDataEnabled("
  
      invoke-virtual {v1, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 3342,3353 ****
--- 3720,3733 ----
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->log(Ljava/lang/String;)V
  
+     .line 1084
      const v1, 0x4201b
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/DataConnectionTracker;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 1085
      .local v0, msg:Landroid/os/Message;
      if-eqz p1, :cond_0
  
***************
*** 3356,3365 ****
--- 3736,3748 ----
      :goto_0
      iput v1, v0, Landroid/os/Message;->arg1:I
  
+     .line 1086
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker;->sendMessage(Landroid/os/Message;)Z
  
+     .line 1087
      return v2
  
+     .line 1085
      :cond_0
      const/4 v1, 0x0
  
