*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/EncodeException.smali	2012-03-23 19:37:36.831446702 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/EncodeException.smali.original	2012-03-23 19:34:46.157334465 -0500
***************
*** 8,15 ****
--- 8,17 ----
      .locals 0
  
      .prologue
+     .line 24
      invoke-direct {p0}, Ljava/lang/Exception;-><init>()V
  
+     .line 25
      return-void
  .end method
  
***************
*** 18,23 ****
--- 20,26 ----
      .parameter "c"
  
      .prologue
+     .line 32
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 44,49 ****
--- 47,53 ----
  
      invoke-direct {p0, v0}, Ljava/lang/Exception;-><init>(Ljava/lang/String;)V
  
+     .line 33
      return-void
  .end method
  
***************
*** 52,58 ****
--- 56,64 ----
      .parameter "s"
  
      .prologue
+     .line 28
      invoke-direct {p0, p1}, Ljava/lang/Exception;-><init>(Ljava/lang/String;)V
  
+     .line 29
      return-void
  .end method
