*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/GsmAlphabet$LanguagePairCount.smali	2012-03-23 19:37:42.071542256 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/GsmAlphabet$LanguagePairCount.smali.original	2012-03-23 19:34:46.255336252 -0500
***************
*** 34,47 ****
--- 34,51 ----
  
      const/4 v4, -0x1
  
+     .line 1007
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1008
      iput p1, p0, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->languageCode:I
  
+     .line 1009
      invoke-static {}, Lcom/android/internal/telephony/GsmAlphabet;->access$000()I
  
      move-result v1
  
+     .line 1010
      .local v1, maxSingleShiftCode:I
      add-int/lit8 v3, v1, 0x1
  
***************
*** 49,60 ****
--- 53,66 ----
  
      iput-object v3, p0, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
+     .line 1011
      add-int/lit8 v3, v1, 0x1
  
      new-array v3, v3, [I
  
      iput-object v3, p0, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->unencodableCounts:[I
  
+     .line 1014
      const/4 v0, 0x1
  
      .local v0, i:I
***************
*** 64,69 ****
--- 70,76 ----
      :goto_0
      if-gt v0, v1, :cond_1
  
+     .line 1015
      invoke-static {}, Lcom/android/internal/telephony/GsmAlphabet;->access$100()[I
  
      move-result-object v3
***************
*** 72,84 ****
--- 79,94 ----
  
      if-ne v3, v0, :cond_0
  
+     .line 1016
      add-int/lit8 v2, v2, 0x1
  
+     .line 1014
      :goto_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 1018
      :cond_0
      iget-object v3, p0, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
***************
*** 86,104 ****
--- 96,118 ----
  
      goto :goto_1
  
+     .line 1024
      :cond_1
      if-ne p1, v5, :cond_3
  
      if-lt v1, v5, :cond_3
  
+     .line 1025
      iget-object v3, p0, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
      aput v4, v3, v5
  
+     .line 1029
      :cond_2
      :goto_2
      return-void
  
+     .line 1026
      :cond_3
      const/4 v3, 0x3
  
***************
*** 106,111 ****
--- 120,126 ----
  
      if-lt v1, v6, :cond_2
  
+     .line 1027
      iget-object v3, p0, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
      aput v4, v3, v6
