*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/GsmAlphabet.smali	2012-03-23 19:37:42.020541326 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/GsmAlphabet.smali.original	2012-03-23 19:34:46.254336233 -0500
***************
*** 44,49 ****
--- 44,50 ----
      .locals 15
  
      .prologue
+     .line 1036
      const/16 v12, 0xe
  
      new-array v12, v12, [Ljava/lang/String;
***************
*** 74,92 ****
  
      const/4 v13, 0x4
  
!     const-string v14, "\u0981\u0982\u0983\u0985\u0986\u0987\u0988\u0989\u098a\u098b\n\u098c \r \u098f\u0990  \u0993\u0994\u0995\u0996\u0997\u0998\u0999\u099a\uffff\u099b\u099c\u099d\u099e !\u099f\u09a0\u09a1\u09a2\u09a3\u09a4)(\u09a5\u09a6,\u09a7.\u09a80123456789:; \u09aa\u09ab?\u09ac\u09ad\u09ae\u09af\u09b0 \u09b2   \u09b6\u09b7\u09b8\u09b9\u09bc\u09bd\u09be\u09bf\u09c0\u09c1\u09c2\u09c3\u09c4  \u09c7\u09c8  \u09cb\u09cc\u09cd\u09ceabcdefghijklmnopqrstuvwxyz\u09d7\u09dc\u09dd\u09f0\u09f1"
  
      aput-object v14, v12, v13
  
      const/4 v13, 0x5
  
!     const-string v14, "\u0a81\u0a82\u0a83\u0a85\u0a86\u0a87\u0a88\u0a89\u0a8a\u0a8b\n\u0a8c\u0a8d\r \u0a8f\u0a90\u0a91 \u0a93\u0a94\u0a95\u0a96\u0a97\u0a98\u0a99\u0a9a\uffff\u0a9b\u0a9c\u0a9d\u0a9e !\u0a9f\u0aa0\u0aa1\u0aa2\u0aa3\u0aa4)(\u0aa5\u0aa6,\u0aa7.\u0aa80123456789:; \u0aaa\u0aab?\u0aac\u0aad\u0aae\u0aaf\u0ab0 \u0ab2\u0ab3 \u0ab5\u0ab6\u0ab7\u0ab8\u0ab9\u0abc\u0abd\u0abe\u0abf\u0ac0\u0ac1\u0ac2\u0ac3\u0ac4\u0ac5 \u0ac7\u0ac8\u0ac9 \u0acb\u0acc\u0acd\u0ad0abcdefghijklmnopqrstuvwxyz\u0ae0\u0ae1\u0ae2\u0ae3\u0af1"
  
      aput-object v14, v12, v13
  
      const/4 v13, 0x6
  
!     const-string v14, "\u0901\u0902\u0903\u0905\u0906\u0907\u0908\u0909\u090a\u090b\n\u090c\u090d\r\u090e\u090f\u0910\u0911\u0912\u0913\u0914\u0915\u0916\u0917\u0918\u0919\u091a\uffff\u091b\u091c\u091d\u091e !\u091f\u0920\u0921\u0922\u0923\u0924)(\u0925\u0926,\u0927.\u09280123456789:;\u0929\u092a\u092b?\u092c\u092d\u092e\u092f\u0930\u0931\u0932\u0933\u0934\u0935\u0936\u0937\u0938\u0939\u093c\u093d\u093e\u093f\u0940\u0941\u0942\u0943\u0944\u0945\u0946\u0947\u0948\u0949\u094a\u094b\u094c\u094d\u0950abcdefghijklmnopqrstuvwxyz\u0972\u097b\u097c\u097e\u097f"
  
      aput-object v14, v12, v13
  
--- 75,93 ----
  
      const/4 v13, 0x4
  
!     const-string/jumbo v14, "\u0981\u0982\u0983\u0985\u0986\u0987\u0988\u0989\u098a\u098b\n\u098c \r \u098f\u0990  \u0993\u0994\u0995\u0996\u0997\u0998\u0999\u099a\uffff\u099b\u099c\u099d\u099e !\u099f\u09a0\u09a1\u09a2\u09a3\u09a4)(\u09a5\u09a6,\u09a7.\u09a80123456789:; \u09aa\u09ab?\u09ac\u09ad\u09ae\u09af\u09b0 \u09b2   \u09b6\u09b7\u09b8\u09b9\u09bc\u09bd\u09be\u09bf\u09c0\u09c1\u09c2\u09c3\u09c4  \u09c7\u09c8  \u09cb\u09cc\u09cd\u09ceabcdefghijklmnopqrstuvwxyz\u09d7\u09dc\u09dd\u09f0\u09f1"
  
      aput-object v14, v12, v13
  
      const/4 v13, 0x5
  
!     const-string/jumbo v14, "\u0a81\u0a82\u0a83\u0a85\u0a86\u0a87\u0a88\u0a89\u0a8a\u0a8b\n\u0a8c\u0a8d\r \u0a8f\u0a90\u0a91 \u0a93\u0a94\u0a95\u0a96\u0a97\u0a98\u0a99\u0a9a\uffff\u0a9b\u0a9c\u0a9d\u0a9e !\u0a9f\u0aa0\u0aa1\u0aa2\u0aa3\u0aa4)(\u0aa5\u0aa6,\u0aa7.\u0aa80123456789:; \u0aaa\u0aab?\u0aac\u0aad\u0aae\u0aaf\u0ab0 \u0ab2\u0ab3 \u0ab5\u0ab6\u0ab7\u0ab8\u0ab9\u0abc\u0abd\u0abe\u0abf\u0ac0\u0ac1\u0ac2\u0ac3\u0ac4\u0ac5 \u0ac7\u0ac8\u0ac9 \u0acb\u0acc\u0acd\u0ad0abcdefghijklmnopqrstuvwxyz\u0ae0\u0ae1\u0ae2\u0ae3\u0af1"
  
      aput-object v14, v12, v13
  
      const/4 v13, 0x6
  
!     const-string/jumbo v14, "\u0901\u0902\u0903\u0905\u0906\u0907\u0908\u0909\u090a\u090b\n\u090c\u090d\r\u090e\u090f\u0910\u0911\u0912\u0913\u0914\u0915\u0916\u0917\u0918\u0919\u091a\uffff\u091b\u091c\u091d\u091e !\u091f\u0920\u0921\u0922\u0923\u0924)(\u0925\u0926,\u0927.\u09280123456789:;\u0929\u092a\u092b?\u092c\u092d\u092e\u092f\u0930\u0931\u0932\u0933\u0934\u0935\u0936\u0937\u0938\u0939\u093c\u093d\u093e\u093f\u0940\u0941\u0942\u0943\u0944\u0945\u0946\u0947\u0948\u0949\u094a\u094b\u094c\u094d\u0950abcdefghijklmnopqrstuvwxyz\u0972\u097b\u097c\u097e\u097f"
  
      aput-object v14, v12, v13
  
***************
*** 104,116 ****
  
      const/16 v13, 0x9
  
!     const-string v14, "\u0b01\u0b02\u0b03\u0b05\u0b06\u0b07\u0b08\u0b09\u0b0a\u0b0b\n\u0b0c \r \u0b0f\u0b10  \u0b13\u0b14\u0b15\u0b16\u0b17\u0b18\u0b19\u0b1a\uffff\u0b1b\u0b1c\u0b1d\u0b1e !\u0b1f\u0b20\u0b21\u0b22\u0b23\u0b24)(\u0b25\u0b26,\u0b27.\u0b280123456789:; \u0b2a\u0b2b?\u0b2c\u0b2d\u0b2e\u0b2f\u0b30 \u0b32\u0b33 \u0b35\u0b36\u0b37\u0b38\u0b39\u0b3c\u0b3d\u0b3e\u0b3f\u0b40\u0b41\u0b42\u0b43\u0b44  \u0b47\u0b48  \u0b4b\u0b4c\u0b4d\u0b56abcdefghijklmnopqrstuvwxyz\u0b57\u0b60\u0b61\u0b62\u0b63"
  
      aput-object v14, v12, v13
  
      const/16 v13, 0xa
  
!     const-string v14, "\u0a01\u0a02\u0a03\u0a05\u0a06\u0a07\u0a08\u0a09\u0a0a \n  \r \u0a0f\u0a10  \u0a13\u0a14\u0a15\u0a16\u0a17\u0a18\u0a19\u0a1a\uffff\u0a1b\u0a1c\u0a1d\u0a1e !\u0a1f\u0a20\u0a21\u0a22\u0a23\u0a24)(\u0a25\u0a26,\u0a27.\u0a280123456789:; \u0a2a\u0a2b?\u0a2c\u0a2d\u0a2e\u0a2f\u0a30 \u0a32\u0a33 \u0a35\u0a36 \u0a38\u0a39\u0a3c \u0a3e\u0a3f\u0a40\u0a41\u0a42    \u0a47\u0a48  \u0a4b\u0a4c\u0a4d\u0a51abcdefghijklmnopqrstuvwxyz\u0a70\u0a71\u0a72\u0a73\u0a74"
  
      aput-object v14, v12, v13
  
--- 105,117 ----
  
      const/16 v13, 0x9
  
!     const-string/jumbo v14, "\u0b01\u0b02\u0b03\u0b05\u0b06\u0b07\u0b08\u0b09\u0b0a\u0b0b\n\u0b0c \r \u0b0f\u0b10  \u0b13\u0b14\u0b15\u0b16\u0b17\u0b18\u0b19\u0b1a\uffff\u0b1b\u0b1c\u0b1d\u0b1e !\u0b1f\u0b20\u0b21\u0b22\u0b23\u0b24)(\u0b25\u0b26,\u0b27.\u0b280123456789:; \u0b2a\u0b2b?\u0b2c\u0b2d\u0b2e\u0b2f\u0b30 \u0b32\u0b33 \u0b35\u0b36\u0b37\u0b38\u0b39\u0b3c\u0b3d\u0b3e\u0b3f\u0b40\u0b41\u0b42\u0b43\u0b44  \u0b47\u0b48  \u0b4b\u0b4c\u0b4d\u0b56abcdefghijklmnopqrstuvwxyz\u0b57\u0b60\u0b61\u0b62\u0b63"
  
      aput-object v14, v12, v13
  
      const/16 v13, 0xa
  
!     const-string/jumbo v14, "\u0a01\u0a02\u0a03\u0a05\u0a06\u0a07\u0a08\u0a09\u0a0a \n  \r \u0a0f\u0a10  \u0a13\u0a14\u0a15\u0a16\u0a17\u0a18\u0a19\u0a1a\uffff\u0a1b\u0a1c\u0a1d\u0a1e !\u0a1f\u0a20\u0a21\u0a22\u0a23\u0a24)(\u0a25\u0a26,\u0a27.\u0a280123456789:; \u0a2a\u0a2b?\u0a2c\u0a2d\u0a2e\u0a2f\u0a30 \u0a32\u0a33 \u0a35\u0a36 \u0a38\u0a39\u0a3c \u0a3e\u0a3f\u0a40\u0a41\u0a42    \u0a47\u0a48  \u0a4b\u0a4c\u0a4d\u0a51abcdefghijklmnopqrstuvwxyz\u0a70\u0a71\u0a72\u0a73\u0a74"
  
      aput-object v14, v12, v13
  
***************
*** 122,139 ****
  
      const/16 v13, 0xc
  
!     const-string v14, "\u0c01\u0c02\u0c03\u0c05\u0c06\u0c07\u0c08\u0c09\u0c0a\u0c0b\n\u0c0c \r\u0c0e\u0c0f\u0c10 \u0c12\u0c13\u0c14\u0c15\u0c16\u0c17\u0c18\u0c19\u0c1a\uffff\u0c1b\u0c1c\u0c1d\u0c1e !\u0c1f\u0c20\u0c21\u0c22\u0c23\u0c24)(\u0c25\u0c26,\u0c27.\u0c280123456789:; \u0c2a\u0c2b?\u0c2c\u0c2d\u0c2e\u0c2f\u0c30\u0c31\u0c32\u0c33 \u0c35\u0c36\u0c37\u0c38\u0c39 \u0c3d\u0c3e\u0c3f\u0c40\u0c41\u0c42\u0c43\u0c44 \u0c46\u0c47\u0c48 \u0c4a\u0c4b\u0c4c\u0c4d\u0c55abcdefghijklmnopqrstuvwxyz\u0c56\u0c60\u0c61\u0c62\u0c63"
  
      aput-object v14, v12, v13
  
      const/16 v13, 0xd
  
!     const-string v14, "\u0627\u0622\u0628\u067b\u0680\u067e\u06a6\u062a\u06c2\u067f\n\u0679\u067d\r\u067a\u067c\u062b\u062c\u0681\u0684\u0683\u0685\u0686\u0687\u062d\u062e\u062f\uffff\u068c\u0688\u0689\u068a !\u068f\u068d\u0630\u0631\u0691\u0693)(\u0699\u0632,\u0696.\u06980123456789:;\u069a\u0633\u0634?\u0635\u0636\u0637\u0638\u0639\u0641\u0642\u06a9\u06aa\u06ab\u06af\u06b3\u06b1\u0644\u0645\u0646\u06ba\u06bb\u06bc\u0648\u06c4\u06d5\u06c1\u06be\u0621\u06cc\u06d0\u06d2\u064d\u0650\u064f\u0657\u0654abcdefghijklmnopqrstuvwxyz\u0655\u0651\u0653\u0656\u0670"
  
      aput-object v14, v12, v13
  
      sput-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageTables:[Ljava/lang/String;
  
      const/16 v12, 0xe
  
      new-array v12, v12, [Ljava/lang/String;
--- 123,141 ----
  
      const/16 v13, 0xc
  
!     const-string/jumbo v14, "\u0c01\u0c02\u0c03\u0c05\u0c06\u0c07\u0c08\u0c09\u0c0a\u0c0b\n\u0c0c \r\u0c0e\u0c0f\u0c10 \u0c12\u0c13\u0c14\u0c15\u0c16\u0c17\u0c18\u0c19\u0c1a\uffff\u0c1b\u0c1c\u0c1d\u0c1e !\u0c1f\u0c20\u0c21\u0c22\u0c23\u0c24)(\u0c25\u0c26,\u0c27.\u0c280123456789:; \u0c2a\u0c2b?\u0c2c\u0c2d\u0c2e\u0c2f\u0c30\u0c31\u0c32\u0c33 \u0c35\u0c36\u0c37\u0c38\u0c39 \u0c3d\u0c3e\u0c3f\u0c40\u0c41\u0c42\u0c43\u0c44 \u0c46\u0c47\u0c48 \u0c4a\u0c4b\u0c4c\u0c4d\u0c55abcdefghijklmnopqrstuvwxyz\u0c56\u0c60\u0c61\u0c62\u0c63"
  
      aput-object v14, v12, v13
  
      const/16 v13, 0xd
  
!     const-string/jumbo v14, "\u0627\u0622\u0628\u067b\u0680\u067e\u06a6\u062a\u06c2\u067f\n\u0679\u067d\r\u067a\u067c\u062b\u062c\u0681\u0684\u0683\u0685\u0686\u0687\u062d\u062e\u062f\uffff\u068c\u0688\u0689\u068a !\u068f\u068d\u0630\u0631\u0691\u0693)(\u0699\u0632,\u0696.\u06980123456789:;\u069a\u0633\u0634?\u0635\u0636\u0637\u0638\u0639\u0641\u0642\u06a9\u06aa\u06ab\u06af\u06b3\u06b1\u0644\u0645\u0646\u06ba\u06bb\u06bc\u0648\u06c4\u06d5\u06c1\u06be\u0621\u06cc\u06d0\u06d2\u064d\u0650\u064f\u0657\u0654abcdefghijklmnopqrstuvwxyz\u0655\u0651\u0653\u0656\u0670"
  
      aput-object v14, v12, v13
  
      sput-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageTables:[Ljava/lang/String;
  
+     .line 1229
      const/16 v12, 0xe
  
      new-array v12, v12, [Ljava/lang/String;
***************
*** 224,233 ****
--- 226,237 ----
  
      sput-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageShiftTables:[Ljava/lang/String;
  
+     .line 1362
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v7
  
+     .line 1364
      .local v7, r:Landroid/content/res/Resources;
      const v12, 0x107002a
  
***************
*** 237,242 ****
--- 241,247 ----
  
      sput-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sEnabledSingleShiftTables:[I
  
+     .line 1365
      const v12, 0x107002b
  
      invoke-virtual {v7, v12}, Landroid/content/res/Resources;->getIntArray(I)[I
***************
*** 245,262 ****
--- 250,271 ----
  
      sput-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sEnabledLockingShiftTables:[I
  
+     .line 1366
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageTables:[Ljava/lang/String;
  
      array-length v6, v12
  
+     .line 1367
      .local v6, numTables:I
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageShiftTables:[Ljava/lang/String;
  
      array-length v5, v12
  
+     .line 1368
      .local v5, numShiftTables:I
      if-eq v6, v5, :cond_0
  
+     .line 1369
      const-string v12, "GSM"
  
      new-instance v13, Ljava/lang/StringBuilder;
***************
*** 289,294 ****
--- 298,304 ----
  
      invoke-static {v12, v13}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1373
      :cond_0
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sEnabledSingleShiftTables:[I
  
***************
*** 296,301 ****
--- 306,312 ----
  
      if-lez v12, :cond_2
  
+     .line 1374
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sEnabledSingleShiftTables:[I
  
      sget-object v13, Lcom/android/internal/telephony/GsmAlphabet;->sEnabledSingleShiftTables:[I
***************
*** 308,333 ****
--- 319,349 ----
  
      sput v12, Lcom/android/internal/telephony/GsmAlphabet;->sHighestEnabledSingleShiftCode:I
  
+     .line 1380
      :goto_0
      new-array v12, v6, [Landroid/util/SparseIntArray;
  
      sput-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
+     .line 1381
      const/4 v3, 0x0
  
      .local v3, i:I
      :goto_1
      if-ge v3, v6, :cond_4
  
+     .line 1382
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageTables:[Ljava/lang/String;
  
      aget-object v10, v12, v3
  
+     .line 1384
      .local v10, table:Ljava/lang/String;
      invoke-virtual {v10}, Ljava/lang/String;->length()I
  
      move-result v11
  
+     .line 1385
      .local v11, tableLen:I
      if-eqz v11, :cond_1
  
***************
*** 335,340 ****
--- 351,357 ----
  
      if-eq v11, v12, :cond_1
  
+     .line 1386
      const-string v12, "GSM"
  
      new-instance v13, Ljava/lang/StringBuilder;
***************
*** 373,405 ****
--- 390,429 ----
  
      invoke-static {v12, v13}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1390
      :cond_1
      new-instance v1, Landroid/util/SparseIntArray;
  
      invoke-direct {v1, v11}, Landroid/util/SparseIntArray;-><init>(I)V
  
+     .line 1391
      .local v1, charToGsmTable:Landroid/util/SparseIntArray;
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
      aput-object v1, v12, v3
  
+     .line 1392
      const/4 v4, 0x0
  
      .local v4, j:I
      :goto_2
      if-ge v4, v11, :cond_3
  
+     .line 1393
      invoke-virtual {v10, v4}, Ljava/lang/String;->charAt(I)C
  
      move-result v0
  
+     .line 1394
      .local v0, c:C
      invoke-virtual {v1, v0, v4}, Landroid/util/SparseIntArray;->put(II)V
  
+     .line 1392
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_2
  
+     .line 1377
      .end local v0           #c:C
      .end local v1           #charToGsmTable:Landroid/util/SparseIntArray;
      .end local v3           #i:I
***************
*** 413,418 ****
--- 437,443 ----
  
      goto :goto_0
  
+     .line 1381
      .restart local v1       #charToGsmTable:Landroid/util/SparseIntArray;
      .restart local v3       #i:I
      .restart local v4       #j:I
***************
*** 423,428 ****
--- 448,454 ----
  
      goto :goto_1
  
+     .line 1398
      .end local v1           #charToGsmTable:Landroid/util/SparseIntArray;
      .end local v4           #j:I
      .end local v10           #table:Ljava/lang/String;
***************
*** 432,451 ****
--- 458,481 ----
  
      sput-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
+     .line 1399
      const/4 v3, 0x0
  
      :goto_3
      if-ge v3, v5, :cond_8
  
+     .line 1400
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageShiftTables:[Ljava/lang/String;
  
      aget-object v8, v12, v3
  
+     .line 1402
      .local v8, shiftTable:Ljava/lang/String;
      invoke-virtual {v8}, Ljava/lang/String;->length()I
  
      move-result v9
  
+     .line 1403
      .local v9, shiftTableLen:I
      if-eqz v9, :cond_5
  
***************
*** 453,458 ****
--- 483,489 ----
  
      if-eq v9, v12, :cond_5
  
+     .line 1404
      const-string v12, "GSM"
  
      new-instance v13, Ljava/lang/StringBuilder;
***************
*** 491,534 ****
--- 522,574 ----
  
      invoke-static {v12, v13}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1408
      :cond_5
      new-instance v2, Landroid/util/SparseIntArray;
  
      invoke-direct {v2, v9}, Landroid/util/SparseIntArray;-><init>(I)V
  
+     .line 1409
      .local v2, charToShiftTable:Landroid/util/SparseIntArray;
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
      aput-object v2, v12, v3
  
+     .line 1410
      const/4 v4, 0x0
  
      .restart local v4       #j:I
      :goto_4
      if-ge v4, v9, :cond_7
  
+     .line 1411
      invoke-virtual {v8, v4}, Ljava/lang/String;->charAt(I)C
  
      move-result v0
  
+     .line 1412
      .restart local v0       #c:C
      const/16 v12, 0x20
  
      if-eq v0, v12, :cond_6
  
+     .line 1413
      invoke-virtual {v2, v0, v4}, Landroid/util/SparseIntArray;->put(II)V
  
+     .line 1410
      :cond_6
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_4
  
+     .line 1399
      .end local v0           #c:C
      :cond_7
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_3
  
+     .line 1417
      .end local v2           #charToShiftTable:Landroid/util/SparseIntArray;
      .end local v4           #j:I
      .end local v8           #shiftTable:Ljava/lang/String;
***************
*** 541,546 ****
--- 581,587 ----
      .locals 0
  
      .prologue
+     .line 46
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 550,555 ****
--- 591,597 ----
      .locals 1
  
      .prologue
+     .line 43
      sget v0, Lcom/android/internal/telephony/GsmAlphabet;->sHighestEnabledSingleShiftCode:I
  
      return v0
***************
*** 559,564 ****
--- 601,607 ----
      .locals 1
  
      .prologue
+     .line 43
      sget-object v0, Lcom/android/internal/telephony/GsmAlphabet;->sEnabledSingleShiftTables:[I
  
      return-object v0
***************
*** 573,578 ****
--- 616,622 ----
  
      const/4 v2, 0x0
  
+     .line 95
      const/4 v1, 0x0
  
      :try_start_0
***************
*** 582,593 ****
--- 626,640 ----
  
      move-result v1
  
+     .line 98
      :goto_0
      return v1
  
+     .line 96
      :catch_0
      move-exception v0
  
+     .line 98
      .local v0, ex:Lcom/android/internal/telephony/EncodeException;
      sget-object v1, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
***************
*** 617,622 ****
--- 664,670 ----
  
      const/4 v2, -0x1
  
+     .line 118
      sget-object v1, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
      aget-object v1, v1, v3
***************
*** 625,633 ****
--- 673,683 ----
  
      move-result v0
  
+     .line 120
      .local v0, ret:I
      if-ne v0, v2, :cond_2
  
+     .line 121
      sget-object v1, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
      aget-object v1, v1, v3
***************
*** 636,651 ****
--- 686,705 ----
  
      move-result v0
  
+     .line 123
      if-ne v0, v2, :cond_1
  
+     .line 124
      if-eqz p1, :cond_0
  
+     .line 125
      new-instance v1, Lcom/android/internal/telephony/EncodeException;
  
      invoke-direct {v1, p0}, Lcom/android/internal/telephony/EncodeException;-><init>(C)V
  
      throw v1
  
+     .line 127
      :cond_0
      sget-object v1, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
***************
*** 655,663 ****
--- 709,719 ----
  
      move-result v1
  
+     .line 134
      :goto_0
      return v1
  
+     .line 130
      :cond_1
      const/16 v1, 0x1b
  
***************
*** 666,671 ****
--- 722,728 ----
      :cond_2
      move v1, v0
  
+     .line 134
      goto :goto_0
  .end method
  
***************
*** 680,685 ****
--- 737,743 ----
  
      const/4 v2, -0x1
  
+     .line 148
      sget-object v1, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
      aget-object v1, v1, v3
***************
*** 688,696 ****
--- 746,756 ----
  
      move-result v0
  
+     .line 150
      .local v0, ret:I
      if-ne v0, v2, :cond_0
  
+     .line 151
      sget-object v1, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
      aget-object v1, v1, v3
***************
*** 699,704 ****
--- 759,765 ----
  
      move-result v0
  
+     .line 154
      .end local v0           #ret:I
      :cond_0
      return v0
***************
*** 711,716 ****
--- 772,778 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 665
      const/4 v2, 0x0
  
      :try_start_0
***************
*** 720,731 ****
--- 782,796 ----
  
      move-result v1
  
+     .line 668
      :goto_0
      return v1
  
+     .line 666
      :catch_0
      move-exception v0
  
+     .line 668
      .local v0, ex:Lcom/android/internal/telephony/EncodeException;
      goto :goto_0
  .end method
***************
*** 747,752 ****
--- 812,818 ----
  
      const/4 v2, -0x1
  
+     .line 683
      sget-object v1, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
      aget-object v1, v1, v3
***************
*** 757,766 ****
--- 823,834 ----
  
      if-eq v1, v2, :cond_1
  
+     .line 695
      :cond_0
      :goto_0
      return v0
  
+     .line 687
      :cond_1
      sget-object v1, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
***************
*** 772,784 ****
--- 840,855 ----
  
      if-eq v1, v2, :cond_2
  
+     .line 688
      const/4 v0, 0x2
  
      goto :goto_0
  
+     .line 691
      :cond_2
      if-eqz p1, :cond_0
  
+     .line 692
      new-instance v0, Lcom/android/internal/telephony/EncodeException;
  
      invoke-direct {v0, p0}, Lcom/android/internal/telephony/EncodeException;-><init>(C)V
***************
*** 792,797 ****
--- 863,869 ----
      .parameter "use7bitOnly"
  
      .prologue
+     .line 758
      sget-object v25, Lcom/android/internal/telephony/GsmAlphabet;->sEnabledSingleShiftTables:[I
  
      move-object/from16 v0, v25
***************
*** 812,821 ****
--- 884,895 ----
  
      if-nez v25, :cond_3
  
+     .line 759
      new-instance v22, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
  
      invoke-direct/range {v22 .. v22}, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;-><init>()V
  
+     .line 760
      .local v22, ted:Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
      const/16 v25, 0x0
  
***************
*** 833,838 ****
--- 907,913 ----
  
      move-result v14
  
+     .line 761
      .local v14, septets:I
      const/16 v25, -0x1
  
***************
*** 840,853 ****
--- 915,931 ----
  
      if-ne v14, v0, :cond_1
  
+     .line 762
      const/16 v22, 0x0
  
+     .line 889
      .end local v14           #septets:I
      .end local v22           #ted:Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
      :cond_0
      :goto_0
      return-object v22
  
+     .line 764
      .restart local v14       #septets:I
      .restart local v22       #ted:Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
      :cond_1
***************
*** 859,874 ****
--- 937,955 ----
  
      iput v0, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->codeUnitSize:I
  
+     .line 765
      move-object/from16 v0, v22
  
      iput v14, v0, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->codeUnitCount:I
  
+     .line 766
      const/16 v25, 0xa0
  
      move/from16 v0, v25
  
      if-le v14, v0, :cond_2
  
+     .line 767
      add-int/lit16 v0, v14, 0x98
  
      move/from16 v25, v0
***************
*** 885,890 ****
--- 966,972 ----
  
      iput v0, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->msgCount:I
  
+     .line 769
      move-object/from16 v0, v22
  
      iget v0, v0, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->msgCount:I
***************
*** 905,910 ****
--- 987,993 ----
  
      iput v0, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->codeUnitsRemaining:I
  
+     .line 775
      :goto_1
      const/16 v25, 0x1
  
***************
*** 916,921 ****
--- 999,1005 ----
  
      goto :goto_0
  
+     .line 772
      :cond_2
      const/16 v25, 0x1
  
***************
*** 925,930 ****
--- 1009,1015 ----
  
      iput v0, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->msgCount:I
  
+     .line 773
      rsub-int v0, v14, 0xa0
  
      move/from16 v25, v0
***************
*** 937,947 ****
--- 1022,1034 ----
  
      goto :goto_1
  
+     .line 779
      .end local v14           #septets:I
      .end local v22           #ted:Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
      :cond_3
      sget v11, Lcom/android/internal/telephony/GsmAlphabet;->sHighestEnabledSingleShiftCode:I
  
+     .line 780
      .local v11, maxSingleShiftCode:I
      new-instance v10, Ljava/util/ArrayList;
  
***************
*** 959,964 ****
--- 1046,1052 ----
  
      invoke-direct {v10, v0}, Ljava/util/ArrayList;-><init>(I)V
  
+     .line 784
      .local v10, lpcList:Ljava/util/List;,"Ljava/util/List<Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;>;"
      new-instance v25, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;
  
***************
*** 970,975 ****
--- 1058,1064 ----
  
      invoke-interface {v10, v0}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 785
      sget-object v4, Lcom/android/internal/telephony/GsmAlphabet;->sEnabledLockingShiftTables:[I
  
      .local v4, arr$:[I
***************
*** 984,989 ****
--- 1073,1079 ----
  
      aget v6, v4, v7
  
+     .line 787
      .local v6, i:I
      if-eqz v6, :cond_4
  
***************
*** 997,1002 ****
--- 1087,1093 ----
  
      if-nez v25, :cond_4
  
+     .line 788
      new-instance v25, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;
  
      move-object/from16 v0, v25
***************
*** 1007,1023 ****
--- 1098,1117 ----
  
      invoke-interface {v10, v0}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 785
      :cond_4
      add-int/lit8 v7, v7, 0x1
  
      goto :goto_2
  
+     .line 792
      .end local v6           #i:I
      :cond_5
      invoke-interface/range {p0 .. p0}, Ljava/lang/CharSequence;->length()I
  
      move-result v19
  
+     .line 794
      .local v19, sz:I
      const/4 v6, 0x0
  
***************
*** 1034,1045 ****
--- 1128,1141 ----
  
      if-nez v25, :cond_e
  
+     .line 795
      move-object/from16 v0, p0
  
      invoke-interface {v0, v6}, Ljava/lang/CharSequence;->charAt(I)C
  
      move-result v5
  
+     .line 796
      .local v5, c:C
      const/16 v25, 0x1b
  
***************
*** 1047,1063 ****
--- 1143,1162 ----
  
      if-ne v5, v0, :cond_7
  
+     .line 797
      const-string v25, "GSM"
  
      const-string v26, "countGsmSeptets() string contains Escape character, ignoring!"
  
      invoke-static/range {v25 .. v26}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 794
      :cond_6
      add-int/lit8 v6, v6, 0x1
  
      goto :goto_3
  
+     .line 801
      :cond_7
      invoke-interface {v10}, Ljava/util/List;->iterator()Ljava/util/Iterator;
  
***************
*** 1077,1082 ****
--- 1176,1182 ----
  
      check-cast v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;
  
+     .line 802
      .local v9, lpc:Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;
      sget-object v25, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
***************
*** 1096,1101 ****
--- 1196,1202 ----
  
      move-result v21
  
+     .line 803
      .local v21, tableIndex:I
      const/16 v25, -0x1
  
***************
*** 1105,1110 ****
--- 1206,1212 ----
  
      if-ne v0, v1, :cond_c
  
+     .line 805
      const/16 v20, 0x0
  
      .local v20, table:I
***************
*** 1113,1118 ****
--- 1215,1221 ----
  
      if-gt v0, v11, :cond_8
  
+     .line 806
      iget-object v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
      move-object/from16 v25, v0
***************
*** 1127,1132 ****
--- 1230,1236 ----
  
      if-eq v0, v1, :cond_9
  
+     .line 807
      sget-object v25, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
      aget-object v25, v25, v20
***************
*** 1141,1146 ****
--- 1245,1251 ----
  
      move-result v18
  
+     .line 808
      .local v18, shiftTableIndex:I
      const/16 v25, -0x1
  
***************
*** 1150,1157 ****
--- 1255,1264 ----
  
      if-ne v0, v1, :cond_b
  
+     .line 809
      if-eqz p1, :cond_a
  
+     .line 811
      iget-object v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
      move-object/from16 v25, v0
***************
*** 1162,1167 ****
--- 1269,1275 ----
  
      aput v26, v25, v20
  
+     .line 812
      iget-object v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->unencodableCounts:[I
  
      move-object/from16 v25, v0
***************
*** 1172,1177 ****
--- 1280,1286 ----
  
      aput v26, v25, v20
  
+     .line 805
      .end local v18           #shiftTableIndex:I
      :cond_9
      :goto_5
***************
*** 1179,1184 ****
--- 1288,1294 ----
  
      goto :goto_4
  
+     .line 815
      .restart local v18       #shiftTableIndex:I
      :cond_a
      iget-object v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
***************
*** 1191,1196 ****
--- 1301,1307 ----
  
      goto :goto_5
  
+     .line 819
      :cond_b
      iget-object v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
***************
*** 1204,1209 ****
--- 1315,1321 ----
  
      goto :goto_5
  
+     .line 825
      .end local v18           #shiftTableIndex:I
      .end local v20           #table:I
      :cond_c
***************
*** 1215,1220 ****
--- 1327,1333 ----
  
      if-gt v0, v11, :cond_8
  
+     .line 826
      iget-object v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
      move-object/from16 v25, v0
***************
*** 1229,1234 ****
--- 1342,1348 ----
  
      if-eq v0, v1, :cond_d
  
+     .line 827
      iget-object v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
      move-object/from16 v25, v0
***************
*** 1239,1249 ****
--- 1353,1365 ----
  
      aput v26, v25, v20
  
+     .line 825
      :cond_d
      add-int/lit8 v20, v20, 0x1
  
      goto :goto_6
  
+     .line 835
      .end local v5           #c:C
      .end local v7           #i$:Ljava/util/Iterator;
      .end local v9           #lpc:Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;
***************
*** 1254,1259 ****
--- 1370,1376 ----
  
      invoke-direct/range {v22 .. v22}, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;-><init>()V
  
+     .line 836
      .restart local v22       #ted:Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
      const v25, 0x7fffffff
  
***************
*** 1263,1268 ****
--- 1380,1386 ----
  
      iput v0, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->msgCount:I
  
+     .line 837
      const/16 v25, 0x1
  
      move/from16 v0, v25
***************
*** 1271,1278 ****
--- 1389,1398 ----
  
      iput v0, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->codeUnitSize:I
  
+     .line 838
      const v12, 0x7fffffff
  
+     .line 839
      .local v12, minUnencodableCount:I
      invoke-interface {v10}, Ljava/util/List;->iterator()Ljava/util/Iterator;
  
***************
*** 1292,1297 ****
--- 1412,1418 ----
  
      check-cast v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;
  
+     .line 840
      .restart local v9       #lpc:Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;
      const/16 v17, 0x0
  
***************
*** 1301,1312 ****
--- 1422,1435 ----
  
      if-gt v0, v11, :cond_f
  
+     .line 841
      iget-object v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->septetCounts:[I
  
      move-object/from16 v25, v0
  
      aget v14, v25, v17
  
+     .line 842
      .restart local v14       #septets:I
      const/16 v25, -0x1
  
***************
*** 1314,1325 ****
--- 1437,1450 ----
  
      if-ne v14, v0, :cond_11
  
+     .line 840
      :cond_10
      :goto_8
      add-int/lit8 v17, v17, 0x1
  
      goto :goto_7
  
+     .line 846
      :cond_11
      iget v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->languageCode:I
  
***************
*** 1329,1336 ****
--- 1454,1463 ----
  
      if-eqz v17, :cond_16
  
+     .line 847
      const/16 v23, 0x8
  
+     .line 855
      .local v23, udhLength:I
      :goto_9
      add-int v25, v14, v23
***************
*** 1343,1359 ****
--- 1470,1491 ----
  
      if-le v0, v1, :cond_19
  
+     .line 856
      if-nez v23, :cond_12
  
+     .line 857
      const/16 v23, 0x1
  
+     .line 859
      :cond_12
      add-int/lit8 v23, v23, 0x6
  
+     .line 860
      move/from16 v0, v23
  
      rsub-int v15, v0, 0xa0
  
+     .line 861
      .local v15, septetsPerMessage:I
      add-int v25, v14, v15
  
***************
*** 1361,1371 ****
--- 1493,1505 ----
  
      div-int v13, v25, v15
  
+     .line 862
      .local v13, msgCount:I
      mul-int v25, v13, v15
  
      sub-int v16, v25, v14
  
+     .line 868
      .end local v15           #septetsPerMessage:I
      .local v16, septetsRemaining:I
      :goto_a
***************
*** 1375,1380 ****
--- 1509,1515 ----
  
      aget v24, v25, v17
  
+     .line 869
      .local v24, unencodableCount:I
      if-eqz p1, :cond_13
  
***************
*** 1382,1387 ****
--- 1517,1523 ----
  
      if-gt v0, v12, :cond_10
  
+     .line 872
      :cond_13
      if-eqz p1, :cond_14
  
***************
*** 1422,1444 ****
--- 1558,1585 ----
  
      if-le v0, v1, :cond_10
  
+     .line 875
      :cond_15
      move/from16 v12, v24
  
+     .line 876
      move-object/from16 v0, v22
  
      iput v13, v0, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->msgCount:I
  
+     .line 877
      move-object/from16 v0, v22
  
      iput v14, v0, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->codeUnitCount:I
  
+     .line 878
      move/from16 v0, v16
  
      move-object/from16 v1, v22
  
      iput v0, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->codeUnitsRemaining:I
  
+     .line 879
      iget v0, v9, Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;->languageCode:I
  
      move/from16 v25, v0
***************
*** 1449,1454 ****
--- 1590,1596 ----
  
      iput v0, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->languageTable:I
  
+     .line 880
      move/from16 v0, v17
  
      move-object/from16 v1, v22
***************
*** 1457,1462 ****
--- 1599,1605 ----
  
      goto :goto_8
  
+     .line 848
      .end local v13           #msgCount:I
      .end local v16           #septetsRemaining:I
      .end local v23           #udhLength:I
***************
*** 1470,1481 ****
--- 1613,1626 ----
  
      if-eqz v17, :cond_18
  
+     .line 849
      :cond_17
      const/16 v23, 0x5
  
      .restart local v23       #udhLength:I
      goto :goto_9
  
+     .line 851
      .end local v23           #udhLength:I
      :cond_18
      const/16 v23, 0x0
***************
*** 1483,1491 ****
--- 1628,1638 ----
      .restart local v23       #udhLength:I
      goto :goto_9
  
+     .line 864
      :cond_19
      const/4 v13, 0x1
  
+     .line 865
      .restart local v13       #msgCount:I
      move/from16 v0, v23
  
***************
*** 1498,1503 ****
--- 1645,1651 ----
      .restart local v16       #septetsRemaining:I
      goto :goto_a
  
+     .line 885
      .end local v9           #lpc:Lcom/android/internal/telephony/GsmAlphabet$LanguagePairCount;
      .end local v13           #msgCount:I
      .end local v14           #septets:I
***************
*** 1519,1524 ****
--- 1667,1673 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 886
      const/16 v22, 0x0
  
      goto/16 :goto_0
***************
*** 1534,1556 ****
--- 1683,1710 ----
      .prologue
      const/4 v6, -0x1
  
+     .line 714
      const/4 v3, 0x0
  
+     .line 715
      .local v3, count:I
      invoke-interface {p0}, Ljava/lang/CharSequence;->length()I
  
      move-result v5
  
+     .line 716
      .local v5, sz:I
      sget-object v7, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
      aget-object v1, v7, p2
  
+     .line 717
      .local v1, charToLanguageTable:Landroid/util/SparseIntArray;
      sget-object v7, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
      aget-object v2, v7, p3
  
+     .line 718
      .local v2, charToShiftTable:Landroid/util/SparseIntArray;
      const/4 v4, 0x0
  
***************
*** 1558,1583 ****
--- 1712,1742 ----
      :goto_0
      if-ge v4, v5, :cond_4
  
+     .line 719
      invoke-interface {p0, v4}, Ljava/lang/CharSequence;->charAt(I)C
  
      move-result v0
  
+     .line 720
      .local v0, c:C
      const/16 v7, 0x1b
  
      if-ne v0, v7, :cond_0
  
+     .line 721
      const-string v7, "GSM"
  
      const-string v8, "countGsmSeptets() string contains Escape character, skipping."
  
      invoke-static {v7, v8}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 718
      :goto_1
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
  
+     .line 724
      :cond_0
      invoke-virtual {v1, v0, v6}, Landroid/util/SparseIntArray;->get(II)I
  
***************
*** 1585,1594 ****
--- 1744,1755 ----
  
      if-eq v7, v6, :cond_1
  
+     .line 725
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_1
  
+     .line 726
      :cond_1
      invoke-virtual {v2, v0, v6}, Landroid/util/SparseIntArray;->get(II)I
  
***************
*** 1596,1608 ****
--- 1757,1772 ----
  
      if-eq v7, v6, :cond_2
  
+     .line 727
      add-int/lit8 v3, v3, 0x2
  
      goto :goto_1
  
+     .line 728
      :cond_2
      if-eqz p1, :cond_3
  
+     .line 729
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_1
***************
*** 1610,1615 ****
--- 1774,1780 ----
      :cond_3
      move v3, v6
  
+     .line 734
      .end local v0           #c:C
      .end local v3           #count:I
      :cond_4
***************
*** 1627,1649 ****
--- 1792,1819 ----
      .prologue
      const/4 v7, -0x1
  
+     .line 909
      const/4 v0, 0x0
  
+     .line 910
      .local v0, accumulator:I
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v5
  
+     .line 912
      .local v5, size:I
      sget-object v6, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
      aget-object v2, v6, p3
  
+     .line 913
      .local v2, charToLangTable:Landroid/util/SparseIntArray;
      sget-object v6, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
      aget-object v1, v6, p4
  
+     .line 914
      .local v1, charToLangShiftTable:Landroid/util/SparseIntArray;
      move v4, p1
  
***************
*** 1651,1656 ****
--- 1821,1827 ----
      :goto_0
      if-ge v4, v5, :cond_3
  
+     .line 915
      invoke-virtual {p0, v4}, Ljava/lang/String;->charAt(I)C
  
      move-result v6
***************
*** 1659,1667 ****
--- 1830,1840 ----
  
      move-result v3
  
+     .line 916
      .local v3, encodedSeptet:I
      if-ne v3, v7, :cond_1
  
+     .line 917
      invoke-virtual {p0, v4}, Ljava/lang/String;->charAt(I)C
  
      move-result v6
***************
*** 1670,1687 ****
--- 1843,1865 ----
  
      move-result v3
  
+     .line 918
      if-ne v3, v7, :cond_0
  
+     .line 920
      add-int/lit8 v0, v0, 0x1
  
+     .line 927
      :goto_1
      if-le v0, p2, :cond_2
  
+     .line 931
      .end local v3           #encodedSeptet:I
      .end local v4           #i:I
      :goto_2
      return v4
  
+     .line 922
      .restart local v3       #encodedSeptet:I
      .restart local v4       #i:I
      :cond_0
***************
*** 1689,1699 ****
--- 1867,1879 ----
  
      goto :goto_1
  
+     .line 925
      :cond_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 914
      :cond_2
      add-int/lit8 v4, v4, 0x1
  
***************
*** 1703,1708 ****
--- 1883,1889 ----
      :cond_3
      move v4, v5
  
+     .line 931
      goto :goto_2
  .end method
  
***************
*** 1710,1715 ****
--- 1891,1897 ----
      .locals 2
  
      .prologue
+     .line 981
      const-class v0, Lcom/android/internal/telephony/GsmAlphabet;
  
      monitor-enter v0
***************
*** 1735,1740 ****
--- 1917,1923 ----
      .locals 2
  
      .prologue
+     .line 970
      const-class v0, Lcom/android/internal/telephony/GsmAlphabet;
  
      monitor-enter v0
***************
*** 1765,1770 ****
--- 1948,1954 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 407
      move-object v0, p0
  
      move v1, p1
***************
*** 1792,1803 ****
--- 1976,1989 ----
      .parameter "shiftTable"
  
      .prologue
+     .line 428
      new-instance v9, Ljava/lang/StringBuilder;
  
      move/from16 v0, p2
  
      invoke-direct {v9, v0}, Ljava/lang/StringBuilder;-><init>(I)V
  
+     .line 430
      .local v9, ret:Ljava/lang/StringBuilder;
      if-ltz p4, :cond_0
  
***************
*** 1809,1814 ****
--- 1995,2001 ----
  
      if-le v0, v12, :cond_1
  
+     .line 431
      :cond_0
      const-string v12, "GSM"
  
***************
*** 1816,1822 ****
  
      invoke-direct {v13}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v14, "unknown language table "
  
      invoke-virtual {v13, v14}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2003,2009 ----
  
      invoke-direct {v13}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v14, "unknown language table "
  
      invoke-virtual {v13, v14}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1840,1847 ****
--- 2027,2036 ----
  
      invoke-static {v12, v13}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 432
      const/16 p4, 0x0
  
+     .line 434
      :cond_1
      if-ltz p5, :cond_2
  
***************
*** 1853,1858 ****
--- 2042,2048 ----
  
      if-le v0, v12, :cond_3
  
+     .line 435
      :cond_2
      const-string v12, "GSM"
  
***************
*** 1860,1866 ****
  
      invoke-direct {v13}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v14, "unknown single shift table "
  
      invoke-virtual {v13, v14}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2050,2056 ----
  
      invoke-direct {v13}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v14, "unknown single shift table "
  
      invoke-virtual {v13, v14}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1884,1905 ****
--- 2074,2100 ----
  
      invoke-static {v12, v13}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 436
      const/16 p5, 0x0
  
+     .line 440
      :cond_3
      const/4 v8, 0x0
  
+     .line 441
      .local v8, prevCharWasEscape:Z
      :try_start_0
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageTables:[Ljava/lang/String;
  
      aget-object v7, v12, p4
  
+     .line 442
      .local v7, languageTableToChar:Ljava/lang/String;
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageShiftTables:[Ljava/lang/String;
  
      aget-object v11, v12, p5
  
+     .line 444
      .local v11, shiftTableToChar:Ljava/lang/String;
      invoke-virtual {v7}, Ljava/lang/String;->isEmpty()Z
  
***************
*** 1907,1919 ****
  
      if-eqz v12, :cond_4
  
      const-string v12, "GSM"
  
      new-instance v13, Ljava/lang/StringBuilder;
  
      invoke-direct {v13}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v14, "no language table for code "
  
      invoke-virtual {v13, v14}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2102,2115 ----
  
      if-eqz v12, :cond_4
  
+     .line 445
      const-string v12, "GSM"
  
      new-instance v13, Ljava/lang/StringBuilder;
  
      invoke-direct {v13}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v14, "no language table for code "
  
      invoke-virtual {v13, v14}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1937,1948 ****
--- 2133,2146 ----
  
      invoke-static {v12, v13}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 446
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageTables:[Ljava/lang/String;
  
      const/4 v13, 0x0
  
      aget-object v7, v12, v13
  
+     .line 448
      :cond_4
      invoke-virtual {v11}, Ljava/lang/String;->isEmpty()Z
  
***************
*** 1950,1962 ****
  
      if-eqz v12, :cond_5
  
      const-string v12, "GSM"
  
      new-instance v13, Ljava/lang/StringBuilder;
  
      invoke-direct {v13}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v14, "no single shift table for code "
  
      invoke-virtual {v13, v14}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2148,2161 ----
  
      if-eqz v12, :cond_5
  
+     .line 449
      const-string v12, "GSM"
  
      new-instance v13, Ljava/lang/StringBuilder;
  
      invoke-direct {v13}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v14, "no single shift table for code "
  
      invoke-virtual {v13, v14}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1980,1991 ****
--- 2179,2192 ----
  
      invoke-static {v12, v13}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 450
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageShiftTables:[Ljava/lang/String;
  
      const/4 v13, 0x0
  
      aget-object v11, v12, v13
  
+     .line 453
      :cond_5
      const/4 v6, 0x0
  
***************
*** 1995,2010 ****
--- 2196,2215 ----
  
      if-ge v6, v0, :cond_b
  
+     .line 454
      mul-int/lit8 v12, v6, 0x7
  
      add-int v1, v12, p3
  
+     .line 456
      .local v1, bitOffset:I
      div-int/lit8 v2, v1, 0x8
  
+     .line 457
      .local v2, byteOffset:I
      rem-int/lit8 v10, v1, 0x8
  
+     .line 460
      .local v10, shift:I
      add-int v12, p1, v2
  
***************
*** 2014,2024 ****
--- 2219,2231 ----
  
      and-int/lit8 v5, v12, 0x7f
  
+     .line 463
      .local v5, gsmVal:I
      const/4 v12, 0x1
  
      if-le v10, v12, :cond_6
  
+     .line 465
      const/16 v12, 0x7f
  
      add-int/lit8 v13, v10, -0x1
***************
*** 2027,2032 ****
--- 2234,2240 ----
  
      and-int/2addr v5, v12
  
+     .line 467
      add-int v12, p1, v2
  
      add-int/lit8 v12, v12, 0x1
***************
*** 2041,2075 ****
--- 2249,2291 ----
  
      or-int/2addr v5, v12
  
+     .line 470
      :cond_6
      if-eqz v8, :cond_9
  
+     .line 471
      const/16 v12, 0x1b
  
      if-ne v5, v12, :cond_7
  
+     .line 472
      const/16 v12, 0x20
  
      invoke-virtual {v9, v12}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 481
      :goto_1
      const/4 v8, 0x0
  
+     .line 453
      :goto_2
      add-int/lit8 v6, v6, 0x1
  
      goto :goto_0
  
+     .line 474
      :cond_7
      invoke-virtual {v11, v5}, Ljava/lang/String;->charAt(I)C
  
      move-result v3
  
+     .line 475
      .local v3, c:C
      const/16 v12, 0x20
  
      if-ne v3, v12, :cond_8
  
+     .line 476
      invoke-virtual {v7, v5}, Ljava/lang/String;->charAt(I)C
  
      move-result v12
***************
*** 2080,2085 ****
--- 2296,2302 ----
  
      goto :goto_1
  
+     .line 488
      .end local v1           #bitOffset:I
      .end local v2           #byteOffset:I
      .end local v3           #c:C
***************
*** 2091,2096 ****
--- 2308,2314 ----
      :catch_0
      move-exception v4
  
+     .line 489
      .local v4, ex:Ljava/lang/RuntimeException;
      const-string v12, "GSM"
  
***************
*** 2098,2109 ****
--- 2316,2330 ----
  
      invoke-static {v12, v13, v4}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 490
      const/4 v12, 0x0
  
+     .line 493
      .end local v4           #ex:Ljava/lang/RuntimeException;
      :goto_3
      return-object v12
  
+     .line 478
      .restart local v1       #bitOffset:I
      .restart local v2       #byteOffset:I
      .restart local v3       #c:C
***************
*** 2118,2133 ****
--- 2339,2357 ----
  
      goto :goto_1
  
+     .line 482
      .end local v3           #c:C
      :cond_9
      const/16 v12, 0x1b
  
      if-ne v5, v12, :cond_a
  
+     .line 483
      const/4 v8, 0x1
  
      goto :goto_2
  
+     .line 485
      :cond_a
      invoke-virtual {v7, v5}, Ljava/lang/String;->charAt(I)C
  
***************
*** 2139,2144 ****
--- 2363,2369 ----
  
      goto :goto_2
  
+     .line 493
      .end local v1           #bitOffset:I
      .end local v2           #byteOffset:I
      .end local v5           #gsmVal:I
***************
*** 2158,2163 ****
--- 2383,2389 ----
      .parameter "length"
  
      .prologue
+     .line 511
      const-string v0, ""
  
      invoke-static {p0, p1, p2, v0}, Lcom/android/internal/telephony/GsmAlphabet;->gsm8BitUnpackedToString([BIILjava/lang/String;)Ljava/lang/String;
***************
*** 2175,2188 ****
--- 2401,2418 ----
      .parameter "characterset"
  
      .prologue
+     .line 526
      const/4 v5, 0x0
  
+     .line 527
      .local v5, isMbcs:Z
      const/4 v2, 0x0
  
+     .line 528
      .local v2, charset:Ljava/nio/charset/Charset;
      const/4 v7, 0x0
  
+     .line 530
      .local v7, mbcsBuffer:Ljava/nio/ByteBuffer;
      invoke-static/range {p3 .. p3}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
***************
*** 2190,2196 ****
  
      if-nez v12, :cond_0
  
!     const-string v12, "us-ascii"
  
      move-object/from16 v0, p3
  
--- 2420,2426 ----
  
      if-nez v12, :cond_0
  
!     const-string/jumbo v12, "us-ascii"
  
      move-object/from16 v0, p3
  
***************
*** 2206,2223 ****
--- 2436,2457 ----
  
      if-eqz v12, :cond_0
  
+     .line 533
      const/4 v5, 0x1
  
+     .line 534
      invoke-static/range {p3 .. p3}, Ljava/nio/charset/Charset;->forName(Ljava/lang/String;)Ljava/nio/charset/Charset;
  
      move-result-object v2
  
+     .line 535
      const/4 v12, 0x2
  
      invoke-static {v12}, Ljava/nio/ByteBuffer;->allocate(I)Ljava/nio/ByteBuffer;
  
      move-result-object v7
  
+     .line 539
      :cond_0
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageTables:[Ljava/lang/String;
  
***************
*** 2225,2230 ****
--- 2459,2465 ----
  
      aget-object v6, v12, v13
  
+     .line 540
      .local v6, languageTableToChar:Ljava/lang/String;
      sget-object v12, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageShiftTables:[Ljava/lang/String;
  
***************
*** 2232,2237 ****
--- 2467,2473 ----
  
      aget-object v11, v12, v13
  
+     .line 542
      .local v11, shiftTableToChar:Ljava/lang/String;
      new-instance v9, Ljava/lang/StringBuilder;
  
***************
*** 2239,2247 ****
--- 2475,2485 ----
  
      invoke-direct {v9, v0}, Ljava/lang/StringBuilder;-><init>(I)V
  
+     .line 543
      .local v9, ret:Ljava/lang/StringBuilder;
      const/4 v8, 0x0
  
+     .line 544
      .local v8, prevWasEscape:Z
      move v3, p1
  
***************
*** 2255,2269 ****
--- 2493,2510 ----
  
      if-ge v4, v12, :cond_1
  
+     .line 547
      aget-byte v12, p0, v4
  
      and-int/lit16 v1, v12, 0xff
  
+     .line 549
      .local v1, c:I
      const/16 v12, 0xff
  
      if-ne v1, v12, :cond_2
  
+     .line 585
      .end local v1           #c:I
      :cond_1
      invoke-virtual {v9}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
***************
*** 2272,2293 ****
--- 2513,2539 ----
  
      return-object v12
  
+     .line 551
      .restart local v1       #c:I
      :cond_2
      const/16 v12, 0x1b
  
      if-ne v1, v12, :cond_4
  
+     .line 552
      if-eqz v8, :cond_3
  
+     .line 556
      const/16 v12, 0x20
  
      invoke-virtual {v9, v12}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 557
      const/4 v8, 0x0
  
      move v3, v4
  
+     .line 544
      .end local v4           #i:I
      .restart local v3       #i:I
      :goto_1
***************
*** 2299,2304 ****
--- 2545,2551 ----
      .restart local v4       #i:I
      goto :goto_0
  
+     .line 559
      :cond_3
      const/4 v8, 0x1
  
***************
*** 2308,2327 ****
--- 2555,2578 ----
      .restart local v3       #i:I
      goto :goto_1
  
+     .line 562
      .end local v3           #i:I
      .restart local v4       #i:I
      :cond_4
      if-eqz v8, :cond_6
  
+     .line 563
      invoke-virtual {v11, v1}, Ljava/lang/String;->charAt(I)C
  
      move-result v10
  
+     .line 564
      .local v10, shiftChar:C
      const/16 v12, 0x20
  
      if-ne v10, v12, :cond_5
  
+     .line 566
      invoke-virtual {v6, v1}, Ljava/lang/String;->charAt(I)C
  
      move-result v12
***************
*** 2331,2336 ****
--- 2582,2588 ----
      :goto_2
      move v3, v4
  
+     .line 581
      .end local v4           #i:I
      .end local v10           #shiftChar:C
      .restart local v3       #i:I
***************
*** 2339,2344 ****
--- 2591,2597 ----
  
      goto :goto_1
  
+     .line 568
      .end local v3           #i:I
      .restart local v4       #i:I
      .restart local v10       #shiftChar:C
***************
*** 2347,2352 ****
--- 2600,2606 ----
  
      goto :goto_2
  
+     .line 571
      .end local v10           #shiftChar:C
      :cond_6
      if-eqz v5, :cond_7
***************
*** 2361,2366 ****
--- 2615,2621 ----
  
      if-lt v12, v13, :cond_8
  
+     .line 572
      :cond_7
      invoke-virtual {v6, v1}, Ljava/lang/String;->charAt(I)C
  
***************
*** 2374,2384 ****
--- 2629,2641 ----
      .restart local v3       #i:I
      goto :goto_3
  
+     .line 575
      .end local v3           #i:I
      .restart local v4       #i:I
      :cond_8
      invoke-virtual {v7}, Ljava/nio/ByteBuffer;->clear()Ljava/nio/Buffer;
  
+     .line 576
      add-int/lit8 v3, v4, 0x1
  
      .end local v4           #i:I
***************
*** 2387,2394 ****
--- 2644,2653 ----
  
      invoke-virtual {v7, p0, v4, v12}, Ljava/nio/ByteBuffer;->put([BII)Ljava/nio/ByteBuffer;
  
+     .line 577
      invoke-virtual {v7}, Ljava/nio/ByteBuffer;->flip()Ljava/nio/Buffer;
  
+     .line 578
      invoke-virtual {v2, v7}, Ljava/nio/charset/Charset;->decode(Ljava/nio/ByteBuffer;)Ljava/nio/CharBuffer;
  
      move-result-object v12
***************
*** 2411,2426 ****
--- 2670,2688 ----
  
      const/16 v1, 0x20
  
+     .line 193
      const/16 v2, 0x1b
  
      if-ne p0, v2, :cond_1
  
      move v0, v1
  
+     .line 203
      :cond_0
      :goto_0
      return v0
  
+     .line 195
      :cond_1
      if-ltz p0, :cond_2
  
***************
*** 2428,2433 ****
--- 2690,2696 ----
  
      if-ge p0, v2, :cond_2
  
+     .line 196
      sget-object v2, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageShiftTables:[Ljava/lang/String;
  
      aget-object v2, v2, v3
***************
*** 2436,2444 ****
--- 2699,2709 ----
  
      move-result v0
  
+     .line 197
      .local v0, c:C
      if-ne v0, v1, :cond_0
  
+     .line 198
      sget-object v1, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageTables:[Ljava/lang/String;
  
      aget-object v1, v1, v3
***************
*** 2453,2458 ****
--- 2718,2724 ----
      :cond_2
      move v0, v1
  
+     .line 203
      goto :goto_0
  .end method
  
***************
*** 2461,2472 ****
--- 2727,2740 ----
      .parameter "gsmChar"
  
      .prologue
+     .line 171
      if-ltz p0, :cond_0
  
      const/16 v0, 0x80
  
      if-ge p0, v0, :cond_0
  
+     .line 172
      sget-object v0, Lcom/android/internal/telephony/GsmAlphabet;->sLanguageTables:[Ljava/lang/String;
  
      const/4 v1, 0x0
***************
*** 2477,2482 ****
--- 2745,2751 ----
  
      move-result v0
  
+     .line 174
      :goto_0
      return v0
  
***************
*** 2493,2503 ****
--- 2762,2775 ----
      .parameter "value"
  
      .prologue
+     .line 384
      div-int/lit8 v0, p1, 0x8
  
+     .line 385
      .local v0, byteOffset:I
      rem-int/lit8 v1, p1, 0x8
  
+     .line 387
      .local v1, shift:I
      add-int/lit8 v0, v0, 0x1
  
***************
*** 2511,2520 ****
--- 2783,2794 ----
  
      aput-byte v2, p0, v0
  
+     .line 389
      const/4 v2, 0x1
  
      if-le v1, v2, :cond_0
  
+     .line 390
      add-int/lit8 v0, v0, 0x1
  
      rsub-int/lit8 v2, v1, 0x8
***************
*** 2525,2530 ****
--- 2799,2805 ----
  
      aput-byte v2, p0, v0
  
+     .line 392
      :cond_0
      return-void
  .end method
***************
*** 2534,2539 ****
--- 2809,2815 ----
      .parameter "tables"
  
      .prologue
+     .line 959
      const-class v0, Lcom/android/internal/telephony/GsmAlphabet;
  
      monitor-enter v0
***************
*** 2543,2552 ****
--- 2819,2830 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 960
      monitor-exit v0
  
      return-void
  
+     .line 959
      :catchall_0
      move-exception v1
  
***************
*** 2560,2565 ****
--- 2838,2844 ----
      .parameter "tables"
  
      .prologue
+     .line 942
      const-class v1, Lcom/android/internal/telephony/GsmAlphabet;
  
      monitor-enter v1
***************
*** 2567,2576 ****
--- 2846,2857 ----
      :try_start_0
      sput-object p0, Lcom/android/internal/telephony/GsmAlphabet;->sEnabledSingleShiftTables:[I
  
+     .line 944
      array-length v0, p0
  
      if-lez v0, :cond_0
  
+     .line 945
      array-length v0, p0
  
      add-int/lit8 v0, v0, -0x1
***************
*** 2581,2591 ****
--- 2862,2874 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 949
      :goto_0
      monitor-exit v1
  
      return-void
  
+     .line 947
      :cond_0
      const/4 v0, 0x0
  
***************
*** 2596,2601 ****
--- 2879,2885 ----
  
      goto :goto_0
  
+     .line 942
      :catchall_0
      move-exception v0
  
***************
*** 2616,2621 ****
--- 2900,2906 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 283
      const/4 v0, 0x1
  
      invoke-static {p0, v1, v0, v1, v1}, Lcom/android/internal/telephony/GsmAlphabet;->stringToGsm7BitPacked(Ljava/lang/String;IZII)[B
***************
*** 2637,2642 ****
--- 2922,2928 ----
      .end annotation
  
      .prologue
+     .line 307
      const/4 v0, 0x0
  
      const/4 v1, 0x1
***************
*** 2662,2671 ****
--- 2948,2959 ----
      .end annotation
  
      .prologue
+     .line 334
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v7
  
+     .line 335
      .local v7, dataLen:I
      if-nez p2, :cond_0
  
***************
*** 2680,2690 ****
--- 2968,2980 ----
  
      move-result v10
  
+     .line 337
      .local v10, septetCount:I
      const/4 v13, -0x1
  
      if-ne v10, v13, :cond_1
  
+     .line 338
      new-instance v13, Lcom/android/internal/telephony/EncodeException;
  
      const-string v14, "countGsmSeptetsUsingTables(): unencodable char"
***************
*** 2693,2712 ****
--- 2983,3006 ----
  
      throw v13
  
+     .line 335
      .end local v10           #septetCount:I
      :cond_0
      const/4 v13, 0x0
  
      goto :goto_0
  
+     .line 340
      .restart local v10       #septetCount:I
      :cond_1
      add-int v10, v10, p1
  
+     .line 341
      const/16 v13, 0xff
  
      if-le v10, v13, :cond_2
  
+     .line 342
      new-instance v13, Lcom/android/internal/telephony/EncodeException;
  
      const-string v14, "Payload cannot exceed 255 septets"
***************
*** 2715,2720 ****
--- 3009,3015 ----
  
      throw v13
  
+     .line 344
      :cond_2
      mul-int/lit8 v13, v10, 0x7
  
***************
*** 2722,2742 ****
--- 3017,3041 ----
  
      div-int/lit8 v3, v13, 0x8
  
+     .line 345
      .local v3, byteCount:I
      add-int/lit8 v13, v3, 0x1
  
      new-array v9, v13, [B
  
+     .line 346
      .local v9, ret:[B
      sget-object v13, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
      aget-object v5, v13, p3
  
+     .line 347
      .local v5, charToLanguageTable:Landroid/util/SparseIntArray;
      sget-object v13, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
      aget-object v6, v13, p4
  
+     .line 348
      .local v6, charToShiftTable:Landroid/util/SparseIntArray;
      const/4 v8, 0x0
  
***************
*** 2746,2761 ****
--- 3045,3063 ----
      .local v11, septets:I
      mul-int/lit8 v2, p1, 0x7
  
+     .line 349
      .local v2, bitOffset:I
      :goto_1
      if-ge v8, v7, :cond_6
  
      if-ge v11, v10, :cond_6
  
+     .line 351
      invoke-virtual {p0, v8}, Ljava/lang/String;->charAt(I)C
  
      move-result v4
  
+     .line 352
      .local v4, c:C
      const/4 v13, -0x1
  
***************
*** 2763,2793 ****
  
      move-result v12
  
      .local v12, v:I
      const/4 v13, -0x1
  
      if-ne v12, v13, :cond_4
  
      const/4 v13, -0x1
  
      invoke-virtual {v6, v4, v13}, Landroid/util/SparseIntArray;->get(II)I
  
      move-result v12
  
      const/4 v13, -0x1
  
      if-ne v12, v13, :cond_5
  
      if-eqz p2, :cond_3
  
      new-instance v13, Lcom/android/internal/telephony/EncodeException;
  
!     const-string v14, "stringToGsm7BitPacked(): unencodable char"
  
      invoke-direct {v13, v14}, Lcom/android/internal/telephony/EncodeException;-><init>(Ljava/lang/String;)V
  
      throw v13
  
      :cond_3
      const/16 v13, 0x20
  
--- 3065,3101 ----
  
      move-result v12
  
+     .line 353
      .local v12, v:I
      const/4 v13, -0x1
  
      if-ne v12, v13, :cond_4
  
+     .line 354
      const/4 v13, -0x1
  
      invoke-virtual {v6, v4, v13}, Landroid/util/SparseIntArray;->get(II)I
  
      move-result v12
  
+     .line 355
      const/4 v13, -0x1
  
      if-ne v12, v13, :cond_5
  
+     .line 356
      if-eqz p2, :cond_3
  
+     .line 357
      new-instance v13, Lcom/android/internal/telephony/EncodeException;
  
!     const-string/jumbo v14, "stringToGsm7BitPacked(): unencodable char"
  
      invoke-direct {v13, v14}, Lcom/android/internal/telephony/EncodeException;-><init>(Ljava/lang/String;)V
  
      throw v13
  
+     .line 359
      :cond_3
      const/16 v13, 0x20
  
***************
*** 2797,2825 ****
--- 3105,3140 ----
  
      move-result v12
  
+     .line 367
      :cond_4
      :goto_2
      invoke-static {v9, v2, v12}, Lcom/android/internal/telephony/GsmAlphabet;->packSmsChar([BII)V
  
+     .line 368
      add-int/lit8 v11, v11, 0x1
  
+     .line 350
      add-int/lit8 v8, v8, 0x1
  
      add-int/lit8 v2, v2, 0x7
  
      goto :goto_1
  
+     .line 362
      :cond_5
      const/16 v13, 0x1b
  
      invoke-static {v9, v2, v13}, Lcom/android/internal/telephony/GsmAlphabet;->packSmsChar([BII)V
  
+     .line 363
      add-int/lit8 v2, v2, 0x7
  
+     .line 364
      add-int/lit8 v11, v11, 0x1
  
      goto :goto_2
  
+     .line 370
      .end local v4           #c:C
      .end local v12           #v:I
      :cond_6
***************
*** 2829,2834 ****
--- 3144,3150 ----
  
      aput-byte v14, v9, v13
  
+     .line 371
      return-object v9
  .end method
  
***************
*** 2845,2850 ****
--- 3161,3167 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 222
      invoke-static {p0, p1, v0, v0}, Lcom/android/internal/telephony/GsmAlphabet;->stringToGsm7BitPackedWithHeader(Ljava/lang/String;[BII)[B
  
      move-result-object v0
***************
*** 2867,2886 ****
--- 3184,3207 ----
      .prologue
      const/4 v4, 0x1
  
+     .line 250
      if-eqz p1, :cond_0
  
      array-length v3, p1
  
      if-nez v3, :cond_1
  
+     .line 251
      :cond_0
      invoke-static {p0, p2, p3}, Lcom/android/internal/telephony/GsmAlphabet;->stringToGsm7BitPacked(Ljava/lang/String;II)[B
  
      move-result-object v2
  
+     .line 263
      :goto_0
      return-object v2
  
+     .line 254
      :cond_1
      array-length v3, p1
  
***************
*** 2888,2903 ****
--- 3209,3227 ----
  
      mul-int/lit8 v0, v3, 0x8
  
+     .line 255
      .local v0, headerBits:I
      add-int/lit8 v3, v0, 0x6
  
      div-int/lit8 v1, v3, 0x7
  
+     .line 257
      .local v1, headerSeptets:I
      invoke-static {p0, v1, v4, p2, p3}, Lcom/android/internal/telephony/GsmAlphabet;->stringToGsm7BitPacked(Ljava/lang/String;IZII)[B
  
      move-result-object v2
  
+     .line 261
      .local v2, ret:[B
      array-length v3, p1
  
***************
*** 2905,2910 ****
--- 3229,3235 ----
  
      aput-byte v3, v2, v4
  
+     .line 262
      const/4 v3, 0x0
  
      const/4 v4, 0x2
***************
*** 2923,2942 ****
--- 3248,3271 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 598
      const/4 v2, 0x1
  
      invoke-static {p0, v2, v3, v3}, Lcom/android/internal/telephony/GsmAlphabet;->countGsmSeptetsUsingTables(Ljava/lang/CharSequence;ZII)I
  
      move-result v1
  
+     .line 601
      .local v1, septets:I
      new-array v0, v1, [B
  
+     .line 603
      .local v0, ret:[B
      array-length v2, v0
  
      invoke-static {p0, v0, v3, v2}, Lcom/android/internal/telephony/GsmAlphabet;->stringToGsm8BitUnpackedField(Ljava/lang/String;[BII)V
  
+     .line 605
      return-object v0
  .end method
  
***************
*** 2954,2971 ****
--- 3283,3304 ----
  
      const/4 v9, -0x1
  
+     .line 620
      move v4, p2
  
+     .line 621
      .local v4, outByteIndex:I
      sget-object v8, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToGsmTables:[Landroid/util/SparseIntArray;
  
      aget-object v1, v8, v10
  
+     .line 622
      .local v1, charToLanguageTable:Landroid/util/SparseIntArray;
      sget-object v8, Lcom/android/internal/telephony/GsmAlphabet;->sCharsToShiftTables:[Landroid/util/SparseIntArray;
  
      aget-object v2, v8, v10
  
+     .line 625
      .local v2, charToShiftTable:Landroid/util/SparseIntArray;
      const/4 v3, 0x0
  
***************
*** 2977,2982 ****
--- 3310,3316 ----
      .local v6, sz:I
      move v5, v4
  
+     .line 626
      .end local v4           #outByteIndex:I
      .local v5, outByteIndex:I
      :goto_0
***************
*** 2986,3015 ****
--- 3320,3356 ----
  
      if-ge v8, p3, :cond_1
  
+     .line 629
      invoke-virtual {p0, v3}, Ljava/lang/String;->charAt(I)C
  
      move-result v0
  
+     .line 631
      .local v0, c:C
      invoke-virtual {v1, v0, v9}, Landroid/util/SparseIntArray;->get(II)I
  
      move-result v7
  
+     .line 633
      .local v7, v:I
      if-ne v7, v9, :cond_4
  
+     .line 634
      invoke-virtual {v2, v0, v9}, Landroid/util/SparseIntArray;->get(II)I
  
      move-result v7
  
+     .line 635
      if-ne v7, v9, :cond_0
  
+     .line 636
      invoke-virtual {v1, v11, v11}, Landroid/util/SparseIntArray;->get(II)I
  
      move-result v7
  
      move v4, v5
  
+     .line 647
      .end local v5           #outByteIndex:I
      .restart local v4       #outByteIndex:I
      :goto_1
***************
*** 3021,3030 ****
--- 3362,3373 ----
  
      aput-byte v8, p1, v4
  
+     .line 627
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 639
      :cond_0
      add-int/lit8 v8, v5, 0x1
  
***************
*** 3032,3037 ****
--- 3375,3381 ----
  
      if-lt v8, p3, :cond_2
  
+     .line 651
      .end local v0           #c:C
      .end local v7           #v:I
      :cond_1
***************
*** 3040,3045 ****
--- 3384,3390 ----
  
      if-ge v8, p3, :cond_3
  
+     .line 652
      add-int/lit8 v4, v5, 0x1
  
      .end local v5           #outByteIndex:I
***************
*** 3052,3057 ****
--- 3397,3403 ----
      .restart local v5       #outByteIndex:I
      goto :goto_2
  
+     .line 643
      .restart local v0       #c:C
      .restart local v7       #v:I
      :cond_2
***************
*** 3065,3070 ****
--- 3411,3417 ----
  
      goto :goto_1
  
+     .line 654
      .end local v0           #c:C
      .end local v4           #outByteIndex:I
      .end local v7           #v:I
