*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccCard$2.smali	2012-03-23 19:37:46.224617987 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccCard$2.smali.original	2012-03-23 19:34:46.324337510 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 565
      iput-object p1, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
***************
*** 44,51 ****
--- 45,54 ----
  
      const/4 v4, 0x2
  
+     .line 571
      const/4 v1, 0x7
  
+     .line 575
      .local v1, serviceClassX:I
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 55,60 ****
--- 58,64 ----
  
      if-nez v2, :cond_0
  
+     .line 576
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v2, v2, Lcom/android/internal/telephony/IccCard;->mLogTag:Ljava/lang/String;
***************
*** 97,110 ****
--- 101,117 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 689
      :goto_0
      return-void
  
+     .line 581
      :cond_0
      iget v2, p1, Landroid/os/Message;->what:I
  
      packed-switch v2, :pswitch_data_0
  
+     .line 687
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v2, v2, Lcom/android/internal/telephony/IccCard;->mLogTag:Ljava/lang/String;
***************
*** 133,147 ****
--- 140,157 ----
  
      goto :goto_0
  
+     .line 583
      :pswitch_0
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iput-object v6, v2, Lcom/android/internal/telephony/IccCard;->mState:Lcom/android/internal/telephony/IccCard$State;
  
+     .line 584
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      invoke-virtual {v2}, Lcom/android/internal/telephony/IccCard;->updateStateProperty()V
  
+     .line 585
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      const-string v3, "NOT_READY"
***************
*** 150,155 ****
--- 160,166 ----
  
      goto :goto_0
  
+     .line 589
      :pswitch_1
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 163,168 ****
--- 174,180 ----
  
      invoke-interface {v2, v3}, Lcom/android/internal/telephony/CommandsInterface;->getIccCardStatus(Landroid/os/Message;)V
  
+     .line 590
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v2, v2, Lcom/android/internal/telephony/IccCard;->mPhone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 179,184 ****
--- 191,197 ----
  
      invoke-interface {v2, v3, v4, v1, v5}, Lcom/android/internal/telephony/CommandsInterface;->queryFacilityLock(Ljava/lang/String;Ljava/lang/String;ILandroid/os/Message;)V
  
+     .line 593
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v2, v2, Lcom/android/internal/telephony/IccCard;->mPhone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 199,204 ****
--- 212,218 ----
  
      goto :goto_0
  
+     .line 598
      :pswitch_2
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 212,217 ****
--- 226,232 ----
  
      invoke-interface {v2, v3}, Lcom/android/internal/telephony/CommandsInterface;->getIccCardStatus(Landroid/os/Message;)V
  
+     .line 599
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v2, v2, Lcom/android/internal/telephony/IccCard;->mPhone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 230,240 ****
--- 245,257 ----
  
      goto/16 :goto_0
  
+     .line 604
      :pswitch_3
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/AsyncResult;
  
+     .line 606
      .local v0, ar:Landroid/os/AsyncResult;
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 243,254 ****
--- 260,273 ----
  
      goto/16 :goto_0
  
+     .line 612
      .end local v0           #ar:Landroid/os/AsyncResult;
      :pswitch_4
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/AsyncResult;
  
+     .line 614
      .restart local v0       #ar:Landroid/os/AsyncResult;
      iget-object v2, v0, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
***************
*** 262,267 ****
--- 281,287 ----
  
      iput-object v3, v2, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
+     .line 616
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v2, v2, Lcom/android/internal/telephony/IccCard;->mPhone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 280,297 ****
--- 300,320 ----
  
      goto/16 :goto_0
  
+     .line 625
      .end local v0           #ar:Landroid/os/AsyncResult;
      :pswitch_5
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/AsyncResult;
  
+     .line 626
      .restart local v0       #ar:Landroid/os/AsyncResult;
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      #calls: Lcom/android/internal/telephony/IccCard;->getIccCardStatusDone(Landroid/os/AsyncResult;)V
      invoke-static {v2, v0}, Lcom/android/internal/telephony/IccCard;->access$100(Lcom/android/internal/telephony/IccCard;Landroid/os/AsyncResult;)V
  
+     .line 627
      iget-object v2, v0, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
      check-cast v2, Landroid/os/Message;
***************
*** 300,311 ****
--- 323,336 ----
  
      goto/16 :goto_0
  
+     .line 630
      .end local v0           #ar:Landroid/os/AsyncResult;
      :pswitch_6
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/AsyncResult;
  
+     .line 631
      .restart local v0       #ar:Landroid/os/AsyncResult;
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 314,325 ****
--- 339,352 ----
  
      goto/16 :goto_0
  
+     .line 634
      .end local v0           #ar:Landroid/os/AsyncResult;
      :pswitch_7
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/AsyncResult;
  
+     .line 635
      .restart local v0       #ar:Landroid/os/AsyncResult;
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 328,344 ****
--- 355,374 ----
  
      goto/16 :goto_0
  
+     .line 638
      .end local v0           #ar:Landroid/os/AsyncResult;
      :pswitch_8
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/AsyncResult;
  
+     .line 639
      .restart local v0       #ar:Landroid/os/AsyncResult;
      iget-object v2, v0, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
      if-nez v2, :cond_2
  
+     .line 640
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v3, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
***************
*** 351,356 ****
--- 381,387 ----
      #setter for: Lcom/android/internal/telephony/IccCard;->mIccPinLocked:Z
      invoke-static {v2, v3}, Lcom/android/internal/telephony/IccCard;->access$402(Lcom/android/internal/telephony/IccCard;Z)Z
  
+     .line 641
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-boolean v2, v2, Lcom/android/internal/telephony/IccCard;->mDbg:Z
***************
*** 387,392 ****
--- 418,424 ----
      #calls: Lcom/android/internal/telephony/IccCard;->log(Ljava/lang/String;)V
      invoke-static {v2, v3}, Lcom/android/internal/telephony/IccCard;->access$000(Lcom/android/internal/telephony/IccCard;Ljava/lang/String;)V
  
+     .line 647
      :cond_1
      :goto_1
      iget-object v2, v0, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
***************
*** 401,406 ****
--- 433,439 ----
  
      iput-object v3, v2, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
+     .line 649
      iget-object v2, v0, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
      check-cast v2, Landroid/os/Message;
***************
*** 409,414 ****
--- 442,448 ----
  
      goto/16 :goto_0
  
+     .line 644
      :cond_2
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 438,454 ****
--- 472,491 ----
  
      goto :goto_1
  
+     .line 652
      .end local v0           #ar:Landroid/os/AsyncResult;
      :pswitch_9
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/AsyncResult;
  
+     .line 654
      .restart local v0       #ar:Landroid/os/AsyncResult;
      iget-object v2, v0, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
      if-nez v2, :cond_4
  
+     .line 655
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v3, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
***************
*** 461,466 ****
--- 498,504 ----
      #setter for: Lcom/android/internal/telephony/IccCard;->mIccFdnEnabled:Z
      invoke-static {v2, v3}, Lcom/android/internal/telephony/IccCard;->access$602(Lcom/android/internal/telephony/IccCard;Z)Z
  
+     .line 656
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-boolean v2, v2, Lcom/android/internal/telephony/IccCard;->mDbg:Z
***************
*** 497,502 ****
--- 535,541 ----
      #calls: Lcom/android/internal/telephony/IccCard;->log(Ljava/lang/String;)V
      invoke-static {v2, v3}, Lcom/android/internal/telephony/IccCard;->access$000(Lcom/android/internal/telephony/IccCard;Ljava/lang/String;)V
  
+     .line 662
      :cond_3
      :goto_2
      iget-object v2, v0, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
***************
*** 511,516 ****
--- 550,556 ----
  
      iput-object v3, v2, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
+     .line 664
      iget-object v2, v0, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
      check-cast v2, Landroid/os/Message;
***************
*** 519,524 ****
--- 559,565 ----
  
      goto/16 :goto_0
  
+     .line 659
      :cond_4
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 548,564 ****
--- 589,608 ----
  
      goto :goto_2
  
+     .line 667
      .end local v0           #ar:Landroid/os/AsyncResult;
      :pswitch_a
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/AsyncResult;
  
+     .line 668
      .restart local v0       #ar:Landroid/os/AsyncResult;
      iget-object v2, v0, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
      if-eqz v2, :cond_5
  
+     .line 669
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v2, v2, Lcom/android/internal/telephony/IccCard;->mLogTag:Ljava/lang/String;
***************
*** 585,590 ****
--- 629,635 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 672
      :cond_5
      iget-object v2, v0, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
***************
*** 598,603 ****
--- 643,649 ----
  
      iput-object v3, v2, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
+     .line 674
      iget-object v2, v0, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
      check-cast v2, Landroid/os/Message;
***************
*** 606,611 ****
--- 652,658 ----
  
      goto/16 :goto_0
  
+     .line 677
      .end local v0           #ar:Landroid/os/AsyncResult;
      :pswitch_b
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
***************
*** 616,621 ****
--- 663,669 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 678
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
      iget-object v2, v2, Lcom/android/internal/telephony/IccCard;->mPhone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 630,635 ****
--- 678,684 ----
  
      goto/16 :goto_0
  
+     .line 681
      :pswitch_c
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 640,645 ****
--- 689,695 ----
  
      goto/16 :goto_0
  
+     .line 684
      :pswitch_d
      iget-object v2, p0, Lcom/android/internal/telephony/IccCard$2;->this$0:Lcom/android/internal/telephony/IccCard;
  
***************
*** 650,655 ****
--- 700,706 ----
  
      goto/16 :goto_0
  
+     .line 581
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_2
