*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccCardStatus$PinState.smali	2012-03-23 19:37:41.097524495 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccCardStatus$PinState.smali.original	2012-03-23 19:34:46.239335961 -0500
***************
*** 54,59 ****
--- 54,60 ----
  
      const/4 v3, 0x0
  
+     .line 40
      new-instance v0, Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      const-string v1, "PINSTATE_UNKNOWN"
***************
*** 62,67 ****
--- 63,69 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->PINSTATE_UNKNOWN:Lcom/android/internal/telephony/IccCardStatus$PinState;
  
+     .line 41
      new-instance v0, Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      const-string v1, "PINSTATE_ENABLED_NOT_VERIFIED"
***************
*** 70,75 ****
--- 72,78 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->PINSTATE_ENABLED_NOT_VERIFIED:Lcom/android/internal/telephony/IccCardStatus$PinState;
  
+     .line 42
      new-instance v0, Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      const-string v1, "PINSTATE_ENABLED_VERIFIED"
***************
*** 78,83 ****
--- 81,87 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->PINSTATE_ENABLED_VERIFIED:Lcom/android/internal/telephony/IccCardStatus$PinState;
  
+     .line 43
      new-instance v0, Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      const-string v1, "PINSTATE_DISABLED"
***************
*** 86,91 ****
--- 90,96 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->PINSTATE_DISABLED:Lcom/android/internal/telephony/IccCardStatus$PinState;
  
+     .line 44
      new-instance v0, Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      const-string v1, "PINSTATE_ENABLED_BLOCKED"
***************
*** 94,99 ****
--- 99,105 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->PINSTATE_ENABLED_BLOCKED:Lcom/android/internal/telephony/IccCardStatus$PinState;
  
+     .line 45
      new-instance v0, Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      const-string v1, "PINSTATE_ENABLED_PERM_BLOCKED"
***************
*** 104,109 ****
--- 110,116 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->PINSTATE_ENABLED_PERM_BLOCKED:Lcom/android/internal/telephony/IccCardStatus$PinState;
  
+     .line 39
      const/4 v0, 0x6
  
      new-array v0, v0, [Lcom/android/internal/telephony/IccCardStatus$PinState;
***************
*** 150,155 ****
--- 157,163 ----
      .end annotation
  
      .prologue
+     .line 39
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 160,165 ****
--- 168,174 ----
      .parameter "name"
  
      .prologue
+     .line 39
      const-class v0, Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 175,180 ****
--- 184,190 ----
      .locals 1
  
      .prologue
+     .line 39
      sget-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->$VALUES:[Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/IccCardStatus$PinState;->clone()Ljava/lang/Object;
***************
*** 192,197 ****
--- 202,208 ----
      .locals 1
  
      .prologue
+     .line 48
      sget-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->PINSTATE_ENABLED_PERM_BLOCKED:Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      if-ne p0, v0, :cond_0
***************
*** 211,216 ****
--- 222,228 ----
      .locals 1
  
      .prologue
+     .line 52
      sget-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->PINSTATE_ENABLED_NOT_VERIFIED:Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      if-ne p0, v0, :cond_0
***************
*** 230,235 ****
--- 242,248 ----
      .locals 1
  
      .prologue
+     .line 56
      sget-object v0, Lcom/android/internal/telephony/IccCardStatus$PinState;->PINSTATE_ENABLED_BLOCKED:Lcom/android/internal/telephony/IccCardStatus$PinState;
  
      if-ne p0, v0, :cond_0
