*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccCardApplication$PersoSubState.smali	2012-03-23 19:37:41.700535491 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccCardApplication$PersoSubState.smali.original	2012-03-23 19:34:46.249336143 -0500
***************
*** 92,97 ****
--- 92,98 ----
  
      const/4 v3, 0x0
  
+     .line 68
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_UNKNOWN"
***************
*** 100,105 ****
--- 101,107 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_UNKNOWN:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 69
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_IN_PROGRESS"
***************
*** 108,113 ****
--- 110,116 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_IN_PROGRESS:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 70
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_READY"
***************
*** 116,121 ****
--- 119,125 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_READY:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 71
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_NETWORK"
***************
*** 124,129 ****
--- 128,134 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_NETWORK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 72
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_NETWORK_SUBSET"
***************
*** 132,137 ****
--- 137,143 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_NETWORK_SUBSET:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 73
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_CORPORATE"
***************
*** 142,147 ****
--- 148,154 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_CORPORATE:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 74
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_SERVICE_PROVIDER"
***************
*** 152,157 ****
--- 159,165 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_SERVICE_PROVIDER:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 75
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_SIM"
***************
*** 162,167 ****
--- 170,176 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_SIM:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 76
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_NETWORK_PUK"
***************
*** 172,177 ****
--- 181,187 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_NETWORK_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 77
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_NETWORK_SUBSET_PUK"
***************
*** 182,187 ****
--- 192,198 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_NETWORK_SUBSET_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 78
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_CORPORATE_PUK"
***************
*** 192,197 ****
--- 203,209 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_CORPORATE_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 79
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_SERVICE_PROVIDER_PUK"
***************
*** 202,207 ****
--- 214,220 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_SERVICE_PROVIDER_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 80
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_SIM_SIM_PUK"
***************
*** 212,217 ****
--- 225,231 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_SIM_SIM_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 81
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_NETWORK1"
***************
*** 222,227 ****
--- 236,242 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_NETWORK1:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 82
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_NETWORK2"
***************
*** 232,237 ****
--- 247,253 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_NETWORK2:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 83
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_HRPD"
***************
*** 242,247 ****
--- 258,264 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_HRPD:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 84
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_CORPORATE"
***************
*** 252,257 ****
--- 269,275 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_CORPORATE:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 85
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_SERVICE_PROVIDER"
***************
*** 262,267 ****
--- 280,286 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_SERVICE_PROVIDER:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 86
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_RUIM"
***************
*** 272,277 ****
--- 291,297 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_RUIM:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 87
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_NETWORK1_PUK"
***************
*** 282,287 ****
--- 302,308 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_NETWORK1_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 88
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_NETWORK2_PUK"
***************
*** 292,297 ****
--- 313,319 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_NETWORK2_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 89
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_HRPD_PUK"
***************
*** 302,307 ****
--- 324,330 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_HRPD_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 90
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_CORPORATE_PUK"
***************
*** 312,317 ****
--- 335,341 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_CORPORATE_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 91
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_SERVICE_PROVIDER_PUK"
***************
*** 322,327 ****
--- 346,352 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_SERVICE_PROVIDER_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 92
      new-instance v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      const-string v1, "PERSOSUBSTATE_RUIM_RUIM_PUK"
***************
*** 332,337 ****
--- 357,363 ----
  
      sput-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_RUIM_RUIM_PUK:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
+     .line 67
      const/16 v0, 0x19
  
      new-array v0, v0, [Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
***************
*** 492,497 ****
--- 518,524 ----
      .end annotation
  
      .prologue
+     .line 67
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 502,507 ****
--- 529,535 ----
      .parameter "name"
  
      .prologue
+     .line 67
      const-class v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 517,522 ****
--- 545,551 ----
      .locals 1
  
      .prologue
+     .line 67
      sget-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->$VALUES:[Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->clone()Ljava/lang/Object;
***************
*** 534,539 ****
--- 563,569 ----
      .locals 1
  
      .prologue
+     .line 95
      sget-object v0, Lcom/android/internal/telephony/IccCardApplication$PersoSubState;->PERSOSUBSTATE_UNKNOWN:Lcom/android/internal/telephony/IccCardApplication$PersoSubState;
  
      if-ne p0, v0, :cond_0
