*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccFileHandler$LoadLinearFixedContext.smali	2012-03-23 19:37:45.714608687 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccFileHandler$LoadLinearFixedContext.smali.original	2012-03-23 19:34:46.319337419 -0500
***************
*** 45,62 ****
--- 45,68 ----
      .parameter "onLoaded"
  
      .prologue
+     .line 105
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 106
      iput p1, p0, Lcom/android/internal/telephony/IccFileHandler$LoadLinearFixedContext;->efid:I
  
+     .line 107
      iput p2, p0, Lcom/android/internal/telephony/IccFileHandler$LoadLinearFixedContext;->recordNum:I
  
+     .line 108
      iput-object p3, p0, Lcom/android/internal/telephony/IccFileHandler$LoadLinearFixedContext;->onLoaded:Landroid/os/Message;
  
+     .line 109
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/internal/telephony/IccFileHandler$LoadLinearFixedContext;->loadAll:Z
  
+     .line 110
      return-void
  .end method
  
***************
*** 68,82 ****
--- 74,94 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 112
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 113
      iput p1, p0, Lcom/android/internal/telephony/IccFileHandler$LoadLinearFixedContext;->efid:I
  
+     .line 114
      iput v0, p0, Lcom/android/internal/telephony/IccFileHandler$LoadLinearFixedContext;->recordNum:I
  
+     .line 115
      iput-boolean v0, p0, Lcom/android/internal/telephony/IccFileHandler$LoadLinearFixedContext;->loadAll:Z
  
+     .line 116
      iput-object p2, p0, Lcom/android/internal/telephony/IccFileHandler$LoadLinearFixedContext;->onLoaded:Landroid/os/Message;
  
+     .line 117
      return-void
  .end method
