*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccIoResult.smali	2012-03-23 19:37:37.361456368 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccIoResult.smali.original	2012-03-23 19:34:46.165334611 -0500
***************
*** 19,30 ****
--- 19,32 ----
      .parameter "hexString"
  
      .prologue
+     .line 36
      invoke-static {p3}, Lcom/android/internal/telephony/IccUtils;->hexStringToBytes(Ljava/lang/String;)[B
  
      move-result-object v0
  
      invoke-direct {p0, p1, p2, v0}, Lcom/android/internal/telephony/IccIoResult;-><init>(II[B)V
  
+     .line 37
      return-void
  .end method
  
***************
*** 35,48 ****
--- 37,55 ----
      .parameter "payload"
  
      .prologue
+     .line 29
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 30
      iput p1, p0, Lcom/android/internal/telephony/IccIoResult;->sw1:I
  
+     .line 31
      iput p2, p0, Lcom/android/internal/telephony/IccIoResult;->sw2:I
  
+     .line 32
      iput-object p3, p0, Lcom/android/internal/telephony/IccIoResult;->payload:[B
  
+     .line 33
      return-void
  .end method
  
***************
*** 52,57 ****
--- 59,65 ----
      .locals 3
  
      .prologue
+     .line 57
      invoke-virtual {p0}, Lcom/android/internal/telephony/IccIoResult;->success()Z
  
      move-result v0
***************
*** 60,80 ****
  
      const/4 v0, 0x0
  
      :goto_0
      return-object v0
  
      :cond_0
      iget v0, p0, Lcom/android/internal/telephony/IccIoResult;->sw1:I
  
      packed-switch v0, :pswitch_data_0
  
      new-instance v0, Lcom/android/internal/telephony/IccException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "sw1:"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 68,91 ----
  
      const/4 v0, 0x0
  
+     .line 67
      :goto_0
      return-object v0
  
+     .line 59
      :cond_0
      iget v0, p0, Lcom/android/internal/telephony/IccIoResult;->sw1:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 67
      new-instance v0, Lcom/android/internal/telephony/IccException;
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "sw1:"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 106,111 ****
--- 117,123 ----
  
      goto :goto_0
  
+     .line 61
      :pswitch_0
      iget v0, p0, Lcom/android/internal/telephony/IccIoResult;->sw2:I
  
***************
*** 113,124 ****
--- 125,138 ----
  
      if-ne v0, v1, :cond_1
  
+     .line 62
      new-instance v0, Lcom/android/internal/telephony/IccFileTypeMismatch;
  
      invoke-direct {v0}, Lcom/android/internal/telephony/IccFileTypeMismatch;-><init>()V
  
      goto :goto_0
  
+     .line 64
      :cond_1
      new-instance v0, Lcom/android/internal/telephony/IccFileNotFound;
  
***************
*** 126,131 ****
--- 140,146 ----
  
      goto :goto_0
  
+     .line 59
      nop
  
      :pswitch_data_0
***************
*** 138,143 ****
--- 153,159 ----
      .locals 2
  
      .prologue
+     .line 50
      iget v0, p0, Lcom/android/internal/telephony/IccIoResult;->sw1:I
  
      const/16 v1, 0x90
***************
*** 178,183 ****
--- 194,200 ----
      .locals 2
  
      .prologue
+     .line 40
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
