*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccVmFixedException.smali	2012-03-23 19:37:45.681608086 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IccVmFixedException.smali.original	2012-03-23 19:34:46.319337419 -0500
***************
*** 8,15 ****
--- 8,17 ----
      .locals 0
  
      .prologue
+     .line 24
      invoke-direct {p0}, Lcom/android/internal/telephony/IccException;-><init>()V
  
+     .line 26
      return-void
  .end method
  
***************
*** 18,24 ****
--- 20,28 ----
      .parameter "s"
  
      .prologue
+     .line 30
      invoke-direct {p0, p1}, Lcom/android/internal/telephony/IccException;-><init>(Ljava/lang/String;)V
  
+     .line 31
      return-void
  .end method
