*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IExtendedNetworkService$Stub$Proxy.smali	2012-03-23 19:37:46.000613903 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/IExtendedNetworkService$Stub$Proxy.smali.original	2012-03-23 19:34:46.322337473 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 106
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 107
      iput-object p1, p0, Lcom/android/internal/telephony/IExtendedNetworkService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 108
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 111
      iget-object v0, p0, Lcom/android/internal/telephony/IExtendedNetworkService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 54,74 ****
--- 58,82 ----
      .end annotation
  
      .prologue
+     .line 202
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 203
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 205
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.IExtendedNetworkService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 206
      iget-object v2, p0, Lcom/android/internal/telephony/IExtendedNetworkService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x4
***************
*** 77,97 ****
--- 85,111 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 207
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 210
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 211
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 213
      return-void
  
+     .line 210
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 211
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 101,106 ****
--- 115,121 ----
      .locals 1
  
      .prologue
+     .line 115
      const-string v0, "com.android.internal.telephony.IExtendedNetworkService"
  
      return-object v0
***************
*** 115,135 ****
--- 130,154 ----
      .end annotation
  
      .prologue
+     .line 142
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 143
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 146
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.IExtendedNetworkService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 147
      iget-object v3, p0, Lcom/android/internal/telephony/IExtendedNetworkService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x2
***************
*** 138,151 ****
--- 157,173 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 148
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 149
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 150
      sget-object v3, Landroid/text/TextUtils;->CHAR_SEQUENCE_CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 156,169 ****
--- 178,195 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 157
      .local v2, _result:Ljava/lang/CharSequence;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 158
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 160
      return-object v2
  
+     .line 153
      .end local v2           #_result:Ljava/lang/CharSequence;
      :cond_0
      const/4 v2, 0x0
***************
*** 171,182 ****
--- 197,210 ----
      .restart local v2       #_result:Ljava/lang/CharSequence;
      goto :goto_0
  
+     .line 157
      .end local v2           #_result:Ljava/lang/CharSequence;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 158
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 192,222 ****
--- 220,257 ----
      .end annotation
  
      .prologue
+     .line 169
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 170
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 173
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.IExtendedNetworkService"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 174
      if-eqz p1, :cond_0
  
+     .line 175
      const/4 v3, 0x1
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 176
      const/4 v3, 0x0
  
      invoke-static {p1, v0, v3}, Landroid/text/TextUtils;->writeToParcel(Ljava/lang/CharSequence;Landroid/os/Parcel;I)V
  
+     .line 181
      :goto_0
      iget-object v3, p0, Lcom/android/internal/telephony/IExtendedNetworkService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 226,239 ****
--- 261,277 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 182
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 183
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_1
  
+     .line 184
      sget-object v3, Landroid/text/TextUtils;->CHAR_SEQUENCE_CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 244,257 ****
--- 282,299 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 191
      .local v2, _result:Ljava/lang/CharSequence;
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 192
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 194
      return-object v2
  
+     .line 179
      .end local v2           #_result:Ljava/lang/CharSequence;
      :cond_0
      const/4 v3, 0x0
***************
*** 263,277 ****
--- 305,322 ----
  
      goto :goto_0
  
+     .line 191
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 192
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
  
+     .line 187
      :cond_1
      const/4 v2, 0x0
  
***************
*** 289,311 ****
--- 334,361 ----
      .end annotation
  
      .prologue
+     .line 124
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 125
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 127
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.IExtendedNetworkService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 128
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 129
      iget-object v2, p0, Lcom/android/internal/telephony/IExtendedNetworkService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x1
***************
*** 314,334 ****
--- 364,390 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 130
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 133
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 134
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 136
      return-void
  
+     .line 133
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 134
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
