*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/ISms$Stub$Proxy.smali	2012-03-23 19:37:42.854556534 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/ISms$Stub$Proxy.smali.original	2012-03-23 19:34:46.267336471 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 218
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 219
      iput-object p1, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 220
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 223
      iget-object v0, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 59,85 ****
--- 63,96 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 293
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 294
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 297
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 298
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 299
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeByteArray([B)V
  
+     .line 300
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeByteArray([B)V
  
+     .line 301
      iget-object v3, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x3
***************
*** 88,95 ****
--- 99,108 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 302
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 303
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 100,119 ****
--- 113,137 ----
  
      const/4 v2, 0x1
  
+     .line 306
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 307
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 309
      return v2
  
+     .line 306
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 307
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 131,153 ****
--- 149,176 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 503
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 504
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 507
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 508
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 509
      iget-object v3, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x8
***************
*** 156,163 ****
--- 179,188 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 510
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 511
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 168,187 ****
--- 193,217 ----
  
      const/4 v2, 0x1
  
+     .line 514
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 515
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 517
      return v2
  
+     .line 514
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 515
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 200,224 ****
--- 230,260 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 564
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 565
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 568
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 569
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 570
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 571
      iget-object v3, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xa
***************
*** 227,234 ****
--- 263,272 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 572
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 573
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 239,258 ****
--- 277,301 ----
  
      const/4 v2, 0x1
  
+     .line 576
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 577
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 579
      return v2
  
+     .line 576
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 577
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 270,292 ****
--- 313,340 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 474
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 475
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 478
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 479
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 480
      iget-object v3, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x7
***************
*** 295,302 ****
--- 343,352 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 481
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 482
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 307,326 ****
--- 357,381 ----
  
      const/4 v2, 0x1
  
+     .line 485
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 486
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 488
      return v2
  
+     .line 485
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 486
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 339,363 ****
--- 394,424 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 533
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 534
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 537
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 538
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 539
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 540
      iget-object v3, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x9
***************
*** 366,373 ****
--- 427,436 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 541
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 542
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 378,397 ****
--- 441,465 ----
  
      const/4 v2, 0x1
  
+     .line 545
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 546
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 548
      return v2
  
+     .line 545
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 546
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 416,436 ****
--- 484,508 ----
      .end annotation
  
      .prologue
+     .line 236
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 237
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 240
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 241
      iget-object v3, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x1
***************
*** 439,446 ****
--- 511,520 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 242
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 243
      sget-object v3, Lcom/android/internal/telephony/SmsRawData;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-virtual {v1, v3}, Landroid/os/Parcel;->createTypedArrayList(Landroid/os/Parcelable$Creator;)Ljava/util/ArrayList;
***************
*** 449,467 ****
--- 523,546 ----
  
      move-result-object v2
  
+     .line 246
      .local v2, _result:Ljava/util/List;,"Ljava/util/List<Lcom/android/internal/telephony/SmsRawData;>;"
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 247
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 249
      return-object v2
  
+     .line 246
      .end local v2           #_result:Ljava/util/List;,"Ljava/util/List<Lcom/android/internal/telephony/SmsRawData;>;"
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 247
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 471,476 ****
--- 550,556 ----
      .locals 1
  
      .prologue
+     .line 227
      const-string v0, "com.android.internal.telephony.ISms"
  
      return-object v0
***************
*** 491,540 ****
--- 571,634 ----
      .end annotation
  
      .prologue
+     .line 336
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 337
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 339
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 340
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 341
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 342
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 343
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeByteArray([B)V
  
+     .line 344
      if-eqz p5, :cond_0
  
+     .line 345
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 346
      const/4 v2, 0x0
  
      invoke-virtual {p5, v0, v2}, Landroid/app/PendingIntent;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 351
      :goto_0
      if-eqz p6, :cond_1
  
+     .line 352
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 353
      const/4 v2, 0x0
  
      invoke-virtual {p6, v0, v2}, Landroid/app/PendingIntent;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 358
      :goto_1
      iget-object v2, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 544,559 ****
--- 638,658 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 359
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 362
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 363
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 365
      return-void
  
+     .line 349
      :cond_0
      const/4 v2, 0x0
  
***************
*** 564,578 ****
--- 663,680 ----
  
      goto :goto_0
  
+     .line 362
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 363
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
  
+     .line 356
      :cond_1
      const/4 v2, 0x0
  
***************
*** 618,623 ****
--- 720,726 ----
      .end annotation
  
      .prologue
+     .line 444
      .local p3, parts:Ljava/util/List;,"Ljava/util/List<Ljava/lang/String;>;"
      .local p4, sentIntents:Ljava/util/List;,"Ljava/util/List<Landroid/app/PendingIntent;>;"
      .local p5, deliveryIntents:Ljava/util/List;,"Ljava/util/List<Landroid/app/PendingIntent;>;"
***************
*** 625,651 ****
--- 728,762 ----
  
      move-result-object v0
  
+     .line 445
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 447
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 448
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 449
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 450
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeStringList(Ljava/util/List;)V
  
+     .line 451
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeTypedList(Ljava/util/List;)V
  
+     .line 452
      invoke-virtual {v0, p5}, Landroid/os/Parcel;->writeTypedList(Ljava/util/List;)V
  
+     .line 453
      iget-object v2, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x6
***************
*** 654,674 ****
--- 765,791 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 454
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 457
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 458
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 460
      return-void
  
+     .line 457
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 458
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 688,735 ****
--- 805,865 ----
      .end annotation
  
      .prologue
+     .line 391
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 392
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 394
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 395
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 396
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 397
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 398
      if-eqz p4, :cond_0
  
+     .line 399
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 400
      const/4 v2, 0x0
  
      invoke-virtual {p4, v0, v2}, Landroid/app/PendingIntent;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 405
      :goto_0
      if-eqz p5, :cond_1
  
+     .line 406
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 407
      const/4 v2, 0x0
  
      invoke-virtual {p5, v0, v2}, Landroid/app/PendingIntent;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 412
      :goto_1
      iget-object v2, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 739,754 ****
--- 869,889 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 413
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 416
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 417
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 419
      return-void
  
+     .line 403
      :cond_0
      const/4 v2, 0x0
  
***************
*** 759,773 ****
--- 894,911 ----
  
      goto :goto_0
  
+     .line 416
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 417
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
  
+     .line 410
      :cond_1
      const/4 v2, 0x0
  
***************
*** 793,819 ****
--- 931,964 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 264
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 265
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 268
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ISms"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 269
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 270
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 271
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeByteArray([B)V
  
+     .line 272
      iget-object v3, p0, Lcom/android/internal/telephony/ISms$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x2
***************
*** 822,829 ****
--- 967,976 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 273
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 274
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 834,853 ****
--- 981,1005 ----
  
      const/4 v2, 0x1
  
+     .line 277
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 278
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 280
      return v2
  
+     .line 277
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 278
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
