*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/ITelephony$Stub$Proxy.smali	2012-03-23 19:37:41.521532227 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/ITelephony$Stub$Proxy.smali.original	2012-03-23 19:34:46.246336087 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 396
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 397
      iput-object p1, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 398
      return-void
  .end method
  
***************
*** 45,65 ****
--- 48,72 ----
      .end annotation
  
      .prologue
+     .line 539
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 540
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 542
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 543
      iget-object v2, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x6
***************
*** 68,88 ****
--- 75,101 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 544
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 547
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 548
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 550
      return-void
  
+     .line 547
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 548
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 92,97 ****
--- 105,111 ----
      .locals 1
  
      .prologue
+     .line 401
      iget-object v0, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 107,129 ****
--- 121,148 ----
      .end annotation
  
      .prologue
+     .line 434
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 435
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 437
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 438
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 439
      iget-object v2, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x2
***************
*** 132,152 ****
--- 151,177 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 440
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 443
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 444
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 446
      return-void
  
+     .line 443
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 444
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 161,181 ****
--- 186,210 ----
      .end annotation
  
      .prologue
+     .line 686
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 687
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 689
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 690
      iget-object v2, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0xd
***************
*** 184,204 ****
--- 213,239 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 691
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 694
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 695
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 697
      return-void
  
+     .line 694
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 695
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 214,236 ****
--- 249,276 ----
      .end annotation
  
      .prologue
+     .line 415
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 416
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 418
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 419
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 420
      iget-object v2, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x1
***************
*** 239,259 ****
--- 279,305 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 421
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 424
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 425
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 427
      return-void
  
+     .line 424
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 425
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 269,291 ****
--- 315,342 ----
      .end annotation
  
      .prologue
+     .line 887
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 888
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 891
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 892
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 893
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x17
***************
*** 294,320 ****
--- 345,378 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 894
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 895
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 898
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 899
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 901
      return v2
  
+     .line 898
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 899
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 331,351 ****
--- 389,413 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 928
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 929
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 932
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 933
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x19
***************
*** 354,361 ****
--- 416,425 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 934
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 935
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 366,385 ****
--- 430,454 ----
  
      const/4 v2, 0x1
  
+     .line 938
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 939
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 941
      return v2
  
+     .line 938
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 939
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 394,414 ****
--- 463,487 ----
      .end annotation
  
      .prologue
+     .line 849
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 850
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 852
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 853
      iget-object v2, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x15
***************
*** 417,437 ****
--- 490,516 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 854
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 857
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 858
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 860
      return-void
  
+     .line 857
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 858
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 447,469 ****
--- 526,553 ----
      .end annotation
  
      .prologue
+     .line 866
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 867
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 870
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 871
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 872
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x16
***************
*** 472,498 ****
--- 556,589 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 873
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 874
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 877
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 878
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 880
      return v2
  
+     .line 877
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 878
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 509,529 ****
--- 600,624 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 908
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 909
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 912
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 913
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x18
***************
*** 532,539 ****
--- 627,636 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 914
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 915
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 544,563 ****
--- 641,665 ----
  
      const/4 v2, 0x1
  
+     .line 918
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 919
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 921
      return v2
  
+     .line 918
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 919
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 572,592 ****
--- 674,698 ----
      .end annotation
  
      .prologue
+     .line 832
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 833
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 835
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 836
      iget-object v2, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x14
***************
*** 595,615 ****
--- 701,727 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 837
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 840
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 841
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 843
      return-void
  
+     .line 840
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 841
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 626,646 ****
--- 738,762 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 507
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 508
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 511
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 512
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x5
***************
*** 649,656 ****
--- 765,774 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 513
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 514
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 661,680 ****
--- 779,803 ----
  
      const/4 v2, 0x1
  
+     .line 517
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 518
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 520
      return v2
  
+     .line 517
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 518
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 689,709 ****
--- 812,836 ----
      .end annotation
  
      .prologue
+     .line 1063
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1064
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1067
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1068
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x20
***************
*** 712,738 ****
--- 839,872 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1069
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1070
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 1073
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1074
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1076
      return v2
  
+     .line 1073
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1074
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 747,767 ****
--- 881,905 ----
      .end annotation
  
      .prologue
+     .line 1007
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1008
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1011
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1012
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x1d
***************
*** 770,796 ****
--- 908,941 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1013
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1014
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 1017
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1018
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1020
      return v2
  
+     .line 1017
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1018
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 805,825 ****
--- 950,974 ----
      .end annotation
  
      .prologue
+     .line 1083
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1084
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1087
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1088
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x21
***************
*** 828,854 ****
--- 977,1010 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1089
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1090
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 1093
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1094
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1096
      return v2
  
+     .line 1093
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1094
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 863,883 ****
--- 1019,1043 ----
      .end annotation
  
      .prologue
+     .line 1105
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1106
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1109
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1110
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x22
***************
*** 886,912 ****
--- 1046,1079 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1111
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1112
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 1115
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1116
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1118
      return v2
  
+     .line 1115
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1116
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 921,941 ****
--- 1088,1112 ----
      .end annotation
  
      .prologue
+     .line 1125
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1126
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1129
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1130
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x23
***************
*** 944,970 ****
--- 1115,1148 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1131
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1132
      invoke-virtual {v1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result-object v2
  
+     .line 1135
      .local v2, _result:Ljava/lang/String;
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1136
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1138
      return-object v2
  
+     .line 1135
      .end local v2           #_result:Ljava/lang/String;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1136
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 979,999 ****
--- 1157,1181 ----
      .end annotation
  
      .prologue
+     .line 965
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 966
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 969
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 970
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x1b
***************
*** 1002,1015 ****
--- 1184,1200 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 971
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 972
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 973
      sget-object v3, Landroid/os/Bundle;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, v1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 1020,1033 ****
--- 1205,1222 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 980
      .local v2, _result:Landroid/os/Bundle;
      :goto_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 981
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 983
      return-object v2
  
+     .line 976
      .end local v2           #_result:Landroid/os/Bundle;
      :cond_0
      const/4 v2, 0x0
***************
*** 1035,1046 ****
--- 1224,1237 ----
      .restart local v2       #_result:Landroid/os/Bundle;
      goto :goto_0
  
+     .line 980
      .end local v2           #_result:Landroid/os/Bundle;
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 981
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1055,1075 ****
--- 1246,1270 ----
      .end annotation
  
      .prologue
+     .line 1024
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1025
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1028
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1029
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x1e
***************
*** 1078,1104 ****
--- 1273,1306 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1030
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1031
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 1034
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1035
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1037
      return v2
  
+     .line 1034
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1035
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1113,1133 ****
--- 1315,1339 ----
      .end annotation
  
      .prologue
+     .line 1041
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1042
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1045
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1046
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x1f
***************
*** 1136,1162 ****
--- 1342,1375 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1047
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1048
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 1051
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1052
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1054
      return v2
  
+     .line 1051
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1052
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1166,1171 ****
--- 1379,1385 ----
      .locals 1
  
      .prologue
+     .line 405
      const-string v0, "com.android.internal.telephony.ITelephony"
  
      return-object v0
***************
*** 1180,1200 ****
--- 1394,1418 ----
      .end annotation
  
      .prologue
+     .line 1232
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1233
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1236
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1237
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x28
***************
*** 1203,1229 ****
--- 1421,1454 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1238
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1239
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 1242
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1243
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1245
      return v2
  
+     .line 1242
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1243
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1248,1268 ****
--- 1473,1497 ----
      .end annotation
  
      .prologue
+     .line 990
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 991
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 994
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 995
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x1c
***************
*** 1271,1278 ****
--- 1500,1509 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 996
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 997
      sget-object v3, Landroid/telephony/NeighboringCellInfo;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-virtual {v1, v3}, Landroid/os/Parcel;->createTypedArrayList(Landroid/os/Parcelable$Creator;)Ljava/util/ArrayList;
***************
*** 1281,1299 ****
--- 1512,1535 ----
  
      move-result-object v2
  
+     .line 1000
      .local v2, _result:Ljava/util/List;,"Ljava/util/List<Landroid/telephony/NeighboringCellInfo;>;"
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1001
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1003
      return-object v2
  
+     .line 1000
      .end local v2           #_result:Ljava/util/List;,"Ljava/util/List<Landroid/telephony/NeighboringCellInfo;>;"
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1001
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1308,1328 ****
--- 1544,1568 ----
      .end annotation
  
      .prologue
+     .line 1187
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1188
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1191
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1192
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x26
***************
*** 1331,1357 ****
--- 1571,1604 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1193
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1194
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 1197
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1198
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1200
      return v2
  
+     .line 1197
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1198
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1366,1386 ****
--- 1613,1637 ----
      .end annotation
  
      .prologue
+     .line 1167
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1168
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1171
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1172
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x25
***************
*** 1389,1415 ****
--- 1640,1673 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1173
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1174
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      move-result v2
  
+     .line 1177
      .local v2, _result:I
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1178
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1180
      return v2
  
+     .line 1177
      .end local v2           #_result:I
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1178
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1427,1449 ****
--- 1685,1712 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 756
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 757
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 760
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 761
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 762
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x10
***************
*** 1452,1459 ****
--- 1715,1724 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 763
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 764
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1464,1483 ****
--- 1729,1753 ----
  
      const/4 v2, 0x1
  
+     .line 767
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 768
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 770
      return v2
  
+     .line 767
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 768
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1494,1514 ****
--- 1764,1788 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1207
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1208
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1211
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1212
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x27
***************
*** 1517,1524 ****
--- 1791,1800 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1213
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1214
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1529,1548 ****
--- 1805,1829 ----
  
      const/4 v2, 0x1
  
+     .line 1217
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1218
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1220
      return v2
  
+     .line 1217
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1218
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1559,1579 ****
--- 1840,1864 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 948
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 949
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 952
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 953
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x1a
***************
*** 1582,1589 ****
--- 1867,1876 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 954
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 955
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1594,1613 ****
--- 1881,1905 ----
  
      const/4 v2, 0x1
  
+     .line 958
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 959
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 961
      return v2
  
+     .line 958
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 959
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1624,1644 ****
--- 1916,1940 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 624
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 625
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 628
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 629
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xa
***************
*** 1647,1654 ****
--- 1943,1952 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 630
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 631
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1659,1678 ****
--- 1957,1981 ----
  
      const/4 v2, 0x1
  
+     .line 634
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 635
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 637
      return v2
  
+     .line 634
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 635
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1689,1709 ****
--- 1992,2016 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 582
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 583
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 586
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 587
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x8
***************
*** 1712,1719 ****
--- 2019,2028 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 588
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 589
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1724,1743 ****
--- 2033,2057 ----
  
      const/4 v2, 0x1
  
+     .line 592
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 593
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 595
      return v2
  
+     .line 592
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 593
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1754,1774 ****
--- 2068,2092 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 645
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 646
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 649
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 650
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xb
***************
*** 1777,1784 ****
--- 2095,2104 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 651
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 652
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1789,1808 ****
--- 2109,2133 ----
  
      const/4 v2, 0x1
  
+     .line 655
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 656
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 658
      return v2
  
+     .line 655
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 656
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1819,1839 ****
--- 2144,2168 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 603
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 604
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 607
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 608
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x9
***************
*** 1842,1849 ****
--- 2171,2180 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 609
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 610
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1854,1873 ****
--- 2185,2209 ----
  
      const/4 v2, 0x1
  
+     .line 613
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 614
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 616
      return v2
  
+     .line 613
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 614
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1884,1904 ****
--- 2220,2244 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 666
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 667
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 670
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 671
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xc
***************
*** 1907,1914 ****
--- 2247,2256 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 672
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 673
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1919,1938 ****
--- 2261,2285 ----
  
      const/4 v2, 0x1
  
+     .line 676
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 677
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 679
      return v2
  
+     .line 676
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 677
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 1949,1969 ****
--- 2296,2320 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1147
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 1148
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 1151
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 1152
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0x24
***************
*** 1972,1979 ****
--- 2323,2332 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 1153
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 1154
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 1984,2003 ****
--- 2337,2361 ----
  
      const/4 v2, 0x1
  
+     .line 1157
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1158
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 1160
      return v2
  
+     .line 1157
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 1158
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 2017,2037 ****
--- 2375,2399 ----
  
      const/4 v3, 0x0
  
+     .line 794
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 795
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 798
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 799
      if-eqz p1, :cond_0
  
      move v4, v2
***************
*** 2039,2044 ****
--- 2401,2407 ----
      :goto_0
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 800
      iget-object v4, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v5, 0x12
***************
*** 2047,2054 ****
--- 2410,2419 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 801
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 802
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 2057,2086 ****
--- 2422,2458 ----
  
      if-eqz v4, :cond_1
  
+     .line 805
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 806
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 808
      return v2
  
      .end local v2           #_result:Z
      :cond_0
      move v4, v3
  
+     .line 799
      goto :goto_0
  
      :cond_1
      move v2, v3
  
+     .line 802
      goto :goto_1
  
+     .line 805
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 806
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 2097,2117 ****
--- 2469,2493 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 456
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 457
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 460
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 461
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v4, 0x3
***************
*** 2120,2127 ****
--- 2496,2505 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 462
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 463
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 2132,2151 ****
--- 2510,2534 ----
  
      const/4 v2, 0x1
  
+     .line 466
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 467
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 469
      return v2
  
+     .line 466
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 467
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 2165,2185 ****
--- 2548,2572 ----
  
      const/4 v3, 0x0
  
+     .line 484
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 485
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 488
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v4, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 489
      if-eqz p1, :cond_0
  
      move v4, v2
***************
*** 2187,2192 ****
--- 2574,2580 ----
      :goto_0
      invoke-virtual {v0, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 490
      iget-object v4, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v5, 0x4
***************
*** 2195,2202 ****
--- 2583,2592 ----
  
      invoke-interface {v4, v5, v0, v1, v6}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 491
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 492
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 2205,2234 ****
--- 2595,2631 ----
  
      if-eqz v4, :cond_1
  
+     .line 495
      .local v2, _result:Z
      :goto_1
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 496
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 498
      return v2
  
      .end local v2           #_result:Z
      :cond_0
      move v4, v3
  
+     .line 489
      goto :goto_0
  
      :cond_1
      move v2, v3
  
+     .line 492
      goto :goto_1
  
+     .line 495
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 496
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 2243,2263 ****
--- 2640,2664 ----
      .end annotation
  
      .prologue
+     .line 564
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 565
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 567
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 568
      iget-object v2, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x7
***************
*** 2266,2286 ****
--- 2667,2693 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 569
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 572
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 573
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 575
      return-void
  
+     .line 572
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 573
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 2298,2320 ****
--- 2705,2732 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 705
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 706
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 709
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 710
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 711
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xe
***************
*** 2323,2330 ****
--- 2735,2744 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 712
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 713
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 2335,2354 ****
--- 2749,2773 ----
  
      const/4 v2, 0x1
  
+     .line 716
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 717
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 719
      return v2
  
+     .line 716
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 717
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 2367,2391 ****
--- 2786,2816 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 730
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 731
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 734
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v3, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 735
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 736
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 737
      iget-object v3, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v4, 0xf
***************
*** 2394,2401 ****
--- 2819,2828 ----
  
      invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 738
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 739
      invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 2406,2425 ****
--- 2833,2857 ----
  
      const/4 v2, 0x1
  
+     .line 742
      .local v2, _result:Z
      :cond_0
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 743
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 745
      return v2
  
+     .line 742
      .end local v2           #_result:Z
      :catchall_0
      move-exception v3
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 743
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v3
***************
*** 2434,2454 ****
--- 2866,2890 ----
      .end annotation
  
      .prologue
+     .line 777
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 778
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 780
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 781
      iget-object v2, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x11
***************
*** 2457,2477 ****
--- 2893,2919 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 782
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 785
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 786
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 788
      return-void
  
+     .line 785
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 786
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 2486,2506 ****
--- 2928,2952 ----
      .end annotation
  
      .prologue
+     .line 815
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 816
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 818
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.telephony.ITelephony"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 819
      iget-object v2, p0, Lcom/android/internal/telephony/ITelephony$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x13
***************
*** 2509,2529 ****
--- 2955,2981 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 820
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 823
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 824
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 826
      return-void
  
+     .line 823
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 824
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
