*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/MmiCode$State.smali	2012-03-23 19:37:45.861611369 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/MmiCode$State.smali.original	2012-03-23 19:34:46.321337455 -0500
***************
*** 48,53 ****
--- 48,54 ----
  
      const/4 v2, 0x0
  
+     .line 28
      new-instance v0, Lcom/android/internal/telephony/MmiCode$State;
  
      const-string v1, "PENDING"
***************
*** 56,61 ****
--- 57,63 ----
  
      sput-object v0, Lcom/android/internal/telephony/MmiCode$State;->PENDING:Lcom/android/internal/telephony/MmiCode$State;
  
+     .line 29
      new-instance v0, Lcom/android/internal/telephony/MmiCode$State;
  
      const-string v1, "CANCELLED"
***************
*** 64,69 ****
--- 66,72 ----
  
      sput-object v0, Lcom/android/internal/telephony/MmiCode$State;->CANCELLED:Lcom/android/internal/telephony/MmiCode$State;
  
+     .line 30
      new-instance v0, Lcom/android/internal/telephony/MmiCode$State;
  
      const-string v1, "COMPLETE"
***************
*** 72,77 ****
--- 75,81 ----
  
      sput-object v0, Lcom/android/internal/telephony/MmiCode$State;->COMPLETE:Lcom/android/internal/telephony/MmiCode$State;
  
+     .line 31
      new-instance v0, Lcom/android/internal/telephony/MmiCode$State;
  
      const-string v1, "FAILED"
***************
*** 80,85 ****
--- 84,90 ----
  
      sput-object v0, Lcom/android/internal/telephony/MmiCode$State;->FAILED:Lcom/android/internal/telephony/MmiCode$State;
  
+     .line 27
      const/4 v0, 0x4
  
      new-array v0, v0, [Lcom/android/internal/telephony/MmiCode$State;
***************
*** 116,121 ****
--- 121,127 ----
      .end annotation
  
      .prologue
+     .line 27
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 126,131 ****
--- 132,138 ----
      .parameter "name"
  
      .prologue
+     .line 27
      const-class v0, Lcom/android/internal/telephony/MmiCode$State;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 141,146 ****
--- 148,154 ----
      .locals 1
  
      .prologue
+     .line 27
      sget-object v0, Lcom/android/internal/telephony/MmiCode$State;->$VALUES:[Lcom/android/internal/telephony/MmiCode$State;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/MmiCode$State;->clone()Ljava/lang/Object;
