*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/OperatorInfo.smali	2012-03-23 19:37:42.764554892 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/OperatorInfo.smali.original	2012-03-23 19:34:46.266336453 -0500
***************
*** 42,47 ****
--- 42,48 ----
      .locals 1
  
      .prologue
+     .line 134
      new-instance v0, Lcom/android/internal/telephony/OperatorInfo$1;
  
      invoke-direct {v0}, Lcom/android/internal/telephony/OperatorInfo$1;-><init>()V
***************
*** 59,78 ****
--- 60,86 ----
      .parameter "state"
  
      .prologue
+     .line 63
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 37
      sget-object v0, Lcom/android/internal/telephony/OperatorInfo$State;->UNKNOWN:Lcom/android/internal/telephony/OperatorInfo$State;
  
      iput-object v0, p0, Lcom/android/internal/telephony/OperatorInfo;->state:Lcom/android/internal/telephony/OperatorInfo$State;
  
+     .line 65
      iput-object p1, p0, Lcom/android/internal/telephony/OperatorInfo;->operatorAlphaLong:Ljava/lang/String;
  
+     .line 66
      iput-object p2, p0, Lcom/android/internal/telephony/OperatorInfo;->operatorAlphaShort:Ljava/lang/String;
  
+     .line 67
      iput-object p3, p0, Lcom/android/internal/telephony/OperatorInfo;->operatorNumeric:Ljava/lang/String;
  
+     .line 69
      iput-object p4, p0, Lcom/android/internal/telephony/OperatorInfo;->state:Lcom/android/internal/telephony/OperatorInfo$State;
  
+     .line 70
      return-void
  .end method
  
***************
*** 84,95 ****
--- 92,105 ----
      .parameter "stateString"
  
      .prologue
+     .line 77
      invoke-static {p4}, Lcom/android/internal/telephony/OperatorInfo;->rilStateToState(Ljava/lang/String;)Lcom/android/internal/telephony/OperatorInfo$State;
  
      move-result-object v0
  
      invoke-direct {p0, p1, p2, p3, v0}, Lcom/android/internal/telephony/OperatorInfo;-><init>(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Lcom/android/internal/telephony/OperatorInfo$State;)V
  
+     .line 79
      return-void
  .end method
  
***************
*** 98,104 ****
      .parameter "s"
  
      .prologue
!     const-string v0, "unknown"
  
      invoke-virtual {p0, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 108,115 ----
      .parameter "s"
  
      .prologue
!     .line 85
!     const-string/jumbo v0, "unknown"
  
      invoke-virtual {p0, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 106,116 ****
--- 117,130 ----
  
      if-eqz v0, :cond_0
  
+     .line 86
      sget-object v0, Lcom/android/internal/telephony/OperatorInfo$State;->UNKNOWN:Lcom/android/internal/telephony/OperatorInfo$State;
  
+     .line 92
      :goto_0
      return-object v0
  
+     .line 87
      :cond_0
      const-string v0, "available"
  
***************
*** 120,129 ****
--- 134,145 ----
  
      if-eqz v0, :cond_1
  
+     .line 88
      sget-object v0, Lcom/android/internal/telephony/OperatorInfo$State;->AVAILABLE:Lcom/android/internal/telephony/OperatorInfo$State;
  
      goto :goto_0
  
+     .line 89
      :cond_1
      const-string v0, "current"
  
***************
*** 133,142 ****
--- 149,160 ----
  
      if-eqz v0, :cond_2
  
+     .line 90
      sget-object v0, Lcom/android/internal/telephony/OperatorInfo$State;->CURRENT:Lcom/android/internal/telephony/OperatorInfo$State;
  
      goto :goto_0
  
+     .line 91
      :cond_2
      const-string v0, "forbidden"
  
***************
*** 146,155 ****
--- 164,175 ----
  
      if-eqz v0, :cond_3
  
+     .line 92
      sget-object v0, Lcom/android/internal/telephony/OperatorInfo$State;->FORBIDDEN:Lcom/android/internal/telephony/OperatorInfo$State;
  
      goto :goto_0
  
+     .line 94
      :cond_3
      new-instance v0, Ljava/lang/RuntimeException;
  
***************
*** 188,193 ****
--- 208,214 ----
      .locals 1
  
      .prologue
+     .line 116
      const/4 v0, 0x0
  
      return v0
***************
*** 197,202 ****
--- 218,224 ----
      .locals 1
  
      .prologue
+     .line 42
      iget-object v0, p0, Lcom/android/internal/telephony/OperatorInfo;->operatorAlphaLong:Ljava/lang/String;
  
      return-object v0
***************
*** 206,211 ****
--- 228,234 ----
      .locals 1
  
      .prologue
+     .line 47
      iget-object v0, p0, Lcom/android/internal/telephony/OperatorInfo;->operatorAlphaShort:Ljava/lang/String;
  
      return-object v0
***************
*** 215,220 ****
--- 238,244 ----
      .locals 1
  
      .prologue
+     .line 52
      iget-object v0, p0, Lcom/android/internal/telephony/OperatorInfo;->operatorNumeric:Ljava/lang/String;
  
      return-object v0
***************
*** 224,229 ****
--- 248,254 ----
      .locals 1
  
      .prologue
+     .line 57
      iget-object v0, p0, Lcom/android/internal/telephony/OperatorInfo;->state:Lcom/android/internal/telephony/OperatorInfo$State;
  
      return-object v0
***************
*** 233,238 ****
--- 258,264 ----
      .locals 2
  
      .prologue
+     .line 101
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 298,318 ****
--- 324,349 ----
      .parameter "flags"
  
      .prologue
+     .line 124
      iget-object v0, p0, Lcom/android/internal/telephony/OperatorInfo;->operatorAlphaLong:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 125
      iget-object v0, p0, Lcom/android/internal/telephony/OperatorInfo;->operatorAlphaShort:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 126
      iget-object v0, p0, Lcom/android/internal/telephony/OperatorInfo;->operatorNumeric:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 127
      iget-object v0, p0, Lcom/android/internal/telephony/OperatorInfo;->state:Lcom/android/internal/telephony/OperatorInfo$State;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeSerializable(Ljava/io/Serializable;)V
  
+     .line 128
      return-void
  .end method
