*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/PhoneStateIntentReceiver.smali	2012-03-23 19:37:42.211544808 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/PhoneStateIntentReceiver.smali.original	2012-03-23 19:34:46.258336307 -0500
***************
*** 51,80 ****
--- 51,86 ----
      .locals 1
  
      .prologue
+     .line 63
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
  
+     .line 49
      sget-object v0, Lcom/android/internal/telephony/Phone$State;->IDLE:Lcom/android/internal/telephony/Phone$State;
  
      iput-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mPhoneState:Lcom/android/internal/telephony/Phone$State;
  
+     .line 50
      new-instance v0, Landroid/telephony/ServiceState;
  
      invoke-direct {v0}, Landroid/telephony/ServiceState;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mServiceState:Landroid/telephony/ServiceState;
  
+     .line 51
      new-instance v0, Landroid/telephony/SignalStrength;
  
      invoke-direct {v0}, Landroid/telephony/SignalStrength;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mSignalStrength:Landroid/telephony/SignalStrength;
  
+     .line 64
      new-instance v0, Landroid/content/IntentFilter;
  
      invoke-direct {v0}, Landroid/content/IntentFilter;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mFilter:Landroid/content/IntentFilter;
  
+     .line 65
      return-void
  .end method
  
***************
*** 84,95 ****
--- 90,105 ----
      .parameter "target"
  
      .prologue
+     .line 68
      invoke-direct {p0}, Lcom/android/internal/telephony/PhoneStateIntentReceiver;-><init>()V
  
+     .line 69
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->setContext(Landroid/content/Context;)V
  
+     .line 70
      invoke-virtual {p0, p2}, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->setTarget(Landroid/os/Handler;)V
  
+     .line 71
      return-void
  .end method
  
***************
*** 99,104 ****
--- 109,115 ----
      .locals 1
  
      .prologue
+     .line 133
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      and-int/lit8 v0, v0, 0x1
***************
*** 120,125 ****
--- 131,137 ----
      .locals 1
  
      .prologue
+     .line 143
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      and-int/lit8 v0, v0, 0x2
***************
*** 141,146 ****
--- 153,159 ----
      .locals 1
  
      .prologue
+     .line 153
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      and-int/lit8 v0, v0, 0x4
***************
*** 162,173 ****
--- 175,188 ----
      .locals 2
  
      .prologue
+     .line 82
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      and-int/lit8 v0, v0, 0x1
  
      if-nez v0, :cond_0
  
+     .line 83
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "client must call notifyPhoneCallState(int)"
***************
*** 176,181 ****
--- 191,197 ----
  
      throw v0
  
+     .line 86
      :cond_0
      iget-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mPhoneState:Lcom/android/internal/telephony/Phone$State;
  
***************
*** 186,197 ****
--- 202,215 ----
      .locals 2
  
      .prologue
+     .line 90
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      and-int/lit8 v0, v0, 0x2
  
      if-nez v0, :cond_0
  
+     .line 91
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "client must call notifyServiceState(int)"
***************
*** 200,205 ****
--- 218,224 ----
  
      throw v0
  
+     .line 94
      :cond_0
      iget-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mServiceState:Landroid/telephony/ServiceState;
  
***************
*** 210,221 ****
--- 229,242 ----
      .locals 2
  
      .prologue
+     .line 119
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      and-int/lit8 v0, v0, 0x4
  
      if-nez v0, :cond_0
  
+     .line 120
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "client must call notifySignalStrength(int)"
***************
*** 224,229 ****
--- 245,251 ----
  
      throw v0
  
+     .line 123
      :cond_0
      iget-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mSignalStrength:Landroid/telephony/SignalStrength;
  
***************
*** 238,249 ****
--- 260,273 ----
      .locals 2
  
      .prologue
+     .line 104
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      and-int/lit8 v0, v0, 0x4
  
      if-nez v0, :cond_0
  
+     .line 105
      new-instance v0, Ljava/lang/RuntimeException;
  
      const-string v1, "client must call notifySignalStrength(int)"
***************
*** 252,257 ****
--- 276,282 ----
  
      throw v0
  
+     .line 108
      :cond_0
      iget-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mSignalStrength:Landroid/telephony/SignalStrength;
  
***************
*** 267,286 ****
--- 292,315 ----
      .parameter "eventWhat"
  
      .prologue
+     .line 127
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      or-int/lit8 v0, v0, 0x1
  
      iput v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
+     .line 128
      iput p1, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mPhoneStateEventWhat:I
  
+     .line 129
      iget-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mFilter:Landroid/content/IntentFilter;
  
      const-string v1, "android.intent.action.PHONE_STATE"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 130
      return-void
  .end method
  
***************
*** 289,308 ****
--- 318,341 ----
      .parameter "eventWhat"
  
      .prologue
+     .line 137
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      or-int/lit8 v0, v0, 0x2
  
      iput v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
+     .line 138
      iput p1, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mServiceStateEventWhat:I
  
+     .line 139
      iget-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mFilter:Landroid/content/IntentFilter;
  
      const-string v1, "android.intent.action.SERVICE_STATE"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 140
      return-void
  .end method
  
***************
*** 311,330 ****
--- 344,367 ----
      .parameter "eventWhat"
  
      .prologue
+     .line 147
      iget v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
      or-int/lit8 v0, v0, 0x4
  
      iput v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mWants:I
  
+     .line 148
      iput p1, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mAsuEventWhat:I
  
+     .line 149
      iget-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mFilter:Landroid/content/IntentFilter;
  
      const-string v1, "android.intent.action.SIG_STR"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 150
      return-void
  .end method
  
***************
*** 334,343 ****
--- 371,382 ----
      .parameter "intent"
  
      .prologue
+     .line 166
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 169
      .local v0, action:Ljava/lang/String;
      :try_start_0
      const-string v4, "android.intent.action.SIG_STR"
***************
*** 348,353 ****
--- 387,393 ----
  
      if-eqz v4, :cond_1
  
+     .line 170
      invoke-virtual {p2}, Landroid/content/Intent;->getExtras()Landroid/os/Bundle;
  
      move-result-object v4
***************
*** 358,363 ****
--- 398,404 ----
  
      iput-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mSignalStrength:Landroid/telephony/SignalStrength;
  
+     .line 172
      iget-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
      if-eqz v4, :cond_0
***************
*** 368,373 ****
--- 409,415 ----
  
      if-eqz v4, :cond_0
  
+     .line 173
      iget-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
      iget v5, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mAsuEventWhat:I
***************
*** 376,391 ****
--- 418,436 ----
  
      move-result-object v2
  
+     .line 174
      .local v2, message:Landroid/os/Message;
      iget-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
      invoke-virtual {v4, v2}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 201
      .end local v2           #message:Landroid/os/Message;
      :cond_0
      :goto_0
      return-void
  
+     .line 176
      :cond_1
      const-string v4, "android.intent.action.PHONE_STATE"
  
***************
*** 395,406 ****
  
      if-eqz v4, :cond_2
  
!     const-string v4, "state"
  
      invoke-virtual {p2, v4}, Landroid/content/Intent;->getStringExtra(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v3
  
      .local v3, phoneState:Ljava/lang/String;
      const-class v4, Lcom/android/internal/telephony/Phone$State;
  
--- 440,453 ----
  
      if-eqz v4, :cond_2
  
!     .line 179
!     const-string/jumbo v4, "state"
  
      invoke-virtual {p2, v4}, Landroid/content/Intent;->getStringExtra(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v3
  
+     .line 180
      .local v3, phoneState:Ljava/lang/String;
      const-class v4, Lcom/android/internal/telephony/Phone$State;
  
***************
*** 412,417 ****
--- 459,465 ----
  
      iput-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mPhoneState:Lcom/android/internal/telephony/Phone$State;
  
+     .line 183
      iget-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
      if-eqz v4, :cond_0
***************
*** 422,427 ****
--- 470,476 ----
  
      if-eqz v4, :cond_0
  
+     .line 184
      iget-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
      iget v5, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mPhoneStateEventWhat:I
***************
*** 430,435 ****
--- 479,485 ----
  
      move-result-object v2
  
+     .line 186
      .restart local v2       #message:Landroid/os/Message;
      iget-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
***************
*** 439,449 ****
--- 489,501 ----
  
      goto :goto_0
  
+     .line 197
      .end local v2           #message:Landroid/os/Message;
      .end local v3           #phoneState:Ljava/lang/String;
      :catch_0
      move-exception v1
  
+     .line 198
      .local v1, ex:Ljava/lang/Exception;
      const-string v4, "PHONE"
  
***************
*** 467,476 ****
--- 519,530 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 199
      invoke-virtual {v1}, Ljava/lang/Exception;->printStackTrace()V
  
      goto :goto_0
  
+     .line 188
      .end local v1           #ex:Ljava/lang/Exception;
      :cond_2
      :try_start_1
***************
*** 482,487 ****
--- 536,542 ----
  
      if-eqz v4, :cond_0
  
+     .line 189
      invoke-virtual {p2}, Landroid/content/Intent;->getExtras()Landroid/os/Bundle;
  
      move-result-object v4
***************
*** 492,497 ****
--- 547,553 ----
  
      iput-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mServiceState:Landroid/telephony/ServiceState;
  
+     .line 191
      iget-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
      if-eqz v4, :cond_0
***************
*** 502,507 ****
--- 558,564 ----
  
      if-eqz v4, :cond_0
  
+     .line 192
      iget-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
      iget v5, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mServiceStateEventWhat:I
***************
*** 510,515 ****
--- 567,573 ----
  
      move-result-object v2
  
+     .line 194
      .restart local v2       #message:Landroid/os/Message;
      iget-object v4, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
***************
*** 524,535 ****
--- 582,595 ----
      .locals 2
  
      .prologue
+     .line 157
      iget-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mContext:Landroid/content/Context;
  
      iget-object v1, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mFilter:Landroid/content/IntentFilter;
  
      invoke-virtual {v0, p0, v1}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 158
      return-void
  .end method
  
***************
*** 538,545 ****
--- 598,607 ----
      .parameter "c"
  
      .prologue
+     .line 74
      iput-object p1, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mContext:Landroid/content/Context;
  
+     .line 75
      return-void
  .end method
  
***************
*** 548,555 ****
--- 610,619 ----
      .parameter "h"
  
      .prologue
+     .line 78
      iput-object p1, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mTarget:Landroid/os/Handler;
  
+     .line 79
      return-void
  .end method
  
***************
*** 557,565 ****
--- 621,631 ----
      .locals 1
  
      .prologue
+     .line 161
      iget-object v0, p0, Lcom/android/internal/telephony/PhoneStateIntentReceiver;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0, p0}, Landroid/content/Context;->unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  
+     .line 162
      return-void
  .end method
