*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/RIL$RILSender.smali	2012-03-23 19:37:42.096542711 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/RIL$RILSender.smali.original	2012-03-23 19:34:46.256336271 -0500
***************
*** 30,45 ****
--- 30,49 ----
      .parameter "looper"
  
      .prologue
+     .line 270
      iput-object p1, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
+     .line 271
      invoke-direct {p0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
+     .line 275
      const/4 v0, 0x4
  
      new-array v0, v0, [B
  
      iput-object v0, p0, Lcom/android/internal/telephony/RIL$RILSender;->dataLength:[B
  
+     .line 272
      return-void
  .end method
  
***************
*** 50,55 ****
--- 54,60 ----
      .parameter "msg"
  
      .prologue
+     .line 287
      move-object/from16 v0, p1
  
      iget-object v10, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
***************
*** 60,68 ****
--- 65,75 ----
  
      check-cast v8, Lcom/android/internal/telephony/RILRequest;
  
+     .line 288
      .local v8, rr:Lcom/android/internal/telephony/RILRequest;
      const/4 v7, 0x0
  
+     .line 290
      .local v7, req:Lcom/android/internal/telephony/RILRequest;
      move-object/from16 v0, p1
  
***************
*** 70,105 ****
--- 77,120 ----
  
      packed-switch v10, :pswitch_data_0
  
+     .line 426
      :cond_0
      :goto_0
      return-void
  
+     .line 297
      :pswitch_0
      const/4 v1, 0x0
  
+     .line 301
      .local v1, alreadySubtracted:Z
      :try_start_0
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget-object v9, v10, Lcom/android/internal/telephony/RIL;->mSocket:Landroid/net/LocalSocket;
  
+     .line 303
      .local v9, s:Landroid/net/LocalSocket;
      if-nez v9, :cond_2
  
+     .line 304
      const/4 v10, 0x1
  
      const/4 v11, 0x0
  
      invoke-virtual {v8, v10, v11}, Lcom/android/internal/telephony/RILRequest;->onError(ILjava/lang/Object;)V
  
+     .line 305
      invoke-virtual {v8}, Lcom/android/internal/telephony/RILRequest;->release()V
  
+     .line 306
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget v10, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesPending:I
  
      if-lez v10, :cond_1
  
+     .line 307
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget v11, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesPending:I
***************
*** 112,120 ****
--- 127,137 ----
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/lang/RuntimeException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 308
      :cond_1
      const/4 v1, 0x1
  
+     .line 364
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      #calls: Lcom/android/internal/telephony/RIL;->releaseWakeLockIfDone()V
***************
*** 122,127 ****
--- 139,145 ----
  
      goto :goto_0
  
+     .line 312
      :cond_2
      :try_start_1
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
***************
*** 134,139 ****
--- 152,158 ----
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_0
      .catch Ljava/lang/RuntimeException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 313
      :try_start_2
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
***************
*** 141,146 ****
--- 160,166 ----
  
      invoke-virtual {v10, v8}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 314
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget v12, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesWaiting:I
***************
*** 149,158 ****
--- 169,180 ----
  
      iput v12, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesWaiting:I
  
+     .line 315
      monitor-exit v11
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
  
+     .line 317
      :try_start_3
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
***************
*** 160,165 ****
--- 182,188 ----
  
      if-lez v10, :cond_3
  
+     .line 318
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget v11, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesPending:I
***************
*** 168,197 ****
--- 191,226 ----
  
      iput v11, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesPending:I
  
+     .line 319
      :cond_3
      const/4 v1, 0x1
  
+     .line 323
      iget-object v10, v8, Lcom/android/internal/telephony/RILRequest;->mp:Landroid/os/Parcel;
  
      invoke-virtual {v10}, Landroid/os/Parcel;->marshall()[B
  
      move-result-object v3
  
+     .line 324
      .local v3, data:[B
      iget-object v10, v8, Lcom/android/internal/telephony/RILRequest;->mp:Landroid/os/Parcel;
  
      invoke-virtual {v10}, Landroid/os/Parcel;->recycle()V
  
+     .line 325
      const/4 v10, 0x0
  
      iput-object v10, v8, Lcom/android/internal/telephony/RILRequest;->mp:Landroid/os/Parcel;
  
+     .line 327
      array-length v10, v3
  
      const/16 v11, 0x2000
  
      if-le v10, v11, :cond_8
  
+     .line 328
      new-instance v10, Ljava/lang/RuntimeException;
  
      new-instance v11, Ljava/lang/StringBuilder;
***************
*** 222,232 ****
--- 251,263 ----
      .catch Ljava/io/IOException; {:try_start_3 .. :try_end_3} :catch_0
      .catch Ljava/lang/RuntimeException; {:try_start_3 .. :try_end_3} :catch_1
  
+     .line 342
      .end local v3           #data:[B
      .end local v9           #s:Landroid/net/LocalSocket;
      :catch_0
      move-exception v4
  
+     .line 343
      .local v4, ex:Ljava/io/IOException;
      :try_start_4
      const-string v10, "RILJ"
***************
*** 235,240 ****
--- 266,272 ----
  
      invoke-static {v10, v11, v4}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 344
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget v11, v8, Lcom/android/internal/telephony/RILRequest;->mSerial:I
***************
*** 244,253 ****
--- 276,287 ----
  
      move-result-object v7
  
+     .line 347
      if-nez v7, :cond_4
  
      if-nez v1, :cond_5
  
+     .line 348
      :cond_4
      const/4 v10, 0x1
  
***************
*** 255,264 ****
--- 289,300 ----
  
      invoke-virtual {v8, v10, v11}, Lcom/android/internal/telephony/RILRequest;->onError(ILjava/lang/Object;)V
  
+     .line 349
      invoke-virtual {v8}, Lcom/android/internal/telephony/RILRequest;->release()V
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_1
  
+     .line 364
      :cond_5
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
***************
*** 267,272 ****
--- 303,309 ----
      #calls: Lcom/android/internal/telephony/RIL;->releaseWakeLockIfDone()V
      invoke-static {v10}, Lcom/android/internal/telephony/RIL;->access$200(Lcom/android/internal/telephony/RIL;)V
  
+     .line 367
      if-nez v1, :cond_0
  
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
***************
*** 275,280 ****
--- 312,318 ----
  
      if-lez v10, :cond_0
  
+     .line 368
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget v11, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesPending:I
***************
*** 285,290 ****
--- 323,329 ----
  
      goto/16 :goto_0
  
+     .line 315
      .restart local v9       #s:Landroid/net/LocalSocket;
      :catchall_0
      move-exception v10
***************
*** 301,310 ****
--- 340,351 ----
      .catch Ljava/io/IOException; {:try_start_6 .. :try_end_6} :catch_0
      .catch Ljava/lang/RuntimeException; {:try_start_6 .. :try_end_6} :catch_1
  
+     .line 351
      .end local v9           #s:Landroid/net/LocalSocket;
      :catch_1
      move-exception v5
  
+     .line 352
      .local v5, exc:Ljava/lang/RuntimeException;
      :try_start_7
      const-string v10, "RILJ"
***************
*** 313,318 ****
--- 354,360 ----
  
      invoke-static {v10, v11, v5}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 353
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget v11, v8, Lcom/android/internal/telephony/RILRequest;->mSerial:I
***************
*** 322,331 ****
--- 364,375 ----
  
      move-result-object v7
  
+     .line 356
      if-nez v7, :cond_6
  
      if-nez v1, :cond_7
  
+     .line 357
      :cond_6
      const/4 v10, 0x2
  
***************
*** 333,347 ****
--- 377,394 ----
  
      invoke-virtual {v8, v10, v11}, Lcom/android/internal/telephony/RILRequest;->onError(ILjava/lang/Object;)V
  
+     .line 358
      invoke-virtual {v8}, Lcom/android/internal/telephony/RILRequest;->release()V
      :try_end_7
      .catchall {:try_start_7 .. :try_end_7} :catchall_1
  
+     .line 364
      :cond_7
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      goto :goto_1
  
+     .line 334
      .end local v5           #exc:Ljava/lang/RuntimeException;
      .restart local v3       #data:[B
      .restart local v9       #s:Landroid/net/LocalSocket;
***************
*** 361,366 ****
--- 408,414 ----
  
      aput-byte v14, v10, v11
  
+     .line 335
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->dataLength:[B
  
      const/4 v11, 0x2
***************
*** 375,380 ****
--- 423,429 ----
  
      aput-byte v12, v10, v11
  
+     .line 336
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->dataLength:[B
  
      const/4 v11, 0x3
***************
*** 387,392 ****
--- 436,442 ----
  
      aput-byte v12, v10, v11
  
+     .line 340
      invoke-virtual {v9}, Landroid/net/LocalSocket;->getOutputStream()Ljava/io/OutputStream;
  
      move-result-object v10
***************
*** 395,400 ****
--- 445,451 ----
  
      invoke-virtual {v10, v11}, Ljava/io/OutputStream;->write([B)V
  
+     .line 341
      invoke-virtual {v9}, Landroid/net/LocalSocket;->getOutputStream()Ljava/io/OutputStream;
  
      move-result-object v10
***************
*** 405,410 ****
--- 456,462 ----
      .catch Ljava/io/IOException; {:try_start_8 .. :try_end_8} :catch_0
      .catch Ljava/lang/RuntimeException; {:try_start_8 .. :try_end_8} :catch_1
  
+     .line 364
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      goto :goto_1
***************
*** 421,426 ****
--- 473,479 ----
  
      throw v10
  
+     .line 376
      .end local v1           #alreadySubtracted:Z
      :pswitch_1
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
***************
*** 429,434 ****
--- 482,488 ----
  
      monitor-enter v11
  
+     .line 377
      :try_start_9
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
***************
*** 440,451 ****
--- 494,507 ----
  
      if-eqz v10, :cond_c
  
+     .line 387
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget v10, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesWaiting:I
  
      if-eqz v10, :cond_a
  
+     .line 388
      const-string v10, "RILJ"
  
      new-instance v12, Ljava/lang/StringBuilder;
***************
*** 484,495 ****
--- 540,553 ----
  
      invoke-static {v10, v12}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 392
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      const/4 v12, 0x0
  
      iput v12, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesWaiting:I
  
+     .line 395
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget-object v12, v10, Lcom/android/internal/telephony/RIL;->mRequestsList:Ljava/util/ArrayList;
***************
*** 498,503 ****
--- 556,562 ----
      :try_end_9
      .catchall {:try_start_9 .. :try_end_9} :catchall_2
  
+     .line 396
      :try_start_a
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
***************
*** 507,512 ****
--- 566,572 ----
  
      move-result v2
  
+     .line 397
      .local v2, count:I
      const-string v10, "RILJ"
  
***************
*** 530,541 ****
--- 590,603 ----
  
      invoke-static {v10, v13}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 400
      const/4 v6, 0x0
  
      .local v6, i:I
      :goto_2
      if-ge v6, v2, :cond_9
  
+     .line 401
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      iget-object v10, v10, Lcom/android/internal/telephony/RIL;->mRequestsList:Ljava/util/ArrayList;
***************
*** 550,555 ****
--- 612,618 ----
  
      move-object v8, v0
  
+     .line 402
      const-string v10, "RILJ"
  
      new-instance v13, Ljava/lang/StringBuilder;
***************
*** 594,608 ****
--- 657,674 ----
  
      invoke-static {v10, v13}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 400
      add-int/lit8 v6, v6, 0x1
  
      goto :goto_2
  
+     .line 405
      :cond_9
      monitor-exit v12
      :try_end_a
      .catchall {:try_start_a .. :try_end_a} :catchall_3
  
+     .line 415
      .end local v2           #count:I
      .end local v6           #i:I
      :cond_a
***************
*** 613,618 ****
--- 679,685 ----
  
      if-eqz v10, :cond_b
  
+     .line 416
      const-string v10, "RILJ"
  
      new-instance v12, Ljava/lang/StringBuilder;
***************
*** 645,656 ****
--- 712,725 ----
  
      invoke-static {v10, v12}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 418
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
      const/4 v12, 0x0
  
      iput v12, v10, Lcom/android/internal/telephony/RIL;->mRequestMessagesPending:I
  
+     .line 421
      :cond_b
      iget-object v10, p0, Lcom/android/internal/telephony/RIL$RILSender;->this$0:Lcom/android/internal/telephony/RIL;
  
***************
*** 658,663 ****
--- 727,733 ----
  
      invoke-virtual {v10}, Landroid/os/PowerManager$WakeLock;->release()V
  
+     .line 423
      :cond_c
      monitor-exit v11
  
***************
*** 672,677 ****
--- 742,748 ----
  
      throw v10
  
+     .line 405
      :catchall_3
      move-exception v10
  
***************
*** 685,690 ****
--- 756,762 ----
      :try_end_d
      .catchall {:try_start_d .. :try_end_d} :catchall_2
  
+     .line 290
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 696,700 ****
--- 768,773 ----
      .locals 0
  
      .prologue
+     .line 281
      return-void
  .end method
