*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/SMSDispatcher.smali	2012-03-23 19:37:41.369529456 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/SMSDispatcher.smali.original	2012-03-23 19:34:46.243336033 -0500
***************
*** 122,144 ****
  
      const/4 v2, 0x0
  
      new-array v0, v3, [Ljava/lang/String;
  
!     const-string v1, "pdu"
  
      aput-object v1, v0, v2
  
      sput-object v0, Lcom/android/internal/telephony/SMSDispatcher;->PDU_PROJECTION:[Ljava/lang/String;
  
      const/4 v0, 0x3
  
      new-array v0, v0, [Ljava/lang/String;
  
!     const-string v1, "pdu"
  
      aput-object v1, v0, v2
  
!     const-string v1, "sequence"
  
      aput-object v1, v0, v3
  
--- 122,146 ----
  
      const/4 v2, 0x0
  
+     .line 80
      new-array v0, v3, [Ljava/lang/String;
  
!     const-string/jumbo v1, "pdu"
  
      aput-object v1, v0, v2
  
      sput-object v0, Lcom/android/internal/telephony/SMSDispatcher;->PDU_PROJECTION:[Ljava/lang/String;
  
+     .line 85
      const/4 v0, 0x3
  
      new-array v0, v0, [Ljava/lang/String;
  
!     const-string/jumbo v1, "pdu"
  
      aput-object v1, v0, v2
  
!     const-string/jumbo v1, "sequence"
  
      aput-object v1, v0, v3
  
***************
*** 150,158 ****
  
      sput-object v0, Lcom/android/internal/telephony/SMSDispatcher;->PDU_SEQUENCE_PORT_PROJECTION:[Ljava/lang/String;
  
      sget-object v0, Landroid/provider/Telephony$Sms;->CONTENT_URI:Landroid/net/Uri;
  
!     const-string v1, "raw"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
--- 152,161 ----
  
      sput-object v0, Lcom/android/internal/telephony/SMSDispatcher;->PDU_SEQUENCE_PORT_PROJECTION:[Ljava/lang/String;
  
+     .line 124
      sget-object v0, Landroid/provider/Telephony$Sms;->CONTENT_URI:Landroid/net/Uri;
  
!     const-string/jumbo v1, "raw"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
***************
*** 160,165 ****
--- 163,169 ----
  
      sput-object v0, Lcom/android/internal/telephony/SMSDispatcher;->mRawUri:Landroid/net/Uri;
  
+     .line 141
      new-instance v0, Ljava/util/Random;
  
      invoke-direct {v0}, Ljava/util/Random;-><init>()V
***************
*** 186,193 ****
--- 190,199 ----
  
      const/4 v1, 0x1
  
+     .line 176
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
  
+     .line 146
      new-instance v0, Ljava/util/ArrayList;
  
      const/4 v3, 0x5
***************
*** 196,239 ****
--- 202,254 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
  
+     .line 158
      iput-boolean v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsCapable:Z
  
+     .line 162
      const/4 v0, -0x1
  
      iput v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
+     .line 229
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->deliveryPendingList:Ljava/util/ArrayList;
  
+     .line 1052
      new-instance v0, Lcom/android/internal/telephony/SMSDispatcher$1;
  
      invoke-direct {v0, p0}, Lcom/android/internal/telephony/SMSDispatcher$1;-><init>(Lcom/android/internal/telephony/SMSDispatcher;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mListener:Landroid/content/DialogInterface$OnClickListener;
  
+     .line 1066
      new-instance v0, Lcom/android/internal/telephony/SMSDispatcher$2;
  
      invoke-direct {v0, p0}, Lcom/android/internal/telephony/SMSDispatcher$2;-><init>(Lcom/android/internal/telephony/SMSDispatcher;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mResultReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 177
      iput-object p1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mPhone:Lcom/android/internal/telephony/Phone;
  
+     .line 178
      new-instance v0, Lcom/android/internal/telephony/WapPushOverSms;
  
      invoke-direct {v0, p1, p0}, Lcom/android/internal/telephony/WapPushOverSms;-><init>(Lcom/android/internal/telephony/Phone;Lcom/android/internal/telephony/SMSDispatcher;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mWapPush:Lcom/android/internal/telephony/WapPushOverSms;
  
+     .line 179
      invoke-virtual {p1}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
+     .line 180
      iget-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
***************
*** 242,257 ****
--- 257,277 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mResolver:Landroid/content/ContentResolver;
  
+     .line 181
      iget-object v0, p1, Lcom/android/internal/telephony/PhoneBase;->mCM:Lcom/android/internal/telephony/CommandsInterface;
  
      iput-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mCm:Lcom/android/internal/telephony/CommandsInterface;
  
+     .line 182
      iput-object p2, p0, Lcom/android/internal/telephony/SMSDispatcher;->mStorageMonitor:Lcom/android/internal/telephony/SmsStorageMonitor;
  
+     .line 183
      iput-object p3, p0, Lcom/android/internal/telephony/SMSDispatcher;->mUsageMonitor:Lcom/android/internal/telephony/SmsUsageMonitor;
  
+     .line 185
      invoke-direct {p0}, Lcom/android/internal/telephony/SMSDispatcher;->createWakelock()V
  
+     .line 187
      iget-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 266,272 ****
  
      iput-boolean v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsCapable:Z
  
!     const-string v0, "telephony.sms.receive"
  
      iget-boolean v3, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsCapable:Z
  
--- 286,293 ----
  
      iput-boolean v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsCapable:Z
  
!     .line 189
!     const-string/jumbo v0, "telephony.sms.receive"
  
      iget-boolean v3, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsCapable:Z
  
***************
*** 281,287 ****
      :goto_0
      iput-boolean v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsReceiveDisabled:Z
  
!     const-string v0, "telephony.sms.send"
  
      iget-boolean v3, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsCapable:Z
  
--- 302,309 ----
      :goto_0
      iput-boolean v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsReceiveDisabled:Z
  
!     .line 191
!     const-string/jumbo v0, "telephony.sms.send"
  
      iget-boolean v3, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsCapable:Z
  
***************
*** 294,299 ****
--- 316,322 ----
      :goto_1
      iput-boolean v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsSendDisabled:Z
  
+     .line 193
      const-string v0, "SMS"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 356,371 ****
--- 379,397 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 196
      return-void
  
      :cond_0
      move v0, v2
  
+     .line 189
      goto :goto_0
  
      :cond_1
      move v1, v2
  
+     .line 191
      goto :goto_1
  .end method
  
***************
*** 375,383 ****
      .prologue
      const/4 v3, 0x1
  
      iget-object v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
!     const-string v2, "power"
  
      invoke-virtual {v1, v2}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
--- 401,410 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 329
      iget-object v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
!     const-string/jumbo v2, "power"
  
      invoke-virtual {v1, v2}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
***************
*** 385,390 ****
--- 412,418 ----
  
      check-cast v0, Landroid/os/PowerManager;
  
+     .line 330
      .local v0, pm:Landroid/os/PowerManager;
      const-string v1, "SMSDispatcher"
  
***************
*** 394,403 ****
--- 422,433 ----
  
      iput-object v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mWakeLock:Landroid/os/PowerManager$WakeLock;
  
+     .line 331
      iget-object v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mWakeLock:Landroid/os/PowerManager$WakeLock;
  
      invoke-virtual {v1, v3}, Landroid/os/PowerManager$WakeLock;->setReferenceCounted(Z)V
  
+     .line 332
      return-void
  .end method
  
***************
*** 406,415 ****
--- 436,447 ----
      .parameter "intent"
  
      .prologue
+     .line 945
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v0
  
+     .line 946
      .local v0, r:Landroid/content/res/Resources;
      if-eqz p0, :cond_0
  
***************
*** 434,445 ****
--- 466,479 ----
      .locals 1
  
      .prologue
+     .line 165
      sget v0, Lcom/android/internal/telephony/SMSDispatcher;->sConcatenatedRef:I
  
      add-int/lit8 v0, v0, 0x1
  
      sput v0, Lcom/android/internal/telephony/SMSDispatcher;->sConcatenatedRef:I
  
+     .line 166
      sget v0, Lcom/android/internal/telephony/SMSDispatcher;->sConcatenatedRef:I
  
      return v0
***************
*** 451,464 ****
--- 485,501 ----
      .parameter "tracker"
  
      .prologue
+     .line 451
      iget-object v0, p1, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
      if-eqz v0, :cond_0
  
+     .line 453
      const/4 v0, 0x3
  
      if-ne p0, v0, :cond_1
  
+     .line 454
      :try_start_0
      iget-object v0, p1, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
***************
*** 466,475 ****
--- 503,514 ----
  
      invoke-virtual {v0, v1}, Landroid/app/PendingIntent;->send(I)V
  
+     .line 460
      :cond_0
      :goto_0
      return-void
  
+     .line 456
      :cond_1
      iget-object v0, p1, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
***************
*** 481,486 ****
--- 520,526 ----
  
      goto :goto_0
  
+     .line 458
      :catch_0
      move-exception v0
  
***************
*** 494,528 ****
      .parameter "response"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.provider.Telephony.SMS_REJECTED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
      .local v0, intent:Landroid/content/Intent;
!     const-string v1, "result"
  
      invoke-virtual {v0, v1, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
      iget-object v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mWakeLock:Landroid/os/PowerManager$WakeLock;
  
      const-wide/16 v2, 0x1388
  
      invoke-virtual {v1, v2, v3}, Landroid/os/PowerManager$WakeLock;->acquire(J)V
  
      iget-object v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
      const-string v2, "android.permission.RECEIVE_SMS"
  
      invoke-virtual {v1, v0, v2}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;Ljava/lang/String;)V
  
      .end local v0           #intent:Landroid/content/Intent;
      :cond_0
      invoke-virtual {p0, p1, p2, p3}, Lcom/android/internal/telephony/SMSDispatcher;->acknowledgeLastIncomingSms(ZILandroid/os/Message;)V
  
      return-void
  .end method
  
--- 534,575 ----
      .parameter "response"
  
      .prologue
+     .line 1010
      if-nez p1, :cond_0
  
+     .line 1012
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.provider.Telephony.SMS_REJECTED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 1013
      .local v0, intent:Landroid/content/Intent;
!     const-string/jumbo v1, "result"
  
      invoke-virtual {v0, v1, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 1014
      iget-object v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mWakeLock:Landroid/os/PowerManager$WakeLock;
  
      const-wide/16 v2, 0x1388
  
      invoke-virtual {v1, v2, v3}, Landroid/os/PowerManager$WakeLock;->acquire(J)V
  
+     .line 1015
      iget-object v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
      const-string v2, "android.permission.RECEIVE_SMS"
  
      invoke-virtual {v1, v0, v2}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;Ljava/lang/String;)V
  
+     .line 1017
      .end local v0           #intent:Landroid/content/Intent;
      :cond_0
      invoke-virtual {p0, p1, p2, p3}, Lcom/android/internal/telephony/SMSDispatcher;->acknowledgeLastIncomingSms(ZILandroid/os/Message;)V
  
+     .line 1018
      return-void
  .end method
  
***************
*** 533,540 ****
--- 580,589 ----
      .prologue
      const/4 v11, 0x0
  
+     .line 967
      iget-object v8, p1, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mData:Ljava/util/HashMap;
  
+     .line 969
      .local v8, map:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/String;Ljava/lang/Object;>;"
      const-string v0, "destination"
  
***************
*** 544,551 ****
  
      check-cast v1, Ljava/lang/String;
  
      .local v1, destinationAddress:Ljava/lang/String;
!     const-string v0, "scaddress"
  
      invoke-virtual {v8, v0}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
--- 593,601 ----
  
      check-cast v1, Ljava/lang/String;
  
+     .line 970
      .local v1, destinationAddress:Ljava/lang/String;
!     const-string/jumbo v0, "scaddress"
  
      invoke-virtual {v8, v0}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
***************
*** 553,560 ****
  
      check-cast v2, Ljava/lang/String;
  
      .local v2, scAddress:Ljava/lang/String;
!     const-string v0, "parts"
  
      invoke-virtual {v8, v0}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
--- 603,611 ----
  
      check-cast v2, Ljava/lang/String;
  
+     .line 972
      .local v2, scAddress:Ljava/lang/String;
!     const-string/jumbo v0, "parts"
  
      invoke-virtual {v8, v0}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
***************
*** 562,569 ****
  
      check-cast v3, Ljava/util/ArrayList;
  
      .local v3, parts:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
!     const-string v0, "sentIntents"
  
      invoke-virtual {v8, v0}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
--- 613,621 ----
  
      check-cast v3, Ljava/util/ArrayList;
  
+     .line 973
      .local v3, parts:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
!     const-string/jumbo v0, "sentIntents"
  
      invoke-virtual {v8, v0}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
***************
*** 571,576 ****
--- 623,629 ----
  
      check-cast v4, Ljava/util/ArrayList;
  
+     .line 974
      .local v4, sentIntents:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/app/PendingIntent;>;"
      const-string v0, "deliveryIntents"
  
***************
*** 580,585 ****
--- 633,639 ----
  
      check-cast v5, Ljava/util/ArrayList;
  
+     .line 977
      .local v5, deliveryIntents:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/app/PendingIntent;>;"
      iget-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mPhone:Lcom/android/internal/telephony/Phone;
  
***************
*** 591,599 ****
--- 645,655 ----
  
      move-result v10
  
+     .line 978
      .local v10, ss:I
      if-eqz v10, :cond_1
  
+     .line 979
      const/4 v7, 0x0
  
      .local v7, i:I
***************
*** 605,612 ****
--- 661,670 ----
      :goto_0
      if-ge v7, v6, :cond_2
  
+     .line 980
      const/4 v9, 0x0
  
+     .line 981
      .local v9, sentIntent:Landroid/app/PendingIntent;
      if-eqz v4, :cond_0
  
***************
*** 616,621 ****
--- 674,680 ----
  
      if-le v0, v7, :cond_0
  
+     .line 982
      invoke-virtual {v4, v7}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v9
***************
*** 623,628 ****
--- 682,688 ----
      .end local v9           #sentIntent:Landroid/app/PendingIntent;
      check-cast v9, Landroid/app/PendingIntent;
  
+     .line 984
      .restart local v9       #sentIntent:Landroid/app/PendingIntent;
      :cond_0
      new-instance v0, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
***************
*** 631,636 ****
--- 691,697 ----
  
      invoke-static {v10, v0}, Lcom/android/internal/telephony/SMSDispatcher;->handleNotInService(ILcom/android/internal/telephony/SMSDispatcher$SmsTracker;)V
  
+     .line 979
      add-int/lit8 v7, v7, 0x1
  
      goto :goto_0
***************
*** 641,648 ****
--- 702,711 ----
      :cond_1
      move-object v0, p0
  
+     .line 989
      invoke-virtual/range {v0 .. v5}, Lcom/android/internal/telephony/SMSDispatcher;->sendMultipartText(Ljava/lang/String;Ljava/lang/String;Ljava/util/ArrayList;Ljava/util/ArrayList;Ljava/util/ArrayList;)V
  
+     .line 990
      :cond_2
      return-void
  .end method
***************
*** 663,674 ****
--- 726,739 ----
      .prologue
      const/4 v6, 0x0
  
+     .line 345
      iget-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mWakeLock:Landroid/os/PowerManager$WakeLock;
  
      const-wide/16 v1, 0x1388
  
      invoke-virtual {v0, v1, v2}, Landroid/os/PowerManager$WakeLock;->acquire(J)V
  
+     .line 346
      iget-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
      iget-object v3, p0, Lcom/android/internal/telephony/SMSDispatcher;->mResultReceiver:Landroid/content/BroadcastReceiver;
***************
*** 685,690 ****
--- 750,756 ----
  
      invoke-virtual/range {v0 .. v7}, Landroid/content/Context;->sendOrderedBroadcast(Landroid/content/Intent;Ljava/lang/String;Landroid/content/BroadcastReceiver;Landroid/os/Handler;ILjava/lang/String;Landroid/os/Bundle;)V
  
+     .line 348
      return-void
  .end method
  
***************
*** 694,712 ****
      .parameter "isEmergencyMessage"
  
      .prologue
      if-eqz p2, :cond_0
  
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.provider.Telephony.SMS_EMERGENCY_CB_RECEIVED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
      .local v0, intent:Landroid/content/Intent;
!     const-string v1, "pdus"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
      const-string v1, "SMS"
  
      new-instance v2, Ljava/lang/StringBuilder;
--- 760,782 ----
      .parameter "isEmergencyMessage"
  
      .prologue
+     .line 1082
      if-eqz p2, :cond_0
  
+     .line 1083
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.provider.Telephony.SMS_EMERGENCY_CB_RECEIVED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 1084
      .local v0, intent:Landroid/content/Intent;
!     const-string/jumbo v1, "pdus"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
+     .line 1085
      const-string v1, "SMS"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 737,749 ****
--- 807,822 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1086
      const-string v1, "android.permission.RECEIVE_EMERGENCY_BROADCAST"
  
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/telephony/SMSDispatcher;->dispatch(Landroid/content/Intent;Ljava/lang/String;)V
  
+     .line 1093
      :goto_0
      return-void
  
+     .line 1088
      .end local v0           #intent:Landroid/content/Intent;
      :cond_0
      new-instance v0, Landroid/content/Intent;
***************
*** 752,762 ****
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
      .restart local v0       #intent:Landroid/content/Intent;
!     const-string v1, "pdus"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
      const-string v1, "SMS"
  
      new-instance v2, Ljava/lang/StringBuilder;
--- 825,837 ----
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 1089
      .restart local v0       #intent:Landroid/content/Intent;
!     const-string/jumbo v1, "pdus"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
+     .line 1090
      const-string v1, "SMS"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 787,792 ****
--- 862,868 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1091
      const-string v1, "android.permission.RECEIVE_SMS"
  
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/telephony/SMSDispatcher;->dispatch(Landroid/content/Intent;Ljava/lang/String;)V
***************
*** 806,815 ****
--- 882,893 ----
  
      const/4 v8, -0x1
  
+     .line 480
      invoke-virtual {p1}, Lcom/android/internal/telephony/SmsMessageBase;->getUserDataHeader()Lcom/android/internal/telephony/SmsHeader;
  
      move-result-object v13
  
+     .line 483
      .local v13, smsHeader:Lcom/android/internal/telephony/SmsHeader;
      if-eqz v13, :cond_0
  
***************
*** 817,827 ****
--- 895,907 ----
  
      if-nez v0, :cond_3
  
+     .line 485
      :cond_0
      const/4 v0, 0x1
  
      new-array v11, v0, [[B
  
+     .line 486
      .local v11, pdus:[[B
      invoke-virtual {p1}, Lcom/android/internal/telephony/SmsMessageBase;->getPdu()[B
  
***************
*** 829,840 ****
--- 909,922 ----
  
      aput-object v0, v11, v9
  
+     .line 488
      if-eqz v13, :cond_2
  
      iget-object v0, v13, Lcom/android/internal/telephony/SmsHeader;->portAddrs:Lcom/android/internal/telephony/SmsHeader$PortAddrs;
  
      if-eqz v0, :cond_2
  
+     .line 489
      iget-object v0, v13, Lcom/android/internal/telephony/SmsHeader;->portAddrs:Lcom/android/internal/telephony/SmsHeader$PortAddrs;
  
      iget v0, v0, Lcom/android/internal/telephony/SmsHeader$PortAddrs;->destPort:I
***************
*** 843,848 ****
--- 925,931 ----
  
      if-ne v0, v1, :cond_1
  
+     .line 491
      iget-object v0, p0, Lcom/android/internal/telephony/SMSDispatcher;->mWapPush:Lcom/android/internal/telephony/WapPushOverSms;
  
      invoke-virtual {p1}, Lcom/android/internal/telephony/SmsMessageBase;->getUserData()[B
***************
*** 853,862 ****
--- 936,947 ----
  
      move-result v8
  
+     .line 505
      .end local v11           #pdus:[[B
      :goto_0
      return v8
  
+     .line 494
      .restart local v11       #pdus:[[B
      :cond_1
      iget-object v0, v13, Lcom/android/internal/telephony/SmsHeader;->portAddrs:Lcom/android/internal/telephony/SmsHeader$PortAddrs;
***************
*** 867,884 ****
--- 952,973 ----
  
      goto :goto_0
  
+     .line 498
      :cond_2
      invoke-virtual {p0, v11}, Lcom/android/internal/telephony/SMSDispatcher;->dispatchPdus([[B)V
  
      goto :goto_0
  
+     .line 503
      .end local v11           #pdus:[[B
      :cond_3
      iget-object v10, v13, Lcom/android/internal/telephony/SmsHeader;->concatRef:Lcom/android/internal/telephony/SmsHeader$ConcatRef;
  
+     .line 504
      .local v10, concatRef:Lcom/android/internal/telephony/SmsHeader$ConcatRef;
      iget-object v12, v13, Lcom/android/internal/telephony/SmsHeader;->portAddrs:Lcom/android/internal/telephony/SmsHeader$PortAddrs;
  
+     .line 505
      .local v12, portAddrs:Lcom/android/internal/telephony/SmsHeader$PortAddrs;
      invoke-virtual {p1}, Lcom/android/internal/telephony/SmsMessageBase;->getPdu()[B
  
***************
*** 917,933 ****
      .parameter "pdus"
  
      .prologue
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.provider.Telephony.SMS_RECEIVED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
      .local v0, intent:Landroid/content/Intent;
!     const-string v1, "pdus"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
      const-string v1, "format"
  
      invoke-virtual {p0}, Lcom/android/internal/telephony/SMSDispatcher;->getFormat()Ljava/lang/String;
--- 1006,1025 ----
      .parameter "pdus"
  
      .prologue
+     .line 669
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.provider.Telephony.SMS_RECEIVED"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 670
      .local v0, intent:Landroid/content/Intent;
!     const-string/jumbo v1, "pdus"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
+     .line 671
      const-string v1, "format"
  
      invoke-virtual {p0}, Lcom/android/internal/telephony/SMSDispatcher;->getFormat()Ljava/lang/String;
***************
*** 936,945 ****
--- 1028,1039 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 672
      const-string v1, "android.permission.RECEIVE_SMS"
  
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/telephony/SMSDispatcher;->dispatch(Landroid/content/Intent;Ljava/lang/String;)V
  
+     .line 673
      return-void
  .end method
  
***************
*** 949,959 ****
      .parameter "port"
  
      .prologue
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "sms://localhost:"
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1043,1054 ----
      .parameter "port"
  
      .prologue
+     .line 682
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "sms://localhost:"
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 971,976 ****
--- 1066,1072 ----
  
      move-result-object v1
  
+     .line 683
      .local v1, uri:Landroid/net/Uri;
      new-instance v0, Landroid/content/Intent;
  
***************
*** 978,988 ****
  
      invoke-direct {v0, v2, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;Landroid/net/Uri;)V
  
      .local v0, intent:Landroid/content/Intent;
!     const-string v2, "pdus"
  
      invoke-virtual {v0, v2, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
      const-string v2, "format"
  
      invoke-virtual {p0}, Lcom/android/internal/telephony/SMSDispatcher;->getFormat()Ljava/lang/String;
--- 1074,1086 ----
  
      invoke-direct {v0, v2, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;Landroid/net/Uri;)V
  
+     .line 684
      .local v0, intent:Landroid/content/Intent;
!     const-string/jumbo v2, "pdus"
  
      invoke-virtual {v0, v2, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
+     .line 685
      const-string v2, "format"
  
      invoke-virtual {p0}, Lcom/android/internal/telephony/SMSDispatcher;->getFormat()Ljava/lang/String;
***************
*** 991,1000 ****
--- 1089,1100 ----
  
      invoke-virtual {v0, v2, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 686
      const-string v2, "android.permission.RECEIVE_SMS"
  
      invoke-virtual {p0, v0, v2}, Lcom/android/internal/telephony/SMSDispatcher;->dispatch(Landroid/content/Intent;Ljava/lang/String;)V
  
+     .line 687
      return-void
  .end method
  
***************
*** 1005,1016 ****
--- 1105,1118 ----
      .locals 2
  
      .prologue
+     .line 219
      const-string v0, "SMS"
  
      const-string v1, "SMSDispatcher finalized"
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 220
      return-void
  .end method
  
***************
*** 1030,1053 ****
--- 1132,1160 ----
  
      const/4 v6, 0x0
  
+     .line 240
      iget v7, p1, Landroid/os/Message;->what:I
  
      packed-switch v7, :pswitch_data_0
  
+     .line 326
      :cond_0
      :goto_0
      return-void
  
+     .line 249
      :pswitch_0
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/os/AsyncResult;
  
+     .line 251
      .local v0, ar:Landroid/os/AsyncResult;
      iget-object v7, v0, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
      if-eqz v7, :cond_1
  
+     .line 252
      const-string v6, "SMS"
  
      new-instance v7, Ljava/lang/StringBuilder;
***************
*** 1074,1084 ****
--- 1181,1193 ----
  
      goto :goto_0
  
+     .line 256
      :cond_1
      iget-object v5, v0, Landroid/os/AsyncResult;->result:Ljava/lang/Object;
  
      check-cast v5, Landroid/telephony/SmsMessage;
  
+     .line 258
      .local v5, sms:Landroid/telephony/SmsMessage;
      :try_start_0
      iget-object v7, v5, Landroid/telephony/SmsMessage;->mWrappedSmsMessage:Lcom/android/internal/telephony/SmsMessageBase;
***************
*** 1087,1099 ****
--- 1196,1211 ----
  
      move-result v3
  
+     .line 259
      .local v3, result:I
      const/4 v7, -0x1
  
      if-eq v3, v7, :cond_0
  
+     .line 262
      if-ne v3, v2, :cond_2
  
+     .line 263
      .local v2, handled:Z
      :goto_1
      const/4 v7, 0x0
***************
*** 1104,1114 ****
--- 1216,1228 ----
  
      goto :goto_0
  
+     .line 265
      .end local v2           #handled:Z
      .end local v3           #result:I
      :catch_0
      move-exception v1
  
+     .line 266
      .local v1, ex:Ljava/lang/RuntimeException;
      const-string v7, "SMS"
  
***************
*** 1116,1121 ****
--- 1230,1236 ----
  
      invoke-static {v7, v8, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 267
      const/4 v7, 0x2
  
      invoke-direct {p0, v6, v7, v9}, Lcom/android/internal/telephony/SMSDispatcher;->notifyAndAcknowledgeLastIncomingSms(ZILandroid/os/Message;)V
***************
*** 1127,1134 ****
--- 1242,1251 ----
      :cond_2
      move v2, v6
  
+     .line 262
      goto :goto_1
  
+     .line 274
      .end local v0           #ar:Landroid/os/AsyncResult;
      .end local v3           #result:I
      .end local v5           #sms:Landroid/telephony/SmsMessage;
***************
*** 1141,1146 ****
--- 1258,1264 ----
  
      goto :goto_0
  
+     .line 278
      :pswitch_2
      iget-object v6, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 1150,1155 ****
--- 1268,1274 ----
  
      goto :goto_0
  
+     .line 282
      :pswitch_3
      iget-object v6, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 1161,1166 ****
--- 1280,1286 ----
  
      goto :goto_0
  
+     .line 286
      :pswitch_4
      iget-object v6, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 1170,1177 ****
--- 1290,1299 ----
  
      invoke-virtual {v6}, Landroid/app/AlertDialog;->dismiss()V
  
+     .line 287
      iput-object v9, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 288
      iget-object v6, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
  
      invoke-virtual {v6}, Ljava/util/ArrayList;->isEmpty()Z
***************
*** 1180,1185 ****
--- 1302,1308 ----
  
      if-nez v6, :cond_0
  
+     .line 290
      :try_start_1
      iget-object v6, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
  
***************
*** 1191,1196 ****
--- 1314,1320 ----
  
      check-cast v4, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
  
+     .line 291
      .local v4, sTracker:Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      iget-object v6, v4, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
***************
*** 1202,1211 ****
--- 1326,1337 ----
  
      goto/16 :goto_0
  
+     .line 292
      .end local v4           #sTracker:Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      :catch_1
      move-exception v1
  
+     .line 293
      .local v1, ex:Landroid/app/PendingIntent$CanceledException;
      const-string v6, "SMS"
  
***************
*** 1215,1220 ****
--- 1341,1347 ----
  
      goto/16 :goto_0
  
+     .line 302
      .end local v1           #ex:Landroid/app/PendingIntent$CanceledException;
      :pswitch_5
      iget-object v6, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
***************
*** 1225,1230 ****
--- 1352,1358 ----
  
      if-nez v6, :cond_0
  
+     .line 303
      iget-object v6, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
  
      iget-object v7, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
***************
*** 1241,1246 ****
--- 1369,1375 ----
  
      check-cast v4, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
  
+     .line 304
      .restart local v4       #sTracker:Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      invoke-virtual {v4}, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->isMultipart()Z
  
***************
*** 1248,1255 ****
--- 1377,1386 ----
  
      if-eqz v6, :cond_3
  
+     .line 305
      invoke-direct {p0, v4}, Lcom/android/internal/telephony/SMSDispatcher;->sendMultipartSms(Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;)V
  
+     .line 309
      :goto_2
      iget-object v6, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 1257,1267 ****
--- 1388,1400 ----
  
      goto/16 :goto_0
  
+     .line 307
      :cond_3
      invoke-virtual {p0, v4}, Lcom/android/internal/telephony/SMSDispatcher;->sendSms(Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;)V
  
      goto :goto_2
  
+     .line 314
      .end local v4           #sTracker:Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      :pswitch_6
      iget-object v6, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
***************
*** 1272,1277 ****
--- 1405,1411 ----
  
      if-nez v6, :cond_0
  
+     .line 317
      :try_start_2
      iget-object v6, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
  
***************
*** 1289,1294 ****
--- 1423,1429 ----
  
      check-cast v4, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
  
+     .line 318
      .restart local v4       #sTracker:Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      iget-object v6, v4, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
***************
*** 1298,1303 ****
--- 1433,1439 ----
      :try_end_2
      .catch Landroid/app/PendingIntent$CanceledException; {:try_start_2 .. :try_end_2} :catch_2
  
+     .line 322
      .end local v4           #sTracker:Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      :goto_3
      iget-object v6, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
***************
*** 1306,1314 ****
--- 1442,1452 ----
  
      goto/16 :goto_0
  
+     .line 319
      :catch_2
      move-exception v1
  
+     .line 320
      .restart local v1       #ex:Landroid/app/PendingIntent$CanceledException;
      const-string v6, "SMS"
  
***************
*** 1318,1323 ****
--- 1456,1462 ----
  
      goto :goto_3
  
+     .line 240
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 1337,1342 ****
--- 1476,1482 ----
      .prologue
      const/4 v5, 0x5
  
+     .line 915
      iget-object v4, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
  
      invoke-virtual {v4}, Ljava/util/ArrayList;->size()I
***************
*** 1345,1350 ****
--- 1485,1491 ----
  
      if-lt v4, v5, :cond_0
  
+     .line 918
      :try_start_0
      iget-object v4, p1, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
***************
*** 1354,1365 ****
--- 1495,1509 ----
      :try_end_0
      .catch Landroid/app/PendingIntent$CanceledException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 942
      :goto_0
      return-void
  
+     .line 919
      :catch_0
      move-exception v2
  
+     .line 920
      .local v2, ex:Landroid/app/PendingIntent$CanceledException;
      const-string v4, "SMS"
  
***************
*** 1369,1380 ****
--- 1513,1526 ----
  
      goto :goto_0
  
+     .line 925
      .end local v2           #ex:Landroid/app/PendingIntent$CanceledException;
      :cond_0
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v3
  
+     .line 927
      .local v3, r:Landroid/content/res/Resources;
      iget-object v4, p1, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
***************
*** 1382,1387 ****
--- 1528,1534 ----
  
      move-result-object v0
  
+     .line 929
      .local v0, appName:Ljava/lang/String;
      new-instance v4, Landroid/app/AlertDialog$Builder;
  
***************
*** 1459,1464 ****
--- 1606,1612 ----
  
      move-result-object v1
  
+     .line 936
      .local v1, d:Landroid/app/AlertDialog;
      invoke-virtual {v1}, Landroid/app/AlertDialog;->getWindow()Landroid/view/Window;
  
***************
*** 1468,1479 ****
--- 1616,1630 ----
  
      invoke-virtual {v4, v5}, Landroid/view/Window;->setType(I)V
  
+     .line 937
      invoke-virtual {v1}, Landroid/app/AlertDialog;->show()V
  
+     .line 939
      iget-object v4, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSTrackers:Ljava/util/ArrayList;
  
      invoke-virtual {v4, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 940
      const/4 v4, 0x6
  
      invoke-virtual {p0, v4, v1}, Lcom/android/internal/telephony/SMSDispatcher;->obtainMessage(ILjava/lang/Object;)Landroid/os/Message;
***************
*** 1496,1554 ****
--- 1647,1718 ----
  
      const/4 v10, -0x1
  
+     .line 360
      iget-object v7, p1, Landroid/os/AsyncResult;->userObj:Ljava/lang/Object;
  
      check-cast v7, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
  
+     .line 361
      .local v7, tracker:Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      iget-object v5, v7, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
+     .line 363
      .local v5, sentIntent:Landroid/app/PendingIntent;
      iget-object v8, p1, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
      if-nez v8, :cond_4
  
+     .line 369
      iget-object v8, v7, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mDeliveryIntent:Landroid/app/PendingIntent;
  
      if-eqz v8, :cond_0
  
+     .line 371
      iget-object v8, p1, Landroid/os/AsyncResult;->result:Ljava/lang/Object;
  
      check-cast v8, Lcom/android/internal/telephony/SmsResponse;
  
      iget v2, v8, Lcom/android/internal/telephony/SmsResponse;->messageRef:I
  
+     .line 372
      .local v2, messageRef:I
      iput v2, v7, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mMessageRef:I
  
+     .line 373
      iget-object v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->deliveryPendingList:Ljava/util/ArrayList;
  
      invoke-virtual {v8, v7}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 376
      .end local v2           #messageRef:I
      :cond_0
      if-eqz v5, :cond_2
  
+     .line 378
      :try_start_0
      iget v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
      if-le v8, v10, :cond_1
  
+     .line 379
      iget v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
      add-int/lit8 v8, v8, -0x1
  
      iput v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
+     .line 382
      :cond_1
      iget v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
      if-nez v8, :cond_3
  
+     .line 383
      new-instance v4, Landroid/content/Intent;
  
      invoke-direct {v4}, Landroid/content/Intent;-><init>()V
  
+     .line 384
      .local v4, sendNext:Landroid/content/Intent;
      const-string v8, "SendNextMsg"
  
***************
*** 1556,1572 ****
--- 1720,1739 ----
  
      invoke-virtual {v4, v8, v9}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     .line 385
      iget-object v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
      const/4 v9, -0x1
  
      invoke-virtual {v5, v8, v9, v4}, Landroid/app/PendingIntent;->send(Landroid/content/Context;ILandroid/content/Intent;)V
  
+     .line 439
      .end local v4           #sendNext:Landroid/content/Intent;
      :cond_2
      :goto_0
      return-void
  
+     .line 387
      :cond_3
      const/4 v8, -0x1
  
***************
*** 1576,1586 ****
--- 1743,1755 ----
  
      goto :goto_0
  
+     .line 389
      :catch_0
      move-exception v8
  
      goto :goto_0
  
+     .line 396
      :cond_4
      iget-object v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mPhone:Lcom/android/internal/telephony/Phone;
  
***************
*** 1592,1604 ****
--- 1761,1776 ----
  
      move-result v6
  
+     .line 398
      .local v6, ss:I
      if-eqz v6, :cond_5
  
+     .line 399
      invoke-static {v6, v7}, Lcom/android/internal/telephony/SMSDispatcher;->handleNotInService(ILcom/android/internal/telephony/SMSDispatcher$SmsTracker;)V
  
      goto :goto_0
  
+     .line 400
      :cond_5
      iget-object v8, p1, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
***************
*** 1618,1633 ****
--- 1790,1808 ----
  
      if-ge v8, v11, :cond_6
  
+     .line 411
      iget v8, v7, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mRetryCount:I
  
      add-int/lit8 v8, v8, 0x1
  
      iput v8, v7, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mRetryCount:I
  
+     .line 412
      invoke-virtual {p0, v11, v7}, Lcom/android/internal/telephony/SMSDispatcher;->obtainMessage(ILjava/lang/Object;)Landroid/os/Message;
  
      move-result-object v3
  
+     .line 413
      .local v3, retryMsg:Landroid/os/Message;
      const-wide/16 v8, 0x7d0
  
***************
*** 1635,1648 ****
--- 1810,1826 ----
  
      goto :goto_0
  
+     .line 414
      .end local v3           #retryMsg:Landroid/os/Message;
      :cond_6
      iget-object v8, v7, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
      if-eqz v8, :cond_2
  
+     .line 415
      const/4 v0, 0x1
  
+     .line 417
      .local v0, error:I
      iget-object v8, p1, Landroid/os/AsyncResult;->exception:Ljava/lang/Throwable;
  
***************
*** 1658,1676 ****
--- 1836,1858 ----
  
      if-ne v8, v9, :cond_7
  
+     .line 419
      const/4 v0, 0x6
  
+     .line 423
      :cond_7
      :try_start_1
      new-instance v1, Landroid/content/Intent;
  
      invoke-direct {v1}, Landroid/content/Intent;-><init>()V
  
+     .line 424
      .local v1, fillIn:Landroid/content/Intent;
      iget-object v8, p1, Landroid/os/AsyncResult;->result:Ljava/lang/Object;
  
      if-eqz v8, :cond_8
  
+     .line 425
      const-string v9, "errorCode"
  
      iget-object v8, p1, Landroid/os/AsyncResult;->result:Ljava/lang/Object;
***************
*** 1681,1708 ****
--- 1863,1895 ----
  
      invoke-virtual {v1, v9, v8}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
+     .line 427
      :cond_8
      iget v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
      if-le v8, v10, :cond_9
  
+     .line 428
      iget v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
      add-int/lit8 v8, v8, -0x1
  
      iput v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
+     .line 431
      :cond_9
      iget v8, p0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
      if-nez v8, :cond_a
  
+     .line 432
      const-string v8, "SendNextMsg"
  
      const/4 v9, 0x1
  
      invoke-virtual {v1, v8, v9}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     .line 435
      :cond_a
      iget-object v8, v7, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;->mSentIntent:Landroid/app/PendingIntent;
  
***************
*** 1714,1719 ****
--- 1901,1907 ----
  
      goto :goto_0
  
+     .line 436
      .end local v1           #fillIn:Landroid/content/Intent;
      :catch_1
      move-exception v8
***************
*** 1733,1756 ****
--- 1921,1949 ----
      .parameter "isCdmaWapPush"
  
      .prologue
+     .line 534
      const/16 v20, 0x0
  
      check-cast v20, [[B
  
+     .line 535
      .local v20, pdus:[[B
      const/4 v9, 0x0
  
+     .line 538
      .local v9, cursor:Landroid/database/Cursor;
      :try_start_0
      invoke-static/range {p3 .. p3}, Ljava/lang/Integer;->toString(I)Ljava/lang/String;
  
      move-result-object v21
  
+     .line 539
      .local v21, refNumber:Ljava/lang/String;
      invoke-static/range {p4 .. p4}, Ljava/lang/Integer;->toString(I)Ljava/lang/String;
  
      move-result-object v22
  
+     .line 542
      .local v22, seqNumber:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 1784,1795 ****
--- 1977,1990 ----
  
      move-result-object v9
  
+     .line 547
      invoke-interface {v9}, Landroid/database/Cursor;->moveToNext()Z
  
      move-result v3
  
      if-eqz v3, :cond_2
  
+     .line 548
      const-string v3, "SMS"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 1838,1854 ****
--- 2033,2052 ----
  
      invoke-static {v3, v4}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 550
      const/4 v3, 0x0
  
      invoke-interface {v9, v3}, Landroid/database/Cursor;->getString(I)Ljava/lang/String;
  
      move-result-object v18
  
+     .line 551
      .local v18, oldPduString:Ljava/lang/String;
      invoke-static/range {v18 .. v18}, Lcom/android/internal/util/HexDump;->hexStringToByteArray(Ljava/lang/String;)[B
  
      move-result-object v17
  
+     .line 552
      .local v17, oldPdu:[B
      move-object/from16 v0, v17
  
***************
*** 1860,1865 ****
--- 2058,2064 ----
  
      if-nez v3, :cond_0
  
+     .line 553
      const-string v3, "SMS"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 1903,1915 ****
--- 2102,2117 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Landroid/database/SQLException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 556
      :cond_0
      const/4 v3, 0x1
  
+     .line 614
      if-eqz v9, :cond_1
  
      invoke-interface {v9}, Landroid/database/Cursor;->close()V
  
+     .line 660
      .end local v17           #oldPdu:[B
      .end local v18           #oldPduString:Ljava/lang/String;
      .end local v21           #refNumber:Ljava/lang/String;
***************
*** 1918,1931 ****
--- 2120,2136 ----
      :goto_0
      return v3
  
+     .line 558
      .restart local v21       #refNumber:Ljava/lang/String;
      .restart local v22       #seqNumber:Ljava/lang/String;
      :cond_2
      :try_start_1
      invoke-interface {v9}, Landroid/database/Cursor;->close()V
  
+     .line 561
      const-string v6, "address=? AND reference_number=?"
  
+     .line 562
      .local v6, where:Ljava/lang/String;
      const/4 v3, 0x2
  
***************
*** 1939,1944 ****
--- 2144,2150 ----
  
      aput-object v21, v7, v3
  
+     .line 563
      .local v7, whereArgs:[Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 1954,1972 ****
--- 2160,2182 ----
  
      move-result-object v9
  
+     .line 565
      invoke-interface {v9}, Landroid/database/Cursor;->getCount()I
  
      move-result v10
  
+     .line 566
      .local v10, cursorCount:I
      add-int/lit8 v3, p5, -0x1
  
      if-eq v10, v3, :cond_4
  
+     .line 568
      new-instance v23, Landroid/content/ContentValues;
  
      invoke-direct/range {v23 .. v23}, Landroid/content/ContentValues;-><init>()V
  
+     .line 569
      .local v23, values:Landroid/content/ContentValues;
      const-string v3, "date"
  
***************
*** 1978,1984 ****
  
      invoke-virtual {v0, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
!     const-string v3, "pdu"
  
      invoke-static/range {p1 .. p1}, Lcom/android/internal/util/HexDump;->toHexString([B)Ljava/lang/String;
  
--- 2188,2195 ----
  
      invoke-virtual {v0, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
!     .line 570
!     const-string/jumbo v3, "pdu"
  
      invoke-static/range {p1 .. p1}, Lcom/android/internal/util/HexDump;->toHexString([B)Ljava/lang/String;
  
***************
*** 1988,1993 ****
--- 2199,2205 ----
  
      invoke-virtual {v0, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 571
      const-string v3, "address"
  
      move-object/from16 v0, v23
***************
*** 1996,2002 ****
  
      invoke-virtual {v0, v3, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v3, "reference_number"
  
      invoke-static/range {p3 .. p3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 2208,2215 ----
  
      invoke-virtual {v0, v3, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 572
!     const-string/jumbo v3, "reference_number"
  
      invoke-static/range {p3 .. p3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 2006,2011 ****
--- 2219,2225 ----
  
      invoke-virtual {v0, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 573
      const-string v3, "count"
  
      invoke-static/range {p5 .. p5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 2016,2022 ****
  
      invoke-virtual {v0, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     const-string v3, "sequence"
  
      invoke-static/range {p4 .. p4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 2230,2237 ----
  
      invoke-virtual {v0, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     .line 574
!     const-string/jumbo v3, "sequence"
  
      invoke-static/range {p4 .. p4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 2026,2037 ****
--- 2241,2254 ----
  
      invoke-virtual {v0, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 575
      const/4 v3, -0x1
  
      move/from16 v0, p8
  
      if-eq v0, v3, :cond_3
  
+     .line 576
      const-string v3, "destination_port"
  
      invoke-static/range {p8 .. p8}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 2042,2047 ****
--- 2259,2265 ----
  
      invoke-virtual {v0, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 578
      :cond_3
      move-object/from16 v0, p0
  
***************
*** 2056,2069 ****
--- 2274,2290 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/database/SQLException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 579
      const/4 v3, 0x1
  
+     .line 614
      if-eqz v9, :cond_1
  
      invoke-interface {v9}, Landroid/database/Cursor;->close()V
  
      goto/16 :goto_0
  
+     .line 583
      .end local v23           #values:Landroid/content/ContentValues;
      :cond_4
      :try_start_2
***************
*** 2073,2097 ****
--- 2294,2324 ----
  
      move-object/from16 v20, v0
  
+     .line 584
      const/4 v15, 0x0
  
      .local v15, i:I
      :goto_1
      if-ge v15, v10, :cond_7
  
+     .line 585
      invoke-interface {v9}, Landroid/database/Cursor;->moveToNext()Z
  
+     .line 586
      const/4 v3, 0x1
  
      invoke-interface {v9, v3}, Landroid/database/Cursor;->getInt(I)I
  
      move-result v11
  
+     .line 588
      .local v11, cursorSequence:I
      if-nez p9, :cond_5
  
+     .line 589
      add-int/lit8 v11, v11, -0x1
  
+     .line 591
      :cond_5
      const/4 v3, 0x0
  
***************
*** 2105,2110 ****
--- 2332,2338 ----
  
      aput-object v3, v20, v11
  
+     .line 596
      if-nez v11, :cond_6
  
      const/4 v3, 0x2
***************
*** 2115,2137 ****
--- 2343,2370 ----
  
      if-nez v3, :cond_6
  
+     .line 597
      const/4 v3, 0x2
  
      invoke-interface {v9, v3}, Landroid/database/Cursor;->getInt(I)I
  
      move-result p8
  
+     .line 584
      :cond_6
      add-int/lit8 v15, v15, 0x1
  
      goto :goto_1
  
+     .line 602
      .end local v11           #cursorSequence:I
      :cond_7
      if-eqz p9, :cond_9
  
+     .line 603
      aput-object p1, v20, p4
  
+     .line 609
      :goto_2
      move-object/from16 v0, p0
  
***************
*** 2144,2160 ****
--- 2377,2397 ----
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
      .catch Landroid/database/SQLException; {:try_start_2 .. :try_end_2} :catch_0
  
+     .line 614
      if-eqz v9, :cond_8
  
      invoke-interface {v9}, Landroid/database/Cursor;->close()V
  
+     .line 618
      :cond_8
      if-eqz p9, :cond_d
  
+     .line 620
      new-instance v19, Ljava/io/ByteArrayOutputStream;
  
      invoke-direct/range {v19 .. v19}, Ljava/io/ByteArrayOutputStream;-><init>()V
  
+     .line 621
      .local v19, output:Ljava/io/ByteArrayOutputStream;
      const/4 v15, 0x0
  
***************
*** 2163,2168 ****
--- 2400,2406 ----
  
      if-ge v15, v0, :cond_b
  
+     .line 623
      aget-object v3, v20, v15
  
      const/4 v4, 0x0
***************
*** 2175,2184 ****
--- 2413,2424 ----
  
      invoke-virtual {v0, v3, v4, v5}, Ljava/io/ByteArrayOutputStream;->write([BII)V
  
+     .line 621
      add-int/lit8 v15, v15, 0x1
  
      goto :goto_3
  
+     .line 605
      .end local v19           #output:Ljava/io/ByteArrayOutputStream;
      :cond_9
      add-int/lit8 v3, p4, -0x1
***************
*** 2191,2196 ****
--- 2431,2437 ----
  
      goto :goto_2
  
+     .line 610
      .end local v6           #where:Ljava/lang/String;
      .end local v7           #whereArgs:[Ljava/lang/String;
      .end local v10           #cursorCount:I
***************
*** 2200,2205 ****
--- 2441,2447 ----
      :catch_0
      move-exception v14
  
+     .line 611
      .local v14, e:Landroid/database/SQLException;
      :try_start_4
      const-string v3, "SMS"
***************
*** 2210,2217 ****
--- 2452,2461 ----
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
  
+     .line 612
      const/4 v3, 0x2
  
+     .line 614
      if-eqz v9, :cond_1
  
      invoke-interface {v9}, Landroid/database/Cursor;->close()V
***************
*** 2229,2234 ****
--- 2473,2479 ----
      :cond_a
      throw v3
  
+     .line 625
      .restart local v6       #where:Ljava/lang/String;
      .restart local v7       #whereArgs:[Ljava/lang/String;
      .restart local v10       #cursorCount:I
***************
*** 2241,2246 ****
--- 2486,2492 ----
  
      move-result-object v13
  
+     .line 628
      .local v13, datagram:[B
      const/16 v3, 0xb84
  
***************
*** 2248,2253 ****
--- 2494,2500 ----
  
      if-ne v0, v3, :cond_c
  
+     .line 630
      move-object/from16 v0, p0
  
      iget-object v3, v0, Lcom/android/internal/telephony/SMSDispatcher;->mWapPush:Lcom/android/internal/telephony/WapPushOverSms;
***************
*** 2258,2263 ****
--- 2505,2511 ----
  
      goto/16 :goto_0
  
+     .line 632
      :cond_c
      const/4 v3, 0x1
  
***************
*** 2265,2274 ****
--- 2513,2524 ----
  
      move-object/from16 v20, v0
  
+     .line 633
      const/4 v3, 0x0
  
      aput-object v13, v20, v3
  
+     .line 635
      move-object/from16 v0, p0
  
      move-object/from16 v1, v20
***************
*** 2277,2286 ****
--- 2527,2538 ----
  
      invoke-virtual {v0, v1, v2}, Lcom/android/internal/telephony/SMSDispatcher;->dispatchPortAddressedPdus([[BI)V
  
+     .line 636
      const/4 v3, -0x1
  
      goto/16 :goto_0
  
+     .line 641
      .end local v13           #datagram:[B
      .end local v19           #output:Ljava/io/ByteArrayOutputStream;
      :cond_d
***************
*** 2290,2305 ****
--- 2542,2560 ----
  
      if-eq v0, v3, :cond_10
  
+     .line 642
      const/16 v3, 0xb84
  
      move/from16 v0, p8
  
      if-ne v0, v3, :cond_f
  
+     .line 644
      new-instance v19, Ljava/io/ByteArrayOutputStream;
  
      invoke-direct/range {v19 .. v19}, Ljava/io/ByteArrayOutputStream;-><init>()V
  
+     .line 645
      .restart local v19       #output:Ljava/io/ByteArrayOutputStream;
      const/4 v15, 0x0
  
***************
*** 2308,2313 ****
--- 2563,2569 ----
  
      if-ge v15, v0, :cond_e
  
+     .line 646
      aget-object v3, v20, v15
  
      invoke-virtual/range {p0 .. p0}, Lcom/android/internal/telephony/SMSDispatcher;->getFormat()Ljava/lang/String;
***************
*** 2318,2328 ****
--- 2574,2586 ----
  
      move-result-object v16
  
+     .line 647
      .local v16, msg:Landroid/telephony/SmsMessage;
      invoke-virtual/range {v16 .. v16}, Landroid/telephony/SmsMessage;->getUserData()[B
  
      move-result-object v12
  
+     .line 648
      .local v12, data:[B
      const/4 v3, 0x0
  
***************
*** 2332,2341 ****
--- 2590,2601 ----
  
      invoke-virtual {v0, v12, v3, v4}, Ljava/io/ByteArrayOutputStream;->write([BII)V
  
+     .line 645
      add-int/lit8 v15, v15, 0x1
  
      goto :goto_4
  
+     .line 651
      .end local v12           #data:[B
      .end local v16           #msg:Landroid/telephony/SmsMessage;
      :cond_e
***************
*** 2353,2358 ****
--- 2613,2619 ----
  
      goto/16 :goto_0
  
+     .line 654
      .end local v19           #output:Ljava/io/ByteArrayOutputStream;
      :cond_f
      move-object/from16 v0, p0
***************
*** 2363,2373 ****
--- 2624,2636 ----
  
      invoke-virtual {v0, v1, v2}, Lcom/android/internal/telephony/SMSDispatcher;->dispatchPortAddressedPdus([[BI)V
  
+     .line 660
      :goto_5
      const/4 v3, -0x1
  
      goto/16 :goto_0
  
+     .line 658
      :cond_10
      move-object/from16 v0, p0
  
***************
*** 2409,2414 ****
--- 2672,2678 ----
      .end annotation
  
      .prologue
+     .line 786
      .local p3, parts:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
      .local p4, sentIntents:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/app/PendingIntent;>;"
      .local p5, deliveryIntents:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/app/PendingIntent;>;"
***************
*** 2418,2438 ****
--- 2682,2707 ----
  
      and-int/lit16 v15, v1, 0xff
  
+     .line 787
      .local v15, refNumber:I
      invoke-virtual/range {p3 .. p3}, Ljava/util/ArrayList;->size()I
  
      move-result v14
  
+     .line 788
      .local v14, msgCount:I
      const/4 v6, 0x0
  
+     .line 790
      .local v6, encoding:I
      move-object/from16 v0, p0
  
      iput v14, v0, Lcom/android/internal/telephony/SMSDispatcher;->mRemainingMessages:I
  
+     .line 792
      new-array v12, v14, [Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
  
+     .line 793
      .local v12, encodingForParts:[Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
      const/4 v13, 0x0
  
***************
*** 2440,2445 ****
--- 2709,2715 ----
      :goto_0
      if-ge v13, v14, :cond_2
  
+     .line 794
      move-object/from16 v0, p3
  
      invoke-virtual {v0, v13}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 2456,2461 ****
--- 2726,2732 ----
  
      move-result-object v11
  
+     .line 795
      .local v11, details:Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
      iget v1, v11, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->codeUnitSize:I
  
***************
*** 2467,2482 ****
--- 2738,2757 ----
  
      if-ne v6, v1, :cond_1
  
+     .line 798
      :cond_0
      iget v6, v11, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->codeUnitSize:I
  
+     .line 800
      :cond_1
      aput-object v11, v12, v13
  
+     .line 793
      add-int/lit8 v13, v13, 0x1
  
      goto :goto_0
  
+     .line 803
      .end local v11           #details:Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;
      :cond_2
      const/4 v13, 0x0
***************
*** 2484,2532 ****
--- 2759,2819 ----
      :goto_1
      if-ge v13, v14, :cond_7
  
+     .line 804
      new-instance v10, Lcom/android/internal/telephony/SmsHeader$ConcatRef;
  
      invoke-direct {v10}, Lcom/android/internal/telephony/SmsHeader$ConcatRef;-><init>()V
  
+     .line 805
      .local v10, concatRef:Lcom/android/internal/telephony/SmsHeader$ConcatRef;
      iput v15, v10, Lcom/android/internal/telephony/SmsHeader$ConcatRef;->refNumber:I
  
+     .line 806
      add-int/lit8 v1, v13, 0x1
  
      iput v1, v10, Lcom/android/internal/telephony/SmsHeader$ConcatRef;->seqNumber:I
  
+     .line 807
      iput v14, v10, Lcom/android/internal/telephony/SmsHeader$ConcatRef;->msgCount:I
  
+     .line 814
      const/4 v1, 0x1
  
      iput-boolean v1, v10, Lcom/android/internal/telephony/SmsHeader$ConcatRef;->isEightBits:Z
  
+     .line 815
      new-instance v5, Lcom/android/internal/telephony/SmsHeader;
  
      invoke-direct {v5}, Lcom/android/internal/telephony/SmsHeader;-><init>()V
  
+     .line 816
      .local v5, smsHeader:Lcom/android/internal/telephony/SmsHeader;
      iput-object v10, v5, Lcom/android/internal/telephony/SmsHeader;->concatRef:Lcom/android/internal/telephony/SmsHeader$ConcatRef;
  
+     .line 819
      const/4 v1, 0x1
  
      if-ne v6, v1, :cond_3
  
+     .line 820
      aget-object v1, v12, v13
  
      iget v1, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->languageTable:I
  
      iput v1, v5, Lcom/android/internal/telephony/SmsHeader;->languageTable:I
  
+     .line 821
      aget-object v1, v12, v13
  
      iget v1, v1, Lcom/android/internal/telephony/SmsMessageBase$TextEncodingDetails;->languageShiftTable:I
  
      iput v1, v5, Lcom/android/internal/telephony/SmsHeader;->languageShiftTable:I
  
+     .line 824
      :cond_3
      const/4 v7, 0x0
  
+     .line 825
      .local v7, sentIntent:Landroid/app/PendingIntent;
      if-eqz p4, :cond_4
  
***************
*** 2536,2541 ****
--- 2823,2829 ----
  
      if-le v1, v13, :cond_4
  
+     .line 826
      move-object/from16 v0, p4
  
      invoke-virtual {v0, v13}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 2545,2554 ****
--- 2833,2844 ----
      .end local v7           #sentIntent:Landroid/app/PendingIntent;
      check-cast v7, Landroid/app/PendingIntent;
  
+     .line 829
      .restart local v7       #sentIntent:Landroid/app/PendingIntent;
      :cond_4
      const/4 v8, 0x0
  
+     .line 830
      .local v8, deliveryIntent:Landroid/app/PendingIntent;
      if-eqz p5, :cond_5
  
***************
*** 2558,2563 ****
--- 2848,2854 ----
  
      if-le v1, v13, :cond_5
  
+     .line 831
      move-object/from16 v0, p5
  
      invoke-virtual {v0, v13}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 2567,2572 ****
--- 2858,2864 ----
      .end local v8           #deliveryIntent:Landroid/app/PendingIntent;
      check-cast v8, Landroid/app/PendingIntent;
  
+     .line 834
      .restart local v8       #deliveryIntent:Landroid/app/PendingIntent;
      :cond_5
      move-object/from16 v0, p3
***************
*** 2592,2606 ****
--- 2884,2901 ----
  
      invoke-virtual/range {v1 .. v9}, Lcom/android/internal/telephony/SMSDispatcher;->sendNewSubmitPdu(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Lcom/android/internal/telephony/SmsHeader;ILandroid/app/PendingIntent;Landroid/app/PendingIntent;Z)V
  
+     .line 803
      add-int/lit8 v13, v13, 0x1
  
      goto :goto_1
  
+     .line 834
      :cond_6
      const/4 v9, 0x0
  
      goto :goto_2
  
+     .line 838
      .end local v5           #smsHeader:Lcom/android/internal/telephony/SmsHeader;
      .end local v7           #sentIntent:Landroid/app/PendingIntent;
      .end local v8           #deliveryIntent:Landroid/app/PendingIntent;
***************
*** 2622,2633 ****
--- 2917,2931 ----
      .prologue
      const/4 v6, 0x4
  
+     .line 870
      iget-boolean v4, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsSendDisabled:Z
  
      if-eqz v4, :cond_2
  
+     .line 871
      if-eqz p3, :cond_0
  
+     .line 873
      const/4 v4, 0x4
  
      :try_start_0
***************
*** 2635,2640 ****
--- 2933,2939 ----
      :try_end_0
      .catch Landroid/app/PendingIntent$CanceledException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 876
      :cond_0
      :goto_0
      const-string v4, "SMS"
***************
*** 2643,2657 ****
--- 2942,2960 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 907
      :cond_1
      :goto_1
      return-void
  
+     .line 880
      :cond_2
      if-nez p2, :cond_3
  
+     .line 881
      if-eqz p3, :cond_1
  
+     .line 883
      const/4 v4, 0x3
  
      :try_start_1
***************
*** 2661,2689 ****
  
      goto :goto_1
  
      :catch_0
      move-exception v4
  
      goto :goto_1
  
      :cond_3
      new-instance v1, Ljava/util/HashMap;
  
      invoke-direct {v1}, Ljava/util/HashMap;-><init>()V
  
      .local v1, map:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/String;Ljava/lang/Object;>;"
!     const-string v4, "smsc"
  
      invoke-virtual {v1, v4, p1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     const-string v4, "pdu"
  
      invoke-virtual {v1, v4, p2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      new-instance v3, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
  
      invoke-direct {v3, v1, p3, p4}, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;-><init>(Ljava/util/HashMap;Landroid/app/PendingIntent;Landroid/app/PendingIntent;)V
  
      .local v3, tracker:Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      iget-object v4, p0, Lcom/android/internal/telephony/SMSDispatcher;->mPhone:Lcom/android/internal/telephony/Phone;
  
--- 2964,2998 ----
  
      goto :goto_1
  
+     .line 884
      :catch_0
      move-exception v4
  
      goto :goto_1
  
+     .line 889
      :cond_3
      new-instance v1, Ljava/util/HashMap;
  
      invoke-direct {v1}, Ljava/util/HashMap;-><init>()V
  
+     .line 890
      .local v1, map:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/String;Ljava/lang/Object;>;"
!     const-string/jumbo v4, "smsc"
  
      invoke-virtual {v1, v4, p1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
!     .line 891
!     const-string/jumbo v4, "pdu"
  
      invoke-virtual {v1, v4, p2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 893
      new-instance v3, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
  
      invoke-direct {v3, v1, p3, p4}, Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;-><init>(Ljava/util/HashMap;Landroid/app/PendingIntent;Landroid/app/PendingIntent;)V
  
+     .line 895
      .local v3, tracker:Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;
      iget-object v4, p0, Lcom/android/internal/telephony/SMSDispatcher;->mPhone:Lcom/android/internal/telephony/Phone;
  
***************
*** 2695,2712 ****
--- 3004,3025 ----
  
      move-result v2
  
+     .line 897
      .local v2, ss:I
      if-eqz v2, :cond_4
  
+     .line 898
      invoke-static {v2, v3}, Lcom/android/internal/telephony/SMSDispatcher;->handleNotInService(ILcom/android/internal/telephony/SMSDispatcher$SmsTracker;)V
  
      goto :goto_1
  
+     .line 900
      :cond_4
      invoke-static {p3}, Lcom/android/internal/telephony/SMSDispatcher;->getAppNameByIntent(Landroid/app/PendingIntent;)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 901
      .local v0, appName:Ljava/lang/String;
      iget-object v4, p0, Lcom/android/internal/telephony/SMSDispatcher;->mUsageMonitor:Lcom/android/internal/telephony/SmsUsageMonitor;
  
***************
*** 2718,2727 ****
--- 3031,3042 ----
  
      if-eqz v4, :cond_5
  
+     .line 902
      invoke-virtual {p0, v3}, Lcom/android/internal/telephony/SMSDispatcher;->sendSms(Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;)V
  
      goto :goto_1
  
+     .line 904
      :cond_5
      invoke-virtual {p0, v6, v3}, Lcom/android/internal/telephony/SMSDispatcher;->obtainMessage(ILjava/lang/Object;)Landroid/os/Message;
  
***************
*** 2731,2736 ****
--- 3046,3052 ----
  
      goto :goto_1
  
+     .line 874
      .end local v0           #appName:Ljava/lang/String;
      .end local v1           #map:Ljava/util/HashMap;,"Ljava/util/HashMap<Ljava/lang/String;Ljava/lang/Object;>;"
      .end local v2           #ss:I
