*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/SmsMessageBase$TextEncodingDetails.smali	2012-03-23 19:37:47.339638320 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/SmsMessageBase$TextEncodingDetails.smali.original	2012-03-23 19:34:46.335337711 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .locals 0
  
      .prologue
+     .line 94
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,49 ****
--- 45,51 ----
      .locals 2
  
      .prologue
+     .line 132
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
