*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/SmsResponse.smali	2012-03-23 19:37:39.679498636 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/SmsResponse.smali.original	2012-03-23 19:34:46.201335267 -0500
***************
*** 19,32 ****
--- 19,37 ----
      .parameter "errorCode"
  
      .prologue
+     .line 35
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 36
      iput p1, p0, Lcom/android/internal/telephony/SmsResponse;->messageRef:I
  
+     .line 37
      iput-object p2, p0, Lcom/android/internal/telephony/SmsResponse;->ackPdu:Ljava/lang/String;
  
+     .line 38
      iput p3, p0, Lcom/android/internal/telephony/SmsResponse;->errorCode:I
  
+     .line 39
      return-void
  .end method
  
***************
*** 36,46 ****
      .locals 3
  
      .prologue
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "{ messageRef = "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 41,52 ----
      .locals 3
  
      .prologue
+     .line 42
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "{ messageRef = "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 76,82 ****
  
      move-result-object v1
  
!     const-string v2, "}"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 82,88 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "}"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 86,91 ****
--- 92,98 ----
  
      move-result-object v0
  
+     .line 46
      .local v0, ret:Ljava/lang/String;
      return-object v0
  .end method
