*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/TelephonyIntents.smali	2012-03-23 19:37:42.992559050 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/TelephonyIntents.smali.original	2012-03-23 19:34:46.274336599 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .locals 0
  
      .prologue
+     .line 26
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
