*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/UUSInfo.smali	2012-03-23 19:37:43.169562278 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/UUSInfo.smali.original	2012-03-23 19:34:46.276336635 -0500
***************
*** 42,61 ****
--- 42,66 ----
      .locals 1
  
      .prologue
+     .line 65
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 66
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/telephony/UUSInfo;->uusType:I
  
+     .line 67
      const/4 v0, 0x4
  
      iput v0, p0, Lcom/android/internal/telephony/UUSInfo;->uusDcs:I
  
+     .line 68
      const/4 v0, 0x0
  
      iput-object v0, p0, Lcom/android/internal/telephony/UUSInfo;->uusData:[B
  
+     .line 69
      return-void
  .end method
  
***************
*** 66,79 ****
--- 71,89 ----
      .parameter "uusData"
  
      .prologue
+     .line 71
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 72
      iput p1, p0, Lcom/android/internal/telephony/UUSInfo;->uusType:I
  
+     .line 73
      iput p2, p0, Lcom/android/internal/telephony/UUSInfo;->uusDcs:I
  
+     .line 74
      iput-object p3, p0, Lcom/android/internal/telephony/UUSInfo;->uusData:[B
  
+     .line 75
      return-void
  .end method
  
***************
*** 83,88 ****
--- 93,99 ----
      .locals 1
  
      .prologue
+     .line 78
      iget v0, p0, Lcom/android/internal/telephony/UUSInfo;->uusDcs:I
  
      return v0
***************
*** 92,97 ****
--- 103,109 ----
      .locals 1
  
      .prologue
+     .line 86
      iget v0, p0, Lcom/android/internal/telephony/UUSInfo;->uusType:I
  
      return v0
***************
*** 101,106 ****
--- 113,119 ----
      .locals 1
  
      .prologue
+     .line 94
      iget-object v0, p0, Lcom/android/internal/telephony/UUSInfo;->uusData:[B
  
      return-object v0
***************
*** 111,118 ****
--- 124,133 ----
      .parameter "uusDcs"
  
      .prologue
+     .line 82
      iput p1, p0, Lcom/android/internal/telephony/UUSInfo;->uusDcs:I
  
+     .line 83
      return-void
  .end method
  
***************
*** 121,128 ****
--- 136,145 ----
      .parameter "uusType"
  
      .prologue
+     .line 90
      iput p1, p0, Lcom/android/internal/telephony/UUSInfo;->uusType:I
  
+     .line 91
      return-void
  .end method
  
***************
*** 131,137 ****
--- 148,156 ----
      .parameter "uusData"
  
      .prologue
+     .line 98
      iput-object p1, p0, Lcom/android/internal/telephony/UUSInfo;->uusData:[B
  
+     .line 99
      return-void
  .end method
