*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/WspTypeDecoder.smali	2012-03-23 19:37:42.423548674 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/WspTypeDecoder.smali.original	2012-03-23 19:34:46.261336361 -0500
***************
*** 84,101 ****
--- 84,104 ----
  
      const/4 v3, 0x0
  
+     .line 34
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      sput-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
+     .line 37
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      sput-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
+     .line 44
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      invoke-static {v3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 106,141 ****
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      invoke-static {v4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
      move-result-object v1
  
!     const-string v2, "text/*"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      invoke-static {v5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
      move-result-object v1
  
!     const-string v2, "text/html"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      invoke-static {v6}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
      move-result-object v1
  
!     const-string v2, "text/plain"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/4 v1, 0x4
--- 109,148 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 45
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      invoke-static {v4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/*"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 46
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      invoke-static {v5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/html"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 47
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      invoke-static {v6}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/plain"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 48
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/4 v1, 0x4
***************
*** 144,153 ****
  
      move-result-object v1
  
!     const-string v2, "text/x-hdml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/4 v1, 0x5
--- 151,161 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/x-hdml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 49
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/4 v1, 0x5
***************
*** 156,165 ****
  
      move-result-object v1
  
!     const-string v2, "text/x-ttml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/4 v1, 0x6
--- 164,174 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/x-ttml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 50
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/4 v1, 0x6
***************
*** 168,187 ****
  
      move-result-object v1
  
!     const-string v2, "text/x-vCalendar"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      invoke-static {v7}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
      move-result-object v1
  
!     const-string v2, "text/x-vCard"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x8
--- 177,198 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/x-vCalendar"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 51
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      invoke-static {v7}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/x-vCard"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 52
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x8
***************
*** 190,199 ****
  
      move-result-object v1
  
!     const-string v2, "text/vnd.wap.wml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x9
--- 201,211 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/vnd.wap.wml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 53
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x9
***************
*** 202,211 ****
  
      move-result-object v1
  
!     const-string v2, "text/vnd.wap.wmlscript"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xa
--- 214,224 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/vnd.wap.wmlscript"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 54
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xa
***************
*** 214,223 ****
  
      move-result-object v1
  
!     const-string v2, "text/vnd.wap.wta-event"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xb
--- 227,237 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/vnd.wap.wta-event"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 55
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xb
***************
*** 226,235 ****
  
      move-result-object v1
  
!     const-string v2, "multipart/*"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xc
--- 240,250 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "multipart/*"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 56
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xc
***************
*** 238,247 ****
  
      move-result-object v1
  
!     const-string v2, "multipart/mixed"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xd
--- 253,263 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "multipart/mixed"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 57
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xd
***************
*** 250,259 ****
  
      move-result-object v1
  
!     const-string v2, "multipart/form-data"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xe
--- 266,276 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "multipart/form-data"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 58
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xe
***************
*** 262,271 ****
  
      move-result-object v1
  
!     const-string v2, "multipart/byterantes"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xf
--- 279,289 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "multipart/byterantes"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 59
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0xf
***************
*** 274,283 ****
  
      move-result-object v1
  
!     const-string v2, "multipart/alternative"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x10
--- 292,302 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "multipart/alternative"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 60
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x10
***************
*** 290,295 ****
--- 309,315 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 61
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x11
***************
*** 302,307 ****
--- 322,328 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 62
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x12
***************
*** 314,319 ****
--- 335,341 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 63
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x13
***************
*** 326,331 ****
--- 348,354 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 64
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x14
***************
*** 338,343 ****
--- 361,367 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 65
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x15
***************
*** 350,355 ****
--- 374,380 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 66
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x16
***************
*** 362,367 ****
--- 387,393 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 67
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x17
***************
*** 374,379 ****
--- 400,406 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 68
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x18
***************
*** 386,391 ****
--- 413,419 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 69
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x19
***************
*** 398,403 ****
--- 426,432 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 70
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x1a
***************
*** 410,415 ****
--- 439,445 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 71
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x1b
***************
*** 422,427 ****
--- 452,458 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 72
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x1c
***************
*** 434,439 ****
--- 465,471 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 73
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x1d
***************
*** 446,451 ****
--- 478,484 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 74
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x1e
***************
*** 458,463 ****
--- 491,497 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 75
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x1f
***************
*** 470,475 ****
--- 504,510 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 76
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x20
***************
*** 482,487 ****
--- 517,523 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 77
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x21
***************
*** 494,499 ****
--- 530,536 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 78
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x22
***************
*** 506,511 ****
--- 543,549 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 79
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x23
***************
*** 518,523 ****
--- 556,562 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 80
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x24
***************
*** 530,535 ****
--- 569,575 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 81
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x25
***************
*** 542,547 ****
--- 582,588 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 82
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x26
***************
*** 554,559 ****
--- 595,601 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 83
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x27
***************
*** 566,571 ****
--- 608,614 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 84
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x28
***************
*** 574,583 ****
  
      move-result-object v1
  
!     const-string v2, "text/xml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x29
--- 617,627 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/xml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 85
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x29
***************
*** 590,595 ****
--- 634,640 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 86
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x2a
***************
*** 602,607 ****
--- 647,653 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 87
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x2b
***************
*** 614,619 ****
--- 660,666 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 88
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x2c
***************
*** 626,631 ****
--- 673,679 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 89
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x2d
***************
*** 634,643 ****
  
      move-result-object v1
  
!     const-string v2, "text/vnd.wap.si"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x2e
--- 682,692 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/vnd.wap.si"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 90
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x2e
***************
*** 650,655 ****
--- 699,705 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 91
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x2f
***************
*** 658,667 ****
  
      move-result-object v1
  
!     const-string v2, "text/vnd.wap.sl"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x30
--- 708,718 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/vnd.wap.sl"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 92
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x30
***************
*** 674,679 ****
--- 725,731 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 93
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x31
***************
*** 682,691 ****
  
      move-result-object v1
  
!     const-string v2, "text/vnd.wap.co"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x32
--- 734,744 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/vnd.wap.co"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 94
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x32
***************
*** 698,703 ****
--- 751,757 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 95
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x33
***************
*** 710,715 ****
--- 764,770 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 96
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x34
***************
*** 722,727 ****
--- 777,783 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 97
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x35
***************
*** 730,739 ****
  
      move-result-object v1
  
!     const-string v2, "text/vnd.wap.connectivity-xml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x36
--- 786,796 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/vnd.wap.connectivity-xml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 98
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x36
***************
*** 746,751 ****
--- 803,809 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 99
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x37
***************
*** 758,763 ****
--- 816,822 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 100
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x38
***************
*** 770,775 ****
--- 829,835 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 101
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x39
***************
*** 782,787 ****
--- 842,848 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 102
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x3a
***************
*** 794,799 ****
--- 855,861 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 103
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x3b
***************
*** 806,811 ****
--- 868,874 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 104
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x3c
***************
*** 818,823 ****
--- 881,887 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 105
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x3d
***************
*** 826,835 ****
  
      move-result-object v1
  
!     const-string v2, "text/css"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x3e
--- 890,900 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/css"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 106
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x3e
***************
*** 842,847 ****
--- 907,913 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 107
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x3f
***************
*** 854,859 ****
--- 920,926 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 108
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x40
***************
*** 866,871 ****
--- 933,939 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 109
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x41
***************
*** 878,883 ****
--- 946,952 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 110
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x42
***************
*** 890,895 ****
--- 959,965 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 111
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x43
***************
*** 902,907 ****
--- 972,978 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 112
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x44
***************
*** 914,919 ****
--- 985,991 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 113
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x45
***************
*** 926,931 ****
--- 998,1004 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 114
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x46
***************
*** 938,943 ****
--- 1011,1017 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 115
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x47
***************
*** 950,955 ****
--- 1024,1030 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 116
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x48
***************
*** 962,967 ****
--- 1037,1043 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 117
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x49
***************
*** 974,979 ****
--- 1050,1056 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 118
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x4a
***************
*** 986,991 ****
--- 1063,1069 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 119
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x4b
***************
*** 998,1003 ****
--- 1076,1082 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 120
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x4c
***************
*** 1010,1015 ****
--- 1089,1095 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 121
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x4d
***************
*** 1022,1027 ****
--- 1102,1108 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 122
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x4e
***************
*** 1034,1039 ****
--- 1115,1121 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 123
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x4f
***************
*** 1046,1051 ****
--- 1128,1134 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 124
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x50
***************
*** 1054,1063 ****
  
      move-result-object v1
  
!     const-string v2, "video/*"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x51
--- 1137,1147 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "video/*"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 125
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x51
***************
*** 1070,1075 ****
--- 1154,1160 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 126
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x52
***************
*** 1082,1087 ****
--- 1167,1173 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 127
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x53
***************
*** 1094,1099 ****
--- 1180,1186 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 128
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x54
***************
*** 1106,1111 ****
--- 1193,1199 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 130
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x201
***************
*** 1118,1123 ****
--- 1206,1212 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 131
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x202
***************
*** 1130,1135 ****
--- 1219,1225 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 132
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x203
***************
*** 1142,1147 ****
--- 1232,1238 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 133
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x204
***************
*** 1154,1159 ****
--- 1245,1251 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 134
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x205
***************
*** 1166,1171 ****
--- 1258,1264 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 135
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x206
***************
*** 1178,1183 ****
--- 1271,1277 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 136
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x207
***************
*** 1190,1195 ****
--- 1284,1290 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 137
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x208
***************
*** 1198,1207 ****
  
      move-result-object v1
  
!     const-string v2, "x-wap.multipart/vnd.uplanet.header-set"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x209
--- 1293,1303 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "x-wap.multipart/vnd.uplanet.header-set"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 138
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x209
***************
*** 1214,1219 ****
--- 1310,1316 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 139
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x20a
***************
*** 1226,1231 ****
--- 1323,1329 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 140
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x20b
***************
*** 1238,1243 ****
--- 1336,1342 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 141
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x20c
***************
*** 1250,1255 ****
--- 1349,1355 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 142
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x300
***************
*** 1262,1267 ****
--- 1362,1368 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 143
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x301
***************
*** 1274,1279 ****
--- 1375,1381 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 144
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x302
***************
*** 1286,1291 ****
--- 1388,1394 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 145
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x303
***************
*** 1298,1303 ****
--- 1401,1407 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 146
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x304
***************
*** 1306,1315 ****
  
      move-result-object v1
  
!     const-string v2, "text/vnd.wap.emn+xml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x305
--- 1410,1420 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/vnd.wap.emn+xml"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 147
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x305
***************
*** 1318,1327 ****
  
      move-result-object v1
  
!     const-string v2, "text/calendar"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x306
--- 1423,1433 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/calendar"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 148
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x306
***************
*** 1334,1339 ****
--- 1440,1446 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 149
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x307
***************
*** 1346,1351 ****
--- 1453,1459 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 150
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x308
***************
*** 1358,1363 ****
--- 1466,1472 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 151
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x309
***************
*** 1366,1375 ****
  
      move-result-object v1
  
!     const-string v2, "text/directory;profile=vCard"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x30a
--- 1475,1485 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "text/directory;profile=vCard"
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 152
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x30a
***************
*** 1382,1387 ****
--- 1492,1498 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 153
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x30b
***************
*** 1394,1399 ****
--- 1505,1511 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 154
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x30c
***************
*** 1406,1411 ****
--- 1518,1524 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 155
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x30d
***************
*** 1418,1423 ****
--- 1531,1537 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 156
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x30e
***************
*** 1430,1435 ****
--- 1544,1550 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 157
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x30f
***************
*** 1442,1447 ****
--- 1557,1563 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 158
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x310
***************
*** 1454,1459 ****
--- 1570,1576 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 159
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x311
***************
*** 1466,1471 ****
--- 1583,1589 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 160
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x312
***************
*** 1478,1483 ****
--- 1596,1602 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 161
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x313
***************
*** 1490,1495 ****
--- 1609,1615 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 162
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x314
***************
*** 1502,1507 ****
--- 1622,1628 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 163
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x315
***************
*** 1514,1519 ****
--- 1635,1641 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 164
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x316
***************
*** 1526,1531 ****
--- 1648,1654 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 165
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x317
***************
*** 1538,1543 ****
--- 1661,1667 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 166
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x318
***************
*** 1550,1555 ****
--- 1674,1680 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 167
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x319
***************
*** 1562,1567 ****
--- 1687,1693 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 168
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x31a
***************
*** 1574,1579 ****
--- 1700,1706 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 169
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
      const/16 v1, 0x31b
***************
*** 1586,1591 ****
--- 1713,1719 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 171
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      invoke-static {v3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 1596,1601 ****
--- 1724,1730 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 172
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      invoke-static {v4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 1606,1611 ****
--- 1735,1741 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 173
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      invoke-static {v5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 1616,1621 ****
--- 1746,1752 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 174
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      invoke-static {v6}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 1626,1631 ****
--- 1757,1763 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 175
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      invoke-static {v7}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 1636,1641 ****
--- 1768,1774 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 176
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x8
***************
*** 1648,1653 ****
--- 1781,1787 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 177
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x9
***************
*** 1660,1665 ****
--- 1794,1800 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 178
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0xe
***************
*** 1672,1677 ****
--- 1807,1813 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 179
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x10
***************
*** 1684,1689 ****
--- 1820,1826 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 180
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x11
***************
*** 1696,1701 ****
--- 1833,1839 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 181
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x12
***************
*** 1708,1713 ****
--- 1846,1852 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 182
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x13
***************
*** 1720,1725 ****
--- 1859,1865 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 183
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x14
***************
*** 1732,1737 ****
--- 1872,1878 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 184
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x15
***************
*** 1744,1749 ****
--- 1885,1891 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 185
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x16
***************
*** 1756,1761 ****
--- 1898,1904 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 186
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x17
***************
*** 1768,1773 ****
--- 1911,1917 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 187
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x18
***************
*** 1780,1785 ****
--- 1924,1930 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 188
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x19
***************
*** 1792,1797 ****
--- 1937,1943 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 189
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x1a
***************
*** 1804,1809 ****
--- 1950,1956 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 190
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x1b
***************
*** 1816,1821 ****
--- 1963,1969 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 191
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x1c
***************
*** 1828,1833 ****
--- 1976,1982 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 192
      sget-object v0, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
      const/16 v1, 0x1d
***************
*** 1840,1845 ****
--- 1989,1995 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 193
      return-void
  .end method
  
***************
*** 1848,1857 ****
--- 1998,2010 ----
      .parameter "pdu"
  
      .prologue
+     .line 206
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 207
      iput-object p1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
+     .line 208
      return-void
  .end method
  
***************
*** 1862,1875 ****
--- 2015,2031 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 541
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      aget-byte v1, v1, p1
  
      if-nez v1, :cond_0
  
+     .line 542
      iput v0, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 545
      :goto_0
      return v0
  
***************
*** 1883,1896 ****
--- 2039,2055 ----
      .locals 3
  
      .prologue
+     .line 555
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
      if-nez v1, :cond_0
  
+     .line 556
      iget-wide v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
      long-to-int v0, v1
  
+     .line 557
      .local v0, binaryContentType:I
      sget-object v1, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_MIME_TYPES:Ljava/util/HashMap;
  
***************
*** 1906,1915 ****
--- 2065,2076 ----
  
      iput-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 561
      .end local v0           #binaryContentType:I
      :goto_0
      return-void
  
+     .line 559
      :cond_0
      const-wide/16 v1, -0x1
  
***************
*** 1929,1949 ****
--- 2090,2116 ----
  
      const/4 v6, 0x0
  
+     .line 468
      const/4 v3, 0x0
  
+     .line 470
      .local v3, totalRead:I
      if-lez p2, :cond_8
  
+     .line 471
      iget-object v8, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      aget-byte v1, v8, p1
  
+     .line 472
      .local v1, nextByte:B
      const/4 v4, 0x0
  
+     .line 473
      .local v4, value:Ljava/lang/String;
      const/4 v2, 0x0
  
+     .line 474
      .local v2, param:Ljava/lang/String;
      and-int/lit16 v8, v1, 0x80
  
***************
*** 1953,1966 ****
--- 2120,2137 ----
  
      if-le v1, v8, :cond_3
  
+     .line 475
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeTokenText(I)Z
  
+     .line 476
      iget-object v2, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 477
      iget v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v3, v6
  
+     .line 503
      :cond_0
      add-int v6, p1, v3
  
***************
*** 1970,1987 ****
--- 2141,2162 ----
  
      if-eqz v6, :cond_5
  
+     .line 504
      iget v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v3, v6
  
+     .line 505
      const/4 v4, 0x0
  
+     .line 523
      :cond_1
      :goto_0
      iget-object v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->contentParameters:Ljava/util/HashMap;
  
      invoke-virtual {v6, v2, v4}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 524
      add-int v6, p1, v3
  
      sub-int v7, p2, v3
***************
*** 1992,1997 ****
--- 2167,2173 ----
  
      move-result v6
  
+     .line 529
      .end local v1           #nextByte:B
      .end local v2           #param:Ljava/lang/String;
      .end local v4           #value:Ljava/lang/String;
***************
*** 1999,2004 ****
--- 2175,2181 ----
      :goto_1
      return v6
  
+     .line 479
      .restart local v1       #nextByte:B
      .restart local v2       #param:Ljava/lang/String;
      .restart local v4       #value:Ljava/lang/String;
***************
*** 2009,2022 ****
--- 2186,2202 ----
  
      if-eqz v8, :cond_2
  
+     .line 480
      iget v8, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v3, v8
  
+     .line 481
      iget-wide v8, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
      long-to-int v5, v8
  
+     .line 482
      .local v5, wellKnownParameterValue:I
      sget-object v8, Lcom/android/internal/telephony/WspTypeDecoder;->WELL_KNOWN_PARAMETERS:Ljava/util/HashMap;
  
***************
*** 2031,2044 ****
      .end local v2           #param:Ljava/lang/String;
      check-cast v2, Ljava/lang/String;
  
      .restart local v2       #param:Ljava/lang/String;
      if-nez v2, :cond_4
  
      new-instance v8, Ljava/lang/StringBuilder;
  
      invoke-direct {v8}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v9, "unassigned/0x"
  
      invoke-virtual {v8, v9}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2211,2226 ----
      .end local v2           #param:Ljava/lang/String;
      check-cast v2, Ljava/lang/String;
  
+     .line 483
      .restart local v2       #param:Ljava/lang/String;
      if-nez v2, :cond_4
  
+     .line 484
      new-instance v8, Ljava/lang/StringBuilder;
  
      invoke-direct {v8}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v9, "unassigned/0x"
  
      invoke-virtual {v8, v9}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 2058,2066 ****
--- 2240,2250 ----
  
      move-result-object v2
  
+     .line 487
      :cond_4
      if-nez v5, :cond_0
  
+     .line 488
      add-int v7, p1, v3
  
      invoke-virtual {p0, v7}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeUintvarInteger(I)Z
***************
*** 2069,2088 ****
--- 2253,2276 ----
  
      if-eqz v7, :cond_2
  
+     .line 489
      iget v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v3, v6
  
+     .line 490
      iget-wide v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
      invoke-static {v6, v7}, Ljava/lang/String;->valueOf(J)Ljava/lang/String;
  
      move-result-object v4
  
+     .line 491
      iget-object v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->contentParameters:Ljava/util/HashMap;
  
      invoke-virtual {v6, v2, v4}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 492
      add-int v6, p1, v3
  
      sub-int v7, p2, v3
***************
*** 2095,2100 ****
--- 2283,2289 ----
  
      goto :goto_1
  
+     .line 506
      .end local v5           #wellKnownParameterValue:I
      :cond_5
      add-int v6, p1, v3
***************
*** 2105,2125 ****
--- 2294,2319 ----
  
      if-eqz v6, :cond_7
  
+     .line 507
      iget v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v3, v6
  
+     .line 508
      iget-wide v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
      long-to-int v0, v6
  
+     .line 509
      .local v0, intValue:I
      if-nez v0, :cond_6
  
+     .line 510
      const-string v4, ""
  
      goto :goto_0
  
+     .line 512
      :cond_6
      invoke-static {v0}, Ljava/lang/String;->valueOf(I)Ljava/lang/String;
  
***************
*** 2127,2144 ****
--- 2321,2342 ----
  
      goto :goto_0
  
+     .line 515
      .end local v0           #intValue:I
      :cond_7
      add-int v6, p1, v3
  
      invoke-virtual {p0, v6}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeTokenText(I)Z
  
+     .line 516
      iget v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v3, v6
  
+     .line 517
      iget-object v4, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 518
      const-string v6, "\""
  
      invoke-virtual {v4, v6}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
***************
*** 2147,2158 ****
--- 2345,2358 ----
  
      if-eqz v6, :cond_1
  
+     .line 520
      invoke-virtual {v4, v7}, Ljava/lang/String;->substring(I)Ljava/lang/String;
  
      move-result-object v4
  
      goto/16 :goto_0
  
+     .line 528
      .end local v1           #nextByte:B
      .end local v2           #param:Ljava/lang/String;
      .end local v4           #value:Ljava/lang/String;
***************
*** 2161,2166 ****
--- 2361,2367 ----
  
      move v6, v7
  
+     .line 529
      goto/16 :goto_1
  .end method
  
***************
*** 2173,2188 ****
--- 2374,2392 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 395
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeShortInteger(I)Z
  
      move-result v1
  
      if-ne v1, v0, :cond_0
  
+     .line 396
      const/4 v1, 0x0
  
      iput-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 399
      :goto_0
      return v0
  
***************
*** 2199,2204 ****
--- 2403,2409 ----
      .parameter "startIndex"
  
      .prologue
+     .line 573
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeIntegerValue(I)Z
  
      move-result v0
***************
*** 2211,2216 ****
--- 2416,2422 ----
      .parameter "startIndex"
  
      .prologue
+     .line 586
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeTextString(I)Z
  
      move-result v0
***************
*** 2227,2238 ****
--- 2433,2446 ----
  
      const/4 v8, 0x0
  
+     .line 417
      new-instance v9, Ljava/util/HashMap;
  
      invoke-direct {v9}, Ljava/util/HashMap;-><init>()V
  
      iput-object v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->contentParameters:Ljava/util/HashMap;
  
+     .line 420
      :try_start_0
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeValueLength(I)Z
  
***************
*** 2240,2269 ****
--- 2448,2484 ----
  
      if-nez v9, :cond_1
  
+     .line 421
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeConstrainedEncoding(I)Z
  
      move-result v1
  
+     .line 422
      .local v1, found:Z
      if-eqz v1, :cond_0
  
+     .line 423
      invoke-direct {p0}, Lcom/android/internal/telephony/WspTypeDecoder;->expandWellKnownMimeType()V
  
+     .line 463
      .end local v1           #found:Z
      :cond_0
      :goto_0
      return v1
  
+     .line 427
      :cond_1
      iget-wide v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
      long-to-int v2, v9
  
+     .line 428
      .local v2, headersLength:I
      invoke-virtual {p0}, Lcom/android/internal/telephony/WspTypeDecoder;->getDecodedDataLength()I
  
      move-result v3
  
+     .line 429
      .local v3, mediaPrefixLength:I
      add-int v9, p1, v3
  
***************
*** 2273,2298 ****
--- 2488,2520 ----
  
      if-ne v9, v1, :cond_3
  
+     .line 430
      iget v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v9, v3
  
      iput v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 431
      iget v5, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 432
      .local v5, readLength:I
      const/4 v9, 0x0
  
      iput-object v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 433
      invoke-direct {p0}, Lcom/android/internal/telephony/WspTypeDecoder;->expandWellKnownMimeType()V
  
+     .line 434
      iget-wide v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 435
      .local v6, wellKnownValue:J
      iget-object v4, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 436
      .local v4, mimeType:Ljava/lang/String;
      iget v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
***************
*** 2312,2329 ****
--- 2534,2555 ----
  
      if-eqz v9, :cond_2
  
+     .line 438
      iget v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v9, v5
  
      iput v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 439
      iput-wide v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 440
      iput-object v4, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
      goto :goto_0
  
+     .line 459
      .end local v2           #headersLength:I
      .end local v3           #mediaPrefixLength:I
      .end local v4           #mimeType:Ljava/lang/String;
***************
*** 2335,2340 ****
--- 2561,2567 ----
      .local v0, e:Ljava/lang/ArrayIndexOutOfBoundsException;
      move v1, v8
  
+     .line 461
      goto :goto_0
  
      .end local v0           #e:Ljava/lang/ArrayIndexOutOfBoundsException;
***************
*** 2346,2353 ****
--- 2573,2582 ----
      :cond_2
      move v1, v8
  
+     .line 443
      goto :goto_0
  
+     .line 445
      .end local v4           #mimeType:Ljava/lang/String;
      .end local v5           #readLength:I
      .end local v6           #wellKnownValue:J
***************
*** 2360,2381 ****
--- 2589,2616 ----
  
      if-ne v9, v1, :cond_4
  
+     .line 446
      iget v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v9, v3
  
      iput v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 447
      iget v5, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 448
      .restart local v5       #readLength:I
      invoke-direct {p0}, Lcom/android/internal/telephony/WspTypeDecoder;->expandWellKnownMimeType()V
  
+     .line 449
      iget-wide v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 450
      .restart local v6       #wellKnownValue:J
      iget-object v4, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 451
      .restart local v4       #mimeType:Ljava/lang/String;
      iget v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
***************
*** 2395,2408 ****
--- 2630,2646 ----
  
      if-eqz v9, :cond_4
  
+     .line 453
      iget v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/2addr v9, v5
  
      iput v9, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 454
      iput-wide v6, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 455
      iput-object v4, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
      :try_end_0
      .catch Ljava/lang/ArrayIndexOutOfBoundsException; {:try_start_0 .. :try_end_0} :catch_0
***************
*** 2415,2420 ****
--- 2653,2659 ----
      :cond_4
      move v1, v8
  
+     .line 463
      goto :goto_0
  .end method
  
***************
*** 2425,2448 ****
--- 2664,2693 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 369
      move v0, p1
  
+     .line 370
      .local v0, index:I
      iput v2, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 371
      const/4 v3, 0x0
  
      iput-object v3, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 372
      iget-object v3, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      array-length v1, v3
  
+     .line 373
      .local v1, length:I
      if-ge v0, v1, :cond_0
  
      const/4 v2, 0x1
  
+     .line 375
      .local v2, rtrn:Z
      :cond_0
      :goto_0
***************
*** 2454,2463 ****
--- 2699,2710 ----
  
      if-eqz v3, :cond_1
  
+     .line 376
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 379
      :cond_1
      sub-int v3, v0, p1
  
***************
*** 2465,2470 ****
--- 2712,2718 ----
  
      iput v3, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 380
      new-instance v3, Ljava/lang/String;
  
      iget-object v4, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
***************
*** 2477,2482 ****
--- 2725,2731 ----
  
      iput-object v3, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 382
      return v2
  .end method
  
***************
*** 2487,2498 ****
--- 2736,2749 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 304
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeShortInteger(I)Z
  
      move-result v1
  
      if-ne v1, v0, :cond_0
  
+     .line 307
      :goto_0
      return v0
  
***************
*** 2509,2541 ****
--- 2760,2799 ----
      .parameter "startIndex"
  
      .prologue
+     .line 281
      iget-object v2, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      aget-byte v2, v2, p1
  
      and-int/lit16 v1, v2, 0xff
  
+     .line 283
      .local v1, lengthMultiOctet:I
      const/16 v2, 0x1e
  
      if-le v1, v2, :cond_0
  
+     .line 284
      const/4 v2, 0x0
  
+     .line 291
      :goto_0
      return v2
  
+     .line 286
      :cond_0
      const-wide/16 v2, 0x0
  
      iput-wide v2, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 287
      const/4 v0, 0x1
  
      .local v0, i:I
      :goto_1
      if-gt v0, v1, :cond_1
  
+     .line 288
      iget-wide v2, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
      const/16 v4, 0x8
***************
*** 2556,2570 ****
--- 2814,2831 ----
  
      iput-wide v2, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 287
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 290
      :cond_1
      add-int/lit8 v2, v1, 0x1
  
      iput v2, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 291
      const/4 v2, 0x1
  
      goto :goto_0
***************
*** 2577,2582 ****
--- 2838,2844 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 263
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      aget-byte v1, v1, p1
***************
*** 2585,2595 ****
--- 2847,2860 ----
  
      if-nez v1, :cond_0
  
+     .line 264
      const/4 v0, 0x0
  
+     .line 268
      :goto_0
      return v0
  
+     .line 266
      :cond_0
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
***************
*** 2601,2606 ****
--- 2866,2872 ----
  
      iput-wide v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 267
      iput v0, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      goto :goto_0
***************
*** 2611,2618 ****
--- 2877,2886 ----
      .parameter "startIndex"
  
      .prologue
+     .line 220
      move v0, p1
  
+     .line 221
      .local v0, index:I
      :goto_0
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
***************
*** 2621,2630 ****
--- 2889,2900 ----
  
      if-eqz v1, :cond_0
  
+     .line 222
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 224
      :cond_0
      sub-int v1, v0, p1
  
***************
*** 2632,2637 ****
--- 2902,2908 ----
  
      iput v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 225
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      aget-byte v1, v1, p1
***************
*** 2640,2645 ****
--- 2911,2917 ----
  
      if-ne v1, v2, :cond_1
  
+     .line 226
      new-instance v1, Ljava/lang/String;
  
      iget-object v2, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
***************
*** 2654,2664 ****
--- 2926,2938 ----
  
      iput-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 230
      :goto_1
      const/4 v1, 0x1
  
      return v1
  
+     .line 228
      :cond_1
      new-instance v1, Ljava/lang/String;
  
***************
*** 2680,2687 ****
--- 2954,2963 ----
      .parameter "startIndex"
  
      .prologue
+     .line 243
      move v0, p1
  
+     .line 244
      .local v0, index:I
      :goto_0
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
***************
*** 2690,2699 ****
--- 2966,2977 ----
  
      if-eqz v1, :cond_0
  
+     .line 245
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 247
      :cond_0
      sub-int v1, v0, p1
  
***************
*** 2701,2706 ****
--- 2979,2985 ----
  
      iput v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 248
      new-instance v1, Ljava/lang/String;
  
      iget-object v2, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
***************
*** 2713,2718 ****
--- 2992,2998 ----
  
      iput-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 250
      const/4 v1, 0x1
  
      return v1
***************
*** 2725,2737 ****
--- 3005,3020 ----
      .prologue
      const/4 v5, 0x7
  
+     .line 320
      move v0, p1
  
+     .line 322
      .local v0, index:I
      const-wide/16 v1, 0x0
  
      iput-wide v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 323
      :goto_0
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
***************
*** 2741,2757 ****
--- 3024,3044 ----
  
      if-eqz v1, :cond_1
  
+     .line 324
      sub-int v1, v0, p1
  
      const/4 v2, 0x4
  
      if-lt v1, v2, :cond_0
  
+     .line 325
      const/4 v1, 0x0
  
+     .line 332
      :goto_1
      return v1
  
+     .line 327
      :cond_0
      iget-wide v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
***************
*** 2769,2778 ****
--- 3056,3067 ----
  
      iput-wide v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 328
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 330
      :cond_1
      iget-wide v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
***************
*** 2790,2801 ****
--- 3079,3092 ----
  
      iput-wide v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 331
      sub-int v1, v0, p1
  
      add-int/lit8 v1, v1, 0x1
  
      iput v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
+     .line 332
      const/4 v1, 0x1
  
      goto :goto_1
***************
*** 2810,2815 ****
--- 3101,3107 ----
  
      const/4 v0, 0x1
  
+     .line 345
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      aget-byte v1, v1, p1
***************
*** 2818,2828 ****
--- 3110,3123 ----
  
      if-le v1, v2, :cond_0
  
+     .line 346
      const/4 v0, 0x0
  
+     .line 355
      :goto_0
      return v0
  
+     .line 348
      :cond_0
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
***************
*** 2830,2835 ****
--- 3125,3131 ----
  
      if-ge v1, v2, :cond_1
  
+     .line 349
      iget-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      aget-byte v1, v1, p1
***************
*** 2838,2852 ****
--- 3134,3151 ----
  
      iput-wide v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 350
      iput v0, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      goto :goto_0
  
+     .line 352
      :cond_1
      add-int/lit8 v1, p1, 0x1
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeUintvarInteger(I)Z
  
+     .line 353
      iget v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      add-int/lit8 v1, v1, 0x1
***************
*** 2863,2878 ****
--- 3162,3180 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 600
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeIntegerValue(I)Z
  
      move-result v1
  
      if-ne v1, v0, :cond_0
  
+     .line 601
      const/4 v1, 0x0
  
      iput-object v1, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
+     .line 604
      :goto_0
      return v0
  
***************
*** 2889,2894 ****
--- 3191,3197 ----
      .parameter "startIndex"
  
      .prologue
+     .line 681
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeTextString(I)Z
  
      move-result v0
***************
*** 2901,2906 ****
--- 3204,3210 ----
      .parameter "startIndex"
  
      .prologue
+     .line 694
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeTextString(I)Z
  
      move-result v0
***************
*** 2922,2927 ****
--- 3226,3232 ----
      .end annotation
  
      .prologue
+     .line 729
      iget-object v0, p0, Lcom/android/internal/telephony/WspTypeDecoder;->contentParameters:Ljava/util/HashMap;
  
      return-object v0
***************
*** 2931,2936 ****
--- 3236,3242 ----
      .locals 1
  
      .prologue
+     .line 701
      iget v0, p0, Lcom/android/internal/telephony/WspTypeDecoder;->dataLength:I
  
      return v0
***************
*** 2940,2945 ****
--- 3246,3252 ----
      .locals 2
  
      .prologue
+     .line 708
      iget-wide v0, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
      return-wide v0
***************
*** 2949,2954 ****
--- 3256,3262 ----
      .locals 1
  
      .prologue
+     .line 715
      iget-object v0, p0, Lcom/android/internal/telephony/WspTypeDecoder;->stringValue:Ljava/lang/String;
  
      return-object v0
***************
*** 2964,2977 ****
--- 3272,3288 ----
  
      const/4 v4, 0x0
  
+     .line 617
      move v2, p1
  
+     .line 620
      .local v2, index:I
      move v2, p1
  
      :goto_0
      if-gt v2, p2, :cond_0
  
+     .line 625
      :try_start_0
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeIntegerValue(I)Z
  
***************
*** 2979,3008 ****
--- 3290,3325 ----
  
      if-eqz v5, :cond_1
  
+     .line 626
      invoke-virtual {p0}, Lcom/android/internal/telephony/WspTypeDecoder;->getValue32()J
  
      move-result-wide v5
  
      long-to-int v1, v5
  
+     .line 628
      .local v1, fieldValue:I
      const/16 v5, 0x2f
  
      if-ne v1, v5, :cond_2
  
+     .line 629
      add-int/lit8 v5, v2, 0x1
  
      int-to-long v5, v5
  
      iput-wide v5, p0, Lcom/android/internal/telephony/WspTypeDecoder;->unsigned32bit:J
  
+     .line 630
      const/4 v4, 0x1
  
+     .line 668
      .end local v1           #fieldValue:I
      :cond_0
      :goto_1
      return v4
  
+     .line 633
      :cond_1
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeTextString(I)Z
  
***************
*** 3010,3015 ****
--- 3327,3333 ----
  
      if-eqz v5, :cond_0
  
+     .line 635
      :cond_2
      invoke-virtual {p0}, Lcom/android/internal/telephony/WspTypeDecoder;->getDecodedDataLength()I
  
***************
*** 3017,3028 ****
--- 3335,3349 ----
  
      add-int/2addr v2, v5
  
+     .line 636
      if-gt v2, p2, :cond_0
  
+     .line 649
      iget-object v5, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      aget-byte v3, v5, v2
  
+     .line 650
      .local v3, val:B
      if-ltz v3, :cond_3
  
***************
*** 3030,3035 ****
--- 3351,3357 ----
  
      if-gt v3, v5, :cond_3
  
+     .line 651
      iget-object v5, p0, Lcom/android/internal/telephony/WspTypeDecoder;->wspData:[B
  
      aget-byte v5, v5, v2
***************
*** 3040,3060 ****
--- 3362,3387 ----
  
      goto :goto_0
  
+     .line 652
      :cond_3
      if-ne v3, v7, :cond_4
  
+     .line 653
      add-int/lit8 v5, v2, 0x1
  
      if-ge v5, p2, :cond_0
  
+     .line 654
      add-int/lit8 v2, v2, 0x1
  
+     .line 655
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeUintvarInteger(I)Z
  
      move-result v5
  
      if-eqz v5, :cond_0
  
+     .line 656
      invoke-virtual {p0}, Lcom/android/internal/telephony/WspTypeDecoder;->getDecodedDataLength()I
  
      move-result v5
***************
*** 3063,3068 ****
--- 3390,3396 ----
  
      goto :goto_0
  
+     .line 657
      :cond_4
      if-ge v7, v3, :cond_5
  
***************
*** 3070,3081 ****
--- 3398,3411 ----
  
      if-gt v3, v5, :cond_5
  
+     .line 658
      invoke-virtual {p0, v2}, Lcom/android/internal/telephony/WspTypeDecoder;->decodeTextString(I)Z
  
      move-result v5
  
      if-eqz v5, :cond_0
  
+     .line 659
      invoke-virtual {p0}, Lcom/android/internal/telephony/WspTypeDecoder;->getDecodedDataLength()I
      :try_end_0
      .catch Ljava/lang/ArrayIndexOutOfBoundsException; {:try_start_0 .. :try_end_0} :catch_0
***************
*** 3086,3100 ****
--- 3416,3433 ----
  
      goto :goto_0
  
+     .line 661
      :cond_5
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 664
      .end local v3           #val:B
      :catch_0
      move-exception v0
  
+     .line 666
      .local v0, e:Ljava/lang/ArrayIndexOutOfBoundsException;
      goto :goto_1
  .end method
