*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/policy/IFaceLockInterface$Stub$Proxy.smali	2012-03-23 19:37:30.755335905 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/policy/IFaceLockInterface$Stub$Proxy.smali.original	2012-03-23 19:34:46.043332386 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 95
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 96
      iput-object p1, p0, Lcom/android/internal/policy/IFaceLockInterface$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 97
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 100
      iget-object v0, p0, Lcom/android/internal/policy/IFaceLockInterface$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 49,54 ****
--- 53,59 ----
      .locals 1
  
      .prologue
+     .line 104
      const-string v0, "com.android.internal.policy.IFaceLockInterface"
  
      return-object v0
***************
*** 64,84 ****
--- 69,93 ----
      .end annotation
  
      .prologue
+     .line 141
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 142
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 144
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.policy.IFaceLockInterface"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 145
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Lcom/android/internal/policy/IFaceLockCallback;->asBinder()Landroid/os/IBinder;
***************
*** 88,93 ****
--- 97,103 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 146
      iget-object v2, p0, Lcom/android/internal/policy/IFaceLockInterface$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x3
***************
*** 96,121 ****
--- 106,138 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 147
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 150
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 151
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 153
      return-void
  
+     .line 145
      :cond_0
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 150
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 151
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 135,165 ****
--- 152,191 ----
      .end annotation
  
      .prologue
+     .line 108
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 109
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 111
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.policy.IFaceLockInterface"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 112
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 113
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 114
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 115
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 116
      invoke-virtual {v0, p5}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 117
      iget-object v2, p0, Lcom/android/internal/policy/IFaceLockInterface$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x1
***************
*** 168,188 ****
--- 194,220 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 118
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 121
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 122
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 124
      return-void
  
+     .line 121
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 122
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 197,217 ****
--- 229,253 ----
      .end annotation
  
      .prologue
+     .line 127
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 128
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 130
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.policy.IFaceLockInterface"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 131
      iget-object v2, p0, Lcom/android/internal/policy/IFaceLockInterface$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x2
***************
*** 220,240 ****
--- 256,282 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 132
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 135
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 136
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 138
      return-void
  
+     .line 135
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 136
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 250,270 ****
--- 292,316 ----
      .end annotation
  
      .prologue
+     .line 156
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 157
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 159
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "com.android.internal.policy.IFaceLockInterface"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 160
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Lcom/android/internal/policy/IFaceLockCallback;->asBinder()Landroid/os/IBinder;
***************
*** 274,279 ****
--- 320,326 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 161
      iget-object v2, p0, Lcom/android/internal/policy/IFaceLockInterface$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x4
***************
*** 282,307 ****
--- 329,361 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 162
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 165
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 166
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 168
      return-void
  
+     .line 160
      :cond_0
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 165
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 166
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
