*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AbsListView$PositionScroller.smali	2012-03-23 19:35:50.488507543 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AbsListView$PositionScroller.smali.original	2012-03-23 19:34:44.864310886 -0500
***************
*** 55,64 ****
--- 55,66 ----
      .parameter
  
      .prologue
+     .line 3901
      iput-object p1, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 3902
      #getter for: Landroid/widget/AbsListView;->mContext:Landroid/content/Context;
      invoke-static {p1}, Landroid/widget/AbsListView;->access$3000(Landroid/widget/AbsListView;)Landroid/content/Context;
  
***************
*** 74,79 ****
--- 76,82 ----
  
      iput v0, p0, Landroid/widget/AbsListView$PositionScroller;->mExtraScroll:I
  
+     .line 3903
      return-void
  .end method
  
***************
*** 83,88 ****
--- 86,92 ----
      .locals 31
  
      .prologue
+     .line 4039
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 117,126 ****
--- 121,132 ----
  
      if-eq v0, v1, :cond_1
  
+     .line 4210
      :cond_0
      :goto_0
      return-void
  
+     .line 4043
      :cond_1
      move-object/from16 v0, p0
  
***************
*** 132,137 ****
--- 138,144 ----
  
      move-result v16
  
+     .line 4044
      .local v16, listHeight:I
      move-object/from16 v0, p0
  
***************
*** 143,148 ****
--- 150,156 ----
  
      iget v6, v0, Landroid/widget/AbsListView;->mFirstPosition:I
  
+     .line 4046
      .local v6, firstPos:I
      move-object/from16 v0, p0
  
***************
*** 154,159 ****
--- 162,168 ----
  
      goto :goto_0
  
+     .line 4048
      :pswitch_0
      move-object/from16 v0, p0
  
***************
*** 167,178 ****
--- 176,190 ----
  
      add-int/lit8 v13, v27, -0x1
  
+     .line 4049
      .local v13, lastViewIndex:I
      add-int v9, v6, v13
  
+     .line 4051
      .local v9, lastPos:I
      if-ltz v13, :cond_0
  
+     .line 4055
      move-object/from16 v0, p0
  
      iget v0, v0, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
***************
*** 183,188 ****
--- 195,201 ----
  
      if-ne v9, v0, :cond_2
  
+     .line 4057
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 197,202 ****
--- 210,216 ----
  
      goto :goto_0
  
+     .line 4061
      :cond_2
      move-object/from16 v0, p0
  
***************
*** 210,228 ****
--- 224,246 ----
  
      move-result-object v10
  
+     .line 4062
      .local v10, lastView:Landroid/view/View;
      invoke-virtual {v10}, Landroid/view/View;->getHeight()I
  
      move-result v12
  
+     .line 4063
      .local v12, lastViewHeight:I
      invoke-virtual {v10}, Landroid/view/View;->getTop()I
  
      move-result v15
  
+     .line 4064
      .local v15, lastViewTop:I
      sub-int v14, v16, v15
  
+     .line 4065
      .local v14, lastViewPixelsShowing:I
      move-object/from16 v0, p0
  
***************
*** 246,251 ****
--- 264,270 ----
  
      iget v5, v0, Landroid/widget/AbsListView$PositionScroller;->mExtraScroll:I
  
+     .line 4067
      .local v5, extraScroll:I
      :goto_1
      move-object/from16 v0, p0
***************
*** 266,275 ****
--- 285,296 ----
  
      invoke-virtual/range {v27 .. v29}, Landroid/widget/AbsListView;->smoothScrollBy(II)V
  
+     .line 4070
      move-object/from16 v0, p0
  
      iput v9, v0, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
  
+     .line 4071
      move-object/from16 v0, p0
  
      iget v0, v0, Landroid/widget/AbsListView$PositionScroller;->mTargetPos:I
***************
*** 280,285 ****
--- 301,307 ----
  
      if-ge v9, v0, :cond_0
  
+     .line 4072
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 294,299 ****
--- 316,322 ----
  
      goto/16 :goto_0
  
+     .line 4065
      .end local v5           #extraScroll:I
      :cond_3
      move-object/from16 v0, p0
***************
*** 314,319 ****
--- 337,343 ----
  
      goto :goto_1
  
+     .line 4078
      .end local v9           #lastPos:I
      .end local v10           #lastView:Landroid/view/View;
      .end local v12           #lastViewHeight:I
***************
*** 323,328 ****
--- 347,353 ----
      :pswitch_1
      const/16 v21, 0x1
  
+     .line 4079
      .local v21, nextViewIndex:I
      move-object/from16 v0, p0
  
***************
*** 334,339 ****
--- 359,365 ----
  
      move-result v3
  
+     .line 4081
      .local v3, childCount:I
      move-object/from16 v0, p0
  
***************
*** 371,378 ****
--- 397,406 ----
  
      if-ge v0, v1, :cond_0
  
+     .line 4085
      add-int/lit8 v18, v6, 0x1
  
+     .line 4087
      .local v18, nextPos:I
      move-object/from16 v0, p0
  
***************
*** 386,391 ****
--- 414,420 ----
  
      if-ne v0, v1, :cond_4
  
+     .line 4089
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 400,405 ****
--- 429,435 ----
  
      goto/16 :goto_0
  
+     .line 4093
      :cond_4
      move-object/from16 v0, p0
  
***************
*** 413,433 ****
--- 443,467 ----
  
      move-result-object v19
  
+     .line 4094
      .local v19, nextView:Landroid/view/View;
      invoke-virtual/range {v19 .. v19}, Landroid/view/View;->getHeight()I
  
      move-result v20
  
+     .line 4095
      .local v20, nextViewHeight:I
      invoke-virtual/range {v19 .. v19}, Landroid/view/View;->getTop()I
  
      move-result v22
  
+     .line 4096
      .local v22, nextViewTop:I
      move-object/from16 v0, p0
  
      iget v5, v0, Landroid/widget/AbsListView$PositionScroller;->mExtraScroll:I
  
+     .line 4097
      .restart local v5       #extraScroll:I
      move-object/from16 v0, p0
  
***************
*** 441,446 ****
--- 475,481 ----
  
      if-ge v0, v1, :cond_5
  
+     .line 4098
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 465,476 ****
--- 500,513 ----
  
      invoke-virtual/range {v27 .. v29}, Landroid/widget/AbsListView;->smoothScrollBy(II)V
  
+     .line 4101
      move/from16 v0, v18
  
      move-object/from16 v1, p0
  
      iput v0, v1, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
  
+     .line 4103
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 485,495 ****
--- 522,534 ----
  
      goto/16 :goto_0
  
+     .line 4105
      :cond_5
      move/from16 v0, v22
  
      if-le v0, v5, :cond_0
  
+     .line 4106
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 508,513 ****
--- 547,553 ----
  
      goto/16 :goto_0
  
+     .line 4113
      .end local v3           #childCount:I
      .end local v5           #extraScroll:I
      .end local v18           #nextPos:I
***************
*** 526,531 ****
--- 566,572 ----
  
      if-ne v6, v0, :cond_6
  
+     .line 4115
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 540,545 ****
--- 581,587 ----
  
      goto/16 :goto_0
  
+     .line 4119
      :cond_6
      move-object/from16 v0, p0
  
***************
*** 553,565 ****
--- 595,610 ----
  
      move-result-object v7
  
+     .line 4120
      .local v7, firstView:Landroid/view/View;
      if-eqz v7, :cond_0
  
+     .line 4123
      invoke-virtual {v7}, Landroid/view/View;->getTop()I
  
      move-result v8
  
+     .line 4124
      .local v8, firstViewTop:I
      if-lez v6, :cond_7
  
***************
*** 567,572 ****
--- 612,618 ----
  
      iget v5, v0, Landroid/widget/AbsListView$PositionScroller;->mExtraScroll:I
  
+     .line 4126
      .restart local v5       #extraScroll:I
      :goto_2
      move-object/from16 v0, p0
***************
*** 585,594 ****
--- 631,642 ----
  
      invoke-virtual/range {v27 .. v29}, Landroid/widget/AbsListView;->smoothScrollBy(II)V
  
+     .line 4128
      move-object/from16 v0, p0
  
      iput v6, v0, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
  
+     .line 4130
      move-object/from16 v0, p0
  
      iget v0, v0, Landroid/widget/AbsListView$PositionScroller;->mTargetPos:I
***************
*** 599,604 ****
--- 647,653 ----
  
      if-le v6, v0, :cond_0
  
+     .line 4131
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 613,618 ****
--- 662,668 ----
  
      goto/16 :goto_0
  
+     .line 4124
      .end local v5           #extraScroll:I
      :cond_7
      move-object/from16 v0, p0
***************
*** 633,638 ****
--- 683,689 ----
  
      goto :goto_2
  
+     .line 4137
      .end local v7           #firstView:Landroid/view/View;
      .end local v8           #firstViewTop:I
      :pswitch_3
***************
*** 648,658 ****
--- 699,712 ----
  
      add-int/lit8 v13, v27, -0x2
  
+     .line 4138
      .restart local v13       #lastViewIndex:I
      if-ltz v13, :cond_0
  
+     .line 4141
      add-int v9, v6, v13
  
+     .line 4143
      .restart local v9       #lastPos:I
      move-object/from16 v0, p0
  
***************
*** 664,669 ****
--- 718,724 ----
  
      if-ne v9, v0, :cond_8
  
+     .line 4145
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 678,683 ****
--- 733,739 ----
  
      goto/16 :goto_0
  
+     .line 4149
      :cond_8
      move-object/from16 v0, p0
  
***************
*** 691,714 ****
--- 747,775 ----
  
      move-result-object v10
  
+     .line 4150
      .restart local v10       #lastView:Landroid/view/View;
      invoke-virtual {v10}, Landroid/view/View;->getHeight()I
  
      move-result v12
  
+     .line 4151
      .restart local v12       #lastViewHeight:I
      invoke-virtual {v10}, Landroid/view/View;->getTop()I
  
      move-result v15
  
+     .line 4152
      .restart local v15       #lastViewTop:I
      sub-int v14, v16, v15
  
+     .line 4153
      .restart local v14       #lastViewPixelsShowing:I
      move-object/from16 v0, p0
  
      iput v9, v0, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
  
+     .line 4154
      move-object/from16 v0, p0
  
      iget v0, v0, Landroid/widget/AbsListView$PositionScroller;->mBoundPos:I
***************
*** 719,724 ****
--- 780,786 ----
  
      if-le v9, v0, :cond_9
  
+     .line 4155
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 747,752 ****
--- 809,815 ----
  
      invoke-virtual/range {v27 .. v29}, Landroid/widget/AbsListView;->smoothScrollBy(II)V
  
+     .line 4156
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 761,766 ****
--- 824,830 ----
  
      goto/16 :goto_0
  
+     .line 4158
      :cond_9
      move-object/from16 v0, p0
  
***************
*** 770,781 ****
--- 834,848 ----
  
      sub-int v2, v16, v27
  
+     .line 4159
      .local v2, bottom:I
      add-int v11, v15, v12
  
+     .line 4160
      .local v11, lastViewBottom:I
      if-le v2, v11, :cond_0
  
+     .line 4161
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 800,805 ****
--- 867,873 ----
  
      goto/16 :goto_0
  
+     .line 4168
      .end local v2           #bottom:I
      .end local v9           #lastPos:I
      .end local v10           #lastView:Landroid/view/View;
***************
*** 819,824 ****
--- 887,893 ----
  
      if-ne v0, v6, :cond_a
  
+     .line 4170
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 833,843 ****
--- 902,914 ----
  
      goto/16 :goto_0
  
+     .line 4174
      :cond_a
      move-object/from16 v0, p0
  
      iput v6, v0, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
  
+     .line 4176
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 848,853 ****
--- 919,925 ----
  
      move-result v3
  
+     .line 4177
      .restart local v3       #childCount:I
      move-object/from16 v0, p0
  
***************
*** 855,877 ****
--- 927,954 ----
  
      move/from16 v23, v0
  
+     .line 4178
      .local v23, position:I
      add-int v27, v6, v3
  
      add-int/lit8 v9, v27, -0x1
  
+     .line 4180
      .restart local v9       #lastPos:I
      const/16 v26, 0x0
  
+     .line 4181
      .local v26, viewTravelCount:I
      move/from16 v0, v23
  
      if-ge v0, v6, :cond_c
  
+     .line 4182
      sub-int v27, v6, v23
  
      add-int/lit8 v26, v27, 0x1
  
+     .line 4188
      :cond_b
      :goto_3
      move/from16 v0, v26
***************
*** 886,891 ****
--- 963,969 ----
  
      div-float v24, v27, v28
  
+     .line 4190
      .local v24, screenTravelCount:F
      invoke-static/range {v24 .. v24}, Ljava/lang/Math;->abs(F)F
  
***************
*** 897,907 ****
--- 975,987 ----
  
      move-result v17
  
+     .line 4191
      .local v17, modifier:F
      move/from16 v0, v23
  
      if-ge v0, v6, :cond_d
  
+     .line 4192
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 946,951 ****
--- 1026,1032 ----
  
      invoke-virtual/range {v27 .. v29}, Landroid/widget/AbsListView;->smoothScrollBy(II)V
  
+     .line 4193
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 960,965 ****
--- 1041,1047 ----
  
      goto/16 :goto_0
  
+     .line 4183
      .end local v17           #modifier:F
      .end local v24           #screenTravelCount:F
      :cond_c
***************
*** 967,976 ****
--- 1049,1060 ----
  
      if-le v0, v9, :cond_b
  
+     .line 4184
      sub-int v26, v23, v9
  
      goto :goto_3
  
+     .line 4194
      .restart local v17       #modifier:F
      .restart local v24       #screenTravelCount:F
      :cond_d
***************
*** 978,983 ****
--- 1062,1068 ----
  
      if-le v0, v9, :cond_e
  
+     .line 4195
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 1016,1021 ****
--- 1101,1107 ----
  
      invoke-virtual/range {v27 .. v29}, Landroid/widget/AbsListView;->smoothScrollBy(II)V
  
+     .line 4196
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 1030,1035 ****
--- 1116,1122 ----
  
      goto/16 :goto_0
  
+     .line 4199
      :cond_e
      move-object/from16 v0, p0
  
***************
*** 1047,1052 ****
--- 1134,1140 ----
  
      move-result v25
  
+     .line 4200
      .local v25, targetTop:I
      move-object/from16 v0, p0
  
***************
*** 1056,1061 ****
--- 1144,1150 ----
  
      sub-int v4, v25, v27
  
+     .line 4201
      .local v4, distance:I
      move-object/from16 v0, p0
  
***************
*** 1113,1118 ****
--- 1202,1208 ----
  
      goto/16 :goto_0
  
+     .line 4046
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 1132,1143 ****
--- 1222,1236 ----
  
      const/4 v4, -0x1
  
+     .line 3906
      invoke-virtual {p0}, Landroid/widget/AbsListView$PositionScroller;->stop()V
  
+     .line 3908
      iget-object v3, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
      iget v0, v3, Landroid/widget/AbsListView;->mFirstPosition:I
  
+     .line 3909
      .local v0, firstPos:I
      iget-object v3, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
***************
*** 1149,1195 ****
--- 1242,1301 ----
  
      add-int/lit8 v1, v3, -0x1
  
+     .line 3912
      .local v1, lastPos:I
      if-gt p1, v0, :cond_1
  
+     .line 3913
      sub-int v3, v0, p1
  
      add-int/lit8 v2, v3, 0x1
  
+     .line 3914
      .local v2, viewTravelCount:I
      const/4 v3, 0x2
  
      iput v3, p0, Landroid/widget/AbsListView$PositionScroller;->mMode:I
  
+     .line 3923
      :goto_0
      if-lez v2, :cond_2
  
+     .line 3924
      div-int v3, v5, v2
  
      iput v3, p0, Landroid/widget/AbsListView$PositionScroller;->mScrollDuration:I
  
+     .line 3928
      :goto_1
      iput p1, p0, Landroid/widget/AbsListView$PositionScroller;->mTargetPos:I
  
+     .line 3929
      iput v4, p0, Landroid/widget/AbsListView$PositionScroller;->mBoundPos:I
  
+     .line 3930
      iput v4, p0, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
  
+     .line 3932
      iget-object v3, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
      invoke-virtual {v3, p0}, Landroid/widget/AbsListView;->post(Ljava/lang/Runnable;)Z
  
+     .line 3933
      .end local v2           #viewTravelCount:I
      :cond_0
      return-void
  
+     .line 3915
      :cond_1
      if-lt p1, v1, :cond_0
  
+     .line 3916
      sub-int v3, p1, v1
  
      add-int/lit8 v2, v3, 0x1
  
+     .line 3917
      .restart local v2       #viewTravelCount:I
      const/4 v3, 0x1
  
***************
*** 1197,1202 ****
--- 1303,1309 ----
  
      goto :goto_0
  
+     .line 3926
      :cond_2
      iput v5, p0, Landroid/widget/AbsListView$PositionScroller;->mScrollDuration:I
  
***************
*** 1215,1235 ****
--- 1322,1348 ----
  
      const/4 v8, 0x1
  
+     .line 3936
      invoke-virtual {p0}, Landroid/widget/AbsListView$PositionScroller;->stop()V
  
+     .line 3938
      if-ne p2, v9, :cond_1
  
+     .line 3939
      invoke-virtual {p0, p1}, Landroid/widget/AbsListView$PositionScroller;->start(I)V
  
+     .line 3994
      :cond_0
      :goto_0
      return-void
  
+     .line 3943
      :cond_1
      iget-object v7, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
      iget v3, v7, Landroid/widget/AbsListView;->mFirstPosition:I
  
+     .line 3944
      .local v3, firstPos:I
      iget-object v7, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
***************
*** 1241,1297 ****
--- 1354,1426 ----
  
      add-int/lit8 v4, v7, -0x1
  
+     .line 3947
      .local v4, lastPos:I
      if-gt p1, v3, :cond_3
  
+     .line 3948
      sub-int v1, v4, p2
  
+     .line 3949
      .local v1, boundPosFromLast:I
      if-lt v1, v8, :cond_0
  
+     .line 3954
      sub-int v7, v3, p1
  
      add-int/lit8 v5, v7, 0x1
  
+     .line 3955
      .local v5, posTravel:I
      add-int/lit8 v2, v1, -0x1
  
+     .line 3956
      .local v2, boundTravel:I
      if-ge v2, v5, :cond_2
  
+     .line 3957
      move v6, v2
  
+     .line 3958
      .local v6, viewTravelCount:I
      const/4 v7, 0x4
  
      iput v7, p0, Landroid/widget/AbsListView$PositionScroller;->mMode:I
  
+     .line 3984
      .end local v1           #boundPosFromLast:I
      :goto_1
      if-lez v6, :cond_5
  
+     .line 3985
      div-int v7, v10, v6
  
      iput v7, p0, Landroid/widget/AbsListView$PositionScroller;->mScrollDuration:I
  
+     .line 3989
      :goto_2
      iput p1, p0, Landroid/widget/AbsListView$PositionScroller;->mTargetPos:I
  
+     .line 3990
      iput p2, p0, Landroid/widget/AbsListView$PositionScroller;->mBoundPos:I
  
+     .line 3991
      iput v9, p0, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
  
+     .line 3993
      iget-object v7, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
      invoke-virtual {v7, p0}, Landroid/widget/AbsListView;->post(Ljava/lang/Runnable;)Z
  
      goto :goto_0
  
+     .line 3960
      .end local v6           #viewTravelCount:I
      .restart local v1       #boundPosFromLast:I
      :cond_2
      move v6, v5
  
+     .line 3961
      .restart local v6       #viewTravelCount:I
      const/4 v7, 0x2
  
***************
*** 1299,1304 ****
--- 1428,1434 ----
  
      goto :goto_1
  
+     .line 3963
      .end local v1           #boundPosFromLast:I
      .end local v2           #boundTravel:I
      .end local v5           #posTravel:I
***************
*** 1306,1328 ****
--- 1436,1465 ----
      :cond_3
      if-lt p1, v4, :cond_0
  
+     .line 3964
      sub-int v0, p2, v3
  
+     .line 3965
      .local v0, boundPosFromFirst:I
      if-lt v0, v8, :cond_0
  
+     .line 3970
      sub-int v7, p1, v4
  
      add-int/lit8 v5, v7, 0x1
  
+     .line 3971
      .restart local v5       #posTravel:I
      add-int/lit8 v2, v0, -0x1
  
+     .line 3972
      .restart local v2       #boundTravel:I
      if-ge v2, v5, :cond_4
  
+     .line 3973
      move v6, v2
  
+     .line 3974
      .restart local v6       #viewTravelCount:I
      const/4 v7, 0x3
  
***************
*** 1330,1344 ****
--- 1467,1484 ----
  
      goto :goto_1
  
+     .line 3976
      .end local v6           #viewTravelCount:I
      :cond_4
      move v6, v5
  
+     .line 3977
      .restart local v6       #viewTravelCount:I
      iput v8, p0, Landroid/widget/AbsListView$PositionScroller;->mMode:I
  
      goto :goto_1
  
+     .line 3987
      .end local v0           #boundPosFromFirst:I
      :cond_5
      iput v10, p0, Landroid/widget/AbsListView$PositionScroller;->mScrollDuration:I
***************
*** 1352,1361 ****
--- 1492,1503 ----
      .parameter "offset"
  
      .prologue
+     .line 3997
      const/16 v0, 0x190
  
      invoke-virtual {p0, p1, p2, v0}, Landroid/widget/AbsListView$PositionScroller;->startWithOffset(III)V
  
+     .line 3998
      return-void
  .end method
  
***************
*** 1368,1391 ****
--- 1510,1541 ----
      .prologue
      const/4 v8, -0x1
  
+     .line 4001
      invoke-virtual {p0}, Landroid/widget/AbsListView$PositionScroller;->stop()V
  
+     .line 4003
      iput p1, p0, Landroid/widget/AbsListView$PositionScroller;->mTargetPos:I
  
+     .line 4004
      iput p2, p0, Landroid/widget/AbsListView$PositionScroller;->mOffsetFromTop:I
  
+     .line 4005
      iput v8, p0, Landroid/widget/AbsListView$PositionScroller;->mBoundPos:I
  
+     .line 4006
      iput v8, p0, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
  
+     .line 4007
      const/4 v6, 0x5
  
      iput v6, p0, Landroid/widget/AbsListView$PositionScroller;->mMode:I
  
+     .line 4009
      iget-object v6, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
      iget v1, v6, Landroid/widget/AbsListView;->mFirstPosition:I
  
+     .line 4010
      .local v1, firstPos:I
      iget-object v6, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
***************
*** 1393,1408 ****
--- 1543,1562 ----
  
      move-result v0
  
+     .line 4011
      .local v0, childCount:I
      add-int v6, v1, v0
  
      add-int/lit8 v2, v6, -0x1
  
+     .line 4014
      .local v2, lastPos:I
      if-ge p1, v1, :cond_0
  
+     .line 4015
      sub-int v5, v1, p1
  
+     .line 4026
      .local v5, viewTravelCount:I
      :goto_0
      int-to-float v6, v5
***************
*** 1411,1416 ****
--- 1565,1571 ----
  
      div-float v3, v6, v7
  
+     .line 4027
      .local v3, screenTravelCount:F
      const/high16 v6, 0x3f80
  
***************
*** 1427,1451 ****
--- 1582,1612 ----
      :goto_1
      iput v6, p0, Landroid/widget/AbsListView$PositionScroller;->mScrollDuration:I
  
+     .line 4029
      iput v8, p0, Landroid/widget/AbsListView$PositionScroller;->mLastSeenPos:I
  
+     .line 4031
      iget-object v6, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
      invoke-virtual {v6, p0}, Landroid/widget/AbsListView;->post(Ljava/lang/Runnable;)Z
  
+     .line 4032
      .end local v3           #screenTravelCount:F
      .end local v5           #viewTravelCount:I
      :goto_2
      return-void
  
+     .line 4016
      :cond_0
      if-le p1, v2, :cond_1
  
+     .line 4017
      sub-int v5, p1, v2
  
      .restart local v5       #viewTravelCount:I
      goto :goto_0
  
+     .line 4020
      .end local v5           #viewTravelCount:I
      :cond_1
      iget-object v6, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
***************
*** 1460,1465 ****
--- 1621,1627 ----
  
      move-result v4
  
+     .line 4021
      .local v4, targetTop:I
      iget-object v6, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
***************
*** 1469,1474 ****
--- 1631,1637 ----
  
      goto :goto_2
  
+     .line 4027
      .end local v4           #targetTop:I
      .restart local v3       #screenTravelCount:F
      .restart local v5       #viewTravelCount:I
***************
*** 1486,1494 ****
--- 1649,1659 ----
      .locals 1
  
      .prologue
+     .line 4035
      iget-object v0, p0, Landroid/widget/AbsListView$PositionScroller;->this$0:Landroid/widget/AbsListView;
  
      invoke-virtual {v0, p0}, Landroid/widget/AbsListView;->removeCallbacks(Ljava/lang/Runnable;)Z
  
+     .line 4036
      return-void
  .end method
