*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserModel.smali	2012-03-23 19:35:54.385578605 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserModel.smali.original	2012-03-23 19:34:44.896311470 -0500
***************
*** 115,120 ****
--- 115,121 ----
      .locals 1
  
      .prologue
+     .line 165
      const-class v0, Landroid/widget/ActivityChooserModel;
  
      invoke-virtual {v0}, Ljava/lang/Class;->getSimpleName()Ljava/lang/String;
***************
*** 123,134 ****
--- 124,137 ----
  
      sput-object v0, Landroid/widget/ActivityChooserModel;->LOG_TAG:Ljava/lang/String;
  
+     .line 226
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      sput-object v0, Landroid/widget/ActivityChooserModel;->sRegistryLock:Ljava/lang/Object;
  
+     .line 231
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
***************
*** 148,209 ****
--- 151,225 ----
  
      const/4 v2, 0x1
  
+     .line 360
      invoke-direct {p0}, Landroid/database/DataSetObservable;-><init>()V
  
+     .line 237
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      iput-object v0, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
+     .line 242
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
+     .line 247
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecords:Ljava/util/List;
  
+     .line 252
      new-instance v0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;
  
      invoke-direct {v0, p0, v1}, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;-><init>(Landroid/widget/ActivityChooserModel;Landroid/widget/ActivityChooserModel$1;)V
  
      iput-object v0, p0, Landroid/widget/ActivityChooserModel;->mPackageMonitor:Lcom/android/internal/content/PackageMonitor;
  
+     .line 272
      new-instance v0, Landroid/widget/ActivityChooserModel$DefaultSorter;
  
      invoke-direct {v0, p0, v1}, Landroid/widget/ActivityChooserModel$DefaultSorter;-><init>(Landroid/widget/ActivityChooserModel;Landroid/widget/ActivityChooserModel$1;)V
  
      iput-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivitySorter:Landroid/widget/ActivityChooserModel$ActivitySorter;
  
+     .line 277
      const/16 v0, 0x32
  
      iput v0, p0, Landroid/widget/ActivityChooserModel;->mHistoryMaxSize:I
  
+     .line 287
      iput-boolean v2, p0, Landroid/widget/ActivityChooserModel;->mCanReadHistoricalData:Z
  
+     .line 298
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/ActivityChooserModel;->mReadShareHistoryCalled:Z
  
+     .line 306
      iput-boolean v2, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecordsChanged:Z
  
+     .line 311
      new-instance v0, Landroid/os/Handler;
  
      invoke-direct {v0}, Landroid/os/Handler;-><init>()V
  
      iput-object v0, p0, Landroid/widget/ActivityChooserModel;->mHandler:Landroid/os/Handler;
  
+     .line 361
      invoke-virtual {p1}, Landroid/content/Context;->getApplicationContext()Landroid/content/Context;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/widget/ActivityChooserModel;->mContext:Landroid/content/Context;
  
+     .line 362
      invoke-static {p2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v0
***************
*** 218,223 ****
--- 234,240 ----
  
      if-nez v0, :cond_0
  
+     .line 364
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 238,243 ****
--- 255,261 ----
  
      iput-object v0, p0, Landroid/widget/ActivityChooserModel;->mHistoryFileName:Ljava/lang/String;
  
+     .line 368
      :goto_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mPackageMonitor:Lcom/android/internal/content/PackageMonitor;
  
***************
*** 245,252 ****
--- 263,272 ----
  
      invoke-virtual {v0, v1, v2}, Lcom/android/internal/content/PackageMonitor;->register(Landroid/content/Context;Z)V
  
+     .line 369
      return-void
  
+     .line 366
      :cond_0
      iput-object p2, p0, Landroid/widget/ActivityChooserModel;->mHistoryFileName:Ljava/lang/String;
  
***************
*** 258,263 ****
--- 278,284 ----
      .parameter "x0"
  
      .prologue
+     .line 98
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->sortActivities()V
  
      return-void
***************
*** 268,273 ****
--- 289,295 ----
      .parameter "x0"
  
      .prologue
+     .line 98
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mHandler:Landroid/os/Handler;
  
      return-object v0
***************
*** 277,282 ****
--- 299,305 ----
      .locals 1
  
      .prologue
+     .line 98
      sget-object v0, Landroid/widget/ActivityChooserModel;->LOG_TAG:Ljava/lang/String;
  
      return-object v0
***************
*** 287,292 ****
--- 310,316 ----
      .parameter "x0"
  
      .prologue
+     .line 98
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->loadActivitiesLocked()V
  
      return-void
***************
*** 298,303 ****
--- 322,328 ----
      .parameter "x1"
  
      .prologue
+     .line 98
      invoke-direct {p0, p1}, Landroid/widget/ActivityChooserModel;->pruneHistoricalRecordsForPackageLocked(Ljava/lang/String;)V
  
      return-void
***************
*** 308,313 ****
--- 333,339 ----
      .parameter "x0"
  
      .prologue
+     .line 98
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mHistoryFileName:Ljava/lang/String;
  
      return-object v0
***************
*** 318,323 ****
--- 344,350 ----
      .parameter "x0"
  
      .prologue
+     .line 98
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 328,333 ****
--- 355,361 ----
      .parameter "x0"
  
      .prologue
+     .line 98
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      return-object v0
***************
*** 338,343 ****
--- 366,372 ----
      .parameter "x0"
  
      .prologue
+     .line 98
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecords:Ljava/util/List;
  
      return-object v0
***************
*** 349,354 ****
--- 378,384 ----
      .parameter "x1"
  
      .prologue
+     .line 98
      iput-boolean p1, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecordsChanged:Z
  
      return p1
***************
*** 359,364 ****
--- 389,395 ----
      .parameter "x0"
  
      .prologue
+     .line 98
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->pruneExcessiveHistoricalRecordsLocked()V
  
      return-void
***************
*** 369,378 ****
--- 400,411 ----
      .parameter "historicalRecord"
  
      .prologue
+     .line 690
      iget-object v2, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 691
      :try_start_0
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecords:Ljava/util/List;
  
***************
*** 380,403 ****
--- 413,443 ----
  
      move-result v0
  
+     .line 692
      .local v0, added:Z
      if-eqz v0, :cond_0
  
+     .line 693
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecordsChanged:Z
  
+     .line 694
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->pruneExcessiveHistoricalRecordsLocked()V
  
+     .line 695
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->persistHistoricalData()V
  
+     .line 696
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->sortActivities()V
  
+     .line 698
      :cond_0
      monitor-exit v2
  
      return v0
  
+     .line 699
      .end local v0           #added:Z
      :catchall_0
      move-exception v1
***************
*** 415,424 ****
--- 455,466 ----
      .parameter "historyFileName"
  
      .prologue
+     .line 343
      sget-object v2, Landroid/widget/ActivityChooserModel;->sRegistryLock:Ljava/lang/Object;
  
      monitor-enter v2
  
+     .line 344
      :try_start_0
      sget-object v1, Landroid/widget/ActivityChooserModel;->sDataModelRegistry:Ljava/util/Map;
  
***************
*** 428,453 ****
--- 470,501 ----
  
      check-cast v0, Landroid/widget/ActivityChooserModel;
  
+     .line 345
      .local v0, dataModel:Landroid/widget/ActivityChooserModel;
      if-nez v0, :cond_0
  
+     .line 346
      new-instance v0, Landroid/widget/ActivityChooserModel;
  
      .end local v0           #dataModel:Landroid/widget/ActivityChooserModel;
      invoke-direct {v0, p0, p1}, Landroid/widget/ActivityChooserModel;-><init>(Landroid/content/Context;Ljava/lang/String;)V
  
+     .line 347
      .restart local v0       #dataModel:Landroid/widget/ActivityChooserModel;
      sget-object v1, Landroid/widget/ActivityChooserModel;->sDataModelRegistry:Ljava/util/Map;
  
      invoke-interface {v1, p1, v0}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 349
      :cond_0
      invoke-direct {v0}, Landroid/widget/ActivityChooserModel;->readHistoricalData()V
  
+     .line 350
      monitor-exit v2
  
      return-object v0
  
+     .line 351
      .end local v0           #dataModel:Landroid/widget/ActivityChooserModel;
      :catchall_0
      move-exception v1
***************
*** 463,476 ****
--- 511,527 ----
      .locals 7
  
      .prologue
+     .line 724
      iget-object v4, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
      invoke-interface {v4}, Ljava/util/List;->clear()V
  
+     .line 725
      iget-object v4, p0, Landroid/widget/ActivityChooserModel;->mIntent:Landroid/content/Intent;
  
      if-eqz v4, :cond_1
  
+     .line 726
      iget-object v4, p0, Landroid/widget/ActivityChooserModel;->mContext:Landroid/content/Context;
  
      invoke-virtual {v4}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
***************
*** 485,495 ****
--- 536,548 ----
  
      move-result-object v3
  
+     .line 728
      .local v3, resolveInfos:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
      invoke-interface {v3}, Ljava/util/List;->size()I
  
      move-result v2
  
+     .line 729
      .local v2, resolveInfoCount:I
      const/4 v0, 0x0
  
***************
*** 497,508 ****
--- 550,563 ----
      :goto_0
      if-ge v0, v2, :cond_0
  
+     .line 730
      invoke-interface {v3, v0}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Landroid/content/pm/ResolveInfo;
  
+     .line 731
      .local v1, resolveInfo:Landroid/content/pm/ResolveInfo;
      iget-object v4, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
***************
*** 512,531 ****
--- 567,590 ----
  
      invoke-interface {v4, v5}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 729
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 733
      .end local v1           #resolveInfo:Landroid/content/pm/ResolveInfo;
      :cond_0
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->sortActivities()V
  
+     .line 737
      .end local v0           #i:I
      .end local v2           #resolveInfoCount:I
      .end local v3           #resolveInfos:Ljava/util/List;,"Ljava/util/List<Landroid/content/pm/ResolveInfo;>;"
      :goto_1
      return-void
  
+     .line 735
      :cond_1
      invoke-virtual {p0}, Landroid/widget/ActivityChooserModel;->notifyChanged()V
  
***************
*** 536,550 ****
--- 595,612 ----
      .locals 4
  
      .prologue
+     .line 583
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 584
      :try_start_0
      iget-boolean v0, p0, Landroid/widget/ActivityChooserModel;->mReadShareHistoryCalled:Z
  
      if-nez v0, :cond_0
  
+     .line 585
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v2, "No preceding call to #readHistoricalData"
***************
*** 553,558 ****
--- 615,621 ----
  
      throw v0
  
+     .line 595
      :catchall_0
      move-exception v0
  
***************
*** 562,587 ****
--- 625,656 ----
  
      throw v0
  
+     .line 587
      :cond_0
      :try_start_1
      iget-boolean v0, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecordsChanged:Z
  
      if-nez v0, :cond_1
  
+     .line 588
      monitor-exit v1
  
+     .line 596
      :goto_0
      return-void
  
+     .line 590
      :cond_1
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecordsChanged:Z
  
+     .line 591
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/ActivityChooserModel;->mCanReadHistoricalData:Z
  
+     .line 592
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mHistoryFileName:Ljava/lang/String;
  
      invoke-static {v0}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
***************
*** 590,595 ****
--- 659,665 ----
  
      if-nez v0, :cond_2
  
+     .line 593
      sget-object v0, Landroid/os/AsyncTask;->SERIAL_EXECUTOR:Ljava/util/concurrent/Executor;
  
      new-instance v2, Landroid/widget/ActivityChooserModel$HistoryPersister;
***************
*** 600,605 ****
--- 670,676 ----
  
      invoke-interface {v0, v2}, Ljava/util/concurrent/Executor;->execute(Ljava/lang/Runnable;)V
  
+     .line 595
      :cond_2
      monitor-exit v1
      :try_end_1
***************
*** 612,619 ****
--- 683,692 ----
      .locals 5
  
      .prologue
+     .line 706
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecords:Ljava/util/List;
  
+     .line 707
      .local v0, choiceRecords:Ljava/util/List;,"Ljava/util/List<Landroid/widget/ActivityChooserModel$HistoricalRecord;>;"
      invoke-interface {v0}, Ljava/util/List;->size()I
  
***************
*** 623,645 ****
--- 696,723 ----
  
      sub-int v2, v3, v4
  
+     .line 708
      .local v2, pruneCount:I
      if-gtz v2, :cond_1
  
+     .line 718
      :cond_0
      return-void
  
+     .line 711
      :cond_1
      const/4 v3, 0x1
  
      iput-boolean v3, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecordsChanged:Z
  
+     .line 712
      const/4 v1, 0x0
  
      .local v1, i:I
      :goto_0
      if-ge v1, v2, :cond_0
  
+     .line 713
      const/4 v3, 0x0
  
      invoke-interface {v0, v3}, Ljava/util/List;->remove(I)Ljava/lang/Object;
***************
*** 648,653 ****
--- 726,732 ----
  
      check-cast v3, Landroid/widget/ActivityChooserModel$HistoricalRecord;
  
+     .line 712
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
***************
*** 658,668 ****
--- 737,750 ----
      .parameter "packageName"
  
      .prologue
+     .line 745
      const/4 v4, 0x0
  
+     .line 747
      .local v4, recordsRemoved:Z
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecords:Ljava/util/List;
  
+     .line 748
      .local v1, historicalRecords:Ljava/util/List;,"Ljava/util/List<Landroid/widget/ActivityChooserModel$HistoricalRecord;>;"
      const/4 v2, 0x0
  
***************
*** 674,685 ****
--- 756,769 ----
  
      if-ge v2, v5, :cond_1
  
+     .line 749
      invoke-interface {v1, v2}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Landroid/widget/ActivityChooserModel$HistoricalRecord;
  
+     .line 750
      .local v0, historicalRecord:Landroid/widget/ActivityChooserModel$HistoricalRecord;
      iget-object v5, v0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->activity:Landroid/content/ComponentName;
  
***************
*** 687,692 ****
--- 771,777 ----
  
      move-result-object v3
  
+     .line 751
      .local v3, recordPackageName:Ljava/lang/String;
      invoke-virtual {v3, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 694,719 ****
--- 779,811 ----
  
      if-eqz v5, :cond_0
  
+     .line 752
      invoke-interface {v1, v0}, Ljava/util/List;->remove(Ljava/lang/Object;)Z
  
+     .line 753
      const/4 v4, 0x1
  
+     .line 748
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 757
      .end local v0           #historicalRecord:Landroid/widget/ActivityChooserModel$HistoricalRecord;
      .end local v3           #recordPackageName:Ljava/lang/String;
      :cond_1
      if-eqz v4, :cond_2
  
+     .line 758
      const/4 v5, 0x1
  
      iput-boolean v5, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecordsChanged:Z
  
+     .line 759
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->sortActivities()V
  
+     .line 761
      :cond_2
      return-void
  .end method
***************
*** 722,731 ****
--- 814,825 ----
      .locals 4
  
      .prologue
+     .line 561
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 562
      :try_start_0
      iget-boolean v0, p0, Landroid/widget/ActivityChooserModel;->mCanReadHistoricalData:Z
  
***************
*** 735,755 ****
--- 829,854 ----
  
      if-nez v0, :cond_1
  
+     .line 563
      :cond_0
      monitor-exit v1
  
+     .line 571
      :goto_0
      return-void
  
+     .line 565
      :cond_1
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/ActivityChooserModel;->mCanReadHistoricalData:Z
  
+     .line 566
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/ActivityChooserModel;->mReadShareHistoryCalled:Z
  
+     .line 567
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mHistoryFileName:Ljava/lang/String;
  
      invoke-static {v0}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
***************
*** 758,763 ****
--- 857,863 ----
  
      if-nez v0, :cond_2
  
+     .line 568
      sget-object v0, Landroid/os/AsyncTask;->SERIAL_EXECUTOR:Ljava/util/concurrent/Executor;
  
      new-instance v2, Landroid/widget/ActivityChooserModel$HistoryLoader;
***************
*** 768,773 ****
--- 868,874 ----
  
      invoke-interface {v0, v2}, Ljava/util/concurrent/Executor;->execute(Ljava/lang/Runnable;)V
  
+     .line 570
      :cond_2
      monitor-exit v1
  
***************
*** 787,796 ****
--- 888,899 ----
      .locals 5
  
      .prologue
+     .line 622
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 623
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivitySorter:Landroid/widget/ActivityChooserModel$ActivitySorter;
  
***************
*** 804,809 ****
--- 907,913 ----
  
      if-nez v0, :cond_0
  
+     .line 624
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivitySorter:Landroid/widget/ActivityChooserModel$ActivitySorter;
  
      iget-object v2, p0, Landroid/widget/ActivityChooserModel;->mIntent:Landroid/content/Intent;
***************
*** 818,830 ****
--- 922,938 ----
  
      invoke-interface {v0, v2, v3, v4}, Landroid/widget/ActivityChooserModel$ActivitySorter;->sort(Landroid/content/Intent;Ljava/util/List;Ljava/util/List;)V
  
+     .line 626
      invoke-virtual {p0}, Landroid/widget/ActivityChooserModel;->notifyChanged()V
  
+     .line 628
      :cond_0
      monitor-exit v1
  
+     .line 629
      return-void
  
+     .line 628
      :catchall_0
      move-exception v0
  
***************
*** 842,847 ****
--- 950,956 ----
      .parameter "index"
  
      .prologue
+     .line 465
      iget-object v6, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
      invoke-interface {v6, p1}, Ljava/util/List;->get(I)Ljava/lang/Object;
***************
*** 850,855 ****
--- 959,965 ----
  
      check-cast v2, Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
  
+     .line 467
      .local v2, chosenActivity:Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
      new-instance v3, Landroid/content/ComponentName;
  
***************
*** 867,872 ****
--- 977,983 ----
  
      invoke-direct {v3, v6, v7}, Landroid/content/ComponentName;-><init>(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 471
      .local v3, chosenName:Landroid/content/ComponentName;
      new-instance v0, Landroid/content/Intent;
  
***************
*** 874,890 ****
--- 985,1005 ----
  
      invoke-direct {v0, v6}, Landroid/content/Intent;-><init>(Landroid/content/Intent;)V
  
+     .line 472
      .local v0, choiceIntent:Landroid/content/Intent;
      invoke-virtual {v0, v3}, Landroid/content/Intent;->setComponent(Landroid/content/ComponentName;)Landroid/content/Intent;
  
+     .line 474
      iget-object v6, p0, Landroid/widget/ActivityChooserModel;->mActivityChoserModelPolicy:Landroid/widget/ActivityChooserModel$OnChooseActivityListener;
  
      if-eqz v6, :cond_0
  
+     .line 476
      new-instance v1, Landroid/content/Intent;
  
      invoke-direct {v1, v0}, Landroid/content/Intent;-><init>(Landroid/content/Intent;)V
  
+     .line 477
      .local v1, choiceIntentCopy:Landroid/content/Intent;
      iget-object v6, p0, Landroid/widget/ActivityChooserModel;->mActivityChoserModelPolicy:Landroid/widget/ActivityChooserModel$OnChooseActivityListener;
  
***************
*** 892,908 ****
--- 1007,1027 ----
  
      move-result v4
  
+     .line 479
      .local v4, handled:Z
      if-eqz v4, :cond_0
  
+     .line 480
      const/4 v0, 0x0
  
+     .line 488
      .end local v0           #choiceIntent:Landroid/content/Intent;
      .end local v1           #choiceIntentCopy:Landroid/content/Intent;
      .end local v4           #handled:Z
      :goto_0
      return-object v0
  
+     .line 484
      .restart local v0       #choiceIntent:Landroid/content/Intent;
      :cond_0
      new-instance v5, Landroid/widget/ActivityChooserModel$HistoricalRecord;
***************
*** 915,920 ****
--- 1034,1040 ----
  
      invoke-direct {v5, v3, v6, v7, v8}, Landroid/widget/ActivityChooserModel$HistoricalRecord;-><init>(Landroid/content/ComponentName;JF)V
  
+     .line 486
      .local v5, historicalRecord:Landroid/widget/ActivityChooserModel$HistoricalRecord;
      invoke-direct {p0, v5}, Landroid/widget/ActivityChooserModel;->addHisoricalRecord(Landroid/widget/ActivityChooserModel$HistoricalRecord;)Z
  
***************
*** 930,941 ****
--- 1050,1064 ----
      .end annotation
  
      .prologue
+     .line 679
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 680
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mPackageMonitor:Lcom/android/internal/content/PackageMonitor;
  
      invoke-virtual {v0}, Lcom/android/internal/content/PackageMonitor;->unregister()V
  
+     .line 681
      return-void
  .end method
  
***************
*** 944,953 ****
--- 1067,1078 ----
      .parameter "index"
  
      .prologue
+     .line 423
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 424
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
***************
*** 963,968 ****
--- 1088,1094 ----
  
      return-object v0
  
+     .line 425
      :catchall_0
      move-exception v0
  
***************
*** 977,986 ****
--- 1103,1114 ----
      .locals 2
  
      .prologue
+     .line 409
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 410
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
***************
*** 992,997 ****
--- 1120,1126 ----
  
      return v0
  
+     .line 411
      :catchall_0
      move-exception v0
  
***************
*** 1007,1019 ****
--- 1136,1151 ----
      .parameter "activity"
  
      .prologue
+     .line 436
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
+     .line 437
      .local v0, activities:Ljava/util/List;,"Ljava/util/List<Landroid/widget/ActivityChooserModel$ActivityResolveInfo;>;"
      invoke-interface {v0}, Ljava/util/List;->size()I
  
      move-result v1
  
+     .line 438
      .local v1, activityCount:I
      const/4 v3, 0x0
  
***************
*** 1021,1042 ****
--- 1153,1178 ----
      :goto_0
      if-ge v3, v1, :cond_1
  
+     .line 439
      invoke-interface {v0, v3}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v2
  
      check-cast v2, Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
  
+     .line 440
      .local v2, currentActivity:Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
      iget-object v4, v2, Landroid/widget/ActivityChooserModel$ActivityResolveInfo;->resolveInfo:Landroid/content/pm/ResolveInfo;
  
      if-ne v4, p1, :cond_0
  
+     .line 444
      .end local v2           #currentActivity:Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
      .end local v3           #i:I
      :goto_1
      return v3
  
+     .line 438
      .restart local v2       #currentActivity:Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
      .restart local v3       #i:I
      :cond_0
***************
*** 1044,1049 ****
--- 1180,1186 ----
  
      goto :goto_0
  
+     .line 444
      .end local v2           #currentActivity:Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
      :cond_1
      const/4 v3, -0x1
***************
*** 1055,1064 ****
--- 1192,1203 ----
      .locals 3
  
      .prologue
+     .line 510
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 511
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
***************
*** 1068,1073 ****
--- 1207,1213 ----
  
      if-nez v0, :cond_0
  
+     .line 512
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
      const/4 v2, 0x0
***************
*** 1082,1097 ****
--- 1222,1241 ----
  
      monitor-exit v1
  
+     .line 515
      :goto_0
      return-object v0
  
+     .line 514
      :cond_0
      monitor-exit v1
  
+     .line 515
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 514
      :catchall_0
      move-exception v0
  
***************
*** 1106,1115 ****
--- 1250,1261 ----
      .locals 2
  
      .prologue
+     .line 661
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 662
      :try_start_0
      iget v0, p0, Landroid/widget/ActivityChooserModel;->mHistoryMaxSize:I
  
***************
*** 1117,1122 ****
--- 1263,1269 ----
  
      return v0
  
+     .line 663
      :catchall_0
      move-exception v0
  
***************
*** 1131,1140 ****
--- 1278,1289 ----
      .locals 2
  
      .prologue
+     .line 672
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 673
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mHistoricalRecords:Ljava/util/List;
  
***************
*** 1146,1151 ****
--- 1295,1301 ----
  
      return v0
  
+     .line 674
      :catchall_0
      move-exception v0
  
***************
*** 1160,1169 ****
--- 1310,1321 ----
      .locals 2
  
      .prologue
+     .line 396
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 397
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mIntent:Landroid/content/Intent;
  
***************
*** 1171,1176 ****
--- 1323,1329 ----
  
      return-object v0
  
+     .line 398
      :catchall_0
      move-exception v0
  
***************
*** 1186,1210 ****
--- 1339,1370 ----
      .parameter "activitySorter"
  
      .prologue
+     .line 606
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 607
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mActivitySorter:Landroid/widget/ActivityChooserModel$ActivitySorter;
  
      if-ne v0, p1, :cond_0
  
+     .line 608
      monitor-exit v1
  
+     .line 613
      :goto_0
      return-void
  
+     .line 610
      :cond_0
      iput-object p1, p0, Landroid/widget/ActivityChooserModel;->mActivitySorter:Landroid/widget/ActivityChooserModel$ActivitySorter;
  
+     .line 611
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->sortActivities()V
  
+     .line 612
      monitor-exit v1
  
      goto :goto_0
***************
*** 1224,1229 ****
--- 1384,1390 ----
      .parameter "index"
  
      .prologue
+     .line 529
      iget-object v5, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
      invoke-interface {v5, p1}, Ljava/util/List;->get(I)Ljava/lang/Object;
***************
*** 1232,1237 ****
--- 1393,1399 ----
  
      check-cast v2, Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
  
+     .line 530
      .local v2, newDefaultActivity:Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
      iget-object v5, p0, Landroid/widget/ActivityChooserModel;->mActivites:Ljava/util/List;
  
***************
*** 1243,1251 ****
--- 1405,1415 ----
  
      check-cast v3, Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
  
+     .line 533
      .local v3, oldDefaultActivity:Landroid/widget/ActivityChooserModel$ActivityResolveInfo;
      if-eqz v3, :cond_0
  
+     .line 535
      iget v5, v3, Landroid/widget/ActivityChooserModel$ActivityResolveInfo;->weight:F
  
      iget v6, v2, Landroid/widget/ActivityChooserModel$ActivityResolveInfo;->weight:F
***************
*** 1256,1261 ****
--- 1420,1426 ----
  
      add-float v4, v5, v6
  
+     .line 541
      .local v4, weight:F
      :goto_0
      new-instance v0, Landroid/content/ComponentName;
***************
*** 1274,1279 ****
--- 1439,1445 ----
  
      invoke-direct {v0, v5, v6}, Landroid/content/ComponentName;-><init>(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 544
      .local v0, defaultName:Landroid/content/ComponentName;
      new-instance v1, Landroid/widget/ActivityChooserModel$HistoricalRecord;
  
***************
*** 1283,1293 ****
--- 1449,1462 ----
  
      invoke-direct {v1, v0, v5, v6, v4}, Landroid/widget/ActivityChooserModel$HistoricalRecord;-><init>(Landroid/content/ComponentName;JF)V
  
+     .line 546
      .local v1, historicalRecord:Landroid/widget/ActivityChooserModel$HistoricalRecord;
      invoke-direct {p0, v1}, Landroid/widget/ActivityChooserModel;->addHisoricalRecord(Landroid/widget/ActivityChooserModel$HistoricalRecord;)Z
  
+     .line 547
      return-void
  
+     .line 538
      .end local v0           #defaultName:Landroid/content/ComponentName;
      .end local v1           #historicalRecord:Landroid/widget/ActivityChooserModel$HistoricalRecord;
      .end local v4           #weight:F
***************
*** 1303,1329 ****
--- 1472,1506 ----
      .parameter "historyMaxSize"
  
      .prologue
+     .line 645
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 646
      :try_start_0
      iget v0, p0, Landroid/widget/ActivityChooserModel;->mHistoryMaxSize:I
  
      if-ne v0, p1, :cond_0
  
+     .line 647
      monitor-exit v1
  
+     .line 653
      :goto_0
      return-void
  
+     .line 649
      :cond_0
      iput p1, p0, Landroid/widget/ActivityChooserModel;->mHistoryMaxSize:I
  
+     .line 650
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->pruneExcessiveHistoricalRecordsLocked()V
  
+     .line 651
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->sortActivities()V
  
+     .line 652
      monitor-exit v1
  
      goto :goto_0
***************
*** 1343,1367 ****
--- 1520,1551 ----
      .parameter "intent"
  
      .prologue
+     .line 381
      iget-object v1, p0, Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 382
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel;->mIntent:Landroid/content/Intent;
  
      if-ne v0, p1, :cond_0
  
+     .line 383
      monitor-exit v1
  
+     .line 388
      :goto_0
      return-void
  
+     .line 385
      :cond_0
      iput-object p1, p0, Landroid/widget/ActivityChooserModel;->mIntent:Landroid/content/Intent;
  
+     .line 386
      invoke-direct {p0}, Landroid/widget/ActivityChooserModel;->loadActivitiesLocked()V
  
+     .line 387
      monitor-exit v1
  
      goto :goto_0
***************
*** 1381,1387 ****
--- 1565,1573 ----
      .parameter "listener"
  
      .prologue
+     .line 497
      iput-object p1, p0, Landroid/widget/ActivityChooserModel;->mActivityChoserModelPolicy:Landroid/widget/ActivityChooserModel$OnChooseActivityListener;
  
+     .line 498
      return-void
  .end method
