*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserModel$HistoryLoader$1.smali	2012-03-23 19:35:46.966443318 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserModel$HistoryLoader$1.smali.original	2012-03-23 19:34:44.837310394 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1055
      iput-object p1, p0, Landroid/widget/ActivityChooserModel$HistoryLoader$1;->this$1:Landroid/widget/ActivityChooserModel$HistoryLoader;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .locals 1
  
      .prologue
+     .line 1057
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$HistoryLoader$1;->this$1:Landroid/widget/ActivityChooserModel$HistoryLoader;
  
      iget-object v0, v0, Landroid/widget/ActivityChooserModel$HistoryLoader;->this$0:Landroid/widget/ActivityChooserModel;
***************
*** 47,52 ****
--- 49,55 ----
      #calls: Landroid/widget/ActivityChooserModel;->pruneExcessiveHistoricalRecordsLocked()V
      invoke-static {v0}, Landroid/widget/ActivityChooserModel;->access$900(Landroid/widget/ActivityChooserModel;)V
  
+     .line 1058
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$HistoryLoader$1;->this$1:Landroid/widget/ActivityChooserModel$HistoryLoader;
  
      iget-object v0, v0, Landroid/widget/ActivityChooserModel$HistoryLoader;->this$0:Landroid/widget/ActivityChooserModel;
***************
*** 54,58 ****
--- 57,62 ----
      #calls: Landroid/widget/ActivityChooserModel;->sortActivities()V
      invoke-static {v0}, Landroid/widget/ActivityChooserModel;->access$1000(Landroid/widget/ActivityChooserModel;)V
  
+     .line 1059
      return-void
  .end method
