*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserView$3.smali	2012-03-23 19:35:58.971662233 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserView$3.smali.original	2012-03-23 19:34:44.931312108 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 239
      iput-object p1, p0, Landroid/widget/ActivityChooserView$3;->this$0:Landroid/widget/ActivityChooserView;
  
      invoke-direct {p0}, Landroid/database/DataSetObserver;-><init>()V
***************
*** 37,48 ****
--- 38,52 ----
      .locals 1
  
      .prologue
+     .line 242
      invoke-super {p0}, Landroid/database/DataSetObserver;->onChanged()V
  
+     .line 243
      iget-object v0, p0, Landroid/widget/ActivityChooserView$3;->this$0:Landroid/widget/ActivityChooserView;
  
      #calls: Landroid/widget/ActivityChooserView;->updateAppearance()V
      invoke-static {v0}, Landroid/widget/ActivityChooserView;->access$400(Landroid/widget/ActivityChooserView;)V
  
+     .line 244
      return-void
  .end method
