*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserModel$DataModelPackageMonitor.smali	2012-03-23 19:35:57.271631233 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserModel$DataModelPackageMonitor.smali.original	2012-03-23 19:34:44.918311872 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1145
      iput-object p1, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      invoke-direct {p0}, Lcom/android/internal/content/PackageMonitor;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 1145
      invoke-direct {p0, p1}, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;-><init>(Landroid/widget/ActivityChooserModel;)V
  
      return-void
***************
*** 50,55 ****
--- 52,58 ----
      .parameter "uid"
  
      .prologue
+     .line 1149
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #getter for: Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
***************
*** 59,74 ****
--- 62,81 ----
  
      monitor-enter v1
  
+     .line 1150
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #calls: Landroid/widget/ActivityChooserModel;->loadActivitiesLocked()V
      invoke-static {v0}, Landroid/widget/ActivityChooserModel;->access$1300(Landroid/widget/ActivityChooserModel;)V
  
+     .line 1151
      monitor-exit v1
  
+     .line 1152
      return-void
  
+     .line 1151
      :catchall_0
      move-exception v0
  
***************
*** 85,90 ****
--- 92,98 ----
      .parameter "reason"
  
      .prologue
+     .line 1156
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #getter for: Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
***************
*** 94,109 ****
--- 102,121 ----
  
      monitor-enter v1
  
+     .line 1157
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #calls: Landroid/widget/ActivityChooserModel;->loadActivitiesLocked()V
      invoke-static {v0}, Landroid/widget/ActivityChooserModel;->access$1300(Landroid/widget/ActivityChooserModel;)V
  
+     .line 1158
      monitor-exit v1
  
+     .line 1159
      return-void
  
+     .line 1158
      :catchall_0
      move-exception v0
  
***************
*** 120,125 ****
--- 132,138 ----
      .parameter "reason"
  
      .prologue
+     .line 1171
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #getter for: Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
***************
*** 129,149 ****
--- 142,167 ----
  
      monitor-enter v1
  
+     .line 1172
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #calls: Landroid/widget/ActivityChooserModel;->pruneHistoricalRecordsForPackageLocked(Ljava/lang/String;)V
      invoke-static {v0, p1}, Landroid/widget/ActivityChooserModel;->access$1400(Landroid/widget/ActivityChooserModel;Ljava/lang/String;)V
  
+     .line 1173
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #calls: Landroid/widget/ActivityChooserModel;->loadActivitiesLocked()V
      invoke-static {v0}, Landroid/widget/ActivityChooserModel;->access$1300(Landroid/widget/ActivityChooserModel;)V
  
+     .line 1174
      monitor-exit v1
  
+     .line 1175
      return-void
  
+     .line 1174
      :catchall_0
      move-exception v0
  
***************
*** 160,165 ****
--- 178,184 ----
      .parameter "uid"
  
      .prologue
+     .line 1163
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #getter for: Landroid/widget/ActivityChooserModel;->mInstanceLock:Ljava/lang/Object;
***************
*** 169,189 ****
--- 188,213 ----
  
      monitor-enter v1
  
+     .line 1164
      :try_start_0
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #calls: Landroid/widget/ActivityChooserModel;->pruneHistoricalRecordsForPackageLocked(Ljava/lang/String;)V
      invoke-static {v0, p1}, Landroid/widget/ActivityChooserModel;->access$1400(Landroid/widget/ActivityChooserModel;Ljava/lang/String;)V
  
+     .line 1165
      iget-object v0, p0, Landroid/widget/ActivityChooserModel$DataModelPackageMonitor;->this$0:Landroid/widget/ActivityChooserModel;
  
      #calls: Landroid/widget/ActivityChooserModel;->loadActivitiesLocked()V
      invoke-static {v0}, Landroid/widget/ActivityChooserModel;->access$1300(Landroid/widget/ActivityChooserModel;)V
  
+     .line 1166
      monitor-exit v1
  
+     .line 1167
      return-void
  
+     .line 1166
      :catchall_0
      move-exception v0
  
