*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserModel$HistoricalRecord.smali	2012-03-23 19:35:58.596655394 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ActivityChooserModel$HistoricalRecord.smali.original	2012-03-23 19:34:44.929312072 -0500
***************
*** 30,43 ****
--- 30,48 ----
      .parameter "weight"
  
      .prologue
+     .line 801
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 802
      iput-object p1, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->activity:Landroid/content/ComponentName;
  
+     .line 803
      iput-wide p2, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->time:J
  
+     .line 804
      iput p4, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->weight:F
  
+     .line 805
      return-void
  .end method
  
***************
*** 48,59 ****
--- 53,66 ----
      .parameter "weight"
  
      .prologue
+     .line 791
      invoke-static {p1}, Landroid/content/ComponentName;->unflattenFromString(Ljava/lang/String;)Landroid/content/ComponentName;
  
      move-result-object v0
  
      invoke-direct {p0, v0, p2, p3, p4}, Landroid/widget/ActivityChooserModel$HistoricalRecord;-><init>(Landroid/content/ComponentName;JF)V
  
+     .line 792
      return-void
  .end method
  
***************
*** 68,86 ****
--- 75,98 ----
  
      const/4 v2, 0x0
  
+     .line 819
      if-ne p0, p1, :cond_1
  
+     .line 842
      :cond_0
      :goto_0
      return v1
  
+     .line 822
      :cond_1
      if-nez p1, :cond_2
  
      move v1, v2
  
+     .line 823
      goto :goto_0
  
+     .line 825
      :cond_2
      invoke-virtual {p0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
  
***************
*** 94,119 ****
--- 106,137 ----
  
      move v1, v2
  
+     .line 826
      goto :goto_0
  
      :cond_3
      move-object v0, p1
  
+     .line 828
      check-cast v0, Landroid/widget/ActivityChooserModel$HistoricalRecord;
  
+     .line 829
      .local v0, other:Landroid/widget/ActivityChooserModel$HistoricalRecord;
      iget-object v3, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->activity:Landroid/content/ComponentName;
  
      if-nez v3, :cond_4
  
+     .line 830
      iget-object v3, v0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->activity:Landroid/content/ComponentName;
  
      if-eqz v3, :cond_5
  
      move v1, v2
  
+     .line 831
      goto :goto_0
  
+     .line 833
      :cond_4
      iget-object v3, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->activity:Landroid/content/ComponentName;
  
***************
*** 127,134 ****
--- 145,154 ----
  
      move v1, v2
  
+     .line 834
      goto :goto_0
  
+     .line 836
      :cond_5
      iget-wide v3, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->time:J
  
***************
*** 140,147 ****
--- 160,169 ----
  
      move v1, v2
  
+     .line 837
      goto :goto_0
  
+     .line 839
      :cond_6
      iget v3, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->weight:F
  
***************
*** 159,164 ****
--- 181,187 ----
  
      move v1, v2
  
+     .line 840
      goto :goto_0
  .end method
  
***************
*** 166,176 ****
--- 189,202 ----
      .locals 8
  
      .prologue
+     .line 809
      const/16 v0, 0x1f
  
+     .line 810
      .local v0, prime:I
      const/4 v1, 0x1
  
+     .line 811
      .local v1, result:I
      iget-object v2, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->activity:Landroid/content/ComponentName;
  
***************
*** 181,186 ****
--- 207,213 ----
      :goto_0
      add-int/lit8 v1, v2, 0x1f
  
+     .line 812
      mul-int/lit8 v2, v1, 0x1f
  
      iget-wide v3, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->time:J
***************
*** 197,202 ****
--- 224,230 ----
  
      add-int v1, v2, v3
  
+     .line 813
      mul-int/lit8 v2, v1, 0x1f
  
      iget v3, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->weight:F
***************
*** 207,214 ****
--- 235,244 ----
  
      add-int v1, v2, v3
  
+     .line 814
      return v1
  
+     .line 811
      :cond_0
      iget-object v2, p0, Landroid/widget/ActivityChooserModel$HistoricalRecord;->activity:Landroid/content/ComponentName;
  
***************
*** 223,237 ****
--- 253,270 ----
      .locals 5
  
      .prologue
+     .line 847
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 848
      .local v0, builder:Ljava/lang/StringBuilder;
      const-string v1, "["
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 849
      const-string v1, "; activity:"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 242,247 ****
--- 275,281 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
  
+     .line 850
      const-string v1, "; time:"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 252,257 ****
--- 286,292 ----
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 851
      const-string v1, "; weight:"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 268,277 ****
--- 303,314 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
  
+     .line 852
      const-string v1, "]"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 853
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
