*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AdapterViewAnimator$CheckForTap.smali	2012-03-23 19:35:57.514635663 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AdapterViewAnimator$CheckForTap.smali.original	2012-03-23 19:34:44.920311908 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 593
      iput-object p1, p0, Landroid/widget/AdapterViewAnimator$CheckForTap;->this$0:Landroid/widget/AdapterViewAnimator;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .locals 3
  
      .prologue
+     .line 595
      iget-object v1, p0, Landroid/widget/AdapterViewAnimator$CheckForTap;->this$0:Landroid/widget/AdapterViewAnimator;
  
      #getter for: Landroid/widget/AdapterViewAnimator;->mTouchMode:I
***************
*** 51,67 ****
--- 53,72 ----
  
      if-ne v1, v2, :cond_0
  
+     .line 596
      iget-object v1, p0, Landroid/widget/AdapterViewAnimator$CheckForTap;->this$0:Landroid/widget/AdapterViewAnimator;
  
      invoke-virtual {v1}, Landroid/widget/AdapterViewAnimator;->getCurrentView()Landroid/view/View;
  
      move-result-object v0
  
+     .line 597
      .local v0, v:Landroid/view/View;
      iget-object v1, p0, Landroid/widget/AdapterViewAnimator$CheckForTap;->this$0:Landroid/widget/AdapterViewAnimator;
  
      invoke-virtual {v1, v0}, Landroid/widget/AdapterViewAnimator;->showTapFeedback(Landroid/view/View;)V
  
+     .line 599
      .end local v0           #v:Landroid/view/View;
      :cond_0
      return-void
