*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AdapterViewAnimator$1$1.smali	2012-03-23 19:35:51.335522987 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AdapterViewAnimator$1$1.smali.original	2012-03-23 19:34:44.871311014 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 636
      iput-object p1, p0, Landroid/widget/AdapterViewAnimator$1$1;->this$1:Landroid/widget/AdapterViewAnimator$1;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,51 ****
--- 41,54 ----
      .locals 5
  
      .prologue
+     .line 638
      iget-object v0, p0, Landroid/widget/AdapterViewAnimator$1$1;->this$1:Landroid/widget/AdapterViewAnimator$1;
  
      iget-object v0, v0, Landroid/widget/AdapterViewAnimator$1;->val$viewData:Landroid/widget/AdapterViewAnimator$ViewAndMetaData;
  
      if-eqz v0, :cond_0
  
+     .line 639
      iget-object v0, p0, Landroid/widget/AdapterViewAnimator$1$1;->this$1:Landroid/widget/AdapterViewAnimator$1;
  
      iget-object v0, v0, Landroid/widget/AdapterViewAnimator$1;->this$0:Landroid/widget/AdapterViewAnimator;
***************
*** 68,76 ****
--- 71,81 ----
  
      invoke-virtual {v0, v1, v2, v3, v4}, Landroid/widget/AdapterViewAnimator;->performItemClick(Landroid/view/View;IJ)Z
  
+     .line 644
      :goto_0
      return-void
  
+     .line 642
      :cond_0
      iget-object v0, p0, Landroid/widget/AdapterViewAnimator$1$1;->this$1:Landroid/widget/AdapterViewAnimator$1;
  
