*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AdapterView$SelectionNotifier.smali	2012-03-23 19:36:00.572691427 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AdapterView$SelectionNotifier.smali.original	2012-03-23 19:34:44.940312272 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 838
      .local p0, this:Landroid/widget/AdapterView$SelectionNotifier;,"Landroid/widget/AdapterView<TT;>.SelectionNotifier;"
      iput-object p1, p0, Landroid/widget/AdapterView$SelectionNotifier;->this$0:Landroid/widget/AdapterView;
  
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "x1"
  
      .prologue
+     .line 838
      .local p0, this:Landroid/widget/AdapterView$SelectionNotifier;,"Landroid/widget/AdapterView<TT;>.SelectionNotifier;"
      invoke-direct {p0, p1}, Landroid/widget/AdapterView$SelectionNotifier;-><init>(Landroid/widget/AdapterView;)V
  
***************
*** 53,58 ****
--- 55,61 ----
      .locals 1
  
      .prologue
+     .line 840
      .local p0, this:Landroid/widget/AdapterView$SelectionNotifier;,"Landroid/widget/AdapterView<TT;>.SelectionNotifier;"
      iget-object v0, p0, Landroid/widget/AdapterView$SelectionNotifier;->this$0:Landroid/widget/AdapterView;
  
***************
*** 60,65 ****
--- 63,69 ----
  
      if-eqz v0, :cond_1
  
+     .line 844
      iget-object v0, p0, Landroid/widget/AdapterView$SelectionNotifier;->this$0:Landroid/widget/AdapterView;
  
      invoke-virtual {v0}, Landroid/widget/AdapterView;->getAdapter()Landroid/widget/Adapter;
***************
*** 68,81 ****
--- 72,88 ----
  
      if-eqz v0, :cond_0
  
+     .line 845
      iget-object v0, p0, Landroid/widget/AdapterView$SelectionNotifier;->this$0:Landroid/widget/AdapterView;
  
      invoke-virtual {v0, p0}, Landroid/widget/AdapterView;->post(Ljava/lang/Runnable;)Z
  
+     .line 850
      :cond_0
      :goto_0
      return-void
  
+     .line 848
      :cond_1
      iget-object v0, p0, Landroid/widget/AdapterView$SelectionNotifier;->this$0:Landroid/widget/AdapterView;
  
