*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AdapterViewFlipper.smali	2012-03-23 19:35:49.256485075 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AdapterViewFlipper.smali.original	2012-03-23 19:34:44.854310704 -0500
***************
*** 48,85 ****
--- 48,97 ----
  
      const/4 v1, 0x0
  
+     .line 58
      invoke-direct {p0, p1}, Landroid/widget/AdapterViewAnimator;-><init>(Landroid/content/Context;)V
  
+     .line 48
      const/16 v0, 0x2710
  
      iput v0, p0, Landroid/widget/AdapterViewFlipper;->mFlipInterval:I
  
+     .line 49
      iput-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mAutoStart:Z
  
+     .line 51
      iput-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mRunning:Z
  
+     .line 52
      iput-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mStarted:Z
  
+     .line 53
      iput-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mVisible:Z
  
+     .line 54
      iput-boolean v2, p0, Landroid/widget/AdapterViewFlipper;->mUserPresent:Z
  
+     .line 55
      iput-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mAdvancedByHost:Z
  
+     .line 77
      new-instance v0, Landroid/widget/AdapterViewFlipper$1;
  
      invoke-direct {v0, p0}, Landroid/widget/AdapterViewFlipper$1;-><init>(Landroid/widget/AdapterViewFlipper;)V
  
      iput-object v0, p0, Landroid/widget/AdapterViewFlipper;->mReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 247
      iput v2, p0, Landroid/widget/AdapterViewFlipper;->FLIP_MSG:I
  
+     .line 249
      new-instance v0, Landroid/widget/AdapterViewFlipper$2;
  
      invoke-direct {v0, p0}, Landroid/widget/AdapterViewFlipper$2;-><init>(Landroid/widget/AdapterViewFlipper;)V
  
      iput-object v0, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
+     .line 59
      return-void
  .end method
  
***************
*** 95,136 ****
--- 107,161 ----
  
      const/4 v2, 0x0
  
+     .line 62
      invoke-direct {p0, p1, p2}, Landroid/widget/AdapterViewAnimator;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 48
      iput v4, p0, Landroid/widget/AdapterViewFlipper;->mFlipInterval:I
  
+     .line 49
      iput-boolean v2, p0, Landroid/widget/AdapterViewFlipper;->mAutoStart:Z
  
+     .line 51
      iput-boolean v2, p0, Landroid/widget/AdapterViewFlipper;->mRunning:Z
  
+     .line 52
      iput-boolean v2, p0, Landroid/widget/AdapterViewFlipper;->mStarted:Z
  
+     .line 53
      iput-boolean v2, p0, Landroid/widget/AdapterViewFlipper;->mVisible:Z
  
+     .line 54
      iput-boolean v3, p0, Landroid/widget/AdapterViewFlipper;->mUserPresent:Z
  
+     .line 55
      iput-boolean v2, p0, Landroid/widget/AdapterViewFlipper;->mAdvancedByHost:Z
  
+     .line 77
      new-instance v1, Landroid/widget/AdapterViewFlipper$1;
  
      invoke-direct {v1, p0}, Landroid/widget/AdapterViewFlipper$1;-><init>(Landroid/widget/AdapterViewFlipper;)V
  
      iput-object v1, p0, Landroid/widget/AdapterViewFlipper;->mReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 247
      iput v3, p0, Landroid/widget/AdapterViewFlipper;->FLIP_MSG:I
  
+     .line 249
      new-instance v1, Landroid/widget/AdapterViewFlipper$2;
  
      invoke-direct {v1, p0}, Landroid/widget/AdapterViewFlipper$2;-><init>(Landroid/widget/AdapterViewFlipper;)V
  
      iput-object v1, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
+     .line 64
      sget-object v1, Lcom/android/internal/R$styleable;->AdapterViewFlipper:[I
  
      invoke-virtual {p1, p2, v1}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[I)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 66
      .local v0, a:Landroid/content/res/TypedArray;
      invoke-virtual {v0, v2, v4}, Landroid/content/res/TypedArray;->getInt(II)I
  
***************
*** 138,153 ****
--- 163,182 ----
  
      iput v1, p0, Landroid/widget/AdapterViewFlipper;->mFlipInterval:I
  
+     .line 68
      invoke-virtual {v0, v3, v2}, Landroid/content/res/TypedArray;->getBoolean(IZ)Z
  
      move-result v1
  
      iput-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mAutoStart:Z
  
+     .line 72
      iput-boolean v3, p0, Landroid/widget/AdapterViewFlipper;->mLoopViews:Z
  
+     .line 74
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 75
      return-void
  .end method
  
***************
*** 157,162 ****
--- 186,192 ----
      .parameter "x1"
  
      .prologue
+     .line 42
      iput-boolean p1, p0, Landroid/widget/AdapterViewFlipper;->mUserPresent:Z
  
      return p1
***************
*** 167,172 ****
--- 197,203 ----
      .parameter "x0"
  
      .prologue
+     .line 42
      invoke-direct {p0}, Landroid/widget/AdapterViewFlipper;->updateRunning()V
  
      return-void
***************
*** 178,183 ****
--- 209,215 ----
      .parameter "x1"
  
      .prologue
+     .line 42
      invoke-direct {p0, p1}, Landroid/widget/AdapterViewFlipper;->updateRunning(Z)V
  
      return-void
***************
*** 188,193 ****
--- 220,226 ----
      .parameter "x0"
  
      .prologue
+     .line 42
      iget-boolean v0, p0, Landroid/widget/AdapterViewFlipper;->mRunning:Z
  
      return v0
***************
*** 197,206 ****
--- 230,241 ----
      .locals 1
  
      .prologue
+     .line 194
      const/4 v0, 0x1
  
      invoke-direct {p0, v0}, Landroid/widget/AdapterViewFlipper;->updateRunning(Z)V
  
+     .line 195
      return-void
  .end method
  
***************
*** 211,216 ****
--- 246,252 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 206
      iget-boolean v3, p0, Landroid/widget/AdapterViewFlipper;->mAdvancedByHost:Z
  
      if-nez v3, :cond_1
***************
*** 233,256 ****
--- 269,297 ----
  
      move v1, v2
  
+     .line 208
      .local v1, running:Z
      :goto_0
      iget-boolean v3, p0, Landroid/widget/AdapterViewFlipper;->mRunning:Z
  
      if-eq v1, v3, :cond_0
  
+     .line 209
      if-eqz v1, :cond_2
  
+     .line 210
      iget v3, p0, Landroid/widget/AdapterViewFlipper;->mWhichChild:I
  
      invoke-virtual {p0, v3, p1}, Landroid/widget/AdapterViewFlipper;->showOnly(IZ)V
  
+     .line 211
      iget-object v3, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v3, v2}, Landroid/os/Handler;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 212
      .local v0, msg:Landroid/os/Message;
      iget-object v2, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
***************
*** 260,278 ****
--- 301,323 ----
  
      invoke-virtual {v2, v0, v3, v4}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 216
      .end local v0           #msg:Landroid/os/Message;
      :goto_1
      iput-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mRunning:Z
  
+     .line 222
      :cond_0
      return-void
  
+     .line 206
      .end local v1           #running:Z
      :cond_1
      const/4 v1, 0x0
  
      goto :goto_0
  
+     .line 214
      .restart local v1       #running:Z
      :cond_2
      iget-object v3, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
***************
*** 288,301 ****
--- 333,349 ----
      .locals 1
  
      .prologue
+     .line 268
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/AdapterViewFlipper;->mAdvancedByHost:Z
  
+     .line 269
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/widget/AdapterViewFlipper;->updateRunning(Z)V
  
+     .line 270
      return-void
  .end method
  
***************
*** 303,308 ****
--- 351,357 ----
      .locals 1
  
      .prologue
+     .line 244
      iget-boolean v0, p0, Landroid/widget/AdapterViewFlipper;->mAutoStart:Z
  
      return v0
***************
*** 312,317 ****
--- 361,367 ----
      .locals 1
  
      .prologue
+     .line 228
      iget-boolean v0, p0, Landroid/widget/AdapterViewFlipper;->mStarted:Z
  
      return v0
***************
*** 321,341 ****
--- 371,396 ----
      .locals 3
  
      .prologue
+     .line 93
      invoke-super {p0}, Landroid/widget/AdapterViewAnimator;->onAttachedToWindow()V
  
+     .line 96
      new-instance v0, Landroid/content/IntentFilter;
  
      invoke-direct {v0}, Landroid/content/IntentFilter;-><init>()V
  
+     .line 97
      .local v0, filter:Landroid/content/IntentFilter;
      const-string v1, "android.intent.action.SCREEN_OFF"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 98
      const-string v1, "android.intent.action.USER_PRESENT"
  
      invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
  
+     .line 99
      invoke-virtual {p0}, Landroid/widget/AdapterViewFlipper;->getContext()Landroid/content/Context;
  
      move-result-object v1
***************
*** 344,355 ****
--- 399,413 ----
  
      invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 101
      iget-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mAutoStart:Z
  
      if-eqz v1, :cond_0
  
+     .line 103
      invoke-virtual {p0}, Landroid/widget/AdapterViewFlipper;->startFlipping()V
  
+     .line 105
      :cond_0
      return-void
  .end method
***************
*** 358,369 ****
--- 416,430 ----
      .locals 2
  
      .prologue
+     .line 109
      invoke-super {p0}, Landroid/widget/AdapterViewAnimator;->onDetachedFromWindow()V
  
+     .line 110
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/AdapterViewFlipper;->mVisible:Z
  
+     .line 112
      invoke-virtual {p0}, Landroid/widget/AdapterViewFlipper;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 372,379 ****
--- 433,442 ----
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  
+     .line 113
      invoke-direct {p0}, Landroid/widget/AdapterViewFlipper;->updateRunning()V
  
+     .line 114
      return-void
  .end method
  
***************
*** 384,391 ****
--- 447,456 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 118
      invoke-super {p0, p1}, Landroid/widget/AdapterViewAnimator;->onWindowVisibilityChanged(I)V
  
+     .line 119
      if-nez p1, :cond_0
  
      const/4 v0, 0x1
***************
*** 393,405 ****
--- 458,473 ----
      :goto_0
      iput-boolean v0, p0, Landroid/widget/AdapterViewFlipper;->mVisible:Z
  
+     .line 120
      invoke-direct {p0, v1}, Landroid/widget/AdapterViewFlipper;->updateRunning(Z)V
  
+     .line 121
      return-void
  
      :cond_0
      move v0, v1
  
+     .line 119
      goto :goto_0
  .end method
  
***************
*** 408,417 ****
--- 476,488 ----
      .parameter "adapter"
  
      .prologue
+     .line 125
      invoke-super {p0, p1}, Landroid/widget/AdapterViewAnimator;->setAdapter(Landroid/widget/Adapter;)V
  
+     .line 126
      invoke-direct {p0}, Landroid/widget/AdapterViewFlipper;->updateRunning()V
  
+     .line 127
      return-void
  .end method
  
***************
*** 420,427 ****
--- 491,500 ----
      .parameter "autoStart"
  
      .prologue
+     .line 236
      iput-boolean p1, p0, Landroid/widget/AdapterViewFlipper;->mAutoStart:Z
  
+     .line 237
      return-void
  .end method
  
***************
*** 430,437 ****
--- 503,512 ----
      .parameter "milliseconds"
  
      .prologue
+     .line 136
      iput p1, p0, Landroid/widget/AdapterViewFlipper;->mFlipInterval:I
  
+     .line 137
      return-void
  .end method
  
***************
*** 443,462 ****
--- 518,541 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 163
      iget-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mRunning:Z
  
      if-eqz v1, :cond_0
  
+     .line 164
      iget-object v1, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v2}, Landroid/os/Handler;->removeMessages(I)V
  
+     .line 165
      iget-object v1, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v2}, Landroid/os/Handler;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 166
      .local v0, msg:Landroid/os/Message;
      iget-object v1, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
***************
*** 466,475 ****
--- 545,556 ----
  
      invoke-virtual {v1, v0, v2, v3}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 168
      .end local v0           #msg:Landroid/os/Message;
      :cond_0
      invoke-super {p0}, Landroid/widget/AdapterViewAnimator;->showNext()V
  
+     .line 169
      return-void
  .end method
  
***************
*** 481,500 ****
--- 562,585 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 179
      iget-boolean v1, p0, Landroid/widget/AdapterViewFlipper;->mRunning:Z
  
      if-eqz v1, :cond_0
  
+     .line 180
      iget-object v1, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v2}, Landroid/os/Handler;->removeMessages(I)V
  
+     .line 181
      iget-object v1, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v2}, Landroid/os/Handler;->obtainMessage(I)Landroid/os/Message;
  
      move-result-object v0
  
+     .line 182
      .local v0, msg:Landroid/os/Message;
      iget-object v1, p0, Landroid/widget/AdapterViewFlipper;->mHandler:Landroid/os/Handler;
  
***************
*** 504,513 ****
--- 589,600 ----
  
      invoke-virtual {v1, v0, v2, v3}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 184
      .end local v0           #msg:Landroid/os/Message;
      :cond_0
      invoke-super {p0}, Landroid/widget/AdapterViewAnimator;->showPrevious()V
  
+     .line 185
      return-void
  .end method
  
***************
*** 515,526 ****
--- 602,616 ----
      .locals 1
  
      .prologue
+     .line 143
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/AdapterViewFlipper;->mStarted:Z
  
+     .line 144
      invoke-direct {p0}, Landroid/widget/AdapterViewFlipper;->updateRunning()V
  
+     .line 145
      return-void
  .end method
  
***************
*** 528,538 ****
--- 618,631 ----
      .locals 1
  
      .prologue
+     .line 151
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/AdapterViewFlipper;->mStarted:Z
  
+     .line 152
      invoke-direct {p0}, Landroid/widget/AdapterViewFlipper;->updateRunning()V
  
+     .line 153
      return-void
  .end method
