*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AppSecurityPermissions$State.smali	2012-03-23 19:35:48.268467060 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AppSecurityPermissions$State.smali.original	2012-03-23 19:34:44.847310576 -0500
***************
*** 48,53 ****
--- 48,54 ----
  
      const/4 v2, 0x0
  
+     .line 58
      new-instance v0, Landroid/widget/AppSecurityPermissions$State;
  
      const-string v1, "NO_PERMS"
***************
*** 56,61 ****
--- 57,63 ----
  
      sput-object v0, Landroid/widget/AppSecurityPermissions$State;->NO_PERMS:Landroid/widget/AppSecurityPermissions$State;
  
+     .line 59
      new-instance v0, Landroid/widget/AppSecurityPermissions$State;
  
      const-string v1, "DANGEROUS_ONLY"
***************
*** 64,69 ****
--- 66,72 ----
  
      sput-object v0, Landroid/widget/AppSecurityPermissions$State;->DANGEROUS_ONLY:Landroid/widget/AppSecurityPermissions$State;
  
+     .line 60
      new-instance v0, Landroid/widget/AppSecurityPermissions$State;
  
      const-string v1, "NORMAL_ONLY"
***************
*** 72,77 ****
--- 75,81 ----
  
      sput-object v0, Landroid/widget/AppSecurityPermissions$State;->NORMAL_ONLY:Landroid/widget/AppSecurityPermissions$State;
  
+     .line 61
      new-instance v0, Landroid/widget/AppSecurityPermissions$State;
  
      const-string v1, "BOTH"
***************
*** 80,85 ****
--- 84,90 ----
  
      sput-object v0, Landroid/widget/AppSecurityPermissions$State;->BOTH:Landroid/widget/AppSecurityPermissions$State;
  
+     .line 57
      const/4 v0, 0x4
  
      new-array v0, v0, [Landroid/widget/AppSecurityPermissions$State;
***************
*** 116,121 ****
--- 121,127 ----
      .end annotation
  
      .prologue
+     .line 57
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 126,131 ****
--- 132,138 ----
      .parameter "name"
  
      .prologue
+     .line 57
      const-class v0, Landroid/widget/AppSecurityPermissions$State;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 141,146 ****
--- 148,154 ----
      .locals 1
  
      .prologue
+     .line 57
      sget-object v0, Landroid/widget/AppSecurityPermissions$State;->$VALUES:[Landroid/widget/AppSecurityPermissions$State;
  
      invoke-virtual {v0}, [Landroid/widget/AppSecurityPermissions$State;->clone()Ljava/lang/Object;
