*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AutoCompleteTextView$PopupDataSetObserver$1.smali	2012-03-23 19:36:00.644692739 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/AutoCompleteTextView$PopupDataSetObserver$1.smali.original	2012-03-23 19:34:44.940312272 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1214
      iput-object p1, p0, Landroid/widget/AutoCompleteTextView$PopupDataSetObserver$1;->this$1:Landroid/widget/AutoCompleteTextView$PopupDataSetObserver;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .locals 3
  
      .prologue
+     .line 1216
      iget-object v1, p0, Landroid/widget/AutoCompleteTextView$PopupDataSetObserver$1;->this$1:Landroid/widget/AutoCompleteTextView$PopupDataSetObserver;
  
      iget-object v1, v1, Landroid/widget/AutoCompleteTextView$PopupDataSetObserver;->this$0:Landroid/widget/AutoCompleteTextView;
***************
*** 49,57 ****
--- 51,61 ----
  
      move-result-object v0
  
+     .line 1217
      .local v0, adapter:Landroid/widget/ListAdapter;
      if-eqz v0, :cond_0
  
+     .line 1220
      iget-object v1, p0, Landroid/widget/AutoCompleteTextView$PopupDataSetObserver$1;->this$1:Landroid/widget/AutoCompleteTextView$PopupDataSetObserver;
  
      iget-object v1, v1, Landroid/widget/AutoCompleteTextView$PopupDataSetObserver;->this$0:Landroid/widget/AutoCompleteTextView;
***************
*** 63,68 ****
--- 67,73 ----
      #calls: Landroid/widget/AutoCompleteTextView;->updateDropDownForFilter(I)V
      invoke-static {v1, v2}, Landroid/widget/AutoCompleteTextView;->access$800(Landroid/widget/AutoCompleteTextView;I)V
  
+     .line 1222
      :cond_0
      return-void
  .end method
