*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/BaseExpandableListAdapter.smali	2012-03-23 19:35:56.130610425 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/BaseExpandableListAdapter.smali.original	2012-03-23 19:34:44.910311726 -0500
***************
*** 16,23 ****
--- 16,25 ----
      .locals 1
  
      .prologue
+     .line 33
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 35
      new-instance v0, Landroid/database/DataSetObservable;
  
      invoke-direct {v0}, Landroid/database/DataSetObservable;-><init>()V
***************
*** 33,38 ****
--- 35,41 ----
      .locals 1
  
      .prologue
+     .line 60
      const/4 v0, 0x1
  
      return v0
***************
*** 44,49 ****
--- 47,53 ----
      .parameter "childPosition"
  
      .prologue
+     .line 112
      const/4 v0, 0x0
  
      return v0
***************
*** 53,58 ****
--- 57,63 ----
      .locals 1
  
      .prologue
+     .line 120
      const/4 v0, 0x1
  
      return v0
***************
*** 64,69 ****
--- 69,75 ----
      .parameter "childId"
  
      .prologue
+     .line 81
      const-wide/high16 v0, -0x8000
  
      const-wide/32 v2, 0x7fffffff
***************
*** 90,95 ****
--- 96,102 ----
      .parameter "groupId"
  
      .prologue
+     .line 96
      const-wide/32 v0, 0x7fffffff
  
      and-long/2addr v0, p1
***************
*** 106,111 ****
--- 113,119 ----
      .parameter "groupPosition"
  
      .prologue
+     .line 128
      const/4 v0, 0x0
  
      return v0
***************
*** 115,120 ****
--- 123,129 ----
      .locals 1
  
      .prologue
+     .line 136
      const/4 v0, 0x1
  
      return v0
***************
*** 124,129 ****
--- 133,139 ----
      .locals 1
  
      .prologue
+     .line 103
      invoke-virtual {p0}, Landroid/widget/BaseExpandableListAdapter;->getGroupCount()I
  
      move-result v0
***************
*** 145,154 ****
--- 155,166 ----
      .locals 1
  
      .prologue
+     .line 56
      iget-object v0, p0, Landroid/widget/BaseExpandableListAdapter;->mDataSetObservable:Landroid/database/DataSetObservable;
  
      invoke-virtual {v0}, Landroid/database/DataSetObservable;->notifyChanged()V
  
+     .line 57
      return-void
  .end method
  
***************
*** 156,165 ****
--- 168,179 ----
      .locals 1
  
      .prologue
+     .line 49
      iget-object v0, p0, Landroid/widget/BaseExpandableListAdapter;->mDataSetObservable:Landroid/database/DataSetObservable;
  
      invoke-virtual {v0}, Landroid/database/DataSetObservable;->notifyInvalidated()V
  
+     .line 50
      return-void
  .end method
  
***************
*** 168,173 ****
--- 182,188 ----
      .parameter "groupPosition"
  
      .prologue
+     .line 64
      return-void
  .end method
  
***************
*** 176,181 ****
--- 191,197 ----
      .parameter "groupPosition"
  
      .prologue
+     .line 67
      return-void
  .end method
  
***************
*** 184,193 ****
--- 200,211 ----
      .parameter "observer"
  
      .prologue
+     .line 38
      iget-object v0, p0, Landroid/widget/BaseExpandableListAdapter;->mDataSetObservable:Landroid/database/DataSetObservable;
  
      invoke-virtual {v0, p1}, Landroid/database/DataSetObservable;->registerObserver(Ljava/lang/Object;)V
  
+     .line 39
      return-void
  .end method
  
***************
*** 196,204 ****
--- 214,224 ----
      .parameter "observer"
  
      .prologue
+     .line 42
      iget-object v0, p0, Landroid/widget/BaseExpandableListAdapter;->mDataSetObservable:Landroid/database/DataSetObservable;
  
      invoke-virtual {v0, p1}, Landroid/database/DataSetObservable;->unregisterObserver(Ljava/lang/Object;)V
  
+     .line 43
      return-void
  .end method
