*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/BaseAdapter.smali	2012-03-23 19:35:52.313540823 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/BaseAdapter.smali.original	2012-03-23 19:34:44.879311160 -0500
***************
*** 16,23 ****
--- 16,25 ----
      .locals 1
  
      .prologue
+     .line 30
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 31
      new-instance v0, Landroid/database/DataSetObservable;
  
      invoke-direct {v0}, Landroid/database/DataSetObservable;-><init>()V
***************
*** 33,38 ****
--- 35,41 ----
      .locals 1
  
      .prologue
+     .line 63
      const/4 v0, 0x1
  
      return v0
***************
*** 45,50 ****
--- 48,54 ----
      .parameter "parent"
  
      .prologue
+     .line 71
      invoke-virtual {p0, p1, p2, p3}, Landroid/widget/BaseAdapter;->getView(ILandroid/view/View;Landroid/view/ViewGroup;)Landroid/view/View;
  
      move-result-object v0
***************
*** 57,62 ****
--- 61,67 ----
      .parameter "position"
  
      .prologue
+     .line 75
      const/4 v0, 0x0
  
      return v0
***************
*** 66,71 ****
--- 71,77 ----
      .locals 1
  
      .prologue
+     .line 79
      const/4 v0, 0x1
  
      return v0
***************
*** 75,80 ****
--- 81,87 ----
      .locals 1
  
      .prologue
+     .line 34
      const/4 v0, 0x0
  
      return v0
***************
*** 84,89 ****
--- 91,97 ----
      .locals 1
  
      .prologue
+     .line 83
      invoke-virtual {p0}, Landroid/widget/BaseAdapter;->getCount()I
  
      move-result v0
***************
*** 106,111 ****
--- 114,120 ----
      .parameter "position"
  
      .prologue
+     .line 67
      const/4 v0, 0x1
  
      return v0
***************
*** 115,124 ****
--- 124,135 ----
      .locals 1
  
      .prologue
+     .line 50
      iget-object v0, p0, Landroid/widget/BaseAdapter;->mDataSetObservable:Landroid/database/DataSetObservable;
  
      invoke-virtual {v0}, Landroid/database/DataSetObservable;->notifyChanged()V
  
+     .line 51
      return-void
  .end method
  
***************
*** 126,135 ****
--- 137,148 ----
      .locals 1
  
      .prologue
+     .line 59
      iget-object v0, p0, Landroid/widget/BaseAdapter;->mDataSetObservable:Landroid/database/DataSetObservable;
  
      invoke-virtual {v0}, Landroid/database/DataSetObservable;->notifyInvalidated()V
  
+     .line 60
      return-void
  .end method
  
***************
*** 138,147 ****
--- 151,162 ----
      .parameter "observer"
  
      .prologue
+     .line 38
      iget-object v0, p0, Landroid/widget/BaseAdapter;->mDataSetObservable:Landroid/database/DataSetObservable;
  
      invoke-virtual {v0, p1}, Landroid/database/DataSetObservable;->registerObserver(Ljava/lang/Object;)V
  
+     .line 39
      return-void
  .end method
  
***************
*** 150,158 ****
--- 165,175 ----
      .parameter "observer"
  
      .prologue
+     .line 42
      iget-object v0, p0, Landroid/widget/BaseAdapter;->mDataSetObservable:Landroid/database/DataSetObservable;
  
      invoke-virtual {v0, p1}, Landroid/database/DataSetObservable;->unregisterObserver(Ljava/lang/Object;)V
  
+     .line 43
      return-void
  .end method
