*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/CalendarView$WeeksAdapter.smali	2012-03-23 19:35:48.839477473 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/CalendarView$WeeksAdapter.smali.original	2012-03-23 19:34:44.851310650 -0500
***************
*** 44,62 ****
--- 44,66 ----
      .parameter "context"
  
      .prologue
+     .line 1033
      iput-object p1, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
  
      invoke-direct {p0}, Landroid/widget/BaseAdapter;-><init>()V
  
+     .line 1029
      invoke-static {}, Ljava/util/Calendar;->getInstance()Ljava/util/Calendar;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedDate:Ljava/util/Calendar;
  
+     .line 1034
      #setter for: Landroid/widget/CalendarView;->mContext:Landroid/content/Context;
      invoke-static {p1, p2}, Landroid/widget/CalendarView;->access$1102(Landroid/widget/CalendarView;Landroid/content/Context;)Landroid/content/Context;
  
+     .line 1035
      new-instance v0, Landroid/view/GestureDetector;
  
      #getter for: Landroid/widget/CalendarView;->mContext:Landroid/content/Context;
***************
*** 72,79 ****
--- 76,85 ----
  
      iput-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mGestureDetector:Landroid/view/GestureDetector;
  
+     .line 1036
      invoke-direct {p0}, Landroid/widget/CalendarView$WeeksAdapter;->init()V
  
+     .line 1037
      return-void
  .end method
  
***************
*** 82,87 ****
--- 88,94 ----
      .parameter "x0"
  
      .prologue
+     .line 1021
      iget-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedDate:Ljava/util/Calendar;
  
      return-object v0
***************
*** 92,97 ****
--- 99,105 ----
      .parameter "x0"
  
      .prologue
+     .line 1021
      invoke-direct {p0}, Landroid/widget/CalendarView$WeeksAdapter;->init()V
  
      return-void
***************
*** 103,108 ****
--- 111,117 ----
      .prologue
      const/4 v2, 0x7
  
+     .line 1043
      iget-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
  
      iget-object v1, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedDate:Ljava/util/Calendar;
***************
*** 114,119 ****
--- 123,129 ----
  
      iput v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedWeek:I
  
+     .line 1044
      iget-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
  
      iget-object v1, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
***************
*** 130,135 ****
--- 140,146 ----
  
      iput v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mTotalWeekCount:I
  
+     .line 1045
      iget-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
  
      #getter for: Landroid/widget/CalendarView;->mMinDate:Ljava/util/Calendar;
***************
*** 170,175 ****
--- 181,187 ----
  
      if-eq v0, v1, :cond_1
  
+     .line 1047
      :cond_0
      iget v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mTotalWeekCount:I
  
***************
*** 177,182 ****
--- 189,195 ----
  
      iput v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mTotalWeekCount:I
  
+     .line 1049
      :cond_1
      return-void
  .end method
***************
*** 186,198 ****
--- 199,214 ----
      .parameter "day"
  
      .prologue
+     .line 1149
      invoke-virtual {p0, p1}, Landroid/widget/CalendarView$WeeksAdapter;->setSelectedDay(Ljava/util/Calendar;)V
  
+     .line 1150
      iget-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
  
      #calls: Landroid/widget/CalendarView;->setMonthDisplayed(Ljava/util/Calendar;)V
      invoke-static {v0, p1}, Landroid/widget/CalendarView;->access$2000(Landroid/widget/CalendarView;Ljava/util/Calendar;)V
  
+     .line 1151
      return-void
  .end method
  
***************
*** 202,207 ****
--- 218,224 ----
      .locals 1
  
      .prologue
+     .line 1076
      iget v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mTotalWeekCount:I
  
      return v0
***************
*** 212,217 ****
--- 229,235 ----
      .parameter "position"
  
      .prologue
+     .line 1081
      const/4 v0, 0x0
  
      return-object v0
***************
*** 222,227 ****
--- 240,246 ----
      .parameter "position"
  
      .prologue
+     .line 1086
      int-to-long v0, p1
  
      return-wide v0
***************
*** 231,236 ****
--- 250,256 ----
      .locals 1
  
      .prologue
+     .line 1071
      iget-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedDate:Ljava/util/Calendar;
  
      return-object v0
***************
*** 245,259 ****
--- 265,283 ----
      .prologue
      const/4 v5, -0x2
  
+     .line 1091
      const/4 v2, 0x0
  
+     .line 1092
      .local v2, weekView:Landroid/widget/CalendarView$WeekView;
      if-eqz p2, :cond_0
  
      move-object v2, p2
  
+     .line 1093
      check-cast v2, Landroid/widget/CalendarView$WeekView;
  
+     .line 1104
      :goto_0
      iget v3, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedWeek:I
  
***************
*** 267,280 ****
--- 291,307 ----
  
      move-result v1
  
+     .line 1106
      .local v1, selectedWeekDay:I
      :goto_1
      iget v3, p0, Landroid/widget/CalendarView$WeeksAdapter;->mFocusedMonth:I
  
      invoke-virtual {v2, p1, v1, v3}, Landroid/widget/CalendarView$WeekView;->init(III)V
  
+     .line 1108
      return-object v2
  
+     .line 1095
      .end local v1           #selectedWeekDay:I
      :cond_0
      new-instance v2, Landroid/widget/CalendarView$WeekView;
***************
*** 291,312 ****
--- 318,344 ----
  
      invoke-direct {v2, v3, v4}, Landroid/widget/CalendarView$WeekView;-><init>(Landroid/widget/CalendarView;Landroid/content/Context;)V
  
+     .line 1096
      .restart local v2       #weekView:Landroid/widget/CalendarView$WeekView;
      new-instance v0, Landroid/widget/AbsListView$LayoutParams;
  
      invoke-direct {v0, v5, v5}, Landroid/widget/AbsListView$LayoutParams;-><init>(II)V
  
+     .line 1099
      .local v0, params:Landroid/widget/AbsListView$LayoutParams;
      invoke-virtual {v2, v0}, Landroid/widget/CalendarView$WeekView;->setLayoutParams(Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 1100
      const/4 v3, 0x1
  
      invoke-virtual {v2, v3}, Landroid/widget/CalendarView$WeekView;->setClickable(Z)V
  
+     .line 1101
      invoke-virtual {v2, p0}, Landroid/widget/CalendarView$WeekView;->setOnTouchListener(Landroid/view/View$OnTouchListener;)V
  
      goto :goto_0
  
+     .line 1104
      .end local v0           #params:Landroid/widget/AbsListView$LayoutParams;
      :cond_1
      const/4 v1, -0x1
***************
*** 322,327 ****
--- 354,360 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 1126
      iget-object v2, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
  
      #getter for: Landroid/widget/CalendarView;->mListView:Landroid/widget/ListView;
***************
*** 345,352 ****
--- 378,387 ----
  
      move-object v0, p1
  
+     .line 1127
      check-cast v0, Landroid/widget/CalendarView$WeekView;
  
+     .line 1129
      .local v0, weekView:Landroid/widget/CalendarView$WeekView;
      invoke-virtual {p2}, Landroid/view/MotionEvent;->getX()F
  
***************
*** 365,375 ****
--- 400,412 ----
  
      if-nez v2, :cond_1
  
+     .line 1140
      .end local v0           #weekView:Landroid/widget/CalendarView$WeekView;
      :cond_0
      :goto_0
      return v1
  
+     .line 1134
      .restart local v0       #weekView:Landroid/widget/CalendarView$WeekView;
      :cond_1
      iget-object v2, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
***************
*** 412,417 ****
--- 449,455 ----
  
      if-nez v2, :cond_0
  
+     .line 1137
      iget-object v2, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
  
      #getter for: Landroid/widget/CalendarView;->mTempDate:Ljava/util/Calendar;
***************
*** 423,428 ****
--- 461,467 ----
  
      goto :goto_0
  
+     .line 1140
      .end local v0           #weekView:Landroid/widget/CalendarView$WeekView;
      :cond_2
      const/4 v1, 0x0
***************
*** 435,450 ****
--- 474,493 ----
      .parameter "month"
  
      .prologue
+     .line 1117
      iget v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mFocusedMonth:I
  
      if-ne v0, p1, :cond_0
  
+     .line 1122
      :goto_0
      return-void
  
+     .line 1120
      :cond_0
      iput p1, p0, Landroid/widget/CalendarView$WeeksAdapter;->mFocusedMonth:I
  
+     .line 1121
      invoke-virtual {p0}, Landroid/widget/CalendarView$WeeksAdapter;->notifyDataSetChanged()V
  
      goto :goto_0
***************
*** 459,464 ****
--- 502,508 ----
  
      const/4 v2, 0x1
  
+     .line 1057
      invoke-virtual {p1, v3}, Ljava/util/Calendar;->get(I)I
  
      move-result v0
***************
*** 483,491 ****
--- 527,537 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 1065
      :goto_0
      return-void
  
+     .line 1061
      :cond_0
      iget-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedDate:Ljava/util/Calendar;
  
***************
*** 495,500 ****
--- 541,547 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 1062
      iget-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->this$0:Landroid/widget/CalendarView;
  
      iget-object v1, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedDate:Ljava/util/Calendar;
***************
*** 506,511 ****
--- 553,559 ----
  
      iput v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedWeek:I
  
+     .line 1063
      iget-object v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mSelectedDate:Ljava/util/Calendar;
  
      const/4 v1, 0x2
***************
*** 516,521 ****
--- 564,570 ----
  
      iput v0, p0, Landroid/widget/CalendarView$WeeksAdapter;->mFocusedMonth:I
  
+     .line 1064
      invoke-virtual {p0}, Landroid/widget/CalendarView$WeeksAdapter;->notifyDataSetChanged()V
  
      goto :goto_0
