*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/CursorTreeAdapter$MyCursorHelper.smali	2012-03-23 19:35:55.170592921 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/CursorTreeAdapter$MyCursorHelper.smali.original	2012-03-23 19:34:44.902311580 -0500
***************
*** 42,61 ****
--- 42,66 ----
      .parameter "cursor"
  
      .prologue
+     .line 408
      iput-object p1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->this$0:Landroid/widget/CursorTreeAdapter;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 409
      if-eqz p2, :cond_1
  
      const/4 v0, 0x1
  
+     .line 410
      .local v0, cursorPresent:Z
      :goto_0
      iput-object p2, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
+     .line 411
      iput-boolean v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataValid:Z
  
+     .line 412
      if-eqz v0, :cond_2
  
      const-string v1, "_id"
***************
*** 67,78 ****
--- 72,85 ----
      :goto_1
      iput v1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mRowIDColumn:I
  
+     .line 413
      new-instance v1, Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver;
  
      invoke-direct {v1, p0}, Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver;-><init>(Landroid/widget/CursorTreeAdapter$MyCursorHelper;)V
  
      iput-object v1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mContentObserver:Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver;
  
+     .line 414
      new-instance v1, Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyDataSetObserver;
  
      const/4 v2, 0x0
***************
*** 81,105 ****
--- 88,118 ----
  
      iput-object v1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataSetObserver:Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyDataSetObserver;
  
+     .line 415
      if-eqz v0, :cond_0
  
+     .line 416
      iget-object v1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mContentObserver:Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver;
  
      invoke-interface {p2, v1}, Landroid/database/Cursor;->registerContentObserver(Landroid/database/ContentObserver;)V
  
+     .line 417
      iget-object v1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataSetObserver:Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyDataSetObserver;
  
      invoke-interface {p2, v1}, Landroid/database/Cursor;->registerDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 419
      :cond_0
      return-void
  
+     .line 409
      .end local v0           #cursorPresent:Z
      :cond_1
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 412
      .restart local v0       #cursorPresent:Z
      :cond_2
      const/4 v1, -0x1
***************
*** 112,117 ****
--- 125,131 ----
      .parameter "x0"
  
      .prologue
+     .line 401
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
      return-object v0
***************
*** 123,128 ****
--- 137,143 ----
      .parameter "x1"
  
      .prologue
+     .line 401
      iput-boolean p1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataValid:Z
  
      return p1
***************
*** 136,163 ****
--- 151,186 ----
      .parameter "releaseCursors"
  
      .prologue
+     .line 454
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
      if-ne p1, v0, :cond_0
  
+     .line 471
      :goto_0
      return-void
  
+     .line 456
      :cond_0
      invoke-virtual {p0}, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->deactivate()V
  
+     .line 457
      iput-object p1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
+     .line 458
      if-eqz p1, :cond_1
  
+     .line 459
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mContentObserver:Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver;
  
      invoke-interface {p1, v0}, Landroid/database/Cursor;->registerContentObserver(Landroid/database/ContentObserver;)V
  
+     .line 460
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataSetObserver:Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyDataSetObserver;
  
      invoke-interface {p1, v0}, Landroid/database/Cursor;->registerDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 461
      const-string v0, "_id"
  
      invoke-interface {p1, v0}, Landroid/database/Cursor;->getColumnIndex(Ljava/lang/String;)I
***************
*** 166,190 ****
--- 189,218 ----
  
      iput v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mRowIDColumn:I
  
+     .line 462
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataValid:Z
  
+     .line 464
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->this$0:Landroid/widget/CursorTreeAdapter;
  
      invoke-virtual {v0, p2}, Landroid/widget/CursorTreeAdapter;->notifyDataSetChanged(Z)V
  
      goto :goto_0
  
+     .line 466
      :cond_1
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mRowIDColumn:I
  
+     .line 467
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataValid:Z
  
+     .line 469
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->this$0:Landroid/widget/CursorTreeAdapter;
  
      invoke-virtual {v0}, Landroid/widget/CursorTreeAdapter;->notifyDataSetInvalidated()V
***************
*** 196,208 ****
--- 224,239 ----
      .locals 2
  
      .prologue
+     .line 474
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
      if-nez v0, :cond_0
  
+     .line 482
      :goto_0
      return-void
  
+     .line 478
      :cond_0
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
***************
*** 210,225 ****
--- 241,259 ----
  
      invoke-interface {v0, v1}, Landroid/database/Cursor;->unregisterContentObserver(Landroid/database/ContentObserver;)V
  
+     .line 479
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
      iget-object v1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataSetObserver:Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyDataSetObserver;
  
      invoke-interface {v0, v1}, Landroid/database/Cursor;->unregisterDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 480
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
+     .line 481
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
***************
*** 231,236 ****
--- 265,271 ----
      .locals 1
  
      .prologue
+     .line 426
      iget-boolean v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataValid:Z
  
      if-eqz v0, :cond_0
***************
*** 239,250 ****
--- 274,287 ----
  
      if-eqz v0, :cond_0
  
+     .line 427
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v0}, Landroid/database/Cursor;->getCount()I
  
      move-result v0
  
+     .line 429
      :goto_0
      return v0
  
***************
*** 258,263 ****
--- 295,301 ----
      .locals 1
  
      .prologue
+     .line 422
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
      return-object v0
***************
*** 270,275 ****
--- 308,314 ----
      .prologue
      const-wide/16 v0, 0x0
  
+     .line 434
      iget-boolean v2, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataValid:Z
  
      if-eqz v2, :cond_0
***************
*** 278,283 ****
--- 317,323 ----
  
      if-eqz v2, :cond_0
  
+     .line 435
      iget-object v2, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
      invoke-interface {v2, p1}, Landroid/database/Cursor;->moveToPosition(I)Z
***************
*** 286,291 ****
--- 326,332 ----
  
      if-eqz v2, :cond_0
  
+     .line 436
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
      iget v1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mRowIDColumn:I
***************
*** 294,299 ****
--- 335,341 ----
  
      move-result-wide v0
  
+     .line 441
      :cond_0
      return-wide v0
  .end method
***************
*** 302,307 ****
--- 344,350 ----
      .locals 1
  
      .prologue
+     .line 485
      iget-boolean v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataValid:Z
  
      if-eqz v0, :cond_0
***************
*** 326,331 ****
--- 369,375 ----
      .parameter "position"
  
      .prologue
+     .line 446
      iget-boolean v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataValid:Z
  
      if-eqz v0, :cond_0
***************
*** 342,349 ****
--- 386,395 ----
  
      if-eqz v0, :cond_0
  
+     .line 447
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mCursor:Landroid/database/Cursor;
  
+     .line 449
      :goto_0
      return-object v0
  
