*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver.smali	2012-03-23 19:35:51.021517261 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver.smali.original	2012-03-23 19:34:44.868310960 -0500
***************
*** 24,31 ****
--- 24,33 ----
      .parameter
  
      .prologue
+     .line 489
      iput-object p1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver;->this$1:Landroid/widget/CursorTreeAdapter$MyCursorHelper;
  
+     .line 490
      iget-object v0, p1, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->this$0:Landroid/widget/CursorTreeAdapter;
  
      #getter for: Landroid/widget/CursorTreeAdapter;->mHandler:Landroid/os/Handler;
***************
*** 35,40 ****
--- 37,43 ----
  
      invoke-direct {p0, v0}, Landroid/database/ContentObserver;-><init>(Landroid/os/Handler;)V
  
+     .line 491
      return-void
  .end method
  
***************
*** 44,49 ****
--- 47,53 ----
      .locals 1
  
      .prologue
+     .line 495
      const/4 v0, 0x1
  
      return v0
***************
*** 54,59 ****
--- 58,64 ----
      .parameter "selfChange"
  
      .prologue
+     .line 500
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver;->this$1:Landroid/widget/CursorTreeAdapter$MyCursorHelper;
  
      iget-object v0, v0, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->this$0:Landroid/widget/CursorTreeAdapter;
***************
*** 74,79 ****
--- 79,85 ----
  
      if-eqz v0, :cond_0
  
+     .line 503
      iget-object v0, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver;->this$1:Landroid/widget/CursorTreeAdapter$MyCursorHelper;
  
      iget-object v1, p0, Landroid/widget/CursorTreeAdapter$MyCursorHelper$MyContentObserver;->this$1:Landroid/widget/CursorTreeAdapter$MyCursorHelper;
***************
*** 90,95 ****
--- 96,102 ----
      #setter for: Landroid/widget/CursorTreeAdapter$MyCursorHelper;->mDataValid:Z
      invoke-static {v0, v1}, Landroid/widget/CursorTreeAdapter$MyCursorHelper;->access$402(Landroid/widget/CursorTreeAdapter$MyCursorHelper;Z)Z
  
+     .line 505
      :cond_0
      return-void
  .end method
