*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/CursorFilter.smali	2012-03-23 19:35:46.872441602 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/CursorFilter.smali.original	2012-03-23 19:34:44.836310376 -0500
***************
*** 21,30 ****
--- 21,33 ----
      .parameter "client"
  
      .prologue
+     .line 38
      invoke-direct {p0}, Landroid/widget/Filter;-><init>()V
  
+     .line 39
      iput-object p1, p0, Landroid/widget/CursorFilter;->mClient:Landroid/widget/CursorFilter$CursorFilterClient;
  
+     .line 40
      return-void
  .end method
  
***************
*** 35,40 ****
--- 38,44 ----
      .parameter "resultValue"
  
      .prologue
+     .line 44
      iget-object v0, p0, Landroid/widget/CursorFilter;->mClient:Landroid/widget/CursorFilter$CursorFilterClient;
  
      check-cast p1, Landroid/database/Cursor;
***************
*** 52,87 ****
--- 56,99 ----
      .parameter "constraint"
  
      .prologue
+     .line 49
      iget-object v2, p0, Landroid/widget/CursorFilter;->mClient:Landroid/widget/CursorFilter$CursorFilterClient;
  
      invoke-interface {v2, p1}, Landroid/widget/CursorFilter$CursorFilterClient;->runQueryOnBackgroundThread(Ljava/lang/CharSequence;)Landroid/database/Cursor;
  
      move-result-object v0
  
+     .line 51
      .local v0, cursor:Landroid/database/Cursor;
      new-instance v1, Landroid/widget/Filter$FilterResults;
  
      invoke-direct {v1}, Landroid/widget/Filter$FilterResults;-><init>()V
  
+     .line 52
      .local v1, results:Landroid/widget/Filter$FilterResults;
      if-eqz v0, :cond_0
  
+     .line 53
      invoke-interface {v0}, Landroid/database/Cursor;->getCount()I
  
      move-result v2
  
      iput v2, v1, Landroid/widget/Filter$FilterResults;->count:I
  
+     .line 54
      iput-object v0, v1, Landroid/widget/Filter$FilterResults;->values:Ljava/lang/Object;
  
+     .line 59
      :goto_0
      return-object v1
  
+     .line 56
      :cond_0
      const/4 v2, 0x0
  
      iput v2, v1, Landroid/widget/Filter$FilterResults;->count:I
  
+     .line 57
      const/4 v2, 0x0
  
      iput-object v2, v1, Landroid/widget/Filter$FilterResults;->values:Ljava/lang/Object;
***************
*** 95,106 ****
--- 107,120 ----
      .parameter "results"
  
      .prologue
+     .line 64
      iget-object v1, p0, Landroid/widget/CursorFilter;->mClient:Landroid/widget/CursorFilter$CursorFilterClient;
  
      invoke-interface {v1}, Landroid/widget/CursorFilter$CursorFilterClient;->getCursor()Landroid/database/Cursor;
  
      move-result-object v0
  
+     .line 66
      .local v0, oldCursor:Landroid/database/Cursor;
      iget-object v1, p2, Landroid/widget/Filter$FilterResults;->values:Ljava/lang/Object;
  
***************
*** 110,115 ****
--- 124,130 ----
  
      if-eq v1, v0, :cond_0
  
+     .line 67
      iget-object v2, p0, Landroid/widget/CursorFilter;->mClient:Landroid/widget/CursorFilter$CursorFilterClient;
  
      iget-object v1, p2, Landroid/widget/Filter$FilterResults;->values:Ljava/lang/Object;
***************
*** 118,123 ****
--- 133,139 ----
  
      invoke-interface {v2, v1}, Landroid/widget/CursorFilter$CursorFilterClient;->changeCursor(Landroid/database/Cursor;)V
  
+     .line 69
      :cond_0
      return-void
  .end method
