*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/DialerFilter.smali	2012-03-23 19:35:58.819659460 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/DialerFilter.smali.original	2012-03-23 19:34:44.930312090 -0500
***************
*** 39,46 ****
--- 39,48 ----
      .parameter "context"
  
      .prologue
+     .line 41
      invoke-direct {p0, p1}, Landroid/widget/RelativeLayout;-><init>(Landroid/content/Context;)V
  
+     .line 42
      return-void
  .end method
  
***************
*** 50,57 ****
--- 52,61 ----
      .parameter "attrs"
  
      .prologue
+     .line 45
      invoke-direct {p0, p1, p2}, Landroid/widget/RelativeLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 46
      return-void
  .end method
  
***************
*** 59,74 ****
--- 63,81 ----
      .locals 2
  
      .prologue
+     .line 263
      iget-object v0, p0, Landroid/widget/DialerFilter;->mPrimary:Landroid/widget/EditText;
  
      iget-object v1, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      if-ne v0, v1, :cond_0
  
+     .line 264
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/widget/DialerFilter;->swapPrimaryAndHint(Z)V
  
+     .line 266
      :cond_0
      return-void
  .end method
***************
*** 77,92 ****
--- 84,102 ----
      .locals 2
  
      .prologue
+     .line 257
      iget-object v0, p0, Landroid/widget/DialerFilter;->mPrimary:Landroid/widget/EditText;
  
      iget-object v1, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      if-ne v0, v1, :cond_0
  
+     .line 258
      const/4 v0, 0x1
  
      invoke-direct {p0, v0}, Landroid/widget/DialerFilter;->swapPrimaryAndHint(Z)V
  
+     .line 260
      :cond_0
      return-void
  .end method
***************
*** 96,107 ****
--- 106,119 ----
      .parameter "makeLettersPrimary"
  
      .prologue
+     .line 269
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v4}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
  
      move-result-object v3
  
+     .line 270
      .local v3, lettersText:Landroid/text/Editable;
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
***************
*** 109,114 ****
--- 121,127 ----
  
      move-result-object v1
  
+     .line 271
      .local v1, digitsText:Landroid/text/Editable;
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
***************
*** 116,121 ****
--- 129,135 ----
  
      move-result-object v2
  
+     .line 272
      .local v2, lettersInput:Landroid/text/method/KeyListener;
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
***************
*** 123,199 ****
--- 137,229 ----
  
      move-result-object v0
  
+     .line 274
      .local v0, digitsInput:Landroid/text/method/KeyListener;
      if-eqz p1, :cond_0
  
+     .line 275
      iget-object v4, p0, Landroid/widget/DialerFilter;->mPrimary:Landroid/widget/EditText;
  
      iput-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
+     .line 276
      iget-object v4, p0, Landroid/widget/DialerFilter;->mHint:Landroid/widget/EditText;
  
      iput-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
+     .line 282
      :goto_0
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v4, v2}, Landroid/widget/EditText;->setKeyListener(Landroid/text/method/KeyListener;)V
  
+     .line 283
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v4, v3}, Landroid/widget/EditText;->setText(Ljava/lang/CharSequence;)V
  
+     .line 284
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v4}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
  
      move-result-object v3
  
+     .line 285
      invoke-interface {v3}, Landroid/text/Editable;->length()I
  
      move-result v4
  
      invoke-static {v3, v4}, Landroid/text/Selection;->setSelection(Landroid/text/Spannable;I)V
  
+     .line 287
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v4, v0}, Landroid/widget/EditText;->setKeyListener(Landroid/text/method/KeyListener;)V
  
+     .line 288
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v4, v1}, Landroid/widget/EditText;->setText(Ljava/lang/CharSequence;)V
  
+     .line 289
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v4}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
  
      move-result-object v1
  
+     .line 290
      invoke-interface {v1}, Landroid/text/Editable;->length()I
  
      move-result v4
  
      invoke-static {v1, v4}, Landroid/text/Selection;->setSelection(Landroid/text/Spannable;I)V
  
+     .line 293
      iget-object v4, p0, Landroid/widget/DialerFilter;->mPrimary:Landroid/widget/EditText;
  
      iget-object v5, p0, Landroid/widget/DialerFilter;->mInputFilters:[Landroid/text/InputFilter;
  
      invoke-virtual {v4, v5}, Landroid/widget/EditText;->setFilters([Landroid/text/InputFilter;)V
  
+     .line 294
      iget-object v4, p0, Landroid/widget/DialerFilter;->mHint:Landroid/widget/EditText;
  
      iget-object v5, p0, Landroid/widget/DialerFilter;->mInputFilters:[Landroid/text/InputFilter;
  
      invoke-virtual {v4, v5}, Landroid/widget/EditText;->setFilters([Landroid/text/InputFilter;)V
  
+     .line 295
      return-void
  
+     .line 278
      :cond_0
      iget-object v4, p0, Landroid/widget/DialerFilter;->mHint:Landroid/widget/EditText;
  
      iput-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
+     .line 279
      iget-object v4, p0, Landroid/widget/DialerFilter;->mPrimary:Landroid/widget/EditText;
  
      iput-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
***************
*** 208,220 ****
--- 238,253 ----
      .parameter "text"
  
      .prologue
+     .line 323
      iget v0, p0, Landroid/widget/DialerFilter;->mMode:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 339
      :goto_0
      return-void
  
+     .line 325
      :pswitch_0
      iget-object v0, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
***************
*** 224,229 ****
--- 257,263 ----
  
      invoke-interface {v0, p1}, Landroid/text/Editable;->append(Ljava/lang/CharSequence;)Landroid/text/Editable;
  
+     .line 326
      iget-object v0, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v0}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
***************
*** 234,239 ****
--- 268,274 ----
  
      goto :goto_0
  
+     .line 331
      :pswitch_1
      iget-object v0, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
***************
*** 245,250 ****
--- 280,286 ----
  
      goto :goto_0
  
+     .line 336
      :pswitch_2
      iget-object v0, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
***************
*** 256,261 ****
--- 292,298 ----
  
      goto :goto_0
  
+     .line 323
      nop
  
      :pswitch_data_0
***************
*** 272,305 ****
--- 309,350 ----
      .locals 2
  
      .prologue
+     .line 347
      iget-object v1, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v1}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
  
      move-result-object v0
  
+     .line 348
      .local v0, text:Landroid/text/Editable;
      invoke-interface {v0}, Landroid/text/Editable;->clear()V
  
+     .line 350
      iget-object v1, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v1}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
  
      move-result-object v0
  
+     .line 351
      invoke-interface {v0}, Landroid/text/Editable;->clear()V
  
+     .line 354
      iget-boolean v1, p0, Landroid/widget/DialerFilter;->mIsQwerty:Z
  
      if-eqz v1, :cond_0
  
+     .line 355
      const/4 v1, 0x1
  
      invoke-virtual {p0, v1}, Landroid/widget/DialerFilter;->setMode(I)V
  
+     .line 359
      :goto_0
      return-void
  
+     .line 357
      :cond_0
      const/4 v1, 0x4
  
***************
*** 312,317 ****
--- 357,363 ----
      .locals 1
  
      .prologue
+     .line 307
      iget-object v0, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v0}, Landroid/widget/EditText;->getVisibility()I
***************
*** 320,331 ****
--- 366,379 ----
  
      if-nez v0, :cond_0
  
+     .line 308
      iget-object v0, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v0}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
  
      move-result-object v0
  
+     .line 310
      :goto_0
      return-object v0
  
***************
*** 339,354 ****
--- 387,405 ----
      .locals 2
  
      .prologue
+     .line 315
      iget v0, p0, Landroid/widget/DialerFilter;->mMode:I
  
      const/4 v1, 0x4
  
      if-eq v0, v1, :cond_0
  
+     .line 316
      invoke-virtual {p0}, Landroid/widget/DialerFilter;->getLetters()Ljava/lang/CharSequence;
  
      move-result-object v0
  
+     .line 318
      :goto_0
      return-object v0
  
***************
*** 364,369 ****
--- 415,421 ----
      .locals 1
  
      .prologue
+     .line 299
      iget-object v0, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v0}, Landroid/widget/EditText;->getVisibility()I
***************
*** 372,383 ****
--- 424,437 ----
  
      if-nez v0, :cond_0
  
+     .line 300
      iget-object v0, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v0}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
  
      move-result-object v0
  
+     .line 302
      :goto_0
      return-object v0
  
***************
*** 391,396 ****
--- 445,451 ----
      .locals 1
  
      .prologue
+     .line 210
      iget v0, p0, Landroid/widget/DialerFilter;->mMode:I
  
      return v0
***************
*** 400,405 ****
--- 455,461 ----
      .locals 1
  
      .prologue
+     .line 103
      iget-boolean v0, p0, Landroid/widget/DialerFilter;->mIsQwerty:Z
  
      return v0
***************
*** 415,422 ****
--- 471,480 ----
  
      const/4 v2, 0x1
  
+     .line 50
      invoke-super {p0}, Landroid/widget/RelativeLayout;->onFinishInflate()V
  
+     .line 53
      new-array v0, v2, [Landroid/text/InputFilter;
  
      new-instance v1, Landroid/text/InputFilter$AllCaps;
***************
*** 427,432 ****
--- 485,491 ----
  
      iput-object v0, p0, Landroid/widget/DialerFilter;->mInputFilters:[Landroid/text/InputFilter;
  
+     .line 55
      const v0, 0x1020005
  
      invoke-virtual {p0, v0}, Landroid/widget/DialerFilter;->findViewById(I)Landroid/view/View;
***************
*** 437,446 ****
--- 496,507 ----
  
      iput-object v0, p0, Landroid/widget/DialerFilter;->mHint:Landroid/widget/EditText;
  
+     .line 56
      iget-object v0, p0, Landroid/widget/DialerFilter;->mHint:Landroid/widget/EditText;
  
      if-nez v0, :cond_0
  
+     .line 57
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "DialerFilter must have a child EditText named hint"
***************
*** 449,454 ****
--- 510,516 ----
  
      throw v0
  
+     .line 59
      :cond_0
      iget-object v0, p0, Landroid/widget/DialerFilter;->mHint:Landroid/widget/EditText;
  
***************
*** 456,465 ****
--- 518,529 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/EditText;->setFilters([Landroid/text/InputFilter;)V
  
+     .line 61
      iget-object v0, p0, Landroid/widget/DialerFilter;->mHint:Landroid/widget/EditText;
  
      iput-object v0, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
+     .line 62
      iget-object v0, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-static {}, Landroid/text/method/TextKeyListener;->getInstance()Landroid/text/method/TextKeyListener;
***************
*** 468,481 ****
--- 532,548 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/EditText;->setKeyListener(Landroid/text/method/KeyListener;)V
  
+     .line 63
      iget-object v0, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v0, v4}, Landroid/widget/EditText;->setMovementMethod(Landroid/text/method/MovementMethod;)V
  
+     .line 64
      iget-object v0, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v0, v3}, Landroid/widget/EditText;->setFocusable(Z)V
  
+     .line 67
      const v0, 0x102000c
  
      invoke-virtual {p0, v0}, Landroid/widget/DialerFilter;->findViewById(I)Landroid/view/View;
***************
*** 486,495 ****
--- 553,564 ----
  
      iput-object v0, p0, Landroid/widget/DialerFilter;->mPrimary:Landroid/widget/EditText;
  
+     .line 68
      iget-object v0, p0, Landroid/widget/DialerFilter;->mPrimary:Landroid/widget/EditText;
  
      if-nez v0, :cond_1
  
+     .line 69
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "DialerFilter must have a child EditText named primary"
***************
*** 498,503 ****
--- 567,573 ----
  
      throw v0
  
+     .line 71
      :cond_1
      iget-object v0, p0, Landroid/widget/DialerFilter;->mPrimary:Landroid/widget/EditText;
  
***************
*** 505,514 ****
--- 575,586 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/EditText;->setFilters([Landroid/text/InputFilter;)V
  
+     .line 73
      iget-object v0, p0, Landroid/widget/DialerFilter;->mPrimary:Landroid/widget/EditText;
  
      iput-object v0, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
+     .line 74
      iget-object v0, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-static {}, Landroid/text/method/DialerKeyListener;->getInstance()Landroid/text/method/DialerKeyListener;
***************
*** 517,530 ****
--- 589,605 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/EditText;->setKeyListener(Landroid/text/method/KeyListener;)V
  
+     .line 75
      iget-object v0, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v0, v4}, Landroid/widget/EditText;->setMovementMethod(Landroid/text/method/MovementMethod;)V
  
+     .line 76
      iget-object v0, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v0, v3}, Landroid/widget/EditText;->setFocusable(Z)V
  
+     .line 79
      const v0, 0x1020006
  
      invoke-virtual {p0, v0}, Landroid/widget/DialerFilter;->findViewById(I)Landroid/view/View;
***************
*** 535,546 ****
--- 610,625 ----
  
      iput-object v0, p0, Landroid/widget/DialerFilter;->mIcon:Landroid/widget/ImageView;
  
+     .line 82
      invoke-virtual {p0, v2}, Landroid/widget/DialerFilter;->setFocusable(Z)V
  
+     .line 85
      iput-boolean v2, p0, Landroid/widget/DialerFilter;->mIsQwerty:Z
  
+     .line 86
      invoke-virtual {p0, v2}, Landroid/widget/DialerFilter;->setMode(I)V
  
+     .line 87
      return-void
  .end method
  
***************
*** 551,562 ****
--- 630,644 ----
      .parameter "previouslyFocusedRect"
  
      .prologue
+     .line 94
      invoke-super {p0, p1, p2, p3}, Landroid/widget/RelativeLayout;->onFocusChanged(ZILandroid/graphics/Rect;)V
  
+     .line 96
      iget-object v0, p0, Landroid/widget/DialerFilter;->mIcon:Landroid/widget/ImageView;
  
      if-eqz v0, :cond_0
  
+     .line 97
      iget-object v1, p0, Landroid/widget/DialerFilter;->mIcon:Landroid/widget/ImageView;
  
      if-eqz p1, :cond_1
***************
*** 566,574 ****
--- 648,658 ----
      :goto_0
      invoke-virtual {v1, v0}, Landroid/widget/ImageView;->setVisibility(I)V
  
+     .line 99
      :cond_0
      return-void
  
+     .line 97
      :cond_1
      const/16 v0, 0x8
  
***************
*** 583,609 ****
--- 667,700 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 108
      const/4 v1, 0x0
  
+     .line 110
      .local v1, handled:Z
      sparse-switch p1, :sswitch_data_0
  
+     .line 154
      iget v4, p0, Landroid/widget/DialerFilter;->mMode:I
  
      packed-switch v4, :pswitch_data_0
  
+     .line 195
      :cond_0
      :goto_0
      :sswitch_0
      if-nez v1, :cond_1
  
+     .line 196
      invoke-super {p0, p1, p2}, Landroid/widget/RelativeLayout;->onKeyDown(ILandroid/view/KeyEvent;)Z
  
      move-result v3
  
+     .line 198
      :cond_1
      return v3
  
+     .line 120
      :sswitch_1
      iget v4, p0, Landroid/widget/DialerFilter;->mMode:I
  
***************
*** 611,616 ****
--- 702,708 ----
  
      goto :goto_0
  
+     .line 122
      :pswitch_0
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
***************
*** 618,623 ****
--- 710,716 ----
  
      move-result v1
  
+     .line 123
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v4, p1, p2}, Landroid/widget/EditText;->onKeyDown(ILandroid/view/KeyEvent;)Z
***************
*** 626,633 ****
--- 719,728 ----
  
      and-int/2addr v1, v4
  
+     .line 124
      goto :goto_0
  
+     .line 127
      :pswitch_1
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
***************
*** 635,640 ****
--- 730,736 ----
  
      move-result v1
  
+     .line 128
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v4}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
***************
*** 657,666 ****
--- 753,764 ----
  
      if-ne v4, v5, :cond_0
  
+     .line 129
      invoke-virtual {p0, v3}, Landroid/widget/DialerFilter;->setMode(I)V
  
      goto :goto_0
  
+     .line 134
      :pswitch_2
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
***************
*** 684,695 ****
--- 782,796 ----
  
      if-ne v4, v5, :cond_2
  
+     .line 135
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v4, p1, p2}, Landroid/widget/EditText;->onKeyDown(ILandroid/view/KeyEvent;)Z
  
+     .line 136
      invoke-virtual {p0, v3}, Landroid/widget/DialerFilter;->setMode(I)V
  
+     .line 138
      :cond_2
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
***************
*** 697,704 ****
--- 798,807 ----
  
      move-result v1
  
+     .line 139
      goto :goto_0
  
+     .line 142
      :pswitch_3
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
***************
*** 706,713 ****
--- 809,818 ----
  
      move-result v1
  
+     .line 143
      goto :goto_0
  
+     .line 146
      :pswitch_4
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
***************
*** 717,722 ****
--- 822,828 ----
  
      goto :goto_0
  
+     .line 156
      :pswitch_5
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
***************
*** 724,748 ****
--- 830,860 ----
  
      move-result v1
  
+     .line 160
      invoke-static {p1}, Landroid/view/KeyEvent;->isModifierKey(I)Z
  
      move-result v4
  
      if-eqz v4, :cond_3
  
+     .line 161
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v4, p1, p2}, Landroid/widget/EditText;->onKeyDown(ILandroid/view/KeyEvent;)Z
  
+     .line 162
      const/4 v1, 0x1
  
+     .line 163
      goto :goto_0
  
+     .line 171
      :cond_3
      invoke-virtual {p2}, Landroid/view/KeyEvent;->isPrintingKey()Z
  
      move-result v2
  
+     .line 172
      .local v2, isPrint:Z
      if-nez v2, :cond_4
  
***************
*** 754,759 ****
--- 866,872 ----
  
      if-ne p1, v4, :cond_0
  
+     .line 174
      :cond_4
      sget-object v4, Landroid/text/method/DialerKeyListener;->CHARACTERS:[C
  
***************
*** 761,769 ****
--- 874,884 ----
  
      move-result v0
  
+     .line 175
      .local v0, c:C
      if-eqz v0, :cond_5
  
+     .line 176
      iget-object v4, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v4, p1, p2}, Landroid/widget/EditText;->onKeyDown(ILandroid/view/KeyEvent;)Z
***************
*** 774,779 ****
--- 889,895 ----
  
      goto/16 :goto_0
  
+     .line 178
      :cond_5
      const/4 v4, 0x2
  
***************
*** 781,786 ****
--- 897,903 ----
  
      goto/16 :goto_0
  
+     .line 185
      .end local v0           #c:C
      .end local v2           #isPrint:Z
      :pswitch_6
***************
*** 790,797 ****
--- 907,916 ----
  
      move-result v1
  
+     .line 186
      goto/16 :goto_0
  
+     .line 190
      :pswitch_7
      iget-object v4, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
***************
*** 801,806 ****
--- 920,926 ----
  
      goto/16 :goto_0
  
+     .line 110
      :sswitch_data_0
      .sparse-switch
          0x13 -> :sswitch_0
***************
*** 812,817 ****
--- 932,938 ----
          0x43 -> :sswitch_1
      .end sparse-switch
  
+     .line 154
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_5
***************
*** 821,826 ****
--- 942,948 ----
          :pswitch_7
      .end packed-switch
  
+     .line 120
      :pswitch_data_1
      .packed-switch 0x1
          :pswitch_0
***************
*** 837,848 ****
--- 959,972 ----
      .parameter "event"
  
      .prologue
+     .line 204
      iget-object v2, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v2, p1, p2}, Landroid/widget/EditText;->onKeyUp(ILandroid/view/KeyEvent;)Z
  
      move-result v0
  
+     .line 205
      .local v0, a:Z
      iget-object v2, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
***************
*** 850,855 ****
--- 974,980 ----
  
      move-result v1
  
+     .line 206
      .local v1, b:Z
      if-nez v0, :cond_0
  
***************
*** 873,878 ****
--- 998,1004 ----
      .parameter "newMode"
  
      .prologue
+     .line 396
      return-void
  .end method
  
***************
*** 881,904 ****
--- 1007,1035 ----
      .parameter "watcher"
  
      .prologue
+     .line 383
      iget v1, p0, Landroid/widget/DialerFilter;->mMode:I
  
      const/4 v2, 0x4
  
      if-eq v1, v2, :cond_0
  
+     .line 384
      iget-object v1, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v1}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
  
      move-result-object v0
  
+     .line 388
      .local v0, text:Landroid/text/Spannable;
      :goto_0
      invoke-interface {v0, p1}, Landroid/text/Spannable;->removeSpan(Ljava/lang/Object;)V
  
+     .line 389
      return-void
  
+     .line 386
      .end local v0           #text:Landroid/text/Spannable;
      :cond_0
      iget-object v1, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
***************
*** 916,921 ****
--- 1047,1053 ----
      .parameter "watcher"
  
      .prologue
+     .line 368
      iget-object v2, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v2}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
***************
*** 925,932 ****
--- 1057,1066 ----
      .local v1, text:Ljava/lang/CharSequence;
      move-object v0, v1
  
+     .line 369
      check-cast v0, Landroid/text/Spannable;
  
+     .line 370
      .local v0, span:Landroid/text/Spannable;
      const/4 v2, 0x0
  
***************
*** 938,943 ****
--- 1072,1078 ----
  
      invoke-interface {v0, p1, v2, v3, v4}, Landroid/text/Spannable;->setSpan(Ljava/lang/Object;III)V
  
+     .line 371
      return-void
  .end method
  
***************
*** 946,962 ****
--- 1081,1101 ----
      .parameter "watcher"
  
      .prologue
+     .line 374
      iget v0, p0, Landroid/widget/DialerFilter;->mMode:I
  
      const/4 v1, 0x4
  
      if-eq v0, v1, :cond_0
  
+     .line 375
      invoke-virtual {p0, p1}, Landroid/widget/DialerFilter;->setLettersWatcher(Landroid/text/TextWatcher;)V
  
+     .line 379
      :goto_0
      return-void
  
+     .line 377
      :cond_0
      invoke-virtual {p0, p1}, Landroid/widget/DialerFilter;->setDigitsWatcher(Landroid/text/TextWatcher;)V
  
***************
*** 968,973 ****
--- 1107,1113 ----
      .parameter "watcher"
  
      .prologue
+     .line 362
      iget-object v2, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v2}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
***************
*** 977,984 ****
--- 1117,1126 ----
      .local v1, text:Ljava/lang/CharSequence;
      move-object v0, v1
  
+     .line 363
      check-cast v0, Landroid/text/Spannable;
  
+     .line 364
      .local v0, span:Landroid/text/Spannable;
      const/4 v2, 0x0
  
***************
*** 990,995 ****
--- 1132,1138 ----
  
      invoke-interface {v0, p1, v2, v3, v4}, Landroid/text/Spannable;->setSpan(Ljava/lang/Object;III)V
  
+     .line 365
      return-void
  .end method
  
***************
*** 1004,1087 ****
--- 1147,1251 ----
  
      const/4 v2, 0x0
  
+     .line 219
      packed-switch p1, :pswitch_data_0
  
+     .line 251
      :goto_0
      iget v0, p0, Landroid/widget/DialerFilter;->mMode:I
  
+     .line 252
      .local v0, oldMode:I
      iput p1, p0, Landroid/widget/DialerFilter;->mMode:I
  
+     .line 253
      invoke-virtual {p0, v0, p1}, Landroid/widget/DialerFilter;->onModeChange(II)V
  
+     .line 254
      return-void
  
+     .line 221
      .end local v0           #oldMode:I
      :pswitch_0
      invoke-direct {p0}, Landroid/widget/DialerFilter;->makeDigitsPrimary()V
  
+     .line 222
      iget-object v1, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setVisibility(I)V
  
+     .line 223
      iget-object v1, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setVisibility(I)V
  
      goto :goto_0
  
+     .line 227
      :pswitch_1
      invoke-direct {p0}, Landroid/widget/DialerFilter;->makeDigitsPrimary()V
  
+     .line 228
      iget-object v1, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v1, v4}, Landroid/widget/EditText;->setVisibility(I)V
  
+     .line 229
      iget-object v1, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setVisibility(I)V
  
      goto :goto_0
  
+     .line 233
      :pswitch_2
      invoke-direct {p0}, Landroid/widget/DialerFilter;->makeLettersPrimary()V
  
+     .line 234
      iget-object v1, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setVisibility(I)V
  
+     .line 235
      iget-object v1, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v1, v4}, Landroid/widget/EditText;->setVisibility(I)V
  
      goto :goto_0
  
+     .line 239
      :pswitch_3
      invoke-direct {p0}, Landroid/widget/DialerFilter;->makeDigitsPrimary()V
  
+     .line 240
      iget-object v1, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v1, v3}, Landroid/widget/EditText;->setVisibility(I)V
  
+     .line 241
      iget-object v1, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setVisibility(I)V
  
      goto :goto_0
  
+     .line 245
      :pswitch_4
      invoke-direct {p0}, Landroid/widget/DialerFilter;->makeLettersPrimary()V
  
+     .line 246
      iget-object v1, p0, Landroid/widget/DialerFilter;->mLetters:Landroid/widget/EditText;
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setVisibility(I)V
  
+     .line 247
      iget-object v1, p0, Landroid/widget/DialerFilter;->mDigits:Landroid/widget/EditText;
  
      invoke-virtual {v1, v3}, Landroid/widget/EditText;->setVisibility(I)V
  
      goto :goto_0
  
+     .line 219
      nop
  
      :pswitch_data_0
