*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/DigitalClock.smali	2012-03-23 19:35:51.849532361 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/DigitalClock.smali.original	2012-03-23 19:34:44.875311088 -0500
***************
*** 37,50 ****
--- 37,54 ----
      .parameter "context"
  
      .prologue
+     .line 52
      invoke-direct {p0, p1}, Landroid/widget/TextView;-><init>(Landroid/content/Context;)V
  
+     .line 47
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/DigitalClock;->mTickerStopped:Z
  
+     .line 53
      invoke-direct {p0, p1}, Landroid/widget/DigitalClock;->initClock(Landroid/content/Context;)V
  
+     .line 54
      return-void
  .end method
  
***************
*** 54,67 ****
--- 58,75 ----
      .parameter "attrs"
  
      .prologue
+     .line 57
      invoke-direct {p0, p1, p2}, Landroid/widget/TextView;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 47
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/DigitalClock;->mTickerStopped:Z
  
+     .line 58
      invoke-direct {p0, p1}, Landroid/widget/DigitalClock;->initClock(Landroid/content/Context;)V
  
+     .line 59
      return-void
  .end method
  
***************
*** 70,75 ****
--- 78,84 ----
      .parameter "x0"
  
      .prologue
+     .line 37
      iget-boolean v0, p0, Landroid/widget/DigitalClock;->mTickerStopped:Z
  
      return v0
***************
*** 80,85 ****
--- 89,95 ----
      .parameter "x0"
  
      .prologue
+     .line 37
      iget-object v0, p0, Landroid/widget/DigitalClock;->mTicker:Ljava/lang/Runnable;
  
      return-object v0
***************
*** 90,95 ****
--- 100,106 ----
      .parameter "x0"
  
      .prologue
+     .line 37
      iget-object v0, p0, Landroid/widget/DigitalClock;->mHandler:Landroid/os/Handler;
  
      return-object v0
***************
*** 100,105 ****
--- 111,117 ----
      .parameter "x0"
  
      .prologue
+     .line 37
      invoke-direct {p0}, Landroid/widget/DigitalClock;->setFormat()V
  
      return-void
***************
*** 109,114 ****
--- 121,127 ----
      .locals 1
  
      .prologue
+     .line 108
      invoke-virtual {p0}, Landroid/widget/DigitalClock;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 125,147 ****
--- 138,164 ----
      .parameter "context"
  
      .prologue
+     .line 62
      iget-object v1, p0, Landroid/widget/DigitalClock;->mContext:Landroid/content/Context;
  
      invoke-virtual {v1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v0
  
+     .line 64
      .local v0, r:Landroid/content/res/Resources;
      iget-object v1, p0, Landroid/widget/DigitalClock;->mCalendar:Ljava/util/Calendar;
  
      if-nez v1, :cond_0
  
+     .line 65
      invoke-static {}, Ljava/util/Calendar;->getInstance()Ljava/util/Calendar;
  
      move-result-object v1
  
      iput-object v1, p0, Landroid/widget/DigitalClock;->mCalendar:Ljava/util/Calendar;
  
+     .line 68
      :cond_0
      new-instance v1, Landroid/widget/DigitalClock$FormatChangeObserver;
  
***************
*** 149,154 ****
--- 166,172 ----
  
      iput-object v1, p0, Landroid/widget/DigitalClock;->mFormatChangeObserver:Landroid/widget/DigitalClock$FormatChangeObserver;
  
+     .line 69
      invoke-virtual {p0}, Landroid/widget/DigitalClock;->getContext()Landroid/content/Context;
  
      move-result-object v1
***************
*** 165,172 ****
--- 183,192 ----
  
      invoke-virtual {v1, v2, v3, v4}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 72
      invoke-direct {p0}, Landroid/widget/DigitalClock;->setFormat()V
  
+     .line 73
      return-void
  .end method
  
***************
*** 174,192 ****
--- 194,216 ----
      .locals 1
  
      .prologue
+     .line 112
      invoke-direct {p0}, Landroid/widget/DigitalClock;->get24HourMode()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 113
      const-string v0, "k:mm:ss"
  
      iput-object v0, p0, Landroid/widget/DigitalClock;->mFormat:Ljava/lang/String;
  
+     .line 117
      :goto_0
      return-void
  
+     .line 115
      :cond_0
      const-string v0, "h:mm:ss aa"
  
***************
*** 201,228 ****
--- 225,258 ----
      .locals 1
  
      .prologue
+     .line 77
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/DigitalClock;->mTickerStopped:Z
  
+     .line 78
      invoke-super {p0}, Landroid/widget/TextView;->onAttachedToWindow()V
  
+     .line 79
      new-instance v0, Landroid/os/Handler;
  
      invoke-direct {v0}, Landroid/os/Handler;-><init>()V
  
      iput-object v0, p0, Landroid/widget/DigitalClock;->mHandler:Landroid/os/Handler;
  
+     .line 84
      new-instance v0, Landroid/widget/DigitalClock$1;
  
      invoke-direct {v0, p0}, Landroid/widget/DigitalClock$1;-><init>(Landroid/widget/DigitalClock;)V
  
      iput-object v0, p0, Landroid/widget/DigitalClock;->mTicker:Ljava/lang/Runnable;
  
+     .line 95
      iget-object v0, p0, Landroid/widget/DigitalClock;->mTicker:Ljava/lang/Runnable;
  
      invoke-interface {v0}, Ljava/lang/Runnable;->run()V
  
+     .line 96
      return-void
  .end method
  
***************
*** 230,240 ****
--- 260,273 ----
      .locals 1
  
      .prologue
+     .line 100
      invoke-super {p0}, Landroid/widget/TextView;->onDetachedFromWindow()V
  
+     .line 101
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/DigitalClock;->mTickerStopped:Z
  
+     .line 102
      return-void
  .end method
