*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/DigitalClock$1.smali	2012-03-23 19:36:01.332705287 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/DigitalClock$1.smali.original	2012-03-23 19:34:44.945312364 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 84
      iput-object p1, p0, Landroid/widget/DigitalClock$1;->this$0:Landroid/widget/DigitalClock;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,47 ****
--- 43,49 ----
      .prologue
      const-wide/16 v7, 0x3e8
  
+     .line 86
      iget-object v4, p0, Landroid/widget/DigitalClock$1;->this$0:Landroid/widget/DigitalClock;
  
      #getter for: Landroid/widget/DigitalClock;->mTickerStopped:Z
***************
*** 51,59 ****
--- 53,63 ----
  
      if-eqz v4, :cond_0
  
+     .line 93
      :goto_0
      return-void
  
+     .line 87
      :cond_0
      iget-object v4, p0, Landroid/widget/DigitalClock$1;->this$0:Landroid/widget/DigitalClock;
  
***************
*** 65,70 ****
--- 69,75 ----
  
      invoke-virtual {v4, v5, v6}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 88
      iget-object v4, p0, Landroid/widget/DigitalClock$1;->this$0:Landroid/widget/DigitalClock;
  
      iget-object v5, p0, Landroid/widget/DigitalClock$1;->this$0:Landroid/widget/DigitalClock;
***************
*** 81,94 ****
--- 86,102 ----
  
      invoke-virtual {v4, v5}, Landroid/widget/DigitalClock;->setText(Ljava/lang/CharSequence;)V
  
+     .line 89
      iget-object v4, p0, Landroid/widget/DigitalClock$1;->this$0:Landroid/widget/DigitalClock;
  
      invoke-virtual {v4}, Landroid/widget/DigitalClock;->invalidate()V
  
+     .line 90
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v2
  
+     .line 91
      .local v2, now:J
      rem-long v4, v2, v7
  
***************
*** 96,101 ****
--- 104,110 ----
  
      add-long v0, v2, v4
  
+     .line 92
      .local v0, next:J
      iget-object v4, p0, Landroid/widget/DigitalClock$1;->this$0:Landroid/widget/DigitalClock;
  
