*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/DoubleDigitManager$1.smali	2012-03-23 19:35:54.874587522 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/DoubleDigitManager$1.smali.original	2012-03-23 19:34:44.899311525 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 52
      iput-object p1, p0, Landroid/widget/DoubleDigitManager$1;->this$0:Landroid/widget/DoubleDigitManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .locals 2
  
      .prologue
+     .line 54
      iget-object v0, p0, Landroid/widget/DoubleDigitManager$1;->this$0:Landroid/widget/DoubleDigitManager;
  
      #getter for: Landroid/widget/DoubleDigitManager;->intermediateDigit:Ljava/lang/Integer;
***************
*** 49,54 ****
--- 51,57 ----
  
      if-eqz v0, :cond_0
  
+     .line 55
      iget-object v0, p0, Landroid/widget/DoubleDigitManager$1;->this$0:Landroid/widget/DoubleDigitManager;
  
      #getter for: Landroid/widget/DoubleDigitManager;->mCallBack:Landroid/widget/DoubleDigitManager$CallBack;
***************
*** 69,74 ****
--- 72,78 ----
  
      invoke-interface {v0, v1}, Landroid/widget/DoubleDigitManager$CallBack;->singleDigitFinal(I)V
  
+     .line 56
      iget-object v0, p0, Landroid/widget/DoubleDigitManager$1;->this$0:Landroid/widget/DoubleDigitManager;
  
      const/4 v1, 0x0
***************
*** 76,81 ****
--- 80,86 ----
      #setter for: Landroid/widget/DoubleDigitManager;->intermediateDigit:Ljava/lang/Integer;
      invoke-static {v0, v1}, Landroid/widget/DoubleDigitManager;->access$002(Landroid/widget/DoubleDigitManager;Ljava/lang/Integer;)Ljava/lang/Integer;
  
+     .line 58
      :cond_0
      return-void
  .end method
