*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ExpandableListConnector.smali	2012-03-23 19:35:52.893551399 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ExpandableListConnector.smali.original	2012-03-23 19:34:44.884311252 -0500
***************
*** 43,68 ****
--- 43,74 ----
      .parameter "expandableListAdapter"
  
      .prologue
+     .line 75
      invoke-direct {p0}, Landroid/widget/BaseAdapter;-><init>()V
  
+     .line 67
      const v0, 0x7fffffff
  
      iput v0, p0, Landroid/widget/ExpandableListConnector;->mMaxExpGroupCount:I
  
+     .line 70
      new-instance v0, Landroid/widget/ExpandableListConnector$MyDataSetObserver;
  
      invoke-direct {v0, p0}, Landroid/widget/ExpandableListConnector$MyDataSetObserver;-><init>(Landroid/widget/ExpandableListConnector;)V
  
      iput-object v0, p0, Landroid/widget/ExpandableListConnector;->mDataSetObserver:Landroid/database/DataSetObserver;
  
+     .line 76
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
+     .line 78
      invoke-virtual {p0, p1}, Landroid/widget/ExpandableListConnector;->setExpandableListAdapter(Landroid/widget/ExpandableListAdapter;)V
  
+     .line 79
      return-void
  .end method
  
***************
*** 73,78 ****
--- 79,85 ----
      .parameter "x2"
  
      .prologue
+     .line 50
      invoke-direct {p0, p1, p2}, Landroid/widget/ExpandableListConnector;->refreshExpGroupMetadataList(ZZ)V
  
      return-void
***************
*** 86,110 ****
--- 93,124 ----
      .prologue
      const/4 v12, -0x1
  
+     .line 518
      iget-object v2, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
+     .line 519
      .local v2, egml:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/widget/ExpandableListConnector$GroupMetadata;>;"
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
  
      move-result v3
  
+     .line 520
      .local v3, egmlSize:I
      const/4 v0, 0x0
  
+     .line 523
      .local v0, curFlPos:I
      const/4 v9, 0x0
  
      iput v9, p0, Landroid/widget/ExpandableListConnector;->mTotalExpChildrenCount:I
  
+     .line 525
      if-eqz p2, :cond_3
  
+     .line 527
      const/4 v8, 0x0
  
+     .line 529
      .local v8, positionsChanged:Z
      add-int/lit8 v5, v3, -0x1
  
***************
*** 112,123 ****
--- 126,139 ----
      :goto_0
      if-ltz v5, :cond_2
  
+     .line 530
      invoke-virtual {v2, v5}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 531
      .local v1, curGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      iget-wide v9, v1, Landroid/widget/ExpandableListConnector$GroupMetadata;->gId:J
  
***************
*** 127,167 ****
--- 143,194 ----
  
      move-result v7
  
+     .line 532
      .local v7, newGPos:I
      iget v9, v1, Landroid/widget/ExpandableListConnector$GroupMetadata;->gPos:I
  
      if-eq v7, v9, :cond_1
  
+     .line 533
      if-ne v7, v12, :cond_0
  
+     .line 535
      invoke-virtual {v2, v5}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 536
      add-int/lit8 v3, v3, -0x1
  
+     .line 539
      :cond_0
      iput v7, v1, Landroid/widget/ExpandableListConnector$GroupMetadata;->gPos:I
  
+     .line 540
      if-nez v8, :cond_1
  
      const/4 v8, 0x1
  
+     .line 529
      :cond_1
      add-int/lit8 v5, v5, -0x1
  
      goto :goto_0
  
+     .line 544
      .end local v1           #curGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      .end local v7           #newGPos:I
      :cond_2
      if-eqz v8, :cond_3
  
+     .line 546
      invoke-static {v2}, Ljava/util/Collections;->sort(Ljava/util/List;)V
  
+     .line 551
      .end local v5           #i:I
      .end local v8           #positionsChanged:Z
      :cond_3
      const/4 v6, 0x0
  
+     .line 552
      .local v6, lastGPos:I
      const/4 v5, 0x0
  
***************
*** 169,180 ****
--- 196,209 ----
      :goto_1
      if-ge v5, v3, :cond_6
  
+     .line 554
      invoke-virtual {v2, v5}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 560
      .restart local v1       #curGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      iget v9, v1, Landroid/widget/ExpandableListConnector$GroupMetadata;->lastChildFlPos:I
  
***************
*** 182,187 ****
--- 211,217 ----
  
      if-eqz p1, :cond_5
  
+     .line 561
      :cond_4
      iget-object v9, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
***************
*** 191,196 ****
--- 221,227 ----
  
      move-result v4
  
+     .line 570
      .local v4, gChildrenCount:I
      :goto_2
      iget v9, p0, Landroid/widget/ExpandableListConnector;->mTotalExpChildrenCount:I
***************
*** 199,222 ****
--- 230,260 ----
  
      iput v9, p0, Landroid/widget/ExpandableListConnector;->mTotalExpChildrenCount:I
  
+     .line 577
      iget v9, v1, Landroid/widget/ExpandableListConnector$GroupMetadata;->gPos:I
  
      sub-int/2addr v9, v6
  
      add-int/2addr v0, v9
  
+     .line 578
      iget v6, v1, Landroid/widget/ExpandableListConnector$GroupMetadata;->gPos:I
  
+     .line 581
      iput v0, v1, Landroid/widget/ExpandableListConnector$GroupMetadata;->flPos:I
  
+     .line 582
      add-int/2addr v0, v4
  
+     .line 583
      iput v0, v1, Landroid/widget/ExpandableListConnector$GroupMetadata;->lastChildFlPos:I
  
+     .line 552
      add-int/lit8 v5, v5, 0x1
  
      goto :goto_1
  
+     .line 566
      .end local v4           #gChildrenCount:I
      :cond_5
      iget v9, v1, Landroid/widget/ExpandableListConnector$GroupMetadata;->lastChildFlPos:I
***************
*** 228,233 ****
--- 266,272 ----
      .restart local v4       #gChildrenCount:I
      goto :goto_2
  
+     .line 585
      .end local v1           #curGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      .end local v4           #gChildrenCount:I
      :cond_6
***************
*** 240,245 ****
--- 279,285 ----
      .locals 1
  
      .prologue
+     .line 370
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      invoke-interface {v0}, Landroid/widget/ExpandableListAdapter;->areAllItemsEnabled()Z
***************
*** 256,261 ****
--- 296,302 ----
      .prologue
      const/4 v3, -0x1
  
+     .line 593
      const/4 v2, 0x2
  
      invoke-static {v2, p1, v3, v3}, Landroid/widget/ExpandableListPosition;->obtain(IIII)Landroid/widget/ExpandableListPosition;
***************
*** 266,284 ****
--- 307,329 ----
  
      move-result-object v0
  
+     .line 595
      .local v0, pm:Landroid/widget/ExpandableListConnector$PositionMetadata;
      if-nez v0, :cond_0
  
      const/4 v1, 0x0
  
+     .line 599
      :goto_0
      return v1
  
+     .line 597
      :cond_0
      invoke-virtual {p0, v0}, Landroid/widget/ExpandableListConnector;->collapseGroup(Landroid/widget/ExpandableListConnector$PositionMetadata;)Z
  
      move-result v1
  
+     .line 598
      .local v1, retValue:Z
      invoke-virtual {v0}, Landroid/widget/ExpandableListConnector$PositionMetadata;->recycle()V
  
***************
*** 292,304 ****
--- 337,352 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 612
      iget-object v1, p1, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
  
      if-nez v1, :cond_0
  
+     .line 626
      :goto_0
      return v0
  
+     .line 615
      :cond_0
      iget-object v1, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
***************
*** 306,315 ****
--- 354,366 ----
  
      invoke-virtual {v1, v2}, Ljava/util/ArrayList;->remove(Ljava/lang/Object;)Z
  
+     .line 618
      invoke-direct {p0, v0, v0}, Landroid/widget/ExpandableListConnector;->refreshExpGroupMetadataList(ZZ)V
  
+     .line 621
      invoke-virtual {p0}, Landroid/widget/ExpandableListConnector;->notifyDataSetChanged()V
  
+     .line 624
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      iget-object v1, p1, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
***************
*** 318,323 ****
--- 369,375 ----
  
      invoke-interface {v0, v1}, Landroid/widget/ExpandableListAdapter;->onGroupCollapsed(I)V
  
+     .line 626
      const/4 v0, 0x1
  
      goto :goto_0
***************
*** 330,335 ****
--- 382,388 ----
      .prologue
      const/4 v3, -0x1
  
+     .line 634
      const/4 v2, 0x2
  
      invoke-static {v2, p1, v3, v3}, Landroid/widget/ExpandableListPosition;->obtain(IIII)Landroid/widget/ExpandableListPosition;
***************
*** 340,353 ****
--- 393,409 ----
  
      move-result-object v0
  
+     .line 636
      .local v0, pm:Landroid/widget/ExpandableListConnector$PositionMetadata;
      invoke-virtual {p0, v0}, Landroid/widget/ExpandableListConnector;->expandGroup(Landroid/widget/ExpandableListConnector$PositionMetadata;)Z
  
      move-result v1
  
+     .line 637
      .local v1, retValue:Z
      invoke-virtual {v0}, Landroid/widget/ExpandableListConnector$PositionMetadata;->recycle()V
  
+     .line 638
      return v1
  .end method
  
***************
*** 360,371 ****
--- 416,429 ----
  
      const/4 v3, 0x0
  
+     .line 646
      iget-object v4, p1, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
  
      iget v4, v4, Landroid/widget/ExpandableListPosition;->groupPos:I
  
      if-gez v4, :cond_0
  
+     .line 648
      new-instance v3, Ljava/lang/RuntimeException;
  
      const-string v4, "Need group"
***************
*** 374,393 ****
--- 432,455 ----
  
      throw v3
  
+     .line 651
      :cond_0
      iget v4, p0, Landroid/widget/ExpandableListConnector;->mMaxExpGroupCount:I
  
      if-nez v4, :cond_2
  
+     .line 690
      :cond_1
      :goto_0
      return v3
  
+     .line 654
      :cond_2
      iget-object v4, p1, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
  
      if-nez v4, :cond_1
  
+     .line 657
      iget-object v4, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
      invoke-virtual {v4}, Ljava/util/ArrayList;->size()I
***************
*** 398,403 ****
--- 460,466 ----
  
      if-lt v4, v5, :cond_3
  
+     .line 661
      iget-object v4, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v3}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 406,411 ****
--- 469,475 ----
  
      check-cast v0, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 663
      .local v0, collapsedGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      iget-object v4, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
***************
*** 413,433 ****
--- 477,501 ----
  
      move-result v1
  
+     .line 665
      .local v1, collapsedIndex:I
      iget v4, v0, Landroid/widget/ExpandableListConnector$GroupMetadata;->gPos:I
  
      invoke-virtual {p0, v4}, Landroid/widget/ExpandableListConnector;->collapseGroup(I)Z
  
+     .line 668
      iget v4, p1, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupInsertIndex:I
  
      if-le v4, v1, :cond_3
  
+     .line 669
      iget v4, p1, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupInsertIndex:I
  
      add-int/lit8 v4, v4, -0x1
  
      iput v4, p1, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupInsertIndex:I
  
+     .line 673
      .end local v0           #collapsedGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      .end local v1           #collapsedIndex:I
      :cond_3
***************
*** 449,454 ****
--- 517,523 ----
  
      move-result-object v2
  
+     .line 679
      .local v2, expandedGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      iget-object v4, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
***************
*** 456,471 ****
--- 525,544 ----
  
      invoke-virtual {v4, v5, v2}, Ljava/util/ArrayList;->add(ILjava/lang/Object;)V
  
+     .line 682
      invoke-direct {p0, v3, v3}, Landroid/widget/ExpandableListConnector;->refreshExpGroupMetadataList(ZZ)V
  
+     .line 685
      invoke-virtual {p0}, Landroid/widget/ExpandableListConnector;->notifyDataSetChanged()V
  
+     .line 688
      iget-object v3, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      iget v4, v2, Landroid/widget/ExpandableListConnector$GroupMetadata;->gPos:I
  
      invoke-interface {v3, v4}, Landroid/widget/ExpandableListAdapter;->onGroupExpanded(I)V
  
+     .line 690
      const/4 v3, 0x1
  
      goto :goto_0
***************
*** 477,482 ****
--- 550,556 ----
      .parameter "seedGroupPosition"
  
      .prologue
+     .line 773
      move-object/from16 v0, p0
  
      iget-object v12, v0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
***************
*** 485,498 ****
--- 559,576 ----
  
      move-result v2
  
+     .line 775
      .local v2, count:I
      if-nez v2, :cond_0
  
+     .line 776
      const/4 v12, -0x1
  
+     .line 845
      :goto_0
      return v12
  
+     .line 780
      :cond_0
      const-wide/high16 v12, -0x8000
  
***************
*** 500,509 ****
--- 578,589 ----
  
      if-nez v12, :cond_1
  
+     .line 781
      const/4 v12, -0x1
  
      goto :goto_0
  
+     .line 785
      :cond_1
      const/4 v12, 0x0
  
***************
*** 513,518 ****
--- 593,599 ----
  
      move-result p3
  
+     .line 786
      add-int/lit8 v12, v2, -0x1
  
      move/from16 v0, p3
***************
*** 521,526 ****
--- 602,608 ----
  
      move-result p3
  
+     .line 788
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v12
***************
*** 529,555 ****
--- 611,644 ----
  
      add-long v3, v12, v14
  
+     .line 793
      .local v3, endTime:J
      move/from16 v5, p3
  
+     .line 796
      .local v5, first:I
      move/from16 v8, p3
  
+     .line 799
      .local v8, last:I
      const/4 v9, 0x0
  
+     .line 809
      .local v9, next:Z
      invoke-virtual/range {p0 .. p0}, Landroid/widget/ExpandableListConnector;->getAdapter()Landroid/widget/ExpandableListAdapter;
  
      move-result-object v1
  
+     .line 810
      .local v1, adapter:Landroid/widget/ExpandableListAdapter;
      if-nez v1, :cond_4
  
+     .line 811
      const/4 v12, -0x1
  
      goto :goto_0
  
+     .line 829
      .local v6, hitFirst:Z
      .local v7, hitLast:Z
      .local v10, rowId:J
***************
*** 560,572 ****
--- 649,665 ----
  
      if-nez v7, :cond_9
  
+     .line 831
      :cond_3
      add-int/lit8 v8, v8, 0x1
  
+     .line 832
      move/from16 p3, v8
  
+     .line 834
      const/4 v9, 0x0
  
+     .line 814
      .end local v6           #hitFirst:Z
      .end local v7           #hitLast:Z
      .end local v10           #rowId:J
***************
*** 580,591 ****
--- 673,686 ----
  
      if-gtz v12, :cond_6
  
+     .line 815
      move/from16 v0, p3
  
      invoke-interface {v1, v0}, Landroid/widget/ExpandableListAdapter;->getGroupId(I)J
  
      move-result-wide v10
  
+     .line 816
      .restart local v10       #rowId:J
      cmp-long v12, v10, p1
  
***************
*** 593,600 ****
--- 688,697 ----
  
      move/from16 v12, p3
  
+     .line 818
      goto :goto_0
  
+     .line 821
      :cond_5
      add-int/lit8 v12, v2, -0x1
  
***************
*** 602,619 ****
--- 699,719 ----
  
      const/4 v7, 0x1
  
+     .line 822
      .restart local v7       #hitLast:Z
      :goto_2
      if-nez v5, :cond_8
  
      const/4 v6, 0x1
  
+     .line 824
      .restart local v6       #hitFirst:Z
      :goto_3
      if-eqz v7, :cond_2
  
      if-eqz v6, :cond_2
  
+     .line 845
      .end local v6           #hitFirst:Z
      .end local v7           #hitLast:Z
      .end local v10           #rowId:J
***************
*** 622,639 ****
--- 722,742 ----
  
      goto :goto_0
  
+     .line 821
      .restart local v10       #rowId:J
      :cond_7
      const/4 v7, 0x0
  
      goto :goto_2
  
+     .line 822
      .restart local v7       #hitLast:Z
      :cond_8
      const/4 v6, 0x0
  
      goto :goto_3
  
+     .line 835
      .restart local v6       #hitFirst:Z
      :cond_9
      if-nez v7, :cond_a
***************
*** 642,652 ****
--- 745,758 ----
  
      if-nez v6, :cond_4
  
+     .line 837
      :cond_a
      add-int/lit8 v5, v5, -0x1
  
+     .line 838
      move/from16 p3, v5
  
+     .line 840
      const/4 v9, 0x1
  
      goto :goto_1
***************
*** 656,661 ****
--- 762,768 ----
      .locals 1
  
      .prologue
+     .line 719
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      return-object v0
***************
*** 665,670 ****
--- 772,778 ----
      .locals 2
  
      .prologue
+     .line 396
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      invoke-interface {v0}, Landroid/widget/ExpandableListAdapter;->getGroupCount()I
***************
*** 691,696 ****
--- 799,805 ----
      .end annotation
  
      .prologue
+     .line 732
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
      return-object v0
***************
*** 700,714 ****
--- 809,826 ----
      .locals 2
  
      .prologue
+     .line 723
      invoke-virtual {p0}, Landroid/widget/ExpandableListConnector;->getAdapter()Landroid/widget/ExpandableListAdapter;
  
      move-result-object v0
  
+     .line 724
      .local v0, adapter:Landroid/widget/ExpandableListAdapter;
      instance-of v1, v0, Landroid/widget/Filterable;
  
      if-eqz v1, :cond_0
  
+     .line 725
      check-cast v0, Landroid/widget/Filterable;
  
      .end local v0           #adapter:Landroid/widget/ExpandableListAdapter;
***************
*** 716,721 ****
--- 828,834 ----
  
      move-result-object v1
  
+     .line 727
      :goto_0
      return-object v1
  
***************
*** 731,759 ****
--- 844,879 ----
      .parameter "pos"
  
      .prologue
+     .line 259
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
      move-object/from16 v20, v0
  
+     .line 260
      .local v20, egml:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/widget/ExpandableListConnector$GroupMetadata;>;"
      invoke-virtual/range {v20 .. v20}, Ljava/util/ArrayList;->size()I
  
      move-result v23
  
+     .line 263
      .local v23, numExpGroups:I
      const/4 v13, 0x0
  
+     .line 264
      .local v13, leftExpGroupIndex:I
      add-int/lit8 v19, v23, -0x1
  
+     .line 265
      .local v19, rightExpGroupIndex:I
      const/16 v22, 0x0
  
+     .line 268
      .local v22, midExpGroupIndex:I
      if-nez v23, :cond_9
  
+     .line 274
      move-object/from16 v0, p1
  
      iget v2, v0, Landroid/widget/ExpandableListPosition;->groupPos:I
***************
*** 780,802 ****
--- 900,926 ----
  
      move/from16 v7, v22
  
+     .line 364
      .end local v22           #midExpGroupIndex:I
      .local v7, midExpGroupIndex:I
      :goto_0
      return-object v2
  
+     .line 283
      :cond_0
      :goto_1
      move/from16 v0, v19
  
      if-gt v13, v0, :cond_5
  
+     .line 284
      sub-int v2, v19, v13
  
      div-int/lit8 v2, v2, 0x2
  
      add-int v7, v2, v13
  
+     .line 285
      move-object/from16 v0, v20
  
      invoke-virtual {v0, v7}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 805,810 ****
--- 929,935 ----
  
      check-cast v6, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 287
      .local v6, midExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      move-object/from16 v0, p1
  
***************
*** 814,823 ****
--- 939,950 ----
  
      if-le v2, v3, :cond_1
  
+     .line 291
      add-int/lit8 v13, v7, 0x1
  
      goto :goto_1
  
+     .line 292
      :cond_1
      move-object/from16 v0, p1
  
***************
*** 827,836 ****
--- 954,965 ----
  
      if-ge v2, v3, :cond_2
  
+     .line 296
      add-int/lit8 v19, v7, -0x1
  
      goto :goto_1
  
+     .line 297
      :cond_2
      move-object/from16 v0, p1
  
***************
*** 840,845 ****
--- 969,975 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 302
      move-object/from16 v0, p1
  
      iget v2, v0, Landroid/widget/ExpandableListPosition;->type:I
***************
*** 848,853 ****
--- 978,984 ----
  
      if-ne v2, v3, :cond_3
  
+     .line 304
      iget v2, v6, Landroid/widget/ExpandableListConnector$GroupMetadata;->flPos:I
  
      move-object/from16 v0, p1
***************
*** 868,873 ****
--- 999,1005 ----
  
      goto :goto_0
  
+     .line 306
      :cond_3
      move-object/from16 v0, p1
  
***************
*** 877,882 ****
--- 1009,1015 ----
  
      if-ne v2, v3, :cond_4
  
+     .line 308
      iget v2, v6, Landroid/widget/ExpandableListConnector$GroupMetadata;->flPos:I
  
      move-object/from16 v0, p1
***************
*** 905,915 ****
--- 1038,1050 ----
  
      goto :goto_0
  
+     .line 312
      :cond_4
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 321
      .end local v6           #midExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      :cond_5
      move-object/from16 v0, p1
***************
*** 920,932 ****
--- 1055,1070 ----
  
      if-eq v2, v3, :cond_6
  
+     .line 323
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 331
      :cond_6
      if-le v13, v7, :cond_7
  
+     .line 342
      add-int/lit8 v2, v13, -0x1
  
      move-object/from16 v0, v20
***************
*** 937,942 ****
--- 1075,1081 ----
  
      check-cast v21, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 343
      .local v21, leftExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      move-object/from16 v0, v21
  
***************
*** 954,959 ****
--- 1093,1099 ----
  
      add-int v8, v2, v3
  
+     .line 347
      .local v8, flPos:I
      move-object/from16 v0, p1
  
***************
*** 975,980 ****
--- 1115,1121 ----
  
      goto/16 :goto_0
  
+     .line 349
      .end local v8           #flPos:I
      .end local v21           #leftExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      :cond_7
***************
*** 982,987 ****
--- 1123,1129 ----
  
      if-ge v0, v7, :cond_8
  
+     .line 357
      add-int/lit8 v19, v19, 0x1
  
      move-object/from16 v0, v20
***************
*** 994,999 ****
--- 1136,1142 ----
  
      check-cast v24, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 358
      .local v24, rightExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      move-object/from16 v0, v24
  
***************
*** 1011,1016 ****
--- 1154,1160 ----
  
      sub-int v8, v2, v3
  
+     .line 361
      .restart local v8       #flPos:I
      move-object/from16 v0, p1
  
***************
*** 1038,1043 ****
--- 1182,1188 ----
  
      goto/16 :goto_0
  
+     .line 364
      .end local v8           #flPos:I
      .end local v24           #rightExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      :cond_8
***************
*** 1060,1069 ****
--- 1205,1216 ----
      .parameter "flatListPos"
  
      .prologue
+     .line 400
      invoke-virtual {p0, p1}, Landroid/widget/ExpandableListConnector;->getUnflattenedPos(I)Landroid/widget/ExpandableListConnector$PositionMetadata;
  
      move-result-object v0
  
+     .line 403
      .local v0, posMetadata:Landroid/widget/ExpandableListConnector$PositionMetadata;
      iget-object v2, v0, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
  
***************
*** 1073,1078 ****
--- 1220,1226 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 404
      iget-object v2, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      iget-object v3, v0, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
***************
*** 1083,1094 ****
--- 1231,1245 ----
  
      move-result-object v1
  
+     .line 414
      .local v1, retValue:Ljava/lang/Object;
      :goto_0
      invoke-virtual {v0}, Landroid/widget/ExpandableListConnector$PositionMetadata;->recycle()V
  
+     .line 416
      return-object v1
  
+     .line 406
      .end local v1           #retValue:Ljava/lang/Object;
      :cond_0
      iget-object v2, v0, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
***************
*** 1099,1104 ****
--- 1250,1256 ----
  
      if-ne v2, v3, :cond_1
  
+     .line 407
      iget-object v2, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      iget-object v3, v0, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
***************
*** 1116,1121 ****
--- 1268,1274 ----
      .restart local v1       #retValue:Ljava/lang/Object;
      goto :goto_0
  
+     .line 411
      .end local v1           #retValue:Ljava/lang/Object;
      :cond_1
      new-instance v2, Ljava/lang/RuntimeException;
***************
*** 1132,1141 ****
--- 1285,1296 ----
      .parameter "flatListPos"
  
      .prologue
+     .line 420
      invoke-virtual {p0, p1}, Landroid/widget/ExpandableListConnector;->getUnflattenedPos(I)Landroid/widget/ExpandableListConnector$PositionMetadata;
  
      move-result-object v4
  
+     .line 421
      .local v4, posMetadata:Landroid/widget/ExpandableListConnector$PositionMetadata;
      iget-object v7, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
***************
*** 1147,1152 ****
--- 1302,1308 ----
  
      move-result-wide v2
  
+     .line 424
      .local v2, groupId:J
      iget-object v7, v4, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
  
***************
*** 1156,1173 ****
--- 1312,1333 ----
  
      if-ne v7, v8, :cond_0
  
+     .line 425
      iget-object v7, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      invoke-interface {v7, v2, v3}, Landroid/widget/ExpandableListAdapter;->getCombinedGroupId(J)J
  
      move-result-wide v5
  
+     .line 435
      .local v5, retValue:J
      :goto_0
      invoke-virtual {v4}, Landroid/widget/ExpandableListConnector$PositionMetadata;->recycle()V
  
+     .line 437
      return-wide v5
  
+     .line 426
      .end local v5           #retValue:J
      :cond_0
      iget-object v7, v4, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
***************
*** 1178,1183 ****
--- 1338,1344 ----
  
      if-ne v7, v8, :cond_1
  
+     .line 427
      iget-object v7, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      iget-object v8, v4, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
***************
*** 1192,1197 ****
--- 1353,1359 ----
  
      move-result-wide v0
  
+     .line 429
      .local v0, childId:J
      iget-object v7, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
***************
*** 1199,1207 ****
--- 1361,1371 ----
  
      move-result-wide v5
  
+     .line 430
      .restart local v5       #retValue:J
      goto :goto_0
  
+     .line 432
      .end local v0           #childId:J
      .end local v5           #retValue:J
      :cond_1
***************
*** 1221,1232 ****
--- 1385,1398 ----
      .prologue
      const/4 v5, 0x2
  
+     .line 464
      invoke-virtual {p0, p1}, Landroid/widget/ExpandableListConnector;->getUnflattenedPos(I)Landroid/widget/ExpandableListConnector$PositionMetadata;
  
      move-result-object v4
  
      iget-object v2, v4, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
  
+     .line 467
      .local v2, pos:Landroid/widget/ExpandableListPosition;
      iget-object v4, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
***************
*** 1234,1261 ****
--- 1400,1433 ----
  
      if-eqz v4, :cond_1
  
+     .line 468
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      check-cast v0, Landroid/widget/HeterogeneousExpandableList;
  
+     .line 470
      .local v0, adapter:Landroid/widget/HeterogeneousExpandableList;
      iget v4, v2, Landroid/widget/ExpandableListPosition;->type:I
  
      if-ne v4, v5, :cond_0
  
+     .line 471
      iget v4, v2, Landroid/widget/ExpandableListPosition;->groupPos:I
  
      invoke-interface {v0, v4}, Landroid/widget/HeterogeneousExpandableList;->getGroupType(I)I
  
      move-result v3
  
+     .line 484
      .end local v0           #adapter:Landroid/widget/HeterogeneousExpandableList;
      .local v3, retValue:I
      :goto_0
      invoke-virtual {v2}, Landroid/widget/ExpandableListPosition;->recycle()V
  
+     .line 486
      return v3
  
+     .line 473
      .end local v3           #retValue:I
      .restart local v0       #adapter:Landroid/widget/HeterogeneousExpandableList;
      :cond_0
***************
*** 1267,1272 ****
--- 1439,1445 ----
  
      move-result v1
  
+     .line 474
      .local v1, childType:I
      invoke-interface {v0}, Landroid/widget/HeterogeneousExpandableList;->getGroupTypeCount()I
  
***************
*** 1277,1282 ****
--- 1450,1456 ----
      .restart local v3       #retValue:I
      goto :goto_0
  
+     .line 477
      .end local v0           #adapter:Landroid/widget/HeterogeneousExpandableList;
      .end local v1           #childType:I
      .end local v3           #retValue:I
***************
*** 1285,1295 ****
--- 1459,1471 ----
  
      if-ne v4, v5, :cond_2
  
+     .line 478
      const/4 v3, 0x0
  
      .restart local v3       #retValue:I
      goto :goto_0
  
+     .line 480
      .end local v3           #retValue:I
      :cond_2
      const/4 v3, 0x1
***************
*** 1303,1329 ****
--- 1479,1512 ----
      .parameter "flPos"
  
      .prologue
+     .line 109
      move-object/from16 v0, p0
  
      iget-object v13, v0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
+     .line 110
      .local v13, egml:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/widget/ExpandableListConnector$GroupMetadata;>;"
      invoke-virtual {v13}, Ljava/util/ArrayList;->size()I
  
      move-result v17
  
+     .line 113
      .local v17, numExpGroups:I
      const/4 v15, 0x0
  
+     .line 114
      .local v15, leftExpGroupIndex:I
      add-int/lit8 v19, v17, -0x1
  
+     .line 115
      .local v19, rightExpGroupIndex:I
      const/16 v16, 0x0
  
+     .line 118
      .local v16, midExpGroupIndex:I
      if-nez v17, :cond_7
  
+     .line 124
      const/4 v2, 0x2
  
      const/4 v4, -0x1
***************
*** 1342,1370 ****
--- 1525,1558 ----
  
      move/from16 v6, v16
  
+     .line 243
      .end local v16           #midExpGroupIndex:I
      .local v6, midExpGroupIndex:I
      :goto_0
      return-object v1
  
+     .line 140
      :cond_0
      :goto_1
      move/from16 v0, v19
  
      if-gt v15, v0, :cond_4
  
+     .line 141
      sub-int v1, v19, v15
  
      div-int/lit8 v1, v1, 0x2
  
      add-int v6, v1, v15
  
+     .line 144
      invoke-virtual {v13, v6}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v5
  
      check-cast v5, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 146
      .local v5, midExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      iget v1, v5, Landroid/widget/ExpandableListConnector$GroupMetadata;->lastChildFlPos:I
  
***************
*** 1372,1381 ****
--- 1560,1571 ----
  
      if-le v0, v1, :cond_1
  
+     .line 151
      add-int/lit8 v15, v6, 0x1
  
      goto :goto_1
  
+     .line 152
      :cond_1
      iget v1, v5, Landroid/widget/ExpandableListConnector$GroupMetadata;->flPos:I
  
***************
*** 1383,1392 ****
--- 1573,1584 ----
  
      if-ge v0, v1, :cond_2
  
+     .line 157
      add-int/lit8 v19, v6, -0x1
  
      goto :goto_1
  
+     .line 158
      :cond_2
      iget v1, v5, Landroid/widget/ExpandableListConnector$GroupMetadata;->flPos:I
  
***************
*** 1394,1399 ****
--- 1586,1592 ----
  
      if-ne v0, v1, :cond_3
  
+     .line 163
      const/4 v2, 0x2
  
      iget v3, v5, Landroid/widget/ExpandableListConnector$GroupMetadata;->gPos:I
***************
*** 1408,1413 ****
--- 1601,1607 ----
  
      goto :goto_0
  
+     .line 165
      :cond_3
      iget v1, v5, Landroid/widget/ExpandableListConnector$GroupMetadata;->lastChildFlPos:I
  
***************
*** 1415,1426 ****
--- 1609,1622 ----
  
      if-gt v0, v1, :cond_0
  
+     .line 175
      iget v1, v5, Landroid/widget/ExpandableListConnector$GroupMetadata;->flPos:I
  
      add-int/lit8 v1, v1, 0x1
  
      sub-int v4, p1, v1
  
+     .line 176
      .local v4, childPos:I
      const/4 v2, 0x1
  
***************
*** 1434,1450 ****
--- 1630,1650 ----
  
      goto :goto_0
  
+     .line 192
      .end local v4           #childPos:I
      .end local v5           #midExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      :cond_4
      const/4 v12, 0x0
  
+     .line 195
      .local v12, insertPosition:I
      const/4 v9, 0x0
  
+     .line 202
      .local v9, groupPos:I
      if-le v15, v6, :cond_5
  
+     .line 210
      add-int/lit8 v1, v15, -0x1
  
      invoke-virtual {v13, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1453,1461 ****
--- 1653,1663 ----
  
      check-cast v14, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 212
      .local v14, leftExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      move v12, v15
  
+     .line 218
      iget v1, v14, Landroid/widget/ExpandableListConnector$GroupMetadata;->lastChildFlPos:I
  
      sub-int v1, p1, v1
***************
*** 1464,1469 ****
--- 1666,1672 ----
  
      add-int v9, v1, v2
  
+     .line 243
      .end local v14           #leftExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      :goto_2
      const/4 v8, 0x2
***************
*** 1480,1490 ****
--- 1683,1695 ----
  
      goto :goto_0
  
+     .line 220
      :cond_5
      move/from16 v0, v19
  
      if-ge v0, v6, :cond_6
  
+     .line 227
      add-int/lit8 v19, v19, 0x1
  
      move/from16 v0, v19
***************
*** 1495,1503 ****
--- 1700,1710 ----
  
      check-cast v18, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 229
      .local v18, rightExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      move/from16 v12, v19
  
+     .line 237
      move-object/from16 v0, v18
  
      iget v1, v0, Landroid/widget/ExpandableListConnector$GroupMetadata;->gPos:I
***************
*** 1510,1517 ****
--- 1717,1726 ----
  
      sub-int v9, v1, v2
  
+     .line 238
      goto :goto_2
  
+     .line 240
      .end local v18           #rightExpGm:Landroid/widget/ExpandableListConnector$GroupMetadata;
      :cond_6
      new-instance v1, Ljava/lang/RuntimeException;
***************
*** 1543,1552 ****
--- 1752,1763 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 441
      invoke-virtual {p0, p1}, Landroid/widget/ExpandableListConnector;->getUnflattenedPos(I)Landroid/widget/ExpandableListConnector$PositionMetadata;
  
      move-result-object v6
  
+     .line 444
      .local v6, posMetadata:Landroid/widget/ExpandableListConnector$PositionMetadata;
      iget-object v0, v6, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
  
***************
*** 1556,1561 ****
--- 1767,1773 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 445
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      iget-object v1, v6, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
***************
*** 1570,1581 ****
--- 1782,1796 ----
  
      move-result-object v7
  
+     .line 457
      .local v7, retValue:Landroid/view/View;
      :goto_0
      invoke-virtual {v6}, Landroid/widget/ExpandableListConnector$PositionMetadata;->recycle()V
  
+     .line 459
      return-object v7
  
+     .line 447
      .end local v7           #retValue:Landroid/view/View;
      :cond_0
      iget-object v0, v6, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
***************
*** 1584,1595 ****
--- 1799,1812 ----
  
      if-ne v0, v3, :cond_2
  
+     .line 448
      iget-object v0, v6, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
  
      iget v0, v0, Landroid/widget/ExpandableListConnector$GroupMetadata;->lastChildFlPos:I
  
      if-ne v0, p1, :cond_1
  
+     .line 450
      .local v3, isLastChild:Z
      :goto_1
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
***************
*** 1610,1618 ****
--- 1827,1837 ----
  
      move-result-object v7
  
+     .line 452
      .restart local v7       #retValue:Landroid/view/View;
      goto :goto_0
  
+     .line 448
      .end local v3           #isLastChild:Z
      .end local v7           #retValue:Landroid/view/View;
      :cond_1
***************
*** 1620,1625 ****
--- 1839,1845 ----
  
      goto :goto_1
  
+     .line 454
      :cond_2
      new-instance v0, Ljava/lang/RuntimeException;
  
***************
*** 1634,1649 ****
--- 1854,1872 ----
      .locals 3
  
      .prologue
+     .line 491
      iget-object v1, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      instance-of v1, v1, Landroid/widget/HeterogeneousExpandableList;
  
      if-eqz v1, :cond_0
  
+     .line 492
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      check-cast v0, Landroid/widget/HeterogeneousExpandableList;
  
+     .line 494
      .local v0, adapter:Landroid/widget/HeterogeneousExpandableList;
      invoke-interface {v0}, Landroid/widget/HeterogeneousExpandableList;->getGroupTypeCount()I
  
***************
*** 1655,1660 ****
--- 1878,1884 ----
  
      add-int/2addr v1, v2
  
+     .line 496
      .end local v0           #adapter:Landroid/widget/HeterogeneousExpandableList;
      :goto_0
      return v1
***************
*** 1669,1674 ****
--- 1893,1899 ----
      .locals 1
  
      .prologue
+     .line 502
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      invoke-interface {v0}, Landroid/widget/ExpandableListAdapter;->hasStableIds()Z
***************
*** 1682,1691 ****
--- 1907,1918 ----
      .locals 2
  
      .prologue
+     .line 757
      invoke-virtual {p0}, Landroid/widget/ExpandableListConnector;->getAdapter()Landroid/widget/ExpandableListAdapter;
  
      move-result-object v0
  
+     .line 758
      .local v0, adapter:Landroid/widget/ExpandableListAdapter;
      if-eqz v0, :cond_0
  
***************
*** 1707,1718 ****
--- 1934,1947 ----
      .parameter "flatListPos"
  
      .prologue
+     .line 375
      invoke-virtual {p0, p1}, Landroid/widget/ExpandableListConnector;->getUnflattenedPos(I)Landroid/widget/ExpandableListConnector$PositionMetadata;
  
      move-result-object v2
  
      iget-object v0, v2, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
  
+     .line 378
      .local v0, pos:Landroid/widget/ExpandableListPosition;
      iget v2, v0, Landroid/widget/ExpandableListPosition;->type:I
  
***************
*** 1720,1725 ****
--- 1949,1955 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 379
      iget-object v2, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      iget v3, v0, Landroid/widget/ExpandableListPosition;->groupPos:I
***************
*** 1730,1741 ****
--- 1960,1974 ----
  
      move-result v1
  
+     .line 385
      .local v1, retValue:Z
      :goto_0
      invoke-virtual {v0}, Landroid/widget/ExpandableListPosition;->recycle()V
  
+     .line 387
      return v1
  
+     .line 382
      .end local v1           #retValue:Z
      :cond_0
      const/4 v1, 0x1
***************
*** 1749,1754 ****
--- 1982,1988 ----
      .parameter "groupPosition"
  
      .prologue
+     .line 700
      iget-object v2, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
***************
*** 1761,1766 ****
--- 1995,2001 ----
      :goto_0
      if-ltz v1, :cond_1
  
+     .line 701
      iget-object v2, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
      invoke-virtual {v2, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1769,1791 ****
--- 2004,2031 ----
  
      check-cast v0, Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 703
      .local v0, groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
      iget v2, v0, Landroid/widget/ExpandableListConnector$GroupMetadata;->gPos:I
  
      if-ne v2, p1, :cond_0
  
+     .line 704
      const/4 v2, 0x1
  
+     .line 708
      .end local v0           #groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
      :goto_1
      return v2
  
+     .line 700
      .restart local v0       #groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
      :cond_0
      add-int/lit8 v1, v1, -0x1
  
      goto :goto_0
  
+     .line 708
      .end local v0           #groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
      :cond_1
      const/4 v2, 0x0
***************
*** 1798,1820 ****
--- 2038,2065 ----
      .parameter "expandableListAdapter"
  
      .prologue
+     .line 87
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      if-eqz v0, :cond_0
  
+     .line 88
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
      iget-object v1, p0, Landroid/widget/ExpandableListConnector;->mDataSetObserver:Landroid/database/DataSetObserver;
  
      invoke-interface {v0, v1}, Landroid/widget/ExpandableListAdapter;->unregisterDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 91
      :cond_0
      iput-object p1, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
+     .line 92
      iget-object v0, p0, Landroid/widget/ExpandableListConnector;->mDataSetObserver:Landroid/database/DataSetObserver;
  
      invoke-interface {p1, v0}, Landroid/widget/ExpandableListAdapter;->registerDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 93
      return-void
  .end method
  
***************
*** 1832,1837 ****
--- 2077,2083 ----
      .end annotation
  
      .prologue
+     .line 737
      .local p1, expandedGroupMetadataList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/widget/ExpandableListConnector$GroupMetadata;>;"
      if-eqz p1, :cond_0
  
***************
*** 1839,1848 ****
--- 2085,2096 ----
  
      if-nez v2, :cond_1
  
+     .line 753
      :cond_0
      :goto_0
      return-void
  
+     .line 743
      :cond_1
      iget-object v2, p0, Landroid/widget/ExpandableListConnector;->mExpandableListAdapter:Landroid/widget/ExpandableListAdapter;
  
***************
*** 1850,1855 ****
--- 2098,2104 ----
  
      move-result v1
  
+     .line 744
      .local v1, numGroups:I
      invoke-virtual {p1}, Ljava/util/ArrayList;->size()I
  
***************
*** 1861,1866 ****
--- 2110,2116 ----
      :goto_1
      if-ltz v0, :cond_2
  
+     .line 745
      invoke-virtual {p1, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
  
      move-result-object v2
***************
*** 1871,1883 ****
--- 2121,2136 ----
  
      if-ge v2, v1, :cond_0
  
+     .line 744
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_1
  
+     .line 751
      :cond_2
      iput-object p1, p0, Landroid/widget/ExpandableListConnector;->mExpGroupMetadataList:Ljava/util/ArrayList;
  
+     .line 752
      const/4 v2, 0x1
  
      const/4 v3, 0x0
***************
*** 1892,1898 ****
--- 2145,2153 ----
      .parameter "maxExpGroupCount"
  
      .prologue
+     .line 715
      iput p1, p0, Landroid/widget/ExpandableListConnector;->mMaxExpGroupCount:I
  
+     .line 716
      return-void
  .end method
