*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ExpandableListConnector$PositionMetadata.smali	2012-03-23 19:35:57.560636502 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ExpandableListConnector$PositionMetadata.smali.original	2012-03-23 19:34:44.921311926 -0500
***************
*** 42,47 ****
--- 42,48 ----
      .locals 2
  
      .prologue
+     .line 951
      new-instance v0, Ljava/util/ArrayList;
  
      const/4 v1, 0x5
***************
*** 57,64 ****
--- 58,67 ----
      .locals 0
  
      .prologue
+     .line 982
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 983
      return-void
  .end method
  
***************
*** 66,75 ****
--- 69,80 ----
      .locals 4
  
      .prologue
+     .line 996
      sget-object v2, Landroid/widget/ExpandableListConnector$PositionMetadata;->sPool:Ljava/util/ArrayList;
  
      monitor-enter v2
  
+     .line 997
      :try_start_0
      sget-object v1, Landroid/widget/ExpandableListConnector$PositionMetadata;->sPool:Ljava/util/ArrayList;
  
***************
*** 79,84 ****
--- 84,90 ----
  
      if-lez v1, :cond_0
  
+     .line 998
      sget-object v1, Landroid/widget/ExpandableListConnector$PositionMetadata;->sPool:Ljava/util/ArrayList;
  
      const/4 v3, 0x0
***************
*** 89,105 ****
--- 95,115 ----
  
      check-cast v0, Landroid/widget/ExpandableListConnector$PositionMetadata;
  
+     .line 1002
      .local v0, pm:Landroid/widget/ExpandableListConnector$PositionMetadata;
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1003
      invoke-direct {v0}, Landroid/widget/ExpandableListConnector$PositionMetadata;->resetState()V
  
+     .line 1004
      .end local v0           #pm:Landroid/widget/ExpandableListConnector$PositionMetadata;
      :goto_0
      return-object v0
  
+     .line 1000
      :cond_0
      :try_start_1
      new-instance v0, Landroid/widget/ExpandableListConnector$PositionMetadata;
***************
*** 110,115 ****
--- 120,126 ----
  
      goto :goto_0
  
+     .line 1002
      :catchall_0
      move-exception v1
  
***************
*** 130,139 ****
--- 141,152 ----
      .parameter "groupInsertIndex"
  
      .prologue
+     .line 987
      invoke-static {}, Landroid/widget/ExpandableListConnector$PositionMetadata;->getRecycledOrCreate()Landroid/widget/ExpandableListConnector$PositionMetadata;
  
      move-result-object v0
  
+     .line 988
      .local v0, pm:Landroid/widget/ExpandableListConnector$PositionMetadata;
      invoke-static {p1, p2, p3, p0}, Landroid/widget/ExpandableListPosition;->obtain(IIII)Landroid/widget/ExpandableListPosition;
  
***************
*** 141,150 ****
--- 154,166 ----
  
      iput-object v1, v0, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
  
+     .line 989
      iput-object p4, v0, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 990
      iput p5, v0, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupInsertIndex:I
  
+     .line 991
      return-object v0
  .end method
  
***************
*** 154,167 ****
--- 170,187 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 974
      iput-object v0, p0, Landroid/widget/ExpandableListConnector$PositionMetadata;->position:Landroid/widget/ExpandableListPosition;
  
+     .line 975
      iput-object v0, p0, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
  
+     .line 976
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupInsertIndex:I
  
+     .line 977
      return-void
  .end method
  
***************
*** 171,176 ****
--- 191,197 ----
      .locals 1
  
      .prologue
+     .line 1022
      iget-object v0, p0, Landroid/widget/ExpandableListConnector$PositionMetadata;->groupMetadata:Landroid/widget/ExpandableListConnector$GroupMetadata;
  
      if-eqz v0, :cond_0
***************
*** 190,199 ****
--- 211,222 ----
      .locals 3
  
      .prologue
+     .line 1008
      sget-object v1, Landroid/widget/ExpandableListConnector$PositionMetadata;->sPool:Ljava/util/ArrayList;
  
      monitor-enter v1
  
+     .line 1009
      :try_start_0
      sget-object v0, Landroid/widget/ExpandableListConnector$PositionMetadata;->sPool:Ljava/util/ArrayList;
  
***************
*** 205,219 ****
--- 228,246 ----
  
      if-ge v0, v2, :cond_0
  
+     .line 1010
      sget-object v0, Landroid/widget/ExpandableListConnector$PositionMetadata;->sPool:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p0}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 1012
      :cond_0
      monitor-exit v1
  
+     .line 1013
      return-void
  
+     .line 1012
      :catchall_0
      move-exception v0
  
