*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ExpandableListPosition.smali	2012-03-23 19:35:49.682492843 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ExpandableListPosition.smali.original	2012-03-23 19:34:44.858310778 -0500
***************
*** 37,42 ****
--- 37,43 ----
      .locals 2
  
      .prologue
+     .line 31
      new-instance v0, Ljava/util/ArrayList;
  
      const/4 v1, 0x5
***************
*** 52,59 ****
--- 53,62 ----
      .locals 0
  
      .prologue
+     .line 73
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 74
      return-void
  .end method
  
***************
*** 61,70 ****
--- 64,75 ----
      .locals 4
  
      .prologue
+     .line 117
      sget-object v2, Landroid/widget/ExpandableListPosition;->sPool:Ljava/util/ArrayList;
  
      monitor-enter v2
  
+     .line 118
      :try_start_0
      sget-object v1, Landroid/widget/ExpandableListPosition;->sPool:Ljava/util/ArrayList;
  
***************
*** 74,79 ****
--- 79,85 ----
  
      if-lez v1, :cond_0
  
+     .line 119
      sget-object v1, Landroid/widget/ExpandableListPosition;->sPool:Ljava/util/ArrayList;
  
      const/4 v3, 0x0
***************
*** 84,100 ****
--- 90,110 ----
  
      check-cast v0, Landroid/widget/ExpandableListPosition;
  
+     .line 123
      .local v0, elp:Landroid/widget/ExpandableListPosition;
      monitor-exit v2
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 124
      invoke-direct {v0}, Landroid/widget/ExpandableListPosition;->resetState()V
  
+     .line 125
      .end local v0           #elp:Landroid/widget/ExpandableListPosition;
      :goto_0
      return-object v0
  
+     .line 121
      :cond_0
      :try_start_1
      new-instance v0, Landroid/widget/ExpandableListPosition;
***************
*** 105,110 ****
--- 115,121 ----
  
      goto :goto_0
  
+     .line 123
      :catchall_0
      move-exception v1
  
***************
*** 123,141 ****
--- 134,158 ----
      .parameter "flatListPos"
  
      .prologue
+     .line 107
      invoke-static {}, Landroid/widget/ExpandableListPosition;->getRecycledOrCreate()Landroid/widget/ExpandableListPosition;
  
      move-result-object v0
  
+     .line 108
      .local v0, elp:Landroid/widget/ExpandableListPosition;
      iput p0, v0, Landroid/widget/ExpandableListPosition;->type:I
  
+     .line 109
      iput p1, v0, Landroid/widget/ExpandableListPosition;->groupPos:I
  
+     .line 110
      iput p2, v0, Landroid/widget/ExpandableListPosition;->childPos:I
  
+     .line 111
      iput p3, v0, Landroid/widget/ExpandableListPosition;->flatListPos:I
  
+     .line 112
      return-object v0
  .end method
  
***************
*** 145,150 ****
--- 162,168 ----
      .parameter "childPosition"
  
      .prologue
+     .line 86
      const/4 v0, 0x1
  
      const/4 v1, 0x0
***************
*** 163,168 ****
--- 181,187 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 82
      const/4 v0, 0x2
  
      invoke-static {v0, p0, v1, v1}, Landroid/widget/ExpandableListPosition;->obtain(IIII)Landroid/widget/ExpandableListPosition;
***************
*** 179,200 ****
--- 198,224 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 90
      const-wide v1, 0xffffffffL
  
      cmp-long v1, p0, v1
  
      if-nez v1, :cond_0
  
+     .line 91
      const/4 v0, 0x0
  
+     .line 103
      :goto_0
      return-object v0
  
+     .line 94
      :cond_0
      invoke-static {}, Landroid/widget/ExpandableListPosition;->getRecycledOrCreate()Landroid/widget/ExpandableListPosition;
  
      move-result-object v0
  
+     .line 95
      .local v0, elp:Landroid/widget/ExpandableListPosition;
      invoke-static {p0, p1}, Landroid/widget/ExpandableListView;->getPackedPositionGroup(J)I
  
***************
*** 202,215 ****
--- 226,242 ----
  
      iput v1, v0, Landroid/widget/ExpandableListPosition;->groupPos:I
  
+     .line 96
      invoke-static {p0, p1}, Landroid/widget/ExpandableListView;->getPackedPositionType(J)I
  
      move-result v1
  
      if-ne v1, v3, :cond_1
  
+     .line 98
      iput v3, v0, Landroid/widget/ExpandableListPosition;->type:I
  
+     .line 99
      invoke-static {p0, p1}, Landroid/widget/ExpandableListView;->getPackedPositionChild(J)I
  
      move-result v1
***************
*** 218,223 ****
--- 245,251 ----
  
      goto :goto_0
  
+     .line 101
      :cond_1
      const/4 v1, 0x2
  
***************
*** 232,245 ****
--- 260,278 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 67
      iput v0, p0, Landroid/widget/ExpandableListPosition;->groupPos:I
  
+     .line 68
      iput v0, p0, Landroid/widget/ExpandableListPosition;->childPos:I
  
+     .line 69
      iput v0, p0, Landroid/widget/ExpandableListPosition;->flatListPos:I
  
+     .line 70
      iput v0, p0, Landroid/widget/ExpandableListPosition;->type:I
  
+     .line 71
      return-void
  .end method
  
***************
*** 249,254 ****
--- 282,288 ----
      .locals 2
  
      .prologue
+     .line 77
      iget v0, p0, Landroid/widget/ExpandableListPosition;->type:I
  
      const/4 v1, 0x1
***************
*** 263,268 ****
--- 297,303 ----
  
      move-result-wide v0
  
+     .line 78
      :goto_0
      return-wide v0
  
***************
*** 280,289 ****
--- 315,326 ----
      .locals 3
  
      .prologue
+     .line 129
      sget-object v1, Landroid/widget/ExpandableListPosition;->sPool:Ljava/util/ArrayList;
  
      monitor-enter v1
  
+     .line 130
      :try_start_0
      sget-object v0, Landroid/widget/ExpandableListPosition;->sPool:Ljava/util/ArrayList;
  
***************
*** 295,309 ****
--- 332,350 ----
  
      if-ge v0, v2, :cond_0
  
+     .line 131
      sget-object v0, Landroid/widget/ExpandableListPosition;->sPool:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p0}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 133
      :cond_0
      monitor-exit v1
  
+     .line 134
      return-void
  
+     .line 133
      :catchall_0
      move-exception v0
  
