*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/FastScroller$ScrollFade.smali	2012-03-23 19:35:52.335541222 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/FastScroller$ScrollFade.smali.original	2012-03-23 19:34:44.879311160 -0500
***************
*** 37,42 ****
--- 37,43 ----
      .parameter
  
      .prologue
+     .line 836
      iput-object p1, p0, Landroid/widget/FastScroller$ScrollFade;->this$0:Landroid/widget/FastScroller;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 52,57 ****
--- 53,59 ----
      .prologue
      const-wide/16 v7, 0xd0
  
+     .line 850
      iget-object v3, p0, Landroid/widget/FastScroller$ScrollFade;->this$0:Landroid/widget/FastScroller;
  
      invoke-virtual {v3}, Landroid/widget/FastScroller;->getState()I
***************
*** 62,77 ****
--- 64,83 ----
  
      if-eq v3, v4, :cond_0
  
+     .line 851
      const/16 v0, 0xd0
  
+     .line 860
      :goto_0
      return v0
  
+     .line 854
      :cond_0
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v1
  
+     .line 855
      .local v1, now:J
      iget-wide v3, p0, Landroid/widget/FastScroller$ScrollFade;->mStartTime:J
  
***************
*** 83,93 ****
--- 89,101 ----
  
      if-lez v3, :cond_1
  
+     .line 856
      const/4 v0, 0x0
  
      .local v0, alpha:I
      goto :goto_0
  
+     .line 858
      .end local v0           #alpha:I
      :cond_1
      iget-wide v3, p0, Landroid/widget/FastScroller$ScrollFade;->mStartTime:J
***************
*** 112,117 ****
--- 120,126 ----
      .locals 2
  
      .prologue
+     .line 864
      iget-object v0, p0, Landroid/widget/FastScroller$ScrollFade;->this$0:Landroid/widget/FastScroller;
  
      invoke-virtual {v0}, Landroid/widget/FastScroller;->getState()I
***************
*** 122,132 ****
--- 131,144 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 865
      invoke-virtual {p0}, Landroid/widget/FastScroller$ScrollFade;->startFade()V
  
+     .line 874
      :goto_0
      return-void
  
+     .line 869
      :cond_0
      invoke-virtual {p0}, Landroid/widget/FastScroller$ScrollFade;->getAlpha()I
  
***************
*** 134,139 ****
--- 146,152 ----
  
      if-lez v0, :cond_1
  
+     .line 870
      iget-object v0, p0, Landroid/widget/FastScroller$ScrollFade;->this$0:Landroid/widget/FastScroller;
  
      iget-object v0, v0, Landroid/widget/FastScroller;->mList:Landroid/widget/AbsListView;
***************
*** 142,147 ****
--- 155,161 ----
  
      goto :goto_0
  
+     .line 872
      :cond_1
      iget-object v0, p0, Landroid/widget/FastScroller$ScrollFade;->this$0:Landroid/widget/FastScroller;
  
***************
*** 156,176 ****
--- 170,194 ----
      .locals 2
  
      .prologue
+     .line 844
      const-wide/16 v0, 0xc8
  
      iput-wide v0, p0, Landroid/widget/FastScroller$ScrollFade;->mFadeDuration:J
  
+     .line 845
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/widget/FastScroller$ScrollFade;->mStartTime:J
  
+     .line 846
      iget-object v0, p0, Landroid/widget/FastScroller$ScrollFade;->this$0:Landroid/widget/FastScroller;
  
      const/4 v1, 0x4
  
      invoke-virtual {v0, v1}, Landroid/widget/FastScroller;->setState(I)V
  
+     .line 847
      return-void
  .end method
