*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Filter$FilterResults.smali	2012-03-23 19:35:51.045517699 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Filter$FilterResults.smali.original	2012-03-23 19:34:44.869310978 -0500
***************
*** 25,31 ****
--- 25,33 ----
      .locals 0
  
      .prologue
+     .line 181
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 183
      return-void
  .end method
