*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Filter.smali	2012-03-23 19:35:50.747512265 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Filter.smali.original	2012-03-23 19:34:44.866310924 -0500
***************
*** 42,55 ****
--- 42,58 ----
      .locals 2
  
      .prologue
+     .line 56
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 51
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      iput-object v0, p0, Landroid/widget/Filter;->mLock:Ljava/lang/Object;
  
+     .line 57
      new-instance v0, Landroid/widget/Filter$ResultsHandler;
  
      const/4 v1, 0x0
***************
*** 58,63 ****
--- 61,67 ----
  
      iput-object v0, p0, Landroid/widget/Filter;->mResultHandler:Landroid/os/Handler;
  
+     .line 58
      return-void
  .end method
  
***************
*** 66,71 ****
--- 70,76 ----
      .parameter "x0"
  
      .prologue
+     .line 39
      iget-object v0, p0, Landroid/widget/Filter;->mResultHandler:Landroid/os/Handler;
  
      return-object v0
***************
*** 76,81 ****
--- 81,87 ----
      .parameter "x0"
  
      .prologue
+     .line 39
      iget-object v0, p0, Landroid/widget/Filter;->mLock:Ljava/lang/Object;
  
      return-object v0
***************
*** 86,91 ****
--- 92,98 ----
      .parameter "x0"
  
      .prologue
+     .line 39
      iget-object v0, p0, Landroid/widget/Filter;->mThreadHandler:Landroid/os/Handler;
  
      return-object v0
***************
*** 97,102 ****
--- 104,110 ----
      .parameter "x1"
  
      .prologue
+     .line 39
      iput-object p1, p0, Landroid/widget/Filter;->mThreadHandler:Landroid/os/Handler;
  
      return-object p1
***************
*** 109,114 ****
--- 117,123 ----
      .parameter "resultValue"
  
      .prologue
+     .line 173
      if-nez p1, :cond_0
  
      const-string v0, ""
***************
*** 129,138 ****
--- 138,149 ----
      .parameter "constraint"
  
      .prologue
+     .line 84
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Landroid/widget/Filter;->filter(Ljava/lang/CharSequence;Landroid/widget/Filter$FilterListener;)V
  
+     .line 85
      return-void
  .end method
  
***************
*** 144,158 ****
--- 155,172 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 102
      iget-object v6, p0, Landroid/widget/Filter;->mLock:Ljava/lang/Object;
  
      monitor-enter v6
  
+     .line 103
      :try_start_0
      iget-object v7, p0, Landroid/widget/Filter;->mThreadHandler:Landroid/os/Handler;
  
      if-nez v7, :cond_0
  
+     .line 104
      new-instance v4, Landroid/os/HandlerThread;
  
      const-string v7, "Filter"
***************
*** 161,169 ****
--- 175,185 ----
  
      invoke-direct {v4, v7, v8}, Landroid/os/HandlerThread;-><init>(Ljava/lang/String;I)V
  
+     .line 106
      .local v4, thread:Landroid/os/HandlerThread;
      invoke-virtual {v4}, Landroid/os/HandlerThread;->start()V
  
+     .line 107
      new-instance v7, Landroid/widget/Filter$RequestHandler;
  
      invoke-virtual {v4}, Landroid/os/HandlerThread;->getLooper()Landroid/os/Looper;
***************
*** 174,179 ****
--- 190,196 ----
  
      iput-object v7, p0, Landroid/widget/Filter;->mThreadHandler:Landroid/os/Handler;
  
+     .line 110
      .end local v4           #thread:Landroid/os/HandlerThread;
      :cond_0
      iget-object v7, p0, Landroid/widget/Filter;->mDelayer:Landroid/widget/Filter$Delayer;
***************
*** 182,187 ****
--- 199,205 ----
  
      const-wide/16 v1, 0x0
  
+     .line 112
      .local v1, delay:J
      :goto_0
      iget-object v7, p0, Landroid/widget/Filter;->mThreadHandler:Landroid/os/Handler;
***************
*** 192,197 ****
--- 210,216 ----
  
      move-result-object v3
  
+     .line 114
      .local v3, message:Landroid/os/Message;
      new-instance v0, Landroid/widget/Filter$RequestArguments;
  
***************
*** 199,204 ****
--- 218,224 ----
  
      invoke-direct {v0, v7}, Landroid/widget/Filter$RequestArguments;-><init>(Landroid/widget/Filter$1;)V
  
+     .line 117
      .local v0, args:Landroid/widget/Filter$RequestArguments;
      if-eqz p1, :cond_1
  
***************
*** 209,238 ****
--- 229,266 ----
      :cond_1
      iput-object v5, v0, Landroid/widget/Filter$RequestArguments;->constraint:Ljava/lang/CharSequence;
  
+     .line 118
      iput-object p2, v0, Landroid/widget/Filter$RequestArguments;->listener:Landroid/widget/Filter$FilterListener;
  
+     .line 119
      iput-object v0, v3, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 121
      iget-object v5, p0, Landroid/widget/Filter;->mThreadHandler:Landroid/os/Handler;
  
      const v7, -0x2f2f0ff3
  
      invoke-virtual {v5, v7}, Landroid/os/Handler;->removeMessages(I)V
  
+     .line 122
      iget-object v5, p0, Landroid/widget/Filter;->mThreadHandler:Landroid/os/Handler;
  
      const v7, -0x21524111
  
      invoke-virtual {v5, v7}, Landroid/os/Handler;->removeMessages(I)V
  
+     .line 123
      iget-object v5, p0, Landroid/widget/Filter;->mThreadHandler:Landroid/os/Handler;
  
      invoke-virtual {v5, v3, v1, v2}, Landroid/os/Handler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 124
      monitor-exit v6
  
+     .line 125
      return-void
  
+     .line 110
      .end local v0           #args:Landroid/widget/Filter$RequestArguments;
      .end local v1           #delay:J
      .end local v3           #message:Landroid/os/Message;
***************
*** 245,250 ****
--- 273,279 ----
  
      goto :goto_0
  
+     .line 124
      :catchall_0
      move-exception v5
  
***************
*** 266,282 ****
--- 295,316 ----
      .parameter "delayer"
  
      .prologue
+     .line 69
      iget-object v1, p0, Landroid/widget/Filter;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 70
      :try_start_0
      iput-object p1, p0, Landroid/widget/Filter;->mDelayer:Landroid/widget/Filter$Delayer;
  
+     .line 71
      monitor-exit v1
  
+     .line 72
      return-void
  
+     .line 71
      :catchall_0
      move-exception v0
  
