*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Filter$ResultsHandler.smali	2012-03-23 19:35:46.510435002 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Filter$ResultsHandler.smali.original	2012-03-23 19:34:44.833310322 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 267
      iput-object p1, p0, Landroid/widget/Filter$ResultsHandler;->this$0:Landroid/widget/Filter;
  
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 267
      invoke-direct {p0, p1}, Landroid/widget/Filter$ResultsHandler;-><init>(Landroid/widget/Filter;)V
  
      return-void
***************
*** 49,58 ****
--- 51,62 ----
      .parameter "msg"
  
      .prologue
+     .line 280
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Landroid/widget/Filter$RequestArguments;
  
+     .line 282
      .local v0, args:Landroid/widget/Filter$RequestArguments;
      iget-object v2, p0, Landroid/widget/Filter$ResultsHandler;->this$0:Landroid/widget/Filter;
  
***************
*** 62,71 ****
--- 66,77 ----
  
      invoke-virtual {v2, v3, v4}, Landroid/widget/Filter;->publishResults(Ljava/lang/CharSequence;Landroid/widget/Filter$FilterResults;)V
  
+     .line 283
      iget-object v2, v0, Landroid/widget/Filter$RequestArguments;->listener:Landroid/widget/Filter$FilterListener;
  
      if-eqz v2, :cond_0
  
+     .line 284
      iget-object v2, v0, Landroid/widget/Filter$RequestArguments;->results:Landroid/widget/Filter$FilterResults;
  
      if-eqz v2, :cond_1
***************
*** 74,89 ****
--- 80,98 ----
  
      iget v1, v2, Landroid/widget/Filter$FilterResults;->count:I
  
+     .line 285
      .local v1, count:I
      :goto_0
      iget-object v2, v0, Landroid/widget/Filter$RequestArguments;->listener:Landroid/widget/Filter$FilterListener;
  
      invoke-interface {v2, v1}, Landroid/widget/Filter$FilterListener;->onFilterComplete(I)V
  
+     .line 287
      .end local v1           #count:I
      :cond_0
      return-void
  
+     .line 284
      :cond_1
      const/4 v1, -0x1
  
