*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/GridLayout$LayoutParams.smali	2012-03-23 19:35:53.786567682 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/GridLayout$LayoutParams.smali.original	2012-03-23 19:34:44.890311360 -0500
***************
*** 62,67 ****
--- 62,68 ----
      .locals 3
  
      .prologue
+     .line 1717
      new-instance v0, Landroid/widget/GridLayout$Interval;
  
      const/high16 v1, -0x8000
***************
*** 72,77 ****
--- 73,79 ----
  
      sput-object v0, Landroid/widget/GridLayout$LayoutParams;->DEFAULT_SPAN:Landroid/widget/GridLayout$Interval;
  
+     .line 1718
      sget-object v0, Landroid/widget/GridLayout$LayoutParams;->DEFAULT_SPAN:Landroid/widget/GridLayout$Interval;
  
      invoke-virtual {v0}, Landroid/widget/GridLayout$Interval;->size()I
***************
*** 87,98 ****
--- 89,102 ----
      .locals 2
  
      .prologue
+     .line 1782
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      sget-object v1, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      invoke-direct {p0, v0, v1}, Landroid/widget/GridLayout$LayoutParams;-><init>(Landroid/widget/GridLayout$Spec;Landroid/widget/GridLayout$Spec;)V
  
+     .line 1783
      return-void
  .end method
  
***************
*** 108,129 ****
--- 112,140 ----
      .parameter "columnSpec"
  
      .prologue
+     .line 1758
      invoke-direct {p0, p1, p2}, Landroid/view/ViewGroup$MarginLayoutParams;-><init>(II)V
  
+     .line 1744
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1750
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1759
      invoke-virtual {p0, p3, p4, p5, p6}, Landroid/widget/GridLayout$LayoutParams;->setMargins(IIII)V
  
+     .line 1760
      iput-object p7, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1761
      iput-object p8, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1762
      return-void
  .end method
  
***************
*** 133,152 ****
--- 144,169 ----
      .parameter "attrs"
  
      .prologue
+     .line 1819
      invoke-direct {p0, p1, p2}, Landroid/view/ViewGroup$MarginLayoutParams;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 1744
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1750
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1820
      invoke-direct {p0, p1, p2}, Landroid/widget/GridLayout$LayoutParams;->reInitSuper(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 1821
      invoke-direct {p0, p1, p2}, Landroid/widget/GridLayout$LayoutParams;->init(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 1822
      return-void
  .end method
  
***************
*** 155,170 ****
--- 172,191 ----
      .parameter "params"
  
      .prologue
+     .line 1791
      invoke-direct {p0, p1}, Landroid/view/ViewGroup$MarginLayoutParams;-><init>(Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 1744
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1750
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1792
      return-void
  .end method
  
***************
*** 173,188 ****
--- 194,213 ----
      .parameter "params"
  
      .prologue
+     .line 1798
      invoke-direct {p0, p1}, Landroid/view/ViewGroup$MarginLayoutParams;-><init>(Landroid/view/ViewGroup$MarginLayoutParams;)V
  
+     .line 1744
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1750
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1799
      return-void
  .end method
  
***************
*** 191,214 ****
--- 216,245 ----
      .parameter "that"
  
      .prologue
+     .line 1805
      invoke-direct {p0, p1}, Landroid/view/ViewGroup$MarginLayoutParams;-><init>(Landroid/view/ViewGroup$MarginLayoutParams;)V
  
+     .line 1744
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1750
      sget-object v0, Landroid/widget/GridLayout$Spec;->UNDEFINED:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1806
      iget-object v0, p1, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1807
      iget-object v0, p1, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1808
      return-void
  .end method
  
***************
*** 222,227 ****
--- 253,259 ----
  
      const/high16 v3, -0x8000
  
+     .line 1773
      move-object v0, p0
  
      move v2, v1
***************
*** 238,243 ****
--- 270,276 ----
  
      invoke-direct/range {v0 .. v8}, Landroid/widget/GridLayout$LayoutParams;-><init>(IIIIIILandroid/widget/GridLayout$Spec;Landroid/widget/GridLayout$Spec;)V
  
+     .line 1776
      return-void
  .end method
  
***************
*** 247,258 ****
--- 280,293 ----
      .parameter "attrs"
  
      .prologue
+     .line 1851
      sget-object v6, Lcom/android/internal/R$styleable;->GridLayout_Layout:[I
  
      invoke-virtual {p1, p2, v6}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[I)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 1853
      .local v0, a:Landroid/content/res/TypedArray;
      const/4 v6, 0x0
  
***************
*** 263,268 ****
--- 298,304 ----
  
      move-result v3
  
+     .line 1855
      .local v3, gravity:I
      const/4 v6, 0x1
  
***************
*** 272,277 ****
--- 308,314 ----
  
      move-result v2
  
+     .line 1856
      .local v2, column:I
      const/4 v6, 0x4
  
***************
*** 281,286 ****
--- 318,324 ----
  
      move-result v1
  
+     .line 1857
      .local v1, colSpan:I
      const/4 v6, 0x1
  
***************
*** 294,299 ****
--- 332,338 ----
  
      iput-object v6, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1859
      const/4 v6, 0x2
  
      const/high16 v7, -0x8000
***************
*** 302,307 ****
--- 341,347 ----
  
      move-result v4
  
+     .line 1860
      .local v4, row:I
      const/4 v6, 0x3
  
***************
*** 311,316 ****
--- 351,357 ----
  
      move-result v5
  
+     .line 1861
      .local v5, rowSpan:I
      const/4 v6, 0x0
  
***************
*** 326,335 ****
--- 367,379 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1863
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 1865
      return-void
  
+     .line 1863
      .end local v1           #colSpan:I
      .end local v2           #column:I
      .end local v3           #gravity:I
***************
*** 349,360 ****
--- 393,406 ----
      .parameter "attrs"
  
      .prologue
+     .line 1836
      sget-object v2, Lcom/android/internal/R$styleable;->ViewGroup_MarginLayout:[I
  
      invoke-virtual {p1, p2, v2}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[I)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 1839
      .local v0, a:Landroid/content/res/TypedArray;
      const/4 v2, 0x2
  
***************
*** 365,370 ****
--- 411,417 ----
  
      move-result v1
  
+     .line 1841
      .local v1, margin:I
      const/4 v2, 0x3
  
***************
*** 374,379 ****
--- 421,427 ----
  
      iput v2, p0, Landroid/widget/GridLayout$LayoutParams;->leftMargin:I
  
+     .line 1842
      const/4 v2, 0x4
  
      invoke-virtual {v0, v2, v1}, Landroid/content/res/TypedArray;->getDimensionPixelSize(II)I
***************
*** 382,387 ****
--- 430,436 ----
  
      iput v2, p0, Landroid/widget/GridLayout$LayoutParams;->topMargin:I
  
+     .line 1843
      const/4 v2, 0x5
  
      invoke-virtual {v0, v2, v1}, Landroid/content/res/TypedArray;->getDimensionPixelSize(II)I
***************
*** 390,395 ****
--- 439,445 ----
  
      iput v2, p0, Landroid/widget/GridLayout$LayoutParams;->rightMargin:I
  
+     .line 1844
      const/4 v2, 0x6
  
      invoke-virtual {v0, v2, v1}, Landroid/content/res/TypedArray;->getDimensionPixelSize(II)I
***************
*** 400,409 ****
--- 450,462 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1846
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 1848
      return-void
  
+     .line 1846
      .end local v1           #margin:I
      :catchall_0
      move-exception v2
***************
*** 424,441 ****
--- 477,497 ----
      .prologue
      const/4 v1, -0x2
  
+     .line 1882
      invoke-virtual {p1, p2, v1}, Landroid/content/res/TypedArray;->getLayoutDimension(II)I
  
      move-result v0
  
      iput v0, p0, Landroid/widget/GridLayout$LayoutParams;->width:I
  
+     .line 1883
      invoke-virtual {p1, p3, v1}, Landroid/content/res/TypedArray;->getLayoutDimension(II)I
  
      move-result v0
  
      iput v0, p0, Landroid/widget/GridLayout$LayoutParams;->height:I
  
+     .line 1884
      return-void
  .end method
  
***************
*** 444,449 ****
--- 500,506 ----
      .parameter "span"
  
      .prologue
+     .line 1891
      iget-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
      invoke-virtual {v0, p1}, Landroid/widget/GridLayout$Spec;->copyWriteSpan(Landroid/widget/GridLayout$Interval;)Landroid/widget/GridLayout$Spec;
***************
*** 452,457 ****
--- 509,515 ----
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1892
      return-void
  .end method
  
***************
*** 460,465 ****
--- 518,524 ----
      .parameter "gravity"
  
      .prologue
+     .line 1876
      iget-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
      const/4 v1, 0x0
***************
*** 474,479 ****
--- 533,539 ----
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1877
      iget-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
      const/4 v1, 0x1
***************
*** 488,493 ****
--- 548,554 ----
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1878
      return-void
  .end method
  
***************
*** 496,501 ****
--- 557,563 ----
      .parameter "span"
  
      .prologue
+     .line 1887
      iget-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
      invoke-virtual {v0, p1}, Landroid/widget/GridLayout$Spec;->copyWriteSpan(Landroid/widget/GridLayout$Interval;)Landroid/widget/GridLayout$Spec;
***************
*** 504,508 ****
--- 566,571 ----
  
      iput-object v0, p0, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 1888
      return-void
  .end method
