*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/GridLayout.smali	2012-03-23 19:35:54.977589401 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/GridLayout.smali.original	2012-03-23 19:34:44.900311544 -0500
***************
*** 111,116 ****
--- 111,117 ----
      .locals 1
  
      .prologue
+     .line 207
      const-class v0, Landroid/widget/GridLayout;
  
      invoke-virtual {v0}, Ljava/lang/Class;->getName()Ljava/lang/String;
***************
*** 119,170 ****
--- 120,181 ----
  
      sput-object v0, Landroid/widget/GridLayout;->TAG:Ljava/lang/String;
  
+     .line 2390
      new-instance v0, Landroid/widget/GridLayout$1;
  
      invoke-direct {v0}, Landroid/widget/GridLayout$1;-><init>()V
  
      sput-object v0, Landroid/widget/GridLayout;->UNDEFINED_ALIGNMENT:Landroid/widget/GridLayout$Alignment;
  
+     .line 2396
      new-instance v0, Landroid/widget/GridLayout$2;
  
      invoke-direct {v0}, Landroid/widget/GridLayout$2;-><init>()V
  
      sput-object v0, Landroid/widget/GridLayout;->LEADING:Landroid/widget/GridLayout$Alignment;
  
+     .line 2402
      new-instance v0, Landroid/widget/GridLayout$3;
  
      invoke-direct {v0}, Landroid/widget/GridLayout$3;-><init>()V
  
      sput-object v0, Landroid/widget/GridLayout;->TRAILING:Landroid/widget/GridLayout$Alignment;
  
+     .line 2412
      sget-object v0, Landroid/widget/GridLayout;->LEADING:Landroid/widget/GridLayout$Alignment;
  
      sput-object v0, Landroid/widget/GridLayout;->TOP:Landroid/widget/GridLayout$Alignment;
  
+     .line 2418
      sget-object v0, Landroid/widget/GridLayout;->TRAILING:Landroid/widget/GridLayout$Alignment;
  
      sput-object v0, Landroid/widget/GridLayout;->BOTTOM:Landroid/widget/GridLayout$Alignment;
  
+     .line 2424
      sget-object v0, Landroid/widget/GridLayout;->TRAILING:Landroid/widget/GridLayout$Alignment;
  
      sput-object v0, Landroid/widget/GridLayout;->RIGHT:Landroid/widget/GridLayout$Alignment;
  
+     .line 2430
      sget-object v0, Landroid/widget/GridLayout;->LEADING:Landroid/widget/GridLayout$Alignment;
  
      sput-object v0, Landroid/widget/GridLayout;->LEFT:Landroid/widget/GridLayout$Alignment;
  
+     .line 2437
      new-instance v0, Landroid/widget/GridLayout$4;
  
      invoke-direct {v0}, Landroid/widget/GridLayout$4;-><init>()V
  
      sput-object v0, Landroid/widget/GridLayout;->CENTER:Landroid/widget/GridLayout$Alignment;
  
+     .line 2450
      new-instance v0, Landroid/widget/GridLayout$5;
  
      invoke-direct {v0}, Landroid/widget/GridLayout$5;-><init>()V
  
      sput-object v0, Landroid/widget/GridLayout;->BASELINE:Landroid/widget/GridLayout$Alignment;
  
+     .line 2500
      new-instance v0, Landroid/widget/GridLayout$6;
  
      invoke-direct {v0}, Landroid/widget/GridLayout$6;-><init>()V
***************
*** 179,188 ****
--- 190,201 ----
      .parameter "context"
  
      .prologue
+     .line 278
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/widget/GridLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 279
      return-void
  .end method
  
***************
*** 192,201 ****
--- 205,216 ----
      .parameter "attrs"
  
      .prologue
+     .line 270
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, p2, v0}, Landroid/widget/GridLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 271
      return-void
  .end method
  
***************
*** 212,239 ****
--- 227,262 ----
  
      const/4 v2, 0x0
  
+     .line 247
      invoke-direct {p0, p1, p2, p3}, Landroid/view/ViewGroup;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 233
      new-instance v1, Landroid/widget/GridLayout$Axis;
  
      invoke-direct {v1, p0, v3, v4}, Landroid/widget/GridLayout$Axis;-><init>(Landroid/widget/GridLayout;ZLandroid/widget/GridLayout$1;)V
  
      iput-object v1, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
+     .line 234
      new-instance v1, Landroid/widget/GridLayout$Axis;
  
      invoke-direct {v1, p0, v2, v4}, Landroid/widget/GridLayout$Axis;-><init>(Landroid/widget/GridLayout;ZLandroid/widget/GridLayout$1;)V
  
      iput-object v1, p0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
  
+     .line 235
      iput-boolean v2, p0, Landroid/widget/GridLayout;->layoutParamsValid:Z
  
+     .line 236
      iput v2, p0, Landroid/widget/GridLayout;->orientation:I
  
+     .line 237
      iput-boolean v2, p0, Landroid/widget/GridLayout;->useDefaultMargins:Z
  
+     .line 238
      iput v3, p0, Landroid/widget/GridLayout;->alignmentMode:I
  
+     .line 251
      invoke-virtual {p1}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v1
***************
*** 246,257 ****
--- 269,282 ----
  
      iput v1, p0, Landroid/widget/GridLayout;->defaultGap:I
  
+     .line 252
      sget-object v1, Lcom/android/internal/R$styleable;->GridLayout:[I
  
      invoke-virtual {p1, p2, v1}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[I)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 254
      .local v0, a:Landroid/content/res/TypedArray;
      const/4 v1, 0x1
  
***************
*** 264,269 ****
--- 289,295 ----
  
      invoke-virtual {p0, v1}, Landroid/widget/GridLayout;->setRowCount(I)V
  
+     .line 255
      const/4 v1, 0x3
  
      const/high16 v2, -0x8000
***************
*** 274,279 ****
--- 300,306 ----
  
      invoke-virtual {p0, v1}, Landroid/widget/GridLayout;->setColumnCount(I)V
  
+     .line 256
      const/4 v1, 0x0
  
      const/4 v2, 0x0
***************
*** 284,289 ****
--- 311,317 ----
  
      invoke-virtual {p0, v1}, Landroid/widget/GridLayout;->setOrientation(I)V
  
+     .line 257
      const/4 v1, 0x5
  
      const/4 v2, 0x0
***************
*** 294,299 ****
--- 322,328 ----
  
      invoke-virtual {p0, v1}, Landroid/widget/GridLayout;->setUseDefaultMargins(Z)V
  
+     .line 258
      const/4 v1, 0x6
  
      const/4 v2, 0x1
***************
*** 304,309 ****
--- 333,339 ----
  
      invoke-virtual {p0, v1}, Landroid/widget/GridLayout;->setAlignmentMode(I)V
  
+     .line 259
      const/4 v1, 0x2
  
      const/4 v2, 0x1
***************
*** 314,319 ****
--- 344,350 ----
  
      invoke-virtual {p0, v1}, Landroid/widget/GridLayout;->setRowOrderPreserved(Z)V
  
+     .line 260
      const/4 v1, 0x4
  
      const/4 v2, 0x1
***************
*** 326,335 ****
--- 357,369 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 262
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 264
      return-void
  
+     .line 262
      :catchall_0
      move-exception v1
  
***************
*** 355,360 ****
--- 389,395 ----
      .local p1, b:[Ljava/lang/Object;,"[TT;"
      const/4 v4, 0x0
  
+     .line 547
      invoke-virtual {p0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
  
      move-result-object v1
***************
*** 379,395 ****
--- 414,433 ----
  
      check-cast v0, [Ljava/lang/Object;
  
+     .line 548
      .local v0, result:[Ljava/lang/Object;,"[TT;"
      array-length v1, p0
  
      invoke-static {p0, v4, v0, v4, v1}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 549
      array-length v1, p0
  
      array-length v2, p1
  
      invoke-static {p1, v4, v0, v1, v2}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 550
      return-object v0
  .end method
  
***************
*** 398,403 ****
--- 436,442 ----
      .parameter "flexibility"
  
      .prologue
+     .line 2512
      and-int/lit8 v0, p0, 0x2
  
      if-eqz v0, :cond_0
***************
*** 420,436 ****
--- 459,479 ----
      .parameter "count"
  
      .prologue
+     .line 644
      invoke-virtual {p0}, Landroid/widget/GridLayout$Interval;->size()I
  
      move-result v1
  
+     .line 645
      .local v1, size:I
      if-nez p2, :cond_0
  
+     .line 649
      .end local v1           #size:I
      :goto_0
      return v1
  
+     .line 648
      .restart local v1       #size:I
      :cond_0
      if-eqz p1, :cond_1
***************
*** 441,446 ****
--- 484,490 ----
  
      move-result v0
  
+     .line 649
      .local v0, min:I
      :goto_1
      sub-int v2, p2, v0
***************
*** 451,456 ****
--- 495,501 ----
  
      goto :goto_0
  
+     .line 648
      .end local v0           #min:I
      :cond_1
      const/4 v0, 0x0
***************
*** 468,482 ****
--- 513,530 ----
      .parameter "paint"
  
      .prologue
+     .line 758
      invoke-virtual {p0}, Landroid/widget/GridLayout;->getPaddingLeft()I
  
      move-result v6
  
+     .line 759
      .local v6, dx:I
      invoke-virtual {p0}, Landroid/widget/GridLayout;->getPaddingTop()I
  
      move-result v7
  
+     .line 760
      .local v7, dy:I
      add-int v0, v6, p2
  
***************
*** 500,505 ****
--- 548,554 ----
  
      invoke-virtual/range {v0 .. v5}, Landroid/graphics/Canvas;->drawLine(FFFFLandroid/graphics/Paint;)V
  
+     .line 761
      return-void
  .end method
  
***************
*** 513,518 ****
--- 562,568 ----
      .parameter "paint"
  
      .prologue
+     .line 764
      int-to-float v1, p1
  
      int-to-float v2, p2
***************
*** 531,536 ****
--- 581,587 ----
  
      invoke-virtual/range {v0 .. v5}, Landroid/graphics/Canvas;->drawRect(FFFFLandroid/graphics/Paint;)V
  
+     .line 765
      return-void
  .end method
  
***************
*** 544,557 ****
--- 595,611 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 621
      array-length v2, p0
  
      if-le p3, v2, :cond_1
  
+     .line 629
      :cond_0
      :goto_0
      return v1
  
+     .line 624
      :cond_1
      move v0, p2
  
***************
*** 559,572 ****
--- 613,629 ----
      :goto_1
      if-ge v0, p3, :cond_2
  
+     .line 625
      aget v2, p0, v0
  
      if-gt v2, p1, :cond_0
  
+     .line 624
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 629
      :cond_2
      const/4 v1, 0x1
  
***************
*** 579,609 ****
--- 636,672 ----
      .parameter "horizontal"
  
      .prologue
+     .line 554
      if-eqz p1, :cond_0
  
      const/4 v1, 0x7
  
+     .line 555
      .local v1, mask:I
      :goto_0
      if-eqz p1, :cond_1
  
      const/4 v2, 0x0
  
+     .line 556
      .local v2, shift:I
      :goto_1
      and-int v3, p0, v1
  
      shr-int v0, v3, v2
  
+     .line 557
      .local v0, flags:I
      packed-switch v0, :pswitch_data_0
  
+     .line 567
      :pswitch_0
      sget-object v3, Landroid/widget/GridLayout;->UNDEFINED_ALIGNMENT:Landroid/widget/GridLayout$Alignment;
  
      :goto_2
      return-object v3
  
+     .line 554
      .end local v0           #flags:I
      .end local v1           #mask:I
      .end local v2           #shift:I
***************
*** 612,623 ****
--- 675,688 ----
  
      goto :goto_0
  
+     .line 555
      .restart local v1       #mask:I
      :cond_1
      const/4 v2, 0x4
  
      goto :goto_1
  
+     .line 559
      .restart local v0       #flags:I
      .restart local v2       #shift:I
      :pswitch_1
***************
*** 625,645 ****
--- 690,714 ----
  
      goto :goto_2
  
+     .line 561
      :pswitch_2
      sget-object v3, Landroid/widget/GridLayout;->TRAILING:Landroid/widget/GridLayout$Alignment;
  
      goto :goto_2
  
+     .line 563
      :pswitch_3
      sget-object v3, Landroid/widget/GridLayout;->FILL:Landroid/widget/GridLayout$Alignment;
  
      goto :goto_2
  
+     .line 565
      :pswitch_4
      sget-object v3, Landroid/widget/GridLayout;->CENTER:Landroid/widget/GridLayout$Alignment;
  
      goto :goto_2
  
+     .line 557
      nop
  
      :pswitch_data_0
***************
*** 661,666 ****
--- 730,736 ----
      .parameter "leading"
  
      .prologue
+     .line 573
      invoke-virtual {p1}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
  
      move-result-object v0
***************
*** 669,676 ****
--- 739,748 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 574
      const/4 v0, 0x0
  
+     .line 576
      :goto_0
      return v0
  
***************
*** 690,695 ****
--- 762,768 ----
      .parameter "leading"
  
      .prologue
+     .line 580
      if-eqz p2, :cond_0
  
      const/4 v0, 0x0
***************
*** 717,744 ****
--- 790,823 ----
  
      const/4 v4, 0x0
  
+     .line 584
      iget-boolean v5, p0, Landroid/widget/GridLayout;->useDefaultMargins:Z
  
      if-nez v5, :cond_0
  
+     .line 592
      :goto_0
      return v4
  
+     .line 587
      :cond_0
      if-eqz p3, :cond_2
  
      iget-object v3, p2, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 588
      .local v3, spec:Landroid/widget/GridLayout$Spec;
      :goto_1
      if-eqz p3, :cond_3
  
      iget-object v0, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
+     .line 589
      .local v0, axis:Landroid/widget/GridLayout$Axis;
      :goto_2
      iget-object v2, v3, Landroid/widget/GridLayout$Spec;->span:Landroid/widget/GridLayout$Interval;
  
+     .line 590
      .local v2, span:Landroid/widget/GridLayout$Interval;
      if-eqz p4, :cond_5
  
***************
*** 746,751 ****
--- 825,831 ----
  
      if-nez v5, :cond_4
  
+     .line 592
      .local v1, isAtEdge:Z
      :cond_1
      :goto_3
***************
*** 755,760 ****
--- 835,841 ----
  
      goto :goto_0
  
+     .line 587
      .end local v0           #axis:Landroid/widget/GridLayout$Axis;
      .end local v1           #isAtEdge:Z
      .end local v2           #span:Landroid/widget/GridLayout$Interval;
***************
*** 764,769 ****
--- 845,851 ----
  
      goto :goto_1
  
+     .line 588
      .restart local v3       #spec:Landroid/widget/GridLayout$Spec;
      :cond_3
      iget-object v0, p0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
***************
*** 775,780 ****
--- 857,863 ----
      :cond_4
      move v1, v4
  
+     .line 590
      goto :goto_3
  
      :cond_5
***************
*** 796,801 ****
--- 879,885 ----
      .parameter "c"
  
      .prologue
+     .line 729
      invoke-virtual {p1}, Landroid/view/View;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
  
      move-result-object v0
***************
*** 812,835 ****
--- 896,924 ----
      .parameter "leading"
  
      .prologue
+     .line 604
      iget v5, p0, Landroid/widget/GridLayout;->alignmentMode:I
  
      const/4 v6, 0x1
  
      if-ne v5, v6, :cond_0
  
+     .line 605
      invoke-virtual {p0, p1, p2, p3}, Landroid/widget/GridLayout;->getMargin1(Landroid/view/View;ZZ)I
  
      move-result v5
  
+     .line 612
      :goto_0
      return v5
  
+     .line 607
      :cond_0
      if-eqz p2, :cond_1
  
      iget-object v0, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
+     .line 608
      .local v0, axis:Landroid/widget/GridLayout$Axis;
      :goto_1
      if-eqz p3, :cond_2
***************
*** 838,854 ****
--- 927,946 ----
  
      move-result-object v3
  
+     .line 609
      .local v3, margins:[I
      :goto_2
      invoke-virtual {p0, p1}, Landroid/widget/GridLayout;->getLayoutParams(Landroid/view/View;)Landroid/widget/GridLayout$LayoutParams;
  
      move-result-object v2
  
+     .line 610
      .local v2, lp:Landroid/widget/GridLayout$LayoutParams;
      if-eqz p2, :cond_3
  
      iget-object v4, v2, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 611
      .local v4, spec:Landroid/widget/GridLayout$Spec;
      :goto_3
      if-eqz p3, :cond_4
***************
*** 857,868 ****
--- 949,962 ----
  
      iget v1, v5, Landroid/widget/GridLayout$Interval;->min:I
  
+     .line 612
      .local v1, index:I
      :goto_4
      aget v5, v3, v1
  
      goto :goto_0
  
+     .line 607
      .end local v0           #axis:Landroid/widget/GridLayout$Axis;
      .end local v1           #index:I
      .end local v2           #lp:Landroid/widget/GridLayout$LayoutParams;
***************
*** 873,878 ****
--- 967,973 ----
  
      goto :goto_1
  
+     .line 608
      .restart local v0       #axis:Landroid/widget/GridLayout$Axis;
      :cond_2
      invoke-virtual {v0}, Landroid/widget/GridLayout$Axis;->getTrailingMargins()[I
***************
*** 881,886 ****
--- 976,982 ----
  
      goto :goto_2
  
+     .line 610
      .restart local v2       #lp:Landroid/widget/GridLayout$LayoutParams;
      .restart local v3       #margins:[I
      :cond_3
***************
*** 888,893 ****
--- 984,990 ----
  
      goto :goto_3
  
+     .line 611
      .restart local v4       #spec:Landroid/widget/GridLayout$Spec;
      :cond_4
      iget-object v5, v4, Landroid/widget/GridLayout$Spec;->span:Landroid/widget/GridLayout$Interval;
***************
*** 903,908 ****
--- 1000,1006 ----
      .parameter "horizontal"
  
      .prologue
+     .line 925
      if-eqz p2, :cond_0
  
      invoke-virtual {p1}, Landroid/view/View;->getMeasuredWidth()I
***************
*** 926,931 ****
--- 1024,1030 ----
      .parameter "horizontal"
  
      .prologue
+     .line 617
      const/4 v0, 0x1
  
      invoke-direct {p0, p1, p2, v0}, Landroid/widget/GridLayout;->getMargin(Landroid/view/View;ZZ)I
***************
*** 947,966 ****
--- 1046,1070 ----
      .locals 1
  
      .prologue
+     .line 712
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/GridLayout;->layoutParamsValid:Z
  
+     .line 713
      iget-object v0, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0}, Landroid/widget/GridLayout$Axis;->invalidateStructure()V
  
+     .line 714
      iget-object v0, p0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0}, Landroid/widget/GridLayout$Axis;->invalidateStructure()V
  
+     .line 716
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateValues()V
  
+     .line 717
      return-void
  .end method
  
***************
*** 968,973 ****
--- 1072,1078 ----
      .locals 1
  
      .prologue
+     .line 722
      iget-object v0, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
      if-eqz v0, :cond_0
***************
*** 976,989 ****
--- 1081,1097 ----
  
      if-eqz v0, :cond_0
  
+     .line 723
      iget-object v0, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0}, Landroid/widget/GridLayout$Axis;->invalidateValues()V
  
+     .line 724
      iget-object v0, p0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0}, Landroid/widget/GridLayout$Axis;->invalidateValues()V
  
+     .line 726
      :cond_0
      return-void
  .end method
***************
*** 994,1001 ****
--- 1102,1111 ----
      .parameter "valueIfEmpty"
  
      .prologue
+     .line 538
      move v2, p1
  
+     .line 539
      .local v2, result:I
      const/4 v1, 0x0
  
***************
*** 1006,1021 ****
--- 1116,1134 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 540
      aget v3, p0, v1
  
      invoke-static {v2, v3}, Ljava/lang/Math;->max(II)I
  
      move-result v2
  
+     .line 539
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 542
      :cond_0
      return v2
  .end method
***************
*** 1029,1034 ****
--- 1142,1148 ----
      .parameter "childHeight"
  
      .prologue
+     .line 855
      iget v2, p0, Landroid/widget/GridLayout;->mPaddingLeft:I
  
      iget v3, p0, Landroid/widget/GridLayout;->mPaddingRight:I
***************
*** 1047,1052 ****
--- 1161,1167 ----
  
      move-result v1
  
+     .line 857
      .local v1, childWidthSpec:I
      iget v2, p0, Landroid/widget/GridLayout;->mPaddingTop:I
  
***************
*** 1066,1074 ****
--- 1181,1191 ----
  
      move-result v0
  
+     .line 859
      .local v0, childHeightSpec:I
      invoke-virtual {p1, v1, v0}, Landroid/view/View;->measure(II)V
  
+     .line 860
      return-void
  .end method
  
***************
*** 1079,1084 ****
--- 1196,1202 ----
      .parameter "firstPass"
  
      .prologue
+     .line 863
      const/16 v17, 0x0
  
      .local v17, i:I
***************
*** 1092,1097 ****
--- 1210,1216 ----
  
      if-ge v0, v13, :cond_7
  
+     .line 864
      move-object/from16 v0, p0
  
      move/from16 v1, v17
***************
*** 1100,1105 ****
--- 1219,1225 ----
  
      move-result-object v3
  
+     .line 865
      .local v3, c:Landroid/view/View;
      move-object/from16 v0, p0
  
***************
*** 1109,1120 ****
--- 1229,1242 ----
  
      if-eqz v2, :cond_1
  
+     .line 863
      :cond_0
      :goto_1
      add-int/lit8 v17, v17, 0x1
  
      goto :goto_0
  
+     .line 866
      :cond_1
      move-object/from16 v0, p0
  
***************
*** 1122,1130 ****
--- 1244,1254 ----
  
      move-result-object v19
  
+     .line 867
      .local v19, lp:Landroid/widget/GridLayout$LayoutParams;
      if-eqz p3, :cond_2
  
+     .line 868
      move-object/from16 v0, v19
  
      iget v6, v0, Landroid/widget/GridLayout$LayoutParams;->width:I
***************
*** 1143,1148 ****
--- 1267,1273 ----
  
      goto :goto_1
  
+     .line 870
      :cond_2
      move-object/from16 v0, p0
  
***************
*** 1152,1157 ****
--- 1277,1283 ----
  
      const/16 v16, 0x1
  
+     .line 871
      .local v16, horizontal:Z
      :goto_2
      if-eqz v16, :cond_4
***************
*** 1162,1167 ****
--- 1288,1294 ----
  
      move-object/from16 v21, v0
  
+     .line 872
      .local v21, spec:Landroid/widget/GridLayout$Spec;
      :goto_3
      move-object/from16 v0, v21
***************
*** 1172,1183 ****
--- 1299,1312 ----
  
      if-ne v2, v4, :cond_0
  
+     .line 873
      move-object/from16 v0, v21
  
      iget-object v0, v0, Landroid/widget/GridLayout$Spec;->span:Landroid/widget/GridLayout$Interval;
  
      move-object/from16 v20, v0
  
+     .line 874
      .local v20, span:Landroid/widget/GridLayout$Interval;
      if-eqz v16, :cond_5
  
***************
*** 1185,1196 ****
--- 1314,1327 ----
  
      iget-object v14, v0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
+     .line 875
      .local v14, axis:Landroid/widget/GridLayout$Axis;
      :goto_4
      invoke-virtual {v14}, Landroid/widget/GridLayout$Axis;->getLocations()[I
  
      move-result-object v18
  
+     .line 876
      .local v18, locations:[I
      move-object/from16 v0, v20
  
***************
*** 1206,1211 ****
--- 1337,1343 ----
  
      sub-int v15, v2, v4
  
+     .line 877
      .local v15, cellSize:I
      move-object/from16 v0, p0
  
***************
*** 1217,1225 ****
--- 1349,1359 ----
  
      sub-int v6, v15, v2
  
+     .line 878
      .local v6, viewSize:I
      if-eqz v16, :cond_6
  
+     .line 879
      move-object/from16 v0, v19
  
      iget v7, v0, Landroid/widget/GridLayout$LayoutParams;->height:I
***************
*** 1234,1239 ****
--- 1368,1374 ----
  
      goto :goto_1
  
+     .line 870
      .end local v6           #viewSize:I
      .end local v14           #axis:Landroid/widget/GridLayout$Axis;
      .end local v15           #cellSize:I
***************
*** 1246,1251 ****
--- 1381,1387 ----
  
      goto :goto_2
  
+     .line 871
      .restart local v16       #horizontal:Z
      :cond_4
      move-object/from16 v0, v19
***************
*** 1256,1261 ****
--- 1392,1398 ----
  
      goto :goto_3
  
+     .line 874
      .restart local v20       #span:Landroid/widget/GridLayout$Interval;
      .restart local v21       #spec:Landroid/widget/GridLayout$Spec;
      :cond_5
***************
*** 1265,1270 ****
--- 1402,1408 ----
  
      goto :goto_4
  
+     .line 881
      .restart local v6       #viewSize:I
      .restart local v14       #axis:Landroid/widget/GridLayout$Axis;
      .restart local v15       #cellSize:I
***************
*** 1288,1293 ****
--- 1426,1432 ----
  
      goto/16 :goto_1
  
+     .line 886
      .end local v3           #c:Landroid/view/View;
      .end local v6           #viewSize:I
      .end local v14           #axis:Landroid/widget/GridLayout$Axis;
***************
*** 1309,1316 ****
--- 1448,1457 ----
      .parameter "value"
  
      .prologue
+     .line 633
      array-length v0, p0
  
+     .line 634
      .local v0, length:I
      invoke-static {p1, v0}, Ljava/lang/Math;->min(II)I
  
***************
*** 1322,1327 ****
--- 1463,1469 ----
  
      invoke-static {p0, v1, v2, p3}, Ljava/util/Arrays;->fill([IIII)V
  
+     .line 635
      return-void
  .end method
  
***************
*** 1330,1335 ****
--- 1472,1478 ----
      .parameter "alignment"
  
      .prologue
+     .line 921
      const/high16 v0, -0x8000
  
      if-ne p1, v0, :cond_0
***************
*** 1350,1355 ****
--- 1493,1499 ----
      .parameter "colSpan"
  
      .prologue
+     .line 638
      new-instance v0, Landroid/widget/GridLayout$Interval;
  
      add-int v1, p1, p2
***************
*** 1358,1363 ****
--- 1502,1508 ----
  
      invoke-virtual {p0, v0}, Landroid/widget/GridLayout$LayoutParams;->setRowSpecSpan(Landroid/widget/GridLayout$Interval;)V
  
+     .line 639
      new-instance v0, Landroid/widget/GridLayout$Interval;
  
      add-int v1, p3, p4
***************
*** 1366,1371 ****
--- 1511,1517 ----
  
      invoke-virtual {p0, v0}, Landroid/widget/GridLayout$LayoutParams;->setColumnSpecSpan(Landroid/widget/GridLayout$Interval;)V
  
+     .line 640
      return-void
  .end method
  
***************
*** 1374,1379 ****
--- 1520,1526 ----
      .parameter "start"
  
      .prologue
+     .line 2326
      const/4 v0, 0x1
  
      invoke-static {p0, v0}, Landroid/widget/GridLayout;->spec(II)Landroid/widget/GridLayout$Spec;
***************
*** 1389,1394 ****
--- 1536,1542 ----
      .parameter "size"
  
      .prologue
+     .line 2314
      sget-object v0, Landroid/widget/GridLayout;->UNDEFINED_ALIGNMENT:Landroid/widget/GridLayout$Alignment;
  
      invoke-static {p0, p1, v0}, Landroid/widget/GridLayout;->spec(IILandroid/widget/GridLayout$Alignment;)Landroid/widget/GridLayout$Spec;
***************
*** 1405,1410 ****
--- 1553,1559 ----
      .parameter "alignment"
  
      .prologue
+     .line 2287
      new-instance v0, Landroid/widget/GridLayout$Spec;
  
      const/high16 v1, -0x8000
***************
*** 1438,1443 ****
--- 1587,1593 ----
      .parameter "alignment"
  
      .prologue
+     .line 2301
      const/4 v0, 0x1
  
      invoke-static {p0, v0, p1}, Landroid/widget/GridLayout;->spec(IILandroid/widget/GridLayout$Alignment;)Landroid/widget/GridLayout$Spec;
***************
*** 1451,1456 ****
--- 1601,1607 ----
      .locals 21
  
      .prologue
+     .line 654
      move-object/from16 v0, p0
  
      iget v0, v0, Landroid/widget/GridLayout;->orientation:I
***************
*** 1461,1466 ****
--- 1612,1618 ----
  
      const/4 v5, 0x1
  
+     .line 655
      .local v5, horizontal:Z
      :goto_0
      if-eqz v5, :cond_4
***************
*** 1469,1474 ****
--- 1621,1627 ----
  
      iget-object v3, v0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
+     .line 656
      .local v3, axis:Landroid/widget/GridLayout$Axis;
      :goto_1
      iget v0, v3, Landroid/widget/GridLayout$Axis;->definedCount:I
***************
*** 1485,1500 ****
--- 1638,1657 ----
  
      iget v4, v3, Landroid/widget/GridLayout$Axis;->definedCount:I
  
+     .line 658
      .local v4, count:I
      :goto_2
      const/4 v8, 0x0
  
+     .line 659
      .local v8, major:I
      const/4 v14, 0x0
  
+     .line 660
      .local v14, minor:I
      new-array v13, v4, [I
  
+     .line 662
      .local v13, maxSizes:[I
      const/4 v6, 0x0
  
***************
*** 1507,1512 ****
--- 1664,1670 ----
      :goto_3
      if-ge v6, v2, :cond_d
  
+     .line 663
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v6}, Landroid/widget/GridLayout;->getChildAt(I)Landroid/view/View;
***************
*** 1521,1548 ****
--- 1679,1713 ----
  
      move-result-object v7
  
+     .line 665
      .local v7, lp:Landroid/widget/GridLayout$LayoutParams;
      if-eqz v5, :cond_6
  
      iget-object v11, v7, Landroid/widget/GridLayout$LayoutParams;->rowSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 666
      .local v11, majorSpec:Landroid/widget/GridLayout$Spec;
      :goto_4
      iget-object v9, v11, Landroid/widget/GridLayout$Spec;->span:Landroid/widget/GridLayout$Interval;
  
+     .line 667
      .local v9, majorRange:Landroid/widget/GridLayout$Interval;
      iget-boolean v12, v11, Landroid/widget/GridLayout$Spec;->startDefined:Z
  
+     .line 668
      .local v12, majorWasDefined:Z
      invoke-virtual {v9}, Landroid/widget/GridLayout$Interval;->size()I
  
      move-result v10
  
+     .line 669
      .local v10, majorSpan:I
      if-eqz v12, :cond_0
  
+     .line 670
      iget v8, v9, Landroid/widget/GridLayout$Interval;->min:I
  
+     .line 673
      :cond_0
      if-eqz v5, :cond_7
  
***************
*** 1550,1561 ****
--- 1715,1728 ----
  
      move-object/from16 v17, v0
  
+     .line 674
      .local v17, minorSpec:Landroid/widget/GridLayout$Spec;
      :goto_5
      move-object/from16 v0, v17
  
      iget-object v15, v0, Landroid/widget/GridLayout$Spec;->span:Landroid/widget/GridLayout$Interval;
  
+     .line 675
      .local v15, minorRange:Landroid/widget/GridLayout$Interval;
      move-object/from16 v0, v17
  
***************
*** 1563,1568 ****
--- 1730,1736 ----
  
      move/from16 v18, v0
  
+     .line 676
      .local v18, minorWasDefined:Z
      move/from16 v0, v18
  
***************
*** 1570,1587 ****
--- 1738,1760 ----
  
      move-result v16
  
+     .line 677
      .local v16, minorSpan:I
      if-eqz v18, :cond_1
  
+     .line 678
      iget v14, v15, Landroid/widget/GridLayout$Interval;->min:I
  
+     .line 681
      :cond_1
      if-eqz v4, :cond_b
  
+     .line 683
      if-eqz v12, :cond_2
  
      if-nez v18, :cond_a
  
+     .line 684
      :cond_2
      :goto_6
      add-int v19, v14, v16
***************
*** 1594,1605 ****
--- 1767,1781 ----
  
      if-nez v19, :cond_a
  
+     .line 685
      if-eqz v18, :cond_8
  
+     .line 686
      add-int/lit8 v8, v8, 0x1
  
      goto :goto_6
  
+     .line 654
      .end local v2           #N:I
      .end local v3           #axis:Landroid/widget/GridLayout$Axis;
      .end local v4           #count:I
***************
*** 1622,1627 ****
--- 1798,1804 ----
  
      goto :goto_0
  
+     .line 655
      .restart local v5       #horizontal:Z
      :cond_4
      move-object/from16 v0, p0
***************
*** 1630,1641 ****
--- 1807,1820 ----
  
      goto :goto_1
  
+     .line 656
      .restart local v3       #axis:Landroid/widget/GridLayout$Axis;
      :cond_5
      const/4 v4, 0x0
  
      goto :goto_2
  
+     .line 665
      .restart local v2       #N:I
      .restart local v4       #count:I
      .restart local v6       #i:I
***************
*** 1648,1653 ****
--- 1827,1833 ----
  
      goto :goto_4
  
+     .line 673
      .restart local v9       #majorRange:Landroid/widget/GridLayout$Interval;
      .restart local v10       #majorSpan:I
      .restart local v11       #majorSpec:Landroid/widget/GridLayout$Spec;
***************
*** 1659,1664 ****
--- 1839,1845 ----
  
      goto :goto_5
  
+     .line 688
      .restart local v15       #minorRange:Landroid/widget/GridLayout$Interval;
      .restart local v16       #minorSpan:I
      .restart local v17       #minorSpec:Landroid/widget/GridLayout$Spec;
***************
*** 1670,1686 ****
--- 1851,1871 ----
  
      if-gt v0, v4, :cond_9
  
+     .line 689
      add-int/lit8 v14, v14, 0x1
  
      goto :goto_6
  
+     .line 691
      :cond_9
      const/4 v14, 0x0
  
+     .line 692
      add-int/lit8 v8, v8, 0x1
  
      goto :goto_6
  
+     .line 697
      :cond_a
      add-int v19, v14, v16
  
***************
*** 1692,1711 ****
--- 1877,1901 ----
  
      invoke-static {v13, v14, v0, v1}, Landroid/widget/GridLayout;->procrusteanFill([IIII)V
  
+     .line 700
      :cond_b
      if-eqz v5, :cond_c
  
+     .line 701
      move/from16 v0, v16
  
      invoke-static {v7, v8, v10, v14, v0}, Landroid/widget/GridLayout;->setCellGroup(Landroid/widget/GridLayout$LayoutParams;IIII)V
  
+     .line 706
      :goto_7
      add-int v14, v14, v16
  
+     .line 662
      add-int/lit8 v6, v6, 0x1
  
      goto/16 :goto_3
  
+     .line 703
      :cond_c
      move/from16 v0, v16
  
***************
*** 1713,1718 ****
--- 1903,1909 ----
  
      goto :goto_7
  
+     .line 708
      .end local v7           #lp:Landroid/widget/GridLayout$LayoutParams;
      .end local v9           #majorRange:Landroid/widget/GridLayout$Interval;
      .end local v10           #majorSpan:I
***************
*** 1725,1730 ****
--- 1916,1922 ----
      :cond_d
      invoke-direct/range {p0 .. p0}, Landroid/widget/GridLayout;->invalidateStructure()V
  
+     .line 709
      return-void
  .end method
  
***************
*** 1734,1739 ****
--- 1926,1932 ----
      .locals 1
  
      .prologue
+     .line 147
      invoke-virtual {p0}, Landroid/widget/GridLayout;->generateDefaultLayoutParams()Landroid/widget/GridLayout$LayoutParams;
  
      move-result-object v0
***************
*** 1745,1750 ****
--- 1938,1944 ----
      .locals 1
  
      .prologue
+     .line 742
      new-instance v0, Landroid/widget/GridLayout$LayoutParams;
  
      invoke-direct {v0}, Landroid/widget/GridLayout$LayoutParams;-><init>()V
***************
*** 1757,1762 ****
--- 1951,1957 ----
      .parameter "x0"
  
      .prologue
+     .line 147
      invoke-virtual {p0, p1}, Landroid/widget/GridLayout;->generateLayoutParams(Landroid/util/AttributeSet;)Landroid/widget/GridLayout$LayoutParams;
  
      move-result-object v0
***************
*** 1769,1774 ****
--- 1964,1970 ----
      .parameter "x0"
  
      .prologue
+     .line 147
      invoke-virtual {p0, p1}, Landroid/widget/GridLayout;->generateLayoutParams(Landroid/view/ViewGroup$LayoutParams;)Landroid/widget/GridLayout$LayoutParams;
  
      move-result-object v0
***************
*** 1781,1786 ****
--- 1977,1983 ----
      .parameter "attrs"
  
      .prologue
+     .line 747
      new-instance v0, Landroid/widget/GridLayout$LayoutParams;
  
      invoke-virtual {p0}, Landroid/widget/GridLayout;->getContext()Landroid/content/Context;
***************
*** 1797,1802 ****
--- 1994,2000 ----
      .parameter "p"
  
      .prologue
+     .line 752
      new-instance v0, Landroid/widget/GridLayout$LayoutParams;
  
      invoke-direct {v0, p1}, Landroid/widget/GridLayout$LayoutParams;-><init>(Landroid/view/ViewGroup$LayoutParams;)V
***************
*** 1810,1815 ****
--- 2008,2014 ----
      .parameter "horizontal"
  
      .prologue
+     .line 942
      sget-object v0, Landroid/widget/GridLayout;->UNDEFINED_ALIGNMENT:Landroid/widget/GridLayout$Alignment;
  
      if-eq p1, v0, :cond_0
***************
*** 1836,1841 ****
--- 2035,2041 ----
      .locals 1
  
      .prologue
+     .line 440
      iget v0, p0, Landroid/widget/GridLayout;->alignmentMode:I
  
      return v0
***************
*** 1845,1850 ****
--- 2045,2051 ----
      .locals 1
  
      .prologue
+     .line 362
      iget-object v0, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0}, Landroid/widget/GridLayout$Axis;->getCount()I
***************
*** 1859,1874 ****
--- 2060,2079 ----
      .parameter "c"
  
      .prologue
+     .line 733
      iget-boolean v0, p0, Landroid/widget/GridLayout;->layoutParamsValid:Z
  
      if-nez v0, :cond_0
  
+     .line 734
      invoke-direct {p0}, Landroid/widget/GridLayout;->validateLayoutParams()V
  
+     .line 735
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/GridLayout;->layoutParamsValid:Z
  
+     .line 737
      :cond_0
      invoke-direct {p0, p1}, Landroid/widget/GridLayout;->getLayoutParams1(Landroid/view/View;)Landroid/widget/GridLayout$LayoutParams;
  
***************
*** 1884,1893 ****
--- 2089,2100 ----
      .parameter "leading"
  
      .prologue
+     .line 596
      invoke-virtual {p0, p1}, Landroid/widget/GridLayout;->getLayoutParams(Landroid/view/View;)Landroid/widget/GridLayout$LayoutParams;
  
      move-result-object v0
  
+     .line 597
      .local v0, lp:Landroid/widget/GridLayout$LayoutParams;
      if-eqz p2, :cond_2
  
***************
*** 1895,1900 ****
--- 2102,2108 ----
  
      iget v1, v0, Landroid/widget/GridLayout$LayoutParams;->leftMargin:I
  
+     .line 600
      .local v1, margin:I
      :goto_0
      const/high16 v2, -0x8000
***************
*** 1909,1914 ****
--- 2117,2123 ----
      :cond_0
      return v1
  
+     .line 597
      :cond_1
      iget v1, v0, Landroid/widget/GridLayout$LayoutParams;->rightMargin:I
  
***************
*** 1933,1946 ****
--- 2142,2158 ----
      .parameter "horizontal"
  
      .prologue
+     .line 929
      invoke-virtual {p0, p1}, Landroid/widget/GridLayout;->isGone(Landroid/view/View;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 930
      const/4 v0, 0x0
  
+     .line 932
      :goto_0
      return v0
  
***************
*** 1962,1967 ****
--- 2174,2180 ----
      .locals 1
  
      .prologue
+     .line 293
      iget v0, p0, Landroid/widget/GridLayout;->orientation:I
  
      return v0
***************
*** 1971,1976 ****
--- 2184,2190 ----
      .locals 1
  
      .prologue
+     .line 329
      iget-object v0, p0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0}, Landroid/widget/GridLayout$Axis;->getCount()I
***************
*** 1984,1989 ****
--- 2198,2204 ----
      .locals 1
  
      .prologue
+     .line 393
      iget-boolean v0, p0, Landroid/widget/GridLayout;->useDefaultMargins:Z
  
      return v0
***************
*** 1993,1998 ****
--- 2208,2214 ----
      .locals 1
  
      .prologue
+     .line 510
      iget-object v0, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0}, Landroid/widget/GridLayout$Axis;->isOrderPreserved()Z
***************
*** 2007,2012 ****
--- 2223,2229 ----
      .parameter "c"
  
      .prologue
+     .line 850
      invoke-virtual {p1}, Landroid/view/View;->getVisibility()I
  
      move-result v0
***************
*** 2030,2035 ****
--- 2247,2253 ----
      .locals 1
  
      .prologue
+     .line 474
      iget-object v0, p0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0}, Landroid/widget/GridLayout$Axis;->isOrderPreserved()Z
***************
*** 2045,2054 ****
--- 2263,2275 ----
      .parameter "visibility"
  
      .prologue
+     .line 843
      invoke-super {p0, p1, p2}, Landroid/view/ViewGroup;->onChildVisibilityChanged(Landroid/view/View;I)V
  
+     .line 844
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateStructure()V
  
+     .line 845
      return-void
  .end method
  
***************
*** 2057,2064 ****
--- 2278,2287 ----
      .parameter "canvas"
  
      .prologue
+     .line 769
      invoke-super {p0, p1}, Landroid/view/ViewGroup;->onDraw(Landroid/graphics/Canvas;)V
  
+     .line 813
      return-void
  .end method
  
***************
*** 2071,2101 ****
--- 2294,2331 ----
      .parameter "bottom"
  
      .prologue
+     .line 962
      sub-int v38, p4, p2
  
+     .line 963
      .local v38, targetWidth:I
      sub-int v37, p5, p3
  
+     .line 965
      .local v37, targetHeight:I
      invoke-virtual/range {p0 .. p0}, Landroid/widget/GridLayout;->getPaddingLeft()I
  
      move-result v30
  
+     .line 966
      .local v30, paddingLeft:I
      invoke-virtual/range {p0 .. p0}, Landroid/widget/GridLayout;->getPaddingTop()I
  
      move-result v32
  
+     .line 967
      .local v32, paddingTop:I
      invoke-virtual/range {p0 .. p0}, Landroid/widget/GridLayout;->getPaddingRight()I
  
      move-result v31
  
+     .line 968
      .local v31, paddingRight:I
      invoke-virtual/range {p0 .. p0}, Landroid/widget/GridLayout;->getPaddingBottom()I
  
      move-result v29
  
+     .line 970
      .local v29, paddingBottom:I
      move-object/from16 v0, p0
  
***************
*** 2109,2114 ****
--- 2339,2345 ----
  
      invoke-virtual/range {v48 .. v49}, Landroid/widget/GridLayout$Axis;->layout(I)V
  
+     .line 971
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
***************
*** 2121,2126 ****
--- 2352,2358 ----
  
      invoke-virtual/range {v48 .. v49}, Landroid/widget/GridLayout$Axis;->layout(I)V
  
+     .line 973
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
***************
*** 2131,2136 ****
--- 2363,2369 ----
  
      move-result-object v20
  
+     .line 974
      .local v20, hLocations:[I
      move-object/from16 v0, p0
  
***************
*** 2142,2147 ****
--- 2375,2381 ----
  
      move-result-object v42
  
+     .line 976
      .local v42, vLocations:[I
      const/16 v22, 0x0
  
***************
*** 2156,2161 ****
--- 2390,2396 ----
  
      if-ge v0, v3, :cond_3
  
+     .line 977
      move-object/from16 v0, p0
  
      move/from16 v1, v22
***************
*** 2164,2169 ****
--- 2399,2405 ----
  
      move-result-object v7
  
+     .line 978
      .local v7, c:Landroid/view/View;
      move-object/from16 v0, p0
  
***************
*** 2173,2183 ****
--- 2409,2421 ----
  
      if-eqz v48, :cond_0
  
+     .line 976
      :goto_1
      add-int/lit8 v22, v22, 0x1
  
      goto :goto_0
  
+     .line 979
      :cond_0
      move-object/from16 v0, p0
  
***************
*** 2185,2195 ****
--- 2423,2435 ----
  
      move-result-object v24
  
+     .line 980
      .local v24, lp:Landroid/widget/GridLayout$LayoutParams;
      move-object/from16 v0, v24
  
      iget-object v14, v0, Landroid/widget/GridLayout$LayoutParams;->columnSpec:Landroid/widget/GridLayout$Spec;
  
+     .line 981
      .local v14, columnSpec:Landroid/widget/GridLayout$Spec;
      move-object/from16 v0, v24
  
***************
*** 2197,2205 ****
--- 2437,2447 ----
  
      move-object/from16 v36, v0
  
+     .line 983
      .local v36, rowSpec:Landroid/widget/GridLayout$Spec;
      iget-object v13, v14, Landroid/widget/GridLayout$Spec;->span:Landroid/widget/GridLayout$Interval;
  
+     .line 984
      .local v13, colSpan:Landroid/widget/GridLayout$Interval;
      move-object/from16 v0, v36
  
***************
*** 2207,2212 ****
--- 2449,2455 ----
  
      move-object/from16 v35, v0
  
+     .line 986
      .local v35, rowSpan:Landroid/widget/GridLayout$Interval;
      iget v0, v13, Landroid/widget/GridLayout$Interval;->min:I
  
***************
*** 2214,2219 ****
--- 2457,2463 ----
  
      aget v44, v20, v48
  
+     .line 987
      .local v44, x1:I
      move-object/from16 v0, v35
  
***************
*** 2223,2228 ****
--- 2467,2473 ----
  
      aget v46, v42, v48
  
+     .line 989
      .local v46, y1:I
      iget v0, v13, Landroid/widget/GridLayout$Interval;->max:I
  
***************
*** 2230,2235 ****
--- 2475,2481 ----
  
      aget v45, v20, v48
  
+     .line 990
      .local v45, x2:I
      move-object/from16 v0, v35
  
***************
*** 2239,2250 ****
--- 2485,2499 ----
  
      aget v47, v42, v48
  
+     .line 992
      .local v47, y2:I
      sub-int v11, v45, v44
  
+     .line 993
      .local v11, cellWidth:I
      sub-int v10, v47, v46
  
+     .line 995
      .local v10, cellHeight:I
      const/16 v48, 0x1
  
***************
*** 2256,2261 ****
--- 2505,2511 ----
  
      move-result v28
  
+     .line 996
      .local v28, pWidth:I
      const/16 v48, 0x0
  
***************
*** 2267,2272 ****
--- 2517,2523 ----
  
      move-result v27
  
+     .line 998
      .local v27, pHeight:I
      iget-object v0, v14, Landroid/widget/GridLayout$Spec;->alignment:Landroid/widget/GridLayout$Alignment;
  
***************
*** 2284,2289 ****
--- 2535,2541 ----
  
      move-result-object v19
  
+     .line 999
      .local v19, hAlign:Landroid/widget/GridLayout$Alignment;
      move-object/from16 v0, v36
  
***************
*** 2303,2308 ****
--- 2555,2561 ----
  
      move-result-object v41
  
+     .line 1003
      .local v41, vAlign:Landroid/widget/GridLayout$Alignment;
      move-object/from16 v0, p0
  
***************
*** 2324,2329 ****
--- 2577,2583 ----
  
      check-cast v12, Landroid/widget/GridLayout$Bounds;
  
+     .line 1004
      .local v12, colBounds:Landroid/widget/GridLayout$Bounds;
      move-object/from16 v0, p0
  
***************
*** 2345,2350 ****
--- 2599,2605 ----
  
      check-cast v34, Landroid/widget/GridLayout$Bounds;
  
+     .line 1008
      .local v34, rowBounds:Landroid/widget/GridLayout$Bounds;
      const/16 v48, 0x0
  
***************
*** 2376,2381 ****
--- 2631,2637 ----
  
      move-result v8
  
+     .line 1010
      .local v8, c2ax:I
      const/16 v48, 0x0
  
***************
*** 2409,2414 ****
--- 2665,2671 ----
  
      move-result v9
  
+     .line 1012
      .local v9, c2ay:I
      const/16 v48, 0x1
  
***************
*** 2424,2429 ****
--- 2681,2687 ----
  
      move-result v23
  
+     .line 1013
      .local v23, leftMargin:I
      const/16 v48, 0x0
  
***************
*** 2439,2444 ****
--- 2697,2703 ----
  
      move-result v39
  
+     .line 1014
      .local v39, topMargin:I
      const/16 v48, 0x1
  
***************
*** 2454,2459 ****
--- 2713,2719 ----
  
      move-result v33
  
+     .line 1015
      .local v33, rightMargin:I
      const/16 v48, 0x0
  
***************
*** 2469,2484 ****
--- 2729,2747 ----
  
      move-result v6
  
+     .line 1018
      .local v6, bottomMargin:I
      add-int v48, v23, v28
  
      add-int v26, v48, v33
  
+     .line 1019
      .local v26, mWidth:I
      add-int v48, v39, v27
  
      add-int v25, v48, v6
  
+     .line 1022
      .local v25, mHeight:I
      move-object/from16 v0, v19
  
***************
*** 2488,2493 ****
--- 2751,2757 ----
  
      move-result v4
  
+     .line 1023
      .local v4, a2vx:I
      move-object/from16 v0, v34
  
***************
*** 2499,2525 ****
--- 2763,2795 ----
  
      move-result v5
  
+     .line 1025
      .local v5, a2vy:I
      add-int v48, v8, v4
  
      add-int v17, v48, v23
  
+     .line 1026
      .local v17, dx:I
      add-int v48, v9, v5
  
      add-int v18, v48, v39
  
+     .line 1028
      .local v18, dy:I
      add-int v48, v23, v33
  
      sub-int v11, v11, v48
  
+     .line 1029
      add-int v48, v39, v6
  
      sub-int v10, v10, v48
  
+     .line 1031
      const/16 v40, 0x1
  
+     .line 1032
      .local v40, type:I
      move-object/from16 v0, v19
  
***************
*** 2531,2536 ****
--- 2801,2807 ----
  
      move-result v43
  
+     .line 1033
      .local v43, width:I
      move-object/from16 v0, v41
  
***************
*** 2542,2557 ****
--- 2813,2831 ----
  
      move-result v21
  
+     .line 1035
      .local v21, height:I
      add-int v48, v30, v44
  
      add-int v15, v48, v17
  
+     .line 1036
      .local v15, cx:I
      add-int v48, v32, v46
  
      add-int v16, v48, v18
  
+     .line 1037
      .local v16, cy:I
      invoke-virtual {v7}, Landroid/view/View;->getMeasuredWidth()I
  
***************
*** 2573,2578 ****
--- 2847,2853 ----
  
      if-eq v0, v1, :cond_2
  
+     .line 1038
      :cond_1
      const/high16 v48, 0x4000
  
***************
*** 2600,2605 ****
--- 2875,2881 ----
  
      invoke-virtual {v7, v0, v1}, Landroid/view/View;->measure(II)V
  
+     .line 1040
      :cond_2
      add-int v48, v15, v43
  
***************
*** 2615,2620 ****
--- 2891,2897 ----
  
      goto/16 :goto_1
  
+     .line 1042
      .end local v4           #a2vx:I
      .end local v5           #a2vy:I
      .end local v6           #bottomMargin:I
***************
*** 2662,2692 ****
--- 2939,2976 ----
      .prologue
      const/4 v8, 0x0
  
+     .line 892
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateValues()V
  
+     .line 894
      const/4 v6, 0x1
  
      invoke-direct {p0, p1, p2, v6}, Landroid/widget/GridLayout;->measureChildrenWithMargins(IIZ)V
  
+     .line 899
      iget v6, p0, Landroid/widget/GridLayout;->orientation:I
  
      if-nez v6, :cond_0
  
+     .line 900
      iget-object v6, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v6, p1}, Landroid/widget/GridLayout$Axis;->getMeasure(I)I
  
      move-result v5
  
+     .line 901
      .local v5, width:I
      invoke-direct {p0, p1, p2, v8}, Landroid/widget/GridLayout;->measureChildrenWithMargins(IIZ)V
  
+     .line 902
      iget-object v6, p0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v6, p2}, Landroid/widget/GridLayout$Axis;->getMeasure(I)I
  
      move-result v1
  
+     .line 909
      .local v1, height:I
      :goto_0
      invoke-virtual {p0}, Landroid/widget/GridLayout;->getPaddingLeft()I
***************
*** 2699,2704 ****
--- 2983,2989 ----
  
      add-int v0, v6, v7
  
+     .line 910
      .local v0, hPadding:I
      invoke-virtual {p0}, Landroid/widget/GridLayout;->getPaddingTop()I
  
***************
*** 2710,2715 ****
--- 2995,3001 ----
  
      add-int v4, v6, v7
  
+     .line 912
      .local v4, vPadding:I
      add-int v6, v0, v5
  
***************
*** 2721,2726 ****
--- 3007,3013 ----
  
      move-result v3
  
+     .line 913
      .local v3, measuredWidth:I
      add-int v6, v4, v1
  
***************
*** 2732,2737 ****
--- 3019,3025 ----
  
      move-result v2
  
+     .line 915
      .local v2, measuredHeight:I
      invoke-static {v3, p1, v8}, Landroid/widget/GridLayout;->resolveSizeAndState(III)I
  
***************
*** 2743,2750 ****
--- 3031,3040 ----
  
      invoke-virtual {p0, v6, v7}, Landroid/widget/GridLayout;->setMeasuredDimension(II)V
  
+     .line 918
      return-void
  
+     .line 904
      .end local v0           #hPadding:I
      .end local v1           #height:I
      .end local v2           #measuredHeight:I
***************
*** 2758,2766 ****
--- 3048,3058 ----
  
      move-result v1
  
+     .line 905
      .restart local v1       #height:I
      invoke-direct {p0, p1, p2, v8}, Landroid/widget/GridLayout;->measureChildrenWithMargins(IIZ)V
  
+     .line 906
      iget-object v6, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v6, p1}, Landroid/widget/GridLayout$Axis;->getMeasure(I)I
***************
*** 2776,2785 ****
--- 3068,3080 ----
      .parameter "child"
  
      .prologue
+     .line 822
      invoke-super {p0, p1}, Landroid/view/ViewGroup;->onViewAdded(Landroid/view/View;)V
  
+     .line 823
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateStructure()V
  
+     .line 824
      return-void
  .end method
  
***************
*** 2788,2797 ****
--- 3083,3095 ----
      .parameter "child"
  
      .prologue
+     .line 831
      invoke-super {p0, p1}, Landroid/view/ViewGroup;->onViewRemoved(Landroid/view/View;)V
  
+     .line 832
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateStructure()V
  
+     .line 833
      return-void
  .end method
  
***************
*** 2799,2808 ****
--- 3097,3109 ----
      .locals 0
  
      .prologue
+     .line 937
      invoke-super {p0}, Landroid/view/ViewGroup;->requestLayout()V
  
+     .line 938
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateValues()V
  
+     .line 939
      return-void
  .end method
  
***************
*** 2811,2820 ****
--- 3112,3124 ----
      .parameter "alignmentMode"
  
      .prologue
+     .line 459
      iput p1, p0, Landroid/widget/GridLayout;->alignmentMode:I
  
+     .line 460
      invoke-virtual {p0}, Landroid/widget/GridLayout;->requestLayout()V
  
+     .line 461
      return-void
  .end method
  
***************
*** 2823,2836 ****
--- 3127,3144 ----
      .parameter "columnCount"
  
      .prologue
+     .line 377
      iget-object v0, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0, p1}, Landroid/widget/GridLayout$Axis;->setCount(I)V
  
+     .line 378
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateStructure()V
  
+     .line 379
      invoke-virtual {p0}, Landroid/widget/GridLayout;->requestLayout()V
  
+     .line 380
      return-void
  .end method
  
***************
*** 2839,2852 ****
--- 3147,3164 ----
      .parameter "columnOrderPreserved"
  
      .prologue
+     .line 530
      iget-object v0, p0, Landroid/widget/GridLayout;->horizontalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0, p1}, Landroid/widget/GridLayout$Axis;->setOrderPreserved(Z)V
  
+     .line 531
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateStructure()V
  
+     .line 532
      invoke-virtual {p0}, Landroid/widget/GridLayout;->requestLayout()V
  
+     .line 533
      return-void
  .end method
  
***************
*** 2855,2870 ****
--- 3167,3187 ----
      .parameter "orientation"
  
      .prologue
+     .line 309
      iget v0, p0, Landroid/widget/GridLayout;->orientation:I
  
      if-eq v0, p1, :cond_0
  
+     .line 310
      iput p1, p0, Landroid/widget/GridLayout;->orientation:I
  
+     .line 311
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateStructure()V
  
+     .line 312
      invoke-virtual {p0}, Landroid/widget/GridLayout;->requestLayout()V
  
+     .line 314
      :cond_0
      return-void
  .end method
***************
*** 2874,2887 ****
--- 3191,3208 ----
      .parameter "rowCount"
  
      .prologue
+     .line 344
      iget-object v0, p0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0, p1}, Landroid/widget/GridLayout$Axis;->setCount(I)V
  
+     .line 345
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateStructure()V
  
+     .line 346
      invoke-virtual {p0}, Landroid/widget/GridLayout;->requestLayout()V
  
+     .line 347
      return-void
  .end method
  
***************
*** 2890,2903 ****
--- 3211,3228 ----
      .parameter "rowOrderPreserved"
  
      .prologue
+     .line 494
      iget-object v0, p0, Landroid/widget/GridLayout;->verticalAxis:Landroid/widget/GridLayout$Axis;
  
      invoke-virtual {v0, p1}, Landroid/widget/GridLayout$Axis;->setOrderPreserved(Z)V
  
+     .line 495
      invoke-direct {p0}, Landroid/widget/GridLayout;->invalidateStructure()V
  
+     .line 496
      invoke-virtual {p0}, Landroid/widget/GridLayout;->requestLayout()V
  
+     .line 497
      return-void
  .end method
  
***************
*** 2906,2914 ****
--- 3231,3242 ----
      .parameter "useDefaultMargins"
  
      .prologue
+     .line 423
      iput-boolean p1, p0, Landroid/widget/GridLayout;->useDefaultMargins:Z
  
+     .line 424
      invoke-virtual {p0}, Landroid/widget/GridLayout;->requestLayout()V
  
+     .line 425
      return-void
  .end method
