*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/GridLayout$Bounds.smali	2012-03-23 19:35:53.813568175 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/GridLayout$Bounds.smali.original	2012-03-23 19:34:44.891311379 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .locals 0
  
      .prologue
+     .line 2048
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 2049
      invoke-virtual {p0}, Landroid/widget/GridLayout$Bounds;->reset()V
  
+     .line 2050
      return-void
  .end method
  
***************
*** 39,44 ****
--- 42,48 ----
      .parameter "x0"
  
      .prologue
+     .line 2043
      invoke-direct {p0}, Landroid/widget/GridLayout$Bounds;-><init>()V
  
      return-void
***************
*** 53,58 ****
--- 57,63 ----
      .parameter "size"
  
      .prologue
+     .line 2073
      iget v0, p0, Landroid/widget/GridLayout$Bounds;->before:I
  
      invoke-virtual {p2, p1, p3}, Landroid/widget/GridLayout$Alignment;->getAlignmentValue(Landroid/view/View;I)I
***************
*** 70,75 ****
--- 75,81 ----
      .parameter "after"
  
      .prologue
+     .line 2059
      iget v0, p0, Landroid/widget/GridLayout$Bounds;->before:I
  
      invoke-static {v0, p1}, Ljava/lang/Math;->max(II)I
***************
*** 78,83 ****
--- 84,90 ----
  
      iput v0, p0, Landroid/widget/GridLayout$Bounds;->before:I
  
+     .line 2060
      iget v0, p0, Landroid/widget/GridLayout$Bounds;->after:I
  
      invoke-static {v0, p2}, Ljava/lang/Math;->max(II)I
***************
*** 86,91 ****
--- 93,99 ----
  
      iput v0, p0, Landroid/widget/GridLayout$Bounds;->after:I
  
+     .line 2061
      return-void
  .end method
  
***************
*** 97,102 ****
--- 105,111 ----
      .parameter "axis"
  
      .prologue
+     .line 2077
      iget v3, p0, Landroid/widget/GridLayout$Bounds;->flexibility:I
  
      invoke-virtual {p2}, Landroid/widget/GridLayout$Spec;->getFlexibility()I
***************
*** 107,118 ****
--- 116,129 ----
  
      iput v3, p0, Landroid/widget/GridLayout$Bounds;->flexibility:I
  
+     .line 2078
      iget-boolean v3, p4, Landroid/widget/GridLayout$Axis;->horizontal:Z
  
      invoke-virtual {p3, p1, v3}, Landroid/widget/GridLayout;->getMeasurementIncludingMargin(Landroid/view/View;Z)I
  
      move-result v2
  
+     .line 2079
      .local v2, size:I
      iget-object v3, p2, Landroid/widget/GridLayout$Spec;->alignment:Landroid/widget/GridLayout$Alignment;
  
***************
*** 122,137 ****
--- 133,151 ----
  
      move-result-object v0
  
+     .line 2081
      .local v0, alignment:Landroid/widget/GridLayout$Alignment;
      invoke-virtual {v0, p1, v2}, Landroid/widget/GridLayout$Alignment;->getAlignmentValue(Landroid/view/View;I)I
  
      move-result v1
  
+     .line 2082
      .local v1, before:I
      sub-int v3, v2, v1
  
      invoke-virtual {p0, v1, v3}, Landroid/widget/GridLayout$Bounds;->include(II)V
  
+     .line 2083
      return-void
  .end method
  
***************
*** 141,154 ****
--- 155,172 ----
      .prologue
      const/high16 v0, -0x8000
  
+     .line 2053
      iput v0, p0, Landroid/widget/GridLayout$Bounds;->before:I
  
+     .line 2054
      iput v0, p0, Landroid/widget/GridLayout$Bounds;->after:I
  
+     .line 2055
      const/4 v0, 0x2
  
      iput v0, p0, Landroid/widget/GridLayout$Bounds;->flexibility:I
  
+     .line 2056
      return-void
  .end method
  
***************
*** 157,164 ****
--- 175,184 ----
      .parameter "min"
  
      .prologue
+     .line 2064
      if-nez p1, :cond_0
  
+     .line 2065
      iget v0, p0, Landroid/widget/GridLayout$Bounds;->flexibility:I
  
      invoke-static {v0}, Landroid/widget/GridLayout;->canStretch(I)Z
***************
*** 167,174 ****
--- 187,196 ----
  
      if-eqz v0, :cond_0
  
+     .line 2066
      const v0, 0x186a0
  
+     .line 2069
      :goto_0
      return v0
  
***************
*** 186,191 ****
--- 208,214 ----
      .locals 2
  
      .prologue
+     .line 2087
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
