*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/HeaderViewListAdapter.smali	2012-03-23 19:35:50.864514399 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/HeaderViewListAdapter.smali.original	2012-03-23 19:34:44.867310942 -0500
***************
*** 55,60 ****
--- 55,61 ----
      .locals 1
  
      .prologue
+     .line 43
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
***************
*** 86,114 ****
--- 87,123 ----
      .end annotation
  
      .prologue
+     .line 52
      .local p1, headerViewInfos:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/widget/ListView$FixedViewInfo;>;"
      .local p2, footerViewInfos:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/widget/ListView$FixedViewInfo;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 53
      iput-object p3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
+     .line 54
      instance-of v0, p3, Landroid/widget/Filterable;
  
      iput-boolean v0, p0, Landroid/widget/HeaderViewListAdapter;->mIsFilterable:Z
  
+     .line 56
      if-nez p1, :cond_0
  
+     .line 57
      sget-object v0, Landroid/widget/HeaderViewListAdapter;->EMPTY_INFO_LIST:Ljava/util/ArrayList;
  
      iput-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
+     .line 62
      :goto_0
      if-nez p2, :cond_1
  
+     .line 63
      sget-object v0, Landroid/widget/HeaderViewListAdapter;->EMPTY_INFO_LIST:Ljava/util/ArrayList;
  
      iput-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mFooterViewInfos:Ljava/util/ArrayList;
  
+     .line 68
      :goto_1
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
***************
*** 131,148 ****
--- 140,161 ----
      :goto_2
      iput-boolean v0, p0, Landroid/widget/HeaderViewListAdapter;->mAreAllFixedViewsSelectable:Z
  
+     .line 71
      return-void
  
+     .line 59
      :cond_0
      iput-object p1, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
      goto :goto_0
  
+     .line 65
      :cond_1
      iput-object p2, p0, Landroid/widget/HeaderViewListAdapter;->mFooterViewInfos:Ljava/util/ArrayList;
  
      goto :goto_1
  
+     .line 68
      :cond_2
      const/4 v0, 0x0
  
***************
*** 163,171 ****
--- 176,186 ----
      .end annotation
  
      .prologue
+     .line 86
      .local p1, infos:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/widget/ListView$FixedViewInfo;>;"
      if-eqz p1, :cond_1
  
+     .line 87
      invoke-virtual {p1}, Ljava/util/ArrayList;->iterator()Ljava/util/Iterator;
  
      move-result-object v0
***************
*** 184,196 ****
--- 199,214 ----
  
      check-cast v1, Landroid/widget/ListView$FixedViewInfo;
  
+     .line 88
      .local v1, info:Landroid/widget/ListView$FixedViewInfo;
      iget-boolean v2, v1, Landroid/widget/ListView$FixedViewInfo;->isSelectable:Z
  
      if-nez v2, :cond_0
  
+     .line 89
      const/4 v2, 0x0
  
+     .line 93
      .end local v0           #i$:Ljava/util/Iterator;
      .end local v1           #info:Landroid/widget/ListView$FixedViewInfo;
      :goto_0
***************
*** 210,219 ****
--- 228,239 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 139
      iget-object v1, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v1, :cond_0
  
+     .line 140
      iget-boolean v1, p0, Landroid/widget/HeaderViewListAdapter;->mAreAllFixedViewsSelectable:Z
  
      if-eqz v1, :cond_1
***************
*** 226,235 ****
--- 246,257 ----
  
      if-eqz v1, :cond_1
  
+     .line 142
      :cond_0
      :goto_0
      return v0
  
+     .line 140
      :cond_1
      const/4 v0, 0x0
  
***************
*** 240,249 ****
--- 262,273 ----
      .locals 2
  
      .prologue
+     .line 131
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v0, :cond_0
  
+     .line 132
      invoke-virtual {p0}, Landroid/widget/HeaderViewListAdapter;->getFootersCount()I
  
      move-result v0
***************
*** 262,267 ****
--- 286,292 ----
  
      add-int/2addr v0, v1
  
+     .line 134
      :goto_0
      return v0
  
***************
*** 283,292 ****
--- 308,319 ----
      .locals 1
  
      .prologue
+     .line 261
      iget-boolean v0, p0, Landroid/widget/HeaderViewListAdapter;->mIsFilterable:Z
  
      if-eqz v0, :cond_0
  
+     .line 262
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      check-cast v0, Landroid/widget/Filterable;
***************
*** 295,300 ****
--- 322,328 ----
  
      move-result-object v0
  
+     .line 264
      :goto_0
      return-object v0
  
***************
*** 308,313 ****
--- 336,342 ----
      .locals 1
  
      .prologue
+     .line 78
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mFooterViewInfos:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->size()I
***************
*** 321,326 ****
--- 350,356 ----
      .locals 1
  
      .prologue
+     .line 74
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->size()I
***************
*** 335,347 ****
--- 365,380 ----
      .parameter "position"
  
      .prologue
+     .line 169
      invoke-virtual {p0}, Landroid/widget/HeaderViewListAdapter;->getHeadersCount()I
  
      move-result v2
  
+     .line 170
      .local v2, numHeaders:I
      if-ge p1, v2, :cond_0
  
+     .line 171
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
      invoke-virtual {v3, p1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 352,379 ****
--- 385,419 ----
  
      iget-object v3, v3, Landroid/widget/ListView$FixedViewInfo;->data:Ljava/lang/Object;
  
+     .line 185
      :goto_0
      return-object v3
  
+     .line 175
      :cond_0
      sub-int v1, p1, v2
  
+     .line 176
      .local v1, adjPosition:I
      const/4 v0, 0x0
  
+     .line 177
      .local v0, adapterCount:I
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v3, :cond_1
  
+     .line 178
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v3}, Landroid/widget/ListAdapter;->getCount()I
  
      move-result v0
  
+     .line 179
      if-ge v1, v0, :cond_1
  
+     .line 180
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v3, v1}, Landroid/widget/ListAdapter;->getItem(I)Ljava/lang/Object;
***************
*** 382,387 ****
--- 422,428 ----
  
      goto :goto_0
  
+     .line 185
      :cond_1
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mFooterViewInfos:Ljava/util/ArrayList;
  
***************
*** 403,412 ****
--- 444,455 ----
      .parameter "position"
  
      .prologue
+     .line 189
      invoke-virtual {p0}, Landroid/widget/HeaderViewListAdapter;->getHeadersCount()I
  
      move-result v2
  
+     .line 190
      .local v2, numHeaders:I
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
***************
*** 414,421 ****
--- 457,466 ----
  
      if-lt p1, v2, :cond_0
  
+     .line 191
      sub-int v1, p1, v2
  
+     .line 192
      .local v1, adjPosition:I
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
***************
*** 423,437 ****
--- 468,485 ----
  
      move-result v0
  
+     .line 193
      .local v0, adapterCount:I
      if-ge v1, v0, :cond_0
  
+     .line 194
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v3, v1}, Landroid/widget/ListAdapter;->getItemId(I)J
  
      move-result-wide v3
  
+     .line 197
      .end local v0           #adapterCount:I
      .end local v1           #adjPosition:I
      :goto_0
***************
*** 448,457 ****
--- 496,507 ----
      .parameter "position"
  
      .prologue
+     .line 229
      invoke-virtual {p0}, Landroid/widget/HeaderViewListAdapter;->getHeadersCount()I
  
      move-result v2
  
+     .line 230
      .local v2, numHeaders:I
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
***************
*** 459,466 ****
--- 509,518 ----
  
      if-lt p1, v2, :cond_0
  
+     .line 231
      sub-int v1, p1, v2
  
+     .line 232
      .local v1, adjPosition:I
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
***************
*** 468,482 ****
--- 520,537 ----
  
      move-result v0
  
+     .line 233
      .local v0, adapterCount:I
      if-ge v1, v0, :cond_0
  
+     .line 234
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v3, v1}, Landroid/widget/ListAdapter;->getItemViewType(I)I
  
      move-result v3
  
+     .line 238
      .end local v0           #adapterCount:I
      .end local v1           #adjPosition:I
      :goto_0
***************
*** 495,507 ****
--- 550,565 ----
      .parameter "parent"
  
      .prologue
+     .line 209
      invoke-virtual {p0}, Landroid/widget/HeaderViewListAdapter;->getHeadersCount()I
  
      move-result v2
  
+     .line 210
      .local v2, numHeaders:I
      if-ge p1, v2, :cond_0
  
+     .line 211
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
      invoke-virtual {v3, p1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 512,539 ****
--- 570,604 ----
  
      iget-object v3, v3, Landroid/widget/ListView$FixedViewInfo;->view:Landroid/view/View;
  
+     .line 225
      :goto_0
      return-object v3
  
+     .line 215
      :cond_0
      sub-int v1, p1, v2
  
+     .line 216
      .local v1, adjPosition:I
      const/4 v0, 0x0
  
+     .line 217
      .local v0, adapterCount:I
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v3, :cond_1
  
+     .line 218
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v3}, Landroid/widget/ListAdapter;->getCount()I
  
      move-result v0
  
+     .line 219
      if-ge v1, v0, :cond_1
  
+     .line 220
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v3, v1, p2, p3}, Landroid/widget/ListAdapter;->getView(ILandroid/view/View;Landroid/view/ViewGroup;)Landroid/view/View;
***************
*** 542,547 ****
--- 607,613 ----
  
      goto :goto_0
  
+     .line 225
      :cond_1
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mFooterViewInfos:Ljava/util/ArrayList;
  
***************
*** 562,577 ****
--- 628,646 ----
      .locals 1
  
      .prologue
+     .line 242
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v0, :cond_0
  
+     .line 243
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v0}, Landroid/widget/ListAdapter;->getViewTypeCount()I
  
      move-result v0
  
+     .line 245
      :goto_0
      return v0
  
***************
*** 585,590 ****
--- 654,660 ----
      .locals 1
  
      .prologue
+     .line 268
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      return-object v0
***************
*** 594,609 ****
--- 664,682 ----
      .locals 1
  
      .prologue
+     .line 201
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v0, :cond_0
  
+     .line 202
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v0}, Landroid/widget/ListAdapter;->hasStableIds()Z
  
      move-result v0
  
+     .line 204
      :goto_0
      return v0
  
***************
*** 617,622 ****
--- 690,696 ----
      .locals 1
  
      .prologue
+     .line 82
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v0, :cond_0
***************
*** 646,658 ****
--- 720,735 ----
      .parameter "position"
  
      .prologue
+     .line 148
      invoke-virtual {p0}, Landroid/widget/HeaderViewListAdapter;->getHeadersCount()I
  
      move-result v2
  
+     .line 149
      .local v2, numHeaders:I
      if-ge p1, v2, :cond_0
  
+     .line 150
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
      invoke-virtual {v3, p1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 663,690 ****
--- 740,774 ----
  
      iget-boolean v3, v3, Landroid/widget/ListView$FixedViewInfo;->isSelectable:Z
  
+     .line 164
      :goto_0
      return v3
  
+     .line 154
      :cond_0
      sub-int v1, p1, v2
  
+     .line 155
      .local v1, adjPosition:I
      const/4 v0, 0x0
  
+     .line 156
      .local v0, adapterCount:I
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v3, :cond_1
  
+     .line 157
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v3}, Landroid/widget/ListAdapter;->getCount()I
  
      move-result v0
  
+     .line 158
      if-ge v1, v0, :cond_1
  
+     .line 159
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v3, v1}, Landroid/widget/ListAdapter;->isEnabled(I)Z
***************
*** 693,698 ****
--- 777,783 ----
  
      goto :goto_0
  
+     .line 164
      :cond_1
      iget-object v3, p0, Landroid/widget/HeaderViewListAdapter;->mFooterViewInfos:Ljava/util/ArrayList;
  
***************
*** 714,727 ****
--- 799,815 ----
      .parameter "observer"
  
      .prologue
+     .line 249
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v0, :cond_0
  
+     .line 250
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v0, p1}, Landroid/widget/ListAdapter;->registerDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 252
      :cond_0
      return-void
  .end method
***************
*** 735,740 ****
--- 823,829 ----
  
      const/4 v2, 0x0
  
+     .line 114
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 747,752 ****
--- 836,842 ----
  
      if-ge v0, v4, :cond_2
  
+     .line 115
      iget-object v4, p0, Landroid/widget/HeaderViewListAdapter;->mFooterViewInfos:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 755,769 ****
--- 845,862 ----
  
      check-cast v1, Landroid/widget/ListView$FixedViewInfo;
  
+     .line 116
      .local v1, info:Landroid/widget/ListView$FixedViewInfo;
      iget-object v4, v1, Landroid/widget/ListView$FixedViewInfo;->view:Landroid/view/View;
  
      if-ne v4, p1, :cond_1
  
+     .line 117
      iget-object v4, p0, Landroid/widget/HeaderViewListAdapter;->mFooterViewInfos:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v0}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 119
      iget-object v4, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
      invoke-direct {p0, v4}, Landroid/widget/HeaderViewListAdapter;->areAllListInfosSelectable(Ljava/util/ArrayList;)Z
***************
*** 785,794 ****
--- 878,889 ----
      :cond_0
      iput-boolean v2, p0, Landroid/widget/HeaderViewListAdapter;->mAreAllFixedViewsSelectable:Z
  
+     .line 127
      .end local v1           #info:Landroid/widget/ListView$FixedViewInfo;
      :goto_1
      return v3
  
+     .line 114
      .restart local v1       #info:Landroid/widget/ListView$FixedViewInfo;
      :cond_1
      add-int/lit8 v0, v0, 0x1
***************
*** 799,804 ****
--- 894,900 ----
      :cond_2
      move v3, v2
  
+     .line 127
      goto :goto_1
  .end method
  
***************
*** 811,816 ****
--- 907,913 ----
  
      const/4 v2, 0x0
  
+     .line 97
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 823,828 ****
--- 920,926 ----
  
      if-ge v0, v4, :cond_2
  
+     .line 98
      iget-object v4, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 831,845 ****
--- 929,946 ----
  
      check-cast v1, Landroid/widget/ListView$FixedViewInfo;
  
+     .line 99
      .local v1, info:Landroid/widget/ListView$FixedViewInfo;
      iget-object v4, v1, Landroid/widget/ListView$FixedViewInfo;->view:Landroid/view/View;
  
      if-ne v4, p1, :cond_1
  
+     .line 100
      iget-object v4, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
      invoke-virtual {v4, v0}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 102
      iget-object v4, p0, Landroid/widget/HeaderViewListAdapter;->mHeaderViewInfos:Ljava/util/ArrayList;
  
      invoke-direct {p0, v4}, Landroid/widget/HeaderViewListAdapter;->areAllListInfosSelectable(Ljava/util/ArrayList;)Z
***************
*** 861,870 ****
--- 962,973 ----
      :cond_0
      iput-boolean v2, p0, Landroid/widget/HeaderViewListAdapter;->mAreAllFixedViewsSelectable:Z
  
+     .line 110
      .end local v1           #info:Landroid/widget/ListView$FixedViewInfo;
      :goto_1
      return v3
  
+     .line 97
      .restart local v1       #info:Landroid/widget/ListView$FixedViewInfo;
      :cond_1
      add-int/lit8 v0, v0, 0x1
***************
*** 875,880 ****
--- 978,984 ----
      :cond_2
      move v3, v2
  
+     .line 110
      goto :goto_1
  .end method
  
***************
*** 883,896 ****
--- 987,1003 ----
      .parameter "observer"
  
      .prologue
+     .line 255
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      if-eqz v0, :cond_0
  
+     .line 256
      iget-object v0, p0, Landroid/widget/HeaderViewListAdapter;->mAdapter:Landroid/widget/ListAdapter;
  
      invoke-interface {v0, p1}, Landroid/widget/ListAdapter;->unregisterDataSetObserver(Landroid/database/DataSetObserver;)V
  
+     .line 258
      :cond_0
      return-void
  .end method
