*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/HorizontalScrollView.smali	2012-03-23 19:35:53.225557452 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/HorizontalScrollView.smali.original	2012-03-23 19:34:44.886311288 -0500
***************
*** 59,68 ****
--- 59,70 ----
      .parameter "context"
  
      .prologue
+     .line 139
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/widget/HorizontalScrollView;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 140
      return-void
  .end method
  
***************
*** 72,81 ****
--- 74,85 ----
      .parameter "attrs"
  
      .prologue
+     .line 143
      const v0, 0x1010353
  
      invoke-direct {p0, p1, p2, v0}, Landroid/widget/HorizontalScrollView;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 144
      return-void
  .end method
  
***************
*** 90,125 ****
--- 94,139 ----
  
      const/4 v2, 0x0
  
+     .line 147
      invoke-direct {p0, p1, p2, p3}, Landroid/widget/FrameLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 72
      new-instance v1, Landroid/graphics/Rect;
  
      invoke-direct {v1}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v1, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
+     .line 86
      iput-boolean v3, p0, Landroid/widget/HorizontalScrollView;->mIsLayoutDirty:Z
  
+     .line 93
      const/4 v1, 0x0
  
      iput-object v1, p0, Landroid/widget/HorizontalScrollView;->mChildToScrollTo:Landroid/view/View;
  
+     .line 100
      iput-boolean v2, p0, Landroid/widget/HorizontalScrollView;->mIsBeingDragged:Z
  
+     .line 117
      iput-boolean v3, p0, Landroid/widget/HorizontalScrollView;->mSmoothScrollingEnabled:Z
  
+     .line 130
      const/4 v1, -0x1
  
      iput v1, p0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
  
+     .line 148
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->initScrollView()V
  
+     .line 150
      sget-object v1, Landroid/R$styleable;->HorizontalScrollView:[I
  
      invoke-virtual {p1, p2, v1, p3, v2}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[III)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 153
      .local v0, a:Landroid/content/res/TypedArray;
      invoke-virtual {v0, v2, v2}, Landroid/content/res/TypedArray;->getBoolean(IZ)Z
  
***************
*** 127,134 ****
--- 141,150 ----
  
      invoke-virtual {p0, v1}, Landroid/widget/HorizontalScrollView;->setFillViewport(Z)V
  
+     .line 155
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 156
      return-void
  .end method
  
***************
*** 138,154 ****
--- 154,174 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 250
      invoke-virtual {p0, v2}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 251
      .local v0, child:Landroid/view/View;
      if-eqz v0, :cond_0
  
+     .line 252
      invoke-virtual {v0}, Landroid/view/View;->getWidth()I
  
      move-result v1
  
+     .line 253
      .local v1, childWidth:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
***************
*** 166,171 ****
--- 186,192 ----
  
      const/4 v2, 0x1
  
+     .line 255
      .end local v1           #childWidth:I
      :cond_0
      return v2
***************
*** 178,201 ****
--- 199,227 ----
      .parameter "child"
  
      .prologue
+     .line 1526
      if-ge p2, p3, :cond_0
  
      if-gez p1, :cond_2
  
+     .line 1527
      :cond_0
      const/4 p1, 0x0
  
+     .line 1532
      .end local p1
      :cond_1
      :goto_0
      return p1
  
+     .line 1529
      .restart local p1
      :cond_2
      add-int v0, p2, p1
  
      if-le v0, p3, :cond_1
  
+     .line 1530
      sub-int p1, p3, p2
  
      goto :goto_0
***************
*** 208,225 ****
--- 234,256 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 1055
      if-eqz p1, :cond_0
  
+     .line 1056
      iget-boolean v0, p0, Landroid/widget/HorizontalScrollView;->mSmoothScrollingEnabled:Z
  
      if-eqz v0, :cond_1
  
+     .line 1057
      invoke-virtual {p0, p1, v1}, Landroid/widget/HorizontalScrollView;->smoothScrollBy(II)V
  
+     .line 1062
      :cond_0
      :goto_0
      return-void
  
+     .line 1059
      :cond_1
      invoke-virtual {p0, p1, v1}, Landroid/widget/HorizontalScrollView;->scrollBy(II)V
  
***************
*** 233,255 ****
--- 264,291 ----
      .parameter "right"
  
      .prologue
+     .line 801
      const/4 v10, 0x2
  
      invoke-virtual {p0, v10}, Landroid/widget/HorizontalScrollView;->getFocusables(I)Ljava/util/ArrayList;
  
      move-result-object v2
  
+     .line 802
      .local v2, focusables:Ljava/util/List;,"Ljava/util/List<Landroid/view/View;>;"
      const/4 v1, 0x0
  
+     .line 811
      .local v1, focusCandidate:Landroid/view/View;
      const/4 v3, 0x0
  
+     .line 813
      .local v3, foundFullyContainedFocusable:Z
      invoke-interface {v2}, Ljava/util/List;->size()I
  
      move-result v0
  
+     .line 814
      .local v0, count:I
      const/4 v4, 0x0
  
***************
*** 257,297 ****
--- 293,342 ----
      :goto_0
      if-ge v4, v0, :cond_8
  
+     .line 815
      invoke-interface {v2, v4}, Ljava/util/List;->get(I)Ljava/lang/Object;
  
      move-result-object v5
  
      check-cast v5, Landroid/view/View;
  
+     .line 816
      .local v5, view:Landroid/view/View;
      invoke-virtual {v5}, Landroid/view/View;->getLeft()I
  
      move-result v8
  
+     .line 817
      .local v8, viewLeft:I
      invoke-virtual {v5}, Landroid/view/View;->getRight()I
  
      move-result v9
  
+     .line 819
      .local v9, viewRight:I
      if-ge p2, v9, :cond_0
  
      if-ge v8, p3, :cond_0
  
+     .line 825
      if-ge p2, v8, :cond_1
  
      if-ge v9, p3, :cond_1
  
      const/4 v7, 0x1
  
+     .line 828
      .local v7, viewIsFullyContained:Z
      :goto_1
      if-nez v1, :cond_2
  
+     .line 830
      move-object v1, v5
  
+     .line 831
      move v3, v7
  
+     .line 814
      .end local v7           #viewIsFullyContained:Z
      :cond_0
      :goto_2
***************
*** 299,309 ****
--- 344,356 ----
  
      goto :goto_0
  
+     .line 825
      :cond_1
      const/4 v7, 0x0
  
      goto :goto_1
  
+     .line 833
      .restart local v7       #viewIsFullyContained:Z
      :cond_2
      if-eqz p1, :cond_3
***************
*** 326,366 ****
--- 373,423 ----
      :cond_4
      const/4 v6, 0x1
  
+     .line 837
      .local v6, viewIsCloserToBoundary:Z
      :goto_3
      if-eqz v3, :cond_6
  
+     .line 838
      if-eqz v7, :cond_0
  
      if-eqz v6, :cond_0
  
+     .line 844
      move-object v1, v5
  
      goto :goto_2
  
+     .line 833
      .end local v6           #viewIsCloserToBoundary:Z
      :cond_5
      const/4 v6, 0x0
  
      goto :goto_3
  
+     .line 847
      .restart local v6       #viewIsCloserToBoundary:Z
      :cond_6
      if-eqz v7, :cond_7
  
+     .line 849
      move-object v1, v5
  
+     .line 850
      const/4 v3, 0x1
  
      goto :goto_2
  
+     .line 851
      :cond_7
      if-eqz v6, :cond_0
  
+     .line 856
      move-object v1, v5
  
      goto :goto_2
  
+     .line 863
      .end local v5           #view:Landroid/view/View;
      .end local v6           #viewIsCloserToBoundary:Z
      .end local v7           #viewIsFullyContained:Z
***************
*** 377,391 ****
--- 434,451 ----
      .parameter "preferredFocusable"
  
      .prologue
+     .line 770
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getHorizontalFadingEdgeLength()I
  
      move-result v3
  
      div-int/lit8 v0, v3, 0x2
  
+     .line 771
      .local v0, fadingEdgeLength:I
      add-int v1, p2, v0
  
+     .line 772
      .local v1, leftWithoutFadingEdge:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
***************
*** 395,400 ****
--- 455,461 ----
  
      sub-int v2, v3, v0
  
+     .line 774
      .local v2, rightWithoutFadingEdge:I
      if-eqz p3, :cond_0
  
***************
*** 410,415 ****
--- 471,477 ----
  
      if-le v3, v1, :cond_0
  
+     .line 780
      .end local p3
      :goto_0
      return-object p3
***************
*** 429,436 ****
--- 491,500 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 738
      const/4 v1, 0x0
  
+     .line 739
      .local v1, scrollRange:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
***************
*** 438,447 ****
--- 502,513 ----
  
      if-lez v2, :cond_0
  
+     .line 740
      invoke-virtual {p0, v5}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 741
      .local v0, child:Landroid/view/View;
      invoke-virtual {v0}, Landroid/view/View;->getWidth()I
  
***************
*** 465,470 ****
--- 531,537 ----
  
      move-result v1
  
+     .line 744
      .end local v0           #child:Landroid/view/View;
      :cond_0
      return v1
***************
*** 478,496 ****
--- 545,567 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 387
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v3
  
      if-lez v3, :cond_0
  
+     .line 388
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
+     .line 389
      .local v1, scrollX:I
      invoke-virtual {p0, v2}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 390
      .local v0, child:Landroid/view/View;
      invoke-virtual {v0}, Landroid/view/View;->getTop()I
  
***************
*** 522,527 ****
--- 593,599 ----
  
      const/4 v2, 0x1
  
+     .line 395
      .end local v0           #child:Landroid/view/View;
      .end local v1           #scrollX:I
      :cond_0
***************
*** 532,550 ****
--- 604,626 ----
      .locals 1
  
      .prologue
+     .line 399
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      if-nez v0, :cond_0
  
+     .line 400
      invoke-static {}, Landroid/view/VelocityTracker;->obtain()Landroid/view/VelocityTracker;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
+     .line 404
      :goto_0
      return-void
  
+     .line 402
      :cond_0
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
***************
*** 557,562 ****
--- 633,639 ----
      .locals 3
  
      .prologue
+     .line 198
      new-instance v1, Landroid/widget/OverScroller;
  
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getContext()Landroid/content/Context;
***************
*** 567,590 ****
--- 644,672 ----
  
      iput-object v1, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
+     .line 199
      const/4 v1, 0x1
  
      invoke-virtual {p0, v1}, Landroid/widget/HorizontalScrollView;->setFocusable(Z)V
  
+     .line 200
      const/high16 v1, 0x4
  
      invoke-virtual {p0, v1}, Landroid/widget/HorizontalScrollView;->setDescendantFocusability(I)V
  
+     .line 201
      const/4 v1, 0x0
  
      invoke-virtual {p0, v1}, Landroid/widget/HorizontalScrollView;->setWillNotDraw(Z)V
  
+     .line 202
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mContext:Landroid/content/Context;
  
      invoke-static {v1}, Landroid/view/ViewConfiguration;->get(Landroid/content/Context;)Landroid/view/ViewConfiguration;
  
      move-result-object v0
  
+     .line 203
      .local v0, configuration:Landroid/view/ViewConfiguration;
      invoke-virtual {v0}, Landroid/view/ViewConfiguration;->getScaledTouchSlop()I
  
***************
*** 592,621 ****
--- 674,708 ----
  
      iput v1, p0, Landroid/widget/HorizontalScrollView;->mTouchSlop:I
  
+     .line 204
      invoke-virtual {v0}, Landroid/view/ViewConfiguration;->getScaledMinimumFlingVelocity()I
  
      move-result v1
  
      iput v1, p0, Landroid/widget/HorizontalScrollView;->mMinimumVelocity:I
  
+     .line 205
      invoke-virtual {v0}, Landroid/view/ViewConfiguration;->getScaledMaximumFlingVelocity()I
  
      move-result v1
  
      iput v1, p0, Landroid/widget/HorizontalScrollView;->mMaximumVelocity:I
  
+     .line 206
      invoke-virtual {v0}, Landroid/view/ViewConfiguration;->getScaledOverscrollDistance()I
  
      move-result v1
  
      iput v1, p0, Landroid/widget/HorizontalScrollView;->mOverscrollDistance:I
  
+     .line 207
      invoke-virtual {v0}, Landroid/view/ViewConfiguration;->getScaledOverflingDistance()I
  
      move-result v1
  
      iput v1, p0, Landroid/widget/HorizontalScrollView;->mOverflingDistance:I
  
+     .line 208
      return-void
  .end method
  
***************
*** 623,638 ****
--- 710,728 ----
      .locals 1
  
      .prologue
+     .line 407
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      if-nez v0, :cond_0
  
+     .line 408
      invoke-static {}, Landroid/view/VelocityTracker;->obtain()Landroid/view/VelocityTracker;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
+     .line 410
      :cond_0
      return-void
  .end method
***************
*** 644,649 ****
--- 734,740 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 1034
      invoke-direct {p0, p1, v0}, Landroid/widget/HorizontalScrollView;->isWithinDeltaOfScreen(Landroid/view/View;I)Z
  
      move-result v1
***************
*** 664,680 ****
--- 755,775 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 1416
      if-ne p1, p2, :cond_1
  
+     .line 1421
      :cond_0
      :goto_0
      return v1
  
+     .line 1420
      :cond_1
      invoke-virtual {p1}, Landroid/view/View;->getParent()Landroid/view/ViewParent;
  
      move-result-object v0
  
+     .line 1421
      .local v0, theParent:Landroid/view/ViewParent;
      instance-of v2, v0, Landroid/view/ViewGroup;
  
***************
*** 701,714 ****
--- 796,812 ----
      .parameter "delta"
  
      .prologue
+     .line 1042
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p1, v0}, Landroid/view/View;->getDrawingRect(Landroid/graphics/Rect;)V
  
+     .line 1043
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0, p1, v0}, Landroid/widget/HorizontalScrollView;->offsetDescendantRectToMyCoords(Landroid/view/View;Landroid/graphics/Rect;)V
  
+     .line 1045
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      iget v0, v0, Landroid/graphics/Rect;->right:I
***************
*** 755,760 ****
--- 853,859 ----
      .parameter "ev"
  
      .prologue
+     .line 649
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getAction()I
  
      move-result v3
***************
*** 765,784 ****
--- 864,887 ----
  
      shr-int/lit8 v2, v3, 0x8
  
+     .line 651
      .local v2, pointerIndex:I
      invoke-virtual {p1, v2}, Landroid/view/MotionEvent;->getPointerId(I)I
  
      move-result v1
  
+     .line 652
      .local v1, pointerId:I
      iget v3, p0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
  
      if-ne v1, v3, :cond_0
  
+     .line 656
      if-nez v2, :cond_1
  
      const/4 v0, 0x1
  
+     .line 657
      .local v0, newPointerIndex:I
      :goto_0
      invoke-virtual {p1, v0}, Landroid/view/MotionEvent;->getX(I)F
***************
*** 787,810 ****
--- 890,918 ----
  
      iput v3, p0, Landroid/widget/HorizontalScrollView;->mLastMotionX:F
  
+     .line 658
      invoke-virtual {p1, v0}, Landroid/view/MotionEvent;->getPointerId(I)I
  
      move-result v3
  
      iput v3, p0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
  
+     .line 659
      iget-object v3, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      if-eqz v3, :cond_0
  
+     .line 660
      iget-object v3, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      invoke-virtual {v3}, Landroid/view/VelocityTracker;->clear()V
  
+     .line 663
      .end local v0           #newPointerIndex:I
      :cond_0
      return-void
  
+     .line 656
      :cond_1
      const/4 v0, 0x0
  
***************
*** 815,832 ****
--- 923,944 ----
      .locals 1
  
      .prologue
+     .line 413
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      if-eqz v0, :cond_0
  
+     .line 414
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      invoke-virtual {v0}, Landroid/view/VelocityTracker;->recycle()V
  
+     .line 415
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
+     .line 417
      :cond_0
      return-void
  .end method
***************
*** 838,858 ****
--- 950,975 ----
      .parameter "right"
  
      .prologue
+     .line 946
      const/4 v4, 0x1
  
+     .line 948
      .local v4, handled:Z
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
      move-result v6
  
+     .line 949
      .local v6, width:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getScrollX()I
  
      move-result v0
  
+     .line 950
      .local v0, containerLeft:I
      add-int v1, v0, v6
  
+     .line 951
      .local v1, containerRight:I
      const/16 v7, 0x11
  
***************
*** 860,883 ****
--- 977,1006 ----
  
      const/4 v3, 0x1
  
+     .line 953
      .local v3, goLeft:Z
      :goto_0
      invoke-direct {p0, v3, p2, p3}, Landroid/widget/HorizontalScrollView;->findFocusableViewInBounds(ZII)Landroid/view/View;
  
      move-result-object v5
  
+     .line 954
      .local v5, newFocused:Landroid/view/View;
      if-nez v5, :cond_0
  
+     .line 955
      move-object v5, p0
  
+     .line 958
      :cond_0
      if-lt p2, v0, :cond_3
  
      if-gt p3, v1, :cond_3
  
+     .line 959
      const/4 v4, 0x0
  
+     .line 965
      :goto_1
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->findFocus()Landroid/view/View;
  
***************
*** 887,895 ****
--- 1010,1020 ----
  
      invoke-virtual {v5, p1}, Landroid/view/View;->requestFocus(I)Z
  
+     .line 967
      :cond_1
      return v4
  
+     .line 951
      .end local v3           #goLeft:Z
      .end local v5           #newFocused:Landroid/view/View;
      :cond_2
***************
*** 897,902 ****
--- 1022,1028 ----
  
      goto :goto_0
  
+     .line 961
      .restart local v3       #goLeft:Z
      .restart local v5       #newFocused:Landroid/view/View;
      :cond_3
***************
*** 904,915 ****
--- 1030,1043 ----
  
      sub-int v2, p2, v0
  
+     .line 962
      .local v2, delta:I
      :goto_2
      invoke-direct {p0, v2}, Landroid/widget/HorizontalScrollView;->doScrollX(I)V
  
      goto :goto_1
  
+     .line 961
      .end local v2           #delta:I
      :cond_4
      sub-int v2, p3, v1
***************
*** 922,948 ****
--- 1050,1082 ----
      .parameter "child"
  
      .prologue
+     .line 1218
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p1, v1}, Landroid/view/View;->getDrawingRect(Landroid/graphics/Rect;)V
  
+     .line 1221
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0, p1, v1}, Landroid/widget/HorizontalScrollView;->offsetDescendantRectToMyCoords(Landroid/view/View;Landroid/graphics/Rect;)V
  
+     .line 1223
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0, v1}, Landroid/widget/HorizontalScrollView;->computeScrollDeltaToGetChildRectOnScreen(Landroid/graphics/Rect;)I
  
      move-result v0
  
+     .line 1225
      .local v0, scrollDelta:I
      if-eqz v0, :cond_0
  
+     .line 1226
      const/4 v1, 0x0
  
      invoke-virtual {p0, v0, v1}, Landroid/widget/HorizontalScrollView;->scrollBy(II)V
  
+     .line 1228
      :cond_0
      return-void
  .end method
***************
*** 955,977 ****
--- 1089,1117 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1239
      invoke-virtual {p0, p1}, Landroid/widget/HorizontalScrollView;->computeScrollDeltaToGetChildRectOnScreen(Landroid/graphics/Rect;)I
  
      move-result v0
  
+     .line 1240
      .local v0, delta:I
      if-eqz v0, :cond_1
  
      const/4 v1, 0x1
  
+     .line 1241
      .local v1, scroll:Z
      :goto_0
      if-eqz v1, :cond_0
  
+     .line 1242
      if-eqz p2, :cond_2
  
+     .line 1243
      invoke-virtual {p0, v0, v2}, Landroid/widget/HorizontalScrollView;->scrollBy(II)V
  
+     .line 1248
      :cond_0
      :goto_1
      return v1
***************
*** 980,987 ****
--- 1120,1129 ----
      :cond_1
      move v1, v2
  
+     .line 1240
      goto :goto_0
  
+     .line 1245
      .restart local v1       #scroll:Z
      :cond_2
      invoke-virtual {p0, v0, v2}, Landroid/widget/HorizontalScrollView;->smoothScrollBy(II)V
***************
*** 996,1007 ****
--- 1138,1151 ----
      .parameter "child"
  
      .prologue
+     .line 212
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v0
  
      if-lez v0, :cond_0
  
+     .line 213
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "HorizontalScrollView can host only one direct child"
***************
*** 1010,1018 ****
--- 1154,1164 ----
  
      throw v0
  
+     .line 216
      :cond_0
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->addView(Landroid/view/View;)V
  
+     .line 217
      return-void
  .end method
  
***************
*** 1022,1033 ****
--- 1168,1181 ----
      .parameter "index"
  
      .prologue
+     .line 221
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v0
  
      if-lez v0, :cond_0
  
+     .line 222
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "HorizontalScrollView can host only one direct child"
***************
*** 1036,1044 ****
--- 1184,1194 ----
  
      throw v0
  
+     .line 225
      :cond_0
      invoke-super {p0, p1, p2}, Landroid/widget/FrameLayout;->addView(Landroid/view/View;I)V
  
+     .line 226
      return-void
  .end method
  
***************
*** 1049,1060 ****
--- 1199,1212 ----
      .parameter "params"
  
      .prologue
+     .line 239
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v0
  
      if-lez v0, :cond_0
  
+     .line 240
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "HorizontalScrollView can host only one direct child"
***************
*** 1063,1071 ****
--- 1215,1225 ----
  
      throw v0
  
+     .line 243
      :cond_0
      invoke-super {p0, p1, p2, p3}, Landroid/widget/FrameLayout;->addView(Landroid/view/View;ILandroid/view/ViewGroup$LayoutParams;)V
  
+     .line 244
      return-void
  .end method
  
***************
*** 1075,1086 ****
--- 1229,1242 ----
      .parameter "params"
  
      .prologue
+     .line 230
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v0
  
      if-lez v0, :cond_0
  
+     .line 231
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "HorizontalScrollView can host only one direct child"
***************
*** 1089,1097 ****
--- 1245,1255 ----
  
      throw v0
  
+     .line 234
      :cond_0
      invoke-super {p0, p1, p2}, Landroid/widget/FrameLayout;->addView(Landroid/view/View;Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 235
      return-void
  .end method
  
***************
*** 1104,1118 ****
--- 1262,1279 ----
  
      const/4 v7, 0x0
  
+     .line 979
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->findFocus()Landroid/view/View;
  
      move-result-object v0
  
+     .line 980
      .local v0, currentFocused:Landroid/view/View;
      if-ne v0, p0, :cond_0
  
      const/4 v0, 0x0
  
+     .line 982
      :cond_0
      invoke-static {}, Landroid/view/FocusFinder;->getInstance()Landroid/view/FocusFinder;
  
***************
*** 1122,1132 ****
--- 1283,1295 ----
  
      move-result-object v4
  
+     .line 984
      .local v4, nextFocused:Landroid/view/View;
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getMaxScrollAmount()I
  
      move-result v3
  
+     .line 986
      .local v3, maxJump:I
      if-eqz v4, :cond_3
  
***************
*** 1136,1160 ****
--- 1299,1329 ----
  
      if-eqz v8, :cond_3
  
+     .line 987
      iget-object v7, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {v4, v7}, Landroid/view/View;->getDrawingRect(Landroid/graphics/Rect;)V
  
+     .line 988
      iget-object v7, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0, v4, v7}, Landroid/widget/HorizontalScrollView;->offsetDescendantRectToMyCoords(Landroid/view/View;Landroid/graphics/Rect;)V
  
+     .line 989
      iget-object v7, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0, v7}, Landroid/widget/HorizontalScrollView;->computeScrollDeltaToGetChildRectOnScreen(Landroid/graphics/Rect;)I
  
      move-result v6
  
+     .line 990
      .local v6, scrollDelta:I
      invoke-direct {p0, v6}, Landroid/widget/HorizontalScrollView;->doScrollX(I)V
  
+     .line 991
      invoke-virtual {v4, p1}, Landroid/view/View;->requestFocus(I)Z
  
+     .line 1014
      :goto_0
      if-eqz v0, :cond_1
  
***************
*** 1170,1188 ****
--- 1339,1362 ----
  
      if-eqz v7, :cond_1
  
+     .line 1021
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getDescendantFocusability()I
  
      move-result v2
  
+     .line 1022
      .local v2, descendantFocusability:I
      const/high16 v7, 0x2
  
      invoke-virtual {p0, v7}, Landroid/widget/HorizontalScrollView;->setDescendantFocusability(I)V
  
+     .line 1023
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->requestFocus()Z
  
+     .line 1024
      invoke-virtual {p0, v2}, Landroid/widget/HorizontalScrollView;->setDescendantFocusability(I)V
  
+     .line 1026
      .end local v2           #descendantFocusability:I
      :cond_1
      const/4 v7, 0x1
***************
*** 1190,1199 ****
--- 1364,1375 ----
      :cond_2
      return v7
  
+     .line 994
      .end local v6           #scrollDelta:I
      :cond_3
      move v6, v3
  
+     .line 996
      .restart local v6       #scrollDelta:I
      const/16 v8, 0x11
  
***************
*** 1205,1218 ****
--- 1381,1397 ----
  
      if-ge v8, v6, :cond_5
  
+     .line 997
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getScrollX()I
  
      move-result v6
  
+     .line 1008
      :cond_4
      :goto_1
      if-eqz v6, :cond_2
  
+     .line 1011
      if-ne p1, v10, :cond_6
  
      move v7, v6
***************
*** 1222,1227 ****
--- 1401,1407 ----
  
      goto :goto_0
  
+     .line 998
      :cond_5
      if-ne p1, v10, :cond_4
  
***************
*** 1231,1236 ****
--- 1411,1417 ----
  
      if-lez v8, :cond_4
  
+     .line 1000
      invoke-virtual {p0, v7}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
      move-result-object v8
***************
*** 1239,1244 ****
--- 1420,1426 ----
  
      move-result v1
  
+     .line 1002
      .local v1, daRight:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getScrollX()I
  
***************
*** 1250,1264 ****
--- 1432,1449 ----
  
      add-int v5, v8, v9
  
+     .line 1004
      .local v5, screenRight:I
      sub-int v8, v1, v5
  
      if-ge v8, v3, :cond_4
  
+     .line 1005
      sub-int v6, v1, v5
  
      goto :goto_1
  
+     .line 1011
      .end local v1           #daRight:I
      .end local v5           #screenRight:I
      :cond_6
***************
*** 1271,1276 ****
--- 1456,1462 ----
      .locals 2
  
      .prologue
+     .line 1130
      const/4 v0, 0x0
  
      invoke-super {p0}, Landroid/widget/FrameLayout;->computeHorizontalScrollOffset()I
***************
*** 1290,1299 ****
--- 1476,1487 ----
      .prologue
      const/4 v7, 0x0
  
+     .line 1110
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v1
  
+     .line 1111
      .local v1, count:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
***************
*** 1307,1319 ****
--- 1495,1510 ----
  
      sub-int v0, v5, v6
  
+     .line 1112
      .local v0, contentWidth:I
      if-nez v1, :cond_0
  
+     .line 1125
      .end local v0           #contentWidth:I
      :goto_0
      return v0
  
+     .line 1116
      .restart local v0       #contentWidth:I
      :cond_0
      invoke-virtual {p0, v7}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
***************
*** 1324,1332 ****
--- 1515,1525 ----
  
      move-result v3
  
+     .line 1117
      .local v3, scrollRange:I
      iget v4, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
+     .line 1118
      .local v4, scrollX:I
      sub-int v5, v3, v0
  
***************
*** 1334,1353 ****
--- 1527,1551 ----
  
      move-result v2
  
+     .line 1119
      .local v2, overscrollRight:I
      if-gez v4, :cond_2
  
+     .line 1120
      sub-int/2addr v3, v4
  
      :cond_1
      :goto_1
      move v0, v3
  
+     .line 1125
      goto :goto_0
  
+     .line 1121
      :cond_2
      if-le v4, v2, :cond_1
  
+     .line 1122
      sub-int v5, v4, v2
  
      add-int/2addr v3, v5
***************
*** 1363,1368 ****
--- 1561,1567 ----
  
      const/4 v6, 0x0
  
+     .line 1164
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
      invoke-virtual {v0}, Landroid/widget/OverScroller;->computeScrollOffset()Z
***************
*** 1371,1381 ****
--- 1570,1583 ----
  
      if-eqz v0, :cond_3
  
+     .line 1181
      iget v3, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
+     .line 1182
      .local v3, oldX:I
      iget v4, p0, Landroid/widget/HorizontalScrollView;->mScrollY:I
  
+     .line 1183
      .local v4, oldY:I
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
***************
*** 1383,1388 ****
--- 1585,1591 ----
  
      move-result v12
  
+     .line 1184
      .local v12, x:I
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
***************
*** 1390,1410 ****
--- 1593,1617 ----
  
      move-result v13
  
+     .line 1186
      .local v13, y:I
      if-ne v3, v12, :cond_0
  
      if-eq v4, v13, :cond_2
  
+     .line 1187
      :cond_0
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->getScrollRange()I
  
      move-result v5
  
+     .line 1188
      .local v5, range:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getOverScrollMode()I
  
      move-result v11
  
+     .line 1189
      .local v11, overscrollMode:I
      if-eqz v11, :cond_1
  
***************
*** 1412,1417 ****
--- 1619,1625 ----
  
      if-lez v5, :cond_4
  
+     .line 1192
      .local v10, canOverscroll:Z
      :cond_1
      :goto_0
***************
*** 1429,1446 ****
--- 1637,1658 ----
  
      invoke-virtual/range {v0 .. v9}, Landroid/widget/HorizontalScrollView;->overScrollBy(IIIIIIIIZ)Z
  
+     .line 1194
      iget v0, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mScrollY:I
  
      invoke-virtual {p0, v0, v1, v3, v4}, Landroid/widget/HorizontalScrollView;->onScrollChanged(IIII)V
  
+     .line 1196
      if-eqz v10, :cond_2
  
+     .line 1197
      if-gez v12, :cond_5
  
      if-ltz v3, :cond_5
  
+     .line 1198
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
***************
*** 1453,1458 ****
--- 1665,1671 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/EdgeEffect;->onAbsorb(I)V
  
+     .line 1205
      .end local v5           #range:I
      .end local v10           #canOverscroll:Z
      .end local v11           #overscrollMode:I
***************
*** 1460,1467 ****
--- 1673,1682 ----
      :goto_1
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->awakenScrollBars()Z
  
+     .line 1208
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->postInvalidate()V
  
+     .line 1210
      .end local v3           #oldX:I
      .end local v4           #oldY:I
      .end local v12           #x:I
***************
*** 1478,1491 ****
--- 1693,1709 ----
      :cond_4
      move v10, v6
  
+     .line 1189
      goto :goto_0
  
+     .line 1199
      .restart local v10       #canOverscroll:Z
      :cond_5
      if-le v12, v5, :cond_2
  
      if-gt v3, v5, :cond_2
  
+     .line 1200
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
  
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
***************
*** 1508,1513 ****
--- 1726,1732 ----
      .prologue
      const/4 v7, 0x0
  
+     .line 1260
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v8
***************
*** 1516,1550 ****
--- 1735,1777 ----
  
      move v5, v7
  
+     .line 1314
      :cond_0
      :goto_0
      return v5
  
+     .line 1262
      :cond_1
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
      move-result v6
  
+     .line 1263
      .local v6, width:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getScrollX()I
  
      move-result v3
  
+     .line 1264
      .local v3, screenLeft:I
      add-int v4, v3, v6
  
+     .line 1266
      .local v4, screenRight:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getHorizontalFadingEdgeLength()I
  
      move-result v1
  
+     .line 1269
      .local v1, fadingEdge:I
      iget v8, p1, Landroid/graphics/Rect;->left:I
  
      if-lez v8, :cond_2
  
+     .line 1270
      add-int/2addr v3, v1
  
+     .line 1274
      :cond_2
      iget v8, p1, Landroid/graphics/Rect;->right:I
  
***************
*** 1558,1568 ****
--- 1785,1798 ----
  
      if-ge v8, v9, :cond_3
  
+     .line 1275
      sub-int/2addr v4, v1
  
+     .line 1278
      :cond_3
      const/4 v5, 0x0
  
+     .line 1280
      .local v5, scrollXDelta:I
      iget v8, p1, Landroid/graphics/Rect;->right:I
  
***************
*** 1572,1589 ****
--- 1802,1822 ----
  
      if-le v8, v3, :cond_5
  
+     .line 1285
      invoke-virtual {p1}, Landroid/graphics/Rect;->width()I
  
      move-result v8
  
      if-le v8, v6, :cond_4
  
+     .line 1287
      iget v8, p1, Landroid/graphics/Rect;->left:I
  
      sub-int/2addr v8, v3
  
      add-int/2addr v5, v8
  
+     .line 1294
      :goto_1
      invoke-virtual {p0, v7}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
***************
*** 1593,1608 ****
--- 1826,1845 ----
  
      move-result v2
  
+     .line 1295
      .local v2, right:I
      sub-int v0, v2, v4
  
+     .line 1296
      .local v0, distanceToRight:I
      invoke-static {v5, v0}, Ljava/lang/Math;->min(II)I
  
      move-result v5
  
+     .line 1298
      goto :goto_0
  
+     .line 1290
      .end local v0           #distanceToRight:I
      .end local v2           #right:I
      :cond_4
***************
*** 1614,1619 ****
--- 1851,1857 ----
  
      goto :goto_1
  
+     .line 1298
      :cond_5
      iget v7, p1, Landroid/graphics/Rect;->left:I
  
***************
*** 1623,1640 ****
--- 1861,1881 ----
  
      if-ge v7, v4, :cond_0
  
+     .line 1303
      invoke-virtual {p1}, Landroid/graphics/Rect;->width()I
  
      move-result v7
  
      if-le v7, v6, :cond_6
  
+     .line 1305
      iget v7, p1, Landroid/graphics/Rect;->right:I
  
      sub-int v7, v4, v7
  
      sub-int/2addr v5, v7
  
+     .line 1312
      :goto_2
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getScrollX()I
  
***************
*** 1648,1653 ****
--- 1889,1895 ----
  
      goto :goto_0
  
+     .line 1308
      :cond_6
      iget v7, p1, Landroid/graphics/Rect;->left:I
  
***************
*** 1663,1668 ****
--- 1905,1911 ----
      .parameter "event"
  
      .prologue
+     .line 334
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->dispatchKeyEvent(Landroid/view/KeyEvent;)Z
  
      move-result v0
***************
*** 1692,1705 ****
--- 1935,1952 ----
      .parameter "canvas"
  
      .prologue
+     .line 1493
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->draw(Landroid/graphics/Canvas;)V
  
+     .line 1494
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
      if-eqz v4, :cond_3
  
+     .line 1495
      iget v2, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
+     .line 1496
      .local v2, scrollX:I
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
***************
*** 1709,1718 ****
--- 1956,1967 ----
  
      if-nez v4, :cond_1
  
+     .line 1497
      invoke-virtual {p1}, Landroid/graphics/Canvas;->save()I
  
      move-result v1
  
+     .line 1498
      .local v1, restoreCount:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getHeight()I
  
***************
*** 1726,1736 ****
--- 1975,1987 ----
  
      sub-int v0, v4, v5
  
+     .line 1500
      .local v0, height:I
      const/high16 v4, 0x4387
  
      invoke-virtual {p1, v4}, Landroid/graphics/Canvas;->rotate(F)V
  
+     .line 1501
      neg-int v4, v0
  
      iget v5, p0, Landroid/widget/HorizontalScrollView;->mPaddingTop:I
***************
*** 1749,1754 ****
--- 2000,2006 ----
  
      invoke-virtual {p1, v4, v5}, Landroid/graphics/Canvas;->translate(FF)V
  
+     .line 1502
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
***************
*** 1757,1762 ****
--- 2009,2015 ----
  
      invoke-virtual {v4, v0, v5}, Landroid/widget/EdgeEffect;->setSize(II)V
  
+     .line 1503
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
      invoke-virtual {v4, p1}, Landroid/widget/EdgeEffect;->draw(Landroid/graphics/Canvas;)Z
***************
*** 1765,1775 ****
--- 2018,2031 ----
  
      if-eqz v4, :cond_0
  
+     .line 1504
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->invalidate()V
  
+     .line 1506
      :cond_0
      invoke-virtual {p1, v1}, Landroid/graphics/Canvas;->restoreToCount(I)V
  
+     .line 1508
      .end local v0           #height:I
      .end local v1           #restoreCount:I
      :cond_1
***************
*** 1781,1795 ****
--- 2037,2054 ----
  
      if-nez v4, :cond_3
  
+     .line 1509
      invoke-virtual {p1}, Landroid/graphics/Canvas;->save()I
  
      move-result v1
  
+     .line 1510
      .restart local v1       #restoreCount:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
      move-result v3
  
+     .line 1511
      .local v3, width:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getHeight()I
  
***************
*** 1803,1813 ****
--- 2062,2074 ----
  
      sub-int v0, v4, v5
  
+     .line 1513
      .restart local v0       #height:I
      const/high16 v4, 0x42b4
  
      invoke-virtual {p1, v4}, Landroid/graphics/Canvas;->rotate(F)V
  
+     .line 1514
      iget v4, p0, Landroid/widget/HorizontalScrollView;->mPaddingTop:I
  
      neg-int v4, v4
***************
*** 1830,1839 ****
--- 2091,2102 ----
  
      invoke-virtual {p1, v4, v5}, Landroid/graphics/Canvas;->translate(FF)V
  
+     .line 1516
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
  
      invoke-virtual {v4, v0, v3}, Landroid/widget/EdgeEffect;->setSize(II)V
  
+     .line 1517
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
  
      invoke-virtual {v4, p1}, Landroid/widget/EdgeEffect;->draw(Landroid/graphics/Canvas;)Z
***************
*** 1842,1852 ****
--- 2105,2118 ----
  
      if-eqz v4, :cond_2
  
+     .line 1518
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->invalidate()V
  
+     .line 1520
      :cond_2
      invoke-virtual {p1, v1}, Landroid/graphics/Canvas;->restoreToCount(I)V
  
+     .line 1523
      .end local v0           #height:I
      .end local v1           #restoreCount:I
      .end local v2           #scrollX:I
***************
*** 1866,1896 ****
--- 2132,2168 ----
  
      const/16 v5, 0x42
  
+     .line 346
      iget-object v6, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {v6}, Landroid/graphics/Rect;->setEmpty()V
  
+     .line 348
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->canScroll()Z
  
      move-result v6
  
      if-nez v6, :cond_2
  
+     .line 349
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->isFocused()Z
  
      move-result v4
  
      if-eqz v4, :cond_1
  
+     .line 350
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->findFocus()Landroid/view/View;
  
      move-result-object v0
  
+     .line 351
      .local v0, currentFocused:Landroid/view/View;
      if-ne v0, p0, :cond_0
  
      const/4 v0, 0x0
  
+     .line 352
      :cond_0
      invoke-static {}, Landroid/view/FocusFinder;->getInstance()Landroid/view/FocusFinder;
  
***************
*** 1900,1905 ****
--- 2172,2178 ----
  
      move-result-object v2
  
+     .line 354
      .local v2, nextFocused:Landroid/view/View;
      if-eqz v2, :cond_1
  
***************
*** 1913,1927 ****
--- 2186,2203 ----
  
      const/4 v3, 0x1
  
+     .line 383
      .end local v0           #currentFocused:Landroid/view/View;
      .end local v2           #nextFocused:Landroid/view/View;
      :cond_1
      :goto_0
      return v3
  
+     .line 360
      :cond_2
      const/4 v1, 0x0
  
+     .line 361
      .local v1, handled:Z
      invoke-virtual {p1}, Landroid/view/KeyEvent;->getAction()I
  
***************
*** 1929,1934 ****
--- 2205,2211 ----
  
      if-nez v3, :cond_3
  
+     .line 362
      invoke-virtual {p1}, Landroid/view/KeyEvent;->getKeyCode()I
  
      move-result v3
***************
*** 1939,1946 ****
--- 2216,2225 ----
      :goto_1
      move v3, v1
  
+     .line 383
      goto :goto_0
  
+     .line 364
      :sswitch_0
      invoke-virtual {p1}, Landroid/view/KeyEvent;->isAltPressed()Z
  
***************
*** 1948,1966 ****
--- 2227,2249 ----
  
      if-nez v3, :cond_4
  
+     .line 365
      invoke-virtual {p0, v4}, Landroid/widget/HorizontalScrollView;->arrowScroll(I)Z
  
      move-result v1
  
      goto :goto_1
  
+     .line 367
      :cond_4
      invoke-virtual {p0, v4}, Landroid/widget/HorizontalScrollView;->fullScroll(I)Z
  
      move-result v1
  
+     .line 369
      goto :goto_1
  
+     .line 371
      :sswitch_1
      invoke-virtual {p1}, Landroid/view/KeyEvent;->isAltPressed()Z
  
***************
*** 1968,1986 ****
--- 2251,2273 ----
  
      if-nez v3, :cond_5
  
+     .line 372
      invoke-virtual {p0, v5}, Landroid/widget/HorizontalScrollView;->arrowScroll(I)Z
  
      move-result v1
  
      goto :goto_1
  
+     .line 374
      :cond_5
      invoke-virtual {p0, v5}, Landroid/widget/HorizontalScrollView;->fullScroll(I)Z
  
      move-result v1
  
+     .line 376
      goto :goto_1
  
+     .line 378
      :sswitch_2
      invoke-virtual {p1}, Landroid/view/KeyEvent;->isShiftPressed()Z
  
***************
*** 2000,2005 ****
--- 2287,2293 ----
  
      goto :goto_2
  
+     .line 362
      :sswitch_data_0
      .sparse-switch
          0x15 -> :sswitch_0
***************
*** 2013,2024 ****
--- 2301,2314 ----
      .parameter "velocityX"
  
      .prologue
+     .line 1432
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v1
  
      if-lez v1, :cond_2
  
+     .line 1433
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
      move-result v1
***************
*** 2035,2040 ****
--- 2325,2331 ----
  
      sub-int v16, v1, v2
  
+     .line 1434
      .local v16, width:I
      const/4 v1, 0x0
  
***************
*** 2048,2053 ****
--- 2339,2345 ----
  
      move-result v15
  
+     .line 1436
      .local v15, right:I
      move-object/from16 v0, p0
  
***************
*** 2085,2100 ****
--- 2377,2395 ----
  
      invoke-virtual/range {v1 .. v11}, Landroid/widget/OverScroller;->fling(IIIIIIIIII)V
  
+     .line 1439
      if-lez p1, :cond_3
  
      const/4 v13, 0x1
  
+     .line 1441
      .local v13, movingRight:Z
      :goto_0
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->findFocus()Landroid/view/View;
  
      move-result-object v12
  
+     .line 1442
      .local v12, currentFocused:Landroid/view/View;
      move-object/from16 v0, p0
  
***************
*** 2110,2123 ****
--- 2405,2422 ----
  
      move-result-object v14
  
+     .line 1445
      .local v14, newFocused:Landroid/view/View;
      if-nez v14, :cond_0
  
+     .line 1446
      move-object/from16 v14, p0
  
+     .line 1449
      :cond_0
      if-eq v14, v12, :cond_1
  
+     .line 1450
      if-eqz v13, :cond_4
  
      const/16 v1, 0x42
***************
*** 2125,2133 ****
--- 2424,2434 ----
      :goto_1
      invoke-virtual {v14, v1}, Landroid/view/View;->requestFocus(I)Z
  
+     .line 1453
      :cond_1
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->invalidate()V
  
+     .line 1455
      .end local v12           #currentFocused:Landroid/view/View;
      .end local v13           #movingRight:Z
      .end local v14           #newFocused:Landroid/view/View;
***************
*** 2136,2141 ****
--- 2437,2443 ----
      :cond_2
      return-void
  
+     .line 1439
      .restart local v15       #right:I
      .restart local v16       #width:I
      :cond_3
***************
*** 2143,2148 ****
--- 2445,2451 ----
  
      goto :goto_0
  
+     .line 1450
      .restart local v12       #currentFocused:Landroid/view/View;
      .restart local v13       #movingRight:Z
      .restart local v14       #newFocused:Landroid/view/View;
***************
*** 2159,2198 ****
--- 2462,2510 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 915
      const/16 v5, 0x42
  
      if-ne p1, v5, :cond_1
  
      const/4 v1, 0x1
  
+     .line 916
      .local v1, right:Z
      :goto_0
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
      move-result v3
  
+     .line 918
      .local v3, width:I
      iget-object v5, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      iput v4, v5, Landroid/graphics/Rect;->left:I
  
+     .line 919
      iget-object v5, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      iput v3, v5, Landroid/graphics/Rect;->right:I
  
+     .line 921
      if-eqz v1, :cond_0
  
+     .line 922
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v0
  
+     .line 923
      .local v0, count:I
      if-lez v0, :cond_0
  
+     .line 924
      invoke-virtual {p0, v4}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
      move-result-object v2
  
+     .line 925
      .local v2, view:Landroid/view/View;
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
***************
*** 2202,2207 ****
--- 2514,2520 ----
  
      iput v5, v4, Landroid/graphics/Rect;->right:I
  
+     .line 926
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      iget-object v5, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
***************
*** 2212,2217 ****
--- 2525,2531 ----
  
      iput v5, v4, Landroid/graphics/Rect;->left:I
  
+     .line 930
      .end local v0           #count:I
      .end local v2           #view:Landroid/view/View;
      :cond_0
***************
*** 2234,2239 ****
--- 2548,2554 ----
      :cond_1
      move v1, v4
  
+     .line 915
      goto :goto_0
  .end method
  
***************
*** 2241,2267 ****
--- 2556,2588 ----
      .locals 3
  
      .prologue
+     .line 160
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v1
  
      if-nez v1, :cond_0
  
+     .line 161
      const/4 v1, 0x0
  
+     .line 169
      :goto_0
      return v1
  
+     .line 164
      :cond_0
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getHorizontalFadingEdgeLength()I
  
      move-result v0
  
+     .line 165
      .local v0, length:I
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
      if-ge v1, v0, :cond_1
  
+     .line 166
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
      int-to-float v1, v1
***************
*** 2272,2277 ****
--- 2593,2599 ----
  
      goto :goto_0
  
+     .line 169
      :cond_1
      const/high16 v1, 0x3f80
  
***************
*** 2282,2287 ****
--- 2604,2610 ----
      .locals 3
  
      .prologue
+     .line 193
      const/high16 v0, 0x3f00
  
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mRight:I
***************
*** 2303,2324 ****
--- 2626,2652 ----
      .locals 5
  
      .prologue
+     .line 174
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v3
  
      if-nez v3, :cond_0
  
+     .line 175
      const/4 v3, 0x0
  
+     .line 185
      :goto_0
      return v3
  
+     .line 178
      :cond_0
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getHorizontalFadingEdgeLength()I
  
      move-result v0
  
+     .line 179
      .local v0, length:I
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
***************
*** 2328,2333 ****
--- 2656,2662 ----
  
      sub-int v1, v3, v4
  
+     .line 180
      .local v1, rightEdge:I
      const/4 v3, 0x0
  
***************
*** 2345,2353 ****
--- 2674,2684 ----
  
      sub-int v2, v3, v1
  
+     .line 181
      .local v2, span:I
      if-ge v2, v0, :cond_1
  
+     .line 182
      int-to-float v3, v2
  
      int-to-float v4, v0
***************
*** 2356,2361 ****
--- 2687,2693 ----
  
      goto :goto_0
  
+     .line 185
      :cond_1
      const/high16 v3, 0x3f80
  
***************
*** 2366,2371 ****
--- 2698,2704 ----
      .locals 1
  
      .prologue
+     .line 267
      iget-boolean v0, p0, Landroid/widget/HorizontalScrollView;->mFillViewport:Z
  
      return v0
***************
*** 2375,2380 ****
--- 2708,2714 ----
      .locals 1
  
      .prologue
+     .line 290
      iget-boolean v0, p0, Landroid/widget/HorizontalScrollView;->mSmoothScrollingEnabled:Z
  
      return v0
***************
*** 2389,2398 ****
--- 2723,2734 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 1135
      invoke-virtual {p1}, Landroid/view/View;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
  
      move-result-object v2
  
+     .line 1140
      .local v2, lp:Landroid/view/ViewGroup$LayoutParams;
      iget v3, p0, Landroid/widget/HorizontalScrollView;->mPaddingTop:I
  
***************
*** 2406,2419 ****
--- 2742,2758 ----
  
      move-result v0
  
+     .line 1143
      .local v0, childHeightMeasureSpec:I
      invoke-static {v5, v5}, Landroid/view/View$MeasureSpec;->makeMeasureSpec(II)I
  
      move-result v1
  
+     .line 1145
      .local v1, childWidthMeasureSpec:I
      invoke-virtual {p1, v1, v0}, Landroid/view/View;->measure(II)V
  
+     .line 1146
      return-void
  .end method
  
***************
*** 2426,2437 ****
--- 2765,2778 ----
      .parameter "heightUsed"
  
      .prologue
+     .line 1151
      invoke-virtual {p1}, Landroid/view/View;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
  
      move-result-object v2
  
      check-cast v2, Landroid/view/ViewGroup$MarginLayoutParams;
  
+     .line 1153
      .local v2, lp:Landroid/view/ViewGroup$MarginLayoutParams;
      iget v3, p0, Landroid/widget/HorizontalScrollView;->mPaddingTop:I
  
***************
*** 2455,2460 ****
--- 2796,2802 ----
  
      move-result v0
  
+     .line 1156
      .local v0, childHeightMeasureSpec:I
      iget v3, v2, Landroid/view/ViewGroup$MarginLayoutParams;->leftMargin:I
  
***************
*** 2468,2476 ****
--- 2810,2820 ----
  
      move-result v1
  
+     .line 1159
      .local v1, childWidthMeasureSpec:I
      invoke-virtual {p1, v1, v0}, Landroid/view/View;->measure(II)V
  
+     .line 1160
      return-void
  .end method
  
***************
*** 2479,2484 ****
--- 2823,2829 ----
      .parameter "event"
  
      .prologue
+     .line 667
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getSource()I
  
      move-result v5
***************
*** 2487,2498 ****
--- 2832,2845 ----
  
      if-eqz v5, :cond_0
  
+     .line 668
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getAction()I
  
      move-result v5
  
      packed-switch v5, :pswitch_data_0
  
+     .line 696
      :cond_0
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->onGenericMotionEvent(Landroid/view/MotionEvent;)Z
  
***************
*** 2501,2511 ****
--- 2848,2860 ----
      :goto_0
      return v5
  
+     .line 670
      :pswitch_0
      iget-boolean v5, p0, Landroid/widget/HorizontalScrollView;->mIsBeingDragged:Z
  
      if-nez v5, :cond_0
  
+     .line 672
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getMetaState()I
  
      move-result v5
***************
*** 2514,2519 ****
--- 2863,2869 ----
  
      if-eqz v5, :cond_2
  
+     .line 673
      const/16 v5, 0x9
  
      invoke-virtual {p1, v5}, Landroid/view/MotionEvent;->getAxisValue(I)F
***************
*** 2522,2527 ****
--- 2872,2878 ----
  
      neg-float v1, v5
  
+     .line 677
      .local v1, hscroll:F
      :goto_1
      const/4 v5, 0x0
***************
*** 2530,2535 ****
--- 2881,2887 ----
  
      if-eqz v5, :cond_0
  
+     .line 678
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getHorizontalScrollFactor()F
  
      move-result v5
***************
*** 2538,2571 ****
--- 2890,2932 ----
  
      float-to-int v0, v5
  
+     .line 679
      .local v0, delta:I
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->getScrollRange()I
  
      move-result v4
  
+     .line 680
      .local v4, range:I
      iget v3, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
+     .line 681
      .local v3, oldScrollX:I
      add-int v2, v3, v0
  
+     .line 682
      .local v2, newScrollX:I
      if-gez v2, :cond_3
  
+     .line 683
      const/4 v2, 0x0
  
+     .line 687
      :cond_1
      :goto_2
      if-eq v2, v3, :cond_0
  
+     .line 688
      iget v5, p0, Landroid/widget/HorizontalScrollView;->mScrollY:I
  
      invoke-super {p0, v2, v5}, Landroid/widget/FrameLayout;->scrollTo(II)V
  
+     .line 689
      const/4 v5, 0x1
  
      goto :goto_0
  
+     .line 675
      .end local v0           #delta:I
      .end local v1           #hscroll:F
      .end local v2           #newScrollX:I
***************
*** 2581,2586 ****
--- 2942,2948 ----
      .restart local v1       #hscroll:F
      goto :goto_1
  
+     .line 684
      .restart local v0       #delta:I
      .restart local v2       #newScrollX:I
      .restart local v3       #oldScrollX:I
***************
*** 2588,2597 ****
--- 2950,2961 ----
      :cond_3
      if-le v2, v4, :cond_1
  
+     .line 685
      move v2, v4
  
      goto :goto_2
  
+     .line 668
      nop
  
      :pswitch_data_0
***************
*** 2605,2612 ****
--- 2969,2978 ----
      .parameter "event"
  
      .prologue
+     .line 729
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->onInitializeAccessibilityEvent(Landroid/view/accessibility/AccessibilityEvent;)V
  
+     .line 730
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->getScrollRange()I
  
      move-result v0
***************
*** 2618,2643 ****
--- 2984,3015 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/view/accessibility/AccessibilityEvent;->setScrollable(Z)V
  
+     .line 731
      iget v0, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
      invoke-virtual {p1, v0}, Landroid/view/accessibility/AccessibilityEvent;->setScrollX(I)V
  
+     .line 732
      iget v0, p0, Landroid/widget/HorizontalScrollView;->mScrollY:I
  
      invoke-virtual {p1, v0}, Landroid/view/accessibility/AccessibilityEvent;->setScrollY(I)V
  
+     .line 733
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->getScrollRange()I
  
      move-result v0
  
      invoke-virtual {p1, v0}, Landroid/view/accessibility/AccessibilityEvent;->setMaxScrollX(I)V
  
+     .line 734
      iget v0, p0, Landroid/widget/HorizontalScrollView;->mScrollY:I
  
      invoke-virtual {p1, v0}, Landroid/view/accessibility/AccessibilityEvent;->setMaxScrollY(I)V
  
+     .line 735
      return-void
  
+     .line 730
      :cond_0
      const/4 v0, 0x0
  
***************
*** 2649,2656 ****
--- 3021,3030 ----
      .parameter "info"
  
      .prologue
+     .line 723
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->onInitializeAccessibilityNodeInfo(Landroid/view/accessibility/AccessibilityNodeInfo;)V
  
+     .line 724
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->getScrollRange()I
  
      move-result v0
***************
*** 2662,2669 ****
--- 3036,3045 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/view/accessibility/AccessibilityNodeInfo;->setScrollable(Z)V
  
+     .line 725
      return-void
  
+     .line 724
      :cond_0
      const/4 v0, 0x0
  
***************
*** 2681,2690 ****
--- 3057,3068 ----
  
      const/4 v3, 0x0
  
+     .line 440
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getAction()I
  
      move-result v7
  
+     .line 441
      .local v7, action:I
      const/4 v1, 0x2
  
***************
*** 2694,2707 ****
--- 3072,3088 ----
  
      if-eqz v1, :cond_0
  
+     .line 527
      :goto_0
      return v0
  
+     .line 445
      :cond_0
      and-int/lit16 v1, v7, 0xff
  
      packed-switch v1, :pswitch_data_0
  
+     .line 527
      :cond_1
      :goto_1
      :pswitch_0
***************
*** 2709,2729 ****
--- 3090,3115 ----
  
      goto :goto_0
  
+     .line 456
      :pswitch_1
      iget v8, p0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
  
+     .line 457
      .local v8, activePointerId:I
      if-eq v8, v2, :cond_1
  
+     .line 462
      invoke-virtual {p1, v8}, Landroid/view/MotionEvent;->findPointerIndex(I)I
  
      move-result v10
  
+     .line 463
      .local v10, pointerIndex:I
      invoke-virtual {p1, v10}, Landroid/view/MotionEvent;->getX(I)F
  
      move-result v11
  
+     .line 464
      .local v11, x:F
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mLastMotionX:F
  
***************
*** 2735,2755 ****
--- 3121,3147 ----
  
      float-to-int v12, v1
  
+     .line 465
      .local v12, xDiff:I
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mTouchSlop:I
  
      if-le v12, v1, :cond_1
  
+     .line 466
      iput-boolean v0, p0, Landroid/widget/HorizontalScrollView;->mIsBeingDragged:Z
  
+     .line 467
      iput v11, p0, Landroid/widget/HorizontalScrollView;->mLastMotionX:F
  
+     .line 468
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->initVelocityTrackerIfNotExists()V
  
+     .line 469
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      invoke-virtual {v1, p1}, Landroid/view/VelocityTracker;->addMovement(Landroid/view/MotionEvent;)V
  
+     .line 470
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mParent:Landroid/view/ViewParent;
  
      if-eqz v1, :cond_1
***************
*** 2760,2765 ****
--- 3152,3158 ----
  
      goto :goto_1
  
+     .line 476
      .end local v8           #activePointerId:I
      .end local v10           #pointerIndex:I
      .end local v11           #x:F
***************
*** 2769,2774 ****
--- 3162,3168 ----
  
      move-result v11
  
+     .line 477
      .restart local v11       #x:F
      float-to-int v1, v11
  
***************
*** 2784,2810 ****
--- 3178,3211 ----
  
      if-nez v1, :cond_2
  
+     .line 478
      iput-boolean v3, p0, Landroid/widget/HorizontalScrollView;->mIsBeingDragged:Z
  
+     .line 479
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->recycleVelocityTracker()V
  
      goto :goto_1
  
+     .line 487
      :cond_2
      iput v11, p0, Landroid/widget/HorizontalScrollView;->mLastMotionX:F
  
+     .line 488
      invoke-virtual {p1, v3}, Landroid/view/MotionEvent;->getPointerId(I)I
  
      move-result v1
  
      iput v1, p0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
  
+     .line 490
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->initOrResetVelocityTracker()V
  
+     .line 491
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      invoke-virtual {v1, p1}, Landroid/view/VelocityTracker;->addMovement(Landroid/view/MotionEvent;)V
  
+     .line 498
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
      invoke-virtual {v1}, Landroid/widget/OverScroller;->isFinished()Z
***************
*** 2820,2831 ****
--- 3221,3235 ----
  
      goto :goto_1
  
+     .line 505
      .end local v11           #x:F
      :pswitch_3
      iput-boolean v3, p0, Landroid/widget/HorizontalScrollView;->mIsBeingDragged:Z
  
+     .line 506
      iput v2, p0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
  
+     .line 507
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
***************
*** 2846,2860 ****
--- 3250,3267 ----
  
      if-eqz v0, :cond_1
  
+     .line 508
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->invalidate()V
  
      goto :goto_1
  
+     .line 512
      :pswitch_4
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getActionIndex()I
  
      move-result v9
  
+     .line 513
      .local v9, index:I
      invoke-virtual {p1, v9}, Landroid/view/MotionEvent;->getX(I)F
  
***************
*** 2862,2867 ****
--- 3269,3275 ----
  
      iput v0, p0, Landroid/widget/HorizontalScrollView;->mLastMotionX:F
  
+     .line 514
      invoke-virtual {p1, v9}, Landroid/view/MotionEvent;->getPointerId(I)I
  
      move-result v0
***************
*** 2870,2879 ****
--- 3278,3289 ----
  
      goto/16 :goto_1
  
+     .line 518
      .end local v9           #index:I
      :pswitch_5
      invoke-direct {p0, p1}, Landroid/widget/HorizontalScrollView;->onSecondaryPointerUp(Landroid/view/MotionEvent;)V
  
+     .line 519
      iget v0, p0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
  
      invoke-virtual {p1, v0}, Landroid/view/MotionEvent;->findPointerIndex(I)I
***************
*** 2888,2893 ****
--- 3298,3304 ----
  
      goto/16 :goto_1
  
+     .line 445
      nop
  
      :pswitch_data_0
***************
*** 2911,2922 ****
--- 3322,3336 ----
      .parameter "b"
  
      .prologue
+     .line 1382
      invoke-super/range {p0 .. p5}, Landroid/widget/FrameLayout;->onLayout(ZIIII)V
  
+     .line 1383
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/HorizontalScrollView;->mIsLayoutDirty:Z
  
+     .line 1385
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mChildToScrollTo:Landroid/view/View;
  
      if-eqz v0, :cond_0
***************
*** 2929,2949 ****
--- 3343,3367 ----
  
      if-eqz v0, :cond_0
  
+     .line 1386
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mChildToScrollTo:Landroid/view/View;
  
      invoke-direct {p0, v0}, Landroid/widget/HorizontalScrollView;->scrollToChild(Landroid/view/View;)V
  
+     .line 1388
      :cond_0
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/widget/HorizontalScrollView;->mChildToScrollTo:Landroid/view/View;
  
+     .line 1391
      iget v0, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mScrollY:I
  
      invoke-virtual {p0, v0, v1}, Landroid/widget/HorizontalScrollView;->scrollTo(II)V
  
+     .line 1392
      return-void
  .end method
  
***************
*** 2953,2993 ****
--- 3371,3420 ----
      .parameter "heightMeasureSpec"
  
      .prologue
+     .line 303
      invoke-super {p0, p1, p2}, Landroid/widget/FrameLayout;->onMeasure(II)V
  
+     .line 305
      iget-boolean v6, p0, Landroid/widget/HorizontalScrollView;->mFillViewport:Z
  
      if-nez v6, :cond_1
  
+     .line 329
      :cond_0
      :goto_0
      return-void
  
+     .line 309
      :cond_1
      invoke-static {p1}, Landroid/view/View$MeasureSpec;->getMode(I)I
  
      move-result v5
  
+     .line 310
      .local v5, widthMode:I
      if-eqz v5, :cond_0
  
+     .line 314
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v6
  
      if-lez v6, :cond_0
  
+     .line 315
      const/4 v6, 0x0
  
      invoke-virtual {p0, v6}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 316
      .local v0, child:Landroid/view/View;
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getMeasuredWidth()I
  
      move-result v4
  
+     .line 317
      .local v4, width:I
      invoke-virtual {v0}, Landroid/view/View;->getMeasuredWidth()I
  
***************
*** 2995,3006 ****
--- 3422,3435 ----
  
      if-ge v6, v4, :cond_0
  
+     .line 318
      invoke-virtual {v0}, Landroid/view/View;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
  
      move-result-object v3
  
      check-cast v3, Landroid/widget/FrameLayout$LayoutParams;
  
+     .line 320
      .local v3, lp:Landroid/widget/FrameLayout$LayoutParams;
      iget v6, p0, Landroid/widget/HorizontalScrollView;->mPaddingTop:I
  
***************
*** 3014,3034 ****
--- 3443,3467 ----
  
      move-result v1
  
+     .line 322
      .local v1, childHeightMeasureSpec:I
      iget v6, p0, Landroid/widget/HorizontalScrollView;->mPaddingLeft:I
  
      sub-int/2addr v4, v6
  
+     .line 323
      iget v6, p0, Landroid/widget/HorizontalScrollView;->mPaddingRight:I
  
      sub-int/2addr v4, v6
  
+     .line 324
      const/high16 v6, 0x4000
  
      invoke-static {v4, v6}, Landroid/view/View$MeasureSpec;->makeMeasureSpec(II)I
  
      move-result v2
  
+     .line 326
      .local v2, childWidthMeasureSpec:I
      invoke-virtual {v0, v2, v1}, Landroid/view/View;->measure(II)V
  
***************
*** 3045,3050 ****
--- 3478,3484 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 708
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
      invoke-virtual {v0}, Landroid/widget/OverScroller;->isFinished()Z
***************
*** 3053,3066 ****
--- 3487,3505 ----
  
      if-nez v0, :cond_1
  
+     .line 709
      iput p1, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
+     .line 710
      iput p2, p0, Landroid/widget/HorizontalScrollView;->mScrollY:I
  
+     .line 711
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->invalidateParentIfNeeded()V
  
+     .line 712
      if-eqz p3, :cond_0
  
+     .line 713
      iget-object v0, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
***************
*** 3077,3088 ****
--- 3516,3530 ----
  
      invoke-virtual/range {v0 .. v6}, Landroid/widget/OverScroller;->springBack(IIIIII)Z
  
+     .line 718
      :cond_0
      :goto_0
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->awakenScrollBars()Z
  
+     .line 719
      return-void
  
+     .line 716
      :cond_1
      invoke-super {p0, p1, p2}, Landroid/widget/FrameLayout;->scrollTo(II)V
  
***************
*** 3097,3108 ****
--- 3539,3553 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 1342
      const/4 v2, 0x2
  
      if-ne p1, v2, :cond_2
  
+     .line 1343
      const/16 p1, 0x42
  
+     .line 1348
      :cond_0
      :goto_0
      if-nez p2, :cond_3
***************
*** 3117,3140 ****
--- 3562,3590 ----
  
      move-result-object v0
  
+     .line 1353
      .local v0, nextFocus:Landroid/view/View;
      :goto_1
      if-nez v0, :cond_4
  
+     .line 1361
      :cond_1
      :goto_2
      return v1
  
+     .line 1344
      .end local v0           #nextFocus:Landroid/view/View;
      :cond_2
      const/4 v2, 0x1
  
      if-ne p1, v2, :cond_0
  
+     .line 1345
      const/16 p1, 0x11
  
      goto :goto_0
  
+     .line 1348
      :cond_3
      invoke-static {}, Landroid/view/FocusFinder;->getInstance()Landroid/view/FocusFinder;
  
***************
*** 3146,3151 ****
--- 3596,3602 ----
  
      goto :goto_1
  
+     .line 1357
      .restart local v0       #nextFocus:Landroid/view/View;
      :cond_4
      invoke-direct {p0, v0}, Landroid/widget/HorizontalScrollView;->isOffScreen(Landroid/view/View;)Z
***************
*** 3154,3159 ****
--- 3605,3611 ----
  
      if-nez v2, :cond_1
  
+     .line 1361
      invoke-virtual {v0, p1, p2}, Landroid/view/View;->requestFocus(ILandroid/graphics/Rect;)Z
  
      move-result v1
***************
*** 3169,3189 ****
--- 3621,3646 ----
      .parameter "oldh"
  
      .prologue
+     .line 1396
      invoke-super {p0, p1, p2, p3, p4}, Landroid/widget/FrameLayout;->onSizeChanged(IIII)V
  
+     .line 1398
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->findFocus()Landroid/view/View;
  
      move-result-object v0
  
+     .line 1399
      .local v0, currentFocused:Landroid/view/View;
      if-eqz v0, :cond_0
  
      if-ne p0, v0, :cond_1
  
+     .line 1410
      :cond_0
      :goto_0
      return-void
  
+     .line 1402
      :cond_1
      iget v3, p0, Landroid/widget/HorizontalScrollView;->mRight:I
  
***************
*** 3191,3196 ****
--- 3648,3654 ----
  
      sub-int v1, v3, v4
  
+     .line 1404
      .local v1, maxJump:I
      invoke-direct {p0, v0, v1}, Landroid/widget/HorizontalScrollView;->isWithinDeltaOfScreen(Landroid/view/View;I)Z
  
***************
*** 3198,3217 ****
--- 3656,3679 ----
  
      if-eqz v3, :cond_0
  
+     .line 1405
      iget-object v3, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {v0, v3}, Landroid/view/View;->getDrawingRect(Landroid/graphics/Rect;)V
  
+     .line 1406
      iget-object v3, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0, v0, v3}, Landroid/widget/HorizontalScrollView;->offsetDescendantRectToMyCoords(Landroid/view/View;Landroid/graphics/Rect;)V
  
+     .line 1407
      iget-object v3, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0, v3}, Landroid/widget/HorizontalScrollView;->computeScrollDeltaToGetChildRectOnScreen(Landroid/graphics/Rect;)I
  
      move-result v2
  
+     .line 1408
      .local v2, scrollDelta:I
      invoke-direct {p0, v2}, Landroid/widget/HorizontalScrollView;->doScrollX(I)V
  
***************
*** 3223,3230 ****
--- 3685,3694 ----
      .parameter "ev"
  
      .prologue
+     .line 532
      invoke-direct/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->initVelocityTrackerIfNotExists()V
  
+     .line 533
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
***************
*** 3233,3242 ****
--- 3697,3708 ----
  
      invoke-virtual {v3, v0}, Landroid/view/VelocityTracker;->addMovement(Landroid/view/MotionEvent;)V
  
+     .line 535
      invoke-virtual/range {p1 .. p1}, Landroid/view/MotionEvent;->getAction()I
  
      move-result v16
  
+     .line 537
      .local v16, action:I
      move/from16 v0, v16
  
***************
*** 3244,3249 ****
--- 3710,3716 ----
  
      packed-switch v3, :pswitch_data_0
  
+     .line 645
      :cond_0
      :goto_0
      :pswitch_0
***************
*** 3252,3257 ****
--- 3719,3725 ----
      :goto_1
      return v3
  
+     .line 539
      :pswitch_1
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
***************
*** 3266,3286 ****
--- 3734,3758 ----
  
      iput-boolean v3, v0, Landroid/widget/HorizontalScrollView;->mIsBeingDragged:Z
  
+     .line 540
      move-object/from16 v0, p0
  
      iget-boolean v3, v0, Landroid/widget/HorizontalScrollView;->mIsBeingDragged:Z
  
      if-nez v3, :cond_2
  
+     .line 541
      const/4 v3, 0x0
  
      goto :goto_1
  
+     .line 539
      :cond_1
      const/4 v3, 0x0
  
      goto :goto_2
  
+     .line 548
      :cond_2
      move-object/from16 v0, p0
  
***************
*** 3292,3303 ****
--- 3764,3777 ----
  
      if-nez v3, :cond_3
  
+     .line 549
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
      invoke-virtual {v3}, Landroid/widget/OverScroller;->abortAnimation()V
  
+     .line 553
      :cond_3
      invoke-virtual/range {p1 .. p1}, Landroid/view/MotionEvent;->getX()F
  
***************
*** 3307,3312 ****
--- 3781,3787 ----
  
      iput v3, v0, Landroid/widget/HorizontalScrollView;->mLastMotionX:F
  
+     .line 554
      const/4 v3, 0x0
  
      move-object/from16 v0, p1
***************
*** 3321,3326 ****
--- 3796,3802 ----
  
      goto :goto_0
  
+     .line 558
      :pswitch_2
      move-object/from16 v0, p0
  
***************
*** 3328,3333 ****
--- 3804,3810 ----
  
      if-eqz v3, :cond_0
  
+     .line 560
      move-object/from16 v0, p0
  
      iget v3, v0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
***************
*** 3338,3343 ****
--- 3815,3821 ----
  
      move-result v17
  
+     .line 561
      .local v17, activePointerIndex:I
      move-object/from16 v0, p1
  
***************
*** 3347,3352 ****
--- 3825,3831 ----
  
      move-result v25
  
+     .line 562
      .local v25, x:F
      move-object/from16 v0, p0
  
***************
*** 3356,3361 ****
--- 3835,3841 ----
  
      float-to-int v4, v3
  
+     .line 563
      .local v4, deltaX:I
      move/from16 v0, v25
  
***************
*** 3363,3374 ****
--- 3843,3856 ----
  
      iput v0, v1, Landroid/widget/HorizontalScrollView;->mLastMotionX:F
  
+     .line 565
      move-object/from16 v0, p0
  
      iget v0, v0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
      move/from16 v20, v0
  
+     .line 566
      .local v20, oldX:I
      move-object/from16 v0, p0
  
***************
*** 3376,3391 ****
--- 3858,3876 ----
  
      move/from16 v21, v0
  
+     .line 567
      .local v21, oldY:I
      invoke-direct/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->getScrollRange()I
  
      move-result v8
  
+     .line 568
      .local v8, range:I
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->getOverScrollMode()I
  
      move-result v22
  
+     .line 569
      .local v22, overscrollMode:I
      if-eqz v22, :cond_4
  
***************
*** 3400,3405 ****
--- 3885,3891 ----
      :cond_4
      const/16 v18, 0x1
  
+     .line 572
      .local v18, canOverscroll:Z
      :goto_3
      const/4 v5, 0x0
***************
*** 3428,3439 ****
--- 3914,3927 ----
  
      if-eqz v3, :cond_5
  
+     .line 575
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      invoke-virtual {v3}, Landroid/view/VelocityTracker;->clear()V
  
+     .line 577
      :cond_5
      move-object/from16 v0, p0
  
***************
*** 3451,3463 ****
--- 3939,3955 ----
  
      invoke-virtual {v0, v3, v5, v1, v2}, Landroid/widget/HorizontalScrollView;->onScrollChanged(IIII)V
  
+     .line 579
      if-eqz v18, :cond_0
  
+     .line 580
      add-int v23, v20, v4
  
+     .line 581
      .local v23, pulledToX:I
      if-gez v23, :cond_9
  
+     .line 582
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
***************
*** 3474,3479 ****
--- 3966,3972 ----
  
      invoke-virtual {v3, v5}, Landroid/widget/EdgeEffect;->onPull(F)V
  
+     .line 583
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
***************
*** 3484,3495 ****
--- 3977,3990 ----
  
      if-nez v3, :cond_6
  
+     .line 584
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
  
      invoke-virtual {v3}, Landroid/widget/EdgeEffect;->onRelease()V
  
+     .line 592
      :cond_6
      :goto_4
      move-object/from16 v0, p0
***************
*** 3518,3528 ****
--- 4013,4025 ----
  
      if-nez v3, :cond_0
  
+     .line 594
      :cond_7
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->invalidate()V
  
      goto/16 :goto_0
  
+     .line 569
      .end local v18           #canOverscroll:Z
      .end local v23           #pulledToX:I
      :cond_8
***************
*** 3530,3535 ****
--- 4027,4033 ----
  
      goto :goto_3
  
+     .line 586
      .restart local v18       #canOverscroll:Z
      .restart local v23       #pulledToX:I
      :cond_9
***************
*** 3537,3542 ****
--- 4035,4041 ----
  
      if-le v0, v8, :cond_6
  
+     .line 587
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
***************
*** 3553,3558 ****
--- 4052,4058 ----
  
      invoke-virtual {v3, v5}, Landroid/widget/EdgeEffect;->onPull(F)V
  
+     .line 588
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
***************
*** 3563,3568 ****
--- 4063,4069 ----
  
      if-nez v3, :cond_6
  
+     .line 589
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
***************
*** 3571,3576 ****
--- 4072,4078 ----
  
      goto :goto_4
  
+     .line 600
      .end local v4           #deltaX:I
      .end local v8           #range:I
      .end local v17           #activePointerIndex:I
***************
*** 3587,3598 ****
--- 4089,4102 ----
  
      if-eqz v3, :cond_0
  
+     .line 601
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/HorizontalScrollView;->mVelocityTracker:Landroid/view/VelocityTracker;
  
      move-object/from16 v24, v0
  
+     .line 602
      .local v24, velocityTracker:Landroid/view/VelocityTracker;
      const/16 v3, 0x3e8
  
***************
*** 3606,3611 ****
--- 4110,4116 ----
  
      invoke-virtual {v0, v3, v5}, Landroid/view/VelocityTracker;->computeCurrentVelocity(IF)V
  
+     .line 603
      move-object/from16 v0, p0
  
      iget v3, v0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
***************
*** 3620,3625 ****
--- 4125,4131 ----
  
      move/from16 v19, v0
  
+     .line 605
      .local v19, initialVelocity:I
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
***************
*** 3627,3632 ****
--- 4133,4139 ----
  
      if-lez v3, :cond_a
  
+     .line 606
      invoke-static/range {v19 .. v19}, Ljava/lang/Math;->abs(I)I
  
      move-result v3
***************
*** 3637,3642 ****
--- 4144,4150 ----
  
      if-le v3, v5, :cond_b
  
+     .line 607
      move/from16 v0, v19
  
      neg-int v3, v0
***************
*** 3645,3650 ****
--- 4153,4159 ----
  
      invoke-virtual {v0, v3}, Landroid/widget/HorizontalScrollView;->fling(I)V
  
+     .line 616
      :cond_a
      :goto_5
      const/4 v3, -0x1
***************
*** 3653,3678 ****
--- 4162,4192 ----
  
      iput v3, v0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
  
+     .line 617
      const/4 v3, 0x0
  
      move-object/from16 v0, p0
  
      iput-boolean v3, v0, Landroid/widget/HorizontalScrollView;->mIsBeingDragged:Z
  
+     .line 618
      invoke-direct/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->recycleVelocityTracker()V
  
+     .line 620
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
      if-eqz v3, :cond_0
  
+     .line 621
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
      invoke-virtual {v3}, Landroid/widget/EdgeEffect;->onRelease()V
  
+     .line 622
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
***************
*** 3681,3686 ****
--- 4195,4201 ----
  
      goto/16 :goto_0
  
+     .line 609
      :cond_b
      move-object/from16 v0, p0
  
***************
*** 3710,3719 ****
--- 4225,4236 ----
  
      if-eqz v3, :cond_a
  
+     .line 611
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->invalidate()V
  
      goto :goto_5
  
+     .line 627
      .end local v19           #initialVelocity:I
      .end local v24           #velocityTracker:Landroid/view/VelocityTracker;
      :pswitch_4
***************
*** 3729,3734 ****
--- 4246,4252 ----
  
      if-lez v3, :cond_0
  
+     .line 628
      move-object/from16 v0, p0
  
      iget-object v9, v0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
***************
*** 3757,3764 ****
--- 4275,4284 ----
  
      if-eqz v3, :cond_c
  
+     .line 629
      invoke-virtual/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->invalidate()V
  
+     .line 631
      :cond_c
      const/4 v3, -0x1
  
***************
*** 3766,3791 ****
--- 4286,4316 ----
  
      iput v3, v0, Landroid/widget/HorizontalScrollView;->mActivePointerId:I
  
+     .line 632
      const/4 v3, 0x0
  
      move-object/from16 v0, p0
  
      iput-boolean v3, v0, Landroid/widget/HorizontalScrollView;->mIsBeingDragged:Z
  
+     .line 633
      invoke-direct/range {p0 .. p0}, Landroid/widget/HorizontalScrollView;->recycleVelocityTracker()V
  
+     .line 635
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
      if-eqz v3, :cond_0
  
+     .line 636
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
      invoke-virtual {v3}, Landroid/widget/EdgeEffect;->onRelease()V
  
+     .line 637
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
***************
*** 3794,3804 ****
--- 4319,4331 ----
  
      goto/16 :goto_0
  
+     .line 642
      :pswitch_5
      invoke-direct/range {p0 .. p1}, Landroid/widget/HorizontalScrollView;->onSecondaryPointerUp(Landroid/view/MotionEvent;)V
  
      goto/16 :goto_0
  
+     .line 537
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_1
***************
*** 3818,3838 ****
--- 4345,4369 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 879
      const/16 v5, 0x42
  
      if-ne p1, v5, :cond_1
  
      const/4 v1, 0x1
  
+     .line 880
      .local v1, right:Z
      :goto_0
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
      move-result v3
  
+     .line 882
      .local v3, width:I
      if-eqz v1, :cond_2
  
+     .line 883
      iget-object v5, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getScrollX()I
***************
*** 3843,3859 ****
--- 4374,4394 ----
  
      iput v6, v5, Landroid/graphics/Rect;->left:I
  
+     .line 884
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v0
  
+     .line 885
      .local v0, count:I
      if-lez v0, :cond_0
  
+     .line 886
      invoke-virtual {p0, v4}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
      move-result-object v2
  
+     .line 887
      .local v2, view:Landroid/view/View;
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
***************
*** 3867,3872 ****
--- 4402,4408 ----
  
      if-le v4, v5, :cond_0
  
+     .line 888
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      invoke-virtual {v2}, Landroid/view/View;->getRight()I
***************
*** 3877,3882 ****
--- 4413,4419 ----
  
      iput v5, v4, Landroid/graphics/Rect;->left:I
  
+     .line 897
      .end local v0           #count:I
      .end local v2           #view:Landroid/view/View;
      :cond_0
***************
*** 3891,3896 ****
--- 4428,4434 ----
  
      iput v5, v4, Landroid/graphics/Rect;->right:I
  
+     .line 899
      iget-object v4, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      iget v4, v4, Landroid/graphics/Rect;->left:I
***************
*** 3910,3917 ****
--- 4448,4457 ----
      :cond_1
      move v1, v4
  
+     .line 879
      goto :goto_0
  
+     .line 892
      .restart local v1       #right:Z
      .restart local v3       #width:I
      :cond_2
***************
*** 3925,3936 ****
--- 4465,4478 ----
  
      iput v6, v5, Landroid/graphics/Rect;->left:I
  
+     .line 893
      iget-object v5, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      iget v5, v5, Landroid/graphics/Rect;->left:I
  
      if-gez v5, :cond_0
  
+     .line 894
      iget-object v5, p0, Landroid/widget/HorizontalScrollView;->mTempRect:Landroid/graphics/Rect;
  
      iput v4, v5, Landroid/graphics/Rect;->left:I
***************
*** 3944,3960 ****
--- 4486,4507 ----
      .parameter "focused"
  
      .prologue
+     .line 1319
      iget-boolean v0, p0, Landroid/widget/HorizontalScrollView;->mIsLayoutDirty:Z
  
      if-nez v0, :cond_0
  
+     .line 1320
      invoke-direct {p0, p2}, Landroid/widget/HorizontalScrollView;->scrollToChild(Landroid/view/View;)V
  
+     .line 1325
      :goto_0
      invoke-super {p0, p1, p2}, Landroid/widget/FrameLayout;->requestChildFocus(Landroid/view/View;Landroid/view/View;)V
  
+     .line 1326
      return-void
  
+     .line 1323
      :cond_0
      iput-object p2, p0, Landroid/widget/HorizontalScrollView;->mChildToScrollTo:Landroid/view/View;
  
***************
*** 3968,3973 ****
--- 4515,4521 ----
      .parameter "immediate"
  
      .prologue
+     .line 1368
      invoke-virtual {p1}, Landroid/view/View;->getLeft()I
  
      move-result v0
***************
*** 3990,3995 ****
--- 4538,4544 ----
  
      invoke-virtual {p2, v0, v1}, Landroid/graphics/Rect;->offset(II)V
  
+     .line 1371
      invoke-direct {p0, p2, p3}, Landroid/widget/HorizontalScrollView;->scrollToChildRect(Landroid/graphics/Rect;Z)Z
  
      move-result v0
***************
*** 4002,4014 ****
--- 4551,4567 ----
      .parameter "disallowIntercept"
  
      .prologue
+     .line 421
      if-eqz p1, :cond_0
  
+     .line 422
      invoke-direct {p0}, Landroid/widget/HorizontalScrollView;->recycleVelocityTracker()V
  
+     .line 424
      :cond_0
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->requestDisallowInterceptTouchEvent(Z)V
  
+     .line 425
      return-void
  .end method
  
***************
*** 4016,4027 ****
--- 4569,4583 ----
      .locals 1
  
      .prologue
+     .line 1376
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/HorizontalScrollView;->mIsLayoutDirty:Z
  
+     .line 1377
      invoke-super {p0}, Landroid/widget/FrameLayout;->requestLayout()V
  
+     .line 1378
      return-void
  .end method
  
***************
*** 4031,4048 ****
--- 4587,4607 ----
      .parameter "y"
  
      .prologue
+     .line 1465
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v1
  
      if-lez v1, :cond_1
  
+     .line 1466
      const/4 v1, 0x0
  
      invoke-virtual {p0, v1}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 1467
      .local v0, child:Landroid/view/View;
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
***************
*** 4064,4069 ****
--- 4623,4629 ----
  
      move-result p1
  
+     .line 1468
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getHeight()I
  
      move-result v1
***************
*** 4084,4089 ****
--- 4644,4650 ----
  
      move-result p2
  
+     .line 1469
      iget v1, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
      if-ne p1, v1, :cond_0
***************
*** 4092,4100 ****
--- 4653,4663 ----
  
      if-eq p2, v1, :cond_1
  
+     .line 1470
      :cond_0
      invoke-super {p0, p1, p2}, Landroid/widget/FrameLayout;->scrollTo(II)V
  
+     .line 1473
      .end local v0           #child:Landroid/view/View;
      :cond_1
      return-void
***************
*** 4105,4118 ****
--- 4668,4685 ----
      .parameter "fillViewport"
  
      .prologue
+     .line 280
      iget-boolean v0, p0, Landroid/widget/HorizontalScrollView;->mFillViewport:Z
  
      if-eq p1, v0, :cond_0
  
+     .line 281
      iput-boolean p1, p0, Landroid/widget/HorizontalScrollView;->mFillViewport:Z
  
+     .line 282
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->requestLayout()V
  
+     .line 284
      :cond_0
      return-void
  .end method
***************
*** 4124,4141 ****
--- 4691,4712 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1477
      const/4 v1, 0x2
  
      if-eq p1, v1, :cond_1
  
+     .line 1478
      iget-object v1, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
      if-nez v1, :cond_0
  
+     .line 1479
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getContext()Landroid/content/Context;
  
      move-result-object v0
  
+     .line 1480
      .local v0, context:Landroid/content/Context;
      new-instance v1, Landroid/widget/EdgeEffect;
  
***************
*** 4143,4164 ****
--- 4714,4740 ----
  
      iput-object v1, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
+     .line 1481
      new-instance v1, Landroid/widget/EdgeEffect;
  
      invoke-direct {v1, v0}, Landroid/widget/EdgeEffect;-><init>(Landroid/content/Context;)V
  
      iput-object v1, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
  
+     .line 1487
      .end local v0           #context:Landroid/content/Context;
      :cond_0
      :goto_0
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->setOverScrollMode(I)V
  
+     .line 1488
      return-void
  
+     .line 1484
      :cond_1
      iput-object v2, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowLeft:Landroid/widget/EdgeEffect;
  
+     .line 1485
      iput-object v2, p0, Landroid/widget/HorizontalScrollView;->mEdgeGlowRight:Landroid/widget/EdgeEffect;
  
      goto :goto_0
***************
*** 4169,4176 ****
--- 4745,4754 ----
      .parameter "smoothScrollingEnabled"
  
      .prologue
+     .line 298
      iput-boolean p1, p0, Landroid/widget/HorizontalScrollView;->mSmoothScrollingEnabled:Z
  
+     .line 299
      return-void
  .end method
  
***************
*** 4178,4183 ****
--- 4756,4762 ----
      .locals 1
  
      .prologue
+     .line 701
      const/4 v0, 0x1
  
      return v0
***************
*** 4191,4205 ****
--- 4770,4787 ----
      .prologue
      const/4 v10, 0x0
  
+     .line 1071
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getChildCount()I
  
      move-result v6
  
      if-nez v6, :cond_0
  
+     .line 1092
      :goto_0
      return-void
  
+     .line 1075
      :cond_0
      invoke-static {}, Landroid/view/animation/AnimationUtils;->currentAnimationTimeMillis()J
  
***************
*** 4209,4214 ****
--- 4791,4797 ----
  
      sub-long v0, v6, v8
  
+     .line 1076
      .local v0, duration:J
      const-wide/16 v6, 0xfa
  
***************
*** 4216,4221 ****
--- 4799,4805 ----
  
      if-lez v6, :cond_1
  
+     .line 1077
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->getWidth()I
  
      move-result v6
***************
*** 4228,4233 ****
--- 4812,4818 ----
  
      sub-int v5, v6, v7
  
+     .line 1078
      .local v5, width:I
      invoke-virtual {p0, v10}, Landroid/widget/HorizontalScrollView;->getChildAt(I)Landroid/view/View;
  
***************
*** 4237,4242 ****
--- 4822,4828 ----
  
      move-result v3
  
+     .line 1079
      .local v3, right:I
      sub-int v6, v3, v5
  
***************
*** 4244,4252 ****
--- 4830,4840 ----
  
      move-result v2
  
+     .line 1080
      .local v2, maxX:I
      iget v4, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
+     .line 1081
      .local v4, scrollX:I
      add-int v6, v4, p1
  
***************
*** 4260,4273 ****
--- 4848,4864 ----
  
      sub-int p1, v6, v4
  
+     .line 1083
      iget-object v6, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
      iget v7, p0, Landroid/widget/HorizontalScrollView;->mScrollY:I
  
      invoke-virtual {v6, v4, v7, p1, v10}, Landroid/widget/OverScroller;->startScroll(IIII)V
  
+     .line 1084
      invoke-virtual {p0}, Landroid/widget/HorizontalScrollView;->invalidate()V
  
+     .line 1091
      .end local v2           #maxX:I
      .end local v3           #right:I
      .end local v4           #scrollX:I
***************
*** 4281,4286 ****
--- 4872,4878 ----
  
      goto :goto_0
  
+     .line 1086
      :cond_1
      iget-object v6, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
***************
*** 4290,4299 ****
--- 4882,4893 ----
  
      if-nez v6, :cond_2
  
+     .line 1087
      iget-object v6, p0, Landroid/widget/HorizontalScrollView;->mScroller:Landroid/widget/OverScroller;
  
      invoke-virtual {v6}, Landroid/widget/OverScroller;->abortAnimation()V
  
+     .line 1089
      :cond_2
      invoke-virtual {p0, p1, p2}, Landroid/widget/HorizontalScrollView;->scrollBy(II)V
  
***************
*** 4306,4311 ****
--- 4900,4906 ----
      .parameter "y"
  
      .prologue
+     .line 1101
      iget v0, p0, Landroid/widget/HorizontalScrollView;->mScrollX:I
  
      sub-int v0, p1, v0
***************
*** 4316,4320 ****
--- 4911,4916 ----
  
      invoke-virtual {p0, v0, v1}, Landroid/widget/HorizontalScrollView;->smoothScrollBy(II)V
  
+     .line 1102
      return-void
  .end method
