*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ListView$FixedViewInfo.smali	2012-03-23 19:35:54.149574303 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ListView$FixedViewInfo.smali.original	2012-03-23 19:34:44.894311434 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 93
      iput-object p1, p0, Landroid/widget/ListView$FixedViewInfo;->this$0:Landroid/widget/ListView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
