*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ListPopupWindow$PopupDataSetObserver.smali	2012-03-23 19:36:00.673693269 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ListPopupWindow$PopupDataSetObserver.smali.original	2012-03-23 19:34:44.941312290 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 1214
      iput-object p1, p0, Landroid/widget/ListPopupWindow$PopupDataSetObserver;->this$0:Landroid/widget/ListPopupWindow;
  
      invoke-direct {p0}, Landroid/database/DataSetObserver;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 1214
      invoke-direct {p0, p1}, Landroid/widget/ListPopupWindow$PopupDataSetObserver;-><init>(Landroid/widget/ListPopupWindow;)V
  
      return-void
***************
*** 48,53 ****
--- 50,56 ----
      .locals 1
  
      .prologue
+     .line 1217
      iget-object v0, p0, Landroid/widget/ListPopupWindow$PopupDataSetObserver;->this$0:Landroid/widget/ListPopupWindow;
  
      invoke-virtual {v0}, Landroid/widget/ListPopupWindow;->isShowing()Z
***************
*** 56,65 ****
--- 59,70 ----
  
      if-eqz v0, :cond_0
  
+     .line 1219
      iget-object v0, p0, Landroid/widget/ListPopupWindow$PopupDataSetObserver;->this$0:Landroid/widget/ListPopupWindow;
  
      invoke-virtual {v0}, Landroid/widget/ListPopupWindow;->show()V
  
+     .line 1221
      :cond_0
      return-void
  .end method
***************
*** 68,76 ****
--- 73,83 ----
      .locals 1
  
      .prologue
+     .line 1225
      iget-object v0, p0, Landroid/widget/ListPopupWindow$PopupDataSetObserver;->this$0:Landroid/widget/ListPopupWindow;
  
      invoke-virtual {v0}, Landroid/widget/ListPopupWindow;->dismiss()V
  
+     .line 1226
      return-void
  .end method
