*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/MediaController$1.smali	2012-03-23 19:36:00.832696168 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/MediaController$1.smali.original	2012-03-23 19:34:44.942312309 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 175
      iput-object p1, p0, Landroid/widget/MediaController$1;->this$0:Landroid/widget/MediaController;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 49,59 ****
--- 50,62 ----
      .parameter "oldBottom"
  
      .prologue
+     .line 179
      iget-object v0, p0, Landroid/widget/MediaController$1;->this$0:Landroid/widget/MediaController;
  
      #calls: Landroid/widget/MediaController;->updateFloatingWindowLayout()V
      invoke-static {v0}, Landroid/widget/MediaController;->access$000(Landroid/widget/MediaController;)V
  
+     .line 180
      iget-object v0, p0, Landroid/widget/MediaController$1;->this$0:Landroid/widget/MediaController;
  
      #getter for: Landroid/widget/MediaController;->mShowing:Z
***************
*** 63,68 ****
--- 66,72 ----
  
      if-eqz v0, :cond_0
  
+     .line 181
      iget-object v0, p0, Landroid/widget/MediaController$1;->this$0:Landroid/widget/MediaController;
  
      #getter for: Landroid/widget/MediaController;->mWindowManager:Landroid/view/WindowManager;
***************
*** 86,91 ****
--- 90,96 ----
  
      invoke-interface {v0, v1, v2}, Landroid/view/WindowManager;->updateViewLayout(Landroid/view/View;Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 183
      :cond_0
      return-void
  .end method
