*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/NumberPicker$SetSelectionCommand.smali	2012-03-23 19:36:00.278686066 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/NumberPicker$SetSelectionCommand.smali.original	2012-03-23 19:34:44.938312236 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .parameter
  
      .prologue
+     .line 1941
      iput-object p1, p0, Landroid/widget/NumberPicker$SetSelectionCommand;->this$0:Landroid/widget/NumberPicker;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "x1"
  
      .prologue
+     .line 1941
      iput p1, p0, Landroid/widget/NumberPicker$SetSelectionCommand;->mSelectionStart:I
  
      return p1
***************
*** 55,60 ****
--- 57,63 ----
      .parameter "x1"
  
      .prologue
+     .line 1941
      iput p1, p0, Landroid/widget/NumberPicker$SetSelectionCommand;->mSelectionEnd:I
  
      return p1
***************
*** 66,71 ****
--- 69,75 ----
      .locals 3
  
      .prologue
+     .line 1947
      iget-object v0, p0, Landroid/widget/NumberPicker$SetSelectionCommand;->this$0:Landroid/widget/NumberPicker;
  
      #getter for: Landroid/widget/NumberPicker;->mInputText:Landroid/widget/EditText;
***************
*** 79,83 ****
--- 83,88 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/widget/EditText;->setSelection(II)V
  
+     .line 1948
      return-void
  .end method
