*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/NumberPicker$3.smali	2012-03-23 19:35:55.963607380 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/NumberPicker$3.smali.original	2012-03-23 19:34:44.909311708 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 599
      iput-object p1, p0, Landroid/widget/NumberPicker$3;->this$0:Landroid/widget/NumberPicker;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 43,48 ****
--- 44,50 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 601
      iget-object v0, p0, Landroid/widget/NumberPicker$3;->this$0:Landroid/widget/NumberPicker;
  
      #getter for: Landroid/widget/NumberPicker;->mInputText:Landroid/widget/EditText;
***************
*** 52,57 ****
--- 54,60 ----
  
      invoke-virtual {v0}, Landroid/widget/EditText;->clearFocus()V
  
+     .line 602
      invoke-virtual {p1}, Landroid/view/View;->getId()I
  
      move-result v0
***************
*** 60,73 ****
--- 63,79 ----
  
      if-ne v0, v1, :cond_0
  
+     .line 603
      iget-object v0, p0, Landroid/widget/NumberPicker$3;->this$0:Landroid/widget/NumberPicker;
  
      #calls: Landroid/widget/NumberPicker;->postChangeCurrentByOneFromLongPress(Z)V
      invoke-static {v0, v2}, Landroid/widget/NumberPicker;->access$200(Landroid/widget/NumberPicker;Z)V
  
+     .line 607
      :goto_0
      return v2
  
+     .line 605
      :cond_0
      iget-object v0, p0, Landroid/widget/NumberPicker$3;->this$0:Landroid/widget/NumberPicker;
  
