*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/NumberPicker$5.smali	2012-03-23 19:35:54.296576982 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/NumberPicker$5.smali.original	2012-03-23 19:34:44.895311452 -0500
***************
*** 26,35 ****
--- 26,37 ----
      .parameter
  
      .prologue
+     .line 673
      iput-object p1, p0, Landroid/widget/NumberPicker$5;->this$0:Landroid/widget/NumberPicker;
  
      invoke-direct {p0}, Landroid/animation/AnimatorListenerAdapter;-><init>()V
  
+     .line 674
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/NumberPicker$5;->mCanceled:Z
***************
*** 44,49 ****
--- 46,52 ----
      .parameter "animation"
  
      .prologue
+     .line 687
      iget-object v0, p0, Landroid/widget/NumberPicker$5;->this$0:Landroid/widget/NumberPicker;
  
      #getter for: Landroid/widget/NumberPicker;->mShowInputControlsAnimator:Landroid/animation/AnimatorSet;
***************
*** 57,66 ****
--- 60,71 ----
  
      if-eqz v0, :cond_0
  
+     .line 688
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/NumberPicker$5;->mCanceled:Z
  
+     .line 690
      :cond_0
      return-void
  .end method
***************
*** 70,79 ****
--- 75,86 ----
      .parameter "animation"
  
      .prologue
+     .line 678
      iget-boolean v0, p0, Landroid/widget/NumberPicker$5;->mCanceled:Z
  
      if-nez v0, :cond_0
  
+     .line 680
      iget-object v0, p0, Landroid/widget/NumberPicker$5;->this$0:Landroid/widget/NumberPicker;
  
      const/4 v1, 0x1
***************
*** 81,90 ****
--- 88,99 ----
      #calls: Landroid/widget/NumberPicker;->setSelectorWheelState(I)V
      invoke-static {v0, v1}, Landroid/widget/NumberPicker;->access$400(Landroid/widget/NumberPicker;I)V
  
+     .line 682
      :cond_0
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/NumberPicker$5;->mCanceled:Z
  
+     .line 683
      return-void
  .end method
