*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/OverScroller.smali	2012-03-23 19:35:55.316595582 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/OverScroller.smali.original	2012-03-23 19:34:44.904311616 -0500
***************
*** 37,46 ****
--- 37,48 ----
      .parameter "context"
  
      .prologue
+     .line 51
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/widget/OverScroller;-><init>(Landroid/content/Context;Landroid/view/animation/Interpolator;)V
  
+     .line 52
      return-void
  .end method
  
***************
*** 50,59 ****
--- 52,63 ----
      .parameter "interpolator"
  
      .prologue
+     .line 61
      const/4 v0, 0x1
  
      invoke-direct {p0, p1, p2, v0}, Landroid/widget/OverScroller;-><init>(Landroid/content/Context;Landroid/view/animation/Interpolator;Z)V
  
+     .line 62
      return-void
  .end method
  
***************
*** 65,74 ****
--- 69,80 ----
      .parameter "bounceCoefficientY"
  
      .prologue
+     .line 95
      const/4 v0, 0x1
  
      invoke-direct {p0, p1, p2, v0}, Landroid/widget/OverScroller;-><init>(Landroid/content/Context;Landroid/view/animation/Interpolator;Z)V
  
+     .line 96
      return-void
  .end method
  
***************
*** 81,88 ****
--- 87,96 ----
      .parameter "flywheel"
  
      .prologue
+     .line 113
      invoke-direct {p0, p1, p2, p5}, Landroid/widget/OverScroller;-><init>(Landroid/content/Context;Landroid/view/animation/Interpolator;Z)V
  
+     .line 114
      return-void
  .end method
  
***************
*** 93,118 ****
--- 101,133 ----
      .parameter "flywheel"
  
      .prologue
+     .line 72
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 73
      iput-object p2, p0, Landroid/widget/OverScroller;->mInterpolator:Landroid/view/animation/Interpolator;
  
+     .line 74
      iput-boolean p3, p0, Landroid/widget/OverScroller;->mFlywheel:Z
  
+     .line 75
      new-instance v0, Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-direct {v0}, Landroid/widget/OverScroller$SplineOverScroller;-><init>()V
  
      iput-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
+     .line 76
      new-instance v0, Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-direct {v0}, Landroid/widget/OverScroller$SplineOverScroller;-><init>()V
  
      iput-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
+     .line 78
      invoke-static {p1}, Landroid/widget/OverScroller$SplineOverScroller;->initFromContext(Landroid/content/Context;)V
  
+     .line 79
      return-void
  .end method
  
***************
*** 122,135 ****
--- 137,153 ----
      .locals 1
  
      .prologue
+     .line 511
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0}, Landroid/widget/OverScroller$SplineOverScroller;->finish()V
  
+     .line 512
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0}, Landroid/widget/OverScroller$SplineOverScroller;->finish()V
  
+     .line 513
      return-void
  .end method
  
***************
*** 137,169 ****
--- 155,194 ----
      .locals 8
  
      .prologue
+     .line 290
      invoke-virtual {p0}, Landroid/widget/OverScroller;->isFinished()Z
  
      move-result v6
  
      if-eqz v6, :cond_0
  
+     .line 291
      const/4 v6, 0x0
  
+     .line 338
      :goto_0
      return v6
  
+     .line 294
      :cond_0
      iget v6, p0, Landroid/widget/OverScroller;->mMode:I
  
      packed-switch v6, :pswitch_data_0
  
+     .line 338
      :cond_1
      :goto_1
      const/4 v6, 0x1
  
      goto :goto_0
  
+     .line 296
      :pswitch_0
      invoke-static {}, Landroid/view/animation/AnimationUtils;->currentAnimationTimeMillis()J
  
      move-result-wide v4
  
+     .line 299
      .local v4, time:J
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
***************
*** 174,179 ****
--- 199,205 ----
  
      sub-long v1, v4, v6
  
+     .line 301
      .local v1, elapsedTime:J
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
***************
*** 182,187 ****
--- 208,214 ----
  
      move-result v0
  
+     .line 302
      .local v0, duration:I
      int-to-long v6, v0
  
***************
*** 189,220 ****
--- 216,253 ----
  
      if-gez v6, :cond_3
  
+     .line 303
      long-to-float v6, v1
  
      int-to-float v7, v0
  
      div-float v3, v6, v7
  
+     .line 305
      .local v3, q:F
      iget-object v6, p0, Landroid/widget/OverScroller;->mInterpolator:Landroid/view/animation/Interpolator;
  
      if-nez v6, :cond_2
  
+     .line 306
      invoke-static {v3}, Landroid/widget/Scroller;->viscousFluid(F)F
  
      move-result v3
  
+     .line 311
      :goto_2
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v6, v3}, Landroid/widget/OverScroller$SplineOverScroller;->updateScroll(F)V
  
+     .line 312
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v6, v3}, Landroid/widget/OverScroller$SplineOverScroller;->updateScroll(F)V
  
      goto :goto_1
  
+     .line 308
      :cond_2
      iget-object v6, p0, Landroid/widget/OverScroller;->mInterpolator:Landroid/view/animation/Interpolator;
  
***************
*** 224,235 ****
--- 257,270 ----
  
      goto :goto_2
  
+     .line 314
      .end local v3           #q:F
      :cond_3
      invoke-virtual {p0}, Landroid/widget/OverScroller;->abortAnimation()V
  
      goto :goto_1
  
+     .line 319
      .end local v0           #duration:I
      .end local v1           #elapsedTime:J
      .end local v4           #time:J
***************
*** 243,248 ****
--- 278,284 ----
  
      if-nez v6, :cond_4
  
+     .line 320
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->update()Z
***************
*** 251,256 ****
--- 287,293 ----
  
      if-nez v6, :cond_4
  
+     .line 321
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->continueWhenFinished()Z
***************
*** 259,268 ****
--- 296,307 ----
  
      if-nez v6, :cond_4
  
+     .line 322
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->finish()V
  
+     .line 327
      :cond_4
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
***************
*** 273,278 ****
--- 312,318 ----
  
      if-nez v6, :cond_1
  
+     .line 328
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->update()Z
***************
*** 281,286 ****
--- 321,327 ----
  
      if-nez v6, :cond_1
  
+     .line 329
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->continueWhenFinished()Z
***************
*** 289,300 ****
--- 330,343 ----
  
      if-nez v6, :cond_1
  
+     .line 330
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->finish()V
  
      goto :goto_1
  
+     .line 294
      nop
  
      :pswitch_data_0
***************
*** 311,324 ****
--- 354,370 ----
      .end annotation
  
      .prologue
+     .line 245
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->extendDuration(I)V
  
+     .line 246
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->extendDuration(I)V
  
+     .line 247
      return-void
  .end method
  
***************
*** 334,339 ****
--- 380,386 ----
      .parameter "maxY"
  
      .prologue
+     .line 401
      const/4 v9, 0x0
  
      const/4 v10, 0x0
***************
*** 358,363 ****
--- 405,411 ----
  
      invoke-virtual/range {v0 .. v10}, Landroid/widget/OverScroller;->fling(IIIIIIIIII)V
  
+     .line 402
      return-void
  .end method
  
***************
*** 375,380 ****
--- 423,429 ----
      .parameter "overY"
  
      .prologue
+     .line 434
      iget-boolean v0, p0, Landroid/widget/OverScroller;->mFlywheel:Z
  
      if-eqz v0, :cond_0
***************
*** 385,390 ****
--- 434,440 ----
  
      if-nez v0, :cond_0
  
+     .line 435
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mCurrVelocity:F
***************
*** 392,397 ****
--- 442,448 ----
  
      move-result v6
  
+     .line 436
      .local v6, oldVelocityX:F
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
***************
*** 400,405 ****
--- 451,457 ----
  
      move-result v7
  
+     .line 437
      .local v7, oldVelocityY:F
      int-to-float v0, p3
  
***************
*** 429,446 ****
--- 481,501 ----
  
      if-nez v0, :cond_0
  
+     .line 439
      int-to-float v0, p3
  
      add-float/2addr v0, v6
  
      float-to-int p3, v0
  
+     .line 440
      int-to-float v0, p4
  
      add-float/2addr v0, v7
  
      float-to-int p4, v0
  
+     .line 444
      .end local v6           #oldVelocityX:F
      .end local v7           #oldVelocityY:F
      :cond_0
***************
*** 448,453 ****
--- 503,509 ----
  
      iput v0, p0, Landroid/widget/OverScroller;->mMode:I
  
+     .line 445
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      move v1, p1
***************
*** 462,467 ****
--- 518,524 ----
  
      invoke-virtual/range {v0 .. v5}, Landroid/widget/OverScroller$SplineOverScroller;->fling(IIIII)V
  
+     .line 446
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      move v1, p2
***************
*** 476,481 ****
--- 533,539 ----
  
      invoke-virtual/range {v0 .. v5}, Landroid/widget/OverScroller$SplineOverScroller;->fling(IIIII)V
  
+     .line 447
      return-void
  .end method
  
***************
*** 484,489 ****
--- 542,548 ----
      .parameter "finished"
  
      .prologue
+     .line 147
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      iget-object v1, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
***************
*** 496,501 ****
--- 555,561 ----
      #setter for: Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
      invoke-static {v0, v1}, Landroid/widget/OverScroller$SplineOverScroller;->access$002(Landroid/widget/OverScroller$SplineOverScroller;Z)Z
  
+     .line 148
      return-void
  .end method
  
***************
*** 503,508 ****
--- 563,569 ----
      .locals 3
  
      .prologue
+     .line 174
      iget-object v1, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mCurrVelocity:F
***************
*** 519,524 ****
--- 580,586 ----
  
      mul-float v0, v1, v2
  
+     .line 175
      .local v0, squaredNorm:F
      iget-object v1, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
***************
*** 538,543 ****
--- 600,606 ----
  
      add-float/2addr v0, v1
  
+     .line 176
      invoke-static {v0}, Landroid/util/FloatMath;->sqrt(F)F
  
      move-result v1
***************
*** 549,554 ****
--- 612,618 ----
      .locals 1
  
      .prologue
+     .line 156
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mCurrentPosition:I
***************
*** 563,568 ****
--- 627,633 ----
      .locals 1
  
      .prologue
+     .line 165
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mCurrentPosition:I
***************
*** 579,584 ****
--- 644,650 ----
      .end annotation
  
      .prologue
+     .line 226
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
***************
*** 604,609 ****
--- 670,676 ----
      .locals 1
  
      .prologue
+     .line 203
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
***************
*** 618,623 ****
--- 685,691 ----
      .locals 1
  
      .prologue
+     .line 212
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
***************
*** 632,637 ****
--- 700,706 ----
      .locals 1
  
      .prologue
+     .line 185
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
***************
*** 646,651 ****
--- 715,721 ----
      .locals 1
  
      .prologue
+     .line 194
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
***************
*** 660,665 ****
--- 730,736 ----
      .locals 1
  
      .prologue
+     .line 135
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
***************
*** 693,698 ****
--- 764,770 ----
      .locals 1
  
      .prologue
+     .line 497
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
***************
*** 748,753 ****
--- 820,826 ----
      .parameter "yvel"
  
      .prologue
+     .line 532
      iget-object v2, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
***************
*** 764,769 ****
--- 837,843 ----
  
      sub-int v0, v2, v3
  
+     .line 533
      .local v0, dx:I
      iget-object v2, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
***************
*** 781,786 ****
--- 855,861 ----
  
      sub-int v1, v2, v3
  
+     .line 534
      .local v1, dy:I
      invoke-virtual {p0}, Landroid/widget/OverScroller;->isFinished()Z
  
***************
*** 834,843 ****
--- 909,920 ----
      .parameter "overX"
  
      .prologue
+     .line 463
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p1, p2, p3}, Landroid/widget/OverScroller$SplineOverScroller;->notifyEdgeReached(III)V
  
+     .line 464
      return-void
  .end method
  
***************
*** 848,857 ****
--- 925,936 ----
      .parameter "overY"
  
      .prologue
+     .line 480
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p1, p2, p3}, Landroid/widget/OverScroller$SplineOverScroller;->notifyEdgeReached(III)V
  
+     .line 481
      return-void
  .end method
  
***************
*** 862,871 ****
--- 941,952 ----
      .end annotation
  
      .prologue
+     .line 264
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->setFinalPosition(I)V
  
+     .line 265
      return-void
  .end method
  
***************
*** 876,885 ****
--- 957,968 ----
      .end annotation
  
      .prologue
+     .line 282
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->setFinalPosition(I)V
  
+     .line 283
      return-void
  .end method
  
***************
*** 888,901 ****
--- 971,987 ----
      .parameter "friction"
  
      .prologue
+     .line 124
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->setFriction(F)V
  
+     .line 125
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->setFriction(F)V
  
+     .line 126
      return-void
  .end method
  
***************
*** 911,924 ****
--- 997,1013 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 391
      iput v2, p0, Landroid/widget/OverScroller;->mMode:I
  
+     .line 394
      iget-object v3, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v3, p1, p3, p4}, Landroid/widget/OverScroller$SplineOverScroller;->springback(III)Z
  
      move-result v0
  
+     .line 395
      .local v0, spingbackX:Z
      iget-object v3, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
***************
*** 926,931 ****
--- 1015,1021 ----
  
      move-result v1
  
+     .line 396
      .local v1, spingbackY:Z
      if-nez v0, :cond_0
  
***************
*** 949,954 ****
--- 1039,1045 ----
      .parameter "dy"
  
      .prologue
+     .line 356
      const/16 v5, 0xfa
  
      move-object v0, p0
***************
*** 963,968 ****
--- 1054,1060 ----
  
      invoke-virtual/range {v0 .. v5}, Landroid/widget/OverScroller;->startScroll(IIIII)V
  
+     .line 357
      return-void
  .end method
  
***************
*** 975,992 ****
--- 1067,1088 ----
      .parameter "duration"
  
      .prologue
+     .line 373
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/widget/OverScroller;->mMode:I
  
+     .line 374
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p1, p3, p5}, Landroid/widget/OverScroller$SplineOverScroller;->startScroll(III)V
  
+     .line 375
      iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
      invoke-virtual {v0, p2, p4, p5}, Landroid/widget/OverScroller$SplineOverScroller;->startScroll(III)V
  
+     .line 376
      return-void
  .end method
  
***************
*** 994,1003 ****
--- 1090,1101 ----
      .locals 8
  
      .prologue
+     .line 523
      invoke-static {}, Landroid/view/animation/AnimationUtils;->currentAnimationTimeMillis()J
  
      move-result-wide v2
  
+     .line 524
      .local v2, time:J
      iget-object v4, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
***************
*** 1017,1022 ****
--- 1115,1121 ----
  
      move-result-wide v0
  
+     .line 525
      .local v0, startTime:J
      sub-long v4, v2, v0
  
