*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/OverScroller$SplineOverScroller.smali	2012-03-23 19:35:56.344614329 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/OverScroller$SplineOverScroller.smali.original	2012-03-23 19:34:44.911311744 -0500
***************
*** 79,84 ****
--- 79,85 ----
      .locals 16
  
      .prologue
+     .line 587
      const-wide v11, 0x3fe8f5c28f5c28f6L
  
      invoke-static {v11, v12}, Ljava/lang/Math;->log(D)D
***************
*** 97,119 ****
--- 98,125 ----
  
      sput v11, Landroid/widget/OverScroller$SplineOverScroller;->DECELERATION_RATE:F
  
+     .line 595
      const/16 v11, 0x65
  
      new-array v11, v11, [F
  
      sput-object v11, Landroid/widget/OverScroller$SplineOverScroller;->SPLINE_POSITION:[F
  
+     .line 596
      const/16 v11, 0x65
  
      new-array v11, v11, [F
  
      sput-object v11, Landroid/widget/OverScroller$SplineOverScroller;->SPLINE_TIME:[F
  
+     .line 603
      const/4 v7, 0x0
  
+     .line 604
      .local v7, x_min:F
      const/4 v10, 0x0
  
+     .line 605
      .local v10, y_min:F
      const/4 v3, 0x0
  
***************
*** 123,137 ****
--- 129,146 ----
  
      if-ge v3, v11, :cond_4
  
+     .line 606
      int-to-float v11, v3
  
      const/high16 v12, 0x42c8
  
      div-float v0, v11, v12
  
+     .line 608
      .local v0, alpha:F
      const/high16 v6, 0x3f80
  
+     .line 611
      .local v6, x_max:F
      :goto_1
      sub-float v11, v6, v7
***************
*** 142,147 ****
--- 151,157 ----
  
      add-float v5, v7, v11
  
+     .line 612
      .local v5, x:F
      const/high16 v11, 0x4040
  
***************
*** 153,158 ****
--- 163,169 ----
  
      mul-float v1, v11, v12
  
+     .line 613
      .local v1, coef:F
      const/high16 v11, 0x3f80
  
***************
*** 176,181 ****
--- 187,193 ----
  
      add-float v4, v11, v12
  
+     .line 614
      .local v4, tx:F
      sub-float v11, v4, v0
  
***************
*** 191,196 ****
--- 203,209 ----
  
      if-gez v11, :cond_0
  
+     .line 618
      sget-object v11, Landroid/widget/OverScroller$SplineOverScroller;->SPLINE_POSITION:[F
  
      const/high16 v12, 0x3f80
***************
*** 213,220 ****
--- 226,235 ----
  
      aput v12, v11, v3
  
+     .line 620
      const/high16 v9, 0x3f80
  
+     .line 623
      .local v9, y_max:F
      :goto_2
      sub-float v11, v9, v10
***************
*** 225,230 ****
--- 240,246 ----
  
      add-float v8, v10, v11
  
+     .line 624
      .local v8, y:F
      const/high16 v11, 0x4040
  
***************
*** 236,241 ****
--- 252,258 ----
  
      mul-float v1, v11, v12
  
+     .line 625
      const/high16 v11, 0x3f80
  
      sub-float/2addr v11, v8
***************
*** 254,259 ****
--- 271,277 ----
  
      add-float v2, v11, v12
  
+     .line 626
      .local v2, dy:F
      sub-float v11, v2, v0
  
***************
*** 269,274 ****
--- 287,293 ----
  
      if-gez v11, :cond_2
  
+     .line 630
      sget-object v11, Landroid/widget/OverScroller$SplineOverScroller;->SPLINE_TIME:[F
  
      const/high16 v12, 0x3f80
***************
*** 295,304 ****
--- 314,325 ----
  
      aput v12, v11, v3
  
+     .line 605
      add-int/lit8 v3, v3, 0x1
  
      goto/16 :goto_0
  
+     .line 615
      .end local v2           #dy:F
      .end local v8           #y:F
      .end local v9           #y_max:F
***************
*** 311,321 ****
--- 332,344 ----
  
      goto/16 :goto_1
  
+     .line 616
      :cond_1
      move v7, v5
  
      goto/16 :goto_1
  
+     .line 627
      .restart local v2       #dy:F
      .restart local v8       #y:F
      .restart local v9       #y_max:F
***************
*** 328,338 ****
--- 351,363 ----
  
      goto :goto_2
  
+     .line 628
      :cond_3
      move v10, v8
  
      goto :goto_2
  
+     .line 632
      .end local v0           #alpha:F
      .end local v1           #coef:F
      .end local v2           #dy:F
***************
*** 356,361 ****
--- 381,387 ----
  
      aput v15, v11, v12
  
+     .line 633
      return-void
  .end method
  
***************
*** 363,384 ****
--- 389,415 ----
      .locals 1
  
      .prologue
+     .line 647
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 576
      invoke-static {}, Landroid/view/ViewConfiguration;->getScrollFriction()F
  
      move-result v0
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFlingFriction:F
  
+     .line 579
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mState:I
  
+     .line 648
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
+     .line 649
      return-void
  .end method
  
***************
*** 387,392 ****
--- 418,424 ----
      .parameter "x0"
  
      .prologue
+     .line 538
      iget-boolean v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
      return v0
***************
*** 398,403 ****
--- 430,436 ----
      .parameter "x1"
  
      .prologue
+     .line 538
      iput-boolean p1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
      return p1
***************
*** 408,413 ****
--- 441,447 ----
      .parameter "x0"
  
      .prologue
+     .line 538
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mCurrentPosition:I
  
      return v0
***************
*** 418,423 ****
--- 452,458 ----
      .parameter "x0"
  
      .prologue
+     .line 538
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mCurrVelocity:F
  
      return v0
***************
*** 428,433 ****
--- 463,469 ----
      .parameter "x0"
  
      .prologue
+     .line 538
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
      return v0
***************
*** 438,443 ****
--- 474,480 ----
      .parameter "x0"
  
      .prologue
+     .line 538
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      return v0
***************
*** 448,453 ****
--- 485,491 ----
      .parameter "x0"
  
      .prologue
+     .line 538
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
      return v0
***************
*** 458,463 ****
--- 496,502 ----
      .parameter "x0"
  
      .prologue
+     .line 538
      iget-wide v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
      return-wide v0
***************
*** 468,473 ****
--- 507,513 ----
      .parameter "x0"
  
      .prologue
+     .line 538
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mState:I
  
      return v0
***************
*** 482,492 ****
--- 522,535 ----
      .prologue
      const/high16 v11, 0x42c8
  
+     .line 667
      sub-int v2, p2, p1
  
+     .line 668
      .local v2, oldDistance:I
      sub-int v1, p3, p1
  
+     .line 669
      .local v1, newDistance:I
      int-to-float v9, v1
  
***************
*** 498,517 ****
--- 541,564 ----
  
      move-result v6
  
+     .line 670
      .local v6, x:F
      mul-float v9, v11, v6
  
      float-to-int v0, v9
  
+     .line 671
      .local v0, index:I
      const/16 v9, 0x64
  
      if-ge v0, v9, :cond_0
  
+     .line 672
      int-to-float v9, v0
  
      div-float v7, v9, v11
  
+     .line 673
      .local v7, x_inf:F
      add-int/lit8 v9, v0, 0x1
  
***************
*** 519,529 ****
--- 566,578 ----
  
      div-float v8, v9, v11
  
+     .line 674
      .local v8, x_sup:F
      sget-object v9, Landroid/widget/OverScroller$SplineOverScroller;->SPLINE_TIME:[F
  
      aget v3, v9, v0
  
+     .line 675
      .local v3, t_inf:F
      sget-object v9, Landroid/widget/OverScroller$SplineOverScroller;->SPLINE_TIME:[F
  
***************
*** 531,536 ****
--- 580,586 ----
  
      aget v4, v9, v10
  
+     .line 676
      .local v4, t_sup:F
      sub-float v9, v6, v7
  
***************
*** 544,549 ****
--- 594,600 ----
  
      add-float v5, v3, v9
  
+     .line 677
      .local v5, timeCoef:F
      iget v9, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
***************
*** 555,560 ****
--- 606,612 ----
  
      iput v9, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
+     .line 679
      .end local v3           #t_inf:F
      .end local v4           #t_sup:F
      .end local v5           #timeCoef:F
***************
*** 571,576 ****
--- 623,629 ----
      .parameter "velocity"
  
      .prologue
+     .line 802
      neg-int v4, p3
  
      int-to-float v4, v4
***************
*** 579,584 ****
--- 632,638 ----
  
      div-float v2, v4, v5
  
+     .line 803
      .local v2, durationToApex:F
      mul-int v4, p3, p3
  
***************
*** 596,601 ****
--- 650,656 ----
  
      div-float v0, v4, v5
  
+     .line 804
      .local v0, distanceToApex:F
      sub-int v4, p2, p1
  
***************
*** 605,610 ****
--- 660,666 ----
  
      int-to-float v1, v4
  
+     .line 805
      .local v1, distanceToEdge:F
      const-wide/high16 v4, 0x4000
  
***************
*** 630,635 ****
--- 686,692 ----
  
      double-to-float v3, v4
  
+     .line 807
      .local v3, totalDuration:F
      iget-wide v4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
***************
*** 647,654 ****
--- 704,713 ----
  
      iput-wide v4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
+     .line 808
      iput p2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
+     .line 809
      iget v4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDeceleration:F
  
      neg-float v4, v4
***************
*** 659,664 ****
--- 718,724 ----
  
      iput v4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
  
+     .line 810
      return-void
  .end method
  
***************
*** 667,672 ****
--- 727,733 ----
      .parameter "velocity"
  
      .prologue
+     .line 659
      if-lez p0, :cond_0
  
      const/high16 v0, -0x3b06
***************
*** 685,690 ****
--- 746,752 ----
      .parameter "velocity"
  
      .prologue
+     .line 784
      const v0, 0x3eb33333
  
      invoke-static {p1}, Ljava/lang/Math;->abs(I)I
***************
*** 717,726 ****
--- 779,790 ----
      .parameter "velocity"
  
      .prologue
+     .line 788
      invoke-direct {p0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->getSplineDeceleration(I)D
  
      move-result-wide v2
  
+     .line 789
      .local v2, l:D
      sget v4, Landroid/widget/OverScroller$SplineOverScroller;->DECELERATION_RATE:F
  
***************
*** 730,735 ****
--- 794,800 ----
  
      sub-double v0, v4, v6
  
+     .line 790
      .local v0, decelMinusOne:D
      iget v4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFlingFriction:F
  
***************
*** 761,770 ****
--- 826,837 ----
      .parameter "velocity"
  
      .prologue
+     .line 795
      invoke-direct {p0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->getSplineDeceleration(I)D
  
      move-result-wide v2
  
+     .line 796
      .local v2, l:D
      sget v4, Landroid/widget/OverScroller$SplineOverScroller;->DECELERATION_RATE:F
  
***************
*** 774,779 ****
--- 841,847 ----
  
      sub-double v0, v4, v6
  
+     .line 797
      .local v0, decelMinusOne:D
      const-wide v4, 0x408f400000000000L
  
***************
*** 795,800 ****
--- 863,869 ----
      .parameter "context"
  
      .prologue
+     .line 636
      invoke-virtual {p0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v1
***************
*** 809,814 ****
--- 878,884 ----
  
      mul-float v0, v1, v2
  
+     .line 637
      .local v0, ppi:F
      const v1, 0x43c10b3d
  
***************
*** 820,825 ****
--- 890,896 ----
  
      sput v1, Landroid/widget/OverScroller$SplineOverScroller;->PHYSICAL_COEF:F
  
+     .line 641
      return-void
  .end method
  
***************
*** 829,834 ****
--- 900,906 ----
      .prologue
      const/high16 v4, 0x4000
  
+     .line 854
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
  
      iget v3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
***************
*** 847,852 ****
--- 919,925 ----
  
      div-float v0, v2, v3
  
+     .line 855
      .local v0, distance:F
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
  
***************
*** 856,861 ****
--- 929,935 ----
  
      move-result v1
  
+     .line 857
      .local v1, sign:F
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mOver:I
  
***************
*** 865,870 ****
--- 939,945 ----
  
      if-lez v2, :cond_0
  
+     .line 859
      neg-float v2, v1
  
      iget v3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
***************
*** 889,907 ****
--- 964,986 ----
  
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDeceleration:F
  
+     .line 860
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mOver:I
  
      int-to-float v0, v2
  
+     .line 863
      :cond_0
      float-to-int v2, v0
  
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mOver:I
  
+     .line 864
      const/4 v2, 0x2
  
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mState:I
  
+     .line 865
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
      iget v3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
***************
*** 916,921 ****
--- 995,1001 ----
  
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
+     .line 866
      const/high16 v2, 0x447a
  
      iget v3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
***************
*** 934,941 ****
--- 1014,1023 ----
  
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
+     .line 867
      return-void
  
+     .line 865
      .restart local v0       #distance:F
      :cond_1
      neg-float v0, v0
***************
*** 951,975 ****
      .parameter "velocity"
  
      .prologue
      if-le p1, p2, :cond_0
  
      move/from16 v0, p3
  
      if-ge p1, v0, :cond_0
  
      const-string v1, "OverScroller"
  
!     const-string v2, "startAfterEdge called from a valid position"
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
      :goto_0
      return-void
  
      :cond_0
      move/from16 v0, p3
  
--- 1033,1062 ----
      .parameter "velocity"
  
      .prologue
+     .line 819
      if-le p1, p2, :cond_0
  
      move/from16 v0, p3
  
      if-ge p1, v0, :cond_0
  
+     .line 820
      const-string v1, "OverScroller"
  
!     const-string/jumbo v2, "startAfterEdge called from a valid position"
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 821
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
+     .line 839
      :goto_0
      return-void
  
+     .line 824
      :cond_0
      move/from16 v0, p3
  
***************
*** 977,992 ****
--- 1064,1082 ----
  
      const/4 v10, 0x1
  
+     .line 825
      .local v10, positive:Z
      :goto_1
      if-eqz v10, :cond_2
  
      move/from16 v7, p3
  
+     .line 826
      .local v7, edge:I
      :goto_2
      sub-int v9, p1, v7
  
+     .line 827
      .local v9, overDistance:I
      mul-int v1, v9, p4
  
***************
*** 994,1009 ****
--- 1084,1102 ----
  
      const/4 v8, 0x1
  
+     .line 828
      .local v8, keepIncreasing:Z
      :goto_3
      if-eqz v8, :cond_4
  
+     .line 830
      move/from16 v0, p4
  
      invoke-direct {p0, p1, v7, v0}, Landroid/widget/OverScroller$SplineOverScroller;->startBounceAfterEdge(III)V
  
      goto :goto_0
  
+     .line 824
      .end local v7           #edge:I
      .end local v8           #keepIncreasing:Z
      .end local v9           #overDistance:I
***************
*** 1017,1024 ****
--- 1110,1119 ----
      :cond_2
      move v7, p2
  
+     .line 825
      goto :goto_2
  
+     .line 827
      .restart local v7       #edge:I
      .restart local v9       #overDistance:I
      :cond_3
***************
*** 1026,1031 ****
--- 1121,1127 ----
  
      goto :goto_3
  
+     .line 832
      .restart local v8       #keepIncreasing:Z
      :cond_4
      move/from16 v0, p4
***************
*** 1034,1039 ****
--- 1130,1136 ----
  
      move-result-wide v11
  
+     .line 833
      .local v11, totalDistance:D
      invoke-static {v9}, Ljava/lang/Math;->abs(I)I
  
***************
*** 1045,1050 ****
--- 1142,1148 ----
  
      if-lez v1, :cond_7
  
+     .line 834
      if-eqz v10, :cond_5
  
      move v4, p2
***************
*** 1077,1082 ****
--- 1175,1181 ----
  
      goto :goto_5
  
+     .line 836
      :cond_7
      move/from16 v0, p4
  
***************
*** 1092,1097 ****
--- 1191,1197 ----
      .parameter "velocity"
  
      .prologue
+     .line 813
      if-nez p3, :cond_0
  
      sub-int v0, p1, p2
***************
*** 1103,1117 ****
--- 1203,1221 ----
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDeceleration:F
  
+     .line 814
      invoke-direct {p0, p1, p2, p3}, Landroid/widget/OverScroller$SplineOverScroller;->fitOnBounceCurve(III)V
  
+     .line 815
      invoke-direct {p0}, Landroid/widget/OverScroller$SplineOverScroller;->onEdgeReached()V
  
+     .line 816
      return-void
  
      :cond_0
      move v0, p3
  
+     .line 813
      goto :goto_0
  .end method
  
***************
*** 1122,1141 ****
--- 1226,1251 ----
      .parameter "velocity"
  
      .prologue
+     .line 735
      const/4 v1, 0x0
  
      iput-boolean v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
+     .line 736
      const/4 v1, 0x1
  
      iput v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mState:I
  
+     .line 737
      iput p1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
+     .line 738
      iput p2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
+     .line 739
      sub-int v0, p1, p2
  
+     .line 740
      .local v0, delta:I
      invoke-static {v0}, Landroid/widget/OverScroller$SplineOverScroller;->getDeceleration(I)F
  
***************
*** 1143,1158 ****
--- 1253,1271 ----
  
      iput v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDeceleration:F
  
+     .line 742
      neg-int v1, v0
  
      iput v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
  
+     .line 743
      invoke-static {v0}, Ljava/lang/Math;->abs(I)I
  
      move-result v1
  
      iput v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mOver:I
  
+     .line 744
      const-wide v1, 0x408f400000000000L
  
      const-wide/high16 v3, -0x4000
***************
*** 1177,1182 ****
--- 1290,1296 ----
  
      iput v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
+     .line 745
      return-void
  .end method
  
***************
*** 1188,1206 ****
--- 1302,1324 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 870
      iget v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mState:I
  
      packed-switch v1, :pswitch_data_0
  
+     .line 894
      :goto_0
      invoke-virtual {p0}, Landroid/widget/OverScroller$SplineOverScroller;->update()Z
  
+     .line 895
      const/4 v0, 0x1
  
      :cond_0
      :pswitch_0
      return v0
  
+     .line 873
      :pswitch_1
      iget v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
***************
*** 1208,1223 ****
--- 1326,1344 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 875
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
+     .line 877
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mCurrVelocity:F
  
      float-to-int v0, v0
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
  
+     .line 878
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
  
      invoke-static {v0}, Landroid/widget/OverScroller$SplineOverScroller;->getDeceleration(I)F
***************
*** 1226,1231 ****
--- 1347,1353 ----
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDeceleration:F
  
+     .line 879
      iget-wide v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
***************
*** 1236,1245 ****
--- 1358,1369 ----
  
      iput-wide v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
+     .line 880
      invoke-direct {p0}, Landroid/widget/OverScroller$SplineOverScroller;->onEdgeReached()V
  
      goto :goto_0
  
+     .line 887
      :pswitch_2
      iget-wide v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
***************
*** 1251,1256 ****
--- 1375,1381 ----
  
      iput-wide v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
+     .line 888
      iget v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
***************
*** 1259,1264 ****
--- 1384,1390 ----
  
      goto :goto_0
  
+     .line 870
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_1
***************
*** 1272,1281 ****
--- 1398,1409 ----
      .parameter "extend"
  
      .prologue
+     .line 709
      invoke-static {}, Landroid/view/animation/AnimationUtils;->currentAnimationTimeMillis()J
  
      move-result-wide v1
  
+     .line 710
      .local v1, time:J
      iget-wide v3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
***************
*** 1283,1297 ****
--- 1411,1428 ----
  
      long-to-int v0, v3
  
+     .line 711
      .local v0, elapsedTime:I
      add-int v3, v0, p1
  
      iput v3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
+     .line 712
      const/4 v3, 0x0
  
      iput-boolean v3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
+     .line 713
      return-void
  .end method
  
***************
*** 1299,1312 ****
--- 1430,1446 ----
      .locals 1
  
      .prologue
+     .line 696
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mCurrentPosition:I
  
+     .line 700
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
+     .line 701
      return-void
  .end method
  
***************
*** 1321,1369 ****
--- 1455,1516 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 748
      iput p5, p0, Landroid/widget/OverScroller$SplineOverScroller;->mOver:I
  
+     .line 749
      iput-boolean v4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
+     .line 750
      iput p2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
  
      int-to-float v2, p2
  
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mCurrVelocity:F
  
+     .line 751
      iput v4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mSplineDuration:I
  
      iput v4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
+     .line 752
      invoke-static {}, Landroid/view/animation/AnimationUtils;->currentAnimationTimeMillis()J
  
      move-result-wide v2
  
      iput-wide v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
+     .line 753
      iput p1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
      iput p1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mCurrentPosition:I
  
+     .line 755
      if-gt p1, p4, :cond_0
  
      if-ge p1, p3, :cond_2
  
+     .line 756
      :cond_0
      invoke-direct {p0, p1, p3, p4, p2}, Landroid/widget/OverScroller$SplineOverScroller;->startAfterEdge(IIII)V
  
+     .line 781
      :cond_1
      :goto_0
      return-void
  
+     .line 760
      :cond_2
      iput v4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mState:I
  
+     .line 761
      const-wide/16 v0, 0x0
  
+     .line 763
      .local v0, totalDistance:D
      if-eqz p2, :cond_3
  
+     .line 764
      invoke-direct {p0, p2}, Landroid/widget/OverScroller$SplineOverScroller;->getSplineFlingDuration(I)I
  
      move-result v2
***************
*** 1372,1381 ****
--- 1519,1530 ----
  
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
+     .line 765
      invoke-direct {p0, p2}, Landroid/widget/OverScroller$SplineOverScroller;->getSplineFlingDistance(I)D
  
      move-result-wide v0
  
+     .line 768
      :cond_3
      int-to-float v2, p2
  
***************
*** 1391,1425 ****
--- 1540,1581 ----
  
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mSplineDistance:I
  
+     .line 769
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mSplineDistance:I
  
      add-int/2addr v2, p1
  
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
+     .line 772
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      if-ge v2, p3, :cond_4
  
+     .line 773
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
      iget v3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      invoke-direct {p0, v2, v3, p3}, Landroid/widget/OverScroller$SplineOverScroller;->adjustDuration(III)V
  
+     .line 774
      iput p3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
+     .line 777
      :cond_4
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      if-le v2, p4, :cond_1
  
+     .line 778
      iget v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
      iget v3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      invoke-direct {p0, v2, v3, p4}, Landroid/widget/OverScroller$SplineOverScroller;->adjustDuration(III)V
  
+     .line 779
      iput p4, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      goto :goto_0
***************
*** 1432,1455 ****
--- 1588,1616 ----
      .parameter "over"
  
      .prologue
+     .line 843
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mState:I
  
      if-nez v0, :cond_0
  
+     .line 844
      iput p3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mOver:I
  
+     .line 845
      invoke-static {}, Landroid/view/animation/AnimationUtils;->currentAnimationTimeMillis()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
+     .line 848
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mCurrVelocity:F
  
      float-to-int v0, v0
  
      invoke-direct {p0, p1, p2, p2, v0}, Landroid/widget/OverScroller$SplineOverScroller;->startAfterEdge(IIII)V
  
+     .line 850
      :cond_0
      return-void
  .end method
***************
*** 1459,1470 ****
--- 1620,1634 ----
      .parameter "position"
  
      .prologue
+     .line 704
      iput p1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
+     .line 705
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
+     .line 706
      return-void
  .end method
  
***************
*** 1473,1480 ****
--- 1637,1646 ----
      .parameter "friction"
  
      .prologue
+     .line 644
      iput p1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFlingFriction:F
  
+     .line 645
      return-void
  .end method
  
***************
*** 1489,1514 ****
--- 1655,1688 ----
  
      const/4 v1, 0x0
  
+     .line 716
      iput-boolean v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
+     .line 718
      iput p1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
      iput p1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
+     .line 719
      iput v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
  
+     .line 721
      invoke-static {}, Landroid/view/animation/AnimationUtils;->currentAnimationTimeMillis()J
  
      move-result-wide v2
  
      iput-wide v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
+     .line 722
      iput v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
+     .line 724
      if-ge p1, p2, :cond_1
  
+     .line 725
      invoke-direct {p0, p1, p2, v1}, Landroid/widget/OverScroller$SplineOverScroller;->startSpringback(III)V
  
+     .line 730
      :cond_0
      :goto_0
      iget-boolean v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
***************
*** 1518,1526 ****
--- 1692,1702 ----
      :goto_1
      return v0
  
+     .line 726
      :cond_1
      if-le p1, p3, :cond_0
  
+     .line 727
      invoke-direct {p0, p1, p3, v1}, Landroid/widget/OverScroller$SplineOverScroller;->startSpringback(III)V
  
      goto :goto_0
***************
*** 1528,1533 ****
--- 1704,1710 ----
      :cond_2
      move v0, v1
  
+     .line 730
      goto :goto_1
  .end method
  
***************
*** 1540,1567 ****
--- 1717,1752 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 682
      iput-boolean v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinished:Z
  
+     .line 684
      iput p1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
+     .line 685
      add-int v0, p1, p2
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
  
+     .line 687
      invoke-static {}, Landroid/view/animation/AnimationUtils;->currentAnimationTimeMillis()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStartTime:J
  
+     .line 688
      iput p3, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDuration:I
  
+     .line 691
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mDeceleration:F
  
+     .line 692
      iput v2, p0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
  
+     .line 693
      return-void
  .end method
  
***************
*** 1569,1578 ****
--- 1754,1765 ----
      .locals 21
  
      .prologue
+     .line 904
      invoke-static {}, Landroid/view/animation/AnimationUtils;->currentAnimationTimeMillis()J
  
      move-result-wide v15
  
+     .line 905
      .local v15, time:J
      move-object/from16 v0, p0
  
***************
*** 1582,1587 ****
--- 1769,1775 ----
  
      sub-long v2, v15, v18
  
+     .line 907
      .local v2, currentTime:J
      move-object/from16 v0, p0
  
***************
*** 1599,1612 ****
--- 1787,1804 ----
  
      if-lez v18, :cond_0
  
+     .line 908
      const/16 v18, 0x0
  
+     .line 951
      :goto_0
      return v18
  
+     .line 911
      :cond_0
      const-wide/16 v6, 0x0
  
+     .line 912
      .local v6, distance:D
      move-object/from16 v0, p0
  
***************
*** 1616,1621 ****
--- 1808,1814 ----
  
      packed-switch v18, :pswitch_data_0
  
+     .line 949
      :goto_1
      move-object/from16 v0, p0
  
***************
*** 1641,1650 ****
--- 1834,1845 ----
  
      iput v0, v1, Landroid/widget/OverScroller$SplineOverScroller;->mCurrentPosition:I
  
+     .line 951
      const/16 v18, 0x1
  
      goto :goto_0
  
+     .line 914
      :pswitch_0
      long-to-float v0, v2
  
***************
*** 1664,1669 ****
--- 1859,1865 ----
  
      div-float v11, v18, v19
  
+     .line 915
      .local v11, t:F
      const/high16 v18, 0x42c8
  
***************
*** 1673,1684 ****
--- 1869,1883 ----
  
      float-to-int v9, v0
  
+     .line 916
      .local v9, index:I
      const/high16 v8, 0x3f80
  
+     .line 917
      .local v8, distanceCoef:F
      const/16 v17, 0x0
  
+     .line 918
      .local v17, velocityCoef:F
      const/16 v18, 0x64
  
***************
*** 1686,1691 ****
--- 1885,1891 ----
  
      if-ge v9, v0, :cond_1
  
+     .line 919
      int-to-float v0, v9
  
      move/from16 v18, v0
***************
*** 1694,1699 ****
--- 1894,1900 ----
  
      div-float v13, v18, v19
  
+     .line 920
      .local v13, t_inf:F
      add-int/lit8 v18, v9, 0x1
  
***************
*** 1707,1717 ****
--- 1908,1920 ----
  
      div-float v14, v18, v19
  
+     .line 921
      .local v14, t_sup:F
      sget-object v18, Landroid/widget/OverScroller$SplineOverScroller;->SPLINE_POSITION:[F
  
      aget v4, v18, v9
  
+     .line 922
      .local v4, d_inf:F
      sget-object v18, Landroid/widget/OverScroller$SplineOverScroller;->SPLINE_POSITION:[F
  
***************
*** 1719,1724 ****
--- 1922,1928 ----
  
      aget v5, v18, v19
  
+     .line 923
      .local v5, d_sup:F
      sub-float v18, v5, v4
  
***************
*** 1726,1737 ****
--- 1930,1943 ----
  
      div-float v17, v18, v19
  
+     .line 924
      sub-float v18, v11, v13
  
      mul-float v18, v18, v17
  
      add-float v8, v4, v18
  
+     .line 927
      .end local v4           #d_inf:F
      .end local v5           #d_sup:F
      .end local v13           #t_inf:F
***************
*** 1755,1760 ****
--- 1961,1967 ----
  
      float-to-double v6, v0
  
+     .line 928
      move-object/from16 v0, p0
  
      iget v0, v0, Landroid/widget/OverScroller$SplineOverScroller;->mSplineDistance:I
***************
*** 1795,1800 ****
--- 2002,2008 ----
  
      goto/16 :goto_1
  
+     .line 933
      .end local v8           #distanceCoef:F
      .end local v9           #index:I
      .end local v11           #t:F
***************
*** 1808,1813 ****
--- 2016,2022 ----
  
      div-float v11, v18, v19
  
+     .line 934
      .restart local v11       #t:F
      move-object/from16 v0, p0
  
***************
*** 1837,1842 ****
--- 2046,2052 ----
  
      iput v0, v1, Landroid/widget/OverScroller$SplineOverScroller;->mCurrVelocity:F
  
+     .line 935
      move-object/from16 v0, p0
  
      iget v0, v0, Landroid/widget/OverScroller$SplineOverScroller;->mVelocity:I
***************
*** 1871,1878 ****
--- 2081,2090 ----
  
      float-to-double v6, v0
  
+     .line 936
      goto/16 :goto_1
  
+     .line 940
      .end local v11           #t:F
      :pswitch_2
      long-to-float v0, v2
***************
*** 1893,1901 ****
--- 2105,2115 ----
  
      div-float v11, v18, v19
  
+     .line 941
      .restart local v11       #t:F
      mul-float v12, v11, v11
  
+     .line 942
      .local v12, t2:F
      move-object/from16 v0, p0
  
***************
*** 1913,1918 ****
--- 2127,2133 ----
  
      move-result v10
  
+     .line 943
      .local v10, sign:F
      move-object/from16 v0, p0
  
***************
*** 1946,1951 ****
--- 2161,2167 ----
  
      float-to-double v6, v0
  
+     .line 944
      move-object/from16 v0, p0
  
      iget v0, v0, Landroid/widget/OverScroller$SplineOverScroller;->mOver:I
***************
*** 1980,1985 ****
--- 2196,2202 ----
  
      goto/16 :goto_1
  
+     .line 912
      nop
  
      :pswitch_data_0
***************
*** 1995,2000 ****
--- 2212,2218 ----
      .parameter "q"
  
      .prologue
+     .line 652
      iget v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mStart:I
  
      iget v1, p0, Landroid/widget/OverScroller$SplineOverScroller;->mFinal:I
***************
*** 2015,2019 ****
--- 2233,2238 ----
  
      iput v0, p0, Landroid/widget/OverScroller$SplineOverScroller;->mCurrentPosition:I
  
+     .line 653
      return-void
  .end method
