*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/QuickContactBadge.smali	2012-03-23 19:35:51.000516879 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/QuickContactBadge.smali.original	2012-03-23 19:34:44.868310960 -0500
***************
*** 63,68 ****
--- 63,69 ----
  
      const/4 v2, 0x0
  
+     .line 59
      new-array v0, v4, [Ljava/lang/String;
  
      const-string v1, "contact_id"
***************
*** 75,80 ****
--- 76,82 ----
  
      sput-object v0, Landroid/widget/QuickContactBadge;->EMAIL_LOOKUP_PROJECTION:[Ljava/lang/String;
  
+     .line 66
      new-array v0, v4, [Ljava/lang/String;
  
      const-string v1, "_id"
***************
*** 95,104 ****
--- 97,108 ----
      .parameter "context"
  
      .prologue
+     .line 74
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/widget/QuickContactBadge;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 75
      return-void
  .end method
  
***************
*** 108,117 ****
--- 112,123 ----
      .parameter "attrs"
  
      .prologue
+     .line 78
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, p2, v0}, Landroid/widget/QuickContactBadge;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 79
      return-void
  .end method
  
***************
*** 122,133 ****
--- 128,142 ----
      .parameter "defStyle"
  
      .prologue
+     .line 82
      invoke-direct {p0, p1, p2, p3}, Landroid/widget/ImageView;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 52
      const/4 v1, 0x0
  
      iput-object v1, p0, Landroid/widget/QuickContactBadge;->mExcludeMimes:[Ljava/lang/String;
  
+     .line 84
      iget-object v1, p0, Landroid/widget/QuickContactBadge;->mContext:Landroid/content/Context;
  
      sget-object v2, Lcom/android/internal/R$styleable;->Theme:[I
***************
*** 136,141 ****
--- 145,151 ----
  
      move-result-object v0
  
+     .line 85
      .local v0, styledAttributes:Landroid/content/res/TypedArray;
      const/16 v1, 0xe4
  
***************
*** 145,152 ****
--- 155,164 ----
  
      iput-object v1, p0, Landroid/widget/QuickContactBadge;->mOverlay:Landroid/graphics/drawable/Drawable;
  
+     .line 87
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 89
      new-instance v1, Landroid/widget/QuickContactBadge$QueryHandler;
  
      iget-object v2, p0, Landroid/widget/QuickContactBadge;->mContext:Landroid/content/Context;
***************
*** 159,166 ****
--- 171,180 ----
  
      iput-object v1, p0, Landroid/widget/QuickContactBadge;->mQueryHandler:Landroid/widget/QuickContactBadge$QueryHandler;
  
+     .line 90
      invoke-virtual {p0, p0}, Landroid/widget/QuickContactBadge;->setOnClickListener(Landroid/view/View$OnClickListener;)V
  
+     .line 91
      return-void
  .end method
  
***************
*** 170,175 ****
--- 184,190 ----
      .parameter "x1"
  
      .prologue
+     .line 44
      iput-object p1, p0, Landroid/widget/QuickContactBadge;->mContactUri:Landroid/net/Uri;
  
      return-object p1
***************
*** 180,185 ****
--- 195,201 ----
      .parameter "x0"
  
      .prologue
+     .line 44
      invoke-direct {p0}, Landroid/widget/QuickContactBadge;->onContactUriChanged()V
  
      return-void
***************
*** 189,194 ****
--- 205,211 ----
      .locals 1
  
      .prologue
+     .line 137
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mContactUri:Landroid/net/Uri;
  
      if-nez v0, :cond_0
***************
*** 217,228 ****
--- 234,247 ----
      .locals 1
  
      .prologue
+     .line 209
      invoke-direct {p0}, Landroid/widget/QuickContactBadge;->isAssigned()Z
  
      move-result v0
  
      invoke-virtual {p0, v0}, Landroid/widget/QuickContactBadge;->setEnabled(Z)V
  
+     .line 210
      return-void
  .end method
  
***************
*** 236,245 ****
--- 255,267 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 176
      iput-object p1, p0, Landroid/widget/QuickContactBadge;->mContactEmail:Ljava/lang/String;
  
+     .line 177
      if-nez p2, :cond_0
  
+     .line 178
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mQueryHandler:Landroid/widget/QuickContactBadge$QueryHandler;
  
      const/4 v1, 0x0
***************
*** 266,277 ****
--- 288,302 ----
  
      invoke-virtual/range {v0 .. v7}, Landroid/widget/QuickContactBadge$QueryHandler;->startQuery(ILjava/lang/Object;Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 185
      :goto_0
      return-void
  
+     .line 182
      :cond_0
      iput-object v2, p0, Landroid/widget/QuickContactBadge;->mContactUri:Landroid/net/Uri;
  
+     .line 183
      invoke-direct {p0}, Landroid/widget/QuickContactBadge;->onContactUriChanged()V
  
      goto :goto_0
***************
*** 285,294 ****
--- 310,322 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 197
      iput-object p1, p0, Landroid/widget/QuickContactBadge;->mContactPhone:Ljava/lang/String;
  
+     .line 198
      if-nez p2, :cond_0
  
+     .line 199
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mQueryHandler:Landroid/widget/QuickContactBadge$QueryHandler;
  
      const/4 v1, 0x1
***************
*** 311,322 ****
--- 339,353 ----
  
      invoke-virtual/range {v0 .. v7}, Landroid/widget/QuickContactBadge$QueryHandler;->startQuery(ILjava/lang/Object;Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 206
      :goto_0
      return-void
  
+     .line 203
      :cond_0
      iput-object v2, p0, Landroid/widget/QuickContactBadge;->mContactUri:Landroid/net/Uri;
  
+     .line 204
      invoke-direct {p0}, Landroid/widget/QuickContactBadge;->onContactUriChanged()V
  
      goto :goto_0
***************
*** 329,342 ****
--- 360,378 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 160
      iput-object p1, p0, Landroid/widget/QuickContactBadge;->mContactUri:Landroid/net/Uri;
  
+     .line 161
      iput-object v0, p0, Landroid/widget/QuickContactBadge;->mContactEmail:Ljava/lang/String;
  
+     .line 162
      iput-object v0, p0, Landroid/widget/QuickContactBadge;->mContactPhone:Ljava/lang/String;
  
+     .line 163
      invoke-direct {p0}, Landroid/widget/QuickContactBadge;->onContactUriChanged()V
  
+     .line 164
      return-void
  .end method
  
***************
*** 344,351 ****
--- 380,389 ----
      .locals 2
  
      .prologue
+     .line 95
      invoke-super {p0}, Landroid/widget/ImageView;->drawableStateChanged()V
  
+     .line 96
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mOverlay:Landroid/graphics/drawable/Drawable;
  
      if-eqz v0, :cond_0
***************
*** 358,363 ****
--- 396,402 ----
  
      if-eqz v0, :cond_0
  
+     .line 97
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mOverlay:Landroid/graphics/drawable/Drawable;
  
      invoke-virtual {p0}, Landroid/widget/QuickContactBadge;->getDrawableState()[I
***************
*** 366,373 ****
--- 405,414 ----
  
      invoke-virtual {v0, v1}, Landroid/graphics/drawable/Drawable;->setState([I)Z
  
+     .line 98
      invoke-virtual {p0}, Landroid/widget/QuickContactBadge;->invalidate()V
  
+     .line 100
      :cond_0
      return-void
  .end method
***************
*** 381,390 ****
--- 422,433 ----
  
      const/4 v5, 0x0
  
+     .line 214
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mContactUri:Landroid/net/Uri;
  
      if-eqz v0, :cond_1
  
+     .line 215
      invoke-virtual {p0}, Landroid/widget/QuickContactBadge;->getContext()Landroid/content/Context;
  
      move-result-object v0
***************
*** 395,409 ****
--- 438,455 ----
  
      invoke-static {v0, p0, v2, v1, v3}, Landroid/provider/ContactsContract$QuickContact;->showQuickContact(Landroid/content/Context;Landroid/view/View;Landroid/net/Uri;I[Ljava/lang/String;)V
  
+     .line 229
      :cond_0
      :goto_0
      return-void
  
+     .line 217
      :cond_1
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mContactEmail:Ljava/lang/String;
  
      if-eqz v0, :cond_2
  
+     .line 218
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mQueryHandler:Landroid/widget/QuickContactBadge$QueryHandler;
  
      const/4 v1, 0x2
***************
*** 432,442 ****
--- 478,490 ----
  
      goto :goto_0
  
+     .line 221
      :cond_2
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mContactPhone:Ljava/lang/String;
  
      if-eqz v0, :cond_0
  
+     .line 222
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mQueryHandler:Landroid/widget/QuickContactBadge$QueryHandler;
  
      iget-object v2, p0, Landroid/widget/QuickContactBadge;->mContactPhone:Ljava/lang/String;
***************
*** 467,484 ****
--- 515,536 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 109
      invoke-super {p0, p1}, Landroid/widget/ImageView;->onDraw(Landroid/graphics/Canvas;)V
  
+     .line 111
      invoke-virtual {p0}, Landroid/widget/QuickContactBadge;->isEnabled()Z
  
      move-result v1
  
      if-nez v1, :cond_1
  
+     .line 133
      :cond_0
      :goto_0
      return-void
  
+     .line 116
      :cond_1
      iget-object v1, p0, Landroid/widget/QuickContactBadge;->mOverlay:Landroid/graphics/drawable/Drawable;
  
***************
*** 500,505 ****
--- 552,558 ----
  
      if-eqz v1, :cond_0
  
+     .line 122
      iget-object v1, p0, Landroid/widget/QuickContactBadge;->mOverlay:Landroid/graphics/drawable/Drawable;
  
      invoke-virtual {p0}, Landroid/widget/QuickContactBadge;->getWidth()I
***************
*** 512,517 ****
--- 565,571 ----
  
      invoke-virtual {v1, v4, v4, v2, v3}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
  
+     .line 124
      iget v1, p0, Landroid/widget/QuickContactBadge;->mPaddingTop:I
  
      if-nez v1, :cond_2
***************
*** 520,539 ****
--- 574,597 ----
  
      if-nez v1, :cond_2
  
+     .line 125
      iget-object v1, p0, Landroid/widget/QuickContactBadge;->mOverlay:Landroid/graphics/drawable/Drawable;
  
      invoke-virtual {v1, p1}, Landroid/graphics/drawable/Drawable;->draw(Landroid/graphics/Canvas;)V
  
      goto :goto_0
  
+     .line 127
      :cond_2
      invoke-virtual {p1}, Landroid/graphics/Canvas;->getSaveCount()I
  
      move-result v0
  
+     .line 128
      .local v0, saveCount:I
      invoke-virtual {p1}, Landroid/graphics/Canvas;->save()I
  
+     .line 129
      iget v1, p0, Landroid/widget/QuickContactBadge;->mPaddingLeft:I
  
      int-to-float v1, v1
***************
*** 544,553 ****
--- 602,613 ----
  
      invoke-virtual {p1, v1, v2}, Landroid/graphics/Canvas;->translate(FF)V
  
+     .line 130
      iget-object v1, p0, Landroid/widget/QuickContactBadge;->mOverlay:Landroid/graphics/drawable/Drawable;
  
      invoke-virtual {v1, p1}, Landroid/graphics/drawable/Drawable;->draw(Landroid/graphics/Canvas;)V
  
+     .line 131
      invoke-virtual {p1, v0}, Landroid/graphics/Canvas;->restoreToCount(I)V
  
      goto :goto_0
***************
*** 558,565 ****
--- 618,627 ----
      .parameter "excludeMimes"
  
      .prologue
+     .line 237
      iput-object p1, p0, Landroid/widget/QuickContactBadge;->mExcludeMimes:[Ljava/lang/String;
  
+     .line 238
      return-void
  .end method
  
***************
*** 567,576 ****
--- 629,640 ----
      .locals 2
  
      .prologue
+     .line 144
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mDefaultAvatar:Landroid/graphics/drawable/Drawable;
  
      if-nez v0, :cond_0
  
+     .line 145
      invoke-virtual {p0}, Landroid/widget/QuickContactBadge;->getResources()Landroid/content/res/Resources;
  
      move-result-object v0
***************
*** 583,593 ****
--- 647,659 ----
  
      iput-object v0, p0, Landroid/widget/QuickContactBadge;->mDefaultAvatar:Landroid/graphics/drawable/Drawable;
  
+     .line 147
      :cond_0
      iget-object v0, p0, Landroid/widget/QuickContactBadge;->mDefaultAvatar:Landroid/graphics/drawable/Drawable;
  
      invoke-virtual {p0, v0}, Landroid/widget/QuickContactBadge;->setImageDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 148
      return-void
  .end method
  
***************
*** 596,600 ****
--- 662,667 ----
      .parameter "size"
  
      .prologue
+     .line 105
      return-void
  .end method
