*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RadioGroup$CheckedStateTracker.smali	2012-03-23 19:35:49.279485495 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RadioGroup$CheckedStateTracker.smali.original	2012-03-23 19:34:44.855310722 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 328
      iput-object p1, p0, Landroid/widget/RadioGroup$CheckedStateTracker;->this$0:Landroid/widget/RadioGroup;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .parameter "x1"
  
      .prologue
+     .line 328
      invoke-direct {p0, p1}, Landroid/widget/RadioGroup$CheckedStateTracker;-><init>(Landroid/widget/RadioGroup;)V
  
      return-void
***************
*** 55,60 ****
--- 57,63 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 331
      iget-object v1, p0, Landroid/widget/RadioGroup$CheckedStateTracker;->this$0:Landroid/widget/RadioGroup;
  
      #getter for: Landroid/widget/RadioGroup;->mProtectFromCheckedChange:Z
***************
*** 64,72 ****
--- 67,77 ----
  
      if-eqz v1, :cond_0
  
+     .line 343
      :goto_0
      return-void
  
+     .line 335
      :cond_0
      iget-object v1, p0, Landroid/widget/RadioGroup$CheckedStateTracker;->this$0:Landroid/widget/RadioGroup;
  
***************
*** 75,80 ****
--- 80,86 ----
      #setter for: Landroid/widget/RadioGroup;->mProtectFromCheckedChange:Z
      invoke-static {v1, v2}, Landroid/widget/RadioGroup;->access$302(Landroid/widget/RadioGroup;Z)Z
  
+     .line 336
      iget-object v1, p0, Landroid/widget/RadioGroup$CheckedStateTracker;->this$0:Landroid/widget/RadioGroup;
  
      #getter for: Landroid/widget/RadioGroup;->mCheckedId:I
***************
*** 86,91 ****
--- 92,98 ----
  
      if-eq v1, v2, :cond_1
  
+     .line 337
      iget-object v1, p0, Landroid/widget/RadioGroup$CheckedStateTracker;->this$0:Landroid/widget/RadioGroup;
  
      iget-object v2, p0, Landroid/widget/RadioGroup$CheckedStateTracker;->this$0:Landroid/widget/RadioGroup;
***************
*** 98,113 ****
--- 105,123 ----
      #calls: Landroid/widget/RadioGroup;->setCheckedStateForView(IZ)V
      invoke-static {v1, v2, v3}, Landroid/widget/RadioGroup;->access$500(Landroid/widget/RadioGroup;IZ)V
  
+     .line 339
      :cond_1
      iget-object v1, p0, Landroid/widget/RadioGroup$CheckedStateTracker;->this$0:Landroid/widget/RadioGroup;
  
      #setter for: Landroid/widget/RadioGroup;->mProtectFromCheckedChange:Z
      invoke-static {v1, v3}, Landroid/widget/RadioGroup;->access$302(Landroid/widget/RadioGroup;Z)Z
  
+     .line 341
      invoke-virtual {p1}, Landroid/widget/CompoundButton;->getId()I
  
      move-result v0
  
+     .line 342
      .local v0, id:I
      iget-object v1, p0, Landroid/widget/RadioGroup$CheckedStateTracker;->this$0:Landroid/widget/RadioGroup;
  
