*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RelativeLayout$DependencyGraph$Node.smali	2012-03-23 19:35:58.360651091 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RelativeLayout$DependencyGraph$Node.smali.original	2012-03-23 19:34:44.927312036 -0500
***************
*** 77,82 ****
--- 77,83 ----
      .locals 2
  
      .prologue
+     .line 1436
      new-instance v0, Landroid/widget/RelativeLayout$DependencyGraph$Node$1;
  
      invoke-direct {v0}, Landroid/widget/RelativeLayout$DependencyGraph$Node$1;-><init>()V
***************
*** 100,113 ****
--- 101,117 ----
      .locals 1
  
      .prologue
+     .line 1413
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1423
      new-instance v0, Ljava/util/HashSet;
  
      invoke-direct {v0}, Ljava/util/HashSet;-><init>()V
  
      iput-object v0, p0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->dependents:Ljava/util/HashSet;
  
+     .line 1428
      new-instance v0, Landroid/util/SparseArray;
  
      invoke-direct {v0}, Landroid/util/SparseArray;-><init>()V
***************
*** 122,127 ****
--- 126,132 ----
      .parameter "view"
  
      .prologue
+     .line 1470
      sget-object v1, Landroid/widget/RelativeLayout$DependencyGraph$Node;->sPool:Landroid/util/Pool;
  
      invoke-interface {v1}, Landroid/util/Pool;->acquire()Landroid/util/Poolable;
***************
*** 130,138 ****
--- 135,145 ----
  
      check-cast v0, Landroid/widget/RelativeLayout$DependencyGraph$Node;
  
+     .line 1471
      .local v0, node:Landroid/widget/RelativeLayout$DependencyGraph$Node;
      iput-object p0, v0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->view:Landroid/view/View;
  
+     .line 1473
      return-object v0
  .end method
  
***************
*** 142,147 ****
--- 149,155 ----
      .locals 1
  
      .prologue
+     .line 1458
      iget-object v0, p0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->mNext:Landroid/widget/RelativeLayout$DependencyGraph$Node;
  
      return-object v0
***************
*** 151,156 ****
--- 159,165 ----
      .locals 1
  
      .prologue
+     .line 1413
      invoke-virtual {p0}, Landroid/widget/RelativeLayout$DependencyGraph$Node;->getNextPoolable()Landroid/widget/RelativeLayout$DependencyGraph$Node;
  
      move-result-object v0
***************
*** 162,167 ****
--- 171,177 ----
      .locals 1
  
      .prologue
+     .line 1462
      iget-boolean v0, p0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->mIsPooled:Z
  
      return v0
***************
*** 171,192 ****
--- 181,207 ----
      .locals 1
  
      .prologue
+     .line 1477
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->view:Landroid/view/View;
  
+     .line 1478
      iget-object v0, p0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->dependents:Ljava/util/HashSet;
  
      invoke-virtual {v0}, Ljava/util/HashSet;->clear()V
  
+     .line 1479
      iget-object v0, p0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->dependencies:Landroid/util/SparseArray;
  
      invoke-virtual {v0}, Landroid/util/SparseArray;->clear()V
  
+     .line 1481
      sget-object v0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->sPool:Landroid/util/Pool;
  
      invoke-interface {v0, p0}, Landroid/util/Pool;->release(Landroid/util/Poolable;)V
  
+     .line 1482
      return-void
  .end method
  
***************
*** 195,202 ****
--- 210,219 ----
      .parameter "element"
  
      .prologue
+     .line 1454
      iput-object p1, p0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->mNext:Landroid/widget/RelativeLayout$DependencyGraph$Node;
  
+     .line 1455
      return-void
  .end method
  
***************
*** 205,210 ****
--- 222,228 ----
      .parameter "x0"
  
      .prologue
+     .line 1413
      check-cast p1, Landroid/widget/RelativeLayout$DependencyGraph$Node;
  
      .end local p1
***************
*** 218,224 ****
--- 236,244 ----
      .parameter "isPooled"
  
      .prologue
+     .line 1466
      iput-boolean p1, p0, Landroid/widget/RelativeLayout$DependencyGraph$Node;->mIsPooled:Z
  
+     .line 1467
      return-void
  .end method
