*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RelativeLayout$LayoutParams.smali	2012-03-23 19:35:50.834513851 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RelativeLayout$LayoutParams.smali.original	2012-03-23 19:34:44.867310942 -0500
***************
*** 120,133 ****
--- 120,136 ----
      .parameter "h"
  
      .prologue
+     .line 1133
      invoke-direct {p0, p1, p2}, Landroid/view/ViewGroup$MarginLayoutParams;-><init>(II)V
  
+     .line 1031
      const/16 v0, 0x10
  
      new-array v0, v0, [I
  
      iput-object v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRules:[I
  
+     .line 1134
      return-void
  .end method
  
***************
*** 141,168 ****
--- 144,177 ----
  
      const/4 v7, 0x0
  
+     .line 1064
      invoke-direct {p0, p1, p2}, Landroid/view/ViewGroup$MarginLayoutParams;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 1031
      const/16 v5, 0x10
  
      new-array v5, v5, [I
  
      iput-object v5, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRules:[I
  
+     .line 1066
      sget-object v5, Lcom/android/internal/R$styleable;->RelativeLayout_Layout:[I
  
      invoke-virtual {p1, p2, v5}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[I)Landroid/content/res/TypedArray;
  
      move-result-object v1
  
+     .line 1069
      .local v1, a:Landroid/content/res/TypedArray;
      iget-object v4, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRules:[I
  
+     .line 1071
      .local v4, rules:[I
      invoke-virtual {v1}, Landroid/content/res/TypedArray;->getIndexCount()I
  
      move-result v0
  
+     .line 1072
      .local v0, N:I
      const/4 v3, 0x0
  
***************
*** 170,187 ****
--- 179,200 ----
      :goto_0
      if-ge v3, v0, :cond_7
  
+     .line 1073
      invoke-virtual {v1, v3}, Landroid/content/res/TypedArray;->getIndex(I)I
  
      move-result v2
  
+     .line 1074
      .local v2, attr:I
      packed-switch v2, :pswitch_data_0
  
+     .line 1072
      :goto_1
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 1076
      :pswitch_0
      invoke-virtual {v1, v2, v7}, Landroid/content/res/TypedArray;->getBoolean(IZ)Z
  
***************
*** 191,196 ****
--- 204,210 ----
  
      goto :goto_1
  
+     .line 1079
      :pswitch_1
      invoke-virtual {v1, v2, v7}, Landroid/content/res/TypedArray;->getResourceId(II)I
  
***************
*** 200,205 ****
--- 214,220 ----
  
      goto :goto_1
  
+     .line 1082
      :pswitch_2
      const/4 v5, 0x1
  
***************
*** 211,216 ****
--- 226,232 ----
  
      goto :goto_1
  
+     .line 1085
      :pswitch_3
      const/4 v5, 0x2
  
***************
*** 222,227 ****
--- 238,244 ----
  
      goto :goto_1
  
+     .line 1088
      :pswitch_4
      const/4 v5, 0x3
  
***************
*** 233,238 ****
--- 250,256 ----
  
      goto :goto_1
  
+     .line 1091
      :pswitch_5
      const/4 v5, 0x4
  
***************
*** 244,249 ****
--- 262,268 ----
  
      goto :goto_1
  
+     .line 1094
      :pswitch_6
      const/4 v5, 0x5
  
***************
*** 255,260 ****
--- 274,280 ----
  
      goto :goto_1
  
+     .line 1097
      :pswitch_7
      const/4 v5, 0x6
  
***************
*** 266,271 ****
--- 286,292 ----
  
      goto :goto_1
  
+     .line 1100
      :pswitch_8
      const/4 v5, 0x7
  
***************
*** 277,282 ****
--- 298,304 ----
  
      goto :goto_1
  
+     .line 1103
      :pswitch_9
      const/16 v5, 0x8
  
***************
*** 288,293 ****
--- 310,316 ----
  
      goto :goto_1
  
+     .line 1106
      :pswitch_a
      const/16 v8, 0x9
  
***************
*** 309,314 ****
--- 332,338 ----
  
      goto :goto_2
  
+     .line 1109
      :pswitch_b
      const/16 v8, 0xa
  
***************
*** 330,335 ****
--- 354,360 ----
  
      goto :goto_3
  
+     .line 1112
      :pswitch_c
      const/16 v8, 0xb
  
***************
*** 351,356 ****
--- 376,382 ----
  
      goto :goto_4
  
+     .line 1115
      :pswitch_d
      const/16 v8, 0xc
  
***************
*** 372,377 ****
--- 398,404 ----
  
      goto :goto_5
  
+     .line 1118
      :pswitch_e
      const/16 v8, 0xd
  
***************
*** 393,398 ****
--- 420,426 ----
  
      goto :goto_6
  
+     .line 1121
      :pswitch_f
      const/16 v8, 0xe
  
***************
*** 414,419 ****
--- 442,448 ----
  
      goto :goto_7
  
+     .line 1124
      :pswitch_10
      const/16 v8, 0xf
  
***************
*** 435,446 ****
--- 464,478 ----
  
      goto :goto_8
  
+     .line 1129
      .end local v2           #attr:I
      :cond_7
      invoke-virtual {v1}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 1130
      return-void
  
+     .line 1074
      nop
  
      :pswitch_data_0
***************
*** 470,483 ****
--- 502,518 ----
      .parameter "source"
  
      .prologue
+     .line 1140
      invoke-direct {p0, p1}, Landroid/view/ViewGroup$MarginLayoutParams;-><init>(Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 1031
      const/16 v0, 0x10
  
      new-array v0, v0, [I
  
      iput-object v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRules:[I
  
+     .line 1141
      return-void
  .end method
  
***************
*** 486,499 ****
--- 521,537 ----
      .parameter "source"
  
      .prologue
+     .line 1147
      invoke-direct {p0, p1}, Landroid/view/ViewGroup$MarginLayoutParams;-><init>(Landroid/view/ViewGroup$MarginLayoutParams;)V
  
+     .line 1031
      const/16 v0, 0x10
  
      new-array v0, v0, [I
  
      iput-object v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRules:[I
  
+     .line 1148
      return-void
  .end method
  
***************
*** 502,507 ****
--- 540,546 ----
      .parameter "x0"
  
      .prologue
+     .line 1030
      iget v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRight:I
  
      return v0
***************
*** 513,518 ****
--- 552,558 ----
      .parameter "x1"
  
      .prologue
+     .line 1030
      iput p1, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRight:I
  
      return p1
***************
*** 524,529 ****
--- 564,570 ----
      .parameter "x1"
  
      .prologue
+     .line 1030
      iget v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRight:I
  
      add-int/2addr v0, p1
***************
*** 538,543 ****
--- 579,585 ----
      .parameter "x0"
  
      .prologue
+     .line 1030
      iget v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mBottom:I
  
      return v0
***************
*** 549,554 ****
--- 591,597 ----
      .parameter "x1"
  
      .prologue
+     .line 1030
      iput p1, p0, Landroid/widget/RelativeLayout$LayoutParams;->mBottom:I
  
      return p1
***************
*** 560,565 ****
--- 603,609 ----
      .parameter "x1"
  
      .prologue
+     .line 1030
      iget v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mBottom:I
  
      add-int/2addr v0, p1
***************
*** 574,579 ****
--- 618,624 ----
      .parameter "x0"
  
      .prologue
+     .line 1030
      iget v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mLeft:I
  
      return v0
***************
*** 585,590 ****
--- 630,636 ----
      .parameter "x1"
  
      .prologue
+     .line 1030
      iput p1, p0, Landroid/widget/RelativeLayout$LayoutParams;->mLeft:I
  
      return p1
***************
*** 596,601 ****
--- 642,648 ----
      .parameter "x1"
  
      .prologue
+     .line 1030
      iget v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mLeft:I
  
      add-int/2addr v0, p1
***************
*** 610,615 ****
--- 657,663 ----
      .parameter "x0"
  
      .prologue
+     .line 1030
      iget v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mTop:I
  
      return v0
***************
*** 621,626 ****
--- 669,675 ----
      .parameter "x1"
  
      .prologue
+     .line 1030
      iput p1, p0, Landroid/widget/RelativeLayout$LayoutParams;->mTop:I
  
      return p1
***************
*** 632,637 ****
--- 681,687 ----
      .parameter "x1"
  
      .prologue
+     .line 1030
      iget v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mTop:I
  
      add-int/2addr v0, p1
***************
*** 646,651 ****
--- 696,702 ----
      .parameter "x0"
  
      .prologue
+     .line 1030
      iget-object v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRules:[I
  
      return-object v0
***************
*** 658,669 ****
--- 709,722 ----
      .parameter "verb"
  
      .prologue
+     .line 1169
      iget-object v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRules:[I
  
      const/4 v1, -0x1
  
      aput v1, v0, p1
  
+     .line 1170
      return-void
  .end method
  
***************
*** 673,682 ****
--- 726,737 ----
      .parameter "anchor"
  
      .prologue
+     .line 1187
      iget-object v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRules:[I
  
      aput p2, v0, p1
  
+     .line 1188
      return-void
  .end method
  
***************
*** 685,690 ****
--- 740,746 ----
      .parameter "output"
  
      .prologue
+     .line 1152
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 742,747 ****
--- 798,804 ----
      .locals 1
  
      .prologue
+     .line 1199
      iget-object v0, p0, Landroid/widget/RelativeLayout$LayoutParams;->mRules:[I
  
      return-object v0
