*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViews$ReflectionActionWithoutParams.smali	2012-03-23 19:35:53.257558036 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViews$ReflectionActionWithoutParams.smali.original	2012-03-23 19:34:44.886311288 -0500
***************
*** 34,49 ****
--- 34,53 ----
      .parameter "methodName"
  
      .prologue
+     .line 579
      iput-object p1, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->this$0:Landroid/widget/RemoteViews;
  
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews$Action;-><init>(Landroid/widget/RemoteViews$1;)V
  
+     .line 580
      iput p2, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->viewId:I
  
+     .line 581
      iput-object p3, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->methodName:Ljava/lang/String;
  
+     .line 582
      return-void
  .end method
  
***************
*** 53,76 ****
--- 57,84 ----
      .parameter "in"
  
      .prologue
+     .line 584
      iput-object p1, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->this$0:Landroid/widget/RemoteViews;
  
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews$Action;-><init>(Landroid/widget/RemoteViews$1;)V
  
+     .line 585
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->viewId:I
  
+     .line 586
      invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->methodName:Ljava/lang/String;
  
+     .line 587
      return-void
  .end method
  
***************
*** 82,104 ****
--- 90,117 ----
      .parameter "rootParent"
  
      .prologue
+     .line 597
      iget v4, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->viewId:I
  
      invoke-virtual {p1, v4}, Landroid/view/View;->findViewById(I)Landroid/view/View;
  
      move-result-object v3
  
+     .line 598
      .local v3, view:Landroid/view/View;
      if-nez v3, :cond_0
  
+     .line 625
      :goto_0
      return-void
  
+     .line 600
      :cond_0
      invoke-virtual {v3}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
  
      move-result-object v1
  
+     .line 603
      .local v1, klass:Ljava/lang/Class;
      :try_start_0
      iget-object v4, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->methodName:Ljava/lang/String;
***************
*** 113,118 ****
--- 126,132 ----
  
      move-result-object v2
  
+     .line 609
      .local v2, method:Ljava/lang/reflect/Method;
      const-class v4, Landroid/view/RemotableViewMethod;
  
***************
*** 122,134 ****
  
      if-nez v4, :cond_1
  
      new-instance v4, Landroid/widget/RemoteViews$ActionException;
  
      new-instance v5, Ljava/lang/StringBuilder;
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v6, "view: "
  
      invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 136,149 ----
  
      if-nez v4, :cond_1
  
+     .line 610
      new-instance v4, Landroid/widget/RemoteViews$ActionException;
  
      new-instance v5, Ljava/lang/StringBuilder;
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v6, "view: "
  
      invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 168,177 ****
--- 183,194 ----
  
      throw v4
  
+     .line 604
      .end local v2           #method:Ljava/lang/reflect/Method;
      :catch_0
      move-exception v0
  
+     .line 605
      .local v0, ex:Ljava/lang/NoSuchMethodException;
      new-instance v4, Landroid/widget/RemoteViews$ActionException;
  
***************
*** 179,185 ****
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v6, "view: "
  
      invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 196,202 ----
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v6, "view: "
  
      invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 219,224 ****
--- 236,242 ----
  
      throw v4
  
+     .line 621
      .end local v0           #ex:Ljava/lang/NoSuchMethodException;
      .restart local v2       #method:Ljava/lang/reflect/Method;
      :cond_1
***************
*** 233,241 ****
--- 251,261 ----
  
      goto :goto_0
  
+     .line 622
      :catch_1
      move-exception v0
  
+     .line 623
      .local v0, ex:Ljava/lang/Exception;
      new-instance v4, Landroid/widget/RemoteViews$ActionException;
  
***************
*** 250,266 ****
--- 270,290 ----
      .parameter "flags"
  
      .prologue
+     .line 590
      const/4 v0, 0x5
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 591
      iget v0, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->viewId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 592
      iget-object v0, p0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;->methodName:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 593
      return-void
  .end method
