*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViews.smali	2012-03-23 19:36:02.132719875 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViews.smali.original	2012-03-23 19:34:44.949312436 -0500
***************
*** 71,76 ****
--- 71,77 ----
      .locals 1
  
      .prologue
+     .line 1661
      new-instance v0, Landroid/widget/RemoteViews$1;
  
      invoke-direct {v0}, Landroid/widget/RemoteViews$1;-><init>()V
***************
*** 89,110 ****
--- 90,116 ----
  
      const/4 v4, 0x0
  
+     .line 974
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 97
      iput-boolean v4, p0, Landroid/widget/RemoteViews;->mIsWidgetCollectionChild:Z
  
+     .line 975
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v5
  
      iput-object v5, p0, Landroid/widget/RemoteViews;->mPackage:Ljava/lang/String;
  
+     .line 976
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
  
      iput v5, p0, Landroid/widget/RemoteViews;->mLayoutId:I
  
+     .line 977
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v5
***************
*** 114,145 ****
--- 120,158 ----
      :goto_0
      iput-boolean v3, p0, Landroid/widget/RemoteViews;->mIsWidgetCollectionChild:Z
  
+     .line 979
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 980
      .local v0, count:I
      if-lez v0, :cond_1
  
+     .line 981
      new-instance v3, Ljava/util/ArrayList;
  
      invoke-direct {v3, v0}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
+     .line 982
      const/4 v1, 0x0
  
      .local v1, i:I
      :goto_1
      if-ge v1, v0, :cond_1
  
+     .line 983
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
+     .line 984
      .local v2, tag:I
      packed-switch v2, :pswitch_data_0
  
+     .line 1013
      :pswitch_0
      new-instance v3, Landroid/widget/RemoteViews$ActionException;
  
***************
*** 177,184 ****
--- 190,199 ----
      :cond_0
      move v3, v4
  
+     .line 977
      goto :goto_0
  
+     .line 986
      .restart local v0       #count:I
      .restart local v1       #i:I
      .restart local v2       #tag:I
***************
*** 191,201 ****
--- 206,218 ----
  
      invoke-virtual {v3, v4}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 982
      :goto_2
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 989
      :pswitch_2
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
***************
*** 207,212 ****
--- 224,230 ----
  
      goto :goto_2
  
+     .line 992
      :pswitch_3
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
***************
*** 218,223 ****
--- 236,242 ----
  
      goto :goto_2
  
+     .line 995
      :pswitch_4
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
***************
*** 229,234 ****
--- 248,254 ----
  
      goto :goto_2
  
+     .line 998
      :pswitch_5
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
***************
*** 240,245 ****
--- 260,266 ----
  
      goto :goto_2
  
+     .line 1001
      :pswitch_6
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
***************
*** 251,256 ****
--- 272,278 ----
  
      goto :goto_2
  
+     .line 1004
      :pswitch_7
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
***************
*** 262,267 ****
--- 284,290 ----
  
      goto :goto_2
  
+     .line 1007
      :pswitch_8
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
***************
*** 273,278 ****
--- 296,302 ----
  
      goto :goto_2
  
+     .line 1010
      :pswitch_9
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
***************
*** 284,289 ****
--- 308,314 ----
  
      goto :goto_2
  
+     .line 1019
      .end local v1           #i:I
      .end local v2           #tag:I
      :cond_1
***************
*** 295,304 ****
--- 320,332 ----
  
      iput-object v3, p0, Landroid/widget/RemoteViews;->mMemoryUsageCounter:Landroid/widget/RemoteViews$MemoryUsageCounter;
  
+     .line 1020
      invoke-direct {p0}, Landroid/widget/RemoteViews;->recalculateMemoryUsage()V
  
+     .line 1021
      return-void
  
+     .line 984
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_1
***************
*** 320,335 ****
--- 348,368 ----
      .parameter "layoutId"
  
      .prologue
+     .line 960
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 97
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/widget/RemoteViews;->mIsWidgetCollectionChild:Z
  
+     .line 961
      iput-object p1, p0, Landroid/widget/RemoteViews;->mPackage:Ljava/lang/String;
  
+     .line 962
      iput p2, p0, Landroid/widget/RemoteViews;->mLayoutId:I
  
+     .line 965
      new-instance v0, Landroid/widget/RemoteViews$MemoryUsageCounter;
  
      const/4 v1, 0x0
***************
*** 338,345 ****
--- 371,380 ----
  
      iput-object v0, p0, Landroid/widget/RemoteViews;->mMemoryUsageCounter:Landroid/widget/RemoteViews$MemoryUsageCounter;
  
+     .line 966
      invoke-direct {p0}, Landroid/widget/RemoteViews;->recalculateMemoryUsage()V
  
+     .line 967
      return-void
  .end method
  
***************
*** 348,353 ****
--- 383,389 ----
      .parameter "x0"
  
      .prologue
+     .line 57
      iget-boolean v0, p0, Landroid/widget/RemoteViews;->mIsWidgetCollectionChild:Z
  
      return v0
***************
*** 358,382 ****
--- 394,423 ----
      .parameter "a"
  
      .prologue
+     .line 1082
      iget-object v0, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      if-nez v0, :cond_0
  
+     .line 1083
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
+     .line 1085
      :cond_0
      iget-object v0, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 1088
      iget-object v0, p0, Landroid/widget/RemoteViews;->mMemoryUsageCounter:Landroid/widget/RemoteViews$MemoryUsageCounter;
  
      invoke-virtual {p1, v0}, Landroid/widget/RemoteViews$Action;->updateMemoryUsageEstimate(Landroid/widget/RemoteViews$MemoryUsageCounter;)V
  
+     .line 1089
      return-void
  .end method
  
***************
*** 386,401 ****
--- 427,445 ----
      .parameter "parent"
  
      .prologue
+     .line 1601
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      if-eqz v3, :cond_0
  
+     .line 1602
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v1
  
+     .line 1603
      .local v1, count:I
      const/4 v2, 0x0
  
***************
*** 403,408 ****
--- 447,453 ----
      :goto_0
      if-ge v2, v1, :cond_0
  
+     .line 1604
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v2}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 411,423 ****
--- 456,471 ----
  
      check-cast v0, Landroid/widget/RemoteViews$Action;
  
+     .line 1605
      .local v0, a:Landroid/widget/RemoteViews$Action;
      invoke-virtual {v0, p1, p2}, Landroid/widget/RemoteViews$Action;->apply(Landroid/view/View;Landroid/view/ViewGroup;)V
  
+     .line 1603
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 1608
      .end local v0           #a:Landroid/widget/RemoteViews$Action;
      .end local v1           #count:I
      .end local v2           #i:I
***************
*** 430,440 ****
--- 478,491 ----
      .parameter "context"
  
      .prologue
+     .line 1612
      iget-object v2, p0, Landroid/widget/RemoteViews;->mPackage:Ljava/lang/String;
  
+     .line 1614
      .local v2, packageName:Ljava/lang/String;
      if-eqz v2, :cond_0
  
+     .line 1616
      const/4 v3, 0x4
  
      :try_start_0
***************
*** 444,457 ****
--- 495,511 ----
  
      move-result-object v0
  
+     .line 1625
      .local v0, c:Landroid/content/Context;
      :goto_0
      return-object v0
  
+     .line 1617
      .end local v0           #c:Landroid/content/Context;
      :catch_0
      move-exception v1
  
+     .line 1618
      .local v1, e:Landroid/content/pm/PackageManager$NameNotFoundException;
      const-string v3, "RemoteViews"
  
***************
*** 481,491 ****
--- 535,548 ----
  
      invoke-static {v3, v4}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1619
      move-object v0, p1
  
+     .line 1620
      .restart local v0       #c:Landroid/content/Context;
      goto :goto_0
  
+     .line 1622
      .end local v0           #c:Landroid/content/Context;
      .end local v1           #e:Landroid/content/pm/PackageManager$NameNotFoundException;
      :cond_0
***************
*** 499,518 ****
--- 556,579 ----
      .locals 4
  
      .prologue
+     .line 1058
      iget-object v2, p0, Landroid/widget/RemoteViews;->mMemoryUsageCounter:Landroid/widget/RemoteViews$MemoryUsageCounter;
  
      invoke-virtual {v2}, Landroid/widget/RemoteViews$MemoryUsageCounter;->clear()V
  
+     .line 1061
      iget-object v2, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      if-eqz v2, :cond_0
  
+     .line 1062
      iget-object v2, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
  
      move-result v0
  
+     .line 1063
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 520,525 ****
--- 581,587 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 1064
      iget-object v2, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      invoke-virtual {v2, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 532,541 ****
--- 594,605 ----
  
      invoke-virtual {v2, v3}, Landroid/widget/RemoteViews$Action;->updateMemoryUsageEstimate(Landroid/widget/RemoteViews$MemoryUsageCounter;)V
  
+     .line 1063
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 1067
      .end local v0           #count:I
      .end local v1           #i:I
      :cond_0
***************
*** 550,561 ****
--- 614,627 ----
      .parameter "nestedView"
  
      .prologue
+     .line 1102
      new-instance v0, Landroid/widget/RemoteViews$ViewGroupAction;
  
      invoke-direct {v0, p0, p1, p2}, Landroid/widget/RemoteViews$ViewGroupAction;-><init>(Landroid/widget/RemoteViews;ILandroid/widget/RemoteViews;)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1103
      return-void
  .end method
  
***************
*** 565,574 ****
--- 631,642 ----
      .parameter "parent"
  
      .prologue
+     .line 1572
      invoke-direct {p0, p1}, Landroid/widget/RemoteViews;->prepareContext(Landroid/content/Context;)Landroid/content/Context;
  
      move-result-object v0
  
+     .line 1574
      .local v0, c:Landroid/content/Context;
      const-string v3, "layout_inflater"
  
***************
*** 578,590 ****
--- 646,661 ----
  
      check-cast v1, Landroid/view/LayoutInflater;
  
+     .line 1577
      .local v1, inflater:Landroid/view/LayoutInflater;
      invoke-virtual {v1, v0}, Landroid/view/LayoutInflater;->cloneInContext(Landroid/content/Context;)Landroid/view/LayoutInflater;
  
      move-result-object v1
  
+     .line 1578
      invoke-virtual {v1, p0}, Landroid/view/LayoutInflater;->setFilter(Landroid/view/LayoutInflater$Filter;)V
  
+     .line 1580
      iget v3, p0, Landroid/widget/RemoteViews;->mLayoutId:I
  
      const/4 v4, 0x0
***************
*** 593,601 ****
--- 664,674 ----
  
      move-result-object v2
  
+     .line 1582
      .local v2, result:Landroid/view/View;
      invoke-direct {p0, v2, p2}, Landroid/widget/RemoteViews;->performApply(Landroid/view/View;Landroid/view/ViewGroup;)V
  
+     .line 1584
      return-object v2
  .end method
  
***************
*** 603,608 ****
--- 676,682 ----
      .locals 3
  
      .prologue
+     .line 1025
      new-instance v0, Landroid/widget/RemoteViews;
  
      iget-object v1, p0, Landroid/widget/RemoteViews;->mPackage:Ljava/lang/String;
***************
*** 611,621 ****
--- 685,697 ----
  
      invoke-direct {v0, v1, v2}, Landroid/widget/RemoteViews;-><init>(Ljava/lang/String;I)V
  
+     .line 1026
      .local v0, that:Landroid/widget/RemoteViews;
      iget-object v1, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      if-eqz v1, :cond_0
  
+     .line 1027
      iget-object v1, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      invoke-virtual {v1}, Ljava/util/ArrayList;->clone()Ljava/lang/Object;
***************
*** 626,634 ****
--- 702,712 ----
  
      iput-object v1, v0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
+     .line 1031
      :cond_0
      invoke-direct {v0}, Landroid/widget/RemoteViews;->recalculateMemoryUsage()V
  
+     .line 1032
      return-object v0
  .end method
  
***************
*** 641,646 ****
--- 719,725 ----
      .end annotation
  
      .prologue
+     .line 57
      invoke-virtual {p0}, Landroid/widget/RemoteViews;->clone()Landroid/widget/RemoteViews;
  
      move-result-object v0
***************
*** 652,657 ****
--- 731,737 ----
      .locals 1
  
      .prologue
+     .line 1638
      const/4 v0, 0x0
  
      return v0
***************
*** 661,666 ****
--- 741,747 ----
      .locals 1
  
      .prologue
+     .line 1073
      iget-object v0, p0, Landroid/widget/RemoteViews;->mMemoryUsageCounter:Landroid/widget/RemoteViews$MemoryUsageCounter;
  
      invoke-virtual {v0}, Landroid/widget/RemoteViews$MemoryUsageCounter;->getBitmapHeapMemoryUsage()I
***************
*** 674,679 ****
--- 755,761 ----
      .locals 1
  
      .prologue
+     .line 1040
      iget v0, p0, Landroid/widget/RemoteViews;->mLayoutId:I
  
      return v0
***************
*** 683,688 ****
--- 765,771 ----
      .locals 1
  
      .prologue
+     .line 1036
      iget-object v0, p0, Landroid/widget/RemoteViews;->mPackage:Ljava/lang/String;
  
      return-object v0
***************
*** 693,698 ****
--- 776,782 ----
      .parameter "clazz"
  
      .prologue
+     .line 1634
      const-class v0, Landroid/widget/RemoteViews$RemoteView;
  
      invoke-virtual {p1, v0}, Ljava/lang/Class;->isAnnotationPresent(Ljava/lang/Class;)Z
***************
*** 708,715 ****
--- 792,801 ----
      .parameter "v"
  
      .prologue
+     .line 1596
      invoke-direct {p0, p1}, Landroid/widget/RemoteViews;->prepareContext(Landroid/content/Context;)Landroid/content/Context;
  
+     .line 1597
      invoke-virtual {p2}, Landroid/view/View;->getParent()Landroid/view/ViewParent;
  
      move-result-object v0
***************
*** 718,723 ****
--- 804,810 ----
  
      invoke-direct {p0, p2, v0}, Landroid/widget/RemoteViews;->performApply(Landroid/view/View;Landroid/view/ViewGroup;)V
  
+     .line 1598
      return-void
  .end method
  
***************
*** 726,731 ****
--- 813,819 ----
      .parameter "viewId"
  
      .prologue
+     .line 1112
      new-instance v0, Landroid/widget/RemoteViews$ViewGroupAction;
  
      const/4 v1, 0x0
***************
*** 734,739 ****
--- 822,828 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1113
      return-void
  .end method
  
***************
*** 744,749 ****
--- 833,839 ----
      .parameter "value"
  
      .prologue
+     .line 1524
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/16 v4, 0xc
***************
*** 760,765 ****
--- 850,856 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1525
      return-void
  .end method
  
***************
*** 770,775 ****
--- 861,867 ----
      .parameter "value"
  
      .prologue
+     .line 1400
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/4 v4, 0x1
***************
*** 788,793 ****
--- 880,886 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1401
      return-void
  .end method
  
***************
*** 798,803 ****
--- 891,897 ----
      .parameter "value"
  
      .prologue
+     .line 1535
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/16 v4, 0xd
***************
*** 814,819 ****
--- 908,914 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1536
      return-void
  .end method
  
***************
*** 824,829 ****
--- 919,925 ----
      .parameter "value"
  
      .prologue
+     .line 1411
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/4 v4, 0x2
***************
*** 842,847 ****
--- 938,944 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1412
      return-void
  .end method
  
***************
*** 852,857 ****
--- 949,955 ----
      .parameter "value"
  
      .prologue
+     .line 1477
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/16 v4, 0x8
***************
*** 870,875 ****
--- 968,974 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1478
      return-void
  .end method
  
***************
*** 880,885 ****
--- 979,985 ----
      .parameter "value"
  
      .prologue
+     .line 1499
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/16 v4, 0xa
***************
*** 896,901 ****
--- 996,1002 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1500
      return-void
  .end method
  
***************
*** 907,924 ****
      .parameter "started"
  
      .prologue
!     const-string v0, "setBase"
  
      invoke-virtual {p0, p1, v0, p2, p3}, Landroid/widget/RemoteViews;->setLong(ILjava/lang/String;J)V
  
!     const-string v0, "setFormat"
  
      invoke-virtual {p0, p1, v0, p4}, Landroid/widget/RemoteViews;->setString(ILjava/lang/String;Ljava/lang/String;)V
  
!     const-string v0, "setStarted"
  
      invoke-virtual {p0, p1, v0, p5}, Landroid/widget/RemoteViews;->setBoolean(ILjava/lang/String;Z)V
  
      return-void
  .end method
  
--- 1008,1029 ----
      .parameter "started"
  
      .prologue
!     .line 1218
!     const-string/jumbo v0, "setBase"
  
      invoke-virtual {p0, p1, v0, p2, p3}, Landroid/widget/RemoteViews;->setLong(ILjava/lang/String;J)V
  
!     .line 1219
!     const-string/jumbo v0, "setFormat"
  
      invoke-virtual {p0, p1, v0, p4}, Landroid/widget/RemoteViews;->setString(ILjava/lang/String;Ljava/lang/String;)V
  
!     .line 1220
!     const-string/jumbo v0, "setStarted"
  
      invoke-virtual {p0, p1, v0, p5}, Landroid/widget/RemoteViews;->setBoolean(ILjava/lang/String;Z)V
  
+     .line 1221
      return-void
  .end method
  
***************
*** 928,937 ****
      .parameter "contentDescription"
  
      .prologue
!     const-string v0, "setContentDescription"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setCharSequence(ILjava/lang/String;Ljava/lang/CharSequence;)V
  
      return-void
  .end method
  
--- 1033,1044 ----
      .parameter "contentDescription"
  
      .prologue
!     .line 1555
!     const-string/jumbo v0, "setContentDescription"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setCharSequence(ILjava/lang/String;Ljava/lang/CharSequence;)V
  
+     .line 1556
      return-void
  .end method
  
***************
*** 941,950 ****
      .parameter "childIndex"
  
      .prologue
!     const-string v0, "setDisplayedChild"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
      return-void
  .end method
  
--- 1048,1059 ----
      .parameter "childIndex"
  
      .prologue
!     .line 1140
!     const-string/jumbo v0, "setDisplayedChild"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
+     .line 1141
      return-void
  .end method
  
***************
*** 955,960 ****
--- 1064,1070 ----
      .parameter "value"
  
      .prologue
+     .line 1466
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/4 v4, 0x7
***************
*** 973,978 ****
--- 1083,1089 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1467
      return-void
  .end method
  
***************
*** 986,991 ****
--- 1097,1103 ----
      .parameter "level"
  
      .prologue
+     .line 1329
      new-instance v0, Landroid/widget/RemoteViews$SetDrawableParameters;
  
      move-object v1, p0
***************
*** 1006,1011 ****
--- 1118,1124 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1331
      return-void
  .end method
  
***************
*** 1015,1026 ****
--- 1128,1141 ----
      .parameter "emptyViewId"
  
      .prologue
+     .line 1200
      new-instance v0, Landroid/widget/RemoteViews$SetEmptyView;
  
      invoke-direct {v0, p0, p1, p2}, Landroid/widget/RemoteViews$SetEmptyView;-><init>(Landroid/widget/RemoteViews;II)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1201
      return-void
  .end method
  
***************
*** 1031,1036 ****
--- 1146,1152 ----
      .parameter "value"
  
      .prologue
+     .line 1455
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/4 v4, 0x6
***************
*** 1049,1054 ****
--- 1165,1171 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1456
      return-void
  .end method
  
***************
*** 1058,1067 ****
      .parameter "bitmap"
  
      .prologue
!     const-string v0, "setImageBitmap"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setBitmap(ILjava/lang/String;Landroid/graphics/Bitmap;)V
  
      return-void
  .end method
  
--- 1175,1186 ----
      .parameter "bitmap"
  
      .prologue
!     .line 1190
!     const-string/jumbo v0, "setImageBitmap"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setBitmap(ILjava/lang/String;Landroid/graphics/Bitmap;)V
  
+     .line 1191
      return-void
  .end method
  
***************
*** 1071,1080 ****
      .parameter "srcId"
  
      .prologue
!     const-string v0, "setImageResource"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
      return-void
  .end method
  
--- 1190,1201 ----
      .parameter "srcId"
  
      .prologue
!     .line 1170
!     const-string/jumbo v0, "setImageResource"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
+     .line 1171
      return-void
  .end method
  
***************
*** 1084,1093 ****
      .parameter "uri"
  
      .prologue
!     const-string v0, "setImageURI"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setUri(ILjava/lang/String;Landroid/net/Uri;)V
  
      return-void
  .end method
  
--- 1205,1216 ----
      .parameter "uri"
  
      .prologue
!     .line 1180
!     const-string/jumbo v0, "setImageURI"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setUri(ILjava/lang/String;Landroid/net/Uri;)V
  
+     .line 1181
      return-void
  .end method
  
***************
*** 1098,1103 ****
--- 1221,1227 ----
      .parameter "value"
  
      .prologue
+     .line 1433
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/4 v4, 0x4
***************
*** 1116,1121 ****
--- 1240,1246 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1434
      return-void
  .end method
  
***************
*** 1126,1131 ****
--- 1251,1257 ----
      .parameter "value"
  
      .prologue
+     .line 1545
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/16 v4, 0xe
***************
*** 1142,1147 ****
--- 1268,1274 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1546
      return-void
  .end method
  
***************
*** 1150,1157 ****
--- 1277,1286 ----
      .parameter "isWidgetCollectionChild"
  
      .prologue
+     .line 1051
      iput-boolean p1, p0, Landroid/widget/RemoteViews;->mIsWidgetCollectionChild:Z
  
+     .line 1052
      return-void
  .end method
  
***************
*** 1162,1167 ****
--- 1291,1297 ----
      .parameter "value"
  
      .prologue
+     .line 1444
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/4 v4, 0x5
***************
*** 1180,1185 ****
--- 1310,1316 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1445
      return-void
  .end method
  
***************
*** 1189,1200 ****
--- 1320,1333 ----
      .parameter "fillInIntent"
  
      .prologue
+     .line 1298
      new-instance v0, Landroid/widget/RemoteViews$SetOnClickFillInIntent;
  
      invoke-direct {v0, p0, p1, p2}, Landroid/widget/RemoteViews$SetOnClickFillInIntent;-><init>(Landroid/widget/RemoteViews;ILandroid/content/Intent;)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1299
      return-void
  .end method
  
***************
*** 1204,1215 ****
--- 1337,1350 ----
      .parameter "pendingIntent"
  
      .prologue
+     .line 1259
      new-instance v0, Landroid/widget/RemoteViews$SetOnClickPendingIntent;
  
      invoke-direct {v0, p0, p1, p2}, Landroid/widget/RemoteViews$SetOnClickPendingIntent;-><init>(Landroid/widget/RemoteViews;ILandroid/app/PendingIntent;)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1260
      return-void
  .end method
  
***************
*** 1219,1230 ****
--- 1354,1367 ----
      .parameter "pendingIntentTemplate"
  
      .prologue
+     .line 1275
      new-instance v0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;
  
      invoke-direct {v0, p0, p1, p2}, Landroid/widget/RemoteViews$SetPendingIntentTemplate;-><init>(Landroid/widget/RemoteViews;ILandroid/app/PendingIntent;)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1276
      return-void
  .end method
  
***************
*** 1236,1255 ****
      .parameter "indeterminate"
  
      .prologue
!     const-string v0, "setIndeterminate"
  
      invoke-virtual {p0, p1, v0, p4}, Landroid/widget/RemoteViews;->setBoolean(ILjava/lang/String;Z)V
  
      if-nez p4, :cond_0
  
!     const-string v0, "setMax"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
!     const-string v0, "setProgress"
  
      invoke-virtual {p0, p1, v0, p3}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
      :cond_0
      return-void
  .end method
--- 1373,1397 ----
      .parameter "indeterminate"
  
      .prologue
!     .line 1238
!     const-string/jumbo v0, "setIndeterminate"
  
      invoke-virtual {p0, p1, v0, p4}, Landroid/widget/RemoteViews;->setBoolean(ILjava/lang/String;Z)V
  
+     .line 1239
      if-nez p4, :cond_0
  
!     .line 1240
!     const-string/jumbo v0, "setMax"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
!     .line 1241
!     const-string/jumbo v0, "setProgress"
  
      invoke-virtual {p0, p1, v0, p3}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
+     .line 1243
      :cond_0
      return-void
  .end method
***************
*** 1260,1269 ****
      .parameter "offset"
  
      .prologue
!     const-string v0, "smoothScrollByOffset"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
      return-void
  .end method
  
--- 1402,1413 ----
      .parameter "offset"
  
      .prologue
!     .line 1389
!     const-string/jumbo v0, "smoothScrollByOffset"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
+     .line 1390
      return-void
  .end method
  
***************
*** 1276,1283 ****
--- 1420,1429 ----
      .end annotation
  
      .prologue
+     .line 1357
      invoke-virtual {p0, p2, p3}, Landroid/widget/RemoteViews;->setRemoteAdapter(ILandroid/content/Intent;)V
  
+     .line 1358
      return-void
  .end method
  
***************
*** 1287,1298 ****
--- 1433,1446 ----
      .parameter "intent"
  
      .prologue
+     .line 1369
      new-instance v0, Landroid/widget/RemoteViews$SetRemoteViewsAdapterIntent;
  
      invoke-direct {v0, p0, p1, p2}, Landroid/widget/RemoteViews$SetRemoteViewsAdapterIntent;-><init>(Landroid/widget/RemoteViews;ILandroid/content/Intent;)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1370
      return-void
  .end method
  
***************
*** 1302,1311 ****
      .parameter "position"
  
      .prologue
!     const-string v0, "smoothScrollToPosition"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
      return-void
  .end method
  
--- 1450,1461 ----
      .parameter "position"
  
      .prologue
!     .line 1379
!     const-string/jumbo v0, "smoothScrollToPosition"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
+     .line 1380
      return-void
  .end method
  
***************
*** 1316,1321 ****
--- 1466,1472 ----
      .parameter "value"
  
      .prologue
+     .line 1422
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/4 v4, 0x3
***************
*** 1334,1339 ****
--- 1485,1491 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1423
      return-void
  .end method
  
***************
*** 1344,1349 ****
--- 1496,1502 ----
      .parameter "value"
  
      .prologue
+     .line 1488
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/16 v4, 0x9
***************
*** 1360,1365 ****
--- 1513,1519 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1489
      return-void
  .end method
  
***************
*** 1369,1378 ****
      .parameter "color"
  
      .prologue
!     const-string v0, "setTextColor"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
      return-void
  .end method
  
--- 1523,1534 ----
      .parameter "color"
  
      .prologue
!     .line 1341
!     const-string/jumbo v0, "setTextColor"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
+     .line 1342
      return-void
  .end method
  
***************
*** 1382,1391 ****
      .parameter "text"
  
      .prologue
!     const-string v0, "setText"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setCharSequence(ILjava/lang/String;Ljava/lang/CharSequence;)V
  
      return-void
  .end method
  
--- 1538,1549 ----
      .parameter "text"
  
      .prologue
!     .line 1160
!     const-string/jumbo v0, "setText"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setCharSequence(ILjava/lang/String;Ljava/lang/CharSequence;)V
  
+     .line 1161
      return-void
  .end method
  
***************
*** 1396,1401 ****
--- 1554,1560 ----
      .parameter "value"
  
      .prologue
+     .line 1510
      new-instance v0, Landroid/widget/RemoteViews$ReflectionAction;
  
      const/16 v4, 0xb
***************
*** 1412,1417 ****
--- 1571,1577 ----
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1511
      return-void
  .end method
  
***************
*** 1421,1430 ****
      .parameter "visibility"
  
      .prologue
!     const-string v0, "setVisibility"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
      return-void
  .end method
  
--- 1581,1592 ----
      .parameter "visibility"
  
      .prologue
!     .line 1150
!     const-string/jumbo v0, "setVisibility"
  
      invoke-virtual {p0, p1, v0, p2}, Landroid/widget/RemoteViews;->setInt(ILjava/lang/String;I)V
  
+     .line 1151
      return-void
  .end method
  
***************
*** 1433,1446 ****
      .parameter "viewId"
  
      .prologue
      new-instance v0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;
  
!     const-string v1, "showNext"
  
      invoke-direct {v0, p0, p1, v1}, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;-><init>(Landroid/widget/RemoteViews;ILjava/lang/String;)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
      return-void
  .end method
  
--- 1595,1610 ----
      .parameter "viewId"
  
      .prologue
+     .line 1121
      new-instance v0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;
  
!     const-string/jumbo v1, "showNext"
  
      invoke-direct {v0, p0, p1, v1}, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;-><init>(Landroid/widget/RemoteViews;ILjava/lang/String;)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1122
      return-void
  .end method
  
***************
*** 1449,1462 ****
      .parameter "viewId"
  
      .prologue
      new-instance v0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;
  
!     const-string v1, "showPrevious"
  
      invoke-direct {v0, p0, p1, v1}, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;-><init>(Landroid/widget/RemoteViews;ILjava/lang/String;)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
      return-void
  .end method
  
--- 1613,1628 ----
      .parameter "viewId"
  
      .prologue
+     .line 1130
      new-instance v0, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;
  
!     const-string/jumbo v1, "showPrevious"
  
      invoke-direct {v0, p0, p1, v1}, Landroid/widget/RemoteViews$ReflectionActionWithoutParams;-><init>(Landroid/widget/RemoteViews;ILjava/lang/String;)V
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews;->addAction(Landroid/widget/RemoteViews$Action;)V
  
+     .line 1131
      return-void
  .end method
  
***************
*** 1468,1481 ****
--- 1634,1650 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 1642
      iget-object v3, p0, Landroid/widget/RemoteViews;->mPackage:Ljava/lang/String;
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 1643
      iget v3, p0, Landroid/widget/RemoteViews;->mLayoutId:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 1644
      iget-boolean v3, p0, Landroid/widget/RemoteViews;->mIsWidgetCollectionChild:Z
  
      if-eqz v3, :cond_0
***************
*** 1485,1510 ****
--- 1654,1684 ----
      :goto_0
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 1646
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      if-eqz v3, :cond_1
  
+     .line 1647
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      invoke-virtual {v3}, Ljava/util/ArrayList;->size()I
  
      move-result v1
  
+     .line 1651
      .local v1, count:I
      :goto_1
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 1652
      const/4 v2, 0x0
  
      .local v2, i:I
      :goto_2
      if-ge v2, v1, :cond_2
  
+     .line 1653
      iget-object v3, p0, Landroid/widget/RemoteViews;->mActions:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v2}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1513,1521 ****
--- 1687,1697 ----
  
      check-cast v0, Landroid/widget/RemoteViews$Action;
  
+     .line 1654
      .local v0, a:Landroid/widget/RemoteViews$Action;
      invoke-virtual {v0, p1, v4}, Landroid/widget/RemoteViews$Action;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 1652
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_2
***************
*** 1526,1539 ****
--- 1702,1718 ----
      :cond_0
      move v3, v4
  
+     .line 1644
      goto :goto_0
  
+     .line 1649
      :cond_1
      const/4 v1, 0x0
  
      .restart local v1       #count:I
      goto :goto_1
  
+     .line 1656
      .restart local v2       #i:I
      :cond_2
      return-void
