*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1.smali	2012-03-23 19:35:50.438506631 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1.smali.original	2012-03-23 19:34:44.864310886 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 155
      iput-object p1, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->this$0:Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection;
  
      iput-object p2, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->val$adapter:Landroid/widget/RemoteViewsAdapter;
***************
*** 45,50 ****
--- 46,52 ----
      .locals 5
  
      .prologue
+     .line 158
      iget-object v2, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->val$adapter:Landroid/widget/RemoteViewsAdapter;
  
      #getter for: Landroid/widget/RemoteViewsAdapter;->mNotifyDataSetChangedAfterOnServiceConnected:Z
***************
*** 54,70 ****
--- 56,75 ----
  
      if-eqz v2, :cond_0
  
+     .line 160
      iget-object v2, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->val$adapter:Landroid/widget/RemoteViewsAdapter;
  
      #calls: Landroid/widget/RemoteViewsAdapter;->onNotifyDataSetChanged()V
      invoke-static {v2}, Landroid/widget/RemoteViewsAdapter;->access$100(Landroid/widget/RemoteViewsAdapter;)V
  
+     .line 204
      :goto_0
      iget-object v2, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->val$adapter:Landroid/widget/RemoteViewsAdapter;
  
      #calls: Landroid/widget/RemoteViewsAdapter;->enqueueDeferredUnbindServiceMessage()V
      invoke-static {v2}, Landroid/widget/RemoteViewsAdapter;->access$700(Landroid/widget/RemoteViewsAdapter;)V
  
+     .line 205
      iget-object v2, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->this$0:Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection;
  
      const/4 v3, 0x1
***************
*** 72,77 ****
--- 77,83 ----
      #setter for: Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection;->mIsConnected:Z
      invoke-static {v2, v3}, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection;->access$802(Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection;Z)Z
  
+     .line 206
      iget-object v2, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->this$0:Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection;
  
      const/4 v3, 0x0
***************
*** 79,87 ****
--- 85,95 ----
      #setter for: Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection;->mIsConnecting:Z
      invoke-static {v2, v3}, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection;->access$902(Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection;Z)Z
  
+     .line 207
      :goto_1
      return-void
  
+     .line 162
      :cond_0
      iget-object v2, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->val$adapter:Landroid/widget/RemoteViewsAdapter;
  
***************
*** 94,99 ****
--- 102,108 ----
  
      move-result-object v1
  
+     .line 165
      .local v1, factory:Lcom/android/internal/widget/IRemoteViewsFactory;
      :try_start_0
      invoke-interface {v1}, Lcom/android/internal/widget/IRemoteViewsFactory;->isCreated()Z
***************
*** 102,112 ****
--- 111,123 ----
  
      if-nez v2, :cond_1
  
+     .line 168
      invoke-interface {v1}, Lcom/android/internal/widget/IRemoteViewsFactory;->onDataSetChanged()V
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/lang/RuntimeException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 184
      :cond_1
      :goto_2
      iget-object v2, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->val$adapter:Landroid/widget/RemoteViewsAdapter;
***************
*** 114,119 ****
--- 125,131 ----
      #calls: Landroid/widget/RemoteViewsAdapter;->updateTemporaryMetaData()V
      invoke-static {v2}, Landroid/widget/RemoteViewsAdapter;->access$300(Landroid/widget/RemoteViewsAdapter;)V
  
+     .line 187
      iget-object v2, p0, Landroid/widget/RemoteViewsAdapter$RemoteViewsAdapterServiceConnection$1;->val$adapter:Landroid/widget/RemoteViewsAdapter;
  
      #getter for: Landroid/widget/RemoteViewsAdapter;->mMainQueue:Landroid/os/Handler;
***************
*** 129,137 ****
--- 141,151 ----
  
      goto :goto_0
  
+     .line 170
      :catch_0
      move-exception v0
  
+     .line 171
      .local v0, e:Landroid/os/RemoteException;
      const-string v2, "RemoteViewsAdapter"
  
***************
*** 161,170 ****
--- 175,186 ----
  
      goto :goto_1
  
+     .line 177
      .end local v0           #e:Landroid/os/RemoteException;
      :catch_1
      move-exception v0
  
+     .line 178
      .local v0, e:Ljava/lang/RuntimeException;
      const-string v2, "RemoteViewsAdapter"
  
