*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViewsAdapter$5.smali	2012-03-23 19:35:52.977552930 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViewsAdapter$5.smali.original	2012-03-23 19:34:44.884311252 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1095
      iput-object p1, p0, Landroid/widget/RemoteViewsAdapter$5;->this$0:Landroid/widget/RemoteViewsAdapter;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,49 ****
--- 41,52 ----
      .locals 1
  
      .prologue
+     .line 1098
      iget-object v0, p0, Landroid/widget/RemoteViewsAdapter$5;->this$0:Landroid/widget/RemoteViewsAdapter;
  
      #calls: Landroid/widget/RemoteViewsAdapter;->onNotifyDataSetChanged()V
      invoke-static {v0}, Landroid/widget/RemoteViewsAdapter;->access$100(Landroid/widget/RemoteViewsAdapter;)V
  
+     .line 1099
      return-void
  .end method
