*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViewsAdapter$3.smali	2012-03-23 19:35:51.488525778 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViewsAdapter$3.smali.original	2012-03-23 19:34:44.872311032 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .parameter
  
      .prologue
+     .line 867
      iput-object p1, p0, Landroid/widget/RemoteViewsAdapter$3;->this$0:Landroid/widget/RemoteViewsAdapter;
  
      iput p2, p0, Landroid/widget/RemoteViewsAdapter$3;->val$position:I
***************
*** 55,60 ****
--- 56,62 ----
      .locals 4
  
      .prologue
+     .line 870
      iget-object v0, p0, Landroid/widget/RemoteViewsAdapter$3;->this$0:Landroid/widget/RemoteViewsAdapter;
  
      #getter for: Landroid/widget/RemoteViewsAdapter;->mRequestedViews:Landroid/widget/RemoteViewsAdapter$RemoteViewsFrameLayoutRefSet;
***************
*** 70,74 ****
--- 72,77 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/widget/RemoteViewsAdapter$RemoteViewsFrameLayoutRefSet;->notifyOnRemoteViewsLoaded(ILandroid/widget/RemoteViews;I)V
  
+     .line 871
      return-void
  .end method
