*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViews$SetPendingIntentTemplate.smali	2012-03-23 19:35:47.872459840 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/RemoteViews$SetPendingIntentTemplate.smali.original	2012-03-23 19:34:44.844310522 -0500
***************
*** 34,49 ****
--- 34,53 ----
      .parameter "pendingIntentTemplate"
  
      .prologue
+     .line 281
      iput-object p1, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->this$0:Landroid/widget/RemoteViews;
  
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews$Action;-><init>(Landroid/widget/RemoteViews$1;)V
  
+     .line 282
      iput p2, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->viewId:I
  
+     .line 283
      iput-object p3, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->pendingIntentTemplate:Landroid/app/PendingIntent;
  
+     .line 284
      return-void
  .end method
  
***************
*** 53,76 ****
--- 57,84 ----
      .parameter "parcel"
  
      .prologue
+     .line 286
      iput-object p1, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->this$0:Landroid/widget/RemoteViews;
  
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/widget/RemoteViews$Action;-><init>(Landroid/widget/RemoteViews$1;)V
  
+     .line 287
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->viewId:I
  
+     .line 288
      invoke-static {p2}, Landroid/app/PendingIntent;->readPendingIntentOrNullFromParcel(Landroid/os/Parcel;)Landroid/app/PendingIntent;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->pendingIntentTemplate:Landroid/app/PendingIntent;
  
+     .line 289
      return-void
  .end method
  
***************
*** 82,99 ****
--- 90,111 ----
      .parameter "rootParent"
  
      .prologue
+     .line 299
      iget v3, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->viewId:I
  
      invoke-virtual {p1, v3}, Landroid/view/View;->findViewById(I)Landroid/view/View;
  
      move-result-object v2
  
+     .line 300
      .local v2, target:Landroid/view/View;
      if-nez v2, :cond_0
  
+     .line 355
      :goto_0
      return-void
  
+     .line 303
      :cond_0
      instance-of v3, v2, Landroid/widget/AdapterView;
  
***************
*** 101,122 ****
--- 113,139 ----
  
      move-object v0, v2
  
+     .line 304
      check-cast v0, Landroid/widget/AdapterView;
  
+     .line 306
      .local v0, av:Landroid/widget/AdapterView;,"Landroid/widget/AdapterView<*>;"
      new-instance v1, Landroid/widget/RemoteViews$SetPendingIntentTemplate$1;
  
      invoke-direct {v1, p0}, Landroid/widget/RemoteViews$SetPendingIntentTemplate$1;-><init>(Landroid/widget/RemoteViews$SetPendingIntentTemplate;)V
  
+     .line 348
      .local v1, listener:Landroid/widget/AdapterView$OnItemClickListener;
      invoke-virtual {v0, v1}, Landroid/widget/AdapterView;->setOnItemClickListener(Landroid/widget/AdapterView$OnItemClickListener;)V
  
+     .line 349
      iget-object v3, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->pendingIntentTemplate:Landroid/app/PendingIntent;
  
      invoke-virtual {v0, v3}, Landroid/widget/AdapterView;->setTag(Ljava/lang/Object;)V
  
      goto :goto_0
  
+     .line 351
      .end local v0           #av:Landroid/widget/AdapterView;,"Landroid/widget/AdapterView<*>;"
      .end local v1           #listener:Landroid/widget/AdapterView$OnItemClickListener;
      :cond_1
***************
*** 159,177 ****
--- 176,198 ----
      .parameter "flags"
  
      .prologue
+     .line 292
      const/16 v0, 0x8
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 293
      iget v0, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->viewId:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 294
      iget-object v0, p0, Landroid/widget/RemoteViews$SetPendingIntentTemplate;->pendingIntentTemplate:Landroid/app/PendingIntent;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, p1, v1}, Landroid/app/PendingIntent;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 295
      return-void
  .end method
