*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ResourceCursorTreeAdapter.smali	2012-03-23 19:35:50.583509274 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ResourceCursorTreeAdapter.smali.original	2012-03-23 19:34:44.865310905 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter "childLayout"
  
      .prologue
+     .line 94
      move-object v0, p0
  
      move-object v1, p1
***************
*** 40,45 ****
--- 41,47 ----
  
      invoke-direct/range {v0 .. v6}, Landroid/widget/ResourceCursorTreeAdapter;-><init>(Landroid/content/Context;Landroid/database/Cursor;IIII)V
  
+     .line 95
      return-void
  .end method
  
***************
*** 52,57 ****
--- 54,60 ----
      .parameter "childLayout"
  
      .prologue
+     .line 78
      move-object v0, p0
  
      move-object v1, p1
***************
*** 68,73 ****
--- 71,77 ----
  
      invoke-direct/range {v0 .. v6}, Landroid/widget/ResourceCursorTreeAdapter;-><init>(Landroid/content/Context;Landroid/database/Cursor;IIII)V
  
+     .line 79
      return-void
  .end method
  
***************
*** 81,96 ****
--- 85,106 ----
      .parameter "lastChildLayout"
  
      .prologue
+     .line 53
      invoke-direct {p0, p2, p1}, Landroid/widget/CursorTreeAdapter;-><init>(Landroid/database/Cursor;Landroid/content/Context;)V
  
+     .line 55
      iput p3, p0, Landroid/widget/ResourceCursorTreeAdapter;->mCollapsedGroupLayout:I
  
+     .line 56
      iput p4, p0, Landroid/widget/ResourceCursorTreeAdapter;->mExpandedGroupLayout:I
  
+     .line 57
      iput p5, p0, Landroid/widget/ResourceCursorTreeAdapter;->mChildLayout:I
  
+     .line 58
      iput p6, p0, Landroid/widget/ResourceCursorTreeAdapter;->mLastChildLayout:I
  
+     .line 60
      const-string v0, "layout_inflater"
  
      invoke-virtual {p1, v0}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 101,106 ****
--- 111,117 ----
  
      iput-object v0, p0, Landroid/widget/ResourceCursorTreeAdapter;->mInflater:Landroid/view/LayoutInflater;
  
+     .line 61
      return-void
  .end method
  
***************
*** 114,119 ****
--- 125,131 ----
      .parameter "parent"
  
      .prologue
+     .line 100
      iget-object v1, p0, Landroid/widget/ResourceCursorTreeAdapter;->mInflater:Landroid/view/LayoutInflater;
  
      if-eqz p3, :cond_0
***************
*** 143,148 ****
--- 155,161 ----
      .parameter "parent"
  
      .prologue
+     .line 105
      iget-object v1, p0, Landroid/widget/ResourceCursorTreeAdapter;->mInflater:Landroid/view/LayoutInflater;
  
      if-eqz p3, :cond_0
