*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/SearchView$9.smali	2012-03-23 19:35:55.396597041 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/SearchView$9.smali.original	2012-03-23 19:34:44.904311616 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1245
      iput-object p1, p0, Landroid/widget/SearchView$9;->this$0:Landroid/widget/SearchView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 53,58 ****
--- 54,60 ----
      .end annotation
  
      .prologue
+     .line 1252
      .local p1, parent:Landroid/widget/AdapterView;,"Landroid/widget/AdapterView<*>;"
      iget-object v0, p0, Landroid/widget/SearchView$9;->this$0:Landroid/widget/SearchView;
  
***************
*** 63,67 ****
--- 65,70 ----
      #calls: Landroid/widget/SearchView;->onItemClicked(IILjava/lang/String;)Z
      invoke-static {v0, p3, v1, v2}, Landroid/widget/SearchView;->access$1800(Landroid/widget/SearchView;IILjava/lang/String;)Z
  
+     .line 1253
      return-void
  .end method
