*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/SearchView$10.smali	2012-03-23 19:35:59.717675835 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/SearchView$10.smali.original	2012-03-23 19:34:44.935312182 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1256
      iput-object p1, p0, Landroid/widget/SearchView$10;->this$0:Landroid/widget/SearchView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 53,64 ****
--- 54,67 ----
      .end annotation
  
      .prologue
+     .line 1263
      .local p1, parent:Landroid/widget/AdapterView;,"Landroid/widget/AdapterView<*>;"
      iget-object v0, p0, Landroid/widget/SearchView$10;->this$0:Landroid/widget/SearchView;
  
      #calls: Landroid/widget/SearchView;->onItemSelected(I)Z
      invoke-static {v0, p3}, Landroid/widget/SearchView;->access$1900(Landroid/widget/SearchView;I)Z
  
+     .line 1264
      return-void
  .end method
  
***************
*** 74,79 ****
--- 77,83 ----
      .end annotation
  
      .prologue
+     .line 1272
      .local p1, parent:Landroid/widget/AdapterView;,"Landroid/widget/AdapterView<*>;"
      return-void
  .end method
