*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/SeekBar.smali	2012-03-23 19:35:57.537636083 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/SeekBar.smali.original	2012-03-23 19:34:44.921311926 -0500
***************
*** 21,30 ****
--- 21,32 ----
      .parameter "context"
  
      .prologue
+     .line 73
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/widget/SeekBar;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 74
      return-void
  .end method
  
***************
*** 34,43 ****
--- 36,47 ----
      .parameter "attrs"
  
      .prologue
+     .line 77
      const v0, 0x101007b
  
      invoke-direct {p0, p1, p2, v0}, Landroid/widget/SeekBar;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 78
      return-void
  .end method
  
***************
*** 48,55 ****
--- 52,61 ----
      .parameter "defStyle"
  
      .prologue
+     .line 81
      invoke-direct {p0, p1, p2, p3}, Landroid/widget/AbsSeekBar;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 82
      return-void
  .end method
  
***************
*** 61,72 ****
--- 67,81 ----
      .parameter "fromUser"
  
      .prologue
+     .line 86
      invoke-super {p0, p1, p2}, Landroid/widget/AbsSeekBar;->onProgressRefresh(FZ)V
  
+     .line 88
      iget-object v0, p0, Landroid/widget/SeekBar;->mOnSeekBarChangeListener:Landroid/widget/SeekBar$OnSeekBarChangeListener;
  
      if-eqz v0, :cond_0
  
+     .line 89
      iget-object v0, p0, Landroid/widget/SeekBar;->mOnSeekBarChangeListener:Landroid/widget/SeekBar$OnSeekBarChangeListener;
  
      invoke-virtual {p0}, Landroid/widget/SeekBar;->getProgress()I
***************
*** 75,80 ****
--- 84,90 ----
  
      invoke-interface {v0, p0, v1, p2}, Landroid/widget/SeekBar$OnSeekBarChangeListener;->onProgressChanged(Landroid/widget/SeekBar;IZ)V
  
+     .line 91
      :cond_0
      return-void
  .end method
***************
*** 83,98 ****
--- 93,112 ----
      .locals 1
  
      .prologue
+     .line 107
      invoke-super {p0}, Landroid/widget/AbsSeekBar;->onStartTrackingTouch()V
  
+     .line 108
      iget-object v0, p0, Landroid/widget/SeekBar;->mOnSeekBarChangeListener:Landroid/widget/SeekBar$OnSeekBarChangeListener;
  
      if-eqz v0, :cond_0
  
+     .line 109
      iget-object v0, p0, Landroid/widget/SeekBar;->mOnSeekBarChangeListener:Landroid/widget/SeekBar$OnSeekBarChangeListener;
  
      invoke-interface {v0, p0}, Landroid/widget/SeekBar$OnSeekBarChangeListener;->onStartTrackingTouch(Landroid/widget/SeekBar;)V
  
+     .line 111
      :cond_0
      return-void
  .end method
***************
*** 101,116 ****
--- 115,134 ----
      .locals 1
  
      .prologue
+     .line 115
      invoke-super {p0}, Landroid/widget/AbsSeekBar;->onStopTrackingTouch()V
  
+     .line 116
      iget-object v0, p0, Landroid/widget/SeekBar;->mOnSeekBarChangeListener:Landroid/widget/SeekBar$OnSeekBarChangeListener;
  
      if-eqz v0, :cond_0
  
+     .line 117
      iget-object v0, p0, Landroid/widget/SeekBar;->mOnSeekBarChangeListener:Landroid/widget/SeekBar$OnSeekBarChangeListener;
  
      invoke-interface {v0, p0}, Landroid/widget/SeekBar$OnSeekBarChangeListener;->onStopTrackingTouch(Landroid/widget/SeekBar;)V
  
+     .line 119
      :cond_0
      return-void
  .end method
***************
*** 120,126 ****
--- 138,146 ----
      .parameter "l"
  
      .prologue
+     .line 102
      iput-object p1, p0, Landroid/widget/SeekBar;->mOnSeekBarChangeListener:Landroid/widget/SeekBar$OnSeekBarChangeListener;
  
+     .line 103
      return-void
  .end method
