*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ShareActionProvider$ShareAcitivityChooserModelPolicy.smali	2012-03-23 19:35:49.711493374 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ShareActionProvider$ShareAcitivityChooserModelPolicy.smali.original	2012-03-23 19:34:44.858310778 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 306
      iput-object p1, p0, Landroid/widget/ShareActionProvider$ShareAcitivityChooserModelPolicy;->this$0:Landroid/widget/ShareActionProvider;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .parameter "x1"
  
      .prologue
+     .line 306
      invoke-direct {p0, p1}, Landroid/widget/ShareActionProvider$ShareAcitivityChooserModelPolicy;-><init>(Landroid/widget/ShareActionProvider;)V
  
      return-void
***************
*** 53,58 ****
--- 55,61 ----
      .parameter "intent"
  
      .prologue
+     .line 309
      iget-object v0, p0, Landroid/widget/ShareActionProvider$ShareAcitivityChooserModelPolicy;->this$0:Landroid/widget/ShareActionProvider;
  
      #getter for: Landroid/widget/ShareActionProvider;->mOnShareTargetSelectedListener:Landroid/widget/ShareActionProvider$OnShareTargetSelectedListener;
***************
*** 62,67 ****
--- 65,71 ----
  
      if-eqz v0, :cond_0
  
+     .line 310
      iget-object v0, p0, Landroid/widget/ShareActionProvider$ShareAcitivityChooserModelPolicy;->this$0:Landroid/widget/ShareActionProvider;
  
      #getter for: Landroid/widget/ShareActionProvider;->mOnShareTargetSelectedListener:Landroid/widget/ShareActionProvider$OnShareTargetSelectedListener;
***************
*** 75,80 ****
--- 79,85 ----
  
      move-result v0
  
+     .line 313
      :goto_0
      return v0
  
