*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/SpellChecker$SpellParser.smali	2012-03-23 19:35:48.760476031 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/SpellChecker$SpellParser.smali.original	2012-03-23 19:34:44.851310650 -0500
***************
*** 26,35 ****
--- 26,37 ----
      .parameter
  
      .prologue
+     .line 399
      iput-object p1, p0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 400
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
***************
*** 45,50 ****
--- 47,53 ----
      .parameter "x1"
  
      .prologue
+     .line 399
      invoke-direct {p0, p1}, Landroid/widget/SpellChecker$SpellParser;-><init>(Landroid/widget/SpellChecker;)V
  
      return-void
***************
*** 66,74 ****
--- 69,79 ----
      .end annotation
  
      .prologue
+     .line 523
      .local p3, spans:[Ljava/lang/Object;,"[TT;"
      array-length v2, p3
  
+     .line 524
      .local v2, length:I
      const/4 v1, 0x0
  
***************
*** 76,109 ****
--- 81,122 ----
      :goto_0
      if-ge v1, v2, :cond_2
  
+     .line 525
      aget-object v3, p3, v1
  
+     .line 526
      .local v3, span:Ljava/lang/Object;,"TT;"
      invoke-interface {p1, v3}, Landroid/text/Editable;->getSpanStart(Ljava/lang/Object;)I
  
      move-result v4
  
+     .line 527
      .local v4, start:I
      if-le v4, p2, :cond_1
  
+     .line 524
      :cond_0
      :goto_1
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 528
      :cond_1
      invoke-interface {p1, v3}, Landroid/text/Editable;->getSpanEnd(Ljava/lang/Object;)I
  
      move-result v0
  
+     .line 529
      .local v0, end:I
      if-lt v0, p2, :cond_0
  
+     .line 530
      invoke-interface {p1, v3}, Landroid/text/Editable;->removeSpan(Ljava/lang/Object;)V
  
      goto :goto_1
  
+     .line 532
      .end local v0           #end:I
      .end local v3           #span:Ljava/lang/Object;,"TT;"
      .end local v4           #start:I
***************
*** 117,122 ****
--- 130,136 ----
      .locals 2
  
      .prologue
+     .line 408
      iget-object v0, p0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
  
      #getter for: Landroid/widget/SpellChecker;->mTextView:Landroid/widget/TextView;
***************
*** 134,139 ****
--- 148,154 ----
  
      invoke-interface {v0, v1}, Landroid/text/Editable;->removeSpan(Ljava/lang/Object;)V
  
+     .line 409
      return-void
  .end method
  
***************
*** 143,148 ****
--- 158,164 ----
      .parameter "end"
  
      .prologue
+     .line 403
      iget-object v0, p0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
  
      #getter for: Landroid/widget/SpellChecker;->mTextView:Landroid/widget/TextView;
***************
*** 162,167 ****
--- 178,184 ----
  
      invoke-interface {v0, v1, p1, p2, v2}, Landroid/text/Editable;->setSpan(Ljava/lang/Object;III)V
  
+     .line 405
      return-void
  .end method
  
***************
*** 169,174 ****
--- 186,192 ----
      .locals 2
  
      .prologue
+     .line 412
      iget-object v0, p0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
  
      #getter for: Landroid/widget/SpellChecker;->mTextView:Landroid/widget/TextView;
***************
*** 205,210 ****
--- 223,229 ----
      .locals 21
  
      .prologue
+     .line 416
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
***************
*** 222,227 ****
--- 241,247 ----
  
      check-cast v4, Landroid/text/Editable;
  
+     .line 418
      .local v4, editable:Landroid/text/Editable;
      move-object/from16 v0, p0
  
***************
*** 235,240 ****
--- 255,261 ----
  
      move-result v12
  
+     .line 419
      .local v12, start:I
      move-object/from16 v0, p0
  
***************
*** 248,253 ****
--- 269,275 ----
  
      move-result v5
  
+     .line 421
      .local v5, end:I
      add-int/lit16 v0, v12, 0x15e
  
***************
*** 259,264 ****
--- 281,287 ----
  
      move-result v16
  
+     .line 422
      .local v16, wordIteratorWindowEnd:I
      move-object/from16 v0, p0
  
***************
*** 277,282 ****
--- 300,306 ----
  
      invoke-virtual {v0, v4, v12, v1}, Landroid/text/method/WordIterator;->setCharSequence(Ljava/lang/CharSequence;II)V
  
+     .line 425
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
***************
*** 294,299 ****
--- 318,324 ----
  
      move-result v17
  
+     .line 427
      .local v17, wordStart:I
      const/16 v18, -0x1
  
***************
*** 303,308 ****
--- 328,334 ----
  
      if-ne v0, v1, :cond_1
  
+     .line 428
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
***************
*** 320,325 ****
--- 346,352 ----
  
      move-result v15
  
+     .line 429
      .local v15, wordEnd:I
      const/16 v18, -0x1
  
***************
*** 327,332 ****
--- 354,360 ----
  
      if-eq v15, v0, :cond_0
  
+     .line 430
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
***************
*** 344,349 ****
--- 372,378 ----
  
      move-result v17
  
+     .line 435
      :cond_0
      :goto_0
      const/16 v18, -0x1
***************
*** 352,357 ****
--- 381,387 ----
  
      if-ne v15, v0, :cond_2
  
+     .line 436
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->mRange:Ljava/lang/Object;
***************
*** 362,370 ****
--- 392,402 ----
  
      invoke-interface {v4, v0}, Landroid/text/Editable;->removeSpan(Ljava/lang/Object;)V
  
+     .line 520
      :goto_1
      return-void
  
+     .line 433
      .end local v15           #wordEnd:I
      :cond_1
      move-object/from16 v0, p0
***************
*** 389,394 ****
--- 421,427 ----
      .restart local v15       #wordEnd:I
      goto :goto_0
  
+     .line 442
      :cond_2
      add-int/lit8 v18, v12, -0x1
  
***************
*** 408,413 ****
--- 441,447 ----
  
      check-cast v11, [Landroid/text/style/SpellCheckSpan;
  
+     .line 444
      .local v11, spellCheckSpans:[Landroid/text/style/SpellCheckSpan;
      add-int/lit8 v18, v12, -0x1
  
***************
*** 427,462 ****
--- 461,504 ----
  
      check-cast v13, [Landroid/text/style/SuggestionSpan;
  
+     .line 447
      .local v13, suggestionSpans:[Landroid/text/style/SuggestionSpan;
      const/4 v14, 0x0
  
+     .line 448
      .local v14, wordCount:I
      const/4 v8, 0x0
  
+     .line 450
      .local v8, scheduleOtherSpellCheck:Z
      :cond_3
      move/from16 v0, v17
  
      if-gt v0, v5, :cond_4
  
+     .line 451
      if-lt v15, v12, :cond_b
  
      move/from16 v0, v17
  
      if-le v15, v0, :cond_b
  
+     .line 452
      const/16 v18, 0x32
  
      move/from16 v0, v18
  
      if-lt v14, v0, :cond_5
  
+     .line 453
      const/4 v8, 0x1
  
+     .line 513
      :cond_4
      :goto_2
      if-eqz v8, :cond_10
  
+     .line 514
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->mRange:Ljava/lang/Object;
***************
*** 473,478 ****
--- 515,521 ----
  
      invoke-interface {v4, v0, v1, v5, v2}, Landroid/text/Editable;->setSpan(Ljava/lang/Object;III)V
  
+     .line 519
      :goto_3
      move-object/from16 v0, p0
  
***************
*** 485,490 ****
--- 528,534 ----
  
      goto :goto_1
  
+     .line 459
      :cond_5
      move/from16 v0, v17
  
***************
*** 492,505 ****
--- 536,552 ----
  
      if-le v15, v12, :cond_6
  
+     .line 460
      move-object/from16 v0, p0
  
      invoke-direct {v0, v4, v12, v11}, Landroid/widget/SpellChecker$SpellParser;->removeSpansAt(Landroid/text/Editable;I[Ljava/lang/Object;)V
  
+     .line 461
      move-object/from16 v0, p0
  
      invoke-direct {v0, v4, v12, v13}, Landroid/widget/SpellChecker$SpellParser;->removeSpansAt(Landroid/text/Editable;I[Ljava/lang/Object;)V
  
+     .line 464
      :cond_6
      move/from16 v0, v17
  
***************
*** 507,526 ****
--- 554,578 ----
  
      if-le v15, v5, :cond_7
  
+     .line 465
      move-object/from16 v0, p0
  
      invoke-direct {v0, v4, v5, v11}, Landroid/widget/SpellChecker$SpellParser;->removeSpansAt(Landroid/text/Editable;I[Ljava/lang/Object;)V
  
+     .line 466
      move-object/from16 v0, p0
  
      invoke-direct {v0, v4, v5, v13}, Landroid/widget/SpellChecker$SpellParser;->removeSpansAt(Landroid/text/Editable;I[Ljava/lang/Object;)V
  
+     .line 470
      :cond_7
      const/4 v3, 0x1
  
+     .line 471
      .local v3, createSpellCheckSpan:Z
      if-ne v15, v12, :cond_8
  
+     .line 472
      const/4 v6, 0x0
  
      .local v6, i:I
***************
*** 533,538 ****
--- 585,591 ----
  
      if-ge v6, v0, :cond_8
  
+     .line 473
      aget-object v18, v11, v6
  
      move-object/from16 v0, v18
***************
*** 541,551 ****
--- 594,607 ----
  
      move-result v9
  
+     .line 474
      .local v9, spanEnd:I
      if-ne v9, v12, :cond_e
  
+     .line 475
      const/4 v3, 0x0
  
+     .line 481
      .end local v6           #i:I
      .end local v9           #spanEnd:I
      :cond_8
***************
*** 553,558 ****
--- 609,615 ----
  
      if-ne v0, v5, :cond_9
  
+     .line 482
      const/4 v6, 0x0
  
      .restart local v6       #i:I
***************
*** 565,570 ****
--- 622,628 ----
  
      if-ge v6, v0, :cond_9
  
+     .line 483
      aget-object v18, v11, v6
  
      move-object/from16 v0, v18
***************
*** 573,588 ****
--- 631,650 ----
  
      move-result v10
  
+     .line 484
      .local v10, spanStart:I
      if-ne v10, v5, :cond_f
  
+     .line 485
      const/4 v3, 0x0
  
+     .line 491
      .end local v6           #i:I
      .end local v10           #spanStart:I
      :cond_9
      if-eqz v3, :cond_a
  
+     .line 492
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
***************
*** 596,608 ****
--- 658,673 ----
      #calls: Landroid/widget/SpellChecker;->addSpellCheckSpan(Landroid/text/Editable;II)V
      invoke-static {v0, v4, v1, v15}, Landroid/widget/SpellChecker;->access$400(Landroid/widget/SpellChecker;Landroid/text/Editable;II)V
  
+     .line 494
      :cond_a
      add-int/lit8 v14, v14, 0x1
  
+     .line 498
      .end local v3           #createSpellCheckSpan:Z
      :cond_b
      move v7, v15
  
+     .line 499
      .local v7, originalWordEnd:I
      move-object/from16 v0, p0
  
***************
*** 621,626 ****
--- 686,692 ----
  
      move-result v15
  
+     .line 500
      move/from16 v0, v16
  
      if-ge v0, v5, :cond_d
***************
*** 635,640 ****
--- 701,707 ----
  
      if-lt v15, v0, :cond_d
  
+     .line 502
      :cond_c
      add-int/lit16 v0, v7, 0x15e
  
***************
*** 646,651 ****
--- 713,719 ----
  
      move-result v16
  
+     .line 503
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
***************
*** 663,668 ****
--- 731,737 ----
  
      invoke-virtual {v0, v4, v7, v1}, Landroid/text/method/WordIterator;->setCharSequence(Ljava/lang/CharSequence;II)V
  
+     .line 504
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
***************
*** 680,685 ****
--- 749,755 ----
  
      move-result v15
  
+     .line 506
      :cond_d
      const/16 v18, -0x1
  
***************
*** 687,692 ****
--- 757,763 ----
  
      if-eq v15, v0, :cond_4
  
+     .line 507
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/SpellChecker$SpellParser;->this$0:Landroid/widget/SpellChecker;
***************
*** 704,709 ****
--- 775,781 ----
  
      move-result v17
  
+     .line 508
      const/16 v18, -0x1
  
      move/from16 v0, v17
***************
*** 714,719 ****
--- 786,792 ----
  
      goto/16 :goto_2
  
+     .line 472
      .end local v7           #originalWordEnd:I
      .restart local v3       #createSpellCheckSpan:Z
      .restart local v6       #i:I
***************
*** 723,728 ****
--- 796,802 ----
  
      goto/16 :goto_4
  
+     .line 482
      .end local v9           #spanEnd:I
      .restart local v10       #spanStart:I
      :cond_f
***************
*** 730,735 ****
--- 804,810 ----
  
      goto/16 :goto_5
  
+     .line 516
      .end local v3           #createSpellCheckSpan:Z
      .end local v6           #i:I
      .end local v10           #spanStart:I
