*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Spinner$DropdownPopup$1.smali	2012-03-23 19:35:55.799604390 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Spinner$DropdownPopup$1.smali.original	2012-03-23 19:34:44.908311689 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 710
      iput-object p1, p0, Landroid/widget/Spinner$DropdownPopup$1;->this$1:Landroid/widget/Spinner$DropdownPopup;
  
      iput-object p2, p0, Landroid/widget/Spinner$DropdownPopup$1;->val$this$0:Landroid/widget/Spinner;
***************
*** 49,63 ****
--- 50,67 ----
      .parameter "id"
  
      .prologue
+     .line 712
      iget-object v0, p0, Landroid/widget/Spinner$DropdownPopup$1;->this$1:Landroid/widget/Spinner$DropdownPopup;
  
      iget-object v0, v0, Landroid/widget/Spinner$DropdownPopup;->this$0:Landroid/widget/Spinner;
  
      invoke-virtual {v0, p3}, Landroid/widget/Spinner;->setSelection(I)V
  
+     .line 713
      iget-object v0, p0, Landroid/widget/Spinner$DropdownPopup$1;->this$1:Landroid/widget/Spinner$DropdownPopup;
  
      invoke-virtual {v0}, Landroid/widget/Spinner$DropdownPopup;->dismiss()V
  
+     .line 714
      return-void
  .end method
