*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabHost$TabSpec.smali	2012-03-23 19:35:49.210484238 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabHost$TabSpec.smali.original	2012-03-23 19:34:44.854310704 -0500
***************
*** 31,42 ****
--- 31,45 ----
      .parameter "tag"
  
      .prologue
+     .line 428
      iput-object p1, p0, Landroid/widget/TabHost$TabSpec;->this$0:Landroid/widget/TabHost;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 429
      iput-object p2, p0, Landroid/widget/TabHost$TabSpec;->mTag:Ljava/lang/String;
  
+     .line 430
      return-void
  .end method
  
***************
*** 47,52 ****
--- 50,56 ----
      .parameter "x2"
  
      .prologue
+     .line 421
      invoke-direct {p0, p1, p2}, Landroid/widget/TabHost$TabSpec;-><init>(Landroid/widget/TabHost;Ljava/lang/String;)V
  
      return-void
***************
*** 57,62 ****
--- 61,67 ----
      .parameter "x0"
  
      .prologue
+     .line 421
      iget-object v0, p0, Landroid/widget/TabHost$TabSpec;->mIndicatorStrategy:Landroid/widget/TabHost$IndicatorStrategy;
  
      return-object v0
***************
*** 67,72 ****
--- 72,78 ----
      .parameter "x0"
  
      .prologue
+     .line 421
      iget-object v0, p0, Landroid/widget/TabHost$TabSpec;->mContentStrategy:Landroid/widget/TabHost$ContentStrategy;
  
      return-object v0
***************
*** 78,83 ****
--- 84,90 ----
      .locals 1
  
      .prologue
+     .line 484
      iget-object v0, p0, Landroid/widget/TabHost$TabSpec;->mTag:Ljava/lang/String;
  
      return-object v0
***************
*** 88,93 ****
--- 95,101 ----
      .parameter "viewId"
  
      .prologue
+     .line 461
      new-instance v0, Landroid/widget/TabHost$ViewIdContentStrategy;
  
      iget-object v1, p0, Landroid/widget/TabHost$TabSpec;->this$0:Landroid/widget/TabHost;
***************
*** 98,103 ****
--- 106,112 ----
  
      iput-object v0, p0, Landroid/widget/TabHost$TabSpec;->mContentStrategy:Landroid/widget/TabHost$ContentStrategy;
  
+     .line 462
      return-object p0
  .end method
  
***************
*** 106,111 ****
--- 115,121 ----
      .parameter "intent"
  
      .prologue
+     .line 478
      new-instance v0, Landroid/widget/TabHost$IntentContentStrategy;
  
      iget-object v1, p0, Landroid/widget/TabHost$TabSpec;->this$0:Landroid/widget/TabHost;
***************
*** 118,123 ****
--- 128,134 ----
  
      iput-object v0, p0, Landroid/widget/TabHost$TabSpec;->mContentStrategy:Landroid/widget/TabHost$ContentStrategy;
  
+     .line 479
      return-object p0
  .end method
  
***************
*** 126,131 ****
--- 137,143 ----
      .parameter "contentFactory"
  
      .prologue
+     .line 470
      new-instance v0, Landroid/widget/TabHost$FactoryContentStrategy;
  
      iget-object v1, p0, Landroid/widget/TabHost$TabSpec;->this$0:Landroid/widget/TabHost;
***************
*** 136,141 ****
--- 148,154 ----
  
      iput-object v0, p0, Landroid/widget/TabHost$TabSpec;->mContentStrategy:Landroid/widget/TabHost$ContentStrategy;
  
+     .line 471
      return-object p0
  .end method
  
***************
*** 144,149 ****
--- 157,163 ----
      .parameter "view"
  
      .prologue
+     .line 452
      new-instance v0, Landroid/widget/TabHost$ViewIndicatorStrategy;
  
      iget-object v1, p0, Landroid/widget/TabHost$TabSpec;->this$0:Landroid/widget/TabHost;
***************
*** 154,159 ****
--- 168,174 ----
  
      iput-object v0, p0, Landroid/widget/TabHost$TabSpec;->mIndicatorStrategy:Landroid/widget/TabHost$IndicatorStrategy;
  
+     .line 453
      return-object p0
  .end method
  
***************
*** 162,167 ****
--- 177,183 ----
      .parameter "label"
  
      .prologue
+     .line 436
      new-instance v0, Landroid/widget/TabHost$LabelIndicatorStrategy;
  
      iget-object v1, p0, Landroid/widget/TabHost$TabSpec;->this$0:Landroid/widget/TabHost;
***************
*** 172,177 ****
--- 188,194 ----
  
      iput-object v0, p0, Landroid/widget/TabHost$TabSpec;->mIndicatorStrategy:Landroid/widget/TabHost$IndicatorStrategy;
  
+     .line 437
      return-object p0
  .end method
  
***************
*** 181,186 ****
--- 198,204 ----
      .parameter "icon"
  
      .prologue
+     .line 444
      new-instance v0, Landroid/widget/TabHost$LabelAndIconIndicatorStrategy;
  
      iget-object v1, p0, Landroid/widget/TabHost$TabSpec;->this$0:Landroid/widget/TabHost;
***************
*** 191,195 ****
--- 209,214 ----
  
      iput-object v0, p0, Landroid/widget/TabHost$TabSpec;->mIndicatorStrategy:Landroid/widget/TabHost$IndicatorStrategy;
  
+     .line 445
      return-object p0
  .end method
