*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabHost$ViewIndicatorStrategy.smali	2012-03-23 19:35:59.906679282 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabHost$ViewIndicatorStrategy.smali.original	2012-03-23 19:34:44.936312200 -0500
***************
*** 30,41 ****
--- 30,44 ----
      .parameter "view"
  
      .prologue
+     .line 600
      iput-object p1, p0, Landroid/widget/TabHost$ViewIndicatorStrategy;->this$0:Landroid/widget/TabHost;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 601
      iput-object p2, p0, Landroid/widget/TabHost$ViewIndicatorStrategy;->mView:Landroid/view/View;
  
+     .line 602
      return-void
  .end method
  
***************
*** 46,51 ****
--- 49,55 ----
      .parameter "x2"
  
      .prologue
+     .line 596
      invoke-direct {p0, p1, p2}, Landroid/widget/TabHost$ViewIndicatorStrategy;-><init>(Landroid/widget/TabHost;Landroid/view/View;)V
  
      return-void
***************
*** 57,62 ****
--- 61,67 ----
      .locals 1
  
      .prologue
+     .line 605
      iget-object v0, p0, Landroid/widget/TabHost$ViewIndicatorStrategy;->mView:Landroid/view/View;
  
      return-object v0
