*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabHost$LabelIndicatorStrategy.smali	2012-03-23 19:35:57.293631634 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabHost$LabelIndicatorStrategy.smali.original	2012-03-23 19:34:44.919311890 -0500
***************
*** 30,41 ****
--- 30,44 ----
      .parameter "label"
  
      .prologue
+     .line 523
      iput-object p1, p0, Landroid/widget/TabHost$LabelIndicatorStrategy;->this$0:Landroid/widget/TabHost;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 524
      iput-object p2, p0, Landroid/widget/TabHost$LabelIndicatorStrategy;->mLabel:Ljava/lang/CharSequence;
  
+     .line 525
      return-void
  .end method
  
***************
*** 46,51 ****
--- 49,55 ----
      .parameter "x2"
  
      .prologue
+     .line 519
      invoke-direct {p0, p1, p2}, Landroid/widget/TabHost$LabelIndicatorStrategy;-><init>(Landroid/widget/TabHost;Ljava/lang/CharSequence;)V
  
      return-void
***************
*** 57,68 ****
--- 61,74 ----
      .locals 7
  
      .prologue
+     .line 528
      iget-object v4, p0, Landroid/widget/TabHost$LabelIndicatorStrategy;->this$0:Landroid/widget/TabHost;
  
      invoke-virtual {v4}, Landroid/widget/TabHost;->getContext()Landroid/content/Context;
  
      move-result-object v0
  
+     .line 529
      .local v0, context:Landroid/content/Context;
      const-string v4, "layout_inflater"
  
***************
*** 72,77 ****
--- 78,84 ----
  
      check-cast v1, Landroid/view/LayoutInflater;
  
+     .line 531
      .local v1, inflater:Landroid/view/LayoutInflater;
      iget-object v4, p0, Landroid/widget/TabHost$LabelIndicatorStrategy;->this$0:Landroid/widget/TabHost;
  
***************
*** 93,98 ****
--- 100,106 ----
  
      move-result-object v2
  
+     .line 535
      .local v2, tabIndicator:Landroid/view/View;
      const v4, 0x1020016
  
***************
*** 102,112 ****
--- 110,122 ----
  
      check-cast v3, Landroid/widget/TextView;
  
+     .line 536
      .local v3, tv:Landroid/widget/TextView;
      iget-object v4, p0, Landroid/widget/TabHost$LabelIndicatorStrategy;->mLabel:Ljava/lang/CharSequence;
  
      invoke-virtual {v3, v4}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 538
      invoke-virtual {v0}, Landroid/content/Context;->getApplicationInfo()Landroid/content/pm/ApplicationInfo;
  
      move-result-object v4
***************
*** 117,126 ****
--- 127,138 ----
  
      if-gt v4, v5, :cond_0
  
+     .line 540
      const v4, 0x108057f
  
      invoke-virtual {v2, v4}, Landroid/view/View;->setBackgroundResource(I)V
  
+     .line 541
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v4
***************
*** 133,138 ****
--- 145,151 ----
  
      invoke-virtual {v3, v4}, Landroid/widget/TextView;->setTextColor(Landroid/content/res/ColorStateList;)V
  
+     .line 544
      :cond_0
      return-object v2
  .end method
