*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabHost$ViewIdContentStrategy.smali	2012-03-23 19:36:00.020681361 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabHost$ViewIdContentStrategy.smali.original	2012-03-23 19:34:44.937312218 -0500
***************
*** 30,39 ****
--- 30,41 ----
      .parameter "viewId"
  
      .prologue
+     .line 616
      iput-object p1, p0, Landroid/widget/TabHost$ViewIdContentStrategy;->this$0:Landroid/widget/TabHost;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 617
      #getter for: Landroid/widget/TabHost;->mTabContent:Landroid/widget/FrameLayout;
      invoke-static {p1}, Landroid/widget/TabHost;->access$100(Landroid/widget/TabHost;)Landroid/widget/FrameLayout;
  
***************
*** 45,62 ****
--- 47,68 ----
  
      iput-object v0, p0, Landroid/widget/TabHost$ViewIdContentStrategy;->mView:Landroid/view/View;
  
+     .line 618
      iget-object v0, p0, Landroid/widget/TabHost$ViewIdContentStrategy;->mView:Landroid/view/View;
  
      if-eqz v0, :cond_0
  
+     .line 619
      iget-object v0, p0, Landroid/widget/TabHost$ViewIdContentStrategy;->mView:Landroid/view/View;
  
      const/16 v1, 0x8
  
      invoke-virtual {v0, v1}, Landroid/view/View;->setVisibility(I)V
  
+     .line 624
      return-void
  
+     .line 621
      :cond_0
      new-instance v0, Ljava/lang/RuntimeException;
  
***************
*** 90,95 ****
--- 96,102 ----
      .parameter "x2"
  
      .prologue
+     .line 612
      invoke-direct {p0, p1, p2}, Landroid/widget/TabHost$ViewIdContentStrategy;-><init>(Landroid/widget/TabHost;I)V
  
      return-void
***************
*** 101,112 ****
--- 108,121 ----
      .locals 2
  
      .prologue
+     .line 627
      iget-object v0, p0, Landroid/widget/TabHost$ViewIdContentStrategy;->mView:Landroid/view/View;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, v1}, Landroid/view/View;->setVisibility(I)V
  
+     .line 628
      iget-object v0, p0, Landroid/widget/TabHost$ViewIdContentStrategy;->mView:Landroid/view/View;
  
      return-object v0
***************
*** 116,126 ****
--- 125,137 ----
      .locals 2
  
      .prologue
+     .line 632
      iget-object v0, p0, Landroid/widget/TabHost$ViewIdContentStrategy;->mView:Landroid/view/View;
  
      const/16 v1, 0x8
  
      invoke-virtual {v0, v1}, Landroid/view/View;->setVisibility(I)V
  
+     .line 633
      return-void
  .end method
