*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TableLayout.smali	2012-03-23 19:35:54.657583565 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TableLayout.smali.original	2012-03-23 19:34:44.898311506 -0500
***************
*** 37,46 ****
--- 37,49 ----
      .parameter "context"
  
      .prologue
+     .line 93
      invoke-direct {p0, p1}, Landroid/widget/LinearLayout;-><init>(Landroid/content/Context;)V
  
+     .line 94
      invoke-direct {p0}, Landroid/widget/TableLayout;->initTableLayout()V
  
+     .line 95
      return-void
  .end method
  
***************
*** 56,102 ****
--- 59,116 ----
  
      const/4 v5, 0x0
  
+     .line 105
      invoke-direct {p0, p1, p2}, Landroid/widget/LinearLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 107
      sget-object v4, Lcom/android/internal/R$styleable;->TableLayout:[I
  
      invoke-virtual {p1, p2, v4}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[I)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 109
      .local v0, a:Landroid/content/res/TypedArray;
      invoke-virtual {v0, v5}, Landroid/content/res/TypedArray;->getString(I)Ljava/lang/String;
  
      move-result-object v3
  
+     .line 110
      .local v3, stretchedColumns:Ljava/lang/String;
      if-eqz v3, :cond_0
  
+     .line 111
      invoke-virtual {v3, v5}, Ljava/lang/String;->charAt(I)C
  
      move-result v4
  
      if-ne v4, v7, :cond_3
  
+     .line 112
      iput-boolean v6, p0, Landroid/widget/TableLayout;->mStretchAllColumns:Z
  
+     .line 118
      :cond_0
      :goto_0
      invoke-virtual {v0, v6}, Landroid/content/res/TypedArray;->getString(I)Ljava/lang/String;
  
      move-result-object v2
  
+     .line 119
      .local v2, shrinkedColumns:Ljava/lang/String;
      if-eqz v2, :cond_1
  
+     .line 120
      invoke-virtual {v2, v5}, Ljava/lang/String;->charAt(I)C
  
      move-result v4
  
      if-ne v4, v7, :cond_4
  
+     .line 121
      iput-boolean v6, p0, Landroid/widget/TableLayout;->mShrinkAllColumns:Z
  
+     .line 127
      :cond_1
      :goto_1
      const/4 v4, 0x2
***************
*** 105,126 ****
--- 119,146 ----
  
      move-result-object v1
  
+     .line 128
      .local v1, collapsedColumns:Ljava/lang/String;
      if-eqz v1, :cond_2
  
+     .line 129
      invoke-static {v1}, Landroid/widget/TableLayout;->parseColumns(Ljava/lang/String;)Landroid/util/SparseBooleanArray;
  
      move-result-object v4
  
      iput-object v4, p0, Landroid/widget/TableLayout;->mCollapsedColumns:Landroid/util/SparseBooleanArray;
  
+     .line 132
      :cond_2
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 133
      invoke-direct {p0}, Landroid/widget/TableLayout;->initTableLayout()V
  
+     .line 134
      return-void
  
+     .line 114
      .end local v1           #collapsedColumns:Ljava/lang/String;
      .end local v2           #shrinkedColumns:Ljava/lang/String;
      :cond_3
***************
*** 132,137 ****
--- 152,158 ----
  
      goto :goto_0
  
+     .line 123
      .restart local v2       #shrinkedColumns:Ljava/lang/String;
      :cond_4
      invoke-static {v2}, Landroid/widget/TableLayout;->parseColumns(Ljava/lang/String;)Landroid/util/SparseBooleanArray;
***************
*** 149,154 ****
--- 170,176 ----
      .parameter "x1"
  
      .prologue
+     .line 74
      invoke-direct {p0, p1}, Landroid/widget/TableLayout;->trackCollapsedColumns(Landroid/view/View;)V
  
      return-void
***************
*** 159,171 ****
--- 181,196 ----
      .parameter "widthMeasureSpec"
  
      .prologue
+     .line 478
      const/4 v7, 0x1
  
+     .line 487
      .local v7, firstRow:Z
      invoke-virtual/range {p0 .. p0}, Landroid/widget/TableLayout;->getChildCount()I
  
      move-result v5
  
+     .line 488
      .local v5, count:I
      const/4 v8, 0x0
  
***************
*** 173,184 ****
--- 198,211 ----
      :goto_0
      if-ge v8, v5, :cond_6
  
+     .line 489
      move-object/from16 v0, p0
  
      invoke-virtual {v0, v8}, Landroid/widget/TableLayout;->getChildAt(I)Landroid/view/View;
  
      move-result-object v4
  
+     .line 490
      .local v4, child:Landroid/view/View;
      invoke-virtual {v4}, Landroid/view/View;->getVisibility()I
  
***************
*** 192,203 ****
--- 219,232 ----
  
      if-ne v0, v1, :cond_1
  
+     .line 488
      :cond_0
      :goto_1
      add-int/lit8 v8, v8, 0x1
  
      goto :goto_0
  
+     .line 494
      :cond_1
      instance-of v0, v4, Landroid/widget/TableRow;
  
***************
*** 207,219 ****
--- 236,251 ----
  
      move-object v15, v4
  
+     .line 495
      check-cast v15, Landroid/widget/TableRow;
  
+     .line 497
      .local v15, row:Landroid/widget/TableRow;
      invoke-virtual {v15}, Landroid/widget/TableRow;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
  
      move-result-object v10
  
+     .line 498
      .local v10, layoutParams:Landroid/view/ViewGroup$LayoutParams;
      const/16 v17, -0x2
  
***************
*** 221,240 ****
--- 253,276 ----
  
      iput v0, v10, Landroid/view/ViewGroup$LayoutParams;->height:I
  
+     .line 500
      move/from16 v0, p1
  
      invoke-virtual {v15, v0}, Landroid/widget/TableRow;->getColumnsWidths(I)[I
  
      move-result-object v16
  
+     .line 501
      .local v16, widths:[I
      move-object/from16 v0, v16
  
      array-length v13, v0
  
+     .line 503
      .local v13, newLength:I
      if-eqz v7, :cond_4
  
+     .line 504
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/widget/TableLayout;->mMaxWidths:[I
***************
*** 259,264 ****
--- 295,301 ----
  
      if-eq v0, v13, :cond_3
  
+     .line 505
      :cond_2
      new-array v0, v13, [I
  
***************
*** 270,275 ****
--- 307,313 ----
  
      iput-object v0, v1, Landroid/widget/TableLayout;->mMaxWidths:[I
  
+     .line 507
      :cond_3
      const/16 v17, 0x0
  
***************
*** 291,300 ****
--- 329,340 ----
  
      invoke-static {v0, v1, v2, v3, v13}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 508
      const/4 v7, 0x0
  
      goto :goto_1
  
+     .line 510
      :cond_4
      move-object/from16 v0, p0
  
***************
*** 306,321 ****
--- 346,365 ----
  
      array-length v11, v0
  
+     .line 511
      .local v11, length:I
      sub-int v6, v13, v11
  
+     .line 514
      .local v6, difference:I
      if-lez v6, :cond_5
  
+     .line 515
      move-object/from16 v0, p0
  
      iget-object v14, v0, Landroid/widget/TableLayout;->mMaxWidths:[I
  
+     .line 516
      .local v14, oldMaxWidths:[I
      new-array v0, v13, [I
  
***************
*** 327,332 ****
--- 371,377 ----
  
      iput-object v0, v1, Landroid/widget/TableLayout;->mMaxWidths:[I
  
+     .line 517
      const/16 v17, 0x0
  
      move-object/from16 v0, p0
***************
*** 351,356 ****
--- 396,402 ----
  
      invoke-static {v14, v0, v1, v2, v3}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 519
      array-length v0, v14
  
      move/from16 v17, v0
***************
*** 375,397 ****
--- 421,447 ----
  
      invoke-static {v0, v1, v2, v3, v6}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 527
      .end local v14           #oldMaxWidths:[I
      :cond_5
      move-object/from16 v0, p0
  
      iget-object v12, v0, Landroid/widget/TableLayout;->mMaxWidths:[I
  
+     .line 528
      .local v12, maxWidths:[I
      invoke-static {v11, v13}, Ljava/lang/Math;->min(II)I
  
      move-result v11
  
+     .line 529
      const/4 v9, 0x0
  
      .local v9, j:I
      :goto_2
      if-ge v9, v11, :cond_0
  
+     .line 530
      aget v17, v12, v9
  
      aget v18, v16, v9
***************
*** 402,411 ****
--- 452,463 ----
  
      aput v17, v12, v9
  
+     .line 529
      add-int/lit8 v9, v9, 0x1
  
      goto :goto_2
  
+     .line 535
      .end local v4           #child:Landroid/view/View;
      .end local v6           #difference:I
      .end local v9           #j:I
***************
*** 423,460 ****
--- 475,519 ----
      .locals 2
  
      .prologue
+     .line 177
      iget-object v0, p0, Landroid/widget/TableLayout;->mCollapsedColumns:Landroid/util/SparseBooleanArray;
  
      if-nez v0, :cond_0
  
+     .line 178
      new-instance v0, Landroid/util/SparseBooleanArray;
  
      invoke-direct {v0}, Landroid/util/SparseBooleanArray;-><init>()V
  
      iput-object v0, p0, Landroid/widget/TableLayout;->mCollapsedColumns:Landroid/util/SparseBooleanArray;
  
+     .line 180
      :cond_0
      iget-object v0, p0, Landroid/widget/TableLayout;->mStretchableColumns:Landroid/util/SparseBooleanArray;
  
      if-nez v0, :cond_1
  
+     .line 181
      new-instance v0, Landroid/util/SparseBooleanArray;
  
      invoke-direct {v0}, Landroid/util/SparseBooleanArray;-><init>()V
  
      iput-object v0, p0, Landroid/widget/TableLayout;->mStretchableColumns:Landroid/util/SparseBooleanArray;
  
+     .line 183
      :cond_1
      iget-object v0, p0, Landroid/widget/TableLayout;->mShrinkableColumns:Landroid/util/SparseBooleanArray;
  
      if-nez v0, :cond_2
  
+     .line 184
      new-instance v0, Landroid/util/SparseBooleanArray;
  
      invoke-direct {v0}, Landroid/util/SparseBooleanArray;-><init>()V
  
      iput-object v0, p0, Landroid/widget/TableLayout;->mShrinkableColumns:Landroid/util/SparseBooleanArray;
  
+     .line 187
      :cond_2
      new-instance v0, Landroid/widget/TableLayout$PassThroughHierarchyChangeListener;
  
***************
*** 464,477 ****
--- 523,539 ----
  
      iput-object v0, p0, Landroid/widget/TableLayout;->mPassThroughListener:Landroid/widget/TableLayout$PassThroughHierarchyChangeListener;
  
+     .line 190
      iget-object v0, p0, Landroid/widget/TableLayout;->mPassThroughListener:Landroid/widget/TableLayout$PassThroughHierarchyChangeListener;
  
      invoke-super {p0, v0}, Landroid/widget/LinearLayout;->setOnHierarchyChangeListener(Landroid/view/ViewGroup$OnHierarchyChangeListener;)V
  
+     .line 192
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/TableLayout;->mInitialized:Z
  
+     .line 193
      return-void
  .end method
  
***************
*** 483,513 ****
--- 545,583 ----
      .parameter "totalWidth"
  
      .prologue
+     .line 574
      const/4 v8, 0x0
  
+     .line 575
      .local v8, skipped:I
      iget-object v6, p0, Landroid/widget/TableLayout;->mMaxWidths:[I
  
+     .line 576
      .local v6, maxWidths:[I
      array-length v5, v6
  
+     .line 577
      .local v5, length:I
      if-eqz p2, :cond_1
  
      move v2, v5
  
+     .line 578
      .local v2, count:I
      :goto_0
      sub-int v9, p3, p4
  
+     .line 579
      .local v9, totalExtraSpace:I
      div-int v3, v9, v2
  
+     .line 583
      .local v3, extraSpace:I
      invoke-virtual {p0}, Landroid/widget/TableLayout;->getChildCount()I
  
      move-result v7
  
+     .line 584
      .local v7, nbChildren:I
      const/4 v4, 0x0
  
***************
*** 515,536 ****
--- 585,611 ----
      :goto_1
      if-ge v4, v7, :cond_2
  
+     .line 585
      invoke-virtual {p0, v4}, Landroid/widget/TableLayout;->getChildAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 586
      .local v0, child:Landroid/view/View;
      instance-of v10, v0, Landroid/widget/TableRow;
  
      if-eqz v10, :cond_0
  
+     .line 587
      invoke-virtual {v0}, Landroid/view/View;->forceLayout()V
  
+     .line 584
      :cond_0
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_1
  
+     .line 577
      .end local v0           #child:Landroid/view/View;
      .end local v2           #count:I
      .end local v3           #extraSpace:I
***************
*** 544,549 ****
--- 619,625 ----
  
      goto :goto_0
  
+     .line 591
      .restart local v2       #count:I
      .restart local v3       #extraSpace:I
      .restart local v4       #i:I
***************
*** 552,566 ****
--- 628,645 ----
      :cond_2
      if-nez p2, :cond_5
  
+     .line 592
      const/4 v4, 0x0
  
      :goto_2
      if-ge v4, v2, :cond_6
  
+     .line 593
      invoke-virtual {p1, v4}, Landroid/util/SparseBooleanArray;->keyAt(I)I
  
      move-result v1
  
+     .line 594
      .local v1, column:I
      invoke-virtual {p1, v4}, Landroid/util/SparseBooleanArray;->valueAt(I)Z
  
***************
*** 568,592 ****
--- 647,676 ----
  
      if-eqz v10, :cond_3
  
+     .line 595
      if-ge v1, v5, :cond_4
  
+     .line 596
      aget v10, v6, v1
  
      add-int/2addr v10, v3
  
      aput v10, v6, v1
  
+     .line 592
      :cond_3
      :goto_3
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_2
  
+     .line 598
      :cond_4
      add-int/lit8 v8, v8, 0x1
  
      goto :goto_3
  
+     .line 603
      .end local v1           #column:I
      :cond_5
      const/4 v4, 0x0
***************
*** 594,629 ****
--- 678,720 ----
      :goto_4
      if-ge v4, v2, :cond_9
  
+     .line 604
      aget v10, v6, v4
  
      add-int/2addr v10, v3
  
      aput v10, v6, v4
  
+     .line 603
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_4
  
+     .line 611
      :cond_6
      if-lez v8, :cond_9
  
      if-ge v8, v2, :cond_9
  
+     .line 613
      mul-int v10, v8, v3
  
      sub-int v11, v2, v8
  
      div-int v3, v10, v11
  
+     .line 614
      const/4 v4, 0x0
  
      :goto_5
      if-ge v4, v2, :cond_9
  
+     .line 615
      invoke-virtual {p1, v4}, Landroid/util/SparseBooleanArray;->keyAt(I)I
  
      move-result v1
  
+     .line 616
      .restart local v1       #column:I
      invoke-virtual {p1, v4}, Landroid/util/SparseBooleanArray;->valueAt(I)Z
  
***************
*** 633,652 ****
--- 724,747 ----
  
      if-ge v1, v5, :cond_7
  
+     .line 617
      aget v10, v6, v1
  
      if-le v3, v10, :cond_8
  
+     .line 618
      const/4 v10, 0x0
  
      aput v10, v6, v1
  
+     .line 614
      :cond_7
      :goto_6
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_5
  
+     .line 620
      :cond_8
      aget v10, v6, v1
  
***************
*** 656,661 ****
--- 751,757 ----
  
      goto :goto_6
  
+     .line 625
      .end local v1           #column:I
      :cond_9
      return-void
***************
*** 666,675 ****
--- 762,773 ----
      .parameter "sequence"
  
      .prologue
+     .line 151
      new-instance v4, Landroid/util/SparseBooleanArray;
  
      invoke-direct {v4}, Landroid/util/SparseBooleanArray;-><init>()V
  
+     .line 152
      .local v4, columns:Landroid/util/SparseBooleanArray;
      const-string v8, "\\s*,\\s*"
  
***************
*** 677,687 ****
--- 775,787 ----
  
      move-result-object v7
  
+     .line 153
      .local v7, pattern:Ljava/util/regex/Pattern;
      invoke-virtual {v7, p0}, Ljava/util/regex/Pattern;->split(Ljava/lang/CharSequence;)[Ljava/lang/String;
  
      move-result-object v1
  
+     .line 155
      .local v1, columnDefs:[Ljava/lang/String;
      move-object v0, v1
  
***************
*** 697,717 ****
--- 797,821 ----
  
      aget-object v2, v0, v5
  
+     .line 157
      .local v2, columnIdentifier:Ljava/lang/String;
      :try_start_0
      invoke-static {v2}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
  
      move-result v3
  
+     .line 159
      .local v3, columnIndex:I
      if-ltz v3, :cond_0
  
+     .line 162
      const/4 v8, 0x1
  
      invoke-virtual {v4, v3, v8}, Landroid/util/SparseBooleanArray;->put(IZ)V
      :try_end_0
      .catch Ljava/lang/NumberFormatException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 155
      .end local v3           #columnIndex:I
      :cond_0
      :goto_1
***************
*** 719,728 ****
--- 823,834 ----
  
      goto :goto_0
  
+     .line 169
      .end local v2           #columnIdentifier:Ljava/lang/String;
      :cond_1
      return-object v4
  
+     .line 164
      .restart local v2       #columnIdentifier:Ljava/lang/String;
      :catch_0
      move-exception v8
***************
*** 734,747 ****
--- 840,856 ----
      .locals 3
  
      .prologue
+     .line 206
      iget-boolean v2, p0, Landroid/widget/TableLayout;->mInitialized:Z
  
      if-eqz v2, :cond_0
  
+     .line 207
      invoke-virtual {p0}, Landroid/widget/TableLayout;->getChildCount()I
  
      move-result v0
  
+     .line 208
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 749,764 ****
--- 858,876 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 209
      invoke-virtual {p0, v1}, Landroid/widget/TableLayout;->getChildAt(I)Landroid/view/View;
  
      move-result-object v2
  
      invoke-virtual {v2}, Landroid/view/View;->requestLayout()V
  
+     .line 208
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 212
      .end local v0           #count:I
      .end local v1           #i:I
      :cond_0
***************
*** 770,786 ****
--- 882,902 ----
      .parameter "widthMeasureSpec"
  
      .prologue
+     .line 549
      iget-object v6, p0, Landroid/widget/TableLayout;->mMaxWidths:[I
  
      if-nez v6, :cond_1
  
+     .line 570
      :cond_0
      :goto_0
      return-void
  
+     .line 554
      :cond_1
      const/4 v4, 0x0
  
+     .line 555
      .local v4, totalWidth:I
      iget-object v0, p0, Landroid/widget/TableLayout;->mMaxWidths:[I
  
***************
*** 796,808 ****
--- 912,927 ----
  
      aget v5, v0, v1
  
+     .line 556
      .local v5, width:I
      add-int/2addr v4, v5
  
+     .line 555
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 559
      .end local v5           #width:I
      :cond_2
      invoke-static {p1}, Landroid/view/View$MeasureSpec;->getSize(I)I
***************
*** 817,822 ****
--- 936,942 ----
  
      sub-int v3, v6, v7
  
+     .line 561
      .local v3, size:I
      if-le v4, v3, :cond_4
  
***************
*** 832,837 ****
--- 952,958 ----
  
      if-lez v6, :cond_4
  
+     .line 564
      :cond_3
      iget-object v6, p0, Landroid/widget/TableLayout;->mShrinkableColumns:Landroid/util/SparseBooleanArray;
  
***************
*** 841,846 ****
--- 962,968 ----
  
      goto :goto_0
  
+     .line 565
      :cond_4
      if-ge v4, v3, :cond_0
  
***************
*** 856,861 ****
--- 978,984 ----
  
      if-lez v6, :cond_0
  
+     .line 568
      :cond_5
      iget-object v6, p0, Landroid/widget/TableLayout;->mStretchableColumns:Landroid/util/SparseBooleanArray;
  
***************
*** 871,892 ****
--- 994,1020 ----
      .parameter "child"
  
      .prologue
+     .line 372
      instance-of v6, p1, Landroid/widget/TableRow;
  
      if-eqz v6, :cond_1
  
      move-object v5, p1
  
+     .line 373
      check-cast v5, Landroid/widget/TableRow;
  
+     .line 374
      .local v5, row:Landroid/widget/TableRow;
      iget-object v0, p0, Landroid/widget/TableLayout;->mCollapsedColumns:Landroid/util/SparseBooleanArray;
  
+     .line 375
      .local v0, collapsedColumns:Landroid/util/SparseBooleanArray;
      invoke-virtual {v0}, Landroid/util/SparseBooleanArray;->size()I
  
      move-result v2
  
+     .line 376
      .local v2, count:I
      const/4 v3, 0x0
  
***************
*** 894,918 ****
--- 1022,1052 ----
      :goto_0
      if-ge v3, v2, :cond_1
  
+     .line 377
      invoke-virtual {v0, v3}, Landroid/util/SparseBooleanArray;->keyAt(I)I
  
      move-result v1
  
+     .line 378
      .local v1, columnIndex:I
      invoke-virtual {v0, v3}, Landroid/util/SparseBooleanArray;->valueAt(I)Z
  
      move-result v4
  
+     .line 382
      .local v4, isCollapsed:Z
      if-eqz v4, :cond_0
  
+     .line 383
      invoke-virtual {v5, v1, v4}, Landroid/widget/TableRow;->setColumnCollapsed(IZ)V
  
+     .line 376
      :cond_0
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 387
      .end local v0           #collapsedColumns:Landroid/util/SparseBooleanArray;
      .end local v1           #columnIndex:I
      .end local v2           #count:I
***************
*** 930,939 ****
--- 1064,1076 ----
      .parameter "child"
  
      .prologue
+     .line 394
      invoke-super {p0, p1}, Landroid/widget/LinearLayout;->addView(Landroid/view/View;)V
  
+     .line 395
      invoke-direct {p0}, Landroid/widget/TableLayout;->requestRowsLayout()V
  
+     .line 396
      return-void
  .end method
  
***************
*** 943,952 ****
--- 1080,1092 ----
      .parameter "index"
  
      .prologue
+     .line 403
      invoke-super {p0, p1, p2}, Landroid/widget/LinearLayout;->addView(Landroid/view/View;I)V
  
+     .line 404
      invoke-direct {p0}, Landroid/widget/TableLayout;->requestRowsLayout()V
  
+     .line 405
      return-void
  .end method
  
***************
*** 957,966 ****
--- 1097,1109 ----
      .parameter "params"
  
      .prologue
+     .line 421
      invoke-super {p0, p1, p2, p3}, Landroid/widget/LinearLayout;->addView(Landroid/view/View;ILandroid/view/ViewGroup$LayoutParams;)V
  
+     .line 422
      invoke-direct {p0}, Landroid/widget/TableLayout;->requestRowsLayout()V
  
+     .line 423
      return-void
  .end method
  
***************
*** 970,979 ****
--- 1113,1125 ----
      .parameter "params"
  
      .prologue
+     .line 412
      invoke-super {p0, p1, p2}, Landroid/widget/LinearLayout;->addView(Landroid/view/View;Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 413
      invoke-direct {p0}, Landroid/widget/TableLayout;->requestRowsLayout()V
  
+     .line 414
      return-void
  .end method
  
***************
*** 982,987 ****
--- 1128,1134 ----
      .parameter "p"
  
      .prologue
+     .line 650
      instance-of v0, p1, Landroid/widget/TableLayout$LayoutParams;
  
      return v0
***************
*** 991,996 ****
--- 1138,1144 ----
      .locals 1
  
      .prologue
+     .line 74
      invoke-virtual {p0}, Landroid/widget/TableLayout;->generateDefaultLayoutParams()Landroid/widget/LinearLayout$LayoutParams;
  
      move-result-object v0
***************
*** 1002,1007 ****
--- 1150,1156 ----
      .locals 1
  
      .prologue
+     .line 642
      new-instance v0, Landroid/widget/TableLayout$LayoutParams;
  
      invoke-direct {v0}, Landroid/widget/TableLayout$LayoutParams;-><init>()V
***************
*** 1014,1019 ****
--- 1163,1169 ----
      .parameter "x0"
  
      .prologue
+     .line 74
      invoke-virtual {p0, p1}, Landroid/widget/TableLayout;->generateLayoutParams(Landroid/util/AttributeSet;)Landroid/widget/TableLayout$LayoutParams;
  
      move-result-object v0
***************
*** 1026,1031 ****
--- 1176,1182 ----
      .parameter "x0"
  
      .prologue
+     .line 74
      invoke-virtual {p0, p1}, Landroid/widget/TableLayout;->generateLayoutParams(Landroid/view/ViewGroup$LayoutParams;)Landroid/widget/LinearLayout$LayoutParams;
  
      move-result-object v0
***************
*** 1038,1043 ****
--- 1189,1195 ----
      .parameter "x0"
  
      .prologue
+     .line 74
      invoke-virtual {p0, p1}, Landroid/widget/TableLayout;->generateLayoutParams(Landroid/util/AttributeSet;)Landroid/widget/TableLayout$LayoutParams;
  
      move-result-object v0
***************
*** 1050,1055 ****
--- 1202,1208 ----
      .parameter "p"
  
      .prologue
+     .line 658
      new-instance v0, Landroid/widget/TableLayout$LayoutParams;
  
      invoke-direct {v0, p1}, Landroid/widget/TableLayout$LayoutParams;-><init>(Landroid/view/ViewGroup$LayoutParams;)V
***************
*** 1062,1067 ****
--- 1215,1221 ----
      .parameter "attrs"
  
      .prologue
+     .line 632
      new-instance v0, Landroid/widget/TableLayout$LayoutParams;
  
      invoke-virtual {p0}, Landroid/widget/TableLayout;->getContext()Landroid/content/Context;
***************
*** 1078,1083 ****
--- 1232,1238 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 304
      iget-object v0, p0, Landroid/widget/TableLayout;->mCollapsedColumns:Landroid/util/SparseBooleanArray;
  
      invoke-virtual {v0, p1}, Landroid/util/SparseBooleanArray;->get(I)Z
***************
*** 1092,1097 ****
--- 1247,1253 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 358
      iget-boolean v0, p0, Landroid/widget/TableLayout;->mShrinkAllColumns:Z
  
      if-nez v0, :cond_0
***************
*** 1121,1126 ****
--- 1277,1283 ----
      .parameter "columnIndex"
  
      .prologue
+     .line 331
      iget-boolean v0, p0, Landroid/widget/TableLayout;->mStretchAllColumns:Z
  
      if-nez v0, :cond_0
***************
*** 1149,1154 ****
--- 1306,1312 ----
      .locals 1
  
      .prologue
+     .line 235
      iget-boolean v0, p0, Landroid/widget/TableLayout;->mShrinkAllColumns:Z
  
      return v0
***************
*** 1158,1163 ****
--- 1316,1322 ----
      .locals 1
  
      .prologue
+     .line 255
      iget-boolean v0, p0, Landroid/widget/TableLayout;->mStretchAllColumns:Z
  
      return v0
***************
*** 1173,1193 ****
--- 1332,1356 ----
      .parameter "totalHeight"
  
      .prologue
+     .line 452
      instance-of v0, p1, Landroid/widget/TableRow;
  
      if-eqz v0, :cond_0
  
      move-object v0, p1
  
+     .line 453
      check-cast v0, Landroid/widget/TableRow;
  
      iget-object v1, p0, Landroid/widget/TableLayout;->mMaxWidths:[I
  
      invoke-virtual {v0, v1}, Landroid/widget/TableRow;->setColumnsWidthConstraints([I)V
  
+     .line 456
      :cond_0
      invoke-super/range {p0 .. p6}, Landroid/widget/LinearLayout;->measureChildBeforeLayout(Landroid/view/View;IIIII)V
  
+     .line 458
      return-void
  .end method
  
***************
*** 1197,1208 ****
--- 1360,1375 ----
      .parameter "heightMeasureSpec"
  
      .prologue
+     .line 465
      invoke-direct {p0, p1}, Landroid/widget/TableLayout;->findLargestCells(I)V
  
+     .line 466
      invoke-direct {p0, p1}, Landroid/widget/TableLayout;->shrinkAndStretchColumns(I)V
  
+     .line 468
      invoke-super {p0, p1, p2}, Landroid/widget/LinearLayout;->measureVertical(II)V
  
+     .line 469
      return-void
  .end method
  
***************
*** 1215,1222 ****
--- 1382,1391 ----
      .parameter "b"
  
      .prologue
+     .line 440
      invoke-virtual {p0}, Landroid/widget/TableLayout;->layoutVertical()V
  
+     .line 441
      return-void
  .end method
  
***************
*** 1226,1233 ****
--- 1395,1404 ----
      .parameter "heightMeasureSpec"
  
      .prologue
+     .line 431
      invoke-virtual {p0, p1, p2}, Landroid/widget/TableLayout;->measureVertical(II)V
  
+     .line 432
      return-void
  .end method
  
***************
*** 1235,1248 ****
--- 1406,1422 ----
      .locals 3
  
      .prologue
+     .line 219
      iget-boolean v2, p0, Landroid/widget/TableLayout;->mInitialized:Z
  
      if-eqz v2, :cond_0
  
+     .line 220
      invoke-virtual {p0}, Landroid/widget/TableLayout;->getChildCount()I
  
      move-result v0
  
+     .line 221
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 1250,1270 ****
--- 1424,1448 ----
      :goto_0
      if-ge v1, v0, :cond_0
  
+     .line 222
      invoke-virtual {p0, v1}, Landroid/widget/TableLayout;->getChildAt(I)Landroid/view/View;
  
      move-result-object v2
  
      invoke-virtual {v2}, Landroid/view/View;->forceLayout()V
  
+     .line 221
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 226
      .end local v0           #count:I
      .end local v1           #i:I
      :cond_0
      invoke-super {p0}, Landroid/widget/LinearLayout;->requestLayout()V
  
+     .line 227
      return-void
  .end method
  
***************
*** 1274,1287 ****
--- 1452,1468 ----
      .parameter "isCollapsed"
  
      .prologue
+     .line 284
      iget-object v3, p0, Landroid/widget/TableLayout;->mCollapsedColumns:Landroid/util/SparseBooleanArray;
  
      invoke-virtual {v3, p1, p2}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 286
      invoke-virtual {p0}, Landroid/widget/TableLayout;->getChildCount()I
  
      move-result v0
  
+     .line 287
      .local v0, count:I
      const/4 v1, 0x0
  
***************
*** 1289,1316 ****
--- 1470,1503 ----
      :goto_0
      if-ge v1, v0, :cond_1
  
+     .line 288
      invoke-virtual {p0, v1}, Landroid/widget/TableLayout;->getChildAt(I)Landroid/view/View;
  
      move-result-object v2
  
+     .line 289
      .local v2, view:Landroid/view/View;
      instance-of v3, v2, Landroid/widget/TableRow;
  
      if-eqz v3, :cond_0
  
+     .line 290
      check-cast v2, Landroid/widget/TableRow;
  
      .end local v2           #view:Landroid/view/View;
      invoke-virtual {v2, p1, p2}, Landroid/widget/TableRow;->setColumnCollapsed(IZ)V
  
+     .line 287
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 294
      :cond_1
      invoke-direct {p0}, Landroid/widget/TableLayout;->requestRowsLayout()V
  
+     .line 295
      return-void
  .end method
  
***************
*** 1320,1331 ****
--- 1507,1521 ----
      .parameter "isShrinkable"
  
      .prologue
+     .line 347
      iget-object v0, p0, Landroid/widget/TableLayout;->mShrinkableColumns:Landroid/util/SparseBooleanArray;
  
      invoke-virtual {v0, p1, p2}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 348
      invoke-direct {p0}, Landroid/widget/TableLayout;->requestRowsLayout()V
  
+     .line 349
      return-void
  .end method
  
***************
*** 1335,1346 ****
--- 1525,1539 ----
      .parameter "isStretchable"
  
      .prologue
+     .line 320
      iget-object v0, p0, Landroid/widget/TableLayout;->mStretchableColumns:Landroid/util/SparseBooleanArray;
  
      invoke-virtual {v0, p1, p2}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 321
      invoke-direct {p0}, Landroid/widget/TableLayout;->requestRowsLayout()V
  
+     .line 322
      return-void
  .end method
  
***************
*** 1349,1359 ****
--- 1542,1554 ----
      .parameter "listener"
  
      .prologue
+     .line 202
      iget-object v0, p0, Landroid/widget/TableLayout;->mPassThroughListener:Landroid/widget/TableLayout$PassThroughHierarchyChangeListener;
  
      #setter for: Landroid/widget/TableLayout$PassThroughHierarchyChangeListener;->mOnHierarchyChangeListener:Landroid/view/ViewGroup$OnHierarchyChangeListener;
      invoke-static {v0, p1}, Landroid/widget/TableLayout$PassThroughHierarchyChangeListener;->access$102(Landroid/widget/TableLayout$PassThroughHierarchyChangeListener;Landroid/view/ViewGroup$OnHierarchyChangeListener;)Landroid/view/ViewGroup$OnHierarchyChangeListener;
  
+     .line 203
      return-void
  .end method
  
***************
*** 1362,1369 ****
--- 1557,1566 ----
      .parameter "shrinkAllColumns"
  
      .prologue
+     .line 246
      iput-boolean p1, p0, Landroid/widget/TableLayout;->mShrinkAllColumns:Z
  
+     .line 247
      return-void
  .end method
  
***************
*** 1372,1378 ****
--- 1569,1577 ----
      .parameter "stretchAllColumns"
  
      .prologue
+     .line 266
      iput-boolean p1, p0, Landroid/widget/TableLayout;->mStretchAllColumns:Z
  
+     .line 267
      return-void
  .end method
