*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabWidget.smali	2012-03-23 19:35:59.451670985 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TabWidget.smali.original	2012-03-23 19:34:44.933312145 -0500
***************
*** 42,51 ****
--- 42,53 ----
      .parameter "context"
  
      .prologue
+     .line 72
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/widget/TabWidget;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 73
      return-void
  .end method
  
***************
*** 55,64 ****
--- 57,68 ----
      .parameter "attrs"
  
      .prologue
+     .line 76
      const v0, 0x1010083
  
      invoke-direct {p0, p1, p2, v0}, Landroid/widget/TabWidget;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 77
      return-void
  .end method
  
***************
*** 73,92 ****
--- 77,102 ----
  
      const/4 v3, 0x1
  
+     .line 80
      invoke-direct {p0, p1, p2, p3}, Landroid/widget/LinearLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 57
      iput v2, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
+     .line 62
      iput-boolean v3, p0, Landroid/widget/TabWidget;->mDrawBottomStrips:Z
  
+     .line 65
      new-instance v1, Landroid/graphics/Rect;
  
      invoke-direct {v1}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v1, p0, Landroid/widget/TabWidget;->mBounds:Landroid/graphics/Rect;
  
+     .line 68
      iput v2, p0, Landroid/widget/TabWidget;->mImposedTabsHeight:I
  
+     .line 82
      sget-object v1, Lcom/android/internal/R$styleable;->TabWidget:[I
  
      const/4 v2, 0x0
***************
*** 95,100 ****
--- 105,111 ----
  
      move-result-object v0
  
+     .line 85
      .local v0, a:Landroid/content/res/TypedArray;
      const/4 v1, 0x3
  
***************
*** 104,115 ****
--- 115,128 ----
  
      invoke-virtual {p0, v1}, Landroid/widget/TabWidget;->setStripEnabled(Z)V
  
+     .line 86
      invoke-virtual {v0, v3}, Landroid/content/res/TypedArray;->getDrawable(I)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
  
      invoke-virtual {p0, v1}, Landroid/widget/TabWidget;->setLeftStripDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 87
      const/4 v1, 0x2
  
      invoke-virtual {v0, v1}, Landroid/content/res/TypedArray;->getDrawable(I)Landroid/graphics/drawable/Drawable;
***************
*** 118,127 ****
--- 131,143 ----
  
      invoke-virtual {p0, v1}, Landroid/widget/TabWidget;->setRightStripDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 89
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 91
      invoke-direct {p0}, Landroid/widget/TabWidget;->initTabWidget()V
  
+     .line 92
      return-void
  .end method
  
***************
*** 130,135 ****
--- 146,152 ----
      .parameter "x0"
  
      .prologue
+     .line 53
      iget-object v0, p0, Landroid/widget/TabWidget;->mSelectionChangedListener:Landroid/widget/TabWidget$OnTabSelectionChanged;
  
      return-object v0
***************
*** 141,155 ****
--- 158,176 ----
      .prologue
      const/4 v4, 0x1
  
+     .line 118
      invoke-virtual {p0, v4}, Landroid/widget/TabWidget;->setChildrenDrawingOrderEnabled(Z)V
  
+     .line 120
      iget-object v0, p0, Landroid/widget/TabWidget;->mContext:Landroid/content/Context;
  
+     .line 121
      .local v0, context:Landroid/content/Context;
      invoke-virtual {v0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v1
  
+     .line 125
      .local v1, resources:Landroid/content/res/Resources;
      invoke-virtual {v0}, Landroid/content/Context;->getApplicationInfo()Landroid/content/pm/ApplicationInfo;
  
***************
*** 161,170 ****
--- 182,193 ----
  
      if-gt v2, v3, :cond_2
  
+     .line 127
      iget-object v2, p0, Landroid/widget/TabWidget;->mLeftStrip:Landroid/graphics/drawable/Drawable;
  
      if-nez v2, :cond_0
  
+     .line 128
      const v2, 0x1080576
  
      invoke-virtual {v1, v2}, Landroid/content/res/Resources;->getDrawable(I)Landroid/graphics/drawable/Drawable;
***************
*** 173,183 ****
--- 196,208 ----
  
      iput-object v2, p0, Landroid/widget/TabWidget;->mLeftStrip:Landroid/graphics/drawable/Drawable;
  
+     .line 131
      :cond_0
      iget-object v2, p0, Landroid/widget/TabWidget;->mRightStrip:Landroid/graphics/drawable/Drawable;
  
      if-nez v2, :cond_1
  
+     .line 132
      const v2, 0x1080578
  
      invoke-virtual {v1, v2}, Landroid/content/res/Resources;->getDrawable(I)Landroid/graphics/drawable/Drawable;
***************
*** 186,204 ****
--- 211,234 ----
  
      iput-object v2, p0, Landroid/widget/TabWidget;->mRightStrip:Landroid/graphics/drawable/Drawable;
  
+     .line 149
      :cond_1
      :goto_0
      invoke-virtual {p0, v4}, Landroid/widget/TabWidget;->setFocusable(Z)V
  
+     .line 150
      invoke-virtual {p0, p0}, Landroid/widget/TabWidget;->setOnFocusChangeListener(Landroid/view/View$OnFocusChangeListener;)V
  
+     .line 151
      return-void
  
+     .line 137
      :cond_2
      iget-object v2, p0, Landroid/widget/TabWidget;->mLeftStrip:Landroid/graphics/drawable/Drawable;
  
      if-nez v2, :cond_3
  
+     .line 138
      const v2, 0x1080575
  
      invoke-virtual {v1, v2}, Landroid/content/res/Resources;->getDrawable(I)Landroid/graphics/drawable/Drawable;
***************
*** 207,217 ****
--- 237,249 ----
  
      iput-object v2, p0, Landroid/widget/TabWidget;->mLeftStrip:Landroid/graphics/drawable/Drawable;
  
+     .line 141
      :cond_3
      iget-object v2, p0, Landroid/widget/TabWidget;->mRightStrip:Landroid/graphics/drawable/Drawable;
  
      if-nez v2, :cond_1
  
+     .line 142
      const v2, 0x1080577
  
      invoke-virtual {v1, v2}, Landroid/content/res/Resources;->getDrawable(I)Landroid/graphics/drawable/Drawable;
***************
*** 234,245 ****
--- 266,279 ----
  
      const/4 v3, 0x0
  
+     .line 462
      invoke-virtual {p1}, Landroid/view/View;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
  
      move-result-object v1
  
      if-nez v1, :cond_0
  
+     .line 463
      new-instance v0, Landroid/widget/LinearLayout$LayoutParams;
  
      const/4 v1, -0x1
***************
*** 248,266 ****
--- 282,306 ----
  
      invoke-direct {v0, v3, v1, v2}, Landroid/widget/LinearLayout$LayoutParams;-><init>(IIF)V
  
+     .line 466
      .local v0, lp:Landroid/widget/LinearLayout$LayoutParams;
      invoke-virtual {v0, v3, v3, v3, v3}, Landroid/widget/LinearLayout$LayoutParams;->setMargins(IIII)V
  
+     .line 467
      invoke-virtual {p1, v0}, Landroid/view/View;->setLayoutParams(Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 471
      .end local v0           #lp:Landroid/widget/LinearLayout$LayoutParams;
      :cond_0
      invoke-virtual {p1, v4}, Landroid/view/View;->setFocusable(Z)V
  
+     .line 472
      invoke-virtual {p1, v4}, Landroid/view/View;->setClickable(Z)V
  
+     .line 474
      invoke-super {p0, p1}, Landroid/widget/LinearLayout;->addView(Landroid/view/View;)V
  
+     .line 478
      new-instance v1, Landroid/widget/TabWidget$TabClickListener;
  
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getTabCount()I
***************
*** 275,282 ****
--- 315,324 ----
  
      invoke-virtual {p1, v1}, Landroid/view/View;->setOnClickListener(Landroid/view/View$OnClickListener;)V
  
+     .line 479
      invoke-virtual {p1, p0}, Landroid/view/View;->setOnFocusChangeListener(Landroid/view/View$OnFocusChangeListener;)V
  
+     .line 480
      return-void
  .end method
  
***************
*** 285,290 ****
--- 327,333 ----
      .parameter "child"
  
      .prologue
+     .line 314
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getTabCount()I
  
      move-result v0
***************
*** 299,309 ****
--- 342,355 ----
  
      if-ne p1, v0, :cond_0
  
+     .line 316
      invoke-virtual {p0}, Landroid/widget/TabWidget;->invalidate()V
  
+     .line 318
      :cond_0
      invoke-super {p0, p1}, Landroid/widget/LinearLayout;->childDrawableStateChanged(Landroid/view/View;)V
  
+     .line 319
      return-void
  .end method
  
***************
*** 314,348 ****
--- 360,402 ----
      .prologue
      const/4 v10, 0x0
  
+     .line 323
      invoke-super {p0, p1}, Landroid/widget/LinearLayout;->dispatchDraw(Landroid/graphics/Canvas;)V
  
+     .line 326
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getTabCount()I
  
      move-result v5
  
      if-nez v5, :cond_1
  
+     .line 357
      :cond_0
      :goto_0
      return-void
  
+     .line 330
      :cond_1
      iget-boolean v5, p0, Landroid/widget/TabWidget;->mDrawBottomStrips:Z
  
      if-eqz v5, :cond_0
  
+     .line 335
      iget v5, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      invoke-virtual {p0, v5}, Landroid/widget/TabWidget;->getChildTabViewAt(I)Landroid/view/View;
  
      move-result-object v4
  
+     .line 337
      .local v4, selectedChild:Landroid/view/View;
      iget-object v1, p0, Landroid/widget/TabWidget;->mLeftStrip:Landroid/graphics/drawable/Drawable;
  
+     .line 338
      .local v1, leftStrip:Landroid/graphics/drawable/Drawable;
      iget-object v3, p0, Landroid/widget/TabWidget;->mRightStrip:Landroid/graphics/drawable/Drawable;
  
+     .line 340
      .local v3, rightStrip:Landroid/graphics/drawable/Drawable;
      invoke-virtual {v4}, Landroid/view/View;->getDrawableState()[I
  
***************
*** 350,367 ****
--- 404,425 ----
  
      invoke-virtual {v1, v5}, Landroid/graphics/drawable/Drawable;->setState([I)Z
  
+     .line 341
      invoke-virtual {v4}, Landroid/view/View;->getDrawableState()[I
  
      move-result-object v5
  
      invoke-virtual {v3, v5}, Landroid/graphics/drawable/Drawable;->setState([I)Z
  
+     .line 343
      iget-boolean v5, p0, Landroid/widget/TabWidget;->mStripMoved:Z
  
      if-eqz v5, :cond_2
  
+     .line 344
      iget-object v0, p0, Landroid/widget/TabWidget;->mBounds:Landroid/graphics/Rect;
  
+     .line 345
      .local v0, bounds:Landroid/graphics/Rect;
      invoke-virtual {v4}, Landroid/view/View;->getLeft()I
  
***************
*** 369,384 ****
--- 427,445 ----
  
      iput v5, v0, Landroid/graphics/Rect;->left:I
  
+     .line 346
      invoke-virtual {v4}, Landroid/view/View;->getRight()I
  
      move-result v5
  
      iput v5, v0, Landroid/graphics/Rect;->right:I
  
+     .line 347
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getHeight()I
  
      move-result v2
  
+     .line 348
      .local v2, myHeight:I
      iget v5, v0, Landroid/graphics/Rect;->left:I
  
***************
*** 402,407 ****
--- 463,469 ----
  
      invoke-virtual {v1, v5, v6, v7, v2}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
  
+     .line 350
      iget v5, v0, Landroid/graphics/Rect;->right:I
  
      invoke-virtual {v3}, Landroid/graphics/drawable/Drawable;->getIntrinsicHeight()I
***************
*** 428,440 ****
--- 490,505 ----
  
      invoke-virtual {v3, v5, v6, v7, v2}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
  
+     .line 352
      iput-boolean v10, p0, Landroid/widget/TabWidget;->mStripMoved:Z
  
+     .line 355
      .end local v0           #bounds:Landroid/graphics/Rect;
      .end local v2           #myHeight:I
      :cond_2
      invoke-virtual {v1, p1}, Landroid/graphics/drawable/Drawable;->draw(Landroid/graphics/Canvas;)V
  
+     .line 356
      invoke-virtual {v3, p1}, Landroid/graphics/drawable/Drawable;->draw(Landroid/graphics/Canvas;)V
  
      goto :goto_0
***************
*** 445,464 ****
--- 510,533 ----
      .parameter "event"
  
      .prologue
+     .line 405
      invoke-virtual {p0, p1}, Landroid/widget/TabWidget;->onPopulateAccessibilityEvent(Landroid/view/accessibility/AccessibilityEvent;)V
  
+     .line 407
      iget v1, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      const/4 v2, -0x1
  
      if-eq v1, v2, :cond_0
  
+     .line 408
      iget v1, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      invoke-virtual {p0, v1}, Landroid/widget/TabWidget;->getChildTabViewAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 409
      .local v0, tabView:Landroid/view/View;
      if-eqz v0, :cond_0
  
***************
*** 468,477 ****
--- 537,548 ----
  
      if-nez v1, :cond_0
  
+     .line 410
      invoke-virtual {v0, p1}, Landroid/view/View;->dispatchPopulateAccessibilityEvent(Landroid/view/accessibility/AccessibilityEvent;)Z
  
      move-result v1
  
+     .line 413
      .end local v0           #tabView:Landroid/view/View;
      :goto_0
      return v1
***************
*** 487,505 ****
--- 558,581 ----
      .parameter "index"
  
      .prologue
+     .line 438
      iget v0, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
+     .line 441
      .local v0, oldTab:I
      invoke-virtual {p0, p1}, Landroid/widget/TabWidget;->setCurrentTab(I)V
  
+     .line 444
      if-eq v0, p1, :cond_0
  
+     .line 445
      invoke-virtual {p0, p1}, Landroid/widget/TabWidget;->getChildTabViewAt(I)Landroid/view/View;
  
      move-result-object v1
  
      invoke-virtual {v1}, Landroid/view/View;->requestFocus()Z
  
+     .line 447
      :cond_0
      return-void
  .end method
***************
*** 510,541 ****
--- 586,623 ----
      .parameter "i"
  
      .prologue
+     .line 102
      iget v0, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      const/4 v1, -0x1
  
      if-ne v0, v1, :cond_1
  
+     .line 112
      .end local p2
      :cond_0
      :goto_0
      return p2
  
+     .line 107
      .restart local p2
      :cond_1
      add-int/lit8 v0, p1, -0x1
  
      if-ne p2, v0, :cond_2
  
+     .line 108
      iget p2, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      goto :goto_0
  
+     .line 109
      :cond_2
      iget v0, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      if-lt p2, v0, :cond_0
  
+     .line 110
      add-int/lit8 p2, p2, 0x1
  
      goto :goto_0
***************
*** 546,551 ****
--- 628,634 ----
      .parameter "index"
  
      .prologue
+     .line 221
      invoke-virtual {p0, p1}, Landroid/widget/TabWidget;->getChildAt(I)Landroid/view/View;
  
      move-result-object v0
***************
*** 557,562 ****
--- 640,646 ----
      .locals 1
  
      .prologue
+     .line 229
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getChildCount()I
  
      move-result v0
***************
*** 568,573 ****
--- 652,658 ----
      .locals 1
  
      .prologue
+     .line 309
      iget-boolean v0, p0, Landroid/widget/TabWidget;->mDrawBottomStrips:Z
  
      return v0
***************
*** 585,590 ****
--- 670,676 ----
      .prologue
      const/high16 v1, 0x4000
  
+     .line 157
      invoke-virtual {p0}, Landroid/widget/TabWidget;->isMeasureWithLargestChildEnabled()Z
  
      move-result v0
***************
*** 595,600 ****
--- 681,687 ----
  
      if-ltz v0, :cond_0
  
+     .line 158
      iget-object v0, p0, Landroid/widget/TabWidget;->mImposedTabWidths:[I
  
      aget v0, v0, p2
***************
*** 605,619 ****
--- 692,709 ----
  
      move-result p3
  
+     .line 160
      iget v0, p0, Landroid/widget/TabWidget;->mImposedTabsHeight:I
  
      invoke-static {v0, v1}, Landroid/view/View$MeasureSpec;->makeMeasureSpec(II)I
  
      move-result p5
  
+     .line 164
      :cond_0
      invoke-super/range {p0 .. p6}, Landroid/widget/LinearLayout;->measureChildBeforeLayout(Landroid/view/View;IIIII)V
  
+     .line 166
      return-void
  .end method
  
***************
*** 627,655 ****
--- 717,752 ----
  
      const/4 v11, 0x0
  
+     .line 170
      invoke-static {p1}, Landroid/view/View$MeasureSpec;->getMode(I)I
  
      move-result v9
  
      if-nez v9, :cond_0
  
+     .line 171
      invoke-super {p0, p1, p2}, Landroid/widget/LinearLayout;->measureHorizontal(II)V
  
+     .line 212
      :goto_0
      return-void
  
+     .line 176
      :cond_0
      invoke-static {v11, v11}, Landroid/view/View$MeasureSpec;->makeMeasureSpec(II)I
  
      move-result v8
  
+     .line 177
      .local v8, unspecifiedWidth:I
      const/4 v9, -0x1
  
      iput v9, p0, Landroid/widget/TabWidget;->mImposedTabsHeight:I
  
+     .line 178
      invoke-super {p0, v8, p2}, Landroid/widget/LinearLayout;->measureHorizontal(II)V
  
+     .line 180
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getMeasuredWidth()I
  
      move-result v9
***************
*** 660,675 ****
--- 757,776 ----
  
      sub-int v5, v9, v10
  
+     .line 181
      .local v5, extraWidth:I
      if-lez v5, :cond_6
  
+     .line 182
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getChildCount()I
  
      move-result v3
  
+     .line 184
      .local v3, count:I
      const/4 v1, 0x0
  
+     .line 185
      .local v1, childCount:I
      const/4 v6, 0x0
  
***************
*** 677,686 ****
--- 778,789 ----
      :goto_1
      if-ge v6, v3, :cond_2
  
+     .line 186
      invoke-virtual {p0, v6}, Landroid/widget/TabWidget;->getChildAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 187
      .local v0, child:Landroid/view/View;
      invoke-virtual {v0}, Landroid/view/View;->getVisibility()I
  
***************
*** 688,707 ****
--- 791,814 ----
  
      if-ne v9, v12, :cond_1
  
+     .line 185
      :goto_2
      add-int/lit8 v6, v6, 0x1
  
      goto :goto_1
  
+     .line 188
      :cond_1
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_2
  
+     .line 191
      .end local v0           #child:Landroid/view/View;
      :cond_2
      if-lez v1, :cond_6
  
+     .line 192
      iget-object v9, p0, Landroid/widget/TabWidget;->mImposedTabWidths:[I
  
      if-eqz v9, :cond_3
***************
*** 712,732 ****
--- 819,843 ----
  
      if-eq v9, v3, :cond_4
  
+     .line 193
      :cond_3
      new-array v9, v3, [I
  
      iput-object v9, p0, Landroid/widget/TabWidget;->mImposedTabWidths:[I
  
+     .line 195
      :cond_4
      const/4 v6, 0x0
  
      :goto_3
      if-ge v6, v3, :cond_6
  
+     .line 196
      invoke-virtual {p0, v6}, Landroid/widget/TabWidget;->getChildAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 197
      .restart local v0       #child:Landroid/view/View;
      invoke-virtual {v0}, Landroid/view/View;->getVisibility()I
  
***************
*** 734,752 ****
--- 845,867 ----
  
      if-ne v9, v12, :cond_5
  
+     .line 195
      :goto_4
      add-int/lit8 v6, v6, 0x1
  
      goto :goto_3
  
+     .line 198
      :cond_5
      invoke-virtual {v0}, Landroid/view/View;->getMeasuredWidth()I
  
      move-result v2
  
+     .line 199
      .local v2, childWidth:I
      div-int v4, v5, v1
  
+     .line 200
      .local v4, delta:I
      sub-int v9, v2, v4
  
***************
*** 754,770 ****
--- 869,889 ----
  
      move-result v7
  
+     .line 201
      .local v7, newWidth:I
      iget-object v9, p0, Landroid/widget/TabWidget;->mImposedTabWidths:[I
  
      aput v7, v9, v6
  
+     .line 203
      sub-int v9, v2, v7
  
      sub-int/2addr v5, v9
  
+     .line 204
      add-int/lit8 v1, v1, -0x1
  
+     .line 205
      iget v9, p0, Landroid/widget/TabWidget;->mImposedTabsHeight:I
  
      invoke-virtual {v0}, Landroid/view/View;->getMeasuredHeight()I
***************
*** 779,784 ****
--- 898,904 ----
  
      goto :goto_4
  
+     .line 211
      .end local v0           #child:Landroid/view/View;
      .end local v1           #childCount:I
      .end local v2           #childWidth:I
***************
*** 798,803 ****
--- 918,924 ----
      .parameter "hasFocus"
  
      .prologue
+     .line 507
      if-ne p1, p0, :cond_1
  
      if-eqz p2, :cond_1
***************
*** 808,813 ****
--- 929,935 ----
  
      if-lez v2, :cond_1
  
+     .line 508
      iget v2, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      invoke-virtual {p0, v2}, Landroid/widget/TabWidget;->getChildTabViewAt(I)Landroid/view/View;
***************
*** 816,865 ****
--- 938,998 ----
  
      invoke-virtual {v2}, Landroid/view/View;->requestFocus()Z
  
+     .line 528
      :cond_0
      :goto_0
      return-void
  
+     .line 512
      :cond_1
      if-eqz p2, :cond_0
  
+     .line 513
      const/4 v0, 0x0
  
+     .line 514
      .local v0, i:I
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getTabCount()I
  
      move-result v1
  
+     .line 515
      .local v1, numTabs:I
      :goto_1
      if-ge v0, v1, :cond_0
  
+     .line 516
      invoke-virtual {p0, v0}, Landroid/widget/TabWidget;->getChildTabViewAt(I)Landroid/view/View;
  
      move-result-object v2
  
      if-ne v2, p1, :cond_2
  
+     .line 517
      invoke-virtual {p0, v0}, Landroid/widget/TabWidget;->setCurrentTab(I)V
  
+     .line 518
      iget-object v2, p0, Landroid/widget/TabWidget;->mSelectionChangedListener:Landroid/widget/TabWidget$OnTabSelectionChanged;
  
      const/4 v3, 0x0
  
      invoke-interface {v2, v0, v3}, Landroid/widget/TabWidget$OnTabSelectionChanged;->onTabSelectionChanged(IZ)V
  
+     .line 519
      invoke-virtual {p0}, Landroid/widget/TabWidget;->isShown()Z
  
      move-result v2
  
      if-eqz v2, :cond_0
  
+     .line 521
      const/16 v2, 0x8
  
      invoke-virtual {p0, v2}, Landroid/widget/TabWidget;->sendAccessibilityEvent(I)V
  
      goto :goto_0
  
+     .line 525
      :cond_2
      add-int/lit8 v0, v0, 0x1
  
***************
*** 871,888 ****
--- 1004,1025 ----
      .parameter "event"
  
      .prologue
+     .line 418
      invoke-super {p0, p1}, Landroid/widget/LinearLayout;->onInitializeAccessibilityEvent(Landroid/view/accessibility/AccessibilityEvent;)V
  
+     .line 419
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getTabCount()I
  
      move-result v0
  
      invoke-virtual {p1, v0}, Landroid/view/accessibility/AccessibilityEvent;->setItemCount(I)V
  
+     .line 420
      iget v0, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      invoke-virtual {p1, v0}, Landroid/view/accessibility/AccessibilityEvent;->setCurrentItemIndex(I)V
  
+     .line 421
      return-void
  .end method
  
***************
*** 894,905 ****
--- 1031,1045 ----
      .parameter "oldh"
  
      .prologue
+     .line 96
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/widget/TabWidget;->mStripMoved:Z
  
+     .line 97
      invoke-super {p0, p1, p2, p3, p4}, Landroid/widget/LinearLayout;->onSizeChanged(IIII)V
  
+     .line 98
      return-void
  .end method
  
***************
*** 907,918 ****
--- 1047,1061 ----
      .locals 1
  
      .prologue
+     .line 484
      invoke-super {p0}, Landroid/widget/LinearLayout;->removeAllViews()V
  
+     .line 485
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
+     .line 486
      return-void
  .end method
  
***************
*** 921,926 ****
--- 1064,1070 ----
      .parameter "event"
  
      .prologue
+     .line 491
      invoke-virtual {p1}, Landroid/view/accessibility/AccessibilityEvent;->getEventType()I
  
      move-result v0
***************
*** 935,945 ****
--- 1079,1092 ----
  
      if-eqz v0, :cond_0
  
+     .line 492
      invoke-virtual {p1}, Landroid/view/accessibility/AccessibilityEvent;->recycle()V
  
+     .line 496
      :goto_0
      return-void
  
+     .line 495
      :cond_0
      invoke-super {p0, p1}, Landroid/widget/LinearLayout;->sendAccessibilityEventUnchecked(Landroid/view/accessibility/AccessibilityEvent;)V
  
***************
*** 953,958 ****
--- 1100,1106 ----
      .prologue
      const/4 v2, 0x1
  
+     .line 387
      if-ltz p1, :cond_0
  
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getTabCount()I
***************
*** 965,974 ****
--- 1113,1124 ----
  
      if-ne p1, v0, :cond_1
  
+     .line 401
      :cond_0
      :goto_0
      return-void
  
+     .line 391
      :cond_1
      iget v0, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
***************
*** 976,981 ****
--- 1126,1132 ----
  
      if-eq v0, v1, :cond_2
  
+     .line 392
      iget v0, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      invoke-virtual {p0, v0}, Landroid/widget/TabWidget;->getChildTabViewAt(I)Landroid/view/View;
***************
*** 986,994 ****
--- 1137,1147 ----
  
      invoke-virtual {v0, v1}, Landroid/view/View;->setSelected(Z)V
  
+     .line 394
      :cond_2
      iput p1, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
+     .line 395
      iget v0, p0, Landroid/widget/TabWidget;->mSelectedTab:I
  
      invoke-virtual {p0, v0}, Landroid/widget/TabWidget;->getChildTabViewAt(I)Landroid/view/View;
***************
*** 997,1010 ****
--- 1150,1166 ----
  
      invoke-virtual {v0, v2}, Landroid/view/View;->setSelected(Z)V
  
+     .line 396
      iput-boolean v2, p0, Landroid/widget/TabWidget;->mStripMoved:Z
  
+     .line 398
      invoke-virtual {p0}, Landroid/widget/TabWidget;->isShown()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 399
      const/4 v0, 0x4
  
      invoke-virtual {p0, v0}, Landroid/widget/TabWidget;->sendAccessibilityEvent(I)V
***************
*** 1017,1022 ****
--- 1173,1179 ----
      .parameter "resId"
  
      .prologue
+     .line 247
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getResources()Landroid/content/res/Resources;
  
      move-result-object v0
***************
*** 1027,1032 ****
--- 1184,1190 ----
  
      invoke-virtual {p0, v0}, Landroid/widget/TabWidget;->setDividerDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 248
      return-void
  .end method
  
***************
*** 1035,1042 ****
--- 1193,1202 ----
      .parameter "drawable"
  
      .prologue
+     .line 238
      invoke-super {p0, p1}, Landroid/widget/LinearLayout;->setDividerDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 239
      return-void
  .end method
  
***************
*** 1045,1056 ****
--- 1205,1219 ----
      .parameter "enabled"
  
      .prologue
+     .line 451
      invoke-super {p0, p1}, Landroid/widget/LinearLayout;->setEnabled(Z)V
  
+     .line 453
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getTabCount()I
  
      move-result v1
  
+     .line 454
      .local v1, count:I
      const/4 v2, 0x0
  
***************
*** 1058,1074 ****
--- 1221,1241 ----
      :goto_0
      if-ge v2, v1, :cond_0
  
+     .line 455
      invoke-virtual {p0, v2}, Landroid/widget/TabWidget;->getChildTabViewAt(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 456
      .local v0, child:Landroid/view/View;
      invoke-virtual {v0, p1}, Landroid/view/View;->setEnabled(Z)V
  
+     .line 454
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 458
      .end local v0           #child:Landroid/view/View;
      :cond_0
      return-void
***************
*** 1079,1084 ****
--- 1246,1252 ----
      .parameter "resId"
  
      .prologue
+     .line 268
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getResources()Landroid/content/res/Resources;
  
      move-result-object v0
***************
*** 1089,1094 ****
--- 1257,1263 ----
  
      invoke-virtual {p0, v0}, Landroid/widget/TabWidget;->setLeftStripDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 269
      return-void
  .end method
  
***************
*** 1097,1108 ****
--- 1266,1281 ----
      .parameter "drawable"
  
      .prologue
+     .line 256
      iput-object p1, p0, Landroid/widget/TabWidget;->mLeftStrip:Landroid/graphics/drawable/Drawable;
  
+     .line 257
      invoke-virtual {p0}, Landroid/widget/TabWidget;->requestLayout()V
  
+     .line 258
      invoke-virtual {p0}, Landroid/widget/TabWidget;->invalidate()V
  
+     .line 259
      return-void
  .end method
  
***************
*** 1111,1116 ****
--- 1284,1290 ----
      .parameter "resId"
  
      .prologue
+     .line 289
      invoke-virtual {p0}, Landroid/widget/TabWidget;->getResources()Landroid/content/res/Resources;
  
      move-result-object v0
***************
*** 1121,1126 ****
--- 1295,1301 ----
  
      invoke-virtual {p0, v0}, Landroid/widget/TabWidget;->setRightStripDrawable(Landroid/graphics/drawable/Drawable;)V
  
+     .line 290
      return-void
  .end method
  
***************
*** 1129,1140 ****
--- 1304,1319 ----
      .parameter "drawable"
  
      .prologue
+     .line 277
      iput-object p1, p0, Landroid/widget/TabWidget;->mRightStrip:Landroid/graphics/drawable/Drawable;
  
+     .line 278
      invoke-virtual {p0}, Landroid/widget/TabWidget;->requestLayout()V
  
+     .line 279
      invoke-virtual {p0}, Landroid/widget/TabWidget;->invalidate()V
  
+     .line 280
      return-void
  .end method
  
***************
*** 1143,1152 ****
--- 1322,1334 ----
      .parameter "stripEnabled"
  
      .prologue
+     .line 300
      iput-boolean p1, p0, Landroid/widget/TabWidget;->mDrawBottomStrips:Z
  
+     .line 301
      invoke-virtual {p0}, Landroid/widget/TabWidget;->invalidate()V
  
+     .line 302
      return-void
  .end method
  
***************
*** 1155,1161 ****
--- 1337,1345 ----
      .parameter "listener"
  
      .prologue
+     .line 502
      iput-object p1, p0, Landroid/widget/TabWidget;->mSelectionChangedListener:Landroid/widget/TabWidget$OnTabSelectionChanged;
  
+     .line 503
      return-void
  .end method
