*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$InsertionHandleView$1.smali	2012-03-23 19:35:52.999553331 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$InsertionHandleView$1.smali.original	2012-03-23 19:34:44.885311270 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 10737
      iput-object p1, p0, Landroid/widget/TextView$InsertionHandleView$1;->this$1:Landroid/widget/TextView$InsertionHandleView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,48 ****
--- 41,51 ----
      .locals 1
  
      .prologue
+     .line 10739
      iget-object v0, p0, Landroid/widget/TextView$InsertionHandleView$1;->this$1:Landroid/widget/TextView$InsertionHandleView;
  
      invoke-virtual {v0}, Landroid/widget/TextView$InsertionHandleView;->hide()V
  
+     .line 10740
      return-void
  .end method
