*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$SavedState.smali	2012-03-23 19:35:59.954680157 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$SavedState.smali.original	2012-03-23 19:34:44.937312218 -0500
***************
*** 44,49 ****
--- 44,50 ----
      .locals 1
  
      .prologue
+     .line 2930
      new-instance v0, Landroid/widget/TextView$SavedState$1;
  
      invoke-direct {v0}, Landroid/widget/TextView$SavedState$1;-><init>()V
***************
*** 58,77 ****
--- 59,82 ----
      .parameter "in"
  
      .prologue
+     .line 2942
      invoke-direct {p0, p1}, Landroid/view/View$BaseSavedState;-><init>(Landroid/os/Parcel;)V
  
+     .line 2943
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/widget/TextView$SavedState;->selStart:I
  
+     .line 2944
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/widget/TextView$SavedState;->selEnd:I
  
+     .line 2945
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
***************
*** 83,88 ****
--- 88,94 ----
      :goto_0
      iput-boolean v0, p0, Landroid/widget/TextView$SavedState;->frozenWithFocus:Z
  
+     .line 2946
      sget-object v0, Landroid/text/TextUtils;->CHAR_SEQUENCE_CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 93,104 ****
--- 99,112 ----
  
      iput-object v0, p0, Landroid/widget/TextView$SavedState;->text:Ljava/lang/CharSequence;
  
+     .line 2948
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 2949
      sget-object v0, Landroid/text/TextUtils;->CHAR_SEQUENCE_CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 109,117 ****
--- 117,127 ----
  
      iput-object v0, p0, Landroid/widget/TextView$SavedState;->error:Ljava/lang/CharSequence;
  
+     .line 2951
      :cond_0
      return-void
  
+     .line 2945
      :cond_1
      const/4 v0, 0x0
  
***************
*** 124,129 ****
--- 134,140 ----
      .parameter "x1"
  
      .prologue
+     .line 2891
      invoke-direct {p0, p1}, Landroid/widget/TextView$SavedState;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 134,141 ****
--- 145,154 ----
      .parameter "superState"
  
      .prologue
+     .line 2899
      invoke-direct {p0, p1}, Landroid/view/View$BaseSavedState;-><init>(Landroid/os/Parcelable;)V
  
+     .line 2900
      return-void
  .end method
  
***************
*** 145,150 ****
--- 158,164 ----
      .locals 3
  
      .prologue
+     .line 2920
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 195,205 ****
--- 209,221 ----
  
      move-result-object v0
  
+     .line 2923
      .local v0, str:Ljava/lang/String;
      iget-object v1, p0, Landroid/widget/TextView$SavedState;->text:Ljava/lang/CharSequence;
  
      if-eqz v1, :cond_0
  
+     .line 2924
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 224,229 ****
--- 240,246 ----
  
      move-result-object v0
  
+     .line 2926
      :cond_0
      new-instance v1, Ljava/lang/StringBuilder;
  
***************
*** 233,239 ****
  
      move-result-object v1
  
!     const-string v2, "}"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 250,256 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "}"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 256,271 ****
--- 273,292 ----
  
      const/4 v2, 0x0
  
+     .line 2904
      invoke-super {p0, p1, p2}, Landroid/view/View$BaseSavedState;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 2905
      iget v0, p0, Landroid/widget/TextView$SavedState;->selStart:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 2906
      iget v0, p0, Landroid/widget/TextView$SavedState;->selEnd:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 2907
      iget-boolean v0, p0, Landroid/widget/TextView$SavedState;->frozenWithFocus:Z
  
      if-eqz v0, :cond_0
***************
*** 275,301 ****
--- 296,329 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 2908
      iget-object v0, p0, Landroid/widget/TextView$SavedState;->text:Ljava/lang/CharSequence;
  
      invoke-static {v0, p1, p2}, Landroid/text/TextUtils;->writeToParcel(Ljava/lang/CharSequence;Landroid/os/Parcel;I)V
  
+     .line 2910
      iget-object v0, p0, Landroid/widget/TextView$SavedState;->error:Ljava/lang/CharSequence;
  
      if-nez v0, :cond_1
  
+     .line 2911
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 2916
      :goto_1
      return-void
  
      :cond_0
      move v0, v2
  
+     .line 2907
      goto :goto_0
  
+     .line 2913
      :cond_1
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 2914
      iget-object v0, p0, Landroid/widget/TextView$SavedState;->error:Ljava/lang/CharSequence;
  
      invoke-static {v0, p1, p2}, Landroid/text/TextUtils;->writeToParcel(Ljava/lang/CharSequence;Landroid/os/Parcel;I)V
