*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$Marquee.smali	2012-03-23 19:35:54.793586045 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$Marquee.smali.original	2012-03-23 19:34:44.899311525 -0500
***************
*** 75,86 ****
--- 75,89 ----
      .parameter "v"
  
      .prologue
+     .line 7425
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
  
+     .line 7414
      const/4 v1, 0x0
  
      iput-byte v1, p0, Landroid/widget/TextView$Marquee;->mStatus:B
  
+     .line 7426
      invoke-virtual {p1}, Landroid/widget/TextView;->getContext()Landroid/content/Context;
  
      move-result-object v1
***************
*** 95,100 ****
--- 98,104 ----
  
      iget v0, v1, Landroid/util/DisplayMetrics;->density:F
  
+     .line 7427
      .local v0, density:F
      const/high16 v1, 0x41f0
  
***************
*** 106,117 ****
--- 110,123 ----
  
      iput v1, p0, Landroid/widget/TextView$Marquee;->mScrollUnit:F
  
+     .line 7428
      new-instance v1, Ljava/lang/ref/WeakReference;
  
      invoke-direct {v1, p1}, Ljava/lang/ref/WeakReference;-><init>(Ljava/lang/Object;)V
  
      iput-object v1, p0, Landroid/widget/TextView$Marquee;->mView:Ljava/lang/ref/WeakReference;
  
+     .line 7429
      return-void
  .end method
  
***************
*** 119,128 ****
--- 125,136 ----
      .locals 2
  
      .prologue
+     .line 7481
      const/4 v1, 0x0
  
      iput v1, p0, Landroid/widget/TextView$Marquee;->mScroll:F
  
+     .line 7482
      iget-object v1, p0, Landroid/widget/TextView$Marquee;->mView:Ljava/lang/ref/WeakReference;
  
      invoke-virtual {v1}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
***************
*** 131,141 ****
--- 139,151 ----
  
      check-cast v0, Landroid/widget/TextView;
  
+     .line 7483
      .local v0, textView:Landroid/widget/TextView;
      if-eqz v0, :cond_0
  
      invoke-virtual {v0}, Landroid/widget/TextView;->invalidate()V
  
+     .line 7484
      :cond_0
      return-void
  .end method
***************
*** 146,151 ****
--- 156,162 ----
      .locals 1
  
      .prologue
+     .line 7512
      iget v0, p0, Landroid/widget/TextView$Marquee;->mGhostOffset:F
  
      return v0
***************
*** 158,198 ****
--- 169,218 ----
      .prologue
      const/4 v1, 0x2
  
+     .line 7433
      iget v0, p1, Landroid/os/Message;->what:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 7450
      :cond_0
      :goto_0
      return-void
  
+     .line 7435
      :pswitch_0
      iput-byte v1, p0, Landroid/widget/TextView$Marquee;->mStatus:B
  
+     .line 7436
      invoke-virtual {p0}, Landroid/widget/TextView$Marquee;->tick()V
  
      goto :goto_0
  
+     .line 7439
      :pswitch_1
      invoke-virtual {p0}, Landroid/widget/TextView$Marquee;->tick()V
  
      goto :goto_0
  
+     .line 7442
      :pswitch_2
      iget-byte v0, p0, Landroid/widget/TextView$Marquee;->mStatus:B
  
      if-ne v0, v1, :cond_0
  
+     .line 7443
      iget v0, p0, Landroid/widget/TextView$Marquee;->mRepeatLimit:I
  
      if-ltz v0, :cond_1
  
+     .line 7444
      iget v0, p0, Landroid/widget/TextView$Marquee;->mRepeatLimit:I
  
      add-int/lit8 v0, v0, -0x1
  
      iput v0, p0, Landroid/widget/TextView$Marquee;->mRepeatLimit:I
  
+     .line 7446
      :cond_1
      iget v0, p0, Landroid/widget/TextView$Marquee;->mRepeatLimit:I
  
***************
*** 200,205 ****
--- 220,226 ----
  
      goto :goto_0
  
+     .line 7433
      nop
  
      :pswitch_data_0
***************
*** 214,219 ****
--- 235,241 ----
      .locals 2
  
      .prologue
+     .line 7524
      iget-byte v0, p0, Landroid/widget/TextView$Marquee;->mStatus:B
  
      const/4 v1, 0x2
***************
*** 235,240 ****
--- 257,263 ----
      .locals 1
  
      .prologue
+     .line 7528
      iget-byte v0, p0, Landroid/widget/TextView$Marquee;->mStatus:B
  
      if-nez v0, :cond_0
***************
*** 254,259 ****
--- 277,283 ----
      .locals 2
  
      .prologue
+     .line 7520
      iget-byte v0, p0, Landroid/widget/TextView$Marquee;->mStatus:B
  
      const/4 v1, 0x2
***************
*** 283,288 ****
--- 307,313 ----
      .locals 2
  
      .prologue
+     .line 7516
      iget v0, p0, Landroid/widget/TextView$Marquee;->mScroll:F
  
      iget v1, p0, Landroid/widget/TextView$Marquee;->mFadeStop:F
***************
*** 309,325 ****
--- 334,355 ----
      .prologue
      const/4 v6, 0x1
  
+     .line 7487
      if-nez p1, :cond_1
  
+     .line 7488
      invoke-virtual {p0}, Landroid/widget/TextView$Marquee;->stop()V
  
+     .line 7509
      :cond_0
      :goto_0
      return-void
  
+     .line 7491
      :cond_1
      iput p1, p0, Landroid/widget/TextView$Marquee;->mRepeatLimit:I
  
+     .line 7492
      iget-object v4, p0, Landroid/widget/TextView$Marquee;->mView:Ljava/lang/ref/WeakReference;
  
      invoke-virtual {v4}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
***************
*** 328,333 ****
--- 358,364 ----
  
      check-cast v2, Landroid/widget/TextView;
  
+     .line 7493
      .local v2, textView:Landroid/widget/TextView;
      if-eqz v2, :cond_0
  
***************
*** 335,346 ****
--- 366,380 ----
  
      if-eqz v4, :cond_0
  
+     .line 7494
      iput-byte v6, p0, Landroid/widget/TextView$Marquee;->mStatus:B
  
+     .line 7495
      const/4 v4, 0x0
  
      iput v4, p0, Landroid/widget/TextView$Marquee;->mScroll:F
  
+     .line 7496
      invoke-virtual {v2}, Landroid/widget/TextView;->getWidth()I
  
      move-result v4
***************
*** 357,362 ****
--- 391,397 ----
  
      sub-int v3, v4, v5
  
+     .line 7498
      .local v3, textWidth:I
      iget-object v4, v2, Landroid/widget/TextView;->mLayout:Landroid/text/Layout;
  
***************
*** 366,371 ****
--- 401,407 ----
  
      move-result v1
  
+     .line 7499
      .local v1, lineWidth:F
      int-to-float v4, v3
  
***************
*** 373,378 ****
--- 409,415 ----
  
      div-float v0, v4, v5
  
+     .line 7500
      .local v0, gap:F
      int-to-float v4, v3
  
***************
*** 382,387 ****
--- 419,425 ----
  
      iput v4, p0, Landroid/widget/TextView$Marquee;->mGhostStart:F
  
+     .line 7501
      iget v4, p0, Landroid/widget/TextView$Marquee;->mGhostStart:F
  
      int-to-float v5, v3
***************
*** 390,399 ****
--- 428,439 ----
  
      iput v4, p0, Landroid/widget/TextView$Marquee;->mMaxScroll:F
  
+     .line 7502
      add-float v4, v1, v0
  
      iput v4, p0, Landroid/widget/TextView$Marquee;->mGhostOffset:F
  
+     .line 7503
      int-to-float v4, v3
  
      const/high16 v5, 0x40c0
***************
*** 404,409 ****
--- 444,450 ----
  
      iput v4, p0, Landroid/widget/TextView$Marquee;->mFadeStop:F
  
+     .line 7504
      iget v4, p0, Landroid/widget/TextView$Marquee;->mGhostStart:F
  
      add-float/2addr v4, v1
***************
*** 412,419 ****
--- 453,462 ----
  
      iput v4, p0, Landroid/widget/TextView$Marquee;->mMaxFadeScroll:F
  
+     .line 7506
      invoke-virtual {v2}, Landroid/widget/TextView;->invalidate()V
  
+     .line 7507
      const-wide/16 v4, 0x4b0
  
      invoke-virtual {p0, v6, v4, v5}, Landroid/widget/TextView$Marquee;->sendEmptyMessageDelayed(IJ)Z
***************
*** 425,448 ****
--- 468,497 ----
      .locals 1
  
      .prologue
+     .line 7473
      const/4 v0, 0x0
  
      iput-byte v0, p0, Landroid/widget/TextView$Marquee;->mStatus:B
  
+     .line 7474
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Landroid/widget/TextView$Marquee;->removeMessages(I)V
  
+     .line 7475
      const/4 v0, 0x3
  
      invoke-virtual {p0, v0}, Landroid/widget/TextView$Marquee;->removeMessages(I)V
  
+     .line 7476
      const/4 v0, 0x2
  
      invoke-virtual {p0, v0}, Landroid/widget/TextView$Marquee;->removeMessages(I)V
  
+     .line 7477
      invoke-direct {p0}, Landroid/widget/TextView$Marquee;->resetScroll()V
  
+     .line 7478
      return-void
  .end method
  
***************
*** 452,468 ****
--- 501,521 ----
      .prologue
      const/4 v3, 0x2
  
+     .line 7453
      iget-byte v1, p0, Landroid/widget/TextView$Marquee;->mStatus:B
  
      if-eq v1, v3, :cond_1
  
+     .line 7470
      :cond_0
      :goto_0
      return-void
  
+     .line 7457
      :cond_1
      invoke-virtual {p0, v3}, Landroid/widget/TextView$Marquee;->removeMessages(I)V
  
+     .line 7459
      iget-object v1, p0, Landroid/widget/TextView$Marquee;->mView:Ljava/lang/ref/WeakReference;
  
      invoke-virtual {v1}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
***************
*** 471,476 ****
--- 524,530 ----
  
      check-cast v0, Landroid/widget/TextView;
  
+     .line 7460
      .local v0, textView:Landroid/widget/TextView;
      if-eqz v0, :cond_0
  
***************
*** 486,491 ****
--- 540,546 ----
  
      if-eqz v1, :cond_0
  
+     .line 7461
      :cond_2
      iget v1, p0, Landroid/widget/TextView$Marquee;->mScroll:F
  
***************
*** 495,500 ****
--- 550,556 ----
  
      iput v1, p0, Landroid/widget/TextView$Marquee;->mScroll:F
  
+     .line 7462
      iget v1, p0, Landroid/widget/TextView$Marquee;->mScroll:F
  
      iget v2, p0, Landroid/widget/TextView$Marquee;->mMaxScroll:F
***************
*** 503,523 ****
--- 559,583 ----
  
      if-lez v1, :cond_3
  
+     .line 7463
      iget v1, p0, Landroid/widget/TextView$Marquee;->mMaxScroll:F
  
      iput v1, p0, Landroid/widget/TextView$Marquee;->mScroll:F
  
+     .line 7464
      const/4 v1, 0x3
  
      const-wide/16 v2, 0x4b0
  
      invoke-virtual {p0, v1, v2, v3}, Landroid/widget/TextView$Marquee;->sendEmptyMessageDelayed(IJ)Z
  
+     .line 7468
      :goto_1
      invoke-virtual {v0}, Landroid/widget/TextView;->invalidate()V
  
      goto :goto_0
  
+     .line 7466
      :cond_3
      const-wide/16 v1, 0x21
  
