*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$InsertionHandleView.smali	2012-03-23 19:36:02.058718524 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$InsertionHandleView.smali.original	2012-03-23 19:34:44.948312418 -0500
***************
*** 37,46 ****
--- 37,49 ----
      .parameter "drawable"
  
      .prologue
+     .line 10713
      iput-object p1, p0, Landroid/widget/TextView$InsertionHandleView;->this$0:Landroid/widget/TextView;
  
+     .line 10714
      invoke-direct {p0, p1, p2, p2}, Landroid/widget/TextView$HandleView;-><init>(Landroid/widget/TextView;Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;)V
  
+     .line 10715
      return-void
  .end method
  
***************
*** 48,65 ****
--- 51,72 ----
      .locals 4
  
      .prologue
+     .line 10735
      invoke-direct {p0}, Landroid/widget/TextView$InsertionHandleView;->removeHiderCallback()V
  
+     .line 10736
      iget-object v0, p0, Landroid/widget/TextView$InsertionHandleView;->mHider:Ljava/lang/Runnable;
  
      if-nez v0, :cond_0
  
+     .line 10737
      new-instance v0, Landroid/widget/TextView$InsertionHandleView$1;
  
      invoke-direct {v0, p0}, Landroid/widget/TextView$InsertionHandleView$1;-><init>(Landroid/widget/TextView$InsertionHandleView;)V
  
      iput-object v0, p0, Landroid/widget/TextView$InsertionHandleView;->mHider:Ljava/lang/Runnable;
  
+     .line 10743
      :cond_0
      iget-object v0, p0, Landroid/widget/TextView$InsertionHandleView;->this$0:Landroid/widget/TextView;
  
***************
*** 69,74 ****
--- 76,82 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/widget/TextView;->postDelayed(Ljava/lang/Runnable;J)Z
  
+     .line 10744
      return-void
  .end method
  
***************
*** 76,91 ****
--- 84,102 ----
      .locals 2
  
      .prologue
+     .line 10747
      iget-object v0, p0, Landroid/widget/TextView$InsertionHandleView;->mHider:Ljava/lang/Runnable;
  
      if-eqz v0, :cond_0
  
+     .line 10748
      iget-object v0, p0, Landroid/widget/TextView$InsertionHandleView;->this$0:Landroid/widget/TextView;
  
      iget-object v1, p0, Landroid/widget/TextView$InsertionHandleView;->mHider:Ljava/lang/Runnable;
  
      invoke-virtual {v0, v1}, Landroid/widget/TextView;->removeCallbacks(Ljava/lang/Runnable;)Z
  
+     .line 10750
      :cond_0
      return-void
  .end method
***************
*** 96,101 ****
--- 107,113 ----
      .locals 1
  
      .prologue
+     .line 10797
      iget-object v0, p0, Landroid/widget/TextView$InsertionHandleView;->this$0:Landroid/widget/TextView;
  
      invoke-virtual {v0}, Landroid/widget/TextView;->getSelectionStart()I
***************
*** 111,116 ****
--- 123,129 ----
      .parameter "isRtlRun"
  
      .prologue
+     .line 10754
      invoke-virtual {p1}, Landroid/graphics/drawable/Drawable;->getIntrinsicWidth()I
  
      move-result v0
***************
*** 124,133 ****
--- 137,149 ----
      .locals 0
  
      .prologue
+     .line 10818
      invoke-super {p0}, Landroid/widget/TextView$HandleView;->onDetached()V
  
+     .line 10819
      invoke-direct {p0}, Landroid/widget/TextView$InsertionHandleView;->removeHiderCallback()V
  
+     .line 10820
      return-void
  .end method
  
***************
*** 135,144 ****
--- 151,163 ----
      .locals 0
  
      .prologue
+     .line 10812
      invoke-super {p0}, Landroid/widget/TextView$HandleView;->onHandleMoved()V
  
+     .line 10813
      invoke-direct {p0}, Landroid/widget/TextView$InsertionHandleView;->removeHiderCallback()V
  
+     .line 10814
      return-void
  .end method
  
***************
*** 147,156 ****
--- 166,177 ----
      .parameter "ev"
  
      .prologue
+     .line 10759
      invoke-super {p0, p1}, Landroid/widget/TextView$HandleView;->onTouchEvent(Landroid/view/MotionEvent;)Z
  
      move-result v3
  
+     .line 10761
      .local v3, result:Z
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getActionMasked()I
  
***************
*** 158,167 ****
--- 179,190 ----
  
      packed-switch v4, :pswitch_data_0
  
+     .line 10792
      :goto_0
      :pswitch_0
      return v3
  
+     .line 10763
      :pswitch_1
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getRawX()F
  
***************
*** 169,174 ****
--- 192,198 ----
  
      iput v4, p0, Landroid/widget/TextView$InsertionHandleView;->mDownPositionX:F
  
+     .line 10764
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getRawY()F
  
      move-result v4
***************
*** 177,182 ****
--- 201,207 ----
  
      goto :goto_0
  
+     .line 10768
      :pswitch_2
      invoke-virtual {p0}, Landroid/widget/TextView$InsertionHandleView;->offsetHasBeenChanged()Z
  
***************
*** 184,189 ****
--- 209,215 ----
  
      if-nez v4, :cond_0
  
+     .line 10769
      iget v4, p0, Landroid/widget/TextView$InsertionHandleView;->mDownPositionX:F
  
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getRawX()F
***************
*** 192,197 ****
--- 218,224 ----
  
      sub-float v0, v4, v5
  
+     .line 10770
      .local v0, deltaX:F
      iget v4, p0, Landroid/widget/TextView$InsertionHandleView;->mDownPositionY:F
  
***************
*** 201,206 ****
--- 228,234 ----
  
      sub-float v1, v4, v5
  
+     .line 10771
      .local v1, deltaY:F
      mul-float v4, v0, v0
  
***************
*** 208,213 ****
--- 236,242 ----
  
      add-float v2, v4, v5
  
+     .line 10772
      .local v2, distanceSquared:F
      iget-object v4, p0, Landroid/widget/TextView$InsertionHandleView;->this$0:Landroid/widget/TextView;
  
***************
*** 222,227 ****
--- 251,257 ----
  
      if-gez v4, :cond_0
  
+     .line 10773
      iget-object v4, p0, Landroid/widget/TextView$InsertionHandleView;->mActionPopupWindow:Landroid/widget/TextView$ActionPopupWindow;
  
      if-eqz v4, :cond_1
***************
*** 234,243 ****
--- 264,275 ----
  
      if-eqz v4, :cond_1
  
+     .line 10775
      iget-object v4, p0, Landroid/widget/TextView$InsertionHandleView;->mActionPopupWindow:Landroid/widget/TextView$ActionPopupWindow;
  
      invoke-virtual {v4}, Landroid/widget/TextView$ActionPopupWindow;->hide()V
  
+     .line 10781
      .end local v0           #deltaX:F
      .end local v1           #deltaY:F
      .end local v2           #distanceSquared:F
***************
*** 247,252 ****
--- 279,285 ----
  
      goto :goto_0
  
+     .line 10777
      .restart local v0       #deltaX:F
      .restart local v1       #deltaY:F
      .restart local v2       #distanceSquared:F
***************
*** 255,260 ****
--- 288,294 ----
  
      goto :goto_1
  
+     .line 10785
      .end local v0           #deltaX:F
      .end local v1           #deltaY:F
      .end local v2           #distanceSquared:F
***************
*** 263,268 ****
--- 297,303 ----
  
      goto :goto_0
  
+     .line 10761
      nop
  
      :pswitch_data_0
***************
*** 278,285 ****
--- 313,322 ----
      .locals 6
  
      .prologue
+     .line 10719
      invoke-super {p0}, Landroid/widget/TextView$HandleView;->show()V
  
+     .line 10721
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v2
***************
*** 290,295 ****
--- 327,333 ----
  
      sub-long v0, v2, v4
  
+     .line 10722
      .local v0, durationSinceCutOrCopy:J
      const-wide/16 v2, 0x3a98
  
***************
*** 297,309 ****
--- 335,350 ----
  
      if-gez v2, :cond_0
  
+     .line 10723
      const/4 v2, 0x0
  
      invoke-virtual {p0, v2}, Landroid/widget/TextView$InsertionHandleView;->showActionPopupWindow(I)V
  
+     .line 10726
      :cond_0
      invoke-direct {p0}, Landroid/widget/TextView$InsertionHandleView;->hideAfterDelay()V
  
+     .line 10727
      return-void
  .end method
  
***************
*** 311,322 ****
--- 352,366 ----
      .locals 1
  
      .prologue
+     .line 10730
      invoke-virtual {p0}, Landroid/widget/TextView$InsertionHandleView;->show()V
  
+     .line 10731
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Landroid/widget/TextView$InsertionHandleView;->showActionPopupWindow(I)V
  
+     .line 10732
      return-void
  .end method
  
***************
*** 326,331 ****
--- 370,376 ----
      .parameter "y"
  
      .prologue
+     .line 10807
      iget-object v0, p0, Landroid/widget/TextView$InsertionHandleView;->this$0:Landroid/widget/TextView;
  
      invoke-virtual {v0, p1, p2}, Landroid/widget/TextView;->getOffsetForPosition(FF)I
***************
*** 336,341 ****
--- 381,387 ----
  
      invoke-virtual {p0, v0, v1}, Landroid/widget/TextView$InsertionHandleView;->positionAtCursorOffset(IZ)V
  
+     .line 10808
      return-void
  .end method
  
***************
*** 344,349 ****
--- 390,396 ----
      .parameter "offset"
  
      .prologue
+     .line 10802
      iget-object v0, p0, Landroid/widget/TextView$InsertionHandleView;->this$0:Landroid/widget/TextView;
  
      #getter for: Landroid/widget/TextView;->mText:Ljava/lang/CharSequence;
***************
*** 355,359 ****
--- 402,407 ----
  
      invoke-static {v0, p1}, Landroid/text/Selection;->setSelection(Landroid/text/Spannable;I)V
  
+     .line 10803
      return-void
  .end method
