*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$SuggestionsPopupWindow$SuggestionSpanComparator.smali	2012-03-23 19:36:00.769695020 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$SuggestionsPopupWindow$SuggestionSpanComparator.smali.original	2012-03-23 19:34:44.941312290 -0500
***************
*** 37,42 ****
--- 37,43 ----
      .parameter
  
      .prologue
+     .line 9679
      iput-object p1, p0, Landroid/widget/TextView$SuggestionsPopupWindow$SuggestionSpanComparator;->this$1:Landroid/widget/TextView$SuggestionsPopupWindow;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 50,55 ****
--- 51,57 ----
      .parameter "x1"
  
      .prologue
+     .line 9679
      invoke-direct {p0, p1}, Landroid/widget/TextView$SuggestionsPopupWindow$SuggestionSpanComparator;-><init>(Landroid/widget/TextView$SuggestionsPopupWindow;)V
  
      return-void
***************
*** 69,92 ****
--- 71,99 ----
  
      const/4 v7, 0x1
  
+     .line 9681
      invoke-virtual {p1}, Landroid/text/style/SuggestionSpan;->getFlags()I
  
      move-result v2
  
+     .line 9682
      .local v2, flag1:I
      invoke-virtual {p2}, Landroid/text/style/SuggestionSpan;->getFlags()I
  
      move-result v3
  
+     .line 9683
      .local v3, flag2:I
      if-eq v2, v3, :cond_7
  
+     .line 9685
      and-int/lit8 v8, v2, 0x1
  
      if-eqz v8, :cond_2
  
      move v0, v7
  
+     .line 9686
      .local v0, easy1:Z
      :goto_0
      and-int/lit8 v8, v3, 0x1
***************
*** 95,100 ****
--- 102,108 ----
  
      move v1, v7
  
+     .line 9687
      .local v1, easy2:Z
      :goto_1
      and-int/lit8 v8, v2, 0x2
***************
*** 103,108 ****
--- 111,117 ----
  
      move v4, v7
  
+     .line 9688
      .local v4, misspelled1:Z
      :goto_2
      and-int/lit8 v8, v3, 0x2
***************
*** 111,122 ****
--- 120,133 ----
  
      move v5, v7
  
+     .line 9689
      .local v5, misspelled2:Z
      :cond_0
      if-eqz v0, :cond_5
  
      if-nez v4, :cond_5
  
+     .line 9695
      .end local v0           #easy1:Z
      .end local v1           #easy2:Z
      .end local v4           #misspelled1:Z
***************
*** 128,147 ****
--- 139,162 ----
      :cond_2
      move v0, v5
  
+     .line 9685
      goto :goto_0
  
      .restart local v0       #easy1:Z
      :cond_3
      move v1, v5
  
+     .line 9686
      goto :goto_1
  
      .restart local v1       #easy2:Z
      :cond_4
      move v4, v5
  
+     .line 9687
      goto :goto_2
  
+     .line 9690
      .restart local v4       #misspelled1:Z
      .restart local v5       #misspelled2:Z
      :cond_5
***************
*** 153,167 ****
--- 168,185 ----
  
      goto :goto_3
  
+     .line 9691
      :cond_6
      if-nez v4, :cond_1
  
+     .line 9692
      if-eqz v5, :cond_7
  
      move v6, v7
  
      goto :goto_3
  
+     .line 9695
      .end local v0           #easy1:Z
      .end local v1           #easy2:Z
      .end local v4           #misspelled1:Z
***************
*** 212,217 ****
--- 230,236 ----
      .parameter "x1"
  
      .prologue
+     .line 9679
      check-cast p1, Landroid/text/style/SuggestionSpan;
  
      .end local p1
