*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$SelectionStartHandleView.smali	2012-03-23 19:35:48.068463413 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$SelectionStartHandleView.smali.original	2012-03-23 19:34:44.845310540 -0500
***************
*** 26,35 ****
--- 26,38 ----
      .parameter "drawableRtl"
  
      .prologue
+     .line 10825
      iput-object p1, p0, Landroid/widget/TextView$SelectionStartHandleView;->this$0:Landroid/widget/TextView;
  
+     .line 10826
      invoke-direct {p0, p1, p2, p3}, Landroid/widget/TextView$HandleView;-><init>(Landroid/widget/TextView;Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;)V
  
+     .line 10827
      return-void
  .end method
  
***************
*** 39,44 ****
--- 42,48 ----
      .locals 1
  
      .prologue
+     .line 10861
      iget-object v0, p0, Landroid/widget/TextView$SelectionStartHandleView;->mActionPopupWindow:Landroid/widget/TextView$ActionPopupWindow;
  
      return-object v0
***************
*** 48,53 ****
--- 52,58 ----
      .locals 1
  
      .prologue
+     .line 10840
      iget-object v0, p0, Landroid/widget/TextView$SelectionStartHandleView;->this$0:Landroid/widget/TextView;
  
      invoke-virtual {v0}, Landroid/widget/TextView;->getSelectionStart()I
***************
*** 63,76 ****
--- 68,84 ----
      .parameter "isRtlRun"
  
      .prologue
+     .line 10831
      if-eqz p2, :cond_0
  
+     .line 10832
      invoke-virtual {p1}, Landroid/graphics/drawable/Drawable;->getIntrinsicWidth()I
  
      move-result v0
  
      div-int/lit8 v0, v0, 0x4
  
+     .line 10834
      :goto_0
      return v0
  
***************
*** 94,105 ****
--- 102,115 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 10851
      iget-object v2, p0, Landroid/widget/TextView$SelectionStartHandleView;->this$0:Landroid/widget/TextView;
  
      invoke-virtual {v2, p1, p2}, Landroid/widget/TextView;->getOffsetForPosition(FF)I
  
      move-result v0
  
+     .line 10854
      .local v0, offset:I
      iget-object v2, p0, Landroid/widget/TextView$SelectionStartHandleView;->this$0:Landroid/widget/TextView;
  
***************
*** 107,112 ****
--- 117,123 ----
  
      move-result v1
  
+     .line 10855
      .local v1, selectionEnd:I
      if-lt v0, v1, :cond_0
  
***************
*** 116,124 ****
--- 127,137 ----
  
      move-result v0
  
+     .line 10857
      :cond_0
      invoke-virtual {p0, v0, v3}, Landroid/widget/TextView$SelectionStartHandleView;->positionAtCursorOffset(IZ)V
  
+     .line 10858
      return-void
  .end method
  
***************
*** 127,132 ****
--- 140,146 ----
      .parameter "offset"
  
      .prologue
+     .line 10845
      iget-object v0, p0, Landroid/widget/TextView$SelectionStartHandleView;->this$0:Landroid/widget/TextView;
  
      #getter for: Landroid/widget/TextView;->mText:Ljava/lang/CharSequence;
***************
*** 144,150 ****
--- 158,166 ----
  
      invoke-static {v0, p1, v1}, Landroid/text/Selection;->setSelection(Landroid/text/Spannable;II)V
  
+     .line 10846
      invoke-virtual {p0}, Landroid/widget/TextView$SelectionStartHandleView;->updateDrawable()V
  
+     .line 10847
      return-void
  .end method
