*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$3.smali	2012-03-23 19:35:48.993480281 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$3.smali.original	2012-03-23 19:34:44.852310668 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 3
  
      .prologue
+     .line 6869
      invoke-static {}, Landroid/text/Layout$Alignment;->values()[Landroid/text/Layout$Alignment;
  
      move-result-object v0
***************
*** 114,119 ****
--- 115,121 ----
      :try_end_4
      .catch Ljava/lang/NoSuchFieldError; {:try_start_4 .. :try_end_4} :catch_7
  
+     .line 6126
      :goto_4
      invoke-static {}, Landroid/widget/TextView$TextAlign;->values()[Landroid/widget/TextView$TextAlign;
  
***************
*** 274,279 ****
--- 276,282 ----
  
      goto :goto_5
  
+     .line 6869
      :catch_7
      move-exception v0
  
