*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$PinnedPopupWindow.smali	2012-03-23 19:35:48.546472130 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TextView$PinnedPopupWindow.smali.original	2012-03-23 19:34:44.849310614 -0500
***************
*** 37,79 ****
--- 37,89 ----
      .prologue
      const/4 v3, -0x2
  
+     .line 9462
      iput-object p1, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 9463
      invoke-virtual {p0}, Landroid/widget/TextView$PinnedPopupWindow;->createPopupWindow()V
  
+     .line 9465
      iget-object v1, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPopupWindow:Landroid/widget/PopupWindow;
  
      const/16 v2, 0x3ea
  
      invoke-virtual {v1, v2}, Landroid/widget/PopupWindow;->setWindowLayoutType(I)V
  
+     .line 9466
      iget-object v1, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPopupWindow:Landroid/widget/PopupWindow;
  
      invoke-virtual {v1, v3}, Landroid/widget/PopupWindow;->setWidth(I)V
  
+     .line 9467
      iget-object v1, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPopupWindow:Landroid/widget/PopupWindow;
  
      invoke-virtual {v1, v3}, Landroid/widget/PopupWindow;->setHeight(I)V
  
+     .line 9469
      invoke-virtual {p0}, Landroid/widget/TextView$PinnedPopupWindow;->initContentView()V
  
+     .line 9471
      new-instance v0, Landroid/view/ViewGroup$LayoutParams;
  
      invoke-direct {v0, v3, v3}, Landroid/view/ViewGroup$LayoutParams;-><init>(II)V
  
+     .line 9473
      .local v0, wrapContent:Landroid/view/ViewGroup$LayoutParams;
      iget-object v1, p0, Landroid/widget/TextView$PinnedPopupWindow;->mContentView:Landroid/view/ViewGroup;
  
      invoke-virtual {v1, v0}, Landroid/view/ViewGroup;->setLayoutParams(Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 9475
      iget-object v1, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPopupWindow:Landroid/widget/PopupWindow;
  
      iget-object v2, p0, Landroid/widget/TextView$PinnedPopupWindow;->mContentView:Landroid/view/ViewGroup;
  
      invoke-virtual {v1, v2}, Landroid/widget/PopupWindow;->setContentView(Landroid/view/View;)V
  
+     .line 9476
      return-void
  .end method
  
***************
*** 81,99 ****
--- 91,113 ----
      .locals 6
  
      .prologue
+     .line 9501
      invoke-virtual {p0}, Landroid/widget/TextView$PinnedPopupWindow;->measureContent()V
  
+     .line 9502
      iget-object v3, p0, Landroid/widget/TextView$PinnedPopupWindow;->mContentView:Landroid/view/ViewGroup;
  
      invoke-virtual {v3}, Landroid/view/ViewGroup;->getMeasuredWidth()I
  
      move-result v2
  
+     .line 9503
      .local v2, width:I
      invoke-virtual {p0}, Landroid/widget/TextView$PinnedPopupWindow;->getTextOffset()I
  
      move-result v1
  
+     .line 9504
      .local v1, offset:I
      iget-object v3, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
  
***************
*** 115,120 ****
--- 129,135 ----
  
      iput v3, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPositionX:I
  
+     .line 9505
      iget v3, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPositionX:I
  
      iget-object v4, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
***************
*** 128,133 ****
--- 143,149 ----
  
      iput v3, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPositionX:I
  
+     .line 9507
      iget-object v3, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
  
      iget-object v3, v3, Landroid/widget/TextView;->mLayout:Landroid/text/Layout;
***************
*** 136,141 ****
--- 152,158 ----
  
      move-result v0
  
+     .line 9508
      .local v0, line:I
      invoke-virtual {p0, v0}, Landroid/widget/TextView$PinnedPopupWindow;->getVerticalLocalPosition(I)I
  
***************
*** 143,148 ****
--- 160,166 ----
  
      iput v3, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPositionY:I
  
+     .line 9509
      iget v3, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPositionY:I
  
      iget-object v4, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
***************
*** 156,161 ****
--- 174,180 ----
  
      iput v3, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPositionY:I
  
+     .line 9510
      return-void
  .end method
  
***************
*** 169,188 ****
--- 188,211 ----
  
      const/4 v5, -0x1
  
+     .line 9513
      iget v4, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPositionX:I
  
      add-int v1, p1, v4
  
+     .line 9514
      .local v1, positionX:I
      iget v4, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPositionY:I
  
      add-int v2, p2, v4
  
+     .line 9516
      .local v2, positionY:I
      invoke-virtual {p0, v2}, Landroid/widget/TextView$PinnedPopupWindow;->clipVertically(I)I
  
      move-result v2
  
+     .line 9519
      iget-object v4, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
  
      #getter for: Landroid/widget/TextView;->mContext:Landroid/content/Context;
***************
*** 198,203 ****
--- 221,227 ----
  
      move-result-object v0
  
+     .line 9520
      .local v0, displayMetrics:Landroid/util/DisplayMetrics;
      iget-object v4, p0, Landroid/widget/TextView$PinnedPopupWindow;->mContentView:Landroid/view/ViewGroup;
  
***************
*** 205,210 ****
--- 229,235 ----
  
      move-result v3
  
+     .line 9521
      .local v3, width:I
      iget v4, v0, Landroid/util/DisplayMetrics;->widthPixels:I
  
***************
*** 214,236 ****
--- 239,266 ----
  
      move-result v1
  
+     .line 9522
      invoke-static {v6, v1}, Ljava/lang/Math;->max(II)I
  
      move-result v1
  
+     .line 9524
      invoke-virtual {p0}, Landroid/widget/TextView$PinnedPopupWindow;->isShowing()Z
  
      move-result v4
  
      if-eqz v4, :cond_0
  
+     .line 9525
      iget-object v4, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPopupWindow:Landroid/widget/PopupWindow;
  
      invoke-virtual {v4, v1, v2, v5, v5}, Landroid/widget/PopupWindow;->update(IIII)V
  
+     .line 9530
      :goto_0
      return-void
  
+     .line 9527
      :cond_0
      iget-object v4, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPopupWindow:Landroid/widget/PopupWindow;
  
***************
*** 259,268 ****
--- 289,300 ----
      .locals 1
  
      .prologue
+     .line 9533
      iget-object v0, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPopupWindow:Landroid/widget/PopupWindow;
  
      invoke-virtual {v0}, Landroid/widget/PopupWindow;->dismiss()V
  
+     .line 9534
      iget-object v0, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
  
      #calls: Landroid/widget/TextView;->getPositionListener()Landroid/widget/TextView$PositionListener;
***************
*** 272,277 ****
--- 304,310 ----
  
      invoke-virtual {v0, p0}, Landroid/widget/TextView$PositionListener;->removeSubscriber(Landroid/widget/TextView$TextViewPositionListener;)V
  
+     .line 9535
      return-void
  .end method
  
***************
*** 282,287 ****
--- 315,321 ----
      .locals 1
  
      .prologue
+     .line 9550
      iget-object v0, p0, Landroid/widget/TextView$PinnedPopupWindow;->mPopupWindow:Landroid/widget/PopupWindow;
  
      invoke-virtual {v0}, Landroid/widget/PopupWindow;->isShowing()Z
***************
*** 297,302 ****
--- 331,337 ----
      .prologue
      const/high16 v4, -0x8000
  
+     .line 9488
      iget-object v1, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
  
      #getter for: Landroid/widget/TextView;->mContext:Landroid/content/Context;
***************
*** 312,317 ****
--- 347,353 ----
  
      move-result-object v0
  
+     .line 9489
      .local v0, displayMetrics:Landroid/util/DisplayMetrics;
      iget-object v1, p0, Landroid/widget/TextView$PinnedPopupWindow;->mContentView:Landroid/view/ViewGroup;
  
***************
*** 329,334 ****
--- 365,371 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/view/ViewGroup;->measure(II)V
  
+     .line 9494
      return-void
  .end method
  
***************
*** 336,341 ****
--- 373,379 ----
      .locals 3
  
      .prologue
+     .line 9479
      iget-object v1, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
  
      #calls: Landroid/widget/TextView;->getPositionListener()Landroid/widget/TextView$PositionListener;
***************
*** 347,354 ****
--- 385,394 ----
  
      invoke-virtual {v1, p0, v2}, Landroid/widget/TextView$PositionListener;->addSubscriber(Landroid/widget/TextView$TextViewPositionListener;Z)V
  
+     .line 9481
      invoke-direct {p0}, Landroid/widget/TextView$PinnedPopupWindow;->computeLocalPosition()V
  
+     .line 9483
      iget-object v1, p0, Landroid/widget/TextView$PinnedPopupWindow;->this$0:Landroid/widget/TextView;
  
      #calls: Landroid/widget/TextView;->getPositionListener()Landroid/widget/TextView$PositionListener;
***************
*** 356,361 ****
--- 396,402 ----
  
      move-result-object v0
  
+     .line 9484
      .local v0, positionListener:Landroid/widget/TextView$PositionListener;
      invoke-virtual {v0}, Landroid/widget/TextView$PositionListener;->getPositionX()I
  
***************
*** 367,372 ****
--- 408,414 ----
  
      invoke-direct {p0, v1, v2}, Landroid/widget/TextView$PinnedPopupWindow;->updatePosition(II)V
  
+     .line 9485
      return-void
  .end method
  
***************
*** 378,383 ****
--- 420,426 ----
      .parameter "parentScrolled"
  
      .prologue
+     .line 9541
      invoke-virtual {p0}, Landroid/widget/TextView$PinnedPopupWindow;->isShowing()Z
  
      move-result v0
***************
*** 397,412 ****
--- 440,459 ----
  
      if-eqz v0, :cond_1
  
+     .line 9542
      if-eqz p4, :cond_0
  
      invoke-direct {p0}, Landroid/widget/TextView$PinnedPopupWindow;->computeLocalPosition()V
  
+     .line 9543
      :cond_0
      invoke-direct {p0, p1, p2}, Landroid/widget/TextView$PinnedPopupWindow;->updatePosition(II)V
  
+     .line 9547
      :goto_0
      return-void
  
+     .line 9545
      :cond_1
      invoke-virtual {p0}, Landroid/widget/TextView$PinnedPopupWindow;->hide()V
  
