*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TimePicker$4.smali	2012-03-23 19:36:00.506690223 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/TimePicker$4.smali.original	2012-03-23 19:34:44.940312272 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 210
      iput-object p1, p0, Landroid/widget/TimePicker$4;->this$0:Landroid/widget/TimePicker;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,48 ****
--- 42,51 ----
      .parameter "button"
  
      .prologue
+     .line 212
      invoke-virtual {p1}, Landroid/view/View;->requestFocus()Z
  
+     .line 213
      iget-object v1, p0, Landroid/widget/TimePicker$4;->this$0:Landroid/widget/TimePicker;
  
      iget-object v0, p0, Landroid/widget/TimePicker$4;->this$0:Landroid/widget/TimePicker;
***************
*** 60,72 ****
--- 63,78 ----
      #setter for: Landroid/widget/TimePicker;->mIsAm:Z
      invoke-static {v1, v0}, Landroid/widget/TimePicker;->access$102(Landroid/widget/TimePicker;Z)Z
  
+     .line 214
      iget-object v0, p0, Landroid/widget/TimePicker$4;->this$0:Landroid/widget/TimePicker;
  
      #calls: Landroid/widget/TimePicker;->updateAmPmControl()V
      invoke-static {v0}, Landroid/widget/TimePicker;->access$200(Landroid/widget/TimePicker;)V
  
+     .line 215
      return-void
  
+     .line 213
      :cond_0
      const/4 v0, 0x0
  
