*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Toast$TN$1.smali	2012-03-23 19:35:46.912442334 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Toast$TN$1.smali.original	2012-03-23 19:34:44.837310394 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 302
      iput-object p1, p0, Landroid/widget/Toast$TN$1;->this$0:Landroid/widget/Toast$TN;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,48 ****
--- 41,51 ----
      .locals 1
  
      .prologue
+     .line 304
      iget-object v0, p0, Landroid/widget/Toast$TN$1;->this$0:Landroid/widget/Toast$TN;
  
      invoke-virtual {v0}, Landroid/widget/Toast$TN;->handleShow()V
  
+     .line 305
      return-void
  .end method
