*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Toast$TN$2.smali	2012-03-23 19:35:49.933497421 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/Toast$TN$2.smali.original	2012-03-23 19:34:44.860310814 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 308
      iput-object p1, p0, Landroid/widget/Toast$TN$2;->this$0:Landroid/widget/Toast$TN;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,54 ****
--- 41,58 ----
      .locals 2
  
      .prologue
+     .line 310
      iget-object v0, p0, Landroid/widget/Toast$TN$2;->this$0:Landroid/widget/Toast$TN;
  
      invoke-virtual {v0}, Landroid/widget/Toast$TN;->handleHide()V
  
+     .line 312
      iget-object v0, p0, Landroid/widget/Toast$TN$2;->this$0:Landroid/widget/Toast$TN;
  
      const/4 v1, 0x0
  
      iput-object v1, v0, Landroid/widget/Toast$TN;->mNextView:Landroid/view/View;
  
+     .line 313
      return-void
  .end method
