*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/VideoView$6.smali	2012-03-23 19:35:46.423433416 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/VideoView$6.smali.original	2012-03-23 19:34:44.833310322 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 446
      iput-object p1, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 48,63 ****
--- 49,67 ----
  
      const/4 v3, 0x0
  
+     .line 450
      iget-object v4, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      #setter for: Landroid/widget/VideoView;->mSurfaceWidth:I
      invoke-static {v4, p3}, Landroid/widget/VideoView;->access$1002(Landroid/widget/VideoView;I)I
  
+     .line 451
      iget-object v4, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      #setter for: Landroid/widget/VideoView;->mSurfaceHeight:I
      invoke-static {v4, p4}, Landroid/widget/VideoView;->access$1102(Landroid/widget/VideoView;I)I
  
+     .line 452
      iget-object v4, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mTargetState:I
***************
*** 71,76 ****
--- 75,81 ----
  
      move v1, v2
  
+     .line 453
      .local v1, isValidState:Z
      :goto_0
      iget-object v4, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
***************
*** 93,98 ****
--- 98,104 ----
  
      move v0, v2
  
+     .line 454
      .local v0, hasValidSize:Z
      :goto_1
      iget-object v2, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
***************
*** 108,113 ****
--- 114,120 ----
  
      if-eqz v0, :cond_1
  
+     .line 455
      iget-object v2, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mSeekWhenPrepared:I
***************
*** 117,122 ****
--- 124,130 ----
  
      if-eqz v2, :cond_0
  
+     .line 456
      iget-object v2, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      iget-object v3, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
***************
*** 128,138 ****
--- 136,148 ----
  
      invoke-virtual {v2, v3}, Landroid/widget/VideoView;->seekTo(I)V
  
+     .line 458
      :cond_0
      iget-object v2, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      invoke-virtual {v2}, Landroid/widget/VideoView;->start()V
  
+     .line 460
      :cond_1
      return-void
  
***************
*** 141,152 ****
--- 151,164 ----
      :cond_2
      move v1, v3
  
+     .line 452
      goto :goto_0
  
      .restart local v1       #isValidState:Z
      :cond_3
      move v0, v3
  
+     .line 453
      goto :goto_1
  .end method
  
***************
*** 155,170 ****
--- 167,185 ----
      .parameter "holder"
  
      .prologue
+     .line 464
      iget-object v0, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      #setter for: Landroid/widget/VideoView;->mSurfaceHolder:Landroid/view/SurfaceHolder;
      invoke-static {v0, p1}, Landroid/widget/VideoView;->access$1902(Landroid/widget/VideoView;Landroid/view/SurfaceHolder;)Landroid/view/SurfaceHolder;
  
+     .line 465
      iget-object v0, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      #calls: Landroid/widget/VideoView;->openVideo()V
      invoke-static {v0}, Landroid/widget/VideoView;->access$2000(Landroid/widget/VideoView;)V
  
+     .line 466
      return-void
  .end method
  
***************
*** 173,178 ****
--- 188,194 ----
      .parameter "holder"
  
      .prologue
+     .line 471
      iget-object v0, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      const/4 v1, 0x0
***************
*** 180,185 ****
--- 196,202 ----
      #setter for: Landroid/widget/VideoView;->mSurfaceHolder:Landroid/view/SurfaceHolder;
      invoke-static {v0, v1}, Landroid/widget/VideoView;->access$1902(Landroid/widget/VideoView;Landroid/view/SurfaceHolder;)Landroid/view/SurfaceHolder;
  
+     .line 472
      iget-object v0, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mMediaController:Landroid/widget/MediaController;
***************
*** 198,203 ****
--- 215,221 ----
  
      invoke-virtual {v0}, Landroid/widget/MediaController;->hide()V
  
+     .line 473
      :cond_0
      iget-object v0, p0, Landroid/widget/VideoView$6;->this$0:Landroid/widget/VideoView;
  
***************
*** 206,210 ****
--- 224,229 ----
      #calls: Landroid/widget/VideoView;->release(Z)V
      invoke-static {v0, v1}, Landroid/widget/VideoView;->access$2100(Landroid/widget/VideoView;Z)V
  
+     .line 474
      return-void
  .end method
