*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/VideoView$2.smali	2012-03-23 19:35:53.949570655 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/VideoView$2.smali.original	2012-03-23 19:34:44.892311398 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 274
      iput-object p1, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 49,66 ****
--- 50,71 ----
  
      const/4 v4, 0x1
  
+     .line 276
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      #setter for: Landroid/widget/VideoView;->mCurrentState:I
      invoke-static {v2, v6}, Landroid/widget/VideoView;->access$202(Landroid/widget/VideoView;I)I
  
+     .line 279
      invoke-virtual {p1, v3, v3}, Landroid/media/MediaPlayer;->getMetadata(ZZ)Landroid/media/Metadata;
  
      move-result-object v0
  
+     .line 282
      .local v0, data:Landroid/media/Metadata;
      if-eqz v0, :cond_a
  
+     .line 283
      iget-object v5, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      invoke-virtual {v0, v4}, Landroid/media/Metadata;->has(I)Z
***************
*** 82,87 ****
--- 87,93 ----
      #setter for: Landroid/widget/VideoView;->mCanPause:Z
      invoke-static {v5, v2}, Landroid/widget/VideoView;->access$302(Landroid/widget/VideoView;Z)Z
  
+     .line 285
      iget-object v5, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      invoke-virtual {v0, v6}, Landroid/media/Metadata;->has(I)Z
***************
*** 103,108 ****
--- 109,115 ----
      #setter for: Landroid/widget/VideoView;->mCanSeekBack:Z
      invoke-static {v5, v2}, Landroid/widget/VideoView;->access$402(Landroid/widget/VideoView;Z)Z
  
+     .line 287
      iget-object v5, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      invoke-virtual {v0, v7}, Landroid/media/Metadata;->has(I)Z
***************
*** 124,129 ****
--- 131,137 ----
      #setter for: Landroid/widget/VideoView;->mCanSeekForward:Z
      invoke-static {v5, v2}, Landroid/widget/VideoView;->access$502(Landroid/widget/VideoView;Z)Z
  
+     .line 293
      :goto_3
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
***************
*** 134,139 ****
--- 142,148 ----
  
      if-eqz v2, :cond_3
  
+     .line 294
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mOnPreparedListener:Landroid/media/MediaPlayer$OnPreparedListener;
***************
*** 150,155 ****
--- 159,165 ----
  
      invoke-interface {v2, v5}, Landroid/media/MediaPlayer$OnPreparedListener;->onPrepared(Landroid/media/MediaPlayer;)V
  
+     .line 296
      :cond_3
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
***************
*** 160,165 ****
--- 170,176 ----
  
      if-eqz v2, :cond_4
  
+     .line 297
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mMediaController:Landroid/widget/MediaController;
***************
*** 169,174 ****
--- 180,186 ----
  
      invoke-virtual {v2, v4}, Landroid/widget/MediaController;->setEnabled(Z)V
  
+     .line 299
      :cond_4
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
***************
*** 179,184 ****
--- 191,197 ----
      #setter for: Landroid/widget/VideoView;->mVideoWidth:I
      invoke-static {v2, v4}, Landroid/widget/VideoView;->access$002(Landroid/widget/VideoView;I)I
  
+     .line 300
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      invoke-virtual {p1}, Landroid/media/MediaPlayer;->getVideoHeight()I
***************
*** 188,193 ****
--- 201,207 ----
      #setter for: Landroid/widget/VideoView;->mVideoHeight:I
      invoke-static {v2, v4}, Landroid/widget/VideoView;->access$102(Landroid/widget/VideoView;I)I
  
+     .line 302
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mSeekWhenPrepared:I
***************
*** 195,207 ****
--- 209,224 ----
  
      move-result v1
  
+     .line 303
      .local v1, seekToPosition:I
      if-eqz v1, :cond_5
  
+     .line 304
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      invoke-virtual {v2, v1}, Landroid/widget/VideoView;->seekTo(I)V
  
+     .line 306
      :cond_5
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
***************
*** 221,226 ****
--- 238,244 ----
  
      if-eqz v2, :cond_d
  
+     .line 308
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      invoke-virtual {v2}, Landroid/widget/VideoView;->getHolder()Landroid/view/SurfaceHolder;
***************
*** 243,248 ****
--- 261,267 ----
  
      invoke-interface {v2, v4, v5}, Landroid/view/SurfaceHolder;->setFixedSize(II)V
  
+     .line 309
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mSurfaceWidth:I
***************
*** 275,280 ****
--- 294,300 ----
  
      if-ne v2, v4, :cond_6
  
+     .line 313
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mTargetState:I
***************
*** 284,293 ****
--- 304,315 ----
  
      if-ne v2, v7, :cond_b
  
+     .line 314
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      invoke-virtual {v2}, Landroid/widget/VideoView;->start()V
  
+     .line 315
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mMediaController:Landroid/widget/MediaController;
***************
*** 297,302 ****
--- 319,325 ----
  
      if-eqz v2, :cond_6
  
+     .line 316
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mMediaController:Landroid/widget/MediaController;
***************
*** 306,311 ****
--- 329,335 ----
  
      invoke-virtual {v2}, Landroid/widget/MediaController;->show()V
  
+     .line 333
      :cond_6
      :goto_4
      return-void
***************
*** 314,331 ****
--- 338,359 ----
      :cond_7
      move v2, v3
  
+     .line 283
      goto/16 :goto_0
  
      :cond_8
      move v2, v3
  
+     .line 285
      goto/16 :goto_1
  
      :cond_9
      move v2, v3
  
+     .line 287
      goto/16 :goto_2
  
+     .line 290
      :cond_a
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
***************
*** 348,353 ****
--- 376,382 ----
  
      goto/16 :goto_3
  
+     .line 318
      .restart local v1       #seekToPosition:I
      :cond_b
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
***************
*** 368,373 ****
--- 397,403 ----
  
      if-lez v2, :cond_6
  
+     .line 320
      :cond_c
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
***************
*** 378,383 ****
--- 408,414 ----
  
      if-eqz v2, :cond_6
  
+     .line 322
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      #getter for: Landroid/widget/VideoView;->mMediaController:Landroid/widget/MediaController;
***************
*** 389,394 ****
--- 420,426 ----
  
      goto :goto_4
  
+     .line 329
      :cond_d
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
***************
*** 399,404 ****
--- 431,437 ----
  
      if-ne v2, v7, :cond_6
  
+     .line 330
      iget-object v2, p0, Landroid/widget/VideoView$2;->this$0:Landroid/widget/VideoView;
  
      invoke-virtual {v2}, Landroid/widget/VideoView;->start()V
