*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/VideoView$4$1.smali	2012-03-23 19:35:51.918533620 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/VideoView$4$1.smali.original	2012-03-23 19:34:44.875311088 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 386
      iput-object p1, p0, Landroid/widget/VideoView$4$1;->this$1:Landroid/widget/VideoView$4;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,47 ****
--- 43,49 ----
      .parameter "whichButton"
  
      .prologue
+     .line 391
      iget-object v0, p0, Landroid/widget/VideoView$4$1;->this$1:Landroid/widget/VideoView$4;
  
      iget-object v0, v0, Landroid/widget/VideoView$4;->this$0:Landroid/widget/VideoView;
***************
*** 53,58 ****
--- 55,61 ----
  
      if-eqz v0, :cond_0
  
+     .line 392
      iget-object v0, p0, Landroid/widget/VideoView$4$1;->this$1:Landroid/widget/VideoView$4;
  
      iget-object v0, v0, Landroid/widget/VideoView$4;->this$0:Landroid/widget/VideoView;
***************
*** 73,78 ****
--- 76,82 ----
  
      invoke-interface {v0, v1}, Landroid/media/MediaPlayer$OnCompletionListener;->onCompletion(Landroid/media/MediaPlayer;)V
  
+     .line 394
      :cond_0
      return-void
  .end method
