*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ZoomButtonsController$5.smali	2012-03-23 19:35:55.342596057 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ZoomButtonsController$5.smali.original	2012-03-23 19:34:44.904311616 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 374
      iput-object p1, p0, Landroid/widget/ZoomButtonsController$5;->this$0:Landroid/widget/ZoomButtonsController;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,50 ****
--- 41,53 ----
      .locals 2
  
      .prologue
+     .line 376
      iget-object v0, p0, Landroid/widget/ZoomButtonsController$5;->this$0:Landroid/widget/ZoomButtonsController;
  
      #calls: Landroid/widget/ZoomButtonsController;->refreshPositioningVariables()V
      invoke-static {v0}, Landroid/widget/ZoomButtonsController;->access$700(Landroid/widget/ZoomButtonsController;)V
  
+     .line 378
      iget-object v0, p0, Landroid/widget/ZoomButtonsController$5;->this$0:Landroid/widget/ZoomButtonsController;
  
      #getter for: Landroid/widget/ZoomButtonsController;->mCallback:Landroid/widget/ZoomButtonsController$OnZoomListener;
***************
*** 54,59 ****
--- 57,63 ----
  
      if-eqz v0, :cond_0
  
+     .line 379
      iget-object v0, p0, Landroid/widget/ZoomButtonsController$5;->this$0:Landroid/widget/ZoomButtonsController;
  
      #getter for: Landroid/widget/ZoomButtonsController;->mCallback:Landroid/widget/ZoomButtonsController$OnZoomListener;
***************
*** 65,70 ****
--- 69,75 ----
  
      invoke-interface {v0, v1}, Landroid/widget/ZoomButtonsController$OnZoomListener;->onVisibilityChanged(Z)V
  
+     .line 381
      :cond_0
      return-void
  .end method
