*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ZoomButtonsController$3.smali	2012-03-23 19:35:55.233594068 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/widget/ZoomButtonsController$3.smali.original	2012-03-23 19:34:44.903311598 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 266
      iput-object p1, p0, Landroid/widget/ZoomButtonsController$3;->this$0:Landroid/widget/ZoomButtonsController;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "v"
  
      .prologue
+     .line 268
      iget-object v0, p0, Landroid/widget/ZoomButtonsController$3;->this$0:Landroid/widget/ZoomButtonsController;
  
      invoke-static {}, Landroid/widget/ZoomButtonsController;->access$400()I
***************
*** 50,55 ****
--- 52,58 ----
      #calls: Landroid/widget/ZoomButtonsController;->dismissControlsDelayed(I)V
      invoke-static {v0, v1}, Landroid/widget/ZoomButtonsController;->access$500(Landroid/widget/ZoomButtonsController;I)V
  
+     .line 269
      iget-object v0, p0, Landroid/widget/ZoomButtonsController$3;->this$0:Landroid/widget/ZoomButtonsController;
  
      #getter for: Landroid/widget/ZoomButtonsController;->mCallback:Landroid/widget/ZoomButtonsController$OnZoomListener;
***************
*** 70,75 ****
--- 73,79 ----
  
      invoke-interface {v0, v1}, Landroid/widget/ZoomButtonsController$OnZoomListener;->onZoom(Z)V
  
+     .line 270
      :cond_0
      return-void
  .end method
