*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper.smali	2012-03-23 19:34:52.073442346 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper.smali.original	2012-03-23 19:34:46.852347144 -0500
***************
*** 33,46 ****
--- 33,50 ----
      .parameter "connection"
  
      .prologue
+     .line 917
      iput-object p1, p0, Lcom/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper;->this$0:Lcom/android/server/accessibility/AccessibilityManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 918
      iput p2, p0, Lcom/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper;->mWindowId:I
  
+     .line 919
      iput-object p3, p0, Lcom/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper;->mConnection:Landroid/view/accessibility/IAccessibilityInteractionConnection;
  
+     .line 920
      return-void
  .end method
  
***************
*** 49,54 ****
--- 53,59 ----
      .parameter "x0"
  
      .prologue
+     .line 912
      iget-object v0, p0, Lcom/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper;->mConnection:Landroid/view/accessibility/IAccessibilityInteractionConnection;
  
      return-object v0
***************
*** 60,73 ****
--- 65,81 ----
      .locals 3
  
      .prologue
+     .line 932
      invoke-virtual {p0}, Lcom/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper;->unlinkToDeath()V
  
+     .line 933
      iget-object v0, p0, Lcom/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper;->this$0:Lcom/android/server/accessibility/AccessibilityManagerService;
  
      iget-object v1, v0, Lcom/android/server/accessibility/AccessibilityManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 934
      :try_start_0
      iget-object v0, p0, Lcom/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper;->this$0:Lcom/android/server/accessibility/AccessibilityManagerService;
  
***************
*** 76,85 ****
--- 84,96 ----
      #calls: Lcom/android/server/accessibility/AccessibilityManagerService;->removeAccessibilityInteractionConnectionLocked(I)V
      invoke-static {v0, v2}, Lcom/android/server/accessibility/AccessibilityManagerService;->access$1100(Lcom/android/server/accessibility/AccessibilityManagerService;I)V
  
+     .line 935
      monitor-exit v1
  
+     .line 936
      return-void
  
+     .line 935
      :catchall_0
      move-exception v0
  
***************
*** 99,104 ****
--- 110,116 ----
      .end annotation
  
      .prologue
+     .line 923
      iget-object v0, p0, Lcom/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper;->mConnection:Landroid/view/accessibility/IAccessibilityInteractionConnection;
  
      invoke-interface {v0}, Landroid/view/accessibility/IAccessibilityInteractionConnection;->asBinder()Landroid/os/IBinder;
***************
*** 109,114 ****
--- 121,127 ----
  
      invoke-interface {v0, p0, v1}, Landroid/os/IBinder;->linkToDeath(Landroid/os/IBinder$DeathRecipient;I)V
  
+     .line 924
      return-void
  .end method
  
***************
*** 116,121 ****
--- 129,135 ----
      .locals 2
  
      .prologue
+     .line 927
      iget-object v0, p0, Lcom/android/server/accessibility/AccessibilityManagerService$AccessibilityConnectionWrapper;->mConnection:Landroid/view/accessibility/IAccessibilityInteractionConnection;
  
      invoke-interface {v0}, Landroid/view/accessibility/IAccessibilityInteractionConnection;->asBinder()Landroid/os/IBinder;
***************
*** 126,130 ****
--- 140,145 ----
  
      invoke-interface {v0, p0, v1}, Landroid/os/IBinder;->unlinkToDeath(Landroid/os/IBinder$DeathRecipient;I)Z
  
+     .line 928
      return-void
  .end method
