*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/accessibility/AccessibilityManagerService$6.smali	2012-03-23 19:34:51.672435033 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/accessibility/AccessibilityManagerService$6.smali.original	2012-03-23 19:34:46.849347079 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 345
      iput-object p1, p0, Lcom/android/server/accessibility/AccessibilityManagerService$6;->this$0:Lcom/android/server/accessibility/AccessibilityManagerService;
  
      iput-object p2, p0, Lcom/android/server/accessibility/AccessibilityManagerService$6;->val$addedClient:Landroid/view/accessibility/IAccessibilityManagerClient;
***************
*** 45,56 ****
--- 46,59 ----
      .locals 3
  
      .prologue
+     .line 347
      iget-object v0, p0, Lcom/android/server/accessibility/AccessibilityManagerService$6;->this$0:Lcom/android/server/accessibility/AccessibilityManagerService;
  
      iget-object v1, v0, Lcom/android/server/accessibility/AccessibilityManagerService;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 348
      :try_start_0
      iget-object v0, p0, Lcom/android/server/accessibility/AccessibilityManagerService$6;->val$addedClient:Landroid/view/accessibility/IAccessibilityManagerClient;
  
***************
*** 62,67 ****
--- 65,71 ----
  
      invoke-interface {v0, p0, v2}, Landroid/os/IBinder;->unlinkToDeath(Landroid/os/IBinder$DeathRecipient;I)Z
  
+     .line 349
      iget-object v0, p0, Lcom/android/server/accessibility/AccessibilityManagerService$6;->this$0:Lcom/android/server/accessibility/AccessibilityManagerService;
  
      iget-object v0, v0, Lcom/android/server/accessibility/AccessibilityManagerService;->mClients:Ljava/util/List;
***************
*** 70,79 ****
--- 74,86 ----
  
      invoke-interface {v0, v2}, Ljava/util/List;->remove(Ljava/lang/Object;)Z
  
+     .line 350
      monitor-exit v1
  
+     .line 351
      return-void
  
+     .line 350
      :catchall_0
      move-exception v0
  
