*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/security/KeyStore.smali	2012-03-23 19:36:19.655039394 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/security/KeyStore.smali.original	2012-03-23 19:34:45.055314369 -0500
***************
*** 44,49 ****
--- 44,50 ----
      .locals 3
  
      .prologue
+     .line 51
      new-instance v0, Landroid/net/LocalSocketAddress;
  
      const-string v1, "keystore"
***************
*** 61,72 ****
--- 62,76 ----
      .locals 1
  
      .prologue
+     .line 56
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 54
      const/4 v0, 0x1
  
      iput v0, p0, Landroid/security/KeyStore;->mError:I
  
+     .line 56
      return-void
  .end method
  
***************
*** 79,84 ****
--- 83,89 ----
  
      const/4 v0, 0x1
  
+     .line 100
      const/16 v2, 0x65
  
      new-array v3, v0, [[B
***************
*** 87,92 ****
--- 92,98 ----
  
      invoke-direct {p0, v2, v3}, Landroid/security/KeyStore;->execute(I[[B)Ljava/util/ArrayList;
  
+     .line 101
      iget v2, p0, Landroid/security/KeyStore;->mError:I
  
      if-ne v2, v0, :cond_0
***************
*** 109,114 ****
--- 115,121 ----
  
      const/4 v0, 0x1
  
+     .line 91
      const/16 v2, 0x64
  
      new-array v3, v0, [[B
***************
*** 117,122 ****
--- 124,130 ----
  
      invoke-direct {p0, v2, v3}, Landroid/security/KeyStore;->execute(I[[B)Ljava/util/ArrayList;
  
+     .line 92
      iget v2, p0, Landroid/security/KeyStore;->mError:I
  
      if-ne v2, v0, :cond_0
***************
*** 143,152 ****
--- 151,162 ----
      .end annotation
  
      .prologue
+     .line 163
      const/4 v11, 0x5
  
      iput v11, p0, Landroid/security/KeyStore;->mError:I
  
+     .line 165
      move-object v0, p2
  
      .local v0, arr$:[[B
***************
*** 161,166 ****
--- 171,177 ----
  
      aget-object v7, v0, v2
  
+     .line 166
      .local v7, parameter:[B
      if-eqz v7, :cond_0
  
***************
*** 170,207 ****
--- 181,226 ----
  
      if-le v11, v12, :cond_1
  
+     .line 167
      :cond_0
      const/4 v10, 0x0
  
+     .line 219
      .end local v7           #parameter:[B
      :goto_1
      return-object v10
  
+     .line 165
      .restart local v7       #parameter:[B
      :cond_1
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 171
      .end local v7           #parameter:[B
      :cond_2
      new-instance v8, Landroid/net/LocalSocket;
  
      invoke-direct {v8}, Landroid/net/LocalSocket;-><init>()V
  
+     .line 173
      .local v8, socket:Landroid/net/LocalSocket;
      :try_start_0
      sget-object v11, Landroid/security/KeyStore;->sAddress:Landroid/net/LocalSocketAddress;
  
      invoke-virtual {v8, v11}, Landroid/net/LocalSocket;->connect(Landroid/net/LocalSocketAddress;)V
  
+     .line 175
      invoke-virtual {v8}, Landroid/net/LocalSocket;->getOutputStream()Ljava/io/OutputStream;
  
      move-result-object v6
  
+     .line 176
      .local v6, out:Ljava/io/OutputStream;
      invoke-virtual {v6, p1}, Ljava/io/OutputStream;->write(I)V
  
+     .line 177
      move-object v0, p2
  
      array-length v5, v0
***************
*** 213,218 ****
--- 232,238 ----
  
      aget-object v7, v0, v2
  
+     .line 178
      .restart local v7       #parameter:[B
      array-length v11, v7
  
***************
*** 220,245 ****
--- 240,272 ----
  
      invoke-virtual {v6, v11}, Ljava/io/OutputStream;->write(I)V
  
+     .line 179
      array-length v11, v7
  
      invoke-virtual {v6, v11}, Ljava/io/OutputStream;->write(I)V
  
+     .line 180
      invoke-virtual {v6, v7}, Ljava/io/OutputStream;->write([B)V
  
+     .line 177
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_2
  
+     .line 182
      .end local v7           #parameter:[B
      :cond_3
      invoke-virtual {v6}, Ljava/io/OutputStream;->flush()V
  
+     .line 183
      invoke-virtual {v8}, Landroid/net/LocalSocket;->shutdownOutput()V
  
+     .line 185
      invoke-virtual {v8}, Landroid/net/LocalSocket;->getInputStream()Ljava/io/InputStream;
  
      move-result-object v3
  
+     .line 186
      .local v3, in:Ljava/io/InputStream;
      invoke-virtual {v3}, Ljava/io/InputStream;->read()I
  
***************
*** 249,266 ****
--- 276,297 ----
  
      if-eq p1, v11, :cond_5
  
+     .line 187
      const/4 v11, -0x1
  
      if-eq p1, v11, :cond_4
  
+     .line 188
      iput p1, p0, Landroid/security/KeyStore;->mError:I
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_4
  
+     .line 190
      :cond_4
      const/4 v10, 0x0
  
+     .line 216
      :try_start_1
      invoke-virtual {v8}, Landroid/net/LocalSocket;->close()V
      :try_end_1
***************
*** 268,284 ****
--- 299,318 ----
  
      goto :goto_1
  
+     .line 217
      :catch_0
      move-exception v11
  
      goto :goto_1
  
+     .line 193
      :cond_5
      :try_start_2
      new-instance v10, Ljava/util/ArrayList;
  
      invoke-direct {v10}, Ljava/util/ArrayList;-><init>()V
  
+     .line 196
      .local v10, values:Ljava/util/ArrayList;,"Ljava/util/ArrayList<[B>;"
      :goto_3
      invoke-virtual {v3}, Ljava/io/InputStream;->read()I
***************
*** 290,295 ****
--- 324,330 ----
  
      if-ne v1, v11, :cond_6
  
+     .line 210
      const/4 v11, 0x1
  
      iput v11, p0, Landroid/security/KeyStore;->mError:I
***************
*** 297,302 ****
--- 332,338 ----
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
      .catch Ljava/io/IOException; {:try_start_2 .. :try_end_2} :catch_4
  
+     .line 216
      :try_start_3
      invoke-virtual {v8}, Landroid/net/LocalSocket;->close()V
      :try_end_3
***************
*** 304,314 ****
--- 340,352 ----
  
      goto :goto_1
  
+     .line 217
      :catch_1
      move-exception v11
  
      goto :goto_1
  
+     .line 199
      :cond_6
      :try_start_4
      invoke-virtual {v3}, Ljava/io/InputStream;->read()I
***************
*** 323,330 ****
--- 361,370 ----
  
      if-ne v4, v11, :cond_7
  
+     .line 200
      const/4 v10, 0x0
  
+     .line 216
      .end local v10           #values:Ljava/util/ArrayList;,"Ljava/util/ArrayList<[B>;"
      :try_start_5
      invoke-virtual {v8}, Landroid/net/LocalSocket;->close()V
***************
*** 333,343 ****
--- 373,385 ----
  
      goto :goto_1
  
+     .line 217
      :catch_2
      move-exception v11
  
      goto :goto_1
  
+     .line 202
      .restart local v10       #values:Ljava/util/ArrayList;,"Ljava/util/ArrayList<[B>;"
      :cond_7
      shl-int/lit8 v11, v1, 0x8
***************
*** 347,352 ****
--- 389,395 ----
      :try_start_6
      new-array v9, v11, [B
  
+     .line 203
      .local v9, value:[B
      const/4 v1, 0x0
  
***************
*** 355,360 ****
--- 398,404 ----
  
      if-ge v1, v11, :cond_9
  
+     .line 204
      array-length v11, v9
  
      sub-int/2addr v11, v1
***************
*** 370,377 ****
--- 414,423 ----
  
      if-ne v4, v11, :cond_8
  
+     .line 205
      const/4 v10, 0x0
  
+     .line 216
      .end local v10           #values:Ljava/util/ArrayList;,"Ljava/util/ArrayList<[B>;"
      :try_start_7
      invoke-virtual {v8}, Landroid/net/LocalSocket;->close()V
***************
*** 380,396 ****
--- 426,445 ----
  
      goto/16 :goto_1
  
+     .line 217
      :catch_3
      move-exception v11
  
      goto/16 :goto_1
  
+     .line 203
      .restart local v10       #values:Ljava/util/ArrayList;,"Ljava/util/ArrayList<[B>;"
      :cond_8
      add-int/2addr v1, v4
  
      goto :goto_4
  
+     .line 208
      :cond_9
      :try_start_8
      invoke-virtual {v10, v9}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
***************
*** 400,405 ****
--- 449,455 ----
  
      goto :goto_3
  
+     .line 212
      .end local v1           #i:I
      .end local v3           #in:Ljava/io/InputStream;
      .end local v4           #j:I
***************
*** 409,432 ****
--- 459,487 ----
      :catch_4
      move-exception v11
  
+     .line 216
      :try_start_9
      invoke-virtual {v8}, Landroid/net/LocalSocket;->close()V
      :try_end_9
      .catch Ljava/io/IOException; {:try_start_9 .. :try_end_9} :catch_5
  
+     .line 219
      :goto_5
      const/4 v10, 0x0
  
      goto/16 :goto_1
  
+     .line 215
      :catchall_0
      move-exception v11
  
+     .line 216
      :try_start_a
      invoke-virtual {v8}, Landroid/net/LocalSocket;->close()V
      :try_end_a
      .catch Ljava/io/IOException; {:try_start_a .. :try_end_a} :catch_6
  
+     .line 217
      :goto_6
      throw v11
  
***************
*** 448,453 ****
--- 503,509 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 73
      const/16 v1, 0x67
  
      const/4 v2, 0x1
***************
*** 460,465 ****
--- 516,522 ----
  
      move-result-object v0
  
+     .line 74
      .local v0, values:Ljava/util/ArrayList;,"Ljava/util/ArrayList<[B>;"
      if-eqz v0, :cond_0
  
***************
*** 490,495 ****
--- 547,553 ----
      .parameter "string"
  
      .prologue
+     .line 223
      sget-object v0, Ljava/nio/charset/Charsets;->UTF_8:Ljava/nio/charset/Charset;
  
      invoke-virtual {p0, v0}, Ljava/lang/String;->getBytes(Ljava/nio/charset/Charset;)[B
***************
*** 503,508 ****
--- 561,567 ----
      .locals 1
  
      .prologue
+     .line 59
      new-instance v0, Landroid/security/KeyStore;
  
      invoke-direct {v0}, Landroid/security/KeyStore;-><init>()V
***************
*** 519,524 ****
--- 578,584 ----
  
      const/4 v0, 0x1
  
+     .line 131
      const/16 v2, 0x70
  
      new-array v3, v0, [[B
***************
*** 527,532 ****
--- 587,593 ----
  
      invoke-direct {p0, v2, v3}, Landroid/security/KeyStore;->execute(I[[B)Ljava/util/ArrayList;
  
+     .line 132
      iget v2, p0, Landroid/security/KeyStore;->mError:I
  
      if-ne v2, v0, :cond_0
***************
*** 550,555 ****
--- 611,617 ----
  
      const/4 v0, 0x1
  
+     .line 82
      const/16 v2, 0x69
  
      const/4 v3, 0x2
***************
*** 562,567 ****
--- 624,630 ----
  
      invoke-direct {p0, v2, v3}, Landroid/security/KeyStore;->execute(I[[B)Ljava/util/ArrayList;
  
+     .line 83
      iget v2, p0, Landroid/security/KeyStore;->mError:I
  
      if-ne v2, v0, :cond_0
***************
*** 580,585 ****
--- 643,649 ----
      .parameter "bytes"
  
      .prologue
+     .line 227
      new-instance v0, Ljava/lang/String;
  
      sget-object v1, Ljava/nio/charset/Charsets;->UTF_8:Ljava/nio/charset/Charset;
***************
*** 598,603 ****
--- 662,668 ----
  
      const/4 v0, 0x1
  
+     .line 145
      const/16 v2, 0x75
  
      new-array v3, v0, [[B
***************
*** 606,611 ****
--- 671,677 ----
  
      invoke-direct {p0, v2, v3}, Landroid/security/KeyStore;->execute(I[[B)Ljava/util/ArrayList;
  
+     .line 146
      iget v2, p0, Landroid/security/KeyStore;->mError:I
  
      if-ne v2, v0, :cond_0
***************
*** 626,631 ****
--- 692,698 ----
      .parameter "key"
  
      .prologue
+     .line 105
      invoke-static {p1}, Landroid/security/KeyStore;->getBytes(Ljava/lang/String;)[B
  
      move-result-object v0
***************
*** 642,647 ****
--- 709,715 ----
      .parameter "key"
  
      .prologue
+     .line 96
      invoke-static {p1}, Landroid/security/KeyStore;->getBytes(Ljava/lang/String;)[B
  
      move-result-object v0
***************
*** 658,663 ****
--- 726,732 ----
      .parameter "key"
  
      .prologue
+     .line 78
      invoke-static {p1}, Landroid/security/KeyStore;->getBytes(Ljava/lang/String;)[B
  
      move-result-object v0
***************
*** 673,678 ****
--- 742,748 ----
      .locals 1
  
      .prologue
+     .line 159
      iget v0, p0, Landroid/security/KeyStore;->mError:I
  
      return v0
***************
*** 684,695 ****
--- 754,767 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 154
      const/16 v1, 0x7a
  
      new-array v2, v0, [[B
  
      invoke-direct {p0, v1, v2}, Landroid/security/KeyStore;->execute(I[[B)Ljava/util/ArrayList;
  
+     .line 155
      iget v1, p0, Landroid/security/KeyStore;->mError:I
  
      const/4 v2, 0x7
***************
*** 710,721 ****
--- 782,795 ----
  
      const/4 v1, 0x0
  
+     .line 140
      const/16 v2, 0x6c
  
      new-array v3, v1, [[B
  
      invoke-direct {p0, v2, v3}, Landroid/security/KeyStore;->execute(I[[B)Ljava/util/ArrayList;
  
+     .line 141
      iget v2, p0, Landroid/security/KeyStore;->mError:I
  
      if-ne v2, v0, :cond_0
***************
*** 734,739 ****
--- 808,814 ----
      .parameter "password"
  
      .prologue
+     .line 136
      invoke-static {p1}, Landroid/security/KeyStore;->getBytes(Ljava/lang/String;)[B
  
      move-result-object v0
***************
*** 751,756 ****
--- 826,832 ----
      .parameter "value"
  
      .prologue
+     .line 87
      invoke-static {p1}, Landroid/security/KeyStore;->getBytes(Ljava/lang/String;)[B
  
      move-result-object v0
***************
*** 770,781 ****
--- 846,859 ----
  
      const/4 v1, 0x0
  
+     .line 126
      const/16 v2, 0x72
  
      new-array v3, v1, [[B
  
      invoke-direct {p0, v2, v3}, Landroid/security/KeyStore;->execute(I[[B)Ljava/util/ArrayList;
  
+     .line 127
      iget v2, p0, Landroid/security/KeyStore;->mError:I
  
      if-ne v2, v0, :cond_0
***************
*** 794,799 ****
--- 872,878 ----
      .parameter "prefix"
  
      .prologue
+     .line 114
      invoke-static {p1}, Landroid/security/KeyStore;->getBytes(Ljava/lang/String;)[B
  
      move-result-object v3
***************
*** 802,820 ****
--- 881,904 ----
  
      move-result-object v2
  
+     .line 115
      .local v2, values:[[B
      if-nez v2, :cond_1
  
+     .line 116
      const/4 v1, 0x0
  
+     .line 122
      :cond_0
      return-object v1
  
+     .line 118
      :cond_1
      array-length v3, v2
  
      new-array v1, v3, [Ljava/lang/String;
  
+     .line 119
      .local v1, strings:[Ljava/lang/String;
      const/4 v0, 0x0
  
***************
*** 824,829 ****
--- 908,914 ----
  
      if-ge v0, v3, :cond_0
  
+     .line 120
      aget-object v3, v2, v0
  
      invoke-static {v3}, Landroid/security/KeyStore;->toString([B)Ljava/lang/String;
***************
*** 832,837 ****
--- 917,923 ----
  
      aput-object v3, v1, v0
  
+     .line 119
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
***************
*** 842,847 ****
--- 928,934 ----
      .parameter "prefix"
  
      .prologue
+     .line 109
      const/16 v1, 0x73
  
      const/4 v2, 0x1
***************
*** 856,861 ****
--- 943,949 ----
  
      move-result-object v0
  
+     .line 110
      .local v0, values:Ljava/util/ArrayList;,"Ljava/util/ArrayList<[B>;"
      if-nez v0, :cond_0
  
***************
*** 886,891 ****
--- 974,980 ----
      .locals 2
  
      .prologue
+     .line 63
      const/16 v0, 0x74
  
      const/4 v1, 0x0
***************
*** 894,903 ****
--- 983,994 ----
  
      invoke-direct {p0, v0, v1}, Landroid/security/KeyStore;->execute(I[[B)Ljava/util/ArrayList;
  
+     .line 64
      iget v0, p0, Landroid/security/KeyStore;->mError:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 68
      new-instance v0, Ljava/lang/AssertionError;
  
      iget v1, p0, Landroid/security/KeyStore;->mError:I
***************
*** 906,927 ****
--- 997,1023 ----
  
      throw v0
  
+     .line 65
      :pswitch_0
      sget-object v0, Landroid/security/KeyStore$State;->UNLOCKED:Landroid/security/KeyStore$State;
  
+     .line 67
      :goto_0
      return-object v0
  
+     .line 66
      :pswitch_1
      sget-object v0, Landroid/security/KeyStore$State;->LOCKED:Landroid/security/KeyStore$State;
  
      goto :goto_0
  
+     .line 67
      :pswitch_2
      sget-object v0, Landroid/security/KeyStore$State;->UNINITIALIZED:Landroid/security/KeyStore$State;
  
      goto :goto_0
  
+     .line 64
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 935,940 ****
--- 1031,1037 ----
      .parameter "password"
  
      .prologue
+     .line 150
      invoke-static {p1}, Landroid/security/KeyStore;->getBytes(Ljava/lang/String;)[B
  
      move-result-object v0
