*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/net/NetworkPolicyManagerService$8.smali	2012-03-23 19:35:07.981732432 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/net/NetworkPolicyManagerService$8.smali.original	2012-03-23 19:34:47.306355419 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1409
      iput-object p1, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 43,63 ****
--- 44,69 ----
      .prologue
      const/4 v11, 0x1
  
+     .line 1412
      iget v10, p1, Landroid/os/Message;->what:I
  
      packed-switch v10, :pswitch_data_0
  
+     .line 1499
      const/4 v10, 0x0
  
      :goto_0
      return v10
  
+     .line 1414
      :pswitch_0
      iget v8, p1, Landroid/os/Message;->arg1:I
  
+     .line 1415
      .local v8, uid:I
      iget v9, p1, Landroid/os/Message;->arg2:I
  
+     .line 1416
      .local v9, uidRules:I
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 70,75 ****
--- 76,82 ----
  
      move-result v3
  
+     .line 1417
      .local v3, length:I
      const/4 v1, 0x0
  
***************
*** 77,82 ****
--- 84,90 ----
      :goto_1
      if-ge v1, v3, :cond_1
  
+     .line 1418
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #getter for: Lcom/android/server/net/NetworkPolicyManagerService;->mListeners:Landroid/os/RemoteCallbackList;
***************
*** 90,109 ****
--- 98,121 ----
  
      check-cast v4, Landroid/net/INetworkPolicyListener;
  
+     .line 1419
      .local v4, listener:Landroid/net/INetworkPolicyListener;
      if-eqz v4, :cond_0
  
+     .line 1421
      :try_start_0
      invoke-interface {v4, v8, v9}, Landroid/net/INetworkPolicyListener;->onUidRulesChanged(II)V
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1417
      :cond_0
      :goto_2
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 1426
      .end local v4           #listener:Landroid/net/INetworkPolicyListener;
      :cond_1
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
***************
*** 117,124 ****
--- 129,138 ----
  
      move v10, v11
  
+     .line 1427
      goto :goto_0
  
+     .line 1430
      .end local v1           #i:I
      .end local v3           #length:I
      .end local v8           #uid:I
***************
*** 132,137 ****
--- 146,152 ----
  
      check-cast v5, [Ljava/lang/String;
  
+     .line 1431
      .local v5, meteredIfaces:[Ljava/lang/String;
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 144,149 ****
--- 159,165 ----
  
      move-result v3
  
+     .line 1432
      .restart local v3       #length:I
      const/4 v1, 0x0
  
***************
*** 151,156 ****
--- 167,173 ----
      :goto_3
      if-ge v1, v3, :cond_3
  
+     .line 1433
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #getter for: Lcom/android/server/net/NetworkPolicyManagerService;->mListeners:Landroid/os/RemoteCallbackList;
***************
*** 164,183 ****
--- 181,204 ----
  
      check-cast v4, Landroid/net/INetworkPolicyListener;
  
+     .line 1434
      .restart local v4       #listener:Landroid/net/INetworkPolicyListener;
      if-eqz v4, :cond_2
  
+     .line 1436
      :try_start_1
      invoke-interface {v4, v5}, Landroid/net/INetworkPolicyListener;->onMeteredIfacesChanged([Ljava/lang/String;)V
      :try_end_1
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 1432
      :cond_2
      :goto_4
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_3
  
+     .line 1441
      .end local v4           #listener:Landroid/net/INetworkPolicyListener;
      :cond_3
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
***************
*** 191,207 ****
--- 212,232 ----
  
      move v10, v11
  
+     .line 1442
      goto :goto_0
  
+     .line 1445
      .end local v1           #i:I
      .end local v3           #length:I
      .end local v5           #meteredIfaces:[Ljava/lang/String;
      :pswitch_2
      iget v6, p1, Landroid/os/Message;->arg1:I
  
+     .line 1446
      .local v6, pid:I
      iget v8, p1, Landroid/os/Message;->arg2:I
  
+     .line 1447
      .restart local v8       #uid:I
      iget-object v10, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 211,216 ****
--- 236,242 ----
  
      move-result v0
  
+     .line 1449
      .local v0, foregroundActivities:Z
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 221,226 ****
--- 247,253 ----
  
      monitor-enter v12
  
+     .line 1454
      :try_start_2
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 235,243 ****
--- 262,272 ----
  
      check-cast v7, Landroid/util/SparseBooleanArray;
  
+     .line 1455
      .local v7, pidForeground:Landroid/util/SparseBooleanArray;
      if-nez v7, :cond_4
  
+     .line 1456
      new-instance v7, Landroid/util/SparseBooleanArray;
  
      .end local v7           #pidForeground:Landroid/util/SparseBooleanArray;
***************
*** 245,250 ****
--- 274,280 ----
  
      invoke-direct {v7, v10}, Landroid/util/SparseBooleanArray;-><init>(I)V
  
+     .line 1457
      .restart local v7       #pidForeground:Landroid/util/SparseBooleanArray;
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 255,274 ****
--- 285,309 ----
  
      invoke-virtual {v10, v8, v7}, Landroid/util/SparseArray;->put(ILjava/lang/Object;)V
  
+     .line 1459
      :cond_4
      invoke-virtual {v7, v6, v0}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 1460
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #calls: Lcom/android/server/net/NetworkPolicyManagerService;->computeUidForegroundLocked(I)V
      invoke-static {v10, v8}, Lcom/android/server/net/NetworkPolicyManagerService;->access$1400(Lcom/android/server/net/NetworkPolicyManagerService;I)V
  
+     .line 1461
      monitor-exit v12
  
      move v10, v11
  
+     .line 1462
      goto/16 :goto_0
  
+     .line 1461
      .end local v7           #pidForeground:Landroid/util/SparseBooleanArray;
      :catchall_0
      move-exception v10
***************
*** 279,293 ****
--- 314,331 ----
  
      throw v10
  
+     .line 1465
      .end local v0           #foregroundActivities:Z
      .end local v6           #pid:I
      .end local v8           #uid:I
      :pswitch_3
      iget v6, p1, Landroid/os/Message;->arg1:I
  
+     .line 1466
      .restart local v6       #pid:I
      iget v8, p1, Landroid/os/Message;->arg2:I
  
+     .line 1468
      .restart local v8       #uid:I
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 298,303 ****
--- 336,342 ----
  
      monitor-enter v12
  
+     .line 1470
      :try_start_3
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 312,334 ****
--- 351,379 ----
  
      check-cast v7, Landroid/util/SparseBooleanArray;
  
+     .line 1471
      .restart local v7       #pidForeground:Landroid/util/SparseBooleanArray;
      if-eqz v7, :cond_5
  
+     .line 1472
      invoke-virtual {v7, v6}, Landroid/util/SparseBooleanArray;->delete(I)V
  
+     .line 1473
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #calls: Lcom/android/server/net/NetworkPolicyManagerService;->computeUidForegroundLocked(I)V
      invoke-static {v10, v8}, Lcom/android/server/net/NetworkPolicyManagerService;->access$1400(Lcom/android/server/net/NetworkPolicyManagerService;I)V
  
+     .line 1475
      :cond_5
      monitor-exit v12
  
      move v10, v11
  
+     .line 1476
      goto/16 :goto_0
  
+     .line 1475
      .end local v7           #pidForeground:Landroid/util/SparseBooleanArray;
      :catchall_1
      move-exception v10
***************
*** 339,344 ****
--- 384,390 ----
  
      throw v10
  
+     .line 1479
      .end local v6           #pid:I
      .end local v8           #uid:I
      :pswitch_4
***************
*** 346,357 ****
--- 392,405 ----
  
      check-cast v2, Ljava/lang/String;
  
+     .line 1481
      .local v2, iface:Ljava/lang/String;
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #calls: Lcom/android/server/net/NetworkPolicyManagerService;->maybeRefreshTrustedTime()V
      invoke-static {v10}, Lcom/android/server/net/NetworkPolicyManagerService;->access$600(Lcom/android/server/net/NetworkPolicyManagerService;)V
  
+     .line 1482
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #getter for: Lcom/android/server/net/NetworkPolicyManagerService;->mRulesLock:Ljava/lang/Object;
***************
*** 361,366 ****
--- 409,415 ----
  
      monitor-enter v12
  
+     .line 1483
      :try_start_4
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 377,382 ****
--- 426,432 ----
  
      if-eqz v10, :cond_6
  
+     .line 1487
      :try_start_5
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 390,395 ****
--- 440,446 ----
      .catchall {:try_start_5 .. :try_end_5} :catchall_2
      .catch Landroid/os/RemoteException; {:try_start_5 .. :try_end_5} :catch_2
  
+     .line 1492
      :goto_5
      :try_start_6
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
***************
*** 397,414 ****
--- 448,469 ----
      #calls: Lcom/android/server/net/NetworkPolicyManagerService;->updateNetworkEnabledLocked()V
      invoke-static {v10}, Lcom/android/server/net/NetworkPolicyManagerService;->access$700(Lcom/android/server/net/NetworkPolicyManagerService;)V
  
+     .line 1493
      iget-object v10, p0, Lcom/android/server/net/NetworkPolicyManagerService$8;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #calls: Lcom/android/server/net/NetworkPolicyManagerService;->updateNotificationsLocked()V
      invoke-static {v10}, Lcom/android/server/net/NetworkPolicyManagerService;->access$800(Lcom/android/server/net/NetworkPolicyManagerService;)V
  
+     .line 1495
      :cond_6
      monitor-exit v12
  
      move v10, v11
  
+     .line 1496
      goto/16 :goto_0
  
+     .line 1495
      :catchall_2
      move-exception v10
  
***************
*** 418,423 ****
--- 473,479 ----
  
      throw v10
  
+     .line 1422
      .end local v2           #iface:Ljava/lang/String;
      .restart local v1       #i:I
      .restart local v3       #length:I
***************
*** 429,434 ****
--- 485,491 ----
  
      goto/16 :goto_2
  
+     .line 1437
      .end local v8           #uid:I
      .end local v9           #uidRules:I
      .restart local v5       #meteredIfaces:[Ljava/lang/String;
***************
*** 437,442 ****
--- 494,500 ----
  
      goto/16 :goto_4
  
+     .line 1488
      .end local v1           #i:I
      .end local v3           #length:I
      .end local v4           #listener:Landroid/net/INetworkPolicyListener;
***************
*** 447,452 ****
--- 505,511 ----
  
      goto :goto_5
  
+     .line 1412
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
