*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/net/NetworkPolicyManagerService$3.smali	2012-03-23 19:35:07.865730317 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/net/NetworkPolicyManagerService$3.smali.original	2012-03-23 19:34:47.305355402 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 360
      iput-object p1, p0, Lcom/android/server/net/NetworkPolicyManagerService$3;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 39,48 ****
--- 40,51 ----
      .parameter "intent"
  
      .prologue
+     .line 366
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 367
      .local v0, action:Ljava/lang/String;
      const-string v2, "android.intent.extra.UID"
  
***************
*** 52,57 ****
--- 55,61 ----
  
      move-result v1
  
+     .line 368
      .local v1, uid:I
      iget-object v2, p0, Lcom/android/server/net/NetworkPolicyManagerService$3;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
***************
*** 62,67 ****
--- 66,72 ----
  
      monitor-enter v3
  
+     .line 369
      :try_start_0
      const-string v2, "android.intent.action.PACKAGE_ADDED"
  
***************
*** 71,87 ****
--- 76,96 ----
  
      if-eqz v2, :cond_1
  
+     .line 373
      iget-object v2, p0, Lcom/android/server/net/NetworkPolicyManagerService$3;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #calls: Lcom/android/server/net/NetworkPolicyManagerService;->updateRulesForUidLocked(I)V
      invoke-static {v2, v1}, Lcom/android/server/net/NetworkPolicyManagerService;->access$300(Lcom/android/server/net/NetworkPolicyManagerService;I)V
  
+     .line 382
      :cond_0
      :goto_0
      monitor-exit v3
  
+     .line 383
      return-void
  
+     .line 375
      :cond_1
      const-string v2, "android.intent.action.UID_REMOVED"
  
***************
*** 91,96 ****
--- 100,106 ----
  
      if-eqz v2, :cond_0
  
+     .line 378
      iget-object v2, p0, Lcom/android/server/net/NetworkPolicyManagerService$3;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #getter for: Lcom/android/server/net/NetworkPolicyManagerService;->mUidPolicy:Landroid/util/SparseIntArray;
***************
*** 100,110 ****
--- 110,122 ----
  
      invoke-virtual {v2, v1}, Landroid/util/SparseIntArray;->delete(I)V
  
+     .line 379
      iget-object v2, p0, Lcom/android/server/net/NetworkPolicyManagerService$3;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #calls: Lcom/android/server/net/NetworkPolicyManagerService;->updateRulesForUidLocked(I)V
      invoke-static {v2, v1}, Lcom/android/server/net/NetworkPolicyManagerService;->access$300(Lcom/android/server/net/NetworkPolicyManagerService;I)V
  
+     .line 380
      iget-object v2, p0, Lcom/android/server/net/NetworkPolicyManagerService$3;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #calls: Lcom/android/server/net/NetworkPolicyManagerService;->writePolicyLocked()V
***************
*** 112,117 ****
--- 124,130 ----
  
      goto :goto_0
  
+     .line 382
      :catchall_0
      move-exception v2
  
