*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/net/NetworkStatsService$4.smali	2012-03-23 19:35:07.888730737 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/net/NetworkStatsService$4.smali.original	2012-03-23 19:34:47.305355402 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 640
      iput-object p1, p0, Lcom/android/server/net/NetworkStatsService$4;->this$0:Lcom/android/server/net/NetworkStatsService;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 39,44 ****
--- 40,46 ----
      .parameter "intent"
  
      .prologue
+     .line 645
      const-string v1, "android.intent.extra.UID"
  
      const/4 v2, 0x0
***************
*** 47,52 ****
--- 49,55 ----
  
      move-result v0
  
+     .line 646
      .local v0, uid:I
      iget-object v1, p0, Lcom/android/server/net/NetworkStatsService$4;->this$0:Lcom/android/server/net/NetworkStatsService;
  
***************
*** 57,62 ****
--- 60,66 ----
  
      monitor-enter v2
  
+     .line 647
      :try_start_0
      iget-object v1, p0, Lcom/android/server/net/NetworkStatsService$4;->this$0:Lcom/android/server/net/NetworkStatsService;
  
***************
*** 69,74 ****
--- 73,79 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_1
  
+     .line 649
      :try_start_1
      iget-object v1, p0, Lcom/android/server/net/NetworkStatsService$4;->this$0:Lcom/android/server/net/NetworkStatsService;
  
***************
*** 77,82 ****
--- 82,88 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 651
      :try_start_2
      iget-object v1, p0, Lcom/android/server/net/NetworkStatsService$4;->this$0:Lcom/android/server/net/NetworkStatsService;
  
***************
*** 87,96 ****
--- 93,105 ----
  
      invoke-virtual {v1}, Landroid/os/PowerManager$WakeLock;->release()V
  
+     .line 653
      monitor-exit v2
  
+     .line 654
      return-void
  
+     .line 651
      :catchall_0
      move-exception v1
  
***************
*** 105,110 ****
--- 114,120 ----
  
      throw v1
  
+     .line 653
      :catchall_1
      move-exception v1
  
