*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/net/NetworkPolicyManagerService$1.smali	2012-03-23 19:35:07.430722385 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/net/NetworkPolicyManagerService$1.smali.original	2012-03-23 19:34:47.300355300 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 336
      iput-object p1, p0, Lcom/android/server/net/NetworkPolicyManagerService$1;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      invoke-direct {p0}, Landroid/app/IProcessObserver$Stub;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .parameter "foregroundActivities"
  
      .prologue
+     .line 339
      iget-object v0, p0, Lcom/android/server/net/NetworkPolicyManagerService$1;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #getter for: Lcom/android/server/net/NetworkPolicyManagerService;->mHandler:Landroid/os/Handler;
***************
*** 59,64 ****
--- 61,67 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 341
      return-void
  .end method
  
***************
*** 68,73 ****
--- 71,77 ----
      .parameter "uid"
  
      .prologue
+     .line 345
      iget-object v0, p0, Lcom/android/server/net/NetworkPolicyManagerService$1;->this$0:Lcom/android/server/net/NetworkPolicyManagerService;
  
      #getter for: Lcom/android/server/net/NetworkPolicyManagerService;->mHandler:Landroid/os/Handler;
***************
*** 83,87 ****
--- 87,92 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 346
      return-void
  .end method
