*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/pm/PackageManagerService$13.smali	2012-03-23 19:35:11.039788196 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/pm/PackageManagerService$13.smali.original	2012-03-23 19:34:47.370356585 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .parameter
  
      .prologue
+     .line 8476
      iput-object p1, p0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
  
      iput p2, p0, Lcom/android/server/pm/PackageManagerService$13;->val$currentStatus:I
***************
*** 50,55 ****
--- 51,57 ----
      .locals 19
  
      .prologue
+     .line 8479
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 60,69 ****
--- 62,73 ----
  
      invoke-virtual {v14, v0}, Lcom/android/server/pm/PackageManagerService$PackageHandler;->removeCallbacks(Ljava/lang/Runnable;)V
  
+     .line 8480
      move-object/from16 v0, p0
  
      iget v11, v0, Lcom/android/server/pm/PackageManagerService$13;->val$currentStatus:I
  
+     .line 8481
      .local v11, returnCode:I
      move-object/from16 v0, p0
  
***************
*** 73,83 ****
--- 77,90 ----
  
      if-ne v14, v15, :cond_1
  
+     .line 8482
      const/4 v12, 0x0
  
+     .line 8483
      .local v12, uidArr:[I
      const/4 v8, 0x0
  
+     .line 8484
      .local v8, pkgList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
      move-object/from16 v0, p0
  
***************
*** 87,92 ****
--- 94,100 ----
  
      monitor-enter v15
  
+     .line 8485
      :try_start_0
      move-object/from16 v0, p0
  
***************
*** 114,122 ****
--- 122,132 ----
  
      check-cast v7, Landroid/content/pm/PackageParser$Package;
  
+     .line 8486
      .local v7, pkg:Landroid/content/pm/PackageParser$Package;
      if-nez v7, :cond_6
  
+     .line 8487
      const-string v14, "PackageManager"
  
      new-instance v16, Ljava/lang/StringBuilder;
***************
*** 159,175 ****
--- 169,189 ----
  
      invoke-static {v14, v0}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 8489
      const/4 v11, -0x2
  
+     .line 8503
      :goto_0
      monitor-exit v15
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 8504
      const/4 v14, 0x1
  
      if-ne v11, v14, :cond_1
  
+     .line 8506
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 183,188 ****
--- 197,203 ----
      #calls: Lcom/android/server/pm/PackageManagerService;->sendResourcesChangedBroadcast(ZLjava/util/ArrayList;[ILandroid/content/IIntentReceiver;)V
      invoke-static {v14, v15, v8, v12, v0}, Lcom/android/server/pm/PackageManagerService;->access$3500(Lcom/android/server/pm/PackageManagerService;ZLjava/util/ArrayList;[ILandroid/content/IIntentReceiver;)V
  
+     .line 8508
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 191,196 ****
--- 206,212 ----
  
      monitor-enter v15
  
+     .line 8509
      :try_start_1
      move-object/from16 v0, p0
  
***************
*** 204,209 ****
--- 220,226 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_2
  
+     .line 8510
      :try_start_2
      move-object/from16 v0, p0
  
***************
*** 232,240 ****
--- 249,259 ----
      .end local v7           #pkg:Landroid/content/pm/PackageParser$Package;
      check-cast v7, Landroid/content/pm/PackageParser$Package;
  
+     .line 8512
      .restart local v7       #pkg:Landroid/content/pm/PackageParser$Package;
      if-nez v7, :cond_8
  
+     .line 8513
      const-string v14, "PackageManager"
  
      new-instance v17, Ljava/lang/StringBuilder;
***************
*** 277,295 ****
--- 296,318 ----
  
      invoke-static {v14, v0}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 8515
      const/4 v11, -0x2
  
+     .line 8582
      :cond_0
      :goto_1
      monitor-exit v16
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_1
  
+     .line 8583
      :try_start_3
      monitor-exit v15
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_2
  
+     .line 8585
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 303,308 ****
--- 326,332 ----
      #calls: Lcom/android/server/pm/PackageManagerService;->sendResourcesChangedBroadcast(ZLjava/util/ArrayList;[ILandroid/content/IIntentReceiver;)V
      invoke-static {v14, v15, v8, v12, v0}, Lcom/android/server/pm/PackageManagerService;->access$3500(Lcom/android/server/pm/PackageManagerService;ZLjava/util/ArrayList;[ILandroid/content/IIntentReceiver;)V
  
+     .line 8588
      .end local v7           #pkg:Landroid/content/pm/PackageParser$Package;
      .end local v8           #pkgList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
      .end local v12           #uidArr:[I
***************
*** 311,316 ****
--- 335,341 ----
  
      if-eq v11, v14, :cond_f
  
+     .line 8590
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->val$mp:Lcom/android/server/pm/PackageManagerService$MoveParams;
***************
*** 319,324 ****
--- 344,350 ----
  
      if-eqz v14, :cond_2
  
+     .line 8591
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->val$mp:Lcom/android/server/pm/PackageManagerService$MoveParams;
***************
*** 329,340 ****
--- 355,368 ----
  
      invoke-virtual {v14, v15}, Lcom/android/server/pm/PackageManagerService$InstallArgs;->doPostInstall(I)I
  
+     .line 8604
      :cond_2
      :goto_2
      const/4 v14, -0x7
  
      if-eq v11, v14, :cond_4
  
+     .line 8605
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 343,348 ****
--- 371,377 ----
  
      monitor-enter v15
  
+     .line 8606
      :try_start_4
      move-object/from16 v0, p0
  
***************
*** 370,387 ****
--- 399,420 ----
  
      check-cast v7, Landroid/content/pm/PackageParser$Package;
  
+     .line 8607
      .restart local v7       #pkg:Landroid/content/pm/PackageParser$Package;
      if-eqz v7, :cond_3
  
+     .line 8608
      const/4 v14, 0x0
  
      iput-boolean v14, v7, Landroid/content/pm/PackageParser$Package;->mOperationPending:Z
  
+     .line 8610
      :cond_3
      monitor-exit v15
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_4
  
+     .line 8613
      .end local v7           #pkg:Landroid/content/pm/PackageParser$Package;
      :cond_4
      move-object/from16 v0, p0
***************
*** 390,398 ****
--- 423,433 ----
  
      iget-object v5, v14, Lcom/android/server/pm/PackageManagerService$MoveParams;->observer:Landroid/content/pm/IPackageMoveObserver;
  
+     .line 8614
      .local v5, observer:Landroid/content/pm/IPackageMoveObserver;
      if-eqz v5, :cond_5
  
+     .line 8616
      :try_start_5
      move-object/from16 v0, p0
  
***************
*** 404,413 ****
--- 439,450 ----
      :try_end_5
      .catch Landroid/os/RemoteException; {:try_start_5 .. :try_end_5} :catch_0
  
+     .line 8621
      :cond_5
      :goto_3
      return-void
  
+     .line 8490
      .end local v5           #observer:Landroid/content/pm/IPackageMoveObserver;
      .restart local v7       #pkg:Landroid/content/pm/PackageParser$Package;
      .restart local v8       #pkgList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
***************
*** 442,447 ****
--- 479,485 ----
  
      if-nez v14, :cond_7
  
+     .line 8491
      const-string v14, "PackageManager"
  
      new-instance v16, Ljava/lang/StringBuilder;
***************
*** 530,539 ****
--- 568,579 ----
  
      invoke-static {v14, v0}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 8495
      const/4 v11, -0x6
  
      goto/16 :goto_0
  
+     .line 8497
      :cond_7
      const/4 v14, 0x1
  
***************
*** 555,560 ****
--- 595,601 ----
      :try_end_6
      .catchall {:try_start_6 .. :try_end_6} :catchall_0
  
+     .line 8500
      .end local v12           #uidArr:[I
      .local v13, uidArr:[I
      :try_start_7
***************
*** 564,569 ****
--- 605,611 ----
      :try_end_7
      .catchall {:try_start_7 .. :try_end_7} :catchall_5
  
+     .line 8501
      .end local v8           #pkgList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
      .local v9, pkgList:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Ljava/lang/String;>;"
      :try_start_8
***************
*** 587,592 ****
--- 629,635 ----
      .restart local v12       #uidArr:[I
      goto/16 :goto_0
  
+     .line 8503
      .end local v7           #pkg:Landroid/content/pm/PackageParser$Package;
      :catchall_0
      move-exception v14
***************
*** 599,604 ****
--- 642,648 ----
  
      throw v14
  
+     .line 8516
      .restart local v7       #pkg:Landroid/content/pm/PackageParser$Package;
      :cond_8
      :try_start_a
***************
*** 630,635 ****
--- 674,680 ----
  
      if-nez v14, :cond_9
  
+     .line 8518
      const-string v14, "PackageManager"
  
      new-instance v17, Ljava/lang/StringBuilder;
***************
*** 718,730 ****
--- 763,778 ----
  
      invoke-static {v14, v0}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 8522
      const/4 v11, -0x6
  
      goto/16 :goto_1
  
+     .line 8524
      :cond_9
      iget-object v6, v7, Landroid/content/pm/PackageParser$Package;->mPath:Ljava/lang/String;
  
+     .line 8525
      .local v6, oldCodePath:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 736,741 ****
--- 784,790 ----
  
      move-result-object v2
  
+     .line 8526
      .local v2, newCodePath:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 747,752 ****
--- 796,802 ----
  
      move-result-object v4
  
+     .line 8527
      .local v4, newResPath:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 758,763 ****
--- 808,814 ----
  
      move-result-object v3
  
+     .line 8530
      .local v3, newNativePath:Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 769,774 ****
--- 820,826 ----
  
      if-nez v14, :cond_d
  
+     .line 8531
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 793,808 ****
--- 845,864 ----
  
      if-gez v14, :cond_c
  
+     .line 8533
      const/4 v11, -0x1
  
+     .line 8545
      :cond_a
      :goto_5
      const/4 v14, 0x1
  
      if-ne v11, v14, :cond_b
  
+     .line 8546
      iput-object v2, v7, Landroid/content/pm/PackageParser$Package;->mPath:Ljava/lang/String;
  
+     .line 8548
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 818,852 ****
--- 874,917 ----
  
      if-eq v14, v0, :cond_b
  
+     .line 8551
      iget-object v14, v7, Landroid/content/pm/PackageParser$Package;->mScanPath:Ljava/lang/String;
  
      iput-object v14, v7, Landroid/content/pm/PackageParser$Package;->mPath:Ljava/lang/String;
  
+     .line 8552
      const/4 v11, -0x1
  
+     .line 8556
      :cond_b
      const/4 v14, 0x1
  
      if-ne v11, v14, :cond_0
  
+     .line 8557
      iput-object v2, v7, Landroid/content/pm/PackageParser$Package;->mScanPath:Ljava/lang/String;
  
+     .line 8558
      iget-object v14, v7, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iput-object v2, v14, Landroid/content/pm/ApplicationInfo;->sourceDir:Ljava/lang/String;
  
+     .line 8559
      iget-object v14, v7, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iput-object v4, v14, Landroid/content/pm/ApplicationInfo;->publicSourceDir:Ljava/lang/String;
  
+     .line 8560
      iget-object v14, v7, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iput-object v3, v14, Landroid/content/pm/ApplicationInfo;->nativeLibraryDir:Ljava/lang/String;
  
+     .line 8561
      iget-object v10, v7, Landroid/content/pm/PackageParser$Package;->mExtras:Ljava/lang/Object;
  
      check-cast v10, Lcom/android/server/pm/PackageSetting;
  
+     .line 8562
      .local v10, ps:Lcom/android/server/pm/PackageSetting;
      new-instance v14, Ljava/io/File;
  
***************
*** 866,871 ****
--- 931,937 ----
  
      iput-object v14, v10, Lcom/android/server/pm/PackageSetting;->codePath:Ljava/io/File;
  
+     .line 8563
      iget-object v14, v10, Lcom/android/server/pm/PackageSetting;->codePath:Ljava/io/File;
  
      invoke-virtual {v14}, Ljava/io/File;->getPath()Ljava/lang/String;
***************
*** 874,879 ****
--- 940,946 ----
  
      iput-object v14, v10, Lcom/android/server/pm/PackageSetting;->codePathString:Ljava/lang/String;
  
+     .line 8564
      new-instance v14, Ljava/io/File;
  
      iget-object v0, v7, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
***************
*** 892,897 ****
--- 959,965 ----
  
      iput-object v14, v10, Lcom/android/server/pm/PackageSetting;->resourcePath:Ljava/io/File;
  
+     .line 8566
      iget-object v14, v10, Lcom/android/server/pm/PackageSetting;->resourcePath:Ljava/io/File;
  
      invoke-virtual {v14}, Ljava/io/File;->getPath()Ljava/lang/String;
***************
*** 900,907 ****
--- 968,977 ----
  
      iput-object v14, v10, Lcom/android/server/pm/PackageSetting;->resourcePathString:Ljava/lang/String;
  
+     .line 8567
      iput-object v3, v10, Lcom/android/server/pm/PackageSetting;->nativeLibraryPathString:Ljava/lang/String;
  
+     .line 8570
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->val$mp:Lcom/android/server/pm/PackageManagerService$MoveParams;
***************
*** 912,917 ****
--- 982,988 ----
  
      if-eqz v14, :cond_e
  
+     .line 8571
      iget-object v14, v7, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iget v0, v14, Landroid/content/pm/ApplicationInfo;->flags:I
***************
*** 926,931 ****
--- 997,1003 ----
  
      iput v0, v14, Landroid/content/pm/ApplicationInfo;->flags:I
  
+     .line 8575
      :goto_6
      iget-object v14, v7, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
***************
*** 933,938 ****
--- 1005,1011 ----
  
      invoke-virtual {v10, v14}, Lcom/android/server/pm/PackageSetting;->setFlags(I)V
  
+     .line 8576
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 941,946 ****
--- 1014,1020 ----
  
      invoke-virtual {v14, v6}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 8577
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 949,954 ****
--- 1023,1029 ----
  
      invoke-virtual {v14, v2, v7}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 8579
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 959,964 ****
--- 1034,1040 ----
  
      goto/16 :goto_1
  
+     .line 8582
      .end local v2           #newCodePath:Ljava/lang/String;
      .end local v3           #newNativePath:Ljava/lang/String;
      .end local v4           #newResPath:Ljava/lang/String;
***************
*** 975,980 ****
--- 1051,1057 ----
      :try_start_b
      throw v14
  
+     .line 8583
      :catchall_2
      move-exception v14
  
***************
*** 984,989 ****
--- 1061,1067 ----
  
      throw v14
  
+     .line 8535
      .restart local v2       #newCodePath:Ljava/lang/String;
      .restart local v3       #newNativePath:Ljava/lang/String;
      .restart local v4       #newResPath:Ljava/lang/String;
***************
*** 1007,1012 ****
--- 1085,1091 ----
  
      goto/16 :goto_5
  
+     .line 8539
      :cond_d
      move-object/from16 v0, p0
  
***************
*** 1032,1041 ****
--- 1111,1122 ----
  
      if-gez v14, :cond_a
  
+     .line 8541
      const/4 v11, -0x1
  
      goto/16 :goto_5
  
+     .line 8573
      .restart local v10       #ps:Lcom/android/server/pm/PackageSetting;
      :cond_e
      iget-object v14, v7, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
***************
*** 1056,1061 ****
--- 1137,1143 ----
  
      goto :goto_6
  
+     .line 8595
      .end local v2           #newCodePath:Ljava/lang/String;
      .end local v3           #newNativePath:Ljava/lang/String;
      .end local v4           #newResPath:Ljava/lang/String;
***************
*** 1071,1076 ****
--- 1153,1159 ----
  
      invoke-virtual {v14}, Ljava/lang/Runtime;->gc()V
  
+     .line 8597
      move-object/from16 v0, p0
  
      iget-object v14, v0, Lcom/android/server/pm/PackageManagerService$13;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 1079,1084 ****
--- 1162,1168 ----
  
      monitor-enter v15
  
+     .line 8598
      :try_start_d
      move-object/from16 v0, p0
  
***************
*** 1092,1097 ****
--- 1176,1182 ----
  
      invoke-virtual {v14, v0}, Lcom/android/server/pm/PackageManagerService$InstallArgs;->doPostDeleteLI(Z)Z
  
+     .line 8599
      monitor-exit v15
  
      goto/16 :goto_2
***************
*** 1105,1110 ****
--- 1190,1196 ----
  
      throw v14
  
+     .line 8610
      :catchall_4
      move-exception v14
  
***************
*** 1115,1124 ****
--- 1201,1212 ----
  
      throw v14
  
+     .line 8617
      .restart local v5       #observer:Landroid/content/pm/IPackageMoveObserver;
      :catch_0
      move-exception v1
  
+     .line 8618
      .local v1, e:Landroid/os/RemoteException;
      const-string v14, "PackageManager"
  
***************
*** 1128,1133 ****
--- 1216,1222 ----
  
      goto/16 :goto_3
  
+     .line 8503
      .end local v1           #e:Landroid/os/RemoteException;
      .end local v5           #observer:Landroid/content/pm/IPackageMoveObserver;
      .restart local v7       #pkg:Landroid/content/pm/PackageParser$Package;
