*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/pm/PackageManagerService$InstallParams$1.smali	2012-03-23 19:35:09.308756630 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/pm/PackageManagerService$InstallParams$1.smali.original	2012-03-23 19:34:47.354356306 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 5562
      iput-object p1, p0, Lcom/android/server/pm/PackageManagerService$InstallParams$1;->this$1:Lcom/android/server/pm/PackageManagerService$InstallParams;
  
      iput p2, p0, Lcom/android/server/pm/PackageManagerService$InstallParams$1;->val$verificationId:I
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "intent"
  
      .prologue
+     .line 5565
      iget-object v1, p0, Lcom/android/server/pm/PackageManagerService$InstallParams$1;->this$1:Lcom/android/server/pm/PackageManagerService$InstallParams;
  
      iget-object v1, v1, Lcom/android/server/pm/PackageManagerService$InstallParams;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 56,66 ****
--- 58,70 ----
  
      move-result-object v0
  
+     .line 5567
      .local v0, msg:Landroid/os/Message;
      iget v1, p0, Lcom/android/server/pm/PackageManagerService$InstallParams$1;->val$verificationId:I
  
      iput v1, v0, Landroid/os/Message;->arg1:I
  
+     .line 5568
      iget-object v1, p0, Lcom/android/server/pm/PackageManagerService$InstallParams$1;->this$1:Lcom/android/server/pm/PackageManagerService$InstallParams;
  
      iget-object v1, v1, Lcom/android/server/pm/PackageManagerService$InstallParams;->this$0:Lcom/android/server/pm/PackageManagerService;
***************
*** 78,82 ****
--- 82,87 ----
  
      invoke-virtual {v1, v0, v2, v3}, Lcom/android/server/pm/PackageManagerService$PackageHandler;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 5569
      return-void
  .end method
