*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/pm/PackageManagerService$PostInstallData.smali	2012-03-23 19:35:10.434777163 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/pm/PackageManagerService$PostInstallData.smali.original	2012-03-23 19:34:47.365356495 -0500
***************
*** 30,42 ****
--- 30,46 ----
      .parameter "_r"
  
      .prologue
+     .line 429
      iput-object p1, p0, Lcom/android/server/pm/PackageManagerService$PostInstallData;->this$0:Lcom/android/server/pm/PackageManagerService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 430
      iput-object p2, p0, Lcom/android/server/pm/PackageManagerService$PostInstallData;->args:Lcom/android/server/pm/PackageManagerService$InstallArgs;
  
+     .line 431
      iput-object p3, p0, Lcom/android/server/pm/PackageManagerService$PostInstallData;->res:Lcom/android/server/pm/PackageManagerService$PackageInstalledInfo;
  
+     .line 432
      return-void
  .end method
