*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/AndroidHttpClient$LoggingConfiguration.smali	2012-03-23 19:37:00.550785125 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/AndroidHttpClient$LoggingConfiguration.smali.original	2012-03-23 19:34:45.481322137 -0500
***************
*** 27,38 ****
--- 27,42 ----
      .parameter "level"
  
      .prologue
+     .line 335
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 336
      iput-object p1, p0, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;->tag:Ljava/lang/String;
  
+     .line 337
      iput p2, p0, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;->level:I
  
+     .line 338
      return-void
  .end method
  
***************
*** 43,48 ****
--- 47,53 ----
      .parameter "x2"
  
      .prologue
+     .line 330
      invoke-direct {p0, p1, p2}, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 53,58 ****
--- 58,64 ----
      .parameter "x0"
  
      .prologue
+     .line 330
      invoke-direct {p0}, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;->isLoggable()Z
  
      move-result v0
***************
*** 66,71 ****
--- 72,78 ----
      .parameter "x1"
  
      .prologue
+     .line 330
      invoke-direct {p0, p1}, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;->println(Ljava/lang/String;)V
  
      return-void
***************
*** 75,80 ****
--- 82,88 ----
      .locals 2
  
      .prologue
+     .line 344
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;->tag:Ljava/lang/String;
  
      iget v1, p0, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;->level:I
***************
*** 91,101 ****
--- 99,111 ----
      .parameter "message"
  
      .prologue
+     .line 351
      iget v0, p0, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;->level:I
  
      iget-object v1, p0, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;->tag:Ljava/lang/String;
  
      invoke-static {v0, v1, p1}, Landroid/util/Log;->println(ILjava/lang/String;Ljava/lang/String;)I
  
+     .line 352
      return-void
  .end method
