*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/AndroidHttpClient.smali	2012-03-23 19:37:00.972792821 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/AndroidHttpClient.smali.original	2012-03-23 19:34:45.487322247 -0500
***************
*** 40,56 ****
      .locals 3
  
      .prologue
      const-wide/16 v0, 0x100
  
      sput-wide v0, Landroid/net/http/AndroidHttpClient;->DEFAULT_SYNC_MIN_GZIP_BYTES:J
  
      const/4 v0, 0x3
  
      new-array v0, v0, [Ljava/lang/String;
  
      const/4 v1, 0x0
  
!     const-string v2, "text/"
  
      aput-object v2, v0, v1
  
--- 40,58 ----
      .locals 3
  
      .prologue
+     .line 81
      const-wide/16 v0, 0x100
  
      sput-wide v0, Landroid/net/http/AndroidHttpClient;->DEFAULT_SYNC_MIN_GZIP_BYTES:J
  
+     .line 88
      const/4 v0, 0x3
  
      new-array v0, v0, [Ljava/lang/String;
  
      const/4 v1, 0x0
  
!     const-string/jumbo v2, "text/"
  
      aput-object v2, v0, v1
  
***************
*** 68,73 ****
--- 70,76 ----
  
      sput-object v0, Landroid/net/http/AndroidHttpClient;->textContentTypes:[Ljava/lang/String;
  
+     .line 95
      new-instance v0, Landroid/net/http/AndroidHttpClient$1;
  
      invoke-direct {v0}, Landroid/net/http/AndroidHttpClient$1;-><init>()V
***************
*** 83,90 ****
--- 86,95 ----
      .parameter "params"
  
      .prologue
+     .line 161
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 158
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "AndroidHttpClient created and never closed"
***************
*** 93,104 ****
--- 98,111 ----
  
      iput-object v0, p0, Landroid/net/http/AndroidHttpClient;->mLeakedException:Ljava/lang/RuntimeException;
  
+     .line 162
      new-instance v0, Landroid/net/http/AndroidHttpClient$2;
  
      invoke-direct {v0, p0, p1, p2}, Landroid/net/http/AndroidHttpClient$2;-><init>(Landroid/net/http/AndroidHttpClient;Lorg/apache/http/conn/ClientConnectionManager;Lorg/apache/http/params/HttpParams;)V
  
      iput-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
+     .line 190
      return-void
  .end method
  
***************
*** 106,111 ****
--- 113,119 ----
      .locals 1
  
      .prologue
+     .line 78
      sget-object v0, Landroid/net/http/AndroidHttpClient;->sThreadCheckInterceptor:Lorg/apache/http/HttpRequestInterceptor;
  
      return-object v0
***************
*** 116,121 ****
--- 124,130 ----
      .parameter "x0"
  
      .prologue
+     .line 78
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->curlConfiguration:Landroid/net/http/AndroidHttpClient$LoggingConfiguration;
  
      return-object v0
***************
*** 132,137 ****
--- 141,147 ----
      .end annotation
  
      .prologue
+     .line 78
      invoke-static {p0, p1}, Landroid/net/http/AndroidHttpClient;->toCurl(Lorg/apache/http/client/methods/HttpUriRequest;Z)Ljava/lang/String;
  
      move-result-object v0
***************
*** 150,155 ****
--- 160,166 ----
      .end annotation
  
      .prologue
+     .line 304
      array-length v3, p0
  
      int-to-long v3, v3
***************
*** 162,191 ****
--- 173,209 ----
  
      if-gez v3, :cond_0
  
+     .line 305
      new-instance v1, Lorg/apache/http/entity/ByteArrayEntity;
  
      invoke-direct {v1, p0}, Lorg/apache/http/entity/ByteArrayEntity;-><init>([B)V
  
+     .line 314
      .local v1, entity:Lorg/apache/http/entity/AbstractHttpEntity;
      :goto_0
      return-object v1
  
+     .line 307
      .end local v1           #entity:Lorg/apache/http/entity/AbstractHttpEntity;
      :cond_0
      new-instance v0, Ljava/io/ByteArrayOutputStream;
  
      invoke-direct {v0}, Ljava/io/ByteArrayOutputStream;-><init>()V
  
+     .line 308
      .local v0, arr:Ljava/io/ByteArrayOutputStream;
      new-instance v2, Ljava/util/zip/GZIPOutputStream;
  
      invoke-direct {v2, v0}, Ljava/util/zip/GZIPOutputStream;-><init>(Ljava/io/OutputStream;)V
  
+     .line 309
      .local v2, zipper:Ljava/io/OutputStream;
      invoke-virtual {v2, p0}, Ljava/io/OutputStream;->write([B)V
  
+     .line 310
      invoke-virtual {v2}, Ljava/io/OutputStream;->close()V
  
+     .line 311
      new-instance v1, Lorg/apache/http/entity/ByteArrayEntity;
  
      invoke-virtual {v0}, Ljava/io/ByteArrayOutputStream;->toByteArray()[B
***************
*** 194,199 ****
--- 212,218 ----
  
      invoke-direct {v1, v3}, Lorg/apache/http/entity/ByteArrayEntity;-><init>([B)V
  
+     .line 312
      .restart local v1       #entity:Lorg/apache/http/entity/AbstractHttpEntity;
      const-string v3, "gzip"
  
***************
*** 207,212 ****
--- 226,232 ----
      .parameter "resolver"
  
      .prologue
+     .line 322
      sget-wide v0, Landroid/net/http/AndroidHttpClient;->DEFAULT_SYNC_MIN_GZIP_BYTES:J
  
      return-wide v0
***************
*** 222,241 ****
--- 242,265 ----
      .end annotation
  
      .prologue
+     .line 221
      invoke-interface {p0}, Lorg/apache/http/HttpEntity;->getContent()Ljava/io/InputStream;
  
      move-result-object v2
  
+     .line 222
      .local v2, responseStream:Ljava/io/InputStream;
      if-nez v2, :cond_0
  
      move-object v3, v2
  
+     .line 229
      .end local v2           #responseStream:Ljava/io/InputStream;
      .local v3, responseStream:Ljava/io/InputStream;
      :goto_0
      return-object v3
  
+     .line 223
      .end local v3           #responseStream:Ljava/io/InputStream;
      .restart local v2       #responseStream:Ljava/io/InputStream;
      :cond_0
***************
*** 243,248 ****
--- 267,273 ----
  
      move-result-object v1
  
+     .line 224
      .local v1, header:Lorg/apache/http/Header;
      if-nez v1, :cond_1
  
***************
*** 252,257 ****
--- 277,283 ----
      .restart local v3       #responseStream:Ljava/io/InputStream;
      goto :goto_0
  
+     .line 225
      .end local v3           #responseStream:Ljava/io/InputStream;
      .restart local v2       #responseStream:Ljava/io/InputStream;
      :cond_1
***************
*** 259,264 ****
--- 285,291 ----
  
      move-result-object v0
  
+     .line 226
      .local v0, contentEncoding:Ljava/lang/String;
      if-nez v0, :cond_2
  
***************
*** 268,273 ****
--- 295,301 ----
      .restart local v3       #responseStream:Ljava/io/InputStream;
      goto :goto_0
  
+     .line 227
      .end local v3           #responseStream:Ljava/io/InputStream;
      .restart local v2       #responseStream:Ljava/io/InputStream;
      :cond_2
***************
*** 292,297 ****
--- 320,326 ----
      :cond_3
      move-object v3, v2
  
+     .line 229
      .end local v2           #responseStream:Ljava/io/InputStream;
      .restart local v3       #responseStream:Ljava/io/InputStream;
      goto :goto_0
***************
*** 304,318 ****
--- 333,350 ----
      .prologue
      const/4 v9, 0x1
  
+     .line 465
      const-string v10, "content-encoding"
  
      invoke-interface {p0, v10}, Lorg/apache/http/client/methods/HttpUriRequest;->getHeaders(Ljava/lang/String;)[Lorg/apache/http/Header;
  
      move-result-object v4
  
+     .line 466
      .local v4, headers:[Lorg/apache/http/Header;
      if-eqz v4, :cond_2
  
+     .line 467
      move-object v0, v4
  
      .local v0, arr$:[Lorg/apache/http/Header;
***************
*** 327,332 ****
--- 359,365 ----
  
      aget-object v3, v0, v5
  
+     .line 468
      .local v3, header:Lorg/apache/http/Header;
      const-string v10, "gzip"
  
***************
*** 340,345 ****
--- 373,379 ----
  
      if-eqz v10, :cond_1
  
+     .line 484
      .end local v0           #arr$:[Lorg/apache/http/Header;
      .end local v3           #header:Lorg/apache/http/Header;
      .end local v5           #i$:I
***************
*** 348,353 ****
--- 382,388 ----
      :goto_1
      return v9
  
+     .line 467
      .restart local v0       #arr$:[Lorg/apache/http/Header;
      .restart local v3       #header:Lorg/apache/http/Header;
      .restart local v5       #i$:I
***************
*** 357,362 ****
--- 392,398 ----
  
      goto :goto_0
  
+     .line 474
      .end local v0           #arr$:[Lorg/apache/http/Header;
      .end local v3           #header:Lorg/apache/http/Header;
      .end local v5           #i$:I
***************
*** 368,375 ****
--- 404,413 ----
  
      move-result-object v4
  
+     .line 475
      if-eqz v4, :cond_0
  
+     .line 476
      move-object v0, v4
  
      .restart local v0       #arr$:[Lorg/apache/http/Header;
***************
*** 390,395 ****
--- 428,434 ----
  
      aget-object v3, v0, v6
  
+     .line 477
      .restart local v3       #header:Lorg/apache/http/Header;
      sget-object v1, Landroid/net/http/AndroidHttpClient;->textContentTypes:[Ljava/lang/String;
  
***************
*** 406,411 ****
--- 445,451 ----
  
      aget-object v2, v1, v5
  
+     .line 478
      .local v2, contentType:Ljava/lang/String;
      invoke-interface {v3}, Lorg/apache/http/Header;->getValue()Ljava/lang/String;
  
***************
*** 417,431 ****
--- 457,474 ----
  
      if-eqz v10, :cond_3
  
+     .line 479
      const/4 v9, 0x0
  
      goto :goto_1
  
+     .line 477
      :cond_3
      add-int/lit8 v5, v5, 0x1
  
      goto :goto_3
  
+     .line 476
      .end local v2           #contentType:Ljava/lang/String;
      :cond_4
      add-int/lit8 v5, v6, 0x1
***************
*** 442,453 ****
--- 485,498 ----
      .parameter "request"
  
      .prologue
+     .line 208
      const-string v0, "Accept-Encoding"
  
      const-string v1, "gzip"
  
      invoke-interface {p0, v0, v1}, Lorg/apache/http/HttpRequest;->addHeader(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 209
      return-void
  .end method
  
***************
*** 456,461 ****
--- 501,507 ----
      .parameter "userAgent"
  
      .prologue
+     .line 153
      const/4 v0, 0x0
  
      invoke-static {p0, v0}, Landroid/net/http/AndroidHttpClient;->newInstance(Ljava/lang/String;Landroid/content/Context;)Landroid/net/http/AndroidHttpClient;
***************
*** 475,509 ****
--- 521,565 ----
  
      const v8, 0xea60
  
+     .line 113
      new-instance v1, Lorg/apache/http/params/BasicHttpParams;
  
      invoke-direct {v1}, Lorg/apache/http/params/BasicHttpParams;-><init>()V
  
+     .line 117
      .local v1, params:Lorg/apache/http/params/HttpParams;
      invoke-static {v1, v5}, Lorg/apache/http/params/HttpConnectionParams;->setStaleCheckingEnabled(Lorg/apache/http/params/HttpParams;Z)V
  
+     .line 119
      invoke-static {v1, v8}, Lorg/apache/http/params/HttpConnectionParams;->setConnectionTimeout(Lorg/apache/http/params/HttpParams;I)V
  
+     .line 120
      invoke-static {v1, v8}, Lorg/apache/http/params/HttpConnectionParams;->setSoTimeout(Lorg/apache/http/params/HttpParams;I)V
  
+     .line 121
      const/16 v4, 0x2000
  
      invoke-static {v1, v4}, Lorg/apache/http/params/HttpConnectionParams;->setSocketBufferSize(Lorg/apache/http/params/HttpParams;I)V
  
+     .line 125
      invoke-static {v1, v5}, Lorg/apache/http/client/params/HttpClientParams;->setRedirecting(Lorg/apache/http/params/HttpParams;Z)V
  
+     .line 128
      if-nez p1, :cond_0
  
      const/4 v3, 0x0
  
+     .line 131
      .local v3, sessionCache:Landroid/net/SSLSessionCache;
      :goto_0
      invoke-static {v1, p0}, Lorg/apache/http/params/HttpProtocolParams;->setUserAgent(Lorg/apache/http/params/HttpParams;Ljava/lang/String;)V
  
+     .line 132
      new-instance v2, Lorg/apache/http/conn/scheme/SchemeRegistry;
  
      invoke-direct {v2}, Lorg/apache/http/conn/scheme/SchemeRegistry;-><init>()V
  
+     .line 133
      .local v2, schemeRegistry:Lorg/apache/http/conn/scheme/SchemeRegistry;
      new-instance v4, Lorg/apache/http/conn/scheme/Scheme;
  
***************
*** 519,524 ****
--- 575,581 ----
  
      invoke-virtual {v2, v4}, Lorg/apache/http/conn/scheme/SchemeRegistry;->register(Lorg/apache/http/conn/scheme/Scheme;)Lorg/apache/http/conn/scheme/Scheme;
  
+     .line 135
      new-instance v4, Lorg/apache/http/conn/scheme/Scheme;
  
      const-string v5, "https"
***************
*** 533,542 ****
--- 590,601 ----
  
      invoke-virtual {v2, v4}, Lorg/apache/http/conn/scheme/SchemeRegistry;->register(Lorg/apache/http/conn/scheme/Scheme;)Lorg/apache/http/conn/scheme/Scheme;
  
+     .line 139
      new-instance v0, Lorg/apache/http/impl/conn/tsccm/ThreadSafeClientConnManager;
  
      invoke-direct {v0, v1, v2}, Lorg/apache/http/impl/conn/tsccm/ThreadSafeClientConnManager;-><init>(Lorg/apache/http/params/HttpParams;Lorg/apache/http/conn/scheme/SchemeRegistry;)V
  
+     .line 144
      .local v0, manager:Lorg/apache/http/conn/ClientConnectionManager;
      new-instance v4, Landroid/net/http/AndroidHttpClient;
  
***************
*** 544,549 ****
--- 603,609 ----
  
      return-object v4
  
+     .line 128
      .end local v0           #manager:Lorg/apache/http/conn/ClientConnectionManager;
      .end local v2           #schemeRegistry:Lorg/apache/http/conn/scheme/SchemeRegistry;
      .end local v3           #sessionCache:Landroid/net/SSLSessionCache;
***************
*** 560,565 ****
--- 620,626 ----
      .parameter "dateString"
  
      .prologue
+     .line 502
      invoke-static {p0}, Lcom/android/internal/http/HttpDateTime;->parse(Ljava/lang/String;)J
  
      move-result-wide v0
***************
*** 578,592 ****
--- 639,656 ----
      .end annotation
  
      .prologue
+     .line 404
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 406
      .local v3, builder:Ljava/lang/StringBuilder;
      const-string v13, "curl "
  
      invoke-virtual {v3, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 408
      invoke-interface/range {p0 .. p0}, Lorg/apache/http/client/methods/HttpUriRequest;->getAllHeaders()[Lorg/apache/http/Header;
  
      move-result-object v1
***************
*** 603,608 ****
--- 667,673 ----
  
      aget-object v7, v1, v8
  
+     .line 409
      .local v7, header:Lorg/apache/http/Header;
      if-nez p1, :cond_1
  
***************
*** 630,646 ****
--- 695,714 ----
  
      if-eqz v13, :cond_1
  
+     .line 408
      :cond_0
      :goto_1
      add-int/lit8 v8, v8, 0x1
  
      goto :goto_0
  
+     .line 414
      :cond_1
      const-string v13, "--header \""
  
      invoke-virtual {v3, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 415
      invoke-virtual {v7}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
      move-result-object v13
***************
*** 651,668 ****
--- 719,739 ----
  
      invoke-virtual {v3, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 416
      const-string v13, "\" "
  
      invoke-virtual {v3, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      goto :goto_1
  
+     .line 419
      .end local v7           #header:Lorg/apache/http/Header;
      :cond_2
      invoke-interface/range {p0 .. p0}, Lorg/apache/http/client/methods/HttpUriRequest;->getURI()Ljava/net/URI;
  
      move-result-object v12
  
+     .line 424
      .local v12, uri:Ljava/net/URI;
      move-object/from16 v0, p0
  
***************
*** 672,688 ****
--- 743,762 ----
  
      move-object/from16 v13, p0
  
+     .line 425
      check-cast v13, Lorg/apache/http/impl/client/RequestWrapper;
  
      invoke-virtual {v13}, Lorg/apache/http/impl/client/RequestWrapper;->getOriginal()Lorg/apache/http/HttpRequest;
  
      move-result-object v10
  
+     .line 426
      .local v10, original:Lorg/apache/http/HttpRequest;
      instance-of v13, v10, Lorg/apache/http/client/methods/HttpUriRequest;
  
      if-eqz v13, :cond_3
  
+     .line 427
      check-cast v10, Lorg/apache/http/client/methods/HttpUriRequest;
  
      .end local v10           #original:Lorg/apache/http/HttpRequest;
***************
*** 690,706 ****
--- 764,784 ----
  
      move-result-object v12
  
+     .line 431
      :cond_3
      const-string v13, "\""
  
      invoke-virtual {v3, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 432
      invoke-virtual {v3, v12}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
  
+     .line 433
      const-string v13, "\""
  
      invoke-virtual {v3, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 435
      move-object/from16 v0, p0
  
      instance-of v13, v0, Lorg/apache/http/HttpEntityEnclosingRequest;
***************
*** 709,721 ****
--- 787,802 ----
  
      move-object/from16 v5, p0
  
+     .line 436
      check-cast v5, Lorg/apache/http/HttpEntityEnclosingRequest;
  
+     .line 438
      .local v5, entityRequest:Lorg/apache/http/HttpEntityEnclosingRequest;
      invoke-interface {v5}, Lorg/apache/http/HttpEntityEnclosingRequest;->getEntity()Lorg/apache/http/HttpEntity;
  
      move-result-object v4
  
+     .line 439
      .local v4, entity:Lorg/apache/http/HttpEntity;
      if-eqz v4, :cond_4
  
***************
*** 725,730 ****
--- 806,812 ----
  
      if-eqz v13, :cond_4
  
+     .line 440
      invoke-interface {v4}, Lorg/apache/http/HttpEntity;->getContentLength()J
  
      move-result-wide v13
***************
*** 735,753 ****
--- 817,839 ----
  
      if-gez v13, :cond_6
  
+     .line 441
      new-instance v11, Ljava/io/ByteArrayOutputStream;
  
      invoke-direct {v11}, Ljava/io/ByteArrayOutputStream;-><init>()V
  
+     .line 442
      .local v11, stream:Ljava/io/ByteArrayOutputStream;
      invoke-interface {v4, v11}, Lorg/apache/http/HttpEntity;->writeTo(Ljava/io/OutputStream;)V
  
+     .line 444
      invoke-static/range {p0 .. p0}, Landroid/net/http/AndroidHttpClient;->isBinaryContent(Lorg/apache/http/client/methods/HttpUriRequest;)Z
  
      move-result v13
  
      if-eqz v13, :cond_5
  
+     .line 445
      invoke-virtual {v11}, Ljava/io/ByteArrayOutputStream;->toByteArray()[B
  
      move-result-object v13
***************
*** 758,763 ****
--- 844,850 ----
  
      move-result-object v2
  
+     .line 446
      .local v2, base64:Ljava/lang/String;
      const/4 v13, 0x0
  
***************
*** 787,796 ****
--- 874,885 ----
  
      invoke-virtual {v3, v13, v14}, Ljava/lang/StringBuilder;->insert(ILjava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 447
      const-string v13, " --data-binary @/tmp/$$.bin"
  
      invoke-virtual {v3, v13}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 460
      .end local v2           #base64:Ljava/lang/String;
      .end local v4           #entity:Lorg/apache/http/HttpEntity;
      .end local v5           #entityRequest:Lorg/apache/http/HttpEntityEnclosingRequest;
***************
*** 803,808 ****
--- 892,898 ----
  
      return-object v13
  
+     .line 449
      .restart local v4       #entity:Lorg/apache/http/HttpEntity;
      .restart local v5       #entityRequest:Lorg/apache/http/HttpEntityEnclosingRequest;
      .restart local v11       #stream:Ljava/io/ByteArrayOutputStream;
***************
*** 811,816 ****
--- 901,907 ----
  
      move-result-object v6
  
+     .line 450
      .local v6, entityString:Ljava/lang/String;
      const-string v13, " --data-ascii \""
  
***************
*** 828,833 ****
--- 919,925 ----
  
      goto :goto_2
  
+     .line 455
      .end local v6           #entityString:Ljava/lang/String;
      .end local v11           #stream:Ljava/io/ByteArrayOutputStream;
      :cond_6
***************
*** 844,863 ****
--- 936,959 ----
      .locals 1
  
      .prologue
+     .line 237
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->mLeakedException:Ljava/lang/RuntimeException;
  
      if-eqz v0, :cond_0
  
+     .line 238
      invoke-virtual {p0}, Landroid/net/http/AndroidHttpClient;->getConnectionManager()Lorg/apache/http/conn/ClientConnectionManager;
  
      move-result-object v0
  
      invoke-interface {v0}, Lorg/apache/http/conn/ClientConnectionManager;->shutdown()V
  
+     .line 239
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/net/http/AndroidHttpClient;->mLeakedException:Ljava/lang/RuntimeException;
  
+     .line 241
      :cond_0
      return-void
  .end method
***************
*** 866,875 ****
--- 962,973 ----
      .locals 1
  
      .prologue
+     .line 380
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/net/http/AndroidHttpClient;->curlConfiguration:Landroid/net/http/AndroidHttpClient$LoggingConfiguration;
  
+     .line 381
      return-void
  .end method
  
***************
*** 879,894 ****
      .parameter "level"
  
      .prologue
      if-nez p1, :cond_0
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "name"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      const/4 v0, 0x2
  
--- 977,995 ----
      .parameter "level"
  
      .prologue
+     .line 365
      if-nez p1, :cond_0
  
+     .line 366
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "name"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 368
      :cond_0
      const/4 v0, 0x2
  
***************
*** 898,903 ****
--- 999,1005 ----
  
      if-le p2, v0, :cond_2
  
+     .line 369
      :cond_1
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
***************
*** 907,912 ****
--- 1009,1015 ----
  
      throw v0
  
+     .line 373
      :cond_2
      new-instance v0, Landroid/net/http/AndroidHttpClient$LoggingConfiguration;
  
***************
*** 916,921 ****
--- 1019,1025 ----
  
      iput-object v0, p0, Landroid/net/http/AndroidHttpClient;->curlConfiguration:Landroid/net/http/AndroidHttpClient$LoggingConfiguration;
  
+     .line 374
      return-void
  .end method
  
***************
*** 944,949 ****
--- 1048,1054 ----
      .end annotation
  
      .prologue
+     .line 285
      .local p3, responseHandler:Lorg/apache/http/client/ResponseHandler;,"Lorg/apache/http/client/ResponseHandler<+TT;>;"
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
***************
*** 982,987 ****
--- 1087,1093 ----
      .end annotation
  
      .prologue
+     .line 291
      .local p3, responseHandler:Lorg/apache/http/client/ResponseHandler;,"Lorg/apache/http/client/ResponseHandler<+TT;>;"
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
***************
*** 1015,1020 ****
--- 1121,1127 ----
      .end annotation
  
      .prologue
+     .line 273
      .local p2, responseHandler:Lorg/apache/http/client/ResponseHandler;,"Lorg/apache/http/client/ResponseHandler<+TT;>;"
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
***************
*** 1051,1056 ****
--- 1158,1164 ----
      .end annotation
  
      .prologue
+     .line 279
      .local p2, responseHandler:Lorg/apache/http/client/ResponseHandler;,"Lorg/apache/http/client/ResponseHandler<+TT;>;"
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
***************
*** 1072,1077 ****
--- 1180,1186 ----
      .end annotation
  
      .prologue
+     .line 262
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
      invoke-interface {v0, p1, p2}, Lorg/apache/http/client/HttpClient;->execute(Lorg/apache/http/HttpHost;Lorg/apache/http/HttpRequest;)Lorg/apache/http/HttpResponse;
***************
*** 1093,1098 ****
--- 1202,1208 ----
      .end annotation
  
      .prologue
+     .line 267
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
      invoke-interface {v0, p1, p2, p3}, Lorg/apache/http/client/HttpClient;->execute(Lorg/apache/http/HttpHost;Lorg/apache/http/HttpRequest;Lorg/apache/http/protocol/HttpContext;)Lorg/apache/http/HttpResponse;
***************
*** 1112,1117 ****
--- 1222,1228 ----
      .end annotation
  
      .prologue
+     .line 252
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
      invoke-interface {v0, p1}, Lorg/apache/http/client/HttpClient;->execute(Lorg/apache/http/client/methods/HttpUriRequest;)Lorg/apache/http/HttpResponse;
***************
*** 1132,1137 ****
--- 1243,1249 ----
      .end annotation
  
      .prologue
+     .line 257
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
      invoke-interface {v0, p1, p2}, Lorg/apache/http/client/HttpClient;->execute(Lorg/apache/http/client/methods/HttpUriRequest;Lorg/apache/http/protocol/HttpContext;)Lorg/apache/http/HttpResponse;
***************
*** 1150,1161 ****
--- 1262,1276 ----
      .end annotation
  
      .prologue
+     .line 194
      invoke-super {p0}, Ljava/lang/Object;->finalize()V
  
+     .line 195
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->mLeakedException:Ljava/lang/RuntimeException;
  
      if-eqz v0, :cond_0
  
+     .line 196
      const-string v0, "AndroidHttpClient"
  
      const-string v1, "Leak found"
***************
*** 1164,1173 ****
--- 1279,1290 ----
  
      invoke-static {v0, v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 197
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/net/http/AndroidHttpClient;->mLeakedException:Ljava/lang/RuntimeException;
  
+     .line 199
      :cond_0
      return-void
  .end method
***************
*** 1176,1181 ****
--- 1293,1299 ----
      .locals 1
  
      .prologue
+     .line 248
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
      invoke-interface {v0}, Lorg/apache/http/client/HttpClient;->getConnectionManager()Lorg/apache/http/conn/ClientConnectionManager;
***************
*** 1189,1194 ****
--- 1307,1313 ----
      .locals 1
  
      .prologue
+     .line 244
      iget-object v0, p0, Landroid/net/http/AndroidHttpClient;->delegate:Lorg/apache/http/client/HttpClient;
  
      invoke-interface {v0}, Lorg/apache/http/client/HttpClient;->getParams()Lorg/apache/http/params/HttpParams;
