*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/CertificateChainValidator.smali	2012-03-23 19:37:00.760788956 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/CertificateChainValidator.smali.original	2012-03-23 19:34:45.484322193 -0500
***************
*** 12,17 ****
--- 12,18 ----
      .locals 1
  
      .prologue
+     .line 53
      new-instance v0, Landroid/net/http/CertificateChainValidator;
  
      invoke-direct {v0}, Landroid/net/http/CertificateChainValidator;-><init>()V
***************
*** 25,30 ****
--- 26,32 ----
      .locals 0
  
      .prologue
+     .line 67
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 41,60 ****
--- 43,68 ----
      .end annotation
  
      .prologue
+     .line 183
      if-eqz p1, :cond_1
  
+     .line 184
      invoke-virtual {p1}, Ljavax/net/ssl/SSLSocket;->getSession()Ljavax/net/ssl/SSLSession;
  
      move-result-object v0
  
+     .line 185
      .local v0, session:Ljavax/net/ssl/SSLSession;
      if-eqz v0, :cond_0
  
+     .line 186
      invoke-interface {v0}, Ljavax/net/ssl/SSLSession;->invalidate()V
  
+     .line 189
      :cond_0
      invoke-virtual {p1}, Ljavax/net/ssl/SSLSocket;->close()V
  
+     .line 192
      .end local v0           #session:Ljavax/net/ssl/SSLSession;
      :cond_1
      new-instance v1, Ljavax/net/ssl/SSLHandshakeException;
***************
*** 76,93 ****
--- 84,104 ----
      .end annotation
  
      .prologue
+     .line 173
      if-eqz p2, :cond_0
  
      .end local p2
      :goto_0
      invoke-direct {p0, p1, p2}, Landroid/net/http/CertificateChainValidator;->closeSocketThrowException(Ljavax/net/ssl/SSLSocket;Ljava/lang/String;)V
  
+     .line 175
      return-void
  
      .restart local p2
      :cond_0
      move-object p2, p3
  
+     .line 173
      goto :goto_0
  .end method
  
***************
*** 95,100 ****
--- 106,112 ----
      .locals 1
  
      .prologue
+     .line 60
      sget-object v0, Landroid/net/http/CertificateChainValidator;->sInstance:Landroid/net/http/CertificateChainValidator;
  
      return-object v0
***************
*** 112,123 ****
--- 124,137 ----
      .end annotation
  
      .prologue
+     .line 120
      if-eqz p0, :cond_0
  
      array-length v2, p0
  
      if-nez v2, :cond_1
  
+     .line 121
      :cond_0
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
***************
*** 127,137 ****
--- 141,153 ----
  
      throw v2
  
+     .line 124
      :cond_1
      array-length v2, p0
  
      new-array v1, v2, [Ljava/security/cert/X509Certificate;
  
+     .line 126
      .local v1, serverCertificates:[Ljava/security/cert/X509Certificate;
      const/4 v0, 0x0
  
***************
*** 141,146 ****
--- 157,163 ----
  
      if-ge v0, v2, :cond_2
  
+     .line 127
      new-instance v2, Lorg/apache/harmony/security/provider/cert/X509CertImpl;
  
      aget-object v3, p0, v0
***************
*** 149,158 ****
--- 166,177 ----
  
      aput-object v2, v1, v0
  
+     .line 126
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 130
      :cond_2
      invoke-static {v1, p1, p2}, Landroid/net/http/CertificateChainValidator;->verifyServerDomainAndCertificates([Ljava/security/cert/X509Certificate;Ljava/lang/String;Ljava/lang/String;)Landroid/net/http/SslError;
  
***************
*** 173,185 ****
--- 192,207 ----
      .end annotation
  
      .prologue
+     .line 145
      const/4 v2, 0x0
  
      aget-object v0, p0, v2
  
+     .line 146
      .local v0, currCertificate:Ljava/security/cert/X509Certificate;
      if-nez v0, :cond_0
  
+     .line 147
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
      const-string v3, "certificate for this site is null"
***************
*** 188,193 ****
--- 210,216 ----
  
      throw v2
  
+     .line 150
      :cond_0
      invoke-static {v0, p1}, Lcom/android/internal/net/DomainNameValidator;->match(Ljava/security/cert/X509Certificate;Ljava/lang/String;)Z
  
***************
*** 195,209 ****
--- 218,235 ----
  
      if-nez v2, :cond_1
  
+     .line 154
      new-instance v2, Landroid/net/http/SslError;
  
      const/4 v3, 0x2
  
      invoke-direct {v2, v3, v0}, Landroid/net/http/SslError;-><init>(ILjava/security/cert/X509Certificate;)V
  
+     .line 165
      :goto_0
      return-object v2
  
+     .line 158
      :cond_1
      :try_start_0
      invoke-static {}, Lorg/apache/harmony/xnet/provider/jsse/SSLParametersImpl;->getDefaultTrustManager()Ljavax/net/ssl/X509TrustManager;
***************
*** 214,226 ****
--- 240,255 ----
      :try_end_0
      .catch Ljava/security/cert/CertificateException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 159
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 160
      :catch_0
      move-exception v1
  
+     .line 165
      .local v1, e:Ljava/security/cert/CertificateException;
      new-instance v2, Landroid/net/http/SslError;
  
***************
*** 247,256 ****
--- 276,287 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 83
      invoke-virtual {p2}, Ljavax/net/ssl/SSLSocket;->getSession()Ljavax/net/ssl/SSLSession;
  
      move-result-object v1
  
+     .line 84
      .local v1, sslSession:Ljavax/net/ssl/SSLSession;
      invoke-interface {v1}, Ljavax/net/ssl/SSLSession;->isValid()Z
  
***************
*** 258,267 ****
--- 289,300 ----
  
      if-nez v2, :cond_0
  
+     .line 85
      const-string v2, "failed to perform SSL handshake"
  
      invoke-direct {p0, p2, v2}, Landroid/net/http/CertificateChainValidator;->closeSocketThrowException(Ljavax/net/ssl/SSLSocket;Ljava/lang/String;)V
  
+     .line 89
      :cond_0
      invoke-virtual {p2}, Ljavax/net/ssl/SSLSocket;->getSession()Ljavax/net/ssl/SSLSession;
  
***************
*** 271,276 ****
--- 304,310 ----
  
      move-result-object v0
  
+     .line 92
      .local v0, peerCertificates:[Ljava/security/cert/Certificate;
      if-eqz v0, :cond_1
  
***************
*** 278,288 ****
--- 312,324 ----
  
      if-nez v2, :cond_3
  
+     .line 93
      :cond_1
      const-string v2, "failed to retrieve peer certificates"
  
      invoke-direct {p0, p2, v2}, Landroid/net/http/CertificateChainValidator;->closeSocketThrowException(Ljavax/net/ssl/SSLSocket;Ljava/lang/String;)V
  
+     .line 105
      :cond_2
      :goto_0
      check-cast v0, [Ljava/security/cert/X509Certificate;
***************
*** 298,311 ****
--- 334,350 ----
  
      return-object v2
  
+     .line 97
      .restart local v0       #peerCertificates:[Ljava/security/cert/Certificate;
      :cond_3
      if-eqz p1, :cond_2
  
+     .line 98
      aget-object v2, v0, v4
  
      if-eqz v2, :cond_2
  
+     .line 99
      new-instance v3, Landroid/net/http/SslCertificate;
  
      aget-object v2, v0, v4
