*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/CharArrayBuffers.smali	2012-03-23 19:37:00.529784742 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/CharArrayBuffers.smali.original	2012-03-23 19:34:45.481322137 -0500
***************
*** 12,17 ****
--- 12,18 ----
      .locals 0
  
      .prologue
+     .line 27
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 28,42 ****
--- 29,46 ----
  
      const/4 v8, 0x0
  
+     .line 41
      invoke-virtual {p0}, Lorg/apache/http/util/CharArrayBuffer;->length()I
  
      move-result v4
  
+     .line 42
      .local v4, len:I
      invoke-virtual {p0}, Lorg/apache/http/util/CharArrayBuffer;->buffer()[C
  
      move-result-object v2
  
+     .line 43
      .local v2, chars:[C
      :goto_0
      if-ge p1, v4, :cond_0
***************
*** 49,63 ****
--- 53,70 ----
  
      if-eqz v9, :cond_0
  
+     .line 44
      add-int/lit8 p1, p1, 0x1
  
      goto :goto_0
  
+     .line 46
      :cond_0
      invoke-virtual {p2}, Ljava/lang/String;->length()I
  
      move-result v6
  
+     .line 47
      .local v6, size:I
      add-int v9, p1, v6
  
***************
*** 65,70 ****
--- 72,78 ----
  
      move v5, v7
  
+     .line 48
      .local v5, ok:Z
      :goto_1
      const/4 v3, 0x0
***************
*** 75,104 ****
--- 83,119 ----
  
      if-ge v3, v6, :cond_4
  
+     .line 49
      add-int v9, p1, v3
  
      aget-char v0, v2, v9
  
+     .line 50
      .local v0, a:C
      invoke-virtual {p2, v3}, Ljava/lang/String;->charAt(I)C
  
      move-result v1
  
+     .line 51
      .local v1, b:C
      if-eq v0, v1, :cond_1
  
+     .line 52
      invoke-static {v0}, Landroid/net/http/CharArrayBuffers;->toLower(C)C
  
      move-result v0
  
+     .line 53
      invoke-static {v1}, Landroid/net/http/CharArrayBuffers;->toLower(C)C
  
      move-result v1
  
+     .line 54
      if-ne v0, v1, :cond_3
  
      move v5, v7
  
+     .line 48
      :cond_1
      :goto_3
      add-int/lit8 v3, v3, 0x1
***************
*** 112,117 ****
--- 127,133 ----
      :cond_2
      move v5, v8
  
+     .line 47
      goto :goto_1
  
      .restart local v0       #a:C
***************
*** 121,128 ****
--- 137,146 ----
      :cond_3
      move v5, v8
  
+     .line 54
      goto :goto_3
  
+     .line 57
      .end local v0           #a:C
      .end local v1           #b:C
      :cond_4
***************
*** 135,152 ****
--- 153,174 ----
      .parameter "ch"
  
      .prologue
+     .line 66
      const/4 v0, 0x0
  
+     .line 67
      .local v0, beginIndex:I
      invoke-virtual {p0}, Lorg/apache/http/util/CharArrayBuffer;->length()I
  
      move-result v3
  
+     .line 68
      .local v3, endIndex:I
      invoke-virtual {p0}, Lorg/apache/http/util/CharArrayBuffer;->buffer()[C
  
      move-result-object v1
  
+     .line 70
      .local v1, chars:[C
      move v4, v0
  
***************
*** 154,169 ****
--- 176,195 ----
      :goto_0
      if-ge v4, v3, :cond_2
  
+     .line 71
      aget-char v2, v1, v4
  
+     .line 72
      .local v2, current:C
      if-ne v2, p1, :cond_0
  
+     .line 80
      .end local v2           #current:C
      .end local v4           #i:I
      :goto_1
      return v4
  
+     .line 74
      .restart local v2       #current:C
      .restart local v4       #i:I
      :cond_0
***************
*** 175,191 ****
--- 201,221 ----
  
      if-gt v2, v5, :cond_1
  
+     .line 76
      add-int/lit8 v5, v2, 0x20
  
      int-to-char v2, v5
  
+     .line 77
      aput-char v2, v1, v4
  
+     .line 70
      :cond_1
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
  
+     .line 80
      .end local v2           #current:C
      :cond_2
      const/4 v4, -0x1
***************
*** 198,203 ****
--- 228,234 ----
      .parameter "c"
  
      .prologue
+     .line 84
      const/16 v0, 0x41
  
      if-lt p0, v0, :cond_0
***************
*** 206,215 ****
--- 237,248 ----
  
      if-gt p0, v0, :cond_0
  
+     .line 85
      add-int/lit8 v0, p0, 0x20
  
      int-to-char p0, v0
  
+     .line 87
      :cond_0
      return p0
  .end method
