*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/Headers.smali	2012-03-23 19:37:00.485783942 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/Headers.smali.original	2012-03-23 19:34:45.480322119 -0500
***************
*** 191,203 ****
      .locals 3
  
      .prologue
      const/16 v0, 0x13
  
      new-array v0, v0, [Ljava/lang/String;
  
      const/4 v1, 0x0
  
!     const-string v2, "transfer-encoding"
  
      aput-object v2, v0, v1
  
--- 191,204 ----
      .locals 3
  
      .prologue
+     .line 128
      const/16 v0, 0x13
  
      new-array v0, v0, [Ljava/lang/String;
  
      const/4 v1, 0x0
  
!     const-string/jumbo v2, "transfer-encoding"
  
      aput-object v2, v0, v1
  
***************
*** 233,251 ****
  
      const/4 v1, 0x6
  
!     const-string v2, "proxy-connection"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x7
  
!     const-string v2, "www-authenticate"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x8
  
!     const-string v2, "proxy-authenticate"
  
      aput-object v2, v0, v1
  
--- 234,252 ----
  
      const/4 v1, 0x6
  
!     const-string/jumbo v2, "proxy-connection"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x7
  
!     const-string/jumbo v2, "www-authenticate"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x8
  
!     const-string/jumbo v2, "proxy-authenticate"
  
      aput-object v2, v0, v1
  
***************
*** 287,311 ****
  
      const/16 v1, 0xf
  
!     const-string v2, "set-cookie"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x10
  
!     const-string v2, "pragma"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x11
  
!     const-string v2, "refresh"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x12
  
!     const-string v2, "x-permitted-cross-domain-policies"
  
      aput-object v2, v0, v1
  
--- 288,312 ----
  
      const/16 v1, 0xf
  
!     const-string/jumbo v2, "set-cookie"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x10
  
!     const-string/jumbo v2, "pragma"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x11
  
!     const-string/jumbo v2, "refresh"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x12
  
!     const-string/jumbo v2, "x-permitted-cross-domain-policies"
  
      aput-object v2, v0, v1
  
***************
*** 320,327 ****
--- 321,330 ----
      .prologue
      const/4 v2, 0x4
  
+     .line 154
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 125
      new-instance v0, Ljava/util/ArrayList;
  
      const/4 v1, 0x2
***************
*** 330,365 ****
--- 333,375 ----
  
      iput-object v0, p0, Landroid/net/http/Headers;->cookies:Ljava/util/ArrayList;
  
+     .line 127
      const/16 v0, 0x13
  
      new-array v0, v0, [Ljava/lang/String;
  
      iput-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
+     .line 151
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0, v2}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v0, p0, Landroid/net/http/Headers;->mExtraHeaderNames:Ljava/util/ArrayList;
  
+     .line 152
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0, v2}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v0, p0, Landroid/net/http/Headers;->mExtraHeaderValues:Ljava/util/ArrayList;
  
+     .line 155
      const-wide/16 v0, 0x0
  
      iput-wide v0, p0, Landroid/net/http/Headers;->transferEncoding:J
  
+     .line 156
      const-wide/16 v0, -0x1
  
      iput-wide v0, p0, Landroid/net/http/Headers;->contentLength:J
  
+     .line 157
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/net/http/Headers;->connectionType:I
  
+     .line 158
      return-void
  .end method
  
***************
*** 369,374 ****
--- 379,385 ----
      .parameter "pos"
  
      .prologue
+     .line 462
      const-string v0, "Close"
  
      invoke-static {p1, p2, v0}, Landroid/net/http/CharArrayBuffers;->containsIgnoreCaseTrimmed(Lorg/apache/http/util/CharArrayBuffer;ILjava/lang/String;)Z
***************
*** 377,390 ****
--- 388,404 ----
  
      if-eqz v0, :cond_1
  
+     .line 464
      const/4 v0, 0x1
  
      iput v0, p0, Landroid/net/http/Headers;->connectionType:I
  
+     .line 469
      :cond_0
      :goto_0
      return-void
  
+     .line 465
      :cond_1
      const-string v0, "Keep-Alive"
  
***************
*** 394,399 ****
--- 408,414 ----
  
      if-eqz v0, :cond_0
  
+     .line 467
      const/4 v0, 0x2
  
      iput v0, p0, Landroid/net/http/Headers;->connectionType:I
***************
*** 407,412 ****
--- 422,428 ----
      .locals 2
  
      .prologue
+     .line 348
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xa
***************
*** 420,425 ****
--- 436,442 ----
      .locals 2
  
      .prologue
+     .line 356
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xc
***************
*** 433,438 ****
--- 450,456 ----
      .locals 1
  
      .prologue
+     .line 320
      iget v0, p0, Landroid/net/http/Headers;->connectionType:I
  
      return v0
***************
*** 442,447 ****
--- 460,466 ----
      .locals 2
  
      .prologue
+     .line 344
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0x9
***************
*** 455,460 ****
--- 474,480 ----
      .locals 2
  
      .prologue
+     .line 328
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v1, 0x3
***************
*** 468,473 ****
--- 488,494 ----
      .locals 2
  
      .prologue
+     .line 316
      iget-wide v0, p0, Landroid/net/http/Headers;->contentLength:J
  
      return-wide v0
***************
*** 477,482 ****
--- 498,504 ----
      .locals 2
  
      .prologue
+     .line 324
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v1, 0x2
***************
*** 490,495 ****
--- 512,518 ----
      .locals 2
  
      .prologue
+     .line 364
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xe
***************
*** 503,508 ****
--- 526,532 ----
      .locals 2
  
      .prologue
+     .line 352
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xb
***************
*** 517,522 ****
--- 541,547 ----
      .parameter "hcb"
  
      .prologue
+     .line 443
      const/4 v2, 0x0
  
      .local v2, i:I
***************
*** 525,548 ****
--- 550,578 ----
  
      if-ge v2, v3, :cond_1
  
+     .line 444
      iget-object v3, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      aget-object v1, v3, v2
  
+     .line 445
      .local v1, h:Ljava/lang/String;
      if-eqz v1, :cond_0
  
+     .line 446
      sget-object v3, Landroid/net/http/Headers;->sHeaderNames:[Ljava/lang/String;
  
      aget-object v3, v3, v2
  
      invoke-interface {p1, v3, v1}, Landroid/net/http/Headers$HeaderCallback;->header(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 443
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 449
      .end local v1           #h:Ljava/lang/String;
      :cond_1
      iget-object v3, p0, Landroid/net/http/Headers;->mExtraHeaderNames:Ljava/util/ArrayList;
***************
*** 551,562 ****
--- 581,594 ----
  
      move-result v0
  
+     .line 450
      .local v0, extraLen:I
      const/4 v2, 0x0
  
      :goto_1
      if-ge v2, v0, :cond_2
  
+     .line 455
      iget-object v3, p0, Landroid/net/http/Headers;->mExtraHeaderNames:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v2}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 575,584 ****
--- 607,618 ----
  
      invoke-interface {p1, v3, v4}, Landroid/net/http/Headers$HeaderCallback;->header(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 450
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_1
  
+     .line 459
      :cond_2
      return-void
  .end method
***************
*** 587,592 ****
--- 621,627 ----
      .locals 2
  
      .prologue
+     .line 360
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xd
***************
*** 600,605 ****
--- 635,641 ----
      .locals 2
  
      .prologue
+     .line 332
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v1, 0x5
***************
*** 613,618 ****
--- 649,655 ----
      .locals 2
  
      .prologue
+     .line 372
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0x10
***************
*** 626,631 ****
--- 663,669 ----
      .locals 2
  
      .prologue
+     .line 340
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0x8
***************
*** 639,644 ****
--- 677,683 ----
      .locals 2
  
      .prologue
+     .line 376
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0x11
***************
*** 661,666 ****
--- 700,706 ----
      .end annotation
  
      .prologue
+     .line 368
      iget-object v0, p0, Landroid/net/http/Headers;->cookies:Ljava/util/ArrayList;
  
      return-object v0
***************
*** 670,675 ****
--- 710,716 ----
      .locals 2
  
      .prologue
+     .line 312
      iget-wide v0, p0, Landroid/net/http/Headers;->transferEncoding:J
  
      return-wide v0
***************
*** 679,684 ****
--- 720,726 ----
      .locals 2
  
      .prologue
+     .line 336
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v1, 0x7
***************
*** 692,697 ****
--- 734,740 ----
      .locals 2
  
      .prologue
+     .line 380
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0x12
***************
*** 712,737 ****
--- 755,785 ----
  
      const/16 v8, 0xc
  
+     .line 161
      const/16 v5, 0x3a
  
      invoke-static {p1, v5}, Landroid/net/http/CharArrayBuffers;->setLowercaseIndexOf(Lorg/apache/http/util/CharArrayBuffer;I)I
  
      move-result v3
  
+     .line 162
      .local v3, pos:I
      const/4 v5, -0x1
  
      if-ne v3, v5, :cond_1
  
+     .line 309
      :cond_0
      :goto_0
      return-void
  
+     .line 165
      :cond_1
      invoke-virtual {p1, v6, v3}, Lorg/apache/http/util/CharArrayBuffer;->substringTrimmed(II)Ljava/lang/String;
  
      move-result-object v2
  
+     .line 166
      .local v2, name:Ljava/lang/String;
      invoke-virtual {v2}, Ljava/lang/String;->length()I
  
***************
*** 739,746 ****
--- 787,796 ----
  
      if-eqz v5, :cond_0
  
+     .line 169
      add-int/lit8 v3, v3, 0x1
  
+     .line 171
      invoke-virtual {p1}, Lorg/apache/http/util/CharArrayBuffer;->length()I
  
      move-result v5
***************
*** 749,754 ****
--- 799,805 ----
  
      move-result-object v4
  
+     .line 176
      .local v4, val:Ljava/lang/String;
      invoke-virtual {v2}, Ljava/lang/String;->hashCode()I
  
***************
*** 756,773 ****
  
      sparse-switch v5, :sswitch_data_0
  
      iget-object v5, p0, Landroid/net/http/Headers;->mExtraHeaderNames:Ljava/util/ArrayList;
  
      invoke-virtual {v5, v2}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
      iget-object v5, p0, Landroid/net/http/Headers;->mExtraHeaderValues:Ljava/util/ArrayList;
  
      invoke-virtual {v5, v4}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
      goto :goto_0
  
      :sswitch_0
!     const-string v5, "transfer-encoding"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 807,827 ----
  
      sparse-switch v5, :sswitch_data_0
  
+     .line 306
      iget-object v5, p0, Landroid/net/http/Headers;->mExtraHeaderNames:Ljava/util/ArrayList;
  
      invoke-virtual {v5, v2}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 307
      iget-object v5, p0, Landroid/net/http/Headers;->mExtraHeaderValues:Ljava/util/ArrayList;
  
      invoke-virtual {v5, v4}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
      goto :goto_0
  
+     .line 178
      :sswitch_0
!     const-string/jumbo v5, "transfer-encoding"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 775,784 ****
--- 829,840 ----
  
      if-eqz v5, :cond_0
  
+     .line 179
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      aput-object v4, v5, v6
  
+     .line 180
      sget-object v5, Lorg/apache/http/message/BasicHeaderValueParser;->DEFAULT:Lorg/apache/http/message/BasicHeaderValueParser;
  
      new-instance v6, Lorg/apache/http/message/ParserCursor;
***************
*** 793,801 ****
--- 849,859 ----
  
      move-result-object v0
  
+     .line 185
      .local v0, encodings:[Lorg/apache/http/HeaderElement;
      array-length v1, v0
  
+     .line 186
      .local v1, len:I
      const-string v5, "identity"
  
***************
*** 805,814 ****
--- 863,874 ----
  
      if-eqz v5, :cond_2
  
+     .line 187
      iput-wide v9, p0, Landroid/net/http/Headers;->transferEncoding:J
  
      goto :goto_0
  
+     .line 188
      :cond_2
      if-lez v1, :cond_3
  
***************
*** 828,844 ****
--- 888,907 ----
  
      if-eqz v5, :cond_3
  
+     .line 191
      const-wide/16 v5, -0x2
  
      iput-wide v5, p0, Landroid/net/http/Headers;->transferEncoding:J
  
      goto :goto_0
  
+     .line 193
      :cond_3
      iput-wide v9, p0, Landroid/net/http/Headers;->transferEncoding:J
  
      goto :goto_0
  
+     .line 198
      .end local v0           #encodings:[Lorg/apache/http/HeaderElement;
      .end local v1           #len:I
      :sswitch_1
***************
*** 850,861 ****
--- 913,926 ----
  
      if-eqz v5, :cond_0
  
+     .line 199
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v6, 0x1
  
      aput-object v4, v5, v6
  
+     .line 201
      :try_start_0
      invoke-static {v4}, Ljava/lang/Long;->parseLong(Ljava/lang/String;)J
  
***************
*** 867,877 ****
--- 932,944 ----
  
      goto :goto_0
  
+     .line 202
      :catch_0
      move-exception v5
  
      goto :goto_0
  
+     .line 211
      :sswitch_2
      const-string v5, "content-type"
  
***************
*** 881,886 ****
--- 948,954 ----
  
      if-eqz v5, :cond_0
  
+     .line 212
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v6, 0x2
***************
*** 889,894 ****
--- 957,963 ----
  
      goto/16 :goto_0
  
+     .line 216
      :sswitch_3
      const-string v5, "content-encoding"
  
***************
*** 898,903 ****
--- 967,973 ----
  
      if-eqz v5, :cond_0
  
+     .line 217
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v6, 0x3
***************
*** 906,911 ****
--- 976,982 ----
  
      goto/16 :goto_0
  
+     .line 221
      :sswitch_4
      const-string v5, "connection"
  
***************
*** 915,930 ****
--- 986,1004 ----
  
      if-eqz v5, :cond_0
  
+     .line 222
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v6, 0x4
  
      aput-object v4, v5, v6
  
+     .line 223
      invoke-direct {p0, p1, v3}, Landroid/net/http/Headers;->setConnectionType(Lorg/apache/http/util/CharArrayBuffer;I)V
  
      goto/16 :goto_0
  
+     .line 227
      :sswitch_5
      const-string v5, "location"
  
***************
*** 934,939 ****
--- 1008,1014 ----
  
      if-eqz v5, :cond_0
  
+     .line 228
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v6, 0x5
***************
*** 942,949 ****
  
      goto/16 :goto_0
  
      :sswitch_6
!     const-string v5, "proxy-connection"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 1017,1025 ----
  
      goto/16 :goto_0
  
+     .line 232
      :sswitch_6
!     const-string/jumbo v5, "proxy-connection"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 951,968 ****
  
      if-eqz v5, :cond_0
  
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v6, 0x6
  
      aput-object v4, v5, v6
  
      invoke-direct {p0, p1, v3}, Landroid/net/http/Headers;->setConnectionType(Lorg/apache/http/util/CharArrayBuffer;I)V
  
      goto/16 :goto_0
  
      :sswitch_7
!     const-string v5, "www-authenticate"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 1027,1047 ----
  
      if-eqz v5, :cond_0
  
+     .line 233
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v6, 0x6
  
      aput-object v4, v5, v6
  
+     .line 234
      invoke-direct {p0, p1, v3}, Landroid/net/http/Headers;->setConnectionType(Lorg/apache/http/util/CharArrayBuffer;I)V
  
      goto/16 :goto_0
  
+     .line 238
      :sswitch_7
!     const-string/jumbo v5, "www-authenticate"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 970,975 ****
--- 1049,1055 ----
  
      if-eqz v5, :cond_0
  
+     .line 239
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v6, 0x7
***************
*** 978,985 ****
  
      goto/16 :goto_0
  
      :sswitch_8
!     const-string v5, "proxy-authenticate"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 1058,1066 ----
  
      goto/16 :goto_0
  
+     .line 243
      :sswitch_8
!     const-string/jumbo v5, "proxy-authenticate"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 987,992 ****
--- 1068,1074 ----
  
      if-eqz v5, :cond_0
  
+     .line 244
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0x8
***************
*** 995,1000 ****
--- 1077,1083 ----
  
      goto/16 :goto_0
  
+     .line 248
      :sswitch_9
      const-string v5, "content-disposition"
  
***************
*** 1004,1009 ****
--- 1087,1093 ----
  
      if-eqz v5, :cond_0
  
+     .line 249
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0x9
***************
*** 1012,1017 ****
--- 1096,1102 ----
  
      goto/16 :goto_0
  
+     .line 253
      :sswitch_a
      const-string v5, "accept-ranges"
  
***************
*** 1021,1026 ****
--- 1106,1112 ----
  
      if-eqz v5, :cond_0
  
+     .line 254
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0xa
***************
*** 1029,1034 ****
--- 1115,1121 ----
  
      goto/16 :goto_0
  
+     .line 258
      :sswitch_b
      const-string v5, "expires"
  
***************
*** 1038,1043 ****
--- 1125,1131 ----
  
      if-eqz v5, :cond_0
  
+     .line 259
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0xb
***************
*** 1046,1051 ****
--- 1134,1140 ----
  
      goto/16 :goto_0
  
+     .line 263
      :sswitch_c
      const-string v5, "cache-control"
  
***************
*** 1055,1060 ****
--- 1144,1150 ----
  
      if-eqz v5, :cond_0
  
+     .line 266
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      aget-object v5, v5, v8
***************
*** 1071,1076 ****
--- 1161,1167 ----
  
      if-lez v5, :cond_4
  
+     .line 268
      new-instance v5, Ljava/lang/StringBuilder;
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1101,1106 ****
--- 1192,1198 ----
  
      goto/16 :goto_0
  
+     .line 270
      :cond_4
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
***************
*** 1108,1113 ****
--- 1200,1206 ----
  
      goto/16 :goto_0
  
+     .line 275
      :sswitch_d
      const-string v5, "last-modified"
  
***************
*** 1117,1122 ****
--- 1210,1216 ----
  
      if-eqz v5, :cond_0
  
+     .line 276
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0xd
***************
*** 1125,1130 ****
--- 1219,1225 ----
  
      goto/16 :goto_0
  
+     .line 280
      :sswitch_e
      const-string v5, "etag"
  
***************
*** 1134,1139 ****
--- 1229,1235 ----
  
      if-eqz v5, :cond_0
  
+     .line 281
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0xe
***************
*** 1142,1149 ****
  
      goto/16 :goto_0
  
      :sswitch_f
!     const-string v5, "set-cookie"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 1238,1246 ----
  
      goto/16 :goto_0
  
+     .line 285
      :sswitch_f
!     const-string/jumbo v5, "set-cookie"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 1151,1170 ****
  
      if-eqz v5, :cond_0
  
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0xf
  
      aput-object v4, v5, v6
  
      iget-object v5, p0, Landroid/net/http/Headers;->cookies:Ljava/util/ArrayList;
  
      invoke-virtual {v5, v4}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
      goto/16 :goto_0
  
      :sswitch_10
!     const-string v5, "pragma"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 1248,1270 ----
  
      if-eqz v5, :cond_0
  
+     .line 286
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0xf
  
      aput-object v4, v5, v6
  
+     .line 287
      iget-object v5, p0, Landroid/net/http/Headers;->cookies:Ljava/util/ArrayList;
  
      invoke-virtual {v5, v4}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
      goto/16 :goto_0
  
+     .line 291
      :sswitch_10
!     const-string/jumbo v5, "pragma"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 1172,1177 ****
--- 1272,1278 ----
  
      if-eqz v5, :cond_0
  
+     .line 292
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0x10
***************
*** 1180,1187 ****
  
      goto/16 :goto_0
  
      :sswitch_11
!     const-string v5, "refresh"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 1281,1289 ----
  
      goto/16 :goto_0
  
+     .line 296
      :sswitch_11
!     const-string/jumbo v5, "refresh"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 1189,1194 ****
--- 1291,1297 ----
  
      if-eqz v5, :cond_0
  
+     .line 297
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0x11
***************
*** 1197,1204 ****
  
      goto/16 :goto_0
  
      :sswitch_12
!     const-string v5, "x-permitted-cross-domain-policies"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 1300,1308 ----
  
      goto/16 :goto_0
  
+     .line 301
      :sswitch_12
!     const-string/jumbo v5, "x-permitted-cross-domain-policies"
  
      invoke-virtual {v2, v5}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 1206,1211 ****
--- 1310,1316 ----
  
      if-eqz v5, :cond_0
  
+     .line 302
      iget-object v5, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v6, 0x12
***************
*** 1214,1219 ****
--- 1319,1325 ----
  
      goto/16 :goto_0
  
+     .line 176
      :sswitch_data_0
      .sparse-switch
          -0x5034229e -> :sswitch_12
***************
*** 1243,1254 ****
--- 1349,1362 ----
      .parameter "value"
  
      .prologue
+     .line 412
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xa
  
      aput-object p1, v0, v1
  
+     .line 413
      return-void
  .end method
  
***************
*** 1257,1268 ****
--- 1365,1378 ----
      .parameter "value"
  
      .prologue
+     .line 420
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xc
  
      aput-object p1, v0, v1
  
+     .line 421
      return-void
  .end method
  
***************
*** 1271,1282 ****
--- 1381,1394 ----
      .parameter "value"
  
      .prologue
+     .line 408
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0x9
  
      aput-object p1, v0, v1
  
+     .line 409
      return-void
  .end method
  
***************
*** 1285,1296 ****
--- 1397,1410 ----
      .parameter "value"
  
      .prologue
+     .line 392
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v1, 0x3
  
      aput-object p1, v0, v1
  
+     .line 393
      return-void
  .end method
  
***************
*** 1299,1306 ****
--- 1413,1422 ----
      .parameter "value"
  
      .prologue
+     .line 384
      iput-wide p1, p0, Landroid/net/http/Headers;->contentLength:J
  
+     .line 385
      return-void
  .end method
  
***************
*** 1309,1320 ****
--- 1425,1438 ----
      .parameter "value"
  
      .prologue
+     .line 388
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v1, 0x2
  
      aput-object p1, v0, v1
  
+     .line 389
      return-void
  .end method
  
***************
*** 1323,1334 ****
--- 1441,1454 ----
      .parameter "value"
  
      .prologue
+     .line 428
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xe
  
      aput-object p1, v0, v1
  
+     .line 429
      return-void
  .end method
  
***************
*** 1337,1348 ****
--- 1457,1470 ----
      .parameter "value"
  
      .prologue
+     .line 416
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xb
  
      aput-object p1, v0, v1
  
+     .line 417
      return-void
  .end method
  
***************
*** 1351,1362 ****
--- 1473,1486 ----
      .parameter "value"
  
      .prologue
+     .line 424
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0xd
  
      aput-object p1, v0, v1
  
+     .line 425
      return-void
  .end method
  
***************
*** 1365,1376 ****
--- 1489,1502 ----
      .parameter "value"
  
      .prologue
+     .line 396
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v1, 0x5
  
      aput-object p1, v0, v1
  
+     .line 397
      return-void
  .end method
  
***************
*** 1379,1390 ****
--- 1505,1518 ----
      .parameter "value"
  
      .prologue
+     .line 404
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0x8
  
      aput-object p1, v0, v1
  
+     .line 405
      return-void
  .end method
  
***************
*** 1393,1404 ****
--- 1521,1534 ----
      .parameter "value"
  
      .prologue
+     .line 400
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/4 v1, 0x7
  
      aput-object p1, v0, v1
  
+     .line 401
      return-void
  .end method
  
***************
*** 1407,1417 ****
--- 1537,1549 ----
      .parameter "value"
  
      .prologue
+     .line 432
      iget-object v0, p0, Landroid/net/http/Headers;->mHeaders:[Ljava/lang/String;
  
      const/16 v1, 0x12
  
      aput-object p1, v0, v1
  
+     .line 433
      return-void
  .end method
