*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/HttpAuthHeader.smali	2012-03-23 19:37:01.237797654 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/HttpAuthHeader.smali.original	2012-03-23 19:34:45.491322321 -0500
***************
*** 55,66 ****
--- 55,70 ----
      .parameter "header"
  
      .prologue
+     .line 116
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 117
      if-eqz p1, :cond_0
  
+     .line 118
      invoke-direct {p0, p1}, Landroid/net/http/HttpAuthHeader;->parseHeader(Ljava/lang/String;)V
  
+     .line 120
      :cond_0
      return-void
  .end method
***************
*** 70,90 ****
--- 74,100 ----
      .parameter "header"
  
      .prologue
+     .line 265
      if-eqz p1, :cond_0
  
+     .line 266
      invoke-direct {p0, p1}, Landroid/net/http/HttpAuthHeader;->parseScheme(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 267
      .local v0, parameters:Ljava/lang/String;
      if-eqz v0, :cond_0
  
+     .line 269
      iget v1, p0, Landroid/net/http/HttpAuthHeader;->mScheme:I
  
      if-eqz v1, :cond_0
  
+     .line 270
      invoke-direct {p0, v0}, Landroid/net/http/HttpAuthHeader;->parseParameters(Ljava/lang/String;)V
  
+     .line 274
      .end local v0           #parameters:Ljava/lang/String;
      :cond_0
      return-void
***************
*** 95,111 ****
--- 105,125 ----
      .parameter "parameter"
  
      .prologue
+     .line 336
      if-eqz p1, :cond_0
  
+     .line 338
      const/16 v3, 0x3d
  
      invoke-virtual {p1, v3}, Ljava/lang/String;->indexOf(I)I
  
      move-result v0
  
+     .line 339
      .local v0, i:I
      if-ltz v0, :cond_0
  
+     .line 340
      const/4 v3, 0x0
  
      invoke-virtual {p1, v3, v0}, Ljava/lang/String;->substring(II)Ljava/lang/String;
***************
*** 116,121 ****
--- 130,136 ----
  
      move-result-object v1
  
+     .line 341
      .local v1, token:Ljava/lang/String;
      add-int/lit8 v3, v0, 0x1
  
***************
*** 131,138 ****
  
      move-result-object v2
  
      .local v2, value:Ljava/lang/String;
!     const-string v3, "realm"
  
      invoke-virtual {v1, v3}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 146,154 ----
  
      move-result-object v2
  
+     .line 350
      .local v2, value:Ljava/lang/String;
!     const-string/jumbo v3, "realm"
  
      invoke-virtual {v1, v3}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 140,147 ****
--- 156,165 ----
  
      if-eqz v3, :cond_1
  
+     .line 351
      iput-object v2, p0, Landroid/net/http/HttpAuthHeader;->mRealm:Ljava/lang/String;
  
+     .line 359
      .end local v0           #i:I
      .end local v1           #token:Ljava/lang/String;
      .end local v2           #value:Ljava/lang/String;
***************
*** 149,154 ****
--- 167,173 ----
      :goto_0
      return-void
  
+     .line 353
      .restart local v0       #i:I
      .restart local v1       #token:Ljava/lang/String;
      .restart local v2       #value:Ljava/lang/String;
***************
*** 159,164 ****
--- 178,184 ----
  
      if-ne v3, v4, :cond_0
  
+     .line 354
      invoke-direct {p0, v1, v2}, Landroid/net/http/HttpAuthHeader;->parseParameter(Ljava/lang/String;Ljava/lang/String;)V
  
      goto :goto_0
***************
*** 170,180 ****
      .parameter "value"
  
      .prologue
      if-eqz p1, :cond_0
  
      if-eqz p2, :cond_0
  
!     const-string v0, "nonce"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 190,202 ----
      .parameter "value"
  
      .prologue
+     .line 366
      if-eqz p1, :cond_0
  
      if-eqz p2, :cond_0
  
!     .line 367
!     const-string/jumbo v0, "nonce"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 182,195 ****
  
      if-eqz v0, :cond_1
  
      iput-object p2, p0, Landroid/net/http/HttpAuthHeader;->mNonce:Ljava/lang/String;
  
      :cond_0
      :goto_0
      return-void
  
      :cond_1
!     const-string v0, "stale"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 204,220 ----
  
      if-eqz v0, :cond_1
  
+     .line 368
      iput-object p2, p0, Landroid/net/http/HttpAuthHeader;->mNonce:Ljava/lang/String;
  
+     .line 392
      :cond_0
      :goto_0
      return-void
  
+     .line 372
      :cond_1
!     const-string/jumbo v0, "stale"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 197,208 ****
  
      if-eqz v0, :cond_2
  
      invoke-direct {p0, p2}, Landroid/net/http/HttpAuthHeader;->parseStale(Ljava/lang/String;)V
  
      goto :goto_0
  
      :cond_2
!     const-string v0, "opaque"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 222,235 ----
  
      if-eqz v0, :cond_2
  
+     .line 373
      invoke-direct {p0, p2}, Landroid/net/http/HttpAuthHeader;->parseStale(Ljava/lang/String;)V
  
      goto :goto_0
  
+     .line 377
      :cond_2
!     const-string/jumbo v0, "opaque"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 210,221 ****
  
      if-eqz v0, :cond_3
  
      iput-object p2, p0, Landroid/net/http/HttpAuthHeader;->mOpaque:Ljava/lang/String;
  
      goto :goto_0
  
      :cond_3
!     const-string v0, "qop"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 237,250 ----
  
      if-eqz v0, :cond_3
  
+     .line 378
      iput-object p2, p0, Landroid/net/http/HttpAuthHeader;->mOpaque:Ljava/lang/String;
  
      goto :goto_0
  
+     .line 382
      :cond_3
!     const-string/jumbo v0, "qop"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 223,228 ****
--- 252,258 ----
  
      if-eqz v0, :cond_4
  
+     .line 383
      invoke-virtual {p2}, Ljava/lang/String;->toLowerCase()Ljava/lang/String;
  
      move-result-object v0
***************
*** 231,236 ****
--- 261,267 ----
  
      goto :goto_0
  
+     .line 387
      :cond_4
      const-string v0, "algorithm"
  
***************
*** 240,245 ****
--- 271,277 ----
  
      if-eqz v0, :cond_0
  
+     .line 388
      invoke-virtual {p2}, Ljava/lang/String;->toLowerCase()Ljava/lang/String;
  
      move-result-object v0
***************
*** 254,261 ****
--- 286,295 ----
      .parameter "parameters"
  
      .prologue
+     .line 316
      if-eqz p1, :cond_1
  
+     .line 319
      :cond_0
      const/16 v1, 0x2c
  
***************
*** 263,280 ****
--- 297,319 ----
  
      move-result v0
  
+     .line 320
      .local v0, i:I
      if-gez v0, :cond_2
  
+     .line 322
      invoke-direct {p0, p1}, Landroid/net/http/HttpAuthHeader;->parseParameter(Ljava/lang/String;)V
  
+     .line 327
      :goto_0
      if-gez v0, :cond_0
  
+     .line 329
      .end local v0           #i:I
      :cond_1
      return-void
  
+     .line 324
      .restart local v0       #i:I
      :cond_2
      const/4 v1, 0x0
***************
*** 285,290 ****
--- 324,330 ----
  
      invoke-direct {p0, v1}, Landroid/net/http/HttpAuthHeader;->parseParameter(Ljava/lang/String;)V
  
+     .line 325
      add-int/lit8 v1, v0, 0x1
  
      invoke-virtual {p1, v1}, Ljava/lang/String;->substring(I)Ljava/lang/String;
***************
*** 299,315 ****
--- 339,359 ----
      .parameter "header"
  
      .prologue
+     .line 284
      if-eqz p1, :cond_2
  
+     .line 285
      const/16 v2, 0x20
  
      invoke-virtual {p1, v2}, Ljava/lang/String;->indexOf(I)I
  
      move-result v0
  
+     .line 286
      .local v0, i:I
      if-ltz v0, :cond_2
  
+     .line 287
      const/4 v2, 0x0
  
      invoke-virtual {p1, v2, v0}, Ljava/lang/String;->substring(II)Ljava/lang/String;
***************
*** 320,325 ****
--- 364,370 ----
  
      move-result-object v1
  
+     .line 288
      .local v1, scheme:Ljava/lang/String;
      const-string v2, "Digest"
  
***************
*** 329,342 ****
  
      if-eqz v2, :cond_1
  
      const/4 v2, 0x2
  
      iput v2, p0, Landroid/net/http/HttpAuthHeader;->mScheme:I
  
!     const-string v2, "md5"
  
      iput-object v2, p0, Landroid/net/http/HttpAuthHeader;->mAlgorithm:Ljava/lang/String;
  
      :cond_0
      :goto_0
      add-int/lit8 v2, v0, 0x1
--- 374,390 ----
  
      if-eqz v2, :cond_1
  
+     .line 289
      const/4 v2, 0x2
  
      iput v2, p0, Landroid/net/http/HttpAuthHeader;->mScheme:I
  
!     .line 292
!     const-string/jumbo v2, "md5"
  
      iput-object v2, p0, Landroid/net/http/HttpAuthHeader;->mAlgorithm:Ljava/lang/String;
  
+     .line 299
      :cond_0
      :goto_0
      add-int/lit8 v2, v0, 0x1
***************
*** 345,355 ****
--- 393,405 ----
  
      move-result-object v2
  
+     .line 303
      .end local v0           #i:I
      .end local v1           #scheme:Ljava/lang/String;
      :goto_1
      return-object v2
  
+     .line 294
      .restart local v0       #i:I
      .restart local v1       #scheme:Ljava/lang/String;
      :cond_1
***************
*** 361,372 ****
--- 411,424 ----
  
      if-eqz v2, :cond_0
  
+     .line 295
      const/4 v2, 0x1
  
      iput v2, p0, Landroid/net/http/HttpAuthHeader;->mScheme:I
  
      goto :goto_0
  
+     .line 303
      .end local v0           #i:I
      .end local v1           #scheme:Ljava/lang/String;
      :cond_2
***************
*** 380,388 ****
      .parameter "value"
  
      .prologue
      if-eqz p1, :cond_0
  
!     const-string v0, "true"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
--- 432,442 ----
      .parameter "value"
  
      .prologue
+     .line 399
      if-eqz p1, :cond_0
  
!     .line 400
!     const-string/jumbo v0, "true"
  
      invoke-virtual {p1, v0}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
  
***************
*** 390,399 ****
--- 444,455 ----
  
      if-eqz v0, :cond_0
  
+     .line 401
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/net/http/HttpAuthHeader;->mStale:Z
  
+     .line 404
      :cond_0
      return-void
  .end method
***************
*** 405,416 ****
--- 461,475 ----
      .prologue
      const/16 v2, 0x22
  
+     .line 412
      if-eqz p0, :cond_0
  
+     .line 413
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v0
  
+     .line 414
      .local v0, len:I
      const/4 v1, 0x2
  
***************
*** 432,437 ****
--- 491,497 ----
  
      if-ne v1, v2, :cond_0
  
+     .line 416
      const/4 v1, 0x1
  
      add-int/lit8 v2, v0, -0x1
***************
*** 440,445 ****
--- 500,506 ----
  
      move-result-object p0
  
+     .line 420
      .end local v0           #len:I
      .end local p0
      :cond_0
***************
*** 452,457 ****
--- 513,519 ----
      .locals 1
  
      .prologue
+     .line 230
      iget-object v0, p0, Landroid/net/http/HttpAuthHeader;->mAlgorithm:Ljava/lang/String;
  
      return-object v0
***************
*** 461,466 ****
--- 523,529 ----
      .locals 1
  
      .prologue
+     .line 207
      iget-object v0, p0, Landroid/net/http/HttpAuthHeader;->mNonce:Ljava/lang/String;
  
      return-object v0
***************
*** 470,475 ****
--- 533,539 ----
      .locals 1
  
      .prologue
+     .line 214
      iget-object v0, p0, Landroid/net/http/HttpAuthHeader;->mOpaque:Ljava/lang/String;
  
      return-object v0
***************
*** 479,484 ****
--- 543,549 ----
      .locals 1
  
      .prologue
+     .line 154
      iget-object v0, p0, Landroid/net/http/HttpAuthHeader;->mPassword:Ljava/lang/String;
  
      return-object v0
***************
*** 488,493 ****
--- 553,559 ----
      .locals 1
  
      .prologue
+     .line 222
      iget-object v0, p0, Landroid/net/http/HttpAuthHeader;->mQop:Ljava/lang/String;
  
      return-object v0
***************
*** 497,502 ****
--- 563,569 ----
      .locals 1
  
      .prologue
+     .line 200
      iget-object v0, p0, Landroid/net/http/HttpAuthHeader;->mRealm:Ljava/lang/String;
  
      return-object v0
***************
*** 506,511 ****
--- 573,579 ----
      .locals 1
  
      .prologue
+     .line 185
      iget v0, p0, Landroid/net/http/HttpAuthHeader;->mScheme:I
  
      return v0
***************
*** 515,520 ****
--- 583,589 ----
      .locals 1
  
      .prologue
+     .line 193
      iget-boolean v0, p0, Landroid/net/http/HttpAuthHeader;->mStale:Z
  
      return v0
***************
*** 524,529 ****
--- 593,599 ----
      .locals 1
  
      .prologue
+     .line 140
      iget-object v0, p0, Landroid/net/http/HttpAuthHeader;->mUsername:Ljava/lang/String;
  
      return-object v0
***************
*** 535,540 ****
--- 605,611 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 168
      iget v1, p0, Landroid/net/http/HttpAuthHeader;->mScheme:I
  
      if-ne v1, v0, :cond_0
***************
*** 552,557 ****
--- 623,629 ----
      .locals 2
  
      .prologue
+     .line 175
      iget v0, p0, Landroid/net/http/HttpAuthHeader;->mScheme:I
  
      const/4 v1, 0x2
***************
*** 573,578 ****
--- 645,651 ----
      .locals 1
  
      .prologue
+     .line 126
      iget-boolean v0, p0, Landroid/net/http/HttpAuthHeader;->mIsProxy:Z
  
      return v0
***************
*** 586,603 ****
--- 659,680 ----
  
      const/4 v0, 0x1
  
+     .line 241
      iget-object v2, p0, Landroid/net/http/HttpAuthHeader;->mRealm:Ljava/lang/String;
  
      if-eqz v2, :cond_3
  
+     .line 242
      iget v2, p0, Landroid/net/http/HttpAuthHeader;->mScheme:I
  
      if-ne v2, v0, :cond_1
  
+     .line 253
      :cond_0
      :goto_0
      return v0
  
+     .line 245
      :cond_1
      iget v2, p0, Landroid/net/http/HttpAuthHeader;->mScheme:I
  
***************
*** 605,613 ****
  
      if-ne v2, v3, :cond_3
  
      iget-object v2, p0, Landroid/net/http/HttpAuthHeader;->mAlgorithm:Ljava/lang/String;
  
!     const-string v3, "md5"
  
      invoke-virtual {v2, v3}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 682,691 ----
  
      if-ne v2, v3, :cond_3
  
+     .line 246
      iget-object v2, p0, Landroid/net/http/HttpAuthHeader;->mAlgorithm:Ljava/lang/String;
  
!     const-string/jumbo v3, "md5"
  
      invoke-virtual {v2, v3}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 637,642 ****
--- 715,721 ----
      :cond_3
      move v0, v1
  
+     .line 253
      goto :goto_0
  .end method
  
***************
*** 645,652 ****
--- 724,733 ----
      .parameter "password"
  
      .prologue
+     .line 161
      iput-object p1, p0, Landroid/net/http/HttpAuthHeader;->mPassword:Ljava/lang/String;
  
+     .line 162
      return-void
  .end method
  
***************
*** 654,663 ****
--- 735,746 ----
      .locals 1
  
      .prologue
+     .line 133
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/net/http/HttpAuthHeader;->mIsProxy:Z
  
+     .line 134
      return-void
  .end method
  
***************
*** 666,672 ****
--- 749,757 ----
      .parameter "username"
  
      .prologue
+     .line 147
      iput-object p1, p0, Landroid/net/http/HttpAuthHeader;->mUsername:Ljava/lang/String;
  
+     .line 148
      return-void
  .end method
