*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/IdleCache.smali	2012-03-23 19:37:00.352781516 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/http/IdleCache.smali.original	2012-03-23 19:34:45.478322083 -0500
***************
*** 44,71 ****
--- 44,79 ----
  
      const/4 v2, 0x0
  
+     .line 56
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 46
      new-array v1, v3, [Landroid/net/http/IdleCache$Entry;
  
      iput-object v1, p0, Landroid/net/http/IdleCache;->mEntries:[Landroid/net/http/IdleCache$Entry;
  
+     .line 48
      iput v2, p0, Landroid/net/http/IdleCache;->mCount:I
  
+     .line 50
      const/4 v1, 0x0
  
      iput-object v1, p0, Landroid/net/http/IdleCache;->mThread:Landroid/net/http/IdleCache$IdleReaper;
  
+     .line 53
      iput v2, p0, Landroid/net/http/IdleCache;->mCached:I
  
+     .line 54
      iput v2, p0, Landroid/net/http/IdleCache;->mReused:I
  
+     .line 57
      const/4 v0, 0x0
  
      .local v0, i:I
      :goto_0
      if-ge v0, v3, :cond_0
  
+     .line 58
      iget-object v1, p0, Landroid/net/http/IdleCache;->mEntries:[Landroid/net/http/IdleCache$Entry;
  
      new-instance v2, Landroid/net/http/IdleCache$Entry;
***************
*** 74,83 ****
--- 82,93 ----
  
      aput-object v2, v1, v0
  
+     .line 57
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 60
      :cond_0
      return-void
  .end method
***************
*** 87,92 ****
--- 97,103 ----
      .parameter "x0"
  
      .prologue
+     .line 30
      iget v0, p0, Landroid/net/http/IdleCache;->mCount:I
  
      return v0
***************
*** 97,102 ****
--- 108,114 ----
      .parameter "x0"
  
      .prologue
+     .line 30
      invoke-direct {p0}, Landroid/net/http/IdleCache;->clearIdle()V
  
      return-void
***************
*** 108,113 ****
--- 120,126 ----
      .parameter "x1"
  
      .prologue
+     .line 30
      iput-object p1, p0, Landroid/net/http/IdleCache;->mThread:Landroid/net/http/IdleCache$IdleReaper;
  
      return-object p1
***************
*** 117,122 ****
--- 130,136 ----
      .locals 6
  
      .prologue
+     .line 130
      monitor-enter p0
  
      :try_start_0
***************
*** 124,133 ****
--- 138,149 ----
  
      if-lez v4, :cond_1
  
+     .line 131
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v2
  
+     .line 132
      .local v2, time:J
      const/4 v1, 0x0
  
***************
*** 137,146 ****
--- 153,164 ----
  
      if-ge v1, v4, :cond_1
  
+     .line 133
      iget-object v4, p0, Landroid/net/http/IdleCache;->mEntries:[Landroid/net/http/IdleCache$Entry;
  
      aget-object v0, v4, v1
  
+     .line 134
      .local v0, entry:Landroid/net/http/IdleCache$Entry;
      iget-object v4, v0, Landroid/net/http/IdleCache$Entry;->mHost:Lorg/apache/http/HttpHost;
  
***************
*** 152,169 ****
--- 170,191 ----
  
      if-lez v4, :cond_0
  
+     .line 135
      const/4 v4, 0x0
  
      iput-object v4, v0, Landroid/net/http/IdleCache$Entry;->mHost:Lorg/apache/http/HttpHost;
  
+     .line 136
      iget-object v4, v0, Landroid/net/http/IdleCache$Entry;->mConnection:Landroid/net/http/Connection;
  
      invoke-virtual {v4}, Landroid/net/http/Connection;->closeConnection()V
  
+     .line 137
      const/4 v4, 0x0
  
      iput-object v4, v0, Landroid/net/http/IdleCache$Entry;->mConnection:Landroid/net/http/Connection;
  
+     .line 138
      iget v4, p0, Landroid/net/http/IdleCache;->mCount:I
  
      add-int/lit8 v4, v4, -0x1
***************
*** 172,182 ****
--- 194,206 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 132
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 142
      .end local v0           #entry:Landroid/net/http/IdleCache$Entry;
      .end local v1           #i:I
      .end local v2           #time:J
***************
*** 185,190 ****
--- 209,215 ----
  
      return-void
  
+     .line 130
      :catchall_0
      move-exception v4
  
***************
*** 203,222 ****
--- 228,251 ----
      .prologue
      const/16 v6, 0x8
  
+     .line 69
      monitor-enter p0
  
      const/4 v2, 0x0
  
+     .line 75
      .local v2, ret:Z
      :try_start_0
      iget v5, p0, Landroid/net/http/IdleCache;->mCount:I
  
      if-ge v5, v6, :cond_0
  
+     .line 76
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v3
  
+     .line 77
      .local v3, time:J
      const/4 v1, 0x0
  
***************
*** 224,260 ****
--- 253,298 ----
      :goto_0
      if-ge v1, v6, :cond_0
  
+     .line 78
      iget-object v5, p0, Landroid/net/http/IdleCache;->mEntries:[Landroid/net/http/IdleCache$Entry;
  
      aget-object v0, v5, v1
  
+     .line 79
      .local v0, entry:Landroid/net/http/IdleCache$Entry;
      iget-object v5, v0, Landroid/net/http/IdleCache$Entry;->mHost:Lorg/apache/http/HttpHost;
  
      if-nez v5, :cond_1
  
+     .line 80
      iput-object p1, v0, Landroid/net/http/IdleCache$Entry;->mHost:Lorg/apache/http/HttpHost;
  
+     .line 81
      iput-object p2, v0, Landroid/net/http/IdleCache$Entry;->mConnection:Landroid/net/http/Connection;
  
+     .line 82
      const-wide/16 v5, 0x1770
  
      add-long/2addr v5, v3
  
      iput-wide v5, v0, Landroid/net/http/IdleCache$Entry;->mTimeout:J
  
+     .line 83
      iget v5, p0, Landroid/net/http/IdleCache;->mCount:I
  
      add-int/lit8 v5, v5, 0x1
  
      iput v5, p0, Landroid/net/http/IdleCache;->mCount:I
  
+     .line 85
      const/4 v2, 0x1
  
+     .line 86
      iget-object v5, p0, Landroid/net/http/IdleCache;->mThread:Landroid/net/http/IdleCache$IdleReaper;
  
      if-nez v5, :cond_0
  
+     .line 87
      new-instance v5, Landroid/net/http/IdleCache$IdleReaper;
  
      const/4 v6, 0x0
***************
*** 263,274 ****
--- 301,314 ----
  
      iput-object v5, p0, Landroid/net/http/IdleCache;->mThread:Landroid/net/http/IdleCache$IdleReaper;
  
+     .line 88
      iget-object v5, p0, Landroid/net/http/IdleCache;->mThread:Landroid/net/http/IdleCache$IdleReaper;
  
      invoke-virtual {v5}, Landroid/net/http/IdleCache$IdleReaper;->start()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 94
      .end local v0           #entry:Landroid/net/http/IdleCache$Entry;
      .end local v1           #i:I
      .end local v3           #time:J
***************
*** 277,282 ****
--- 317,323 ----
  
      return v2
  
+     .line 77
      .restart local v0       #entry:Landroid/net/http/IdleCache$Entry;
      .restart local v1       #i:I
      .restart local v3       #time:J
***************
*** 285,290 ****
--- 326,332 ----
  
      goto :goto_0
  
+     .line 69
      .end local v0           #entry:Landroid/net/http/IdleCache$Entry;
      .end local v1           #i:I
      .end local v3           #time:J
***************
*** 300,305 ****
--- 342,348 ----
      .locals 3
  
      .prologue
+     .line 118
      monitor-enter p0
  
      const/4 v1, 0x0
***************
*** 315,341 ****
--- 358,390 ----
  
      if-ge v1, v2, :cond_1
  
+     .line 119
      iget-object v2, p0, Landroid/net/http/IdleCache;->mEntries:[Landroid/net/http/IdleCache$Entry;
  
      aget-object v0, v2, v1
  
+     .line 120
      .local v0, entry:Landroid/net/http/IdleCache$Entry;
      iget-object v2, v0, Landroid/net/http/IdleCache$Entry;->mHost:Lorg/apache/http/HttpHost;
  
      if-eqz v2, :cond_0
  
+     .line 121
      const/4 v2, 0x0
  
      iput-object v2, v0, Landroid/net/http/IdleCache$Entry;->mHost:Lorg/apache/http/HttpHost;
  
+     .line 122
      iget-object v2, v0, Landroid/net/http/IdleCache$Entry;->mConnection:Landroid/net/http/Connection;
  
      invoke-virtual {v2}, Landroid/net/http/Connection;->closeConnection()V
  
+     .line 123
      const/4 v2, 0x0
  
      iput-object v2, v0, Landroid/net/http/IdleCache$Entry;->mConnection:Landroid/net/http/Connection;
  
+     .line 124
      iget v2, p0, Landroid/net/http/IdleCache;->mCount:I
  
      add-int/lit8 v2, v2, -0x1
***************
*** 344,360 ****
--- 393,412 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 118
      :cond_0
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 127
      .end local v0           #entry:Landroid/net/http/IdleCache$Entry;
      :cond_1
      monitor-exit p0
  
      return-void
  
+     .line 118
      :catchall_0
      move-exception v2
  
***************
*** 368,383 ****
--- 420,438 ----
      .parameter "host"
  
      .prologue
+     .line 98
      monitor-enter p0
  
      const/4 v3, 0x0
  
+     .line 100
      .local v3, ret:Landroid/net/http/Connection;
      :try_start_0
      iget v4, p0, Landroid/net/http/IdleCache;->mCount:I
  
      if-lez v4, :cond_0
  
+     .line 101
      const/4 v2, 0x0
  
      .local v2, i:I
***************
*** 386,398 ****
--- 441,456 ----
  
      if-ge v2, v4, :cond_0
  
+     .line 102
      iget-object v4, p0, Landroid/net/http/IdleCache;->mEntries:[Landroid/net/http/IdleCache$Entry;
  
      aget-object v1, v4, v2
  
+     .line 103
      .local v1, entry:Landroid/net/http/IdleCache$Entry;
      iget-object v0, v1, Landroid/net/http/IdleCache$Entry;->mHost:Lorg/apache/http/HttpHost;
  
+     .line 104
      .local v0, eHost:Lorg/apache/http/HttpHost;
      if-eqz v0, :cond_1
  
***************
*** 402,417 ****
--- 460,479 ----
  
      if-eqz v4, :cond_1
  
+     .line 105
      iget-object v3, v1, Landroid/net/http/IdleCache$Entry;->mConnection:Landroid/net/http/Connection;
  
+     .line 106
      const/4 v4, 0x0
  
      iput-object v4, v1, Landroid/net/http/IdleCache$Entry;->mHost:Lorg/apache/http/HttpHost;
  
+     .line 107
      const/4 v4, 0x0
  
      iput-object v4, v1, Landroid/net/http/IdleCache$Entry;->mConnection:Landroid/net/http/Connection;
  
+     .line 108
      iget v4, p0, Landroid/net/http/IdleCache;->mCount:I
  
      add-int/lit8 v4, v4, -0x1
***************
*** 420,425 ****
--- 482,488 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 114
      .end local v0           #eHost:Lorg/apache/http/HttpHost;
      .end local v1           #entry:Landroid/net/http/IdleCache$Entry;
      .end local v2           #i:I
***************
*** 428,433 ****
--- 491,497 ----
  
      return-object v3
  
+     .line 101
      .restart local v0       #eHost:Lorg/apache/http/HttpHost;
      .restart local v1       #entry:Landroid/net/http/IdleCache$Entry;
      .restart local v2       #i:I
***************
*** 436,441 ****
--- 500,506 ----
  
      goto :goto_0
  
+     .line 98
      .end local v0           #eHost:Lorg/apache/http/HttpHost;
      .end local v1           #entry:Landroid/net/http/IdleCache$Entry;
      .end local v2           #i:I
