*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/AsyncService$AsyncServiceInfo.smali	2012-03-23 19:37:29.492312874 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/AsyncService$AsyncServiceInfo.smali.original	2012-03-23 19:34:46.011331803 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 0
  
      .prologue
+     .line 52
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
