*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/AsyncService.smali	2012-03-23 19:37:29.536313676 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/AsyncService.smali.original	2012-03-23 19:34:46.012331821 -0500
***************
*** 34,41 ****
--- 34,43 ----
      .locals 0
  
      .prologue
+     .line 35
      invoke-direct {p0}, Landroid/app/Service;-><init>()V
  
+     .line 52
      return-void
  .end method
  
***************
*** 48,53 ****
--- 50,56 ----
      .locals 1
  
      .prologue
+     .line 76
      iget-object v0, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
      return-object v0
***************
*** 58,63 ****
--- 61,67 ----
      .parameter "intent"
  
      .prologue
+     .line 126
      iget-object v0, p0, Lcom/android/internal/util/AsyncService;->mMessenger:Landroid/os/Messenger;
  
      invoke-virtual {v0}, Landroid/os/Messenger;->getBinder()Landroid/os/IBinder;
***************
*** 71,90 ****
--- 75,98 ----
      .locals 2
  
      .prologue
+     .line 84
      invoke-super {p0}, Landroid/app/Service;->onCreate()V
  
+     .line 85
      invoke-virtual {p0}, Lcom/android/internal/util/AsyncService;->createHandler()Lcom/android/internal/util/AsyncService$AsyncServiceInfo;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/internal/util/AsyncService;->mAsyncServiceInfo:Lcom/android/internal/util/AsyncService$AsyncServiceInfo;
  
+     .line 86
      iget-object v0, p0, Lcom/android/internal/util/AsyncService;->mAsyncServiceInfo:Lcom/android/internal/util/AsyncService$AsyncServiceInfo;
  
      iget-object v0, v0, Lcom/android/internal/util/AsyncService$AsyncServiceInfo;->mHandler:Landroid/os/Handler;
  
      iput-object v0, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
+     .line 87
      new-instance v0, Landroid/os/Messenger;
  
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
***************
*** 93,98 ****
--- 101,107 ----
  
      iput-object v0, p0, Lcom/android/internal/util/AsyncService;->mMessenger:Landroid/os/Messenger;
  
+     .line 88
      return-void
  .end method
  
***************
*** 100,126 ****
      .locals 3
  
      .prologue
      const-string v1, "AsyncService"
  
!     const-string v2, "onDestroy"
  
      invoke-static {v1, v2}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1}, Landroid/os/Handler;->obtainMessage()Landroid/os/Message;
  
      move-result-object v0
  
      .local v0, msg:Landroid/os/Message;
      const/high16 v1, 0x100
  
      iput v1, v0, Landroid/os/Message;->what:I
  
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
      return-void
  .end method
  
--- 109,140 ----
      .locals 3
  
      .prologue
+     .line 114
      const-string v1, "AsyncService"
  
!     const-string/jumbo v2, "onDestroy"
  
      invoke-static {v1, v2}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 116
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1}, Landroid/os/Handler;->obtainMessage()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 117
      .local v0, msg:Landroid/os/Message;
      const/high16 v1, 0x100
  
      iput v1, v0, Landroid/os/Message;->what:I
  
+     .line 118
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 119
      return-void
  .end method
  
***************
*** 131,163 ****
      .parameter "startId"
  
      .prologue
      const-string v1, "AsyncService"
  
!     const-string v2, "onStartCommand"
  
      invoke-static {v1, v2}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1}, Landroid/os/Handler;->obtainMessage()Landroid/os/Message;
  
      move-result-object v0
  
      .local v0, msg:Landroid/os/Message;
      const v1, 0xffffff
  
      iput v1, v0, Landroid/os/Message;->what:I
  
      iput p2, v0, Landroid/os/Message;->arg1:I
  
      iput p3, v0, Landroid/os/Message;->arg2:I
  
      iput-object p1, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mAsyncServiceInfo:Lcom/android/internal/util/AsyncService$AsyncServiceInfo;
  
      iget v1, v1, Lcom/android/internal/util/AsyncService$AsyncServiceInfo;->mRestartFlags:I
--- 145,185 ----
      .parameter "startId"
  
      .prologue
+     .line 95
      const-string v1, "AsyncService"
  
!     const-string/jumbo v2, "onStartCommand"
  
      invoke-static {v1, v2}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 97
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1}, Landroid/os/Handler;->obtainMessage()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 98
      .local v0, msg:Landroid/os/Message;
      const v1, 0xffffff
  
      iput v1, v0, Landroid/os/Message;->what:I
  
+     .line 99
      iput p2, v0, Landroid/os/Message;->arg1:I
  
+     .line 100
      iput p3, v0, Landroid/os/Message;->arg2:I
  
+     .line 101
      iput-object p1, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 102
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v1, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 104
      iget-object v1, p0, Lcom/android/internal/util/AsyncService;->mAsyncServiceInfo:Lcom/android/internal/util/AsyncService$AsyncServiceInfo;
  
      iget v1, v1, Lcom/android/internal/util/AsyncService$AsyncServiceInfo;->mRestartFlags:I
