*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/CharSequences.smali	2012-03-23 19:37:30.334328228 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/CharSequences.smali.original	2012-03-23 19:34:46.031332167 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 22
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 19,24 ****
--- 20,26 ----
      .parameter "another"
  
      .prologue
+     .line 119
      invoke-interface {p0}, Ljava/lang/CharSequence;->length()I
  
      move-result v4
***************
*** 28,39 ****
--- 30,43 ----
  
      move-result v0
  
+     .line 120
      .local v0, anotherLen:I
      const/4 v5, 0x0
  
      .local v5, myPos:I
      const/4 v1, 0x0
  
+     .line 121
      .local v1, anotherPos:I
      if-ge v4, v0, :cond_0
  
***************
*** 47,57 ****
--- 51,63 ----
      .local v2, anotherPos:I
      move v6, v5
  
+     .line 123
      .end local v5           #myPos:I
      .local v6, myPos:I
      :goto_1
      if-ge v6, v3, :cond_1
  
+     .line 124
      add-int/lit8 v5, v6, 0x1
  
      .end local v6           #myPos:I
***************
*** 81,86 ****
--- 87,93 ----
      .local v7, result:I
      if-eqz v7, :cond_2
  
+     .line 129
      .end local v7           #result:I
      :goto_2
      return v7
***************
*** 89,96 ****
--- 96,105 ----
      :cond_0
      move v3, v0
  
+     .line 121
      goto :goto_0
  
+     .line 129
      .end local v1           #anotherPos:I
      .end local v5           #myPos:I
      .restart local v2       #anotherPos:I
***************
*** 130,135 ****
--- 139,145 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 97
      invoke-interface {p0}, Ljava/lang/CharSequence;->length()I
  
      move-result v3
***************
*** 140,154 ****
--- 150,167 ----
  
      if-eq v3, v4, :cond_1
  
+     .line 107
      :cond_0
      :goto_0
      return v2
  
+     .line 101
      :cond_1
      invoke-interface {p0}, Ljava/lang/CharSequence;->length()I
  
      move-result v1
  
+     .line 102
      .local v1, length:I
      const/4 v0, 0x0
  
***************
*** 156,161 ****
--- 169,175 ----
      :goto_1
      if-ge v0, v1, :cond_2
  
+     .line 103
      invoke-interface {p0, v0}, Ljava/lang/CharSequence;->charAt(I)C
  
      move-result v3
***************
*** 166,175 ****
--- 180,191 ----
  
      if-ne v3, v4, :cond_0
  
+     .line 102
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 107
      :cond_2
      const/4 v2, 0x1
  
***************
*** 181,186 ****
--- 197,203 ----
      .parameter "bytes"
  
      .prologue
+     .line 31
      new-instance v0, Lcom/android/internal/util/CharSequences$1;
  
      invoke-direct {v0, p0}, Lcom/android/internal/util/CharSequences$1;-><init>([B)V
***************
*** 195,204 ****
--- 212,223 ----
      .parameter "end"
  
      .prologue
+     .line 63
      array-length v0, p0
  
      invoke-static {p1, p2, v0}, Lcom/android/internal/util/CharSequences;->validate(III)V
  
+     .line 64
      new-instance v0, Lcom/android/internal/util/CharSequences$2;
  
      invoke-direct {v0, p0, p1, p2}, Lcom/android/internal/util/CharSequences$2;-><init>([BII)V
***************
*** 213,218 ****
--- 232,238 ----
      .parameter "length"
  
      .prologue
+     .line 87
      if-gez p0, :cond_0
  
      new-instance v0, Ljava/lang/IndexOutOfBoundsException;
***************
*** 221,226 ****
--- 241,247 ----
  
      throw v0
  
+     .line 88
      :cond_0
      if-gez p1, :cond_1
  
***************
*** 230,235 ****
--- 251,257 ----
  
      throw v0
  
+     .line 89
      :cond_1
      if-le p1, p2, :cond_2
  
***************
*** 239,244 ****
--- 261,267 ----
  
      throw v0
  
+     .line 90
      :cond_2
      if-le p0, p1, :cond_3
  
***************
*** 248,253 ****
--- 271,277 ----
  
      throw v0
  
+     .line 91
      :cond_3
      return-void
  .end method
