*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/CharSequences$1.smali	2012-03-23 19:37:30.180325420 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/CharSequences$1.smali.original	2012-03-23 19:34:46.029332131 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 31
      iput-object p1, p0, Lcom/android/internal/util/CharSequences$1;->val$bytes:[B
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "index"
  
      .prologue
+     .line 33
      iget-object v0, p0, Lcom/android/internal/util/CharSequences$1;->val$bytes:[B
  
      aget-byte v0, v0, p1
***************
*** 54,59 ****
--- 56,62 ----
      .locals 1
  
      .prologue
+     .line 37
      iget-object v0, p0, Lcom/android/internal/util/CharSequences$1;->val$bytes:[B
  
      array-length v0, v0
***************
*** 67,72 ****
--- 70,76 ----
      .parameter "end"
  
      .prologue
+     .line 41
      iget-object v0, p0, Lcom/android/internal/util/CharSequences$1;->val$bytes:[B
  
      invoke-static {v0, p1, p2}, Lcom/android/internal/util/CharSequences;->forAsciiBytes([BII)Ljava/lang/CharSequence;
***************
*** 80,85 ****
--- 84,90 ----
      .locals 2
  
      .prologue
+     .line 45
      new-instance v0, Ljava/lang/String;
  
      iget-object v1, p0, Lcom/android/internal/util/CharSequences$1;->val$bytes:[B
