*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/Protocol.smali	2012-03-23 19:37:30.553332222 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/Protocol.smali.original	2012-03-23 19:34:46.035332241 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .locals 0
  
      .prologue
+     .line 34
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
