*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/StateMachine$SmHandler.smali	2012-03-23 19:37:29.770317944 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/StateMachine$SmHandler.smali.original	2012-03-23 19:34:46.015331875 -0500
***************
*** 82,87 ****
--- 82,88 ----
      .locals 1
  
      .prologue
+     .line 625
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
***************
*** 99,154 ****
--- 100,167 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 1097
      invoke-direct {p0, p1}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
+     .line 622
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
+     .line 631
      new-instance v0, Lcom/android/internal/util/StateMachine$ProcessedMessages;
  
      invoke-direct {v0}, Lcom/android/internal/util/StateMachine$ProcessedMessages;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mProcessedMessages:Lcom/android/internal/util/StateMachine$ProcessedMessages;
  
+     .line 640
      const/4 v0, -0x1
  
      iput v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
  
+     .line 649
      new-instance v0, Lcom/android/internal/util/StateMachine$SmHandler$HaltingState;
  
      invoke-direct {v0, p0, v1}, Lcom/android/internal/util/StateMachine$SmHandler$HaltingState;-><init>(Lcom/android/internal/util/StateMachine$SmHandler;Lcom/android/internal/util/StateMachine$1;)V
  
      iput-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mHaltingState:Lcom/android/internal/util/StateMachine$SmHandler$HaltingState;
  
+     .line 652
      new-instance v0, Lcom/android/internal/util/StateMachine$SmHandler$QuittingState;
  
      invoke-direct {v0, p0, v1}, Lcom/android/internal/util/StateMachine$SmHandler$QuittingState;-><init>(Lcom/android/internal/util/StateMachine$SmHandler;Lcom/android/internal/util/StateMachine$1;)V
  
      iput-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mQuittingState:Lcom/android/internal/util/StateMachine$SmHandler$QuittingState;
  
+     .line 683
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateInfo:Ljava/util/HashMap;
  
+     .line 693
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDeferredMessages:Ljava/util/ArrayList;
  
+     .line 1098
      iput-object p2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mSm:Lcom/android/internal/util/StateMachine;
  
+     .line 1100
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mHaltingState:Lcom/android/internal/util/StateMachine$SmHandler$HaltingState;
  
      invoke-direct {p0, v0, v1}, Lcom/android/internal/util/StateMachine$SmHandler;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 1101
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mQuittingState:Lcom/android/internal/util/StateMachine$SmHandler$QuittingState;
  
      invoke-direct {p0, v0, v1}, Lcom/android/internal/util/StateMachine$SmHandler;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 1102
      return-void
  .end method
  
***************
*** 159,164 ****
--- 172,178 ----
      .parameter "x2"
  
      .prologue
+     .line 619
      invoke-direct {p0, p1, p2}, Lcom/android/internal/util/StateMachine$SmHandler;-><init>(Landroid/os/Looper;Lcom/android/internal/util/StateMachine;)V
  
      return-void
***************
*** 170,175 ****
--- 184,190 ----
      .parameter "x1"
  
      .prologue
+     .line 619
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler;->setInitialState(Lcom/android/internal/util/State;)V
  
      return-void
***************
*** 181,186 ****
--- 196,202 ----
      .parameter "x1"
  
      .prologue
+     .line 619
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler;->transitionTo(Lcom/android/internal/util/IState;)V
  
      return-void
***************
*** 191,196 ****
--- 207,213 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mHaltingState:Lcom/android/internal/util/StateMachine$SmHandler$HaltingState;
  
      return-object v0
***************
*** 202,207 ****
--- 219,225 ----
      .parameter "x1"
  
      .prologue
+     .line 619
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler;->deferMessage(Landroid/os/Message;)V
  
      return-void
***************
*** 212,217 ****
--- 230,236 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      iget-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      return v0
***************
*** 223,228 ****
--- 242,248 ----
      .parameter "x1"
  
      .prologue
+     .line 619
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler;->setProcessedMessagesSize(I)V
  
      return-void
***************
*** 233,238 ****
--- 253,259 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->getProcessedMessagesSize()I
  
      move-result v0
***************
*** 245,250 ****
--- 266,272 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->getProcessedMessagesCount()I
  
      move-result v0
***************
*** 258,263 ****
--- 280,286 ----
      .parameter "x1"
  
      .prologue
+     .line 619
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler;->getProcessedMessageInfo(I)Lcom/android/internal/util/StateMachine$ProcessedMessageInfo;
  
      move-result-object v0
***************
*** 270,275 ****
--- 293,299 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->quit()V
  
      return-void
***************
*** 280,285 ****
--- 304,310 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mSm:Lcom/android/internal/util/StateMachine;
  
      return-object v0
***************
*** 291,296 ****
--- 316,322 ----
      .parameter "x1"
  
      .prologue
+     .line 619
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler;->isQuit(Landroid/os/Message;)Z
  
      move-result v0
***************
*** 303,308 ****
--- 329,335 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->isDbg()Z
  
      move-result v0
***************
*** 316,321 ****
--- 343,349 ----
      .parameter "x1"
  
      .prologue
+     .line 619
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler;->setDbg(Z)V
  
      return-void
***************
*** 326,331 ****
--- 354,360 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->completeConstruction()V
  
      return-void
***************
*** 338,343 ****
--- 367,373 ----
      .parameter "x2"
  
      .prologue
+     .line 619
      invoke-direct {p0, p1, p2}, Lcom/android/internal/util/StateMachine$SmHandler;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      move-result-object v0
***************
*** 350,355 ****
--- 380,386 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->getCurrentMessage()Landroid/os/Message;
  
      move-result-object v0
***************
*** 362,367 ****
--- 393,399 ----
      .parameter "x0"
  
      .prologue
+     .line 619
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->getCurrentState()Lcom/android/internal/util/IState;
  
      move-result-object v0
***************
*** 377,386 ****
--- 409,420 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 1060
      iget-boolean v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v2, :cond_0
  
+     .line 1061
      const-string v3, "StateMachine"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 422,433 ****
--- 456,470 ----
  
      invoke-static {v3, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1064
      :cond_0
      const/4 v0, 0x0
  
+     .line 1065
      .local v0, parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      if-eqz p2, :cond_1
  
+     .line 1066
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateInfo:Ljava/util/HashMap;
  
      invoke-virtual {v2, p2}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 437,449 ****
--- 474,489 ----
      .end local v0           #parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      check-cast v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 1067
      .restart local v0       #parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      if-nez v0, :cond_1
  
+     .line 1069
      invoke-direct {p0, p2, v5}, Lcom/android/internal/util/StateMachine$SmHandler;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      move-result-object v0
  
+     .line 1072
      :cond_1
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateInfo:Ljava/util/HashMap;
  
***************
*** 453,471 ****
--- 493,515 ----
  
      check-cast v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 1073
      .local v1, stateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      if-nez v1, :cond_2
  
+     .line 1074
      new-instance v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      .end local v1           #stateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      invoke-direct {v1, p0, v5}, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;-><init>(Lcom/android/internal/util/StateMachine$SmHandler;Lcom/android/internal/util/StateMachine$1;)V
  
+     .line 1075
      .restart local v1       #stateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateInfo:Ljava/util/HashMap;
  
      invoke-virtual {v2, p1, v1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 1079
      :cond_2
      iget-object v2, v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
***************
*** 475,488 ****
  
      if-eq v2, v0, :cond_4
  
      new-instance v2, Ljava/lang/RuntimeException;
  
!     const-string v3, "state already added"
  
      invoke-direct {v2, v3}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
      .end local v0           #parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      .end local v1           #stateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      :cond_3
--- 519,534 ----
  
      if-eq v2, v0, :cond_4
  
+     .line 1081
      new-instance v2, Ljava/lang/RuntimeException;
  
!     const-string/jumbo v3, "state already added"
  
      invoke-direct {v2, v3}, Ljava/lang/RuntimeException;-><init>(Ljava/lang/String;)V
  
      throw v2
  
+     .line 1061
      .end local v0           #parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      .end local v1           #stateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      :cond_3
***************
*** 492,508 ****
--- 538,558 ----
  
      goto :goto_0
  
+     .line 1083
      .restart local v0       #parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      .restart local v1       #stateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      :cond_4
      iput-object p1, v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->state:Lcom/android/internal/util/State;
  
+     .line 1084
      iput-object v0, v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 1085
      const/4 v2, 0x0
  
      iput-boolean v2, v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->active:Z
  
+     .line 1086
      iget-boolean v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v2, :cond_5
***************
*** 529,534 ****
--- 579,585 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1087
      :cond_5
      return-object v1
  .end method
***************
*** 539,548 ****
--- 590,601 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 810
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mSm:Lcom/android/internal/util/StateMachine;
  
      invoke-virtual {v0}, Lcom/android/internal/util/StateMachine;->quitting()V
  
+     .line 811
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mSm:Lcom/android/internal/util/StateMachine;
  
      #getter for: Lcom/android/internal/util/StateMachine;->mSmThread:Landroid/os/HandlerThread;
***************
*** 552,598 ****
--- 605,664 ----
  
      if-eqz v0, :cond_0
  
+     .line 813
      invoke-virtual {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->getLooper()Landroid/os/Looper;
  
      move-result-object v0
  
      invoke-virtual {v0}, Landroid/os/Looper;->quit()V
  
+     .line 814
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mSm:Lcom/android/internal/util/StateMachine;
  
      #setter for: Lcom/android/internal/util/StateMachine;->mSmThread:Landroid/os/HandlerThread;
      invoke-static {v0, v1}, Lcom/android/internal/util/StateMachine;->access$302(Lcom/android/internal/util/StateMachine;Landroid/os/HandlerThread;)Landroid/os/HandlerThread;
  
+     .line 817
      :cond_0
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mSm:Lcom/android/internal/util/StateMachine;
  
      #setter for: Lcom/android/internal/util/StateMachine;->mSmHandler:Lcom/android/internal/util/StateMachine$SmHandler;
      invoke-static {v0, v1}, Lcom/android/internal/util/StateMachine;->access$402(Lcom/android/internal/util/StateMachine;Lcom/android/internal/util/StateMachine$SmHandler;)Lcom/android/internal/util/StateMachine$SmHandler;
  
+     .line 818
      iput-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mSm:Lcom/android/internal/util/StateMachine;
  
+     .line 819
      iput-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mMsg:Landroid/os/Message;
  
+     .line 820
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mProcessedMessages:Lcom/android/internal/util/StateMachine$ProcessedMessages;
  
      invoke-virtual {v0}, Lcom/android/internal/util/StateMachine$ProcessedMessages;->cleanup()V
  
+     .line 821
      iput-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 822
      iput-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mTempStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 823
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateInfo:Ljava/util/HashMap;
  
      invoke-virtual {v0}, Ljava/util/HashMap;->clear()V
  
+     .line 824
      iput-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mInitialState:Lcom/android/internal/util/State;
  
+     .line 825
      iput-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDestState:Lcom/android/internal/util/State;
  
+     .line 826
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDeferredMessages:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
  
+     .line 827
      return-void
  .end method
  
***************
*** 600,605 ****
--- 666,672 ----
      .locals 8
  
      .prologue
+     .line 833
      iget-boolean v5, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v5, :cond_0
***************
*** 610,618 ****
--- 677,687 ----
  
      invoke-static {v5, v6}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 839
      :cond_0
      const/4 v3, 0x0
  
+     .line 840
      .local v3, maxDepth:I
      iget-object v5, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateInfo:Ljava/util/HashMap;
  
***************
*** 639,647 ****
--- 708,718 ----
  
      check-cast v4, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 841
      .local v4, si:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      const/4 v0, 0x0
  
+     .line 842
      .local v0, depth:I
      move-object v1, v4
  
***************
*** 649,667 ****
--- 720,743 ----
      :goto_1
      if-eqz v1, :cond_2
  
+     .line 843
      iget-object v1, v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 842
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 845
      :cond_2
      if-ge v3, v0, :cond_1
  
+     .line 846
      move v3, v0
  
      goto :goto_0
  
+     .line 849
      .end local v0           #depth:I
      .end local v1           #i:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      .end local v4           #si:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
***************
*** 692,712 ****
--- 768,793 ----
  
      invoke-static {v5, v6}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 851
      :cond_4
      new-array v5, v3, [Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      iput-object v5, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 852
      new-array v5, v3, [Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      iput-object v5, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mTempStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 853
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->setupInitialStateStack()V
  
+     .line 859
      const/4 v5, 0x1
  
      iput-boolean v5, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mIsConstructionCompleted:Z
  
+     .line 860
      const/4 v5, -0x2
  
      invoke-virtual {p0, v5}, Lcom/android/internal/util/StateMachine$SmHandler;->obtainMessage(I)Landroid/os/Message;
***************
*** 715,726 ****
--- 796,810 ----
  
      iput-object v5, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mMsg:Landroid/os/Message;
  
+     .line 861
      const/4 v5, 0x0
  
      invoke-direct {p0, v5}, Lcom/android/internal/util/StateMachine$SmHandler;->invokeEnterMethods(I)V
  
+     .line 866
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->performTransitions()V
  
+     .line 868
      iget-boolean v5, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v5, :cond_5
***************
*** 731,736 ****
--- 815,821 ----
  
      invoke-static {v5, v6}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 869
      :cond_5
      return-void
  .end method
***************
*** 740,745 ****
--- 825,831 ----
      .parameter "msg"
  
      .prologue
+     .line 1118
      iget-boolean v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v1, :cond_0
***************
*** 768,785 ****
--- 854,875 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1121
      :cond_0
      invoke-virtual {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->obtainMessage()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 1122
      .local v0, newMsg:Landroid/os/Message;
      invoke-virtual {v0, p1}, Landroid/os/Message;->copyFrom(Landroid/os/Message;)V
  
+     .line 1124
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDeferredMessages:Ljava/util/ArrayList;
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 1125
      return-void
  .end method
  
***************
*** 787,792 ****
--- 877,883 ----
      .locals 1
  
      .prologue
+     .line 1040
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mMsg:Landroid/os/Message;
  
      return-object v0
***************
*** 796,801 ****
--- 887,893 ----
      .locals 2
  
      .prologue
+     .line 1047
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      iget v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
***************
*** 812,817 ****
--- 904,910 ----
      .parameter "index"
  
      .prologue
+     .line 1165
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mProcessedMessages:Lcom/android/internal/util/StateMachine$ProcessedMessages;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/util/StateMachine$ProcessedMessages;->get(I)Lcom/android/internal/util/StateMachine$ProcessedMessageInfo;
***************
*** 825,830 ****
--- 918,924 ----
      .locals 1
  
      .prologue
+     .line 1160
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mProcessedMessages:Lcom/android/internal/util/StateMachine$ProcessedMessages;
  
      invoke-virtual {v0}, Lcom/android/internal/util/StateMachine$ProcessedMessages;->count()I
***************
*** 838,843 ****
--- 932,938 ----
      .locals 1
  
      .prologue
+     .line 1155
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mProcessedMessages:Lcom/android/internal/util/StateMachine$ProcessedMessages;
  
      invoke-virtual {v0}, Lcom/android/internal/util/StateMachine$ProcessedMessages;->size()I
***************
*** 852,857 ****
--- 947,953 ----
      .parameter "stateStackEnteringIndex"
  
      .prologue
+     .line 931
      move v0, p1
  
      .local v0, i:I
***************
*** 860,865 ****
--- 956,962 ----
  
      if-gt v0, v1, :cond_1
  
+     .line 932
      iget-boolean v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v1, :cond_0
***************
*** 896,901 ****
--- 993,999 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 933
      :cond_0
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
***************
*** 905,910 ****
--- 1003,1009 ----
  
      invoke-virtual {v1}, Lcom/android/internal/util/State;->enter()V
  
+     .line 934
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      aget-object v1, v1, v0
***************
*** 913,922 ****
--- 1012,1023 ----
  
      iput-boolean v2, v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->active:Z
  
+     .line 931
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 936
      :cond_1
      return-void
  .end method
***************
*** 926,931 ****
--- 1027,1033 ----
      .parameter "commonStateInfo"
  
      .prologue
+     .line 917
      :goto_0
      iget v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
  
***************
*** 939,944 ****
--- 1041,1047 ----
  
      if-eq v1, p1, :cond_1
  
+     .line 919
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      iget v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
***************
*** 947,952 ****
--- 1050,1056 ----
  
      iget-object v0, v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->state:Lcom/android/internal/util/State;
  
+     .line 920
      .local v0, curState:Lcom/android/internal/util/State;
      iget-boolean v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
***************
*** 978,986 ****
--- 1082,1092 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 921
      :cond_0
      invoke-virtual {v0}, Lcom/android/internal/util/State;->exit()V
  
+     .line 922
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      iget v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
***************
*** 991,996 ****
--- 1097,1103 ----
  
      iput-boolean v2, v1, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->active:Z
  
+     .line 923
      iget v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
  
      add-int/lit8 v1, v1, -0x1
***************
*** 999,1004 ****
--- 1106,1112 ----
  
      goto :goto_0
  
+     .line 925
      .end local v0           #curState:Lcom/android/internal/util/State;
      :cond_1
      return-void
***************
*** 1008,1013 ****
--- 1116,1122 ----
      .locals 1
  
      .prologue
+     .line 1140
      iget-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      return v0
***************
*** 1018,1023 ****
--- 1127,1133 ----
      .parameter "msg"
  
      .prologue
+     .line 1135
      iget v0, p1, Landroid/os/Message;->what:I
  
      const/4 v1, -0x1
***************
*** 1045,1050 ****
--- 1155,1161 ----
      .locals 5
  
      .prologue
+     .line 948
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDeferredMessages:Ljava/util/ArrayList;
  
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
***************
*** 1057,1062 ****
--- 1168,1174 ----
      :goto_0
      if-ltz v1, :cond_1
  
+     .line 949
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDeferredMessages:Ljava/util/ArrayList;
  
      invoke-virtual {v2, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1065,1070 ****
--- 1177,1183 ----
  
      check-cast v0, Landroid/os/Message;
  
+     .line 950
      .local v0, curMsg:Landroid/os/Message;
      iget-boolean v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
***************
*** 1076,1082 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "moveDeferredMessageAtFrontOfQueue; what="
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1189,1195 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "moveDeferredMessageAtFrontOfQueue; what="
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1094,1112 ****
--- 1207,1229 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 951
      :cond_0
      invoke-virtual {p0, v0}, Lcom/android/internal/util/StateMachine$SmHandler;->sendMessageAtFrontOfQueue(Landroid/os/Message;)Z
  
+     .line 948
      add-int/lit8 v1, v1, -0x1
  
      goto :goto_0
  
+     .line 953
      .end local v0           #curMsg:Landroid/os/Message;
      :cond_1
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDeferredMessages:Ljava/util/ArrayList;
  
      invoke-virtual {v2}, Ljava/util/ArrayList;->clear()V
  
+     .line 954
      return-void
  .end method
  
***************
*** 1114,1135 ****
--- 1231,1257 ----
      .locals 7
  
      .prologue
+     .line 964
      iget v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
  
      add-int/lit8 v2, v3, 0x1
  
+     .line 965
      .local v2, startingIndex:I
      iget v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mTempStateStackCount:I
  
      add-int/lit8 v0, v3, -0x1
  
+     .line 966
      .local v0, i:I
      move v1, v2
  
+     .line 967
      .local v1, j:I
      :goto_0
      if-ltz v0, :cond_1
  
+     .line 968
      iget-boolean v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v3, :cond_0
***************
*** 1140,1146 ****
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v5, "moveTempStackToStateStack: i="
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1262,1268 ----
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v5, "moveTempStackToStateStack: i="
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1166,1171 ****
--- 1288,1294 ----
  
      invoke-static {v3, v4}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 969
      :cond_0
      iget-object v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
***************
*** 1175,1202 ****
  
      aput-object v4, v3, v1
  
      add-int/lit8 v1, v1, 0x1
  
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_0
  
      :cond_1
      add-int/lit8 v3, v1, -0x1
  
      iput v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
  
      iget-boolean v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v3, :cond_2
  
      const-string v3, "StateMachine"
  
      new-instance v4, Ljava/lang/StringBuilder;
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v5, "moveTempStackToStateStack: X mStateStackTop="
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1298,1330 ----
  
      aput-object v4, v3, v1
  
+     .line 970
      add-int/lit8 v1, v1, 0x1
  
+     .line 971
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_0
  
+     .line 974
      :cond_1
      add-int/lit8 v3, v1, -0x1
  
      iput v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
  
+     .line 975
      iget-boolean v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v3, :cond_2
  
+     .line 976
      const-string v3, "StateMachine"
  
      new-instance v4, Ljava/lang/StringBuilder;
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v5, "moveTempStackToStateStack: X mStateStackTop="
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1246,1251 ****
--- 1374,1380 ----
  
      invoke-static {v3, v4}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 980
      :cond_2
      return v2
  .end method
***************
*** 1254,1267 ****
--- 1383,1399 ----
      .locals 5
  
      .prologue
+     .line 756
      const/4 v1, 0x0
  
+     .line 757
      .local v1, destState:Lcom/android/internal/util/State;
      :goto_0
      iget-object v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDestState:Lcom/android/internal/util/State;
  
      if-eqz v3, :cond_1
  
+     .line 758
      iget-boolean v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v3, :cond_0
***************
*** 1272,1322 ****
--- 1404,1467 ----
  
      invoke-static {v3, v4}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 764
      :cond_0
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDestState:Lcom/android/internal/util/State;
  
+     .line 765
      const/4 v3, 0x0
  
      iput-object v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDestState:Lcom/android/internal/util/State;
  
+     .line 772
      invoke-direct {p0, v1}, Lcom/android/internal/util/StateMachine$SmHandler;->setupTempStateStackWithStatesToEnter(Lcom/android/internal/util/State;)Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      move-result-object v0
  
+     .line 773
      .local v0, commonStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      invoke-direct {p0, v0}, Lcom/android/internal/util/StateMachine$SmHandler;->invokeExitMethods(Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;)V
  
+     .line 774
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->moveTempStateStackToStateStack()I
  
      move-result v2
  
+     .line 775
      .local v2, stateStackEnteringIndex:I
      invoke-direct {p0, v2}, Lcom/android/internal/util/StateMachine$SmHandler;->invokeEnterMethods(I)V
  
+     .line 784
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->moveDeferredMessageAtFrontOfQueue()V
  
      goto :goto_0
  
+     .line 791
      .end local v0           #commonStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      .end local v2           #stateStackEnteringIndex:I
      :cond_1
      if-eqz v1, :cond_2
  
+     .line 792
      iget-object v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mQuittingState:Lcom/android/internal/util/StateMachine$SmHandler$QuittingState;
  
      if-ne v1, v3, :cond_3
  
+     .line 793
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->cleanupAfterQuitting()V
  
+     .line 804
      :cond_2
      :goto_1
      return-void
  
+     .line 795
      :cond_3
      iget-object v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mHaltingState:Lcom/android/internal/util/StateMachine$SmHandler$HaltingState;
  
      if-ne v1, v3, :cond_2
  
+     .line 801
      iget-object v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mSm:Lcom/android/internal/util/StateMachine;
  
      invoke-virtual {v3}, Lcom/android/internal/util/StateMachine;->halting()V
***************
*** 1331,1354 ****
      .prologue
      const/4 v5, 0x0
  
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      iget v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
  
      aget-object v0, v2, v3
  
      .local v0, curStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      iget-boolean v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v2, :cond_0
  
      const-string v2, "StateMachine"
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "processMsg: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1476,1502 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 877
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      iget v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
  
      aget-object v0, v2, v3
  
+     .line 878
      .local v0, curStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      iget-boolean v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v2, :cond_0
  
+     .line 879
      const-string v2, "StateMachine"
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "processMsg: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1370,1375 ****
--- 1518,1524 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 881
      :cond_0
      :goto_0
      iget-object v2, v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->state:Lcom/android/internal/util/State;
***************
*** 1380,1406 ****
--- 1529,1562 ----
  
      if-nez v2, :cond_1
  
+     .line 885
      iget-object v0, v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 886
      if-nez v0, :cond_2
  
+     .line 890
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mSm:Lcom/android/internal/util/StateMachine;
  
      invoke-virtual {v2, p1}, Lcom/android/internal/util/StateMachine;->unhandledMessage(Landroid/os/Message;)V
  
+     .line 891
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler;->isQuit(Landroid/os/Message;)Z
  
      move-result v2
  
      if-eqz v2, :cond_1
  
+     .line 892
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mQuittingState:Lcom/android/internal/util/StateMachine$SmHandler$QuittingState;
  
      invoke-direct {p0, v2}, Lcom/android/internal/util/StateMachine$SmHandler;->transitionTo(Lcom/android/internal/util/IState;)V
  
+     .line 904
      :cond_1
      if-eqz v0, :cond_3
  
+     .line 905
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      iget v3, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
***************
*** 1409,1414 ****
--- 1565,1571 ----
  
      iget-object v1, v2, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->state:Lcom/android/internal/util/State;
  
+     .line 906
      .local v1, orgState:Lcom/android/internal/util/State;
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mProcessedMessages:Lcom/android/internal/util/StateMachine$ProcessedMessages;
  
***************
*** 1416,1437 ****
  
      invoke-virtual {v2, p1, v3, v1}, Lcom/android/internal/util/StateMachine$ProcessedMessages;->add(Landroid/os/Message;Lcom/android/internal/util/State;Lcom/android/internal/util/State;)V
  
      .end local v1           #orgState:Lcom/android/internal/util/State;
      :goto_1
      return-void
  
      :cond_2
      iget-boolean v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v2, :cond_0
  
      const-string v2, "StateMachine"
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "processMsg: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1573,1597 ----
  
      invoke-virtual {v2, p1, v3, v1}, Lcom/android/internal/util/StateMachine$ProcessedMessages;->add(Landroid/os/Message;Lcom/android/internal/util/State;Lcom/android/internal/util/State;)V
  
+     .line 910
      .end local v1           #orgState:Lcom/android/internal/util/State;
      :goto_1
      return-void
  
+     .line 896
      :cond_2
      iget-boolean v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v2, :cond_0
  
+     .line 897
      const-string v2, "StateMachine"
  
      new-instance v3, Ljava/lang/StringBuilder;
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "processMsg: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1455,1460 ****
--- 1615,1621 ----
  
      goto :goto_0
  
+     .line 908
      :cond_3
      iget-object v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mProcessedMessages:Lcom/android/internal/util/StateMachine$ProcessedMessages;
  
***************
*** 1467,1482 ****
      .locals 2
  
      .prologue
      iget-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v0, :cond_0
  
      const-string v0, "StateMachine"
  
!     const-string v1, "quit:"
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
      :cond_0
      const/4 v0, -0x1
  
--- 1628,1645 ----
      .locals 2
  
      .prologue
+     .line 1129
      iget-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v0, :cond_0
  
      const-string v0, "StateMachine"
  
!     const-string/jumbo v1, "quit:"
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1130
      :cond_0
      const/4 v0, -0x1
  
***************
*** 1488,1493 ****
--- 1651,1657 ----
  
      invoke-virtual {p0, v0}, Lcom/android/internal/util/StateMachine$SmHandler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 1131
      return-void
  .end method
  
***************
*** 1496,1503 ****
--- 1660,1669 ----
      .parameter "dbg"
  
      .prologue
+     .line 1145
      iput-boolean p1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
+     .line 1146
      return-void
  .end method
  
***************
*** 1506,1511 ****
--- 1672,1678 ----
      .parameter "initialState"
  
      .prologue
+     .line 1106
      iget-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v0, :cond_0
***************
*** 1516,1522 ****
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "setInitialState: initialState"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1683,1689 ----
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "setInitialState: initialState"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1536,1544 ****
--- 1703,1713 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1107
      :cond_0
      iput-object p1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mInitialState:Lcom/android/internal/util/State;
  
+     .line 1108
      return-void
  .end method
  
***************
*** 1547,1556 ****
--- 1716,1727 ----
      .parameter "maxSize"
  
      .prologue
+     .line 1150
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mProcessedMessages:Lcom/android/internal/util/StateMachine$ProcessedMessages;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/util/StateMachine$ProcessedMessages;->setSize(I)V
  
+     .line 1151
      return-void
  .end method
  
***************
*** 1558,1574 ****
      .locals 4
  
      .prologue
      iget-boolean v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v1, :cond_0
  
      const-string v1, "StateMachine"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "setupInitialStateStack: E mInitialState="
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1729,1747 ----
      .locals 4
  
      .prologue
+     .line 1019
      iget-boolean v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v1, :cond_0
  
+     .line 1020
      const-string v1, "StateMachine"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "setupInitialStateStack: E mInitialState="
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1590,1595 ****
--- 1763,1769 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1024
      :cond_0
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateInfo:Ljava/util/HashMap;
  
***************
*** 1601,1606 ****
--- 1775,1781 ----
  
      check-cast v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 1025
      .local v0, curStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      const/4 v1, 0x0
  
***************
*** 1609,1622 ****
--- 1784,1800 ----
      :goto_0
      if-eqz v0, :cond_1
  
+     .line 1026
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mTempStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      iget v2, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mTempStateStackCount:I
  
      aput-object v0, v1, v2
  
+     .line 1027
      iget-object v0, v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 1025
      iget v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mTempStateStackCount:I
  
      add-int/lit8 v1, v1, 0x1
***************
*** 1625,1637 ****
--- 1803,1818 ----
  
      goto :goto_0
  
+     .line 1031
      :cond_1
      const/4 v1, -0x1
  
      iput v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateStackTopIndex:I
  
+     .line 1033
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->moveTempStateStackToStateStack()I
  
+     .line 1034
      return-void
  .end method
  
***************
*** 1640,1649 ****
--- 1821,1832 ----
      .parameter "destState"
  
      .prologue
+     .line 1001
      const/4 v1, 0x0
  
      iput v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mTempStateStackCount:I
  
+     .line 1002
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mStateInfo:Ljava/util/HashMap;
  
      invoke-virtual {v1, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1652,1657 ****
--- 1835,1841 ----
  
      check-cast v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 1004
      .local v0, curStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
      :cond_0
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mTempStateStack:[Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
***************
*** 1664,1689 ****
  
      aput-object v0, v1, v2
  
      iget-object v0, v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
      if-eqz v0, :cond_1
  
      iget-boolean v1, v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->active:Z
  
      if-eqz v1, :cond_0
  
      :cond_1
      iget-boolean v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v1, :cond_2
  
      const-string v1, "StateMachine"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "setupTempStateStackWithStatesToEnter: X mTempStateStackCount="
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1848,1877 ----
  
      aput-object v0, v1, v2
  
+     .line 1005
      iget-object v0, v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->parentStateInfo:Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;
  
+     .line 1006
      if-eqz v0, :cond_1
  
      iget-boolean v1, v0, Lcom/android/internal/util/StateMachine$SmHandler$StateInfo;->active:Z
  
      if-eqz v1, :cond_0
  
+     .line 1008
      :cond_1
      iget-boolean v1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v1, :cond_2
  
+     .line 1009
      const-string v1, "StateMachine"
  
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "setupTempStateStackWithStatesToEnter: X mTempStateStackCount="
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1711,1716 ****
--- 1899,1905 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1012
      :cond_2
      return-object v0
  .end method
***************
*** 1720,1730 ****
--- 1909,1921 ----
      .parameter "destState"
  
      .prologue
+     .line 1112
      check-cast p1, Lcom/android/internal/util/State;
  
      .end local p1
      iput-object p1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDestState:Lcom/android/internal/util/State;
  
+     .line 1113
      iget-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v0, :cond_0
***************
*** 1757,1762 ****
--- 1948,1954 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1114
      :cond_0
      return-void
  .end method
***************
*** 1768,1773 ****
--- 1960,1966 ----
      .parameter "msg"
  
      .prologue
+     .line 724
      iget-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v0, :cond_0
***************
*** 1796,1808 ****
--- 1989,2004 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 727
      :cond_0
      iput-object p1, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mMsg:Landroid/os/Message;
  
+     .line 732
      iget-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mIsConstructionCompleted:Z
  
      if-nez v0, :cond_2
  
+     .line 733
      const-string v0, "StateMachine"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 1825,1839 ****
--- 2021,2039 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 745
      :cond_1
      :goto_0
      return-void
  
+     .line 741
      :cond_2
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler;->processMsg(Landroid/os/Message;)V
  
+     .line 742
      invoke-direct {p0}, Lcom/android/internal/util/StateMachine$SmHandler;->performTransitions()V
  
+     .line 744
      iget-boolean v0, p0, Lcom/android/internal/util/StateMachine$SmHandler;->mDbg:Z
  
      if-eqz v0, :cond_1
