*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/StateMachine$SmHandler$QuittingState.smali	2012-03-23 19:37:30.510331438 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/StateMachine$SmHandler$QuittingState.smali.original	2012-03-23 19:34:46.034332222 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 709
      iput-object p1, p0, Lcom/android/internal/util/StateMachine$SmHandler$QuittingState;->this$0:Lcom/android/internal/util/StateMachine$SmHandler;
  
      invoke-direct {p0}, Lcom/android/internal/util/State;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 709
      invoke-direct {p0, p1}, Lcom/android/internal/util/StateMachine$SmHandler$QuittingState;-><init>(Lcom/android/internal/util/StateMachine$SmHandler;)V
  
      return-void
***************
*** 49,54 ****
--- 51,57 ----
      .parameter "msg"
  
      .prologue
+     .line 712
      const/4 v0, 0x0
  
      return v0
