*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/StateMachine$ProcessedMessages.smali	2012-03-23 19:37:30.617333388 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/StateMachine$ProcessedMessages.smali.original	2012-03-23 19:34:46.041332349 -0500
***************
*** 44,65 ****
--- 44,71 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 543
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 535
      new-instance v0, Ljava/util/Vector;
  
      invoke-direct {v0}, Ljava/util/Vector;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMessages:Ljava/util/Vector;
  
+     .line 536
      const/16 v0, 0x14
  
      iput v0, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMaxSize:I
  
+     .line 537
      iput v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mOldestIndex:I
  
+     .line 538
      iput v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mCount:I
  
+     .line 544
      return-void
  .end method
  
***************
*** 72,83 ****
--- 78,91 ----
      .parameter "orgState"
  
      .prologue
+     .line 604
      iget v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mCount:I
  
      add-int/lit8 v1, v1, 0x1
  
      iput v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mCount:I
  
+     .line 605
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMessages:Ljava/util/Vector;
  
      invoke-virtual {v1}, Ljava/util/Vector;->size()I
***************
*** 88,93 ****
--- 96,102 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 606
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMessages:Ljava/util/Vector;
  
      new-instance v2, Lcom/android/internal/util/StateMachine$ProcessedMessageInfo;
***************
*** 96,104 ****
--- 105,115 ----
  
      invoke-virtual {v1, v2}, Ljava/util/Vector;->add(Ljava/lang/Object;)Z
  
+     .line 615
      :goto_0
      return-void
  
+     .line 608
      :cond_0
      iget-object v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMessages:Ljava/util/Vector;
  
***************
*** 110,115 ****
--- 121,127 ----
  
      check-cast v0, Lcom/android/internal/util/StateMachine$ProcessedMessageInfo;
  
+     .line 609
      .local v0, pmi:Lcom/android/internal/util/StateMachine$ProcessedMessageInfo;
      iget v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mOldestIndex:I
  
***************
*** 117,132 ****
--- 129,147 ----
  
      iput v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mOldestIndex:I
  
+     .line 610
      iget v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mOldestIndex:I
  
      iget v2, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMaxSize:I
  
      if-lt v1, v2, :cond_1
  
+     .line 611
      const/4 v1, 0x0
  
      iput v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mOldestIndex:I
  
+     .line 613
      :cond_1
      invoke-virtual {v0, p1, p2, p3}, Lcom/android/internal/util/StateMachine$ProcessedMessageInfo;->update(Landroid/os/Message;Lcom/android/internal/util/State;Lcom/android/internal/util/State;)V
  
***************
*** 137,146 ****
--- 152,163 ----
      .locals 1
  
      .prologue
+     .line 575
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMessages:Ljava/util/Vector;
  
      invoke-virtual {v0}, Ljava/util/Vector;->clear()V
  
+     .line 576
      return-void
  .end method
  
***************
*** 148,153 ****
--- 165,171 ----
      .locals 1
  
      .prologue
+     .line 568
      iget v0, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mCount:I
  
      return v0
***************
*** 158,176 ****
--- 176,198 ----
      .parameter "index"
  
      .prologue
+     .line 584
      iget v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mOldestIndex:I
  
      add-int v0, v1, p1
  
+     .line 585
      .local v0, nextIndex:I
      iget v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMaxSize:I
  
      if-lt v0, v1, :cond_0
  
+     .line 586
      iget v1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMaxSize:I
  
      sub-int/2addr v0, v1
  
+     .line 588
      :cond_0
      invoke-virtual {p0}, Lcom/android/internal/util/StateMachine$ProcessedMessages;->size()I
  
***************
*** 178,185 ****
--- 200,209 ----
  
      if-lt v0, v1, :cond_1
  
+     .line 589
      const/4 v1, 0x0
  
+     .line 591
      :goto_0
      return-object v1
  
***************
*** 200,215 ****
--- 224,243 ----
      .parameter "maxSize"
  
      .prologue
+     .line 552
      iput p1, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMaxSize:I
  
+     .line 553
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mCount:I
  
+     .line 554
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMessages:Ljava/util/Vector;
  
      invoke-virtual {v0}, Ljava/util/Vector;->clear()V
  
+     .line 555
      return-void
  .end method
  
***************
*** 217,222 ****
--- 245,251 ----
      .locals 1
  
      .prologue
+     .line 561
      iget-object v0, p0, Lcom/android/internal/util/StateMachine$ProcessedMessages;->mMessages:Ljava/util/Vector;
  
      invoke-virtual {v0}, Ljava/util/Vector;->size()I
