*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/TypedProperties.smali	2012-03-23 19:37:30.589332878 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/util/TypedProperties.smali.original	2012-03-23 19:34:46.041332349 -0500
***************
*** 59,64 ****
--- 59,65 ----
      .locals 2
  
      .prologue
+     .line 88
      new-instance v0, Ljava/lang/String;
  
      const-string v1, "<TypedProperties:NULL_STRING>"
***************
*** 74,81 ****
--- 75,84 ----
      .locals 0
  
      .prologue
+     .line 343
      invoke-direct {p0}, Ljava/util/HashMap;-><init>()V
  
+     .line 344
      return-void
  .end method
  
***************
*** 94,174 ****
--- 97,196 ----
  
      const/4 v3, 0x1
  
+     .line 39
      new-instance v0, Ljava/io/StreamTokenizer;
  
      invoke-direct {v0, p0}, Ljava/io/StreamTokenizer;-><init>(Ljava/io/Reader;)V
  
+     .line 42
      .local v0, st:Ljava/io/StreamTokenizer;
      invoke-virtual {v0}, Ljava/io/StreamTokenizer;->resetSyntax()V
  
+     .line 50
      const/16 v1, 0x30
  
      const/16 v2, 0x39
  
      invoke-virtual {v0, v1, v2}, Ljava/io/StreamTokenizer;->wordChars(II)V
  
+     .line 51
      const/16 v1, 0x41
  
      const/16 v2, 0x5a
  
      invoke-virtual {v0, v1, v2}, Ljava/io/StreamTokenizer;->wordChars(II)V
  
+     .line 52
      const/16 v1, 0x61
  
      const/16 v2, 0x7a
  
      invoke-virtual {v0, v1, v2}, Ljava/io/StreamTokenizer;->wordChars(II)V
  
+     .line 53
      const/16 v1, 0x5f
  
      const/16 v2, 0x5f
  
      invoke-virtual {v0, v1, v2}, Ljava/io/StreamTokenizer;->wordChars(II)V
  
+     .line 54
      const/16 v1, 0x24
  
      const/16 v2, 0x24
  
      invoke-virtual {v0, v1, v2}, Ljava/io/StreamTokenizer;->wordChars(II)V
  
+     .line 55
      const/16 v1, 0x2e
  
      const/16 v2, 0x2e
  
      invoke-virtual {v0, v1, v2}, Ljava/io/StreamTokenizer;->wordChars(II)V
  
+     .line 56
      const/16 v1, 0x2d
  
      const/16 v2, 0x2d
  
      invoke-virtual {v0, v1, v2}, Ljava/io/StreamTokenizer;->wordChars(II)V
  
+     .line 57
      const/16 v1, 0x2b
  
      const/16 v2, 0x2b
  
      invoke-virtual {v0, v1, v2}, Ljava/io/StreamTokenizer;->wordChars(II)V
  
+     .line 60
      const/16 v1, 0x3d
  
      invoke-virtual {v0, v1}, Ljava/io/StreamTokenizer;->ordinaryChar(I)V
  
+     .line 63
      invoke-virtual {v0, v7, v7}, Ljava/io/StreamTokenizer;->whitespaceChars(II)V
  
+     .line 64
      invoke-virtual {v0, v4, v4}, Ljava/io/StreamTokenizer;->whitespaceChars(II)V
  
+     .line 65
      invoke-virtual {v0, v5, v5}, Ljava/io/StreamTokenizer;->whitespaceChars(II)V
  
+     .line 66
      invoke-virtual {v0, v6, v6}, Ljava/io/StreamTokenizer;->whitespaceChars(II)V
  
+     .line 67
      const/16 v1, 0x22
  
      invoke-virtual {v0, v1}, Ljava/io/StreamTokenizer;->quoteChar(I)V
  
+     .line 70
      invoke-virtual {v0, v3}, Ljava/io/StreamTokenizer;->slashStarComments(Z)V
  
+     .line 71
      invoke-virtual {v0, v3}, Ljava/io/StreamTokenizer;->slashSlashComments(Z)V
  
+     .line 73
      return-object v0
  .end method
  
***************
*** 177,183 ****
      .parameter "typeName"
  
      .prologue
!     const-string v0, "unset"
  
      invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 199,206 ----
      .parameter "typeName"
  
      .prologue
!     .line 111
!     const-string/jumbo v0, "unset"
  
      invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 185,195 ****
--- 208,221 ----
  
      if-eqz v0, :cond_0
  
+     .line 112
      const/16 v0, 0x78
  
+     .line 130
      :goto_0
      return v0
  
+     .line 113
      :cond_0
      const-string v0, "boolean"
  
***************
*** 199,208 ****
--- 225,236 ----
  
      if-eqz v0, :cond_1
  
+     .line 114
      const/16 v0, 0x5a
  
      goto :goto_0
  
+     .line 115
      :cond_1
      const-string v0, "byte"
  
***************
*** 212,223 ****
  
      if-eqz v0, :cond_2
  
      const/16 v0, 0x149
  
      goto :goto_0
  
      :cond_2
!     const-string v0, "short"
  
      invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 240,253 ----
  
      if-eqz v0, :cond_2
  
+     .line 116
      const/16 v0, 0x149
  
      goto :goto_0
  
+     .line 117
      :cond_2
!     const-string/jumbo v0, "short"
  
      invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 225,234 ****
--- 255,266 ----
  
      if-eqz v0, :cond_3
  
+     .line 118
      const/16 v0, 0x249
  
      goto :goto_0
  
+     .line 119
      :cond_3
      const-string v0, "int"
  
***************
*** 238,247 ****
--- 270,281 ----
  
      if-eqz v0, :cond_4
  
+     .line 120
      const/16 v0, 0x449
  
      goto :goto_0
  
+     .line 121
      :cond_4
      const-string v0, "long"
  
***************
*** 251,260 ****
--- 285,296 ----
  
      if-eqz v0, :cond_5
  
+     .line 122
      const/16 v0, 0x849
  
      goto :goto_0
  
+     .line 123
      :cond_5
      const-string v0, "float"
  
***************
*** 264,273 ****
--- 300,311 ----
  
      if-eqz v0, :cond_6
  
+     .line 124
      const/16 v0, 0x446
  
      goto :goto_0
  
+     .line 125
      :cond_6
      const-string v0, "double"
  
***************
*** 277,286 ****
--- 315,326 ----
  
      if-eqz v0, :cond_7
  
+     .line 126
      const/16 v0, 0x846
  
      goto :goto_0
  
+     .line 127
      :cond_7
      const-string v0, "String"
  
***************
*** 290,299 ****
--- 330,341 ----
  
      if-eqz v0, :cond_8
  
+     .line 128
      const/16 v0, 0x734c
  
      goto :goto_0
  
+     .line 130
      :cond_8
      const/4 v0, -0x1
  
***************
*** 333,345 ****
--- 375,390 ----
  
      const/4 v10, -0x3
  
+     .line 142
      invoke-static {p0}, Lcom/android/internal/util/TypedProperties;->initTokenizer(Ljava/io/Reader;)Ljava/io/StreamTokenizer;
  
      move-result-object v4
  
+     .line 147
      .local v4, st:Ljava/io/StreamTokenizer;
      const-string v0, "[a-zA-Z_$][0-9a-zA-Z_$]*"
  
+     .line 148
      .local v0, identifierPattern:Ljava/lang/String;
      const-string v8, "([a-zA-Z_$][0-9a-zA-Z_$]*\\.)*[a-zA-Z_$][0-9a-zA-Z_$]*"
  
***************
*** 347,374 ****
  
      move-result-object v3
  
      .local v3, propertyNamePattern:Ljava/util/regex/Pattern;
      :cond_0
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
      .local v5, token:I
      if-ne v5, v11, :cond_1
  
      return-void
  
      :cond_1
      if-eq v5, v10, :cond_2
  
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
!     const-string v9, "type name"
  
      invoke-direct {v8, v4, v9}, Lcom/android/internal/util/TypedProperties$ParseException;-><init>(Ljava/io/StreamTokenizer;Ljava/lang/String;)V
  
      throw v8
  
      :cond_2
      iget-object v8, v4, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
--- 392,425 ----
  
      move-result-object v3
  
+     .line 156
      .local v3, propertyNamePattern:Ljava/util/regex/Pattern;
      :cond_0
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
+     .line 157
      .local v5, token:I
      if-ne v5, v11, :cond_1
  
+     .line 222
      return-void
  
+     .line 160
      :cond_1
      if-eq v5, v10, :cond_2
  
+     .line 161
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
!     const-string/jumbo v9, "type name"
  
      invoke-direct {v8, v4, v9}, Lcom/android/internal/util/TypedProperties$ParseException;-><init>(Ljava/io/StreamTokenizer;Ljava/lang/String;)V
  
      throw v8
  
+     .line 163
      :cond_2
      iget-object v8, v4, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
***************
*** 376,405 ****
  
      move-result v6
  
      .local v6, type:I
      if-ne v6, v11, :cond_3
  
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
!     const-string v9, "valid type name"
  
      invoke-direct {v8, v4, v9}, Lcom/android/internal/util/TypedProperties$ParseException;-><init>(Ljava/io/StreamTokenizer;Ljava/lang/String;)V
  
      throw v8
  
      :cond_3
      iput-object v13, v4, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
      if-ne v6, v12, :cond_4
  
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
      const/16 v8, 0x28
  
      if-eq v5, v8, :cond_4
  
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v9, "\'(\'"
--- 427,463 ----
  
      move-result v6
  
+     .line 164
      .local v6, type:I
      if-ne v6, v11, :cond_3
  
+     .line 165
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
!     const-string/jumbo v9, "valid type name"
  
      invoke-direct {v8, v4, v9}, Lcom/android/internal/util/TypedProperties$ParseException;-><init>(Ljava/io/StreamTokenizer;Ljava/lang/String;)V
  
      throw v8
  
+     .line 167
      :cond_3
      iput-object v13, v4, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
+     .line 169
      if-ne v6, v12, :cond_4
  
+     .line 171
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
+     .line 172
      const/16 v8, 0x28
  
      if-eq v5, v8, :cond_4
  
+     .line 173
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v9, "\'(\'"
***************
*** 408,431 ****
  
      throw v8
  
      :cond_4
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
      if-eq v5, v10, :cond_5
  
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
!     const-string v9, "property name"
  
      invoke-direct {v8, v4, v9}, Lcom/android/internal/util/TypedProperties$ParseException;-><init>(Ljava/io/StreamTokenizer;Ljava/lang/String;)V
  
      throw v8
  
      :cond_5
      iget-object v2, v4, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
      .local v2, propertyName:Ljava/lang/String;
      invoke-virtual {v3, v2}, Ljava/util/regex/Pattern;->matcher(Ljava/lang/CharSequence;)Ljava/util/regex/Matcher;
  
--- 466,494 ----
  
      throw v8
  
+     .line 178
      :cond_4
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
+     .line 179
      if-eq v5, v10, :cond_5
  
+     .line 180
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
!     const-string/jumbo v9, "property name"
  
      invoke-direct {v8, v4, v9}, Lcom/android/internal/util/TypedProperties$ParseException;-><init>(Ljava/io/StreamTokenizer;Ljava/lang/String;)V
  
      throw v8
  
+     .line 182
      :cond_5
      iget-object v2, v4, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
+     .line 183
      .local v2, propertyName:Ljava/lang/String;
      invoke-virtual {v3, v2}, Ljava/util/regex/Pattern;->matcher(Ljava/lang/CharSequence;)Ljava/util/regex/Matcher;
  
***************
*** 437,463 ****
  
      if-nez v8, :cond_6
  
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
!     const-string v9, "valid property name"
  
      invoke-direct {v8, v4, v9}, Lcom/android/internal/util/TypedProperties$ParseException;-><init>(Ljava/io/StreamTokenizer;Ljava/lang/String;)V
  
      throw v8
  
      :cond_6
      iput-object v13, v4, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
      if-ne v6, v12, :cond_8
  
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
      const/16 v8, 0x29
  
      if-eq v5, v8, :cond_7
  
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v9, "\')\'"
--- 500,532 ----
  
      if-nez v8, :cond_6
  
+     .line 184
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
!     const-string/jumbo v9, "valid property name"
  
      invoke-direct {v8, v4, v9}, Lcom/android/internal/util/TypedProperties$ParseException;-><init>(Ljava/io/StreamTokenizer;Ljava/lang/String;)V
  
      throw v8
  
+     .line 186
      :cond_6
      iput-object v13, v4, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
+     .line 188
      if-ne v6, v12, :cond_8
  
+     .line 190
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
+     .line 191
      const/16 v8, 0x29
  
      if-eq v5, v8, :cond_7
  
+     .line 192
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v9, "\')\'"
***************
*** 466,483 ****
--- 535,556 ----
  
      throw v8
  
+     .line 194
      :cond_7
      invoke-interface {p1, v2}, Ljava/util/Map;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 217
      :goto_0
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
+     .line 218
      const/16 v8, 0x3b
  
      if-eq v5, v8, :cond_0
  
+     .line 219
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v9, "\';\'"
***************
*** 486,500 ****
--- 559,576 ----
  
      throw v8
  
+     .line 197
      :cond_8
      invoke-virtual {v4}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v5
  
+     .line 198
      const/16 v8, 0x3d
  
      if-eq v5, v8, :cond_9
  
+     .line 199
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v9, "\'=\'"
***************
*** 503,521 ****
--- 579,601 ----
  
      throw v8
  
+     .line 203
      :cond_9
      invoke-static {v4, v6}, Lcom/android/internal/util/TypedProperties;->parseValue(Ljava/io/StreamTokenizer;I)Ljava/lang/Object;
  
      move-result-object v7
  
+     .line 204
      .local v7, value:Ljava/lang/Object;
      invoke-interface {p1, v2}, Ljava/util/Map;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v1
  
+     .line 205
      .local v1, oldValue:Ljava/lang/Object;
      if-eqz v1, :cond_a
  
+     .line 208
      invoke-virtual {v7}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
  
      move-result-object v8
***************
*** 526,531 ****
--- 606,612 ----
  
      if-eq v8, v9, :cond_a
  
+     .line 209
      new-instance v8, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v9, "(property previously declared as a different type)"
***************
*** 534,539 ****
--- 615,621 ----
  
      throw v8
  
+     .line 213
      :cond_a
      invoke-interface {p1, v2, v7}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
***************
*** 553,569 ****
--- 635,655 ----
      .prologue
      const/4 v9, -0x3
  
+     .line 234
      invoke-virtual {p0}, Ljava/io/StreamTokenizer;->nextToken()I
  
      move-result v3
  
+     .line 236
      .local v3, token:I
      const/16 v7, 0x5a
  
      if-ne p1, v7, :cond_3
  
+     .line 237
      if-eq v3, v9, :cond_0
  
+     .line 238
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v8, "boolean constant"
***************
*** 572,579 ****
  
      throw v7
  
      :cond_0
!     const-string v7, "true"
  
      iget-object v8, p0, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
--- 658,666 ----
  
      throw v7
  
+     .line 241
      :cond_0
!     const-string/jumbo v7, "true"
  
      iget-object v8, p0, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
***************
*** 583,593 ****
--- 670,683 ----
  
      if-eqz v7, :cond_1
  
+     .line 242
      sget-object v7, Ljava/lang/Boolean;->TRUE:Ljava/lang/Boolean;
  
+     .line 330
      :goto_0
      return-object v7
  
+     .line 243
      :cond_1
      const-string v7, "false"
  
***************
*** 599,608 ****
--- 689,700 ----
  
      if-eqz v7, :cond_2
  
+     .line 244
      sget-object v7, Ljava/lang/Boolean;->FALSE:Ljava/lang/Boolean;
  
      goto :goto_0
  
+     .line 247
      :cond_2
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
***************
*** 612,617 ****
--- 704,710 ----
  
      throw v7
  
+     .line 248
      :cond_3
      and-int/lit16 v7, p1, 0xff
  
***************
*** 619,626 ****
--- 712,721 ----
  
      if-ne v7, v8, :cond_d
  
+     .line 249
      if-eq v3, v9, :cond_4
  
+     .line 250
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v8, "integer constant"
***************
*** 629,634 ****
--- 724,730 ----
  
      throw v7
  
+     .line 259
      :cond_4
      :try_start_0
      iget-object v7, p0, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
***************
*** 643,656 ****
--- 739,755 ----
  
      move-result-wide v4
  
+     .line 265
      .local v4, value:J
      shr-int/lit8 v7, p1, 0x8
  
      and-int/lit16 v6, v7, 0xff
  
+     .line 266
      .local v6, width:I
      packed-switch v6, :pswitch_data_0
  
+     .line 288
      :pswitch_0
      new-instance v7, Ljava/lang/IllegalStateException;
  
***************
*** 676,686 ****
--- 775,787 ----
  
      throw v7
  
+     .line 260
      .end local v4           #value:J
      .end local v6           #width:I
      :catch_0
      move-exception v2
  
+     .line 261
      .local v2, ex:Ljava/lang/NumberFormatException;
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
***************
*** 690,695 ****
--- 791,797 ----
  
      throw v7
  
+     .line 268
      .end local v2           #ex:Ljava/lang/NumberFormatException;
      .restart local v4       #value:J
      .restart local v6       #width:I
***************
*** 706,711 ****
--- 808,814 ----
  
      if-lez v7, :cond_6
  
+     .line 269
      :cond_5
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
***************
*** 715,720 ****
--- 818,824 ----
  
      throw v7
  
+     .line 271
      :cond_6
      new-instance v7, Ljava/lang/Byte;
  
***************
*** 726,731 ****
--- 830,836 ----
  
      goto :goto_0
  
+     .line 273
      :pswitch_2
      const-wide/16 v7, -0x8000
  
***************
*** 739,744 ****
--- 844,850 ----
  
      if-lez v7, :cond_8
  
+     .line 274
      :cond_7
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
***************
*** 748,753 ****
--- 854,860 ----
  
      throw v7
  
+     .line 276
      :cond_8
      new-instance v7, Ljava/lang/Short;
  
***************
*** 759,764 ****
--- 866,872 ----
  
      goto/16 :goto_0
  
+     .line 278
      :pswitch_3
      const-wide/32 v7, -0x80000000
  
***************
*** 772,777 ****
--- 880,886 ----
  
      if-lez v7, :cond_a
  
+     .line 279
      :cond_9
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
***************
*** 781,786 ****
--- 890,896 ----
  
      throw v7
  
+     .line 281
      :cond_a
      new-instance v7, Ljava/lang/Integer;
  
***************
*** 790,795 ****
--- 900,906 ----
  
      goto/16 :goto_0
  
+     .line 283
      :pswitch_4
      const-wide/high16 v7, -0x8000
  
***************
*** 803,808 ****
--- 914,920 ----
  
      if-lez v7, :cond_c
  
+     .line 284
      :cond_b
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
***************
*** 812,817 ****
--- 924,930 ----
  
      throw v7
  
+     .line 286
      :cond_c
      new-instance v7, Ljava/lang/Long;
  
***************
*** 819,824 ****
--- 932,938 ----
  
      goto/16 :goto_0
  
+     .line 291
      .end local v4           #value:J
      .end local v6           #width:I
      :cond_d
***************
*** 828,835 ****
--- 942,951 ----
  
      if-ne v7, v8, :cond_12
  
+     .line 292
      if-eq v3, v9, :cond_e
  
+     .line 293
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
      const-string v8, "float constant"
***************
*** 838,843 ****
--- 954,960 ----
  
      throw v7
  
+     .line 306
      :cond_e
      :try_start_1
      iget-object v7, p0, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
***************
*** 848,853 ****
--- 965,971 ----
  
      move-result-wide v4
  
+     .line 312
      .local v4, value:D
      shr-int/lit8 v7, p1, 0x8
  
***************
*** 857,866 ****
--- 975,986 ----
  
      if-ne v7, v8, :cond_11
  
+     .line 314
      invoke-static {v4, v5}, Ljava/lang/Math;->abs(D)D
  
      move-result-wide v0
  
+     .line 315
      .local v0, absValue:D
      const-wide/16 v7, 0x0
  
***************
*** 880,885 ****
--- 1000,1006 ----
  
      if-nez v7, :cond_10
  
+     .line 316
      const-wide/high16 v7, 0x36a0
  
      cmpg-double v7, v0, v7
***************
*** 892,897 ****
--- 1013,1019 ----
  
      if-lez v7, :cond_10
  
+     .line 317
      :cond_f
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
***************
*** 901,911 ****
--- 1023,1035 ----
  
      throw v7
  
+     .line 307
      .end local v0           #absValue:D
      .end local v4           #value:D
      :catch_1
      move-exception v2
  
+     .line 308
      .restart local v2       #ex:Ljava/lang/NumberFormatException;
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
***************
*** 915,920 ****
--- 1039,1045 ----
  
      throw v7
  
+     .line 320
      .end local v2           #ex:Ljava/lang/NumberFormatException;
      .restart local v0       #absValue:D
      .restart local v4       #value:D
***************
*** 927,932 ****
--- 1052,1058 ----
  
      goto/16 :goto_0
  
+     .line 323
      .end local v0           #absValue:D
      :cond_11
      new-instance v7, Ljava/lang/Double;
***************
*** 935,958 ****
  
      goto/16 :goto_0
  
      .end local v4           #value:D
      :cond_12
      const/16 v7, 0x734c
  
      if-ne p1, v7, :cond_15
  
      const/16 v7, 0x22
  
      if-ne v3, v7, :cond_13
  
      iget-object v7, p0, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
      goto/16 :goto_0
  
      :cond_13
      if-ne v3, v9, :cond_14
  
!     const-string v7, "null"
  
      iget-object v8, p0, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
--- 1061,1088 ----
  
      goto/16 :goto_0
  
+     .line 325
      .end local v4           #value:D
      :cond_12
      const/16 v7, 0x734c
  
      if-ne p1, v7, :cond_15
  
+     .line 327
      const/16 v7, 0x22
  
      if-ne v3, v7, :cond_13
  
+     .line 328
      iget-object v7, p0, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
      goto/16 :goto_0
  
+     .line 329
      :cond_13
      if-ne v3, v9, :cond_14
  
!     const-string/jumbo v7, "null"
  
      iget-object v8, p0, Ljava/io/StreamTokenizer;->sval:Ljava/lang/String;
  
***************
*** 962,971 ****
--- 1092,1103 ----
  
      if-eqz v7, :cond_14
  
+     .line 330
      sget-object v7, Lcom/android/internal/util/TypedProperties;->NULL_STRING:Ljava/lang/String;
  
      goto/16 :goto_0
  
+     .line 332
      :cond_14
      new-instance v7, Lcom/android/internal/util/TypedProperties$ParseException;
  
***************
*** 975,980 ****
--- 1107,1113 ----
  
      throw v7
  
+     .line 335
      :cond_15
      new-instance v7, Ljava/lang/IllegalStateException;
  
***************
*** 1000,1005 ****
--- 1133,1139 ----
  
      throw v7
  
+     .line 266
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_1
***************
*** 1020,1036 ****
--- 1154,1174 ----
      .parameter "key"
  
      .prologue
+     .line 403
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 404
      .local v0, value:Ljava/lang/Object;
      sget-object v1, Lcom/android/internal/util/TypedProperties;->NULL_STRING:Ljava/lang/String;
  
      if-ne v0, v1, :cond_0
  
+     .line 405
      const/4 v0, 0x0
  
+     .line 407
      .end local v0           #value:Ljava/lang/Object;
      :cond_0
      return-object v0
***************
*** 1041,1046 ****
--- 1179,1185 ----
      .parameter "property"
  
      .prologue
+     .line 600
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Lcom/android/internal/util/TypedProperties;->getBoolean(Ljava/lang/String;Z)Z
***************
*** 1056,1073 ****
--- 1195,1216 ----
      .parameter "def"
  
      .prologue
+     .line 435
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 436
      .local v0, value:Ljava/lang/Object;
      if-nez v0, :cond_0
  
+     .line 440
      .end local v0           #value:Ljava/lang/Object;
      .end local p2
      :goto_0
      return p2
  
+     .line 439
      .restart local v0       #value:Ljava/lang/Object;
      .restart local p2
      :cond_0
***************
*** 1075,1080 ****
--- 1218,1224 ----
  
      if-eqz v1, :cond_1
  
+     .line 440
      check-cast v0, Ljava/lang/Boolean;
  
      .end local v0           #value:Ljava/lang/Object;
***************
*** 1084,1089 ****
--- 1228,1234 ----
  
      goto :goto_0
  
+     .line 442
      .restart local v0       #value:Ljava/lang/Object;
      :cond_1
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
***************
*** 1100,1105 ****
--- 1245,1251 ----
      .parameter "property"
  
      .prologue
+     .line 612
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Lcom/android/internal/util/TypedProperties;->getByte(Ljava/lang/String;B)B
***************
*** 1115,1132 ****
--- 1261,1282 ----
      .parameter "def"
  
      .prologue
+     .line 455
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 456
      .local v0, value:Ljava/lang/Object;
      if-nez v0, :cond_0
  
+     .line 460
      .end local v0           #value:Ljava/lang/Object;
      .end local p2
      :goto_0
      return p2
  
+     .line 459
      .restart local v0       #value:Ljava/lang/Object;
      .restart local p2
      :cond_0
***************
*** 1134,1139 ****
--- 1284,1290 ----
  
      if-eqz v1, :cond_1
  
+     .line 460
      check-cast v0, Ljava/lang/Byte;
  
      .end local v0           #value:Ljava/lang/Object;
***************
*** 1143,1148 ****
--- 1294,1300 ----
  
      goto :goto_0
  
+     .line 462
      .restart local v0       #value:Ljava/lang/Object;
      :cond_1
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
***************
*** 1159,1164 ****
--- 1311,1317 ----
      .parameter "property"
  
      .prologue
+     .line 672
      const-wide/16 v0, 0x0
  
      invoke-virtual {p0, p1, v0, v1}, Lcom/android/internal/util/TypedProperties;->getDouble(Ljava/lang/String;D)D
***************
*** 1174,1191 ****
--- 1327,1348 ----
      .parameter "def"
  
      .prologue
+     .line 555
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 556
      .local v0, value:Ljava/lang/Object;
      if-nez v0, :cond_0
  
+     .line 560
      .end local v0           #value:Ljava/lang/Object;
      .end local p2
      :goto_0
      return-wide p2
  
+     .line 559
      .restart local v0       #value:Ljava/lang/Object;
      .restart local p2
      :cond_0
***************
*** 1193,1198 ****
--- 1350,1356 ----
  
      if-eqz v1, :cond_1
  
+     .line 560
      check-cast v0, Ljava/lang/Double;
  
      .end local v0           #value:Ljava/lang/Object;
***************
*** 1202,1207 ****
--- 1360,1366 ----
  
      goto :goto_0
  
+     .line 562
      .restart local v0       #value:Ljava/lang/Object;
      :cond_1
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
***************
*** 1218,1223 ****
--- 1377,1383 ----
      .parameter "property"
  
      .prologue
+     .line 660
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Lcom/android/internal/util/TypedProperties;->getFloat(Ljava/lang/String;F)F
***************
*** 1233,1250 ****
--- 1393,1414 ----
      .parameter "def"
  
      .prologue
+     .line 535
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 536
      .local v0, value:Ljava/lang/Object;
      if-nez v0, :cond_0
  
+     .line 540
      .end local v0           #value:Ljava/lang/Object;
      .end local p2
      :goto_0
      return p2
  
+     .line 539
      .restart local v0       #value:Ljava/lang/Object;
      .restart local p2
      :cond_0
***************
*** 1252,1257 ****
--- 1416,1422 ----
  
      if-eqz v1, :cond_1
  
+     .line 540
      check-cast v0, Ljava/lang/Float;
  
      .end local v0           #value:Ljava/lang/Object;
***************
*** 1261,1266 ****
--- 1426,1432 ----
  
      goto :goto_0
  
+     .line 542
      .restart local v0       #value:Ljava/lang/Object;
      :cond_1
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
***************
*** 1277,1282 ****
--- 1443,1449 ----
      .parameter "property"
  
      .prologue
+     .line 636
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Lcom/android/internal/util/TypedProperties;->getInt(Ljava/lang/String;I)I
***************
*** 1292,1309 ****
--- 1459,1480 ----
      .parameter "def"
  
      .prologue
+     .line 495
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 496
      .local v0, value:Ljava/lang/Object;
      if-nez v0, :cond_0
  
+     .line 500
      .end local v0           #value:Ljava/lang/Object;
      .end local p2
      :goto_0
      return p2
  
+     .line 499
      .restart local v0       #value:Ljava/lang/Object;
      .restart local p2
      :cond_0
***************
*** 1311,1316 ****
--- 1482,1488 ----
  
      if-eqz v1, :cond_1
  
+     .line 500
      check-cast v0, Ljava/lang/Integer;
  
      .end local v0           #value:Ljava/lang/Object;
***************
*** 1320,1325 ****
--- 1492,1498 ----
  
      goto :goto_0
  
+     .line 502
      .restart local v0       #value:Ljava/lang/Object;
      :cond_1
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
***************
*** 1336,1341 ****
--- 1509,1515 ----
      .parameter "property"
  
      .prologue
+     .line 648
      const-wide/16 v0, 0x0
  
      invoke-virtual {p0, p1, v0, v1}, Lcom/android/internal/util/TypedProperties;->getLong(Ljava/lang/String;J)J
***************
*** 1351,1368 ****
--- 1525,1546 ----
      .parameter "def"
  
      .prologue
+     .line 515
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 516
      .local v0, value:Ljava/lang/Object;
      if-nez v0, :cond_0
  
+     .line 520
      .end local v0           #value:Ljava/lang/Object;
      .end local p2
      :goto_0
      return-wide p2
  
+     .line 519
      .restart local v0       #value:Ljava/lang/Object;
      .restart local p2
      :cond_0
***************
*** 1370,1375 ****
--- 1548,1554 ----
  
      if-eqz v1, :cond_1
  
+     .line 520
      check-cast v0, Ljava/lang/Long;
  
      .end local v0           #value:Ljava/lang/Object;
***************
*** 1379,1384 ****
--- 1558,1564 ----
  
      goto :goto_0
  
+     .line 522
      .restart local v0       #value:Ljava/lang/Object;
      :cond_1
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
***************
*** 1395,1400 ****
--- 1575,1581 ----
      .parameter "property"
  
      .prologue
+     .line 624
      const/4 v0, 0x0
  
      invoke-virtual {p0, p1, v0}, Lcom/android/internal/util/TypedProperties;->getShort(Ljava/lang/String;S)S
***************
*** 1410,1427 ****
--- 1591,1612 ----
      .parameter "def"
  
      .prologue
+     .line 475
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 476
      .local v0, value:Ljava/lang/Object;
      if-nez v0, :cond_0
  
+     .line 480
      .end local v0           #value:Ljava/lang/Object;
      .end local p2
      :goto_0
      return p2
  
+     .line 479
      .restart local v0       #value:Ljava/lang/Object;
      .restart local p2
      :cond_0
***************
*** 1429,1434 ****
--- 1614,1620 ----
  
      if-eqz v1, :cond_1
  
+     .line 480
      check-cast v0, Ljava/lang/Short;
  
      .end local v0           #value:Ljava/lang/Object;
***************
*** 1438,1448 ****
  
      goto :goto_0
  
      .restart local v0       #value:Ljava/lang/Object;
      :cond_1
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
  
!     const-string v2, "short"
  
      invoke-direct {v1, p1, v0, v2}, Lcom/android/internal/util/TypedProperties$TypeException;-><init>(Ljava/lang/String;Ljava/lang/Object;Ljava/lang/String;)V
  
--- 1624,1635 ----
  
      goto :goto_0
  
+     .line 482
      .restart local v0       #value:Ljava/lang/Object;
      :cond_1
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
  
!     const-string/jumbo v2, "short"
  
      invoke-direct {v1, p1, v0, v2}, Lcom/android/internal/util/TypedProperties$TypeException;-><init>(Ljava/lang/String;Ljava/lang/Object;Ljava/lang/String;)V
  
***************
*** 1454,1459 ****
--- 1641,1647 ----
      .parameter "property"
  
      .prologue
+     .line 684
      const-string v0, ""
  
      invoke-virtual {p0, p1, v0}, Lcom/android/internal/util/TypedProperties;->getString(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
***************
*** 1469,1486 ****
--- 1657,1678 ----
      .parameter "def"
  
      .prologue
+     .line 575
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 576
      .local v0, value:Ljava/lang/Object;
      if-nez v0, :cond_0
  
+     .line 582
      .end local v0           #value:Ljava/lang/Object;
      .end local p2
      :goto_0
      return-object p2
  
+     .line 579
      .restart local v0       #value:Ljava/lang/Object;
      .restart local p2
      :cond_0
***************
*** 1488,1502 ****
--- 1680,1697 ----
  
      if-ne v0, v1, :cond_1
  
+     .line 580
      const/4 p2, 0x0
  
      goto :goto_0
  
+     .line 581
      :cond_1
      instance-of v1, v0, Ljava/lang/String;
  
      if-eqz v1, :cond_2
  
+     .line 582
      check-cast v0, Ljava/lang/String;
  
      .end local v0           #value:Ljava/lang/Object;
***************
*** 1504,1514 ****
  
      goto :goto_0
  
      .restart local v0       #value:Ljava/lang/Object;
      :cond_2
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
  
!     const-string v2, "string"
  
      invoke-direct {v1, p1, v0, v2}, Lcom/android/internal/util/TypedProperties$TypeException;-><init>(Ljava/lang/String;Ljava/lang/Object;Ljava/lang/String;)V
  
--- 1699,1710 ----
  
      goto :goto_0
  
+     .line 584
      .restart local v0       #value:Ljava/lang/Object;
      :cond_2
      new-instance v1, Lcom/android/internal/util/TypedProperties$TypeException;
  
!     const-string/jumbo v2, "string"
  
      invoke-direct {v1, p1, v0, v2}, Lcom/android/internal/util/TypedProperties$TypeException;-><init>(Ljava/lang/String;Ljava/lang/Object;Ljava/lang/String;)V
  
***************
*** 1520,1555 ****
--- 1716,1760 ----
      .parameter "property"
  
      .prologue
+     .line 703
      invoke-super {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 704
      .local v0, value:Ljava/lang/Object;
      if-nez v0, :cond_0
  
+     .line 705
      const/4 v1, -0x1
  
+     .line 712
      :goto_0
      return v1
  
+     .line 707
      :cond_0
      sget-object v1, Lcom/android/internal/util/TypedProperties;->NULL_STRING:Ljava/lang/String;
  
      if-ne v0, v1, :cond_1
  
+     .line 708
      const/4 v1, 0x0
  
      goto :goto_0
  
+     .line 709
      :cond_1
      instance-of v1, v0, Ljava/lang/String;
  
      if-eqz v1, :cond_2
  
+     .line 710
      const/4 v1, 0x1
  
      goto :goto_0
  
+     .line 712
      :cond_2
      const/4 v1, -0x2
  
***************
*** 1566,1572 ****
--- 1771,1779 ----
      .end annotation
  
      .prologue
+     .line 398
      invoke-static {p1, p0}, Lcom/android/internal/util/TypedProperties;->parse(Ljava/io/Reader;Ljava/util/Map;)V
  
+     .line 399
      return-void
  .end method
