*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/content/PackageHelper.smali	2012-03-23 19:37:48.341656591 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/content/PackageHelper.smali.original	2012-03-23 19:34:46.341337820 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .locals 0
  
      .prologue
+     .line 34
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 51,60 ****
--- 52,63 ----
      .prologue
      const/4 v9, 0x0
  
+     .line 64
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
      move-result-object v0
  
+     .line 67
      .local v0, mountService:Landroid/os/storage/IMountService;
      const-string v1, "PackageHelper"
  
***************
*** 84,89 ****
--- 87,93 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 70
      :try_start_0
      const-string v3, "fat"
  
***************
*** 99,107 ****
--- 103,113 ----
  
      move-result v8
  
+     .line 72
      .local v8, rc:I
      if-eqz v8, :cond_0
  
+     .line 73
      const-string v1, "PackageHelper"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 126,141 ****
--- 132,150 ----
  
      move-object v6, v9
  
+     .line 83
      .end local v8           #rc:I
      :goto_0
      return-object v6
  
+     .line 76
      .restart local v8       #rc:I
      :cond_0
      invoke-interface {v0, p1}, Landroid/os/storage/IMountService;->getSecureContainerPath(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v6
  
+     .line 77
      .local v6, cachePath:Ljava/lang/String;
      const-string v1, "PackageHelper"
  
***************
*** 173,183 ****
--- 182,194 ----
  
      goto :goto_0
  
+     .line 80
      .end local v6           #cachePath:Ljava/lang/String;
      .end local v8           #rc:I
      :catch_0
      move-exception v7
  
+     .line 81
      .local v7, e:Landroid/os/RemoteException;
      const-string v1, "PackageHelper"
  
***************
*** 187,192 ****
--- 198,204 ----
  
      move-object v6, v9
  
+     .line 83
      goto :goto_0
  .end method
  
***************
*** 199,204 ****
--- 211,217 ----
  
      const/4 v2, 0x0
  
+     .line 167
      :try_start_0
      const-string v4, "PackageHelper"
  
***************
*** 222,227 ****
--- 235,241 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 168
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
      move-result-object v4
***************
*** 232,240 ****
--- 246,256 ----
  
      move-result v1
  
+     .line 169
      .local v1, rc:I
      if-eqz v1, :cond_0
  
+     .line 170
      const-string v3, "PackageHelper"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 259,264 ****
--- 275,281 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 178
      .end local v1           #rc:I
      :goto_0
      return v2
***************
*** 267,278 ****
--- 284,298 ----
      :cond_0
      move v2, v3
  
+     .line 173
      goto :goto_0
  
+     .line 174
      .end local v1           #rc:I
      :catch_0
      move-exception v0
  
+     .line 175
      .local v0, e:Landroid/os/RemoteException;
      const-string v3, "PackageHelper"
  
***************
*** 316,321 ****
--- 336,342 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 152
      :try_start_0
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
***************
*** 325,333 ****
--- 346,356 ----
  
      move-result v1
  
+     .line 153
      .local v1, rc:I
      if-eqz v1, :cond_0
  
+     .line 154
      const-string v3, "PackageHelper"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 352,371 ****
--- 375,398 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 162
      .end local v1           #rc:I
      :goto_0
      return v2
  
+     .line 157
      .restart local v1       #rc:I
      :cond_0
      const/4 v2, 0x1
  
      goto :goto_0
  
+     .line 158
      .end local v1           #rc:I
      :catch_0
      move-exception v0
  
+     .line 159
      .local v0, e:Landroid/os/RemoteException;
      const-string v3, "PackageHelper"
  
***************
*** 406,427 ****
      .locals 3
  
      .prologue
!     const-string v1, "mount"
  
      invoke-static {v1}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
      move-result-object v0
  
      .local v0, service:Landroid/os/IBinder;
      if-eqz v0, :cond_0
  
      invoke-static {v0}, Landroid/os/storage/IMountService$Stub;->asInterface(Landroid/os/IBinder;)Landroid/os/storage/IMountService;
  
      move-result-object v1
  
      :goto_0
      return-object v1
  
      :cond_0
      const-string v1, "PackageHelper"
  
--- 433,459 ----
      .locals 3
  
      .prologue
!     .line 52
!     const-string/jumbo v1, "mount"
  
      invoke-static {v1}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 53
      .local v0, service:Landroid/os/IBinder;
      if-eqz v0, :cond_0
  
+     .line 54
      invoke-static {v0}, Landroid/os/storage/IMountService$Stub;->asInterface(Landroid/os/IBinder;)Landroid/os/storage/IMountService;
  
      move-result-object v1
  
+     .line 58
      :goto_0
      return-object v1
  
+     .line 56
      :cond_0
      const-string v1, "PackageHelper"
  
***************
*** 429,434 ****
--- 461,467 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 58
      const/4 v1, 0x0
  
      goto :goto_0
***************
*** 439,444 ****
--- 472,478 ----
      .parameter "cid"
  
      .prologue
+     .line 132
      :try_start_0
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
***************
*** 450,461 ****
--- 484,498 ----
  
      move-result-object v1
  
+     .line 137
      :goto_0
      return-object v1
  
+     .line 133
      :catch_0
      move-exception v0
  
+     .line 134
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "PackageHelper"
  
***************
*** 489,494 ****
--- 526,532 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 137
      const/4 v1, 0x0
  
      goto :goto_0
***************
*** 499,504 ****
--- 537,543 ----
      .parameter "cid"
  
      .prologue
+     .line 142
      :try_start_0
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
***************
*** 510,521 ****
--- 549,563 ----
  
      move-result-object v1
  
+     .line 147
      :goto_0
      return-object v1
  
+     .line 143
      :catch_0
      move-exception v0
  
+     .line 144
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "PackageHelper"
  
***************
*** 549,554 ****
--- 591,597 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 147
      const/4 v1, 0x0
  
      goto :goto_0
***************
*** 558,563 ****
--- 601,607 ----
      .locals 4
  
      .prologue
+     .line 183
      :try_start_0
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
***************
*** 569,582 ****
--- 613,629 ----
  
      move-result-object v1
  
+     .line 188
      .local v0, e:Landroid/os/RemoteException;
      :goto_0
      return-object v1
  
+     .line 184
      .end local v0           #e:Landroid/os/RemoteException;
      :catch_0
      move-exception v0
  
+     .line 185
      .restart local v0       #e:Landroid/os/RemoteException;
      const-string v1, "PackageHelper"
  
***************
*** 600,605 ****
--- 647,653 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 188
      const/4 v1, 0x0
  
      goto :goto_0
***************
*** 610,615 ****
--- 658,664 ----
      .parameter "cid"
  
      .prologue
+     .line 193
      :try_start_0
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
***************
*** 621,632 ****
--- 670,684 ----
  
      move-result v1
  
+     .line 197
      :goto_0
      return v1
  
+     .line 194
      :catch_0
      move-exception v0
  
+     .line 195
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "PackageHelper"
  
***************
*** 656,661 ****
--- 708,714 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 197
      const/4 v1, 0x0
  
      goto :goto_0
***************
*** 670,675 ****
--- 723,729 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 88
      :try_start_0
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
***************
*** 679,687 ****
--- 733,743 ----
  
      move-result v1
  
+     .line 89
      .local v1, rc:I
      if-eqz v1, :cond_0
  
+     .line 90
      const-string v3, "PackageHelper"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 714,723 ****
--- 770,781 ----
  
      invoke-static {v3, v4}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 97
      .end local v1           #rc:I
      :goto_0
      return-object v2
  
+     .line 93
      .restart local v1       #rc:I
      :cond_0
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
***************
*** 732,741 ****
--- 790,801 ----
  
      goto :goto_0
  
+     .line 94
      .end local v1           #rc:I
      :catch_0
      move-exception v0
  
+     .line 95
      .local v0, e:Landroid/os/RemoteException;
      const-string v3, "PackageHelper"
  
***************
*** 754,759 ****
--- 814,820 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 116
      :try_start_0
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
***************
*** 763,771 ****
--- 824,834 ----
  
      move-result v1
  
+     .line 117
      .local v1, rc:I
      if-eqz v1, :cond_0
  
+     .line 118
      const-string v3, "PackageHelper"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 792,798 ****
  
      move-result-object v4
  
!     const-string v5, "with rc "
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 855,861 ----
  
      move-result-object v4
  
!     const-string/jumbo v5, "with rc "
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 810,829 ****
--- 873,896 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 127
      .end local v1           #rc:I
      :goto_0
      return v2
  
+     .line 122
      .restart local v1       #rc:I
      :cond_0
      const/4 v2, 0x1
  
      goto :goto_0
  
+     .line 123
      .end local v1           #rc:I
      :catch_0
      move-exception v0
  
+     .line 124
      .local v0, e:Landroid/os/RemoteException;
      const-string v3, "PackageHelper"
  
***************
*** 879,884 ****
--- 946,952 ----
  
      const/4 v2, 0x0
  
+     .line 102
      :try_start_0
      invoke-static {}, Lcom/android/internal/content/PackageHelper;->getMountService()Landroid/os/storage/IMountService;
  
***************
*** 890,898 ****
--- 958,968 ----
  
      move-result v1
  
+     .line 103
      .local v1, rc:I
      if-eqz v1, :cond_0
  
+     .line 104
      const-string v3, "PackageHelper"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 927,932 ****
--- 997,1003 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 111
      .end local v1           #rc:I
      :goto_0
      return v2
***************
*** 935,946 ****
--- 1006,1020 ----
      :cond_0
      move v2, v3
  
+     .line 107
      goto :goto_0
  
+     .line 108
      .end local v1           #rc:I
      :catch_0
      move-exception v0
  
+     .line 109
      .local v0, e:Landroid/os/RemoteException;
      const-string v3, "PackageHelper"
  
