*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/DayOfMonthCursor.smali	2012-03-23 19:35:17.388903972 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/DayOfMonthCursor.smali.original	2012-03-23 19:34:44.459303501 -0500
***************
*** 18,37 ****
--- 18,41 ----
      .parameter "weekStartDay"
  
      .prologue
+     .line 49
      invoke-direct {p0, p1, p2, p4}, Landroid/util/MonthDisplayHelper;-><init>(III)V
  
+     .line 50
      invoke-virtual {p0, p3}, Landroid/util/DayOfMonthCursor;->getRowOf(I)I
  
      move-result v0
  
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 51
      invoke-virtual {p0, p3}, Landroid/util/DayOfMonthCursor;->getColumnOf(I)I
  
      move-result v0
  
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
+     .line 52
      return-void
  .end method
  
***************
*** 43,48 ****
--- 47,53 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 121
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      add-int/lit8 v1, v1, 0x1
***************
*** 55,74 ****
--- 60,84 ----
  
      if-eqz v1, :cond_0
  
+     .line 123
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      add-int/lit8 v1, v1, 0x1
  
      iput v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 132
      :goto_0
      return v0
  
+     .line 127
      :cond_0
      invoke-virtual {p0}, Landroid/util/DayOfMonthCursor;->nextMonth()V
  
+     .line 128
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 129
      :goto_1
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
***************
*** 80,85 ****
--- 90,96 ----
  
      if-nez v0, :cond_1
  
+     .line 130
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      add-int/lit8 v0, v0, 0x1
***************
*** 88,93 ****
--- 99,105 ----
  
      goto :goto_1
  
+     .line 132
      :cond_1
      const/4 v0, 0x1
  
***************
*** 98,103 ****
--- 110,116 ----
      .locals 1
  
      .prologue
+     .line 60
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
      return v0
***************
*** 107,112 ****
--- 120,126 ----
      .locals 2
  
      .prologue
+     .line 69
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
***************
*** 122,127 ****
--- 136,142 ----
      .locals 2
  
      .prologue
+     .line 77
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
***************
*** 132,151 ****
--- 147,171 ----
  
      if-eqz v0, :cond_0
  
+     .line 78
      const/4 v0, 0x0
  
+     .line 83
      :goto_0
      return v0
  
+     .line 80
      :cond_0
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      if-nez v0, :cond_1
  
+     .line 81
      const/4 v0, -0x1
  
      goto :goto_0
  
+     .line 83
      :cond_1
      const/4 v0, 0x1
  
***************
*** 156,161 ****
--- 176,182 ----
      .locals 1
  
      .prologue
+     .line 56
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      return v0
***************
*** 167,172 ****
--- 188,194 ----
      .parameter "column"
  
      .prologue
+     .line 92
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      if-ne v0, p1, :cond_0
***************
*** 190,209 ****
--- 212,235 ----
      .locals 3
  
      .prologue
+     .line 141
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
      if-nez v1, :cond_0
  
+     .line 142
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      add-int/lit8 v1, v1, -0x1
  
      iput v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 143
      const/4 v1, 0x6
  
      iput v1, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
+     .line 148
      :goto_0
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
***************
*** 215,225 ****
--- 241,254 ----
  
      if-eqz v1, :cond_1
  
+     .line 149
      const/4 v1, 0x0
  
+     .line 157
      :goto_1
      return v1
  
+     .line 145
      :cond_0
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
***************
*** 229,241 ****
--- 258,273 ----
  
      goto :goto_0
  
+     .line 153
      :cond_1
      invoke-virtual {p0}, Landroid/util/DayOfMonthCursor;->previousMonth()V
  
+     .line 154
      invoke-virtual {p0}, Landroid/util/DayOfMonthCursor;->getNumberOfDaysInMonth()I
  
      move-result v0
  
+     .line 155
      .local v0, lastDay:I
      invoke-virtual {p0, v0}, Landroid/util/DayOfMonthCursor;->getRowOf(I)I
  
***************
*** 243,254 ****
--- 275,288 ----
  
      iput v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 156
      invoke-virtual {p0, v0}, Landroid/util/DayOfMonthCursor;->getColumnOf(I)I
  
      move-result v1
  
      iput v1, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
+     .line 157
      const/4 v1, 0x1
  
      goto :goto_1
***************
*** 260,279 ****
--- 294,317 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 166
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
      const/4 v2, 0x6
  
      if-ne v1, v2, :cond_0
  
+     .line 167
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      add-int/lit8 v1, v1, 0x1
  
      iput v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 168
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
+     .line 173
      :goto_0
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
***************
*** 285,293 ****
--- 323,333 ----
  
      if-eqz v1, :cond_1
  
+     .line 184
      :goto_1
      return v0
  
+     .line 170
      :cond_0
      iget v1, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
***************
*** 297,309 ****
--- 337,353 ----
  
      goto :goto_0
  
+     .line 178
      :cond_1
      invoke-virtual {p0}, Landroid/util/DayOfMonthCursor;->nextMonth()V
  
+     .line 179
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 180
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
+     .line 181
      :goto_2
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
***************
*** 315,320 ****
--- 359,365 ----
  
      if-nez v0, :cond_2
  
+     .line 182
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
      add-int/lit8 v0, v0, 0x1
***************
*** 323,328 ****
--- 368,374 ----
  
      goto :goto_2
  
+     .line 184
      :cond_2
      const/4 v0, 0x1
  
***************
*** 334,351 ****
--- 380,400 ----
      .parameter "dayOfMonth"
  
      .prologue
+     .line 87
      invoke-virtual {p0, p1}, Landroid/util/DayOfMonthCursor;->getRowOf(I)I
  
      move-result v0
  
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 88
      invoke-virtual {p0, p1}, Landroid/util/DayOfMonthCursor;->getColumnOf(I)I
  
      move-result v0
  
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
+     .line 89
      return-void
  .end method
  
***************
*** 355,364 ****
--- 404,416 ----
      .parameter "col"
  
      .prologue
+     .line 64
      iput p1, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 65
      iput p2, p0, Landroid/util/DayOfMonthCursor;->mColumn:I
  
+     .line 66
      return-void
  .end method
  
***************
*** 366,371 ****
--- 418,424 ----
      .locals 2
  
      .prologue
+     .line 101
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      add-int/lit8 v0, v0, -0x1
***************
*** 378,401 ****
--- 431,460 ----
  
      if-eqz v0, :cond_0
  
+     .line 103
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      add-int/lit8 v0, v0, -0x1
  
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 104
      const/4 v0, 0x0
  
+     .line 112
      :goto_0
      return v0
  
+     .line 107
      :cond_0
      invoke-virtual {p0}, Landroid/util/DayOfMonthCursor;->previousMonth()V
  
+     .line 108
      const/4 v0, 0x5
  
      iput v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
+     .line 109
      :goto_1
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
***************
*** 407,412 ****
--- 466,472 ----
  
      if-nez v0, :cond_1
  
+     .line 110
      iget v0, p0, Landroid/util/DayOfMonthCursor;->mRow:I
  
      add-int/lit8 v0, v0, -0x1
***************
*** 415,420 ****
--- 475,481 ----
  
      goto :goto_1
  
+     .line 112
      :cond_1
      const/4 v0, 0x1
  
