*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/EventLog.smali	2012-03-23 19:35:16.999896878 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/EventLog.smali.original	2012-03-23 19:34:44.453303392 -0500
***************
*** 52,59 ****
--- 52,61 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 52
      sput-object v0, Landroid/util/EventLog;->sTagCodes:Ljava/util/HashMap;
  
+     .line 53
      sput-object v0, Landroid/util/EventLog;->sTagNames:Ljava/util/HashMap;
  
      return-void
***************
*** 63,68 ****
--- 65,71 ----
      .locals 0
  
      .prologue
+     .line 45
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 73,80 ****
--- 76,85 ----
      .parameter "name"
  
      .prologue
+     .line 211
      invoke-static {}, Landroid/util/EventLog;->readTagsFile()V
  
+     .line 212
      sget-object v1, Landroid/util/EventLog;->sTagCodes:Ljava/util/HashMap;
  
      invoke-virtual {v1, p0}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 83,88 ****
--- 88,94 ----
  
      check-cast v0, Ljava/lang/Integer;
  
+     .line 213
      .local v0, code:Ljava/lang/Integer;
      if-eqz v0, :cond_0
  
***************
*** 104,111 ****
--- 110,119 ----
      .parameter "tag"
  
      .prologue
+     .line 201
      invoke-static {}, Landroid/util/EventLog;->readTagsFile()V
  
+     .line 202
      sget-object v0, Landroid/util/EventLog;->sTagNames:Ljava/util/HashMap;
  
      invoke-static {p0}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 143,148 ****
--- 151,157 ----
      .locals 13
  
      .prologue
+     .line 220
      const-class v10, Landroid/util/EventLog;
  
      monitor-enter v10
***************
*** 158,163 ****
--- 167,173 ----
  
      if-eqz v9, :cond_1
  
+     .line 256
      .local v0, comment:Ljava/util/regex/Pattern;
      .local v6, reader:Ljava/io/BufferedReader;
      .local v8, tag:Ljava/util/regex/Pattern;
***************
*** 167,172 ****
--- 177,183 ----
  
      return-void
  
+     .line 222
      .end local v0           #comment:Ljava/util/regex/Pattern;
      .end local v6           #reader:Ljava/io/BufferedReader;
      .end local v8           #tag:Ljava/util/regex/Pattern;
***************
*** 178,195 ****
--- 189,209 ----
  
      sput-object v9, Landroid/util/EventLog;->sTagCodes:Ljava/util/HashMap;
  
+     .line 223
      new-instance v9, Ljava/util/HashMap;
  
      invoke-direct {v9}, Ljava/util/HashMap;-><init>()V
  
      sput-object v9, Landroid/util/EventLog;->sTagNames:Ljava/util/HashMap;
  
+     .line 225
      const-string v9, "^\\s*(#.*)?$"
  
      invoke-static {v9}, Ljava/util/regex/Pattern;->compile(Ljava/lang/String;)Ljava/util/regex/Pattern;
  
      move-result-object v0
  
+     .line 226
      .restart local v0       #comment:Ljava/util/regex/Pattern;
      const-string v9, "^\\s*(\\d+)\\s+(\\w+)\\s*(\\(.*\\))?\\s*$"
  
***************
*** 199,207 ****
--- 213,223 ----
  
      move-result-object v8
  
+     .line 227
      .restart local v8       #tag:Ljava/util/regex/Pattern;
      const/4 v6, 0x0
  
+     .line 231
      .restart local v6       #reader:Ljava/io/BufferedReader;
      :try_start_2
      new-instance v7, Ljava/io/BufferedReader;
***************
*** 219,224 ****
--- 235,241 ----
      .catchall {:try_start_2 .. :try_end_2} :catchall_2
      .catch Ljava/io/IOException; {:try_start_2 .. :try_end_2} :catch_5
  
+     .line 232
      .end local v6           #reader:Ljava/io/BufferedReader;
      .local v7, reader:Ljava/io/BufferedReader;
      :cond_2
***************
*** 231,236 ****
--- 248,254 ----
      .local v2, line:Ljava/lang/String;
      if-eqz v2, :cond_5
  
+     .line 233
      invoke-virtual {v0, v2}, Ljava/util/regex/Pattern;->matcher(Ljava/lang/CharSequence;)Ljava/util/regex/Matcher;
  
      move-result-object v9
***************
*** 241,250 ****
--- 259,270 ----
  
      if-nez v9, :cond_2
  
+     .line 235
      invoke-virtual {v8, v2}, Ljava/util/regex/Pattern;->matcher(Ljava/lang/CharSequence;)Ljava/util/regex/Matcher;
  
      move-result-object v3
  
+     .line 236
      .local v3, m:Ljava/util/regex/Matcher;
      invoke-virtual {v3}, Ljava/util/regex/Matcher;->matches()Z
  
***************
*** 252,257 ****
--- 272,278 ----
  
      if-nez v9, :cond_3
  
+     .line 237
      const-string v9, "EventLog"
  
      new-instance v11, Ljava/lang/StringBuilder;
***************
*** 279,284 ****
--- 300,306 ----
  
      goto :goto_1
  
+     .line 250
      .end local v2           #line:Ljava/lang/String;
      .end local v3           #m:Ljava/util/regex/Matcher;
      :catch_0
***************
*** 286,291 ****
--- 308,314 ----
  
      move-object v6, v7
  
+     .line 251
      .end local v7           #reader:Ljava/io/BufferedReader;
      .local v1, e:Ljava/io/IOException;
      .restart local v6       #reader:Ljava/io/BufferedReader;
***************
*** 299,304 ****
--- 322,328 ----
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_2
  
+     .line 254
      if-eqz v6, :cond_0
  
      :try_start_5
***************
*** 314,319 ****
--- 338,344 ----
  
      goto :goto_0
  
+     .line 242
      .end local v1           #e:Ljava/io/IOException;
      .end local v6           #reader:Ljava/io/BufferedReader;
      .restart local v2       #line:Ljava/lang/String;
***************
*** 331,336 ****
--- 356,362 ----
  
      move-result v5
  
+     .line 243
      .local v5, num:I
      const/4 v9, 0x2
  
***************
*** 338,343 ****
--- 364,370 ----
  
      move-result-object v4
  
+     .line 244
      .local v4, name:Ljava/lang/String;
      sget-object v9, Landroid/util/EventLog;->sTagCodes:Ljava/util/HashMap;
  
***************
*** 347,352 ****
--- 374,380 ----
  
      invoke-virtual {v9, v4, v11}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 245
      sget-object v9, Landroid/util/EventLog;->sTagNames:Ljava/util/HashMap;
  
      invoke-static {v5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 361,371 ****
--- 389,401 ----
  
      goto :goto_1
  
+     .line 246
      .end local v4           #name:Ljava/lang/String;
      .end local v5           #num:I
      :catch_2
      move-exception v1
  
+     .line 247
      .local v1, e:Ljava/lang/NumberFormatException;
      :try_start_7
      const-string v9, "EventLog"
***************
*** 395,400 ****
--- 425,431 ----
  
      goto :goto_1
  
+     .line 254
      .end local v1           #e:Ljava/lang/NumberFormatException;
      .end local v2           #line:Ljava/lang/String;
      .end local v3           #m:Ljava/util/regex/Matcher;
***************
*** 421,426 ****
--- 452,458 ----
      :try_end_9
      .catchall {:try_start_9 .. :try_end_9} :catchall_1
  
+     .line 220
      .end local v6           #reader:Ljava/io/BufferedReader;
      .end local v8           #tag:Ljava/util/regex/Pattern;
      :catchall_1
***************
*** 430,435 ****
--- 462,468 ----
  
      throw v9
  
+     .line 254
      .restart local v2       #line:Ljava/lang/String;
      .restart local v7       #reader:Ljava/io/BufferedReader;
      .restart local v8       #tag:Ljava/util/regex/Pattern;
***************
*** 456,465 ****
--- 489,500 ----
  
      move-object v6, v7
  
+     .line 255
      .end local v7           #reader:Ljava/io/BufferedReader;
      .restart local v6       #reader:Ljava/io/BufferedReader;
      goto/16 :goto_0
  
+     .line 254
      .end local v2           #line:Ljava/lang/String;
      :catch_4
      move-exception v11
***************
*** 471,476 ****
--- 506,512 ----
  
      goto :goto_3
  
+     .line 250
      :catch_5
      move-exception v1
  
