*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/EventLog$Event.smali	2012-03-23 19:35:16.623890022 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/EventLog$Event.smali.original	2012-03-23 19:34:44.448303301 -0500
***************
*** 50,63 ****
--- 50,66 ----
      .parameter "data"
  
      .prologue
+     .line 77
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 78
      invoke-static {p1}, Ljava/nio/ByteBuffer;->wrap([B)Ljava/nio/ByteBuffer;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
+     .line 79
      iget-object v0, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
      invoke-static {}, Ljava/nio/ByteOrder;->nativeOrder()Ljava/nio/ByteOrder;
***************
*** 66,71 ****
--- 69,75 ----
  
      invoke-virtual {v0, v1}, Ljava/nio/ByteBuffer;->order(Ljava/nio/ByteOrder;)Ljava/nio/ByteBuffer;
  
+     .line 80
      return-void
  .end method
  
***************
*** 73,87 ****
--- 77,94 ----
      .locals 9
  
      .prologue
+     .line 120
      iget-object v6, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
      invoke-virtual {v6}, Ljava/nio/ByteBuffer;->get()B
  
      move-result v5
  
+     .line 121
      .local v5, type:B
      packed-switch v5, :pswitch_data_0
  
+     .line 147
      new-instance v6, Ljava/lang/IllegalArgumentException;
  
      new-instance v7, Ljava/lang/StringBuilder;
***************
*** 106,111 ****
--- 113,119 ----
  
      throw v6
  
+     .line 123
      :pswitch_0
      iget-object v6, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
***************
*** 117,126 ****
--- 125,136 ----
  
      move-result-object v0
  
+     .line 144
      :cond_0
      :goto_0
      return-object v0
  
+     .line 126
      :pswitch_1
      iget-object v6, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
***************
*** 134,139 ****
--- 144,150 ----
  
      goto :goto_0
  
+     .line 130
      :pswitch_2
      :try_start_0
      iget-object v6, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
***************
*** 142,147 ****
--- 153,159 ----
  
      move-result v3
  
+     .line 131
      .local v3, length:I
      iget-object v6, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
***************
*** 149,154 ****
--- 161,167 ----
  
      move-result v4
  
+     .line 132
      .local v4, start:I
      iget-object v6, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
***************
*** 156,161 ****
--- 169,175 ----
  
      invoke-virtual {v6, v7}, Ljava/nio/ByteBuffer;->position(I)Ljava/nio/Buffer;
  
+     .line 133
      new-instance v0, Ljava/lang/String;
  
      iget-object v6, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
***************
*** 172,182 ****
--- 186,198 ----
  
      goto :goto_0
  
+     .line 134
      .end local v3           #length:I
      .end local v4           #start:I
      :catch_0
      move-exception v1
  
+     .line 135
      .local v1, e:Ljava/io/UnsupportedEncodingException;
      const-string v6, "EventLog"
  
***************
*** 184,193 ****
--- 200,211 ----
  
      invoke-static {v6, v7, v1}, Landroid/util/Log;->wtf(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 136
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 140
      .end local v1           #e:Ljava/io/UnsupportedEncodingException;
      :pswitch_3
      iget-object v6, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
***************
*** 196,209 ****
--- 214,230 ----
  
      move-result v3
  
+     .line 141
      .restart local v3       #length:I
      if-gez v3, :cond_1
  
      add-int/lit16 v3, v3, 0x100
  
+     .line 142
      :cond_1
      new-array v0, v3, [Ljava/lang/Object;
  
+     .line 143
      .local v0, array:[Ljava/lang/Object;
      const/4 v2, 0x0
  
***************
*** 221,226 ****
--- 242,248 ----
  
      goto :goto_1
  
+     .line 121
      nop
  
      :pswitch_data_0
***************
*** 240,245 ****
--- 262,268 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 106
      monitor-enter p0
  
      :try_start_0
***************
*** 257,268 ****
--- 280,293 ----
  
      invoke-virtual {v2, v3}, Ljava/nio/ByteBuffer;->limit(I)Ljava/nio/Buffer;
  
+     .line 107
      iget-object v2, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
      const/16 v3, 0x18
  
      invoke-virtual {v2, v3}, Ljava/nio/ByteBuffer;->position(I)Ljava/nio/Buffer;
  
+     .line 108
      invoke-direct {p0}, Landroid/util/EventLog$Event;->decodeObject()Ljava/lang/Object;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 271,284 ****
--- 296,312 ----
  
      move-result-object v1
  
+     .line 114
      :goto_0
      monitor-exit p0
  
      return-object v1
  
+     .line 109
      :catch_0
      move-exception v0
  
+     .line 110
      .local v0, e:Ljava/lang/IllegalArgumentException;
      :try_start_1
      const-string v2, "EventLog"
***************
*** 311,316 ****
--- 339,345 ----
  
      goto :goto_0
  
+     .line 106
      .end local v0           #e:Ljava/lang/IllegalArgumentException;
      :catchall_0
      move-exception v1
***************
*** 319,327 ****
--- 348,358 ----
  
      throw v1
  
+     .line 112
      :catch_1
      move-exception v0
  
+     .line 113
      .local v0, e:Ljava/nio/BufferUnderflowException;
      :try_start_2
      const-string v2, "EventLog"
***************
*** 359,364 ****
--- 390,396 ----
      .locals 2
  
      .prologue
+     .line 84
      iget-object v0, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
      const/4 v1, 0x4
***************
*** 374,379 ****
--- 406,412 ----
      .locals 2
  
      .prologue
+     .line 100
      iget-object v0, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
      const/16 v1, 0x14
***************
*** 389,394 ****
--- 422,428 ----
      .locals 2
  
      .prologue
+     .line 89
      iget-object v0, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
      const/16 v1, 0x8
***************
*** 404,409 ****
--- 438,444 ----
      .locals 4
  
      .prologue
+     .line 94
      iget-object v0, p0, Landroid/util/EventLog$Event;->mBuffer:Ljava/nio/ByteBuffer;
  
      const/16 v1, 0xc
