*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/Singleton.smali	2012-03-23 19:35:17.458905248 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/Singleton.smali.original	2012-03-23 19:34:44.460303519 -0500
***************
*** 29,34 ****
--- 29,35 ----
      .locals 0
  
      .prologue
+     .line 26
      .local p0, this:Landroid/util/Singleton;,"Landroid/util/Singleton<TT;>;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
***************
*** 54,73 ****
--- 55,78 ----
      .end annotation
  
      .prologue
+     .line 32
      .local p0, this:Landroid/util/Singleton;,"Landroid/util/Singleton<TT;>;"
      monitor-enter p0
  
+     .line 33
      :try_start_0
      iget-object v0, p0, Landroid/util/Singleton;->mInstance:Ljava/lang/Object;
  
      if-nez v0, :cond_0
  
+     .line 34
      invoke-virtual {p0}, Landroid/util/Singleton;->create()Ljava/lang/Object;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/util/Singleton;->mInstance:Ljava/lang/Object;
  
+     .line 36
      :cond_0
      iget-object v0, p0, Landroid/util/Singleton;->mInstance:Ljava/lang/Object;
  
***************
*** 75,80 ****
--- 80,86 ----
  
      return-object v0
  
+     .line 37
      :catchall_0
      move-exception v0
  
