*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/TimingLogger.smali	2012-03-23 19:35:17.115898993 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/TimingLogger.smali.original	2012-03-23 19:34:44.455303429 -0500
***************
*** 40,49 ****
--- 40,52 ----
      .parameter "label"
  
      .prologue
+     .line 76
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 77
      invoke-virtual {p0, p1, p2}, Landroid/util/TimingLogger;->reset(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 78
      return-void
  .end method
  
***************
*** 54,71 ****
--- 57,78 ----
      .parameter "splitLabel"
  
      .prologue
+     .line 123
      iget-boolean v2, p0, Landroid/util/TimingLogger;->mDisabled:Z
  
      if-eqz v2, :cond_0
  
+     .line 127
      :goto_0
      return-void
  
+     .line 124
      :cond_0
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v0
  
+     .line 125
      .local v0, now:J
      iget-object v2, p0, Landroid/util/TimingLogger;->mSplits:Ljava/util/ArrayList;
  
***************
*** 75,80 ****
--- 82,88 ----
  
      invoke-virtual {v2, v3}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 126
      iget-object v2, p0, Landroid/util/TimingLogger;->mSplitLabels:Ljava/util/ArrayList;
  
      invoke-virtual {v2, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
***************
*** 86,98 ****
--- 94,109 ----
      .locals 12
  
      .prologue
+     .line 135
      iget-boolean v8, p0, Landroid/util/TimingLogger;->mDisabled:Z
  
      if-eqz v8, :cond_0
  
+     .line 147
      :goto_0
      return-void
  
+     .line 136
      :cond_0
      iget-object v8, p0, Landroid/util/TimingLogger;->mTag:Ljava/lang/String;
  
***************
*** 118,123 ****
--- 129,135 ----
  
      invoke-static {v8, v9}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 137
      iget-object v8, p0, Landroid/util/TimingLogger;->mSplits:Ljava/util/ArrayList;
  
      const/4 v9, 0x0
***************
*** 132,140 ****
--- 144,154 ----
  
      move-result-wide v0
  
+     .line 138
      .local v0, first:J
      move-wide v3, v0
  
+     .line 139
      .local v3, now:J
      const/4 v2, 0x1
  
***************
*** 148,153 ****
--- 162,168 ----
  
      if-ge v2, v8, :cond_1
  
+     .line 140
      iget-object v8, p0, Landroid/util/TimingLogger;->mSplits:Ljava/util/ArrayList;
  
      invoke-virtual {v8, v2}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 160,165 ****
--- 175,181 ----
  
      move-result-wide v3
  
+     .line 141
      iget-object v8, p0, Landroid/util/TimingLogger;->mSplitLabels:Ljava/util/ArrayList;
  
      invoke-virtual {v8, v2}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 168,173 ****
--- 184,190 ----
  
      check-cast v7, Ljava/lang/String;
  
+     .line 142
      .local v7, splitLabel:Ljava/lang/String;
      iget-object v8, p0, Landroid/util/TimingLogger;->mSplits:Ljava/util/ArrayList;
  
***************
*** 183,188 ****
--- 200,206 ----
  
      move-result-wide v5
  
+     .line 144
      .local v5, prev:J
      iget-object v8, p0, Landroid/util/TimingLogger;->mTag:Ljava/lang/String;
  
***************
*** 224,233 ****
--- 242,253 ----
  
      invoke-static {v8, v9}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 139
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_1
  
+     .line 146
      .end local v5           #prev:J
      .end local v7           #splitLabel:Ljava/lang/String;
      :cond_1
***************
*** 274,279 ****
--- 294,300 ----
      .locals 2
  
      .prologue
+     .line 103
      iget-object v0, p0, Landroid/util/TimingLogger;->mTag:Ljava/lang/String;
  
      const/4 v1, 0x2
***************
*** 289,323 ****
--- 310,351 ----
      :goto_0
      iput-boolean v0, p0, Landroid/util/TimingLogger;->mDisabled:Z
  
+     .line 104
      iget-boolean v0, p0, Landroid/util/TimingLogger;->mDisabled:Z
  
      if-eqz v0, :cond_1
  
+     .line 113
      :goto_1
      return-void
  
+     .line 103
      :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
  
+     .line 105
      :cond_1
      iget-object v0, p0, Landroid/util/TimingLogger;->mSplits:Ljava/util/ArrayList;
  
      if-nez v0, :cond_2
  
+     .line 106
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/util/TimingLogger;->mSplits:Ljava/util/ArrayList;
  
+     .line 107
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/util/TimingLogger;->mSplitLabels:Ljava/util/ArrayList;
  
+     .line 112
      :goto_2
      const/4 v0, 0x0
  
***************
*** 325,335 ****
--- 353,365 ----
  
      goto :goto_1
  
+     .line 109
      :cond_2
      iget-object v0, p0, Landroid/util/TimingLogger;->mSplits:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
  
+     .line 110
      iget-object v0, p0, Landroid/util/TimingLogger;->mSplitLabels:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->clear()V
***************
*** 343,353 ****
--- 373,387 ----
      .parameter "label"
  
      .prologue
+     .line 89
      iput-object p1, p0, Landroid/util/TimingLogger;->mTag:Ljava/lang/String;
  
+     .line 90
      iput-object p2, p0, Landroid/util/TimingLogger;->mLabel:Ljava/lang/String;
  
+     .line 91
      invoke-virtual {p0}, Landroid/util/TimingLogger;->reset()V
  
+     .line 92
      return-void
  .end method
