*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/TypedValue.smali	2012-03-23 19:35:16.600889603 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/util/TypedValue.smali.original	2012-03-23 19:34:44.447303283 -0500
***************
*** 118,134 ****
  
      const/4 v3, 0x0
  
      new-array v0, v6, [F
  
      fill-array-data v0, :array_0
  
      sput-object v0, Landroid/util/TypedValue;->RADIX_MULTS:[F
  
      const/4 v0, 0x6
  
      new-array v0, v0, [Ljava/lang/String;
  
!     const-string v1, "px"
  
      aput-object v1, v0, v3
  
--- 118,136 ----
  
      const/4 v3, 0x0
  
+     .line 191
      new-array v0, v6, [F
  
      fill-array-data v0, :array_0
  
      sput-object v0, Landroid/util/TypedValue;->RADIX_MULTS:[F
  
+     .line 417
      const/4 v0, 0x6
  
      new-array v0, v0, [Ljava/lang/String;
  
!     const-string/jumbo v1, "px"
  
      aput-object v1, v0, v3
  
***************
*** 136,148 ****
  
      aput-object v1, v0, v4
  
!     const-string v1, "sp"
  
      aput-object v1, v0, v5
  
      const/4 v1, 0x3
  
!     const-string v2, "pt"
  
      aput-object v2, v0, v1
  
--- 138,150 ----
  
      aput-object v1, v0, v4
  
!     const-string/jumbo v1, "sp"
  
      aput-object v1, v0, v5
  
      const/4 v1, 0x3
  
!     const-string/jumbo v2, "pt"
  
      aput-object v2, v0, v1
  
***************
*** 152,163 ****
  
      const/4 v1, 0x5
  
!     const-string v2, "mm"
  
      aput-object v2, v0, v1
  
      sput-object v0, Landroid/util/TypedValue;->DIMENSION_UNIT_STRS:[Ljava/lang/String;
  
      new-array v0, v5, [Ljava/lang/String;
  
      const-string v1, "%"
--- 154,166 ----
  
      const/4 v1, 0x5
  
!     const-string/jumbo v2, "mm"
  
      aput-object v2, v0, v1
  
      sput-object v0, Landroid/util/TypedValue;->DIMENSION_UNIT_STRS:[Ljava/lang/String;
  
+     .line 420
      new-array v0, v5, [Ljava/lang/String;
  
      const-string v1, "%"
***************
*** 172,177 ****
--- 175,181 ----
  
      return-void
  
+     .line 191
      nop
  
      :array_0
***************
*** 187,194 ****
--- 191,200 ----
      .locals 1
  
      .prologue
+     .line 23
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 174
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/util/TypedValue;->changingConfigurations:I
***************
*** 203,210 ****
--- 209,218 ----
      .parameter "metrics"
  
      .prologue
+     .line 323
      packed-switch p0, :pswitch_data_0
  
+     .line 337
      const/4 p1, 0x0
  
      .end local p1
***************
*** 212,217 ****
--- 220,226 ----
      :pswitch_0
      return p1
  
+     .line 327
      .restart local p1
      :pswitch_1
      iget v0, p2, Landroid/util/DisplayMetrics;->density:F
***************
*** 220,225 ****
--- 229,235 ----
  
      goto :goto_0
  
+     .line 329
      :pswitch_2
      iget v0, p2, Landroid/util/DisplayMetrics;->scaledDensity:F
  
***************
*** 227,232 ****
--- 237,243 ----
  
      goto :goto_0
  
+     .line 331
      :pswitch_3
      iget v0, p2, Landroid/util/DisplayMetrics;->xdpi:F
  
***************
*** 238,243 ****
--- 249,255 ----
  
      goto :goto_0
  
+     .line 333
      :pswitch_4
      iget v0, p2, Landroid/util/DisplayMetrics;->xdpi:F
  
***************
*** 245,250 ****
--- 257,263 ----
  
      goto :goto_0
  
+     .line 335
      :pswitch_5
      iget v0, p2, Landroid/util/DisplayMetrics;->xdpi:F
  
***************
*** 256,261 ****
--- 269,275 ----
  
      goto :goto_0
  
+     .line 323
      nop
  
      :pswitch_data_0
***************
*** 279,286 ****
--- 293,302 ----
  
      const/16 v2, 0x1f
  
+     .line 436
      packed-switch p0, :pswitch_data_0
  
+     .line 457
      :pswitch_0
      const/16 v1, 0x1c
  
***************
*** 288,293 ****
--- 304,310 ----
  
      if-gt p0, v2, :cond_2
  
+     .line 458
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 310,320 ****
--- 327,339 ----
  
      move-result-object v0
  
+     .line 463
      :cond_0
      :goto_0
      :pswitch_1
      return-object v0
  
+     .line 440
      :pswitch_2
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 336,341 ****
--- 355,361 ----
  
      goto :goto_0
  
+     .line 442
      :pswitch_3
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 357,362 ****
--- 377,383 ----
  
      goto :goto_0
  
+     .line 444
      :pswitch_4
      invoke-static {p1}, Ljava/lang/Float;->intBitsToFloat(I)F
  
***************
*** 368,373 ****
--- 389,395 ----
  
      goto :goto_0
  
+     .line 446
      :pswitch_5
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 403,408 ****
--- 425,431 ----
  
      goto :goto_0
  
+     .line 449
      :pswitch_6
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 442,447 ****
--- 465,471 ----
  
      goto :goto_0
  
+     .line 452
      :pswitch_7
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 467,476 ****
  
      goto/16 :goto_0
  
      :pswitch_8
      if-eqz p1, :cond_1
  
!     const-string v0, "true"
  
      goto/16 :goto_0
  
--- 491,501 ----
  
      goto/16 :goto_0
  
+     .line 454
      :pswitch_8
      if-eqz p1, :cond_1
  
!     const-string/jumbo v0, "true"
  
      goto/16 :goto_0
  
***************
*** 479,484 ****
--- 504,510 ----
  
      goto/16 :goto_0
  
+     .line 459
      :cond_2
      const/16 v1, 0x10
  
***************
*** 486,497 ****
--- 512,525 ----
  
      if-gt p0, v2, :cond_0
  
+     .line 460
      invoke-static {p1}, Ljava/lang/Integer;->toString(I)Ljava/lang/String;
  
      move-result-object v0
  
      goto/16 :goto_0
  
+     .line 436
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_1
***************
*** 522,527 ****
--- 550,556 ----
      .parameter "metrics"
  
      .prologue
+     .line 229
      shr-int/lit8 v0, p0, 0x0
  
      and-int/lit8 v0, v0, 0xf
***************
*** 543,552 ****
--- 572,583 ----
      .parameter "metrics"
  
      .prologue
+     .line 295
      invoke-static {p0, p1}, Landroid/util/TypedValue;->complexToDimension(ILandroid/util/DisplayMetrics;)F
  
      move-result v0
  
+     .line 296
      .local v0, res:F
      sget-object v1, Ljava/lang/System;->out:Ljava/io/PrintStream;
  
***************
*** 626,631 ****
--- 657,663 ----
  
      invoke-virtual {v1, v2}, Ljava/io/PrintStream;->println(Ljava/lang/String;)V
  
+     .line 304
      return v0
  .end method
  
***************
*** 635,640 ****
--- 667,673 ----
      .parameter "metrics"
  
      .prologue
+     .line 254
      shr-int/lit8 v0, p0, 0x0
  
      and-int/lit8 v0, v0, 0xf
***************
*** 660,669 ****
--- 693,704 ----
      .prologue
      const/4 v4, 0x0
  
+     .line 281
      invoke-static {p0}, Landroid/util/TypedValue;->complexToFloat(I)F
  
      move-result v2
  
+     .line 282
      .local v2, value:F
      shr-int/lit8 v3, p0, 0x0
  
***************
*** 673,678 ****
--- 708,714 ----
  
      move-result v0
  
+     .line 286
      .local v0, f:F
      const/high16 v3, 0x3f00
  
***************
*** 680,692 ****
--- 716,731 ----
  
      float-to-int v1, v3
  
+     .line 287
      .local v1, res:I
      if-eqz v1, :cond_0
  
+     .line 290
      .end local v1           #res:I
      :goto_0
      return v1
  
+     .line 288
      .restart local v1       #res:I
      :cond_0
      cmpl-float v3, v2, v4
***************
*** 697,702 ****
--- 736,742 ----
  
      goto :goto_0
  
+     .line 289
      :cond_1
      cmpl-float v3, v2, v4
  
***************
*** 706,711 ****
--- 746,752 ----
  
      goto :goto_0
  
+     .line 290
      :cond_2
      const/4 v1, -0x1
  
***************
*** 717,722 ****
--- 758,764 ----
      .parameter "complex"
  
      .prologue
+     .line 208
      and-int/lit16 v0, p0, -0x100
  
      int-to-float v0, v0
***************
*** 741,757 ****
--- 783,802 ----
      .parameter "pbase"
  
      .prologue
+     .line 373
      shr-int/lit8 v0, p0, 0x0
  
      and-int/lit8 v0, v0, 0xf
  
      packed-switch v0, :pswitch_data_0
  
+     .line 379
      const/4 v0, 0x0
  
      :goto_0
      return v0
  
+     .line 375
      :pswitch_0
      invoke-static {p0}, Landroid/util/TypedValue;->complexToFloat(I)F
  
***************
*** 761,766 ****
--- 806,812 ----
  
      goto :goto_0
  
+     .line 377
      :pswitch_1
      invoke-static {p0}, Landroid/util/TypedValue;->complexToFloat(I)F
  
***************
*** 770,775 ****
--- 816,822 ----
  
      goto :goto_0
  
+     .line 373
      nop
  
      :pswitch_data_0
***************
*** 785,799 ****
--- 832,850 ----
      .locals 2
  
      .prologue
+     .line 410
      iget v0, p0, Landroid/util/TypedValue;->type:I
  
+     .line 411
      .local v0, t:I
      const/4 v1, 0x3
  
      if-ne v0, v1, :cond_0
  
+     .line 412
      iget-object v1, p0, Landroid/util/TypedValue;->string:Ljava/lang/CharSequence;
  
+     .line 414
      :goto_0
      return-object v1
  
***************
*** 812,817 ****
--- 863,869 ----
      .parameter "metrics"
  
      .prologue
+     .line 352
      iget v0, p0, Landroid/util/TypedValue;->data:I
  
      invoke-static {v0, p1}, Landroid/util/TypedValue;->complexToDimension(ILandroid/util/DisplayMetrics;)F
***************
*** 825,830 ****
--- 877,883 ----
      .locals 1
  
      .prologue
+     .line 186
      iget v0, p0, Landroid/util/TypedValue;->data:I
  
      invoke-static {v0}, Ljava/lang/Float;->intBitsToFloat(I)F
***************
*** 840,845 ****
--- 893,899 ----
      .parameter "pbase"
  
      .prologue
+     .line 397
      iget v0, p0, Landroid/util/TypedValue;->data:I
  
      invoke-static {v0, p1, p2}, Landroid/util/TypedValue;->complexToFraction(IFF)F
***************
*** 854,883 ****
--- 908,944 ----
      .parameter "other"
  
      .prologue
+     .line 468
      iget v0, p1, Landroid/util/TypedValue;->type:I
  
      iput v0, p0, Landroid/util/TypedValue;->type:I
  
+     .line 469
      iget-object v0, p1, Landroid/util/TypedValue;->string:Ljava/lang/CharSequence;
  
      iput-object v0, p0, Landroid/util/TypedValue;->string:Ljava/lang/CharSequence;
  
+     .line 470
      iget v0, p1, Landroid/util/TypedValue;->data:I
  
      iput v0, p0, Landroid/util/TypedValue;->data:I
  
+     .line 471
      iget v0, p1, Landroid/util/TypedValue;->assetCookie:I
  
      iput v0, p0, Landroid/util/TypedValue;->assetCookie:I
  
+     .line 472
      iget v0, p1, Landroid/util/TypedValue;->resourceId:I
  
      iput v0, p0, Landroid/util/TypedValue;->resourceId:I
  
+     .line 473
      iget v0, p1, Landroid/util/TypedValue;->density:I
  
      iput v0, p0, Landroid/util/TypedValue;->density:I
  
+     .line 474
      return-void
  .end method
  
***************
*** 885,894 ****
--- 946,957 ----
      .locals 3
  
      .prologue
+     .line 478
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 479
      .local v0, sb:Ljava/lang/StringBuilder;
      const-string v1, "TypedValue{t=0x"
  
***************
*** 904,909 ****
--- 967,973 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 480
      const-string v1, "/d=0x"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 918,929 ****
--- 982,995 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 481
      iget v1, p0, Landroid/util/TypedValue;->type:I
  
      const/4 v2, 0x3
  
      if-ne v1, v2, :cond_0
  
+     .line 482
      const-string v1, " \""
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 945,955 ****
--- 1011,1023 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 484
      :cond_0
      iget v1, p0, Landroid/util/TypedValue;->assetCookie:I
  
      if-eqz v1, :cond_1
  
+     .line 485
      const-string v1, " a="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 960,970 ****
--- 1028,1040 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 487
      :cond_1
      iget v1, p0, Landroid/util/TypedValue;->resourceId:I
  
      if-eqz v1, :cond_2
  
+     .line 488
      const-string v1, " r=0x"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 979,995 ****
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      :cond_2
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
  
      return-object v1
  
      :cond_3
      const-string v1, "<null>"
  
--- 1049,1068 ----
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 490
      :cond_2
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 491
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
  
      return-object v1
  
+     .line 482
      :cond_3
      const-string v1, "<null>"
  
