*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/admin/DeviceAdminInfo$PolicyInfo.smali	2012-03-23 19:36:36.880353495 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/admin/DeviceAdminInfo$PolicyInfo.smali.original	2012-03-23 19:34:45.252317962 -0500
***************
*** 33,47 ****
--- 33,53 ----
      .parameter "descriptionIn"
  
      .prologue
+     .line 148
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 149
      iput p1, p0, Landroid/app/admin/DeviceAdminInfo$PolicyInfo;->ident:I
  
+     .line 150
      iput-object p2, p0, Landroid/app/admin/DeviceAdminInfo$PolicyInfo;->tag:Ljava/lang/String;
  
+     .line 151
      iput p3, p0, Landroid/app/admin/DeviceAdminInfo$PolicyInfo;->label:I
  
+     .line 152
      iput p4, p0, Landroid/app/admin/DeviceAdminInfo$PolicyInfo;->description:I
  
+     .line 153
      return-void
  .end method
