*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/admin/DeviceAdminReceiver.smali	2012-03-23 19:36:36.859353114 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/admin/DeviceAdminReceiver.smali.original	2012-03-23 19:34:45.252317962 -0500
***************
*** 38,47 ****
--- 38,49 ----
      .locals 1
  
      .prologue
+     .line 54
      const-string v0, "DevicePolicy"
  
      sput-object v0, Landroid/app/admin/DeviceAdminReceiver;->TAG:Ljava/lang/String;
  
+     .line 55
      const/4 v0, 0x0
  
      sput-boolean v0, Landroid/app/admin/DeviceAdminReceiver;->localLOGV:Z
***************
*** 53,58 ****
--- 55,61 ----
      .locals 0
  
      .prologue
+     .line 53
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
  
      return-void
***************
*** 65,79 ****
--- 68,86 ----
      .parameter "context"
  
      .prologue
+     .line 175
      iget-object v0, p0, Landroid/app/admin/DeviceAdminReceiver;->mManager:Landroid/app/admin/DevicePolicyManager;
  
      if-eqz v0, :cond_0
  
+     .line 176
      iget-object v0, p0, Landroid/app/admin/DeviceAdminReceiver;->mManager:Landroid/app/admin/DevicePolicyManager;
  
+     .line 180
      :goto_0
      return-object v0
  
+     .line 178
      :cond_0
      const-string v0, "device_policy"
  
***************
*** 85,90 ****
--- 92,98 ----
  
      iput-object v0, p0, Landroid/app/admin/DeviceAdminReceiver;->mManager:Landroid/app/admin/DevicePolicyManager;
  
+     .line 180
      iget-object v0, p0, Landroid/app/admin/DeviceAdminReceiver;->mManager:Landroid/app/admin/DevicePolicyManager;
  
      goto :goto_0
***************
*** 95,109 ****
--- 103,121 ----
      .parameter "context"
  
      .prologue
+     .line 189
      iget-object v0, p0, Landroid/app/admin/DeviceAdminReceiver;->mWho:Landroid/content/ComponentName;
  
      if-eqz v0, :cond_0
  
+     .line 190
      iget-object v0, p0, Landroid/app/admin/DeviceAdminReceiver;->mWho:Landroid/content/ComponentName;
  
+     .line 193
      :goto_0
      return-object v0
  
+     .line 192
      :cond_0
      new-instance v0, Landroid/content/ComponentName;
  
***************
*** 115,120 ****
--- 127,133 ----
  
      iput-object v0, p0, Landroid/app/admin/DeviceAdminReceiver;->mWho:Landroid/content/ComponentName;
  
+     .line 193
      iget-object v0, p0, Landroid/app/admin/DeviceAdminReceiver;->mWho:Landroid/content/ComponentName;
  
      goto :goto_0
***************
*** 126,131 ****
--- 139,145 ----
      .parameter "intent"
  
      .prologue
+     .line 218
      const/4 v0, 0x0
  
      return-object v0
***************
*** 137,142 ****
--- 151,157 ----
      .parameter "intent"
  
      .prologue
+     .line 230
      return-void
  .end method
  
***************
*** 146,151 ****
--- 161,167 ----
      .parameter "intent"
  
      .prologue
+     .line 204
      return-void
  .end method
  
***************
*** 155,160 ****
--- 171,177 ----
      .parameter "intent"
  
      .prologue
+     .line 242
      return-void
  .end method
  
***************
*** 164,169 ****
--- 181,187 ----
      .parameter "intent"
  
      .prologue
+     .line 285
      return-void
  .end method
  
***************
*** 173,178 ****
--- 191,197 ----
      .parameter "intent"
  
      .prologue
+     .line 253
      return-void
  .end method
  
***************
*** 182,187 ****
--- 201,207 ----
      .parameter "intent"
  
      .prologue
+     .line 264
      return-void
  .end method
  
***************
*** 191,200 ****
--- 211,222 ----
      .parameter "intent"
  
      .prologue
+     .line 294
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 295
      .local v0, action:Ljava/lang/String;
      const-string v3, "android.app.action.ACTION_PASSWORD_CHANGED"
  
***************
*** 204,215 ****
--- 226,240 ----
  
      if-eqz v3, :cond_1
  
+     .line 296
      invoke-virtual {p0, p1, p2}, Landroid/app/admin/DeviceAdminReceiver;->onPasswordChanged(Landroid/content/Context;Landroid/content/Intent;)V
  
+     .line 314
      :cond_0
      :goto_0
      return-void
  
+     .line 297
      :cond_1
      const-string v3, "android.app.action.ACTION_PASSWORD_FAILED"
  
***************
*** 219,228 ****
--- 244,255 ----
  
      if-eqz v3, :cond_2
  
+     .line 298
      invoke-virtual {p0, p1, p2}, Landroid/app/admin/DeviceAdminReceiver;->onPasswordFailed(Landroid/content/Context;Landroid/content/Intent;)V
  
      goto :goto_0
  
+     .line 299
      :cond_2
      const-string v3, "android.app.action.ACTION_PASSWORD_SUCCEEDED"
  
***************
*** 232,241 ****
--- 259,270 ----
  
      if-eqz v3, :cond_3
  
+     .line 300
      invoke-virtual {p0, p1, p2}, Landroid/app/admin/DeviceAdminReceiver;->onPasswordSucceeded(Landroid/content/Context;Landroid/content/Intent;)V
  
      goto :goto_0
  
+     .line 301
      :cond_3
      const-string v3, "android.app.action.DEVICE_ADMIN_ENABLED"
  
***************
*** 245,254 ****
--- 274,285 ----
  
      if-eqz v3, :cond_4
  
+     .line 302
      invoke-virtual {p0, p1, p2}, Landroid/app/admin/DeviceAdminReceiver;->onEnabled(Landroid/content/Context;Landroid/content/Intent;)V
  
      goto :goto_0
  
+     .line 303
      :cond_4
      const-string v3, "android.app.action.DEVICE_ADMIN_DISABLE_REQUESTED"
  
***************
*** 258,276 ****
--- 289,311 ----
  
      if-eqz v3, :cond_5
  
+     .line 304
      invoke-virtual {p0, p1, p2}, Landroid/app/admin/DeviceAdminReceiver;->onDisableRequested(Landroid/content/Context;Landroid/content/Intent;)Ljava/lang/CharSequence;
  
      move-result-object v2
  
+     .line 305
      .local v2, res:Ljava/lang/CharSequence;
      if-eqz v2, :cond_0
  
+     .line 306
      const/4 v3, 0x1
  
      invoke-virtual {p0, v3}, Landroid/app/admin/DeviceAdminReceiver;->getResultExtras(Z)Landroid/os/Bundle;
  
      move-result-object v1
  
+     .line 307
      .local v1, extras:Landroid/os/Bundle;
      const-string v3, "android.app.extra.DISABLE_WARNING"
  
***************
*** 278,283 ****
--- 313,319 ----
  
      goto :goto_0
  
+     .line 309
      .end local v1           #extras:Landroid/os/Bundle;
      .end local v2           #res:Ljava/lang/CharSequence;
      :cond_5
***************
*** 289,298 ****
--- 325,336 ----
  
      if-eqz v3, :cond_6
  
+     .line 310
      invoke-virtual {p0, p1, p2}, Landroid/app/admin/DeviceAdminReceiver;->onDisabled(Landroid/content/Context;Landroid/content/Intent;)V
  
      goto :goto_0
  
+     .line 311
      :cond_6
      const-string v3, "android.app.action.ACTION_PASSWORD_EXPIRING"
  
***************
*** 302,307 ****
--- 340,346 ----
  
      if-eqz v3, :cond_0
  
+     .line 312
      invoke-virtual {p0, p1, p2}, Landroid/app/admin/DeviceAdminReceiver;->onPasswordExpiring(Landroid/content/Context;Landroid/content/Intent;)V
  
      goto :goto_0
