*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Double4.smali	2012-03-23 19:35:33.008188789 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Double4.smali.original	2012-03-23 19:34:44.706308006 -0500
***************
*** 18,25 ****
--- 18,27 ----
      .locals 0
  
      .prologue
+     .line 29
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 30
      return-void
  .end method
  
***************
*** 31,45 ****
--- 33,53 ----
      .parameter "initW"
  
      .prologue
+     .line 32
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 33
      iput-wide p1, p0, Landroid/renderscript/Double4;->x:D
  
+     .line 34
      iput-wide p3, p0, Landroid/renderscript/Double4;->y:D
  
+     .line 35
      iput-wide p5, p0, Landroid/renderscript/Double4;->z:D
  
+     .line 36
      iput-wide p7, p0, Landroid/renderscript/Double4;->w:D
  
+     .line 37
      return-void
  .end method
