*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Element$DataType.smali	2012-03-23 19:35:32.951187750 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Element$DataType.smali.original	2012-03-23 19:34:44.705307988 -0500
***************
*** 102,107 ****
--- 102,108 ----
  
      const/4 v5, 0x4
  
+     .line 79
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "FLOAT_32"
***************
*** 112,117 ****
--- 113,119 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->FLOAT_32:Landroid/renderscript/Element$DataType;
  
+     .line 80
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "FLOAT_64"
***************
*** 122,127 ****
--- 124,130 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->FLOAT_64:Landroid/renderscript/Element$DataType;
  
+     .line 81
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "SIGNED_8"
***************
*** 130,135 ****
--- 133,139 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->SIGNED_8:Landroid/renderscript/Element$DataType;
  
+     .line 82
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "SIGNED_16"
***************
*** 142,147 ****
--- 146,152 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->SIGNED_16:Landroid/renderscript/Element$DataType;
  
+     .line 83
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "SIGNED_32"
***************
*** 152,157 ****
--- 157,163 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->SIGNED_32:Landroid/renderscript/Element$DataType;
  
+     .line 84
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "SIGNED_64"
***************
*** 164,169 ****
--- 170,176 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->SIGNED_64:Landroid/renderscript/Element$DataType;
  
+     .line 85
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "UNSIGNED_8"
***************
*** 174,179 ****
--- 181,187 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->UNSIGNED_8:Landroid/renderscript/Element$DataType;
  
+     .line 86
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "UNSIGNED_16"
***************
*** 186,191 ****
--- 194,200 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->UNSIGNED_16:Landroid/renderscript/Element$DataType;
  
+     .line 87
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "UNSIGNED_32"
***************
*** 196,201 ****
--- 205,211 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->UNSIGNED_32:Landroid/renderscript/Element$DataType;
  
+     .line 88
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "UNSIGNED_64"
***************
*** 208,213 ****
--- 218,224 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->UNSIGNED_64:Landroid/renderscript/Element$DataType;
  
+     .line 90
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "BOOLEAN"
***************
*** 220,225 ****
--- 231,237 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->BOOLEAN:Landroid/renderscript/Element$DataType;
  
+     .line 92
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "UNSIGNED_5_6_5"
***************
*** 232,237 ****
--- 244,250 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->UNSIGNED_5_6_5:Landroid/renderscript/Element$DataType;
  
+     .line 93
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "UNSIGNED_5_5_5_1"
***************
*** 244,249 ****
--- 257,263 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->UNSIGNED_5_5_5_1:Landroid/renderscript/Element$DataType;
  
+     .line 94
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "UNSIGNED_4_4_4_4"
***************
*** 256,261 ****
--- 270,276 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->UNSIGNED_4_4_4_4:Landroid/renderscript/Element$DataType;
  
+     .line 96
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "MATRIX_4X4"
***************
*** 268,273 ****
--- 283,289 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->MATRIX_4X4:Landroid/renderscript/Element$DataType;
  
+     .line 97
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "MATRIX_3X3"
***************
*** 282,287 ****
--- 298,304 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->MATRIX_3X3:Landroid/renderscript/Element$DataType;
  
+     .line 98
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "MATRIX_2X2"
***************
*** 292,297 ****
--- 309,315 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->MATRIX_2X2:Landroid/renderscript/Element$DataType;
  
+     .line 100
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_ELEMENT"
***************
*** 304,309 ****
--- 322,328 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_ELEMENT:Landroid/renderscript/Element$DataType;
  
+     .line 101
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_TYPE"
***************
*** 316,321 ****
--- 335,341 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_TYPE:Landroid/renderscript/Element$DataType;
  
+     .line 102
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_ALLOCATION"
***************
*** 328,333 ****
--- 348,354 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_ALLOCATION:Landroid/renderscript/Element$DataType;
  
+     .line 103
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_SAMPLER"
***************
*** 340,345 ****
--- 361,367 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_SAMPLER:Landroid/renderscript/Element$DataType;
  
+     .line 104
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_SCRIPT"
***************
*** 352,357 ****
--- 374,380 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_SCRIPT:Landroid/renderscript/Element$DataType;
  
+     .line 105
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_MESH"
***************
*** 364,369 ****
--- 387,393 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_MESH:Landroid/renderscript/Element$DataType;
  
+     .line 106
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_PROGRAM_FRAGMENT"
***************
*** 376,381 ****
--- 400,406 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_PROGRAM_FRAGMENT:Landroid/renderscript/Element$DataType;
  
+     .line 107
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_PROGRAM_VERTEX"
***************
*** 388,393 ****
--- 413,419 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_PROGRAM_VERTEX:Landroid/renderscript/Element$DataType;
  
+     .line 108
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_PROGRAM_RASTER"
***************
*** 400,405 ****
--- 426,432 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_PROGRAM_RASTER:Landroid/renderscript/Element$DataType;
  
+     .line 109
      new-instance v0, Landroid/renderscript/Element$DataType;
  
      const-string v1, "RS_PROGRAM_STORE"
***************
*** 412,417 ****
--- 439,445 ----
  
      sput-object v0, Landroid/renderscript/Element$DataType;->RS_PROGRAM_STORE:Landroid/renderscript/Element$DataType;
  
+     .line 77
      const/16 v0, 0x1b
  
      new-array v0, v0, [Landroid/renderscript/Element$DataType;
***************
*** 586,597 ****
--- 614,629 ----
      .end annotation
  
      .prologue
+     .line 113
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 114
      iput p3, p0, Landroid/renderscript/Element$DataType;->mID:I
  
+     .line 115
      iput p4, p0, Landroid/renderscript/Element$DataType;->mSize:I
  
+     .line 116
      return-void
  .end method
  
***************
*** 600,605 ****
--- 632,638 ----
      .parameter "name"
  
      .prologue
+     .line 77
      const-class v0, Landroid/renderscript/Element$DataType;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 615,620 ****
--- 648,654 ----
      .locals 1
  
      .prologue
+     .line 77
      sget-object v0, Landroid/renderscript/Element$DataType;->$VALUES:[Landroid/renderscript/Element$DataType;
  
      invoke-virtual {v0}, [Landroid/renderscript/Element$DataType;->clone()Ljava/lang/Object;
