*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/FileA3D$1.smali	2012-03-23 19:35:33.474197286 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/FileA3D$1.smali.original	2012-03-23 19:34:44.710308078 -0500
***************
*** 23,28 ****
--- 23,29 ----
      .locals 3
  
      .prologue
+     .line 139
      invoke-static {}, Landroid/renderscript/FileA3D$EntryType;->values()[Landroid/renderscript/FileA3D$EntryType;
  
      move-result-object v0
