*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/FileA3D$IndexEntry.smali	2012-03-23 19:35:31.438160159 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/FileA3D$IndexEntry.smali.original	2012-03-23 19:34:44.693307768 -0500
***************
*** 38,59 ****
--- 38,67 ----
      .parameter "type"
  
      .prologue
+     .line 149
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 150
      iput-object p1, p0, Landroid/renderscript/FileA3D$IndexEntry;->mRS:Landroid/renderscript/RenderScript;
  
+     .line 151
      iput p2, p0, Landroid/renderscript/FileA3D$IndexEntry;->mIndex:I
  
+     .line 152
      iput p3, p0, Landroid/renderscript/FileA3D$IndexEntry;->mID:I
  
+     .line 153
      iput-object p4, p0, Landroid/renderscript/FileA3D$IndexEntry;->mName:Ljava/lang/String;
  
+     .line 154
      iput-object p5, p0, Landroid/renderscript/FileA3D$IndexEntry;->mEntryType:Landroid/renderscript/FileA3D$EntryType;
  
+     .line 155
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/renderscript/FileA3D$IndexEntry;->mLoadedObj:Landroid/renderscript/BaseObj;
  
+     .line 156
      return-void
  .end method
  
***************
*** 65,70 ****
--- 73,79 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 125
      const-class v2, Landroid/renderscript/FileA3D$IndexEntry;
  
      monitor-enter v2
***************
*** 74,89 ****
--- 83,101 ----
  
      if-eqz v3, :cond_1
  
+     .line 126
      iget-object v1, p1, Landroid/renderscript/FileA3D$IndexEntry;->mLoadedObj:Landroid/renderscript/BaseObj;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 146
      :cond_0
      :goto_0
      monitor-exit v2
  
      return-object v1
  
+     .line 130
      :cond_1
      :try_start_1
      iget-object v3, p1, Landroid/renderscript/FileA3D$IndexEntry;->mEntryType:Landroid/renderscript/FileA3D$EntryType;
***************
*** 92,97 ****
--- 104,110 ----
  
      if-eq v3, v4, :cond_0
  
+     .line 134
      iget v3, p1, Landroid/renderscript/FileA3D$IndexEntry;->mID:I
  
      iget v4, p1, Landroid/renderscript/FileA3D$IndexEntry;->mIndex:I
***************
*** 100,108 ****
--- 113,123 ----
  
      move-result v0
  
+     .line 135
      .local v0, objectID:I
      if-eqz v0, :cond_0
  
+     .line 139
      sget-object v1, Landroid/renderscript/FileA3D$1;->$SwitchMap$android$renderscript$FileA3D$EntryType:[I
  
      iget-object v3, p1, Landroid/renderscript/FileA3D$IndexEntry;->mEntryType:Landroid/renderscript/FileA3D$EntryType;
***************
*** 115,129 ****
--- 130,147 ----
  
      packed-switch v1, :pswitch_data_0
  
+     .line 145
      :goto_1
      iget-object v1, p1, Landroid/renderscript/FileA3D$IndexEntry;->mLoadedObj:Landroid/renderscript/BaseObj;
  
      invoke-virtual {v1}, Landroid/renderscript/BaseObj;->updateFromNative()V
  
+     .line 146
      iget-object v1, p1, Landroid/renderscript/FileA3D$IndexEntry;->mLoadedObj:Landroid/renderscript/BaseObj;
  
      goto :goto_0
  
+     .line 141
      :pswitch_0
      new-instance v1, Landroid/renderscript/Mesh;
  
***************
*** 135,140 ****
--- 153,159 ----
  
      goto :goto_1
  
+     .line 125
      .end local v0           #objectID:I
      :catchall_0
      move-exception v1
***************
*** 143,148 ****
--- 162,168 ----
  
      throw v1
  
+     .line 139
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 155,160 ****
--- 175,181 ----
      .locals 1
  
      .prologue
+     .line 101
      iget-object v0, p0, Landroid/renderscript/FileA3D$IndexEntry;->mEntryType:Landroid/renderscript/FileA3D$EntryType;
  
      return-object v0
***************
*** 164,169 ****
--- 185,191 ----
      .locals 1
  
      .prologue
+     .line 121
      invoke-virtual {p0}, Landroid/renderscript/FileA3D$IndexEntry;->getObject()Landroid/renderscript/BaseObj;
  
      move-result-object v0
***************
*** 177,182 ****
--- 199,205 ----
      .locals 1
  
      .prologue
+     .line 91
      iget-object v0, p0, Landroid/renderscript/FileA3D$IndexEntry;->mName:Ljava/lang/String;
  
      return-object v0
***************
*** 186,201 ****
--- 209,227 ----
      .locals 2
  
      .prologue
+     .line 109
      iget-object v1, p0, Landroid/renderscript/FileA3D$IndexEntry;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-virtual {v1}, Landroid/renderscript/RenderScript;->validate()V
  
+     .line 110
      iget-object v1, p0, Landroid/renderscript/FileA3D$IndexEntry;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-static {v1, p0}, Landroid/renderscript/FileA3D$IndexEntry;->internalCreate(Landroid/renderscript/RenderScript;Landroid/renderscript/FileA3D$IndexEntry;)Landroid/renderscript/BaseObj;
  
      move-result-object v0
  
+     .line 111
      .local v0, obj:Landroid/renderscript/BaseObj;
      return-object v0
  .end method
