*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Font$FontFamily.smali	2012-03-23 19:35:32.207174182 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Font$FontFamily.smali.original	2012-03-23 19:34:44.700307896 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .locals 0
  
      .prologue
+     .line 64
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "x0"
  
      .prologue
+     .line 64
      invoke-direct {p0}, Landroid/renderscript/Font$FontFamily;-><init>()V
  
      return-void
