*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Mesh$TriangleMeshBuilder.smali	2012-03-23 19:35:32.289175678 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Mesh$TriangleMeshBuilder.smali.original	2012-03-23 19:34:44.701307914 -0500
***************
*** 74,119 ****
--- 74,137 ----
  
      const/4 v1, 0x0
  
+     .line 548
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 522
      iput v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mNX:F
  
+     .line 523
      iput v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mNY:F
  
+     .line 524
      const/high16 v0, -0x4080
  
      iput v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mNZ:F
  
+     .line 525
      iput v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mS0:F
  
+     .line 526
      iput v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mT0:F
  
+     .line 527
      iput v2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mR:F
  
+     .line 528
      iput v2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mG:F
  
+     .line 529
      iput v2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mB:F
  
+     .line 530
      iput v2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mA:F
  
+     .line 549
      iput-object p1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
+     .line 550
      iput v3, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
  
+     .line 551
      iput v3, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexCount:I
  
+     .line 552
      new-array v0, v4, [F
  
      iput-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
+     .line 553
      new-array v0, v4, [S
  
      iput-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexData:[S
  
+     .line 554
      iput p2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxSize:I
  
+     .line 555
      iput p3, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
+     .line 557
      const/4 v0, 0x2
  
      if-lt p2, v0, :cond_0
***************
*** 122,127 ****
--- 140,146 ----
  
      if-le p2, v0, :cond_1
  
+     .line 558
      :cond_0
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
***************
*** 131,136 ****
--- 150,156 ----
  
      throw v0
  
+     .line 560
      :cond_1
      return-void
  .end method
***************
*** 139,154 ****
--- 159,177 ----
      .locals 3
  
      .prologue
+     .line 571
      iget v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
      and-int/lit8 v0, v0, 0x1
  
      if-eqz v0, :cond_0
  
+     .line 572
      const/4 v0, 0x4
  
      invoke-direct {p0, v0}, Landroid/renderscript/Mesh$TriangleMeshBuilder;->makeSpace(I)V
  
+     .line 573
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 161,166 ****
--- 184,190 ----
  
      aput v2, v0, v1
  
+     .line 574
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 173,178 ****
--- 197,203 ----
  
      aput v2, v0, v1
  
+     .line 575
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 185,190 ****
--- 210,216 ----
  
      aput v2, v0, v1
  
+     .line 576
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 197,202 ****
--- 223,229 ----
  
      aput v2, v0, v1
  
+     .line 578
      :cond_0
      iget v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
***************
*** 204,213 ****
--- 231,242 ----
  
      if-eqz v0, :cond_1
  
+     .line 579
      const/4 v0, 0x2
  
      invoke-direct {p0, v0}, Landroid/renderscript/Mesh$TriangleMeshBuilder;->makeSpace(I)V
  
+     .line 580
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 220,225 ****
--- 249,255 ----
  
      aput v2, v0, v1
  
+     .line 581
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 232,237 ****
--- 262,268 ----
  
      aput v2, v0, v1
  
+     .line 583
      :cond_1
      iget v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
***************
*** 239,248 ****
--- 270,281 ----
  
      if-eqz v0, :cond_2
  
+     .line 584
      const/4 v0, 0x3
  
      invoke-direct {p0, v0}, Landroid/renderscript/Mesh$TriangleMeshBuilder;->makeSpace(I)V
  
+     .line 585
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 255,260 ****
--- 288,294 ----
  
      aput v2, v0, v1
  
+     .line 586
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 267,272 ****
--- 301,307 ----
  
      aput v2, v0, v1
  
+     .line 587
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 279,284 ****
--- 314,320 ----
  
      aput v2, v0, v1
  
+     .line 589
      :cond_2
      return-void
  .end method
***************
*** 290,295 ****
--- 326,332 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 563
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
  
      add-int/2addr v1, p1
***************
*** 300,305 ****
--- 337,343 ----
  
      if-lt v1, v2, :cond_0
  
+     .line 564
      iget-object v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      array-length v1, v1
***************
*** 308,313 ****
--- 346,352 ----
  
      new-array v0, v1, [F
  
+     .line 565
      .local v0, t:[F
      iget-object v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
***************
*** 317,324 ****
--- 356,365 ----
  
      invoke-static {v1, v3, v0, v3, v2}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 566
      iput-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
+     .line 568
      .end local v0           #t:[F
      :cond_0
      return-void
***************
*** 335,340 ****
--- 376,382 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 700
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
  
      if-ge p1, v1, :cond_0
***************
*** 353,358 ****
--- 395,401 ----
  
      if-gez p3, :cond_1
  
+     .line 703
      :cond_0
      new-instance v1, Ljava/lang/IllegalStateException;
  
***************
*** 362,367 ****
--- 405,411 ----
  
      throw v1
  
+     .line 705
      :cond_1
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexCount:I
  
***************
*** 373,378 ****
--- 417,423 ----
  
      if-lt v1, v2, :cond_2
  
+     .line 706
      iget-object v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexData:[S
  
      array-length v1, v1
***************
*** 381,386 ****
--- 426,432 ----
  
      new-array v0, v1, [S
  
+     .line 707
      .local v0, t:[S
      iget-object v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexData:[S
  
***************
*** 390,397 ****
--- 436,445 ----
  
      invoke-static {v1, v3, v0, v3, v2}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 708
      iput-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexData:[S
  
+     .line 710
      .end local v0           #t:[S
      :cond_2
      iget-object v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexData:[S
***************
*** 406,411 ****
--- 454,460 ----
  
      aput-short v3, v1, v2
  
+     .line 711
      iget-object v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexData:[S
  
      iget v2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexCount:I
***************
*** 418,423 ****
--- 467,473 ----
  
      aput-short v3, v1, v2
  
+     .line 712
      iget-object v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexData:[S
  
      iget v2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mIndexCount:I
***************
*** 430,435 ****
--- 480,486 ----
  
      aput-short v3, v1, v2
  
+     .line 713
      return-object p0
  .end method
  
***************
*** 441,450 ****
--- 492,503 ----
      .prologue
      const/4 v1, 0x2
  
+     .line 601
      iget v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxSize:I
  
      if-eq v0, v1, :cond_0
  
+     .line 602
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "add mistmatch with declared components."
***************
*** 453,461 ****
--- 506,516 ----
  
      throw v0
  
+     .line 604
      :cond_0
      invoke-direct {p0, v1}, Landroid/renderscript/Mesh$TriangleMeshBuilder;->makeSpace(I)V
  
+     .line 605
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 466,471 ****
--- 521,527 ----
  
      aput p1, v0, v1
  
+     .line 606
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 476,483 ****
--- 532,541 ----
  
      aput p2, v0, v1
  
+     .line 607
      invoke-direct {p0}, Landroid/renderscript/Mesh$TriangleMeshBuilder;->latch()V
  
+     .line 608
      return-object p0
  .end method
  
***************
*** 490,499 ****
--- 548,559 ----
      .prologue
      const/4 v1, 0x3
  
+     .line 622
      iget v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxSize:I
  
      if-eq v0, v1, :cond_0
  
+     .line 623
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "add mistmatch with declared components."
***************
*** 502,510 ****
--- 562,572 ----
  
      throw v0
  
+     .line 625
      :cond_0
      invoke-direct {p0, v1}, Landroid/renderscript/Mesh$TriangleMeshBuilder;->makeSpace(I)V
  
+     .line 626
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 515,520 ****
--- 577,583 ----
  
      aput p1, v0, v1
  
+     .line 627
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 525,530 ****
--- 588,594 ----
  
      aput p2, v0, v1
  
+     .line 628
      iget-object v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxData:[F
  
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxCount:I
***************
*** 535,542 ****
--- 599,608 ----
  
      aput p3, v0, v1
  
+     .line 629
      invoke-direct {p0}, Landroid/renderscript/Mesh$TriangleMeshBuilder;->latch()V
  
+     .line 630
      return-object p0
  .end method
  
***************
*** 549,563 ****
--- 615,632 ----
  
      const/4 v8, 0x0
  
+     .line 731
      new-instance v0, Landroid/renderscript/Element$Builder;
  
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-direct {v0, v5}, Landroid/renderscript/Element$Builder;-><init>(Landroid/renderscript/RenderScript;)V
  
+     .line 732
      .local v0, b:Landroid/renderscript/Element$Builder;
      iget v1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mVtxSize:I
  
+     .line 733
      .local v1, floatCount:I
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
***************
*** 569,586 ****
  
      move-result-object v5
  
!     const-string v6, "position"
  
      invoke-virtual {v0, v5, v6}, Landroid/renderscript/Element$Builder;->add(Landroid/renderscript/Element;Ljava/lang/String;)Landroid/renderscript/Element$Builder;
  
      iget v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
      and-int/lit8 v5, v5, 0x1
  
      if-eqz v5, :cond_0
  
      add-int/lit8 v1, v1, 0x4
  
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-static {v5}, Landroid/renderscript/Element;->F32_4(Landroid/renderscript/RenderScript;)Landroid/renderscript/Element;
--- 638,658 ----
  
      move-result-object v5
  
!     const-string/jumbo v6, "position"
  
      invoke-virtual {v0, v5, v6}, Landroid/renderscript/Element$Builder;->add(Landroid/renderscript/Element;Ljava/lang/String;)Landroid/renderscript/Element$Builder;
  
+     .line 736
      iget v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
      and-int/lit8 v5, v5, 0x1
  
      if-eqz v5, :cond_0
  
+     .line 737
      add-int/lit8 v1, v1, 0x4
  
+     .line 738
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-static {v5}, Landroid/renderscript/Element;->F32_4(Landroid/renderscript/RenderScript;)Landroid/renderscript/Element;
***************
*** 591,596 ****
--- 663,669 ----
  
      invoke-virtual {v0, v5, v6}, Landroid/renderscript/Element$Builder;->add(Landroid/renderscript/Element;Ljava/lang/String;)Landroid/renderscript/Element$Builder;
  
+     .line 740
      :cond_0
      iget v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
***************
*** 598,615 ****
  
      if-eqz v5, :cond_1
  
      add-int/lit8 v1, v1, 0x2
  
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-static {v5}, Landroid/renderscript/Element;->F32_2(Landroid/renderscript/RenderScript;)Landroid/renderscript/Element;
  
      move-result-object v5
  
!     const-string v6, "texture0"
  
      invoke-virtual {v0, v5, v6}, Landroid/renderscript/Element$Builder;->add(Landroid/renderscript/Element;Ljava/lang/String;)Landroid/renderscript/Element$Builder;
  
      :cond_1
      iget v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
--- 671,691 ----
  
      if-eqz v5, :cond_1
  
+     .line 741
      add-int/lit8 v1, v1, 0x2
  
+     .line 742
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-static {v5}, Landroid/renderscript/Element;->F32_2(Landroid/renderscript/RenderScript;)Landroid/renderscript/Element;
  
      move-result-object v5
  
!     const-string/jumbo v6, "texture0"
  
      invoke-virtual {v0, v5, v6}, Landroid/renderscript/Element$Builder;->add(Landroid/renderscript/Element;Ljava/lang/String;)Landroid/renderscript/Element$Builder;
  
+     .line 744
      :cond_1
      iget v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
***************
*** 617,634 ****
  
      if-eqz v5, :cond_2
  
      add-int/lit8 v1, v1, 0x3
  
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-static {v5}, Landroid/renderscript/Element;->F32_3(Landroid/renderscript/RenderScript;)Landroid/renderscript/Element;
  
      move-result-object v5
  
!     const-string v6, "normal"
  
      invoke-virtual {v0, v5, v6}, Landroid/renderscript/Element$Builder;->add(Landroid/renderscript/Element;Ljava/lang/String;)Landroid/renderscript/Element$Builder;
  
      :cond_2
      invoke-virtual {v0}, Landroid/renderscript/Element$Builder;->create()Landroid/renderscript/Element;
  
--- 693,713 ----
  
      if-eqz v5, :cond_2
  
+     .line 745
      add-int/lit8 v1, v1, 0x3
  
+     .line 746
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-static {v5}, Landroid/renderscript/Element;->F32_3(Landroid/renderscript/RenderScript;)Landroid/renderscript/Element;
  
      move-result-object v5
  
!     const-string/jumbo v6, "normal"
  
      invoke-virtual {v0, v5, v6}, Landroid/renderscript/Element$Builder;->add(Landroid/renderscript/Element;Ljava/lang/String;)Landroid/renderscript/Element$Builder;
  
+     .line 748
      :cond_2
      invoke-virtual {v0}, Landroid/renderscript/Element$Builder;->create()Landroid/renderscript/Element;
  
***************
*** 636,648 ****
--- 715,731 ----
  
      iput-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mElement:Landroid/renderscript/Element;
  
+     .line 750
      const/4 v4, 0x1
  
+     .line 751
      .local v4, usage:I
      if-eqz p1, :cond_3
  
+     .line 752
      or-int/lit8 v4, v4, 0x4
  
+     .line 755
      :cond_3
      new-instance v3, Landroid/renderscript/Mesh$Builder;
  
***************
*** 650,655 ****
--- 733,739 ----
  
      invoke-direct {v3, v5, v4}, Landroid/renderscript/Mesh$Builder;-><init>(Landroid/renderscript/RenderScript;I)V
  
+     .line 756
      .local v3, smb:Landroid/renderscript/Mesh$Builder;
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mElement:Landroid/renderscript/Element;
  
***************
*** 659,664 ****
--- 743,749 ----
  
      invoke-virtual {v3, v5, v6}, Landroid/renderscript/Mesh$Builder;->addVertexType(Landroid/renderscript/Element;I)Landroid/renderscript/Mesh$Builder;
  
+     .line 757
      iget-object v5, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-static {v5}, Landroid/renderscript/Element;->U16(Landroid/renderscript/RenderScript;)Landroid/renderscript/Element;
***************
*** 671,680 ****
--- 756,767 ----
  
      invoke-virtual {v3, v5, v6, v7}, Landroid/renderscript/Mesh$Builder;->addIndexSetType(Landroid/renderscript/Element;ILandroid/renderscript/Mesh$Primitive;)Landroid/renderscript/Mesh$Builder;
  
+     .line 759
      invoke-virtual {v3}, Landroid/renderscript/Mesh$Builder;->create()Landroid/renderscript/Mesh;
  
      move-result-object v2
  
+     .line 761
      .local v2, sm:Landroid/renderscript/Mesh;
      invoke-virtual {v2, v8}, Landroid/renderscript/Mesh;->getVertexAllocation(I)Landroid/renderscript/Allocation;
  
***************
*** 688,703 ****
--- 775,794 ----
  
      invoke-virtual {v5, v8, v6, v7}, Landroid/renderscript/Allocation;->copy1DRangeFromUnchecked(II[F)V
  
+     .line 762
      if-eqz p1, :cond_4
  
+     .line 763
      if-eqz p1, :cond_4
  
+     .line 764
      invoke-virtual {v2, v8}, Landroid/renderscript/Mesh;->getVertexAllocation(I)Landroid/renderscript/Allocation;
  
      move-result-object v5
  
      invoke-virtual {v5, v9}, Landroid/renderscript/Allocation;->syncAll(I)V
  
+     .line 768
      :cond_4
      invoke-virtual {v2, v8}, Landroid/renderscript/Mesh;->getIndexSetAllocation(I)Landroid/renderscript/Allocation;
  
***************
*** 709,722 ****
--- 800,816 ----
  
      invoke-virtual {v5, v8, v6, v7}, Landroid/renderscript/Allocation;->copy1DRangeFromUnchecked(II[S)V
  
+     .line 769
      if-eqz p1, :cond_5
  
+     .line 770
      invoke-virtual {v2, v8}, Landroid/renderscript/Mesh;->getIndexSetAllocation(I)Landroid/renderscript/Allocation;
  
      move-result-object v5
  
      invoke-virtual {v5, v9}, Landroid/renderscript/Allocation;->syncAll(I)V
  
+     .line 773
      :cond_5
      return-object v2
  .end method
***************
*** 729,740 ****
--- 823,836 ----
      .parameter "a"
  
      .prologue
+     .line 680
      iget v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
      and-int/lit8 v0, v0, 0x1
  
      if-nez v0, :cond_0
  
+     .line 681
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "add mistmatch with declared components."
***************
*** 743,757 ****
--- 839,858 ----
  
      throw v0
  
+     .line 683
      :cond_0
      iput p1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mR:F
  
+     .line 684
      iput p2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mG:F
  
+     .line 685
      iput p3, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mB:F
  
+     .line 686
      iput p4, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mA:F
  
+     .line 687
      return-object p0
  .end method
  
***************
*** 762,773 ****
--- 863,876 ----
      .parameter "z"
  
      .prologue
+     .line 660
      iget v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
      and-int/lit8 v0, v0, 0x2
  
      if-nez v0, :cond_0
  
+     .line 661
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "add mistmatch with declared components."
***************
*** 776,788 ****
--- 879,895 ----
  
      throw v0
  
+     .line 663
      :cond_0
      iput p1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mNX:F
  
+     .line 664
      iput p2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mNY:F
  
+     .line 665
      iput p3, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mNZ:F
  
+     .line 666
      return-object p0
  .end method
  
***************
*** 792,803 ****
--- 899,912 ----
      .parameter "t"
  
      .prologue
+     .line 642
      iget v0, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mFlags:I
  
      and-int/lit16 v0, v0, 0x100
  
      if-nez v0, :cond_0
  
+     .line 643
      new-instance v0, Ljava/lang/IllegalStateException;
  
      const-string v1, "add mistmatch with declared components."
***************
*** 806,815 ****
--- 915,927 ----
  
      throw v0
  
+     .line 645
      :cond_0
      iput p1, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mS0:F
  
+     .line 646
      iput p2, p0, Landroid/renderscript/Mesh$TriangleMeshBuilder;->mT0:F
  
+     .line 647
      return-object p0
  .end method
