*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Program$ProgramParam.smali	2012-03-23 19:35:32.180173690 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Program$ProgramParam.smali.original	2012-03-23 19:34:44.700307896 -0500
***************
*** 52,57 ****
--- 52,58 ----
  
      const/4 v2, 0x0
  
+     .line 57
      new-instance v0, Landroid/renderscript/Program$ProgramParam;
  
      const-string v1, "INPUT"
***************
*** 60,65 ****
--- 61,67 ----
  
      sput-object v0, Landroid/renderscript/Program$ProgramParam;->INPUT:Landroid/renderscript/Program$ProgramParam;
  
+     .line 58
      new-instance v0, Landroid/renderscript/Program$ProgramParam;
  
      const-string v1, "OUTPUT"
***************
*** 68,73 ****
--- 70,76 ----
  
      sput-object v0, Landroid/renderscript/Program$ProgramParam;->OUTPUT:Landroid/renderscript/Program$ProgramParam;
  
+     .line 59
      new-instance v0, Landroid/renderscript/Program$ProgramParam;
  
      const-string v1, "CONSTANT"
***************
*** 76,81 ****
--- 79,85 ----
  
      sput-object v0, Landroid/renderscript/Program$ProgramParam;->CONSTANT:Landroid/renderscript/Program$ProgramParam;
  
+     .line 60
      new-instance v0, Landroid/renderscript/Program$ProgramParam;
  
      const-string v1, "TEXTURE_TYPE"
***************
*** 84,89 ****
--- 88,94 ----
  
      sput-object v0, Landroid/renderscript/Program$ProgramParam;->TEXTURE_TYPE:Landroid/renderscript/Program$ProgramParam;
  
+     .line 56
      const/4 v0, 0x4
  
      new-array v0, v0, [Landroid/renderscript/Program$ProgramParam;
***************
*** 121,130 ****
--- 126,138 ----
      .end annotation
  
      .prologue
+     .line 63
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 64
      iput p3, p0, Landroid/renderscript/Program$ProgramParam;->mID:I
  
+     .line 65
      return-void
  .end method
  
***************
*** 133,138 ****
--- 141,147 ----
      .parameter "name"
  
      .prologue
+     .line 56
      const-class v0, Landroid/renderscript/Program$ProgramParam;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 148,153 ****
--- 157,163 ----
      .locals 1
  
      .prologue
+     .line 56
      sget-object v0, Landroid/renderscript/Program$ProgramParam;->$VALUES:[Landroid/renderscript/Program$ProgramParam;
  
      invoke-virtual {v0}, [Landroid/renderscript/Program$ProgramParam;->clone()Ljava/lang/Object;
