*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/ProgramFragmentFixedFunction$1.smali	2012-03-23 19:35:31.834167380 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/ProgramFragmentFixedFunction$1.smali.original	2012-03-23 19:34:44.697307842 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 3
  
      .prologue
+     .line 148
      invoke-static {}, Landroid/renderscript/ProgramFragmentFixedFunction$Builder$EnvMode;->values()[Landroid/renderscript/ProgramFragmentFixedFunction$Builder$EnvMode;
  
      move-result-object v0
***************
*** 82,87 ****
--- 83,89 ----
      :try_end_2
      .catch Ljava/lang/NoSuchFieldError; {:try_start_2 .. :try_end_2} :catch_4
  
+     .line 150
      :goto_2
      invoke-static {}, Landroid/renderscript/ProgramFragmentFixedFunction$Builder$Format;->values()[Landroid/renderscript/ProgramFragmentFixedFunction$Builder$Format;
  
***************
*** 179,184 ****
--- 181,187 ----
  
      goto :goto_3
  
+     .line 148
      :catch_4
      move-exception v0
  
