*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/ProgramStore$BlendDstFunc.smali	2012-03-23 19:35:31.415159740 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/ProgramStore$BlendDstFunc.smali.original	2012-03-23 19:34:44.693307768 -0500
***************
*** 62,67 ****
--- 62,68 ----
  
      const/4 v4, 0x0
  
+     .line 123
      new-instance v0, Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      const-string v1, "ZERO"
***************
*** 70,75 ****
--- 71,77 ----
  
      sput-object v0, Landroid/renderscript/ProgramStore$BlendDstFunc;->ZERO:Landroid/renderscript/ProgramStore$BlendDstFunc;
  
+     .line 124
      new-instance v0, Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      const-string v1, "ONE"
***************
*** 78,83 ****
--- 80,86 ----
  
      sput-object v0, Landroid/renderscript/ProgramStore$BlendDstFunc;->ONE:Landroid/renderscript/ProgramStore$BlendDstFunc;
  
+     .line 125
      new-instance v0, Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      const-string v1, "SRC_COLOR"
***************
*** 86,91 ****
--- 89,95 ----
  
      sput-object v0, Landroid/renderscript/ProgramStore$BlendDstFunc;->SRC_COLOR:Landroid/renderscript/ProgramStore$BlendDstFunc;
  
+     .line 126
      new-instance v0, Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      const-string v1, "ONE_MINUS_SRC_COLOR"
***************
*** 94,99 ****
--- 98,104 ----
  
      sput-object v0, Landroid/renderscript/ProgramStore$BlendDstFunc;->ONE_MINUS_SRC_COLOR:Landroid/renderscript/ProgramStore$BlendDstFunc;
  
+     .line 127
      new-instance v0, Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      const-string v1, "SRC_ALPHA"
***************
*** 102,107 ****
--- 107,113 ----
  
      sput-object v0, Landroid/renderscript/ProgramStore$BlendDstFunc;->SRC_ALPHA:Landroid/renderscript/ProgramStore$BlendDstFunc;
  
+     .line 128
      new-instance v0, Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      const-string v1, "ONE_MINUS_SRC_ALPHA"
***************
*** 114,119 ****
--- 120,126 ----
  
      sput-object v0, Landroid/renderscript/ProgramStore$BlendDstFunc;->ONE_MINUS_SRC_ALPHA:Landroid/renderscript/ProgramStore$BlendDstFunc;
  
+     .line 129
      new-instance v0, Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      const-string v1, "DST_ALPHA"
***************
*** 126,131 ****
--- 133,139 ----
  
      sput-object v0, Landroid/renderscript/ProgramStore$BlendDstFunc;->DST_ALPHA:Landroid/renderscript/ProgramStore$BlendDstFunc;
  
+     .line 130
      new-instance v0, Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      const-string v1, "ONE_MINUS_DST_ALPHA"
***************
*** 138,143 ****
--- 146,152 ----
  
      sput-object v0, Landroid/renderscript/ProgramStore$BlendDstFunc;->ONE_MINUS_DST_ALPHA:Landroid/renderscript/ProgramStore$BlendDstFunc;
  
+     .line 122
      const/16 v0, 0x8
  
      new-array v0, v0, [Landroid/renderscript/ProgramStore$BlendDstFunc;
***************
*** 197,206 ****
--- 206,218 ----
      .end annotation
  
      .prologue
+     .line 133
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 134
      iput p3, p0, Landroid/renderscript/ProgramStore$BlendDstFunc;->mID:I
  
+     .line 135
      return-void
  .end method
  
***************
*** 209,214 ****
--- 221,227 ----
      .parameter "name"
  
      .prologue
+     .line 122
      const-class v0, Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 224,229 ****
--- 237,243 ----
      .locals 1
  
      .prologue
+     .line 122
      sget-object v0, Landroid/renderscript/ProgramStore$BlendDstFunc;->$VALUES:[Landroid/renderscript/ProgramStore$BlendDstFunc;
  
      invoke-virtual {v0}, [Landroid/renderscript/ProgramStore$BlendDstFunc;->clone()Ljava/lang/Object;
