*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/ProgramFragmentFixedFunction.smali	2012-03-23 19:35:33.119190813 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/ProgramFragmentFixedFunction.smali.original	2012-03-23 19:34:44.707308024 -0500
***************
*** 20,26 ****
--- 20,28 ----
      .parameter "rs"
  
      .prologue
+     .line 33
      invoke-direct {p0, p1, p2}, Landroid/renderscript/ProgramFragment;-><init>(ILandroid/renderscript/RenderScript;)V
  
+     .line 34
      return-void
  .end method
