*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/ProgramFragment$Builder.smali	2012-03-23 19:35:32.443178486 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/ProgramFragment$Builder.smali.original	2012-03-23 19:34:44.702307932 -0500
***************
*** 20,27 ****
--- 20,29 ----
      .parameter "rs"
  
      .prologue
+     .line 51
      invoke-direct {p0, p1}, Landroid/renderscript/Program$BaseProgramBuilder;-><init>(Landroid/renderscript/RenderScript;)V
  
+     .line 52
      return-void
  .end method
  
***************
*** 31,40 ****
--- 33,44 ----
      .locals 8
  
      .prologue
+     .line 60
      iget-object v6, p0, Landroid/renderscript/ProgramFragment$Builder;->mRS:Landroid/renderscript/RenderScript;
  
      invoke-virtual {v6}, Landroid/renderscript/RenderScript;->validate()V
  
+     .line 61
      iget v6, p0, Landroid/renderscript/ProgramFragment$Builder;->mInputCount:I
  
      iget v7, p0, Landroid/renderscript/ProgramFragment$Builder;->mOutputCount:I
***************
*** 53,61 ****
--- 57,67 ----
  
      new-array v5, v6, [I
  
+     .line 62
      .local v5, tmp:[I
      const/4 v2, 0x0
  
+     .line 64
      .local v2, idx:I
      const/4 v0, 0x0
  
***************
*** 65,70 ****
--- 71,77 ----
  
      if-ge v0, v6, :cond_0
  
+     .line 65
      add-int/lit8 v3, v2, 0x1
  
      .end local v2           #idx:I
***************
*** 75,80 ****
--- 82,88 ----
  
      aput v6, v5, v2
  
+     .line 66
      add-int/lit8 v2, v3, 0x1
  
      .end local v3           #idx:I
***************
*** 89,98 ****
--- 97,108 ----
  
      aput v6, v5, v3
  
+     .line 64
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 68
      :cond_0
      const/4 v0, 0x0
  
***************
*** 101,106 ****
--- 111,117 ----
  
      if-ge v0, v6, :cond_1
  
+     .line 69
      add-int/lit8 v3, v2, 0x1
  
      .end local v2           #idx:I
***************
*** 111,116 ****
--- 122,128 ----
  
      aput v6, v5, v2
  
+     .line 70
      add-int/lit8 v2, v3, 0x1
  
      .end local v3           #idx:I
***************
*** 125,134 ****
--- 137,148 ----
  
      aput v6, v5, v3
  
+     .line 68
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 72
      :cond_1
      const/4 v0, 0x0
  
***************
*** 137,142 ****
--- 151,157 ----
  
      if-ge v0, v6, :cond_2
  
+     .line 73
      add-int/lit8 v3, v2, 0x1
  
      .end local v2           #idx:I
***************
*** 147,152 ****
--- 162,168 ----
  
      aput v6, v5, v2
  
+     .line 74
      add-int/lit8 v2, v3, 0x1
  
      .end local v3           #idx:I
***************
*** 161,170 ****
--- 177,188 ----
  
      aput v6, v5, v3
  
+     .line 72
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_2
  
+     .line 76
      :cond_2
      const/4 v0, 0x0
  
***************
*** 173,178 ****
--- 191,197 ----
  
      if-ge v0, v6, :cond_3
  
+     .line 77
      add-int/lit8 v3, v2, 0x1
  
      .end local v2           #idx:I
***************
*** 183,188 ****
--- 202,208 ----
  
      aput v6, v5, v2
  
+     .line 78
      add-int/lit8 v2, v3, 0x1
  
      .end local v3           #idx:I
***************
*** 195,204 ****
--- 215,226 ----
  
      aput v6, v5, v3
  
+     .line 76
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_3
  
+     .line 81
      :cond_3
      iget-object v6, p0, Landroid/renderscript/ProgramFragment$Builder;->mRS:Landroid/renderscript/RenderScript;
  
***************
*** 208,213 ****
--- 230,236 ----
  
      move-result v1
  
+     .line 82
      .local v1, id:I
      new-instance v4, Landroid/renderscript/ProgramFragment;
  
***************
*** 215,222 ****
--- 238,247 ----
  
      invoke-direct {v4, v1, v6}, Landroid/renderscript/ProgramFragment;-><init>(ILandroid/renderscript/RenderScript;)V
  
+     .line 83
      .local v4, pf:Landroid/renderscript/ProgramFragment;
      invoke-virtual {p0, v4}, Landroid/renderscript/ProgramFragment$Builder;->initProgram(Landroid/renderscript/Program;)V
  
+     .line 84
      return-object v4
  .end method
