*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/RenderScript$RSMessageHandler.smali	2012-03-23 19:35:32.635181988 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/RenderScript$RSMessageHandler.smali.original	2012-03-23 19:34:44.703307950 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .locals 0
  
      .prologue
+     .line 691
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 41,45 ****
--- 42,47 ----
      .locals 0
  
      .prologue
+     .line 696
      return-void
  .end method
