*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/RenderScript$RSErrorHandler.smali	2012-03-23 19:35:32.065171593 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/RenderScript$RSErrorHandler.smali.original	2012-03-23 19:34:44.699307878 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .locals 0
  
      .prologue
+     .line 719
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 39,43 ****
--- 40,45 ----
      .locals 0
  
      .prologue
+     .line 723
      return-void
  .end method
