*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/RSTextureView.smali	2012-03-23 19:35:33.411196138 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/RSTextureView.smali.original	2012-03-23 19:34:44.709308060 -0500
***************
*** 18,27 ****
--- 18,30 ----
      .parameter "context"
  
      .prologue
+     .line 46
      invoke-direct {p0, p1}, Landroid/view/TextureView;-><init>(Landroid/content/Context;)V
  
+     .line 47
      invoke-direct {p0}, Landroid/renderscript/RSTextureView;->init()V
  
+     .line 49
      return-void
  .end method
  
***************
*** 31,40 ****
--- 34,46 ----
      .parameter "attrs"
  
      .prologue
+     .line 57
      invoke-direct {p0, p1, p2}, Landroid/view/TextureView;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 58
      invoke-direct {p0}, Landroid/renderscript/RSTextureView;->init()V
  
+     .line 60
      return-void
  .end method
  
***************
*** 42,49 ****
--- 48,57 ----
      .locals 0
  
      .prologue
+     .line 63
      invoke-virtual {p0, p0}, Landroid/renderscript/RSTextureView;->setSurfaceTextureListener(Landroid/view/TextureView$SurfaceTextureListener;)V
  
+     .line 65
      return-void
  .end method
  
***************
*** 54,59 ****
--- 62,68 ----
      .parameter "sc"
  
      .prologue
+     .line 140
      new-instance v0, Landroid/renderscript/RenderScriptGL;
  
      invoke-virtual {p0}, Landroid/renderscript/RSTextureView;->getContext()Landroid/content/Context;
***************
*** 62,74 ****
--- 71,86 ----
  
      invoke-direct {v0, v1, p1}, Landroid/renderscript/RenderScriptGL;-><init>(Landroid/content/Context;Landroid/renderscript/RenderScriptGL$SurfaceConfig;)V
  
+     .line 141
      .local v0, rs:Landroid/renderscript/RenderScriptGL;
      invoke-virtual {p0, v0}, Landroid/renderscript/RSTextureView;->setRenderScriptGL(Landroid/renderscript/RenderScriptGL;)V
  
+     .line 142
      iget-object v1, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
  
      if-eqz v1, :cond_0
  
+     .line 143
      iget-object v1, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      iget-object v2, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
***************
*** 83,88 ****
--- 95,101 ----
  
      invoke-virtual {v1, v2, v3, v4}, Landroid/renderscript/RenderScriptGL;->setSurfaceTexture(Landroid/graphics/SurfaceTexture;II)V
  
+     .line 145
      :cond_0
      return-object v0
  .end method
***************
*** 91,104 ****
--- 104,120 ----
      .locals 1
  
      .prologue
+     .line 153
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      invoke-virtual {v0}, Landroid/renderscript/RenderScriptGL;->destroy()V
  
+     .line 154
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
+     .line 155
      return-void
  .end method
  
***************
*** 106,111 ****
--- 122,128 ----
      .locals 1
  
      .prologue
+     .line 176
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      return-object v0
***************
*** 118,135 ****
--- 135,156 ----
      .parameter "height"
  
      .prologue
+     .line 70
      iput-object p1, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
  
+     .line 72
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      if-eqz v0, :cond_0
  
+     .line 73
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      iget-object v1, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
  
      invoke-virtual {v0, v1, p2, p3}, Landroid/renderscript/RenderScriptGL;->setSurfaceTexture(Landroid/graphics/SurfaceTexture;II)V
  
+     .line 75
      :cond_0
      return-void
  .end method
***************
*** 141,158 ****
--- 162,183 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 90
      iput-object p1, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
  
+     .line 92
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      if-eqz v0, :cond_0
  
+     .line 93
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, v1, v2, v2}, Landroid/renderscript/RenderScriptGL;->setSurfaceTexture(Landroid/graphics/SurfaceTexture;II)V
  
+     .line 96
      :cond_0
      const/4 v0, 0x1
  
***************
*** 166,183 ****
--- 191,212 ----
      .parameter "height"
  
      .prologue
+     .line 80
      iput-object p1, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
  
+     .line 82
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      if-eqz v0, :cond_0
  
+     .line 83
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      iget-object v1, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
  
      invoke-virtual {v0, v1, p2, p3}, Landroid/renderscript/RenderScriptGL;->setSurfaceTexture(Landroid/graphics/SurfaceTexture;II)V
  
+     .line 85
      :cond_0
      return-void
  .end method
***************
*** 187,194 ****
--- 216,225 ----
      .parameter "surface"
  
      .prologue
+     .line 102
      iput-object p1, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
  
+     .line 103
      return-void
  .end method
  
***************
*** 196,209 ****
--- 227,243 ----
      .locals 1
  
      .prologue
+     .line 112
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      if-eqz v0, :cond_0
  
+     .line 113
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      invoke-virtual {v0}, Landroid/renderscript/RenderScriptGL;->pause()V
  
+     .line 115
      :cond_0
      return-void
  .end method
***************
*** 212,225 ****
--- 246,262 ----
      .locals 1
  
      .prologue
+     .line 125
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      if-eqz v0, :cond_0
  
+     .line 126
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      invoke-virtual {v0}, Landroid/renderscript/RenderScriptGL;->resume()V
  
+     .line 128
      :cond_0
      return-void
  .end method
***************
*** 229,240 ****
--- 266,280 ----
      .parameter "rs"
  
      .prologue
+     .line 164
      iput-object p1, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
+     .line 165
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
  
      if-eqz v0, :cond_0
  
+     .line 166
      iget-object v0, p0, Landroid/renderscript/RSTextureView;->mRS:Landroid/renderscript/RenderScriptGL;
  
      iget-object v1, p0, Landroid/renderscript/RSTextureView;->mSurfaceTexture:Landroid/graphics/SurfaceTexture;
***************
*** 249,254 ****
--- 289,295 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/renderscript/RenderScriptGL;->setSurfaceTexture(Landroid/graphics/SurfaceTexture;II)V
  
+     .line 168
      :cond_0
      return-void
  .end method
