*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Type$Builder.smali	2012-03-23 19:35:32.904186892 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/renderscript/Type$Builder.smali.original	2012-03-23 19:34:44.705307988 -0500
***************
*** 37,54 ****
--- 37,60 ----
      .parameter "e"
  
      .prologue
+     .line 214
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 200
      const/4 v0, 0x1
  
      iput v0, p0, Landroid/renderscript/Type$Builder;->mDimX:I
  
+     .line 215
      invoke-virtual {p2}, Landroid/renderscript/Element;->checkValid()V
  
+     .line 216
      iput-object p1, p0, Landroid/renderscript/Type$Builder;->mRS:Landroid/renderscript/RenderScript;
  
+     .line 217
      iput-object p2, p0, Landroid/renderscript/Type$Builder;->mElement:Landroid/renderscript/Element;
  
+     .line 218
      return-void
  .end method
  
***************
*** 60,69 ****
--- 66,77 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 259
      iget v0, p0, Landroid/renderscript/Type$Builder;->mDimZ:I
  
      if-lez v0, :cond_2
  
+     .line 260
      iget v0, p0, Landroid/renderscript/Type$Builder;->mDimX:I
  
      if-lt v0, v1, :cond_0
***************
*** 72,77 ****
--- 80,86 ----
  
      if-ge v0, v1, :cond_1
  
+     .line 261
      :cond_0
      new-instance v0, Landroid/renderscript/RSInvalidStateException;
  
***************
*** 81,91 ****
--- 90,102 ----
  
      throw v0
  
+     .line 263
      :cond_1
      iget-boolean v0, p0, Landroid/renderscript/Type$Builder;->mDimFaces:Z
  
      if-eqz v0, :cond_2
  
+     .line 264
      new-instance v0, Landroid/renderscript/RSInvalidStateException;
  
      const-string v1, "Cube maps not supported with 3D types."
***************
*** 94,108 ****
--- 105,122 ----
  
      throw v0
  
+     .line 267
      :cond_2
      iget v0, p0, Landroid/renderscript/Type$Builder;->mDimY:I
  
      if-lez v0, :cond_3
  
+     .line 268
      iget v0, p0, Landroid/renderscript/Type$Builder;->mDimX:I
  
      if-ge v0, v1, :cond_3
  
+     .line 269
      new-instance v0, Landroid/renderscript/RSInvalidStateException;
  
      const-string v1, "X dimension required when Y is present."
***************
*** 111,125 ****
--- 125,142 ----
  
      throw v0
  
+     .line 272
      :cond_3
      iget-boolean v0, p0, Landroid/renderscript/Type$Builder;->mDimFaces:Z
  
      if-eqz v0, :cond_4
  
+     .line 273
      iget v0, p0, Landroid/renderscript/Type$Builder;->mDimY:I
  
      if-ge v0, v1, :cond_4
  
+     .line 274
      new-instance v0, Landroid/renderscript/RSInvalidStateException;
  
      const-string v1, "Cube maps require 2D Types."
***************
*** 128,133 ****
--- 145,151 ----
  
      throw v0
  
+     .line 278
      :cond_4
      iget-object v0, p0, Landroid/renderscript/Type$Builder;->mRS:Landroid/renderscript/RenderScript;
  
***************
*** 151,156 ****
--- 169,175 ----
  
      move-result v7
  
+     .line 279
      .local v7, id:I
      new-instance v8, Landroid/renderscript/Type;
  
***************
*** 158,190 ****
--- 177,217 ----
  
      invoke-direct {v8, v7, v0}, Landroid/renderscript/Type;-><init>(ILandroid/renderscript/RenderScript;)V
  
+     .line 280
      .local v8, t:Landroid/renderscript/Type;
      iget-object v0, p0, Landroid/renderscript/Type$Builder;->mElement:Landroid/renderscript/Element;
  
      iput-object v0, v8, Landroid/renderscript/Type;->mElement:Landroid/renderscript/Element;
  
+     .line 281
      iget v0, p0, Landroid/renderscript/Type$Builder;->mDimX:I
  
      iput v0, v8, Landroid/renderscript/Type;->mDimX:I
  
+     .line 282
      iget v0, p0, Landroid/renderscript/Type$Builder;->mDimY:I
  
      iput v0, v8, Landroid/renderscript/Type;->mDimY:I
  
+     .line 283
      iget v0, p0, Landroid/renderscript/Type$Builder;->mDimZ:I
  
      iput v0, v8, Landroid/renderscript/Type;->mDimZ:I
  
+     .line 284
      iget-boolean v0, p0, Landroid/renderscript/Type$Builder;->mDimMipmaps:Z
  
      iput-boolean v0, v8, Landroid/renderscript/Type;->mDimMipmaps:Z
  
+     .line 285
      iget-boolean v0, p0, Landroid/renderscript/Type$Builder;->mDimFaces:Z
  
      iput-boolean v0, v8, Landroid/renderscript/Type;->mDimFaces:Z
  
+     .line 287
      invoke-virtual {v8}, Landroid/renderscript/Type;->calcElementCount()V
  
+     .line 288
      return-object v8
  .end method
  
***************
*** 193,200 ****
--- 220,229 ----
      .parameter "value"
  
      .prologue
+     .line 248
      iput-boolean p1, p0, Landroid/renderscript/Type$Builder;->mDimFaces:Z
  
+     .line 249
      return-object p0
  .end method
  
***************
*** 203,210 ****
--- 232,241 ----
      .parameter "value"
  
      .prologue
+     .line 243
      iput-boolean p1, p0, Landroid/renderscript/Type$Builder;->mDimMipmaps:Z
  
+     .line 244
      return-object p0
  .end method
  
***************
*** 213,222 ****
--- 244,255 ----
      .parameter "value"
  
      .prologue
+     .line 227
      const/4 v0, 0x1
  
      if-ge p1, v0, :cond_0
  
+     .line 228
      new-instance v0, Landroid/renderscript/RSIllegalArgumentException;
  
      const-string v1, "Values of less than 1 for Dimension X are not valid."
***************
*** 225,233 ****
--- 258,268 ----
  
      throw v0
  
+     .line 230
      :cond_0
      iput p1, p0, Landroid/renderscript/Type$Builder;->mDimX:I
  
+     .line 231
      return-object p0
  .end method
  
***************
*** 236,245 ****
--- 271,282 ----
      .parameter "value"
  
      .prologue
+     .line 235
      const/4 v0, 0x1
  
      if-ge p1, v0, :cond_0
  
+     .line 236
      new-instance v0, Landroid/renderscript/RSIllegalArgumentException;
  
      const-string v1, "Values of less than 1 for Dimension Y are not valid."
***************
*** 248,255 ****
--- 285,294 ----
  
      throw v0
  
+     .line 238
      :cond_0
      iput p1, p0, Landroid/renderscript/Type$Builder;->mDimY:I
  
+     .line 239
      return-object p0
  .end method
