*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/drm/DrmInfoRequest.smali	2012-03-23 19:37:16.460075232 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/drm/DrmInfoRequest.smali.original	2012-03-23 19:34:45.614324563 -0500
***************
*** 42,59 ****
--- 42,64 ----
      .parameter "mimeType"
  
      .prologue
+     .line 67
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 59
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Landroid/drm/DrmInfoRequest;->mRequestInformation:Ljava/util/HashMap;
  
+     .line 68
      iput p1, p0, Landroid/drm/DrmInfoRequest;->mInfoType:I
  
+     .line 69
      iput-object p2, p0, Landroid/drm/DrmInfoRequest;->mMimeType:Ljava/lang/String;
  
+     .line 70
      return-void
  .end method
  
***************
*** 62,80 ****
--- 67,90 ----
      .parameter "infoType"
  
      .prologue
+     .line 145
      const/4 v0, 0x0
  
+     .line 147
      .local v0, isValid:Z
      packed-switch p0, :pswitch_data_0
  
+     .line 155
      :goto_0
      return v0
  
+     .line 152
      :pswitch_0
      const/4 v0, 0x1
  
      goto :goto_0
  
+     .line 147
      nop
  
      :pswitch_data_0
***************
*** 93,98 ****
--- 103,109 ----
      .parameter "key"
  
      .prologue
+     .line 109
      iget-object v0, p0, Landroid/drm/DrmInfoRequest;->mRequestInformation:Ljava/util/HashMap;
  
      invoke-virtual {v0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 106,111 ****
--- 117,123 ----
      .locals 1
  
      .prologue
+     .line 87
      iget v0, p0, Landroid/drm/DrmInfoRequest;->mInfoType:I
  
      return v0
***************
*** 115,120 ****
--- 127,133 ----
      .locals 1
  
      .prologue
+     .line 78
      iget-object v0, p0, Landroid/drm/DrmInfoRequest;->mMimeType:Ljava/lang/String;
  
      return-object v0
***************
*** 124,129 ****
--- 137,143 ----
      .locals 2
  
      .prologue
+     .line 140
      iget-object v0, p0, Landroid/drm/DrmInfoRequest;->mMimeType:Ljava/lang/String;
  
      if-eqz v0, :cond_0
***************
*** 174,179 ****
--- 188,194 ----
      .end annotation
  
      .prologue
+     .line 129
      iget-object v0, p0, Landroid/drm/DrmInfoRequest;->mRequestInformation:Ljava/util/HashMap;
  
      invoke-virtual {v0}, Ljava/util/HashMap;->values()Ljava/util/Collection;
***************
*** 200,205 ****
--- 215,221 ----
      .end annotation
  
      .prologue
+     .line 119
      iget-object v0, p0, Landroid/drm/DrmInfoRequest;->mRequestInformation:Ljava/util/HashMap;
  
      invoke-virtual {v0}, Ljava/util/HashMap;->keySet()Ljava/util/Set;
***************
*** 219,227 ****
--- 235,245 ----
      .parameter "value"
  
      .prologue
+     .line 97
      iget-object v0, p0, Landroid/drm/DrmInfoRequest;->mRequestInformation:Ljava/util/HashMap;
  
      invoke-virtual {v0, p1, p2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 98
      return-void
  .end method
