*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/drm/DrmStore.smali	2012-03-23 19:37:17.033085680 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/drm/DrmStore.smali.original	2012-03-23 19:34:45.618324636 -0500
***************
*** 20,26 ****
--- 20,28 ----
      .locals 0
  
      .prologue
+     .line 23
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 186
      return-void
  .end method
