*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/ExtractButton.smali	2012-03-23 19:36:22.647093953 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/ExtractButton.smali.original	2012-03-23 19:34:45.081314843 -0500
***************
*** 9,18 ****
--- 9,20 ----
      .parameter "context"
  
      .prologue
+     .line 28
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Landroid/widget/Button;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 29
      return-void
  .end method
  
***************
*** 22,31 ****
--- 24,35 ----
      .parameter "attrs"
  
      .prologue
+     .line 32
      const v0, 0x1010048
  
      invoke-direct {p0, p1, p2, v0}, Landroid/widget/Button;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 33
      return-void
  .end method
  
***************
*** 36,43 ****
--- 40,49 ----
      .parameter "defStyle"
  
      .prologue
+     .line 36
      invoke-direct {p0, p1, p2, p3}, Landroid/widget/Button;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 37
      return-void
  .end method
  
***************
*** 47,52 ****
--- 53,59 ----
      .locals 1
  
      .prologue
+     .line 44
      invoke-virtual {p0}, Landroid/inputmethodservice/ExtractButton;->isEnabled()Z
  
      move-result v0
