*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/IInputMethodSessionWrapper.smali	2012-03-23 19:36:23.615111604 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/IInputMethodSessionWrapper.smali.original	2012-03-23 19:34:45.087314953 -0500
***************
*** 55,70 ****
--- 55,74 ----
      .parameter "inputMethodSession"
  
      .prologue
+     .line 70
      invoke-direct {p0}, Lcom/android/internal/view/IInputMethodSession$Stub;-><init>()V
  
+     .line 71
      new-instance v0, Lcom/android/internal/os/HandlerCaller;
  
      invoke-direct {v0, p1, p0}, Lcom/android/internal/os/HandlerCaller;-><init>(Landroid/content/Context;Lcom/android/internal/os/HandlerCaller$Callback;)V
  
      iput-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
+     .line 72
      iput-object p2, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
+     .line 73
      return-void
  .end method
  
***************
*** 76,81 ****
--- 80,86 ----
      .parameter "data"
  
      .prologue
+     .line 186
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 88,93 ****
--- 93,99 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 187
      return-void
  .end method
  
***************
*** 98,103 ****
--- 104,110 ----
      .parameter "callback"
  
      .prologue
+     .line 160
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 110,115 ****
--- 117,123 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 162
      return-void
  .end method
  
***************
*** 120,125 ****
--- 128,134 ----
      .parameter "callback"
  
      .prologue
+     .line 165
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 132,137 ****
--- 141,147 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 167
      return-void
  .end method
  
***************
*** 140,145 ****
--- 150,156 ----
      .parameter "completions"
  
      .prologue
+     .line 150
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 152,157 ****
--- 163,169 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 152
      return-void
  .end method
  
***************
*** 162,179 ****
--- 174,195 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 80
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
      if-nez v1, :cond_0
  
+     .line 143
      :goto_0
      return-void
  
+     .line 82
      :cond_0
      iget v1, p1, Landroid/os/Message;->what:I
  
      sparse-switch v1, :sswitch_data_0
  
+     .line 142
      const-string v0, "InputMethodWrapper"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 200,205 ****
--- 216,222 ----
  
      goto :goto_0
  
+     .line 84
      :sswitch_0
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
***************
*** 207,212 ****
--- 224,230 ----
  
      goto :goto_0
  
+     .line 87
      :sswitch_1
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
***************
*** 220,225 ****
--- 238,244 ----
  
      goto :goto_0
  
+     .line 90
      :sswitch_2
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
***************
*** 233,243 ****
--- 252,264 ----
  
      goto :goto_0
  
+     .line 94
      :sswitch_3
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v7, Lcom/android/internal/os/HandlerCaller$SomeArgs;
  
+     .line 95
      .local v7, args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      iget-object v2, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
***************
*** 257,274 ****
--- 278,298 ----
  
      invoke-interface {v2, v3, v0, v4}, Landroid/view/inputmethod/InputMethodSession;->dispatchKeyEvent(ILandroid/view/KeyEvent;Landroid/view/inputmethod/InputMethodSession$EventCallback;)V
  
+     .line 99
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      invoke-virtual {v0, v7}, Lcom/android/internal/os/HandlerCaller;->recycleArgs(Lcom/android/internal/os/HandlerCaller$SomeArgs;)V
  
      goto :goto_0
  
+     .line 103
      .end local v7           #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      :sswitch_4
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v7, Lcom/android/internal/os/HandlerCaller$SomeArgs;
  
+     .line 104
      .restart local v7       #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      iget-object v2, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
***************
*** 288,305 ****
--- 312,332 ----
  
      invoke-interface {v2, v3, v0, v4}, Landroid/view/inputmethod/InputMethodSession;->dispatchTrackballEvent(ILandroid/view/MotionEvent;Landroid/view/inputmethod/InputMethodSession$EventCallback;)V
  
+     .line 108
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      invoke-virtual {v0, v7}, Lcom/android/internal/os/HandlerCaller;->recycleArgs(Lcom/android/internal/os/HandlerCaller$SomeArgs;)V
  
      goto :goto_0
  
+     .line 112
      .end local v7           #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      :sswitch_5
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v7, Lcom/android/internal/os/HandlerCaller$SomeArgs;
  
+     .line 113
      .restart local v7       #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
***************
*** 317,328 ****
--- 344,357 ----
  
      invoke-interface/range {v0 .. v6}, Landroid/view/inputmethod/InputMethodSession;->updateSelection(IIIIII)V
  
+     .line 115
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      invoke-virtual {v0, v7}, Lcom/android/internal/os/HandlerCaller;->recycleArgs(Lcom/android/internal/os/HandlerCaller$SomeArgs;)V
  
      goto/16 :goto_0
  
+     .line 119
      .end local v7           #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      :sswitch_6
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
***************
*** 335,345 ****
--- 364,376 ----
  
      goto/16 :goto_0
  
+     .line 123
      :sswitch_7
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v7, Lcom/android/internal/os/HandlerCaller$SomeArgs;
  
+     .line 124
      .restart local v7       #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      iget-object v2, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
***************
*** 353,364 ****
--- 384,397 ----
  
      invoke-interface {v2, v0, v1}, Landroid/view/inputmethod/InputMethodSession;->appPrivateCommand(Ljava/lang/String;Landroid/os/Bundle;)V
  
+     .line 126
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      invoke-virtual {v0, v7}, Lcom/android/internal/os/HandlerCaller;->recycleArgs(Lcom/android/internal/os/HandlerCaller$SomeArgs;)V
  
      goto/16 :goto_0
  
+     .line 130
      .end local v7           #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      :sswitch_8
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
***************
*** 371,376 ****
--- 404,410 ----
  
      goto/16 :goto_0
  
+     .line 134
      :sswitch_9
      const/4 v0, 0x0
  
***************
*** 378,383 ****
--- 412,418 ----
  
      goto/16 :goto_0
  
+     .line 138
      :sswitch_a
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
***************
*** 395,400 ****
--- 430,436 ----
  
      goto :goto_1
  
+     .line 82
      :sswitch_data_0
      .sparse-switch
          0x3c -> :sswitch_0
***************
*** 415,420 ****
--- 451,457 ----
      .locals 3
  
      .prologue
+     .line 146
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 427,432 ****
--- 464,470 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 147
      return-void
  .end method
  
***************
*** 434,439 ****
--- 472,478 ----
      .locals 3
  
      .prologue
+     .line 194
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 446,451 ****
--- 485,491 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 195
      return-void
  .end method
  
***************
*** 453,458 ****
--- 493,499 ----
      .locals 1
  
      .prologue
+     .line 76
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mInputMethodSession:Landroid/view/inputmethod/InputMethodSession;
  
      return-object v0
***************
*** 464,469 ****
--- 505,511 ----
      .parameter "hideFlags"
  
      .prologue
+     .line 190
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 476,481 ****
--- 518,524 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 191
      return-void
  .end method
  
***************
*** 484,489 ****
--- 527,533 ----
      .parameter "newCursor"
  
      .prologue
+     .line 181
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 496,501 ****
--- 540,546 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 183
      return-void
  .end method
  
***************
*** 505,510 ****
--- 550,556 ----
      .parameter "text"
  
      .prologue
+     .line 155
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 517,522 ****
--- 563,569 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 157
      return-void
  .end method
  
***************
*** 530,535 ****
--- 577,583 ----
      .parameter "candidatesEnd"
  
      .prologue
+     .line 171
      iget-object v8, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 554,559 ****
--- 602,608 ----
  
      invoke-virtual {v8, v0}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 174
      return-void
  .end method
  
***************
*** 562,567 ****
--- 611,617 ----
      .parameter "focusChanged"
  
      .prologue
+     .line 177
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v2, p0, Landroid/inputmethodservice/IInputMethodSessionWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 579,586 ****
--- 629,638 ----
  
      invoke-virtual {v1, v0}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 178
      return-void
  
+     .line 177
      :cond_0
      const/4 v0, 0x0
  
