*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/IInputMethodWrapper.smali	2012-03-23 19:36:22.704094992 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/IInputMethodWrapper.smali.original	2012-03-23 19:34:45.082314862 -0500
***************
*** 80,93 ****
--- 80,96 ----
      .parameter "inputMethod"
  
      .prologue
+     .line 102
      invoke-direct {p0}, Lcom/android/internal/view/IInputMethod$Stub;-><init>()V
  
+     .line 103
      new-instance v0, Ljava/lang/ref/WeakReference;
  
      invoke-direct {v0, p1}, Ljava/lang/ref/WeakReference;-><init>(Ljava/lang/Object;)V
  
      iput-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mTarget:Ljava/lang/ref/WeakReference;
  
+     .line 104
      new-instance v0, Lcom/android/internal/os/HandlerCaller;
  
      invoke-virtual {p1}, Landroid/inputmethodservice/AbstractInputMethodService;->getApplicationContext()Landroid/content/Context;
***************
*** 98,109 ****
--- 101,114 ----
  
      iput-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
+     .line 105
      new-instance v0, Ljava/lang/ref/WeakReference;
  
      invoke-direct {v0, p2}, Ljava/lang/ref/WeakReference;-><init>(Ljava/lang/Object;)V
  
      iput-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mInputMethod:Ljava/lang/ref/WeakReference;
  
+     .line 106
      invoke-virtual {p1}, Landroid/inputmethodservice/AbstractInputMethodService;->getApplicationInfo()Landroid/content/pm/ApplicationInfo;
  
      move-result-object v0
***************
*** 112,117 ****
--- 117,123 ----
  
      iput v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mTargetSdkVersion:I
  
+     .line 107
      return-void
  .end method
  
***************
*** 122,127 ****
--- 128,134 ----
      .parameter "token"
  
      .prologue
+     .line 223
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 134,139 ****
--- 141,147 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 224
      return-void
  .end method
  
***************
*** 142,147 ****
--- 150,156 ----
      .parameter "binding"
  
      .prologue
+     .line 227
      new-instance v0, Lcom/android/internal/view/InputConnectionWrapper;
  
      invoke-virtual {p1}, Landroid/view/inputmethod/InputBinding;->getConnectionToken()Landroid/os/IBinder;
***************
*** 154,164 ****
--- 163,175 ----
  
      invoke-direct {v0, v2}, Lcom/android/internal/view/InputConnectionWrapper;-><init>(Lcom/android/internal/view/IInputContext;)V
  
+     .line 229
      .local v0, ic:Landroid/view/inputmethod/InputConnection;
      new-instance v1, Landroid/view/inputmethod/InputBinding;
  
      invoke-direct {v1, v0, p1}, Landroid/view/inputmethod/InputBinding;-><init>(Landroid/view/inputmethod/InputConnection;Landroid/view/inputmethod/InputBinding;)V
  
+     .line 230
      .local v1, nu:Landroid/view/inputmethod/InputBinding;
      iget-object v2, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
***************
*** 172,177 ****
--- 183,189 ----
  
      invoke-virtual {v2, v3}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 231
      return-void
  .end method
  
***************
*** 180,185 ****
--- 192,198 ----
      .parameter "subtype"
  
      .prologue
+     .line 283
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 192,197 ****
--- 205,211 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 285
      return-void
  .end method
  
***************
*** 200,205 ****
--- 214,220 ----
      .parameter "callback"
  
      .prologue
+     .line 248
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 212,217 ****
--- 227,233 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 249
      return-void
  .end method
  
***************
*** 224,229 ****
--- 240,246 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 197
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mTarget:Ljava/lang/ref/WeakReference;
  
      invoke-virtual {v0}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
***************
*** 232,244 ****
--- 249,264 ----
  
      check-cast v7, Landroid/inputmethodservice/AbstractInputMethodService;
  
+     .line 198
      .local v7, target:Landroid/inputmethodservice/AbstractInputMethodService;
      if-nez v7, :cond_1
  
+     .line 220
      :cond_0
      :goto_0
      return-void
  
+     .line 201
      :cond_1
      const-string v0, "android.permission.DUMP"
  
***************
*** 248,253 ****
--- 268,274 ----
  
      if-eqz v0, :cond_2
  
+     .line 204
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 288,298 ****
--- 309,321 ----
  
      goto :goto_0
  
+     .line 210
      :cond_2
      new-instance v5, Ljava/util/concurrent/CountDownLatch;
  
      invoke-direct {v5, v1}, Ljava/util/concurrent/CountDownLatch;-><init>(I)V
  
+     .line 211
      .local v5, latch:Ljava/util/concurrent/CountDownLatch;
      iget-object v8, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
***************
*** 310,315 ****
--- 333,339 ----
  
      invoke-virtual {v8, v0}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 214
      const-wide/16 v0, 0x5
  
      :try_start_0
***************
*** 321,326 ****
--- 345,351 ----
  
      if-nez v0, :cond_0
  
+     .line 215
      const-string v0, "Timeout waiting for dump"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
***************
*** 329,337 ****
--- 354,364 ----
  
      goto :goto_0
  
+     .line 217
      :catch_0
      move-exception v6
  
+     .line 218
      .local v6, e:Ljava/lang/InterruptedException;
      const-string v0, "Interrupted waiting for dump"
  
***************
*** 349,354 ****
--- 376,382 ----
  
      const/4 v8, 0x1
  
+     .line 114
      iget-object v7, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mInputMethod:Ljava/lang/ref/WeakReference;
  
      invoke-virtual {v7}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
***************
*** 357,362 ****
--- 385,391 ----
  
      check-cast v5, Landroid/view/inputmethod/InputMethod;
  
+     .line 116
      .local v5, inputMethod:Landroid/view/inputmethod/InputMethod;
      if-nez v5, :cond_1
  
***************
*** 364,369 ****
--- 393,399 ----
  
      if-eq v7, v8, :cond_1
  
+     .line 117
      const-string v7, "InputMethodWrapper"
  
      new-instance v8, Ljava/lang/StringBuilder;
***************
*** 388,402 ****
--- 418,435 ----
  
      invoke-static {v7, v8}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 194
      :cond_0
      :goto_0
      return-void
  
+     .line 121
      :cond_1
      iget v7, p1, Landroid/os/Message;->what:I
  
      sparse-switch v7, :sswitch_data_0
  
+     .line 193
      const-string v7, "InputMethodWrapper"
  
      new-instance v8, Ljava/lang/StringBuilder;
***************
*** 423,428 ****
--- 456,462 ----
  
      goto :goto_0
  
+     .line 123
      :sswitch_0
      iget-object v7, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mTarget:Ljava/lang/ref/WeakReference;
  
***************
*** 432,444 ****
--- 466,481 ----
  
      check-cast v6, Landroid/inputmethodservice/AbstractInputMethodService;
  
+     .line 124
      .local v6, target:Landroid/inputmethodservice/AbstractInputMethodService;
      if-eqz v6, :cond_0
  
+     .line 127
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;
  
+     .line 129
      .local v0, args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      :try_start_0
      iget-object v7, v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;->arg1:Ljava/lang/Object;
***************
*** 459,469 ****
--- 496,508 ----
      :try_end_0
      .catch Ljava/lang/RuntimeException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 134
      :goto_1
      iget-object v8, v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;->arg4:Ljava/lang/Object;
  
      monitor-enter v8
  
+     .line 135
      :try_start_1
      iget-object v7, v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;->arg4:Ljava/lang/Object;
  
***************
*** 471,476 ****
--- 510,516 ----
  
      invoke-virtual {v7}, Ljava/util/concurrent/CountDownLatch;->countDown()V
  
+     .line 136
      monitor-exit v8
  
      goto :goto_0
***************
*** 484,492 ****
--- 524,534 ----
  
      throw v7
  
+     .line 131
      :catch_0
      move-exception v1
  
+     .line 132
      .local v1, e:Ljava/lang/RuntimeException;
      iget-object v7, v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;->arg2:Ljava/lang/Object;
  
***************
*** 514,519 ****
--- 556,562 ----
  
      goto :goto_1
  
+     .line 141
      .end local v0           #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      .end local v1           #e:Ljava/lang/RuntimeException;
      .end local v6           #target:Landroid/inputmethodservice/AbstractInputMethodService;
***************
*** 526,531 ****
--- 569,575 ----
  
      goto :goto_0
  
+     .line 145
      :sswitch_2
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 535,555 ****
--- 579,603 ----
  
      goto :goto_0
  
+     .line 149
      :sswitch_3
      invoke-interface {v5}, Landroid/view/inputmethod/InputMethod;->unbindInput()V
  
      goto :goto_0
  
+     .line 152
      :sswitch_4
      iget-object v0, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;
  
+     .line 153
      .restart local v0       #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      iget-object v4, v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;->arg1:Ljava/lang/Object;
  
      check-cast v4, Lcom/android/internal/view/IInputContext;
  
+     .line 154
      .local v4, inputContext:Lcom/android/internal/view/IInputContext;
      if-eqz v4, :cond_2
  
***************
*** 557,577 ****
--- 605,629 ----
  
      invoke-direct {v2, v4}, Lcom/android/internal/view/InputConnectionWrapper;-><init>(Lcom/android/internal/view/IInputContext;)V
  
+     .line 156
      .local v2, ic:Landroid/view/inputmethod/InputConnection;
      :cond_2
      iget-object v3, v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;->arg2:Ljava/lang/Object;
  
      check-cast v3, Landroid/view/inputmethod/EditorInfo;
  
+     .line 157
      .local v3, info:Landroid/view/inputmethod/EditorInfo;
      iget v7, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mTargetSdkVersion:I
  
      invoke-virtual {v3, v7}, Landroid/view/inputmethod/EditorInfo;->makeCompatible(I)V
  
+     .line 158
      invoke-interface {v5, v2, v3}, Landroid/view/inputmethod/InputMethod;->startInput(Landroid/view/inputmethod/InputConnection;Landroid/view/inputmethod/EditorInfo;)V
  
      goto/16 :goto_0
  
+     .line 162
      .end local v0           #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      .end local v2           #ic:Landroid/view/inputmethod/InputConnection;
      .end local v3           #info:Landroid/view/inputmethod/EditorInfo;
***************
*** 581,591 ****
--- 633,645 ----
  
      check-cast v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;
  
+     .line 163
      .restart local v0       #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      iget-object v4, v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;->arg1:Ljava/lang/Object;
  
      check-cast v4, Lcom/android/internal/view/IInputContext;
  
+     .line 164
      .restart local v4       #inputContext:Lcom/android/internal/view/IInputContext;
      if-eqz v4, :cond_3
  
***************
*** 593,613 ****
--- 647,671 ----
  
      invoke-direct {v2, v4}, Lcom/android/internal/view/InputConnectionWrapper;-><init>(Lcom/android/internal/view/IInputContext;)V
  
+     .line 166
      .restart local v2       #ic:Landroid/view/inputmethod/InputConnection;
      :cond_3
      iget-object v3, v0, Lcom/android/internal/os/HandlerCaller$SomeArgs;->arg2:Ljava/lang/Object;
  
      check-cast v3, Landroid/view/inputmethod/EditorInfo;
  
+     .line 167
      .restart local v3       #info:Landroid/view/inputmethod/EditorInfo;
      iget v7, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mTargetSdkVersion:I
  
      invoke-virtual {v3, v7}, Landroid/view/inputmethod/EditorInfo;->makeCompatible(I)V
  
+     .line 168
      invoke-interface {v5, v2, v3}, Landroid/view/inputmethod/InputMethod;->restartInput(Landroid/view/inputmethod/InputConnection;Landroid/view/inputmethod/EditorInfo;)V
  
      goto/16 :goto_0
  
+     .line 172
      .end local v0           #args:Lcom/android/internal/os/HandlerCaller$SomeArgs;
      .end local v2           #ic:Landroid/view/inputmethod/InputConnection;
      .end local v3           #info:Landroid/view/inputmethod/EditorInfo;
***************
*** 629,634 ****
--- 687,693 ----
  
      goto/16 :goto_0
  
+     .line 177
      :sswitch_7
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 648,653 ****
--- 707,713 ----
  
      goto :goto_2
  
+     .line 181
      :sswitch_8
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 657,662 ****
--- 717,723 ----
  
      goto/16 :goto_0
  
+     .line 184
      :sswitch_9
      iget v8, p1, Landroid/os/Message;->arg1:I
  
***************
*** 668,673 ****
--- 729,735 ----
  
      goto/16 :goto_0
  
+     .line 187
      :sswitch_a
      iget v8, p1, Landroid/os/Message;->arg1:I
  
***************
*** 679,684 ****
--- 741,747 ----
  
      goto/16 :goto_0
  
+     .line 190
      :sswitch_b
      iget-object v7, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
***************
*** 688,693 ****
--- 751,757 ----
  
      goto/16 :goto_0
  
+     .line 121
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_0
***************
*** 709,714 ****
--- 773,779 ----
      .locals 1
  
      .prologue
+     .line 110
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mInputMethod:Ljava/lang/ref/WeakReference;
  
      invoke-virtual {v0}, Ljava/lang/ref/WeakReference;->get()Ljava/lang/Object;
***************
*** 726,731 ****
--- 791,797 ----
      .parameter "resultReceiver"
  
      .prologue
+     .line 278
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 738,743 ****
--- 804,810 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 280
      return-void
  .end method
  
***************
*** 747,752 ****
--- 814,820 ----
      .parameter "attribute"
  
      .prologue
+     .line 243
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 759,764 ****
--- 827,833 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 245
      return-void
  .end method
  
***************
*** 767,772 ****
--- 836,842 ----
      .parameter "session"
  
      .prologue
+     .line 264
      :try_start_0
      move-object v0, p1
  
***************
*** 778,783 ****
--- 848,854 ----
  
      move-result-object v2
  
+     .line 266
      .local v2, ls:Landroid/view/inputmethod/InputMethodSession;
      iget-object v3, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
***************
*** 793,805 ****
--- 864,879 ----
      :try_end_0
      .catch Ljava/lang/ClassCastException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 270
      .end local v2           #ls:Landroid/view/inputmethod/InputMethodSession;
      :goto_0
      return-void
  
+     .line 267
      :catch_0
      move-exception v1
  
+     .line 268
      .local v1, e:Ljava/lang/ClassCastException;
      const-string v3, "InputMethodWrapper"
  
***************
*** 832,837 ****
--- 906,912 ----
      .parameter "enabled"
  
      .prologue
+     .line 253
      :try_start_0
      move-object v0, p1
  
***************
*** 843,848 ****
--- 918,924 ----
  
      move-result-object v2
  
+     .line 255
      .local v2, ls:Landroid/view/inputmethod/InputMethodSession;
      iget-object v4, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
***************
*** 863,882 ****
--- 939,962 ----
      :try_end_0
      .catch Ljava/lang/ClassCastException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 260
      .end local v2           #ls:Landroid/view/inputmethod/InputMethodSession;
      :goto_1
      return-void
  
+     .line 255
      .restart local v2       #ls:Landroid/view/inputmethod/InputMethodSession;
      :cond_0
      const/4 v3, 0x0
  
      goto :goto_0
  
+     .line 257
      .end local v2           #ls:Landroid/view/inputmethod/InputMethodSession;
      :catch_0
      move-exception v1
  
+     .line 258
      .local v1, e:Ljava/lang/ClassCastException;
      const-string v3, "InputMethodWrapper"
  
***************
*** 909,914 ****
--- 989,995 ----
      .parameter "resultReceiver"
  
      .prologue
+     .line 273
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 921,926 ****
--- 1002,1008 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 275
      return-void
  .end method
  
***************
*** 930,935 ****
--- 1012,1018 ----
      .parameter "attribute"
  
      .prologue
+     .line 238
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 942,947 ****
--- 1025,1031 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 240
      return-void
  .end method
  
***************
*** 949,954 ****
--- 1033,1039 ----
      .locals 3
  
      .prologue
+     .line 234
      iget-object v0, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v1, p0, Landroid/inputmethodservice/IInputMethodWrapper;->mCaller:Lcom/android/internal/os/HandlerCaller;
***************
*** 961,965 ****
--- 1046,1051 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/os/HandlerCaller;->executeOrSendMessage(Landroid/os/Message;)V
  
+     .line 235
      return-void
  .end method
