*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/IInputMethodWrapper$Notifier.smali	2012-03-23 19:36:23.492109362 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/IInputMethodWrapper$Notifier.smali.original	2012-03-23 19:34:45.087314953 -0500
***************
*** 23,28 ****
--- 23,29 ----
      .locals 0
  
      .prologue
+     .line 75
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
