*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/KeyboardView$3.smali	2012-03-23 19:36:23.689112954 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/KeyboardView$3.smali.original	2012-03-23 19:34:45.088314971 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1090
      iput-object p1, p0, Landroid/inputmethodservice/KeyboardView$3;->this$0:Landroid/inputmethodservice/KeyboardView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,47 ****
--- 43,49 ----
      .parameter "keyCodes"
  
      .prologue
+     .line 1092
      iget-object v0, p0, Landroid/inputmethodservice/KeyboardView$3;->this$0:Landroid/inputmethodservice/KeyboardView;
  
      #getter for: Landroid/inputmethodservice/KeyboardView;->mKeyboardActionListener:Landroid/inputmethodservice/KeyboardView$OnKeyboardActionListener;
***************
*** 51,61 ****
--- 53,65 ----
  
      invoke-interface {v0, p1, p2}, Landroid/inputmethodservice/KeyboardView$OnKeyboardActionListener;->onKey(I[I)V
  
+     .line 1093
      iget-object v0, p0, Landroid/inputmethodservice/KeyboardView$3;->this$0:Landroid/inputmethodservice/KeyboardView;
  
      #calls: Landroid/inputmethodservice/KeyboardView;->dismissPopupKeyboard()V
      invoke-static {v0}, Landroid/inputmethodservice/KeyboardView;->access$1400(Landroid/inputmethodservice/KeyboardView;)V
  
+     .line 1094
      return-void
  .end method
  
***************
*** 64,69 ****
--- 68,74 ----
      .parameter "primaryCode"
  
      .prologue
+     .line 1106
      iget-object v0, p0, Landroid/inputmethodservice/KeyboardView$3;->this$0:Landroid/inputmethodservice/KeyboardView;
  
      #getter for: Landroid/inputmethodservice/KeyboardView;->mKeyboardActionListener:Landroid/inputmethodservice/KeyboardView$OnKeyboardActionListener;
***************
*** 73,78 ****
--- 78,84 ----
  
      invoke-interface {v0, p1}, Landroid/inputmethodservice/KeyboardView$OnKeyboardActionListener;->onPress(I)V
  
+     .line 1107
      return-void
  .end method
  
***************
*** 81,86 ****
--- 87,93 ----
      .parameter "primaryCode"
  
      .prologue
+     .line 1109
      iget-object v0, p0, Landroid/inputmethodservice/KeyboardView$3;->this$0:Landroid/inputmethodservice/KeyboardView;
  
      #getter for: Landroid/inputmethodservice/KeyboardView;->mKeyboardActionListener:Landroid/inputmethodservice/KeyboardView$OnKeyboardActionListener;
***************
*** 90,95 ****
--- 97,103 ----
  
      invoke-interface {v0, p1}, Landroid/inputmethodservice/KeyboardView$OnKeyboardActionListener;->onRelease(I)V
  
+     .line 1110
      return-void
  .end method
  
***************
*** 98,103 ****
--- 106,112 ----
      .parameter "text"
  
      .prologue
+     .line 1097
      iget-object v0, p0, Landroid/inputmethodservice/KeyboardView$3;->this$0:Landroid/inputmethodservice/KeyboardView;
  
      #getter for: Landroid/inputmethodservice/KeyboardView;->mKeyboardActionListener:Landroid/inputmethodservice/KeyboardView$OnKeyboardActionListener;
***************
*** 107,117 ****
--- 116,128 ----
  
      invoke-interface {v0, p1}, Landroid/inputmethodservice/KeyboardView$OnKeyboardActionListener;->onText(Ljava/lang/CharSequence;)V
  
+     .line 1098
      iget-object v0, p0, Landroid/inputmethodservice/KeyboardView$3;->this$0:Landroid/inputmethodservice/KeyboardView;
  
      #calls: Landroid/inputmethodservice/KeyboardView;->dismissPopupKeyboard()V
      invoke-static {v0}, Landroid/inputmethodservice/KeyboardView;->access$1400(Landroid/inputmethodservice/KeyboardView;)V
  
+     .line 1099
      return-void
  .end method
  
***************
*** 119,124 ****
--- 130,136 ----
      .locals 0
  
      .prologue
+     .line 1104
      return-void
  .end method
  
***************
*** 126,131 ****
--- 138,144 ----
      .locals 0
  
      .prologue
+     .line 1101
      return-void
  .end method
  
***************
*** 133,138 ****
--- 146,152 ----
      .locals 0
  
      .prologue
+     .line 1102
      return-void
  .end method
  
***************
*** 140,144 ****
--- 154,159 ----
      .locals 0
  
      .prologue
+     .line 1103
      return-void
  .end method
