*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/Keyboard$Key.smali	2012-03-23 19:36:23.127102706 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/inputmethodservice/Keyboard$Key.smali.original	2012-03-23 19:34:45.084314899 -0500
***************
*** 79,90 ****
--- 79,92 ----
  
      const/4 v2, 0x0
  
+     .line 285
      new-array v0, v4, [I
  
      fill-array-data v0, :array_0
  
      sput-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_NORMAL_ON:[I
  
+     .line 290
      const/4 v0, 0x3
  
      new-array v0, v0, [I
***************
*** 93,98 ****
--- 95,101 ----
  
      sput-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_PRESSED_ON:[I
  
+     .line 296
      new-array v0, v3, [I
  
      const v1, 0x101009f
***************
*** 101,116 ****
--- 104,122 ----
  
      sput-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_NORMAL_OFF:[I
  
+     .line 300
      new-array v0, v4, [I
  
      fill-array-data v0, :array_2
  
      sput-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_PRESSED_OFF:[I
  
+     .line 305
      new-array v0, v2, [I
  
      sput-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_NORMAL:[I
  
+     .line 308
      new-array v0, v3, [I
  
      const v1, 0x10100a7
***************
*** 121,132 ****
--- 127,140 ----
  
      return-void
  
+     .line 285
      :array_0
      .array-data 0x4
          0x9ft 0x0t 0x1t 0x1t
          0xa0t 0x0t 0x1t 0x1t
      .end array-data
  
+     .line 290
      :array_1
      .array-data 0x4
          0xa7t 0x0t 0x1t 0x1t
***************
*** 134,139 ****
--- 142,148 ----
          0xa0t 0x0t 0x1t 0x1t
      .end array-data
  
+     .line 300
      :array_2
      .array-data 0x4
          0xa7t 0x0t 0x1t 0x1t
***************
*** 158,169 ****
--- 167,182 ----
  
      const/4 v5, 0x0
  
+     .line 331
      invoke-direct {p0, p2}, Landroid/inputmethodservice/Keyboard$Key;-><init>(Landroid/inputmethodservice/Keyboard$Row;)V
  
+     .line 333
      iput p3, p0, Landroid/inputmethodservice/Keyboard$Key;->x:I
  
+     .line 334
      iput p4, p0, Landroid/inputmethodservice/Keyboard$Key;->y:I
  
+     .line 336
      invoke-static {p5}, Landroid/util/Xml;->asAttributeSet(Lorg/xmlpull/v1/XmlPullParser;)Landroid/util/AttributeSet;
  
      move-result-object v2
***************
*** 174,179 ****
--- 187,193 ----
  
      move-result-object v0
  
+     .line 339
      .local v0, a:Landroid/content/res/TypedArray;
      iget-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->keyboard:Landroid/inputmethodservice/Keyboard;
  
***************
*** 190,195 ****
--- 204,210 ----
  
      iput v2, p0, Landroid/inputmethodservice/Keyboard$Key;->width:I
  
+     .line 342
      iget-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->keyboard:Landroid/inputmethodservice/Keyboard;
  
      #getter for: Landroid/inputmethodservice/Keyboard;->mDisplayHeight:I
***************
*** 205,210 ****
--- 220,226 ----
  
      iput v2, p0, Landroid/inputmethodservice/Keyboard$Key;->height:I
  
+     .line 345
      iget-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->keyboard:Landroid/inputmethodservice/Keyboard;
  
      #getter for: Landroid/inputmethodservice/Keyboard;->mDisplayWidth:I
***************
*** 220,227 ****
--- 236,245 ----
  
      iput v2, p0, Landroid/inputmethodservice/Keyboard$Key;->gap:I
  
+     .line 348
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 349
      invoke-static {p5}, Landroid/util/Xml;->asAttributeSet(Lorg/xmlpull/v1/XmlPullParser;)Landroid/util/AttributeSet;
  
      move-result-object v2
***************
*** 232,237 ****
--- 250,256 ----
  
      move-result-object v0
  
+     .line 351
      iget v2, p0, Landroid/inputmethodservice/Keyboard$Key;->x:I
  
      iget v3, p0, Landroid/inputmethodservice/Keyboard$Key;->gap:I
***************
*** 240,252 ****
--- 259,274 ----
  
      iput v2, p0, Landroid/inputmethodservice/Keyboard$Key;->x:I
  
+     .line 352
      new-instance v1, Landroid/util/TypedValue;
  
      invoke-direct {v1}, Landroid/util/TypedValue;-><init>()V
  
+     .line 353
      .local v1, codesValue:Landroid/util/TypedValue;
      invoke-virtual {v0, v5, v1}, Landroid/content/res/TypedArray;->getValue(ILandroid/util/TypedValue;)Z
  
+     .line 355
      iget v2, v1, Landroid/util/TypedValue;->type:I
  
      const/16 v3, 0x10
***************
*** 259,264 ****
--- 281,287 ----
  
      if-ne v2, v3, :cond_5
  
+     .line 357
      :cond_0
      new-array v2, v6, [I
  
***************
*** 268,273 ****
--- 291,297 ----
  
      iput-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->codes:[I
  
+     .line 362
      :cond_1
      :goto_0
      const/4 v2, 0x7
***************
*** 278,287 ****
--- 302,313 ----
  
      iput-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->iconPreview:Landroid/graphics/drawable/Drawable;
  
+     .line 363
      iget-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->iconPreview:Landroid/graphics/drawable/Drawable;
  
      if-eqz v2, :cond_2
  
+     .line 364
      iget-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->iconPreview:Landroid/graphics/drawable/Drawable;
  
      iget-object v3, p0, Landroid/inputmethodservice/Keyboard$Key;->iconPreview:Landroid/graphics/drawable/Drawable;
***************
*** 298,303 ****
--- 324,330 ----
  
      invoke-virtual {v2, v5, v5, v3, v4}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
  
+     .line 367
      :cond_2
      invoke-virtual {v0, v7}, Landroid/content/res/TypedArray;->getText(I)Ljava/lang/CharSequence;
  
***************
*** 305,316 ****
--- 332,345 ----
  
      iput-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->popupCharacters:Ljava/lang/CharSequence;
  
+     .line 369
      invoke-virtual {v0, v6, v5}, Landroid/content/res/TypedArray;->getResourceId(II)I
  
      move-result v2
  
      iput v2, p0, Landroid/inputmethodservice/Keyboard$Key;->popupResId:I
  
+     .line 371
      const/4 v2, 0x6
  
      invoke-virtual {v0, v2, v5}, Landroid/content/res/TypedArray;->getBoolean(IZ)Z
***************
*** 319,324 ****
--- 348,354 ----
  
      iput-boolean v2, p0, Landroid/inputmethodservice/Keyboard$Key;->repeatable:Z
  
+     .line 373
      const/4 v2, 0x4
  
      invoke-virtual {v0, v2, v5}, Landroid/content/res/TypedArray;->getBoolean(IZ)Z
***************
*** 327,332 ****
--- 357,363 ----
  
      iput-boolean v2, p0, Landroid/inputmethodservice/Keyboard$Key;->modifier:Z
  
+     .line 375
      const/4 v2, 0x5
  
      invoke-virtual {v0, v2, v5}, Landroid/content/res/TypedArray;->getBoolean(IZ)Z
***************
*** 335,346 ****
--- 366,379 ----
  
      iput-boolean v2, p0, Landroid/inputmethodservice/Keyboard$Key;->sticky:Z
  
+     .line 377
      invoke-virtual {v0, v8, v5}, Landroid/content/res/TypedArray;->getInt(II)I
  
      move-result v2
  
      iput v2, p0, Landroid/inputmethodservice/Keyboard$Key;->edgeFlags:I
  
+     .line 378
      iget v2, p0, Landroid/inputmethodservice/Keyboard$Key;->edgeFlags:I
  
      iget v3, p2, Landroid/inputmethodservice/Keyboard$Row;->rowEdgeFlags:I
***************
*** 349,354 ****
--- 382,388 ----
  
      iput v2, p0, Landroid/inputmethodservice/Keyboard$Key;->edgeFlags:I
  
+     .line 380
      const/16 v2, 0xa
  
      invoke-virtual {v0, v2}, Landroid/content/res/TypedArray;->getDrawable(I)Landroid/graphics/drawable/Drawable;
***************
*** 357,366 ****
--- 391,402 ----
  
      iput-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->icon:Landroid/graphics/drawable/Drawable;
  
+     .line 382
      iget-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->icon:Landroid/graphics/drawable/Drawable;
  
      if-eqz v2, :cond_3
  
+     .line 383
      iget-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->icon:Landroid/graphics/drawable/Drawable;
  
      iget-object v3, p0, Landroid/inputmethodservice/Keyboard$Key;->icon:Landroid/graphics/drawable/Drawable;
***************
*** 377,382 ****
--- 413,419 ----
  
      invoke-virtual {v2, v5, v5, v3, v4}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
  
+     .line 385
      :cond_3
      const/16 v2, 0x9
  
***************
*** 386,391 ****
--- 423,429 ----
  
      iput-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->label:Ljava/lang/CharSequence;
  
+     .line 386
      const/16 v2, 0x8
  
      invoke-virtual {v0, v2}, Landroid/content/res/TypedArray;->getText(I)Ljava/lang/CharSequence;
***************
*** 394,399 ****
--- 432,438 ----
  
      iput-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->text:Ljava/lang/CharSequence;
  
+     .line 388
      iget-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->codes:[I
  
      if-nez v2, :cond_4
***************
*** 406,411 ****
--- 445,451 ----
  
      if-nez v2, :cond_4
  
+     .line 389
      new-array v2, v6, [I
  
      iget-object v3, p0, Landroid/inputmethodservice/Keyboard$Key;->label:Ljava/lang/CharSequence;
***************
*** 418,433 ****
--- 458,477 ----
  
      iput-object v2, p0, Landroid/inputmethodservice/Keyboard$Key;->codes:[I
  
+     .line 391
      :cond_4
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 392
      return-void
  
+     .line 358
      :cond_5
      iget v2, v1, Landroid/util/TypedValue;->type:I
  
      if-ne v2, v8, :cond_1
  
+     .line 359
      iget-object v2, v1, Landroid/util/TypedValue;->string:Ljava/lang/CharSequence;
  
      invoke-virtual {v2}, Ljava/lang/Object;->toString()Ljava/lang/String;
***************
*** 448,455 ****
--- 492,501 ----
      .parameter "parent"
  
      .prologue
+     .line 313
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 314
      #getter for: Landroid/inputmethodservice/Keyboard$Row;->parent:Landroid/inputmethodservice/Keyboard;
      invoke-static {p1}, Landroid/inputmethodservice/Keyboard$Row;->access$600(Landroid/inputmethodservice/Keyboard$Row;)Landroid/inputmethodservice/Keyboard;
  
***************
*** 457,478 ****
--- 503,529 ----
  
      iput-object v0, p0, Landroid/inputmethodservice/Keyboard$Key;->keyboard:Landroid/inputmethodservice/Keyboard;
  
+     .line 315
      iget v0, p1, Landroid/inputmethodservice/Keyboard$Row;->defaultHeight:I
  
      iput v0, p0, Landroid/inputmethodservice/Keyboard$Key;->height:I
  
+     .line 316
      iget v0, p1, Landroid/inputmethodservice/Keyboard$Row;->defaultWidth:I
  
      iput v0, p0, Landroid/inputmethodservice/Keyboard$Key;->width:I
  
+     .line 317
      iget v0, p1, Landroid/inputmethodservice/Keyboard$Row;->defaultHorizontalGap:I
  
      iput v0, p0, Landroid/inputmethodservice/Keyboard$Key;->gap:I
  
+     .line 318
      iget v0, p1, Landroid/inputmethodservice/Keyboard$Row;->rowEdgeFlags:I
  
      iput v0, p0, Landroid/inputmethodservice/Keyboard$Key;->edgeFlags:I
  
+     .line 319
      return-void
  .end method
  
***************
*** 482,532 ****
--- 533,595 ----
      .locals 2
  
      .prologue
+     .line 479
      sget-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_NORMAL:[I
  
+     .line 481
      .local v0, states:[I
      iget-boolean v1, p0, Landroid/inputmethodservice/Keyboard$Key;->on:Z
  
      if-eqz v1, :cond_2
  
+     .line 482
      iget-boolean v1, p0, Landroid/inputmethodservice/Keyboard$Key;->pressed:Z
  
      if-eqz v1, :cond_1
  
+     .line 483
      sget-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_PRESSED_ON:[I
  
+     .line 500
      :cond_0
      :goto_0
      return-object v0
  
+     .line 485
      :cond_1
      sget-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_NORMAL_ON:[I
  
      goto :goto_0
  
+     .line 488
      :cond_2
      iget-boolean v1, p0, Landroid/inputmethodservice/Keyboard$Key;->sticky:Z
  
      if-eqz v1, :cond_4
  
+     .line 489
      iget-boolean v1, p0, Landroid/inputmethodservice/Keyboard$Key;->pressed:Z
  
      if-eqz v1, :cond_3
  
+     .line 490
      sget-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_PRESSED_OFF:[I
  
      goto :goto_0
  
+     .line 492
      :cond_3
      sget-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_NORMAL_OFF:[I
  
      goto :goto_0
  
+     .line 495
      :cond_4
      iget-boolean v1, p0, Landroid/inputmethodservice/Keyboard$Key;->pressed:Z
  
      if-eqz v1, :cond_0
  
+     .line 496
      sget-object v0, Landroid/inputmethodservice/Keyboard$Key;->KEY_STATE_PRESSED:[I
  
      goto :goto_0
***************
*** 542,547 ****
--- 605,611 ----
  
      const/4 v5, 0x0
  
+     .line 447
      iget v6, p0, Landroid/inputmethodservice/Keyboard$Key;->edgeFlags:I
  
      and-int/lit8 v6, v6, 0x1
***************
*** 550,555 ****
--- 614,620 ----
  
      move v1, v4
  
+     .line 448
      .local v1, leftEdge:Z
      :goto_0
      iget v6, p0, Landroid/inputmethodservice/Keyboard$Key;->edgeFlags:I
***************
*** 560,565 ****
--- 625,631 ----
  
      move v2, v4
  
+     .line 449
      .local v2, rightEdge:Z
      :goto_1
      iget v6, p0, Landroid/inputmethodservice/Keyboard$Key;->edgeFlags:I
***************
*** 570,575 ****
--- 636,642 ----
  
      move v3, v4
  
+     .line 450
      .local v3, topEdge:Z
      :goto_2
      iget v6, p0, Landroid/inputmethodservice/Keyboard$Key;->edgeFlags:I
***************
*** 580,585 ****
--- 647,653 ----
  
      move v0, v4
  
+     .line 451
      .local v0, bottomEdge:Z
      :goto_3
      iget v6, p0, Landroid/inputmethodservice/Keyboard$Key;->x:I
***************
*** 641,646 ****
--- 709,715 ----
  
      if-lt p2, v6, :cond_8
  
+     .line 457
      :cond_3
      :goto_4
      return v4
***************
*** 652,681 ****
--- 721,755 ----
      :cond_4
      move v1, v5
  
+     .line 447
      goto :goto_0
  
      .restart local v1       #leftEdge:Z
      :cond_5
      move v2, v5
  
+     .line 448
      goto :goto_1
  
      .restart local v2       #rightEdge:Z
      :cond_6
      move v3, v5
  
+     .line 449
      goto :goto_2
  
      .restart local v3       #topEdge:Z
      :cond_7
      move v0, v5
  
+     .line 450
      goto :goto_3
  
      .restart local v0       #bottomEdge:Z
      :cond_8
      move v4, v5
  
+     .line 457
      goto :goto_4
  .end method
  
***************
*** 683,688 ****
--- 757,763 ----
      .locals 1
  
      .prologue
+     .line 400
      iget-boolean v0, p0, Landroid/inputmethodservice/Keyboard$Key;->pressed:Z
  
      if-nez v0, :cond_0
***************
*** 692,699 ****
--- 767,776 ----
      :goto_0
      iput-boolean v0, p0, Landroid/inputmethodservice/Keyboard$Key;->pressed:Z
  
+     .line 401
      return-void
  
+     .line 400
      :cond_0
      const/4 v0, 0x0
  
***************
*** 709,714 ****
--- 786,792 ----
  
      const/4 v2, 0x0
  
+     .line 410
      iget-boolean v0, p0, Landroid/inputmethodservice/Keyboard$Key;->pressed:Z
  
      if-nez v0, :cond_1
***************
*** 718,727 ****
--- 796,807 ----
      :goto_0
      iput-boolean v0, p0, Landroid/inputmethodservice/Keyboard$Key;->pressed:Z
  
+     .line 411
      iget-boolean v0, p0, Landroid/inputmethodservice/Keyboard$Key;->sticky:Z
  
      if-eqz v0, :cond_0
  
+     .line 412
      iget-boolean v0, p0, Landroid/inputmethodservice/Keyboard$Key;->on:Z
  
      if-nez v0, :cond_2
***************
*** 729,745 ****
--- 809,828 ----
      :goto_1
      iput-boolean v1, p0, Landroid/inputmethodservice/Keyboard$Key;->on:Z
  
+     .line 414
      :cond_0
      return-void
  
      :cond_1
      move v0, v2
  
+     .line 410
      goto :goto_0
  
      :cond_2
      move v1, v2
  
+     .line 412
      goto :goto_1
  .end method
  
***************
*** 748,758 ****
--- 831,844 ----
      .parameter "value"
  
      .prologue
+     .line 417
      const/4 v0, 0x0
  
+     .line 418
      .local v0, count:I
      const/4 v2, 0x0
  
+     .line 419
      .local v2, lastIndex:I
      invoke-virtual {p1}, Ljava/lang/String;->length()I
  
***************
*** 760,767 ****
--- 846,855 ----
  
      if-lez v6, :cond_0
  
+     .line 420
      add-int/lit8 v0, v0, 0x1
  
+     .line 421
      :goto_0
      const-string v6, ","
  
***************
*** 773,794 ****
--- 861,887 ----
  
      if-lez v2, :cond_0
  
+     .line 422
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 425
      :cond_0
      new-array v5, v0, [I
  
+     .line 426
      .local v5, values:[I
      const/4 v0, 0x0
  
+     .line 427
      new-instance v4, Ljava/util/StringTokenizer;
  
      const-string v6, ","
  
      invoke-direct {v4, p1, v6}, Ljava/util/StringTokenizer;-><init>(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 428
      .local v4, st:Ljava/util/StringTokenizer;
      :goto_1
      invoke-virtual {v4}, Ljava/util/StringTokenizer;->hasMoreTokens()Z
***************
*** 797,802 ****
--- 890,896 ----
  
      if-eqz v6, :cond_1
  
+     .line 430
      add-int/lit8 v1, v0, 0x1
  
      .end local v0           #count:I
***************
*** 816,830 ****
--- 910,927 ----
  
      move v0, v1
  
+     .line 433
      .end local v1           #count:I
      .restart local v0       #count:I
      goto :goto_1
  
+     .line 431
      .end local v0           #count:I
      .restart local v1       #count:I
      :catch_0
      move-exception v3
  
+     .line 432
      .local v3, nfe:Ljava/lang/NumberFormatException;
      const-string v6, "Keyboard"
  
***************
*** 850,859 ****
--- 947,958 ----
  
      move v0, v1
  
+     .line 433
      .end local v1           #count:I
      .restart local v0       #count:I
      goto :goto_1
  
+     .line 435
      .end local v3           #nfe:Ljava/lang/NumberFormatException;
      :cond_1
      return-object v5
***************
*** 865,870 ****
--- 964,970 ----
      .parameter "y"
  
      .prologue
+     .line 468
      iget v2, p0, Landroid/inputmethodservice/Keyboard$Key;->x:I
  
      iget v3, p0, Landroid/inputmethodservice/Keyboard$Key;->width:I
***************
*** 875,880 ****
--- 975,981 ----
  
      sub-int v0, v2, p1
  
+     .line 469
      .local v0, xDist:I
      iget v2, p0, Landroid/inputmethodservice/Keyboard$Key;->y:I
  
***************
*** 886,891 ****
--- 987,993 ----
  
      sub-int v1, v2, p2
  
+     .line 470
      .local v1, yDist:I
      mul-int v2, v0, v0
  
