*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/GlobalActions$ToggleAction$State.smali	2012-03-23 19:35:13.899840348 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/GlobalActions$ToggleAction$State.smali.original	2012-03-23 19:34:44.403302481 -0500
***************
*** 52,57 ****
--- 52,58 ----
  
      const/4 v2, 0x0
  
+     .line 379
      new-instance v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
      const-string v1, "Off"
***************
*** 60,65 ****
--- 61,67 ----
  
      sput-object v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->Off:Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
+     .line 380
      new-instance v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
      const-string v1, "TurningOn"
***************
*** 68,73 ****
--- 70,76 ----
  
      sput-object v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->TurningOn:Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
+     .line 381
      new-instance v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
      const-string v1, "TurningOff"
***************
*** 76,81 ****
--- 79,85 ----
  
      sput-object v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->TurningOff:Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
+     .line 382
      new-instance v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
      const-string v1, "On"
***************
*** 84,89 ****
--- 88,94 ----
  
      sput-object v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->On:Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
+     .line 378
      const/4 v0, 0x4
  
      new-array v0, v0, [Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
***************
*** 121,130 ****
--- 126,138 ----
      .end annotation
  
      .prologue
+     .line 386
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
+     .line 387
      iput-boolean p3, p0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->inTransition:Z
  
+     .line 388
      return-void
  .end method
  
***************
*** 133,138 ****
--- 141,147 ----
      .parameter "name"
  
      .prologue
+     .line 378
      const-class v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 148,153 ****
--- 157,163 ----
      .locals 1
  
      .prologue
+     .line 378
      sget-object v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->$VALUES:[Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
      invoke-virtual {v0}, [Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->clone()Ljava/lang/Object;
***************
*** 165,170 ****
--- 175,181 ----
      .locals 1
  
      .prologue
+     .line 391
      iget-boolean v0, p0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->inTransition:Z
  
      return v0
