*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/GlobalActions$1.smali	2012-03-23 19:35:14.635853769 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/GlobalActions$1.smali.original	2012-03-23 19:34:44.414302681 -0500
***************
*** 29,34 ****
--- 29,35 ----
      .parameter "x4"
  
      .prologue
+     .line 125
      iput-object p1, p0, Lcom/android/internal/policy/impl/GlobalActions$1;->this$0:Lcom/android/internal/policy/impl/GlobalActions;
  
      move-object v0, p0
***************
*** 55,60 ****
--- 56,62 ----
      .parameter "buttonOn"
  
      .prologue
+     .line 144
      const-string v0, "ril.cdma.inecmmode"
  
      invoke-static {v0}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
***************
*** 67,72 ****
--- 69,75 ----
  
      if-nez v0, :cond_0
  
+     .line 146
      if-eqz p1, :cond_1
  
      sget-object v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->TurningOn:Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
***************
*** 74,79 ****
--- 77,83 ----
      :goto_0
      iput-object v0, p0, Lcom/android/internal/policy/impl/GlobalActions$1;->mState:Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
+     .line 147
      iget-object v0, p0, Lcom/android/internal/policy/impl/GlobalActions$1;->this$0:Lcom/android/internal/policy/impl/GlobalActions;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/GlobalActions$1;->mState:Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
***************
*** 81,89 ****
--- 85,95 ----
      #setter for: Lcom/android/internal/policy/impl/GlobalActions;->mAirplaneState:Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
      invoke-static {v0, v1}, Lcom/android/internal/policy/impl/GlobalActions;->access$302(Lcom/android/internal/policy/impl/GlobalActions;Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;)Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
+     .line 149
      :cond_0
      return-void
  
+     .line 146
      :cond_1
      sget-object v0, Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;->TurningOff:Lcom/android/internal/policy/impl/GlobalActions$ToggleAction$State;
  
***************
*** 95,100 ****
--- 101,107 ----
      .parameter "on"
  
      .prologue
+     .line 128
      const-string v1, "ril.cdma.inecmmode"
  
      invoke-static {v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
***************
*** 107,112 ****
--- 114,120 ----
  
      if-eqz v1, :cond_0
  
+     .line 130
      iget-object v1, p0, Lcom/android/internal/policy/impl/GlobalActions$1;->this$0:Lcom/android/internal/policy/impl/GlobalActions;
  
      const/4 v2, 0x1
***************
*** 114,119 ****
--- 122,128 ----
      #setter for: Lcom/android/internal/policy/impl/GlobalActions;->mIsWaitingForEcmExit:Z
      invoke-static {v1, v2}, Lcom/android/internal/policy/impl/GlobalActions;->access$002(Lcom/android/internal/policy/impl/GlobalActions;Z)Z
  
+     .line 132
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.intent.action.ACTION_SHOW_NOTICE_ECM_BLOCK_OTHERS"
***************
*** 122,132 ****
--- 131,143 ----
  
      invoke-direct {v0, v1, v2}, Landroid/content/Intent;-><init>(Ljava/lang/String;Landroid/net/Uri;)V
  
+     .line 134
      .local v0, ecmDialogIntent:Landroid/content/Intent;
      const/high16 v1, 0x1000
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 135
      iget-object v1, p0, Lcom/android/internal/policy/impl/GlobalActions$1;->this$0:Lcom/android/internal/policy/impl/GlobalActions;
  
      #getter for: Lcom/android/internal/policy/impl/GlobalActions;->mContext:Landroid/content/Context;
***************
*** 136,145 ****
--- 147,158 ----
  
      invoke-virtual {v1, v0}, Landroid/content/Context;->startActivity(Landroid/content/Intent;)V
  
+     .line 139
      .end local v0           #ecmDialogIntent:Landroid/content/Intent;
      :goto_0
      return-void
  
+     .line 137
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/GlobalActions$1;->this$0:Lcom/android/internal/policy/impl/GlobalActions;
  
***************
*** 153,158 ****
--- 166,172 ----
      .locals 1
  
      .prologue
+     .line 156
      const/4 v0, 0x0
  
      return v0
***************
*** 162,167 ****
--- 176,182 ----
      .locals 1
  
      .prologue
+     .line 152
      const/4 v0, 0x1
  
      return v0
