*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/KeyguardViewManager$KeyguardViewHost.smali	2012-03-23 19:35:14.937859277 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/KeyguardViewManager$KeyguardViewHost.smali.original	2012-03-23 19:34:44.420302791 -0500
***************
*** 25,34 ****
--- 25,37 ----
      .parameter "callback"
  
      .prologue
+     .line 90
      invoke-direct {p0, p1}, Landroid/widget/FrameLayout;-><init>(Landroid/content/Context;)V
  
+     .line 91
      iput-object p2, p0, Lcom/android/internal/policy/impl/KeyguardViewManager$KeyguardViewHost;->mCallback:Lcom/android/internal/policy/impl/KeyguardViewCallback;
  
+     .line 92
      return-void
  .end method
  
***************
*** 39,44 ****
--- 42,48 ----
      .parameter "x2"
  
      .prologue
+     .line 86
      invoke-direct {p0, p1, p2}, Lcom/android/internal/policy/impl/KeyguardViewManager$KeyguardViewHost;-><init>(Landroid/content/Context;Lcom/android/internal/policy/impl/KeyguardViewCallback;)V
  
      return-void
***************
*** 51,61 ****
--- 55,68 ----
      .parameter "canvas"
  
      .prologue
+     .line 96
      invoke-super {p0, p1}, Landroid/widget/FrameLayout;->dispatchDraw(Landroid/graphics/Canvas;)V
  
+     .line 97
      iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewManager$KeyguardViewHost;->mCallback:Lcom/android/internal/policy/impl/KeyguardViewCallback;
  
      invoke-interface {v0}, Lcom/android/internal/policy/impl/KeyguardViewCallback;->keyguardDoneDrawing()V
  
+     .line 98
      return-void
  .end method
