*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/KeyguardStatusViewManager$3.smali	2012-03-23 19:35:14.891858438 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/KeyguardStatusViewManager$3.smali.original	2012-03-23 19:34:44.419302772 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 3
  
      .prologue
+     .line 472
      invoke-static {}, Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;->values()[Lcom/android/internal/policy/impl/KeyguardStatusViewManager$StatusMode;
  
      move-result-object v0
***************
*** 146,151 ****
--- 147,153 ----
      :try_end_6
      .catch Ljava/lang/NoSuchFieldError; {:try_start_6 .. :try_end_6} :catch_8
  
+     .line 433
      :goto_6
      invoke-static {}, Lcom/android/internal/telephony/IccCard$State;->values()[Lcom/android/internal/telephony/IccCard$State;
  
***************
*** 327,332 ****
--- 329,335 ----
  
      goto :goto_7
  
+     .line 472
      :catch_8
      move-exception v0
  
