*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/LockPatternKeyguardView$4.smali	2012-03-23 19:35:13.429831778 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/LockPatternKeyguardView$4.smali.original	2012-03-23 19:34:44.397302371 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1279
      iput-object p1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,47 ****
--- 43,49 ----
      .parameter "iservice"
  
      .prologue
+     .line 1283
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      invoke-static {p2}, Lcom/android/internal/policy/IFaceLockInterface$Stub;->asInterface(Landroid/os/IBinder;)Lcom/android/internal/policy/IFaceLockInterface;
***************
*** 51,56 ****
--- 53,59 ----
      #setter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mFaceLockService:Lcom/android/internal/policy/IFaceLockInterface;
      invoke-static {v0, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->access$2302(Lcom/android/internal/policy/impl/LockPatternKeyguardView;Lcom/android/internal/policy/IFaceLockInterface;)Lcom/android/internal/policy/IFaceLockInterface;
  
+     .line 1286
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
***************
*** 70,75 ****
--- 73,79 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 1294
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      #getter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mFaceLockAreaView:Landroid/view/View;
***************
*** 79,84 ****
--- 83,89 ----
  
      if-eqz v0, :cond_0
  
+     .line 1295
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
***************
*** 138,150 ****
--- 143,158 ----
  
      invoke-virtual/range {v0 .. v5}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->startFaceLock(Landroid/os/IBinder;IIII)V
  
+     .line 1299
      :cond_0
      :goto_0
      return-void
  
+     .line 1287
      :catch_0
      move-exception v6
  
+     .line 1288
      .local v6, e:Landroid/os/RemoteException;
      const-string v0, "LockPatternKeyguardView"
  
***************
*** 172,177 ****
--- 180,186 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1289
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      const/4 v1, 0x0
***************
*** 179,184 ****
--- 188,194 ----
      #setter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mFaceLockService:Lcom/android/internal/policy/IFaceLockInterface;
      invoke-static {v0, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->access$2302(Lcom/android/internal/policy/impl/LockPatternKeyguardView;Lcom/android/internal/policy/IFaceLockInterface;)Lcom/android/internal/policy/IFaceLockInterface;
  
+     .line 1290
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      const/4 v1, 0x0
***************
*** 196,201 ****
--- 206,212 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 1304
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      #getter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mFaceLockServiceRunningLock:Ljava/lang/Object;
***************
*** 205,210 ****
--- 216,222 ----
  
      monitor-enter v1
  
+     .line 1305
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
***************
*** 213,218 ****
--- 225,231 ----
      #setter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mFaceLockService:Lcom/android/internal/policy/IFaceLockInterface;
      invoke-static {v0, v2}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->access$2302(Lcom/android/internal/policy/impl/LockPatternKeyguardView;Lcom/android/internal/policy/IFaceLockInterface;)Lcom/android/internal/policy/IFaceLockInterface;
  
+     .line 1306
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      const/4 v2, 0x0
***************
*** 220,242 ****
--- 233,260 ----
      #setter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mFaceLockServiceRunning:Z
      invoke-static {v0, v2}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->access$1302(Lcom/android/internal/policy/impl/LockPatternKeyguardView;Z)Z
  
+     .line 1307
      monitor-exit v1
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1308
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$4;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      #setter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mBoundToFaceLockService:Z
      invoke-static {v0, v3}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->access$2502(Lcom/android/internal/policy/impl/LockPatternKeyguardView;Z)Z
  
+     .line 1309
      const-string v0, "LockPatternKeyguardView"
  
      const-string v1, "Unexpected disconnect from FaceLock service"
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 1310
      return-void
  
+     .line 1307
      :catchall_0
      move-exception v0
  
