*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/LockScreen$MultiWaveViewMethods.smali	2012-03-23 19:35:15.279865513 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/LockScreen$MultiWaveViewMethods.smali.original	2012-03-23 19:34:44.425302881 -0500
***************
*** 35,46 ****
--- 35,49 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 189
      iput-object p1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->this$0:Lcom/android/internal/policy/impl/LockScreen;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 190
      iput-object p2, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->mMultiWaveView:Lcom/android/internal/widget/multiwaveview/MultiWaveView;
  
+     .line 191
      #getter for: Lcom/android/internal/policy/impl/LockScreen;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
      invoke-static {p1}, Lcom/android/internal/policy/impl/LockScreen;->access$600(Lcom/android/internal/policy/impl/LockScreen;)Lcom/android/internal/widget/LockPatternUtils;
  
***************
*** 56,75 ****
--- 59,83 ----
  
      move-result v0
  
+     .line 193
      .local v0, cameraDisabled:Z
      if-eqz v0, :cond_0
  
+     .line 194
      const-string v2, "LockScreen"
  
      const-string v3, "Camera disabled by Device Policy"
  
      invoke-static {v2, v3}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 195
      iput-boolean v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->mCameraDisabled:Z
  
+     .line 202
      :goto_0
      return-void
  
+     .line 199
      :cond_0
      iget-object v2, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->mMultiWaveView:Lcom/android/internal/widget/multiwaveview/MultiWaveView;
  
***************
*** 98,103 ****
--- 106,112 ----
      .locals 1
  
      .prologue
+     .line 252
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->mMultiWaveView:Lcom/android/internal/widget/multiwaveview/MultiWaveView;
  
      return-object v0
***************
*** 109,114 ****
--- 118,124 ----
      .parameter "handle"
  
      .prologue
+     .line 218
      return-void
  .end method
  
***************
*** 118,125 ****
--- 128,137 ----
      .parameter "handle"
  
      .prologue
+     .line 246
      if-eqz p2, :cond_0
  
+     .line 247
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->this$0:Lcom/android/internal/policy/impl/LockScreen;
  
      #getter for: Lcom/android/internal/policy/impl/LockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
***************
*** 129,134 ****
--- 141,147 ----
  
      invoke-interface {v0}, Lcom/android/internal/policy/impl/KeyguardScreenCallback;->pokeWakelock()V
  
+     .line 249
      :cond_0
      return-void
  .end method
***************
*** 139,144 ****
--- 152,158 ----
      .parameter "handle"
  
      .prologue
+     .line 222
      return-void
  .end method
  
***************
*** 148,159 ****
--- 162,175 ----
      .parameter "target"
  
      .prologue
+     .line 225
      if-eqz p2, :cond_0
  
      const/4 v1, 0x1
  
      if-ne p2, v1, :cond_2
  
+     .line 226
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->this$0:Lcom/android/internal/policy/impl/LockScreen;
  
***************
*** 164,173 ****
--- 180,191 ----
  
      invoke-interface {v1}, Lcom/android/internal/policy/impl/KeyguardScreenCallback;->goToUnlockScreen()V
  
+     .line 240
      :cond_1
      :goto_0
      return-void
  
+     .line 227
      :cond_2
      const/4 v1, 0x2
  
***************
*** 177,198 ****
--- 195,220 ----
  
      if-ne p2, v1, :cond_1
  
+     .line 228
      :cond_3
      iget-boolean v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->mCameraDisabled:Z
  
      if-nez v1, :cond_4
  
+     .line 230
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.media.action.STILL_IMAGE_CAMERA"
  
      invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 231
      .local v0, intent:Landroid/content/Intent;
      const/high16 v1, 0x1000
  
      invoke-virtual {v0, v1}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 232
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->this$0:Lcom/android/internal/policy/impl/LockScreen;
  
      #getter for: Lcom/android/internal/policy/impl/LockScreen;->mContext:Landroid/content/Context;
***************
*** 202,207 ****
--- 224,230 ----
  
      invoke-virtual {v1, v0}, Landroid/content/Context;->startActivity(Landroid/content/Intent;)V
  
+     .line 233
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->this$0:Lcom/android/internal/policy/impl/LockScreen;
  
      #getter for: Lcom/android/internal/policy/impl/LockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
***************
*** 213,218 ****
--- 236,242 ----
  
      goto :goto_0
  
+     .line 235
      .end local v0           #intent:Landroid/content/Intent;
      :cond_4
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->this$0:Lcom/android/internal/policy/impl/LockScreen;
***************
*** 220,225 ****
--- 244,250 ----
      #calls: Lcom/android/internal/policy/impl/LockScreen;->toggleRingMode()V
      invoke-static {v1}, Lcom/android/internal/policy/impl/LockScreen;->access$300(Lcom/android/internal/policy/impl/LockScreen;)V
  
+     .line 236
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->this$0:Lcom/android/internal/policy/impl/LockScreen;
  
      #getter for: Lcom/android/internal/policy/impl/LockScreen;->mUnlockWidgetMethods:Lcom/android/internal/policy/impl/LockScreen$UnlockWidgetCommonMethods;
***************
*** 229,234 ****
--- 254,260 ----
  
      invoke-interface {v1}, Lcom/android/internal/policy/impl/LockScreen$UnlockWidgetCommonMethods;->updateResources()V
  
+     .line 237
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->this$0:Lcom/android/internal/policy/impl/LockScreen;
  
      #getter for: Lcom/android/internal/policy/impl/LockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
***************
*** 245,254 ****
--- 271,282 ----
      .locals 1
  
      .prologue
+     .line 260
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->mMultiWaveView:Lcom/android/internal/widget/multiwaveview/MultiWaveView;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/multiwaveview/MultiWaveView;->ping()V
  
+     .line 261
      return-void
  .end method
  
***************
*** 257,266 ****
--- 285,296 ----
      .parameter "animate"
  
      .prologue
+     .line 256
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->mMultiWaveView:Lcom/android/internal/widget/multiwaveview/MultiWaveView;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/widget/multiwaveview/MultiWaveView;->reset(Z)V
  
+     .line 257
      return-void
  .end method
  
***************
*** 268,277 ****
--- 298,309 ----
      .locals 2
  
      .prologue
+     .line 206
      iget-boolean v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->mCameraDisabled:Z
  
      if-eqz v1, :cond_1
  
+     .line 208
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->this$0:Lcom/android/internal/policy/impl/LockScreen;
  
      #getter for: Lcom/android/internal/policy/impl/LockScreen;->mSilentMode:Z
***************
*** 283,302 ****
--- 315,338 ----
  
      const v0, 0x107000b
  
+     .line 213
      .local v0, resId:I
      :goto_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockScreen$MultiWaveViewMethods;->mMultiWaveView:Lcom/android/internal/widget/multiwaveview/MultiWaveView;
  
      invoke-virtual {v1, v0}, Lcom/android/internal/widget/multiwaveview/MultiWaveView;->setTargetResources(I)V
  
+     .line 214
      return-void
  
+     .line 208
      .end local v0           #resId:I
      :cond_0
      const v0, 0x107000e
  
      goto :goto_0
  
+     .line 211
      :cond_1
      const v0, 0x1070010
  
