*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PasswordUnlockScreen.smali	2012-03-23 19:35:15.989878461 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PasswordUnlockScreen.smali.original	2012-03-23 19:34:44.436303082 -0500
***************
*** 49,88 ****
--- 49,97 ----
      .parameter "callback"
  
      .prologue
+     .line 88
      invoke-direct {p0, p1}, Landroid/widget/LinearLayout;-><init>(Landroid/content/Context;)V
  
+     .line 78
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mUseSystemIME:Z
  
+     .line 90
      move-object/from16 v0, p2
  
      iget v1, v0, Landroid/content/res/Configuration;->hardKeyboardHidden:I
  
      iput v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCreationHardKeyboardHidden:I
  
+     .line 91
      move-object/from16 v0, p2
  
      iget v1, v0, Landroid/content/res/Configuration;->orientation:I
  
      iput v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCreationOrientation:I
  
+     .line 92
      move-object/from16 v0, p4
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
+     .line 93
      move-object/from16 v0, p5
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
+     .line 94
      move-object/from16 v0, p3
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
+     .line 96
      invoke-static {p1}, Landroid/view/LayoutInflater;->from(Landroid/content/Context;)Landroid/view/LayoutInflater;
  
      move-result-object v9
  
+     .line 97
      .local v9, layoutInflater:Landroid/view/LayoutInflater;
      iget v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCreationOrientation:I
  
***************
*** 90,101 ****
--- 99,112 ----
  
      if-eq v1, v2, :cond_4
  
+     .line 98
      const v1, 0x109004f
  
      const/4 v2, 0x1
  
      invoke-virtual {v9, v1, p0, v2}, Landroid/view/LayoutInflater;->inflate(ILandroid/view/ViewGroup;Z)Landroid/view/View;
  
+     .line 103
      :goto_0
      new-instance v1, Lcom/android/internal/policy/impl/KeyguardStatusViewManager;
  
***************
*** 113,122 ****
--- 124,135 ----
  
      iput-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mStatusViewManager:Lcom/android/internal/policy/impl/KeyguardStatusViewManager;
  
+     .line 106
      invoke-virtual/range {p3 .. p3}, Lcom/android/internal/widget/LockPatternUtils;->getKeyguardStoredPasswordQuality()I
  
      move-result v12
  
+     .line 107
      .local v12, quality:I
      const/high16 v1, 0x4
  
***************
*** 136,141 ****
--- 149,155 ----
      :goto_1
      iput-boolean v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mIsAlpha:Z
  
+     .line 111
      const v1, 0x10201ff
  
      invoke-virtual {p0, v1}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->findViewById(I)Landroid/view/View;
***************
*** 146,151 ****
--- 160,166 ----
  
      iput-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardView:Lcom/android/internal/widget/PasswordEntryKeyboardView;
  
+     .line 112
      const v1, 0x102029a
  
      invoke-virtual {p0, v1}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->findViewById(I)Landroid/view/View;
***************
*** 156,165 ****
--- 171,182 ----
  
      iput-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
+     .line 113
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      invoke-virtual {v1, p0}, Landroid/widget/EditText;->setOnEditorActionListener(Landroid/widget/TextView$OnEditorActionListener;)V
  
+     .line 115
      new-instance v1, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;
  
      iget-object v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardView:Lcom/android/internal/widget/PasswordEntryKeyboardView;
***************
*** 170,175 ****
--- 187,193 ----
  
      iput-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardHelper:Lcom/android/internal/widget/PasswordEntryKeyboardHelper;
  
+     .line 116
      iget-object v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardHelper:Lcom/android/internal/widget/PasswordEntryKeyboardHelper;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mContext:Landroid/content/Context;
***************
*** 193,227 ****
--- 211,252 ----
      :goto_2
      invoke-virtual {v2, v1}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->setEnableHaptics(Z)V
  
+     .line 120
      const/4 v7, 0x0
  
+     .line 121
      .local v7, imeOrDeleteButtonVisible:Z
      iget-boolean v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mIsAlpha:Z
  
      if-eqz v1, :cond_7
  
+     .line 123
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardHelper:Lcom/android/internal/widget/PasswordEntryKeyboardHelper;
  
      const/4 v2, 0x0
  
      invoke-virtual {v1, v2}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->setKeyboardMode(I)V
  
+     .line 124
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardView:Lcom/android/internal/widget/PasswordEntryKeyboardView;
  
      const/16 v2, 0x8
  
      invoke-virtual {v1, v2}, Lcom/android/internal/widget/PasswordEntryKeyboardView;->setVisibility(I)V
  
+     .line 146
      :cond_1
      :goto_3
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      invoke-virtual {v1}, Landroid/widget/EditText;->requestFocus()Z
  
+     .line 149
      iget-boolean v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mIsAlpha:Z
  
      if-eqz v1, :cond_9
  
+     .line 150
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      invoke-static {}, Landroid/text/method/TextKeyListener;->getInstance()Landroid/text/method/TextKeyListener;
***************
*** 230,241 ****
--- 255,268 ----
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setKeyListener(Landroid/text/method/KeyListener;)V
  
+     .line 151
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      const/16 v2, 0x81
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setInputType(I)V
  
+     .line 164
      :goto_4
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
***************
*** 245,250 ****
--- 272,278 ----
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setOnClickListener(Landroid/view/View$OnClickListener;)V
  
+     .line 169
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      new-instance v2, Lcom/android/internal/policy/impl/PasswordUnlockScreen$3;
***************
*** 253,264 ****
--- 281,294 ----
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->addTextChangedListener(Landroid/text/TextWatcher;)V
  
+     .line 184
      const v1, 0x102029c
  
      invoke-virtual {p0, v1}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->findViewById(I)Landroid/view/View;
  
      move-result-object v13
  
+     .line 185
      .local v13, switchImeButton:Landroid/view/View;
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->getContext()Landroid/content/Context;
  
***************
*** 272,277 ****
--- 302,308 ----
  
      check-cast v8, Landroid/view/inputmethod/InputMethodManager;
  
+     .line 187
      .local v8, imm:Landroid/view/inputmethod/InputMethodManager;
      iget-boolean v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mIsAlpha:Z
  
***************
*** 287,313 ****
--- 318,350 ----
  
      if-eqz v1, :cond_2
  
+     .line 188
      const/4 v1, 0x0
  
      invoke-virtual {v13, v1}, Landroid/view/View;->setVisibility(I)V
  
+     .line 189
      const/4 v7, 0x1
  
+     .line 190
      new-instance v1, Lcom/android/internal/policy/impl/PasswordUnlockScreen$4;
  
      invoke-direct {v1, p0, v8}, Lcom/android/internal/policy/impl/PasswordUnlockScreen$4;-><init>(Lcom/android/internal/policy/impl/PasswordUnlockScreen;Landroid/view/inputmethod/InputMethodManager;)V
  
      invoke-virtual {v13, v1}, Landroid/view/View;->setOnClickListener(Landroid/view/View$OnClickListener;)V
  
+     .line 200
      :cond_2
      if-nez v7, :cond_3
  
+     .line 201
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      invoke-virtual {v1}, Landroid/widget/EditText;->getLayoutParams()Landroid/view/ViewGroup$LayoutParams;
  
      move-result-object v10
  
+     .line 202
      .local v10, params:Landroid/view/ViewGroup$LayoutParams;
      instance-of v1, v10, Landroid/view/ViewGroup$MarginLayoutParams;
  
***************
*** 315,334 ****
--- 352,375 ----
  
      move-object v1, v10
  
+     .line 203
      check-cast v1, Landroid/view/ViewGroup$MarginLayoutParams;
  
      const/4 v2, 0x0
  
      iput v2, v1, Landroid/view/ViewGroup$MarginLayoutParams;->leftMargin:I
  
+     .line 204
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      invoke-virtual {v1, v10}, Landroid/widget/EditText;->setLayoutParams(Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 207
      .end local v10           #params:Landroid/view/ViewGroup$LayoutParams;
      :cond_3
      return-void
  
+     .line 100
      .end local v7           #imeOrDeleteButtonVisible:Z
      .end local v8           #imm:Landroid/view/inputmethod/InputMethodManager;
      .end local v12           #quality:I
***************
*** 342,358 ****
--- 383,402 ----
  
      goto/16 :goto_0
  
+     .line 107
      .restart local v12       #quality:I
      :cond_5
      const/4 v1, 0x0
  
      goto/16 :goto_1
  
+     .line 116
      :cond_6
      const/4 v1, 0x0
  
      goto/16 :goto_2
  
+     .line 127
      .restart local v7       #imeOrDeleteButtonVisible:Z
      :cond_7
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardHelper:Lcom/android/internal/widget/PasswordEntryKeyboardHelper;
***************
*** 361,366 ****
--- 405,411 ----
  
      invoke-virtual {v1, v2}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->setKeyboardMode(I)V
  
+     .line 128
      iget-object v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardView:Lcom/android/internal/widget/PasswordEntryKeyboardView;
  
      iget v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCreationHardKeyboardHidden:I
***************
*** 374,394 ****
--- 419,444 ----
      :goto_5
      invoke-virtual {v2, v1}, Lcom/android/internal/widget/PasswordEntryKeyboardView;->setVisibility(I)V
  
+     .line 133
      const v1, 0x102029b
  
      invoke-virtual {p0, v1}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->findViewById(I)Landroid/view/View;
  
      move-result-object v11
  
+     .line 134
      .local v11, pinDelete:Landroid/view/View;
      if-eqz v11, :cond_1
  
+     .line 135
      const/4 v1, 0x0
  
      invoke-virtual {v11, v1}, Landroid/view/View;->setVisibility(I)V
  
+     .line 136
      const/4 v7, 0x1
  
+     .line 137
      new-instance v1, Lcom/android/internal/policy/impl/PasswordUnlockScreen$1;
  
      invoke-direct {v1, p0}, Lcom/android/internal/policy/impl/PasswordUnlockScreen$1;-><init>(Lcom/android/internal/policy/impl/PasswordUnlockScreen;)V
***************
*** 397,408 ****
--- 447,460 ----
  
      goto/16 :goto_3
  
+     .line 128
      .end local v11           #pinDelete:Landroid/view/View;
      :cond_8
      const/4 v1, 0x0
  
      goto :goto_5
  
+     .line 156
      :cond_9
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
***************
*** 412,417 ****
--- 464,470 ----
  
      invoke-virtual {v1, v2}, Landroid/widget/EditText;->setKeyListener(Landroid/text/method/KeyListener;)V
  
+     .line 157
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      const/16 v2, 0x12
***************
*** 426,431 ****
--- 479,485 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardHelper:Lcom/android/internal/widget/PasswordEntryKeyboardHelper;
  
      return-object v0
***************
*** 436,441 ****
--- 490,496 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
      return-object v0
***************
*** 446,451 ****
--- 501,507 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-boolean v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mResuming:Z
  
      return v0
***************
*** 456,461 ****
--- 512,518 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mStatusViewManager:Lcom/android/internal/policy/impl/KeyguardStatusViewManager;
  
      return-object v0
***************
*** 466,471 ****
--- 523,529 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      return-object v0
***************
*** 476,481 ****
--- 534,540 ----
      .parameter "x0"
  
      .prologue
+     .line 60
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardView:Lcom/android/internal/widget/PasswordEntryKeyboardView;
  
      return-object v0
***************
*** 488,505 ****
--- 547,568 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 323
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      invoke-virtual {v0, v1}, Landroid/widget/EditText;->setEnabled(Z)V
  
+     .line 324
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardView:Lcom/android/internal/widget/PasswordEntryKeyboardView;
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/PasswordEntryKeyboardView;->setEnabled(Z)V
  
+     .line 325
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v6
  
+     .line 326
      .local v6, elapsedRealtime:J
      new-instance v0, Lcom/android/internal/policy/impl/PasswordUnlockScreen$5;
  
***************
*** 513,518 ****
--- 576,582 ----
  
      invoke-virtual {v0}, Lcom/android/internal/policy/impl/PasswordUnlockScreen$5;->start()Landroid/os/CountDownTimer;
  
+     .line 344
      return-void
  .end method
  
***************
*** 526,538 ****
--- 590,605 ----
  
      const/4 v10, 0x1
  
+     .line 218
      invoke-virtual {p1}, Landroid/view/inputmethod/InputMethodManager;->getEnabledInputMethodList()Ljava/util/List;
  
      move-result-object v1
  
+     .line 221
      .local v1, enabledImis:Ljava/util/List;,"Ljava/util/List<Landroid/view/inputmethod/InputMethodInfo;>;"
      const/4 v2, 0x0
  
+     .line 223
      .local v2, filteredImisCount:I
      invoke-interface {v1}, Ljava/util/List;->iterator()Ljava/util/Iterator;
  
***************
*** 552,570 ****
--- 619,641 ----
  
      check-cast v5, Landroid/view/inputmethod/InputMethodInfo;
  
+     .line 225
      .local v5, imi:Landroid/view/inputmethod/InputMethodInfo;
      if-le v2, v10, :cond_1
  
+     .line 251
      .end local v5           #imi:Landroid/view/inputmethod/InputMethodInfo;
      :goto_1
      return v10
  
+     .line 226
      .restart local v5       #imi:Landroid/view/inputmethod/InputMethodInfo;
      :cond_1
      invoke-virtual {p1, v5, v10}, Landroid/view/inputmethod/InputMethodManager;->getEnabledInputMethodSubtypeList(Landroid/view/inputmethod/InputMethodInfo;Z)Ljava/util/List;
  
      move-result-object v8
  
+     .line 229
      .local v8, subtypes:Ljava/util/List;,"Ljava/util/List<Landroid/view/inputmethod/InputMethodSubtype;>;"
      invoke-interface {v8}, Ljava/util/List;->isEmpty()Z
  
***************
*** 572,584 ****
--- 643,659 ----
  
      if-eqz v11, :cond_2
  
+     .line 230
      add-int/lit8 v2, v2, 0x1
  
+     .line 231
      goto :goto_0
  
+     .line 234
      :cond_2
      const/4 v0, 0x0
  
+     .line 235
      .local v0, auxCount:I
      invoke-interface {v8}, Ljava/util/List;->iterator()Ljava/util/Iterator;
  
***************
*** 599,604 ****
--- 674,680 ----
  
      check-cast v7, Landroid/view/inputmethod/InputMethodSubtype;
  
+     .line 236
      .local v7, subtype:Landroid/view/inputmethod/InputMethodSubtype;
      invoke-virtual {v7}, Landroid/view/inputmethod/InputMethodSubtype;->isAuxiliary()Z
  
***************
*** 606,615 ****
--- 682,693 ----
  
      if-eqz v11, :cond_3
  
+     .line 237
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_2
  
+     .line 240
      .end local v7           #subtype:Landroid/view/inputmethod/InputMethodSubtype;
      :cond_4
      invoke-interface {v8}, Ljava/util/List;->size()I
***************
*** 618,623 ****
--- 696,702 ----
  
      sub-int v6, v11, v0
  
+     .line 245
      .local v6, nonAuxCount:I
      if-gtz v6, :cond_5
  
***************
*** 625,635 ****
--- 704,717 ----
  
      if-le v0, v10, :cond_0
  
+     .line 246
      :cond_5
      add-int/lit8 v2, v2, 0x1
  
+     .line 247
      goto :goto_0
  
+     .line 251
      .end local v0           #auxCount:I
      .end local v4           #i$:Ljava/util/Iterator;
      .end local v5           #imi:Landroid/view/inputmethod/InputMethodInfo;
***************
*** 665,670 ****
--- 747,753 ----
      .prologue
      const v5, 0x10402e4
  
+     .line 297
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      invoke-virtual {v3}, Landroid/widget/EditText;->getText()Landroid/text/Editable;
***************
*** 675,680 ****
--- 758,764 ----
  
      move-result-object v2
  
+     .line 298
      .local v2, entry:Ljava/lang/String;
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
***************
*** 684,711 ****
--- 768,800 ----
  
      if-eqz v3, :cond_1
  
+     .line 299
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
      const/4 v4, 0x1
  
      invoke-interface {v3, v4}, Lcom/android/internal/policy/impl/KeyguardScreenCallback;->keyguardDone(Z)V
  
+     .line 300
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
      invoke-interface {v3}, Lcom/android/internal/policy/impl/KeyguardScreenCallback;->reportSuccessfulUnlockAttempt()V
  
+     .line 301
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mStatusViewManager:Lcom/android/internal/policy/impl/KeyguardStatusViewManager;
  
      const/4 v4, 0x0
  
      invoke-virtual {v3, v4}, Lcom/android/internal/policy/impl/KeyguardStatusViewManager;->setInstructionText(Ljava/lang/String;)V
  
+     .line 302
      invoke-static {}, Landroid/security/KeyStore;->getInstance()Landroid/security/KeyStore;
  
      move-result-object v3
  
      invoke-virtual {v3, v2}, Landroid/security/KeyStore;->password(Ljava/lang/String;)Z
  
+     .line 318
      :cond_0
      :goto_0
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
***************
*** 714,721 ****
--- 803,812 ----
  
      invoke-virtual {v3, v4}, Landroid/widget/EditText;->setText(Ljava/lang/CharSequence;)V
  
+     .line 319
      return-void
  
+     .line 303
      :cond_1
      invoke-virtual {v2}, Ljava/lang/String;->length()I
  
***************
*** 725,734 ****
--- 816,827 ----
  
      if-le v3, v4, :cond_3
  
+     .line 306
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
      invoke-interface {v3}, Lcom/android/internal/policy/impl/KeyguardScreenCallback;->reportFailedUnlockAttempt()V
  
+     .line 307
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
      invoke-virtual {v3}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getFailedAttempts()I
***************
*** 739,753 ****
--- 832,849 ----
  
      if-nez v3, :cond_2
  
+     .line 309
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
      invoke-virtual {v3}, Lcom/android/internal/widget/LockPatternUtils;->setLockoutAttemptDeadline()J
  
      move-result-wide v0
  
+     .line 310
      .local v0, deadline:J
      invoke-direct {p0, v0, v1}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->handleAttemptLockout(J)V
  
+     .line 312
      .end local v0           #deadline:J
      :cond_2
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mStatusViewManager:Lcom/android/internal/policy/impl/KeyguardStatusViewManager;
***************
*** 762,767 ****
--- 858,864 ----
  
      goto :goto_0
  
+     .line 314
      :cond_3
      invoke-virtual {v2}, Ljava/lang/String;->length()I
  
***************
*** 769,774 ****
--- 866,872 ----
  
      if-lez v3, :cond_0
  
+     .line 315
      iget-object v3, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mStatusViewManager:Lcom/android/internal/policy/impl/KeyguardStatusViewManager;
  
      iget-object v4, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mContext:Landroid/content/Context;
***************
*** 788,797 ****
--- 886,897 ----
      .locals 1
  
      .prologue
+     .line 293
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
      invoke-virtual {v0, p0}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->removeCallback(Ljava/lang/Object;)V
  
+     .line 294
      return-void
  .end method
  
***************
*** 799,804 ****
--- 899,905 ----
      .locals 1
  
      .prologue
+     .line 265
      iget-boolean v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mIsAlpha:Z
  
      return v0
***************
*** 808,815 ****
--- 909,918 ----
      .locals 3
  
      .prologue
+     .line 354
      invoke-super {p0}, Landroid/widget/LinearLayout;->onAttachedToWindow()V
  
+     .line 355
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->getResources()Landroid/content/res/Resources;
  
      move-result-object v1
***************
*** 818,823 ****
--- 921,927 ----
  
      move-result-object v0
  
+     .line 356
      .local v0, config:Landroid/content/res/Configuration;
      iget v1, v0, Landroid/content/res/Configuration;->orientation:I
  
***************
*** 831,841 ****
--- 935,947 ----
  
      if-eq v1, v2, :cond_1
  
+     .line 358
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
      invoke-interface {v1, v0}, Lcom/android/internal/policy/impl/KeyguardScreenCallback;->recreateMe(Landroid/content/res/Configuration;)V
  
+     .line 360
      :cond_1
      return-void
  .end method
***************
*** 845,852 ****
--- 951,960 ----
      .parameter "newConfig"
  
      .prologue
+     .line 365
      invoke-super {p0, p1}, Landroid/widget/LinearLayout;->onConfigurationChanged(Landroid/content/res/Configuration;)V
  
+     .line 366
      iget v0, p1, Landroid/content/res/Configuration;->orientation:I
  
      iget v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCreationOrientation:I
***************
*** 859,869 ****
--- 967,979 ----
  
      if-eq v0, v1, :cond_1
  
+     .line 368
      :cond_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
      invoke-interface {v0, p1}, Lcom/android/internal/policy/impl/KeyguardScreenCallback;->recreateMe(Landroid/content/res/Configuration;)V
  
+     .line 370
      :cond_1
      return-void
  .end method
***************
*** 875,880 ****
--- 985,991 ----
      .parameter "event"
  
      .prologue
+     .line 379
      if-eqz p2, :cond_0
  
      const/4 v0, 0x6
***************
*** 885,895 ****
--- 996,1009 ----
  
      if-ne p2, v0, :cond_1
  
+     .line 381
      :cond_0
      invoke-direct {p0}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->verifyPasswordAndUnlock()V
  
+     .line 382
      const/4 v0, 0x1
  
+     .line 384
      :goto_0
      return v0
  
***************
*** 905,914 ****
--- 1019,1030 ----
      .parameter "event"
  
      .prologue
+     .line 348
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
      invoke-interface {v0}, Lcom/android/internal/policy/impl/KeyguardScreenCallback;->pokeWakelock()V
  
+     .line 349
      const/4 v0, 0x0
  
      return v0
***************
*** 919,924 ****
--- 1035,1041 ----
      .parameter "isKeyboardOpen"
  
      .prologue
+     .line 374
      iget-object v1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardView:Lcom/android/internal/widget/PasswordEntryKeyboardView;
  
      if-eqz p1, :cond_0
***************
*** 928,935 ****
--- 1045,1054 ----
      :goto_0
      invoke-virtual {v1, v0}, Lcom/android/internal/widget/PasswordEntryKeyboardView;->setVisibility(I)V
  
+     .line 375
      return-void
  
+     .line 374
      :cond_0
      const/4 v0, 0x0
  
***************
*** 940,949 ****
--- 1059,1070 ----
      .locals 1
  
      .prologue
+     .line 270
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mStatusViewManager:Lcom/android/internal/policy/impl/KeyguardStatusViewManager;
  
      invoke-virtual {v0}, Lcom/android/internal/policy/impl/KeyguardStatusViewManager;->onPause()V
  
+     .line 271
      return-void
  .end method
  
***************
*** 953,958 ****
--- 1074,1080 ----
      .parameter "previouslyFocusedRect"
  
      .prologue
+     .line 260
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      invoke-virtual {v0, p1, p2}, Landroid/widget/EditText;->requestFocus(ILandroid/graphics/Rect;)Z
***************
*** 966,995 ****
--- 1088,1123 ----
      .locals 4
  
      .prologue
+     .line 275
      const/4 v2, 0x1
  
      iput-boolean v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mResuming:Z
  
+     .line 277
      iget-object v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mStatusViewManager:Lcom/android/internal/policy/impl/KeyguardStatusViewManager;
  
      invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardStatusViewManager;->onResume()V
  
+     .line 280
      iget-object v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      const-string v3, ""
  
      invoke-virtual {v2, v3}, Landroid/widget/EditText;->setText(Ljava/lang/CharSequence;)V
  
+     .line 281
      iget-object v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
  
      invoke-virtual {v2}, Landroid/widget/EditText;->requestFocus()Z
  
+     .line 284
      iget-object v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
      invoke-virtual {v2}, Lcom/android/internal/widget/LockPatternUtils;->getLockoutAttemptDeadline()J
  
      move-result-wide v0
  
+     .line 285
      .local v0, deadline:J
      const-wide/16 v2, 0x0
  
***************
*** 997,1008 ****
--- 1125,1139 ----
  
      if-eqz v2, :cond_0
  
+     .line 286
      invoke-direct {p0, v0, v1}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->handleAttemptLockout(J)V
  
+     .line 288
      :cond_0
      const/4 v2, 0x0
  
      iput-boolean v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mResuming:Z
  
+     .line 289
      return-void
  .end method
