*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PasswordUnlockScreen$5.smali	2012-03-23 19:35:12.222809768 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PasswordUnlockScreen$5.smali.original	2012-03-23 19:34:44.379302043 -0500
***************
*** 26,31 ****
--- 26,32 ----
      .parameter "x1"
  
      .prologue
+     .line 326
      iput-object p1, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen$5;->this$0:Lcom/android/internal/policy/impl/PasswordUnlockScreen;
  
      invoke-direct {p0, p2, p3, p4, p5}, Landroid/os/CountDownTimer;-><init>(JJ)V
***************
*** 41,46 ****
--- 42,48 ----
      .prologue
      const/4 v1, 0x1
  
+     .line 339
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen$5;->this$0:Lcom/android/internal/policy/impl/PasswordUnlockScreen;
  
      #getter for: Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mPasswordEntry:Landroid/widget/EditText;
***************
*** 50,55 ****
--- 52,58 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/EditText;->setEnabled(Z)V
  
+     .line 340
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen$5;->this$0:Lcom/android/internal/policy/impl/PasswordUnlockScreen;
  
      #getter for: Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mKeyboardView:Lcom/android/internal/widget/PasswordEntryKeyboardView;
***************
*** 59,64 ****
--- 62,68 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/PasswordEntryKeyboardView;->setEnabled(Z)V
  
+     .line 341
      iget-object v0, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen$5;->this$0:Lcom/android/internal/policy/impl/PasswordUnlockScreen;
  
      #getter for: Lcom/android/internal/policy/impl/PasswordUnlockScreen;->mStatusViewManager:Lcom/android/internal/policy/impl/KeyguardStatusViewManager;
***************
*** 68,73 ****
--- 72,78 ----
  
      invoke-virtual {v0}, Lcom/android/internal/policy/impl/KeyguardStatusViewManager;->resetStatusInfo()V
  
+     .line 342
      return-void
  .end method
  
***************
*** 76,87 ****
--- 81,94 ----
      .parameter "millisUntilFinished"
  
      .prologue
+     .line 330
      const-wide/16 v2, 0x3e8
  
      div-long v2, p1, v2
  
      long-to-int v1, v2
  
+     .line 331
      .local v1, secondsRemaining:I
      iget-object v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen$5;->this$0:Lcom/android/internal/policy/impl/PasswordUnlockScreen;
  
***************
*** 107,112 ****
--- 114,120 ----
  
      move-result-object v0
  
+     .line 334
      .local v0, instructions:Ljava/lang/String;
      iget-object v2, p0, Lcom/android/internal/policy/impl/PasswordUnlockScreen$5;->this$0:Lcom/android/internal/policy/impl/PasswordUnlockScreen;
  
***************
*** 117,121 ****
--- 125,130 ----
  
      invoke-virtual {v2, v0}, Lcom/android/internal/policy/impl/KeyguardStatusViewManager;->setInstructionText(Ljava/lang/String;)V
  
+     .line 335
      return-void
  .end method
