*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PatternUnlockScreen$FooterMode.smali	2012-03-23 19:35:14.524851747 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PatternUnlockScreen$FooterMode.smali.original	2012-03-23 19:34:44.413302663 -0500
***************
*** 44,49 ****
--- 44,50 ----
  
      const/4 v2, 0x0
  
+     .line 102
      new-instance v0, Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;
  
      const-string v1, "Normal"
***************
*** 52,57 ****
--- 53,59 ----
  
      sput-object v0, Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;->Normal:Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;
  
+     .line 103
      new-instance v0, Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;
  
      const-string v1, "ForgotLockPattern"
***************
*** 60,65 ****
--- 62,68 ----
  
      sput-object v0, Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;->ForgotLockPattern:Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;
  
+     .line 104
      new-instance v0, Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;
  
      const-string v1, "VerifyUnlocked"
***************
*** 68,73 ****
--- 71,77 ----
  
      sput-object v0, Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;->VerifyUnlocked:Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;
  
+     .line 101
      const/4 v0, 0x3
  
      new-array v0, v0, [Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;
***************
*** 100,105 ****
--- 104,110 ----
      .end annotation
  
      .prologue
+     .line 101
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 110,115 ****
--- 115,121 ----
      .parameter "name"
  
      .prologue
+     .line 101
      const-class v0, Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 125,130 ****
--- 131,137 ----
      .locals 1
  
      .prologue
+     .line 101
      sget-object v0, Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;->$VALUES:[Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;
  
      invoke-virtual {v0}, [Lcom/android/internal/policy/impl/PatternUnlockScreen$FooterMode;->clone()Ljava/lang/Object;
