*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$PassHeadsetKey.smali	2012-03-23 19:35:15.738873883 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$PassHeadsetKey.smali.original	2012-03-23 19:34:44.432303009 -0500
***************
*** 30,41 ****
--- 30,44 ----
      .parameter "keyEvent"
  
      .prologue
+     .line 2987
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$PassHeadsetKey;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 2988
      iput-object p2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$PassHeadsetKey;->mKeyEvent:Landroid/view/KeyEvent;
  
+     .line 2989
      return-void
  .end method
  
***************
*** 47,64 ****
--- 50,70 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 2992
      invoke-static {}, Landroid/app/ActivityManagerNative;->isSystemReady()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 2993
      new-instance v1, Landroid/content/Intent;
  
      const-string v0, "android.intent.action.MEDIA_BUTTON"
  
      invoke-direct {v1, v0, v2}, Landroid/content/Intent;-><init>(Ljava/lang/String;Landroid/net/Uri;)V
  
+     .line 2994
      .local v1, intent:Landroid/content/Intent;
      const-string v0, "android.intent.extra.KEY_EVENT"
  
***************
*** 66,71 ****
--- 72,78 ----
  
      invoke-virtual {v1, v0, v3}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 2995
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$PassHeadsetKey;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iget-object v0, v0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
***************
*** 86,91 ****
--- 93,99 ----
  
      invoke-virtual/range {v0 .. v7}, Landroid/content/Context;->sendOrderedBroadcast(Landroid/content/Intent;Ljava/lang/String;Landroid/content/BroadcastReceiver;Landroid/os/Handler;ILjava/lang/String;Landroid/os/Bundle;)V
  
+     .line 2998
      .end local v1           #intent:Landroid/content/Intent;
      :cond_0
      return-void
