*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$DrawableFeatureState.smali	2012-03-23 19:35:12.288810971 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$DrawableFeatureState.smali.original	2012-03-23 19:34:44.380302061 -0500
***************
*** 42,54 ****
--- 42,59 ----
      .prologue
      const/16 v0, 0xff
  
+     .line 3117
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 3135
      iput v0, p0, Lcom/android/internal/policy/impl/PhoneWindow$DrawableFeatureState;->alpha:I
  
+     .line 3137
      iput v0, p0, Lcom/android/internal/policy/impl/PhoneWindow$DrawableFeatureState;->curAlpha:I
  
+     .line 3118
      iput p1, p0, Lcom/android/internal/policy/impl/PhoneWindow$DrawableFeatureState;->featureId:I
  
+     .line 3119
      return-void
  .end method
