*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$RotationWatcher$1.smali	2012-03-23 19:35:12.767819707 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$RotationWatcher$1.smali.original	2012-03-23 19:34:44.386302171 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 3414
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindow$RotationWatcher$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindow$RotationWatcher;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,48 ****
--- 41,51 ----
      .locals 1
  
      .prologue
+     .line 3416
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindow$RotationWatcher$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindow$RotationWatcher;
  
      invoke-virtual {v0}, Lcom/android/internal/policy/impl/PhoneWindow$RotationWatcher;->dispatchRotationChanged()V
  
+     .line 3417
      return-void
  .end method
