*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$15.smali	2012-03-23 19:35:12.675818028 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$15.smali.original	2012-03-23 19:34:44.385302152 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 3048
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$15;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iput-object p2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$15;->val$screenOnListener:Landroid/view/WindowManagerPolicy$ScreenOnListener;
***************
*** 46,53 ****
--- 47,56 ----
      .parameter "windowToken"
  
      .prologue
+     .line 3050
      if-eqz p1, :cond_0
  
+     .line 3052
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$15;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 61,69 ****
--- 64,74 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 3071
      :goto_0
      return-void
  
+     .line 3065
      :cond_0
      const-string v0, "WindowManager"
  
***************
*** 71,86 ****
--- 76,94 ----
  
      invoke-static {v0, v1}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 3066
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$15;->val$screenOnListener:Landroid/view/WindowManagerPolicy$ScreenOnListener;
  
      invoke-interface {v0}, Landroid/view/WindowManagerPolicy$ScreenOnListener;->onScreenOn()V
  
+     .line 3067
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$15;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iget-object v1, v0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 3068
      :try_start_1
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$15;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 88,93 ****
--- 96,102 ----
  
      iput-boolean v2, v0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mScreenOnFully:Z
  
+     .line 3069
      monitor-exit v1
  
      goto :goto_0
***************
*** 101,106 ****
--- 110,116 ----
  
      throw v0
  
+     .line 3062
      :catch_0
      move-exception v0
  
