*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$SettingsObserver.smali	2012-03-23 19:35:14.772856268 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$SettingsObserver.smali.original	2012-03-23 19:34:44.417302735 -0500
***************
*** 25,34 ****
--- 25,37 ----
      .parameter "handler"
  
      .prologue
+     .line 467
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$SettingsObserver;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
+     .line 468
      invoke-direct {p0, p2}, Landroid/database/ContentObserver;-><init>(Landroid/os/Handler;)V
  
+     .line 469
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 472
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$SettingsObserver;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iget-object v1, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
***************
*** 48,53 ****
--- 52,58 ----
  
      move-result-object v0
  
+     .line 473
      .local v0, resolver:Landroid/content/ContentResolver;
      const-string v1, "end_button_behavior"
  
***************
*** 57,62 ****
--- 62,68 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 475
      const-string v1, "incall_power_button_behavior"
  
      invoke-static {v1}, Landroid/provider/Settings$Secure;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 65,70 ****
--- 71,77 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 477
      const-string v1, "accelerometer_rotation"
  
      invoke-static {v1}, Landroid/provider/Settings$System;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 73,78 ****
--- 80,86 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 479
      const-string v1, "user_rotation"
  
      invoke-static {v1}, Landroid/provider/Settings$System;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 81,86 ****
--- 89,95 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 481
      const-string v1, "screen_off_timeout"
  
      invoke-static {v1}, Landroid/provider/Settings$System;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 89,94 ****
--- 98,104 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 483
      const-string v1, "window_orientation_listener_log"
  
      invoke-static {v1}, Landroid/provider/Settings$System;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 97,102 ****
--- 107,113 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 485
      const-string v1, "pointer_location"
  
      invoke-static {v1}, Landroid/provider/Settings$System;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 105,110 ****
--- 116,122 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 487
      const-string v1, "default_input_method"
  
      invoke-static {v1}, Landroid/provider/Settings$Secure;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 113,118 ****
--- 125,131 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 489
      const-string v1, "fancy_rotation_anim"
  
      invoke-static {v1}, Landroid/provider/Settings$System;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 121,130 ****
--- 134,145 ----
  
      invoke-virtual {v0, v1, v2, p0}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
  
+     .line 491
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$SettingsObserver;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      invoke-virtual {v1}, Lcom/android/internal/policy/impl/PhoneWindowManager;->updateSettings()V
  
+     .line 492
      return-void
  .end method
  
***************
*** 133,147 ****
--- 148,165 ----
      .parameter "selfChange"
  
      .prologue
+     .line 495
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$SettingsObserver;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      invoke-virtual {v0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->updateSettings()V
  
+     .line 496
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$SettingsObserver;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/PhoneWindowManager;->updateRotation(Z)V
  
+     .line 497
      return-void
  .end method
