*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback.smali	2012-03-23 19:35:15.871876309 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback.smali.original	2012-03-23 19:34:44.434303045 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1701
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,45 ****
--- 41,47 ----
      .parameter "x1"
  
      .prologue
+     .line 1701
      invoke-direct {p0, p1}, Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;-><init>(Lcom/android/internal/policy/impl/PhoneWindow;)V
  
      return-void
***************
*** 55,69 ****
--- 57,74 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 1704
      invoke-virtual {p1}, Lcom/android/internal/view/menu/MenuBuilder;->getRootMenu()Lcom/android/internal/view/menu/MenuBuilder;
  
      move-result-object v2
  
+     .line 1705
      .local v2, parentMenu:Landroid/view/Menu;
      if-eq v2, p1, :cond_2
  
      move v0, v3
  
+     .line 1706
      .local v0, isSubMenu:Z
      :goto_0
      iget-object v4, p0, Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
***************
*** 78,88 ****
--- 83,96 ----
  
      move-result-object v1
  
+     .line 1707
      .local v1, panel:Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;
      if-eqz v1, :cond_1
  
+     .line 1708
      if-eqz v0, :cond_3
  
+     .line 1709
      iget-object v4, p0, Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
  
      iget v5, v1, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;->featureId:I
***************
*** 90,103 ****
--- 98,114 ----
      #calls: Lcom/android/internal/policy/impl/PhoneWindow;->callOnPanelClosed(ILcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;Landroid/view/Menu;)V
      invoke-static {v4, v5, v1, v2}, Lcom/android/internal/policy/impl/PhoneWindow;->access$300(Lcom/android/internal/policy/impl/PhoneWindow;ILcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;Landroid/view/Menu;)V
  
+     .line 1710
      iget-object v4, p0, Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
  
      invoke-virtual {v4, v1, v3}, Lcom/android/internal/policy/impl/PhoneWindow;->closePanel(Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;Z)V
  
+     .line 1717
      :cond_1
      :goto_1
      return-void
  
+     .line 1705
      .end local v0           #isSubMenu:Z
      .end local v1           #panel:Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;
      .restart local p1
***************
*** 106,111 ****
--- 117,123 ----
  
      goto :goto_0
  
+     .line 1714
      .end local p1
      .restart local v0       #isSubMenu:Z
      .restart local v1       #panel:Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;
***************
*** 124,129 ****
--- 136,142 ----
      .prologue
      const/16 v2, 0x8
  
+     .line 1721
      if-nez p1, :cond_0
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
***************
*** 134,145 ****
--- 147,160 ----
  
      if-eqz v1, :cond_0
  
+     .line 1722
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindow$PanelMenuPresenterCallback;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
  
      invoke-virtual {v1}, Lcom/android/internal/policy/impl/PhoneWindow;->getCallback()Landroid/view/Window$Callback;
  
      move-result-object v0
  
+     .line 1723
      .local v0, cb:Landroid/view/Window$Callback;
      if-eqz v0, :cond_0
  
***************
*** 151,158 ****
--- 166,175 ----
  
      if-nez v1, :cond_0
  
+     .line 1724
      invoke-interface {v0, v2, p1}, Landroid/view/Window$Callback;->onMenuOpened(ILandroid/view/Menu;)Z
  
+     .line 1728
      .end local v0           #cb:Landroid/view/Window$Callback;
      :cond_0
      const/4 v1, 0x1
