*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$18$1.smali	2012-03-23 19:35:15.234864692 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$18$1.smali.original	2012-03-23 19:34:44.425302881 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .parameter "x0"
  
      .prologue
+     .line 3405
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$18$1;->this$1:Lcom/android/internal/policy/impl/PhoneWindowManager$18;
  
      invoke-direct {p0, p2}, Landroid/app/ProgressDialog;-><init>(Landroid/content/Context;)V
***************
*** 39,44 ****
--- 40,46 ----
      .parameter "ev"
  
      .prologue
+     .line 3421
      const/4 v0, 0x1
  
      return v0
***************
*** 49,54 ****
--- 51,57 ----
      .parameter "event"
  
      .prologue
+     .line 3409
      const/4 v0, 0x1
  
      return v0
***************
*** 59,64 ****
--- 62,68 ----
      .parameter "event"
  
      .prologue
+     .line 3412
      const/4 v0, 0x1
  
      return v0
***************
*** 69,74 ****
--- 73,79 ----
      .parameter "event"
  
      .prologue
+     .line 3425
      const/4 v0, 0x1
  
      return v0
***************
*** 79,84 ****
--- 84,90 ----
      .parameter "ev"
  
      .prologue
+     .line 3415
      const/4 v0, 0x1
  
      return v0
***************
*** 89,94 ****
--- 95,101 ----
      .parameter "ev"
  
      .prologue
+     .line 3418
      const/4 v0, 0x1
  
      return v0
