*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$21.smali	2012-03-23 19:35:14.499851289 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$21.smali.original	2012-03-23 19:34:44.412302645 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .parameter
  
      .prologue
+     .line 3719
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$21;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iput p2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$21;->val$visibility:I
***************
*** 50,61 ****
--- 51,64 ----
      .locals 3
  
      .prologue
+     .line 3721
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$21;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iget-object v1, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBarService:Lcom/android/internal/statusbar/IStatusBarService;
  
      if-nez v1, :cond_0
  
+     .line 3722
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$21;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      const-string v2, "statusbar"
***************
*** 70,75 ****
--- 73,79 ----
  
      iput-object v2, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBarService:Lcom/android/internal/statusbar/IStatusBarService;
  
+     .line 3725
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$21;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 77,82 ****
--- 81,87 ----
  
      if-eqz v1, :cond_1
  
+     .line 3727
      :try_start_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$21;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 86,91 ****
--- 91,97 ----
  
      invoke-interface {v1, v2}, Lcom/android/internal/statusbar/IStatusBarService;->setSystemUiVisibility(I)V
  
+     .line 3728
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$21;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iget-object v1, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBarService:Lcom/android/internal/statusbar/IStatusBarService;
***************
*** 96,108 ****
--- 102,117 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 3734
      :cond_1
      :goto_0
      return-void
  
+     .line 3729
      :catch_0
      move-exception v0
  
+     .line 3731
      .local v0, e:Landroid/os/RemoteException;
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$21;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
