*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState$1.smali	2012-03-23 19:35:14.078843612 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState$1.smali.original	2012-03-23 19:34:44.406302535 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 3399
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "in"
  
      .prologue
+     .line 3401
      #calls: Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState;->readFromParcel(Landroid/os/Parcel;)Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState;
      invoke-static {p1}, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState;->access$1700(Landroid/os/Parcel;)Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState;
  
***************
*** 57,62 ****
--- 59,65 ----
      .parameter "x0"
  
      .prologue
+     .line 3399
      invoke-virtual {p0, p1}, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState$1;->createFromParcel(Landroid/os/Parcel;)Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState;
  
      move-result-object v0
***************
*** 69,74 ****
--- 72,78 ----
      .parameter "size"
  
      .prologue
+     .line 3405
      new-array v0, p1, [Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState;
  
      return-object v0
***************
*** 79,84 ****
--- 83,89 ----
      .parameter "x0"
  
      .prologue
+     .line 3399
      invoke-virtual {p0, p1}, Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState$1;->newArray(I)[Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState$SavedState;
  
      move-result-object v0
