*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$5.smali	2012-03-23 19:35:13.547833929 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$5.smali.original	2012-03-23 19:34:44.399302407 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 717
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iput p2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->val$behavior:I
***************
*** 45,56 ****
--- 46,59 ----
      .locals 3
  
      .prologue
+     .line 720
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iget-object v0, v0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mRecentAppsDialog:Lcom/android/internal/policy/impl/RecentApplicationsDialog;
  
      if-nez v0, :cond_0
  
+     .line 721
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      new-instance v1, Lcom/android/internal/policy/impl/RecentApplicationsDialog;
***************
*** 63,68 ****
--- 66,72 ----
  
      iput-object v1, v0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mRecentAppsDialog:Lcom/android/internal/policy/impl/RecentApplicationsDialog;
  
+     .line 723
      :cond_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 74,87 ****
--- 78,94 ----
  
      if-eqz v0, :cond_1
  
+     .line 724
      iget v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->val$behavior:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 752
      :goto_0
      :pswitch_0
      return-void
  
+     .line 726
      :pswitch_1
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 91,96 ****
--- 98,104 ----
  
      goto :goto_0
  
+     .line 729
      :pswitch_2
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 100,105 ****
--- 108,114 ----
  
      goto :goto_0
  
+     .line 736
      :cond_1
      iget v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->val$behavior:I
  
***************
*** 107,112 ****
--- 116,122 ----
  
      goto :goto_0
  
+     .line 738
      :pswitch_3
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 116,121 ****
--- 126,132 ----
  
      goto :goto_0
  
+     .line 742
      :pswitch_4
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
***************
*** 128,133 ****
--- 139,145 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 745
      :goto_1
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$5;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 137,147 ****
--- 149,161 ----
  
      goto :goto_0
  
+     .line 743
      :catch_0
      move-exception v0
  
      goto :goto_1
  
+     .line 724
      nop
  
      :pswitch_data_0
***************
*** 151,156 ****
--- 165,171 ----
          :pswitch_2
      .end packed-switch
  
+     .line 736
      :pswitch_data_1
      .packed-switch 0x0
          :pswitch_3
