*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$1.smali	2012-03-23 19:35:13.507833201 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$1.smali.original	2012-03-23 19:34:44.398302389 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 351
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      invoke-direct {p0}, Lcom/android/internal/view/BaseInputHandler;-><init>()V
***************
*** 39,46 ****
--- 40,49 ----
      .parameter "finishedCallback"
  
      .prologue
+     .line 354
      const/4 v0, 0x0
  
+     .line 356
      .local v0, handled:Z
      :try_start_0
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getSource()I
***************
*** 51,56 ****
--- 54,60 ----
  
      if-eqz v1, :cond_1
  
+     .line 357
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iget-object v2, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mLock:Ljava/lang/Object;
***************
*** 59,64 ****
--- 63,69 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_1
  
+     .line 358
      :try_start_1
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 66,89 ****
--- 71,100 ----
  
      if-eqz v1, :cond_0
  
+     .line 359
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      iget-object v1, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mPointerLocationView:Lcom/android/internal/widget/PointerLocationView;
  
      invoke-virtual {v1, p1}, Lcom/android/internal/widget/PointerLocationView;->addPointerEvent(Landroid/view/MotionEvent;)V
  
+     .line 360
      const/4 v0, 0x1
  
+     .line 362
      :cond_0
      monitor-exit v2
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 365
      :cond_1
      invoke-virtual {p2, v0}, Landroid/view/InputQueue$FinishedCallback;->finished(Z)V
  
+     .line 367
      return-void
  
+     .line 362
      :catchall_0
      move-exception v1
  
***************
*** 97,102 ****
--- 108,114 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_1
  
+     .line 365
      :catchall_1
      move-exception v1
  
