*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$7.smali	2012-03-23 19:35:15.303865951 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$7.smali.original	2012-03-23 19:34:44.426302899 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 1823
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$7;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 40,44 ****
--- 41,46 ----
      .locals 0
  
      .prologue
+     .line 1825
      return-void
  .end method
