*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$14.smali	2012-03-23 19:35:13.711836921 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindowManager$14.smali.original	2012-03-23 19:34:44.401302443 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 3007
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$14;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      invoke-direct {p0}, Landroid/content/BroadcastReceiver;-><init>()V
***************
*** 39,44 ****
--- 40,46 ----
      .parameter "intent"
  
      .prologue
+     .line 3009
      const-string v1, "android.intent.action.DOCK_EVENT"
  
      invoke-virtual {p2}, Landroid/content/Intent;->getAction()Ljava/lang/String;
***************
*** 51,56 ****
--- 53,59 ----
  
      if-eqz v1, :cond_0
  
+     .line 3010
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$14;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      const-string v2, "android.intent.extra.DOCK_STATE"
***************
*** 63,68 ****
--- 66,72 ----
  
      iput v2, v1, Lcom/android/internal/policy/impl/PhoneWindowManager;->mDockMode:I
  
+     .line 3020
      :goto_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$14;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 70,81 ****
--- 74,88 ----
  
      invoke-virtual {v1, v2}, Lcom/android/internal/policy/impl/PhoneWindowManager;->updateRotation(Z)V
  
+     .line 3021
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$14;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
      invoke-virtual {v1}, Lcom/android/internal/policy/impl/PhoneWindowManager;->updateOrientationListenerLp()V
  
+     .line 3022
      return-void
  
+     .line 3014
      :cond_0
      :try_start_0
      const-string v1, "uimode"
***************
*** 88,93 ****
--- 95,101 ----
  
      move-result-object v0
  
+     .line 3016
      .local v0, uiModeService:Landroid/app/IUiModeManager;
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager$14;->this$0:Lcom/android/internal/policy/impl/PhoneWindowManager;
  
***************
*** 101,106 ****
--- 109,115 ----
  
      goto :goto_0
  
+     .line 3017
      .end local v0           #uiModeService:Landroid/app/IUiModeManager;
      :catch_0
      move-exception v1
