*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneFallbackEventHandler.smali	2012-03-23 19:35:12.984823664 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneFallbackEventHandler.smali.original	2012-03-23 19:34:44.389302225 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .locals 1
  
      .prologue
+     .line 35
      const-string v0, "PhoneFallbackEventHandler"
  
      sput-object v0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->TAG:Ljava/lang/String;
***************
*** 43,52 ****
--- 44,56 ----
      .parameter "context"
  
      .prologue
+     .line 46
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 47
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
+     .line 48
      return-void
  .end method
  
***************
*** 57,78 ****
--- 61,87 ----
      .parameter "event"
  
      .prologue
+     .line 61
      invoke-virtual {p1}, Landroid/view/KeyEvent;->getAction()I
  
      move-result v0
  
+     .line 62
      .local v0, action:I
      invoke-virtual {p1}, Landroid/view/KeyEvent;->getKeyCode()I
  
      move-result v1
  
+     .line 64
      .local v1, keyCode:I
      if-nez v0, :cond_0
  
+     .line 65
      invoke-virtual {p0, v1, p1}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->onKeyDown(ILandroid/view/KeyEvent;)Z
  
      move-result v2
  
+     .line 67
      :goto_0
      return v2
  
***************
*** 88,97 ****
--- 97,108 ----
      .locals 2
  
      .prologue
+     .line 281
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mAudioManager:Landroid/media/AudioManager;
  
      if-nez v0, :cond_0
  
+     .line 282
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      const-string v1, "audio"
***************
*** 104,109 ****
--- 115,121 ----
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mAudioManager:Landroid/media/AudioManager;
  
+     .line 284
      :cond_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mAudioManager:Landroid/media/AudioManager;
  
***************
*** 114,123 ****
--- 126,137 ----
      .locals 2
  
      .prologue
+     .line 274
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mKeyguardManager:Landroid/app/KeyguardManager;
  
      if-nez v0, :cond_0
  
+     .line 275
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      const-string v1, "keyguard"
***************
*** 130,135 ****
--- 144,150 ----
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mKeyguardManager:Landroid/app/KeyguardManager;
  
+     .line 277
      :cond_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mKeyguardManager:Landroid/app/KeyguardManager;
  
***************
*** 140,149 ****
--- 155,166 ----
      .locals 2
  
      .prologue
+     .line 259
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mSearchManager:Landroid/app/SearchManager;
  
      if-nez v0, :cond_0
  
+     .line 260
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      const-string v1, "search"
***************
*** 156,161 ****
--- 173,179 ----
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mSearchManager:Landroid/app/SearchManager;
  
+     .line 262
      :cond_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mSearchManager:Landroid/app/SearchManager;
  
***************
*** 166,175 ****
--- 184,195 ----
      .locals 2
  
      .prologue
+     .line 266
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mTelephonyManager:Landroid/telephony/TelephonyManager;
  
      if-nez v0, :cond_0
  
+     .line 267
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      const-string v1, "phone"
***************
*** 182,187 ****
--- 202,208 ----
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mTelephonyManager:Landroid/telephony/TelephonyManager;
  
+     .line 270
      :cond_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mTelephonyManager:Landroid/telephony/TelephonyManager;
  
***************
*** 202,213 ****
--- 223,236 ----
  
      const/4 v4, 0x1
  
+     .line 76
      iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mView:Landroid/view/View;
  
      invoke-virtual {v6}, Landroid/view/View;->getKeyDispatcherState()Landroid/view/KeyEvent$DispatcherState;
  
      move-result-object v1
  
+     .line 78
      .local v1, dispatcher:Landroid/view/KeyEvent$DispatcherState;
      sparse-switch p1, :sswitch_data_0
  
***************
*** 215,224 ****
--- 238,249 ----
      :goto_0
      move v4, v5
  
+     .line 180
      :cond_1
      :goto_1
      return v4
  
+     .line 82
      :sswitch_0
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->getAudioManager()Landroid/media/AudioManager;
  
***************
*** 230,235 ****
--- 255,261 ----
  
      goto :goto_1
  
+     .line 92
      :sswitch_1
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->getTelephonyManager()Landroid/telephony/TelephonyManager;
  
***************
*** 241,246 ****
--- 267,273 ----
  
      if-nez v5, :cond_1
  
+     .line 103
      :sswitch_2
      new-instance v3, Landroid/content/Intent;
  
***************
*** 248,264 ****
--- 275,294 ----
  
      invoke-direct {v3, v5, v7}, Landroid/content/Intent;-><init>(Ljava/lang/String;Landroid/net/Uri;)V
  
+     .line 104
      .local v3, intent:Landroid/content/Intent;
      const-string v5, "android.intent.extra.KEY_EVENT"
  
      invoke-virtual {v3, v5, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 105
      iget-object v5, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      invoke-virtual {v5, v3, v7}, Landroid/content/Context;->sendOrderedBroadcast(Landroid/content/Intent;Ljava/lang/String;)V
  
      goto :goto_1
  
+     .line 110
      .end local v3           #intent:Landroid/content/Intent;
      :sswitch_3
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->getKeyguardManager()Landroid/app/KeyguardManager;
***************
*** 273,288 ****
--- 303,321 ----
  
      if-eqz v1, :cond_0
  
+     .line 113
      invoke-virtual {p2}, Landroid/view/KeyEvent;->getRepeatCount()I
  
      move-result v6
  
      if-nez v6, :cond_2
  
+     .line 114
      invoke-virtual {v1, p2, p0}, Landroid/view/KeyEvent$DispatcherState;->startTracking(Landroid/view/KeyEvent;Ljava/lang/Object;)V
  
      goto :goto_1
  
+     .line 115
      :cond_2
      invoke-virtual {p2}, Landroid/view/KeyEvent;->isLongPress()Z
  
***************
*** 296,319 ****
--- 329,358 ----
  
      if-eqz v6, :cond_1
  
+     .line 116
      invoke-virtual {v1, p2}, Landroid/view/KeyEvent$DispatcherState;->performedLongPress(Landroid/view/KeyEvent;)V
  
+     .line 117
      iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mView:Landroid/view/View;
  
      invoke-virtual {v6, v5}, Landroid/view/View;->performHapticFeedback(I)Z
  
+     .line 119
      new-instance v3, Landroid/content/Intent;
  
      const-string v5, "android.intent.action.VOICE_COMMAND"
  
      invoke-direct {v3, v5}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 120
      .restart local v3       #intent:Landroid/content/Intent;
      invoke-virtual {v3, v8}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 122
      :try_start_0
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->sendCloseSystemWindows()V
  
+     .line 123
      iget-object v5, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      invoke-virtual {v5, v3}, Landroid/content/Context;->startActivity(Landroid/content/Intent;)V
***************
*** 322,335 ****
--- 361,377 ----
  
      goto :goto_1
  
+     .line 124
      :catch_0
      move-exception v2
  
+     .line 125
      .local v2, e:Landroid/content/ActivityNotFoundException;
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->startCallActivity()V
  
      goto :goto_1
  
+     .line 132
      .end local v2           #e:Landroid/content/ActivityNotFoundException;
      .end local v3           #intent:Landroid/content/Intent;
      :sswitch_4
***************
*** 345,360 ****
--- 387,405 ----
  
      if-eqz v1, :cond_0
  
+     .line 135
      invoke-virtual {p2}, Landroid/view/KeyEvent;->getRepeatCount()I
  
      move-result v6
  
      if-nez v6, :cond_3
  
+     .line 136
      invoke-virtual {v1, p2, p0}, Landroid/view/KeyEvent$DispatcherState;->startTracking(Landroid/view/KeyEvent;Ljava/lang/Object;)V
  
      goto :goto_1
  
+     .line 137
      :cond_3
      invoke-virtual {p2}, Landroid/view/KeyEvent;->isLongPress()Z
  
***************
*** 368,398 ****
--- 413,450 ----
  
      if-eqz v6, :cond_1
  
+     .line 138
      invoke-virtual {v1, p2}, Landroid/view/KeyEvent$DispatcherState;->performedLongPress(Landroid/view/KeyEvent;)V
  
+     .line 139
      iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mView:Landroid/view/View;
  
      invoke-virtual {v6, v5}, Landroid/view/View;->performHapticFeedback(I)Z
  
+     .line 140
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->sendCloseSystemWindows()V
  
+     .line 142
      new-instance v3, Landroid/content/Intent;
  
      const-string v5, "android.intent.action.CAMERA_BUTTON"
  
      invoke-direct {v3, v5, v7}, Landroid/content/Intent;-><init>(Ljava/lang/String;Landroid/net/Uri;)V
  
+     .line 143
      .restart local v3       #intent:Landroid/content/Intent;
      const-string v5, "android.intent.extra.KEY_EVENT"
  
      invoke-virtual {v3, v5, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 144
      iget-object v5, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      invoke-virtual {v5, v3, v7}, Landroid/content/Context;->sendOrderedBroadcast(Landroid/content/Intent;Ljava/lang/String;)V
  
      goto/16 :goto_1
  
+     .line 150
      .end local v3           #intent:Landroid/content/Intent;
      :sswitch_5
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->getKeyguardManager()Landroid/app/KeyguardManager;
***************
*** 407,422 ****
--- 459,477 ----
  
      if-eqz v1, :cond_0
  
+     .line 153
      invoke-virtual {p2}, Landroid/view/KeyEvent;->getRepeatCount()I
  
      move-result v6
  
      if-nez v6, :cond_4
  
+     .line 154
      invoke-virtual {v1, p2, p0}, Landroid/view/KeyEvent$DispatcherState;->startTracking(Landroid/view/KeyEvent;Ljava/lang/Object;)V
  
      goto/16 :goto_0
  
+     .line 155
      :cond_4
      invoke-virtual {p2}, Landroid/view/KeyEvent;->isLongPress()Z
  
***************
*** 430,435 ****
--- 485,491 ----
  
      if-eqz v6, :cond_0
  
+     .line 156
      iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      invoke-virtual {v6}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
***************
*** 440,445 ****
--- 496,502 ----
  
      move-result-object v0
  
+     .line 157
      .local v0, config:Landroid/content/res/Configuration;
      iget v6, v0, Landroid/content/res/Configuration;->keyboard:I
  
***************
*** 451,456 ****
--- 508,514 ----
  
      if-ne v6, v7, :cond_0
  
+     .line 160
      :cond_5
      new-instance v3, Landroid/content/Intent;
  
***************
*** 458,466 ****
--- 516,526 ----
  
      invoke-direct {v3, v6}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 161
      .restart local v3       #intent:Landroid/content/Intent;
      invoke-virtual {v3, v8}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 163
      :try_start_1
      iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mView:Landroid/view/View;
  
***************
*** 468,496 ****
--- 528,562 ----
  
      invoke-virtual {v6, v7}, Landroid/view/View;->performHapticFeedback(I)Z
  
+     .line 164
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->sendCloseSystemWindows()V
  
+     .line 165
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->getSearchManager()Landroid/app/SearchManager;
  
      move-result-object v6
  
      invoke-virtual {v6}, Landroid/app/SearchManager;->stopSearch()V
  
+     .line 166
      iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      invoke-virtual {v6, v3}, Landroid/content/Context;->startActivity(Landroid/content/Intent;)V
  
+     .line 170
      invoke-virtual {v1, p2}, Landroid/view/KeyEvent$DispatcherState;->performedLongPress(Landroid/view/KeyEvent;)V
      :try_end_1
      .catch Landroid/content/ActivityNotFoundException; {:try_start_1 .. :try_end_1} :catch_1
  
      goto/16 :goto_1
  
+     .line 172
      :catch_1
      move-exception v4
  
      goto/16 :goto_0
  
+     .line 78
      nop
  
      :sswitch_data_0
***************
*** 525,544 ****
--- 591,615 ----
  
      const/4 v3, 0x1
  
+     .line 187
      iget-object v4, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mView:Landroid/view/View;
  
      invoke-virtual {v4}, Landroid/view/View;->getKeyDispatcherState()Landroid/view/KeyEvent$DispatcherState;
  
      move-result-object v1
  
+     .line 188
      .local v1, dispatcher:Landroid/view/KeyEvent$DispatcherState;
      if-eqz v1, :cond_0
  
+     .line 189
      invoke-virtual {v1, p2}, Landroid/view/KeyEvent$DispatcherState;->handleUpEvent(Landroid/view/KeyEvent;)V
  
+     .line 192
      :cond_0
      sparse-switch p1, :sswitch_data_0
  
+     .line 244
      :cond_1
      const/4 v3, 0x0
  
***************
*** 546,551 ****
--- 617,623 ----
      :goto_0
      return v3
  
+     .line 196
      :sswitch_0
      invoke-virtual {p2}, Landroid/view/KeyEvent;->isCanceled()Z
  
***************
*** 553,558 ****
--- 625,631 ----
  
      if-nez v4, :cond_2
  
+     .line 197
      iget-object v4, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      const-string v5, "audio"
***************
*** 563,571 ****
--- 636,646 ----
  
      check-cast v0, Landroid/media/AudioManager;
  
+     .line 199
      .local v0, audioManager:Landroid/media/AudioManager;
      if-eqz v0, :cond_2
  
+     .line 200
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->getAudioManager()Landroid/media/AudioManager;
  
      move-result-object v4
***************
*** 576,581 ****
--- 651,657 ----
  
      goto :goto_0
  
+     .line 218
      .end local v0           #audioManager:Landroid/media/AudioManager;
      :sswitch_1
      new-instance v2, Landroid/content/Intent;
***************
*** 584,600 ****
--- 660,679 ----
  
      invoke-direct {v2, v4, v5}, Landroid/content/Intent;-><init>(Ljava/lang/String;Landroid/net/Uri;)V
  
+     .line 219
      .local v2, intent:Landroid/content/Intent;
      const-string v4, "android.intent.extra.KEY_EVENT"
  
      invoke-virtual {v2, v4, p2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Landroid/os/Parcelable;)Landroid/content/Intent;
  
+     .line 220
      iget-object v4, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      invoke-virtual {v4, v2, v5}, Landroid/content/Context;->sendOrderedBroadcast(Landroid/content/Intent;Ljava/lang/String;)V
  
      goto :goto_0
  
+     .line 225
      .end local v2           #intent:Landroid/content/Intent;
      :sswitch_2
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->getKeyguardManager()Landroid/app/KeyguardManager;
***************
*** 607,612 ****
--- 686,692 ----
  
      if-nez v4, :cond_1
  
+     .line 228
      invoke-virtual {p2}, Landroid/view/KeyEvent;->isTracking()Z
  
      move-result v4
***************
*** 621,626 ****
--- 701,707 ----
  
      goto :goto_0
  
+     .line 235
      :sswitch_3
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->getKeyguardManager()Landroid/app/KeyguardManager;
  
***************
*** 632,637 ****
--- 713,719 ----
  
      if-nez v4, :cond_1
  
+     .line 238
      invoke-virtual {p2}, Landroid/view/KeyEvent;->isTracking()Z
  
      move-result v4
***************
*** 644,653 ****
--- 726,737 ----
  
      if-nez v4, :cond_2
  
+     .line 239
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->startCallActivity()V
  
      goto :goto_0
  
+     .line 192
      nop
  
      :sswitch_data_0
***************
*** 676,681 ****
--- 760,766 ----
      .parameter "event"
  
      .prologue
+     .line 55
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->getAudioManager()Landroid/media/AudioManager;
  
      move-result-object v0
***************
*** 688,693 ****
--- 773,779 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/media/AudioManager;->preDispatchKeyEvent(II)V
  
+     .line 57
      return-void
  .end method
  
***************
*** 695,706 ****
--- 781,794 ----
      .locals 2
  
      .prologue
+     .line 288
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
      const/4 v1, 0x0
  
      invoke-static {v0, v1}, Lcom/android/internal/policy/impl/PhoneWindowManager;->sendCloseSystemWindows(Landroid/content/Context;Ljava/lang/String;)V
  
+     .line 289
      return-void
  .end method
  
***************
*** 709,716 ****
--- 797,806 ----
      .parameter "v"
  
      .prologue
+     .line 51
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mView:Landroid/view/View;
  
+     .line 52
      return-void
  .end method
  
***************
*** 718,736 ****
--- 808,830 ----
      .locals 4
  
      .prologue
+     .line 248
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->sendCloseSystemWindows()V
  
+     .line 249
      new-instance v1, Landroid/content/Intent;
  
      const-string v2, "android.intent.action.CALL_BUTTON"
  
      invoke-direct {v1, v2}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 250
      .local v1, intent:Landroid/content/Intent;
      const/high16 v2, 0x1000
  
      invoke-virtual {v1, v2}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 252
      :try_start_0
      iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->mContext:Landroid/content/Context;
  
***************
*** 738,749 ****
--- 832,846 ----
      :try_end_0
      .catch Landroid/content/ActivityNotFoundException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 256
      :goto_0
      return-void
  
+     .line 253
      :catch_0
      move-exception v0
  
+     .line 254
      .local v0, e:Landroid/content/ActivityNotFoundException;
      sget-object v2, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;->TAG:Ljava/lang/String;
  
