*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$1.smali	2012-03-23 19:35:15.913877075 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/PhoneWindow$1.smali.original	2012-03-23 19:34:44.435303063 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 2828
      iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindow$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,47 ****
--- 43,49 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 2831
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindow$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
  
      #calls: Lcom/android/internal/policy/impl/PhoneWindow;->getPanelState(IZ)Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;
***************
*** 49,54 ****
--- 51,57 ----
  
      move-result-object v0
  
+     .line 2832
      .local v0, st:Lcom/android/internal/policy/impl/PhoneWindow$PanelFeatureState;
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindow$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
  
***************
*** 64,69 ****
--- 67,73 ----
  
      if-nez v1, :cond_1
  
+     .line 2833
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindow$1;->this$0:Lcom/android/internal/policy/impl/PhoneWindow;
  
***************
*** 71,76 ****
--- 75,81 ----
  
      invoke-virtual {v1, v2}, Lcom/android/internal/policy/impl/PhoneWindow;->invalidatePanelMenu(I)V
  
+     .line 2835
      :cond_1
      return-void
  .end method
