*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/RecentApplicationsDialog$1.smali	2012-03-23 19:35:13.758837778 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/RecentApplicationsDialog$1.smali.original	2012-03-23 19:34:44.401302443 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 64
      iput-object p1, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog$1;->this$0:Lcom/android/internal/policy/impl/RecentApplicationsDialog;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,47 ****
--- 43,49 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 67
      iget-object v4, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog$1;->this$0:Lcom/android/internal/policy/impl/RecentApplicationsDialog;
  
      iget-object v0, v4, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
***************
*** 58,72 ****
--- 60,78 ----
  
      aget-object v2, v0, v1
  
+     .line 68
      .local v2, icon:Landroid/widget/TextView;
      invoke-virtual {v2, v5, v5, v5, v5}, Landroid/widget/TextView;->setCompoundDrawables(Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;)V
  
+     .line 69
      invoke-virtual {v2, v5}, Landroid/widget/TextView;->setTag(Ljava/lang/Object;)V
  
+     .line 67
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 71
      .end local v2           #icon:Landroid/widget/TextView;
      :cond_0
      return-void
