*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/RecentApplicationsDialog$RecentTag.smali	2012-03-23 19:35:15.826875488 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/RecentApplicationsDialog$RecentTag.smali.original	2012-03-23 19:34:44.434303045 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .parameter
  
      .prologue
+     .line 58
      iput-object p1, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;->this$0:Lcom/android/internal/policy/impl/RecentApplicationsDialog;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
