*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/RecentApplicationsDialog.smali	2012-03-23 19:35:15.136862907 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/android.policy.jar.out/smali/com/android/internal/policy/impl/RecentApplicationsDialog.smali.original	2012-03-23 19:34:44.423302845 -0500
***************
*** 44,59 ****
--- 44,62 ----
      .parameter "context"
  
      .prologue
+     .line 75
      const v0, 0x10302f8
  
      invoke-direct {p0, p1, v0}, Landroid/app/Dialog;-><init>(Landroid/content/Context;I)V
  
+     .line 54
      const/16 v0, 0x8
  
      new-array v0, v0, [Landroid/widget/TextView;
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
+     .line 56
      new-instance v0, Landroid/content/IntentFilter;
  
      const-string v1, "android.intent.action.CLOSE_SYSTEM_DIALOGS"
***************
*** 62,85 ****
--- 65,92 ----
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mBroadcastIntentFilter:Landroid/content/IntentFilter;
  
+     .line 63
      new-instance v0, Landroid/os/Handler;
  
      invoke-direct {v0}, Landroid/os/Handler;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mHandler:Landroid/os/Handler;
  
+     .line 64
      new-instance v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog$1;
  
      invoke-direct {v0, p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog$1;-><init>(Lcom/android/internal/policy/impl/RecentApplicationsDialog;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mCleanup:Ljava/lang/Runnable;
  
+     .line 335
      new-instance v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog$2;
  
      invoke-direct {v0, p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog$2;-><init>(Lcom/android/internal/policy/impl/RecentApplicationsDialog;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mBroadcastReceiver:Landroid/content/BroadcastReceiver;
  
+     .line 77
      return-void
  .end method
  
***************
*** 87,101 ****
--- 94,111 ----
      .locals 24
  
      .prologue
+     .line 257
      invoke-virtual/range {p0 .. p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->getContext()Landroid/content/Context;
  
      move-result-object v6
  
+     .line 258
      .local v6, context:Landroid/content/Context;
      invoke-virtual {v6}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
  
      move-result-object v15
  
+     .line 259
      .local v15, pm:Landroid/content/pm/PackageManager;
      const-string v21, "activity"
  
***************
*** 107,112 ****
--- 117,123 ----
  
      check-cast v5, Landroid/app/ActivityManager;
  
+     .line 261
      .local v5, am:Landroid/app/ActivityManager;
      const/16 v21, 0x10
  
***************
*** 120,125 ****
--- 131,137 ----
  
      move-result-object v16
  
+     .line 264
      .local v16, recentTasks:Ljava/util/List;,"Ljava/util/List<Landroid/app/ActivityManager$RecentTaskInfo;>;"
      new-instance v21, Landroid/content/Intent;
  
***************
*** 143,148 ****
--- 155,161 ----
  
      move-result-object v7
  
+     .line 268
      .local v7, homeInfo:Landroid/content/pm/ActivityInfo;
      new-instance v10, Lcom/android/internal/policy/impl/IconUtilities;
  
***************
*** 154,167 ****
--- 167,183 ----
  
      invoke-direct {v10, v0}, Lcom/android/internal/policy/impl/IconUtilities;-><init>(Landroid/content/Context;)V
  
+     .line 273
      .local v10, iconUtilities:Lcom/android/internal/policy/impl/IconUtilities;
      const/4 v11, 0x0
  
+     .line 274
      .local v11, index:I
      invoke-interface/range {v16 .. v16}, Ljava/util/List;->size()I
  
      move-result v14
  
+     .line 275
      .local v14, numTasks:I
      const/4 v8, 0x0
  
***************
*** 175,180 ****
--- 191,197 ----
  
      if-ge v11, v0, :cond_3
  
+     .line 276
      move-object/from16 v0, v16
  
      invoke-interface {v0, v8}, Ljava/util/List;->get(I)Ljava/lang/Object;
***************
*** 183,188 ****
--- 200,206 ----
  
      check-cast v12, Landroid/app/ActivityManager$RecentTaskInfo;
  
+     .line 281
      .local v12, info:Landroid/app/ActivityManager$RecentTaskInfo;
      new-instance v13, Landroid/content/Intent;
  
***************
*** 194,199 ****
--- 212,218 ----
  
      invoke-direct {v13, v0}, Landroid/content/Intent;-><init>(Landroid/content/Intent;)V
  
+     .line 282
      .local v13, intent:Landroid/content/Intent;
      iget-object v0, v12, Landroid/app/ActivityManager$RecentTaskInfo;->origActivity:Landroid/content/ComponentName;
  
***************
*** 201,206 ****
--- 220,226 ----
  
      if-eqz v21, :cond_0
  
+     .line 283
      iget-object v0, v12, Landroid/app/ActivityManager$RecentTaskInfo;->origActivity:Landroid/content/ComponentName;
  
      move-object/from16 v21, v0
***************
*** 209,217 ****
--- 229,239 ----
  
      invoke-virtual {v13, v0}, Landroid/content/Intent;->setComponent(Landroid/content/ComponentName;)Landroid/content/Intent;
  
+     .line 287
      :cond_0
      if-eqz v7, :cond_2
  
+     .line 288
      iget-object v0, v7, Landroid/content/pm/ActivityInfo;->packageName:Ljava/lang/String;
  
      move-object/from16 v21, v0
***************
*** 248,259 ****
--- 270,283 ----
  
      if-eqz v21, :cond_2
  
+     .line 275
      :cond_1
      :goto_1
      add-int/lit8 v8, v8, 0x1
  
      goto :goto_0
  
+     .line 296
      :cond_2
      invoke-virtual {v13}, Landroid/content/Intent;->getFlags()I
  
***************
*** 271,276 ****
--- 295,301 ----
  
      invoke-virtual {v13, v0}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
+     .line 298
      const/16 v21, 0x0
  
      move/from16 v0, v21
***************
*** 279,291 ****
--- 304,319 ----
  
      move-result-object v17
  
+     .line 299
      .local v17, resolveInfo:Landroid/content/pm/ResolveInfo;
      if-eqz v17, :cond_1
  
+     .line 300
      move-object/from16 v0, v17
  
      iget-object v4, v0, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
+     .line 301
      .local v4, activityInfo:Landroid/content/pm/ActivityInfo;
      invoke-virtual {v4, v15}, Landroid/content/pm/ActivityInfo;->loadLabel(Landroid/content/pm/PackageManager;)Ljava/lang/CharSequence;
  
***************
*** 295,305 ****
--- 323,335 ----
  
      move-result-object v19
  
+     .line 302
      .local v19, title:Ljava/lang/String;
      invoke-virtual {v4, v15}, Landroid/content/pm/ActivityInfo;->loadIcon(Landroid/content/pm/PackageManager;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v9
  
+     .line 304
      .local v9, icon:Landroid/graphics/drawable/Drawable;
      if-eqz v19, :cond_1
  
***************
*** 311,316 ****
--- 341,347 ----
  
      if-eqz v9, :cond_1
  
+     .line 305
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
***************
*** 319,324 ****
--- 350,356 ----
  
      aget-object v20, v21, v11
  
+     .line 306
      .local v20, tv:Landroid/widget/TextView;
      move-object/from16 v0, v20
  
***************
*** 326,335 ****
--- 358,369 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/TextView;->setText(Ljava/lang/CharSequence;)V
  
+     .line 307
      invoke-virtual {v10, v9}, Lcom/android/internal/policy/impl/IconUtilities;->createIconDrawable(Landroid/graphics/drawable/Drawable;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v9
  
+     .line 308
      const/16 v21, 0x0
  
      const/16 v22, 0x0
***************
*** 346,351 ****
--- 380,386 ----
  
      invoke-virtual {v0, v1, v9, v2, v3}, Landroid/widget/TextView;->setCompoundDrawables(Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;Landroid/graphics/drawable/Drawable;)V
  
+     .line 309
      new-instance v18, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;
  
      move-object/from16 v0, v18
***************
*** 354,388 ****
--- 389,431 ----
  
      invoke-direct {v0, v1}, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;-><init>(Lcom/android/internal/policy/impl/RecentApplicationsDialog;)V
  
+     .line 310
      .local v18, tag:Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;
      move-object/from16 v0, v18
  
      iput-object v12, v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;->info:Landroid/app/ActivityManager$RecentTaskInfo;
  
+     .line 311
      move-object/from16 v0, v18
  
      iput-object v13, v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;->intent:Landroid/content/Intent;
  
+     .line 312
      move-object/from16 v0, v20
  
      move-object/from16 v1, v18
  
      invoke-virtual {v0, v1}, Landroid/widget/TextView;->setTag(Ljava/lang/Object;)V
  
+     .line 313
      const/16 v21, 0x0
  
      invoke-virtual/range {v20 .. v21}, Landroid/widget/TextView;->setVisibility(I)V
  
+     .line 314
      const/16 v21, 0x0
  
      invoke-virtual/range {v20 .. v21}, Landroid/widget/TextView;->setPressed(Z)V
  
+     .line 315
      invoke-virtual/range {v20 .. v20}, Landroid/widget/TextView;->clearFocus()V
  
+     .line 316
      add-int/lit8 v11, v11, 0x1
  
      goto/16 :goto_1
  
+     .line 322
      .end local v4           #activityInfo:Landroid/content/pm/ActivityInfo;
      .end local v9           #icon:Landroid/graphics/drawable/Drawable;
      .end local v12           #info:Landroid/app/ActivityManager$RecentTaskInfo;
***************
*** 409,414 ****
--- 452,458 ----
  
      invoke-virtual {v0, v1}, Landroid/view/View;->setVisibility(I)V
  
+     .line 325
      :goto_3
      const/16 v21, 0x8
  
***************
*** 416,421 ****
--- 460,466 ----
  
      if-ge v11, v0, :cond_5
  
+     .line 326
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
***************
*** 428,442 ****
--- 473,490 ----
  
      invoke-virtual/range {v21 .. v22}, Landroid/widget/TextView;->setVisibility(I)V
  
+     .line 325
      add-int/lit8 v11, v11, 0x1
  
      goto :goto_3
  
+     .line 322
      :cond_4
      const/16 v21, 0x8
  
      goto :goto_2
  
+     .line 328
      :cond_5
      return-void
  .end method
***************
*** 446,457 ****
--- 494,507 ----
      .parameter "tag"
  
      .prologue
+     .line 202
      iget-object v2, p1, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;->info:Landroid/app/ActivityManager$RecentTaskInfo;
  
      iget v2, v2, Landroid/app/ActivityManager$RecentTaskInfo;->id:I
  
      if-ltz v2, :cond_1
  
+     .line 204
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->getContext()Landroid/content/Context;
  
      move-result-object v2
***************
*** 464,469 ****
--- 514,520 ----
  
      check-cast v0, Landroid/app/ActivityManager;
  
+     .line 206
      .local v0, am:Landroid/app/ActivityManager;
      iget-object v2, p1, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;->info:Landroid/app/ActivityManager$RecentTaskInfo;
  
***************
*** 473,494 ****
--- 524,549 ----
  
      invoke-virtual {v0, v2, v3}, Landroid/app/ActivityManager;->moveTaskToFront(II)V
  
+     .line 216
      .end local v0           #am:Landroid/app/ActivityManager;
      :cond_0
      :goto_0
      return-void
  
+     .line 207
      :cond_1
      iget-object v2, p1, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;->intent:Landroid/content/Intent;
  
      if-eqz v2, :cond_0
  
+     .line 208
      iget-object v2, p1, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;->intent:Landroid/content/Intent;
  
      const v3, 0x104000
  
      invoke-virtual {v2, v3}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 211
      :try_start_0
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->getContext()Landroid/content/Context;
  
***************
*** 502,510 ****
--- 557,567 ----
  
      goto :goto_0
  
+     .line 212
      :catch_0
      move-exception v1
  
+     .line 213
      .local v1, e:Landroid/content/ActivityNotFoundException;
      const-string v2, "Recent"
  
***************
*** 521,533 ****
--- 578,593 ----
      .locals 4
  
      .prologue
+     .line 168
      iget-object v3, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      array-length v1, v3
  
+     .line 169
      .local v1, numIcons:I
      const/4 v2, 0x0
  
+     .line 170
      .local v2, tag:Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;
      const/4 v0, 0x0
  
***************
*** 535,540 ****
--- 595,601 ----
      :goto_0
      if-ge v0, v1, :cond_0
  
+     .line 171
      iget-object v3, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      aget-object v3, v3, v0
***************
*** 545,560 ****
--- 606,626 ----
  
      if-eqz v3, :cond_2
  
+     .line 181
      :cond_0
      if-eqz v2, :cond_1
  
+     .line 182
      invoke-direct {p0, v2}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->switchTo(Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;)V
  
+     .line 184
      :cond_1
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->dismiss()V
  
+     .line 185
      return-void
  
+     .line 174
      :cond_2
      if-eqz v0, :cond_3
  
***************
*** 568,573 ****
--- 634,640 ----
  
      if-eqz v3, :cond_4
  
+     .line 175
      :cond_3
      iget-object v3, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
***************
*** 580,585 ****
--- 647,653 ----
      .end local v2           #tag:Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;
      check-cast v2, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;
  
+     .line 176
      .restart local v2       #tag:Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;
      iget-object v3, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
***************
*** 591,596 ****
--- 659,665 ----
  
      if-nez v3, :cond_0
  
+     .line 170
      :cond_4
      add-int/lit8 v0, v0, 0x1
  
***************
*** 602,607 ****
--- 671,677 ----
      .parameter "v"
  
      .prologue
+     .line 191
      iget-object v0, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      .local v0, arr$:[Landroid/widget/TextView;
***************
*** 616,640 ****
--- 686,716 ----
  
      aget-object v1, v0, v2
  
+     .line 192
      .local v1, b:Landroid/widget/TextView;
      if-ne v1, p1, :cond_1
  
+     .line 193
      invoke-virtual {v1}, Landroid/widget/TextView;->getTag()Ljava/lang/Object;
  
      move-result-object v4
  
      check-cast v4, Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;
  
+     .line 194
      .local v4, tag:Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;
      invoke-direct {p0, v4}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->switchTo(Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;)V
  
+     .line 198
      .end local v1           #b:Landroid/widget/TextView;
      .end local v4           #tag:Lcom/android/internal/policy/impl/RecentApplicationsDialog$RecentTag;
      :cond_0
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->dismiss()V
  
+     .line 199
      return-void
  
+     .line 191
      .restart local v1       #b:Landroid/widget/TextView;
      :cond_1
      add-int/lit8 v2, v2, 0x1
***************
*** 657,673 ****
--- 733,753 ----
  
      const/4 v8, -0x1
  
+     .line 87
      invoke-super {p0, p1}, Landroid/app/Dialog;->onCreate(Landroid/os/Bundle;)V
  
+     .line 89
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->getContext()Landroid/content/Context;
  
      move-result-object v2
  
+     .line 91
      .local v2, context:Landroid/content/Context;
      sget-object v7, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->sStatusBar:Landroid/app/StatusBarManager;
  
      if-nez v7, :cond_0
  
+     .line 92
      const-string v7, "statusbar"
  
      invoke-virtual {v2, v7}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 678,718 ****
--- 758,810 ----
  
      sput-object v7, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->sStatusBar:Landroid/app/StatusBarManager;
  
+     .line 95
      :cond_0
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->getWindow()Landroid/view/Window;
  
      move-result-object v6
  
+     .line 96
      .local v6, window:Landroid/view/Window;
      invoke-virtual {v6, v10}, Landroid/view/Window;->requestFeature(I)Z
  
+     .line 97
      const/16 v7, 0x7d8
  
      invoke-virtual {v6, v7}, Landroid/view/Window;->setType(I)V
  
+     .line 98
      invoke-virtual {v6, v12, v12}, Landroid/view/Window;->setFlags(II)V
  
+     .line 100
      const-string v7, "Recents"
  
      invoke-virtual {v6, v7}, Landroid/view/Window;->setTitle(Ljava/lang/CharSequence;)V
  
+     .line 102
      const v7, 0x109007c
  
      invoke-virtual {p0, v7}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->setContentView(I)V
  
+     .line 104
      invoke-virtual {v6}, Landroid/view/Window;->getAttributes()Landroid/view/WindowManager$LayoutParams;
  
      move-result-object v5
  
+     .line 105
      .local v5, params:Landroid/view/WindowManager$LayoutParams;
      iput v8, v5, Landroid/view/WindowManager$LayoutParams;->width:I
  
+     .line 106
      iput v8, v5, Landroid/view/WindowManager$LayoutParams;->height:I
  
+     .line 107
      invoke-virtual {v6, v5}, Landroid/view/Window;->setAttributes(Landroid/view/WindowManager$LayoutParams;)V
  
+     .line 108
      invoke-virtual {v6, v9, v11}, Landroid/view/Window;->setFlags(II)V
  
+     .line 110
      iget-object v8, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      const v7, 0x10202dd
***************
*** 725,730 ****
--- 817,823 ----
  
      aput-object v7, v8, v9
  
+     .line 111
      iget-object v8, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      const v7, 0x1020019
***************
*** 737,742 ****
--- 830,836 ----
  
      aput-object v7, v8, v10
  
+     .line 112
      iget-object v8, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      const v7, 0x102001a
***************
*** 749,754 ****
--- 843,849 ----
  
      aput-object v7, v8, v11
  
+     .line 113
      iget-object v8, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      const/4 v9, 0x3
***************
*** 763,768 ****
--- 858,864 ----
  
      aput-object v7, v8, v9
  
+     .line 114
      iget-object v8, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      const/4 v9, 0x4
***************
*** 777,782 ****
--- 873,879 ----
  
      aput-object v7, v8, v9
  
+     .line 115
      iget-object v8, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      const/4 v9, 0x5
***************
*** 791,796 ****
--- 888,894 ----
  
      aput-object v7, v8, v9
  
+     .line 116
      iget-object v8, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      const/4 v9, 0x6
***************
*** 805,810 ****
--- 903,909 ----
  
      aput-object v7, v8, v9
  
+     .line 117
      iget-object v8, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      const/4 v9, 0x7
***************
*** 819,824 ****
--- 918,924 ----
  
      aput-object v7, v8, v9
  
+     .line 118
      const v7, 0x10202dc
  
      invoke-virtual {p0, v7}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->findViewById(I)Landroid/view/View;
***************
*** 827,832 ****
--- 927,933 ----
  
      iput-object v7, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mNoAppsText:Landroid/view/View;
  
+     .line 120
      iget-object v0, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      .local v0, arr$:[Landroid/widget/TextView;
***************
*** 841,853 ****
--- 942,957 ----
  
      aget-object v1, v0, v3
  
+     .line 121
      .local v1, b:Landroid/widget/TextView;
      invoke-virtual {v1, p0}, Landroid/widget/TextView;->setOnClickListener(Landroid/view/View$OnClickListener;)V
  
+     .line 120
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 123
      .end local v1           #b:Landroid/widget/TextView;
      :cond_1
      return-void
***************
*** 861,882 ****
--- 965,991 ----
      .prologue
      const/4 v6, 0x1
  
+     .line 127
      const/16 v7, 0x3d
  
      if-ne p1, v7, :cond_7
  
+     .line 131
      invoke-virtual {p2}, Landroid/view/KeyEvent;->isShiftPressed()Z
  
      move-result v0
  
+     .line 132
      .local v0, backward:Z
      iget-object v7, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      array-length v5, v7
  
+     .line 133
      .local v5, numIcons:I
      const/4 v4, 0x0
  
+     .line 134
      .local v4, numButtons:I
      :goto_0
      if-ge v4, v5, :cond_0
***************
*** 891,907 ****
--- 1000,1020 ----
  
      if-nez v7, :cond_0
  
+     .line 135
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
  
+     .line 137
      :cond_0
      if-eqz v4, :cond_2
  
+     .line 138
      if-eqz v0, :cond_3
  
      add-int/lit8 v3, v4, -0x1
  
+     .line 139
      .local v3, nextFocus:I
      :goto_1
      const/4 v2, 0x0
***************
*** 910,915 ****
--- 1023,1029 ----
      :goto_2
      if-ge v2, v4, :cond_1
  
+     .line 140
      iget-object v7, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      aget-object v7, v7, v2
***************
*** 920,939 ****
--- 1034,1057 ----
  
      if-eqz v7, :cond_5
  
+     .line 141
      if-eqz v0, :cond_4
  
+     .line 142
      add-int v7, v2, v4
  
      add-int/lit8 v7, v7, -0x1
  
      rem-int v3, v7, v4
  
+     .line 149
      :cond_1
      :goto_3
      if-eqz v0, :cond_6
  
      move v1, v6
  
+     .line 150
      .local v1, direction:I
      :goto_4
      iget-object v7, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
***************
*** 946,951 ****
--- 1064,1070 ----
  
      if-eqz v7, :cond_2
  
+     .line 151
      iget-object v7, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mIcons:[Landroid/widget/TextView;
  
      aget-object v7, v7, v3
***************
*** 956,961 ****
--- 1075,1081 ----
  
      invoke-virtual {v7, v8}, Landroid/widget/TextView;->playSoundEffect(I)V
  
+     .line 161
      .end local v0           #backward:Z
      .end local v1           #direction:I
      .end local v2           #i:I
***************
*** 966,971 ****
--- 1086,1092 ----
      :goto_5
      return v6
  
+     .line 138
      .restart local v0       #backward:Z
      .restart local v4       #numButtons:I
      .restart local v5       #numIcons:I
***************
*** 974,979 ****
--- 1095,1101 ----
  
      goto :goto_1
  
+     .line 144
      .restart local v2       #i:I
      .restart local v3       #nextFocus:I
      :cond_4
***************
*** 981,998 ****
--- 1103,1124 ----
  
      rem-int v3, v7, v4
  
+     .line 146
      goto :goto_3
  
+     .line 139
      :cond_5
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_2
  
+     .line 149
      :cond_6
      const/4 v1, 0x2
  
      goto :goto_4
  
+     .line 161
      .end local v0           #backward:Z
      .end local v2           #i:I
      .end local v3           #nextFocus:I
***************
*** 1010,1029 ****
--- 1136,1160 ----
      .locals 3
  
      .prologue
+     .line 223
      invoke-super {p0}, Landroid/app/Dialog;->onStart()V
  
+     .line 224
      invoke-direct {p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->reloadButtons()V
  
+     .line 225
      sget-object v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->sStatusBar:Landroid/app/StatusBarManager;
  
      if-eqz v0, :cond_0
  
+     .line 226
      sget-object v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->sStatusBar:Landroid/app/StatusBarManager;
  
      const/high16 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/app/StatusBarManager;->disable(I)V
  
+     .line 230
      :cond_0
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->getContext()Landroid/content/Context;
  
***************
*** 1035,1046 ****
--- 1166,1179 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
  
+     .line 232
      iget-object v0, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mHandler:Landroid/os/Handler;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mCleanup:Ljava/lang/Runnable;
  
      invoke-virtual {v0, v1}, Landroid/os/Handler;->removeCallbacks(Ljava/lang/Runnable;)V
  
+     .line 233
      return-void
  .end method
  
***************
*** 1048,1065 ****
--- 1181,1202 ----
      .locals 4
  
      .prologue
+     .line 240
      invoke-super {p0}, Landroid/app/Dialog;->onStop()V
  
+     .line 242
      sget-object v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->sStatusBar:Landroid/app/StatusBarManager;
  
      if-eqz v0, :cond_0
  
+     .line 243
      sget-object v0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->sStatusBar:Landroid/app/StatusBarManager;
  
      const/4 v1, 0x0
  
      invoke-virtual {v0, v1}, Landroid/app/StatusBarManager;->disable(I)V
  
+     .line 247
      :cond_0
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->getContext()Landroid/content/Context;
  
***************
*** 1069,1074 ****
--- 1206,1212 ----
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->unregisterReceiver(Landroid/content/BroadcastReceiver;)V
  
+     .line 249
      iget-object v0, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mHandler:Landroid/os/Handler;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/RecentApplicationsDialog;->mCleanup:Ljava/lang/Runnable;
***************
*** 1077,1081 ****
--- 1215,1220 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/os/Handler;->postDelayed(Ljava/lang/Runnable;J)Z
  
+     .line 250
      return-void
  .end method
