*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter.smali	2012-03-23 19:37:46.409621361 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter.smali.original	2012-03-23 19:34:46.327337565 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 884
      iput-object p1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;->this$0:Lcom/android/internal/telephony/sip/SipPhone;
  
      invoke-direct {p0}, Landroid/net/sip/SipAudioCall$Listener;-><init>()V
***************
*** 37,42 ****
--- 38,44 ----
      .parameter "x1"
  
      .prologue
+     .line 884
      invoke-direct {p0, p1}, Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;-><init>(Lcom/android/internal/telephony/sip/SipPhone;)V
  
      return-void
***************
*** 49,58 ****
--- 51,62 ----
      .parameter "call"
  
      .prologue
+     .line 897
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->BUSY:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;->onCallEnded(Lcom/android/internal/telephony/Connection$DisconnectCause;)V
  
+     .line 898
      return-void
  .end method
  
***************
*** 61,66 ****
--- 65,71 ----
      .parameter "call"
  
      .prologue
+     .line 890
      invoke-virtual {p1}, Landroid/net/sip/SipAudioCall;->isInCall()Z
  
      move-result v0
***************
*** 72,79 ****
--- 77,86 ----
      :goto_0
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;->onCallEnded(Lcom/android/internal/telephony/Connection$DisconnectCause;)V
  
+     .line 893
      return-void
  
+     .line 890
      :cond_0
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->INCOMING_MISSED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
***************
*** 90,97 ****
--- 97,106 ----
      .parameter "errorMessage"
  
      .prologue
+     .line 903
      packed-switch p2, :pswitch_data_0
  
+     .line 932
      :pswitch_0
      const-string v0, "SipPhone"
  
***************
*** 129,141 ****
--- 138,153 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 934
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->ERROR_UNSPECIFIED:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;->onError(Lcom/android/internal/telephony/Connection$DisconnectCause;)V
  
+     .line 936
      :goto_0
      return-void
  
+     .line 905
      :pswitch_1
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->SERVER_UNREACHABLE:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
***************
*** 143,148 ****
--- 155,161 ----
  
      goto :goto_0
  
+     .line 908
      :pswitch_2
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->NUMBER_UNREACHABLE:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
***************
*** 150,155 ****
--- 163,169 ----
  
      goto :goto_0
  
+     .line 911
      :pswitch_3
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->INVALID_NUMBER:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
***************
*** 157,162 ****
--- 171,177 ----
  
      goto :goto_0
  
+     .line 915
      :pswitch_4
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->TIMED_OUT:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
***************
*** 164,169 ****
--- 179,185 ----
  
      goto :goto_0
  
+     .line 918
      :pswitch_5
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->LOST_SIGNAL:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
***************
*** 171,176 ****
--- 187,193 ----
  
      goto :goto_0
  
+     .line 921
      :pswitch_6
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->INVALID_CREDENTIALS:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
***************
*** 178,183 ****
--- 195,201 ----
  
      goto :goto_0
  
+     .line 924
      :pswitch_7
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->OUT_OF_NETWORK:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
***************
*** 185,190 ****
--- 203,209 ----
  
      goto :goto_0
  
+     .line 927
      :pswitch_8
      sget-object v0, Lcom/android/internal/telephony/Connection$DisconnectCause;->SERVER_ERROR:Lcom/android/internal/telephony/Connection$DisconnectCause;
  
***************
*** 192,197 ****
--- 211,217 ----
  
      goto :goto_0
  
+     .line 903
      nop
  
      :pswitch_data_0
