*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/sip/SipPhone$SipConnection.smali	2012-03-23 19:37:46.751627596 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/sip/SipPhone$SipConnection.smali.original	2012-03-23 19:34:46.330337619 -0500
***************
*** 40,45 ****
--- 40,46 ----
      .parameter "callee"
  
      .prologue
+     .line 698
      #calls: Lcom/android/internal/telephony/sip/SipPhone;->getUriString(Landroid/net/sip/SipProfile;)Ljava/lang/String;
      invoke-static {p1, p3}, Lcom/android/internal/telephony/sip/SipPhone;->access$1300(Lcom/android/internal/telephony/sip/SipPhone;Landroid/net/sip/SipProfile;)Ljava/lang/String;
  
***************
*** 47,52 ****
--- 48,54 ----
  
      invoke-direct {p0, p1, p2, p3, v0}, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$SipCall;Landroid/net/sip/SipProfile;Ljava/lang/String;)V
  
+     .line 699
      return-void
  .end method
  
***************
*** 58,87 ****
--- 60,98 ----
      .parameter "originalNumber"
  
      .prologue
+     .line 690
      iput-object p1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->this$0:Lcom/android/internal/telephony/sip/SipPhone;
  
+     .line 691
      invoke-direct {p0, p4}, Lcom/android/internal/telephony/sip/SipConnectionBase;-><init>(Ljava/lang/String;)V
  
+     .line 611
      sget-object v0, Lcom/android/internal/telephony/Call$State;->IDLE:Lcom/android/internal/telephony/Call$State;
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mState:Lcom/android/internal/telephony/Call$State;
  
+     .line 614
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mIncoming:Z
  
+     .line 616
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection$1;
  
      invoke-direct {v0, p0}, Lcom/android/internal/telephony/sip/SipPhone$SipConnection$1;-><init>(Lcom/android/internal/telephony/sip/SipPhone$SipConnection;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mAdapter:Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;
  
+     .line 692
      iput-object p2, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mOwner:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
+     .line 693
      iput-object p3, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mPeer:Landroid/net/sip/SipProfile;
  
+     .line 694
      iput-object p4, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mOriginalNumber:Ljava/lang/String;
  
+     .line 695
      return-void
  .end method
  
***************
*** 90,95 ****
--- 101,107 ----
      .parameter "x0"
  
      .prologue
+     .line 608
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mState:Lcom/android/internal/telephony/Call$State;
  
      return-object v0
***************
*** 100,105 ****
--- 112,118 ----
      .parameter "x0"
  
      .prologue
+     .line 608
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      return-object v0
***************
*** 111,116 ****
--- 124,130 ----
      .parameter "x1"
  
      .prologue
+     .line 608
      iput-object p1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      return-object p1
***************
*** 121,126 ****
--- 135,141 ----
      .parameter "x0"
  
      .prologue
+     .line 608
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mPeer:Landroid/net/sip/SipProfile;
  
      return-object v0
***************
*** 131,136 ****
--- 146,152 ----
      .parameter "x0"
  
      .prologue
+     .line 608
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mOwner:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      return-object v0
***************
*** 147,152 ****
--- 163,169 ----
      .end annotation
  
      .prologue
+     .line 722
      :try_start_0
      iget-object v1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
***************
*** 156,166 ****
--- 173,186 ----
      :try_end_0
      .catch Landroid/net/sip/SipException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 726
      return-void
  
+     .line 723
      :catch_0
      move-exception v0
  
+     .line 724
      .local v0, e:Landroid/net/sip/SipException;
      new-instance v1, Lcom/android/internal/telephony/CallStateException;
  
***************
*** 192,199 ****
--- 212,221 ----
      .parameter "owner"
  
      .prologue
+     .line 729
      iput-object p1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mOwner:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
+     .line 730
      return-void
  .end method
  
***************
*** 206,215 ****
--- 228,239 ----
      .end annotation
  
      .prologue
+     .line 738
      sget-object v0, Lcom/android/internal/telephony/Call$State;->DIALING:Lcom/android/internal/telephony/Call$State;
  
      invoke-virtual {p0, v0}, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->setState(Lcom/android/internal/telephony/Call$State;)V
  
+     .line 739
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->this$0:Lcom/android/internal/telephony/sip/SipPhone;
  
      #getter for: Lcom/android/internal/telephony/sip/SipPhone;->mSipManager:Landroid/net/sip/SipManager;
***************
*** 236,247 ****
--- 260,273 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
+     .line 741
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      iget-object v1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mAdapter:Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;
  
      invoke-virtual {v0, v1}, Landroid/net/sip/SipAudioCall;->setListener(Landroid/net/sip/SipAudioCall$Listener;)V
  
+     .line 742
      return-void
  .end method
  
***************
*** 249,254 ****
--- 275,281 ----
      .locals 1
  
      .prologue
+     .line 796
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mOriginalNumber:Ljava/lang/String;
  
      return-object v0
***************
*** 258,269 ****
--- 285,298 ----
      .locals 1
  
      .prologue
+     .line 733
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      if-nez v0, :cond_0
  
      const/4 v0, 0x0
  
+     .line 734
      :goto_0
      return-object v0
  
***************
*** 281,286 ****
--- 310,316 ----
      .locals 1
  
      .prologue
+     .line 608
      invoke-virtual {p0}, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->getCall()Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      move-result-object v0
***************
*** 292,297 ****
--- 322,328 ----
      .locals 1
  
      .prologue
+     .line 801
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mOwner:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      return-object v0
***************
*** 301,312 ****
--- 332,345 ----
      .locals 2
  
      .prologue
+     .line 703
      iget-object v1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mPeer:Landroid/net/sip/SipProfile;
  
      invoke-virtual {v1}, Landroid/net/sip/SipProfile;->getDisplayName()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 704
      .local v0, displayName:Ljava/lang/String;
      invoke-static {v0}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
***************
*** 325,330 ****
--- 358,364 ----
      .locals 1
  
      .prologue
+     .line 770
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      if-nez v0, :cond_0
***************
*** 348,353 ****
--- 382,388 ----
      .locals 1
  
      .prologue
+     .line 710
      sget v0, Lcom/android/internal/telephony/Connection;->PRESENTATION_ALLOWED:I
  
      return v0
***************
*** 357,362 ****
--- 392,398 ----
      .locals 1
  
      .prologue
+     .line 806
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mOwner:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->getPhone()Lcom/android/internal/telephony/Phone;
***************
*** 370,375 ****
--- 406,412 ----
      .locals 1
  
      .prologue
+     .line 783
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mState:Lcom/android/internal/telephony/Call$State;
  
      return-object v0
***************
*** 384,393 ****
--- 421,432 ----
      .end annotation
  
      .prologue
+     .line 811
      const-class v4, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v4
  
+     .line 812
      :try_start_0
      const-string v2, "SipPhone"
  
***************
*** 447,452 ****
--- 486,492 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 815
      iget-object v2, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mState:Lcom/android/internal/telephony/Call$State;
  
      invoke-virtual {v2}, Lcom/android/internal/telephony/Call$State;->isAlive()Z
***************
*** 459,483 ****
--- 499,529 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 831
      :goto_0
      return-void
  
+     .line 817
      :cond_0
      :try_start_1
      iget-object v1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
+     .line 818
      .local v1, sipAudioCall:Landroid/net/sip/SipAudioCall;
      if-eqz v1, :cond_1
  
+     .line 819
      const/4 v2, 0x0
  
      invoke-virtual {v1, v2}, Landroid/net/sip/SipAudioCall;->setListener(Landroid/net/sip/SipAudioCall$Listener;)V
  
+     .line 820
      invoke-virtual {v1}, Landroid/net/sip/SipAudioCall;->endCall()V
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_1
      .catch Landroid/net/sip/SipException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 825
      :cond_1
      :try_start_2
      iget-object v3, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mAdapter:Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;
***************
*** 500,505 ****
--- 546,552 ----
      :goto_1
      invoke-virtual {v3, v2}, Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;->onCallEnded(Lcom/android/internal/telephony/Connection$DisconnectCause;)V
  
+     .line 830
      monitor-exit v4
  
      goto :goto_0
***************
*** 514,522 ****
--- 561,571 ----
  
      throw v2
  
+     .line 822
      :catch_0
      move-exception v0
  
+     .line 823
      .local v0, e:Landroid/net/sip/SipException;
      :try_start_3
      new-instance v2, Lcom/android/internal/telephony/CallStateException;
***************
*** 545,550 ****
--- 594,600 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_1
  
+     .line 825
      .end local v0           #e:Landroid/net/sip/SipException;
      :catchall_1
      move-exception v2
***************
*** 595,604 ****
--- 645,656 ----
      .end annotation
  
      .prologue
+     .line 745
      sget-object v1, Lcom/android/internal/telephony/Call$State;->HOLDING:Lcom/android/internal/telephony/Call$State;
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->setState(Lcom/android/internal/telephony/Call$State;)V
  
+     .line 747
      :try_start_0
      iget-object v1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
***************
*** 608,618 ****
--- 660,673 ----
      :try_end_0
      .catch Landroid/net/sip/SipException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 751
      return-void
  
+     .line 748
      :catch_0
      move-exception v0
  
+     .line 749
      .local v0, e:Landroid/net/sip/SipException;
      new-instance v1, Lcom/android/internal/telephony/CallStateException;
  
***************
*** 645,662 ****
--- 700,722 ----
      .parameter "newState"
  
      .prologue
+     .line 714
      invoke-virtual {p0, p2}, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->setState(Lcom/android/internal/telephony/Call$State;)V
  
+     .line 715
      iput-object p1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
+     .line 716
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mAdapter:Lcom/android/internal/telephony/sip/SipPhone$SipAudioCallAdapter;
  
      invoke-virtual {p1, v0}, Landroid/net/sip/SipAudioCall;->setListener(Landroid/net/sip/SipAudioCall$Listener;)V
  
+     .line 717
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mIncoming:Z
  
+     .line 718
      return-void
  .end method
  
***************
*** 664,669 ****
--- 724,730 ----
      .locals 1
  
      .prologue
+     .line 788
      iget-boolean v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mIncoming:Z
  
      return v0
***************
*** 678,687 ****
--- 739,750 ----
      .end annotation
  
      .prologue
+     .line 835
      const-class v4, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v4
  
+     .line 836
      :try_start_0
      invoke-virtual {p0}, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->getPhone()Lcom/android/internal/telephony/Phone;
  
***************
*** 701,706 ****
--- 764,770 ----
  
      move-object v1, v3
  
+     .line 839
      .local v1, call:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
      :goto_0
      invoke-virtual {v1}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->getState()Lcom/android/internal/telephony/Call$State;
***************
*** 711,716 ****
--- 775,781 ----
  
      if-eq v3, v5, :cond_1
  
+     .line 840
      new-instance v3, Lcom/android/internal/telephony/CallStateException;
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 739,744 ****
--- 804,810 ----
  
      throw v3
  
+     .line 862
      .end local v1           #call:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
      :catchall_0
      move-exception v3
***************
*** 749,754 ****
--- 815,821 ----
  
      throw v3
  
+     .line 836
      :cond_0
      :try_start_1
      iget-object v3, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->this$0:Lcom/android/internal/telephony/sip/SipPhone;
***************
*** 763,768 ****
--- 830,836 ----
  
      goto :goto_0
  
+     .line 844
      .restart local v1       #call:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
      :cond_1
      const-string v3, "SipPhone"
***************
*** 771,777 ****
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v6, "separate conn: "
  
      invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 839,845 ----
  
      invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v6, "separate conn: "
  
      invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 815,840 ****
--- 883,914 ----
  
      invoke-static {v3, v5}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 849
      invoke-virtual {p0}, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->getPhone()Lcom/android/internal/telephony/Phone;
  
      move-result-object v2
  
+     .line 850
      .local v2, originalPhone:Lcom/android/internal/telephony/Phone;
      #calls: Lcom/android/internal/telephony/sip/SipPhone$SipCall;->getAudioGroup()Landroid/net/rtp/AudioGroup;
      invoke-static {v1}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->access$1500(Lcom/android/internal/telephony/sip/SipPhone$SipCall;)Landroid/net/rtp/AudioGroup;
  
      move-result-object v0
  
+     .line 851
      .local v0, audioGroup:Landroid/net/rtp/AudioGroup;
      #calls: Lcom/android/internal/telephony/sip/SipPhone$SipCall;->add(Lcom/android/internal/telephony/sip/SipPhone$SipConnection;)V
      invoke-static {v1, p0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->access$1600(Lcom/android/internal/telephony/sip/SipPhone$SipCall;Lcom/android/internal/telephony/sip/SipPhone$SipConnection;)V
  
+     .line 852
      iget-object v3, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      invoke-virtual {v3, v0}, Landroid/net/sip/SipAudioCall;->setAudioGroup(Landroid/net/rtp/AudioGroup;)V
  
+     .line 856
      invoke-interface {v2}, Lcom/android/internal/telephony/Phone;->switchHoldingAndActive()V
  
+     .line 859
      iget-object v3, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->this$0:Lcom/android/internal/telephony/sip/SipPhone;
  
      invoke-virtual {v3}, Lcom/android/internal/telephony/sip/SipPhone;->getForegroundCall()Lcom/android/internal/telephony/Call;
***************
*** 844,860 ****
--- 918,938 ----
      .end local v1           #call:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
      check-cast v1, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
+     .line 860
      .restart local v1       #call:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
      iget-object v3, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      invoke-virtual {v3}, Landroid/net/sip/SipAudioCall;->startAudio()V
  
+     .line 861
      invoke-virtual {v1, p0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->onConnectionStateChanged(Lcom/android/internal/telephony/sip/SipPhone$SipConnection;)V
  
+     .line 862
      monitor-exit v4
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 863
      return-void
  .end method
  
***************
*** 863,868 ****
--- 941,947 ----
      .parameter "muted"
  
      .prologue
+     .line 764
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      if-eqz v0, :cond_0
***************
*** 875,884 ****
--- 954,965 ----
  
      if-eq p1, v0, :cond_0
  
+     .line 765
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      invoke-virtual {v0}, Landroid/net/sip/SipAudioCall;->toggleMute()V
  
+     .line 767
      :cond_0
      return-void
  .end method
***************
*** 888,903 ****
--- 969,988 ----
      .parameter "state"
  
      .prologue
+     .line 776
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mState:Lcom/android/internal/telephony/Call$State;
  
      if-ne p1, v0, :cond_0
  
+     .line 779
      :goto_0
      return-void
  
+     .line 777
      :cond_0
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipConnectionBase;->setState(Lcom/android/internal/telephony/Call$State;)V
  
+     .line 778
      iput-object p1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mState:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
***************
*** 913,926 ****
--- 998,1014 ----
      .end annotation
  
      .prologue
+     .line 754
      iget-object v1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
      invoke-virtual {v1, p1}, Landroid/net/sip/SipAudioCall;->setAudioGroup(Landroid/net/rtp/AudioGroup;)V
  
+     .line 755
      sget-object v1, Lcom/android/internal/telephony/Call$State;->ACTIVE:Lcom/android/internal/telephony/Call$State;
  
      invoke-virtual {p0, v1}, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->setState(Lcom/android/internal/telephony/Call$State;)V
  
+     .line 757
      :try_start_0
      iget-object v1, p0, Lcom/android/internal/telephony/sip/SipPhone$SipConnection;->mSipAudioCall:Landroid/net/sip/SipAudioCall;
  
***************
*** 930,940 ****
--- 1018,1031 ----
      :try_end_0
      .catch Landroid/net/sip/SipException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 761
      return-void
  
+     .line 758
      :catch_0
      move-exception v0
  
+     .line 759
      .local v0, e:Landroid/net/sip/SipException;
      new-instance v1, Lcom/android/internal/telephony/CallStateException;
  
***************
*** 942,948 ****
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "unhold(): "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1033,1039 ----
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "unhold(): "
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
