*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/sip/SipPhone.smali	2012-03-23 19:37:46.558624078 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/sip/SipPhone.smali.original	2012-03-23 19:34:46.328337583 -0500
***************
*** 48,80 ****
      .prologue
      const/4 v3, 0x0
  
      invoke-direct {p0, p1, p2}, Lcom/android/internal/telephony/sip/SipPhoneBase;-><init>(Landroid/content/Context;Lcom/android/internal/telephony/PhoneNotifier;)V
  
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-direct {v0, p0, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$1;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-direct {v0, p0, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$1;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-direct {v0, p0, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$1;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->backgroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      const-string v0, "SipPhone"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "new SipPhone: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 48,85 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 64
      invoke-direct {p0, p1, p2}, Lcom/android/internal/telephony/sip/SipPhoneBase;-><init>(Landroid/content/Context;Lcom/android/internal/telephony/PhoneNotifier;)V
  
+     .line 56
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-direct {v0, p0, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$1;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
+     .line 57
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-direct {v0, p0, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$1;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
+     .line 58
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-direct {v0, p0, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$1;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->backgroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
+     .line 66
      const-string v0, "SipPhone"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "new SipPhone: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 94,125 ****
--- 99,136 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 67
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-direct {v0, p0, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$1;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
+     .line 68
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-direct {v0, p0, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$1;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
+     .line 69
      new-instance v0, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-direct {v0, p0, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;-><init>(Lcom/android/internal/telephony/sip/SipPhone;Lcom/android/internal/telephony/sip/SipPhone$1;)V
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->backgroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
+     .line 70
      iput-object p3, p0, Lcom/android/internal/telephony/sip/SipPhone;->mProfile:Landroid/net/sip/SipProfile;
  
+     .line 71
      invoke-static {p1}, Landroid/net/sip/SipManager;->newInstance(Landroid/content/Context;)Landroid/net/sip/SipManager;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->mSipManager:Landroid/net/sip/SipManager;
  
+     .line 72
      return-void
  .end method
  
***************
*** 128,133 ****
--- 139,145 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->mProfile:Landroid/net/sip/SipProfile;
  
      return-object v0
***************
*** 138,143 ****
--- 150,156 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-static {p0}, Lcom/android/internal/telephony/sip/SipPhone;->getCallStateFrom(Landroid/net/sip/SipAudioCall;)Lcom/android/internal/telephony/Call$State;
  
      move-result-object v0
***************
*** 150,155 ****
--- 163,169 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      return-object v0
***************
*** 161,166 ****
--- 175,181 ----
      .parameter "x1"
  
      .prologue
+     .line 48
      invoke-direct {p0, p1}, Lcom/android/internal/telephony/sip/SipPhone;->getUriString(Landroid/net/sip/SipProfile;)Ljava/lang/String;
  
      move-result-object v0
***************
*** 173,178 ****
--- 188,194 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->mSipManager:Landroid/net/sip/SipManager;
  
      return-object v0
***************
*** 184,189 ****
--- 200,206 ----
      .parameter "x1"
  
      .prologue
+     .line 48
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/sip/SipPhone;->notifyNewRingingConnectionP(Lcom/android/internal/telephony/Connection;)V
  
      return-void
***************
*** 194,199 ****
--- 211,217 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      return-object v0
***************
*** 204,209 ****
--- 222,228 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 215,220 ****
--- 234,240 ----
      .parameter "x1"
  
      .prologue
+     .line 48
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/sip/SipPhone;->notifyDisconnectP(Lcom/android/internal/telephony/Connection;)V
  
      return-void
***************
*** 230,243 ****
--- 250,266 ----
      .end annotation
  
      .prologue
+     .line 165
      invoke-virtual {p0}, Lcom/android/internal/telephony/sip/SipPhone;->clearDisconnected()V
  
+     .line 167
      invoke-virtual {p0}, Lcom/android/internal/telephony/sip/SipPhone;->canDial()Z
  
      move-result v2
  
      if-nez v2, :cond_0
  
+     .line 168
      new-instance v2, Lcom/android/internal/telephony/CallStateException;
  
      const-string v3, "cannot dial in current state"
***************
*** 246,251 ****
--- 269,275 ----
  
      throw v2
  
+     .line 170
      :cond_0
      iget-object v2, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 257,264 ****
--- 281,290 ----
  
      if-ne v2, v3, :cond_1
  
+     .line 171
      invoke-virtual {p0}, Lcom/android/internal/telephony/sip/SipPhone;->switchHoldingAndActive()V
  
+     .line 173
      :cond_1
      iget-object v2, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 270,275 ****
--- 296,302 ----
  
      if-eq v2, v3, :cond_2
  
+     .line 175
      new-instance v2, Lcom/android/internal/telephony/CallStateException;
  
      const-string v3, "cannot dial in current state"
***************
*** 278,283 ****
--- 305,311 ----
  
      throw v2
  
+     .line 178
      :cond_2
      iget-object v2, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 285,290 ****
--- 313,319 ----
  
      invoke-virtual {v2, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->setMute(Z)V
  
+     .line 180
      :try_start_0
      iget-object v2, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 294,306 ****
--- 323,338 ----
  
      move-result-object v0
  
+     .line 181
      .local v0, c:Lcom/android/internal/telephony/Connection;
      return-object v0
  
+     .line 182
      .end local v0           #c:Lcom/android/internal/telephony/Connection;
      :catch_0
      move-exception v1
  
+     .line 183
      .local v1, e:Landroid/net/sip/SipException;
      const-string v2, "SipPhone"
  
***************
*** 308,313 ****
--- 340,346 ----
  
      invoke-static {v2, v3, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 184
      new-instance v2, Lcom/android/internal/telephony/CallStateException;
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 338,343 ****
--- 371,377 ----
      .parameter "sipAudioCall"
  
      .prologue
+     .line 868
      invoke-virtual {p0}, Landroid/net/sip/SipAudioCall;->isOnHold()Z
  
      move-result v1
***************
*** 346,362 ****
--- 380,400 ----
  
      sget-object v1, Lcom/android/internal/telephony/Call$State;->HOLDING:Lcom/android/internal/telephony/Call$State;
  
+     .line 880
      :goto_0
      return-object v1
  
+     .line 869
      :cond_0
      invoke-virtual {p0}, Landroid/net/sip/SipAudioCall;->getState()I
  
      move-result v0
  
+     .line 870
      .local v0, sessionState:I
      packed-switch v0, :pswitch_data_0
  
+     .line 879
      :pswitch_0
      const-string v1, "SipPhone"
  
***************
*** 380,419 ****
--- 418,465 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 880
      sget-object v1, Lcom/android/internal/telephony/Call$State;->DISCONNECTED:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 871
      :pswitch_1
      sget-object v1, Lcom/android/internal/telephony/Call$State;->IDLE:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 873
      :pswitch_2
      sget-object v1, Lcom/android/internal/telephony/Call$State;->INCOMING:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 874
      :pswitch_3
      sget-object v1, Lcom/android/internal/telephony/Call$State;->DIALING:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 875
      :pswitch_4
      sget-object v1, Lcom/android/internal/telephony/Call$State;->ALERTING:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 876
      :pswitch_5
      sget-object v1, Lcom/android/internal/telephony/Call$State;->DISCONNECTING:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 877
      :pswitch_6
      sget-object v1, Lcom/android/internal/telephony/Call$State;->ACTIVE:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 870
      nop
  
      :pswitch_data_0
***************
*** 435,444 ****
--- 481,492 ----
      .parameter "p"
  
      .prologue
+     .line 342
      invoke-virtual {p1}, Landroid/net/sip/SipProfile;->getSipDomain()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 344
      .local v0, domain:Ljava/lang/String;
      const-string v1, ":5060"
  
***************
*** 448,453 ****
--- 496,502 ----
  
      if-eqz v1, :cond_0
  
+     .line 345
      const/4 v1, 0x0
  
      invoke-virtual {v0}, Ljava/lang/String;->length()I
***************
*** 460,465 ****
--- 509,515 ----
  
      move-result-object v0
  
+     .line 347
      .end local v0           #domain:Ljava/lang/String;
      :cond_0
      return-object v0
***************
*** 470,475 ****
--- 520,526 ----
      .parameter "p"
  
      .prologue
+     .line 338
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 514,523 ****
--- 565,576 ----
      .end annotation
  
      .prologue
+     .line 133
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 134
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 539,544 ****
--- 592,598 ----
  
      if-ne v0, v2, :cond_1
  
+     .line 136
      :cond_0
      const-string v0, "SipPhone"
  
***************
*** 546,574 ****
  
      invoke-static {v0, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      const/4 v2, 0x0
  
      invoke-virtual {v0, v2}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->setMute(Z)V
  
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->acceptCall()V
  
      monitor-exit v1
  
      return-void
  
      :cond_1
      new-instance v0, Lcom/android/internal/telephony/CallStateException;
  
!     const-string v2, "phone not ringing"
  
      invoke-direct {v0, v2}, Lcom/android/internal/telephony/CallStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :catchall_0
      move-exception v0
  
--- 600,634 ----
  
      invoke-static {v0, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 138
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      const/4 v2, 0x0
  
      invoke-virtual {v0, v2}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->setMute(Z)V
  
+     .line 139
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->acceptCall()V
  
+     .line 143
      monitor-exit v1
  
+     .line 144
      return-void
  
+     .line 141
      :cond_1
      new-instance v0, Lcom/android/internal/telephony/CallStateException;
  
!     const-string/jumbo v2, "phone not ringing"
  
      invoke-direct {v0, v2}, Lcom/android/internal/telephony/CallStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 143
      :catchall_0
      move-exception v0
  
***************
*** 585,590 ****
--- 645,651 ----
      .parameter "x1"
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2}, Lcom/android/internal/telephony/sip/SipPhoneBase;->activateCellBroadcastSms(ILandroid/os/Message;)V
  
      return-void
***************
*** 594,599 ****
--- 655,661 ----
      .locals 1
  
      .prologue
+     .line 198
      const/4 v0, 0x1
  
      return v0
***************
*** 603,608 ****
--- 665,671 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->canDial()Z
  
      move-result v0
***************
*** 617,626 ****
--- 680,691 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 95
      const-class v6, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v6
  
+     .line 96
      :try_start_0
      instance-of v7, p1, Landroid/net/sip/SipAudioCall;
  
***************
*** 628,636 ****
--- 693,703 ----
  
      monitor-exit v6
  
+     .line 128
      :goto_0
      return v5
  
+     .line 97
      :cond_0
      iget-object v7, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 648,653 ****
--- 715,721 ----
  
      goto :goto_0
  
+     .line 129
      :catchall_0
      move-exception v5
  
***************
*** 657,662 ****
--- 725,731 ----
  
      throw v5
  
+     .line 101
      :cond_1
      :try_start_1
      iget-object v7, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
***************
*** 683,694 ****
--- 752,765 ----
  
      if-eqz v7, :cond_2
  
+     .line 103
      monitor-exit v6
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
      goto :goto_0
  
+     .line 107
      :cond_2
      :try_start_2
      move-object v0, p1
***************
*** 697,702 ****
--- 768,774 ----
  
      move-object v4, v0
  
+     .line 108
      .local v4, sipAudioCall:Landroid/net/sip/SipAudioCall;
      const-string v7, "SipPhone"
  
***************
*** 728,733 ****
--- 800,806 ----
  
      invoke-static {v7, v8}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 110
      invoke-virtual {v4}, Landroid/net/sip/SipAudioCall;->getLocalProfile()Landroid/net/sip/SipProfile;
  
      move-result-object v7
***************
*** 736,741 ****
--- 809,815 ----
  
      move-result-object v2
  
+     .line 111
      .local v2, localUri:Ljava/lang/String;
      iget-object v7, p0, Lcom/android/internal/telephony/sip/SipPhone;->mProfile:Landroid/net/sip/SipProfile;
  
***************
*** 749,754 ****
--- 823,829 ----
  
      if-eqz v7, :cond_4
  
+     .line 112
      iget-object v7, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v7}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->getState()Lcom/android/internal/telephony/Call$State;
***************
*** 759,769 ****
--- 834,846 ----
  
      move-result v3
  
+     .line 113
      .local v3, makeCallWait:Z
      iget-object v7, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v7, v4, v3}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->initIncomingCall(Landroid/net/sip/SipAudioCall;Z)V
  
+     .line 114
      invoke-virtual {v4}, Landroid/net/sip/SipAudioCall;->getState()I
  
      move-result v7
***************
*** 772,783 ****
--- 849,862 ----
  
      if-eq v7, v8, :cond_3
  
+     .line 117
      const-string v7, "SipPhone"
  
      const-string v8, "    call cancelled !!"
  
      invoke-static {v7, v8}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 118
      iget-object v7, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v7}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->reset()V
***************
*** 785,790 ****
--- 864,870 ----
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
      .catch Ljava/lang/Exception; {:try_start_2 .. :try_end_2} :catch_0
  
+     .line 120
      :cond_3
      const/4 v5, 0x1
  
***************
*** 793,809 ****
--- 873,892 ----
  
      goto/16 :goto_0
  
+     .line 122
      .end local v2           #localUri:Ljava/lang/String;
      .end local v3           #makeCallWait:Z
      .end local v4           #sipAudioCall:Landroid/net/sip/SipAudioCall;
      :catch_0
      move-exception v1
  
+     .line 126
      .local v1, e:Ljava/lang/Exception;
      iget-object v7, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v7}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->reset()V
  
+     .line 128
      .end local v1           #e:Ljava/lang/Exception;
      :cond_4
      monitor-exit v6
***************
*** 817,822 ****
--- 900,906 ----
      .locals 1
  
      .prologue
+     .line 224
      const/4 v0, 0x0
  
      return v0
***************
*** 826,856 ****
--- 910,949 ----
      .locals 2
  
      .prologue
+     .line 232
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 233
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->clearDisconnected()V
  
+     .line 234
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->clearDisconnected()V
  
+     .line 235
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->backgroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->clearDisconnected()V
  
+     .line 237
      invoke-virtual {p0}, Lcom/android/internal/telephony/sip/SipPhone;->updatePhoneState()V
  
+     .line 238
      invoke-virtual {p0}, Lcom/android/internal/telephony/sip/SipPhone;->notifyPreciseCallStateChanged()V
  
+     .line 239
      monitor-exit v1
  
+     .line 240
      return-void
  
+     .line 239
      :catchall_0
      move-exception v0
  
***************
*** 870,879 ****
--- 963,974 ----
      .end annotation
  
      .prologue
+     .line 202
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 203
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 895,900 ****
--- 990,996 ----
  
      if-eq v0, v2, :cond_1
  
+     .line 205
      :cond_0
      new-instance v0, Lcom/android/internal/telephony/CallStateException;
  
***************
*** 902,908 ****
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v3, "wrong state to merge calls: fg="
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 998,1004 ----
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v3, "wrong state to merge calls: fg="
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 942,947 ****
--- 1038,1044 ----
  
      throw v0
  
+     .line 210
      :catchall_0
      move-exception v0
  
***************
*** 951,956 ****
--- 1048,1054 ----
  
      throw v0
  
+     .line 209
      :cond_1
      :try_start_1
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
***************
*** 959,968 ****
--- 1057,1068 ----
  
      invoke-virtual {v0, v2}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->merge(Lcom/android/internal/telephony/sip/SipPhone$SipCall;)V
  
+     .line 210
      monitor-exit v1
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 211
      return-void
  .end method
  
***************
*** 976,990 ****
--- 1076,1093 ----
      .end annotation
  
      .prologue
+     .line 214
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 215
      :try_start_0
      instance-of v0, p1, Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      if-nez v0, :cond_0
  
+     .line 216
      new-instance v0, Lcom/android/internal/telephony/CallStateException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 1025,1030 ****
--- 1128,1134 ----
  
      throw v0
  
+     .line 220
      .end local p1
      :catchall_0
      move-exception v0
***************
*** 1035,1040 ****
--- 1139,1145 ----
  
      throw v0
  
+     .line 219
      .restart local p1
      :cond_0
      :try_start_1
***************
*** 1045,1054 ****
--- 1150,1161 ----
      .end local p1
      invoke-virtual {v0, p1}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->merge(Lcom/android/internal/telephony/sip/SipPhone$SipCall;)V
  
+     .line 220
      monitor-exit v1
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 221
      return-void
  .end method
  
***************
*** 1062,1071 ****
--- 1169,1180 ----
      .end annotation
  
      .prologue
+     .line 158
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 159
      :try_start_0
      invoke-direct {p0, p1}, Lcom/android/internal/telephony/sip/SipPhone;->dialInternal(Ljava/lang/String;)Lcom/android/internal/telephony/Connection;
  
***************
*** 1075,1080 ****
--- 1184,1190 ----
  
      return-object v0
  
+     .line 160
      :catchall_0
      move-exception v0
  
***************
*** 1096,1101 ****
--- 1206,1212 ----
      .end annotation
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2}, Lcom/android/internal/telephony/sip/SipPhoneBase;->dial(Ljava/lang/String;Lcom/android/internal/telephony/UUSInfo;)Lcom/android/internal/telephony/Connection;
  
      move-result-object v0
***************
*** 1107,1112 ****
--- 1218,1224 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->disableDataConnectivity()Z
  
      move-result v0
***************
*** 1118,1123 ****
--- 1230,1236 ----
      .locals 0
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->disableLocationUpdates()V
  
      return-void
***************
*** 1127,1132 ****
--- 1240,1246 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->enableDataConnectivity()Z
  
      move-result v0
***************
*** 1138,1143 ****
--- 1252,1258 ----
      .locals 0
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->enableLocationUpdates()V
  
      return-void
***************
*** 1148,1153 ****
--- 1263,1269 ----
      .parameter "phone"
  
      .prologue
+     .line 91
      invoke-virtual {p0}, Lcom/android/internal/telephony/sip/SipPhone;->getSipUri()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1168,1180 ****
--- 1284,1299 ----
      .parameter "o"
  
      .prologue
+     .line 76
      if-ne p1, p0, :cond_0
  
      const/4 v1, 0x1
  
+     .line 79
      :goto_0
      return v1
  
+     .line 77
      :cond_0
      instance-of v1, p1, Lcom/android/internal/telephony/sip/SipPhone;
  
***************
*** 1187,1194 ****
--- 1306,1315 ----
      :cond_1
      move-object v0, p1
  
+     .line 78
      check-cast v0, Lcom/android/internal/telephony/sip/SipPhone;
  
+     .line 79
      .local v0, that:Lcom/android/internal/telephony/sip/SipPhone;
      iget-object v1, p0, Lcom/android/internal/telephony/sip/SipPhone;->mProfile:Landroid/net/sip/SipProfile;
  
***************
*** 1218,1223 ****
--- 1339,1345 ----
      .end annotation
  
      .prologue
+     .line 229
      return-void
  .end method
  
***************
*** 1226,1231 ****
--- 1348,1354 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getAvailableNetworks(Landroid/os/Message;)V
  
      return-void
***************
*** 1235,1240 ****
--- 1358,1364 ----
      .locals 1
  
      .prologue
+     .line 323
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->backgroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      return-object v0
***************
*** 1244,1249 ****
--- 1368,1374 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getCallForwardingIndicator()Z
  
      move-result v0
***************
*** 1257,1262 ****
--- 1382,1388 ----
      .parameter "x1"
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getCallForwardingOption(ILandroid/os/Message;)V
  
      return-void
***************
*** 1269,1278 ****
--- 1395,1407 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 285
      invoke-static {p1, v0, v0}, Landroid/os/AsyncResult;->forMessage(Landroid/os/Message;Ljava/lang/Object;Ljava/lang/Throwable;)Landroid/os/AsyncResult;
  
+     .line 286
      invoke-virtual {p1}, Landroid/os/Message;->sendToTarget()V
  
+     .line 287
      return-void
  .end method
  
***************
*** 1281,1286 ****
--- 1410,1416 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getCellBroadcastSmsConfig(Landroid/os/Message;)V
  
      return-void
***************
*** 1290,1295 ****
--- 1420,1426 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getCellLocation()Landroid/telephony/CellLocation;
  
      move-result-object v0
***************
*** 1301,1306 ****
--- 1432,1438 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getCurrentDataConnectionList()Ljava/util/List;
  
      move-result-object v0
***************
*** 1312,1317 ****
--- 1444,1450 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getDataActivityState()Lcom/android/internal/telephony/Phone$DataActivityState;
  
      move-result-object v0
***************
*** 1324,1329 ****
--- 1457,1463 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getDataCallList(Landroid/os/Message;)V
  
      return-void
***************
*** 1333,1338 ****
--- 1467,1473 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getDataConnectionState()Lcom/android/internal/telephony/Phone$DataState;
  
      move-result-object v0
***************
*** 1345,1350 ****
--- 1480,1486 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getDataConnectionState(Ljava/lang/String;)Lcom/android/internal/telephony/Phone$DataState;
  
      move-result-object v0
***************
*** 1356,1361 ****
--- 1492,1498 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getDataRoamingEnabled()Z
  
      move-result v0
***************
*** 1367,1372 ****
--- 1504,1510 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getDeviceId()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1378,1383 ****
--- 1516,1522 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getDeviceSvn()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1389,1394 ****
--- 1528,1534 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getEsn()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1400,1405 ****
--- 1540,1546 ----
      .locals 1
  
      .prologue
+     .line 319
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      return-object v0
***************
*** 1409,1414 ****
--- 1550,1556 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getIccCard()Lcom/android/internal/telephony/IccCard;
  
      move-result-object v0
***************
*** 1420,1425 ****
--- 1562,1568 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getIccFileHandler()Lcom/android/internal/telephony/IccFileHandler;
  
      move-result-object v0
***************
*** 1431,1436 ****
--- 1574,1580 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getIccPhoneBookInterfaceManager()Lcom/android/internal/telephony/IccPhoneBookInterfaceManager;
  
      move-result-object v0
***************
*** 1442,1447 ****
--- 1586,1592 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getIccRecordsLoaded()Z
  
      move-result v0
***************
*** 1453,1458 ****
--- 1598,1604 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getIccSerialNumber()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1464,1469 ****
--- 1610,1616 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getIccSmsInterfaceManager()Lcom/android/internal/telephony/IccSmsInterfaceManager;
  
      move-result-object v0
***************
*** 1475,1480 ****
--- 1622,1628 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getImei()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1486,1491 ****
--- 1634,1640 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getLine1AlphaTag()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1497,1502 ****
--- 1646,1652 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getLine1Number()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1509,1514 ****
--- 1659,1665 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getLinkProperties(Ljava/lang/String;)Landroid/net/LinkProperties;
  
      move-result-object v0
***************
*** 1520,1525 ****
--- 1671,1677 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getMeid()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1531,1536 ****
--- 1683,1689 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getMessageWaitingIndicator()Z
  
      move-result v0
***************
*** 1542,1547 ****
--- 1695,1701 ----
      .locals 1
  
      .prologue
+     .line 313
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->getState()Lcom/android/internal/telephony/Call$State;
***************
*** 1578,1583 ****
--- 1732,1738 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getNeighboringCids(Landroid/os/Message;)V
  
      return-void
***************
*** 1590,1599 ****
--- 1745,1757 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 272
      invoke-static {p1, v0, v0}, Landroid/os/AsyncResult;->forMessage(Landroid/os/Message;Ljava/lang/Object;Ljava/lang/Throwable;)Landroid/os/AsyncResult;
  
+     .line 273
      invoke-virtual {p1}, Landroid/os/Message;->sendToTarget()V
  
+     .line 274
      return-void
  .end method
  
***************
*** 1601,1606 ****
--- 1759,1765 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getPendingMmiCodes()Ljava/util/List;
  
      move-result-object v0
***************
*** 1612,1617 ****
--- 1771,1777 ----
      .locals 2
  
      .prologue
+     .line 83
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1643,1648 ****
--- 1803,1809 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getPhoneSubInfo()Lcom/android/internal/telephony/PhoneSubInfo;
  
      move-result-object v0
***************
*** 1654,1659 ****
--- 1815,1821 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getPhoneType()I
  
      move-result v0
***************
*** 1665,1670 ****
--- 1827,1833 ----
      .locals 1
  
      .prologue
+     .line 327
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      return-object v0
***************
*** 1674,1679 ****
--- 1837,1843 ----
      .locals 1
  
      .prologue
+     .line 333
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getServiceState()Landroid/telephony/ServiceState;
  
      move-result-object v0
***************
*** 1685,1690 ****
--- 1849,1855 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getSignalStrength()Landroid/telephony/SignalStrength;
  
      move-result-object v0
***************
*** 1696,1701 ****
--- 1861,1867 ----
      .locals 1
  
      .prologue
+     .line 87
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->mProfile:Landroid/net/sip/SipProfile;
  
      invoke-virtual {v0}, Landroid/net/sip/SipProfile;->getUriString()Ljava/lang/String;
***************
*** 1709,1714 ****
--- 1875,1881 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getState()Lcom/android/internal/telephony/Phone$State;
  
      move-result-object v0
***************
*** 1720,1725 ****
--- 1887,1893 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getSubscriberId()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1731,1736 ****
--- 1899,1905 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getVoiceMailAlphaTag()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1742,1747 ****
--- 1911,1917 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->getVoiceMailNumber()Ljava/lang/String;
  
      move-result-object v0
***************
*** 1759,1764 ****
--- 1929,1935 ----
      .end annotation
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->handleInCallMmiCommands(Ljava/lang/String;)Z
  
      move-result v0
***************
*** 1771,1776 ****
--- 1942,1948 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->handlePinMmi(Ljava/lang/String;)Z
  
      move-result v0
***************
*** 1782,1787 ****
--- 1954,1960 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->isDataConnectivityPossible()Z
  
      move-result v0
***************
*** 1793,1798 ****
--- 1966,1972 ----
      .locals 1
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->needsOtaServiceProvisioning()Z
  
      move-result v0
***************
*** 1804,1809 ****
--- 1978,1984 ----
      .locals 0
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->notifyCallForwardingIndicator()V
  
      return-void
***************
*** 1816,1821 ****
--- 1991,1997 ----
      .parameter "x2"
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2, p3}, Lcom/android/internal/telephony/sip/SipPhoneBase;->registerForRingbackTone(Landroid/os/Handler;ILjava/lang/Object;)V
  
      return-void
***************
*** 1828,1833 ****
--- 2004,2010 ----
      .parameter "x2"
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2, p3}, Lcom/android/internal/telephony/sip/SipPhoneBase;->registerForSuppServiceNotification(Landroid/os/Handler;ILjava/lang/Object;)V
  
      return-void
***************
*** 1842,1851 ****
--- 2019,2030 ----
      .end annotation
  
      .prologue
+     .line 147
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 148
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 1859,1887 ****
  
      if-eqz v0, :cond_0
  
      const-string v0, "SipPhone"
  
!     const-string v2, "rejectCall"
  
      invoke-static {v0, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->rejectCall()V
  
      monitor-exit v1
  
      return-void
  
      :cond_0
      new-instance v0, Lcom/android/internal/telephony/CallStateException;
  
!     const-string v2, "phone not ringing"
  
      invoke-direct {v0, v2}, Lcom/android/internal/telephony/CallStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :catchall_0
      move-exception v0
  
--- 2038,2072 ----
  
      if-eqz v0, :cond_0
  
+     .line 149
      const-string v0, "SipPhone"
  
!     const-string/jumbo v2, "rejectCall"
  
      invoke-static {v0, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 150
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->ringingCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->rejectCall()V
  
+     .line 154
      monitor-exit v1
  
+     .line 155
      return-void
  
+     .line 152
      :cond_0
      new-instance v0, Lcom/android/internal/telephony/CallStateException;
  
!     const-string/jumbo v2, "phone not ringing"
  
      invoke-direct {v0, v2}, Lcom/android/internal/telephony/CallStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 154
      :catchall_0
      move-exception v0
  
***************
*** 1897,1902 ****
--- 2082,2088 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->saveClirSetting(I)V
  
      return-void
***************
*** 1908,1913 ****
--- 2094,2100 ----
      .parameter "x1"
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2}, Lcom/android/internal/telephony/sip/SipPhoneBase;->selectNetworkManually(Lcom/android/internal/telephony/OperatorInfo;Landroid/os/Message;)V
  
      return-void
***************
*** 1918,1929 ****
--- 2105,2118 ----
      .parameter "dtmfString"
  
      .prologue
+     .line 267
      const-string v0, "SipPhone"
  
      const-string v1, "[SipPhone] sendBurstDtmf() is a CDMA method"
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 268
      return-void
  .end method
  
***************
*** 1932,1950 ****
      .parameter "c"
  
      .prologue
      invoke-static {p1}, Landroid/telephony/PhoneNumberUtils;->is12Key(C)Z
  
      move-result v0
  
      if-nez v0, :cond_1
  
      const-string v0, "SipPhone"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "sendDtmf called with invalid character \'"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2121,2141 ----
      .parameter "c"
  
      .prologue
+     .line 243
      invoke-static {p1}, Landroid/telephony/PhoneNumberUtils;->is12Key(C)Z
  
      move-result v0
  
      if-nez v0, :cond_1
  
+     .line 244
      const-string v0, "SipPhone"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "sendDtmf called with invalid character \'"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1966,1975 ****
--- 2157,2168 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 251
      :cond_0
      :goto_0
      return-void
  
+     .line 246
      :cond_1
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 1983,1997 ****
--- 2176,2193 ----
  
      if-eqz v0, :cond_0
  
+     .line 247
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 248
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->sendDtmf(C)V
  
+     .line 249
      monitor-exit v1
  
      goto :goto_0
***************
*** 2011,2016 ****
--- 2207,2213 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->sendUssdResponse(Ljava/lang/String;)V
  
      return-void
***************
*** 2025,2030 ****
--- 2222,2228 ----
      .parameter "x4"
  
      .prologue
+     .line 48
      invoke-super/range {p0 .. p5}, Lcom/android/internal/telephony/sip/SipPhoneBase;->setCallForwardingOption(IILjava/lang/String;ILandroid/os/Message;)V
  
      return-void
***************
*** 2036,2047 ****
--- 2234,2247 ----
      .parameter "onComplete"
  
      .prologue
+     .line 291
      const-string v0, "SipPhone"
  
      const-string v1, "call waiting not supported"
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 292
      return-void
  .end method
  
***************
*** 2051,2056 ****
--- 2251,2257 ----
      .parameter "x1"
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2}, Lcom/android/internal/telephony/sip/SipPhoneBase;->setCellBroadcastSmsConfig([ILandroid/os/Message;)V
  
      return-void
***************
*** 2061,2066 ****
--- 2262,2268 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->setDataRoamingEnabled(Z)V
  
      return-void
***************
*** 2071,2089 ****
--- 2273,2296 ----
      .parameter "enabled"
  
      .prologue
+     .line 301
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 302
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->setAudioGroupMode()V
  
+     .line 303
      monitor-exit v1
  
+     .line 304
      return-void
  
+     .line 303
      :catchall_0
      move-exception v0
  
***************
*** 2101,2106 ****
--- 2308,2314 ----
      .parameter "x2"
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2, p3}, Lcom/android/internal/telephony/sip/SipPhoneBase;->setLine1Number(Ljava/lang/String;Ljava/lang/String;Landroid/os/Message;)V
  
      return-void
***************
*** 2111,2129 ****
--- 2319,2342 ----
      .parameter "muted"
  
      .prologue
+     .line 307
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 308
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->setMute(Z)V
  
+     .line 309
      monitor-exit v1
  
+     .line 310
      return-void
  
+     .line 309
      :catchall_0
      move-exception v0
  
***************
*** 2139,2144 ****
--- 2352,2358 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->setNetworkSelectionModeAutomatic(Landroid/os/Message;)V
  
      return-void
***************
*** 2151,2156 ****
--- 2365,2371 ----
      .parameter "x2"
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2, p3}, Lcom/android/internal/telephony/sip/SipPhoneBase;->setOnPostDialCharacter(Landroid/os/Handler;ILjava/lang/Object;)V
  
      return-void
***************
*** 2164,2173 ****
--- 2379,2391 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 279
      invoke-static {p2, v0, v0}, Landroid/os/AsyncResult;->forMessage(Landroid/os/Message;Ljava/lang/Object;Ljava/lang/Throwable;)Landroid/os/AsyncResult;
  
+     .line 280
      invoke-virtual {p2}, Landroid/os/Message;->sendToTarget()V
  
+     .line 281
      return-void
  .end method
  
***************
*** 2176,2181 ****
--- 2394,2400 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->setRadioPower(Z)V
  
      return-void
***************
*** 2188,2193 ****
--- 2407,2413 ----
      .parameter "x2"
  
      .prologue
+     .line 48
      invoke-super {p0, p1, p2, p3}, Lcom/android/internal/telephony/sip/SipPhoneBase;->setVoiceMailNumber(Ljava/lang/String;Ljava/lang/String;Landroid/os/Message;)V
  
      return-void
***************
*** 2198,2216 ****
      .parameter "c"
  
      .prologue
      invoke-static {p1}, Landroid/telephony/PhoneNumberUtils;->is12Key(C)Z
  
      move-result v0
  
      if-nez v0, :cond_0
  
      const-string v0, "SipPhone"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "startDtmf called with invalid character \'"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 2418,2438 ----
      .parameter "c"
  
      .prologue
+     .line 254
      invoke-static {p1}, Landroid/telephony/PhoneNumberUtils;->is12Key(C)Z
  
      move-result v0
  
      if-nez v0, :cond_0
  
+     .line 255
      const-string v0, "SipPhone"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "startDtmf called with invalid character \'"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 2232,2240 ****
--- 2454,2464 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 260
      :goto_0
      return-void
  
+     .line 258
      :cond_0
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/sip/SipPhone;->sendDtmf(C)V
  
***************
*** 2245,2250 ****
--- 2469,2475 ----
      .locals 0
  
      .prologue
+     .line 264
      return-void
  .end method
  
***************
*** 2257,2272 ****
--- 2482,2500 ----
      .end annotation
  
      .prologue
+     .line 189
      const-string v0, "SipPhone"
  
      const-string v1, " ~~~~~~  switch fg and bg"
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 190
      const-class v1, Lcom/android/internal/telephony/sip/SipPhone;
  
      monitor-enter v1
  
+     .line 191
      :try_start_0
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 2274,2279 ****
--- 2502,2508 ----
  
      invoke-virtual {v0, v2}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->switchWith(Lcom/android/internal/telephony/sip/SipPhone$SipCall;)V
  
+     .line 192
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->backgroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->getState()Lcom/android/internal/telephony/Call$State;
***************
*** 2290,2295 ****
--- 2519,2525 ----
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->hold()V
  
+     .line 193
      :cond_0
      iget-object v0, p0, Lcom/android/internal/telephony/sip/SipPhone;->foregroundCall:Lcom/android/internal/telephony/sip/SipPhone$SipCall;
  
***************
*** 2307,2317 ****
--- 2537,2550 ----
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/sip/SipPhone$SipCall;->unhold()V
  
+     .line 194
      :cond_1
      monitor-exit v1
  
+     .line 195
      return-void
  
+     .line 194
      :catchall_0
      move-exception v0
  
***************
*** 2327,2332 ****
--- 2560,2566 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->unregisterForRingbackTone(Landroid/os/Handler;)V
  
      return-void
***************
*** 2337,2342 ****
--- 2571,2577 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-super {p0, p1}, Lcom/android/internal/telephony/sip/SipPhoneBase;->unregisterForSuppServiceNotification(Landroid/os/Handler;)V
  
      return-void
***************
*** 2346,2351 ****
--- 2581,2587 ----
      .locals 0
  
      .prologue
+     .line 48
      invoke-super {p0}, Lcom/android/internal/telephony/sip/SipPhoneBase;->updateServiceLocation()V
  
      return-void
