*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/srec/UlawEncoderInputStream.smali	2012-03-23 19:36:31.847261718 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/srec/UlawEncoderInputStream.smali.original	2012-03-23 19:34:45.172316503 -0500
***************
*** 32,59 ****
--- 32,67 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 135
      invoke-direct {p0}, Ljava/io/InputStream;-><init>()V
  
+     .line 41
      iput v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mMax:I
  
+     .line 43
      const/16 v0, 0x400
  
      new-array v0, v0, [B
  
      iput-object v0, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBuf:[B
  
+     .line 44
      iput v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBufCount:I
  
+     .line 46
      const/4 v0, 0x1
  
      new-array v0, v0, [B
  
      iput-object v0, p0, Landroid/speech/srec/UlawEncoderInputStream;->mOneByte:[B
  
+     .line 136
      iput-object p1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mIn:Ljava/io/InputStream;
  
+     .line 137
      iput p2, p0, Landroid/speech/srec/UlawEncoderInputStream;->mMax:I
  
+     .line 138
      return-void
  .end method
  
***************
*** 67,81 ****
--- 75,92 ----
      .parameter "max"
  
      .prologue
+     .line 76
      if-gtz p5, :cond_0
  
      const/16 p5, 0x2000
  
+     .line 78
      :cond_0
      const/high16 v6, 0x2000
  
      div-int v0, v6, p5
  
+     .line 80
      .local v0, coef:I
      const/4 v1, 0x0
  
***************
*** 91,96 ****
--- 102,108 ----
      :goto_0
      if-ge v1, p4, :cond_14
  
+     .line 81
      add-int/lit8 p1, v3, 0x1
  
      .end local v3           #pcmOffset:I
***************
*** 109,125 ****
--- 121,141 ----
  
      add-int v2, v6, v7
  
+     .line 82
      .local v2, pcm:I
      mul-int v6, v2, v0
  
      shr-int/lit8 v2, v6, 0x10
  
+     .line 85
      if-ltz v2, :cond_a
  
+     .line 86
      if-gtz v2, :cond_1
  
      const/16 v4, 0xff
  
+     .line 108
      .local v4, ulaw:I
      :goto_1
      add-int/lit8 p3, v5, 0x1
***************
*** 130,135 ****
--- 146,152 ----
  
      aput-byte v6, p2, v5
  
+     .line 80
      add-int/lit8 v1, v1, 0x1
  
      move v5, p3
***************
*** 138,143 ****
--- 155,161 ----
      .restart local v5       #ulawOffset:I
      goto :goto_0
  
+     .line 86
      .end local v4           #ulaw:I
      :cond_1
      const/16 v6, 0x1e
***************
*** 248,253 ****
--- 266,272 ----
  
      goto :goto_1
  
+     .line 97
      :cond_a
      const/4 v6, -0x1
  
***************
*** 369,374 ****
--- 388,394 ----
  
      goto :goto_2
  
+     .line 110
      .end local v2           #pcm:I
      :cond_14
      return-void
***************
*** 381,388 ****
--- 401,410 ----
      .parameter "length"
  
      .prologue
+     .line 121
      const/4 v1, 0x0
  
+     .line 122
      .local v1, max:I
      const/4 v0, 0x0
  
***************
*** 394,399 ****
--- 416,422 ----
      :goto_0
      if-ge v0, p2, :cond_2
  
+     .line 123
      add-int/lit8 p1, v2, 0x1
  
      .end local v2           #offset:I
***************
*** 412,432 ****
--- 435,459 ----
  
      add-int v3, v4, v5
  
+     .line 124
      .local v3, pcm:I
      if-gez v3, :cond_0
  
      neg-int v3, v3
  
+     .line 125
      :cond_0
      if-le v3, v1, :cond_1
  
      move v1, v3
  
+     .line 122
      :cond_1
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 127
      .end local v3           #pcm:I
      :cond_2
      return v1
***************
*** 443,448 ****
--- 470,476 ----
      .end annotation
  
      .prologue
+     .line 185
      iget-object v0, p0, Landroid/speech/srec/UlawEncoderInputStream;->mIn:Ljava/io/InputStream;
  
      invoke-virtual {v0}, Ljava/io/InputStream;->available()I
***************
*** 467,485 ****
--- 495,518 ----
      .end annotation
  
      .prologue
+     .line 176
      iget-object v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mIn:Ljava/io/InputStream;
  
      if-eqz v1, :cond_0
  
+     .line 177
      iget-object v0, p0, Landroid/speech/srec/UlawEncoderInputStream;->mIn:Ljava/io/InputStream;
  
+     .line 178
      .local v0, in:Ljava/io/InputStream;
      const/4 v1, 0x0
  
      iput-object v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mIn:Ljava/io/InputStream;
  
+     .line 179
      invoke-virtual {v0}, Ljava/io/InputStream;->close()V
  
+     .line 181
      .end local v0           #in:Ljava/io/InputStream;
      :cond_0
      return-void
***************
*** 498,503 ****
--- 531,537 ----
  
      const/4 v1, -0x1
  
+     .line 169
      iget-object v2, p0, Landroid/speech/srec/UlawEncoderInputStream;->mOneByte:[B
  
      const/4 v3, 0x1
***************
*** 506,514 ****
--- 540,550 ----
  
      move-result v0
  
+     .line 170
      .local v0, n:I
      if-ne v0, v1, :cond_0
  
+     .line 171
      :goto_0
      return v1
  
***************
*** 532,537 ****
--- 568,574 ----
      .end annotation
  
      .prologue
+     .line 164
      const/4 v0, 0x0
  
      array-length v1, p1
***************
*** 557,574 ****
      .prologue
      const/4 v0, -0x1
  
      iget-object v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mIn:Ljava/io/InputStream;
  
      if-nez v1, :cond_1
  
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string v1, "not open"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      .local v4, n:I
      :cond_0
      iget v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBufCount:I
--- 594,613 ----
      .prologue
      const/4 v0, -0x1
  
+     .line 142
      iget-object v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mIn:Ljava/io/InputStream;
  
      if-nez v1, :cond_1
  
      new-instance v0, Ljava/lang/IllegalStateException;
  
!     const-string/jumbo v1, "not open"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 148
      .local v4, n:I
      :cond_0
      iget v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBufCount:I
***************
*** 577,582 ****
--- 616,622 ----
  
      iput v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBufCount:I
  
+     .line 145
      .end local v4           #n:I
      :cond_1
      iget v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBufCount:I
***************
*** 585,590 ****
--- 625,631 ----
  
      if-ge v1, v2, :cond_2
  
+     .line 146
      iget-object v1, p0, Landroid/speech/srec/UlawEncoderInputStream;->mIn:Ljava/io/InputStream;
  
      iget-object v2, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBuf:[B
***************
*** 609,620 ****
--- 650,664 ----
  
      move-result v4
  
+     .line 147
      .restart local v4       #n:I
      if-ne v4, v0, :cond_0
  
+     .line 159
      :goto_0
      return v0
  
+     .line 152
      .end local v4           #n:I
      :cond_2
      iget v0, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBufCount:I
***************
*** 625,630 ****
--- 669,675 ----
  
      move-result v4
  
+     .line 153
      .restart local v4       #n:I
      iget-object v0, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBuf:[B
  
***************
*** 638,643 ****
--- 683,689 ----
  
      invoke-static/range {v0 .. v5}, Landroid/speech/srec/UlawEncoderInputStream;->encode([BI[BIII)V
  
+     .line 156
      iget v0, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBufCount:I
  
      mul-int/lit8 v1, v4, 0x2
***************
*** 646,651 ****
--- 692,698 ----
  
      iput v0, p0, Landroid/speech/srec/UlawEncoderInputStream;->mBufCount:I
  
+     .line 157
      const/4 v6, 0x0
  
      .local v6, i:I
***************
*** 673,677 ****
--- 720,725 ----
      :cond_3
      move v0, v4
  
+     .line 159
      goto :goto_0
  .end method
