*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/AbsActionBarView$VisibilityAnimListener.smali	2012-03-23 19:37:32.882374693 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/AbsActionBarView$VisibilityAnimListener.smali.original	2012-03-23 19:34:46.087333189 -0500
***************
*** 31,40 ****
--- 31,42 ----
      .parameter
  
      .prologue
+     .line 240
      iput-object p1, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->this$0:Lcom/android/internal/widget/AbsActionBarView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 241
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->mCanceled:Z
***************
*** 49,58 ****
--- 51,62 ----
      .parameter "animation"
  
      .prologue
+     .line 269
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->mCanceled:Z
  
+     .line 270
      return-void
  .end method
  
***************
*** 61,74 ****
--- 65,81 ----
      .parameter "animation"
  
      .prologue
+     .line 258
      iget-boolean v0, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->mCanceled:Z
  
      if-eqz v0, :cond_1
  
+     .line 265
      :cond_0
      :goto_0
      return-void
  
+     .line 260
      :cond_1
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->this$0:Lcom/android/internal/widget/AbsActionBarView;
  
***************
*** 76,87 ****
--- 83,96 ----
  
      iput-object v1, v0, Lcom/android/internal/widget/AbsActionBarView;->mVisibilityAnim:Landroid/animation/Animator;
  
+     .line 261
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->this$0:Lcom/android/internal/widget/AbsActionBarView;
  
      iget v1, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->mFinalVisibility:I
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/AbsActionBarView;->setVisibility(I)V
  
+     .line 262
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->this$0:Lcom/android/internal/widget/AbsActionBarView;
  
      iget-object v0, v0, Lcom/android/internal/widget/AbsActionBarView;->mSplitView:Lcom/android/internal/widget/ActionBarContainer;
***************
*** 94,99 ****
--- 103,109 ----
  
      if-eqz v0, :cond_0
  
+     .line 263
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->this$0:Lcom/android/internal/widget/AbsActionBarView;
  
      iget-object v0, v0, Lcom/android/internal/widget/AbsActionBarView;->mMenuView:Lcom/android/internal/view/menu/ActionMenuView;
***************
*** 110,115 ****
--- 120,126 ----
      .parameter "animation"
  
      .prologue
+     .line 274
      return-void
  .end method
  
***************
*** 120,135 ****
--- 131,150 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 251
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->this$0:Lcom/android/internal/widget/AbsActionBarView;
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/AbsActionBarView;->setVisibility(I)V
  
+     .line 252
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->this$0:Lcom/android/internal/widget/AbsActionBarView;
  
      iput-object p1, v0, Lcom/android/internal/widget/AbsActionBarView;->mVisibilityAnim:Landroid/animation/Animator;
  
+     .line 253
      iput-boolean v1, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->mCanceled:Z
  
+     .line 254
      return-void
  .end method
  
***************
*** 138,144 ****
--- 153,161 ----
      .parameter "visibility"
  
      .prologue
+     .line 245
      iput p1, p0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->mFinalVisibility:I
  
+     .line 246
      return-object p0
  .end method
