*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/AbsActionBarView.smali	2012-03-23 19:37:33.128379179 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/AbsActionBarView.smali.original	2012-03-23 19:34:46.091333261 -0500
***************
*** 40,45 ****
--- 40,46 ----
      .locals 1
  
      .prologue
+     .line 45
      new-instance v0, Landroid/view/animation/DecelerateInterpolator;
  
      invoke-direct {v0}, Landroid/view/animation/DecelerateInterpolator;-><init>()V
***************
*** 54,67 ****
--- 55,71 ----
      .parameter "context"
  
      .prologue
+     .line 50
      invoke-direct {p0, p1}, Landroid/view/ViewGroup;-><init>(Landroid/content/Context;)V
  
+     .line 43
      new-instance v0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
      invoke-direct {v0, p0}, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;-><init>(Lcom/android/internal/widget/AbsActionBarView;)V
  
      iput-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisAnimListener:Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
+     .line 51
      return-void
  .end method
  
***************
*** 71,84 ****
--- 75,91 ----
      .parameter "attrs"
  
      .prologue
+     .line 54
      invoke-direct {p0, p1, p2}, Landroid/view/ViewGroup;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 43
      new-instance v0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
      invoke-direct {v0, p0}, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;-><init>(Lcom/android/internal/widget/AbsActionBarView;)V
  
      iput-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisAnimListener:Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
+     .line 55
      return-void
  .end method
  
***************
*** 89,102 ****
--- 96,112 ----
      .parameter "defStyle"
  
      .prologue
+     .line 58
      invoke-direct {p0, p1, p2, p3}, Landroid/view/ViewGroup;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
+     .line 43
      new-instance v0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
      invoke-direct {v0, p0}, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;-><init>(Lcom/android/internal/widget/AbsActionBarView;)V
  
      iput-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisAnimListener:Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
+     .line 59
      return-void
  .end method
  
***************
*** 117,141 ****
--- 127,157 ----
  
      const/4 v6, 0x0
  
+     .line 120
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisibilityAnim:Landroid/animation/Animator;
  
      if-eqz v3, :cond_0
  
+     .line 121
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisibilityAnim:Landroid/animation/Animator;
  
      invoke-virtual {v3}, Landroid/animation/Animator;->cancel()V
  
+     .line 123
      :cond_0
      if-nez p1, :cond_3
  
+     .line 124
      invoke-virtual {p0}, Lcom/android/internal/widget/AbsActionBarView;->getVisibility()I
  
      move-result v3
  
      if-eqz v3, :cond_1
  
+     .line 125
      invoke-virtual {p0, v6}, Lcom/android/internal/widget/AbsActionBarView;->setAlpha(F)V
  
+     .line 126
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mSplitView:Lcom/android/internal/widget/ActionBarContainer;
  
      if-eqz v3, :cond_1
***************
*** 144,153 ****
--- 160,171 ----
  
      if-eqz v3, :cond_1
  
+     .line 127
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mMenuView:Lcom/android/internal/view/menu/ActionMenuView;
  
      invoke-virtual {v3, v6}, Lcom/android/internal/view/menu/ActionMenuView;->setAlpha(F)V
  
+     .line 130
      :cond_1
      const-string v3, "alpha"
  
***************
*** 159,171 ****
--- 177,192 ----
  
      move-result-object v0
  
+     .line 131
      .local v0, anim:Landroid/animation/ObjectAnimator;
      invoke-virtual {v0, v8, v9}, Landroid/animation/ObjectAnimator;->setDuration(J)Landroid/animation/ObjectAnimator;
  
+     .line 132
      sget-object v3, Lcom/android/internal/widget/AbsActionBarView;->sAlphaInterpolator:Landroid/animation/TimeInterpolator;
  
      invoke-virtual {v0, v3}, Landroid/animation/ObjectAnimator;->setInterpolator(Landroid/animation/TimeInterpolator;)V
  
+     .line 133
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mSplitView:Lcom/android/internal/widget/ActionBarContainer;
  
      if-eqz v3, :cond_2
***************
*** 174,183 ****
--- 195,206 ----
  
      if-eqz v3, :cond_2
  
+     .line 134
      new-instance v1, Landroid/animation/AnimatorSet;
  
      invoke-direct {v1}, Landroid/animation/AnimatorSet;-><init>()V
  
+     .line 135
      .local v1, set:Landroid/animation/AnimatorSet;
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mMenuView:Lcom/android/internal/view/menu/ActionMenuView;
  
***************
*** 191,199 ****
--- 214,224 ----
  
      move-result-object v2
  
+     .line 136
      .local v2, splitAnim:Landroid/animation/ObjectAnimator;
      invoke-virtual {v2, v8, v9}, Landroid/animation/ObjectAnimator;->setDuration(J)Landroid/animation/ObjectAnimator;
  
+     .line 137
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisAnimListener:Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
      invoke-virtual {v3, p1}, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->withFinalVisibility(I)Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
***************
*** 202,220 ****
--- 227,249 ----
  
      invoke-virtual {v1, v3}, Landroid/animation/AnimatorSet;->addListener(Landroid/animation/Animator$AnimatorListener;)V
  
+     .line 138
      invoke-virtual {v1, v0}, Landroid/animation/AnimatorSet;->play(Landroid/animation/Animator;)Landroid/animation/AnimatorSet$Builder;
  
      move-result-object v3
  
      invoke-virtual {v3, v2}, Landroid/animation/AnimatorSet$Builder;->with(Landroid/animation/Animator;)Landroid/animation/AnimatorSet$Builder;
  
+     .line 139
      invoke-virtual {v1}, Landroid/animation/AnimatorSet;->start()V
  
+     .line 160
      .end local v1           #set:Landroid/animation/AnimatorSet;
      .end local v2           #splitAnim:Landroid/animation/ObjectAnimator;
      :goto_0
      return-void
  
+     .line 141
      :cond_2
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisAnimListener:Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
***************
*** 224,233 ****
--- 253,264 ----
  
      invoke-virtual {v0, v3}, Landroid/animation/ObjectAnimator;->addListener(Landroid/animation/Animator$AnimatorListener;)V
  
+     .line 142
      invoke-virtual {v0}, Landroid/animation/ObjectAnimator;->start()V
  
      goto :goto_0
  
+     .line 145
      .end local v0           #anim:Landroid/animation/ObjectAnimator;
      :cond_3
      const-string v3, "alpha"
***************
*** 240,252 ****
--- 271,286 ----
  
      move-result-object v0
  
+     .line 146
      .restart local v0       #anim:Landroid/animation/ObjectAnimator;
      invoke-virtual {v0, v8, v9}, Landroid/animation/ObjectAnimator;->setDuration(J)Landroid/animation/ObjectAnimator;
  
+     .line 147
      sget-object v3, Lcom/android/internal/widget/AbsActionBarView;->sAlphaInterpolator:Landroid/animation/TimeInterpolator;
  
      invoke-virtual {v0, v3}, Landroid/animation/ObjectAnimator;->setInterpolator(Landroid/animation/TimeInterpolator;)V
  
+     .line 148
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mSplitView:Lcom/android/internal/widget/ActionBarContainer;
  
      if-eqz v3, :cond_4
***************
*** 255,264 ****
--- 289,300 ----
  
      if-eqz v3, :cond_4
  
+     .line 149
      new-instance v1, Landroid/animation/AnimatorSet;
  
      invoke-direct {v1}, Landroid/animation/AnimatorSet;-><init>()V
  
+     .line 150
      .restart local v1       #set:Landroid/animation/AnimatorSet;
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mMenuView:Lcom/android/internal/view/menu/ActionMenuView;
  
***************
*** 272,280 ****
--- 308,318 ----
  
      move-result-object v2
  
+     .line 151
      .restart local v2       #splitAnim:Landroid/animation/ObjectAnimator;
      invoke-virtual {v2, v8, v9}, Landroid/animation/ObjectAnimator;->setDuration(J)Landroid/animation/ObjectAnimator;
  
+     .line 152
      iget-object v3, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisAnimListener:Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
      invoke-virtual {v3, p1}, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->withFinalVisibility(I)Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
***************
*** 283,298 ****
--- 321,339 ----
  
      invoke-virtual {v1, v3}, Landroid/animation/AnimatorSet;->addListener(Landroid/animation/Animator$AnimatorListener;)V
  
+     .line 153
      invoke-virtual {v1, v0}, Landroid/animation/AnimatorSet;->play(Landroid/animation/Animator;)Landroid/animation/AnimatorSet$Builder;
  
      move-result-object v3
  
      invoke-virtual {v3, v2}, Landroid/animation/AnimatorSet$Builder;->with(Landroid/animation/Animator;)Landroid/animation/AnimatorSet$Builder;
  
+     .line 154
      invoke-virtual {v1}, Landroid/animation/AnimatorSet;->start()V
  
      goto :goto_0
  
+     .line 156
      .end local v1           #set:Landroid/animation/AnimatorSet;
      .end local v2           #splitAnim:Landroid/animation/ObjectAnimator;
      :cond_4
***************
*** 304,309 ****
--- 345,351 ----
  
      invoke-virtual {v0, v3}, Landroid/animation/ObjectAnimator;->addListener(Landroid/animation/Animator$AnimatorListener;)V
  
+     .line 157
      invoke-virtual {v0}, Landroid/animation/ObjectAnimator;->start()V
  
      goto :goto_0
***************
*** 313,326 ****
--- 355,371 ----
      .locals 1
  
      .prologue
+     .line 204
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      if-eqz v0, :cond_0
  
+     .line 205
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      invoke-virtual {v0}, Lcom/android/internal/view/menu/ActionMenuPresenter;->dismissPopupMenus()Z
  
+     .line 207
      :cond_0
      return-void
  .end method
***************
*** 329,342 ****
--- 374,390 ----
      .locals 1
  
      .prologue
+     .line 113
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisibilityAnim:Landroid/animation/Animator;
  
      if-eqz v0, :cond_0
  
+     .line 114
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisAnimListener:Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;
  
      iget v0, v0, Lcom/android/internal/widget/AbsActionBarView$VisibilityAnimListener;->mFinalVisibility:I
  
+     .line 116
      :goto_0
      return v0
  
***************
*** 352,357 ****
--- 400,406 ----
      .locals 1
  
      .prologue
+     .line 102
      iget v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mContentHeight:I
  
      return v0
***************
*** 361,376 ****
--- 410,428 ----
      .locals 1
  
      .prologue
+     .line 186
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      if-eqz v0, :cond_0
  
+     .line 187
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      invoke-virtual {v0}, Lcom/android/internal/view/menu/ActionMenuPresenter;->hideOverflowMenu()Z
  
      move-result v0
  
+     .line 189
      :goto_0
      return v0
  
***************
*** 384,399 ****
--- 436,454 ----
      .locals 1
  
      .prologue
+     .line 193
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      if-eqz v0, :cond_0
  
+     .line 194
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      invoke-virtual {v0}, Lcom/android/internal/view/menu/ActionMenuPresenter;->isOverflowMenuShowing()Z
  
      move-result v0
  
+     .line 196
      :goto_0
      return v0
  
***************
*** 407,412 ****
--- 462,468 ----
      .locals 1
  
      .prologue
+     .line 200
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      if-eqz v0, :cond_0
***************
*** 438,443 ****
--- 494,500 ----
      .parameter "spacing"
  
      .prologue
+     .line 211
      const/high16 v0, -0x8000
  
      invoke-static {p2, v0}, Landroid/view/View$MeasureSpec;->makeMeasureSpec(II)I
***************
*** 446,459 ****
--- 503,519 ----
  
      invoke-virtual {p1, v0, p3}, Landroid/view/View;->measure(II)V
  
+     .line 214
      invoke-virtual {p1}, Landroid/view/View;->getMeasuredWidth()I
  
      move-result v0
  
      sub-int/2addr p2, v0
  
+     .line 215
      sub-int/2addr p2, p4
  
+     .line 217
      const/4 v0, 0x0
  
      invoke-static {v0, p2}, Ljava/lang/Math;->max(II)I
***************
*** 470,477 ****
--- 530,539 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 63
      invoke-super {p0, p1}, Landroid/view/ViewGroup;->onConfigurationChanged(Landroid/content/res/Configuration;)V
  
+     .line 67
      invoke-virtual {p0}, Lcom/android/internal/widget/AbsActionBarView;->getContext()Landroid/content/Context;
  
      move-result-object v1
***************
*** 486,491 ****
--- 548,554 ----
  
      move-result-object v0
  
+     .line 69
      .local v0, a:Landroid/content/res/TypedArray;
      const/4 v1, 0x4
  
***************
*** 495,506 ****
--- 558,572 ----
  
      invoke-virtual {p0, v1}, Lcom/android/internal/widget/AbsActionBarView;->setContentHeight(I)V
  
+     .line 70
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 71
      iget-boolean v1, p0, Lcom/android/internal/widget/AbsActionBarView;->mSplitWhenNarrow:Z
  
      if-eqz v1, :cond_0
  
+     .line 72
      invoke-virtual {p0}, Lcom/android/internal/widget/AbsActionBarView;->getContext()Landroid/content/Context;
  
      move-result-object v1
***************
*** 517,531 ****
--- 583,600 ----
  
      invoke-virtual {p0, v1}, Lcom/android/internal/widget/AbsActionBarView;->setSplitActionBar(Z)V
  
+     .line 75
      :cond_0
      iget-object v1, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      if-eqz v1, :cond_1
  
+     .line 76
      iget-object v1, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      invoke-virtual {v1, p1}, Lcom/android/internal/view/menu/ActionMenuPresenter;->onConfigurationChanged(Landroid/content/res/Configuration;)V
  
+     .line 78
      :cond_1
      return-void
  .end method
***************
*** 538,552 ****
--- 607,624 ----
      .parameter "contentHeight"
  
      .prologue
+     .line 221
      invoke-virtual {p1}, Landroid/view/View;->getMeasuredWidth()I
  
      move-result v2
  
+     .line 222
      .local v2, childWidth:I
      invoke-virtual {p1}, Landroid/view/View;->getMeasuredHeight()I
  
      move-result v0
  
+     .line 223
      .local v0, childHeight:I
      sub-int v3, p4, v0
  
***************
*** 554,559 ****
--- 626,632 ----
  
      add-int v1, p3, v3
  
+     .line 225
      .local v1, childTop:I
      add-int v3, p2, v2
  
***************
*** 561,566 ****
--- 634,640 ----
  
      invoke-virtual {p1, p2, v1, v3, v4}, Landroid/view/View;->layout(IIII)V
  
+     .line 227
      return v2
  .end method
  
***************
*** 572,586 ****
--- 646,663 ----
      .parameter "contentHeight"
  
      .prologue
+     .line 231
      invoke-virtual {p1}, Landroid/view/View;->getMeasuredWidth()I
  
      move-result v2
  
+     .line 232
      .local v2, childWidth:I
      invoke-virtual {p1}, Landroid/view/View;->getMeasuredHeight()I
  
      move-result v0
  
+     .line 233
      .local v0, childHeight:I
      sub-int v3, p4, v0
  
***************
*** 588,593 ****
--- 665,671 ----
  
      add-int v1, p3, v3
  
+     .line 235
      .local v1, childTop:I
      sub-int v3, p2, v2
  
***************
*** 595,600 ****
--- 673,679 ----
  
      invoke-virtual {p1, v3, v1, p2, v4}, Landroid/view/View;->layout(IIII)V
  
+     .line 237
      return v2
  .end method
  
***************
*** 602,613 ****
--- 681,694 ----
      .locals 1
  
      .prologue
+     .line 178
      new-instance v0, Lcom/android/internal/widget/AbsActionBarView$1;
  
      invoke-direct {v0, p0}, Lcom/android/internal/widget/AbsActionBarView$1;-><init>(Lcom/android/internal/widget/AbsActionBarView;)V
  
      invoke-virtual {p0, v0}, Lcom/android/internal/widget/AbsActionBarView;->post(Ljava/lang/Runnable;)Z
  
+     .line 183
      return-void
  .end method
  
***************
*** 616,625 ****
--- 697,709 ----
      .parameter "height"
  
      .prologue
+     .line 97
      iput p1, p0, Lcom/android/internal/widget/AbsActionBarView;->mContentHeight:I
  
+     .line 98
      invoke-virtual {p0}, Lcom/android/internal/widget/AbsActionBarView;->requestLayout()V
  
+     .line 99
      return-void
  .end method
  
***************
*** 628,635 ****
--- 712,721 ----
      .parameter "split"
  
      .prologue
+     .line 85
      iput-boolean p1, p0, Lcom/android/internal/widget/AbsActionBarView;->mSplitActionBar:Z
  
+     .line 86
      return-void
  .end method
  
***************
*** 638,645 ****
--- 724,733 ----
      .parameter "splitView"
  
      .prologue
+     .line 106
      iput-object p1, p0, Lcom/android/internal/widget/AbsActionBarView;->mSplitView:Lcom/android/internal/widget/ActionBarContainer;
  
+     .line 107
      return-void
  .end method
  
***************
*** 648,655 ****
--- 736,745 ----
      .parameter "splitWhenNarrow"
  
      .prologue
+     .line 93
      iput-boolean p1, p0, Lcom/android/internal/widget/AbsActionBarView;->mSplitWhenNarrow:Z
  
+     .line 94
      return-void
  .end method
  
***************
*** 658,674 ****
--- 748,768 ----
      .parameter "visibility"
  
      .prologue
+     .line 164
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisibilityAnim:Landroid/animation/Animator;
  
      if-eqz v0, :cond_0
  
+     .line 165
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mVisibilityAnim:Landroid/animation/Animator;
  
      invoke-virtual {v0}, Landroid/animation/Animator;->end()V
  
+     .line 167
      :cond_0
      invoke-super {p0, p1}, Landroid/view/ViewGroup;->setVisibility(I)V
  
+     .line 168
      return-void
  .end method
  
***************
*** 676,691 ****
--- 770,788 ----
      .locals 1
  
      .prologue
+     .line 171
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      if-eqz v0, :cond_0
  
+     .line 172
      iget-object v0, p0, Lcom/android/internal/widget/AbsActionBarView;->mActionMenuPresenter:Lcom/android/internal/view/menu/ActionMenuPresenter;
  
      invoke-virtual {v0}, Lcom/android/internal/view/menu/ActionMenuPresenter;->showOverflowMenu()Z
  
      move-result v0
  
+     .line 174
      :goto_0
      return v0
  
