*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter.smali	2012-03-23 19:37:33.498385923 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter.smali.original	2012-03-23 19:34:46.098333389 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .parameter
  
      .prologue
+     .line 1310
      iput-object p1, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "x1"
  
      .prologue
+     .line 1310
      invoke-direct {p0, p1}, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;-><init>(Lcom/android/internal/widget/ActionBarView;)V
  
      return-void
***************
*** 63,68 ****
--- 65,71 ----
  
      const/4 v2, 0x0
  
+     .line 1400
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mExpandedActionView:Landroid/view/View;
***************
*** 71,76 ****
--- 74,80 ----
  
      if-eqz v0, :cond_0
  
+     .line 1401
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mExpandedActionView:Landroid/view/View;
***************
*** 79,84 ****
--- 83,89 ----
  
      invoke-interface {v0}, Landroid/view/CollapsibleActionView;->onActionViewCollapsed()V
  
+     .line 1404
      :cond_0
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 88,93 ****
--- 93,99 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/ActionBarView;->removeView(Landroid/view/View;)V
  
+     .line 1405
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      iget-object v1, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
***************
*** 99,108 ****
--- 105,116 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/ActionBarView;->removeView(Landroid/view/View;)V
  
+     .line 1406
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      iput-object v3, v0, Lcom/android/internal/widget/ActionBarView;->mExpandedActionView:Landroid/view/View;
  
+     .line 1407
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      #getter for: Lcom/android/internal/widget/ActionBarView;->mDisplayOptions:I
***************
*** 114,119 ****
--- 122,128 ----
  
      if-eqz v0, :cond_1
  
+     .line 1408
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      #getter for: Lcom/android/internal/widget/ActionBarView;->mHomeLayout:Lcom/android/internal/widget/ActionBarView$HomeView;
***************
*** 123,128 ****
--- 132,138 ----
  
      invoke-virtual {v0, v2}, Lcom/android/internal/widget/ActionBarView$HomeView;->setVisibility(I)V
  
+     .line 1410
      :cond_1
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 135,140 ****
--- 145,151 ----
  
      if-eqz v0, :cond_2
  
+     .line 1411
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      #getter for: Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
***************
*** 144,154 ****
--- 155,167 ----
  
      if-nez v0, :cond_6
  
+     .line 1412
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      #calls: Lcom/android/internal/widget/ActionBarView;->initTitle()V
      invoke-static {v0}, Lcom/android/internal/widget/ActionBarView;->access$1300(Lcom/android/internal/widget/ActionBarView;)V
  
+     .line 1417
      :cond_2
      :goto_0
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
***************
*** 171,176 ****
--- 184,190 ----
  
      if-ne v0, v1, :cond_3
  
+     .line 1418
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      #getter for: Lcom/android/internal/widget/ActionBarView;->mTabScrollView:Lcom/android/internal/widget/ScrollingTabContainerView;
***************
*** 180,185 ****
--- 194,200 ----
  
      invoke-virtual {v0, v2}, Lcom/android/internal/widget/ScrollingTabContainerView;->setVisibility(I)V
  
+     .line 1420
      :cond_3
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 199,204 ****
--- 214,220 ----
  
      if-ne v0, v4, :cond_4
  
+     .line 1421
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      #getter for: Lcom/android/internal/widget/ActionBarView;->mSpinner:Landroid/widget/Spinner;
***************
*** 208,213 ****
--- 224,230 ----
  
      invoke-virtual {v0, v2}, Landroid/widget/Spinner;->setVisibility(I)V
  
+     .line 1423
      :cond_4
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 229,234 ****
--- 246,252 ----
  
      if-eqz v0, :cond_5
  
+     .line 1424
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      #getter for: Lcom/android/internal/widget/ActionBarView;->mCustomNavView:Landroid/view/View;
***************
*** 238,243 ****
--- 256,262 ----
  
      invoke-virtual {v0, v2}, Landroid/view/View;->setVisibility(I)V
  
+     .line 1426
      :cond_5
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 248,263 ****
--- 267,287 ----
  
      invoke-virtual {v0, v3}, Lcom/android/internal/widget/ActionBarView$HomeView;->setIcon(Landroid/graphics/drawable/Drawable;)V
  
+     .line 1427
      iput-object v3, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mCurrentExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 1428
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/ActionBarView;->requestLayout()V
  
+     .line 1429
      invoke-virtual {p2, v2}, Lcom/android/internal/view/menu/MenuItemImpl;->setActionViewExpanded(Z)V
  
+     .line 1431
      return v4
  
+     .line 1414
      :cond_6
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 281,286 ****
--- 305,311 ----
  
      const/16 v3, 0x8
  
+     .line 1372
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      invoke-virtual {p2}, Lcom/android/internal/view/menu/MenuItemImpl;->getActionView()Landroid/view/View;
***************
*** 289,294 ****
--- 314,320 ----
  
      iput-object v1, v0, Lcom/android/internal/widget/ActionBarView;->mExpandedActionView:Landroid/view/View;
  
+     .line 1373
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      #getter for: Lcom/android/internal/widget/ActionBarView;->mExpandedHomeLayout:Lcom/android/internal/widget/ActionBarView$HomeView;
***************
*** 319,326 ****
--- 345,354 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/ActionBarView$HomeView;->setIcon(Landroid/graphics/drawable/Drawable;)V
  
+     .line 1374
      iput-object p2, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mCurrentExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
+     .line 1375
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mExpandedActionView:Landroid/view/View;
***************
*** 333,338 ****
--- 361,367 ----
  
      if-eq v0, v1, :cond_0
  
+     .line 1376
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      iget-object v1, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
***************
*** 341,346 ****
--- 370,376 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/ActionBarView;->addView(Landroid/view/View;)V
  
+     .line 1378
      :cond_0
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 357,362 ****
--- 387,393 ----
  
      if-eq v0, v1, :cond_1
  
+     .line 1379
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      iget-object v1, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
***************
*** 368,373 ****
--- 399,405 ----
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/ActionBarView;->addView(Landroid/view/View;)V
  
+     .line 1381
      :cond_1
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 378,383 ****
--- 410,416 ----
  
      invoke-virtual {v0, v3}, Lcom/android/internal/widget/ActionBarView$HomeView;->setVisibility(I)V
  
+     .line 1382
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      #getter for: Lcom/android/internal/widget/ActionBarView;->mTitleLayout:Landroid/widget/LinearLayout;
***************
*** 396,401 ****
--- 429,435 ----
  
      invoke-virtual {v0, v3}, Landroid/widget/LinearLayout;->setVisibility(I)V
  
+     .line 1383
      :cond_2
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 415,420 ****
--- 449,455 ----
  
      invoke-virtual {v0, v3}, Lcom/android/internal/widget/ScrollingTabContainerView;->setVisibility(I)V
  
+     .line 1384
      :cond_3
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 434,439 ****
--- 469,475 ----
  
      invoke-virtual {v0, v3}, Landroid/widget/Spinner;->setVisibility(I)V
  
+     .line 1385
      :cond_4
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
***************
*** 453,465 ****
--- 489,504 ----
  
      invoke-virtual {v0, v3}, Landroid/view/View;->setVisibility(I)V
  
+     .line 1386
      :cond_5
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/ActionBarView;->requestLayout()V
  
+     .line 1387
      invoke-virtual {p2, v4}, Lcom/android/internal/view/menu/MenuItemImpl;->setActionViewExpanded(Z)V
  
+     .line 1389
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mExpandedActionView:Landroid/view/View;
***************
*** 468,473 ****
--- 507,513 ----
  
      if-eqz v0, :cond_6
  
+     .line 1390
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->this$0:Lcom/android/internal/widget/ActionBarView;
  
      iget-object v0, v0, Lcom/android/internal/widget/ActionBarView;->mExpandedActionView:Landroid/view/View;
***************
*** 476,481 ****
--- 516,522 ----
  
      invoke-interface {v0}, Landroid/view/CollapsibleActionView;->onActionViewExpanded()V
  
+     .line 1393
      :cond_6
      return v4
  .end method
***************
*** 484,489 ****
--- 525,531 ----
      .locals 1
  
      .prologue
+     .line 1367
      const/4 v0, 0x0
  
      return v0
***************
*** 493,498 ****
--- 535,541 ----
      .locals 1
  
      .prologue
+     .line 1436
      const/4 v0, 0x0
  
      return v0
***************
*** 503,508 ****
--- 546,552 ----
      .parameter "root"
  
      .prologue
+     .line 1325
      const/4 v0, 0x0
  
      return-object v0
***************
*** 514,519 ****
--- 558,564 ----
      .parameter "menu"
  
      .prologue
+     .line 1317
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mMenu:Lcom/android/internal/view/menu/MenuBuilder;
  
      if-eqz v0, :cond_0
***************
*** 522,536 ****
--- 567,584 ----
  
      if-eqz v0, :cond_0
  
+     .line 1318
      iget-object v0, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mMenu:Lcom/android/internal/view/menu/MenuBuilder;
  
      iget-object v1, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mCurrentExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
      invoke-virtual {v0, v1}, Lcom/android/internal/view/menu/MenuBuilder;->collapseItemActionView(Lcom/android/internal/view/menu/MenuItemImpl;)Z
  
+     .line 1320
      :cond_0
      iput-object p2, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mMenu:Lcom/android/internal/view/menu/MenuBuilder;
  
+     .line 1321
      return-void
  .end method
  
***************
*** 540,545 ****
--- 588,594 ----
      .parameter "allMenusAreClosing"
  
      .prologue
+     .line 1363
      return-void
  .end method
  
***************
*** 548,553 ****
--- 597,603 ----
      .parameter "state"
  
      .prologue
+     .line 1446
      return-void
  .end method
  
***************
*** 555,560 ****
--- 605,611 ----
      .locals 1
  
      .prologue
+     .line 1441
      const/4 v0, 0x0
  
      return-object v0
***************
*** 565,570 ****
--- 616,622 ----
      .parameter "subMenu"
  
      .prologue
+     .line 1358
      const/4 v0, 0x0
  
      return v0
***************
*** 575,580 ****
--- 627,633 ----
      .parameter "cb"
  
      .prologue
+     .line 1354
      return-void
  .end method
  
***************
*** 583,605 ****
--- 636,663 ----
      .parameter "cleared"
  
      .prologue
+     .line 1331
      iget-object v4, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mCurrentExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
      if-eqz v4, :cond_1
  
+     .line 1332
      const/4 v1, 0x0
  
+     .line 1334
      .local v1, found:Z
      iget-object v4, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mMenu:Lcom/android/internal/view/menu/MenuBuilder;
  
      if-eqz v4, :cond_0
  
+     .line 1335
      iget-object v4, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mMenu:Lcom/android/internal/view/menu/MenuBuilder;
  
      invoke-virtual {v4}, Lcom/android/internal/view/menu/MenuBuilder;->size()I
  
      move-result v0
  
+     .line 1336
      .local v0, count:I
      const/4 v2, 0x0
  
***************
*** 607,641 ****
--- 665,706 ----
      :goto_0
      if-ge v2, v0, :cond_0
  
+     .line 1337
      iget-object v4, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mMenu:Lcom/android/internal/view/menu/MenuBuilder;
  
      invoke-virtual {v4, v2}, Lcom/android/internal/view/menu/MenuBuilder;->getItem(I)Landroid/view/MenuItem;
  
      move-result-object v3
  
+     .line 1338
      .local v3, item:Landroid/view/MenuItem;
      iget-object v4, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mCurrentExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
      if-ne v3, v4, :cond_2
  
+     .line 1339
      const/4 v1, 0x1
  
+     .line 1345
      .end local v0           #count:I
      .end local v2           #i:I
      .end local v3           #item:Landroid/view/MenuItem;
      :cond_0
      if-nez v1, :cond_1
  
+     .line 1347
      iget-object v4, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mMenu:Lcom/android/internal/view/menu/MenuBuilder;
  
      iget-object v5, p0, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->mCurrentExpandedItem:Lcom/android/internal/view/menu/MenuItemImpl;
  
      invoke-virtual {p0, v4, v5}, Lcom/android/internal/widget/ActionBarView$ExpandedActionViewMenuPresenter;->collapseItemActionView(Lcom/android/internal/view/menu/MenuBuilder;Lcom/android/internal/view/menu/MenuItemImpl;)Z
  
+     .line 1350
      .end local v1           #found:Z
      :cond_1
      return-void
  
+     .line 1336
      .restart local v0       #count:I
      .restart local v1       #found:Z
      .restart local v2       #i:I
