*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/IRemoteViewsFactory$Stub.smali	2012-03-23 19:37:31.538350184 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/IRemoteViewsFactory$Stub.smali.original	2012-03-23 19:34:46.059332677 -0500
***************
*** 52,63 ****
--- 52,66 ----
      .locals 1
  
      .prologue
+     .line 15
      invoke-direct {p0}, Landroid/os/Binder;-><init>()V
  
+     .line 16
      const-string v0, "com.android.internal.widget.IRemoteViewsFactory"
  
      invoke-virtual {p0, p0, v0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->attachInterface(Landroid/os/IInterface;Ljava/lang/String;)V
  
+     .line 17
      return-void
  .end method
  
***************
*** 66,78 ****
--- 69,85 ----
      .parameter "obj"
  
      .prologue
+     .line 24
      if-nez p0, :cond_0
  
+     .line 25
      const/4 v0, 0x0
  
+     .line 31
      :goto_0
      return-object v0
  
+     .line 27
      :cond_0
      const-string v1, "com.android.internal.widget.IRemoteViewsFactory"
  
***************
*** 80,85 ****
--- 87,93 ----
  
      move-result-object v0
  
+     .line 28
      .local v0, iin:Landroid/os/IInterface;
      if-eqz v0, :cond_1
  
***************
*** 87,96 ****
--- 95,106 ----
  
      if-eqz v1, :cond_1
  
+     .line 29
      check-cast v0, Lcom/android/internal/widget/IRemoteViewsFactory;
  
      goto :goto_0
  
+     .line 31
      :cond_1
      new-instance v0, Lcom/android/internal/widget/IRemoteViewsFactory$Stub$Proxy;
  
***************
*** 106,111 ****
--- 116,122 ----
      .locals 0
  
      .prologue
+     .line 35
      return-object p0
  .end method
  
***************
*** 126,133 ****
--- 137,146 ----
  
      const/4 v4, 0x1
  
+     .line 39
      sparse-switch p1, :sswitch_data_0
  
+     .line 145
      invoke-super {p0, p1, p2, p3, p4}, Landroid/os/Binder;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
      move-result v4
***************
*** 135,140 ****
--- 148,154 ----
      :goto_0
      return v4
  
+     .line 43
      :sswitch_0
      const-string v3, "com.android.internal.widget.IRemoteViewsFactory"
  
***************
*** 142,178 ****
--- 156,200 ----
  
      goto :goto_0
  
+     .line 48
      :sswitch_1
      const-string v3, "com.android.internal.widget.IRemoteViewsFactory"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 49
      invoke-virtual {p0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->onDataSetChanged()V
  
+     .line 50
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
      goto :goto_0
  
+     .line 55
      :sswitch_2
      const-string v3, "com.android.internal.widget.IRemoteViewsFactory"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 56
      invoke-virtual {p0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->onDataSetChangedAsync()V
  
      goto :goto_0
  
+     .line 61
      :sswitch_3
      const-string v3, "com.android.internal.widget.IRemoteViewsFactory"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 63
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v3
  
      if-eqz v3, :cond_0
  
+     .line 64
      sget-object v3, Landroid/content/Intent;->CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v3, p2}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 181,192 ****
--- 203,216 ----
  
      check-cast v0, Landroid/content/Intent;
  
+     .line 69
      .local v0, _arg0:Landroid/content/Intent;
      :goto_1
      invoke-virtual {p0, v0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->onDestroy(Landroid/content/Intent;)V
  
      goto :goto_0
  
+     .line 67
      .end local v0           #_arg0:Landroid/content/Intent;
      :cond_0
      const/4 v0, 0x0
***************
*** 194,247 ****
--- 218,284 ----
      .restart local v0       #_arg0:Landroid/content/Intent;
      goto :goto_1
  
+     .line 74
      .end local v0           #_arg0:Landroid/content/Intent;
      :sswitch_4
      const-string v3, "com.android.internal.widget.IRemoteViewsFactory"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 75
      invoke-virtual {p0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->getCount()I
  
      move-result v1
  
+     .line 76
      .local v1, _result:I
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 77
      invoke-virtual {p3, v1}, Landroid/os/Parcel;->writeInt(I)V
  
      goto :goto_0
  
+     .line 82
      .end local v1           #_result:I
      :sswitch_5
      const-string v5, "com.android.internal.widget.IRemoteViewsFactory"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 84
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 85
      .local v0, _arg0:I
      invoke-virtual {p0, v0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->getViewAt(I)Landroid/widget/RemoteViews;
  
      move-result-object v1
  
+     .line 86
      .local v1, _result:Landroid/widget/RemoteViews;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 87
      if-eqz v1, :cond_1
  
+     .line 88
      invoke-virtual {p3, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 89
      invoke-virtual {v1, p3, v4}, Landroid/widget/RemoteViews;->writeToParcel(Landroid/os/Parcel;I)V
  
      goto :goto_0
  
+     .line 92
      :cond_1
      invoke-virtual {p3, v3}, Landroid/os/Parcel;->writeInt(I)V
  
      goto :goto_0
  
+     .line 98
      .end local v0           #_arg0:I
      .end local v1           #_result:Landroid/widget/RemoteViews;
      :sswitch_6
***************
*** 249,313 ****
--- 286,366 ----
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 99
      invoke-virtual {p0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->getLoadingView()Landroid/widget/RemoteViews;
  
      move-result-object v1
  
+     .line 100
      .restart local v1       #_result:Landroid/widget/RemoteViews;
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 101
      if-eqz v1, :cond_2
  
+     .line 102
      invoke-virtual {p3, v4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 103
      invoke-virtual {v1, p3, v4}, Landroid/widget/RemoteViews;->writeToParcel(Landroid/os/Parcel;I)V
  
      goto :goto_0
  
+     .line 106
      :cond_2
      invoke-virtual {p3, v3}, Landroid/os/Parcel;->writeInt(I)V
  
      goto :goto_0
  
+     .line 112
      .end local v1           #_result:Landroid/widget/RemoteViews;
      :sswitch_7
      const-string v3, "com.android.internal.widget.IRemoteViewsFactory"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 113
      invoke-virtual {p0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->getViewTypeCount()I
  
      move-result v1
  
+     .line 114
      .local v1, _result:I
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 115
      invoke-virtual {p3, v1}, Landroid/os/Parcel;->writeInt(I)V
  
      goto/16 :goto_0
  
+     .line 120
      .end local v1           #_result:I
      :sswitch_8
      const-string v3, "com.android.internal.widget.IRemoteViewsFactory"
  
      invoke-virtual {p2, v3}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 122
      invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
+     .line 123
      .restart local v0       #_arg0:I
      invoke-virtual {p0, v0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->getItemId(I)J
  
      move-result-wide v1
  
+     .line 124
      .local v1, _result:J
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 125
      invoke-virtual {p3, v1, v2}, Landroid/os/Parcel;->writeLong(J)V
  
      goto/16 :goto_0
  
+     .line 130
      .end local v0           #_arg0:I
      .end local v1           #_result:J
      :sswitch_9
***************
*** 315,327 ****
--- 368,383 ----
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 131
      invoke-virtual {p0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->hasStableIds()Z
  
      move-result v1
  
+     .line 132
      .local v1, _result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 133
      if-eqz v1, :cond_3
  
      move v3, v4
***************
*** 331,349 ****
--- 387,409 ----
  
      goto/16 :goto_0
  
+     .line 138
      .end local v1           #_result:Z
      :sswitch_a
      const-string v5, "com.android.internal.widget.IRemoteViewsFactory"
  
      invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
  
+     .line 139
      invoke-virtual {p0}, Lcom/android/internal/widget/IRemoteViewsFactory$Stub;->isCreated()Z
  
      move-result v1
  
+     .line 140
      .restart local v1       #_result:Z
      invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
  
+     .line 141
      if-eqz v1, :cond_4
  
      move v3, v4
***************
*** 353,358 ****
--- 413,419 ----
  
      goto/16 :goto_0
  
+     .line 39
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_1
