*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/LockPatternView$Cell.smali	2012-03-23 19:37:31.345346663 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/LockPatternView$Cell.smali.original	2012-03-23 19:34:46.056332623 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .prologue
      const/4 v4, 0x3
  
+     .line 140
      filled-new-array {v4, v4}, [I
  
      move-result-object v2
***************
*** 45,62 ****
--- 46,66 ----
  
      sput-object v2, Lcom/android/internal/widget/LockPatternView$Cell;->sCells:[[Lcom/android/internal/widget/LockPatternView$Cell;
  
+     .line 142
      const/4 v0, 0x0
  
      .local v0, i:I
      :goto_0
      if-ge v0, v4, :cond_1
  
+     .line 143
      const/4 v1, 0x0
  
      .local v1, j:I
      :goto_1
      if-ge v1, v4, :cond_0
  
+     .line 144
      sget-object v2, Lcom/android/internal/widget/LockPatternView$Cell;->sCells:[[Lcom/android/internal/widget/LockPatternView$Cell;
  
      aget-object v2, v2, v0
***************
*** 67,81 ****
--- 71,88 ----
  
      aput-object v3, v2, v1
  
+     .line 143
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
  
+     .line 142
      :cond_0
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 147
      .end local v1           #j:I
      :cond_1
      return-void
***************
*** 87,100 ****
--- 94,112 ----
      .parameter "column"
  
      .prologue
+     .line 153
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 154
      invoke-static {p1, p2}, Lcom/android/internal/widget/LockPatternView$Cell;->checkRange(II)V
  
+     .line 155
      iput p1, p0, Lcom/android/internal/widget/LockPatternView$Cell;->row:I
  
+     .line 156
      iput p2, p0, Lcom/android/internal/widget/LockPatternView$Cell;->column:I
  
+     .line 157
      return-void
  .end method
  
***************
*** 106,129 ****
      .prologue
      const/4 v0, 0x2
  
      if-ltz p0, :cond_0
  
      if-le p0, v0, :cond_1
  
      :cond_0
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string v1, "row must be in range 0-2"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_1
      if-ltz p1, :cond_2
  
      if-le p1, v0, :cond_3
  
      :cond_2
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
--- 118,145 ----
      .prologue
      const/4 v0, 0x2
  
+     .line 177
      if-ltz p0, :cond_0
  
      if-le p0, v0, :cond_1
  
+     .line 178
      :cond_0
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v1, "row must be in range 0-2"
  
      invoke-direct {v0, v1}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 180
      :cond_1
      if-ltz p1, :cond_2
  
      if-le p1, v0, :cond_3
  
+     .line 181
      :cond_2
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
***************
*** 133,138 ****
--- 149,155 ----
  
      throw v0
  
+     .line 183
      :cond_3
      return-void
  .end method
***************
*** 143,148 ****
--- 160,166 ----
      .parameter "column"
  
      .prologue
+     .line 172
      const-class v1, Lcom/android/internal/widget/LockPatternView$Cell;
  
      monitor-enter v1
***************
*** 150,155 ****
--- 168,174 ----
      :try_start_0
      invoke-static {p0, p1}, Lcom/android/internal/widget/LockPatternView$Cell;->checkRange(II)V
  
+     .line 173
      sget-object v0, Lcom/android/internal/widget/LockPatternView$Cell;->sCells:[[Lcom/android/internal/widget/LockPatternView$Cell;
  
      aget-object v0, v0, p0
***************
*** 162,167 ****
--- 181,187 ----
  
      return-object v0
  
+     .line 172
      :catchall_0
      move-exception v0
  
***************
*** 176,181 ****
--- 196,202 ----
      .locals 1
  
      .prologue
+     .line 164
      iget v0, p0, Lcom/android/internal/widget/LockPatternView$Cell;->column:I
  
      return v0
***************
*** 185,190 ****
--- 206,212 ----
      .locals 1
  
      .prologue
+     .line 160
      iget v0, p0, Lcom/android/internal/widget/LockPatternView$Cell;->row:I
  
      return v0
***************
*** 194,199 ****
--- 216,222 ----
      .locals 2
  
      .prologue
+     .line 186
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
