*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/LockPatternView$SavedState.smali	2012-03-23 19:37:33.994394970 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/LockPatternView$SavedState.smali.original	2012-03-23 19:34:46.105333517 -0500
***************
*** 44,49 ****
--- 44,50 ----
      .locals 1
  
      .prologue
+     .line 1172
      new-instance v0, Lcom/android/internal/widget/LockPatternView$SavedState$1;
  
      invoke-direct {v0}, Lcom/android/internal/widget/LockPatternView$SavedState$1;-><init>()V
***************
*** 60,79 ****
--- 61,84 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 1134
      invoke-direct {p0, p1}, Landroid/view/View$BaseSavedState;-><init>(Landroid/os/Parcel;)V
  
+     .line 1135
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mSerializedPattern:Ljava/lang/String;
  
+     .line 1136
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mDisplayMode:I
  
+     .line 1137
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->readValue(Ljava/lang/ClassLoader;)Ljava/lang/Object;
  
      move-result-object v0
***************
*** 86,91 ****
--- 91,97 ----
  
      iput-boolean v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mInputEnabled:Z
  
+     .line 1138
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->readValue(Ljava/lang/ClassLoader;)Ljava/lang/Object;
  
      move-result-object v0
***************
*** 98,103 ****
--- 104,110 ----
  
      iput-boolean v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mInStealthMode:Z
  
+     .line 1139
      invoke-virtual {p1, v1}, Landroid/os/Parcel;->readValue(Ljava/lang/ClassLoader;)Ljava/lang/Object;
  
      move-result-object v0
***************
*** 110,115 ****
--- 117,123 ----
  
      iput-boolean v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mTactileFeedbackEnabled:Z
  
+     .line 1140
      return-void
  .end method
  
***************
*** 119,124 ****
--- 127,133 ----
      .parameter "x1"
  
      .prologue
+     .line 1109
      invoke-direct {p0, p1}, Lcom/android/internal/widget/LockPatternView$SavedState;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 134,151 ****
--- 143,167 ----
      .parameter "tactileFeedbackEnabled"
  
      .prologue
+     .line 1122
      invoke-direct {p0, p1}, Landroid/view/View$BaseSavedState;-><init>(Landroid/os/Parcelable;)V
  
+     .line 1123
      iput-object p2, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mSerializedPattern:Ljava/lang/String;
  
+     .line 1124
      iput p3, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mDisplayMode:I
  
+     .line 1125
      iput-boolean p4, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mInputEnabled:Z
  
+     .line 1126
      iput-boolean p5, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mInStealthMode:Z
  
+     .line 1127
      iput-boolean p6, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mTactileFeedbackEnabled:Z
  
+     .line 1128
      return-void
  .end method
  
***************
*** 160,165 ****
--- 176,182 ----
      .parameter "x6"
  
      .prologue
+     .line 1109
      invoke-direct/range {p0 .. p6}, Lcom/android/internal/widget/LockPatternView$SavedState;-><init>(Landroid/os/Parcelable;Ljava/lang/String;IZZZ)V
  
      return-void
***************
*** 171,176 ****
--- 188,194 ----
      .locals 1
  
      .prologue
+     .line 1147
      iget v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mDisplayMode:I
  
      return v0
***************
*** 180,185 ****
--- 198,204 ----
      .locals 1
  
      .prologue
+     .line 1143
      iget-object v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mSerializedPattern:Ljava/lang/String;
  
      return-object v0
***************
*** 189,194 ****
--- 208,214 ----
      .locals 1
  
      .prologue
+     .line 1155
      iget-boolean v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mInStealthMode:Z
  
      return v0
***************
*** 198,203 ****
--- 218,224 ----
      .locals 1
  
      .prologue
+     .line 1151
      iget-boolean v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mInputEnabled:Z
  
      return v0
***************
*** 207,212 ****
--- 228,234 ----
      .locals 1
  
      .prologue
+     .line 1159
      iget-boolean v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mTactileFeedbackEnabled:Z
  
      return v0
***************
*** 218,233 ****
--- 240,259 ----
      .parameter "flags"
  
      .prologue
+     .line 1164
      invoke-super {p0, p1, p2}, Landroid/view/View$BaseSavedState;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 1165
      iget-object v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mSerializedPattern:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 1166
      iget v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mDisplayMode:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 1167
      iget-boolean v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mInputEnabled:Z
  
      invoke-static {v0}, Ljava/lang/Boolean;->valueOf(Z)Ljava/lang/Boolean;
***************
*** 236,241 ****
--- 262,268 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeValue(Ljava/lang/Object;)V
  
+     .line 1168
      iget-boolean v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mInStealthMode:Z
  
      invoke-static {v0}, Ljava/lang/Boolean;->valueOf(Z)Ljava/lang/Boolean;
***************
*** 244,249 ****
--- 271,277 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeValue(Ljava/lang/Object;)V
  
+     .line 1169
      iget-boolean v0, p0, Lcom/android/internal/widget/LockPatternView$SavedState;->mTactileFeedbackEnabled:Z
  
      invoke-static {v0}, Ljava/lang/Boolean;->valueOf(Z)Ljava/lang/Boolean;
***************
*** 252,256 ****
--- 280,285 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeValue(Ljava/lang/Object;)V
  
+     .line 1170
      return-void
  .end method
