*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/PasswordEntryKeyboardHelper.smali	2012-03-23 19:37:34.112397121 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/PasswordEntryKeyboardHelper.smali.original	2012-03-23 19:34:46.107333553 -0500
***************
*** 54,63 ****
--- 54,65 ----
      .parameter "targetView"
  
      .prologue
+     .line 59
      const/4 v0, 0x1
  
      invoke-direct {p0, p1, p2, p3, v0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;-><init>(Landroid/content/Context;Landroid/inputmethodservice/KeyboardView;Landroid/view/View;Z)V
  
+     .line 60
      return-void
  .end method
  
***************
*** 71,90 ****
--- 73,100 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 63
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 45
      iput v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardMode:I
  
+     .line 46
      iput v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
+     .line 56
      iput-boolean v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mEnableHaptics:Z
  
+     .line 64
      iput-object p1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
  
+     .line 65
      iput-object p3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mTargetView:Landroid/view/View;
  
+     .line 66
      iput-object p2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
+     .line 67
      if-nez p4, :cond_0
  
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
***************
*** 99,114 ****
--- 109,128 ----
  
      if-ne v0, v1, :cond_1
  
+     .line 69
      :cond_0
      invoke-direct {p0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->createKeyboards()V
  
+     .line 74
      :goto_0
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
      invoke-virtual {v0, p0}, Landroid/inputmethodservice/KeyboardView;->setOnKeyboardActionListener(Landroid/inputmethodservice/KeyboardView$OnKeyboardActionListener;)V
  
+     .line 75
      return-void
  
+     .line 71
      :cond_1
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
***************
*** 139,144 ****
--- 153,159 ----
  
      const/4 v3, 0x1
  
+     .line 109
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 149,154 ****
--- 164,170 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mNumericKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 110
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 159,168 ****
--- 175,186 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 112
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/PasswordEntryKeyboard;->enableShiftLock()V
  
+     .line 114
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 173,186 ****
--- 191,207 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 117
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/PasswordEntryKeyboard;->enableShiftLock()V
  
+     .line 118
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0, v3}, Lcom/android/internal/widget/PasswordEntryKeyboard;->setShifted(Z)Z
  
+     .line 120
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 191,200 ****
--- 212,223 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 121
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/PasswordEntryKeyboard;->enableShiftLock()V
  
+     .line 123
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 205,218 ****
--- 228,244 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 125
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/PasswordEntryKeyboard;->enableShiftLock()V
  
+     .line 126
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0, v3}, Lcom/android/internal/widget/PasswordEntryKeyboard;->setShifted(Z)Z
  
+     .line 127
      return-void
  .end method
  
***************
*** 226,231 ****
--- 252,258 ----
  
      const/4 v6, 0x1
  
+     .line 86
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 236,241 ****
--- 263,269 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mNumericKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 88
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 250,259 ****
--- 278,289 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 90
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/PasswordEntryKeyboard;->enableShiftLock()V
  
+     .line 92
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 268,281 ****
--- 298,314 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 95
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/PasswordEntryKeyboard;->enableShiftLock()V
  
+     .line 96
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0, v6}, Lcom/android/internal/widget/PasswordEntryKeyboard;->setShifted(Z)Z
  
+     .line 98
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 286,295 ****
--- 319,330 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 100
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/PasswordEntryKeyboard;->enableShiftLock()V
  
+     .line 102
      new-instance v0, Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 300,313 ****
--- 335,351 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 104
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/PasswordEntryKeyboard;->enableShiftLock()V
  
+     .line 105
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      invoke-virtual {v0, v6}, Lcom/android/internal/widget/PasswordEntryKeyboard;->setShifted(Z)Z
  
+     .line 106
      return-void
  .end method
  
***************
*** 317,322 ****
--- 355,361 ----
      .parameter "keyCodes"
  
      .prologue
+     .line 269
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
      invoke-virtual {v0}, Landroid/inputmethodservice/KeyboardView;->isShifted()Z
***************
*** 333,345 ****
--- 372,387 ----
  
      if-eq p1, v0, :cond_0
  
+     .line 270
      invoke-static {p1}, Ljava/lang/Character;->toUpperCase(I)I
  
      move-result p1
  
+     .line 272
      :cond_0
      invoke-direct {p0, p1}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->sendKeyEventsToTarget(I)V
  
+     .line 273
      return-void
  .end method
  
***************
*** 347,352 ****
--- 389,395 ----
      .locals 0
  
      .prologue
+     .line 277
      return-void
  .end method
  
***************
*** 354,368 ****
--- 397,414 ----
      .locals 3
  
      .prologue
+     .line 222
      iget-object v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
      invoke-virtual {v2}, Landroid/inputmethodservice/KeyboardView;->getKeyboard()Landroid/inputmethodservice/Keyboard;
  
      move-result-object v0
  
+     .line 223
      .local v0, current:Landroid/inputmethodservice/Keyboard;
      const/4 v1, 0x0
  
+     .line 224
      .local v1, next:Landroid/inputmethodservice/Keyboard;
      iget-object v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
***************
*** 372,395 ****
--- 418,447 ----
  
      if-ne v0, v2, :cond_3
  
+     .line 225
      :cond_0
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 229
      :cond_1
      :goto_0
      if-eqz v1, :cond_2
  
+     .line 230
      iget-object v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
      invoke-virtual {v2, v1}, Landroid/inputmethodservice/KeyboardView;->setKeyboard(Landroid/inputmethodservice/Keyboard;)V
  
+     .line 231
      const/4 v2, 0x0
  
      iput v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
+     .line 233
      :cond_2
      return-void
  
+     .line 226
      :cond_3
      iget-object v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
***************
*** 399,404 ****
--- 451,457 ----
  
      if-ne v0, v2, :cond_1
  
+     .line 227
      :cond_4
      iget-object v1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
***************
*** 415,428 ****
--- 468,484 ----
  
      const/4 v4, 0x1
  
+     .line 241
      iget-object v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
      if-nez v5, :cond_1
  
+     .line 265
      :cond_0
      :goto_0
      return-void
  
+     .line 244
      :cond_1
      iget-object v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
***************
*** 430,438 ****
--- 486,496 ----
  
      move-result-object v0
  
+     .line 245
      .local v0, current:Landroid/inputmethodservice/Keyboard;
      const/4 v2, 0x0
  
+     .line 246
      .local v2, next:Lcom/android/internal/widget/PasswordEntryKeyboard;
      iget-object v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
***************
*** 445,456 ****
--- 503,516 ----
      :cond_2
      move v1, v4
  
+     .line 248
      .local v1, isAlphaMode:Z
      :goto_1
      iget v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
      if-nez v5, :cond_8
  
+     .line 249
      if-eqz v1, :cond_6
  
      move v5, v4
***************
*** 458,477 ****
--- 518,542 ----
      :goto_2
      iput v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
+     .line 250
      if-eqz v1, :cond_7
  
      iget-object v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
+     .line 258
      :cond_3
      :goto_3
      if-eqz v2, :cond_0
  
+     .line 259
      if-eq v2, v0, :cond_4
  
+     .line 260
      iget-object v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
      invoke-virtual {v5, v2}, Landroid/inputmethodservice/KeyboardView;->setKeyboard(Landroid/inputmethodservice/Keyboard;)V
  
+     .line 262
      :cond_4
      iget v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
***************
*** 482,487 ****
--- 547,553 ----
      :goto_4
      invoke-virtual {v2, v5}, Lcom/android/internal/widget/PasswordEntryKeyboard;->setShiftLocked(Z)V
  
+     .line 263
      iget-object v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
      iget v6, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
***************
*** 497,522 ****
--- 563,594 ----
      :cond_5
      move v1, v3
  
+     .line 246
      goto :goto_1
  
      .restart local v1       #isAlphaMode:Z
      :cond_6
      move v5, v6
  
+     .line 249
      goto :goto_2
  
+     .line 250
      :cond_7
      iget-object v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mSymbolsKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
  
      goto :goto_3
  
+     .line 251
      :cond_8
      iget v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
      if-ne v5, v4, :cond_a
  
+     .line 252
      iput v6, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
+     .line 253
      if-eqz v1, :cond_9
  
      iget-object v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboardShifted:Lcom/android/internal/widget/PasswordEntryKeyboard;
***************
*** 529,541 ****
--- 601,616 ----
  
      goto :goto_6
  
+     .line 254
      :cond_a
      iget v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
      if-ne v5, v6, :cond_3
  
+     .line 255
      iput v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
+     .line 256
      if-eqz v1, :cond_b
  
      iget-object v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mQwertyKeyboard:Lcom/android/internal/widget/PasswordEntryKeyboard;
***************
*** 551,561 ****
--- 626,638 ----
      :cond_c
      move v5, v3
  
+     .line 262
      goto :goto_4
  
      :cond_d
      move v4, v3
  
+     .line 263
      goto :goto_5
  .end method
  
***************
*** 563,572 ****
--- 640,651 ----
      .locals 3
  
      .prologue
+     .line 284
      iget-boolean v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mEnableHaptics:Z
  
      if-eqz v0, :cond_0
  
+     .line 285
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
      const/4 v1, 0x1
***************
*** 575,580 ****
--- 654,660 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/inputmethodservice/KeyboardView;->performHapticFeedback(II)Z
  
+     .line 289
      :cond_0
      return-void
  .end method
***************
*** 584,595 ****
--- 664,677 ----
      .parameter "character"
  
      .prologue
+     .line 150
      iget-object v5, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mTargetView:Landroid/view/View;
  
      invoke-virtual {v5}, Landroid/view/View;->getHandler()Landroid/os/Handler;
  
      move-result-object v3
  
+     .line 151
      .local v3, handler:Landroid/os/Handler;
      const/4 v5, -0x1
  
***************
*** 611,621 ****
--- 693,706 ----
  
      move-result-object v2
  
+     .line 153
      .local v2, events:[Landroid/view/KeyEvent;
      if-eqz v2, :cond_0
  
+     .line 154
      array-length v0, v2
  
+     .line 155
      .local v0, N:I
      const/4 v4, 0x0
  
***************
*** 623,630 ****
--- 708,717 ----
      :goto_0
      if-ge v4, v0, :cond_0
  
+     .line 156
      aget-object v1, v2, v4
  
+     .line 157
      .local v1, event:Landroid/view/KeyEvent;
      invoke-virtual {v1}, Landroid/view/KeyEvent;->getFlags()I
  
***************
*** 638,643 ****
--- 725,731 ----
  
      move-result-object v1
  
+     .line 159
      const/16 v5, 0x3ed
  
      invoke-virtual {v3, v5, v1}, Landroid/os/Handler;->obtainMessage(ILjava/lang/Object;)Landroid/os/Message;
***************
*** 646,655 ****
--- 734,745 ----
  
      invoke-virtual {v3, v5}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 155
      add-int/lit8 v4, v4, 0x1
  
      goto :goto_0
  
+     .line 162
      .end local v0           #N:I
      .end local v1           #event:Landroid/view/KeyEvent;
      .end local v4           #i:I
***************
*** 663,674 ****
--- 753,767 ----
      .locals 1
  
      .prologue
+     .line 236
      const/16 v0, 0x43
  
      invoke-virtual {p0, v0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->sendDownUpKeyEvents(I)V
  
+     .line 237
      invoke-direct {p0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->performHapticFeedback()V
  
+     .line 238
      return-void
  .end method
  
***************
*** 676,681 ****
--- 769,775 ----
      .locals 1
  
      .prologue
+     .line 82
      iget v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardMode:I
  
      if-nez v0, :cond_0
***************
*** 697,730 ****
--- 791,832 ----
      .parameter "keyCodes"
  
      .prologue
+     .line 178
      const/4 v0, -0x5
  
      if-ne p1, v0, :cond_1
  
+     .line 179
      invoke-virtual {p0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->handleBackspace()V
  
+     .line 196
      :cond_0
      :goto_0
      return-void
  
+     .line 180
      :cond_1
      const/4 v0, -0x1
  
      if-ne p1, v0, :cond_2
  
+     .line 181
      invoke-direct {p0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->handleShift()V
  
      goto :goto_0
  
+     .line 182
      :cond_2
      const/4 v0, -0x3
  
      if-ne p1, v0, :cond_3
  
+     .line 183
      invoke-direct {p0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->handleClose()V
  
      goto :goto_0
  
+     .line 185
      :cond_3
      const/4 v0, -0x2
  
***************
*** 734,756 ****
--- 836,863 ----
  
      if-eqz v0, :cond_4
  
+     .line 186
      invoke-direct {p0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->handleModeChange()V
  
      goto :goto_0
  
+     .line 188
      :cond_4
      invoke-direct {p0, p1, p2}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->handleCharacter(I[I)V
  
+     .line 190
      iget v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
      const/4 v1, 0x1
  
      if-ne v0, v1, :cond_0
  
+     .line 192
      const/4 v0, 0x2
  
      iput v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
+     .line 193
      invoke-direct {p0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->handleShift()V
  
      goto :goto_0
***************
*** 761,768 ****
--- 868,877 ----
      .parameter "primaryCode"
  
      .prologue
+     .line 280
      invoke-direct {p0}, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->performHapticFeedback()V
  
+     .line 281
      return-void
  .end method
  
***************
*** 771,776 ****
--- 880,886 ----
      .parameter "primaryCode"
  
      .prologue
+     .line 293
      return-void
  .end method
  
***************
*** 779,784 ****
--- 889,895 ----
      .parameter "text"
  
      .prologue
+     .line 297
      return-void
  .end method
  
***************
*** 787,796 ****
--- 898,909 ----
      .parameter "keyEventCode"
  
      .prologue
+     .line 165
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v1
  
+     .line 166
      .local v1, eventTime:J
      iget-object v0, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mTargetView:Landroid/view/View;
  
***************
*** 798,803 ****
--- 911,917 ----
  
      move-result-object v12
  
+     .line 167
      .local v12, handler:Landroid/os/Handler;
      const/16 v13, 0x3f3
  
***************
*** 827,832 ****
--- 941,947 ----
  
      invoke-virtual {v12, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 171
      const/16 v13, 0x3f3
  
      new-instance v0, Landroid/view/KeyEvent;
***************
*** 855,860 ****
--- 970,976 ----
  
      invoke-virtual {v12, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 175
      return-void
  .end method
  
***************
*** 863,870 ****
--- 979,988 ----
      .parameter "enabled"
  
      .prologue
+     .line 78
      iput-boolean p1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mEnableHaptics:Z
  
+     .line 79
      return-void
  .end method
  
***************
*** 877,889 ****
--- 995,1011 ----
  
      const/4 v2, 0x0
  
+     .line 130
      packed-switch p1, :pswitch_data_0
  
+     .line 146
      :goto_0
      iput p1, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardMode:I
  
+     .line 147
      return-void
  
+     .line 132
      :pswitch_0
      iget-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
***************
*** 891,905 ****
  
      invoke-virtual {v3, v4}, Landroid/inputmethodservice/KeyboardView;->setKeyboard(Landroid/inputmethodservice/Keyboard;)V
  
      iput v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
      iget-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
  
      invoke-virtual {v3}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v3
  
!     const-string v4, "show_password"
  
      invoke-static {v3, v4, v1}, Landroid/provider/Settings$System;->getInt(Landroid/content/ContentResolver;Ljava/lang/String;I)I
  
--- 1013,1029 ----
  
      invoke-virtual {v3, v4}, Landroid/inputmethodservice/KeyboardView;->setKeyboard(Landroid/inputmethodservice/Keyboard;)V
  
+     .line 133
      iput v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
+     .line 134
      iget-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
  
      invoke-virtual {v3}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v3
  
!     const-string/jumbo v4, "show_password"
  
      invoke-static {v3, v4, v1}, Landroid/provider/Settings$System;->getInt(Landroid/content/ContentResolver;Ljava/lang/String;I)I
  
***************
*** 907,916 ****
--- 1031,1042 ----
  
      if-eqz v3, :cond_1
  
+     .line 137
      .local v1, visiblePassword:Z
      :goto_1
      const/4 v0, 0x0
  
+     .line 138
      .local v0, enablePreview:Z
      iget-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
***************
*** 926,933 ****
--- 1052,1061 ----
      :cond_1
      move v1, v2
  
+     .line 134
      goto :goto_1
  
+     .line 141
      :pswitch_1
      iget-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
***************
*** 935,948 ****
--- 1063,1079 ----
  
      invoke-virtual {v3, v4}, Landroid/inputmethodservice/KeyboardView;->setKeyboard(Landroid/inputmethodservice/Keyboard;)V
  
+     .line 142
      iput v2, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardState:I
  
+     .line 143
      iget-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mKeyboardView:Landroid/inputmethodservice/KeyboardView;
  
      invoke-virtual {v3, v2}, Landroid/inputmethodservice/KeyboardView;->setPreviewEnabled(Z)V
  
      goto :goto_0
  
+     .line 130
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
***************
*** 955,962 ****
--- 1086,1095 ----
      .parameter "id"
  
      .prologue
+     .line 203
      const/4 v2, 0x0
  
+     .line 205
      .local v2, tmpArray:[I
      :try_start_0
      iget-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mContext:Landroid/content/Context;
***************
*** 971,993 ****
--- 1104,1132 ----
  
      move-result-object v2
  
+     .line 211
      :cond_0
      :goto_0
      if-nez v2, :cond_2
  
+     .line 212
      const/4 v3, 0x0
  
      iput-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mVibratePattern:[J
  
+     .line 219
      :cond_1
      return-void
  
+     .line 206
      :catch_0
      move-exception v0
  
+     .line 207
      .local v0, e:Landroid/content/res/Resources$NotFoundException;
      if-eqz p1, :cond_0
  
+     .line 208
      const-string v3, "PasswordEntryKeyboardHelper"
  
      const-string v4, "Vibrate pattern missing"
***************
*** 996,1001 ****
--- 1135,1141 ----
  
      goto :goto_0
  
+     .line 215
      .end local v0           #e:Landroid/content/res/Resources$NotFoundException;
      :cond_2
      array-length v3, v2
***************
*** 1004,1009 ****
--- 1144,1150 ----
  
      iput-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mVibratePattern:[J
  
+     .line 216
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 1012,1017 ****
--- 1153,1159 ----
  
      if-ge v1, v3, :cond_1
  
+     .line 217
      iget-object v3, p0, Lcom/android/internal/widget/PasswordEntryKeyboardHelper;->mVibratePattern:[J
  
      aget v4, v2, v1
***************
*** 1020,1025 ****
--- 1162,1168 ----
  
      aput-wide v4, v3, v1
  
+     .line 216
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_1
***************
*** 1029,1034 ****
--- 1172,1178 ----
      .locals 0
  
      .prologue
+     .line 301
      return-void
  .end method
  
***************
*** 1036,1041 ****
--- 1180,1186 ----
      .locals 0
  
      .prologue
+     .line 305
      return-void
  .end method
  
***************
*** 1043,1048 ****
--- 1188,1194 ----
      .locals 0
  
      .prologue
+     .line 309
      return-void
  .end method
  
***************
*** 1050,1054 ****
--- 1196,1201 ----
      .locals 0
  
      .prologue
+     .line 313
      return-void
  .end method
