*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/PointerLocationView$PointerState.smali	2012-03-23 19:37:33.592387639 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/PointerLocationView$PointerState.smali.original	2012-03-23 19:34:46.099333407 -0500
***************
*** 41,62 ****
--- 41,67 ----
      .prologue
      const/16 v1, 0x20
  
+     .line 38
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 40
      new-array v0, v1, [F
  
      iput-object v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceX:[F
  
+     .line 41
      new-array v0, v1, [F
  
      iput-object v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceY:[F
  
+     .line 48
      new-instance v0, Landroid/view/MotionEvent$PointerCoords;
  
      invoke-direct {v0}, Landroid/view/MotionEvent$PointerCoords;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mCoords:Landroid/view/MotionEvent$PointerCoords;
  
+     .line 56
      new-instance v0, Landroid/view/VelocityTracker$Estimator;
  
      invoke-direct {v0}, Landroid/view/VelocityTracker$Estimator;-><init>()V
***************
*** 71,76 ****
--- 76,82 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceCount:I
  
      return v0
***************
*** 81,86 ****
--- 87,93 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget-boolean v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mCurDown:Z
  
      return v0
***************
*** 92,97 ****
--- 99,105 ----
      .parameter "x1"
  
      .prologue
+     .line 38
      iput-boolean p1, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mCurDown:Z
  
      return p1
***************
*** 102,107 ****
--- 110,116 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget-object v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mCoords:Landroid/view/MotionEvent$PointerCoords;
  
      return-object v0
***************
*** 112,117 ****
--- 121,127 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget-object v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceX:[F
  
      return-object v0
***************
*** 122,127 ****
--- 132,138 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget-object v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceY:[F
  
      return-object v0
***************
*** 132,137 ****
--- 143,149 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mXVelocity:F
  
      return v0
***************
*** 143,148 ****
--- 155,161 ----
      .parameter "x1"
  
      .prologue
+     .line 38
      iput p1, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mXVelocity:F
  
      return p1
***************
*** 153,158 ****
--- 166,172 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mYVelocity:F
  
      return v0
***************
*** 164,169 ****
--- 178,184 ----
      .parameter "x1"
  
      .prologue
+     .line 38
      iput p1, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mYVelocity:F
  
      return p1
***************
*** 174,179 ****
--- 189,195 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget-object v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mEstimator:Landroid/view/VelocityTracker$Estimator;
  
      return-object v0
***************
*** 184,189 ****
--- 200,206 ----
      .parameter "x0"
  
      .prologue
+     .line 38
      iget v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mToolType:I
  
      return v0
***************
*** 195,200 ****
--- 212,218 ----
      .parameter "x1"
  
      .prologue
+     .line 38
      iput p1, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mToolType:I
  
      return p1
***************
*** 210,228 ****
--- 228,251 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 63
      iget-object v3, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceX:[F
  
      array-length v2, v3
  
+     .line 64
      .local v2, traceCapacity:I
      iget v3, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceCount:I
  
      if-ne v3, v2, :cond_0
  
+     .line 65
      mul-int/lit8 v2, v2, 0x2
  
+     .line 66
      new-array v0, v2, [F
  
+     .line 67
      .local v0, newTraceX:[F
      iget-object v3, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceX:[F
  
***************
*** 230,239 ****
--- 253,265 ----
  
      invoke-static {v3, v5, v0, v5, v4}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 68
      iput-object v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceX:[F
  
+     .line 70
      new-array v1, v2, [F
  
+     .line 71
      .local v1, newTraceY:[F
      iget-object v3, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceY:[F
  
***************
*** 241,248 ****
--- 267,276 ----
  
      invoke-static {v3, v5, v1, v5, v4}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 72
      iput-object v1, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceY:[F
  
+     .line 75
      .end local v0           #newTraceX:[F
      .end local v1           #newTraceY:[F
      :cond_0
***************
*** 252,269 ****
--- 280,300 ----
  
      aput p1, v3, v4
  
+     .line 76
      iget-object v3, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceY:[F
  
      iget v4, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceCount:I
  
      aput p2, v3, v4
  
+     .line 77
      iget v3, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceCount:I
  
      add-int/lit8 v3, v3, 0x1
  
      iput v3, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceCount:I
  
+     .line 78
      return-void
  .end method
  
***************
*** 271,279 ****
--- 302,312 ----
      .locals 1
  
      .prologue
+     .line 59
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/widget/PointerLocationView$PointerState;->mTraceCount:I
  
+     .line 60
      return-void
  .end method
