*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/SlidingTab.smali	2012-03-23 19:37:32.719371720 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/SlidingTab.smali.original	2012-03-23 19:34:46.079333043 -0500
***************
*** 76,85 ****
--- 76,87 ----
      .parameter "context"
  
      .prologue
+     .line 437
      const/4 v0, 0x0
  
      invoke-direct {p0, p1, v0}, Lcom/android/internal/widget/SlidingTab;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 438
      return-void
  .end method
  
***************
*** 95,128 ****
--- 97,139 ----
  
      const/4 v3, 0x0
  
+     .line 444
      invoke-direct {p0, p1, p2}, Landroid/view/ViewGroup;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 64
      iput-boolean v2, p0, Lcom/android/internal/widget/SlidingTab;->mHoldLeftOnTransition:Z
  
+     .line 65
      iput-boolean v2, p0, Lcom/android/internal/widget/SlidingTab;->mHoldRightOnTransition:Z
  
+     .line 68
      iput v3, p0, Lcom/android/internal/widget/SlidingTab;->mGrabbedState:I
  
+     .line 69
      iput-boolean v3, p0, Lcom/android/internal/widget/SlidingTab;->mTriggered:Z
  
+     .line 90
      new-instance v2, Lcom/android/internal/widget/SlidingTab$1;
  
      invoke-direct {v2, p0}, Lcom/android/internal/widget/SlidingTab$1;-><init>(Lcom/android/internal/widget/SlidingTab;)V
  
      iput-object v2, p0, Lcom/android/internal/widget/SlidingTab;->mAnimationDoneListener:Landroid/view/animation/Animation$AnimationListener;
  
+     .line 447
      new-instance v2, Landroid/graphics/Rect;
  
      invoke-direct {v2}, Landroid/graphics/Rect;-><init>()V
  
      iput-object v2, p0, Lcom/android/internal/widget/SlidingTab;->mTmpRect:Landroid/graphics/Rect;
  
+     .line 449
      sget-object v2, Lcom/android/internal/R$styleable;->SlidingTab:[I
  
      invoke-virtual {p1, p2, v2}, Landroid/content/Context;->obtainStyledAttributes(Landroid/util/AttributeSet;[I)Landroid/content/res/TypedArray;
  
      move-result-object v0
  
+     .line 450
      .local v0, a:Landroid/content/res/TypedArray;
      invoke-virtual {v0, v3, v3}, Landroid/content/res/TypedArray;->getInt(II)I
  
***************
*** 130,141 ****
--- 141,155 ----
  
      iput v2, p0, Lcom/android/internal/widget/SlidingTab;->mOrientation:I
  
+     .line 451
      invoke-virtual {v0}, Landroid/content/res/TypedArray;->recycle()V
  
+     .line 453
      invoke-virtual {p0}, Lcom/android/internal/widget/SlidingTab;->getResources()Landroid/content/res/Resources;
  
      move-result-object v1
  
+     .line 454
      .local v1, r:Landroid/content/res/Resources;
      invoke-virtual {v1}, Landroid/content/res/Resources;->getDisplayMetrics()Landroid/util/DisplayMetrics;
  
***************
*** 145,150 ****
--- 159,165 ----
  
      iput v2, p0, Lcom/android/internal/widget/SlidingTab;->mDensity:F
  
+     .line 457
      new-instance v2, Lcom/android/internal/widget/SlidingTab$Slider;
  
      const v3, 0x1080387
***************
*** 155,160 ****
--- 170,176 ----
  
      iput-object v2, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 461
      new-instance v2, Lcom/android/internal/widget/SlidingTab$Slider;
  
      const v3, 0x1080390
***************
*** 165,170 ****
--- 181,187 ----
  
      iput-object v2, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 467
      return-void
  .end method
  
***************
*** 173,178 ****
--- 190,196 ----
      .parameter "x0"
  
      .prologue
+     .line 51
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->onAnimationDone()V
  
      return-void
***************
*** 184,189 ****
--- 202,208 ----
      .parameter "x1"
  
      .prologue
+     .line 51
      iput-boolean p1, p0, Lcom/android/internal/widget/SlidingTab;->mAnimating:Z
  
      return p1
***************
*** 194,199 ****
--- 213,219 ----
      .parameter "x0"
  
      .prologue
+     .line 51
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->resetView()V
  
      return-void
***************
*** 204,209 ****
--- 224,230 ----
      .parameter "x0"
  
      .prologue
+     .line 51
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mAnimationDoneListener:Landroid/view/animation/Animation$AnimationListener;
  
      return-object v0
***************
*** 214,219 ****
--- 235,241 ----
      .parameter "x0"
  
      .prologue
+     .line 51
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      return-object v0
***************
*** 224,229 ****
--- 246,252 ----
      .parameter "x0"
  
      .prologue
+     .line 51
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      return-object v0
***************
*** 237,266 ****
--- 260,298 ----
  
      const/4 v2, 0x0
  
+     .line 622
      iput-boolean v2, p0, Lcom/android/internal/widget/SlidingTab;->mTracking:Z
  
+     .line 623
      iput-boolean v2, p0, Lcom/android/internal/widget/SlidingTab;->mTriggered:Z
  
+     .line 624
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mOtherSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/SlidingTab$Slider;->show(Z)V
  
+     .line 625
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, v2}, Lcom/android/internal/widget/SlidingTab$Slider;->reset(Z)V
  
+     .line 626
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/SlidingTab$Slider;->hideTarget()V
  
+     .line 627
      iput-object v3, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 628
      iput-object v3, p0, Lcom/android/internal/widget/SlidingTab;->mOtherSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 629
      invoke-direct {p0, v2}, Lcom/android/internal/widget/SlidingTab;->setGrabbedState(I)V
  
+     .line 630
      return-void
  .end method
  
***************
*** 269,286 ****
--- 301,322 ----
      .parameter "whichHandle"
  
      .prologue
+     .line 835
      const-wide/16 v0, 0x28
  
      invoke-direct {p0, v0, v1}, Lcom/android/internal/widget/SlidingTab;->vibrate(J)V
  
+     .line 836
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mOnTriggerListener:Lcom/android/internal/widget/SlidingTab$OnTriggerListener;
  
      if-eqz v0, :cond_0
  
+     .line 837
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mOnTriggerListener:Lcom/android/internal/widget/SlidingTab$OnTriggerListener;
  
      invoke-interface {v0, p0, p1}, Lcom/android/internal/widget/SlidingTab$OnTriggerListener;->onTrigger(Landroid/view/View;I)V
  
+     .line 839
      :cond_0
      return-void
  .end method
***************
*** 289,294 ****
--- 325,331 ----
      .locals 1
  
      .prologue
+     .line 712
      iget v0, p0, Lcom/android/internal/widget/SlidingTab;->mOrientation:I
  
      if-nez v0, :cond_0
***************
*** 309,318 ****
--- 346,357 ----
      .parameter "msg"
  
      .prologue
+     .line 866
      const-string v0, "SlidingTab"
  
      invoke-static {v0, p1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 867
      return-void
  .end method
  
***************
*** 322,327 ****
--- 361,367 ----
      .parameter "y"
  
      .prologue
+     .line 731
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      #getter for: Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
***************
*** 329,334 ****
--- 369,375 ----
  
      move-result-object v3
  
+     .line 732
      .local v3, handle:Landroid/view/View;
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
***************
*** 337,342 ****
--- 378,384 ----
  
      move-result-object v0
  
+     .line 733
      .local v0, content:Landroid/view/View;
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
***************
*** 344,349 ****
--- 386,392 ----
  
      if-eqz v4, :cond_0
  
+     .line 734
      float-to-int v4, p1
  
      invoke-virtual {v3}, Landroid/view/View;->getLeft()I
***************
*** 360,376 ****
--- 403,424 ----
  
      sub-int v1, v4, v5
  
+     .line 735
      .local v1, deltaX:I
      invoke-virtual {v3, v1}, Landroid/view/View;->offsetLeftAndRight(I)V
  
+     .line 736
      invoke-virtual {v0, v1}, Landroid/view/View;->offsetLeftAndRight(I)V
  
+     .line 742
      .end local v1           #deltaX:I
      :goto_0
      invoke-virtual {p0}, Lcom/android/internal/widget/SlidingTab;->invalidate()V
  
+     .line 743
      return-void
  
+     .line 738
      :cond_0
      float-to-int v4, p2
  
***************
*** 388,396 ****
--- 436,446 ----
  
      sub-int v2, v4, v5
  
+     .line 739
      .local v2, deltaY:I
      invoke-virtual {v3, v2}, Landroid/view/View;->offsetTopAndBottom(I)V
  
+     .line 740
      invoke-virtual {v0, v2}, Landroid/view/View;->offsetTopAndBottom(I)V
  
      goto :goto_0
***************
*** 400,411 ****
--- 450,464 ----
      .locals 1
  
      .prologue
+     .line 702
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->resetView()V
  
+     .line 703
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/internal/widget/SlidingTab;->mAnimating:Z
  
+     .line 704
      return-void
  .end method
  
***************
*** 415,428 ****
--- 468,484 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 716
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/SlidingTab$Slider;->reset(Z)V
  
+     .line 717
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, v1}, Lcom/android/internal/widget/SlidingTab$Slider;->reset(Z)V
  
+     .line 719
      return-void
  .end method
  
***************
*** 431,452 ****
--- 487,513 ----
      .parameter "newState"
  
      .prologue
+     .line 857
      iget v0, p0, Lcom/android/internal/widget/SlidingTab;->mGrabbedState:I
  
      if-eq p1, v0, :cond_0
  
+     .line 858
      iput p1, p0, Lcom/android/internal/widget/SlidingTab;->mGrabbedState:I
  
+     .line 859
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mOnTriggerListener:Lcom/android/internal/widget/SlidingTab$OnTriggerListener;
  
      if-eqz v0, :cond_0
  
+     .line 860
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mOnTriggerListener:Lcom/android/internal/widget/SlidingTab$OnTriggerListener;
  
      iget v1, p0, Lcom/android/internal/widget/SlidingTab;->mGrabbedState:I
  
      invoke-interface {v0, p0, v1}, Lcom/android/internal/widget/SlidingTab$OnTriggerListener;->onGrabbedStateChange(Landroid/view/View;I)V
  
+     .line 863
      :cond_0
      return-void
  .end method
***************
*** 456,461 ****
--- 517,523 ----
      .parameter "duration"
  
      .prologue
+     .line 814
      monitor-enter p0
  
      :try_start_0
***************
*** 463,473 ****
  
      if-nez v0, :cond_0
  
      invoke-virtual {p0}, Lcom/android/internal/widget/SlidingTab;->getContext()Landroid/content/Context;
  
      move-result-object v0
  
!     const-string v1, "vibrator"
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
--- 525,536 ----
  
      if-nez v0, :cond_0
  
+     .line 815
      invoke-virtual {p0}, Lcom/android/internal/widget/SlidingTab;->getContext()Landroid/content/Context;
  
      move-result-object v0
  
!     const-string/jumbo v1, "vibrator"
  
      invoke-virtual {v0, v1}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
  
***************
*** 477,482 ****
--- 540,546 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mVibrator:Landroid/os/Vibrator;
  
+     .line 818
      :cond_0
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mVibrator:Landroid/os/Vibrator;
  
***************
*** 484,493 ****
--- 548,559 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 819
      monitor-exit p0
  
      return-void
  
+     .line 814
      :catchall_0
      move-exception v0
  
***************
*** 505,510 ****
--- 571,577 ----
      .prologue
      const/high16 v1, -0x3db8
  
+     .line 707
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
      move-result v0
***************
*** 577,596 ****
--- 644,667 ----
  
      const/4 v10, 0x1
  
+     .line 506
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getAction()I
  
      move-result v0
  
+     .line 507
      .local v0, action:I
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getX()F
  
      move-result v5
  
+     .line 508
      .local v5, x:F
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getY()F
  
      move-result v6
  
+     .line 510
      .local v6, y:F
      iget-boolean v11, p0, Lcom/android/internal/widget/SlidingTab;->mAnimating:Z
  
***************
*** 598,606 ****
--- 669,679 ----
  
      move v7, v9
  
+     .line 549
      :goto_0
      return v7
  
+     .line 514
      :cond_0
      iget-object v11, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
***************
*** 609,619 ****
--- 682,694 ----
  
      move-result-object v1
  
+     .line 515
      .local v1, leftHandle:Landroid/view/View;
      iget-object v11, p0, Lcom/android/internal/widget/SlidingTab;->mTmpRect:Landroid/graphics/Rect;
  
      invoke-virtual {v1, v11}, Landroid/view/View;->getHitRect(Landroid/graphics/Rect;)V
  
+     .line 516
      iget-object v11, p0, Lcom/android/internal/widget/SlidingTab;->mTmpRect:Landroid/graphics/Rect;
  
      float-to-int v12, v5
***************
*** 624,629 ****
--- 699,705 ----
  
      move-result v2
  
+     .line 518
      .local v2, leftHit:Z
      iget-object v11, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
***************
*** 632,642 ****
--- 708,720 ----
  
      move-result-object v3
  
+     .line 519
      .local v3, rightHandle:Landroid/view/View;
      iget-object v11, p0, Lcom/android/internal/widget/SlidingTab;->mTmpRect:Landroid/graphics/Rect;
  
      invoke-virtual {v3, v11}, Landroid/view/View;->getHitRect(Landroid/graphics/Rect;)V
  
+     .line 520
      iget-object v11, p0, Lcom/android/internal/widget/SlidingTab;->mTmpRect:Landroid/graphics/Rect;
  
      float-to-int v12, v5
***************
*** 647,652 ****
--- 725,731 ----
  
      move-result v4
  
+     .line 522
      .local v4, rightHit:Z
      iget-boolean v11, p0, Lcom/android/internal/widget/SlidingTab;->mTracking:Z
  
***************
*** 658,692 ****
--- 737,781 ----
  
      move v7, v9
  
+     .line 523
      goto :goto_0
  
+     .line 526
      :cond_1
      packed-switch v0, :pswitch_data_0
  
      :goto_1
      move v7, v10
  
+     .line 549
      goto :goto_0
  
+     .line 528
      :pswitch_0
      iput-boolean v10, p0, Lcom/android/internal/widget/SlidingTab;->mTracking:Z
  
+     .line 529
      iput-boolean v9, p0, Lcom/android/internal/widget/SlidingTab;->mTriggered:Z
  
+     .line 530
      const-wide/16 v11, 0x1e
  
      invoke-direct {p0, v11, v12}, Lcom/android/internal/widget/SlidingTab;->vibrate(J)V
  
+     .line 531
      if-eqz v2, :cond_3
  
+     .line 532
      iget-object v9, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      iput-object v9, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 533
      iget-object v9, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      iput-object v9, p0, Lcom/android/internal/widget/SlidingTab;->mOtherSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 534
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
      move-result v9
***************
*** 696,712 ****
--- 785,805 ----
      :goto_2
      iput v7, p0, Lcom/android/internal/widget/SlidingTab;->mThreshold:F
  
+     .line 535
      invoke-direct {p0, v10}, Lcom/android/internal/widget/SlidingTab;->setGrabbedState(I)V
  
+     .line 542
      :goto_3
      iget-object v7, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v7, v10}, Lcom/android/internal/widget/SlidingTab$Slider;->setState(I)V
  
+     .line 543
      iget-object v7, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v7}, Lcom/android/internal/widget/SlidingTab$Slider;->showTarget()V
  
+     .line 544
      iget-object v7, p0, Lcom/android/internal/widget/SlidingTab;->mOtherSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v7}, Lcom/android/internal/widget/SlidingTab$Slider;->hide()V
***************
*** 716,732 ****
--- 809,829 ----
      :cond_2
      move v7, v8
  
+     .line 534
      goto :goto_2
  
+     .line 537
      :cond_3
      iget-object v9, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      iput-object v9, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 538
      iget-object v9, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      iput-object v9, p0, Lcom/android/internal/widget/SlidingTab;->mOtherSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 539
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
      move-result v9
***************
*** 736,741 ****
--- 833,839 ----
      :goto_4
      iput v8, p0, Lcom/android/internal/widget/SlidingTab;->mThreshold:F
  
+     .line 540
      const/4 v7, 0x2
  
      invoke-direct {p0, v7}, Lcom/android/internal/widget/SlidingTab;->setGrabbedState(I)V
***************
*** 745,752 ****
--- 843,852 ----
      :cond_4
      move v8, v7
  
+     .line 539
      goto :goto_4
  
+     .line 526
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
***************
*** 762,772 ****
--- 862,875 ----
      .parameter "b"
  
      .prologue
+     .line 723
      if-nez p1, :cond_0
  
+     .line 728
      :goto_0
      return-void
  
+     .line 726
      :cond_0
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
***************
*** 789,794 ****
--- 892,898 ----
  
      invoke-virtual/range {v0 .. v5}, Lcom/android/internal/widget/SlidingTab$Slider;->layout(IIIII)V
  
+     .line 727
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
***************
*** 812,822 ****
--- 916,928 ----
  
      goto :goto_0
  
+     .line 726
      :cond_1
      const/4 v5, 0x3
  
      goto :goto_1
  
+     .line 727
      :cond_2
      const/4 v5, 0x2
  
***************
*** 829,868 ****
--- 935,982 ----
      .parameter "heightMeasureSpec"
  
      .prologue
+     .line 471
      invoke-static {p1}, Landroid/view/View$MeasureSpec;->getMode(I)I
  
      move-result v8
  
+     .line 472
      .local v8, widthSpecMode:I
      invoke-static {p1}, Landroid/view/View$MeasureSpec;->getSize(I)I
  
      move-result v9
  
+     .line 474
      .local v9, widthSpecSize:I
      invoke-static {p2}, Landroid/view/View$MeasureSpec;->getMode(I)I
  
      move-result v1
  
+     .line 475
      .local v1, heightSpecMode:I
      invoke-static {p2}, Landroid/view/View$MeasureSpec;->getSize(I)I
  
      move-result v2
  
+     .line 486
      .local v2, heightSpecSize:I
      iget-object v10, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v10}, Lcom/android/internal/widget/SlidingTab$Slider;->measure()V
  
+     .line 487
      iget-object v10, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v10}, Lcom/android/internal/widget/SlidingTab$Slider;->measure()V
  
+     .line 488
      iget-object v10, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v10}, Lcom/android/internal/widget/SlidingTab$Slider;->getTabWidth()I
  
      move-result v4
  
+     .line 489
      .local v4, leftTabWidth:I
      iget-object v10, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
***************
*** 870,875 ****
--- 984,990 ----
  
      move-result v6
  
+     .line 490
      .local v6, rightTabWidth:I
      iget-object v10, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
***************
*** 877,882 ****
--- 992,998 ----
  
      move-result v3
  
+     .line 491
      .local v3, leftTabHeight:I
      iget-object v10, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
***************
*** 884,889 ****
--- 1000,1006 ----
  
      move-result v5
  
+     .line 494
      .local v5, rightTabHeight:I
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
***************
*** 891,913 ****
--- 1008,1035 ----
  
      if-eqz v10, :cond_0
  
+     .line 495
      add-int v10, v4, v6
  
      invoke-static {v9, v10}, Ljava/lang/Math;->max(II)I
  
      move-result v7
  
+     .line 496
      .local v7, width:I
      invoke-static {v3, v5}, Ljava/lang/Math;->max(II)I
  
      move-result v0
  
+     .line 501
      .local v0, height:I
      :goto_0
      invoke-virtual {p0, v7, v0}, Lcom/android/internal/widget/SlidingTab;->setMeasuredDimension(II)V
  
+     .line 502
      return-void
  
+     .line 498
      .end local v0           #height:I
      .end local v7           #width:I
      :cond_0
***************
*** 915,920 ****
--- 1037,1043 ----
  
      move-result v7
  
+     .line 499
      .restart local v7       #width:I
      add-int v10, v3, v5
  
***************
*** 937,963 ****
--- 1060,1092 ----
  
      const/4 v9, 0x0
  
+     .line 577
      iget-boolean v7, p0, Lcom/android/internal/widget/SlidingTab;->mTracking:Z
  
      if-eqz v7, :cond_0
  
+     .line 578
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getAction()I
  
      move-result v0
  
+     .line 579
      .local v0, action:I
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getX()F
  
      move-result v5
  
+     .line 580
      .local v5, x:F
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getY()F
  
      move-result v6
  
+     .line 582
      .local v6, y:F
      packed-switch v0, :pswitch_data_0
  
+     .line 618
      .end local v0           #action:I
      .end local v5           #x:F
      .end local v6           #y:F
***************
*** 979,984 ****
--- 1108,1114 ----
      :cond_2
      return v9
  
+     .line 584
      .restart local v0       #action:I
      .restart local v5       #x:F
      .restart local v6       #y:F
***************
*** 989,996 ****
--- 1119,1128 ----
  
      if-eqz v7, :cond_f
  
+     .line 585
      invoke-direct {p0, v5, v6}, Lcom/android/internal/widget/SlidingTab;->moveHandle(FF)V
  
+     .line 586
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
      move-result v7
***************
*** 999,1004 ****
--- 1131,1137 ----
  
      move v2, v5
  
+     .line 587
      .local v2, position:F
      :goto_1
      iget v11, p0, Lcom/android/internal/widget/SlidingTab;->mThreshold:F
***************
*** 1018,1023 ****
--- 1151,1157 ----
  
      mul-float v3, v11, v7
  
+     .line 589
      .local v3, target:F
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
***************
*** 1025,1030 ****
--- 1159,1165 ----
  
      if-eqz v7, :cond_8
  
+     .line 590
      iget-object v7, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      iget-object v11, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
***************
*** 1037,1042 ****
--- 1172,1178 ----
  
      move v4, v8
  
+     .line 596
      .local v4, thresholdReached:Z
      :goto_3
      iget-boolean v7, p0, Lcom/android/internal/widget/SlidingTab;->mTriggered:Z
***************
*** 1045,1058 ****
--- 1181,1198 ----
  
      if-eqz v4, :cond_0
  
+     .line 597
      iput-boolean v8, p0, Lcom/android/internal/widget/SlidingTab;->mTriggered:Z
  
+     .line 598
      iput-boolean v9, p0, Lcom/android/internal/widget/SlidingTab;->mTracking:Z
  
+     .line 599
      iget-object v7, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v7, v10}, Lcom/android/internal/widget/SlidingTab$Slider;->setState(I)V
  
+     .line 600
      iget-object v7, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      iget-object v11, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
***************
*** 1061,1066 ****
--- 1201,1207 ----
  
      move v1, v8
  
+     .line 601
      .local v1, isLeft:Z
      :goto_4
      if-eqz v1, :cond_d
***************
*** 1070,1075 ****
--- 1211,1217 ----
      :goto_5
      invoke-direct {p0, v7}, Lcom/android/internal/widget/SlidingTab;->dispatchTriggerEvent(I)V
  
+     .line 604
      if-eqz v1, :cond_e
  
      iget-boolean v7, p0, Lcom/android/internal/widget/SlidingTab;->mHoldLeftOnTransition:Z
***************
*** 1077,1082 ****
--- 1219,1225 ----
      :goto_6
      invoke-virtual {p0, v7}, Lcom/android/internal/widget/SlidingTab;->startAnimating(Z)V
  
+     .line 605
      invoke-direct {p0, v9}, Lcom/android/internal/widget/SlidingTab;->setGrabbedState(I)V
  
      goto :goto_0
***************
*** 1088,1095 ****
--- 1231,1240 ----
      :cond_3
      move v2, v6
  
+     .line 586
      goto :goto_1
  
+     .line 587
      .restart local v2       #position:F
      :cond_4
      invoke-virtual {p0}, Lcom/android/internal/widget/SlidingTab;->getHeight()I
***************
*** 1102,1107 ****
--- 1247,1253 ----
      :cond_5
      move v4, v9
  
+     .line 590
      goto :goto_3
  
      :cond_6
***************
*** 1118,1123 ****
--- 1264,1270 ----
  
      goto :goto_3
  
+     .line 593
      :cond_8
      iget-object v7, p0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
***************
*** 1159,1177 ****
--- 1306,1328 ----
      :cond_c
      move v1, v9
  
+     .line 600
      goto :goto_4
  
      .restart local v1       #isLeft:Z
      :cond_d
      move v7, v10
  
+     .line 601
      goto :goto_5
  
+     .line 604
      :cond_e
      iget-boolean v7, p0, Lcom/android/internal/widget/SlidingTab;->mHoldRightOnTransition:Z
  
      goto :goto_6
  
+     .line 613
      .end local v1           #isLeft:Z
      .end local v2           #position:F
      .end local v3           #target:F
***************
*** 1182,1187 ****
--- 1333,1339 ----
  
      goto/16 :goto_0
  
+     .line 582
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_1
***************
*** 1196,1203 ****
--- 1348,1357 ----
      .parameter "visibility"
  
      .prologue
+     .line 843
      invoke-super {p0, p1, p2}, Landroid/view/ViewGroup;->onVisibilityChanged(Landroid/view/View;I)V
  
+     .line 846
      if-ne p1, p0, :cond_0
  
      if-eqz p2, :cond_0
***************
*** 1206,1213 ****
--- 1360,1369 ----
  
      if-eqz v0, :cond_0
  
+     .line 848
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->cancelGrab()V
  
+     .line 850
      :cond_0
      return-void
  .end method
***************
*** 1217,1236 ****
--- 1373,1397 ----
      .parameter "animate"
  
      .prologue
+     .line 559
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/widget/SlidingTab$Slider;->reset(Z)V
  
+     .line 560
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/widget/SlidingTab$Slider;->reset(Z)V
  
+     .line 561
      if-nez p1, :cond_0
  
+     .line 562
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/internal/widget/SlidingTab;->mAnimating:Z
  
+     .line 564
      :cond_0
      return-void
  .end method
***************
*** 1241,1250 ****
--- 1402,1414 ----
      .parameter "holdRight"
  
      .prologue
+     .line 806
      iput-boolean p1, p0, Lcom/android/internal/widget/SlidingTab;->mHoldLeftOnTransition:Z
  
+     .line 807
      iput-boolean p2, p0, Lcom/android/internal/widget/SlidingTab;->mHoldRightOnTransition:Z
  
+     .line 808
      return-void
  .end method
  
***************
*** 1253,1268 ****
--- 1417,1435 ----
      .parameter "resId"
  
      .prologue
+     .line 770
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 771
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/widget/SlidingTab$Slider;->setHintText(I)V
  
+     .line 773
      :cond_0
      return-void
  .end method
***************
*** 1275,1300 ****
--- 1442,1473 ----
      .parameter "tabId"
  
      .prologue
+     .line 757
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/widget/SlidingTab$Slider;->setIcon(I)V
  
+     .line 758
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p2}, Lcom/android/internal/widget/SlidingTab$Slider;->setTarget(I)V
  
+     .line 759
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p3}, Lcom/android/internal/widget/SlidingTab$Slider;->setBarBackgroundResource(I)V
  
+     .line 760
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p4}, Lcom/android/internal/widget/SlidingTab$Slider;->setTabBackgroundResource(I)V
  
+     .line 761
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/SlidingTab$Slider;->updateDrawableStates()V
  
+     .line 762
      return-void
  .end method
  
***************
*** 1303,1310 ****
--- 1476,1485 ----
      .parameter "listener"
  
      .prologue
+     .line 827
      iput-object p1, p0, Lcom/android/internal/widget/SlidingTab;->mOnTriggerListener:Lcom/android/internal/widget/SlidingTab$OnTriggerListener;
  
+     .line 828
      return-void
  .end method
  
***************
*** 1313,1328 ****
--- 1488,1506 ----
      .parameter "resId"
  
      .prologue
+     .line 800
      invoke-direct {p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 801
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/widget/SlidingTab$Slider;->setHintText(I)V
  
+     .line 803
      :cond_0
      return-void
  .end method
***************
*** 1335,1360 ****
--- 1513,1544 ----
      .parameter "tabId"
  
      .prologue
+     .line 787
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p1}, Lcom/android/internal/widget/SlidingTab$Slider;->setIcon(I)V
  
+     .line 788
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p2}, Lcom/android/internal/widget/SlidingTab$Slider;->setTarget(I)V
  
+     .line 789
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p3}, Lcom/android/internal/widget/SlidingTab$Slider;->setBarBackgroundResource(I)V
  
+     .line 790
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0, p4}, Lcom/android/internal/widget/SlidingTab$Slider;->setTabBackgroundResource(I)V
  
+     .line 791
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
      invoke-virtual {v0}, Lcom/android/internal/widget/SlidingTab$Slider;->updateDrawableStates()V
  
+     .line 792
      return-void
  .end method
  
***************
*** 1363,1368 ****
--- 1547,1553 ----
      .parameter "visibility"
  
      .prologue
+     .line 569
      invoke-virtual {p0}, Lcom/android/internal/widget/SlidingTab;->getVisibility()I
  
      move-result v0
***************
*** 1373,1385 ****
--- 1558,1573 ----
  
      if-ne p1, v0, :cond_0
  
+     .line 570
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Lcom/android/internal/widget/SlidingTab;->reset(Z)V
  
+     .line 572
      :cond_0
      invoke-super {p0, p1}, Landroid/view/ViewGroup;->setVisibility(I)V
  
+     .line 573
      return-void
  .end method
  
***************
*** 1388,1393 ****
--- 1576,1582 ----
      .parameter "holdAfter"
  
      .prologue
+     .line 633
      const/16 v19, 0x1
  
      move/from16 v0, v19
***************
*** 1396,1410 ****
--- 1585,1602 ----
  
      iput-boolean v0, v1, Lcom/android/internal/widget/SlidingTab;->mAnimating:Z
  
+     .line 636
      move-object/from16 v0, p0
  
      iget-object v12, v0, Lcom/android/internal/widget/SlidingTab;->mCurrentSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 637
      .local v12, slider:Lcom/android/internal/widget/SlidingTab$Slider;
      move-object/from16 v0, p0
  
      iget-object v10, v0, Lcom/android/internal/widget/SlidingTab;->mOtherSlider:Lcom/android/internal/widget/SlidingTab$Slider;
  
+     .line 640
      .local v10, other:Lcom/android/internal/widget/SlidingTab$Slider;
      invoke-direct/range {p0 .. p0}, Lcom/android/internal/widget/SlidingTab;->isHorizontal()Z
  
***************
*** 1412,1417 ****
--- 1604,1610 ----
  
      if-eqz v19, :cond_2
  
+     .line 641
      #getter for: Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
      invoke-static {v12}, Lcom/android/internal/widget/SlidingTab$Slider;->access$100(Lcom/android/internal/widget/SlidingTab$Slider;)Landroid/widget/ImageView;
  
***************
*** 1421,1426 ****
--- 1614,1620 ----
  
      move-result v11
  
+     .line 642
      .local v11, right:I
      #getter for: Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
      invoke-static {v12}, Lcom/android/internal/widget/SlidingTab$Slider;->access$100(Lcom/android/internal/widget/SlidingTab$Slider;)Landroid/widget/ImageView;
***************
*** 1431,1436 ****
--- 1625,1631 ----
  
      move-result v18
  
+     .line 643
      .local v18, width:I
      #getter for: Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
      invoke-static {v12}, Lcom/android/internal/widget/SlidingTab$Slider;->access$100(Lcom/android/internal/widget/SlidingTab$Slider;)Landroid/widget/ImageView;
***************
*** 1441,1456 ****
--- 1636,1654 ----
  
      move-result v9
  
+     .line 644
      .local v9, left:I
      invoke-virtual/range {p0 .. p0}, Lcom/android/internal/widget/SlidingTab;->getWidth()I
  
      move-result v17
  
+     .line 645
      .local v17, viewWidth:I
      if-eqz p1, :cond_0
  
      const/4 v8, 0x0
  
+     .line 646
      .local v8, holdOffset:I
      :goto_0
      move-object/from16 v0, p0
***************
*** 1471,1480 ****
--- 1669,1680 ----
  
      neg-int v5, v0
  
+     .line 648
      .local v5, dx:I
      :goto_1
      const/4 v6, 0x0
  
+     .line 659
      .end local v9           #left:I
      .end local v11           #right:I
      .end local v17           #viewWidth:I
***************
*** 1505,1510 ****
--- 1705,1711 ----
  
      invoke-direct {v14, v0, v1, v2, v3}, Landroid/view/animation/TranslateAnimation;-><init>(FFFF)V
  
+     .line 660
      .local v14, trans1:Landroid/view/animation/Animation;
      const-wide/16 v19, 0xfa
  
***************
*** 1512,1517 ****
--- 1713,1719 ----
  
      invoke-virtual {v14, v0, v1}, Landroid/view/animation/Animation;->setDuration(J)V
  
+     .line 661
      new-instance v19, Landroid/view/animation/LinearInterpolator;
  
      invoke-direct/range {v19 .. v19}, Landroid/view/animation/LinearInterpolator;-><init>()V
***************
*** 1520,1531 ****
--- 1722,1735 ----
  
      invoke-virtual {v14, v0}, Landroid/view/animation/Animation;->setInterpolator(Landroid/view/animation/Interpolator;)V
  
+     .line 662
      const/16 v19, 0x1
  
      move/from16 v0, v19
  
      invoke-virtual {v14, v0}, Landroid/view/animation/Animation;->setFillAfter(Z)V
  
+     .line 663
      new-instance v15, Landroid/view/animation/TranslateAnimation;
  
      const/16 v19, 0x0
***************
*** 1550,1555 ****
--- 1754,1760 ----
  
      invoke-direct {v15, v0, v1, v2, v3}, Landroid/view/animation/TranslateAnimation;-><init>(FFFF)V
  
+     .line 664
      .local v15, trans2:Landroid/view/animation/Animation;
      const-wide/16 v19, 0xfa
  
***************
*** 1557,1562 ****
--- 1762,1768 ----
  
      invoke-virtual {v15, v0, v1}, Landroid/view/animation/Animation;->setDuration(J)V
  
+     .line 665
      new-instance v19, Landroid/view/animation/LinearInterpolator;
  
      invoke-direct/range {v19 .. v19}, Landroid/view/animation/LinearInterpolator;-><init>()V
***************
*** 1565,1576 ****
--- 1771,1784 ----
  
      invoke-virtual {v15, v0}, Landroid/view/animation/Animation;->setInterpolator(Landroid/view/animation/Interpolator;)V
  
+     .line 666
      const/16 v19, 0x1
  
      move/from16 v0, v19
  
      invoke-virtual {v15, v0}, Landroid/view/animation/Animation;->setFillAfter(Z)V
  
+     .line 668
      new-instance v19, Lcom/android/internal/widget/SlidingTab$2;
  
      move-object/from16 v0, v19
***************
*** 1585,1594 ****
--- 1793,1805 ----
  
      invoke-virtual {v14, v0}, Landroid/view/animation/Animation;->setAnimationListener(Landroid/view/animation/Animation$AnimationListener;)V
  
+     .line 697
      invoke-virtual {v12}, Lcom/android/internal/widget/SlidingTab$Slider;->hideTarget()V
  
+     .line 698
      invoke-virtual {v12, v14, v15}, Lcom/android/internal/widget/SlidingTab$Slider;->startAnimation(Landroid/view/animation/Animation;Landroid/view/animation/Animation;)V
  
+     .line 699
      return-void
  
      .end local v5           #dx:I
***************
*** 1603,1610 ****
--- 1814,1823 ----
      :cond_0
      move/from16 v8, v18
  
+     .line 645
      goto/16 :goto_0
  
+     .line 646
      .restart local v8       #holdOffset:I
      :cond_1
      sub-int v19, v17, v9
***************
*** 1615,1620 ****
--- 1828,1834 ----
  
      goto :goto_1
  
+     .line 650
      .end local v8           #holdOffset:I
      .end local v9           #left:I
      .end local v11           #right:I
***************
*** 1630,1635 ****
--- 1844,1850 ----
  
      move-result v13
  
+     .line 651
      .local v13, top:I
      #getter for: Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
      invoke-static {v12}, Lcom/android/internal/widget/SlidingTab$Slider;->access$100(Lcom/android/internal/widget/SlidingTab$Slider;)Landroid/widget/ImageView;
***************
*** 1640,1645 ****
--- 1855,1861 ----
  
      move-result v4
  
+     .line 652
      .local v4, bottom:I
      #getter for: Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
      invoke-static {v12}, Lcom/android/internal/widget/SlidingTab$Slider;->access$100(Lcom/android/internal/widget/SlidingTab$Slider;)Landroid/widget/ImageView;
***************
*** 1650,1669 ****
--- 1866,1889 ----
  
      move-result v7
  
+     .line 653
      .local v7, height:I
      invoke-virtual/range {p0 .. p0}, Lcom/android/internal/widget/SlidingTab;->getHeight()I
  
      move-result v16
  
+     .line 654
      .local v16, viewHeight:I
      if-eqz p1, :cond_3
  
      const/4 v8, 0x0
  
+     .line 655
      .restart local v8       #holdOffset:I
      :goto_3
      const/4 v5, 0x0
  
+     .line 656
      .restart local v5       #dx:I
      move-object/from16 v0, p0
  
***************
*** 1689,1696 ****
--- 1909,1918 ----
      :cond_3
      move v8, v7
  
+     .line 654
      goto :goto_3
  
+     .line 656
      .restart local v5       #dx:I
      .restart local v8       #holdOffset:I
      :cond_4
