*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/SlidingTab$Slider.smali	2012-03-23 19:37:32.550368638 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/SlidingTab$Slider.smali.original	2012-03-23 19:34:46.076332987 -0500
***************
*** 59,72 ****
--- 59,76 ----
  
      const/4 v3, -0x2
  
+     .line 188
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 176
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->currentState:I
  
+     .line 177
      iput v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->alignment:I
  
+     .line 190
      new-instance v0, Landroid/widget/ImageView;
  
      invoke-virtual {p1}, Landroid/view/ViewGroup;->getContext()Landroid/content/Context;
***************
*** 77,92 ****
--- 81,99 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
+     .line 191
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v0, p2}, Landroid/widget/ImageView;->setBackgroundResource(I)V
  
+     .line 192
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      sget-object v1, Landroid/widget/ImageView$ScaleType;->CENTER:Landroid/widget/ImageView$ScaleType;
  
      invoke-virtual {v0, v1}, Landroid/widget/ImageView;->setScaleType(Landroid/widget/ImageView$ScaleType;)V
  
+     .line 193
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      new-instance v1, Landroid/view/ViewGroup$LayoutParams;
***************
*** 95,100 ****
--- 102,108 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/ImageView;->setLayoutParams(Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 197
      new-instance v0, Landroid/widget/TextView;
  
      invoke-virtual {p1}, Landroid/view/ViewGroup;->getContext()Landroid/content/Context;
***************
*** 105,110 ****
--- 113,119 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
+     .line 198
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      new-instance v1, Landroid/view/ViewGroup$LayoutParams;
***************
*** 115,124 ****
--- 124,135 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/TextView;->setLayoutParams(Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 200
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v0, p3}, Landroid/widget/TextView;->setBackgroundResource(I)V
  
+     .line 201
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {p1}, Landroid/view/ViewGroup;->getContext()Landroid/content/Context;
***************
*** 129,134 ****
--- 140,146 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/widget/TextView;->setTextAppearance(Landroid/content/Context;I)V
  
+     .line 205
      new-instance v0, Landroid/widget/ImageView;
  
      invoke-virtual {p1}, Landroid/view/ViewGroup;->getContext()Landroid/content/Context;
***************
*** 139,154 ****
--- 151,169 ----
  
      iput-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
+     .line 206
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      invoke-virtual {v0, p4}, Landroid/widget/ImageView;->setImageResource(I)V
  
+     .line 207
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      sget-object v1, Landroid/widget/ImageView$ScaleType;->CENTER:Landroid/widget/ImageView$ScaleType;
  
      invoke-virtual {v0, v1}, Landroid/widget/ImageView;->setScaleType(Landroid/widget/ImageView$ScaleType;)V
  
+     .line 208
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      new-instance v1, Landroid/view/ViewGroup$LayoutParams;
***************
*** 157,178 ****
--- 172,198 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/ImageView;->setLayoutParams(Landroid/view/ViewGroup$LayoutParams;)V
  
+     .line 210
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      invoke-virtual {v0, v4}, Landroid/widget/ImageView;->setVisibility(I)V
  
+     .line 212
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      invoke-virtual {p1, v0}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
  
+     .line 213
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {p1, v0}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
  
+     .line 214
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {p1, v0}, Landroid/view/ViewGroup;->addView(Landroid/view/View;)V
  
+     .line 215
      return-void
  .end method
  
***************
*** 181,186 ****
--- 201,207 ----
      .parameter "x0"
  
      .prologue
+     .line 156
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      return-object v0
***************
*** 191,196 ****
--- 212,218 ----
      .parameter "x0"
  
      .prologue
+     .line 156
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      return-object v0
***************
*** 202,207 ****
--- 224,230 ----
      .locals 1
  
      .prologue
+     .line 415
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v0}, Landroid/widget/ImageView;->getMeasuredHeight()I
***************
*** 215,220 ****
--- 238,244 ----
      .locals 1
  
      .prologue
+     .line 407
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v0}, Landroid/widget/ImageView;->getMeasuredWidth()I
***************
*** 234,239 ****
--- 258,264 ----
  
      const/4 v1, 0x0
  
+     .line 234
      iget v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->alignment:I
  
      if-eqz v5, :cond_0
***************
*** 245,250 ****
--- 270,276 ----
      :cond_0
      move v2, v4
  
+     .line 235
      .local v2, horiz:Z
      :goto_0
      if-eqz v2, :cond_3
***************
*** 263,272 ****
--- 289,300 ----
  
      sub-int v0, v5, v6
  
+     .line 237
      .local v0, dx:I
      :goto_1
      if-eqz v2, :cond_4
  
+     .line 240
      .local v1, dy:I
      :goto_2
      new-instance v3, Landroid/view/animation/TranslateAnimation;
***************
*** 277,303 ****
--- 305,337 ----
  
      invoke-direct {v3, v7, v5, v7, v6}, Landroid/view/animation/TranslateAnimation;-><init>(FFFF)V
  
+     .line 241
      .local v3, trans:Landroid/view/animation/Animation;
      const-wide/16 v5, 0xfa
  
      invoke-virtual {v3, v5, v6}, Landroid/view/animation/Animation;->setDuration(J)V
  
+     .line 242
      invoke-virtual {v3, v4}, Landroid/view/animation/Animation;->setFillAfter(Z)V
  
+     .line 243
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v4, v3}, Landroid/widget/ImageView;->startAnimation(Landroid/view/animation/Animation;)V
  
+     .line 244
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v4, v3}, Landroid/widget/TextView;->startAnimation(Landroid/view/animation/Animation;)V
  
+     .line 245
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      const/4 v5, 0x4
  
      invoke-virtual {v4, v5}, Landroid/widget/ImageView;->setVisibility(I)V
  
+     .line 246
      return-void
  
      .end local v0           #dx:I
***************
*** 307,314 ****
--- 341,350 ----
      :cond_1
      move v2, v1
  
+     .line 234
      goto :goto_0
  
+     .line 235
      .restart local v2       #horiz:Z
      :cond_2
      iget v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->alignment_value:I
***************
*** 328,333 ****
--- 364,370 ----
  
      goto :goto_1
  
+     .line 237
      .restart local v0       #dx:I
      :cond_4
      iget v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->alignment:I
***************
*** 366,381 ****
--- 403,421 ----
      .locals 2
  
      .prologue
+     .line 431
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      invoke-virtual {v0}, Landroid/widget/ImageView;->clearAnimation()V
  
+     .line 432
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      const/4 v1, 0x4
  
      invoke-virtual {v0, v1}, Landroid/widget/ImageView;->setVisibility(I)V
  
+     .line 433
      return-void
  .end method
  
***************
*** 388,399 ****
--- 428,441 ----
      .parameter "alignment"
  
      .prologue
+     .line 334
      move/from16 v0, p5
  
      move-object/from16 v1, p0
  
      iput v0, v1, Lcom/android/internal/widget/SlidingTab$Slider;->alignment:I
  
+     .line 335
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
***************
*** 404,419 ****
--- 446,464 ----
  
      move-result-object v14
  
+     .line 336
      .local v14, tabBackground:Landroid/graphics/drawable/Drawable;
      invoke-virtual {v14}, Landroid/graphics/drawable/Drawable;->getIntrinsicWidth()I
  
      move-result v7
  
+     .line 337
      .local v7, handleWidth:I
      invoke-virtual {v14}, Landroid/graphics/drawable/Drawable;->getIntrinsicHeight()I
  
      move-result v6
  
+     .line 338
      .local v6, handleHeight:I
      move-object/from16 v0, p0
  
***************
*** 425,446 ****
--- 470,496 ----
  
      move-result-object v16
  
+     .line 339
      .local v16, targetDrawable:Landroid/graphics/drawable/Drawable;
      invoke-virtual/range {v16 .. v16}, Landroid/graphics/drawable/Drawable;->getIntrinsicWidth()I
  
      move-result v21
  
+     .line 340
      .local v21, targetWidth:I
      invoke-virtual/range {v16 .. v16}, Landroid/graphics/drawable/Drawable;->getIntrinsicHeight()I
  
      move-result v17
  
+     .line 341
      .local v17, targetHeight:I
      sub-int v11, p3, p1
  
+     .line 342
      .local v11, parentWidth:I
      sub-int v10, p4, p2
  
+     .line 344
      .local v10, parentHeight:I
      const v23, 0x3f2aaaab
  
***************
*** 462,467 ****
--- 512,518 ----
  
      add-int v9, v23, v24
  
+     .line 345
      .local v9, leftTarget:I
      const v23, 0x3eaaaaaa
  
***************
*** 481,494 ****
--- 532,548 ----
  
      sub-int v13, v23, v24
  
+     .line 346
      .local v13, rightTarget:I
      sub-int v23, v11, v7
  
      div-int/lit8 v8, v23, 0x2
  
+     .line 347
      .local v8, left:I
      add-int v12, v8, v7
  
+     .line 349
      .local v12, right:I
      if-eqz p5, :cond_0
  
***************
*** 500,526 ****
--- 554,586 ----
  
      if-ne v0, v1, :cond_2
  
+     .line 351
      :cond_0
      sub-int v23, v10, v17
  
      div-int/lit8 v20, v23, 0x2
  
+     .line 352
      .local v20, targetTop:I
      add-int v15, v20, v17
  
+     .line 353
      .local v15, targetBottom:I
      sub-int v23, v10, v6
  
      div-int/lit8 v22, v23, 0x2
  
+     .line 354
      .local v22, top:I
      add-int v23, v10, v6
  
      div-int/lit8 v5, v23, 0x2
  
+     .line 355
      .local v5, bottom:I
      if-nez p5, :cond_1
  
+     .line 356
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
***************
*** 537,542 ****
--- 597,603 ----
  
      invoke-virtual {v0, v1, v2, v7, v5}, Landroid/widget/ImageView;->layout(IIII)V
  
+     .line 357
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
***************
*** 557,562 ****
--- 618,624 ----
  
      invoke-virtual {v0, v1, v2, v3, v5}, Landroid/widget/TextView;->layout(IIII)V
  
+     .line 358
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
***************
*** 567,572 ****
--- 629,635 ----
  
      invoke-virtual/range {v23 .. v24}, Landroid/widget/TextView;->setGravity(I)V
  
+     .line 359
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
***************
*** 583,599 ****
--- 646,665 ----
  
      invoke-virtual {v0, v9, v1, v2, v15}, Landroid/widget/ImageView;->layout(IIII)V
  
+     .line 360
      move/from16 v0, p1
  
      move-object/from16 v1, p0
  
      iput v0, v1, Lcom/android/internal/widget/SlidingTab$Slider;->alignment_value:I
  
+     .line 386
      .end local v15           #targetBottom:I
      .end local v20           #targetTop:I
      :goto_0
      return-void
  
+     .line 362
      .restart local v15       #targetBottom:I
      .restart local v20       #targetTop:I
      :cond_1
***************
*** 613,618 ****
--- 679,685 ----
  
      invoke-virtual {v0, v1, v2, v11, v5}, Landroid/widget/ImageView;->layout(IIII)V
  
+     .line 363
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
***************
*** 629,634 ****
--- 696,702 ----
  
      invoke-virtual {v0, v11, v1, v2, v5}, Landroid/widget/TextView;->layout(IIII)V
  
+     .line 364
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
***************
*** 645,650 ****
--- 713,719 ----
  
      invoke-virtual {v0, v13, v1, v2, v15}, Landroid/widget/ImageView;->layout(IIII)V
  
+     .line 365
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
***************
*** 655,660 ****
--- 724,730 ----
  
      invoke-virtual/range {v23 .. v24}, Landroid/widget/TextView;->setGravity(I)V
  
+     .line 366
      move/from16 v0, p3
  
      move-object/from16 v1, p0
***************
*** 663,668 ****
--- 733,739 ----
  
      goto :goto_0
  
+     .line 370
      .end local v5           #bottom:I
      .end local v15           #targetBottom:I
      .end local v20           #targetTop:I
***************
*** 672,682 ****
--- 743,755 ----
  
      div-int/lit8 v18, v23, 0x2
  
+     .line 371
      .local v18, targetLeft:I
      add-int v23, v11, v21
  
      div-int/lit8 v19, v23, 0x2
  
+     .line 372
      .local v19, targetRight:I
      const v23, 0x3f2aaaab
  
***************
*** 698,703 ****
--- 771,777 ----
  
      sub-int v22, v23, v17
  
+     .line 373
      .restart local v22       #top:I
      const v23, 0x3eaaaaaa
  
***************
*** 717,722 ****
--- 791,797 ----
  
      sub-int v5, v23, v24
  
+     .line 374
      .restart local v5       #bottom:I
      const/16 v23, 0x2
  
***************
*** 726,731 ****
--- 801,807 ----
  
      if-ne v0, v1, :cond_3
  
+     .line 375
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
***************
*** 740,745 ****
--- 816,822 ----
  
      invoke-virtual {v0, v8, v1, v12, v6}, Landroid/widget/ImageView;->layout(IIII)V
  
+     .line 376
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
***************
*** 758,763 ****
--- 835,841 ----
  
      invoke-virtual {v0, v8, v1, v12, v2}, Landroid/widget/TextView;->layout(IIII)V
  
+     .line 377
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
***************
*** 778,783 ****
--- 856,862 ----
  
      invoke-virtual {v0, v1, v2, v3, v4}, Landroid/widget/ImageView;->layout(IIII)V
  
+     .line 378
      move/from16 v0, p2
  
      move-object/from16 v1, p0
***************
*** 786,791 ****
--- 865,871 ----
  
      goto/16 :goto_0
  
+     .line 380
      :cond_3
      move-object/from16 v0, p0
  
***************
*** 801,806 ****
--- 881,887 ----
  
      invoke-virtual {v0, v8, v1, v12, v10}, Landroid/widget/ImageView;->layout(IIII)V
  
+     .line 381
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
***************
*** 815,820 ****
--- 896,902 ----
  
      invoke-virtual {v0, v8, v10, v12, v1}, Landroid/widget/TextView;->layout(IIII)V
  
+     .line 382
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
***************
*** 833,838 ****
--- 915,921 ----
  
      invoke-virtual {v0, v1, v5, v2, v3}, Landroid/widget/ImageView;->layout(IIII)V
  
+     .line 383
      move/from16 v0, p4
  
      move-object/from16 v1, p0
***************
*** 848,853 ****
--- 931,937 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 396
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-static {v3, v3}, Landroid/view/View$MeasureSpec;->makeMeasureSpec(II)I
***************
*** 860,865 ****
--- 944,950 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/widget/ImageView;->measure(II)V
  
+     .line 398
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-static {v3, v3}, Landroid/view/View$MeasureSpec;->makeMeasureSpec(II)I
***************
*** 872,877 ****
--- 957,963 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/widget/TextView;->measure(II)V
  
+     .line 400
      return-void
  .end method
  
***************
*** 886,897 ****
--- 972,986 ----
  
      const/4 v4, 0x0
  
+     .line 290
      invoke-virtual {p0, v4}, Lcom/android/internal/widget/SlidingTab$Slider;->setState(I)V
  
+     .line 291
      iget-object v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v5, v4}, Landroid/widget/TextView;->setVisibility(I)V
  
+     .line 292
      iget-object v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      iget-object v6, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
***************
*** 904,919 ****
--- 993,1011 ----
  
      invoke-virtual {v5, v6, v7}, Landroid/widget/TextView;->setTextAppearance(Landroid/content/Context;I)V
  
+     .line 293
      iget-object v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v5, v4}, Landroid/widget/ImageView;->setVisibility(I)V
  
+     .line 294
      iget-object v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      const/4 v6, 0x4
  
      invoke-virtual {v5, v6}, Landroid/widget/ImageView;->setVisibility(I)V
  
+     .line 295
      iget v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->alignment:I
  
      if-eqz v5, :cond_0
***************
*** 922,927 ****
--- 1014,1020 ----
  
      if-ne v5, v2, :cond_1
  
+     .line 296
      .local v2, horiz:Z
      :cond_0
      :goto_0
***************
*** 941,956 ****
--- 1034,1052 ----
  
      sub-int v0, v5, v6
  
+     .line 298
      .local v0, dx:I
      :goto_1
      if-eqz v2, :cond_4
  
      move v1, v4
  
+     .line 300
      .local v1, dy:I
      :goto_2
      if-eqz p1, :cond_6
  
+     .line 301
      new-instance v3, Landroid/view/animation/TranslateAnimation;
  
      int-to-float v5, v0
***************
*** 959,979 ****
--- 1055,1080 ----
  
      invoke-direct {v3, v8, v5, v8, v6}, Landroid/view/animation/TranslateAnimation;-><init>(FFFF)V
  
+     .line 302
      .local v3, trans:Landroid/view/animation/TranslateAnimation;
      const-wide/16 v5, 0xfa
  
      invoke-virtual {v3, v5, v6}, Landroid/view/animation/TranslateAnimation;->setDuration(J)V
  
+     .line 303
      invoke-virtual {v3, v4}, Landroid/view/animation/TranslateAnimation;->setFillAfter(Z)V
  
+     .line 304
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v4, v3}, Landroid/widget/TextView;->startAnimation(Landroid/view/animation/Animation;)V
  
+     .line 305
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v4, v3}, Landroid/widget/ImageView;->startAnimation(Landroid/view/animation/Animation;)V
  
+     .line 318
      .end local v3           #trans:Landroid/view/animation/TranslateAnimation;
      :goto_3
      return-void
***************
*** 984,991 ****
--- 1085,1094 ----
      :cond_1
      move v2, v4
  
+     .line 295
      goto :goto_0
  
+     .line 296
      .restart local v2       #horiz:Z
      :cond_2
      iget v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->alignment_value:I
***************
*** 1005,1010 ****
--- 1108,1114 ----
  
      goto :goto_1
  
+     .line 298
      .restart local v0       #dx:I
      :cond_4
      iget v5, p0, Lcom/android/internal/widget/SlidingTab$Slider;->alignment:I
***************
*** 1038,1075 ****
--- 1142,1187 ----
  
      goto :goto_2
  
+     .line 307
      .restart local v1       #dy:I
      :cond_6
      if-eqz v2, :cond_7
  
+     .line 308
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v4, v0}, Landroid/widget/TextView;->offsetLeftAndRight(I)V
  
+     .line 309
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v4, v0}, Landroid/widget/ImageView;->offsetLeftAndRight(I)V
  
+     .line 314
      :goto_4
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v4}, Landroid/widget/TextView;->clearAnimation()V
  
+     .line 315
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v4}, Landroid/widget/ImageView;->clearAnimation()V
  
+     .line 316
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      invoke-virtual {v4}, Landroid/widget/ImageView;->clearAnimation()V
  
      goto :goto_3
  
+     .line 311
      :cond_7
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v4, v1}, Landroid/widget/TextView;->offsetTopAndBottom(I)V
  
+     .line 312
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v4, v1}, Landroid/widget/ImageView;->offsetTopAndBottom(I)V
***************
*** 1082,1091 ****
--- 1194,1205 ----
      .parameter "barId"
  
      .prologue
+     .line 226
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v0, p1}, Landroid/widget/TextView;->setBackgroundResource(I)V
  
+     .line 227
      return-void
  .end method
  
***************
*** 1094,1103 ****
--- 1208,1219 ----
      .parameter "resId"
  
      .prologue
+     .line 230
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v0, p1}, Landroid/widget/TextView;->setText(I)V
  
+     .line 231
      return-void
  .end method
  
***************
*** 1106,1115 ****
--- 1222,1233 ----
      .parameter "iconId"
  
      .prologue
+     .line 218
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v0, p1}, Landroid/widget/ImageView;->setImageResource(I)V
  
+     .line 219
      return-void
  .end method
  
***************
*** 1122,1127 ****
--- 1240,1246 ----
  
      const/4 v2, 0x1
  
+     .line 265
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      if-ne p1, v2, :cond_2
***************
*** 1131,1136 ****
--- 1250,1256 ----
      :goto_0
      invoke-virtual {v4, v1}, Landroid/widget/TextView;->setPressed(Z)V
  
+     .line 266
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      if-ne p1, v2, :cond_3
***************
*** 1140,1155 ****
--- 1260,1278 ----
      :goto_1
      invoke-virtual {v4, v1}, Landroid/widget/ImageView;->setPressed(Z)V
  
+     .line 267
      const/4 v1, 0x2
  
      if-ne p1, v1, :cond_4
  
+     .line 268
      new-array v0, v2, [I
  
      const v1, 0x10100a2
  
      aput v1, v0, v3
  
+     .line 269
      .local v0, activeState:[I
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
***************
*** 1163,1168 ****
--- 1286,1292 ----
  
      if-eqz v1, :cond_0
  
+     .line 270
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v1}, Landroid/widget/TextView;->getBackground()Landroid/graphics/drawable/Drawable;
***************
*** 1171,1176 ****
--- 1295,1301 ----
  
      invoke-virtual {v1, v0}, Landroid/graphics/drawable/Drawable;->setState([I)Z
  
+     .line 272
      :cond_0
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
***************
*** 1184,1189 ****
--- 1309,1315 ----
  
      if-eqz v1, :cond_1
  
+     .line 273
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v1}, Landroid/widget/ImageView;->getBackground()Landroid/graphics/drawable/Drawable;
***************
*** 1192,1197 ****
--- 1318,1324 ----
  
      invoke-virtual {v1, v0}, Landroid/graphics/drawable/Drawable;->setState([I)Z
  
+     .line 275
      :cond_1
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
***************
*** 1205,1226 ****
--- 1332,1358 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/widget/TextView;->setTextAppearance(Landroid/content/Context;I)V
  
+     .line 279
      .end local v0           #activeState:[I
      :goto_2
      iput p1, p0, Lcom/android/internal/widget/SlidingTab$Slider;->currentState:I
  
+     .line 280
      return-void
  
      :cond_2
      move v1, v3
  
+     .line 265
      goto :goto_0
  
      :cond_3
      move v1, v3
  
+     .line 266
      goto :goto_1
  
+     .line 277
      :cond_4
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
***************
*** 1242,1251 ****
--- 1374,1385 ----
      .parameter "tabId"
  
      .prologue
+     .line 222
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v0, p1}, Landroid/widget/ImageView;->setBackgroundResource(I)V
  
+     .line 223
      return-void
  .end method
  
***************
*** 1254,1263 ****
--- 1388,1399 ----
      .parameter "targetId"
  
      .prologue
+     .line 321
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      invoke-virtual {v0, p1}, Landroid/widget/ImageView;->setImageResource(I)V
  
+     .line 322
      return-void
  .end method
  
***************
*** 1272,1287 ****
--- 1408,1427 ----
  
      const/4 v1, 0x0
  
+     .line 249
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v4, v1}, Landroid/widget/TextView;->setVisibility(I)V
  
+     .line 250
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v4, v1}, Landroid/widget/ImageView;->setVisibility(I)V
  
+     .line 252
      if-eqz p1, :cond_1
  
+     .line 253
      iget v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->alignment:I
  
      if-eqz v4, :cond_0
***************
*** 1290,1295 ****
--- 1430,1436 ----
  
      if-ne v4, v2, :cond_2
  
+     .line 254
      .local v2, horiz:Z
      :cond_0
      :goto_0
***************
*** 1305,1314 ****
--- 1446,1457 ----
  
      move-result v0
  
+     .line 255
      .local v0, dx:I
      :goto_1
      if-eqz v2, :cond_5
  
+     .line 257
      .local v1, dy:I
      :goto_2
      new-instance v3, Landroid/view/animation/TranslateAnimation;
***************
*** 1323,1341 ****
--- 1466,1488 ----
  
      invoke-direct {v3, v4, v6, v5, v6}, Landroid/view/animation/TranslateAnimation;-><init>(FFFF)V
  
+     .line 258
      .local v3, trans:Landroid/view/animation/Animation;
      const-wide/16 v4, 0xfa
  
      invoke-virtual {v3, v4, v5}, Landroid/view/animation/Animation;->setDuration(J)V
  
+     .line 259
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v4, v3}, Landroid/widget/ImageView;->startAnimation(Landroid/view/animation/Animation;)V
  
+     .line 260
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v4, v3}, Landroid/widget/TextView;->startAnimation(Landroid/view/animation/Animation;)V
  
+     .line 262
      .end local v0           #dx:I
      .end local v1           #dy:I
      .end local v2           #horiz:Z
***************
*** 1346,1353 ****
--- 1493,1502 ----
      :cond_2
      move v2, v1
  
+     .line 253
      goto :goto_0
  
+     .line 254
      .restart local v2       #horiz:Z
      :cond_3
      iget-object v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
***************
*** 1365,1370 ****
--- 1514,1520 ----
  
      goto :goto_1
  
+     .line 255
      .restart local v0       #dx:I
      :cond_5
      iget v4, p0, Lcom/android/internal/widget/SlidingTab$Slider;->alignment:I
***************
*** 1397,1402 ****
--- 1547,1553 ----
      .locals 3
  
      .prologue
+     .line 283
      new-instance v0, Landroid/view/animation/AlphaAnimation;
  
      const/4 v1, 0x0
***************
*** 1405,1425 ****
--- 1556,1580 ----
  
      invoke-direct {v0, v1, v2}, Landroid/view/animation/AlphaAnimation;-><init>(FF)V
  
+     .line 284
      .local v0, alphaAnim:Landroid/view/animation/AlphaAnimation;
      const-wide/16 v1, 0x1f4
  
      invoke-virtual {v0, v1, v2}, Landroid/view/animation/AlphaAnimation;->setDuration(J)V
  
+     .line 285
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      invoke-virtual {v1, v0}, Landroid/widget/ImageView;->startAnimation(Landroid/view/animation/Animation;)V
  
+     .line 286
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$Slider;->target:Landroid/widget/ImageView;
  
      const/4 v2, 0x0
  
      invoke-virtual {v1, v2}, Landroid/widget/ImageView;->setVisibility(I)V
  
+     .line 287
      return-void
  .end method
  
***************
*** 1429,1442 ****
--- 1584,1600 ----
      .parameter "anim2"
  
      .prologue
+     .line 426
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->tab:Landroid/widget/ImageView;
  
      invoke-virtual {v0, p1}, Landroid/widget/ImageView;->startAnimation(Landroid/view/animation/Animation;)V
  
+     .line 427
      iget-object v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->text:Landroid/widget/TextView;
  
      invoke-virtual {v0, p2}, Landroid/widget/TextView;->startAnimation(Landroid/view/animation/Animation;)V
  
+     .line 428
      return-void
  .end method
  
***************
*** 1444,1452 ****
--- 1602,1612 ----
      .locals 1
  
      .prologue
+     .line 389
      iget v0, p0, Lcom/android/internal/widget/SlidingTab$Slider;->currentState:I
  
      invoke-virtual {p0, v0}, Lcom/android/internal/widget/SlidingTab$Slider;->setState(I)V
  
+     .line 390
      return-void
  .end method
