*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/SlidingTab$2.smali	2012-03-23 19:37:32.775372741 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/SlidingTab$2.smali.original	2012-03-23 19:34:46.080333061 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .parameter
  
      .prologue
+     .line 668
      iput-object p1, p0, Lcom/android/internal/widget/SlidingTab$2;->this$0:Lcom/android/internal/widget/SlidingTab;
  
      iput-boolean p2, p0, Lcom/android/internal/widget/SlidingTab$2;->val$holdAfter:Z
***************
*** 56,65 ****
--- 57,68 ----
      .parameter "animation"
  
      .prologue
+     .line 671
      iget-boolean v1, p0, Lcom/android/internal/widget/SlidingTab$2;->val$holdAfter:Z
  
      if-eqz v1, :cond_0
  
+     .line 672
      new-instance v0, Landroid/view/animation/TranslateAnimation;
  
      iget v1, p0, Lcom/android/internal/widget/SlidingTab$2;->val$dx:I
***************
*** 80,90 ****
--- 83,95 ----
  
      invoke-direct {v0, v1, v2, v3, v4}, Landroid/view/animation/TranslateAnimation;-><init>(FFFF)V
  
+     .line 673
      .local v0, anim:Landroid/view/animation/Animation;
      const-wide/16 v1, 0x3e8
  
      invoke-virtual {v0, v1, v2}, Landroid/view/animation/Animation;->setDuration(J)V
  
+     .line 674
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$2;->this$0:Lcom/android/internal/widget/SlidingTab;
  
      const/4 v2, 0x0
***************
*** 92,97 ****
--- 97,103 ----
      #setter for: Lcom/android/internal/widget/SlidingTab;->mAnimating:Z
      invoke-static {v1, v2}, Lcom/android/internal/widget/SlidingTab;->access$202(Lcom/android/internal/widget/SlidingTab;Z)Z
  
+     .line 680
      :goto_0
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$2;->this$0:Lcom/android/internal/widget/SlidingTab;
  
***************
*** 102,107 ****
--- 108,114 ----
  
      invoke-virtual {v0, v1}, Landroid/view/animation/Animation;->setAnimationListener(Landroid/view/animation/Animation$AnimationListener;)V
  
+     .line 683
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$2;->this$0:Lcom/android/internal/widget/SlidingTab;
  
      #getter for: Lcom/android/internal/widget/SlidingTab;->mLeftSlider:Lcom/android/internal/widget/SlidingTab$Slider;
***************
*** 111,116 ****
--- 118,124 ----
  
      invoke-virtual {v1, v0, v0}, Lcom/android/internal/widget/SlidingTab$Slider;->startAnimation(Landroid/view/animation/Animation;Landroid/view/animation/Animation;)V
  
+     .line 684
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$2;->this$0:Lcom/android/internal/widget/SlidingTab;
  
      #getter for: Lcom/android/internal/widget/SlidingTab;->mRightSlider:Lcom/android/internal/widget/SlidingTab$Slider;
***************
*** 120,127 ****
--- 128,137 ----
  
      invoke-virtual {v1, v0, v0}, Lcom/android/internal/widget/SlidingTab$Slider;->startAnimation(Landroid/view/animation/Animation;Landroid/view/animation/Animation;)V
  
+     .line 685
      return-void
  
+     .line 676
      .end local v0           #anim:Landroid/view/animation/Animation;
      :cond_0
      new-instance v0, Landroid/view/animation/AlphaAnimation;
***************
*** 132,142 ****
--- 142,154 ----
  
      invoke-direct {v0, v1, v2}, Landroid/view/animation/AlphaAnimation;-><init>(FF)V
  
+     .line 677
      .restart local v0       #anim:Landroid/view/animation/Animation;
      const-wide/16 v1, 0xfa
  
      invoke-virtual {v0, v1, v2}, Landroid/view/animation/Animation;->setDuration(J)V
  
+     .line 678
      iget-object v1, p0, Lcom/android/internal/widget/SlidingTab$2;->this$0:Lcom/android/internal/widget/SlidingTab;
  
      #calls: Lcom/android/internal/widget/SlidingTab;->resetView()V
***************
*** 150,155 ****
--- 162,168 ----
      .parameter "animation"
  
      .prologue
+     .line 689
      return-void
  .end method
  
***************
*** 158,162 ****
--- 171,176 ----
      .parameter "animation"
  
      .prologue
+     .line 693
      return-void
  .end method
