*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/TransportControlView$SavedState.smali	2012-03-23 19:37:33.929393784 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/TransportControlView$SavedState.smali.original	2012-03-23 19:34:46.104333499 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .locals 1
  
      .prologue
+     .line 396
      new-instance v0, Lcom/android/internal/widget/TransportControlView$SavedState$1;
  
      invoke-direct {v0}, Lcom/android/internal/widget/TransportControlView$SavedState$1;-><init>()V
***************
*** 50,57 ****
--- 51,60 ----
      .parameter "in"
  
      .prologue
+     .line 386
      invoke-direct {p0, p1}, Landroid/view/View$BaseSavedState;-><init>(Landroid/os/Parcel;)V
  
+     .line 387
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
***************
*** 63,70 ****
--- 66,75 ----
      :goto_0
      iput-boolean v0, p0, Lcom/android/internal/widget/TransportControlView$SavedState;->wasShowing:Z
  
+     .line 388
      return-void
  
+     .line 387
      :cond_0
      const/4 v0, 0x0
  
***************
*** 77,82 ****
--- 82,88 ----
      .parameter "x1"
  
      .prologue
+     .line 378
      invoke-direct {p0, p1}, Lcom/android/internal/widget/TransportControlView$SavedState;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 87,94 ****
--- 93,102 ----
      .parameter "superState"
  
      .prologue
+     .line 382
      invoke-direct {p0, p1}, Landroid/view/View$BaseSavedState;-><init>(Landroid/os/Parcelable;)V
  
+     .line 383
      return-void
  .end method
  
***************
*** 100,107 ****
--- 108,117 ----
      .parameter "flags"
  
      .prologue
+     .line 392
      invoke-super {p0, p1, p2}, Landroid/view/View$BaseSavedState;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 393
      iget-boolean v0, p0, Lcom/android/internal/widget/TransportControlView$SavedState;->wasShowing:Z
  
      if-eqz v0, :cond_0
***************
*** 111,118 ****
--- 121,130 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 394
      return-void
  
+     .line 393
      :cond_0
      const/4 v0, 0x0
  
