*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/TransportControlView$Metadata.smali	2012-03-23 19:37:32.974376370 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/TransportControlView$Metadata.smali.original	2012-03-23 19:34:46.089333225 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .parameter
  
      .prologue
+     .line 253
      iput-object p1, p0, Lcom/android/internal/widget/TransportControlView$Metadata;->this$0:Lcom/android/internal/widget/TransportControlView;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "x0"
  
      .prologue
+     .line 253
      iget-object v0, p0, Lcom/android/internal/widget/TransportControlView$Metadata;->trackTitle:Ljava/lang/String;
  
      return-object v0
***************
*** 55,60 ****
--- 57,63 ----
      .parameter "x1"
  
      .prologue
+     .line 253
      iput-object p1, p0, Lcom/android/internal/widget/TransportControlView$Metadata;->trackTitle:Ljava/lang/String;
  
      return-object p1
***************
*** 65,70 ****
--- 68,74 ----
      .parameter "x0"
  
      .prologue
+     .line 253
      iget-object v0, p0, Lcom/android/internal/widget/TransportControlView$Metadata;->albumTitle:Ljava/lang/String;
  
      return-object v0
***************
*** 76,81 ****
--- 80,86 ----
      .parameter "x1"
  
      .prologue
+     .line 253
      iput-object p1, p0, Lcom/android/internal/widget/TransportControlView$Metadata;->albumTitle:Ljava/lang/String;
  
      return-object p1
***************
*** 86,91 ****
--- 91,97 ----
      .parameter "x0"
  
      .prologue
+     .line 253
      iget-object v0, p0, Lcom/android/internal/widget/TransportControlView$Metadata;->bitmap:Landroid/graphics/Bitmap;
  
      return-object v0
***************
*** 97,102 ****
--- 103,109 ----
      .parameter "x1"
  
      .prologue
+     .line 253
      iput-object p1, p0, Lcom/android/internal/widget/TransportControlView$Metadata;->bitmap:Landroid/graphics/Bitmap;
  
      return-object p1
***************
*** 107,112 ****
--- 114,120 ----
      .parameter "x0"
  
      .prologue
+     .line 253
      iget-object v0, p0, Lcom/android/internal/widget/TransportControlView$Metadata;->artist:Ljava/lang/String;
  
      return-object v0
***************
*** 118,123 ****
--- 126,132 ----
      .parameter "x1"
  
      .prologue
+     .line 253
      iput-object p1, p0, Lcom/android/internal/widget/TransportControlView$Metadata;->artist:Ljava/lang/String;
  
      return-object p1
***************
*** 129,134 ****
--- 138,144 ----
      .locals 2
  
      .prologue
+     .line 260
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
