*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/preference/YesNoPreference$SavedState.smali	2012-03-23 19:37:52.683735770 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/preference/YesNoPreference$SavedState.smali.original	2012-03-23 19:34:46.388338676 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .locals 1
  
      .prologue
+     .line 139
      new-instance v0, Lcom/android/internal/preference/YesNoPreference$SavedState$1;
  
      invoke-direct {v0}, Lcom/android/internal/preference/YesNoPreference$SavedState$1;-><init>()V
***************
*** 52,59 ****
--- 53,62 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 125
      invoke-direct {p0, p1}, Landroid/preference/Preference$BaseSavedState;-><init>(Landroid/os/Parcel;)V
  
+     .line 126
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
***************
*** 63,70 ****
--- 66,75 ----
      :goto_0
      iput-boolean v0, p0, Lcom/android/internal/preference/YesNoPreference$SavedState;->wasPositiveResult:Z
  
+     .line 127
      return-void
  
+     .line 126
      :cond_0
      const/4 v0, 0x0
  
***************
*** 76,83 ****
--- 81,90 ----
      .parameter "superState"
  
      .prologue
+     .line 136
      invoke-direct {p0, p1}, Landroid/preference/Preference$BaseSavedState;-><init>(Landroid/os/Parcelable;)V
  
+     .line 137
      return-void
  .end method
  
***************
*** 89,96 ****
--- 96,105 ----
      .parameter "flags"
  
      .prologue
+     .line 131
      invoke-super {p0, p1, p2}, Landroid/preference/Preference$BaseSavedState;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 132
      iget-boolean v0, p0, Lcom/android/internal/preference/YesNoPreference$SavedState;->wasPositiveResult:Z
  
      if-eqz v0, :cond_0
***************
*** 100,107 ****
--- 109,118 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 133
      return-void
  
+     .line 132
      :cond_0
      const/4 v0, 0x0
  
