*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/GsmCall.smali	2012-03-23 19:37:35.712426297 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/GsmCall.smali.original	2012-03-23 19:34:46.138334119 -0500
***************
*** 32,47 ****
--- 32,51 ----
      .parameter "owner"
  
      .prologue
+     .line 56
      invoke-direct {p0}, Lcom/android/internal/telephony/Call;-><init>()V
  
+     .line 34
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
+     .line 57
      iput-object p1, p0, Lcom/android/internal/telephony/gsm/GsmCall;->owner:Lcom/android/internal/telephony/gsm/GsmCallTracker;
  
+     .line 58
      return-void
  .end method
  
***************
*** 50,55 ****
--- 54,60 ----
      .parameter "dcState"
  
      .prologue
+     .line 42
      sget-object v0, Lcom/android/internal/telephony/gsm/GsmCall$1;->$SwitchMap$com$android$internal$telephony$DriverCall$State:[I
  
      invoke-virtual {p0}, Lcom/android/internal/telephony/DriverCall$State;->ordinal()I
***************
*** 60,65 ****
--- 65,71 ----
  
      packed-switch v0, :pswitch_data_0
  
+     .line 49
      new-instance v0, Ljava/lang/RuntimeException;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 84,120 ****
--- 90,134 ----
  
      throw v0
  
+     .line 43
      :pswitch_0
      sget-object v0, Lcom/android/internal/telephony/Call$State;->ACTIVE:Lcom/android/internal/telephony/Call$State;
  
+     .line 48
      :goto_0
      return-object v0
  
+     .line 44
      :pswitch_1
      sget-object v0, Lcom/android/internal/telephony/Call$State;->HOLDING:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 45
      :pswitch_2
      sget-object v0, Lcom/android/internal/telephony/Call$State;->DIALING:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 46
      :pswitch_3
      sget-object v0, Lcom/android/internal/telephony/Call$State;->ALERTING:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 47
      :pswitch_4
      sget-object v0, Lcom/android/internal/telephony/Call$State;->INCOMING:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 48
      :pswitch_5
      sget-object v0, Lcom/android/internal/telephony/Call$State;->WAITING:Lcom/android/internal/telephony/Call$State;
  
      goto :goto_0
  
+     .line 42
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
***************
*** 134,143 ****
--- 148,159 ----
      .parameter "dc"
  
      .prologue
+     .line 99
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 101
      iget-object v0, p2, Lcom/android/internal/telephony/DriverCall;->state:Lcom/android/internal/telephony/DriverCall$State;
  
      invoke-static {v0}, Lcom/android/internal/telephony/gsm/GsmCall;->stateFromDCState(Lcom/android/internal/telephony/DriverCall$State;)Lcom/android/internal/telephony/Call$State;
***************
*** 146,151 ****
--- 162,168 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
+     .line 102
      return-void
  .end method
  
***************
*** 155,166 ****
--- 172,186 ----
      .parameter "state"
  
      .prologue
+     .line 106
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 108
      iput-object p2, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
+     .line 109
      return-void
  .end method
  
***************
*** 168,173 ****
--- 188,194 ----
      .locals 4
  
      .prologue
+     .line 195
      iget-object v2, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v2}, Ljava/util/ArrayList;->size()I
***************
*** 180,185 ****
--- 201,207 ----
      :goto_0
      if-ltz v1, :cond_1
  
+     .line 196
      iget-object v2, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v2, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 188,193 ****
--- 210,216 ----
  
      check-cast v0, Lcom/android/internal/telephony/gsm/GsmConnection;
  
+     .line 198
      .local v0, cn:Lcom/android/internal/telephony/gsm/GsmConnection;
      invoke-virtual {v0}, Lcom/android/internal/telephony/gsm/GsmConnection;->getState()Lcom/android/internal/telephony/Call$State;
  
***************
*** 197,211 ****
--- 220,237 ----
  
      if-ne v2, v3, :cond_0
  
+     .line 199
      iget-object v2, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v2, v1}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 195
      :cond_0
      add-int/lit8 v1, v1, -0x1
  
      goto :goto_0
  
+     .line 203
      .end local v0           #cn:Lcom/android/internal/telephony/gsm/GsmConnection;
      :cond_1
      iget-object v2, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
***************
*** 216,225 ****
--- 242,253 ----
  
      if-nez v2, :cond_2
  
+     .line 204
      sget-object v2, Lcom/android/internal/telephony/Call$State;->IDLE:Lcom/android/internal/telephony/Call$State;
  
      iput-object v2, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
+     .line 206
      :cond_2
      return-void
  .end method
***************
*** 229,242 ****
--- 257,273 ----
      .parameter "conn"
  
      .prologue
+     .line 116
      iget-object v3, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
      sget-object v4, Lcom/android/internal/telephony/Call$State;->DISCONNECTED:Lcom/android/internal/telephony/Call$State;
  
      if-eq v3, v4, :cond_1
  
+     .line 119
      const/4 v0, 0x1
  
+     .line 121
      .local v0, hasOnlyDisconnectedConnections:Z
      const/4 v1, 0x0
  
***************
*** 251,256 ****
--- 282,288 ----
      :goto_0
      if-ge v1, v2, :cond_0
  
+     .line 122
      iget-object v3, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 267,287 ****
--- 299,324 ----
  
      if-eq v3, v4, :cond_2
  
+     .line 125
      const/4 v0, 0x0
  
+     .line 130
      :cond_0
      if-eqz v0, :cond_1
  
+     .line 131
      sget-object v3, Lcom/android/internal/telephony/Call$State;->DISCONNECTED:Lcom/android/internal/telephony/Call$State;
  
      iput-object v3, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
+     .line 134
      .end local v0           #hasOnlyDisconnectedConnections:Z
      .end local v1           #i:I
      .end local v2           #s:I
      :cond_1
      return-void
  
+     .line 121
      .restart local v0       #hasOnlyDisconnectedConnections:Z
      .restart local v1       #i:I
      .restart local v2       #s:I
***************
*** 296,305 ****
--- 333,344 ----
      .parameter "conn"
  
      .prologue
+     .line 139
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->remove(Ljava/lang/Object;)Z
  
+     .line 141
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->size()I
***************
*** 308,317 ****
--- 347,358 ----
  
      if-nez v0, :cond_0
  
+     .line 142
      sget-object v0, Lcom/android/internal/telephony/Call$State;->IDLE:Lcom/android/internal/telephony/Call$State;
  
      iput-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
+     .line 144
      :cond_0
      return-void
  .end method
***************
*** 320,325 ****
--- 361,367 ----
      .locals 0
  
      .prologue
+     .line 61
      return-void
  .end method
  
***************
*** 336,341 ****
--- 378,384 ----
      .end annotation
  
      .prologue
+     .line 68
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      return-object v0
***************
*** 345,350 ****
--- 388,394 ----
      .locals 1
  
      .prologue
+     .line 73
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->owner:Lcom/android/internal/telephony/gsm/GsmCallTracker;
  
      iget-object v0, v0, Lcom/android/internal/telephony/gsm/GsmCallTracker;->phone:Lcom/android/internal/telephony/gsm/GSMPhone;
***************
*** 361,370 ****
--- 405,416 ----
      .end annotation
  
      .prologue
+     .line 87
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->owner:Lcom/android/internal/telephony/gsm/GsmCallTracker;
  
      invoke-virtual {v0, p0}, Lcom/android/internal/telephony/gsm/GsmCallTracker;->hangup(Lcom/android/internal/telephony/gsm/GsmCall;)V
  
+     .line 88
      return-void
  .end method
  
***************
*** 372,377 ****
--- 418,424 ----
      .locals 2
  
      .prologue
+     .line 167
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->size()I
***************
*** 399,404 ****
--- 446,452 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 78
      iget-object v1, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v1}, Ljava/util/ArrayList;->size()I
***************
*** 420,425 ****
--- 468,474 ----
      .locals 4
  
      .prologue
+     .line 180
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 429,438 ****
--- 478,489 ----
  
      move-result v2
  
+     .line 181
      .local v2, s:I
      :goto_0
      if-ge v1, v2, :cond_0
  
+     .line 183
      iget-object v3, p0, Lcom/android/internal/telephony/gsm/GsmCall;->connections:Ljava/util/ArrayList;
  
      invoke-virtual {v3, v1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 441,459 ****
--- 492,514 ----
  
      check-cast v0, Lcom/android/internal/telephony/gsm/GsmConnection;
  
+     .line 185
      .local v0, cn:Lcom/android/internal/telephony/gsm/GsmConnection;
      invoke-virtual {v0}, Lcom/android/internal/telephony/gsm/GsmConnection;->onHangupLocal()V
  
+     .line 181
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 187
      .end local v0           #cn:Lcom/android/internal/telephony/gsm/GsmConnection;
      :cond_0
      sget-object v3, Lcom/android/internal/telephony/Call$State;->DISCONNECTING:Lcom/android/internal/telephony/Call$State;
  
      iput-object v3, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
+     .line 188
      return-void
  .end method
  
***************
*** 461,466 ****
--- 516,522 ----
      .locals 1
  
      .prologue
+     .line 92
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
      invoke-virtual {v0}, Lcom/android/internal/telephony/Call$State;->toString()Ljava/lang/String;
***************
*** 476,483 ****
--- 532,541 ----
      .parameter "dc"
  
      .prologue
+     .line 149
      const/4 v0, 0x0
  
+     .line 151
      .local v0, changed:Z
      iget-object v2, p2, Lcom/android/internal/telephony/DriverCall;->state:Lcom/android/internal/telephony/DriverCall$State;
  
***************
*** 485,499 ****
--- 543,561 ----
  
      move-result-object v1
  
+     .line 153
      .local v1, newState:Lcom/android/internal/telephony/Call$State;
      iget-object v2, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
      if-eq v1, v2, :cond_0
  
+     .line 154
      iput-object v1, p0, Lcom/android/internal/telephony/gsm/GsmCall;->state:Lcom/android/internal/telephony/Call$State;
  
+     .line 155
      const/4 v0, 0x1
  
+     .line 158
      :cond_0
      return v0
  .end method
