*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/GsmConnection$MyHandler.smali	2012-03-23 19:37:36.358438077 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/GsmConnection$MyHandler.smali.original	2012-03-23 19:34:46.150334337 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .parameter "l"
  
      .prologue
+     .line 95
      iput-object p1, p0, Lcom/android/internal/telephony/gsm/GsmConnection$MyHandler;->this$0:Lcom/android/internal/telephony/gsm/GsmConnection;
  
      invoke-direct {p0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
***************
*** 39,51 ****
--- 40,55 ----
      .parameter "msg"
  
      .prologue
+     .line 100
      iget v0, p1, Landroid/os/Message;->what:I
  
      packed-switch v0, :pswitch_data_0
  
+     .line 110
      :goto_0
      return-void
  
+     .line 104
      :pswitch_0
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmConnection$MyHandler;->this$0:Lcom/android/internal/telephony/gsm/GsmConnection;
  
***************
*** 54,59 ****
--- 58,64 ----
  
      goto :goto_0
  
+     .line 107
      :pswitch_1
      iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmConnection$MyHandler;->this$0:Lcom/android/internal/telephony/gsm/GsmConnection;
  
***************
*** 62,67 ****
--- 67,73 ----
  
      goto :goto_0
  
+     .line 100
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_0
