*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/GsmServiceStateTracker$4.smali	2012-03-23 19:37:36.192435051 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/GsmServiceStateTracker$4.smali.original	2012-03-23 19:34:46.148334301 -0500
***************
*** 23,28 ****
--- 23,29 ----
      .locals 3
  
      .prologue
+     .line 689
      invoke-static {}, Lcom/android/internal/telephony/CommandsInterface$RadioState;->values()[Lcom/android/internal/telephony/CommandsInterface$RadioState;
  
      move-result-object v0
