*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/SIMFileHandler.smali	2012-03-23 19:37:35.587424017 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/SIMFileHandler.smali.original	2012-03-23 19:34:46.135334063 -0500
***************
*** 20,29 ****
--- 20,32 ----
      .parameter "phone"
  
      .prologue
+     .line 40
      invoke-direct {p0, p1}, Lcom/android/internal/telephony/IccFileHandler;-><init>(Lcom/android/internal/telephony/PhoneBase;)V
  
+     .line 41
      iput-object p1, p0, Lcom/android/internal/telephony/gsm/SIMFileHandler;->mPhone:Lcom/android/internal/telephony/Phone;
  
+     .line 42
      return-void
  .end method
  
***************
*** 33,40 ****
--- 36,45 ----
      .locals 0
  
      .prologue
+     .line 45
      invoke-super {p0}, Lcom/android/internal/telephony/IccFileHandler;->dispose()V
  
+     .line 46
      return-void
  .end method
  
***************
*** 42,53 ****
--- 47,60 ----
      .locals 2
  
      .prologue
+     .line 49
      const-string v0, "GSM"
  
      const-string v1, "SIMFileHandler finalized"
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 50
      return-void
  .end method
  
***************
*** 56,76 ****
--- 63,88 ----
      .parameter "efid"
  
      .prologue
+     .line 62
      sparse-switch p1, :sswitch_data_0
  
+     .line 91
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/gsm/SIMFileHandler;->getCommonIccEFPath(I)Ljava/lang/String;
  
      move-result-object v1
  
+     .line 92
      .local v1, path:Ljava/lang/String;
      if-nez v1, :cond_0
  
+     .line 96
      iget-object v2, p0, Lcom/android/internal/telephony/gsm/SIMFileHandler;->phone:Lcom/android/internal/telephony/PhoneBase;
  
      invoke-virtual {v2}, Lcom/android/internal/telephony/PhoneBase;->getIccCard()Lcom/android/internal/telephony/IccCard;
  
      move-result-object v0
  
+     .line 97
      .local v0, card:Lcom/android/internal/telephony/IccCard;
      if-eqz v0, :cond_1
  
***************
*** 82,115 ****
--- 94,134 ----
  
      if-eqz v2, :cond_1
  
+     .line 98
      const-string v1, "3F007F105F3A"
  
+     .line 102
      .end local v0           #card:Lcom/android/internal/telephony/IccCard;
      .end local v1           #path:Ljava/lang/String;
      :cond_0
      :goto_0
      return-object v1
  
+     .line 64
      :sswitch_0
      const-string v1, "3F007F10"
  
      goto :goto_0
  
+     .line 76
      :sswitch_1
      const-string v1, "3F007F20"
  
      goto :goto_0
  
+     .line 85
      :sswitch_2
      const-string v1, "3F007F20"
  
      goto :goto_0
  
+     .line 89
      :sswitch_3
      const-string v1, "3F007F105F3A"
  
      goto :goto_0
  
+     .line 100
      .restart local v0       #card:Lcom/android/internal/telephony/IccCard;
      .restart local v1       #path:Ljava/lang/String;
      :cond_1
***************
*** 121,126 ****
--- 140,146 ----
  
      goto :goto_0
  
+     .line 62
      :sswitch_data_0
      .sparse-switch
          0x4f30 -> :sswitch_3
***************
*** 150,157 ****
--- 170,179 ----
      .parameter "msg"
  
      .prologue
+     .line 56
      invoke-super {p0, p1}, Lcom/android/internal/telephony/IccFileHandler;->handleMessage(Landroid/os/Message;)V
  
+     .line 57
      return-void
  .end method
  
***************
*** 160,165 ****
--- 182,188 ----
      .parameter "msg"
  
      .prologue
+     .line 106
      const-string v0, "GSM"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 182,187 ****
--- 205,211 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 107
      return-void
  .end method
  
***************
*** 190,195 ****
--- 214,220 ----
      .parameter "msg"
  
      .prologue
+     .line 110
      const-string v0, "GSM"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 212,216 ****
--- 237,242 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 111
      return-void
  .end method
