*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/UsimServiceTable$UsimService.smali	2012-03-23 19:37:35.121415521 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/telephony/gsm/UsimServiceTable$UsimService.smali.original	2012-03-23 19:34:46.124333863 -0500
***************
*** 234,239 ****
--- 234,240 ----
  
      const/4 v3, 0x0
  
+     .line 27
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "PHONEBOOK"
***************
*** 242,247 ****
--- 243,249 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->PHONEBOOK:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 28
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "FDN"
***************
*** 250,255 ****
--- 252,258 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->FDN:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 29
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "FDN_EXTENSION"
***************
*** 258,263 ****
--- 261,267 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->FDN_EXTENSION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 30
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "SDN"
***************
*** 266,271 ****
--- 270,276 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->SDN:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 31
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "SDN_EXTENSION"
***************
*** 274,279 ****
--- 279,285 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->SDN_EXTENSION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 32
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "BDN"
***************
*** 284,289 ****
--- 290,296 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->BDN:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 33
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "BDN_EXTENSION"
***************
*** 294,299 ****
--- 301,307 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->BDN_EXTENSION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 34
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "OUTGOING_CALL_INFO"
***************
*** 304,309 ****
--- 312,318 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->OUTGOING_CALL_INFO:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 35
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "INCOMING_CALL_INFO"
***************
*** 314,319 ****
--- 323,329 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->INCOMING_CALL_INFO:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 36
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "SM_STORAGE"
***************
*** 324,329 ****
--- 334,340 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->SM_STORAGE:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 37
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "SM_STATUS_REPORTS"
***************
*** 334,339 ****
--- 345,351 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->SM_STATUS_REPORTS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 38
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "SM_SERVICE_PARAMS"
***************
*** 344,349 ****
--- 356,362 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->SM_SERVICE_PARAMS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 39
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "ADVICE_OF_CHARGE"
***************
*** 354,359 ****
--- 367,373 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->ADVICE_OF_CHARGE:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 40
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "CAP_CONFIG_PARAMS_2"
***************
*** 364,369 ****
--- 378,384 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->CAP_CONFIG_PARAMS_2:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 41
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "CB_MESSAGE_ID"
***************
*** 374,379 ****
--- 389,395 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->CB_MESSAGE_ID:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 42
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "CB_MESSAGE_ID_RANGES"
***************
*** 384,389 ****
--- 400,406 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->CB_MESSAGE_ID_RANGES:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 43
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "GROUP_ID_LEVEL_1"
***************
*** 394,399 ****
--- 411,417 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->GROUP_ID_LEVEL_1:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 44
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "GROUP_ID_LEVEL_2"
***************
*** 404,409 ****
--- 422,428 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->GROUP_ID_LEVEL_2:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 45
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "SPN"
***************
*** 414,419 ****
--- 433,439 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->SPN:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 46
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "USER_PLMN_SELECT"
***************
*** 424,429 ****
--- 444,450 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->USER_PLMN_SELECT:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 47
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MSISDN"
***************
*** 434,439 ****
--- 455,461 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MSISDN:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 48
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IMAGE"
***************
*** 444,449 ****
--- 466,472 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IMAGE:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 49
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "LOCALISED_SERVICE_AREAS"
***************
*** 454,459 ****
--- 477,483 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->LOCALISED_SERVICE_AREAS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 50
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "EMLPP"
***************
*** 464,469 ****
--- 488,494 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->EMLPP:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 51
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "EMLPP_AUTO_ANSWER"
***************
*** 474,479 ****
--- 499,505 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->EMLPP_AUTO_ANSWER:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 52
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "RFU"
***************
*** 484,489 ****
--- 510,516 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->RFU:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 53
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "GSM_ACCESS"
***************
*** 494,499 ****
--- 521,527 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->GSM_ACCESS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 54
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "DATA_DL_VIA_SMS_PP"
***************
*** 504,509 ****
--- 532,538 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->DATA_DL_VIA_SMS_PP:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 55
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "DATA_DL_VIA_SMS_CB"
***************
*** 514,519 ****
--- 543,549 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->DATA_DL_VIA_SMS_CB:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 56
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "CALL_CONTROL_BY_USIM"
***************
*** 524,529 ****
--- 554,560 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->CALL_CONTROL_BY_USIM:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 57
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MO_SMS_CONTROL_BY_USIM"
***************
*** 534,539 ****
--- 565,571 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MO_SMS_CONTROL_BY_USIM:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 58
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "RUN_AT_COMMAND"
***************
*** 544,549 ****
--- 576,582 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->RUN_AT_COMMAND:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 59
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IGNORED_1"
***************
*** 554,559 ****
--- 587,593 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IGNORED_1:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 60
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "ENABLED_SERVICES_TABLE"
***************
*** 564,569 ****
--- 598,604 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->ENABLED_SERVICES_TABLE:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 61
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "APN_CONTROL_LIST"
***************
*** 574,579 ****
--- 609,615 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->APN_CONTROL_LIST:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 62
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "DEPERSONALISATION_CONTROL_KEYS"
***************
*** 584,589 ****
--- 620,626 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->DEPERSONALISATION_CONTROL_KEYS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 63
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "COOPERATIVE_NETWORK_LIST"
***************
*** 594,599 ****
--- 631,637 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->COOPERATIVE_NETWORK_LIST:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 64
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "GSM_SECURITY_CONTEXT"
***************
*** 604,609 ****
--- 642,648 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->GSM_SECURITY_CONTEXT:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 65
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "CPBCCH_INFO"
***************
*** 614,619 ****
--- 653,659 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->CPBCCH_INFO:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 66
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "INVESTIGATION_SCAN"
***************
*** 624,629 ****
--- 664,670 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->INVESTIGATION_SCAN:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 67
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MEXE"
***************
*** 634,639 ****
--- 675,681 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MEXE:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 68
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "OPERATOR_PLMN_SELECT"
***************
*** 644,649 ****
--- 686,692 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->OPERATOR_PLMN_SELECT:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 69
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "HPLMN_SELECT"
***************
*** 654,659 ****
--- 697,703 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->HPLMN_SELECT:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 70
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "EXTENSION_5"
***************
*** 664,669 ****
--- 708,714 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->EXTENSION_5:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 71
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "PLMN_NETWORK_NAME"
***************
*** 674,679 ****
--- 719,725 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->PLMN_NETWORK_NAME:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 72
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "OPERATOR_PLMN_LIST"
***************
*** 684,689 ****
--- 730,736 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->OPERATOR_PLMN_LIST:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 73
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MBDN"
***************
*** 694,699 ****
--- 741,747 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MBDN:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 74
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MWI_STATUS"
***************
*** 704,709 ****
--- 752,758 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MWI_STATUS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 75
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "CFI_STATUS"
***************
*** 714,719 ****
--- 763,769 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->CFI_STATUS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 76
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IGNORED_2"
***************
*** 724,729 ****
--- 774,780 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IGNORED_2:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 77
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "SERVICE_PROVIDER_DISPLAY_INFO"
***************
*** 734,739 ****
--- 785,791 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->SERVICE_PROVIDER_DISPLAY_INFO:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 78
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MMS_NOTIFICATION"
***************
*** 744,749 ****
--- 796,802 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MMS_NOTIFICATION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 79
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MMS_NOTIFICATION_EXTENSION"
***************
*** 754,759 ****
--- 807,813 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MMS_NOTIFICATION_EXTENSION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 80
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "GPRS_CALL_CONTROL_BY_USIM"
***************
*** 764,769 ****
--- 818,824 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->GPRS_CALL_CONTROL_BY_USIM:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 81
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MMS_CONNECTIVITY_PARAMS"
***************
*** 774,779 ****
--- 829,835 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MMS_CONNECTIVITY_PARAMS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 82
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "NETWORK_INDICATION_OF_ALERTING"
***************
*** 784,789 ****
--- 840,846 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->NETWORK_INDICATION_OF_ALERTING:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 83
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "VGCS_GROUP_ID_LIST"
***************
*** 794,799 ****
--- 851,857 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->VGCS_GROUP_ID_LIST:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 84
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "VBS_GROUP_ID_LIST"
***************
*** 804,809 ****
--- 862,868 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->VBS_GROUP_ID_LIST:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 85
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "PSEUDONYM"
***************
*** 814,819 ****
--- 873,879 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->PSEUDONYM:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 86
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IWLAN_USER_PLMN_SELECT"
***************
*** 824,829 ****
--- 884,890 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IWLAN_USER_PLMN_SELECT:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 87
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IWLAN_OPERATOR_PLMN_SELECT"
***************
*** 834,839 ****
--- 895,901 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IWLAN_OPERATOR_PLMN_SELECT:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 88
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "USER_WSID_LIST"
***************
*** 844,849 ****
--- 906,912 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->USER_WSID_LIST:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 89
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "OPERATOR_WSID_LIST"
***************
*** 854,859 ****
--- 917,923 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->OPERATOR_WSID_LIST:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 90
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "VGCS_SECURITY"
***************
*** 864,869 ****
--- 928,934 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->VGCS_SECURITY:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 91
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "VBS_SECURITY"
***************
*** 874,879 ****
--- 939,945 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->VBS_SECURITY:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 92
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "WLAN_REAUTH_IDENTITY"
***************
*** 884,889 ****
--- 950,956 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->WLAN_REAUTH_IDENTITY:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 93
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MM_STORAGE"
***************
*** 894,899 ****
--- 961,967 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MM_STORAGE:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 94
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "GBA"
***************
*** 904,909 ****
--- 972,978 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->GBA:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 95
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "MBMS_SECURITY"
***************
*** 914,919 ****
--- 983,989 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->MBMS_SECURITY:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 96
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "DATA_DL_VIA_USSD"
***************
*** 924,929 ****
--- 994,1000 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->DATA_DL_VIA_USSD:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 97
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "EQUIVALENT_HPLMN"
***************
*** 934,939 ****
--- 1005,1011 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->EQUIVALENT_HPLMN:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 98
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "TERMINAL_PROFILE_AFTER_UICC_ACTIVATION"
***************
*** 944,949 ****
--- 1016,1022 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->TERMINAL_PROFILE_AFTER_UICC_ACTIVATION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 99
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "EQUIVALENT_HPLMN_PRESENTATION"
***************
*** 954,959 ****
--- 1027,1033 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->EQUIVALENT_HPLMN_PRESENTATION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 100
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "LAST_RPLMN_SELECTION_INDICATION"
***************
*** 964,969 ****
--- 1038,1044 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->LAST_RPLMN_SELECTION_INDICATION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 101
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "OMA_BCAST_PROFILE"
***************
*** 974,979 ****
--- 1049,1055 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->OMA_BCAST_PROFILE:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 102
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "GBA_LOCAL_KEY_ESTABLISHMENT"
***************
*** 984,989 ****
--- 1060,1066 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->GBA_LOCAL_KEY_ESTABLISHMENT:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 103
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "TERMINAL_APPLICATIONS"
***************
*** 994,999 ****
--- 1071,1077 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->TERMINAL_APPLICATIONS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 104
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "SPN_ICON"
***************
*** 1004,1009 ****
--- 1082,1088 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->SPN_ICON:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 105
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "PLMN_NETWORK_NAME_ICON"
***************
*** 1014,1019 ****
--- 1093,1099 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->PLMN_NETWORK_NAME_ICON:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 106
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "USIM_IP_CONNECTION_PARAMS"
***************
*** 1024,1029 ****
--- 1104,1110 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->USIM_IP_CONNECTION_PARAMS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 107
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IWLAN_HOME_ID_LIST"
***************
*** 1034,1039 ****
--- 1115,1121 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IWLAN_HOME_ID_LIST:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 108
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IWLAN_EQUIVALENT_HPLMN_PRESENTATION"
***************
*** 1044,1049 ****
--- 1126,1132 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IWLAN_EQUIVALENT_HPLMN_PRESENTATION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 109
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IWLAN_HPLMN_PRIORITY_INDICATION"
***************
*** 1054,1059 ****
--- 1137,1143 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IWLAN_HPLMN_PRIORITY_INDICATION:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 110
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IWLAN_LAST_REGISTERED_PLMN"
***************
*** 1064,1069 ****
--- 1148,1154 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IWLAN_LAST_REGISTERED_PLMN:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 111
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "EPS_MOBILITY_MANAGEMENT_INFO"
***************
*** 1074,1079 ****
--- 1159,1165 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->EPS_MOBILITY_MANAGEMENT_INFO:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 112
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "ALLOWED_CSG_LISTS_AND_INDICATIONS"
***************
*** 1084,1089 ****
--- 1170,1176 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->ALLOWED_CSG_LISTS_AND_INDICATIONS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 113
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "CALL_CONTROL_ON_EPS_PDN_CONNECTION_BY_USIM"
***************
*** 1094,1099 ****
--- 1181,1187 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->CALL_CONTROL_ON_EPS_PDN_CONNECTION_BY_USIM:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 114
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "HPLMN_DIRECT_ACCESS"
***************
*** 1104,1109 ****
--- 1192,1198 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->HPLMN_DIRECT_ACCESS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 115
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "ECALL_DATA"
***************
*** 1114,1119 ****
--- 1203,1209 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->ECALL_DATA:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 116
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "OPERATOR_CSG_LISTS_AND_INDICATIONS"
***************
*** 1124,1129 ****
--- 1214,1220 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->OPERATOR_CSG_LISTS_AND_INDICATIONS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 117
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "SM_OVER_IP"
***************
*** 1134,1139 ****
--- 1225,1231 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->SM_OVER_IP:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 118
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "CSG_DISPLAY_CONTROL"
***************
*** 1144,1149 ****
--- 1236,1242 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->CSG_DISPLAY_CONTROL:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 119
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "IMS_COMMUNICATION_CONTROL_BY_USIM"
***************
*** 1154,1159 ****
--- 1247,1253 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->IMS_COMMUNICATION_CONTROL_BY_USIM:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 120
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "EXTENDED_TERMINAL_APPLICATIONS"
***************
*** 1164,1169 ****
--- 1258,1264 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->EXTENDED_TERMINAL_APPLICATIONS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 121
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "UICC_ACCESS_TO_IMS"
***************
*** 1174,1179 ****
--- 1269,1275 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->UICC_ACCESS_TO_IMS:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 122
      new-instance v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      const-string v1, "NAS_CONFIG_BY_USIM"
***************
*** 1184,1189 ****
--- 1280,1286 ----
  
      sput-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->NAS_CONFIG_BY_USIM:Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
+     .line 26
      const/16 v0, 0x60
  
      new-array v0, v0, [Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
***************
*** 1770,1775 ****
--- 1867,1873 ----
      .end annotation
  
      .prologue
+     .line 26
      invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
  
      return-void
***************
*** 1780,1785 ****
--- 1878,1884 ----
      .parameter "name"
  
      .prologue
+     .line 26
      const-class v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
***************
*** 1795,1800 ****
--- 1894,1900 ----
      .locals 1
  
      .prologue
+     .line 26
      sget-object v0, Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->$VALUES:[Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;
  
      invoke-virtual {v0}, [Lcom/android/internal/telephony/gsm/UsimServiceTable$UsimService;->clone()Ljava/lang/Object;
