*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/IRecognitionService$Stub$Proxy.smali	2012-03-23 19:36:32.071265802 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/IRecognitionService$Stub$Proxy.smali.original	2012-03-23 19:34:45.175316557 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 92
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 93
      iput-object p1, p0, Landroid/speech/IRecognitionService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 94
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 97
      iget-object v0, p0, Landroid/speech/IRecognitionService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 57,72 ****
--- 61,79 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 158
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 160
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.speech.IRecognitionService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 161
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Landroid/speech/IRecognitionListener;->asBinder()Landroid/os/IBinder;
***************
*** 76,81 ****
--- 83,89 ----
      :cond_0
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 162
      iget-object v1, p0, Landroid/speech/IRecognitionService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x3
***************
*** 88,97 ****
--- 96,108 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 165
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 167
      return-void
  
+     .line 165
      :catchall_0
      move-exception v1
  
***************
*** 104,109 ****
--- 115,121 ----
      .locals 1
  
      .prologue
+     .line 101
      const-string v0, "android.speech.IRecognitionService"
  
      return-object v0
***************
*** 122,147 ****
--- 134,165 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 115
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 117
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.speech.IRecognitionService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 118
      if-eqz p1, :cond_1
  
+     .line 119
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 120
      const/4 v2, 0x0
  
      invoke-virtual {p1, v0, v2}, Landroid/content/Intent;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 125
      :goto_0
      if-eqz p2, :cond_0
  
***************
*** 152,157 ****
--- 170,176 ----
      :cond_0
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 126
      iget-object v1, p0, Landroid/speech/IRecognitionService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x1
***************
*** 164,173 ****
--- 183,195 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 129
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 131
      return-void
  
+     .line 123
      :cond_1
      const/4 v2, 0x0
  
***************
*** 178,183 ****
--- 200,206 ----
  
      goto :goto_0
  
+     .line 129
      :catchall_0
      move-exception v1
  
***************
*** 198,213 ****
--- 221,239 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 141
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 143
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.speech.IRecognitionService"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 144
      if-eqz p1, :cond_0
  
      invoke-interface {p1}, Landroid/speech/IRecognitionListener;->asBinder()Landroid/os/IBinder;
***************
*** 217,222 ****
--- 243,249 ----
      :cond_0
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 145
      iget-object v1, p0, Landroid/speech/IRecognitionService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x2
***************
*** 229,238 ****
--- 256,268 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 148
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 150
      return-void
  
+     .line 148
      :catchall_0
      move-exception v1
  
