*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/IRecognitionListener$Stub$Proxy.smali	2012-03-23 19:36:32.027265000 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/IRecognitionListener$Stub$Proxy.smali.original	2012-03-23 19:34:45.175316557 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 148
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 149
      iput-object p1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 150
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 153
      iget-object v0, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 49,54 ****
--- 53,59 ----
      .locals 1
  
      .prologue
+     .line 157
      const-string v0, "android.speech.IRecognitionListener"
  
      return-object v0
***************
*** 63,78 ****
--- 68,86 ----
      .end annotation
  
      .prologue
+     .line 187
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 189
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.speech.IRecognitionListener"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 190
      iget-object v1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x2
***************
*** 85,94 ****
--- 93,105 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 193
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 195
      return-void
  
+     .line 193
      :catchall_0
      move-exception v1
  
***************
*** 107,124 ****
--- 118,139 ----
      .end annotation
  
      .prologue
+     .line 220
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 222
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.speech.IRecognitionListener"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 223
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeByteArray([B)V
  
+     .line 224
      iget-object v1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x4
***************
*** 131,140 ****
--- 146,158 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 227
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 229
      return-void
  
+     .line 227
      :catchall_0
      move-exception v1
  
***************
*** 152,167 ****
--- 170,188 ----
      .end annotation
  
      .prologue
+     .line 235
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 237
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.speech.IRecognitionListener"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 238
      iget-object v1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x5
***************
*** 174,183 ****
--- 195,207 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 241
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 243
      return-void
  
+     .line 241
      :catchall_0
      move-exception v1
  
***************
*** 196,213 ****
--- 220,241 ----
      .end annotation
  
      .prologue
+     .line 251
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 253
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.speech.IRecognitionListener"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 254
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 255
      iget-object v1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x6
***************
*** 220,229 ****
--- 248,260 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 258
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 260
      return-void
  
+     .line 258
      :catchall_0
      move-exception v1
  
***************
*** 243,270 ****
--- 274,308 ----
      .end annotation
  
      .prologue
+     .line 315
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 317
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.speech.IRecognitionListener"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 318
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 319
      if-eqz p2, :cond_0
  
+     .line 320
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 321
      const/4 v1, 0x0
  
      invoke-virtual {p2, v0, v1}, Landroid/os/Bundle;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 326
      :goto_0
      iget-object v1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 278,287 ****
--- 316,328 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 329
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 331
      return-void
  
+     .line 324
      :cond_0
      const/4 v1, 0x0
  
***************
*** 292,297 ****
--- 333,339 ----
  
      goto :goto_0
  
+     .line 329
      :catchall_0
      move-exception v1
  
***************
*** 310,335 ****
--- 352,383 ----
      .end annotation
  
      .prologue
+     .line 291
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 293
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.speech.IRecognitionListener"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 294
      if-eqz p1, :cond_0
  
+     .line 295
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 296
      const/4 v1, 0x0
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 301
      :goto_0
      iget-object v1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 343,352 ****
--- 391,403 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 304
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 306
      return-void
  
+     .line 299
      :cond_0
      const/4 v1, 0x0
  
***************
*** 357,362 ****
--- 408,414 ----
  
      goto :goto_0
  
+     .line 304
      :catchall_0
      move-exception v1
  
***************
*** 375,400 ****
--- 427,458 ----
      .end annotation
  
      .prologue
+     .line 166
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 168
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.speech.IRecognitionListener"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 169
      if-eqz p1, :cond_0
  
+     .line 170
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 171
      const/4 v1, 0x0
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 176
      :goto_0
      iget-object v1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 408,417 ****
--- 466,478 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 179
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 181
      return-void
  
+     .line 174
      :cond_0
      const/4 v1, 0x0
  
***************
*** 422,427 ****
--- 483,489 ----
  
      goto :goto_0
  
+     .line 179
      :catchall_0
      move-exception v1
  
***************
*** 440,465 ****
--- 502,533 ----
      .end annotation
  
      .prologue
+     .line 268
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 270
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.speech.IRecognitionListener"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 271
      if-eqz p1, :cond_0
  
+     .line 272
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 273
      const/4 v1, 0x0
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Bundle;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 278
      :goto_0
      iget-object v1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 473,482 ****
--- 541,553 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 281
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 283
      return-void
  
+     .line 276
      :cond_0
      const/4 v1, 0x0
  
***************
*** 487,492 ****
--- 558,564 ----
  
      goto :goto_0
  
+     .line 281
      :catchall_0
      move-exception v1
  
***************
*** 505,522 ****
--- 577,598 ----
      .end annotation
  
      .prologue
+     .line 203
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 205
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.speech.IRecognitionListener"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 206
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeFloat(F)V
  
+     .line 207
      iget-object v1, p0, Landroid/speech/IRecognitionListener$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x3
***************
*** 529,538 ****
--- 605,617 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 210
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 212
      return-void
  
+     .line 210
      :catchall_0
      move-exception v1
  
