*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/RecognizerResultsIntent.smali	2012-03-23 19:36:33.841298079 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/RecognizerResultsIntent.smali.original	2012-03-23 19:34:45.195316923 -0500
***************
*** 24,30 ****
--- 24,32 ----
      .locals 0
  
      .prologue
+     .line 39
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 41
      return-void
  .end method
