*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/RecognizerIntent.smali	2012-03-23 19:36:32.048265384 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/RecognizerIntent.smali.original	2012-03-23 19:34:45.175316557 -0500
***************
*** 68,75 ****
--- 68,77 ----
      .locals 0
  
      .prologue
+     .line 44
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 46
      return-void
  .end method
  
***************
*** 80,91 ****
--- 82,95 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 289
      new-instance v3, Landroid/content/Intent;
  
      const-string v4, "android.speech.action.WEB_SEARCH"
  
      invoke-direct {v3, v4}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 290
      .local v3, voiceSearchIntent:Landroid/content/Intent;
      invoke-virtual {p0}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
  
***************
*** 97,102 ****
--- 101,107 ----
  
      move-result-object v2
  
+     .line 292
      .local v2, ri:Landroid/content/pm/ResolveInfo;
      if-eqz v2, :cond_0
  
***************
*** 110,119 ****
--- 115,126 ----
  
      if-nez v4, :cond_1
  
+     .line 299
      :cond_0
      :goto_0
      return-object v1
  
+     .line 294
      :cond_1
      iget-object v4, v2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
***************
*** 125,139 ****
--- 132,149 ----
  
      move-result-object v0
  
+     .line 295
      .local v0, className:Ljava/lang/String;
      if-eqz v0, :cond_0
  
+     .line 297
      new-instance v1, Landroid/content/Intent;
  
      const-string v4, "android.speech.action.GET_LANGUAGE_DETAILS"
  
      invoke-direct {v1, v4}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
  
+     .line 298
      .local v1, detailsIntent:Landroid/content/Intent;
      new-instance v4, Landroid/content/ComponentName;
  
