*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/RecognitionService$Callback.smali	2012-03-23 19:36:31.797260806 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/RecognitionService$Callback.smali.original	2012-03-23 19:34:45.170316467 -0500
***************
*** 27,38 ****
--- 27,41 ----
      .parameter "listener"
  
      .prologue
+     .line 218
      iput-object p1, p0, Landroid/speech/RecognitionService$Callback;->this$0:Landroid/speech/RecognitionService;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 219
      iput-object p2, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
+     .line 220
      return-void
  .end method
  
***************
*** 43,48 ****
--- 46,52 ----
      .parameter "x2"
  
      .prologue
+     .line 215
      invoke-direct {p0, p1, p2}, Landroid/speech/RecognitionService$Callback;-><init>(Landroid/speech/RecognitionService;Landroid/speech/IRecognitionListener;)V
  
      return-void
***************
*** 53,58 ****
--- 57,63 ----
      .parameter "x0"
  
      .prologue
+     .line 215
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
      return-object v0
***************
*** 69,78 ****
--- 74,85 ----
      .end annotation
  
      .prologue
+     .line 227
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
      invoke-interface {v0}, Landroid/speech/IRecognitionListener;->onBeginningOfSpeech()V
  
+     .line 228
      return-void
  .end method
  
***************
*** 86,95 ****
--- 93,104 ----
      .end annotation
  
      .prologue
+     .line 238
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
      invoke-interface {v0, p1}, Landroid/speech/IRecognitionListener;->onBufferReceived([B)V
  
+     .line 239
      return-void
  .end method
  
***************
*** 102,111 ****
--- 111,122 ----
      .end annotation
  
      .prologue
+     .line 245
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
      invoke-interface {v0}, Landroid/speech/IRecognitionListener;->onEndOfSpeech()V
  
+     .line 246
      return-void
  .end method
  
***************
*** 119,124 ****
--- 130,136 ----
      .end annotation
  
      .prologue
+     .line 254
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->this$0:Landroid/speech/RecognitionService;
  
      #getter for: Landroid/speech/RecognitionService;->mHandler:Landroid/os/Handler;
***************
*** 134,143 ****
--- 146,157 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 255
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
      invoke-interface {v0, p1}, Landroid/speech/IRecognitionListener;->onError(I)V
  
+     .line 256
      return-void
  .end method
  
***************
*** 151,160 ****
--- 165,176 ----
      .end annotation
  
      .prologue
+     .line 270
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
      invoke-interface {v0, p1}, Landroid/speech/IRecognitionListener;->onPartialResults(Landroid/os/Bundle;)V
  
+     .line 271
      return-void
  .end method
  
***************
*** 168,177 ****
--- 184,195 ----
      .end annotation
  
      .prologue
+     .line 280
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
      invoke-interface {v0, p1}, Landroid/speech/IRecognitionListener;->onReadyForSpeech(Landroid/os/Bundle;)V
  
+     .line 281
      return-void
  .end method
  
***************
*** 185,190 ****
--- 203,209 ----
      .end annotation
  
      .prologue
+     .line 291
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->this$0:Landroid/speech/RecognitionService;
  
      #getter for: Landroid/speech/RecognitionService;->mHandler:Landroid/os/Handler;
***************
*** 200,209 ****
--- 219,230 ----
  
      invoke-virtual {v0}, Landroid/os/Message;->sendToTarget()V
  
+     .line 292
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
      invoke-interface {v0, p1}, Landroid/speech/IRecognitionListener;->onResults(Landroid/os/Bundle;)V
  
+     .line 293
      return-void
  .end method
  
***************
*** 217,225 ****
--- 238,248 ----
      .end annotation
  
      .prologue
+     .line 302
      iget-object v0, p0, Landroid/speech/RecognitionService$Callback;->mListener:Landroid/speech/IRecognitionListener;
  
      invoke-interface {v0, p1}, Landroid/speech/IRecognitionListener;->onRmsChanged(F)V
  
+     .line 303
      return-void
  .end method
