*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/RecognitionService.smali	2012-03-23 19:36:32.223268574 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/speech/RecognitionService.smali.original	2012-03-23 19:34:45.177316595 -0500
***************
*** 44,67 ****
--- 44,72 ----
      .locals 1
  
      .prologue
+     .line 36
      invoke-direct {p0}, Landroid/app/Service;-><init>()V
  
+     .line 57
      new-instance v0, Landroid/speech/RecognitionService$RecognitionServiceBinder;
  
      invoke-direct {v0, p0}, Landroid/speech/RecognitionService$RecognitionServiceBinder;-><init>(Landroid/speech/RecognitionService;)V
  
      iput-object v0, p0, Landroid/speech/RecognitionService;->mBinder:Landroid/speech/RecognitionService$RecognitionServiceBinder;
  
+     .line 63
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
+     .line 73
      new-instance v0, Landroid/speech/RecognitionService$1;
  
      invoke-direct {v0, p0}, Landroid/speech/RecognitionService$1;-><init>(Landroid/speech/RecognitionService;)V
  
      iput-object v0, p0, Landroid/speech/RecognitionService;->mHandler:Landroid/os/Handler;
  
+     .line 307
      return-void
  .end method
  
***************
*** 72,77 ****
--- 77,83 ----
      .parameter "x2"
  
      .prologue
+     .line 36
      invoke-direct {p0, p1, p2}, Landroid/speech/RecognitionService;->dispatchStartListening(Landroid/content/Intent;Landroid/speech/IRecognitionListener;)V
  
      return-void
***************
*** 83,88 ****
--- 89,95 ----
      .parameter "x1"
  
      .prologue
+     .line 36
      invoke-direct {p0, p1}, Landroid/speech/RecognitionService;->dispatchStopListening(Landroid/speech/IRecognitionListener;)V
  
      return-void
***************
*** 94,99 ****
--- 101,107 ----
      .parameter "x1"
  
      .prologue
+     .line 36
      invoke-direct {p0, p1}, Landroid/speech/RecognitionService;->dispatchCancel(Landroid/speech/IRecognitionListener;)V
  
      return-void
***************
*** 104,109 ****
--- 112,118 ----
      .parameter "x0"
  
      .prologue
+     .line 36
      invoke-direct {p0}, Landroid/speech/RecognitionService;->dispatchClearCallback()V
  
      return-void
***************
*** 114,119 ****
--- 123,129 ----
      .parameter "x0"
  
      .prologue
+     .line 36
      iget-object v0, p0, Landroid/speech/RecognitionService;->mHandler:Landroid/os/Handler;
  
      return-object v0
***************
*** 125,130 ****
--- 135,141 ----
      .parameter "x1"
  
      .prologue
+     .line 36
      invoke-direct {p0, p1}, Landroid/speech/RecognitionService;->checkPermissions(Landroid/speech/IRecognitionListener;)Z
  
      move-result v0
***************
*** 137,142 ****
--- 148,154 ----
      .parameter "listener"
  
      .prologue
+     .line 160
      const-string v1, "android.permission.RECORD_AUDIO"
  
      invoke-virtual {p0, v1}, Landroid/speech/RecognitionService;->checkCallingOrSelfPermission(Ljava/lang/String;)I
***************
*** 145,155 ****
--- 157,170 ----
  
      if-nez v1, :cond_0
  
+     .line 162
      const/4 v1, 0x1
  
+     .line 170
      :goto_0
      return v1
  
+     .line 165
      :cond_0
      :try_start_0
      const-string v1, "RecognitionService"
***************
*** 158,181 ****
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
      const/16 v1, 0x9
  
      invoke-interface {p1, v1}, Landroid/speech/IRecognitionListener;->onError(I)V
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
      :goto_1
      const/4 v1, 0x0
  
      goto :goto_0
  
      :catch_0
      move-exception v0
  
      .local v0, re:Landroid/os/RemoteException;
      const-string v1, "RecognitionService"
  
!     const-string v2, "sending ERROR_INSUFFICIENT_PERMISSIONS message failed"
  
      invoke-static {v1, v2, v0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
--- 173,200 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 166
      const/16 v1, 0x9
  
      invoke-interface {p1, v1}, Landroid/speech/IRecognitionListener;->onError(I)V
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 170
      :goto_1
      const/4 v1, 0x0
  
      goto :goto_0
  
+     .line 167
      :catch_0
      move-exception v0
  
+     .line 168
      .local v0, re:Landroid/os/RemoteException;
      const-string v1, "RecognitionService"
  
!     const-string/jumbo v2, "sending ERROR_INSUFFICIENT_PERMISSIONS message failed"
  
      invoke-static {v1, v2, v0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
***************
*** 187,199 ****
--- 206,221 ----
      .parameter "listener"
  
      .prologue
+     .line 126
      iget-object v0, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
      if-nez v0, :cond_0
  
+     .line 135
      :goto_0
      return-void
  
+     .line 128
      :cond_0
      iget-object v0, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
***************
*** 212,217 ****
--- 234,240 ----
  
      if-eq v0, v1, :cond_1
  
+     .line 129
      const-string v0, "RecognitionService"
  
      const-string v1, "cancel called by client who did not call startListening - ignoring"
***************
*** 220,230 ****
--- 243,255 ----
  
      goto :goto_0
  
+     .line 131
      :cond_1
      iget-object v0, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
      invoke-virtual {p0, v0}, Landroid/speech/RecognitionService;->onCancel(Landroid/speech/RecognitionService$Callback;)V
  
+     .line 132
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
***************
*** 236,245 ****
--- 261,272 ----
      .locals 1
  
      .prologue
+     .line 138
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
+     .line 139
      return-void
  .end method
  
***************
*** 249,258 ****
--- 276,287 ----
      .parameter "listener"
  
      .prologue
+     .line 95
      iget-object v1, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
      if-nez v1, :cond_0
  
+     .line 97
      new-instance v1, Landroid/speech/RecognitionService$Callback;
  
      const/4 v2, 0x0
***************
*** 261,273 ****
--- 290,305 ----
  
      iput-object v1, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
+     .line 98
      iget-object v1, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
      invoke-virtual {p0, p1, v1}, Landroid/speech/RecognitionService;->onStartListening(Landroid/content/Intent;Landroid/speech/RecognitionService$Callback;)V
  
+     .line 107
      :goto_0
      return-void
  
+     .line 101
      :cond_0
      const/16 v1, 0x8
  
***************
*** 276,281 ****
--- 308,314 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 105
      :goto_1
      const-string v1, "RecognitionService"
  
***************
*** 285,297 ****
  
      goto :goto_0
  
      :catch_0
      move-exception v0
  
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "RecognitionService"
  
!     const-string v2, "onError call from startListening failed"
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
--- 318,332 ----
  
      goto :goto_0
  
+     .line 102
      :catch_0
      move-exception v0
  
+     .line 103
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "RecognitionService"
  
!     const-string/jumbo v2, "onError call from startListening failed"
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
***************
*** 303,326 ****
      .parameter "listener"
  
      .prologue
      :try_start_0
      iget-object v1, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
      if-nez v1, :cond_0
  
      const/4 v1, 0x5
  
      invoke-interface {p1, v1}, Landroid/speech/IRecognitionListener;->onError(I)V
  
      const-string v1, "RecognitionService"
  
!     const-string v2, "stopListening called with no preceding startListening - ignoring"
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
      :goto_0
      return-void
  
      :cond_0
      iget-object v1, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
--- 338,366 ----
      .parameter "listener"
  
      .prologue
+     .line 111
      :try_start_0
      iget-object v1, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
      if-nez v1, :cond_0
  
+     .line 112
      const/4 v1, 0x5
  
      invoke-interface {p1, v1}, Landroid/speech/IRecognitionListener;->onError(I)V
  
+     .line 113
      const-string v1, "RecognitionService"
  
!     const-string/jumbo v2, "stopListening called with no preceding startListening - ignoring"
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 123
      :goto_0
      return-void
  
+     .line 114
      :cond_0
      iget-object v1, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
***************
*** 339,351 ****
  
      if-eq v1, v2, :cond_1
  
      const/16 v1, 0x8
  
      invoke-interface {p1, v1}, Landroid/speech/IRecognitionListener;->onError(I)V
  
      const-string v1, "RecognitionService"
  
!     const-string v2, "stopListening called by other caller than startListening - ignoring"
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
      :try_end_0
--- 379,393 ----
  
      if-eq v1, v2, :cond_1
  
+     .line 115
      const/16 v1, 0x8
  
      invoke-interface {p1, v1}, Landroid/speech/IRecognitionListener;->onError(I)V
  
+     .line 116
      const-string v1, "RecognitionService"
  
!     const-string/jumbo v2, "stopListening called by other caller than startListening - ignoring"
  
      invoke-static {v1, v2}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
      :try_end_0
***************
*** 353,370 ****
  
      goto :goto_0
  
      :catch_0
      move-exception v0
  
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "RecognitionService"
  
!     const-string v2, "onError call from stopListening failed"
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
      goto :goto_0
  
      .end local v0           #e:Landroid/os/RemoteException;
      :cond_1
      :try_start_1
--- 395,415 ----
  
      goto :goto_0
  
+     .line 120
      :catch_0
      move-exception v0
  
+     .line 121
      .local v0, e:Landroid/os/RemoteException;
      const-string v1, "RecognitionService"
  
!     const-string/jumbo v2, "onError call from stopListening failed"
  
      invoke-static {v1, v2}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
      goto :goto_0
  
+     .line 118
      .end local v0           #e:Landroid/os/RemoteException;
      :cond_1
      :try_start_1
***************
*** 384,389 ****
--- 429,435 ----
      .parameter "intent"
  
      .prologue
+     .line 198
      iget-object v0, p0, Landroid/speech/RecognitionService;->mBinder:Landroid/speech/RecognitionService$RecognitionServiceBinder;
  
      return-object v0
***************
*** 396,411 ****
--- 442,461 ----
      .locals 1
  
      .prologue
+     .line 204
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/speech/RecognitionService;->mCurrentCallback:Landroid/speech/RecognitionService$Callback;
  
+     .line 205
      iget-object v0, p0, Landroid/speech/RecognitionService;->mBinder:Landroid/speech/RecognitionService$RecognitionServiceBinder;
  
      invoke-virtual {v0}, Landroid/speech/RecognitionService$RecognitionServiceBinder;->clearReference()V
  
+     .line 206
      invoke-super {p0}, Landroid/app/Service;->onDestroy()V
  
+     .line 207
      return-void
  .end method
  
