*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/CameraSound.smali	2012-03-23 19:36:02.567727806 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/CameraSound.smali.original	2012-03-23 19:34:44.951312474 -0500
***************
*** 34,41 ****
--- 34,43 ----
      .locals 0
  
      .prologue
+     .line 66
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 67
      return-void
  .end method
  
***************
*** 46,61 ****
--- 48,66 ----
      .parameter "soundId"
  
      .prologue
+     .line 91
      iget-object v0, p0, Landroid/hardware/CameraSound;->mCameraSoundPlayers:[Landroid/hardware/CameraSound$CameraSoundPlayer;
  
      if-nez v0, :cond_0
  
+     .line 92
      const/4 v0, 0x4
  
      new-array v0, v0, [Landroid/hardware/CameraSound$CameraSoundPlayer;
  
      iput-object v0, p0, Landroid/hardware/CameraSound;->mCameraSoundPlayers:[Landroid/hardware/CameraSound$CameraSoundPlayer;
  
+     .line 94
      :cond_0
      iget-object v0, p0, Landroid/hardware/CameraSound;->mCameraSoundPlayers:[Landroid/hardware/CameraSound$CameraSoundPlayer;
  
***************
*** 63,68 ****
--- 68,74 ----
  
      if-nez v0, :cond_1
  
+     .line 95
      iget-object v0, p0, Landroid/hardware/CameraSound;->mCameraSoundPlayers:[Landroid/hardware/CameraSound$CameraSoundPlayer;
  
      new-instance v1, Landroid/hardware/CameraSound$CameraSoundPlayer;
***************
*** 71,76 ****
--- 77,83 ----
  
      aput-object v1, v0, p1
  
+     .line 97
      :cond_1
      iget-object v0, p0, Landroid/hardware/CameraSound;->mCameraSoundPlayers:[Landroid/hardware/CameraSound$CameraSoundPlayer;
  
***************
*** 78,83 ****
--- 85,91 ----
  
      invoke-virtual {v0}, Landroid/hardware/CameraSound$CameraSoundPlayer;->play()V
  
+     .line 98
      return-void
  .end method
  
***************
*** 85,94 ****
--- 93,104 ----
      .locals 5
  
      .prologue
+     .line 101
      iget-object v4, p0, Landroid/hardware/CameraSound;->mCameraSoundPlayers:[Landroid/hardware/CameraSound$CameraSoundPlayer;
  
      if-eqz v4, :cond_2
  
+     .line 102
      iget-object v0, p0, Landroid/hardware/CameraSound;->mCameraSoundPlayers:[Landroid/hardware/CameraSound$CameraSoundPlayer;
  
      .local v0, arr$:[Landroid/hardware/CameraSound$CameraSoundPlayer;
***************
*** 103,124 ****
--- 113,139 ----
  
      aget-object v1, v0, v2
  
+     .line 103
      .local v1, csp:Landroid/hardware/CameraSound$CameraSoundPlayer;
      if-eqz v1, :cond_0
  
+     .line 104
      invoke-virtual {v1}, Landroid/hardware/CameraSound$CameraSoundPlayer;->release()V
  
+     .line 102
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 107
      .end local v1           #csp:Landroid/hardware/CameraSound$CameraSoundPlayer;
      :cond_1
      const/4 v4, 0x0
  
      iput-object v4, p0, Landroid/hardware/CameraSound;->mCameraSoundPlayers:[Landroid/hardware/CameraSound$CameraSoundPlayer;
  
+     .line 109
      .end local v0           #arr$:[Landroid/hardware/CameraSound$CameraSoundPlayer;
      .end local v2           #i$:I
      .end local v3           #len$:I
