*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/CameraSound$CameraSoundPlayer.smali	2012-03-23 19:36:04.356760430 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/CameraSound$CameraSoundPlayer.smali.original	2012-03-23 19:34:44.960312638 -0500
***************
*** 47,57 ****
      .parameter "soundId"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      iput p1, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mSoundId:I
  
!     const-string v0, "ro.camera.sound.forced"
  
      const-string v1, "0"
  
--- 47,60 ----
      .parameter "soundId"
  
      .prologue
+     .line 180
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 181
      iput p1, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mSoundId:I
  
!     .line 182
!     const-string/jumbo v0, "ro.camera.sound.forced"
  
      const-string v1, "0"
  
***************
*** 67,79 ****
--- 70,85 ----
  
      if-eqz v0, :cond_0
  
+     .line 183
      const/4 v0, 0x3
  
      iput v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mAudioStreamType:I
  
+     .line 187
      :goto_0
      return-void
  
+     .line 185
      :cond_0
      const/4 v0, 0x7
  
***************
*** 88,95 ****
--- 94,103 ----
      .locals 0
  
      .prologue
+     .line 220
      invoke-virtual {p0}, Landroid/hardware/CameraSound$CameraSoundPlayer;->release()V
  
+     .line 221
      return-void
  .end method
  
***************
*** 97,119 ****
--- 105,132 ----
      .locals 1
  
      .prologue
+     .line 190
      iget-object v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mThread:Ljava/lang/Thread;
  
      if-nez v0, :cond_0
  
+     .line 191
      new-instance v0, Ljava/lang/Thread;
  
      invoke-direct {v0, p0}, Ljava/lang/Thread;-><init>(Ljava/lang/Runnable;)V
  
      iput-object v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mThread:Ljava/lang/Thread;
  
+     .line 192
      iget-object v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mThread:Ljava/lang/Thread;
  
      invoke-virtual {v0}, Ljava/lang/Thread;->start()V
  
+     .line 194
      :cond_0
      monitor-enter p0
  
+     .line 195
      :try_start_0
      iget v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayCount:I
  
***************
*** 121,132 ****
--- 134,149 ----
  
      iput v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayCount:I
  
+     .line 196
      invoke-virtual {p0}, Ljava/lang/Object;->notifyAll()V
  
+     .line 197
      monitor-exit p0
  
+     .line 198
      return-void
  
+     .line 197
      :catchall_0
      move-exception v0
  
***************
*** 143,165 ****
--- 160,188 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 201
      iget-object v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mThread:Ljava/lang/Thread;
  
      if-eqz v0, :cond_0
  
+     .line 202
      monitor-enter p0
  
+     .line 203
      const/4 v0, 0x1
  
      :try_start_0
      iput-boolean v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mExit:Z
  
+     .line 204
      invoke-virtual {p0}, Ljava/lang/Object;->notifyAll()V
  
+     .line 205
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 207
      :try_start_1
      iget-object v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mThread:Ljava/lang/Thread;
  
***************
*** 167,189 ****
--- 190,218 ----
      :try_end_1
      .catch Ljava/lang/InterruptedException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 210
      :goto_0
      iput-object v1, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mThread:Ljava/lang/Thread;
  
+     .line 212
      :cond_0
      iget-object v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayer:Landroid/media/MediaPlayer;
  
      if-eqz v0, :cond_1
  
+     .line 213
      iget-object v0, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayer:Landroid/media/MediaPlayer;
  
      invoke-virtual {v0}, Landroid/media/MediaPlayer;->release()V
  
+     .line 214
      iput-object v1, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayer:Landroid/media/MediaPlayer;
  
+     .line 216
      :cond_1
      return-void
  
+     .line 205
      :catchall_0
      move-exception v0
  
***************
*** 194,199 ****
--- 223,229 ----
  
      throw v0
  
+     .line 208
      :catch_0
      move-exception v0
  
***************
*** 204,213 ****
--- 234,245 ----
      .locals 5
  
      .prologue
+     .line 131
      iget v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mSoundId:I
  
      packed-switch v2, :pswitch_data_0
  
+     .line 145
      const-string v2, "CameraSound"
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 238,249 ****
--- 270,284 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 164
      :goto_0
      return-void
  
+     .line 133
      :pswitch_0
      const-string v1, "/system/media/audio/ui/camera_click.ogg"
  
+     .line 148
      .local v1, soundFilePath:Ljava/lang/String;
      :goto_1
      new-instance v2, Landroid/media/MediaPlayer;
***************
*** 252,257 ****
--- 287,293 ----
  
      iput-object v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayer:Landroid/media/MediaPlayer;
  
+     .line 150
      :try_start_0
      iget-object v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayer:Landroid/media/MediaPlayer;
  
***************
*** 259,296 ****
--- 295,339 ----
  
      invoke-virtual {v2, v3}, Landroid/media/MediaPlayer;->setAudioStreamType(I)V
  
+     .line 151
      iget-object v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayer:Landroid/media/MediaPlayer;
  
      invoke-virtual {v2, v1}, Landroid/media/MediaPlayer;->setDataSource(Ljava/lang/String;)V
  
+     .line 152
      iget-object v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayer:Landroid/media/MediaPlayer;
  
      const/4 v3, 0x0
  
      invoke-virtual {v2, v3}, Landroid/media/MediaPlayer;->setLooping(Z)V
  
+     .line 153
      iget-object v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayer:Landroid/media/MediaPlayer;
  
      invoke-virtual {v2}, Landroid/media/MediaPlayer;->prepare()V
      :try_end_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 161
      :goto_2
      :try_start_1
      monitor-enter p0
      :try_end_1
      .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 163
      :goto_3
      :try_start_2
      iget-boolean v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mExit:Z
  
      if-eqz v2, :cond_0
  
+     .line 164
      monitor-exit p0
  
      goto :goto_0
  
+     .line 172
      :catchall_0
      move-exception v2
  
***************
*** 303,311 ****
--- 346,356 ----
      :try_end_3
      .catch Ljava/lang/Exception; {:try_start_3 .. :try_end_3} :catch_0
  
+     .line 174
      :catch_0
      move-exception v0
  
+     .line 175
      .local v0, e:Ljava/lang/Exception;
      const-string v2, "CameraSound"
  
***************
*** 333,363 ****
--- 378,416 ----
  
      goto :goto_2
  
+     .line 136
      .end local v0           #e:Ljava/lang/Exception;
      .end local v1           #soundFilePath:Ljava/lang/String;
      :pswitch_1
      const-string v1, "/system/media/audio/ui/camera_focus.ogg"
  
+     .line 137
      .restart local v1       #soundFilePath:Ljava/lang/String;
      goto :goto_1
  
+     .line 139
      .end local v1           #soundFilePath:Ljava/lang/String;
      :pswitch_2
      const-string v1, "/system/media/audio/ui/VideoRecord.ogg"
  
+     .line 140
      .restart local v1       #soundFilePath:Ljava/lang/String;
      goto :goto_1
  
+     .line 142
      .end local v1           #soundFilePath:Ljava/lang/String;
      :pswitch_3
      const-string v1, "/system/media/audio/ui/VideoRecord.ogg"
  
+     .line 143
      .restart local v1       #soundFilePath:Ljava/lang/String;
      goto :goto_1
  
+     .line 154
      :catch_1
      move-exception v0
  
+     .line 155
      .local v0, e:Ljava/io/IOException;
      const-string v2, "CameraSound"
  
***************
*** 385,390 ****
--- 438,444 ----
  
      goto :goto_0
  
+     .line 165
      .end local v0           #e:Ljava/io/IOException;
      :cond_0
      :try_start_4
***************
*** 392,401 ****
--- 446,457 ----
  
      if-gtz v2, :cond_1
  
+     .line 166
      invoke-virtual {p0}, Ljava/lang/Object;->wait()V
  
      goto :goto_3
  
+     .line 168
      :cond_1
      iget v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayCount:I
  
***************
*** 403,412 ****
--- 459,470 ----
  
      iput v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayCount:I
  
+     .line 172
      monitor-exit p0
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
  
+     .line 173
      :try_start_5
      iget-object v2, p0, Landroid/hardware/CameraSound$CameraSoundPlayer;->mPlayer:Landroid/media/MediaPlayer;
  
***************
*** 416,421 ****
--- 474,480 ----
  
      goto :goto_2
  
+     .line 131
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
