*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/Camera$Area.smali	2012-03-23 19:36:03.281740826 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/Camera$Area.smali.original	2012-03-23 19:34:44.955312546 -0500
***************
*** 27,38 ****
--- 27,42 ----
      .parameter "weight"
  
      .prologue
+     .line 1390
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1391
      iput-object p1, p0, Landroid/hardware/Camera$Area;->rect:Landroid/graphics/Rect;
  
+     .line 1392
      iput p2, p0, Landroid/hardware/Camera$Area;->weight:I
  
+     .line 1393
      return-void
  .end method
  
***************
*** 45,54 ****
--- 49,60 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 1403
      instance-of v2, p1, Landroid/hardware/Camera$Area;
  
      if-nez v2, :cond_1
  
+     .line 1412
      :cond_0
      :goto_0
      return v1
***************
*** 56,72 ****
--- 62,82 ----
      :cond_1
      move-object v0, p1
  
+     .line 1406
      check-cast v0, Landroid/hardware/Camera$Area;
  
+     .line 1407
      .local v0, a:Landroid/hardware/Camera$Area;
      iget-object v2, p0, Landroid/hardware/Camera$Area;->rect:Landroid/graphics/Rect;
  
      if-nez v2, :cond_3
  
+     .line 1408
      iget-object v2, v0, Landroid/hardware/Camera$Area;->rect:Landroid/graphics/Rect;
  
      if-nez v2, :cond_0
  
+     .line 1412
      :cond_2
      iget v2, p0, Landroid/hardware/Camera$Area;->weight:I
  
***************
*** 78,83 ****
--- 88,94 ----
  
      goto :goto_0
  
+     .line 1410
      :cond_3
      iget-object v2, p0, Landroid/hardware/Camera$Area;->rect:Landroid/graphics/Rect;
  
