*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/SensorManager$ListenerDelegate$1.smali	2012-03-23 19:36:03.822750692 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/SensorManager$ListenerDelegate$1.smali.original	2012-03-23 19:34:44.957312582 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .parameter
  
      .prologue
+     .line 553
      iput-object p1, p0, Landroid/hardware/SensorManager$ListenerDelegate$1;->this$1:Landroid/hardware/SensorManager$ListenerDelegate;
  
      iput-object p3, p0, Landroid/hardware/SensorManager$ListenerDelegate$1;->val$this$0:Landroid/hardware/SensorManager;
***************
*** 44,53 ****
--- 45,56 ----
      .parameter "msg"
  
      .prologue
+     .line 556
      iget-object v2, p1, Landroid/os/Message;->obj:Ljava/lang/Object;
  
      check-cast v2, Landroid/hardware/SensorEvent;
  
+     .line 557
      .local v2, t:Landroid/hardware/SensorEvent;
      iget-object v3, v2, Landroid/hardware/SensorEvent;->sensor:Landroid/hardware/Sensor;
  
***************
*** 55,60 ****
--- 58,64 ----
  
      move-result v1
  
+     .line 559
      .local v1, handle:I
      iget-object v3, v2, Landroid/hardware/SensorEvent;->sensor:Landroid/hardware/Sensor;
  
***************
*** 64,69 ****
--- 68,74 ----
  
      packed-switch v3, :pswitch_data_0
  
+     .line 572
      iget-object v3, p0, Landroid/hardware/SensorManager$ListenerDelegate$1;->this$1:Landroid/hardware/SensorManager$ListenerDelegate;
  
      iget-object v3, v3, Landroid/hardware/SensorManager$ListenerDelegate;->mFirstEvent:Landroid/util/SparseBooleanArray;
***************
*** 74,79 ****
--- 79,85 ----
  
      if-nez v3, :cond_0
  
+     .line 573
      iget-object v3, p0, Landroid/hardware/SensorManager$ListenerDelegate$1;->this$1:Landroid/hardware/SensorManager$ListenerDelegate;
  
      iget-object v3, v3, Landroid/hardware/SensorManager$ListenerDelegate;->mFirstEvent:Landroid/util/SparseBooleanArray;
***************
*** 82,87 ****
--- 88,94 ----
  
      invoke-virtual {v3, v1, v4}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 574
      iget-object v3, p0, Landroid/hardware/SensorManager$ListenerDelegate$1;->this$1:Landroid/hardware/SensorManager$ListenerDelegate;
  
      #getter for: Landroid/hardware/SensorManager$ListenerDelegate;->mSensorEventListener:Landroid/hardware/SensorEventListener;
***************
*** 95,100 ****
--- 102,108 ----
  
      invoke-interface {v3, v4, v5}, Landroid/hardware/SensorEventListener;->onAccuracyChanged(Landroid/hardware/Sensor;I)V
  
+     .line 580
      :cond_0
      :goto_0
      iget-object v3, p0, Landroid/hardware/SensorManager$ListenerDelegate$1;->this$1:Landroid/hardware/SensorManager$ListenerDelegate;
***************
*** 106,119 ****
--- 114,130 ----
  
      invoke-interface {v3, v2}, Landroid/hardware/SensorEventListener;->onSensorChanged(Landroid/hardware/SensorEvent;)V
  
+     .line 581
      invoke-static {}, Landroid/hardware/SensorManager;->access$200()Landroid/hardware/SensorManager$SensorEventPool;
  
      move-result-object v3
  
      invoke-virtual {v3, v2}, Landroid/hardware/SensorManager$SensorEventPool;->returnToPool(Landroid/hardware/SensorEvent;)V
  
+     .line 582
      return-void
  
+     .line 564
      :pswitch_0
      iget-object v3, p0, Landroid/hardware/SensorManager$ListenerDelegate$1;->this$1:Landroid/hardware/SensorManager$ListenerDelegate;
  
***************
*** 123,128 ****
--- 134,140 ----
  
      move-result v0
  
+     .line 565
      .local v0, accuracy:I
      iget v3, v2, Landroid/hardware/SensorEvent;->accuracy:I
  
***************
*** 132,137 ****
--- 144,150 ----
  
      if-eq v0, v3, :cond_0
  
+     .line 566
      iget-object v3, p0, Landroid/hardware/SensorManager$ListenerDelegate$1;->this$1:Landroid/hardware/SensorManager$ListenerDelegate;
  
      iget-object v3, v3, Landroid/hardware/SensorManager$ListenerDelegate;->mSensorAccuracies:Landroid/util/SparseIntArray;
***************
*** 140,145 ****
--- 153,159 ----
  
      invoke-virtual {v3, v1, v4}, Landroid/util/SparseIntArray;->put(II)V
  
+     .line 567
      iget-object v3, p0, Landroid/hardware/SensorManager$ListenerDelegate$1;->this$1:Landroid/hardware/SensorManager$ListenerDelegate;
  
      #getter for: Landroid/hardware/SensorManager$ListenerDelegate;->mSensorEventListener:Landroid/hardware/SensorEventListener;
***************
*** 155,160 ****
--- 169,175 ----
  
      goto :goto_0
  
+     .line 559
      nop
  
      :pswitch_data_0
