*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/Sensor.smali	2012-03-23 19:36:03.790750108 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/Sensor.smali.original	2012-03-23 19:34:44.957312582 -0500
***************
*** 66,73 ****
--- 66,75 ----
      .locals 0
  
      .prologue
+     .line 137
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 138
      return-void
  .end method
  
***************
*** 77,82 ****
--- 79,85 ----
      .locals 1
  
      .prologue
+     .line 199
      iget v0, p0, Landroid/hardware/Sensor;->mHandle:I
  
      return v0
***************
*** 86,91 ****
--- 89,95 ----
      .locals 1
  
      .prologue
+     .line 212
      iget v0, p0, Landroid/hardware/Sensor;->mLegacyType:I
  
      return v0
***************
*** 95,100 ****
--- 99,105 ----
      .locals 1
  
      .prologue
+     .line 172
      iget v0, p0, Landroid/hardware/Sensor;->mMaxRange:F
  
      return v0
***************
*** 104,109 ****
--- 109,115 ----
      .locals 1
  
      .prologue
+     .line 195
      iget v0, p0, Landroid/hardware/Sensor;->mMinDelay:I
  
      return v0
***************
*** 113,118 ****
--- 119,125 ----
      .locals 1
  
      .prologue
+     .line 144
      iget-object v0, p0, Landroid/hardware/Sensor;->mName:Ljava/lang/String;
  
      return-object v0
***************
*** 122,127 ****
--- 129,135 ----
      .locals 1
  
      .prologue
+     .line 186
      iget v0, p0, Landroid/hardware/Sensor;->mPower:F
  
      return v0
***************
*** 131,136 ****
--- 139,145 ----
      .locals 1
  
      .prologue
+     .line 179
      iget v0, p0, Landroid/hardware/Sensor;->mResolution:F
  
      return v0
***************
*** 140,145 ****
--- 149,155 ----
      .locals 1
  
      .prologue
+     .line 158
      iget v0, p0, Landroid/hardware/Sensor;->mType:I
  
      return v0
***************
*** 149,154 ****
--- 159,165 ----
      .locals 1
  
      .prologue
+     .line 151
      iget-object v0, p0, Landroid/hardware/Sensor;->mVendor:Ljava/lang/String;
  
      return-object v0
***************
*** 158,163 ****
--- 169,175 ----
      .locals 1
  
      .prologue
+     .line 165
      iget v0, p0, Landroid/hardware/Sensor;->mVersion:I
  
      return v0
***************
*** 168,175 ****
--- 180,189 ----
      .parameter "legacyType"
  
      .prologue
+     .line 208
      iput p1, p0, Landroid/hardware/Sensor;->mLegacyType:I
  
+     .line 209
      return-void
  .end method
  
***************
*** 179,187 ****
--- 193,204 ----
      .parameter "res"
  
      .prologue
+     .line 203
      iput p1, p0, Landroid/hardware/Sensor;->mMaxRange:F
  
+     .line 204
      iput p2, p0, Landroid/hardware/Sensor;->mResolution:F
  
+     .line 205
      return-void
  .end method
