*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/SensorManager$ListenerDelegate.smali	2012-03-23 19:36:03.901752133 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/hardware/SensorManager$ListenerDelegate.smali.original	2012-03-23 19:34:44.958312600 -0500
***************
*** 48,89 ****
--- 48,97 ----
      .parameter "handler"
  
      .prologue
+     .line 547
      iput-object p1, p0, Landroid/hardware/SensorManager$ListenerDelegate;->this$0:Landroid/hardware/SensorManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 541
      new-instance v1, Ljava/util/ArrayList;
  
      invoke-direct {v1}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v1, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensorList:Ljava/util/ArrayList;
  
+     .line 543
      new-instance v1, Landroid/util/SparseBooleanArray;
  
      invoke-direct {v1}, Landroid/util/SparseBooleanArray;-><init>()V
  
      iput-object v1, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensors:Landroid/util/SparseBooleanArray;
  
+     .line 544
      new-instance v1, Landroid/util/SparseBooleanArray;
  
      invoke-direct {v1}, Landroid/util/SparseBooleanArray;-><init>()V
  
      iput-object v1, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mFirstEvent:Landroid/util/SparseBooleanArray;
  
+     .line 545
      new-instance v1, Landroid/util/SparseIntArray;
  
      invoke-direct {v1}, Landroid/util/SparseIntArray;-><init>()V
  
      iput-object v1, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensorAccuracies:Landroid/util/SparseIntArray;
  
+     .line 548
      iput-object p2, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensorEventListener:Landroid/hardware/SensorEventListener;
  
+     .line 549
      if-eqz p4, :cond_0
  
      invoke-virtual {p4}, Landroid/os/Handler;->getLooper()Landroid/os/Looper;
  
      move-result-object v0
  
+     .line 553
      .local v0, looper:Landroid/os/Looper;
      :goto_0
      new-instance v1, Landroid/hardware/SensorManager$ListenerDelegate$1;
***************
*** 92,101 ****
--- 100,112 ----
  
      iput-object v1, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mHandler:Landroid/os/Handler;
  
+     .line 584
      invoke-virtual {p0, p3}, Landroid/hardware/SensorManager$ListenerDelegate;->addSensor(Landroid/hardware/Sensor;)V
  
+     .line 585
      return-void
  
+     .line 549
      .end local v0           #looper:Landroid/os/Looper;
      :cond_0
      iget-object v0, p1, Landroid/hardware/SensorManager;->mMainLooper:Landroid/os/Looper;
***************
*** 108,113 ****
--- 119,125 ----
      .parameter "x0"
  
      .prologue
+     .line 539
      iget-object v0, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensorEventListener:Landroid/hardware/SensorEventListener;
  
      return-object v0
***************
*** 120,125 ****
--- 132,138 ----
      .parameter "sensor"
  
      .prologue
+     .line 592
      iget-object v0, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensors:Landroid/util/SparseBooleanArray;
  
      invoke-virtual {p1}, Landroid/hardware/Sensor;->getHandle()I
***************
*** 130,139 ****
--- 143,154 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/util/SparseBooleanArray;->put(IZ)V
  
+     .line 593
      iget-object v0, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensorList:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 594
      return-void
  .end method
  
***************
*** 141,146 ****
--- 156,162 ----
      .locals 1
  
      .prologue
+     .line 588
      iget-object v0, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensorEventListener:Landroid/hardware/SensorEventListener;
  
      return-object v0
***************
*** 159,164 ****
--- 175,181 ----
      .end annotation
  
      .prologue
+     .line 604
      iget-object v0, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensorList:Ljava/util/ArrayList;
  
      return-object v0
***************
*** 169,174 ****
--- 186,192 ----
      .parameter "sensor"
  
      .prologue
+     .line 601
      iget-object v0, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensors:Landroid/util/SparseBooleanArray;
  
      invoke-virtual {p1}, Landroid/hardware/Sensor;->getHandle()I
***************
*** 196,201 ****
--- 214,220 ----
  
      const/4 v5, 0x0
  
+     .line 608
      invoke-static {}, Landroid/hardware/SensorManager;->access$200()Landroid/hardware/SensorManager$SensorEventPool;
  
      move-result-object v3
***************
*** 204,246 ****
--- 223,277 ----
  
      move-result-object v1
  
+     .line 609
      .local v1, t:Landroid/hardware/SensorEvent;
      iget-object v2, v1, Landroid/hardware/SensorEvent;->values:[F
  
+     .line 610
      .local v2, v:[F
      aget v3, p2, v5
  
      aput v3, v2, v5
  
+     .line 611
      aget v3, p2, v4
  
      aput v3, v2, v4
  
+     .line 612
      aget v3, p2, v6
  
      aput v3, v2, v6
  
+     .line 613
      aget-wide v3, p3, v5
  
      iput-wide v3, v1, Landroid/hardware/SensorEvent;->timestamp:J
  
+     .line 614
      iput p4, v1, Landroid/hardware/SensorEvent;->accuracy:I
  
+     .line 615
      iput-object p1, v1, Landroid/hardware/SensorEvent;->sensor:Landroid/hardware/Sensor;
  
+     .line 616
      invoke-static {}, Landroid/os/Message;->obtain()Landroid/os/Message;
  
      move-result-object v0
  
+     .line 617
      .local v0, msg:Landroid/os/Message;
      iput v5, v0, Landroid/os/Message;->what:I
  
+     .line 618
      iput-object v1, v0, Landroid/os/Message;->obj:Ljava/lang/Object;
  
+     .line 619
      iget-object v3, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mHandler:Landroid/os/Handler;
  
      invoke-virtual {v3, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
  
+     .line 620
      return-void
  .end method
  
***************
*** 249,254 ****
--- 280,286 ----
      .parameter "sensor"
  
      .prologue
+     .line 596
      iget-object v0, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensors:Landroid/util/SparseBooleanArray;
  
      invoke-virtual {p1}, Landroid/hardware/Sensor;->getHandle()I
***************
*** 257,266 ****
--- 289,300 ----
  
      invoke-virtual {v0, v1}, Landroid/util/SparseBooleanArray;->delete(I)V
  
+     .line 597
      iget-object v0, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensorList:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->remove(Ljava/lang/Object;)Z
  
+     .line 598
      iget-object v0, p0, Landroid/hardware/SensorManager$ListenerDelegate;->mSensors:Landroid/util/SparseBooleanArray;
  
      invoke-virtual {v0}, Landroid/util/SparseBooleanArray;->size()I
