*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/ddm/DdmHandleNativeHeap.smali	2012-03-23 19:37:28.863301404 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/ddm/DdmHandleNativeHeap.smali.original	2012-03-23 19:34:45.999331583 -0500
***************
*** 14,19 ****
--- 14,20 ----
      .locals 1
  
      .prologue
+     .line 30
      const-string v0, "NHGT"
  
      invoke-static {v0}, Landroid/ddm/DdmHandleNativeHeap;->type(Ljava/lang/String;)I
***************
*** 22,27 ****
--- 23,29 ----
  
      sput v0, Landroid/ddm/DdmHandleNativeHeap;->CHUNK_NHGT:I
  
+     .line 32
      new-instance v0, Landroid/ddm/DdmHandleNativeHeap;
  
      invoke-direct {v0}, Landroid/ddm/DdmHandleNativeHeap;-><init>()V
***************
*** 35,40 ****
--- 37,43 ----
      .locals 0
  
      .prologue
+     .line 36
      invoke-direct {p0}, Lorg/apache/harmony/dalvik/ddmc/ChunkHandler;-><init>()V
  
      return-void
***************
*** 48,60 ****
--- 51,66 ----
      .parameter "request"
  
      .prologue
+     .line 78
      invoke-direct {p0}, Landroid/ddm/DdmHandleNativeHeap;->getLeakInfo()[B
  
      move-result-object v0
  
+     .line 80
      .local v0, data:[B
      if-eqz v0, :cond_0
  
+     .line 82
      const-string v1, "ddm-nativeheap"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 85,90 ****
--- 91,97 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 83
      new-instance v1, Lorg/apache/harmony/dalvik/ddmc/Chunk;
  
      const-string v2, "NHGT"
***************
*** 99,104 ****
--- 106,112 ----
  
      invoke-direct {v1, v2, v0, v3, v4}, Lorg/apache/harmony/dalvik/ddmc/Chunk;-><init>(I[BII)V
  
+     .line 86
      :goto_0
      return-object v1
  
***************
*** 118,129 ****
--- 126,139 ----
      .locals 2
  
      .prologue
+     .line 42
      sget v0, Landroid/ddm/DdmHandleNativeHeap;->CHUNK_NHGT:I
  
      sget-object v1, Landroid/ddm/DdmHandleNativeHeap;->mInstance:Landroid/ddm/DdmHandleNativeHeap;
  
      invoke-static {v0, v1}, Lorg/apache/harmony/dalvik/ddmc/DdmServer;->registerHandler(ILorg/apache/harmony/dalvik/ddmc/ChunkHandler;)V
  
+     .line 43
      return-void
  .end method
  
***************
*** 133,138 ****
--- 143,149 ----
      .locals 0
  
      .prologue
+     .line 49
      return-void
  .end method
  
***************
*** 140,145 ****
--- 151,157 ----
      .locals 0
  
      .prologue
+     .line 55
      return-void
  .end method
  
***************
*** 148,153 ****
--- 160,166 ----
      .parameter "request"
  
      .prologue
+     .line 61
      const-string v1, "ddm-nativeheap"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 182,200 ****
--- 195,217 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 62
      iget v0, p1, Lorg/apache/harmony/dalvik/ddmc/Chunk;->type:I
  
+     .line 64
      .local v0, type:I
      sget v1, Landroid/ddm/DdmHandleNativeHeap;->CHUNK_NHGT:I
  
      if-ne v0, v1, :cond_0
  
+     .line 65
      invoke-direct {p0, p1}, Landroid/ddm/DdmHandleNativeHeap;->handleNHGT(Lorg/apache/harmony/dalvik/ddmc/Chunk;)Lorg/apache/harmony/dalvik/ddmc/Chunk;
  
      move-result-object v1
  
      return-object v1
  
+     .line 67
      :cond_0
      new-instance v1, Ljava/lang/RuntimeException;
  
