*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/ddm/DdmHandleExit.smali	2012-03-23 19:37:28.884301786 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/ddm/DdmHandleExit.smali.original	2012-03-23 19:34:46.000331602 -0500
***************
*** 14,19 ****
--- 14,20 ----
      .locals 1
  
      .prologue
+     .line 30
      const-string v0, "EXIT"
  
      invoke-static {v0}, Landroid/ddm/DdmHandleExit;->type(Ljava/lang/String;)I
***************
*** 22,27 ****
--- 23,29 ----
  
      sput v0, Landroid/ddm/DdmHandleExit;->CHUNK_EXIT:I
  
+     .line 32
      new-instance v0, Landroid/ddm/DdmHandleExit;
  
      invoke-direct {v0}, Landroid/ddm/DdmHandleExit;-><init>()V
***************
*** 35,40 ****
--- 37,43 ----
      .locals 0
  
      .prologue
+     .line 36
      invoke-direct {p0}, Lorg/apache/harmony/dalvik/ddmc/ChunkHandler;-><init>()V
  
      return-void
***************
*** 44,55 ****
--- 47,60 ----
      .locals 2
  
      .prologue
+     .line 42
      sget v0, Landroid/ddm/DdmHandleExit;->CHUNK_EXIT:I
  
      sget-object v1, Landroid/ddm/DdmHandleExit;->mInstance:Landroid/ddm/DdmHandleExit;
  
      invoke-static {v0, v1}, Lorg/apache/harmony/dalvik/ddmc/DdmServer;->registerHandler(ILorg/apache/harmony/dalvik/ddmc/ChunkHandler;)V
  
+     .line 43
      return-void
  .end method
  
***************
*** 59,64 ****
--- 64,70 ----
      .locals 0
  
      .prologue
+     .line 49
      return-void
  .end method
  
***************
*** 66,71 ****
--- 72,78 ----
      .locals 0
  
      .prologue
+     .line 55
      return-void
  .end method
  
***************
*** 74,88 ****
--- 81,98 ----
      .parameter "request"
  
      .prologue
+     .line 67
      invoke-static {p1}, Landroid/ddm/DdmHandleExit;->wrapChunk(Lorg/apache/harmony/dalvik/ddmc/Chunk;)Ljava/nio/ByteBuffer;
  
      move-result-object v0
  
+     .line 69
      .local v0, in:Ljava/nio/ByteBuffer;
      invoke-virtual {v0}, Ljava/nio/ByteBuffer;->getInt()I
  
      move-result v1
  
+     .line 71
      .local v1, statusCode:I
      invoke-static {}, Ljava/lang/Runtime;->getRuntime()Ljava/lang/Runtime;
  
***************
*** 90,95 ****
--- 100,106 ----
  
      invoke-virtual {v2, v1}, Ljava/lang/Runtime;->halt(I)V
  
+     .line 74
      const/4 v2, 0x0
  
      return-object v2
