*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/inputmethod/ComposingText.smali	2012-03-23 19:37:22.064177422 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/inputmethod/ComposingText.smali.original	2012-03-23 19:34:45.891329615 -0500
***************
*** 11,16 ****
--- 11,17 ----
      .locals 0
  
      .prologue
+     .line 39
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
