*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/inputmethod/InputMethodManager$ControlledInputConnectionWrapper.smali	2012-03-23 19:37:22.344182527 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/view/inputmethod/InputMethodManager$ControlledInputConnectionWrapper.smali.original	2012-03-23 19:34:45.894329669 -0500
***************
*** 26,35 ****
--- 26,38 ----
      .parameter "inputMethodManager"
  
      .prologue
+     .line 392
      invoke-direct {p0, p1, p2}, Lcom/android/internal/view/IInputConnectionWrapper;-><init>(Landroid/os/Looper;Landroid/view/inputmethod/InputConnection;)V
  
+     .line 393
      iput-object p3, p0, Landroid/view/inputmethod/InputMethodManager$ControlledInputConnectionWrapper;->mParentInputMethodManager:Landroid/view/inputmethod/InputMethodManager;
  
+     .line 394
      return-void
  .end method
  
***************
*** 39,44 ****
--- 42,48 ----
      .locals 1
  
      .prologue
+     .line 398
      iget-object v0, p0, Landroid/view/inputmethod/InputMethodManager$ControlledInputConnectionWrapper;->mParentInputMethodManager:Landroid/view/inputmethod/InputMethodManager;
  
      iget-boolean v0, v0, Landroid/view/inputmethod/InputMethodManager;->mActive:Z
