*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/format/DateUtils.smali	2012-03-23 19:35:19.640945039 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/format/DateUtils.smali.original	2012-03-23 19:34:44.487304012 -0500
***************
*** 147,206 ****
--- 147,216 ----
  
      const/4 v1, 0x7
  
+     .line 38
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      sput-object v0, Landroid/text/format/DateUtils;->sLock:Ljava/lang/Object;
  
+     .line 39
      new-array v0, v1, [I
  
      fill-array-data v0, :array_0
  
      sput-object v0, Landroid/text/format/DateUtils;->sDaysLong:[I
  
+     .line 48
      new-array v0, v1, [I
  
      fill-array-data v0, :array_1
  
      sput-object v0, Landroid/text/format/DateUtils;->sDaysMedium:[I
  
+     .line 57
      new-array v0, v1, [I
  
      fill-array-data v0, :array_2
  
      sput-object v0, Landroid/text/format/DateUtils;->sDaysShort:[I
  
+     .line 66
      new-array v0, v1, [I
  
      fill-array-data v0, :array_3
  
      sput-object v0, Landroid/text/format/DateUtils;->sDaysShortest:[I
  
+     .line 75
      new-array v0, v2, [I
  
      fill-array-data v0, :array_4
  
      sput-object v0, Landroid/text/format/DateUtils;->sMonthsStandaloneLong:[I
  
+     .line 89
      new-array v0, v2, [I
  
      fill-array-data v0, :array_5
  
      sput-object v0, Landroid/text/format/DateUtils;->sMonthsLong:[I
  
+     .line 103
      new-array v0, v2, [I
  
      fill-array-data v0, :array_6
  
      sput-object v0, Landroid/text/format/DateUtils;->sMonthsMedium:[I
  
+     .line 117
      new-array v0, v2, [I
  
      fill-array-data v0, :array_7
  
      sput-object v0, Landroid/text/format/DateUtils;->sMonthsShortest:[I
  
+     .line 131
      const/4 v0, 0x2
  
      new-array v0, v0, [I
***************
*** 209,220 ****
--- 219,232 ----
  
      sput-object v0, Landroid/text/format/DateUtils;->sAmPm:[I
  
+     .line 210
      new-array v0, v3, [I
  
      fill-array-data v0, :array_9
  
      sput-object v0, Landroid/text/format/DateUtils;->sameYearTable:[I
  
+     .line 236
      new-array v0, v3, [I
  
      fill-array-data v0, :array_a
***************
*** 223,228 ****
--- 235,241 ----
  
      return-void
  
+     .line 39
      nop
  
      :array_0
***************
*** 236,241 ****
--- 249,255 ----
          0x60t 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 48
      :array_1
      .array-data 0x4
          0x61t 0x0t 0x4t 0x1t
***************
*** 247,252 ****
--- 261,267 ----
          0x67t 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 57
      :array_2
      .array-data 0x4
          0x68t 0x0t 0x4t 0x1t
***************
*** 258,263 ****
--- 273,279 ----
          0x6et 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 66
      :array_3
      .array-data 0x4
          0x6ft 0x0t 0x4t 0x1t
***************
*** 269,274 ****
--- 285,291 ----
          0x75t 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 75
      :array_4
      .array-data 0x4
          0x2at 0x0t 0x4t 0x1t
***************
*** 285,290 ****
--- 302,308 ----
          0x35t 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 89
      :array_5
      .array-data 0x4
          0x36t 0x0t 0x4t 0x1t
***************
*** 301,306 ****
--- 319,325 ----
          0x41t 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 103
      :array_6
      .array-data 0x4
          0x42t 0x0t 0x4t 0x1t
***************
*** 317,322 ****
--- 336,342 ----
          0x4dt 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 117
      :array_7
      .array-data 0x4
          0x4et 0x0t 0x4t 0x1t
***************
*** 333,344 ****
--- 353,366 ----
          0x59t 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 131
      :array_8
      .array-data 0x4
          0x76t 0x0t 0x4t 0x1t
          0x77t 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 210
      :array_9
      .array-data 0x4
          0x9ft 0x0t 0x4t 0x1t
***************
*** 359,364 ****
--- 381,387 ----
          0x95t 0x0t 0x4t 0x1t
      .end array-data
  
+     .line 236
      :array_a
      .array-data 0x4
          0xaat 0x0t 0x4t 0x1t
***************
*** 384,389 ****
--- 407,413 ----
      .locals 0
  
      .prologue
+     .line 36
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 395,408 ****
--- 419,435 ----
      .parameter "rval"
  
      .prologue
+     .line 964
      invoke-virtual {p0}, Ljava/util/Calendar;->clear()V
  
+     .line 965
      invoke-virtual {p1}, Ljava/util/Calendar;->getTimeInMillis()J
  
      move-result-wide v0
  
      invoke-virtual {p0, v0, v1}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 966
      return-void
  .end method
  
***************
*** 414,419 ****
--- 441,447 ----
      .parameter "flags"
  
      .prologue
+     .line 989
      new-instance v1, Ljava/util/Formatter;
  
      new-instance v0, Ljava/lang/StringBuilder;
***************
*** 437,442 ****
--- 465,471 ----
  
      move v6, p5
  
+     .line 990
      invoke-static/range {v0 .. v6}, Landroid/text/format/DateUtils;->formatDateRange(Landroid/content/Context;Ljava/util/Formatter;JJI)Ljava/util/Formatter;
  
      move-result-object v0
***************
*** 457,462 ****
--- 486,492 ----
      .parameter "flags"
  
      .prologue
+     .line 1013
      const/4 v7, 0x0
  
      move-object v0, p0
***************
*** 486,495 ****
--- 516,527 ----
      .parameter "timeZone"
  
      .prologue
+     .line 1173
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v38
  
+     .line 1174
      .local v38, res:Landroid/content/res/Resources;
      and-int/lit8 v62, p6, 0x1
  
***************
*** 497,502 ****
--- 529,535 ----
  
      const/16 v41, 0x1
  
+     .line 1175
      .local v41, showTime:Z
      :goto_0
      and-int/lit8 v62, p6, 0x2
***************
*** 505,510 ****
--- 538,544 ----
  
      const/16 v42, 0x1
  
+     .line 1176
      .local v42, showWeekDay:Z
      :goto_1
      and-int/lit8 v62, p6, 0x4
***************
*** 513,518 ****
--- 547,553 ----
  
      const/16 v43, 0x1
  
+     .line 1177
      .local v43, showYear:Z
      :goto_2
      and-int/lit8 v62, p6, 0x8
***************
*** 521,526 ****
--- 556,562 ----
  
      const/16 v36, 0x1
  
+     .line 1178
      .local v36, noYear:Z
      :goto_3
      move/from16 v0, p6
***************
*** 533,538 ****
--- 569,575 ----
  
      const/16 v60, 0x1
  
+     .line 1179
      .local v60, useUTC:Z
      :goto_4
      const v62, 0x88000
***************
*** 543,548 ****
--- 580,586 ----
  
      const/4 v6, 0x1
  
+     .line 1180
      .local v6, abbrevWeekDay:Z
      :goto_5
      const/high16 v62, 0x9
***************
*** 553,558 ****
--- 591,597 ----
  
      const/4 v4, 0x1
  
+     .line 1181
      .local v4, abbrevMonth:Z
      :goto_6
      and-int/lit8 v62, p6, 0x20
***************
*** 561,566 ****
--- 600,606 ----
  
      const/16 v34, 0x1
  
+     .line 1182
      .local v34, noMonthDay:Z
      :goto_7
      const/high16 v62, 0x2
***************
*** 571,576 ****
--- 611,617 ----
  
      const/16 v37, 0x1
  
+     .line 1187
      .local v37, numericDate:Z
      :goto_8
      cmp-long v62, p2, p4
***************
*** 579,588 ****
--- 620,631 ----
  
      const/16 v31, 0x1
  
+     .line 1190
      .local v31, isInstant:Z
      :goto_9
      if-eqz p7, :cond_f
  
+     .line 1191
      new-instance v44, Landroid/text/format/Time;
  
      move-object/from16 v0, v44
***************
*** 591,596 ****
--- 634,640 ----
  
      invoke-direct {v0, v1}, Landroid/text/format/Time;-><init>(Ljava/lang/String;)V
  
+     .line 1197
      .local v44, startDate:Landroid/text/format/Time;
      :goto_a
      move-object/from16 v0, v44
***************
*** 599,611 ****
--- 643,659 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/text/format/Time;->set(J)V
  
+     .line 1201
      if-eqz v31, :cond_11
  
+     .line 1202
      move-object/from16 v14, v44
  
+     .line 1203
      .local v14, endDate:Landroid/text/format/Time;
      const/4 v12, 0x0
  
+     .line 1232
      .local v12, dayDistance:I
      :goto_b
      if-nez v31, :cond_1
***************
*** 636,641 ****
--- 684,690 ----
  
      if-gt v12, v0, :cond_1
  
+     .line 1235
      :cond_0
      iget v0, v14, Landroid/text/format/Time;->monthDay:I
  
***************
*** 647,658 ****
--- 696,709 ----
  
      iput v0, v14, Landroid/text/format/Time;->monthDay:I
  
+     .line 1236
      const/16 v62, 0x1
  
      move/from16 v0, v62
  
      invoke-virtual {v14, v0}, Landroid/text/format/Time;->normalize(Z)J
  
+     .line 1239
      :cond_1
      move-object/from16 v0, v44
  
***************
*** 660,665 ****
--- 711,717 ----
  
      move/from16 v46, v0
  
+     .line 1240
      .local v46, startDay:I
      move-object/from16 v0, v44
  
***************
*** 667,672 ****
--- 719,725 ----
  
      move/from16 v49, v0
  
+     .line 1241
      .local v49, startMonthNum:I
      move-object/from16 v0, v44
  
***************
*** 674,710 ****
--- 727,773 ----
  
      move/from16 v55, v0
  
+     .line 1243
      .local v55, startYear:I
      iget v0, v14, Landroid/text/format/Time;->monthDay:I
  
      move/from16 v16, v0
  
+     .line 1244
      .local v16, endDay:I
      iget v0, v14, Landroid/text/format/Time;->month:I
  
      move/from16 v19, v0
  
+     .line 1245
      .local v19, endMonthNum:I
      iget v0, v14, Landroid/text/format/Time;->year:I
  
      move/from16 v25, v0
  
+     .line 1247
      .local v25, endYear:I
      const-string v54, ""
  
+     .line 1248
      .local v54, startWeekDayString:Ljava/lang/String;
      const-string v24, ""
  
+     .line 1249
      .local v24, endWeekDayString:Ljava/lang/String;
      if-eqz v42, :cond_2
  
+     .line 1250
      const-string v61, ""
  
+     .line 1251
      .local v61, weekDayFormat:Ljava/lang/String;
      if-eqz v6, :cond_14
  
+     .line 1252
      const-string v61, "%a"
  
+     .line 1256
      :goto_c
      move-object/from16 v0, v44
  
***************
*** 714,739 ****
--- 777,809 ----
  
      move-result-object v54
  
+     .line 1257
      if-eqz v31, :cond_15
  
      move-object/from16 v24, v54
  
+     .line 1260
      .end local v61           #weekDayFormat:Ljava/lang/String;
      :cond_2
      :goto_d
      const-string v53, ""
  
+     .line 1261
      .local v53, startTimeString:Ljava/lang/String;
      const-string v23, ""
  
+     .line 1262
      .local v23, endTimeString:Ljava/lang/String;
      if-eqz v41, :cond_4
  
+     .line 1263
      const-string v52, ""
  
+     .line 1264
      .local v52, startTimeFormat:Ljava/lang/String;
      const-string v22, ""
  
+     .line 1265
      .local v22, endTimeFormat:Ljava/lang/String;
      move/from16 v0, p6
  
***************
*** 745,750 ****
--- 815,821 ----
  
      const/16 v28, 0x1
  
+     .line 1266
      .local v28, force24Hour:Z
      :goto_e
      and-int/lit8 v62, p6, 0x40
***************
*** 753,768 ****
--- 824,843 ----
  
      const/16 v27, 0x1
  
+     .line 1268
      .local v27, force12Hour:Z
      :goto_f
      if-eqz v28, :cond_18
  
+     .line 1269
      const/16 v59, 0x1
  
+     .line 1275
      .local v59, use24Hour:Z
      :goto_10
      if-eqz v59, :cond_1a
  
+     .line 1276
      const v62, 0x104007b
  
      move-object/from16 v0, v38
***************
*** 775,780 ****
--- 850,856 ----
  
      move-object/from16 v52, v22
  
+     .line 1347
      :cond_3
      :goto_11
      move-object/from16 v0, v44
***************
*** 785,794 ****
--- 861,872 ----
  
      move-result-object v53
  
+     .line 1348
      if-eqz v31, :cond_2e
  
      move-object/from16 v23, v53
  
+     .line 1355
      .end local v22           #endTimeFormat:Ljava/lang/String;
      .end local v27           #force12Hour:Z
      .end local v28           #force24Hour:Z
***************
*** 798,806 ****
--- 876,886 ----
      :goto_12
      if-eqz v43, :cond_2f
  
+     .line 1372
      :goto_13
      if-eqz v37, :cond_33
  
+     .line 1373
      const v62, 0x1040080
  
      move-object/from16 v0, v38
***************
*** 811,822 ****
--- 891,905 ----
  
      move-result-object v13
  
+     .line 1404
      .local v13, defaultDateFormat:Ljava/lang/String;
      :goto_14
      if-eqz v42, :cond_3c
  
+     .line 1405
      if-eqz v41, :cond_3b
  
+     .line 1406
      const v62, 0x1040099
  
      move-object/from16 v0, v38
***************
*** 827,832 ****
--- 910,916 ----
  
      move-result-object v29
  
+     .line 1418
      .local v29, fullFormat:Ljava/lang/String;
      :goto_15
      if-eqz v34, :cond_3e
***************
*** 843,848 ****
--- 927,933 ----
  
      if-ne v0, v1, :cond_3e
  
+     .line 1420
      const-string v62, "%s"
  
      const/16 v63, 0x1
***************
*** 873,881 ****
--- 958,968 ----
  
      move-result-object v62
  
+     .line 1556
      :goto_16
      return-object v62
  
+     .line 1174
      .end local v4           #abbrevMonth:Z
      .end local v6           #abbrevWeekDay:Z
      .end local v12           #dayDistance:I
***************
*** 906,969 ****
--- 993,1067 ----
  
      goto/16 :goto_0
  
+     .line 1175
      .restart local v41       #showTime:Z
      :cond_6
      const/16 v42, 0x0
  
      goto/16 :goto_1
  
+     .line 1176
      .restart local v42       #showWeekDay:Z
      :cond_7
      const/16 v43, 0x0
  
      goto/16 :goto_2
  
+     .line 1177
      .restart local v43       #showYear:Z
      :cond_8
      const/16 v36, 0x0
  
      goto/16 :goto_3
  
+     .line 1178
      .restart local v36       #noYear:Z
      :cond_9
      const/16 v60, 0x0
  
      goto/16 :goto_4
  
+     .line 1179
      .restart local v60       #useUTC:Z
      :cond_a
      const/4 v6, 0x0
  
      goto/16 :goto_5
  
+     .line 1180
      .restart local v6       #abbrevWeekDay:Z
      :cond_b
      const/4 v4, 0x0
  
      goto/16 :goto_6
  
+     .line 1181
      .restart local v4       #abbrevMonth:Z
      :cond_c
      const/16 v34, 0x0
  
      goto/16 :goto_7
  
+     .line 1182
      .restart local v34       #noMonthDay:Z
      :cond_d
      const/16 v37, 0x0
  
      goto/16 :goto_8
  
+     .line 1187
      .restart local v37       #numericDate:Z
      :cond_e
      const/16 v31, 0x0
  
      goto/16 :goto_9
  
+     .line 1192
      .restart local v31       #isInstant:Z
      :cond_f
      if-eqz v60, :cond_10
  
+     .line 1193
      new-instance v44, Landroid/text/format/Time;
  
      const-string v62, "UTC"
***************
*** 977,982 ****
--- 1075,1081 ----
      .restart local v44       #startDate:Landroid/text/format/Time;
      goto/16 :goto_a
  
+     .line 1195
      .end local v44           #startDate:Landroid/text/format/Time;
      :cond_10
      new-instance v44, Landroid/text/format/Time;
***************
*** 986,1006 ****
--- 1085,1109 ----
      .restart local v44       #startDate:Landroid/text/format/Time;
      goto/16 :goto_a
  
+     .line 1205
      :cond_11
      if-eqz p7, :cond_12
  
+     .line 1206
      new-instance v14, Landroid/text/format/Time;
  
      move-object/from16 v0, p7
  
      invoke-direct {v14, v0}, Landroid/text/format/Time;-><init>(Ljava/lang/String;)V
  
+     .line 1212
      .restart local v14       #endDate:Landroid/text/format/Time;
      :goto_17
      move-wide/from16 v0, p4
  
      invoke-virtual {v14, v0, v1}, Landroid/text/format/Time;->set(J)V
  
+     .line 1213
      move-object/from16 v0, v44
  
      iget-wide v0, v0, Landroid/text/format/Time;->gmtoff:J
***************
*** 1015,1020 ****
--- 1118,1124 ----
  
      move-result v47
  
+     .line 1214
      .local v47, startJulianDay:I
      iget-wide v0, v14, Landroid/text/format/Time;->gmtoff:J
  
***************
*** 1028,1039 ****
--- 1132,1145 ----
  
      move-result v17
  
+     .line 1215
      .local v17, endJulianDay:I
      sub-int v12, v17, v47
  
      .restart local v12       #dayDistance:I
      goto/16 :goto_b
  
+     .line 1207
      .end local v12           #dayDistance:I
      .end local v14           #endDate:Landroid/text/format/Time;
      .end local v17           #endJulianDay:I
***************
*** 1041,1046 ****
--- 1147,1153 ----
      :cond_12
      if-eqz v60, :cond_13
  
+     .line 1208
      new-instance v14, Landroid/text/format/Time;
  
      const-string v62, "UTC"
***************
*** 1052,1057 ****
--- 1159,1165 ----
      .restart local v14       #endDate:Landroid/text/format/Time;
      goto :goto_17
  
+     .line 1210
      .end local v14           #endDate:Landroid/text/format/Time;
      :cond_13
      new-instance v14, Landroid/text/format/Time;
***************
*** 1061,1066 ****
--- 1169,1175 ----
      .restart local v14       #endDate:Landroid/text/format/Time;
      goto :goto_17
  
+     .line 1254
      .restart local v12       #dayDistance:I
      .restart local v16       #endDay:I
      .restart local v19       #endMonthNum:I
***************
*** 1076,1081 ****
--- 1185,1191 ----
  
      goto/16 :goto_c
  
+     .line 1257
      :cond_15
      move-object/from16 v0, v61
  
***************
*** 1085,1090 ****
--- 1195,1201 ----
  
      goto/16 :goto_d
  
+     .line 1265
      .end local v61           #weekDayFormat:Ljava/lang/String;
      .restart local v22       #endTimeFormat:Ljava/lang/String;
      .restart local v23       #endTimeString:Ljava/lang/String;
***************
*** 1095,1115 ****
--- 1206,1230 ----
  
      goto/16 :goto_e
  
+     .line 1266
      .restart local v28       #force24Hour:Z
      :cond_17
      const/16 v27, 0x0
  
      goto/16 :goto_f
  
+     .line 1270
      .restart local v27       #force12Hour:Z
      :cond_18
      if-eqz v27, :cond_19
  
+     .line 1271
      const/16 v59, 0x0
  
      .restart local v59       #use24Hour:Z
      goto/16 :goto_10
  
+     .line 1273
      .end local v59           #use24Hour:Z
      :cond_19
      invoke-static/range {p0 .. p0}, Landroid/text/format/DateFormat;->is24HourFormat(Landroid/content/Context;)Z
***************
*** 1119,1124 ****
--- 1234,1240 ----
      .restart local v59       #use24Hour:Z
      goto/16 :goto_10
  
+     .line 1279
      :cond_1a
      const v62, 0x84000
  
***************
*** 1128,1133 ****
--- 1244,1250 ----
  
      const/4 v5, 0x1
  
+     .line 1280
      .local v5, abbrevTime:Z
      :goto_18
      move/from16 v0, p6
***************
*** 1140,1145 ****
--- 1257,1263 ----
  
      const/4 v7, 0x1
  
+     .line 1281
      .local v7, capAMPM:Z
      :goto_19
      move/from16 v0, p6
***************
*** 1152,1157 ****
--- 1270,1276 ----
  
      const/16 v35, 0x1
  
+     .line 1282
      .local v35, noNoon:Z
      :goto_1a
      move/from16 v0, p6
***************
*** 1164,1169 ****
--- 1283,1289 ----
  
      const/4 v9, 0x1
  
+     .line 1283
      .local v9, capNoon:Z
      :goto_1b
      move/from16 v0, p6
***************
*** 1176,1181 ****
--- 1296,1302 ----
  
      const/16 v33, 0x1
  
+     .line 1284
      .local v33, noMidnight:Z
      :goto_1c
      move/from16 v0, p6
***************
*** 1188,1193 ****
--- 1309,1315 ----
  
      const/4 v8, 0x1
  
+     .line 1286
      .local v8, capMidnight:Z
      :goto_1d
      move-object/from16 v0, v44
***************
*** 1208,1213 ****
--- 1330,1336 ----
  
      const/16 v51, 0x1
  
+     .line 1287
      .local v51, startOnTheHour:Z
      :goto_1e
      iget v0, v14, Landroid/text/format/Time;->minute:I
***************
*** 1224,1237 ****
--- 1347,1363 ----
  
      const/16 v21, 0x1
  
+     .line 1288
      .local v21, endOnTheHour:Z
      :goto_1f
      if-eqz v5, :cond_25
  
      if-eqz v51, :cond_25
  
+     .line 1289
      if-eqz v7, :cond_24
  
+     .line 1290
      const v62, 0x1040315
  
      move-object/from16 v0, v38
***************
*** 1242,1256 ****
--- 1368,1386 ----
  
      move-result-object v52
  
+     .line 1306
      :goto_20
      if-nez v31, :cond_1b
  
+     .line 1307
      if-eqz v5, :cond_28
  
      if-eqz v21, :cond_28
  
+     .line 1308
      if-eqz v7, :cond_27
  
+     .line 1309
      const v62, 0x1040315
  
      move-object/from16 v0, v38
***************
*** 1261,1266 ****
--- 1391,1397 ----
  
      move-result-object v22
  
+     .line 1321
      :goto_21
      iget v0, v14, Landroid/text/format/Time;->hour:I
  
***************
*** 1278,1285 ****
--- 1409,1418 ----
  
      if-nez v35, :cond_2b
  
+     .line 1322
      if-eqz v9, :cond_2a
  
+     .line 1323
      const v62, 0x104038f
  
      move-object/from16 v0, v38
***************
*** 1290,1295 ****
--- 1423,1429 ----
  
      move-result-object v22
  
+     .line 1336
      :cond_1b
      :goto_22
      move-object/from16 v0, v44
***************
*** 1310,1317 ****
--- 1444,1453 ----
  
      if-nez v35, :cond_3
  
+     .line 1337
      if-eqz v9, :cond_2d
  
+     .line 1338
      const v62, 0x104038f
  
      move-object/from16 v0, v38
***************
*** 1324,1329 ****
--- 1460,1466 ----
  
      goto/16 :goto_11
  
+     .line 1279
      .end local v5           #abbrevTime:Z
      .end local v7           #capAMPM:Z
      .end local v8           #capMidnight:Z
***************
*** 1337,1384 ****
--- 1474,1529 ----
  
      goto/16 :goto_18
  
+     .line 1280
      .restart local v5       #abbrevTime:Z
      :cond_1d
      const/4 v7, 0x0
  
      goto/16 :goto_19
  
+     .line 1281
      .restart local v7       #capAMPM:Z
      :cond_1e
      const/16 v35, 0x0
  
      goto/16 :goto_1a
  
+     .line 1282
      .restart local v35       #noNoon:Z
      :cond_1f
      const/4 v9, 0x0
  
      goto/16 :goto_1b
  
+     .line 1283
      .restart local v9       #capNoon:Z
      :cond_20
      const/16 v33, 0x0
  
      goto/16 :goto_1c
  
+     .line 1284
      .restart local v33       #noMidnight:Z
      :cond_21
      const/4 v8, 0x0
  
      goto/16 :goto_1d
  
+     .line 1286
      .restart local v8       #capMidnight:Z
      :cond_22
      const/16 v51, 0x0
  
      goto/16 :goto_1e
  
+     .line 1287
      .restart local v51       #startOnTheHour:Z
      :cond_23
      const/16 v21, 0x0
  
      goto :goto_1f
  
+     .line 1292
      .restart local v21       #endOnTheHour:Z
      :cond_24
      const v62, 0x1040314
***************
*** 1393,1401 ****
--- 1538,1548 ----
  
      goto :goto_20
  
+     .line 1295
      :cond_25
      if-eqz v7, :cond_26
  
+     .line 1296
      const v62, 0x104007d
  
      move-object/from16 v0, v38
***************
*** 1408,1413 ****
--- 1555,1561 ----
  
      goto/16 :goto_20
  
+     .line 1298
      :cond_26
      const v62, 0x104007c
  
***************
*** 1421,1426 ****
--- 1569,1575 ----
  
      goto/16 :goto_20
  
+     .line 1311
      :cond_27
      const v62, 0x1040314
  
***************
*** 1434,1442 ****
--- 1583,1593 ----
  
      goto/16 :goto_21
  
+     .line 1314
      :cond_28
      if-eqz v7, :cond_29
  
+     .line 1315
      const v62, 0x104007d
  
      move-object/from16 v0, v38
***************
*** 1449,1454 ****
--- 1600,1606 ----
  
      goto/16 :goto_21
  
+     .line 1317
      :cond_29
      const v62, 0x104007c
  
***************
*** 1462,1467 ****
--- 1614,1620 ----
  
      goto/16 :goto_21
  
+     .line 1325
      :cond_2a
      const v62, 0x104038e
  
***************
*** 1475,1480 ****
--- 1628,1634 ----
  
      goto/16 :goto_22
  
+     .line 1327
      :cond_2b
      iget v0, v14, Landroid/text/format/Time;->hour:I
  
***************
*** 1486,1493 ****
--- 1640,1649 ----
  
      if-nez v33, :cond_1b
  
+     .line 1328
      if-eqz v8, :cond_2c
  
+     .line 1329
      const v62, 0x1040391
  
      move-object/from16 v0, v38
***************
*** 1500,1505 ****
--- 1656,1662 ----
  
      goto/16 :goto_22
  
+     .line 1331
      :cond_2c
      const v62, 0x1040390
  
***************
*** 1513,1518 ****
--- 1670,1676 ----
  
      goto/16 :goto_22
  
+     .line 1340
      :cond_2d
      const v62, 0x104038e
  
***************
*** 1526,1531 ****
--- 1684,1690 ----
  
      goto/16 :goto_11
  
+     .line 1348
      .end local v5           #abbrevTime:Z
      .end local v7           #capAMPM:Z
      .end local v8           #capMidnight:Z
***************
*** 1543,1548 ****
--- 1702,1708 ----
  
      goto/16 :goto_12
  
+     .line 1359
      .end local v22           #endTimeFormat:Ljava/lang/String;
      .end local v27           #force12Hour:Z
      .end local v28           #force24Hour:Z
***************
*** 1551,1560 ****
--- 1711,1722 ----
      :cond_2f
      if-eqz v36, :cond_30
  
+     .line 1361
      const/16 v43, 0x0
  
      goto/16 :goto_13
  
+     .line 1362
      :cond_30
      move/from16 v0, v55
  
***************
*** 1562,1579 ****
--- 1724,1745 ----
  
      if-eq v0, v1, :cond_31
  
+     .line 1363
      const/16 v43, 0x1
  
      goto/16 :goto_13
  
+     .line 1366
      :cond_31
      new-instance v10, Landroid/text/format/Time;
  
      invoke-direct {v10}, Landroid/text/format/Time;-><init>()V
  
+     .line 1367
      .local v10, currentTime:Landroid/text/format/Time;
      invoke-virtual {v10}, Landroid/text/format/Time;->setToNow()V
  
+     .line 1368
      iget v0, v10, Landroid/text/format/Time;->year:I
  
      move/from16 v62, v0
***************
*** 1594,1607 ****
--- 1760,1777 ----
  
      goto :goto_23
  
+     .line 1374
      .end local v10           #currentTime:Landroid/text/format/Time;
      :cond_33
      if-eqz v43, :cond_37
  
+     .line 1375
      if-eqz v4, :cond_35
  
+     .line 1376
      if-eqz v34, :cond_34
  
+     .line 1377
      const v62, 0x104008e
  
      move-object/from16 v0, v38
***************
*** 1615,1620 ****
--- 1785,1791 ----
      .restart local v13       #defaultDateFormat:Ljava/lang/String;
      goto/16 :goto_14
  
+     .line 1379
      .end local v13           #defaultDateFormat:Ljava/lang/String;
      :cond_34
      const v62, 0x1040088
***************
*** 1630,1639 ****
--- 1801,1812 ----
      .restart local v13       #defaultDateFormat:Ljava/lang/String;
      goto/16 :goto_14
  
+     .line 1382
      .end local v13           #defaultDateFormat:Ljava/lang/String;
      :cond_35
      if-eqz v34, :cond_36
  
+     .line 1383
      const v62, 0x104008b
  
      move-object/from16 v0, v38
***************
*** 1647,1652 ****
--- 1820,1826 ----
      .restart local v13       #defaultDateFormat:Ljava/lang/String;
      goto/16 :goto_14
  
+     .line 1385
      .end local v13           #defaultDateFormat:Ljava/lang/String;
      :cond_36
      const v62, 0x1040083
***************
*** 1662,1673 ****
--- 1836,1850 ----
      .restart local v13       #defaultDateFormat:Ljava/lang/String;
      goto/16 :goto_14
  
+     .line 1389
      .end local v13           #defaultDateFormat:Ljava/lang/String;
      :cond_37
      if-eqz v4, :cond_39
  
+     .line 1390
      if-eqz v34, :cond_38
  
+     .line 1391
      const v62, 0x104008d
  
      move-object/from16 v0, v38
***************
*** 1681,1686 ****
--- 1858,1864 ----
      .restart local v13       #defaultDateFormat:Ljava/lang/String;
      goto/16 :goto_14
  
+     .line 1393
      .end local v13           #defaultDateFormat:Ljava/lang/String;
      :cond_38
      const v62, 0x104008c
***************
*** 1696,1705 ****
--- 1874,1885 ----
      .restart local v13       #defaultDateFormat:Ljava/lang/String;
      goto/16 :goto_14
  
+     .line 1396
      .end local v13           #defaultDateFormat:Ljava/lang/String;
      :cond_39
      if-eqz v34, :cond_3a
  
+     .line 1397
      const v62, 0x104008a
  
      move-object/from16 v0, v38
***************
*** 1713,1718 ****
--- 1893,1899 ----
      .restart local v13       #defaultDateFormat:Ljava/lang/String;
      goto/16 :goto_14
  
+     .line 1399
      .end local v13           #defaultDateFormat:Ljava/lang/String;
      :cond_3a
      const v62, 0x1040089
***************
*** 1728,1733 ****
--- 1909,1915 ----
      .restart local v13       #defaultDateFormat:Ljava/lang/String;
      goto/16 :goto_14
  
+     .line 1408
      :cond_3b
      const v62, 0x104009a
  
***************
*** 1742,1751 ****
--- 1924,1935 ----
      .restart local v29       #fullFormat:Ljava/lang/String;
      goto/16 :goto_15
  
+     .line 1411
      .end local v29           #fullFormat:Ljava/lang/String;
      :cond_3c
      if-eqz v41, :cond_3d
  
+     .line 1412
      const v62, 0x104009b
  
      move-object/from16 v0, v38
***************
*** 1759,1764 ****
--- 1943,1949 ----
      .restart local v29       #fullFormat:Ljava/lang/String;
      goto/16 :goto_15
  
+     .line 1414
      .end local v29           #fullFormat:Ljava/lang/String;
      :cond_3d
      const v62, 0x1040090
***************
*** 1774,1779 ****
--- 1959,1965 ----
      .restart local v29       #fullFormat:Ljava/lang/String;
      goto/16 :goto_15
  
+     .line 1423
      :cond_3e
      move/from16 v0, v55
  
***************
*** 1783,1788 ****
--- 1969,1975 ----
  
      if-eqz v34, :cond_40
  
+     .line 1427
      :cond_3f
      move-object/from16 v0, v44
  
***************
*** 1790,1800 ****
--- 1977,1989 ----
  
      move-result-object v45
  
+     .line 1428
      .local v45, startDateString:Ljava/lang/String;
      invoke-virtual {v14, v13}, Landroid/text/format/Time;->format(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v15
  
+     .line 1432
      .local v15, endDateString:Ljava/lang/String;
      const/16 v62, 0x6
  
***************
*** 1840,1852 ****
--- 2029,2044 ----
  
      goto/16 :goto_16
  
+     .line 1439
      .end local v15           #endDateString:Ljava/lang/String;
      .end local v45           #startDateString:Ljava/lang/String;
      :cond_40
      if-eqz v37, :cond_45
  
+     .line 1440
      const-string v32, "%m"
  
+     .line 1447
      .local v32, monthFormat:Ljava/lang/String;
      :goto_24
      move-object/from16 v0, v44
***************
*** 1857,1862 ****
--- 2049,2055 ----
  
      move-result-object v50
  
+     .line 1448
      .local v50, startMonthString:Ljava/lang/String;
      const-string v62, "%-d"
  
***************
*** 1868,1873 ****
--- 2061,2067 ----
  
      move-result-object v48
  
+     .line 1449
      .local v48, startMonthDayString:Ljava/lang/String;
      const-string v62, "%Y"
  
***************
*** 1879,1901 ****
--- 2073,2099 ----
  
      move-result-object v56
  
+     .line 1451
      .local v56, startYearString:Ljava/lang/String;
      if-eqz v31, :cond_47
  
      const/16 v20, 0x0
  
+     .line 1452
      .local v20, endMonthString:Ljava/lang/String;
      :goto_25
      if-eqz v31, :cond_48
  
      const/16 v18, 0x0
  
+     .line 1453
      .local v18, endMonthDayString:Ljava/lang/String;
      :goto_26
      if-eqz v31, :cond_49
  
      const/16 v26, 0x0
  
+     .line 1455
      .local v26, endYearString:Ljava/lang/String;
      :goto_27
      move/from16 v0, v49
***************
*** 1904,1941 ****
--- 2102,2147 ----
  
      if-eq v0, v1, :cond_4a
  
+     .line 1461
      const/16 v30, 0x0
  
+     .line 1462
      .local v30, index:I
      if-eqz v42, :cond_41
  
      const/16 v30, 0x1
  
+     .line 1463
      :cond_41
      if-eqz v43, :cond_42
  
      add-int/lit8 v30, v30, 0x2
  
+     .line 1464
      :cond_42
      if-eqz v41, :cond_43
  
      add-int/lit8 v30, v30, 0x4
  
+     .line 1465
      :cond_43
      if-eqz v37, :cond_44
  
      add-int/lit8 v30, v30, 0x8
  
+     .line 1466
      :cond_44
      sget-object v62, Landroid/text/format/DateUtils;->sameYearTable:[I
  
      aget v39, v62, v30
  
+     .line 1467
      .local v39, resId:I
      invoke-virtual/range {v38 .. v39}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
  
      move-result-object v29
  
+     .line 1471
      const/16 v62, 0xa
  
      move/from16 v0, v62
***************
*** 1996,2001 ****
--- 2202,2208 ----
  
      goto/16 :goto_16
  
+     .line 1441
      .end local v18           #endMonthDayString:Ljava/lang/String;
      .end local v20           #endMonthString:Ljava/lang/String;
      .end local v26           #endYearString:Ljava/lang/String;
***************
*** 2008,2013 ****
--- 2215,2221 ----
      :cond_45
      if-eqz v4, :cond_46
  
+     .line 1442
      const v62, 0x10400af
  
      move-object/from16 v0, v38
***************
*** 2021,2026 ****
--- 2229,2235 ----
      .restart local v32       #monthFormat:Ljava/lang/String;
      goto/16 :goto_24
  
+     .line 1445
      .end local v32           #monthFormat:Ljava/lang/String;
      :cond_46
      const-string v32, "%B"
***************
*** 2028,2033 ****
--- 2237,2243 ----
      .restart local v32       #monthFormat:Ljava/lang/String;
      goto/16 :goto_24
  
+     .line 1451
      .restart local v48       #startMonthDayString:Ljava/lang/String;
      .restart local v50       #startMonthString:Ljava/lang/String;
      .restart local v56       #startYearString:Ljava/lang/String;
***************
*** 2040,2045 ****
--- 2250,2256 ----
  
      goto :goto_25
  
+     .line 1452
      .restart local v20       #endMonthString:Ljava/lang/String;
      :cond_48
      const-string v62, "%-d"
***************
*** 2052,2057 ****
--- 2263,2269 ----
  
      goto/16 :goto_26
  
+     .line 1453
      .restart local v18       #endMonthDayString:Ljava/lang/String;
      :cond_49
      const-string v62, "%Y"
***************
*** 2064,2069 ****
--- 2276,2282 ----
  
      goto/16 :goto_27
  
+     .line 1478
      .restart local v26       #endYearString:Ljava/lang/String;
      :cond_4a
      move/from16 v0, v46
***************
*** 2072,2109 ****
--- 2285,2330 ----
  
      if-eq v0, v1, :cond_4f
  
+     .line 1480
      const/16 v30, 0x0
  
+     .line 1481
      .restart local v30       #index:I
      if-eqz v42, :cond_4b
  
      const/16 v30, 0x1
  
+     .line 1482
      :cond_4b
      if-eqz v43, :cond_4c
  
      add-int/lit8 v30, v30, 0x2
  
+     .line 1483
      :cond_4c
      if-eqz v41, :cond_4d
  
      add-int/lit8 v30, v30, 0x4
  
+     .line 1484
      :cond_4d
      if-eqz v37, :cond_4e
  
      add-int/lit8 v30, v30, 0x8
  
+     .line 1485
      :cond_4e
      sget-object v62, Landroid/text/format/DateUtils;->sameMonthTable:[I
  
      aget v39, v62, v30
  
+     .line 1486
      .restart local v39       #resId:I
      invoke-virtual/range {v38 .. v39}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
  
      move-result-object v29
  
+     .line 1490
      const/16 v62, 0xa
  
      move/from16 v0, v62
***************
*** 2164,2169 ****
--- 2385,2391 ----
  
      goto/16 :goto_16
  
+     .line 1498
      .end local v30           #index:I
      .end local v39           #resId:I
      :cond_4f
***************
*** 2173,2178 ****
--- 2395,2401 ----
  
      const/16 v40, 0x1
  
+     .line 1501
      .local v40, showDate:Z
      :goto_28
      if-nez v41, :cond_50
***************
*** 2183,2217 ****
--- 2406,2451 ----
  
      const/16 v40, 0x1
  
+     .line 1504
      :cond_50
      const-string v58, ""
  
+     .line 1505
      .local v58, timeString:Ljava/lang/String;
      if-eqz v41, :cond_51
  
+     .line 1508
      if-eqz v31, :cond_54
  
+     .line 1511
      move-object/from16 v58, v53
  
+     .line 1521
      :cond_51
      :goto_29
      const-string v29, ""
  
+     .line 1522
      const-string v11, ""
  
+     .line 1523
      .local v11, dateString:Ljava/lang/String;
      if-eqz v40, :cond_58
  
+     .line 1524
      move-object/from16 v0, v44
  
      invoke-virtual {v0, v13}, Landroid/text/format/Time;->format(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v11
  
+     .line 1525
      if-eqz v42, :cond_56
  
+     .line 1526
      if-eqz v41, :cond_55
  
+     .line 1528
      const v62, 0x104009c
  
      move-object/from16 v0, v38
***************
*** 2222,2227 ****
--- 2456,2462 ----
  
      move-result-object v29
  
+     .line 1556
      :cond_52
      :goto_2a
      const/16 v62, 0x3
***************
*** 2256,2261 ****
--- 2491,2497 ----
  
      goto/16 :goto_16
  
+     .line 1498
      .end local v11           #dateString:Ljava/lang/String;
      .end local v40           #showDate:Z
      .end local v58           #timeString:Ljava/lang/String;
***************
*** 2264,2269 ****
--- 2500,2506 ----
  
      goto :goto_28
  
+     .line 1514
      .restart local v40       #showDate:Z
      .restart local v58       #timeString:Ljava/lang/String;
      :cond_54
***************
*** 2277,2282 ****
--- 2514,2520 ----
  
      move-result-object v57
  
+     .line 1516
      .local v57, timeFormat:Ljava/lang/String;
      const/16 v62, 0x2
  
***************
*** 2304,2309 ****
--- 2542,2548 ----
  
      goto :goto_29
  
+     .line 1531
      .end local v57           #timeFormat:Ljava/lang/String;
      .restart local v11       #dateString:Ljava/lang/String;
      :cond_55
***************
*** 2319,2327 ****
--- 2558,2568 ----
  
      goto :goto_2a
  
+     .line 1534
      :cond_56
      if-eqz v41, :cond_57
  
+     .line 1536
      const v62, 0x1040087
  
      move-object/from16 v0, v38
***************
*** 2334,2339 ****
--- 2575,2581 ----
  
      goto :goto_2a
  
+     .line 1539
      :cond_57
      const-string v62, "%s"
  
***************
*** 2361,2371 ****
--- 2603,2616 ----
  
      goto/16 :goto_16
  
+     .line 1542
      :cond_58
      if-eqz v42, :cond_5a
  
+     .line 1543
      if-eqz v41, :cond_59
  
+     .line 1545
      const v62, 0x104009e
  
      move-object/from16 v0, v38
***************
*** 2378,2383 ****
--- 2623,2629 ----
  
      goto/16 :goto_2a
  
+     .line 1548
      :cond_59
      const-string v62, "%s"
  
***************
*** 2405,2413 ****
--- 2651,2661 ----
  
      goto/16 :goto_16
  
+     .line 1550
      :cond_5a
      if-eqz v41, :cond_52
  
+     .line 1551
      const-string v62, "%s"
  
      const/16 v63, 0x1
***************
*** 2442,2447 ****
--- 2690,2696 ----
      .parameter "flags"
  
      .prologue
+     .line 1602
      move-object v0, p0
  
      move-wide v1, p1
***************
*** 2462,2467 ****
--- 2711,2717 ----
      .parameter "elapsedSeconds"
  
      .prologue
+     .line 678
      const/4 v0, 0x0
  
      invoke-static {v0, p0, p1}, Landroid/text/format/DateUtils;->formatElapsedTime(Ljava/lang/StringBuilder;J)Ljava/lang/String;
***************
*** 2477,2492 ****
--- 2727,2747 ----
      .parameter "elapsedSeconds"
  
      .prologue
+     .line 689
      invoke-static {}, Landroid/text/format/DateUtils;->initFormatStrings()V
  
+     .line 691
      const-wide/16 v2, 0x0
  
+     .line 692
      .local v2, hours:J
      const-wide/16 v4, 0x0
  
+     .line 693
      .local v4, minutes:J
      const-wide/16 v6, 0x0
  
+     .line 695
      .local v6, seconds:J
      const-wide/16 v0, 0xe10
  
***************
*** 2494,2509 ****
--- 2749,2767 ----
  
      if-ltz v0, :cond_0
  
+     .line 696
      const-wide/16 v0, 0xe10
  
      div-long v2, p1, v0
  
+     .line 697
      const-wide/16 v0, 0xe10
  
      mul-long/2addr v0, v2
  
      sub-long/2addr p1, v0
  
+     .line 699
      :cond_0
      const-wide/16 v0, 0x3c
  
***************
*** 2511,2535 ****
--- 2769,2798 ----
  
      if-ltz v0, :cond_1
  
+     .line 700
      const-wide/16 v0, 0x3c
  
      div-long v4, p1, v0
  
+     .line 701
      const-wide/16 v0, 0x3c
  
      mul-long/2addr v0, v4
  
      sub-long/2addr p1, v0
  
+     .line 703
      :cond_1
      move-wide v6, p1
  
+     .line 706
      const-wide/16 v0, 0x0
  
      cmp-long v0, v2, v0
  
      if-lez v0, :cond_2
  
+     .line 707
      sget-object v1, Landroid/text/format/DateUtils;->sElapsedFormatHMMSS:Ljava/lang/String;
  
      move-object v0, p0
***************
*** 2538,2543 ****
--- 2801,2807 ----
  
      move-result-object v0
  
+     .line 709
      :goto_0
      return-object v0
  
***************
*** 2571,2576 ****
--- 2835,2841 ----
  
      const-wide/16 v3, 0xa
  
+     .line 751
      const-string v1, "%1$02d:%2$02d"
  
      invoke-virtual {v1, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
***************
*** 2579,2589 ****
--- 2844,2857 ----
  
      if-eqz v1, :cond_3
  
+     .line 752
      move-object v0, p0
  
+     .line 753
      .local v0, sb:Ljava/lang/StringBuilder;
      if-nez v0, :cond_0
  
+     .line 754
      new-instance v0, Ljava/lang/StringBuilder;
  
      .end local v0           #sb:Ljava/lang/StringBuilder;
***************
*** 2591,2604 ****
--- 2859,2875 ----
  
      invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(I)V
  
+     .line 758
      .restart local v0       #sb:Ljava/lang/StringBuilder;
      :goto_0
      cmp-long v1, p2, v3
  
      if-gez v1, :cond_1
  
+     .line 759
      invoke-virtual {v0, v6}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 763
      :goto_1
      rem-long v1, p2, v3
  
***************
*** 2608,2623 ****
--- 2879,2898 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 764
      const/16 v1, 0x3a
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 765
      cmp-long v1, p4, v3
  
      if-gez v1, :cond_2
  
+     .line 766
      invoke-virtual {v0, v6}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 770
      :goto_2
      rem-long v1, p4, v3
  
***************
*** 2627,2646 ****
--- 2902,2925 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 771
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 773
      .end local v0           #sb:Ljava/lang/StringBuilder;
      :goto_3
      return-object v1
  
+     .line 756
      .restart local v0       #sb:Ljava/lang/StringBuilder;
      :cond_0
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->setLength(I)V
  
      goto :goto_0
  
+     .line 761
      :cond_1
      div-long v1, p2, v3
  
***************
*** 2652,2657 ****
--- 2931,2937 ----
  
      goto :goto_1
  
+     .line 768
      :cond_2
      div-long v1, p4, v3
  
***************
*** 2663,2668 ****
--- 2943,2949 ----
  
      goto :goto_2
  
+     .line 773
      .end local v0           #sb:Ljava/lang/StringBuilder;
      :cond_3
      const/4 v1, 0x2
***************
*** 2707,2712 ****
--- 2988,2994 ----
  
      const-wide/16 v3, 0xa
  
+     .line 718
      const-string v1, "%1$d:%2$02d:%3$02d"
  
      invoke-virtual {v1, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
***************
*** 2715,2725 ****
--- 2997,3010 ----
  
      if-eqz v1, :cond_3
  
+     .line 719
      move-object v0, p0
  
+     .line 720
      .local v0, sb:Ljava/lang/StringBuilder;
      if-nez v0, :cond_0
  
+     .line 721
      new-instance v0, Ljava/lang/StringBuilder;
  
      .end local v0           #sb:Ljava/lang/StringBuilder;
***************
*** 2727,2744 ****
--- 3012,3034 ----
  
      invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(I)V
  
+     .line 725
      .restart local v0       #sb:Ljava/lang/StringBuilder;
      :goto_0
      invoke-virtual {v0, p2, p3}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
  
+     .line 726
      invoke-virtual {v0, v7}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 727
      cmp-long v1, p4, v3
  
      if-gez v1, :cond_1
  
+     .line 728
      invoke-virtual {v0, v6}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 732
      :goto_1
      rem-long v1, p4, v3
  
***************
*** 2748,2761 ****
--- 3038,3055 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 733
      invoke-virtual {v0, v7}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 734
      cmp-long v1, p6, v3
  
      if-gez v1, :cond_2
  
+     .line 735
      invoke-virtual {v0, v6}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 739
      :goto_2
      rem-long v1, p6, v3
  
***************
*** 2765,2784 ****
--- 3059,3082 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 740
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 742
      .end local v0           #sb:Ljava/lang/StringBuilder;
      :goto_3
      return-object v1
  
+     .line 723
      .restart local v0       #sb:Ljava/lang/StringBuilder;
      :cond_0
      invoke-virtual {v0, v5}, Ljava/lang/StringBuilder;->setLength(I)V
  
      goto :goto_0
  
+     .line 730
      :cond_1
      div-long v1, p4, v3
  
***************
*** 2790,2795 ****
--- 3088,3094 ----
  
      goto :goto_1
  
+     .line 737
      :cond_2
      div-long v1, p6, v3
  
***************
*** 2801,2806 ****
--- 3100,3106 ----
  
      goto :goto_2
  
+     .line 742
      .end local v0           #sb:Ljava/lang/StringBuilder;
      :cond_3
      const/4 v1, 0x3
***************
*** 2850,2874 ****
--- 3150,3180 ----
  
      const/4 v5, 0x1
  
+     .line 800
      new-instance v2, Ljava/util/GregorianCalendar;
  
      invoke-direct {v2}, Ljava/util/GregorianCalendar;-><init>()V
  
+     .line 801
      .local v2, thenCal:Ljava/util/Calendar;
      invoke-virtual {v2, p0, p1}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 802
      invoke-virtual {v2}, Ljava/util/Calendar;->getTime()Ljava/util/Date;
  
      move-result-object v3
  
+     .line 803
      .local v3, thenDate:Ljava/util/Date;
      new-instance v1, Ljava/util/GregorianCalendar;
  
      invoke-direct {v1}, Ljava/util/GregorianCalendar;-><init>()V
  
+     .line 804
      .local v1, nowCal:Ljava/util/Calendar;
      invoke-virtual {v1, p2, p3}, Ljava/util/Calendar;->setTimeInMillis(J)V
  
+     .line 808
      invoke-virtual {v2, v5}, Ljava/util/Calendar;->get(I)I
  
      move-result v4
***************
*** 2899,2908 ****
--- 3205,3216 ----
  
      if-ne v4, v5, :cond_0
  
+     .line 811
      invoke-static {p5}, Ljava/text/DateFormat;->getTimeInstance(I)Ljava/text/DateFormat;
  
      move-result-object v0
  
+     .line 815
      .local v0, f:Ljava/text/DateFormat;
      :goto_0
      invoke-virtual {v0, v3}, Ljava/text/DateFormat;->format(Ljava/util/Date;)Ljava/lang/String;
***************
*** 2911,2916 ****
--- 3219,3225 ----
  
      return-object v4
  
+     .line 813
      .end local v0           #f:Ljava/text/DateFormat;
      :cond_0
      invoke-static {p4}, Ljava/text/DateFormat;->getDateInstance(I)Ljava/text/DateFormat;
***************
*** 2926,2935 ****
--- 3235,3246 ----
      .parameter "ampm"
  
      .prologue
+     .line 335
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v0
  
+     .line 336
      .local v0, r:Landroid/content/res/Resources;
      sget-object v1, Landroid/text/format/DateUtils;->sAmPm:[I
  
***************
*** 2950,2965 ****
--- 3261,3280 ----
      .parameter "abbrev"
  
      .prologue
+     .line 315
      sparse-switch p1, :sswitch_data_0
  
+     .line 321
      sget-object v0, Landroid/text/format/DateUtils;->sDaysMedium:[I
  
+     .line 324
      .local v0, list:[I
      :goto_0
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v1
  
+     .line 325
      .local v1, r:Landroid/content/res/Resources;
      add-int/lit8 v2, p0, -0x1
  
***************
*** 2971,2976 ****
--- 3286,3292 ----
  
      return-object v2
  
+     .line 316
      .end local v0           #list:[I
      .end local v1           #r:Landroid/content/res/Resources;
      :sswitch_0
***************
*** 2979,2984 ****
--- 3295,3301 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 317
      .end local v0           #list:[I
      :sswitch_1
      sget-object v0, Landroid/text/format/DateUtils;->sDaysMedium:[I
***************
*** 2986,2991 ****
--- 3303,3309 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 318
      .end local v0           #list:[I
      :sswitch_2
      sget-object v0, Landroid/text/format/DateUtils;->sDaysShort:[I
***************
*** 2993,2998 ****
--- 3311,3317 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 319
      .end local v0           #list:[I
      :sswitch_3
      sget-object v0, Landroid/text/format/DateUtils;->sDaysShort:[I
***************
*** 3000,3005 ****
--- 3319,3325 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 320
      .end local v0           #list:[I
      :sswitch_4
      sget-object v0, Landroid/text/format/DateUtils;->sDaysShortest:[I
***************
*** 3007,3012 ****
--- 3327,3333 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 315
      nop
  
      :sswitch_data_0
***************
*** 3025,3040 ****
--- 3346,3365 ----
      .parameter "abbrev"
  
      .prologue
+     .line 355
      sparse-switch p1, :sswitch_data_0
  
+     .line 361
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsMedium:[I
  
+     .line 364
      .local v0, list:[I
      :goto_0
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v1
  
+     .line 365
      .local v1, r:Landroid/content/res/Resources;
      add-int/lit8 v2, p0, 0x0
  
***************
*** 3046,3051 ****
--- 3371,3377 ----
  
      return-object v2
  
+     .line 356
      .end local v0           #list:[I
      .end local v1           #r:Landroid/content/res/Resources;
      :sswitch_0
***************
*** 3054,3059 ****
--- 3380,3386 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 357
      .end local v0           #list:[I
      :sswitch_1
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsMedium:[I
***************
*** 3061,3066 ****
--- 3388,3394 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 358
      .end local v0           #list:[I
      :sswitch_2
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsMedium:[I
***************
*** 3068,3073 ****
--- 3396,3402 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 359
      .end local v0           #list:[I
      :sswitch_3
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsMedium:[I
***************
*** 3075,3080 ****
--- 3404,3410 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 360
      .end local v0           #list:[I
      :sswitch_4
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsShortest:[I
***************
*** 3082,3087 ****
--- 3412,3418 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 355
      nop
  
      :sswitch_data_0
***************
*** 3100,3105 ****
--- 3431,3437 ----
      .parameter "date2"
  
      .prologue
+     .line 537
      const-class v3, Landroid/text/format/DateUtils;
  
      monitor-enter v3
***************
*** 3109,3125 ****
--- 3441,3460 ----
  
      if-nez v2, :cond_0
  
+     .line 538
      new-instance v2, Landroid/text/format/Time;
  
      invoke-direct {v2}, Landroid/text/format/Time;-><init>()V
  
      sput-object v2, Landroid/text/format/DateUtils;->sThenTime:Landroid/text/format/Time;
  
+     .line 540
      :cond_0
      sget-object v2, Landroid/text/format/DateUtils;->sThenTime:Landroid/text/format/Time;
  
      invoke-virtual {v2, p0, p1}, Landroid/text/format/Time;->set(J)V
  
+     .line 541
      sget-object v2, Landroid/text/format/DateUtils;->sThenTime:Landroid/text/format/Time;
  
      iget-wide v4, v2, Landroid/text/format/Time;->gmtoff:J
***************
*** 3128,3138 ****
--- 3463,3475 ----
  
      move-result v0
  
+     .line 542
      .local v0, day1:I
      sget-object v2, Landroid/text/format/DateUtils;->sThenTime:Landroid/text/format/Time;
  
      invoke-virtual {v2, p2, p3}, Landroid/text/format/Time;->set(J)V
  
+     .line 543
      sget-object v2, Landroid/text/format/DateUtils;->sThenTime:Landroid/text/format/Time;
  
      iget-wide v4, v2, Landroid/text/format/Time;->gmtoff:J
***************
*** 3141,3146 ****
--- 3478,3484 ----
  
      move-result v1
  
+     .line 544
      .local v1, day2:I
      sub-int v2, v1, v0
  
***************
*** 3156,3161 ****
--- 3494,3500 ----
  
      return-wide v4
  
+     .line 537
      .end local v0           #day1:I
      .end local v1           #day2:I
      :catchall_0
***************
*** 3175,3189 ****
--- 3514,3531 ----
      .parameter "flags"
  
      .prologue
+     .line 572
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v15
  
+     .line 574
      .local v15, r:Landroid/content/res/Resources;
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v13
  
+     .line 575
      .local v13, now:J
      sub-long v3, v13, p1
  
***************
*** 3191,3196 ****
--- 3533,3539 ----
  
      move-result-wide v11
  
+     .line 580
      .local v11, duration:J
      const-wide/32 v3, 0x240c8400
  
***************
*** 3198,3205 ****
--- 3541,3550 ----
  
      if-lez v3, :cond_1
  
+     .line 581
      const-wide/32 p5, 0x240c8400
  
+     .line 586
      :cond_0
      :goto_0
      const/4 v8, 0x1
***************
*** 3214,3219 ****
--- 3559,3565 ----
  
      move-result-object v18
  
+     .line 589
      .local v18, timeClause:Ljava/lang/CharSequence;
      cmp-long v3, v11, p5
  
***************
*** 3227,3236 ****
--- 3573,3584 ----
  
      move/from16 v9, p7
  
+     .line 590
      invoke-static/range {v3 .. v9}, Landroid/text/format/DateUtils;->getRelativeTimeSpanString(JJJI)Ljava/lang/CharSequence;
  
      move-result-object v16
  
+     .line 591
      .local v16, relativeClause:Ljava/lang/CharSequence;
      const v3, 0x104038d
  
***************
*** 3250,3260 ****
--- 3598,3610 ----
  
      move-result-object v17
  
+     .line 597
      .end local v16           #relativeClause:Ljava/lang/CharSequence;
      .local v17, result:Ljava/lang/String;
      :goto_1
      return-object v17
  
+     .line 582
      .end local v17           #result:Ljava/lang/String;
      .end local v18           #timeClause:Ljava/lang/CharSequence;
      :cond_1
***************
*** 3264,3273 ****
--- 3614,3625 ----
  
      if-gez v3, :cond_0
  
+     .line 583
      const-wide/32 p5, 0x5265c00
  
      goto :goto_0
  
+     .line 593
      .restart local v18       #timeClause:Ljava/lang/CharSequence;
      :cond_2
      const/4 v3, 0x0
***************
*** 3280,3285 ****
--- 3632,3638 ----
  
      move-result-object v10
  
+     .line 594
      .local v10, dateClause:Ljava/lang/CharSequence;
      const v3, 0x1040086
  
***************
*** 3310,3335 ****
--- 3663,3694 ----
      .parameter "today"
  
      .prologue
+     .line 611
      new-instance v7, Landroid/text/format/Time;
  
      invoke-direct {v7}, Landroid/text/format/Time;-><init>()V
  
+     .line 612
      .local v7, startTime:Landroid/text/format/Time;
      invoke-virtual {v7, p1, p2}, Landroid/text/format/Time;->set(J)V
  
+     .line 613
      new-instance v1, Landroid/text/format/Time;
  
      invoke-direct {v1}, Landroid/text/format/Time;-><init>()V
  
+     .line 614
      .local v1, currentTime:Landroid/text/format/Time;
      invoke-virtual {v1, p3, p4}, Landroid/text/format/Time;->set(J)V
  
+     .line 616
      iget-wide v8, v7, Landroid/text/format/Time;->gmtoff:J
  
      invoke-static {p1, p2, v8, v9}, Landroid/text/format/Time;->getJulianDay(JJ)I
  
      move-result v6
  
+     .line 617
      .local v6, startDay:I
      iget-wide v8, v1, Landroid/text/format/Time;->gmtoff:J
  
***************
*** 3337,3342 ****
--- 3696,3702 ----
  
      move-result v0
  
+     .line 619
      .local v0, currentDay:I
      sub-int v8, v0, v6
  
***************
*** 3344,3349 ****
--- 3704,3710 ----
  
      move-result v2
  
+     .line 620
      .local v2, days:I
      cmp-long v8, p3, p1
  
***************
*** 3351,3379 ****
--- 3712,3746 ----
  
      const/4 v4, 0x1
  
+     .line 622
      .local v4, past:Z
      :goto_0
      const/4 v8, 0x1
  
      if-ne v2, v8, :cond_2
  
+     .line 623
      if-eqz v4, :cond_1
  
+     .line 624
      const v8, 0x1040078
  
      invoke-virtual {p0, v8}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
  
      move-result-object v8
  
+     .line 640
      :goto_1
      return-object v8
  
+     .line 620
      .end local v4           #past:Z
      :cond_0
      const/4 v4, 0x0
  
      goto :goto_0
  
+     .line 626
      .restart local v4       #past:Z
      :cond_1
      const v8, 0x104007a
***************
*** 3384,3392 ****
--- 3751,3761 ----
  
      goto :goto_1
  
+     .line 628
      :cond_2
      if-nez v2, :cond_3
  
+     .line 629
      const v8, 0x1040079
  
      invoke-virtual {p0, v8}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
***************
*** 3395,3411 ****
--- 3764,3784 ----
  
      goto :goto_1
  
+     .line 633
      :cond_3
      if-eqz v4, :cond_4
  
+     .line 634
      const v5, 0x1130004
  
+     .line 639
      .local v5, resId:I
      :goto_2
      invoke-virtual {p0, v5, v2}, Landroid/content/res/Resources;->getQuantityString(II)Ljava/lang/String;
  
      move-result-object v3
  
+     .line 640
      .local v3, format:Ljava/lang/String;
      const/4 v8, 0x1
  
***************
*** 3425,3430 ****
--- 3798,3804 ----
  
      goto :goto_1
  
+     .line 636
      .end local v3           #format:Ljava/lang/String;
      .end local v5           #resId:I
      :cond_4
***************
*** 3439,3444 ****
--- 3813,3819 ----
      .parameter "startTime"
  
      .prologue
+     .line 409
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v2
***************
*** 3461,3466 ****
--- 3836,3842 ----
      .parameter "minResolution"
  
      .prologue
+     .line 425
      const v6, 0x10014
  
      .local v6, flags:I
***************
*** 3470,3475 ****
--- 3846,3852 ----
  
      move-wide v4, p4
  
+     .line 426
      invoke-static/range {v0 .. v6}, Landroid/text/format/DateUtils;->getRelativeTimeSpanString(JJJI)Ljava/lang/CharSequence;
  
      move-result-object v0
***************
*** 3485,3494 ****
--- 3862,3873 ----
      .parameter "flags"
  
      .prologue
+     .line 451
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v13
  
+     .line 452
      .local v13, r:Landroid/content/res/Resources;
      const/high16 v0, 0xc
  
***************
*** 3498,3503 ****
--- 3877,3883 ----
  
      const/4 v6, 0x1
  
+     .line 454
      .local v6, abbrevRelative:Z
      :goto_0
      cmp-long v0, p2, p0
***************
*** 3506,3511 ****
--- 3886,3892 ----
  
      const/4 v12, 0x1
  
+     .line 455
      .local v12, past:Z
      :goto_1
      sub-long v0, p2, p0
***************
*** 3514,3519 ****
--- 3895,3901 ----
  
      move-result-wide v9
  
+     .line 459
      .local v9, duration:J
      const-wide/32 v0, 0xea60
  
***************
*** 3527,3543 ****
--- 3909,3930 ----
  
      if-gez v0, :cond_5
  
+     .line 460
      const-wide/16 v0, 0x3e8
  
      div-long v7, v9, v0
  
+     .line 461
      .local v7, count:J
      if-eqz v12, :cond_3
  
+     .line 462
      if-eqz v6, :cond_2
  
+     .line 463
      const v14, 0x1130009
  
+     .line 525
      .local v14, resId:I
      :goto_2
      long-to-int v0, v7
***************
*** 3546,3551 ****
--- 3933,3939 ----
  
      move-result-object v11
  
+     .line 526
      .local v11, format:Ljava/lang/String;
      const/4 v0, 0x1
  
***************
*** 3569,3574 ****
--- 3957,3963 ----
      :goto_3
      return-object v0
  
+     .line 452
      .end local v6           #abbrevRelative:Z
      .end local v9           #duration:J
      .end local v12           #past:Z
***************
*** 3577,3588 ****
--- 3966,3979 ----
  
      goto :goto_0
  
+     .line 454
      .restart local v6       #abbrevRelative:Z
      :cond_1
      const/4 v12, 0x0
  
      goto :goto_1
  
+     .line 465
      .restart local v7       #count:J
      .restart local v9       #duration:J
      .restart local v12       #past:Z
***************
*** 3592,3606 ****
--- 3983,4000 ----
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 468
      .end local v14           #resId:I
      :cond_3
      if-eqz v6, :cond_4
  
+     .line 469
      const v14, 0x113000d
  
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 471
      .end local v14           #resId:I
      :cond_4
      const v14, 0x1130005
***************
*** 3608,3613 ****
--- 4002,4008 ----
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 474
      .end local v7           #count:J
      .end local v14           #resId:I
      :cond_5
***************
*** 3623,3642 ****
--- 4018,4042 ----
  
      if-gez v0, :cond_9
  
+     .line 475
      const-wide/32 v0, 0xea60
  
      div-long v7, v9, v0
  
+     .line 476
      .restart local v7       #count:J
      if-eqz v12, :cond_7
  
+     .line 477
      if-eqz v6, :cond_6
  
+     .line 478
      const v14, 0x113000a
  
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 480
      .end local v14           #resId:I
      :cond_6
      const v14, 0x1130001
***************
*** 3644,3658 ****
--- 4044,4061 ----
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 483
      .end local v14           #resId:I
      :cond_7
      if-eqz v6, :cond_8
  
+     .line 484
      const v14, 0x113000e
  
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 486
      .end local v14           #resId:I
      :cond_8
      const v14, 0x1130006
***************
*** 3660,3665 ****
--- 4063,4069 ----
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 489
      .end local v7           #count:J
      .end local v14           #resId:I
      :cond_9
***************
*** 3675,3694 ****
--- 4079,4103 ----
  
      if-gez v0, :cond_d
  
+     .line 490
      const-wide/32 v0, 0x36ee80
  
      div-long v7, v9, v0
  
+     .line 491
      .restart local v7       #count:J
      if-eqz v12, :cond_b
  
+     .line 492
      if-eqz v6, :cond_a
  
+     .line 493
      const v14, 0x113000b
  
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 495
      .end local v14           #resId:I
      :cond_a
      const v14, 0x1130002
***************
*** 3696,3710 ****
--- 4105,4122 ----
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 498
      .end local v14           #resId:I
      :cond_b
      if-eqz v6, :cond_c
  
+     .line 499
      const v14, 0x113000f
  
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 501
      .end local v14           #resId:I
      :cond_c
      const v14, 0x1130007
***************
*** 3712,3717 ****
--- 4124,4130 ----
      .restart local v14       #resId:I
      goto :goto_2
  
+     .line 504
      .end local v7           #count:J
      .end local v14           #resId:I
      :cond_d
***************
*** 3727,3746 ****
--- 4140,4164 ----
  
      if-gez v0, :cond_11
  
+     .line 505
      invoke-static/range {p0 .. p3}, Landroid/text/format/DateUtils;->getNumberOfDaysPassed(JJ)J
  
      move-result-wide v7
  
+     .line 506
      .restart local v7       #count:J
      if-eqz v12, :cond_f
  
+     .line 507
      if-eqz v6, :cond_e
  
+     .line 508
      const v14, 0x113000c
  
      .restart local v14       #resId:I
      goto/16 :goto_2
  
+     .line 510
      .end local v14           #resId:I
      :cond_e
      const v14, 0x1130004
***************
*** 3748,3762 ****
--- 4166,4183 ----
      .restart local v14       #resId:I
      goto/16 :goto_2
  
+     .line 513
      .end local v14           #resId:I
      :cond_f
      if-eqz v6, :cond_10
  
+     .line 514
      const v14, 0x1130010
  
      .restart local v14       #resId:I
      goto/16 :goto_2
  
+     .line 516
      .end local v14           #resId:I
      :cond_10
      const v14, 0x1130008
***************
*** 3764,3769 ****
--- 4185,4191 ----
      .restart local v14       #resId:I
      goto/16 :goto_2
  
+     .line 522
      .end local v7           #count:J
      .end local v14           #resId:I
      :cond_11
***************
*** 3788,3793 ****
--- 4210,4216 ----
      .parameter "millis"
  
      .prologue
+     .line 1673
      const/4 v0, 0x0
  
      invoke-static {p0, p1, p2, v0}, Landroid/text/format/DateUtils;->getRelativeTimeSpanString(Landroid/content/Context;JZ)Ljava/lang/CharSequence;
***************
*** 3804,3854 ****
--- 4227,4287 ----
      .parameter "withPreposition"
  
      .prologue
+     .line 1623
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v8
  
+     .line 1624
      .local v8, now:J
      sub-long v13, v8, p1
  
+     .line 1626
      .local v13, span:J
      const-class v15, Landroid/text/format/DateUtils;
  
      monitor-enter v15
  
+     .line 1627
      :try_start_0
      sget-object v2, Landroid/text/format/DateUtils;->sNowTime:Landroid/text/format/Time;
  
      if-nez v2, :cond_0
  
+     .line 1628
      new-instance v2, Landroid/text/format/Time;
  
      invoke-direct {v2}, Landroid/text/format/Time;-><init>()V
  
      sput-object v2, Landroid/text/format/DateUtils;->sNowTime:Landroid/text/format/Time;
  
+     .line 1631
      :cond_0
      sget-object v2, Landroid/text/format/DateUtils;->sThenTime:Landroid/text/format/Time;
  
      if-nez v2, :cond_1
  
+     .line 1632
      new-instance v2, Landroid/text/format/Time;
  
      invoke-direct {v2}, Landroid/text/format/Time;-><init>()V
  
      sput-object v2, Landroid/text/format/DateUtils;->sThenTime:Landroid/text/format/Time;
  
+     .line 1635
      :cond_1
      sget-object v2, Landroid/text/format/DateUtils;->sNowTime:Landroid/text/format/Time;
  
      invoke-virtual {v2, v8, v9}, Landroid/text/format/Time;->set(J)V
  
+     .line 1636
      sget-object v2, Landroid/text/format/DateUtils;->sThenTime:Landroid/text/format/Time;
  
      move-wide/from16 v0, p1
  
      invoke-virtual {v2, v0, v1}, Landroid/text/format/Time;->set(J)V
  
+     .line 1639
      const-wide/32 v2, 0x5265c00
  
      cmp-long v2, v13, v2
***************
*** 3865,3870 ****
--- 4298,4304 ----
  
      if-ne v2, v3, :cond_3
  
+     .line 1641
      const/4 v7, 0x1
  
      .local v7, flags:I
***************
*** 3874,3894 ****
--- 4308,4333 ----
  
      move-wide/from16 v5, p1
  
+     .line 1642
      invoke-static/range {v2 .. v7}, Landroid/text/format/DateUtils;->formatDateRange(Landroid/content/Context;JJI)Ljava/lang/String;
  
      move-result-object v12
  
+     .line 1643
      .local v12, result:Ljava/lang/String;
      const v10, 0x104037f
  
+     .line 1657
      .local v10, prepositionId:I
      :goto_0
      if-eqz p3, :cond_2
  
+     .line 1658
      invoke-virtual/range {p0 .. p0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v11
  
+     .line 1659
      .local v11, res:Landroid/content/res/Resources;
      const/4 v2, 0x1
  
***************
*** 3902,3913 ****
--- 4341,4355 ----
  
      move-result-object v12
  
+     .line 1661
      .end local v11           #res:Landroid/content/res/Resources;
      :cond_2
      monitor-exit v15
  
+     .line 1662
      return-object v12
  
+     .line 1644
      .end local v7           #flags:I
      .end local v10           #prepositionId:I
      .end local v12           #result:Ljava/lang/String;
***************
*** 3922,3927 ****
--- 4364,4370 ----
  
      if-eq v2, v3, :cond_4
  
+     .line 1646
      const v7, 0x20014
  
      .restart local v7       #flags:I
***************
*** 3931,3946 ****
--- 4374,4393 ----
  
      move-wide/from16 v5, p1
  
+     .line 1647
      invoke-static/range {v2 .. v7}, Landroid/text/format/DateUtils;->formatDateRange(Landroid/content/Context;JJI)Ljava/lang/String;
  
      move-result-object v12
  
+     .line 1650
      .restart local v12       #result:Ljava/lang/String;
      const v10, 0x104037e
  
+     .line 1651
      .restart local v10       #prepositionId:I
      goto :goto_0
  
+     .line 1653
      .end local v7           #flags:I
      .end local v10           #prepositionId:I
      .end local v12           #result:Ljava/lang/String;
***************
*** 3954,3969 ****
--- 4401,4419 ----
  
      move-wide/from16 v5, p1
  
+     .line 1654
      invoke-static/range {v2 .. v7}, Landroid/text/format/DateUtils;->formatDateRange(Landroid/content/Context;JJI)Ljava/lang/String;
  
      move-result-object v12
  
+     .line 1655
      .restart local v12       #result:Ljava/lang/String;
      const v10, 0x104037e
  
      .restart local v10       #prepositionId:I
      goto :goto_0
  
+     .line 1661
      .end local v7           #flags:I
      .end local v10           #prepositionId:I
      .end local v12           #result:Ljava/lang/String;
***************
*** 3983,3998 ****
--- 4433,4452 ----
      .parameter "abbrev"
  
      .prologue
+     .line 388
      sparse-switch p1, :sswitch_data_0
  
+     .line 395
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsMedium:[I
  
+     .line 398
      .local v0, list:[I
      :goto_0
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v1
  
+     .line 399
      .local v1, r:Landroid/content/res/Resources;
      add-int/lit8 v2, p0, 0x0
  
***************
*** 4004,4017 ****
--- 4458,4474 ----
  
      return-object v2
  
+     .line 389
      .end local v0           #list:[I
      .end local v1           #r:Landroid/content/res/Resources;
      :sswitch_0
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsStandaloneLong:[I
  
+     .line 390
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 391
      .end local v0           #list:[I
      :sswitch_1
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsMedium:[I
***************
*** 4019,4024 ****
--- 4476,4482 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 392
      .end local v0           #list:[I
      :sswitch_2
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsMedium:[I
***************
*** 4026,4031 ****
--- 4484,4490 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 393
      .end local v0           #list:[I
      :sswitch_3
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsMedium:[I
***************
*** 4033,4038 ****
--- 4492,4498 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 394
      .end local v0           #list:[I
      :sswitch_4
      sget-object v0, Landroid/text/format/DateUtils;->sMonthsShortest:[I
***************
*** 4040,4045 ****
--- 4500,4506 ----
      .restart local v0       #list:[I
      goto :goto_0
  
+     .line 388
      nop
  
      :sswitch_data_0
***************
*** 4056,4072 ****
--- 4517,4538 ----
      .locals 2
  
      .prologue
+     .line 644
      sget-object v1, Landroid/text/format/DateUtils;->sLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 645
      :try_start_0
      invoke-static {}, Landroid/text/format/DateUtils;->initFormatStringsLocked()V
  
+     .line 646
      monitor-exit v1
  
+     .line 647
      return-void
  
+     .line 646
      :catchall_0
      move-exception v0
  
***************
*** 4081,4095 ****
--- 4547,4564 ----
      .locals 3
  
      .prologue
+     .line 650
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v1
  
+     .line 651
      .local v1, r:Landroid/content/res/Resources;
      invoke-virtual {v1}, Landroid/content/res/Resources;->getConfiguration()Landroid/content/res/Configuration;
  
      move-result-object v0
  
+     .line 652
      .local v0, cfg:Landroid/content/res/Configuration;
      sget-object v2, Landroid/text/format/DateUtils;->sLastConfig:Landroid/content/res/Configuration;
  
***************
*** 4103,4111 ****
--- 4572,4582 ----
  
      if-nez v2, :cond_1
  
+     .line 653
      :cond_0
      sput-object v0, Landroid/text/format/DateUtils;->sLastConfig:Landroid/content/res/Configuration;
  
+     .line 654
      const/4 v2, 0x3
  
      invoke-static {v2}, Ljava/text/DateFormat;->getTimeInstance(I)Ljava/text/DateFormat;
***************
*** 4114,4119 ****
--- 4585,4591 ----
  
      sput-object v2, Landroid/text/format/DateUtils;->sStatusTimeFormat:Ljava/text/DateFormat;
  
+     .line 655
      const v2, 0x1040392
  
      invoke-virtual {v1, v2}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
***************
*** 4122,4127 ****
--- 4594,4600 ----
  
      sput-object v2, Landroid/text/format/DateUtils;->sElapsedFormatMMSS:Ljava/lang/String;
  
+     .line 656
      const v2, 0x1040393
  
      invoke-virtual {v1, v2}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
***************
*** 4130,4135 ****
--- 4603,4609 ----
  
      sput-object v2, Landroid/text/format/DateUtils;->sElapsedFormatHMMSS:Ljava/lang/String;
  
+     .line 658
      :cond_1
      return-void
  .end method
***************
*** 4139,4159 ****
--- 4613,4639 ----
      .parameter "when"
  
      .prologue
+     .line 834
      new-instance v3, Landroid/text/format/Time;
  
      invoke-direct {v3}, Landroid/text/format/Time;-><init>()V
  
+     .line 835
      .local v3, time:Landroid/text/format/Time;
      invoke-virtual {v3, p0, p1}, Landroid/text/format/Time;->set(J)V
  
+     .line 837
      iget v2, v3, Landroid/text/format/Time;->year:I
  
+     .line 838
      .local v2, thenYear:I
      iget v0, v3, Landroid/text/format/Time;->month:I
  
+     .line 839
      .local v0, thenMonth:I
      iget v1, v3, Landroid/text/format/Time;->monthDay:I
  
+     .line 841
      .local v1, thenMonthDay:I
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
***************
*** 4161,4166 ****
--- 4641,4647 ----
  
      invoke-virtual {v3, v4, v5}, Landroid/text/format/Time;->set(J)V
  
+     .line 842
      iget v4, v3, Landroid/text/format/Time;->year:I
  
      if-ne v2, v4, :cond_0
***************
*** 4193,4198 ****
--- 4674,4680 ----
  
      const/4 v0, 0x1
  
+     .line 854
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
      move-result v1
***************
*** 4209,4218 ****
--- 4691,4702 ----
  
      if-ne v1, v3, :cond_1
  
+     .line 861
      :cond_0
      :goto_0
      return v0
  
+     .line 857
      :cond_1
      invoke-virtual {p0}, Ljava/lang/String;->length()I
  
***************
*** 4230,4235 ****
--- 4714,4720 ----
  
      if-eq v1, v3, :cond_0
  
+     .line 861
      :cond_2
      const/4 v0, 0x0
  
***************
*** 4241,4248 ****
--- 4726,4735 ----
      .parameter "zulu"
  
      .prologue
+     .line 824
      if-eqz p0, :cond_0
  
+     .line 825
      const-string v0, "GMT"
  
      invoke-static {v0}, Ljava/util/TimeZone;->getTimeZone(Ljava/lang/String;)Ljava/util/TimeZone;
***************
*** 4253,4258 ****
--- 4740,4746 ----
  
      move-result-object v0
  
+     .line 827
      :goto_0
      return-object v0
  
***************
*** 4269,4281 ****
--- 4757,4772 ----
      .parameter "millis"
  
      .prologue
+     .line 666
      sget-object v1, Landroid/text/format/DateUtils;->sLock:Ljava/lang/Object;
  
      monitor-enter v1
  
+     .line 667
      :try_start_0
      invoke-static {}, Landroid/text/format/DateUtils;->initFormatStringsLocked()V
  
+     .line 668
      sget-object v0, Landroid/text/format/DateUtils;->sStatusTimeFormat:Ljava/text/DateFormat;
  
      invoke-static {p0, p1}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
***************
*** 4290,4295 ****
--- 4781,4787 ----
  
      return-object v0
  
+     .line 669
      :catchall_0
      move-exception v0
  
***************
*** 4305,4310 ****
--- 4797,4803 ----
      .parameter "digit"
  
      .prologue
+     .line 778
      const-wide/16 v0, 0x30
  
      add-long/2addr v0, p0
***************
*** 4321,4337 ****
--- 4814,4833 ----
      .parameter "cal"
  
      .prologue
+     .line 875
      const-string v2, "GMT"
  
      invoke-static {v2}, Ljava/util/TimeZone;->getTimeZone(Ljava/lang/String;)Ljava/util/TimeZone;
  
      move-result-object v1
  
+     .line 876
      .local v1, tz:Ljava/util/TimeZone;
      new-instance v0, Ljava/util/GregorianCalendar;
  
      invoke-direct {v0, v1}, Ljava/util/GregorianCalendar;-><init>(Ljava/util/TimeZone;)V
  
+     .line 877
      .local v0, c:Ljava/util/GregorianCalendar;
      invoke-virtual {p0}, Ljava/util/Calendar;->getTimeInMillis()J
  
***************
*** 4339,4344 ****
--- 4835,4841 ----
  
      invoke-virtual {v0, v2, v3}, Ljava/util/GregorianCalendar;->setTimeInMillis(J)V
  
+     .line 878
      const/4 v2, 0x1
  
      invoke-static {v0, v2}, Landroid/text/format/DateUtils;->writeDateTime(Ljava/util/Calendar;Z)Ljava/lang/String;
***************
*** 4364,4373 ****
--- 4861,4872 ----
  
      const/4 v3, 0x1
  
+     .line 918
      invoke-virtual {p0, v3}, Ljava/util/Calendar;->get(I)I
  
      move-result v0
  
+     .line 919
      .local v0, n:I
      const/4 v1, 0x3
  
***************
*** 4379,4386 ****
--- 4878,4887 ----
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 920
      div-int/lit8 v0, v0, 0xa
  
+     .line 921
      rem-int/lit8 v1, v0, 0xa
  
      add-int/lit8 v1, v1, 0x30
***************
*** 4389,4396 ****
--- 4890,4899 ----
  
      invoke-virtual {p1, v4, v1}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 922
      div-int/lit8 v0, v0, 0xa
  
+     .line 923
      rem-int/lit8 v1, v0, 0xa
  
      add-int/lit8 v1, v1, 0x30
***************
*** 4399,4406 ****
--- 4902,4911 ----
  
      invoke-virtual {p1, v3, v1}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 924
      div-int/lit8 v0, v0, 0xa
  
+     .line 925
      const/4 v1, 0x0
  
      rem-int/lit8 v2, v0, 0xa
***************
*** 4411,4422 ****
--- 4916,4929 ----
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 927
      invoke-virtual {p0, v4}, Ljava/util/Calendar;->get(I)I
  
      move-result v1
  
      add-int/lit8 v0, v1, 0x1
  
+     .line 928
      rem-int/lit8 v1, v0, 0xa
  
      add-int/lit8 v1, v1, 0x30
***************
*** 4425,4432 ****
--- 4932,4941 ----
  
      invoke-virtual {p1, v5, v1}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 929
      div-int/lit8 v0, v0, 0xa
  
+     .line 930
      const/4 v1, 0x4
  
      rem-int/lit8 v2, v0, 0xa
***************
*** 4437,4446 ****
--- 4946,4957 ----
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 932
      invoke-virtual {p0, v5}, Ljava/util/Calendar;->get(I)I
  
      move-result v0
  
+     .line 933
      const/4 v1, 0x7
  
      rem-int/lit8 v2, v0, 0xa
***************
*** 4451,4458 ****
--- 4962,4971 ----
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 934
      div-int/lit8 v0, v0, 0xa
  
+     .line 935
      const/4 v1, 0x6
  
      rem-int/lit8 v2, v0, 0xa
***************
*** 4463,4478 ****
--- 4976,4994 ----
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 937
      const/16 v1, 0x8
  
      const/16 v2, 0x54
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 939
      invoke-virtual {p0, v6}, Ljava/util/Calendar;->get(I)I
  
      move-result v0
  
+     .line 940
      const/16 v1, 0xa
  
      rem-int/lit8 v2, v0, 0xa
***************
*** 4483,4490 ****
--- 4999,5008 ----
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 941
      div-int/lit8 v0, v0, 0xa
  
+     .line 942
      const/16 v1, 0x9
  
      rem-int/lit8 v2, v0, 0xa
***************
*** 4495,4504 ****
--- 5013,5024 ----
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 944
      invoke-virtual {p0, v7}, Ljava/util/Calendar;->get(I)I
  
      move-result v0
  
+     .line 945
      rem-int/lit8 v1, v0, 0xa
  
      add-int/lit8 v1, v1, 0x30
***************
*** 4507,4514 ****
--- 5027,5036 ----
  
      invoke-virtual {p1, v7, v1}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 946
      div-int/lit8 v0, v0, 0xa
  
+     .line 947
      rem-int/lit8 v1, v0, 0xa
  
      add-int/lit8 v1, v1, 0x30
***************
*** 4517,4528 ****
--- 5039,5052 ----
  
      invoke-virtual {p1, v6, v1}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 949
      const/16 v1, 0xd
  
      invoke-virtual {p0, v1}, Ljava/util/Calendar;->get(I)I
  
      move-result v0
  
+     .line 950
      const/16 v1, 0xe
  
      rem-int/lit8 v2, v0, 0xa
***************
*** 4533,4540 ****
--- 5057,5066 ----
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 951
      div-int/lit8 v0, v0, 0xa
  
+     .line 952
      const/16 v1, 0xd
  
      rem-int/lit8 v2, v0, 0xa
***************
*** 4545,4550 ****
--- 5071,5077 ----
  
      invoke-virtual {p1, v1, v2}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 954
      invoke-virtual {p1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
***************
*** 4562,4582 ****
--- 5089,5115 ----
  
      const/16 v2, 0xf
  
+     .line 893
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 894
      .local v0, sb:Ljava/lang/StringBuilder;
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->ensureCapacity(I)V
  
+     .line 895
      if-eqz p1, :cond_0
  
+     .line 896
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->setLength(I)V
  
+     .line 897
      const/16 v1, 0x5a
  
      invoke-virtual {v0, v2, v1}, Ljava/lang/StringBuilder;->setCharAt(IC)V
  
+     .line 901
      :goto_0
      invoke-static {p0, v0}, Landroid/text/format/DateUtils;->writeDateTime(Ljava/util/Calendar;Ljava/lang/StringBuilder;)Ljava/lang/String;
  
***************
*** 4584,4589 ****
--- 5117,5123 ----
  
      return-object v1
  
+     .line 899
      :cond_0
      invoke-virtual {v0, v2}, Ljava/lang/StringBuilder;->setLength(I)V
  
