*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Activity$NonConfigurationInstances.smali	2012-03-23 19:36:37.059356761 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Activity$NonConfigurationInstances.smali.original	2012-03-23 19:34:45.255318017 -0500
***************
*** 57,62 ****
--- 57,63 ----
      .locals 0
  
      .prologue
+     .line 682
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
