*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityThread$ProviderClientRecord.smali	2012-03-23 19:36:41.543438528 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityThread$ProviderClientRecord.smali.original	2012-03-23 19:34:45.304318911 -0500
***************
*** 36,51 ****
--- 36,56 ----
      .parameter "localProvider"
  
      .prologue
+     .line 282
      iput-object p1, p0, Landroid/app/ActivityThread$ProviderClientRecord;->this$0:Landroid/app/ActivityThread;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 283
      iput-object p2, p0, Landroid/app/ActivityThread$ProviderClientRecord;->mName:Ljava/lang/String;
  
+     .line 284
      iput-object p3, p0, Landroid/app/ActivityThread$ProviderClientRecord;->mProvider:Landroid/content/IContentProvider;
  
+     .line 285
      iput-object p4, p0, Landroid/app/ActivityThread$ProviderClientRecord;->mLocalProvider:Landroid/content/ContentProvider;
  
+     .line 286
      return-void
  .end method
  
***************
*** 55,60 ****
--- 60,66 ----
      .locals 3
  
      .prologue
+     .line 289
      iget-object v0, p0, Landroid/app/ActivityThread$ProviderClientRecord;->this$0:Landroid/app/ActivityThread;
  
      iget-object v1, p0, Landroid/app/ActivityThread$ProviderClientRecord;->mName:Ljava/lang/String;
***************
*** 63,67 ****
--- 69,74 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/app/ActivityThread;->removeDeadProvider(Ljava/lang/String;Landroid/content/IContentProvider;)V
  
+     .line 290
      return-void
  .end method
