*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityGroup.smali	2012-03-23 19:36:37.037356359 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityGroup.smali.original	2012-03-23 19:34:45.255318017 -0500
***************
*** 23,32 ****
--- 23,34 ----
      .locals 1
  
      .prologue
+     .line 43
      const/4 v0, 0x1
  
      invoke-direct {p0, v0}, Landroid/app/ActivityGroup;-><init>(Z)V
  
+     .line 44
      return-void
  .end method
  
***************
*** 35,48 ****
--- 37,53 ----
      .parameter "singleActivityMode"
  
      .prologue
+     .line 46
      invoke-direct {p0}, Landroid/app/Activity;-><init>()V
  
+     .line 47
      new-instance v0, Landroid/app/LocalActivityManager;
  
      invoke-direct {v0, p0, p1}, Landroid/app/LocalActivityManager;-><init>(Landroid/app/Activity;Z)V
  
      iput-object v0, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
+     .line 48
      return-void
  .end method
  
***************
*** 56,78 ****
--- 61,89 ----
      .parameter "data"
  
      .prologue
+     .line 113
      if-eqz p1, :cond_0
  
+     .line 114
      iget-object v1, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      invoke-virtual {v1, p1}, Landroid/app/LocalActivityManager;->getActivity(Ljava/lang/String;)Landroid/app/Activity;
  
      move-result-object v0
  
+     .line 121
      .local v0, act:Landroid/app/Activity;
      if-eqz v0, :cond_0
  
+     .line 122
      invoke-virtual {v0, p2, p3, p4}, Landroid/app/Activity;->onActivityResult(IILandroid/content/Intent;)V
  
+     .line 127
      .end local v0           #act:Landroid/app/Activity;
      :goto_0
      return-void
  
+     .line 126
      :cond_0
      invoke-super {p0, p1, p2, p3, p4}, Landroid/app/Activity;->dispatchActivityResult(Ljava/lang/String;IILandroid/content/Intent;)V
  
***************
*** 83,88 ****
--- 94,100 ----
      .locals 1
  
      .prologue
+     .line 103
      iget-object v0, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      invoke-virtual {v0}, Landroid/app/LocalActivityManager;->getCurrentActivity()Landroid/app/Activity;
***************
*** 96,101 ****
--- 108,114 ----
      .locals 1
  
      .prologue
+     .line 107
      iget-object v0, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      return-object v0
***************
*** 106,113 ****
--- 119,128 ----
      .parameter "savedInstanceState"
  
      .prologue
+     .line 52
      invoke-super {p0, p1}, Landroid/app/Activity;->onCreate(Landroid/os/Bundle;)V
  
+     .line 53
      if-eqz p1, :cond_0
  
      const-string v1, "android:states"
***************
*** 116,129 ****
--- 131,147 ----
  
      move-result-object v0
  
+     .line 55
      .local v0, states:Landroid/os/Bundle;
      :goto_0
      iget-object v1, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      invoke-virtual {v1, v0}, Landroid/app/LocalActivityManager;->dispatchCreate(Landroid/os/Bundle;)V
  
+     .line 56
      return-void
  
+     .line 53
      .end local v0           #states:Landroid/os/Bundle;
      :cond_0
      const/4 v0, 0x0
***************
*** 135,142 ****
--- 153,162 ----
      .locals 2
  
      .prologue
+     .line 87
      invoke-super {p0}, Landroid/app/Activity;->onDestroy()V
  
+     .line 88
      iget-object v0, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      invoke-virtual {p0}, Landroid/app/ActivityGroup;->isFinishing()Z
***************
*** 145,150 ****
--- 165,171 ----
  
      invoke-virtual {v0, v1}, Landroid/app/LocalActivityManager;->dispatchDestroy(Z)V
  
+     .line 89
      return-void
  .end method
  
***************
*** 152,159 ****
--- 173,182 ----
      .locals 2
  
      .prologue
+     .line 75
      invoke-super {p0}, Landroid/app/Activity;->onPause()V
  
+     .line 76
      iget-object v0, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      invoke-virtual {p0}, Landroid/app/ActivityGroup;->isFinishing()Z
***************
*** 162,167 ****
--- 185,191 ----
  
      invoke-virtual {v0, v1}, Landroid/app/LocalActivityManager;->dispatchPause(Z)V
  
+     .line 77
      return-void
  .end method
  
***************
*** 169,180 ****
--- 193,207 ----
      .locals 1
  
      .prologue
+     .line 60
      invoke-super {p0}, Landroid/app/Activity;->onResume()V
  
+     .line 61
      iget-object v0, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      invoke-virtual {v0}, Landroid/app/LocalActivityManager;->dispatchResume()V
  
+     .line 62
      return-void
  .end method
  
***************
*** 192,197 ****
--- 219,225 ----
      .end annotation
  
      .prologue
+     .line 99
      iget-object v0, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      invoke-virtual {v0}, Landroid/app/LocalActivityManager;->dispatchRetainNonConfigurationInstance()Ljava/util/HashMap;
***************
*** 206,226 ****
--- 234,259 ----
      .parameter "outState"
  
      .prologue
+     .line 66
      invoke-super {p0, p1}, Landroid/app/Activity;->onSaveInstanceState(Landroid/os/Bundle;)V
  
+     .line 67
      iget-object v1, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      invoke-virtual {v1}, Landroid/app/LocalActivityManager;->saveInstanceState()Landroid/os/Bundle;
  
      move-result-object v0
  
+     .line 68
      .local v0, state:Landroid/os/Bundle;
      if-eqz v0, :cond_0
  
+     .line 69
      const-string v1, "android:states"
  
      invoke-virtual {p1, v1, v0}, Landroid/os/Bundle;->putBundle(Ljava/lang/String;Landroid/os/Bundle;)V
  
+     .line 71
      :cond_0
      return-void
  .end method
***************
*** 229,239 ****
--- 262,275 ----
      .locals 1
  
      .prologue
+     .line 81
      invoke-super {p0}, Landroid/app/Activity;->onStop()V
  
+     .line 82
      iget-object v0, p0, Landroid/app/ActivityGroup;->mLocalActivityManager:Landroid/app/LocalActivityManager;
  
      invoke-virtual {v0}, Landroid/app/LocalActivityManager;->dispatchStop()V
  
+     .line 83
      return-void
  .end method
