*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityThread$ResourcesKey.smali	2012-03-23 19:36:42.291452168 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityThread$ResourcesKey.smali.original	2012-03-23 19:34:45.309319001 -0500
***************
*** 29,40 ****
--- 29,44 ----
      .parameter "scale"
  
      .prologue
+     .line 1360
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1361
      iput-object p1, p0, Landroid/app/ActivityThread$ResourcesKey;->mResDir:Ljava/lang/String;
  
+     .line 1362
      iput p2, p0, Landroid/app/ActivityThread$ResourcesKey;->mScale:F
  
+     .line 1363
      iget-object v0, p0, Landroid/app/ActivityThread$ResourcesKey;->mResDir:Ljava/lang/String;
  
      invoke-virtual {v0}, Ljava/lang/String;->hashCode()I
***************
*** 55,60 ****
--- 59,65 ----
  
      iput v0, p0, Landroid/app/ActivityThread$ResourcesKey;->mHash:I
  
+     .line 1364
      return-void
  .end method
  
***************
*** 67,76 ****
--- 72,83 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 1373
      instance-of v2, p1, Landroid/app/ActivityThread$ResourcesKey;
  
      if-nez v2, :cond_1
  
+     .line 1377
      :cond_0
      :goto_0
      return v1
***************
*** 78,85 ****
--- 85,94 ----
      :cond_1
      move-object v0, p1
  
+     .line 1376
      check-cast v0, Landroid/app/ActivityThread$ResourcesKey;
  
+     .line 1377
      .local v0, peer:Landroid/app/ActivityThread$ResourcesKey;
      iget-object v2, p0, Landroid/app/ActivityThread$ResourcesKey;->mResDir:Ljava/lang/String;
  
***************
*** 108,113 ****
--- 117,123 ----
      .locals 1
  
      .prologue
+     .line 1368
      iget v0, p0, Landroid/app/ActivityThread$ResourcesKey;->mHash:I
  
      return v0
