*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityThread$DumpHeapData.smali	2012-03-23 19:36:39.057393195 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityThread$DumpHeapData.smali.original	2012-03-23 19:34:45.284318545 -0500
***************
*** 25,30 ****
--- 25,31 ----
      .locals 0
  
      .prologue
+     .line 472
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
