*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityThread$2.smali	2012-03-23 19:36:42.267451730 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ActivityThread$2.smali.original	2012-03-23 19:34:45.309319001 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 4353
      iput-object p1, p0, Landroid/app/ActivityThread$2;->this$0:Landroid/app/ActivityThread;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,52 ****
--- 42,55 ----
      .parameter "newConfig"
  
      .prologue
+     .line 4355
      iget-object v0, p0, Landroid/app/ActivityThread$2;->this$0:Landroid/app/ActivityThread;
  
      iget-object v1, v0, Landroid/app/ActivityThread;->mPackages:Ljava/util/HashMap;
  
      monitor-enter v1
  
+     .line 4359
      :try_start_0
      iget-object v0, p0, Landroid/app/ActivityThread$2;->this$0:Landroid/app/ActivityThread;
  
***************
*** 58,63 ****
--- 61,67 ----
  
      if-eqz v0, :cond_1
  
+     .line 4362
      iget-object v0, p0, Landroid/app/ActivityThread$2;->this$0:Landroid/app/ActivityThread;
  
      iget-object v0, v0, Landroid/app/ActivityThread;->mPendingConfiguration:Landroid/content/res/Configuration;
***************
*** 74,84 ****
--- 78,90 ----
  
      if-eqz v0, :cond_1
  
+     .line 4364
      :cond_0
      iget-object v0, p0, Landroid/app/ActivityThread$2;->this$0:Landroid/app/ActivityThread;
  
      iput-object p1, v0, Landroid/app/ActivityThread;->mPendingConfiguration:Landroid/content/res/Configuration;
  
+     .line 4366
      iget-object v0, p0, Landroid/app/ActivityThread$2;->this$0:Landroid/app/ActivityThread;
  
      const/16 v2, 0x76
***************
*** 86,96 ****
--- 92,105 ----
      #calls: Landroid/app/ActivityThread;->queueOrSendMessage(ILjava/lang/Object;)V
      invoke-static {v0, v2, p1}, Landroid/app/ActivityThread;->access$300(Landroid/app/ActivityThread;ILjava/lang/Object;)V
  
+     .line 4369
      :cond_1
      monitor-exit v1
  
+     .line 4370
      return-void
  
+     .line 4369
      :catchall_0
      move-exception v0
  
***************
*** 105,110 ****
--- 114,120 ----
      .locals 0
  
      .prologue
+     .line 4372
      return-void
  .end method
  
***************
*** 113,117 ****
--- 123,128 ----
      .parameter "level"
  
      .prologue
+     .line 4374
      return-void
  .end method
