*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/AlarmManager.smali	2012-03-23 19:36:39.276397187 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/AlarmManager.smali.original	2012-03-23 19:34:45.285318563 -0500
***************
*** 33,42 ****
--- 33,45 ----
      .parameter "service"
  
      .prologue
+     .line 90
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 91
      iput-object p1, p0, Landroid/app/AlarmManager;->mService:Landroid/app/IAlarmManager;
  
+     .line 92
      return-void
  .end method
  
***************
*** 47,52 ****
--- 50,56 ----
      .parameter "operation"
  
      .prologue
+     .line 276
      :try_start_0
      iget-object v0, p0, Landroid/app/AlarmManager;->mService:Landroid/app/IAlarmManager;
  
***************
*** 54,62 ****
--- 58,68 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 279
      :goto_0
      return-void
  
+     .line 277
      :catch_0
      move-exception v0
  
***************
*** 70,75 ****
--- 76,82 ----
      .parameter "operation"
  
      .prologue
+     .line 139
      :try_start_0
      iget-object v0, p0, Landroid/app/AlarmManager;->mService:Landroid/app/IAlarmManager;
  
***************
*** 77,85 ****
--- 84,94 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 142
      :goto_0
      return-void
  
+     .line 140
      :catch_0
      move-exception v0
  
***************
*** 94,99 ****
--- 103,109 ----
      .parameter "operation"
  
      .prologue
+     .line 259
      :try_start_0
      iget-object v0, p0, Landroid/app/AlarmManager;->mService:Landroid/app/IAlarmManager;
  
***************
*** 109,117 ****
--- 119,129 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 262
      :goto_0
      return-void
  
+     .line 260
      :catch_0
      move-exception v0
  
***************
*** 126,131 ****
--- 138,144 ----
      .parameter "operation"
  
      .prologue
+     .line 193
      :try_start_0
      iget-object v0, p0, Landroid/app/AlarmManager;->mService:Landroid/app/IAlarmManager;
  
***************
*** 141,149 ****
--- 154,164 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 196
      :goto_0
      return-void
  
+     .line 194
      :catch_0
      move-exception v0
  
***************
*** 155,160 ****
--- 170,176 ----
      .parameter "millis"
  
      .prologue
+     .line 289
      :try_start_0
      iget-object v0, p0, Landroid/app/AlarmManager;->mService:Landroid/app/IAlarmManager;
  
***************
*** 162,170 ****
--- 178,188 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 292
      :goto_0
      return-void
  
+     .line 290
      :catch_0
      move-exception v0
  
***************
*** 176,181 ****
--- 194,200 ----
      .parameter "timeZone"
  
      .prologue
+     .line 302
      :try_start_0
      iget-object v0, p0, Landroid/app/AlarmManager;->mService:Landroid/app/IAlarmManager;
  
***************
*** 183,191 ****
--- 202,212 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 305
      :goto_0
      return-void
  
+     .line 303
      :catch_0
      move-exception v0
  
