*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/AppGlobals.smali	2012-03-23 19:36:33.948300030 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/AppGlobals.smali.original	2012-03-23 19:34:45.197316959 -0500
***************
*** 8,13 ****
--- 8,14 ----
      .locals 0
  
      .prologue
+     .line 25
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 17,22 ****
--- 18,24 ----
      .locals 1
  
      .prologue
+     .line 31
      invoke-static {}, Landroid/app/ActivityThread;->currentApplication()Landroid/app/Application;
  
      move-result-object v0
***************
*** 28,33 ****
--- 30,36 ----
      .locals 1
  
      .prologue
+     .line 39
      invoke-static {}, Landroid/app/ActivityThread;->currentPackageName()Ljava/lang/String;
  
      move-result-object v0
***************
*** 41,57 ****
--- 44,64 ----
      .parameter "defaultValue"
  
      .prologue
+     .line 58
      invoke-static {}, Landroid/app/ActivityThread;->currentActivityThread()Landroid/app/ActivityThread;
  
      move-result-object v0
  
+     .line 59
      .local v0, currentActivityThread:Landroid/app/ActivityThread;
      if-eqz v0, :cond_0
  
+     .line 60
      invoke-virtual {v0, p0, p1}, Landroid/app/ActivityThread;->getIntCoreSetting(Ljava/lang/String;I)I
  
      move-result p1
  
+     .line 62
      .end local p1
      :cond_0
      return p1
***************
*** 61,66 ****
--- 68,74 ----
      .locals 1
  
      .prologue
+     .line 47
      invoke-static {}, Landroid/app/ActivityThread;->getPackageManager()Landroid/content/pm/IPackageManager;
  
      move-result-object v0
