*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Application.smali	2012-03-23 19:36:39.844407546 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Application.smali.original	2012-03-23 19:34:45.291318673 -0500
***************
*** 45,66 ****
--- 45,70 ----
      .locals 1
  
      .prologue
+     .line 63
      const/4 v0, 0x0
  
      invoke-direct {p0, v0}, Landroid/content/ContextWrapper;-><init>(Landroid/content/Context;)V
  
+     .line 44
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/app/Application;->mComponentCallbacks:Ljava/util/ArrayList;
  
+     .line 46
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/app/Application;->mActivityLifecycleCallbacks:Ljava/util/ArrayList;
  
+     .line 64
      return-void
  .end method
  
***************
*** 68,80 ****
--- 72,87 ----
      .locals 3
  
      .prologue
+     .line 226
      const/4 v0, 0x0
  
+     .line 227
      .local v0, callbacks:[Ljava/lang/Object;
      iget-object v2, p0, Landroid/app/Application;->mActivityLifecycleCallbacks:Ljava/util/ArrayList;
  
      monitor-enter v2
  
+     .line 228
      :try_start_0
      iget-object v1, p0, Landroid/app/Application;->mActivityLifecycleCallbacks:Ljava/util/ArrayList;
  
***************
*** 84,100 ****
--- 91,111 ----
  
      if-lez v1, :cond_0
  
+     .line 229
      iget-object v1, p0, Landroid/app/Application;->mActivityLifecycleCallbacks:Ljava/util/ArrayList;
  
      invoke-virtual {v1}, Ljava/util/ArrayList;->toArray()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 231
      :cond_0
      monitor-exit v2
  
+     .line 232
      return-object v0
  
+     .line 231
      :catchall_0
      move-exception v1
  
***************
*** 109,121 ****
--- 120,135 ----
      .locals 3
  
      .prologue
+     .line 216
      const/4 v0, 0x0
  
+     .line 217
      .local v0, callbacks:[Ljava/lang/Object;
      iget-object v2, p0, Landroid/app/Application;->mComponentCallbacks:Ljava/util/ArrayList;
  
      monitor-enter v2
  
+     .line 218
      :try_start_0
      iget-object v1, p0, Landroid/app/Application;->mComponentCallbacks:Ljava/util/ArrayList;
  
***************
*** 125,141 ****
--- 139,159 ----
  
      if-lez v1, :cond_0
  
+     .line 219
      iget-object v1, p0, Landroid/app/Application;->mComponentCallbacks:Ljava/util/ArrayList;
  
      invoke-virtual {v1}, Ljava/util/ArrayList;->toArray()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 221
      :cond_0
      monitor-exit v2
  
+     .line 222
      return-object v0
  
+     .line 221
      :catchall_0
      move-exception v1
  
***************
*** 153,160 ****
--- 171,180 ----
      .parameter "context"
  
      .prologue
+     .line 146
      invoke-virtual {p0, p1}, Landroid/app/Application;->attachBaseContext(Landroid/content/Context;)V
  
+     .line 147
      invoke-static {p1}, Landroid/app/ContextImpl;->getImpl(Landroid/content/Context;)Landroid/app/ContextImpl;
  
      move-result-object v0
***************
*** 163,168 ****
--- 183,189 ----
  
      iput-object v0, p0, Landroid/app/Application;->mLoadedApk:Landroid/app/LoadedApk;
  
+     .line 148
      return-void
  .end method
  
***************
*** 172,184 ****
--- 193,208 ----
      .parameter "savedInstanceState"
  
      .prologue
+     .line 151
      invoke-direct {p0}, Landroid/app/Application;->collectActivityLifecycleCallbacks()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 152
      .local v0, callbacks:[Ljava/lang/Object;
      if-eqz v0, :cond_0
  
+     .line 153
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 187,202 ****
--- 211,229 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 154
      aget-object v2, v0, v1
  
      check-cast v2, Landroid/app/Application$ActivityLifecycleCallbacks;
  
      invoke-interface {v2, p1, p2}, Landroid/app/Application$ActivityLifecycleCallbacks;->onActivityCreated(Landroid/app/Activity;Landroid/os/Bundle;)V
  
+     .line 153
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 158
      .end local v1           #i:I
      :cond_0
      return-void
***************
*** 207,219 ****
--- 234,249 ----
      .parameter "activity"
  
      .prologue
+     .line 207
      invoke-direct {p0}, Landroid/app/Application;->collectActivityLifecycleCallbacks()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 208
      .local v0, callbacks:[Ljava/lang/Object;
      if-eqz v0, :cond_0
  
+     .line 209
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 222,237 ****
--- 252,270 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 210
      aget-object v2, v0, v1
  
      check-cast v2, Landroid/app/Application$ActivityLifecycleCallbacks;
  
      invoke-interface {v2, p1}, Landroid/app/Application$ActivityLifecycleCallbacks;->onActivityDestroyed(Landroid/app/Activity;)V
  
+     .line 209
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 213
      .end local v1           #i:I
      :cond_0
      return-void
***************
*** 242,254 ****
--- 275,290 ----
      .parameter "activity"
  
      .prologue
+     .line 179
      invoke-direct {p0}, Landroid/app/Application;->collectActivityLifecycleCallbacks()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 180
      .local v0, callbacks:[Ljava/lang/Object;
      if-eqz v0, :cond_0
  
+     .line 181
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 257,272 ****
--- 293,311 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 182
      aget-object v2, v0, v1
  
      check-cast v2, Landroid/app/Application$ActivityLifecycleCallbacks;
  
      invoke-interface {v2, p1}, Landroid/app/Application$ActivityLifecycleCallbacks;->onActivityPaused(Landroid/app/Activity;)V
  
+     .line 181
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 185
      .end local v1           #i:I
      :cond_0
      return-void
***************
*** 277,289 ****
--- 316,331 ----
      .parameter "activity"
  
      .prologue
+     .line 170
      invoke-direct {p0}, Landroid/app/Application;->collectActivityLifecycleCallbacks()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 171
      .local v0, callbacks:[Ljava/lang/Object;
      if-eqz v0, :cond_0
  
+     .line 172
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 292,307 ****
--- 334,352 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 173
      aget-object v2, v0, v1
  
      check-cast v2, Landroid/app/Application$ActivityLifecycleCallbacks;
  
      invoke-interface {v2, p1}, Landroid/app/Application$ActivityLifecycleCallbacks;->onActivityResumed(Landroid/app/Activity;)V
  
+     .line 172
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 176
      .end local v1           #i:I
      :cond_0
      return-void
***************
*** 313,325 ****
--- 358,373 ----
      .parameter "outState"
  
      .prologue
+     .line 197
      invoke-direct {p0}, Landroid/app/Application;->collectActivityLifecycleCallbacks()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 198
      .local v0, callbacks:[Ljava/lang/Object;
      if-eqz v0, :cond_0
  
+     .line 199
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 328,343 ****
--- 376,394 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 200
      aget-object v2, v0, v1
  
      check-cast v2, Landroid/app/Application$ActivityLifecycleCallbacks;
  
      invoke-interface {v2, p1, p2}, Landroid/app/Application$ActivityLifecycleCallbacks;->onActivitySaveInstanceState(Landroid/app/Activity;Landroid/os/Bundle;)V
  
+     .line 199
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 204
      .end local v1           #i:I
      :cond_0
      return-void
***************
*** 348,360 ****
--- 399,414 ----
      .parameter "activity"
  
      .prologue
+     .line 161
      invoke-direct {p0}, Landroid/app/Application;->collectActivityLifecycleCallbacks()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 162
      .local v0, callbacks:[Ljava/lang/Object;
      if-eqz v0, :cond_0
  
+     .line 163
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 363,378 ****
--- 417,435 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 164
      aget-object v2, v0, v1
  
      check-cast v2, Landroid/app/Application$ActivityLifecycleCallbacks;
  
      invoke-interface {v2, p1}, Landroid/app/Application$ActivityLifecycleCallbacks;->onActivityStarted(Landroid/app/Activity;)V
  
+     .line 163
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 167
      .end local v1           #i:I
      :cond_0
      return-void
***************
*** 383,395 ****
--- 440,455 ----
      .parameter "activity"
  
      .prologue
+     .line 188
      invoke-direct {p0}, Landroid/app/Application;->collectActivityLifecycleCallbacks()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 189
      .local v0, callbacks:[Ljava/lang/Object;
      if-eqz v0, :cond_0
  
+     .line 190
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 398,413 ****
--- 458,476 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 191
      aget-object v2, v0, v1
  
      check-cast v2, Landroid/app/Application$ActivityLifecycleCallbacks;
  
      invoke-interface {v2, p1}, Landroid/app/Application$ActivityLifecycleCallbacks;->onActivityStopped(Landroid/app/Activity;)V
  
+     .line 190
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 194
      .end local v1           #i:I
      :cond_0
      return-void
***************
*** 418,430 ****
--- 481,496 ----
      .parameter "newConfig"
  
      .prologue
+     .line 87
      invoke-direct {p0}, Landroid/app/Application;->collectComponentCallbacks()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 88
      .local v0, callbacks:[Ljava/lang/Object;
      if-eqz v0, :cond_0
  
+     .line 89
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 433,448 ****
--- 499,517 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 90
      aget-object v2, v0, v1
  
      check-cast v2, Landroid/content/ComponentCallbacks;
  
      invoke-interface {v2, p1}, Landroid/content/ComponentCallbacks;->onConfigurationChanged(Landroid/content/res/Configuration;)V
  
+     .line 89
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 93
      .end local v1           #i:I
      :cond_0
      return-void
***************
*** 452,457 ****
--- 521,527 ----
      .locals 0
  
      .prologue
+     .line 75
      return-void
  .end method
  
***************
*** 459,471 ****
--- 529,544 ----
      .locals 3
  
      .prologue
+     .line 96
      invoke-direct {p0}, Landroid/app/Application;->collectComponentCallbacks()[Ljava/lang/Object;
  
      move-result-object v0
  
+     .line 97
      .local v0, callbacks:[Ljava/lang/Object;
      if-eqz v0, :cond_0
  
+     .line 98
      const/4 v1, 0x0
  
      .local v1, i:I
***************
*** 474,489 ****
--- 547,565 ----
  
      if-ge v1, v2, :cond_0
  
+     .line 99
      aget-object v2, v0, v1
  
      check-cast v2, Landroid/content/ComponentCallbacks;
  
      invoke-interface {v2}, Landroid/content/ComponentCallbacks;->onLowMemory()V
  
+     .line 98
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 102
      .end local v1           #i:I
      :cond_0
      return-void
***************
*** 493,498 ****
--- 569,575 ----
      .locals 0
  
      .prologue
+     .line 84
      return-void
  .end method
  
***************
*** 501,513 ****
--- 578,593 ----
      .parameter "level"
  
      .prologue
+     .line 105
      invoke-direct {p0}, Landroid/app/Application;->collectComponentCallbacks()[Ljava/lang/Object;
  
      move-result-object v1
  
+     .line 106
      .local v1, callbacks:[Ljava/lang/Object;
      if-eqz v1, :cond_1
  
+     .line 107
      const/4 v2, 0x0
  
      .local v2, i:I
***************
*** 516,538 ****
--- 596,623 ----
  
      if-ge v2, v3, :cond_1
  
+     .line 108
      aget-object v0, v1, v2
  
+     .line 109
      .local v0, c:Ljava/lang/Object;
      instance-of v3, v0, Landroid/content/ComponentCallbacks2;
  
      if-eqz v3, :cond_0
  
+     .line 110
      check-cast v0, Landroid/content/ComponentCallbacks2;
  
      .end local v0           #c:Ljava/lang/Object;
      invoke-interface {v0, p1}, Landroid/content/ComponentCallbacks2;->onTrimMemory(I)V
  
+     .line 107
      :cond_0
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 114
      .end local v2           #i:I
      :cond_1
      return-void
***************
*** 543,561 ****
--- 628,651 ----
      .parameter "callback"
  
      .prologue
+     .line 129
      iget-object v1, p0, Landroid/app/Application;->mActivityLifecycleCallbacks:Ljava/util/ArrayList;
  
      monitor-enter v1
  
+     .line 130
      :try_start_0
      iget-object v0, p0, Landroid/app/Application;->mActivityLifecycleCallbacks:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 131
      monitor-exit v1
  
+     .line 132
      return-void
  
+     .line 131
      :catchall_0
      move-exception v0
  
***************
*** 571,589 ****
--- 661,684 ----
      .parameter "callback"
  
      .prologue
+     .line 117
      iget-object v1, p0, Landroid/app/Application;->mComponentCallbacks:Ljava/util/ArrayList;
  
      monitor-enter v1
  
+     .line 118
      :try_start_0
      iget-object v0, p0, Landroid/app/Application;->mComponentCallbacks:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 119
      monitor-exit v1
  
+     .line 120
      return-void
  
+     .line 119
      :catchall_0
      move-exception v0
  
***************
*** 599,617 ****
--- 694,717 ----
      .parameter "callback"
  
      .prologue
+     .line 135
      iget-object v1, p0, Landroid/app/Application;->mActivityLifecycleCallbacks:Ljava/util/ArrayList;
  
      monitor-enter v1
  
+     .line 136
      :try_start_0
      iget-object v0, p0, Landroid/app/Application;->mActivityLifecycleCallbacks:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->remove(Ljava/lang/Object;)Z
  
+     .line 137
      monitor-exit v1
  
+     .line 138
      return-void
  
+     .line 137
      :catchall_0
      move-exception v0
  
***************
*** 627,645 ****
--- 727,750 ----
      .parameter "callback"
  
      .prologue
+     .line 123
      iget-object v1, p0, Landroid/app/Application;->mComponentCallbacks:Ljava/util/ArrayList;
  
      monitor-enter v1
  
+     .line 124
      :try_start_0
      iget-object v0, p0, Landroid/app/Application;->mComponentCallbacks:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->remove(Ljava/lang/Object;)Z
  
+     .line 125
      monitor-exit v1
  
+     .line 126
      return-void
  
+     .line 125
      :catchall_0
      move-exception v0
  
