*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ApplicationErrorReport$AnrInfo.smali	2012-03-23 19:36:44.080484791 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ApplicationErrorReport$AnrInfo.smali.original	2012-03-23 19:34:45.323319257 -0500
***************
*** 27,34 ****
--- 27,36 ----
      .locals 0
  
      .prologue
+     .line 424
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 425
      return-void
  .end method
  
***************
*** 37,62 ****
--- 39,69 ----
      .parameter "in"
  
      .prologue
+     .line 430
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 431
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/app/ApplicationErrorReport$AnrInfo;->activity:Ljava/lang/String;
  
+     .line 432
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/app/ApplicationErrorReport$AnrInfo;->cause:Ljava/lang/String;
  
+     .line 433
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/app/ApplicationErrorReport$AnrInfo;->info:Ljava/lang/String;
  
+     .line 434
      return-void
  .end method
  
***************
*** 68,73 ****
--- 75,81 ----
      .parameter "prefix"
  
      .prologue
+     .line 449
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 94,99 ****
--- 102,108 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 450
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 120,125 ****
--- 129,135 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 451
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 146,151 ****
--- 156,162 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 452
      return-void
  .end method
  
***************
*** 155,171 ****
--- 166,186 ----
      .parameter "flags"
  
      .prologue
+     .line 440
      iget-object v0, p0, Landroid/app/ApplicationErrorReport$AnrInfo;->activity:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 441
      iget-object v0, p0, Landroid/app/ApplicationErrorReport$AnrInfo;->cause:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 442
      iget-object v0, p0, Landroid/app/ApplicationErrorReport$AnrInfo;->info:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 443
      return-void
  .end method
