*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ContextImpl$24.smali	2012-03-23 19:36:41.637440242 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ContextImpl$24.smali.original	2012-03-23 19:34:45.305318929 -0500
***************
*** 19,24 ****
--- 19,25 ----
      .locals 0
  
      .prologue
+     .line 393
      invoke-direct {p0}, Landroid/app/ContextImpl$ServiceFetcher;-><init>()V
  
      return-void
***************
*** 31,36 ****
--- 32,38 ----
      .parameter "ctx"
  
      .prologue
+     .line 395
      new-instance v0, Landroid/app/StatusBarManager;
  
      invoke-virtual {p1}, Landroid/app/ContextImpl;->getOuterContext()Landroid/content/Context;
