*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/DownloadManager$Request.smali	2012-03-23 19:36:38.427381706 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/DownloadManager$Request.smali.original	2012-03-23 19:34:45.277318417 -0500
***************
*** 78,83 ****
--- 78,84 ----
      .locals 1
  
      .prologue
+     .line 331
      const-class v0, Landroid/app/DownloadManager;
  
      invoke-virtual {v0}, Ljava/lang/Class;->desiredAssertionStatus()Z
***************
*** 108,148 ****
--- 109,161 ----
  
      const/4 v2, 0x0
  
+     .line 401
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 346
      new-instance v1, Ljava/util/ArrayList;
  
      invoke-direct {v1}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v1, p0, Landroid/app/DownloadManager$Request;->mRequestHeaders:Ljava/util/List;
  
+     .line 350
      iput-boolean v3, p0, Landroid/app/DownloadManager$Request;->mRoamingAllowed:Z
  
+     .line 351
      const/4 v1, -0x1
  
      iput v1, p0, Landroid/app/DownloadManager$Request;->mAllowedNetworkTypes:I
  
+     .line 352
      iput-boolean v3, p0, Landroid/app/DownloadManager$Request;->mIsVisibleInDownloadsUi:Z
  
+     .line 353
      iput-boolean v2, p0, Landroid/app/DownloadManager$Request;->mScannable:Z
  
+     .line 354
      iput-boolean v2, p0, Landroid/app/DownloadManager$Request;->mUseSystemCache:Z
  
+     .line 396
      iput v2, p0, Landroid/app/DownloadManager$Request;->mNotificationVisibility:I
  
+     .line 402
      if-nez p1, :cond_0
  
+     .line 403
      new-instance v1, Ljava/lang/NullPointerException;
  
      invoke-direct {v1}, Ljava/lang/NullPointerException;-><init>()V
  
      throw v1
  
+     .line 405
      :cond_0
      invoke-virtual {p1}, Landroid/net/Uri;->getScheme()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 406
      .local v0, scheme:Ljava/lang/String;
      if-eqz v0, :cond_1
  
***************
*** 162,167 ****
--- 175,181 ----
  
      if-nez v1, :cond_2
  
+     .line 407
      :cond_1
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
***************
*** 187,195 ****
--- 201,211 ----
  
      throw v1
  
+     .line 409
      :cond_2
      iput-object p1, p0, Landroid/app/DownloadManager$Request;->mUri:Landroid/net/Uri;
  
+     .line 410
      return-void
  .end method
  
***************
*** 202,235 ****
--- 218,261 ----
  
      const/4 v1, 0x0
  
+     .line 412
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 346
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/app/DownloadManager$Request;->mRequestHeaders:Ljava/util/List;
  
+     .line 350
      iput-boolean v2, p0, Landroid/app/DownloadManager$Request;->mRoamingAllowed:Z
  
+     .line 351
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/app/DownloadManager$Request;->mAllowedNetworkTypes:I
  
+     .line 352
      iput-boolean v2, p0, Landroid/app/DownloadManager$Request;->mIsVisibleInDownloadsUi:Z
  
+     .line 353
      iput-boolean v1, p0, Landroid/app/DownloadManager$Request;->mScannable:Z
  
+     .line 354
      iput-boolean v1, p0, Landroid/app/DownloadManager$Request;->mUseSystemCache:Z
  
+     .line 396
      iput v1, p0, Landroid/app/DownloadManager$Request;->mNotificationVisibility:I
  
+     .line 413
      invoke-static {p1}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/app/DownloadManager$Request;->mUri:Landroid/net/Uri;
  
+     .line 414
      return-void
  .end method
  
***************
*** 238,245 ****
--- 264,273 ----
      .parameter "values"
  
      .prologue
+     .line 684
      const/4 v3, 0x0
  
+     .line 685
      .local v3, index:I
      iget-object v4, p0, Landroid/app/DownloadManager$Request;->mRequestHeaders:Ljava/util/List;
  
***************
*** 261,266 ****
--- 289,295 ----
  
      check-cast v0, Landroid/util/Pair;
  
+     .line 686
      .local v0, header:Landroid/util/Pair;,"Landroid/util/Pair<Ljava/lang/String;Ljava/lang/String;>;"
      new-instance v5, Ljava/lang/StringBuilder;
  
***************
*** 292,297 ****
--- 321,327 ----
  
      move-result-object v1
  
+     .line 687
      .local v1, headerString:Ljava/lang/String;
      new-instance v4, Ljava/lang/StringBuilder;
  
***************
*** 313,322 ****
--- 343,355 ----
  
      invoke-virtual {p1, v4, v1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 688
      add-int/lit8 v3, v3, 0x1
  
+     .line 689
      goto :goto_0
  
+     .line 690
      .end local v0           #header:Landroid/util/Pair;,"Landroid/util/Pair<Ljava/lang/String;Ljava/lang/String;>;"
      .end local v1           #headerString:Ljava/lang/String;
      :cond_0
***************
*** 330,343 ****
--- 363,379 ----
      .parameter "value"
  
      .prologue
+     .line 693
      if-eqz p3, :cond_0
  
+     .line 694
      invoke-virtual {p3}, Ljava/lang/Object;->toString()Ljava/lang/String;
  
      move-result-object v0
  
      invoke-virtual {p1, p2, v0}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 696
      :cond_0
      return-void
  .end method
***************
*** 348,363 ****
      .parameter "subPath"
  
      .prologue
      if-nez p2, :cond_0
  
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string v1, "subPath cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
      :cond_0
      invoke-static {p1}, Landroid/net/Uri;->fromFile(Ljava/io/File;)Landroid/net/Uri;
  
--- 384,402 ----
      .parameter "subPath"
  
      .prologue
+     .line 501
      if-nez p2, :cond_0
  
+     .line 502
      new-instance v0, Ljava/lang/NullPointerException;
  
!     const-string/jumbo v1, "subPath cannot be null"
  
      invoke-direct {v0, v1}, Ljava/lang/NullPointerException;-><init>(Ljava/lang/String;)V
  
      throw v0
  
+     .line 504
      :cond_0
      invoke-static {p1}, Landroid/net/Uri;->fromFile(Ljava/io/File;)Landroid/net/Uri;
  
***************
*** 369,374 ****
--- 408,414 ----
  
      iput-object v0, p0, Landroid/app/DownloadManager$Request;->mDestinationUri:Landroid/net/Uri;
  
+     .line 505
      return-void
  .end method
  
***************
*** 380,387 ****
--- 420,429 ----
      .parameter "value"
  
      .prologue
+     .line 525
      if-nez p1, :cond_0
  
+     .line 526
      new-instance v0, Ljava/lang/NullPointerException;
  
      const-string v1, "header cannot be null"
***************
*** 390,395 ****
--- 432,438 ----
  
      throw v0
  
+     .line 528
      :cond_0
      const-string v0, ":"
  
***************
*** 399,404 ****
--- 442,448 ----
  
      if-eqz v0, :cond_1
  
+     .line 529
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "header may not contain \':\'"
***************
*** 407,417 ****
--- 451,464 ----
  
      throw v0
  
+     .line 531
      :cond_1
      if-nez p2, :cond_2
  
+     .line 532
      const-string p2, ""
  
+     .line 534
      :cond_2
      iget-object v0, p0, Landroid/app/DownloadManager$Request;->mRequestHeaders:Ljava/util/List;
  
***************
*** 421,426 ****
--- 468,474 ----
  
      invoke-interface {v0, v1}, Ljava/util/List;->add(Ljava/lang/Object;)Z
  
+     .line 535
      return-object p0
  .end method
  
***************
*** 428,437 ****
--- 476,487 ----
      .locals 1
  
      .prologue
+     .line 512
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/app/DownloadManager$Request;->mScannable:Z
  
+     .line 513
      return-void
  .end method
  
***************
*** 440,447 ****
--- 490,499 ----
      .parameter "flags"
  
      .prologue
+     .line 618
      iput p1, p0, Landroid/app/DownloadManager$Request;->mAllowedNetworkTypes:I
  
+     .line 619
      return-object p0
  .end method
  
***************
*** 450,457 ****
--- 502,511 ----
      .parameter "allowed"
  
      .prologue
+     .line 629
      iput-boolean p1, p0, Landroid/app/DownloadManager$Request;->mRoamingAllowed:Z
  
+     .line 630
      return-object p0
  .end method
  
***************
*** 460,467 ****
--- 514,523 ----
      .parameter "description"
  
      .prologue
+     .line 554
      iput-object p1, p0, Landroid/app/DownloadManager$Request;->mDescription:Ljava/lang/CharSequence;
  
+     .line 555
      return-object p0
  .end method
  
***************
*** 472,483 ****
--- 528,541 ----
      .parameter "subPath"
  
      .prologue
+     .line 466
      invoke-virtual {p1, p2}, Landroid/content/Context;->getExternalFilesDir(Ljava/lang/String;)Ljava/io/File;
  
      move-result-object v0
  
      invoke-direct {p0, v0, p3}, Landroid/app/DownloadManager$Request;->setDestinationFromBase(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 467
      return-object p0
  .end method
  
***************
*** 487,496 ****
--- 545,556 ----
      .parameter "subPath"
  
      .prologue
+     .line 484
      invoke-static {p1}, Landroid/os/Environment;->getExternalStoragePublicDirectory(Ljava/lang/String;)Ljava/io/File;
  
      move-result-object v0
  
+     .line 485
      .local v0, file:Ljava/io/File;
      invoke-virtual {v0}, Ljava/io/File;->exists()Z
  
***************
*** 498,509 ****
--- 558,571 ----
  
      if-eqz v1, :cond_0
  
+     .line 486
      invoke-virtual {v0}, Ljava/io/File;->isDirectory()Z
  
      move-result v1
  
      if-nez v1, :cond_1
  
+     .line 487
      new-instance v1, Ljava/lang/IllegalStateException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 532,537 ****
--- 594,600 ----
  
      throw v1
  
+     .line 491
      :cond_0
      invoke-virtual {v0}, Ljava/io/File;->mkdir()Z
  
***************
*** 539,544 ****
--- 602,608 ----
  
      if-nez v1, :cond_1
  
+     .line 492
      new-instance v1, Ljava/lang/IllegalStateException;
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 567,575 ****
--- 631,641 ----
  
      throw v1
  
+     .line 496
      :cond_1
      invoke-direct {p0, v0, p2}, Landroid/app/DownloadManager$Request;->setDestinationFromBase(Ljava/io/File;Ljava/lang/String;)V
  
+     .line 497
      return-object p0
  .end method
  
***************
*** 577,586 ****
--- 643,654 ----
      .locals 1
  
      .prologue
+     .line 448
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/app/DownloadManager$Request;->mUseSystemCache:Z
  
+     .line 449
      return-object p0
  .end method
  
***************
*** 589,596 ****
--- 657,666 ----
      .parameter "uri"
  
      .prologue
+     .line 430
      iput-object p1, p0, Landroid/app/DownloadManager$Request;->mDestinationUri:Landroid/net/Uri;
  
+     .line 431
      return-object p0
  .end method
  
***************
*** 599,606 ****
--- 669,678 ----
      .parameter "mimeType"
  
      .prologue
+     .line 566
      iput-object p1, p0, Landroid/app/DownloadManager$Request;->mMimeType:Ljava/lang/String;
  
+     .line 567
      return-object p0
  .end method
  
***************
*** 609,616 ****
--- 681,690 ----
      .parameter "visibility"
  
      .prologue
+     .line 607
      iput p1, p0, Landroid/app/DownloadManager$Request;->mNotificationVisibility:I
  
+     .line 608
      return-object p0
  .end method
  
***************
*** 621,626 ****
--- 695,701 ----
      .end annotation
  
      .prologue
+     .line 585
      if-eqz p1, :cond_0
  
      const/4 v0, 0x0
***************
*** 647,654 ****
--- 722,731 ----
      .parameter "title"
  
      .prologue
+     .line 545
      iput-object p1, p0, Landroid/app/DownloadManager$Request;->mTitle:Ljava/lang/CharSequence;
  
+     .line 546
      return-object p0
  .end method
  
***************
*** 657,664 ****
--- 734,743 ----
      .parameter "isVisible"
  
      .prologue
+     .line 640
      iput-boolean p1, p0, Landroid/app/DownloadManager$Request;->mIsVisibleInDownloadsUi:Z
  
+     .line 641
      return-object p0
  .end method
  
***************
*** 669,678 ****
--- 748,759 ----
      .prologue
      const/4 v2, 0x2
  
+     .line 648
      new-instance v0, Landroid/content/ContentValues;
  
      invoke-direct {v0}, Landroid/content/ContentValues;-><init>()V
  
+     .line 649
      .local v0, values:Landroid/content/ContentValues;
      sget-boolean v1, Landroid/app/DownloadManager$Request;->$assertionsDisabled:Z
  
***************
*** 688,695 ****
  
      throw v1
  
      :cond_0
!     const-string v1, "uri"
  
      iget-object v3, p0, Landroid/app/DownloadManager$Request;->mUri:Landroid/net/Uri;
  
--- 769,777 ----
  
      throw v1
  
+     .line 650
      :cond_0
!     const-string/jumbo v1, "uri"
  
      iget-object v3, p0, Landroid/app/DownloadManager$Request;->mUri:Landroid/net/Uri;
  
***************
*** 699,704 ****
--- 781,787 ----
  
      invoke-virtual {v0, v1, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 651
      const-string v1, "is_public_api"
  
      const/4 v3, 0x1
***************
*** 709,722 ****
  
      invoke-virtual {v0, v1, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Boolean;)V
  
!     const-string v1, "notificationpackage"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
      iget-object v1, p0, Landroid/app/DownloadManager$Request;->mDestinationUri:Landroid/net/Uri;
  
      if-eqz v1, :cond_3
  
      const-string v1, "destination"
  
      const/4 v3, 0x4
--- 792,808 ----
  
      invoke-virtual {v0, v1, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Boolean;)V
  
!     .line 652
!     const-string/jumbo v1, "notificationpackage"
  
      invoke-virtual {v0, v1, p1}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 654
      iget-object v1, p0, Landroid/app/DownloadManager$Request;->mDestinationUri:Landroid/net/Uri;
  
      if-eqz v1, :cond_3
  
+     .line 655
      const-string v1, "destination"
  
      const/4 v3, 0x4
***************
*** 727,732 ****
--- 813,819 ----
  
      invoke-virtual {v0, v1, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 656
      const-string v1, "hint"
  
      iget-object v3, p0, Landroid/app/DownloadManager$Request;->mDestinationUri:Landroid/net/Uri;
***************
*** 737,744 ****
  
      invoke-virtual {v0, v1, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
      :goto_0
!     const-string v1, "scanned"
  
      iget-boolean v3, p0, Landroid/app/DownloadManager$Request;->mScannable:Z
  
--- 824,832 ----
  
      invoke-virtual {v0, v1, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 664
      :goto_0
!     const-string/jumbo v1, "scanned"
  
      iget-boolean v3, p0, Landroid/app/DownloadManager$Request;->mScannable:Z
  
***************
*** 753,758 ****
--- 841,847 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 667
      iget-object v1, p0, Landroid/app/DownloadManager$Request;->mRequestHeaders:Ljava/util/List;
  
      invoke-interface {v1}, Ljava/util/List;->isEmpty()Z
***************
*** 761,788 ****
  
      if-nez v1, :cond_2
  
      invoke-direct {p0, v0}, Landroid/app/DownloadManager$Request;->encodeHttpHeaders(Landroid/content/ContentValues;)V
  
      :cond_2
!     const-string v1, "title"
  
      iget-object v2, p0, Landroid/app/DownloadManager$Request;->mTitle:Ljava/lang/CharSequence;
  
      invoke-direct {p0, v0, v1, v2}, Landroid/app/DownloadManager$Request;->putIfNonNull(Landroid/content/ContentValues;Ljava/lang/String;Ljava/lang/Object;)V
  
      const-string v1, "description"
  
      iget-object v2, p0, Landroid/app/DownloadManager$Request;->mDescription:Ljava/lang/CharSequence;
  
      invoke-direct {p0, v0, v1, v2}, Landroid/app/DownloadManager$Request;->putIfNonNull(Landroid/content/ContentValues;Ljava/lang/String;Ljava/lang/Object;)V
  
!     const-string v1, "mimetype"
  
      iget-object v2, p0, Landroid/app/DownloadManager$Request;->mMimeType:Ljava/lang/String;
  
      invoke-direct {p0, v0, v1, v2}, Landroid/app/DownloadManager$Request;->putIfNonNull(Landroid/content/ContentValues;Ljava/lang/String;Ljava/lang/Object;)V
  
!     const-string v1, "visibility"
  
      iget v2, p0, Landroid/app/DownloadManager$Request;->mNotificationVisibility:I
  
--- 850,882 ----
  
      if-nez v1, :cond_2
  
+     .line 668
      invoke-direct {p0, v0}, Landroid/app/DownloadManager$Request;->encodeHttpHeaders(Landroid/content/ContentValues;)V
  
+     .line 671
      :cond_2
!     const-string/jumbo v1, "title"
  
      iget-object v2, p0, Landroid/app/DownloadManager$Request;->mTitle:Ljava/lang/CharSequence;
  
      invoke-direct {p0, v0, v1, v2}, Landroid/app/DownloadManager$Request;->putIfNonNull(Landroid/content/ContentValues;Ljava/lang/String;Ljava/lang/Object;)V
  
+     .line 672
      const-string v1, "description"
  
      iget-object v2, p0, Landroid/app/DownloadManager$Request;->mDescription:Ljava/lang/CharSequence;
  
      invoke-direct {p0, v0, v1, v2}, Landroid/app/DownloadManager$Request;->putIfNonNull(Landroid/content/ContentValues;Ljava/lang/String;Ljava/lang/Object;)V
  
!     .line 673
!     const-string/jumbo v1, "mimetype"
  
      iget-object v2, p0, Landroid/app/DownloadManager$Request;->mMimeType:Ljava/lang/String;
  
      invoke-direct {p0, v0, v1, v2}, Landroid/app/DownloadManager$Request;->putIfNonNull(Landroid/content/ContentValues;Ljava/lang/String;Ljava/lang/Object;)V
  
!     .line 675
!     const-string/jumbo v1, "visibility"
  
      iget v2, p0, Landroid/app/DownloadManager$Request;->mNotificationVisibility:I
  
***************
*** 792,797 ****
--- 886,892 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 676
      const-string v1, "allowed_network_types"
  
      iget v2, p0, Landroid/app/DownloadManager$Request;->mAllowedNetworkTypes:I
***************
*** 802,807 ****
--- 897,903 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 677
      const-string v1, "allow_roaming"
  
      iget-boolean v2, p0, Landroid/app/DownloadManager$Request;->mRoamingAllowed:Z
***************
*** 812,817 ****
--- 908,914 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Boolean;)V
  
+     .line 678
      const-string v1, "is_visible_in_downloads_ui"
  
      iget-boolean v2, p0, Landroid/app/DownloadManager$Request;->mIsVisibleInDownloadsUi:Z
***************
*** 822,829 ****
--- 919,928 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Boolean;)V
  
+     .line 680
      return-object v0
  
+     .line 658
      :cond_3
      const-string v3, "destination"
  
