*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/DownloadManager.smali	2012-03-23 19:36:36.497346511 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/DownloadManager.smali.original	2012-03-23 19:34:45.246317853 -0500
***************
*** 108,113 ****
--- 108,114 ----
      .locals 3
  
      .prologue
+     .line 301
      const/16 v0, 0xf
  
      new-array v0, v0, [Ljava/lang/String;
***************
*** 126,132 ****
  
      const/4 v1, 0x2
  
!     const-string v2, "mediaprovider_uri"
  
      aput-object v2, v0, v1
  
--- 127,133 ----
  
      const/4 v1, 0x2
  
!     const-string/jumbo v2, "mediaprovider_uri"
  
      aput-object v2, v0, v1
  
***************
*** 138,144 ****
  
      const/4 v1, 0x4
  
!     const-string v2, "title"
  
      aput-object v2, v0, v1
  
--- 139,145 ----
  
      const/4 v1, 0x4
  
!     const-string/jumbo v2, "title"
  
      aput-object v2, v0, v1
  
***************
*** 150,162 ****
  
      const/4 v1, 0x6
  
!     const-string v2, "uri"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x7
  
!     const-string v2, "status"
  
      aput-object v2, v0, v1
  
--- 151,163 ----
  
      const/4 v1, 0x6
  
!     const-string/jumbo v2, "uri"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x7
  
!     const-string/jumbo v2, "status"
  
      aput-object v2, v0, v1
  
***************
*** 168,180 ****
  
      const/16 v1, 0x9
  
!     const-string v2, "mimetype AS media_type"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0xa
  
!     const-string v2, "total_bytes AS total_size"
  
      aput-object v2, v0, v1
  
--- 169,181 ----
  
      const/16 v1, 0x9
  
!     const-string/jumbo v2, "mimetype AS media_type"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0xa
  
!     const-string/jumbo v2, "total_bytes AS total_size"
  
      aput-object v2, v0, v1
  
***************
*** 213,228 ****
--- 214,234 ----
      .parameter "packageName"
  
      .prologue
+     .line 857
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 852
      sget-object v0, Landroid/provider/Downloads$Impl;->CONTENT_URI:Landroid/net/Uri;
  
      iput-object v0, p0, Landroid/app/DownloadManager;->mBaseUri:Landroid/net/Uri;
  
+     .line 858
      iput-object p1, p0, Landroid/app/DownloadManager;->mResolver:Landroid/content/ContentResolver;
  
+     .line 859
      iput-object p2, p0, Landroid/app/DownloadManager;->mPackageName:Ljava/lang/String;
  
+     .line 860
      return-void
  .end method
  
***************
*** 231,236 ****
--- 237,243 ----
      .parameter "context"
  
      .prologue
+     .line 1075
      :try_start_0
      invoke-virtual {p0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
***************
*** 248,259 ****
--- 255,269 ----
  
      move-result-object v1
  
+     .line 1078
      :goto_0
      return-object v1
  
+     .line 1077
      :catch_0
      move-exception v0
  
+     .line 1078
      .local v0, exc:Landroid/provider/Settings$SettingNotFoundException;
      const/4 v1, 0x0
  
***************
*** 265,270 ****
--- 275,281 ----
      .parameter "context"
  
      .prologue
+     .line 1093
      :try_start_0
      invoke-virtual {p0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
***************
*** 282,293 ****
--- 293,307 ----
  
      move-result-object v1
  
+     .line 1096
      :goto_0
      return-object v1
  
+     .line 1095
      :catch_0
      move-exception v0
  
+     .line 1096
      .local v0, exc:Landroid/provider/Settings$SettingNotFoundException;
      const/4 v1, 0x0
  
***************
*** 299,308 ****
--- 313,324 ----
      .parameter "ids"
  
      .prologue
+     .line 1193
      array-length v2, p0
  
      new-array v1, v2, [Ljava/lang/String;
  
+     .line 1194
      .local v1, whereArgs:[Ljava/lang/String;
      const/4 v0, 0x0
  
***************
*** 312,317 ****
--- 328,334 ----
  
      if-ge v0, v2, :cond_0
  
+     .line 1195
      aget-wide v2, p0, v0
  
      invoke-static {v2, v3}, Ljava/lang/Long;->toString(J)Ljava/lang/String;
***************
*** 320,329 ****
--- 337,348 ----
  
      aput-object v2, v1, v0
  
+     .line 1194
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 1197
      :cond_0
      return-object v1
  .end method
***************
*** 333,347 ****
--- 352,369 ----
      .parameter "ids"
  
      .prologue
+     .line 1176
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 1177
      .local v1, whereClause:Ljava/lang/StringBuilder;
      const-string v2, "("
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 1178
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 350,379 ****
--- 372,408 ----
  
      if-ge v0, v2, :cond_1
  
+     .line 1179
      if-lez v0, :cond_0
  
+     .line 1180
      const-string v2, "OR "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 1182
      :cond_0
      const-string v2, "_id"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 1183
      const-string v2, " = ? "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 1178
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 1185
      :cond_1
      const-string v2, ")"
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 1186
      invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v2
***************
*** 387,398 ****
--- 416,429 ----
      .parameter "val"
  
      .prologue
+     .line 1160
      invoke-static {p1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
      move-result v0
  
      if-eqz v0, :cond_0
  
+     .line 1161
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 417,422 ****
--- 448,454 ----
  
      throw v0
  
+     .line 1163
      :cond_0
      return-void
  .end method
***************
*** 436,463 ****
      .prologue
      const/4 v4, 0x2
  
!     const-string v3, "title"
  
      invoke-static {v3, p1}, Landroid/app/DownloadManager;->validateArgumentIsNonEmpty(Ljava/lang/String;Ljava/lang/String;)V
  
      const-string v3, "description"
  
      invoke-static {v3, p2}, Landroid/app/DownloadManager;->validateArgumentIsNonEmpty(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v3, "path"
  
      invoke-static {v3, p5}, Landroid/app/DownloadManager;->validateArgumentIsNonEmpty(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v3, "mimeType"
  
      invoke-static {v3, p4}, Landroid/app/DownloadManager;->validateArgumentIsNonEmpty(Ljava/lang/String;Ljava/lang/String;)V
  
      const-wide/16 v5, 0x0
  
      cmp-long v3, p6, v5
  
      if-gtz v3, :cond_0
  
      new-instance v3, Ljava/lang/IllegalArgumentException;
  
      const-string v4, " invalid value for param: totalBytes"
--- 468,501 ----
      .prologue
      const/4 v4, 0x2
  
!     .line 1126
!     const-string/jumbo v3, "title"
  
      invoke-static {v3, p1}, Landroid/app/DownloadManager;->validateArgumentIsNonEmpty(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1127
      const-string v3, "description"
  
      invoke-static {v3, p2}, Landroid/app/DownloadManager;->validateArgumentIsNonEmpty(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 1128
!     const-string/jumbo v3, "path"
  
      invoke-static {v3, p5}, Landroid/app/DownloadManager;->validateArgumentIsNonEmpty(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 1129
!     const-string/jumbo v3, "mimeType"
  
      invoke-static {v3, p4}, Landroid/app/DownloadManager;->validateArgumentIsNonEmpty(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1130
      const-wide/16 v5, 0x0
  
      cmp-long v3, p6, v5
  
      if-gtz v3, :cond_0
  
+     .line 1131
      new-instance v3, Ljava/lang/IllegalArgumentException;
  
      const-string v4, " invalid value for param: totalBytes"
***************
*** 466,475 ****
  
      throw v3
  
      :cond_0
      new-instance v3, Landroid/app/DownloadManager$Request;
  
!     const-string v5, "non-dwnldmngr-download-dont-retry2download"
  
      invoke-direct {v3, v5}, Landroid/app/DownloadManager$Request;-><init>(Ljava/lang/String;)V
  
--- 504,514 ----
  
      throw v3
  
+     .line 1135
      :cond_0
      new-instance v3, Landroid/app/DownloadManager$Request;
  
!     const-string/jumbo v5, "non-dwnldmngr-download-dont-retry2download"
  
      invoke-direct {v3, v5}, Landroid/app/DownloadManager$Request;-><init>(Ljava/lang/String;)V
  
***************
*** 485,490 ****
--- 524,530 ----
  
      move-result-object v1
  
+     .line 1139
      .local v1, request:Landroid/app/DownloadManager$Request;
      const/4 v3, 0x0
  
***************
*** 492,497 ****
--- 532,538 ----
  
      move-result-object v2
  
+     .line 1140
      .local v2, values:Landroid/content/ContentValues;
      const-string v3, "destination"
  
***************
*** 503,513 ****
  
      invoke-virtual {v2, v3, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
      const-string v3, "_data"
  
      invoke-virtual {v2, v3, p5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v3, "status"
  
      const/16 v5, 0xc8
  
--- 544,556 ----
  
      invoke-virtual {v2, v3, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 1142
      const-string v3, "_data"
  
      invoke-virtual {v2, v3, p5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 1143
!     const-string/jumbo v3, "status"
  
      const/16 v5, 0xc8
  
***************
*** 517,523 ****
  
      invoke-virtual {v2, v3, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     const-string v3, "total_bytes"
  
      invoke-static {p6, p7}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
  
--- 560,567 ----
  
      invoke-virtual {v2, v3, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     .line 1144
!     const-string/jumbo v3, "total_bytes"
  
      invoke-static {p6, p7}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
  
***************
*** 525,531 ****
  
      invoke-virtual {v2, v3, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
!     const-string v5, "scanned"
  
      if-eqz p3, :cond_2
  
--- 569,576 ----
  
      invoke-virtual {v2, v3, v5}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
!     .line 1145
!     const-string/jumbo v5, "scanned"
  
      if-eqz p3, :cond_2
  
***************
*** 538,544 ****
  
      invoke-virtual {v2, v5, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     const-string v3, "visibility"
  
      if-eqz p8, :cond_1
  
--- 583,590 ----
  
      invoke-virtual {v2, v5, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     .line 1148
!     const-string/jumbo v3, "visibility"
  
      if-eqz p8, :cond_1
  
***************
*** 551,556 ****
--- 597,603 ----
  
      invoke-virtual {v2, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 1150
      iget-object v3, p0, Landroid/app/DownloadManager;->mResolver:Landroid/content/ContentResolver;
  
      sget-object v4, Landroid/provider/Downloads$Impl;->CONTENT_URI:Landroid/net/Uri;
***************
*** 559,569 ****
--- 606,619 ----
  
      move-result-object v0
  
+     .line 1151
      .local v0, downloadUri:Landroid/net/Uri;
      if-nez v0, :cond_3
  
+     .line 1152
      const-wide/16 v3, -0x1
  
+     .line 1154
      :goto_1
      return-wide v3
  
***************
*** 571,578 ****
--- 621,630 ----
      :cond_2
      move v3, v4
  
+     .line 1145
      goto :goto_0
  
+     .line 1154
      .restart local v0       #downloadUri:Landroid/net/Uri;
      :cond_3
      invoke-virtual {v0}, Landroid/net/Uri;->getLastPathSegment()Ljava/lang/String;
***************
*** 591,602 ****
--- 643,656 ----
      .parameter "request"
  
      .prologue
+     .line 884
      iget-object v4, p0, Landroid/app/DownloadManager;->mPackageName:Ljava/lang/String;
  
      invoke-virtual {p1, v4}, Landroid/app/DownloadManager$Request;->toContentValues(Ljava/lang/String;)Landroid/content/ContentValues;
  
      move-result-object v3
  
+     .line 885
      .local v3, values:Landroid/content/ContentValues;
      iget-object v4, p0, Landroid/app/DownloadManager;->mResolver:Landroid/content/ContentResolver;
  
***************
*** 606,611 ****
--- 660,666 ----
  
      move-result-object v0
  
+     .line 886
      .local v0, downloadUri:Landroid/net/Uri;
      invoke-virtual {v0}, Landroid/net/Uri;->getLastPathSegment()Ljava/lang/String;
  
***************
*** 615,620 ****
--- 670,676 ----
  
      move-result-wide v1
  
+     .line 887
      .local v1, id:J
      return-wide v1
  .end method
***************
*** 624,629 ****
--- 680,686 ----
      .parameter "id"
  
      .prologue
+     .line 1169
      iget-object v0, p0, Landroid/app/DownloadManager;->mBaseUri:Landroid/net/Uri;
  
      invoke-static {v0, p1, p2}, Landroid/content/ContentUris;->withAppendedId(Landroid/net/Uri;J)Landroid/net/Uri;
***************
*** 640,645 ****
--- 697,703 ----
      .prologue
      const/4 v2, 0x0
  
+     .line 1018
      new-instance v3, Landroid/app/DownloadManager$Query;
  
      invoke-direct {v3}, Landroid/app/DownloadManager$Query;-><init>()V
***************
*** 656,664 ****
--- 714,724 ----
  
      move-result-object v1
  
+     .line 1019
      .local v1, query:Landroid/app/DownloadManager$Query;
      const/4 v0, 0x0
  
+     .line 1021
      .local v0, cursor:Landroid/database/Cursor;
      :try_start_0
      invoke-virtual {p0, v1}, Landroid/app/DownloadManager;->query(Landroid/app/DownloadManager$Query;)Landroid/database/Cursor;
***************
*** 667,682 ****
--- 727,747 ----
  
      move-result-object v0
  
+     .line 1022
      if-nez v0, :cond_1
  
+     .line 1029
      if-eqz v0, :cond_0
  
+     .line 1030
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
+     .line 1034
      :cond_0
      :goto_0
      return-object v2
  
+     .line 1025
      :cond_1
      :try_start_1
      invoke-interface {v0}, Landroid/database/Cursor;->moveToFirst()Z
***************
*** 685,691 ****
  
      if-eqz v3, :cond_2
  
!     const-string v2, "media_type"
  
      invoke-interface {v0, v2}, Landroid/database/Cursor;->getColumnIndexOrThrow(Ljava/lang/String;)I
  
--- 750,757 ----
  
      if-eqz v3, :cond_2
  
!     .line 1026
!     const-string/jumbo v2, "media_type"
  
      invoke-interface {v0, v2}, Landroid/database/Cursor;->getColumnIndexOrThrow(Ljava/lang/String;)I
  
***************
*** 697,720 ****
--- 763,792 ----
  
      move-result-object v2
  
+     .line 1029
      if-eqz v0, :cond_0
  
+     .line 1030
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 1029
      :cond_2
      if-eqz v0, :cond_0
  
+     .line 1030
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 1029
      :catchall_0
      move-exception v2
  
      if-eqz v0, :cond_3
  
+     .line 1030
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      :cond_3
***************
*** 730,735 ****
--- 802,808 ----
  
      const/4 v10, 0x1
  
+     .line 966
      new-instance v7, Landroid/app/DownloadManager$Query;
  
      invoke-direct {v7}, Landroid/app/DownloadManager$Query;-><init>()V
***************
*** 744,752 ****
--- 817,827 ----
  
      move-result-object v4
  
+     .line 967
      .local v4, query:Landroid/app/DownloadManager$Query;
      const/4 v0, 0x0
  
+     .line 969
      .local v0, cursor:Landroid/database/Cursor;
      :try_start_0
      invoke-virtual {p0, v4}, Landroid/app/DownloadManager;->query(Landroid/app/DownloadManager$Query;)Landroid/database/Cursor;
***************
*** 755,770 ****
--- 830,850 ----
  
      move-result-object v0
  
+     .line 970
      if-nez v0, :cond_1
  
+     .line 997
      if-eqz v0, :cond_0
  
+     .line 998
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
+     .line 1002
      :cond_0
      :goto_0
      return-object v6
  
+     .line 973
      :cond_1
      :try_start_1
      invoke-interface {v0}, Landroid/database/Cursor;->moveToFirst()Z
***************
*** 773,779 ****
  
      if-eqz v7, :cond_4
  
!     const-string v7, "status"
  
      invoke-interface {v0, v7}, Landroid/database/Cursor;->getColumnIndexOrThrow(Ljava/lang/String;)I
  
--- 853,860 ----
  
      if-eqz v7, :cond_4
  
!     .line 974
!     const-string/jumbo v7, "status"
  
      invoke-interface {v0, v7}, Landroid/database/Cursor;->getColumnIndexOrThrow(Ljava/lang/String;)I
  
***************
*** 783,804 ****
--- 864,889 ----
  
      move-result v5
  
+     .line 975
      .local v5, status:I
      const/16 v7, 0x8
  
      if-ne v7, v5, :cond_4
  
+     .line 976
      const-string v6, "destination"
  
      invoke-interface {v0, v6}, Landroid/database/Cursor;->getColumnIndexOrThrow(Ljava/lang/String;)I
  
      move-result v2
  
+     .line 978
      .local v2, indx:I
      invoke-interface {v0, v2}, Landroid/database/Cursor;->getInt(I)I
  
      move-result v1
  
+     .line 982
      .local v1, destination:I
      if-eq v1, v10, :cond_2
  
***************
*** 814,819 ****
--- 899,905 ----
  
      if-ne v1, v6, :cond_3
  
+     .line 987
      :cond_2
      sget-object v6, Landroid/provider/Downloads$Impl;->CONTENT_URI:Landroid/net/Uri;
  
***************
*** 823,834 ****
--- 909,923 ----
  
      move-result-object v6
  
+     .line 997
      if-eqz v0, :cond_0
  
+     .line 998
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 990
      :cond_3
      :try_start_2
      const-string v6, "local_filename"
***************
*** 841,846 ****
--- 930,936 ----
  
      move-result-object v3
  
+     .line 992
      .local v3, path:Ljava/lang/String;
      new-instance v6, Ljava/io/File;
  
***************
*** 852,863 ****
--- 942,956 ----
  
      move-result-object v6
  
+     .line 997
      if-eqz v0, :cond_0
  
+     .line 998
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 997
      .end local v1           #destination:I
      .end local v2           #indx:I
      .end local v3           #path:Ljava/lang/String;
***************
*** 865,879 ****
--- 958,975 ----
      :cond_4
      if-eqz v0, :cond_0
  
+     .line 998
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 997
      :catchall_0
      move-exception v6
  
      if-eqz v0, :cond_5
  
+     .line 998
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
      :cond_5
***************
*** 889,900 ****
--- 985,998 ----
  
      const/4 v3, 0x1
  
+     .line 900
      if-eqz p1, :cond_0
  
      array-length v1, p1
  
      if-nez v1, :cond_1
  
+     .line 902
      :cond_0
      new-instance v1, Ljava/lang/IllegalArgumentException;
  
***************
*** 904,914 ****
--- 1002,1014 ----
  
      throw v1
  
+     .line 904
      :cond_1
      new-instance v0, Landroid/content/ContentValues;
  
      invoke-direct {v0}, Landroid/content/ContentValues;-><init>()V
  
+     .line 905
      .local v0, values:Landroid/content/ContentValues;
      const-string v1, "deleted"
  
***************
*** 918,927 ****
--- 1018,1029 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 908
      array-length v1, p1
  
      if-ne v1, v3, :cond_2
  
+     .line 909
      iget-object v1, p0, Landroid/app/DownloadManager;->mResolver:Landroid/content/ContentResolver;
  
      iget-object v2, p0, Landroid/app/DownloadManager;->mBaseUri:Landroid/net/Uri;
***************
*** 938,943 ****
--- 1040,1046 ----
  
      move-result v1
  
+     .line 912
      :goto_0
      return v1
  
***************
*** 971,983 ****
      .end annotation
  
      .prologue
      iget-object v0, p0, Landroid/app/DownloadManager;->mResolver:Landroid/content/ContentResolver;
  
      invoke-virtual {p0, p1, p2}, Landroid/app/DownloadManager;->getDownloadUri(J)Landroid/net/Uri;
  
      move-result-object v1
  
!     const-string v2, "r"
  
      invoke-virtual {v0, v1, v2}, Landroid/content/ContentResolver;->openFileDescriptor(Landroid/net/Uri;Ljava/lang/String;)Landroid/os/ParcelFileDescriptor;
  
--- 1074,1087 ----
      .end annotation
  
      .prologue
+     .line 949
      iget-object v0, p0, Landroid/app/DownloadManager;->mResolver:Landroid/content/ContentResolver;
  
      invoke-virtual {p0, p1, p2}, Landroid/app/DownloadManager;->getDownloadUri(J)Landroid/net/Uri;
  
      move-result-object v1
  
!     const-string/jumbo v2, "r"
  
      invoke-virtual {v0, v1, v2}, Landroid/content/ContentResolver;->openFileDescriptor(Landroid/net/Uri;Ljava/lang/String;)Landroid/os/ParcelFileDescriptor;
  
***************
*** 991,996 ****
--- 1095,1101 ----
      .parameter "query"
  
      .prologue
+     .line 935
      iget-object v1, p0, Landroid/app/DownloadManager;->mResolver:Landroid/content/ContentResolver;
  
      sget-object v2, Landroid/app/DownloadManager;->UNDERLYING_COLUMNS:[Ljava/lang/String;
***************
*** 1001,1011 ****
--- 1106,1119 ----
  
      move-result-object v0
  
+     .line 936
      .local v0, underlyingCursor:Landroid/database/Cursor;
      if-nez v0, :cond_0
  
+     .line 937
      const/4 v1, 0x0
  
+     .line 939
      :goto_0
      return-object v1
  
***************
*** 1024,1029 ****
--- 1132,1138 ----
      .parameter "ids"
  
      .prologue
+     .line 925
      invoke-virtual {p0, p1}, Landroid/app/DownloadManager;->markRowDeleted([J)I
  
      move-result v0
***************
*** 1036,1041 ****
--- 1145,1151 ----
      .parameter "ids"
  
      .prologue
+     .line 1044
      new-instance v3, Landroid/app/DownloadManager$Query;
  
      invoke-direct {v3}, Landroid/app/DownloadManager$Query;-><init>()V
***************
*** 1048,1053 ****
--- 1158,1164 ----
  
      move-result-object v0
  
+     .line 1046
      .local v0, cursor:Landroid/database/Cursor;
      :try_start_0
      invoke-interface {v0}, Landroid/database/Cursor;->moveToFirst()Z
***************
*** 1059,1065 ****
  
      if-nez v3, :cond_1
  
!     const-string v3, "status"
  
      invoke-interface {v0, v3}, Landroid/database/Cursor;->getColumnIndex(Ljava/lang/String;)I
  
--- 1170,1177 ----
  
      if-nez v3, :cond_1
  
!     .line 1047
!     const-string/jumbo v3, "status"
  
      invoke-interface {v0, v3}, Landroid/database/Cursor;->getColumnIndex(Ljava/lang/String;)I
  
***************
*** 1069,1074 ****
--- 1181,1187 ----
  
      move-result v1
  
+     .line 1048
      .local v1, status:I
      const/16 v3, 0x8
  
***************
*** 1078,1083 ****
--- 1191,1197 ----
  
      if-eq v1, v3, :cond_0
  
+     .line 1049
      new-instance v3, Ljava/lang/IllegalArgumentException;
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 1114,1119 ****
--- 1228,1234 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 1054
      .end local v1           #status:I
      :catchall_0
      move-exception v3
***************
*** 1122,1127 ****
--- 1237,1243 ----
  
      throw v3
  
+     .line 1046
      .restart local v1       #status:I
      :cond_0
      :try_start_1
***************
*** 1131,1144 ****
--- 1247,1263 ----
  
      goto :goto_0
  
+     .line 1054
      .end local v1           #status:I
      :cond_1
      invoke-interface {v0}, Landroid/database/Cursor;->close()V
  
+     .line 1057
      new-instance v2, Landroid/content/ContentValues;
  
      invoke-direct {v2}, Landroid/content/ContentValues;-><init>()V
  
+     .line 1058
      .local v2, values:Landroid/content/ContentValues;
      const-string v3, "current_bytes"
  
***************
*** 1150,1156 ****
  
      invoke-virtual {v2, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     const-string v3, "total_bytes"
  
      const/4 v4, -0x1
  
--- 1269,1276 ----
  
      invoke-virtual {v2, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     .line 1059
!     const-string/jumbo v3, "total_bytes"
  
      const/4 v4, -0x1
  
***************
*** 1160,1170 ****
  
      invoke-virtual {v2, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
      const-string v3, "_data"
  
      invoke-virtual {v2, v3}, Landroid/content/ContentValues;->putNull(Ljava/lang/String;)V
  
!     const-string v3, "status"
  
      const/16 v4, 0xbe
  
--- 1280,1292 ----
  
      invoke-virtual {v2, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 1060
      const-string v3, "_data"
  
      invoke-virtual {v2, v3}, Landroid/content/ContentValues;->putNull(Ljava/lang/String;)V
  
!     .line 1061
!     const-string/jumbo v3, "status"
  
      const/16 v4, 0xbe
  
***************
*** 1174,1179 ****
--- 1296,1302 ----
  
      invoke-virtual {v2, v3, v4}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 1062
      iget-object v3, p0, Landroid/app/DownloadManager;->mResolver:Landroid/content/ContentResolver;
  
      iget-object v4, p0, Landroid/app/DownloadManager;->mBaseUri:Landroid/net/Uri;
***************
*** 1188,1193 ****
--- 1311,1317 ----
  
      invoke-virtual {v3, v4, v2, v5, v6}, Landroid/content/ContentResolver;->update(Landroid/net/Uri;Landroid/content/ContentValues;Ljava/lang/String;[Ljava/lang/String;)I
  
+     .line 1063
      return-void
  .end method
  
***************
*** 1196,1210 ****
--- 1320,1338 ----
      .parameter "accessAllDownloads"
  
      .prologue
+     .line 868
      if-eqz p1, :cond_0
  
+     .line 869
      sget-object v0, Landroid/provider/Downloads$Impl;->ALL_DOWNLOADS_CONTENT_URI:Landroid/net/Uri;
  
      iput-object v0, p0, Landroid/app/DownloadManager;->mBaseUri:Landroid/net/Uri;
  
+     .line 873
      :goto_0
      return-void
  
+     .line 871
      :cond_0
      sget-object v0, Landroid/provider/Downloads$Impl;->CONTENT_URI:Landroid/net/Uri;
  
