*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ExpandableListActivity.smali	2012-03-23 19:36:42.840462178 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/ExpandableListActivity.smali.original	2012-03-23 19:34:45.314319093 -0500
***************
*** 22,29 ****
--- 22,31 ----
      .locals 1
  
      .prologue
+     .line 155
      invoke-direct {p0}, Landroid/app/Activity;-><init>()V
  
+     .line 161
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/app/ExpandableListActivity;->mFinishedStart:Z
***************
*** 35,47 ****
--- 37,52 ----
      .locals 1
  
      .prologue
+     .line 270
      iget-object v0, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      if-eqz v0, :cond_0
  
+     .line 274
      :goto_0
      return-void
  
+     .line 273
      :cond_0
      const v0, 0x1090001
  
***************
*** 56,61 ****
--- 61,67 ----
      .locals 1
  
      .prologue
+     .line 266
      iget-object v0, p0, Landroid/app/ExpandableListActivity;->mAdapter:Landroid/widget/ExpandableListAdapter;
  
      return-object v0
***************
*** 65,72 ****
--- 71,80 ----
      .locals 1
  
      .prologue
+     .line 257
      invoke-direct {p0}, Landroid/app/ExpandableListActivity;->ensureList()V
  
+     .line 258
      iget-object v0, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      return-object v0
***************
*** 76,81 ****
--- 84,90 ----
      .locals 2
  
      .prologue
+     .line 282
      iget-object v0, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      invoke-virtual {v0}, Landroid/widget/ExpandableListView;->getSelectedId()J
***************
*** 89,94 ****
--- 98,104 ----
      .locals 2
  
      .prologue
+     .line 297
      iget-object v0, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      invoke-virtual {v0}, Landroid/widget/ExpandableListView;->getSelectedPosition()J
***************
*** 107,112 ****
--- 117,123 ----
      .parameter "id"
  
      .prologue
+     .line 183
      const/4 v0, 0x0
  
      return v0
***************
*** 116,129 ****
--- 127,143 ----
      .locals 3
  
      .prologue
+     .line 218
      invoke-super {p0}, Landroid/app/Activity;->onContentChanged()V
  
+     .line 219
      const v1, 0x1020004
  
      invoke-virtual {p0, v1}, Landroid/app/ExpandableListActivity;->findViewById(I)Landroid/view/View;
  
      move-result-object v0
  
+     .line 220
      .local v0, emptyView:Landroid/view/View;
      const v1, 0x102000a
  
***************
*** 135,144 ****
--- 149,160 ----
  
      iput-object v1, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
+     .line 221
      iget-object v1, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      if-nez v1, :cond_0
  
+     .line 222
      new-instance v1, Ljava/lang/RuntimeException;
  
      const-string v2, "Your content must have a ExpandableListView whose id attribute is \'android.R.id.list\'"
***************
*** 147,185 ****
--- 163,210 ----
  
      throw v1
  
+     .line 226
      :cond_0
      if-eqz v0, :cond_1
  
+     .line 227
      iget-object v1, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      invoke-virtual {v1, v0}, Landroid/widget/ExpandableListView;->setEmptyView(Landroid/view/View;)V
  
+     .line 229
      :cond_1
      iget-object v1, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      invoke-virtual {v1, p0}, Landroid/widget/ExpandableListView;->setOnChildClickListener(Landroid/widget/ExpandableListView$OnChildClickListener;)V
  
+     .line 230
      iget-object v1, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      invoke-virtual {v1, p0}, Landroid/widget/ExpandableListView;->setOnGroupExpandListener(Landroid/widget/ExpandableListView$OnGroupExpandListener;)V
  
+     .line 231
      iget-object v1, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      invoke-virtual {v1, p0}, Landroid/widget/ExpandableListView;->setOnGroupCollapseListener(Landroid/widget/ExpandableListView$OnGroupCollapseListener;)V
  
+     .line 233
      iget-boolean v1, p0, Landroid/app/ExpandableListActivity;->mFinishedStart:Z
  
      if-eqz v1, :cond_2
  
+     .line 234
      iget-object v1, p0, Landroid/app/ExpandableListActivity;->mAdapter:Landroid/widget/ExpandableListAdapter;
  
      invoke-virtual {p0, v1}, Landroid/app/ExpandableListActivity;->setListAdapter(Landroid/widget/ExpandableListAdapter;)V
  
+     .line 236
      :cond_2
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/app/ExpandableListActivity;->mFinishedStart:Z
  
+     .line 237
      return-void
  .end method
  
***************
*** 190,195 ****
--- 215,221 ----
      .parameter "menuInfo"
  
      .prologue
+     .line 174
      return-void
  .end method
  
***************
*** 198,203 ****
--- 224,230 ----
      .parameter "groupPosition"
  
      .prologue
+     .line 190
      return-void
  .end method
  
***************
*** 206,211 ****
--- 233,239 ----
      .parameter "groupPosition"
  
      .prologue
+     .line 196
      return-void
  .end method
  
***************
*** 214,223 ****
--- 242,254 ----
      .parameter "state"
  
      .prologue
+     .line 206
      invoke-direct {p0}, Landroid/app/ExpandableListActivity;->ensureList()V
  
+     .line 207
      invoke-super {p0, p1}, Landroid/app/Activity;->onRestoreInstanceState(Landroid/os/Bundle;)V
  
+     .line 208
      return-void
  .end method
  
***************
*** 226,246 ****
--- 257,284 ----
      .parameter "adapter"
  
      .prologue
+     .line 243
      monitor-enter p0
  
+     .line 244
      :try_start_0
      invoke-direct {p0}, Landroid/app/ExpandableListActivity;->ensureList()V
  
+     .line 245
      iput-object p1, p0, Landroid/app/ExpandableListActivity;->mAdapter:Landroid/widget/ExpandableListAdapter;
  
+     .line 246
      iget-object v0, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      invoke-virtual {v0, p1}, Landroid/widget/ExpandableListView;->setAdapter(Landroid/widget/ExpandableListAdapter;)V
  
+     .line 247
      monitor-exit p0
  
+     .line 248
      return-void
  
+     .line 247
      :catchall_0
      move-exception v0
  
***************
*** 258,263 ****
--- 296,302 ----
      .parameter "shouldExpandGroup"
  
      .prologue
+     .line 312
      iget-object v0, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      invoke-virtual {v0, p1, p2, p3}, Landroid/widget/ExpandableListView;->setSelectedChild(IIZ)Z
***************
*** 272,280 ****
--- 311,321 ----
      .parameter "groupPosition"
  
      .prologue
+     .line 320
      iget-object v0, p0, Landroid/app/ExpandableListActivity;->mList:Landroid/widget/ExpandableListView;
  
      invoke-virtual {v0, p1}, Landroid/widget/ExpandableListView;->setSelectedGroup(I)V
  
+     .line 321
      return-void
  .end method
