*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/INotificationManager$Stub$Proxy.smali	2012-03-23 19:36:41.359435172 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/INotificationManager$Stub$Proxy.smali.original	2012-03-23 19:34:45.303318892 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 179
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 180
      iput-object p1, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 181
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 184
      iget-object v0, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 55,77 ****
--- 59,86 ----
      .end annotation
  
      .prologue
+     .line 235
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 236
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 238
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.INotificationManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 239
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 240
      iget-object v2, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x3
***************
*** 80,100 ****
--- 89,115 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 241
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 244
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 245
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 247
      return-void
  
+     .line 244
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 245
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 111,135 ****
--- 126,156 ----
      .end annotation
  
      .prologue
+     .line 219
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 220
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 222
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.INotificationManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 223
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 224
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 225
      iget-object v2, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x2
***************
*** 138,158 ****
--- 159,185 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 226
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 229
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 230
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 232
      return-void
  
+     .line 229
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 230
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 170,196 ****
--- 197,230 ----
      .end annotation
  
      .prologue
+     .line 336
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 337
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 339
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.INotificationManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 340
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 341
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 342
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 343
      iget-object v2, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/16 v3, 0x8
***************
*** 199,219 ****
--- 233,259 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 344
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 347
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 348
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 350
      return-void
  
+     .line 347
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 348
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 230,252 ****
--- 270,297 ----
      .end annotation
  
      .prologue
+     .line 267
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 268
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 270
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.INotificationManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 271
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 272
      if-eqz p2, :cond_0
  
      invoke-interface {p2}, Landroid/app/ITransientNotification;->asBinder()Landroid/os/IBinder;
***************
*** 256,261 ****
--- 301,307 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 273
      iget-object v2, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x5
***************
*** 264,289 ****
--- 310,342 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 274
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 277
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 278
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 280
      return-void
  
+     .line 272
      :cond_0
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 277
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 278
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 302,339 ****
--- 355,402 ----
      .end annotation
  
      .prologue
+     .line 193
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 194
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 196
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.INotificationManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 197
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 198
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 199
      if-eqz p3, :cond_0
  
+     .line 200
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 201
      const/4 v2, 0x0
  
      invoke-virtual {p3, v0, v2}, Landroid/app/Notification;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 206
      :goto_0
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeIntArray([I)V
  
+     .line 207
      iget-object v2, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x1
***************
*** 342,359 ****
--- 405,428 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 208
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 209
      invoke-virtual {v1, p4}, Landroid/os/Parcel;->readIntArray([I)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 212
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 213
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 215
      return-void
  
+     .line 204
      :cond_0
      const/4 v2, 0x0
  
***************
*** 364,374 ****
--- 433,445 ----
  
      goto :goto_0
  
+     .line 212
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 213
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 388,427 ****
--- 459,509 ----
      .end annotation
  
      .prologue
+     .line 283
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 284
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 286
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.INotificationManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 287
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 288
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 289
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 290
      if-eqz p4, :cond_0
  
+     .line 291
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 292
      const/4 v2, 0x0
  
      invoke-virtual {p4, v0, v2}, Landroid/app/Notification;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 297
      :goto_0
      invoke-virtual {v0, p5}, Landroid/os/Parcel;->writeIntArray([I)V
  
+     .line 298
      iget-object v2, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x6
***************
*** 430,447 ****
--- 512,535 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 299
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 300
      invoke-virtual {v1, p5}, Landroid/os/Parcel;->readIntArray([I)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 303
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 304
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 306
      return-void
  
+     .line 295
      :cond_0
      const/4 v2, 0x0
  
***************
*** 452,462 ****
--- 540,552 ----
  
      goto :goto_0
  
+     .line 303
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 304
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 477,518 ****
--- 567,620 ----
      .end annotation
  
      .prologue
+     .line 309
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 310
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 312
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.INotificationManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 313
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 314
      invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 315
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 316
      invoke-virtual {v0, p4}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 317
      if-eqz p5, :cond_0
  
+     .line 318
      const/4 v2, 0x1
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 319
      const/4 v2, 0x0
  
      invoke-virtual {p5, v0, v2}, Landroid/app/Notification;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 324
      :goto_0
      invoke-virtual {v0, p6}, Landroid/os/Parcel;->writeIntArray([I)V
  
+     .line 325
      iget-object v2, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x7
***************
*** 521,538 ****
--- 623,646 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 326
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
  
+     .line 327
      invoke-virtual {v1, p6}, Landroid/os/Parcel;->readIntArray([I)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 330
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 331
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 333
      return-void
  
+     .line 322
      :cond_0
      const/4 v2, 0x0
  
***************
*** 543,553 ****
--- 651,663 ----
  
      goto :goto_0
  
+     .line 330
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 331
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 565,587 ****
--- 675,702 ----
      .end annotation
  
      .prologue
+     .line 250
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 251
      .local v0, _data:Landroid/os/Parcel;
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v1
  
+     .line 253
      .local v1, _reply:Landroid/os/Parcel;
      :try_start_0
      const-string v2, "android.app.INotificationManager"
  
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 254
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 255
      if-eqz p2, :cond_0
  
      invoke-interface {p2}, Landroid/app/ITransientNotification;->asBinder()Landroid/os/IBinder;
***************
*** 591,598 ****
--- 706,715 ----
      :goto_0
      invoke-virtual {v0, v2}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
  
+     .line 256
      invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 257
      iget-object v2, p0, Landroid/app/INotificationManager$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v3, 0x4
***************
*** 601,626 ****
--- 718,750 ----
  
      invoke-interface {v2, v3, v0, v1, v4}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
  
+     .line 258
      invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 261
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 262
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 264
      return-void
  
+     .line 255
      :cond_0
      const/4 v2, 0x0
  
      goto :goto_0
  
+     .line 261
      :catchall_0
      move-exception v2
  
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
+     .line 262
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
      throw v2
***************
*** 630,635 ****
--- 754,760 ----
      .locals 1
  
      .prologue
+     .line 188
      const-string v0, "android.app.INotificationManager"
  
      return-object v0
