*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Instrumentation$1MenuRunnable.smali	2012-03-23 19:36:39.355398630 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Instrumentation$1MenuRunnable.smali.original	2012-03-23 19:34:45.286318582 -0500
***************
*** 38,53 ****
--- 38,58 ----
      .parameter "_flags"
  
      .prologue
+     .line 743
      iput-object p1, p0, Landroid/app/Instrumentation$1MenuRunnable;->this$0:Landroid/app/Instrumentation;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 744
      iput-object p2, p0, Landroid/app/Instrumentation$1MenuRunnable;->activity:Landroid/app/Activity;
  
+     .line 745
      iput p3, p0, Landroid/app/Instrumentation$1MenuRunnable;->identifier:I
  
+     .line 746
      iput p4, p0, Landroid/app/Instrumentation$1MenuRunnable;->flags:I
  
+     .line 747
      return-void
  .end method
  
***************
*** 57,68 ****
--- 62,75 ----
      .locals 4
  
      .prologue
+     .line 750
      iget-object v1, p0, Landroid/app/Instrumentation$1MenuRunnable;->activity:Landroid/app/Activity;
  
      invoke-virtual {v1}, Landroid/app/Activity;->getWindow()Landroid/view/Window;
  
      move-result-object v0
  
+     .line 752
      .local v0, win:Landroid/view/Window;
      const/4 v1, 0x0
  
***************
*** 76,80 ****
--- 83,88 ----
  
      iput-boolean v1, p0, Landroid/app/Instrumentation$1MenuRunnable;->returnValue:Z
  
+     .line 756
      return-void
  .end method
