*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Instrumentation$1ContextMenuRunnable.smali	2012-03-23 19:36:42.314452588 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Instrumentation$1ContextMenuRunnable.smali.original	2012-03-23 19:34:45.309319001 -0500
***************
*** 38,53 ****
--- 38,58 ----
      .parameter "_flags"
  
      .prologue
+     .line 806
      iput-object p1, p0, Landroid/app/Instrumentation$1ContextMenuRunnable;->this$0:Landroid/app/Instrumentation;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 807
      iput-object p2, p0, Landroid/app/Instrumentation$1ContextMenuRunnable;->activity:Landroid/app/Activity;
  
+     .line 808
      iput p3, p0, Landroid/app/Instrumentation$1ContextMenuRunnable;->identifier:I
  
+     .line 809
      iput p4, p0, Landroid/app/Instrumentation$1ContextMenuRunnable;->flags:I
  
+     .line 810
      return-void
  .end method
  
***************
*** 57,68 ****
--- 62,75 ----
      .locals 3
  
      .prologue
+     .line 813
      iget-object v1, p0, Landroid/app/Instrumentation$1ContextMenuRunnable;->activity:Landroid/app/Activity;
  
      invoke-virtual {v1}, Landroid/app/Activity;->getWindow()Landroid/view/Window;
  
      move-result-object v0
  
+     .line 814
      .local v0, win:Landroid/view/Window;
      iget v1, p0, Landroid/app/Instrumentation$1ContextMenuRunnable;->identifier:I
  
***************
*** 74,78 ****
--- 81,86 ----
  
      iput-boolean v1, p0, Landroid/app/Instrumentation$1ContextMenuRunnable;->returnValue:Z
  
+     .line 817
      return-void
  .end method
