*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Instrumentation$ActivityWaiter.smali	2012-03-23 19:36:38.377380794 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Instrumentation$ActivityWaiter.smali.original	2012-03-23 19:34:45.276318399 -0500
***************
*** 26,34 ****
--- 26,37 ----
      .parameter "_intent"
  
      .prologue
+     .line 1592
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1593
      iput-object p1, p0, Landroid/app/Instrumentation$ActivityWaiter;->intent:Landroid/content/Intent;
  
+     .line 1594
      return-void
  .end method
