*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/IThumbnailReceiver$Stub$Proxy.smali	2012-03-23 19:36:43.962482638 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/IThumbnailReceiver$Stub$Proxy.smali.original	2012-03-23 19:34:45.322319239 -0500
***************
*** 27,36 ****
--- 27,39 ----
      .parameter "remote"
  
      .prologue
+     .line 85
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 86
      iput-object p1, p0, Landroid/app/IThumbnailReceiver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
+     .line 87
      return-void
  .end method
  
***************
*** 40,45 ****
--- 43,49 ----
      .locals 1
  
      .prologue
+     .line 90
      iget-object v0, p0, Landroid/app/IThumbnailReceiver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      return-object v0
***************
*** 54,69 ****
--- 58,76 ----
      .end annotation
  
      .prologue
+     .line 124
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 126
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.IThumbnailReceiver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 127
      iget-object v1, p0, Landroid/app/IThumbnailReceiver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
      const/4 v2, 0x2
***************
*** 76,85 ****
--- 83,95 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 130
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 132
      return-void
  
+     .line 130
      :catchall_0
      move-exception v1
  
***************
*** 92,97 ****
--- 102,108 ----
      .locals 1
  
      .prologue
+     .line 94
      const-string v0, "android.app.IThumbnailReceiver"
  
      return-object v0
***************
*** 109,147 ****
--- 120,168 ----
      .end annotation
  
      .prologue
+     .line 98
      invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
  
      move-result-object v0
  
+     .line 100
      .local v0, _data:Landroid/os/Parcel;
      :try_start_0
      const-string v1, "android.app.IThumbnailReceiver"
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
  
+     .line 101
      invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 102
      if-eqz p2, :cond_0
  
+     .line 103
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 104
      const/4 v1, 0x0
  
      invoke-virtual {p2, v0, v1}, Landroid/graphics/Bitmap;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 109
      :goto_0
      if-eqz p3, :cond_1
  
+     .line 110
      const/4 v1, 0x1
  
      invoke-virtual {v0, v1}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 111
      const/4 v1, 0x0
  
      invoke-static {p3, v0, v1}, Landroid/text/TextUtils;->writeToParcel(Ljava/lang/CharSequence;Landroid/os/Parcel;I)V
  
+     .line 116
      :goto_1
      iget-object v1, p0, Landroid/app/IThumbnailReceiver$Stub$Proxy;->mRemote:Landroid/os/IBinder;
  
***************
*** 155,164 ****
--- 176,188 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 119
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
  
+     .line 121
      return-void
  
+     .line 107
      :cond_0
      const/4 v1, 0x0
  
***************
*** 169,174 ****
--- 193,199 ----
  
      goto :goto_0
  
+     .line 119
      :catchall_0
      move-exception v1
  
***************
*** 176,181 ****
--- 201,207 ----
  
      throw v1
  
+     .line 114
      :cond_1
      const/4 v1, 0x0
  
