*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/KeyguardManager$1.smali	2012-03-23 19:36:35.553329298 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/KeyguardManager$1.smali.original	2012-03-23 19:34:45.236317670 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 206
      iput-object p1, p0, Landroid/app/KeyguardManager$1;->this$0:Landroid/app/KeyguardManager;
  
      iput-object p2, p0, Landroid/app/KeyguardManager$1;->val$callback:Landroid/app/KeyguardManager$OnKeyguardExitResult;
***************
*** 48,56 ****
--- 49,59 ----
      .end annotation
  
      .prologue
+     .line 208
      iget-object v0, p0, Landroid/app/KeyguardManager$1;->val$callback:Landroid/app/KeyguardManager$OnKeyguardExitResult;
  
      invoke-interface {v0, p1}, Landroid/app/KeyguardManager$OnKeyguardExitResult;->onKeyguardExitResult(Z)V
  
+     .line 209
      return-void
  .end method
