*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/LauncherActivity$1.smali	2012-03-23 19:36:36.646349229 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/LauncherActivity$1.smali.original	2012-03-23 19:34:45.248317889 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 373
      iput-object p1, p0, Landroid/app/LauncherActivity$1;->this$0:Landroid/app/LauncherActivity;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,49 ****
--- 42,52 ----
      .parameter "v"
  
      .prologue
+     .line 375
      iget-object v0, p0, Landroid/app/LauncherActivity$1;->this$0:Landroid/app/LauncherActivity;
  
      invoke-virtual {v0}, Landroid/app/LauncherActivity;->finish()V
  
+     .line 376
      return-void
  .end method
