*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/LauncherActivity$ListItem.smali	2012-03-23 19:36:37.785370000 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/LauncherActivity$ListItem.smali.original	2012-03-23 19:34:45.268318253 -0500
***************
*** 33,40 ****
--- 33,42 ----
      .locals 0
  
      .prologue
+     .line 89
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 90
      return-void
  .end method
  
***************
*** 45,67 ****
--- 47,75 ----
      .parameter "resizer"
  
      .prologue
+     .line 73
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 74
      iput-object p2, p0, Landroid/app/LauncherActivity$ListItem;->resolveInfo:Landroid/content/pm/ResolveInfo;
  
+     .line 75
      invoke-virtual {p2, p1}, Landroid/content/pm/ResolveInfo;->loadLabel(Landroid/content/pm/PackageManager;)Ljava/lang/CharSequence;
  
      move-result-object v1
  
      iput-object v1, p0, Landroid/app/LauncherActivity$ListItem;->label:Ljava/lang/CharSequence;
  
+     .line 76
      iget-object v0, p2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
+     .line 77
      .local v0, ci:Landroid/content/pm/ComponentInfo;
      if-nez v0, :cond_0
  
      iget-object v0, p2, Landroid/content/pm/ResolveInfo;->serviceInfo:Landroid/content/pm/ServiceInfo;
  
+     .line 78
      :cond_0
      iget-object v1, p0, Landroid/app/LauncherActivity$ListItem;->label:Ljava/lang/CharSequence;
  
***************
*** 69,83 ****
--- 77,94 ----
  
      if-eqz v0, :cond_1
  
+     .line 79
      iget-object v1, p2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
  
      iget-object v1, v1, Landroid/content/pm/ActivityInfo;->name:Ljava/lang/String;
  
      iput-object v1, p0, Landroid/app/LauncherActivity$ListItem;->label:Ljava/lang/CharSequence;
  
+     .line 82
      :cond_1
      if-eqz p3, :cond_2
  
+     .line 83
      invoke-virtual {p2, p1}, Landroid/content/pm/ResolveInfo;->loadIcon(Landroid/content/pm/PackageManager;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v1
***************
*** 88,93 ****
--- 99,105 ----
  
      iput-object v1, p0, Landroid/app/LauncherActivity$ListItem;->icon:Landroid/graphics/drawable/Drawable;
  
+     .line 85
      :cond_2
      iget-object v1, v0, Landroid/content/pm/ComponentInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
***************
*** 95,103 ****
--- 107,117 ----
  
      iput-object v1, p0, Landroid/app/LauncherActivity$ListItem;->packageName:Ljava/lang/String;
  
+     .line 86
      iget-object v1, v0, Landroid/content/pm/ComponentInfo;->name:Ljava/lang/String;
  
      iput-object v1, p0, Landroid/app/LauncherActivity$ListItem;->className:Ljava/lang/String;
  
+     .line 87
      return-void
  .end method
