*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/LoaderManagerImpl$LoaderInfo.smali	2012-03-23 19:36:44.551493380 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/LoaderManagerImpl$LoaderInfo.smali.original	2012-03-23 19:34:45.326319311 -0500
***************
*** 96,112 ****
--- 96,117 ----
      .end annotation
  
      .prologue
+     .line 239
      .local p4, callbacks:Landroid/app/LoaderManager$LoaderCallbacks;,"Landroid/app/LoaderManager$LoaderCallbacks<Ljava/lang/Object;>;"
      iput-object p1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 240
      iput p2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mId:I
  
+     .line 241
      iput-object p3, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mArgs:Landroid/os/Bundle;
  
+     .line 242
      iput-object p4, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mCallbacks:Landroid/app/LoaderManager$LoaderCallbacks;
  
+     .line 243
      return-void
  .end method
  
***************
*** 129,141 ****
--- 134,149 ----
      .end annotation
  
      .prologue
+     .line 429
      .local p1, loader:Landroid/content/Loader;,"Landroid/content/Loader<Ljava/lang/Object;>;"
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mCallbacks:Landroid/app/LoaderManager$LoaderCallbacks;
  
      if-eqz v1, :cond_3
  
+     .line 430
      const/4 v0, 0x0
  
+     .line 431
      .local v0, lastBecause:Ljava/lang/String;
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
***************
*** 143,148 ****
--- 151,157 ----
  
      if-eqz v1, :cond_0
  
+     .line 432
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v1, v1, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
***************
*** 151,166 ****
  
      iget-object v0, v1, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v1, v1, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
  
      iget-object v1, v1, Landroid/app/Activity;->mFragments:Landroid/app/FragmentManagerImpl;
  
!     const-string v2, "onLoadFinished"
  
      iput-object v2, v1, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
      :cond_0
      :try_start_0
      sget-boolean v1, Landroid/app/LoaderManagerImpl;->DEBUG:Z
--- 160,177 ----
  
      iget-object v0, v1, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
+     .line 433
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v1, v1, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
  
      iget-object v1, v1, Landroid/app/Activity;->mFragments:Landroid/app/FragmentManagerImpl;
  
!     const-string/jumbo v2, "onLoadFinished"
  
      iput-object v2, v1, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
+     .line 436
      :cond_0
      :try_start_0
      sget-boolean v1, Landroid/app/LoaderManagerImpl;->DEBUG:Z
***************
*** 203,208 ****
--- 214,220 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 438
      :cond_1
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mCallbacks:Landroid/app/LoaderManager$LoaderCallbacks;
  
***************
*** 210,221 ****
--- 222,235 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 440
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v1, v1, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
  
      if-eqz v1, :cond_2
  
+     .line 441
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v1, v1, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
***************
*** 224,238 ****
--- 238,255 ----
  
      iput-object v0, v1, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
+     .line 444
      :cond_2
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mDeliveredData:Z
  
+     .line 446
      .end local v0           #lastBecause:Ljava/lang/String;
      :cond_3
      return-void
  
+     .line 440
      .restart local v0       #lastBecause:Ljava/lang/String;
      :catchall_0
      move-exception v1
***************
*** 243,248 ****
--- 260,266 ----
  
      if-eqz v2, :cond_4
  
+     .line 441
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v2, v2, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
***************
*** 263,268 ****
--- 281,287 ----
  
      const/4 v5, 0x0
  
+     .line 338
      sget-boolean v2, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v2, :cond_0
***************
*** 289,304 ****
--- 308,327 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 339
      :cond_0
      const/4 v2, 0x1
  
      iput-boolean v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mDestroyed:Z
  
+     .line 340
      iget-boolean v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mDeliveredData:Z
  
+     .line 341
      .local v1, needReset:Z
      iput-boolean v5, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mDeliveredData:Z
  
+     .line 342
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mCallbacks:Landroid/app/LoaderManager$LoaderCallbacks;
  
      if-eqz v2, :cond_3
***************
*** 313,318 ****
--- 336,342 ----
  
      if-eqz v1, :cond_3
  
+     .line 343
      sget-boolean v2, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v2, :cond_1
***************
*** 339,347 ****
--- 363,373 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 344
      :cond_1
      const/4 v0, 0x0
  
+     .line 345
      .local v0, lastBecause:Ljava/lang/String;
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
***************
*** 349,354 ****
--- 375,381 ----
  
      if-eqz v2, :cond_2
  
+     .line 346
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v2, v2, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
***************
*** 357,372 ****
  
      iget-object v0, v2, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v2, v2, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
  
      iget-object v2, v2, Landroid/app/Activity;->mFragments:Landroid/app/FragmentManagerImpl;
  
!     const-string v3, "onLoaderReset"
  
      iput-object v3, v2, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
      :cond_2
      :try_start_0
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mCallbacks:Landroid/app/LoaderManager$LoaderCallbacks;
--- 384,401 ----
  
      iget-object v0, v2, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
+     .line 347
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v2, v2, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
  
      iget-object v2, v2, Landroid/app/Activity;->mFragments:Landroid/app/FragmentManagerImpl;
  
!     const-string/jumbo v3, "onLoaderReset"
  
      iput-object v3, v2, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
+     .line 350
      :cond_2
      :try_start_0
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mCallbacks:Landroid/app/LoaderManager$LoaderCallbacks;
***************
*** 377,388 ****
--- 406,419 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 352
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v2, v2, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
  
      if-eqz v2, :cond_3
  
+     .line 353
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v2, v2, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
***************
*** 391,435 ****
--- 422,478 ----
  
      iput-object v0, v2, Landroid/app/FragmentManagerImpl;->mNoTransactionsBecause:Ljava/lang/String;
  
+     .line 357
      .end local v0           #lastBecause:Ljava/lang/String;
      :cond_3
      iput-object v6, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mCallbacks:Landroid/app/LoaderManager$LoaderCallbacks;
  
+     .line 358
      iput-object v6, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mData:Ljava/lang/Object;
  
+     .line 359
      iput-boolean v5, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mHaveData:Z
  
+     .line 360
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      if-eqz v2, :cond_5
  
+     .line 361
      iget-boolean v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mListenerRegistered:Z
  
      if-eqz v2, :cond_4
  
+     .line 362
      iput-boolean v5, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mListenerRegistered:Z
  
+     .line 363
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      invoke-virtual {v2, p0}, Landroid/content/Loader;->unregisterListener(Landroid/content/Loader$OnLoadCompleteListener;)V
  
+     .line 365
      :cond_4
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      invoke-virtual {v2}, Landroid/content/Loader;->reset()V
  
+     .line 367
      :cond_5
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mPendingLoader:Landroid/app/LoaderManagerImpl$LoaderInfo;
  
      if-eqz v2, :cond_6
  
+     .line 368
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mPendingLoader:Landroid/app/LoaderManagerImpl$LoaderInfo;
  
      invoke-virtual {v2}, Landroid/app/LoaderManagerImpl$LoaderInfo;->destroy()V
  
+     .line 370
      :cond_6
      return-void
  
+     .line 352
      .restart local v0       #lastBecause:Ljava/lang/String;
      :catchall_0
      move-exception v2
***************
*** 440,445 ****
--- 483,489 ----
  
      if-eqz v3, :cond_7
  
+     .line 353
      iget-object v3, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v3, v3, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
***************
*** 460,465 ****
--- 504,510 ----
      .parameter "args"
  
      .prologue
+     .line 462
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mId="
***************
*** 470,475 ****
--- 515,521 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 463
      const-string v0, " mArgs="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 478,483 ****
--- 524,530 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 464
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mCallbacks="
***************
*** 488,493 ****
--- 535,541 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 465
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mLoader="
***************
*** 498,507 ****
--- 546,557 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 466
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      if-eqz v0, :cond_0
  
+     .line 467
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 524,529 ****
--- 574,580 ----
  
      invoke-virtual {v0, v1, p2, p3, p4}, Landroid/content/Loader;->dump(Ljava/lang/String;Ljava/io/FileDescriptor;Ljava/io/PrintWriter;[Ljava/lang/String;)V
  
+     .line 469
      :cond_0
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mHaveData:Z
  
***************
*** 533,538 ****
--- 584,590 ----
  
      if-eqz v0, :cond_2
  
+     .line 470
      :cond_1
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 544,549 ****
--- 596,602 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 471
      const-string v0, "  mDeliveredData="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 552,557 ****
--- 605,611 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 472
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mData="
***************
*** 562,567 ****
--- 616,622 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 474
      :cond_2
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 573,578 ****
--- 628,634 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 475
      const-string v0, " mReportNextStart="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 581,586 ****
--- 637,643 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 476
      const-string v0, " mDestroyed="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 589,594 ****
--- 646,652 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 477
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mRetaining="
***************
*** 599,604 ****
--- 657,663 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 478
      const-string v0, " mRetainingStarted="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 607,612 ****
--- 666,672 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 479
      const-string v0, " mListenerRegistered="
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 615,630 ****
--- 675,693 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 480
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mPendingLoader:Landroid/app/LoaderManagerImpl$LoaderInfo;
  
      if-eqz v0, :cond_3
  
+     .line 481
      invoke-virtual {p3, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "Pending Loader "
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 482
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mPendingLoader:Landroid/app/LoaderManagerImpl$LoaderInfo;
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/Object;)V
***************
*** 633,638 ****
--- 696,702 ----
  
      invoke-virtual {p3, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 483
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mPendingLoader:Landroid/app/LoaderManagerImpl$LoaderInfo;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 655,660 ****
--- 719,725 ----
  
      invoke-virtual {v0, v1, p2, p3, p4}, Landroid/app/LoaderManagerImpl$LoaderInfo;->dump(Ljava/lang/String;Ljava/io/FileDescriptor;Ljava/io/PrintWriter;[Ljava/lang/String;)V
  
+     .line 485
      :cond_3
      return-void
  .end method
***************
*** 663,672 ****
--- 728,739 ----
      .locals 3
  
      .prologue
+     .line 289
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mRetaining:Z
  
      if-eqz v0, :cond_1
  
+     .line 290
      sget-boolean v0, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v0, :cond_0
***************
*** 693,715 ****
--- 760,787 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 291
      :cond_0
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mRetaining:Z
  
+     .line 292
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
      iget-boolean v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mRetainingStarted:Z
  
      if-eq v0, v1, :cond_1
  
+     .line 293
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
      if-nez v0, :cond_1
  
+     .line 297
      invoke-virtual {p0}, Landroid/app/LoaderManagerImpl$LoaderInfo;->stop()V
  
+     .line 302
      :cond_1
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
***************
*** 723,734 ****
--- 795,808 ----
  
      if-nez v0, :cond_2
  
+     .line 309
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mData:Ljava/lang/Object;
  
      invoke-virtual {p0, v0, v1}, Landroid/app/LoaderManagerImpl$LoaderInfo;->callOnLoadFinished(Landroid/content/Loader;Ljava/lang/Object;)V
  
+     .line 311
      :cond_2
      return-void
  .end method
***************
*** 753,758 ****
--- 827,833 ----
      .local p1, loader:Landroid/content/Loader;,"Landroid/content/Loader<Ljava/lang/Object;>;"
      const/4 v5, 0x0
  
+     .line 373
      sget-boolean v2, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v2, :cond_0
***************
*** 763,769 ****
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v4, "onLoadComplete: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 838,844 ----
  
      invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v4, "onLoadComplete: "
  
      invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 779,789 ****
--- 854,866 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 375
      :cond_0
      iget-boolean v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mDestroyed:Z
  
      if-eqz v2, :cond_2
  
+     .line 376
      sget-boolean v2, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v2, :cond_1
***************
*** 794,803 ****
--- 871,882 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 426
      :cond_1
      :goto_0
      return-void
  
+     .line 380
      :cond_2
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
***************
*** 811,816 ****
--- 890,896 ----
  
      if-eq v2, p0, :cond_3
  
+     .line 383
      sget-boolean v2, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v2, :cond_1
***************
*** 823,834 ****
--- 903,917 ----
  
      goto :goto_0
  
+     .line 387
      :cond_3
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mPendingLoader:Landroid/app/LoaderManagerImpl$LoaderInfo;
  
+     .line 388
      .local v1, pending:Landroid/app/LoaderManagerImpl$LoaderInfo;
      if-eqz v1, :cond_5
  
+     .line 392
      sget-boolean v2, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v2, :cond_4
***************
*** 855,863 ****
--- 938,948 ----
  
      invoke-static {v2, v3}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 393
      :cond_4
      iput-object v5, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mPendingLoader:Landroid/app/LoaderManagerImpl$LoaderInfo;
  
+     .line 394
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v2, v2, Landroid/app/LoaderManagerImpl;->mLoaders:Landroid/util/SparseArray;
***************
*** 866,879 ****
--- 951,967 ----
  
      invoke-virtual {v2, v3, v5}, Landroid/util/SparseArray;->put(ILjava/lang/Object;)V
  
+     .line 395
      invoke-virtual {p0}, Landroid/app/LoaderManagerImpl$LoaderInfo;->destroy()V
  
+     .line 396
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      invoke-virtual {v2, v1}, Landroid/app/LoaderManagerImpl;->installLoader(Landroid/app/LoaderManagerImpl$LoaderInfo;)V
  
      goto :goto_0
  
+     .line 402
      :cond_5
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mData:Ljava/lang/Object;
  
***************
*** 883,901 ****
--- 971,994 ----
  
      if-nez v2, :cond_7
  
+     .line 403
      :cond_6
      iput-object p2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mData:Ljava/lang/Object;
  
+     .line 404
      const/4 v2, 0x1
  
      iput-boolean v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mHaveData:Z
  
+     .line 405
      iget-boolean v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
      if-eqz v2, :cond_7
  
+     .line 406
      invoke-virtual {p0, p1, p2}, Landroid/app/LoaderManagerImpl$LoaderInfo;->callOnLoadFinished(Landroid/content/Loader;Ljava/lang/Object;)V
  
+     .line 416
      :cond_7
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
***************
*** 909,925 ****
--- 1002,1022 ----
  
      check-cast v0, Landroid/app/LoaderManagerImpl$LoaderInfo;
  
+     .line 417
      .local v0, info:Landroid/app/LoaderManagerImpl$LoaderInfo;
      if-eqz v0, :cond_8
  
      if-eq v0, p0, :cond_8
  
+     .line 418
      const/4 v2, 0x0
  
      iput-boolean v2, v0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mDeliveredData:Z
  
+     .line 419
      invoke-virtual {v0}, Landroid/app/LoaderManagerImpl$LoaderInfo;->destroy()V
  
+     .line 420
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v2, v2, Landroid/app/LoaderManagerImpl;->mInactiveLoaders:Landroid/util/SparseArray;
***************
*** 928,933 ****
--- 1025,1031 ----
  
      invoke-virtual {v2, v3}, Landroid/util/SparseArray;->remove(I)V
  
+     .line 423
      :cond_8
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
***************
*** 943,948 ****
--- 1041,1047 ----
  
      if-nez v2, :cond_1
  
+     .line 424
      iget-object v2, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->this$0:Landroid/app/LoaderManagerImpl;
  
      iget-object v2, v2, Landroid/app/LoaderManagerImpl;->mActivity:Landroid/app/Activity;
***************
*** 958,985 ****
--- 1057,1090 ----
      .locals 2
  
      .prologue
+     .line 314
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
      if-eqz v0, :cond_0
  
+     .line 315
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mReportNextStart:Z
  
      if-eqz v0, :cond_0
  
+     .line 316
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mReportNextStart:Z
  
+     .line 317
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mHaveData:Z
  
      if-eqz v0, :cond_0
  
+     .line 318
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mData:Ljava/lang/Object;
  
      invoke-virtual {p0, v0, v1}, Landroid/app/LoaderManagerImpl$LoaderInfo;->callOnLoadFinished(Landroid/content/Loader;Ljava/lang/Object;)V
  
+     .line 322
      :cond_0
      return-void
  .end method
***************
*** 988,993 ****
--- 1093,1099 ----
      .locals 3
  
      .prologue
+     .line 281
      sget-boolean v0, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v0, :cond_0
***************
*** 1014,1036 ****
--- 1120,1147 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 282
      :cond_0
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mRetaining:Z
  
+     .line 283
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
      iput-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mRetainingStarted:Z
  
+     .line 284
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
+     .line 285
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mCallbacks:Landroid/app/LoaderManager$LoaderCallbacks;
  
+     .line 286
      return-void
  .end method
  
***************
*** 1040,1045 ****
--- 1151,1157 ----
      .prologue
      const/4 v3, 0x1
  
+     .line 246
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mRetaining:Z
  
      if-eqz v0, :cond_1
***************
*** 1048,1066 ****
--- 1160,1183 ----
  
      if-eqz v0, :cond_1
  
+     .line 250
      iput-boolean v3, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
+     .line 278
      :cond_0
      :goto_0
      return-void
  
+     .line 254
      :cond_1
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
      if-nez v0, :cond_0
  
+     .line 259
      iput-boolean v3, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
+     .line 261
      sget-boolean v0, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v0, :cond_2
***************
*** 1087,1092 ****
--- 1204,1210 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 262
      :cond_2
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
***************
*** 1096,1101 ****
--- 1214,1220 ----
  
      if-eqz v0, :cond_3
  
+     .line 263
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mCallbacks:Landroid/app/LoaderManager$LoaderCallbacks;
  
      iget v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mId:I
***************
*** 1108,1118 ****
--- 1227,1239 ----
  
      iput-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
+     .line 265
      :cond_3
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      if-eqz v0, :cond_0
  
+     .line 266
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
***************
*** 1141,1146 ****
--- 1262,1268 ----
  
      if-nez v0, :cond_4
  
+     .line 268
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 1167,1185 ****
--- 1289,1311 ----
  
      throw v0
  
+     .line 272
      :cond_4
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mListenerRegistered:Z
  
      if-nez v0, :cond_5
  
+     .line 273
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      iget v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mId:I
  
      invoke-virtual {v0, v1, p0}, Landroid/content/Loader;->registerListener(ILandroid/content/Loader$OnLoadCompleteListener;)V
  
+     .line 274
      iput-boolean v3, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mListenerRegistered:Z
  
+     .line 276
      :cond_5
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
***************
*** 1194,1199 ****
--- 1320,1326 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 325
      sget-boolean v0, Landroid/app/LoaderManagerImpl;->DEBUG:Z
  
      if-eqz v0, :cond_0
***************
*** 1220,1232 ****
--- 1347,1362 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 326
      :cond_0
      iput-boolean v3, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mStarted:Z
  
+     .line 327
      iget-boolean v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mRetaining:Z
  
      if-nez v0, :cond_1
  
+     .line 328
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      if-eqz v0, :cond_1
***************
*** 1235,1250 ****
--- 1365,1384 ----
  
      if-eqz v0, :cond_1
  
+     .line 330
      iput-boolean v3, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mListenerRegistered:Z
  
+     .line 331
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      invoke-virtual {v0, p0}, Landroid/content/Loader;->unregisterListener(Landroid/content/Loader$OnLoadCompleteListener;)V
  
+     .line 332
      iget-object v0, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      invoke-virtual {v0}, Landroid/content/Loader;->stopLoading()V
  
+     .line 335
      :cond_1
      return-void
  .end method
***************
*** 1253,1269 ****
--- 1387,1406 ----
      .locals 2
  
      .prologue
+     .line 450
      new-instance v0, Ljava/lang/StringBuilder;
  
      const/16 v1, 0x40
  
      invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(I)V
  
+     .line 451
      .local v0, sb:Ljava/lang/StringBuilder;
      const-string v1, "LoaderInfo{"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 452
      invoke-static {p0}, Ljava/lang/System;->identityHashCode(Ljava/lang/Object;)I
  
      move-result v1
***************
*** 1274,1299 ****
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      const-string v1, " #"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      iget v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mId:I
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
      const-string v1, " : "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      invoke-static {v1, v0}, Landroid/util/DebugUtils;->buildShortClassTag(Ljava/lang/Object;Ljava/lang/StringBuilder;)V
  
!     const-string v1, "}}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
--- 1411,1442 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 453
      const-string v1, " #"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 454
      iget v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mId:I
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 455
      const-string v1, " : "
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 456
      iget-object v1, p0, Landroid/app/LoaderManagerImpl$LoaderInfo;->mLoader:Landroid/content/Loader;
  
      invoke-static {v1, v0}, Landroid/util/DebugUtils;->buildShortClassTag(Ljava/lang/Object;Ljava/lang/StringBuilder;)V
  
!     .line 457
!     const-string/jumbo v1, "}}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 458
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
