*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/LoadedApk$WarningContextClassLoader.smali	2012-03-23 19:36:35.950336538 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/LoadedApk$WarningContextClassLoader.smali.original	2012-03-23 19:34:45.239317725 -0500
***************
*** 23,28 ****
--- 23,29 ----
      .locals 1
  
      .prologue
+     .line 376
      const/4 v0, 0x0
  
      sput-boolean v0, Landroid/app/LoadedApk$WarningContextClassLoader;->warned:Z
***************
*** 34,39 ****
--- 35,41 ----
      .locals 0
  
      .prologue
+     .line 374
      invoke-direct {p0}, Ljava/lang/ClassLoader;-><init>()V
  
      return-void
***************
*** 44,49 ****
--- 46,52 ----
      .parameter "x0"
  
      .prologue
+     .line 374
      invoke-direct {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;-><init>()V
  
      return-void
***************
*** 54,71 ****
--- 57,78 ----
      .parameter "methodName"
  
      .prologue
+     .line 379
      sget-boolean v0, Landroid/app/LoadedApk$WarningContextClassLoader;->warned:Z
  
      if-eqz v0, :cond_0
  
+     .line 390
      :goto_0
      return-void
  
+     .line 382
      :cond_0
      const/4 v0, 0x1
  
      sput-boolean v0, Landroid/app/LoadedApk$WarningContextClassLoader;->warned:Z
  
+     .line 383
      invoke-static {}, Ljava/lang/Thread;->currentThread()Ljava/lang/Thread;
  
      move-result-object v0
***************
*** 76,81 ****
--- 83,89 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/Thread;->setContextClassLoader(Ljava/lang/ClassLoader;)V
  
+     .line 384
      const-string v0, "ActivityThread"
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 110,122 ****
  
      move-result-object v1
  
!     const-string v2, "that host multiple applications. You should explicitly "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      move-result-object v1
  
!     const-string v2, "specify a context class loader. For example: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 118,130 ----
  
      move-result-object v1
  
!     const-string/jumbo v2, "that host multiple applications. You should explicitly "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
      move-result-object v1
  
!     const-string/jumbo v2, "specify a context class loader. For example: "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 143,158 ****
--- 151,169 ----
      .locals 1
  
      .prologue
+     .line 428
      const-string v0, "clearAssertionStatus"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
+     .line 429
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
  
      invoke-virtual {v0}, Ljava/lang/ClassLoader;->clearAssertionStatus()V
  
+     .line 430
      return-void
  .end method
  
***************
*** 161,170 ****
--- 172,183 ----
      .parameter "resName"
  
      .prologue
+     .line 393
      const-string v0, "getResource"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
+     .line 394
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
***************
*** 181,190 ****
--- 194,205 ----
      .parameter "resName"
  
      .prologue
+     .line 403
      const-string v0, "getResourceAsStream"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
+     .line 404
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
***************
*** 218,227 ****
--- 233,244 ----
      .end annotation
  
      .prologue
+     .line 398
      const-string v0, "getResources"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
+     .line 399
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
***************
*** 253,262 ****
--- 270,281 ----
      .end annotation
  
      .prologue
+     .line 408
      const-string v0, "loadClass"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
+     .line 409
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
***************
*** 274,289 ****
      .parameter "enable"
  
      .prologue
!     const-string v0, "setClassAssertionStatus"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
  
      invoke-virtual {v0, p1, p2}, Ljava/lang/ClassLoader;->setClassAssertionStatus(Ljava/lang/String;Z)V
  
      return-void
  .end method
  
--- 293,311 ----
      .parameter "enable"
  
      .prologue
!     .line 413
!     const-string/jumbo v0, "setClassAssertionStatus"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
+     .line 414
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
  
      invoke-virtual {v0, p1, p2}, Ljava/lang/ClassLoader;->setClassAssertionStatus(Ljava/lang/String;Z)V
  
+     .line 415
      return-void
  .end method
  
***************
*** 292,307 ****
      .parameter "enable"
  
      .prologue
!     const-string v0, "setDefaultAssertionStatus"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
  
      invoke-virtual {v0, p1}, Ljava/lang/ClassLoader;->setDefaultAssertionStatus(Z)V
  
      return-void
  .end method
  
--- 314,332 ----
      .parameter "enable"
  
      .prologue
!     .line 423
!     const-string/jumbo v0, "setDefaultAssertionStatus"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
+     .line 424
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
  
      invoke-virtual {v0, p1}, Ljava/lang/ClassLoader;->setDefaultAssertionStatus(Z)V
  
+     .line 425
      return-void
  .end method
  
***************
*** 311,325 ****
      .parameter "enable"
  
      .prologue
!     const-string v0, "setPackageAssertionStatus"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
  
      invoke-virtual {v0, p1, p2}, Ljava/lang/ClassLoader;->setPackageAssertionStatus(Ljava/lang/String;Z)V
  
      return-void
  .end method
--- 336,353 ----
      .parameter "enable"
  
      .prologue
!     .line 418
!     const-string/jumbo v0, "setPackageAssertionStatus"
  
      invoke-direct {p0, v0}, Landroid/app/LoadedApk$WarningContextClassLoader;->warn(Ljava/lang/String;)V
  
+     .line 419
      invoke-virtual {p0}, Landroid/app/LoadedApk$WarningContextClassLoader;->getParent()Ljava/lang/ClassLoader;
  
      move-result-object v0
  
      invoke-virtual {v0, p1, p2}, Ljava/lang/ClassLoader;->setPackageAssertionStatus(Ljava/lang/String;Z)V
  
+     .line 420
      return-void
  .end method
