*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/NativeActivity$NativeContentView.smali	2012-03-23 19:36:38.084375453 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/NativeActivity$NativeContentView.smali.original	2012-03-23 19:34:45.272318327 -0500
***************
*** 24,31 ****
--- 24,33 ----
      .parameter "context"
  
      .prologue
+     .line 114
      invoke-direct {p0, p1}, Landroid/view/View;-><init>(Landroid/content/Context;)V
  
+     .line 115
      return-void
  .end method
  
***************
*** 35,41 ****
--- 37,45 ----
      .parameter "attrs"
  
      .prologue
+     .line 118
      invoke-direct {p0, p1, p2}, Landroid/view/View;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 119
      return-void
  .end method
