*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/PendingIntent$CanceledException.smali	2012-03-23 19:36:35.021319596 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/PendingIntent$CanceledException.smali.original	2012-03-23 19:34:45.207317141 -0500
***************
*** 19,26 ****
--- 19,28 ----
      .locals 0
  
      .prologue
+     .line 101
      invoke-direct {p0}, Landroid/util/AndroidException;-><init>()V
  
+     .line 102
      return-void
  .end method
  
***************
*** 29,36 ****
--- 31,40 ----
      .parameter "cause"
  
      .prologue
+     .line 109
      invoke-direct {p0, p1}, Landroid/util/AndroidException;-><init>(Ljava/lang/Exception;)V
  
+     .line 110
      return-void
  .end method
  
***************
*** 39,45 ****
--- 43,51 ----
      .parameter "name"
  
      .prologue
+     .line 105
      invoke-direct {p0, p1}, Landroid/util/AndroidException;-><init>(Ljava/lang/String;)V
  
+     .line 106
      return-void
  .end method
