*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/PendingIntent$FinishedDispatcher.smali	2012-03-23 19:36:42.215450782 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/PendingIntent$FinishedDispatcher.smali.original	2012-03-23 19:34:45.308318983 -0500
***************
*** 41,54 ****
--- 41,59 ----
      .parameter "handler"
  
      .prologue
+     .line 144
      invoke-direct {p0}, Landroid/content/IIntentReceiver$Stub;-><init>()V
  
+     .line 145
      iput-object p1, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mPendingIntent:Landroid/app/PendingIntent;
  
+     .line 146
      iput-object p2, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mWho:Landroid/app/PendingIntent$OnFinished;
  
+     .line 147
      iput-object p3, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mHandler:Landroid/os/Handler;
  
+     .line 148
      return-void
  .end method
  
***************
*** 64,86 ****
--- 69,99 ----
      .parameter "sticky"
  
      .prologue
+     .line 151
      iput-object p1, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mIntent:Landroid/content/Intent;
  
+     .line 152
      iput p2, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mResultCode:I
  
+     .line 153
      iput-object p3, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mResultData:Ljava/lang/String;
  
+     .line 154
      iput-object p4, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mResultExtras:Landroid/os/Bundle;
  
+     .line 155
      iget-object v0, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mHandler:Landroid/os/Handler;
  
      if-nez v0, :cond_0
  
+     .line 156
      invoke-virtual {p0}, Landroid/app/PendingIntent$FinishedDispatcher;->run()V
  
+     .line 160
      :goto_0
      return-void
  
+     .line 158
      :cond_0
      iget-object v0, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mHandler:Landroid/os/Handler;
  
***************
*** 93,98 ****
--- 106,112 ----
      .locals 6
  
      .prologue
+     .line 162
      iget-object v0, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mWho:Landroid/app/PendingIntent$OnFinished;
  
      iget-object v1, p0, Landroid/app/PendingIntent$FinishedDispatcher;->mPendingIntent:Landroid/app/PendingIntent;
***************
*** 107,111 ****
--- 121,126 ----
  
      invoke-interface/range {v0 .. v5}, Landroid/app/PendingIntent$OnFinished;->onSendFinished(Landroid/app/PendingIntent;Landroid/content/Intent;ILjava/lang/String;Landroid/os/Bundle;)V
  
+     .line 164
      return-void
  .end method
