*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SearchDialog$4.smali	2012-03-23 19:36:38.588384643 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SearchDialog$4.smali.original	2012-03-23 19:34:45.280318473 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .parameter
  
      .prologue
+     .line 682
      iput-object p1, p0, Landroid/app/SearchDialog$4;->this$0:Landroid/app/SearchDialog;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 41,46 ****
--- 42,48 ----
      .parameter "newText"
  
      .prologue
+     .line 690
      const/4 v0, 0x0
  
      return v0
***************
*** 51,60 ****
--- 53,64 ----
      .parameter "query"
  
      .prologue
+     .line 685
      iget-object v0, p0, Landroid/app/SearchDialog$4;->this$0:Landroid/app/SearchDialog;
  
      invoke-virtual {v0}, Landroid/app/SearchDialog;->dismiss()V
  
+     .line 686
      const/4 v0, 0x0
  
      return v0
