*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SearchableInfo$1.smali	2012-03-23 19:36:38.840389238 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SearchableInfo$1.smali.original	2012-03-23 19:34:45.282318509 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 778
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 44,49 ****
--- 45,51 ----
      .parameter "in"
  
      .prologue
+     .line 780
      new-instance v0, Landroid/app/SearchableInfo;
  
      invoke-direct {v0, p1}, Landroid/app/SearchableInfo;-><init>(Landroid/os/Parcel;)V
***************
*** 56,61 ****
--- 58,64 ----
      .parameter "x0"
  
      .prologue
+     .line 778
      invoke-virtual {p0, p1}, Landroid/app/SearchableInfo$1;->createFromParcel(Landroid/os/Parcel;)Landroid/app/SearchableInfo;
  
      move-result-object v0
***************
*** 68,73 ****
--- 71,77 ----
      .parameter "size"
  
      .prologue
+     .line 784
      new-array v0, p1, [Landroid/app/SearchableInfo;
  
      return-object v0
***************
*** 78,83 ****
--- 82,88 ----
      .parameter "x0"
  
      .prologue
+     .line 778
      invoke-virtual {p0, p1}, Landroid/app/SearchableInfo$1;->newArray(I)[Landroid/app/SearchableInfo;
  
      move-result-object v0
