*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SearchDialog$SearchBar.smali	2012-03-23 19:36:35.321325068 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SearchDialog$SearchBar.smali.original	2012-03-23 19:34:45.233317615 -0500
***************
*** 24,31 ****
--- 24,33 ----
      .parameter "context"
  
      .prologue
+     .line 630
      invoke-direct {p0, p1}, Landroid/widget/LinearLayout;-><init>(Landroid/content/Context;)V
  
+     .line 631
      return-void
  .end method
  
***************
*** 35,42 ****
--- 37,46 ----
      .parameter "attrs"
  
      .prologue
+     .line 626
      invoke-direct {p0, p1, p2}, Landroid/widget/LinearLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;)V
  
+     .line 627
      return-void
  .end method
  
***************
*** 47,54 ****
--- 51,60 ----
      .parameter "searchDialog"
  
      .prologue
+     .line 634
      iput-object p1, p0, Landroid/app/SearchDialog$SearchBar;->mSearchDialog:Landroid/app/SearchDialog;
  
+     .line 635
      return-void
  .end method
  
***************
*** 58,63 ****
--- 64,70 ----
      .parameter "callback"
  
      .prologue
+     .line 642
      const/4 v0, 0x0
  
      return-object v0
