*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Service.smali	2012-03-23 19:36:42.192450362 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/Service.smali.original	2012-03-23 19:34:45.308318983 -0500
***************
*** 47,68 ****
--- 47,76 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 283
      invoke-direct {p0, v0}, Landroid/content/ContextWrapper;-><init>(Landroid/content/Context;)V
  
+     .line 695
      iput-object v0, p0, Landroid/app/Service;->mThread:Landroid/app/ActivityThread;
  
+     .line 696
      iput-object v0, p0, Landroid/app/Service;->mClassName:Ljava/lang/String;
  
+     .line 697
      iput-object v0, p0, Landroid/app/Service;->mToken:Landroid/os/IBinder;
  
+     .line 698
      iput-object v0, p0, Landroid/app/Service;->mApplication:Landroid/app/Application;
  
+     .line 699
      iput-object v0, p0, Landroid/app/Service;->mActivityManager:Landroid/app/IActivityManager;
  
+     .line 700
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Landroid/app/Service;->mStartCompatibility:Z
  
+     .line 284
      return-void
  .end method
  
***************
*** 78,98 ****
--- 86,113 ----
      .parameter "activityManager"
  
      .prologue
+     .line 680
      invoke-virtual {p0, p1}, Landroid/app/Service;->attachBaseContext(Landroid/content/Context;)V
  
+     .line 681
      iput-object p2, p0, Landroid/app/Service;->mThread:Landroid/app/ActivityThread;
  
+     .line 682
      iput-object p3, p0, Landroid/app/Service;->mClassName:Ljava/lang/String;
  
+     .line 683
      iput-object p4, p0, Landroid/app/Service;->mToken:Landroid/os/IBinder;
  
+     .line 684
      iput-object p5, p0, Landroid/app/Service;->mApplication:Landroid/app/Application;
  
+     .line 685
      check-cast p6, Landroid/app/IActivityManager;
  
      .end local p6
      iput-object p6, p0, Landroid/app/Service;->mActivityManager:Landroid/app/IActivityManager;
  
+     .line 686
      invoke-virtual {p0}, Landroid/app/Service;->getApplicationInfo()Landroid/content/pm/ApplicationInfo;
  
      move-result-object v0
***************
*** 108,115 ****
--- 123,132 ----
      :goto_0
      iput-boolean v0, p0, Landroid/app/Service;->mStartCompatibility:Z
  
+     .line 688
      return-void
  
+     .line 686
      :cond_0
      const/4 v0, 0x0
  
***************
*** 123,132 ****
      .parameter "args"
  
      .prologue
!     const-string v0, "nothing to dump"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
      return-void
  .end method
  
--- 140,151 ----
      .parameter "args"
  
      .prologue
!     .line 668
!     const-string/jumbo v0, "nothing to dump"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 669
      return-void
  .end method
  
***************
*** 134,139 ****
--- 153,159 ----
      .locals 1
  
      .prologue
+     .line 288
      iget-object v0, p0, Landroid/app/Service;->mApplication:Landroid/app/Application;
  
      return-object v0
***************
*** 143,148 ****
--- 163,169 ----
      .locals 1
  
      .prologue
+     .line 691
      iget-object v0, p0, Landroid/app/Service;->mClassName:Ljava/lang/String;
  
      return-object v0
***************
*** 156,161 ****
--- 177,183 ----
      .parameter "newConfig"
  
      .prologue
+     .line 452
      return-void
  .end method
  
***************
*** 163,168 ****
--- 185,191 ----
      .locals 0
  
      .prologue
+     .line 295
      return-void
  .end method
  
***************
*** 170,175 ****
--- 193,199 ----
      .locals 0
  
      .prologue
+     .line 449
      return-void
  .end method
  
***************
*** 177,182 ****
--- 201,207 ----
      .locals 0
  
      .prologue
+     .line 455
      return-void
  .end method
  
***************
*** 185,190 ****
--- 210,216 ----
      .parameter "intent"
  
      .prologue
+     .line 512
      return-void
  .end method
  
***************
*** 196,201 ****
--- 222,228 ----
      .end annotation
  
      .prologue
+     .line 302
      return-void
  .end method
  
***************
*** 206,213 ****
--- 233,242 ----
      .parameter "startId"
  
      .prologue
+     .line 438
      invoke-virtual {p0, p1, p3}, Landroid/app/Service;->onStart(Landroid/content/Intent;I)V
  
+     .line 439
      iget-boolean v0, p0, Landroid/app/Service;->mStartCompatibility:Z
  
      if-eqz v0, :cond_0
***************
*** 228,233 ****
--- 257,263 ----
      .parameter "rootIntent"
  
      .prologue
+     .line 525
      return-void
  .end method
  
***************
*** 236,241 ****
--- 266,272 ----
      .parameter "level"
  
      .prologue
+     .line 458
      return-void
  .end method
  
***************
*** 244,249 ****
--- 275,281 ----
      .parameter "intent"
  
      .prologue
+     .line 497
      const/4 v0, 0x0
  
      return v0
***************
*** 256,268 ****
      .end annotation
  
      .prologue
      const-string v0, "Service"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "setForeground: ignoring old API call on "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 288,301 ----
      .end annotation
  
      .prologue
+     .line 602
      const-string v0, "Service"
  
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "setForeground: ignoring old API call on "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 286,291 ****
--- 319,325 ----
  
      invoke-static {v0, v1}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 603
      return-void
  .end method
  
***************
*** 295,300 ****
--- 329,335 ----
      .parameter "notification"
  
      .prologue
+     .line 631
      :try_start_0
      iget-object v0, p0, Landroid/app/Service;->mActivityManager:Landroid/app/IActivityManager;
  
***************
*** 316,324 ****
--- 351,361 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 636
      :goto_0
      return-void
  
+     .line 634
      :catch_0
      move-exception v0
  
***************
*** 330,335 ****
--- 367,373 ----
      .parameter "removeNotification"
  
      .prologue
+     .line 648
      :try_start_0
      iget-object v0, p0, Landroid/app/Service;->mActivityManager:Landroid/app/IActivityManager;
  
***************
*** 351,359 ****
--- 389,399 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 653
      :goto_0
      return-void
  
+     .line 651
      :catch_0
      move-exception v0
  
***************
*** 364,373 ****
--- 404,415 ----
      .locals 1
  
      .prologue
+     .line 534
      const/4 v0, -0x1
  
      invoke-virtual {p0, v0}, Landroid/app/Service;->stopSelf(I)V
  
+     .line 535
      return-void
  .end method
  
***************
*** 376,388 ****
--- 418,433 ----
      .parameter "startId"
  
      .prologue
+     .line 543
      iget-object v0, p0, Landroid/app/Service;->mActivityManager:Landroid/app/IActivityManager;
  
      if-nez v0, :cond_0
  
+     .line 551
      :goto_0
      return-void
  
+     .line 547
      :cond_0
      :try_start_0
      iget-object v0, p0, Landroid/app/Service;->mActivityManager:Landroid/app/IActivityManager;
***************
*** 401,406 ****
--- 446,452 ----
  
      goto :goto_0
  
+     .line 549
      :catch_0
      move-exception v0
  
***************
*** 414,426 ****
--- 460,475 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 575
      iget-object v1, p0, Landroid/app/Service;->mActivityManager:Landroid/app/IActivityManager;
  
      if-nez v1, :cond_0
  
+     .line 583
      :goto_0
      return v0
  
+     .line 579
      :cond_0
      :try_start_0
      iget-object v1, p0, Landroid/app/Service;->mActivityManager:Landroid/app/IActivityManager;
***************
*** 441,446 ****
--- 490,496 ----
  
      goto :goto_0
  
+     .line 581
      :catch_0
      move-exception v1
  
