*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SharedPreferencesImpl$EditorImpl$2.smali	2012-03-23 19:36:36.282342591 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SharedPreferencesImpl$EditorImpl$2.smali.original	2012-03-23 19:34:45.242317779 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 352
      iput-object p1, p0, Landroid/app/SharedPreferencesImpl$EditorImpl$2;->this$1:Landroid/app/SharedPreferencesImpl$EditorImpl;
  
      iput-object p2, p0, Landroid/app/SharedPreferencesImpl$EditorImpl$2;->val$awaitCommit:Ljava/lang/Runnable;
***************
*** 45,57 ****
--- 46,61 ----
      .locals 1
  
      .prologue
+     .line 354
      iget-object v0, p0, Landroid/app/SharedPreferencesImpl$EditorImpl$2;->val$awaitCommit:Ljava/lang/Runnable;
  
      invoke-interface {v0}, Ljava/lang/Runnable;->run()V
  
+     .line 355
      iget-object v0, p0, Landroid/app/SharedPreferencesImpl$EditorImpl$2;->val$awaitCommit:Ljava/lang/Runnable;
  
      invoke-static {v0}, Landroid/app/QueuedWork;->remove(Ljava/lang/Runnable;)V
  
+     .line 356
      return-void
  .end method
