*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SharedPreferencesImpl.smali	2012-03-23 19:36:41.936445693 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/SharedPreferencesImpl.smali.original	2012-03-23 19:34:45.306318947 -0500
***************
*** 70,75 ****
--- 70,76 ----
      .locals 1
  
      .prologue
+     .line 67
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
***************
*** 87,128 ****
--- 88,141 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 71
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 61
      iput v1, p0, Landroid/app/SharedPreferencesImpl;->mDiskWritesInFlight:I
  
+     .line 62
      iput-boolean v1, p0, Landroid/app/SharedPreferencesImpl;->mLoaded:Z
  
+     .line 66
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
  
      iput-object v0, p0, Landroid/app/SharedPreferencesImpl;->mWritingToDiskLock:Ljava/lang/Object;
  
+     .line 68
      new-instance v0, Ljava/util/WeakHashMap;
  
      invoke-direct {v0}, Ljava/util/WeakHashMap;-><init>()V
  
      iput-object v0, p0, Landroid/app/SharedPreferencesImpl;->mListeners:Ljava/util/WeakHashMap;
  
+     .line 72
      iput-object p1, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
+     .line 73
      invoke-static {p1}, Landroid/app/SharedPreferencesImpl;->makeBackupFile(Ljava/io/File;)Ljava/io/File;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/app/SharedPreferencesImpl;->mBackupFile:Ljava/io/File;
  
+     .line 74
      iput p2, p0, Landroid/app/SharedPreferencesImpl;->mMode:I
  
+     .line 75
      iput-boolean v1, p0, Landroid/app/SharedPreferencesImpl;->mLoaded:Z
  
+     .line 76
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
+     .line 77
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->startLoadFromDisk()V
  
+     .line 78
      return-void
  .end method
  
***************
*** 131,136 ****
--- 144,150 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->loadFromDiskLocked()V
  
      return-void
***************
*** 143,148 ****
--- 157,163 ----
      .parameter "x2"
  
      .prologue
+     .line 48
      invoke-direct {p0, p1, p2}, Landroid/app/SharedPreferencesImpl;->enqueueDiskWrite(Landroid/app/SharedPreferencesImpl$MemoryCommitResult;Ljava/lang/Runnable;)V
  
      return-void
***************
*** 153,158 ****
--- 168,174 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget v0, p0, Landroid/app/SharedPreferencesImpl;->mDiskWritesInFlight:I
  
      return v0
***************
*** 163,168 ****
--- 179,185 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget v0, p0, Landroid/app/SharedPreferencesImpl;->mDiskWritesInFlight:I
  
      add-int/lit8 v1, v0, 0x1
***************
*** 177,182 ****
--- 194,200 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget v0, p0, Landroid/app/SharedPreferencesImpl;->mDiskWritesInFlight:I
  
      add-int/lit8 v1, v0, -0x1
***************
*** 191,196 ****
--- 209,215 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget-object v0, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
      return-object v0
***************
*** 202,207 ****
--- 221,227 ----
      .parameter "x1"
  
      .prologue
+     .line 48
      iput-object p1, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
      return-object p1
***************
*** 212,217 ****
--- 232,238 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget-object v0, p0, Landroid/app/SharedPreferencesImpl;->mListeners:Ljava/util/WeakHashMap;
  
      return-object v0
***************
*** 222,227 ****
--- 243,249 ----
      .parameter "x0"
  
      .prologue
+     .line 48
      iget-object v0, p0, Landroid/app/SharedPreferencesImpl;->mWritingToDiskLock:Ljava/lang/Object;
  
      return-object v0
***************
*** 233,238 ****
--- 255,261 ----
      .parameter "x1"
  
      .prologue
+     .line 48
      invoke-direct {p0, p1}, Landroid/app/SharedPreferencesImpl;->writeToFile(Landroid/app/SharedPreferencesImpl$MemoryCommitResult;)V
  
      return-void
***************
*** 242,263 ****
--- 265,290 ----
      .locals 1
  
      .prologue
+     .line 180
      iget-boolean v0, p0, Landroid/app/SharedPreferencesImpl;->mLoaded:Z
  
      if-nez v0, :cond_0
  
+     .line 184
      invoke-static {}, Ldalvik/system/BlockGuard;->getThreadPolicy()Ldalvik/system/BlockGuard$Policy;
  
      move-result-object v0
  
      invoke-interface {v0}, Ldalvik/system/BlockGuard$Policy;->onReadFromDisk()V
  
+     .line 186
      :cond_0
      :goto_0
      iget-boolean v0, p0, Landroid/app/SharedPreferencesImpl;->mLoaded:Z
  
      if-nez v0, :cond_1
  
+     .line 188
      :try_start_0
      invoke-virtual {p0}, Ljava/lang/Object;->wait()V
      :try_end_0
***************
*** 265,275 ****
--- 292,304 ----
  
      goto :goto_0
  
+     .line 189
      :catch_0
      move-exception v0
  
      goto :goto_0
  
+     .line 192
      :cond_1
      return-void
  .end method
***************
*** 281,288 ****
--- 310,319 ----
      .prologue
      const/4 v7, -0x1
  
+     .line 521
      const/4 v3, 0x0
  
+     .line 523
      .local v3, str:Ljava/io/FileOutputStream;
      :try_start_0
      new-instance v4, Ljava/io/FileOutputStream;
***************
*** 300,316 ****
--- 331,351 ----
      :goto_0
      move-object v5, v3
  
+     .line 540
      :goto_1
      return-object v5
  
+     .line 524
      :catch_0
      move-exception v0
  
+     .line 525
      .local v0, e:Ljava/io/FileNotFoundException;
      invoke-virtual {p0}, Ljava/io/File;->getParentFile()Ljava/io/File;
  
      move-result-object v2
  
+     .line 526
      .local v2, parent:Ljava/io/File;
      invoke-virtual {v2}, Ljava/io/File;->mkdir()Z
  
***************
*** 318,323 ****
--- 353,359 ----
  
      if-nez v5, :cond_0
  
+     .line 527
      const-string v5, "SharedPreferencesImpl"
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 340,349 ****
--- 376,387 ----
  
      invoke-static {v5, v6}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 528
      const/4 v5, 0x0
  
      goto :goto_1
  
+     .line 530
      :cond_0
      invoke-virtual {v2}, Ljava/io/File;->getPath()Ljava/lang/String;
  
***************
*** 353,358 ****
--- 391,397 ----
  
      invoke-static {v5, v6, v7, v7}, Landroid/os/FileUtils;->setPermissions(Ljava/lang/String;III)I
  
+     .line 535
      :try_start_1
      new-instance v4, Ljava/io/FileOutputStream;
  
***************
*** 364,376 ****
--- 403,418 ----
      .restart local v4       #str:Ljava/io/FileOutputStream;
      move-object v3, v4
  
+     .line 538
      .end local v4           #str:Ljava/io/FileOutputStream;
      .restart local v3       #str:Ljava/io/FileOutputStream;
      goto :goto_0
  
+     .line 536
      :catch_1
      move-exception v1
  
+     .line 537
      .local v1, e2:Ljava/io/FileNotFoundException;
      const-string v5, "SharedPreferencesImpl"
  
***************
*** 407,430 ****
--- 449,478 ----
  
      const/4 v3, 0x1
  
+     .line 488
      new-instance v2, Landroid/app/SharedPreferencesImpl$2;
  
      invoke-direct {v2, p0, p1, p2}, Landroid/app/SharedPreferencesImpl$2;-><init>(Landroid/app/SharedPreferencesImpl;Landroid/app/SharedPreferencesImpl$MemoryCommitResult;Ljava/lang/Runnable;)V
  
+     .line 502
      .local v2, writeToDiskRunnable:Ljava/lang/Runnable;
      if-nez p2, :cond_0
  
      move v0, v3
  
+     .line 506
      .local v0, isFromSyncCommit:Z
      :goto_0
      if-eqz v0, :cond_2
  
+     .line 507
      const/4 v1, 0x0
  
+     .line 508
      .local v1, wasEmpty:Z
      monitor-enter p0
  
+     .line 509
      :try_start_0
      iget v5, p0, Landroid/app/SharedPreferencesImpl;->mDiskWritesInFlight:I
  
***************
*** 432,446 ****
--- 480,498 ----
  
      move v1, v3
  
+     .line 510
      :goto_1
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 511
      if-eqz v1, :cond_2
  
+     .line 512
      invoke-interface {v2}, Ljava/lang/Runnable;->run()V
  
+     .line 518
      .end local v1           #wasEmpty:Z
      :goto_2
      return-void
***************
*** 449,454 ****
--- 501,507 ----
      :cond_0
      move v0, v4
  
+     .line 502
      goto :goto_0
  
      .restart local v0       #isFromSyncCommit:Z
***************
*** 456,463 ****
--- 509,518 ----
      :cond_1
      move v1, v4
  
+     .line 509
      goto :goto_1
  
+     .line 510
      :catchall_0
      move-exception v3
  
***************
*** 468,473 ****
--- 523,529 ----
  
      throw v3
  
+     .line 517
      .end local v1           #wasEmpty:Z
      :cond_2
      invoke-static {}, Landroid/app/QueuedWork;->singleThreadExecutor()Ljava/util/concurrent/ExecutorService;
***************
*** 487,513 ****
--- 543,576 ----
  
      const/4 v1, 0x0
  
+     .line 151
      monitor-enter p0
  
+     .line 152
      :try_start_0
      iget v3, p0, Landroid/app/SharedPreferencesImpl;->mDiskWritesInFlight:I
  
      if-lez v3, :cond_0
  
+     .line 155
      monitor-exit p0
  
+     .line 163
      :goto_0
      return v1
  
+     .line 157
      :cond_0
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 158
      new-instance v0, Landroid/os/FileUtils$FileStatus;
  
      invoke-direct {v0}, Landroid/os/FileUtils$FileStatus;-><init>()V
  
+     .line 159
      .local v0, stat:Landroid/os/FileUtils$FileStatus;
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
***************
*** 523,530 ****
--- 586,595 ----
  
      move v1, v2
  
+     .line 160
      goto :goto_0
  
+     .line 157
      .end local v0           #stat:Landroid/os/FileUtils$FileStatus;
      :catchall_0
      move-exception v1
***************
*** 536,545 ****
--- 601,612 ----
  
      throw v1
  
+     .line 162
      .restart local v0       #stat:Landroid/os/FileUtils$FileStatus;
      :cond_1
      monitor-enter p0
  
+     .line 163
      :try_start_2
      iget-wide v3, p0, Landroid/app/SharedPreferencesImpl;->mStatTimestamp:J
  
***************
*** 565,570 ****
--- 632,638 ----
  
      goto :goto_0
  
+     .line 164
      :catchall_1
      move-exception v1
  
***************
*** 579,591 ****
--- 647,662 ----
      .locals 7
  
      .prologue
+     .line 94
      iget-boolean v4, p0, Landroid/app/SharedPreferencesImpl;->mLoaded:Z
  
      if-eqz v4, :cond_0
  
+     .line 132
      :goto_0
      return-void
  
+     .line 97
      :cond_0
      iget-object v4, p0, Landroid/app/SharedPreferencesImpl;->mBackupFile:Ljava/io/File;
  
***************
*** 595,610 ****
--- 666,684 ----
  
      if-eqz v4, :cond_1
  
+     .line 98
      iget-object v4, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
      invoke-virtual {v4}, Ljava/io/File;->delete()Z
  
+     .line 99
      iget-object v4, p0, Landroid/app/SharedPreferencesImpl;->mBackupFile:Ljava/io/File;
  
      iget-object v5, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
      invoke-virtual {v4, v5}, Ljava/io/File;->renameTo(Ljava/io/File;)Z
  
+     .line 103
      :cond_1
      iget-object v4, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
***************
*** 622,627 ****
--- 696,702 ----
  
      if-nez v4, :cond_2
  
+     .line 104
      const-string v4, "SharedPreferencesImpl"
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 652,665 ****
--- 727,743 ----
  
      invoke-static {v4, v5}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 107
      :cond_2
      const/4 v1, 0x0
  
+     .line 108
      .local v1, map:Ljava/util/Map;
      new-instance v2, Landroid/os/FileUtils$FileStatus;
  
      invoke-direct {v2}, Landroid/os/FileUtils$FileStatus;-><init>()V
  
+     .line 109
      .local v2, stat:Landroid/os/FileUtils$FileStatus;
      iget-object v4, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
***************
*** 681,686 ****
--- 759,765 ----
  
      if-eqz v4, :cond_3
  
+     .line 111
      :try_start_0
      new-instance v3, Ljava/io/BufferedInputStream;
  
***************
*** 694,710 ****
--- 773,792 ----
  
      invoke-direct {v3, v4, v5}, Ljava/io/BufferedInputStream;-><init>(Ljava/io/InputStream;I)V
  
+     .line 113
      .local v3, str:Ljava/io/BufferedInputStream;
      invoke-static {v3}, Lcom/android/internal/util/XmlUtils;->readMapXml(Ljava/io/InputStream;)Ljava/util/HashMap;
  
      move-result-object v1
  
+     .line 114
      invoke-virtual {v3}, Ljava/io/BufferedInputStream;->close()V
      :try_end_0
      .catch Lorg/xmlpull/v1/XmlPullParserException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/io/FileNotFoundException; {:try_start_0 .. :try_end_0} :catch_1
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_2
  
+     .line 123
      .end local v3           #str:Ljava/io/BufferedInputStream;
      :cond_3
      :goto_1
***************
*** 712,737 ****
--- 794,826 ----
  
      iput-boolean v4, p0, Landroid/app/SharedPreferencesImpl;->mLoaded:Z
  
+     .line 124
      if-eqz v1, :cond_4
  
+     .line 125
      iput-object v1, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
+     .line 126
      iget-wide v4, v2, Landroid/os/FileUtils$FileStatus;->mtime:J
  
      iput-wide v4, p0, Landroid/app/SharedPreferencesImpl;->mStatTimestamp:J
  
+     .line 127
      iget-wide v4, v2, Landroid/os/FileUtils$FileStatus;->size:J
  
      iput-wide v4, p0, Landroid/app/SharedPreferencesImpl;->mStatSize:J
  
+     .line 131
      :goto_2
      invoke-virtual {p0}, Ljava/lang/Object;->notifyAll()V
  
      goto/16 :goto_0
  
+     .line 115
      :catch_0
      move-exception v0
  
+     .line 116
      .local v0, e:Lorg/xmlpull/v1/XmlPullParserException;
      const-string v4, "SharedPreferencesImpl"
  
***************
*** 741,750 ****
--- 830,841 ----
  
      goto :goto_1
  
+     .line 117
      .end local v0           #e:Lorg/xmlpull/v1/XmlPullParserException;
      :catch_1
      move-exception v0
  
+     .line 118
      .local v0, e:Ljava/io/FileNotFoundException;
      const-string v4, "SharedPreferencesImpl"
  
***************
*** 754,763 ****
--- 845,856 ----
  
      goto :goto_1
  
+     .line 119
      .end local v0           #e:Ljava/io/FileNotFoundException;
      :catch_2
      move-exception v0
  
+     .line 120
      .local v0, e:Ljava/io/IOException;
      const-string v4, "SharedPreferencesImpl"
  
***************
*** 767,772 ****
--- 860,866 ----
  
      goto :goto_1
  
+     .line 129
      .end local v0           #e:Ljava/io/IOException;
      :cond_4
      new-instance v4, Ljava/util/HashMap;
***************
*** 783,788 ****
--- 877,883 ----
      .parameter "prefsFile"
  
      .prologue
+     .line 135
      new-instance v0, Ljava/io/File;
  
      new-instance v1, Ljava/lang/StringBuilder;
***************
*** 816,832 ****
--- 911,931 ----
      .locals 2
  
      .prologue
+     .line 81
      monitor-enter p0
  
+     .line 82
      const/4 v0, 0x0
  
      :try_start_0
      iput-boolean v0, p0, Landroid/app/SharedPreferencesImpl;->mLoaded:Z
  
+     .line 83
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 84
      new-instance v0, Landroid/app/SharedPreferencesImpl$1;
  
      const-string v1, "SharedPreferencesImpl-load"
***************
*** 835,842 ****
--- 934,943 ----
  
      invoke-virtual {v0}, Landroid/app/SharedPreferencesImpl$1;->start()V
  
+     .line 91
      return-void
  
+     .line 83
      :catchall_0
      move-exception v0
  
***************
*** 857,862 ****
--- 958,964 ----
  
      const/4 v6, 0x0
  
+     .line 546
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
      invoke-virtual {v3}, Ljava/io/File;->exists()Z
***************
*** 865,879 ****
--- 967,985 ----
  
      if-eqz v3, :cond_2
  
+     .line 547
      iget-boolean v3, p1, Landroid/app/SharedPreferencesImpl$MemoryCommitResult;->changesMade:Z
  
      if-nez v3, :cond_0
  
+     .line 552
      invoke-virtual {p1, v4}, Landroid/app/SharedPreferencesImpl$MemoryCommitResult;->setDiskWriteResult(Z)V
  
+     .line 603
      :goto_0
      return-void
  
+     .line 555
      :cond_0
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mBackupFile:Ljava/io/File;
  
***************
*** 883,888 ****
--- 989,995 ----
  
      if-nez v3, :cond_1
  
+     .line 556
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
      iget-object v4, p0, Landroid/app/SharedPreferencesImpl;->mBackupFile:Ljava/io/File;
***************
*** 893,898 ****
--- 1000,1006 ----
  
      if-nez v3, :cond_2
  
+     .line 557
      const-string v3, "SharedPreferencesImpl"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 929,943 ****
--- 1037,1054 ----
  
      invoke-static {v3, v4}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 559
      invoke-virtual {p1, v6}, Landroid/app/SharedPreferencesImpl$MemoryCommitResult;->setDiskWriteResult(Z)V
  
      goto :goto_0
  
+     .line 563
      :cond_1
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
      invoke-virtual {v3}, Ljava/io/File;->delete()Z
  
+     .line 571
      :cond_2
      :try_start_0
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
***************
*** 946,954 ****
--- 1057,1067 ----
  
      move-result-object v2
  
+     .line 572
      .local v2, str:Ljava/io/FileOutputStream;
      if-nez v2, :cond_4
  
+     .line 573
      const/4 v3, 0x0
  
      invoke-virtual {p1, v3}, Landroid/app/SharedPreferencesImpl$MemoryCommitResult;->setDiskWriteResult(Z)V
***************
*** 958,974 ****
  
      goto :goto_0
  
      .end local v2           #str:Ljava/io/FileOutputStream;
      :catch_0
      move-exception v0
  
      .local v0, e:Lorg/xmlpull/v1/XmlPullParserException;
      const-string v3, "SharedPreferencesImpl"
  
!     const-string v4, "writeToFile: Got exception:"
  
      invoke-static {v3, v4, v0}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
      .end local v0           #e:Lorg/xmlpull/v1/XmlPullParserException;
      :goto_1
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
--- 1071,1090 ----
  
      goto :goto_0
  
+     .line 591
      .end local v2           #str:Ljava/io/FileOutputStream;
      :catch_0
      move-exception v0
  
+     .line 592
      .local v0, e:Lorg/xmlpull/v1/XmlPullParserException;
      const-string v3, "SharedPreferencesImpl"
  
!     const-string/jumbo v4, "writeToFile: Got exception:"
  
      invoke-static {v3, v4, v0}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 597
      .end local v0           #e:Lorg/xmlpull/v1/XmlPullParserException;
      :goto_1
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
***************
*** 979,984 ****
--- 1095,1101 ----
  
      if-eqz v3, :cond_3
  
+     .line 598
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
      invoke-virtual {v3}, Ljava/io/File;->delete()Z
***************
*** 987,992 ****
--- 1104,1110 ----
  
      if-nez v3, :cond_3
  
+     .line 599
      const-string v3, "SharedPreferencesImpl"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 1011,1021 ****
--- 1129,1141 ----
  
      invoke-static {v3, v4}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 602
      :cond_3
      invoke-virtual {p1, v6}, Landroid/app/SharedPreferencesImpl$MemoryCommitResult;->setDiskWriteResult(Z)V
  
      goto/16 :goto_0
  
+     .line 576
      .restart local v2       #str:Ljava/io/FileOutputStream;
      :cond_4
      :try_start_1
***************
*** 1023,1032 ****
--- 1143,1155 ----
  
      invoke-static {v3, v2}, Lcom/android/internal/util/XmlUtils;->writeMapXml(Ljava/util/Map;Ljava/io/OutputStream;)V
  
+     .line 577
      invoke-static {v2}, Landroid/os/FileUtils;->sync(Ljava/io/FileOutputStream;)Z
  
+     .line 578
      invoke-virtual {v2}, Ljava/io/FileOutputStream;->close()V
  
+     .line 579
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
      invoke-virtual {v3}, Ljava/io/File;->getPath()Ljava/lang/String;
***************
*** 1039,1048 ****
--- 1162,1173 ----
  
      invoke-static {v3, v4, v5}, Landroid/app/ContextImpl;->setFilePermissionsFromMode(Ljava/lang/String;II)V
  
+     .line 580
      new-instance v1, Landroid/os/FileUtils$FileStatus;
  
      invoke-direct {v1}, Landroid/os/FileUtils$FileStatus;-><init>()V
  
+     .line 581
      .local v1, stat:Landroid/os/FileUtils$FileStatus;
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mFile:Ljava/io/File;
  
***************
*** 1056,1085 ****
--- 1181,1216 ----
  
      if-eqz v3, :cond_5
  
+     .line 582
      monitor-enter p0
      :try_end_1
      .catch Lorg/xmlpull/v1/XmlPullParserException; {:try_start_1 .. :try_end_1} :catch_0
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_1
  
+     .line 583
      :try_start_2
      iget-wide v3, v1, Landroid/os/FileUtils$FileStatus;->mtime:J
  
      iput-wide v3, p0, Landroid/app/SharedPreferencesImpl;->mStatTimestamp:J
  
+     .line 584
      iget-wide v3, v1, Landroid/os/FileUtils$FileStatus;->size:J
  
      iput-wide v3, p0, Landroid/app/SharedPreferencesImpl;->mStatSize:J
  
+     .line 585
      monitor-exit p0
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
  
+     .line 588
      :cond_5
      :try_start_3
      iget-object v3, p0, Landroid/app/SharedPreferencesImpl;->mBackupFile:Ljava/io/File;
  
      invoke-virtual {v3}, Ljava/io/File;->delete()Z
  
+     .line 589
      const/4 v3, 0x1
  
      invoke-virtual {p1, v3}, Landroid/app/SharedPreferencesImpl$MemoryCommitResult;->setDiskWriteResult(Z)V
***************
*** 1089,1108 ****
  
      goto/16 :goto_0
  
      .end local v1           #stat:Landroid/os/FileUtils$FileStatus;
      .end local v2           #str:Ljava/io/FileOutputStream;
      :catch_1
      move-exception v0
  
      .local v0, e:Ljava/io/IOException;
      const-string v3, "SharedPreferencesImpl"
  
!     const-string v4, "writeToFile: Got exception:"
  
      invoke-static {v3, v4, v0}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
      goto :goto_1
  
      .end local v0           #e:Ljava/io/IOException;
      .restart local v1       #stat:Landroid/os/FileUtils$FileStatus;
      .restart local v2       #str:Ljava/io/FileOutputStream;
--- 1220,1242 ----
  
      goto/16 :goto_0
  
+     .line 593
      .end local v1           #stat:Landroid/os/FileUtils$FileStatus;
      .end local v2           #str:Ljava/io/FileOutputStream;
      :catch_1
      move-exception v0
  
+     .line 594
      .local v0, e:Ljava/io/IOException;
      const-string v3, "SharedPreferencesImpl"
  
!     const-string/jumbo v4, "writeToFile: Got exception:"
  
      invoke-static {v3, v4, v0}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
      goto :goto_1
  
+     .line 585
      .end local v0           #e:Ljava/io/IOException;
      .restart local v1       #stat:Landroid/os/FileUtils$FileStatus;
      .restart local v2       #str:Ljava/io/FileOutputStream;
***************
*** 1128,1138 ****
--- 1262,1275 ----
      .parameter "key"
  
      .prologue
+     .line 248
      monitor-enter p0
  
+     .line 249
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->awaitLoadedLocked()V
  
+     .line 250
      iget-object v0, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
      invoke-interface {v0, p1}, Ljava/util/Map;->containsKey(Ljava/lang/Object;)Z
***************
*** 1143,1148 ****
--- 1280,1286 ----
  
      return v0
  
+     .line 251
      :catchall_0
      move-exception v0
  
***************
*** 1157,1177 ****
--- 1295,1320 ----
      .locals 1
  
      .prologue
+     .line 262
      monitor-enter p0
  
+     .line 263
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->awaitLoadedLocked()V
  
+     .line 264
      monitor-exit p0
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 266
      new-instance v0, Landroid/app/SharedPreferencesImpl$EditorImpl;
  
      invoke-direct {v0, p0}, Landroid/app/SharedPreferencesImpl$EditorImpl;-><init>(Landroid/app/SharedPreferencesImpl;)V
  
      return-object v0
  
+     .line 264
      :catchall_0
      move-exception v0
  
***************
*** 1196,1206 ****
--- 1339,1352 ----
      .end annotation
  
      .prologue
+     .line 195
      monitor-enter p0
  
+     .line 196
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->awaitLoadedLocked()V
  
+     .line 198
      new-instance v0, Ljava/util/HashMap;
  
      iget-object v1, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
***************
*** 1211,1216 ****
--- 1357,1363 ----
  
      return-object v0
  
+     .line 199
      :catchall_0
      move-exception v0
  
***************
*** 1227,1237 ****
--- 1374,1387 ----
      .parameter "defValue"
  
      .prologue
+     .line 240
      monitor-enter p0
  
+     .line 241
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->awaitLoadedLocked()V
  
+     .line 242
      iget-object v1, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
      invoke-interface {v1, p1}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1240,1245 ****
--- 1390,1396 ----
  
      check-cast v0, Ljava/lang/Boolean;
  
+     .line 243
      .local v0, v:Ljava/lang/Boolean;
      if-eqz v0, :cond_0
  
***************
*** 1253,1258 ****
--- 1404,1410 ----
  
      return p2
  
+     .line 244
      .end local v0           #v:Ljava/lang/Boolean;
      :catchall_0
      move-exception v1
***************
*** 1270,1280 ****
--- 1422,1435 ----
      .parameter "defValue"
  
      .prologue
+     .line 233
      monitor-enter p0
  
+     .line 234
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->awaitLoadedLocked()V
  
+     .line 235
      iget-object v1, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
      invoke-interface {v1, p1}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1283,1288 ****
--- 1438,1444 ----
  
      check-cast v0, Ljava/lang/Float;
  
+     .line 236
      .local v0, v:Ljava/lang/Float;
      if-eqz v0, :cond_0
  
***************
*** 1296,1301 ****
--- 1452,1458 ----
  
      return p2
  
+     .line 237
      .end local v0           #v:Ljava/lang/Float;
      :catchall_0
      move-exception v1
***************
*** 1313,1323 ****
--- 1470,1483 ----
      .parameter "defValue"
  
      .prologue
+     .line 219
      monitor-enter p0
  
+     .line 220
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->awaitLoadedLocked()V
  
+     .line 221
      iget-object v1, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
      invoke-interface {v1, p1}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1326,1331 ****
--- 1486,1492 ----
  
      check-cast v0, Ljava/lang/Integer;
  
+     .line 222
      .local v0, v:Ljava/lang/Integer;
      if-eqz v0, :cond_0
  
***************
*** 1339,1344 ****
--- 1500,1506 ----
  
      return p2
  
+     .line 223
      .end local v0           #v:Ljava/lang/Integer;
      :catchall_0
      move-exception v1
***************
*** 1356,1366 ****
--- 1518,1531 ----
      .parameter "defValue"
  
      .prologue
+     .line 226
      monitor-enter p0
  
+     .line 227
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->awaitLoadedLocked()V
  
+     .line 228
      iget-object v1, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
      invoke-interface {v1, p1}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1369,1374 ****
--- 1534,1540 ----
  
      check-cast v0, Ljava/lang/Long;
  
+     .line 229
      .local v0, v:Ljava/lang/Long;
      if-eqz v0, :cond_0
  
***************
*** 1382,1387 ****
--- 1548,1554 ----
  
      return-wide p2
  
+     .line 230
      .end local v0           #v:Ljava/lang/Long;
      :catchall_0
      move-exception v1
***************
*** 1399,1409 ****
--- 1566,1579 ----
      .parameter "defValue"
  
      .prologue
+     .line 203
      monitor-enter p0
  
+     .line 204
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->awaitLoadedLocked()V
  
+     .line 205
      iget-object v1, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
      invoke-interface {v1, p1}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1412,1417 ****
--- 1582,1588 ----
  
      check-cast v0, Ljava/lang/String;
  
+     .line 206
      .local v0, v:Ljava/lang/String;
      if-eqz v0, :cond_0
  
***************
*** 1427,1432 ****
--- 1598,1604 ----
  
      goto :goto_0
  
+     .line 207
      .end local v0           #v:Ljava/lang/String;
      :catchall_0
      move-exception v1
***************
*** 1458,1469 ****
--- 1630,1644 ----
      .end annotation
  
      .prologue
+     .line 211
      .local p2, defValues:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
      monitor-enter p0
  
+     .line 212
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->awaitLoadedLocked()V
  
+     .line 213
      iget-object v1, p0, Landroid/app/SharedPreferencesImpl;->mMap:Ljava/util/Map;
  
      invoke-interface {v1, p1}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 1472,1477 ****
--- 1647,1653 ----
  
      check-cast v0, Ljava/util/Set;
  
+     .line 214
      .local v0, v:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
      if-eqz v0, :cond_0
  
***************
*** 1487,1492 ****
--- 1663,1669 ----
  
      goto :goto_0
  
+     .line 215
      .end local v0           #v:Ljava/util/Set;,"Ljava/util/Set<Ljava/lang/String;>;"
      :catchall_0
      move-exception v1
***************
*** 1503,1510 ****
--- 1680,1689 ----
      .parameter "listener"
  
      .prologue
+     .line 168
      monitor-enter p0
  
+     .line 169
      :try_start_0
      iget-object v0, p0, Landroid/app/SharedPreferencesImpl;->mListeners:Ljava/util/WeakHashMap;
  
***************
*** 1512,1521 ****
--- 1691,1703 ----
  
      invoke-virtual {v0, p1, v1}, Ljava/util/WeakHashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 170
      monitor-exit p0
  
+     .line 171
      return-void
  
+     .line 170
      :catchall_0
      move-exception v0
  
***************
*** 1530,1537 ****
--- 1712,1721 ----
      .locals 1
  
      .prologue
+     .line 139
      monitor-enter p0
  
+     .line 141
      :try_start_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->hasFileChangedUnexpectedly()Z
  
***************
*** 1539,1552 ****
--- 1723,1740 ----
  
      if-nez v0, :cond_0
  
+     .line 142
      monitor-exit p0
  
+     .line 146
      :goto_0
      return-void
  
+     .line 144
      :cond_0
      invoke-direct {p0}, Landroid/app/SharedPreferencesImpl;->startLoadFromDisk()V
  
+     .line 145
      monitor-exit p0
  
      goto :goto_0
***************
*** 1566,1582 ****
--- 1754,1775 ----
      .parameter "listener"
  
      .prologue
+     .line 174
      monitor-enter p0
  
+     .line 175
      :try_start_0
      iget-object v0, p0, Landroid/app/SharedPreferencesImpl;->mListeners:Ljava/util/WeakHashMap;
  
      invoke-virtual {v0, p1}, Ljava/util/WeakHashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 176
      monitor-exit p0
  
+     .line 177
      return-void
  
+     .line 176
      :catchall_0
      move-exception v0
  
