*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/WallpaperManager$1.smali	2012-03-23 19:36:39.196395729 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/app/WallpaperManager$1.smali.original	2012-03-23 19:34:45.285318563 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .parameter
  
      .prologue
+     .line 629
      iput-object p1, p0, Landroid/app/WallpaperManager$1;->this$0:Landroid/app/WallpaperManager;
  
      iput-object p2, p0, Landroid/app/WallpaperManager$1;->val$fWindowToken:Landroid/os/IBinder;
***************
*** 55,60 ****
--- 56,62 ----
      .locals 6
  
      .prologue
+     .line 633
      :try_start_0
      iget-object v0, p0, Landroid/app/WallpaperManager$1;->this$0:Landroid/app/WallpaperManager;
  
***************
*** 96,109 ****
--- 98,114 ----
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_1
      .catch Ljava/lang/IllegalArgumentException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 642
      :goto_0
      return-void
  
+     .line 638
      :catch_0
      move-exception v0
  
      goto :goto_0
  
+     .line 636
      :catch_1
      move-exception v0
  
