*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/ApplicationInfo.smali	2012-03-23 19:36:29.705222658 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/ApplicationInfo.smali.original	2012-03-23 19:34:45.144315993 -0500
***************
*** 130,135 ****
--- 130,136 ----
      .locals 1
  
      .prologue
+     .line 564
      new-instance v0, Landroid/content/pm/ApplicationInfo$1;
  
      invoke-direct {v0}, Landroid/content/pm/ApplicationInfo$1;-><init>()V
***************
*** 145,172 ****
--- 146,183 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 495
      invoke-direct {p0}, Landroid/content/pm/PackageItemInfo;-><init>()V
  
+     .line 98
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->uiOptions:I
  
+     .line 328
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->flags:I
  
+     .line 336
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->requiresSmallestWidthDp:I
  
+     .line 344
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->compatibleWidthLimitDp:I
  
+     .line 352
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->largestWidthLimitDp:I
  
+     .line 415
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/content/pm/ApplicationInfo;->enabled:Z
  
+     .line 421
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->enabledSetting:I
  
+     .line 427
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->installLocation:I
  
+     .line 496
      return-void
  .end method
  
***************
*** 177,296 ****
--- 188,340 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 499
      invoke-direct {p0, p1}, Landroid/content/pm/PackageItemInfo;-><init>(Landroid/content/pm/PackageItemInfo;)V
  
+     .line 98
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->uiOptions:I
  
+     .line 328
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->flags:I
  
+     .line 336
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->requiresSmallestWidthDp:I
  
+     .line 344
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->compatibleWidthLimitDp:I
  
+     .line 352
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->largestWidthLimitDp:I
  
+     .line 415
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Landroid/content/pm/ApplicationInfo;->enabled:Z
  
+     .line 421
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->enabledSetting:I
  
+     .line 427
      const/4 v0, -0x1
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->installLocation:I
  
+     .line 500
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->taskAffinity:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->taskAffinity:Ljava/lang/String;
  
+     .line 501
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->permission:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->permission:Ljava/lang/String;
  
+     .line 502
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->processName:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->processName:Ljava/lang/String;
  
+     .line 503
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->className:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->className:Ljava/lang/String;
  
+     .line 504
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->theme:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->theme:I
  
+     .line 505
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->flags:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->flags:I
  
+     .line 506
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->requiresSmallestWidthDp:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->requiresSmallestWidthDp:I
  
+     .line 507
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->compatibleWidthLimitDp:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->compatibleWidthLimitDp:I
  
+     .line 508
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->largestWidthLimitDp:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->largestWidthLimitDp:I
  
+     .line 509
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->sourceDir:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->sourceDir:Ljava/lang/String;
  
+     .line 510
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->publicSourceDir:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->publicSourceDir:Ljava/lang/String;
  
+     .line 511
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->nativeLibraryDir:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->nativeLibraryDir:Ljava/lang/String;
  
+     .line 512
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->resourceDirs:[Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->resourceDirs:[Ljava/lang/String;
  
+     .line 513
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->sharedLibraryFiles:[Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->sharedLibraryFiles:[Ljava/lang/String;
  
+     .line 514
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->dataDir:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->dataDir:Ljava/lang/String;
  
+     .line 515
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->uid:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->uid:I
  
+     .line 516
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->targetSdkVersion:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->targetSdkVersion:I
  
+     .line 517
      iget-boolean v0, p1, Landroid/content/pm/ApplicationInfo;->enabled:Z
  
      iput-boolean v0, p0, Landroid/content/pm/ApplicationInfo;->enabled:Z
  
+     .line 518
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->enabledSetting:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->enabledSetting:I
  
+     .line 519
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->installLocation:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->installLocation:I
  
+     .line 520
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->manageSpaceActivityName:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->manageSpaceActivityName:Ljava/lang/String;
  
+     .line 521
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->descriptionRes:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->descriptionRes:I
  
+     .line 522
      iget v0, p1, Landroid/content/pm/ApplicationInfo;->uiOptions:I
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->uiOptions:I
  
+     .line 523
      return-void
  .end method
  
***************
*** 303,430 ****
--- 347,501 ----
  
      const/4 v1, 0x0
  
+     .line 575
      invoke-direct {p0, p1}, Landroid/content/pm/PackageItemInfo;-><init>(Landroid/os/Parcel;)V
  
+     .line 98
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->uiOptions:I
  
+     .line 328
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->flags:I
  
+     .line 336
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->requiresSmallestWidthDp:I
  
+     .line 344
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->compatibleWidthLimitDp:I
  
+     .line 352
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->largestWidthLimitDp:I
  
+     .line 415
      iput-boolean v0, p0, Landroid/content/pm/ApplicationInfo;->enabled:Z
  
+     .line 421
      iput v1, p0, Landroid/content/pm/ApplicationInfo;->enabledSetting:I
  
+     .line 427
      const/4 v2, -0x1
  
      iput v2, p0, Landroid/content/pm/ApplicationInfo;->installLocation:I
  
+     .line 576
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->taskAffinity:Ljava/lang/String;
  
+     .line 577
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->permission:Ljava/lang/String;
  
+     .line 578
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->processName:Ljava/lang/String;
  
+     .line 579
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->className:Ljava/lang/String;
  
+     .line 580
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      iput v2, p0, Landroid/content/pm/ApplicationInfo;->theme:I
  
+     .line 581
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      iput v2, p0, Landroid/content/pm/ApplicationInfo;->flags:I
  
+     .line 582
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      iput v2, p0, Landroid/content/pm/ApplicationInfo;->requiresSmallestWidthDp:I
  
+     .line 583
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      iput v2, p0, Landroid/content/pm/ApplicationInfo;->compatibleWidthLimitDp:I
  
+     .line 584
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      iput v2, p0, Landroid/content/pm/ApplicationInfo;->largestWidthLimitDp:I
  
+     .line 585
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->sourceDir:Ljava/lang/String;
  
+     .line 586
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->publicSourceDir:Ljava/lang/String;
  
+     .line 587
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->nativeLibraryDir:Ljava/lang/String;
  
+     .line 588
      invoke-virtual {p1}, Landroid/os/Parcel;->readStringArray()[Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->resourceDirs:[Ljava/lang/String;
  
+     .line 589
      invoke-virtual {p1}, Landroid/os/Parcel;->readStringArray()[Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->sharedLibraryFiles:[Ljava/lang/String;
  
+     .line 590
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/ApplicationInfo;->dataDir:Ljava/lang/String;
  
+     .line 591
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      iput v2, p0, Landroid/content/pm/ApplicationInfo;->uid:I
  
+     .line 592
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
  
      iput v2, p0, Landroid/content/pm/ApplicationInfo;->targetSdkVersion:I
  
+     .line 593
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v2
***************
*** 434,480 ****
--- 505,559 ----
      :goto_0
      iput-boolean v0, p0, Landroid/content/pm/ApplicationInfo;->enabled:Z
  
+     .line 594
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->enabledSetting:I
  
+     .line 595
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->installLocation:I
  
+     .line 596
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->manageSpaceActivityName:Ljava/lang/String;
  
+     .line 597
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo;->backupAgentName:Ljava/lang/String;
  
+     .line 598
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->descriptionRes:I
  
+     .line 599
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->uiOptions:I
  
+     .line 600
      return-void
  
      :cond_0
      move v0, v1
  
+     .line 593
      goto :goto_0
  .end method
  
***************
*** 484,489 ****
--- 563,569 ----
      .parameter "x1"
  
      .prologue
+     .line 34
      invoke-direct {p0, p1}, Landroid/content/pm/ApplicationInfo;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 498,503 ****
--- 578,584 ----
  
      const/4 v2, 0x0
  
+     .line 648
      :try_start_0
      iget-object v3, p0, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
  
***************
*** 511,527 ****
--- 592,612 ----
  
      if-nez v3, :cond_0
  
+     .line 650
      :goto_0
      return v1
  
      :cond_0
      move v1, v2
  
+     .line 648
      goto :goto_0
  
+     .line 649
      :catch_0
      move-exception v0
  
+     .line 650
      .local v0, ex:Landroid/content/pm/PackageManager$NameNotFoundException;
      goto :goto_0
  .end method
***************
*** 532,537 ****
--- 617,623 ----
      .locals 1
  
      .prologue
+     .line 533
      const/4 v0, 0x0
  
      return v0
***************
*** 541,546 ****
--- 627,633 ----
      .locals 2
  
      .prologue
+     .line 629
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->flags:I
  
      const v1, 0x83e00
***************
*** 549,554 ****
--- 636,642 ----
  
      iput v0, p0, Landroid/content/pm/ApplicationInfo;->flags:I
  
+     .line 632
      return-void
  .end method
  
***************
*** 558,569 ****
--- 646,660 ----
      .parameter "prefix"
  
      .prologue
+     .line 430
      invoke-super {p0, p1, p2}, Landroid/content/pm/PackageItemInfo;->dumpFront(Landroid/util/Printer;Ljava/lang/String;)V
  
+     .line 431
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->className:Ljava/lang/String;
  
      if-eqz v0, :cond_0
  
+     .line 432
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 590,600 ****
--- 681,693 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 434
      :cond_0
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->permission:Ljava/lang/String;
  
      if-eqz v0, :cond_1
  
+     .line 435
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 603,609 ****
  
      move-result-object v0
  
!     const-string v1, "permission="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 696,702 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "permission="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 621,626 ****
--- 714,720 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 437
      :cond_1
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 630,636 ****
  
      move-result-object v0
  
!     const-string v1, "processName="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 724,730 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "processName="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 648,653 ****
--- 742,748 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 438
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 656,662 ****
  
      move-result-object v0
  
!     const-string v1, "taskAffinity="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 751,757 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "taskAffinity="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 674,679 ****
--- 769,775 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 439
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 682,688 ****
  
      move-result-object v0
  
!     const-string v1, "uid="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 778,784 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "uid="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 732,737 ****
--- 828,834 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 441
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 740,746 ****
  
      move-result-object v0
  
!     const-string v1, "requiresSmallestWidthDp="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 837,843 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "requiresSmallestWidthDp="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 782,787 ****
--- 879,885 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 444
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 790,796 ****
  
      move-result-object v0
  
!     const-string v1, "sourceDir="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 888,894 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "sourceDir="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 808,821 ****
--- 906,922 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 445
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->sourceDir:Ljava/lang/String;
  
      if-nez v0, :cond_8
  
+     .line 446
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->publicSourceDir:Ljava/lang/String;
  
      if-eqz v0, :cond_2
  
+     .line 447
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 824,830 ****
  
      move-result-object v0
  
!     const-string v1, "publicSourceDir="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 925,931 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "publicSourceDir="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 842,853 ****
--- 943,956 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 452
      :cond_2
      :goto_0
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->resourceDirs:[Ljava/lang/String;
  
      if-eqz v0, :cond_3
  
+     .line 453
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 856,862 ****
  
      move-result-object v0
  
!     const-string v1, "resourceDirs="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 959,965 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "resourceDirs="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 874,879 ****
--- 977,983 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 455
      :cond_3
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 901,910 ****
--- 1005,1016 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 456
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->sharedLibraryFiles:[Ljava/lang/String;
  
      if-eqz v0, :cond_4
  
+     .line 457
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 913,919 ****
  
      move-result-object v0
  
!     const-string v1, "sharedLibraryFiles="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1019,1025 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "sharedLibraryFiles="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 931,936 ****
--- 1037,1043 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 459
      :cond_4
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 970,979 ****
--- 1077,1088 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 460
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->manageSpaceActivityName:Ljava/lang/String;
  
      if-eqz v0, :cond_5
  
+     .line 461
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 982,988 ****
  
      move-result-object v0
  
!     const-string v1, "manageSpaceActivityName="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1091,1097 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "manageSpaceActivityName="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1000,1010 ****
--- 1109,1121 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 463
      :cond_5
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->descriptionRes:I
  
      if-eqz v0, :cond_6
  
+     .line 464
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1035,1045 ****
--- 1146,1158 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 466
      :cond_6
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->uiOptions:I
  
      if-eqz v0, :cond_7
  
+     .line 467
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1048,1054 ****
  
      move-result-object v0
  
!     const-string v1, "uiOptions=0x"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1161,1167 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "uiOptions=0x"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1070,1080 ****
--- 1183,1196 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 469
      :cond_7
      invoke-super {p0, p1, p2}, Landroid/content/pm/PackageItemInfo;->dumpBack(Landroid/util/Printer;Ljava/lang/String;)V
  
+     .line 470
      return-void
  
+     .line 449
      :cond_8
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->sourceDir:Ljava/lang/String;
  
***************
*** 1086,1091 ****
--- 1202,1208 ----
  
      if-nez v0, :cond_2
  
+     .line 450
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1094,1100 ****
  
      move-result-object v0
  
!     const-string v1, "publicSourceDir="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1211,1217 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "publicSourceDir="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1119,1124 ****
--- 1236,1242 ----
      .locals 0
  
      .prologue
+     .line 658
      return-object p0
  .end method
  
***************
*** 1127,1132 ****
--- 1245,1251 ----
      .parameter "pm"
  
      .prologue
+     .line 638
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->flags:I
  
      const/high16 v1, 0x4
***************
*** 1141,1146 ****
--- 1260,1266 ----
  
      if-eqz v0, :cond_0
  
+     .line 640
      invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
  
      move-result-object v0
***************
*** 1151,1156 ****
--- 1271,1277 ----
  
      move-result-object v0
  
+     .line 643
      :goto_0
      return-object v0
  
***************
*** 1167,1176 ****
--- 1288,1299 ----
      .parameter "pm"
  
      .prologue
+     .line 614
      iget v1, p0, Landroid/content/pm/ApplicationInfo;->descriptionRes:I
  
      if-eqz v1, :cond_0
  
+     .line 615
      iget-object v1, p0, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
  
      iget v2, p0, Landroid/content/pm/ApplicationInfo;->descriptionRes:I
***************
*** 1179,1187 ****
--- 1302,1312 ----
  
      move-result-object v0
  
+     .line 616
      .local v0, label:Ljava/lang/CharSequence;
      if-eqz v0, :cond_0
  
+     .line 620
      .end local v0           #label:Ljava/lang/CharSequence;
      :goto_0
      return-object v0
***************
*** 1196,1201 ****
--- 1321,1327 ----
      .locals 2
  
      .prologue
+     .line 527
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 1230,1236 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 1356,1362 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 1249,1324 ****
--- 1375,1469 ----
      .parameter "parcelableFlags"
  
      .prologue
+     .line 537
      invoke-super {p0, p1, p2}, Landroid/content/pm/PackageItemInfo;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 538
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->taskAffinity:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 539
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->permission:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 540
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->processName:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 541
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->className:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 542
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->theme:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 543
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->flags:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 544
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->requiresSmallestWidthDp:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 545
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->compatibleWidthLimitDp:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 546
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->largestWidthLimitDp:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 547
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->sourceDir:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 548
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->publicSourceDir:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 549
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->nativeLibraryDir:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 550
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->resourceDirs:[Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStringArray([Ljava/lang/String;)V
  
+     .line 551
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->sharedLibraryFiles:[Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeStringArray([Ljava/lang/String;)V
  
+     .line 552
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->dataDir:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 553
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->uid:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 554
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->targetSdkVersion:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 555
      iget-boolean v0, p0, Landroid/content/pm/ApplicationInfo;->enabled:Z
  
      if-eqz v0, :cond_0
***************
*** 1328,1359 ****
--- 1473,1512 ----
      :goto_0
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 556
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->enabledSetting:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 557
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->installLocation:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 558
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->manageSpaceActivityName:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 559
      iget-object v0, p0, Landroid/content/pm/ApplicationInfo;->backupAgentName:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 560
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->descriptionRes:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 561
      iget v0, p0, Landroid/content/pm/ApplicationInfo;->uiOptions:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 562
      return-void
  
+     .line 555
      :cond_0
      const/4 v0, 0x0
  
