*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/ApplicationInfo$DisplayNameComparator.smali	2012-03-23 19:36:30.470236608 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/ApplicationInfo$DisplayNameComparator.smali.original	2012-03-23 19:34:45.151316121 -0500
***************
*** 39,54 ****
--- 39,58 ----
      .parameter "pm"
  
      .prologue
+     .line 474
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 491
      invoke-static {}, Ljava/text/Collator;->getInstance()Ljava/text/Collator;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/pm/ApplicationInfo$DisplayNameComparator;->sCollator:Ljava/text/Collator;
  
+     .line 475
      iput-object p1, p0, Landroid/content/pm/ApplicationInfo$DisplayNameComparator;->mPM:Landroid/content/pm/PackageManager;
  
+     .line 476
      return-void
  .end method
  
***************
*** 60,76 ****
--- 64,84 ----
      .parameter "ab"
  
      .prologue
+     .line 479
      iget-object v2, p0, Landroid/content/pm/ApplicationInfo$DisplayNameComparator;->mPM:Landroid/content/pm/PackageManager;
  
      invoke-virtual {v2, p1}, Landroid/content/pm/PackageManager;->getApplicationLabel(Landroid/content/pm/ApplicationInfo;)Ljava/lang/CharSequence;
  
      move-result-object v0
  
+     .line 480
      .local v0, sa:Ljava/lang/CharSequence;
      if-nez v0, :cond_0
  
+     .line 481
      iget-object v0, p1, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
  
+     .line 483
      :cond_0
      iget-object v2, p0, Landroid/content/pm/ApplicationInfo$DisplayNameComparator;->mPM:Landroid/content/pm/PackageManager;
  
***************
*** 78,88 ****
--- 86,99 ----
  
      move-result-object v1
  
+     .line 484
      .local v1, sb:Ljava/lang/CharSequence;
      if-nez v1, :cond_1
  
+     .line 485
      iget-object v1, p2, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
  
+     .line 488
      :cond_1
      iget-object v2, p0, Landroid/content/pm/ApplicationInfo$DisplayNameComparator;->sCollator:Ljava/text/Collator;
  
***************
*** 107,112 ****
--- 118,124 ----
      .parameter "x1"
  
      .prologue
+     .line 472
      check-cast p1, Landroid/content/pm/ApplicationInfo;
  
      .end local p1
