*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/ConfigurationInfo.smali	2012-03-23 19:36:29.061210914 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/ConfigurationInfo.smali.original	2012-03-23 19:34:45.138315883 -0500
***************
*** 42,47 ****
--- 42,48 ----
      .locals 1
  
      .prologue
+     .line 116
      new-instance v0, Landroid/content/pm/ConfigurationInfo$1;
  
      invoke-direct {v0}, Landroid/content/pm/ConfigurationInfo$1;-><init>()V
***************
*** 55,66 ****
--- 56,70 ----
      .locals 1
  
      .prologue
+     .line 83
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 71
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqInputFeatures:I
  
+     .line 84
      return-void
  .end method
  
***************
*** 69,100 ****
--- 73,112 ----
      .parameter "orig"
  
      .prologue
+     .line 86
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 71
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqInputFeatures:I
  
+     .line 87
      iget v0, p1, Landroid/content/pm/ConfigurationInfo;->reqTouchScreen:I
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqTouchScreen:I
  
+     .line 88
      iget v0, p1, Landroid/content/pm/ConfigurationInfo;->reqKeyboardType:I
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqKeyboardType:I
  
+     .line 89
      iget v0, p1, Landroid/content/pm/ConfigurationInfo;->reqNavigation:I
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqNavigation:I
  
+     .line 90
      iget v0, p1, Landroid/content/pm/ConfigurationInfo;->reqInputFeatures:I
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqInputFeatures:I
  
+     .line 91
      iget v0, p1, Landroid/content/pm/ConfigurationInfo;->reqGlEsVersion:I
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqGlEsVersion:I
  
+     .line 92
      return-void
  .end method
  
***************
*** 103,144 ****
--- 115,164 ----
      .parameter "source"
  
      .prologue
+     .line 126
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 71
      const/4 v0, 0x0
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqInputFeatures:I
  
+     .line 127
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqTouchScreen:I
  
+     .line 128
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqKeyboardType:I
  
+     .line 129
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqNavigation:I
  
+     .line 130
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqInputFeatures:I
  
+     .line 131
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ConfigurationInfo;->reqGlEsVersion:I
  
+     .line 132
      return-void
  .end method
  
***************
*** 148,153 ****
--- 168,174 ----
      .parameter "x1"
  
      .prologue
+     .line 27
      invoke-direct {p0, p1}, Landroid/content/pm/ConfigurationInfo;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 159,164 ****
--- 180,186 ----
      .locals 1
  
      .prologue
+     .line 105
      const/4 v0, 0x0
  
      return v0
***************
*** 168,173 ****
--- 190,196 ----
      .locals 4
  
      .prologue
+     .line 141
      iget v2, p0, Landroid/content/pm/ConfigurationInfo;->reqGlEsVersion:I
  
      const/high16 v3, -0x1
***************
*** 176,181 ****
--- 199,205 ----
  
      shr-int/lit8 v0, v2, 0x10
  
+     .line 142
      .local v0, major:I
      iget v2, p0, Landroid/content/pm/ConfigurationInfo;->reqGlEsVersion:I
  
***************
*** 183,188 ****
--- 207,213 ----
  
      and-int v1, v2, v3
  
+     .line 143
      .local v1, minor:I
      new-instance v2, Ljava/lang/StringBuilder;
  
***************
*** 221,226 ****
--- 246,252 ----
      .locals 2
  
      .prologue
+     .line 95
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 303,309 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 329,335 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 322,346 ****
--- 348,378 ----
      .parameter "parcelableFlags"
  
      .prologue
+     .line 109
      iget v0, p0, Landroid/content/pm/ConfigurationInfo;->reqTouchScreen:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 110
      iget v0, p0, Landroid/content/pm/ConfigurationInfo;->reqKeyboardType:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 111
      iget v0, p0, Landroid/content/pm/ConfigurationInfo;->reqNavigation:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 112
      iget v0, p0, Landroid/content/pm/ConfigurationInfo;->reqInputFeatures:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 113
      iget v0, p0, Landroid/content/pm/ConfigurationInfo;->reqGlEsVersion:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 114
      return-void
  .end method
