*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/FeatureInfo.smali	2012-03-23 19:36:29.602220780 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/FeatureInfo.smali.original	2012-03-23 19:34:45.143315975 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .locals 1
  
      .prologue
+     .line 90
      new-instance v0, Landroid/content/pm/FeatureInfo$1;
  
      invoke-direct {v0}, Landroid/content/pm/FeatureInfo$1;-><init>()V
***************
*** 49,56 ****
--- 50,59 ----
      .locals 0
  
      .prologue
+     .line 58
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 59
      return-void
  .end method
  
***************
*** 59,78 ****
--- 62,86 ----
      .parameter "orig"
  
      .prologue
+     .line 61
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 62
      iget-object v0, p1, Landroid/content/pm/FeatureInfo;->name:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/FeatureInfo;->name:Ljava/lang/String;
  
+     .line 63
      iget v0, p1, Landroid/content/pm/FeatureInfo;->reqGlEsVersion:I
  
      iput v0, p0, Landroid/content/pm/FeatureInfo;->reqGlEsVersion:I
  
+     .line 64
      iget v0, p1, Landroid/content/pm/FeatureInfo;->flags:I
  
      iput v0, p0, Landroid/content/pm/FeatureInfo;->flags:I
  
+     .line 65
      return-void
  .end method
  
***************
*** 81,106 ****
--- 89,119 ----
      .parameter "source"
  
      .prologue
+     .line 100
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 101
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/pm/FeatureInfo;->name:Ljava/lang/String;
  
+     .line 102
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/FeatureInfo;->reqGlEsVersion:I
  
+     .line 103
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/FeatureInfo;->flags:I
  
+     .line 104
      return-void
  .end method
  
***************
*** 110,115 ****
--- 123,129 ----
      .parameter "x1"
  
      .prologue
+     .line 28
      invoke-direct {p0, p1}, Landroid/content/pm/FeatureInfo;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 121,126 ****
--- 135,141 ----
      .locals 1
  
      .prologue
+     .line 81
      const/4 v0, 0x0
  
      return v0
***************
*** 130,135 ****
--- 145,151 ----
      .locals 4
  
      .prologue
+     .line 113
      iget v2, p0, Landroid/content/pm/FeatureInfo;->reqGlEsVersion:I
  
      const/high16 v3, -0x1
***************
*** 138,143 ****
--- 154,160 ----
  
      shr-int/lit8 v0, v2, 0x10
  
+     .line 114
      .local v0, major:I
      iget v2, p0, Landroid/content/pm/FeatureInfo;->reqGlEsVersion:I
  
***************
*** 145,150 ****
--- 162,168 ----
  
      and-int v1, v2, v3
  
+     .line 115
      .local v1, minor:I
      new-instance v2, Ljava/lang/StringBuilder;
  
***************
*** 183,192 ****
--- 201,212 ----
      .locals 2
  
      .prologue
+     .line 68
      iget-object v0, p0, Landroid/content/pm/FeatureInfo;->name:Ljava/lang/String;
  
      if-eqz v0, :cond_0
  
+     .line 69
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 237,243 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 257,263 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 247,252 ****
--- 267,273 ----
  
      move-result-object v0
  
+     .line 73
      :goto_0
      return-object v0
  
***************
*** 303,309 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 324,330 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 322,338 ****
--- 343,363 ----
      .parameter "parcelableFlags"
  
      .prologue
+     .line 85
      iget-object v0, p0, Landroid/content/pm/FeatureInfo;->name:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 86
      iget v0, p0, Landroid/content/pm/FeatureInfo;->reqGlEsVersion:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 87
      iget v0, p0, Landroid/content/pm/FeatureInfo;->flags:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 88
      return-void
  .end method
