*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$IntentInfo.smali	2012-03-23 19:36:30.444236134 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$IntentInfo.smali.original	2012-03-23 19:34:45.151316121 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .locals 0
  
      .prologue
+     .line 3452
      invoke-direct {p0}, Landroid/content/IntentFilter;-><init>()V
  
      return-void
