*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$ServiceIntentInfo.smali	2012-03-23 19:36:30.640239708 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$ServiceIntentInfo.smali.original	2012-03-23 19:34:45.151316121 -0500
***************
*** 24,33 ****
--- 24,36 ----
      .parameter "_service"
  
      .prologue
+     .line 3477
      invoke-direct {p0}, Landroid/content/pm/PackageParser$IntentInfo;-><init>()V
  
+     .line 3478
      iput-object p1, p0, Landroid/content/pm/PackageParser$ServiceIntentInfo;->service:Landroid/content/pm/PackageParser$Service;
  
+     .line 3479
      return-void
  .end method
  
***************
*** 37,42 ****
--- 40,46 ----
      .locals 2
  
      .prologue
+     .line 3482
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 75,81 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 79,85 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
