*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$ActivityIntentInfo.smali	2012-03-23 19:36:29.959227290 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$ActivityIntentInfo.smali.original	2012-03-23 19:34:45.147316047 -0500
***************
*** 24,33 ****
--- 24,36 ----
      .parameter "_activity"
  
      .prologue
+     .line 3463
      invoke-direct {p0}, Landroid/content/pm/PackageParser$IntentInfo;-><init>()V
  
+     .line 3464
      iput-object p1, p0, Landroid/content/pm/PackageParser$ActivityIntentInfo;->activity:Landroid/content/pm/PackageParser$Activity;
  
+     .line 3465
      return-void
  .end method
  
***************
*** 37,42 ****
--- 40,46 ----
      .locals 2
  
      .prologue
+     .line 3468
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 75,81 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 79,85 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
