*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageItemInfo$DisplayNameComparator.smali	2012-03-23 19:36:30.727241294 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageItemInfo$DisplayNameComparator.smali.original	2012-03-23 19:34:45.152316139 -0500
***************
*** 39,54 ****
--- 39,58 ----
      .parameter "pm"
  
      .prologue
+     .line 273
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 285
      invoke-static {}, Ljava/text/Collator;->getInstance()Ljava/text/Collator;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/pm/PackageItemInfo$DisplayNameComparator;->sCollator:Ljava/text/Collator;
  
+     .line 274
      iput-object p1, p0, Landroid/content/pm/PackageItemInfo$DisplayNameComparator;->mPM:Landroid/content/pm/PackageManager;
  
+     .line 275
      return-void
  .end method
  
***************
*** 60,76 ****
--- 64,83 ----
      .parameter "ab"
  
      .prologue
+     .line 278
      iget-object v2, p0, Landroid/content/pm/PackageItemInfo$DisplayNameComparator;->mPM:Landroid/content/pm/PackageManager;
  
      invoke-virtual {p1, v2}, Landroid/content/pm/PackageItemInfo;->loadLabel(Landroid/content/pm/PackageManager;)Ljava/lang/CharSequence;
  
      move-result-object v0
  
+     .line 279
      .local v0, sa:Ljava/lang/CharSequence;
      if-nez v0, :cond_0
  
      iget-object v0, p1, Landroid/content/pm/PackageItemInfo;->name:Ljava/lang/String;
  
+     .line 280
      :cond_0
      iget-object v2, p0, Landroid/content/pm/PackageItemInfo$DisplayNameComparator;->mPM:Landroid/content/pm/PackageManager;
  
***************
*** 78,88 ****
--- 85,97 ----
  
      move-result-object v1
  
+     .line 281
      .local v1, sb:Ljava/lang/CharSequence;
      if-nez v1, :cond_1
  
      iget-object v1, p2, Landroid/content/pm/PackageItemInfo;->name:Ljava/lang/String;
  
+     .line 282
      :cond_1
      iget-object v2, p0, Landroid/content/pm/PackageItemInfo$DisplayNameComparator;->sCollator:Ljava/text/Collator;
  
***************
*** 107,112 ****
--- 116,122 ----
      .parameter "x1"
  
      .prologue
+     .line 271
      check-cast p1, Landroid/content/pm/PackageItemInfo;
  
      .end local p1
