*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$PermissionGroup.smali	2012-03-23 19:36:30.604239052 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$PermissionGroup.smali.original	2012-03-23 19:34:45.151316121 -0500
***************
*** 33,46 ****
--- 33,49 ----
      .parameter "_owner"
  
      .prologue
+     .line 3204
      invoke-direct {p0, p1}, Landroid/content/pm/PackageParser$Component;-><init>(Landroid/content/pm/PackageParser$Package;)V
  
+     .line 3205
      new-instance v0, Landroid/content/pm/PermissionGroupInfo;
  
      invoke-direct {v0}, Landroid/content/pm/PermissionGroupInfo;-><init>()V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$PermissionGroup;->info:Landroid/content/pm/PermissionGroupInfo;
  
+     .line 3206
      return-void
  .end method
  
***************
*** 50,59 ****
--- 53,65 ----
      .parameter "_info"
  
      .prologue
+     .line 3209
      invoke-direct {p0, p1}, Landroid/content/pm/PackageParser$Component;-><init>(Landroid/content/pm/PackageParser$Package;)V
  
+     .line 3210
      iput-object p2, p0, Landroid/content/pm/PackageParser$PermissionGroup;->info:Landroid/content/pm/PermissionGroupInfo;
  
+     .line 3211
      return-void
  .end method
  
***************
*** 64,75 ****
--- 70,84 ----
      .parameter "packageName"
  
      .prologue
+     .line 3214
      invoke-super {p0, p1}, Landroid/content/pm/PackageParser$Component;->setPackageName(Ljava/lang/String;)V
  
+     .line 3215
      iget-object v0, p0, Landroid/content/pm/PackageParser$PermissionGroup;->info:Landroid/content/pm/PermissionGroupInfo;
  
      iput-object p1, v0, Landroid/content/pm/PermissionGroupInfo;->packageName:Ljava/lang/String;
  
+     .line 3216
      return-void
  .end method
  
***************
*** 77,82 ****
--- 86,92 ----
      .locals 2
  
      .prologue
+     .line 3219
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 113,119 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 123,129 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
