*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$Activity.smali	2012-03-23 19:36:29.554219904 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$Activity.smali.original	2012-03-23 19:34:45.142315956 -0500
***************
*** 34,43 ****
--- 34,46 ----
      .parameter "_info"
  
      .prologue
+     .line 3311
      invoke-direct {p0, p1, p2}, Landroid/content/pm/PackageParser$Component;-><init>(Landroid/content/pm/PackageParser$ParseComponentArgs;Landroid/content/pm/ComponentInfo;)V
  
+     .line 3312
      iput-object p2, p0, Landroid/content/pm/PackageParser$Activity;->info:Landroid/content/pm/ActivityInfo;
  
+     .line 3313
      iget-object v0, p0, Landroid/content/pm/PackageParser$Activity;->info:Landroid/content/pm/ActivityInfo;
  
      iget-object v1, p1, Landroid/content/pm/PackageParser$ParseComponentArgs;->owner:Landroid/content/pm/PackageParser$Package;
***************
*** 46,51 ****
--- 49,55 ----
  
      iput-object v1, v0, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
+     .line 3314
      return-void
  .end method
  
***************
*** 56,67 ****
--- 60,74 ----
      .parameter "packageName"
  
      .prologue
+     .line 3317
      invoke-super {p0, p1}, Landroid/content/pm/PackageParser$Component;->setPackageName(Ljava/lang/String;)V
  
+     .line 3318
      iget-object v0, p0, Landroid/content/pm/PackageParser$Activity;->info:Landroid/content/pm/ActivityInfo;
  
      iput-object p1, v0, Landroid/content/pm/ActivityInfo;->packageName:Ljava/lang/String;
  
+     .line 3319
      return-void
  .end method
  
***************
*** 69,74 ****
--- 76,82 ----
      .locals 2
  
      .prologue
+     .line 3322
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 105,111 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 113,119 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
