*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$Package.smali	2012-03-23 19:36:29.933226816 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PackageParser$Package.smali.original	2012-03-23 19:34:45.146316029 -0500
***************
*** 231,332 ****
--- 231,358 ----
  
      const/4 v1, 0x0
  
+     .line 3016
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 2931
      new-instance v0, Landroid/content/pm/ApplicationInfo;
  
      invoke-direct {v0}, Landroid/content/pm/ApplicationInfo;-><init>()V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
+     .line 2933
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0, v1}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->permissions:Ljava/util/ArrayList;
  
+     .line 2934
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0, v1}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->permissionGroups:Ljava/util/ArrayList;
  
+     .line 2935
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0, v1}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->activities:Ljava/util/ArrayList;
  
+     .line 2936
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0, v1}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->receivers:Ljava/util/ArrayList;
  
+     .line 2937
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0, v1}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->providers:Ljava/util/ArrayList;
  
+     .line 2938
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0, v1}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->services:Ljava/util/ArrayList;
  
+     .line 2939
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0, v1}, Ljava/util/ArrayList;-><init>(I)V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->instrumentation:Ljava/util/ArrayList;
  
+     .line 2941
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->requestedPermissions:Ljava/util/ArrayList;
  
+     .line 2945
      iput-object v2, p0, Landroid/content/pm/PackageParser$Package;->usesLibraries:Ljava/util/ArrayList;
  
+     .line 2946
      iput-object v2, p0, Landroid/content/pm/PackageParser$Package;->usesOptionalLibraries:Ljava/util/ArrayList;
  
+     .line 2947
      iput-object v2, p0, Landroid/content/pm/PackageParser$Package;->usesLibraryFiles:[Ljava/lang/String;
  
+     .line 2949
      iput-object v2, p0, Landroid/content/pm/PackageParser$Package;->mOriginalPackages:Ljava/util/ArrayList;
  
+     .line 2950
      iput-object v2, p0, Landroid/content/pm/PackageParser$Package;->mRealPackage:Ljava/lang/String;
  
+     .line 2951
      iput-object v2, p0, Landroid/content/pm/PackageParser$Package;->mAdoptPermissions:Ljava/util/ArrayList;
  
+     .line 2954
      iput-object v2, p0, Landroid/content/pm/PackageParser$Package;->mAppMetaData:Landroid/os/Bundle;
  
+     .line 2976
      iput v1, p0, Landroid/content/pm/PackageParser$Package;->mPreferredOrder:I
  
+     .line 2986
      iput v1, p0, Landroid/content/pm/PackageParser$Package;->mSetEnabled:I
  
+     .line 2989
      iput-boolean v1, p0, Landroid/content/pm/PackageParser$Package;->mSetStopped:Z
  
+     .line 3000
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Landroid/content/pm/PackageParser$Package;->configPreferences:Ljava/util/ArrayList;
  
+     .line 3006
      iput-object v2, p0, Landroid/content/pm/PackageParser$Package;->reqFeatures:Ljava/util/ArrayList;
  
+     .line 3017
      iput-object p1, p0, Landroid/content/pm/PackageParser$Package;->packageName:Ljava/lang/String;
  
+     .line 3018
      iget-object v0, p0, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iput-object p1, v0, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
  
+     .line 3019
      iget-object v0, p0, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      const/4 v1, -0x1
  
      iput v1, v0, Landroid/content/pm/ApplicationInfo;->uid:I
  
+     .line 3020
      return-void
  .end method
  
***************
*** 337,348 ****
--- 363,377 ----
      .parameter "newName"
  
      .prologue
+     .line 3023
      iput-object p1, p0, Landroid/content/pm/PackageParser$Package;->packageName:Ljava/lang/String;
  
+     .line 3024
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->applicationInfo:Landroid/content/pm/ApplicationInfo;
  
      iput-object p1, v1, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
  
+     .line 3025
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->permissions:Ljava/util/ArrayList;
  
      invoke-virtual {v1}, Ljava/util/ArrayList;->size()I
***************
*** 355,360 ****
--- 384,390 ----
      :goto_0
      if-ltz v0, :cond_0
  
+     .line 3026
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->permissions:Ljava/util/ArrayList;
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 365,374 ****
--- 395,406 ----
  
      invoke-virtual {v1, p1}, Landroid/content/pm/PackageParser$Permission;->setPackageName(Ljava/lang/String;)V
  
+     .line 3025
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_0
  
+     .line 3028
      :cond_0
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->permissionGroups:Ljava/util/ArrayList;
  
***************
*** 381,386 ****
--- 413,419 ----
      :goto_1
      if-ltz v0, :cond_1
  
+     .line 3029
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->permissionGroups:Ljava/util/ArrayList;
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 391,400 ****
--- 424,435 ----
  
      invoke-virtual {v1, p1}, Landroid/content/pm/PackageParser$PermissionGroup;->setPackageName(Ljava/lang/String;)V
  
+     .line 3028
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_1
  
+     .line 3031
      :cond_1
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->activities:Ljava/util/ArrayList;
  
***************
*** 407,412 ****
--- 442,448 ----
      :goto_2
      if-ltz v0, :cond_2
  
+     .line 3032
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->activities:Ljava/util/ArrayList;
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 417,426 ****
--- 453,464 ----
  
      invoke-virtual {v1, p1}, Landroid/content/pm/PackageParser$Activity;->setPackageName(Ljava/lang/String;)V
  
+     .line 3031
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_2
  
+     .line 3034
      :cond_2
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->receivers:Ljava/util/ArrayList;
  
***************
*** 433,438 ****
--- 471,477 ----
      :goto_3
      if-ltz v0, :cond_3
  
+     .line 3035
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->receivers:Ljava/util/ArrayList;
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 443,452 ****
--- 482,493 ----
  
      invoke-virtual {v1, p1}, Landroid/content/pm/PackageParser$Activity;->setPackageName(Ljava/lang/String;)V
  
+     .line 3034
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_3
  
+     .line 3037
      :cond_3
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->providers:Ljava/util/ArrayList;
  
***************
*** 459,464 ****
--- 500,506 ----
      :goto_4
      if-ltz v0, :cond_4
  
+     .line 3038
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->providers:Ljava/util/ArrayList;
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 469,478 ****
--- 511,522 ----
  
      invoke-virtual {v1, p1}, Landroid/content/pm/PackageParser$Provider;->setPackageName(Ljava/lang/String;)V
  
+     .line 3037
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_4
  
+     .line 3040
      :cond_4
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->services:Ljava/util/ArrayList;
  
***************
*** 485,490 ****
--- 529,535 ----
      :goto_5
      if-ltz v0, :cond_5
  
+     .line 3041
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->services:Ljava/util/ArrayList;
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 495,504 ****
--- 540,551 ----
  
      invoke-virtual {v1, p1}, Landroid/content/pm/PackageParser$Service;->setPackageName(Ljava/lang/String;)V
  
+     .line 3040
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_5
  
+     .line 3043
      :cond_5
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->instrumentation:Ljava/util/ArrayList;
  
***************
*** 511,516 ****
--- 558,564 ----
      :goto_6
      if-ltz v0, :cond_6
  
+     .line 3044
      iget-object v1, p0, Landroid/content/pm/PackageParser$Package;->instrumentation:Ljava/util/ArrayList;
  
      invoke-virtual {v1, v0}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 521,530 ****
--- 569,580 ----
  
      invoke-virtual {v1, p1}, Landroid/content/pm/PackageParser$Instrumentation;->setPackageName(Ljava/lang/String;)V
  
+     .line 3043
      add-int/lit8 v0, v0, -0x1
  
      goto :goto_6
  
+     .line 3046
      :cond_6
      return-void
  .end method
***************
*** 533,538 ****
--- 583,589 ----
      .locals 2
  
      .prologue
+     .line 3049
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 567,573 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 618,624 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
