*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PermissionGroupInfo.smali	2012-03-23 19:36:29.373216604 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/PermissionGroupInfo.smali.original	2012-03-23 19:34:45.141315937 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .locals 1
  
      .prologue
+     .line 93
      new-instance v0, Landroid/content/pm/PermissionGroupInfo$1;
  
      invoke-direct {v0}, Landroid/content/pm/PermissionGroupInfo$1;-><init>()V
***************
*** 43,50 ****
--- 44,53 ----
      .locals 0
  
      .prologue
+     .line 44
      invoke-direct {p0}, Landroid/content/pm/PackageItemInfo;-><init>()V
  
+     .line 45
      return-void
  .end method
  
***************
*** 53,68 ****
--- 56,75 ----
      .parameter "orig"
  
      .prologue
+     .line 48
      invoke-direct {p0, p1}, Landroid/content/pm/PackageItemInfo;-><init>(Landroid/content/pm/PackageItemInfo;)V
  
+     .line 49
      iget v0, p1, Landroid/content/pm/PermissionGroupInfo;->descriptionRes:I
  
      iput v0, p0, Landroid/content/pm/PermissionGroupInfo;->descriptionRes:I
  
+     .line 50
      iget-object v0, p1, Landroid/content/pm/PermissionGroupInfo;->nonLocalizedDescription:Ljava/lang/CharSequence;
  
      iput-object v0, p0, Landroid/content/pm/PermissionGroupInfo;->nonLocalizedDescription:Ljava/lang/CharSequence;
  
+     .line 51
      return-void
  .end method
  
***************
*** 71,84 ****
--- 78,94 ----
      .parameter "source"
  
      .prologue
+     .line 104
      invoke-direct {p0, p1}, Landroid/content/pm/PackageItemInfo;-><init>(Landroid/os/Parcel;)V
  
+     .line 105
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/PermissionGroupInfo;->descriptionRes:I
  
+     .line 106
      sget-object v0, Landroid/text/TextUtils;->CHAR_SEQUENCE_CREATOR:Landroid/os/Parcelable$Creator;
  
      invoke-interface {v0, p1}, Landroid/os/Parcelable$Creator;->createFromParcel(Landroid/os/Parcel;)Ljava/lang/Object;
***************
*** 89,94 ****
--- 99,105 ----
  
      iput-object v0, p0, Landroid/content/pm/PermissionGroupInfo;->nonLocalizedDescription:Ljava/lang/CharSequence;
  
+     .line 107
      return-void
  .end method
  
***************
*** 98,103 ****
--- 109,115 ----
      .parameter "x1"
  
      .prologue
+     .line 28
      invoke-direct {p0, p1}, Landroid/content/pm/PermissionGroupInfo;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 109,114 ****
--- 121,127 ----
      .locals 1
  
      .prologue
+     .line 84
      const/4 v0, 0x0
  
      return v0
***************
*** 121,141 ****
--- 134,159 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 65
      iget-object v2, p0, Landroid/content/pm/PermissionGroupInfo;->nonLocalizedDescription:Ljava/lang/CharSequence;
  
      if-eqz v2, :cond_1
  
+     .line 66
      iget-object v0, p0, Landroid/content/pm/PermissionGroupInfo;->nonLocalizedDescription:Ljava/lang/CharSequence;
  
+     .line 74
      :cond_0
      :goto_0
      return-object v0
  
+     .line 68
      :cond_1
      iget v2, p0, Landroid/content/pm/PermissionGroupInfo;->descriptionRes:I
  
      if-eqz v2, :cond_2
  
+     .line 69
      iget-object v2, p0, Landroid/content/pm/PermissionGroupInfo;->packageName:Ljava/lang/String;
  
      iget v3, p0, Landroid/content/pm/PermissionGroupInfo;->descriptionRes:I
***************
*** 144,149 ****
--- 162,168 ----
  
      move-result-object v0
  
+     .line 70
      .local v0, label:Ljava/lang/CharSequence;
      if-nez v0, :cond_0
  
***************
*** 151,156 ****
--- 170,176 ----
      :cond_2
      move-object v0, v1
  
+     .line 74
      goto :goto_0
  .end method
  
***************
*** 158,163 ****
--- 178,184 ----
      .locals 2
  
      .prologue
+     .line 78
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 192,198 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 213,219 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 211,225 ****
--- 232,250 ----
      .parameter "parcelableFlags"
  
      .prologue
+     .line 88
      invoke-super {p0, p1, p2}, Landroid/content/pm/PackageItemInfo;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 89
      iget v0, p0, Landroid/content/pm/PermissionGroupInfo;->descriptionRes:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 90
      iget-object v0, p0, Landroid/content/pm/PermissionGroupInfo;->nonLocalizedDescription:Ljava/lang/CharSequence;
  
      invoke-static {v0, p1, p2}, Landroid/text/TextUtils;->writeToParcel(Ljava/lang/CharSequence;Landroid/os/Parcel;I)V
  
+     .line 91
      return-void
  .end method
