*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/RegisteredServicesCache$ServiceInfo.smali	2012-03-23 19:36:29.983227728 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/RegisteredServicesCache$ServiceInfo.smali.original	2012-03-23 19:34:45.147316047 -0500
***************
*** 52,67 ****
--- 52,72 ----
      .end annotation
  
      .prologue
+     .line 186
      .local p0, this:Landroid/content/pm/RegisteredServicesCache$ServiceInfo;,"Landroid/content/pm/RegisteredServicesCache$ServiceInfo<TV;>;"
      .local p1, type:Ljava/lang/Object;,"TV;"
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 187
      iput-object p1, p0, Landroid/content/pm/RegisteredServicesCache$ServiceInfo;->type:Ljava/lang/Object;
  
+     .line 188
      iput-object p2, p0, Landroid/content/pm/RegisteredServicesCache$ServiceInfo;->componentName:Landroid/content/ComponentName;
  
+     .line 189
      iput p3, p0, Landroid/content/pm/RegisteredServicesCache$ServiceInfo;->uid:I
  
+     .line 190
      return-void
  .end method
  
***************
*** 71,76 ****
--- 76,82 ----
      .locals 2
  
      .prologue
+     .line 194
      .local p0, this:Landroid/content/pm/RegisteredServicesCache$ServiceInfo;,"Landroid/content/pm/RegisteredServicesCache$ServiceInfo<TV;>;"
      new-instance v0, Ljava/lang/StringBuilder;
  
