*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/RegisteredServicesCache$2.smali	2012-03-23 19:36:29.728223078 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/RegisteredServicesCache$2.smali.original	2012-03-23 19:34:45.144315993 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .parameter
  
      .prologue
+     .line 169
      .local p0, this:Landroid/content/pm/RegisteredServicesCache$2;,"Landroid/content/pm/RegisteredServicesCache.2;"
      iput-object p1, p0, Landroid/content/pm/RegisteredServicesCache$2;->this$0:Landroid/content/pm/RegisteredServicesCache;
  
***************
*** 56,61 ****
--- 57,63 ----
      .locals 3
  
      .prologue
+     .line 171
      .local p0, this:Landroid/content/pm/RegisteredServicesCache$2;,"Landroid/content/pm/RegisteredServicesCache.2;"
      iget-object v0, p0, Landroid/content/pm/RegisteredServicesCache$2;->val$listener2:Landroid/content/pm/RegisteredServicesCacheListener;
  
***************
*** 65,69 ****
--- 67,72 ----
  
      invoke-interface {v0, v1, v2}, Landroid/content/pm/RegisteredServicesCacheListener;->onServiceChanged(Ljava/lang/Object;Z)V
  
+     .line 172
      return-void
  .end method
