*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/RegisteredServicesCache$1.smali	2012-03-23 19:36:29.036210458 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/RegisteredServicesCache$1.smali.original	2012-03-23 19:34:45.138315883 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 107
      .local p0, this:Landroid/content/pm/RegisteredServicesCache$1;,"Landroid/content/pm/RegisteredServicesCache.1;"
      iput-object p1, p0, Landroid/content/pm/RegisteredServicesCache$1;->this$0:Landroid/content/pm/RegisteredServicesCache;
  
***************
*** 40,49 ****
--- 41,52 ----
      .parameter "intent"
  
      .prologue
+     .line 110
      .local p0, this:Landroid/content/pm/RegisteredServicesCache$1;,"Landroid/content/pm/RegisteredServicesCache.1;"
      iget-object v0, p0, Landroid/content/pm/RegisteredServicesCache$1;->this$0:Landroid/content/pm/RegisteredServicesCache;
  
      invoke-virtual {v0}, Landroid/content/pm/RegisteredServicesCache;->generateServicesMap()V
  
+     .line 111
      return-void
  .end method
