*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/ServiceInfo.smali	2012-03-23 19:36:30.336234164 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/ServiceInfo.smali.original	2012-03-23 19:34:45.150316102 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 1
  
      .prologue
+     .line 83
      new-instance v0, Landroid/content/pm/ServiceInfo$1;
  
      invoke-direct {v0}, Landroid/content/pm/ServiceInfo$1;-><init>()V
***************
*** 45,52 ****
--- 46,55 ----
      .locals 0
  
      .prologue
+     .line 52
      invoke-direct {p0}, Landroid/content/pm/ComponentInfo;-><init>()V
  
+     .line 53
      return-void
  .end method
  
***************
*** 55,70 ****
--- 58,77 ----
      .parameter "orig"
  
      .prologue
+     .line 56
      invoke-direct {p0, p1}, Landroid/content/pm/ComponentInfo;-><init>(Landroid/content/pm/ComponentInfo;)V
  
+     .line 57
      iget-object v0, p1, Landroid/content/pm/ServiceInfo;->permission:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/ServiceInfo;->permission:Ljava/lang/String;
  
+     .line 58
      iget v0, p1, Landroid/content/pm/ServiceInfo;->flags:I
  
      iput v0, p0, Landroid/content/pm/ServiceInfo;->flags:I
  
+     .line 59
      return-void
  .end method
  
***************
*** 73,92 ****
--- 80,103 ----
      .parameter "source"
  
      .prologue
+     .line 94
      invoke-direct {p0, p1}, Landroid/content/pm/ComponentInfo;-><init>(Landroid/os/Parcel;)V
  
+     .line 95
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/pm/ServiceInfo;->permission:Ljava/lang/String;
  
+     .line 96
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/ServiceInfo;->flags:I
  
+     .line 97
      return-void
  .end method
  
***************
*** 96,101 ****
--- 107,113 ----
      .parameter "x1"
  
      .prologue
+     .line 28
      invoke-direct {p0, p1}, Landroid/content/pm/ServiceInfo;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 107,112 ****
--- 119,125 ----
      .locals 1
  
      .prologue
+     .line 74
      const/4 v0, 0x0
  
      return v0
***************
*** 118,125 ****
--- 131,140 ----
      .parameter "prefix"
  
      .prologue
+     .line 62
      invoke-super {p0, p1, p2}, Landroid/content/pm/ComponentInfo;->dumpFront(Landroid/util/Printer;Ljava/lang/String;)V
  
+     .line 63
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 128,134 ****
  
      move-result-object v0
  
!     const-string v1, "permission="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 143,149 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "permission="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 146,151 ****
--- 161,167 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 64
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 176,181 ****
--- 192,198 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 65
      return-void
  .end method
  
***************
*** 183,188 ****
--- 200,206 ----
      .locals 2
  
      .prologue
+     .line 68
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 217,223 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 235,241 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 236,250 ****
--- 254,272 ----
      .parameter "parcelableFlags"
  
      .prologue
+     .line 78
      invoke-super {p0, p1, p2}, Landroid/content/pm/ComponentInfo;->writeToParcel(Landroid/os/Parcel;I)V
  
+     .line 79
      iget-object v0, p0, Landroid/content/pm/ServiceInfo;->permission:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 80
      iget v0, p0, Landroid/content/pm/ServiceInfo;->flags:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 81
      return-void
  .end method
