*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/UserInfo.smali	2012-03-23 19:36:29.518219248 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/UserInfo.smali.original	2012-03-23 19:34:45.142315956 -0500
***************
*** 38,43 ****
--- 38,44 ----
      .locals 1
  
      .prologue
+     .line 90
      new-instance v0, Landroid/content/pm/UserInfo$1;
  
      invoke-direct {v0}, Landroid/content/pm/UserInfo$1;-><init>()V
***************
*** 51,58 ****
--- 52,61 ----
      .locals 0
  
      .prologue
+     .line 66
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 67
      return-void
  .end method
  
***************
*** 63,76 ****
--- 66,84 ----
      .parameter "flags"
  
      .prologue
+     .line 48
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 49
      iput p1, p0, Landroid/content/pm/UserInfo;->id:I
  
+     .line 50
      iput-object p2, p0, Landroid/content/pm/UserInfo;->name:Ljava/lang/String;
  
+     .line 51
      iput p3, p0, Landroid/content/pm/UserInfo;->flags:I
  
+     .line 52
      return-void
  .end method
  
***************
*** 79,98 ****
--- 87,111 ----
      .parameter "orig"
  
      .prologue
+     .line 69
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 70
      iget-object v0, p1, Landroid/content/pm/UserInfo;->name:Ljava/lang/String;
  
      iput-object v0, p0, Landroid/content/pm/UserInfo;->name:Ljava/lang/String;
  
+     .line 71
      iget v0, p1, Landroid/content/pm/UserInfo;->id:I
  
      iput v0, p0, Landroid/content/pm/UserInfo;->id:I
  
+     .line 72
      iget v0, p1, Landroid/content/pm/UserInfo;->flags:I
  
      iput v0, p0, Landroid/content/pm/UserInfo;->flags:I
  
+     .line 73
      return-void
  .end method
  
***************
*** 101,126 ****
--- 114,144 ----
      .parameter "source"
  
      .prologue
+     .line 100
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 101
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/UserInfo;->id:I
  
+     .line 102
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/pm/UserInfo;->name:Ljava/lang/String;
  
+     .line 103
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Landroid/content/pm/UserInfo;->flags:I
  
+     .line 104
      return-void
  .end method
  
***************
*** 130,135 ****
--- 148,154 ----
      .parameter "x1"
  
      .prologue
+     .line 26
      invoke-direct {p0, p1}, Landroid/content/pm/UserInfo;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 141,146 ****
--- 160,166 ----
      .locals 1
  
      .prologue
+     .line 81
      const/4 v0, 0x0
  
      return v0
***************
*** 150,155 ****
--- 170,176 ----
      .locals 2
  
      .prologue
+     .line 59
      iget v0, p0, Landroid/content/pm/UserInfo;->flags:I
  
      and-int/lit8 v0, v0, 0x2
***************
*** 173,178 ****
--- 194,200 ----
      .locals 2
  
      .prologue
+     .line 63
      iget v0, p0, Landroid/content/pm/UserInfo;->flags:I
  
      and-int/lit8 v0, v0, 0x4
***************
*** 198,203 ****
--- 220,226 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 55
      iget v1, p0, Landroid/content/pm/UserInfo;->flags:I
  
      and-int/lit8 v1, v1, 0x1
***************
*** 217,222 ****
--- 240,246 ----
      .locals 2
  
      .prologue
+     .line 77
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 261,267 ****
  
      move-result-object v0
  
!     const-string v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 285,291 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "}"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 280,296 ****
--- 304,324 ----
      .parameter "parcelableFlags"
  
      .prologue
+     .line 85
      iget v0, p0, Landroid/content/pm/UserInfo;->id:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 86
      iget-object v0, p0, Landroid/content/pm/UserInfo;->name:Ljava/lang/String;
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 87
      iget v0, p0, Landroid/content/pm/UserInfo;->flags:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 88
      return-void
  .end method
