*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/VerifierDeviceIdentity.smali	2012-03-23 19:36:28.480200319 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/content/pm/VerifierDeviceIdentity.smali.original	2012-03-23 19:34:45.132315773 -0500
***************
*** 38,43 ****
--- 38,44 ----
      .locals 1
  
      .prologue
+     .line 94
      const/16 v0, 0x20
  
      new-array v0, v0, [C
***************
*** 46,51 ****
--- 47,53 ----
  
      sput-object v0, Landroid/content/pm/VerifierDeviceIdentity;->ENCODE:[C
  
+     .line 230
      new-instance v0, Landroid/content/pm/VerifierDeviceIdentity$1;
  
      invoke-direct {v0}, Landroid/content/pm/VerifierDeviceIdentity$1;-><init>()V
***************
*** 54,59 ****
--- 56,62 ----
  
      return-void
  
+     .line 94
      nop
  
      :array_0
***************
*** 98,113 ****
--- 101,120 ----
      .parameter "identity"
  
      .prologue
+     .line 59
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 60
      iput-wide p1, p0, Landroid/content/pm/VerifierDeviceIdentity;->mIdentity:J
  
+     .line 61
      invoke-static {p1, p2}, Landroid/content/pm/VerifierDeviceIdentity;->encodeBase32(J)Ljava/lang/String;
  
      move-result-object v0
  
      iput-object v0, p0, Landroid/content/pm/VerifierDeviceIdentity;->mIdentityString:Ljava/lang/String;
  
+     .line 62
      return-void
  .end method
  
***************
*** 116,136 ****
--- 123,148 ----
      .parameter "source"
  
      .prologue
+     .line 64
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 65
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
+     .line 67
      .local v0, identity:J
      iput-wide v0, p0, Landroid/content/pm/VerifierDeviceIdentity;->mIdentity:J
  
+     .line 68
      invoke-static {v0, v1}, Landroid/content/pm/VerifierDeviceIdentity;->encodeBase32(J)Ljava/lang/String;
  
      move-result-object v2
  
      iput-object v2, p0, Landroid/content/pm/VerifierDeviceIdentity;->mIdentityString:Ljava/lang/String;
  
+     .line 69
      return-void
  .end method
  
***************
*** 140,145 ****
--- 152,158 ----
      .parameter "x1"
  
      .prologue
+     .line 35
      invoke-direct {p0, p1}, Landroid/content/pm/VerifierDeviceIdentity;-><init>(Landroid/os/Parcel;)V
  
      return-void
***************
*** 157,170 ****
--- 170,187 ----
      .prologue
      const/16 v11, 0xd
  
+     .line 138
      const-wide/16 v4, 0x0
  
+     .line 139
      .local v4, output:J
      const/4 v3, 0x0
  
+     .line 141
      .local v3, numParsed:I
      array-length v0, p0
  
+     .line 142
      .local v0, N:I
      const/4 v2, 0x0
  
***************
*** 172,179 ****
--- 189,198 ----
      :goto_0
      if-ge v2, v0, :cond_8
  
+     .line 143
      aget-byte v1, p0, v2
  
+     .line 150
      .local v1, group:I
      const/16 v7, 0x41
  
***************
*** 183,190 ****
--- 202,211 ----
  
      if-gt v1, v7, :cond_0
  
+     .line 151
      add-int/lit8 v6, v1, -0x41
  
+     .line 169
      .local v6, value:I
      :goto_1
      const/4 v7, 0x5
***************
*** 195,210 ****
--- 216,235 ----
  
      or-long v4, v7, v9
  
+     .line 170
      add-int/lit8 v3, v3, 0x1
  
+     .line 172
      const/4 v7, 0x1
  
      if-ne v3, v7, :cond_7
  
+     .line 173
      and-int/lit8 v7, v6, 0xf
  
      if-eq v7, v6, :cond_2
  
+     .line 174
      new-instance v7, Ljava/lang/IllegalArgumentException;
  
      const-string v8, "illegal start character; will overflow"
***************
*** 213,218 ****
--- 238,244 ----
  
      throw v7
  
+     .line 152
      .end local v6           #value:I
      :cond_0
      const/16 v7, 0x32
***************
*** 223,244 ****
--- 249,274 ----
  
      if-gt v1, v7, :cond_1
  
+     .line 153
      add-int/lit8 v6, v1, -0x18
  
      .restart local v6       #value:I
      goto :goto_1
  
+     .line 154
      .end local v6           #value:I
      :cond_1
      const/16 v7, 0x2d
  
      if-ne v1, v7, :cond_3
  
+     .line 142
      :cond_2
      add-int/lit8 v2, v2, 0x1
  
      goto :goto_0
  
+     .line 156
      :cond_3
      const/16 v7, 0x61
  
***************
*** 248,280 ****
--- 278,316 ----
  
      if-gt v1, v7, :cond_4
  
+     .line 158
      add-int/lit8 v6, v1, -0x61
  
      .restart local v6       #value:I
      goto :goto_1
  
+     .line 159
      .end local v6           #value:I
      :cond_4
      const/16 v7, 0x30
  
      if-ne v1, v7, :cond_5
  
+     .line 161
      const/16 v6, 0xe
  
      .restart local v6       #value:I
      goto :goto_1
  
+     .line 162
      .end local v6           #value:I
      :cond_5
      const/16 v7, 0x31
  
      if-ne v1, v7, :cond_6
  
+     .line 164
      const/16 v6, 0x8
  
      .restart local v6       #value:I
      goto :goto_1
  
+     .line 166
      .end local v6           #value:I
      :cond_6
      new-instance v7, Ljava/lang/IllegalArgumentException;
***************
*** 301,331 ****
  
      throw v7
  
      .restart local v6       #value:I
      :cond_7
      if-le v3, v11, :cond_2
  
      new-instance v7, Ljava/lang/IllegalArgumentException;
  
!     const-string v8, "too long; should have 13 characters"
  
      invoke-direct {v7, v8}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v7
  
      .end local v1           #group:I
      .end local v6           #value:I
      :cond_8
      if-eq v3, v11, :cond_9
  
      new-instance v7, Ljava/lang/IllegalArgumentException;
  
!     const-string v8, "too short; should have 13 characters"
  
      invoke-direct {v7, v8}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v7
  
      :cond_9
      return-wide v4
  .end method
--- 337,372 ----
  
      throw v7
  
+     .line 176
      .restart local v6       #value:I
      :cond_7
      if-le v3, v11, :cond_2
  
+     .line 177
      new-instance v7, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v8, "too long; should have 13 characters"
  
      invoke-direct {v7, v8}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v7
  
+     .line 181
      .end local v1           #group:I
      .end local v6           #value:I
      :cond_8
      if-eq v3, v11, :cond_9
  
+     .line 182
      new-instance v7, Ljava/lang/IllegalArgumentException;
  
!     const-string/jumbo v8, "too short; should have 13 characters"
  
      invoke-direct {v7, v8}, Ljava/lang/IllegalArgumentException;-><init>(Ljava/lang/String;)V
  
      throw v7
  
+     .line 185
      :cond_9
      return-wide v4
  .end method
***************
*** 335,350 ****
--- 376,395 ----
      .parameter "input"
  
      .prologue
+     .line 104
      sget-object v0, Landroid/content/pm/VerifierDeviceIdentity;->ENCODE:[C
  
+     .line 110
      .local v0, alphabet:[C
      const/16 v5, 0x10
  
      new-array v1, v5, [C
  
+     .line 112
      .local v1, encoded:[C
      array-length v4, v1
  
+     .line 113
      .local v4, index:I
      const/4 v3, 0x0
  
***************
*** 354,359 ****
--- 399,405 ----
  
      if-ge v3, v5, :cond_1
  
+     .line 120
      if-lez v3, :cond_0
  
      rem-int/lit8 v5, v3, 0x4
***************
*** 362,373 ****
--- 408,421 ----
  
      if-ne v5, v6, :cond_0
  
+     .line 121
      add-int/lit8 v4, v4, -0x1
  
      const/16 v5, 0x2d
  
      aput-char v5, v1, v4
  
+     .line 127
      :cond_0
      const-wide/16 v5, 0x1f
  
***************
*** 375,395 ****
--- 423,447 ----
  
      long-to-int v2, v5
  
+     .line 128
      .local v2, group:I
      const/4 v5, 0x5
  
      ushr-long/2addr p0, v5
  
+     .line 130
      add-int/lit8 v4, v4, -0x1
  
      aget-char v5, v0, v2
  
      aput-char v5, v1, v4
  
+     .line 113
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 133
      .end local v2           #group:I
      :cond_1
      invoke-static {v1}, Ljava/lang/String;->valueOf([C)Ljava/lang/String;
***************
*** 403,412 ****
--- 455,466 ----
      .locals 2
  
      .prologue
+     .line 77
      new-instance v0, Ljava/security/SecureRandom;
  
      invoke-direct {v0}, Ljava/security/SecureRandom;-><init>()V
  
+     .line 78
      .local v0, sr:Ljava/security/SecureRandom;
      invoke-static {v0}, Landroid/content/pm/VerifierDeviceIdentity;->generate(Ljava/util/Random;)Landroid/content/pm/VerifierDeviceIdentity;
  
***************
*** 420,429 ****
--- 474,485 ----
      .parameter "rng"
  
      .prologue
+     .line 90
      invoke-virtual {p0}, Ljava/util/Random;->nextLong()J
  
      move-result-wide v0
  
+     .line 91
      .local v0, identity:J
      new-instance v2, Landroid/content/pm/VerifierDeviceIdentity;
  
***************
*** 442,447 ****
--- 498,504 ----
      .end annotation
  
      .prologue
+     .line 212
      :try_start_0
      const-string v2, "US-ASCII"
  
***************
*** 451,456 ****
--- 508,514 ----
  
      move-result-object v1
  
+     .line 217
      .local v1, input:[B
      new-instance v2, Landroid/content/pm/VerifierDeviceIdentity;
  
***************
*** 462,471 ****
--- 520,531 ----
  
      return-object v2
  
+     .line 213
      .end local v1           #input:[B
      :catch_0
      move-exception v0
  
+     .line 214
      .local v0, e:Ljava/io/UnsupportedEncodingException;
      new-instance v2, Ljava/lang/IllegalArgumentException;
  
***************
*** 482,487 ****
--- 542,548 ----
      .locals 1
  
      .prologue
+     .line 222
      const/4 v0, 0x0
  
      return v0
***************
*** 494,503 ****
--- 555,566 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 195
      instance-of v2, p1, Landroid/content/pm/VerifierDeviceIdentity;
  
      if-nez v2, :cond_1
  
+     .line 200
      :cond_0
      :goto_0
      return v1
***************
*** 505,512 ****
--- 568,577 ----
      :cond_1
      move-object v0, p1
  
+     .line 199
      check-cast v0, Landroid/content/pm/VerifierDeviceIdentity;
  
+     .line 200
      .local v0, o:Landroid/content/pm/VerifierDeviceIdentity;
      iget-wide v2, p0, Landroid/content/pm/VerifierDeviceIdentity;->mIdentity:J
  
***************
*** 525,530 ****
--- 590,596 ----
      .locals 2
  
      .prologue
+     .line 190
      iget-wide v0, p0, Landroid/content/pm/VerifierDeviceIdentity;->mIdentity:J
  
      long-to-int v0, v0
***************
*** 536,541 ****
--- 602,608 ----
      .locals 1
  
      .prologue
+     .line 205
      iget-object v0, p0, Landroid/content/pm/VerifierDeviceIdentity;->mIdentityString:Ljava/lang/String;
  
      return-object v0
***************
*** 547,555 ****
--- 614,624 ----
      .parameter "flags"
  
      .prologue
+     .line 227
      iget-wide v0, p0, Landroid/content/pm/VerifierDeviceIdentity;->mIdentity:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 228
      return-void
  .end method
