*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Browser$SearchColumns.smali	2012-03-23 19:35:27.721092379 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Browser$SearchColumns.smali.original	2012-03-23 19:34:44.603306127 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .locals 0
  
      .prologue
+     .line 641
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
