*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/BrowserContract$Combined.smali	2012-03-23 19:35:25.954060157 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/BrowserContract$Combined.smali.original	2012-03-23 19:34:44.588305853 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .locals 2
  
      .prologue
+     .line 496
      sget-object v0, Landroid/provider/BrowserContract;->AUTHORITY_URI:Landroid/net/Uri;
  
      const-string v1, "combined"
***************
*** 47,52 ****
--- 48,54 ----
      .locals 0
  
      .prologue
+     .line 491
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
