*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$Events.smali	2012-03-23 19:35:27.336085358 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$Events.smali.original	2012-03-23 19:34:44.600306073 -0500
***************
*** 47,52 ****
--- 47,53 ----
  
      const/4 v3, 0x0
  
+     .line 1506
      const-string v0, "content://com.android.calendar/events"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 55,60 ****
--- 56,62 ----
  
      sput-object v0, Landroid/provider/CalendarContract$Events;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 1514
      const-string v0, "content://com.android.calendar/exception"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 63,68 ****
--- 65,71 ----
  
      sput-object v0, Landroid/provider/CalendarContract$Events;->CONTENT_EXCEPTION_URI:Landroid/net/Uri;
  
+     .line 1534
      const/16 v0, 0x18
  
      new-array v0, v0, [Ljava/lang/String;
***************
*** 191,208 ****
  
      const/16 v1, 0x16
  
!     const-string v2, "sync_events"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x17
  
!     const-string v2, "visible"
  
      aput-object v2, v0, v1
  
      sput-object v0, Landroid/provider/CalendarContract$Events;->PROVIDER_WRITABLE_COLUMNS:[Ljava/lang/String;
  
      const/16 v0, 0xc
  
      new-array v0, v0, [Ljava/lang/String;
--- 194,212 ----
  
      const/16 v1, 0x16
  
!     const-string/jumbo v2, "sync_events"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x17
  
!     const-string/jumbo v2, "visible"
  
      aput-object v2, v0, v1
  
      sput-object v0, Landroid/provider/CalendarContract$Events;->PROVIDER_WRITABLE_COLUMNS:[Ljava/lang/String;
  
+     .line 1568
      const/16 v0, 0xc
  
      new-array v0, v0, [Ljava/lang/String;
***************
*** 215,271 ****
  
      aput-object v1, v0, v4
  
!     const-string v1, "sync_data1"
  
      aput-object v1, v0, v5
  
!     const-string v1, "sync_data2"
  
      aput-object v1, v0, v6
  
!     const-string v1, "sync_data3"
  
      aput-object v1, v0, v7
  
      const/4 v1, 0x5
  
!     const-string v2, "sync_data4"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x6
  
!     const-string v2, "sync_data5"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x7
  
!     const-string v2, "sync_data6"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x8
  
!     const-string v2, "sync_data7"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x9
  
!     const-string v2, "sync_data8"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0xa
  
!     const-string v2, "sync_data9"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0xb
  
!     const-string v2, "sync_data10"
  
      aput-object v2, v0, v1
  
--- 219,275 ----
  
      aput-object v1, v0, v4
  
!     const-string/jumbo v1, "sync_data1"
  
      aput-object v1, v0, v5
  
!     const-string/jumbo v1, "sync_data2"
  
      aput-object v1, v0, v6
  
!     const-string/jumbo v1, "sync_data3"
  
      aput-object v1, v0, v7
  
      const/4 v1, 0x5
  
!     const-string/jumbo v2, "sync_data4"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x6
  
!     const-string/jumbo v2, "sync_data5"
  
      aput-object v2, v0, v1
  
      const/4 v1, 0x7
  
!     const-string/jumbo v2, "sync_data6"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x8
  
!     const-string/jumbo v2, "sync_data7"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0x9
  
!     const-string/jumbo v2, "sync_data8"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0xa
  
!     const-string/jumbo v2, "sync_data9"
  
      aput-object v2, v0, v1
  
      const/16 v1, 0xb
  
!     const-string/jumbo v2, "sync_data10"
  
      aput-object v2, v0, v1
  
***************
*** 278,283 ****
--- 282,288 ----
      .locals 0
  
      .prologue
+     .line 1520
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
