*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$Colors.smali	2012-03-23 19:35:27.122081456 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$Colors.smali.original	2012-03-23 19:34:44.598306036 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .locals 1
  
      .prologue
+     .line 2362
      const-string v0, "content://com.android.calendar/colors"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 43,49 ****
--- 44,52 ----
      .locals 0
  
      .prologue
+     .line 2367
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 2368
      return-void
  .end method
