*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$Instances.smali	2012-03-23 19:35:26.196064571 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$Instances.smali.original	2012-03-23 19:34:44.590305891 -0500
***************
*** 54,59 ****
--- 54,60 ----
      .locals 3
  
      .prologue
+     .line 1593
      const/4 v0, 0x1
  
      new-array v0, v0, [Ljava/lang/String;
***************
*** 66,71 ****
--- 67,73 ----
  
      sput-object v0, Landroid/provider/CalendarContract$Instances;->WHERE_CALENDARS_ARGS:[Ljava/lang/String;
  
+     .line 1662
      const-string v0, "content://com.android.calendar/instances/when"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 74,79 ****
--- 76,82 ----
  
      sput-object v0, Landroid/provider/CalendarContract$Instances;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 1669
      const-string v0, "content://com.android.calendar/instances/whenbyday"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 82,87 ****
--- 85,91 ----
  
      sput-object v0, Landroid/provider/CalendarContract$Instances;->CONTENT_BY_DAY_URI:Landroid/net/Uri;
  
+     .line 1676
      const-string v0, "content://com.android.calendar/instances/search"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 90,95 ****
--- 94,100 ----
  
      sput-object v0, Landroid/provider/CalendarContract$Instances;->CONTENT_SEARCH_URI:Landroid/net/Uri;
  
+     .line 1683
      const-string v0, "content://com.android.calendar/instances/searchbyday"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 105,110 ****
--- 110,116 ----
      .locals 0
  
      .prologue
+     .line 1600
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 118,139 ****
      .parameter "end"
  
      .prologue
      sget-object v0, Landroid/provider/CalendarContract$Instances;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-virtual {v0}, Landroid/net/Uri;->buildUpon()Landroid/net/Uri$Builder;
  
      move-result-object v6
  
      .local v6, builder:Landroid/net/Uri$Builder;
      invoke-static {v6, p2, p3}, Landroid/content/ContentUris;->appendId(Landroid/net/Uri$Builder;J)Landroid/net/Uri$Builder;
  
      invoke-static {v6, p4, p5}, Landroid/content/ContentUris;->appendId(Landroid/net/Uri$Builder;J)Landroid/net/Uri$Builder;
  
      invoke-virtual {v6}, Landroid/net/Uri$Builder;->build()Landroid/net/Uri;
  
      move-result-object v1
  
!     const-string v3, "visible=?"
  
      sget-object v4, Landroid/provider/CalendarContract$Instances;->WHERE_CALENDARS_ARGS:[Ljava/lang/String;
  
--- 124,149 ----
      .parameter "end"
  
      .prologue
+     .line 1619
      sget-object v0, Landroid/provider/CalendarContract$Instances;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-virtual {v0}, Landroid/net/Uri;->buildUpon()Landroid/net/Uri$Builder;
  
      move-result-object v6
  
+     .line 1620
      .local v6, builder:Landroid/net/Uri$Builder;
      invoke-static {v6, p2, p3}, Landroid/content/ContentUris;->appendId(Landroid/net/Uri$Builder;J)Landroid/net/Uri$Builder;
  
+     .line 1621
      invoke-static {v6, p4, p5}, Landroid/content/ContentUris;->appendId(Landroid/net/Uri$Builder;J)Landroid/net/Uri$Builder;
  
+     .line 1622
      invoke-virtual {v6}, Landroid/net/Uri$Builder;->build()Landroid/net/Uri;
  
      move-result-object v1
  
!     const-string/jumbo v3, "visible=?"
  
      sget-object v4, Landroid/provider/CalendarContract$Instances;->WHERE_CALENDARS_ARGS:[Ljava/lang/String;
  
***************
*** 159,184 ****
      .parameter "searchQuery"
  
      .prologue
      sget-object v0, Landroid/provider/CalendarContract$Instances;->CONTENT_SEARCH_URI:Landroid/net/Uri;
  
      invoke-virtual {v0}, Landroid/net/Uri;->buildUpon()Landroid/net/Uri$Builder;
  
      move-result-object v6
  
      .local v6, builder:Landroid/net/Uri$Builder;
      invoke-static {v6, p2, p3}, Landroid/content/ContentUris;->appendId(Landroid/net/Uri$Builder;J)Landroid/net/Uri$Builder;
  
      invoke-static {v6, p4, p5}, Landroid/content/ContentUris;->appendId(Landroid/net/Uri$Builder;J)Landroid/net/Uri$Builder;
  
      invoke-virtual {v6, p6}, Landroid/net/Uri$Builder;->appendPath(Ljava/lang/String;)Landroid/net/Uri$Builder;
  
      move-result-object v6
  
      invoke-virtual {v6}, Landroid/net/Uri$Builder;->build()Landroid/net/Uri;
  
      move-result-object v1
  
!     const-string v3, "visible=?"
  
      sget-object v4, Landroid/provider/CalendarContract$Instances;->WHERE_CALENDARS_ARGS:[Ljava/lang/String;
  
--- 169,199 ----
      .parameter "searchQuery"
  
      .prologue
+     .line 1648
      sget-object v0, Landroid/provider/CalendarContract$Instances;->CONTENT_SEARCH_URI:Landroid/net/Uri;
  
      invoke-virtual {v0}, Landroid/net/Uri;->buildUpon()Landroid/net/Uri$Builder;
  
      move-result-object v6
  
+     .line 1649
      .local v6, builder:Landroid/net/Uri$Builder;
      invoke-static {v6, p2, p3}, Landroid/content/ContentUris;->appendId(Landroid/net/Uri$Builder;J)Landroid/net/Uri$Builder;
  
+     .line 1650
      invoke-static {v6, p4, p5}, Landroid/content/ContentUris;->appendId(Landroid/net/Uri$Builder;J)Landroid/net/Uri$Builder;
  
+     .line 1651
      invoke-virtual {v6, p6}, Landroid/net/Uri$Builder;->appendPath(Ljava/lang/String;)Landroid/net/Uri$Builder;
  
      move-result-object v6
  
+     .line 1652
      invoke-virtual {v6}, Landroid/net/Uri$Builder;->build()Landroid/net/Uri;
  
      move-result-object v1
  
!     const-string/jumbo v3, "visible=?"
  
      sget-object v4, Landroid/provider/CalendarContract$Instances;->WHERE_CALENDARS_ARGS:[Ljava/lang/String;
  
