*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$CalendarCache.smali	2012-03-23 19:35:25.549052773 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$CalendarCache.smali.original	2012-03-23 19:34:44.585305799 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .locals 1
  
      .prologue
+     .line 1759
      const-string v0, "content://com.android.calendar/properties"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 51,56 ****
--- 52,58 ----
      .locals 0
  
      .prologue
+     .line 1765
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
