*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$EventsEntity$EntityIteratorImpl.smali	2012-03-23 19:35:30.089135560 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract$EventsEntity$EntityIteratorImpl.smali.original	2012-03-23 19:34:44.623306492 -0500
***************
*** 63,80 ****
  
      const/4 v3, 0x0
  
      new-array v0, v5, [Ljava/lang/String;
  
!     const-string v1, "minutes"
  
      aput-object v1, v0, v3
  
!     const-string v1, "method"
  
      aput-object v1, v0, v4
  
      sput-object v0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->REMINDERS_PROJECTION:[Ljava/lang/String;
  
      const/4 v0, 0x5
  
      new-array v0, v0, [Ljava/lang/String;
--- 63,82 ----
  
      const/4 v3, 0x0
  
+     .line 1200
      new-array v0, v5, [Ljava/lang/String;
  
!     const-string/jumbo v1, "minutes"
  
      aput-object v1, v0, v3
  
!     const-string/jumbo v1, "method"
  
      aput-object v1, v0, v4
  
      sput-object v0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->REMINDERS_PROJECTION:[Ljava/lang/String;
  
+     .line 1207
      const/4 v0, 0x5
  
      new-array v0, v0, [Ljava/lang/String;
***************
*** 103,119 ****
  
      sput-object v0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->ATTENDEES_PROJECTION:[Ljava/lang/String;
  
      new-array v0, v6, [Ljava/lang/String;
  
      const-string v1, "_id"
  
      aput-object v1, v0, v3
  
!     const-string v1, "name"
  
      aput-object v1, v0, v4
  
!     const-string v1, "value"
  
      aput-object v1, v0, v5
  
--- 105,122 ----
  
      sput-object v0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->ATTENDEES_PROJECTION:[Ljava/lang/String;
  
+     .line 1219
      new-array v0, v6, [Ljava/lang/String;
  
      const-string v1, "_id"
  
      aput-object v1, v0, v3
  
!     const-string/jumbo v1, "name"
  
      aput-object v1, v0, v4
  
!     const-string/jumbo v1, "value"
  
      aput-object v1, v0, v5
  
***************
*** 128,141 ****
--- 131,148 ----
      .parameter "provider"
  
      .prologue
+     .line 1237
      invoke-direct {p0, p1}, Landroid/content/CursorEntityIterator;-><init>(Landroid/database/Cursor;)V
  
+     .line 1238
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->mResolver:Landroid/content/ContentResolver;
  
+     .line 1239
      iput-object p2, p0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->mProvider:Landroid/content/ContentProviderClient;
  
+     .line 1240
      return-void
  .end method
  
***************
*** 145,158 ****
--- 152,169 ----
      .parameter "resolver"
  
      .prologue
+     .line 1231
      invoke-direct {p0, p1}, Landroid/content/CursorEntityIterator;-><init>(Landroid/database/Cursor;)V
  
+     .line 1232
      iput-object p2, p0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->mResolver:Landroid/content/ContentResolver;
  
+     .line 1233
      const/4 v0, 0x0
  
      iput-object v0, p0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->mProvider:Landroid/content/ContentProviderClient;
  
+     .line 1234
      return-void
  .end method
  
***************
*** 168,173 ****
--- 179,185 ----
      .end annotation
  
      .prologue
+     .line 1245
      const-string v1, "_id"
  
      move-object/from16 v0, p1
***************
*** 182,192 ****
--- 194,206 ----
  
      move-result-wide v10
  
+     .line 1246
      .local v10, eventId:J
      new-instance v8, Landroid/content/ContentValues;
  
      invoke-direct {v8}, Landroid/content/ContentValues;-><init>()V
  
+     .line 1247
      .local v8, cv:Landroid/content/ContentValues;
      const-string v1, "_id"
  
***************
*** 196,529 ****
  
      invoke-virtual {v8, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
      const-string v1, "calendar_id"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "title"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "description"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "eventLocation"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "eventStatus"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "selfAttendeeStatus"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "dtstart"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "dtend"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "duration"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "eventTimezone"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "eventEndTimezone"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "allDay"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "accessLevel"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "availability"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "hasAlarm"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "hasExtendedProperties"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "rrule"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "rdate"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "exrule"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "exdate"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "original_sync_id"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "original_id"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "originalInstanceTime"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "originalAllDay"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "lastDate"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "hasAttendeeData"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "guestsCanInviteOthers"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "guestsCanModify"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "guestsCanSeeGuests"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "organizer"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "_sync_id"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "dirty"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "lastSynced"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "deleted"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data1"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data2"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data3"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data4"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data5"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data6"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data7"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data8"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data9"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     const-string v1, "sync_data10"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync1"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync2"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync3"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync4"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync5"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync6"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync7"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync8"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync9"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      const-string v1, "cal_sync10"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
      new-instance v9, Landroid/content/Entity;
  
      invoke-direct {v9, v8}, Landroid/content/Entity;-><init>(Landroid/content/ContentValues;)V
  
      .local v9, entity:Landroid/content/Entity;
      move-object/from16 v0, p0
  
--- 210,599 ----
  
      invoke-virtual {v8, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
+     .line 1248
      const-string v1, "calendar_id"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1249
!     const-string/jumbo v1, "title"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1250
      const-string v1, "description"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1251
      const-string v1, "eventLocation"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1252
      const-string v1, "eventStatus"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1253
!     const-string/jumbo v1, "selfAttendeeStatus"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1254
      const-string v1, "dtstart"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1255
      const-string v1, "dtend"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1256
      const-string v1, "duration"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1257
      const-string v1, "eventTimezone"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1258
      const-string v1, "eventEndTimezone"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1259
      const-string v1, "allDay"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1260
      const-string v1, "accessLevel"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1261
      const-string v1, "availability"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1262
      const-string v1, "hasAlarm"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1263
      const-string v1, "hasExtendedProperties"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1265
!     const-string/jumbo v1, "rrule"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1266
!     const-string/jumbo v1, "rdate"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1267
      const-string v1, "exrule"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1268
      const-string v1, "exdate"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1269
!     const-string/jumbo v1, "original_sync_id"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1270
!     const-string/jumbo v1, "original_id"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1271
!     const-string/jumbo v1, "originalInstanceTime"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1273
!     const-string/jumbo v1, "originalAllDay"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1274
      const-string v1, "lastDate"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1275
      const-string v1, "hasAttendeeData"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1276
      const-string v1, "guestsCanInviteOthers"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1278
      const-string v1, "guestsCanModify"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1279
      const-string v1, "guestsCanSeeGuests"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1280
!     const-string/jumbo v1, "organizer"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1281
      const-string v1, "_sync_id"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1282
      const-string v1, "dirty"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1283
      const-string v1, "lastSynced"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorLongToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1284
      const-string v1, "deleted"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorIntToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1285
!     const-string/jumbo v1, "sync_data1"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1286
!     const-string/jumbo v1, "sync_data2"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1287
!     const-string/jumbo v1, "sync_data3"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1288
!     const-string/jumbo v1, "sync_data4"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1289
!     const-string/jumbo v1, "sync_data5"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1290
!     const-string/jumbo v1, "sync_data6"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1291
!     const-string/jumbo v1, "sync_data7"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1292
!     const-string/jumbo v1, "sync_data8"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1293
!     const-string/jumbo v1, "sync_data9"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
!     .line 1294
!     const-string/jumbo v1, "sync_data10"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1295
      const-string v1, "cal_sync1"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1296
      const-string v1, "cal_sync2"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1297
      const-string v1, "cal_sync3"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1298
      const-string v1, "cal_sync4"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1299
      const-string v1, "cal_sync5"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1300
      const-string v1, "cal_sync6"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1301
      const-string v1, "cal_sync7"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1302
      const-string v1, "cal_sync8"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1303
      const-string v1, "cal_sync9"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1304
      const-string v1, "cal_sync10"
  
      move-object/from16 v0, p1
  
      invoke-static {v0, v8, v1}, Landroid/database/DatabaseUtils;->cursorStringToContentValuesIfPresent(Landroid/database/Cursor;Landroid/content/ContentValues;Ljava/lang/String;)V
  
+     .line 1306
      new-instance v9, Landroid/content/Entity;
  
      invoke-direct {v9, v8}, Landroid/content/Entity;-><init>(Landroid/content/ContentValues;)V
  
+     .line 1308
      .local v9, entity:Landroid/content/Entity;
      move-object/from16 v0, p0
  
***************
*** 531,536 ****
--- 601,607 ----
  
      if-eqz v1, :cond_0
  
+     .line 1309
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->mResolver:Landroid/content/ContentResolver;
***************
*** 559,564 ****
--- 630,636 ----
  
      move-result-object v14
  
+     .line 1320
      .local v14, subCursor:Landroid/database/Cursor;
      :goto_0
      :try_start_0
***************
*** 568,579 ****
  
      if-eqz v1, :cond_1
  
      new-instance v13, Landroid/content/ContentValues;
  
      invoke-direct {v13}, Landroid/content/ContentValues;-><init>()V
  
      .local v13, reminderValues:Landroid/content/ContentValues;
!     const-string v1, "minutes"
  
      const/4 v2, 0x0
  
--- 640,653 ----
  
      if-eqz v1, :cond_1
  
+     .line 1321
      new-instance v13, Landroid/content/ContentValues;
  
      invoke-direct {v13}, Landroid/content/ContentValues;-><init>()V
  
+     .line 1322
      .local v13, reminderValues:Landroid/content/ContentValues;
!     const-string/jumbo v1, "minutes"
  
      const/4 v2, 0x0
  
***************
*** 587,593 ****
  
      invoke-virtual {v13, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     const-string v1, "method"
  
      const/4 v2, 0x1
  
--- 661,668 ----
  
      invoke-virtual {v13, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     .line 1323
!     const-string/jumbo v1, "method"
  
      const/4 v2, 0x1
  
***************
*** 601,606 ****
--- 676,682 ----
  
      invoke-virtual {v13, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 1324
      sget-object v1, Landroid/provider/CalendarContract$Reminders;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-virtual {v9, v1, v13}, Landroid/content/Entity;->addSubValue(Landroid/net/Uri;Landroid/content/ContentValues;)V
***************
*** 609,614 ****
--- 685,691 ----
  
      goto :goto_0
  
+     .line 1327
      .end local v13           #reminderValues:Landroid/content/ContentValues;
      :catchall_0
      move-exception v1
***************
*** 617,622 ****
--- 694,700 ----
  
      throw v1
  
+     .line 1314
      .end local v14           #subCursor:Landroid/database/Cursor;
      :cond_0
      move-object/from16 v0, p0
***************
*** 650,664 ****
--- 728,745 ----
      .restart local v14       #subCursor:Landroid/database/Cursor;
      goto :goto_0
  
+     .line 1327
      :cond_1
      invoke-interface {v14}, Landroid/database/Cursor;->close()V
  
+     .line 1330
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->mResolver:Landroid/content/ContentResolver;
  
      if-eqz v1, :cond_2
  
+     .line 1331
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->mResolver:Landroid/content/ContentResolver;
***************
*** 687,692 ****
--- 768,774 ----
  
      move-result-object v14
  
+     .line 1342
      :goto_1
      :try_start_1
      invoke-interface {v14}, Landroid/database/Cursor;->moveToNext()Z
***************
*** 695,704 ****
--- 777,788 ----
  
      if-eqz v1, :cond_3
  
+     .line 1343
      new-instance v7, Landroid/content/ContentValues;
  
      invoke-direct {v7}, Landroid/content/ContentValues;-><init>()V
  
+     .line 1344
      .local v7, attendeeValues:Landroid/content/ContentValues;
      const-string v1, "attendeeName"
  
***************
*** 710,715 ****
--- 794,800 ----
  
      invoke-virtual {v7, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1346
      const-string v1, "attendeeEmail"
  
      const/4 v2, 0x1
***************
*** 720,725 ****
--- 805,811 ----
  
      invoke-virtual {v7, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1348
      const-string v1, "attendeeRelationship"
  
      const/4 v2, 0x2
***************
*** 734,739 ****
--- 820,826 ----
  
      invoke-virtual {v7, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 1350
      const-string v1, "attendeeType"
  
      const/4 v2, 0x3
***************
*** 748,753 ****
--- 835,841 ----
  
      invoke-virtual {v7, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 1352
      const-string v1, "attendeeStatus"
  
      const/4 v2, 0x4
***************
*** 762,767 ****
--- 850,856 ----
  
      invoke-virtual {v7, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 1354
      sget-object v1, Landroid/provider/CalendarContract$Attendees;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-virtual {v9, v1, v7}, Landroid/content/Entity;->addSubValue(Landroid/net/Uri;Landroid/content/ContentValues;)V
***************
*** 770,775 ****
--- 859,865 ----
  
      goto :goto_1
  
+     .line 1357
      .end local v7           #attendeeValues:Landroid/content/ContentValues;
      :catchall_1
      move-exception v1
***************
*** 778,783 ****
--- 868,874 ----
  
      throw v1
  
+     .line 1336
      :cond_2
      move-object/from16 v0, p0
  
***************
*** 809,823 ****
--- 900,917 ----
  
      goto :goto_1
  
+     .line 1357
      :cond_3
      invoke-interface {v14}, Landroid/database/Cursor;->close()V
  
+     .line 1360
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->mResolver:Landroid/content/ContentResolver;
  
      if-eqz v1, :cond_4
  
+     .line 1361
      move-object/from16 v0, p0
  
      iget-object v1, v0, Landroid/provider/CalendarContract$EventsEntity$EntityIteratorImpl;->mResolver:Landroid/content/ContentResolver;
***************
*** 846,851 ****
--- 940,946 ----
  
      move-result-object v14
  
+     .line 1372
      :goto_2
      :try_start_2
      invoke-interface {v14}, Landroid/database/Cursor;->moveToNext()Z
***************
*** 854,863 ****
--- 949,960 ----
  
      if-eqz v1, :cond_5
  
+     .line 1373
      new-instance v12, Landroid/content/ContentValues;
  
      invoke-direct {v12}, Landroid/content/ContentValues;-><init>()V
  
+     .line 1374
      .local v12, extendedValues:Landroid/content/ContentValues;
      const-string v1, "_id"
  
***************
*** 869,875 ****
  
      invoke-virtual {v12, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v1, "name"
  
      const/4 v2, 0x1
  
--- 966,973 ----
  
      invoke-virtual {v12, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 1376
!     const-string/jumbo v1, "name"
  
      const/4 v2, 0x1
  
***************
*** 879,885 ****
  
      invoke-virtual {v12, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v1, "value"
  
      const/4 v2, 0x2
  
--- 977,984 ----
  
      invoke-virtual {v12, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 1378
!     const-string/jumbo v1, "value"
  
      const/4 v2, 0x2
  
***************
*** 889,894 ****
--- 988,994 ----
  
      invoke-virtual {v12, v1, v2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 1380
      sget-object v1, Landroid/provider/CalendarContract$ExtendedProperties;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-virtual {v9, v1, v12}, Landroid/content/Entity;->addSubValue(Landroid/net/Uri;Landroid/content/ContentValues;)V
***************
*** 897,902 ****
--- 997,1003 ----
  
      goto :goto_2
  
+     .line 1383
      .end local v12           #extendedValues:Landroid/content/ContentValues;
      :catchall_2
      move-exception v1
***************
*** 905,910 ****
--- 1006,1012 ----
  
      throw v1
  
+     .line 1366
      :cond_4
      move-object/from16 v0, p0
  
***************
*** 936,945 ****
--- 1038,1050 ----
  
      goto :goto_2
  
+     .line 1383
      :cond_5
      invoke-interface {v14}, Landroid/database/Cursor;->close()V
  
+     .line 1386
      invoke-interface/range {p1 .. p1}, Landroid/database/Cursor;->moveToNext()Z
  
+     .line 1387
      return-object v9
  .end method
