*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract.smali	2012-03-23 19:35:26.367067689 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CalendarContract.smali.original	2012-03-23 19:34:44.592305927 -0500
***************
*** 63,68 ****
--- 63,69 ----
      .locals 1
  
      .prologue
+     .line 128
      const-string v0, "content://com.android.calendar"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 78,83 ****
--- 79,85 ----
      .locals 0
  
      .prologue
+     .line 158
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
