*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CallLog.smali	2012-03-23 19:35:27.868095059 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CallLog.smali.original	2012-03-23 19:34:44.604306145 -0500
***************
*** 22,27 ****
--- 22,28 ----
      .locals 1
  
      .prologue
+     .line 41
      const-string v0, "content://call_log"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 37,43 ****
--- 38,46 ----
      .locals 0
  
      .prologue
+     .line 35
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 47
      return-void
  .end method
