*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CallLog$Calls.smali	2012-03-23 19:35:28.504106656 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/CallLog$Calls.smali.original	2012-03-23 19:34:44.610306255 -0500
***************
*** 80,85 ****
--- 80,86 ----
      .locals 3
  
      .prologue
+     .line 51
      const-string v0, "content://call_log/calls"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 88,93 ****
--- 89,95 ----
  
      sput-object v0, Landroid/provider/CallLog$Calls;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 57
      const-string v0, "content://call_log/calls/filter"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 96,101 ****
--- 98,104 ----
  
      sput-object v0, Landroid/provider/CallLog$Calls;->CONTENT_FILTER_URI:Landroid/net/Uri;
  
+     .line 79
      sget-object v0, Landroid/provider/CallLog$Calls;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-virtual {v0}, Landroid/net/Uri;->buildUpon()Landroid/net/Uri$Builder;
***************
*** 104,110 ****
  
      const-string v1, "allow_voicemails"
  
!     const-string v2, "true"
  
      invoke-virtual {v0, v1, v2}, Landroid/net/Uri$Builder;->appendQueryParameter(Ljava/lang/String;Ljava/lang/String;)Landroid/net/Uri$Builder;
  
--- 107,113 ----
  
      const-string v1, "allow_voicemails"
  
!     const-string/jumbo v2, "true"
  
      invoke-virtual {v0, v1, v2}, Landroid/net/Uri$Builder;->appendQueryParameter(Ljava/lang/String;Ljava/lang/String;)Landroid/net/Uri$Builder;
  
***************
*** 123,128 ****
--- 126,132 ----
      .locals 0
  
      .prologue
+     .line 47
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 139,148 ****
--- 143,154 ----
      .parameter "duration"
  
      .prologue
+     .line 265
      invoke-virtual/range {p1 .. p1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v1
  
+     .line 269
      .local v1, resolver:Landroid/content/ContentResolver;
      sget v2, Lcom/android/internal/telephony/Connection;->PRESENTATION_RESTRICTED:I
  
***************
*** 150,163 ****
--- 156,172 ----
  
      if-ne v0, v2, :cond_5
  
+     .line 270
      const-string p2, "-2"
  
+     .line 271
      if-eqz p0, :cond_0
  
      const-string v2, ""
  
      iput-object v2, p0, Lcom/android/internal/telephony/CallerInfo;->name:Ljava/lang/String;
  
+     .line 281
      :cond_0
      :goto_0
      new-instance v12, Landroid/content/ContentValues;
***************
*** 166,179 ****
  
      invoke-direct {v12, v2}, Landroid/content/ContentValues;-><init>(I)V
  
      .local v12, values:Landroid/content/ContentValues;
!     const-string v2, "number"
  
      move-object/from16 v0, p2
  
      invoke-virtual {v12, v2, v0}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v2, "type"
  
      invoke-static/range {p4 .. p4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 175,190 ----
  
      invoke-direct {v12, v2}, Landroid/content/ContentValues;-><init>(I)V
  
+     .line 283
      .local v12, values:Landroid/content/ContentValues;
!     const-string/jumbo v2, "number"
  
      move-object/from16 v0, p2
  
      invoke-virtual {v12, v2, v0}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 284
!     const-string/jumbo v2, "type"
  
      invoke-static/range {p4 .. p4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 181,186 ****
--- 192,198 ----
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 285
      const-string v2, "date"
  
      invoke-static/range {p5 .. p6}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
***************
*** 189,194 ****
--- 201,207 ----
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
+     .line 286
      const-string v2, "duration"
  
      move/from16 v0, p7
***************
*** 201,207 ****
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
!     const-string v2, "new"
  
      const/4 v3, 0x1
  
--- 214,221 ----
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
!     .line 287
!     const-string/jumbo v2, "new"
  
      const/4 v3, 0x1
  
***************
*** 211,222 ****
--- 225,238 ----
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 288
      const/4 v2, 0x3
  
      move/from16 v0, p4
  
      if-ne v0, v2, :cond_1
  
+     .line 289
      const-string v2, "is_read"
  
      const/4 v3, 0x0
***************
*** 227,242 ****
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
      :cond_1
      if-eqz p0, :cond_2
  
!     const-string v2, "name"
  
      iget-object v3, p0, Lcom/android/internal/telephony/CallerInfo;->name:Ljava/lang/String;
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v2, "numbertype"
  
      iget v3, p0, Lcom/android/internal/telephony/CallerInfo;->numberType:I
  
--- 243,261 ----
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
+     .line 291
      :cond_1
      if-eqz p0, :cond_2
  
!     .line 292
!     const-string/jumbo v2, "name"
  
      iget-object v3, p0, Lcom/android/internal/telephony/CallerInfo;->name:Ljava/lang/String;
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 293
!     const-string/jumbo v2, "numbertype"
  
      iget v3, p0, Lcom/android/internal/telephony/CallerInfo;->numberType:I
  
***************
*** 246,257 ****
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     const-string v2, "numberlabel"
  
      iget-object v3, p0, Lcom/android/internal/telephony/CallerInfo;->numberLabel:Ljava/lang/String;
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
      :cond_2
      if-eqz p0, :cond_4
  
--- 265,278 ----
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Integer;)V
  
!     .line 294
!     const-string/jumbo v2, "numberlabel"
  
      iget-object v3, p0, Lcom/android/internal/telephony/CallerInfo;->numberLabel:Ljava/lang/String;
  
      invoke-virtual {v12, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 297
      :cond_2
      if-eqz p0, :cond_4
  
***************
*** 263,274 ****
--- 284,298 ----
  
      if-lez v2, :cond_4
  
+     .line 306
      iget-object v2, p0, Lcom/android/internal/telephony/CallerInfo;->normalizedNumber:Ljava/lang/String;
  
      if-eqz v2, :cond_8
  
+     .line 307
      iget-object v9, p0, Lcom/android/internal/telephony/CallerInfo;->normalizedNumber:Ljava/lang/String;
  
+     .line 308
      .local v9, normalizedPhoneNumber:Ljava/lang/String;
      sget-object v2, Landroid/provider/ContactsContract$CommonDataKinds$Phone;->CONTENT_URI:Landroid/net/Uri;
  
***************
*** 308,318 ****
--- 332,344 ----
  
      move-result-object v7
  
+     .line 322
      .end local v9           #normalizedPhoneNumber:Ljava/lang/String;
      .local v7, cursor:Landroid/database/Cursor;
      :goto_1
      if-eqz v7, :cond_4
  
+     .line 324
      :try_start_0
      invoke-interface {v7}, Landroid/database/Cursor;->getCount()I
  
***************
*** 326,331 ****
--- 352,358 ----
  
      if-eqz v2, :cond_3
  
+     .line 325
      sget-object v2, Landroid/provider/ContactsContract$DataUsageFeedback;->FEEDBACK_URI:Landroid/net/Uri;
  
      invoke-virtual {v2}, Landroid/net/Uri;->buildUpon()Landroid/net/Uri$Builder;
***************
*** 342,348 ****
  
      move-result-object v2
  
!     const-string v3, "type"
  
      const-string v4, "call"
  
--- 369,375 ----
  
      move-result-object v2
  
!     const-string/jumbo v3, "type"
  
      const-string v4, "call"
  
***************
*** 354,359 ****
--- 381,387 ----
  
      move-result-object v8
  
+     .line 330
      .local v8, feedbackUri:Landroid/net/Uri;
      new-instance v2, Landroid/content/ContentValues;
  
***************
*** 367,376 ****
--- 395,406 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 333
      .end local v8           #feedbackUri:Landroid/net/Uri;
      :cond_3
      invoke-interface {v7}, Landroid/database/Cursor;->close()V
  
+     .line 338
      .end local v7           #cursor:Landroid/database/Cursor;
      :cond_4
      sget-object v2, Landroid/provider/CallLog$Calls;->CONTENT_URI:Landroid/net/Uri;
***************
*** 379,389 ****
--- 409,422 ----
  
      move-result-object v11
  
+     .line 340
      .local v11, result:Landroid/net/Uri;
      invoke-static/range {p1 .. p1}, Landroid/provider/CallLog$Calls;->removeExpiredEntries(Landroid/content/Context;)V
  
+     .line 342
      return-object v11
  
+     .line 272
      .end local v11           #result:Landroid/net/Uri;
      .end local v12           #values:Landroid/content/ContentValues;
      :cond_5
***************
*** 393,400 ****
--- 426,435 ----
  
      if-ne v0, v2, :cond_6
  
+     .line 273
      const-string p2, "-3"
  
+     .line 274
      if-eqz p0, :cond_0
  
      const-string v2, ""
***************
*** 403,408 ****
--- 438,444 ----
  
      goto/16 :goto_0
  
+     .line 275
      :cond_6
      invoke-static/range {p2 .. p2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
  
***************
*** 416,424 ****
--- 452,462 ----
  
      if-ne v0, v2, :cond_0
  
+     .line 277
      :cond_7
      const-string p2, "-1"
  
+     .line 278
      if-eqz p0, :cond_0
  
      const-string v2, ""
***************
*** 427,432 ****
--- 465,471 ----
  
      goto/16 :goto_0
  
+     .line 314
      .restart local v12       #values:Landroid/content/ContentValues;
      :cond_8
      iget-object v2, p0, Lcom/android/internal/telephony/CallerInfo;->phoneNumber:Ljava/lang/String;
***************
*** 435,440 ****
--- 474,480 ----
  
      iget-object v10, p0, Lcom/android/internal/telephony/CallerInfo;->phoneNumber:Ljava/lang/String;
  
+     .line 315
      .local v10, phoneNumber:Ljava/lang/String;
      :goto_2
      sget-object v2, Landroid/provider/ContactsContract$CommonDataKinds$Phone;->CONTENT_URI:Landroid/net/Uri;
***************
*** 483,490 ****
--- 523,532 ----
      :cond_9
      move-object/from16 v10, p2
  
+     .line 314
      goto :goto_2
  
+     .line 333
      .restart local v7       #cursor:Landroid/database/Cursor;
      :catchall_0
      move-exception v2
***************
*** 499,511 ****
--- 541,556 ----
      .parameter "context"
  
      .prologue
+     .line 352
      invoke-virtual {p0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v0
  
+     .line 353
      .local v0, resolver:Landroid/content/ContentResolver;
      const/4 v6, 0x0
  
+     .line 355
      .local v6, c:Landroid/database/Cursor;
      :try_start_0
      sget-object v1, Landroid/provider/CallLog$Calls;->CONTENT_URI:Landroid/net/Uri;
***************
*** 516,526 ****
  
      const/4 v3, 0x0
  
!     const-string v4, "number"
  
      aput-object v4, v2, v3
  
!     const-string v3, "type = 2"
  
      const/4 v4, 0x0
  
--- 561,571 ----
  
      const/4 v3, 0x0
  
!     const-string/jumbo v4, "number"
  
      aput-object v4, v2, v3
  
!     const-string/jumbo v3, "type = 2"
  
      const/4 v4, 0x0
  
***************
*** 530,535 ****
--- 575,581 ----
  
      move-result-object v6
  
+     .line 361
      if-eqz v6, :cond_0
  
      invoke-interface {v6}, Landroid/database/Cursor;->moveToFirst()Z
***************
*** 538,548 ****
--- 584,596 ----
  
      if-nez v1, :cond_2
  
+     .line 362
      :cond_0
      const-string v1, ""
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 366
      if-eqz v6, :cond_1
  
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
***************
*** 551,556 ****
--- 599,605 ----
      :goto_0
      return-object v1
  
+     .line 364
      :cond_2
      const/4 v1, 0x0
  
***************
*** 561,566 ****
--- 610,616 ----
  
      move-result-object v1
  
+     .line 366
      if-eqz v6, :cond_1
  
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
***************
*** 583,592 ****
--- 633,644 ----
      .parameter "context"
  
      .prologue
+     .line 371
      invoke-virtual {p0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v0
  
+     .line 372
      .local v0, resolver:Landroid/content/ContentResolver;
      sget-object v1, Landroid/provider/CallLog$Calls;->CONTENT_URI:Landroid/net/Uri;
  
***************
*** 596,600 ****
--- 648,653 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/ContentResolver;->delete(Landroid/net/Uri;Ljava/lang/String;[Ljava/lang/String;)I
  
+     .line 375
      return-void
  .end method
