*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$Contacts$Entity.smali	2012-03-23 19:35:25.640054432 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$Contacts$Entity.smali.original	2012-03-23 19:34:44.586305817 -0500
***************
*** 39,45 ****
--- 39,47 ----
      .locals 0
  
      .prologue
+     .line 1656
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1657
      return-void
  .end method
