*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$Im.smali	2012-03-23 19:35:28.040098196 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$Im.smali.original	2012-03-23 19:34:44.606306182 -0500
***************
*** 57,62 ****
--- 57,63 ----
      .locals 0
  
      .prologue
+     .line 5920
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 69,74 ****
--- 70,76 ----
      .parameter "label"
  
      .prologue
+     .line 6007
      const/4 v1, -0x1
  
      if-ne p1, v1, :cond_0
***************
*** 79,94 ****
--- 81,99 ----
  
      if-nez v1, :cond_0
  
+     .line 6011
      .end local p2
      :goto_0
      return-object p2
  
+     .line 6010
      .restart local p2
      :cond_0
      invoke-static {p1}, Landroid/provider/ContactsContract$CommonDataKinds$Im;->getProtocolLabelResource(I)I
  
      move-result v0
  
+     .line 6011
      .local v0, labelRes:I
      invoke-virtual {p0, v0}, Landroid/content/res/Resources;->getText(I)Ljava/lang/CharSequence;
  
***************
*** 102,159 ****
--- 107,176 ----
      .parameter "type"
  
      .prologue
+     .line 5986
      packed-switch p0, :pswitch_data_0
  
+     .line 5996
      const v0, 0x10402b1
  
      :goto_0
      return v0
  
+     .line 5987
      :pswitch_0
      const v0, 0x10402b2
  
      goto :goto_0
  
+     .line 5988
      :pswitch_1
      const v0, 0x10402b3
  
      goto :goto_0
  
+     .line 5989
      :pswitch_2
      const v0, 0x10402b4
  
      goto :goto_0
  
+     .line 5990
      :pswitch_3
      const v0, 0x10402b5
  
      goto :goto_0
  
+     .line 5991
      :pswitch_4
      const v0, 0x10402b6
  
      goto :goto_0
  
+     .line 5992
      :pswitch_5
      const v0, 0x10402b7
  
      goto :goto_0
  
+     .line 5993
      :pswitch_6
      const v0, 0x10402b8
  
      goto :goto_0
  
+     .line 5994
      :pswitch_7
      const v0, 0x10402b9
  
      goto :goto_0
  
+     .line 5995
      :pswitch_8
      const v0, 0x10402ba
  
      goto :goto_0
  
+     .line 5986
      nop
  
      :pswitch_data_0
***************
*** 177,182 ****
--- 194,200 ----
      .parameter "label"
  
      .prologue
+     .line 5973
      if-nez p1, :cond_0
  
      invoke-static {p2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
***************
*** 185,200 ****
--- 203,221 ----
  
      if-nez v1, :cond_0
  
+     .line 5977
      .end local p2
      :goto_0
      return-object p2
  
+     .line 5976
      .restart local p2
      :cond_0
      invoke-static {p1}, Landroid/provider/ContactsContract$CommonDataKinds$Im;->getTypeLabelResource(I)I
  
      move-result v0
  
+     .line 5977
      .local v0, labelRes:I
      invoke-virtual {p0, v0}, Landroid/content/res/Resources;->getText(I)Ljava/lang/CharSequence;
  
***************
*** 208,235 ****
--- 229,262 ----
      .parameter "type"
  
      .prologue
+     .line 5958
      packed-switch p0, :pswitch_data_0
  
+     .line 5962
      const v0, 0x10402ad
  
      :goto_0
      return v0
  
+     .line 5959
      :pswitch_0
      const v0, 0x10402ae
  
      goto :goto_0
  
+     .line 5960
      :pswitch_1
      const v0, 0x10402af
  
      goto :goto_0
  
+     .line 5961
      :pswitch_2
      const v0, 0x10402b0
  
      goto :goto_0
  
+     .line 5958
      nop
  
      :pswitch_data_0
