*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$PhotoFiles.smali	2012-03-23 19:35:26.794075475 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$PhotoFiles.smali.original	2012-03-23 19:34:44.596305999 -0500
***************
*** 23,29 ****
--- 23,31 ----
      .locals 0
  
      .prologue
+     .line 3627
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 3628
      return-void
  .end method
