*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$SipAddress.smali	2012-03-23 19:35:27.974096993 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$SipAddress.smali.original	2012-03-23 19:34:44.605306163 -0500
***************
*** 35,40 ****
--- 35,41 ----
      .locals 0
  
      .prologue
+     .line 6677
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 47,52 ****
--- 48,54 ----
      .parameter "label"
  
      .prologue
+     .line 6713
      if-nez p1, :cond_0
  
      invoke-static {p2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
***************
*** 55,70 ****
--- 57,75 ----
  
      if-nez v1, :cond_0
  
+     .line 6717
      .end local p2
      :goto_0
      return-object p2
  
+     .line 6716
      .restart local p2
      :cond_0
      invoke-static {p1}, Landroid/provider/ContactsContract$CommonDataKinds$SipAddress;->getTypeLabelResource(I)I
  
      move-result v0
  
+     .line 6717
      .local v0, labelRes:I
      invoke-virtual {p0, v0}, Landroid/content/res/Resources;->getText(I)Ljava/lang/CharSequence;
  
***************
*** 78,105 ****
--- 83,116 ----
      .parameter "type"
  
      .prologue
+     .line 6698
      packed-switch p0, :pswitch_data_0
  
+     .line 6702
      const v0, 0x10402cd
  
      :goto_0
      return v0
  
+     .line 6699
      :pswitch_0
      const v0, 0x10402ce
  
      goto :goto_0
  
+     .line 6700
      :pswitch_1
      const v0, 0x10402cf
  
      goto :goto_0
  
+     .line 6701
      :pswitch_2
      const v0, 0x10402d0
  
      goto :goto_0
  
+     .line 6698
      nop
  
      :pswitch_data_0
