*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Contacts$Intents$Insert.smali	2012-03-23 19:35:28.619108754 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Contacts$Intents$Insert.smali.original	2012-03-23 19:34:44.611306273 -0500
***************
*** 161,167 ****
--- 161,169 ----
      .end annotation
  
      .prologue
+     .line 2199
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 2200
      return-void
  .end method
