*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Contacts$ContactMethods.smali	2012-03-23 19:35:30.196137512 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Contacts$ContactMethods.smali.original	2012-03-23 19:34:44.624306511 -0500
***************
*** 135,140 ****
--- 135,141 ----
      .locals 1
  
      .prologue
+     .line 1472
      const-string v0, "content://contacts/contact_methods"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 143,148 ****
--- 144,150 ----
  
      sput-object v0, Landroid/provider/Contacts$ContactMethods;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 1480
      const-string v0, "content://contacts/contact_methods/email"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 158,163 ****
--- 160,166 ----
      .locals 0
  
      .prologue
+     .line 1391
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 170,184 ****
      .end annotation
  
      .prologue
      if-nez p0, :cond_0
  
      const/4 v0, 0x0
  
      :goto_0
      return-object v0
  
      :cond_0
!     const-string v0, "pre:"
  
      invoke-virtual {p0, v0}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
  
--- 173,191 ----
      .end annotation
  
      .prologue
+     .line 1320
      if-nez p0, :cond_0
  
+     .line 1321
      const/4 v0, 0x0
  
+     .line 1329
      :goto_0
      return-object v0
  
+     .line 1324
      :cond_0
!     const-string/jumbo v0, "pre:"
  
      invoke-virtual {p0, v0}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
  
***************
*** 186,191 ****
--- 193,199 ----
  
      if-eqz v0, :cond_1
  
+     .line 1325
      const/4 v0, 0x4
  
      invoke-virtual {p0, v0}, Ljava/lang/String;->substring(I)Ljava/lang/String;
***************
*** 202,207 ****
--- 210,216 ----
  
      goto :goto_0
  
+     .line 1328
      :cond_1
      const-string v0, "custom:"
  
***************
*** 211,216 ****
--- 220,226 ----
  
      if-eqz v0, :cond_2
  
+     .line 1329
      const/4 v0, 0x7
  
      invoke-virtual {p0, v0}, Ljava/lang/String;->substring(I)Ljava/lang/String;
***************
*** 219,224 ****
--- 229,235 ----
  
      goto :goto_0
  
+     .line 1332
      :cond_2
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
***************
*** 226,232 ****
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v2, "the value is not a valid encoded protocol, "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 237,243 ----
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v2, "the value is not a valid encoded protocol, "
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 252,257 ****
--- 263,269 ----
      .end annotation
  
      .prologue
+     .line 1312
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 280,290 ****
      .end annotation
  
      .prologue
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string v1, "pre:"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 292,303 ----
      .end annotation
  
      .prologue
+     .line 1304
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
!     const-string/jumbo v1, "pre:"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 313,336 ****
--- 326,355 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 1399
      const-string v0, ""
  
+     .line 1400
      .local v0, display:Ljava/lang/CharSequence;
      packed-switch p1, :pswitch_data_0
  
+     .line 1436
      const v3, 0x104000f
  
      invoke-virtual {p0, v3}, Landroid/content/Context;->getString(I)Ljava/lang/String;
  
      move-result-object v0
  
+     .line 1438
      :cond_0
      :goto_0
      return-object v0
  
+     .line 1402
      :pswitch_0
      if-eqz p2, :cond_1
  
+     .line 1403
      invoke-virtual {p0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v3
***************
*** 341,346 ****
--- 360,366 ----
  
      move-result-object v2
  
+     .line 1406
      .local v2, labels:[Ljava/lang/CharSequence;
      add-int/lit8 v3, p2, -0x1
  
***************
*** 351,364 ****
--- 371,387 ----
  
      goto :goto_0
  
+     .line 1407
      :catch_0
      move-exception v1
  
+     .line 1408
      .local v1, e:Ljava/lang/ArrayIndexOutOfBoundsException;
      aget-object v0, v2, v5
  
      goto :goto_0
  
+     .line 1411
      .end local v1           #e:Ljava/lang/ArrayIndexOutOfBoundsException;
      .end local v2           #labels:[Ljava/lang/CharSequence;
      :cond_1
***************
*** 368,380 ****
--- 391,406 ----
  
      if-nez v3, :cond_0
  
+     .line 1412
      move-object v0, p3
  
      goto :goto_0
  
+     .line 1419
      :pswitch_1
      if-eqz p2, :cond_2
  
+     .line 1420
      invoke-virtual {p0}, Landroid/content/Context;->getResources()Landroid/content/res/Resources;
  
      move-result-object v3
***************
*** 385,390 ****
--- 411,417 ----
  
      move-result-object v2
  
+     .line 1423
      .restart local v2       #labels:[Ljava/lang/CharSequence;
      add-int/lit8 v3, p2, -0x1
  
***************
*** 395,408 ****
--- 422,438 ----
  
      goto :goto_0
  
+     .line 1424
      :catch_1
      move-exception v1
  
+     .line 1425
      .restart local v1       #e:Ljava/lang/ArrayIndexOutOfBoundsException;
      aget-object v0, v2, v5
  
      goto :goto_0
  
+     .line 1428
      .end local v1           #e:Ljava/lang/ArrayIndexOutOfBoundsException;
      .end local v2           #labels:[Ljava/lang/CharSequence;
      :cond_2
***************
*** 412,421 ****
--- 442,453 ----
  
      if-nez v3, :cond_0
  
+     .line 1429
      move-object v0, p3
  
      goto :goto_0
  
+     .line 1400
      nop
  
      :pswitch_data_0
***************
*** 432,484 ****
--- 464,527 ----
      .end annotation
  
      .prologue
+     .line 1367
      packed-switch p0, :pswitch_data_0
  
+     .line 1385
      const/4 v0, 0x0
  
      :goto_0
      return-object v0
  
+     .line 1369
      :pswitch_0
      const-string v0, "GTalk"
  
      goto :goto_0
  
+     .line 1371
      :pswitch_1
      const-string v0, "AIM"
  
      goto :goto_0
  
+     .line 1373
      :pswitch_2
      const-string v0, "MSN"
  
      goto :goto_0
  
+     .line 1375
      :pswitch_3
      const-string v0, "Yahoo"
  
      goto :goto_0
  
+     .line 1377
      :pswitch_4
      const-string v0, "ICQ"
  
      goto :goto_0
  
+     .line 1379
      :pswitch_5
      const-string v0, "JABBER"
  
      goto :goto_0
  
+     .line 1381
      :pswitch_6
      const-string v0, "SKYPE"
  
      goto :goto_0
  
+     .line 1383
      :pswitch_7
      const-string v0, "QQ"
  
      goto :goto_0
  
+     .line 1367
      nop
  
      :pswitch_data_0
***************
*** 508,517 ****
--- 551,562 ----
      .prologue
      const/4 v7, 0x0
  
+     .line 1453
      invoke-virtual {p1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v3
  
+     .line 1455
      .local v3, resolver:Landroid/content/ContentResolver;
      new-instance v4, Landroid/content/ContentValues;
  
***************
*** 519,524 ****
--- 564,570 ----
  
      invoke-direct {v4, v5}, Landroid/content/ContentValues;-><init>(I)V
  
+     .line 1456
      .local v4, values:Landroid/content/ContentValues;
      const-string v5, "data"
  
***************
*** 528,533 ****
--- 574,580 ----
  
      invoke-virtual {v4, v5, v6}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Double;)V
  
+     .line 1457
      const-string v5, "aux_data"
  
      invoke-static {p6, p7}, Ljava/lang/Double;->valueOf(D)Ljava/lang/Double;
***************
*** 536,555 ****
--- 583,606 ----
  
      invoke-virtual {v4, v5, v6}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Double;)V
  
+     .line 1458
      sget-object v5, Landroid/provider/Contacts$ContactMethods;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-virtual {v3, v5, v4}, Landroid/content/ContentResolver;->insert(Landroid/net/Uri;Landroid/content/ContentValues;)Landroid/net/Uri;
  
      move-result-object v0
  
+     .line 1459
      .local v0, loc:Landroid/net/Uri;
      invoke-static {v0}, Landroid/content/ContentUris;->parseId(Landroid/net/Uri;)J
  
      move-result-wide v1
  
+     .line 1462
      .local v1, locId:J
      invoke-virtual {v4}, Landroid/content/ContentValues;->clear()V
  
+     .line 1463
      const-string v5, "aux_data"
  
      invoke-static {v1, v2}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
***************
*** 558,563 ****
--- 609,615 ----
  
      invoke-virtual {v4, v5, v6}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
+     .line 1464
      sget-object v5, Landroid/provider/Contacts$ContactMethods;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-static {v5, p2, p3}, Landroid/content/ContentUris;->withAppendedId(Landroid/net/Uri;J)Landroid/net/Uri;
***************
*** 566,570 ****
--- 618,623 ----
  
      invoke-virtual {v3, v5, v4, v7, v7}, Landroid/content/ContentResolver;->update(Landroid/net/Uri;Landroid/content/ContentValues;Ljava/lang/String;[Ljava/lang/String;)I
  
+     .line 1465
      return-void
  .end method
