*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$Contacts.smali	2012-03-23 19:35:25.442050820 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$Contacts.smali.original	2012-03-23 19:34:44.584305781 -0500
***************
*** 64,69 ****
--- 64,70 ----
      .locals 2
  
      .prologue
+     .line 1411
      sget-object v0, Landroid/provider/ContactsContract;->AUTHORITY_URI:Landroid/net/Uri;
  
      const-string v1, "contacts"
***************
*** 74,79 ****
--- 75,81 ----
  
      sput-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 1429
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "lookup"
***************
*** 84,89 ****
--- 86,92 ----
  
      sput-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_LOOKUP_URI:Landroid/net/Uri;
  
+     .line 1440
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "as_vcard"
***************
*** 94,99 ****
--- 97,103 ----
  
      sput-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_VCARD_URI:Landroid/net/Uri;
  
+     .line 1468
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "as_multi_vcard"
***************
*** 104,109 ****
--- 108,114 ----
  
      sput-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_MULTI_VCARD_URI:Landroid/net/Uri;
  
+     .line 1556
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "filter"
***************
*** 114,122 ****
  
      sput-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_FILTER_URI:Landroid/net/Uri;
  
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
!     const-string v1, "strequent"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
--- 119,128 ----
  
      sput-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_FILTER_URI:Landroid/net/Uri;
  
+     .line 1564
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
!     const-string/jumbo v1, "strequent"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
***************
*** 124,129 ****
--- 130,136 ----
  
      sput-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_STREQUENT_URI:Landroid/net/Uri;
  
+     .line 1571
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "frequent"
***************
*** 134,139 ****
--- 141,147 ----
  
      sput-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_FREQUENT_URI:Landroid/net/Uri;
  
+     .line 1580
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_STREQUENT_URI:Landroid/net/Uri;
  
      const-string v1, "filter"
***************
*** 144,149 ****
--- 152,158 ----
  
      sput-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_STREQUENT_FILTER_URI:Landroid/net/Uri;
  
+     .line 1583
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "group"
***************
*** 161,166 ****
--- 170,176 ----
      .locals 0
  
      .prologue
+     .line 1406
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 172,177 ****
--- 182,188 ----
      .parameter "lookupKey"
  
      .prologue
+     .line 1503
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_LOOKUP_URI:Landroid/net/Uri;
  
      invoke-static {v0, p2}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
***************
*** 197,202 ****
--- 208,214 ----
  
      const/4 v3, 0x0
  
+     .line 1479
      const/4 v0, 0x2
  
      new-array v2, v0, [Ljava/lang/String;
***************
*** 221,232 ****
--- 233,247 ----
  
      move-result-object v6
  
+     .line 1482
      .local v6, c:Landroid/database/Cursor;
      if-nez v6, :cond_0
  
+     .line 1495
      :goto_0
      return-object v3
  
+     .line 1487
      :cond_0
      :try_start_0
      invoke-interface {v6}, Landroid/database/Cursor;->moveToFirst()Z
***************
*** 235,246 ****
--- 250,263 ----
  
      if-eqz v0, :cond_1
  
+     .line 1488
      const/4 v0, 0x0
  
      invoke-interface {v6, v0}, Landroid/database/Cursor;->getString(I)Ljava/lang/String;
  
      move-result-object v9
  
+     .line 1489
      .local v9, lookupKey:Ljava/lang/String;
      const/4 v0, 0x1
  
***************
*** 248,253 ****
--- 265,271 ----
  
      move-result-wide v7
  
+     .line 1490
      .local v7, contactId:J
      invoke-static {v7, v8, v9}, Landroid/provider/ContactsContract$Contacts;->getLookupUri(JLjava/lang/String;)Landroid/net/Uri;
      :try_end_0
***************
*** 255,260 ****
--- 273,279 ----
  
      move-result-object v3
  
+     .line 1493
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
***************
*** 284,295 ****
--- 303,317 ----
  
      const/4 v3, 0x0
  
+     .line 1513
      if-nez p1, :cond_1
  
+     .line 1530
      :cond_0
      :goto_0
      return-object v3
  
+     .line 1517
      :cond_1
      const/4 v0, 0x1
  
***************
*** 311,319 ****
--- 333,343 ----
  
      move-result-object v6
  
+     .line 1518
      .local v6, c:Landroid/database/Cursor;
      if-eqz v6, :cond_0
  
+     .line 1523
      :try_start_0
      invoke-interface {v6}, Landroid/database/Cursor;->moveToFirst()Z
  
***************
*** 321,332 ****
--- 345,358 ----
  
      if-eqz v0, :cond_2
  
+     .line 1524
      const/4 v0, 0x0
  
      invoke-interface {v6, v0}, Landroid/database/Cursor;->getLong(I)J
  
      move-result-wide v7
  
+     .line 1525
      .local v7, contactId:J
      sget-object v0, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
***************
*** 336,341 ****
--- 362,368 ----
  
      move-result-object v3
  
+     .line 1528
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
***************
*** 362,378 ****
--- 389,408 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 1543
      sget-object v2, Landroid/provider/ContactsContract$Contacts;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-static {v2, p1, p2}, Landroid/content/ContentUris;->withAppendedId(Landroid/net/Uri;J)Landroid/net/Uri;
  
      move-result-object v0
  
+     .line 1544
      .local v0, uri:Landroid/net/Uri;
      new-instance v1, Landroid/content/ContentValues;
  
      invoke-direct {v1}, Landroid/content/ContentValues;-><init>()V
  
+     .line 1546
      .local v1, values:Landroid/content/ContentValues;
      const-string v2, "last_time_contacted"
  
***************
*** 386,393 ****
--- 416,425 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/Long;)V
  
+     .line 1547
      invoke-virtual {p0, v0, v1, v5, v5}, Landroid/content/ContentResolver;->update(Landroid/net/Uri;Landroid/content/ContentValues;Ljava/lang/String;[Ljava/lang/String;)I
  
+     .line 1548
      return-void
  .end method
  
***************
*** 397,402 ****
--- 429,435 ----
      .parameter "contactUri"
  
      .prologue
+     .line 2001
      const/4 v0, 0x0
  
      invoke-static {p0, p1, v0}, Landroid/provider/ContactsContract$Contacts;->openContactPhotoInputStream(Landroid/content/ContentResolver;Landroid/net/Uri;Z)Ljava/io/InputStream;
***************
*** 417,438 ****
  
      const/4 v3, 0x0
  
      if-eqz p2, :cond_1
  
      const-string v0, "display_photo"
  
      invoke-static {p1, v0}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v8
  
      .local v8, displayPhotoUri:Landroid/net/Uri;
      :try_start_0
!     const-string v0, "r"
  
      invoke-virtual {p0, v8, v0}, Landroid/content/ContentResolver;->openAssetFileDescriptor(Landroid/net/Uri;Ljava/lang/String;)Landroid/content/res/AssetFileDescriptor;
  
      move-result-object v9
  
      .local v9, fd:Landroid/content/res/AssetFileDescriptor;
      invoke-virtual {v9}, Landroid/content/res/AssetFileDescriptor;->createInputStream()Ljava/io/FileInputStream;
      :try_end_0
--- 450,475 ----
  
      const/4 v3, 0x0
  
+     .line 1954
      if-eqz p2, :cond_1
  
+     .line 1955
      const-string v0, "display_photo"
  
      invoke-static {p1, v0}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v8
  
+     .line 1959
      .local v8, displayPhotoUri:Landroid/net/Uri;
      :try_start_0
!     const-string/jumbo v0, "r"
  
      invoke-virtual {p0, v8, v0}, Landroid/content/ContentResolver;->openAssetFileDescriptor(Landroid/net/Uri;Ljava/lang/String;)Landroid/content/res/AssetFileDescriptor;
  
      move-result-object v9
  
+     .line 1960
      .local v9, fd:Landroid/content/res/AssetFileDescriptor;
      invoke-virtual {v9}, Landroid/content/res/AssetFileDescriptor;->createInputStream()Ljava/io/FileInputStream;
      :try_end_0
***************
*** 440,466 ****
  
      move-result-object v3
  
      .end local v8           #displayPhotoUri:Landroid/net/Uri;
      .end local v9           #fd:Landroid/content/res/AssetFileDescriptor;
      :cond_0
      :goto_0
      return-object v3
  
      .restart local v8       #displayPhotoUri:Landroid/net/Uri;
      :catch_0
      move-exception v0
  
      .end local v8           #displayPhotoUri:Landroid/net/Uri;
      :cond_1
!     const-string v0, "photo"
  
      invoke-static {p1, v0}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v1
  
      .local v1, photoUri:Landroid/net/Uri;
      if-eqz v1, :cond_0
  
      const/4 v0, 0x1
  
      new-array v2, v0, [Ljava/lang/String;
--- 477,508 ----
  
      move-result-object v3
  
+     .line 1985
      .end local v8           #displayPhotoUri:Landroid/net/Uri;
      .end local v9           #fd:Landroid/content/res/AssetFileDescriptor;
      :cond_0
      :goto_0
      return-object v3
  
+     .line 1961
      .restart local v8       #displayPhotoUri:Landroid/net/Uri;
      :catch_0
      move-exception v0
  
+     .line 1966
      .end local v8           #displayPhotoUri:Landroid/net/Uri;
      :cond_1
!     const-string/jumbo v0, "photo"
  
      invoke-static {p1, v0}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v1
  
+     .line 1967
      .local v1, photoUri:Landroid/net/Uri;
      if-eqz v1, :cond_0
  
+     .line 1970
      const/4 v0, 0x1
  
      new-array v2, v0, [Ljava/lang/String;
***************
*** 479,484 ****
--- 521,527 ----
  
      move-result-object v6
  
+     .line 1975
      .local v6, cursor:Landroid/database/Cursor;
      if-eqz v6, :cond_2
  
***************
*** 491,503 ****
--- 534,549 ----
  
      if-nez v0, :cond_3
  
+     .line 1984
      :cond_2
      if-eqz v6, :cond_0
  
+     .line 1985
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 1978
      :cond_3
      const/4 v0, 0x0
  
***************
*** 508,522 ****
--- 554,572 ----
  
      move-result-object v7
  
+     .line 1979
      .local v7, data:[B
      if-nez v7, :cond_4
  
+     .line 1984
      if-eqz v6, :cond_0
  
+     .line 1985
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 1982
      :cond_4
      :try_start_3
      new-instance v3, Ljava/io/ByteArrayInputStream;
***************
*** 525,542 ****
--- 575,596 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
+     .line 1984
      if-eqz v6, :cond_0
  
+     .line 1985
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      goto :goto_0
  
+     .line 1984
      .end local v7           #data:[B
      :catchall_0
      move-exception v0
  
      if-eqz v6, :cond_5
  
+     .line 1985
      invoke-interface {v6}, Landroid/database/Cursor;->close()V
  
      :cond_5
