*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$SearchSnippetColumns.smali	2012-03-23 19:35:27.553089315 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$SearchSnippetColumns.smali.original	2012-03-23 19:34:44.601306091 -0500
***************
*** 27,32 ****
--- 27,33 ----
      .locals 0
  
      .prologue
+     .line 4962
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
