*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Contacts$People$ContactMethods.smali	2012-03-23 19:35:26.569071372 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Contacts$People$ContactMethods.smali.original	2012-03-23 19:34:44.593305945 -0500
***************
*** 39,44 ****
--- 39,45 ----
      .locals 0
  
      .prologue
+     .line 695
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
