*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$SyncState.smali	2012-03-23 19:35:26.963078556 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$SyncState.smali.original	2012-03-23 19:34:44.597306017 -0500
***************
*** 28,36 ****
      .locals 2
  
      .prologue
      sget-object v0, Landroid/provider/ContactsContract;->AUTHORITY_URI:Landroid/net/Uri;
  
!     const-string v1, "syncstate"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
--- 28,37 ----
      .locals 2
  
      .prologue
+     .line 646
      sget-object v0, Landroid/provider/ContactsContract;->AUTHORITY_URI:Landroid/net/Uri;
  
!     const-string/jumbo v1, "syncstate"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
***************
*** 45,50 ****
--- 46,52 ----
      .locals 0
  
      .prologue
+     .line 638
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 61,66 ****
--- 63,69 ----
      .end annotation
  
      .prologue
+     .line 654
      sget-object v0, Landroid/provider/ContactsContract$SyncState;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-static {p0, v0, p1}, Landroid/provider/SyncStateContract$Helpers;->get(Landroid/content/ContentProviderClient;Landroid/net/Uri;Landroid/accounts/Account;)[B
***************
*** 94,99 ****
--- 97,103 ----
      .end annotation
  
      .prologue
+     .line 662
      sget-object v0, Landroid/provider/ContactsContract$SyncState;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-static {p0, v0, p1}, Landroid/provider/SyncStateContract$Helpers;->getWithUri(Landroid/content/ContentProviderClient;Landroid/net/Uri;Landroid/accounts/Account;)Landroid/util/Pair;
***************
*** 109,114 ****
--- 113,119 ----
      .parameter "data"
  
      .prologue
+     .line 677
      sget-object v0, Landroid/provider/ContactsContract$SyncState;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-static {v0, p0, p1}, Landroid/provider/SyncStateContract$Helpers;->newSetOperation(Landroid/net/Uri;Landroid/accounts/Account;[B)Landroid/content/ContentProviderOperation;
***************
*** 130,138 ****
--- 135,145 ----
      .end annotation
  
      .prologue
+     .line 670
      sget-object v0, Landroid/provider/ContactsContract$SyncState;->CONTENT_URI:Landroid/net/Uri;
  
      invoke-static {p0, v0, p1, p2}, Landroid/provider/SyncStateContract$Helpers;->set(Landroid/content/ContentProviderClient;Landroid/net/Uri;Landroid/accounts/Account;[B)V
  
+     .line 671
      return-void
  .end method
