*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$Email.smali	2012-03-23 19:35:30.151136690 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$Email.smali.original	2012-03-23 19:34:44.624306511 -0500
***************
*** 47,52 ****
--- 47,53 ----
      .locals 2
  
      .prologue
+     .line 5549
      sget-object v0, Landroid/provider/ContactsContract$Data;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "emails"
***************
*** 57,62 ****
--- 58,64 ----
  
      sput-object v0, Landroid/provider/ContactsContract$CommonDataKinds$Email;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 5567
      sget-object v0, Landroid/provider/ContactsContract$CommonDataKinds$Email;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "lookup"
***************
*** 67,72 ****
--- 69,75 ----
  
      sput-object v0, Landroid/provider/ContactsContract$CommonDataKinds$Email;->CONTENT_LOOKUP_URI:Landroid/net/Uri;
  
+     .line 5587
      sget-object v0, Landroid/provider/ContactsContract$CommonDataKinds$Email;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "filter"
***************
*** 84,89 ****
--- 87,93 ----
      .locals 0
  
      .prologue
+     .line 5534
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 96,101 ****
--- 100,106 ----
      .parameter "label"
  
      .prologue
+     .line 5628
      if-nez p1, :cond_0
  
      invoke-static {p2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
***************
*** 104,119 ****
--- 109,127 ----
  
      if-nez v1, :cond_0
  
+     .line 5632
      .end local p2
      :goto_0
      return-object p2
  
+     .line 5631
      .restart local p2
      :cond_0
      invoke-static {p1}, Landroid/provider/ContactsContract$CommonDataKinds$Email;->getTypeLabelResource(I)I
  
      move-result v0
  
+     .line 5632
      .local v0, labelRes:I
      invoke-virtual {p0, v0}, Landroid/content/res/Resources;->getText(I)Ljava/lang/CharSequence;
  
***************
*** 127,159 ****
--- 135,174 ----
      .parameter "type"
  
      .prologue
+     .line 5612
      packed-switch p0, :pswitch_data_0
  
+     .line 5617
      const v0, 0x10402a4
  
      :goto_0
      return v0
  
+     .line 5613
      :pswitch_0
      const v0, 0x10402a5
  
      goto :goto_0
  
+     .line 5614
      :pswitch_1
      const v0, 0x10402a6
  
      goto :goto_0
  
+     .line 5615
      :pswitch_2
      const v0, 0x10402a7
  
      goto :goto_0
  
+     .line 5616
      :pswitch_3
      const v0, 0x10402a8
  
      goto :goto_0
  
+     .line 5612
      nop
  
      :pswitch_data_0
