*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$StructuredPostal.smali	2012-03-23 19:35:28.942114644 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$StructuredPostal.smali.original	2012-03-23 19:34:44.613306309 -0500
***************
*** 53,61 ****
      .locals 2
  
      .prologue
      sget-object v0, Landroid/provider/ContactsContract$Data;->CONTENT_URI:Landroid/net/Uri;
  
!     const-string v1, "postals"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
--- 53,62 ----
      .locals 2
  
      .prologue
+     .line 5743
      sget-object v0, Landroid/provider/ContactsContract$Data;->CONTENT_URI:Landroid/net/Uri;
  
!     const-string/jumbo v1, "postals"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
***************
*** 70,77 ****
--- 71,80 ----
      .locals 0
  
      .prologue
+     .line 5726
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 5727
      return-void
  .end method
  
***************
*** 82,87 ****
--- 85,91 ----
      .parameter "label"
  
      .prologue
+     .line 5835
      if-nez p1, :cond_0
  
      invoke-static {p2}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
***************
*** 90,105 ****
--- 94,112 ----
  
      if-nez v1, :cond_0
  
+     .line 5839
      .end local p2
      :goto_0
      return-object p2
  
+     .line 5838
      .restart local p2
      :cond_0
      invoke-static {p1}, Landroid/provider/ContactsContract$CommonDataKinds$StructuredPostal;->getTypeLabelResource(I)I
  
      move-result v0
  
+     .line 5839
      .local v0, labelRes:I
      invoke-virtual {p0, v0}, Landroid/content/res/Resources;->getText(I)Ljava/lang/CharSequence;
  
***************
*** 113,140 ****
--- 120,153 ----
      .parameter "type"
  
      .prologue
+     .line 5820
      packed-switch p0, :pswitch_data_0
  
+     .line 5824
      const v0, 0x10402a9
  
      :goto_0
      return v0
  
+     .line 5821
      :pswitch_0
      const v0, 0x10402aa
  
      goto :goto_0
  
+     .line 5822
      :pswitch_1
      const v0, 0x10402ab
  
      goto :goto_0
  
+     .line 5823
      :pswitch_2
      const v0, 0x10402ac
  
      goto :goto_0
  
+     .line 5820
      nop
  
      :pswitch_data_0
