*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$Photo.smali	2012-03-23 19:35:28.378104360 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$CommonDataKinds$Photo.smali.original	2012-03-23 19:34:44.609306237 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .locals 0
  
      .prologue
+     .line 6435
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
