*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$Intents$Insert.smali	2012-03-23 19:35:28.988115483 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$Intents$Insert.smali.original	2012-03-23 19:34:44.614306327 -0500
***************
*** 81,86 ****
--- 81,87 ----
      .locals 0
  
      .prologue
+     .line 7973
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
