*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Contacts$GroupMembership.smali	2012-03-23 19:35:25.793057221 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Contacts$GroupMembership.smali.original	2012-03-23 19:34:44.587305835 -0500
***************
*** 83,88 ****
--- 83,89 ----
      .locals 1
  
      .prologue
+     .line 1053
      const-string v0, "content://contacts/groupmembership"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 91,96 ****
--- 92,98 ----
  
      sput-object v0, Landroid/provider/Contacts$GroupMembership;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 1061
      const-string v0, "content://contacts/groupmembershipraw"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 106,111 ****
--- 108,114 ----
      .locals 0
  
      .prologue
+     .line 1046
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
