*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$PhoneLookup.smali	2012-03-23 19:35:26.409068455 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$PhoneLookup.smali.original	2012-03-23 19:34:44.592305927 -0500
***************
*** 33,41 ****
      .locals 2
  
      .prologue
      sget-object v0, Landroid/provider/ContactsContract;->AUTHORITY_URI:Landroid/net/Uri;
  
!     const-string v1, "phone_lookup"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
--- 33,42 ----
      .locals 2
  
      .prologue
+     .line 4666
      sget-object v0, Landroid/provider/ContactsContract;->AUTHORITY_URI:Landroid/net/Uri;
  
!     const-string/jumbo v1, "phone_lookup"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
***************
*** 50,55 ****
--- 51,57 ----
      .locals 0
  
      .prologue
+     .line 4657
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
