*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$StreamItems$StreamItemPhotos.smali	2012-03-23 19:35:30.316139701 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/ContactsContract$StreamItems$StreamItemPhotos.smali.original	2012-03-23 19:34:44.625306529 -0500
***************
*** 31,37 ****
--- 31,39 ----
      .locals 0
  
      .prologue
+     .line 3234
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 3235
      return-void
  .end method
