*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Downloads.smali	2012-03-23 19:35:30.981151826 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Downloads.smali.original	2012-03-23 19:34:44.631306637 -0500
***************
*** 16,21 ****
--- 16,22 ----
      .locals 0
  
      .prologue
+     .line 29
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
