*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Downloads$Impl.smali	2012-03-23 19:35:26.502070150 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Downloads$Impl.smali.original	2012-03-23 19:34:44.593305945 -0500
***************
*** 198,203 ****
--- 198,204 ----
      .locals 1
  
      .prologue
+     .line 88
      const-string v0, "content://downloads/my_downloads"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 206,211 ****
--- 207,213 ----
  
      sput-object v0, Landroid/provider/Downloads$Impl;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 95
      const-string v0, "content://downloads/all_downloads"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 214,219 ****
--- 216,222 ----
  
      sput-object v0, Landroid/provider/Downloads$Impl;->ALL_DOWNLOADS_CONTENT_URI:Landroid/net/Uri;
  
+     .line 105
      const-string v0, "content://downloads/public_downloads"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 229,234 ****
--- 232,238 ----
      .locals 0
  
      .prologue
+     .line 41
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 241,246 ****
--- 245,251 ----
      .prologue
      const/4 v0, 0x1
  
+     .line 538
      if-eq p0, v0, :cond_0
  
      const/4 v1, 0x3
***************
*** 262,267 ****
--- 267,273 ----
      .parameter "status"
  
      .prologue
+     .line 521
      const/16 v0, 0x190
  
      if-lt p0, v0, :cond_0
***************
*** 286,291 ****
--- 292,298 ----
      .parameter "status"
  
      .prologue
+     .line 547
      const/16 v0, 0xc8
  
      if-lt p0, v0, :cond_0
***************
*** 320,325 ****
--- 327,333 ----
      .parameter "status"
  
      .prologue
+     .line 514
      const/16 v0, 0x190
  
      if-lt p0, v0, :cond_0
***************
*** 344,349 ****
--- 352,358 ----
      .parameter "status"
  
      .prologue
+     .line 500
      const/16 v0, 0x64
  
      if-lt p0, v0, :cond_0
***************
*** 368,373 ****
--- 377,383 ----
      .parameter "status"
  
      .prologue
+     .line 528
      const/16 v0, 0x1f4
  
      if-lt p0, v0, :cond_0
***************
*** 392,397 ****
--- 402,408 ----
      .parameter "status"
  
      .prologue
+     .line 507
      const/16 v0, 0xc8
  
      if-lt p0, v0, :cond_0
