*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/DrmStore$Audio.smali	2012-03-23 19:35:30.538143748 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/DrmStore$Audio.smali.original	2012-03-23 19:34:44.628306583 -0500
***************
*** 26,31 ****
--- 26,32 ----
      .locals 1
  
      .prologue
+     .line 91
      const-string v0, "content://drm/audio"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
