*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/MediaStore$Video$Thumbnails.smali	2012-03-23 19:35:26.039061707 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/MediaStore$Video$Thumbnails.smali.original	2012-03-23 19:34:44.589305872 -0500
***************
*** 46,51 ****
--- 46,52 ----
      .locals 1
  
      .prologue
+     .line 1985
      const-string v0, "internal"
  
      invoke-static {v0}, Landroid/provider/MediaStore$Video$Thumbnails;->getContentUri(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 54,59 ****
--- 55,61 ----
  
      sput-object v0, Landroid/provider/MediaStore$Video$Thumbnails;->INTERNAL_CONTENT_URI:Landroid/net/Uri;
  
+     .line 1992
      const-string v0, "external"
  
      invoke-static {v0}, Landroid/provider/MediaStore$Video$Thumbnails;->getContentUri(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 69,74 ****
--- 71,77 ----
      .locals 0
  
      .prologue
+     .line 1907
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 80,85 ****
--- 83,89 ----
      .parameter "origId"
  
      .prologue
+     .line 1917
      sget-object v3, Landroid/provider/MediaStore$Video$Thumbnails;->EXTERNAL_CONTENT_URI:Landroid/net/Uri;
  
      const-wide/16 v4, 0x0
***************
*** 90,95 ****
--- 94,100 ----
  
      invoke-static/range {v0 .. v5}, Landroid/provider/MediaStore$InternalThumbnails;->cancelThumbnailRequest(Landroid/content/ContentResolver;JLandroid/net/Uri;J)V
  
+     .line 1919
      return-void
  .end method
  
***************
*** 100,105 ****
--- 105,111 ----
      .parameter "groupId"
  
      .prologue
+     .line 1967
      sget-object v3, Landroid/provider/MediaStore$Video$Thumbnails;->EXTERNAL_CONTENT_URI:Landroid/net/Uri;
  
      move-object v0, p0
***************
*** 110,115 ****
--- 116,122 ----
  
      invoke-static/range {v0 .. v5}, Landroid/provider/MediaStore$InternalThumbnails;->cancelThumbnailRequest(Landroid/content/ContentResolver;JLandroid/net/Uri;J)V
  
+     .line 1968
      return-void
  .end method
  
***************
*** 118,123 ****
--- 125,131 ----
      .parameter "volumeName"
  
      .prologue
+     .line 1978
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 157,162 ****
--- 165,171 ----
      .parameter "options"
  
      .prologue
+     .line 1934
      const-wide/16 v3, 0x0
  
      sget-object v7, Landroid/provider/MediaStore$Video$Thumbnails;->EXTERNAL_CONTENT_URI:Landroid/net/Uri;
***************
*** 187,192 ****
--- 196,202 ----
      .parameter "options"
  
      .prologue
+     .line 1953
      sget-object v7, Landroid/provider/MediaStore$Video$Thumbnails;->EXTERNAL_CONTENT_URI:Landroid/net/Uri;
  
      const/4 v8, 0x1
