*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/MediaStore$Audio$Media.smali	2012-03-23 19:35:26.153063786 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/MediaStore$Audio$Media.smali.original	2012-03-23 19:34:44.590305891 -0500
***************
*** 36,41 ****
--- 36,42 ----
      .locals 1
  
      .prologue
+     .line 1260
      const-string v0, "internal"
  
      invoke-static {v0}, Landroid/provider/MediaStore$Audio$Media;->getContentUri(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 44,49 ****
--- 45,51 ----
  
      sput-object v0, Landroid/provider/MediaStore$Audio$Media;->INTERNAL_CONTENT_URI:Landroid/net/Uri;
  
+     .line 1267
      const-string v0, "external"
  
      invoke-static {v0}, Landroid/provider/MediaStore$Audio$Media;->getContentUri(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 59,64 ****
--- 61,67 ----
      .locals 0
  
      .prologue
+     .line 1239
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 69,74 ****
--- 72,78 ----
      .parameter "volumeName"
  
      .prologue
+     .line 1248
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 105,110 ****
--- 109,115 ----
      .parameter "path"
  
      .prologue
+     .line 1253
      invoke-static {}, Landroid/os/Environment;->getExternalStorageDirectory()Ljava/io/File;
  
      move-result-object v0
