*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/MediaStore$Audio$Albums.smali	2012-03-23 19:35:27.221083262 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/MediaStore$Audio$Albums.smali.original	2012-03-23 19:34:44.599306055 -0500
***************
*** 35,40 ****
--- 35,41 ----
      .locals 1
  
      .prologue
+     .line 1721
      const-string v0, "internal"
  
      invoke-static {v0}, Landroid/provider/MediaStore$Audio$Albums;->getContentUri(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 43,48 ****
--- 44,50 ----
  
      sput-object v0, Landroid/provider/MediaStore$Audio$Albums;->INTERNAL_CONTENT_URI:Landroid/net/Uri;
  
+     .line 1728
      const-string v0, "external"
  
      invoke-static {v0}, Landroid/provider/MediaStore$Audio$Albums;->getContentUri(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 58,63 ****
--- 60,66 ----
      .locals 0
  
      .prologue
+     .line 1705
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 68,73 ****
--- 71,77 ----
      .parameter "volumeName"
  
      .prologue
+     .line 1714
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
