*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/SearchRecentSuggestions$1.smali	2012-03-23 19:35:25.489051677 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/SearchRecentSuggestions$1.smali.original	2012-03-23 19:34:44.585305799 -0500
***************
*** 31,36 ****
--- 31,37 ----
      .parameter
  
      .prologue
+     .line 162
      iput-object p1, p0, Landroid/provider/SearchRecentSuggestions$1;->this$0:Landroid/provider/SearchRecentSuggestions;
  
      iput-object p3, p0, Landroid/provider/SearchRecentSuggestions$1;->val$queryString:Ljava/lang/String;
***************
*** 48,53 ****
--- 49,55 ----
      .locals 3
  
      .prologue
+     .line 165
      iget-object v0, p0, Landroid/provider/SearchRecentSuggestions$1;->this$0:Landroid/provider/SearchRecentSuggestions;
  
      iget-object v1, p0, Landroid/provider/SearchRecentSuggestions$1;->val$queryString:Ljava/lang/String;
***************
*** 57,67 ****
--- 59,71 ----
      #calls: Landroid/provider/SearchRecentSuggestions;->saveRecentQueryBlocking(Ljava/lang/String;Ljava/lang/String;)V
      invoke-static {v0, v1, v2}, Landroid/provider/SearchRecentSuggestions;->access$000(Landroid/provider/SearchRecentSuggestions;Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 166
      invoke-static {}, Landroid/provider/SearchRecentSuggestions;->access$100()Ljava/util/concurrent/Semaphore;
  
      move-result-object v0
  
      invoke-virtual {v0}, Ljava/util/concurrent/Semaphore;->release()V
  
+     .line 167
      return-void
  .end method
