*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Settings.smali	2012-03-23 19:35:25.702055561 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Settings.smali.original	2012-03-23 19:34:44.587305835 -0500
***************
*** 111,118 ****
--- 111,120 ----
      .locals 0
  
      .prologue
+     .line 54
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 4163
      return-void
  .end method
  
***************
*** 121,126 ****
--- 123,129 ----
      .parameter "androidId"
  
      .prologue
+     .line 4368
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
