*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Settings$NameValueTable.smali	2012-03-23 19:35:27.403086581 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Settings$NameValueTable.smali.original	2012-03-23 19:34:44.600306073 -0500
***************
*** 28,33 ****
--- 28,34 ----
      .locals 0
  
      .prologue
+     .line 628
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 39,44 ****
--- 40,46 ----
      .parameter "name"
  
      .prologue
+     .line 648
      invoke-static {p0, p1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
      move-result-object v0
***************
*** 54,86 ****
      .parameter "value"
  
      .prologue
      :try_start_0
      new-instance v1, Landroid/content/ContentValues;
  
      invoke-direct {v1}, Landroid/content/ContentValues;-><init>()V
  
      .local v1, values:Landroid/content/ContentValues;
!     const-string v2, "name"
  
      invoke-virtual {v1, v2, p2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     const-string v2, "value"
  
      invoke-virtual {v1, v2, p3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
      invoke-virtual {p0, p1, v1}, Landroid/content/ContentResolver;->insert(Landroid/net/Uri;Landroid/content/ContentValues;)Landroid/net/Uri;
      :try_end_0
      .catch Landroid/database/SQLException; {:try_start_0 .. :try_end_0} :catch_0
  
      const/4 v2, 0x1
  
      .end local v1           #values:Landroid/content/ContentValues;
      :goto_0
      return v2
  
      :catch_0
      move-exception v0
  
      .local v0, e:Landroid/database/SQLException;
      const-string v2, "Settings"
  
--- 56,96 ----
      .parameter "value"
  
      .prologue
+     .line 636
      :try_start_0
      new-instance v1, Landroid/content/ContentValues;
  
      invoke-direct {v1}, Landroid/content/ContentValues;-><init>()V
  
+     .line 637
      .local v1, values:Landroid/content/ContentValues;
!     const-string/jumbo v2, "name"
  
      invoke-virtual {v1, v2, p2}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
!     .line 638
!     const-string/jumbo v2, "value"
  
      invoke-virtual {v1, v2, p3}, Landroid/content/ContentValues;->put(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 639
      invoke-virtual {p0, p1, v1}, Landroid/content/ContentResolver;->insert(Landroid/net/Uri;Landroid/content/ContentValues;)Landroid/net/Uri;
      :try_end_0
      .catch Landroid/database/SQLException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 640
      const/4 v2, 0x1
  
+     .line 643
      .end local v1           #values:Landroid/content/ContentValues;
      :goto_0
      return v2
  
+     .line 641
      :catch_0
      move-exception v0
  
+     .line 642
      .local v0, e:Landroid/database/SQLException;
      const-string v2, "Settings"
  
***************
*** 114,119 ****
--- 124,130 ----
  
      invoke-static {v2, v3, v0}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 643
      const/4 v2, 0x0
  
      goto :goto_0
