*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/SocialContract$Activities.smali	2012-03-23 19:35:26.107062948 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/SocialContract$Activities.smali.original	2012-03-23 19:34:44.590305891 -0500
***************
*** 35,40 ****
--- 35,41 ----
      .locals 2
  
      .prologue
+     .line 157
      sget-object v0, Landroid/provider/SocialContract;->AUTHORITY_URI:Landroid/net/Uri;
  
      const-string v1, "activities"
***************
*** 45,50 ****
--- 46,52 ----
  
      sput-object v0, Landroid/provider/SocialContract$Activities;->CONTENT_URI:Landroid/net/Uri;
  
+     .line 163
      sget-object v0, Landroid/provider/SocialContract$Activities;->CONTENT_URI:Landroid/net/Uri;
  
      const-string v1, "authored_by"
***************
*** 55,60 ****
--- 57,63 ----
  
      sput-object v0, Landroid/provider/SocialContract$Activities;->CONTENT_AUTHORED_BY_URI:Landroid/net/Uri;
  
+     .line 171
      sget-object v0, Landroid/provider/SocialContract;->AUTHORITY_URI:Landroid/net/Uri;
  
      const-string v1, "contact_status"
***************
*** 72,78 ****
--- 75,83 ----
      .locals 0
  
      .prologue
+     .line 151
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 152
      return-void
  .end method
