*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Telephony$Sms$Inbox.smali	2012-03-23 19:35:25.814057604 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Telephony$Sms$Inbox.smali.original	2012-03-23 19:34:44.588305853 -0500
***************
*** 29,34 ****
--- 29,35 ----
      .locals 1
  
      .prologue
+     .line 325
      const-string v0, "content://sms/inbox"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 44,49 ****
--- 45,51 ----
      .locals 0
  
      .prologue
+     .line 321
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 59,64 ****
--- 61,67 ----
      .parameter "read"
  
      .prologue
+     .line 347
      sget-object v1, Landroid/provider/Telephony$Sms$Inbox;->CONTENT_URI:Landroid/net/Uri;
  
      const/4 v7, 0x0
