*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Telephony$Carriers.smali	2012-03-23 19:35:25.726055999 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Telephony$Carriers.smali.original	2012-03-23 19:34:44.587305835 -0500
***************
*** 74,79 ****
--- 74,80 ----
      .locals 1
  
      .prologue
+     .line 1721
      const-string v0, "content://telephony/carriers"
  
      invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
***************
*** 89,94 ****
--- 90,96 ----
      .locals 0
  
      .prologue
+     .line 1717
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
