*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Telephony$Mms$Rate.smali	2012-03-23 19:35:29.660127738 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Telephony$Mms$Rate.smali.original	2012-03-23 19:34:44.620306437 -0500
***************
*** 25,33 ****
      .locals 2
  
      .prologue
      sget-object v0, Landroid/provider/Telephony$Mms;->CONTENT_URI:Landroid/net/Uri;
  
!     const-string v1, "rate"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
--- 25,34 ----
      .locals 2
  
      .prologue
+     .line 1566
      sget-object v0, Landroid/provider/Telephony$Mms;->CONTENT_URI:Landroid/net/Uri;
  
!     const-string/jumbo v1, "rate"
  
      invoke-static {v0, v1}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
  
***************
*** 42,47 ****
--- 43,49 ----
      .locals 0
  
      .prologue
+     .line 1565
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
