*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Telephony$Intents.smali	2012-03-23 19:35:28.198101077 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/provider/Telephony$Intents.smali.original	2012-03-23 19:34:44.607306201 -0500
***************
*** 33,39 ****
--- 33,41 ----
      .locals 0
  
      .prologue
+     .line 1800
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 1802
      return-void
  .end method
