*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/BatteryStatsImpl$Timer.smali	2012-03-23 19:37:50.593697658 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/BatteryStatsImpl$Timer.smali.original	2012-03-23 19:34:46.368338313 -0500
***************
*** 64,78 ****
--- 64,83 ----
      .end annotation
  
      .prologue
+     .line 540
      .local p2, unpluggables:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/os/BatteryStatsImpl$Unpluggable;>;"
      invoke-direct {p0}, Landroid/os/BatteryStats$Timer;-><init>()V
  
+     .line 541
      iput p1, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mType:I
  
+     .line 542
      iput-object p2, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggables:Ljava/util/ArrayList;
  
+     .line 543
      invoke-virtual {p2, p0}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 544
      return-void
  .end method
  
***************
*** 94,152 ****
--- 99,170 ----
      .end annotation
  
      .prologue
+     .line 525
      .local p2, unpluggables:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/os/BatteryStatsImpl$Unpluggable;>;"
      invoke-direct {p0}, Landroid/os/BatteryStats$Timer;-><init>()V
  
+     .line 526
      iput p1, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mType:I
  
+     .line 527
      iput-object p2, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggables:Ljava/util/ArrayList;
  
+     .line 529
      invoke-virtual {p3}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mCount:I
  
+     .line 530
      invoke-virtual {p3}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLoadedCount:I
  
+     .line 531
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLastCount:I
  
+     .line 532
      invoke-virtual {p3}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedCount:I
  
+     .line 533
      invoke-virtual {p3}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mTotalTime:J
  
+     .line 534
      invoke-virtual {p3}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLoadedTime:J
  
+     .line 535
      const-wide/16 v0, 0x0
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLastTime:J
  
+     .line 536
      invoke-virtual {p3}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedTime:J
  
+     .line 537
      invoke-virtual {p2, p0}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 538
      return-void
  .end method
  
***************
*** 157,176 ****
--- 175,199 ----
      .parameter "batteryRealtime"
  
      .prologue
+     .line 612
      if-nez p1, :cond_0
  
+     .line 613
      const/4 v0, 0x0
  
      invoke-virtual {p0, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 619
      :goto_0
      return-void
  
+     .line 616
      :cond_0
      const/4 v0, 0x1
  
      invoke-virtual {p0, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 618
      invoke-virtual {p1, p0, p2, p3}, Lcom/android/internal/os/BatteryStatsImpl$Timer;->writeToParcel(Landroid/os/Parcel;J)V
  
      goto :goto_0
***************
*** 188,197 ****
--- 211,222 ----
      .locals 1
  
      .prologue
+     .line 564
      iget-object v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggables:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p0}, Ljava/util/ArrayList;->remove(Ljava/lang/Object;)Z
  
+     .line 565
      return-void
  .end method
  
***************
*** 200,236 ****
--- 225,269 ----
      .parameter "which"
  
      .prologue
+     .line 641
      const/4 v1, 0x1
  
      if-ne p1, v1, :cond_1
  
+     .line 642
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLastCount:I
  
+     .line 652
      .local v0, val:I
      :cond_0
      :goto_0
      return v0
  
+     .line 644
      .end local v0           #val:I
      :cond_1
      invoke-virtual {p0}, Lcom/android/internal/os/BatteryStatsImpl$Timer;->computeCurrentCountLocked()I
  
      move-result v0
  
+     .line 645
      .restart local v0       #val:I
      const/4 v1, 0x3
  
      if-ne p1, v1, :cond_2
  
+     .line 646
      iget v1, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedCount:I
  
      sub-int/2addr v0, v1
  
      goto :goto_0
  
+     .line 647
      :cond_2
      if-eqz p1, :cond_0
  
+     .line 648
      iget v1, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLoadedCount:I
  
      sub-int/2addr v0, v1
***************
*** 244,280 ****
--- 277,321 ----
      .parameter "which"
  
      .prologue
+     .line 624
      const/4 v2, 0x1
  
      if-ne p3, v2, :cond_1
  
+     .line 625
      iget-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLastTime:J
  
+     .line 635
      .local v0, val:J
      :cond_0
      :goto_0
      return-wide v0
  
+     .line 627
      .end local v0           #val:J
      :cond_1
      invoke-virtual {p0, p1, p2}, Lcom/android/internal/os/BatteryStatsImpl$Timer;->computeRunTimeLocked(J)J
  
      move-result-wide v0
  
+     .line 628
      .restart local v0       #val:J
      const/4 v2, 0x3
  
      if-ne p3, v2, :cond_2
  
+     .line 629
      iget-wide v2, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedTime:J
  
      sub-long/2addr v0, v2
  
      goto :goto_0
  
+     .line 630
      :cond_2
      if-eqz p3, :cond_0
  
+     .line 631
      iget-wide v2, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLoadedTime:J
  
      sub-long/2addr v0, v2
***************
*** 288,293 ****
--- 329,335 ----
      .parameter "prefix"
  
      .prologue
+     .line 656
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 350,355 ****
--- 392,398 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 659
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 358,364 ****
  
      move-result-object v0
  
!     const-string v1, "mTotalTime="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 401,407 ----
  
      move-result-object v0
  
!     const-string/jumbo v1, "mTotalTime="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 388,393 ****
--- 431,437 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 661
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 426,431 ****
--- 470,476 ----
  
      invoke-interface {p1, v0}, Landroid/util/Printer;->println(Ljava/lang/String;)V
  
+     .line 663
      return-void
  .end method
  
***************
*** 435,452 ****
--- 480,500 ----
      .parameter "batteryRealtime"
  
      .prologue
+     .line 596
      invoke-virtual {p0, p3, p4}, Lcom/android/internal/os/BatteryStatsImpl$Timer;->computeRunTimeLocked(J)J
  
      move-result-wide v0
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mTotalTime:J
  
+     .line 597
      invoke-virtual {p0}, Lcom/android/internal/os/BatteryStatsImpl$Timer;->computeCurrentCountLocked()I
  
      move-result v0
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mCount:I
  
+     .line 602
      return-void
  .end method
  
***************
*** 455,460 ****
--- 503,509 ----
      .parameter "in"
  
      .prologue
+     .line 675
      invoke-virtual {p1}, Landroid/os/Parcel;->readLong()J
  
      move-result-wide v0
***************
*** 467,480 ****
--- 516,532 ----
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mTotalTime:J
  
+     .line 676
      const-wide/16 v0, 0x0
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLastTime:J
  
+     .line 677
      iget-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mTotalTime:J
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedTime:J
  
+     .line 678
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v0
***************
*** 483,496 ****
--- 535,551 ----
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mCount:I
  
+     .line 679
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLastCount:I
  
+     .line 680
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mCount:I
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedCount:I
  
+     .line 681
      return-void
  .end method
  
***************
*** 500,505 ****
--- 555,561 ----
      .parameter "detachIfReset"
  
      .prologue
+     .line 555
      const-wide/16 v0, 0x0
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLastTime:J
***************
*** 508,513 ****
--- 564,570 ----
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mTotalTime:J
  
+     .line 556
      const/4 v0, 0x0
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLastCount:I
***************
*** 516,525 ****
--- 573,585 ----
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mCount:I
  
+     .line 557
      if-eqz p2, :cond_0
  
+     .line 558
      invoke-virtual {p0}, Lcom/android/internal/os/BatteryStatsImpl$Timer;->detach()V
  
+     .line 560
      :cond_0
      const/4 v0, 0x1
  
***************
*** 532,547 ****
--- 592,610 ----
      .parameter "batteryRealtime"
  
      .prologue
+     .line 582
      invoke-virtual {p0, p3, p4}, Lcom/android/internal/os/BatteryStatsImpl$Timer;->computeRunTimeLocked(J)J
  
      move-result-wide v0
  
      iput-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedTime:J
  
+     .line 583
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mCount:I
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedCount:I
  
+     .line 589
      return-void
  .end method
  
***************
*** 551,560 ****
--- 614,625 ----
      .parameter "batteryRealtime"
  
      .prologue
+     .line 667
      invoke-virtual {p0, p2, p3}, Lcom/android/internal/os/BatteryStatsImpl$Timer;->computeRunTimeLocked(J)J
  
      move-result-wide v0
  
+     .line 669
      .local v0, runTime:J
      const-wide/16 v2, 0x1f4
  
***************
*** 566,575 ****
--- 631,642 ----
  
      invoke-virtual {p1, v2, v3}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 670
      iget v2, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mCount:I
  
      invoke-virtual {p1, v2}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 671
      return-void
  .end method
  
***************
*** 579,609 ****
--- 646,683 ----
      .parameter "batteryRealtime"
  
      .prologue
+     .line 568
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mCount:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 569
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLoadedCount:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 570
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedCount:I
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 571
      invoke-virtual {p0, p2, p3}, Lcom/android/internal/os/BatteryStatsImpl$Timer;->computeRunTimeLocked(J)J
  
      move-result-wide v0
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 572
      iget-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mLoadedTime:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 573
      iget-wide v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Timer;->mUnpluggedTime:J
  
      invoke-virtual {p1, v0, v1}, Landroid/os/Parcel;->writeLong(J)V
  
+     .line 574
      return-void
  .end method
