*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/BatteryStatsImpl$Uid$Pkg.smali	2012-03-23 19:37:51.842720432 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/BatteryStatsImpl$Uid$Pkg.smali.original	2012-03-23 19:34:46.379338514 -0500
***************
*** 53,74 ****
--- 53,78 ----
      .parameter
  
      .prologue
+     .line 3514
      iput-object p1, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->this$1:Lcom/android/internal/os/BatteryStatsImpl$Uid;
  
      invoke-direct {p0}, Landroid/os/BatteryStats$Uid$Pkg;-><init>()V
  
+     .line 3512
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mServiceStats:Ljava/util/HashMap;
  
+     .line 3515
      iget-object v0, p1, Lcom/android/internal/os/BatteryStatsImpl$Uid;->this$0:Lcom/android/internal/os/BatteryStatsImpl;
  
      iget-object v0, v0, Lcom/android/internal/os/BatteryStatsImpl;->mUnpluggables:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p0}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 3516
      return-void
  .end method
  
***************
*** 78,83 ****
--- 82,88 ----
      .locals 1
  
      .prologue
+     .line 3526
      iget-object v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->this$1:Lcom/android/internal/os/BatteryStatsImpl$Uid;
  
      iget-object v0, v0, Lcom/android/internal/os/BatteryStatsImpl$Uid;->this$0:Lcom/android/internal/os/BatteryStatsImpl;
***************
*** 86,91 ****
--- 91,97 ----
  
      invoke-virtual {v0, p0}, Ljava/util/ArrayList;->remove(Ljava/lang/Object;)Z
  
+     .line 3527
      return-void
  .end method
  
***************
*** 93,98 ****
--- 99,105 ----
      .locals 1
  
      .prologue
+     .line 3838
      iget-object v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->this$1:Lcom/android/internal/os/BatteryStatsImpl$Uid;
  
      iget-object v0, v0, Lcom/android/internal/os/BatteryStatsImpl$Uid;->this$0:Lcom/android/internal/os/BatteryStatsImpl;
***************
*** 115,120 ****
--- 122,128 ----
      .end annotation
  
      .prologue
+     .line 3562
      iget-object v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mServiceStats:Ljava/util/HashMap;
  
      return-object v0
***************
*** 125,161 ****
--- 133,177 ----
      .parameter "which"
  
      .prologue
+     .line 3568
      const/4 v1, 0x1
  
      if-ne p1, v1, :cond_1
  
+     .line 3569
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mLastWakeups:I
  
+     .line 3579
      .local v0, val:I
      :cond_0
      :goto_0
      return v0
  
+     .line 3571
      .end local v0           #val:I
      :cond_1
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mWakeups:I
  
+     .line 3572
      .restart local v0       #val:I
      const/4 v1, 0x2
  
      if-ne p1, v1, :cond_2
  
+     .line 3573
      iget v1, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mLoadedWakeups:I
  
      sub-int/2addr v0, v1
  
      goto :goto_0
  
+     .line 3574
      :cond_2
      const/4 v1, 0x3
  
      if-ne p1, v1, :cond_0
  
+     .line 3575
      iget v1, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mUnpluggedWakeups:I
  
      sub-int/2addr v0, v1
***************
*** 167,178 ****
--- 183,196 ----
      .locals 1
  
      .prologue
+     .line 3842
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mWakeups:I
  
      add-int/lit8 v0, v0, 0x1
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mWakeups:I
  
+     .line 3843
      return-void
  .end method
  
***************
*** 180,185 ****
--- 198,204 ----
      .locals 1
  
      .prologue
+     .line 3846
      new-instance v0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;
  
      invoke-direct {v0, p0}, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;-><init>(Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;)V
***************
*** 193,198 ****
--- 212,218 ----
      .parameter "batteryRealtime"
  
      .prologue
+     .line 3523
      return-void
  .end method
  
***************
*** 201,263 ****
--- 221,296 ----
      .parameter "in"
  
      .prologue
+     .line 3530
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
  
      iput v4, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mWakeups:I
  
+     .line 3531
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
  
      iput v4, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mLoadedWakeups:I
  
+     .line 3532
      const/4 v4, 0x0
  
      iput v4, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mLastWakeups:I
  
+     .line 3533
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v4
  
      iput v4, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mUnpluggedWakeups:I
  
+     .line 3535
      invoke-virtual {p1}, Landroid/os/Parcel;->readInt()I
  
      move-result v1
  
+     .line 3536
      .local v1, numServs:I
      iget-object v4, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mServiceStats:Ljava/util/HashMap;
  
      invoke-virtual {v4}, Ljava/util/HashMap;->clear()V
  
+     .line 3537
      const/4 v0, 0x0
  
      .local v0, m:I
      :goto_0
      if-ge v0, v1, :cond_0
  
+     .line 3538
      invoke-virtual {p1}, Landroid/os/Parcel;->readString()Ljava/lang/String;
  
      move-result-object v3
  
+     .line 3539
      .local v3, serviceName:Ljava/lang/String;
      new-instance v2, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;
  
      invoke-direct {v2, p0}, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;-><init>(Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;)V
  
+     .line 3540
      .local v2, serv:Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;
      iget-object v4, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mServiceStats:Ljava/util/HashMap;
  
      invoke-virtual {v4, v3, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 3542
      invoke-virtual {v2, p1}, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;->readFromParcelLocked(Landroid/os/Parcel;)V
  
+     .line 3537
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 3544
      .end local v2           #serv:Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;
      .end local v3           #serviceName:Ljava/lang/String;
      :cond_0
***************
*** 270,279 ****
--- 303,314 ----
      .parameter "batteryRealtime"
  
      .prologue
+     .line 3519
      iget v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mWakeups:I
  
      iput v0, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mUnpluggedWakeups:I
  
+     .line 3520
      return-void
  .end method
  
***************
*** 282,299 ****
--- 317,338 ----
      .parameter "out"
  
      .prologue
+     .line 3547
      iget v3, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mWakeups:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 3548
      iget v3, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mLoadedWakeups:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 3549
      iget v3, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mUnpluggedWakeups:I
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 3551
      iget-object v3, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mServiceStats:Ljava/util/HashMap;
  
      invoke-virtual {v3}, Ljava/util/HashMap;->size()I
***************
*** 302,307 ****
--- 341,347 ----
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeInt(I)V
  
+     .line 3552
      iget-object v3, p0, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg;->mServiceStats:Ljava/util/HashMap;
  
      invoke-virtual {v3}, Ljava/util/HashMap;->entrySet()Ljava/util/Set;
***************
*** 326,331 ****
--- 366,372 ----
  
      check-cast v2, Ljava/util/Map$Entry;
  
+     .line 3553
      .local v2, servEntry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;>;"
      invoke-interface {v2}, Ljava/util/Map$Entry;->getKey()Ljava/lang/Object;
  
***************
*** 335,351 ****
--- 376,395 ----
  
      invoke-virtual {p1, v3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     .line 3554
      invoke-interface {v2}, Ljava/util/Map$Entry;->getValue()Ljava/lang/Object;
  
      move-result-object v1
  
      check-cast v1, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;
  
+     .line 3556
      .local v1, serv:Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;
      invoke-virtual {v1, p1}, Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;->writeToParcelLocked(Landroid/os/Parcel;)V
  
      goto :goto_0
  
+     .line 3558
      .end local v1           #serv:Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;
      .end local v2           #servEntry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/String;Lcom/android/internal/os/BatteryStatsImpl$Uid$Pkg$Serv;>;"
      :cond_0
