*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/HandlerCaller$MyHandler.smali	2012-03-23 19:37:50.837702107 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/HandlerCaller$MyHandler.smali.original	2012-03-23 19:34:46.370338350 -0500
***************
*** 25,34 ****
--- 25,37 ----
      .parameter "looper"
  
      .prologue
+     .line 55
      iput-object p1, p0, Lcom/android/internal/os/HandlerCaller$MyHandler;->this$0:Lcom/android/internal/os/HandlerCaller;
  
+     .line 56
      invoke-direct {p0, p2}, Landroid/os/Handler;-><init>(Landroid/os/Looper;)V
  
+     .line 57
      return-void
  .end method
  
***************
*** 39,49 ****
--- 42,54 ----
      .parameter "msg"
  
      .prologue
+     .line 61
      iget-object v0, p0, Lcom/android/internal/os/HandlerCaller$MyHandler;->this$0:Lcom/android/internal/os/HandlerCaller;
  
      iget-object v0, v0, Lcom/android/internal/os/HandlerCaller;->mCallback:Lcom/android/internal/os/HandlerCaller$Callback;
  
      invoke-interface {v0, p1}, Lcom/android/internal/os/HandlerCaller$Callback;->executeMessage(Landroid/os/Message;)V
  
+     .line 62
      return-void
  .end method
