*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/ProcessStats$1.smali	2012-03-23 19:37:50.565697146 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/ProcessStats$1.smali.original	2012-03-23 19:34:46.368338313 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 234
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 49,60 ****
--- 50,63 ----
  
      const/4 v2, -0x1
  
+     .line 237
      iget v4, p1, Lcom/android/internal/os/ProcessStats$Stats;->rel_utime:I
  
      iget v5, p1, Lcom/android/internal/os/ProcessStats$Stats;->rel_stime:I
  
      add-int v0, v4, v5
  
+     .line 238
      .local v0, ta:I
      iget v4, p2, Lcom/android/internal/os/ProcessStats$Stats;->rel_utime:I
  
***************
*** 62,72 ****
--- 65,78 ----
  
      add-int v1, v4, v5
  
+     .line 239
      .local v1, tb:I
      if-eq v0, v1, :cond_2
  
+     .line 240
      if-le v0, v1, :cond_1
  
+     .line 248
      :cond_0
      :goto_0
      return v2
***************
*** 74,81 ****
--- 80,89 ----
      :cond_1
      move v2, v3
  
+     .line 240
      goto :goto_0
  
+     .line 242
      :cond_2
      iget-boolean v4, p1, Lcom/android/internal/os/ProcessStats$Stats;->added:Z
  
***************
*** 83,88 ****
--- 91,97 ----
  
      if-eq v4, v5, :cond_3
  
+     .line 243
      iget-boolean v4, p1, Lcom/android/internal/os/ProcessStats$Stats;->added:Z
  
      if-nez v4, :cond_0
***************
*** 91,96 ****
--- 100,106 ----
  
      goto :goto_0
  
+     .line 245
      :cond_3
      iget-boolean v4, p1, Lcom/android/internal/os/ProcessStats$Stats;->removed:Z
  
***************
*** 98,103 ****
--- 108,114 ----
  
      if-eq v4, v5, :cond_4
  
+     .line 246
      iget-boolean v4, p1, Lcom/android/internal/os/ProcessStats$Stats;->added:Z
  
      if-nez v4, :cond_0
***************
*** 106,111 ****
--- 117,123 ----
  
      goto :goto_0
  
+     .line 248
      :cond_4
      const/4 v2, 0x0
  
***************
*** 118,123 ****
--- 130,136 ----
      .parameter "x1"
  
      .prologue
+     .line 234
      check-cast p1, Lcom/android/internal/os/ProcessStats$Stats;
  
      .end local p1
