*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/ProcessStats.smali	2012-03-23 19:37:51.659717095 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/ProcessStats.smali.original	2012-03-23 19:34:46.377338477 -0500
***************
*** 153,158 ****
--- 153,159 ----
      .locals 1
  
      .prologue
+     .line 40
      const/16 v0, 0xf
  
      new-array v0, v0, [I
***************
*** 161,166 ****
--- 162,168 ----
  
      sput-object v0, Lcom/android/internal/os/ProcessStats;->PROCESS_STATS_FORMAT:[I
  
+     .line 68
      const/16 v0, 0x16
  
      new-array v0, v0, [I
***************
*** 169,174 ****
--- 171,177 ----
  
      sput-object v0, Lcom/android/internal/os/ProcessStats;->PROCESS_FULL_STATS_FORMAT:[I
  
+     .line 102
      const/16 v0, 0x8
  
      new-array v0, v0, [I
***************
*** 177,182 ****
--- 180,186 ----
  
      sput-object v0, Lcom/android/internal/os/ProcessStats;->SYSTEM_CPU_FORMAT:[I
  
+     .line 115
      const/4 v0, 0x3
  
      new-array v0, v0, [I
***************
*** 185,190 ****
--- 189,195 ----
  
      sput-object v0, Lcom/android/internal/os/ProcessStats;->LOAD_AVERAGE_FORMAT:[I
  
+     .line 234
      new-instance v0, Lcom/android/internal/os/ProcessStats$1;
  
      invoke-direct {v0}, Lcom/android/internal/os/ProcessStats$1;-><init>()V
***************
*** 193,198 ****
--- 198,204 ----
  
      return-void
  
+     .line 40
      nop
  
      :array_0
***************
*** 214,219 ****
--- 220,226 ----
          0x20t 0x20t 0x0t 0x0t
      .end array-data
  
+     .line 68
      :array_1
      .array-data 0x4
          0x20t 0x0t 0x0t 0x0t
***************
*** 240,245 ****
--- 247,253 ----
          0x20t 0x20t 0x0t 0x0t
      .end array-data
  
+     .line 102
      :array_2
      .array-data 0x4
          0x20t 0x1t 0x0t 0x0t
***************
*** 252,257 ****
--- 260,266 ----
          0x20t 0x20t 0x0t 0x0t
      .end array-data
  
+     .line 115
      :array_3
      .array-data 0x4
          0x20t 0x40t 0x0t 0x0t
***************
*** 271,336 ****
--- 280,361 ----
  
      const/4 v1, 0x0
  
+     .line 253
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 64
      new-array v0, v2, [J
  
      iput-object v0, p0, Lcom/android/internal/os/ProcessStats;->mProcessStatsData:[J
  
+     .line 66
      new-array v0, v2, [J
  
      iput-object v0, p0, Lcom/android/internal/os/ProcessStats;->mSinglePidStatsData:[J
  
+     .line 99
      new-array v0, v3, [Ljava/lang/String;
  
      iput-object v0, p0, Lcom/android/internal/os/ProcessStats;->mProcessFullStatsStringData:[Ljava/lang/String;
  
+     .line 100
      new-array v0, v3, [J
  
      iput-object v0, p0, Lcom/android/internal/os/ProcessStats;->mProcessFullStatsData:[J
  
+     .line 113
      const/4 v0, 0x7
  
      new-array v0, v0, [J
  
      iput-object v0, p0, Lcom/android/internal/os/ProcessStats;->mSystemCpuData:[J
  
+     .line 121
      const/4 v0, 0x3
  
      new-array v0, v0, [F
  
      iput-object v0, p0, Lcom/android/internal/os/ProcessStats;->mLoadAverageData:[F
  
+     .line 125
      iput v1, p0, Lcom/android/internal/os/ProcessStats;->mLoad1:F
  
+     .line 126
      iput v1, p0, Lcom/android/internal/os/ProcessStats;->mLoad5:F
  
+     .line 127
      iput v1, p0, Lcom/android/internal/os/ProcessStats;->mLoad15:F
  
+     .line 151
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/os/ProcessStats;->mProcStats:Ljava/util/ArrayList;
  
+     .line 152
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/internal/os/ProcessStats;->mWorkingProcs:Ljava/util/ArrayList;
  
+     .line 155
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Lcom/android/internal/os/ProcessStats;->mFirst:Z
  
+     .line 157
      const/16 v0, 0x200
  
      new-array v0, v0, [B
  
      iput-object v0, p0, Lcom/android/internal/os/ProcessStats;->mBuffer:[B
  
+     .line 254
      iput-boolean p1, p0, Lcom/android/internal/os/ProcessStats;->mIncludeThreads:Z
  
+     .line 255
      return-void
  .end method
  
***************
*** 354,359 ****
--- 379,385 ----
      .end annotation
  
      .prologue
+     .line 341
      .local p5, allProcs:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/internal/os/ProcessStats$Stats;>;"
      move-object/from16 v0, p1
  
***************
*** 363,382 ****
--- 389,412 ----
  
      move-result-object v17
  
+     .line 342
      .local v17, pids:[I
      if-nez v17, :cond_1
  
      const/4 v9, 0x0
  
+     .line 343
      .local v9, NP:I
      :goto_0
      invoke-virtual/range {p5 .. p5}, Ljava/util/ArrayList;->size()I
  
      move-result v10
  
+     .line 344
      .local v10, NS:I
      const/4 v11, 0x0
  
+     .line 345
      .local v11, curStatsIndex:I
      const/4 v12, 0x0
  
***************
*** 384,401 ****
--- 414,436 ----
      :goto_1
      if-ge v12, v9, :cond_0
  
+     .line 346
      aget v5, v17, v12
  
+     .line 347
      .local v5, pid:I
      if-gez v5, :cond_2
  
+     .line 348
      move v9, v5
  
+     .line 504
      .end local v5           #pid:I
      :cond_0
      :goto_2
      if-ge v11, v10, :cond_f
  
+     .line 506
      move-object/from16 v0, p5
  
      invoke-virtual {v0, v11}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 404,409 ****
--- 439,445 ----
  
      check-cast v20, Lcom/android/internal/os/ProcessStats$Stats;
  
+     .line 507
      .local v20, st:Lcom/android/internal/os/ProcessStats$Stats;
      const/4 v3, 0x0
  
***************
*** 411,454 ****
--- 447,499 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_utime:I
  
+     .line 508
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_stime:I
  
+     .line 509
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_minfaults:I
  
+     .line 510
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_majfaults:I
  
+     .line 511
      const/4 v3, 0x1
  
      move-object/from16 v0, v20
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->removed:Z
  
+     .line 512
      const/4 v3, 0x1
  
      move-object/from16 v0, v20
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->working:Z
  
+     .line 513
      move-object/from16 v0, p5
  
      invoke-virtual {v0, v11}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 514
      add-int/lit8 v10, v10, -0x1
  
+     .line 516
      goto :goto_2
  
+     .line 342
      .end local v9           #NP:I
      .end local v10           #NS:I
      .end local v11           #curStatsIndex:I
***************
*** 461,466 ****
--- 506,512 ----
  
      goto :goto_0
  
+     .line 351
      .restart local v5       #pid:I
      .restart local v9       #NP:I
      .restart local v10       #NS:I
***************
*** 479,484 ****
--- 525,531 ----
  
      move-object/from16 v20, v3
  
+     .line 353
      .restart local v20       #st:Lcom/android/internal/os/ProcessStats$Stats;
      :goto_3
      if-eqz v20, :cond_9
***************
*** 489,518 ****
--- 536,571 ----
  
      if-ne v3, v5, :cond_9
  
+     .line 355
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->added:Z
  
+     .line 356
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->working:Z
  
+     .line 357
      add-int/lit8 v11, v11, 0x1
  
+     .line 362
      move-object/from16 v0, v20
  
      iget-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->interesting:Z
  
      if-eqz v3, :cond_3
  
+     .line 363
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v23
  
+     .line 365
      .local v23, uptime:J
      move-object/from16 v0, p0
  
***************
*** 520,525 ****
--- 573,579 ----
  
      move-object/from16 v18, v0
  
+     .line 366
      .local v18, procStats:[J
      move-object/from16 v0, v20
  
***************
*** 543,548 ****
--- 597,603 ----
  
      if-nez v3, :cond_5
  
+     .line 345
      .end local v18           #procStats:[J
      .end local v23           #uptime:J
      :cond_3
***************
*** 551,562 ****
--- 606,619 ----
  
      goto/16 :goto_1
  
+     .line 351
      .end local v20           #st:Lcom/android/internal/os/ProcessStats$Stats;
      :cond_4
      const/16 v20, 0x0
  
      goto :goto_3
  
+     .line 371
      .restart local v18       #procStats:[J
      .restart local v20       #st:Lcom/android/internal/os/ProcessStats$Stats;
      .restart local v23       #uptime:J
***************
*** 565,585 ****
--- 622,646 ----
  
      aget-wide v15, v18, v3
  
+     .line 372
      .local v15, minfaults:J
      const/4 v3, 0x1
  
      aget-wide v13, v18, v3
  
+     .line 373
      .local v13, majfaults:J
      const/4 v3, 0x2
  
      aget-wide v25, v18, v3
  
+     .line 374
      .local v25, utime:J
      const/4 v3, 0x3
  
      aget-wide v21, v18, v3
  
+     .line 376
      .local v21, stime:J
      move-object/from16 v0, v20
  
***************
*** 597,632 ****
--- 658,699 ----
  
      if-nez v3, :cond_6
  
+     .line 377
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_utime:I
  
+     .line 378
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_stime:I
  
+     .line 379
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_minfaults:I
  
+     .line 380
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_majfaults:I
  
+     .line 381
      move-object/from16 v0, v20
  
      iget-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->active:Z
  
      if-eqz v3, :cond_3
  
+     .line 382
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
***************
*** 635,640 ****
--- 702,708 ----
  
      goto :goto_4
  
+     .line 387
      :cond_6
      move-object/from16 v0, v20
  
***************
*** 642,656 ****
--- 710,727 ----
  
      if-nez v3, :cond_7
  
+     .line 388
      const/4 v3, 0x1
  
      move-object/from16 v0, v20
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->active:Z
  
+     .line 391
      :cond_7
      if-gez p2, :cond_8
  
+     .line 392
      move-object/from16 v0, v20
  
      iget-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->cmdlineFile:Ljava/lang/String;
***************
*** 661,672 ****
--- 732,745 ----
  
      invoke-direct {v0, v1, v3}, Lcom/android/internal/os/ProcessStats;->getName(Lcom/android/internal/os/ProcessStats$Stats;Ljava/lang/String;)V
  
+     .line 393
      move-object/from16 v0, v20
  
      iget-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->threadStats:Ljava/util/ArrayList;
  
      if-eqz v3, :cond_8
  
+     .line 394
      move-object/from16 v0, v20
  
      iget-object v4, v0, Lcom/android/internal/os/ProcessStats$Stats;->threadsDir:Ljava/lang/String;
***************
*** 691,696 ****
--- 764,770 ----
  
      iput-object v3, v0, Lcom/android/internal/os/ProcessStats;->mCurThreadPids:[I
  
+     .line 405
      :cond_8
      move-object/from16 v0, v20
  
***************
*** 702,713 ****
--- 776,789 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_uptime:J
  
+     .line 406
      move-wide/from16 v0, v23
  
      move-object/from16 v2, v20
  
      iput-wide v0, v2, Lcom/android/internal/os/ProcessStats$Stats;->base_uptime:J
  
+     .line 407
      move-object/from16 v0, v20
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_utime:J
***************
*** 720,725 ****
--- 796,802 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_utime:I
  
+     .line 408
      move-object/from16 v0, v20
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_stime:J
***************
*** 732,749 ****
--- 809,829 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_stime:I
  
+     .line 409
      move-wide/from16 v0, v25
  
      move-object/from16 v2, v20
  
      iput-wide v0, v2, Lcom/android/internal/os/ProcessStats$Stats;->base_utime:J
  
+     .line 410
      move-wide/from16 v0, v21
  
      move-object/from16 v2, v20
  
      iput-wide v0, v2, Lcom/android/internal/os/ProcessStats$Stats;->base_stime:J
  
+     .line 411
      move-object/from16 v0, v20
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_minfaults:J
***************
*** 756,761 ****
--- 836,842 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_minfaults:I
  
+     .line 412
      move-object/from16 v0, v20
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_majfaults:J
***************
*** 768,781 ****
--- 849,865 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_majfaults:I
  
+     .line 413
      move-object/from16 v0, v20
  
      iput-wide v15, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_minfaults:J
  
+     .line 414
      move-object/from16 v0, v20
  
      iput-wide v13, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_majfaults:J
  
+     .line 415
      const/4 v3, 0x1
  
      move-object/from16 v0, v20
***************
*** 784,789 ****
--- 868,874 ----
  
      goto/16 :goto_4
  
+     .line 421
      .end local v13           #majfaults:J
      .end local v15           #minfaults:J
      .end local v18           #procStats:[J
***************
*** 799,804 ****
--- 884,890 ----
  
      if-le v3, v5, :cond_e
  
+     .line 423
      :cond_a
      new-instance v20, Lcom/android/internal/os/ProcessStats$Stats;
  
***************
*** 813,818 ****
--- 899,905 ----
  
      invoke-direct {v0, v5, v1, v3}, Lcom/android/internal/os/ProcessStats$Stats;-><init>(IIZ)V
  
+     .line 424
      .restart local v20       #st:Lcom/android/internal/os/ProcessStats$Stats;
      move-object/from16 v0, p5
  
***************
*** 820,835 ****
--- 907,926 ----
  
      invoke-virtual {v0, v11, v1}, Ljava/util/ArrayList;->add(ILjava/lang/Object;)V
  
+     .line 425
      add-int/lit8 v11, v11, 0x1
  
+     .line 426
      add-int/lit8 v10, v10, 0x1
  
+     .line 431
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/os/ProcessStats;->mProcessFullStatsStringData:[Ljava/lang/String;
  
      move-object/from16 v19, v0
  
+     .line 432
      .local v19, procStatsString:[Ljava/lang/String;
      move-object/from16 v0, p0
  
***************
*** 837,842 ****
--- 928,934 ----
  
      move-object/from16 v18, v0
  
+     .line 433
      .restart local v18       #procStats:[J
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
***************
*** 846,851 ****
--- 938,944 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_uptime:J
  
+     .line 434
      move-object/from16 v0, v20
  
      iget-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->statFile:Ljava/lang/String;
***************
*** 868,879 ****
--- 961,974 ----
  
      if-eqz v3, :cond_c
  
+     .line 442
      const/4 v3, 0x1
  
      move-object/from16 v0, v20
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->interesting:Z
  
+     .line 443
      const/4 v3, 0x0
  
      aget-object v3, v19, v3
***************
*** 882,887 ****
--- 977,983 ----
  
      iput-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->baseName:Ljava/lang/String;
  
+     .line 444
      const/4 v3, 0x1
  
      aget-wide v3, v18, v3
***************
*** 890,895 ****
--- 986,992 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_minfaults:J
  
+     .line 445
      const/4 v3, 0x2
  
      aget-wide v3, v18, v3
***************
*** 898,903 ****
--- 995,1001 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_majfaults:J
  
+     .line 446
      const/4 v3, 0x3
  
      aget-wide v3, v18, v3
***************
*** 906,911 ****
--- 1004,1010 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_utime:J
  
+     .line 447
      const/4 v3, 0x4
  
      aget-wide v3, v18, v3
***************
*** 914,922 ****
--- 1013,1023 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_stime:J
  
+     .line 460
      :goto_5
      if-gez p2, :cond_d
  
+     .line 461
      move-object/from16 v0, v20
  
      iget-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->cmdlineFile:Ljava/lang/String;
***************
*** 927,938 ****
--- 1028,1041 ----
  
      invoke-direct {v0, v1, v3}, Lcom/android/internal/os/ProcessStats;->getName(Lcom/android/internal/os/ProcessStats$Stats;Ljava/lang/String;)V
  
+     .line 462
      move-object/from16 v0, v20
  
      iget-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->threadStats:Ljava/util/ArrayList;
  
      if-eqz v3, :cond_b
  
+     .line 463
      move-object/from16 v0, v20
  
      iget-object v4, v0, Lcom/android/internal/os/ProcessStats$Stats;->threadsDir:Ljava/lang/String;
***************
*** 957,962 ****
--- 1060,1066 ----
  
      iput-object v3, v0, Lcom/android/internal/os/ProcessStats;->mCurThreadPids:[I
  
+     .line 475
      :cond_b
      :goto_6
      const/4 v3, 0x0
***************
*** 965,994 ****
--- 1069,1103 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_utime:I
  
+     .line 476
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_stime:I
  
+     .line 477
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_minfaults:I
  
+     .line 478
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_majfaults:I
  
+     .line 479
      const/4 v3, 0x1
  
      move-object/from16 v0, v20
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->added:Z
  
+     .line 480
      if-nez p3, :cond_3
  
      move-object/from16 v0, v20
***************
*** 997,1002 ****
--- 1106,1112 ----
  
      if-eqz v3, :cond_3
  
+     .line 481
      const/4 v3, 0x1
  
      move-object/from16 v0, v20
***************
*** 1005,1010 ****
--- 1115,1121 ----
  
      goto/16 :goto_4
  
+     .line 454
      :cond_c
      const-string v3, "ProcessStats"
  
***************
*** 1028,1039 ****
--- 1139,1152 ----
  
      invoke-static {v3, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 455
      const-string v3, "<unknown>"
  
      move-object/from16 v0, v20
  
      iput-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->baseName:Ljava/lang/String;
  
+     .line 456
      const-wide/16 v3, 0x0
  
      move-object/from16 v0, v20
***************
*** 1044,1049 ****
--- 1157,1163 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->base_utime:J
  
+     .line 457
      const-wide/16 v3, 0x0
  
      move-object/from16 v0, v20
***************
*** 1056,1061 ****
--- 1170,1176 ----
  
      goto/16 :goto_5
  
+     .line 466
      :cond_d
      move-object/from16 v0, v20
  
***************
*** 1063,1068 ****
--- 1178,1184 ----
  
      if-eqz v3, :cond_b
  
+     .line 467
      move-object/from16 v0, v20
  
      iget-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->baseName:Ljava/lang/String;
***************
*** 1071,1076 ****
--- 1187,1193 ----
  
      iput-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->name:Ljava/lang/String;
  
+     .line 468
      move-object/from16 v0, v20
  
      iget-object v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->name:Ljava/lang/String;
***************
*** 1087,1092 ****
--- 1204,1210 ----
  
      goto :goto_6
  
+     .line 487
      .end local v18           #procStats:[J
      .end local v19           #procStatsString:[Ljava/lang/String;
      :cond_e
***************
*** 1096,1141 ****
--- 1214,1269 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_utime:I
  
+     .line 488
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_stime:I
  
+     .line 489
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_minfaults:I
  
+     .line 490
      const/4 v3, 0x0
  
      move-object/from16 v0, v20
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->rel_majfaults:I
  
+     .line 491
      const/4 v3, 0x1
  
      move-object/from16 v0, v20
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->removed:Z
  
+     .line 492
      const/4 v3, 0x1
  
      move-object/from16 v0, v20
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats$Stats;->working:Z
  
+     .line 493
      move-object/from16 v0, p5
  
      invoke-virtual {v0, v11}, Ljava/util/ArrayList;->remove(I)Ljava/lang/Object;
  
+     .line 494
      add-int/lit8 v10, v10, -0x1
  
+     .line 500
      add-int/lit8 v12, v12, -0x1
  
+     .line 501
      goto/16 :goto_4
  
+     .line 518
      .end local v5           #pid:I
      .end local v20           #st:Lcom/android/internal/os/ProcessStats$Stats;
      :cond_f
***************
*** 1151,1174 ****
--- 1279,1310 ----
  
      const/4 v12, 0x0
  
+     .line 557
      move-object v6, p1
  
+     .line 558
      .local v6, tempTimes:[J
      iget-object v5, p0, Lcom/android/internal/os/ProcessStats;->mCpuSpeeds:[J
  
+     .line 559
      .local v5, tempSpeeds:[J
      const/16 v0, 0x14
  
+     .line 560
      .local v0, MAX_SPEEDS:I
      if-nez p1, :cond_0
  
+     .line 561
      new-array v6, v13, [J
  
+     .line 562
      new-array v5, v13, [J
  
+     .line 564
      :cond_0
      const/4 v3, 0x0
  
+     .line 565
      .local v3, speed:I
      const-string v10, "/sys/devices/system/cpu/cpu0/cpufreq/stats/time_in_state"
  
***************
*** 1176,1190 ****
--- 1312,1329 ----
  
      move-result-object v1
  
+     .line 567
      .local v1, file:Ljava/lang/String;
      if-eqz v1, :cond_2
  
+     .line 568
      new-instance v4, Ljava/util/StringTokenizer;
  
      const-string v10, "\n "
  
      invoke-direct {v4, v1, v10}, Ljava/util/StringTokenizer;-><init>(Ljava/lang/String;Ljava/lang/String;)V
  
+     .line 569
      .local v4, st:Ljava/util/StringTokenizer;
      :cond_1
      :goto_0
***************
*** 1194,1254 ****
--- 1333,1409 ----
  
      if-eqz v10, :cond_2
  
+     .line 570
      invoke-virtual {v4}, Ljava/util/StringTokenizer;->nextToken()Ljava/lang/String;
  
      move-result-object v7
  
+     .line 572
      .local v7, token:Ljava/lang/String;
      :try_start_0
      invoke-static {v7}, Ljava/lang/Long;->parseLong(Ljava/lang/String;)J
  
      move-result-wide v8
  
+     .line 573
      .local v8, val:J
      aput-wide v8, v5, v3
  
+     .line 574
      invoke-virtual {v4}, Ljava/util/StringTokenizer;->nextToken()Ljava/lang/String;
  
      move-result-object v7
  
+     .line 575
      invoke-static {v7}, Ljava/lang/Long;->parseLong(Ljava/lang/String;)J
  
      move-result-wide v8
  
+     .line 576
      aput-wide v8, v6, v3
      :try_end_0
      .catch Ljava/lang/NumberFormatException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 577
      add-int/lit8 v3, v3, 0x1
  
+     .line 578
      if-ne v3, v13, :cond_1
  
+     .line 588
      .end local v4           #st:Ljava/util/StringTokenizer;
      .end local v7           #token:Ljava/lang/String;
      .end local v8           #val:J
      :cond_2
      if-nez p1, :cond_3
  
+     .line 589
      new-array p1, v3, [J
  
+     .line 590
      new-array v10, v3, [J
  
      iput-object v10, p0, Lcom/android/internal/os/ProcessStats;->mCpuSpeeds:[J
  
+     .line 591
      iget-object v10, p0, Lcom/android/internal/os/ProcessStats;->mCpuSpeeds:[J
  
      invoke-static {v5, v12, v10, v12, v3}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 592
      invoke-static {v6, v12, p1, v12, v3}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 594
      :cond_3
      return-object p1
  
+     .line 583
      .restart local v4       #st:Ljava/util/StringTokenizer;
      .restart local v7       #token:Ljava/lang/String;
      :catch_0
      move-exception v2
  
+     .line 584
      .local v2, nfe:Ljava/lang/NumberFormatException;
      const-string v10, "ProcessStats"
  
***************
*** 1265,1272 ****
--- 1420,1429 ----
      .parameter "cmdlineFile"
  
      .prologue
+     .line 836
      iget-object v2, p1, Lcom/android/internal/os/ProcessStats$Stats;->name:Ljava/lang/String;
  
+     .line 837
      .local v2, newName:Ljava/lang/String;
      iget-object v3, p1, Lcom/android/internal/os/ProcessStats$Stats;->name:Ljava/lang/String;
  
***************
*** 1292,1297 ****
--- 1449,1455 ----
  
      if-eqz v3, :cond_2
  
+     .line 839
      :cond_0
      const/4 v3, 0x0
  
***************
*** 1299,1304 ****
--- 1457,1463 ----
  
      move-result-object v0
  
+     .line 840
      .local v0, cmdName:Ljava/lang/String;
      if-eqz v0, :cond_1
  
***************
*** 1310,1323 ****
--- 1469,1485 ----
  
      if-le v3, v4, :cond_1
  
+     .line 841
      move-object v2, v0
  
+     .line 842
      const-string v3, "/"
  
      invoke-virtual {v2, v3}, Ljava/lang/String;->lastIndexOf(Ljava/lang/String;)I
  
      move-result v1
  
+     .line 843
      .local v1, i:I
      if-lez v1, :cond_1
  
***************
*** 1329,1346 ****
--- 1491,1512 ----
  
      if-ge v1, v3, :cond_1
  
+     .line 844
      add-int/lit8 v3, v1, 0x1
  
      invoke-virtual {v2, v3}, Ljava/lang/String;->substring(I)Ljava/lang/String;
  
      move-result-object v2
  
+     .line 847
      .end local v1           #i:I
      :cond_1
      if-nez v2, :cond_2
  
+     .line 848
      iget-object v2, p1, Lcom/android/internal/os/ProcessStats$Stats;->baseName:Ljava/lang/String;
  
+     .line 851
      .end local v0           #cmdName:Ljava/lang/String;
      :cond_2
      iget-object v3, p1, Lcom/android/internal/os/ProcessStats$Stats;->name:Ljava/lang/String;
***************
*** 1355,1363 ****
--- 1521,1531 ----
  
      if-nez v3, :cond_4
  
+     .line 852
      :cond_3
      iput-object v2, p1, Lcom/android/internal/os/ProcessStats$Stats;->name:Ljava/lang/String;
  
+     .line 853
      iget-object v3, p1, Lcom/android/internal/os/ProcessStats$Stats;->name:Ljava/lang/String;
  
      invoke-virtual {p0, v3}, Lcom/android/internal/os/ProcessStats;->onMeasureProcessName(Ljava/lang/String;)I
***************
*** 1366,1371 ****
--- 1534,1540 ----
  
      iput v3, p1, Lcom/android/internal/os/ProcessStats$Stats;->nameWidth:I
  
+     .line 855
      :cond_4
      return-void
  .end method
***************
*** 1386,1397 ****
--- 1555,1569 ----
      .parameter "majFaults"
  
      .prologue
+     .line 756
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 757
      if-nez p5, :cond_0
  
      const/4 p5, 0x1
  
+     .line 758
      :cond_0
      add-int v1, p6, p7
  
***************
*** 1411,1435 ****
--- 1583,1614 ----
  
      invoke-direct/range {v1 .. v6}, Lcom/android/internal/os/ProcessStats;->printRatio(Ljava/io/PrintWriter;JJ)V
  
+     .line 759
      const-string v1, "% "
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 760
      if-ltz p3, :cond_1
  
+     .line 761
      invoke-virtual {p1, p3}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 762
      const-string v1, "/"
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 764
      :cond_1
      invoke-virtual {p1, p4}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 765
      const-string v1, ": "
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 766
      int-to-long v3, p6
  
      int-to-long v5, p5
***************
*** 1440,1449 ****
--- 1619,1630 ----
  
      invoke-direct/range {v1 .. v6}, Lcom/android/internal/os/ProcessStats;->printRatio(Ljava/io/PrintWriter;JJ)V
  
+     .line 767
      const-string v1, "% user + "
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 768
      int-to-long v3, p7
  
      int-to-long v5, p5
***************
*** 1454,1469 ****
--- 1635,1654 ----
  
      invoke-direct/range {v1 .. v6}, Lcom/android/internal/os/ProcessStats;->printRatio(Ljava/io/PrintWriter;JJ)V
  
+     .line 769
      const-string v1, "% kernel"
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 770
      if-lez p8, :cond_2
  
+     .line 771
      const-string v1, " + "
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 772
      int-to-long v3, p8
  
      int-to-long v5, p5
***************
*** 1474,1490 ****
--- 1659,1679 ----
  
      invoke-direct/range {v1 .. v6}, Lcom/android/internal/os/ProcessStats;->printRatio(Ljava/io/PrintWriter;JJ)V
  
+     .line 773
      const-string v1, "% iowait"
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 775
      :cond_2
      if-lez p9, :cond_3
  
+     .line 776
      const-string v1, " + "
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 777
      move/from16 v0, p9
  
      int-to-long v3, v0
***************
*** 1497,1513 ****
--- 1686,1706 ----
  
      invoke-direct/range {v1 .. v6}, Lcom/android/internal/os/ProcessStats;->printRatio(Ljava/io/PrintWriter;JJ)V
  
+     .line 778
      const-string v1, "% irq"
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 780
      :cond_3
      if-lez p10, :cond_4
  
+     .line 781
      const-string v1, " + "
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 782
      move/from16 v0, p10
  
      int-to-long v3, v0
***************
*** 1520,1571 ****
--- 1713,1777 ----
  
      invoke-direct/range {v1 .. v6}, Lcom/android/internal/os/ProcessStats;->printRatio(Ljava/io/PrintWriter;JJ)V
  
+     .line 783
      const-string v1, "% softirq"
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 785
      :cond_4
      if-gtz p11, :cond_5
  
      if-lez p12, :cond_7
  
+     .line 786
      :cond_5
      const-string v1, " / faults:"
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 787
      if-lez p11, :cond_6
  
+     .line 788
      const-string v1, " "
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 789
      move/from16 v0, p11
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 790
      const-string v1, " minor"
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 792
      :cond_6
      if-lez p12, :cond_7
  
+     .line 793
      const-string v1, " "
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 794
      move/from16 v0, p12
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 795
      const-string v1, " major"
  
      invoke-virtual {p1, v1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 798
      :cond_7
      invoke-virtual {p1}, Ljava/io/PrintWriter;->println()V
  
+     .line 799
      return-void
  .end method
  
***************
*** 1578,1603 ****
--- 1784,1815 ----
      .prologue
      const-wide/16 v8, 0xa
  
+     .line 741
      const-wide/16 v6, 0x3e8
  
      mul-long/2addr v6, p2
  
      div-long v4, v6, p4
  
+     .line 742
      .local v4, thousands:J
      div-long v0, v4, v8
  
+     .line 743
      .local v0, hundreds:J
      invoke-virtual {p1, v0, v1}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 744
      cmp-long v6, v0, v8
  
      if-gez v6, :cond_0
  
+     .line 745
      mul-long v6, v0, v8
  
      sub-long v2, v4, v6
  
+     .line 746
      .local v2, remainder:J
      const-wide/16 v6, 0x0
  
***************
*** 1605,1616 ****
--- 1817,1831 ----
  
      if-eqz v6, :cond_0
  
+     .line 747
      const/16 v6, 0x2e
  
      invoke-virtual {p1, v6}, Ljava/io/PrintWriter;->print(C)V
  
+     .line 748
      invoke-virtual {p1, v2, v3}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 751
      .end local v2           #remainder:J
      :cond_0
      return-void
***************
*** 1622,1634 ****
--- 1837,1852 ----
      .parameter "endChar"
  
      .prologue
+     .line 805
      invoke-static {}, Landroid/os/StrictMode;->allowThreadDiskReads()Landroid/os/StrictMode$ThreadPolicy;
  
      move-result-object v4
  
+     .line 806
      .local v4, savedPolicy:Landroid/os/StrictMode$ThreadPolicy;
      const/4 v1, 0x0
  
+     .line 808
      .local v1, is:Ljava/io/FileInputStream;
      :try_start_0
      new-instance v2, Ljava/io/FileInputStream;
***************
*** 1639,1644 ****
--- 1857,1863 ----
      .catch Ljava/io/FileNotFoundException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 809
      .end local v1           #is:Ljava/io/FileInputStream;
      .local v2, is:Ljava/io/FileInputStream;
      :try_start_1
***************
*** 1648,1670 ****
--- 1867,1894 ----
  
      move-result v3
  
+     .line 810
      .local v3, len:I
      invoke-virtual {v2}, Ljava/io/FileInputStream;->close()V
  
+     .line 812
      if-lez v3, :cond_3
  
+     .line 814
      const/4 v0, 0x0
  
      .local v0, i:I
      :goto_0
      if-ge v0, v3, :cond_0
  
+     .line 815
      iget-object v5, p0, Lcom/android/internal/os/ProcessStats;->mBuffer:[B
  
      aget-byte v5, v5, v0
  
      if-ne v5, p2, :cond_2
  
+     .line 819
      :cond_0
      new-instance v5, Ljava/lang/String;
  
***************
*** 1678,1696 ****
--- 1902,1924 ----
      .catch Ljava/io/FileNotFoundException; {:try_start_1 .. :try_end_1} :catch_8
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_7
  
+     .line 824
      if-eqz v2, :cond_1
  
+     .line 826
      :try_start_2
      invoke-virtual {v2}, Ljava/io/FileInputStream;->close()V
      :try_end_2
      .catch Ljava/io/IOException; {:try_start_2 .. :try_end_2} :catch_2
  
+     .line 830
      :cond_1
      :goto_1
      invoke-static {v4}, Landroid/os/StrictMode;->setThreadPolicy(Landroid/os/StrictMode$ThreadPolicy;)V
  
      move-object v1, v2
  
+     .line 832
      .end local v0           #i:I
      .end local v2           #is:Ljava/io/FileInputStream;
      .end local v3           #len:I
***************
*** 1698,1703 ****
--- 1926,1932 ----
      :goto_2
      return-object v5
  
+     .line 814
      .end local v1           #is:Ljava/io/FileInputStream;
      .restart local v0       #i:I
      .restart local v2       #is:Ljava/io/FileInputStream;
***************
*** 1707,1727 ****
--- 1936,1960 ----
  
      goto :goto_0
  
+     .line 824
      .end local v0           #i:I
      :cond_3
      if-eqz v2, :cond_4
  
+     .line 826
      :try_start_3
      invoke-virtual {v2}, Ljava/io/FileInputStream;->close()V
      :try_end_3
      .catch Ljava/io/IOException; {:try_start_3 .. :try_end_3} :catch_3
  
+     .line 830
      :cond_4
      :goto_3
      invoke-static {v4}, Landroid/os/StrictMode;->setThreadPolicy(Landroid/os/StrictMode$ThreadPolicy;)V
  
      move-object v1, v2
  
+     .line 832
      .end local v2           #is:Ljava/io/FileInputStream;
      .end local v3           #len:I
      .restart local v1       #is:Ljava/io/FileInputStream;
***************
*** 1730,1786 ****
--- 1963,2031 ----
  
      goto :goto_2
  
+     .line 821
      :catch_0
      move-exception v5
  
+     .line 824
      :goto_5
      if-eqz v1, :cond_5
  
+     .line 826
      :try_start_4
      invoke-virtual {v1}, Ljava/io/FileInputStream;->close()V
      :try_end_4
      .catch Ljava/io/IOException; {:try_start_4 .. :try_end_4} :catch_4
  
+     .line 830
      :cond_5
      :goto_6
      invoke-static {v4}, Landroid/os/StrictMode;->setThreadPolicy(Landroid/os/StrictMode$ThreadPolicy;)V
  
      goto :goto_4
  
+     .line 822
      :catch_1
      move-exception v5
  
+     .line 824
      :goto_7
      if-eqz v1, :cond_6
  
+     .line 826
      :try_start_5
      invoke-virtual {v1}, Ljava/io/FileInputStream;->close()V
      :try_end_5
      .catch Ljava/io/IOException; {:try_start_5 .. :try_end_5} :catch_5
  
+     .line 830
      :cond_6
      :goto_8
      invoke-static {v4}, Landroid/os/StrictMode;->setThreadPolicy(Landroid/os/StrictMode$ThreadPolicy;)V
  
      goto :goto_4
  
+     .line 824
      :catchall_0
      move-exception v5
  
      :goto_9
      if-eqz v1, :cond_7
  
+     .line 826
      :try_start_6
      invoke-virtual {v1}, Ljava/io/FileInputStream;->close()V
      :try_end_6
      .catch Ljava/io/IOException; {:try_start_6 .. :try_end_6} :catch_6
  
+     .line 830
      :cond_7
      :goto_a
      invoke-static {v4}, Landroid/os/StrictMode;->setThreadPolicy(Landroid/os/StrictMode$ThreadPolicy;)V
  
      throw v5
  
+     .line 827
      .end local v1           #is:Ljava/io/FileInputStream;
      .restart local v0       #i:I
      .restart local v2       #is:Ljava/io/FileInputStream;
***************
*** 1814,1819 ****
--- 2059,2065 ----
  
      goto :goto_a
  
+     .line 824
      .end local v1           #is:Ljava/io/FileInputStream;
      .restart local v2       #is:Ljava/io/FileInputStream;
      :catchall_1
***************
*** 1825,1830 ****
--- 2071,2077 ----
      .restart local v1       #is:Ljava/io/FileInputStream;
      goto :goto_9
  
+     .line 822
      .end local v1           #is:Ljava/io/FileInputStream;
      .restart local v2       #is:Ljava/io/FileInputStream;
      :catch_7
***************
*** 1836,1841 ****
--- 2083,2089 ----
      .restart local v1       #is:Ljava/io/FileInputStream;
      goto :goto_7
  
+     .line 821
      .end local v1           #is:Ljava/io/FileInputStream;
      .restart local v2       #is:Ljava/io/FileInputStream;
      :catch_8
***************
*** 1856,1875 ****
--- 2104,2127 ----
      .prologue
      const/4 v8, 0x1
  
+     .line 627
      iget-boolean v6, p0, Lcom/android/internal/os/ProcessStats;->mWorkingProcsSorted:Z
  
      if-nez v6, :cond_4
  
+     .line 628
      iget-object v6, p0, Lcom/android/internal/os/ProcessStats;->mWorkingProcs:Ljava/util/ArrayList;
  
      invoke-virtual {v6}, Ljava/util/ArrayList;->clear()V
  
+     .line 629
      iget-object v6, p0, Lcom/android/internal/os/ProcessStats;->mProcStats:Ljava/util/ArrayList;
  
      invoke-virtual {v6}, Ljava/util/ArrayList;->size()I
  
      move-result v1
  
+     .line 630
      .local v1, N:I
      const/4 v2, 0x0
  
***************
*** 1877,1882 ****
--- 2129,2135 ----
      :goto_0
      if-ge v2, v1, :cond_3
  
+     .line 631
      iget-object v6, p0, Lcom/android/internal/os/ProcessStats;->mProcStats:Ljava/util/ArrayList;
  
      invoke-virtual {v6, v2}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1885,1899 ****
--- 2138,2155 ----
  
      check-cast v4, Lcom/android/internal/os/ProcessStats$Stats;
  
+     .line 632
      .local v4, stats:Lcom/android/internal/os/ProcessStats$Stats;
      iget-boolean v6, v4, Lcom/android/internal/os/ProcessStats$Stats;->working:Z
  
      if-eqz v6, :cond_2
  
+     .line 633
      iget-object v6, p0, Lcom/android/internal/os/ProcessStats;->mWorkingProcs:Ljava/util/ArrayList;
  
      invoke-virtual {v6, v4}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 634
      iget-object v6, v4, Lcom/android/internal/os/ProcessStats$Stats;->threadStats:Ljava/util/ArrayList;
  
      if-eqz v6, :cond_2
***************
*** 1906,1921 ****
--- 2162,2180 ----
  
      if-le v6, v8, :cond_2
  
+     .line 635
      iget-object v6, v4, Lcom/android/internal/os/ProcessStats$Stats;->workingThreads:Ljava/util/ArrayList;
  
      invoke-virtual {v6}, Ljava/util/ArrayList;->clear()V
  
+     .line 636
      iget-object v6, v4, Lcom/android/internal/os/ProcessStats$Stats;->threadStats:Ljava/util/ArrayList;
  
      invoke-virtual {v6}, Ljava/util/ArrayList;->size()I
  
      move-result v0
  
+     .line 637
      .local v0, M:I
      const/4 v3, 0x0
  
***************
*** 1923,1928 ****
--- 2182,2188 ----
      :goto_1
      if-ge v3, v0, :cond_1
  
+     .line 638
      iget-object v6, v4, Lcom/android/internal/os/ProcessStats$Stats;->threadStats:Ljava/util/ArrayList;
  
      invoke-virtual {v6, v3}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 1931,1950 ****
--- 2191,2214 ----
  
      check-cast v5, Lcom/android/internal/os/ProcessStats$Stats;
  
+     .line 639
      .local v5, tstats:Lcom/android/internal/os/ProcessStats$Stats;
      iget-boolean v6, v5, Lcom/android/internal/os/ProcessStats$Stats;->working:Z
  
      if-eqz v6, :cond_0
  
+     .line 640
      iget-object v6, v4, Lcom/android/internal/os/ProcessStats$Stats;->workingThreads:Ljava/util/ArrayList;
  
      invoke-virtual {v6, v5}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 637
      :cond_0
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_1
  
+     .line 643
      .end local v5           #tstats:Lcom/android/internal/os/ProcessStats$Stats;
      :cond_1
      iget-object v6, v4, Lcom/android/internal/os/ProcessStats$Stats;->workingThreads:Ljava/util/ArrayList;
***************
*** 1953,1958 ****
--- 2217,2223 ----
  
      invoke-static {v6, v7}, Ljava/util/Collections;->sort(Ljava/util/List;Ljava/util/Comparator;)V
  
+     .line 630
      .end local v0           #M:I
      .end local v3           #j:I
      :cond_2
***************
*** 1960,1965 ****
--- 2225,2231 ----
  
      goto :goto_0
  
+     .line 647
      .end local v4           #stats:Lcom/android/internal/os/ProcessStats$Stats;
      :cond_3
      iget-object v6, p0, Lcom/android/internal/os/ProcessStats;->mWorkingProcs:Ljava/util/ArrayList;
***************
*** 1968,1975 ****
--- 2234,2243 ----
  
      invoke-static {v6, v7}, Ljava/util/Collections;->sort(Ljava/util/List;Ljava/util/Comparator;)V
  
+     .line 648
      iput-boolean v8, p0, Lcom/android/internal/os/ProcessStats;->mWorkingProcsSorted:Z
  
+     .line 650
      .end local v1           #N:I
      .end local v2           #i:I
      :cond_4
***************
*** 1980,1985 ****
--- 2248,2254 ----
      .locals 1
  
      .prologue
+     .line 653
      iget-object v0, p0, Lcom/android/internal/os/ProcessStats;->mProcStats:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->size()I
***************
*** 1993,2000 ****
--- 2262,2271 ----
      .locals 1
  
      .prologue
+     .line 661
      invoke-virtual {p0}, Lcom/android/internal/os/ProcessStats;->buildWorkingProcs()V
  
+     .line 662
      iget-object v0, p0, Lcom/android/internal/os/ProcessStats;->mWorkingProcs:Ljava/util/ArrayList;
  
      invoke-virtual {v0}, Ljava/util/ArrayList;->size()I
***************
*** 2011,2016 ****
--- 2282,2288 ----
      .prologue
      const/4 v6, 0x0
  
+     .line 522
      new-instance v4, Ljava/lang/StringBuilder;
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 2035,2043 ****
--- 2307,2317 ----
  
      move-result-object v0
  
+     .line 523
      .local v0, statFile:Ljava/lang/String;
      iget-object v1, p0, Lcom/android/internal/os/ProcessStats;->mSinglePidStatsData:[J
  
+     .line 524
      .local v1, statsData:[J
      sget-object v4, Lcom/android/internal/os/ProcessStats;->PROCESS_STATS_FORMAT:[I
  
***************
*** 2047,2052 ****
--- 2321,2327 ----
  
      if-eqz v4, :cond_0
  
+     .line 526
      const/4 v4, 0x2
  
      aget-wide v4, v1, v4
***************
*** 2057,2062 ****
--- 2332,2338 ----
  
      add-long v2, v4, v6
  
+     .line 530
      :goto_0
      return-wide v2
  
***************
*** 2070,2079 ****
--- 2346,2357 ----
      .locals 8
  
      .prologue
+     .line 539
      iget-object v3, p0, Lcom/android/internal/os/ProcessStats;->mCpuSpeedTimes:[J
  
      if-nez v3, :cond_0
  
+     .line 540
      const/4 v3, 0x0
  
      invoke-direct {p0, v3}, Lcom/android/internal/os/ProcessStats;->getCpuSpeedTimes([J)[J
***************
*** 2082,2087 ****
--- 2360,2366 ----
  
      iput-object v3, p0, Lcom/android/internal/os/ProcessStats;->mCpuSpeedTimes:[J
  
+     .line 541
      iget-object v3, p0, Lcom/android/internal/os/ProcessStats;->mCpuSpeedTimes:[J
  
      array-length v3, v3
***************
*** 2090,2095 ****
--- 2369,2375 ----
  
      iput-object v3, p0, Lcom/android/internal/os/ProcessStats;->mRelCpuSpeedTimes:[J
  
+     .line 542
      const/4 v0, 0x0
  
      .local v0, i:I
***************
*** 2100,2121 ****
--- 2380,2405 ----
  
      if-ge v0, v3, :cond_1
  
+     .line 543
      iget-object v3, p0, Lcom/android/internal/os/ProcessStats;->mRelCpuSpeedTimes:[J
  
      const-wide/16 v4, 0x1
  
      aput-wide v4, v3, v0
  
+     .line 542
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_0
  
+     .line 546
      .end local v0           #i:I
      :cond_0
      iget-object v3, p0, Lcom/android/internal/os/ProcessStats;->mRelCpuSpeedTimes:[J
  
      invoke-direct {p0, v3}, Lcom/android/internal/os/ProcessStats;->getCpuSpeedTimes([J)[J
  
+     .line 547
      const/4 v0, 0x0
  
      .restart local v0       #i:I
***************
*** 2126,2135 ****
--- 2410,2421 ----
  
      if-ge v0, v3, :cond_1
  
+     .line 548
      iget-object v3, p0, Lcom/android/internal/os/ProcessStats;->mRelCpuSpeedTimes:[J
  
      aget-wide v1, v3, v0
  
+     .line 549
      .local v1, temp:J
      iget-object v3, p0, Lcom/android/internal/os/ProcessStats;->mRelCpuSpeedTimes:[J
  
***************
*** 2143,2156 ****
--- 2429,2445 ----
  
      aput-wide v4, v3, v0
  
+     .line 550
      iget-object v3, p0, Lcom/android/internal/os/ProcessStats;->mCpuSpeedTimes:[J
  
      aput-wide v1, v3, v0
  
+     .line 547
      add-int/lit8 v0, v0, 0x1
  
      goto :goto_1
  
+     .line 553
      .end local v1           #temp:J
      :cond_1
      iget-object v3, p0, Lcom/android/internal/os/ProcessStats;->mRelCpuSpeedTimes:[J
***************
*** 2162,2167 ****
--- 2451,2457 ----
      .locals 1
  
      .prologue
+     .line 618
      iget v0, p0, Lcom/android/internal/os/ProcessStats;->mRelIdleTime:I
  
      return v0
***************
*** 2171,2176 ****
--- 2461,2467 ----
      .locals 1
  
      .prologue
+     .line 606
      iget v0, p0, Lcom/android/internal/os/ProcessStats;->mRelIoWaitTime:I
  
      return v0
***************
*** 2180,2185 ****
--- 2471,2477 ----
      .locals 1
  
      .prologue
+     .line 610
      iget v0, p0, Lcom/android/internal/os/ProcessStats;->mRelIrqTime:I
  
      return v0
***************
*** 2189,2194 ****
--- 2481,2487 ----
      .locals 1
  
      .prologue
+     .line 614
      iget v0, p0, Lcom/android/internal/os/ProcessStats;->mRelSoftIrqTime:I
  
      return v0
***************
*** 2198,2203 ****
--- 2491,2497 ----
      .locals 1
  
      .prologue
+     .line 602
      iget v0, p0, Lcom/android/internal/os/ProcessStats;->mRelSystemTime:I
  
      return v0
***************
*** 2207,2212 ****
--- 2501,2507 ----
      .locals 1
  
      .prologue
+     .line 598
      iget v0, p0, Lcom/android/internal/os/ProcessStats;->mRelUserTime:I
  
      return v0
***************
*** 2217,2222 ****
--- 2512,2518 ----
      .parameter "index"
  
      .prologue
+     .line 657
      iget-object v0, p0, Lcom/android/internal/os/ProcessStats;->mProcStats:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 2232,2237 ****
--- 2528,2534 ----
      .locals 3
  
      .prologue
+     .line 622
      iget v0, p0, Lcom/android/internal/os/ProcessStats;->mRelUserTime:I
  
      iget v1, p0, Lcom/android/internal/os/ProcessStats;->mRelSystemTime:I
***************
*** 2274,2279 ****
--- 2571,2577 ----
      .parameter "index"
  
      .prologue
+     .line 666
      iget-object v0, p0, Lcom/android/internal/os/ProcessStats;->mWorkingProcs:Ljava/util/ArrayList;
  
      invoke-virtual {v0, p1}, Ljava/util/ArrayList;->get(I)Ljava/lang/Object;
***************
*** 2289,2300 ****
--- 2587,2601 ----
      .locals 1
  
      .prologue
+     .line 266
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Lcom/android/internal/os/ProcessStats;->mFirst:Z
  
+     .line 267
      invoke-virtual {p0}, Lcom/android/internal/os/ProcessStats;->update()V
  
+     .line 268
      return-void
  .end method
  
***************
*** 2305,2310 ****
--- 2606,2612 ----
      .parameter "load15"
  
      .prologue
+     .line 258
      return-void
  .end method
  
***************
*** 2313,2318 ****
--- 2615,2621 ----
      .parameter "name"
  
      .prologue
+     .line 261
      const/4 v0, 0x0
  
      return v0
***************
*** 2322,2361 ****
--- 2625,2673 ----
      .locals 3
  
      .prologue
+     .line 670
      new-instance v1, Ljava/io/StringWriter;
  
      invoke-direct {v1}, Ljava/io/StringWriter;-><init>()V
  
+     .line 671
      .local v1, sw:Ljava/io/StringWriter;
      new-instance v0, Ljava/io/PrintWriter;
  
      invoke-direct {v0, v1}, Ljava/io/PrintWriter;-><init>(Ljava/io/Writer;)V
  
+     .line 672
      .local v0, pw:Ljava/io/PrintWriter;
      const-string v2, "Load: "
  
      invoke-virtual {v0, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 673
      iget v2, p0, Lcom/android/internal/os/ProcessStats;->mLoad1:F
  
      invoke-virtual {v0, v2}, Ljava/io/PrintWriter;->print(F)V
  
+     .line 674
      const-string v2, " / "
  
      invoke-virtual {v0, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 675
      iget v2, p0, Lcom/android/internal/os/ProcessStats;->mLoad5:F
  
      invoke-virtual {v0, v2}, Ljava/io/PrintWriter;->print(F)V
  
+     .line 676
      const-string v2, " / "
  
      invoke-virtual {v0, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 677
      iget v2, p0, Lcom/android/internal/os/ProcessStats;->mLoad15:F
  
      invoke-virtual {v0, v2}, Ljava/io/PrintWriter;->println(F)V
  
+     .line 678
      invoke-virtual {v1}, Ljava/io/StringWriter;->toString()Ljava/lang/String;
  
      move-result-object v2
***************
*** 2368,2379 ****
--- 2680,2694 ----
      .parameter "now"
  
      .prologue
+     .line 682
      invoke-virtual/range {p0 .. p0}, Lcom/android/internal/os/ProcessStats;->buildWorkingProcs()V
  
+     .line 684
      new-instance v26, Ljava/io/StringWriter;
  
      invoke-direct/range {v26 .. v26}, Ljava/io/StringWriter;-><init>()V
  
+     .line 685
      .local v26, sw:Ljava/io/StringWriter;
      new-instance v3, Ljava/io/PrintWriter;
  
***************
*** 2381,2391 ****
--- 2696,2708 ----
  
      invoke-direct {v3, v0}, Ljava/io/PrintWriter;-><init>(Ljava/io/Writer;)V
  
+     .line 687
      .local v3, pw:Ljava/io/PrintWriter;
      const-string v2, "CPU usage from "
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 688
      move-object/from16 v0, p0
  
      iget-wide v4, v0, Lcom/android/internal/os/ProcessStats;->mLastSampleTime:J
***************
*** 2394,2399 ****
--- 2711,2717 ----
  
      if-lez v2, :cond_1
  
+     .line 689
      move-object/from16 v0, p0
  
      iget-wide v4, v0, Lcom/android/internal/os/ProcessStats;->mLastSampleTime:J
***************
*** 2402,2411 ****
  
      invoke-virtual {v3, v4, v5}, Ljava/io/PrintWriter;->print(J)V
  
!     const-string v2, "ms to "
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      move-object/from16 v0, p0
  
      iget-wide v4, v0, Lcom/android/internal/os/ProcessStats;->mCurrentSampleTime:J
--- 2720,2731 ----
  
      invoke-virtual {v3, v4, v5}, Ljava/io/PrintWriter;->print(J)V
  
!     .line 690
!     const-string/jumbo v2, "ms to "
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 691
      move-object/from16 v0, p0
  
      iget-wide v4, v0, Lcom/android/internal/os/ProcessStats;->mCurrentSampleTime:J
***************
*** 2414,2423 ****
  
      invoke-virtual {v3, v4, v5}, Ljava/io/PrintWriter;->print(J)V
  
!     const-string v2, "ms ago"
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      :goto_0
      move-object/from16 v0, p0
  
--- 2734,2745 ----
  
      invoke-virtual {v3, v4, v5}, Ljava/io/PrintWriter;->print(J)V
  
!     .line 692
!     const-string/jumbo v2, "ms ago"
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 700
      :goto_0
      move-object/from16 v0, p0
  
***************
*** 2429,2434 ****
--- 2751,2757 ----
  
      sub-long v23, v4, v6
  
+     .line 701
      .local v23, sampleTime:J
      move-object/from16 v0, p0
  
***************
*** 2440,2445 ****
--- 2763,2769 ----
  
      sub-long v21, v4, v6
  
+     .line 702
      .local v21, sampleRealTime:J
      const-wide/16 v4, 0x64
  
***************
*** 2447,2452 ****
--- 2771,2777 ----
  
      div-long v19, v4, v21
  
+     .line 703
      .local v19, percAwake:J
      const-wide/16 v4, 0x64
  
***************
*** 2454,2476 ****
--- 2779,2806 ----
  
      if-eqz v2, :cond_0
  
+     .line 704
      const-string v2, " with "
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 705
      move-wide/from16 v0, v19
  
      invoke-virtual {v3, v0, v1}, Ljava/io/PrintWriter;->print(J)V
  
+     .line 706
      const-string v2, "% awake"
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 708
      :cond_0
      const-string v2, ":"
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 710
      move-object/from16 v0, p0
  
      iget v2, v0, Lcom/android/internal/os/ProcessStats;->mRelUserTime:I
***************
*** 2505,2510 ****
--- 2835,2841 ----
  
      add-int v27, v2, v4
  
+     .line 716
      .local v27, totalTime:I
      move-object/from16 v0, p0
  
***************
*** 2514,2519 ****
--- 2845,2851 ----
  
      move-result v16
  
+     .line 717
      .local v16, N:I
      const/16 v17, 0x0
  
***************
*** 2525,2530 ****
--- 2857,2863 ----
  
      if-ge v0, v1, :cond_7
  
+     .line 718
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/internal/os/ProcessStats;->mWorkingProcs:Ljava/util/ArrayList;
***************
*** 2537,2542 ****
--- 2870,2876 ----
  
      check-cast v25, Lcom/android/internal/os/ProcessStats$Stats;
  
+     .line 719
      .local v25, st:Lcom/android/internal/os/ProcessStats$Stats;
      move-object/from16 v0, v25
  
***************
*** 2593,2598 ****
--- 2927,2933 ----
  
      invoke-direct/range {v2 .. v14}, Lcom/android/internal/os/ProcessStats;->printProcessCPU(Ljava/io/PrintWriter;Ljava/lang/String;ILjava/lang/String;IIIIIIII)V
  
+     .line 722
      move-object/from16 v0, v25
  
      iget-boolean v2, v0, Lcom/android/internal/os/ProcessStats$Stats;->removed:Z
***************
*** 2605,2610 ****
--- 2940,2946 ----
  
      if-eqz v2, :cond_6
  
+     .line 723
      move-object/from16 v0, v25
  
      iget-object v2, v0, Lcom/android/internal/os/ProcessStats$Stats;->workingThreads:Ljava/util/ArrayList;
***************
*** 2613,2618 ****
--- 2949,2955 ----
  
      move-result v15
  
+     .line 724
      .local v15, M:I
      const/16 v18, 0x0
  
***************
*** 2622,2627 ****
--- 2959,2965 ----
  
      if-ge v0, v15, :cond_6
  
+     .line 725
      move-object/from16 v0, v25
  
      iget-object v2, v0, Lcom/android/internal/os/ProcessStats$Stats;->workingThreads:Ljava/util/ArrayList;
***************
*** 2634,2639 ****
--- 2972,2978 ----
  
      check-cast v28, Lcom/android/internal/os/ProcessStats$Stats;
  
+     .line 726
      .local v28, tst:Lcom/android/internal/os/ProcessStats$Stats;
      move-object/from16 v0, v28
  
***************
*** 2686,2695 ****
--- 3025,3036 ----
  
      invoke-direct/range {v2 .. v14}, Lcom/android/internal/os/ProcessStats;->printProcessCPU(Ljava/io/PrintWriter;Ljava/lang/String;ILjava/lang/String;IIIIIIII)V
  
+     .line 724
      add-int/lit8 v18, v18, 0x1
  
      goto :goto_3
  
+     .line 694
      .end local v15           #M:I
      .end local v16           #N:I
      .end local v17           #i:I
***************
*** 2709,2718 ****
  
      invoke-virtual {v3, v4, v5}, Ljava/io/PrintWriter;->print(J)V
  
!     const-string v2, "ms to "
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      move-object/from16 v0, p0
  
      iget-wide v4, v0, Lcom/android/internal/os/ProcessStats;->mCurrentSampleTime:J
--- 3050,3061 ----
  
      invoke-virtual {v3, v4, v5}, Ljava/io/PrintWriter;->print(J)V
  
!     .line 695
!     const-string/jumbo v2, "ms to "
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
+     .line 696
      move-object/from16 v0, p0
  
      iget-wide v4, v0, Lcom/android/internal/os/ProcessStats;->mCurrentSampleTime:J
***************
*** 2721,2732 ****
  
      invoke-virtual {v3, v4, v5}, Ljava/io/PrintWriter;->print(J)V
  
!     const-string v2, "ms later"
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      goto/16 :goto_0
  
      .restart local v16       #N:I
      .restart local v17       #i:I
      .restart local v19       #percAwake:J
--- 3064,3077 ----
  
      invoke-virtual {v3, v4, v5}, Ljava/io/PrintWriter;->print(J)V
  
!     .line 697
!     const-string/jumbo v2, "ms later"
  
      invoke-virtual {v3, v2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      goto/16 :goto_0
  
+     .line 719
      .restart local v16       #N:I
      .restart local v17       #i:I
      .restart local v19       #percAwake:J
***************
*** 2750,2755 ****
--- 3095,3101 ----
  
      goto/16 :goto_2
  
+     .line 726
      .restart local v15       #M:I
      .restart local v18       #j:I
      .restart local v28       #tst:Lcom/android/internal/os/ProcessStats$Stats;
***************
*** 2769,2774 ****
--- 3115,3121 ----
  
      goto :goto_4
  
+     .line 717
      .end local v15           #M:I
      .end local v18           #j:I
      .end local v28           #tst:Lcom/android/internal/os/ProcessStats$Stats;
***************
*** 2777,2782 ****
--- 3124,3130 ----
  
      goto/16 :goto_1
  
+     .line 734
      .end local v25           #st:Lcom/android/internal/os/ProcessStats$Stats;
      :cond_7
      const-string v4, ""
***************
*** 2815,2820 ****
--- 3163,3169 ----
  
      invoke-direct/range {v2 .. v14}, Lcom/android/internal/os/ProcessStats;->printProcessCPU(Ljava/io/PrintWriter;Ljava/lang/String;ILjava/lang/String;IIIIIIII)V
  
+     .line 737
      invoke-virtual/range {v26 .. v26}, Ljava/io/StringWriter;->toString()Ljava/lang/String;
  
      move-result-object v2
***************
*** 2826,2831 ****
--- 3175,3181 ----
      .locals 26
  
      .prologue
+     .line 272
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mCurrentSampleTime:J
***************
*** 2834,2839 ****
--- 3184,3190 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mLastSampleTime:J
  
+     .line 273
      invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
  
      move-result-wide v3
***************
*** 2842,2847 ****
--- 3193,3199 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mCurrentSampleTime:J
  
+     .line 274
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mCurrentSampleRealTime:J
***************
*** 2850,2855 ****
--- 3202,3208 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mLastSampleRealTime:J
  
+     .line 275
      invoke-static {}, Landroid/os/SystemClock;->elapsedRealtime()J
  
      move-result-wide v3
***************
*** 2858,2869 ****
--- 3211,3224 ----
  
      iput-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mCurrentSampleRealTime:J
  
+     .line 277
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/os/ProcessStats;->mSystemCpuData:[J
  
      move-object/from16 v21, v0
  
+     .line 278
      .local v21, sysCpu:[J
      const-string v3, "/proc/stat"
  
***************
*** 2881,2886 ****
--- 3236,3242 ----
  
      if-eqz v3, :cond_0
  
+     .line 281
      const/4 v3, 0x0
  
      aget-wide v3, v21, v3
***************
*** 2891,2921 ****
--- 3247,3283 ----
  
      add-long v24, v3, v5
  
+     .line 283
      .local v24, usertime:J
      const/4 v3, 0x2
  
      aget-wide v22, v21, v3
  
+     .line 285
      .local v22, systemtime:J
      const/4 v3, 0x3
  
      aget-wide v9, v21, v3
  
+     .line 287
      .local v9, idletime:J
      const/4 v3, 0x4
  
      aget-wide v11, v21, v3
  
+     .line 288
      .local v11, iowaittime:J
      const/4 v3, 0x5
  
      aget-wide v13, v21, v3
  
+     .line 289
      .local v13, irqtime:J
      const/4 v3, 0x6
  
      aget-wide v19, v21, v3
  
+     .line 291
      .local v19, softirqtime:J
      move-object/from16 v0, p0
  
***************
*** 2929,2934 ****
--- 3291,3297 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats;->mRelUserTime:I
  
+     .line 292
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mBaseSystemTime:J
***************
*** 2941,2946 ****
--- 3304,3310 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats;->mRelSystemTime:I
  
+     .line 293
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mBaseIoWaitTime:J
***************
*** 2953,2958 ****
--- 3317,3323 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats;->mRelIoWaitTime:I
  
+     .line 294
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mBaseIrqTime:J
***************
*** 2965,2970 ****
--- 3330,3336 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats;->mRelIrqTime:I
  
+     .line 295
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mBaseSoftIrqTime:J
***************
*** 2977,2982 ****
--- 3343,3349 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats;->mRelSoftIrqTime:I
  
+     .line 296
      move-object/from16 v0, p0
  
      iget-wide v3, v0, Lcom/android/internal/os/ProcessStats;->mBaseIdleTime:J
***************
*** 2989,3024 ****
--- 3356,3398 ----
  
      iput v3, v0, Lcom/android/internal/os/ProcessStats;->mRelIdleTime:I
  
+     .line 307
      move-wide/from16 v0, v24
  
      move-object/from16 v2, p0
  
      iput-wide v0, v2, Lcom/android/internal/os/ProcessStats;->mBaseUserTime:J
  
+     .line 308
      move-wide/from16 v0, v22
  
      move-object/from16 v2, p0
  
      iput-wide v0, v2, Lcom/android/internal/os/ProcessStats;->mBaseSystemTime:J
  
+     .line 309
      move-object/from16 v0, p0
  
      iput-wide v11, v0, Lcom/android/internal/os/ProcessStats;->mBaseIoWaitTime:J
  
+     .line 310
      move-object/from16 v0, p0
  
      iput-wide v13, v0, Lcom/android/internal/os/ProcessStats;->mBaseIrqTime:J
  
+     .line 311
      move-wide/from16 v0, v19
  
      move-object/from16 v2, p0
  
      iput-wide v0, v2, Lcom/android/internal/os/ProcessStats;->mBaseSoftIrqTime:J
  
+     .line 312
      move-object/from16 v0, p0
  
      iput-wide v9, v0, Lcom/android/internal/os/ProcessStats;->mBaseIdleTime:J
  
+     .line 315
      .end local v9           #idletime:J
      .end local v11           #iowaittime:J
      .end local v13           #irqtime:J
***************
*** 3052,3063 ****
--- 3426,3439 ----
  
      iput-object v3, v0, Lcom/android/internal/os/ProcessStats;->mCurPids:[I
  
+     .line 317
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/os/ProcessStats;->mLoadAverageData:[F
  
      move-object/from16 v18, v0
  
+     .line 318
      .local v18, loadAverages:[F
      const-string v3, "/proc/loadavg"
  
***************
*** 3075,3094 ****
--- 3451,3474 ----
  
      if-eqz v3, :cond_2
  
+     .line 320
      const/4 v3, 0x0
  
      aget v15, v18, v3
  
+     .line 321
      .local v15, load1:F
      const/4 v3, 0x1
  
      aget v17, v18, v3
  
+     .line 322
      .local v17, load5:F
      const/4 v3, 0x2
  
      aget v16, v18, v3
  
+     .line 323
      .local v16, load15:F
      move-object/from16 v0, p0
  
***************
*** 3114,3136 ****
--- 3494,3520 ----
  
      if-eqz v3, :cond_2
  
+     .line 324
      :cond_1
      move-object/from16 v0, p0
  
      iput v15, v0, Lcom/android/internal/os/ProcessStats;->mLoad1:F
  
+     .line 325
      move/from16 v0, v17
  
      move-object/from16 v1, p0
  
      iput v0, v1, Lcom/android/internal/os/ProcessStats;->mLoad5:F
  
+     .line 326
      move/from16 v0, v16
  
      move-object/from16 v1, p0
  
      iput v0, v1, Lcom/android/internal/os/ProcessStats;->mLoad15:F
  
+     .line 327
      move-object/from16 v0, p0
  
      move/from16 v1, v17
***************
*** 3139,3144 ****
--- 3523,3529 ----
  
      invoke-virtual {v0, v15, v1, v2}, Lcom/android/internal/os/ProcessStats;->onLoadChanged(FFF)V
  
+     .line 334
      .end local v15           #load1:F
      .end local v16           #load15:F
      .end local v17           #load5:F
***************
*** 3149,3159 ****
--- 3534,3546 ----
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats;->mWorkingProcsSorted:Z
  
+     .line 335
      const/4 v3, 0x0
  
      move-object/from16 v0, p0
  
      iput-boolean v3, v0, Lcom/android/internal/os/ProcessStats;->mFirst:Z
  
+     .line 336
      return-void
  .end method
