*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/RuntimeInit$UncaughtHandler.smali	2012-03-23 19:37:52.078724737 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/RuntimeInit$UncaughtHandler.smali.original	2012-03-23 19:34:46.381338550 -0500
***************
*** 22,27 ****
--- 22,28 ----
      .locals 0
  
      .prologue
+     .line 58
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 32,37 ****
--- 33,39 ----
      .parameter "x0"
  
      .prologue
+     .line 58
      invoke-direct {p0}, Lcom/android/internal/os/RuntimeInit$UncaughtHandler;-><init>()V
  
      return-void
***************
*** 47,52 ****
--- 49,55 ----
      .prologue
      const/16 v4, 0xa
  
+     .line 62
      :try_start_0
      invoke-static {}, Lcom/android/internal/os/RuntimeInit;->access$000()Z
      :try_end_0
***************
*** 57,85 ****
--- 60,94 ----
  
      if-eqz v1, :cond_0
  
+     .line 82
      invoke-static {}, Landroid/os/Process;->myPid()I
  
      move-result v1
  
      invoke-static {v1}, Landroid/os/Process;->killProcess(I)V
  
+     .line 83
      invoke-static {v4}, Ljava/lang/System;->exit(I)V
  
+     .line 85
      :goto_0
      return-void
  
+     .line 63
      :cond_0
      const/4 v1, 0x1
  
      :try_start_1
      invoke-static {v1}, Lcom/android/internal/os/RuntimeInit;->access$002(Z)Z
  
+     .line 65
      invoke-static {}, Lcom/android/internal/os/RuntimeInit;->access$100()Landroid/os/IBinder;
  
      move-result-object v1
  
      if-nez v1, :cond_1
  
+     .line 66
      const-string v1, "AndroidRuntime"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 106,111 ****
--- 115,121 ----
  
      invoke-static {v1, v2, p2}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 72
      :goto_1
      invoke-static {}, Landroid/app/ActivityManagerNative;->getDefault()Landroid/app/IActivityManager;
  
***************
*** 124,139 ****
--- 134,152 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Ljava/lang/Throwable; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 82
      invoke-static {}, Landroid/os/Process;->myPid()I
  
      move-result v1
  
      invoke-static {v1}, Landroid/os/Process;->killProcess(I)V
  
+     .line 83
      invoke-static {v4}, Ljava/lang/System;->exit(I)V
  
      goto :goto_0
  
+     .line 68
      :cond_1
      :try_start_2
      const-string v1, "AndroidRuntime"
***************
*** 167,175 ****
--- 180,190 ----
  
      goto :goto_1
  
+     .line 74
      :catch_0
      move-exception v0
  
+     .line 76
      .local v0, t2:Ljava/lang/Throwable;
      :try_start_3
      const-string v1, "AndroidRuntime"
***************
*** 181,186 ****
--- 196,202 ----
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
      .catch Ljava/lang/Throwable; {:try_start_3 .. :try_end_3} :catch_1
  
+     .line 82
      :goto_2
      invoke-static {}, Landroid/os/Process;->myPid()I
  
***************
*** 188,197 ****
--- 204,215 ----
  
      invoke-static {v1}, Landroid/os/Process;->killProcess(I)V
  
+     .line 83
      invoke-static {v4}, Ljava/lang/System;->exit(I)V
  
      goto :goto_0
  
+     .line 82
      .end local v0           #t2:Ljava/lang/Throwable;
      :catchall_0
      move-exception v1
***************
*** 202,211 ****
--- 220,231 ----
  
      invoke-static {v2}, Landroid/os/Process;->killProcess(I)V
  
+     .line 83
      invoke-static {v4}, Ljava/lang/System;->exit(I)V
  
      throw v1
  
+     .line 77
      .restart local v0       #t2:Ljava/lang/Throwable;
      :catch_1
      move-exception v1
