*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/SamplingProfilerIntegration.smali	2012-03-23 19:37:51.405712465 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/SamplingProfilerIntegration.smali.original	2012-03-23 19:34:46.375338441 -0500
***************
*** 34,46 ****
  
      const/4 v3, 0x0
  
      new-instance v1, Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-direct {v1, v3}, Ljava/util/concurrent/atomic/AtomicBoolean;-><init>(Z)V
  
      sput-object v1, Lcom/android/internal/os/SamplingProfilerIntegration;->pending:Ljava/util/concurrent/atomic/AtomicBoolean;
  
!     const-string v1, "persist.sys.profiler_ms"
  
      invoke-static {v1, v3}, Landroid/os/SystemProperties;->getInt(Ljava/lang/String;I)I
  
--- 34,48 ----
  
      const/4 v3, 0x0
  
+     .line 53
      new-instance v1, Ljava/util/concurrent/atomic/AtomicBoolean;
  
      invoke-direct {v1, v3}, Ljava/util/concurrent/atomic/AtomicBoolean;-><init>(Z)V
  
      sput-object v1, Lcom/android/internal/os/SamplingProfilerIntegration;->pending:Ljava/util/concurrent/atomic/AtomicBoolean;
  
!     .line 56
!     const-string/jumbo v1, "persist.sys.profiler_ms"
  
      invoke-static {v1, v3}, Landroid/os/SystemProperties;->getInt(Ljava/lang/String;I)I
  
***************
*** 48,54 ****
  
      sput v1, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfilerMilliseconds:I
  
!     const-string v1, "persist.sys.profiler_depth"
  
      const/4 v2, 0x4
  
--- 50,57 ----
  
      sput v1, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfilerMilliseconds:I
  
!     .line 57
!     const-string/jumbo v1, "persist.sys.profiler_depth"
  
      const/4 v2, 0x4
  
***************
*** 58,86 ****
--- 61,96 ----
  
      sput v1, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfilerDepth:I
  
+     .line 58
      sget v1, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfilerMilliseconds:I
  
      if-lez v1, :cond_1
  
+     .line 59
      new-instance v0, Ljava/io/File;
  
      const-string v1, "/data/snapshots"
  
      invoke-direct {v0, v1}, Ljava/io/File;-><init>(Ljava/lang/String;)V
  
+     .line 60
      .local v0, dir:Ljava/io/File;
      invoke-virtual {v0}, Ljava/io/File;->mkdirs()Z
  
+     .line 62
      invoke-virtual {v0, v4, v3}, Ljava/io/File;->setWritable(ZZ)Z
  
+     .line 64
      invoke-virtual {v0, v4, v3}, Ljava/io/File;->setExecutable(ZZ)Z
  
+     .line 65
      invoke-virtual {v0}, Ljava/io/File;->isDirectory()Z
  
      move-result v1
  
      if-eqz v1, :cond_0
  
+     .line 66
      new-instance v1, Lcom/android/internal/os/SamplingProfilerIntegration$1;
  
      invoke-direct {v1}, Lcom/android/internal/os/SamplingProfilerIntegration$1;-><init>()V
***************
*** 91,98 ****
--- 101,110 ----
  
      sput-object v1, Lcom/android/internal/os/SamplingProfilerIntegration;->snapshotWriter:Ljava/util/concurrent/Executor;
  
+     .line 71
      sput-boolean v4, Lcom/android/internal/os/SamplingProfilerIntegration;->enabled:Z
  
+     .line 72
      const-string v1, "SamplingProfilerIntegration"
  
      new-instance v2, Ljava/lang/StringBuilder;
***************
*** 117,132 ****
--- 129,148 ----
  
      invoke-static {v1, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 84
      .end local v0           #dir:Ljava/io/File;
      :goto_0
      return-void
  
+     .line 75
      .restart local v0       #dir:Ljava/io/File;
      :cond_0
      sput-object v5, Lcom/android/internal/os/SamplingProfilerIntegration;->snapshotWriter:Ljava/util/concurrent/Executor;
  
+     .line 76
      sput-boolean v4, Lcom/android/internal/os/SamplingProfilerIntegration;->enabled:Z
  
+     .line 77
      const-string v1, "SamplingProfilerIntegration"
  
      const-string v2, "Profiling setup failed. Could not create /data/snapshots"
***************
*** 135,146 ****
--- 151,165 ----
  
      goto :goto_0
  
+     .line 80
      .end local v0           #dir:Ljava/io/File;
      :cond_1
      sput-object v5, Lcom/android/internal/os/SamplingProfilerIntegration;->snapshotWriter:Ljava/util/concurrent/Executor;
  
+     .line 81
      sput-boolean v3, Lcom/android/internal/os/SamplingProfilerIntegration;->enabled:Z
  
+     .line 82
      const-string v1, "SamplingProfilerIntegration"
  
      const-string v2, "Profiling disabled."
***************
*** 154,159 ****
--- 173,179 ----
      .locals 0
  
      .prologue
+     .line 41
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 165,170 ****
--- 185,191 ----
      .parameter "x1"
  
      .prologue
+     .line 41
      invoke-static {p0, p1}, Lcom/android/internal/os/SamplingProfilerIntegration;->writeSnapshotFile(Ljava/lang/String;Landroid/content/pm/PackageInfo;)V
  
      return-void
***************
*** 174,179 ****
--- 195,201 ----
      .locals 1
  
      .prologue
+     .line 41
      sget-object v0, Lcom/android/internal/os/SamplingProfilerIntegration;->pending:Ljava/util/concurrent/atomic/AtomicBoolean;
  
      return-object v0
***************
*** 186,195 ****
--- 208,219 ----
      .parameter "out"
  
      .prologue
+     .line 216
      const-string v0, "Version: 3"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintStream;->println(Ljava/lang/String;)V
  
+     .line 217
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 210,217 ****
--- 234,243 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintStream;->println(Ljava/lang/String;)V
  
+     .line 218
      if-eqz p1, :cond_0
  
+     .line 219
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 234,239 ****
--- 260,266 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintStream;->println(Ljava/lang/String;)V
  
+     .line 220
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 256,261 ****
--- 283,289 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintStream;->println(Ljava/lang/String;)V
  
+     .line 222
      :cond_0
      new-instance v0, Ljava/lang/StringBuilder;
  
***************
*** 279,286 ****
--- 307,316 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintStream;->println(Ljava/lang/String;)V
  
+     .line 224
      invoke-virtual {p2}, Ljava/io/PrintStream;->println()V
  
+     .line 225
      return-void
  .end method
  
***************
*** 288,293 ****
--- 318,324 ----
      .locals 1
  
      .prologue
+     .line 93
      sget-boolean v0, Lcom/android/internal/os/SamplingProfilerIntegration;->enabled:Z
  
      return v0
***************
*** 297,311 ****
--- 328,345 ----
      .locals 7
  
      .prologue
+     .line 100
      sget-boolean v2, Lcom/android/internal/os/SamplingProfilerIntegration;->enabled:Z
  
      if-nez v2, :cond_0
  
+     .line 113
      .local v0, group:Ljava/lang/ThreadGroup;
      .local v1, threadSet:Ldalvik/system/profiler/SamplingProfiler$ThreadSet;
      :goto_0
      return-void
  
+     .line 103
      .end local v0           #group:Ljava/lang/ThreadGroup;
      .end local v1           #threadSet:Ldalvik/system/profiler/SamplingProfiler$ThreadSet;
      :cond_0
***************
*** 313,318 ****
--- 347,353 ----
  
      if-eqz v2, :cond_1
  
+     .line 104
      const-string v2, "SamplingProfilerIntegration"
  
      new-instance v3, Ljava/lang/StringBuilder;
***************
*** 343,348 ****
--- 378,384 ----
  
      goto :goto_0
  
+     .line 108
      :cond_1
      invoke-static {}, Ljava/lang/Thread;->currentThread()Ljava/lang/Thread;
  
***************
*** 352,362 ****
--- 388,400 ----
  
      move-result-object v0
  
+     .line 109
      .restart local v0       #group:Ljava/lang/ThreadGroup;
      invoke-static {v0}, Ldalvik/system/profiler/SamplingProfiler;->newThreadGroupTheadSet(Ljava/lang/ThreadGroup;)Ldalvik/system/profiler/SamplingProfiler$ThreadSet;
  
      move-result-object v1
  
+     .line 110
      .restart local v1       #threadSet:Ldalvik/system/profiler/SamplingProfiler$ThreadSet;
      new-instance v2, Ldalvik/system/profiler/SamplingProfiler;
  
***************
*** 366,377 ****
--- 404,417 ----
  
      sput-object v2, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfiler:Ldalvik/system/profiler/SamplingProfiler;
  
+     .line 111
      sget-object v2, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfiler:Ldalvik/system/profiler/SamplingProfiler;
  
      sget v3, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfilerMilliseconds:I
  
      invoke-virtual {v2, v3}, Ldalvik/system/profiler/SamplingProfiler;->start(I)V
  
+     .line 112
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v2
***************
*** 387,405 ****
--- 427,449 ----
      .parameter "packageInfo"
  
      .prologue
+     .line 119
      sget-boolean v0, Lcom/android/internal/os/SamplingProfilerIntegration;->enabled:Z
  
      if-nez v0, :cond_1
  
+     .line 144
      :cond_0
      :goto_0
      return-void
  
+     .line 122
      :cond_1
      sget-object v0, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfiler:Ldalvik/system/profiler/SamplingProfiler;
  
      if-nez v0, :cond_2
  
+     .line 123
      const-string v0, "SamplingProfilerIntegration"
  
      const-string v1, "SamplingProfilerIntegration is not started"
***************
*** 408,413 ****
--- 452,458 ----
  
      goto :goto_0
  
+     .line 133
      :cond_2
      sget-object v0, Lcom/android/internal/os/SamplingProfilerIntegration;->pending:Ljava/util/concurrent/atomic/AtomicBoolean;
  
***************
*** 421,426 ****
--- 466,472 ----
  
      if-eqz v0, :cond_0
  
+     .line 134
      sget-object v0, Lcom/android/internal/os/SamplingProfilerIntegration;->snapshotWriter:Ljava/util/concurrent/Executor;
  
      new-instance v1, Lcom/android/internal/os/SamplingProfilerIntegration$2;
***************
*** 438,455 ****
--- 484,505 ----
      .parameter "packageInfo"
  
      .prologue
+     .line 163
      sget-boolean v10, Lcom/android/internal/os/SamplingProfilerIntegration;->enabled:Z
  
      if-nez v10, :cond_0
  
+     .line 199
      :goto_0
      return-void
  
+     .line 166
      :cond_0
      sget-object v10, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfiler:Ldalvik/system/profiler/SamplingProfiler;
  
      invoke-virtual {v10}, Ldalvik/system/profiler/SamplingProfiler;->stop()V
  
+     .line 174
      const-string v10, ":"
  
      const-string v11, "."
***************
*** 458,463 ****
--- 508,514 ----
  
      move-result-object v3
  
+     .line 175
      .local v3, name:Ljava/lang/String;
      new-instance v10, Ljava/lang/StringBuilder;
  
***************
*** 495,508 ****
--- 546,562 ----
  
      move-result-object v7
  
+     .line 176
      .local v7, path:Ljava/lang/String;
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v8
  
+     .line 177
      .local v8, start:J
      const/4 v5, 0x0
  
+     .line 179
      .local v5, outputStream:Ljava/io/OutputStream;
      :try_start_0
      new-instance v6, Ljava/io/BufferedOutputStream;
***************
*** 516,521 ****
--- 570,576 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 180
      .end local v5           #outputStream:Ljava/io/OutputStream;
      .local v6, outputStream:Ljava/io/OutputStream;
      :try_start_1
***************
*** 523,537 ****
--- 578,595 ----
  
      invoke-direct {v4, v6}, Ljava/io/PrintStream;-><init>(Ljava/io/OutputStream;)V
  
+     .line 181
      .local v4, out:Ljava/io/PrintStream;
      invoke-static {v3, p1, v4}, Lcom/android/internal/os/SamplingProfilerIntegration;->generateSnapshotHeader(Ljava/lang/String;Landroid/content/pm/PackageInfo;Ljava/io/PrintStream;)V
  
+     .line 182
      invoke-virtual {v4}, Ljava/io/PrintStream;->checkError()Z
  
      move-result v10
  
      if-eqz v10, :cond_1
  
+     .line 183
      new-instance v10, Ljava/io/IOException;
  
      invoke-direct {v10}, Ljava/io/IOException;-><init>()V
***************
*** 541,552 ****
--- 599,612 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_1
      .catch Ljava/io/IOException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 186
      .end local v4           #out:Ljava/io/PrintStream;
      :catch_0
      move-exception v0
  
      move-object v5, v6
  
+     .line 187
      .end local v6           #outputStream:Ljava/io/OutputStream;
      .local v0, e:Ljava/io/IOException;
      .restart local v5       #outputStream:Ljava/io/OutputStream;
***************
*** 576,585 ****
--- 636,647 ----
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
  
+     .line 190
      invoke-static {v5}, Llibcore/io/IoUtils;->closeQuietly(Ljava/lang/AutoCloseable;)V
  
      goto :goto_0
  
+     .line 185
      .end local v0           #e:Ljava/io/IOException;
      .end local v5           #outputStream:Ljava/io/OutputStream;
      .restart local v4       #out:Ljava/io/PrintStream;
***************
*** 597,604 ****
--- 659,668 ----
      .catchall {:try_start_3 .. :try_end_3} :catchall_1
      .catch Ljava/io/IOException; {:try_start_3 .. :try_end_3} :catch_0
  
+     .line 190
      invoke-static {v6}, Llibcore/io/IoUtils;->closeQuietly(Ljava/lang/AutoCloseable;)V
  
+     .line 194
      new-instance v10, Ljava/io/File;
  
      invoke-direct {v10, v7}, Ljava/io/File;-><init>(Ljava/lang/String;)V
***************
*** 609,620 ****
--- 673,686 ----
  
      invoke-virtual {v10, v11, v12}, Ljava/io/File;->setReadable(ZZ)Z
  
+     .line 196
      invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
  
      move-result-wide v10
  
      sub-long v1, v10, v8
  
+     .line 197
      .local v1, elapsed:J
      const-string v10, "SamplingProfilerIntegration"
  
***************
*** 642,648 ****
  
      move-result-object v11
  
!     const-string v12, "ms."
  
      invoke-virtual {v11, v12}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
--- 708,714 ----
  
      move-result-object v11
  
!     const-string/jumbo v12, "ms."
  
      invoke-virtual {v11, v12}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
***************
*** 654,659 ****
--- 720,726 ----
  
      invoke-static {v10, v11}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 198
      sget-object v10, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfiler:Ldalvik/system/profiler/SamplingProfiler;
  
      sget v11, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfilerMilliseconds:I
***************
*** 662,667 ****
--- 729,735 ----
  
      goto/16 :goto_0
  
+     .line 190
      .end local v1           #elapsed:J
      .end local v4           #out:Ljava/io/PrintStream;
      .end local v6           #outputStream:Ljava/io/OutputStream;
***************
*** 685,690 ****
--- 753,759 ----
      .restart local v5       #outputStream:Ljava/io/OutputStream;
      goto :goto_2
  
+     .line 186
      :catch_1
      move-exception v0
  
***************
*** 697,720 ****
      .prologue
      const/4 v1, 0x0
  
      sget-boolean v0, Lcom/android/internal/os/SamplingProfilerIntegration;->enabled:Z
  
      if-nez v0, :cond_0
  
      :goto_0
      return-void
  
      :cond_0
!     const-string v0, "zygote"
  
      invoke-static {v0, v1}, Lcom/android/internal/os/SamplingProfilerIntegration;->writeSnapshotFile(Ljava/lang/String;Landroid/content/pm/PackageInfo;)V
  
      sget-object v0, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfiler:Ldalvik/system/profiler/SamplingProfiler;
  
      invoke-virtual {v0}, Ldalvik/system/profiler/SamplingProfiler;->shutdown()V
  
      sput-object v1, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfiler:Ldalvik/system/profiler/SamplingProfiler;
  
      const-wide/16 v0, 0x0
  
      sput-wide v0, Lcom/android/internal/os/SamplingProfilerIntegration;->startMillis:J
--- 766,795 ----
      .prologue
      const/4 v1, 0x0
  
+     .line 150
      sget-boolean v0, Lcom/android/internal/os/SamplingProfilerIntegration;->enabled:Z
  
      if-nez v0, :cond_0
  
+     .line 157
      :goto_0
      return-void
  
+     .line 153
      :cond_0
!     const-string/jumbo v0, "zygote"
  
      invoke-static {v0, v1}, Lcom/android/internal/os/SamplingProfilerIntegration;->writeSnapshotFile(Ljava/lang/String;Landroid/content/pm/PackageInfo;)V
  
+     .line 154
      sget-object v0, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfiler:Ldalvik/system/profiler/SamplingProfiler;
  
      invoke-virtual {v0}, Ldalvik/system/profiler/SamplingProfiler;->shutdown()V
  
+     .line 155
      sput-object v1, Lcom/android/internal/os/SamplingProfilerIntegration;->samplingProfiler:Ldalvik/system/profiler/SamplingProfiler;
  
+     .line 156
      const-wide/16 v0, 0x0
  
      sput-wide v0, Lcom/android/internal/os/SamplingProfilerIntegration;->startMillis:J
