*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/ZygoteConnection$Arguments.smali	2012-03-23 19:37:51.243709510 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/ZygoteConnection$Arguments.smali.original	2012-03-23 19:34:46.374338423 -0500
***************
*** 72,85 ****
--- 72,90 ----
      .prologue
      const/4 v0, 0x0
  
+     .line 372
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 318
      iput v0, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->uid:I
  
+     .line 322
      iput v0, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->gid:I
  
+     .line 373
      invoke-direct {p0, p1}, Lcom/android/internal/os/ZygoteConnection$Arguments;->parseArgs([Ljava/lang/String;)V
  
+     .line 374
      return-void
  .end method
  
***************
*** 99,114 ****
--- 104,123 ----
  
      const/4 v11, 0x1
  
+     .line 386
      const/4 v3, 0x0
  
+     .line 388
      .local v3, curArg:I
      :goto_0
      array-length v9, p1
  
      if-ge v3, v9, :cond_0
  
+     .line 389
      aget-object v0, p1, v3
  
+     .line 391
      .local v0, arg:Ljava/lang/String;
      const-string v9, "--"
  
***************
*** 118,125 ****
--- 127,136 ----
  
      if-eqz v9, :cond_1
  
+     .line 392
      add-int/lit8 v3, v3, 0x1
  
+     .line 516
      .end local v0           #arg:Ljava/lang/String;
      :cond_0
      iget-boolean v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->runtimeInit:Z
***************
*** 130,135 ****
--- 141,147 ----
  
      if-eqz v9, :cond_1e
  
+     .line 517
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "--runtime-init and -classpath are incompatible"
***************
*** 138,143 ****
--- 150,156 ----
  
      throw v9
  
+     .line 394
      .restart local v0       #arg:Ljava/lang/String;
      :cond_1
      const-string v9, "--setuid="
***************
*** 148,157 ****
--- 161,172 ----
  
      if-eqz v9, :cond_4
  
+     .line 395
      iget-boolean v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->uidSpecified:Z
  
      if-eqz v9, :cond_2
  
+     .line 396
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "Duplicate arg specified"
***************
*** 160,168 ****
--- 175,185 ----
  
      throw v9
  
+     .line 399
      :cond_2
      iput-boolean v11, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->uidSpecified:Z
  
+     .line 400
      invoke-virtual {v0, v12}, Ljava/lang/String;->indexOf(I)I
  
      move-result v9
***************
*** 179,190 ****
--- 196,209 ----
  
      iput v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->uid:I
  
+     .line 388
      :cond_3
      :goto_1
      add-int/lit8 v3, v3, 0x1
  
      goto :goto_0
  
+     .line 402
      :cond_4
      const-string v9, "--setgid="
  
***************
*** 194,203 ****
--- 213,224 ----
  
      if-eqz v9, :cond_6
  
+     .line 403
      iget-boolean v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->gidSpecified:Z
  
      if-eqz v9, :cond_5
  
+     .line 404
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "Duplicate arg specified"
***************
*** 206,214 ****
--- 227,237 ----
  
      throw v9
  
+     .line 407
      :cond_5
      iput-boolean v11, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->gidSpecified:Z
  
+     .line 408
      invoke-virtual {v0, v12}, Ljava/lang/String;->indexOf(I)I
  
      move-result v9
***************
*** 227,232 ****
--- 250,256 ----
  
      goto :goto_1
  
+     .line 410
      :cond_6
      const-string v9, "--target-sdk-version="
  
***************
*** 236,245 ****
--- 260,271 ----
  
      if-eqz v9, :cond_8
  
+     .line 411
      iget-boolean v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->targetSdkVersionSpecified:Z
  
      if-eqz v9, :cond_7
  
+     .line 412
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "Duplicate target-sdk-version specified"
***************
*** 248,256 ****
--- 274,284 ----
  
      throw v9
  
+     .line 415
      :cond_7
      iput-boolean v11, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->targetSdkVersionSpecified:Z
  
+     .line 416
      invoke-virtual {v0, v12}, Ljava/lang/String;->indexOf(I)I
  
      move-result v9
***************
*** 269,274 ****
--- 297,303 ----
  
      goto :goto_1
  
+     .line 418
      :cond_8
      const-string v9, "--enable-debugger"
  
***************
*** 278,283 ****
--- 307,313 ----
  
      if-eqz v9, :cond_9
  
+     .line 419
      iget v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->debugFlags:I
  
      or-int/lit8 v9, v9, 0x1
***************
*** 286,291 ****
--- 316,322 ----
  
      goto :goto_1
  
+     .line 420
      :cond_9
      const-string v9, "--enable-safemode"
  
***************
*** 295,300 ****
--- 326,332 ----
  
      if-eqz v9, :cond_a
  
+     .line 421
      iget v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->debugFlags:I
  
      or-int/lit8 v9, v9, 0x8
***************
*** 303,308 ****
--- 335,341 ----
  
      goto :goto_1
  
+     .line 422
      :cond_a
      const-string v9, "--enable-checkjni"
  
***************
*** 312,317 ****
--- 345,351 ----
  
      if-eqz v9, :cond_b
  
+     .line 423
      iget v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->debugFlags:I
  
      or-int/lit8 v9, v9, 0x2
***************
*** 320,325 ****
--- 354,360 ----
  
      goto :goto_1
  
+     .line 424
      :cond_b
      const-string v9, "--enable-jni-logging"
  
***************
*** 329,334 ****
--- 364,370 ----
  
      if-eqz v9, :cond_c
  
+     .line 425
      iget v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->debugFlags:I
  
      or-int/lit8 v9, v9, 0x10
***************
*** 337,342 ****
--- 373,379 ----
  
      goto/16 :goto_1
  
+     .line 426
      :cond_c
      const-string v9, "--enable-assert"
  
***************
*** 346,351 ****
--- 383,389 ----
  
      if-eqz v9, :cond_d
  
+     .line 427
      iget v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->debugFlags:I
  
      or-int/lit8 v9, v9, 0x4
***************
*** 354,359 ****
--- 392,398 ----
  
      goto/16 :goto_1
  
+     .line 428
      :cond_d
      const-string v9, "--peer-wait"
  
***************
*** 363,372 ****
--- 402,413 ----
  
      if-eqz v9, :cond_e
  
+     .line 429
      iput-boolean v11, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->peerWait:Z
  
      goto/16 :goto_1
  
+     .line 430
      :cond_e
      const-string v9, "--runtime-init"
  
***************
*** 376,385 ****
--- 417,428 ----
  
      if-eqz v9, :cond_f
  
+     .line 431
      iput-boolean v11, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->runtimeInit:Z
  
      goto/16 :goto_1
  
+     .line 432
      :cond_f
      const-string v9, "--capabilities="
  
***************
*** 389,398 ****
--- 432,443 ----
  
      if-eqz v9, :cond_12
  
+     .line 433
      iget-boolean v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->capabilitiesSpecified:Z
  
      if-eqz v9, :cond_10
  
+     .line 434
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "Duplicate arg specified"
***************
*** 401,409 ****
--- 446,456 ----
  
      throw v9
  
+     .line 437
      :cond_10
      iput-boolean v11, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->capabilitiesSpecified:Z
  
+     .line 438
      invoke-virtual {v0, v12}, Ljava/lang/String;->indexOf(I)I
  
      move-result v9
***************
*** 414,419 ****
--- 461,467 ----
  
      move-result-object v1
  
+     .line 440
      .local v1, capString:Ljava/lang/String;
      const-string v9, ","
  
***************
*** 423,433 ****
--- 471,483 ----
  
      move-result-object v2
  
+     .line 442
      .local v2, capStrings:[Ljava/lang/String;
      array-length v9, v2
  
      if-ne v9, v11, :cond_11
  
+     .line 443
      aget-object v9, v2, v13
  
      invoke-static {v9}, Ljava/lang/Long;->decode(Ljava/lang/String;)Ljava/lang/Long;
***************
*** 440,451 ****
--- 490,503 ----
  
      iput-wide v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->effectiveCapabilities:J
  
+     .line 444
      iget-wide v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->effectiveCapabilities:J
  
      iput-wide v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->permittedCapabilities:J
  
      goto/16 :goto_1
  
+     .line 446
      :cond_11
      aget-object v9, v2, v13
  
***************
*** 459,464 ****
--- 511,517 ----
  
      iput-wide v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->permittedCapabilities:J
  
+     .line 447
      aget-object v9, v2, v11
  
      invoke-static {v9}, Ljava/lang/Long;->decode(Ljava/lang/String;)Ljava/lang/Long;
***************
*** 473,478 ****
--- 526,532 ----
  
      goto/16 :goto_1
  
+     .line 449
      .end local v1           #capString:Ljava/lang/String;
      .end local v2           #capStrings:[Ljava/lang/String;
      :cond_12
***************
*** 484,489 ****
--- 538,544 ----
  
      if-eqz v9, :cond_16
  
+     .line 451
      invoke-virtual {v0, v12}, Ljava/lang/String;->indexOf(I)I
  
      move-result v9
***************
*** 500,505 ****
--- 555,561 ----
  
      move-result-object v6
  
+     .line 454
      .local v6, limitStrings:[Ljava/lang/String;
      array-length v9, v6
  
***************
*** 507,512 ****
--- 563,569 ----
  
      if-eq v9, v10, :cond_13
  
+     .line 455
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "--rlimit= should have 3 comma-delimited ints"
***************
*** 515,525 ****
--- 572,584 ----
  
      throw v9
  
+     .line 458
      :cond_13
      array-length v9, v6
  
      new-array v8, v9, [I
  
+     .line 460
      .local v8, rlimitTuple:[I
      const/4 v5, 0x0
  
***************
*** 529,534 ****
--- 588,594 ----
  
      if-ge v5, v9, :cond_14
  
+     .line 461
      aget-object v9, v6, v5
  
      invoke-static {v9}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
***************
*** 537,557 ****
--- 597,621 ----
  
      aput v9, v8, v5
  
+     .line 460
      add-int/lit8 v5, v5, 0x1
  
      goto :goto_2
  
+     .line 464
      :cond_14
      iget-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->rlimits:Ljava/util/ArrayList;
  
      if-nez v9, :cond_15
  
+     .line 465
      new-instance v9, Ljava/util/ArrayList;
  
      invoke-direct {v9}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->rlimits:Ljava/util/ArrayList;
  
+     .line 468
      :cond_15
      iget-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->rlimits:Ljava/util/ArrayList;
  
***************
*** 559,564 ****
--- 623,629 ----
  
      goto/16 :goto_1
  
+     .line 469
      .end local v5           #i:I
      .end local v6           #limitStrings:[Ljava/lang/String;
      .end local v8           #rlimitTuple:[I
***************
*** 571,580 ****
--- 636,647 ----
  
      if-eqz v9, :cond_18
  
+     .line 470
      iget-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->classpath:Ljava/lang/String;
  
      if-eqz v9, :cond_17
  
+     .line 471
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "Duplicate arg specified"
***************
*** 583,588 ****
--- 650,656 ----
  
      throw v9
  
+     .line 475
      :cond_17
      add-int/lit8 v3, v3, 0x1
  
***************
*** 595,603 ****
--- 663,673 ----
  
      goto/16 :goto_1
  
+     .line 476
      :catch_0
      move-exception v4
  
+     .line 477
      .local v4, ex:Ljava/lang/IndexOutOfBoundsException;
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
***************
*** 607,612 ****
--- 677,683 ----
  
      throw v9
  
+     .line 480
      .end local v4           #ex:Ljava/lang/IndexOutOfBoundsException;
      :cond_18
      const-string v9, "--setgroups="
***************
*** 617,626 ****
--- 688,699 ----
  
      if-eqz v9, :cond_1a
  
+     .line 481
      iget-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->gids:[I
  
      if-eqz v9, :cond_19
  
+     .line 482
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "Duplicate arg specified"
***************
*** 629,634 ****
--- 702,708 ----
  
      throw v9
  
+     .line 486
      :cond_19
      invoke-virtual {v0, v12}, Ljava/lang/String;->indexOf(I)I
  
***************
*** 646,651 ****
--- 720,726 ----
  
      move-result-object v7
  
+     .line 489
      .local v7, params:[Ljava/lang/String;
      array-length v9, v7
  
***************
*** 653,658 ****
--- 728,734 ----
  
      iput-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->gids:[I
  
+     .line 491
      array-length v9, v7
  
      add-int/lit8 v5, v9, -0x1
***************
*** 661,666 ****
--- 737,743 ----
      :goto_3
      if-ltz v5, :cond_3
  
+     .line 492
      iget-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->gids:[I
  
      aget-object v10, v7, v5
***************
*** 671,680 ****
--- 748,759 ----
  
      aput v10, v9, v5
  
+     .line 491
      add-int/lit8 v5, v5, -0x1
  
      goto :goto_3
  
+     .line 494
      .end local v5           #i:I
      .end local v7           #params:[Ljava/lang/String;
      :cond_1a
***************
*** 686,695 ****
--- 765,776 ----
  
      if-eqz v9, :cond_1c
  
+     .line 495
      iget-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->invokeWith:Ljava/lang/String;
  
      if-eqz v9, :cond_1b
  
+     .line 496
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "Duplicate arg specified"
***************
*** 698,703 ****
--- 779,785 ----
  
      throw v9
  
+     .line 500
      :cond_1b
      add-int/lit8 v3, v3, 0x1
  
***************
*** 710,718 ****
--- 792,802 ----
  
      goto/16 :goto_1
  
+     .line 501
      :catch_1
      move-exception v4
  
+     .line 502
      .restart local v4       #ex:Ljava/lang/IndexOutOfBoundsException;
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
***************
*** 722,727 ****
--- 806,812 ----
  
      throw v9
  
+     .line 505
      .end local v4           #ex:Ljava/lang/IndexOutOfBoundsException;
      :cond_1c
      const-string v9, "--nice-name="
***************
*** 732,741 ****
--- 817,828 ----
  
      if-eqz v9, :cond_0
  
+     .line 506
      iget-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->niceName:Ljava/lang/String;
  
      if-eqz v9, :cond_1d
  
+     .line 507
      new-instance v9, Ljava/lang/IllegalArgumentException;
  
      const-string v10, "Duplicate arg specified"
***************
*** 744,749 ****
--- 831,837 ----
  
      throw v9
  
+     .line 510
      :cond_1d
      invoke-virtual {v0, v12}, Ljava/lang/String;->indexOf(I)I
  
***************
*** 759,764 ****
--- 847,853 ----
  
      goto/16 :goto_1
  
+     .line 521
      .end local v0           #arg:Ljava/lang/String;
      :cond_1e
      array-length v9, p1
***************
*** 769,774 ****
--- 858,864 ----
  
      iput-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->remainingArgs:[Ljava/lang/String;
  
+     .line 523
      iget-object v9, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->remainingArgs:[Ljava/lang/String;
  
      iget-object v10, p0, Lcom/android/internal/os/ZygoteConnection$Arguments;->remainingArgs:[Ljava/lang/String;
***************
*** 777,781 ****
--- 867,872 ----
  
      invoke-static {p1, v3, v9, v13, v10}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
+     .line 525
      return-void
  .end method
