*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/ZygoteInit$MethodAndArgsCaller.smali	2012-03-23 19:37:52.120725503 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/os/ZygoteInit$MethodAndArgsCaller.smali.original	2012-03-23 19:34:46.381338550 -0500
***************
*** 30,41 ****
--- 30,45 ----
      .parameter "args"
  
      .prologue
+     .line 777
      invoke-direct {p0}, Ljava/lang/Exception;-><init>()V
  
+     .line 778
      iput-object p1, p0, Lcom/android/internal/os/ZygoteInit$MethodAndArgsCaller;->mMethod:Ljava/lang/reflect/Method;
  
+     .line 779
      iput-object p2, p0, Lcom/android/internal/os/ZygoteInit$MethodAndArgsCaller;->mArgs:[Ljava/lang/String;
  
+     .line 780
      return-void
  .end method
  
***************
*** 45,50 ****
--- 49,55 ----
      .locals 7
  
      .prologue
+     .line 784
      :try_start_0
      iget-object v2, p0, Lcom/android/internal/os/ZygoteInit$MethodAndArgsCaller;->mMethod:Ljava/lang/reflect/Method;
  
***************
*** 65,75 ****
--- 70,83 ----
      .catch Ljava/lang/IllegalAccessException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/lang/reflect/InvocationTargetException; {:try_start_0 .. :try_end_0} :catch_1
  
+     .line 796
      return-void
  
+     .line 785
      :catch_0
      move-exception v1
  
+     .line 786
      .local v1, ex:Ljava/lang/IllegalAccessException;
      new-instance v2, Ljava/lang/RuntimeException;
  
***************
*** 77,112 ****
--- 85,127 ----
  
      throw v2
  
+     .line 787
      .end local v1           #ex:Ljava/lang/IllegalAccessException;
      :catch_1
      move-exception v1
  
+     .line 788
      .local v1, ex:Ljava/lang/reflect/InvocationTargetException;
      invoke-virtual {v1}, Ljava/lang/reflect/InvocationTargetException;->getCause()Ljava/lang/Throwable;
  
      move-result-object v0
  
+     .line 789
      .local v0, cause:Ljava/lang/Throwable;
      instance-of v2, v0, Ljava/lang/RuntimeException;
  
      if-eqz v2, :cond_0
  
+     .line 790
      check-cast v0, Ljava/lang/RuntimeException;
  
      .end local v0           #cause:Ljava/lang/Throwable;
      throw v0
  
+     .line 791
      .restart local v0       #cause:Ljava/lang/Throwable;
      :cond_0
      instance-of v2, v0, Ljava/lang/Error;
  
      if-eqz v2, :cond_1
  
+     .line 792
      check-cast v0, Ljava/lang/Error;
  
      .end local v0           #cause:Ljava/lang/Throwable;
      throw v0
  
+     .line 794
      .restart local v0       #cause:Ljava/lang/Throwable;
      :cond_1
      new-instance v2, Ljava/lang/RuntimeException;
