*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/test/AndroidTestCase.smali	2012-03-23 19:36:07.260813384 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/test/AndroidTestCase.smali.original	2012-03-23 19:34:44.974312892 -0500
***************
*** 14,19 ****
--- 14,20 ----
      .locals 0
  
      .prologue
+     .line 30
      invoke-direct {p0}, Ljunit/framework/TestCase;-><init>()V
  
      return-void
***************
*** 28,44 ****
--- 29,49 ----
      .parameter "permission"
  
      .prologue
+     .line 91
      new-instance v1, Landroid/content/Intent;
  
      invoke-direct {v1}, Landroid/content/Intent;-><init>()V
  
+     .line 92
      .local v1, intent:Landroid/content/Intent;
      invoke-virtual {v1, p1, p2}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
  
+     .line 93
      const/high16 v2, 0x1000
  
      invoke-virtual {v1, v2}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
  
+     .line 96
      :try_start_0
      invoke-virtual {p0}, Landroid/test/AndroidTestCase;->getContext()Landroid/content/Context;
  
***************
*** 46,51 ****
--- 51,57 ----
  
      invoke-virtual {v2, v1}, Landroid/content/Context;->startActivity(Landroid/content/Intent;)V
  
+     .line 97
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 68,81 ****
      :try_end_0
      .catch Ljava/lang/SecurityException; {:try_start_0 .. :try_end_0} :catch_0
  
      :goto_0
      return-void
  
      :catch_0
      move-exception v0
  
      .local v0, expected:Ljava/lang/SecurityException;
!     const-string v2, "security exception\'s error message."
  
      invoke-virtual {v0}, Ljava/lang/SecurityException;->getMessage()Ljava/lang/String;
  
--- 74,90 ----
      :try_end_0
      .catch Ljava/lang/SecurityException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 103
      :goto_0
      return-void
  
+     .line 98
      :catch_0
      move-exception v0
  
+     .line 99
      .local v0, expected:Ljava/lang/SecurityException;
!     const-string/jumbo v2, "security exception\'s error message."
  
      invoke-virtual {v0}, Ljava/lang/SecurityException;->getMessage()Ljava/lang/String;
  
***************
*** 83,88 ****
--- 92,98 ----
  
      invoke-static {v2, v3}, Landroid/test/AndroidTestCase;->assertNotNull(Ljava/lang/String;Ljava/lang/Object;)V
  
+     .line 100
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 126,131 ****
--- 136,142 ----
      .parameter "permission"
  
      .prologue
+     .line 115
      :try_start_0
      invoke-virtual {p0}, Landroid/test/AndroidTestCase;->getContext()Landroid/content/Context;
  
***************
*** 147,152 ****
--- 158,164 ----
  
      invoke-virtual/range {v0 .. v5}, Landroid/content/ContentResolver;->query(Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)Landroid/database/Cursor;
  
+     .line 116
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 169,182 ****
      :try_end_0
      .catch Ljava/lang/SecurityException; {:try_start_0 .. :try_end_0} :catch_0
  
      :goto_0
      return-void
  
      :catch_0
      move-exception v6
  
      .local v6, expected:Ljava/lang/SecurityException;
!     const-string v0, "security exception\'s error message."
  
      invoke-virtual {v6}, Ljava/lang/SecurityException;->getMessage()Ljava/lang/String;
  
--- 181,197 ----
      :try_end_0
      .catch Ljava/lang/SecurityException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 122
      :goto_0
      return-void
  
+     .line 117
      :catch_0
      move-exception v6
  
+     .line 118
      .local v6, expected:Ljava/lang/SecurityException;
!     const-string/jumbo v0, "security exception\'s error message."
  
      invoke-virtual {v6}, Ljava/lang/SecurityException;->getMessage()Ljava/lang/String;
  
***************
*** 184,189 ****
--- 199,205 ----
  
      invoke-static {v0, v1}, Landroid/test/AndroidTestCase;->assertNotNull(Ljava/lang/String;Ljava/lang/Object;)V
  
+     .line 119
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 227,232 ****
--- 243,249 ----
      .parameter "permission"
  
      .prologue
+     .line 134
      :try_start_0
      invoke-virtual {p0}, Landroid/test/AndroidTestCase;->getContext()Landroid/content/Context;
  
***************
*** 242,247 ****
--- 259,265 ----
  
      invoke-virtual {v1, p1, v2}, Landroid/content/ContentResolver;->insert(Landroid/net/Uri;Landroid/content/ContentValues;)Landroid/net/Uri;
  
+     .line 135
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 264,277 ****
      :try_end_0
      .catch Ljava/lang/SecurityException; {:try_start_0 .. :try_end_0} :catch_0
  
      :goto_0
      return-void
  
      :catch_0
      move-exception v0
  
      .local v0, expected:Ljava/lang/SecurityException;
!     const-string v1, "security exception\'s error message."
  
      invoke-virtual {v0}, Ljava/lang/SecurityException;->getMessage()Ljava/lang/String;
  
--- 282,298 ----
      :try_end_0
      .catch Ljava/lang/SecurityException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 141
      :goto_0
      return-void
  
+     .line 136
      :catch_0
      move-exception v0
  
+     .line 137
      .local v0, expected:Ljava/lang/SecurityException;
!     const-string/jumbo v1, "security exception\'s error message."
  
      invoke-virtual {v0}, Ljava/lang/SecurityException;->getMessage()Ljava/lang/String;
  
***************
*** 279,284 ****
--- 300,306 ----
  
      invoke-static {v1, v2}, Landroid/test/AndroidTestCase;->assertNotNull(Ljava/lang/String;Ljava/lang/Object;)V
  
+     .line 138
      new-instance v1, Ljava/lang/StringBuilder;
  
      invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 320,325 ****
--- 342,348 ----
      .locals 1
  
      .prologue
+     .line 55
      iget-object v0, p0, Landroid/test/AndroidTestCase;->mContext:Landroid/content/Context;
  
      return-object v0
***************
*** 329,334 ****
--- 352,358 ----
      .locals 1
  
      .prologue
+     .line 74
      iget-object v0, p0, Landroid/test/AndroidTestCase;->mTestContext:Landroid/content/Context;
  
      return-object v0
***************
*** 352,357 ****
--- 376,382 ----
      .end annotation
  
      .prologue
+     .line 155
      .local p1, testCaseClass:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
      invoke-virtual {p0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
  
***************
*** 361,366 ****
--- 386,392 ----
  
      move-result-object v4
  
+     .line 156
      .local v4, fields:[Ljava/lang/reflect/Field;
      move-object v0, v4
  
***************
*** 376,386 ****
--- 402,414 ----
  
      aget-object v2, v0, v5
  
+     .line 157
      .local v2, field:Ljava/lang/reflect/Field;
      invoke-virtual {v2}, Ljava/lang/reflect/Field;->getDeclaringClass()Ljava/lang/Class;
  
      move-result-object v3
  
+     .line 158
      .local v3, fieldClass:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
      invoke-virtual {p1, v3}, Ljava/lang/Class;->isAssignableFrom(Ljava/lang/Class;)Z
  
***************
*** 398,414 ****
--- 426,445 ----
  
      if-nez v7, :cond_0
  
+     .line 160
      const/4 v7, 0x1
  
      :try_start_0
      invoke-virtual {v2, v7}, Ljava/lang/reflect/Field;->setAccessible(Z)V
  
+     .line 161
      const/4 v7, 0x0
  
      invoke-virtual {v2, p0, v7}, Ljava/lang/reflect/Field;->set(Ljava/lang/Object;Ljava/lang/Object;)V
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 166
      :goto_1
      invoke-virtual {v2, p0}, Ljava/lang/reflect/Field;->get(Ljava/lang/Object;)Ljava/lang/Object;
  
***************
*** 416,435 ****
--- 447,470 ----
  
      if-eqz v7, :cond_0
  
+     .line 167
      const-string v7, "TestCase"
  
      const-string v8, "Error: Could not nullify field!"
  
      invoke-static {v7, v8}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 156
      :cond_0
      add-int/lit8 v5, v5, 0x1
  
      goto :goto_0
  
+     .line 162
      :catch_0
      move-exception v1
  
+     .line 163
      .local v1, e:Ljava/lang/Exception;
      const-string v7, "TestCase"
  
***************
*** 439,444 ****
--- 474,480 ----
  
      goto :goto_1
  
+     .line 171
      .end local v1           #e:Ljava/lang/Exception;
      .end local v2           #field:Ljava/lang/reflect/Field;
      .end local v3           #fieldClass:Ljava/lang/Class;,"Ljava/lang/Class<*>;"
***************
*** 451,458 ****
--- 487,496 ----
      .parameter "context"
  
      .prologue
+     .line 51
      iput-object p1, p0, Landroid/test/AndroidTestCase;->mContext:Landroid/content/Context;
  
+     .line 52
      return-void
  .end method
  
***************
*** 461,468 ****
--- 499,508 ----
      .parameter "context"
  
      .prologue
+     .line 67
      iput-object p1, p0, Landroid/test/AndroidTestCase;->mTestContext:Landroid/content/Context;
  
+     .line 68
      return-void
  .end method
  
***************
*** 475,482 ****
--- 515,524 ----
      .end annotation
  
      .prologue
+     .line 37
      invoke-super {p0}, Ljunit/framework/TestCase;->setUp()V
  
+     .line 38
      return-void
  .end method
  
***************
*** 489,496 ****
--- 531,540 ----
      .end annotation
  
      .prologue
+     .line 42
      invoke-super {p0}, Ljunit/framework/TestCase;->tearDown()V
  
+     .line 43
      return-void
  .end method
  
***************
*** 498,508 ****
--- 542,554 ----
      .locals 2
  
      .prologue
+     .line 46
      const-string v0, "Context is null. setContext should be called before tests are run"
  
      iget-object v1, p0, Landroid/test/AndroidTestCase;->mContext:Landroid/content/Context;
  
      invoke-static {v0, v1}, Landroid/test/AndroidTestCase;->assertNotNull(Ljava/lang/String;Ljava/lang/Object;)V
  
+     .line 48
      return-void
  .end method
