*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/test/InstrumentationTestCase$1.smali	2012-03-23 19:36:07.185812018 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/test/InstrumentationTestCase$1.smali.original	2012-03-23 19:34:44.974312892 -0500
***************
*** 33,38 ****
--- 33,39 ----
      .parameter
  
      .prologue
+     .line 135
      iput-object p1, p0, Landroid/test/InstrumentationTestCase$1;->this$0:Landroid/test/InstrumentationTestCase;
  
      iput-object p2, p0, Landroid/test/InstrumentationTestCase$1;->val$r:Ljava/lang/Runnable;
***************
*** 50,55 ****
--- 51,57 ----
      .locals 3
  
      .prologue
+     .line 138
      :try_start_0
      iget-object v1, p0, Landroid/test/InstrumentationTestCase$1;->val$r:Ljava/lang/Runnable;
  
***************
*** 57,68 ****
--- 59,73 ----
      :try_end_0
      .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 142
      :goto_0
      return-void
  
+     .line 139
      :catch_0
      move-exception v0
  
+     .line 140
      .local v0, throwable:Ljava/lang/Throwable;
      iget-object v1, p0, Landroid/test/InstrumentationTestCase$1;->val$exceptions:[Ljava/lang/Throwable;
  
