*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/test/InstrumentationTestCase$2.smali	2012-03-23 19:36:06.948807696 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/test/InstrumentationTestCase$2.smali.original	2012-03-23 19:34:44.972312856 -0500
***************
*** 39,44 ****
--- 39,45 ----
      .parameter
  
      .prologue
+     .line 186
      iput-object p1, p0, Landroid/test/InstrumentationTestCase$2;->this$0:Landroid/test/InstrumentationTestCase;
  
      iput-object p2, p0, Landroid/test/InstrumentationTestCase$2;->val$testMethod:Ljava/lang/reflect/Method;
***************
*** 60,65 ****
--- 61,67 ----
      .locals 5
  
      .prologue
+     .line 189
      :try_start_0
      iget-object v1, p0, Landroid/test/InstrumentationTestCase$2;->this$0:Landroid/test/InstrumentationTestCase;
  
***************
*** 74,85 ****
--- 76,90 ----
      :try_end_0
      .catch Ljava/lang/Throwable; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 193
      :goto_0
      return-void
  
+     .line 190
      :catch_0
      move-exception v0
  
+     .line 191
      .local v0, throwable:Ljava/lang/Throwable;
      iget-object v1, p0, Landroid/test/InstrumentationTestCase$2;->val$exceptions:[Ljava/lang/Throwable;
  
