*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/gles_jni/EGLContextImpl.smali	2012-03-23 19:37:58.456841041 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/gles_jni/EGLContextImpl.smali.original	2012-03-23 19:34:46.441339644 -0500
***************
*** 15,30 ****
--- 15,34 ----
      .parameter "ctx"
  
      .prologue
+     .line 26
      invoke-direct {p0}, Ljavax/microedition/khronos/egl/EGLContext;-><init>()V
  
+     .line 27
      iput p1, p0, Lcom/google/android/gles_jni/EGLContextImpl;->mEGLContext:I
  
+     .line 28
      new-instance v0, Lcom/google/android/gles_jni/GLImpl;
  
      invoke-direct {v0}, Lcom/google/android/gles_jni/GLImpl;-><init>()V
  
      iput-object v0, p0, Lcom/google/android/gles_jni/EGLContextImpl;->mGLContext:Lcom/google/android/gles_jni/GLImpl;
  
+     .line 29
      return-void
  .end method
  
***************
*** 39,50 ****
--- 43,57 ----
  
      const/4 v2, 0x0
  
+     .line 38
      if-ne p0, p1, :cond_1
  
+     .line 43
      :cond_0
      :goto_0
      return v1
  
+     .line 39
      :cond_1
      if-eqz p1, :cond_2
  
***************
*** 66,73 ****
--- 73,82 ----
      :cond_3
      move-object v0, p1
  
+     .line 41
      check-cast v0, Lcom/google/android/gles_jni/EGLContextImpl;
  
+     .line 43
      .local v0, that:Lcom/google/android/gles_jni/EGLContextImpl;
      iget v3, p0, Lcom/google/android/gles_jni/EGLContextImpl;->mEGLContext:I
  
***************
*** 84,89 ****
--- 93,99 ----
      .locals 1
  
      .prologue
+     .line 33
      iget-object v0, p0, Lcom/google/android/gles_jni/EGLContextImpl;->mGLContext:Lcom/google/android/gles_jni/GLImpl;
  
      return-object v0
***************
*** 93,98 ****
--- 103,109 ----
      .locals 1
  
      .prologue
+     .line 48
      iget v0, p0, Lcom/google/android/gles_jni/EGLContextImpl;->mEGLContext:I
  
      return v0
