*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/util/Linkify$4.smali	2012-03-23 19:35:18.591925909 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/util/Linkify$4.smali.original	2012-03-23 19:34:44.474303775 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 481
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 49,64 ****
--- 50,68 ----
  
      const/4 v0, -0x1
  
+     .line 483
      iget v2, p1, Landroid/text/util/LinkSpec;->start:I
  
      iget v3, p2, Landroid/text/util/LinkSpec;->start:I
  
      if-ge v2, v3, :cond_1
  
+     .line 499
      :cond_0
      :goto_0
      return v0
  
+     .line 487
      :cond_1
      iget v2, p1, Landroid/text/util/LinkSpec;->start:I
  
***************
*** 68,75 ****
--- 72,81 ----
  
      move v0, v1
  
+     .line 488
      goto :goto_0
  
+     .line 491
      :cond_2
      iget v2, p1, Landroid/text/util/LinkSpec;->end:I
  
***************
*** 79,86 ****
--- 85,94 ----
  
      move v0, v1
  
+     .line 492
      goto :goto_0
  
+     .line 495
      :cond_3
      iget v1, p1, Landroid/text/util/LinkSpec;->end:I
  
***************
*** 88,93 ****
--- 96,102 ----
  
      if-gt v1, v2, :cond_0
  
+     .line 499
      const/4 v0, 0x0
  
      goto :goto_0
***************
*** 99,104 ****
--- 108,114 ----
      .parameter "x1"
  
      .prologue
+     .line 481
      check-cast p1, Landroid/text/util/LinkSpec;
  
      .end local p1
***************
*** 117,122 ****
--- 127,133 ----
      .parameter "o"
  
      .prologue
+     .line 503
      const/4 v0, 0x0
  
      return v0
