*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/util/LinkSpec.smali	2012-03-23 19:35:18.670927349 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/text/util/LinkSpec.smali.original	2012-03-23 19:34:44.475303793 -0500
***************
*** 16,21 ****
--- 16,22 ----
      .locals 0
  
      .prologue
+     .line 539
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
