*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/gesture/InstanceLearner$1.smali	2012-03-23 19:35:46.118427855 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/gesture/InstanceLearner$1.smali.original	2012-03-23 19:34:44.829310248 -0500
***************
*** 32,37 ****
--- 32,38 ----
      .locals 0
  
      .prologue
+     .line 29
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 45,74 ****
--- 46,83 ----
      .parameter "object2"
  
      .prologue
+     .line 31
      iget-wide v0, p1, Landroid/gesture/Prediction;->score:D
  
+     .line 32
      .local v0, score1:D
      iget-wide v2, p2, Landroid/gesture/Prediction;->score:D
  
+     .line 33
      .local v2, score2:D
      cmpl-double v4, v0, v2
  
      if-lez v4, :cond_0
  
+     .line 34
      const/4 v4, -0x1
  
+     .line 38
      :goto_0
      return v4
  
+     .line 35
      :cond_0
      cmpg-double v4, v0, v2
  
      if-gez v4, :cond_1
  
+     .line 36
      const/4 v4, 0x1
  
      goto :goto_0
  
+     .line 38
      :cond_1
      const/4 v4, 0x0
  
***************
*** 81,86 ****
--- 90,96 ----
      .parameter "x1"
  
      .prologue
+     .line 29
      check-cast p1, Landroid/gesture/Prediction;
  
      .end local p1
