*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/BlackFrame$BlackSurface.smali	2012-03-23 19:35:06.801710916 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/BlackFrame$BlackSurface.smali.original	2012-03-23 19:34:47.125352119 -0500
***************
*** 41,59 ****
--- 41,65 ----
      .end annotation
  
      .prologue
+     .line 36
      iput-object p1, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->this$0:Lcom/android/server/wm/BlackFrame;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 37
      iput p4, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->left:I
  
+     .line 38
      iput p5, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->top:I
  
+     .line 39
      sub-int v5, p6, p4
  
+     .line 40
      .local v5, w:I
      sub-int v6, p7, p5
  
+     .line 41
      .local v6, h:I
      new-instance v0, Landroid/view/Surface;
  
***************
*** 73,88 ****
--- 79,97 ----
  
      iput-object v0, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->surface:Landroid/view/Surface;
  
+     .line 46
      iget-object v0, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->surface:Landroid/view/Surface;
  
      const/high16 v1, 0x3f80
  
      invoke-virtual {v0, v1}, Landroid/view/Surface;->setAlpha(F)V
  
+     .line 47
      iget-object v0, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->surface:Landroid/view/Surface;
  
      invoke-virtual {v0, p3}, Landroid/view/Surface;->setLayer(I)V
  
+     .line 48
      return-void
  .end method
  
***************
*** 96,105 ****
--- 105,116 ----
  
      const/4 v1, 0x0
  
+     .line 71
      iget-object v0, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->surface:Landroid/view/Surface;
  
      invoke-virtual {v0, v2, v1, v1, v2}, Landroid/view/Surface;->setMatrix(FFFF)V
  
+     .line 72
      return-void
  .end method
  
***************
*** 108,113 ****
--- 119,125 ----
      .parameter "matrix"
  
      .prologue
+     .line 51
      iget-object v0, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->this$0:Lcom/android/server/wm/BlackFrame;
  
      iget-object v0, v0, Lcom/android/server/wm/BlackFrame;->mTmpMatrix:Landroid/graphics/Matrix;
***************
*** 122,133 ****
--- 134,147 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/graphics/Matrix;->setTranslate(FF)V
  
+     .line 52
      iget-object v0, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->this$0:Lcom/android/server/wm/BlackFrame;
  
      iget-object v0, v0, Lcom/android/server/wm/BlackFrame;->mTmpMatrix:Landroid/graphics/Matrix;
  
      invoke-virtual {v0, p1}, Landroid/graphics/Matrix;->postConcat(Landroid/graphics/Matrix;)Z
  
+     .line 53
      iget-object v0, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->this$0:Lcom/android/server/wm/BlackFrame;
  
      iget-object v0, v0, Lcom/android/server/wm/BlackFrame;->mTmpMatrix:Landroid/graphics/Matrix;
***************
*** 138,143 ****
--- 152,158 ----
  
      invoke-virtual {v0, v1}, Landroid/graphics/Matrix;->getValues([F)V
  
+     .line 54
      iget-object v0, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->surface:Landroid/view/Surface;
  
      iget-object v1, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->this$0:Lcom/android/server/wm/BlackFrame;
***************
*** 158,163 ****
--- 173,179 ----
  
      invoke-virtual {v0, v1, v2}, Landroid/view/Surface;->setPosition(FF)V
  
+     .line 56
      iget-object v0, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->surface:Landroid/view/Surface;
  
      iget-object v1, p0, Lcom/android/server/wm/BlackFrame$BlackSurface;->this$0:Lcom/android/server/wm/BlackFrame;
***************
*** 194,198 ****
--- 210,215 ----
  
      invoke-virtual {v0, v1, v2, v3, v4}, Landroid/view/Surface;->setMatrix(FFFF)V
  
+     .line 68
      return-void
  .end method
