*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/DimSurface.smali	2012-03-23 19:35:06.778710495 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/DimSurface.smali.original	2012-03-23 19:34:47.125352119 -0500
***************
*** 27,44 ****
--- 27,50 ----
  
      const/4 v0, -0x1
  
+     .line 33
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 28
      iput-boolean v1, p0, Lcom/android/server/wm/DimSurface;->mDimShown:Z
  
+     .line 29
      iput v1, p0, Lcom/android/server/wm/DimSurface;->mDimColor:I
  
+     .line 30
      iput v0, p0, Lcom/android/server/wm/DimSurface;->mLayer:I
  
+     .line 34
      iget-object v0, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
      if-nez v0, :cond_0
  
+     .line 39
      :try_start_0
      new-instance v0, Landroid/view/Surface;
  
***************
*** 62,67 ****
--- 68,74 ----
  
      iput-object v0, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
+     .line 43
      iget-object v0, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
      const/4 v1, 0x0
***************
*** 70,82 ****
--- 77,92 ----
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 48
      :cond_0
      :goto_0
      return-void
  
+     .line 44
      :catch_0
      move-exception v9
  
+     .line 45
      .local v9, e:Ljava/lang/Exception;
      const-string v0, "WindowManager"
  
***************
*** 93,106 ****
--- 103,119 ----
      .locals 3
  
      .prologue
+     .line 83
      iget-boolean v1, p0, Lcom/android/server/wm/DimSurface;->mDimShown:Z
  
      if-eqz v1, :cond_0
  
+     .line 84
      const/4 v1, 0x0
  
      iput-boolean v1, p0, Lcom/android/server/wm/DimSurface;->mDimShown:Z
  
+     .line 87
      :try_start_0
      iget-object v1, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
***************
*** 108,120 ****
--- 121,136 ----
      :try_end_0
      .catch Ljava/lang/RuntimeException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 92
      :cond_0
      :goto_0
      return-void
  
+     .line 88
      :catch_0
      move-exception v0
  
+     .line 89
      .local v0, e:Ljava/lang/RuntimeException;
      const-string v1, "WindowManager"
  
***************
*** 131,136 ****
--- 147,153 ----
      .parameter "pw"
  
      .prologue
+     .line 95
      invoke-virtual {p2, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mDimSurface="
***************
*** 141,146 ****
--- 158,164 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 96
      invoke-virtual {p2, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mDimShown="
***************
*** 151,156 ****
--- 169,175 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 97
      const-string v0, " mLayer="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 159,164 ****
--- 178,184 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 98
      const-string v0, " mDimColor=0x"
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 171,176 ****
--- 191,197 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/String;)V
  
+     .line 99
      invoke-virtual {p2, p1}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mLastDimWidth="
***************
*** 181,186 ****
--- 202,208 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 100
      const-string v0, " mLastDimWidth="
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 189,194 ****
--- 211,217 ----
  
      invoke-virtual {p2, v0}, Ljava/io/PrintWriter;->println(I)V
  
+     .line 101
      return-void
  .end method
  
***************
*** 200,218 ****
--- 223,246 ----
      .parameter "color"
  
      .prologue
+     .line 54
      iget-boolean v1, p0, Lcom/android/server/wm/DimSurface;->mDimShown:Z
  
      if-nez v1, :cond_1
  
+     .line 57
      const/4 v1, 0x1
  
      iput-boolean v1, p0, Lcom/android/server/wm/DimSurface;->mDimShown:Z
  
+     .line 59
      :try_start_0
      iput p1, p0, Lcom/android/server/wm/DimSurface;->mLastDimWidth:I
  
+     .line 60
      iput p2, p0, Lcom/android/server/wm/DimSurface;->mLastDimHeight:I
  
+     .line 61
      iget-object v1, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
      const/4 v2, 0x0
***************
*** 221,247 ****
--- 249,281 ----
  
      invoke-virtual {v1, v2, v3}, Landroid/view/Surface;->setPosition(II)V
  
+     .line 62
      iget-object v1, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
      invoke-virtual {v1, p1, p2}, Landroid/view/Surface;->setSize(II)V
  
+     .line 63
      iget-object v1, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
      invoke-virtual {v1, p3}, Landroid/view/Surface;->setLayer(I)V
  
+     .line 64
      iget-object v1, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
      invoke-virtual {v1}, Landroid/view/Surface;->show()V
      :try_end_0
      .catch Ljava/lang/RuntimeException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 80
      :cond_0
      :goto_0
      return-void
  
+     .line 65
      :catch_0
      move-exception v0
  
+     .line 66
      .local v0, e:Ljava/lang/RuntimeException;
      const-string v1, "WindowManager"
  
***************
*** 251,256 ****
--- 285,291 ----
  
      goto :goto_0
  
+     .line 68
      .end local v0           #e:Ljava/lang/RuntimeException;
      :cond_1
      iget v1, p0, Lcom/android/server/wm/DimSurface;->mLastDimWidth:I
***************
*** 269,291 ****
--- 304,333 ----
  
      if-eq v1, p3, :cond_0
  
+     .line 72
      :cond_2
      iput p1, p0, Lcom/android/server/wm/DimSurface;->mLastDimWidth:I
  
+     .line 73
      iput p2, p0, Lcom/android/server/wm/DimSurface;->mLastDimHeight:I
  
+     .line 74
      iput p3, p0, Lcom/android/server/wm/DimSurface;->mLayer:I
  
+     .line 75
      iput p4, p0, Lcom/android/server/wm/DimSurface;->mDimColor:I
  
+     .line 76
      iget-object v1, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
      invoke-virtual {v1, p1, p2}, Landroid/view/Surface;->setSize(II)V
  
+     .line 77
      iget-object v1, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
      invoke-virtual {v1, p3}, Landroid/view/Surface;->setLayer(I)V
  
+     .line 78
      iget-object v1, p0, Lcom/android/server/wm/DimSurface;->mDimSurface:Landroid/view/Surface;
  
      shr-int/lit8 v2, p4, 0x18
