*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/InputManager$InputFilterHost.smali	2012-03-23 19:35:05.947695341 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/InputManager$InputFilterHost.smali.original	2012-03-23 19:34:47.117351971 -0500
***************
*** 29,34 ****
--- 29,35 ----
      .parameter
  
      .prologue
+     .line 500
      iput-object p1, p0, Lcom/android/server/wm/InputManager$InputFilterHost;->this$0:Lcom/android/server/wm/InputManager;
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
***************
*** 42,47 ****
--- 43,49 ----
      .parameter "x1"
  
      .prologue
+     .line 500
      invoke-direct {p0, p1}, Lcom/android/server/wm/InputManager$InputFilterHost;-><init>(Lcom/android/server/wm/InputManager;)V
  
      return-void
***************
*** 53,62 ****
--- 55,66 ----
      .locals 1
  
      .prologue
+     .line 504
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Lcom/android/server/wm/InputManager$InputFilterHost;->mDisconnected:Z
  
+     .line 505
      return-void
  .end method
  
***************
*** 66,73 ****
--- 70,79 ----
      .parameter "policyFlags"
  
      .prologue
+     .line 508
      if-nez p1, :cond_0
  
+     .line 509
      new-instance v0, Ljava/lang/IllegalArgumentException;
  
      const-string v1, "event must not be null"
***************
*** 76,81 ****
--- 82,88 ----
  
      throw v0
  
+     .line 512
      :cond_0
      iget-object v0, p0, Lcom/android/server/wm/InputManager$InputFilterHost;->this$0:Lcom/android/server/wm/InputManager;
  
***************
*** 83,93 ****
--- 90,102 ----
  
      monitor-enter v6
  
+     .line 513
      :try_start_0
      iget-boolean v0, p0, Lcom/android/server/wm/InputManager$InputFilterHost;->mDisconnected:Z
  
      if-nez v0, :cond_1
  
+     .line 514
      const/4 v1, 0x0
  
      const/4 v2, 0x0
***************
*** 105,115 ****
--- 114,127 ----
      #calls: Lcom/android/server/wm/InputManager;->nativeInjectInputEvent(Landroid/view/InputEvent;IIIII)I
      invoke-static/range {v0 .. v5}, Lcom/android/server/wm/InputManager;->access$200(Landroid/view/InputEvent;IIIII)I
  
+     .line 517
      :cond_1
      monitor-exit v6
  
+     .line 518
      return-void
  
+     .line 517
      :catchall_0
      move-exception v0
  
