*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/Session.smali	2012-03-23 19:35:05.692690691 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/Session.smali.original	2012-03-23 19:34:47.115351935 -0500
***************
*** 36,74 ****
--- 36,85 ----
      .prologue
      const/4 v5, 0x0
  
+     .line 62
      invoke-direct {p0}, Landroid/view/IWindowSession$Stub;-><init>()V
  
+     .line 58
      iput v5, p0, Lcom/android/server/wm/Session;->mNumWindow:I
  
+     .line 59
      iput-boolean v5, p0, Lcom/android/server/wm/Session;->mClientDead:Z
  
+     .line 63
      iput-object p1, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
+     .line 64
      iput-object p2, p0, Lcom/android/server/wm/Session;->mClient:Lcom/android/internal/view/IInputMethodClient;
  
+     .line 65
      iput-object p3, p0, Lcom/android/server/wm/Session;->mInputContext:Lcom/android/internal/view/IInputContext;
  
+     .line 66
      invoke-static {}, Landroid/os/Binder;->getCallingUid()I
  
      move-result v5
  
      iput v5, p0, Lcom/android/server/wm/Session;->mUid:I
  
+     .line 67
      invoke-static {}, Landroid/os/Binder;->getCallingPid()I
  
      move-result v5
  
      iput v5, p0, Lcom/android/server/wm/Session;->mPid:I
  
+     .line 68
      new-instance v4, Ljava/lang/StringBuilder;
  
      invoke-direct {v4}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 69
      .local v4, sb:Ljava/lang/StringBuilder;
      const-string v5, "Session{"
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 70
      invoke-static {p0}, Ljava/lang/System;->identityHashCode(Ljava/lang/Object;)I
  
      move-result v5
***************
*** 79,108 ****
--- 90,125 ----
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 71
      const-string v5, " uid "
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 72
      iget v5, p0, Lcom/android/server/wm/Session;->mUid:I
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
+     .line 73
      const-string v5, "}"
  
      invoke-virtual {v4, v5}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 74
      invoke-virtual {v4}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v5
  
      iput-object v5, p0, Lcom/android/server/wm/Session;->mStringName:Ljava/lang/String;
  
+     .line 76
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v6, v5, Lcom/android/server/wm/WindowManagerService;->mWindowMap:Ljava/util/HashMap;
  
      monitor-enter v6
  
+     .line 77
      :try_start_0
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 116,127 ****
--- 133,146 ----
  
      if-eqz v5, :cond_0
  
+     .line 78
      const-string v5, "input_method"
  
      invoke-static {v5}, Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder;
  
      move-result-object v0
  
+     .line 80
      .local v0, b:Landroid/os/IBinder;
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 131,146 ****
--- 150,168 ----
  
      iput-object v7, v5, Lcom/android/server/wm/WindowManagerService;->mInputMethodManager:Lcom/android/internal/view/IInputMethodManager;
  
+     .line 82
      .end local v0           #b:Landroid/os/IBinder;
      :cond_0
      monitor-exit v6
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 83
      invoke-static {}, Landroid/os/Binder;->clearCallingIdentity()J
  
      move-result-wide v2
  
+     .line 87
      .local v2, ident:J
      :try_start_1
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 149,154 ****
--- 171,177 ----
  
      if-eqz v5, :cond_1
  
+     .line 88
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v5, v5, Lcom/android/server/wm/WindowManagerService;->mInputMethodManager:Lcom/android/internal/view/IInputMethodManager;
***************
*** 159,164 ****
--- 182,188 ----
  
      invoke-interface {v5, p2, p3, v6, v7}, Lcom/android/internal/view/IInputMethodManager;->addClient(Lcom/android/internal/view/IInputMethodClient;Lcom/android/internal/view/IInputContext;II)V
  
+     .line 93
      :goto_0
      invoke-interface {p2}, Lcom/android/internal/view/IInputMethodClient;->asBinder()Landroid/os/IBinder;
  
***************
*** 171,181 ****
--- 195,208 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_1
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_0
  
+     .line 103
      invoke-static {v2, v3}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
+     .line 105
      :goto_1
      return-void
  
+     .line 82
      .end local v2           #ident:J
      :catchall_0
      move-exception v5
***************
*** 187,192 ****
--- 214,220 ----
  
      throw v5
  
+     .line 91
      .restart local v2       #ident:J
      :cond_1
      const/4 v5, 0x0
***************
*** 199,207 ****
--- 227,237 ----
  
      goto :goto_0
  
+     .line 94
      :catch_0
      move-exception v1
  
+     .line 97
      .local v1, e:Landroid/os/RemoteException;
      :try_start_4
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 210,215 ****
--- 240,246 ----
  
      if-eqz v5, :cond_2
  
+     .line 98
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v5, v5, Lcom/android/server/wm/WindowManagerService;->mInputMethodManager:Lcom/android/internal/view/IInputMethodManager;
***************
*** 219,224 ****
--- 250,256 ----
      .catchall {:try_start_4 .. :try_end_4} :catchall_1
      .catch Landroid/os/RemoteException; {:try_start_4 .. :try_end_4} :catch_1
  
+     .line 103
      :cond_2
      :goto_2
      invoke-static {v2, v3}, Landroid/os/Binder;->restoreCallingIdentity(J)V
***************
*** 233,238 ****
--- 265,271 ----
  
      throw v5
  
+     .line 100
      .restart local v1       #e:Landroid/os/RemoteException;
      :catch_1
      move-exception v5
***************
*** 252,257 ****
--- 285,291 ----
      .parameter "outInputChannel"
  
      .prologue
+     .line 139
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      move-object v1, p0
***************
*** 284,289 ****
--- 318,324 ----
      .parameter "outContentInsets"
  
      .prologue
+     .line 145
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      const/4 v7, 0x0
***************
*** 311,316 ****
--- 346,352 ----
      .locals 3
  
      .prologue
+     .line 125
      :try_start_0
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 318,323 ****
--- 354,360 ----
  
      if-eqz v0, :cond_0
  
+     .line 126
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v0, v0, Lcom/android/server/wm/WindowManagerService;->mInputMethodManager:Lcom/android/internal/view/IInputMethodManager;
***************
*** 328,333 ****
--- 365,371 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 130
      :cond_0
      :goto_0
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 336,341 ****
--- 374,380 ----
  
      monitor-enter v1
  
+     .line 131
      :try_start_1
      iget-object v0, p0, Lcom/android/server/wm/Session;->mClient:Lcom/android/internal/view/IInputMethodClient;
  
***************
*** 347,362 ****
--- 386,406 ----
  
      invoke-interface {v0, p0, v2}, Landroid/os/IBinder;->unlinkToDeath(Landroid/os/IBinder$DeathRecipient;I)Z
  
+     .line 132
      const/4 v0, 0x1
  
      iput-boolean v0, p0, Lcom/android/server/wm/Session;->mClientDead:Z
  
+     .line 133
      invoke-virtual {p0}, Lcom/android/server/wm/Session;->killSessionLocked()V
  
+     .line 134
      monitor-exit v1
  
+     .line 135
      return-void
  
+     .line 134
      :catchall_0
      move-exception v0
  
***************
*** 366,371 ****
--- 410,416 ----
  
      throw v0
  
+     .line 128
      :catch_0
      move-exception v0
  
***************
*** 377,382 ****
--- 422,428 ----
      .parameter "window"
  
      .prologue
+     .line 348
      return-void
  .end method
  
***************
*** 385,390 ****
--- 431,437 ----
      .parameter "window"
  
      .prologue
+     .line 354
      return-void
  .end method
  
***************
*** 394,399 ****
--- 441,447 ----
      .parameter "prefix"
  
      .prologue
+     .line 430
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "mNumWindow="
***************
*** 404,409 ****
--- 452,458 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 431
      const-string v0, " mClientDead="
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 412,417 ****
--- 461,467 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 432
      const-string v0, " mSurfaceSession="
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 420,425 ****
--- 470,476 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 433
      return-void
  .end method
  
***************
*** 428,437 ****
--- 479,490 ----
      .parameter "window"
  
      .prologue
+     .line 191
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      invoke-virtual {v0, p0, p1}, Lcom/android/server/wm/WindowManagerService;->finishDrawingWindow(Lcom/android/server/wm/Session;Landroid/view/IWindow;)V
  
+     .line 192
      return-void
  .end method
  
***************
*** 441,450 ****
--- 494,505 ----
      .parameter "outDisplayFrame"
  
      .prologue
+     .line 185
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      invoke-virtual {v0, p0, p1, p2}, Lcom/android/server/wm/WindowManagerService;->getWindowDisplayFrame(Lcom/android/server/wm/Session;Landroid/view/IWindow;Landroid/graphics/Rect;)V
  
+     .line 186
      return-void
  .end method
  
***************
*** 452,463 ****
--- 507,520 ----
      .locals 2
  
      .prologue
+     .line 201
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v1, v0, Lcom/android/server/wm/WindowManagerService;->mWindowMap:Ljava/util/HashMap;
  
      monitor-enter v1
  
+     .line 202
      :try_start_0
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 467,472 ****
--- 524,530 ----
  
      return v0
  
+     .line 203
      :catchall_0
      move-exception v0
  
***************
*** 481,486 ****
--- 539,545 ----
      .locals 4
  
      .prologue
+     .line 409
      iget v1, p0, Lcom/android/server/wm/Session;->mNumWindow:I
  
      if-gtz v1, :cond_0
***************
*** 489,504 ****
--- 548,566 ----
  
      if-eqz v1, :cond_0
  
+     .line 410
      iget-object v1, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v1, v1, Lcom/android/server/wm/WindowManagerService;->mSessions:Ljava/util/HashSet;
  
      invoke-virtual {v1, p0}, Ljava/util/HashSet;->remove(Ljava/lang/Object;)Z
  
+     .line 411
      iget-object v1, p0, Lcom/android/server/wm/Session;->mSurfaceSession:Landroid/view/SurfaceSession;
  
      if-eqz v1, :cond_0
  
+     .line 418
      :try_start_0
      iget-object v1, p0, Lcom/android/server/wm/Session;->mSurfaceSession:Landroid/view/SurfaceSession;
  
***************
*** 506,522 ****
--- 568,588 ----
      :try_end_0
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 424
      :goto_0
      const/4 v1, 0x0
  
      iput-object v1, p0, Lcom/android/server/wm/Session;->mSurfaceSession:Landroid/view/SurfaceSession;
  
+     .line 427
      :cond_0
      return-void
  
+     .line 419
      :catch_0
      move-exception v0
  
+     .line 420
      .local v0, e:Ljava/lang/Exception;
      const-string v1, "WindowManager"
  
***************
*** 582,587 ****
--- 648,654 ----
      .end annotation
  
      .prologue
+     .line 111
      :try_start_0
      invoke-super {p0, p1, p2, p3, p4}, Landroid/view/IWindowSession$Stub;->onTransact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
      :try_end_0
***************
*** 591,610 ****
--- 658,681 ----
  
      return v1
  
+     .line 112
      :catch_0
      move-exception v0
  
+     .line 114
      .local v0, e:Ljava/lang/RuntimeException;
      instance-of v1, v0, Ljava/lang/SecurityException;
  
      if-nez v1, :cond_0
  
+     .line 115
      const-string v1, "WindowManager"
  
      const-string v2, "Window Session Crash"
  
      invoke-static {v1, v2, v0}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
  
+     .line 117
      :cond_0
      throw v0
  .end method
***************
*** 614,619 ****
--- 685,691 ----
      .parameter "window"
  
      .prologue
+     .line 171
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      invoke-virtual {v0, p0, p1}, Lcom/android/server/wm/WindowManagerService;->outOfMemoryWindow(Lcom/android/server/wm/Session;Landroid/view/IWindow;)Z
***************
*** 628,637 ****
--- 700,711 ----
      .parameter "window"
  
      .prologue
+     .line 167
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      invoke-virtual {v0, p0, p1}, Lcom/android/server/wm/WindowManagerService;->performDeferredDestroyWindow(Lcom/android/server/wm/Session;Landroid/view/IWindow;)V
  
+     .line 168
      return-void
  .end method
  
***************
*** 650,661 ****
--- 724,737 ----
  
      const/4 v2, 0x0
  
+     .line 234
      iget-object v4, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v4, v4, Lcom/android/server/wm/WindowManagerService;->mWindowMap:Ljava/util/HashMap;
  
      monitor-enter v4
  
+     .line 235
      :try_start_0
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 663,674 ****
--- 739,752 ----
  
      if-nez v5, :cond_0
  
+     .line 236
      const-string v2, "WindowManager"
  
      const-string v3, "No drag prepared"
  
      invoke-static {v2, v3}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 237
      new-instance v2, Ljava/lang/IllegalStateException;
  
      const-string v3, "performDrag() without prepareDrag()"
***************
*** 677,682 ****
--- 755,761 ----
  
      throw v2
  
+     .line 299
      :catchall_0
      move-exception v2
  
***************
*** 686,691 ****
--- 765,771 ----
  
      throw v2
  
+     .line 240
      :cond_0
      :try_start_1
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 696,707 ****
--- 776,789 ----
  
      if-eq p2, v5, :cond_1
  
+     .line 241
      const-string v2, "WindowManager"
  
      const-string v3, "Performing mismatched drag"
  
      invoke-static {v2, v3}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 242
      new-instance v2, Ljava/lang/IllegalStateException;
  
      const-string v3, "performDrag() does not match prepareDrag()"
***************
*** 710,715 ****
--- 792,798 ----
  
      throw v2
  
+     .line 245
      :cond_1
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 721,729 ****
--- 804,814 ----
  
      move-result-object v0
  
+     .line 246
      .local v0, callingWin:Lcom/android/server/wm/WindowState;
      if-nez v0, :cond_2
  
+     .line 247
      const-string v3, "WindowManager"
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 746,756 ****
--- 831,844 ----
  
      invoke-static {v3, v5}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 248
      monitor-exit v4
  
+     .line 301
      :goto_0
      return v2
  
+     .line 255
      :cond_2
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 764,775 ****
--- 852,865 ----
  
      invoke-virtual {v5, v6, v7}, Lcom/android/server/wm/WindowManagerService$H;->removeMessages(ILjava/lang/Object;)V
  
+     .line 263
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v5, v5, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
      invoke-virtual {v5}, Lcom/android/server/wm/DragState;->register()V
  
+     .line 264
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v5, v5, Lcom/android/server/wm/WindowManagerService;->mInputMonitor:Lcom/android/server/wm/InputMonitor;
***************
*** 778,783 ****
--- 868,874 ----
  
      invoke-virtual {v5, v6}, Lcom/android/server/wm/InputMonitor;->updateInputWindowsLw(Z)V
  
+     .line 265
      iget-object v5, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v5, v5, Lcom/android/server/wm/WindowManagerService;->mInputManager:Lcom/android/server/wm/InputManager;
***************
*** 796,819 ****
--- 887,914 ----
  
      if-nez v5, :cond_3
  
+     .line 267
      const-string v3, "WindowManager"
  
      const-string v5, "Unable to transfer touch focus"
  
      invoke-static {v3, v5}, Landroid/util/Slog;->e(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 268
      iget-object v3, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v3, v3, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
      invoke-virtual {v3}, Lcom/android/server/wm/DragState;->unregister()V
  
+     .line 269
      iget-object v3, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      const/4 v5, 0x0
  
      iput-object v5, v3, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
+     .line 270
      iget-object v3, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v3, v3, Lcom/android/server/wm/WindowManagerService;->mInputMonitor:Lcom/android/server/wm/InputMonitor;
***************
*** 822,831 ****
--- 917,928 ----
  
      invoke-virtual {v3, v5}, Lcom/android/server/wm/InputMonitor;->updateInputWindowsLw(Z)V
  
+     .line 271
      monitor-exit v4
  
      goto :goto_0
  
+     .line 274
      :cond_3
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 833,879 ****
--- 930,984 ----
  
      iput-object p7, v2, Lcom/android/server/wm/DragState;->mData:Landroid/content/ClipData;
  
+     .line 275
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v2, v2, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
      iput p3, v2, Lcom/android/server/wm/DragState;->mCurrentX:F
  
+     .line 276
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v2, v2, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
      iput p4, v2, Lcom/android/server/wm/DragState;->mCurrentY:F
  
+     .line 277
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v2, v2, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
      invoke-virtual {v2, p3, p4}, Lcom/android/server/wm/DragState;->broadcastDragStartedLw(FF)V
  
+     .line 280
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v2, v2, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
      iput p5, v2, Lcom/android/server/wm/DragState;->mThumbOffsetX:F
  
+     .line 281
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v2, v2, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
      iput p6, v2, Lcom/android/server/wm/DragState;->mThumbOffsetY:F
  
+     .line 284
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v2, v2, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
      iget-object v1, v2, Lcom/android/server/wm/DragState;->mSurface:Landroid/view/Surface;
  
+     .line 287
      .local v1, surface:Landroid/view/Surface;
      invoke-static {}, Landroid/view/Surface;->openTransaction()V
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 289
      sub-float v2, p3, p5
  
      sub-float v5, p4, p6
***************
*** 881,890 ****
--- 986,997 ----
      :try_start_2
      invoke-virtual {v1, v2, v5}, Landroid/view/Surface;->setPosition(FF)V
  
+     .line 291
      const v2, 0x3f350481
  
      invoke-virtual {v1, v2}, Landroid/view/Surface;->setAlpha(F)V
  
+     .line 292
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v2, v2, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
***************
*** 895,913 ****
--- 1002,1025 ----
  
      invoke-virtual {v1, v2}, Landroid/view/Surface;->setLayer(I)V
  
+     .line 293
      invoke-virtual {v1}, Landroid/view/Surface;->show()V
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_1
  
+     .line 295
      :try_start_3
      invoke-static {}, Landroid/view/Surface;->closeTransaction()V
  
+     .line 299
      monitor-exit v4
  
      move v2, v3
  
+     .line 301
      goto/16 :goto_0
  
+     .line 295
      :catchall_1
      move-exception v2
  
***************
*** 925,936 ****
--- 1037,1050 ----
      .parameter "always"
  
      .prologue
+     .line 208
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v3, v2, Lcom/android/server/wm/WindowManagerService;->mWindowMap:Ljava/util/HashMap;
  
      monitor-enter v3
  
+     .line 209
      :try_start_0
      invoke-static {}, Landroid/os/Binder;->clearCallingIdentity()J
      :try_end_0
***************
*** 938,943 ****
--- 1052,1058 ----
  
      move-result-wide v0
  
+     .line 211
      .local v0, ident:J
      :try_start_1
      iget-object v2, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 958,963 ****
--- 1073,1079 ----
  
      move-result v2
  
+     .line 215
      :try_start_2
      invoke-static {v0, v1}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
***************
*** 972,977 ****
--- 1088,1094 ----
  
      throw v2
  
+     .line 217
      .end local v0           #ident:J
      :catchall_1
      move-exception v2
***************
*** 992,997 ****
--- 1109,1115 ----
      .parameter "outSurface"
  
      .prologue
+     .line 223
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v2, p0, Lcom/android/server/wm/Session;->mSurfaceSession:Landroid/view/SurfaceSession;
***************
*** 1029,1034 ****
--- 1147,1153 ----
      .parameter "outSurface"
  
      .prologue
+     .line 158
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      move-object v1, p0
***************
*** 1061,1066 ****
--- 1180,1186 ----
  
      move-result v14
  
+     .line 163
      .local v14, res:I
      return v14
  .end method
***************
*** 1070,1079 ****
--- 1190,1201 ----
      .parameter "window"
  
      .prologue
+     .line 149
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      invoke-virtual {v0, p0, p1}, Lcom/android/server/wm/WindowManagerService;->removeWindow(Lcom/android/server/wm/Session;Landroid/view/IWindow;)V
  
+     .line 150
      return-void
  .end method
  
***************
*** 1083,1092 ****
--- 1205,1216 ----
      .parameter "consumed"
  
      .prologue
+     .line 305
      invoke-interface {p1}, Landroid/view/IWindow;->asBinder()Landroid/os/IBinder;
  
      move-result-object v3
  
+     .line 310
      .local v3, token:Landroid/os/IBinder;
      iget-object v4, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 1094,1099 ****
--- 1218,1224 ----
  
      monitor-enter v5
  
+     .line 311
      :try_start_0
      invoke-static {}, Landroid/os/Binder;->clearCallingIdentity()J
      :try_end_0
***************
*** 1101,1106 ****
--- 1226,1232 ----
  
      move-result-wide v1
  
+     .line 313
      .local v1, ident:J
      :try_start_1
      iget-object v4, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 1109,1114 ****
--- 1235,1241 ----
  
      if-nez v4, :cond_0
  
+     .line 316
      const-string v4, "WindowManager"
  
      const-string v6, "Drop result given but no drag in progress"
***************
*** 1117,1122 ****
--- 1244,1250 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 339
      :try_start_2
      invoke-static {v1, v2}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
***************
*** 1124,1132 ****
--- 1252,1262 ----
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_1
  
+     .line 342
      :goto_0
      return-void
  
+     .line 320
      :cond_0
      :try_start_3
      iget-object v4, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 1137,1142 ****
--- 1267,1273 ----
  
      if-eq v4, v3, :cond_1
  
+     .line 322
      const-string v4, "WindowManager"
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 1159,1164 ****
--- 1290,1296 ----
  
      invoke-static {v4, v6}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 323
      new-instance v4, Ljava/lang/IllegalStateException;
  
      const-string v6, "reportDropResult() by non-recipient"
***************
*** 1169,1174 ****
--- 1301,1307 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
+     .line 339
      :catchall_0
      move-exception v4
  
***************
*** 1177,1182 ****
--- 1310,1316 ----
  
      throw v4
  
+     .line 341
      .end local v1           #ident:J
      :catchall_1
      move-exception v4
***************
*** 1187,1192 ****
--- 1321,1327 ----
  
      throw v4
  
+     .line 329
      .restart local v1       #ident:J
      :cond_1
      :try_start_5
***************
*** 1202,1207 ****
--- 1337,1343 ----
  
      invoke-virtual {v4, v6, v7}, Lcom/android/server/wm/WindowManagerService$H;->removeMessages(ILjava/lang/Object;)V
  
+     .line 330
      iget-object v4, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      const/4 v6, 0x0
***************
*** 1212,1220 ****
--- 1348,1358 ----
  
      move-result-object v0
  
+     .line 331
      .local v0, callingWin:Lcom/android/server/wm/WindowState;
      if-nez v0, :cond_2
  
+     .line 332
      const-string v4, "WindowManager"
  
      new-instance v6, Ljava/lang/StringBuilder;
***************
*** 1239,1244 ****
--- 1377,1383 ----
      :try_end_5
      .catchall {:try_start_5 .. :try_end_5} :catchall_0
  
+     .line 339
      :try_start_6
      invoke-static {v1, v2}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
***************
*** 1248,1253 ****
--- 1387,1393 ----
  
      goto :goto_0
  
+     .line 336
      :cond_2
      :try_start_7
      iget-object v4, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 1256,1261 ****
--- 1396,1402 ----
  
      iput-boolean p2, v4, Lcom/android/server/wm/DragState;->mDragResult:Z
  
+     .line 337
      iget-object v4, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v4, v4, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
***************
*** 1264,1272 ****
--- 1405,1415 ----
      :try_end_7
      .catchall {:try_start_7 .. :try_end_7} :catchall_0
  
+     .line 339
      :try_start_8
      invoke-static {v1, v2}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
+     .line 341
      monitor-exit v5
      :try_end_8
      .catchall {:try_start_8 .. :try_end_8} :catchall_1
***************
*** 1285,1296 ****
--- 1428,1441 ----
      .parameter "sync"
  
      .prologue
+     .line 375
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v10, v0, Lcom/android/server/wm/WindowManagerService;->mWindowMap:Ljava/util/HashMap;
  
      monitor-enter v10
  
+     .line 376
      :try_start_0
      invoke-static {}, Landroid/os/Binder;->clearCallingIdentity()J
      :try_end_0
***************
*** 1298,1303 ****
--- 1443,1449 ----
  
      move-result-wide v8
  
+     .line 378
      .local v8, ident:J
      :try_start_1
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 1328,1333 ****
--- 1474,1480 ----
  
      move-result-object v0
  
+     .line 382
      :try_start_2
      invoke-static {v8, v9}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
***************
*** 1342,1347 ****
--- 1489,1495 ----
  
      throw v0
  
+     .line 384
      .end local v8           #ident:J
      :catchall_1
      move-exception v0
***************
*** 1358,1378 ****
--- 1506,1531 ----
      .parameter "mode"
  
      .prologue
+     .line 195
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v1, v0, Lcom/android/server/wm/WindowManagerService;->mWindowMap:Ljava/util/HashMap;
  
      monitor-enter v1
  
+     .line 196
      :try_start_0
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iput-boolean p1, v0, Lcom/android/server/wm/WindowManagerService;->mInTouchMode:Z
  
+     .line 197
      monitor-exit v1
  
+     .line 198
      return-void
  
+     .line 197
      :catchall_0
      move-exception v0
  
***************
*** 1392,1397 ****
--- 1545,1551 ----
      .parameter "touchableArea"
  
      .prologue
+     .line 180
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      move-object v1, p0
***************
*** 1408,1413 ****
--- 1562,1568 ----
  
      invoke-virtual/range {v0 .. v6}, Lcom/android/server/wm/WindowManagerService;->setInsetsWindow(Lcom/android/server/wm/Session;Landroid/view/IWindow;ILandroid/graphics/Rect;Landroid/graphics/Rect;Landroid/graphics/Region;)V
  
+     .line 182
      return-void
  .end method
  
***************
*** 1417,1426 ****
--- 1572,1583 ----
      .parameter "region"
  
      .prologue
+     .line 175
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      invoke-virtual {v0, p0, p1, p2}, Lcom/android/server/wm/WindowManagerService;->setTransparentRegionWindow(Lcom/android/server/wm/Session;Landroid/view/IWindow;Landroid/graphics/Region;)V
  
+     .line 176
      return-void
  .end method
  
***************
*** 1433,1444 ****
--- 1590,1603 ----
      .parameter "yStep"
  
      .prologue
+     .line 357
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v8, v0, Lcom/android/server/wm/WindowManagerService;->mWindowMap:Ljava/util/HashMap;
  
      monitor-enter v8
  
+     .line 358
      :try_start_0
      invoke-static {}, Landroid/os/Binder;->clearCallingIdentity()J
      :try_end_0
***************
*** 1446,1451 ****
--- 1605,1611 ----
  
      move-result-wide v6
  
+     .line 360
      .local v6, ident:J
      :try_start_1
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
***************
*** 1470,1482 ****
--- 1630,1646 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 364
      :try_start_2
      invoke-static {v6, v7}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
+     .line 366
      monitor-exit v8
  
+     .line 367
      return-void
  
+     .line 364
      :catchall_0
      move-exception v0
  
***************
*** 1484,1489 ****
--- 1648,1654 ----
  
      throw v0
  
+     .line 366
      .end local v6           #ident:J
      :catchall_1
      move-exception v0
***************
*** 1499,1504 ****
--- 1664,1670 ----
      .locals 1
  
      .prologue
+     .line 437
      iget-object v0, p0, Lcom/android/server/wm/Session;->mStringName:Ljava/lang/String;
  
      return-object v0
***************
*** 1510,1519 ****
--- 1676,1687 ----
      .parameter "result"
  
      .prologue
+     .line 388
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      invoke-virtual {v0, p1, p2}, Lcom/android/server/wm/WindowManagerService;->wallpaperCommandComplete(Landroid/os/IBinder;Landroid/os/Bundle;)V
  
+     .line 389
      return-void
  .end method
  
***************
*** 1522,1531 ****
--- 1690,1701 ----
      .parameter "window"
  
      .prologue
+     .line 370
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      invoke-virtual {v0, p1}, Lcom/android/server/wm/WindowManagerService;->wallpaperOffsetsComplete(Landroid/os/IBinder;)V
  
+     .line 371
      return-void
  .end method
  
***************
*** 1533,1554 ****
--- 1703,1728 ----
      .locals 1
  
      .prologue
+     .line 392
      iget-object v0, p0, Lcom/android/server/wm/Session;->mSurfaceSession:Landroid/view/SurfaceSession;
  
      if-nez v0, :cond_0
  
+     .line 395
      new-instance v0, Landroid/view/SurfaceSession;
  
      invoke-direct {v0}, Landroid/view/SurfaceSession;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/wm/Session;->mSurfaceSession:Landroid/view/SurfaceSession;
  
+     .line 398
      iget-object v0, p0, Lcom/android/server/wm/Session;->mService:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v0, v0, Lcom/android/server/wm/WindowManagerService;->mSessions:Ljava/util/HashSet;
  
      invoke-virtual {v0, p0}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 400
      :cond_0
      iget v0, p0, Lcom/android/server/wm/Session;->mNumWindow:I
  
***************
*** 1556,1561 ****
--- 1730,1736 ----
  
      iput v0, p0, Lcom/android/server/wm/Session;->mNumWindow:I
  
+     .line 401
      return-void
  .end method
  
***************
*** 1563,1575 ****
--- 1738,1753 ----
      .locals 1
  
      .prologue
+     .line 404
      iget v0, p0, Lcom/android/server/wm/Session;->mNumWindow:I
  
      add-int/lit8 v0, v0, -0x1
  
      iput v0, p0, Lcom/android/server/wm/Session;->mNumWindow:I
  
+     .line 405
      invoke-virtual {p0}, Lcom/android/server/wm/Session;->killSessionLocked()V
  
+     .line 406
      return-void
  .end method
