*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/WindowManagerService$3.smali	2012-03-23 19:35:05.871693956 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/WindowManagerService$3.smali.original	2012-03-23 19:34:47.117351971 -0500
***************
*** 24,29 ****
--- 24,30 ----
      .parameter
  
      .prologue
+     .line 578
      iput-object p1, p0, Lcom/android/server/wm/WindowManagerService$3;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      invoke-direct {p0}, Lcom/android/internal/view/BaseInputHandler;-><init>()V
***************
*** 39,46 ****
--- 40,49 ----
      .parameter "finishedCallback"
  
      .prologue
+     .line 581
      const/4 v2, 0x0
  
+     .line 583
      .local v2, handled:Z
      :try_start_0
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getSource()I
***************
*** 57,74 ****
--- 60,81 ----
  
      if-eqz v5, :cond_1
  
+     .line 585
      const/4 v1, 0x0
  
+     .line 586
      .local v1, endDrag:Z
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getRawX()F
  
      move-result v3
  
+     .line 587
      .local v3, newX:F
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getRawY()F
  
      move-result v4
  
+     .line 589
      .local v4, newY:F
      invoke-virtual {p1}, Landroid/view/MotionEvent;->getAction()I
  
***************
*** 76,85 ****
--- 83,94 ----
  
      packed-switch v5, :pswitch_data_0
  
+     .line 617
      :goto_0
      :pswitch_0
      if-eqz v1, :cond_0
  
+     .line 620
      iget-object v5, p0, Lcom/android/server/wm/WindowManagerService$3;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      iget-object v6, v5, Lcom/android/server/wm/WindowManagerService;->mWindowMap:Ljava/util/HashMap;
***************
*** 89,94 ****
--- 98,104 ----
      .catchall {:try_start_0 .. :try_end_0} :catchall_2
      .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 621
      :try_start_1
      iget-object v5, p0, Lcom/android/server/wm/WindowManagerService$3;->this$0:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 96,117 ****
--- 106,132 ----
  
      invoke-virtual {v5}, Lcom/android/server/wm/DragState;->endDragLw()V
  
+     .line 622
      monitor-exit v6
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_3
  
+     .line 625
      :cond_0
      const/4 v2, 0x1
  
+     .line 630
      .end local v1           #endDrag:Z
      .end local v3           #newX:F
      .end local v4           #newY:F
      :cond_1
      invoke-virtual {p2, v2}, Landroid/view/InputQueue$FinishedCallback;->finished(Z)V
  
+     .line 632
      :goto_1
      return-void
  
+     .line 597
      .restart local v1       #endDrag:Z
      .restart local v3       #newX:F
      .restart local v4       #newY:F
***************
*** 126,131 ****
--- 141,147 ----
      .catchall {:try_start_2 .. :try_end_2} :catchall_2
      .catch Ljava/lang/Exception; {:try_start_2 .. :try_end_2} :catch_0
  
+     .line 599
      :try_start_3
      iget-object v5, p0, Lcom/android/server/wm/WindowManagerService$3;->this$0:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 133,138 ****
--- 149,155 ----
  
      invoke-virtual {v5, v3, v4}, Lcom/android/server/wm/DragState;->notifyMoveLw(FF)V
  
+     .line 600
      monitor-exit v6
  
      goto :goto_0
***************
*** 150,161 ****
--- 167,180 ----
      .catchall {:try_start_4 .. :try_end_4} :catchall_2
      .catch Ljava/lang/Exception; {:try_start_4 .. :try_end_4} :catch_0
  
+     .line 627
      .end local v1           #endDrag:Z
      .end local v3           #newX:F
      .end local v4           #newY:F
      :catch_0
      move-exception v0
  
+     .line 628
      .local v0, e:Ljava/lang/Exception;
      :try_start_5
      const-string v5, "WindowManager"
***************
*** 166,175 ****
--- 185,196 ----
      :try_end_5
      .catchall {:try_start_5 .. :try_end_5} :catchall_2
  
+     .line 630
      invoke-virtual {p2, v2}, Landroid/view/InputQueue$FinishedCallback;->finished(Z)V
  
      goto :goto_1
  
+     .line 606
      .end local v0           #e:Ljava/lang/Exception;
      .restart local v1       #endDrag:Z
      .restart local v3       #newX:F
***************
*** 185,190 ****
--- 206,212 ----
      .catchall {:try_start_6 .. :try_end_6} :catchall_2
      .catch Ljava/lang/Exception; {:try_start_6 .. :try_end_6} :catch_0
  
+     .line 607
      :try_start_7
      iget-object v5, p0, Lcom/android/server/wm/WindowManagerService$3;->this$0:Lcom/android/server/wm/WindowManagerService;
  
***************
*** 194,199 ****
--- 216,222 ----
  
      move-result v1
  
+     .line 608
      monitor-exit v6
  
      goto :goto_0
***************
*** 211,216 ****
--- 234,240 ----
      .catchall {:try_start_8 .. :try_end_8} :catchall_2
      .catch Ljava/lang/Exception; {:try_start_8 .. :try_end_8} :catch_0
  
+     .line 630
      .end local v1           #endDrag:Z
      .end local v3           #newX:F
      .end local v4           #newY:F
***************
*** 221,226 ****
--- 245,251 ----
  
      throw v5
  
+     .line 613
      .restart local v1       #endDrag:Z
      .restart local v3       #newX:F
      .restart local v4       #newY:F
***************
*** 229,234 ****
--- 254,260 ----
  
      goto :goto_0
  
+     .line 622
      :catchall_3
      move-exception v5
  
***************
*** 243,248 ****
--- 269,275 ----
      .catchall {:try_start_a .. :try_end_a} :catchall_2
      .catch Ljava/lang/Exception; {:try_start_a .. :try_end_a} :catch_0
  
+     .line 589
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_0
