*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/WindowToken.smali	2012-03-23 19:35:05.764692006 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/WindowToken.smali.original	2012-03-23 19:34:47.116351953 -0500
***************
*** 51,76 ****
--- 51,84 ----
      .parameter "_explicit"
  
      .prologue
+     .line 78
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 51
      new-instance v0, Ljava/util/ArrayList;
  
      invoke-direct {v0}, Ljava/util/ArrayList;-><init>()V
  
      iput-object v0, p0, Lcom/android/server/wm/WindowToken;->windows:Ljava/util/ArrayList;
  
+     .line 54
      const/4 v0, 0x0
  
      iput-boolean v0, p0, Lcom/android/server/wm/WindowToken;->paused:Z
  
+     .line 79
      iput-object p1, p0, Lcom/android/server/wm/WindowToken;->service:Lcom/android/server/wm/WindowManagerService;
  
+     .line 80
      iput-object p2, p0, Lcom/android/server/wm/WindowToken;->token:Landroid/os/IBinder;
  
+     .line 81
      iput p3, p0, Lcom/android/server/wm/WindowToken;->windowType:I
  
+     .line 82
      iput-boolean p4, p0, Lcom/android/server/wm/WindowToken;->explicit:Z
  
+     .line 83
      return-void
  .end method
  
***************
*** 82,87 ****
--- 90,96 ----
      .parameter "prefix"
  
      .prologue
+     .line 86
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "token="
***************
*** 92,97 ****
--- 101,107 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 87
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "windows="
***************
*** 102,107 ****
--- 112,118 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Ljava/lang/Object;)V
  
+     .line 88
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
      const-string v0, "windowType="
***************
*** 112,117 ****
--- 123,129 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(I)V
  
+     .line 89
      const-string v0, " hidden="
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 120,125 ****
--- 132,138 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 90
      const-string v0, " hasVisible="
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 128,133 ****
--- 141,147 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 91
      iget-boolean v0, p0, Lcom/android/server/wm/WindowToken;->waitingToShow:Z
  
      if-nez v0, :cond_0
***************
*** 144,149 ****
--- 158,164 ----
  
      if-eqz v0, :cond_1
  
+     .line 92
      :cond_0
      invoke-virtual {p1, p2}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
  
***************
*** 155,160 ****
--- 170,176 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 93
      const-string v0, " waitingToHide="
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 163,168 ****
--- 179,185 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 94
      const-string v0, " sendingToBottom="
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 171,176 ****
--- 188,194 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Z)V
  
+     .line 95
      const-string v0, " sendingToTop="
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->print(Ljava/lang/String;)V
***************
*** 179,184 ****
--- 197,203 ----
  
      invoke-virtual {p1, v0}, Ljava/io/PrintWriter;->println(Z)V
  
+     .line 97
      :cond_1
      return-void
  .end method
***************
*** 187,205 ****
--- 206,228 ----
      .locals 2
  
      .prologue
+     .line 101
      iget-object v1, p0, Lcom/android/server/wm/WindowToken;->stringName:Ljava/lang/String;
  
      if-nez v1, :cond_0
  
+     .line 102
      new-instance v0, Ljava/lang/StringBuilder;
  
      invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
  
+     .line 103
      .local v0, sb:Ljava/lang/StringBuilder;
      const-string v1, "WindowToken{"
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 104
      invoke-static {p0}, Ljava/lang/System;->identityHashCode(Ljava/lang/Object;)I
  
      move-result v1
***************
*** 210,215 ****
--- 233,239 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
  
+     .line 105
      const-string v1, " token="
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
***************
*** 222,233 ****
--- 246,259 ----
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
  
+     .line 106
      invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
  
      move-result-object v1
  
      iput-object v1, p0, Lcom/android/server/wm/WindowToken;->stringName:Ljava/lang/String;
  
+     .line 108
      .end local v0           #sb:Ljava/lang/StringBuilder;
      :cond_0
      iget-object v1, p0, Lcom/android/server/wm/WindowToken;->stringName:Ljava/lang/String;
