*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/WindowManagerService$H.smali	2012-03-23 19:35:05.634689633 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/WindowManagerService$H.smali.original	2012-03-23 19:34:47.114351917 -0500
***************
*** 72,81 ****
--- 72,83 ----
      .parameter
  
      .prologue
+     .line 6527
      iput-object p1, p0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      invoke-direct {p0}, Landroid/os/Handler;-><init>()V
  
+     .line 6528
      return-void
  .end method
  
***************
*** 86,102 ****
--- 88,107 ----
      .parameter "msg"
  
      .prologue
+     .line 6532
      move-object/from16 v0, p1
  
      iget v2, v0, Landroid/os/Message;->what:I
  
      packed-switch v2, :pswitch_data_0
  
+     .line 6933
      :cond_0
      :goto_0
      :pswitch_0
      return-void
  
+     .line 6537
      :pswitch_1
      move-object/from16 v0, p0
  
***************
*** 106,111 ****
--- 111,117 ----
  
      monitor-enter v3
  
+     .line 6538
      :try_start_0
      move-object/from16 v0, p0
  
***************
*** 113,118 ****
--- 119,125 ----
  
      iget-object v15, v2, Lcom/android/server/wm/WindowManagerService;->mLastFocus:Lcom/android/server/wm/WindowState;
  
+     .line 6539
      .local v15, lastFocus:Lcom/android/server/wm/WindowState;
      move-object/from16 v0, p0
  
***************
*** 122,136 ****
--- 129,146 ----
  
      move-object/from16 v17, v0
  
+     .line 6540
      .local v17, newFocus:Lcom/android/server/wm/WindowState;
      move-object/from16 v0, v17
  
      if-ne v15, v0, :cond_1
  
+     .line 6542
      monitor-exit v3
  
      goto :goto_0
  
+     .line 6553
      .end local v15           #lastFocus:Lcom/android/server/wm/WindowState;
      .end local v17           #newFocus:Lcom/android/server/wm/WindowState;
      :catchall_0
***************
*** 142,147 ****
--- 152,158 ----
  
      throw v2
  
+     .line 6544
      .restart local v15       #lastFocus:Lcom/android/server/wm/WindowState;
      .restart local v17       #newFocus:Lcom/android/server/wm/WindowState;
      :cond_1
***************
*** 154,159 ****
--- 165,171 ----
  
      iput-object v0, v2, Lcom/android/server/wm/WindowManagerService;->mLastFocus:Lcom/android/server/wm/WindowState;
  
+     .line 6547
      if-eqz v17, :cond_2
  
      if-eqz v15, :cond_2
***************
*** 164,169 ****
--- 176,182 ----
  
      if-nez v2, :cond_2
  
+     .line 6550
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 172,190 ****
--- 185,208 ----
  
      invoke-virtual {v2, v15}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
  
+     .line 6551
      const/4 v15, 0x0
  
+     .line 6553
      :cond_2
      monitor-exit v3
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
+     .line 6555
      move-object/from16 v0, v17
  
      if-eq v15, v0, :cond_0
  
+     .line 6558
      if-eqz v17, :cond_3
  
+     .line 6561
      :try_start_2
      move-object/from16 v0, v17
  
***************
*** 202,207 ****
--- 220,226 ----
      :try_end_2
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_a
  
+     .line 6565
      :goto_1
      move-object/from16 v0, p0
  
***************
*** 210,218 ****
--- 229,239 ----
      #calls: Lcom/android/server/wm/WindowManagerService;->notifyFocusChanged()V
      invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$400(Lcom/android/server/wm/WindowManagerService;)V
  
+     .line 6568
      :cond_3
      if-eqz v15, :cond_0
  
+     .line 6571
      :try_start_3
      iget-object v2, v15, Lcom/android/server/wm/WindowState;->mClient:Landroid/view/IWindow;
  
***************
*** 230,240 ****
--- 251,263 ----
  
      goto :goto_0
  
+     .line 6572
      :catch_0
      move-exception v2
  
      goto :goto_0
  
+     .line 6582
      .end local v15           #lastFocus:Lcom/android/server/wm/WindowState;
      .end local v17           #newFocus:Lcom/android/server/wm/WindowState;
      :pswitch_2
***************
*** 246,251 ****
--- 269,275 ----
  
      monitor-enter v3
  
+     .line 6583
      :try_start_4
      move-object/from16 v0, p0
  
***************
*** 255,260 ****
--- 279,285 ----
  
      move-object/from16 v16, v0
  
+     .line 6584
      .local v16, losers:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/server/wm/WindowState;>;"
      move-object/from16 v0, p0
  
***************
*** 266,279 ****
--- 291,307 ----
  
      iput-object v4, v2, Lcom/android/server/wm/WindowManagerService;->mLosingFocus:Ljava/util/ArrayList;
  
+     .line 6585
      monitor-exit v3
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_1
  
+     .line 6587
      invoke-virtual/range {v16 .. v16}, Ljava/util/ArrayList;->size()I
  
      move-result v11
  
+     .line 6588
      .local v11, N:I
      const/4 v14, 0x0
  
***************
*** 281,286 ****
--- 309,315 ----
      :goto_2
      if-ge v14, v11, :cond_0
  
+     .line 6591
      :try_start_5
      move-object/from16 v0, v16
  
***************
*** 304,314 ****
--- 333,345 ----
      :try_end_5
      .catch Landroid/os/RemoteException; {:try_start_5 .. :try_end_5} :catch_9
  
+     .line 6588
      :goto_3
      add-int/lit8 v14, v14, 0x1
  
      goto :goto_2
  
+     .line 6585
      .end local v11           #N:I
      .end local v14           #i:I
      .end local v16           #losers:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/server/wm/WindowState;>;"
***************
*** 322,327 ****
--- 353,359 ----
  
      throw v2
  
+     .line 6599
      :pswitch_3
      move-object/from16 v0, p0
  
***************
*** 331,336 ****
--- 363,369 ----
  
      monitor-enter v3
  
+     .line 6600
      :try_start_7
      move-object/from16 v0, p0
  
***************
*** 340,345 ****
--- 373,379 ----
  
      iput-boolean v4, v2, Lcom/android/server/wm/WindowManagerService;->mAnimationPending:Z
  
+     .line 6601
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 347,352 ****
--- 381,387 ----
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
      invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$500(Lcom/android/server/wm/WindowManagerService;)V
  
+     .line 6602
      monitor-exit v3
  
      goto/16 :goto_0
***************
*** 360,365 ****
--- 395,401 ----
  
      throw v2
  
+     .line 6606
      :pswitch_4
      move-object/from16 v0, p1
  
***************
*** 369,374 ****
--- 405,411 ----
  
      check-cast v29, Lcom/android/server/wm/AppWindowToken;
  
+     .line 6607
      .local v29, wtoken:Lcom/android/server/wm/AppWindowToken;
      move-object/from16 v0, v29
  
***************
*** 376,386 ****
--- 413,426 ----
  
      move-object/from16 v23, v0
  
+     .line 6609
      .local v23, sd:Lcom/android/server/wm/StartingData;
      if-eqz v23, :cond_0
  
+     .line 6617
      const/16 v26, 0x0
  
+     .line 6619
      .local v26, view:Landroid/view/View;
      :try_start_8
      move-object/from16 v0, p0
***************
*** 427,437 ****
--- 467,480 ----
  
      move-result-object v26
  
+     .line 6626
      :goto_4
      if-eqz v26, :cond_0
  
+     .line 6627
      const/4 v12, 0x0
  
+     .line 6629
      .local v12, abort:Z
      move-object/from16 v0, p0
  
***************
*** 441,446 ****
--- 484,490 ----
  
      monitor-enter v3
  
+     .line 6630
      :try_start_9
      move-object/from16 v0, v29
  
***************
*** 454,459 ****
--- 498,504 ----
  
      if-nez v2, :cond_6
  
+     .line 6633
      :cond_4
      move-object/from16 v0, v29
  
***************
*** 461,488 ****
--- 506,539 ----
  
      if-eqz v2, :cond_5
  
+     .line 6638
      const/4 v2, 0x0
  
      move-object/from16 v0, v29
  
      iput-object v2, v0, Lcom/android/server/wm/AppWindowToken;->startingWindow:Lcom/android/server/wm/WindowState;
  
+     .line 6639
      const/4 v2, 0x0
  
      move-object/from16 v0, v29
  
      iput-object v2, v0, Lcom/android/server/wm/AppWindowToken;->startingData:Lcom/android/server/wm/StartingData;
  
+     .line 6640
      const/4 v12, 0x1
  
+     .line 6650
      :cond_5
      :goto_5
      monitor-exit v3
      :try_end_9
      .catchall {:try_start_9 .. :try_end_9} :catchall_3
  
+     .line 6652
      if-eqz v12, :cond_0
  
+     .line 6654
      :try_start_a
      move-object/from16 v0, p0
  
***************
*** 502,510 ****
--- 553,563 ----
  
      goto/16 :goto_0
  
+     .line 6655
      :catch_1
      move-exception v13
  
+     .line 6656
      .local v13, e:Ljava/lang/Exception;
      const-string v2, "WindowManager"
  
***************
*** 514,524 ****
--- 567,579 ----
  
      goto/16 :goto_0
  
+     .line 6622
      .end local v12           #abort:Z
      .end local v13           #e:Ljava/lang/Exception;
      :catch_2
      move-exception v13
  
+     .line 6623
      .restart local v13       #e:Ljava/lang/Exception;
      const-string v2, "WindowManager"
  
***************
*** 528,533 ****
--- 583,589 ----
  
      goto :goto_4
  
+     .line 6643
      .end local v13           #e:Ljava/lang/Exception;
      .restart local v12       #abort:Z
      :cond_6
***************
*** 540,545 ****
--- 596,602 ----
  
      goto :goto_5
  
+     .line 6650
      :catchall_3
      move-exception v2
  
***************
*** 549,554 ****
--- 606,612 ----
  
      throw v2
  
+     .line 6663
      .end local v12           #abort:Z
      .end local v23           #sd:Lcom/android/server/wm/StartingData;
      .end local v26           #view:Landroid/view/View;
***************
*** 562,573 ****
--- 620,634 ----
  
      check-cast v29, Lcom/android/server/wm/AppWindowToken;
  
+     .line 6664
      .restart local v29       #wtoken:Lcom/android/server/wm/AppWindowToken;
      const/16 v25, 0x0
  
+     .line 6665
      .local v25, token:Landroid/os/IBinder;
      const/16 v26, 0x0
  
+     .line 6666
      .restart local v26       #view:Landroid/view/View;
      move-object/from16 v0, p0
  
***************
*** 577,582 ****
--- 638,644 ----
  
      monitor-enter v3
  
+     .line 6671
      :try_start_c
      move-object/from16 v0, v29
  
***************
*** 584,626 ****
--- 646,696 ----
  
      if-eqz v2, :cond_7
  
+     .line 6672
      move-object/from16 v0, v29
  
      iget-object v0, v0, Lcom/android/server/wm/AppWindowToken;->startingView:Landroid/view/View;
  
      move-object/from16 v26, v0
  
+     .line 6673
      move-object/from16 v0, v29
  
      iget-object v0, v0, Lcom/android/server/wm/AppWindowToken;->token:Landroid/os/IBinder;
  
      move-object/from16 v25, v0
  
+     .line 6674
      const/4 v2, 0x0
  
      move-object/from16 v0, v29
  
      iput-object v2, v0, Lcom/android/server/wm/AppWindowToken;->startingData:Lcom/android/server/wm/StartingData;
  
+     .line 6675
      const/4 v2, 0x0
  
      move-object/from16 v0, v29
  
      iput-object v2, v0, Lcom/android/server/wm/AppWindowToken;->startingView:Landroid/view/View;
  
+     .line 6676
      const/4 v2, 0x0
  
      move-object/from16 v0, v29
  
      iput-object v2, v0, Lcom/android/server/wm/AppWindowToken;->startingWindow:Lcom/android/server/wm/WindowState;
  
+     .line 6678
      :cond_7
      monitor-exit v3
      :try_end_c
      .catchall {:try_start_c .. :try_end_c} :catchall_4
  
+     .line 6679
      if-eqz v26, :cond_0
  
+     .line 6681
      :try_start_d
      move-object/from16 v0, p0
  
***************
*** 638,646 ****
--- 708,718 ----
  
      goto/16 :goto_0
  
+     .line 6682
      :catch_3
      move-exception v13
  
+     .line 6683
      .restart local v13       #e:Ljava/lang/Exception;
      const-string v2, "WindowManager"
  
***************
*** 650,655 ****
--- 722,728 ----
  
      goto/16 :goto_0
  
+     .line 6678
      .end local v13           #e:Ljava/lang/Exception;
      :catchall_4
      move-exception v2
***************
*** 661,675 ****
--- 734,751 ----
  
      throw v2
  
+     .line 6689
      .end local v25           #token:Landroid/os/IBinder;
      .end local v26           #view:Landroid/view/View;
      .end local v29           #wtoken:Lcom/android/server/wm/AppWindowToken;
      :pswitch_6
      const/16 v25, 0x0
  
+     .line 6690
      .restart local v25       #token:Landroid/os/IBinder;
      const/16 v26, 0x0
  
+     .line 6692
      .restart local v26       #view:Landroid/view/View;
      :goto_6
      move-object/from16 v0, p0
***************
*** 680,685 ****
--- 756,762 ----
  
      monitor-enter v3
  
+     .line 6693
      :try_start_f
      move-object/from16 v0, p0
  
***************
*** 691,703 ****
--- 768,783 ----
  
      move-result v11
  
+     .line 6694
      .restart local v11       #N:I
      if-gtz v11, :cond_8
  
+     .line 6695
      monitor-exit v3
  
      goto/16 :goto_0
  
+     .line 6713
      .end local v11           #N:I
      :catchall_5
      move-exception v2
***************
*** 708,713 ****
--- 788,794 ----
  
      throw v2
  
+     .line 6697
      .restart local v11       #N:I
      :cond_8
      :try_start_10
***************
*** 725,730 ****
--- 806,812 ----
  
      check-cast v29, Lcom/android/server/wm/AppWindowToken;
  
+     .line 6704
      .restart local v29       #wtoken:Lcom/android/server/wm/AppWindowToken;
      move-object/from16 v0, v29
  
***************
*** 732,741 ****
--- 814,825 ----
  
      if-nez v2, :cond_9
  
+     .line 6705
      monitor-exit v3
  
      goto :goto_6
  
+     .line 6708
      :cond_9
      move-object/from16 v0, v29
  
***************
*** 743,776 ****
--- 827,866 ----
  
      move-object/from16 v26, v0
  
+     .line 6709
      move-object/from16 v0, v29
  
      iget-object v0, v0, Lcom/android/server/wm/AppWindowToken;->token:Landroid/os/IBinder;
  
      move-object/from16 v25, v0
  
+     .line 6710
      const/4 v2, 0x0
  
      move-object/from16 v0, v29
  
      iput-object v2, v0, Lcom/android/server/wm/AppWindowToken;->startingData:Lcom/android/server/wm/StartingData;
  
+     .line 6711
      const/4 v2, 0x0
  
      move-object/from16 v0, v29
  
      iput-object v2, v0, Lcom/android/server/wm/AppWindowToken;->startingView:Landroid/view/View;
  
+     .line 6712
      const/4 v2, 0x0
  
      move-object/from16 v0, v29
  
      iput-object v2, v0, Lcom/android/server/wm/AppWindowToken;->startingWindow:Lcom/android/server/wm/WindowState;
  
+     .line 6713
      monitor-exit v3
      :try_end_10
      .catchall {:try_start_10 .. :try_end_10} :catchall_5
  
+     .line 6716
      :try_start_11
      move-object/from16 v0, p0
  
***************
*** 788,796 ****
--- 878,888 ----
  
      goto :goto_6
  
+     .line 6717
      :catch_4
      move-exception v13
  
+     .line 6718
      .restart local v13       #e:Ljava/lang/Exception;
      const-string v2, "WindowManager"
  
***************
*** 800,805 ****
--- 892,898 ----
  
      goto :goto_6
  
+     .line 6724
      .end local v11           #N:I
      .end local v13           #e:Ljava/lang/Exception;
      .end local v25           #token:Landroid/os/IBinder;
***************
*** 814,819 ****
--- 907,913 ----
  
      check-cast v29, Lcom/android/server/wm/AppWindowToken;
  
+     .line 6729
      .restart local v29       #wtoken:Lcom/android/server/wm/AppWindowToken;
      :try_start_12
      move-object/from16 v0, v29
***************
*** 826,836 ****
--- 920,932 ----
  
      goto/16 :goto_0
  
+     .line 6730
      :catch_5
      move-exception v2
  
      goto/16 :goto_0
  
+     .line 6735
      .end local v29           #wtoken:Lcom/android/server/wm/AppWindowToken;
      :pswitch_8
      move-object/from16 v0, p1
***************
*** 841,846 ****
--- 937,943 ----
  
      check-cast v29, Lcom/android/server/wm/AppWindowToken;
  
+     .line 6737
      .restart local v29       #wtoken:Lcom/android/server/wm/AppWindowToken;
      move-object/from16 v0, p1
  
***************
*** 850,855 ****
--- 947,953 ----
  
      const/16 v20, 0x1
  
+     .line 6738
      .local v20, nowVisible:Z
      :goto_7
      move-object/from16 v0, p1
***************
*** 860,869 ****
--- 958,969 ----
  
      const/16 v19, 0x1
  
+     .line 6745
      .local v19, nowGone:Z
      :goto_8
      if-eqz v20, :cond_c
  
+     .line 6746
      :try_start_13
      move-object/from16 v0, v29
  
***************
*** 873,883 ****
--- 973,985 ----
  
      goto/16 :goto_0
  
+     .line 6750
      :catch_6
      move-exception v2
  
      goto/16 :goto_0
  
+     .line 6737
      .end local v19           #nowGone:Z
      .end local v20           #nowVisible:Z
      :cond_a
***************
*** 885,896 ****
--- 987,1000 ----
  
      goto :goto_7
  
+     .line 6738
      .restart local v20       #nowVisible:Z
      :cond_b
      const/16 v19, 0x0
  
      goto :goto_8
  
+     .line 6748
      .restart local v19       #nowGone:Z
      :cond_c
      move-object/from16 v0, v29
***************
*** 903,908 ****
--- 1007,1013 ----
  
      goto/16 :goto_0
  
+     .line 6755
      .end local v19           #nowGone:Z
      .end local v20           #nowVisible:Z
      .end local v29           #wtoken:Lcom/android/server/wm/AppWindowToken;
***************
*** 915,920 ****
--- 1020,1026 ----
  
      monitor-enter v3
  
+     .line 6756
      :try_start_14
      const-string v2, "WindowManager"
  
***************
*** 922,927 ****
--- 1028,1034 ----
  
      invoke-static {v2, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 6757
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 932,944 ****
--- 1039,1054 ----
  
      move-result v14
  
+     .line 6758
      .restart local v14       #i:I
      :cond_d
      :goto_9
      if-lez v14, :cond_e
  
+     .line 6759
      add-int/lit8 v14, v14, -0x1
  
+     .line 6760
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 951,956 ****
--- 1061,1067 ----
  
      check-cast v27, Lcom/android/server/wm/WindowState;
  
+     .line 6761
      .local v27, w:Lcom/android/server/wm/WindowState;
      move-object/from16 v0, v27
  
***************
*** 958,969 ****
--- 1069,1082 ----
  
      if-eqz v2, :cond_d
  
+     .line 6762
      const/4 v2, 0x0
  
      move-object/from16 v0, v27
  
      iput-boolean v2, v0, Lcom/android/server/wm/WindowState;->mOrientationChanging:Z
  
+     .line 6763
      const-string v2, "WindowManager"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 990,995 ****
--- 1103,1109 ----
  
      goto :goto_9
  
+     .line 6767
      .end local v14           #i:I
      .end local v27           #w:Lcom/android/server/wm/WindowState;
      :catchall_6
***************
*** 1001,1006 ****
--- 1115,1121 ----
  
      throw v2
  
+     .line 6766
      .restart local v14       #i:I
      :cond_e
      :try_start_15
***************
*** 1011,1022 ****
--- 1126,1139 ----
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
      invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$500(Lcom/android/server/wm/WindowManagerService;)V
  
+     .line 6767
      monitor-exit v3
      :try_end_15
      .catchall {:try_start_15 .. :try_end_15} :catchall_6
  
      goto/16 :goto_0
  
+     .line 6774
      .end local v14           #i:I
      :pswitch_a
      move-object/from16 v0, p0
***************
*** 1027,1032 ****
--- 1144,1150 ----
  
      monitor-enter v3
  
+     .line 6775
      :try_start_16
      move-object/from16 v0, p0
  
***************
*** 1034,1039 ****
--- 1152,1158 ----
  
      move-object/from16 v21, v0
  
+     .line 6776
      .local v21, oldHold:Lcom/android/server/wm/Session;
      move-object/from16 v0, p1
  
***************
*** 1043,1048 ****
--- 1162,1168 ----
  
      check-cast v18, Lcom/android/server/wm/Session;
  
+     .line 6777
      .local v18, newHold:Lcom/android/server/wm/Session;
      move-object/from16 v0, v18
  
***************
*** 1050,1067 ****
--- 1170,1191 ----
  
      iput-object v0, v1, Lcom/android/server/wm/WindowManagerService$H;->mLastReportedHold:Lcom/android/server/wm/Session;
  
+     .line 6778
      monitor-exit v3
      :try_end_16
      .catchall {:try_start_16 .. :try_end_16} :catchall_7
  
+     .line 6780
      move-object/from16 v0, v21
  
      move-object/from16 v1, v18
  
      if-eq v0, v1, :cond_0
  
+     .line 6782
      if-eqz v21, :cond_f
  
+     .line 6783
      :try_start_17
      move-object/from16 v0, p0
  
***************
*** 1081,1089 ****
--- 1205,1215 ----
  
      invoke-interface {v2, v3, v4, v5, v6}, Lcom/android/internal/app/IBatteryStats;->noteStopWakelock(IILjava/lang/String;I)V
  
+     .line 6787
      :cond_f
      if-eqz v18, :cond_0
  
+     .line 6788
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1106,1116 ****
--- 1232,1244 ----
  
      goto/16 :goto_0
  
+     .line 6792
      :catch_7
      move-exception v2
  
      goto/16 :goto_0
  
+     .line 6778
      .end local v18           #newHold:Lcom/android/server/wm/Session;
      .end local v21           #oldHold:Lcom/android/server/wm/Session;
      :catchall_7
***************
*** 1123,1128 ****
--- 1251,1257 ----
  
      throw v2
  
+     .line 6799
      :pswitch_b
      move-object/from16 v0, p0
  
***************
*** 1132,1137 ****
--- 1261,1267 ----
  
      monitor-enter v3
  
+     .line 6800
      :try_start_19
      move-object/from16 v0, p0
  
***************
*** 1143,1148 ****
--- 1273,1279 ----
  
      if-eq v2, v4, :cond_10
  
+     .line 6803
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1151,1156 ****
--- 1282,1288 ----
  
      iput-boolean v4, v2, Lcom/android/server/wm/WindowManagerService;->mAppTransitionReady:Z
  
+     .line 6804
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1159,1164 ****
--- 1291,1297 ----
  
      iput-boolean v4, v2, Lcom/android/server/wm/WindowManagerService;->mAppTransitionTimeout:Z
  
+     .line 6805
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1166,1171 ****
--- 1299,1305 ----
      #calls: Lcom/android/server/wm/WindowManagerService;->performLayoutAndPlaceSurfacesLocked()V
      invoke-static {v2}, Lcom/android/server/wm/WindowManagerService;->access$500(Lcom/android/server/wm/WindowManagerService;)V
  
+     .line 6807
      :cond_10
      monitor-exit v3
  
***************
*** 1180,1185 ****
--- 1314,1320 ----
  
      throw v2
  
+     .line 6812
      :pswitch_c
      move-object/from16 v0, p0
  
***************
*** 1201,1206 ****
--- 1336,1342 ----
  
      invoke-static {v2, v3, v4}, Landroid/provider/Settings$System;->putFloat(Landroid/content/ContentResolver;Ljava/lang/String;F)Z
  
+     .line 6814
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1223,1228 ****
--- 1359,1365 ----
  
      goto/16 :goto_0
  
+     .line 6820
      :pswitch_d
      move-object/from16 v0, p0
  
***************
*** 1232,1237 ****
--- 1369,1375 ----
  
      monitor-enter v3
  
+     .line 6821
      :try_start_1a
      move-object/from16 v0, p0
  
***************
*** 1241,1246 ****
--- 1379,1385 ----
  
      if-eqz v2, :cond_11
  
+     .line 6824
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1263,1272 ****
--- 1402,1413 ----
  
      invoke-virtual {v2, v4, v5, v6}, Lcom/android/server/wm/WindowManagerService$H;->sendMessageDelayed(Landroid/os/Message;J)Z
  
+     .line 6826
      monitor-exit v3
  
      goto/16 :goto_0
  
+     .line 6833
      :catchall_9
      move-exception v2
  
***************
*** 1276,1281 ****
--- 1417,1423 ----
  
      throw v2
  
+     .line 6830
      :cond_11
      :try_start_1b
      move-object/from16 v0, p0
***************
*** 1286,1300 ****
--- 1428,1445 ----
  
      if-eqz v2, :cond_12
  
+     .line 6831
      monitor-exit v3
  
      goto/16 :goto_0
  
+     .line 6833
      :cond_12
      monitor-exit v3
      :try_end_1b
      .catchall {:try_start_1b .. :try_end_1b} :catchall_9
  
+     .line 6834
      invoke-static {}, Ljava/lang/Runtime;->getRuntime()Ljava/lang/Runtime;
  
      move-result-object v2
***************
*** 1303,1308 ****
--- 1448,1454 ----
  
      goto/16 :goto_0
  
+     .line 6839
      :pswitch_e
      move-object/from16 v0, p0
  
***************
*** 1312,1317 ****
--- 1458,1464 ----
  
      goto/16 :goto_0
  
+     .line 6844
      :pswitch_f
      move-object/from16 v0, p0
  
***************
*** 1321,1326 ****
--- 1468,1474 ----
  
      monitor-enter v3
  
+     .line 6845
      :try_start_1c
      const-string v2, "WindowManager"
  
***************
*** 1328,1333 ****
--- 1476,1482 ----
  
      invoke-static {v2, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 6846
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1338,1350 ****
--- 1487,1502 ----
  
      move-result v14
  
+     .line 6847
      .restart local v14       #i:I
      :cond_13
      :goto_a
      if-lez v14, :cond_14
  
+     .line 6848
      add-int/lit8 v14, v14, -0x1
  
+     .line 6849
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1357,1362 ****
--- 1509,1515 ----
  
      check-cast v24, Lcom/android/server/wm/AppWindowToken;
  
+     .line 6850
      .local v24, tok:Lcom/android/server/wm/AppWindowToken;
      move-object/from16 v0, v24
  
***************
*** 1364,1369 ****
--- 1517,1523 ----
  
      if-eqz v2, :cond_13
  
+     .line 6851
      const-string v2, "WindowManager"
  
      new-instance v4, Ljava/lang/StringBuilder;
***************
*** 1388,1393 ****
--- 1542,1548 ----
  
      invoke-static {v2, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 6852
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1402,1407 ****
--- 1557,1563 ----
  
      goto :goto_a
  
+     .line 6855
      .end local v14           #i:I
      .end local v24           #tok:Lcom/android/server/wm/AppWindowToken;
      :catchall_a
***************
*** 1422,1427 ****
--- 1578,1584 ----
  
      goto/16 :goto_0
  
+     .line 6860
      .end local v14           #i:I
      :pswitch_10
      const/16 v2, 0x12
***************
*** 1430,1435 ****
--- 1587,1593 ----
  
      invoke-virtual {v0, v2}, Lcom/android/server/wm/WindowManagerService$H;->removeMessages(I)V
  
+     .line 6861
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1438,1443 ****
--- 1596,1602 ----
  
      goto/16 :goto_0
  
+     .line 6866
      :pswitch_11
      move-object/from16 v0, p0
  
***************
*** 1450,1455 ****
--- 1609,1615 ----
  
      if-eqz v2, :cond_0
  
+     .line 6867
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1458,1463 ****
--- 1618,1624 ----
  
      monitor-enter v3
  
+     .line 6868
      :try_start_1e
      move-object/from16 v0, p0
  
***************
*** 1468,1477 ****
--- 1629,1640 ----
      #setter for: Lcom/android/server/wm/WindowManagerService;->mWindowsChanged:Z
      invoke-static {v2, v4}, Lcom/android/server/wm/WindowManagerService;->access$602(Lcom/android/server/wm/WindowManagerService;Z)Z
  
+     .line 6869
      monitor-exit v3
      :try_end_1e
      .catchall {:try_start_1e .. :try_end_1e} :catchall_b
  
+     .line 6870
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1481,1486 ****
--- 1644,1650 ----
  
      goto/16 :goto_0
  
+     .line 6869
      :catchall_b
      move-exception v2
  
***************
*** 1491,1496 ****
--- 1655,1661 ----
  
      throw v2
  
+     .line 6876
      :pswitch_12
      move-object/from16 v0, p1
  
***************
*** 1500,1505 ****
--- 1665,1671 ----
  
      check-cast v28, Landroid/os/IBinder;
  
+     .line 6880
      .local v28, win:Landroid/os/IBinder;
      move-object/from16 v0, p0
  
***************
*** 1509,1514 ****
--- 1675,1681 ----
  
      monitor-enter v3
  
+     .line 6882
      :try_start_20
      move-object/from16 v0, p0
  
***************
*** 1518,1523 ****
--- 1685,1691 ----
  
      if-eqz v2, :cond_15
  
+     .line 6883
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1526,1531 ****
--- 1694,1700 ----
  
      invoke-virtual {v2}, Lcom/android/server/wm/DragState;->unregister()V
  
+     .line 6884
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1536,1541 ****
--- 1705,1711 ----
  
      invoke-virtual {v2, v4}, Lcom/android/server/wm/InputMonitor;->updateInputWindowsLw(Z)V
  
+     .line 6885
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1544,1549 ****
--- 1714,1720 ----
  
      invoke-virtual {v2}, Lcom/android/server/wm/DragState;->reset()V
  
+     .line 6886
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1552,1557 ****
--- 1723,1729 ----
  
      iput-object v4, v2, Lcom/android/server/wm/WindowManagerService;->mDragState:Lcom/android/server/wm/DragState;
  
+     .line 6888
      :cond_15
      monitor-exit v3
  
***************
*** 1566,1571 ****
--- 1738,1744 ----
  
      throw v2
  
+     .line 6893
      .end local v28           #win:Landroid/os/IBinder;
      :pswitch_13
      move-object/from16 v0, p1
***************
*** 1576,1581 ****
--- 1749,1755 ----
  
      check-cast v28, Landroid/os/IBinder;
  
+     .line 6897
      .restart local v28       #win:Landroid/os/IBinder;
      move-object/from16 v0, p0
  
***************
*** 1585,1590 ****
--- 1759,1765 ----
  
      monitor-enter v3
  
+     .line 6899
      :try_start_21
      move-object/from16 v0, p0
  
***************
*** 1594,1599 ****
--- 1769,1775 ----
  
      if-eqz v2, :cond_16
  
+     .line 6900
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1604,1609 ****
--- 1780,1786 ----
  
      iput-boolean v4, v2, Lcom/android/server/wm/DragState;->mDragResult:Z
  
+     .line 6901
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1612,1617 ****
--- 1789,1795 ----
  
      invoke-virtual {v2}, Lcom/android/server/wm/DragState;->endDragLw()V
  
+     .line 6903
      :cond_16
      monitor-exit v3
  
***************
*** 1626,1631 ****
--- 1804,1810 ----
  
      throw v2
  
+     .line 6908
      .end local v28           #win:Landroid/os/IBinder;
      :pswitch_14
      move-object/from16 v0, p0
***************
*** 1636,1641 ****
--- 1815,1821 ----
  
      goto/16 :goto_0
  
+     .line 6913
      :pswitch_15
      move-object/from16 v0, p0
  
***************
*** 1645,1650 ****
--- 1825,1831 ----
  
      goto/16 :goto_0
  
+     .line 6919
      :pswitch_16
      move-object/from16 v0, p0
  
***************
*** 1654,1659 ****
--- 1835,1841 ----
  
      monitor-enter v3
  
+     .line 6920
      :try_start_22
      move-object/from16 v0, p1
  
***************
*** 1663,1668 ****
--- 1845,1851 ----
  
      check-cast v22, Landroid/util/Pair;
  
+     .line 6921
      .local v22, pair:Landroid/util/Pair;,"Landroid/util/Pair<Lcom/android/server/wm/WindowState;Landroid/os/IRemoteCallback;>;"
      const-string v2, "WindowManager"
  
***************
*** 1690,1695 ****
--- 1873,1879 ----
  
      invoke-static {v2, v4}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
  
+     .line 6922
      move-object/from16 v0, p0
  
      iget-object v2, v0, Lcom/android/server/wm/WindowManagerService$H;->this$0:Lcom/android/server/wm/WindowManagerService;
***************
*** 1704,1713 ****
--- 1888,1899 ----
  
      if-nez v2, :cond_17
  
+     .line 6923
      monitor-exit v3
  
      goto/16 :goto_0
  
+     .line 6925
      .end local v22           #pair:Landroid/util/Pair;,"Landroid/util/Pair<Lcom/android/server/wm/WindowState;Landroid/os/IRemoteCallback;>;"
      :catchall_e
      move-exception v2
***************
*** 1725,1730 ****
--- 1911,1917 ----
      :try_end_23
      .catchall {:try_start_23 .. :try_end_23} :catchall_e
  
+     .line 6927
      :try_start_24
      move-object/from16 v0, v22
  
***************
*** 1740,1750 ****
--- 1927,1939 ----
  
      goto/16 :goto_0
  
+     .line 6928
      :catch_8
      move-exception v2
  
      goto/16 :goto_0
  
+     .line 6592
      .end local v22           #pair:Landroid/util/Pair;,"Landroid/util/Pair<Lcom/android/server/wm/WindowState;Landroid/os/IRemoteCallback;>;"
      .restart local v11       #N:I
      .restart local v14       #i:I
***************
*** 1754,1759 ****
--- 1943,1949 ----
  
      goto/16 :goto_3
  
+     .line 6562
      .end local v11           #N:I
      .end local v14           #i:I
      .end local v16           #losers:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Lcom/android/server/wm/WindowState;>;"
***************
*** 1764,1769 ****
--- 1954,1960 ----
  
      goto/16 :goto_1
  
+     .line 6532
      :pswitch_data_0
      .packed-switch 0x2
          :pswitch_1
