*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/WindowManagerService$4.smali	2012-03-23 19:35:05.222682121 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/services.jar.out/smali/com/android/server/wm/WindowManagerService$4.smali.original	2012-03-23 19:34:47.111351862 -0500
***************
*** 30,35 ****
--- 30,36 ----
      .parameter
  
      .prologue
+     .line 4587
      iput-object p1, p0, Lcom/android/server/wm/WindowManagerService$4;->this$0:Lcom/android/server/wm/WindowManagerService;
  
      iput-object p2, p0, Lcom/android/server/wm/WindowManagerService$4;->val$callback:Landroid/view/IOnKeyguardExitResult;
***************
*** 46,51 ****
--- 47,53 ----
      .parameter "success"
  
      .prologue
+     .line 4590
      :try_start_0
      iget-object v0, p0, Lcom/android/server/wm/WindowManagerService$4;->val$callback:Landroid/view/IOnKeyguardExitResult;
  
***************
*** 53,61 ****
--- 55,65 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
+     .line 4594
      :goto_0
      return-void
  
+     .line 4591
      :catch_0
      move-exception v0
  
