*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/mms/util/PduCache.smali	2012-03-23 19:37:57.291819797 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/google/android/mms/util/PduCache.smali.original	2012-03-23 19:34:46.430339443 -0500
***************
*** 108,113 ****
--- 108,114 ----
  
      const/4 v4, 0x2
  
+     .line 53
      new-instance v0, Landroid/content/UriMatcher;
  
      const/4 v1, -0x1
***************
*** 116,124 ****
  
      sput-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
      const/4 v2, 0x0
  
--- 117,126 ----
  
      sput-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
+     .line 54
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
      const/4 v2, 0x0
  
***************
*** 126,184 ****
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
      const-string v2, "#"
  
      invoke-virtual {v0, v1, v2, v6}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
      const-string v2, "inbox"
  
      invoke-virtual {v0, v1, v2, v4}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
      const-string v2, "inbox/#"
  
      invoke-virtual {v0, v1, v2, v7}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
!     const-string v2, "sent"
  
      invoke-virtual {v0, v1, v2, v5}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
!     const-string v2, "sent/#"
  
      const/4 v3, 0x5
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
      const-string v2, "drafts"
  
      invoke-virtual {v0, v1, v2, v8}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
      const-string v2, "drafts/#"
  
--- 128,193 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 55
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
      const-string v2, "#"
  
      invoke-virtual {v0, v1, v2, v6}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 56
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
      const-string v2, "inbox"
  
      invoke-virtual {v0, v1, v2, v4}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 57
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
      const-string v2, "inbox/#"
  
      invoke-virtual {v0, v1, v2, v7}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 58
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
!     const-string/jumbo v2, "sent"
  
      invoke-virtual {v0, v1, v2, v5}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 59
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
!     const-string/jumbo v2, "sent/#"
  
      const/4 v3, 0x5
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 60
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
      const-string v2, "drafts"
  
      invoke-virtual {v0, v1, v2, v8}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 61
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
      const-string v2, "drafts/#"
  
***************
*** 186,214 ****
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
!     const-string v2, "outbox"
  
      const/16 v3, 0x8
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms"
  
!     const-string v2, "outbox/#"
  
      const/16 v3, 0x9
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms-sms"
  
      const-string v2, "conversations"
  
--- 195,226 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 62
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
!     const-string/jumbo v2, "outbox"
  
      const/16 v3, 0x8
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 63
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms"
  
!     const-string/jumbo v2, "outbox/#"
  
      const/16 v3, 0x9
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 64
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms-sms"
  
      const-string v2, "conversations"
  
***************
*** 216,224 ****
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string v1, "mms-sms"
  
      const-string v2, "conversations/#"
  
--- 228,237 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 65
      sget-object v0, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
!     const-string/jumbo v1, "mms-sms"
  
      const-string v2, "conversations/#"
  
***************
*** 226,237 ****
--- 239,252 ----
  
      invoke-virtual {v0, v1, v2, v3}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
  
+     .line 67
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      sput-object v0, Lcom/google/android/mms/util/PduCache;->MATCH_TO_MSGBOX_ID_MAP:Ljava/util/HashMap;
  
+     .line 68
      sget-object v0, Lcom/google/android/mms/util/PduCache;->MATCH_TO_MSGBOX_ID_MAP:Ljava/util/HashMap;
  
      invoke-static {v4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 244,249 ****
--- 259,265 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 69
      sget-object v0, Lcom/google/android/mms/util/PduCache;->MATCH_TO_MSGBOX_ID_MAP:Ljava/util/HashMap;
  
      invoke-static {v5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 256,261 ****
--- 272,278 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 70
      sget-object v0, Lcom/google/android/mms/util/PduCache;->MATCH_TO_MSGBOX_ID_MAP:Ljava/util/HashMap;
  
      invoke-static {v8}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
***************
*** 268,273 ****
--- 285,291 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 71
      sget-object v0, Lcom/google/android/mms/util/PduCache;->MATCH_TO_MSGBOX_ID_MAP:Ljava/util/HashMap;
  
      const/16 v1, 0x8
***************
*** 282,287 ****
--- 300,306 ----
  
      invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 72
      return-void
  .end method
  
***************
*** 289,308 ****
--- 308,331 ----
      .locals 1
  
      .prologue
+     .line 77
      invoke-direct {p0}, Lcom/google/android/mms/util/AbstractCache;-><init>()V
  
+     .line 78
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/google/android/mms/util/PduCache;->mMessageBoxes:Ljava/util/HashMap;
  
+     .line 79
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      iput-object v0, p0, Lcom/google/android/mms/util/PduCache;->mThreads:Ljava/util/HashMap;
  
+     .line 80
      return-void
  .end method
  
***************
*** 310,315 ****
--- 333,339 ----
      .locals 2
  
      .prologue
+     .line 83
      const-class v1, Lcom/google/android/mms/util/PduCache;
  
      monitor-enter v1
***************
*** 319,330 ****
--- 343,356 ----
  
      if-nez v0, :cond_0
  
+     .line 87
      new-instance v0, Lcom/google/android/mms/util/PduCache;
  
      invoke-direct {v0}, Lcom/google/android/mms/util/PduCache;-><init>()V
  
      sput-object v0, Lcom/google/android/mms/util/PduCache;->sInstance:Lcom/google/android/mms/util/PduCache;
  
+     .line 89
      :cond_0
      sget-object v0, Lcom/google/android/mms/util/PduCache;->sInstance:Lcom/google/android/mms/util/PduCache;
      :try_end_0
***************
*** 334,339 ****
--- 360,366 ----
  
      return-object v0
  
+     .line 83
      :catchall_0
      move-exception v0
  
***************
*** 347,383 ****
--- 374,419 ----
      .parameter "uri"
  
      .prologue
+     .line 171
      sget-object v3, Lcom/google/android/mms/util/PduCache;->URI_MATCHER:Landroid/content/UriMatcher;
  
      invoke-virtual {v3, p1}, Landroid/content/UriMatcher;->match(Landroid/net/Uri;)I
  
      move-result v0
  
+     .line 172
      .local v0, match:I
      const/4 v2, 0x0
  
+     .line 174
      .local v2, normalizedKey:Landroid/net/Uri;
      packed-switch v0, :pswitch_data_0
  
+     .line 186
      :pswitch_0
      const/4 v3, 0x0
  
+     .line 192
      :goto_0
      return-object v3
  
+     .line 176
      :pswitch_1
      move-object v2, p1
  
      :goto_1
      move-object v3, v2
  
+     .line 192
      goto :goto_0
  
+     .line 182
      :pswitch_2
      invoke-virtual {p1}, Landroid/net/Uri;->getLastPathSegment()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 183
      .local v1, msgId:Ljava/lang/String;
      sget-object v3, Landroid/provider/Telephony$Mms;->CONTENT_URI:Landroid/net/Uri;
  
***************
*** 385,392 ****
--- 421,430 ----
  
      move-result-object v2
  
+     .line 184
      goto :goto_1
  
+     .line 174
      :pswitch_data_0
      .packed-switch 0x1
          :pswitch_1
***************
*** 406,413 ****
--- 444,453 ----
      .parameter "msgBoxId"
  
      .prologue
+     .line 200
      if-eqz p1, :cond_1
  
+     .line 201
      iget-object v4, p0, Lcom/google/android/mms/util/PduCache;->mMessageBoxes:Ljava/util/HashMap;
  
      invoke-virtual {v4, p1}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
***************
*** 416,424 ****
--- 456,466 ----
  
      check-cast v3, Ljava/util/HashSet;
  
+     .line 202
      .local v3, msgBox:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      if-eqz v3, :cond_1
  
+     .line 203
      invoke-virtual {v3}, Ljava/util/HashSet;->iterator()Ljava/util/Iterator;
  
      move-result-object v1
***************
*** 438,443 ****
--- 480,486 ----
  
      check-cast v2, Landroid/net/Uri;
  
+     .line 204
      .local v2, key:Landroid/net/Uri;
      invoke-super {p0, v2}, Lcom/google/android/mms/util/AbstractCache;->purge(Ljava/lang/Object;)Ljava/lang/Object;
  
***************
*** 445,457 ****
--- 488,503 ----
  
      check-cast v0, Lcom/google/android/mms/util/PduCacheEntry;
  
+     .line 205
      .local v0, entry:Lcom/google/android/mms/util/PduCacheEntry;
      if-eqz v0, :cond_0
  
+     .line 206
      invoke-direct {p0, v2, v0}, Lcom/google/android/mms/util/PduCache;->removeFromThreads(Landroid/net/Uri;Lcom/google/android/mms/util/PduCacheEntry;)V
  
      goto :goto_0
  
+     .line 211
      .end local v0           #entry:Lcom/google/android/mms/util/PduCacheEntry;
      .end local v1           #i$:Ljava/util/Iterator;
      .end local v2           #key:Landroid/net/Uri;
***************
*** 465,470 ****
--- 511,517 ----
      .parameter "threadId"
  
      .prologue
+     .line 225
      iget-object v4, p0, Lcom/google/android/mms/util/PduCache;->mThreads:Ljava/util/HashMap;
  
      invoke-static {p1, p2}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
***************
*** 477,485 ****
--- 524,534 ----
  
      check-cast v3, Ljava/util/HashSet;
  
+     .line 226
      .local v3, thread:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      if-eqz v3, :cond_1
  
+     .line 227
      invoke-virtual {v3}, Ljava/util/HashSet;->iterator()Ljava/util/Iterator;
  
      move-result-object v1
***************
*** 499,504 ****
--- 548,554 ----
  
      check-cast v2, Landroid/net/Uri;
  
+     .line 228
      .local v2, key:Landroid/net/Uri;
      invoke-super {p0, v2}, Lcom/google/android/mms/util/AbstractCache;->purge(Ljava/lang/Object;)Ljava/lang/Object;
  
***************
*** 506,518 ****
--- 556,571 ----
  
      check-cast v0, Lcom/google/android/mms/util/PduCacheEntry;
  
+     .line 229
      .local v0, entry:Lcom/google/android/mms/util/PduCacheEntry;
      if-eqz v0, :cond_0
  
+     .line 230
      invoke-direct {p0, v2, v0}, Lcom/google/android/mms/util/PduCache;->removeFromMessageBoxes(Landroid/net/Uri;Lcom/google/android/mms/util/PduCacheEntry;)V
  
      goto :goto_0
  
+     .line 234
      .end local v0           #entry:Lcom/google/android/mms/util/PduCacheEntry;
      .end local v1           #i$:Ljava/util/Iterator;
      .end local v2           #key:Landroid/net/Uri;
***************
*** 525,543 ****
--- 578,601 ----
      .parameter "key"
  
      .prologue
+     .line 149
      invoke-super {p0, p1}, Lcom/google/android/mms/util/AbstractCache;->purge(Ljava/lang/Object;)Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Lcom/google/android/mms/util/PduCacheEntry;
  
+     .line 150
      .local v0, entry:Lcom/google/android/mms/util/PduCacheEntry;
      if-eqz v0, :cond_0
  
+     .line 151
      invoke-direct {p0, p1, v0}, Lcom/google/android/mms/util/PduCache;->removeFromThreads(Landroid/net/Uri;Lcom/google/android/mms/util/PduCacheEntry;)V
  
+     .line 152
      invoke-direct {p0, p1, v0}, Lcom/google/android/mms/util/PduCache;->removeFromMessageBoxes(Landroid/net/Uri;Lcom/google/android/mms/util/PduCacheEntry;)V
  
+     .line 155
      .end local v0           #entry:Lcom/google/android/mms/util/PduCacheEntry;
      :goto_0
      return-object v0
***************
*** 555,560 ****
--- 613,619 ----
      .parameter "entry"
  
      .prologue
+     .line 237
      iget-object v1, p0, Lcom/google/android/mms/util/PduCache;->mThreads:Ljava/util/HashMap;
  
      invoke-virtual {p2}, Lcom/google/android/mms/util/PduCacheEntry;->getMessageBox()I
***************
*** 573,583 ****
--- 632,645 ----
  
      check-cast v0, Ljava/util/HashSet;
  
+     .line 238
      .local v0, msgBox:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      if-eqz v0, :cond_0
  
+     .line 239
      invoke-virtual {v0, p1}, Ljava/util/HashSet;->remove(Ljava/lang/Object;)Z
  
+     .line 241
      :cond_0
      return-void
  .end method
***************
*** 588,593 ****
--- 650,656 ----
      .parameter "entry"
  
      .prologue
+     .line 214
      iget-object v1, p0, Lcom/google/android/mms/util/PduCache;->mThreads:Ljava/util/HashMap;
  
      invoke-virtual {p2}, Lcom/google/android/mms/util/PduCacheEntry;->getThreadId()J
***************
*** 604,614 ****
--- 667,680 ----
  
      check-cast v0, Ljava/util/HashSet;
  
+     .line 215
      .local v0, thread:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      if-eqz v0, :cond_0
  
+     .line 216
      invoke-virtual {v0, p1}, Ljava/util/HashSet;->remove(Ljava/lang/Object;)Z
  
+     .line 218
      :cond_0
      return-void
  .end method
***************
*** 622,627 ****
--- 688,694 ----
      .prologue
      const/4 v3, 0x0
  
+     .line 119
      monitor-enter p0
  
      :try_start_0
***************
*** 633,648 ****
--- 700,718 ----
  
      move-result v0
  
+     .line 120
      .local v0, match:I
      packed-switch v0, :pswitch_data_0
  
      move-object v2, v3
  
+     .line 144
      :goto_0
      monitor-exit p0
  
      return-object v2
  
+     .line 122
      :pswitch_0
      :try_start_1
      invoke-direct {p0, p1}, Lcom/google/android/mms/util/PduCache;->purgeSingleEntry(Landroid/net/Uri;)Lcom/google/android/mms/util/PduCacheEntry;
***************
*** 651,661 ****
--- 721,733 ----
  
      goto :goto_0
  
+     .line 127
      :pswitch_1
      invoke-virtual {p1}, Landroid/net/Uri;->getLastPathSegment()Ljava/lang/String;
  
      move-result-object v1
  
+     .line 128
      .local v1, msgId:Ljava/lang/String;
      sget-object v2, Landroid/provider/Telephony$Mms;->CONTENT_URI:Landroid/net/Uri;
  
***************
*** 669,682 ****
--- 741,757 ----
  
      goto :goto_0
  
+     .line 132
      .end local v1           #msgId:Ljava/lang/String;
      :pswitch_2
      invoke-virtual {p0}, Lcom/google/android/mms/util/PduCache;->purgeAll()V
  
      move-object v2, v3
  
+     .line 133
      goto :goto_0
  
+     .line 138
      :pswitch_3
      sget-object v2, Lcom/google/android/mms/util/PduCache;->MATCH_TO_MSGBOX_ID_MAP:Ljava/util/HashMap;
  
***************
*** 694,701 ****
--- 769,778 ----
  
      move-object v2, v3
  
+     .line 139
      goto :goto_0
  
+     .line 141
      :pswitch_4
      invoke-static {p1}, Landroid/content/ContentUris;->parseId(Landroid/net/Uri;)J
  
***************
*** 707,714 ****
--- 784,793 ----
  
      move-object v2, v3
  
+     .line 142
      goto :goto_0
  
+     .line 119
      .end local v0           #match:I
      :catchall_0
      move-exception v2
***************
*** 717,722 ****
--- 796,802 ----
  
      throw v2
  
+     .line 120
      :pswitch_data_0
      .packed-switch 0x0
          :pswitch_2
***************
*** 739,744 ****
--- 819,825 ----
      .parameter "x0"
  
      .prologue
+     .line 29
      check-cast p1, Landroid/net/Uri;
  
      .end local p1
***************
*** 753,777 ****
--- 834,863 ----
      .locals 1
  
      .prologue
+     .line 160
      monitor-enter p0
  
      :try_start_0
      invoke-super {p0}, Lcom/google/android/mms/util/AbstractCache;->purgeAll()V
  
+     .line 162
      iget-object v0, p0, Lcom/google/android/mms/util/PduCache;->mMessageBoxes:Ljava/util/HashMap;
  
      invoke-virtual {v0}, Ljava/util/HashMap;->clear()V
  
+     .line 163
      iget-object v0, p0, Lcom/google/android/mms/util/PduCache;->mThreads:Ljava/util/HashMap;
  
      invoke-virtual {v0}, Ljava/util/HashMap;->clear()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 164
      monitor-exit p0
  
      return-void
  
+     .line 160
      :catchall_0
      move-exception v0
  
***************
*** 786,791 ****
--- 872,878 ----
      .parameter "entry"
  
      .prologue
+     .line 94
      monitor-enter p0
  
      :try_start_0
***************
*** 793,798 ****
--- 880,886 ----
  
      move-result v2
  
+     .line 95
      .local v2, msgBoxId:I
      iget-object v7, p0, Lcom/google/android/mms/util/PduCache;->mMessageBoxes:Ljava/util/HashMap;
  
***************
*** 806,819 ****
--- 894,910 ----
  
      check-cast v1, Ljava/util/HashSet;
  
+     .line 96
      .local v1, msgBox:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      if-nez v1, :cond_0
  
+     .line 97
      new-instance v1, Ljava/util/HashSet;
  
      .end local v1           #msgBox:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      invoke-direct {v1}, Ljava/util/HashSet;-><init>()V
  
+     .line 98
      .restart local v1       #msgBox:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      iget-object v7, p0, Lcom/google/android/mms/util/PduCache;->mMessageBoxes:Ljava/util/HashMap;
  
***************
*** 823,833 ****
--- 914,926 ----
  
      invoke-virtual {v7, v8, v1}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 101
      :cond_0
      invoke-virtual {p2}, Lcom/google/android/mms/util/PduCacheEntry;->getThreadId()J
  
      move-result-wide v5
  
+     .line 102
      .local v5, threadId:J
      iget-object v7, p0, Lcom/google/android/mms/util/PduCache;->mThreads:Ljava/util/HashMap;
  
***************
*** 841,854 ****
--- 934,950 ----
  
      check-cast v4, Ljava/util/HashSet;
  
+     .line 103
      .local v4, thread:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      if-nez v4, :cond_1
  
+     .line 104
      new-instance v4, Ljava/util/HashSet;
  
      .end local v4           #thread:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      invoke-direct {v4}, Ljava/util/HashSet;-><init>()V
  
+     .line 105
      .restart local v4       #thread:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      iget-object v7, p0, Lcom/google/android/mms/util/PduCache;->mThreads:Ljava/util/HashMap;
  
***************
*** 858,887 ****
--- 954,990 ----
  
      invoke-virtual {v7, v8, v4}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 108
      :cond_1
      invoke-direct {p0, p1}, Lcom/google/android/mms/util/PduCache;->normalizeKey(Landroid/net/Uri;)Landroid/net/Uri;
  
      move-result-object v0
  
+     .line 109
      .local v0, finalKey:Landroid/net/Uri;
      invoke-super {p0, v0, p2}, Lcom/google/android/mms/util/AbstractCache;->put(Ljava/lang/Object;Ljava/lang/Object;)Z
  
      move-result v3
  
+     .line 110
      .local v3, result:Z
      if-eqz v3, :cond_2
  
+     .line 111
      invoke-virtual {v1, v0}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
  
+     .line 112
      invoke-virtual {v4, v0}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
+     .line 114
      :cond_2
      monitor-exit p0
  
      return v3
  
+     .line 94
      .end local v0           #finalKey:Landroid/net/Uri;
      .end local v1           #msgBox:Ljava/util/HashSet;,"Ljava/util/HashSet<Landroid/net/Uri;>;"
      .end local v2           #msgBoxId:I
***************
*** 902,907 ****
--- 1005,1011 ----
      .parameter "x1"
  
      .prologue
+     .line 29
      check-cast p1, Landroid/net/Uri;
  
      .end local p1
