*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/dhcp/DhcpNakPacket.smali	2012-03-23 19:36:52.804643875 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/android/net/dhcp/DhcpNakPacket.smali.original	2012-03-23 19:34:45.383320351 -0500
***************
*** 14,19 ****
--- 14,20 ----
      .parameter "clientMac"
  
      .prologue
+     .line 33
      sget-object v2, Ljava/net/Inet4Address;->ANY:Ljava/net/InetAddress;
  
      sget-object v3, Ljava/net/Inet4Address;->ANY:Ljava/net/InetAddress;
***************
*** 32,37 ****
--- 33,39 ----
  
      invoke-direct/range {v0 .. v7}, Landroid/net/dhcp/DhcpPacket;-><init>(ILjava/net/InetAddress;Ljava/net/InetAddress;Ljava/net/InetAddress;Ljava/net/InetAddress;[BZ)V
  
+     .line 35
      return-void
  .end method
  
***************
*** 44,61 ****
--- 46,67 ----
      .parameter "srcUdp"
  
      .prologue
+     .line 46
      const/16 v0, 0x5dc
  
      invoke-static {v0}, Ljava/nio/ByteBuffer;->allocate(I)Ljava/nio/ByteBuffer;
  
      move-result-object v6
  
+     .line 47
      .local v6, result:Ljava/nio/ByteBuffer;
      iget-object v2, p0, Landroid/net/dhcp/DhcpNakPacket;->mClientIp:Ljava/net/InetAddress;
  
+     .line 48
      .local v2, destIp:Ljava/net/InetAddress;
      iget-object v3, p0, Landroid/net/dhcp/DhcpNakPacket;->mYourIp:Ljava/net/InetAddress;
  
+     .line 50
      .local v3, srcIp:Ljava/net/InetAddress;
      const/4 v7, 0x2
  
***************
*** 71,78 ****
--- 77,86 ----
  
      invoke-virtual/range {v0 .. v8}, Landroid/net/dhcp/DhcpNakPacket;->fillInPacket(ILjava/net/InetAddress;Ljava/net/InetAddress;SSLjava/nio/ByteBuffer;BZ)V
  
+     .line 52
      invoke-virtual {v6}, Ljava/nio/ByteBuffer;->flip()Ljava/nio/Buffer;
  
+     .line 53
      return-object v6
  .end method
  
***************
*** 81,88 ****
--- 89,98 ----
      .parameter "machine"
  
      .prologue
+     .line 70
      invoke-interface {p1}, Landroid/net/dhcp/DhcpStateMachine;->onNakReceived()V
  
+     .line 71
      return-void
  .end method
  
***************
*** 91,116 ****
--- 101,131 ----
      .parameter "buffer"
  
      .prologue
+     .line 60
      const/16 v0, 0x35
  
      const/4 v1, 0x6
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/net/dhcp/DhcpNakPacket;->addTlv(Ljava/nio/ByteBuffer;BB)V
  
+     .line 61
      const/16 v0, 0x36
  
      iget-object v1, p0, Landroid/net/dhcp/DhcpNakPacket;->mServerIdentifier:Ljava/net/InetAddress;
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/net/dhcp/DhcpNakPacket;->addTlv(Ljava/nio/ByteBuffer;BLjava/net/InetAddress;)V
  
+     .line 62
      const/16 v0, 0x38
  
      iget-object v1, p0, Landroid/net/dhcp/DhcpNakPacket;->mMessage:Ljava/lang/String;
  
      invoke-virtual {p0, p1, v0, v1}, Landroid/net/dhcp/DhcpNakPacket;->addTlv(Ljava/nio/ByteBuffer;BLjava/lang/String;)V
  
+     .line 63
      invoke-virtual {p0, p1}, Landroid/net/dhcp/DhcpNakPacket;->addTlvEnd(Ljava/nio/ByteBuffer;)V
  
+     .line 64
      return-void
  .end method
  
***************
*** 118,127 ****
--- 133,144 ----
      .locals 3
  
      .prologue
+     .line 38
      invoke-super {p0}, Landroid/net/dhcp/DhcpPacket;->toString()Ljava/lang/String;
  
      move-result-object v0
  
+     .line 39
      .local v0, s:Ljava/lang/String;
      new-instance v1, Ljava/lang/StringBuilder;
  
