*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/multiwaveview/Ease$Quint$2.smali	2012-03-23 19:37:31.876356346 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/multiwaveview/Ease$Quint$2.smali.original	2012-03-23 19:34:46.066332805 -0500
***************
*** 22,27 ****
--- 22,28 ----
      .locals 0
  
      .prologue
+     .line 100
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      return-void
***************
*** 36,41 ****
--- 37,43 ----
      .prologue
      const/high16 v1, 0x3f80
  
+     .line 102
      div-float v0, p1, v1
  
      sub-float p1, v0, v1
