*** /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/multiwaveview/Tweener.smali	2012-03-23 19:37:32.219362602 -0500
--- /home/dhacker29/patchrom/spyder/temp/dst_smali_orig/framework.jar.out/smali/com/android/internal/widget/multiwaveview/Tweener.smali.original	2012-03-23 19:34:46.070332879 -0500
***************
*** 32,43 ****
--- 32,45 ----
      .locals 1
  
      .prologue
+     .line 38
      new-instance v0, Ljava/util/HashMap;
  
      invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
  
      sput-object v0, Lcom/android/internal/widget/multiwaveview/Tweener;->sTweens:Ljava/util/HashMap;
  
+     .line 138
      new-instance v0, Lcom/android/internal/widget/multiwaveview/Tweener$1;
  
      invoke-direct {v0}, Lcom/android/internal/widget/multiwaveview/Tweener$1;-><init>()V
***************
*** 52,61 ****
--- 54,66 ----
      .parameter "anim"
  
      .prologue
+     .line 40
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     .line 41
      iput-object p1, p0, Lcom/android/internal/widget/multiwaveview/Tweener;->animator:Landroid/animation/ObjectAnimator;
  
+     .line 42
      return-void
  .end method
  
***************
*** 64,69 ****
--- 69,75 ----
      .parameter "x0"
  
      .prologue
+     .line 33
      invoke-static {p0}, Lcom/android/internal/widget/multiwaveview/Tweener;->remove(Landroid/animation/Animator;)V
  
      return-void
***************
*** 74,79 ****
--- 80,86 ----
      .parameter "animator"
  
      .prologue
+     .line 45
      sget-object v2, Lcom/android/internal/widget/multiwaveview/Tweener;->sTweens:Ljava/util/HashMap;
  
      invoke-virtual {v2}, Ljava/util/HashMap;->entrySet()Ljava/util/Set;
***************
*** 84,89 ****
--- 91,97 ----
  
      move-result-object v1
  
+     .line 46
      .local v1, iter:Ljava/util/Iterator;,"Ljava/util/Iterator<Ljava/util/Map$Entry<Ljava/lang/Object;Lcom/android/internal/widget/multiwaveview/Tweener;>;>;"
      :cond_0
      invoke-interface {v1}, Ljava/util/Iterator;->hasNext()Z
***************
*** 92,103 ****
--- 100,113 ----
  
      if-eqz v2, :cond_1
  
+     .line 47
      invoke-interface {v1}, Ljava/util/Iterator;->next()Ljava/lang/Object;
  
      move-result-object v0
  
      check-cast v0, Ljava/util/Map$Entry;
  
+     .line 48
      .local v0, entry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/Object;Lcom/android/internal/widget/multiwaveview/Tweener;>;"
      invoke-interface {v0}, Ljava/util/Map$Entry;->getValue()Ljava/lang/Object;
  
***************
*** 109,116 ****
--- 119,128 ----
  
      if-ne v2, p0, :cond_0
  
+     .line 51
      invoke-interface {v1}, Ljava/util/Iterator;->remove()V
  
+     .line 55
      .end local v0           #entry:Ljava/util/Map$Entry;,"Ljava/util/Map$Entry<Ljava/lang/Object;Lcom/android/internal/widget/multiwaveview/Tweener;>;"
      :cond_1
      return-void
***************
*** 133,138 ****
--- 145,151 ----
      .end annotation
  
      .prologue
+     .line 162
      .local p0, props:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/animation/PropertyValuesHolder;>;"
      move-object v0, p1
  
***************
*** 148,153 ****
--- 161,167 ----
  
      aget-object v2, v0, v1
  
+     .line 163
      .local v2, killobject:Ljava/lang/Object;
      sget-object v5, Lcom/android/internal/widget/multiwaveview/Tweener;->sTweens:Ljava/util/HashMap;
  
***************
*** 157,171 ****
--- 171,189 ----
  
      check-cast v4, Lcom/android/internal/widget/multiwaveview/Tweener;
  
+     .line 164
      .local v4, tween:Lcom/android/internal/widget/multiwaveview/Tweener;
      if-eqz v4, :cond_0
  
+     .line 165
      iget-object v5, v4, Lcom/android/internal/widget/multiwaveview/Tweener;->animator:Landroid/animation/ObjectAnimator;
  
      invoke-virtual {v5}, Landroid/animation/ObjectAnimator;->cancel()V
  
+     .line 166
      if-eqz p0, :cond_1
  
+     .line 167
      iget-object v6, v4, Lcom/android/internal/widget/multiwaveview/Tweener;->animator:Landroid/animation/ObjectAnimator;
  
      invoke-virtual {p0}, Ljava/util/ArrayList;->size()I
***************
*** 182,193 ****
--- 200,213 ----
  
      invoke-virtual {v6, v5}, Landroid/animation/ObjectAnimator;->setValues([Landroid/animation/PropertyValuesHolder;)V
  
+     .line 162
      :cond_0
      :goto_1
      add-int/lit8 v1, v1, 0x1
  
      goto :goto_0
  
+     .line 170
      :cond_1
      sget-object v5, Lcom/android/internal/widget/multiwaveview/Tweener;->sTweens:Ljava/util/HashMap;
  
***************
*** 195,200 ****
--- 215,221 ----
  
      goto :goto_1
  
+     .line 174
      .end local v2           #killobject:Ljava/lang/Object;
      .end local v4           #tween:Lcom/android/internal/widget/multiwaveview/Tweener;
      :cond_2
***************
*** 205,214 ****
--- 226,237 ----
      .locals 1
  
      .prologue
+     .line 158
      sget-object v0, Lcom/android/internal/widget/multiwaveview/Tweener;->sTweens:Ljava/util/HashMap;
  
      invoke-virtual {v0}, Ljava/util/HashMap;->clear()V
  
+     .line 159
      return-void
  .end method
  
***************
*** 219,235 ****
--- 242,263 ----
      .parameter "vars"
  
      .prologue
+     .line 58
      const-wide/16 v3, 0x0
  
+     .line 59
      .local v3, delay:J
      const/4 v12, 0x0
  
+     .line 60
      .local v12, updateListener:Landroid/animation/ValueAnimator$AnimatorUpdateListener;
      const/4 v9, 0x0
  
+     .line 61
      .local v9, listener:Landroid/animation/Animator$AnimatorListener;
      const/4 v7, 0x0
  
+     .line 64
      .local v7, interpolator:Landroid/animation/TimeInterpolator;
      new-instance v10, Ljava/util/ArrayList;
  
***************
*** 241,246 ****
--- 269,275 ----
  
      invoke-direct {v10, v14}, Ljava/util/ArrayList;-><init>(I)V
  
+     .line 65
      .local v10, props:Ljava/util/ArrayList;,"Ljava/util/ArrayList<Landroid/animation/PropertyValuesHolder;>;"
      const/4 v6, 0x0
  
***************
*** 252,263 ****
--- 281,294 ----
  
      if-ge v6, v14, :cond_b
  
+     .line 66
      aget-object v14, p3, v6
  
      instance-of v14, v14, Ljava/lang/String;
  
      if-nez v14, :cond_0
  
+     .line 67
      new-instance v14, Ljava/lang/IllegalArgumentException;
  
      new-instance v15, Ljava/lang/StringBuilder;
***************
*** 284,301 ****
  
      throw v14
  
      :cond_0
      aget-object v8, p3, v6
  
      check-cast v8, Ljava/lang/String;
  
      .local v8, key:Ljava/lang/String;
      add-int/lit8 v14, v6, 0x1
  
      aget-object v13, p3, v14
  
      .local v13, value:Ljava/lang/Object;
!     const-string v14, "simultaneousTween"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 315,335 ----
  
      throw v14
  
+     .line 69
      :cond_0
      aget-object v8, p3, v6
  
      check-cast v8, Ljava/lang/String;
  
+     .line 70
      .local v8, key:Ljava/lang/String;
      add-int/lit8 v14, v6, 0x1
  
      aget-object v13, p3, v14
  
+     .line 71
      .local v13, value:Ljava/lang/Object;
!     const-string/jumbo v14, "simultaneousTween"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 303,308 ****
--- 337,343 ----
  
      if-eqz v14, :cond_2
  
+     .line 65
      .end local v13           #value:Ljava/lang/Object;
      :cond_1
      :goto_1
***************
*** 310,315 ****
--- 345,351 ----
  
      goto :goto_0
  
+     .line 73
      .restart local v13       #value:Ljava/lang/Object;
      :cond_2
      const-string v14, "ease"
***************
*** 322,333 ****
  
      move-object v7, v13
  
      check-cast v7, Landroid/animation/TimeInterpolator;
  
      goto :goto_1
  
      :cond_3
!     const-string v14, "onUpdate"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 358,371 ----
  
      move-object v7, v13
  
+     .line 74
      check-cast v7, Landroid/animation/TimeInterpolator;
  
      goto :goto_1
  
+     .line 75
      :cond_3
!     const-string/jumbo v14, "onUpdate"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 335,341 ****
  
      if-nez v14, :cond_4
  
!     const-string v14, "onUpdateListener"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 373,379 ----
  
      if-nez v14, :cond_4
  
!     const-string/jumbo v14, "onUpdateListener"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 346,357 ****
      :cond_4
      move-object v12, v13
  
      check-cast v12, Landroid/animation/ValueAnimator$AnimatorUpdateListener;
  
      goto :goto_1
  
      :cond_5
!     const-string v14, "onComplete"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 384,397 ----
      :cond_4
      move-object v12, v13
  
+     .line 76
      check-cast v12, Landroid/animation/ValueAnimator$AnimatorUpdateListener;
  
      goto :goto_1
  
+     .line 77
      :cond_5
!     const-string/jumbo v14, "onComplete"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 359,365 ****
  
      if-nez v14, :cond_6
  
!     const-string v14, "onCompleteListener"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 399,405 ----
  
      if-nez v14, :cond_6
  
!     const-string/jumbo v14, "onCompleteListener"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 370,379 ****
--- 410,421 ----
      :cond_6
      move-object v9, v13
  
+     .line 78
      check-cast v9, Landroid/animation/Animator$AnimatorListener;
  
      goto :goto_1
  
+     .line 79
      :cond_7
      const-string v14, "delay"
  
***************
*** 383,388 ****
--- 425,431 ----
  
      if-eqz v14, :cond_8
  
+     .line 80
      check-cast v13, Ljava/lang/Number;
  
      .end local v13           #value:Ljava/lang/Object;
***************
*** 392,400 ****
  
      goto :goto_1
  
      .restart local v13       #value:Ljava/lang/Object;
      :cond_8
!     const-string v14, "syncWith"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
--- 435,444 ----
  
      goto :goto_1
  
+     .line 81
      .restart local v13       #value:Ljava/lang/Object;
      :cond_8
!     const-string/jumbo v14, "syncWith"
  
      invoke-virtual {v14, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
  
***************
*** 402,411 ****
--- 446,457 ----
  
      if-nez v14, :cond_1
  
+     .line 83
      instance-of v14, v13, [F
  
      if-eqz v14, :cond_9
  
+     .line 84
      const/4 v14, 0x2
  
      new-array v15, v14, [F
***************
*** 445,456 ****
--- 491,504 ----
  
      goto :goto_1
  
+     .line 86
      .restart local v13       #value:Ljava/lang/Object;
      :cond_9
      instance-of v14, v13, Ljava/lang/Number;
  
      if-eqz v14, :cond_a
  
+     .line 87
      check-cast v13, Ljava/lang/Number;
  
      .end local v13           #value:Ljava/lang/Object;
***************
*** 458,463 ****
--- 506,512 ----
  
      move-result v5
  
+     .line 88
      .local v5, floatValue:F
      const/4 v14, 0x1
  
***************
*** 475,480 ****
--- 524,530 ----
  
      goto/16 :goto_1
  
+     .line 90
      .end local v5           #floatValue:F
      .restart local v13       #value:Ljava/lang/Object;
      :cond_a
***************
*** 516,521 ****
--- 566,572 ----
  
      throw v14
  
+     .line 96
      .end local v8           #key:Ljava/lang/String;
      .end local v13           #value:Ljava/lang/Object;
      :cond_b
***************
*** 529,540 ****
--- 580,594 ----
  
      check-cast v11, Lcom/android/internal/widget/multiwaveview/Tweener;
  
+     .line 97
      .local v11, tween:Lcom/android/internal/widget/multiwaveview/Tweener;
      const/4 v2, 0x0
  
+     .line 98
      .local v2, anim:Landroid/animation/ObjectAnimator;
      if-nez v11, :cond_f
  
+     .line 99
      invoke-virtual {v10}, Ljava/util/ArrayList;->size()I
  
      move-result v14
***************
*** 553,563 ****
--- 607,619 ----
  
      move-result-object v2
  
+     .line 101
      new-instance v11, Lcom/android/internal/widget/multiwaveview/Tweener;
  
      .end local v11           #tween:Lcom/android/internal/widget/multiwaveview/Tweener;
      invoke-direct {v11, v2}, Lcom/android/internal/widget/multiwaveview/Tweener;-><init>(Landroid/animation/ObjectAnimator;)V
  
+     .line 102
      .restart local v11       #tween:Lcom/android/internal/widget/multiwaveview/Tweener;
      sget-object v14, Lcom/android/internal/widget/multiwaveview/Tweener;->sTweens:Ljava/util/HashMap;
  
***************
*** 565,604 ****
--- 621,674 ----
  
      invoke-virtual {v14, v0, v11}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
  
+     .line 109
      :goto_2
      if-eqz v7, :cond_c
  
+     .line 110
      invoke-virtual {v2, v7}, Landroid/animation/ObjectAnimator;->setInterpolator(Landroid/animation/TimeInterpolator;)V
  
+     .line 114
      :cond_c
      invoke-virtual {v2, v3, v4}, Landroid/animation/ObjectAnimator;->setStartDelay(J)V
  
+     .line 115
      move-wide/from16 v0, p1
  
      invoke-virtual {v2, v0, v1}, Landroid/animation/ObjectAnimator;->setDuration(J)Landroid/animation/ObjectAnimator;
  
+     .line 116
      if-eqz v12, :cond_d
  
+     .line 117
      invoke-virtual {v2}, Landroid/animation/ObjectAnimator;->removeAllUpdateListeners()V
  
+     .line 118
      invoke-virtual {v2, v12}, Landroid/animation/ObjectAnimator;->addUpdateListener(Landroid/animation/ValueAnimator$AnimatorUpdateListener;)V
  
+     .line 120
      :cond_d
      if-eqz v9, :cond_e
  
+     .line 121
      invoke-virtual {v2}, Landroid/animation/ObjectAnimator;->removeAllListeners()V
  
+     .line 122
      invoke-virtual {v2, v9}, Landroid/animation/ObjectAnimator;->addListener(Landroid/animation/Animator$AnimatorListener;)V
  
+     .line 124
      :cond_e
      sget-object v14, Lcom/android/internal/widget/multiwaveview/Tweener;->mCleanupListener:Landroid/animation/Animator$AnimatorListener;
  
      invoke-virtual {v2, v14}, Landroid/animation/ObjectAnimator;->addListener(Landroid/animation/Animator$AnimatorListener;)V
  
+     .line 125
      invoke-virtual {v2}, Landroid/animation/ObjectAnimator;->start()V
  
+     .line 127
      return-object v11
  
+     .line 105
      :cond_f
      sget-object v14, Lcom/android/internal/widget/multiwaveview/Tweener;->sTweens:Ljava/util/HashMap;
  
***************
*** 612,617 ****
--- 682,688 ----
  
      iget-object v2, v14, Lcom/android/internal/widget/multiwaveview/Tweener;->animator:Landroid/animation/ObjectAnimator;
  
+     .line 106
      const/4 v14, 0x1
  
      new-array v14, v14, [Ljava/lang/Object;
***************
*** 634,639 ****
--- 705,711 ----
      .parameter "vars"
  
      .prologue
+     .line 134
      invoke-static {p1, p2, p3, p4}, Lcom/android/internal/widget/multiwaveview/Tweener;->to(Ljava/lang/Object;J[Ljava/lang/Object;)Lcom/android/internal/widget/multiwaveview/Tweener;
  
      move-result-object v0
